// @wix/headless-restaurants-olo/dist/react/core/ItemDetails.js
import { jsx as _jsx } from "react/jsx-runtime";
import { useState } from 'react';
import { useService, WixServices } from '@wix/services-manager-react';
import { createServicesMap } from '@wix/services-manager';
import { ItemService, ItemServiceDefinition, loadItemServiceConfig, } from '../../services/item-details-service.js';
import { OLOSettingsServiceDefinition } from '../../services/olo-settings-service.js';
export const Root = ({ children, itemDetailsServiceConfig, }) => {
    const service = useService(OLOSettingsServiceDefinition);
    const selectedItem = service.selectedItem?.get();
    let config = itemDetailsServiceConfig;
    if (!config) {
        config = loadItemServiceConfig({
            item: selectedItem,
            operationId: service.operation?.get()?._id ?? '',
        });
    }
    if (config.item) {
        service.selectedItem?.set(config.item);
    }
    return (_jsx(WixServices, { servicesMap: createServicesMap().addService(ItemServiceDefinition, ItemService, config), children: children({ item: itemDetailsServiceConfig?.item ?? selectedItem }) }));
};
export const ModifiersRepeater = ({ children, }) => {
    const service = useService(ItemServiceDefinition);
    const item = service.item?.get();
    // TODO: Check if modifiers exist on item type - might be in a different property
    const modifiers = item?.modifiers || [];
    const hasModifiers = modifiers.length > 0;
    return children({ modifiers, hasModifiers });
};
export const VariantsRepeater = ({ children, }) => {
    const service = useService(ItemServiceDefinition);
    const item = service.item?.get();
    // TODO: Check if variants exist on item type - might be in a different property
    const variants = item?.variants || [];
    const hasVariants = variants.length > 0;
    return children({ variants, hasVariants });
};
export const SpecialRequest = ({ children, }) => {
    const [value, setValue] = useState('');
    const service = useService(ItemServiceDefinition);
    const onChange = (newValue) => {
        setValue(newValue);
        service.updateSpecialRequest(newValue);
    };
    return children({
        value,
        onChange,
        // placeholder: 'Any special requests or dietary restrictions?',
        // maxLength: 200
    });
};
export const LineItemComponent = ({ children, }) => {
    const service = useService(ItemServiceDefinition);
    const lineItem = service.lineItem?.get?.() ?? {};
    return children({ lineItem });
};
export const QuantityComponent = ({ children, }) => {
    const service = useService(ItemServiceDefinition);
    const quantity = service.quantity?.get?.() ?? 1;
    const increment = () => service.quantity?.set?.(quantity + 1);
    const decrement = () => service.quantity?.set?.(quantity - 1);
    const setQuantity = (quantity) => {
        service.updateQuantity?.(quantity);
    };
    const canIncrement = true;
    const canDecrement = quantity > 1;
    const onValueChange = (value) => {
        service.updateQuantity?.(value);
    };
    return children({
        quantity,
        increment,
        decrement,
        setQuantity,
        canIncrement,
        canDecrement,
        onValueChange,
    });
};
