// @wix/design-system/dist/esm/AnalyticsSummaryCard/AnalyticsSummaryCard.st.css.js

import "../Foundation/stylable/colors.st.css.js";
import "../common/Focusable/Focusable.st.css.js";

import { classesRuntime, statesRuntime } from "./../stylable-esm-runtime.js";



var _namespace_ = "wds_1_226_0_AnalyticsSummaryCard";
var _style_ = classesRuntime.bind(null, _namespace_);

export var cssStates = statesRuntime.bind(null, _namespace_);
export var style = _style_;
export var st = _style_;

export var namespace = _namespace_;
export var classes = {"root":"wds_1_226_0_AnalyticsSummaryCard__root","title":"wds_1_226_0_AnalyticsSummaryCard__title","valueLineContainer":"wds_1_226_0_AnalyticsSummaryCard__valueLineContainer","valueAndTrend":"wds_1_226_0_AnalyticsSummaryCard__valueAndTrend","trend":"wds_1_226_0_AnalyticsSummaryCard__trend","loader":"wds_1_226_0_AnalyticsSummaryCard__loader","ctaButton":"wds_1_226_0_AnalyticsSummaryCard__ctaButton","footer":"wds_1_226_0_AnalyticsSummaryCard__footer","sparkline":"wds_1_226_0_AnalyticsSummaryCard__sparkline"};
export var keyframes = {}; 
export var layers = {};
export var containers = {};
export var stVars = {}; 
export var vars = {"wds-space-300":"--wds-space-300","wds-space-400":"--wds-space-400","wds-space-500":"--wds-space-500","wds-border-radius-300":"--wds-border-radius-300","wds-color-border-standard-secondary":"--wds-color-border-standard-secondary","wds-color-border-standard-secondary-hover":"--wds-color-border-standard-secondary-hover"};
