// @wix/auto_sdk_comments_widgets/build/internal/es/index.mjs
// src/comments-categories-v1-comments-widget-widgets.public.ts
import { renameKeysFromRESTResponseToSDKResponse as renameKeysFromRESTResponseToSDKResponse2 } from "@wix/sdk-runtime/rename-all-nested-keys";
import { transformRESTPageURLV2ToSDKPageURLV2 as transformRESTPageURLV2ToSDKPageURLV22 } from "@wix/sdk-runtime/transformations/page-url-v2";
import { transformRESTTimestampToSDKTimestamp as transformRESTTimestampToSDKTimestamp2 } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths as transformPaths3 } from "@wix/sdk-runtime/transformations/transform-paths";
import { EventDefinition } from "@wix/sdk-types";

// src/comments-categories-v1-comments-widget-widgets.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/comments-categories-v1-comments-widget-widgets.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixCommentsCategoriesV1CommentsWidgetRegistryUrl(opts) {
  const domainToMappings = {
    _: [
      {
        srcPath: "/_api/comments-widget-registry",
        destPath: "/v1/comments/widgets"
      }
    ],
    "dev._base_domain_": [
      {
        srcPath: "/_api/comments-widget-registry/v1/comments/widgets",
        destPath: "/v1/comments/widgets"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/comments-widget-registry/v1/comments/widgets",
        destPath: "/v1/comments/widgets"
      },
      {
        srcPath: "/_api/comments/widgets/v1",
        destPath: "/v1/comments/widgets"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/comments-widget-registry/v1/comments/widgets",
        destPath: "/v1/comments/widgets"
      },
      {
        srcPath: "/_api/comments/widgets/v1",
        destPath: "/v1/comments/widgets"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/comments-widget-registry/v1/comments/widgets",
        destPath: "/v1/comments/widgets"
      },
      {
        srcPath: "/_api/comments/widgets/v1",
        destPath: "/v1/comments/widgets"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/comments-widget-registry/v1/comments/widgets",
        destPath: "/v1/comments/widgets"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/comments-widget-registry/v1/comments/widgets",
        destPath: "/v1/comments/widgets"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/api/v1/comments/widgets",
        destPath: "/v1/comments/widgets"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/comments-widget-registry",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_comments_widgets";
function registerCommentsWidget(payload) {
  function __registerCommentsWidget({ host }) {
    const metadata = {
      entityFqdn: "wix.comments.categories.v1.comments_widget",
      method: "POST",
      methodFqn: "wix.comments.categories.v1.CommentsWidgetRegistry.RegisterCommentsWidget",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixCommentsCategoriesV1CommentsWidgetRegistryUrl({
        protoPath: "/v1/comments/widgets/register",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "commentsWidget.createdDate" },
            { path: "commentsWidget.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __registerCommentsWidget;
}
function queryCommentsWidgetResources(payload) {
  function __queryCommentsWidgetResources({ host }) {
    const metadata = {
      entityFqdn: "wix.comments.categories.v1.comments_widget",
      method: "POST",
      methodFqn: "wix.comments.categories.v1.CommentsWidgetRegistry.QueryCommentsWidgetResources",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixCommentsCategoriesV1CommentsWidgetRegistryUrl({
        protoPath: "/v1/comments/widgets/resources/query",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __queryCommentsWidgetResources;
}
function createCommentsWidget(payload) {
  function __createCommentsWidget({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "commentsWidget.createdDate" },
          { path: "commentsWidget.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.comments.categories.v1.comments_widget",
      method: "POST",
      methodFqn: "wix.comments.categories.v1.CommentsWidgetRegistry.CreateCommentsWidget",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixCommentsCategoriesV1CommentsWidgetRegistryUrl({
        protoPath: "/v1/comments/widgets",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "commentsWidget.createdDate" },
            { path: "commentsWidget.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createCommentsWidget;
}
function getCommentsWidget(payload) {
  function __getCommentsWidget({ host }) {
    const metadata = {
      entityFqdn: "wix.comments.categories.v1.comments_widget",
      method: "GET",
      methodFqn: "wix.comments.categories.v1.CommentsWidgetRegistry.GetCommentsWidget",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixCommentsCategoriesV1CommentsWidgetRegistryUrl({
        protoPath: "/v1/comments/widgets/{commentsWidgetId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "commentsWidget.createdDate" },
            { path: "commentsWidget.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getCommentsWidget;
}
function updateCommentsWidget(payload) {
  function __updateCommentsWidget({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "commentsWidget.createdDate" },
          { path: "commentsWidget.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.comments.categories.v1.comments_widget",
      method: "PATCH",
      methodFqn: "wix.comments.categories.v1.CommentsWidgetRegistry.UpdateCommentsWidget",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixCommentsCategoriesV1CommentsWidgetRegistryUrl({
        protoPath: "/v1/comments/widgets/{commentsWidget.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "commentsWidget.createdDate" },
            { path: "commentsWidget.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateCommentsWidget;
}
function deleteCommentsWidget(payload) {
  function __deleteCommentsWidget({ host }) {
    const metadata = {
      entityFqdn: "wix.comments.categories.v1.comments_widget",
      method: "DELETE",
      methodFqn: "wix.comments.categories.v1.CommentsWidgetRegistry.DeleteCommentsWidget",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixCommentsCategoriesV1CommentsWidgetRegistryUrl({
        protoPath: "/v1/comments/widgets/{commentsWidgetId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteCommentsWidget;
}
function queryCommentsWidget(payload) {
  function __queryCommentsWidget({ host }) {
    const metadata = {
      entityFqdn: "wix.comments.categories.v1.comments_widget",
      method: "POST",
      methodFqn: "wix.comments.categories.v1.CommentsWidgetRegistry.QueryCommentsWidget",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixCommentsCategoriesV1CommentsWidgetRegistryUrl({
        protoPath: "/v1/comments/widgets/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "commentsWidgets.createdDate" },
            { path: "commentsWidgets.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryCommentsWidget;
}

// src/comments-categories-v1-comments-widget-widgets.universal.ts
import { transformSDKPageURLV2ToRESTPageURLV2 } from "@wix/sdk-runtime/transformations/page-url-v2";
import { transformRESTPageURLV2ToSDKPageURLV2 } from "@wix/sdk-runtime/transformations/page-url-v2";
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function registerCommentsWidget2(categoryId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    categoryId,
    resourceId: options?.resourceId,
    componentId: options?.componentId,
    dynamicPageTitle: options?.dynamicPageTitle,
    pageId: options?.pageId,
    urlPath: options?.urlPath
  });
  const reqOpts = registerCommentsWidget(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTPageURLV2ToSDKPageURLV2,
          paths: [{ path: "commentsWidget.urlPath" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          categoryId: "$[0]",
          resourceId: "$[1].resourceId",
          componentId: "$[1].componentId",
          dynamicPageTitle: "$[1].dynamicPageTitle",
          pageId: "$[1].pageId",
          urlPath: "$[1].urlPath"
        },
        singleArgumentUnchanged: false
      },
      ["categoryId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function queryCommentsWidgetResources2(query) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ query });
  const reqOpts = queryCommentsWidgetResources(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["query"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function createCommentsWidget2(commentsWidget) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({ commentsWidget }),
    [
      {
        transformFn: transformSDKPageURLV2ToRESTPageURLV2,
        paths: [{ path: "commentsWidget.urlPath" }]
      }
    ]
  );
  const reqOpts = createCommentsWidget(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTPageURLV2ToSDKPageURLV2,
          paths: [{ path: "commentsWidget.urlPath" }]
        }
      ])
    )?.commentsWidget;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { commentsWidget: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["commentsWidget"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getCommentsWidget2(commentsWidgetId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    commentsWidgetId
  });
  const reqOpts = getCommentsWidget(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTPageURLV2ToSDKPageURLV2,
          paths: [{ path: "commentsWidget.urlPath" }]
        }
      ])
    )?.commentsWidget;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { commentsWidgetId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["commentsWidgetId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateCommentsWidget2(_id, commentsWidget) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({
      commentsWidget: { ...commentsWidget, id: _id }
    }),
    [
      {
        transformFn: transformSDKPageURLV2ToRESTPageURLV2,
        paths: [{ path: "commentsWidget.urlPath" }]
      }
    ]
  );
  const reqOpts = updateCommentsWidget(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTPageURLV2ToSDKPageURLV2,
          paths: [{ path: "commentsWidget.urlPath" }]
        }
      ])
    )?.commentsWidget;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { commentsWidget: "$[1]" },
        explicitPathsToArguments: { "commentsWidget.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "commentsWidget"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteCommentsWidget2(commentsWidgetId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    commentsWidgetId,
    revision: options?.revision
  });
  const reqOpts = deleteCommentsWidget(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          commentsWidgetId: "$[0]",
          revision: "$[1].revision"
        },
        singleArgumentUnchanged: false
      },
      ["commentsWidgetId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryCommentsWidget2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryCommentsWidget(
        payload
      );
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [
          {
            transformFn: transformRESTPageURLV2ToSDKPageURLV2,
            paths: [{ path: "commentsWidgets.urlPath" }]
          }
        ])
      );
      return {
        items: transformedData?.commentsWidgets,
        pagingMetadata: transformedData?.metadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}

// src/comments-categories-v1-comments-widget-widgets.public.ts
function registerCommentsWidget3(httpClient) {
  return (categoryId, options) => registerCommentsWidget2(
    categoryId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function queryCommentsWidgetResources3(httpClient) {
  return (query) => queryCommentsWidgetResources2(
    query,
    // @ts-ignore
    { httpClient }
  );
}
function createCommentsWidget3(httpClient) {
  return (commentsWidget) => createCommentsWidget2(
    commentsWidget,
    // @ts-ignore
    { httpClient }
  );
}
function getCommentsWidget3(httpClient) {
  return (commentsWidgetId) => getCommentsWidget2(
    commentsWidgetId,
    // @ts-ignore
    { httpClient }
  );
}
function updateCommentsWidget3(httpClient) {
  return (_id, commentsWidget) => updateCommentsWidget2(
    _id,
    commentsWidget,
    // @ts-ignore
    { httpClient }
  );
}
function deleteCommentsWidget3(httpClient) {
  return (commentsWidgetId, options) => deleteCommentsWidget2(
    commentsWidgetId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function queryCommentsWidget3(httpClient) {
  return () => queryCommentsWidget2(
    // @ts-ignore
    { httpClient }
  );
}
var onCommentsWidgetCreated = EventDefinition(
  "wix.comments.categories.v1.comments_widget_created",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTPageURLV2ToSDKPageURLV22,
        paths: [{ path: "entity.urlPath" }]
      },
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onCommentsWidgetDeleted = EventDefinition(
  "wix.comments.categories.v1.comments_widget_deleted",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTPageURLV2ToSDKPageURLV22,
        paths: [{ path: "undefined.urlPath" }]
      },
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "undefined.createdDate" },
          { path: "undefined.updatedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onCommentsWidgetUpdated = EventDefinition(
  "wix.comments.categories.v1.comments_widget_updated",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTPageURLV2ToSDKPageURLV22,
        paths: [{ path: "entity.urlPath" }]
      },
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();

// src/comments-categories-v1-comments-widget-widgets.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
import { createEventModule } from "@wix/sdk-runtime/event-definition-modules";
var registerCommentsWidget4 = /* @__PURE__ */ createRESTModule(registerCommentsWidget3);
var queryCommentsWidgetResources4 = /* @__PURE__ */ createRESTModule(queryCommentsWidgetResources3);
var createCommentsWidget4 = /* @__PURE__ */ createRESTModule(createCommentsWidget3);
var getCommentsWidget4 = /* @__PURE__ */ createRESTModule(getCommentsWidget3);
var updateCommentsWidget4 = /* @__PURE__ */ createRESTModule(updateCommentsWidget3);
var deleteCommentsWidget4 = /* @__PURE__ */ createRESTModule(deleteCommentsWidget3);
var queryCommentsWidget4 = /* @__PURE__ */ createRESTModule(queryCommentsWidget3);
var onCommentsWidgetCreated2 = createEventModule(
  onCommentsWidgetCreated
);
var onCommentsWidgetDeleted2 = createEventModule(
  onCommentsWidgetDeleted
);
var onCommentsWidgetUpdated2 = createEventModule(
  onCommentsWidgetUpdated
);
export {
  SortOrder,
  WebhookIdentityType,
  createCommentsWidget4 as createCommentsWidget,
  deleteCommentsWidget4 as deleteCommentsWidget,
  getCommentsWidget4 as getCommentsWidget,
  onCommentsWidgetCreated2 as onCommentsWidgetCreated,
  onCommentsWidgetDeleted2 as onCommentsWidgetDeleted,
  onCommentsWidgetUpdated2 as onCommentsWidgetUpdated,
  queryCommentsWidget4 as queryCommentsWidget,
  queryCommentsWidgetResources4 as queryCommentsWidgetResources,
  registerCommentsWidget4 as registerCommentsWidget,
  updateCommentsWidget4 as updateCommentsWidget
};
