// @wix/communication-channel/dist/cjs/index.js
"use strict";

exports.__esModule = true;
exports.serve = exports.open = exports.handshakeCode = void 0;
var _serialization = require("./serialization");
var _comlink = require("comlink");
const handshakeCode = exports.handshakeCode = 'wix-sdk-bo-hello-handshake';
/**
 * Initializes a new channel with a target.
 * @param postMessage - Target's function to receive messages.
 * @param origin - Target's origin.
 */
const open = ({
  postMessage,
  origin
}) => {
  const {
    port1: widgetPort,
    port2: hostPort
  } = new MessageChannel();
  const protocolVersion = '1.0.0';
  const message = {
    code: handshakeCode,
    version: protocolVersion,
    port: hostPort
  };
  postMessage(message, origin, [hostPort]);
  const channel = (0, _serialization.serializeAllMethodsIn)((0, _comlink.wrap)(widgetPort));
  return {
    channel,
    close: () => channel[_comlink.releaseProxy]()
  };
};

/**
 * Serves api as a response over the communication channel to given port.
 * @param api - The api object that has the methods inside it.
 * @param port - The port on which the channel was initiated and needs to receive the api.
 */
exports.open = open;
const serve = ({
  api,
  port
}) => {
  (0, _comlink.expose)((0, _serialization.deserializeAllMethodsIn)(api), port);
};
exports.serve = serve;
