// @wix/headless-node/dist/esm/host-module.js
import { createVeloMonitoringClient } from '@wix/monitoring-velo';
import { getPassThroughHeaders } from './headers';
import { resolveAuthToken } from './utils/auth-utils';
export const createHostModule = () => {
  return {
    __type: 'host',
    create: () => {
      return {};
    },
    host: function (_temp) {
      let {
        req,
        waitUntil,
        essentials
      } = _temp === void 0 ? {} : _temp;
      const passThroughHeaders = getPassThroughHeaders(req);
      return {
        essentials: {
          passThroughHeaders,
          language: (essentials == null ? void 0 : essentials.language) ?? 'en',
          locale: (essentials == null ? void 0 : essentials.locale) ?? 'en-US',
          timezone: (essentials == null ? void 0 : essentials.timezone) ?? 'UTC'
        },
        getMonitoringClient: () => {
          return createVeloMonitoringClient({
            req: req,
            namespace: 'Headless',
            producer: 'backend',
            getAuthToken: async () => resolveAuthToken(),
            onBatchQueueCreated: batchFlushPromise => waitUntil == null ? void 0 : waitUntil(batchFlushPromise)
          });
        }
      };
    }
  };
};
