// @wix/auto_sdk_restaurants_menus/build/es/index.mjs
// src/restaurants-menus-v1-menu-menus.public.ts
import { renameKeysFromRESTResponseToSDKResponse as renameKeysFromRESTResponseToSDKResponse2 } from "@wix/sdk-runtime/rename-all-nested-keys";
import { transformRESTTimestampToSDKTimestamp as transformRESTTimestampToSDKTimestamp2 } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths as transformPaths3 } from "@wix/sdk-runtime/transformations/transform-paths";
import { EventDefinition } from "@wix/sdk-types";

// src/restaurants-menus-v1-menu-menus.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/restaurants-menus-v1-menu-menus.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressRestaurantsMenusMenuV1RestaurantsDuplicateMenuRestaurantsDuplicateMenuUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/menus",
        destPath: "/v1/menus"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/menus",
        destPath: "/v1/bulk/menus"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/menus",
        destPath: "/v1/menus"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/menus",
        destPath: "/v1/bulk/menus"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/menus-menu",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/menus",
        destPath: "/v1/bulk/menus"
      },
      {
        srcPath: "/restaurants/menus/v1/menus",
        destPath: "/v1/menus"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
function resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusSiteUrlRestaurantsMenusSiteUrlUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/menus",
        destPath: "/v1/menus"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/menus",
        destPath: "/v1/bulk/menus"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/menus",
        destPath: "/v1/menus"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/menus",
        destPath: "/v1/bulk/menus"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/menus-menu",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/menus",
        destPath: "/v1/bulk/menus"
      },
      {
        srcPath: "/restaurants/menus/v1/menus",
        destPath: "/v1/menus"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
function resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusMenuUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/menus",
        destPath: "/v1/menus"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/menus",
        destPath: "/v1/bulk/menus"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/menus",
        destPath: "/v1/menus"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/menus",
        destPath: "/v1/bulk/menus"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/menus-menu",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/menus",
        destPath: "/v1/bulk/menus"
      },
      {
        srcPath: "/restaurants/menus/v1/menus",
        destPath: "/v1/menus"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_menus";
function createMenu(payload) {
  function __createMenu({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "menu.createdDate" },
          { path: "menu.updatedDate" },
          { path: "menu.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.RestaurantsMenusMenu.CreateMenu",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusMenuUrl({
        protoPath: "/v1/menus",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "menu.createdDate" },
            { path: "menu.updatedDate" },
            { path: "menu.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createMenu;
}
function bulkCreateMenus(payload) {
  function __bulkCreateMenus({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "menus.createdDate" },
          { path: "menus.updatedDate" },
          { path: "menus.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.RestaurantsMenusMenu.BulkCreateMenus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusMenuUrl({
        protoPath: "/v1/bulk/menus/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.menu.createdDate" },
            { path: "results.menu.updatedDate" },
            { path: "results.menu.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateMenus;
}
function getMenu(payload) {
  function __getMenu({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.RestaurantsMenusMenu.GetMenu",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusMenuUrl({
        protoPath: "/v1/menus/{menuId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "menu.createdDate" },
            { path: "menu.updatedDate" },
            { path: "menu.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getMenu;
}
function listMenus(payload) {
  function __listMenus({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.RestaurantsMenusMenu.ListMenus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusMenuUrl({
        protoPath: "/v1/menus",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "menus.createdDate" },
            { path: "menus.updatedDate" },
            { path: "menus.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listMenus;
}
function queryMenus(payload) {
  function __queryMenus({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.RestaurantsMenusMenu.QueryMenus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusMenuUrl({
        protoPath: "/v1/menus/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "menus.createdDate" },
            { path: "menus.updatedDate" },
            { path: "menus.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryMenus;
}
function updateMenu(payload) {
  function __updateMenu({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "menu.createdDate" },
          { path: "menu.updatedDate" },
          { path: "menu.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "PATCH",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.RestaurantsMenusMenu.UpdateMenu",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusMenuUrl({
        protoPath: "/v1/menus/{menu.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "menu.createdDate" },
            { path: "menu.updatedDate" },
            { path: "menu.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateMenu;
}
function bulkUpdateMenu(payload) {
  function __bulkUpdateMenu({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "menus.mask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "menus.menu.createdDate" },
          { path: "menus.menu.updatedDate" },
          { path: "menus.menu.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.RestaurantsMenusMenu.BulkUpdateMenu",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusMenuUrl({
        protoPath: "/v1/bulk/menus/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.menu.createdDate" },
            { path: "results.menu.updatedDate" },
            { path: "results.menu.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateMenu;
}
function updateExtendedFields(payload) {
  function __updateExtendedFields({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.RestaurantsMenusMenu.UpdateExtendedFields",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusMenuUrl({
        protoPath: "/v1/menus/{id}/updateExtendedFields",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __updateExtendedFields;
}
function deleteMenu(payload) {
  function __deleteMenu({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "DELETE",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.RestaurantsMenusMenu.DeleteMenu",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusMenuUrl({
        protoPath: "/v1/menus/{menuId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteMenu;
}
function getMenuSiteUrl(payload) {
  function __getMenuSiteUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.restaurants_menus_site_url.RestaurantsMenusSiteUrl.GetMenuSiteUrl",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusSiteUrlRestaurantsMenusSiteUrlUrl(
        { protoPath: "/v1/menus-site-url/menus/{id}", data: payload, host }
      ),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getMenuSiteUrl;
}
function queryMenusSiteUrl(payload) {
  function __queryMenusSiteUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.restaurants_menus_site_url.RestaurantsMenusSiteUrl.QueryMenusSiteUrl",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusSiteUrlRestaurantsMenusSiteUrlUrl(
        { protoPath: "/v1/menus-site-url/menus/query", data: payload, host }
      ),
      data: payload
    };
    return metadata;
  }
  return __queryMenusSiteUrl;
}
function duplicateMenu(payload) {
  function __duplicateMenu({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.restaurants_duplicate_menu.RestaurantsDuplicateMenu.DuplicateMenu",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsDuplicateMenuRestaurantsDuplicateMenuUrl(
        { protoPath: "/v1/menus/{id}/duplicate", data: payload, host }
      ),
      data: payload
    };
    return metadata;
  }
  return __duplicateMenu;
}

// src/restaurants-menus-v1-menu-menus.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createMenu2(menu) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ menu });
  const reqOpts = createMenu(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.menu;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { menu: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["menu"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCreateMenus2(menus, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    menus,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkCreateMenus(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          menus: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["menus", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getMenu2(menuId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ menuId });
  const reqOpts = getMenu(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.menu;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { menuId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["menuId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listMenus2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    menuIds: options?.menuIds,
    paging: options?.paging,
    onlyVisible: options?.onlyVisible
  });
  const reqOpts = listMenus(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          menuIds: "$[0].menuIds",
          paging: "$[0].paging",
          onlyVisible: "$[0].onlyVisible"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryMenus2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryMenus(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [])
      );
      return {
        items: transformedData?.menus,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function updateMenu2(_id, menu) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    menu: { ...menu, id: _id }
  });
  const reqOpts = updateMenu(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.menu;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { menu: "$[1]" },
        explicitPathsToArguments: { "menu.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "menu"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateMenu2(menus, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    menus,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkUpdateMenu(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          menus: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["menus", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateExtendedFields2(_id, namespace, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    id: _id,
    namespace,
    namespaceData: options?.namespaceData
  });
  const reqOpts = updateExtendedFields(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          id: "$[0]",
          namespace: "$[1]",
          namespaceData: "$[2].namespaceData"
        },
        singleArgumentUnchanged: false
      },
      ["_id", "namespace", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteMenu2(menuId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ menuId });
  const reqOpts = deleteMenu(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { menuId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["menuId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getMenuSiteUrl2(_id) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ id: _id });
  const reqOpts = getMenuSiteUrl(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { id: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function queryMenusSiteUrl2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    query: options?.query
  });
  const reqOpts = queryMenusSiteUrl(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0].query" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function duplicateMenu2(_id, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    id: _id,
    businessLocationIds: options?.businessLocationIds,
    duplicateSubEntities: options?.duplicateSubEntities,
    menuName: options?.menuName
  });
  const reqOpts = duplicateMenu(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          id: "$[0]",
          businessLocationIds: "$[1].businessLocationIds",
          duplicateSubEntities: "$[1].duplicateSubEntities",
          menuName: "$[1].menuName"
        },
        singleArgumentUnchanged: false
      },
      ["_id", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/restaurants-menus-v1-menu-menus.public.ts
function createMenu3(httpClient) {
  return (menu) => createMenu2(
    menu,
    // @ts-ignore
    { httpClient }
  );
}
function bulkCreateMenus3(httpClient) {
  return (menus, options) => bulkCreateMenus2(
    menus,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function getMenu3(httpClient) {
  return (menuId) => getMenu2(
    menuId,
    // @ts-ignore
    { httpClient }
  );
}
function listMenus3(httpClient) {
  return (options) => listMenus2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function queryMenus3(httpClient) {
  return () => queryMenus2(
    // @ts-ignore
    { httpClient }
  );
}
function updateMenu3(httpClient) {
  return (_id, menu) => updateMenu2(
    _id,
    menu,
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdateMenu3(httpClient) {
  return (menus, options) => bulkUpdateMenu2(
    menus,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function updateExtendedFields3(httpClient) {
  return (_id, namespace, options) => updateExtendedFields2(
    _id,
    namespace,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function deleteMenu3(httpClient) {
  return (menuId) => deleteMenu2(
    menuId,
    // @ts-ignore
    { httpClient }
  );
}
function getMenuSiteUrl3(httpClient) {
  return (_id) => getMenuSiteUrl2(
    _id,
    // @ts-ignore
    { httpClient }
  );
}
function queryMenusSiteUrl3(httpClient) {
  return (options) => queryMenusSiteUrl2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function duplicateMenu3(httpClient) {
  return (_id, options) => duplicateMenu2(
    _id,
    options,
    // @ts-ignore
    { httpClient }
  );
}
var onMenuCreated = EventDefinition(
  "wix.restaurants.menus.v1.menu_created",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.externalReferenceInfo.lastSyncDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onMenuDeleted = EventDefinition(
  "wix.restaurants.menus.v1.menu_deleted",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "undefined.createdDate" },
          { path: "undefined.updatedDate" },
          { path: "undefined.externalReferenceInfo.lastSyncDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onMenuUpdated = EventDefinition(
  "wix.restaurants.menus.v1.menu_updated",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.externalReferenceInfo.lastSyncDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();

// src/restaurants-menus-v1-menu-menus.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
import { createEventModule } from "@wix/sdk-runtime/event-definition-modules";
var createMenu4 = /* @__PURE__ */ createRESTModule(createMenu3);
var bulkCreateMenus4 = /* @__PURE__ */ createRESTModule(bulkCreateMenus3);
var getMenu4 = /* @__PURE__ */ createRESTModule(getMenu3);
var listMenus4 = /* @__PURE__ */ createRESTModule(listMenus3);
var queryMenus4 = /* @__PURE__ */ createRESTModule(queryMenus3);
var updateMenu4 = /* @__PURE__ */ createRESTModule(updateMenu3);
var bulkUpdateMenu4 = /* @__PURE__ */ createRESTModule(bulkUpdateMenu3);
var updateExtendedFields4 = /* @__PURE__ */ createRESTModule(updateExtendedFields3);
var deleteMenu4 = /* @__PURE__ */ createRESTModule(deleteMenu3);
var getMenuSiteUrl4 = /* @__PURE__ */ createRESTModule(getMenuSiteUrl3);
var queryMenusSiteUrl4 = /* @__PURE__ */ createRESTModule(queryMenusSiteUrl3);
var duplicateMenu4 = /* @__PURE__ */ createRESTModule(duplicateMenu3);
var onMenuCreated2 = createEventModule(onMenuCreated);
var onMenuDeleted2 = createEventModule(onMenuDeleted);
var onMenuUpdated2 = createEventModule(onMenuUpdated);
export {
  SortOrder,
  WebhookIdentityType,
  bulkCreateMenus4 as bulkCreateMenus,
  bulkUpdateMenu4 as bulkUpdateMenu,
  createMenu4 as createMenu,
  deleteMenu4 as deleteMenu,
  duplicateMenu4 as duplicateMenu,
  getMenu4 as getMenu,
  getMenuSiteUrl4 as getMenuSiteUrl,
  listMenus4 as listMenus,
  onMenuCreated2 as onMenuCreated,
  onMenuDeleted2 as onMenuDeleted,
  onMenuUpdated2 as onMenuUpdated,
  queryMenus4 as queryMenus,
  queryMenusSiteUrl4 as queryMenusSiteUrl,
  updateExtendedFields4 as updateExtendedFields,
  updateMenu4 as updateMenu
};
