// @wix/headless-ecom/dist/react/CartCoupon.js
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
/**
 * @fileoverview Cart Coupon Components
 *
 * This module provides coupon-related components for cart functionality.
 * These components manage local coupon input state separately from applied coupons.
 */
import React from 'react';
import { Slot } from '@radix-ui/react-slot';
import { DataComponentTags } from '../data-component-tags.js';
import { Coupon as CoreCoupon } from './core/CurrentCart.js';
// Components that render actual DOM elements get test IDs on their rendered elements
var TestIds;
(function (TestIds) {
    TestIds["couponInput"] = "coupon-input";
    TestIds["couponTrigger"] = "coupon-trigger";
    TestIds["couponClear"] = "coupon-clear";
})(TestIds || (TestIds = {}));
const CouponContext = React.createContext(null);
/**
 * Hook to access coupon context
 */
function useCouponContext() {
    const context = React.useContext(CouponContext);
    if (!context) {
        throw new Error('useCouponContext must be used within a Cart.Coupon.Root component');
    }
    return context;
}
/**
 * Root component that provides coupon context to its children.
 * Manages local input state separately from the applied coupon in the cart.
 * Shows input/apply when no coupon is applied, shows coupon display when applied.
 *
 * @component
 * @example
 * ```tsx
 * <Cart.Coupon.Root>
 *   <Cart.Coupon.Input placeholder="Enter coupon code" />
 *   <Cart.Coupon.Trigger>Apply</Cart.Coupon.Trigger>
 *   <Cart.Coupon.Clear />
 * </Cart.Coupon.Root>
 * ```
 */
export const Root = React.forwardRef(({ children, defaultValue = '' }, ref) => {
    const [contextInputValue, setContextInputValue] = React.useState(defaultValue);
    const contextValue = {
        contextInputValue,
        setContextInputValue,
    };
    return (_jsx(CouponContext.Provider, { value: contextValue, children: _jsx("div", { ref: ref, "data-component-tag": DataComponentTags.cartCouponRoot, children: children }) }));
});
Root.displayName = 'CartCoupon.Root';
/**
 * Coupon code input field.
 * Automatically hides when a coupon is already applied.
 *
 * @component
 * @example
 * ```tsx
 * // Must be used within Cart.Coupon.Root container
 * <Cart.Coupon.Root>
 *   <Cart.Coupon.Input
 *     placeholder="Enter coupon code"
 *     className="px-3 py-2 border rounded-lg"
 *   />
 *   <Cart.Coupon.Trigger>Apply</Cart.Coupon.Trigger>
 *   <Cart.Coupon.Clear />
 * </Cart.Coupon.Root>
 *
 * // Custom rendering with asChild
 * <Cart.Coupon.Root>
 *   <Cart.Coupon.Input asChild>
 *     {React.forwardRef(({value, onChange, ...props}, ref) => (
 *       <input
 *         ref={ref}
 *         {...props}
 *         type="text"
 *         value={value}
 *         onChange={(e) => onChange(e.target.value)}
 *         className="px-3 py-2 border rounded-lg focus:ring-2 focus:ring-brand-primary"
 *       />
 *     ))}
 *   </Cart.Coupon.Input>
 * </Cart.Coupon.Root>
 * ```
 */
export const Input = React.forwardRef(({ asChild, children, placeholder = 'Enter coupon code', className, ...props }, ref) => {
    const { contextInputValue, setContextInputValue } = useCouponContext();
    return (_jsx(CoreCoupon, { children: (renderProps) => {
            const { appliedCoupon } = renderProps;
            // Hide input if coupon is already applied
            if (appliedCoupon) {
                return null;
            }
            const inputProps = {
                value: contextInputValue,
                onChange: setContextInputValue,
            };
            if (asChild && children) {
                return children(inputProps, ref);
            }
            const Comp = asChild ? Slot : 'input';
            return (_jsx(Comp, { ref: ref, type: "text", value: contextInputValue, onChange: (e) => setContextInputValue(e.target.value), placeholder: placeholder, className: className, "data-testid": TestIds.couponInput, ...props }));
        } }));
});
/**
 * Apply coupon button.
 * Automatically hides when a coupon is already applied.
 *
 * @component
 * @example
 * ```tsx
 * // Must be used within Cart.Coupon.Root container
 * <Cart.Coupon.Root>
 *   <Cart.Coupon.Input placeholder="Enter coupon code" />
 *   <Cart.Coupon.Trigger className="btn-primary px-4 py-2">Apply</Cart.Coupon.Trigger>
 *   <Cart.Coupon.Clear />
 * </Cart.Coupon.Root>
 *
 * // Custom rendering with asChild
 * <Cart.Coupon.Root>
 *   <Cart.Coupon.Input placeholder="Enter coupon code" />
 *   <Cart.Coupon.Trigger asChild>
 *     {React.forwardRef(({apply, disabled, isLoading, onClick, ...props}, ref) => (
 *       <button
 *         ref={ref}
 *         {...props}
 *         disabled={disabled}
 *         onClick={onClick}
 *         className="btn-primary px-4 py-2 disabled:opacity-50"
 *       >
 *         {isLoading ? 'Applying...' : 'Apply'}
 *       </button>
 *     ))}
 *   </Cart.Coupon.Trigger>
 * </Cart.Coupon.Root>
 * ```
 */
export const Trigger = React.forwardRef(({ asChild, children, className, ...props }, ref) => {
    const { contextInputValue } = useCouponContext();
    return (_jsx(CoreCoupon, { children: (renderProps) => {
            const { apply, isLoading, appliedCoupon } = renderProps;
            // Hide trigger if coupon is already applied
            if (appliedCoupon) {
                return null;
            }
            const disabled = isLoading || !contextInputValue.trim();
            const triggerProps = {
                disabled,
                isLoading,
                apply,
                appliedCoupon: null, // Not needed for trigger
                onClick: async () => {
                    if (contextInputValue.trim()) {
                        await apply(contextInputValue.trim());
                    }
                },
            };
            if (asChild && typeof children === 'function') {
                return children(triggerProps, ref);
            }
            const Comp = asChild ? Slot : 'button';
            return (_jsx(Comp, { ref: ref, onClick: triggerProps.onClick, disabled: disabled, className: className, "data-testid": TestIds.couponTrigger, "data-loading": isLoading, ...props, children: !asChild && typeof children !== 'function'
                    ? children || (isLoading ? 'Applying...' : 'Apply')
                    : null }));
        } }));
});
/**
 * Display applied coupon with remove option.
 * Shows the applied coupon in a styled container with a remove button.
 *
 * @component
 * @example
 * ```tsx
 * // Must be used within Cart.Coupon.Root container
 * <Cart.Coupon.Root>
 *   <Cart.Coupon.Input placeholder="Enter coupon code" />
 *   <Cart.Coupon.Trigger>Apply</Cart.Coupon.Trigger>
 *   <Cart.Coupon.Clear />
 * </Cart.Coupon.Root>
 *
 * // Custom rendering with asChild
 * <Cart.Coupon.Root>
 *   <Cart.Coupon.Clear asChild>
 *     {React.forwardRef(({onClick, disabled, isLoading, appliedCoupon, ...props}, ref) => (
 *       <div ref={ref} {...props} className="flex items-center justify-between px-4 py-3 bg-gray-50 border border-gray-200 rounded-lg">
 *         <span className="text-green-600 font-medium text-sm">Coupon: {appliedCoupon}</span>
 *         <button
 *           onClick={onClick}
 *           disabled={disabled}
 *           className="text-orange-400 hover:text-orange-500 text-sm font-medium disabled:opacity-50"
 *         >
 *           {isLoading ? 'Removing...' : 'Remove'}
 *         </button>
 *       </div>
 *     ))}
 *   </Cart.Coupon.Clear>
 * </Cart.Coupon.Root>
 * ```
 */
export const Clear = React.forwardRef(({ asChild, children, className, ...props }, ref) => {
    const { setContextInputValue } = useCouponContext();
    return (_jsx(CoreCoupon, { children: (renderProps) => {
            const { remove, appliedCoupon, isLoading } = renderProps;
            const disabled = !appliedCoupon || isLoading;
            const clearProps = {
                onClick: async () => {
                    await remove();
                    setContextInputValue(''); // Clear the input when removing coupon
                },
                appliedCoupon,
                disabled,
                isLoading,
            };
            // Only render if there's an applied coupon
            if (!appliedCoupon) {
                return null;
            }
            if (asChild && typeof children === 'function') {
                return children(clearProps, ref);
            }
            const Comp = asChild ? Slot : 'div';
            return (_jsx(Comp, { ref: ref, className: className, "data-testid": TestIds.couponClear, ...props, children: !asChild && typeof children !== 'function' ? (_jsxs("div", { className: "flex items-center justify-between p-3 bg-status-success-light border border-status-success rounded-lg", children: [_jsxs("span", { className: "text-status-success text-sm font-medium", children: ["Coupon: ", appliedCoupon] }), _jsx("button", { onClick: remove, disabled: isLoading, className: "text-status-error hover:text-status-error/80 text-sm disabled:opacity-50", children: isLoading ? 'Removing...' : 'Remove' })] })) : null }));
        } }));
});
/**
 * Raw component that exposes all coupon render props without any UI wrapper.
 * Provides direct access to all coupon functionality and state.
 *
 * @component
 * @example
 * ```tsx
 * <Cart.Coupon.Raw>
 *   {({ appliedCoupon, apply, remove, isLoading, error }) => (
 *     <div>
 *       {appliedCoupon ? (
 *         <div>
 *           <span>Applied: {appliedCoupon}</span>
 *           <button onClick={remove} disabled={isLoading}>
 *             {isLoading ? 'Removing...' : 'Remove'}
 *           </button>
 *         </div>
 *       ) : (
 *         <div>
 *           <input
 *             type="text"
 *             onKeyDown={(e) => {
 *               if (e.key === 'Enter') {
 *                 apply(e.currentTarget.value);
 *               }
 *             }}
 *           />
 *           <button
 *             onClick={() => apply('DISCOUNT10')}
 *             disabled={isLoading}
 *           >
 *             {isLoading ? 'Applying...' : 'Apply Sample'}
 *           </button>
 *         </div>
 *       )}
 *       {error && <div className="error">{error}</div>}
 *     </div>
 *   )}
 * </Cart.Coupon.Raw>
 * ```
 */
export const Raw = ({ children }) => {
    return _jsx(CoreCoupon, { children: children });
};
