// @wix/headless-ecom/dist/react/LineItem.js
import { jsx as _jsx, Fragment as _Fragment } from "react/jsx-runtime";
import React from 'react';
import { AsChildSlot } from '@wix/headless-utils/react';
import { DataComponentTags } from '../data-component-tags.js';
import { WixMediaImage } from '@wix/headless-media/react';
import * as SelectedOption from './SelectedOption.js';
import { Item as CoreItem } from './core/CurrentCart.js';
import { Quantity as QuantityComponent } from '@wix/headless-components/react';
const LineItemContext = React.createContext(null);
function useLineItemContext() {
    const context = React.useContext(LineItemContext);
    if (!context) {
        throw new Error('useLineItemContext must be used within a LineItem.Root component');
    }
    return context;
}
/**
 * Root component for a cart line item that provides the LineItem context to its children
 *
 * @example
 * ```tsx
 * <LineItem.Root item={cartItem}>
 *   <LineItem.Image />
 *   <LineItem.Title />
 *   <LineItem.Quantity />
 * </LineItem.Root>
 * ```
 */
export const Root = React.forwardRef((props, ref) => {
    const { asChild, children, item, ...otherProps } = props;
    const contextValue = {
        lineItem: item,
    };
    return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, "data-testid": TestIds.lineItemRoot, "data-component-tag": DataComponentTags.lineItemRoot, customElement: children, customElementProps: { item }, ...otherProps, children: _jsx(LineItemContext.Provider, { value: contextValue, children: _jsx("div", { "data-component-tag": DataComponentTags.lineItemRoot, children: children }) }) }));
});
Root.displayName = 'LineItem.Root';
var TestIds;
(function (TestIds) {
    TestIds["lineItemRoot"] = "line-item-root";
    TestIds["lineItemTitle"] = "line-item-title";
    TestIds["lineItemImage"] = "line-item-image";
    TestIds["lineItemQuantity"] = "line-item-quantity";
    TestIds["lineItemSelectedOptions"] = "line-item-selected-options";
    TestIds["selectedOption"] = "selected-option";
})(TestIds || (TestIds = {}));
/**
 * Displays the line item title/product name with customizable rendering options following the V2 API.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <LineItem.Title className="text-lg font-medium" />
 *
 * // asChild with primitive element
 * <LineItem.Title asChild>
 *   <h3 className="text-lg font-medium" />
 * </LineItem.Title>
 *
 * // asChild with React component
 * <LineItem.Title asChild>
 *   {React.forwardRef(({title, ...props}, ref) => (
 *     <h3 ref={ref} {...props} className="text-lg font-medium">
 *       {title}
 *     </h3>
 *   ))}
 * </LineItem.Title>
 * ```
 */
export const Title = React.forwardRef((props, ref) => {
    const { asChild, children, ...otherProps } = props;
    const { lineItem } = useLineItemContext();
    return (_jsx(CoreItem, { item: lineItem, children: ({ title }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, "data-testid": TestIds.lineItemTitle, customElement: children, customElementProps: { title }, content: title, ...otherProps, children: _jsx("span", { children: title }) })) }));
});
Title.displayName = 'LineItem.Title';
/**
 * Displays the line item product image using WixMediaImage for optimization.
 * Supports custom rendering via asChild pattern with specific src/alt props.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage with WixMediaImage
 * <LineItem.Image className="w-16 h-16 rounded-lg object-cover" />
 *
 * // Custom rendering with specific props
 * <LineItem.Image asChild>
 *   {React.forwardRef(({ src, alt }, ref) => (
 *     <img
 *       ref={ref}
 *       src={src}
 *       alt={alt}
 *       className="w-16 h-16 rounded-lg object-cover custom-image"
 *     />
 *   ))}
 * </LineItem.Image>
 * ```
 */
export const Image = React.forwardRef((props, ref) => {
    const { asChild, children, ...otherProps } = props;
    const { lineItem } = useLineItemContext();
    return (_jsx(CoreItem, { item: lineItem, children: ({ image, title }) => {
            const src = image || '';
            const alt = title || 'Product image';
            if (asChild && children) {
                // Call the ForwardRefRenderFunction with the specific props
                return children({ src, alt }, ref);
            }
            return (_jsx(WixMediaImage, { ref: ref, media: { image: lineItem?.image }, alt: alt, "data-testid": TestIds.lineItemImage, ...otherProps }));
        } }));
});
Image.displayName = 'LineItem.Image';
/**
 * Container for line item quantity selection controls.
 * Wraps children with Quantity.Root and connects to line item context for quantity operations.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage with quantity controls
 * <LineItem.Quantity steps={1}>
 *   <Quantity.Decrement />
 *   <Quantity.Input />
 *   <Quantity.Increment />
 *   <Quantity.Reset />
 * </LineItem.Quantity>
 * ```
 */
export const Quantity = React.forwardRef((props, ref) => {
    const { asChild, children, ...otherProps } = props;
    const { lineItem } = useLineItemContext();
    return (_jsx(CoreItem, { item: lineItem, children: ({ quantity, onValueChange, remove }) => (_jsx(QuantityComponent.Root, { ref: ref, initialValue: quantity, reset: remove, onValueChange: onValueChange, "data-testid": TestIds.lineItemQuantity, ...otherProps, children: children })) }));
});
Quantity.displayName = 'LineItem.Quantity';
/**
 * Container for selected options display. Does not render when there are no selected options.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <LineItem.SelectedOptions>
 *   <LineItem.SelectedOptionRepeater />
 * </LineItem.SelectedOptions>
 *
 * // asChild with primitive element
 * <LineItem.SelectedOptions asChild>
 *   <section className="selected-options-section">
 *     <LineItem.SelectedOptionRepeater />
 *   </section>
 * </LineItem.SelectedOptions>
 *
 * // asChild with React component
 * <LineItem.SelectedOptions asChild>
 *   {React.forwardRef((props, ref) => (
 *     <section ref={ref} {...props} className="selected-options-section">
 *       {props.children}
 *     </section>
 *   ))}
 * </LineItem.SelectedOptions>
 * ```
 */
export const SelectedOptions = React.forwardRef((props, ref) => {
    const { asChild, children, ...otherProps } = props;
    const { lineItem } = useLineItemContext();
    return (_jsx(CoreItem, { item: lineItem, children: ({ selectedOptions }) => {
            if (selectedOptions.length === 0) {
                return null;
            }
            return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, "data-testid": TestIds.lineItemSelectedOptions, customElement: children, customElementProps: { selectedOptions }, ...otherProps, children: _jsx("div", { children: children }) }));
        } }));
});
SelectedOptions.displayName = 'LineItem.SelectedOptions';
/**
 * Renders a list of selected options. Maps over selected options and renders SelectedOption.Root for each.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <LineItem.SelectedOptionRepeater>
 *   <SelectedOption.Text />
 *   <SelectedOption.Color />
 * </LineItem.SelectedOptionRepeater>
 * ```
 */
export function SelectedOptionRepeater(props) {
    const { children } = props;
    const { lineItem } = useLineItemContext();
    return (_jsx(CoreItem, { item: lineItem, children: ({ selectedOptions }) => {
            if (selectedOptions.length === 0) {
                return null;
            }
            // Convert CoreItem selectedOptions to SelectedOption service format
            const convertedOptions = selectedOptions.map((option) => {
                if (typeof option.value === 'object' &&
                    'name' in option.value &&
                    'code' in option.value) {
                    return {
                        name: option.name,
                        type: 'color',
                        value: {
                            name: option.value.name,
                            code: option.value.code,
                        },
                    };
                }
                else {
                    return {
                        name: option.name,
                        type: 'text',
                        value: String(option.value),
                    };
                }
            });
            return (_jsx(_Fragment, { children: convertedOptions.map((option, index) => (_jsx(SelectedOption.Root, { option: option, "data-testid": TestIds.selectedOption, children: children }, `${option.name}-${index}`))) }));
        } }));
}
SelectedOptionRepeater.displayName = 'LineItem.SelectedOptionRepeater';
