// @wix/headless-ecom/dist/react/SelectedOption.js
import { jsx as _jsx } from "react/jsx-runtime";
import { WixServices, useService } from '@wix/services-manager-react';
import React from 'react';
import { DataComponentTags } from '../data-component-tags.js';
import { SelectedOptionServiceDefinition, SelectedOptionService, isTextOption, isColorOption, } from '../services/selected-option-service.js';
import { createServicesMap } from '@wix/services-manager';
import { AsChildSlot } from '@wix/headless-utils/react';
var TestIds;
(function (TestIds) {
    TestIds["selectedOptionText"] = "selected-option-text";
    TestIds["selectedOptionColor"] = "selected-option-color";
})(TestIds || (TestIds = {}));
/**
 * Root component for a selected option that provides the SelectedOption service context to its children
 *
 * @example
 * ```tsx
 * <SelectedOption.Root option={selectedOption}>
 *   <SelectedOption.Text />
 *   <SelectedOption.Color />
 * </SelectedOption.Root>
 * ```
 */
export const Root = (props) => {
    const { children, option } = props;
    return (_jsx(WixServices, { servicesMap: createServicesMap().addService(SelectedOptionServiceDefinition, SelectedOptionService, {
            selectedOption: option,
        }), children: _jsx(AsChildSlot, { "data-component-tag": DataComponentTags.selectedOptionRoot, children: children }) }));
};
Root.displayName = 'SelectedOption.Root';
/**
 * Displays text-based selected option.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <SelectedOption.Text className="text-sm text-content-secondary" />
 *
 * // Custom rendering with forwardRef
 * <SelectedOption.Text asChild>
 *   {React.forwardRef(({name, value, ...props}, ref) => (
 *     <span ref={ref} {...props} className="text-sm text-content-secondary">
 *       {name}: {value}
 *     </span>
 *   ))}
 * </SelectedOption.Text>
 * ```
 */
export const Text = React.forwardRef((props, ref) => {
    const { asChild, children, ...otherProps } = props;
    const selectedOptionService = useService(SelectedOptionServiceDefinition);
    const selectedOption = selectedOptionService.selectedOption.get();
    // Only render if this is a text option
    if (!isTextOption(selectedOption)) {
        return null;
    }
    const name = selectedOption.name;
    const value = selectedOption.value;
    const defaultContent = `${name}: ${value}`;
    return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, "data-testid": TestIds.selectedOptionText, customElement: children, customElementProps: { name, value }, content: defaultContent, ...otherProps, children: _jsx("span", { children: defaultContent }) }));
});
Text.displayName = 'SelectedOption.Text';
/**
 * Displays color-based selected option.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage - renders "OptionName: ColorName"
 * <SelectedOption.Color className="text-sm text-content-secondary" />
 *
 * // Custom rendering with forwardRef - provides name and colorCode for custom display
 * <SelectedOption.Color asChild>
 *   {React.forwardRef(({name, colorCode, ...props}, ref) => (
 *     <div ref={ref} {...props} className="flex items-center gap-2 text-sm text-content-secondary">
 *       <div
 *         className="w-4 h-4 rounded-full border"
 *         style={{ backgroundColor: colorCode }}
 *       />
 *       <span>{name}</span>
 *     </div>
 *   ))}
 * </SelectedOption.Color>
 * ```
 */
export const Color = React.forwardRef((props, ref) => {
    const { asChild, children, ...otherProps } = props;
    const selectedOptionService = useService(SelectedOptionServiceDefinition);
    const selectedOption = selectedOptionService.selectedOption.get();
    // Only render if this is a color option
    if (!isColorOption(selectedOption)) {
        return null;
    }
    const name = selectedOption.name;
    const colorName = selectedOption.value.name;
    const colorCode = selectedOption.value.code;
    const defaultContent = `${name}: ${colorName}`;
    return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, "data-testid": TestIds.selectedOptionColor, customElement: children, customElementProps: { name, colorCode }, content: defaultContent, ...otherProps, children: _jsx("span", { children: defaultContent }) }));
});
Color.displayName = 'SelectedOption.Color';
