// @wix/auto-patterns/dist/cjs/components/AutoPatternsTable/AutoPatternsTable.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.AutoPatternsTable = void 0;
var _react = _interopRequireDefault(require("react"));
var _patterns = require("@wix/patterns");
var _hooks = require("../../hooks");
var _jsxFileName = "/home/builduser/work/73d19ce3378ce4dd/packages/auto-patterns/dist/cjs/components/AutoPatternsTable/AutoPatternsTable.tsx";
const AutoPatternsTable = ({
  configuration
}) => {
  const {
    state,
    columns,
    customColumns,
    search,
    title,
    emptyState,
    filters,
    actionCell,
    bulkActionToolbar,
    onRowClick,
    stickyColumns,
    stickySelectionColumn
  } = (0, _hooks.useTableFeatures)(configuration);
  return /*#__PURE__*/_react.default.createElement(_patterns.Table, {
    dataHook: "auto-patterns-table",
    search: search,
    title: title,
    state: state,
    columns: columns,
    customColumns: customColumns,
    actionCell: actionCell,
    filters: filters,
    emptyState: emptyState,
    bulkActionToolbar: bulkActionToolbar,
    onRowClick: onRowClick,
    horizontalScroll: true,
    stickyColumns: stickyColumns,
    stickySelectionColumn: stickySelectionColumn,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 29,
      columnNumber: 5
    }
  });
};
exports.AutoPatternsTable = AutoPatternsTable;
