// @wix/fast-gallery-core/dist/cjs/components/media/DefaultImageRenderer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.DefaultImageRenderer = void 0;
var _react = _interopRequireDefault(require("react"));
var _clsx = _interopRequireDefault(require("clsx"));
var _printOnlySource = require("./printOnlySource");
var _DefaultImageRendererModule = _interopRequireDefault(require("./DefaultImageRenderer.module.scss"));
var _jsxFileName = "/home/builduser/work/d3ecddb38d0a3df1/packages/fast-gallery-packages/fast-gallery-core/dist/cjs/components/media/DefaultImageRenderer.tsx";
const DefaultImageRenderer = ({
  data,
  selectors
}) => {
  const focalPoint = data.metaData.focalPoint || {
    x: 0.5,
    y: 0.5
  };
  const styleVariables = {
    '--backgroundPosition': `${focalPoint.x * 100}% ${focalPoint.y * 100}%`
  };
  if (typeof data.mediaUrl === 'string') {
    return /*#__PURE__*/_react.default.createElement("picture", {
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 19,
        columnNumber: 7
      }
    }, /*#__PURE__*/_react.default.createElement("img", {
      className: (0, _clsx.default)(_DefaultImageRendererModule.default.image, selectors == null ? void 0 : selectors.image),
      src: data.mediaUrl,
      alt: data.metaData.title,
      style: styleVariables,
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 20,
        columnNumber: 9
      }
    }));
  } else if (typeof data.mediaUrl === 'object') {
    return /*#__PURE__*/_react.default.createElement("picture", {
      id: `multi_picture_${data.id}`,
      key: `multi_picture_${data.id}`,
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 30,
        columnNumber: 7
      }
    }, data.mediaUrl.map((src, index) => src.forPrinting ? /*#__PURE__*/_react.default.createElement(_printOnlySource.PrintOnlyImageSource, {
      key: `print-only-image-${index}`,
      srcSet: src.dpr,
      type: `image/${src.type}`,
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 33,
        columnNumber: 13
      }
    }) : /*#__PURE__*/_react.default.createElement("source", {
      key: `image-source-${index}`,
      srcSet: src.dpr || src.url,
      type: `image/${src.type}`,
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 39,
        columnNumber: 13
      }
    })), /*#__PURE__*/_react.default.createElement("img", {
      className: (0, _clsx.default)(_DefaultImageRendererModule.default.image, selectors == null ? void 0 : selectors.image),
      src: data.mediaUrl[data.mediaUrl.length - 1].url,
      alt: data.metaData.title,
      style: styleVariables,
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 46,
        columnNumber: 9
      }
    }));
  } else {
    return null;
  }
};
exports.DefaultImageRenderer = DefaultImageRenderer;
