// @wix/fast-gallery-core/dist/cjs/components/media/MediaRenderer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.default = void 0;
var _react = _interopRequireDefault(require("react"));
var _clsx = _interopRequireDefault(require("clsx"));
var _lib = require("../../lib");
var _VideoRenderer = _interopRequireDefault(require("./VideoRenderer"));
var _utils = require("../utils");
var _DefaultImageRenderer = require("./DefaultImageRenderer");
var _MediaRendererModule = _interopRequireDefault(require("./MediaRenderer.module.scss"));
var _jsxFileName = "/home/builduser/work/d3ecddb38d0a3df1/packages/fast-gallery-packages/fast-gallery-core/dist/cjs/components/media/MediaRenderer.tsx";
const MediaRenderer = ({
  data,
  selectors,
  customRenderers,
  options
}) => {
  const {
    metaData
  } = data;
  const getInnerMediaItem = data => {
    const props = {
      data,
      selectors,
      customRenderers
    };
    switch (data.type) {
      case _lib.MediaTypes.VIDEO:
        return (0, _utils.renderCustomOrDefaultComp)(customRenderers == null ? void 0 : customRenderers.videoRenderer, _VideoRenderer.default, props);
      case _lib.MediaTypes.IMAGE:
      default:
        return (0, _utils.renderCustomOrDefaultComp)(customRenderers == null ? void 0 : customRenderers.imageRenderer, _DefaultImageRenderer.DefaultImageRenderer, props);
    }
  };
  const innerComp = /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, getInnerMediaItem(data), /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _clsx.default)(_MediaRendererModule.default.overlay, selectors == null ? void 0 : selectors.overlay),
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 43,
      columnNumber: 7
    }
  }));
  const linkParams = (0, _utils.getLinkParams)({
    link: metaData.link,
    itemClickAction: options == null ? void 0 : options.itemClickAction
  });
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, linkParams != null && linkParams.href ? /*#__PURE__*/_react.default.createElement("a", {
    href: linkParams.href,
    target: linkParams.target,
    rel: "playground noreferrer",
    tabIndex: -1,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 54,
      columnNumber: 9
    }
  }, innerComp) : /*#__PURE__*/_react.default.createElement("span", {
    "aria-hidden": true // hide it from assistive technologies for SEO
    ,
    rel: "playground noreferrer",
    tabIndex: -1,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 63,
      columnNumber: 9
    }
  }, innerComp));
};
var _default = exports.default = MediaRenderer;
