// @wix/fast-gallery-core/dist/cjs/components/media/printOnlySource.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.PrintOnlyImageSource = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireDefault(require("react"));
var _jsxFileName = "/home/builduser/work/d3ecddb38d0a3df1/packages/fast-gallery-packages/fast-gallery-core/dist/cjs/components/media/printOnlySource.tsx";
class PrintOnlyImageSource extends _react.default.Component {
  constructor(props) {
    super(props);
    this.state = {
      isPrinting: false
    };
  }
  componentDidMount() {
    window.addEventListener('beforeprint', () => {
      this.setState({
        isPrinting: true
      });
    });
    window.addEventListener('afterprint', () => {
      this.setState({
        isPrinting: false
      });
    });
  }
  render() {
    return this.state.isPrinting ? /*#__PURE__*/_react.default.createElement("source", (0, _extends2.default)({}, this.props, {
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 23,
        columnNumber: 36
      }
    })) : null;
  }
}
exports.PrintOnlyImageSource = PrintOnlyImageSource;
