// @wix/auto_sdk_restaurants_item-labels/build/internal/es/meta.mjs
// src/restaurants-menus-v1-item-label-item-labels.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressRestaurantsMenusItemLabelsV1RestaurantsMenusItemLabelsUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-labels",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-menus-item-labels",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-labels",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-labels",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-menus-item-labels",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-menus-item-labels",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-menus-item-labels",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-labels",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-menus-item-labels",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/item-labels",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/labels",
        destPath: "/v1/labels"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-labels",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_item-labels";
function createLabel(payload) {
  function __createLabel({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "label.createdDate" },
          { path: "label.updatedDate" },
          { path: "label.icon.urlExpirationDate" },
          { path: "label.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_label",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_item_labels.v1.RestaurantsMenusItemLabels.CreateLabel",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemLabelsV1RestaurantsMenusItemLabelsUrl(
        { protoPath: "/v1/labels", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "label.createdDate" },
            { path: "label.updatedDate" },
            { path: "label.icon.urlExpirationDate" },
            { path: "label.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createLabel;
}
function getLabel(payload) {
  function __getLabel({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_label",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.menus_item_labels.v1.RestaurantsMenusItemLabels.GetLabel",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemLabelsV1RestaurantsMenusItemLabelsUrl(
        { protoPath: "/v1/labels/{labelId}", data: payload, host }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "label.createdDate" },
            { path: "label.updatedDate" },
            { path: "label.icon.urlExpirationDate" },
            { path: "label.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getLabel;
}
function listLabels(payload) {
  function __listLabels({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_label",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.menus_item_labels.v1.RestaurantsMenusItemLabels.ListLabels",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemLabelsV1RestaurantsMenusItemLabelsUrl(
        { protoPath: "/v1/labels", data: payload, host }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "labels.createdDate" },
            { path: "labels.updatedDate" },
            { path: "labels.icon.urlExpirationDate" },
            { path: "labels.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listLabels;
}
function queryLabels(payload) {
  function __queryLabels({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_label",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_item_labels.v1.RestaurantsMenusItemLabels.QueryLabels",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemLabelsV1RestaurantsMenusItemLabelsUrl(
        { protoPath: "/v1/labels/query", data: payload, host }
      ),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "labels.createdDate" },
            { path: "labels.updatedDate" },
            { path: "labels.icon.urlExpirationDate" },
            { path: "labels.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryLabels;
}
function countLabels(payload) {
  function __countLabels({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_label",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_item_labels.v1.RestaurantsMenusItemLabels.CountLabels",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemLabelsV1RestaurantsMenusItemLabelsUrl(
        { protoPath: "/v1/labels/count", data: payload, host }
      ),
      data: payload
    };
    return metadata;
  }
  return __countLabels;
}
function updateLabel(payload) {
  function __updateLabel({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "label.createdDate" },
          { path: "label.updatedDate" },
          { path: "label.icon.urlExpirationDate" },
          { path: "label.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_label",
      method: "PATCH",
      methodFqn: "com.wixpress.restaurants.menus_item_labels.v1.RestaurantsMenusItemLabels.UpdateLabel",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemLabelsV1RestaurantsMenusItemLabelsUrl(
        { protoPath: "/v1/labels/{label.id}", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "label.createdDate" },
            { path: "label.updatedDate" },
            { path: "label.icon.urlExpirationDate" },
            { path: "label.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateLabel;
}
function deleteLabel(payload) {
  function __deleteLabel({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_label",
      method: "DELETE",
      methodFqn: "com.wixpress.restaurants.menus_item_labels.v1.RestaurantsMenusItemLabels.DeleteLabel",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemLabelsV1RestaurantsMenusItemLabelsUrl(
        { protoPath: "/v1/labels/{labelId}", data: payload, host }
      ),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteLabel;
}
function bulkDeleteLabels(payload) {
  function __bulkDeleteLabels({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_label",
      method: "DELETE",
      methodFqn: "com.wixpress.restaurants.menus_item_labels.v1.RestaurantsMenusItemLabels.BulkDeleteLabels",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemLabelsV1RestaurantsMenusItemLabelsUrl(
        { protoPath: "/v1/bulk/labels/delete", data: payload, host }
      ),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __bulkDeleteLabels;
}

// src/restaurants-menus-v1-item-label-item-labels.types.ts
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/restaurants-menus-v1-item-label-item-labels.meta.ts
function createLabel2() {
  const payload = {};
  const getRequestOptions = createLabel(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/labels",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getLabel2() {
  const payload = { labelId: ":labelId" };
  const getRequestOptions = getLabel(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/labels/{labelId}",
    pathParams: { labelId: "labelId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listLabels2() {
  const payload = {};
  const getRequestOptions = listLabels(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/labels",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryLabels2() {
  const payload = {};
  const getRequestOptions = queryLabels(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/labels/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function countLabels2() {
  const payload = {};
  const getRequestOptions = countLabels(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/labels/count",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateLabel2() {
  const payload = { label: { id: ":labelId" } };
  const getRequestOptions = updateLabel(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/labels/{label.id}",
    pathParams: { labelId: "labelId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteLabel2() {
  const payload = { labelId: ":labelId" };
  const getRequestOptions = deleteLabel(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/labels/{labelId}",
    pathParams: { labelId: "labelId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkDeleteLabels2() {
  const payload = {};
  const getRequestOptions = bulkDeleteLabels(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/bulk/labels/delete",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  SortOrder as SortOrderOriginal,
  WebhookIdentityType as WebhookIdentityTypeOriginal,
  bulkDeleteLabels2 as bulkDeleteLabels,
  countLabels2 as countLabels,
  createLabel2 as createLabel,
  deleteLabel2 as deleteLabel,
  getLabel2 as getLabel,
  listLabels2 as listLabels,
  queryLabels2 as queryLabels,
  updateLabel2 as updateLabel
};
