// @wix/patterns/dist/cjs/components/FormPage/FormPageProvider.js
"use strict";

exports.__esModule = true;
exports.FormPageProvider = FormPageProvider;
var _react = _interopRequireWildcard(require("react"));
var _react2 = require("@wix/bex-core/react");
var _PageWrapper = require("../PageWrapper");
var _bexCore = require("@wix/bex-core");
var _reactHookForm = require("react-hook-form");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/FormPage/FormPageProvider.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function FormPageProvider({
  children,
  state
}) {
  const newState = (0, _reactHookForm.useForm)();
  const [formPageState] = (0, _react.useState)(() => state ?? new _bexCore.FormPageState({
    form: newState
  }));
  const {
    form: {
      formState: {
        isSubmitSuccessful,
        isDirty
      }
    }
  } = formPageState;

  /* formState is wrapped with a Proxy to improve render performance and skip extra logic if specific state is not subscribed to.
  Therefore, make sure you invoke or read it before a render in order to enable the state update. */
  (0, _react.useEffect)(() => {
    formPageState._isSubmitSuccessful = isSubmitSuccessful;
    formPageState._isFormDirty = isDirty;
  }, [isSubmitSuccessful, isDirty]);
  return /*#__PURE__*/_react.default.createElement(_react2.FormPageContextProvider, {
    value: formPageState,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 33,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(_PageWrapper.PageWrapperBase, {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 34,
      columnNumber: 7
    }
  }, children));
}
