// @wix/monitoring-browser-panorama/build/index.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var index_exports = {};
__export(index_exports, {
  createPanoramaBrowserClient: () => createPanoramaBrowserClient
});
module.exports = __toCommonJS(index_exports);

// src/client.ts
var import_monitoring_common_sentry = require("@wix/monitoring-common-sentry");
var PanoramaBrowserClient = class {
  constructor(options) {
    this.options = options;
    this.captureException = (error, captureContext) => {
      const { panoramaClient } = this.options;
      if (!(error instanceof Error)) {
        error = new Error(error);
      }
      panoramaClient.errorMonitor().reportError(error, this.contextToData(captureContext));
    };
    this.captureMessage = (message, captureContext) => {
      const { panoramaClient } = this.options;
      const logger = panoramaClient.logger();
      let loggerByLevel;
      switch (captureContext?.level) {
        case "error":
          loggerByLevel = logger.error;
          break;
        case "warning":
          loggerByLevel = logger.warn;
          break;
        default:
          loggerByLevel = logger.info;
      }
      loggerByLevel(message, this.contextToData(captureContext));
    };
    this.startSpan = (spanOptions, callback) => {
      const { panoramaClient } = this.options;
      const { name } = spanOptions;
      let res;
      const transaction = panoramaClient.transaction(name, {
        id: (0, import_monitoring_common_sentry.generateRandomId)()
      });
      const data = this.contextToData(spanOptions);
      transaction.start(data);
      try {
        res = callback();
        if ((0, import_monitoring_common_sentry.isPromise)(res)) {
          res = res.then((_res) => {
            transaction.finish(data);
            return _res;
          }).catch((e) => {
            this.reportErrorInContextOfSpan(e, spanOptions);
            throw e;
          });
        } else {
          transaction.finish(data);
        }
      } catch (e) {
        this.reportErrorInContextOfSpan(e, spanOptions);
        throw e;
      }
      return res;
    };
    this.addBreadcrumb = (breadcrumb) => {
      const { panoramaClient } = this.options;
      panoramaClient.errorMonitor().addBreadcrumb(breadcrumb);
    };
  }
  contextToData(captureContext) {
    return captureContext && {
      ...captureContext?.tags && { tags: captureContext.tags },
      ...captureContext?.contexts && { context: captureContext.contexts },
      ...captureContext?.level && {
        severity: captureContext.level
      }
    };
  }
  reportErrorInContextOfSpan(error, spanOptions) {
    const { panoramaClient } = this.options;
    const { name } = spanOptions;
    const data = this.contextToData({
      ...spanOptions,
      contexts: { transaction: { name } }
    });
    panoramaClient.errorMonitor().reportError(error, data);
  }
  startSpanManual(spanOptions) {
    const { panoramaClient } = this.options;
    const { name } = spanOptions;
    const transaction = panoramaClient.transaction(name, {
      id: (0, import_monitoring_common_sentry.generateRandomId)()
    });
    const data = this.contextToData(spanOptions);
    transaction.start(data);
    let reported = false;
    const reportOnce = (cb) => {
      if (reported) {
        return;
      }
      reported = true;
      cb();
    };
    return {
      end: () => {
        reportOnce(() => transaction.finish(data));
      },
      fail: (error) => {
        reportOnce(() => this.reportErrorInContextOfSpan(error, spanOptions));
      }
    };
  }
  endSpanManual(spanOptions) {
    const { panoramaClient } = this.options;
    const { name } = spanOptions;
    panoramaClient.transaction(name).finish();
  }
};

// src/factory.ts
var import_monitoring = require("@wix/monitoring");
var createPanoramaBrowserClient = (options) => {
  const { panoramaClient } = options;
  if (!panoramaClient) {
    return (0, import_monitoring.createFallbackMonitoringClient)("Missing Panorama client");
  }
  return new PanoramaBrowserClient(options);
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  createPanoramaBrowserClient
});
