// @wix/auto_sdk_restaurants_operations/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  ActivationTypeOriginal: () => ActivationType,
  AsapFutureHandlingTypeOriginal: () => AsapFutureHandlingType,
  AsapPacingWindowTypeOriginal: () => AsapPacingWindowType,
  AsapPreorderTypeOriginal: () => AsapPreorderType,
  DayOfWeekOriginal: () => DayOfWeek,
  DeleteStatusOriginal: () => DeleteStatus,
  EntitiesDayOfWeekOriginal: () => EntitiesDayOfWeek,
  FulfillmentTimeTypeOriginal: () => FulfillmentTimeType,
  FulfillmentTimesDisplayTypeOriginal: () => FulfillmentTimesDisplayType,
  FulfillmentTimesTypeOriginal: () => FulfillmentTimesType,
  FulfillmentTypeOriginal: () => FulfillmentType,
  MethodTypeOriginal: () => MethodType,
  NamespaceOriginal: () => Namespace,
  OnlineOrderingStatusTypeOriginal: () => OnlineOrderingStatusType,
  OrderSchedulingTypeOriginal: () => OrderSchedulingType,
  PlacementTypeOriginal: () => PlacementType,
  PreorderPacingWindowTypeOriginal: () => PreorderPacingWindowType,
  PreparationTimePreparationTimeTypeOriginal: () => PreparationTimePreparationTimeType,
  PreparationTimeTypeOriginal: () => PreparationTimeType,
  ResolutionMethodOriginal: () => ResolutionMethod,
  SchedulingTypeOriginal: () => SchedulingType,
  SiteCreatedContextOriginal: () => SiteCreatedContext,
  SortOrderOriginal: () => SortOrder,
  StateOriginal: () => State,
  SubdivisionTypeOriginal: () => SubdivisionType,
  TimeSlotForFulfillmentTimeSlotStatusOriginal: () => TimeSlotForFulfillmentTimeSlotStatus,
  TimeSlotStatusOriginal: () => TimeSlotStatus,
  TimeUnitOriginal: () => TimeUnit,
  ViolationTypeOriginal: () => ViolationType,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  bulkUpdateOperationTags: () => bulkUpdateOperationTags2,
  bulkUpdateOperationTagsByFilter: () => bulkUpdateOperationTagsByFilter2,
  calculateAvailableDatesInRange: () => calculateAvailableDatesInRange2,
  calculateAvailableTimeSlotsForDate: () => calculateAvailableTimeSlotsForDate2,
  calculateFirstAvailableTimeSlotPerFulfillmentType: () => calculateFirstAvailableTimeSlotPerFulfillmentType2,
  calculateFirstAvailableTimeSlotsForItem: () => calculateFirstAvailableTimeSlotsForItem2,
  calculateFirstAvailableTimeSlotsPerMenu: () => calculateFirstAvailableTimeSlotsPerMenu2,
  calculateFirstAvailableTimeSlotsPerOperation: () => calculateFirstAvailableTimeSlotsPerOperation2,
  createOperation: () => createOperation2,
  deleteOperation: () => deleteOperation2,
  getOperation: () => getOperation2,
  listAvailableDatesInRange: () => listAvailableDatesInRange2,
  listAvailableFulfillmentOptions: () => listAvailableFulfillmentOptions2,
  listAvailableTimeSlotsForDate: () => listAvailableTimeSlotsForDate2,
  listFirstAvailableTimeSlotForFulfillmentTypes: () => listFirstAvailableTimeSlotForFulfillmentTypes2,
  listFirstAvailableTimeSlotsForMenus: () => listFirstAvailableTimeSlotsForMenus2,
  listFirstAvailableTimeSlotsForOperations: () => listFirstAvailableTimeSlotsForOperations2,
  listOperations: () => listOperations2,
  queryOperation: () => queryOperation2,
  updateOperation: () => updateOperation2,
  validateOperationAddress: () => validateOperationAddress2
});
module.exports = __toCommonJS(meta_exports);

// src/restaurants-operations-v1-operation-operations.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixRestaurantsOperationsV1OperationsServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-operations/v1",
        destPath: "/v1"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-operations/v1",
        destPath: "/v1"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-operations/v1",
        destPath: "/v1"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-operations/v1",
        destPath: "/v1"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants-operations/v1",
        destPath: "/v1"
      },
      {
        srcPath: "/restaurants/operations/v1/operations",
        destPath: "/v1/operations"
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-operations/v1",
        destPath: "/v1"
      }
    ],
    _base_domain_: [
      {
        srcPath: "/_api/restaurants-operations/v1",
        destPath: "/v1"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-operations/v1",
        destPath: "/v1"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-operations/v1",
        destPath: "/v1"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-operations/v1",
        destPath: "/v1"
      }
    ],
    "users._base_domain_": [
      {
        srcPath: "/_api/restaurants-operations/v1",
        destPath: "/v1"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-operations",
        destPath: ""
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-operations",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_operations";
function createOperation(payload) {
  function __createOperation({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "operation.createdDate" },
          { path: "operation.updatedDate" },
          { path: "operation.pausedUntilOptions.time" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          {
            path: "operation.businessLocationDetails.address.geocode.latitude"
          },
          {
            path: "operation.businessLocationDetails.address.geocode.longitude"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "POST",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.CreateOperation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "operation.createdDate" },
            { path: "operation.updatedDate" },
            { path: "operation.pausedUntilOptions.time" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "operation.businessLocationDetails.address.geocode.latitude"
            },
            {
              path: "operation.businessLocationDetails.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createOperation;
}
function getOperation(payload) {
  function __getOperation({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "GET",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.GetOperation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/{operationId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "operation.createdDate" },
            { path: "operation.updatedDate" },
            { path: "operation.pausedUntilOptions.time" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "operation.businessLocationDetails.address.geocode.latitude"
            },
            {
              path: "operation.businessLocationDetails.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getOperation;
}
function updateOperation(payload) {
  function __updateOperation({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "operation.createdDate" },
          { path: "operation.updatedDate" },
          { path: "operation.pausedUntilOptions.time" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          {
            path: "operation.businessLocationDetails.address.geocode.latitude"
          },
          {
            path: "operation.businessLocationDetails.address.geocode.longitude"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "PATCH",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.UpdateOperation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/{operation.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "operation.createdDate" },
            { path: "operation.updatedDate" },
            { path: "operation.pausedUntilOptions.time" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "operation.businessLocationDetails.address.geocode.latitude"
            },
            {
              path: "operation.businessLocationDetails.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateOperation;
}
function deleteOperation(payload) {
  function __deleteOperation({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "DELETE",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.DeleteOperation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/{operationId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteOperation;
}
function queryOperation(payload) {
  function __queryOperation({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "POST",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.QueryOperation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "operations.createdDate" },
            { path: "operations.updatedDate" },
            { path: "operations.pausedUntilOptions.time" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "operations.businessLocationDetails.address.geocode.latitude"
            },
            {
              path: "operations.businessLocationDetails.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryOperation;
}
function listOperations(payload) {
  function __listOperations({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "GET",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.ListOperations",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "operations.createdDate" },
            { path: "operations.updatedDate" },
            { path: "operations.pausedUntilOptions.time" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "operations.businessLocationDetails.address.geocode.latitude"
            },
            {
              path: "operations.businessLocationDetails.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listOperations;
}
function listAvailableFulfillmentOptions(payload) {
  function __listAvailableFulfillmentOptions({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "deliveryAddress.geocode.latitude" },
          { path: "deliveryAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "GET",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.ListAvailableFulfillmentOptions",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/{operationId}/available-fulfillment-options",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "fulfillmentOptions.pickupOptions.address.geocode.latitude"
            },
            {
              path: "fulfillmentOptions.pickupOptions.address.geocode.longitude"
            }
          ]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "fulfillmentOptions.availability.startTime" },
            { path: "fulfillmentOptions.availability.endTime" },
            { path: "fulfillmentOptions.availability.exceptions.startTime" },
            { path: "fulfillmentOptions.availability.exceptions.endTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listAvailableFulfillmentOptions;
}
function listFirstAvailableTimeSlotForFulfillmentTypes(payload) {
  function __listFirstAvailableTimeSlotForFulfillmentTypes({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "deliveryAddress.geocode.latitude" },
          { path: "deliveryAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "GET",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.ListFirstAvailableTimeSlotForFulfillmentTypes",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/{operationId}/first-available-time-slot-for-fulfillment-types",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "timeSlots.startTime" },
            { path: "timeSlots.endTime" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "timeSlots.fulfillmentAddress.address.geocode.latitude" },
            {
              path: "timeSlots.fulfillmentAddress.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listFirstAvailableTimeSlotForFulfillmentTypes;
}
function calculateFirstAvailableTimeSlotPerFulfillmentType(payload) {
  function __calculateFirstAvailableTimeSlotPerFulfillmentType({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "deliveryAddress.geocode.latitude" },
          { path: "deliveryAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "GET",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.CalculateFirstAvailableTimeSlotPerFulfillmentType",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/{operationId}/first-available-time-slot-per-fulfillment-type",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "timeslotsPerFulfillmentType.timeSlot.startTime" },
            { path: "timeslotsPerFulfillmentType.timeSlot.endTime" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "timeslotsPerFulfillmentType.fulfillmentInfo.address.geocode.latitude"
            },
            {
              path: "timeslotsPerFulfillmentType.fulfillmentInfo.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __calculateFirstAvailableTimeSlotPerFulfillmentType;
}
function listFirstAvailableTimeSlotsForOperations(payload) {
  function __listFirstAvailableTimeSlotsForOperations({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "deliveryAddress.geocode.latitude" },
          { path: "deliveryAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "POST",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.ListFirstAvailableTimeSlotsForOperations",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/first-available-time-slots-for-operations",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "timeSlots.timeSlots.startTime" },
            { path: "timeSlots.timeSlots.endTime" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "timeSlots.timeSlots.fulfillmentAddress.address.geocode.latitude"
            },
            {
              path: "timeSlots.timeSlots.fulfillmentAddress.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listFirstAvailableTimeSlotsForOperations;
}
function calculateFirstAvailableTimeSlotsPerOperation(payload) {
  function __calculateFirstAvailableTimeSlotsPerOperation({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "deliveryAddress.geocode.latitude" },
          { path: "deliveryAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "POST",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.CalculateFirstAvailableTimeSlotsPerOperation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/first-available-time-slots-per-operations",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            {
              path: "timeSlotsPerOperation.timeslotsPerFulfillmentType.timeSlot.startTime"
            },
            {
              path: "timeSlotsPerOperation.timeslotsPerFulfillmentType.timeSlot.endTime"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "timeSlotsPerOperation.timeslotsPerFulfillmentType.fulfillmentInfo.address.geocode.latitude"
            },
            {
              path: "timeSlotsPerOperation.timeslotsPerFulfillmentType.fulfillmentInfo.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __calculateFirstAvailableTimeSlotsPerOperation;
}
function listFirstAvailableTimeSlotsForMenus(payload) {
  function __listFirstAvailableTimeSlotsForMenus({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "deliveryAddress.geocode.latitude" },
          { path: "deliveryAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "GET",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.ListFirstAvailableTimeSlotsForMenus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/{operationId}/first-available-time-slots-for-menus",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            {
              path: "timeSlotsPerMenu.timeslotsPerFulfillmentType.startTime"
            },
            { path: "timeSlotsPerMenu.timeslotsPerFulfillmentType.endTime" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "timeSlotsPerMenu.timeslotsPerFulfillmentType.fulfillmentAddress.address.geocode.latitude"
            },
            {
              path: "timeSlotsPerMenu.timeslotsPerFulfillmentType.fulfillmentAddress.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listFirstAvailableTimeSlotsForMenus;
}
function calculateFirstAvailableTimeSlotsPerMenu(payload) {
  function __calculateFirstAvailableTimeSlotsPerMenu({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "deliveryAddress.geocode.latitude" },
          { path: "deliveryAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "GET",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.CalculateFirstAvailableTimeSlotsPerMenu",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/{operationId}/first-available-time-slots-per-menus",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            {
              path: "timeSlotsPerMenu.timeslotsPerFulfillmentType.timeSlot.startTime"
            },
            {
              path: "timeSlotsPerMenu.timeslotsPerFulfillmentType.timeSlot.endTime"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "timeSlotsPerMenu.timeslotsPerFulfillmentType.fulfillmentInfo.address.geocode.latitude"
            },
            {
              path: "timeSlotsPerMenu.timeslotsPerFulfillmentType.fulfillmentInfo.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __calculateFirstAvailableTimeSlotsPerMenu;
}
function calculateFirstAvailableTimeSlotsForItem(payload) {
  function __calculateFirstAvailableTimeSlotsForItem({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "deliveryAddress.geocode.latitude" },
          { path: "deliveryAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "GET",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.CalculateFirstAvailableTimeSlotsForItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/{operationId}/first-available-time-slots-per-item",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "timeslotsPerFulfillmentType.timeSlot.startTime" },
            { path: "timeslotsPerFulfillmentType.timeSlot.endTime" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "timeslotsPerFulfillmentType.fulfillmentInfo.address.geocode.latitude"
            },
            {
              path: "timeslotsPerFulfillmentType.fulfillmentInfo.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __calculateFirstAvailableTimeSlotsForItem;
}
function listAvailableTimeSlotsForDate(payload) {
  function __listAvailableTimeSlotsForDate({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "deliveryAddress.geocode.latitude" },
          { path: "deliveryAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "GET",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.ListAvailableTimeSlotsForDate",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/{operationId}/available-time-slots-for-date",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "timeSlots.startTime" },
            { path: "timeSlots.endTime" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "timeSlots.fulfillmentAddress.address.geocode.latitude" },
            {
              path: "timeSlots.fulfillmentAddress.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listAvailableTimeSlotsForDate;
}
function calculateAvailableTimeSlotsForDate(payload) {
  function __calculateAvailableTimeSlotsForDate({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "deliveryAddress.geocode.latitude" },
          { path: "deliveryAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "GET",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.CalculateAvailableTimeSlotsForDate",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/{operationId}/available-time-slots-per-date",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "timeslotsPerFulfillmentType.timeSlot.startTime" },
            { path: "timeslotsPerFulfillmentType.timeSlot.endTime" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "timeslotsPerFulfillmentType.fulfillmentInfo.address.geocode.latitude"
            },
            {
              path: "timeslotsPerFulfillmentType.fulfillmentInfo.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __calculateAvailableTimeSlotsForDate;
}
function listAvailableDatesInRange(payload) {
  function __listAvailableDatesInRange({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "deliveryAddress.geocode.latitude" },
          { path: "deliveryAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "GET",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.ListAvailableDatesInRange",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/{operationId}/available-dates-in-range",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData)
    };
    return metadata;
  }
  return __listAvailableDatesInRange;
}
function calculateAvailableDatesInRange(payload) {
  function __calculateAvailableDatesInRange({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "deliveryAddress.geocode.latitude" },
          { path: "deliveryAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "GET",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.CalculateAvailableDatesInRange",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/{operationId}/available-dates",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData)
    };
    return metadata;
  }
  return __calculateAvailableDatesInRange;
}
function validateOperationAddress(payload) {
  function __validateOperationAddress({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "GET",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.ValidateOperationAddress",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/{operationId}/validate-address",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __validateOperationAddress;
}
function bulkUpdateOperationTags(payload) {
  function __bulkUpdateOperationTags({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "POST",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.BulkUpdateOperationTags",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/bulk/operations/update-tags",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateOperationTags;
}
function bulkUpdateOperationTagsByFilter(payload) {
  function __bulkUpdateOperationTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "POST",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.BulkUpdateOperationTagsByFilter",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/bulk/operations/update-tags-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateOperationTagsByFilter;
}

// src/restaurants-operations-v1-operation-operations.types.ts
var SchedulingType = /* @__PURE__ */ ((SchedulingType2) => {
  SchedulingType2["UNKNOWN_SCHEDULING"] = "UNKNOWN_SCHEDULING";
  SchedulingType2["ASAP"] = "ASAP";
  SchedulingType2["PREORDER"] = "PREORDER";
  return SchedulingType2;
})(SchedulingType || {});
var PreparationTimeType = /* @__PURE__ */ ((PreparationTimeType2) => {
  PreparationTimeType2["UNKNOWN_PREPARATION_TIME"] = "UNKNOWN_PREPARATION_TIME";
  PreparationTimeType2["MAX"] = "MAX";
  PreparationTimeType2["RANGE"] = "RANGE";
  return PreparationTimeType2;
})(PreparationTimeType || {});
var TimeUnit = /* @__PURE__ */ ((TimeUnit2) => {
  TimeUnit2["UNKNOWN_TIME_UNIT"] = "UNKNOWN_TIME_UNIT";
  TimeUnit2["MINUTES"] = "MINUTES";
  TimeUnit2["HOURS"] = "HOURS";
  TimeUnit2["DAYS"] = "DAYS";
  return TimeUnit2;
})(TimeUnit || {});
var AsapPreorderType = /* @__PURE__ */ ((AsapPreorderType2) => {
  AsapPreorderType2["UNKNOWN_ASAP_PREORDER"] = "UNKNOWN_ASAP_PREORDER";
  AsapPreorderType2["NO_PREORDER"] = "NO_PREORDER";
  AsapPreorderType2["BUSINESS_DAYS_PREORDER"] = "BUSINESS_DAYS_PREORDER";
  return AsapPreorderType2;
})(AsapPreorderType || {});
var AsapPacingWindowType = /* @__PURE__ */ ((AsapPacingWindowType2) => {
  AsapPacingWindowType2["UNKNOWN_ASAP_PACING_DURATION"] = "UNKNOWN_ASAP_PACING_DURATION";
  AsapPacingWindowType2["FIXED_15_MINUTE"] = "FIXED_15_MINUTE";
  return AsapPacingWindowType2;
})(AsapPacingWindowType || {});
var ActivationType = /* @__PURE__ */ ((ActivationType2) => {
  ActivationType2["UNKNOWN_SCHEDULE"] = "UNKNOWN_SCHEDULE";
  ActivationType2["ALWAYS_ACTIVE"] = "ALWAYS_ACTIVE";
  ActivationType2["SCHEDULE"] = "SCHEDULE";
  return ActivationType2;
})(ActivationType || {});
var EntitiesDayOfWeek = /* @__PURE__ */ ((EntitiesDayOfWeek2) => {
  EntitiesDayOfWeek2["MON"] = "MON";
  EntitiesDayOfWeek2["TUE"] = "TUE";
  EntitiesDayOfWeek2["WED"] = "WED";
  EntitiesDayOfWeek2["THU"] = "THU";
  EntitiesDayOfWeek2["FRI"] = "FRI";
  EntitiesDayOfWeek2["SAT"] = "SAT";
  EntitiesDayOfWeek2["SUN"] = "SUN";
  return EntitiesDayOfWeek2;
})(EntitiesDayOfWeek || {});
var MethodType = /* @__PURE__ */ ((MethodType2) => {
  MethodType2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  MethodType2["TIME_BOUNDED"] = "TIME_BOUNDED";
  MethodType2["WEEKLY_SCHEDULE"] = "WEEKLY_SCHEDULE";
  return MethodType2;
})(MethodType || {});
var FulfillmentTimesType = /* @__PURE__ */ ((FulfillmentTimesType2) => {
  FulfillmentTimesType2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  FulfillmentTimesType2["TIME_WINDOWS"] = "TIME_WINDOWS";
  return FulfillmentTimesType2;
})(FulfillmentTimesType || {});
var PreorderPacingWindowType = /* @__PURE__ */ ((PreorderPacingWindowType2) => {
  PreorderPacingWindowType2["UNKNOWN_PREORDER_PACING_WINDOW"] = "UNKNOWN_PREORDER_PACING_WINDOW";
  PreorderPacingWindowType2["DAILY"] = "DAILY";
  return PreorderPacingWindowType2;
})(PreorderPacingWindowType || {});
var OnlineOrderingStatusType = /* @__PURE__ */ ((OnlineOrderingStatusType2) => {
  OnlineOrderingStatusType2["UNDEFINED_ONLINE_ORDERING_STATUS"] = "UNDEFINED_ONLINE_ORDERING_STATUS";
  OnlineOrderingStatusType2["ENABLED"] = "ENABLED";
  OnlineOrderingStatusType2["DISABLED"] = "DISABLED";
  OnlineOrderingStatusType2["PAUSED_UNTIL"] = "PAUSED_UNTIL";
  return OnlineOrderingStatusType2;
})(OnlineOrderingStatusType || {});
var FulfillmentType = /* @__PURE__ */ ((FulfillmentType2) => {
  FulfillmentType2["UNDEFINED_FULFILLMENT_TYPE"] = "UNDEFINED_FULFILLMENT_TYPE";
  FulfillmentType2["PICKUP"] = "PICKUP";
  FulfillmentType2["DELIVERY"] = "DELIVERY";
  return FulfillmentType2;
})(FulfillmentType || {});
var PreparationTimePreparationTimeType = /* @__PURE__ */ ((PreparationTimePreparationTimeType2) => {
  PreparationTimePreparationTimeType2["UNKNOWN_PREPARATION_TIME"] = "UNKNOWN_PREPARATION_TIME";
  PreparationTimePreparationTimeType2["MAX_TIME"] = "MAX_TIME";
  PreparationTimePreparationTimeType2["TIME_RANGE"] = "TIME_RANGE";
  return PreparationTimePreparationTimeType2;
})(PreparationTimePreparationTimeType || {});
var AsapFutureHandlingType = /* @__PURE__ */ ((AsapFutureHandlingType2) => {
  AsapFutureHandlingType2["UNKNOWN_ASAP_FUTURE_HANDLING"] = "UNKNOWN_ASAP_FUTURE_HANDLING";
  AsapFutureHandlingType2["NO_FUTURE_HANDLING"] = "NO_FUTURE_HANDLING";
  AsapFutureHandlingType2["BUSINESS_DAYS_AHEAD_HANDLING"] = "BUSINESS_DAYS_AHEAD_HANDLING";
  return AsapFutureHandlingType2;
})(AsapFutureHandlingType || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var SubdivisionType = /* @__PURE__ */ ((SubdivisionType2) => {
  SubdivisionType2["UNKNOWN_SUBDIVISION_TYPE"] = "UNKNOWN_SUBDIVISION_TYPE";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_1"] = "ADMINISTRATIVE_AREA_LEVEL_1";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_2"] = "ADMINISTRATIVE_AREA_LEVEL_2";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_3"] = "ADMINISTRATIVE_AREA_LEVEL_3";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_4"] = "ADMINISTRATIVE_AREA_LEVEL_4";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_5"] = "ADMINISTRATIVE_AREA_LEVEL_5";
  SubdivisionType2["COUNTRY"] = "COUNTRY";
  return SubdivisionType2;
})(SubdivisionType || {});
var FulfillmentTimeType = /* @__PURE__ */ ((FulfillmentTimeType2) => {
  FulfillmentTimeType2["UNDEFINED_FULFILLMENT_TIME"] = "UNDEFINED_FULFILLMENT_TIME";
  FulfillmentTimeType2["MAX_TIME"] = "MAX_TIME";
  FulfillmentTimeType2["DURATION_RANGE"] = "DURATION_RANGE";
  return FulfillmentTimeType2;
})(FulfillmentTimeType || {});
var FulfillmentTimesDisplayType = /* @__PURE__ */ ((FulfillmentTimesDisplayType2) => {
  FulfillmentTimesDisplayType2["UNDEFINED_FULFILLMENT_TIMES_DISPLAY"] = "UNDEFINED_FULFILLMENT_TIMES_DISPLAY";
  FulfillmentTimesDisplayType2["TIME_WINDOWS"] = "TIME_WINDOWS";
  return FulfillmentTimesDisplayType2;
})(FulfillmentTimesDisplayType || {});
var TimeSlotStatus = /* @__PURE__ */ ((TimeSlotStatus2) => {
  TimeSlotStatus2["AVAILABLE"] = "AVAILABLE";
  TimeSlotStatus2["BLOCKED"] = "BLOCKED";
  return TimeSlotStatus2;
})(TimeSlotStatus || {});
var OrderSchedulingType = /* @__PURE__ */ ((OrderSchedulingType2) => {
  OrderSchedulingType2["ASAP"] = "ASAP";
  OrderSchedulingType2["PREORDER"] = "PREORDER";
  return OrderSchedulingType2;
})(OrderSchedulingType || {});
var TimeSlotForFulfillmentTimeSlotStatus = /* @__PURE__ */ ((TimeSlotForFulfillmentTimeSlotStatus2) => {
  TimeSlotForFulfillmentTimeSlotStatus2["AVAILABLE"] = "AVAILABLE";
  TimeSlotForFulfillmentTimeSlotStatus2["BLOCKED"] = "BLOCKED";
  return TimeSlotForFulfillmentTimeSlotStatus2;
})(TimeSlotForFulfillmentTimeSlotStatus || {});
var ViolationType = /* @__PURE__ */ ((ViolationType2) => {
  ViolationType2["UNKNOWN"] = "UNKNOWN";
  ViolationType2["NO_ADDRESS"] = "NO_ADDRESS";
  ViolationType2["MISSING_FORMATTED_ADDRESS"] = "MISSING_FORMATTED_ADDRESS";
  ViolationType2["INVALID_GEOCODE"] = "INVALID_GEOCODE";
  ViolationType2["MISSING_COUNTRY"] = "MISSING_COUNTRY";
  ViolationType2["MISSING_SUBDIVISION"] = "MISSING_SUBDIVISION";
  return ViolationType2;
})(ViolationType || {});
var PlacementType = /* @__PURE__ */ ((PlacementType2) => {
  PlacementType2["BEFORE"] = "BEFORE";
  PlacementType2["AFTER"] = "AFTER";
  PlacementType2["REPLACE"] = "REPLACE";
  return PlacementType2;
})(PlacementType || {});
var DayOfWeek = /* @__PURE__ */ ((DayOfWeek2) => {
  DayOfWeek2["MONDAY"] = "MONDAY";
  DayOfWeek2["TUESDAY"] = "TUESDAY";
  DayOfWeek2["WEDNESDAY"] = "WEDNESDAY";
  DayOfWeek2["THURSDAY"] = "THURSDAY";
  DayOfWeek2["FRIDAY"] = "FRIDAY";
  DayOfWeek2["SATURDAY"] = "SATURDAY";
  DayOfWeek2["SUNDAY"] = "SUNDAY";
  return DayOfWeek2;
})(DayOfWeek || {});
var ResolutionMethod = /* @__PURE__ */ ((ResolutionMethod2) => {
  ResolutionMethod2["QUERY_PARAM"] = "QUERY_PARAM";
  ResolutionMethod2["SUBDOMAIN"] = "SUBDOMAIN";
  ResolutionMethod2["SUBDIRECTORY"] = "SUBDIRECTORY";
  return ResolutionMethod2;
})(ResolutionMethod || {});
var State = /* @__PURE__ */ ((State2) => {
  State2["UNKNOWN"] = "UNKNOWN";
  State2["ENABLED"] = "ENABLED";
  State2["DISABLED"] = "DISABLED";
  State2["PENDING"] = "PENDING";
  State2["DEMO"] = "DEMO";
  return State2;
})(State || {});
var SiteCreatedContext = /* @__PURE__ */ ((SiteCreatedContext2) => {
  SiteCreatedContext2["OTHER"] = "OTHER";
  SiteCreatedContext2["FROM_TEMPLATE"] = "FROM_TEMPLATE";
  SiteCreatedContext2["DUPLICATE_BY_SITE_TRANSFER"] = "DUPLICATE_BY_SITE_TRANSFER";
  SiteCreatedContext2["DUPLICATE"] = "DUPLICATE";
  SiteCreatedContext2["OLD_SITE_TRANSFER"] = "OLD_SITE_TRANSFER";
  SiteCreatedContext2["FLASH"] = "FLASH";
  return SiteCreatedContext2;
})(SiteCreatedContext || {});
var Namespace = /* @__PURE__ */ ((Namespace2) => {
  Namespace2["UNKNOWN_NAMESPACE"] = "UNKNOWN_NAMESPACE";
  Namespace2["WIX"] = "WIX";
  Namespace2["SHOUT_OUT"] = "SHOUT_OUT";
  Namespace2["ALBUMS"] = "ALBUMS";
  Namespace2["WIX_STORES_TEST_DRIVE"] = "WIX_STORES_TEST_DRIVE";
  Namespace2["HOTELS"] = "HOTELS";
  Namespace2["CLUBS"] = "CLUBS";
  Namespace2["ONBOARDING_DRAFT"] = "ONBOARDING_DRAFT";
  Namespace2["DEV_SITE"] = "DEV_SITE";
  Namespace2["LOGOS"] = "LOGOS";
  Namespace2["VIDEO_MAKER"] = "VIDEO_MAKER";
  Namespace2["PARTNER_DASHBOARD"] = "PARTNER_DASHBOARD";
  Namespace2["DEV_CENTER_COMPANY"] = "DEV_CENTER_COMPANY";
  Namespace2["HTML_DRAFT"] = "HTML_DRAFT";
  Namespace2["SITELESS_BUSINESS"] = "SITELESS_BUSINESS";
  Namespace2["CREATOR_ECONOMY"] = "CREATOR_ECONOMY";
  Namespace2["DASHBOARD_FIRST"] = "DASHBOARD_FIRST";
  Namespace2["ANYWHERE"] = "ANYWHERE";
  Namespace2["HEADLESS"] = "HEADLESS";
  Namespace2["ACCOUNT_MASTER_CMS"] = "ACCOUNT_MASTER_CMS";
  Namespace2["RISE"] = "RISE";
  Namespace2["BRANDED_FIRST"] = "BRANDED_FIRST";
  Namespace2["NOWNIA"] = "NOWNIA";
  Namespace2["UGC_TEMPLATE"] = "UGC_TEMPLATE";
  Namespace2["CODUX"] = "CODUX";
  Namespace2["MEDIA_DESIGN_CREATOR"] = "MEDIA_DESIGN_CREATOR";
  Namespace2["SHARED_BLOG_ENTERPRISE"] = "SHARED_BLOG_ENTERPRISE";
  Namespace2["STANDALONE_FORMS"] = "STANDALONE_FORMS";
  Namespace2["STANDALONE_EVENTS"] = "STANDALONE_EVENTS";
  Namespace2["MIMIR"] = "MIMIR";
  Namespace2["TWINS"] = "TWINS";
  Namespace2["NANO"] = "NANO";
  return Namespace2;
})(Namespace || {});
var DeleteStatus = /* @__PURE__ */ ((DeleteStatus2) => {
  DeleteStatus2["UNKNOWN"] = "UNKNOWN";
  DeleteStatus2["TRASH"] = "TRASH";
  DeleteStatus2["DELETED"] = "DELETED";
  DeleteStatus2["PENDING_PURGE"] = "PENDING_PURGE";
  DeleteStatus2["PURGED_EXTERNALLY"] = "PURGED_EXTERNALLY";
  return DeleteStatus2;
})(DeleteStatus || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/restaurants-operations-v1-operation-operations.meta.ts
function createOperation2() {
  const payload = {};
  const getRequestOptions = createOperation(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/operations",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getOperation2() {
  const payload = { operationId: ":operationId" };
  const getRequestOptions = getOperation(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/operations/{operationId}",
    pathParams: { operationId: "operationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateOperation2() {
  const payload = { operation: { id: ":operationId" } };
  const getRequestOptions = updateOperation(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/operations/{operation.id}",
    pathParams: { operationId: "operationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteOperation2() {
  const payload = { operationId: ":operationId" };
  const getRequestOptions = deleteOperation(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/operations/{operationId}",
    pathParams: { operationId: "operationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryOperation2() {
  const payload = {};
  const getRequestOptions = queryOperation(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/operations/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listOperations2() {
  const payload = {};
  const getRequestOptions = listOperations(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/operations",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listAvailableFulfillmentOptions2() {
  const payload = { operationId: ":operationId" };
  const getRequestOptions = listAvailableFulfillmentOptions(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/operations/{operationId}/available-fulfillment-options",
    pathParams: { operationId: "operationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listFirstAvailableTimeSlotForFulfillmentTypes2() {
  const payload = { operationId: ":operationId" };
  const getRequestOptions = listFirstAvailableTimeSlotForFulfillmentTypes(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/operations/{operationId}/first-available-time-slot-for-fulfillment-types",
    pathParams: { operationId: "operationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function calculateFirstAvailableTimeSlotPerFulfillmentType2() {
  const payload = { operationId: ":operationId" };
  const getRequestOptions = calculateFirstAvailableTimeSlotPerFulfillmentType(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/operations/{operationId}/first-available-time-slot-per-fulfillment-type",
    pathParams: { operationId: "operationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listFirstAvailableTimeSlotsForOperations2() {
  const payload = {};
  const getRequestOptions = listFirstAvailableTimeSlotsForOperations(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/operations/first-available-time-slots-for-operations",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function calculateFirstAvailableTimeSlotsPerOperation2() {
  const payload = {};
  const getRequestOptions = calculateFirstAvailableTimeSlotsPerOperation(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/operations/first-available-time-slots-per-operations",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listFirstAvailableTimeSlotsForMenus2() {
  const payload = { operationId: ":operationId" };
  const getRequestOptions = listFirstAvailableTimeSlotsForMenus(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/operations/{operationId}/first-available-time-slots-for-menus",
    pathParams: { operationId: "operationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function calculateFirstAvailableTimeSlotsPerMenu2() {
  const payload = { operationId: ":operationId" };
  const getRequestOptions = calculateFirstAvailableTimeSlotsPerMenu(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/operations/{operationId}/first-available-time-slots-per-menus",
    pathParams: { operationId: "operationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function calculateFirstAvailableTimeSlotsForItem2() {
  const payload = { operationId: ":operationId" };
  const getRequestOptions = calculateFirstAvailableTimeSlotsForItem(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/operations/{operationId}/first-available-time-slots-per-item",
    pathParams: { operationId: "operationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listAvailableTimeSlotsForDate2() {
  const payload = { operationId: ":operationId" };
  const getRequestOptions = listAvailableTimeSlotsForDate(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/operations/{operationId}/available-time-slots-for-date",
    pathParams: { operationId: "operationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function calculateAvailableTimeSlotsForDate2() {
  const payload = { operationId: ":operationId" };
  const getRequestOptions = calculateAvailableTimeSlotsForDate(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/operations/{operationId}/available-time-slots-per-date",
    pathParams: { operationId: "operationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listAvailableDatesInRange2() {
  const payload = { operationId: ":operationId" };
  const getRequestOptions = listAvailableDatesInRange(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/operations/{operationId}/available-dates-in-range",
    pathParams: { operationId: "operationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function calculateAvailableDatesInRange2() {
  const payload = { operationId: ":operationId" };
  const getRequestOptions = calculateAvailableDatesInRange(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/operations/{operationId}/available-dates",
    pathParams: { operationId: "operationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function validateOperationAddress2() {
  const payload = { operationId: ":operationId" };
  const getRequestOptions = validateOperationAddress(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/operations/{operationId}/validate-address",
    pathParams: { operationId: "operationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateOperationTags2() {
  const payload = {};
  const getRequestOptions = bulkUpdateOperationTags(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/operations/update-tags",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateOperationTagsByFilter2() {
  const payload = {};
  const getRequestOptions = bulkUpdateOperationTagsByFilter(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/operations/update-tags-by-filter",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ActivationTypeOriginal,
  AsapFutureHandlingTypeOriginal,
  AsapPacingWindowTypeOriginal,
  AsapPreorderTypeOriginal,
  DayOfWeekOriginal,
  DeleteStatusOriginal,
  EntitiesDayOfWeekOriginal,
  FulfillmentTimeTypeOriginal,
  FulfillmentTimesDisplayTypeOriginal,
  FulfillmentTimesTypeOriginal,
  FulfillmentTypeOriginal,
  MethodTypeOriginal,
  NamespaceOriginal,
  OnlineOrderingStatusTypeOriginal,
  OrderSchedulingTypeOriginal,
  PlacementTypeOriginal,
  PreorderPacingWindowTypeOriginal,
  PreparationTimePreparationTimeTypeOriginal,
  PreparationTimeTypeOriginal,
  ResolutionMethodOriginal,
  SchedulingTypeOriginal,
  SiteCreatedContextOriginal,
  SortOrderOriginal,
  StateOriginal,
  SubdivisionTypeOriginal,
  TimeSlotForFulfillmentTimeSlotStatusOriginal,
  TimeSlotStatusOriginal,
  TimeUnitOriginal,
  ViolationTypeOriginal,
  WebhookIdentityTypeOriginal,
  bulkUpdateOperationTags,
  bulkUpdateOperationTagsByFilter,
  calculateAvailableDatesInRange,
  calculateAvailableTimeSlotsForDate,
  calculateFirstAvailableTimeSlotPerFulfillmentType,
  calculateFirstAvailableTimeSlotsForItem,
  calculateFirstAvailableTimeSlotsPerMenu,
  calculateFirstAvailableTimeSlotsPerOperation,
  createOperation,
  deleteOperation,
  getOperation,
  listAvailableDatesInRange,
  listAvailableFulfillmentOptions,
  listAvailableTimeSlotsForDate,
  listFirstAvailableTimeSlotForFulfillmentTypes,
  listFirstAvailableTimeSlotsForMenus,
  listFirstAvailableTimeSlotsForOperations,
  listOperations,
  queryOperation,
  updateOperation,
  validateOperationAddress
});
