// @wix/auto_sdk_payments_wix-payments-account/build/internal/es/index.typings.mjs
// src/payments-mcp-v1-wix-payments-account-wix-payments-account.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/payments-mcp-v1-wix-payments-account-wix-payments-account.http.ts
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixPaymentsMcpV1WixPaymentsAccountServiceUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/payments/v1/wix-payments-account",
        destPath: "/v1/wix-payments-account"
      },
      {
        srcPath: "/payments/mcp/v1/wix-payments-account",
        destPath: "/v1/wix-payments-account"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_wix-payments-account";
function connectWixPaymentsAccount(payload) {
  function __connectWixPaymentsAccount({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.mcp.v1.wix_payments_account",
      method: "POST",
      methodFqn: "wix.payments.mcp.v1.WixPaymentsAccountService.ConnectWixPaymentsAccount",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsMcpV1WixPaymentsAccountServiceUrl({
        protoPath: "/v1/wix-payments-account/connect",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __connectWixPaymentsAccount;
}

// src/payments-mcp-v1-wix-payments-account-wix-payments-account.universal.ts
async function connectWixPaymentsAccount2(account) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ account });
  const reqOpts = connectWixPaymentsAccount(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { account: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["account"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
export {
  connectWixPaymentsAccount2 as connectWixPaymentsAccount
};
