// @wix/auto_sdk_payments_wix-payments-account/build/internal/es/meta.mjs
// src/payments-mcp-v1-wix-payments-account-wix-payments-account.http.ts
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixPaymentsMcpV1WixPaymentsAccountServiceUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/payments/v1/wix-payments-account",
        destPath: "/v1/wix-payments-account"
      },
      {
        srcPath: "/payments/mcp/v1/wix-payments-account",
        destPath: "/v1/wix-payments-account"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_wix-payments-account";
function connectWixPaymentsAccount(payload) {
  function __connectWixPaymentsAccount({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.mcp.v1.wix_payments_account",
      method: "POST",
      methodFqn: "wix.payments.mcp.v1.WixPaymentsAccountService.ConnectWixPaymentsAccount",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsMcpV1WixPaymentsAccountServiceUrl({
        protoPath: "/v1/wix-payments-account/connect",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __connectWixPaymentsAccount;
}

// src/payments-mcp-v1-wix-payments-account-wix-payments-account.meta.ts
function connectWixPaymentsAccount2() {
  const payload = {};
  const getRequestOptions = connectWixPaymentsAccount(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/wix-payments-account/connect",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  connectWixPaymentsAccount2 as connectWixPaymentsAccount
};
