// @wix/ricos/dist/es/plugin-table/toolbar/modals/BorderSelectionDropdown.js
import {
  AllBorders,
  AllBordersSmall,
  SquareRatio,
  SquareRatioSmall
} from "@wix/wix-ui-icons-common";
import React, { useContext } from "react";
import { RicosContext } from "./../../../ricos-context/index.js";
import { useModalService } from "./../../../ricos-modals/modal-service-provider.js";
import { DropdownModal } from "./../../../_shared/dropdown-modal/dropdown-modal.js";
import { ListItemSelect } from "./../../../_shared/list-item-select/list-item-select.js";
import { BORDER_TYPES, TABLE_BUTTONS_MODALS_ID } from "../../consts.js";
import { BORDER_COLOR_ALL, BORDER_COLOR_AROUND } from "./data-hooks.js";
const BorderSelectionDropdown = ({ onClick }) => {
  const { t, isMobile } = useContext(RicosContext);
  const modalService = useModalService();
  return /* @__PURE__ */ React.createElement(
    DropdownModal,
    {
      onCloseButtonClick: () => modalService.closeModal(TABLE_BUTTONS_MODALS_ID.BORDER),
      title: t("TablePlugin_Toolbar_BorderPanel_Header_Title"),
      options: [
        /* @__PURE__ */ React.createElement(
          ListItemSelect,
          {
            key: "border",
            dataHook: BORDER_COLOR_AROUND,
            prefix: isMobile ? /* @__PURE__ */ React.createElement(SquareRatio, null) : /* @__PURE__ */ React.createElement(SquareRatioSmall, null),
            title: t("TablePlugin_Toolbar_BorderPanel_OutsideBorder"),
            onSelect: () => onClick(BORDER_TYPES.outsideBorders)
          }
        ),
        /* @__PURE__ */ React.createElement(
          ListItemSelect,
          {
            key: "outsideBorder",
            dataHook: BORDER_COLOR_ALL,
            prefix: isMobile ? /* @__PURE__ */ React.createElement(AllBorders, null) : /* @__PURE__ */ React.createElement(AllBordersSmall, null),
            title: t("TablePlugin_Toolbar_BorderPanel_AllBorders"),
            onSelect: () => onClick(BORDER_TYPES.borders)
          }
        )
      ]
    }
  );
};
export {
  BorderSelectionDropdown
};
