// @wix/auto-patterns/dist/cjs/components/AutoPatternsApp/AutoPatternsApp.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.AutoPatternsApp = void 0;
var _react = _interopRequireDefault(require("react"));
var _RootAppProvider = require("../../providers/RootAppProvider");
var _AutoPatternsRoutes = require("../AutoPatternsRoute/AutoPatternsRoutes");
var _jsxFileName = "/home/builduser/work/73d19ce3378ce4dd/packages/auto-patterns/dist/cjs/components/AutoPatternsApp/AutoPatternsApp.tsx";
const AutoPatternsApp = ({
  configuration
}) => {
  return /*#__PURE__*/_react.default.createElement(_RootAppProvider.RootAppProvider, {
    appConfig: configuration,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 14,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(_AutoPatternsRoutes.AutoPatternsRoutes, {
    config: configuration,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 15,
      columnNumber: 7
    }
  }));
};
exports.AutoPatternsApp = AutoPatternsApp;
