// @wix/ricos/dist/cjs/ricos-editor/toolbars/utils/biUtils.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var biUtils_exports = {};
__export(biUtils_exports, {
  getBiFunctions: () => getBiFunctions
});
module.exports = __toCommonJS(biUtils_exports);
var import_ricos_content = require("./../../../ricos-content/index.js");
var import_wix_rich_content_common = require("./../../../wix-rich-content-common/index.js");
const getBiFunctions = (helpers, contentId) => ({
  onInlineToolbarOpen: (toolbarType) => helpers?.onInlineToolbarOpen?.({
    toolbarType,
    version: import_ricos_content.Version.currentVersion,
    contentId
  }),
  onToolbarButtonClick: (name, toolbarType, value, pluginId) => {
    helpers?.onToolbarButtonClick?.({
      buttonName: name,
      pluginId,
      type: toolbarType,
      value: value === void 0 ? void 0 : typeof value === "boolean" ? `${!value}` : value,
      version: import_ricos_content.Version.currentVersion,
      contentId
    });
    if (pluginId && value && typeof value === "string") {
      helpers?.onPluginAdd?.(pluginId, "FormattingToolbar", import_ricos_content.Version.currentVersion, contentId);
      helpers?.onPluginAddSuccess?.(
        pluginId,
        "FormattingToolbar",
        value,
        import_ricos_content.Version.currentVersion,
        contentId
      );
    }
  },
  onAddPluginLink: (data, plugin_id) => {
    const { url, anchor, target, rel } = data;
    const params = anchor ? { anchor, category: import_wix_rich_content_common.ANCHOR_CATEGORY } : {
      link: url,
      rel,
      newTab: target === "_blank",
      category: import_wix_rich_content_common.WEB_ADDRESS_CATEGORY
    };
    helpers.onPluginAction?.(import_wix_rich_content_common.ADD_PLUGIN_LINK_BI, {
      plugin_id,
      params,
      version: import_ricos_content.Version.currentVersion,
      contentId
    });
  }
});
