// @wix/auto_sdk_pricing-plans_plans-v-3/build/internal/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  AggregationType: () => AggregationType,
  AppliedAt: () => AppliedAt,
  Cascade: () => Cascade,
  DayOfWeek: () => DayOfWeek,
  EndType: () => EndType,
  Interval: () => Interval,
  MissingValues: () => MissingValues,
  Mode: () => Mode,
  NestedAggregationType: () => NestedAggregationType,
  Period: () => Period,
  PlacementType: () => PlacementType,
  PriceType: () => PriceType,
  RecurrenceType: () => RecurrenceType,
  ResolutionMethod: () => ResolutionMethod,
  RulesType: () => RulesType,
  ScalarType: () => ScalarType,
  SortDirection: () => SortDirection,
  SortOrder: () => SortOrder,
  SortType: () => SortType,
  StartType: () => StartType,
  Status: () => Status,
  TaxableAddress: () => TaxableAddress,
  Type: () => Type,
  V1DayOfWeek: () => V1DayOfWeek,
  ValidationStrategy: () => ValidationStrategy,
  Visibility: () => Visibility,
  WebhookIdentityType: () => WebhookIdentityType,
  bulkUpdatePlans: () => bulkUpdatePlans2,
  changePrice: () => changePrice2,
  countPlans: () => countPlans2,
  createPlan: () => createPlan2,
  deletePlan: () => deletePlan2,
  getPlan: () => getPlan2,
  queryPlans: () => queryPlans2,
  searchPlans: () => searchPlans2,
  updatePlan: () => updatePlan2
});
module.exports = __toCommonJS(index_typings_exports);

// src/pricing-plans-v3-plan-plans-v-3.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/pricing-plans-v3-plan-plans-v-3.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixPricingPlansPlansV3PlanServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/pricing-plans/v3",
        destPath: "/v3"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/pricing-plans/v3",
        destPath: "/v3"
      }
    ],
    _: [
      {
        srcPath: "/_api/pricing-plans/v3",
        destPath: "/v3"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/pricing-plans/v3",
        destPath: "/v3"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/pricing-plans",
        destPath: "/v3"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_pricing-plans_plans-v-3";
function createPlan(payload) {
  function __createPlan({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.pricing_plans.v3.plan",
      method: "POST",
      methodFqn: "wix.pricing_plans.plans.v3.PlanService.CreatePlan",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixPricingPlansPlansV3PlanServiceUrl({
        protoPath: "/v3/plans",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __createPlan;
}
function getPlan(payload) {
  function __getPlan({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v3.plan",
      method: "GET",
      methodFqn: "wix.pricing_plans.plans.v3.PlanService.GetPlan",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixPricingPlansPlansV3PlanServiceUrl({
        protoPath: "/v3/plans/{planId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __getPlan;
}
function updatePlan(payload) {
  function __updatePlan({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.pricing_plans.v3.plan",
      method: "PATCH",
      methodFqn: "wix.pricing_plans.plans.v3.PlanService.UpdatePlan",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixPricingPlansPlansV3PlanServiceUrl({
        protoPath: "/v3/plans/{plan.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __updatePlan;
}
function bulkUpdatePlans(payload) {
  function __bulkUpdatePlans({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "plans.fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "plans.plan.createdDate" },
          { path: "plans.plan.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.pricing_plans.v3.plan",
      method: "POST",
      methodFqn: "wix.pricing_plans.plans.v3.PlanService.BulkUpdatePlans",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixPricingPlansPlansV3PlanServiceUrl({
        protoPath: "/v3/bulk/plans/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdatePlans;
}
function deletePlan(payload) {
  function __deletePlan({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v3.plan",
      method: "DELETE",
      methodFqn: "wix.pricing_plans.plans.v3.PlanService.DeletePlan",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixPricingPlansPlansV3PlanServiceUrl({
        protoPath: "/v3/plans/{planId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deletePlan;
}
function queryPlans(payload) {
  function __queryPlans({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v3.plan",
      method: "GET",
      methodFqn: "wix.pricing_plans.plans.v3.PlanService.QueryPlans",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixPricingPlansPlansV3PlanServiceUrl({
        protoPath: "/v3/plans/query",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload, true),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "plans.createdDate" },
            { path: "plans.updatedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixPricingPlansPlansV3PlanServiceUrl({
            protoPath: "/v3/plans/query",
            data: payload,
            host
          }),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryPlans;
}
function searchPlans(payload) {
  function __searchPlans({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "search.aggregations.range.buckets.from" },
          { path: "search.aggregations.range.buckets.to" },
          {
            path: "search.aggregations.nested.nestedAggregations.range.buckets.from"
          },
          {
            path: "search.aggregations.nested.nestedAggregations.range.buckets.to"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.pricing_plans.v3.plan",
      method: "GET",
      methodFqn: "wix.pricing_plans.plans.v3.PlanService.SearchPlans",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixPricingPlansPlansV3PlanServiceUrl({
        protoPath: "/v3/plans/search",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData, true),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "plans.createdDate" },
            { path: "plans.updatedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixPricingPlansPlansV3PlanServiceUrl({
            protoPath: "/v3/plans/search",
            data: serializedData,
            host
          }),
          data: serializedData
        }
      ]
    };
    return metadata;
  }
  return __searchPlans;
}
function countPlans(payload) {
  function __countPlans({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v3.plan",
      method: "POST",
      methodFqn: "wix.pricing_plans.plans.v3.PlanService.CountPlans",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixPricingPlansPlansV3PlanServiceUrl({
        protoPath: "/v3/plans/count",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __countPlans;
}
function changePrice(payload) {
  function __changePrice({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v3.plan",
      method: "POST",
      methodFqn: "wix.pricing_plans.plans.v3.PlanService.ChangePrice",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixPricingPlansPlansV3PlanServiceUrl({
        protoPath: "/v3/plans/{planId}/change-price",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __changePrice;
}

// src/pricing-plans-v3-plan-plans-v-3.universal.ts
var import_image = require("@wix/sdk-runtime/transformations/image");
var import_image2 = require("@wix/sdk-runtime/transformations/image");
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var PriceType = /* @__PURE__ */ ((PriceType2) => {
  PriceType2["UNKNOWN_PRICE_TYPE"] = "UNKNOWN_PRICE_TYPE";
  PriceType2["FIXED_AMOUNT"] = "FIXED_AMOUNT";
  return PriceType2;
})(PriceType || {});
var AppliedAt = /* @__PURE__ */ ((AppliedAt2) => {
  AppliedAt2["UNKNOWN_APPLIED_AT"] = "UNKNOWN_APPLIED_AT";
  AppliedAt2["FIRST_PAYMENT"] = "FIRST_PAYMENT";
  return AppliedAt2;
})(AppliedAt || {});
var Period = /* @__PURE__ */ ((Period2) => {
  Period2["UNKNOWN_PERIOD"] = "UNKNOWN_PERIOD";
  Period2["DAY"] = "DAY";
  Period2["WEEK"] = "WEEK";
  Period2["MONTH"] = "MONTH";
  Period2["YEAR"] = "YEAR";
  return Period2;
})(Period || {});
var StartType = /* @__PURE__ */ ((StartType2) => {
  StartType2["UNKNOWN_START_TYPE"] = "UNKNOWN_START_TYPE";
  StartType2["ON_PURCHASE"] = "ON_PURCHASE";
  StartType2["CUSTOM"] = "CUSTOM";
  return StartType2;
})(StartType || {});
var EndType = /* @__PURE__ */ ((EndType2) => {
  EndType2["UNKNOWN_END_TYPE"] = "UNKNOWN_END_TYPE";
  EndType2["UNTIL_CANCELLED"] = "UNTIL_CANCELLED";
  EndType2["CYCLES_COMPLETED"] = "CYCLES_COMPLETED";
  return EndType2;
})(EndType || {});
var Visibility = /* @__PURE__ */ ((Visibility2) => {
  Visibility2["UNKNOWN_VISIBILITY"] = "UNKNOWN_VISIBILITY";
  Visibility2["PUBLIC"] = "PUBLIC";
  Visibility2["PRIVATE"] = "PRIVATE";
  return Visibility2;
})(Visibility || {});
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["UNKNOWN_STATUS"] = "UNKNOWN_STATUS";
  Status2["ACTIVE"] = "ACTIVE";
  return Status2;
})(Status || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  Type2["PLAN"] = "PLAN";
  Type2["ADD_ON"] = "ADD_ON";
  return Type2;
})(Type || {});
var TaxableAddress = /* @__PURE__ */ ((TaxableAddress2) => {
  TaxableAddress2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  TaxableAddress2["BILLING"] = "BILLING";
  TaxableAddress2["BUSINESS"] = "BUSINESS";
  return TaxableAddress2;
})(TaxableAddress || {});
var ValidationStrategy = /* @__PURE__ */ ((ValidationStrategy2) => {
  ValidationStrategy2["UNKNOWN_VALIDATION_STRATEGY"] = "UNKNOWN_VALIDATION_STRATEGY";
  ValidationStrategy2["ON_BEHALF"] = "ON_BEHALF";
  ValidationStrategy2["PREVIEW"] = "PREVIEW";
  return ValidationStrategy2;
})(ValidationStrategy || {});
var RulesType = /* @__PURE__ */ ((RulesType2) => {
  RulesType2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  RulesType2["AVAILABLE_IMMEDIATELY"] = "AVAILABLE_IMMEDIATELY";
  RulesType2["FIXED_DATES"] = "FIXED_DATES";
  RulesType2["RECURRENCE"] = "RECURRENCE";
  return RulesType2;
})(RulesType || {});
var RecurrenceType = /* @__PURE__ */ ((RecurrenceType2) => {
  RecurrenceType2["UNKNOWN_RECURRENCE_TYPE"] = "UNKNOWN_RECURRENCE_TYPE";
  RecurrenceType2["WEEKLY"] = "WEEKLY";
  RecurrenceType2["MONTHLY"] = "MONTHLY";
  return RecurrenceType2;
})(RecurrenceType || {});
var V1DayOfWeek = /* @__PURE__ */ ((V1DayOfWeek2) => {
  V1DayOfWeek2["UNKNOWN_DAY_OF_WEEK"] = "UNKNOWN_DAY_OF_WEEK";
  V1DayOfWeek2["SUNDAY"] = "SUNDAY";
  V1DayOfWeek2["MONDAY"] = "MONDAY";
  V1DayOfWeek2["TUESDAY"] = "TUESDAY";
  V1DayOfWeek2["WEDNESDAY"] = "WEDNESDAY";
  V1DayOfWeek2["THURSDAY"] = "THURSDAY";
  V1DayOfWeek2["FRIDAY"] = "FRIDAY";
  V1DayOfWeek2["SATURDAY"] = "SATURDAY";
  return V1DayOfWeek2;
})(V1DayOfWeek || {});
var PlacementType = /* @__PURE__ */ ((PlacementType2) => {
  PlacementType2["BEFORE"] = "BEFORE";
  PlacementType2["AFTER"] = "AFTER";
  PlacementType2["REPLACE"] = "REPLACE";
  return PlacementType2;
})(PlacementType || {});
var DayOfWeek = /* @__PURE__ */ ((DayOfWeek2) => {
  DayOfWeek2["MONDAY"] = "MONDAY";
  DayOfWeek2["TUESDAY"] = "TUESDAY";
  DayOfWeek2["WEDNESDAY"] = "WEDNESDAY";
  DayOfWeek2["THURSDAY"] = "THURSDAY";
  DayOfWeek2["FRIDAY"] = "FRIDAY";
  DayOfWeek2["SATURDAY"] = "SATURDAY";
  DayOfWeek2["SUNDAY"] = "SUNDAY";
  return DayOfWeek2;
})(DayOfWeek || {});
var ResolutionMethod = /* @__PURE__ */ ((ResolutionMethod2) => {
  ResolutionMethod2["QUERY_PARAM"] = "QUERY_PARAM";
  ResolutionMethod2["SUBDOMAIN"] = "SUBDOMAIN";
  ResolutionMethod2["SUBDIRECTORY"] = "SUBDIRECTORY";
  return ResolutionMethod2;
})(ResolutionMethod || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
var Cascade = /* @__PURE__ */ ((Cascade2) => {
  Cascade2["NEW_SUBSCRIPTIONS"] = "NEW_SUBSCRIPTIONS";
  Cascade2["ALL_SUBSCRIPTIONS"] = "ALL_SUBSCRIPTIONS";
  return Cascade2;
})(Cascade || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var SortType = /* @__PURE__ */ ((SortType2) => {
  SortType2["COUNT"] = "COUNT";
  SortType2["VALUE"] = "VALUE";
  return SortType2;
})(SortType || {});
var SortDirection = /* @__PURE__ */ ((SortDirection2) => {
  SortDirection2["DESC"] = "DESC";
  SortDirection2["ASC"] = "ASC";
  return SortDirection2;
})(SortDirection || {});
var MissingValues = /* @__PURE__ */ ((MissingValues2) => {
  MissingValues2["EXCLUDE"] = "EXCLUDE";
  MissingValues2["INCLUDE"] = "INCLUDE";
  return MissingValues2;
})(MissingValues || {});
var ScalarType = /* @__PURE__ */ ((ScalarType2) => {
  ScalarType2["UNKNOWN_SCALAR_TYPE"] = "UNKNOWN_SCALAR_TYPE";
  ScalarType2["COUNT_DISTINCT"] = "COUNT_DISTINCT";
  ScalarType2["MIN"] = "MIN";
  ScalarType2["MAX"] = "MAX";
  return ScalarType2;
})(ScalarType || {});
var NestedAggregationType = /* @__PURE__ */ ((NestedAggregationType2) => {
  NestedAggregationType2["UNKNOWN_AGGREGATION_TYPE"] = "UNKNOWN_AGGREGATION_TYPE";
  NestedAggregationType2["VALUE"] = "VALUE";
  NestedAggregationType2["RANGE"] = "RANGE";
  NestedAggregationType2["SCALAR"] = "SCALAR";
  NestedAggregationType2["DATE_HISTOGRAM"] = "DATE_HISTOGRAM";
  return NestedAggregationType2;
})(NestedAggregationType || {});
var Interval = /* @__PURE__ */ ((Interval2) => {
  Interval2["UNKNOWN_INTERVAL"] = "UNKNOWN_INTERVAL";
  Interval2["YEAR"] = "YEAR";
  Interval2["MONTH"] = "MONTH";
  Interval2["WEEK"] = "WEEK";
  Interval2["DAY"] = "DAY";
  Interval2["HOUR"] = "HOUR";
  Interval2["MINUTE"] = "MINUTE";
  Interval2["SECOND"] = "SECOND";
  return Interval2;
})(Interval || {});
var AggregationType = /* @__PURE__ */ ((AggregationType2) => {
  AggregationType2["UNKNOWN_AGGREGATION_TYPE"] = "UNKNOWN_AGGREGATION_TYPE";
  AggregationType2["VALUE"] = "VALUE";
  AggregationType2["RANGE"] = "RANGE";
  AggregationType2["SCALAR"] = "SCALAR";
  AggregationType2["DATE_HISTOGRAM"] = "DATE_HISTOGRAM";
  AggregationType2["NESTED"] = "NESTED";
  return AggregationType2;
})(AggregationType || {});
var Mode = /* @__PURE__ */ ((Mode2) => {
  Mode2["OR"] = "OR";
  Mode2["AND"] = "AND";
  return Mode2;
})(Mode || {});
async function createPlan2(plan, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      plan,
      idempotencyKey: options?.idempotencyKey
    }),
    [
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [{ path: "plan.image" }]
      }
    ]
  );
  const reqOpts = createPlan(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "plan.image" }]
        }
      ])
    )?.plan;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          plan: "$[0]",
          idempotencyKey: "$[1].idempotencyKey"
        },
        singleArgumentUnchanged: false
      },
      ["plan", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getPlan2(planId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ planId });
  const reqOpts = getPlan(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "plan.image" }]
        }
      ])
    )?.plan;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { planId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["planId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updatePlan2(_id, plan, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      plan: { ...plan, id: _id },
      cascade: options?.cascade
    }),
    [
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [{ path: "plan.image" }]
      }
    ]
  );
  const reqOpts = updatePlan(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "plan.image" }]
        }
      ])
    )?.plan;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { plan: "$[1]" },
        explicitPathsToArguments: {
          "plan.id": "$[0]",
          cascade: "$[2].cascade"
        },
        singleArgumentUnchanged: false
      },
      ["_id", "plan", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdatePlans2(plans, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      plans,
      returnEntity: options?.returnEntity,
      cascade: options?.cascade
    }),
    [
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [{ path: "plans.plan.image" }]
      }
    ]
  );
  const reqOpts = bulkUpdatePlans(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "results.item.image" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          plans: "$[0]",
          returnEntity: "$[1].returnEntity",
          cascade: "$[1].cascade"
        },
        singleArgumentUnchanged: false
      },
      ["plans", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deletePlan2(planId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ planId });
  const reqOpts = deletePlan(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { planId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["planId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryPlans2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryPlans(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [
          {
            transformFn: import_image2.transformRESTImageToSDKImage,
            paths: [{ path: "plans.image" }]
          }
        ])
      );
      return {
        items: transformedData?.plans,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function searchPlans2(search) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ search });
  const reqOpts = searchPlans(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "plans.image" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { search: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["search"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function countPlans2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    filter: options?.filter,
    search: options?.search
  });
  const reqOpts = countPlans(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          filter: "$[0].filter",
          search: "$[0].search"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function changePrice2(planId, pricingVariantId, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    planId,
    pricingVariantId,
    pricingStrategies: options?.pricingStrategies,
    revision: options?.revision,
    cascade: options?.cascade
  });
  const reqOpts = changePrice(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "plan.image" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          planId: "$[0]",
          pricingVariantId: "$[1]",
          pricingStrategies: "$[2].pricingStrategies",
          revision: "$[2].revision",
          cascade: "$[2].cascade"
        },
        singleArgumentUnchanged: false
      },
      ["planId", "pricingVariantId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AggregationType,
  AppliedAt,
  Cascade,
  DayOfWeek,
  EndType,
  Interval,
  MissingValues,
  Mode,
  NestedAggregationType,
  Period,
  PlacementType,
  PriceType,
  RecurrenceType,
  ResolutionMethod,
  RulesType,
  ScalarType,
  SortDirection,
  SortOrder,
  SortType,
  StartType,
  Status,
  TaxableAddress,
  Type,
  V1DayOfWeek,
  ValidationStrategy,
  Visibility,
  WebhookIdentityType,
  bulkUpdatePlans,
  changePrice,
  countPlans,
  createPlan,
  deletePlan,
  getPlan,
  queryPlans,
  searchPlans,
  updatePlan
});
