// @wix/auto_sdk_seo_seo-sitemap-entries/build/internal/es/index.typings.mjs
// src/promote-seo-v1-headless-sitemap-entry-seo-sitemap-entries.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/promote-seo-v1-headless-sitemap-entry-seo-sitemap-entries.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixPromoteSeoHeadlessSitemapServiceV1SeoHeadlessSitemapServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/seo-headless-sitemap/v1/list-static-sitemap-pages",
        destPath: "/v1/list-static-sitemap-pages"
      }
    ],
    _: [
      {
        srcPath: "/_api/seo-headless-sitemap/v1/list-sitemap-pages",
        destPath: "/v1/list-sitemap-pages"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/seo-headless-sitemap/v1/list-sitemap-pages",
        destPath: "/v1/list-sitemap-pages"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/v1/list-sitemap-pages",
        destPath: "/v1/list-sitemap-pages"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_seo_seo-sitemap-entries";
function listSitemapPages(payload) {
  function __listSitemapPages({ host }) {
    const metadata = {
      entityFqdn: "wix.promote.seo.v1.headless_sitemap_entry",
      method: "GET",
      methodFqn: "wix.promote.seo.headless.sitemap.service.v1.SeoHeadlessSitemapService.ListSitemapPages",
      packageName: PACKAGE_NAME,
      url: resolveWixPromoteSeoHeadlessSitemapServiceV1SeoHeadlessSitemapServiceUrl(
        { protoPath: "/v1/list-sitemap-pages", data: payload, host }
      ),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __listSitemapPages;
}

// src/promote-seo-v1-headless-sitemap-entry-seo-sitemap-entries.universal.ts
var ItemType = /* @__PURE__ */ ((ItemType2) => {
  ItemType2["STORES_PRODUCT"] = "STORES_PRODUCT";
  ItemType2["BLOG_POST"] = "BLOG_POST";
  ItemType2["BLOG_CATEGORY"] = "BLOG_CATEGORY";
  ItemType2["BLOG_TAGS"] = "BLOG_TAGS";
  ItemType2["BLOG_ARCHIVE"] = "BLOG_ARCHIVE";
  ItemType2["FORUM_POST"] = "FORUM_POST";
  ItemType2["FORUM_CATEGORY"] = "FORUM_CATEGORY";
  ItemType2["GROUPS_POST"] = "GROUPS_POST";
  ItemType2["GROUPS_PAGE"] = "GROUPS_PAGE";
  ItemType2["EVENTS_PAGE"] = "EVENTS_PAGE";
  ItemType2["BOOKINGS_SERVICE"] = "BOOKINGS_SERVICE";
  ItemType2["MEMBERS_AREA_PROFILE"] = "MEMBERS_AREA_PROFILE";
  ItemType2["CHALLENGES_PAGE"] = "CHALLENGES_PAGE";
  ItemType2["STORES_CATEGORY"] = "STORES_CATEGORY";
  ItemType2["STORES_SUB_CATEGORY"] = "STORES_SUB_CATEGORY";
  ItemType2["PORTFOLIO_COLLECTIONS"] = "PORTFOLIO_COLLECTIONS";
  ItemType2["PORTFOLIO_PROJECTS"] = "PORTFOLIO_PROJECTS";
  ItemType2["PRICING_PLANS"] = "PRICING_PLANS";
  ItemType2["RESTAURANTS_MENU_PAGE"] = "RESTAURANTS_MENU_PAGE";
  return ItemType2;
})(ItemType || {});
async function listSitemapPages2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    itemType: options?.itemType,
    paging: options?.paging
  });
  const reqOpts = listSitemapPages(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          itemType: "$[0].itemType",
          paging: "$[0].paging"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
export {
  ItemType,
  listSitemapPages2 as listSitemapPages
};
