// @wix/auto_sdk_benefit-programs_items/build/internal/es/meta.mjs
// src/benefit-programs-v1-item-items.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixBenefitProgramsV1ItemItemServiceUrl(opts) {
  const domainToMappings = {
    _: [
      {
        srcPath: "/_api/benefit-programs-items",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-items",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-items",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/benefit-programs-items",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/benefit-programs-items",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/items",
        destPath: "/v1/bulk/items"
      },
      {
        srcPath: "/_api/benefit-programs/v1/items",
        destPath: "/v1/items"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-items",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-items",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/benefit-programs-items",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/benefit-programs/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/benefit-programs/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/benefit-programs/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_benefit-programs_items";
function createItem(payload) {
  function __createItem({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "item.createdDate" }, { path: "item.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.CreateItem",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/items",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "item.createdDate" }, { path: "item.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __createItem;
}
function bulkCreateItems(payload) {
  function __bulkCreateItems({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "items.createdDate" }, { path: "items.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.BulkCreateItems",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/bulk/items/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateItems;
}
function deleteItem(payload) {
  function __deleteItem({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.DeleteItem",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/items/{itemId}/delete",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __deleteItem;
}
function bulkDeleteItems(payload) {
  function __bulkDeleteItems({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.BulkDeleteItems",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/bulk/items/delete",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkDeleteItems;
}
function bulkDeleteItemsByFilter(payload) {
  function __bulkDeleteItemsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.BulkDeleteItemsByFilter",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/bulk/items/delete-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkDeleteItemsByFilter;
}
function updateItem(payload) {
  function __updateItem({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "item.createdDate" }, { path: "item.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "PATCH",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.UpdateItem",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/items/{item.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "item.createdDate" }, { path: "item.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __updateItem;
}
function bulkUpdateItems(payload) {
  function __bulkUpdateItems({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "items.fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "items.item.createdDate" },
          { path: "items.item.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.BulkUpdateItems",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/bulk/items/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateItems;
}
function getItem(payload) {
  function __getItem({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.GetItem",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/items/{itemId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "item.createdDate" }, { path: "item.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __getItem;
}
function listItems(payload) {
  function __listItems({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.ListItems",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/items",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "items.createdDate" },
            { path: "items.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listItems;
}
function queryItems(payload) {
  function __queryItems({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.QueryItems",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/items/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "items.createdDate" },
            { path: "items.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryItems;
}
function countItems(payload) {
  function __countItems({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.CountItems",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/items/count",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __countItems;
}

// src/benefit-programs-v1-item-items.meta.ts
function createItem2() {
  const payload = {};
  const getRequestOptions = createItem(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/items",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkCreateItems2() {
  const payload = {};
  const getRequestOptions = bulkCreateItems(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/items/create",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteItem2() {
  const payload = { itemId: ":itemId" };
  const getRequestOptions = deleteItem(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/items/{itemId}/delete",
    pathParams: { itemId: "itemId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkDeleteItems2() {
  const payload = {};
  const getRequestOptions = bulkDeleteItems(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/items/delete",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkDeleteItemsByFilter2() {
  const payload = {};
  const getRequestOptions = bulkDeleteItemsByFilter(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/items/delete-by-filter",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateItem2() {
  const payload = { item: { id: ":itemId" } };
  const getRequestOptions = updateItem(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/items/{item.id}",
    pathParams: { itemId: "itemId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateItems2() {
  const payload = {};
  const getRequestOptions = bulkUpdateItems(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/items/update",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getItem2() {
  const payload = { itemId: ":itemId" };
  const getRequestOptions = getItem(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/items/{itemId}",
    pathParams: { itemId: "itemId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listItems2() {
  const payload = {};
  const getRequestOptions = listItems(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/items",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryItems2() {
  const payload = {};
  const getRequestOptions = queryItems(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/items/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function countItems2() {
  const payload = {};
  const getRequestOptions = countItems(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/items/count",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  bulkCreateItems2 as bulkCreateItems,
  bulkDeleteItems2 as bulkDeleteItems,
  bulkDeleteItemsByFilter2 as bulkDeleteItemsByFilter,
  bulkUpdateItems2 as bulkUpdateItems,
  countItems2 as countItems,
  createItem2 as createItem,
  deleteItem2 as deleteItem,
  getItem2 as getItem,
  listItems2 as listItems,
  queryItems2 as queryItems,
  updateItem2 as updateItem
};
