// @wix/auto_sdk_restaurants_discount/build/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  DayOfWeek: () => DayOfWeek,
  DiscountPlatform: () => DiscountPlatform,
  DiscountType: () => DiscountType,
  FulfillmentType: () => FulfillmentType,
  WebhookIdentityType: () => WebhookIdentityType,
  createDiscount: () => createDiscount2,
  getDiscount: () => getDiscount2,
  listDiscounts: () => listDiscounts2,
  updateDiscount: () => updateDiscount2
});
module.exports = __toCommonJS(index_typings_exports);

// src/restaurants-catalogs-v3-discount-discount.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/restaurants-catalogs-v3-discount-discount.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressRestaurantsCatalogsV3DiscountsApiUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      },
      {
        srcPath: "/restaurants/v3/catalog",
        destPath: "/v3/catalog"
      },
      {
        srcPath: "/restaurants/v3/draft-catalogs",
        destPath: "/v3/draft-catalogs"
      },
      {
        srcPath: "/restaurants/v3/catalogs_test",
        destPath: "/v3/catalogs"
      },
      {
        srcPath: "/_api/restaurants/v3/catalogs_test",
        destPath: "/v3/catalogs"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    "users._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      },
      {
        srcPath: "/restaurants/v3/catalog",
        destPath: "/v3/catalog"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_discount";
function getDiscount(payload) {
  function __getDiscount({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.discount",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.DiscountsApi.GetDiscount",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3DiscountsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/discounts/{discountId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getDiscount;
}
function updateDiscount(payload) {
  function __updateDiscount({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.discount",
      method: "PATCH",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.DiscountsApi.UpdateDiscount",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3DiscountsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/discounts/{discount.id}",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __updateDiscount;
}
function createDiscount(payload) {
  function __createDiscount({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.discount",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.DiscountsApi.CreateDiscount",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3DiscountsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/discounts",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __createDiscount;
}
function listDiscounts(payload) {
  function __listDiscounts({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.discount",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.DiscountsApi.ListDiscounts",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3DiscountsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/discounts",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData)
    };
    return metadata;
  }
  return __listDiscounts;
}

// src/restaurants-catalogs-v3-discount-discount.universal.ts
var DiscountType = /* @__PURE__ */ ((DiscountType2) => {
  DiscountType2["UNSPECIFIED_TYPE"] = "UNSPECIFIED_TYPE";
  DiscountType2["OFF_ITEM"] = "OFF_ITEM";
  DiscountType2["OFF_ORDER"] = "OFF_ORDER";
  return DiscountType2;
})(DiscountType || {});
var FulfillmentType = /* @__PURE__ */ ((FulfillmentType2) => {
  FulfillmentType2["UNSPECIFIED_FULFILLMENT_TYPE"] = "UNSPECIFIED_FULFILLMENT_TYPE";
  FulfillmentType2["DELIVERY"] = "DELIVERY";
  FulfillmentType2["PICKUP_OR_DINE_IN"] = "PICKUP_OR_DINE_IN";
  return FulfillmentType2;
})(FulfillmentType || {});
var DiscountPlatform = /* @__PURE__ */ ((DiscountPlatform2) => {
  DiscountPlatform2["UNSPECIFIED_PLATFORM"] = "UNSPECIFIED_PLATFORM";
  DiscountPlatform2["SITE"] = "SITE";
  DiscountPlatform2["MOBILE_SITE"] = "MOBILE_SITE";
  DiscountPlatform2["CALL_CENTER"] = "CALL_CENTER";
  return DiscountPlatform2;
})(DiscountPlatform || {});
var DayOfWeek = /* @__PURE__ */ ((DayOfWeek2) => {
  DayOfWeek2["UNDEFINED"] = "UNDEFINED";
  DayOfWeek2["SUN"] = "SUN";
  DayOfWeek2["MON"] = "MON";
  DayOfWeek2["TUE"] = "TUE";
  DayOfWeek2["WED"] = "WED";
  DayOfWeek2["THU"] = "THU";
  DayOfWeek2["FRI"] = "FRI";
  DayOfWeek2["SAT"] = "SAT";
  return DayOfWeek2;
})(DayOfWeek || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function getDiscount2(identifiers) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogId: identifiers?.catalogId,
    discountId: identifiers?.discountId
  });
  const reqOpts = getDiscount(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.discount;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          catalogId: "$[0].catalogId",
          discountId: "$[0].discountId"
        },
        singleArgumentUnchanged: false
      },
      ["identifiers"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateDiscount2(identifiers, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogId: identifiers?.catalogId,
    discount: { ...options?.discount, id: identifiers?.discountId }
  });
  const reqOpts = updateDiscount(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.discount;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { discount: "$[1].discount" },
        explicitPathsToArguments: {
          catalogId: "$[0].catalogId",
          "discount.id": "$[0].discountId"
        },
        singleArgumentUnchanged: false
      },
      ["identifiers", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function createDiscount2(catalogId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogId,
    discount: options?.discount
  });
  const reqOpts = createDiscount(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.discount;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          catalogId: "$[0]",
          discount: "$[1].discount"
        },
        singleArgumentUnchanged: false
      },
      ["catalogId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listDiscounts2(catalogId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogId,
    fieldMask: options?.fieldMask,
    active: options?.active
  });
  const reqOpts = listDiscounts(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          catalogId: "$[0]",
          fieldMask: "$[1].fieldMask",
          active: "$[1].active"
        },
        singleArgumentUnchanged: false
      },
      ["catalogId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  DayOfWeek,
  DiscountPlatform,
  DiscountType,
  FulfillmentType,
  WebhookIdentityType,
  createDiscount,
  getDiscount,
  listDiscounts,
  updateDiscount
});
