// @wix/auto_sdk_payments_saved-payment-methods/build/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  CardBrand: () => CardBrand,
  RequestedFields: () => RequestedFields,
  WebhookIdentityType: () => WebhookIdentityType,
  deleteSavedPaymentMethod: () => deleteSavedPaymentMethod2,
  getSavedPaymentMethod: () => getSavedPaymentMethod2,
  listSavedPaymentMethods: () => listSavedPaymentMethods2,
  markSavedPaymentMethodPrimary: () => markSavedPaymentMethodPrimary2,
  updateSavedPaymentMethod: () => updateSavedPaymentMethod2,
  upsertSavedPaymentMethod: () => upsertSavedPaymentMethod2
});
module.exports = __toCommonJS(index_typings_exports);

// src/payments-saved-payment-methods-v1-saved-payment-method-saved-payment-methods.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/payments-saved-payment-methods-v1-saved-payment-method-saved-payment-methods.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixPaymentsSavedPaymentMethodsV1SavedPaymentMethodServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/payments/saved-payment-methods",
        destPath: ""
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/payments/saved-payment-methods",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/payments/saved-payment-methods",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_saved-payment-methods";
function upsertSavedPaymentMethod(payload) {
  function __upsertSavedPaymentMethod({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "savedPaymentMethod.createdDate" },
          { path: "savedPaymentMethod.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.payments.saved_payment_methods.v1.saved_payment_method",
      method: "POST",
      methodFqn: "wix.payments.saved_payment_methods.v1.SavedPaymentMethodService.UpsertSavedPaymentMethod",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsSavedPaymentMethodsV1SavedPaymentMethodServiceUrl({
        protoPath: "/v1/saved-payment-methods/upsert",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "savedPaymentMethod.createdDate" },
            { path: "savedPaymentMethod.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __upsertSavedPaymentMethod;
}
function getSavedPaymentMethod(payload) {
  function __getSavedPaymentMethod({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.saved_payment_methods.v1.saved_payment_method",
      method: "GET",
      methodFqn: "wix.payments.saved_payment_methods.v1.SavedPaymentMethodService.GetSavedPaymentMethod",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsSavedPaymentMethodsV1SavedPaymentMethodServiceUrl({
        protoPath: "/v1/saved-payment-methods/{savedPaymentMethodId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "savedPaymentMethod.createdDate" },
            { path: "savedPaymentMethod.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getSavedPaymentMethod;
}
function deleteSavedPaymentMethod(payload) {
  function __deleteSavedPaymentMethod({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.saved_payment_methods.v1.saved_payment_method",
      method: "DELETE",
      methodFqn: "wix.payments.saved_payment_methods.v1.SavedPaymentMethodService.DeleteSavedPaymentMethod",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsSavedPaymentMethodsV1SavedPaymentMethodServiceUrl({
        protoPath: "/v1/saved-payment-methods/{savedPaymentMethodId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteSavedPaymentMethod;
}
function listSavedPaymentMethods(payload) {
  function __listSavedPaymentMethods({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.saved_payment_methods.v1.saved_payment_method",
      method: "GET",
      methodFqn: "wix.payments.saved_payment_methods.v1.SavedPaymentMethodService.ListSavedPaymentMethods",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsSavedPaymentMethodsV1SavedPaymentMethodServiceUrl({
        protoPath: "/v1/saved-payment-methods",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "savedPaymentMethods.createdDate" },
            { path: "savedPaymentMethods.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listSavedPaymentMethods;
}
function updateSavedPaymentMethod(payload) {
  function __updateSavedPaymentMethod({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "savedPaymentMethod.createdDate" },
          { path: "savedPaymentMethod.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.payments.saved_payment_methods.v1.saved_payment_method",
      method: "PATCH",
      methodFqn: "wix.payments.saved_payment_methods.v1.SavedPaymentMethodService.UpdateSavedPaymentMethod",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsSavedPaymentMethodsV1SavedPaymentMethodServiceUrl({
        protoPath: "/v1/saved-payment-methods/{savedPaymentMethod.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "savedPaymentMethod.createdDate" },
            { path: "savedPaymentMethod.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateSavedPaymentMethod;
}
function markSavedPaymentMethodPrimary(payload) {
  function __markSavedPaymentMethodPrimary({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.saved_payment_methods.v1.saved_payment_method",
      method: "GET",
      methodFqn: "wix.payments.saved_payment_methods.v1.SavedPaymentMethodService.MarkSavedPaymentMethodPrimary",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsSavedPaymentMethodsV1SavedPaymentMethodServiceUrl({
        protoPath: "/v1/saved-payment-methods/{savedPaymentMethodId}/mark-primary",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __markSavedPaymentMethodPrimary;
}

// src/payments-saved-payment-methods-v1-saved-payment-method-saved-payment-methods.universal.ts
var CardBrand = /* @__PURE__ */ ((CardBrand2) => {
  CardBrand2["UNKNOWN_CARD_BRAND"] = "UNKNOWN_CARD_BRAND";
  CardBrand2["AMEX"] = "AMEX";
  CardBrand2["DANKORT"] = "DANKORT";
  CardBrand2["DINERS"] = "DINERS";
  CardBrand2["DISCOVER"] = "DISCOVER";
  CardBrand2["ISRACARD"] = "ISRACARD";
  CardBrand2["JCB"] = "JCB";
  CardBrand2["MAESTRO"] = "MAESTRO";
  CardBrand2["MASTERCARD"] = "MASTERCARD";
  CardBrand2["UNIONPAY"] = "UNIONPAY";
  CardBrand2["VISA"] = "VISA";
  CardBrand2["RUPAY"] = "RUPAY";
  return CardBrand2;
})(CardBrand || {});
var RequestedFields = /* @__PURE__ */ ((RequestedFields2) => {
  RequestedFields2["UNKNOWN_REQUESTED_FIELD"] = "UNKNOWN_REQUESTED_FIELD";
  RequestedFields2["SENSITIVE_INFO"] = "SENSITIVE_INFO";
  return RequestedFields2;
})(RequestedFields || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function upsertSavedPaymentMethod2(savedPaymentMethod, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    savedPaymentMethod,
    temporaryCardToken: options?.temporaryCardToken
  });
  const reqOpts = upsertSavedPaymentMethod(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          savedPaymentMethod: "$[0]",
          temporaryCardToken: "$[1].temporaryCardToken"
        },
        singleArgumentUnchanged: false
      },
      ["savedPaymentMethod", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getSavedPaymentMethod2(savedPaymentMethodId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    savedPaymentMethodId,
    fields: options?.fields
  });
  const reqOpts = getSavedPaymentMethod(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.savedPaymentMethod;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          savedPaymentMethodId: "$[0]",
          fields: "$[1].fields"
        },
        singleArgumentUnchanged: false
      },
      ["savedPaymentMethodId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteSavedPaymentMethod2(savedPaymentMethodId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    savedPaymentMethodId
  });
  const reqOpts = deleteSavedPaymentMethod(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { savedPaymentMethodId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["savedPaymentMethodId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listSavedPaymentMethods2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    siteMemberId: options?.siteMemberId,
    paymentMethodTypeId: options?.paymentMethodTypeId,
    merchantUseAllowed: options?.merchantUseAllowed,
    fields: options?.fields,
    paging: options?.paging
  });
  const reqOpts = listSavedPaymentMethods(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          siteMemberId: "$[0].siteMemberId",
          paymentMethodTypeId: "$[0].paymentMethodTypeId",
          merchantUseAllowed: "$[0].merchantUseAllowed",
          fields: "$[0].fields",
          paging: "$[0].paging"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateSavedPaymentMethod2(_id, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    savedPaymentMethod: { ...options?.savedPaymentMethod, id: _id }
  });
  const reqOpts = updateSavedPaymentMethod(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.savedPaymentMethod;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {
          savedPaymentMethod: "$[1].savedPaymentMethod"
        },
        explicitPathsToArguments: { "savedPaymentMethod.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function markSavedPaymentMethodPrimary2(savedPaymentMethodId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    savedPaymentMethodId
  });
  const reqOpts = markSavedPaymentMethodPrimary(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { savedPaymentMethodId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["savedPaymentMethodId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  CardBrand,
  RequestedFields,
  WebhookIdentityType,
  deleteSavedPaymentMethod,
  getSavedPaymentMethod,
  listSavedPaymentMethods,
  markSavedPaymentMethodPrimary,
  updateSavedPaymentMethod,
  upsertSavedPaymentMethod
});
