// @wix/auto_sdk_seo_robots-txt/build/es/meta.mjs
// src/promote-seo-v2-robots-txt-robots-txt.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressPromoteSeoRobotsServerV2RobotsServiceV2Url(opts) {
  const domainToMappings = {
    "www._base_domain_": [
      {
        srcPath: "/_api/promote-seo-robots-server",
        destPath: "/api"
      },
      {
        srcPath: "/_api/promote-seo-robots-server/v2/robots",
        destPath: "/v2/robots"
      },
      {
        srcPath: "/_api/promote-seo-robots-server/v2/ads",
        destPath: "/v2/ads"
      },
      {
        srcPath: "/_api/promote-seo-robots-server/v2/llms",
        destPath: "/v2/llms"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/promote-seo-robots-server",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/promote-seo-robots-server",
        destPath: "/api"
      },
      {
        srcPath: "/_api/promote-seo-robots-server/v2/robots",
        destPath: "/v2/robots"
      },
      {
        srcPath: "/_api/promote-seo-robots-server/v2/ads",
        destPath: "/v2/ads"
      },
      {
        srcPath: "/_api/promote-seo-robots-server/v2/llms",
        destPath: "/v2/llms"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/promote-seo-robots-server/v2/robots",
        destPath: "/v2/robots"
      },
      {
        srcPath: "/promote-seo-robots-server/v2/ads",
        destPath: "/v2/ads"
      },
      {
        srcPath: "/promote-seo-txt-file-server/v2/ads",
        destPath: "/v2/ads"
      },
      {
        srcPath: "/promote-seo-txt-file-server/v2/robots",
        destPath: "/v2/robots"
      },
      {
        srcPath: "/promote-seo-robots-server/v2/llms",
        destPath: "/v2/llms"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/promote/marketing/v2/ads",
        destPath: "/v2/ads"
      },
      {
        srcPath: "/promote/seo/v2/llms",
        destPath: "/v2/llms"
      },
      {
        srcPath: "/promote/seo/v2/robots",
        destPath: "/v2/robots"
      }
    ],
    _: [
      {
        srcPath: "/promote/marketing/v2/ads",
        destPath: "/v2/ads"
      },
      {
        srcPath: "/promote/seo/v2/llms",
        destPath: "/v2/llms"
      },
      {
        srcPath: "/promote/seo/v2/robots",
        destPath: "/v2/robots"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_seo_robots-txt";
function getRobotsTxt(payload) {
  function __getRobotsTxt({ host }) {
    const metadata = {
      entityFqdn: "wix.promote.seo.v2.robots_txt",
      method: "GET",
      methodFqn: "com.wixpress.promote.seo.robots.server.v2.RobotsServiceV2.GetRobotsTxt",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressPromoteSeoRobotsServerV2RobotsServiceV2Url({
        protoPath: "/v2/robots",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getRobotsTxt;
}
function updateRobotsTxt(payload) {
  function __updateRobotsTxt({ host }) {
    const metadata = {
      entityFqdn: "wix.promote.seo.v2.robots_txt",
      method: "PUT",
      methodFqn: "com.wixpress.promote.seo.robots.server.v2.RobotsServiceV2.UpdateRobotsTxt",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressPromoteSeoRobotsServerV2RobotsServiceV2Url({
        protoPath: "/v2/robots",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __updateRobotsTxt;
}

// src/promote-seo-v2-robots-txt-robots-txt.meta.ts
function getRobotsTxt2() {
  const payload = {};
  const getRequestOptions = getRobotsTxt(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/robots",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateRobotsTxt2() {
  const payload = {};
  const getRequestOptions = updateRobotsTxt(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PUT",
    path: "/v2/robots",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  getRobotsTxt2 as getRobotsTxt,
  updateRobotsTxt2 as updateRobotsTxt
};
