// @wix/auto_sdk_events_rsvp-v-2/build/es/meta.mjs
// src/events-v2-rsvp-rsvp-v-2.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKFloatToRESTFloat } from "@wix/sdk-runtime/transformations/float";
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixEventsRsvpsV2RsvpManagementUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/events-rsvps",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/events/v2/rsvps",
        destPath: "/v2/rsvps"
      },
      {
        srcPath: "/events/v2/bulk/rsvps",
        destPath: "/v2/bulk/rsvps"
      }
    ],
    _: [
      {
        srcPath: "/_api/events-rsvps",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/events-rsvps",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_events_rsvp-v-2";
function searchRsvps(payload) {
  function __searchRsvps({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "search.aggregations.range.buckets.from" },
          { path: "search.aggregations.range.buckets.to" },
          {
            path: "search.aggregations.nested.nestedAggregations.range.buckets.from"
          },
          {
            path: "search.aggregations.nested.nestedAggregations.range.buckets.to"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v2.rsvp",
      method: "POST",
      methodFqn: "wix.events.rsvps.v2.RsvpManagement.SearchRsvps",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsRsvpsV2RsvpManagementUrl({
        protoPath: "/v2/rsvps/search",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "rsvps.createdDate" },
            { path: "rsvps.updatedDate" },
            { path: "rsvps.form.inputValues.dateTime" },
            { path: "rsvps.guestList.guests.checkInDetails.checkInDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "rsvps.form.inputValues.number" },
            {
              path: "rsvps.form.inputValues.address.address.geocode.latitude"
            },
            {
              path: "rsvps.form.inputValues.address.address.geocode.longitude"
            },
            { path: "aggregationData.results.ranges.results.from" },
            { path: "aggregationData.results.ranges.results.to" },
            {
              path: "aggregationData.results.groupedByValue.results.nestedResults.ranges.results.from"
            },
            {
              path: "aggregationData.results.groupedByValue.results.nestedResults.ranges.results.to"
            },
            {
              path: "aggregationData.results.nested.results.results.*.range.from"
            },
            {
              path: "aggregationData.results.nested.results.results.*.range.to"
            },
            { path: "aggregationData.results.scalar.value" },
            {
              path: "aggregationData.results.groupedByValue.results.nestedResults.scalar.value"
            },
            {
              path: "aggregationData.results.nested.results.results.*.scalar.value"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __searchRsvps;
}
function queryRsvps(payload) {
  function __queryRsvps({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v2.rsvp",
      method: "POST",
      methodFqn: "wix.events.rsvps.v2.RsvpManagement.QueryRsvps",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsRsvpsV2RsvpManagementUrl({
        protoPath: "/v2/rsvps/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "rsvps.createdDate" },
            { path: "rsvps.updatedDate" },
            { path: "rsvps.form.inputValues.dateTime" },
            { path: "rsvps.guestList.guests.checkInDetails.checkInDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "rsvps.form.inputValues.number" },
            {
              path: "rsvps.form.inputValues.address.address.geocode.latitude"
            },
            {
              path: "rsvps.form.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryRsvps;
}
function countRsvps(payload) {
  function __countRsvps({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v2.rsvp",
      method: "POST",
      methodFqn: "wix.events.rsvps.v2.RsvpManagement.CountRsvps",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsRsvpsV2RsvpManagementUrl({
        protoPath: "/v2/rsvps/count",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __countRsvps;
}
function getRsvp(payload) {
  function __getRsvp({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v2.rsvp",
      method: "GET",
      methodFqn: "wix.events.rsvps.v2.RsvpManagement.GetRsvp",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsRsvpsV2RsvpManagementUrl({
        protoPath: "/v2/rsvps/{rsvpId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "rsvp.createdDate" },
            { path: "rsvp.updatedDate" },
            { path: "rsvp.form.inputValues.dateTime" },
            { path: "rsvp.guestList.guests.checkInDetails.checkInDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "rsvp.form.inputValues.number" },
            {
              path: "rsvp.form.inputValues.address.address.geocode.latitude"
            },
            {
              path: "rsvp.form.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getRsvp;
}
function createRsvp(payload) {
  function __createRsvp({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "rsvp.createdDate" },
          { path: "rsvp.updatedDate" },
          { path: "rsvp.form.inputValues.dateTime" },
          { path: "rsvp.guestList.guests.checkInDetails.checkInDate" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "rsvp.form.inputValues.number" },
          { path: "rsvp.form.inputValues.address.address.geocode.latitude" },
          { path: "rsvp.form.inputValues.address.address.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v2.rsvp",
      method: "POST",
      methodFqn: "wix.events.rsvps.v2.RsvpManagement.CreateRsvp",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsRsvpsV2RsvpManagementUrl({
        protoPath: "/v2/rsvps",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "rsvp.createdDate" },
            { path: "rsvp.updatedDate" },
            { path: "rsvp.form.inputValues.dateTime" },
            { path: "rsvp.guestList.guests.checkInDetails.checkInDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "rsvp.form.inputValues.number" },
            {
              path: "rsvp.form.inputValues.address.address.geocode.latitude"
            },
            {
              path: "rsvp.form.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createRsvp;
}
function updateRsvp(payload) {
  function __updateRsvp({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "rsvp.createdDate" },
          { path: "rsvp.updatedDate" },
          { path: "rsvp.form.inputValues.dateTime" },
          { path: "rsvp.guestList.guests.checkInDetails.checkInDate" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "rsvp.form.inputValues.number" },
          { path: "rsvp.form.inputValues.address.address.geocode.latitude" },
          { path: "rsvp.form.inputValues.address.address.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v2.rsvp",
      method: "PATCH",
      methodFqn: "wix.events.rsvps.v2.RsvpManagement.UpdateRsvp",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsRsvpsV2RsvpManagementUrl({
        protoPath: "/v2/rsvps/{rsvp.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "rsvp.createdDate" },
            { path: "rsvp.updatedDate" },
            { path: "rsvp.form.inputValues.dateTime" },
            { path: "rsvp.guestList.guests.checkInDetails.checkInDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "rsvp.form.inputValues.number" },
            {
              path: "rsvp.form.inputValues.address.address.geocode.latitude"
            },
            {
              path: "rsvp.form.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateRsvp;
}
function deleteRsvp(payload) {
  function __deleteRsvp({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v2.rsvp",
      method: "DELETE",
      methodFqn: "wix.events.rsvps.v2.RsvpManagement.DeleteRsvp",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsRsvpsV2RsvpManagementUrl({
        protoPath: "/v2/rsvps/{rsvpId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteRsvp;
}
function bulkUpdateRsvps(payload) {
  function __bulkUpdateRsvps({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "rsvps.fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "rsvps.rsvp.createdDate" },
          { path: "rsvps.rsvp.updatedDate" },
          { path: "rsvps.rsvp.form.inputValues.dateTime" },
          { path: "rsvps.rsvp.guestList.guests.checkInDetails.checkInDate" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "rsvps.rsvp.form.inputValues.number" },
          {
            path: "rsvps.rsvp.form.inputValues.address.address.geocode.latitude"
          },
          {
            path: "rsvps.rsvp.form.inputValues.address.address.geocode.longitude"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v2.rsvp",
      method: "PATCH",
      methodFqn: "wix.events.rsvps.v2.RsvpManagement.BulkUpdateRsvps",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsRsvpsV2RsvpManagementUrl({
        protoPath: "/v2/bulk/rsvps/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.rsvp.createdDate" },
            { path: "results.rsvp.updatedDate" },
            { path: "results.rsvp.form.inputValues.dateTime" },
            {
              path: "results.rsvp.guestList.guests.checkInDetails.checkInDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.rsvp.form.inputValues.number" },
            {
              path: "results.rsvp.form.inputValues.address.address.geocode.latitude"
            },
            {
              path: "results.rsvp.form.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateRsvps;
}
function bulkDeleteRsvpsByFilter(payload) {
  function __bulkDeleteRsvpsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v2.rsvp",
      method: "POST",
      methodFqn: "wix.events.rsvps.v2.RsvpManagement.BulkDeleteRsvpsByFilter",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsRsvpsV2RsvpManagementUrl({
        protoPath: "/v2/bulk/rsvps/delete-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkDeleteRsvpsByFilter;
}
function checkInRsvpGuests(payload) {
  function __checkInRsvpGuests({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v2.rsvp",
      method: "POST",
      methodFqn: "wix.events.rsvps.v2.RsvpManagement.CheckInRsvpGuests",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsRsvpsV2RsvpManagementUrl({
        protoPath: "/v2/rsvps/{rsvpId}/check-in",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "rsvp.createdDate" },
            { path: "rsvp.updatedDate" },
            { path: "rsvp.form.inputValues.dateTime" },
            { path: "rsvp.guestList.guests.checkInDetails.checkInDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "rsvp.form.inputValues.number" },
            {
              path: "rsvp.form.inputValues.address.address.geocode.latitude"
            },
            {
              path: "rsvp.form.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __checkInRsvpGuests;
}
function cancelRsvpGuestsCheckIn(payload) {
  function __cancelRsvpGuestsCheckIn({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v2.rsvp",
      method: "POST",
      methodFqn: "wix.events.rsvps.v2.RsvpManagement.CancelRsvpGuestsCheckIn",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsRsvpsV2RsvpManagementUrl({
        protoPath: "/v2/rsvps/{rsvpId}/cancel-check-in",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "rsvp.createdDate" },
            { path: "rsvp.updatedDate" },
            { path: "rsvp.form.inputValues.dateTime" },
            { path: "rsvp.guestList.guests.checkInDetails.checkInDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "rsvp.form.inputValues.number" },
            {
              path: "rsvp.form.inputValues.address.address.geocode.latitude"
            },
            {
              path: "rsvp.form.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __cancelRsvpGuestsCheckIn;
}
function listRsvpSummary(payload) {
  function __listRsvpSummary({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v2.rsvp",
      method: "GET",
      methodFqn: "wix.events.rsvps.v2.RsvpManagement.ListRsvpSummary",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsRsvpsV2RsvpManagementUrl({
        protoPath: "/v2/rsvps/summaries",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __listRsvpSummary;
}

// src/events-v2-rsvp-rsvp-v-2.types.ts
var SubdivisionType = /* @__PURE__ */ ((SubdivisionType2) => {
  SubdivisionType2["UNKNOWN_SUBDIVISION_TYPE"] = "UNKNOWN_SUBDIVISION_TYPE";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_1"] = "ADMINISTRATIVE_AREA_LEVEL_1";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_2"] = "ADMINISTRATIVE_AREA_LEVEL_2";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_3"] = "ADMINISTRATIVE_AREA_LEVEL_3";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_4"] = "ADMINISTRATIVE_AREA_LEVEL_4";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_5"] = "ADMINISTRATIVE_AREA_LEVEL_5";
  SubdivisionType2["COUNTRY"] = "COUNTRY";
  return SubdivisionType2;
})(SubdivisionType || {});
var RsvpStatus = /* @__PURE__ */ ((RsvpStatus2) => {
  RsvpStatus2["UNKNOWN_RSVP_STATUS"] = "UNKNOWN_RSVP_STATUS";
  RsvpStatus2["YES"] = "YES";
  RsvpStatus2["NO"] = "NO";
  RsvpStatus2["WAITLIST"] = "WAITLIST";
  return RsvpStatus2;
})(RsvpStatus || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var SortType = /* @__PURE__ */ ((SortType2) => {
  SortType2["COUNT"] = "COUNT";
  SortType2["VALUE"] = "VALUE";
  return SortType2;
})(SortType || {});
var SortDirection = /* @__PURE__ */ ((SortDirection2) => {
  SortDirection2["DESC"] = "DESC";
  SortDirection2["ASC"] = "ASC";
  return SortDirection2;
})(SortDirection || {});
var MissingValues = /* @__PURE__ */ ((MissingValues2) => {
  MissingValues2["EXCLUDE"] = "EXCLUDE";
  MissingValues2["INCLUDE"] = "INCLUDE";
  return MissingValues2;
})(MissingValues || {});
var ScalarType = /* @__PURE__ */ ((ScalarType2) => {
  ScalarType2["UNKNOWN_SCALAR_TYPE"] = "UNKNOWN_SCALAR_TYPE";
  ScalarType2["COUNT_DISTINCT"] = "COUNT_DISTINCT";
  ScalarType2["MIN"] = "MIN";
  ScalarType2["MAX"] = "MAX";
  return ScalarType2;
})(ScalarType || {});
var NestedAggregationType = /* @__PURE__ */ ((NestedAggregationType2) => {
  NestedAggregationType2["UNKNOWN_AGGREGATION_TYPE"] = "UNKNOWN_AGGREGATION_TYPE";
  NestedAggregationType2["VALUE"] = "VALUE";
  NestedAggregationType2["RANGE"] = "RANGE";
  NestedAggregationType2["SCALAR"] = "SCALAR";
  NestedAggregationType2["DATE_HISTOGRAM"] = "DATE_HISTOGRAM";
  return NestedAggregationType2;
})(NestedAggregationType || {});
var Interval = /* @__PURE__ */ ((Interval2) => {
  Interval2["UNKNOWN_INTERVAL"] = "UNKNOWN_INTERVAL";
  Interval2["YEAR"] = "YEAR";
  Interval2["MONTH"] = "MONTH";
  Interval2["WEEK"] = "WEEK";
  Interval2["DAY"] = "DAY";
  Interval2["HOUR"] = "HOUR";
  Interval2["MINUTE"] = "MINUTE";
  Interval2["SECOND"] = "SECOND";
  return Interval2;
})(Interval || {});
var AggregationType = /* @__PURE__ */ ((AggregationType2) => {
  AggregationType2["UNKNOWN_AGGREGATION_TYPE"] = "UNKNOWN_AGGREGATION_TYPE";
  AggregationType2["VALUE"] = "VALUE";
  AggregationType2["RANGE"] = "RANGE";
  AggregationType2["SCALAR"] = "SCALAR";
  AggregationType2["DATE_HISTOGRAM"] = "DATE_HISTOGRAM";
  AggregationType2["NESTED"] = "NESTED";
  return AggregationType2;
})(AggregationType || {});
var Mode = /* @__PURE__ */ ((Mode2) => {
  Mode2["OR"] = "OR";
  Mode2["AND"] = "AND";
  return Mode2;
})(Mode || {});
var RequestedFields = /* @__PURE__ */ ((RequestedFields2) => {
  RequestedFields2["UNKNOWN_REQUESTED_FIELD"] = "UNKNOWN_REQUESTED_FIELD";
  RequestedFields2["DETAILS"] = "DETAILS";
  RequestedFields2["FORM"] = "FORM";
  RequestedFields2["CONTACT_DETAILS"] = "CONTACT_DETAILS";
  return RequestedFields2;
})(RequestedFields || {});
var LocationType = /* @__PURE__ */ ((LocationType2) => {
  LocationType2["VENUE"] = "VENUE";
  LocationType2["ONLINE"] = "ONLINE";
  return LocationType2;
})(LocationType || {});
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["ONE_TIME"] = "ONE_TIME";
  Status2["RECURRING"] = "RECURRING";
  Status2["RECURRING_NEXT"] = "RECURRING_NEXT";
  Status2["RECURRING_LAST_ENDED"] = "RECURRING_LAST_ENDED";
  Status2["RECURRING_LAST_CANCELED"] = "RECURRING_LAST_CANCELED";
  return Status2;
})(Status || {});
var EventStatus = /* @__PURE__ */ ((EventStatus2) => {
  EventStatus2["SCHEDULED"] = "SCHEDULED";
  EventStatus2["STARTED"] = "STARTED";
  EventStatus2["ENDED"] = "ENDED";
  EventStatus2["CANCELED"] = "CANCELED";
  return EventStatus2;
})(EventStatus || {});
var EventType = /* @__PURE__ */ ((EventType2) => {
  EventType2["NA_EVENT_TYPE"] = "NA_EVENT_TYPE";
  EventType2["RSVP"] = "RSVP";
  EventType2["TICKETS"] = "TICKETS";
  EventType2["EXTERNAL"] = "EXTERNAL";
  EventType2["NO_REGISTRATION"] = "NO_REGISTRATION";
  return EventType2;
})(EventType || {});
var RegistrationStatus = /* @__PURE__ */ ((RegistrationStatus2) => {
  RegistrationStatus2["NA_REGISTRATION_STATUS"] = "NA_REGISTRATION_STATUS";
  RegistrationStatus2["CLOSED"] = "CLOSED";
  RegistrationStatus2["CLOSED_MANUALLY"] = "CLOSED_MANUALLY";
  RegistrationStatus2["OPEN_RSVP"] = "OPEN_RSVP";
  RegistrationStatus2["OPEN_RSVP_WAITLIST"] = "OPEN_RSVP_WAITLIST";
  RegistrationStatus2["OPEN_TICKETS"] = "OPEN_TICKETS";
  RegistrationStatus2["OPEN_EXTERNAL"] = "OPEN_EXTERNAL";
  RegistrationStatus2["SCHEDULED_RSVP"] = "SCHEDULED_RSVP";
  return RegistrationStatus2;
})(RegistrationStatus || {});
var RsvpStatusOptions = /* @__PURE__ */ ((RsvpStatusOptions2) => {
  RsvpStatusOptions2["YES_ONLY"] = "YES_ONLY";
  RsvpStatusOptions2["YES_AND_NO"] = "YES_AND_NO";
  return RsvpStatusOptions2;
})(RsvpStatusOptions || {});
var TaxType = /* @__PURE__ */ ((TaxType2) => {
  TaxType2["INCLUDED"] = "INCLUDED";
  TaxType2["ADDED"] = "ADDED";
  TaxType2["ADDED_AT_CHECKOUT"] = "ADDED_AT_CHECKOUT";
  return TaxType2;
})(TaxType || {});
var CheckoutType = /* @__PURE__ */ ((CheckoutType2) => {
  CheckoutType2["UNKNOWN_CHECKOUT_TYPE"] = "UNKNOWN_CHECKOUT_TYPE";
  CheckoutType2["EVENTS_APP"] = "EVENTS_APP";
  CheckoutType2["ECOMM_PLATFORM"] = "ECOMM_PLATFORM";
  return CheckoutType2;
})(CheckoutType || {});
var VisitorType = /* @__PURE__ */ ((VisitorType2) => {
  VisitorType2["VISITOR"] = "VISITOR";
  VisitorType2["MEMBER"] = "MEMBER";
  VisitorType2["VISITOR_OR_MEMBER"] = "VISITOR_OR_MEMBER";
  return VisitorType2;
})(VisitorType || {});
var InputControlType = /* @__PURE__ */ ((InputControlType2) => {
  InputControlType2["INPUT"] = "INPUT";
  InputControlType2["TEXTAREA"] = "TEXTAREA";
  InputControlType2["DROPDOWN"] = "DROPDOWN";
  InputControlType2["RADIO"] = "RADIO";
  InputControlType2["CHECKBOX"] = "CHECKBOX";
  InputControlType2["NAME"] = "NAME";
  InputControlType2["GUEST_CONTROL"] = "GUEST_CONTROL";
  InputControlType2["ADDRESS_SHORT"] = "ADDRESS_SHORT";
  InputControlType2["ADDRESS_FULL"] = "ADDRESS_FULL";
  InputControlType2["DATE"] = "DATE";
  return InputControlType2;
})(InputControlType || {});
var ValueType = /* @__PURE__ */ ((ValueType2) => {
  ValueType2["TEXT"] = "TEXT";
  ValueType2["NUMBER"] = "NUMBER";
  ValueType2["TEXT_ARRAY"] = "TEXT_ARRAY";
  ValueType2["DATE_TIME"] = "DATE_TIME";
  ValueType2["ADDRESS"] = "ADDRESS";
  return ValueType2;
})(ValueType || {});
var ConferenceType = /* @__PURE__ */ ((ConferenceType2) => {
  ConferenceType2["MEETING"] = "MEETING";
  ConferenceType2["WEBINAR"] = "WEBINAR";
  return ConferenceType2;
})(ConferenceType || {});
var State = /* @__PURE__ */ ((State2) => {
  State2["MANUAL"] = "MANUAL";
  State2["AUTO"] = "AUTO";
  State2["RECURRING_EVENT"] = "RECURRING_EVENT";
  State2["HIDDEN"] = "HIDDEN";
  return State2;
})(State || {});
var NodeType = /* @__PURE__ */ ((NodeType2) => {
  NodeType2["PARAGRAPH"] = "PARAGRAPH";
  NodeType2["TEXT"] = "TEXT";
  NodeType2["HEADING"] = "HEADING";
  NodeType2["BULLETED_LIST"] = "BULLETED_LIST";
  NodeType2["ORDERED_LIST"] = "ORDERED_LIST";
  NodeType2["LIST_ITEM"] = "LIST_ITEM";
  NodeType2["BLOCKQUOTE"] = "BLOCKQUOTE";
  NodeType2["CODE_BLOCK"] = "CODE_BLOCK";
  NodeType2["VIDEO"] = "VIDEO";
  NodeType2["DIVIDER"] = "DIVIDER";
  NodeType2["FILE"] = "FILE";
  NodeType2["GALLERY"] = "GALLERY";
  NodeType2["GIF"] = "GIF";
  NodeType2["HTML"] = "HTML";
  NodeType2["IMAGE"] = "IMAGE";
  NodeType2["LINK_PREVIEW"] = "LINK_PREVIEW";
  NodeType2["MAP"] = "MAP";
  NodeType2["POLL"] = "POLL";
  NodeType2["APP_EMBED"] = "APP_EMBED";
  NodeType2["BUTTON"] = "BUTTON";
  NodeType2["COLLAPSIBLE_LIST"] = "COLLAPSIBLE_LIST";
  NodeType2["TABLE"] = "TABLE";
  NodeType2["EMBED"] = "EMBED";
  NodeType2["COLLAPSIBLE_ITEM"] = "COLLAPSIBLE_ITEM";
  NodeType2["COLLAPSIBLE_ITEM_TITLE"] = "COLLAPSIBLE_ITEM_TITLE";
  NodeType2["COLLAPSIBLE_ITEM_BODY"] = "COLLAPSIBLE_ITEM_BODY";
  NodeType2["TABLE_CELL"] = "TABLE_CELL";
  NodeType2["TABLE_ROW"] = "TABLE_ROW";
  NodeType2["EXTERNAL"] = "EXTERNAL";
  NodeType2["AUDIO"] = "AUDIO";
  NodeType2["CAPTION"] = "CAPTION";
  NodeType2["LAYOUT"] = "LAYOUT";
  NodeType2["LAYOUT_CELL"] = "LAYOUT_CELL";
  return NodeType2;
})(NodeType || {});
var WidthType = /* @__PURE__ */ ((WidthType2) => {
  WidthType2["CONTENT"] = "CONTENT";
  WidthType2["SMALL"] = "SMALL";
  WidthType2["ORIGINAL"] = "ORIGINAL";
  WidthType2["FULL_WIDTH"] = "FULL_WIDTH";
  return WidthType2;
})(WidthType || {});
var PluginContainerDataAlignment = /* @__PURE__ */ ((PluginContainerDataAlignment2) => {
  PluginContainerDataAlignment2["CENTER"] = "CENTER";
  PluginContainerDataAlignment2["LEFT"] = "LEFT";
  PluginContainerDataAlignment2["RIGHT"] = "RIGHT";
  return PluginContainerDataAlignment2;
})(PluginContainerDataAlignment || {});
var ButtonDataType = /* @__PURE__ */ ((ButtonDataType2) => {
  ButtonDataType2["LINK"] = "LINK";
  ButtonDataType2["ACTION"] = "ACTION";
  return ButtonDataType2;
})(ButtonDataType || {});
var Target = /* @__PURE__ */ ((Target2) => {
  Target2["SELF"] = "SELF";
  Target2["BLANK"] = "BLANK";
  Target2["PARENT"] = "PARENT";
  Target2["TOP"] = "TOP";
  return Target2;
})(Target || {});
var TextAlignment = /* @__PURE__ */ ((TextAlignment2) => {
  TextAlignment2["AUTO"] = "AUTO";
  TextAlignment2["LEFT"] = "LEFT";
  TextAlignment2["RIGHT"] = "RIGHT";
  TextAlignment2["CENTER"] = "CENTER";
  TextAlignment2["JUSTIFY"] = "JUSTIFY";
  return TextAlignment2;
})(TextAlignment || {});
var LineStyle = /* @__PURE__ */ ((LineStyle2) => {
  LineStyle2["SINGLE"] = "SINGLE";
  LineStyle2["DOUBLE"] = "DOUBLE";
  LineStyle2["DASHED"] = "DASHED";
  LineStyle2["DOTTED"] = "DOTTED";
  return LineStyle2;
})(LineStyle || {});
var Width = /* @__PURE__ */ ((Width2) => {
  Width2["LARGE"] = "LARGE";
  Width2["MEDIUM"] = "MEDIUM";
  Width2["SMALL"] = "SMALL";
  return Width2;
})(Width || {});
var DividerDataAlignment = /* @__PURE__ */ ((DividerDataAlignment2) => {
  DividerDataAlignment2["CENTER"] = "CENTER";
  DividerDataAlignment2["LEFT"] = "LEFT";
  DividerDataAlignment2["RIGHT"] = "RIGHT";
  return DividerDataAlignment2;
})(DividerDataAlignment || {});
var ViewMode = /* @__PURE__ */ ((ViewMode2) => {
  ViewMode2["NONE"] = "NONE";
  ViewMode2["FULL"] = "FULL";
  ViewMode2["MINI"] = "MINI";
  return ViewMode2;
})(ViewMode || {});
var LayoutType = /* @__PURE__ */ ((LayoutType2) => {
  LayoutType2["COLLAGE"] = "COLLAGE";
  LayoutType2["MASONRY"] = "MASONRY";
  LayoutType2["GRID"] = "GRID";
  LayoutType2["THUMBNAIL"] = "THUMBNAIL";
  LayoutType2["SLIDER"] = "SLIDER";
  LayoutType2["SLIDESHOW"] = "SLIDESHOW";
  LayoutType2["PANORAMA"] = "PANORAMA";
  LayoutType2["COLUMN"] = "COLUMN";
  LayoutType2["MAGIC"] = "MAGIC";
  LayoutType2["FULLSIZE"] = "FULLSIZE";
  return LayoutType2;
})(LayoutType || {});
var Orientation = /* @__PURE__ */ ((Orientation2) => {
  Orientation2["ROWS"] = "ROWS";
  Orientation2["COLUMNS"] = "COLUMNS";
  return Orientation2;
})(Orientation || {});
var Crop = /* @__PURE__ */ ((Crop2) => {
  Crop2["FILL"] = "FILL";
  Crop2["FIT"] = "FIT";
  return Crop2;
})(Crop || {});
var ThumbnailsAlignment = /* @__PURE__ */ ((ThumbnailsAlignment2) => {
  ThumbnailsAlignment2["TOP"] = "TOP";
  ThumbnailsAlignment2["RIGHT"] = "RIGHT";
  ThumbnailsAlignment2["BOTTOM"] = "BOTTOM";
  ThumbnailsAlignment2["LEFT"] = "LEFT";
  ThumbnailsAlignment2["NONE"] = "NONE";
  return ThumbnailsAlignment2;
})(ThumbnailsAlignment || {});
var GIFType = /* @__PURE__ */ ((GIFType2) => {
  GIFType2["NORMAL"] = "NORMAL";
  GIFType2["STICKER"] = "STICKER";
  return GIFType2;
})(GIFType || {});
var Source = /* @__PURE__ */ ((Source2) => {
  Source2["HTML"] = "HTML";
  Source2["ADSENSE"] = "ADSENSE";
  Source2["AI"] = "AI";
  return Source2;
})(Source || {});
var StylesPosition = /* @__PURE__ */ ((StylesPosition2) => {
  StylesPosition2["START"] = "START";
  StylesPosition2["END"] = "END";
  StylesPosition2["TOP"] = "TOP";
  StylesPosition2["HIDDEN"] = "HIDDEN";
  return StylesPosition2;
})(StylesPosition || {});
var MapType = /* @__PURE__ */ ((MapType2) => {
  MapType2["ROADMAP"] = "ROADMAP";
  MapType2["SATELITE"] = "SATELITE";
  MapType2["HYBRID"] = "HYBRID";
  MapType2["TERRAIN"] = "TERRAIN";
  return MapType2;
})(MapType || {});
var ViewRole = /* @__PURE__ */ ((ViewRole2) => {
  ViewRole2["CREATOR"] = "CREATOR";
  ViewRole2["VOTERS"] = "VOTERS";
  ViewRole2["EVERYONE"] = "EVERYONE";
  return ViewRole2;
})(ViewRole || {});
var VoteRole = /* @__PURE__ */ ((VoteRole2) => {
  VoteRole2["SITE_MEMBERS"] = "SITE_MEMBERS";
  VoteRole2["ALL"] = "ALL";
  return VoteRole2;
})(VoteRole || {});
var PollLayoutType = /* @__PURE__ */ ((PollLayoutType2) => {
  PollLayoutType2["LIST"] = "LIST";
  PollLayoutType2["GRID"] = "GRID";
  return PollLayoutType2;
})(PollLayoutType || {});
var PollLayoutDirection = /* @__PURE__ */ ((PollLayoutDirection2) => {
  PollLayoutDirection2["LTR"] = "LTR";
  PollLayoutDirection2["RTL"] = "RTL";
  return PollLayoutDirection2;
})(PollLayoutDirection || {});
var BackgroundType = /* @__PURE__ */ ((BackgroundType2) => {
  BackgroundType2["COLOR"] = "COLOR";
  BackgroundType2["IMAGE"] = "IMAGE";
  BackgroundType2["GRADIENT"] = "GRADIENT";
  return BackgroundType2;
})(BackgroundType || {});
var DecorationType = /* @__PURE__ */ ((DecorationType2) => {
  DecorationType2["BOLD"] = "BOLD";
  DecorationType2["ITALIC"] = "ITALIC";
  DecorationType2["UNDERLINE"] = "UNDERLINE";
  DecorationType2["SPOILER"] = "SPOILER";
  DecorationType2["ANCHOR"] = "ANCHOR";
  DecorationType2["MENTION"] = "MENTION";
  DecorationType2["LINK"] = "LINK";
  DecorationType2["COLOR"] = "COLOR";
  DecorationType2["FONT_SIZE"] = "FONT_SIZE";
  DecorationType2["EXTERNAL"] = "EXTERNAL";
  DecorationType2["STRIKETHROUGH"] = "STRIKETHROUGH";
  DecorationType2["SUPERSCRIPT"] = "SUPERSCRIPT";
  DecorationType2["SUBSCRIPT"] = "SUBSCRIPT";
  return DecorationType2;
})(DecorationType || {});
var FontType = /* @__PURE__ */ ((FontType2) => {
  FontType2["PX"] = "PX";
  FontType2["EM"] = "EM";
  return FontType2;
})(FontType || {});
var Position = /* @__PURE__ */ ((Position2) => {
  Position2["START"] = "START";
  Position2["END"] = "END";
  Position2["TOP"] = "TOP";
  return Position2;
})(Position || {});
var AspectRatio = /* @__PURE__ */ ((AspectRatio2) => {
  AspectRatio2["SQUARE"] = "SQUARE";
  AspectRatio2["RECTANGLE"] = "RECTANGLE";
  return AspectRatio2;
})(AspectRatio || {});
var Resizing = /* @__PURE__ */ ((Resizing2) => {
  Resizing2["FILL"] = "FILL";
  Resizing2["FIT"] = "FIT";
  return Resizing2;
})(Resizing || {});
var Placement = /* @__PURE__ */ ((Placement2) => {
  Placement2["IMAGE"] = "IMAGE";
  Placement2["PRODUCT_INFO"] = "PRODUCT_INFO";
  return Placement2;
})(Placement || {});
var CardStylesType = /* @__PURE__ */ ((CardStylesType2) => {
  CardStylesType2["CONTAINED"] = "CONTAINED";
  CardStylesType2["FRAMELESS"] = "FRAMELESS";
  return CardStylesType2;
})(CardStylesType || {});
var Alignment = /* @__PURE__ */ ((Alignment2) => {
  Alignment2["START"] = "START";
  Alignment2["CENTER"] = "CENTER";
  Alignment2["END"] = "END";
  return Alignment2;
})(Alignment || {});
var Layout = /* @__PURE__ */ ((Layout2) => {
  Layout2["STACKED"] = "STACKED";
  Layout2["SIDE_BY_SIDE"] = "SIDE_BY_SIDE";
  return Layout2;
})(Layout || {});
var AppType = /* @__PURE__ */ ((AppType2) => {
  AppType2["PRODUCT"] = "PRODUCT";
  AppType2["EVENT"] = "EVENT";
  AppType2["BOOKING"] = "BOOKING";
  return AppType2;
})(AppType || {});
var InitialExpandedItems = /* @__PURE__ */ ((InitialExpandedItems2) => {
  InitialExpandedItems2["FIRST"] = "FIRST";
  InitialExpandedItems2["ALL"] = "ALL";
  InitialExpandedItems2["NONE"] = "NONE";
  return InitialExpandedItems2;
})(InitialExpandedItems || {});
var Direction = /* @__PURE__ */ ((Direction2) => {
  Direction2["LTR"] = "LTR";
  Direction2["RTL"] = "RTL";
  return Direction2;
})(Direction || {});
var VerticalAlignment = /* @__PURE__ */ ((VerticalAlignment2) => {
  VerticalAlignment2["TOP"] = "TOP";
  VerticalAlignment2["MIDDLE"] = "MIDDLE";
  VerticalAlignment2["BOTTOM"] = "BOTTOM";
  return VerticalAlignment2;
})(VerticalAlignment || {});
var NullValue = /* @__PURE__ */ ((NullValue2) => {
  NullValue2["NULL_VALUE"] = "NULL_VALUE";
  return NullValue2;
})(NullValue || {});
var Scaling = /* @__PURE__ */ ((Scaling2) => {
  Scaling2["AUTO"] = "AUTO";
  Scaling2["CONTAIN"] = "CONTAIN";
  Scaling2["COVER"] = "COVER";
  return Scaling2;
})(Scaling || {});
var ImagePosition = /* @__PURE__ */ ((ImagePosition2) => {
  ImagePosition2["CENTER"] = "CENTER";
  ImagePosition2["CENTER_LEFT"] = "CENTER_LEFT";
  ImagePosition2["CENTER_RIGHT"] = "CENTER_RIGHT";
  ImagePosition2["TOP"] = "TOP";
  ImagePosition2["TOP_LEFT"] = "TOP_LEFT";
  ImagePosition2["TOP_RIGHT"] = "TOP_RIGHT";
  ImagePosition2["BOTTOM"] = "BOTTOM";
  ImagePosition2["BOTTOM_LEFT"] = "BOTTOM_LEFT";
  ImagePosition2["BOTTOM_RIGHT"] = "BOTTOM_RIGHT";
  return ImagePosition2;
})(ImagePosition || {});
var VerticalAlignmentAlignment = /* @__PURE__ */ ((VerticalAlignmentAlignment2) => {
  VerticalAlignmentAlignment2["TOP"] = "TOP";
  VerticalAlignmentAlignment2["MIDDLE"] = "MIDDLE";
  VerticalAlignmentAlignment2["BOTTOM"] = "BOTTOM";
  return VerticalAlignmentAlignment2;
})(VerticalAlignmentAlignment || {});
var ResponsivenessBehaviour = /* @__PURE__ */ ((ResponsivenessBehaviour2) => {
  ResponsivenessBehaviour2["STACK"] = "STACK";
  ResponsivenessBehaviour2["WRAP"] = "WRAP";
  return ResponsivenessBehaviour2;
})(ResponsivenessBehaviour || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNKNOWN_BADGE_TYPE"] = "UNKNOWN_BADGE_TYPE";
  Type2["FIRST_PRIORITY"] = "FIRST_PRIORITY";
  Type2["SECOND_PRIORITY"] = "SECOND_PRIORITY";
  Type2["THIRD_PRIORITY"] = "THIRD_PRIORITY";
  return Type2;
})(Type || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/events-v2-rsvp-rsvp-v-2.meta.ts
function searchRsvps2() {
  const payload = {};
  const getRequestOptions = searchRsvps(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/rsvps/search",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryRsvps2() {
  const payload = {};
  const getRequestOptions = queryRsvps(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/rsvps/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function countRsvps2() {
  const payload = {};
  const getRequestOptions = countRsvps(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/rsvps/count",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getRsvp2() {
  const payload = { rsvpId: ":rsvpId" };
  const getRequestOptions = getRsvp(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/rsvps/{rsvpId}",
    pathParams: { rsvpId: "rsvpId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createRsvp2() {
  const payload = {};
  const getRequestOptions = createRsvp(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/rsvps",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateRsvp2() {
  const payload = { rsvp: { id: ":rsvpId" } };
  const getRequestOptions = updateRsvp(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v2/rsvps/{rsvp.id}",
    pathParams: { rsvpId: "rsvpId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteRsvp2() {
  const payload = { rsvpId: ":rsvpId" };
  const getRequestOptions = deleteRsvp(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v2/rsvps/{rsvpId}",
    pathParams: { rsvpId: "rsvpId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateRsvps2() {
  const payload = {};
  const getRequestOptions = bulkUpdateRsvps(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v2/bulk/rsvps/update",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkDeleteRsvpsByFilter2() {
  const payload = {};
  const getRequestOptions = bulkDeleteRsvpsByFilter(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/bulk/rsvps/delete-by-filter",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function checkInRsvpGuests2() {
  const payload = { rsvpId: ":rsvpId" };
  const getRequestOptions = checkInRsvpGuests(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/rsvps/{rsvpId}/check-in",
    pathParams: { rsvpId: "rsvpId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function cancelRsvpGuestsCheckIn2() {
  const payload = { rsvpId: ":rsvpId" };
  const getRequestOptions = cancelRsvpGuestsCheckIn(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/rsvps/{rsvpId}/cancel-check-in",
    pathParams: { rsvpId: "rsvpId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listRsvpSummary2() {
  const payload = {};
  const getRequestOptions = listRsvpSummary(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/rsvps/summaries",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  AggregationType as AggregationTypeOriginal,
  Alignment as AlignmentOriginal,
  AppType as AppTypeOriginal,
  AspectRatio as AspectRatioOriginal,
  BackgroundType as BackgroundTypeOriginal,
  ButtonDataType as ButtonDataTypeOriginal,
  CardStylesType as CardStylesTypeOriginal,
  CheckoutType as CheckoutTypeOriginal,
  ConferenceType as ConferenceTypeOriginal,
  Crop as CropOriginal,
  DecorationType as DecorationTypeOriginal,
  Direction as DirectionOriginal,
  DividerDataAlignment as DividerDataAlignmentOriginal,
  EventStatus as EventStatusOriginal,
  EventType as EventTypeOriginal,
  FontType as FontTypeOriginal,
  GIFType as GIFTypeOriginal,
  ImagePosition as ImagePositionOriginal,
  InitialExpandedItems as InitialExpandedItemsOriginal,
  InputControlType as InputControlTypeOriginal,
  Interval as IntervalOriginal,
  Layout as LayoutOriginal,
  LayoutType as LayoutTypeOriginal,
  LineStyle as LineStyleOriginal,
  LocationType as LocationTypeOriginal,
  MapType as MapTypeOriginal,
  MissingValues as MissingValuesOriginal,
  Mode as ModeOriginal,
  NestedAggregationType as NestedAggregationTypeOriginal,
  NodeType as NodeTypeOriginal,
  NullValue as NullValueOriginal,
  Orientation as OrientationOriginal,
  Placement as PlacementOriginal,
  PluginContainerDataAlignment as PluginContainerDataAlignmentOriginal,
  PollLayoutDirection as PollLayoutDirectionOriginal,
  PollLayoutType as PollLayoutTypeOriginal,
  Position as PositionOriginal,
  RegistrationStatus as RegistrationStatusOriginal,
  RequestedFields as RequestedFieldsOriginal,
  Resizing as ResizingOriginal,
  ResponsivenessBehaviour as ResponsivenessBehaviourOriginal,
  RsvpStatusOptions as RsvpStatusOptionsOriginal,
  RsvpStatus as RsvpStatusOriginal,
  ScalarType as ScalarTypeOriginal,
  Scaling as ScalingOriginal,
  SortDirection as SortDirectionOriginal,
  SortOrder as SortOrderOriginal,
  SortType as SortTypeOriginal,
  Source as SourceOriginal,
  State as StateOriginal,
  Status as StatusOriginal,
  StylesPosition as StylesPositionOriginal,
  SubdivisionType as SubdivisionTypeOriginal,
  Target as TargetOriginal,
  TaxType as TaxTypeOriginal,
  TextAlignment as TextAlignmentOriginal,
  ThumbnailsAlignment as ThumbnailsAlignmentOriginal,
  Type as TypeOriginal,
  ValueType as ValueTypeOriginal,
  VerticalAlignmentAlignment as VerticalAlignmentAlignmentOriginal,
  VerticalAlignment as VerticalAlignmentOriginal,
  ViewMode as ViewModeOriginal,
  ViewRole as ViewRoleOriginal,
  VisitorType as VisitorTypeOriginal,
  VoteRole as VoteRoleOriginal,
  WebhookIdentityType as WebhookIdentityTypeOriginal,
  Width as WidthOriginal,
  WidthType as WidthTypeOriginal,
  bulkDeleteRsvpsByFilter2 as bulkDeleteRsvpsByFilter,
  bulkUpdateRsvps2 as bulkUpdateRsvps,
  cancelRsvpGuestsCheckIn2 as cancelRsvpGuestsCheckIn,
  checkInRsvpGuests2 as checkInRsvpGuests,
  countRsvps2 as countRsvps,
  createRsvp2 as createRsvp,
  deleteRsvp2 as deleteRsvp,
  getRsvp2 as getRsvp,
  listRsvpSummary2 as listRsvpSummary,
  queryRsvps2 as queryRsvps,
  searchRsvps2 as searchRsvps,
  updateRsvp2 as updateRsvp
};
