// @wix/headless-online-programs/cjs/dist/services/instructors-service.js
import { defineService, implementService } from '@wix/services-definitions';
import { SignalsServiceDefinition, } from '@wix/services-definitions/core-services/signals';
import { instructors } from '@wix/online-programs';
/**
 * Service definition for the Instructors service.
 * This defines the contract that the InstructorsService must implement.
 *
 * @constant
 */
export const InstructorsServiceDefinition = defineService('program-instructors');
/**
 * Implementation of the Instructors service that manages reactive instructors data.
 * This service provides signals for instructors data, loading state, and error handling,
 * along with methods to dynamically load instructors by program IDs.
 *
 * @example
 * ```tsx
 * import { InstructorsService, InstructorsServiceDefinition } from '@wix/online-programs/services';
 * import { useService } from '@wix/services-manager-react';
 *
 * function InstructorsComponent({ instructorsConfig }) {
 *   return (
 *     <ServiceProvider services={createServicesMap([
 *       [InstructorsServiceDefinition, InstructorsService.withConfig(instructorsConfig)]
 *     ])}>
 *       <InstructorsDisplay />
 *     </ServiceProvider>
 *   );
 * }
 *
 * function InstructorsDisplay() {
 *   const instructorsService = useService(InstructorsServiceDefinition);
 *   const instructors = instructorsService.instructors.get();
 *   const isLoading = instructorsService.isLoading.get();
 *   const error = instructorsService.error.get();
 *
 *   if (isLoading) return <div>Loading...</div>;
 *   if (error) return <div>Error: {error}</div>;
 *
 *   return (
 *     <div>
 *       {instructors.map(instructor => (
 *         <div key={instructor._id}>{instructor.name}</div>
 *       ))}
 *     </div>
 *   );
 * }
 * ```
 */
export const InstructorsService = implementService.withConfig()(InstructorsServiceDefinition, ({ getService, config }) => {
    const signalsService = getService(SignalsServiceDefinition);
    const instructors = signalsService.signal(config.instructors);
    const isLoading = signalsService.signal(!!config.programIds);
    const error = signalsService.signal(null);
    const loadInstructorsByProgramIds = async (programIds) => {
        isLoading.set(true);
        const instructorsResponse = await listInstructorsByProgramIds(programIds);
        if (!instructorsResponse) {
            instructors.set([]);
            error.set('Failed to load instructors');
        }
        else {
            instructors.set(instructorsResponse);
            error.set(null);
        }
        isLoading.set(false);
    };
    // Load instructors on initialization if programIds are provided
    if (config.programIds && config.programIds.length > 0) {
        loadInstructorsByProgramIds(config.programIds);
    }
    return {
        instructors,
        isLoading,
        error,
        loadInstructorsByProgramIds,
    };
});
/**
 * Internal helper function to load instructors by program IDs from the Wix Instructors API.
 * Fetches instructors data for specific programs.
 *
 * @private
 * @param {string[]} programIds - The program IDs to load instructors for
 * @returns {Promise<Instructor[]>} Instructors response from the API
 */
const listInstructorsByProgramIds = async (programIds) => {
    const instructorsResponse = await instructors.listInstructors({
        programIdsFilter: programIds,
    });
    return instructorsResponse.instructors;
};
/**
 * Loads instructors service configuration from the Wix Instructors API for SSR initialization.
 * This function is designed to be used during Server-Side Rendering (SSR) to preload
 * instructors by program IDs that will be used to configure the InstructorsService.
 *
 * @param programIds The program IDs to load instructors for
 * @returns Promise that resolves to InstructorsServiceConfigResult (success with config or notFound)
 *
 * @example
 * ```astro
 * ---
 * // Astro page example - pages/program/[id].astro
 * import { loadInstructorsServiceConfig } from '@wix/online-programs/services';
 * import { Program } from '@wix/online-programs/components';
 *
 * // Get program ID from URL params
 * const { id } = Astro.params;
 *
 * // Load instructors data during SSR
 * const instructorsServiceConfigResult = await loadInstructorsServiceConfig([id]);
 *
 * // Handle not found case
 * if (instructorsServiceConfigResult.type === 'notFound') {
 *   return Astro.redirect('/404');
 * }
 * ---
 *
 * <Program.Root program={programServiceConfigResult.config.program!}>
 *   <Program.Instructors instructors={instructorsServiceConfigResult.config.instructors!}>
 *     <Program.Instructor.Name />
 *   </Program.Instructors>
 * </Program.Root>
 * ```
 *
 * @example
 * ```tsx
 * // Next.js page example - pages/program/[id].tsx
 * import { GetServerSideProps } from 'next';
 * import { loadInstructorsServiceConfig } from '@wix/online-programs/services';
 * import { Program } from '@wix/online-programs/components';
 *
 * interface ProgramPageProps {
 *   instructorsServiceConfig: InstructorsServiceConfig;
 * }
 *
 * export const getServerSideProps: GetServerSideProps<ProgramPageProps> = async ({ params }) => {
 *   const id = params?.id as string;
 *
 *   // Load instructors data during SSR
 *   const instructorsServiceConfigResult = await loadInstructorsServiceConfig([id]);
 *
 *   // Handle not found case
 *   if (instructorsServiceConfigResult.type === 'notFound') {
 *     return {
 *       notFound: true,
 *     };
 *   }
 *
 *   return {
 *     props: {
 *       instructorsServiceConfig: instructorsServiceConfigResult.config,
 *     },
 *   };
 * };
 *
 * export default function ProgramPage({ instructorsServiceConfig }: ProgramPageProps) {
 *   return (
 *     <Program.Root program={programServiceConfig.program!}>
 *       <Program.Instructors instructors={instructorsServiceConfig.instructors!}>
 *         <Program.Instructor.Name />
 *       </Program.Instructors>
 *     </Program.Root>
 *   );
 * }
 * ```
 */
export async function loadInstructorsServiceConfig(programIds) {
    try {
        const instructorsResponse = await listInstructorsByProgramIds(programIds);
        if (!instructorsResponse || instructorsResponse.length === 0) {
            return { type: 'notFound' };
        }
        return {
            type: 'success',
            config: {
                instructors: instructorsResponse,
                programIds,
            },
        };
    }
    catch (error) {
        console.error(`Failed to load instructors for program IDs "${programIds.join(', ')}":`, error);
        return { type: 'notFound' };
    }
}
