// @wix/auto_sdk_seo_robots-txt/build/internal/es/index.mjs
// src/promote-seo-v2-robots-txt-robots-txt.public.ts
import { renameKeysFromRESTResponseToSDKResponse as renameKeysFromRESTResponseToSDKResponse2 } from "@wix/sdk-runtime/rename-all-nested-keys";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { EventDefinition } from "@wix/sdk-types";

// src/promote-seo-v2-robots-txt-robots-txt.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/promote-seo-v2-robots-txt-robots-txt.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressPromoteSeoRobotsServerV2RobotsServiceV2Url(opts) {
  const domainToMappings = {
    "www._base_domain_": [
      {
        srcPath: "/_api/promote-seo-robots-server",
        destPath: "/api"
      },
      {
        srcPath: "/_api/promote-seo-robots-server/v2/robots",
        destPath: "/v2/robots"
      },
      {
        srcPath: "/_api/promote-seo-robots-server/v2/ads",
        destPath: "/v2/ads"
      },
      {
        srcPath: "/_api/promote-seo-robots-server/v2/llms",
        destPath: "/v2/llms"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/promote-seo-robots-server",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/promote-seo-robots-server",
        destPath: "/api"
      },
      {
        srcPath: "/_api/promote-seo-robots-server/v2/robots",
        destPath: "/v2/robots"
      },
      {
        srcPath: "/_api/promote-seo-robots-server/v2/ads",
        destPath: "/v2/ads"
      },
      {
        srcPath: "/_api/promote-seo-robots-server/v2/llms",
        destPath: "/v2/llms"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/promote-seo-robots-server/v2/robots",
        destPath: "/v2/robots"
      },
      {
        srcPath: "/promote-seo-robots-server/v2/ads",
        destPath: "/v2/ads"
      },
      {
        srcPath: "/promote-seo-txt-file-server/v2/ads",
        destPath: "/v2/ads"
      },
      {
        srcPath: "/promote-seo-txt-file-server/v2/robots",
        destPath: "/v2/robots"
      },
      {
        srcPath: "/promote-seo-robots-server/v2/llms",
        destPath: "/v2/llms"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/promote/marketing/v2/ads",
        destPath: "/v2/ads"
      },
      {
        srcPath: "/promote/seo/v2/llms",
        destPath: "/v2/llms"
      },
      {
        srcPath: "/promote/seo/v2/robots",
        destPath: "/v2/robots"
      }
    ],
    _: [
      {
        srcPath: "/promote/marketing/v2/ads",
        destPath: "/v2/ads"
      },
      {
        srcPath: "/promote/seo/v2/llms",
        destPath: "/v2/llms"
      },
      {
        srcPath: "/promote/seo/v2/robots",
        destPath: "/v2/robots"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_seo_robots-txt";
function getRobotsTxt(payload) {
  function __getRobotsTxt({ host }) {
    const metadata = {
      entityFqdn: "wix.promote.seo.v2.robots_txt",
      method: "GET",
      methodFqn: "com.wixpress.promote.seo.robots.server.v2.RobotsServiceV2.GetRobotsTxt",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressPromoteSeoRobotsServerV2RobotsServiceV2Url({
        protoPath: "/v2/robots",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getRobotsTxt;
}
function updateRobotsTxt(payload) {
  function __updateRobotsTxt({ host }) {
    const metadata = {
      entityFqdn: "wix.promote.seo.v2.robots_txt",
      method: "PUT",
      methodFqn: "com.wixpress.promote.seo.robots.server.v2.RobotsServiceV2.UpdateRobotsTxt",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressPromoteSeoRobotsServerV2RobotsServiceV2Url({
        protoPath: "/v2/robots",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __updateRobotsTxt;
}

// src/promote-seo-v2-robots-txt-robots-txt.universal.ts
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function getRobotsTxt2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    subdomain: options?.subdomain
  });
  const reqOpts = getRobotsTxt(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { subdomain: "$[0].subdomain" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateRobotsTxt2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    robotsTxt: options?.robotsTxt
  });
  const reqOpts = updateRobotsTxt(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { robotsTxt: "$[0].robotsTxt" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/promote-seo-v2-robots-txt-robots-txt.public.ts
function getRobotsTxt3(httpClient) {
  return (options) => getRobotsTxt2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function updateRobotsTxt3(httpClient) {
  return (options) => updateRobotsTxt2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
var onRobotsTxtUpdated = EventDefinition(
  "wix.promote.seo.robots_txt_updated",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [{ path: "metadata.eventTime" }]
      }
    ])
  )
)();

// src/promote-seo-v2-robots-txt-robots-txt.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
import { createEventModule } from "@wix/sdk-runtime/event-definition-modules";
var getRobotsTxt4 = /* @__PURE__ */ createRESTModule(getRobotsTxt3);
var updateRobotsTxt4 = /* @__PURE__ */ createRESTModule(updateRobotsTxt3);
var onRobotsTxtUpdated2 = createEventModule(onRobotsTxtUpdated);
export {
  WebhookIdentityType,
  getRobotsTxt4 as getRobotsTxt,
  onRobotsTxtUpdated2 as onRobotsTxtUpdated,
  updateRobotsTxt4 as updateRobotsTxt
};
