// @wix/auto_sdk_forms_form-spam-submission-reports/build/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  ErrorType: () => ErrorType,
  ReportReason: () => ReportReason,
  SortOrder: () => SortOrder,
  SubmissionStatus: () => SubmissionStatus,
  WebhookIdentityType: () => WebhookIdentityType,
  bulkDeleteFormSpamSubmissionReport: () => bulkDeleteFormSpamSubmissionReport2,
  bulkDeleteFormSpamSubmissionReportByFilter: () => bulkDeleteFormSpamSubmissionReportByFilter2,
  bulkReportNotSpamSubmission: () => bulkReportNotSpamSubmission2,
  bulkReportSpamSubmission: () => bulkReportSpamSubmission2,
  checkForSpam: () => checkForSpam2,
  countFormSpamSubmissionReports: () => countFormSpamSubmissionReports2,
  createFormSpamSubmissionReport: () => createFormSpamSubmissionReport2,
  deleteFormSpamSubmissionReport: () => deleteFormSpamSubmissionReport2,
  getFormSpamSubmissionReport: () => getFormSpamSubmissionReport2,
  queryFormSpamSubmissionReportsByNamespace: () => queryFormSpamSubmissionReportsByNamespace2,
  reportNotSpamSubmission: () => reportNotSpamSubmission2,
  reportSpamSubmission: () => reportSpamSubmission2
});
module.exports = __toCommonJS(index_typings_exports);

// src/forms-v4-form-spam-submission-report-form-spam-submission-reports.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/forms-v4-form-spam-submission-report-form-spam-submission-reports.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixFormsV4FormSpamSubmissionReportServiceUrl(opts) {
  const domainToMappings = {
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/form-spam-submission-report-service",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/form-spam-submission-report-service",
        destPath: ""
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/form-spam-submission-report-service",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/form-spam-submission-report-service",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/form-spam-submission-report-service",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/form-spam-submission-report-service",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/form-spam-submission-report-service",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/form-spam-submission-report-service",
        destPath: ""
      },
      {
        srcPath: "/form-spam-submission-report-service",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/form-spam-submission-report-service",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/form-spam-submission-report-service",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_forms_form-spam-submission-reports";
function checkForSpam(payload) {
  function __checkForSpam({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "submission.createdDate" },
          { path: "submission.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "POST",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.CheckForSpam",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/form-spam-submission-reports/check-for-spam",
        data: serializedData,
        host
      }),
      data: serializedData
    };
    return metadata;
  }
  return __checkForSpam;
}
function createFormSpamSubmissionReport(payload) {
  function __createFormSpamSubmissionReport({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "formSpamSubmissionReport.createdDate" },
          { path: "formSpamSubmissionReport.reportedDate" },
          { path: "formSpamSubmissionReport.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "POST",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.CreateFormSpamSubmissionReport",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/form-spam-submission-reports",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "formSpamSubmissionReport.createdDate" },
            { path: "formSpamSubmissionReport.reportedDate" },
            { path: "formSpamSubmissionReport.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createFormSpamSubmissionReport;
}
function getFormSpamSubmissionReport(payload) {
  function __getFormSpamSubmissionReport({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "GET",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.GetFormSpamSubmissionReport",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/form-spam-submission-reports/{formSpamSubmissionReportId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "formSpamSubmissionReport.createdDate" },
            { path: "formSpamSubmissionReport.reportedDate" },
            { path: "formSpamSubmissionReport.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getFormSpamSubmissionReport;
}
function deleteFormSpamSubmissionReport(payload) {
  function __deleteFormSpamSubmissionReport({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "DELETE",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.DeleteFormSpamSubmissionReport",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/form-spam-submission-reports/{formSpamSubmissionReportId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteFormSpamSubmissionReport;
}
function bulkDeleteFormSpamSubmissionReport(payload) {
  function __bulkDeleteFormSpamSubmissionReport({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "POST",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.BulkDeleteFormSpamSubmissionReport",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/bulk/form-spam-submission-reports/delete",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkDeleteFormSpamSubmissionReport;
}
function bulkDeleteFormSpamSubmissionReportByFilter(payload) {
  function __bulkDeleteFormSpamSubmissionReportByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "POST",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.BulkDeleteFormSpamSubmissionReportByFilter",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/bulk/form-spam-submission-reports/delete-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkDeleteFormSpamSubmissionReportByFilter;
}
function reportNotSpamSubmission(payload) {
  function __reportNotSpamSubmission({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "POST",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.ReportNotSpamSubmission",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/form-spam-submission-reports/{formSpamSubmissionReportId}/report-not-spam-submission",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "submission.createdDate" },
            { path: "submission.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __reportNotSpamSubmission;
}
function bulkReportNotSpamSubmission(payload) {
  function __bulkReportNotSpamSubmission({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "POST",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.BulkReportNotSpamSubmission",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/bulk/form-spam-submission-reports/report-not-spam-submission",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkReportNotSpamSubmission;
}
function reportSpamSubmission(payload) {
  function __reportSpamSubmission({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "POST",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.ReportSpamSubmission",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/submissions/{submissionId}/report-spam-submission",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "formSpamSubmissionReport.createdDate" },
            { path: "formSpamSubmissionReport.reportedDate" },
            { path: "formSpamSubmissionReport.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __reportSpamSubmission;
}
function bulkReportSpamSubmission(payload) {
  function __bulkReportSpamSubmission({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "POST",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.BulkReportSpamSubmission",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/bulk/submissions/report-spam-submission",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.reportedDate" },
            { path: "results.item.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkReportSpamSubmission;
}
function queryFormSpamSubmissionReportsByNamespace(payload) {
  function __queryFormSpamSubmissionReportsByNamespace({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "POST",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.QueryFormSpamSubmissionReportsByNamespace",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/form-spam-submission-reports/namespace/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "formSpamSubmissionReports.createdDate" },
            { path: "formSpamSubmissionReports.reportedDate" },
            { path: "formSpamSubmissionReports.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryFormSpamSubmissionReportsByNamespace;
}
function countFormSpamSubmissionReports(payload) {
  function __countFormSpamSubmissionReports({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "POST",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.CountFormSpamSubmissionReports",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/form-spam-submission-reports/count",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __countFormSpamSubmissionReports;
}

// src/forms-v4-form-spam-submission-report-form-spam-submission-reports.universal.ts
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var ReportReason = /* @__PURE__ */ ((ReportReason2) => {
  ReportReason2["UNKNOWN_REASON"] = "UNKNOWN_REASON";
  ReportReason2["EMAIL_QUOTA_REACHED"] = "EMAIL_QUOTA_REACHED";
  ReportReason2["IP_BLOCKLISTED"] = "IP_BLOCKLISTED";
  ReportReason2["EMAIL_BLOCKLISTED"] = "EMAIL_BLOCKLISTED";
  ReportReason2["AI_REPORTED"] = "AI_REPORTED";
  ReportReason2["MANUALLY_REPORTED"] = "MANUALLY_REPORTED";
  ReportReason2["MODERATION_SPI"] = "MODERATION_SPI";
  return ReportReason2;
})(ReportReason || {});
var SubmissionStatus = /* @__PURE__ */ ((SubmissionStatus2) => {
  SubmissionStatus2["UNKNOWN_SUBMISSION_STATUS"] = "UNKNOWN_SUBMISSION_STATUS";
  SubmissionStatus2["PENDING"] = "PENDING";
  SubmissionStatus2["CONFIRMED"] = "CONFIRMED";
  SubmissionStatus2["PAYMENT_WAITING"] = "PAYMENT_WAITING";
  SubmissionStatus2["PAYMENT_CANCELED"] = "PAYMENT_CANCELED";
  return SubmissionStatus2;
})(SubmissionStatus || {});
var ErrorType = /* @__PURE__ */ ((ErrorType2) => {
  ErrorType2["UNKNOWN_ERROR"] = "UNKNOWN_ERROR";
  ErrorType2["TYPE_ERROR"] = "TYPE_ERROR";
  ErrorType2["REQUIRED_VALUE_ERROR"] = "REQUIRED_VALUE_ERROR";
  ErrorType2["UNKNOWN_VALUE_ERROR"] = "UNKNOWN_VALUE_ERROR";
  ErrorType2["MAX_LENGTH_ERROR"] = "MAX_LENGTH_ERROR";
  ErrorType2["MIN_LENGTH_ERROR"] = "MIN_LENGTH_ERROR";
  ErrorType2["PATTERN_ERROR"] = "PATTERN_ERROR";
  ErrorType2["FORMAT_ERROR"] = "FORMAT_ERROR";
  ErrorType2["MAX_VALUE_ERROR"] = "MAX_VALUE_ERROR";
  ErrorType2["MIN_VALUE_ERROR"] = "MIN_VALUE_ERROR";
  ErrorType2["MULTIPLE_OF_VALUE_ERROR"] = "MULTIPLE_OF_VALUE_ERROR";
  ErrorType2["MIN_ITEMS_ERROR"] = "MIN_ITEMS_ERROR";
  ErrorType2["MAX_ITEMS_ERROR"] = "MAX_ITEMS_ERROR";
  ErrorType2["NOT_ALLOWED_VALUE_ERROR"] = "NOT_ALLOWED_VALUE_ERROR";
  ErrorType2["FIELDS_COMPATIBILITY_ERROR"] = "FIELDS_COMPATIBILITY_ERROR";
  ErrorType2["DISABLED_FORM_ERROR"] = "DISABLED_FORM_ERROR";
  ErrorType2["FORMS_COUNT_RESTRICTIONS_ERROR"] = "FORMS_COUNT_RESTRICTIONS_ERROR";
  ErrorType2["FIELDS_COUNT_RESTRICTIONS_ERROR"] = "FIELDS_COUNT_RESTRICTIONS_ERROR";
  ErrorType2["STEPS_COUNT_RESTRICTIONS_ERROR"] = "STEPS_COUNT_RESTRICTIONS_ERROR";
  ErrorType2["RULES_COUNT_RESTRICTIONS_ERROR"] = "RULES_COUNT_RESTRICTIONS_ERROR";
  ErrorType2["FILE_UPLOAD_RESTRICTIONS_ERROR"] = "FILE_UPLOAD_RESTRICTIONS_ERROR";
  ErrorType2["INVALID_PHONE_COUNTRY_CODE_ERROR"] = "INVALID_PHONE_COUNTRY_CODE_ERROR";
  ErrorType2["INVALID_STAFF_ID_ERROR"] = "INVALID_STAFF_ID_ERROR";
  ErrorType2["INVALID_LOCATION_ID_ERROR"] = "INVALID_LOCATION_ID_ERROR";
  ErrorType2["MISSING_SERVICE_OPTION_ERROR"] = "MISSING_SERVICE_OPTION_ERROR";
  ErrorType2["INVALID_SERVICE_OPTIONS_ERROR"] = "INVALID_SERVICE_OPTIONS_ERROR";
  ErrorType2["NO_AVAILABLE_SERVICE_OPTIONS_ERROR"] = "NO_AVAILABLE_SERVICE_OPTIONS_ERROR";
  return ErrorType2;
})(ErrorType || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function checkForSpam2(submission) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    submission
  });
  const reqOpts = checkForSpam(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { submission: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["submission"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function createFormSpamSubmissionReport2(formSpamSubmissionReport) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    formSpamSubmissionReport
  });
  const reqOpts = createFormSpamSubmissionReport(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.formSpamSubmissionReport;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { formSpamSubmissionReport: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["formSpamSubmissionReport"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getFormSpamSubmissionReport2(formSpamSubmissionReportId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    formSpamSubmissionReportId
  });
  const reqOpts = getFormSpamSubmissionReport(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.formSpamSubmissionReport;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { formSpamSubmissionReportId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["formSpamSubmissionReportId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteFormSpamSubmissionReport2(formSpamSubmissionReportId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    formSpamSubmissionReportId
  });
  const reqOpts = deleteFormSpamSubmissionReport(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { formSpamSubmissionReportId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["formSpamSubmissionReportId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkDeleteFormSpamSubmissionReport2(formId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    formId,
    formSpamSubmissionReportIds: options?.formSpamSubmissionReportIds
  });
  const reqOpts = bulkDeleteFormSpamSubmissionReport(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          formId: "$[0]",
          formSpamSubmissionReportIds: "$[1].formSpamSubmissionReportIds"
        },
        singleArgumentUnchanged: false
      },
      ["formId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkDeleteFormSpamSubmissionReportByFilter2(filter) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ filter });
  const reqOpts = bulkDeleteFormSpamSubmissionReportByFilter(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { filter: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["filter"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function reportNotSpamSubmission2(formSpamSubmissionReportId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    formSpamSubmissionReportId
  });
  const reqOpts = reportNotSpamSubmission(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { formSpamSubmissionReportId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["formSpamSubmissionReportId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkReportNotSpamSubmission2(formId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    formId,
    formSpamSubmissionReportIds: options?.formSpamSubmissionReportIds
  });
  const reqOpts = bulkReportNotSpamSubmission(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          formId: "$[0]",
          formSpamSubmissionReportIds: "$[1].formSpamSubmissionReportIds"
        },
        singleArgumentUnchanged: false
      },
      ["formId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function reportSpamSubmission2(submissionId, reportReason, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    submissionId,
    reportReason,
    reportReasonDescription: options?.reportReasonDescription
  });
  const reqOpts = reportSpamSubmission(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          submissionId: "$[0]",
          reportReason: "$[1]",
          reportReasonDescription: "$[2].reportReasonDescription"
        },
        singleArgumentUnchanged: false
      },
      ["submissionId", "reportReason", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkReportSpamSubmission2(formId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    formId,
    submissionIds: options?.submissionIds,
    reportReason: options?.reportReason,
    returnEntity: options?.returnEntity,
    reportReasonDescription: options?.reportReasonDescription
  });
  const reqOpts = bulkReportSpamSubmission(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          formId: "$[0]",
          submissionIds: "$[1].submissionIds",
          reportReason: "$[1].reportReason",
          returnEntity: "$[1].returnEntity",
          reportReasonDescription: "$[1].reportReasonDescription"
        },
        singleArgumentUnchanged: false
      },
      ["formId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryFormSpamSubmissionReportsByNamespace2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryFormSpamSubmissionReportsByNamespace(
        payload
      );
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [])
      );
      return {
        items: transformedData?.formSpamSubmissionReports,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function countFormSpamSubmissionReports2(formIds, namespace) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    formIds,
    namespace
  });
  const reqOpts = countFormSpamSubmissionReports(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { formIds: "$[0]", namespace: "$[1]" },
        singleArgumentUnchanged: false
      },
      ["formIds", "namespace"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ErrorType,
  ReportReason,
  SortOrder,
  SubmissionStatus,
  WebhookIdentityType,
  bulkDeleteFormSpamSubmissionReport,
  bulkDeleteFormSpamSubmissionReportByFilter,
  bulkReportNotSpamSubmission,
  bulkReportSpamSubmission,
  checkForSpam,
  countFormSpamSubmissionReports,
  createFormSpamSubmissionReport,
  deleteFormSpamSubmissionReport,
  getFormSpamSubmissionReport,
  queryFormSpamSubmissionReportsByNamespace,
  reportNotSpamSubmission,
  reportSpamSubmission
});
