// @wix/unidriver-puppeteer/dist/esm/UniDriverClass.js
/* eslint-disable @typescript-eslint/no-unsafe-member-access */
/* eslint-disable @typescript-eslint/no-unsafe-call */
/* eslint-disable @typescript-eslint/no-unsafe-return */
import { UniDriverBase, bindUniDriverApiMethods } from '@wix/unidriver-common';
import { UniDriverClassList } from './UniDriverClassList';
import { enterText } from './utils/enterText';
import { getPage } from './utils/getPage';
import { applyShims } from './utils/applyShims';
export class UniDriverClass extends UniDriverBase {
    constructor(ctx) {
        super(ctx);
        Object.defineProperty(this, "type", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: 'puppeteer'
        });
        Object.defineProperty(this, "defaultTimeBudget", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: 5_000
        });
        Object.defineProperty(this, "api", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: {
                Class: UniDriverClass,
                ListClass: UniDriverClassList,
                getType: () => 'puppeteer',
                unwrap: async (base) => base,
                getNative: async (base) => {
                    return {
                        element: base,
                        page: await getPage(this.ctx),
                        selector: (await this.ctx.selector) ?? '',
                    };
                },
                getElementsBySelector: async (base, selector) => {
                    return base.$$(selector);
                },
                click: async (base, options) => {
                    return base.click({ clickCount: options?.type === 'double' ? 2 : 1 });
                },
                text: async (base) => {
                    return base.evaluate((_base) => {
                        return String(_base.textContent ?? '');
                    });
                },
                value: async (base) => {
                    return base.evaluate((_base) => {
                        const value = _base.value;
                        return typeof value === 'number' ? String(value) : (value ?? '');
                    });
                },
                enterText: async (base, value, options) => {
                    return enterText(base, value, options);
                },
                enterValue: async (base, value, options) => {
                    return enterText(base, value, {
                        ...options,
                        clear: options.shouldClear,
                    });
                },
                pressKey: async (base, keyMeta) => {
                    return base.press(keyMeta.key);
                },
                attr: async (base, attr) => {
                    return base.evaluate((_base, _attr) => {
                        return _base.getAttribute(_attr);
                    }, attr);
                },
                hasClass: async (base, className) => {
                    return base.evaluate((_base, _className) => {
                        return _base.classList.contains(_className);
                    }, className);
                },
                prop: async (base, prop) => {
                    return base.evaluate((_base, _prop) => {
                        return _base[_prop];
                    }, prop);
                },
                mouse: {
                    hover: async (base) => {
                        return base.hover();
                    },
                    press: async (base) => {
                        return this.fireEvent(base, 'mousedown');
                    },
                    release: async (base) => {
                        return this.fireEvent(base, 'mouseup');
                    },
                    leave: async (base) => {
                        await this.fireEvent(base, 'mouseout');
                        await this.fireEvent(base, 'mouseleave');
                    },
                    moveTo: async (base, to) => {
                        const toElement = await to.unwrap();
                        const boundingBox = await toElement.boundingBox();
                        if (boundingBox) {
                            return base.evaluate((_base, _boundingBox) => {
                                const { x, y } = _boundingBox;
                                _base.dispatchEvent(new MouseEvent('mousemove', {
                                    clientX: x,
                                    clientY: y,
                                    bubbles: true,
                                    cancelable: false,
                                }));
                            }, {
                                x: boundingBox.x,
                                y: boundingBox.y,
                            });
                        }
                        else {
                            throw new Error(`Cannot find target element`);
                        }
                    },
                },
            }
        });
        applyShims();
        bindUniDriverApiMethods(this, ['isDisplayed', 'scrollIntoView']);
    }
    async isDisplayed() {
        const base = await this.getBaseElement();
        return base.isIntersectingViewport();
    }
    async scrollIntoView() {
        const base = await this.getBaseElement();
        await base.hover();
    }
    fireEvent(base, eventName) {
        return base.evaluate((_base, _eventName) => {
            _base.dispatchEvent(new Event(_eventName, {
                bubbles: true,
                cancelable: true,
            }));
        }, eventName);
    }
}
