// @wix/bex-core/dist/esm/state/StringFilterState/StringFilterState.js
import { EventEmitter } from 'events';
import { action, computed, makeObservable, observable } from 'mobx';
import { refreshFilter } from '../Filter/refreshFilter';
export class StringFilterState {
    constructor(params) {
        this.events = new EventEmitter();
        this._toArray = (value) => {
            return [value];
        };
        this.name = params.name ?? '';
        this.persistent = params.persistent;
        this.defaultValue = params.defaultValue;
        this.itemKey = (item) => item;
        this.itemName = (item) => item;
        this._value = params.initialValue ?? this.defaultValue;
        this.matches =
            params.matches ??
                ((item, value) => value == null || value === '' || typeof item === 'string'
                    ? value.toLowerCase().includes(item.toLowerCase())
                    : false);
        this.equals = params.equals;
        makeObservable(this, {
            _value: observable.ref,
            persistent: observable.ref,
            isEmpty: computed,
            size: computed,
            value: computed,
            reset: action,
            clone: action,
            setValue: action,
            changeValue: action,
            scheduleRefresh: action,
        });
    }
    get value() {
        return this._value;
    }
    get toArray() {
        return [this._value];
    }
    get isEmpty() {
        return !this._value.length;
    }
    get size() {
        return this.isEmpty ? 0 : 1;
    }
    get toQueryString() {
        return this._value;
    }
    applyFromQueryString(str) {
        this.setValue(str);
    }
    encode(value) {
        return value;
    }
    decode(raw) {
        if (typeof raw === 'string') {
            return raw;
        }
        return '';
    }
    setValue(value, { emitEvents, ...options } = {}) {
        this._value = value;
        if (emitEvents) {
            for (const event of emitEvents) {
                this.events.emit(event, options);
            }
        }
    }
    changeValue(value, { emitEvents = [], ...options } = {}) {
        this.setValue(value, { ...options, emitEvents: ['change', ...emitEvents] });
    }
    refresh(value, options) {
        refreshFilter(this, value, options);
    }
    scheduleRefresh(value) {
        this.changeValue(value, { emitEvents: ['scheduleRefresh'] });
    }
    reset(options = {}) {
        this.setValue(this.defaultValue, options);
    }
    remove(_, options = {}) {
        this.reset(options);
    }
    hasDiff(item) {
        return this.value !== item;
    }
    clone(params) {
        const { defaultValue, value: initialValue, matches, persistent, name, } = this;
        return new StringFilterState({
            defaultValue,
            initialValue,
            matches,
            persistent,
            name,
            ...params,
        });
    }
}
export const stringFilter = (params = {}) => new StringFilterState({
    defaultValue: '',
    ...params,
});
