// @wix/ricos/dist/cjs/plugin-ai/exports/image-alt-text-input/alt-text-input.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var alt_text_input_exports = {};
__export(alt_text_input_exports, {
  createAltTextInput: () => createAltTextInput
});
module.exports = __toCommonJS(alt_text_input_exports);
var import_design_system = require("@wix/design-system");
var import_wix_ui_icons_common = require("@wix/wix-ui-icons-common");
var import_react = __toESM(require("react"));
var import_ricos_context = require("./../../../ricos-context/index.js");
var import_data_hooks = require("../../data-hooks.js");
var import_get_image_url = require("../../utils/get-image-url.js");
var import_use_ai_request_service = require("../../utils/use-ai-request-service.js");
const AltTextInput = ({
  image,
  altText,
  onChange,
  config,
  tooltipAppendTo,
  logBi,
  disabled
}) => {
  const { t, localeContent } = (0, import_react.useContext)(import_ricos_context.RicosContext);
  const { fetchPrompt } = (0, import_use_ai_request_service.useAiRequestService)(config);
  const [requestState, setRequestState] = import_react.default.useState({ type: "INIT" });
  const inputRef = import_react.default.useRef(null);
  const isDestroyed = import_react.default.useRef(false);
  import_react.default.useEffect(() => {
    isDestroyed.current = false;
    return () => {
      isDestroyed.current = true;
    };
  }, []);
  const generateAltText = async () => {
    if (!image) {
      return;
    }
    setRequestState({ type: "LOADING" });
    inputRef.current?.calculateComputedRows();
    const result = await fetchPrompt({
      promptSlug: "generate-alt-text",
      promptParams: {
        imageUrl: (0, import_get_image_url.getImageUrl)({ image }),
        siteLanguage: localeContent
      },
      additionalTags: "image alt text generation"
    });
    if (isDestroyed.current) {
      return;
    }
    if (result.type === "SUCCESS") {
      setRequestState({ type: "INIT" });
      logBi({
        dataHook: import_data_hooks.AI_EXPORTS_ALT_TEXT_GENERATE_BUTTON,
        value: result.results?.[0] ?? altText
      });
      onChange(result.results?.[0] ?? altText);
      inputRef.current?.calculateComputedRows();
      return;
    }
    setRequestState({ type: "ERROR", error: t("AIPlugin_Prompt_GenerateAltText_Error") });
  };
  return /* @__PURE__ */ import_react.default.createElement(
    import_design_system.FormField,
    {
      label: t("ImageSettings_Alt_Label"),
      infoContent: /* @__PURE__ */ import_react.default.createElement(import_design_system.Box, { direction: "vertical", gap: "SP1" }, /* @__PURE__ */ import_react.default.createElement(import_design_system.Text, { light: true, size: "small" }, t("ImageSettings_Alt_Label_Tooltip"))),
      infoTooltipProps: {
        appendTo: tooltipAppendTo ?? ((el) => Boolean(el.getAttribute("data-hook")?.includes("ricos-portal")))
      },
      suffix: image ? /* @__PURE__ */ import_react.default.createElement(import_design_system.Box, { direction: "vertical", gap: "SP1" }, /* @__PURE__ */ import_react.default.createElement(
        import_design_system.TextButton,
        {
          dataHook: import_data_hooks.AI_EXPORTS_ALT_TEXT_GENERATE_BUTTON,
          disabled: requestState.type === "LOADING" || disabled,
          size: "tiny",
          prefixIcon: /* @__PURE__ */ import_react.default.createElement(import_wix_ui_icons_common.SparklesSmall, null),
          onClick: () => {
            logBi({ dataHook: import_data_hooks.AI_EXPORTS_ALT_TEXT_GENERATE_BUTTON });
            generateAltText();
          }
        },
        t("AIPlugin_ImagePlugin_AltText_Generate_Text")
      )) : void 0
    },
    /* @__PURE__ */ import_react.default.createElement(
      import_design_system.InputArea,
      {
        ref: inputRef,
        dataHook: import_data_hooks.AI_EXPORTS_ALT_TEXT_INPUT_AREA,
        placeholder: t("ImageSettings_Alt_Input_Placeholder"),
        onChange: (e) => {
          logBi({ dataHook: import_data_hooks.AI_EXPORTS_ALT_TEXT_INPUT_AREA, value: e.target.value });
          onChange(e.target.value);
        },
        value: requestState.type === "LOADING" ? t("AIPlugin_ImagePlugin_AltText_InProgress") : altText,
        rows: 3,
        readOnly: requestState.type === "LOADING",
        status: requestState.type === "ERROR" ? "error" : requestState.type === "LOADING" ? "loading" : void 0,
        statusMessage: requestState.type === "ERROR" ? requestState.error : void 0,
        tooltipPlacement: "top-end",
        size: "small",
        disabled
      }
    )
  );
};
const createAltTextInput = (config) => {
  return (props) => /* @__PURE__ */ import_react.default.createElement(AltTextInput, { ...props, config });
};
