// @wix/ricos/dist/cjs/plugin-audio/viewer/audio-embed-viewer.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var audio_embed_viewer_exports = {};
__export(audio_embed_viewer_exports, {
  default: () => audio_embed_viewer_default
});
module.exports = __toCommonJS(audio_embed_viewer_exports);
var import_react = __toESM(require("react"));
var import_ricos_content = require("./../../ricos-content/index.js");
var import_ricos_context = require("./../../ricos-context/index.js");
var import_next_schema = require("./../../ricos-schema/next-schema/index.js");
var import_html_viewer = require("../../plugin-html/viewer/html-viewer.js");
var import_consts = require("../consts.js");
const AudioEmbedViewer = ({ node, settings }) => {
  const pluginsEvents = (0, import_react.useContext)(import_ricos_context.PluginsEventsContext) || {};
  const nodeId = node.id;
  const html = node.audioData?.html;
  const url = node.audioData?.audio?.src?.url ?? "";
  const htmlCompProps = {
    htmlData: {
      html,
      source: import_next_schema.HTMLData_Source.HTML,
      containerData: {
        width: {
          custom: "100%"
        },
        height: {
          custom: "100%"
        },
        url
      }
    },
    settings
  };
  (0, import_react.useEffect)(() => {
    nodeId && pluginsEvents?.publishMediaPluginsDetails?.({
      pluginId: import_ricos_content.AUDIO_TYPE,
      nodeId,
      creator: void 0,
      title: void 0,
      track_duration: void 0,
      type: import_consts.AUDIO_TYPES.spotify,
      url
    });
  }, []);
  return /* @__PURE__ */ import_react.default.createElement(import_html_viewer.HtmlViewer, { ...htmlCompProps });
};
var audio_embed_viewer_default = AudioEmbedViewer;
