// @wix/auto_sdk_data_tasks/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  EnvironmentOriginal: () => Environment,
  OperationOriginal: () => Operation,
  StatusOriginal: () => Status,
  TypeOriginal: () => Type,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  cancelTask: () => cancelTask2,
  createTask: () => createTask2,
  getTask: () => getTask2,
  listTasks: () => listTasks2
});
module.exports = __toCommonJS(meta_exports);

// src/data-autocms-v2-background-task-tasks.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixCloudAutocmsTasksV1TaskServiceUrl(opts) {
  const domainToMappings = {
    "editor._base_domain_": [
      {
        srcPath: "/_api/autocms/v1/tasks",
        destPath: "/v1"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/autocms/v1/tasks",
        destPath: "/v1"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/autocms/v1/tasks",
        destPath: "/v1"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/autocms/v1/tasks",
        destPath: "/v1"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/autocms/v1/tasks",
        destPath: "/v1"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/autocms/v1/tasks",
        destPath: "/v1"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/cms/v1/tasks",
        destPath: "/v1"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/autocms/v1/tasks",
        destPath: "/v1"
      }
    ],
    _: [
      {
        srcPath: "/_api/autocms/v1/tasks",
        destPath: "/v1"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_data_tasks";
function createTask(payload) {
  function __createTask({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "task.startedAt" },
          { path: "task.finishedAt" },
          {
            path: "task.updatePublishStatusOptions.scheduleDraftStatusOptions.date"
          },
          {
            path: "task.updatePublishStatusOptions.schedulePublishedStatusOptions.date"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.data.autocms.v2.background_task",
      method: "POST",
      methodFqn: "wix.cloud.autocms.tasks.v1.TaskService.CreateTask",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixCloudAutocmsTasksV1TaskServiceUrl({
        protoPath: "/v1",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "task.startedAt" },
            { path: "task.finishedAt" },
            {
              path: "task.updatePublishStatusOptions.scheduleDraftStatusOptions.date"
            },
            {
              path: "task.updatePublishStatusOptions.schedulePublishedStatusOptions.date"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createTask;
}
function listTasks(payload) {
  function __listTasks({ host }) {
    const metadata = {
      entityFqdn: "wix.data.autocms.v2.background_task",
      method: "GET",
      methodFqn: "wix.cloud.autocms.tasks.v1.TaskService.ListTasks",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixCloudAutocmsTasksV1TaskServiceUrl({
        protoPath: "/v1",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "tasks.startedAt" },
            { path: "tasks.finishedAt" },
            {
              path: "tasks.updatePublishStatusOptions.scheduleDraftStatusOptions.date"
            },
            {
              path: "tasks.updatePublishStatusOptions.schedulePublishedStatusOptions.date"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listTasks;
}
function getTask(payload) {
  function __getTask({ host }) {
    const metadata = {
      entityFqdn: "wix.data.autocms.v2.background_task",
      method: "GET",
      methodFqn: "wix.cloud.autocms.tasks.v1.TaskService.GetTask",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixCloudAutocmsTasksV1TaskServiceUrl({
        protoPath: "/v1/{taskId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "task.startedAt" },
            { path: "task.finishedAt" },
            {
              path: "task.updatePublishStatusOptions.scheduleDraftStatusOptions.date"
            },
            {
              path: "task.updatePublishStatusOptions.schedulePublishedStatusOptions.date"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getTask;
}
function cancelTask(payload) {
  function __cancelTask({ host }) {
    const metadata = {
      entityFqdn: "wix.data.autocms.v2.background_task",
      method: "DELETE",
      methodFqn: "wix.cloud.autocms.tasks.v1.TaskService.CancelTask",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixCloudAutocmsTasksV1TaskServiceUrl({
        protoPath: "/v1/{taskId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "task.startedAt" },
            { path: "task.finishedAt" },
            {
              path: "task.updatePublishStatusOptions.scheduleDraftStatusOptions.date"
            },
            {
              path: "task.updatePublishStatusOptions.schedulePublishedStatusOptions.date"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __cancelTask;
}

// src/data-autocms-v2-background-task-tasks.types.ts
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNKNOWN"] = "UNKNOWN";
  Type2["DELETE_BY_QUERY"] = "DELETE_BY_QUERY";
  Type2["COPY_FIELD_DATA"] = "COPY_FIELD_DATA";
  Type2["EDIT_FIELD_DATA"] = "EDIT_FIELD_DATA";
  Type2["UPDATE_PUBLISH_STATUS"] = "UPDATE_PUBLISH_STATUS";
  return Type2;
})(Type || {});
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["NEW"] = "NEW";
  Status2["RUNNING"] = "RUNNING";
  Status2["COMPLETED"] = "COMPLETED";
  Status2["FAILED"] = "FAILED";
  return Status2;
})(Status || {});
var Environment = /* @__PURE__ */ ((Environment2) => {
  Environment2["UNKNOWN_ENVIRONMENT"] = "UNKNOWN_ENVIRONMENT";
  Environment2["SANDBOX"] = "SANDBOX";
  Environment2["SANDBOX_PREFERRED"] = "SANDBOX_PREFERRED";
  Environment2["LIVE"] = "LIVE";
  return Environment2;
})(Environment || {});
var Operation = /* @__PURE__ */ ((Operation2) => {
  Operation2["UNDEFINED"] = "UNDEFINED";
  Operation2["SET_PUBLISHED_STATUS"] = "SET_PUBLISHED_STATUS";
  Operation2["SET_DRAFT_STATUS"] = "SET_DRAFT_STATUS";
  Operation2["SCHEDULE_PUBLISHED_STATUS"] = "SCHEDULE_PUBLISHED_STATUS";
  Operation2["SCHEDULE_DRAFT_STATUS"] = "SCHEDULE_DRAFT_STATUS";
  Operation2["CANCEL_SCHEDULING"] = "CANCEL_SCHEDULING";
  return Operation2;
})(Operation || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/data-autocms-v2-background-task-tasks.meta.ts
function createTask2() {
  const payload = {};
  const getRequestOptions = createTask(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listTasks2() {
  const payload = {};
  const getRequestOptions = listTasks(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getTask2() {
  const payload = { taskId: ":taskId" };
  const getRequestOptions = getTask(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/{taskId}",
    pathParams: { taskId: "taskId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function cancelTask2() {
  const payload = { taskId: ":taskId" };
  const getRequestOptions = cancelTask(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/{taskId}",
    pathParams: { taskId: "taskId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  EnvironmentOriginal,
  OperationOriginal,
  StatusOriginal,
  TypeOriginal,
  WebhookIdentityTypeOriginal,
  cancelTask,
  createTask,
  getTask,
  listTasks
});
