// @wix/auto_sdk_payments_payment-acceptance-configurations/build/es/meta.mjs
// src/payments-payment-acceptance-configurations-v1-payment-acceptance-configuration-payment-acceptance-configurations.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixPaymentsPaymentAcceptanceConfigurationsV1PaymentAcceptanceConfigurationServiceUrl(opts) {
  const domainToMappings = {};
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_payment-acceptance-configurations";
function createPaymentAcceptanceConfiguration(payload) {
  function __createPaymentAcceptanceConfiguration({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "paymentAcceptanceConfiguration.createdDate" },
          { path: "paymentAcceptanceConfiguration.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.payments.payment_acceptance_configurations.v1.payment_acceptance_configuration",
      method: "POST",
      methodFqn: "wix.payments.payment_acceptance_configurations.v1.PaymentAcceptanceConfigurationService.CreatePaymentAcceptanceConfiguration",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPaymentAcceptanceConfigurationsV1PaymentAcceptanceConfigurationServiceUrl(
        {
          protoPath: "/v1/payment-acceptance-configurations",
          data: serializedData,
          host
        }
      ),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "paymentAcceptanceConfiguration.createdDate" },
            { path: "paymentAcceptanceConfiguration.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createPaymentAcceptanceConfiguration;
}
function getPaymentAcceptanceConfiguration(payload) {
  function __getPaymentAcceptanceConfiguration({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.payment_acceptance_configurations.v1.payment_acceptance_configuration",
      method: "GET",
      methodFqn: "wix.payments.payment_acceptance_configurations.v1.PaymentAcceptanceConfigurationService.GetPaymentAcceptanceConfiguration",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPaymentAcceptanceConfigurationsV1PaymentAcceptanceConfigurationServiceUrl(
        {
          protoPath: "/v1/payment-acceptance-configurations/{paymentAcceptanceConfigurationId}",
          data: payload,
          host
        }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "paymentAcceptanceConfiguration.createdDate" },
            { path: "paymentAcceptanceConfiguration.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getPaymentAcceptanceConfiguration;
}
function updatePaymentAcceptanceConfiguration(payload) {
  function __updatePaymentAcceptanceConfiguration({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "paymentAcceptanceConfiguration.createdDate" },
          { path: "paymentAcceptanceConfiguration.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.payments.payment_acceptance_configurations.v1.payment_acceptance_configuration",
      method: "PATCH",
      methodFqn: "wix.payments.payment_acceptance_configurations.v1.PaymentAcceptanceConfigurationService.UpdatePaymentAcceptanceConfiguration",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPaymentAcceptanceConfigurationsV1PaymentAcceptanceConfigurationServiceUrl(
        {
          protoPath: "/v1/payment-acceptance-configurations/{paymentAcceptanceConfiguration.id}",
          data: serializedData,
          host
        }
      ),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "paymentAcceptanceConfiguration.createdDate" },
            { path: "paymentAcceptanceConfiguration.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updatePaymentAcceptanceConfiguration;
}
function deletePaymentAcceptanceConfiguration(payload) {
  function __deletePaymentAcceptanceConfiguration({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.payment_acceptance_configurations.v1.payment_acceptance_configuration",
      method: "DELETE",
      methodFqn: "wix.payments.payment_acceptance_configurations.v1.PaymentAcceptanceConfigurationService.DeletePaymentAcceptanceConfiguration",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPaymentAcceptanceConfigurationsV1PaymentAcceptanceConfigurationServiceUrl(
        {
          protoPath: "/v1/payment-acceptance-configurations/{paymentAcceptanceConfigurationId}",
          data: payload,
          host
        }
      ),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deletePaymentAcceptanceConfiguration;
}
function queryPaymentAcceptanceConfigurations(payload) {
  function __queryPaymentAcceptanceConfigurations({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.payment_acceptance_configurations.v1.payment_acceptance_configuration",
      method: "POST",
      methodFqn: "wix.payments.payment_acceptance_configurations.v1.PaymentAcceptanceConfigurationService.QueryPaymentAcceptanceConfigurations",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPaymentAcceptanceConfigurationsV1PaymentAcceptanceConfigurationServiceUrl(
        {
          protoPath: "/v1/payment-acceptance-configurations/query",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "paymentAcceptanceConfigurations.createdDate" },
            { path: "paymentAcceptanceConfigurations.updatedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixPaymentsPaymentAcceptanceConfigurationsV1PaymentAcceptanceConfigurationServiceUrl(
            {
              protoPath: "/v1/payment-acceptance-configurations/query",
              data: payload,
              host
            }
          ),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryPaymentAcceptanceConfigurations;
}
function bulkUpdatePaymentAcceptanceConfigurationTags(payload) {
  function __bulkUpdatePaymentAcceptanceConfigurationTags({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.payment_acceptance_configurations.v1.payment_acceptance_configuration",
      method: "POST",
      methodFqn: "wix.payments.payment_acceptance_configurations.v1.PaymentAcceptanceConfigurationService.BulkUpdatePaymentAcceptanceConfigurationTags",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPaymentAcceptanceConfigurationsV1PaymentAcceptanceConfigurationServiceUrl(
        {
          protoPath: "/v1/bulk/payment-acceptance-configurations/update-tags",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdatePaymentAcceptanceConfigurationTags;
}
function bulkUpdatePaymentAcceptanceConfigurationTagsByFilter(payload) {
  function __bulkUpdatePaymentAcceptanceConfigurationTagsByFilter({
    host
  }) {
    const metadata = {
      entityFqdn: "wix.payments.payment_acceptance_configurations.v1.payment_acceptance_configuration",
      method: "POST",
      methodFqn: "wix.payments.payment_acceptance_configurations.v1.PaymentAcceptanceConfigurationService.BulkUpdatePaymentAcceptanceConfigurationTagsByFilter",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPaymentAcceptanceConfigurationsV1PaymentAcceptanceConfigurationServiceUrl(
        {
          protoPath: "/v1/bulk/payment-acceptance-configurations/update-tags-by-filter",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdatePaymentAcceptanceConfigurationTagsByFilter;
}

// src/payments-payment-acceptance-configurations-v1-payment-acceptance-configuration-payment-acceptance-configurations.meta.ts
function createPaymentAcceptanceConfiguration2() {
  const payload = {};
  const getRequestOptions = createPaymentAcceptanceConfiguration(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/payment-acceptance-configurations",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getPaymentAcceptanceConfiguration2() {
  const payload = {
    paymentAcceptanceConfigurationId: ":paymentAcceptanceConfigurationId"
  };
  const getRequestOptions = getPaymentAcceptanceConfiguration(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/payment-acceptance-configurations/{paymentAcceptanceConfigurationId}",
    pathParams: {
      paymentAcceptanceConfigurationId: "paymentAcceptanceConfigurationId"
    },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updatePaymentAcceptanceConfiguration2() {
  const payload = {
    paymentAcceptanceConfiguration: { id: ":paymentAcceptanceConfigurationId" }
  };
  const getRequestOptions = updatePaymentAcceptanceConfiguration(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/payment-acceptance-configurations/{paymentAcceptanceConfiguration.id}",
    pathParams: {
      paymentAcceptanceConfigurationId: "paymentAcceptanceConfigurationId"
    },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deletePaymentAcceptanceConfiguration2() {
  const payload = {
    paymentAcceptanceConfigurationId: ":paymentAcceptanceConfigurationId"
  };
  const getRequestOptions = deletePaymentAcceptanceConfiguration(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/payment-acceptance-configurations/{paymentAcceptanceConfigurationId}",
    pathParams: {
      paymentAcceptanceConfigurationId: "paymentAcceptanceConfigurationId"
    },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryPaymentAcceptanceConfigurations2() {
  const payload = {};
  const getRequestOptions = queryPaymentAcceptanceConfigurations(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/payment-acceptance-configurations/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdatePaymentAcceptanceConfigurationTags2() {
  const payload = {};
  const getRequestOptions = bulkUpdatePaymentAcceptanceConfigurationTags(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/payment-acceptance-configurations/update-tags",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdatePaymentAcceptanceConfigurationTagsByFilter2() {
  const payload = {};
  const getRequestOptions = bulkUpdatePaymentAcceptanceConfigurationTagsByFilter(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/payment-acceptance-configurations/update-tags-by-filter",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  bulkUpdatePaymentAcceptanceConfigurationTags2 as bulkUpdatePaymentAcceptanceConfigurationTags,
  bulkUpdatePaymentAcceptanceConfigurationTagsByFilter2 as bulkUpdatePaymentAcceptanceConfigurationTagsByFilter,
  createPaymentAcceptanceConfiguration2 as createPaymentAcceptanceConfiguration,
  deletePaymentAcceptanceConfiguration2 as deletePaymentAcceptanceConfiguration,
  getPaymentAcceptanceConfiguration2 as getPaymentAcceptanceConfiguration,
  queryPaymentAcceptanceConfigurations2 as queryPaymentAcceptanceConfigurations,
  updatePaymentAcceptanceConfiguration2 as updatePaymentAcceptanceConfiguration
};
