// @wix/feedback-loop-structure/dist/cjs/validator.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.validate = void 0;
const zod_1 = require("zod");
const dropdown_1 = require("./components/dropdown");
const faces_rating_bar_1 = require("./components/faces-rating-bar");
const segmented_toggle_1 = require("./components/segmented-toggle");
const text_input_1 = require("./components/text-input");
const thumbnails_1 = require("./components/thumbnails");
const markdown_1 = require("./components/markdown");
const schema = zod_1.z.object({
    id: zod_1.z.string(),
    title: zod_1.z.string(),
    components: zod_1.z
        .array(zod_1.z.discriminatedUnion('type', [
        dropdown_1.DropdownSchema,
        faces_rating_bar_1.FacesRatingBarSchema,
        segmented_toggle_1.SegmentedToggleSchema,
        text_input_1.TextInputSchema,
        thumbnails_1.ThumbnailsSchema,
        markdown_1.MarkdownSchema,
    ]))
        .min(1)
        .superRefine((val, ctx) => {
        const uniqueQuestionIds = new Set();
        const allQuestionIds = [];
        val.forEach((component) => {
            if ('questionId' in component) {
                uniqueQuestionIds.add(component.questionId);
                allQuestionIds.push(component.questionId);
            }
        });
        if (allQuestionIds.length === 0) {
            ctx.addIssue({
                code: zod_1.z.ZodIssueCode.custom,
                message: 'You are required to provide at least one component that defines a question',
            });
        }
        if (allQuestionIds.length !== uniqueQuestionIds.size) {
            ctx.addIssue({
                code: zod_1.z.ZodIssueCode.custom,
                message: 'Question ids must be unique',
            });
        }
    }),
});
const validate = (structure) => {
    const result = schema.safeParse(structure);
    if (!result.success) {
        console.error(result.error);
    }
    return result;
};
exports.validate = validate;
