// @wix/ricos/dist/cjs/plugin-button/components/button-color-picker.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var button_color_picker_exports = {};
__export(button_color_picker_exports, {
  ButtonColorPicker: () => ButtonColorPicker
});
module.exports = __toCommonJS(button_color_picker_exports);
var import_react = __toESM(require("react"));
var import_ricos_context = require("./../../ricos-context/index.js");
var import_ModalContext = require("./../../ricos-context/ModalContext.js");
var import_bi_context_provider = require("./../../_shared/bi-context-provider/bi-context-provider.js");
var import_use_node_attrs = require("./../../_shared/use-node-attrs.js");
var import_wix_rich_content_common = require("./../../wix-rich-content-common/index.js");
var import_new_color_picker = require("./../../ui-components/color-picker/new-color-picker/index.js");
const ButtonColorPicker = ({
  modalId,
  nodeId,
  nodeType,
  colorType,
  isHoverState,
  referenceElement,
  config,
  defaultTextColor,
  defaultBackgroundColor,
  onColorSelected,
  onColorAdded: onColorAddedProp
}) => {
  const modalService = (0, import_react.useContext)(import_ModalContext.ModalServiceContext);
  const { t } = (0, import_react.useContext)(import_ricos_context.RicosContext);
  const { pmNodeAttrs } = (0, import_use_node_attrs.useNodeAttrs)(nodeId, nodeType);
  const logBi = (0, import_bi_context_provider.useBiNext)({
    modalId,
    pluginId: nodeType === "LINK_BUTTON" ? "linkButton" : "actionButton",
    nodeId
  });
  const themeColors = (0, import_wix_rich_content_common.resolveColorsPalette)(
    config.colors ? Object.values(config.colors) : [],
    void 0,
    57
  );
  const resolveCustomColors = (args) => (0, import_wix_rich_content_common.resolveColorsPalette)(args.getColorFn?.() ?? [], args.currentColor, 57);
  const [allCustomColors, setAllCustomColors] = (0, import_react.useState)(() => ({
    textColor: resolveCustomColors({
      currentColor: pmNodeAttrs?.styles?.textColor ?? defaultTextColor,
      getColorFn: config.getTextColors
    }),
    borderColor: resolveCustomColors({
      currentColor: pmNodeAttrs?.styles?.borderColor ?? defaultBackgroundColor,
      getColorFn: config.getBorderColors
    }),
    backgroundColor: resolveCustomColors({
      currentColor: pmNodeAttrs?.styles?.backgroundColor ?? defaultBackgroundColor,
      getColorFn: config.getBackgroundColors
    })
  }));
  const onColorAdded = (color) => {
    onColorAddedProp(color);
    setAllCustomColors((prev) => ({ ...prev, [colorType]: [...prev[colorType], color] }));
    onColorSelected(color);
  };
  const closeModal = () => modalService.closeModal(modalId);
  const onResetColor = () => onColorSelected(void 0);
  const currentColor = pmNodeAttrs?.styles?.[`${colorType}${isHoverState ? "Hover" : ""}`]?.toUpperCase();
  const customColors = allCustomColors[colorType];
  return /* @__PURE__ */ import_react.default.createElement(
    import_new_color_picker.NewColorPicker,
    {
      themeColors,
      customColors,
      currentColor,
      title: t("ButtonColourModal_Title"),
      closeModal,
      onColorSelected,
      onColorAdded,
      onResetColor,
      referenceElement,
      logBi
    }
  );
};
