// @wix/consent-policy-manager/dist/cjs/consent-policy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getCurrentConsentPolicy = getCurrentConsentPolicy;
exports.resetConsentPolicy = resetConsentPolicy;
exports.onConsentPolicyChanged = onConsentPolicyChanged;
exports.setConsentPolicy = setConsentPolicy;
const consts_js_1 = require("./consts.js");
/**
 * Gets the site visitor's consent policy regarding allowed cookies and 3rd-party data transfers for GDPR or CCPA compliance.
 *
 * Retrieves the site visitor's consent policy details, including which cookies are allowed and whether data transfer to 3rd parties is permitted.
 *
 * @returns The current consent policy details, or undefined if not available
 *
 * @example Get a member's consent policy
 * ```typescript
 * import { consentPolicyManager } from '@wix/site';
 *
 * const policyDetails = consentPolicyManager.getCurrentConsentPolicy();
 * const policy = policyDetails.policy;
 *
 * ```
 */
function getCurrentConsentPolicy() {
    const consentPolicyObject = getConsentPolicyObject();
    if (!consentPolicyObject?.getCurrentConsentPolicy) {
        console.warn("[ConsentPolicy] getCurrentConsentPolicy is not available");
        return;
    }
    try {
        return consentPolicyObject.getCurrentConsentPolicy();
    }
    catch (error) {
        console.error("[ConsentPolicy] Error getting current consent policy:", error);
        throw error;
    }
}
/**
 * Removes the current policy from the site visitor's browser
 * and resets the site visitor's consent policy to the default policy for the site.
 *
 * @returns boolean indicating if the consent policy was successfully reset, or undefined if not available
 *
 * @example Reset the current consent policy
 * ```typescript
 * import { consentPolicyManager } from '@wix/site';
 *
 * const success = consentPolicyManager.resetPolicy();
 * ```
 */
function resetConsentPolicy() {
    const consentPolicyObject = getConsentPolicyObject();
    if (!consentPolicyObject?.resetPolicy) {
        console.warn("[ConsentPolicy] resetPolicy is not available");
        return;
    }
    try {
        return consentPolicyObject.resetPolicy();
    }
    catch (error) {
        console.error("[ConsentPolicy] Error resetting consent policy:", error);
        throw error;
    }
}
/**
 * Triggered when a site visitor's consent policy was changed using setConsentPolicy() or reset using resetConsentPolicy().
 *
 * Use the onConsentPolicyChanged() method for code you want to run after the site visitor's current consent policy was changed.
 *
 * @param handler - The function to run when the consent policy changes
 *
 * @requiredField handler
 *
 * @example
 * ```typescript
 * import { consentPolicyManager } from '@wix/site';
 *
 * consentPolicyManager.onConsentPolicyChanged((event) => {
 *   const policy = event.policy;
 *   console.log(event);
 * });
 *
 * // Full event object:
 * // {
 * //   "defaultPolicy": false,
 * //   "policy": {
 * //     "functional": false,
 * //     "analytics": true,
 * //     "advertising": false,
 * //     "dataToThirdParty": false,
 * //     "essential": true
 * //   }
 * // }
 * ```
 */
function onConsentPolicyChanged(handler) {
    const consentPolicyObject = getConsentPolicyObject();
    if (consentPolicyObject?.onConsentPolicyChanged) {
        return consentPolicyObject.onConsentPolicyChanged(handler);
    }
    if (typeof document === "undefined") {
        console.warn("[ConsentPolicy] onConsentPolicyChanged can only be used in browser environment, document is undefined");
        return;
    }
    try {
        document.addEventListener(consts_js_1.EVENTS.CONSENT_POLICY_CHANGED, ((event) => {
            handler(event.detail);
        }));
    }
    catch (error) {
        console.error("[ConsentPolicy] Error registering consent policy change handler:", error);
        throw error;
    }
}
/**
 * Sets the current site visitor's consent policy
 * regarding allowed cookies and data transfer to 3rd parties,
 * such as for GDPR or CCPA purposes.
 *
 * Changes to the consent policy take affect after the page is refreshed.
 *
 * @param policy - An object representing the cookies of the site visitor's consent policy.
 * @param successCallback - Optional callback function called on success with PolicyDetails
 * @param errorCallback - Optional callback function called on error with error message
 *
 * @requiredField policy
 *
 * @returns Promise that resolves with the updated PolicyDetails when the consent policy is successfully set, or undefined if not available
 *
 * @example Set a member's consent policy
 * ```typescript
 * import { consentPolicyManager } from '@wix/site';
 *
 * const policy = {
 *   essential: true,
 *   functional: true,
 *   analytics: true,
 *   advertising: false,
 *   dataToThirdParty: false,
 * };
 *
 * const successCallback = (policyDetails: PolicyDetails) => {
 *   console.log(policyDetails);
 * };
 *
 * const errorCallback = (error: string) => {
 *   console.log(error);
 * };
 *
 * const policyDetails = await consentPolicyManager.setConsentPolicy(policy, successCallback, errorCallback);
 * console.log(policyDetails);
 * ```
 */
async function setConsentPolicy(policy, successCallback, errorCallback) {
    const consentPolicyObject = getConsentPolicyObject();
    if (!consentPolicyObject?.setConsentPolicy) {
        console.warn("[ConsentPolicy] setConsentPolicy is not available");
        return;
    }
    try {
        return consentPolicyObject.setConsentPolicy(policy, successCallback || (() => { }), errorCallback || (() => { }));
    }
    catch (error) {
        console.error("[ConsentPolicy] Error setting consent policy:", error);
        throw error;
    }
}
function getConsentPolicyObject() {
    const maybeConsentPolicyObject = 
    // @ts-expect-error - ignoring globals type error
    typeof globals !== "undefined"
        // @ts-expect-error - ignoring globals type error
        ? globals?.$ns?.['wix-window']?.consentPolicy
        : undefined;
    if (maybeConsentPolicyObject) {
        return maybeConsentPolicyObject;
    }
    if (typeof window === "undefined") {
        console.warn("[ConsentPolicy] can only be used in browser environment");
        return;
    }
    return window?.consentPolicyManager;
}
