// @wix/auto_sdk_members_custom-field-applications/build/es/meta.mjs
// src/members-v1-custom-field-application-custom-field-applications.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(opts) {
  const domainToMappings = {
    _: [
      {
        srcPath: "/_api/members/v1/custom-fields-applications",
        destPath: "/v1/custom-fields-applications"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/members/v1/custom-fields-applications",
        destPath: "/v1/custom-fields-applications"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/members/v1/custom-fields-applications",
        destPath: "/v1/custom-fields-applications"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/members/v1/custom-fields-applications",
        destPath: "/v1/custom-fields-applications"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_custom-field-applications";
function createCustomFieldApplication(payload) {
  function __createCustomFieldApplication({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_application",
      method: "POST",
      methodFqn: "com.wixpress.members.customfields.applications.CustomFieldApplications.CreateCustomFieldApplication",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(
        { protoPath: "/v1/custom-fields-applications", data: payload, host }
      ),
      data: payload
    };
    return metadata;
  }
  return __createCustomFieldApplication;
}
function deleteCustomFieldApplication(payload) {
  function __deleteCustomFieldApplication({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_application",
      method: "DELETE",
      methodFqn: "com.wixpress.members.customfields.applications.CustomFieldApplications.DeleteCustomFieldApplication",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(
        {
          protoPath: "/v1/custom-fields-applications/{customFieldId}",
          data: payload,
          host
        }
      ),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteCustomFieldApplication;
}
function updateCustomFieldApplication(payload) {
  function __updateCustomFieldApplication({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_application",
      method: "PATCH",
      methodFqn: "com.wixpress.members.customfields.applications.CustomFieldApplications.UpdateCustomFieldApplication",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(
        {
          protoPath: "/v1/custom-fields-applications/{application.customFieldId}",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __updateCustomFieldApplication;
}
function getCustomFieldApplication(payload) {
  function __getCustomFieldApplication({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_application",
      method: "GET",
      methodFqn: "com.wixpress.members.customfields.applications.CustomFieldApplications.GetCustomFieldApplication",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(
        {
          protoPath: "/v1/custom-fields-applications/{customFieldId}",
          data: payload,
          host
        }
      ),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getCustomFieldApplication;
}
function getCustomFieldApplications(payload) {
  function __getCustomFieldApplications({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_application",
      method: "POST",
      methodFqn: "com.wixpress.members.customfields.applications.CustomFieldApplications.GetCustomFieldApplications",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(
        {
          protoPath: "/v1/custom-fields-applications/applications",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __getCustomFieldApplications;
}
function getMembersCustomFieldApplications(payload) {
  function __getMembersCustomFieldApplications({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_application",
      method: "POST",
      methodFqn: "com.wixpress.members.customfields.applications.CustomFieldApplications.GetMembersCustomFieldApplications",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(
        {
          protoPath: "/v1/custom-fields-applications/members",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.applications.customField.createdDate" },
            { path: "results.applications.customField.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getMembersCustomFieldApplications;
}
function getRolesCustomFieldApplications(payload) {
  function __getRolesCustomFieldApplications({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_application",
      method: "POST",
      methodFqn: "com.wixpress.members.customfields.applications.CustomFieldApplications.GetRolesCustomFieldApplications",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(
        {
          protoPath: "/v1/custom-fields-applications/roles",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.applications.customField.createdDate" },
            { path: "results.applications.customField.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getRolesCustomFieldApplications;
}

// src/members-v1-custom-field-application-custom-field-applications.types.ts
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNKNOWN"] = "UNKNOWN";
  Type2["ROLE"] = "ROLE";
  Type2["BADGE"] = "BADGE";
  Type2["PRICING_PLAN"] = "PRICING_PLAN";
  Type2["MEMBER"] = "MEMBER";
  return Type2;
})(Type || {});
var ExclusionType = /* @__PURE__ */ ((ExclusionType2) => {
  ExclusionType2["UNKNOWN"] = "UNKNOWN";
  ExclusionType2["ROLE"] = "ROLE";
  ExclusionType2["BADGE"] = "BADGE";
  ExclusionType2["PRICING_PLAN"] = "PRICING_PLAN";
  ExclusionType2["MEMBER"] = "MEMBER";
  return ExclusionType2;
})(ExclusionType || {});
var Privacy = /* @__PURE__ */ ((Privacy2) => {
  Privacy2["UNKNOWN"] = "UNKNOWN";
  Privacy2["PUBLIC"] = "PUBLIC";
  Privacy2["PRIVATE"] = "PRIVATE";
  return Privacy2;
})(Privacy || {});
var FieldTypeType = /* @__PURE__ */ ((FieldTypeType2) => {
  FieldTypeType2["UNKNOWN"] = "UNKNOWN";
  FieldTypeType2["TEXT"] = "TEXT";
  FieldTypeType2["NUMBER"] = "NUMBER";
  FieldTypeType2["DATE"] = "DATE";
  FieldTypeType2["URL"] = "URL";
  FieldTypeType2["SOCIAL"] = "SOCIAL";
  return FieldTypeType2;
})(FieldTypeType || {});
var SocialTypeType = /* @__PURE__ */ ((SocialTypeType2) => {
  SocialTypeType2["UNKNOWN"] = "UNKNOWN";
  SocialTypeType2["FACEBOOK"] = "FACEBOOK";
  SocialTypeType2["INSTAGRAM"] = "INSTAGRAM";
  SocialTypeType2["LINKEDIN"] = "LINKEDIN";
  SocialTypeType2["TWITTER"] = "TWITTER";
  SocialTypeType2["YOUTUBE"] = "YOUTUBE";
  SocialTypeType2["PINTEREST"] = "PINTEREST";
  SocialTypeType2["TIKTOK"] = "TIKTOK";
  SocialTypeType2["DEVIANTART"] = "DEVIANTART";
  SocialTypeType2["SOUNDCLOUD"] = "SOUNDCLOUD";
  SocialTypeType2["TUMBLR"] = "TUMBLR";
  SocialTypeType2["VIMEO"] = "VIMEO";
  SocialTypeType2["VKONTAKTE"] = "VKONTAKTE";
  SocialTypeType2["ODNOKLASSNIKI"] = "ODNOKLASSNIKI";
  SocialTypeType2["OTHER"] = "OTHER";
  return SocialTypeType2;
})(SocialTypeType || {});
var Origin = /* @__PURE__ */ ((Origin2) => {
  Origin2["UNKNOWN"] = "UNKNOWN";
  Origin2["CUSTOM"] = "CUSTOM";
  Origin2["CONTACT"] = "CONTACT";
  Origin2["SYSTEM"] = "SYSTEM";
  return Origin2;
})(Origin || {});
var AppliesToAppliesTo = /* @__PURE__ */ ((AppliesToAppliesTo2) => {
  AppliesToAppliesTo2["ALL_MEMBERS"] = "ALL_MEMBERS";
  AppliesToAppliesTo2["SELECTED_MEMBERS"] = "SELECTED_MEMBERS";
  return AppliesToAppliesTo2;
})(AppliesToAppliesTo || {});
var Section = /* @__PURE__ */ ((Section2) => {
  Section2["GENERAL"] = "GENERAL";
  Section2["SOCIAL"] = "SOCIAL";
  Section2["DISPLAY_INFO"] = "DISPLAY_INFO";
  Section2["ADDRESS"] = "ADDRESS";
  return Section2;
})(Section || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/members-v1-custom-field-application-custom-field-applications.meta.ts
function createCustomFieldApplication2() {
  const payload = {};
  const getRequestOptions = createCustomFieldApplication(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/custom-fields-applications",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteCustomFieldApplication2() {
  const payload = { customFieldId: ":customFieldId" };
  const getRequestOptions = deleteCustomFieldApplication(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/custom-fields-applications/{customFieldId}",
    pathParams: { customFieldId: "customFieldId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateCustomFieldApplication2() {
  const payload = {
    application: { customFieldId: ":applicationCustomFieldId" }
  };
  const getRequestOptions = updateCustomFieldApplication(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/custom-fields-applications/{application.customFieldId}",
    pathParams: { applicationCustomFieldId: "applicationCustomFieldId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getCustomFieldApplication2() {
  const payload = { customFieldId: ":customFieldId" };
  const getRequestOptions = getCustomFieldApplication(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/custom-fields-applications/{customFieldId}",
    pathParams: { customFieldId: "customFieldId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getCustomFieldApplications2() {
  const payload = {};
  const getRequestOptions = getCustomFieldApplications(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/custom-fields-applications/applications",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getMembersCustomFieldApplications2() {
  const payload = {};
  const getRequestOptions = getMembersCustomFieldApplications(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/custom-fields-applications/members",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getRolesCustomFieldApplications2() {
  const payload = {};
  const getRequestOptions = getRolesCustomFieldApplications(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/custom-fields-applications/roles",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  AppliesToAppliesTo as AppliesToAppliesToOriginal,
  ExclusionType as ExclusionTypeOriginal,
  FieldTypeType as FieldTypeTypeOriginal,
  Origin as OriginOriginal,
  Privacy as PrivacyOriginal,
  Section as SectionOriginal,
  SocialTypeType as SocialTypeTypeOriginal,
  Type as TypeOriginal,
  WebhookIdentityType as WebhookIdentityTypeOriginal,
  createCustomFieldApplication2 as createCustomFieldApplication,
  deleteCustomFieldApplication2 as deleteCustomFieldApplication,
  getCustomFieldApplication2 as getCustomFieldApplication,
  getCustomFieldApplications2 as getCustomFieldApplications,
  getMembersCustomFieldApplications2 as getMembersCustomFieldApplications,
  getRolesCustomFieldApplications2 as getRolesCustomFieldApplications,
  updateCustomFieldApplication2 as updateCustomFieldApplication
};
