// @wix/auto_sdk_seo_robots-txt/build/internal/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  WebhookIdentityType: () => WebhookIdentityType,
  getRobotsTxt: () => getRobotsTxt4,
  onRobotsTxtUpdated: () => onRobotsTxtUpdated2,
  updateRobotsTxt: () => updateRobotsTxt4
});
module.exports = __toCommonJS(index_exports);

// src/promote-seo-v2-robots-txt-robots-txt.public.ts
var import_rename_all_nested_keys2 = require("@wix/sdk-runtime/rename-all-nested-keys");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_sdk_types = require("@wix/sdk-types");

// src/promote-seo-v2-robots-txt-robots-txt.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/promote-seo-v2-robots-txt-robots-txt.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressPromoteSeoRobotsServerV2RobotsServiceV2Url(opts) {
  const domainToMappings = {
    "www._base_domain_": [
      {
        srcPath: "/_api/promote-seo-robots-server",
        destPath: "/api"
      },
      {
        srcPath: "/_api/promote-seo-robots-server/v2/robots",
        destPath: "/v2/robots"
      },
      {
        srcPath: "/_api/promote-seo-robots-server/v2/ads",
        destPath: "/v2/ads"
      },
      {
        srcPath: "/_api/promote-seo-robots-server/v2/llms",
        destPath: "/v2/llms"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/promote-seo-robots-server",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/promote-seo-robots-server",
        destPath: "/api"
      },
      {
        srcPath: "/_api/promote-seo-robots-server/v2/robots",
        destPath: "/v2/robots"
      },
      {
        srcPath: "/_api/promote-seo-robots-server/v2/ads",
        destPath: "/v2/ads"
      },
      {
        srcPath: "/_api/promote-seo-robots-server/v2/llms",
        destPath: "/v2/llms"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/promote-seo-robots-server/v2/robots",
        destPath: "/v2/robots"
      },
      {
        srcPath: "/promote-seo-robots-server/v2/ads",
        destPath: "/v2/ads"
      },
      {
        srcPath: "/promote-seo-txt-file-server/v2/ads",
        destPath: "/v2/ads"
      },
      {
        srcPath: "/promote-seo-txt-file-server/v2/robots",
        destPath: "/v2/robots"
      },
      {
        srcPath: "/promote-seo-robots-server/v2/llms",
        destPath: "/v2/llms"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/promote/marketing/v2/ads",
        destPath: "/v2/ads"
      },
      {
        srcPath: "/promote/seo/v2/llms",
        destPath: "/v2/llms"
      },
      {
        srcPath: "/promote/seo/v2/robots",
        destPath: "/v2/robots"
      }
    ],
    _: [
      {
        srcPath: "/promote/marketing/v2/ads",
        destPath: "/v2/ads"
      },
      {
        srcPath: "/promote/seo/v2/llms",
        destPath: "/v2/llms"
      },
      {
        srcPath: "/promote/seo/v2/robots",
        destPath: "/v2/robots"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_seo_robots-txt";
function getRobotsTxt(payload) {
  function __getRobotsTxt({ host }) {
    const metadata = {
      entityFqdn: "wix.promote.seo.v2.robots_txt",
      method: "GET",
      methodFqn: "com.wixpress.promote.seo.robots.server.v2.RobotsServiceV2.GetRobotsTxt",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressPromoteSeoRobotsServerV2RobotsServiceV2Url({
        protoPath: "/v2/robots",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getRobotsTxt;
}
function updateRobotsTxt(payload) {
  function __updateRobotsTxt({ host }) {
    const metadata = {
      entityFqdn: "wix.promote.seo.v2.robots_txt",
      method: "PUT",
      methodFqn: "com.wixpress.promote.seo.robots.server.v2.RobotsServiceV2.UpdateRobotsTxt",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressPromoteSeoRobotsServerV2RobotsServiceV2Url({
        protoPath: "/v2/robots",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __updateRobotsTxt;
}

// src/promote-seo-v2-robots-txt-robots-txt.universal.ts
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function getRobotsTxt2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    subdomain: options?.subdomain
  });
  const reqOpts = getRobotsTxt(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { subdomain: "$[0].subdomain" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateRobotsTxt2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    robotsTxt: options?.robotsTxt
  });
  const reqOpts = updateRobotsTxt(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { robotsTxt: "$[0].robotsTxt" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/promote-seo-v2-robots-txt-robots-txt.public.ts
function getRobotsTxt3(httpClient) {
  return (options) => getRobotsTxt2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function updateRobotsTxt3(httpClient) {
  return (options) => updateRobotsTxt2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
var onRobotsTxtUpdated = (0, import_sdk_types.EventDefinition)(
  "wix.promote.seo.robots_txt_updated",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths.transformPaths)(event, [
      {
        transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
        paths: [{ path: "metadata.eventTime" }]
      }
    ])
  )
)();

// src/promote-seo-v2-robots-txt-robots-txt.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var import_event_definition_modules = require("@wix/sdk-runtime/event-definition-modules");
var getRobotsTxt4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getRobotsTxt3);
var updateRobotsTxt4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(updateRobotsTxt3);
var onRobotsTxtUpdated2 = (0, import_event_definition_modules.createEventModule)(onRobotsTxtUpdated);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  WebhookIdentityType,
  getRobotsTxt,
  onRobotsTxtUpdated,
  updateRobotsTxt
});
