// pro-gallery-lib/dist/cjs/core/helpers/dimensionsHelper.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const index_js_1 = tslib_1.__importDefault(require("../../common/constants/index.js"));
const layoutParams_structure_galleryRatio_includeExternalInfo_js_1 = tslib_1.__importDefault(require("../../settings/options/layoutParams_structure_galleryRatio_includeExternalInfo.js"));
const optionsMap_js_1 = tslib_1.__importDefault(require("./optionsMap.js"));
class DimensionsHelper {
    constructor() {
        this.options = {};
        this.container = {};
        this.id = '';
        this._cache = {};
    }
    getOrPutInCache(field, createValue) {
        if (this._cache[field])
            return this._cache[field];
        this._cache[field] = createValue();
        return this._cache[field];
    }
    dumpCache() {
        this._cache = {};
    }
    updateParams({ options, container, id }) {
        this.dumpCache();
        this.id = id || this.id;
        this.options = options || this.options;
        this.container = container || this.container;
        this.fixHeightForHorizontalGalleryIfNeeded();
    }
    getGalleryDimensions() {
        return this.getOrPutInCache('galleryDimensions', () => {
            const res = {
                galleryWidth: Math.ceil(this.getGalleryWidth()),
                galleryHeight: Math.ceil(this.getGalleryHeight()),
                scrollBase: this.container.scrollBase ? Math.ceil(this.container.scrollBase) : 0,
                height: Math.ceil(this.container.height),
                width: Math.ceil(this.container.width),
            };
            if (this.container.externalScrollBase) {
                //if was provided from the wrapper
                res.scrollBase += this.container.externalScrollBase;
            }
            if (this.options[optionsMap_js_1.default.layoutParams.thumbnails.enable] &&
                this.options[optionsMap_js_1.default.layoutParams.thumbnails.position] ===
                    index_js_1.default[optionsMap_js_1.default.layoutParams.thumbnails.position].OUTSIDE_GALLERY) {
                res.galleryHeight -= this.getThumbnailHeightDelta();
                res.galleryWidth -= this.getThumbnailWidthDelta();
            }
            if (this.options[optionsMap_js_1.default.layoutParams.thumbnails.enable]) {
                res.navigationBarHeight = this.getThumbnailHeightDelta() || res.galleryHeight;
                res.navigationBarWidth = this.getThumbnailWidthDelta() || res.galleryWidth;
            }
            return res;
        });
    }
    getGalleryWidth() {
        return this.getOrPutInCache('galleryWidth', () => {
            let width = Math.floor(this.container.width) + this.getDimensionFix() * 2; //add margins to width and then remove them in css negative margins
            if (this.options[optionsMap_js_1.default.layoutParams.navigationArrows.position] ===
                index_js_1.default[optionsMap_js_1.default.layoutParams.navigationArrows.position].OUTSIDE_GALLERY &&
                this.options[optionsMap_js_1.default.layoutParams.structure.scrollDirection] ===
                    index_js_1.default[optionsMap_js_1.default.layoutParams.structure.scrollDirection].HORIZONTAL) {
                width -=
                    2 *
                        (this.options[optionsMap_js_1.default.layoutParams.navigationArrows.size] +
                            40 +
                            this.options[optionsMap_js_1.default.layoutParams.structure.itemSpacing] / 2);
            }
            return width;
        });
    }
    getGalleryHeight() {
        return this.getOrPutInCache('galleryHeight', () => {
            //const offsetTop = this.options.scrollDirection === GALLERY_CONSTS[optionsMap.layoutParams.structure.scrollDirection].HORIZONTAL ? this.container.offsetTop : 0;
            const dimensionFix = () => this.options[optionsMap_js_1.default.layoutParams.structure.scrollDirection] ===
                index_js_1.default[optionsMap_js_1.default.layoutParams.structure.scrollDirection].HORIZONTAL
                ? this.getDimensionFix()
                : 0;
            const res = Math.floor((this.container.height > 0 ? this.container.height : 0) + dimensionFix());
            return res;
        });
    }
    getDimensionFix() {
        return this.getOrPutInCache('dimensionFix', () => {
            return (Number(this.options[optionsMap_js_1.default.layoutParams.structure.itemSpacing] / 2) -
                Number(this.options[optionsMap_js_1.default.layoutParams.structure.gallerySpacing]));
        });
    }
    getGalleryRatio() {
        return this.getOrPutInCache('galleryRatio', () => {
            const res = this.getGalleryDimensions();
            return res.galleryWidth / res.galleryHeight;
        });
    }
    getThumbnailSize() {
        const fixedThumbnailSize = this.options[optionsMap_js_1.default.layoutParams.thumbnails.size] +
            this.options[optionsMap_js_1.default.layoutParams.structure.gallerySpacing] +
            this.options[optionsMap_js_1.default.layoutParams.thumbnails.marginToGallery];
        return fixedThumbnailSize;
    }
    getThumbnailHeightDelta() {
        switch (this.options[optionsMap_js_1.default.layoutParams.thumbnails.alignment]) {
            case index_js_1.default[optionsMap_js_1.default.layoutParams.thumbnails.alignment].TOP:
            case index_js_1.default[optionsMap_js_1.default.layoutParams.thumbnails.alignment].BOTTOM:
                return this.getThumbnailSize();
            case index_js_1.default[optionsMap_js_1.default.layoutParams.thumbnails.alignment].RIGHT:
            case index_js_1.default[optionsMap_js_1.default.layoutParams.thumbnails.alignment].LEFT:
                return 0;
            default:
                break;
        }
    }
    getThumbnailWidthDelta() {
        switch (this.options[optionsMap_js_1.default.layoutParams.thumbnails.alignment]) {
            case index_js_1.default[optionsMap_js_1.default.layoutParams.thumbnails.alignment].TOP:
            case index_js_1.default[optionsMap_js_1.default.layoutParams.thumbnails.alignment].BOTTOM:
                return 0;
            case index_js_1.default[optionsMap_js_1.default.layoutParams.thumbnails.alignment].RIGHT:
            case index_js_1.default[optionsMap_js_1.default.layoutParams.thumbnails.alignment].LEFT:
                return this.getThumbnailSize();
            default:
                break;
        }
    }
    fixHeightForHorizontalGalleryIfNeeded() {
        if (this.options[optionsMap_js_1.default.layoutParams.structure.scrollDirection] ===
            index_js_1.default[optionsMap_js_1.default.layoutParams.structure.scrollDirection].HORIZONTAL &&
            this.options[optionsMap_js_1.default.layoutParams.structure.galleryRatio.value] > 0) {
            if (this.options[optionsMap_js_1.default.layoutParams.thumbnails.enable] &&
                this.options[optionsMap_js_1.default.layoutParams.thumbnails.position] ===
                    index_js_1.default[optionsMap_js_1.default.layoutParams.thumbnails.position].OUTSIDE_GALLERY) {
                switch (this.options[optionsMap_js_1.default.layoutParams.thumbnails.alignment]) {
                    case index_js_1.default[optionsMap_js_1.default.layoutParams.thumbnails.alignment].TOP:
                    case index_js_1.default[optionsMap_js_1.default.layoutParams.thumbnails.alignment].BOTTOM:
                        this.container.height =
                            this.container.width * this.options[optionsMap_js_1.default.layoutParams.structure.galleryRatio.value] +
                                this.getThumbnailHeightDelta();
                        break;
                    case index_js_1.default[optionsMap_js_1.default.layoutParams.thumbnails.alignment].RIGHT:
                    case index_js_1.default[optionsMap_js_1.default.layoutParams.thumbnails.alignment].LEFT:
                        this.container.height =
                            (this.container.width - this.getThumbnailWidthDelta()) *
                                this.options[optionsMap_js_1.default.layoutParams.structure.galleryRatio.value];
                        break;
                    default:
                        break;
                }
            }
            else {
                this.container.height =
                    this.container.width * this.options[optionsMap_js_1.default.layoutParams.structure.galleryRatio.value];
                if (!this.options[optionsMap_js_1.default.layoutParams.structure.galleryRatio.includeExternalInfo] &&
                    layoutParams_structure_galleryRatio_includeExternalInfo_js_1.default.isRelevant(this.options)) {
                    this.container.height += this.options.externalInfoHeight;
                }
            }
        }
    }
}
exports.default = new DimensionsHelper();
