// pro-gallery-lib/dist/cjs/core/helpers/migratorStore.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.reverseBooleanTo = exports.namingChange = exports.reverseBooleans = exports.changeNames = exports.nameChangedStylingParams = exports.reversedBehaviourParams = exports.nameChangedBehaviourParams = exports.reversedLayoutParams = exports.nameChangedLayoutParams = void 0;
const tslib_1 = require("tslib");
const optionsMap_js_1 = tslib_1.__importDefault(require("./optionsMap.js"));
exports.nameChangedLayoutParams = [
    ['galleryLayout', optionsMap_js_1.default.layoutParams.structure.galleryLayout],
    ['imageMargin', optionsMap_js_1.default.layoutParams.structure.itemSpacing],
    ['groupsPerStrip', optionsMap_js_1.default.layoutParams.groups.numberOfGroupsPerRow],
    ['cubeImages', optionsMap_js_1.default.layoutParams.crop.enable],
    ['smartCrop', optionsMap_js_1.default.layoutParams.crop.enableSmartCrop],
    ['cropOnlyFill', optionsMap_js_1.default.layoutParams.crop.cropOnlyFill],
    ['minItemSize', optionsMap_js_1.default.layoutParams.targetItemSize.minimum],
    ['scatter', optionsMap_js_1.default.layoutParams.structure.scatter.randomScatter],
    ['rotatingScatter', optionsMap_js_1.default.layoutParams.structure.scatter.manualScatter],
    ['numberOfImagesPerCol', optionsMap_js_1.default.layoutParams.structure.numberOfGridRows],
    //['collageAmount', optionsMap.layoutParams.groups.amount], //This doesnt really exist. need to eradicate as a refactor
    ['collageDensity', optionsMap_js_1.default.layoutParams.groups.density],
    ['chooseBestGroup', optionsMap_js_1.default.layoutParams.groups.groupByOrientation],
    ['groupSize', optionsMap_js_1.default.layoutParams.groups.groupSize],
    ['hasThumbnails', optionsMap_js_1.default.layoutParams.thumbnails.enable],
    ['thumbnailSpacings', optionsMap_js_1.default.layoutParams.thumbnails.spacing],
    ['thumbnailSize', optionsMap_js_1.default.layoutParams.thumbnails.size],
    ['showArrows', optionsMap_js_1.default.layoutParams.navigationArrows.enable],
    ['arrowsPadding', optionsMap_js_1.default.layoutParams.navigationArrows.padding],
    ['arrowsVerticalPosition', optionsMap_js_1.default.layoutParams.navigationArrows.verticalAlignment],
    ['arrowsSize', optionsMap_js_1.default.layoutParams.navigationArrows.size],
    ['imageInfoType', optionsMap_js_1.default.layoutParams.info.layout],
    //['textBoxWidthPercent', optionsMap.layoutParams.info.widthByPercent],
    ['textImageSpace', optionsMap_js_1.default.layoutParams.info.spacing],
    ['textBoxBorderWidth', optionsMap_js_1.default.layoutParams.info.border.width],
    ['textBoxBorderColor', optionsMap_js_1.default.layoutParams.info.border.color],
    ['textBoxBorderRadius', optionsMap_js_1.default.layoutParams.info.border.radius],
];
exports.reversedLayoutParams = [['useMaxDimensions', optionsMap_js_1.default.layoutParams.structure.enableStreching]];
exports.nameChangedBehaviourParams = [
    ['magnificationLevel', optionsMap_js_1.default.behaviourParams.item.content.magnificationValue],
    ['tiltAngleValue', optionsMap_js_1.default.behaviourParams.item.content.tiltAngleValue],
    ['videoLoop', optionsMap_js_1.default.behaviourParams.item.video.loop],
    ['showVideoPlayButton', optionsMap_js_1.default.behaviourParams.item.video.enablePlayButton],
    ['showVideoControls', optionsMap_js_1.default.behaviourParams.item.video.enableControls],
    ['enableVideoPlaceholder', optionsMap_js_1.default.behaviourParams.item.video.enablePlaceholder],
    ['overlayAnimation', optionsMap_js_1.default.behaviourParams.item.overlay.hoverAnimation],
    ['overlayPosition', optionsMap_js_1.default.behaviourParams.item.overlay.position],
    ['overlaySize', optionsMap_js_1.default.behaviourParams.item.overlay.size],
    ['overlaySizeType', optionsMap_js_1.default.behaviourParams.item.overlay.sizeUnits],
    ['overlayPadding', optionsMap_js_1.default.behaviourParams.item.overlay.padding],
    ['overlayBackground', optionsMap_js_1.default.behaviourParams.item.overlay.backgroundColor],
    ['imageHoverAnimation', optionsMap_js_1.default.behaviourParams.item.content.hoverAnimation],
    ['imagePlacementAnimation', optionsMap_js_1.default.behaviourParams.item.content.placementAnimation],
    ['imageLoadingMode', optionsMap_js_1.default.behaviourParams.item.content.loader],
    ['scrollSnap', optionsMap_js_1.default.behaviourParams.gallery.horizontal.enableScrollSnap],
    ['scrollAnimation', optionsMap_js_1.default.behaviourParams.gallery.scrollAnimation],
    ['shouldIndexDirectShareLinkInSEO', optionsMap_js_1.default.behaviourParams.gallery.enableIndexingShareLinks],
    ['loadMoreButtonText', optionsMap_js_1.default.behaviourParams.gallery.vertical.loadMore.text],
    ['slideAnimation', optionsMap_js_1.default.behaviourParams.gallery.horizontal.slideAnimation],
    ['slideTransition', optionsMap_js_1.default.behaviourParams.gallery.horizontal.slideTransition],
    ['scrollDuration', optionsMap_js_1.default.behaviourParams.gallery.horizontal.navigationDuration],
    ['slideshowLoop', optionsMap_js_1.default.behaviourParams.gallery.horizontal.loop],
    ['autoSlideshowInterval', optionsMap_js_1.default.behaviourParams.gallery.horizontal.autoSlide.interval],
    ['pauseAutoSlideshowOnHover', optionsMap_js_1.default.behaviourParams.gallery.horizontal.autoSlide.pauseOnHover],
    ['autoSlideshowContinuousSpeed', optionsMap_js_1.default.behaviourParams.gallery.horizontal.autoSlide.speed],
    ['allowSlideshowCounter', optionsMap_js_1.default.behaviourParams.gallery.horizontal.slideshowInfo.enableCounter],
    ['playButtonForAutoSlideShow', optionsMap_js_1.default.behaviourParams.gallery.horizontal.slideshowInfo.enablePlayButton],
];
exports.reversedBehaviourParams = [
    ['enableInfiniteScroll', optionsMap_js_1.default.behaviourParams.gallery.vertical.loadMore.enable],
    ['allowContextMenu', optionsMap_js_1.default.behaviourParams.gallery.blockContextMenu],
    ['enableScroll', optionsMap_js_1.default.behaviourParams.gallery.horizontal.blockScroll], //requires a reversal! (blocks instead of allowing),
];
exports.nameChangedStylingParams = [
    ['itemShadowBlur', optionsMap_js_1.default.stylingParams.itemShadowBlur],
    ['itemShadowDirection', optionsMap_js_1.default.stylingParams.itemShadowDirection],
    ['itemShadowOpacityAndColor', optionsMap_js_1.default.stylingParams.itemShadowOpacityAndColor],
    ['arrowsColor', optionsMap_js_1.default.stylingParams.arrowsColor],
    ['itemShadowSize', optionsMap_js_1.default.stylingParams.itemShadowSize],
    ['itemEnableShadow', optionsMap_js_1.default.stylingParams.itemEnableShadow],
    ['itemBorderRadius', optionsMap_js_1.default.stylingParams.itemBorderRadius],
    ['itemBorderColor', optionsMap_js_1.default.stylingParams.itemBorderColor],
    ['itemBorderWidth', optionsMap_js_1.default.stylingParams.itemBorderWidth],
];
//---------tooling---------//
function changeNames(obj, pairsArray) {
    let _obj = Object.assign({}, obj);
    for (const [oldName, newName] of pairsArray) {
        _obj = namingChange(_obj, oldName, newName);
    }
    return _obj;
}
exports.changeNames = changeNames;
function reverseBooleans(obj, pairsArray) {
    let _obj = Object.assign({}, obj);
    for (const [oldName, newName] of pairsArray) {
        _obj = reverseBooleanTo(_obj, oldName, newName);
    }
    return _obj;
}
exports.reverseBooleans = reverseBooleans;
function namingChange(obj, oldName, newName) {
    if (typeof obj[newName] === 'undefined' && typeof obj[oldName] !== 'undefined') {
        obj[newName] = obj[oldName];
    }
    delete obj[oldName];
    return obj;
}
exports.namingChange = namingChange;
function reverseBooleanTo(obj, oldName, newName) {
    if (typeof obj[newName] === 'undefined' && typeof obj[oldName] !== 'undefined') {
        obj[newName] = !obj[oldName];
    }
    delete obj[oldName];
    return obj;
}
exports.reverseBooleanTo = reverseBooleanTo;
