// pro-gallery-lib/dist/cjs/core/helpers/optionsConverter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.extendNestedOptionsToIncludeOldAndNew = exports.addMigratedOptions = exports.migrateOptions = void 0;
const tslib_1 = require("tslib");
const optionsUtils_js_1 = require("./optionsUtils.js");
const layoutParams_structure_galleryLayout_js_1 = require("../../common/constants/layoutParams_structure_galleryLayout.js");
const optionsMap_js_1 = tslib_1.__importDefault(require("./optionsMap.js"));
const optionsBackwardConverter_js_1 = require("./optionsBackwardConverter.js");
const migratorStore_js_1 = require("./migratorStore.js");
function extendNestedOptionsToIncludeOldAndNew(nestedOptions) {
    let flatOptions = (0, optionsUtils_js_1.flattenObject)(nestedOptions);
    let populatedFlatOptions = (0, optionsBackwardConverter_js_1.addOldOptions)(addMigratedOptions(flatOptions));
    return Object.assign(Object.assign({}, (0, optionsUtils_js_1.flatToNested)(populatedFlatOptions)), populatedFlatOptions);
}
exports.extendNestedOptionsToIncludeOldAndNew = extendNestedOptionsToIncludeOldAndNew;
function addMigratedOptions(flatOptions) {
    const flat_migrated = migrateOptions(flatOptions);
    let flat_combinedOptions = Object.assign(Object.assign({}, (0, optionsUtils_js_1.trimUndefinedValues_flat)(flat_migrated)), (0, optionsUtils_js_1.trimUndefinedValues_flat)(flatOptions));
    delete flat_combinedOptions.oldRefactoredOptionInCore;
    return flat_combinedOptions;
}
exports.addMigratedOptions = addMigratedOptions;
function migrateOptions(flatOptionsObject) {
    let migratedOptions = Object.assign({}, flatOptionsObject);
    // let newStyles = flattenObject(cloneDeep(oldStyles));
    ///----------- LAYOUT -------------///
    migratedOptions = (0, migratorStore_js_1.changeNames)(migratedOptions, migratorStore_js_1.nameChangedLayoutParams);
    migratedOptions = (0, migratorStore_js_1.reverseBooleans)(migratedOptions, migratorStore_js_1.reversedLayoutParams);
    migratedOptions = process_old_to_new_ThumbnailAlignment(migratedOptions);
    migratedOptions = process_old_to_new_ScrollDirection(migratedOptions);
    migratedOptions = process_old_to_new_LayoutOrientation(migratedOptions);
    migratedOptions = process_old_to_new_groupsOrder(migratedOptions);
    migratedOptions = process_old_to_new_repeatingGroupTypes(migratedOptions);
    migratedOptions = process_old_to_new_AllowedGroupTypes(migratedOptions);
    migratedOptions = process_old_to_new_NumberOfColumns(migratedOptions); // fixedColumns || numberOfImagesPerRow
    migratedOptions = process_old_to_new_targetItemSizeUnit(migratedOptions);
    migratedOptions = process_old_to_new_targetItemSizeValue(migratedOptions);
    migratedOptions = process_old_to_new_CroppedAlignment(migratedOptions);
    migratedOptions = process_old_to_new_CropRatio(migratedOptions);
    migratedOptions = process_old_to_new_textBoxSizeMode(migratedOptions);
    migratedOptions = process_old_to_new_columnRatios(migratedOptions);
    migratedOptions = process_old_to_new_cropMethod(migratedOptions);
    migratedOptions = process_old_to_new_responsiveMode(migratedOptions);
    migratedOptions = process_old_to_new_gallerySpacing(migratedOptions);
    migratedOptions = process_old_to_new_slideshowInfoSize(migratedOptions);
    migratedOptions = process_old_to_new_arrowsPosition(migratedOptions);
    ///----------- BEHAVIOUR -------------///
    migratedOptions = (0, migratorStore_js_1.changeNames)(migratedOptions, migratorStore_js_1.nameChangedBehaviourParams);
    migratedOptions = (0, migratorStore_js_1.reverseBooleans)(migratedOptions, migratorStore_js_1.reversedBehaviourParams);
    migratedOptions = process_old_to_new_ClickAction(migratedOptions);
    migratedOptions = process_old_to_new_VideoPlayTrigger(migratedOptions);
    migratedOptions = process_old_to_new_VideoVolume(migratedOptions);
    migratedOptions = process_old_to_new_VideoSpeed(migratedOptions);
    migratedOptions = process_old_to_new_OverlayHoveringBehaviour(migratedOptions);
    migratedOptions = process_old_to_new_InfoPlacement(migratedOptions);
    migratedOptions = process_old_to_new_layoutDirection(migratedOptions);
    migratedOptions = process_old_to_new_LoadMoreAmount(migratedOptions);
    migratedOptions = process_old_to_new_AutoSlideBehaviour(migratedOptions);
    migratedOptions = process_old_to_new_galleryTextAlign(migratedOptions);
    ///----------- STYLING -------------///
    migratedOptions = (0, migratorStore_js_1.changeNames)(migratedOptions, migratorStore_js_1.nameChangedStylingParams);
    delete migratedOptions.fullscreen;
    return migratedOptions;
}
exports.migrateOptions = migrateOptions;
//----- refactor functions ----------//
function process_old_to_new_columnRatios(obj) {
    var _a, _b, _c;
    let _obj = obj;
    _obj = (0, migratorStore_js_1.namingChange)(_obj, 'columnWidths', optionsMap_js_1.default.layoutParams.structure.columnRatios);
    if (((_a = _obj[optionsMap_js_1.default.layoutParams.structure.columnRatios]) === null || _a === void 0 ? void 0 : _a.length) === 0) {
        _obj[optionsMap_js_1.default.layoutParams.structure.columnRatios] = [];
    }
    else {
        _obj[optionsMap_js_1.default.layoutParams.structure.columnRatios] = _obj[optionsMap_js_1.default.layoutParams.structure.columnRatios]
            ? ((_b = _obj[optionsMap_js_1.default.layoutParams.structure.columnRatios]) === null || _b === void 0 ? void 0 : _b.split)
                ? [...(((_c = _obj[optionsMap_js_1.default.layoutParams.structure.columnRatios]) === null || _c === void 0 ? void 0 : _c.split(',')) || []).map(Number)]
                : _obj[optionsMap_js_1.default.layoutParams.structure.columnRatios]
            : undefined;
    }
    return _obj;
}
function process_old_to_new_targetItemSizeValue(obj) {
    let _obj = obj;
    let unit = _obj[optionsMap_js_1.default.layoutParams.targetItemSize.unit];
    let key;
    switch (unit) {
        case 'PIXEL':
            key = 'gallerySizePx';
            break;
        case 'SMART':
            key = 'gallerySize';
            break;
        case 'PERCENT':
            key = 'gallerySizeRatio';
            break;
    }
    _obj = (0, migratorStore_js_1.namingChange)(_obj, key, optionsMap_js_1.default.layoutParams.targetItemSize.value);
    delete _obj['gallerySizePx'];
    delete _obj['gallerySizeRatio'];
    delete _obj['gallerySize'];
    return _obj;
}
function process_old_to_new_cropMethod(obj) {
    var _a;
    let _obj = obj;
    _obj = (0, migratorStore_js_1.namingChange)(_obj, 'cubeType', optionsMap_js_1.default.layoutParams.crop.method);
    _obj[optionsMap_js_1.default.layoutParams.crop.method] = (_a = _obj[optionsMap_js_1.default.layoutParams.crop.method]) === null || _a === void 0 ? void 0 : _a.toUpperCase();
    return _obj;
}
function process_old_to_new_ThumbnailAlignment(obj) {
    var _a;
    //['galleryThumbnailsAlignment', optionsMap.layoutParams.thumbnails.alignment'],
    let _obj = obj;
    _obj = (0, migratorStore_js_1.namingChange)(_obj, 'galleryThumbnailsAlignment', optionsMap_js_1.default.layoutParams.thumbnails.alignment);
    _obj[optionsMap_js_1.default.layoutParams.thumbnails.alignment] =
        (_a = _obj[optionsMap_js_1.default.layoutParams.thumbnails.alignment]) === null || _a === void 0 ? void 0 : _a.toUpperCase();
    return _obj;
}
function process_old_to_new_VideoPlayTrigger(obj) {
    var _a;
    let _obj = obj;
    _obj = (0, migratorStore_js_1.namingChange)(_obj, 'videoPlay', optionsMap_js_1.default.behaviourParams.item.video.playTrigger);
    _obj[optionsMap_js_1.default.behaviourParams.item.video.playTrigger] =
        (_a = _obj[optionsMap_js_1.default.behaviourParams.item.video.playTrigger]) === null || _a === void 0 ? void 0 : _a.toUpperCase();
    return _obj;
}
function process_old_to_new_targetItemSizeUnit(obj) {
    let _obj = obj;
    _obj = (0, migratorStore_js_1.namingChange)(_obj, 'gallerySizeType', optionsMap_js_1.default.layoutParams.targetItemSize.unit);
    switch (_obj[optionsMap_js_1.default.layoutParams.targetItemSize.unit]) {
        case 'px':
            _obj[optionsMap_js_1.default.layoutParams.targetItemSize.unit] = 'PIXEL';
            break;
        case 'ratio':
            _obj[optionsMap_js_1.default.layoutParams.targetItemSize.unit] = 'PERCENT';
            break;
        case 'smart':
            _obj[optionsMap_js_1.default.layoutParams.targetItemSize.unit] = 'SMART';
            break;
    }
    return _obj;
}
function process_old_to_new_VideoVolume(obj) {
    let _obj = obj;
    _obj = (0, migratorStore_js_1.namingChange)(_obj, 'videoSound', optionsMap_js_1.default.behaviourParams.item.video.volume);
    if (typeof _obj[optionsMap_js_1.default.behaviourParams.item.video.volume] !== 'undefined') {
        _obj[optionsMap_js_1.default.behaviourParams.item.video.volume] = Number(_obj[optionsMap_js_1.default.behaviourParams.item.video.volume]);
    }
    return _obj;
}
function process_old_to_new_VideoSpeed(obj) {
    let _obj = obj;
    _obj = (0, migratorStore_js_1.namingChange)(_obj, 'videoSpeed', optionsMap_js_1.default.behaviourParams.item.video.speed);
    _obj[optionsMap_js_1.default.behaviourParams.item.video.speed] =
        Number(_obj[optionsMap_js_1.default.behaviourParams.item.video.speed]) >= 0
            ? Number(_obj[optionsMap_js_1.default.behaviourParams.item.video.speed])
            : undefined;
    return _obj;
}
function process_old_to_new_gallerySpacing(obj) {
    if (typeof obj[optionsMap_js_1.default.layoutParams.structure.gallerySpacing] !== 'undefined') {
        return obj;
    }
    let _obj = obj;
    let spacingVal;
    if (_obj['layoutParams_gallerySpacing'] >= 0) {
        spacingVal = _obj['layoutParams_gallerySpacing'];
    }
    else if (_obj['galleryMargin'] >= 0) {
        spacingVal = _obj['galleryMargin'];
    }
    _obj[optionsMap_js_1.default.layoutParams.structure.gallerySpacing] = spacingVal;
    delete _obj['layoutParams_gallerySpacing'];
    delete _obj['galleryMargin'];
    return _obj;
}
function process_old_to_new_arrowsPosition(obj) {
    let _obj = obj;
    _obj = (0, migratorStore_js_1.namingChange)(_obj, 'arrowsPosition', optionsMap_js_1.default.layoutParams.navigationArrows.position);
    switch (_obj[optionsMap_js_1.default.layoutParams.navigationArrows.position]) {
        case 0:
            _obj[optionsMap_js_1.default.layoutParams.navigationArrows.position] = 'ON_GALLERY';
            break;
        case 1:
            _obj[optionsMap_js_1.default.layoutParams.navigationArrows.position] = 'OUTSIDE_GALLERY';
            break;
        case 2:
            _obj[optionsMap_js_1.default.layoutParams.navigationArrows.position] = 'MOUSE_CURSOR';
            break;
        default:
            break;
    }
    return _obj;
}
function process_old_to_new_responsiveMode(obj) {
    let _obj = obj;
    _obj = (0, migratorStore_js_1.namingChange)(_obj, 'gridStyle', optionsMap_js_1.default.layoutParams.structure.responsiveMode);
    switch (_obj[optionsMap_js_1.default.layoutParams.structure.responsiveMode]) {
        case 0:
            _obj[optionsMap_js_1.default.layoutParams.structure.responsiveMode] = 'FIT_TO_SCREEN';
            break;
        case 1:
            _obj[optionsMap_js_1.default.layoutParams.structure.responsiveMode] = 'SET_ITEMS_PER_ROW';
            break;
        default:
            break;
    }
    return _obj;
}
function process_old_to_new_ScrollDirection(obj) {
    let _obj = obj;
    _obj = (0, migratorStore_js_1.namingChange)(_obj, 'scrollDirection', optionsMap_js_1.default.layoutParams.structure.scrollDirection);
    switch (_obj[optionsMap_js_1.default.layoutParams.structure.scrollDirection]) {
        case 0:
            _obj[optionsMap_js_1.default.layoutParams.structure.scrollDirection] = 'VERTICAL';
            break;
        case 1:
            _obj[optionsMap_js_1.default.layoutParams.structure.scrollDirection] = 'HORIZONTAL';
            break;
        default:
            break;
    }
    return _obj;
}
function process_old_to_new_layoutDirection(obj) {
    let _obj = obj;
    _obj = (0, migratorStore_js_1.namingChange)(_obj, 'isRTL', optionsMap_js_1.default.behaviourParams.gallery.layoutDirection);
    switch (_obj[optionsMap_js_1.default.behaviourParams.gallery.layoutDirection]) {
        case true:
            _obj[optionsMap_js_1.default.behaviourParams.gallery.layoutDirection] = 'RIGHT_TO_LEFT';
            break;
        case false:
            _obj[optionsMap_js_1.default.behaviourParams.gallery.layoutDirection] = 'LEFT_TO_RIGHT';
            break;
        default:
            break;
    }
    return _obj;
}
function process_old_to_new_slideshowInfoSize(obj) {
    let _obj = obj;
    const isSlideshow = (0, layoutParams_structure_galleryLayout_js_1.isLayout)('SLIDESHOW')({
        galleryLayout: obj['galleryLayout'] >= -3 ? obj['galleryLayout'] : obj[optionsMap_js_1.default.layoutParams.structure.galleryLayout],
    });
    if (isSlideshow) {
        _obj = (0, migratorStore_js_1.namingChange)(_obj, 'slideshowInfoSize', optionsMap_js_1.default.layoutParams.info.height);
    }
    else {
        _obj = (0, migratorStore_js_1.namingChange)(_obj, 'textBoxHeight', optionsMap_js_1.default.layoutParams.info.height);
    }
    delete _obj['slideshowInfoSize'];
    delete _obj['textBoxHeight'];
    return _obj;
}
function process_old_to_new_textBoxSizeMode(obj) {
    let _obj = obj;
    _obj = (0, migratorStore_js_1.namingChange)(_obj, 'calculateTextBoxWidthMode', optionsMap_js_1.default.layoutParams.info.sizeUnits);
    switch (_obj[optionsMap_js_1.default.layoutParams.info.sizeUnits]) {
        case 'PERCENT':
            _obj[optionsMap_js_1.default.layoutParams.info.sizeUnits] = 'PERCENT';
            _obj = (0, migratorStore_js_1.namingChange)(_obj, 'textBoxWidthPercent', optionsMap_js_1.default.layoutParams.info.width);
            delete _obj['textBoxWidth'];
            break;
        case 'MANUAL':
            _obj[optionsMap_js_1.default.layoutParams.info.sizeUnits] = 'PIXEL';
            _obj = (0, migratorStore_js_1.namingChange)(_obj, 'textBoxWidth', optionsMap_js_1.default.layoutParams.info.width);
            delete _obj['textBoxWidthPercent'];
            break;
        default:
            break;
    }
    return _obj;
}
function process_old_to_new_LayoutOrientation(obj) {
    let _obj = obj;
    _obj = (0, migratorStore_js_1.namingChange)(_obj, 'isVertical', optionsMap_js_1.default.layoutParams.structure.layoutOrientation);
    switch (_obj[optionsMap_js_1.default.layoutParams.structure.layoutOrientation]) {
        case true:
            _obj[optionsMap_js_1.default.layoutParams.structure.layoutOrientation] = 'VERTICAL';
            break;
        case false:
            _obj[optionsMap_js_1.default.layoutParams.structure.layoutOrientation] = 'HORIZONTAL';
            break;
        default:
            break;
    }
    return _obj;
}
function process_old_to_new_groupsOrder(obj) {
    let _obj = obj;
    _obj = (0, migratorStore_js_1.namingChange)(_obj, 'placeGroupsLtr', optionsMap_js_1.default.layoutParams.structure.groupsOrder);
    switch (_obj[optionsMap_js_1.default.layoutParams.structure.groupsOrder]) {
        case true:
            _obj[optionsMap_js_1.default.layoutParams.structure.groupsOrder] = 'LEFT_TO_RIGHT';
            break;
        case false:
            _obj[optionsMap_js_1.default.layoutParams.structure.groupsOrder] = 'BY_HEIGHT';
            break;
        default:
            break;
    }
    // 'RIGHT_TO_LEFT' doesnt exist yet.
    return _obj;
}
function process_old_to_new_galleryTextAlign(obj) {
    var _a;
    let _obj = obj;
    _obj = (0, migratorStore_js_1.namingChange)(_obj, 'galleryTextAlign', optionsMap_js_1.default.behaviourParams.gallery.horizontal.slideshowInfo.buttonsAlignment);
    _obj[optionsMap_js_1.default.behaviourParams.gallery.horizontal.slideshowInfo.buttonsAlignment] =
        (_a = _obj[optionsMap_js_1.default.behaviourParams.gallery.horizontal.slideshowInfo.buttonsAlignment]) === null || _a === void 0 ? void 0 : _a.toUpperCase();
    return _obj;
}
function process_old_to_new_LoadMoreAmount(obj) {
    var _a;
    let _obj = obj;
    _obj = (0, migratorStore_js_1.namingChange)(_obj, 'loadMoreAmount', optionsMap_js_1.default.behaviourParams.gallery.vertical.loadMore.amount);
    _obj[optionsMap_js_1.default.behaviourParams.gallery.vertical.loadMore.amount] =
        (_a = _obj[optionsMap_js_1.default.behaviourParams.gallery.vertical.loadMore.amount]) === null || _a === void 0 ? void 0 : _a.toUpperCase();
    return _obj;
}
function process_old_to_new_CroppedAlignment(obj) {
    let _obj = obj;
    _obj = (0, migratorStore_js_1.namingChange)(_obj, 'cubeFitPosition', optionsMap_js_1.default.layoutParams.crop.alignment);
    switch (_obj[optionsMap_js_1.default.layoutParams.crop.alignment]) {
        case 'MIDDLE':
            _obj[optionsMap_js_1.default.layoutParams.crop.alignment] = 'CENTER';
            break;
        default:
            break;
    }
    return _obj;
}
function process_old_to_new_OverlayHoveringBehaviour(obj) {
    let _obj = obj;
    _obj = (0, migratorStore_js_1.namingChange)(_obj, 'hoveringBehaviour', optionsMap_js_1.default.behaviourParams.item.overlay.hoveringBehaviour);
    switch (_obj[optionsMap_js_1.default.behaviourParams.item.overlay.hoveringBehaviour]) {
        case 'NO_CHANGE':
            _obj[optionsMap_js_1.default.behaviourParams.item.overlay.hoveringBehaviour] = 'ALWAYS_SHOW';
            break;
        case 'NEVER_SHOW':
            _obj[optionsMap_js_1.default.behaviourParams.item.overlay.hoveringBehaviour] = 'NEVER_SHOW';
            break;
        default:
            break;
    }
    return _obj;
}
function process_old_to_new_InfoPlacement(obj) {
    let _obj = obj;
    _obj = (0, migratorStore_js_1.namingChange)(_obj, 'titlePlacement', optionsMap_js_1.default.layoutParams.info.placement);
    switch (_obj[optionsMap_js_1.default.layoutParams.info.placement]) {
        case 'SHOW_ON_HOVER':
            _obj[optionsMap_js_1.default.layoutParams.info.placement] = 'OVERLAY';
            break;
        case 'SHOW_BELOW':
            _obj[optionsMap_js_1.default.layoutParams.info.placement] = 'BELOW';
            break;
        case 'SHOW_ABOVE':
            _obj[optionsMap_js_1.default.layoutParams.info.placement] = 'ABOVE';
            break;
        case 'SHOW_ON_THE_RIGHT':
            _obj[optionsMap_js_1.default.layoutParams.info.placement] = 'RIGHT';
            break;
        case 'SHOW_ON_THE_LEFT':
            _obj[optionsMap_js_1.default.layoutParams.info.placement] = 'LEFT';
            break;
        case 'ALTERNATE_HORIZONTAL':
            _obj[optionsMap_js_1.default.layoutParams.info.placement] = 'ALTERNATE_HORIZONTALLY';
            break;
        case 'ALTERNATE_VERTICAL':
            _obj[optionsMap_js_1.default.layoutParams.info.placement] = 'ALTERNATE_VERTICALLY';
            break;
        default:
            break;
    }
    return _obj;
}
function process_old_to_new_ClickAction(obj) {
    var _a;
    let _obj = obj;
    _obj = (0, migratorStore_js_1.namingChange)(_obj, 'itemClick', optionsMap_js_1.default.behaviourParams.item.clickAction);
    _obj[optionsMap_js_1.default.behaviourParams.item.clickAction] = (_a = _obj[optionsMap_js_1.default.behaviourParams.item.clickAction]) === null || _a === void 0 ? void 0 : _a.toUpperCase();
    switch (_obj[optionsMap_js_1.default.behaviourParams.item.clickAction]) {
        case 'FULLSCREEN':
        case 'EXPAND':
            _obj[optionsMap_js_1.default.behaviourParams.item.clickAction] = 'ACTION';
            break;
        default:
            break;
    }
    return _obj;
}
function process_old_to_new_AutoSlideBehaviour(obj) {
    if (typeof obj[optionsMap_js_1.default.behaviourParams.gallery.horizontal.autoSlide.behaviour] !== 'undefined') {
        return obj;
    }
    let _obj = obj;
    let isAutoSlide = _obj['isAutoSlideshow'];
    let autoSlideshowType = _obj['autoSlideshowType'];
    let finalVal;
    if (typeof isAutoSlide === 'undefined') {
        finalVal = undefined;
    }
    else {
        if (!isAutoSlide) {
            finalVal = 'OFF';
        }
        else {
            if (autoSlideshowType === 'interval') {
                finalVal = 'INTERVAL';
            }
            else {
                finalVal = 'CONTINUOUS';
            }
        }
    }
    _obj[optionsMap_js_1.default.behaviourParams.gallery.horizontal.autoSlide.behaviour] = finalVal;
    delete _obj['isAutoSlideshow'];
    delete _obj['autoSlideshowType'];
    return _obj;
}
function process_old_to_new_CropRatio(obj) {
    if (typeof obj[optionsMap_js_1.default.layoutParams.crop.ratios] !== 'undefined') {
        return obj;
    }
    let _obj = obj;
    let repeatingVal = _obj['rotatingCropRatios'];
    let val = _obj['cubeRatio'] || _obj['layoutParams_cropRatio'];
    let finalVal;
    if (typeof repeatingVal === 'string' && repeatingVal !== '') {
        finalVal = repeatingVal;
    }
    else {
        finalVal = val;
    }
    const getVal = (ele) => {
        if (Number(ele) >= 0) {
            return Number(ele);
        }
        else {
            return ele;
        }
    };
    _obj[optionsMap_js_1.default.layoutParams.crop.ratios] = finalVal && String(finalVal).split(',').map(getVal);
    delete _obj['cubeRatio'];
    delete _obj['layoutParams_cropRatio'];
    delete _obj['rotatingCropRatios'];
    return _obj;
}
function process_old_to_new_AllowedGroupTypes(obj) {
    var _a;
    if (typeof obj[optionsMap_js_1.default.layoutParams.groups.allowedGroupTypes] !== 'undefined') {
        return obj;
    }
    let _obj = obj;
    _obj[optionsMap_js_1.default.layoutParams.groups.allowedGroupTypes] = ((_a = _obj['groupTypes']) === null || _a === void 0 ? void 0 : _a.split)
        ? _obj['groupTypes'].split(',')
        : _obj['groupTypes']
            ? _obj['groupTypes']
            : undefined;
    delete _obj['groupTypes'];
    return _obj;
}
function process_old_to_new_repeatingGroupTypes(obj) {
    if (typeof obj[optionsMap_js_1.default.layoutParams.groups.repeatingGroupTypes] !== 'undefined') {
        return obj;
    }
    let _obj = obj;
    let repeatingVal = _obj['rotatingGroupTypes'] || _obj['layoutParams_repeatingGroupTypes'];
    let finalVal;
    if (typeof repeatingVal === 'string' && repeatingVal !== '') {
        finalVal = repeatingVal.split(',');
    }
    else if (typeof repeatingVal === 'string' && repeatingVal === '') {
        finalVal = [];
    }
    else {
        finalVal = undefined;
    }
    _obj[optionsMap_js_1.default.layoutParams.groups.repeatingGroupTypes] = finalVal;
    delete _obj['layoutParams_repeatingGroupTypes'];
    delete _obj['rotatingGroupTypes'];
    return _obj;
}
function process_old_to_new_NumberOfColumns(obj) {
    if (typeof obj[optionsMap_js_1.default.layoutParams.structure.numberOfColumns] !== 'undefined') {
        return obj;
    }
    let _obj = obj;
    const numberOfImagesPerRow = obj['numberOfImagesPerRow'];
    const finalVal = numberOfImagesPerRow >= 0 ? numberOfImagesPerRow : undefined;
    _obj[optionsMap_js_1.default.layoutParams.structure.numberOfColumns] = finalVal;
    delete _obj['numberOfImagesPerRow'];
    return _obj;
}
