// @wix/headless-faq/cjs/dist/react/core/Faq.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Root = Root;
exports.Name = Name;
exports.Item = Item;
exports.Answer = Answer;
const jsx_runtime_1 = require("react/jsx-runtime");
const services_manager_react_1 = require("@wix/services-manager-react");
const services_manager_1 = require("@wix/services-manager");
const index_js_1 = require("../../services/index.js");
/**
 * Root component that provides the FAQ service context to its children.
 * This component sets up the necessary services for rendering and managing a single FAQ entry's data.
 *
 * @component
 * @example
 * ```tsx
 * import { Faq } from '@wix/faq/components';
 *
 * function FaqPage() {
 *   return (
 *     <Faq.Root faqServiceConfig={{ faqs: [myFaq], categoryId: 'cat-1' }}>
 *       <Faq.Name>
 *         {({ question }) => (
 *           <h1 className="text-4xl font-bold">{question}</h1>
 *         )}
 *       </Faq.Name>
 *     </Faq.Root>
 *   );
 * }
 * ```
 */
function Root(props) {
    return ((0, jsx_runtime_1.jsx)(services_manager_react_1.WixServices, { servicesMap: (0, services_manager_1.createServicesMap)().addService(index_js_1.FaqServiceDefinition, index_js_1.FaqService, props.faqServiceConfig), children: props.children }));
}
/**
 * Headless component for FAQ question display
 *
 * @component
 * @example
 * ```tsx
 * import { Faq } from '@wix/faq/components';
 *
 * function FaqHeader() {
 *   return (
 *     <Faq.Name>
 *       {({ question }) => (
 *         <h1>{question}</h1>
 *       )}
 *     </Faq.Name>
 *   );
 * }
 * ```
 */
function Name(props) {
    const service = (0, services_manager_react_1.useService)(index_js_1.FaqServiceDefinition);
    const faqs = service.faqs.get();
    const faq = faqs[0]; // For single FAQ display, take the first FAQ
    const question = faq?.question || '';
    return props.children({
        question,
    });
}
/**
 * Headless component for FAQ item wrapper
 *
 * @component
 * @example
 * ```tsx
 * import { Faq } from '@wix/faq/components';
 *
 * function FaqItem() {
 *   return (
 *     <Faq.Item>
 *       {({ id }) => (
 *         <AccordionItem value={id}>
 *           {children}
 *         </AccordionItem>
 *       )}
 *     </Faq.Item>
 *   );
 * }
 * ```
 */
function Item(props) {
    const service = (0, services_manager_react_1.useService)(index_js_1.FaqServiceDefinition);
    const faqs = service.faqs.get();
    const faq = faqs[0]; // For single FAQ display, take the first FAQ
    const id = faq?._id || `faq-${Math.random()}`;
    return props.children({
        id,
    });
}
/**
 * Headless component for FAQ answer display
 *
 * @component
 * @example
 * ```tsx
 * import { Faq } from '@wix/faq/components';
 *
 * function FaqContent() {
 *   return (
 *     <Faq.Answer>
 *       {({ answer }) => (
 *         <p>{answer}</p>
 *       )}
 *     </Faq.Answer>
 *   );
 * }
 * ```
 */
function Answer(props) {
    const service = (0, services_manager_react_1.useService)(index_js_1.FaqServiceDefinition);
    const faqs = service.faqs.get();
    const faq = faqs[0]; // For single FAQ display, take the first FAQ
    const answer = faq?.plainText || faq?.draftjs || '';
    return props.children({
        answer,
    });
}
