// @wix/headless-events/cjs/dist/services/event-service.js
import { defineService, implementService } from '@wix/services-definitions';
import { SignalsServiceDefinition, } from '@wix/services-definitions/core-services/signals';
import { wixEventsV2 } from '@wix/events';
export const EventServiceDefinition = defineService('event');
export const EventService = implementService.withConfig()(EventServiceDefinition, ({ getService, config }) => {
    const signalsService = getService(SignalsServiceDefinition);
    const event = signalsService.signal(config.event);
    return {
        event,
    };
});
export async function loadEventServiceConfig({ slug, }) {
    try {
        const { event } = await wixEventsV2.getEventBySlug(slug, {
            fields: [
                wixEventsV2.RequestedFields.DETAILS,
                wixEventsV2.RequestedFields.REGISTRATION,
                wixEventsV2.RequestedFields.TEXTS,
                wixEventsV2.RequestedFields.FORM,
            ],
        });
        if (event) {
            return {
                type: 'success',
                config: {
                    event,
                },
            };
        }
    }
    catch (error) {
        console.error(`Failed to load event with "${slug}" slug`, error);
    }
    return {
        type: 'notFound',
    };
}
