// @wix/headless-events/cjs/dist/services/occurrence-list-service.js
import { defineService, implementService } from '@wix/services-definitions';
import { SignalsServiceDefinition, } from '@wix/services-definitions/core-services/signals';
import { wixEventsV2 } from '@wix/events';
import { getErrorMessage } from '../utils/errors.js';
import { queryEvents } from './event-list-service.js';
export const OccurrenceListServiceDefinition = defineService('occurrenceList');
export const OccurrenceListService = implementService.withConfig()(OccurrenceListServiceDefinition, ({ getService, config }) => {
    const signalsService = getService(SignalsServiceDefinition);
    const occurrences = signalsService.signal(config.occurrences);
    const isLoadingMore = signalsService.signal(false);
    const error = signalsService.signal(null);
    const pageSize = signalsService.signal(config.pageSize);
    const currentPage = signalsService.signal(config.currentPage);
    const totalPages = signalsService.signal(config.totalPages);
    const hasMoreOccurrences = signalsService.computed(() => currentPage.get() + 1 < totalPages.get());
    const loadMoreOccurrences = async () => {
        if (!config.recurringCategoryId) {
            return;
        }
        isLoadingMore.set(true);
        error.set(null);
        try {
            const offset = pageSize.get() * (currentPage.get() + 1);
            const response = await queryOccurrences({
                offset,
                recurringCategoryId: config.recurringCategoryId,
            });
            occurrences.set([...occurrences.get(), ...response.items]);
            pageSize.set(response.pageSize);
            currentPage.set(response.currentPage);
            totalPages.set(response.totalPages);
        }
        catch (err) {
            error.set(getErrorMessage(err));
        }
        finally {
            isLoadingMore.set(false);
        }
    };
    return {
        occurrences,
        isLoadingMore,
        error,
        pageSize,
        currentPage,
        totalPages,
        hasMoreOccurrences,
        loadMoreOccurrences,
    };
});
export async function loadOccurrenceListServiceConfig({ recurringCategoryId, }) {
    if (recurringCategoryId) {
        const response = await queryOccurrences({ recurringCategoryId });
        return {
            recurringCategoryId,
            occurrences: response.items,
            pageSize: response.pageSize,
            currentPage: response.currentPage,
            totalPages: response.totalPages,
        };
    }
    return {
        recurringCategoryId,
        occurrences: [],
        pageSize: 0,
        currentPage: 0,
        totalPages: 0,
    };
}
async function queryOccurrences({ recurringCategoryId, offset = 0, }) {
    return queryEvents({
        offset,
        categoryId: recurringCategoryId,
        status: [wixEventsV2.Status.UPCOMING, wixEventsV2.Status.STARTED],
    });
}
