// @wix/headless-events/cjs/dist/services/order-service.js
import { defineService, implementService } from '@wix/services-definitions';
import { SignalsServiceDefinition, } from '@wix/services-definitions/core-services/signals';
import { orders } from '@wix/events';
import { poll } from '../utils/poll.js';
export const OrderServiceDefinition = defineService('order');
export const OrderService = implementService.withConfig()(OrderServiceDefinition, ({ getService, config }) => {
    const signalsService = getService(SignalsServiceDefinition);
    const order = signalsService.signal(config.order);
    const isPolling = signalsService.signal(false);
    const { eventId, orderNumber } = config;
    const pollOrder = async () => {
        if (isOrderReady(order.get())) {
            return;
        }
        isPolling.set(true);
        try {
            await poll({
                callback: async () => {
                    const response = await loadOrder({ eventId, orderNumber });
                    order.set(response);
                    return isOrderReady(response);
                },
                intervalMs: 2000,
                totalMs: 15000,
            });
        }
        catch (error) {
            console.log(`Error polling order: ${error}`);
        }
        finally {
            isPolling.set(false);
        }
    };
    if (config.order) {
        pollOrder();
    }
    return {
        order,
        isPolling,
    };
});
export async function loadOrderServiceConfig({ eventId, orderNumber, }) {
    const order = await loadOrder({ eventId, orderNumber });
    return { order, eventId, orderNumber };
}
export function isOrderReady(order) {
    return order.ticketsQuantity === order.tickets?.length;
}
function loadOrder({ eventId, orderNumber }) {
    return orders.getOrder({ eventId, orderNumber }, {
        fieldset: [
            orders.OrderFieldset.TICKETS,
            orders.OrderFieldset.DETAILS,
            orders.OrderFieldset.INVOICE,
        ],
    });
}
