// @wix/ricos/dist/cjs/wix-rich-content-common/devTools/tabs/ricos-content-editor/code-mirror-theme.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var code_mirror_theme_exports = {};
__export(code_mirror_theme_exports, {
  config: () => config,
  defaultSettingsSolarizedDark: () => defaultSettingsSolarizedDark,
  solarizedDarkInit: () => solarizedDarkInit,
  wixDocsCodeTheme: () => wixDocsCodeTheme
});
module.exports = __toCommonJS(code_mirror_theme_exports);
var import_highlight = require("@lezer/highlight");
var import_codemirror_themes = require("@uiw/codemirror-themes");
const config = {
  background: "rgb(0,6,36)",
  foreground: "rgb(134,138,165)",
  selection: "#004454AA",
  selectionMatch: "#005A6FAA",
  cursor: "#D30102",
  dropdownBackground: "#00212B",
  dropdownBorder: "#2AA19899",
  activeLine: "#00cafe11",
  matchingBracket: "#073642",
  keyword: "#859900",
  storage: "#93A1A1",
  variable: "#268BD2",
  parameter: "#268BD2",
  function: "#268BD2",
  string: "rgb(140, 180, 255)",
  constant: "#CB4B16",
  type: "#859900",
  class: "#268BD2",
  number: "#D33682",
  comment: "#586E75",
  heading: "#268BD2",
  invalid: "#DC322F",
  regexp: "#DC322F",
  tag: "#268BD2"
};
const c = config;
const defaultSettingsSolarizedDark = {
  background: c.background,
  foreground: c.foreground,
  caret: c.cursor,
  selection: c.selection,
  selectionMatch: c.selection,
  gutterBackground: c.background,
  gutterForeground: c.foreground,
  gutterBorder: "transparent",
  lineHighlight: c.activeLine
};
const solarizedDarkInit = (options) => {
  const { theme = "dark", settings = {}, styles = [] } = options || {};
  return (0, import_codemirror_themes.createTheme)({
    theme,
    settings: {
      ...defaultSettingsSolarizedDark,
      ...settings
    },
    styles: [
      { tag: import_highlight.tags.keyword, color: c.keyword },
      { tag: [import_highlight.tags.name, import_highlight.tags.deleted, import_highlight.tags.character, import_highlight.tags.macroName], color: c.variable },
      { tag: [import_highlight.tags.propertyName], color: c.function },
      {
        tag: [import_highlight.tags.processingInstruction, import_highlight.tags.string, import_highlight.tags.inserted, import_highlight.tags.special(import_highlight.tags.string)],
        color: c.string
      },
      { tag: [import_highlight.tags.function(import_highlight.tags.variableName), import_highlight.tags.labelName], color: c.function },
      { tag: [import_highlight.tags.color, import_highlight.tags.constant(import_highlight.tags.name), import_highlight.tags.standard(import_highlight.tags.name)], color: c.constant },
      { tag: [import_highlight.tags.definition(import_highlight.tags.name), import_highlight.tags.separator], color: c.variable },
      { tag: [import_highlight.tags.className], color: c.class },
      {
        tag: [import_highlight.tags.number, import_highlight.tags.changed, import_highlight.tags.annotation, import_highlight.tags.modifier, import_highlight.tags.self, import_highlight.tags.namespace],
        color: c.number
      },
      { tag: [import_highlight.tags.typeName], color: c.type, fontStyle: c.type },
      { tag: [import_highlight.tags.operator, import_highlight.tags.operatorKeyword], color: c.keyword },
      { tag: [import_highlight.tags.url, import_highlight.tags.escape, import_highlight.tags.regexp, import_highlight.tags.link], color: c.regexp },
      { tag: [import_highlight.tags.meta, import_highlight.tags.comment], color: c.comment },
      { tag: import_highlight.tags.tagName, color: c.tag },
      { tag: import_highlight.tags.strong, fontWeight: "bold" },
      { tag: import_highlight.tags.emphasis, fontStyle: "italic" },
      { tag: import_highlight.tags.link, textDecoration: "underline" },
      { tag: import_highlight.tags.heading, fontWeight: "bold", color: c.heading },
      { tag: [import_highlight.tags.atom, import_highlight.tags.bool, import_highlight.tags.special(import_highlight.tags.variableName)], color: c.variable },
      { tag: import_highlight.tags.invalid, color: c.invalid },
      { tag: import_highlight.tags.strikethrough, textDecoration: "line-through" },
      { tag: import_highlight.tags.punctuation, color: "rgb(237, 162, 0)" },
      { tag: import_highlight.tags.propertyName, color: "rgb(54, 172, 170)" },
      ...styles
    ]
  });
};
const wixDocsCodeTheme = solarizedDarkInit();
