// @wix/auto_sdk_forms_form-spam-submission-reports/build/es/index.mjs
// src/forms-v4-form-spam-submission-report-form-spam-submission-reports.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/forms-v4-form-spam-submission-report-form-spam-submission-reports.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixFormsV4FormSpamSubmissionReportServiceUrl(opts) {
  const domainToMappings = {
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/form-spam-submission-report-service",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/form-spam-submission-report-service",
        destPath: ""
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/form-spam-submission-report-service",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/form-spam-submission-report-service",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/form-spam-submission-report-service",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/form-spam-submission-report-service",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/form-spam-submission-report-service",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/form-spam-submission-report-service",
        destPath: ""
      },
      {
        srcPath: "/form-spam-submission-report-service",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/form-spam-submission-report-service",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/form-spam-submission-report-service",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_forms_form-spam-submission-reports";
function checkForSpam(payload) {
  function __checkForSpam({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "submission.createdDate" },
          { path: "submission.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "POST",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.CheckForSpam",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/form-spam-submission-reports/check-for-spam",
        data: serializedData,
        host
      }),
      data: serializedData
    };
    return metadata;
  }
  return __checkForSpam;
}
function createFormSpamSubmissionReport(payload) {
  function __createFormSpamSubmissionReport({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "formSpamSubmissionReport.createdDate" },
          { path: "formSpamSubmissionReport.reportedDate" },
          { path: "formSpamSubmissionReport.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "POST",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.CreateFormSpamSubmissionReport",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/form-spam-submission-reports",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "formSpamSubmissionReport.createdDate" },
            { path: "formSpamSubmissionReport.reportedDate" },
            { path: "formSpamSubmissionReport.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createFormSpamSubmissionReport;
}
function getFormSpamSubmissionReport(payload) {
  function __getFormSpamSubmissionReport({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "GET",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.GetFormSpamSubmissionReport",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/form-spam-submission-reports/{formSpamSubmissionReportId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "formSpamSubmissionReport.createdDate" },
            { path: "formSpamSubmissionReport.reportedDate" },
            { path: "formSpamSubmissionReport.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getFormSpamSubmissionReport;
}
function deleteFormSpamSubmissionReport(payload) {
  function __deleteFormSpamSubmissionReport({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "DELETE",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.DeleteFormSpamSubmissionReport",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/form-spam-submission-reports/{formSpamSubmissionReportId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteFormSpamSubmissionReport;
}
function bulkDeleteFormSpamSubmissionReport(payload) {
  function __bulkDeleteFormSpamSubmissionReport({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "POST",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.BulkDeleteFormSpamSubmissionReport",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/bulk/form-spam-submission-reports/delete",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkDeleteFormSpamSubmissionReport;
}
function bulkDeleteFormSpamSubmissionReportByFilter(payload) {
  function __bulkDeleteFormSpamSubmissionReportByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "POST",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.BulkDeleteFormSpamSubmissionReportByFilter",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/bulk/form-spam-submission-reports/delete-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkDeleteFormSpamSubmissionReportByFilter;
}
function reportNotSpamSubmission(payload) {
  function __reportNotSpamSubmission({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "POST",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.ReportNotSpamSubmission",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/form-spam-submission-reports/{formSpamSubmissionReportId}/report-not-spam-submission",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "submission.createdDate" },
            { path: "submission.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __reportNotSpamSubmission;
}
function bulkReportNotSpamSubmission(payload) {
  function __bulkReportNotSpamSubmission({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "POST",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.BulkReportNotSpamSubmission",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/bulk/form-spam-submission-reports/report-not-spam-submission",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkReportNotSpamSubmission;
}
function reportSpamSubmission(payload) {
  function __reportSpamSubmission({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "POST",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.ReportSpamSubmission",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/submissions/{submissionId}/report-spam-submission",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "formSpamSubmissionReport.createdDate" },
            { path: "formSpamSubmissionReport.reportedDate" },
            { path: "formSpamSubmissionReport.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __reportSpamSubmission;
}
function bulkReportSpamSubmission(payload) {
  function __bulkReportSpamSubmission({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "POST",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.BulkReportSpamSubmission",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/bulk/submissions/report-spam-submission",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.reportedDate" },
            { path: "results.item.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkReportSpamSubmission;
}
function queryFormSpamSubmissionReportsByNamespace(payload) {
  function __queryFormSpamSubmissionReportsByNamespace({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "POST",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.QueryFormSpamSubmissionReportsByNamespace",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/form-spam-submission-reports/namespace/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "formSpamSubmissionReports.createdDate" },
            { path: "formSpamSubmissionReports.reportedDate" },
            { path: "formSpamSubmissionReports.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryFormSpamSubmissionReportsByNamespace;
}
function countFormSpamSubmissionReports(payload) {
  function __countFormSpamSubmissionReports({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "POST",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.CountFormSpamSubmissionReports",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/form-spam-submission-reports/count",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __countFormSpamSubmissionReports;
}

// src/forms-v4-form-spam-submission-report-form-spam-submission-reports.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var ReportReason = /* @__PURE__ */ ((ReportReason2) => {
  ReportReason2["UNKNOWN_REASON"] = "UNKNOWN_REASON";
  ReportReason2["EMAIL_QUOTA_REACHED"] = "EMAIL_QUOTA_REACHED";
  ReportReason2["IP_BLOCKLISTED"] = "IP_BLOCKLISTED";
  ReportReason2["EMAIL_BLOCKLISTED"] = "EMAIL_BLOCKLISTED";
  ReportReason2["AI_REPORTED"] = "AI_REPORTED";
  ReportReason2["MANUALLY_REPORTED"] = "MANUALLY_REPORTED";
  ReportReason2["MODERATION_SPI"] = "MODERATION_SPI";
  return ReportReason2;
})(ReportReason || {});
var SubmissionStatus = /* @__PURE__ */ ((SubmissionStatus2) => {
  SubmissionStatus2["UNKNOWN_SUBMISSION_STATUS"] = "UNKNOWN_SUBMISSION_STATUS";
  SubmissionStatus2["PENDING"] = "PENDING";
  SubmissionStatus2["CONFIRMED"] = "CONFIRMED";
  SubmissionStatus2["PAYMENT_WAITING"] = "PAYMENT_WAITING";
  SubmissionStatus2["PAYMENT_CANCELED"] = "PAYMENT_CANCELED";
  return SubmissionStatus2;
})(SubmissionStatus || {});
var ErrorType = /* @__PURE__ */ ((ErrorType2) => {
  ErrorType2["UNKNOWN_ERROR"] = "UNKNOWN_ERROR";
  ErrorType2["TYPE_ERROR"] = "TYPE_ERROR";
  ErrorType2["REQUIRED_VALUE_ERROR"] = "REQUIRED_VALUE_ERROR";
  ErrorType2["UNKNOWN_VALUE_ERROR"] = "UNKNOWN_VALUE_ERROR";
  ErrorType2["MAX_LENGTH_ERROR"] = "MAX_LENGTH_ERROR";
  ErrorType2["MIN_LENGTH_ERROR"] = "MIN_LENGTH_ERROR";
  ErrorType2["PATTERN_ERROR"] = "PATTERN_ERROR";
  ErrorType2["FORMAT_ERROR"] = "FORMAT_ERROR";
  ErrorType2["MAX_VALUE_ERROR"] = "MAX_VALUE_ERROR";
  ErrorType2["MIN_VALUE_ERROR"] = "MIN_VALUE_ERROR";
  ErrorType2["MULTIPLE_OF_VALUE_ERROR"] = "MULTIPLE_OF_VALUE_ERROR";
  ErrorType2["MIN_ITEMS_ERROR"] = "MIN_ITEMS_ERROR";
  ErrorType2["MAX_ITEMS_ERROR"] = "MAX_ITEMS_ERROR";
  ErrorType2["NOT_ALLOWED_VALUE_ERROR"] = "NOT_ALLOWED_VALUE_ERROR";
  ErrorType2["FIELDS_COMPATIBILITY_ERROR"] = "FIELDS_COMPATIBILITY_ERROR";
  ErrorType2["DISABLED_FORM_ERROR"] = "DISABLED_FORM_ERROR";
  ErrorType2["FORMS_COUNT_RESTRICTIONS_ERROR"] = "FORMS_COUNT_RESTRICTIONS_ERROR";
  ErrorType2["FIELDS_COUNT_RESTRICTIONS_ERROR"] = "FIELDS_COUNT_RESTRICTIONS_ERROR";
  ErrorType2["STEPS_COUNT_RESTRICTIONS_ERROR"] = "STEPS_COUNT_RESTRICTIONS_ERROR";
  ErrorType2["RULES_COUNT_RESTRICTIONS_ERROR"] = "RULES_COUNT_RESTRICTIONS_ERROR";
  ErrorType2["FILE_UPLOAD_RESTRICTIONS_ERROR"] = "FILE_UPLOAD_RESTRICTIONS_ERROR";
  ErrorType2["INVALID_PHONE_COUNTRY_CODE_ERROR"] = "INVALID_PHONE_COUNTRY_CODE_ERROR";
  ErrorType2["INVALID_STAFF_ID_ERROR"] = "INVALID_STAFF_ID_ERROR";
  ErrorType2["INVALID_LOCATION_ID_ERROR"] = "INVALID_LOCATION_ID_ERROR";
  ErrorType2["MISSING_SERVICE_OPTION_ERROR"] = "MISSING_SERVICE_OPTION_ERROR";
  ErrorType2["INVALID_SERVICE_OPTIONS_ERROR"] = "INVALID_SERVICE_OPTIONS_ERROR";
  ErrorType2["NO_AVAILABLE_SERVICE_OPTIONS_ERROR"] = "NO_AVAILABLE_SERVICE_OPTIONS_ERROR";
  return ErrorType2;
})(ErrorType || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function checkForSpam2(submission) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    submission
  });
  const reqOpts = checkForSpam(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { submission: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["submission"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function createFormSpamSubmissionReport2(formSpamSubmissionReport) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    formSpamSubmissionReport
  });
  const reqOpts = createFormSpamSubmissionReport(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.formSpamSubmissionReport;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { formSpamSubmissionReport: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["formSpamSubmissionReport"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getFormSpamSubmissionReport2(formSpamSubmissionReportId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    formSpamSubmissionReportId
  });
  const reqOpts = getFormSpamSubmissionReport(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.formSpamSubmissionReport;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { formSpamSubmissionReportId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["formSpamSubmissionReportId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteFormSpamSubmissionReport2(formSpamSubmissionReportId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    formSpamSubmissionReportId
  });
  const reqOpts = deleteFormSpamSubmissionReport(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { formSpamSubmissionReportId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["formSpamSubmissionReportId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkDeleteFormSpamSubmissionReport2(formId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    formId,
    formSpamSubmissionReportIds: options?.formSpamSubmissionReportIds
  });
  const reqOpts = bulkDeleteFormSpamSubmissionReport(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          formId: "$[0]",
          formSpamSubmissionReportIds: "$[1].formSpamSubmissionReportIds"
        },
        singleArgumentUnchanged: false
      },
      ["formId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkDeleteFormSpamSubmissionReportByFilter2(filter) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ filter });
  const reqOpts = bulkDeleteFormSpamSubmissionReportByFilter(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { filter: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["filter"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function reportNotSpamSubmission2(formSpamSubmissionReportId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    formSpamSubmissionReportId
  });
  const reqOpts = reportNotSpamSubmission(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { formSpamSubmissionReportId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["formSpamSubmissionReportId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkReportNotSpamSubmission2(formId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    formId,
    formSpamSubmissionReportIds: options?.formSpamSubmissionReportIds
  });
  const reqOpts = bulkReportNotSpamSubmission(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          formId: "$[0]",
          formSpamSubmissionReportIds: "$[1].formSpamSubmissionReportIds"
        },
        singleArgumentUnchanged: false
      },
      ["formId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function reportSpamSubmission2(submissionId, reportReason, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    submissionId,
    reportReason,
    reportReasonDescription: options?.reportReasonDescription
  });
  const reqOpts = reportSpamSubmission(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          submissionId: "$[0]",
          reportReason: "$[1]",
          reportReasonDescription: "$[2].reportReasonDescription"
        },
        singleArgumentUnchanged: false
      },
      ["submissionId", "reportReason", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkReportSpamSubmission2(formId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    formId,
    submissionIds: options?.submissionIds,
    reportReason: options?.reportReason,
    returnEntity: options?.returnEntity,
    reportReasonDescription: options?.reportReasonDescription
  });
  const reqOpts = bulkReportSpamSubmission(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          formId: "$[0]",
          submissionIds: "$[1].submissionIds",
          reportReason: "$[1].reportReason",
          returnEntity: "$[1].returnEntity",
          reportReasonDescription: "$[1].reportReasonDescription"
        },
        singleArgumentUnchanged: false
      },
      ["formId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryFormSpamSubmissionReportsByNamespace2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryFormSpamSubmissionReportsByNamespace(
        payload
      );
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [])
      );
      return {
        items: transformedData?.formSpamSubmissionReports,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function countFormSpamSubmissionReports2(formIds, namespace) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    formIds,
    namespace
  });
  const reqOpts = countFormSpamSubmissionReports(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { formIds: "$[0]", namespace: "$[1]" },
        singleArgumentUnchanged: false
      },
      ["formIds", "namespace"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/forms-v4-form-spam-submission-report-form-spam-submission-reports.public.ts
function checkForSpam3(httpClient) {
  return (submission) => checkForSpam2(
    submission,
    // @ts-ignore
    { httpClient }
  );
}
function createFormSpamSubmissionReport3(httpClient) {
  return (formSpamSubmissionReport) => createFormSpamSubmissionReport2(
    formSpamSubmissionReport,
    // @ts-ignore
    { httpClient }
  );
}
function getFormSpamSubmissionReport3(httpClient) {
  return (formSpamSubmissionReportId) => getFormSpamSubmissionReport2(
    formSpamSubmissionReportId,
    // @ts-ignore
    { httpClient }
  );
}
function deleteFormSpamSubmissionReport3(httpClient) {
  return (formSpamSubmissionReportId) => deleteFormSpamSubmissionReport2(
    formSpamSubmissionReportId,
    // @ts-ignore
    { httpClient }
  );
}
function bulkDeleteFormSpamSubmissionReport3(httpClient) {
  return (formId, options) => bulkDeleteFormSpamSubmissionReport2(
    formId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkDeleteFormSpamSubmissionReportByFilter3(httpClient) {
  return (filter) => bulkDeleteFormSpamSubmissionReportByFilter2(
    filter,
    // @ts-ignore
    { httpClient }
  );
}
function reportNotSpamSubmission3(httpClient) {
  return (formSpamSubmissionReportId) => reportNotSpamSubmission2(
    formSpamSubmissionReportId,
    // @ts-ignore
    { httpClient }
  );
}
function bulkReportNotSpamSubmission3(httpClient) {
  return (formId, options) => bulkReportNotSpamSubmission2(
    formId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function reportSpamSubmission3(httpClient) {
  return (submissionId, reportReason, options) => reportSpamSubmission2(
    submissionId,
    reportReason,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkReportSpamSubmission3(httpClient) {
  return (formId, options) => bulkReportSpamSubmission2(
    formId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function queryFormSpamSubmissionReportsByNamespace3(httpClient) {
  return () => queryFormSpamSubmissionReportsByNamespace2(
    // @ts-ignore
    { httpClient }
  );
}
function countFormSpamSubmissionReports3(httpClient) {
  return (formIds, namespace) => countFormSpamSubmissionReports2(
    formIds,
    namespace,
    // @ts-ignore
    { httpClient }
  );
}

// src/forms-v4-form-spam-submission-report-form-spam-submission-reports.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
var checkForSpam4 = /* @__PURE__ */ createRESTModule(checkForSpam3);
var createFormSpamSubmissionReport4 = /* @__PURE__ */ createRESTModule(createFormSpamSubmissionReport3);
var getFormSpamSubmissionReport4 = /* @__PURE__ */ createRESTModule(getFormSpamSubmissionReport3);
var deleteFormSpamSubmissionReport4 = /* @__PURE__ */ createRESTModule(deleteFormSpamSubmissionReport3);
var bulkDeleteFormSpamSubmissionReport4 = /* @__PURE__ */ createRESTModule(bulkDeleteFormSpamSubmissionReport3);
var bulkDeleteFormSpamSubmissionReportByFilter4 = /* @__PURE__ */ createRESTModule(
  bulkDeleteFormSpamSubmissionReportByFilter3
);
var reportNotSpamSubmission4 = /* @__PURE__ */ createRESTModule(reportNotSpamSubmission3);
var bulkReportNotSpamSubmission4 = /* @__PURE__ */ createRESTModule(bulkReportNotSpamSubmission3);
var reportSpamSubmission4 = /* @__PURE__ */ createRESTModule(reportSpamSubmission3);
var bulkReportSpamSubmission4 = /* @__PURE__ */ createRESTModule(bulkReportSpamSubmission3);
var queryFormSpamSubmissionReportsByNamespace4 = /* @__PURE__ */ createRESTModule(
  queryFormSpamSubmissionReportsByNamespace3
);
var countFormSpamSubmissionReports4 = /* @__PURE__ */ createRESTModule(countFormSpamSubmissionReports3);
export {
  ErrorType,
  ReportReason,
  SortOrder,
  SubmissionStatus,
  WebhookIdentityType,
  bulkDeleteFormSpamSubmissionReport4 as bulkDeleteFormSpamSubmissionReport,
  bulkDeleteFormSpamSubmissionReportByFilter4 as bulkDeleteFormSpamSubmissionReportByFilter,
  bulkReportNotSpamSubmission4 as bulkReportNotSpamSubmission,
  bulkReportSpamSubmission4 as bulkReportSpamSubmission,
  checkForSpam4 as checkForSpam,
  countFormSpamSubmissionReports4 as countFormSpamSubmissionReports,
  createFormSpamSubmissionReport4 as createFormSpamSubmissionReport,
  deleteFormSpamSubmissionReport4 as deleteFormSpamSubmissionReport,
  getFormSpamSubmissionReport4 as getFormSpamSubmissionReport,
  queryFormSpamSubmissionReportsByNamespace4 as queryFormSpamSubmissionReportsByNamespace,
  reportNotSpamSubmission4 as reportNotSpamSubmission,
  reportSpamSubmission4 as reportSpamSubmission
};
