// @wix/auto_sdk_forms_form-spam-submission-reports/build/es/meta.mjs
// src/forms-v4-form-spam-submission-report-form-spam-submission-reports.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixFormsV4FormSpamSubmissionReportServiceUrl(opts) {
  const domainToMappings = {
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/form-spam-submission-report-service",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/form-spam-submission-report-service",
        destPath: ""
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/form-spam-submission-report-service",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/form-spam-submission-report-service",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/form-spam-submission-report-service",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/form-spam-submission-report-service",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/form-spam-submission-report-service",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/form-spam-submission-report-service",
        destPath: ""
      },
      {
        srcPath: "/form-spam-submission-report-service",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/form-spam-submission-report-service",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/form-spam-submission-report-service",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_forms_form-spam-submission-reports";
function checkForSpam(payload) {
  function __checkForSpam({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "submission.createdDate" },
          { path: "submission.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "POST",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.CheckForSpam",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/form-spam-submission-reports/check-for-spam",
        data: serializedData,
        host
      }),
      data: serializedData
    };
    return metadata;
  }
  return __checkForSpam;
}
function createFormSpamSubmissionReport(payload) {
  function __createFormSpamSubmissionReport({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "formSpamSubmissionReport.createdDate" },
          { path: "formSpamSubmissionReport.reportedDate" },
          { path: "formSpamSubmissionReport.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "POST",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.CreateFormSpamSubmissionReport",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/form-spam-submission-reports",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "formSpamSubmissionReport.createdDate" },
            { path: "formSpamSubmissionReport.reportedDate" },
            { path: "formSpamSubmissionReport.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createFormSpamSubmissionReport;
}
function getFormSpamSubmissionReport(payload) {
  function __getFormSpamSubmissionReport({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "GET",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.GetFormSpamSubmissionReport",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/form-spam-submission-reports/{formSpamSubmissionReportId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "formSpamSubmissionReport.createdDate" },
            { path: "formSpamSubmissionReport.reportedDate" },
            { path: "formSpamSubmissionReport.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getFormSpamSubmissionReport;
}
function deleteFormSpamSubmissionReport(payload) {
  function __deleteFormSpamSubmissionReport({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "DELETE",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.DeleteFormSpamSubmissionReport",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/form-spam-submission-reports/{formSpamSubmissionReportId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteFormSpamSubmissionReport;
}
function bulkDeleteFormSpamSubmissionReport(payload) {
  function __bulkDeleteFormSpamSubmissionReport({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "POST",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.BulkDeleteFormSpamSubmissionReport",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/bulk/form-spam-submission-reports/delete",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkDeleteFormSpamSubmissionReport;
}
function bulkDeleteFormSpamSubmissionReportByFilter(payload) {
  function __bulkDeleteFormSpamSubmissionReportByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "POST",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.BulkDeleteFormSpamSubmissionReportByFilter",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/bulk/form-spam-submission-reports/delete-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkDeleteFormSpamSubmissionReportByFilter;
}
function reportNotSpamSubmission(payload) {
  function __reportNotSpamSubmission({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "POST",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.ReportNotSpamSubmission",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/form-spam-submission-reports/{formSpamSubmissionReportId}/report-not-spam-submission",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "submission.createdDate" },
            { path: "submission.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __reportNotSpamSubmission;
}
function bulkReportNotSpamSubmission(payload) {
  function __bulkReportNotSpamSubmission({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "POST",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.BulkReportNotSpamSubmission",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/bulk/form-spam-submission-reports/report-not-spam-submission",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkReportNotSpamSubmission;
}
function reportSpamSubmission(payload) {
  function __reportSpamSubmission({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "POST",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.ReportSpamSubmission",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/submissions/{submissionId}/report-spam-submission",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "formSpamSubmissionReport.createdDate" },
            { path: "formSpamSubmissionReport.reportedDate" },
            { path: "formSpamSubmissionReport.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __reportSpamSubmission;
}
function bulkReportSpamSubmission(payload) {
  function __bulkReportSpamSubmission({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "POST",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.BulkReportSpamSubmission",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/bulk/submissions/report-spam-submission",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.reportedDate" },
            { path: "results.item.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkReportSpamSubmission;
}
function queryFormSpamSubmissionReportsByNamespace(payload) {
  function __queryFormSpamSubmissionReportsByNamespace({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "POST",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.QueryFormSpamSubmissionReportsByNamespace",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/form-spam-submission-reports/namespace/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "formSpamSubmissionReports.createdDate" },
            { path: "formSpamSubmissionReports.reportedDate" },
            { path: "formSpamSubmissionReports.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryFormSpamSubmissionReportsByNamespace;
}
function countFormSpamSubmissionReports(payload) {
  function __countFormSpamSubmissionReports({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.v4.form_spam_submission_report",
      method: "POST",
      methodFqn: "wix.forms.v4.FormSpamSubmissionReportService.CountFormSpamSubmissionReports",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsV4FormSpamSubmissionReportServiceUrl({
        protoPath: "/v4/form-spam-submission-reports/count",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __countFormSpamSubmissionReports;
}

// src/forms-v4-form-spam-submission-report-form-spam-submission-reports.types.ts
var ReportReason = /* @__PURE__ */ ((ReportReason2) => {
  ReportReason2["UNKNOWN_REASON"] = "UNKNOWN_REASON";
  ReportReason2["EMAIL_QUOTA_REACHED"] = "EMAIL_QUOTA_REACHED";
  ReportReason2["IP_BLOCKLISTED"] = "IP_BLOCKLISTED";
  ReportReason2["EMAIL_BLOCKLISTED"] = "EMAIL_BLOCKLISTED";
  ReportReason2["AI_REPORTED"] = "AI_REPORTED";
  ReportReason2["MANUALLY_REPORTED"] = "MANUALLY_REPORTED";
  ReportReason2["MODERATION_SPI"] = "MODERATION_SPI";
  return ReportReason2;
})(ReportReason || {});
var SubmissionStatus = /* @__PURE__ */ ((SubmissionStatus2) => {
  SubmissionStatus2["UNKNOWN_SUBMISSION_STATUS"] = "UNKNOWN_SUBMISSION_STATUS";
  SubmissionStatus2["PENDING"] = "PENDING";
  SubmissionStatus2["CONFIRMED"] = "CONFIRMED";
  SubmissionStatus2["PAYMENT_WAITING"] = "PAYMENT_WAITING";
  SubmissionStatus2["PAYMENT_CANCELED"] = "PAYMENT_CANCELED";
  return SubmissionStatus2;
})(SubmissionStatus || {});
var ErrorType = /* @__PURE__ */ ((ErrorType2) => {
  ErrorType2["UNKNOWN_ERROR"] = "UNKNOWN_ERROR";
  ErrorType2["TYPE_ERROR"] = "TYPE_ERROR";
  ErrorType2["REQUIRED_VALUE_ERROR"] = "REQUIRED_VALUE_ERROR";
  ErrorType2["UNKNOWN_VALUE_ERROR"] = "UNKNOWN_VALUE_ERROR";
  ErrorType2["MAX_LENGTH_ERROR"] = "MAX_LENGTH_ERROR";
  ErrorType2["MIN_LENGTH_ERROR"] = "MIN_LENGTH_ERROR";
  ErrorType2["PATTERN_ERROR"] = "PATTERN_ERROR";
  ErrorType2["FORMAT_ERROR"] = "FORMAT_ERROR";
  ErrorType2["MAX_VALUE_ERROR"] = "MAX_VALUE_ERROR";
  ErrorType2["MIN_VALUE_ERROR"] = "MIN_VALUE_ERROR";
  ErrorType2["MULTIPLE_OF_VALUE_ERROR"] = "MULTIPLE_OF_VALUE_ERROR";
  ErrorType2["MIN_ITEMS_ERROR"] = "MIN_ITEMS_ERROR";
  ErrorType2["MAX_ITEMS_ERROR"] = "MAX_ITEMS_ERROR";
  ErrorType2["NOT_ALLOWED_VALUE_ERROR"] = "NOT_ALLOWED_VALUE_ERROR";
  ErrorType2["FIELDS_COMPATIBILITY_ERROR"] = "FIELDS_COMPATIBILITY_ERROR";
  ErrorType2["DISABLED_FORM_ERROR"] = "DISABLED_FORM_ERROR";
  ErrorType2["FORMS_COUNT_RESTRICTIONS_ERROR"] = "FORMS_COUNT_RESTRICTIONS_ERROR";
  ErrorType2["FIELDS_COUNT_RESTRICTIONS_ERROR"] = "FIELDS_COUNT_RESTRICTIONS_ERROR";
  ErrorType2["STEPS_COUNT_RESTRICTIONS_ERROR"] = "STEPS_COUNT_RESTRICTIONS_ERROR";
  ErrorType2["RULES_COUNT_RESTRICTIONS_ERROR"] = "RULES_COUNT_RESTRICTIONS_ERROR";
  ErrorType2["FILE_UPLOAD_RESTRICTIONS_ERROR"] = "FILE_UPLOAD_RESTRICTIONS_ERROR";
  ErrorType2["INVALID_PHONE_COUNTRY_CODE_ERROR"] = "INVALID_PHONE_COUNTRY_CODE_ERROR";
  ErrorType2["INVALID_STAFF_ID_ERROR"] = "INVALID_STAFF_ID_ERROR";
  ErrorType2["INVALID_LOCATION_ID_ERROR"] = "INVALID_LOCATION_ID_ERROR";
  ErrorType2["MISSING_SERVICE_OPTION_ERROR"] = "MISSING_SERVICE_OPTION_ERROR";
  ErrorType2["INVALID_SERVICE_OPTIONS_ERROR"] = "INVALID_SERVICE_OPTIONS_ERROR";
  ErrorType2["NO_AVAILABLE_SERVICE_OPTIONS_ERROR"] = "NO_AVAILABLE_SERVICE_OPTIONS_ERROR";
  return ErrorType2;
})(ErrorType || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/forms-v4-form-spam-submission-report-form-spam-submission-reports.meta.ts
function checkForSpam2() {
  const payload = {};
  const getRequestOptions = checkForSpam(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v4/form-spam-submission-reports/check-for-spam",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createFormSpamSubmissionReport2() {
  const payload = {};
  const getRequestOptions = createFormSpamSubmissionReport(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v4/form-spam-submission-reports",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getFormSpamSubmissionReport2() {
  const payload = {
    formSpamSubmissionReportId: ":formSpamSubmissionReportId"
  };
  const getRequestOptions = getFormSpamSubmissionReport(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v4/form-spam-submission-reports/{formSpamSubmissionReportId}",
    pathParams: { formSpamSubmissionReportId: "formSpamSubmissionReportId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteFormSpamSubmissionReport2() {
  const payload = {
    formSpamSubmissionReportId: ":formSpamSubmissionReportId"
  };
  const getRequestOptions = deleteFormSpamSubmissionReport(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v4/form-spam-submission-reports/{formSpamSubmissionReportId}",
    pathParams: { formSpamSubmissionReportId: "formSpamSubmissionReportId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkDeleteFormSpamSubmissionReport2() {
  const payload = {};
  const getRequestOptions = bulkDeleteFormSpamSubmissionReport(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v4/bulk/form-spam-submission-reports/delete",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkDeleteFormSpamSubmissionReportByFilter2() {
  const payload = {};
  const getRequestOptions = bulkDeleteFormSpamSubmissionReportByFilter(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v4/bulk/form-spam-submission-reports/delete-by-filter",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function reportNotSpamSubmission2() {
  const payload = {
    formSpamSubmissionReportId: ":formSpamSubmissionReportId"
  };
  const getRequestOptions = reportNotSpamSubmission(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v4/form-spam-submission-reports/{formSpamSubmissionReportId}/report-not-spam-submission",
    pathParams: { formSpamSubmissionReportId: "formSpamSubmissionReportId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkReportNotSpamSubmission2() {
  const payload = {};
  const getRequestOptions = bulkReportNotSpamSubmission(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v4/bulk/form-spam-submission-reports/report-not-spam-submission",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function reportSpamSubmission2() {
  const payload = { submissionId: ":submissionId" };
  const getRequestOptions = reportSpamSubmission(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v4/submissions/{submissionId}/report-spam-submission",
    pathParams: { submissionId: "submissionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkReportSpamSubmission2() {
  const payload = {};
  const getRequestOptions = bulkReportSpamSubmission(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v4/bulk/submissions/report-spam-submission",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryFormSpamSubmissionReportsByNamespace2() {
  const payload = {};
  const getRequestOptions = queryFormSpamSubmissionReportsByNamespace(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v4/form-spam-submission-reports/namespace/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function countFormSpamSubmissionReports2() {
  const payload = {};
  const getRequestOptions = countFormSpamSubmissionReports(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v4/form-spam-submission-reports/count",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  ErrorType as ErrorTypeOriginal,
  ReportReason as ReportReasonOriginal,
  SortOrder as SortOrderOriginal,
  SubmissionStatus as SubmissionStatusOriginal,
  WebhookIdentityType as WebhookIdentityTypeOriginal,
  bulkDeleteFormSpamSubmissionReport2 as bulkDeleteFormSpamSubmissionReport,
  bulkDeleteFormSpamSubmissionReportByFilter2 as bulkDeleteFormSpamSubmissionReportByFilter,
  bulkReportNotSpamSubmission2 as bulkReportNotSpamSubmission,
  bulkReportSpamSubmission2 as bulkReportSpamSubmission,
  checkForSpam2 as checkForSpam,
  countFormSpamSubmissionReports2 as countFormSpamSubmissionReports,
  createFormSpamSubmissionReport2 as createFormSpamSubmissionReport,
  deleteFormSpamSubmissionReport2 as deleteFormSpamSubmissionReport,
  getFormSpamSubmissionReport2 as getFormSpamSubmissionReport,
  queryFormSpamSubmissionReportsByNamespace2 as queryFormSpamSubmissionReportsByNamespace,
  reportNotSpamSubmission2 as reportNotSpamSubmission,
  reportSpamSubmission2 as reportSpamSubmission
};
