// @wix/auto_sdk_online-programs_participants/build/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  BenefitType: () => BenefitType,
  Event: () => Event,
  EventType: () => EventType,
  Initiator: () => Initiator,
  JoinPath: () => JoinPath,
  OrderItemCategory: () => OrderItemCategory,
  RefundStatus: () => RefundStatus,
  RemovalReason: () => RemovalReason,
  ScheduledActionType: () => ScheduledActionType,
  SortOrder: () => SortOrder,
  State: () => State,
  Status: () => Status,
  SubdivisionType: () => SubdivisionType,
  TransactionStatus: () => TransactionStatus,
  VatType: () => VatType,
  WebhookIdentityType: () => WebhookIdentityType,
  applyCoupon: () => applyCoupon4,
  bulkAddAllToProgram: () => bulkAddAllToProgram4,
  bulkAddToProgram: () => bulkAddToProgram4,
  bulkApproveAllParticipant: () => bulkApproveAllParticipant4,
  bulkApproveParticipant: () => bulkApproveParticipant4,
  bulkInviteAllToProgram: () => bulkInviteAllToProgram4,
  bulkInviteToProgram: () => bulkInviteToProgram4,
  deleteParticipant: () => deleteParticipant4,
  exportParticipantsStats: () => exportParticipantsStats4,
  getAccessStatus: () => getAccessStatus4,
  getCompletionStats: () => getCompletionStats4,
  getParticipant: () => getParticipant4,
  getPremiumStats: () => getPremiumStats4,
  issueCertificate: () => issueCertificate4,
  joinProgram: () => joinProgram4,
  leaveProgram: () => leaveProgram4,
  onParticipantAdded: () => onParticipantAdded2,
  onParticipantApproved: () => onParticipantApproved2,
  onParticipantDeleted: () => onParticipantDeleted2,
  onParticipantInvited: () => onParticipantInvited2,
  onParticipantJoined: () => onParticipantJoined2,
  onParticipantLeft: () => onParticipantLeft2,
  onParticipantRejected: () => onParticipantRejected2,
  queryParticipants: () => queryParticipants4,
  revertCoupon: () => revertCoupon4
});
module.exports = __toCommonJS(index_exports);

// src/online-programs-v3-participant-participants.public.ts
var import_rename_all_nested_keys2 = require("@wix/sdk-runtime/rename-all-nested-keys");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths3 = require("@wix/sdk-runtime/transformations/transform-paths");
var import_sdk_types = require("@wix/sdk-types");

// src/online-programs-v3-participant-participants.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/online-programs-v3-participant-participants.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixOnlineprogramsParticipantsV3ParticipantExportsServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/participants-service",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
function resolveWixOnlineprogramsParticipantsV3ManageParticipantsServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/participants-service",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
function resolveWixOnlineprogramsParticipantsV3ParticipantsServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/participants-service",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_online-programs_participants";
function joinProgram(payload) {
  function __joinProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "POST",
      methodFqn: "wix.onlineprograms.participants.v3.ParticipantsService.JoinProgram",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ParticipantsServiceUrl({
        protoPath: "/v3/participants/join",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "participant.createdDate" },
            { path: "participant.updatedDate" },
            { path: "participant.certificate.issuedDate" }
          ]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [{ path: "participant.progress.completionPercentage" }]
        }
      ])
    };
    return metadata;
  }
  return __joinProgram;
}
function getParticipant(payload) {
  function __getParticipant({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "GET",
      methodFqn: "wix.onlineprograms.participants.v3.ParticipantsService.GetParticipant",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ParticipantsServiceUrl({
        protoPath: "/v3/participants/{participantId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "participant.createdDate" },
            { path: "participant.updatedDate" },
            { path: "participant.certificate.issuedDate" }
          ]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [{ path: "participant.progress.completionPercentage" }]
        }
      ])
    };
    return metadata;
  }
  return __getParticipant;
}
function leaveProgram(payload) {
  function __leaveProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "POST",
      methodFqn: "wix.onlineprograms.participants.v3.ParticipantsService.LeaveProgram",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ParticipantsServiceUrl({
        protoPath: "/v3/participants/{participantId}/leave",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __leaveProgram;
}
function queryParticipants(payload) {
  function __queryParticipants({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "POST",
      methodFqn: "wix.onlineprograms.participants.v3.ParticipantsService.QueryParticipants",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ParticipantsServiceUrl({
        protoPath: "/v3/participants/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "participants.createdDate" },
            { path: "participants.updatedDate" },
            { path: "participants.certificate.issuedDate" }
          ]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [{ path: "participants.progress.completionPercentage" }]
        }
      ])
    };
    return metadata;
  }
  return __queryParticipants;
}
function getAccessStatus(payload) {
  function __getAccessStatus({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "GET",
      methodFqn: "wix.onlineprograms.participants.v3.ParticipantsService.GetAccessStatus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ParticipantsServiceUrl({
        protoPath: "/v3/participants/access",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getAccessStatus;
}
function applyCoupon(payload) {
  function __applyCoupon({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "POST",
      methodFqn: "wix.onlineprograms.participants.v3.ParticipantsService.ApplyCoupon",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ParticipantsServiceUrl({
        protoPath: "/v3/participants/{participantId}/coupons/{couponCode}/apply",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [
            { path: "subTotal" },
            { path: "discount" },
            { path: "total" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __applyCoupon;
}
function revertCoupon(payload) {
  function __revertCoupon({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "POST",
      methodFqn: "wix.onlineprograms.participants.v3.ParticipantsService.RevertCoupon",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ParticipantsServiceUrl({
        protoPath: "/v3/participants/{participantId}/coupons/{couponId}/revert",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [{ path: "total" }]
        }
      ])
    };
    return metadata;
  }
  return __revertCoupon;
}
function bulkInviteToProgram(payload) {
  function __bulkInviteToProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "POST",
      methodFqn: "wix.onlineprograms.participants.v3.ManageParticipantsService.BulkInviteToProgram",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ManageParticipantsServiceUrl({
        protoPath: "/v3/bulk/participants/invite",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.certificate.issuedDate" }
          ]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [{ path: "results.item.progress.completionPercentage" }]
        }
      ])
    };
    return metadata;
  }
  return __bulkInviteToProgram;
}
function bulkInviteAllToProgram(payload) {
  function __bulkInviteAllToProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "POST",
      methodFqn: "wix.onlineprograms.participants.v3.ManageParticipantsService.BulkInviteAllToProgram",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ManageParticipantsServiceUrl({
        protoPath: "/v3/bulk/participants/invite-all",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkInviteAllToProgram;
}
function bulkAddToProgram(payload) {
  function __bulkAddToProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "POST",
      methodFqn: "wix.onlineprograms.participants.v3.ManageParticipantsService.BulkAddToProgram",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ManageParticipantsServiceUrl({
        protoPath: "/v3/bulk/participants/add",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.certificate.issuedDate" }
          ]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [{ path: "results.item.progress.completionPercentage" }]
        }
      ])
    };
    return metadata;
  }
  return __bulkAddToProgram;
}
function bulkAddAllToProgram(payload) {
  function __bulkAddAllToProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "POST",
      methodFqn: "wix.onlineprograms.participants.v3.ManageParticipantsService.BulkAddAllToProgram",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ManageParticipantsServiceUrl({
        protoPath: "/v3/bulk/participants/add-all",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkAddAllToProgram;
}
function bulkApproveParticipant(payload) {
  function __bulkApproveParticipant({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "POST",
      methodFqn: "wix.onlineprograms.participants.v3.ManageParticipantsService.BulkApproveParticipant",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ManageParticipantsServiceUrl({
        protoPath: "/v3/bulk/participants/approve",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.certificate.issuedDate" }
          ]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [{ path: "results.item.progress.completionPercentage" }]
        }
      ])
    };
    return metadata;
  }
  return __bulkApproveParticipant;
}
function bulkApproveAllParticipant(payload) {
  function __bulkApproveAllParticipant({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "POST",
      methodFqn: "wix.onlineprograms.participants.v3.ManageParticipantsService.BulkApproveAllParticipant",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ManageParticipantsServiceUrl({
        protoPath: "/v3/bulk/participants/approve-all",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkApproveAllParticipant;
}
function deleteParticipant(payload) {
  function __deleteParticipant({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "DELETE",
      methodFqn: "wix.onlineprograms.participants.v3.ManageParticipantsService.DeleteParticipant",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ManageParticipantsServiceUrl({
        protoPath: "/v3/participants/{participantId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteParticipant;
}
function issueCertificate(payload) {
  function __issueCertificate({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "POST",
      methodFqn: "wix.onlineprograms.participants.v3.ManageParticipantsService.IssueCertificate",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ManageParticipantsServiceUrl({
        protoPath: "/v3/participants/{participantId}/issue-certificate",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "participant.createdDate" },
            { path: "participant.updatedDate" },
            { path: "participant.certificate.issuedDate" }
          ]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [{ path: "participant.progress.completionPercentage" }]
        }
      ])
    };
    return metadata;
  }
  return __issueCertificate;
}
function getCompletionStats(payload) {
  function __getCompletionStats({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "GET",
      methodFqn: "wix.onlineprograms.participants.v3.ManageParticipantsService.GetCompletionStats",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ManageParticipantsServiceUrl({
        protoPath: "/v3/stats/completion/{programId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [{ path: "completionRate" }]
        }
      ])
    };
    return metadata;
  }
  return __getCompletionStats;
}
function getPremiumStats(payload) {
  function __getPremiumStats({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "GET",
      methodFqn: "wix.onlineprograms.participants.v3.ManageParticipantsService.GetPremiumStats",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ManageParticipantsServiceUrl({
        protoPath: "/v3/stats/premium",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getPremiumStats;
}
function exportParticipantsStats(payload) {
  function __exportParticipantsStats({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "GET",
      methodFqn: "wix.onlineprograms.participants.v3.ParticipantExportsService.ExportParticipantsStats",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ParticipantExportsServiceUrl({
        protoPath: "/v3/participants/export/{programId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __exportParticipantsStats;
}

// src/online-programs-v3-participant-participants.universal.ts
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var JoinPath = /* @__PURE__ */ ((JoinPath2) => {
  JoinPath2["UNKNOWN_JOIN_PATH"] = "UNKNOWN_JOIN_PATH";
  JoinPath2["FREE"] = "FREE";
  JoinPath2["ADDED"] = "ADDED";
  JoinPath2["SINGLE_PAYMENT"] = "SINGLE_PAYMENT";
  JoinPath2["PAID_PLAN"] = "PAID_PLAN";
  JoinPath2["FREE_COUPON"] = "FREE_COUPON";
  return JoinPath2;
})(JoinPath || {});
var State = /* @__PURE__ */ ((State2) => {
  State2["UNKNOWN_STATE"] = "UNKNOWN_STATE";
  State2["INVITED"] = "INVITED";
  State2["APPROVAL_PENDING"] = "APPROVAL_PENDING";
  State2["PAYMENT_PENDING"] = "PAYMENT_PENDING";
  State2["JOINED"] = "JOINED";
  State2["COMPLETED"] = "COMPLETED";
  State2["FAILED"] = "FAILED";
  State2["SUSPENDED"] = "SUSPENDED";
  State2["AUTO_REMOVED"] = "AUTO_REMOVED";
  return State2;
})(State || {});
var RemovalReason = /* @__PURE__ */ ((RemovalReason2) => {
  RemovalReason2["UNKNOWN_REMOVAL_REASON"] = "UNKNOWN_REMOVAL_REASON";
  RemovalReason2["LEGITIMATE_LEAVE"] = "LEGITIMATE_LEAVE";
  RemovalReason2["CLEANUP_FOR_REJOIN"] = "CLEANUP_FOR_REJOIN";
  RemovalReason2["CLEANUP_FOR_MANUAL_ADD"] = "CLEANUP_FOR_MANUAL_ADD";
  return RemovalReason2;
})(RemovalReason || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var EventType = /* @__PURE__ */ ((EventType2) => {
  EventType2["UNDEFINED"] = "UNDEFINED";
  EventType2["TRANSACTION_STATUS_CHANGED"] = "TRANSACTION_STATUS_CHANGED";
  EventType2["REFUND_STATUS_CHANGED"] = "REFUND_STATUS_CHANGED";
  EventType2["RECURRING_PAYMENT_STATUS_CHANGED"] = "RECURRING_PAYMENT_STATUS_CHANGED";
  EventType2["TRANSACTION_UPDATED"] = "TRANSACTION_UPDATED";
  EventType2["TRANSACTION_CREATED"] = "TRANSACTION_CREATED";
  return EventType2;
})(EventType || {});
var OrderItemCategory = /* @__PURE__ */ ((OrderItemCategory2) => {
  OrderItemCategory2["UNDEFINED"] = "UNDEFINED";
  OrderItemCategory2["PHYSICAL"] = "PHYSICAL";
  OrderItemCategory2["DIGITAL"] = "DIGITAL";
  return OrderItemCategory2;
})(OrderItemCategory || {});
var SubdivisionType = /* @__PURE__ */ ((SubdivisionType2) => {
  SubdivisionType2["UNKNOWN_SUBDIVISION_TYPE"] = "UNKNOWN_SUBDIVISION_TYPE";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_1"] = "ADMINISTRATIVE_AREA_LEVEL_1";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_2"] = "ADMINISTRATIVE_AREA_LEVEL_2";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_3"] = "ADMINISTRATIVE_AREA_LEVEL_3";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_4"] = "ADMINISTRATIVE_AREA_LEVEL_4";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_5"] = "ADMINISTRATIVE_AREA_LEVEL_5";
  SubdivisionType2["COUNTRY"] = "COUNTRY";
  return SubdivisionType2;
})(SubdivisionType || {});
var VatType = /* @__PURE__ */ ((VatType2) => {
  VatType2["UNSPECIFIED"] = "UNSPECIFIED";
  VatType2["CPF"] = "CPF";
  VatType2["CNPJ"] = "CNPJ";
  return VatType2;
})(VatType || {});
var TransactionStatus = /* @__PURE__ */ ((TransactionStatus2) => {
  TransactionStatus2["UNDEFINED"] = "UNDEFINED";
  TransactionStatus2["APPROVED"] = "APPROVED";
  TransactionStatus2["PENDING"] = "PENDING";
  TransactionStatus2["PENDING_MERCHANT"] = "PENDING_MERCHANT";
  TransactionStatus2["CANCELED"] = "CANCELED";
  TransactionStatus2["DECLINED"] = "DECLINED";
  TransactionStatus2["REFUNDED"] = "REFUNDED";
  TransactionStatus2["PARTIALLY_REFUNDED"] = "PARTIALLY_REFUNDED";
  TransactionStatus2["AUTHORIZED"] = "AUTHORIZED";
  TransactionStatus2["VOIDED"] = "VOIDED";
  return TransactionStatus2;
})(TransactionStatus || {});
var ScheduledActionType = /* @__PURE__ */ ((ScheduledActionType2) => {
  ScheduledActionType2["VOID"] = "VOID";
  ScheduledActionType2["CAPTURE"] = "CAPTURE";
  return ScheduledActionType2;
})(ScheduledActionType || {});
var RefundStatus = /* @__PURE__ */ ((RefundStatus2) => {
  RefundStatus2["UNDEFINED"] = "UNDEFINED";
  RefundStatus2["APPROVED"] = "APPROVED";
  RefundStatus2["PENDING"] = "PENDING";
  RefundStatus2["DECLINED"] = "DECLINED";
  return RefundStatus2;
})(RefundStatus || {});
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["UNDEFINED"] = "UNDEFINED";
  Status2["PENDING"] = "PENDING";
  Status2["ACTIVE"] = "ACTIVE";
  Status2["CANCELLED"] = "CANCELLED";
  Status2["FINISHED"] = "FINISHED";
  Status2["SUSPENDED"] = "SUSPENDED";
  return Status2;
})(Status || {});
var Initiator = /* @__PURE__ */ ((Initiator2) => {
  Initiator2["UNDEFINED"] = "UNDEFINED";
  Initiator2["BUYER"] = "BUYER";
  Initiator2["MERCHANT"] = "MERCHANT";
  Initiator2["PAYMENT"] = "PAYMENT";
  Initiator2["SETUP"] = "SETUP";
  return Initiator2;
})(Initiator || {});
var BenefitType = /* @__PURE__ */ ((BenefitType2) => {
  BenefitType2["UNDEFINED"] = "UNDEFINED";
  BenefitType2["LIMITED"] = "LIMITED";
  BenefitType2["UNLIMITED"] = "UNLIMITED";
  return BenefitType2;
})(BenefitType || {});
var Event = /* @__PURE__ */ ((Event2) => {
  Event2["Updated"] = "Updated";
  Event2["Deleted"] = "Deleted";
  Event2["Created"] = "Created";
  return Event2;
})(Event || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function joinProgram2(programId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    programId
  });
  const reqOpts = joinProgram(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { programId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["programId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getParticipant2(participantId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    participantId
  });
  const reqOpts = getParticipant(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.participant;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { participantId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["participantId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function leaveProgram2(participantId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    participantId
  });
  const reqOpts = leaveProgram(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { participantId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["participantId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryParticipants2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryParticipants(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [])
      );
      return {
        items: transformedData?.participants,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function getAccessStatus2() {
  const { httpClient, sideEffects } = arguments[0];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({});
  const reqOpts = getAccessStatus(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {},
        singleArgumentUnchanged: false
      },
      []
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function applyCoupon2(identifiers, orderId) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    participantId: identifiers?.participantId,
    couponCode: identifiers?.couponCode,
    orderId
  });
  const reqOpts = applyCoupon(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          participantId: "$[0].participantId",
          couponCode: "$[0].couponCode",
          orderId: "$[1]"
        },
        singleArgumentUnchanged: false
      },
      ["identifiers", "orderId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function revertCoupon2(identifiers, orderId) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    participantId: identifiers?.participantId,
    couponId: identifiers?.couponId,
    orderId
  });
  const reqOpts = revertCoupon(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          participantId: "$[0].participantId",
          couponId: "$[0].couponId",
          orderId: "$[1]"
        },
        singleArgumentUnchanged: false
      },
      ["identifiers", "orderId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkInviteToProgram2(programId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    programId,
    memberIds: options?.memberIds,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkInviteToProgram(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          programId: "$[0]",
          memberIds: "$[1].memberIds",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["programId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkInviteAllToProgram2(programId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    programId,
    excludeMemberIds: options?.excludeMemberIds
  });
  const reqOpts = bulkInviteAllToProgram(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          programId: "$[0]",
          excludeMemberIds: "$[1].excludeMemberIds"
        },
        singleArgumentUnchanged: false
      },
      ["programId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkAddToProgram2(programId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    programId,
    memberIds: options?.memberIds,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkAddToProgram(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          programId: "$[0]",
          memberIds: "$[1].memberIds",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["programId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkAddAllToProgram2(programId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    programId,
    excludeMemberIds: options?.excludeMemberIds
  });
  const reqOpts = bulkAddAllToProgram(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          programId: "$[0]",
          excludeMemberIds: "$[1].excludeMemberIds"
        },
        singleArgumentUnchanged: false
      },
      ["programId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkApproveParticipant2(participantIds, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    participantIds,
    programId: options?.programId,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkApproveParticipant(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          participantIds: "$[0]",
          programId: "$[1].programId",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["participantIds", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkApproveAllParticipant2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    programId: options?.programId,
    excludeParticipantIds: options?.excludeParticipantIds
  });
  const reqOpts = bulkApproveAllParticipant(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          programId: "$[0].programId",
          excludeParticipantIds: "$[0].excludeParticipantIds"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteParticipant2(participantId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    participantId
  });
  const reqOpts = deleteParticipant(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { participantId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["participantId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function issueCertificate2(participantId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    participantId
  });
  const reqOpts = issueCertificate(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { participantId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["participantId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getCompletionStats2(programId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    programId
  });
  const reqOpts = getCompletionStats(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { programId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["programId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getPremiumStats2() {
  const { httpClient, sideEffects } = arguments[0];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({});
  const reqOpts = getPremiumStats(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {},
        singleArgumentUnchanged: false
      },
      []
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function exportParticipantsStats2(programId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    programId
  });
  const reqOpts = exportParticipantsStats(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { programId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["programId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/online-programs-v3-participant-participants.public.ts
function joinProgram3(httpClient) {
  return (programId) => joinProgram2(
    programId,
    // @ts-ignore
    { httpClient }
  );
}
function getParticipant3(httpClient) {
  return (participantId) => getParticipant2(
    participantId,
    // @ts-ignore
    { httpClient }
  );
}
function leaveProgram3(httpClient) {
  return (participantId) => leaveProgram2(
    participantId,
    // @ts-ignore
    { httpClient }
  );
}
function queryParticipants3(httpClient) {
  return () => queryParticipants2(
    // @ts-ignore
    { httpClient }
  );
}
function getAccessStatus3(httpClient) {
  return () => getAccessStatus2(
    // @ts-ignore
    { httpClient }
  );
}
function applyCoupon3(httpClient) {
  return (identifiers, orderId) => applyCoupon2(
    identifiers,
    orderId,
    // @ts-ignore
    { httpClient }
  );
}
function revertCoupon3(httpClient) {
  return (identifiers, orderId) => revertCoupon2(
    identifiers,
    orderId,
    // @ts-ignore
    { httpClient }
  );
}
function bulkInviteToProgram3(httpClient) {
  return (programId, options) => bulkInviteToProgram2(
    programId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkInviteAllToProgram3(httpClient) {
  return (programId, options) => bulkInviteAllToProgram2(
    programId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkAddToProgram3(httpClient) {
  return (programId, options) => bulkAddToProgram2(
    programId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkAddAllToProgram3(httpClient) {
  return (programId, options) => bulkAddAllToProgram2(
    programId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkApproveParticipant3(httpClient) {
  return (participantIds, options) => bulkApproveParticipant2(
    participantIds,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkApproveAllParticipant3(httpClient) {
  return (options) => bulkApproveAllParticipant2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function deleteParticipant3(httpClient) {
  return (participantId) => deleteParticipant2(
    participantId,
    // @ts-ignore
    { httpClient }
  );
}
function issueCertificate3(httpClient) {
  return (participantId) => issueCertificate2(
    participantId,
    // @ts-ignore
    { httpClient }
  );
}
function getCompletionStats3(httpClient) {
  return (programId) => getCompletionStats2(
    programId,
    // @ts-ignore
    { httpClient }
  );
}
function getPremiumStats3(httpClient) {
  return () => getPremiumStats2(
    // @ts-ignore
    { httpClient }
  );
}
function exportParticipantsStats3(httpClient) {
  return (programId) => exportParticipantsStats2(
    programId,
    // @ts-ignore
    { httpClient }
  );
}
var onParticipantJoined = (0, import_sdk_types.EventDefinition)(
  "wix.online_programs.v3.participant_participant_joined",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "data.participant.createdDate" },
          { path: "data.participant.updatedDate" },
          { path: "data.participant.certificate.issuedDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_float2.transformRESTFloatToSDKFloat,
        paths: [{ path: "data.participant.progress.completionPercentage" }]
      }
    ])
  )
)();
var onParticipantLeft = (0, import_sdk_types.EventDefinition)(
  "wix.online_programs.v3.participant_participant_left",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
        paths: [{ path: "metadata.eventTime" }]
      }
    ])
  )
)();
var onParticipantDeleted = (0, import_sdk_types.EventDefinition)(
  "wix.online_programs.v3.participant_deleted",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.certificate.issuedDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_float2.transformRESTFloatToSDKFloat,
        paths: [{ path: "entity.progress.completionPercentage" }]
      }
    ])
  )
)();
var onParticipantAdded = (0, import_sdk_types.EventDefinition)(
  "wix.online_programs.v3.participant_participant_added",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
        paths: [{ path: "metadata.eventTime" }]
      }
    ])
  )
)();
var onParticipantApproved = (0, import_sdk_types.EventDefinition)(
  "wix.online_programs.v3.participant_participant_approved",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
        paths: [{ path: "metadata.eventTime" }]
      }
    ])
  )
)();
var onParticipantInvited = (0, import_sdk_types.EventDefinition)(
  "wix.online_programs.v3.participant_participant_invited",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
        paths: [{ path: "metadata.eventTime" }]
      }
    ])
  )
)();
var onParticipantRejected = (0, import_sdk_types.EventDefinition)(
  "wix.online_programs.v3.participant_participant_rejected",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
        paths: [{ path: "metadata.eventTime" }]
      }
    ])
  )
)();

// src/online-programs-v3-participant-participants.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var import_event_definition_modules = require("@wix/sdk-runtime/event-definition-modules");
var joinProgram4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(joinProgram3);
var getParticipant4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getParticipant3);
var leaveProgram4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(leaveProgram3);
var queryParticipants4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(queryParticipants3);
var getAccessStatus4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getAccessStatus3);
var applyCoupon4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(applyCoupon3);
var revertCoupon4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(revertCoupon3);
var bulkInviteToProgram4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkInviteToProgram3);
var bulkInviteAllToProgram4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkInviteAllToProgram3);
var bulkAddToProgram4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkAddToProgram3);
var bulkAddAllToProgram4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkAddAllToProgram3);
var bulkApproveParticipant4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkApproveParticipant3);
var bulkApproveAllParticipant4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkApproveAllParticipant3);
var deleteParticipant4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(deleteParticipant3);
var issueCertificate4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(issueCertificate3);
var getCompletionStats4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getCompletionStats3);
var getPremiumStats4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getPremiumStats3);
var exportParticipantsStats4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(exportParticipantsStats3);
var onParticipantJoined2 = (0, import_event_definition_modules.createEventModule)(
  onParticipantJoined
);
var onParticipantLeft2 = (0, import_event_definition_modules.createEventModule)(onParticipantLeft);
var onParticipantDeleted2 = (0, import_event_definition_modules.createEventModule)(
  onParticipantDeleted
);
var onParticipantAdded2 = (0, import_event_definition_modules.createEventModule)(onParticipantAdded);
var onParticipantApproved2 = (0, import_event_definition_modules.createEventModule)(
  onParticipantApproved
);
var onParticipantInvited2 = (0, import_event_definition_modules.createEventModule)(
  onParticipantInvited
);
var onParticipantRejected2 = (0, import_event_definition_modules.createEventModule)(
  onParticipantRejected
);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  BenefitType,
  Event,
  EventType,
  Initiator,
  JoinPath,
  OrderItemCategory,
  RefundStatus,
  RemovalReason,
  ScheduledActionType,
  SortOrder,
  State,
  Status,
  SubdivisionType,
  TransactionStatus,
  VatType,
  WebhookIdentityType,
  applyCoupon,
  bulkAddAllToProgram,
  bulkAddToProgram,
  bulkApproveAllParticipant,
  bulkApproveParticipant,
  bulkInviteAllToProgram,
  bulkInviteToProgram,
  deleteParticipant,
  exportParticipantsStats,
  getAccessStatus,
  getCompletionStats,
  getParticipant,
  getPremiumStats,
  issueCertificate,
  joinProgram,
  leaveProgram,
  onParticipantAdded,
  onParticipantApproved,
  onParticipantDeleted,
  onParticipantInvited,
  onParticipantJoined,
  onParticipantLeft,
  onParticipantRejected,
  queryParticipants,
  revertCoupon
});
