// @wix/auto_sdk_online-programs_participants/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  BenefitTypeOriginal: () => BenefitType,
  EventOriginal: () => Event,
  EventTypeOriginal: () => EventType,
  InitiatorOriginal: () => Initiator,
  JoinPathOriginal: () => JoinPath,
  OrderItemCategoryOriginal: () => OrderItemCategory,
  RefundStatusOriginal: () => RefundStatus,
  RemovalReasonOriginal: () => RemovalReason,
  ScheduledActionTypeOriginal: () => ScheduledActionType,
  SortOrderOriginal: () => SortOrder,
  StateOriginal: () => State,
  StatusOriginal: () => Status,
  SubdivisionTypeOriginal: () => SubdivisionType,
  TransactionStatusOriginal: () => TransactionStatus,
  VatTypeOriginal: () => VatType,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  applyCoupon: () => applyCoupon2,
  bulkAddAllToProgram: () => bulkAddAllToProgram2,
  bulkAddToProgram: () => bulkAddToProgram2,
  bulkApproveAllParticipant: () => bulkApproveAllParticipant2,
  bulkApproveParticipant: () => bulkApproveParticipant2,
  bulkInviteAllToProgram: () => bulkInviteAllToProgram2,
  bulkInviteToProgram: () => bulkInviteToProgram2,
  deleteParticipant: () => deleteParticipant2,
  exportParticipantsStats: () => exportParticipantsStats2,
  getAccessStatus: () => getAccessStatus2,
  getCompletionStats: () => getCompletionStats2,
  getParticipant: () => getParticipant2,
  getPremiumStats: () => getPremiumStats2,
  issueCertificate: () => issueCertificate2,
  joinProgram: () => joinProgram2,
  leaveProgram: () => leaveProgram2,
  queryParticipants: () => queryParticipants2,
  revertCoupon: () => revertCoupon2
});
module.exports = __toCommonJS(meta_exports);

// src/online-programs-v3-participant-participants.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixOnlineprogramsParticipantsV3ParticipantExportsServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/participants-service",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
function resolveWixOnlineprogramsParticipantsV3ManageParticipantsServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/participants-service",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
function resolveWixOnlineprogramsParticipantsV3ParticipantsServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/participants-service",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/participants-service",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_online-programs_participants";
function joinProgram(payload) {
  function __joinProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "POST",
      methodFqn: "wix.onlineprograms.participants.v3.ParticipantsService.JoinProgram",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ParticipantsServiceUrl({
        protoPath: "/v3/participants/join",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "participant.createdDate" },
            { path: "participant.updatedDate" },
            { path: "participant.certificate.issuedDate" }
          ]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [{ path: "participant.progress.completionPercentage" }]
        }
      ])
    };
    return metadata;
  }
  return __joinProgram;
}
function getParticipant(payload) {
  function __getParticipant({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "GET",
      methodFqn: "wix.onlineprograms.participants.v3.ParticipantsService.GetParticipant",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ParticipantsServiceUrl({
        protoPath: "/v3/participants/{participantId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "participant.createdDate" },
            { path: "participant.updatedDate" },
            { path: "participant.certificate.issuedDate" }
          ]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [{ path: "participant.progress.completionPercentage" }]
        }
      ])
    };
    return metadata;
  }
  return __getParticipant;
}
function leaveProgram(payload) {
  function __leaveProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "POST",
      methodFqn: "wix.onlineprograms.participants.v3.ParticipantsService.LeaveProgram",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ParticipantsServiceUrl({
        protoPath: "/v3/participants/{participantId}/leave",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __leaveProgram;
}
function queryParticipants(payload) {
  function __queryParticipants({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "POST",
      methodFqn: "wix.onlineprograms.participants.v3.ParticipantsService.QueryParticipants",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ParticipantsServiceUrl({
        protoPath: "/v3/participants/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "participants.createdDate" },
            { path: "participants.updatedDate" },
            { path: "participants.certificate.issuedDate" }
          ]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [{ path: "participants.progress.completionPercentage" }]
        }
      ])
    };
    return metadata;
  }
  return __queryParticipants;
}
function getAccessStatus(payload) {
  function __getAccessStatus({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "GET",
      methodFqn: "wix.onlineprograms.participants.v3.ParticipantsService.GetAccessStatus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ParticipantsServiceUrl({
        protoPath: "/v3/participants/access",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getAccessStatus;
}
function applyCoupon(payload) {
  function __applyCoupon({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "POST",
      methodFqn: "wix.onlineprograms.participants.v3.ParticipantsService.ApplyCoupon",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ParticipantsServiceUrl({
        protoPath: "/v3/participants/{participantId}/coupons/{couponCode}/apply",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [
            { path: "subTotal" },
            { path: "discount" },
            { path: "total" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __applyCoupon;
}
function revertCoupon(payload) {
  function __revertCoupon({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "POST",
      methodFqn: "wix.onlineprograms.participants.v3.ParticipantsService.RevertCoupon",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ParticipantsServiceUrl({
        protoPath: "/v3/participants/{participantId}/coupons/{couponId}/revert",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [{ path: "total" }]
        }
      ])
    };
    return metadata;
  }
  return __revertCoupon;
}
function bulkInviteToProgram(payload) {
  function __bulkInviteToProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "POST",
      methodFqn: "wix.onlineprograms.participants.v3.ManageParticipantsService.BulkInviteToProgram",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ManageParticipantsServiceUrl({
        protoPath: "/v3/bulk/participants/invite",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.certificate.issuedDate" }
          ]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [{ path: "results.item.progress.completionPercentage" }]
        }
      ])
    };
    return metadata;
  }
  return __bulkInviteToProgram;
}
function bulkInviteAllToProgram(payload) {
  function __bulkInviteAllToProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "POST",
      methodFqn: "wix.onlineprograms.participants.v3.ManageParticipantsService.BulkInviteAllToProgram",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ManageParticipantsServiceUrl({
        protoPath: "/v3/bulk/participants/invite-all",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkInviteAllToProgram;
}
function bulkAddToProgram(payload) {
  function __bulkAddToProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "POST",
      methodFqn: "wix.onlineprograms.participants.v3.ManageParticipantsService.BulkAddToProgram",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ManageParticipantsServiceUrl({
        protoPath: "/v3/bulk/participants/add",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.certificate.issuedDate" }
          ]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [{ path: "results.item.progress.completionPercentage" }]
        }
      ])
    };
    return metadata;
  }
  return __bulkAddToProgram;
}
function bulkAddAllToProgram(payload) {
  function __bulkAddAllToProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "POST",
      methodFqn: "wix.onlineprograms.participants.v3.ManageParticipantsService.BulkAddAllToProgram",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ManageParticipantsServiceUrl({
        protoPath: "/v3/bulk/participants/add-all",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkAddAllToProgram;
}
function bulkApproveParticipant(payload) {
  function __bulkApproveParticipant({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "POST",
      methodFqn: "wix.onlineprograms.participants.v3.ManageParticipantsService.BulkApproveParticipant",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ManageParticipantsServiceUrl({
        protoPath: "/v3/bulk/participants/approve",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.certificate.issuedDate" }
          ]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [{ path: "results.item.progress.completionPercentage" }]
        }
      ])
    };
    return metadata;
  }
  return __bulkApproveParticipant;
}
function bulkApproveAllParticipant(payload) {
  function __bulkApproveAllParticipant({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "POST",
      methodFqn: "wix.onlineprograms.participants.v3.ManageParticipantsService.BulkApproveAllParticipant",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ManageParticipantsServiceUrl({
        protoPath: "/v3/bulk/participants/approve-all",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkApproveAllParticipant;
}
function deleteParticipant(payload) {
  function __deleteParticipant({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "DELETE",
      methodFqn: "wix.onlineprograms.participants.v3.ManageParticipantsService.DeleteParticipant",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ManageParticipantsServiceUrl({
        protoPath: "/v3/participants/{participantId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteParticipant;
}
function issueCertificate(payload) {
  function __issueCertificate({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "POST",
      methodFqn: "wix.onlineprograms.participants.v3.ManageParticipantsService.IssueCertificate",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ManageParticipantsServiceUrl({
        protoPath: "/v3/participants/{participantId}/issue-certificate",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "participant.createdDate" },
            { path: "participant.updatedDate" },
            { path: "participant.certificate.issuedDate" }
          ]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [{ path: "participant.progress.completionPercentage" }]
        }
      ])
    };
    return metadata;
  }
  return __issueCertificate;
}
function getCompletionStats(payload) {
  function __getCompletionStats({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "GET",
      methodFqn: "wix.onlineprograms.participants.v3.ManageParticipantsService.GetCompletionStats",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ManageParticipantsServiceUrl({
        protoPath: "/v3/stats/completion/{programId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [{ path: "completionRate" }]
        }
      ])
    };
    return metadata;
  }
  return __getCompletionStats;
}
function getPremiumStats(payload) {
  function __getPremiumStats({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "GET",
      methodFqn: "wix.onlineprograms.participants.v3.ManageParticipantsService.GetPremiumStats",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ManageParticipantsServiceUrl({
        protoPath: "/v3/stats/premium",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getPremiumStats;
}
function exportParticipantsStats(payload) {
  function __exportParticipantsStats({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.participant",
      method: "GET",
      methodFqn: "wix.onlineprograms.participants.v3.ParticipantExportsService.ExportParticipantsStats",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsParticipantsV3ParticipantExportsServiceUrl({
        protoPath: "/v3/participants/export/{programId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __exportParticipantsStats;
}

// src/online-programs-v3-participant-participants.types.ts
var JoinPath = /* @__PURE__ */ ((JoinPath2) => {
  JoinPath2["UNKNOWN_JOIN_PATH"] = "UNKNOWN_JOIN_PATH";
  JoinPath2["FREE"] = "FREE";
  JoinPath2["ADDED"] = "ADDED";
  JoinPath2["SINGLE_PAYMENT"] = "SINGLE_PAYMENT";
  JoinPath2["PAID_PLAN"] = "PAID_PLAN";
  JoinPath2["FREE_COUPON"] = "FREE_COUPON";
  return JoinPath2;
})(JoinPath || {});
var State = /* @__PURE__ */ ((State2) => {
  State2["UNKNOWN_STATE"] = "UNKNOWN_STATE";
  State2["INVITED"] = "INVITED";
  State2["APPROVAL_PENDING"] = "APPROVAL_PENDING";
  State2["PAYMENT_PENDING"] = "PAYMENT_PENDING";
  State2["JOINED"] = "JOINED";
  State2["COMPLETED"] = "COMPLETED";
  State2["FAILED"] = "FAILED";
  State2["SUSPENDED"] = "SUSPENDED";
  State2["AUTO_REMOVED"] = "AUTO_REMOVED";
  return State2;
})(State || {});
var RemovalReason = /* @__PURE__ */ ((RemovalReason2) => {
  RemovalReason2["UNKNOWN_REMOVAL_REASON"] = "UNKNOWN_REMOVAL_REASON";
  RemovalReason2["LEGITIMATE_LEAVE"] = "LEGITIMATE_LEAVE";
  RemovalReason2["CLEANUP_FOR_REJOIN"] = "CLEANUP_FOR_REJOIN";
  RemovalReason2["CLEANUP_FOR_MANUAL_ADD"] = "CLEANUP_FOR_MANUAL_ADD";
  return RemovalReason2;
})(RemovalReason || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var EventType = /* @__PURE__ */ ((EventType2) => {
  EventType2["UNDEFINED"] = "UNDEFINED";
  EventType2["TRANSACTION_STATUS_CHANGED"] = "TRANSACTION_STATUS_CHANGED";
  EventType2["REFUND_STATUS_CHANGED"] = "REFUND_STATUS_CHANGED";
  EventType2["RECURRING_PAYMENT_STATUS_CHANGED"] = "RECURRING_PAYMENT_STATUS_CHANGED";
  EventType2["TRANSACTION_UPDATED"] = "TRANSACTION_UPDATED";
  EventType2["TRANSACTION_CREATED"] = "TRANSACTION_CREATED";
  return EventType2;
})(EventType || {});
var OrderItemCategory = /* @__PURE__ */ ((OrderItemCategory2) => {
  OrderItemCategory2["UNDEFINED"] = "UNDEFINED";
  OrderItemCategory2["PHYSICAL"] = "PHYSICAL";
  OrderItemCategory2["DIGITAL"] = "DIGITAL";
  return OrderItemCategory2;
})(OrderItemCategory || {});
var SubdivisionType = /* @__PURE__ */ ((SubdivisionType2) => {
  SubdivisionType2["UNKNOWN_SUBDIVISION_TYPE"] = "UNKNOWN_SUBDIVISION_TYPE";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_1"] = "ADMINISTRATIVE_AREA_LEVEL_1";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_2"] = "ADMINISTRATIVE_AREA_LEVEL_2";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_3"] = "ADMINISTRATIVE_AREA_LEVEL_3";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_4"] = "ADMINISTRATIVE_AREA_LEVEL_4";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_5"] = "ADMINISTRATIVE_AREA_LEVEL_5";
  SubdivisionType2["COUNTRY"] = "COUNTRY";
  return SubdivisionType2;
})(SubdivisionType || {});
var VatType = /* @__PURE__ */ ((VatType2) => {
  VatType2["UNSPECIFIED"] = "UNSPECIFIED";
  VatType2["CPF"] = "CPF";
  VatType2["CNPJ"] = "CNPJ";
  return VatType2;
})(VatType || {});
var TransactionStatus = /* @__PURE__ */ ((TransactionStatus2) => {
  TransactionStatus2["UNDEFINED"] = "UNDEFINED";
  TransactionStatus2["APPROVED"] = "APPROVED";
  TransactionStatus2["PENDING"] = "PENDING";
  TransactionStatus2["PENDING_MERCHANT"] = "PENDING_MERCHANT";
  TransactionStatus2["CANCELED"] = "CANCELED";
  TransactionStatus2["DECLINED"] = "DECLINED";
  TransactionStatus2["REFUNDED"] = "REFUNDED";
  TransactionStatus2["PARTIALLY_REFUNDED"] = "PARTIALLY_REFUNDED";
  TransactionStatus2["AUTHORIZED"] = "AUTHORIZED";
  TransactionStatus2["VOIDED"] = "VOIDED";
  return TransactionStatus2;
})(TransactionStatus || {});
var ScheduledActionType = /* @__PURE__ */ ((ScheduledActionType2) => {
  ScheduledActionType2["VOID"] = "VOID";
  ScheduledActionType2["CAPTURE"] = "CAPTURE";
  return ScheduledActionType2;
})(ScheduledActionType || {});
var RefundStatus = /* @__PURE__ */ ((RefundStatus2) => {
  RefundStatus2["UNDEFINED"] = "UNDEFINED";
  RefundStatus2["APPROVED"] = "APPROVED";
  RefundStatus2["PENDING"] = "PENDING";
  RefundStatus2["DECLINED"] = "DECLINED";
  return RefundStatus2;
})(RefundStatus || {});
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["UNDEFINED"] = "UNDEFINED";
  Status2["PENDING"] = "PENDING";
  Status2["ACTIVE"] = "ACTIVE";
  Status2["CANCELLED"] = "CANCELLED";
  Status2["FINISHED"] = "FINISHED";
  Status2["SUSPENDED"] = "SUSPENDED";
  return Status2;
})(Status || {});
var Initiator = /* @__PURE__ */ ((Initiator2) => {
  Initiator2["UNDEFINED"] = "UNDEFINED";
  Initiator2["BUYER"] = "BUYER";
  Initiator2["MERCHANT"] = "MERCHANT";
  Initiator2["PAYMENT"] = "PAYMENT";
  Initiator2["SETUP"] = "SETUP";
  return Initiator2;
})(Initiator || {});
var BenefitType = /* @__PURE__ */ ((BenefitType2) => {
  BenefitType2["UNDEFINED"] = "UNDEFINED";
  BenefitType2["LIMITED"] = "LIMITED";
  BenefitType2["UNLIMITED"] = "UNLIMITED";
  return BenefitType2;
})(BenefitType || {});
var Event = /* @__PURE__ */ ((Event2) => {
  Event2["Updated"] = "Updated";
  Event2["Deleted"] = "Deleted";
  Event2["Created"] = "Created";
  return Event2;
})(Event || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/online-programs-v3-participant-participants.meta.ts
function joinProgram2() {
  const payload = {};
  const getRequestOptions = joinProgram(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/participants/join",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getParticipant2() {
  const payload = { participantId: ":participantId" };
  const getRequestOptions = getParticipant(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/participants/{participantId}",
    pathParams: { participantId: "participantId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function leaveProgram2() {
  const payload = { participantId: ":participantId" };
  const getRequestOptions = leaveProgram(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/participants/{participantId}/leave",
    pathParams: { participantId: "participantId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryParticipants2() {
  const payload = {};
  const getRequestOptions = queryParticipants(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/participants/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getAccessStatus2() {
  const payload = {};
  const getRequestOptions = getAccessStatus(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/participants/access",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function applyCoupon2() {
  const payload = {
    participantId: ":participantId",
    couponCode: ":couponCode"
  };
  const getRequestOptions = applyCoupon(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/participants/{participantId}/coupons/{couponCode}/apply",
    pathParams: { participantId: "participantId", couponCode: "couponCode" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function revertCoupon2() {
  const payload = {
    participantId: ":participantId",
    couponId: ":couponId"
  };
  const getRequestOptions = revertCoupon(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/participants/{participantId}/coupons/{couponId}/revert",
    pathParams: { participantId: "participantId", couponId: "couponId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkInviteToProgram2() {
  const payload = {};
  const getRequestOptions = bulkInviteToProgram(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/bulk/participants/invite",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkInviteAllToProgram2() {
  const payload = {};
  const getRequestOptions = bulkInviteAllToProgram(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/bulk/participants/invite-all",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkAddToProgram2() {
  const payload = {};
  const getRequestOptions = bulkAddToProgram(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/bulk/participants/add",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkAddAllToProgram2() {
  const payload = {};
  const getRequestOptions = bulkAddAllToProgram(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/bulk/participants/add-all",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkApproveParticipant2() {
  const payload = {};
  const getRequestOptions = bulkApproveParticipant(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/bulk/participants/approve",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkApproveAllParticipant2() {
  const payload = {};
  const getRequestOptions = bulkApproveAllParticipant(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/bulk/participants/approve-all",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteParticipant2() {
  const payload = { participantId: ":participantId" };
  const getRequestOptions = deleteParticipant(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v3/participants/{participantId}",
    pathParams: { participantId: "participantId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function issueCertificate2() {
  const payload = { participantId: ":participantId" };
  const getRequestOptions = issueCertificate(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/participants/{participantId}/issue-certificate",
    pathParams: { participantId: "participantId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getCompletionStats2() {
  const payload = { programId: ":programId" };
  const getRequestOptions = getCompletionStats(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/stats/completion/{programId}",
    pathParams: { programId: "programId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getPremiumStats2() {
  const payload = {};
  const getRequestOptions = getPremiumStats(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/stats/premium",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function exportParticipantsStats2() {
  const payload = { programId: ":programId" };
  const getRequestOptions = exportParticipantsStats(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/participants/export/{programId}",
    pathParams: { programId: "programId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  BenefitTypeOriginal,
  EventOriginal,
  EventTypeOriginal,
  InitiatorOriginal,
  JoinPathOriginal,
  OrderItemCategoryOriginal,
  RefundStatusOriginal,
  RemovalReasonOriginal,
  ScheduledActionTypeOriginal,
  SortOrderOriginal,
  StateOriginal,
  StatusOriginal,
  SubdivisionTypeOriginal,
  TransactionStatusOriginal,
  VatTypeOriginal,
  WebhookIdentityTypeOriginal,
  applyCoupon,
  bulkAddAllToProgram,
  bulkAddToProgram,
  bulkApproveAllParticipant,
  bulkApproveParticipant,
  bulkInviteAllToProgram,
  bulkInviteToProgram,
  deleteParticipant,
  exportParticipantsStats,
  getAccessStatus,
  getCompletionStats,
  getParticipant,
  getPremiumStats,
  issueCertificate,
  joinProgram,
  leaveProgram,
  queryParticipants,
  revertCoupon
});
