// @wix/unidriver-common/dist/esm/UniDriverBase/errors.js
const ErrorType = {
    NO_ELEMENTS_WITH_SELECTOR: 'no-elements-with-selector',
    MULTIPLE_ELEMENTS_WITH_SELECTOR: 'multiple-elements-with-selector',
};
export class NoElementWithSelectorError extends Error {
    constructor(params) {
        const { selector, failedPath, fullPath } = params;
        const finalMessage = (() => {
            let message = `Cannot find element with selector: ${selector}`;
            if (failedPath) {
                message += `\r\nFailed path: ${NoElementWithSelectorError.convertSelectorPathToString(failedPath)}`;
            }
            if (fullPath) {
                message += `\r\nFull path: ${NoElementWithSelectorError.convertSelectorPathToString(fullPath)}`;
            }
            return message;
        })();
        super(finalMessage);
        Object.defineProperty(this, "type", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: ErrorType.NO_ELEMENTS_WITH_SELECTOR
        });
        Object.defineProperty(this, "failedPath", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "fullPath", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this.failedPath = failedPath ?? fullPath;
        this.fullPath = fullPath;
    }
}
Object.defineProperty(NoElementWithSelectorError, "convertSelectorPathToString", {
    enumerable: true,
    configurable: true,
    writable: true,
    value: (selectorsPath) => {
        return selectorsPath
            .reduce((acc, item) => {
            if (item.startsWith(':')) {
                acc[acc.length - 1] = `${String(acc.at(-1))}${item}`;
            }
            else {
                acc.push(item);
            }
            return acc;
        }, [])
            .join(' > ');
    }
});
export class MultipleElementsWithSelectorError extends Error {
    constructor(count, selector) {
        super(`Found ${String(count)} elements with selector [${selector}]. Only 1 is expected. This is either a bug or not-specific-enough selector`);
        Object.defineProperty(this, "type", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: ErrorType.MULTIPLE_ELEMENTS_WITH_SELECTOR
        });
    }
}
const assertType = (error, errorType) => {
    const type = error instanceof Error ? error.type : '';
    return type === errorType;
};
export const isNoElementWithSelectorError = (error) => {
    return assertType(error, ErrorType.NO_ELEMENTS_WITH_SELECTOR);
};
export const isMultipleElementsWithSelectorError = (error) => {
    return assertType(error, ErrorType.MULTIPLE_ELEMENTS_WITH_SELECTOR);
};
