// @wix/unidriver-common/dist/esm/UniDriverBase/UniDriverBase.js
import { keyMeta, unidriverConfig } from '@wix/unidriver-core/internal';
import { UniDriverBaseCommon } from './UniDriverBaseCommon/UniDriverBaseCommon';
import { isMultipleElementsWithSelectorError, isNoElementWithSelectorError, MultipleElementsWithSelectorError, NoElementWithSelectorError, } from './errors';
export class UniDriverBase extends UniDriverBaseCommon {
    constructor() {
        super(...arguments);
        Object.defineProperty(this, "mouse", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: {
                hover: async () => {
                    await this.runWithinTimeBudget(async () => {
                        return this.api.mouse.hover(await this.getBaseElement());
                    });
                },
                leave: async () => {
                    await this.runWithinTimeBudget(async () => {
                        return this.api.mouse.leave(await this.getBaseElement());
                    });
                },
                moveTo: async (to) => {
                    await this.runWithinTimeBudget(async () => {
                        return this.api.mouse.moveTo(await this.getBaseElement(), to);
                    });
                },
                press: async () => {
                    await this.runWithinTimeBudget(async () => {
                        return this.api.mouse.press(await this.getBaseElement());
                    });
                },
                release: async () => {
                    await this.runWithinTimeBudget(async () => {
                        return this.api.mouse.release(await this.getBaseElement());
                    });
                },
            }
        });
        Object.defineProperty(this, "api", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        /** @deprecated use .unwrap() */
        Object.defineProperty(this, "getNative", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: async () => {
                return this.runWithinTimeBudget(async () => {
                    return this.api.getNative ? this.api.getNative(await this.getBaseElement()) : await this.getBaseElement();
                });
            }
        });
    }
    $(selector) {
        return this.s(unidriverConfig.queryByAttribute(selector));
    }
    $$(selector) {
        return this.ss(unidriverConfig.queryByAttribute(selector));
    }
    async attr(attr) {
        return this.runWithinTimeBudget(async () => {
            return this.api.attr(await this.getBaseElement(), attr);
        });
    }
    awaited(timeBudget = 10 /* run at least once */) {
        return new this.api.Class({
            base: this.ctx.base,
            parent: this.ctx.parent,
            selector: this.ctx.parent?.selector,
            timeBudget,
        });
    }
    async exists() {
        try {
            await this.getBaseElement();
            return true;
        }
        catch (e) {
            if (isMultipleElementsWithSelectorError(e)) {
                throw e;
            }
            else {
                return false;
            }
        }
    }
    async click(options) {
        return this.runWithinTimeBudget(async () => {
            return this.api.click(await this.getBaseElement(), options);
        });
    }
    async enterText(value, options) {
        await this.runWithinTimeBudget(async () => {
            return this.api.enterText(await this.getBaseElement(), value, {
                delay: 0,
                clear: true,
                ...(options ?? {}),
            });
        });
    }
    /** Internal API */
    getType() {
        return this.api.getType();
    }
    async hasClass(className) {
        return this.runWithinTimeBudget(async () => {
            return this.api.hasClass(await this.getBaseElement(), className);
        });
    }
    async pressKey(key) {
        await this.runWithinTimeBudget(async () => {
            const meta = keyMeta[key];
            return this.api.pressKey(await this.getBaseElement(), {
                key: meta.key,
                keyCode: 'keyCode' in meta ? Number(meta.keyCode) : undefined,
            });
        });
    }
    async prop(prop) {
        return this.runWithinTimeBudget(async () => {
            return this.api.prop(await this.getBaseElement(), prop);
        });
    }
    s(selector) {
        return new this.api.Class({
            base: async () => {
                try {
                    return await this.getElementBySelector(selector);
                }
                catch (e) {
                    if (isNoElementWithSelectorError(e)) {
                        throw new NoElementWithSelectorError({
                            failedPath: e.failedPath,
                            fullPath: [...(await this.selectorsPath), selector],
                            selector,
                        });
                    }
                    throw e;
                }
            },
            parent: this.ctx,
            selector,
        });
    }
    ss(selector) {
        return new this.api.ListClass({
            base: async () => {
                return this.api.getElementsBySelector(await this.getBaseElement(), selector);
            },
            parent: this.ctx,
            selector,
        });
    }
    async text() {
        return this.runWithinTimeBudget(async () => {
            return this.api.text(await this.getBaseElement());
        });
    }
    async unwrap() {
        return this.runWithinTimeBudget(async () => {
            return (await this.getBaseElement());
        });
    }
    async value() {
        return this.runWithinTimeBudget(async () => {
            return this.api.value(await this.getBaseElement());
        });
    }
    async getBaseElement() {
        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-call
        const base = await (typeof this.ctx.base === 'function' ? this.ctx.base() : this.ctx.base);
        if (!base) {
            throw new NoElementWithSelectorError({ selector: (await this.ctx.selector) ?? '' });
        }
        return base;
    }
    async getElementBySelector(selector) {
        const elements = await this.api.getElementsBySelector(await this.getBaseElement(), selector);
        if (!elements.length) {
            throw new NoElementWithSelectorError({ selector });
        }
        if (elements.length > 1) {
            throw new MultipleElementsWithSelectorError(elements.length, selector);
        }
        return elements[0];
    }
    async getElementsBySelector(selector) {
        const elements = await this.api.getElementsBySelector(await this.getBaseElement(), selector);
        if (elements.length === 0) {
            throw new NoElementWithSelectorError({ selector });
        }
        return elements;
    }
    // DEPRECATED SECTION
    /** @deprecated use .prop() */
    async _prop(prop) {
        return this.prop(prop);
    }
    /** @deprecated use .enterText() */
    async enterValue(value, options) {
        await this.runWithinTimeBudget(async () => {
            return this.api.enterValue(await this.getBaseElement(), value, {
                delay: 0,
                shouldClear: true,
                ...(options ?? {}),
            });
        });
    }
    /** @deprecated use .$() */
    async get(selector, options) {
        return this.runWithinTimeBudget(async () => {
            await this.getElementBySelector(selector);
            return new this.api.Class({
                base: () => this.getElementBySelector(selector),
                parent: this.ctx,
                selector,
            });
        }, options?.timeout);
    }
    /** @deprecated use .$$() */
    async getAll(selector, options) {
        return this.runWithinTimeBudget(async () => {
            await this.getElementsBySelector(selector);
            return new this.api.ListClass({
                base: () => this.getElementsBySelector(selector),
                parent: this.ctx,
                selector,
            });
        }, options?.timeout);
    }
    /** @deprecated use .mouse.hover() */
    async hover() {
        return this.mouse.hover();
    }
    /** @deprecated use .$/$$() */
    async wait(timeBudget) {
        return this.runWithinTimeBudget(async () => {
            const isExisting = await this.exists();
            if (!isExisting) {
                throw new Error('Element is not found');
            }
        }, timeBudget);
    }
}
