// @wix/ricos/dist/es/ricos-editor/toolbars/utils/biUtils.js
import { Version } from "./../../../ricos-content/index.js";
import {
  ADD_PLUGIN_LINK_BI,
  ANCHOR_CATEGORY,
  WEB_ADDRESS_CATEGORY
} from "./../../../wix-rich-content-common/index.js";
const getBiFunctions = (helpers, contentId) => ({
  onInlineToolbarOpen: (toolbarType) => helpers?.onInlineToolbarOpen?.({
    toolbarType,
    version: Version.currentVersion,
    contentId
  }),
  onToolbarButtonClick: (name, toolbarType, value, pluginId) => {
    helpers?.onToolbarButtonClick?.({
      buttonName: name,
      pluginId,
      type: toolbarType,
      value: value === void 0 ? void 0 : typeof value === "boolean" ? `${!value}` : value,
      version: Version.currentVersion,
      contentId
    });
    if (pluginId && value && typeof value === "string") {
      helpers?.onPluginAdd?.(pluginId, "FormattingToolbar", Version.currentVersion, contentId);
      helpers?.onPluginAddSuccess?.(
        pluginId,
        "FormattingToolbar",
        value,
        Version.currentVersion,
        contentId
      );
    }
  },
  onAddPluginLink: (data, plugin_id) => {
    const { url, anchor, target, rel } = data;
    const params = anchor ? { anchor, category: ANCHOR_CATEGORY } : {
      link: url,
      rel,
      newTab: target === "_blank",
      category: WEB_ADDRESS_CATEGORY
    };
    helpers.onPluginAction?.(ADD_PLUGIN_LINK_BI, {
      plugin_id,
      params,
      version: Version.currentVersion,
      contentId
    });
  }
});
export {
  getBiFunctions
};
