// @wix/patterns/dist/cjs/components/FormPageCard/FormPageCard.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.FormPageCard = void 0;
exports._FormPageCard = _FormPageCard;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireWildcard(require("react"));
var _providers = require("../../providers");
var _mobxReactLite = require("mobx-react-lite");
var _CardContainer = require("../CardContainer");
var _bexUtils = require("@wix/bex-utils");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/FormPageCard/FormPageCard.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _FormPageCard(props) {
  const {
    status: statusProp,
    id,
    registerScrollableItem,
    ...rest
  } = props;
  const pageState = (0, _providers.useFormPageContext)();
  const status = !pageState || pageState.isFetching ? 'loading' : statusProp;
  const [highlight, setHighlight] = (0, _react.useState)();
  const timeoutRef = (0, _react.useRef)();
  (0, _react.useEffect)(() => {
    return () => {
      window.clearTimeout(timeoutRef.current);
    };
  });
  return /*#__PURE__*/_react.default.createElement("div", {
    id: id,
    ref: ref => {
      if (id && ref) {
        registerScrollableItem({
          ref,
          itemName: id,
          containerName: id,
          afterScroll: () => {
            setHighlight('focus');
            timeoutRef.current = window.setTimeout(() => {
              setHighlight(undefined);
            }, 3000);
          }
        });
      }
    },
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 23,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(_CardContainer.CardContainer, (0, _extends2.default)({
    status: status
  }, rest, {
    highlight: highlight,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 42,
      columnNumber: 7
    }
  })));
}
_FormPageCard.displayName = 'FormPageCard';
const FormPageCard = exports.FormPageCard = (0, _bexUtils.withFocusable)((0, _mobxReactLite.observer)(_FormPageCard));
