// @wix/auto_sdk_restaurants_order-pacing-order/build/es/index.mjs
// src/restaurants-order-pacing-v1-order-order-pacing-order.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/restaurants-order-pacing-v1-order-order-pacing-order.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressRestaurantsOrderPacingOrderPacingServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-order-pacing-pacer",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-order-pacing-pacer",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-order-pacing-pacer",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/order-pacing-pacer",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/v1/order-pacing-pacer",
        destPath: "/v1"
      },
      {
        srcPath: "/restaurants/order-pacing-pacer/v1",
        destPath: "/v1"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-order-pacing-pacer",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/order-pacing-pacer/v1",
        destPath: "/v1"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/order-pacing-pacer/v1",
        destPath: "/v1"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/order-pacing-pacer/v1",
        destPath: "/v1"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_order-pacing-order";
function getCurrentDelay(payload) {
  function __getCurrentDelay({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.order_pacing.v1.order",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.order.pacing.OrderPacingService.GetCurrentDelay",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsOrderPacingOrderPacingServiceUrl({
        protoPath: "/v1/current-delay",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getCurrentDelay;
}
function getCurrentDelayWithValidityPeriod(payload) {
  function __getCurrentDelayWithValidityPeriod({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.order_pacing.v1.order",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.order.pacing.OrderPacingService.GetCurrentDelayWithValidityPeriod",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsOrderPacingOrderPacingServiceUrl({
        protoPath: "/v1/current-delay-with-validity-period",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getCurrentDelayWithValidityPeriod;
}
function listBusyBlocks(payload) {
  function __listBusyBlocks({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "from" }, { path: "until" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.order_pacing.v1.order",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.order.pacing.OrderPacingService.ListBusyBlocks",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsOrderPacingOrderPacingServiceUrl({
        protoPath: "/v1/busy-blocks",
        data: serializedData,
        host
      }),
      params: toURLSearchParams(serializedData),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "busyBlocks.startTime" },
            { path: "busyBlocks.endTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listBusyBlocks;
}

// src/restaurants-order-pacing-v1-order-order-pacing-order.universal.ts
async function getCurrentDelay2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    locationId: options?.locationId,
    maxAcceptableDelayTimeInMinutes: options?.maxAcceptableDelayTimeInMinutes
  });
  const reqOpts = getCurrentDelay(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          locationId: "$[0].locationId",
          maxAcceptableDelayTimeInMinutes: "$[0].maxAcceptableDelayTimeInMinutes"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getCurrentDelayWithValidityPeriod2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    locationId: options?.locationId,
    maxAcceptableDelayTimeInMinutes: options?.maxAcceptableDelayTimeInMinutes
  });
  const reqOpts = getCurrentDelayWithValidityPeriod(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          locationId: "$[0].locationId",
          maxAcceptableDelayTimeInMinutes: "$[0].maxAcceptableDelayTimeInMinutes"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listBusyBlocks2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    locationId: options?.locationId,
    from: options?.from,
    until: options?.until,
    orderFulfillmentTimeInMinutes: options?.orderFulfillmentTimeInMinutes
  });
  const reqOpts = listBusyBlocks(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          locationId: "$[0].locationId",
          from: "$[0].from",
          until: "$[0].until",
          orderFulfillmentTimeInMinutes: "$[0].orderFulfillmentTimeInMinutes"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/restaurants-order-pacing-v1-order-order-pacing-order.public.ts
function getCurrentDelay3(httpClient) {
  return (options) => getCurrentDelay2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function getCurrentDelayWithValidityPeriod3(httpClient) {
  return (options) => getCurrentDelayWithValidityPeriod2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function listBusyBlocks3(httpClient) {
  return (options) => listBusyBlocks2(
    options,
    // @ts-ignore
    { httpClient }
  );
}

// src/restaurants-order-pacing-v1-order-order-pacing-order.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
var getCurrentDelay4 = /* @__PURE__ */ createRESTModule(getCurrentDelay3);
var getCurrentDelayWithValidityPeriod4 = /* @__PURE__ */ createRESTModule(getCurrentDelayWithValidityPeriod3);
var listBusyBlocks4 = /* @__PURE__ */ createRESTModule(listBusyBlocks3);
export {
  getCurrentDelay4 as getCurrentDelay,
  getCurrentDelayWithValidityPeriod4 as getCurrentDelayWithValidityPeriod,
  listBusyBlocks4 as listBusyBlocks
};
