// @wix/wix-data-items-common/dist/esm/api/expressions/Expression.js
export class FieldExpressionImpl {
    constructor(fieldPath) {
        this.fieldPath = fieldPath;
    }
    /** @private */
    build() {
        return { fieldPath: this.fieldPath };
    }
}
export class TextExpressionImpl {
    constructor(value) {
        this.value = value;
    }
    /** @private */
    build() {
        return { text: this.value };
    }
}
export class NumericExpressionImpl {
    constructor(value) {
        this.value = value;
    }
    /** @private */
    build() {
        return { numeric: this.value };
    }
}
export class AddExpressionImpl {
    constructor(expressions) {
        this.expressions = expressions;
    }
    /** @private */
    build() {
        return {
            add: {
                expressions: this.expressions.map((e) => e.build()),
            },
        };
    }
}
export class SubtractExpressionImpl {
    constructor(firstExpression, secondExpression) {
        this.firstExpression = firstExpression;
        this.secondExpression = secondExpression;
    }
    /** @private */
    build() {
        return {
            subtract: {
                firstExpression: this.firstExpression.build(),
                secondExpression: this.secondExpression.build(),
            },
        };
    }
}
export class MultiplyExpressionImpl {
    constructor(expressions) {
        this.expressions = expressions;
    }
    /** @private */
    build() {
        return {
            multiply: {
                expressions: this.expressions.map((e) => e.build()),
            },
        };
    }
}
export class DivideExpressionImpl {
    constructor(firstExpression, secondExpression) {
        this.firstExpression = firstExpression;
        this.secondExpression = secondExpression;
    }
    /** @private */
    build() {
        return {
            divide: {
                firstExpression: this.firstExpression.build(),
                secondExpression: this.secondExpression.build(),
            },
        };
    }
}
export class SumExpressionImpl {
    constructor(expressions) {
        this.expressions = expressions;
    }
    /** @private */
    build() {
        return {
            sum: {
                expressions: this.expressions.map((e) => e.build()),
            },
        };
    }
}
export class AbsExpressionImpl {
    constructor(expression) {
        this.expression = expression;
    }
    /** @private */
    build() {
        return {
            abs: {
                expression: this.expression.build(),
            },
        };
    }
}
export class ModExpressionImpl {
    constructor(firstExpression, secondExpression) {
        this.firstExpression = firstExpression;
        this.secondExpression = secondExpression;
    }
    /** @private */
    build() {
        return {
            mod: {
                firstExpression: this.firstExpression.build(),
                secondExpression: this.secondExpression.build(),
            },
        };
    }
}
export class FloorExpressionImpl {
    constructor(expression) {
        this.expression = expression;
    }
    /** @private */
    build() {
        return {
            floor: {
                expression: this.expression.build(),
            },
        };
    }
}
export class CeilExpressionImpl {
    constructor(expression) {
        this.expression = expression;
    }
    /** @private */
    build() {
        return {
            ceil: {
                expression: this.expression.build(),
            },
        };
    }
}
export class ConcatExpressionImpl {
    constructor(expressions) {
        this.expressions = expressions;
    }
    /** @private */
    build() {
        return {
            concat: {
                expressions: this.expressions.map((e) => e.build()),
            },
        };
    }
}
export class StringifyExpressionImpl {
    constructor(expression) {
        this.expression = expression;
    }
    /** @private */
    build() {
        return {
            stringify: {
                expression: this.expression.build(),
            },
        };
    }
}
export class ToLowerExpressionImpl {
    constructor(expression) {
        this.expression = expression;
    }
    /** @private */
    build() {
        return {
            toLower: {
                expression: this.expression.build(),
            },
        };
    }
}
export class ToUpperExpressionImpl {
    constructor(expression) {
        this.expression = expression;
    }
    /** @private */
    build() {
        return {
            toUpper: {
                expression: this.expression.build(),
            },
        };
    }
}
export class SubstringExpressionImpl {
    constructor(expression, startExpression, lengthExpression) {
        this.expression = expression;
        this.startExpression = startExpression;
        this.lengthExpression = lengthExpression;
    }
    /** @private */
    build() {
        const proto = {
            substring: {
                expression: this.expression.build(),
                startExpression: this.startExpression.build(),
            },
        };
        if (this.lengthExpression) {
            proto.substring.lengthExpression = this.lengthExpression.build();
        }
        return proto;
    }
}
export class LengthExpressionImpl {
    constructor(expression) {
        this.expression = expression;
    }
    /** @private */
    build() {
        return {
            length: { expression: this.expression.build() },
        };
    }
}
