// @wix/ricos/dist/cjs/plugin-figure/tiptap/caption-tiptap-extension.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var caption_tiptap_extension_exports = {};
__export(caption_tiptap_extension_exports, {
  captionTiptapExtension: () => captionTiptapExtension
});
module.exports = __toCommonJS(caption_tiptap_extension_exports);
var import_ricos_content = require("./../../ricos-content/index.js");
var import_plugin_commons = require("./../../plugin-commons/index.js");
var import_caption = require("./caption.js");
const captionTiptapExtension = (0, import_plugin_commons.createRicosNodeExtension)({
  type: "react-node",
  groups: ["not-overtypable", "custom-styles"],
  name: import_ricos_content.TIPTAP_CAPTION_TYPE,
  Component: import_caption.Caption,
  placeholder: {
    predicate: ({ doc, pos, selection }) => {
      const $pos = doc.resolve(pos);
      const parentNode = $pos.parent;
      if (parentNode.type.name !== import_ricos_content.TIPTAP_FIGURE_TYPE) {
        return false;
      }
      const isCurrNodeSelected = selection.from <= pos && selection.to >= pos;
      const isCaptionFocused = selection.from === selection.to;
      return isCurrNodeSelected && !isCaptionFocused;
    },
    content: "Figure_Caption_Input_Placeholder",
    alwaysVisible: true,
    alignCenter: true
  },
  createExtensionConfig() {
    return {
      name: this.name,
      group: "figureSecondChildContent",
      content: "text*",
      isolating: true,
      draggable: false,
      selectable: true,
      addAttributes() {
        return {
          textStyle: {
            default: {
              textAlignment: "CENTER",
              lineHeight: 1.5
            }
          },
          // Using `styleDefaults` instead of a `style` object because
          // Tiptap requires the default value of `style` to be a string.
          // For more context, see:
          // https://github.com/ueberdosis/tiptap/blob/v2.8.0/packages/core/src/utilities/mergeAttributes.ts#L26
          styleDefaults: {
            default: {
              paddingTop: "16px",
              paddingBottom: "16px"
            }
          }
        };
      },
      parseHTML() {
        return [
          {
            tag: "figcaption"
          }
        ];
      },
      renderHTML() {
        return ["figcaption", {}, 0];
      }
    };
  }
});
