// @wix/ricos/dist/cjs/plugin-figure/tiptap/caption.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var caption_exports = {};
__export(caption_exports, {
  Caption: () => Caption
});
module.exports = __toCommonJS(caption_exports);
var import_classnames = __toESM(require("classnames"));
var import_react = __toESM(require("react"));
var import_context = require("./../../pm-converters/context.js");
var import_ricos_content = require("./../../ricos-content/index.js");
var import_ricos_context = require("./../../ricos-context/index.js");
var import_ricos_types = require("./../../ricos-types/index.js");
var import_is_node_selected = require("./../../_shared/is-node-selected.js");
var import_line_spacing_utils = require("./../../_shared/line-spacing-utils.js");
var import_node_styles = require("./../../_shared/node-styles.js");
var import_data_hooks = require("../data-hooks.js");
var import_caption_scss = __toESM(require("./caption-scss.js"));
var import_use_editor_selection = require("./use-editor-selection.js");
const Caption = ({ NodeViewContent, editor, node }) => {
  const { isMobile } = (0, import_react.useContext)(import_ricos_context.RicosContext);
  const pluginsEvents = (0, import_react.useContext)(import_ricos_context.PluginsEventsContext);
  const pluginExports = (0, import_ricos_context.usePluginExportsContext)();
  const selectedNodes = (0, import_use_editor_selection.useEditorSelection)(editor).selectedNodes;
  const captionNodePos = editor.$doc.querySelector(import_ricos_content.TIPTAP_CAPTION_TYPE, {
    id: node.attrs.id
  });
  const converters = (0, import_context.useConverters)();
  if (!captionNodePos?.parent?.node) {
    console.error("Caption must have a parent.");
    return null;
  }
  const ricosParentNode = converters.fromPmNode(captionNodePos.parent.node.toJSON());
  const AiButton = pluginExports.aiIntegrations?.image?.captionTrigger;
  const isCaptionSelected = selectedNodes.find(
    (selectedNode) => selectedNode.attrs.id === node.attrs?.id
  );
  const isParentSelected = selectedNodes.find(
    (selectedNode) => selectedNode.attrs.id === ricosParentNode?.id
  );
  (0, import_react.useEffect)(() => {
    if (node.content.firstChild?.text || isCaptionSelected || isParentSelected) {
      return;
    }
    pluginsEvents?.publishPluginLayoutShift({ nodeId: node.attrs.id });
  }, [isCaptionSelected, isParentSelected]);
  const isVisible = isCaptionSelected || isParentSelected || !!node.content.firstChild;
  const hasAiButton = isCaptionSelected && ricosParentNode?.type === import_ricos_types.Node_Type.IMAGE && AiButton;
  const isFocused = (0, import_is_node_selected.isNodeSelected)(editor.state, node.attrs.id);
  const className = (0, import_classnames.default)(
    import_caption_scss.default.caption,
    (0, import_node_styles.getPluginContainerClassNames)(isMobile),
    (0, import_node_styles.getPluginFocusClassNames)(isFocused)
  );
  return /* @__PURE__ */ import_react.default.createElement(
    "div",
    {
      className,
      style: isVisible ? void 0 : {
        display: "none"
      }
    },
    hasAiButton ? /* @__PURE__ */ import_react.default.createElement("div", { className: import_caption_scss.default.aiButtonContainer }, /* @__PURE__ */ import_react.default.createElement(
      AiButton,
      {
        node: ricosParentNode,
        onCaptionChange: (caption) => {
          const { range } = captionNodePos;
          const { tr } = editor.state;
          tr.insertText(caption, range.from, range.to);
          editor.view.dispatch(tr);
        }
      }
    )) : null,
    /* @__PURE__ */ import_react.default.createElement(
      NodeViewContent,
      {
        dir: "auto",
        "data-hook": import_data_hooks.CAPTION,
        as: "figcaption",
        style: {
          width: "100%",
          overflow: "auto",
          ...node.attrs?.textStyle?.lineHeight && {
            lineHeight: (0, import_line_spacing_utils.normalizeLineHeight)(node.attrs?.textStyle?.lineHeight)
          },
          textAlign: node.attrs?.textStyle?.textAlignment,
          paddingTop: node.attrs?.style?.paddingTop ?? node.attrs?.styleDefaults?.paddingTop,
          paddingBottom: node.attrs?.style?.paddingBottom ?? node.attrs?.styleDefaults?.paddingBottom
        }
      }
    )
  );
};
