// @wix/auto_sdk_seo_seo-sitemap-entries/build/es/meta.mjs
// src/promote-seo-v1-headless-sitemap-entry-seo-sitemap-entries.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixPromoteSeoHeadlessSitemapServiceV1SeoHeadlessSitemapServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/seo-headless-sitemap/v1/list-static-sitemap-pages",
        destPath: "/v1/list-static-sitemap-pages"
      }
    ],
    _: [
      {
        srcPath: "/_api/seo-headless-sitemap/v1/list-sitemap-pages",
        destPath: "/v1/list-sitemap-pages"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/seo-headless-sitemap/v1/list-sitemap-pages",
        destPath: "/v1/list-sitemap-pages"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/v1/list-sitemap-pages",
        destPath: "/v1/list-sitemap-pages"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_seo_seo-sitemap-entries";
function listSitemapPages(payload) {
  function __listSitemapPages({ host }) {
    const metadata = {
      entityFqdn: "wix.promote.seo.v1.headless_sitemap_entry",
      method: "GET",
      methodFqn: "wix.promote.seo.headless.sitemap.service.v1.SeoHeadlessSitemapService.ListSitemapPages",
      packageName: PACKAGE_NAME,
      url: resolveWixPromoteSeoHeadlessSitemapServiceV1SeoHeadlessSitemapServiceUrl(
        { protoPath: "/v1/list-sitemap-pages", data: payload, host }
      ),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __listSitemapPages;
}

// src/promote-seo-v1-headless-sitemap-entry-seo-sitemap-entries.meta.ts
function listSitemapPages2() {
  const payload = {};
  const getRequestOptions = listSitemapPages(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/list-sitemap-pages",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  listSitemapPages2 as listSitemapPages
};
