// @wix/patterns/dist/cjs/components/ActionControl/ActionButton.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.ActionButton = ActionButton;
var _designSystem = require("@wix/design-system");
var _react = _interopRequireDefault(require("react"));
var _DisabledTooltip = require("../DisabledTooltip");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/ActionControl/ActionButton.tsx";
function ActionButton(props) {
  const {
    actionItem,
    suffixIcon,
    size,
    priority,
    fullWidth
  } = props;
  const {
    tooltip,
    prefixIcon,
    label,
    onClick,
    dataHook,
    disabled
  } = actionItem;
  return /*#__PURE__*/_react.default.createElement(_DisabledTooltip.DisabledTooltip, {
    content: tooltip,
    dataHook: `${dataHook}-tooltip`,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 20,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.Button, {
    skin: "standard",
    priority: priority,
    size: size,
    prefixIcon: prefixIcon,
    ellipsis: true,
    onClick: onClick,
    dataHook: dataHook,
    disabled: disabled,
    suffixIcon: suffixIcon,
    fullWidth: fullWidth,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 21,
      columnNumber: 7
    }
  }, label));
}
