// @wix/patterns/dist/cjs/components/ActionControl/ActionPopover.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.ActionPopover = ActionPopover;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _designSystem = require("@wix/design-system");
var _react = _interopRequireWildcard(require("react"));
var _wixUiIconsCommon = require("@wix/wix-ui-icons-common");
var _DisabledTooltip = require("../DisabledTooltip");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/ActionControl/ActionPopover.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function ActionPopover(props) {
  const {
    actionItem,
    size,
    priority,
    onPopoverClick,
    fullWidth
  } = props;
  const {
    dataHook,
    subItems,
    popoverMenuProps
  } = actionItem;
  const allActions = (0, _react.useMemo)(() => {
    return subItems == null ? void 0 : subItems.filter(subItemGroup => subItemGroup == null ? void 0 : subItemGroup.length).reduce((result, group, index, array) => {
      return [...result, ...group.map(item => {
        return /*#__PURE__*/_react.default.createElement(_designSystem.PopoverMenu.MenuItem, {
          prefixIcon: item.prefixIcon,
          disabled: item.disabled,
          text: item.label,
          key: item.label,
          subtitle: item.subtitle,
          dataHook: item.dataHook,
          onClick: item.onClick,
          __self: this,
          __source: {
            fileName: _jsxFileName,
            lineNumber: 25,
            columnNumber: 15
          }
        });
      }), ...(index !== array.length - 1 ? [/*#__PURE__*/_react.default.createElement(_designSystem.PopoverMenu.Divider, {
        key: index,
        __self: this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 37,
          columnNumber: 16
        }
      })] : [])];
    }, []);
  }, [subItems]);
  return /*#__PURE__*/_react.default.createElement(_designSystem.PopoverMenu, (0, _extends2.default)({}, popoverMenuProps, {
    dataHook: dataHook ? `${dataHook}-popover` : undefined,
    triggerElement: ({
      toggle
    }) => /*#__PURE__*/_react.default.createElement(_DisabledTooltip.DisabledTooltip, {
      content: actionItem.tooltip,
      dataHook: dataHook ? `${dataHook}-tooltip` : undefined,
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 48,
        columnNumber: 9
      }
    }, /*#__PURE__*/_react.default.createElement(_designSystem.Button, {
      skin: "standard",
      priority: priority,
      size: size,
      fullWidth: fullWidth,
      dataHook: actionItem.dataHook,
      onClick: () => {
        onPopoverClick == null || onPopoverClick();
        toggle();
      },
      ellipsis: true,
      prefixIcon: actionItem.prefixIcon,
      disabled: actionItem.disabled,
      suffixIcon: /*#__PURE__*/_react.default.createElement(_wixUiIconsCommon.ChevronDown, {
        __self: this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 65,
          columnNumber: 25
        }
      }),
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 52,
        columnNumber: 11
      }
    }, actionItem.label)),
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 44,
      columnNumber: 5
    }
  }), allActions);
}
