// @wix/bex-core/dist/esm/state/CustomFilterState/CustomFilterState.js
import { EventEmitter } from 'events';
import { action, computed, makeObservable, observable } from 'mobx';
import { refreshFilter } from '../Filter/refreshFilter';
export class CustomFilterState {
    constructor(params) {
        this.events = new EventEmitter();
        this.name = params.name ?? '';
        this.persistent = params.persistent;
        this.defaultValue = params.defaultValue;
        this._value = params.initialValue ?? params.defaultValue;
        this._isEmpty = params.isEmpty;
        this._size = params.size ?? ((value) => (this._isEmpty(value) ? 0 : 1));
        this._toQueryString = params.toQueryString;
        this._fromQueryString = params.fromQueryString;
        this._toArray = params.toArray;
        this._remove = params.remove ?? (() => this.defaultValue);
        this._hasDiff = params.hasDiff ?? ((data) => data !== this.value);
        this.itemKey = params.itemKey;
        this.itemName = params.itemName;
        this.decode = params.decode;
        this.encode = params.encode;
        this.matches = params.matches ?? (() => false);
        this.equals = params.equals;
        makeObservable(this, {
            _value: observable.ref,
            persistent: observable.ref,
            isEmpty: computed,
            size: computed,
            value: computed,
            reset: action,
            setValue: action,
            changeValue: action,
            scheduleRefresh: action,
            clone: action,
        });
    }
    get value() {
        return this._value;
    }
    get toArray() {
        return this._toArray(this._value);
    }
    get isEmpty() {
        return this._isEmpty(this.value);
    }
    get size() {
        return this._size(this.value);
    }
    setValue(value, { emitEvents, ...options } = {}) {
        this._value = value;
        if (emitEvents) {
            for (const event of emitEvents) {
                this.events.emit(event, options);
            }
        }
    }
    changeValue(value, { emitEvents = [], ...options } = {}) {
        this.setValue(value, { ...options, emitEvents: ['change', ...emitEvents] });
    }
    refresh(value, options) {
        refreshFilter(this, value, options);
    }
    scheduleRefresh(value) {
        this.changeValue(value, { emitEvents: ['scheduleRefresh'] });
    }
    get toQueryString() {
        const { _toQueryString, encode, value } = this;
        if (_toQueryString) {
            return _toQueryString(value);
        }
        return JSON.stringify(encode(value));
    }
    applyFromQueryString(str) {
        const { _fromQueryString, decode } = this;
        if (_fromQueryString) {
            this.setValue(_fromQueryString(str));
        }
        else {
            try {
                const raw = JSON.parse(str);
                this.setValue(decode(raw));
            }
            catch (e) { }
        }
    }
    reset(options = {}) {
        this.setValue(this.defaultValue, options);
    }
    remove(items, options = {}) {
        this.setValue(this._remove(items, this.value), options);
    }
    hasDiff(data) {
        return this._hasDiff(data);
    }
    clone(params) {
        const { itemName, itemKey, encode, decode, _isEmpty: isEmpty, _remove: remove, _size: size, _fromQueryString: fromQueryString, _toArray: toArray, _toQueryString: toQueryString, defaultValue, value: initialValue, matches, persistent, name, } = this;
        return new CustomFilterState({
            itemName,
            itemKey,
            encode,
            decode,
            isEmpty,
            remove,
            size,
            fromQueryString,
            toArray,
            toQueryString,
            defaultValue,
            initialValue,
            matches,
            persistent,
            name,
            ...params,
        });
    }
}
export const customFilter = (params) => new CustomFilterState(params);
