// @stylable/uni-driver/cjs/uni-driver.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const stateMiddleDelimiter = '-';
class StylableUnidriverUtil {
    constructor(stylesheet) {
        this.stylesheet = stylesheet;
    }
    async hasStyleState(base, stateName, param = true) {
        const stateClass = this.stylesheet.cssStates({ [stateName]: param });
        return base.hasClass(stateClass);
    }
    async getStyleState(base, stateName) {
        const className = (await base.attr('class')) || '';
        if (!className.includes(stateName)) {
            return null;
        }
        const classList = className.trim().split(/\s+/);
        const booleanState = this.stylesheet.cssStates({ [stateName]: true });
        if (classList.includes(booleanState)) {
            return true;
        }
        const baseState = this.getBaseStateWithParam(stateName);
        let paramValue = '';
        classList.forEach(cls => {
            if (!paramValue) {
                paramValue = this.getStateValueFromClassName(cls, baseState);
            }
        });
        return paramValue ? paramValue : null;
    }
    getStateValueFromClassName(cls, baseState) {
        if (cls.startsWith(baseState)) {
            const param = cls.slice(baseState.length);
            const paramIndex = param.indexOf(stateMiddleDelimiter);
            if (paramIndex !== -1) {
                return param.slice(paramIndex + 1);
            }
        }
        return '';
    }
    getBaseStateWithParam(stateName) {
        const singleCharState = 'x';
        return this.stylesheet.cssStates({ [stateName]: singleCharState }).slice(0, -3);
    }
}
exports.StylableUnidriverUtil = StylableUnidriverUtil;
