// @wix/bex-utils/dist/esm/FocusableWrapper/FocusableService.js
export class FocusableItemsService {
    constructor() {
        this.getItemRef = (itemName) => this.focusableItems[itemName] && this.focusableItems[itemName].ref;
        this.getItemAfterScroll = (itemName) => this.focusableItems[itemName] && this.focusableItems[itemName].afterScroll;
        this.getContainerRefByItem = (itemName) => {
            const containerName = this.focusableItems[itemName]?.containerName;
            const container = this.scrollableContainers[containerName];
            return container?.ref;
        };
        this.getContainerRefByName = (containerName) => {
            const container = this.scrollableContainers[containerName];
            return container?.ref;
        };
        this.focusableItems = {};
        this.scrollableContainers = {};
    }
    addItem({ ref, itemName, containerName, afterScroll }) {
        this.focusableItems[itemName] = {
            itemName,
            ref,
            containerName,
            afterScroll,
        };
    }
    addContainer({ containerName, ref }) {
        this.scrollableContainers[containerName] = {
            containerName,
            ref,
        };
    }
}
