// @wix/bex-utils/dist/esm/FocusableWrapper/FocusableWrapper.js
import React, { useCallback, useEffect, useMemo } from 'react';
import { FocusableItemsService } from './FocusableService';
import { scrollAndFocus } from './scrollAndFocus';
export const getAnchorParam = () => {
    const searchParams = new URLSearchParams(window.location.search);
    return searchParams.get('anchor');
};
export const FocusableContext = React.createContext(null);
export const FocusableWrapper = ({ scrollableRef, getScrollableTrigger, getDoesItemExist, children, }) => {
    const focusableItemsService = useMemo(() => new FocusableItemsService(), []);
    const scrollIfNeeded = useCallback(() => {
        const focusedItemName = getAnchorParam();
        scrollByItemName(focusedItemName);
    }, [scrollableRef]);
    const scrollByItemName = useCallback((focusedItemName) => {
        if (!focusedItemName) {
            return;
        }
        const containerRefByItem = focusableItemsService.getContainerRefByItem(focusedItemName);
        if (containerRefByItem) {
            const focusedItemRef = focusableItemsService.getItemRef(focusedItemName);
            const afterScroll = focusableItemsService.getItemAfterScroll(focusedItemName);
            return scrollAndFocus({
                scrollableRef,
                containerRef: containerRefByItem,
                focusedItemRef,
                afterScroll,
            });
        }
        const containerRef = focusableItemsService.getContainerRefByName(focusedItemName);
        if (containerRef) {
            scrollAndFocus({
                scrollableRef,
                containerRef,
                focusedItemRef: containerRef,
            });
        }
    }, [focusableItemsService, scrollableRef]);
    const doesItemExist = useCallback(() => {
        const focusedItemName = getAnchorParam();
        if (!focusedItemName) {
            return false;
        }
        return Boolean(focusableItemsService.getContainerRefByItem(focusedItemName));
    }, [focusableItemsService]);
    useEffect(() => {
        getScrollableTrigger(scrollIfNeeded);
        getDoesItemExist(doesItemExist);
    }, [getScrollableTrigger, getDoesItemExist, scrollIfNeeded, doesItemExist]);
    const contextValue = useMemo(() => ({
        focusable: {
            registerScrollableItem: (scrollableItem) => {
                focusableItemsService.addContainer(scrollableItem);
                focusableItemsService.addItem(scrollableItem);
            },
        },
    }), [focusableItemsService]);
    return (React.createElement(FocusableContext.Provider, { value: contextValue }, children));
};
