// @wix/auto_sdk_restaurants_restaurant-locations/build/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  DayOfWeek: () => DayOfWeek,
  SubdivisionType: () => SubdivisionType,
  getRestaurantLocation: () => getRestaurantLocation4,
  listRestaurantLocations: () => listRestaurantLocations4
});
module.exports = __toCommonJS(index_exports);

// src/restaurants-restaurantlocations-v3-restaurant-location-restaurant-locations.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/restaurants-restaurantlocations-v3-restaurant-location-restaurant-locations.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressRestaurantsRestaurantlocationsV3RestaurantLocationsServiceUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      },
      {
        srcPath: "/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      },
      {
        srcPath: "/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      }
    ],
    _base_domain_: [
      {
        srcPath: "/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      }
    ],
    "users._base_domain_": [
      {
        srcPath: "/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurant-locations",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_restaurant-locations";
function getRestaurantLocation(payload) {
  function __getRestaurantLocation({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.restaurantlocations.v3.restaurant_location",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.restaurantlocations.v3.RestaurantLocationsService.GetRestaurantLocation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsRestaurantlocationsV3RestaurantLocationsServiceUrl(
        {
          protoPath: "/v3/restaurant-locations/{restaurantLocationId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [
            { path: "restaurantLocation.location.address.geocode.latitude" },
            { path: "restaurantLocation.location.address.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getRestaurantLocation;
}
function listRestaurantLocations(payload) {
  function __listRestaurantLocations({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.restaurantlocations.v3.restaurant_location",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.restaurantlocations.v3.RestaurantLocationsService.ListRestaurantLocations",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsRestaurantlocationsV3RestaurantLocationsServiceUrl(
        { protoPath: "/v3/restaurant-locations", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [
            { path: "restaurantLocations.location.address.geocode.latitude" },
            {
              path: "restaurantLocations.location.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listRestaurantLocations;
}

// src/restaurants-restaurantlocations-v3-restaurant-location-restaurant-locations.universal.ts
var import_address = require("@wix/sdk-runtime/transformations/address");
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var SubdivisionType = /* @__PURE__ */ ((SubdivisionType2) => {
  SubdivisionType2["UNKNOWN_SUBDIVISION_TYPE"] = "UNKNOWN_SUBDIVISION_TYPE";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_1"] = "ADMINISTRATIVE_AREA_LEVEL_1";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_2"] = "ADMINISTRATIVE_AREA_LEVEL_2";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_3"] = "ADMINISTRATIVE_AREA_LEVEL_3";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_4"] = "ADMINISTRATIVE_AREA_LEVEL_4";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_5"] = "ADMINISTRATIVE_AREA_LEVEL_5";
  SubdivisionType2["COUNTRY"] = "COUNTRY";
  return SubdivisionType2;
})(SubdivisionType || {});
var DayOfWeek = /* @__PURE__ */ ((DayOfWeek2) => {
  DayOfWeek2["MON"] = "MON";
  DayOfWeek2["TUE"] = "TUE";
  DayOfWeek2["WED"] = "WED";
  DayOfWeek2["THU"] = "THU";
  DayOfWeek2["FRI"] = "FRI";
  DayOfWeek2["SAT"] = "SAT";
  DayOfWeek2["SUN"] = "SUN";
  return DayOfWeek2;
})(DayOfWeek || {});
async function getRestaurantLocation2(restaurantLocationId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    restaurantLocationId
  });
  const reqOpts = getRestaurantLocation(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_address.transformRESTAddressToSDKAddress,
          paths: [{ path: "restaurantLocation.location.address" }]
        }
      ])
    )?.restaurantLocation;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { restaurantLocationId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["restaurantLocationId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listRestaurantLocations2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    locationIds: options?.locationIds,
    includeArchived: options?.includeArchived,
    default: options?.default
  });
  const reqOpts = listRestaurantLocations(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_address.transformRESTAddressToSDKAddress,
          paths: [{ path: "restaurantLocations.location.address" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          locationIds: "$[0].locationIds",
          includeArchived: "$[0].includeArchived",
          default: "$[0].default"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/restaurants-restaurantlocations-v3-restaurant-location-restaurant-locations.public.ts
function getRestaurantLocation3(httpClient) {
  return (restaurantLocationId) => getRestaurantLocation2(
    restaurantLocationId,
    // @ts-ignore
    { httpClient }
  );
}
function listRestaurantLocations3(httpClient) {
  return (options) => listRestaurantLocations2(
    options,
    // @ts-ignore
    { httpClient }
  );
}

// src/restaurants-restaurantlocations-v3-restaurant-location-restaurant-locations.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var getRestaurantLocation4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getRestaurantLocation3);
var listRestaurantLocations4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(listRestaurantLocations3);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  DayOfWeek,
  SubdivisionType,
  getRestaurantLocation,
  listRestaurantLocations
});
