// @wix/events/build/internal/cjs/meta.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  categories: () => categories,
  eventsSettings: () => eventsSettings,
  forms: () => forms,
  guests: () => guests,
  notifications: () => notifications,
  orders: () => orders,
  policies: () => policies,
  rsvp: () => rsvp,
  rsvpV2: () => rsvpV2,
  schedule: () => schedule,
  scheduleBookmarks: () => scheduleBookmarks,
  staffMembers: () => staffMembers,
  ticketDefinitions: () => ticketDefinitions,
  ticketDefinitionsV2: () => ticketDefinitionsV2,
  ticketReservations: () => ticketReservations,
  tickets: () => tickets,
  wixEventsV2: () => wixEventsV2
});
module.exports = __toCommonJS(meta_exports);
var guests = __toESM(require("@wix/auto_sdk_events_guests/meta"));
var notifications = __toESM(require("@wix/auto_sdk_events_notifications/meta"));
var scheduleBookmarks = __toESM(require("@wix/auto_sdk_events_schedule-bookmarks/meta"));
var schedule = __toESM(require("@wix/auto_sdk_events_schedule/meta"));
var staffMembers = __toESM(require("@wix/auto_sdk_events_staff-members/meta"));
var categories = __toESM(require("@wix/auto_sdk_events_categories/meta"));
var eventsSettings = __toESM(require("@wix/auto_sdk_events_events-settings/meta"));
var forms = __toESM(require("@wix/auto_sdk_events_forms/meta"));
var orders = __toESM(require("@wix/auto_sdk_events_orders/meta"));
var rsvp = __toESM(require("@wix/auto_sdk_events_rsvp/meta"));
var tickets = __toESM(require("@wix/auto_sdk_events_tickets/meta"));
var ticketDefinitions = __toESM(require("@wix/auto_sdk_events_ticket-definitions/meta"));
var ticketReservations = __toESM(require("@wix/auto_sdk_events_ticket-reservations/meta"));
var policies = __toESM(require("@wix/auto_sdk_events_policies/meta"));
var rsvpV2 = __toESM(require("@wix/auto_sdk_events_rsvp-v-2/meta"));
var wixEventsV2 = __toESM(require("@wix/auto_sdk_events_wix-events-v-2/meta"));
var ticketDefinitionsV2 = __toESM(require("@wix/auto_sdk_events_ticket-definitions-v-2/meta"));
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  categories,
  eventsSettings,
  forms,
  guests,
  notifications,
  orders,
  policies,
  rsvp,
  rsvpV2,
  schedule,
  scheduleBookmarks,
  staffMembers,
  ticketDefinitions,
  ticketDefinitionsV2,
  ticketReservations,
  tickets,
  wixEventsV2
});
