// @wix/headless-components/cjs/dist/react/generic-list.js
import { jsx as _jsx } from "react/jsx-runtime";
import React from 'react';
import { AsChildSlot } from '@wix/headless-utils/react';
const GenericListContext = React.createContext(null);
function useGenericListContext() {
    const context = React.useContext(GenericListContext);
    if (!context) {
        throw new Error('useGenericListContext must be used within a GenericList.Root component');
    }
    return context;
}
/**
 * Container for list components that provides data context and manages list state.
 * Renders items inside another model's context and supports multiple display variants,
 * empty states, load more functionality, and totals display.
 *
 * @component
 */
export const Root = React.forwardRef((props, ref) => {
    const { items, loadMore, hasMore = false, isLoading = false, variant, children, className, ...otherProps } = props;
    if (!children) {
        throw new Error('GenericList.Root requires children');
    }
    const contextValue = React.useMemo(() => ({
        items,
        hasMore,
        isLoading,
        loadMore,
        variant,
    }), [items, hasMore, isLoading, loadMore, variant]);
    return (_jsx(GenericListContext.Provider, { value: contextValue, children: _jsx(AsChildSlot, { ref: ref, className: className, "data-has-items": items.length > 0, "data-is-loading": isLoading, "data-has-more": hasMore, "data-variant": variant, customElement: children, ...otherProps, children: children }) }));
});
Root.displayName = 'GenericList.Root';
/**
 * Container for list items that handles empty state display and provides structure
 * for item rendering. Does not render if list is empty unless emptyState is provided.
 *
 * @component
 */
export const Items = React.forwardRef((props, ref) => {
    const { emptyState, children, className, ...otherProps } = props;
    const { items } = useGenericListContext();
    if (!children) {
        throw new Error('GenericList.Items requires children');
    }
    const hasItems = items.length > 0;
    // Show empty state if no items and emptyState is provided
    if (!hasItems) {
        return emptyState || null;
    }
    return (_jsx(AsChildSlot, { ref: ref, className: className, "data-empty": !hasItems, customElement: children, ...otherProps, children: children }));
});
Items.displayName = 'GenericList.Items';
/**
 * Load more button component that appears when more items can be loaded.
 * Automatically handles loading state and disables when no more items are available.
 *
 * @component
 */
export const LoadMore = React.forwardRef((props, ref) => {
    const { label, loadingState, children, className, ...otherProps } = props;
    const { hasMore, isLoading, loadMore } = useGenericListContext();
    // Either children or label must be provided
    if (!children && !label) {
        throw new Error('GenericList.LoadMore requires either children or label');
    }
    // Don't render if no more items to load
    if (!hasMore) {
        return null;
    }
    const handleClick = () => {
        if (loadMore && !isLoading) {
            loadMore();
        }
    };
    // Determine what content to render
    const buttonContent = isLoading ? loadingState : label;
    const content = buttonContent;
    return (_jsx(AsChildSlot, { ref: ref, asChild: true, className: className, "data-has-more": hasMore, "data-is-loading": isLoading, disabled: isLoading, onClick: handleClick, customElement: children, customElementProps: {
            label,
            loadingState,
            isLoading,
            hasMore,
            loadMore,
        }, content: content, ...otherProps }));
});
LoadMore.displayName = 'GenericList.LoadMore';
/**
 * Displays totals information about the list (total items and displayed items).
 * Provides data for custom rendering patterns.
 *
 * @component
 */
export const Totals = React.forwardRef((props, ref) => {
    const { children, className, ...otherProps } = props;
    const { items } = useGenericListContext();
    const displayedItems = items.length;
    const content = displayedItems;
    return (_jsx(AsChildSlot, { ref: ref, className: className, "data-displayed-items": displayedItems, customElement: children, customElementProps: { displayedItems }, asChild: true, content: content, ...otherProps }));
});
Totals.displayName = 'GenericList.Totals';
/**
 * Generic repeater component that maps over items from GenericList context.
 * This component provides a reusable pattern for rendering lists of items
 * where each item needs to be wrapped in a specific Root component.
 * Supports asChild pattern for advanced layout components.
 *
 * @component
 */
export const Repeater = (props) => {
    const { ref, children, itemWrapper = ({ item, children }) => (_jsx(React.Fragment, { children: children }, item['id'] || item['_id'])), asChild = false, className, ...otherProps } = props;
    const { items, variant } = useGenericListContext();
    if (items.length === 0)
        return null;
    if (asChild) {
        const asChildRenderProps = {
            items,
            variant,
            itemWrapper,
        };
        return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, customElement: children, customElementProps: asChildRenderProps, ...otherProps }));
    }
    return items.map((item, index) => itemWrapper({ item, index, children: children }));
};
Repeater.displayName = 'GenericList.Repeater';
export const Actions = {
    LoadMore,
};
export const GenericList = {
    Root,
    Items,
    Repeater,
    Actions,
    Totals,
};
export { Actions as GenericListActions };
