// @wix/headless-components/cjs/dist/react/quantity.js
import { jsx as _jsx } from "react/jsx-runtime";
import React from 'react';
/**
 * Utility function to handle children rendering.
 * TODO - use AsChildSlot instead
 */
function renderChildren({ children, props, ref, }) {
    // Early return if no children provided
    if (!children)
        return null;
    // Handle React element pattern
    if (React.isValidElement(children)) {
        return children;
    }
    // Handle render function pattern
    if (typeof children === 'function') {
        return children(props, ref);
    }
    // Fallback for unknown patterns
    return children;
}
var TestIds;
(function (TestIds) {
    TestIds["quantityRoot"] = "quantity-root";
    TestIds["quantityIncrement"] = "quantity-increment";
    TestIds["quantityDecrement"] = "quantity-decrement";
    TestIds["quantityInput"] = "quantity-input";
    TestIds["quantityReset"] = "quantity-reset";
})(TestIds || (TestIds = {}));
const QuantityContext = React.createContext(null);
function useQuantityContext() {
    const context = React.useContext(QuantityContext);
    if (!context) {
        throw new Error('useQuantityContext must be used within a Quantity.Root component');
    }
    return context;
}
/**
 * Root component that provides quantity context to its children.
 * Manages the quantity state and provides increment/decrement functionality.
 *
 * @component
 * @example
 * ```tsx
 * <Quantity.Root steps={1} onValueChange={(value) => updateQuantity(value)}>
 *   <Quantity.Decrement />
 *   <Quantity.Input />
 *   <Quantity.Increment />
 *   <Quantity.Reset />
 * </Quantity.Root>
 * ```
 */
export const Root = React.forwardRef((props, ref) => {
    const { children, steps = 1, onValueChange, initialValue = 1, min = 1, max = Infinity, reset, className, ...otherProps } = props;
    const contextValue = {
        value: initialValue,
        steps,
        onValueChange,
        reset,
    };
    return (_jsx(QuantityContext.Provider, { value: contextValue, children: _jsx("div", { ref: ref, "data-testid": TestIds.quantityRoot, className: className, ...otherProps, children: children }) }));
});
Root.displayName = 'Quantity.Root';
/**
 * Button to increment the quantity value.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Quantity.Increment className="px-3 py-2 border rounded hover:bg-surface-primary" />
 *
 * // Custom rendering with forwardRef
 * <Quantity.Increment asChild>
 *   {React.forwardRef((props, ref) => (
 *     <button ref={ref} {...props} className="custom-increment-button">
 *       <Plus className="h-4 w-4" />
 *     </button>
 *   ))}
 * </Quantity.Increment>
 * ```
 */
export const Increment = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    const { onValueChange, value, steps } = useQuantityContext();
    if (asChild && children) {
        return renderChildren({ children, props: {}, ref });
    }
    return (_jsx("button", { ref: ref, onClick: () => onValueChange(value + steps), "data-testid": TestIds.quantityIncrement, className: className, ...otherProps, children: typeof children === 'string' ? children : '+' }));
});
Increment.displayName = 'Quantity.Increment';
/**
 * Button to decrement the quantity value.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Quantity.Decrement className="px-3 py-2 border rounded hover:bg-surface-primary" />
 *
 * // Custom rendering with forwardRef
 * <Quantity.Decrement asChild>
 *   {React.forwardRef((props, ref) => (
 *     <button ref={ref} {...props} className="custom-decrement-button">
 *       <Minus className="h-4 w-4" />
 *     </button>
 *   ))}
 * </Quantity.Decrement>
 * ```
 */
export const Decrement = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    const { onValueChange, value, steps } = useQuantityContext();
    if (asChild && children) {
        return renderChildren({ children, props: {}, ref });
    }
    return (_jsx("button", { ref: ref, onClick: () => onValueChange(value - steps), "data-testid": TestIds.quantityDecrement, className: className, ...otherProps, children: typeof children === 'string' ? children : '-' }));
});
Decrement.displayName = 'Quantity.Decrement';
/**
 * Input field for directly editing the quantity value.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Quantity.Input className="px-4 py-2 border text-center min-w-16 focus:ring-2" />
 *
 * // Custom rendering with forwardRef
 * <Quantity.Input asChild>
 *   {React.forwardRef(({value, onChange, ...props}, ref) => (
 *     <input
 *       ref={ref}
 *       {...props}
 *       type="number"
 *       value={value}
 *       onChange={(e) => onChange(parseInt(e.target.value) || 1)}
 *       className="custom-input-style"
 *       min="1"
 *     />
 *   ))}
 * </Quantity.Input>
 * ```
 */
export const Input = React.forwardRef((props, ref) => {
    const { asChild, children, disabled = false, className, ...otherProps } = props;
    const { value, onValueChange } = useQuantityContext();
    const handleChange = (newValue) => {
        if (!isNaN(newValue) && newValue > 0) {
            onValueChange(newValue);
        }
    };
    const renderProps = {
        value,
        onChange: handleChange,
    };
    if (asChild && children) {
        return renderChildren({ children, props: renderProps, ref });
    }
    return (_jsx("input", { ref: ref, type: "text" // avoid number input spinner padding issues
        , value: value, onChange: (e) => handleChange(parseInt(e.target.value) || 1), disabled: disabled, "data-testid": TestIds.quantityInput, className: className, min: "1", ...otherProps }));
});
Input.displayName = 'Quantity.Input';
/**
 * Button to reset the quantity to its initial value.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Quantity.Reset className="px-3 py-2 text-sm text-red-600 hover:bg-red-50">
 *   Reset
 * </Quantity.Reset>
 *
 * // Custom rendering with forwardRef
 * <Quantity.Reset asChild>
 *   {React.forwardRef((props, ref) => (
 *     <button ref={ref} {...props} className="custom-reset-button">
 *       <RotateCcw className="h-4 w-4" />
 *     </button>
 *   ))}
 * </Quantity.Reset>
 * ```
 */
export const Reset = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    const { reset } = useQuantityContext();
    if (asChild && children) {
        return renderChildren({ children, props: {}, ref });
    }
    return (_jsx("button", { ref: ref, onClick: reset, "data-testid": TestIds.quantityReset, className: className, ...otherProps, children: typeof children === 'string' ? children : 'Reset' }));
});
Reset.displayName = 'Quantity.Reset';
