// @wix/bex-utils/@wix/ambassador-app-settings-v1-settings/build/cjs/builders.impl.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.aUpdateResponse = exports.aUpdateRequest = exports.aSetResponse = exports.aScope = exports.aSetRequest = exports.aGetResponse = exports.aGetRequest = exports.aGetGlobalSettingsForMigrationResponse = exports.aGetGlobalSettingsForMigrationRequest = exports.aSetGlobalSettingsForMigrationResponse = exports.aSetGlobalSettingsForMigrationRequest = exports.aGetComponentSettingsItem = exports.aGetGlobalSettingsResponse = exports.aGetGlobalSettingsRequest = exports.aSetGlobalSettingsResponse = exports.aSetComponentSettingsItem = exports.aSetGlobalSettingsRequest = exports.aComponentSettings = exports.aBulkGetComponentSettingsResponse = exports.aBulkGetComponentSettingsRequest = exports.aGetComponentSettingsResponse = exports.aGetComponentSettingsRequest = exports.aSetComponentSettingsForMigrationResponse = exports.aSetComponentSettingsForMigrationRequest = exports.aSetComponentSettingsResponse = exports.aLanguageKey = exports.aTranslatedLanguage = exports.aSetComponentSettingsRequest = exports.aHost = exports.aState = exports.aAppSettings = void 0;
var metro_common_builders_1 = require("@wix/metro-common-builders");
var types_impl_1 = require("./types.impl");
// @ts-ignore
var callsite_1 = __importDefault(require("callsite"));
function aAppSettings(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        state: aState(),
        host: aHost(),
        settings: {},
        translations: { translations: (0, metro_common_builders_1.someString)() },
    }, overrideObject);
}
exports.aAppSettings = aAppSettings;
function aState(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.State.NR; }
    return enumFirstItem;
}
exports.aState = aState;
function aHost(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.Host.NA; }
    return enumFirstItem;
}
exports.aHost = aHost;
function aSetComponentSettingsRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        componentId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '36' }],
        }),
        settings: {},
        state: aState(),
        host: aHost(),
        translations: [aTranslatedLanguage()],
        states: [aState()],
    }, overrideObject);
}
exports.aSetComponentSettingsRequest = aSetComponentSettingsRequest;
function aTranslatedLanguage(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        languageKey: aLanguageKey(),
        translations: { translations: (0, metro_common_builders_1.someString)() },
    }, overrideObject);
}
exports.aTranslatedLanguage = aTranslatedLanguage;
function aLanguageKey(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        languageCode: (0, metro_common_builders_1.someString)(),
        regionCode: (0, metro_common_builders_1.someString)(),
        scriptVariant: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aLanguageKey = aLanguageKey;
function aSetComponentSettingsResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        settings: {},
        host: aHost(),
        translations: [aTranslatedLanguage()],
        states: [aState()],
    }, overrideObject);
}
exports.aSetComponentSettingsResponse = aSetComponentSettingsResponse;
function aSetComponentSettingsForMigrationRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        componentId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '36' }],
        }),
        settings: {},
        state: aState(),
        host: aHost(),
        translations: [aTranslatedLanguage()],
        states: [aState()],
        appDefId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        instanceId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
exports.aSetComponentSettingsForMigrationRequest = aSetComponentSettingsForMigrationRequest;
function aSetComponentSettingsForMigrationResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        settings: {},
        host: aHost(),
        translations: [aTranslatedLanguage()],
        states: [aState()],
    }, overrideObject);
}
exports.aSetComponentSettingsForMigrationResponse = aSetComponentSettingsForMigrationResponse;
function aGetComponentSettingsRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        componentId: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '36' },
            ],
        }),
        state: aState(),
        host: aHost(),
        languageKey: aLanguageKey(),
        includeGlobalSettings: true,
    }, overrideObject);
}
exports.aGetComponentSettingsRequest = aGetComponentSettingsRequest;
function aGetComponentSettingsResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        settings: {},
        state: aState(),
        host: aHost(),
        translations: { translations: (0, metro_common_builders_1.someString)() },
        globalSettings: aAppSettings(),
    }, overrideObject);
}
exports.aGetComponentSettingsResponse = aGetComponentSettingsResponse;
function aBulkGetComponentSettingsRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        componentIds: [
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MIN_LENGTH', value: '1' },
                    { type: 'MAX_LENGTH', value: '36' },
                    { type: 'MIN_SIZE', value: '1' },
                    { type: 'MAX_SIZE', value: '100' },
                ],
            }),
        ],
        state: aState(),
        host: aHost(),
        languageKey: aLanguageKey(),
    }, overrideObject);
}
exports.aBulkGetComponentSettingsRequest = aBulkGetComponentSettingsRequest;
function aBulkGetComponentSettingsResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        componentSettings: [aComponentSettings()],
    }, overrideObject);
}
exports.aBulkGetComponentSettingsResponse = aBulkGetComponentSettingsResponse;
function aComponentSettings(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        componentId: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '36' },
            ],
        }),
        settings: {},
        state: aState(),
        host: aHost(),
        translations: { translations: (0, metro_common_builders_1.someString)() },
    }, overrideObject);
}
exports.aComponentSettings = aComponentSettings;
function aSetGlobalSettingsRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        settings: {},
        componentSettings: [aSetComponentSettingsItem()],
        state: aState(),
        host: aHost(),
        translations: [aTranslatedLanguage()],
        states: [aState()],
    }, overrideObject);
}
exports.aSetGlobalSettingsRequest = aSetGlobalSettingsRequest;
function aSetComponentSettingsItem(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        state: aState(),
        host: aHost(),
        componentId: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '36' },
            ],
        }),
        settings: {},
        translations: [aTranslatedLanguage()],
    }, overrideObject);
}
exports.aSetComponentSettingsItem = aSetComponentSettingsItem;
function aSetGlobalSettingsResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        settings: {},
        componentSettings: [aSetComponentSettingsItem()],
        host: aHost(),
        translations: [aTranslatedLanguage()],
        states: [aState()],
    }, overrideObject);
}
exports.aSetGlobalSettingsResponse = aSetGlobalSettingsResponse;
function aGetGlobalSettingsRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        state: aState(),
        host: aHost(),
        includeComponentSettings: true,
        languageKey: aLanguageKey(),
    }, overrideObject);
}
exports.aGetGlobalSettingsRequest = aGetGlobalSettingsRequest;
function aGetGlobalSettingsResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        state: aState(),
        host: aHost(),
        settings: {},
        componentSettings: [aGetComponentSettingsItem()],
        translations: { translations: (0, metro_common_builders_1.someString)() },
    }, overrideObject);
}
exports.aGetGlobalSettingsResponse = aGetGlobalSettingsResponse;
function aGetComponentSettingsItem(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        state: aState(),
        host: aHost(),
        componentId: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '36' },
            ],
        }),
        settings: {},
        translations: { translations: (0, metro_common_builders_1.someString)() },
    }, overrideObject);
}
exports.aGetComponentSettingsItem = aGetComponentSettingsItem;
function aSetGlobalSettingsForMigrationRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        settings: {},
        componentSettings: [aSetComponentSettingsItem()],
        state: aState(),
        host: aHost(),
        translations: [aTranslatedLanguage()],
        states: [aState()],
        appDefId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        instanceId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
exports.aSetGlobalSettingsForMigrationRequest = aSetGlobalSettingsForMigrationRequest;
function aSetGlobalSettingsForMigrationResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        settings: {},
        componentSettings: [aSetComponentSettingsItem()],
        host: aHost(),
        translations: [aTranslatedLanguage()],
        states: [aState()],
    }, overrideObject);
}
exports.aSetGlobalSettingsForMigrationResponse = aSetGlobalSettingsForMigrationResponse;
function aGetGlobalSettingsForMigrationRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        state: aState(),
        host: aHost(),
        includeComponentSettings: true,
        languageKey: aLanguageKey(),
        appDefId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        instanceId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
exports.aGetGlobalSettingsForMigrationRequest = aGetGlobalSettingsForMigrationRequest;
function aGetGlobalSettingsForMigrationResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        state: aState(),
        host: aHost(),
        settings: {},
        componentSettings: [aGetComponentSettingsItem()],
        translations: [aTranslatedLanguage()],
    }, overrideObject);
}
exports.aGetGlobalSettingsForMigrationResponse = aGetGlobalSettingsForMigrationResponse;
function aGetRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        externalAppId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        externalComponentId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        fields: (0, metro_common_builders_1.someRepeated)(),
    }, overrideObject);
}
exports.aGetRequest = aGetRequest;
function aGetResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        appSettings: {},
        componentSettings: {},
    }, overrideObject);
}
exports.aGetResponse = aGetResponse;
function aSetRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        externalId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        scope: aScope(),
        data: {},
    }, overrideObject);
}
exports.aSetRequest = aSetRequest;
function aScope(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.Scope.APP; }
    return enumFirstItem;
}
exports.aScope = aScope;
function aSetResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({}, overrideObject);
}
exports.aSetResponse = aSetResponse;
function aUpdateRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        externalId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        scope: aScope(),
        fields: (0, metro_common_builders_1.someRepeated)(),
        data: {},
    }, overrideObject);
}
exports.aUpdateRequest = aUpdateRequest;
function aUpdateResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        newExternalId: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aUpdateResponse = aUpdateResponse;
function isCurrentBuilderWasPreviouslyCalled() {
    var callChainFunctionNames = (0, callsite_1.default)().map(function (item) {
        return item.getFunctionName();
    });
    var callChainWithoutCurrentBuilder = callChainFunctionNames.splice(2);
    var currentBuilderName = callChainFunctionNames[1];
    return callChainWithoutCurrentBuilder.some(function (functionName) { return functionName === currentBuilderName; });
}
__exportStar(require("@wix/metro-common-builders"), exports);
