// @wix/bex-utils/@wix/ambassador-app-settings-v1-settings/build/cjs/meta.impl.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.update = exports.set = exports.get = exports.getGlobalSettings = exports.setGlobalSettings = exports.bulkGetComponentSettings = exports.getComponentSettings = exports.setComponentSettings = void 0;
var httpWixAppSettingsV1Settings = __importStar(require("./http.impl"));
function setComponentSettings() {
    var payload = { componentId: ':componentId' };
    var getRequestOptions = httpWixAppSettingsV1Settings.setComponentSettings(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'PATCH',
        path: '/api/v1/settings/components/{componentId}',
        pathParams: { componentId: 'componentId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.setComponentSettings = setComponentSettings;
function getComponentSettings() {
    var payload = { componentId: ':componentId' };
    var getRequestOptions = httpWixAppSettingsV1Settings.getComponentSettings(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/api/v1/settings/components/{componentId}',
        pathParams: { componentId: 'componentId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getComponentSettings = getComponentSettings;
function bulkGetComponentSettings() {
    var payload = {};
    var getRequestOptions = httpWixAppSettingsV1Settings.bulkGetComponentSettings(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/api/v1/settings/components/bulk',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkGetComponentSettings = bulkGetComponentSettings;
function setGlobalSettings() {
    var payload = {};
    var getRequestOptions = httpWixAppSettingsV1Settings.setGlobalSettings(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'PATCH',
        path: '/api/v1/settings/instances',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.setGlobalSettings = setGlobalSettings;
function getGlobalSettings() {
    var payload = {};
    var getRequestOptions = httpWixAppSettingsV1Settings.getGlobalSettings(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/api/v1/settings/instances',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getGlobalSettings = getGlobalSettings;
function get() {
    var payload = {};
    var getRequestOptions = httpWixAppSettingsV1Settings.get(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/api/v2/settings',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.get = get;
function set() {
    var payload = { scope: ':scope', externalId: ':externalId' };
    var getRequestOptions = httpWixAppSettingsV1Settings.set(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'PUT',
        path: '/api/v2/settings/{scope}/{externalId}',
        pathParams: { scope: 'scope', externalId: 'externalId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.set = set;
function update() {
    var payload = { scope: ':scope', externalId: ':externalId' };
    var getRequestOptions = httpWixAppSettingsV1Settings.update(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'PATCH',
        path: '/api/v2/settings/{scope}/{externalId}',
        pathParams: { scope: 'scope', externalId: 'externalId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.update = update;
