// @wix/auto_sdk_restaurants_operation-groups/build/internal/es/index.typings.mjs
// src/restaurants-v1-operation-group-operation-groups.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/restaurants-v1-operation-group-operation-groups.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixRestaurantsV1OperationGroupServiceUrl(opts) {
  const domainToMappings = {
    "dev._base_domain_": [
      {
        srcPath: "/_api/restaurants-operation-group",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-operation-group",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-operation-group",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-operation-group",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-operation-group",
        destPath: ""
      },
      {
        srcPath: "/restaurants/v1/bulk/operation-groups",
        destPath: "/v1/bulk/operation-groups"
      },
      {
        srcPath: "/restaurants/v1/operation-groups",
        destPath: "/v1/operation-groups"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-operation-group",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-operation-group",
        destPath: ""
      },
      {
        srcPath: "/restaurants/v1/bulk/operation-groups",
        destPath: "/v1/bulk/operation-groups"
      },
      {
        srcPath: "/restaurants/v1/operation-groups",
        destPath: "/v1/operation-groups"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-operation-group",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/v1/operation-groups",
        destPath: "/v1/operation-groups"
      },
      {
        srcPath: "/restaurants/v1/bulk/operation-groups",
        destPath: "/v1/bulk/operation-groups"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_operation-groups";
function createOperationGroup(payload) {
  function __createOperationGroup({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "operationGroup.createdDate" },
          { path: "operationGroup.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.operation_group",
      method: "POST",
      methodFqn: "wix.restaurants.v1.OperationGroupService.CreateOperationGroup",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1OperationGroupServiceUrl({
        protoPath: "/v1/operation-groups",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "operationGroup.createdDate" },
            { path: "operationGroup.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createOperationGroup;
}
function getOperationGroup(payload) {
  function __getOperationGroup({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.operation_group",
      method: "GET",
      methodFqn: "wix.restaurants.v1.OperationGroupService.GetOperationGroup",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1OperationGroupServiceUrl({
        protoPath: "/v1/operation-groups/{operationGroupId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "operationGroup.createdDate" },
            { path: "operationGroup.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getOperationGroup;
}
function updateOperationGroup(payload) {
  function __updateOperationGroup({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "operationGroup.createdDate" },
          { path: "operationGroup.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.operation_group",
      method: "PATCH",
      methodFqn: "wix.restaurants.v1.OperationGroupService.UpdateOperationGroup",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1OperationGroupServiceUrl({
        protoPath: "/v1/operation-groups/{operationGroup.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "operationGroup.createdDate" },
            { path: "operationGroup.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateOperationGroup;
}
function deleteOperationGroup(payload) {
  function __deleteOperationGroup({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.operation_group",
      method: "DELETE",
      methodFqn: "wix.restaurants.v1.OperationGroupService.DeleteOperationGroup",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1OperationGroupServiceUrl({
        protoPath: "/v1/operation-groups/{operationGroupId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteOperationGroup;
}
function queryOperationGroups(payload) {
  function __queryOperationGroups({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.operation_group",
      method: "GET",
      methodFqn: "wix.restaurants.v1.OperationGroupService.QueryOperationGroups",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1OperationGroupServiceUrl({
        protoPath: "/v1/operation-groups/query",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload, true),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "operationGroups.createdDate" },
            { path: "operationGroups.updatedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixRestaurantsV1OperationGroupServiceUrl({
            protoPath: "/v1/operation-groups/query",
            data: payload,
            host
          }),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryOperationGroups;
}
function bulkCreateOperationGroups(payload) {
  function __bulkCreateOperationGroups({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "operationGroups.createdDate" },
          { path: "operationGroups.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.operation_group",
      method: "POST",
      methodFqn: "wix.restaurants.v1.OperationGroupService.BulkCreateOperationGroups",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1OperationGroupServiceUrl({
        protoPath: "/v1/bulk/operation-groups/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateOperationGroups;
}
function bulkUpdateOperationGroups(payload) {
  function __bulkUpdateOperationGroups({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "operationGroups.fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "operationGroups.operationGroup.createdDate" },
          { path: "operationGroups.operationGroup.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.operation_group",
      method: "POST",
      methodFqn: "wix.restaurants.v1.OperationGroupService.BulkUpdateOperationGroups",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1OperationGroupServiceUrl({
        protoPath: "/v1/bulk/operation-groups/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateOperationGroups;
}
function bulkDeleteOperationGroups(payload) {
  function __bulkDeleteOperationGroups({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.operation_group",
      method: "POST",
      methodFqn: "wix.restaurants.v1.OperationGroupService.BulkDeleteOperationGroups",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1OperationGroupServiceUrl({
        protoPath: "/v1/bulk/operation-groups/delete",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkDeleteOperationGroups;
}
function bulkUpdateOperationGroupTags(payload) {
  function __bulkUpdateOperationGroupTags({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.operation_group",
      method: "POST",
      methodFqn: "wix.restaurants.v1.OperationGroupService.BulkUpdateOperationGroupTags",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1OperationGroupServiceUrl({
        protoPath: "/v1/bulk/operation-groups/update-tags",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateOperationGroupTags;
}
function bulkUpdateOperationGroupTagsByFilter(payload) {
  function __bulkUpdateOperationGroupTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.operation_group",
      method: "POST",
      methodFqn: "wix.restaurants.v1.OperationGroupService.BulkUpdateOperationGroupTagsByFilter",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1OperationGroupServiceUrl({
        protoPath: "/v1/bulk/operation-groups/update-tags-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateOperationGroupTagsByFilter;
}

// src/restaurants-v1-operation-group-operation-groups.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createOperationGroup2(operationGroup) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    operationGroup
  });
  const reqOpts = createOperationGroup(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.operationGroup;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { operationGroup: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["operationGroup"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getOperationGroup2(operationGroupId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    operationGroupId
  });
  const reqOpts = getOperationGroup(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.operationGroup;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { operationGroupId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["operationGroupId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateOperationGroup2(_id, operationGroup) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    operationGroup: { ...operationGroup, id: _id }
  });
  const reqOpts = updateOperationGroup(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.operationGroup;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { operationGroup: "$[1]" },
        explicitPathsToArguments: { "operationGroup.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "operationGroup"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteOperationGroup2(operationGroupId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    operationGroupId
  });
  const reqOpts = deleteOperationGroup(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { operationGroupId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["operationGroupId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryOperationGroups2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryOperationGroups(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [])
      );
      return {
        items: transformedData?.operationGroups,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function bulkCreateOperationGroups2(operationGroups, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    operationGroups,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkCreateOperationGroups(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          operationGroups: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["operationGroups", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateOperationGroups2(operationGroups, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    operationGroups,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkUpdateOperationGroups(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          operationGroups: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["operationGroups", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkDeleteOperationGroups2(operationGroupIds) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    operationGroupIds
  });
  const reqOpts = bulkDeleteOperationGroups(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { operationGroupIds: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["operationGroupIds"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateOperationGroupTags2(operationGroupIds, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    operationGroupIds,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateOperationGroupTags(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          operationGroupIds: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["operationGroupIds", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateOperationGroupTagsByFilter2(filter, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    filter,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateOperationGroupTagsByFilter(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          filter: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["filter", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
export {
  SortOrder,
  WebhookIdentityType,
  bulkCreateOperationGroups2 as bulkCreateOperationGroups,
  bulkDeleteOperationGroups2 as bulkDeleteOperationGroups,
  bulkUpdateOperationGroupTags2 as bulkUpdateOperationGroupTags,
  bulkUpdateOperationGroupTagsByFilter2 as bulkUpdateOperationGroupTagsByFilter,
  bulkUpdateOperationGroups2 as bulkUpdateOperationGroups,
  createOperationGroup2 as createOperationGroup,
  deleteOperationGroup2 as deleteOperationGroup,
  getOperationGroup2 as getOperationGroup,
  queryOperationGroups2 as queryOperationGroups,
  updateOperationGroup2 as updateOperationGroup
};
