// @wix/auto_sdk_restaurants_operation-groups/build/internal/es/index.mjs
// src/restaurants-v1-operation-group-operation-groups.public.ts
import { renameKeysFromRESTResponseToSDKResponse as renameKeysFromRESTResponseToSDKResponse2 } from "@wix/sdk-runtime/rename-all-nested-keys";
import { transformRESTTimestampToSDKTimestamp as transformRESTTimestampToSDKTimestamp2 } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths as transformPaths3 } from "@wix/sdk-runtime/transformations/transform-paths";
import { EventDefinition } from "@wix/sdk-types";

// src/restaurants-v1-operation-group-operation-groups.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/restaurants-v1-operation-group-operation-groups.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixRestaurantsV1OperationGroupServiceUrl(opts) {
  const domainToMappings = {
    "dev._base_domain_": [
      {
        srcPath: "/_api/restaurants-operation-group",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-operation-group",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-operation-group",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-operation-group",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-operation-group",
        destPath: ""
      },
      {
        srcPath: "/restaurants/v1/bulk/operation-groups",
        destPath: "/v1/bulk/operation-groups"
      },
      {
        srcPath: "/restaurants/v1/operation-groups",
        destPath: "/v1/operation-groups"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-operation-group",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-operation-group",
        destPath: ""
      },
      {
        srcPath: "/restaurants/v1/bulk/operation-groups",
        destPath: "/v1/bulk/operation-groups"
      },
      {
        srcPath: "/restaurants/v1/operation-groups",
        destPath: "/v1/operation-groups"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-operation-group",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/v1/operation-groups",
        destPath: "/v1/operation-groups"
      },
      {
        srcPath: "/restaurants/v1/bulk/operation-groups",
        destPath: "/v1/bulk/operation-groups"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_operation-groups";
function createOperationGroup(payload) {
  function __createOperationGroup({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "operationGroup.createdDate" },
          { path: "operationGroup.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.operation_group",
      method: "POST",
      methodFqn: "wix.restaurants.v1.OperationGroupService.CreateOperationGroup",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1OperationGroupServiceUrl({
        protoPath: "/v1/operation-groups",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "operationGroup.createdDate" },
            { path: "operationGroup.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createOperationGroup;
}
function getOperationGroup(payload) {
  function __getOperationGroup({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.operation_group",
      method: "GET",
      methodFqn: "wix.restaurants.v1.OperationGroupService.GetOperationGroup",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1OperationGroupServiceUrl({
        protoPath: "/v1/operation-groups/{operationGroupId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "operationGroup.createdDate" },
            { path: "operationGroup.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getOperationGroup;
}
function updateOperationGroup(payload) {
  function __updateOperationGroup({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "operationGroup.createdDate" },
          { path: "operationGroup.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.operation_group",
      method: "PATCH",
      methodFqn: "wix.restaurants.v1.OperationGroupService.UpdateOperationGroup",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1OperationGroupServiceUrl({
        protoPath: "/v1/operation-groups/{operationGroup.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "operationGroup.createdDate" },
            { path: "operationGroup.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateOperationGroup;
}
function deleteOperationGroup(payload) {
  function __deleteOperationGroup({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.operation_group",
      method: "DELETE",
      methodFqn: "wix.restaurants.v1.OperationGroupService.DeleteOperationGroup",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1OperationGroupServiceUrl({
        protoPath: "/v1/operation-groups/{operationGroupId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteOperationGroup;
}
function queryOperationGroups(payload) {
  function __queryOperationGroups({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.operation_group",
      method: "GET",
      methodFqn: "wix.restaurants.v1.OperationGroupService.QueryOperationGroups",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1OperationGroupServiceUrl({
        protoPath: "/v1/operation-groups/query",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload, true),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "operationGroups.createdDate" },
            { path: "operationGroups.updatedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixRestaurantsV1OperationGroupServiceUrl({
            protoPath: "/v1/operation-groups/query",
            data: payload,
            host
          }),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryOperationGroups;
}
function bulkCreateOperationGroups(payload) {
  function __bulkCreateOperationGroups({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "operationGroups.createdDate" },
          { path: "operationGroups.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.operation_group",
      method: "POST",
      methodFqn: "wix.restaurants.v1.OperationGroupService.BulkCreateOperationGroups",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1OperationGroupServiceUrl({
        protoPath: "/v1/bulk/operation-groups/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateOperationGroups;
}
function bulkUpdateOperationGroups(payload) {
  function __bulkUpdateOperationGroups({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "operationGroups.fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "operationGroups.operationGroup.createdDate" },
          { path: "operationGroups.operationGroup.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.operation_group",
      method: "POST",
      methodFqn: "wix.restaurants.v1.OperationGroupService.BulkUpdateOperationGroups",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1OperationGroupServiceUrl({
        protoPath: "/v1/bulk/operation-groups/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateOperationGroups;
}
function bulkDeleteOperationGroups(payload) {
  function __bulkDeleteOperationGroups({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.operation_group",
      method: "POST",
      methodFqn: "wix.restaurants.v1.OperationGroupService.BulkDeleteOperationGroups",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1OperationGroupServiceUrl({
        protoPath: "/v1/bulk/operation-groups/delete",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkDeleteOperationGroups;
}
function bulkUpdateOperationGroupTags(payload) {
  function __bulkUpdateOperationGroupTags({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.operation_group",
      method: "POST",
      methodFqn: "wix.restaurants.v1.OperationGroupService.BulkUpdateOperationGroupTags",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1OperationGroupServiceUrl({
        protoPath: "/v1/bulk/operation-groups/update-tags",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateOperationGroupTags;
}
function bulkUpdateOperationGroupTagsByFilter(payload) {
  function __bulkUpdateOperationGroupTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.operation_group",
      method: "POST",
      methodFqn: "wix.restaurants.v1.OperationGroupService.BulkUpdateOperationGroupTagsByFilter",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1OperationGroupServiceUrl({
        protoPath: "/v1/bulk/operation-groups/update-tags-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateOperationGroupTagsByFilter;
}

// src/restaurants-v1-operation-group-operation-groups.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createOperationGroup2(operationGroup) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    operationGroup
  });
  const reqOpts = createOperationGroup(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.operationGroup;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { operationGroup: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["operationGroup"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getOperationGroup2(operationGroupId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    operationGroupId
  });
  const reqOpts = getOperationGroup(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.operationGroup;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { operationGroupId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["operationGroupId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateOperationGroup2(_id, operationGroup) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    operationGroup: { ...operationGroup, id: _id }
  });
  const reqOpts = updateOperationGroup(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.operationGroup;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { operationGroup: "$[1]" },
        explicitPathsToArguments: { "operationGroup.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "operationGroup"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteOperationGroup2(operationGroupId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    operationGroupId
  });
  const reqOpts = deleteOperationGroup(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { operationGroupId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["operationGroupId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryOperationGroups2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryOperationGroups(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [])
      );
      return {
        items: transformedData?.operationGroups,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function bulkCreateOperationGroups2(operationGroups, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    operationGroups,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkCreateOperationGroups(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          operationGroups: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["operationGroups", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateOperationGroups2(operationGroups, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    operationGroups,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkUpdateOperationGroups(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          operationGroups: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["operationGroups", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkDeleteOperationGroups2(operationGroupIds) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    operationGroupIds
  });
  const reqOpts = bulkDeleteOperationGroups(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { operationGroupIds: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["operationGroupIds"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateOperationGroupTags2(operationGroupIds, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    operationGroupIds,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateOperationGroupTags(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          operationGroupIds: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["operationGroupIds", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateOperationGroupTagsByFilter2(filter, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    filter,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateOperationGroupTagsByFilter(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          filter: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["filter", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/restaurants-v1-operation-group-operation-groups.public.ts
function createOperationGroup3(httpClient) {
  return (operationGroup) => createOperationGroup2(
    operationGroup,
    // @ts-ignore
    { httpClient }
  );
}
function getOperationGroup3(httpClient) {
  return (operationGroupId) => getOperationGroup2(
    operationGroupId,
    // @ts-ignore
    { httpClient }
  );
}
function updateOperationGroup3(httpClient) {
  return (_id, operationGroup) => updateOperationGroup2(
    _id,
    operationGroup,
    // @ts-ignore
    { httpClient }
  );
}
function deleteOperationGroup3(httpClient) {
  return (operationGroupId) => deleteOperationGroup2(
    operationGroupId,
    // @ts-ignore
    { httpClient }
  );
}
function queryOperationGroups3(httpClient) {
  return () => queryOperationGroups2(
    // @ts-ignore
    { httpClient }
  );
}
function bulkCreateOperationGroups3(httpClient) {
  return (operationGroups, options) => bulkCreateOperationGroups2(
    operationGroups,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdateOperationGroups3(httpClient) {
  return (operationGroups, options) => bulkUpdateOperationGroups2(
    operationGroups,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkDeleteOperationGroups3(httpClient) {
  return (operationGroupIds) => bulkDeleteOperationGroups2(
    operationGroupIds,
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdateOperationGroupTags3(httpClient) {
  return (operationGroupIds, options) => bulkUpdateOperationGroupTags2(
    operationGroupIds,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdateOperationGroupTagsByFilter3(httpClient) {
  return (filter, options) => bulkUpdateOperationGroupTagsByFilter2(
    filter,
    options,
    // @ts-ignore
    { httpClient }
  );
}
var onOperationGroupCreated = EventDefinition(
  "wix.restaurants.v1.operation_group_created",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onOperationGroupDeleted = EventDefinition(
  "wix.restaurants.v1.operation_group_deleted",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onOperationGroupUpdated = EventDefinition(
  "wix.restaurants.v1.operation_group_updated",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();

// src/restaurants-v1-operation-group-operation-groups.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
import { createEventModule } from "@wix/sdk-runtime/event-definition-modules";
var createOperationGroup4 = /* @__PURE__ */ createRESTModule(createOperationGroup3);
var getOperationGroup4 = /* @__PURE__ */ createRESTModule(getOperationGroup3);
var updateOperationGroup4 = /* @__PURE__ */ createRESTModule(updateOperationGroup3);
var deleteOperationGroup4 = /* @__PURE__ */ createRESTModule(deleteOperationGroup3);
var queryOperationGroups4 = /* @__PURE__ */ createRESTModule(queryOperationGroups3);
var bulkCreateOperationGroups4 = /* @__PURE__ */ createRESTModule(bulkCreateOperationGroups3);
var bulkUpdateOperationGroups4 = /* @__PURE__ */ createRESTModule(bulkUpdateOperationGroups3);
var bulkDeleteOperationGroups4 = /* @__PURE__ */ createRESTModule(bulkDeleteOperationGroups3);
var bulkUpdateOperationGroupTags4 = /* @__PURE__ */ createRESTModule(bulkUpdateOperationGroupTags3);
var bulkUpdateOperationGroupTagsByFilter4 = /* @__PURE__ */ createRESTModule(bulkUpdateOperationGroupTagsByFilter3);
var onOperationGroupCreated2 = createEventModule(
  onOperationGroupCreated
);
var onOperationGroupDeleted2 = createEventModule(
  onOperationGroupDeleted
);
var onOperationGroupUpdated2 = createEventModule(
  onOperationGroupUpdated
);
export {
  SortOrder,
  WebhookIdentityType,
  bulkCreateOperationGroups4 as bulkCreateOperationGroups,
  bulkDeleteOperationGroups4 as bulkDeleteOperationGroups,
  bulkUpdateOperationGroupTags4 as bulkUpdateOperationGroupTags,
  bulkUpdateOperationGroupTagsByFilter4 as bulkUpdateOperationGroupTagsByFilter,
  bulkUpdateOperationGroups4 as bulkUpdateOperationGroups,
  createOperationGroup4 as createOperationGroup,
  deleteOperationGroup4 as deleteOperationGroup,
  getOperationGroup4 as getOperationGroup,
  onOperationGroupCreated2 as onOperationGroupCreated,
  onOperationGroupDeleted2 as onOperationGroupDeleted,
  onOperationGroupUpdated2 as onOperationGroupUpdated,
  queryOperationGroups4 as queryOperationGroups,
  updateOperationGroup4 as updateOperationGroup
};
