// @wix/auto_sdk_restaurants_operation-groups/build/internal/es/meta.mjs
// src/restaurants-v1-operation-group-operation-groups.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixRestaurantsV1OperationGroupServiceUrl(opts) {
  const domainToMappings = {
    "dev._base_domain_": [
      {
        srcPath: "/_api/restaurants-operation-group",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-operation-group",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-operation-group",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-operation-group",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-operation-group",
        destPath: ""
      },
      {
        srcPath: "/restaurants/v1/bulk/operation-groups",
        destPath: "/v1/bulk/operation-groups"
      },
      {
        srcPath: "/restaurants/v1/operation-groups",
        destPath: "/v1/operation-groups"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-operation-group",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-operation-group",
        destPath: ""
      },
      {
        srcPath: "/restaurants/v1/bulk/operation-groups",
        destPath: "/v1/bulk/operation-groups"
      },
      {
        srcPath: "/restaurants/v1/operation-groups",
        destPath: "/v1/operation-groups"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-operation-group",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/v1/operation-groups",
        destPath: "/v1/operation-groups"
      },
      {
        srcPath: "/restaurants/v1/bulk/operation-groups",
        destPath: "/v1/bulk/operation-groups"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_operation-groups";
function createOperationGroup(payload) {
  function __createOperationGroup({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "operationGroup.createdDate" },
          { path: "operationGroup.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.operation_group",
      method: "POST",
      methodFqn: "wix.restaurants.v1.OperationGroupService.CreateOperationGroup",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1OperationGroupServiceUrl({
        protoPath: "/v1/operation-groups",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "operationGroup.createdDate" },
            { path: "operationGroup.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createOperationGroup;
}
function getOperationGroup(payload) {
  function __getOperationGroup({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.operation_group",
      method: "GET",
      methodFqn: "wix.restaurants.v1.OperationGroupService.GetOperationGroup",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1OperationGroupServiceUrl({
        protoPath: "/v1/operation-groups/{operationGroupId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "operationGroup.createdDate" },
            { path: "operationGroup.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getOperationGroup;
}
function updateOperationGroup(payload) {
  function __updateOperationGroup({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "operationGroup.createdDate" },
          { path: "operationGroup.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.operation_group",
      method: "PATCH",
      methodFqn: "wix.restaurants.v1.OperationGroupService.UpdateOperationGroup",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1OperationGroupServiceUrl({
        protoPath: "/v1/operation-groups/{operationGroup.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "operationGroup.createdDate" },
            { path: "operationGroup.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateOperationGroup;
}
function deleteOperationGroup(payload) {
  function __deleteOperationGroup({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.operation_group",
      method: "DELETE",
      methodFqn: "wix.restaurants.v1.OperationGroupService.DeleteOperationGroup",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1OperationGroupServiceUrl({
        protoPath: "/v1/operation-groups/{operationGroupId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteOperationGroup;
}
function queryOperationGroups(payload) {
  function __queryOperationGroups({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.operation_group",
      method: "GET",
      methodFqn: "wix.restaurants.v1.OperationGroupService.QueryOperationGroups",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1OperationGroupServiceUrl({
        protoPath: "/v1/operation-groups/query",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload, true),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "operationGroups.createdDate" },
            { path: "operationGroups.updatedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixRestaurantsV1OperationGroupServiceUrl({
            protoPath: "/v1/operation-groups/query",
            data: payload,
            host
          }),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryOperationGroups;
}
function bulkCreateOperationGroups(payload) {
  function __bulkCreateOperationGroups({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "operationGroups.createdDate" },
          { path: "operationGroups.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.operation_group",
      method: "POST",
      methodFqn: "wix.restaurants.v1.OperationGroupService.BulkCreateOperationGroups",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1OperationGroupServiceUrl({
        protoPath: "/v1/bulk/operation-groups/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateOperationGroups;
}
function bulkUpdateOperationGroups(payload) {
  function __bulkUpdateOperationGroups({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "operationGroups.fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "operationGroups.operationGroup.createdDate" },
          { path: "operationGroups.operationGroup.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.operation_group",
      method: "POST",
      methodFqn: "wix.restaurants.v1.OperationGroupService.BulkUpdateOperationGroups",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1OperationGroupServiceUrl({
        protoPath: "/v1/bulk/operation-groups/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateOperationGroups;
}
function bulkDeleteOperationGroups(payload) {
  function __bulkDeleteOperationGroups({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.operation_group",
      method: "POST",
      methodFqn: "wix.restaurants.v1.OperationGroupService.BulkDeleteOperationGroups",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1OperationGroupServiceUrl({
        protoPath: "/v1/bulk/operation-groups/delete",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkDeleteOperationGroups;
}
function bulkUpdateOperationGroupTags(payload) {
  function __bulkUpdateOperationGroupTags({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.operation_group",
      method: "POST",
      methodFqn: "wix.restaurants.v1.OperationGroupService.BulkUpdateOperationGroupTags",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1OperationGroupServiceUrl({
        protoPath: "/v1/bulk/operation-groups/update-tags",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateOperationGroupTags;
}
function bulkUpdateOperationGroupTagsByFilter(payload) {
  function __bulkUpdateOperationGroupTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.operation_group",
      method: "POST",
      methodFqn: "wix.restaurants.v1.OperationGroupService.BulkUpdateOperationGroupTagsByFilter",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1OperationGroupServiceUrl({
        protoPath: "/v1/bulk/operation-groups/update-tags-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateOperationGroupTagsByFilter;
}

// src/restaurants-v1-operation-group-operation-groups.types.ts
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/restaurants-v1-operation-group-operation-groups.meta.ts
function createOperationGroup2() {
  const payload = {};
  const getRequestOptions = createOperationGroup(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/operation-groups",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getOperationGroup2() {
  const payload = { operationGroupId: ":operationGroupId" };
  const getRequestOptions = getOperationGroup(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/operation-groups/{operationGroupId}",
    pathParams: { operationGroupId: "operationGroupId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateOperationGroup2() {
  const payload = { operationGroup: { id: ":operationGroupId" } };
  const getRequestOptions = updateOperationGroup(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/operation-groups/{operationGroup.id}",
    pathParams: { operationGroupId: "operationGroupId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteOperationGroup2() {
  const payload = { operationGroupId: ":operationGroupId" };
  const getRequestOptions = deleteOperationGroup(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/operation-groups/{operationGroupId}",
    pathParams: { operationGroupId: "operationGroupId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryOperationGroups2() {
  const payload = {};
  const getRequestOptions = queryOperationGroups(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/operation-groups/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkCreateOperationGroups2() {
  const payload = {};
  const getRequestOptions = bulkCreateOperationGroups(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/operation-groups/create",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateOperationGroups2() {
  const payload = {};
  const getRequestOptions = bulkUpdateOperationGroups(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/operation-groups/update",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkDeleteOperationGroups2() {
  const payload = {};
  const getRequestOptions = bulkDeleteOperationGroups(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/operation-groups/delete",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateOperationGroupTags2() {
  const payload = {};
  const getRequestOptions = bulkUpdateOperationGroupTags(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/operation-groups/update-tags",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateOperationGroupTagsByFilter2() {
  const payload = {};
  const getRequestOptions = bulkUpdateOperationGroupTagsByFilter(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/operation-groups/update-tags-by-filter",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  SortOrder as SortOrderOriginal,
  WebhookIdentityType as WebhookIdentityTypeOriginal,
  bulkCreateOperationGroups2 as bulkCreateOperationGroups,
  bulkDeleteOperationGroups2 as bulkDeleteOperationGroups,
  bulkUpdateOperationGroupTags2 as bulkUpdateOperationGroupTags,
  bulkUpdateOperationGroupTagsByFilter2 as bulkUpdateOperationGroupTagsByFilter,
  bulkUpdateOperationGroups2 as bulkUpdateOperationGroups,
  createOperationGroup2 as createOperationGroup,
  deleteOperationGroup2 as deleteOperationGroup,
  getOperationGroup2 as getOperationGroup,
  queryOperationGroups2 as queryOperationGroups,
  updateOperationGroup2 as updateOperationGroup
};
