// @wix/headless-online-programs/cjs/dist/react/CategoryList.js
import { jsx as _jsx, Fragment as _Fragment } from "react/jsx-runtime";
import React from 'react';
import { useService } from '@wix/services-manager-react';
import { CategoryListServiceDefinition, } from '../services/category-list-service.js';
import * as CoreCategoryList from './core/CategoryList.js';
import { Category } from './Category.js';
var TestIds;
(function (TestIds) {
    TestIds["categoryListItems"] = "program-category-list-items";
    TestIds["categoryListItem"] = "program-category-list-item";
})(TestIds || (TestIds = {}));
/**
 * Root component that provides the CategoryList service for rendering category lists.
 *
 * @order 1
 * @component
 * @example
 * ```tsx
 * import { CategoryList } from '@wix/online-programs/components';
 *
 * function CategoryListPage(props) {
 *  const { categories } = props;
 *
 *   return (
 *     <CategoryList.Root categoryListConfig={{ categories }}>
 *       <CategoryList.Categories>
 *         <CategoryList.CategoryRepeater>
 *           <Category.Label />
 *           <Category.Id />
 *         </CategoryList.CategoryRepeater>
 *       </CategoryList.Categories>
 *     </CategoryList.Root>
 *   );
 * }
 * ```
 */
function Root(props) {
    const { children, categoryListConfig } = props;
    const serviceConfig = categoryListConfig || {
        categories: [],
    };
    return (_jsx(CoreCategoryList.Root, { categoryListConfig: serviceConfig, children: children }));
}
Root.displayName = 'CategoryList.Root';
/**
 * Raw component that provides direct access to category list data.
 * Should only be used when you need custom access to list data.
 *
 * @component
 * @example
 * ```tsx
 * <CategoryList.Root categoryListConfig={{ categories }}>
 *  <CategoryList.Raw>
 *    {({ categories }) => (
 *      <div>
 *        {categories.map((category) => (
 *          <div key={category._id}>{category.label}</div>
 *        ))}
 *      </div>
 *    )}
 *  </CategoryList.Raw>
 * </CategoryList.Root>
 * ```
 */
const Raw = React.forwardRef((props, _ref) => {
    const { children } = props;
    const categoryListService = useService(CategoryListServiceDefinition);
    const categories = categoryListService.categories.get();
    return typeof children === 'function' ? children({ categories }) : children;
});
Raw.displayName = 'CategoryList.Raw';
/**
 * Container for the category list with empty state support.
 * Follows List Container Level pattern.
 *
 * @component
 * @example
 * ```tsx
 * <CategoryList.Categories emptyState={<div>No categories found</div>}>
 *   <CategoryList.CategoryRepeater>
 *     <Category.Label />
 *     <Category.Id />
 *   </CategoryList.CategoryRepeater>
 * </CategoryList.Categories>
 * ```
 */
const Categories = React.forwardRef((props, ref) => {
    const { children, className, emptyState } = props;
    const categoryListService = useService(CategoryListServiceDefinition);
    const categories = categoryListService.categories.get();
    const hasCategories = categories.length > 0;
    if (!hasCategories) {
        return emptyState || null;
    }
    const attributes = {
        'data-testid': TestIds.categoryListItems,
    };
    return (_jsx("div", { ref: ref, className: className, ...attributes, children: children }));
});
Categories.displayName = 'CategoryList.Categories';
/**
 * Repeater component that renders Category.Root for each category.
 * Follows Repeater Level pattern.
 * Note: Repeater components do NOT support asChild as per architecture rules.
 *
 * @component
 * @example
 * ```tsx
 * <CategoryList.CategoryRepeater>
 *   <Category.Label />
 *   <Category.Id />
 * </CategoryList.CategoryRepeater>
 * ```
 */
const CategoryRepeater = React.forwardRef((props, _ref) => {
    const { children } = props;
    const categoryListService = useService(CategoryListServiceDefinition);
    const categories = categoryListService.categories.get();
    return (_jsx(_Fragment, { children: categories.map((category) => (_jsx(Category.Root, { category: category, "data-testid": TestIds.categoryListItem, children: children }, category._id))) }));
});
CategoryRepeater.displayName = 'CategoryList.CategoryRepeater';
/**
 * Compound component for CategoryList with all sub-components
 */
export const CategoryList = {
    Root,
    Raw,
    Categories,
    CategoryRepeater,
};
