// @wix/headless-online-programs/cjs/dist/react/Instructor.js
import { jsx as _jsx } from "react/jsx-runtime";
import React from 'react';
import { AsChildSlot } from '@wix/headless-utils/react';
const InstructorContext = React.createContext(null);
function useInstructorContext() {
    const context = React.useContext(InstructorContext);
    if (!context) {
        throw new Error('useInstructorContext must be used within a Instructor.Root component');
    }
    return context;
}
var TestIds;
(function (TestIds) {
    TestIds["instructor"] = "program-instructor";
    TestIds["instructorName"] = "program-instructor-name";
    TestIds["instructorDescription"] = "program-instructor-description";
    TestIds["instructorImage"] = "program-instructor-image";
})(TestIds || (TestIds = {}));
/**
 * Root component that provides all necessary service contexts for a complete instructor experience.
 *
 * @order 1
 * @component
 * @example
 * ```tsx
 * import { Instructor } from '@wix/online-programs/components';
 *
 * function InstructorCard({ instructor }) {
 *   return (
 *     <Instructor.Root instructor={instructor}>
 *       <Instructor.Name />
 *       <Instructor.Description />
 *     </Instructor.Root>
 *   );
 * }
 * ```
 */
function Root(props) {
    const { asChild, children, instructor } = props;
    return (_jsx(InstructorContext.Provider, { value: { instructor }, children: _jsx(AsChildSlot, { asChild: asChild, "data-testid": TestIds.instructor, customElement: children, customElementProps: { instructor }, content: instructor.name, children: _jsx("div", { children: children }) }) }));
}
/**
 * Displays the instructor name with customizable rendering following the documented API.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Instructor.Name />
 *
 * // asChild with primitive
 * <Instructor.Name asChild>
 *   <p />
 * </Instructor.Name>
 *
 * // asChild with React component
 * <Instructor.Name asChild>
 *   {React.forwardRef(({ name, slug }, ref) => (
 *     <p ref={ref}>
 *       {name}
 *     </p>
 *   ))}
 * </Instructor.Name>
 * ```
 */
const Name = React.forwardRef((props, ref) => {
    const { asChild, children } = props;
    const { instructor } = useInstructorContext();
    const name = instructor.name || '';
    const slug = instructor.slug || null;
    return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, "data-testid": TestIds.instructorName, customElement: children, customElementProps: { name, slug }, content: name, children: _jsx("p", { children: name }) }));
});
Name.displayName = 'Instructor.Name';
/**
 * Displays the instructor description with customizable rendering following the documented API.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Instructor.Description />
 *
 * // asChild with primitive
 * <Instructor.Description asChild>
 *   <ps />
 * </Instructor.Description>
 *
 * // asChild with React component
 * <Instructor.Description asChild>
 *   {React.forwardRef(({ description }, ref) => (
 *     <p ref={ref}>
 *       {description}
 *     </p>
 *   ))}
 * </Instructor.Description>
 * ```
 */
const Description = React.forwardRef((props, ref) => {
    const { asChild, children } = props;
    const { instructor } = useInstructorContext();
    const description = instructor.description || '';
    return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, "data-testid": TestIds.instructorDescription, customElement: children, customElementProps: { description }, content: description, children: _jsx("p", { children: description }) }));
});
Description.displayName = 'Instructor.Description';
/**
 * Displays the instructor's profile photo with customizable rendering following the documented API.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Instructor.Image />
 *
 * // asChild with primitive
 * <Instructor.Image asChild>
 *   <img />
 * </Instructor.Image>
 *
 * // asChild with React component
 * <Instructor.Image asChild>
 *   {React.forwardRef(({ src, altText }, ref) => (
 *     <img
 *       ref={ref}
 *       src={src}
 *       alt={altText}
 *     />
 *   ))}
 * </Instructor.Image>
 * ```
 */
const Image = React.forwardRef((props, ref) => {
    const { asChild, children, className } = props;
    const { instructor } = useInstructorContext();
    if (!instructor.photo) {
        return null;
    }
    const src = instructor.photo.url || '';
    const altText = instructor.photoAltText || '';
    return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.instructorImage, customElement: children, customElementProps: {
            src,
            altText,
        }, children: _jsx("img", { src: src, alt: altText }) }));
});
Image.displayName = 'Instructor.Image';
/**
 * Compound component for Instructor with all sub-components
 */
export const Instructor = {
    Root,
    Name,
    Description,
    Image,
};
