// @wix/headless-online-programs/cjs/dist/react/Program.js
import { jsx as _jsx } from "react/jsx-runtime";
import React from 'react';
import { useService } from '@wix/services-manager-react';
import { AsChildSlot } from '@wix/headless-utils/react';
import { WixMediaImage } from '@wix/headless-media/react';
import * as CoreProgram from './core/Program.js';
import { Instructor } from './Instructor.js';
import { InstructorsServiceDefinition } from '../services/instructors-service.js';
var TestIds;
(function (TestIds) {
    TestIds["programTitle"] = "program-title";
    TestIds["programImage"] = "program-image";
    TestIds["programDuration"] = "program-duration";
    TestIds["programDescription"] = "program-description";
    TestIds["programPrice"] = "program-price";
    TestIds["programStepCount"] = "program-step-count";
    TestIds["programSectionCount"] = "program-section-count";
    TestIds["programInstructors"] = "program-instructors";
    TestIds["programInstructorRepeater"] = "program-instructor-repeater";
    TestIds["programRaw"] = "program-raw";
    TestIds["programId"] = "program-id";
})(TestIds || (TestIds = {}));
/**
 * Root component that provides all necessary service contexts for a complete program experience.
 *
 * @order 1
 * @component
 * @example
 * ```tsx
 * import { Program } from '@wix/online-programs/components';
 *
 * function ProgramPage({ program }) {
 *   return (
 *     <Program.Root program={program}>
 *       <Program.Title className="text-4xl font-bold" />
 *     </Program.Root>
 *   );
 * }
 * ```
 */
function Root(props) {
    const { children, program } = props;
    return (_jsx(CoreProgram.Root, { programServiceConfig: { program }, children: children }));
}
/**
 * Provides direct access to program data. Should be used only in rare cases and never by Wix implementations.
 *
 * @component
 * @example
 * ```tsx
 * // asChild with React component
 * <Program.Raw asChild>
 *   {React.forwardRef(({ program }, ref) => (
 *     <div ref={ref}>
 *       <span>Program ID: {program._id}</span>
 *       <span>Title: {program.description?.title || 'No title'}</span>
 *       <span>Price: {program.price?.value || 'No price'}</span>
 *     </div>
 *   ))}
 * </Program.Raw>
 * ```
 */
const Raw = React.forwardRef((props, ref) => {
    const { asChild, children, className } = props;
    return (_jsx(CoreProgram.Raw, { children: ({ program }) => {
            // Raw component should not render anything by default when not using asChild
            if (!asChild || !children) {
                return null;
            }
            return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.programRaw, customElement: children, customElementProps: { program } }));
        } }));
});
Raw.displayName = 'Program.Raw';
/**
 * Displays the program ID with customizable rendering following the documented API.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Program.Id />
 *
 * // asChild with React component
 * <Program.Id asChild>
 *   {React.forwardRef(({ id }, ref) => (
 *     <a ref={ref} href={`/program/${id}`}>
 *       View Program
 *     </a>
 *   ))}
 * </Program.Id>
 * ```
 */
const Id = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreProgram.Id, { children: ({ id }) => {
            return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.programId, customElement: children, customElementProps: { id }, content: id, ...otherProps, children: _jsx("span", { children: id }) }));
        } }));
});
Id.displayName = 'Program.Id';
/**
 * Displays the program title with customizable rendering following the documented API.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Program.Title className="text-4xl font-bold" />
 *
 * // asChild with primitive
 * <Program.Title asChild>
 *   <h1 className="text-4xl font-bold" />
 * </Program.Title>
 *
 * // asChild with react component
 * <Program.Title asChild>
 *   {React.forwardRef(({ title, ...props }, ref) => (
 *     <h1 ref={ref} { ...props } className="text-4xl font-bold">
 *       {title}
 *     </h1>
 *   ))}
 * </Program.Title>
 * ```
 */
const Title = React.forwardRef((props, ref) => {
    const { asChild, className, children } = props;
    return (_jsx(CoreProgram.Title, { children: ({ title }) => {
            return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.programTitle, customElement: children, customElementProps: { title }, content: title, children: _jsx("h1", { children: title }) }));
        } }));
});
Title.displayName = 'Program.Title';
/**
 * Displays the program description text with customizable rendering.
 * Data source: program.description.details
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Program.Description />
 *
 * // asChild with primitive
 * <Program.Description asChild>
 *   <p />
 * </Program.Description>
 *
 * // Custom rendering
 * <Program.Description asChild>
 *   {React.forwardRef(({ description, ...props }, ref) => (
 *     <p ref={ref} {...props}>
 *       {description}
 *     </p>
 *   ))}
 * </Program.Description>
 * ```
 */
const Description = React.forwardRef((props, ref) => {
    const { asChild, children, className } = props;
    return (_jsx(CoreProgram.Description, { children: ({ description }) => {
            return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.programDescription, customElement: children, customElementProps: { description }, content: description, children: _jsx("p", { children: description }) }));
        } }));
});
Description.displayName = 'Program.Description';
/**
 * Displays the program image using WixMediaImage for optimization.
 * Supports custom rendering via asChild pattern with specific src/alt props.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage with WixMediaImage
 * <Program.Image className="w-full h-48 object-cover rounded-lg" />
 *
 * // asChild with React component
 * <Program.Image asChild>
 *   {React.forwardRef(({ src, alt }, ref) => (
 *     <img ref={ref} src={src} alt={alt} />
 *   ))}
 * </Program.Image>
 * ```
 */
const Image = React.forwardRef((props, ref) => {
    const { asChild, children, ...otherProps } = props;
    return (_jsx(CoreProgram.Image, { children: ({ src, alt }) => {
            if (asChild && children) {
                // Call the ForwardRefRenderFunction with the specific props
                return children({ src, alt }, ref);
            }
            return (_jsx(WixMediaImage, { ref: ref, media: { image: src }, alt: alt, "data-testid": TestIds.programImage, ...otherProps }));
        } }));
});
Image.displayName = 'Program.Image';
/**
 * Displays the program duration in days with customizable rendering.
 * Data source: program.timeline field. If the program is self-paced, duration has no limit.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Program.DurationInDays />
 *
 * // asChild with primitive
 * <Program.DurationInDays asChild>
 *   <p />
 * </Program.DurationInDays>
 *
 * // asChild with React component
 * <Program.DurationInDays asChild>
 *   {React.forwardRef(({ durationInDays, isSelfPaced, ...props }, ref) => (
 *     <p ref={ref} {...props}>
 *       {isSelfPaced ? 'No Time Limit' : `${durationInDays} days`}
 *     </p>
 *   ))}
 * </Program.DurationInDays>
 * ```
 */
const DurationInDays = React.forwardRef((props, ref) => {
    const { asChild, children } = props;
    return (_jsx(CoreProgram.DurationInDays, { children: ({ durationInDays, isSelfPaced }) => {
            const dataAttributes = {
                'data-testid': TestIds.programDuration,
                'data-type': isSelfPaced ? 'self-paced' : 'time-limited',
            };
            return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, customElement: children, customElementProps: { durationInDays, isSelfPaced }, content: durationInDays, ...dataAttributes, children: _jsx("div", { children: durationInDays }) }));
        } }));
});
DurationInDays.displayName = 'Program.DurationInDays';
/**
 * Displays the program price with customizable rendering following the documented API.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Program.Price />
 *
 * // asChild with primitive
 * <Program.Price asChild>
 *   <span />
 * </Program.Price>
 *
 * // asChild with React component
 * <Program.Price asChild>
 *   {React.forwardRef(({ price, currency }, ref) => (
 *     <div ref={ref}>
 *       <span>{price}</span>
 *       <span>{currency}</span>
 *     </div>
 *   ))}
 * </Program.Price>
 *
 * // With free program handling
 * <Program.Price asChild>
 *   {React.forwardRef(({ price, formattedPrice, ...props }, ref) => (
 *     <span ref={ref} {...props} className="text-2xl font-bold text-brand-primary">
 *       {price ? formattedPrice : 'Free'}
 *     </span>
 *   ))}
 * </Program.Price>
 * ```
 */
const Price = React.forwardRef((props, ref) => {
    const { asChild, children, className } = props;
    return (_jsx(CoreProgram.Price, { children: ({ price, formattedPrice, currency }) => {
            return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.programPrice, customElement: children, customElementProps: {
                    formattedPrice,
                    price,
                    currency,
                }, content: formattedPrice, children: _jsx("span", { children: formattedPrice }) }));
        } }));
});
Price.displayName = 'Program.Price';
/**
 * Displays the program steps count with customizable rendering.
 * Data source: program.contentSummary.stepCount
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Program.StepCount className="text-content-secondary" />
 *
 * // asChild with primitive
 * <Program.StepCount asChild>
 *   <p className="text-content-secondary" />
 * </Program.StepCount>
 *
 * // asChild with React component
 * <Program.StepCount asChild>
 *   {React.forwardRef(({ stepCount }, ref) => (
 *     <p ref={ref}>
 *       {stepCount} Steps
 *     </p>
 *   ))}
 * </Program.StepCount>
 * ```
 */
const StepCount = React.forwardRef((props, ref) => {
    const { asChild, children, className } = props;
    return (_jsx(CoreProgram.StepCount, { children: ({ stepCount }) => {
            return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.programStepCount, customElement: children, customElementProps: { stepCount }, content: stepCount, children: _jsx("p", { children: stepCount }) }));
        } }));
});
StepCount.displayName = 'Program.StepCount';
/**
 * Displays the program section count with customizable rendering.
 * Data source: program.contentSummary.sectionCount
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Program.SectionCount />
 *
 * // asChild with primitive
 * <Program.SectionCount asChild>
 *   <p />
 * </Program.SectionCount>
 *
 * // asChild with React component
 * <Program.SectionCount asChild>
 *   {React.forwardRef(({ sectionCount }, ref) => (
 *     <p ref={ref}>
 *       {sectionCount} Sections
 *     </p>
 *   ))}
 * </Program.SectionCount>
 * ```
 */
const SectionCount = React.forwardRef((props, ref) => {
    const { asChild, children, className } = props;
    return (_jsx(CoreProgram.SectionCount, { children: ({ sectionCount }) => {
            return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.programSectionCount, customElement: children, customElementProps: { sectionCount }, content: sectionCount, children: _jsx("div", { children: sectionCount }) }));
        } }));
});
SectionCount.displayName = 'Program.SectionCount';
/**
 * Displays the program instructors with customizable rendering following the documented API.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Program.Instructors instructors={instructors}>
 *  <Program.InstructorRepeater>
 *    <Instructor.Name />
 *    <Instructor.Description />
 *  </Program.InstructorRepeater>
 * </Program.Instructors>
 * ```
 */
const Instructors = React.forwardRef((props, ref) => {
    const { asChild, children, instructors = [], emptyState } = props;
    return (_jsx(CoreProgram.Instructors, { instructorsServiceConfig: { instructors }, children: ({ instructors }) => {
            if (!instructors.length) {
                return emptyState || null;
            }
            return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, "data-testid": TestIds.programInstructors, customElement: children, customElementProps: { instructors }, children: children }));
        } }));
});
Instructors.displayName = 'Program.Instructors';
/**
 * Repeater component that renders children for each instructor.
 * Maps over instructors and renders Instructor.Root for each.
 *
 * @component
 * @example
 * ```tsx
 * // With React elements
 * <Program.InstructorRepeater>
 *   <Instructor.Name />
 *   <Instructor.Description />
 * </Program.InstructorRepeater>
 *
 * // With custom wrapper
 * <Program.InstructorRepeater>
 *   <div className="instructor-card">
 *     <Instructor.Name />
 *     <Instructor.Description />
 *   </div>
 * </Program.InstructorRepeater>
 * ```
 */
const InstructorRepeater = React.forwardRef((props, ref) => {
    const { asChild, children, className } = props;
    const service = useService(InstructorsServiceDefinition);
    const instructors = service.instructors.get();
    return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.programInstructorRepeater, customElement: children, customElementProps: {}, children: _jsx("div", { children: instructors.map((instructor, index) => {
                return (_jsx(Instructor.Root, { instructor: instructor, children: children }, instructor.userId || index));
            }) }) }));
});
InstructorRepeater.displayName = 'Program.InstructorRepeater';
/**
 * Compound component for Program with all sub-components
 */
export const Program = {
    Root,
    Raw,
    Id,
    Title,
    Image,
    DurationInDays,
    Price,
    Description,
    StepCount,
    SectionCount,
    Instructors,
    InstructorRepeater,
};
