// @wix/headless-online-programs/cjs/dist/react/ProgramList.js
import { jsx as _jsx, Fragment as _Fragment } from "react/jsx-runtime";
import React from 'react';
import { useService } from '@wix/services-manager-react';
import { ProgramListServiceDefinition, } from '../services/program-list-service.js';
import * as CoreProgramList from './core/ProgramList.js';
import { Program } from './Program.js';
var TestIds;
(function (TestIds) {
    TestIds["programListItems"] = "program-list-items";
    TestIds["programListItem"] = "program-list-item";
})(TestIds || (TestIds = {}));
/**
 * Root component that provides the ProgramList service for rendering program lists.
 *
 * @order 1
 * @component
 * @example
 * ```tsx
 * import { ProgramList } from '@wix/online-programs/components';
 *
 * function ProgramListPage(props) {
 *  const { programs } = props;
 *
 *   return (
 *     <ProgramList.Root programListConfig={{ programs }}>
 *       <ProgramList.Programs>
 *         <ProgramList.ProgramRepeater>
 *           <Program.Title />
 *           <Program.Description />
 *         </ProgramList.ProgramRepeater>
 *       </ProgramList.Programs>
 *     </ProgramList.Root>
 *   );
 * }
 * ```
 */
function Root(props) {
    const { children, programListConfig } = props;
    const serviceConfig = programListConfig || {
        programs: [],
    };
    return (_jsx(CoreProgramList.Root, { programListConfig: serviceConfig, children: children }));
}
Root.displayName = 'ProgramList.Root';
/**
 * Raw component that provides direct access to program list data.
 * Should only be used when you need custom access to list data.
 *
 * @component
 * @example
 * ```tsx
 * <ProgramList.Root programListConfig={{ programs }}>
 *  <ProgramList.Raw>
 *    {({ programs }) => (
 *      <div>
 *        {programs.map((program) => (
 *          <div key={program._id}>{program.description?.title || 'No title'}</div>
 *        ))}
 *      </div>
 *    )}
 *  </ProgramList.Raw>
 * </ProgramList.Root>
 * ```
 */
const Raw = React.forwardRef((props, _ref) => {
    const { children } = props;
    const programListService = useService(ProgramListServiceDefinition);
    const programs = programListService.programs.get();
    return typeof children === 'function' ? children({ programs }) : children;
});
Raw.displayName = 'ProgramList.Raw';
/**
 * Container for the program list with empty state support.
 * Follows List Container Level pattern.
 *
 * @component
 * @example
 * ```tsx
 * <ProgramList.Programs emptyState={<div>No programs found</div>}>
 *   <ProgramList.ProgramRepeater>
 *     <Program.Title />
 *     <Program.Description />
 *   </ProgramList.ProgramRepeater>
 * </ProgramList.Programs>
 * ```
 */
const Programs = React.forwardRef((props, ref) => {
    const { children, className, emptyState } = props;
    const programListService = useService(ProgramListServiceDefinition);
    const programs = programListService.programs.get();
    const hasPrograms = programs.length > 0;
    if (!hasPrograms) {
        return emptyState || null;
    }
    const attributes = {
        'data-testid': TestIds.programListItems,
    };
    return (_jsx("div", { ref: ref, className: className, ...attributes, children: children }));
});
Programs.displayName = 'ProgramList.Programs';
/**
 * Repeater component that renders Program.Root for each program.
 * Follows Repeater Level pattern.
 * Note: Repeater components do NOT support asChild as per architecture rules.
 *
 * @component
 * @example
 * ```tsx
 * <ProgramList.ProgramRepeater>
 *   <Program.Title />
 *   <Program.Description />
 * </ProgramList.ProgramRepeater>
 * ```
 */
const ProgramRepeater = React.forwardRef((props, _ref) => {
    const { children } = props;
    const programListService = useService(ProgramListServiceDefinition);
    const programs = programListService.programs.get();
    return (_jsx(_Fragment, { children: programs.map((program) => (_jsx(Program.Root, { program: program, "data-testid": TestIds.programListItem, children: children }, program._id))) }));
});
ProgramRepeater.displayName = 'ProgramList.ProgramRepeater';
/**
 * Compound component for ProgramList with all sub-components
 */
export const ProgramList = {
    Root,
    Raw,
    Programs,
    ProgramRepeater,
};
