// @wix/auto_sdk_payments_dispute-evidence-documents/build/internal/es/meta.mjs
// src/payments-dispute-evidence-documents-v1-dispute-evidence-document-dispute-evidence-documents.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(opts) {
  const domainToMappings = {};
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_dispute-evidence-documents";
function createDisputeEvidenceDocument(payload) {
  function __createDisputeEvidenceDocument({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "disputeEvidenceDocument.createdDate" },
          { path: "disputeEvidenceDocument.updatedDate" },
          { path: "disputeEvidenceDocument.submittedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.CreateDisputeEvidenceDocument",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents",
          data: serializedData,
          host
        }
      ),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "disputeEvidenceDocument.createdDate" },
            { path: "disputeEvidenceDocument.updatedDate" },
            { path: "disputeEvidenceDocument.submittedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createDisputeEvidenceDocument;
}
function generateDisputeEvidenceDocumentUploadUrl(payload) {
  function __generateDisputeEvidenceDocumentUploadUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.GenerateDisputeEvidenceDocumentUploadUrl",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}/generate-upload-url",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __generateDisputeEvidenceDocumentUploadUrl;
}
function attachDisputeEvidenceDocumentFile(payload) {
  function __attachDisputeEvidenceDocumentFile({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.AttachDisputeEvidenceDocumentFile",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}/attach-file",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "disputeEvidenceDocument.createdDate" },
            { path: "disputeEvidenceDocument.updatedDate" },
            { path: "disputeEvidenceDocument.submittedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __attachDisputeEvidenceDocumentFile;
}
function getDisputeEvidenceDocument(payload) {
  function __getDisputeEvidenceDocument({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "GET",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.GetDisputeEvidenceDocument",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}",
          data: payload,
          host
        }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "disputeEvidenceDocument.createdDate" },
            { path: "disputeEvidenceDocument.updatedDate" },
            { path: "disputeEvidenceDocument.submittedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getDisputeEvidenceDocument;
}
function generateDisputeEvidenceDocumentDownloadUrl(payload) {
  function __generateDisputeEvidenceDocumentDownloadUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.GenerateDisputeEvidenceDocumentDownloadUrl",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}/generate-download-url",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "urlExpirationDate" }]
        }
      ])
    };
    return metadata;
  }
  return __generateDisputeEvidenceDocumentDownloadUrl;
}
function markDisputeEvidenceDocumentSubmitted(payload) {
  function __markDisputeEvidenceDocumentSubmitted({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.MarkDisputeEvidenceDocumentSubmitted",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}/mark-submitted",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __markDisputeEvidenceDocumentSubmitted;
}
function deleteDisputeEvidenceDocument(payload) {
  function __deleteDisputeEvidenceDocument({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "DELETE",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.DeleteDisputeEvidenceDocument",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}",
          data: payload,
          host
        }
      ),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteDisputeEvidenceDocument;
}
function queryDisputeEvidenceDocuments(payload) {
  function __queryDisputeEvidenceDocuments({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.QueryDisputeEvidenceDocuments",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents/query",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "disputeEvidenceDocuments.createdDate" },
            { path: "disputeEvidenceDocuments.updatedDate" },
            { path: "disputeEvidenceDocuments.submittedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
            {
              protoPath: "/v1/dispute-evidence-documents/query",
              data: payload,
              host
            }
          ),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryDisputeEvidenceDocuments;
}
function bulkUpdateDisputeEvidenceDocumentTags(payload) {
  function __bulkUpdateDisputeEvidenceDocumentTags({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.BulkUpdateDisputeEvidenceDocumentTags",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/bulk/dispute-evidence-documents/update-tags",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateDisputeEvidenceDocumentTags;
}
function bulkUpdateDisputeEvidenceDocumentTagsByFilter(payload) {
  function __bulkUpdateDisputeEvidenceDocumentTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.BulkUpdateDisputeEvidenceDocumentTagsByFilter",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/bulk/dispute-evidence-documents/update-tags-by-filter",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateDisputeEvidenceDocumentTagsByFilter;
}

// src/payments-dispute-evidence-documents-v1-dispute-evidence-document-dispute-evidence-documents.meta.ts
function createDisputeEvidenceDocument2() {
  const payload = {};
  const getRequestOptions = createDisputeEvidenceDocument(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/dispute-evidence-documents",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function generateDisputeEvidenceDocumentUploadUrl2() {
  const payload = {
    disputeEvidenceDocumentId: ":disputeEvidenceDocumentId"
  };
  const getRequestOptions = generateDisputeEvidenceDocumentUploadUrl(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}/generate-upload-url",
    pathParams: { disputeEvidenceDocumentId: "disputeEvidenceDocumentId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function attachDisputeEvidenceDocumentFile2() {
  const payload = {
    disputeEvidenceDocumentId: ":disputeEvidenceDocumentId"
  };
  const getRequestOptions = attachDisputeEvidenceDocumentFile(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}/attach-file",
    pathParams: { disputeEvidenceDocumentId: "disputeEvidenceDocumentId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getDisputeEvidenceDocument2() {
  const payload = {
    disputeEvidenceDocumentId: ":disputeEvidenceDocumentId"
  };
  const getRequestOptions = getDisputeEvidenceDocument(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}",
    pathParams: { disputeEvidenceDocumentId: "disputeEvidenceDocumentId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function generateDisputeEvidenceDocumentDownloadUrl2() {
  const payload = {
    disputeEvidenceDocumentId: ":disputeEvidenceDocumentId"
  };
  const getRequestOptions = generateDisputeEvidenceDocumentDownloadUrl(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}/generate-download-url",
    pathParams: { disputeEvidenceDocumentId: "disputeEvidenceDocumentId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function markDisputeEvidenceDocumentSubmitted2() {
  const payload = {
    disputeEvidenceDocumentId: ":disputeEvidenceDocumentId"
  };
  const getRequestOptions = markDisputeEvidenceDocumentSubmitted(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}/mark-submitted",
    pathParams: { disputeEvidenceDocumentId: "disputeEvidenceDocumentId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteDisputeEvidenceDocument2() {
  const payload = {
    disputeEvidenceDocumentId: ":disputeEvidenceDocumentId"
  };
  const getRequestOptions = deleteDisputeEvidenceDocument(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}",
    pathParams: { disputeEvidenceDocumentId: "disputeEvidenceDocumentId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryDisputeEvidenceDocuments2() {
  const payload = {};
  const getRequestOptions = queryDisputeEvidenceDocuments(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/dispute-evidence-documents/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateDisputeEvidenceDocumentTags2() {
  const payload = {};
  const getRequestOptions = bulkUpdateDisputeEvidenceDocumentTags(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/dispute-evidence-documents/update-tags",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateDisputeEvidenceDocumentTagsByFilter2() {
  const payload = {};
  const getRequestOptions = bulkUpdateDisputeEvidenceDocumentTagsByFilter(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/dispute-evidence-documents/update-tags-by-filter",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  attachDisputeEvidenceDocumentFile2 as attachDisputeEvidenceDocumentFile,
  bulkUpdateDisputeEvidenceDocumentTags2 as bulkUpdateDisputeEvidenceDocumentTags,
  bulkUpdateDisputeEvidenceDocumentTagsByFilter2 as bulkUpdateDisputeEvidenceDocumentTagsByFilter,
  createDisputeEvidenceDocument2 as createDisputeEvidenceDocument,
  deleteDisputeEvidenceDocument2 as deleteDisputeEvidenceDocument,
  generateDisputeEvidenceDocumentDownloadUrl2 as generateDisputeEvidenceDocumentDownloadUrl,
  generateDisputeEvidenceDocumentUploadUrl2 as generateDisputeEvidenceDocumentUploadUrl,
  getDisputeEvidenceDocument2 as getDisputeEvidenceDocument,
  markDisputeEvidenceDocumentSubmitted2 as markDisputeEvidenceDocumentSubmitted,
  queryDisputeEvidenceDocuments2 as queryDisputeEvidenceDocuments
};
