// @wix/auto_sdk_restaurants_item-modifier-groups/build/internal/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  SortOrder: () => SortOrder,
  WebhookIdentityType: () => WebhookIdentityType,
  bulkCreateModifierGroups: () => bulkCreateModifierGroups2,
  bulkDeleteModifierGroups: () => bulkDeleteModifierGroups2,
  bulkUpdateModifierGroups: () => bulkUpdateModifierGroups2,
  countModifierGroups: () => countModifierGroups2,
  createModifierGroup: () => createModifierGroup2,
  deleteModifierGroup: () => deleteModifierGroup2,
  getModifierGroup: () => getModifierGroup2,
  listModifierGroups: () => listModifierGroups2,
  queryModifierGroups: () => queryModifierGroups2,
  updateModifierGroup: () => updateModifierGroup2
});
module.exports = __toCommonJS(index_typings_exports);

// src/restaurants-menus-v1-item-modifier-group-item-modifier-groups.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/restaurants-menus-v1-item-modifier-group-item-modifier-groups.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl(opts) {
  const domainToMappings = {
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-modifier-group",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers-groups/delete",
        destPath: "/v1/bulk/modifiers-groups/delete"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers-groups/update",
        destPath: "/v1/bulk/modifiers-groups/update"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifier-groups/create",
        destPath: "/v1/bulk/modifier-groups/create"
      },
      {
        srcPath: "/restaurants/menus/v1/modifier-groups",
        destPath: "/v1/modifier-groups"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-modifier-group",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-modifier-group",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-modifier-group",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-modifier-group",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-modifier-group",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-modifier-group",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers-groups/delete",
        destPath: "/v1/bulk/modifiers-groups/delete"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers-groups/update",
        destPath: "/v1/bulk/modifiers-groups/update"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifier-groups/create",
        destPath: "/v1/bulk/modifier-groups/create"
      },
      {
        srcPath: "/restaurants/menus/v1/modifier-groups",
        destPath: "/v1/modifier-groups"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-modifier-group",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-modifier-group",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/item-modifier-group",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifier-groups/create",
        destPath: "/v1/bulk/modifier-groups/create"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers-groups/delete",
        destPath: "/v1/bulk/modifiers-groups/delete"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers-groups/update",
        destPath: "/v1/bulk/modifiers-groups/update"
      },
      {
        srcPath: "/restaurants/menus/v1/modifier-groups",
        destPath: "/v1/modifier-groups"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-modifier-group",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_item-modifier-groups";
function createModifierGroup(payload) {
  function __createModifierGroup({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "modifierGroup.createdDate" },
          { path: "modifierGroup.updatedDate" },
          { path: "modifierGroup.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier_group",
      method: "POST",
      methodFqn: "wix.restaurants.menus.v1.ModifierGroupsService.CreateModifierGroup",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl({
        protoPath: "/v1/modifier-groups",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "modifierGroup.createdDate" },
            { path: "modifierGroup.updatedDate" },
            { path: "modifierGroup.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createModifierGroup;
}
function getModifierGroup(payload) {
  function __getModifierGroup({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier_group",
      method: "GET",
      methodFqn: "wix.restaurants.menus.v1.ModifierGroupsService.GetModifierGroup",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl({
        protoPath: "/v1/modifier-groups/{modifierGroupId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "modifierGroup.createdDate" },
            { path: "modifierGroup.updatedDate" },
            { path: "modifierGroup.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getModifierGroup;
}
function listModifierGroups(payload) {
  function __listModifierGroups({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier_group",
      method: "GET",
      methodFqn: "wix.restaurants.menus.v1.ModifierGroupsService.ListModifierGroups",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl({
        protoPath: "/v1/modifier-groups",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "modifierGroups.createdDate" },
            { path: "modifierGroups.updatedDate" },
            { path: "modifierGroups.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listModifierGroups;
}
function queryModifierGroups(payload) {
  function __queryModifierGroups({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier_group",
      method: "POST",
      methodFqn: "wix.restaurants.menus.v1.ModifierGroupsService.QueryModifierGroups",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl({
        protoPath: "/v1/modifier-groups/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "modifierGroups.createdDate" },
            { path: "modifierGroups.updatedDate" },
            { path: "modifierGroups.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryModifierGroups;
}
function countModifierGroups(payload) {
  function __countModifierGroups({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier_group",
      method: "POST",
      methodFqn: "wix.restaurants.menus.v1.ModifierGroupsService.CountModifierGroups",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl({
        protoPath: "/v1/modifier-groups/count",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __countModifierGroups;
}
function updateModifierGroup(payload) {
  function __updateModifierGroup({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "modifierGroup.createdDate" },
          { path: "modifierGroup.updatedDate" },
          { path: "modifierGroup.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier_group",
      method: "PATCH",
      methodFqn: "wix.restaurants.menus.v1.ModifierGroupsService.UpdateModifierGroup",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl({
        protoPath: "/v1/modifier-groups/{modifierGroup.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "modifierGroup.createdDate" },
            { path: "modifierGroup.updatedDate" },
            { path: "modifierGroup.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateModifierGroup;
}
function deleteModifierGroup(payload) {
  function __deleteModifierGroup({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier_group",
      method: "DELETE",
      methodFqn: "wix.restaurants.menus.v1.ModifierGroupsService.DeleteModifierGroup",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl({
        protoPath: "/v1/modifier-groups/{modifierGroupId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteModifierGroup;
}
function bulkCreateModifierGroups(payload) {
  function __bulkCreateModifierGroups({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "modifierGroups.createdDate" },
          { path: "modifierGroups.updatedDate" },
          { path: "modifierGroups.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier_group",
      method: "POST",
      methodFqn: "wix.restaurants.menus.v1.ModifierGroupsService.BulkCreateModifierGroups",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl({
        protoPath: "/v1/bulk/modifier-groups/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.modifierGroup.createdDate" },
            { path: "results.modifierGroup.updatedDate" },
            {
              path: "results.modifierGroup.externalReferenceInfo.lastSyncDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateModifierGroups;
}
function bulkUpdateModifierGroups(payload) {
  function __bulkUpdateModifierGroups({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "modifierGroups.mask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "modifierGroups.modifierGroup.createdDate" },
          { path: "modifierGroups.modifierGroup.updatedDate" },
          {
            path: "modifierGroups.modifierGroup.externalReferenceInfo.lastSyncDate"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier_group",
      method: "POST",
      methodFqn: "wix.restaurants.menus.v1.ModifierGroupsService.BulkUpdateModifierGroups",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl({
        protoPath: "/v1/bulk/modifiers-groups/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.modifierGroup.createdDate" },
            { path: "results.modifierGroup.updatedDate" },
            {
              path: "results.modifierGroup.externalReferenceInfo.lastSyncDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateModifierGroups;
}
function bulkDeleteModifierGroups(payload) {
  function __bulkDeleteModifierGroups({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier_group",
      method: "DELETE",
      methodFqn: "wix.restaurants.menus.v1.ModifierGroupsService.BulkDeleteModifierGroups",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl({
        protoPath: "/v1/bulk/modifiers-groups/delete",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __bulkDeleteModifierGroups;
}

// src/restaurants-menus-v1-item-modifier-group-item-modifier-groups.universal.ts
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createModifierGroup2(modifierGroup) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    modifierGroup
  });
  const reqOpts = createModifierGroup(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.modifierGroup;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { modifierGroup: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["modifierGroup"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getModifierGroup2(modifierGroupId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    modifierGroupId
  });
  const reqOpts = getModifierGroup(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.modifierGroup;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { modifierGroupId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["modifierGroupId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listModifierGroups2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    modifierGroupIds: options?.modifierGroupIds,
    paging: options?.paging
  });
  const reqOpts = listModifierGroups(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          modifierGroupIds: "$[0].modifierGroupIds",
          paging: "$[0].paging"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryModifierGroups2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryModifierGroups(
        payload
      );
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [])
      );
      return {
        items: transformedData?.modifierGroups,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function countModifierGroups2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    filter: options?.filter
  });
  const reqOpts = countModifierGroups(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { filter: "$[0].filter" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateModifierGroup2(_id, modifierGroup) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    modifierGroup: { ...modifierGroup, id: _id }
  });
  const reqOpts = updateModifierGroup(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.modifierGroup;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { modifierGroup: "$[1]" },
        explicitPathsToArguments: { "modifierGroup.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "modifierGroup"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteModifierGroup2(modifierGroupId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    modifierGroupId
  });
  const reqOpts = deleteModifierGroup(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { modifierGroupId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["modifierGroupId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCreateModifierGroups2(modifierGroups, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    modifierGroups,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkCreateModifierGroups(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          modifierGroups: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["modifierGroups", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateModifierGroups2(modifierGroups, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    modifierGroups,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkUpdateModifierGroups(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          modifierGroups: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["modifierGroups", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkDeleteModifierGroups2(ids) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ ids });
  const reqOpts = bulkDeleteModifierGroups(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { ids: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["ids"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  SortOrder,
  WebhookIdentityType,
  bulkCreateModifierGroups,
  bulkDeleteModifierGroups,
  bulkUpdateModifierGroups,
  countModifierGroups,
  createModifierGroup,
  deleteModifierGroup,
  getModifierGroup,
  listModifierGroups,
  queryModifierGroups,
  updateModifierGroup
});
