// @wix/headless-bookings/dist/react/core/Service.js
import { jsx as _jsx } from "react/jsx-runtime";
import { useService, WixServices } from '@wix/services-manager-react';
import { createServicesMap } from '@wix/services-manager';
import { ServiceService, ServiceServiceDefinition, } from '../../services/service-service.js';
import { WixMediaImage } from '@wix/headless-media/react';
function Content(props) {
    const serviceService = useService(ServiceServiceDefinition);
    const service = serviceService.service.get();
    return props.children({ service });
}
function Loading(props) {
    const serviceService = useService(ServiceServiceDefinition);
    const isLoading = serviceService.isLoading.get();
    return props.children({ isLoading });
}
export function Name(props) {
    const serviceService = useService(ServiceServiceDefinition);
    const service = serviceService.service.get();
    if (!service)
        return null;
    return props.children({ name: service.name || '' });
}
export function Description(props) {
    const serviceService = useService(ServiceServiceDefinition);
    const service = serviceService.service.get();
    if (!service)
        return null;
    return props.children({ description: service.description || null });
}
export function Price(props) {
    const serviceService = useService(ServiceServiceDefinition);
    const service = serviceService.service.get();
    if (!service)
        return null;
    const price = service.payment?.fixed?.price
        ? {
            value: Number(service.payment.fixed.price.value),
            currency: service.payment.fixed.price.currency || '',
        }
        : null;
    return props.children({ price });
}
export function Image() {
    const serviceService = useService(ServiceServiceDefinition);
    const service = serviceService.service.get();
    if (!service || !service.media?.mainMedia?.image)
        return null;
    //WA until the issue with media will be solved
    const image = service.media?.mainMedia?.image?.replace(/v1\/[\w-]+\//, 'v1/');
    return _jsx(WixMediaImage, { media: { image }, alt: service.name || '' });
}
export function Root(props) {
    const { serviceConfig, children } = props;
    const services = createServicesMap().addService(ServiceServiceDefinition, ServiceService, serviceConfig);
    return _jsx(WixServices, { servicesMap: services, children: children });
}
// Set display names for components
Root.displayName = 'Service.Root';
Content.displayName = 'Service.Content';
Loading.displayName = 'Service.Loading';
Name.displayName = 'Service.Name';
Description.displayName = 'Service.Description';
Price.displayName = 'Service.Price';
Image.displayName = 'Service.Image';
export const Service = {
    Root,
    Content,
    Loading,
    Name,
    Description,
    Price,
    Image,
};
