// @wix/headless-bookings/dist/react/core/ServiceList.js
import { jsx as _jsx } from "react/jsx-runtime";
import { useService, WixServices } from '@wix/services-manager-react';
import { createServicesMap } from '@wix/services-manager';
import { ServicesListService, ServicesListServiceDefinition, } from '../../services/services-list-service.js';
/**
 * Root component that provides ServiceList service context to its children.
 * This component sets up the necessary services for managing services list state,
 * including filtering, sorting, and pagination functionality.
 *
 * @order 1
 * @component
 * @example
 * ```tsx
 * import { ServiceList, Service } from '@wix/headless-services/react';
 *
 * function ServiceListPage() {
 *   return (
 *     <ServiceList.Root
 *       servicesListConfig={{
 *         services: myServices,
 *         searchOptions: {
 *           cursorPaging: { limit: 10 },
 *           sort: [{ fieldName: 'name', order: 'ASC' }]
 *         },
 *         pagingMetadata: { count: 10, hasNext: true }
 *       }}
 *     >
 *       <ServiceList.Options>
 *         <ServiceList.ServiceRepeater>
 *           <Service.Name>
 *             {({ name }) => <h3>{name}</h3>}
 *           </Service.Name>
 *           <Service.Price>
 *             {({ price }) => price && <div>{price.value} {price.currency}</div>}
 *           </Service.Price>
 *         </ServiceList.ServiceRepeater>
 *       </ServiceList.Options>
 *     </ServiceList.Root>
 *   );
 * }
 * ```
 */
export function Root(props) {
    return (_jsx(WixServices, { servicesMap: createServicesMap().addService(ServicesListServiceDefinition, ServicesListService, props.servicesListConfig), children: props.children }));
}
/**
 * Component that renders content when the services list is empty.
 * Only displays its children when there are no services, no loading state, and no errors.
 *
 * @component
 * @example
 * ```tsx
 * import { ServiceList } from '@wix/headless-services/react';
 *
 * function EmptyServicesMessage() {
 *   return (
 *     <ServiceList.EmptyState>
 *       {() => (
 *         <div className="empty-state">
 *           <h3>No services found</h3>
 *           <p>Try adjusting your search or filter criteria</p>
 *           <button>Clear Filters</button>
 *         </div>
 *       )}
 *     </ServiceList.EmptyState>
 *   );
 * }
 * ```
 */
export function EmptyState(props) {
    const { isLoading, error, services: servicesList, } = useService(ServicesListServiceDefinition);
    const isLoadingValue = isLoading.get();
    const errorValue = error.get();
    const servicesValue = servicesList.get();
    if (!isLoadingValue && !errorValue && servicesValue.length === 0) {
        return typeof props.children === 'function'
            ? props.children({})
            : props.children;
    }
    return null;
}
/**
 * Component that renders content during loading state.
 * Only displays its children when the services list is currently loading.
 *
 * @component
 * @example
 * ```tsx
 * import { ServiceList } from '@wix/headless-services/react';
 *
 * function ServicesLoading() {
 *   return (
 *     <ServiceList.Loading>
 *       {() => (
 *         <div className="loading-spinner">
 *           <div>Loading services...</div>
 *           <div className="spinner"></div>
 *         </div>
 *       )}
 *     </ServiceList.Loading>
 *   );
 * }
 * ```
 */
export function Loading(props) {
    const { isLoading } = useService(ServicesListServiceDefinition);
    const isLoadingValue = isLoading.get();
    if (isLoadingValue) {
        return typeof props.children === 'function'
            ? props.children({})
            : props.children;
    }
    return null;
}
/**
 * Component that renders content when there's an error loading services.
 * Only displays its children when an error has occurred.
 *
 * @component
 * @example
 * ```tsx
 * import { ServiceList } from '@wix/headless-services/react';
 *
 * function ServicesError() {
 *   return (
 *     <ServiceList.Error>
 *       {({ error }) => (
 *         <div className="error-state">
 *           <h3>Error loading services</h3>
 *           <p>{error}</p>
 *           <button onClick={() => window.location.reload()}>
 *             Try Again
 *           </button>
 *         </div>
 *       )}
 *     </ServiceList.Error>
 *   );
 * }
 * ```
 */
export function Error(props) {
    const { error } = useService(ServicesListServiceDefinition);
    const errorValue = error.get();
    if (errorValue) {
        return typeof props.children === 'function'
            ? props.children({ error: errorValue })
            : props.children;
    }
    return null;
}
/**
 * Component that renders content for each service in the list.
 * Maps over all services and provides each service through a service context.
 * Only renders when services are successfully loaded (not loading, no error, and has services).
 *
 * @component
 * @example
 * ```tsx
 * import { ServiceList, Service } from '@wix/headless-services/react';
 *
 * function ServicesGrid() {
 *   return (
 *     <ServiceList.ItemContent>
 *       {({ service }) => (
 *         <div className="service-card">
 *           <Service.Image>
 *             {({ image, alt }) => <img src={image} alt={alt} />}
 *           </Service.Image>
 *           <Service.Name>
 *             {({ name }) => <h3>{name}</h3>}
 *           </Service.Name>
 *           <Service.Price>
 *             {({ price }) => price && <p>{price.value} {price.currency}</p>}
 *           </Service.Price>
 *           <button>Book Now</button>
 *         </div>
 *       )}
 *     </ServiceList.ItemContent>
 *   );
 * }
 * ```
 */
export function ItemContent(props) {
    const { services: servicesList, isLoading, error, } = useService(ServicesListServiceDefinition);
    const servicesValue = servicesList.get();
    if (isLoading.get() || error.get() || servicesValue.length === 0) {
        return null;
    }
    return servicesValue.map((service) => (_jsx(WixServices, { servicesMap: createServicesMap(), children: typeof props.children === 'function'
            ? props.children({ service })
            : props.children }, service._id)));
}
export function Items(props) {
    const { services: servicesList } = useService(ServicesListServiceDefinition);
    const servicesValue = servicesList.get();
    return typeof props.children === 'function'
        ? props.children({ services: servicesValue })
        : props.children;
}
