// @wix/auto-patterns/dist/esm/components/AutoPatternsEntityPage/Fields/DateTime.js
import React, { useRef, useMemo, useState, useEffect } from 'react';
import { Cell, DatePicker, FormField, Layout, TimeInput } from '@wix/design-system';
import { useWixPatternsContainer, useSelector, useEntityPageContext } from '@wix/patterns';
import { useController } from '@wix/patterns/form';
export const DateTime = props => {
  var _pageState$entity, _field$validation2;
  const {
    field,
    dataHook,
    inputRef
  } = props;
  const {
    translate: t,
    ...container
  } = useWixPatternsContainer();
  useSelector(() => container.initTask.status);
  const pageState = useEntityPageContext();
  const controller = useController({
    name: (field == null ? void 0 : field.id) ?? '',
    control: pageState.form.control,
    defaultValue: (_pageState$entity = pageState.entity) == null ? void 0 : _pageState$entity[(field == null ? void 0 : field.id) ?? ''],
    rules: {
      validate: v => {
        var _field$validation;
        if (field != null && (_field$validation = field.validation) != null && _field$validation.required && (v === null || v === undefined || v === '')) {
          return t('cairo.fieldValidation.requiredField');
        }
        return true;
      }
    }
  });
  const date = useMemo(() => {
    if (!controller.field.value) {
      return undefined;
    }
    const d = new Date(controller.field.value);
    return isNaN(d.getTime()) ? undefined : d;
  }, [controller.field.value]);
  const [dateStatusMessage, setDateStatusMessage] = useState('');
  const [timeStatusMessage, setTimeStatusMessage] = useState('');
  const ref = useRef({});
  const dateRef = useRef({});
  const timeRef = useRef({});
  const [dateValue, setDateValue] = useState(date == null ? void 0 : date.getDate());
  const [timeValue, setTimeValue] = useState(date == null ? void 0 : date.getTime());
  useEffect(() => {
    ref.current.focus = () => {
      if (dateRef.current.invalid) {
        return dateRef.current.focus == null ? void 0 : dateRef.current.focus();
      }
      return timeRef.current.focus == null ? void 0 : timeRef.current.focus();
    };
    inputRef == null || inputRef(ref.current);
  }, [inputRef]);
  return /*#__PURE__*/React.createElement(FormField, {
    label: field.displayName,
    required: (_field$validation2 = field.validation) == null ? void 0 : _field$validation2.required,
    dataHook: dataHook
  }, /*#__PURE__*/React.createElement(Layout, null, /*#__PURE__*/React.createElement(Cell, {
    span: 6
  }, /*#__PURE__*/React.createElement(DatePicker, {
    ref: internalRef => {
      dateRef.current.focus = () => {
        var _internalRef$state;
        return (// TODO: PR to WSR needed to expose as public API
          // @ts-expect-error
          internalRef == null || (_internalRef$state = internalRef.state) == null || (_internalRef$state = _internalRef$state.inputRef) == null || _internalRef$state.focus == null ? void 0 : _internalRef$state.focus()
        );
      };
    },
    width: "100%",
    value: date
    // onChange is triggered only when you have valid, non-empty value. and onValidate is triggered on every change so we use onValidate
    ,
    onChange: () => {},
    onValidate: _ref => {
      let {
        validationType,
        value: newValue,
        format
      } = _ref;
      if (validationType === 'valid') {
        const dateWithNewDate = date ? new Date(date) : new Date(newValue);
        const _value = new Date(newValue);
        dateWithNewDate.setFullYear(_value.getFullYear());
        dateWithNewDate.setMonth(_value.getMonth());
        dateWithNewDate.setDate(_value.getDate());
        controller.field.onChange(dateWithNewDate.toISOString());
        setDateValue(new Date(newValue).getDate());
        setDateStatusMessage('');
        dateRef.current.invalid = false;
      } else if (validationType === 'formatError' && newValue === '') {
        var _field$validation3;
        if ((_field$validation3 = field.validation) != null && _field$validation3.required) {
          setDateStatusMessage(t('cairo.fieldValidation.requiredField'));
          dateRef.current.invalid = true;
        } else {
          setDateStatusMessage('');
          dateRef.current.invalid = false;
        }
        setDateValue(newValue);
        if (!timeValue) {
          controller.field.onChange(null);
        }
      } else {
        setDateStatusMessage(t('cairo.customFields.fieldType.date.invalid.value', {
          dateFormat: format ?? ''
        }));
        dateRef.current.invalid = true;
      }
      ref.current.invalid = dateRef.current.invalid || timeRef.current.invalid;
    },
    status: controller.fieldState.error || dateStatusMessage ? 'error' : undefined,
    statusMessage: controller.fieldState.error || dateStatusMessage,
    dataHook: `datetime-date-${field.id}`
  })), /*#__PURE__*/React.createElement(Cell, {
    span: 6
  }, /*#__PURE__*/React.createElement(TimeInput, {
    ref: internalRef => {
      timeRef.current.focus = () => internalRef == null ? void 0 : internalRef.focus();
    },
    dataHook: `datetime-time-${field.id}`,
    value: date ?? null,
    status: controller.fieldState.error || timeStatusMessage ? 'error' : undefined,
    statusMessage: controller.fieldState.error || timeStatusMessage,
    onInvalid: () => {
      timeRef.current.invalid = true;
      setTimeStatusMessage(t('cairo.customFields.fieldType.dateAndTime.invalid.value'));
      ref.current.invalid = dateRef.current.invalid || timeRef.current.invalid;
    },
    onChange: _ref2 => {
      let {
        date: newValue
      } = _ref2;
      timeRef.current.invalid = false;
      ref.current.invalid = dateRef.current.invalid || timeRef.current.invalid;
      if (newValue) {
        const dateWithNewTime = date ? new Date(date) : newValue;
        dateWithNewTime.setHours(newValue.getHours());
        dateWithNewTime.setMinutes(newValue.getMinutes());
        dateWithNewTime.setSeconds(newValue.getSeconds());
        dateWithNewTime.setMilliseconds(newValue.getMilliseconds());
        controller.field.onChange(dateWithNewTime.toISOString());
        setTimeStatusMessage('');
        setDateStatusMessage('');
      } else {
        if (!newValue) {
          var _field$validation4;
          setTimeValue('');
          if ((_field$validation4 = field.validation) != null && _field$validation4.required) {
            setTimeStatusMessage(t('cairo.fieldValidation.requiredField'));
            timeRef.current.invalid = true;
          } else {
            setTimeStatusMessage('');
            timeRef.current.invalid = false;
          }
          if (!ref.current.invalid) {
            if (!dateValue) {
              controller.field.onChange(null);
            }
          }
        }
      }
    }
  }))));
};
