// @wix/auto-patterns/dist/esm/components/AutoPatternsEntityPage/Fields/FormFieldInput.js
import React from 'react';
import { ShortText } from './ShortText';
import { useEntityPageContext } from '@wix/patterns';
import { useController } from '@wix/patterns/form';
import { Checkbox } from './Checkbox';
import { DateInput } from './DateInput';
import { DateTime } from './DateTime';
import { Number } from './Number';
import { LongText } from './LongText';
import { Url } from './Url';
import { ImageInput } from './ImageInput';
export const EDITABLE_INPUT_ELEMENTS = {
  SHORT_TEXT: ShortText,
  LONG_TEXT: LongText,
  BOOLEAN: Checkbox,
  DATE: DateInput,
  DATETIME: DateTime,
  NUMBER: Number,
  URL: Url,
  IMAGE: ImageInput
};
export const FormFieldInput = props => {
  var _pageState$entity;
  const {
    field,
    inputRef
  } = props;
  const pageState = useEntityPageContext();
  const controller = useController({
    name: (field == null ? void 0 : field.id) ?? '',
    control: pageState.form.control,
    defaultValue: (_pageState$entity = pageState.entity) == null ? void 0 : _pageState$entity[(field == null ? void 0 : field.id) ?? '']
  });
  if (!EDITABLE_INPUT_ELEMENTS.hasOwnProperty(field.type) || !pageState) {
    return null;
  }
  const FieldInput = EDITABLE_INPUT_ELEMENTS[field.type];
  return /*#__PURE__*/React.createElement(FieldInput, {
    field: field,
    value: controller.field.value,
    onChange: controller.field.onChange,
    dataHook: `form-field-${field.id}`,
    inputRef: inputRef
  });
};
