// @wix/auto-patterns/dist/esm/components/AutoPatternsEntityPage/Fields/ImageInput.js
import React from 'react';
import { FormField, ImageViewer } from '@wix/design-system';
import { useEntityPageContext, useSelector, useWixPatternsContainer } from '@wix/patterns';
import { useController } from '@wix/patterns/form';
import { getImageUrl } from '../../../utils/media/getImageUrl';
import { dashboard } from '@wix/dashboard';
export const ImageInput = props => {
  var _pageState$entity, _field$validation2, _controller$fieldStat;
  const {
    field,
    dataHook
  } = props;
  const {
    translate: t,
    ...container
  } = useWixPatternsContainer();
  useSelector(() => container.initTask.status);
  const pageState = useEntityPageContext();
  async function openMedia() {
    try {
      var _chosenMediaItems$ite;
      const chosenMediaItems = await dashboard.openMediaManager({
        multiSelect: false,
        category: 'IMAGE'
      });
      const newImage = chosenMediaItems == null || (_chosenMediaItems$ite = chosenMediaItems.items[0]) == null || (_chosenMediaItems$ite = _chosenMediaItems$ite.media) == null || (_chosenMediaItems$ite = _chosenMediaItems$ite.image) == null ? void 0 : _chosenMediaItems$ite.image;
      controller.field.onChange(newImage);
    } catch (error) {
      console.error('Error opening Media Manager or no files selected:', error);
    }
  }
  const controller = useController({
    name: (field == null ? void 0 : field.id) ?? '',
    control: pageState.form.control,
    defaultValue: (_pageState$entity = pageState.entity) == null ? void 0 : _pageState$entity[(field == null ? void 0 : field.id) ?? ''],
    rules: {
      validate: value => {
        var _field$validation;
        if ((_field$validation = field.validation) != null && _field$validation.required && (value === null || value === undefined || value === '')) {
          return t('cairo.fieldValidation.requiredField');
        }
        return true;
      }
    }
  });
  const imageUrl = controller.field.value && getImageUrl(controller.field.value);
  return /*#__PURE__*/React.createElement(FormField, {
    label: field.displayName,
    required: (_field$validation2 = field.validation) == null ? void 0 : _field$validation2.required,
    status: controller.fieldState.error ? 'error' : undefined,
    statusMessage: (_controller$fieldStat = controller.fieldState.error) == null ? void 0 : _controller$fieldStat.message,
    dataHook: dataHook
  }, /*#__PURE__*/React.createElement(ImageViewer, {
    imageUrl: imageUrl,
    dataHook: `image-${field.id}`,
    onAddImage: () => openMedia(),
    onUpdateImage: () => openMedia(),
    updateImageInfo: t('cairo.replace-image.tooltip'),
    removeImageInfo: t('cairo.remove-image.tooltip'),
    addImageInfo: t('cairo.add-image.tooltip'),
    height: "120px",
    width: "25%",
    fit: "cover",
    onRemoveImage: () => controller.field.onChange(null)
  }));
};
