// @wix/auto-patterns/dist/esm/components/AutoPatternsEntityPage/Fields/LongText.js
import React from 'react';
import { FormField, InputArea } from '@wix/design-system';
import { useEntityPageContext, useSelector, useWixPatternsContainer } from '@wix/patterns';
import { useController } from '@wix/patterns/form';
export const LongText = props => {
  var _pageState$entity, _field$validation2, _controller$fieldStat;
  const {
    field,
    dataHook
  } = props;
  const {
    translate: t,
    ...container
  } = useWixPatternsContainer();
  useSelector(() => container.initTask.status);
  const pageState = useEntityPageContext();
  const controller = useController({
    name: (field == null ? void 0 : field.id) ?? '',
    control: pageState.form.control,
    defaultValue: (_pageState$entity = pageState.entity) == null ? void 0 : _pageState$entity[(field == null ? void 0 : field.id) ?? ''],
    rules: {
      validate: value => {
        var _field$validation;
        if ((_field$validation = field.validation) != null && _field$validation.required && (value === null || value === undefined || value === '')) {
          return t('cairo.fieldValidation.requiredField');
        }
        return true;
      }
    }
  });
  return /*#__PURE__*/React.createElement(FormField, {
    label: field.displayName,
    required: (_field$validation2 = field.validation) == null ? void 0 : _field$validation2.required,
    status: controller.fieldState.error ? 'error' : undefined,
    statusMessage: (_controller$fieldStat = controller.fieldState.error) == null ? void 0 : _controller$fieldStat.message,
    dataHook: dataHook
  }, /*#__PURE__*/React.createElement(InputArea, {
    value: controller.field.value,
    onChange: e => controller.field.onChange == null ? void 0 : controller.field.onChange(e.target.value),
    onBlur: controller.field.onBlur,
    ref: controller.field.ref,
    dataHook: `long-text-${field.id}`
  }));
};
