// @wix/auto-patterns/dist/esm/components/AutoPatternsEntityPage/Fields/Number.js
import React from 'react';
import { FormField, NumberInput } from '@wix/design-system';
import { useEntityPageContext, useSelector, useWixPatternsContainer } from '@wix/patterns';
import { useController } from '@wix/patterns/form';
export const Number = props => {
  var _field$validation, _pageState$entity, _field$validation3, _controller$fieldStat;
  const {
    field,
    dataHook
  } = props;
  const {
    translate: t,
    ...container
  } = useWixPatternsContainer();
  useSelector(() => container.initTask.status);
  const pageState = useEntityPageContext();
  const numberRange = (_field$validation = field.validation) == null ? void 0 : _field$validation.numberRange;
  const min = numberRange == null ? void 0 : numberRange.min;
  const max = numberRange == null ? void 0 : numberRange.max;
  const controller = useController({
    name: (field == null ? void 0 : field.id) ?? '',
    control: pageState.form.control,
    defaultValue: (_pageState$entity = pageState.entity) == null ? void 0 : _pageState$entity[(field == null ? void 0 : field.id) ?? ''],
    rules: {
      validate: value => {
        var _field$validation2;
        if ((_field$validation2 = field.validation) != null && _field$validation2.required && (value === null || value === undefined || value === '')) {
          return t('cairo.fieldValidation.requiredField');
        }
        if (value !== null && value !== undefined) {
          if (min !== undefined && value < min) {
            return t('cairo.fieldValidation.minNumber', {
              minAmount: min
            });
          }
          if (max !== undefined && value > max) {
            return t('cairo.fieldValidation.maxNumber', {
              maxAmount: max
            });
          }
        }
        return true;
      }
    }
  });
  return /*#__PURE__*/React.createElement(FormField, {
    label: field.displayName,
    required: (_field$validation3 = field.validation) == null ? void 0 : _field$validation3.required,
    status: controller.fieldState.error ? 'error' : undefined,
    statusMessage: (_controller$fieldStat = controller.fieldState.error) == null ? void 0 : _controller$fieldStat.message,
    dataHook: dataHook
  }, /*#__PURE__*/React.createElement(NumberInput, {
    value: controller.field.value ?? undefined,
    onChange: e => controller.field.onChange(e),
    onBlur: controller.field.onBlur,
    inputRef: controller.field.ref,
    dataHook: `number-${field.id}`
  }));
};
