// @wix/auto-patterns/dist/esm/components/AutoPatternsEntityPage/Fields/ShortText.js
import React from 'react';
import { FormField, Input } from '@wix/design-system';
import { useEntityPageContext, useSelector, useWixPatternsContainer } from '@wix/patterns';
import { useController } from '@wix/patterns/form';
export const ShortText = props => {
  var _field$validation, _pageState$entity, _controller$field$val, _field$validation3, _controller$fieldStat;
  const {
    field,
    dataHook
  } = props;
  const {
    translate: t,
    ...container
  } = useWixPatternsContainer();
  useSelector(() => container.initTask.status);
  const pageState = useEntityPageContext();
  const stringLengthRange = (_field$validation = field.validation) == null ? void 0 : _field$validation.stringLengthRange;
  const minLength = stringLengthRange == null ? void 0 : stringLengthRange.minLength;
  const maxLength = stringLengthRange == null ? void 0 : stringLengthRange.maxLength;
  const controller = useController({
    name: (field == null ? void 0 : field.id) ?? '',
    control: pageState.form.control,
    defaultValue: (_pageState$entity = pageState.entity) == null ? void 0 : _pageState$entity[(field == null ? void 0 : field.id) ?? ''],
    rules: {
      validate: value => {
        var _field$validation2;
        if ((_field$validation2 = field.validation) != null && _field$validation2.required && (value === null || value === undefined || value === '')) {
          return t('cairo.fieldValidation.requiredField');
        }
        if (minLength && value && value.length < minLength) {
          return t('cairo.fieldValidation.minCharacter', {
            charMin: minLength
          });
        } else if (maxLength && value && value.length > maxLength) {
          return t('cairo.fieldValidation.maxCharacter', {
            charMax: maxLength
          });
        }
        return true;
      }
    }
  });
  const charCount = maxLength ? maxLength - ((_controller$field$val = controller.field.value) == null ? void 0 : _controller$field$val.length) : undefined;
  return /*#__PURE__*/React.createElement(FormField, {
    label: field.displayName,
    required: (_field$validation3 = field.validation) == null ? void 0 : _field$validation3.required,
    charCount: charCount,
    status: controller.fieldState.error ? 'error' : undefined,
    statusMessage: (_controller$fieldStat = controller.fieldState.error) == null ? void 0 : _controller$fieldStat.message,
    dataHook: dataHook
  }, /*#__PURE__*/React.createElement(Input, {
    value: controller.field.value,
    onChange: e => controller.field.onChange == null ? void 0 : controller.field.onChange(e.target.value),
    onBlur: controller.field.onBlur,
    dataHook: `short-text-${field.id}`,
    ref: controller.field.ref
  }));
};
