// @wix/auto-patterns/dist/esm/components/AutoPatternsEntityPage/Fields/Url.js
import React from 'react';
import { FormField, Input } from '@wix/design-system';
import { Link } from '@wix/wix-ui-icons-common';
import { useEntityPageContext, useSelector, useWixPatternsContainer } from '@wix/patterns';
import { useController } from '@wix/patterns/form';
export const Url = props => {
  var _pageState$entity, _field$validation2, _controller$fieldStat;
  const {
    field,
    dataHook
  } = props;
  const {
    translate: t,
    ...container
  } = useWixPatternsContainer();
  useSelector(() => container.initTask.status);
  const pageState = useEntityPageContext();

  // Same regex as data-extensions https://github.com/wix-private/server-infra/blob/a6c3009d39bbb0312c929f68ee81c559e1eb5b4e/iptf/dataextensions/src/main/scala/com/wixpress/infra/dataextensions/runtime/validator/Formats.scala#L31
  const uriPattern =
  // eslint-disable-next-line no-useless-escape
  /^[A-Za-z][A-Za-z0-9+.-]*:(\/\/([A-Za-z0-9._~\-%!$&'()*+,;=:]*@)?[A-Za-z0-9._~\-!$&'()*+,;=%:\[\]]*(:[0-9]*)?)?[A-Za-z0-9._~\-%!$&'()*+,;=:@\/]*([?][A-Za-z0-9._~\-%!$&'()*+,;=:@\/?]*)?([#][A-Za-z0-9._~\-%!$&'()*+,;=:@\/?]*)?$/;
  const controller = useController({
    name: (field == null ? void 0 : field.id) ?? '',
    control: pageState.form.control,
    defaultValue: (_pageState$entity = pageState.entity) == null ? void 0 : _pageState$entity[(field == null ? void 0 : field.id) ?? ''],
    rules: {
      validate: value => {
        var _field$validation;
        if ((_field$validation = field.validation) != null && _field$validation.required && (value === null || value === undefined || value === '')) {
          return t('cairo.fieldValidation.requiredField');
        } else if (value) {
          return uriPattern.test(value) || t('cairo.customFields.url-input.error');
        }
        return true;
      }
    }
  });
  return /*#__PURE__*/React.createElement(FormField, {
    label: field.displayName,
    required: (_field$validation2 = field.validation) == null ? void 0 : _field$validation2.required,
    dataHook: dataHook,
    status: controller.fieldState.error ? 'error' : undefined,
    statusMessage: (_controller$fieldStat = controller.fieldState.error) == null ? void 0 : _controller$fieldStat.message
  }, /*#__PURE__*/React.createElement(Input, {
    prefix: /*#__PURE__*/React.createElement(Input.IconAffix, null, /*#__PURE__*/React.createElement(Link, null)),
    placeholder: t('cairo.customFields.url-input.placeholder'),
    dataHook: `url-${field.id}`,
    value: controller.field.value,
    onChange: e => controller.field.onChange == null ? void 0 : controller.field.onChange(e.target.value),
    onBlur: controller.field.onBlur,
    ref: controller.field.ref
  }));
};
