// @wix/auto_sdk_online-programs_quizzes/build/internal/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  FileFormat: () => FileFormat,
  WebhookIdentityType: () => WebhookIdentityType,
  bulkCloneQuiz: () => bulkCloneQuiz2,
  bulkCreateQuiz: () => bulkCreateQuiz2,
  cloneQuiz: () => cloneQuiz2,
  createQuiz: () => createQuiz2,
  deleteQuiz: () => deleteQuiz2,
  getQuiz: () => getQuiz2
});
module.exports = __toCommonJS(index_typings_exports);

// src/achievements-quizzes-v1-quiz-quizzes.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/achievements-quizzes-v1-quiz-quizzes.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixAchievementsQuizzesV1QuizServiceUrl(opts) {
  const domainToMappings = {};
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_online-programs_quizzes";
function createQuiz(payload) {
  function __createQuiz({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "quiz.createdDate" }, { path: "quiz.updatedDate" }]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "quiz.fields.score" },
          { path: "quiz.fields.numeric.rightAnswer" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.achievements.quizzes.v1.quiz",
      method: "POST",
      methodFqn: "wix.achievements.quizzes.v1.QuizService.CreateQuiz",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAchievementsQuizzesV1QuizServiceUrl({
        protoPath: "/v1/quizzes",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "quiz.createdDate" }, { path: "quiz.updatedDate" }]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "quiz.fields.score" },
            { path: "quiz.fields.numeric.rightAnswer" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createQuiz;
}
function cloneQuiz(payload) {
  function __cloneQuiz({ host }) {
    const metadata = {
      entityFqdn: "wix.achievements.quizzes.v1.quiz",
      method: "POST",
      methodFqn: "wix.achievements.quizzes.v1.QuizService.CloneQuiz",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAchievementsQuizzesV1QuizServiceUrl({
        protoPath: "/v1/quizzes/{quizId}/clone",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "quiz.createdDate" }, { path: "quiz.updatedDate" }]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "quiz.fields.score" },
            { path: "quiz.fields.numeric.rightAnswer" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __cloneQuiz;
}
function bulkCreateQuiz(payload) {
  function __bulkCreateQuiz({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "quizzes.createdDate" },
          { path: "quizzes.updatedDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "quizzes.fields.score" },
          { path: "quizzes.fields.numeric.rightAnswer" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.achievements.quizzes.v1.quiz",
      method: "POST",
      methodFqn: "wix.achievements.quizzes.v1.QuizService.BulkCreateQuiz",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAchievementsQuizzesV1QuizServiceUrl({
        protoPath: "/v1/bulk/quizzes/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.fields.score" },
            { path: "results.item.fields.numeric.rightAnswer" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateQuiz;
}
function bulkCloneQuiz(payload) {
  function __bulkCloneQuiz({ host }) {
    const metadata = {
      entityFqdn: "wix.achievements.quizzes.v1.quiz",
      method: "POST",
      methodFqn: "wix.achievements.quizzes.v1.QuizService.BulkCloneQuiz",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAchievementsQuizzesV1QuizServiceUrl({
        protoPath: "/v1/bulk/quizzes/clone",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.fields.score" },
            { path: "results.item.fields.numeric.rightAnswer" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCloneQuiz;
}
function getQuiz(payload) {
  function __getQuiz({ host }) {
    const metadata = {
      entityFqdn: "wix.achievements.quizzes.v1.quiz",
      method: "GET",
      methodFqn: "wix.achievements.quizzes.v1.QuizService.GetQuiz",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAchievementsQuizzesV1QuizServiceUrl({
        protoPath: "/v1/quizzes/{quizId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "quiz.createdDate" }, { path: "quiz.updatedDate" }]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "quiz.fields.score" },
            { path: "quiz.fields.numeric.rightAnswer" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getQuiz;
}
function deleteQuiz(payload) {
  function __deleteQuiz({ host }) {
    const metadata = {
      entityFqdn: "wix.achievements.quizzes.v1.quiz",
      method: "DELETE",
      methodFqn: "wix.achievements.quizzes.v1.QuizService.DeleteQuiz",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAchievementsQuizzesV1QuizServiceUrl({
        protoPath: "/v1/quizzes/{quizId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "quiz.createdDate" }, { path: "quiz.updatedDate" }]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "quiz.fields.score" },
            { path: "quiz.fields.numeric.rightAnswer" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __deleteQuiz;
}

// src/achievements-quizzes-v1-quiz-quizzes.universal.ts
var FileFormat = /* @__PURE__ */ ((FileFormat2) => {
  FileFormat2["IMAGE"] = "IMAGE";
  FileFormat2["VIDEO"] = "VIDEO";
  FileFormat2["AUDIO"] = "AUDIO";
  FileFormat2["DOCUMENT"] = "DOCUMENT";
  FileFormat2["ARCHIVE"] = "ARCHIVE";
  return FileFormat2;
})(FileFormat || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createQuiz2(quiz) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ quiz });
  const reqOpts = createQuiz(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.quiz;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { quiz: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["quiz"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function cloneQuiz2(quizId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ quizId });
  const reqOpts = cloneQuiz(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { quizId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["quizId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCreateQuiz2(quizzes, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    quizzes,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkCreateQuiz(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          quizzes: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["quizzes", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCloneQuiz2(quizIds, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    quizIds,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkCloneQuiz(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          quizIds: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["quizIds", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getQuiz2(quizId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ quizId });
  const reqOpts = getQuiz(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.quiz;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { quizId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["quizId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteQuiz2(quizId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ quizId });
  const reqOpts = deleteQuiz(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { quizId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["quizId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  FileFormat,
  WebhookIdentityType,
  bulkCloneQuiz,
  bulkCreateQuiz,
  cloneQuiz,
  createQuiz,
  deleteQuiz,
  getQuiz
});
