// @wix/auto_sdk_events_events-settings/build/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  Action: () => Action,
  AppStateState: () => AppStateState,
  CheckoutType: () => CheckoutType,
  DayOfWeek: () => DayOfWeek,
  DeleteStatus: () => DeleteStatus,
  FeaturePeriod: () => FeaturePeriod,
  Namespace: () => Namespace,
  PatchField: () => PatchField,
  PlacementType: () => PlacementType,
  PremiumFeature: () => PremiumFeature,
  RequestedFields: () => RequestedFields,
  ResolutionMethod: () => ResolutionMethod,
  SiteCreatedContext: () => SiteCreatedContext,
  State: () => State,
  Type: () => Type,
  WebhookIdentityType: () => WebhookIdentityType,
  getEventsSettings: () => getEventsSettings4,
  onEventsSettingsUpdated: () => onEventsSettingsUpdated2,
  updateEventsSettings: () => updateEventsSettings4
});
module.exports = __toCommonJS(index_exports);

// src/events-v1-events-settings-events-settings.public.ts
var import_rename_all_nested_keys2 = require("@wix/sdk-runtime/rename-all-nested-keys");
var import_page_url_v23 = require("@wix/sdk-runtime/transformations/page-url-v2");
var import_timestamp3 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths3 = require("@wix/sdk-runtime/transformations/transform-paths");
var import_sdk_types = require("@wix/sdk-types");

// src/events-v1-events-settings-events-settings.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/events-v1-events-settings-events-settings.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEventsSettingsV1EventsSettingsServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/events-settings",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/events-settings",
        destPath: ""
      }
    ],
    "events.wixapps.net": [
      {
        srcPath: "/_api/events-settings",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/events-settings",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/events-settings",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/events/v1/settings",
        destPath: "/v1/events-settings"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/events-settings",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/events-settings",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_events_events-settings";
function getEventsSettings(payload) {
  function __getEventsSettings({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.events_settings",
      method: "GET",
      methodFqn: "wix.events.settings.v1.EventsSettingsService.GetEventsSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsSettingsV1EventsSettingsServiceUrl({
        protoPath: "/v1/events-settings",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "eventsSettings.createdDate" },
            { path: "eventsSettings.updatedDate" },
            { path: "eventsSettings.appState.stateUpdatedDate" },
            {
              path: "eventsSettings.serviceFeeSettings.serviceFeeActivationDate"
            },
            {
              path: "eventsSettings.serviceFeeSettings.serviceFeeActivationAgreementDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getEventsSettings;
}
function updateEventsSettings(payload) {
  function __updateEventsSettings({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "eventsSettings.createdDate" },
          { path: "eventsSettings.updatedDate" },
          { path: "eventsSettings.appState.stateUpdatedDate" },
          {
            path: "eventsSettings.serviceFeeSettings.serviceFeeActivationDate"
          },
          {
            path: "eventsSettings.serviceFeeSettings.serviceFeeActivationAgreementDate"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v1.events_settings",
      method: "PATCH",
      methodFqn: "wix.events.settings.v1.EventsSettingsService.UpdateEventsSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsSettingsV1EventsSettingsServiceUrl({
        protoPath: "/v1/events-settings/{eventsSettings.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "eventsSettings.createdDate" },
            { path: "eventsSettings.updatedDate" },
            { path: "eventsSettings.appState.stateUpdatedDate" },
            {
              path: "eventsSettings.serviceFeeSettings.serviceFeeActivationDate"
            },
            {
              path: "eventsSettings.serviceFeeSettings.serviceFeeActivationAgreementDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateEventsSettings;
}

// src/events-v1-events-settings-events-settings.universal.ts
var import_page_url_v2 = require("@wix/sdk-runtime/transformations/page-url-v2");
var import_page_url_v22 = require("@wix/sdk-runtime/transformations/page-url-v2");
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var AppStateState = /* @__PURE__ */ ((AppStateState2) => {
  AppStateState2["UNKNOWN_STATE"] = "UNKNOWN_STATE";
  AppStateState2["ENABLED"] = "ENABLED";
  AppStateState2["DISABLED"] = "DISABLED";
  return AppStateState2;
})(AppStateState || {});
var PremiumFeature = /* @__PURE__ */ ((PremiumFeature2) => {
  PremiumFeature2["UNKNOWN_PREMIUM_FEATURE"] = "UNKNOWN_PREMIUM_FEATURE";
  PremiumFeature2["SELL_TICKETS"] = "SELL_TICKETS";
  PremiumFeature2["INCLUDE_URL_IN_EMAIL"] = "INCLUDE_URL_IN_EMAIL";
  PremiumFeature2["HIDE_FOOTER_IN_EMAIL"] = "HIDE_FOOTER_IN_EMAIL";
  return PremiumFeature2;
})(PremiumFeature || {});
var Action = /* @__PURE__ */ ((Action2) => {
  Action2["UNKNOWN_ACTION"] = "UNKNOWN_ACTION";
  Action2["CAPTURE"] = "CAPTURE";
  Action2["VOID"] = "VOID";
  return Action2;
})(Action || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  Type2["ENABLED"] = "ENABLED";
  Type2["DISABLED"] = "DISABLED";
  return Type2;
})(Type || {});
var CheckoutType = /* @__PURE__ */ ((CheckoutType2) => {
  CheckoutType2["UNKNOWN_CHECKOUT_TYPE"] = "UNKNOWN_CHECKOUT_TYPE";
  CheckoutType2["EVENTS_APP"] = "EVENTS_APP";
  CheckoutType2["ECOMM_PLATFORM"] = "ECOMM_PLATFORM";
  return CheckoutType2;
})(CheckoutType || {});
var RequestedFields = /* @__PURE__ */ ((RequestedFields2) => {
  RequestedFields2["UNKNOWN_REQUESTED_FIELD"] = "UNKNOWN_REQUESTED_FIELD";
  RequestedFields2["SITE_INFO"] = "SITE_INFO";
  RequestedFields2["EVENTS_INFO"] = "EVENTS_INFO";
  RequestedFields2["SERVICE_FEE_SETTINGS"] = "SERVICE_FEE_SETTINGS";
  RequestedFields2["PAYMENT_SETTINGS"] = "PAYMENT_SETTINGS";
  return RequestedFields2;
})(RequestedFields || {});
var State = /* @__PURE__ */ ((State2) => {
  State2["UNKNOWN"] = "UNKNOWN";
  State2["ENABLED"] = "ENABLED";
  State2["DISABLED"] = "DISABLED";
  State2["PENDING"] = "PENDING";
  State2["DEMO"] = "DEMO";
  return State2;
})(State || {});
var SiteCreatedContext = /* @__PURE__ */ ((SiteCreatedContext2) => {
  SiteCreatedContext2["OTHER"] = "OTHER";
  SiteCreatedContext2["FROM_TEMPLATE"] = "FROM_TEMPLATE";
  SiteCreatedContext2["DUPLICATE_BY_SITE_TRANSFER"] = "DUPLICATE_BY_SITE_TRANSFER";
  SiteCreatedContext2["DUPLICATE"] = "DUPLICATE";
  SiteCreatedContext2["OLD_SITE_TRANSFER"] = "OLD_SITE_TRANSFER";
  SiteCreatedContext2["FLASH"] = "FLASH";
  return SiteCreatedContext2;
})(SiteCreatedContext || {});
var Namespace = /* @__PURE__ */ ((Namespace2) => {
  Namespace2["UNKNOWN_NAMESPACE"] = "UNKNOWN_NAMESPACE";
  Namespace2["WIX"] = "WIX";
  Namespace2["SHOUT_OUT"] = "SHOUT_OUT";
  Namespace2["ALBUMS"] = "ALBUMS";
  Namespace2["WIX_STORES_TEST_DRIVE"] = "WIX_STORES_TEST_DRIVE";
  Namespace2["HOTELS"] = "HOTELS";
  Namespace2["CLUBS"] = "CLUBS";
  Namespace2["ONBOARDING_DRAFT"] = "ONBOARDING_DRAFT";
  Namespace2["DEV_SITE"] = "DEV_SITE";
  Namespace2["LOGOS"] = "LOGOS";
  Namespace2["VIDEO_MAKER"] = "VIDEO_MAKER";
  Namespace2["PARTNER_DASHBOARD"] = "PARTNER_DASHBOARD";
  Namespace2["DEV_CENTER_COMPANY"] = "DEV_CENTER_COMPANY";
  Namespace2["HTML_DRAFT"] = "HTML_DRAFT";
  Namespace2["SITELESS_BUSINESS"] = "SITELESS_BUSINESS";
  Namespace2["CREATOR_ECONOMY"] = "CREATOR_ECONOMY";
  Namespace2["DASHBOARD_FIRST"] = "DASHBOARD_FIRST";
  Namespace2["ANYWHERE"] = "ANYWHERE";
  Namespace2["HEADLESS"] = "HEADLESS";
  Namespace2["ACCOUNT_MASTER_CMS"] = "ACCOUNT_MASTER_CMS";
  Namespace2["RISE"] = "RISE";
  Namespace2["BRANDED_FIRST"] = "BRANDED_FIRST";
  Namespace2["NOWNIA"] = "NOWNIA";
  Namespace2["UGC_TEMPLATE"] = "UGC_TEMPLATE";
  Namespace2["CODUX"] = "CODUX";
  Namespace2["MEDIA_DESIGN_CREATOR"] = "MEDIA_DESIGN_CREATOR";
  Namespace2["SHARED_BLOG_ENTERPRISE"] = "SHARED_BLOG_ENTERPRISE";
  Namespace2["STANDALONE_FORMS"] = "STANDALONE_FORMS";
  Namespace2["STANDALONE_EVENTS"] = "STANDALONE_EVENTS";
  Namespace2["MIMIR"] = "MIMIR";
  Namespace2["TWINS"] = "TWINS";
  Namespace2["NANO"] = "NANO";
  return Namespace2;
})(Namespace || {});
var DeleteStatus = /* @__PURE__ */ ((DeleteStatus2) => {
  DeleteStatus2["UNKNOWN"] = "UNKNOWN";
  DeleteStatus2["TRASH"] = "TRASH";
  DeleteStatus2["DELETED"] = "DELETED";
  DeleteStatus2["PENDING_PURGE"] = "PENDING_PURGE";
  DeleteStatus2["PURGED_EXTERNALLY"] = "PURGED_EXTERNALLY";
  return DeleteStatus2;
})(DeleteStatus || {});
var PlacementType = /* @__PURE__ */ ((PlacementType2) => {
  PlacementType2["BEFORE"] = "BEFORE";
  PlacementType2["AFTER"] = "AFTER";
  PlacementType2["REPLACE"] = "REPLACE";
  return PlacementType2;
})(PlacementType || {});
var DayOfWeek = /* @__PURE__ */ ((DayOfWeek2) => {
  DayOfWeek2["MONDAY"] = "MONDAY";
  DayOfWeek2["TUESDAY"] = "TUESDAY";
  DayOfWeek2["WEDNESDAY"] = "WEDNESDAY";
  DayOfWeek2["THURSDAY"] = "THURSDAY";
  DayOfWeek2["FRIDAY"] = "FRIDAY";
  DayOfWeek2["SATURDAY"] = "SATURDAY";
  DayOfWeek2["SUNDAY"] = "SUNDAY";
  return DayOfWeek2;
})(DayOfWeek || {});
var ResolutionMethod = /* @__PURE__ */ ((ResolutionMethod2) => {
  ResolutionMethod2["QUERY_PARAM"] = "QUERY_PARAM";
  ResolutionMethod2["SUBDOMAIN"] = "SUBDOMAIN";
  ResolutionMethod2["SUBDIRECTORY"] = "SUBDIRECTORY";
  return ResolutionMethod2;
})(ResolutionMethod || {});
var FeaturePeriod = /* @__PURE__ */ ((FeaturePeriod2) => {
  FeaturePeriod2["NO_PERIOD"] = "NO_PERIOD";
  FeaturePeriod2["MILLISECOND"] = "MILLISECOND";
  FeaturePeriod2["SECOND"] = "SECOND";
  FeaturePeriod2["MINUTE"] = "MINUTE";
  FeaturePeriod2["HOUR"] = "HOUR";
  FeaturePeriod2["DAY"] = "DAY";
  FeaturePeriod2["WEEK"] = "WEEK";
  FeaturePeriod2["MONTH"] = "MONTH";
  FeaturePeriod2["YEAR"] = "YEAR";
  return FeaturePeriod2;
})(FeaturePeriod || {});
var PatchField = /* @__PURE__ */ ((PatchField2) => {
  PatchField2["UNKNOWN_PATCH_FIELD"] = "UNKNOWN_PATCH_FIELD";
  PatchField2["FEE_ENABLED"] = "FEE_ENABLED";
  PatchField2["FEE_ACTIVATION_DATE"] = "FEE_ACTIVATION_DATE";
  PatchField2["FEE_ACTIVATION_AGREEMENT_DATE"] = "FEE_ACTIVATION_AGREEMENT_DATE";
  return PatchField2;
})(PatchField || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function getEventsSettings2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    fields: options?.fields
  });
  const reqOpts = getEventsSettings(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_page_url_v22.transformRESTPageURLV2ToSDKPageURLV2,
          paths: [
            { path: "eventsSettings.eventPages.eventPageUrl" },
            { path: "eventsSettings.eventPages.schedulePageUrl" },
            { path: "eventsSettings.eventPages.standalonePageUrl" },
            { path: "eventsSettings.eventPages.basePageUrl" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { fields: "$[0].fields" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateEventsSettings2(_id, eventsSettings, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      eventsSettings: { ...eventsSettings, id: _id },
      fields: options?.fields
    }),
    [
      {
        transformFn: import_page_url_v2.transformSDKPageURLV2ToRESTPageURLV2,
        paths: [
          { path: "eventsSettings.eventPages.eventPageUrl" },
          { path: "eventsSettings.eventPages.schedulePageUrl" },
          { path: "eventsSettings.eventPages.standalonePageUrl" },
          { path: "eventsSettings.eventPages.basePageUrl" }
        ]
      }
    ]
  );
  const reqOpts = updateEventsSettings(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_page_url_v22.transformRESTPageURLV2ToSDKPageURLV2,
          paths: [
            { path: "eventsSettings.eventPages.eventPageUrl" },
            { path: "eventsSettings.eventPages.schedulePageUrl" },
            { path: "eventsSettings.eventPages.standalonePageUrl" },
            { path: "eventsSettings.eventPages.basePageUrl" }
          ]
        }
      ])
    )?.eventsSettings;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { eventsSettings: "$[1]" },
        explicitPathsToArguments: {
          "eventsSettings.id": "$[0]",
          fields: "$[2].fields"
        },
        singleArgumentUnchanged: false
      },
      ["_id", "eventsSettings", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/events-v1-events-settings-events-settings.public.ts
function getEventsSettings3(httpClient) {
  return (options) => getEventsSettings2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function updateEventsSettings3(httpClient) {
  return (_id, eventsSettings, options) => updateEventsSettings2(
    _id,
    eventsSettings,
    options,
    // @ts-ignore
    { httpClient }
  );
}
var onEventsSettingsUpdated = (0, import_sdk_types.EventDefinition)(
  "wix.events.v1.events_settings_updated",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.appState.stateUpdatedDate" },
          { path: "entity.serviceFeeSettings.serviceFeeActivationDate" },
          {
            path: "entity.serviceFeeSettings.serviceFeeActivationAgreementDate"
          },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_page_url_v23.transformRESTPageURLV2ToSDKPageURLV2,
        paths: [
          { path: "entity.eventPages.eventPageUrl" },
          { path: "entity.eventPages.schedulePageUrl" },
          { path: "entity.eventPages.standalonePageUrl" },
          { path: "entity.eventPages.basePageUrl" }
        ]
      }
    ])
  )
)();

// src/events-v1-events-settings-events-settings.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var import_event_definition_modules = require("@wix/sdk-runtime/event-definition-modules");
var getEventsSettings4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getEventsSettings3);
var updateEventsSettings4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(updateEventsSettings3);
var onEventsSettingsUpdated2 = (0, import_event_definition_modules.createEventModule)(
  onEventsSettingsUpdated
);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Action,
  AppStateState,
  CheckoutType,
  DayOfWeek,
  DeleteStatus,
  FeaturePeriod,
  Namespace,
  PatchField,
  PlacementType,
  PremiumFeature,
  RequestedFields,
  ResolutionMethod,
  SiteCreatedContext,
  State,
  Type,
  WebhookIdentityType,
  getEventsSettings,
  onEventsSettingsUpdated,
  updateEventsSettings
});
