// @wix/fast-gallery-core/dist/esm/lib/utils/itemBuilder.js
import { serializeStyles } from '@emotion/serialize';
import { ImageFit, } from '../types';
import { cssStringToBuilderReturnType } from './commonUtils';
const getContentCss = (infoPosition, 
// enableTitle: boolean,
contentItemsAlignment) => {
    const isFlexDirRow = ['LEFT', 'RIGHT', 'ON_IMAGE'].includes(infoPosition);
    const flexDirectionMap = {
        ABOVE: 'column-reverse',
        BELOW: 'column',
        LEFT: 'row-reverse',
        RIGHT: 'row',
        ON_IMAGE: 'row',
    };
    return {
        '.content': {
            justifyContent: isFlexDirRow ? 'center' : contentItemsAlignment,
            alignItems: isFlexDirRow ? contentItemsAlignment : 'center',
            flexDirection: flexDirectionMap[infoPosition] || 'column',
        },
        '.info': {
        // display: enableTitle ? 'block' : 'none',
        },
        ...(infoPosition === 'ON_IMAGE'
            ? {
                '.content .info, .content .counter': {
                    position: 'absolute',
                    textAlign: 'center',
                    width: '100%',
                    color: '#000000',
                    top: '50%',
                    left: '50%',
                    transform: 'translate(-50%, -50%)',
                },
            }
            : {}),
    };
};
const getMediaCss = (imageFit) => {
    return {
        '.media': {
            height: imageFit === ImageFit.crop ? '100%' : 'auto',
        },
        '.image': {
            objectFit: imageFit === ImageFit.crop ? 'cover' : 'contain',
        },
    };
};
const getDetailsCss = (infoPosition) => {
    if (infoPosition === 'ON_IMAGE') {
        return {
            '.details': {
                position: 'absolute',
                height: '100%',
            },
        };
    }
    else {
        return {};
    }
};
export const buildItemStyles = (styles = {}) => {
    const { infoPosition = 'ABOVE', 
    // enableTitle = true,
    imageFit = ImageFit.crop, contentItemsAlignment = 'flex-start', } = {
        ...styles,
    };
    const contentRule = serializeStyles([
        getContentCss(infoPosition, contentItemsAlignment),
    ]).styles;
    const mediaRule = serializeStyles([getMediaCss(imageFit)]).styles;
    const detailsRule = serializeStyles([getDetailsCss(infoPosition)]).styles;
    return cssStringToBuilderReturnType(contentRule + mediaRule + detailsRule);
};
