// @wix/headless-pricing-plans/cjs/dist/services/plan-list-service.js
import { defineService, implementService } from '@wix/services-definitions';
import { SignalsServiceDefinition, } from '@wix/services-definitions/core-services/signals';
import { plansV3 } from '@wix/pricing-plans';
import { enhancePlanData } from './plan-service.js';
export const PlanListServiceDefinition = defineService('planListService');
export const PlanListService = implementService.withConfig()(PlanListServiceDefinition, ({ getService, config }) => {
    const signalsService = getService(SignalsServiceDefinition);
    const isLoadingSignal = signalsService.signal(false);
    const errorSignal = signalsService.signal(null);
    const configHasPlans = 'plans' in config;
    const planListSignal = signalsService.signal(configHasPlans ? config.plans : []);
    if (!configHasPlans) {
        loadPlanList(config.planIds);
    }
    async function loadPlanList(planIds) {
        isLoadingSignal.set(true);
        errorSignal.set(null);
        try {
            const plans = await fetchAndEnhancePlans(planIds);
            planListSignal.set(plans);
        }
        catch (error) {
            console.error('Error loading plan list:', error);
            errorSignal.set(error instanceof Error ? error : new Error(error));
        }
        finally {
            isLoadingSignal.set(false);
        }
    }
    return {
        planListSignal: planListSignal,
        isLoadingSignal: isLoadingSignal,
        errorSignal: errorSignal,
    };
});
async function fetchAndEnhancePlans(planIds) {
    try {
        let request = plansV3.queryPlans().eq('visibility', 'PUBLIC');
        if (planIds?.length) {
            request = request.in('_id', planIds);
        }
        const result = await request.find();
        return result.items.map((plan) => enhancePlanData(plan));
    }
    catch (error) {
        console.log('Error fetching and enhancing plans:', error);
        throw error;
    }
}
export async function loadPlanListServiceConfig(planIds) {
    const plans = await fetchAndEnhancePlans(planIds);
    return { plans };
}
