// @wix/headless-pricing-plans/cjs/dist/services/plan-paywall-service.js
import { defineService, implementService, } from '@wix/services-definitions';
import { SignalsServiceDefinition, } from '@wix/services-definitions/core-services/signals';
import { orders } from '@wix/pricing-plans';
import { auth } from '@wix/essentials';
export const PlanPaywallServiceDefinition = defineService('planPaywallService');
export const PlanPaywallService = implementService.withConfig()(PlanPaywallServiceDefinition, ({ getService, config }) => {
    const signalsService = getService(SignalsServiceDefinition);
    const isLoadingSignal = signalsService.signal(false);
    const errorSignal = signalsService.signal(null);
    const memberOrdersSignal = signalsService.signal(config.memberOrders ?? null);
    const hasAccessSignal = signalsService.computed(() => {
        const memberOrders = memberOrdersSignal.get();
        if (memberOrders) {
            // Additional filtering in case provided orders are not only active
            const activePlanOrders = memberOrders.filter((order) => order.status === 'ACTIVE' &&
                config.requiredPlanIds.includes(order.planId));
            return activePlanOrders.length > 0;
        }
        return !!memberOrdersSignal.get()?.length;
    });
    if (!config.memberOrders) {
        loadMemberOrders(config.requiredPlanIds);
    }
    async function loadMemberOrders(requiredPlanIds) {
        try {
            isLoadingSignal.set(true);
            errorSignal.set(null);
            const isLoggedIn = isMemberLoggedIn();
            if (!isLoggedIn) {
                memberOrdersSignal.set(null);
                return;
            }
            const memberOrders = await fetchMemberOrders(requiredPlanIds);
            memberOrdersSignal.set(memberOrders);
        }
        catch (error) {
            errorSignal.set(error instanceof Error
                ? error.message
                : 'Failed to check member access');
        }
        finally {
            isLoadingSignal.set(false);
        }
    }
    return {
        isLoadingSignal,
        errorSignal,
        hasAccessSignal,
    };
});
async function fetchMemberOrders(planIds) {
    try {
        const memberOrders = await orders.memberListOrders({
            planIds,
            orderStatuses: ['ACTIVE'],
        });
        return memberOrders?.orders ?? null;
    }
    catch (error) {
        console.error('Error fetching member orders:', error);
        throw error;
    }
}
function isMemberLoggedIn() {
    return auth.getContextualAuth().loggedIn();
}
export async function loadPlanPaywallServiceConfig(requiredPlanIds) {
    if (!isMemberLoggedIn()) {
        return { memberOrders: [], requiredPlanIds };
    }
    const memberOrders = await fetchMemberOrders(requiredPlanIds);
    return { memberOrders: memberOrders ?? [], requiredPlanIds };
}
