// @wix/auto_sdk_payments_wix-payments-account/build/internal/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  connectWixPaymentsAccount: () => connectWixPaymentsAccount4
});
module.exports = __toCommonJS(index_exports);

// src/payments-mcp-v1-wix-payments-account-wix-payments-account.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/payments-mcp-v1-wix-payments-account-wix-payments-account.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
function resolveWixPaymentsMcpV1WixPaymentsAccountServiceUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/payments/v1/wix-payments-account",
        destPath: "/v1/wix-payments-account"
      },
      {
        srcPath: "/payments/mcp/v1/wix-payments-account",
        destPath: "/v1/wix-payments-account"
      }
    ]
  };
  return (0, import_rest_modules.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_wix-payments-account";
function connectWixPaymentsAccount(payload) {
  function __connectWixPaymentsAccount({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.mcp.v1.wix_payments_account",
      method: "POST",
      methodFqn: "wix.payments.mcp.v1.WixPaymentsAccountService.ConnectWixPaymentsAccount",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsMcpV1WixPaymentsAccountServiceUrl({
        protoPath: "/v1/wix-payments-account/connect",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __connectWixPaymentsAccount;
}

// src/payments-mcp-v1-wix-payments-account-wix-payments-account.universal.ts
async function connectWixPaymentsAccount2(account) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ account });
  const reqOpts = connectWixPaymentsAccount(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { account: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["account"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/payments-mcp-v1-wix-payments-account-wix-payments-account.public.ts
function connectWixPaymentsAccount3(httpClient) {
  return (account) => connectWixPaymentsAccount2(
    account,
    // @ts-ignore
    { httpClient }
  );
}

// src/payments-mcp-v1-wix-payments-account-wix-payments-account.context.ts
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
var connectWixPaymentsAccount4 = /* @__PURE__ */ (0, import_rest_modules2.createRESTModule)(connectWixPaymentsAccount3);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  connectWixPaymentsAccount
});
