// pro-gallery/dist/cjs/components/item/3d/3dItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importStar(require("react"));
const hooks_js_1 = require("./hooks.js");
const GalleryUI_js_1 = require("../media/GalleryUI.js");
function ThreeDItem(props) {
    const { canvasRef, isLoaded } = (0, hooks_js_1.use3DItem)(props);
    const canvas = (react_1.default.createElement("canvas", { key: 'canvas', width: '100%', height: '100%', ref: canvasRef, style: {
            background: 'transparent',
            opacity: isLoaded ? 1 : 0,
            transition: 'opacity 200ms',
        } }));
    useOrbitControlsClickPropagation(props);
    return (react_1.default.createElement(react_1.default.Fragment, null,
        canvas,
        props.thumbnailWithOverride({
            extraClasses: [props.placeholderExtraClasses, isLoaded ? 'three-d-loaded' : 'three-d-loading'].join(' '),
            overlay: react_1.default.createElement(GalleryUI_js_1.GalleryUI, { type: 'rotateArrow', size: 60 }),
        }),
        props.hover));
}
exports.default = ThreeDItem;
function useOrbitControlsClickPropagation(props) {
    (0, react_1.useEffect)(() => {
        const container = props.itemContainer.current;
        if (!props.shouldPlay && container) {
            const onMouseDown = (e) => {
                e.stopImmediatePropagation();
                const itemWrapper = container.querySelector('.gallery-item-wrapper');
                if (itemWrapper) {
                    itemWrapper.click();
                }
            };
            container.addEventListener('mousedown', onMouseDown);
            return () => container.removeEventListener('mousedown', onMouseDown);
        }
        return;
    }, [props.shouldPlay]);
}
