// @wix/auto_sdk_members_members-about/build/es/meta.mjs
// src/members-about-v2-member-about-members-about.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKFloatToRESTFloat } from "@wix/sdk-runtime/transformations/float";
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixMembersAboutV2MembersAboutUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/members-about-api",
        destPath: ""
      },
      {
        srcPath: "/members/v2/abouts",
        destPath: "/v2/abouts"
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/members-about-api",
        destPath: ""
      },
      {
        srcPath: "/members/v2/abouts",
        destPath: "/v2/abouts"
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/members-about-api",
        destPath: ""
      },
      {
        srcPath: "/members/v2/abouts",
        destPath: "/v2/abouts"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/members/v2/abouts",
        destPath: "/v2/abouts"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_members-about";
function createMemberAbout(payload) {
  function __createMemberAbout({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          {
            path: "memberAbout.content.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "memberAbout.content.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "memberAbout.content.nodes.galleryData.items.video.thumbnail.duration"
          },
          { path: "memberAbout.content.nodes.galleryData.options.item.ratio" },
          { path: "memberAbout.content.nodes.imageData.image.duration" },
          { path: "memberAbout.content.nodes.mapData.mapSettings.lat" },
          { path: "memberAbout.content.nodes.mapData.mapSettings.lng" },
          { path: "memberAbout.content.nodes.pollData.poll.image.duration" },
          {
            path: "memberAbout.content.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "memberAbout.content.nodes.pollData.design.poll.background.image.duration"
          },
          { path: "memberAbout.content.nodes.appEmbedData.image.duration" },
          { path: "memberAbout.content.nodes.videoData.video.duration" },
          { path: "memberAbout.content.nodes.videoData.thumbnail.duration" },
          { path: "memberAbout.content.nodes.audioData.audio.duration" },
          { path: "memberAbout.content.nodes.audioData.coverImage.duration" }
        ]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "memberAbout.content.metadata.createdTimestamp" },
          { path: "memberAbout.content.metadata.updatedTimestamp" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.members.about.v2.member_about",
      method: "POST",
      methodFqn: "wix.members.about.v2.MembersAbout.CreateMemberAbout",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixMembersAboutV2MembersAboutUrl({
        protoPath: "/v2/abouts",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "memberAbout.content.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.options.item.ratio"
            },
            { path: "memberAbout.content.nodes.imageData.image.duration" },
            { path: "memberAbout.content.nodes.mapData.mapSettings.lat" },
            { path: "memberAbout.content.nodes.mapData.mapSettings.lng" },
            {
              path: "memberAbout.content.nodes.pollData.poll.image.duration"
            },
            {
              path: "memberAbout.content.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "memberAbout.content.nodes.pollData.design.poll.background.image.duration"
            },
            { path: "memberAbout.content.nodes.appEmbedData.image.duration" },
            { path: "memberAbout.content.nodes.videoData.video.duration" },
            {
              path: "memberAbout.content.nodes.videoData.thumbnail.duration"
            },
            { path: "memberAbout.content.nodes.audioData.audio.duration" },
            {
              path: "memberAbout.content.nodes.audioData.coverImage.duration"
            }
          ]
        },
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "memberAbout.content.metadata.createdTimestamp" },
            { path: "memberAbout.content.metadata.updatedTimestamp" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createMemberAbout;
}
function updateMemberAbout(payload) {
  function __updateMemberAbout({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          {
            path: "memberAbout.content.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "memberAbout.content.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "memberAbout.content.nodes.galleryData.items.video.thumbnail.duration"
          },
          { path: "memberAbout.content.nodes.galleryData.options.item.ratio" },
          { path: "memberAbout.content.nodes.imageData.image.duration" },
          { path: "memberAbout.content.nodes.mapData.mapSettings.lat" },
          { path: "memberAbout.content.nodes.mapData.mapSettings.lng" },
          { path: "memberAbout.content.nodes.pollData.poll.image.duration" },
          {
            path: "memberAbout.content.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "memberAbout.content.nodes.pollData.design.poll.background.image.duration"
          },
          { path: "memberAbout.content.nodes.appEmbedData.image.duration" },
          { path: "memberAbout.content.nodes.videoData.video.duration" },
          { path: "memberAbout.content.nodes.videoData.thumbnail.duration" },
          { path: "memberAbout.content.nodes.audioData.audio.duration" },
          { path: "memberAbout.content.nodes.audioData.coverImage.duration" }
        ]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "memberAbout.content.metadata.createdTimestamp" },
          { path: "memberAbout.content.metadata.updatedTimestamp" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.members.about.v2.member_about",
      method: "PATCH",
      methodFqn: "wix.members.about.v2.MembersAbout.UpdateMemberAbout",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixMembersAboutV2MembersAboutUrl({
        protoPath: "/v2/abouts/{memberAbout.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "memberAbout.content.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.options.item.ratio"
            },
            { path: "memberAbout.content.nodes.imageData.image.duration" },
            { path: "memberAbout.content.nodes.mapData.mapSettings.lat" },
            { path: "memberAbout.content.nodes.mapData.mapSettings.lng" },
            {
              path: "memberAbout.content.nodes.pollData.poll.image.duration"
            },
            {
              path: "memberAbout.content.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "memberAbout.content.nodes.pollData.design.poll.background.image.duration"
            },
            { path: "memberAbout.content.nodes.appEmbedData.image.duration" },
            { path: "memberAbout.content.nodes.videoData.video.duration" },
            {
              path: "memberAbout.content.nodes.videoData.thumbnail.duration"
            },
            { path: "memberAbout.content.nodes.audioData.audio.duration" },
            {
              path: "memberAbout.content.nodes.audioData.coverImage.duration"
            }
          ]
        },
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "memberAbout.content.metadata.createdTimestamp" },
            { path: "memberAbout.content.metadata.updatedTimestamp" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateMemberAbout;
}
function deleteMemberAbout(payload) {
  function __deleteMemberAbout({ host }) {
    const metadata = {
      entityFqdn: "wix.members.about.v2.member_about",
      method: "DELETE",
      methodFqn: "wix.members.about.v2.MembersAbout.DeleteMemberAbout",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixMembersAboutV2MembersAboutUrl({
        protoPath: "/v2/abouts/{id}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteMemberAbout;
}
function getMemberAbout(payload) {
  function __getMemberAbout({ host }) {
    const metadata = {
      entityFqdn: "wix.members.about.v2.member_about",
      method: "GET",
      methodFqn: "wix.members.about.v2.MembersAbout.GetMemberAbout",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixMembersAboutV2MembersAboutUrl({
        protoPath: "/v2/abouts/{id}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "memberAbout.content.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.options.item.ratio"
            },
            { path: "memberAbout.content.nodes.imageData.image.duration" },
            { path: "memberAbout.content.nodes.mapData.mapSettings.lat" },
            { path: "memberAbout.content.nodes.mapData.mapSettings.lng" },
            {
              path: "memberAbout.content.nodes.pollData.poll.image.duration"
            },
            {
              path: "memberAbout.content.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "memberAbout.content.nodes.pollData.design.poll.background.image.duration"
            },
            { path: "memberAbout.content.nodes.appEmbedData.image.duration" },
            { path: "memberAbout.content.nodes.videoData.video.duration" },
            {
              path: "memberAbout.content.nodes.videoData.thumbnail.duration"
            },
            { path: "memberAbout.content.nodes.audioData.audio.duration" },
            {
              path: "memberAbout.content.nodes.audioData.coverImage.duration"
            }
          ]
        },
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "memberAbout.content.metadata.createdTimestamp" },
            { path: "memberAbout.content.metadata.updatedTimestamp" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getMemberAbout;
}
function getMyMemberAbout(payload) {
  function __getMyMemberAbout({ host }) {
    const metadata = {
      entityFqdn: "wix.members.about.v2.member_about",
      method: "GET",
      methodFqn: "wix.members.about.v2.MembersAbout.GetMyMemberAbout",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixMembersAboutV2MembersAboutUrl({
        protoPath: "/v2/abouts/my",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "memberAbout.content.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.options.item.ratio"
            },
            { path: "memberAbout.content.nodes.imageData.image.duration" },
            { path: "memberAbout.content.nodes.mapData.mapSettings.lat" },
            { path: "memberAbout.content.nodes.mapData.mapSettings.lng" },
            {
              path: "memberAbout.content.nodes.pollData.poll.image.duration"
            },
            {
              path: "memberAbout.content.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "memberAbout.content.nodes.pollData.design.poll.background.image.duration"
            },
            { path: "memberAbout.content.nodes.appEmbedData.image.duration" },
            { path: "memberAbout.content.nodes.videoData.video.duration" },
            {
              path: "memberAbout.content.nodes.videoData.thumbnail.duration"
            },
            { path: "memberAbout.content.nodes.audioData.audio.duration" },
            {
              path: "memberAbout.content.nodes.audioData.coverImage.duration"
            }
          ]
        },
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "memberAbout.content.metadata.createdTimestamp" },
            { path: "memberAbout.content.metadata.updatedTimestamp" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getMyMemberAbout;
}
function queryMemberAbouts(payload) {
  function __queryMemberAbouts({ host }) {
    const metadata = {
      entityFqdn: "wix.members.about.v2.member_about",
      method: "POST",
      methodFqn: "wix.members.about.v2.MembersAbout.QueryMemberAbouts",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixMembersAboutV2MembersAboutUrl({
        protoPath: "/v2/abouts/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "memberAbouts.content.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "memberAbouts.content.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "memberAbouts.content.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "memberAbouts.content.nodes.galleryData.options.item.ratio"
            },
            { path: "memberAbouts.content.nodes.imageData.image.duration" },
            { path: "memberAbouts.content.nodes.mapData.mapSettings.lat" },
            { path: "memberAbouts.content.nodes.mapData.mapSettings.lng" },
            {
              path: "memberAbouts.content.nodes.pollData.poll.image.duration"
            },
            {
              path: "memberAbouts.content.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "memberAbouts.content.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "memberAbouts.content.nodes.appEmbedData.image.duration"
            },
            { path: "memberAbouts.content.nodes.videoData.video.duration" },
            {
              path: "memberAbouts.content.nodes.videoData.thumbnail.duration"
            },
            { path: "memberAbouts.content.nodes.audioData.audio.duration" },
            {
              path: "memberAbouts.content.nodes.audioData.coverImage.duration"
            }
          ]
        },
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "memberAbouts.content.metadata.createdTimestamp" },
            { path: "memberAbouts.content.metadata.updatedTimestamp" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryMemberAbouts;
}

// src/members-about-v2-member-about-members-about.types.ts
var NodeType = /* @__PURE__ */ ((NodeType2) => {
  NodeType2["PARAGRAPH"] = "PARAGRAPH";
  NodeType2["TEXT"] = "TEXT";
  NodeType2["HEADING"] = "HEADING";
  NodeType2["BULLETED_LIST"] = "BULLETED_LIST";
  NodeType2["ORDERED_LIST"] = "ORDERED_LIST";
  NodeType2["LIST_ITEM"] = "LIST_ITEM";
  NodeType2["BLOCKQUOTE"] = "BLOCKQUOTE";
  NodeType2["CODE_BLOCK"] = "CODE_BLOCK";
  NodeType2["VIDEO"] = "VIDEO";
  NodeType2["DIVIDER"] = "DIVIDER";
  NodeType2["FILE"] = "FILE";
  NodeType2["GALLERY"] = "GALLERY";
  NodeType2["GIF"] = "GIF";
  NodeType2["HTML"] = "HTML";
  NodeType2["IMAGE"] = "IMAGE";
  NodeType2["LINK_PREVIEW"] = "LINK_PREVIEW";
  NodeType2["MAP"] = "MAP";
  NodeType2["POLL"] = "POLL";
  NodeType2["APP_EMBED"] = "APP_EMBED";
  NodeType2["BUTTON"] = "BUTTON";
  NodeType2["COLLAPSIBLE_LIST"] = "COLLAPSIBLE_LIST";
  NodeType2["TABLE"] = "TABLE";
  NodeType2["EMBED"] = "EMBED";
  NodeType2["COLLAPSIBLE_ITEM"] = "COLLAPSIBLE_ITEM";
  NodeType2["COLLAPSIBLE_ITEM_TITLE"] = "COLLAPSIBLE_ITEM_TITLE";
  NodeType2["COLLAPSIBLE_ITEM_BODY"] = "COLLAPSIBLE_ITEM_BODY";
  NodeType2["TABLE_CELL"] = "TABLE_CELL";
  NodeType2["TABLE_ROW"] = "TABLE_ROW";
  NodeType2["EXTERNAL"] = "EXTERNAL";
  NodeType2["AUDIO"] = "AUDIO";
  NodeType2["CAPTION"] = "CAPTION";
  NodeType2["LAYOUT"] = "LAYOUT";
  NodeType2["LAYOUT_CELL"] = "LAYOUT_CELL";
  return NodeType2;
})(NodeType || {});
var WidthType = /* @__PURE__ */ ((WidthType2) => {
  WidthType2["CONTENT"] = "CONTENT";
  WidthType2["SMALL"] = "SMALL";
  WidthType2["ORIGINAL"] = "ORIGINAL";
  WidthType2["FULL_WIDTH"] = "FULL_WIDTH";
  return WidthType2;
})(WidthType || {});
var PluginContainerDataAlignment = /* @__PURE__ */ ((PluginContainerDataAlignment2) => {
  PluginContainerDataAlignment2["CENTER"] = "CENTER";
  PluginContainerDataAlignment2["LEFT"] = "LEFT";
  PluginContainerDataAlignment2["RIGHT"] = "RIGHT";
  return PluginContainerDataAlignment2;
})(PluginContainerDataAlignment || {});
var ButtonDataType = /* @__PURE__ */ ((ButtonDataType2) => {
  ButtonDataType2["LINK"] = "LINK";
  ButtonDataType2["ACTION"] = "ACTION";
  return ButtonDataType2;
})(ButtonDataType || {});
var Target = /* @__PURE__ */ ((Target2) => {
  Target2["SELF"] = "SELF";
  Target2["BLANK"] = "BLANK";
  Target2["PARENT"] = "PARENT";
  Target2["TOP"] = "TOP";
  return Target2;
})(Target || {});
var TextAlignment = /* @__PURE__ */ ((TextAlignment2) => {
  TextAlignment2["AUTO"] = "AUTO";
  TextAlignment2["LEFT"] = "LEFT";
  TextAlignment2["RIGHT"] = "RIGHT";
  TextAlignment2["CENTER"] = "CENTER";
  TextAlignment2["JUSTIFY"] = "JUSTIFY";
  return TextAlignment2;
})(TextAlignment || {});
var LineStyle = /* @__PURE__ */ ((LineStyle2) => {
  LineStyle2["SINGLE"] = "SINGLE";
  LineStyle2["DOUBLE"] = "DOUBLE";
  LineStyle2["DASHED"] = "DASHED";
  LineStyle2["DOTTED"] = "DOTTED";
  return LineStyle2;
})(LineStyle || {});
var Width = /* @__PURE__ */ ((Width2) => {
  Width2["LARGE"] = "LARGE";
  Width2["MEDIUM"] = "MEDIUM";
  Width2["SMALL"] = "SMALL";
  return Width2;
})(Width || {});
var DividerDataAlignment = /* @__PURE__ */ ((DividerDataAlignment2) => {
  DividerDataAlignment2["CENTER"] = "CENTER";
  DividerDataAlignment2["LEFT"] = "LEFT";
  DividerDataAlignment2["RIGHT"] = "RIGHT";
  return DividerDataAlignment2;
})(DividerDataAlignment || {});
var ViewMode = /* @__PURE__ */ ((ViewMode2) => {
  ViewMode2["NONE"] = "NONE";
  ViewMode2["FULL"] = "FULL";
  ViewMode2["MINI"] = "MINI";
  return ViewMode2;
})(ViewMode || {});
var LayoutType = /* @__PURE__ */ ((LayoutType2) => {
  LayoutType2["COLLAGE"] = "COLLAGE";
  LayoutType2["MASONRY"] = "MASONRY";
  LayoutType2["GRID"] = "GRID";
  LayoutType2["THUMBNAIL"] = "THUMBNAIL";
  LayoutType2["SLIDER"] = "SLIDER";
  LayoutType2["SLIDESHOW"] = "SLIDESHOW";
  LayoutType2["PANORAMA"] = "PANORAMA";
  LayoutType2["COLUMN"] = "COLUMN";
  LayoutType2["MAGIC"] = "MAGIC";
  LayoutType2["FULLSIZE"] = "FULLSIZE";
  return LayoutType2;
})(LayoutType || {});
var Orientation = /* @__PURE__ */ ((Orientation2) => {
  Orientation2["ROWS"] = "ROWS";
  Orientation2["COLUMNS"] = "COLUMNS";
  return Orientation2;
})(Orientation || {});
var Crop = /* @__PURE__ */ ((Crop2) => {
  Crop2["FILL"] = "FILL";
  Crop2["FIT"] = "FIT";
  return Crop2;
})(Crop || {});
var ThumbnailsAlignment = /* @__PURE__ */ ((ThumbnailsAlignment2) => {
  ThumbnailsAlignment2["TOP"] = "TOP";
  ThumbnailsAlignment2["RIGHT"] = "RIGHT";
  ThumbnailsAlignment2["BOTTOM"] = "BOTTOM";
  ThumbnailsAlignment2["LEFT"] = "LEFT";
  ThumbnailsAlignment2["NONE"] = "NONE";
  return ThumbnailsAlignment2;
})(ThumbnailsAlignment || {});
var GIFType = /* @__PURE__ */ ((GIFType2) => {
  GIFType2["NORMAL"] = "NORMAL";
  GIFType2["STICKER"] = "STICKER";
  return GIFType2;
})(GIFType || {});
var Source = /* @__PURE__ */ ((Source2) => {
  Source2["HTML"] = "HTML";
  Source2["ADSENSE"] = "ADSENSE";
  return Source2;
})(Source || {});
var StylesPosition = /* @__PURE__ */ ((StylesPosition2) => {
  StylesPosition2["START"] = "START";
  StylesPosition2["END"] = "END";
  StylesPosition2["TOP"] = "TOP";
  StylesPosition2["HIDDEN"] = "HIDDEN";
  return StylesPosition2;
})(StylesPosition || {});
var MapType = /* @__PURE__ */ ((MapType2) => {
  MapType2["ROADMAP"] = "ROADMAP";
  MapType2["SATELITE"] = "SATELITE";
  MapType2["HYBRID"] = "HYBRID";
  MapType2["TERRAIN"] = "TERRAIN";
  return MapType2;
})(MapType || {});
var ViewRole = /* @__PURE__ */ ((ViewRole2) => {
  ViewRole2["CREATOR"] = "CREATOR";
  ViewRole2["VOTERS"] = "VOTERS";
  ViewRole2["EVERYONE"] = "EVERYONE";
  return ViewRole2;
})(ViewRole || {});
var VoteRole = /* @__PURE__ */ ((VoteRole2) => {
  VoteRole2["SITE_MEMBERS"] = "SITE_MEMBERS";
  VoteRole2["ALL"] = "ALL";
  return VoteRole2;
})(VoteRole || {});
var PollLayoutType = /* @__PURE__ */ ((PollLayoutType2) => {
  PollLayoutType2["LIST"] = "LIST";
  PollLayoutType2["GRID"] = "GRID";
  return PollLayoutType2;
})(PollLayoutType || {});
var PollLayoutDirection = /* @__PURE__ */ ((PollLayoutDirection2) => {
  PollLayoutDirection2["LTR"] = "LTR";
  PollLayoutDirection2["RTL"] = "RTL";
  return PollLayoutDirection2;
})(PollLayoutDirection || {});
var BackgroundType = /* @__PURE__ */ ((BackgroundType2) => {
  BackgroundType2["COLOR"] = "COLOR";
  BackgroundType2["IMAGE"] = "IMAGE";
  BackgroundType2["GRADIENT"] = "GRADIENT";
  return BackgroundType2;
})(BackgroundType || {});
var DecorationType = /* @__PURE__ */ ((DecorationType2) => {
  DecorationType2["BOLD"] = "BOLD";
  DecorationType2["ITALIC"] = "ITALIC";
  DecorationType2["UNDERLINE"] = "UNDERLINE";
  DecorationType2["SPOILER"] = "SPOILER";
  DecorationType2["ANCHOR"] = "ANCHOR";
  DecorationType2["MENTION"] = "MENTION";
  DecorationType2["LINK"] = "LINK";
  DecorationType2["COLOR"] = "COLOR";
  DecorationType2["FONT_SIZE"] = "FONT_SIZE";
  DecorationType2["EXTERNAL"] = "EXTERNAL";
  DecorationType2["STRIKETHROUGH"] = "STRIKETHROUGH";
  DecorationType2["SUPERSCRIPT"] = "SUPERSCRIPT";
  DecorationType2["SUBSCRIPT"] = "SUBSCRIPT";
  return DecorationType2;
})(DecorationType || {});
var FontType = /* @__PURE__ */ ((FontType2) => {
  FontType2["PX"] = "PX";
  FontType2["EM"] = "EM";
  return FontType2;
})(FontType || {});
var Position = /* @__PURE__ */ ((Position2) => {
  Position2["START"] = "START";
  Position2["END"] = "END";
  Position2["TOP"] = "TOP";
  return Position2;
})(Position || {});
var AspectRatio = /* @__PURE__ */ ((AspectRatio2) => {
  AspectRatio2["SQUARE"] = "SQUARE";
  AspectRatio2["RECTANGLE"] = "RECTANGLE";
  return AspectRatio2;
})(AspectRatio || {});
var Resizing = /* @__PURE__ */ ((Resizing2) => {
  Resizing2["FILL"] = "FILL";
  Resizing2["FIT"] = "FIT";
  return Resizing2;
})(Resizing || {});
var Placement = /* @__PURE__ */ ((Placement2) => {
  Placement2["IMAGE"] = "IMAGE";
  Placement2["PRODUCT_INFO"] = "PRODUCT_INFO";
  return Placement2;
})(Placement || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["CONTAINED"] = "CONTAINED";
  Type2["FRAMELESS"] = "FRAMELESS";
  return Type2;
})(Type || {});
var Alignment = /* @__PURE__ */ ((Alignment2) => {
  Alignment2["START"] = "START";
  Alignment2["CENTER"] = "CENTER";
  Alignment2["END"] = "END";
  return Alignment2;
})(Alignment || {});
var Layout = /* @__PURE__ */ ((Layout2) => {
  Layout2["STACKED"] = "STACKED";
  Layout2["SIDE_BY_SIDE"] = "SIDE_BY_SIDE";
  return Layout2;
})(Layout || {});
var AppType = /* @__PURE__ */ ((AppType2) => {
  AppType2["PRODUCT"] = "PRODUCT";
  AppType2["EVENT"] = "EVENT";
  AppType2["BOOKING"] = "BOOKING";
  return AppType2;
})(AppType || {});
var InitialExpandedItems = /* @__PURE__ */ ((InitialExpandedItems2) => {
  InitialExpandedItems2["FIRST"] = "FIRST";
  InitialExpandedItems2["ALL"] = "ALL";
  InitialExpandedItems2["NONE"] = "NONE";
  return InitialExpandedItems2;
})(InitialExpandedItems || {});
var Direction = /* @__PURE__ */ ((Direction2) => {
  Direction2["LTR"] = "LTR";
  Direction2["RTL"] = "RTL";
  return Direction2;
})(Direction || {});
var VerticalAlignment = /* @__PURE__ */ ((VerticalAlignment2) => {
  VerticalAlignment2["TOP"] = "TOP";
  VerticalAlignment2["MIDDLE"] = "MIDDLE";
  VerticalAlignment2["BOTTOM"] = "BOTTOM";
  return VerticalAlignment2;
})(VerticalAlignment || {});
var NullValue = /* @__PURE__ */ ((NullValue2) => {
  NullValue2["NULL_VALUE"] = "NULL_VALUE";
  return NullValue2;
})(NullValue || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/members-about-v2-member-about-members-about.meta.ts
function createMemberAbout2() {
  const payload = {};
  const getRequestOptions = createMemberAbout(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/abouts",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateMemberAbout2() {
  const payload = { memberAbout: { id: ":memberAboutId" } };
  const getRequestOptions = updateMemberAbout(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v2/abouts/{memberAbout.id}",
    pathParams: { memberAboutId: "memberAboutId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteMemberAbout2() {
  const payload = { id: ":id" };
  const getRequestOptions = deleteMemberAbout(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v2/abouts/{id}",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getMemberAbout2() {
  const payload = { id: ":id" };
  const getRequestOptions = getMemberAbout(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/abouts/{id}",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getMyMemberAbout2() {
  const payload = {};
  const getRequestOptions = getMyMemberAbout(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/abouts/my",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryMemberAbouts2() {
  const payload = {};
  const getRequestOptions = queryMemberAbouts(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/abouts/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  Alignment as AlignmentOriginal,
  AppType as AppTypeOriginal,
  AspectRatio as AspectRatioOriginal,
  BackgroundType as BackgroundTypeOriginal,
  ButtonDataType as ButtonDataTypeOriginal,
  Crop as CropOriginal,
  DecorationType as DecorationTypeOriginal,
  Direction as DirectionOriginal,
  DividerDataAlignment as DividerDataAlignmentOriginal,
  FontType as FontTypeOriginal,
  GIFType as GIFTypeOriginal,
  InitialExpandedItems as InitialExpandedItemsOriginal,
  Layout as LayoutOriginal,
  LayoutType as LayoutTypeOriginal,
  LineStyle as LineStyleOriginal,
  MapType as MapTypeOriginal,
  NodeType as NodeTypeOriginal,
  NullValue as NullValueOriginal,
  Orientation as OrientationOriginal,
  Placement as PlacementOriginal,
  PluginContainerDataAlignment as PluginContainerDataAlignmentOriginal,
  PollLayoutDirection as PollLayoutDirectionOriginal,
  PollLayoutType as PollLayoutTypeOriginal,
  Position as PositionOriginal,
  Resizing as ResizingOriginal,
  SortOrder as SortOrderOriginal,
  Source as SourceOriginal,
  StylesPosition as StylesPositionOriginal,
  Target as TargetOriginal,
  TextAlignment as TextAlignmentOriginal,
  ThumbnailsAlignment as ThumbnailsAlignmentOriginal,
  Type as TypeOriginal,
  VerticalAlignment as VerticalAlignmentOriginal,
  ViewMode as ViewModeOriginal,
  ViewRole as ViewRoleOriginal,
  VoteRole as VoteRoleOriginal,
  WebhookIdentityType as WebhookIdentityTypeOriginal,
  Width as WidthOriginal,
  WidthType as WidthTypeOriginal,
  createMemberAbout2 as createMemberAbout,
  deleteMemberAbout2 as deleteMemberAbout,
  getMemberAbout2 as getMemberAbout,
  getMyMemberAbout2 as getMyMemberAbout,
  queryMemberAbouts2 as queryMemberAbouts,
  updateMemberAbout2 as updateMemberAbout
};
