// @wix/auto_sdk_bookings_booking-fees/build/internal/es/meta.mjs
// src/bookings-fees-v1-booking-fee-booking-fees.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixBookingsFeesV1BookingFeesUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/booking-fees",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/booking-fees",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/_api/booking-fees",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/booking-fees",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_bookings_booking-fees";
function listBookingFeesByBookingIds(payload) {
  function __listBookingFeesByBookingIds({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.fees.v1.booking_fee",
      method: "GET",
      methodFqn: "wix.bookings.fees.v1.BookingFees.ListBookingFeesByBookingIds",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsFeesV1BookingFeesUrl({
        protoPath: "/v1/booking-fees",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __listBookingFeesByBookingIds;
}
function applyBookingFeesToOrder(payload) {
  function __applyBookingFeesToOrder({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.fees.v1.booking_fee",
      method: "POST",
      methodFqn: "wix.bookings.fees.v1.BookingFees.ApplyBookingFeesToOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsFeesV1BookingFeesUrl({
        protoPath: "/v1/booking-fees/apply",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __applyBookingFeesToOrder;
}
function collectAppliedBookingFees(payload) {
  function __collectAppliedBookingFees({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.fees.v1.booking_fee",
      method: "POST",
      methodFqn: "wix.bookings.fees.v1.BookingFees.CollectAppliedBookingFees",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsFeesV1BookingFeesUrl({
        protoPath: "/v1/booking-fees/collect",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __collectAppliedBookingFees;
}

// src/bookings-fees-v1-booking-fee-booking-fees.types.ts
var BookingFeeStatus = /* @__PURE__ */ ((BookingFeeStatus2) => {
  BookingFeeStatus2["UNKNOWN_STATUS"] = "UNKNOWN_STATUS";
  BookingFeeStatus2["PREVIEW"] = "PREVIEW";
  BookingFeeStatus2["NOT_YET_APPLIED_TO_ORDER"] = "NOT_YET_APPLIED_TO_ORDER";
  BookingFeeStatus2["APPLIED_TO_ORDER"] = "APPLIED_TO_ORDER";
  return BookingFeeStatus2;
})(BookingFeeStatus || {});
var Trigger = /* @__PURE__ */ ((Trigger2) => {
  Trigger2["UNKNOWN_TRIGGER"] = "UNKNOWN_TRIGGER";
  Trigger2["NOT_ATTENDED"] = "NOT_ATTENDED";
  Trigger2["BOOKING_CANCELED"] = "BOOKING_CANCELED";
  return Trigger2;
})(Trigger || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/bookings-fees-v1-booking-fee-booking-fees.meta.ts
function listBookingFeesByBookingIds2() {
  const payload = {};
  const getRequestOptions = listBookingFeesByBookingIds(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/booking-fees",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function applyBookingFeesToOrder2() {
  const payload = {};
  const getRequestOptions = applyBookingFeesToOrder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/booking-fees/apply",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function collectAppliedBookingFees2() {
  const payload = {};
  const getRequestOptions = collectAppliedBookingFees(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/booking-fees/collect",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  BookingFeeStatus as BookingFeeStatusOriginal,
  Trigger as TriggerOriginal,
  WebhookIdentityType as WebhookIdentityTypeOriginal,
  applyBookingFeesToOrder2 as applyBookingFeesToOrder,
  collectAppliedBookingFees2 as collectAppliedBookingFees,
  listBookingFeesByBookingIds2 as listBookingFeesByBookingIds
};
