// @wix/patterns/dist/cjs/components/CollectionFilter/CollectionFilterState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.CollectionFilterState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _util = require("@wix/bex-core/util");
var _bexCore = require("@wix/bex-core");
var _bi = require("@wix/bex-core/bi");
var _events = require("events");
var _mobx = require("mobx");
class CollectionFilterState {
  constructor(params) {
    (0, _defineProperty2.default)(this, "filter", void 0);
    (0, _defineProperty2.default)(this, "toolbarFiltersState", void 0);
    (0, _defineProperty2.default)(this, "location", void 0);
    (0, _defineProperty2.default)(this, "events", new _events.EventEmitter());
    this.filter = params.filter;
    this.toolbarFiltersState = params.toolbarFiltersState;
    this.location = params.location;
  }
  _reportFilterToggleBiOnNextResult(collection, params) {
    const {
      filter,
      toolbarFiltersState,
      location,
      events
    } = this;
    const {
      toolbar
    } = toolbarFiltersState;
    const onFetch = () => {
      events.off('fetch', onFetch);
      // const currentFilters = ??? remember filters before query result?
      return (0, _mobx.action)(() => {
        const {
          actionType,
          clickedValueKey
        } = params ?? {};
        toolbar.reportBi((0, _bexCore.withoutDefaults)(_bi.cairoFilterToggled)({
          ...toolbar._commonDynamicBiParams(),
          ...collection._commonDynamicBiParams(),
          origin: location,
          filterValue: clickedValueKey,
          filterName: filter.name,
          actionType,
          numFiltersActive: toolbar.collection.query.activeFiltersCount,
          filteredListSize: toolbar.collection.result.total,
          isCustomField: !!filter.isCustomField
        }));
      });
    };
    events.on('fetch', onFetch);
  }
  init() {
    const {
      filter,
      toolbarFiltersState
    } = this;
    const {
      toolbar
    } = toolbarFiltersState;
    const {
      multi: {
        collections
      }
    } = toolbar;
    const disposers = [...collections.filter(({
      query: {
        filtersArray
      }
    }) => filtersArray.includes(filter)).flatMap(collection => [(0, _util.addEventListener)(filter.events, 'beforeRefresh', params => {
      this._reportFilterToggleBiOnNextResult(collection, params);
    }), (0, _util.addEventListener)(collection.emitter, 'fetch', () => {
      const ls = this.events.listeners('fetch').map(l => l());
      return () => {
        ls.forEach(l => l());
      };
    })])];
    return () => {
      disposers.forEach(d => d());
    };
  }
}
exports.CollectionFilterState = CollectionFilterState;
