// @wix/auto_sdk_payments_disputes/build/internal/es/index.mjs
// src/payments-disputes-v1-dispute-disputes.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/payments-disputes-v1-dispute-disputes.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixPaymentsDisputesV1DisputeServiceUrl(opts) {
  const domainToMappings = {};
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_disputes";
function getDispute(payload) {
  function __getDispute({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.disputes.v1.dispute",
      method: "GET",
      methodFqn: "wix.payments.disputes.v1.DisputeService.GetDispute",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputesV1DisputeServiceUrl({
        protoPath: "/v1/disputes/{disputeId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "dispute.dueDate" },
            { path: "dispute.createdDate" },
            { path: "dispute.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getDispute;
}
function queryDisputes(payload) {
  function __queryDisputes({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.disputes.v1.dispute",
      method: "POST",
      methodFqn: "wix.payments.disputes.v1.DisputeService.QueryDisputes",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputesV1DisputeServiceUrl({
        protoPath: "/v1/disputes/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "disputes.dueDate" },
            { path: "disputes.createdDate" },
            { path: "disputes.updatedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixPaymentsDisputesV1DisputeServiceUrl({
            protoPath: "/v1/disputes/query",
            data: payload,
            host
          }),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryDisputes;
}
function getDisputeActions(payload) {
  function __getDisputeActions({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.disputes.v1.dispute",
      method: "GET",
      methodFqn: "wix.payments.disputes.v1.DisputeService.GetDisputeActions",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputesV1DisputeServiceUrl({
        protoPath: "/v1/disputes/{disputeId}/actions",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "actions.dueDate" }]
        }
      ])
    };
    return metadata;
  }
  return __getDisputeActions;
}
function acceptDispute(payload) {
  function __acceptDispute({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.disputes.v1.dispute",
      method: "POST",
      methodFqn: "wix.payments.disputes.v1.DisputeService.AcceptDispute",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputesV1DisputeServiceUrl({
        protoPath: "/v1/disputes/{disputeId}/accept",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "dispute.dueDate" },
            { path: "dispute.createdDate" },
            { path: "dispute.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __acceptDispute;
}
function submitDisputeEvidence(payload) {
  function __submitDisputeEvidence({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.disputes.v1.dispute",
      method: "POST",
      methodFqn: "wix.payments.disputes.v1.DisputeService.SubmitDisputeEvidence",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputesV1DisputeServiceUrl({
        protoPath: "/v1/disputes/{disputeId}/submit-evidence",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "dispute.dueDate" },
            { path: "dispute.createdDate" },
            { path: "dispute.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __submitDisputeEvidence;
}
function bulkUpdateDisputeTags(payload) {
  function __bulkUpdateDisputeTags({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.disputes.v1.dispute",
      method: "POST",
      methodFqn: "wix.payments.disputes.v1.DisputeService.BulkUpdateDisputeTags",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputesV1DisputeServiceUrl({
        protoPath: "/v1/bulk/disputes/update-tags",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateDisputeTags;
}
function bulkUpdateDisputeTagsByFilter(payload) {
  function __bulkUpdateDisputeTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.disputes.v1.dispute",
      method: "POST",
      methodFqn: "wix.payments.disputes.v1.DisputeService.BulkUpdateDisputeTagsByFilter",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputesV1DisputeServiceUrl({
        protoPath: "/v1/bulk/disputes/update-tags-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateDisputeTagsByFilter;
}

// src/payments-disputes-v1-dispute-disputes.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var DisputeStage = /* @__PURE__ */ ((DisputeStage2) => {
  DisputeStage2["UNKNOWN_DISPUTE_STAGE"] = "UNKNOWN_DISPUTE_STAGE";
  DisputeStage2["RFI"] = "RFI";
  DisputeStage2["CHARGEBACK"] = "CHARGEBACK";
  return DisputeStage2;
})(DisputeStage || {});
var DisputeReason = /* @__PURE__ */ ((DisputeReason2) => {
  DisputeReason2["UNKNOWN_DISPUTE_REASON"] = "UNKNOWN_DISPUTE_REASON";
  DisputeReason2["FRAUD_CARD_PRESENT"] = "FRAUD_CARD_PRESENT";
  DisputeReason2["FRAUD_CARD_ABSENT"] = "FRAUD_CARD_ABSENT";
  DisputeReason2["DUPLICATE_PROCESSING"] = "DUPLICATE_PROCESSING";
  DisputeReason2["SERVICES_NOT_PROVIDED"] = "SERVICES_NOT_PROVIDED";
  DisputeReason2["CANCELED_RECURRING"] = "CANCELED_RECURRING";
  DisputeReason2["NOT_AS_DESCRIBED"] = "NOT_AS_DESCRIBED";
  DisputeReason2["COUNTERFEIT"] = "COUNTERFEIT";
  DisputeReason2["MISREPRESENTATION"] = "MISREPRESENTATION";
  DisputeReason2["CANCELED"] = "CANCELED";
  DisputeReason2["OTHER"] = "OTHER";
  return DisputeReason2;
})(DisputeReason || {});
var DisputeStatus = /* @__PURE__ */ ((DisputeStatus2) => {
  DisputeStatus2["UNKNOWN_DISPUTE_STATUS"] = "UNKNOWN_DISPUTE_STATUS";
  DisputeStatus2["WAITING_MERCHANT"] = "WAITING_MERCHANT";
  DisputeStatus2["UNDER_REVIEW"] = "UNDER_REVIEW";
  DisputeStatus2["WAITING_BUYER"] = "WAITING_BUYER";
  DisputeStatus2["WON"] = "WON";
  DisputeStatus2["LOST"] = "LOST";
  return DisputeStatus2;
})(DisputeStatus || {});
var SellerProtection = /* @__PURE__ */ ((SellerProtection2) => {
  SellerProtection2["UNKNOWN_SELLER_PROTECTION"] = "UNKNOWN_SELLER_PROTECTION";
  SellerProtection2["NOT_ELIGIBLE"] = "NOT_ELIGIBLE";
  SellerProtection2["ELIGIBLE"] = "ELIGIBLE";
  SellerProtection2["EXTENDED"] = "EXTENDED";
  return SellerProtection2;
})(SellerProtection || {});
var DisputeChannel = /* @__PURE__ */ ((DisputeChannel2) => {
  DisputeChannel2["UNKNOWN_DISPUTE_CHANNEL"] = "UNKNOWN_DISPUTE_CHANNEL";
  DisputeChannel2["INTERNAL"] = "INTERNAL";
  DisputeChannel2["EXTERNAL"] = "EXTERNAL";
  return DisputeChannel2;
})(DisputeChannel || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var DisputeActionType = /* @__PURE__ */ ((DisputeActionType2) => {
  DisputeActionType2["UNKNOWN_DISPUTE_ACTION"] = "UNKNOWN_DISPUTE_ACTION";
  DisputeActionType2["ACCEPT"] = "ACCEPT";
  DisputeActionType2["SUBMIT_EVIDENCE"] = "SUBMIT_EVIDENCE";
  return DisputeActionType2;
})(DisputeActionType || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function getDispute2(disputeId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    disputeId
  });
  const reqOpts = getDispute(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.dispute;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { disputeId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["disputeId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryDisputes2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryDisputes(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [])
      );
      return {
        items: transformedData?.disputes,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function getDisputeActions2(disputeId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    disputeId
  });
  const reqOpts = getDisputeActions(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { disputeId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["disputeId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function acceptDispute2(disputeId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    disputeId
  });
  const reqOpts = acceptDispute(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { disputeId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["disputeId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function submitDisputeEvidence2(disputeId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    disputeId
  });
  const reqOpts = submitDisputeEvidence(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { disputeId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["disputeId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateDisputeTags2(ids, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    ids,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateDisputeTags(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          ids: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["ids", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateDisputeTagsByFilter2(filter, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    filter,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateDisputeTagsByFilter(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          filter: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["filter", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/payments-disputes-v1-dispute-disputes.public.ts
function getDispute3(httpClient) {
  return (disputeId) => getDispute2(
    disputeId,
    // @ts-ignore
    { httpClient }
  );
}
function queryDisputes3(httpClient) {
  return () => queryDisputes2(
    // @ts-ignore
    { httpClient }
  );
}
function getDisputeActions3(httpClient) {
  return (disputeId) => getDisputeActions2(
    disputeId,
    // @ts-ignore
    { httpClient }
  );
}
function acceptDispute3(httpClient) {
  return (disputeId) => acceptDispute2(
    disputeId,
    // @ts-ignore
    { httpClient }
  );
}
function submitDisputeEvidence3(httpClient) {
  return (disputeId) => submitDisputeEvidence2(
    disputeId,
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdateDisputeTags3(httpClient) {
  return (ids, options) => bulkUpdateDisputeTags2(
    ids,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdateDisputeTagsByFilter3(httpClient) {
  return (filter, options) => bulkUpdateDisputeTagsByFilter2(
    filter,
    options,
    // @ts-ignore
    { httpClient }
  );
}

// src/payments-disputes-v1-dispute-disputes.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
var getDispute4 = /* @__PURE__ */ createRESTModule(getDispute3);
var queryDisputes4 = /* @__PURE__ */ createRESTModule(queryDisputes3);
var getDisputeActions4 = /* @__PURE__ */ createRESTModule(getDisputeActions3);
var acceptDispute4 = /* @__PURE__ */ createRESTModule(acceptDispute3);
var submitDisputeEvidence4 = /* @__PURE__ */ createRESTModule(submitDisputeEvidence3);
var bulkUpdateDisputeTags4 = /* @__PURE__ */ createRESTModule(bulkUpdateDisputeTags3);
var bulkUpdateDisputeTagsByFilter4 = /* @__PURE__ */ createRESTModule(bulkUpdateDisputeTagsByFilter3);
export {
  DisputeActionType,
  DisputeChannel,
  DisputeReason,
  DisputeStage,
  DisputeStatus,
  SellerProtection,
  SortOrder,
  WebhookIdentityType,
  acceptDispute4 as acceptDispute,
  bulkUpdateDisputeTags4 as bulkUpdateDisputeTags,
  bulkUpdateDisputeTagsByFilter4 as bulkUpdateDisputeTagsByFilter,
  getDispute4 as getDispute,
  getDisputeActions4 as getDisputeActions,
  queryDisputes4 as queryDisputes,
  submitDisputeEvidence4 as submitDisputeEvidence
};
