// @wix/auto_sdk_payments_disputes/build/internal/es/meta.mjs
// src/payments-disputes-v1-dispute-disputes.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixPaymentsDisputesV1DisputeServiceUrl(opts) {
  const domainToMappings = {};
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_disputes";
function getDispute(payload) {
  function __getDispute({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.disputes.v1.dispute",
      method: "GET",
      methodFqn: "wix.payments.disputes.v1.DisputeService.GetDispute",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputesV1DisputeServiceUrl({
        protoPath: "/v1/disputes/{disputeId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "dispute.dueDate" },
            { path: "dispute.createdDate" },
            { path: "dispute.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getDispute;
}
function queryDisputes(payload) {
  function __queryDisputes({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.disputes.v1.dispute",
      method: "POST",
      methodFqn: "wix.payments.disputes.v1.DisputeService.QueryDisputes",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputesV1DisputeServiceUrl({
        protoPath: "/v1/disputes/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "disputes.dueDate" },
            { path: "disputes.createdDate" },
            { path: "disputes.updatedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixPaymentsDisputesV1DisputeServiceUrl({
            protoPath: "/v1/disputes/query",
            data: payload,
            host
          }),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryDisputes;
}
function getDisputeActions(payload) {
  function __getDisputeActions({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.disputes.v1.dispute",
      method: "GET",
      methodFqn: "wix.payments.disputes.v1.DisputeService.GetDisputeActions",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputesV1DisputeServiceUrl({
        protoPath: "/v1/disputes/{disputeId}/actions",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "actions.dueDate" }]
        }
      ])
    };
    return metadata;
  }
  return __getDisputeActions;
}
function acceptDispute(payload) {
  function __acceptDispute({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.disputes.v1.dispute",
      method: "POST",
      methodFqn: "wix.payments.disputes.v1.DisputeService.AcceptDispute",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputesV1DisputeServiceUrl({
        protoPath: "/v1/disputes/{disputeId}/accept",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "dispute.dueDate" },
            { path: "dispute.createdDate" },
            { path: "dispute.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __acceptDispute;
}
function submitDisputeEvidence(payload) {
  function __submitDisputeEvidence({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.disputes.v1.dispute",
      method: "POST",
      methodFqn: "wix.payments.disputes.v1.DisputeService.SubmitDisputeEvidence",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputesV1DisputeServiceUrl({
        protoPath: "/v1/disputes/{disputeId}/submit-evidence",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "dispute.dueDate" },
            { path: "dispute.createdDate" },
            { path: "dispute.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __submitDisputeEvidence;
}
function bulkUpdateDisputeTags(payload) {
  function __bulkUpdateDisputeTags({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.disputes.v1.dispute",
      method: "POST",
      methodFqn: "wix.payments.disputes.v1.DisputeService.BulkUpdateDisputeTags",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputesV1DisputeServiceUrl({
        protoPath: "/v1/bulk/disputes/update-tags",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateDisputeTags;
}
function bulkUpdateDisputeTagsByFilter(payload) {
  function __bulkUpdateDisputeTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.disputes.v1.dispute",
      method: "POST",
      methodFqn: "wix.payments.disputes.v1.DisputeService.BulkUpdateDisputeTagsByFilter",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputesV1DisputeServiceUrl({
        protoPath: "/v1/bulk/disputes/update-tags-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateDisputeTagsByFilter;
}

// src/payments-disputes-v1-dispute-disputes.meta.ts
function getDispute2() {
  const payload = { disputeId: ":disputeId" };
  const getRequestOptions = getDispute(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/disputes/{disputeId}",
    pathParams: { disputeId: "disputeId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryDisputes2() {
  const payload = {};
  const getRequestOptions = queryDisputes(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/disputes/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getDisputeActions2() {
  const payload = { disputeId: ":disputeId" };
  const getRequestOptions = getDisputeActions(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/disputes/{disputeId}/actions",
    pathParams: { disputeId: "disputeId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function acceptDispute2() {
  const payload = { disputeId: ":disputeId" };
  const getRequestOptions = acceptDispute(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/disputes/{disputeId}/accept",
    pathParams: { disputeId: "disputeId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function submitDisputeEvidence2() {
  const payload = { disputeId: ":disputeId" };
  const getRequestOptions = submitDisputeEvidence(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/disputes/{disputeId}/submit-evidence",
    pathParams: { disputeId: "disputeId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateDisputeTags2() {
  const payload = {};
  const getRequestOptions = bulkUpdateDisputeTags(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/disputes/update-tags",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateDisputeTagsByFilter2() {
  const payload = {};
  const getRequestOptions = bulkUpdateDisputeTagsByFilter(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/disputes/update-tags-by-filter",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  acceptDispute2 as acceptDispute,
  bulkUpdateDisputeTags2 as bulkUpdateDisputeTags,
  bulkUpdateDisputeTagsByFilter2 as bulkUpdateDisputeTagsByFilter,
  getDispute2 as getDispute,
  getDisputeActions2 as getDisputeActions,
  queryDisputes2 as queryDisputes,
  submitDisputeEvidence2 as submitDisputeEvidence
};
