// @wix/bex-core/dist/esm/testkit/createCountriesBackend.js
import { Chance } from 'chance';
import { snakeCase, escapeRegExp } from 'lodash';
import { InMemoryBackend } from '../InMemoryBackend';
export const createCountriesBackend = ({ createOne, ...params } = {}) => {
    const chance = new Chance();
    const countries = chance
        .unique(chance.country.bind(chance, { full: true }), chance.countries().length)
        .sort()
        .map((name) => ({
        id: snakeCase(name),
        name,
    }));
    function createCountry(index, { ...rest } = {}) {
        return {
            ...countries[index],
            ...rest,
        };
    }
    return new InMemoryBackend({
        enableTotal: params?.enableTotal,
        createOne: createOne ?? ((index) => createCountry(index)),
        itemKey: (item) => item.id,
        total: countries.length,
        predicate: ({ search }) => {
            const rgx = new RegExp(escapeRegExp(search), 'i');
            return (item) => {
                if (search && !rgx.test(item.name)) {
                    return false;
                }
                return true;
            };
        },
        ...params,
    });
};
