// monocle-ts-esm/lib/At.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.atRecord = exports.atReadonlySet = exports.atReadonlyMap = exports.atReadonlyRecord = exports.fromIso = exports.at = void 0;
const O = __importStar(require("fp-ts-esm/Option"));
const pipeable_1 = require("fp-ts-esm/pipeable");
const RM = __importStar(require("fp-ts-esm/ReadonlyMap"));
const RS = __importStar(require("fp-ts-esm/ReadonlySet"));
const _ = __importStar(require("./internal.js"));
// -------------------------------------------------------------------------------------
// constructors
// -------------------------------------------------------------------------------------
/**
 * @category constructors
 * @since 2.3.8
 */
exports.at = _.at;
/**
 * Lift an instance of `At` using an `Iso`.
 *
 * @category constructors
 * @since 2.3.0
 */
const fromIso = (iso) => (sia) => (0, exports.at)((i) => (0, pipeable_1.pipe)(iso, _.isoAsLens, _.lensComposeLens(sia.at(i))));
exports.fromIso = fromIso;
/**
 * @category constructors
 * @since 2.3.7
 */
exports.atReadonlyRecord = _.atReadonlyRecord;
/**
 * @category constructors
 * @since 2.3.7
 */
const atReadonlyMap = (E) => {
    const lookupE = RM.lookup(E);
    const deleteAtE = RM.deleteAt(E);
    const insertAtE = RM.insertAt(E);
    return () => (0, exports.at)((key) => _.lens((s) => lookupE(key, s), O.fold(() => deleteAtE(key), (a) => insertAtE(key, a))));
};
exports.atReadonlyMap = atReadonlyMap;
/**
 * @category constructors
 * @since 2.3.7
 */
const atReadonlySet = (E) => {
    const elemE = RS.elem(E);
    const insertE = RS.insert(E);
    const removeE = RS.remove(E);
    return (0, exports.at)((a) => {
        const insert = insertE(a);
        const remove = removeE(a);
        return _.lens((s) => elemE(a, s), (b) => (s) => (b ? insert(s) : remove(s)));
    });
};
exports.atReadonlySet = atReadonlySet;
// -------------------------------------------------------------------------------------
// deprecated
// -------------------------------------------------------------------------------------
/**
 * Use `atReadonlyRecord` instead.
 *
 * @category constructors
 * @since 2.3.2
 * @deprecated
 */
exports.atRecord = _.atReadonlyRecord;
