// @wix/auto_sdk_forms_form-submissions/build/internal/es/index.typings.mjs
// src/service-plugins-types.ts
var SubmissionStatus = /* @__PURE__ */ ((SubmissionStatus2) => {
  SubmissionStatus2["UNKNOWN_SUBMISSION_STATUS"] = "UNKNOWN_SUBMISSION_STATUS";
  SubmissionStatus2["PENDING"] = "PENDING";
  SubmissionStatus2["CONFIRMED"] = "CONFIRMED";
  SubmissionStatus2["PAYMENT_WAITING"] = "PAYMENT_WAITING";
  SubmissionStatus2["PAYMENT_CANCELED"] = "PAYMENT_CANCELED";
  return SubmissionStatus2;
})(SubmissionStatus || {});
var ActionType = /* @__PURE__ */ ((ActionType2) => {
  ActionType2["UNKNOWN_ACTION"] = "UNKNOWN_ACTION";
  ActionType2["UPDATE"] = "UPDATE";
  ActionType2["CREATE"] = "CREATE";
  return ActionType2;
})(ActionType || {});
var SubmissionErrorType = /* @__PURE__ */ ((SubmissionErrorType2) => {
  SubmissionErrorType2["UNKNOWN_ERROR"] = "UNKNOWN_ERROR";
  SubmissionErrorType2["TYPE_ERROR"] = "TYPE_ERROR";
  SubmissionErrorType2["REQUIRED_VALUE_ERROR"] = "REQUIRED_VALUE_ERROR";
  SubmissionErrorType2["UNKNOWN_VALUE_ERROR"] = "UNKNOWN_VALUE_ERROR";
  SubmissionErrorType2["MAX_LENGTH_ERROR"] = "MAX_LENGTH_ERROR";
  SubmissionErrorType2["MIN_LENGTH_ERROR"] = "MIN_LENGTH_ERROR";
  SubmissionErrorType2["PATTERN_ERROR"] = "PATTERN_ERROR";
  SubmissionErrorType2["FORMAT_ERROR"] = "FORMAT_ERROR";
  SubmissionErrorType2["MAX_VALUE_ERROR"] = "MAX_VALUE_ERROR";
  SubmissionErrorType2["MIN_VALUE_ERROR"] = "MIN_VALUE_ERROR";
  SubmissionErrorType2["MULTIPLE_OF_VALUE_ERROR"] = "MULTIPLE_OF_VALUE_ERROR";
  SubmissionErrorType2["MIN_ITEMS_ERROR"] = "MIN_ITEMS_ERROR";
  SubmissionErrorType2["MAX_ITEMS_ERROR"] = "MAX_ITEMS_ERROR";
  SubmissionErrorType2["NOT_ALLOWED_VALUE_ERROR"] = "NOT_ALLOWED_VALUE_ERROR";
  SubmissionErrorType2["DISABLED_FORM_ERROR"] = "DISABLED_FORM_ERROR";
  return SubmissionErrorType2;
})(SubmissionErrorType || {});
var IdentityType = /* @__PURE__ */ ((IdentityType2) => {
  IdentityType2["UNKNOWN"] = "UNKNOWN";
  IdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  IdentityType2["MEMBER"] = "MEMBER";
  IdentityType2["WIX_USER"] = "WIX_USER";
  IdentityType2["APP"] = "APP";
  return IdentityType2;
})(IdentityType || {});
export {
  ActionType,
  IdentityType,
  SubmissionErrorType,
  SubmissionStatus
};
