// @wix/bex-core/dist/esm/state/ArchiveFieldModalState.js
import { makeAutoObservable } from 'mobx';
import { ConditionalModalState } from './ConditionalModalState';
import { DataExtensionStateBIReporter } from './DataExtensionStateBIReporter';
export class ArchiveFieldModalState {
    constructor(params) {
        this.biReporter = new DataExtensionStateBIReporter({
            reportBi: params.reportBi,
        });
        this.modal = new ConditionalModalState({
            onConfirm: async () => {
                if (!this.modal.value) {
                    return;
                }
                this.biReporter.reportArchiveFieldModalCtaClicked('Archive');
                await this.dataExtension.archiveField({
                    ...this.modal.value,
                    reportBi: this.biReporter.reportBi,
                });
            },
            onClose: () => {
                this.biReporter.reportArchiveFieldModalCtaClicked('Cancel');
            },
        });
        this.dataExtension = params.dataExtension;
        makeAutoObservable(this);
    }
}
