// @wix/bex-core/dist/esm/state/BulkSelectState.js
import { SelectState } from './SelectState';
import { action, computed, makeObservable, observable, reaction, runInAction, } from 'mobx';
import { Snapshot } from './snapshot';
import { addEventListener } from '../util';
import { EventEmitter } from 'events';
export class BulkSelectState {
    get allSelected() {
        return this.select.defaultStatus.isChecked;
    }
    get hasUnchecked() {
        return !!this.uncheckedValues.length;
    }
    constructor(params) {
        this.events = new EventEmitter();
        this.openModal = () => {
            this.events.emit('openModal');
        };
        this.closeModal = () => {
            this.events.emit('closeModal');
        };
        this.toggleIsSelectAll = action(() => {
            const { allSelected, select: { isEmptyEnabledItems }, } = this;
            if (allSelected || !isEmptyEnabledItems) {
                this.deselectAll();
            }
            else {
                this.selectAll();
            }
        });
        this.collection = params.collection;
        this.scrollToTop = params.scrollToTop;
        this.disableAutoSelectAllCount = params.disableAutoSelectAllCount;
        this.useNewSelectAllLogic = params.useNewSelectAllLogic;
        this.select = new SelectState({
            keyGetter: (item) => item.key,
        });
        this.totalSnapshot = new Snapshot({
            expression: () => this.allSelected,
            effect: () => this.collection.result.total,
        });
        makeObservable(this, {
            allSelected: computed,
            disableAutoSelectAllCount: observable.ref,
            selectAll: action,
            selectOne: action,
            selectOneItem: action,
            deselectOne: action,
            deselectOneItem: action,
            resetToDefaults: action.bound,
            hasUnchecked: computed,
            selectedIds: computed,
            uncheckedIds: computed,
            selectedValues: computed,
            selectedKeyedItems: computed,
            uncheckedValues: computed,
            uncheckedInCurrentCollection: computed,
            selectedCountOrTotal: computed,
            checkedSet: computed,
            indeterminateSet: computed,
        });
    }
    getCollectionItem(key) {
        return this.collection.result._keyedItemsMap.get(key);
    }
    get selectedIds() {
        return this.select.selectedKeys;
    }
    get uncheckedIds() {
        return this.select.toArray
            .filter(({ status }) => !status.isChecked)
            .map(({ value }) => value.key);
    }
    get selectedCount() {
        return this.select.selectedCount;
    }
    get allSelectedItemsFetched() {
        return this.selectedCountOrTotal <= this.collection.result.items.length;
    }
    get selectedValues() {
        return this.select.selectedValues.map(({ item }) => item);
    }
    get checkedSet() {
        return new Set(this.select.toArray
            .filter(({ status }) => status.isChecked)
            .map(({ value }) => value.key));
    }
    get indeterminateSet() {
        return new Set(this.select.toArray
            .filter(({ status }) => status.isIndeterminate)
            .map(({ value }) => value.key));
    }
    get uncheckedValues() {
        return this.select.toArray
            .filter(({ status }) => !status.isChecked)
            .map(({ value }) => value.item);
    }
    isChecked(key) {
        const { select } = this;
        return select.isChecked(key);
    }
    selectAll() {
        const { select, collection } = this;
        select.defaultStatus = select.defaultStatus.final();
        const { result: { keyedItems }, } = collection;
        select.finalizeMany(keyedItems);
    }
    deselectAll() {
        const { select } = this;
        runInAction(() => {
            select.defaultStatus = select.defaultStatus.start();
        });
        select.startAll();
    }
    selectOneItem(item) {
        const { select, useNewSelectAllLogic } = this;
        select.set(item);
        if (!useNewSelectAllLogic) {
            runInAction(() => {
                select.defaultStatus = select.defaultStatus.start();
            });
        }
    }
    selectOne(key) {
        const { collection } = this;
        const { result } = collection;
        const keyedItem = result.get(key);
        if (keyedItem) {
            this.selectOneItem(keyedItem);
        }
    }
    deselectOneItem(keyedItem) {
        const { select, useNewSelectAllLogic } = this;
        select.delete(keyedItem);
        if (!useNewSelectAllLogic) {
            runInAction(() => {
                select.defaultStatus = select.defaultStatus.start();
            });
        }
    }
    deselectOne(key) {
        const { collection } = this;
        const { result } = collection;
        const keyedItem = result.get(key);
        if (keyedItem == null) {
            return;
        }
        this.deselectOneItem(keyedItem);
    }
    resetToDefaults() {
        const { select } = this;
        runInAction(() => {
            select.defaultStatus = select.defaultStatus.start();
        });
        select.clear();
    }
    init() {
        const disposers = [
            this.totalSnapshot.init(),
            addEventListener(this.collection.emitter, 'ready', () => {
                this.totalSnapshot.forceUpdate();
            }),
            reaction(() => this.collection.result.keyedItems, (keyedItems) => {
                const { select, allSelected, disableAutoSelectAllCount, collection: { result: { _keyedItemsMap }, }, } = this;
                if (allSelected && !disableAutoSelectAllCount) {
                    select.syncValues(keyedItems);
                }
                else {
                    select.syncValues(select.selectedKeys
                        .map((key) => _keyedItemsMap.get(key))
                        .filter((e) => e != null));
                }
            }),
        ];
        return () => {
            disposers.forEach((disposer) => disposer());
        };
    }
    get selectedKeyedItems() {
        return this.selectedIds
            .map((id) => this.collection.result.get(id))
            .filter((item) => !!item);
    }
    get isEmpty() {
        return this.select.isEmpty;
    }
    get uncheckedInCurrentCollection() {
        const { uncheckedValues, collection } = this;
        const { itemKey } = collection;
        return uncheckedValues.filter((item) => collection.result.has(itemKey(item)));
    }
    get selectedCountOrTotal() {
        const { allSelected, disableAutoSelectAllCount, selectedIds, useNewSelectAllLogic, uncheckedInCurrentCollection, collection: { result: { total }, }, } = this;
        if (!allSelected || disableAutoSelectAllCount) {
            return selectedIds.length;
        }
        if (useNewSelectAllLogic) {
            return total - uncheckedInCurrentCollection.length;
        }
        return total;
    }
    toggle(keyedItem) {
        this.select.toggle(keyedItem);
    }
}
