// @wix/bex-core/dist/esm/state/BulkActionModalState.js
import { ConditionalModalState } from './ConditionalModalState';
export class BulkActionModalState {
    constructor(params) {
        this.actionModal = new ConditionalModalState();
        this.openModal = () => {
            this.actionModal.modal.open();
        };
        this.closeModal = () => {
            this.actionModal.modal.close();
        };
        this.closeModalAndScrollTop = () => {
            this.closeModal();
            this.scrollToTop();
        };
        this.openConfirmModal = (props) => {
            const { translate: t } = this;
            this.actionModal.open({
                title: t('cairo.confirmModal.title'),
                primaryButtonText: t('cairo.confirm.button'),
                secondaryButtonText: t('cairo.cancel.button'),
                ...props,
            });
        };
        this.translate = params.translate;
        this.scrollToTop = params.scrollToTop;
    }
    get isModalOpen() {
        return this.actionModal.isOpen;
    }
}
