// @wix/bex-core/dist/esm/state/CollectionOptimisticSequenceExecutor.js
import { PromisesSequence } from './PromisesSequence';
import { EventEmitter } from 'events';
export class CollectionOptimisticSequenceExecutor {
    constructor(params) {
        this.seq = new PromisesSequence();
        this.removeToast = null;
        this.events = new EventEmitter();
        this.showToast = params.showToast;
        this.translate = params.translate;
        this.errorMonitor = params.errorMonitor;
        this.onRollback = params.onRollback;
    }
    removeAndShowToast(toastConfig) {
        const { removeToast, showToast } = this;
        removeToast?.();
        this.removeToast = showToast(toastConfig).remove;
    }
    async sequence(params) {
        const { seq, errorMonitor, translate: t, onRollback } = this;
        const { submit, onEventuallyUpdated } = params;
        try {
            await seq.enqueue(async () => {
                await submit();
                onEventuallyUpdated?.();
            });
        }
        catch (err) {
            console.error(err);
            errorMonitor.captureException(err);
            const errorToast = onRollback(err);
            if (errorToast) {
                this.removeAndShowToast({
                    ...errorToast,
                    type: 'ERROR',
                    timeout: 'NONE',
                    action: {
                        uiType: 'LINK',
                        text: t('cairo.toast.retry'),
                        removeToastOnClick: true,
                        ...errorToast.action,
                        onClick: () => {
                            this.removeToast?.();
                            this.removeToast = null;
                            errorToast.action?.onClick?.();
                        },
                    },
                });
                const currentToast = this.removeToast;
                new Promise((resolve) => setTimeout(resolve, 12000)).then(() => {
                    if (currentToast === this.removeToast) {
                        currentToast?.();
                    }
                });
            }
        }
    }
}
