// @wix/bex-core/dist/esm/state/DataExtensionState.js
import { DataExtensionSchemaService, } from '../services';
import { TaskState } from './TaskState';
import { action, computed, makeObservable, observable, runInAction, } from 'mobx';
import { DataExtensionStateBIReporter } from './DataExtensionStateBIReporter';
import { fqdnToString } from '../util';
import { includesAllReadPermissions } from './DataExtension/DataExtensionField';
export const MAX_CUSTOM_FIELDS = 50;
export class DataExtensionState {
    get fqdnString() {
        return fqdnToString(this.fqdn);
    }
    get isFqdnValid() {
        return this.fqdn.resource !== undefined;
    }
    constructor(params) {
        this.initTask = new TaskState();
        this._dataExtensionSchemasBase = [];
        this.container = params.container;
        this.disableUserDefinedWriting = params.disableUserDefinedWriting;
        this.filterable = params.filterable;
        this.closeSidePanel = params.closeSidePanel;
        this.fqdn = params.fqdn;
        this._extensionPoint = params.extensionPoint;
        this.dataExtensionService = new DataExtensionSchemaService(params.container);
        this.containsPii = params.containsPii;
        this.defaultPermissions = params.defaultPermissions;
        makeObservable(this, {
            allUserCustomFields: computed,
            allAppsCustomFields: computed,
            allCustomFields: computed,
            filterableCustomFields: computed,
            availableUserCustomFields: computed,
            availableAppsCustomFields: computed,
            availableCustomFields: computed,
            isReachedMaxCustomFields: computed,
            disableAddCustomField: computed,
            disableAddCustomFieldContent: computed,
            _dataExtensionSchemasBase: observable.ref,
            _dataExtensionSchemas: computed,
            userSchema: computed,
            appSchemas: computed,
            _upsertSchema: action,
            containsPii: observable,
            defaultPermissions: observable,
        });
    }
    get hasWritePermissions() {
        if (this._hasWritePermissions == null) {
            this._hasWritePermissions = this.container.user.permissions.includes('DATA_EXTENSION_SCHEMA.WRITE');
        }
        return this._hasWritePermissions;
    }
    get disableAddCustomField() {
        return !this.hasWritePermissions || this.isReachedMaxCustomFields;
    }
    get disableAddCustomFieldContent() {
        if (!this.hasWritePermissions) {
            return this.container.translate('cairo.customFields.writePermission.tooltip');
        }
        if (this.isReachedMaxCustomFields) {
            return this.container.translate('cairo.customFields.maxFields.tooltip', {
                max: MAX_CUSTOM_FIELDS,
            });
        }
        return '';
    }
    async archiveField({ id, origin, reportBi, }) {
        const biReporter = new DataExtensionStateBIReporter({
            reportBi,
        });
        if (id == null) {
            return;
        }
        try {
            const { dataExtensionSchema } = await this.dataExtensionService.archiveCustomField({
                schema: this.userSchema,
                fqdn: this.fqdnString,
                id,
            });
            if (dataExtensionSchema) {
                this._upsertSchema(dataExtensionSchema);
            }
            this.container.showToast?.({
                message: this.container.translate('cairo.customFields.archive.success.toast'),
                type: 'SUCCESS',
                biName: 'cairo-archive-custom-field-success-toast',
            });
        }
        catch (e) {
            this.container.showToast?.({
                message: this.container.translate('cairo.customFields.archive.technicalError'),
                action: {
                    uiType: 'LINK',
                    text: this.container.translate('cairo.toast.retry'),
                    onClick: () => {
                        biReporter.reportArchiveTryAgain();
                        this.archiveField({ id, origin, reportBi });
                    },
                    removeToastOnClick: true,
                },
                type: 'ERROR',
                biName: 'cairo-archive-custom-field-error-toast',
            });
            biReporter.reportArchiveFieldFailed({
                formValues: this.extractFieldValuesFromSchema(id),
                origin,
            });
        }
    }
    extractFieldValuesFromSchema(fieldId) {
        if (!this.userSchema) {
            return;
        }
        const customField = DataExtensionSchemaService.toCustomField(this.userSchema, fieldId);
        if (!customField || !customField.type || !customField.title) {
            return;
        }
        const { read, write } = customField['x-wix-permissions'];
        const isPiiField = customField['x-wix-pii']?.enabled;
        const readApps = read?.includes('apps');
        const readUsers = read?.includes('users-of-users');
        const writeApps = write?.includes('apps');
        const writeUsers = write?.includes('users-of-users');
        return {
            type: DataExtensionSchemaService.toCustomType(customField),
            name: customField.title,
            key: customField.id,
            readApps,
            readUsers,
            writeApps,
            writeUsers,
            containsPii: Boolean(isPiiField),
        };
    }
    _toCustomField({ key, schema, editable, owner, namespace, }) {
        if (!DataExtensionSchemaService.toCustomType(schema)) {
            return;
        }
        return {
            id: key,
            label: schema?.title || key,
            maxLength: schema?.maxLength,
            options: schema?.enum,
            type: DataExtensionSchemaService.toCustomType(schema),
            editable,
            filterable: schema['x-wix-filterable'],
            owner,
            properties: schema?.properties ?? {},
            namespace,
            permissions: schema['x-wix-permissions'],
        };
    }
    isSupportedFieldType(type) {
        if (!type) {
            return false;
        }
        return ['string', 'number', 'integer', 'boolean', 'array'].includes(type);
    }
    getAvailableCustomFieldsInSchema(properties, { editable, requiredPermissions, owner, namespace, }) {
        const filteredFields = Object.keys(properties).filter((key) => {
            const field = properties[key];
            const isAllowedType = this.isSupportedFieldType(field.type) ||
                (field.type === 'object' && field['x-wix-object-type'] === 'FILE/V1');
            const isPermitted = requiredPermissions == null ||
                includesAllReadPermissions(field, requiredPermissions);
            return !field['x-wix-archived'] && isAllowedType && isPermitted;
        });
        const sorted = filteredFields.sort((a, b) => {
            const aVal = new Date(properties[a]['x-wix-created-date']);
            const bVal = new Date(properties[b]['x-wix-created-date']);
            if (isNaN(bVal.valueOf()) || isNaN(aVal.valueOf())) {
                return 0;
            }
            return aVal.valueOf() - bVal.valueOf();
        });
        return sorted
            .map((key) => this._toCustomField({
            key,
            schema: properties[key],
            editable,
            owner,
            namespace,
        }))
            .filter(Boolean);
    }
    /**
     * Available custom fields, excluding archived fields
     */
    get availableUserCustomFields() {
        const properties = this.userSchema?.jsonSchema?.properties || {};
        return this.getAvailableCustomFieldsInSchema(properties, {
            editable: true,
            owner: 'user',
            namespace: '_user_fields',
        });
    }
    get availableAppsCustomFields() {
        return (this.appSchemas?.reduce((prev, schema) => {
            return [...prev, ...this.getAvailableCustomFieldsForApp(schema)];
        }, []) ?? []);
    }
    getSchemaAndFieldsByNamespace(namespace) {
        const { _dataExtensionSchemas, dataExtensionService: { userFieldsNamespace }, } = this;
        const schema = _dataExtensionSchemas.find((s) => s.namespace === namespace);
        if (!schema) {
            return {};
        }
        const fields = this.getAvailableCustomFieldsInSchema(schema.jsonSchema?.properties || {}, {
            editable: false,
            owner: namespace === userFieldsNamespace ? 'user' : 'app',
            namespace,
        });
        return {
            schema,
            fields,
        };
    }
    getAvailableCustomFieldsForApp(appSchema) {
        const properties = appSchema.jsonSchema?.properties || {};
        return this.getAvailableCustomFieldsInSchema(properties, {
            editable: false,
            requiredPermissions: ['users'],
            owner: 'app',
            namespace: appSchema.namespace,
        });
    }
    get availableCustomFields() {
        return [
            ...this.availableUserCustomFields,
            ...this.availableAppsCustomFields,
        ];
    }
    get allUserCustomFields() {
        const properties = this.userSchema?.jsonSchema?.properties || {};
        return Object.keys(properties)
            .map((key) => this._toCustomField({
            key,
            schema: properties[key],
            editable: true,
            owner: 'user',
            namespace: '_user_fields',
        }))
            .filter(Boolean);
    }
    get allAppsCustomFields() {
        return (this.appSchemas?.reduce((prev, appSchema) => {
            const properties = appSchema?.jsonSchema?.properties || {};
            return prev.concat(Object.keys(properties)
                .map((key) => this._toCustomField({
                key,
                schema: properties[key],
                editable: false,
                owner: 'app',
                namespace: appSchema.namespace,
            }))
                .filter(Boolean));
        }, []) ?? []);
    }
    get allCustomFields() {
        return [...this.allUserCustomFields, ...this.allAppsCustomFields];
    }
    get filterableCustomFields() {
        if (!this.filterable) {
            return [];
        }
        return this.availableCustomFields.filter((field) => field.filterable);
    }
    get isReachedMaxCustomFields() {
        return this.allUserCustomFields.length >= MAX_CUSTOM_FIELDS;
    }
    get allNamespaces() {
        return [
            this.userSchema?.namespace,
            ...(this.appSchemas || []).map((schema) => schema.namespace),
        ].filter(Boolean);
    }
    init() {
        const { initTask } = this;
        initTask.runOnce(async () => {
            await this.fetchSchemas();
        });
        return () => { };
    }
    _upsertSchema(newSchema) {
        const { _dataExtensionSchemasBase } = this;
        const index = _dataExtensionSchemasBase.findIndex((schema) => schema.namespace === newSchema?.namespace);
        if (index === -1) {
            this._dataExtensionSchemasBase = [
                ..._dataExtensionSchemasBase,
                newSchema,
            ];
        }
        else {
            this._dataExtensionSchemasBase = _dataExtensionSchemasBase.map((schema) => schema.namespace === newSchema.namespace ? newSchema : schema);
        }
    }
    get userSchema() {
        const { _dataExtensionSchemas, dataExtensionService: { userFieldsNamespace }, } = this;
        return _dataExtensionSchemas.find((s) => s.namespace === userFieldsNamespace);
    }
    get _dataExtensionSchemas() {
        const { _extensionPoint, _dataExtensionSchemasBase } = this;
        return _dataExtensionSchemasBase.filter((s) => _extensionPoint
            ? s.extensionPoint === _extensionPoint
            : s.extensionPoint == null || s.extensionPoint === 'ROOT');
    }
    get appSchemas() {
        const { _dataExtensionSchemas, dataExtensionService: { userFieldsNamespace }, } = this;
        return _dataExtensionSchemas.filter((s) => s.namespace !== userFieldsNamespace);
    }
    async fetchSchemas() {
        const { dataExtensionService } = this;
        if (this.isFqdnValid) {
            const { dataExtensionSchemas } = await dataExtensionService.fetchSchemas(this.fqdnString);
            runInAction(() => {
                this._dataExtensionSchemasBase = dataExtensionSchemas ?? [];
            });
        }
    }
}
