// @wix/bex-core/dist/esm/state/FormPageState.js
import { computed, makeObservable, observable } from 'mobx';
import { TaskState } from './TaskState';
import { WidgetsFormState } from './WidgetsFormState';
export class FormPageState {
    constructor(props) {
        this.initTask = new TaskState();
        this.submitTask = new TaskState();
        this._isSubmitSuccessful = undefined;
        this._isFormDirty = false;
        this._form = props.form;
        this._widgets = props.widgets ?? new WidgetsFormState();
        makeObservable(this, {
            isFetching: computed,
            _isFormDirty: observable.ref,
            isDirty: computed,
        });
    }
    set isFormDirty(flag) {
        this._isFormDirty = flag;
    }
    get isDirty() {
        return this._isFormDirty || this._widgets.isDirty;
    }
    get form() {
        return this._form;
    }
    get isFetching() {
        return this.initTask.status.isLoading || this.initTask.status.isIdle;
    }
    set isSubmitSuccessful(flag) {
        this._isSubmitSuccessful = flag;
    }
    retryFetch() {
        this.initTask.runOnce();
    }
    /**
     * For backward compatibility, keep this as validate, but it only validates widgets.
     */
    validate() {
        return this._widgets.validate();
    }
    registerWidget(key, state) {
        this._widgets.registerWidget(key, state);
    }
    getWidgetState(key) {
        return this._widgets.get(key);
    }
}
