// @wix/bex-core/dist/esm/state/ModalState.js
import { ToggleState } from './ToggleState';
import { action, computed, makeObservable, observable, runInAction, } from 'mobx';
import { queryStatus } from '../model';
import { EventEmitter } from 'events';
export class ModalState {
    constructor(params = {}) {
        this.openToggle = new ToggleState();
        this.toggle = this.openToggle.toggle;
        this.confirming = queryStatus.idle;
        this.events = new EventEmitter();
        this.requestClose = () => {
            this.events.emit('requestClose');
        };
        makeObservable(this, {
            confirming: observable,
            open: action.bound,
            confirm: action.bound,
            close: action.bound,
            isOpen: computed,
        });
        this.onConfirm = params.onConfirm;
        this.onClose = params.onClose;
    }
    open() {
        this.events.emit('open');
        this.openToggle.setOn();
    }
    promise() {
        return new Promise((resolve) => {
            this.events.once('close', (options) => {
                resolve(options);
            });
        });
    }
    async confirm() {
        try {
            this.confirming = queryStatus.loading;
            await this.onConfirm?.();
            runInAction(() => {
                this.confirming = queryStatus.success;
                this.events.emit('close', {
                    result: 'confirmed',
                });
                return this.openToggle.setOff();
            });
        }
        catch (e) {
            runInAction(() => {
                this.confirming = queryStatus.error;
            });
        }
    }
    close() {
        this.onClose?.();
        this.events.emit('close', { result: 'dismissed' });
        return this.openToggle.setOff();
    }
    decline() {
        this.onClose?.();
        this.events.emit('close', { result: 'declined' });
        return this.openToggle.setOff();
    }
    get isOpen() {
        return this.openToggle.isOn;
    }
}
