// @wix/bex-core/dist/esm/state/onItemsCreated.js
import { action } from 'mobx';
import { newItemCreationEnd } from '@wix/bex-utils/@wix/bi-logger-os-data/v2';
async function _onItemsCreated(collection, { items, reportBi }) {
    const { queryClient, query, itemKey } = collection;
    query.resetFilters();
    query.resetPage();
    const { queryKey } = collection.queryKeyOptions;
    const existingData = queryClient.getQueryData(queryKey);
    if (existingData) {
        queryClient.setQueryData(queryKey, {
            ...existingData,
            items: [...items, ...existingData.items],
            total: (existingData.total ?? existingData.items.length) + items.length,
            ...(existingData.available && {
                available: existingData.available + items.length,
            }),
        });
    }
    await collection.clearResultAndMoveToStart();
    reportBi(newItemCreationEnd({
        numSelectedItems: collection.bulkSelect.select.selectedCount,
        itemId: JSON.stringify(items.map(itemKey)),
        resultType: 'success',
    }));
    collection.scrollableContent?.scrollTo?.({
        top: 0,
        behavior: 'smooth',
    });
}
export const onItemsCreated = action(_onItemsCreated);
