// @wix/bex-core/dist/esm/state/OnlineState.js
import { EventEmitter } from 'events';
export class OnlineState {
    constructor(params) {
        this.events = new EventEmitter();
        this._removeOfflineToastTimeoutId = null;
        this._removeOfflineToast = null;
        this.onlineManager = params.onlineManager;
        this.focusManager = params.focusManager;
        this.showToast = params.showToast;
        this.translate = params.translate;
    }
    get isOnline() {
        return this.onlineManager.isOnline();
    }
    showOfflineToast() {
        const { showToast, translate: t } = this;
        if (showToast == null) {
            return;
        }
        this._removeOfflineToast?.();
        this._removeOfflineToast = showToast({
            type: 'ERROR',
            message: t('cairo.offline'),
            biName: 'cairo-offline-error',
            timeout: 'NONE',
        }).remove;
        // this._removeOfflineToastTimeoutId = setTimeout(() => {
        //   this._removeOfflineToastTimeoutId = null;
        //   this._removeOfflineToast?.();
        // }, 6000);
    }
    init() {
        const { focusManager, onlineManager, events } = this;
        const disposers = [
            onlineManager.subscribe(() => {
                const { isOnline } = this;
                events.emit('change', isOnline);
                if (!isOnline) {
                    this.showOfflineToast();
                }
                else {
                    this._removeOfflineToast?.();
                }
            }),
            focusManager.subscribe(() => {
                const { isOnline } = this;
                if (focusManager.isFocused() && isOnline && this._removeOfflineToast) {
                    this._removeOfflineToast();
                }
            }),
        ];
        return () => {
            disposers.forEach((d) => d());
            if (this._removeOfflineToast) {
                this._removeOfflineToast();
            }
            if (this._removeOfflineToastTimeoutId != null) {
                clearTimeout(this._removeOfflineToastTimeoutId);
            }
        };
    }
}
