// @wix/bex-core/dist/esm/state/PageState.js
import { reactive } from '@wix/bex-utils';
import { action, makeObservable, observable } from 'mobx';
import { addResizeObserver } from './resizeObserverCompat';
export const MOBILE_BREAKPOINT = 768;
export class PageState {
    constructor() {
        this._sidePanelOpen = reactive(false);
        this._isMobile = false;
        this.modalsContainerRef = makeObservable({ current: null }, { current: observable.ref });
        this.stackingRootRef = { current: null };
        this.scrollableContentRef = { current: null };
        makeObservable(this, {
            _isMobile: observable.ref,
            _updateIsMobile: action.bound,
        });
        this._updateIsMobile();
    }
    _updateIsMobile() {
        if (typeof window !== 'undefined') {
            this._isMobile = window.innerWidth <= MOBILE_BREAKPOINT;
        }
    }
    get isMobile() {
        return this._isMobile;
    }
    set sidePanelOpen(value) {
        this._sidePanelOpen.value = value;
    }
    init() {
        if (typeof window === 'undefined') {
            return;
        }
        this._updateIsMobile();
        const container = this.stackingRootRef.current ?? document.body;
        const disposers = [addResizeObserver(container, this._updateIsMobile)];
        return () => {
            disposers.forEach((d) => d?.());
        };
    }
}
