// @wix/bex-core/dist/esm/state/PromisesSequence.js
import { EventEmitter } from 'events';
export class PromisesSequence {
    constructor() {
        this.fns = [];
        this.events = new EventEmitter();
        this.state = {
            running: false,
        };
    }
    async enqueue(fn) {
        const { fns, events, state } = this;
        fns.push(fn);
        if (state.running) {
            return new Promise((resolve) => {
                events.once('success', resolve);
            });
        }
        events.emit('start');
        state.running = true;
        let next;
        try {
            while (fns.length) {
                next = fns.shift();
                await next?.();
            }
            events.emit('success');
            events.emit('done');
            state.running = false;
        }
        catch (e) {
            if (next) {
                fns.unshift(next);
            }
            events.emit('done');
            state.running = false;
            throw e;
        }
    }
    clear() {
        this.fns.splice(0);
    }
}
