// @wix/bex-core/dist/esm/state/PromiseState.js
import { queryStatus } from '../model';
import { action, makeObservable, observable, runInAction } from 'mobx';
import once from 'lodash/once';
export class PromiseState {
    constructor(params) {
        this.status = {
            ...queryStatus.idle,
            error: null,
            isFetching: false,
        };
        this.fetchOnce = once(() => this.fetch());
        this.createPromise = params.createPromise;
        makeObservable(this, {
            status: observable.ref,
            fetch: action,
        });
    }
    async fetch() {
        try {
            this.status = {
                ...queryStatus.loading,
                error: null,
                isFetching: true,
            };
            const data = await this.createPromise();
            runInAction(() => {
                this.status = {
                    ...queryStatus.success,
                    error: null,
                    isFetching: false,
                    data,
                };
            });
        }
        catch (error) {
            runInAction(() => {
                this.status = {
                    ...queryStatus.error,
                    error,
                    isFetching: false,
                };
            });
        }
    }
}
