// @wix/bex-core/dist/esm/state/SelectionStatusWixPatternsPreset.js
class SelectionStatusWixPatternsPreset {
    constructor() {
        this.unchecked = {
            id: 'unchecked',
            toggle: () => this.checked,
            start: () => this.unchecked,
            final: () => this.checked,
            isChecked: false,
            isIndeterminate: false,
        };
        this.checked = {
            id: 'checked',
            toggle: () => this.unchecked,
            start: () => this.unchecked,
            final: () => this.checked,
            isChecked: true,
            isIndeterminate: false,
        };
        this.indeterminateUnchecked = {
            id: 'indeterminateUnchecked',
            toggle: () => this.indeterminate,
            start: () => this.indeterminateUnchecked,
            final: () => this.indeterminateChecked,
            isChecked: false,
            isIndeterminate: false,
        };
        this.indeterminate = {
            id: 'indeterminate',
            toggle: () => this.indeterminateChecked,
            start: () => this.indeterminateUnchecked,
            final: () => this.indeterminateChecked,
            isChecked: true,
            isIndeterminate: true,
        };
        this.indeterminateChecked = {
            id: 'indeterminateChecked',
            toggle: () => this.indeterminateUnchecked,
            start: () => this.indeterminateUnchecked,
            final: () => this.indeterminateChecked,
            isChecked: true,
            isIndeterminate: false,
        };
    }
}
export const selectionStatusWixPatternsPreset = new SelectionStatusWixPatternsPreset();
