// @wix/bex-core/dist/esm/state/TaskState.js
import { queryStatus } from '../model';
import { action, makeObservable, observable, runInAction } from 'mobx';
export class TaskState {
    constructor() {
        this.status = {
            ...queryStatus.idle,
        };
        makeObservable(this, {
            status: observable.ref,
            run: action,
            runOnce: action,
            reset: action,
        });
    }
    runOnce(fn, onError) {
        if (fn) {
            this._fn = fn;
        }
        if (onError) {
            this._onError = onError;
        }
        if (!this._fn) {
            return;
        }
        const { status } = this;
        if (status.isLoading || status.isSuccess) {
            return;
        }
        return this.run(this._fn, this._onError);
    }
    get errorStatus() {
        const { status } = this;
        return status.isError ? status : undefined;
    }
    reset() {
        this.status = {
            ...queryStatus.idle,
        };
    }
    async run(fn, onError) {
        const promise = fn();
        this.status = {
            ...queryStatus.loading,
            promise,
        };
        try {
            const data = await promise;
            runInAction(() => {
                if (promise === this.status.promise) {
                    this.status = {
                        ...queryStatus.success,
                        error: null,
                        data,
                    };
                }
            });
        }
        catch (err) {
            onError?.(err);
            const error = err;
            runInAction(() => {
                if (promise === this.status.promise) {
                    this.status = {
                        ...queryStatus.error,
                        error,
                    };
                }
            });
        }
    }
}
