// @wix/bex-core/dist/esm/state/TextOverflowState.js
import { action, makeObservable, observable, reaction } from 'mobx';
import { addResizeObserver } from './resizeObserverCompat';
export class TextOverflowState {
    constructor(params) {
        this.node = null;
        this.isOverflow = false;
        this._updateIsOverflow = action(() => {
            const { node } = this;
            if (!node) {
                return;
            }
            this.isOverflow = node.clientWidth + 1 < node.scrollWidth;
        });
        this.observable = params.observable;
        makeObservable(this, {
            isOverflow: observable.ref,
            calculateOverflow: action.bound,
            init: action,
        });
    }
    init(node) {
        this.node = node;
        const disposers = [
            reaction(this.observable, () => {
                this.calculateOverflow();
            }, {
                fireImmediately: true,
            }),
            addResizeObserver(this.node, this.calculateOverflow),
        ];
        return () => {
            disposers.forEach((disposer) => disposer?.());
        };
    }
    calculateOverflow() {
        requestAnimationFrame(this._updateIsOverflow);
    }
}
