// @wix/bex-core/dist/esm/state/ToggleState.js
import { action, computed, makeObservable, observable } from 'mobx';
import { EventEmitter } from 'events';
export class ToggleState {
    constructor(initialValue = false) {
        this.events = new EventEmitter();
        this.value = initialValue;
        makeObservable(this, {
            value: observable,
            isOn: computed,
            isOff: computed,
            toggle: action.bound,
            setOff: action.bound,
            setOn: action.bound,
        });
    }
    get isOn() {
        return this.value;
    }
    get isOff() {
        return !this.value;
    }
    toggle(value = !this.value) {
        if (this.value !== value) {
            this.value = value;
            this.events.emit('toggle');
        }
    }
    setOff() {
        this.toggle(false);
        // return new Promise<void>((resolve) => {
        //   this.events.once('afterClose', resolve);
        // });
    }
    setOn() {
        this.toggle(true);
        this.events.emit('open');
    }
}
