// @wix/patterns/dist/cjs/components/CustomFieldFilters/buildCustomFieldFilterElement.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.buildCustomFieldFilterElement = void 0;
var _react = _interopRequireDefault(require("react"));
var _DateCustomFieldFilter = require("./DateCustomFieldFilter");
var _DateTimeCustomFieldFilter = require("./DateTimeCustomFieldFilter");
var _CheckboxCustomFieldFilter = require("./CheckboxCustomFieldFilter");
var _NumberRangeFilter = require("../NumberRangeFilter");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CustomFieldFilters/buildCustomFieldFilterElement.tsx";
const buildCustomFieldFilterElement = (state, customField) => {
  const customFieldFilterComponents = {
    date: _DateCustomFieldFilter.DateCustomFieldFilter,
    dateTime: _DateTimeCustomFieldFilter.DateTimeCustomFieldFilter,
    checkbox: _CheckboxCustomFieldFilter.CheckboxCustomFieldFilter,
    integer: _NumberRangeFilter.NumberRangeFilter,
    decimal: ({
      filter
    }) => /*#__PURE__*/_react.default.createElement(_NumberRangeFilter.NumberRangeFilter, {
      filter: filter,
      decimal: true,
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 21,
        columnNumber: 30
      }
    })
  };
  const CustomFieldFilterComponent = customFieldFilterComponents[customField.type];
  if (!CustomFieldFilterComponent) {
    return null;
  }
  const label = customField.label || customField.id;
  return /*#__PURE__*/_react.default.createElement(CustomFieldFilterComponent, {
    key: customField.id,
    filter: state.collection.filters[customField.id],
    accordionItemProps: {
      label
    },
    toolbarItemProps: {
      label
    },
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 34,
      columnNumber: 5
    }
  });
};
exports.buildCustomFieldFilterElement = buildCustomFieldFilterElement;
