// @wix/patterns/dist/cjs/components/CustomFieldFilters/CheckboxCustomFieldFilter.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.CheckboxCustomFieldFilter = void 0;
var _react = _interopRequireDefault(require("react"));
var _RadioGroupFilter = require("../RadioGroupFilter");
var _react2 = require("@wix/bex-core/react");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CustomFieldFilters/CheckboxCustomFieldFilter.tsx";
const CheckboxCustomFieldFilter = ({
  filter
}) => {
  const {
    translate: t
  } = (0, _react2.useWixPatternsContainer)();
  return /*#__PURE__*/_react.default.createElement(_RadioGroupFilter.RadioGroupFilter, {
    filter: filter,
    data: [{
      id: 'checked',
      name: t('cairo.customFields.fieldType.checkbox.Selected.checked'),
      value: true
    }, {
      id: 'unchecked',
      name: t('cairo.customFields.fieldType.checkbox.Selected.notChecked'),
      value: false
    }],
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 14,
      columnNumber: 5
    }
  });
};
exports.CheckboxCustomFieldFilter = CheckboxCustomFieldFilter;
