// @wix/patterns/dist/cjs/components/CustomFieldFilters/DateCustomFieldFilterBase.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.DateCustomFieldFilterBase = exports.DATE_CUSTOM_FIELD_OPTIONS = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireDefault(require("react"));
var _DateRangeFilter = require("../DateRangeFilter");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CustomFieldFilters/DateCustomFieldFilterBase.tsx";
const DATE_CUSTOM_FIELD_OPTIONS = exports.DATE_CUSTOM_FIELD_OPTIONS = ['SEVEN_DAYS', 'FOURTEEN_DAYS', 'MONTH', 'TODAY', 'NEXT_SEVEN_DAYS', 'NEXT_THIRTY_DAYS'];
const DateCustomFieldFilterBase = props => {
  return /*#__PURE__*/_react.default.createElement(_DateRangeFilter.DateRangeFilter, (0, _extends2.default)({
    dateRangeOptions: DATE_CUSTOM_FIELD_OPTIONS,
    popoverProps: {
      appendTo: 'parent'
    }
  }, props, {
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 15,
      columnNumber: 5
    }
  }));
};
exports.DateCustomFieldFilterBase = DateCustomFieldFilterBase;
