// @wix/patterns/dist/cjs/components/CustomFieldFilters/DateTimeCustomFieldFilter.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.DateTimeCustomFieldFilter = void 0;
var _react = _interopRequireDefault(require("react"));
var _DateRangeFilter = require("../DateRangeFilter");
var _DateCustomFieldFilterBase = require("./DateCustomFieldFilterBase");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CustomFieldFilters/DateTimeCustomFieldFilter.tsx";
const DateTimeCustomFieldFilter = ({
  filter
}) => {
  return /*#__PURE__*/_react.default.createElement(_DateCustomFieldFilterBase.DateCustomFieldFilterBase, {
    filter: filter,
    filterByTime: true,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 12,
      columnNumber: 10
    }
  });
};
exports.DateTimeCustomFieldFilter = DateTimeCustomFieldFilter;
DateTimeCustomFieldFilter.renderPresetToolbarTag = _DateRangeFilter.DateRangeFilter.renderPresetToolbarTag;
