// @wix/bex-core/dist/cjs/state/CollectionPageState/CollectionPageStateBIReporter.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.CollectionPageStateBIReporter = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _bi = require("../../exports/bi");
var _util = require("../../util");
class CollectionPageStateBIReporter {
  constructor(params) {
    (0, _defineProperty2.default)(this, "collectionPageState", void 0);
    (0, _defineProperty2.default)(this, "reportBI", void 0);
    (0, _defineProperty2.default)(this, "appLoaded", void 0);
    this.collectionPageState = params.collectionPageState;
    this.reportBI = params.reportBI;
  }
  init() {
    const startTime = performance.now();
    this.appLoaded = this.reportOnLoadStart(startTime);
  }
  reportOnLoadStart(startTime) {
    this.reportBI((0, _util.withoutDefaults)(_bi.loadStart)({
      ...this._commonDynamicBiParams()
    }));
    return () => {
      this.reportOnLoadEnd(Math.round(performance.now() - startTime));
    };
  }
  reportOnLoadEnd(endTime) {
    this.reportBI((0, _util.withoutDefaults)(_bi.loadEnd)({
      ...this._commonDynamicBiParams(),
      loadingTime: endTime
    }));
  }
  _commonDynamicBiParams() {
    return {
      url: window.location.href,
      route: window.location.pathname,
      routerUsage: this.collectionPageState.withinRouter
    };
  }
}
exports.CollectionPageStateBIReporter = CollectionPageStateBIReporter;
