// @wix/bex-core/dist/cjs/state/CollectionPageState/CollectionPageState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.CollectionPageState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _mobx = require("mobx");
var _PageState = require("../PageState");
var _resizeObserverCompat = require("../resizeObserverCompat");
var _CollectionPageStateBIReporter = require("./CollectionPageStateBIReporter");
class CollectionPageState {
  constructor(params) {
    (0, _defineProperty2.default)(this, "pageTitle", void 0);
    (0, _defineProperty2.default)(this, "hasHeader", false);
    (0, _defineProperty2.default)(this, "total", false);
    (0, _defineProperty2.default)(this, "hideHeaderTotal", false);
    (0, _defineProperty2.default)(this, "scrollTopBeforeUnmount", null);
    (0, _defineProperty2.default)(this, "withinRouter", void 0);
    (0, _defineProperty2.default)(this, "startTime", performance.now());
    (0, _defineProperty2.default)(this, "bi", void 0);
    (0, _defineProperty2.default)(this, "pageState", new _PageState.PageState());
    this.pageTitle = params.pageTitle;
    this.hasHeader = params.hasHeader;
    this.hideHeaderTotal = params.hideHeaderTotal;
    this.withinRouter = params.withinRouter;
    if (params.reportBI) {
      this.bi = new _CollectionPageStateBIReporter.CollectionPageStateBIReporter({
        reportBI: params.reportBI,
        collectionPageState: this
      });
    }
    (0, _mobx.makeObservable)(this, {
      pageTitle: _mobx.observable.ref,
      hasHeader: _mobx.observable.ref,
      hideHeaderTotal: _mobx.observable.ref,
      total: _mobx.observable.ref
    });
  }
  init() {
    var _this$bi, _this$bi2;
    (_this$bi = this.bi) == null || _this$bi.init();
    const scrollableContent = this.pageState.scrollableContentRef.current;
    if (!scrollableContent) {
      return () => {};
    }
    const disposer = (0, _resizeObserverCompat.addResizeObserver)(scrollableContent, () => {
      if (this.scrollTopBeforeUnmount === null) {
        return;
      }
      if (scrollableContent.scrollHeight >= this.scrollTopBeforeUnmount) {
        scrollableContent.scrollTo({
          top: this.scrollTopBeforeUnmount
        });
        this.scrollTopBeforeUnmount = null;
      }
    });
    (_this$bi2 = this.bi) == null || _this$bi2.appLoaded == null || _this$bi2.appLoaded();
    return () => {
      disposer == null || disposer();
    };
  }
}
exports.CollectionPageState = CollectionPageState;
