// @wix/design-system/dist/cjs/utils/test-utils/visual/addStories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.addStories = void 0;
var _react = require("@storybook/react");
var _react2 = _interopRequireWildcard(require("react"));
var _reactFrameComponent = _interopRequireWildcard(require("react-frame-component"));
var _WixDesignSystemProvider = _interopRequireDefault(require("../../../WixDesignSystemProvider"));
var _jsxFileName = "/home/builduser/work/57e038ea7326c1ec/packages/wix-design-system/dist/cjs/utils/test-utils/visual/addStories.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function _interopRequireWildcard(e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (var _t in e) "default" !== _t && {}.hasOwnProperty.call(e, _t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, _t)) && (i.get || i.set) ? o(f, _t, i) : f[_t] = e[_t]); return f; })(e, t); }
var CustomIframe = _ref => {
  var {
    children
  } = _ref;
  var [bodyClassName, setBodyClassName] = _react2.default.useState(document.body.className);

  // copy everything in the outer document's head into the inner document to set the styles
  // https://github.com/ryanseddon/react-frame-component/issues/37#issuecomment-215271946
  var initialContent = "<!DOCTYPE html><html><head>".concat(document.head.innerHTML, "</head><body><div></div></body></html>");
  (0, _react2.useEffect)(() => {
    setTimeout(() => {
      // because WDS provider classes are immediately available
      setBodyClassName(document.body.className);
    }, 100);
  }, []);
  return /*#__PURE__*/_react2.default.createElement(_reactFrameComponent.default, {
    width: 480,
    height: 1200,
    initialContent: initialContent,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 23,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react2.default.createElement(_reactFrameComponent.FrameContextConsumer, {
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 24,
      columnNumber: 7
    }
  }, _ref2 => {
    var {
      document
    } = _ref2;
    if (document) {
      document.body.className = bodyClassName;
    }
    return children;
  }));
};
var addStories = _ref3 => {
  var {
    title,
    describe,
    it,
    render,
    isMobile
  } = _ref3;
  var story = (0, _react.storiesOf)("".concat(title).concat(describe ? '/' + describe : ''), module);
  story.add(it, () => render());
  if (isMobile) {
    story.add("".concat(it, "-screen-small"), () => /*#__PURE__*/_react2.default.createElement(_react2.default.Fragment, null, /*#__PURE__*/_react2.default.createElement(CustomIframe, {
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 57,
        columnNumber: 9
      }
    }, render()), /*#__PURE__*/_react2.default.createElement(CustomIframe, {
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 58,
        columnNumber: 9
      }
    }, /*#__PURE__*/_react2.default.createElement(_WixDesignSystemProvider.default, {
      mobile: true,
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 59,
        columnNumber: 11
      }
    }, render()))));
  }
};
exports.addStories = addStories;
