// @wix/auto_sdk_events_orders/build/internal/es/index.mjs
// src/events-v1-order-orders.public.ts
import { renameKeysFromRESTResponseToSDKResponse as renameKeysFromRESTResponseToSDKResponse2 } from "@wix/sdk-runtime/rename-all-nested-keys";
import { transformRESTAddressToSDKAddress as transformRESTAddressToSDKAddress2 } from "@wix/sdk-runtime/transformations/address";
import { transformRESTFloatToSDKFloat as transformRESTFloatToSDKFloat2 } from "@wix/sdk-runtime/transformations/float";
import { transformRESTTimestampToSDKTimestamp as transformRESTTimestampToSDKTimestamp2 } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths as transformPaths3 } from "@wix/sdk-runtime/transformations/transform-paths";
import { EventDefinition } from "@wix/sdk-types";

// src/events-v1-order-orders.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/events-v1-order-orders.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKFloatToRESTFloat } from "@wix/sdk-runtime/transformations/float";
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixEventsTicketingCheckoutServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/events/v2/ticket-definitions",
        destPath: "/v2/ticket-definitions"
      },
      {
        srcPath: "/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v2/ticket-definitions",
        destPath: "/v2/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/events-ticketing-v1-proxy",
        destPath: "/api"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/events/v2/ticket-definitions",
        destPath: "/v2/ticket-definitions"
      },
      {
        srcPath: "/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    _: [
      {
        srcPath: "/_api/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v2/ticket-definition",
        destPath: "/v2/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/events-ticketing-v1-proxy/v1",
        destPath: "/v1"
      },
      {
        srcPath: "/_api/events-ticketing-v1-proxy/v2",
        destPath: "/v2"
      },
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v2/ticket-definitions",
        destPath: "/v2/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v2/ticket-definitions",
        destPath: "/v2/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v2/ticket-definitions",
        destPath: "/v2/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    "api._base_domain_": [
      {
        srcPath: "/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    "editor-flow.wixapps.net": [
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
function resolveWixEventsTicketingOrderManagementUrl(opts) {
  const domainToMappings = {
    "events._base_domain_": [
      {
        srcPath: "",
        destPath: ""
      },
      {
        srcPath: "/_api/wix-events-web",
        destPath: "/api"
      }
    ],
    "www.wixevents.com": [
      {
        srcPath: "/_api/wix-events-web",
        destPath: "/api"
      },
      {
        srcPath: "/api",
        destPath: "/api"
      },
      {
        srcPath: "/viewEvent",
        destPath: "/viewEvent"
      },
      {
        srcPath: "/viewRsvpDashboard",
        destPath: "/viewRsvpDashboard"
      },
      {
        srcPath: "/viewOrderDashboard",
        destPath: "/viewOrderDashboard"
      },
      {
        srcPath: "/viewSummarySubscription",
        destPath: "/viewSummarySubscription"
      },
      {
        srcPath: "/check-in",
        destPath: "/check-in"
      },
      {
        srcPath: "/v2",
        destPath: "/v2"
      },
      {
        srcPath: "/export",
        destPath: "/export"
      },
      {
        srcPath: "/tickets",
        destPath: "/tickets"
      },
      {
        srcPath: "/payment-redirect",
        destPath: "/payment-redirect"
      },
      {
        srcPath: "/admin",
        destPath: "/admin"
      },
      {
        srcPath: "/ticket",
        destPath: "/ticket"
      },
      {
        srcPath: "/media",
        destPath: "/media"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/wix-events-web",
        destPath: ""
      }
    ],
    "events.wixapps.net": [
      {
        srcPath: "/_api/wix-events-web",
        destPath: "/api"
      },
      {
        srcPath: "/events",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/wix-events-web",
        destPath: "/api"
      },
      {
        srcPath: "/_api/wix-events-web/v2",
        destPath: "/v2"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/wix-events-web",
        destPath: "/api"
      },
      {
        srcPath: "/events",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/wix-events-web",
        destPath: "/api"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v2/events",
        destPath: "/v2/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/events",
        destPath: "/v1/bulk/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      },
      {
        srcPath: "/events/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/events/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/events/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/events/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/events/v1/bulk/events",
        destPath: "/v1/bulk/events"
      },
      {
        srcPath: "/events/v1/reports",
        destPath: "/v1/reports"
      },
      {
        srcPath: "/events/v1/tickets",
        destPath: "/v1/tickets"
      },
      {
        srcPath: "/events/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/events/v1/web",
        destPath: "/v1/web"
      },
      {
        srcPath: "/events/v2/tickets",
        destPath: "/v2/tickets"
      },
      {
        srcPath: "/events/v2/events",
        destPath: "/v2/events"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/events/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/events/v1/tickets",
        destPath: "/v1/tickets"
      },
      {
        srcPath: "/events/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/events/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/events/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/events/v2/events",
        destPath: "/v2/events"
      },
      {
        srcPath: "/events/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      }
    ],
    "api._base_domain_": [
      {
        srcPath: "/events/v2",
        destPath: "/v2"
      },
      {
        srcPath: "/events/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/events/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/events/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/events/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/events/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/events/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    _: [
      {
        srcPath: "/_api/wix-events-web/v2",
        destPath: "/v2"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/events/v2/events",
        destPath: "/v2/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/wix-events-web/v2",
        destPath: "/v2"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "editor-flow.wixapps.net": [
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_events_orders";
function listOrders(payload) {
  function __listOrders({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "GET",
      methodFqn: "wix.events.ticketing.OrderManagement.ListOrders",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingOrderManagementUrl({
        protoPath: "/v1/orders",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "orders.created" },
            { path: "orders.updated" },
            { path: "orders.checkoutForm.inputValues.dateTime" },
            { path: "orders.tickets.checkIn.created" },
            { path: "orders.tickets.guestDetails.form.inputValues.dateTime" },
            {
              path: "orders.paymentDetails.transaction.scheduledActionExecutionDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "orders.checkoutForm.inputValues.number" },
            {
              path: "orders.checkoutForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "orders.checkoutForm.inputValues.address.address.geocode.longitude"
            },
            { path: "orders.tickets.guestDetails.form.inputValues.number" },
            {
              path: "orders.tickets.guestDetails.form.inputValues.address.address.geocode.latitude"
            },
            {
              path: "orders.tickets.guestDetails.form.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listOrders;
}
function getOrder(payload) {
  function __getOrder({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "GET",
      methodFqn: "wix.events.ticketing.OrderManagement.GetOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingOrderManagementUrl({
        protoPath: "/v1/events/{eventId}/orders/{orderNumber}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.created" },
            { path: "order.updated" },
            { path: "order.checkoutForm.inputValues.dateTime" },
            { path: "order.tickets.checkIn.created" },
            { path: "order.tickets.guestDetails.form.inputValues.dateTime" },
            {
              path: "order.paymentDetails.transaction.scheduledActionExecutionDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "order.checkoutForm.inputValues.number" },
            {
              path: "order.checkoutForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "order.checkoutForm.inputValues.address.address.geocode.longitude"
            },
            { path: "order.tickets.guestDetails.form.inputValues.number" },
            {
              path: "order.tickets.guestDetails.form.inputValues.address.address.geocode.latitude"
            },
            {
              path: "order.tickets.guestDetails.form.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getOrder;
}
function updateOrder(payload) {
  function __updateOrder({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fields" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "expires" },
          { path: "checkoutForm.inputValues.dateTime" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "checkoutForm.inputValues.number" },
          { path: "checkoutForm.inputValues.address.address.geocode.latitude" },
          {
            path: "checkoutForm.inputValues.address.address.geocode.longitude"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "PATCH",
      methodFqn: "wix.events.ticketing.OrderManagement.UpdateOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingOrderManagementUrl({
        protoPath: "/v1/events/{eventId}/orders/{orderNumber}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.created" },
            { path: "order.updated" },
            { path: "order.checkoutForm.inputValues.dateTime" },
            { path: "order.tickets.checkIn.created" },
            { path: "order.tickets.guestDetails.form.inputValues.dateTime" },
            {
              path: "order.paymentDetails.transaction.scheduledActionExecutionDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "order.checkoutForm.inputValues.number" },
            {
              path: "order.checkoutForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "order.checkoutForm.inputValues.address.address.geocode.longitude"
            },
            { path: "order.tickets.guestDetails.form.inputValues.number" },
            {
              path: "order.tickets.guestDetails.form.inputValues.address.address.geocode.latitude"
            },
            {
              path: "order.tickets.guestDetails.form.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateOrder;
}
function bulkUpdateOrders(payload) {
  function __bulkUpdateOrders({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fields" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "PATCH",
      methodFqn: "wix.events.ticketing.OrderManagement.BulkUpdateOrders",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingOrderManagementUrl({
        protoPath: "/v1/events/{eventId}/orders",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "orders.created" },
            { path: "orders.updated" },
            { path: "orders.checkoutForm.inputValues.dateTime" },
            { path: "orders.tickets.checkIn.created" },
            { path: "orders.tickets.guestDetails.form.inputValues.dateTime" },
            {
              path: "orders.paymentDetails.transaction.scheduledActionExecutionDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "orders.checkoutForm.inputValues.number" },
            {
              path: "orders.checkoutForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "orders.checkoutForm.inputValues.address.address.geocode.longitude"
            },
            { path: "orders.tickets.guestDetails.form.inputValues.number" },
            {
              path: "orders.tickets.guestDetails.form.inputValues.address.address.geocode.latitude"
            },
            {
              path: "orders.tickets.guestDetails.form.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateOrders;
}
function confirmOrder(payload) {
  function __confirmOrder({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "POST",
      methodFqn: "wix.events.ticketing.OrderManagement.ConfirmOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingOrderManagementUrl({
        protoPath: "/v1/events/{eventId}/orders/confirm",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "orders.created" },
            { path: "orders.updated" },
            { path: "orders.checkoutForm.inputValues.dateTime" },
            { path: "orders.tickets.checkIn.created" },
            { path: "orders.tickets.guestDetails.form.inputValues.dateTime" },
            {
              path: "orders.paymentDetails.transaction.scheduledActionExecutionDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "orders.checkoutForm.inputValues.number" },
            {
              path: "orders.checkoutForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "orders.checkoutForm.inputValues.address.address.geocode.longitude"
            },
            { path: "orders.tickets.guestDetails.form.inputValues.number" },
            {
              path: "orders.tickets.guestDetails.form.inputValues.address.address.geocode.latitude"
            },
            {
              path: "orders.tickets.guestDetails.form.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __confirmOrder;
}
function getSummary(payload) {
  function __getSummary({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "GET",
      methodFqn: "wix.events.ticketing.OrderManagement.GetSummary",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingOrderManagementUrl({
        protoPath: "/v1/orders/summary",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getSummary;
}
function captureAuthorizedPayment(payload) {
  function __captureAuthorizedPayment({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "POST",
      methodFqn: "wix.events.ticketing.OrderManagement.CaptureAuthorizedPayment",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingOrderManagementUrl({
        protoPath: "/v1/events/orders/{orderNumber}/capture",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __captureAuthorizedPayment;
}
function voidAuthorizedPayment(payload) {
  function __voidAuthorizedPayment({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "POST",
      methodFqn: "wix.events.ticketing.OrderManagement.VoidAuthorizedPayment",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingOrderManagementUrl({
        protoPath: "/v1/events/orders/{orderNumber}/void",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __voidAuthorizedPayment;
}
function getCheckoutOptions(payload) {
  function __getCheckoutOptions({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "GET",
      methodFqn: "wix.events.ticketing.CheckoutService.GetCheckoutOptions",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingCheckoutServiceUrl({
        protoPath: "/v1/checkout/options",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getCheckoutOptions;
}
function listAvailableTickets(payload) {
  function __listAvailableTickets({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "overrideRequestTime" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "GET",
      methodFqn: "wix.events.ticketing.CheckoutService.ListAvailableTickets",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingCheckoutServiceUrl({
        protoPath: "/v1/checkout/available-tickets",
        data: serializedData,
        host
      }),
      params: toURLSearchParams(serializedData),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "definitions.salePeriod.startDate" },
            { path: "definitions.salePeriod.endDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listAvailableTickets;
}
function queryAvailableTickets(payload) {
  function __queryAvailableTickets({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "POST",
      methodFqn: "wix.events.ticketing.CheckoutService.QueryAvailableTickets",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingCheckoutServiceUrl({
        protoPath: "/v1/checkout/available-tickets/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "definitions.salePeriod.startDate" },
            { path: "definitions.salePeriod.endDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryAvailableTickets;
}
function createReservation(payload) {
  function __createReservation({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "overrideRequestTime" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "POST",
      methodFqn: "wix.events.ticketing.CheckoutService.CreateReservation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingCheckoutServiceUrl({
        protoPath: "/v1/checkout/reservations",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "expires" },
            { path: "reservations.ticket.salePeriod.startDate" },
            { path: "reservations.ticket.salePeriod.endDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createReservation;
}
function cancelReservation(payload) {
  function __cancelReservation({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "DELETE",
      methodFqn: "wix.events.ticketing.CheckoutService.CancelReservation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingCheckoutServiceUrl({
        protoPath: "/v1/checkout/reservations/{id}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __cancelReservation;
}
function getInvoice(payload) {
  function __getInvoice({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "POST",
      methodFqn: "wix.events.ticketing.CheckoutService.GetInvoice",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingCheckoutServiceUrl({
        protoPath: "/v1/checkout/invoices/{reservationId}",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "expires" },
            { path: "reservations.ticket.salePeriod.startDate" },
            { path: "reservations.ticket.salePeriod.endDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getInvoice;
}
function checkout(payload) {
  function __checkout({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "overrideRequestTime" },
          { path: "guests.form.inputValues.dateTime" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "guests.form.inputValues.number" },
          { path: "guests.form.inputValues.address.address.geocode.latitude" },
          { path: "guests.form.inputValues.address.address.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "POST",
      methodFqn: "wix.events.ticketing.CheckoutService.Checkout",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingCheckoutServiceUrl({
        protoPath: "/v1/checkout",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "expires" },
            { path: "order.created" },
            { path: "order.updated" },
            { path: "order.checkoutForm.inputValues.dateTime" },
            { path: "order.tickets.checkIn.created" },
            { path: "order.tickets.guestDetails.form.inputValues.dateTime" },
            {
              path: "order.paymentDetails.transaction.scheduledActionExecutionDate"
            },
            { path: "reservations.ticket.salePeriod.startDate" },
            { path: "reservations.ticket.salePeriod.endDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "order.checkoutForm.inputValues.number" },
            {
              path: "order.checkoutForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "order.checkoutForm.inputValues.address.address.geocode.longitude"
            },
            { path: "order.tickets.guestDetails.form.inputValues.number" },
            {
              path: "order.tickets.guestDetails.form.inputValues.address.address.geocode.latitude"
            },
            {
              path: "order.tickets.guestDetails.form.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __checkout;
}
function updateCheckout(payload) {
  function __updateCheckout({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "guests.form.inputValues.number" },
          { path: "guests.form.inputValues.address.address.geocode.latitude" },
          { path: "guests.form.inputValues.address.address.geocode.longitude" }
        ]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "guests.form.inputValues.dateTime" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "PATCH",
      methodFqn: "wix.events.ticketing.CheckoutService.UpdateCheckout",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingCheckoutServiceUrl({
        protoPath: "/v1/checkout/{orderNumber}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.created" },
            { path: "order.updated" },
            { path: "order.checkoutForm.inputValues.dateTime" },
            { path: "order.tickets.checkIn.created" },
            { path: "order.tickets.guestDetails.form.inputValues.dateTime" },
            {
              path: "order.paymentDetails.transaction.scheduledActionExecutionDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "order.checkoutForm.inputValues.number" },
            {
              path: "order.checkoutForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "order.checkoutForm.inputValues.address.address.geocode.longitude"
            },
            { path: "order.tickets.guestDetails.form.inputValues.number" },
            {
              path: "order.tickets.guestDetails.form.inputValues.address.address.geocode.latitude"
            },
            {
              path: "order.tickets.guestDetails.form.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateCheckout;
}
function posCheckout(payload) {
  function __posCheckout({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "POST",
      methodFqn: "wix.events.ticketing.CheckoutService.PosCheckout",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingCheckoutServiceUrl({
        protoPath: "/v1/pos-checkout",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "expires" },
            { path: "order.created" },
            { path: "order.updated" },
            { path: "order.checkoutForm.inputValues.dateTime" },
            { path: "order.tickets.checkIn.created" },
            { path: "order.tickets.guestDetails.form.inputValues.dateTime" },
            {
              path: "order.paymentDetails.transaction.scheduledActionExecutionDate"
            },
            { path: "reservations.ticket.salePeriod.startDate" },
            { path: "reservations.ticket.salePeriod.endDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "order.checkoutForm.inputValues.number" },
            {
              path: "order.checkoutForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "order.checkoutForm.inputValues.address.address.geocode.longitude"
            },
            { path: "order.tickets.guestDetails.form.inputValues.number" },
            {
              path: "order.tickets.guestDetails.form.inputValues.address.address.geocode.latitude"
            },
            {
              path: "order.tickets.guestDetails.form.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __posCheckout;
}

// src/events-v1-order-orders.universal.ts
import { transformSDKAddressToRESTAddress } from "@wix/sdk-runtime/transformations/address";
import { transformRESTAddressToSDKAddress } from "@wix/sdk-runtime/transformations/address";
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var SubdivisionType = /* @__PURE__ */ ((SubdivisionType2) => {
  SubdivisionType2["UNKNOWN_SUBDIVISION_TYPE"] = "UNKNOWN_SUBDIVISION_TYPE";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_1"] = "ADMINISTRATIVE_AREA_LEVEL_1";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_2"] = "ADMINISTRATIVE_AREA_LEVEL_2";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_3"] = "ADMINISTRATIVE_AREA_LEVEL_3";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_4"] = "ADMINISTRATIVE_AREA_LEVEL_4";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_5"] = "ADMINISTRATIVE_AREA_LEVEL_5";
  SubdivisionType2["COUNTRY"] = "COUNTRY";
  return SubdivisionType2;
})(SubdivisionType || {});
var OrderStatus = /* @__PURE__ */ ((OrderStatus2) => {
  OrderStatus2["NA_ORDER_STATUS"] = "NA_ORDER_STATUS";
  OrderStatus2["FREE"] = "FREE";
  OrderStatus2["PENDING"] = "PENDING";
  OrderStatus2["PAID"] = "PAID";
  OrderStatus2["OFFLINE_PENDING"] = "OFFLINE_PENDING";
  OrderStatus2["INITIATED"] = "INITIATED";
  OrderStatus2["CANCELED"] = "CANCELED";
  OrderStatus2["DECLINED"] = "DECLINED";
  OrderStatus2["AUTHORIZED"] = "AUTHORIZED";
  OrderStatus2["VOIDED"] = "VOIDED";
  OrderStatus2["PARTIALLY_PAID"] = "PARTIALLY_PAID";
  return OrderStatus2;
})(OrderStatus || {});
var ChannelType = /* @__PURE__ */ ((ChannelType2) => {
  ChannelType2["ONLINE"] = "ONLINE";
  ChannelType2["OFFLINE_POS"] = "OFFLINE_POS";
  return ChannelType2;
})(ChannelType || {});
var TaxType = /* @__PURE__ */ ((TaxType2) => {
  TaxType2["INCLUDED"] = "INCLUDED";
  TaxType2["ADDED"] = "ADDED";
  TaxType2["ADDED_AT_CHECKOUT"] = "ADDED_AT_CHECKOUT";
  return TaxType2;
})(TaxType || {});
var FeeName = /* @__PURE__ */ ((FeeName2) => {
  FeeName2["WIX_FEE"] = "WIX_FEE";
  return FeeName2;
})(FeeName || {});
var FeeType = /* @__PURE__ */ ((FeeType2) => {
  FeeType2["FEE_ADDED"] = "FEE_ADDED";
  FeeType2["FEE_INCLUDED"] = "FEE_INCLUDED";
  FeeType2["FEE_ADDED_AT_CHECKOUT"] = "FEE_ADDED_AT_CHECKOUT";
  return FeeType2;
})(FeeType || {});
var ScheduledActionEnumAction = /* @__PURE__ */ ((ScheduledActionEnumAction2) => {
  ScheduledActionEnumAction2["UNKNOWN_ACTION"] = "UNKNOWN_ACTION";
  ScheduledActionEnumAction2["CAPTURE"] = "CAPTURE";
  ScheduledActionEnumAction2["VOID"] = "VOID";
  return ScheduledActionEnumAction2;
})(ScheduledActionEnumAction || {});
var Action = /* @__PURE__ */ ((Action2) => {
  Action2["ARCHIVE"] = "ARCHIVE";
  Action2["UNARCHIVE"] = "UNARCHIVE";
  Action2["CONFIRM"] = "CONFIRM";
  Action2["CAPTURE"] = "CAPTURE";
  Action2["VOID"] = "VOID";
  return Action2;
})(Action || {});
var OrderType = /* @__PURE__ */ ((OrderType2) => {
  OrderType2["UNASSIGNED_TICKETS"] = "UNASSIGNED_TICKETS";
  OrderType2["ASSIGNED_TICKETS"] = "ASSIGNED_TICKETS";
  return OrderType2;
})(OrderType || {});
var OrderFieldset = /* @__PURE__ */ ((OrderFieldset2) => {
  OrderFieldset2["TICKETS"] = "TICKETS";
  OrderFieldset2["DETAILS"] = "DETAILS";
  OrderFieldset2["FORM"] = "FORM";
  OrderFieldset2["INVOICE"] = "INVOICE";
  return OrderFieldset2;
})(OrderFieldset || {});
var OrderTag = /* @__PURE__ */ ((OrderTag2) => {
  OrderTag2["CONFIRMED"] = "CONFIRMED";
  OrderTag2["UNCONFIRMED"] = "UNCONFIRMED";
  OrderTag2["MEMBER"] = "MEMBER";
  OrderTag2["ARCHIVED"] = "ARCHIVED";
  OrderTag2["NON_ARCHIVED"] = "NON_ARCHIVED";
  OrderTag2["FULLY_CHECKED_IN"] = "FULLY_CHECKED_IN";
  OrderTag2["NOT_FULLY_CHECKED_IN"] = "NOT_FULLY_CHECKED_IN";
  return OrderTag2;
})(OrderTag || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
var ReservationStatus = /* @__PURE__ */ ((ReservationStatus2) => {
  ReservationStatus2["RESERVATION_PENDING"] = "RESERVATION_PENDING";
  ReservationStatus2["RESERVATION_CONFIRMED"] = "RESERVATION_CONFIRMED";
  ReservationStatus2["RESERVATION_CANCELED"] = "RESERVATION_CANCELED";
  ReservationStatus2["RESERVATION_CANCELED_MANUALLY"] = "RESERVATION_CANCELED_MANUALLY";
  ReservationStatus2["RESERVATION_EXPIRED"] = "RESERVATION_EXPIRED";
  return ReservationStatus2;
})(ReservationStatus || {});
var State = /* @__PURE__ */ ((State2) => {
  State2["INCLUDE_HIDDEN_NOT_ON_SALE"] = "INCLUDE_HIDDEN_NOT_ON_SALE";
  return State2;
})(State || {});
var TicketSaleStatus = /* @__PURE__ */ ((TicketSaleStatus2) => {
  TicketSaleStatus2["SALE_SCHEDULED"] = "SALE_SCHEDULED";
  TicketSaleStatus2["SALE_STARTED"] = "SALE_STARTED";
  TicketSaleStatus2["SALE_ENDED"] = "SALE_ENDED";
  return TicketSaleStatus2;
})(TicketSaleStatus || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["STANDARD"] = "STANDARD";
  Type2["DONATION"] = "DONATION";
  return Type2;
})(Type || {});
var TicketDefinitionFieldset = /* @__PURE__ */ ((TicketDefinitionFieldset2) => {
  TicketDefinitionFieldset2["POLICY"] = "POLICY";
  TicketDefinitionFieldset2["DASHBOARD"] = "DASHBOARD";
  return TicketDefinitionFieldset2;
})(TicketDefinitionFieldset || {});
async function listOrders2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    offset: options?.offset,
    limit: options?.limit,
    fieldset: options?.fieldset,
    status: options?.status,
    eventId: options?.eventId,
    orderNumber: options?.orderNumber,
    memberId: options?.memberId,
    facet: options?.facet,
    searchPhrase: options?.searchPhrase,
    eventCreatorId: options?.eventCreatorId,
    sort: options?.sort,
    tag: options?.tag,
    contactId: options?.contactId
  });
  const reqOpts = listOrders(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTAddressToSDKAddress,
          paths: [
            { path: "orders.checkoutForm.inputValues.address.address" },
            {
              path: "orders.tickets.guestDetails.form.inputValues.address.address"
            }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          offset: "$[0].offset",
          limit: "$[0].limit",
          fieldset: "$[0].fieldset",
          status: "$[0].status",
          eventId: "$[0].eventId",
          orderNumber: "$[0].orderNumber",
          memberId: "$[0].memberId",
          facet: "$[0].facet",
          searchPhrase: "$[0].searchPhrase",
          eventCreatorId: "$[0].eventCreatorId",
          sort: "$[0].sort",
          tag: "$[0].tag",
          contactId: "$[0].contactId"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getOrder2(identifiers, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    eventId: identifiers?.eventId,
    orderNumber: identifiers?.orderNumber,
    fieldset: options?.fieldset
  });
  const reqOpts = getOrder(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTAddressToSDKAddress,
          paths: [
            { path: "order.checkoutForm.inputValues.address.address" },
            {
              path: "order.tickets.guestDetails.form.inputValues.address.address"
            }
          ]
        }
      ])
    )?.order;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          eventId: "$[0].eventId",
          orderNumber: "$[0].orderNumber",
          fieldset: "$[1].fieldset"
        },
        singleArgumentUnchanged: false
      },
      ["identifiers", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateOrder2(identifiers, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({
      eventId: identifiers?.eventId,
      orderNumber: identifiers?.orderNumber,
      fields: options?.fields,
      checkoutForm: options?.checkoutForm,
      archived: options?.archived
    }),
    [
      {
        transformFn: transformSDKAddressToRESTAddress,
        paths: [{ path: "checkoutForm.inputValues.address.address" }]
      }
    ]
  );
  const reqOpts = updateOrder(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTAddressToSDKAddress,
          paths: [
            { path: "order.checkoutForm.inputValues.address.address" },
            {
              path: "order.tickets.guestDetails.form.inputValues.address.address"
            }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          eventId: "$[0].eventId",
          orderNumber: "$[0].orderNumber",
          fields: "$[1].fields",
          checkoutForm: "$[1].checkoutForm",
          archived: "$[1].archived"
        },
        singleArgumentUnchanged: false
      },
      ["identifiers", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateOrders2(eventId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    eventId,
    orderNumber: options?.orderNumber,
    fields: options?.fields,
    archived: options?.archived
  });
  const reqOpts = bulkUpdateOrders(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTAddressToSDKAddress,
          paths: [
            { path: "orders.checkoutForm.inputValues.address.address" },
            {
              path: "orders.tickets.guestDetails.form.inputValues.address.address"
            }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          eventId: "$[0]",
          orderNumber: "$[1].orderNumber",
          fields: "$[1].fields",
          archived: "$[1].archived"
        },
        singleArgumentUnchanged: false
      },
      ["eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function confirmOrder2(eventId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    eventId,
    orderNumber: options?.orderNumber
  });
  const reqOpts = confirmOrder(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTAddressToSDKAddress,
          paths: [
            { path: "orders.checkoutForm.inputValues.address.address" },
            {
              path: "orders.tickets.guestDetails.form.inputValues.address.address"
            }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          eventId: "$[0]",
          orderNumber: "$[1].orderNumber"
        },
        singleArgumentUnchanged: false
      },
      ["eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getSummary2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    eventId: options?.eventId
  });
  const reqOpts = getSummary(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { eventId: "$[0].eventId" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function captureAuthorizedPayment2(orderNumber, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    orderNumber,
    eventId: options?.eventId
  });
  const reqOpts = captureAuthorizedPayment(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          orderNumber: "$[0]",
          eventId: "$[1].eventId"
        },
        singleArgumentUnchanged: false
      },
      ["orderNumber", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function voidAuthorizedPayment2(orderNumber, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    orderNumber,
    eventId: options?.eventId
  });
  const reqOpts = voidAuthorizedPayment(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          orderNumber: "$[0]",
          eventId: "$[1].eventId"
        },
        singleArgumentUnchanged: false
      },
      ["orderNumber", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getCheckoutOptions2() {
  const { httpClient, sideEffects } = arguments[0];
  const payload = renameKeysFromSDKRequestToRESTRequest({});
  const reqOpts = getCheckoutOptions(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {},
        singleArgumentUnchanged: false
      },
      []
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listAvailableTickets2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    eventId: options?.eventId,
    offset: options?.offset,
    limit: options?.limit,
    sort: options?.sort,
    state: options?.state
  });
  const reqOpts = listAvailableTickets(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          eventId: "$[0].eventId",
          offset: "$[0].offset",
          limit: "$[0].limit",
          sort: "$[0].sort",
          state: "$[0].state"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function queryAvailableTickets2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    offset: options?.offset,
    limit: options?.limit,
    filter: options?.filter,
    fieldset: options?.fieldset,
    sort: options?.sort
  });
  const reqOpts = queryAvailableTickets(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          offset: "$[0].offset",
          limit: "$[0].limit",
          filter: "$[0].filter",
          fieldset: "$[0].fieldset",
          sort: "$[0].sort"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function createReservation2(eventId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    eventId,
    ticketQuantities: options?.ticketQuantities,
    ignoreLimits: options?.ignoreLimits,
    allowHiddenTickets: options?.allowHiddenTickets
  });
  const reqOpts = createReservation(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          eventId: "$[0]",
          ticketQuantities: "$[1].ticketQuantities",
          ignoreLimits: "$[1].ignoreLimits",
          allowHiddenTickets: "$[1].allowHiddenTickets"
        },
        singleArgumentUnchanged: false
      },
      ["eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function cancelReservation2(_id, eventId) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    id: _id,
    eventId
  });
  const reqOpts = cancelReservation(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { id: "$[0]", eventId: "$[1]" },
        singleArgumentUnchanged: false
      },
      ["_id", "eventId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getInvoice2(reservationId, eventId, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    reservationId,
    eventId,
    withDiscount: options?.withDiscount,
    paidPlanBenefit: options?.paidPlanBenefit
  });
  const reqOpts = getInvoice(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          reservationId: "$[0]",
          eventId: "$[1]",
          withDiscount: "$[2].withDiscount",
          paidPlanBenefit: "$[2].paidPlanBenefit"
        },
        singleArgumentUnchanged: false
      },
      ["reservationId", "eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function checkout2(eventId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({
      eventId,
      reservationId: options?.reservationId,
      memberId: options?.memberId,
      discount: options?.discount,
      buyer: options?.buyer,
      guests: options?.guests,
      paidPlanBenefit: options?.paidPlanBenefit,
      options: options?.options,
      marketingConsent: options?.marketingConsent
    }),
    [
      {
        transformFn: transformSDKAddressToRESTAddress,
        paths: [{ path: "guests.form.inputValues.address.address" }]
      }
    ]
  );
  const reqOpts = checkout(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTAddressToSDKAddress,
          paths: [
            { path: "order.checkoutForm.inputValues.address.address" },
            {
              path: "order.tickets.guestDetails.form.inputValues.address.address"
            }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          eventId: "$[0]",
          reservationId: "$[1].reservationId",
          memberId: "$[1].memberId",
          discount: "$[1].discount",
          buyer: "$[1].buyer",
          guests: "$[1].guests",
          paidPlanBenefit: "$[1].paidPlanBenefit",
          options: "$[1].options",
          marketingConsent: "$[1].marketingConsent"
        },
        singleArgumentUnchanged: false
      },
      ["eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateCheckout2(orderNumber, eventId, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({
      orderNumber,
      eventId,
      buyer: options?.buyer,
      guests: options?.guests,
      memberId: options?.memberId,
      discount: options?.discount,
      paidPlanBenefit: options?.paidPlanBenefit
    }),
    [
      {
        transformFn: transformSDKAddressToRESTAddress,
        paths: [{ path: "guests.form.inputValues.address.address" }]
      }
    ]
  );
  const reqOpts = updateCheckout(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTAddressToSDKAddress,
          paths: [
            { path: "order.checkoutForm.inputValues.address.address" },
            {
              path: "order.tickets.guestDetails.form.inputValues.address.address"
            }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          orderNumber: "$[0]",
          eventId: "$[1]",
          buyer: "$[2].buyer",
          guests: "$[2].guests",
          memberId: "$[2].memberId",
          discount: "$[2].discount",
          paidPlanBenefit: "$[2].paidPlanBenefit"
        },
        singleArgumentUnchanged: false
      },
      ["orderNumber", "eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function posCheckout2(eventId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    eventId,
    reservationId: options?.reservationId,
    paymentDetailsId: options?.paymentDetailsId
  });
  const reqOpts = posCheckout(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTAddressToSDKAddress,
          paths: [
            { path: "order.checkoutForm.inputValues.address.address" },
            {
              path: "order.tickets.guestDetails.form.inputValues.address.address"
            }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          eventId: "$[0]",
          reservationId: "$[1].reservationId",
          paymentDetailsId: "$[1].paymentDetailsId"
        },
        singleArgumentUnchanged: false
      },
      ["eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/events-v1-order-orders.public.ts
function listOrders3(httpClient) {
  return (options) => listOrders2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function getOrder3(httpClient) {
  return (identifiers, options) => getOrder2(
    identifiers,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function updateOrder3(httpClient) {
  return (identifiers, options) => updateOrder2(
    identifiers,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdateOrders3(httpClient) {
  return (eventId, options) => bulkUpdateOrders2(
    eventId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function confirmOrder3(httpClient) {
  return (eventId, options) => confirmOrder2(
    eventId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function getSummary3(httpClient) {
  return (options) => getSummary2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function captureAuthorizedPayment3(httpClient) {
  return (orderNumber, options) => captureAuthorizedPayment2(
    orderNumber,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function voidAuthorizedPayment3(httpClient) {
  return (orderNumber, options) => voidAuthorizedPayment2(
    orderNumber,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function getCheckoutOptions3(httpClient) {
  return () => getCheckoutOptions2(
    // @ts-ignore
    { httpClient }
  );
}
function listAvailableTickets3(httpClient) {
  return (options) => listAvailableTickets2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function queryAvailableTickets3(httpClient) {
  return (options) => queryAvailableTickets2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function createReservation3(httpClient) {
  return (eventId, options) => createReservation2(
    eventId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function cancelReservation3(httpClient) {
  return (_id, eventId) => cancelReservation2(
    _id,
    eventId,
    // @ts-ignore
    { httpClient }
  );
}
function getInvoice3(httpClient) {
  return (reservationId, eventId, options) => getInvoice2(
    reservationId,
    eventId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function checkout3(httpClient) {
  return (eventId, options) => checkout2(
    eventId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function updateCheckout3(httpClient) {
  return (orderNumber, eventId, options) => updateCheckout2(
    orderNumber,
    eventId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function posCheckout3(httpClient) {
  return (eventId, options) => posCheckout2(
    eventId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
var onOrderDeleted = EventDefinition(
  "wix.events.ticketing.events.OrderDeleted",
  false,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "data.timestamp" },
          { path: "data.created" },
          { path: "data.updated" },
          { path: "data.tickets.checkIn.created" },
          { path: "data.tickets.form.inputValues.dateTime" }
        ]
      },
      {
        transformFn: transformRESTFloatToSDKFloat2,
        paths: [{ path: "data.tickets.form.inputValues.number" }]
      },
      {
        transformFn: transformRESTAddressToSDKAddress2,
        paths: [{ path: "data.tickets.form.inputValues.address.address" }]
      }
    ])
  )
)();
var onOrderUpdated = EventDefinition(
  "wix.events.ticketing.events.OrderUpdated",
  false,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "data.timestamp" },
          { path: "data.created" },
          { path: "data.updated" },
          { path: "data.checkoutForm.inputValues.dateTime" },
          { path: "data.tickets.checkIn.created" },
          { path: "data.tickets.form.inputValues.dateTime" }
        ]
      },
      {
        transformFn: transformRESTFloatToSDKFloat2,
        paths: [
          { path: "data.checkoutForm.inputValues.number" },
          { path: "data.tickets.form.inputValues.number" }
        ]
      },
      {
        transformFn: transformRESTAddressToSDKAddress2,
        paths: [
          { path: "data.checkoutForm.inputValues.address.address" },
          { path: "data.tickets.form.inputValues.address.address" }
        ]
      }
    ])
  )
)();
var onOrderConfirmed = EventDefinition(
  "wix.events.ticketing.events.OrderConfirmed",
  false,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "data.timestamp" },
          { path: "data.created" },
          { path: "data.checkoutForm.inputValues.dateTime" },
          { path: "data.tickets.checkIn.created" },
          { path: "data.tickets.form.inputValues.dateTime" }
        ]
      },
      {
        transformFn: transformRESTFloatToSDKFloat2,
        paths: [
          { path: "data.checkoutForm.inputValues.number" },
          { path: "data.tickets.form.inputValues.number" }
        ]
      },
      {
        transformFn: transformRESTAddressToSDKAddress2,
        paths: [
          { path: "data.checkoutForm.inputValues.address.address" },
          { path: "data.tickets.form.inputValues.address.address" }
        ]
      }
    ])
  )
)();
var onOrderInitiated = EventDefinition(
  "wix.events.ticketing.events.OrderInitiated",
  false,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "data.timestamp" },
          { path: "data.created" },
          { path: "data.updated" },
          { path: "data.checkoutForm.inputValues.dateTime" }
        ]
      },
      {
        transformFn: transformRESTFloatToSDKFloat2,
        paths: [{ path: "data.checkoutForm.inputValues.number" }]
      },
      {
        transformFn: transformRESTAddressToSDKAddress2,
        paths: [{ path: "data.checkoutForm.inputValues.address.address" }]
      }
    ])
  )
)();
var onOrderReservationCreated = EventDefinition(
  "wix.events.ticketing.events.ReservationCreated",
  false,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "data.timestamp" },
          { path: "data.expires" },
          { path: "data.updatedDate" },
          { path: "data.quantities.updatedDate" },
          { path: "data.counts.timestamp" }
        ]
      }
    ])
  )
)();
var onOrderReservationUpdated = EventDefinition(
  "wix.events.ticketing.events.ReservationUpdated",
  false,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "data.timestamp" },
          { path: "data.expires" },
          { path: "data.updatedDate" },
          { path: "data.quantities.updatedDate" },
          { path: "data.counts.timestamp" }
        ]
      }
    ])
  )
)();

// src/events-v1-order-orders.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
import { createEventModule } from "@wix/sdk-runtime/event-definition-modules";
var listOrders4 = /* @__PURE__ */ createRESTModule(listOrders3);
var getOrder4 = /* @__PURE__ */ createRESTModule(getOrder3);
var updateOrder4 = /* @__PURE__ */ createRESTModule(updateOrder3);
var bulkUpdateOrders4 = /* @__PURE__ */ createRESTModule(bulkUpdateOrders3);
var confirmOrder4 = /* @__PURE__ */ createRESTModule(confirmOrder3);
var getSummary4 = /* @__PURE__ */ createRESTModule(getSummary3);
var captureAuthorizedPayment4 = /* @__PURE__ */ createRESTModule(captureAuthorizedPayment3);
var voidAuthorizedPayment4 = /* @__PURE__ */ createRESTModule(voidAuthorizedPayment3);
var getCheckoutOptions4 = /* @__PURE__ */ createRESTModule(getCheckoutOptions3);
var listAvailableTickets4 = /* @__PURE__ */ createRESTModule(listAvailableTickets3);
var queryAvailableTickets4 = /* @__PURE__ */ createRESTModule(queryAvailableTickets3);
var createReservation4 = /* @__PURE__ */ createRESTModule(createReservation3);
var cancelReservation4 = /* @__PURE__ */ createRESTModule(cancelReservation3);
var getInvoice4 = /* @__PURE__ */ createRESTModule(getInvoice3);
var checkout4 = /* @__PURE__ */ createRESTModule(checkout3);
var updateCheckout4 = /* @__PURE__ */ createRESTModule(updateCheckout3);
var posCheckout4 = /* @__PURE__ */ createRESTModule(posCheckout3);
var onOrderDeleted2 = createEventModule(onOrderDeleted);
var onOrderUpdated2 = createEventModule(onOrderUpdated);
var onOrderConfirmed2 = createEventModule(onOrderConfirmed);
var onOrderInitiated2 = createEventModule(onOrderInitiated);
var onOrderReservationCreated2 = createEventModule(
  onOrderReservationCreated
);
var onOrderReservationUpdated2 = createEventModule(
  onOrderReservationUpdated
);
export {
  Action,
  ChannelType,
  FeeName,
  FeeType,
  OrderFieldset,
  OrderStatus,
  OrderTag,
  OrderType,
  ReservationStatus,
  ScheduledActionEnumAction,
  State,
  SubdivisionType,
  TaxType,
  TicketDefinitionFieldset,
  TicketSaleStatus,
  Type,
  WebhookIdentityType,
  bulkUpdateOrders4 as bulkUpdateOrders,
  cancelReservation4 as cancelReservation,
  captureAuthorizedPayment4 as captureAuthorizedPayment,
  checkout4 as checkout,
  confirmOrder4 as confirmOrder,
  createReservation4 as createReservation,
  getCheckoutOptions4 as getCheckoutOptions,
  getInvoice4 as getInvoice,
  getOrder4 as getOrder,
  getSummary4 as getSummary,
  listAvailableTickets4 as listAvailableTickets,
  listOrders4 as listOrders,
  onOrderConfirmed2 as onOrderConfirmed,
  onOrderDeleted2 as onOrderDeleted,
  onOrderInitiated2 as onOrderInitiated,
  onOrderReservationCreated2 as onOrderReservationCreated,
  onOrderReservationUpdated2 as onOrderReservationUpdated,
  onOrderUpdated2 as onOrderUpdated,
  posCheckout4 as posCheckout,
  queryAvailableTickets4 as queryAvailableTickets,
  updateCheckout4 as updateCheckout,
  updateOrder4 as updateOrder,
  voidAuthorizedPayment4 as voidAuthorizedPayment
};
