// @wix/auto_sdk_events_orders/build/internal/es/meta.mjs
// src/events-v1-order-orders.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKFloatToRESTFloat } from "@wix/sdk-runtime/transformations/float";
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixEventsTicketingCheckoutServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/events/v2/ticket-definitions",
        destPath: "/v2/ticket-definitions"
      },
      {
        srcPath: "/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v2/ticket-definitions",
        destPath: "/v2/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/events-ticketing-v1-proxy",
        destPath: "/api"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/events/v2/ticket-definitions",
        destPath: "/v2/ticket-definitions"
      },
      {
        srcPath: "/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    _: [
      {
        srcPath: "/_api/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v2/ticket-definition",
        destPath: "/v2/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/events-ticketing-v1-proxy/v1",
        destPath: "/v1"
      },
      {
        srcPath: "/_api/events-ticketing-v1-proxy/v2",
        destPath: "/v2"
      },
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v2/ticket-definitions",
        destPath: "/v2/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v2/ticket-definitions",
        destPath: "/v2/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v2/ticket-definitions",
        destPath: "/v2/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    "api._base_domain_": [
      {
        srcPath: "/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    "editor-flow.wixapps.net": [
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
function resolveWixEventsTicketingOrderManagementUrl(opts) {
  const domainToMappings = {
    "events._base_domain_": [
      {
        srcPath: "",
        destPath: ""
      },
      {
        srcPath: "/_api/wix-events-web",
        destPath: "/api"
      }
    ],
    "www.wixevents.com": [
      {
        srcPath: "/_api/wix-events-web",
        destPath: "/api"
      },
      {
        srcPath: "/api",
        destPath: "/api"
      },
      {
        srcPath: "/viewEvent",
        destPath: "/viewEvent"
      },
      {
        srcPath: "/viewRsvpDashboard",
        destPath: "/viewRsvpDashboard"
      },
      {
        srcPath: "/viewOrderDashboard",
        destPath: "/viewOrderDashboard"
      },
      {
        srcPath: "/viewSummarySubscription",
        destPath: "/viewSummarySubscription"
      },
      {
        srcPath: "/check-in",
        destPath: "/check-in"
      },
      {
        srcPath: "/v2",
        destPath: "/v2"
      },
      {
        srcPath: "/export",
        destPath: "/export"
      },
      {
        srcPath: "/tickets",
        destPath: "/tickets"
      },
      {
        srcPath: "/payment-redirect",
        destPath: "/payment-redirect"
      },
      {
        srcPath: "/admin",
        destPath: "/admin"
      },
      {
        srcPath: "/ticket",
        destPath: "/ticket"
      },
      {
        srcPath: "/media",
        destPath: "/media"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/wix-events-web",
        destPath: ""
      }
    ],
    "events.wixapps.net": [
      {
        srcPath: "/_api/wix-events-web",
        destPath: "/api"
      },
      {
        srcPath: "/events",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/wix-events-web",
        destPath: "/api"
      },
      {
        srcPath: "/_api/wix-events-web/v2",
        destPath: "/v2"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/wix-events-web",
        destPath: "/api"
      },
      {
        srcPath: "/events",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/wix-events-web",
        destPath: "/api"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v2/events",
        destPath: "/v2/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/events",
        destPath: "/v1/bulk/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      },
      {
        srcPath: "/events/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/events/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/events/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/events/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/events/v1/bulk/events",
        destPath: "/v1/bulk/events"
      },
      {
        srcPath: "/events/v1/reports",
        destPath: "/v1/reports"
      },
      {
        srcPath: "/events/v1/tickets",
        destPath: "/v1/tickets"
      },
      {
        srcPath: "/events/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/events/v1/web",
        destPath: "/v1/web"
      },
      {
        srcPath: "/events/v2/tickets",
        destPath: "/v2/tickets"
      },
      {
        srcPath: "/events/v2/events",
        destPath: "/v2/events"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/events/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/events/v1/tickets",
        destPath: "/v1/tickets"
      },
      {
        srcPath: "/events/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/events/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/events/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/events/v2/events",
        destPath: "/v2/events"
      },
      {
        srcPath: "/events/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      }
    ],
    "api._base_domain_": [
      {
        srcPath: "/events/v2",
        destPath: "/v2"
      },
      {
        srcPath: "/events/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/events/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/events/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/events/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/events/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/events/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    _: [
      {
        srcPath: "/_api/wix-events-web/v2",
        destPath: "/v2"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/events/v2/events",
        destPath: "/v2/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/wix-events-web/v2",
        destPath: "/v2"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "editor-flow.wixapps.net": [
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_events_orders";
function listOrders(payload) {
  function __listOrders({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "GET",
      methodFqn: "wix.events.ticketing.OrderManagement.ListOrders",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingOrderManagementUrl({
        protoPath: "/v1/orders",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "orders.created" },
            { path: "orders.updated" },
            { path: "orders.checkoutForm.inputValues.dateTime" },
            { path: "orders.tickets.checkIn.created" },
            { path: "orders.tickets.guestDetails.form.inputValues.dateTime" },
            {
              path: "orders.paymentDetails.transaction.scheduledActionExecutionDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "orders.checkoutForm.inputValues.number" },
            {
              path: "orders.checkoutForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "orders.checkoutForm.inputValues.address.address.geocode.longitude"
            },
            { path: "orders.tickets.guestDetails.form.inputValues.number" },
            {
              path: "orders.tickets.guestDetails.form.inputValues.address.address.geocode.latitude"
            },
            {
              path: "orders.tickets.guestDetails.form.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listOrders;
}
function getOrder(payload) {
  function __getOrder({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "GET",
      methodFqn: "wix.events.ticketing.OrderManagement.GetOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingOrderManagementUrl({
        protoPath: "/v1/events/{eventId}/orders/{orderNumber}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.created" },
            { path: "order.updated" },
            { path: "order.checkoutForm.inputValues.dateTime" },
            { path: "order.tickets.checkIn.created" },
            { path: "order.tickets.guestDetails.form.inputValues.dateTime" },
            {
              path: "order.paymentDetails.transaction.scheduledActionExecutionDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "order.checkoutForm.inputValues.number" },
            {
              path: "order.checkoutForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "order.checkoutForm.inputValues.address.address.geocode.longitude"
            },
            { path: "order.tickets.guestDetails.form.inputValues.number" },
            {
              path: "order.tickets.guestDetails.form.inputValues.address.address.geocode.latitude"
            },
            {
              path: "order.tickets.guestDetails.form.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getOrder;
}
function updateOrder(payload) {
  function __updateOrder({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fields" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "expires" },
          { path: "checkoutForm.inputValues.dateTime" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "checkoutForm.inputValues.number" },
          { path: "checkoutForm.inputValues.address.address.geocode.latitude" },
          {
            path: "checkoutForm.inputValues.address.address.geocode.longitude"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "PATCH",
      methodFqn: "wix.events.ticketing.OrderManagement.UpdateOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingOrderManagementUrl({
        protoPath: "/v1/events/{eventId}/orders/{orderNumber}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.created" },
            { path: "order.updated" },
            { path: "order.checkoutForm.inputValues.dateTime" },
            { path: "order.tickets.checkIn.created" },
            { path: "order.tickets.guestDetails.form.inputValues.dateTime" },
            {
              path: "order.paymentDetails.transaction.scheduledActionExecutionDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "order.checkoutForm.inputValues.number" },
            {
              path: "order.checkoutForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "order.checkoutForm.inputValues.address.address.geocode.longitude"
            },
            { path: "order.tickets.guestDetails.form.inputValues.number" },
            {
              path: "order.tickets.guestDetails.form.inputValues.address.address.geocode.latitude"
            },
            {
              path: "order.tickets.guestDetails.form.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateOrder;
}
function bulkUpdateOrders(payload) {
  function __bulkUpdateOrders({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fields" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "PATCH",
      methodFqn: "wix.events.ticketing.OrderManagement.BulkUpdateOrders",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingOrderManagementUrl({
        protoPath: "/v1/events/{eventId}/orders",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "orders.created" },
            { path: "orders.updated" },
            { path: "orders.checkoutForm.inputValues.dateTime" },
            { path: "orders.tickets.checkIn.created" },
            { path: "orders.tickets.guestDetails.form.inputValues.dateTime" },
            {
              path: "orders.paymentDetails.transaction.scheduledActionExecutionDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "orders.checkoutForm.inputValues.number" },
            {
              path: "orders.checkoutForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "orders.checkoutForm.inputValues.address.address.geocode.longitude"
            },
            { path: "orders.tickets.guestDetails.form.inputValues.number" },
            {
              path: "orders.tickets.guestDetails.form.inputValues.address.address.geocode.latitude"
            },
            {
              path: "orders.tickets.guestDetails.form.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateOrders;
}
function confirmOrder(payload) {
  function __confirmOrder({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "POST",
      methodFqn: "wix.events.ticketing.OrderManagement.ConfirmOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingOrderManagementUrl({
        protoPath: "/v1/events/{eventId}/orders/confirm",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "orders.created" },
            { path: "orders.updated" },
            { path: "orders.checkoutForm.inputValues.dateTime" },
            { path: "orders.tickets.checkIn.created" },
            { path: "orders.tickets.guestDetails.form.inputValues.dateTime" },
            {
              path: "orders.paymentDetails.transaction.scheduledActionExecutionDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "orders.checkoutForm.inputValues.number" },
            {
              path: "orders.checkoutForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "orders.checkoutForm.inputValues.address.address.geocode.longitude"
            },
            { path: "orders.tickets.guestDetails.form.inputValues.number" },
            {
              path: "orders.tickets.guestDetails.form.inputValues.address.address.geocode.latitude"
            },
            {
              path: "orders.tickets.guestDetails.form.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __confirmOrder;
}
function getSummary(payload) {
  function __getSummary({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "GET",
      methodFqn: "wix.events.ticketing.OrderManagement.GetSummary",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingOrderManagementUrl({
        protoPath: "/v1/orders/summary",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getSummary;
}
function captureAuthorizedPayment(payload) {
  function __captureAuthorizedPayment({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "POST",
      methodFqn: "wix.events.ticketing.OrderManagement.CaptureAuthorizedPayment",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingOrderManagementUrl({
        protoPath: "/v1/events/orders/{orderNumber}/capture",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __captureAuthorizedPayment;
}
function voidAuthorizedPayment(payload) {
  function __voidAuthorizedPayment({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "POST",
      methodFqn: "wix.events.ticketing.OrderManagement.VoidAuthorizedPayment",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingOrderManagementUrl({
        protoPath: "/v1/events/orders/{orderNumber}/void",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __voidAuthorizedPayment;
}
function getCheckoutOptions(payload) {
  function __getCheckoutOptions({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "GET",
      methodFqn: "wix.events.ticketing.CheckoutService.GetCheckoutOptions",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingCheckoutServiceUrl({
        protoPath: "/v1/checkout/options",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getCheckoutOptions;
}
function listAvailableTickets(payload) {
  function __listAvailableTickets({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "overrideRequestTime" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "GET",
      methodFqn: "wix.events.ticketing.CheckoutService.ListAvailableTickets",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingCheckoutServiceUrl({
        protoPath: "/v1/checkout/available-tickets",
        data: serializedData,
        host
      }),
      params: toURLSearchParams(serializedData),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "definitions.salePeriod.startDate" },
            { path: "definitions.salePeriod.endDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listAvailableTickets;
}
function queryAvailableTickets(payload) {
  function __queryAvailableTickets({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "POST",
      methodFqn: "wix.events.ticketing.CheckoutService.QueryAvailableTickets",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingCheckoutServiceUrl({
        protoPath: "/v1/checkout/available-tickets/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "definitions.salePeriod.startDate" },
            { path: "definitions.salePeriod.endDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryAvailableTickets;
}
function createReservation(payload) {
  function __createReservation({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "overrideRequestTime" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "POST",
      methodFqn: "wix.events.ticketing.CheckoutService.CreateReservation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingCheckoutServiceUrl({
        protoPath: "/v1/checkout/reservations",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "expires" },
            { path: "reservations.ticket.salePeriod.startDate" },
            { path: "reservations.ticket.salePeriod.endDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createReservation;
}
function cancelReservation(payload) {
  function __cancelReservation({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "DELETE",
      methodFqn: "wix.events.ticketing.CheckoutService.CancelReservation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingCheckoutServiceUrl({
        protoPath: "/v1/checkout/reservations/{id}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __cancelReservation;
}
function getInvoice(payload) {
  function __getInvoice({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "POST",
      methodFqn: "wix.events.ticketing.CheckoutService.GetInvoice",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingCheckoutServiceUrl({
        protoPath: "/v1/checkout/invoices/{reservationId}",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "expires" },
            { path: "reservations.ticket.salePeriod.startDate" },
            { path: "reservations.ticket.salePeriod.endDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getInvoice;
}
function checkout(payload) {
  function __checkout({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "overrideRequestTime" },
          { path: "guests.form.inputValues.dateTime" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "guests.form.inputValues.number" },
          { path: "guests.form.inputValues.address.address.geocode.latitude" },
          { path: "guests.form.inputValues.address.address.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "POST",
      methodFqn: "wix.events.ticketing.CheckoutService.Checkout",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingCheckoutServiceUrl({
        protoPath: "/v1/checkout",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "expires" },
            { path: "order.created" },
            { path: "order.updated" },
            { path: "order.checkoutForm.inputValues.dateTime" },
            { path: "order.tickets.checkIn.created" },
            { path: "order.tickets.guestDetails.form.inputValues.dateTime" },
            {
              path: "order.paymentDetails.transaction.scheduledActionExecutionDate"
            },
            { path: "reservations.ticket.salePeriod.startDate" },
            { path: "reservations.ticket.salePeriod.endDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "order.checkoutForm.inputValues.number" },
            {
              path: "order.checkoutForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "order.checkoutForm.inputValues.address.address.geocode.longitude"
            },
            { path: "order.tickets.guestDetails.form.inputValues.number" },
            {
              path: "order.tickets.guestDetails.form.inputValues.address.address.geocode.latitude"
            },
            {
              path: "order.tickets.guestDetails.form.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __checkout;
}
function updateCheckout(payload) {
  function __updateCheckout({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "guests.form.inputValues.number" },
          { path: "guests.form.inputValues.address.address.geocode.latitude" },
          { path: "guests.form.inputValues.address.address.geocode.longitude" }
        ]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "guests.form.inputValues.dateTime" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "PATCH",
      methodFqn: "wix.events.ticketing.CheckoutService.UpdateCheckout",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingCheckoutServiceUrl({
        protoPath: "/v1/checkout/{orderNumber}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.created" },
            { path: "order.updated" },
            { path: "order.checkoutForm.inputValues.dateTime" },
            { path: "order.tickets.checkIn.created" },
            { path: "order.tickets.guestDetails.form.inputValues.dateTime" },
            {
              path: "order.paymentDetails.transaction.scheduledActionExecutionDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "order.checkoutForm.inputValues.number" },
            {
              path: "order.checkoutForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "order.checkoutForm.inputValues.address.address.geocode.longitude"
            },
            { path: "order.tickets.guestDetails.form.inputValues.number" },
            {
              path: "order.tickets.guestDetails.form.inputValues.address.address.geocode.latitude"
            },
            {
              path: "order.tickets.guestDetails.form.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateCheckout;
}
function posCheckout(payload) {
  function __posCheckout({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.order",
      method: "POST",
      methodFqn: "wix.events.ticketing.CheckoutService.PosCheckout",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingCheckoutServiceUrl({
        protoPath: "/v1/pos-checkout",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "expires" },
            { path: "order.created" },
            { path: "order.updated" },
            { path: "order.checkoutForm.inputValues.dateTime" },
            { path: "order.tickets.checkIn.created" },
            { path: "order.tickets.guestDetails.form.inputValues.dateTime" },
            {
              path: "order.paymentDetails.transaction.scheduledActionExecutionDate"
            },
            { path: "reservations.ticket.salePeriod.startDate" },
            { path: "reservations.ticket.salePeriod.endDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "order.checkoutForm.inputValues.number" },
            {
              path: "order.checkoutForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "order.checkoutForm.inputValues.address.address.geocode.longitude"
            },
            { path: "order.tickets.guestDetails.form.inputValues.number" },
            {
              path: "order.tickets.guestDetails.form.inputValues.address.address.geocode.latitude"
            },
            {
              path: "order.tickets.guestDetails.form.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __posCheckout;
}

// src/events-v1-order-orders.types.ts
var SubdivisionType = /* @__PURE__ */ ((SubdivisionType2) => {
  SubdivisionType2["UNKNOWN_SUBDIVISION_TYPE"] = "UNKNOWN_SUBDIVISION_TYPE";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_1"] = "ADMINISTRATIVE_AREA_LEVEL_1";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_2"] = "ADMINISTRATIVE_AREA_LEVEL_2";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_3"] = "ADMINISTRATIVE_AREA_LEVEL_3";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_4"] = "ADMINISTRATIVE_AREA_LEVEL_4";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_5"] = "ADMINISTRATIVE_AREA_LEVEL_5";
  SubdivisionType2["COUNTRY"] = "COUNTRY";
  return SubdivisionType2;
})(SubdivisionType || {});
var OrderStatus = /* @__PURE__ */ ((OrderStatus2) => {
  OrderStatus2["NA_ORDER_STATUS"] = "NA_ORDER_STATUS";
  OrderStatus2["FREE"] = "FREE";
  OrderStatus2["PENDING"] = "PENDING";
  OrderStatus2["PAID"] = "PAID";
  OrderStatus2["OFFLINE_PENDING"] = "OFFLINE_PENDING";
  OrderStatus2["INITIATED"] = "INITIATED";
  OrderStatus2["CANCELED"] = "CANCELED";
  OrderStatus2["DECLINED"] = "DECLINED";
  OrderStatus2["AUTHORIZED"] = "AUTHORIZED";
  OrderStatus2["VOIDED"] = "VOIDED";
  OrderStatus2["PARTIALLY_PAID"] = "PARTIALLY_PAID";
  return OrderStatus2;
})(OrderStatus || {});
var ChannelType = /* @__PURE__ */ ((ChannelType2) => {
  ChannelType2["ONLINE"] = "ONLINE";
  ChannelType2["OFFLINE_POS"] = "OFFLINE_POS";
  return ChannelType2;
})(ChannelType || {});
var TaxType = /* @__PURE__ */ ((TaxType2) => {
  TaxType2["INCLUDED"] = "INCLUDED";
  TaxType2["ADDED"] = "ADDED";
  TaxType2["ADDED_AT_CHECKOUT"] = "ADDED_AT_CHECKOUT";
  return TaxType2;
})(TaxType || {});
var FeeName = /* @__PURE__ */ ((FeeName2) => {
  FeeName2["WIX_FEE"] = "WIX_FEE";
  return FeeName2;
})(FeeName || {});
var FeeType = /* @__PURE__ */ ((FeeType2) => {
  FeeType2["FEE_ADDED"] = "FEE_ADDED";
  FeeType2["FEE_INCLUDED"] = "FEE_INCLUDED";
  FeeType2["FEE_ADDED_AT_CHECKOUT"] = "FEE_ADDED_AT_CHECKOUT";
  return FeeType2;
})(FeeType || {});
var ScheduledActionEnumAction = /* @__PURE__ */ ((ScheduledActionEnumAction2) => {
  ScheduledActionEnumAction2["UNKNOWN_ACTION"] = "UNKNOWN_ACTION";
  ScheduledActionEnumAction2["CAPTURE"] = "CAPTURE";
  ScheduledActionEnumAction2["VOID"] = "VOID";
  return ScheduledActionEnumAction2;
})(ScheduledActionEnumAction || {});
var Action = /* @__PURE__ */ ((Action2) => {
  Action2["ARCHIVE"] = "ARCHIVE";
  Action2["UNARCHIVE"] = "UNARCHIVE";
  Action2["CONFIRM"] = "CONFIRM";
  Action2["CAPTURE"] = "CAPTURE";
  Action2["VOID"] = "VOID";
  return Action2;
})(Action || {});
var OrderType = /* @__PURE__ */ ((OrderType2) => {
  OrderType2["UNASSIGNED_TICKETS"] = "UNASSIGNED_TICKETS";
  OrderType2["ASSIGNED_TICKETS"] = "ASSIGNED_TICKETS";
  return OrderType2;
})(OrderType || {});
var OrderFieldset = /* @__PURE__ */ ((OrderFieldset2) => {
  OrderFieldset2["TICKETS"] = "TICKETS";
  OrderFieldset2["DETAILS"] = "DETAILS";
  OrderFieldset2["FORM"] = "FORM";
  OrderFieldset2["INVOICE"] = "INVOICE";
  return OrderFieldset2;
})(OrderFieldset || {});
var OrderTag = /* @__PURE__ */ ((OrderTag2) => {
  OrderTag2["CONFIRMED"] = "CONFIRMED";
  OrderTag2["UNCONFIRMED"] = "UNCONFIRMED";
  OrderTag2["MEMBER"] = "MEMBER";
  OrderTag2["ARCHIVED"] = "ARCHIVED";
  OrderTag2["NON_ARCHIVED"] = "NON_ARCHIVED";
  OrderTag2["FULLY_CHECKED_IN"] = "FULLY_CHECKED_IN";
  OrderTag2["NOT_FULLY_CHECKED_IN"] = "NOT_FULLY_CHECKED_IN";
  return OrderTag2;
})(OrderTag || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
var ReservationStatus = /* @__PURE__ */ ((ReservationStatus2) => {
  ReservationStatus2["RESERVATION_PENDING"] = "RESERVATION_PENDING";
  ReservationStatus2["RESERVATION_CONFIRMED"] = "RESERVATION_CONFIRMED";
  ReservationStatus2["RESERVATION_CANCELED"] = "RESERVATION_CANCELED";
  ReservationStatus2["RESERVATION_CANCELED_MANUALLY"] = "RESERVATION_CANCELED_MANUALLY";
  ReservationStatus2["RESERVATION_EXPIRED"] = "RESERVATION_EXPIRED";
  return ReservationStatus2;
})(ReservationStatus || {});
var State = /* @__PURE__ */ ((State2) => {
  State2["INCLUDE_HIDDEN_NOT_ON_SALE"] = "INCLUDE_HIDDEN_NOT_ON_SALE";
  return State2;
})(State || {});
var TicketSaleStatus = /* @__PURE__ */ ((TicketSaleStatus2) => {
  TicketSaleStatus2["SALE_SCHEDULED"] = "SALE_SCHEDULED";
  TicketSaleStatus2["SALE_STARTED"] = "SALE_STARTED";
  TicketSaleStatus2["SALE_ENDED"] = "SALE_ENDED";
  return TicketSaleStatus2;
})(TicketSaleStatus || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["STANDARD"] = "STANDARD";
  Type2["DONATION"] = "DONATION";
  return Type2;
})(Type || {});
var TicketDefinitionFieldset = /* @__PURE__ */ ((TicketDefinitionFieldset2) => {
  TicketDefinitionFieldset2["POLICY"] = "POLICY";
  TicketDefinitionFieldset2["DASHBOARD"] = "DASHBOARD";
  return TicketDefinitionFieldset2;
})(TicketDefinitionFieldset || {});

// src/events-v1-order-orders.meta.ts
function listOrders2() {
  const payload = {};
  const getRequestOptions = listOrders(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/orders",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getOrder2() {
  const payload = { eventId: ":eventId", orderNumber: ":orderNumber" };
  const getRequestOptions = getOrder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/events/{eventId}/orders/{orderNumber}",
    pathParams: { eventId: "eventId", orderNumber: "orderNumber" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateOrder2() {
  const payload = { eventId: ":eventId", orderNumber: ":orderNumber" };
  const getRequestOptions = updateOrder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/events/{eventId}/orders/{orderNumber}",
    pathParams: { eventId: "eventId", orderNumber: "orderNumber" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateOrders2() {
  const payload = { eventId: ":eventId" };
  const getRequestOptions = bulkUpdateOrders(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/events/{eventId}/orders",
    pathParams: { eventId: "eventId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function confirmOrder2() {
  const payload = { eventId: ":eventId" };
  const getRequestOptions = confirmOrder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/events/{eventId}/orders/confirm",
    pathParams: { eventId: "eventId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getSummary2() {
  const payload = {};
  const getRequestOptions = getSummary(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/orders/summary",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function captureAuthorizedPayment2() {
  const payload = { orderNumber: ":orderNumber" };
  const getRequestOptions = captureAuthorizedPayment(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/events/orders/{orderNumber}/capture",
    pathParams: { orderNumber: "orderNumber" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function voidAuthorizedPayment2() {
  const payload = { orderNumber: ":orderNumber" };
  const getRequestOptions = voidAuthorizedPayment(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/events/orders/{orderNumber}/void",
    pathParams: { orderNumber: "orderNumber" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getCheckoutOptions2() {
  const payload = {};
  const getRequestOptions = getCheckoutOptions(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/checkout/options",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listAvailableTickets2() {
  const payload = {};
  const getRequestOptions = listAvailableTickets(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/checkout/available-tickets",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryAvailableTickets2() {
  const payload = {};
  const getRequestOptions = queryAvailableTickets(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/checkout/available-tickets/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createReservation2() {
  const payload = {};
  const getRequestOptions = createReservation(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/checkout/reservations",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function cancelReservation2() {
  const payload = { id: ":id" };
  const getRequestOptions = cancelReservation(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/checkout/reservations/{id}",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getInvoice2() {
  const payload = { reservationId: ":reservationId" };
  const getRequestOptions = getInvoice(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/checkout/invoices/{reservationId}",
    pathParams: { reservationId: "reservationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function checkout2() {
  const payload = {};
  const getRequestOptions = checkout(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/checkout",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateCheckout2() {
  const payload = { orderNumber: ":orderNumber" };
  const getRequestOptions = updateCheckout(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/checkout/{orderNumber}",
    pathParams: { orderNumber: "orderNumber" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function posCheckout2() {
  const payload = {};
  const getRequestOptions = posCheckout(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/pos-checkout",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  Action as ActionOriginal,
  ChannelType as ChannelTypeOriginal,
  FeeName as FeeNameOriginal,
  FeeType as FeeTypeOriginal,
  OrderFieldset as OrderFieldsetOriginal,
  OrderStatus as OrderStatusOriginal,
  OrderTag as OrderTagOriginal,
  OrderType as OrderTypeOriginal,
  ReservationStatus as ReservationStatusOriginal,
  ScheduledActionEnumAction as ScheduledActionEnumActionOriginal,
  State as StateOriginal,
  SubdivisionType as SubdivisionTypeOriginal,
  TaxType as TaxTypeOriginal,
  TicketDefinitionFieldset as TicketDefinitionFieldsetOriginal,
  TicketSaleStatus as TicketSaleStatusOriginal,
  Type as TypeOriginal,
  WebhookIdentityType as WebhookIdentityTypeOriginal,
  bulkUpdateOrders2 as bulkUpdateOrders,
  cancelReservation2 as cancelReservation,
  captureAuthorizedPayment2 as captureAuthorizedPayment,
  checkout2 as checkout,
  confirmOrder2 as confirmOrder,
  createReservation2 as createReservation,
  getCheckoutOptions2 as getCheckoutOptions,
  getInvoice2 as getInvoice,
  getOrder2 as getOrder,
  getSummary2 as getSummary,
  listAvailableTickets2 as listAvailableTickets,
  listOrders2 as listOrders,
  posCheckout2 as posCheckout,
  queryAvailableTickets2 as queryAvailableTickets,
  updateCheckout2 as updateCheckout,
  updateOrder2 as updateOrder,
  voidAuthorizedPayment2 as voidAuthorizedPayment
};
