// @wix/design-system/dist/cjs/Page/test/examples/BMPageContainer.st.css.js

require("../../../Foundation/stylable/colors.st.css.js");
const { classesRuntime, statesRuntime } = require("./../../../stylable-cjs-runtime.js");



var _namespace_ = "wds_1_226_0_BMPageContainer";
var _style_ = classesRuntime.bind(null, _namespace_);

module.exports.cssStates = statesRuntime.bind(null, _namespace_);
module.exports.style = _style_;
module.exports.st = _style_;

module.exports.namespace = _namespace_;
module.exports.classes = {"root":"wds_1_226_0_BMPageContainer__root","header":"wds_1_226_0_BMPageContainer__header","body":"wds_1_226_0_BMPageContainer__body","sideBar":"wds_1_226_0_BMPageContainer__sideBar","sideBarContent":"wds_1_226_0_BMPageContainer__sideBarContent","rightSide":"wds_1_226_0_BMPageContainer__rightSide","mainContentContainer":"wds_1_226_0_BMPageContainer__mainContentContainer","mainContent":"wds_1_226_0_BMPageContainer__mainContent","withLoadingIndicatorWrapper":"wds_1_226_0_BMPageContainer__withLoadingIndicatorWrapper","withLoadingIndicatorContainer":"wds_1_226_0_BMPageContainer__withLoadingIndicatorContainer"};
module.exports.keyframes = {}; 
module.exports.layers = {};
module.exports.containers = {};
module.exports.stVars = {}; 
module.exports.vars = {};
