// @wix/patterns/dist/cjs/components/FiltersPanel/FiltersPanelFooter.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.FiltersPanelFooter = void 0;
var _designSystem = require("@wix/design-system");
var _react = _interopRequireDefault(require("react"));
var _mobxReactLite = require("mobx-react-lite");
var _FiltersPanelAppliedFiltersSection = require("./FiltersPanelAppliedFiltersSection");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/FiltersPanel/FiltersPanelFooter.tsx";
function _FiltersPanelFooter(props) {
  var _state$AddApplyFilter;
  const {
    state,
    onClearButtonClicked
  } = props;
  const FiltersPanelSelectedFiltersSection = (_state$AddApplyFilter = state.AddApplyFiltersButton) == null ? void 0 : _state$AddApplyFilter.FiltersPanelSelectedFiltersSection;
  return /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
    gap: "SP2",
    paddingTop: "SP3",
    paddingBottom: "SP3",
    paddingLeft: "SP4",
    paddingRight: "SP4",
    verticalAlign: "middle",
    width: "100%",
    boxSizing: "border-box",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 22,
      columnNumber: 5
    }
  }, FiltersPanelSelectedFiltersSection ? /*#__PURE__*/_react.default.createElement(FiltersPanelSelectedFiltersSection, {
    state: state,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 33,
      columnNumber: 9
    }
  }) : /*#__PURE__*/_react.default.createElement(_FiltersPanelAppliedFiltersSection.FiltersPanelAppliedFiltersSection, {
    state: state,
    onClearButtonClicked: onClearButtonClicked,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 35,
      columnNumber: 9
    }
  }));
}
const FiltersPanelFooter = exports.FiltersPanelFooter = (0, _mobxReactLite.observer)(_FiltersPanelFooter);
