// @wix/patterns/dist/cjs/components/FiltersPanel/FiltersPanelSelectedFiltersSection.js
"use strict";

exports.__esModule = true;
exports.FiltersPanelSelectedFiltersSection = void 0;
var _designSystem = require("@wix/design-system");
var _react = _interopRequireWildcard(require("react"));
var _mobxReactLite = require("mobx-react-lite");
var _react2 = require("@wix/bex-core/react");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/FiltersPanel/FiltersPanelSelectedFiltersSection.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _FiltersPanelSelectedFiltersSection(props) {
  const {
    state
  } = props;
  const container = (0, _react2.useWixPatternsContainer)();
  const {
    translate: t
  } = container;
  const {
    pendingQuery
  } = state;
  (0, _react.useEffect)(() => pendingQuery == null ? void 0 : pendingQuery.init(), []);
  if (pendingQuery == null) {
    return null;
  }
  const {
    trackedFiltersCount,
    hasChanged
  } = pendingQuery;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_designSystem.Text, {
    weight: "bold",
    dataHook: "applied-filters",
    size: "small",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 37,
      columnNumber: 7
    }
  }, hasChanged ? t('cairo.appliedFilters.selected', {
    num: trackedFiltersCount
  }) : t('cairo.appliedFilters', {
    num: trackedFiltersCount
  })), /*#__PURE__*/_react.default.createElement(_designSystem.TextButton, {
    dataHook: "clear-all-filters",
    size: "small",
    onClick: () => {
      pendingQuery.clearSelected();
    },
    disabled: !trackedFiltersCount,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 47,
      columnNumber: 7
    }
  }, t('cairo.clear')), /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
    align: "right",
    flex: 1,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 58,
      columnNumber: 7
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.Button, {
    dataHook: "apply-button",
    ellipsis: true,
    disabled: !hasChanged,
    onClick: () => {
      pendingQuery.apply();
    },
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 59,
      columnNumber: 9
    }
  }, t('cair.appliedFilters.Apply.CTA'))));
}
const FiltersPanelSelectedFiltersSection = exports.FiltersPanelSelectedFiltersSection = (0, _mobxReactLite.observer)(_FiltersPanelSelectedFiltersSection);
