// @wix/patterns/dist/cjs/components/FiltersPanel/FilterAccordionTitle.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.FilterAccordionTitle = void 0;
var _designSystem = require("@wix/design-system");
var _react = _interopRequireDefault(require("react"));
var _mobxReactLite = require("mobx-react-lite");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/FiltersPanel/FilterAccordionTitle.tsx";
function _FilterAccordionTitle(props) {
  const {
    state,
    filterProps
  } = props;
  const {
    accordionItemProps
  } = filterProps;
  const filter = state.getPendingFilter(props.filter);
  return /*#__PURE__*/_react.default.createElement(_designSystem.Text, {
    dataHook: `accordion-item-title-${filter.name}`,
    ellipsis: true,
    weight: "normal",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 21,
      columnNumber: 5
    }
  }, `${(accordionItemProps == null ? void 0 : accordionItemProps.label) ?? ''}${filter.size > 0 ? ` (${filter.size})` : ''}`);
}
const FilterAccordionTitle = exports.FilterAccordionTitle = (0, _mobxReactLite.observer)(_FilterAccordionTitle);
