// @wix/ricos/dist/cjs/ricos-common-plugins/blockquote/blockquote-converter.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var blockquote_converter_exports = {};
__export(blockquote_converter_exports, {
  blockquoteConverter: () => blockquoteConverter
});
module.exports = __toCommonJS(blockquote_converter_exports);
var import_v1 = require("./../../ricos-schema/legacy/wix/rich_content/v1/index.js");
var import_blockquote_defaults_json = __toESM(require("./../../ricos-schema/legacy/statics/blockquote.defaults.json.js"));
const toParagraphFields = (node) => {
  const { id: paragraphId, paragraphData = {} } = node;
  return { paragraphData, paragraphId };
};
const toParagraphNode = (fields) => (nodes) => {
  return {
    id: fields.paragraphId,
    type: import_v1.Node_Type.PARAGRAPH,
    paragraphData: fields.paragraphData,
    nodes
  };
};
const blockquoteConverter = {
  toPm: {
    types: [import_v1.Node_Type.BLOCKQUOTE],
    // @ts-expect-error $TSFixMe
    convert: (node, visit) => {
      const { id, style, blockquoteData } = node || {};
      const firstNode = node.nodes?.[0] || { id: "", paragraphData: {} };
      const { paragraphData, paragraphId } = toParagraphFields(firstNode);
      return {
        type: import_v1.Node_Type.BLOCKQUOTE,
        attrs: {
          paragraphId,
          textStyle: paragraphData.textStyle,
          indentation: blockquoteData?.indentation ?? import_blockquote_defaults_json.default.indentation,
          ...style ? { style } : {},
          id
        },
        // skip internal paragraph, go to text nodes
        content: node.nodes?.[0] ? visit(node.nodes[0]) : []
      };
    }
  },
  fromPm: {
    types: [import_v1.Node_Type.BLOCKQUOTE],
    // @ts-expect-error $TSFixMe
    convert: (node, visit) => {
      const { id, style, paragraphId, textStyle, indentation } = node.attrs || {};
      const paragraphNode = toParagraphNode({
        paragraphData: { indentation: 0, textStyle },
        paragraphId
      })(visit(node));
      return {
        type: import_v1.Node_Type.BLOCKQUOTE,
        id,
        nodes: [paragraphNode],
        ...style ? { style } : {},
        blockquoteData: {
          indentation
        }
      };
    }
  }
};
