// @wix/ricos/dist/cjs/ricos-common-plugins/blockquote/blockquote.testkit.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var blockquote_testkit_exports = {};
__export(blockquote_testkit_exports, {
  createBlockquoteTestkit: () => createBlockquoteTestkit
});
module.exports = __toCommonJS(blockquote_testkit_exports);
var import_wix_rich_content_common = require("./../../wix-rich-content-common/index.js");
var import_data_hooks = require("./data-hooks.js");
const createBlockquoteTestkit = () => {
  return {
    content: {},
    toolbars: {
      createTextButtonToolbars: (getTextButtonBase) => ({
        isActive: async () => {
          const active = getTextButtonBase().$((0, import_wix_rich_content_common.dataHook)([import_data_hooks.TEXT_BUTTON_BLOCKQUOTE, "active"]));
          return active.exists();
        },
        isDisabled: async () => {
          const disabled = getTextButtonBase().$((0, import_wix_rich_content_common.dataHook)([import_data_hooks.TEXT_BUTTON_BLOCKQUOTE, "disabled"]));
          return disabled.exists();
        },
        click: async () => {
          const button = getTextButtonBase().$((0, import_wix_rich_content_common.dataHook)(import_data_hooks.TEXT_BUTTON_BLOCKQUOTE));
          await button.click();
        }
      })
    }
  };
};
