// @wix/astro/build/dependencies/astro-auth/backend-runtime/chunk-RTCWF7UX.js
// src/runtime/backend/contextualClient.ts
import { createClient, OAuthStrategy } from "@wix/sdk";
import { WIX_CLIENT_ID } from "astro:env/client";

// src/utils/authAsyncLocalStorage.ts
import { AsyncLocalStorage } from "async_hooks";
var authAsyncLocalStorage = new AsyncLocalStorage();

// src/runtime/backend/hostProxy.ts
import { headlessNode } from "@wix/headless-node";
var defaultHost = headlessNode.host({});
var resolveHostForCurrentRequest = () => {
  const store = authAsyncLocalStorage.getStore();
  return store?.sdkHost ?? defaultHost;
};
var hostProxy = new Proxy(
  {},
  {
    get(target, prop) {
      const host = resolveHostForCurrentRequest();
      const value = host[prop];
      if (typeof value !== "function") {
        return value;
      }
      return function(...args) {
        const anotherHost = resolveHostForCurrentRequest();
        return (
          // eslint-disable-next-line @typescript-eslint/no-unsafe-function-type
          anotherHost[prop].apply(
            anotherHost,
            args
          )
        );
      };
    }
  }
);

// src/runtime/backend/contextualClient.ts
var authProxy = new Proxy(
  {},
  {
    get(target, prop) {
      const auth = OAuthStrategy({
        clientId: WIX_CLIENT_ID
      });
      const value = auth[prop];
      if (typeof value !== "function") {
        return value;
      }
      return function(...args) {
        const store = authAsyncLocalStorage.getStore();
        if (store?.sessionTokens) {
          auth.setTokens(store.sessionTokens);
        }
        return value.apply(auth, args);
      };
    }
  }
);
var contextualClient = createClient({
  auth: authProxy,
  host: hostProxy
});

// src/runtime/backend/elevatedContextualClient.ts
import { AppStrategy, createClient as createClient2 } from "@wix/sdk";
import { WIX_CLIENT_ID as WIX_CLIENT_ID2 } from "astro:env/client";
import {
  WIX_CLIENT_INSTANCE_ID,
  WIX_CLIENT_PUBLIC_KEY,
  WIX_CLIENT_SECRET
} from "astro:env/server";
var elevatedContextualClient = createClient2({
  auth: AppStrategy({
    appId: WIX_CLIENT_ID2,
    appSecret: WIX_CLIENT_SECRET,
    instanceId: WIX_CLIENT_INSTANCE_ID,
    publicKey: WIX_CLIENT_PUBLIC_KEY
  }),
  host: hostProxy
});

// src/runtime/backend/setupContextualClient.ts
contextualClient.enableContext("global");
elevatedContextualClient.enableContext("global", { elevated: true });

export {
  authAsyncLocalStorage
};
