// @wix/fast-gallery-core/dist/cjs/lib/assets/images.js
"use strict";

exports.__esModule = true;
exports.imagesWithSecondMedia = exports.imagesAndVideos = exports.images = void 0;
var _galleryTypes = require("../types/galleryTypes");
const testImages = [{
  id: '8b72558253b2502b401bb46e5599f22a',
  index: '0',
  mediaUrl: 'https://static.wixstatic.com/media/a9ff3b_1aa079fe615b4a0fbb9949a524ebc326~mv2.jpg',
  metaData: {
    width: 1870,
    height: 1054,
    title: '',
    description: '',
    alt: 'First image',
    link: {
      url: 'https://static.wixstatic.com/media/a9ff3b_1aa079fe615b4a0fbb9949a524ebc326~mv2.jpg',
      target: '_self'
    }
  },
  type: _galleryTypes.MediaTypes.IMAGE
}, {
  id: '8b72558253b2502b401bb46e5599f23a',
  index: '1',
  mediaUrl: 'https://static.wixstatic.com/media/ea3d98_23ae925ffe8a4751951eb7f504a07ca3~mv2.jpg',
  metaData: {
    width: 1870,
    height: 1054,
    title: '',
    description: '',
    alt: 'Second image',
    link: {
      url: 'https://static.wixstatic.com/media/ea3d98_23ae925ffe8a4751951eb7f504a07ca3~mv2.jpg',
      target: '_self'
    }
  },
  type: _galleryTypes.MediaTypes.IMAGE
}, {
  id: '8b72558253b2502b401bb46e5599f22b',
  index: '2',
  mediaUrl: 'https://static.wixstatic.com/media/ea3d98_be64ed766f36409c892af71ab9533c6c~mv2.jpg',
  metaData: {
    width: 699,
    height: 1054,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://static.wixstatic.com/media/ea3d98_be64ed766f36409c892af71ab9533c6c~mv2.jpg',
      target: '_self'
    }
  },
  type: _galleryTypes.MediaTypes.IMAGE
}, {
  id: '8b72558253b2502b401bb46e5599f22c',
  index: '3',
  mediaUrl: 'https://static.wixstatic.com/media/ea3d98_89da6c402e594614a2c1a58d70441976~mv2.jpg',
  metaData: {
    width: 1054,
    height: 1054,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://static.wixstatic.com/media/ea3d98_89da6c402e594614a2c1a58d70441976~mv2.jpg',
      target: '_self'
    }
  },
  type: _galleryTypes.MediaTypes.IMAGE
}, {
  id: '8b72558253b2502b401bb46e5599f22d',
  index: '4',
  mediaUrl: 'https://static.wixstatic.com/media/ea3d98_4ba763c2c25242e2bd485f2a10841f2b~mv2.jpg',
  metaData: {
    width: 699,
    height: 1054,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://static.wixstatic.com/media/ea3d98_4ba763c2c25242e2bd485f2a10841f2b~mv2.jpg',
      target: '_self'
    }
  },
  type: _galleryTypes.MediaTypes.IMAGE
}, {
  id: '8b72558253b2502b401bb46e5599f22e',
  index: '5',
  mediaUrl: 'https://static.wixstatic.com/media/ea3d98_6aec3ae64e00451c813f6b87bef43250~mv2.jpg',
  metaData: {
    width: 699,
    height: 1054,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://static.wixstatic.com/media/ea3d98_6aec3ae64e00451c813f6b87bef43250~mv2.jpg',
      target: '_self'
    }
  },
  type: _galleryTypes.MediaTypes.IMAGE
}, {
  id: '8b72558253b2502b401bb46e5599f22f',
  index: '6',
  mediaUrl: 'https://static.wixstatic.com/media/ea3d98_f8b3cecc6598487cb501c73b5a744411~mv2.jpg',
  metaData: {
    width: 935,
    height: 1054,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://static.wixstatic.com/media/ea3d98_f8b3cecc6598487cb501c73b5a744411~mv2.jpg',
      target: '_self'
    }
  },
  type: _galleryTypes.MediaTypes.IMAGE
}, {
  id: '8b72558253b2502b401bb46e5599f22g',
  index: '7',
  mediaUrl: 'https://static.wixstatic.com/media/ea3d98_11ef8ac0bbae4d5483222fec20b05d12~mv2.jpg',
  metaData: {
    width: 699,
    height: 1054,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://static.wixstatic.com/media/ea3d98_11ef8ac0bbae4d5483222fec20b05d12~mv2.jpg',
      target: '_self'
    }
  },
  type: _galleryTypes.MediaTypes.IMAGE
}, {
  id: '8b72558253b2502b401bb46e5599f22h',
  index: '8',
  mediaUrl: 'https://static.wixstatic.com/media/ea3d98_c72c4c47268c423d959db5e25c8eec93~mv2.jpg',
  metaData: {
    width: 1054,
    height: 1870,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://static.wixstatic.com/media/ea3d98_c72c4c47268c423d959db5e25c8eec93~mv2.jpg',
      target: '_self'
    }
  },
  type: _galleryTypes.MediaTypes.IMAGE
}, {
  id: '8b72558253b2502b401bb46e5599f22i',
  index: '9',
  mediaUrl: 'https://static.wixstatic.com/media/ea3d98_5bdeafbee0c04267847ee8ea3b36bf39~mv2.jpg',
  metaData: {
    width: 1054,
    height: 1054,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://static.wixstatic.com/media/ea3d98_5bdeafbee0c04267847ee8ea3b36bf39~mv2.jpg',
      target: '_self'
    }
  },
  type: _galleryTypes.MediaTypes.IMAGE
}, {
  id: '8b72558253b2502b401bb46e5599f22j',
  index: '10',
  mediaUrl: 'https://static.wixstatic.com/media/ea3d98_1bb813ee4e9c4556aa3f247fbe4dc3bc~mv2.jpg',
  metaData: {
    width: 699,
    height: 1054,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://static.wixstatic.com/media/ea3d98_1bb813ee4e9c4556aa3f247fbe4dc3bc~mv2.jpg',
      target: '_self'
    }
  },
  type: _galleryTypes.MediaTypes.IMAGE
}, {
  id: '8b72558253b2502b401bb46e5599f11j',
  index: '11',
  mediaUrl: 'https://static.wixstatic.com/media/ea3d98_0c64d9d70b57468190612b8fcfd8f08c~mv2.jpg',
  metaData: {
    width: 1870,
    height: 1054,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://static.wixstatic.com/media/ea3d98_0c64d9d70b57468190612b8fcfd8f08c~mv2.jpg',
      target: '_self'
    }
  },
  type: _galleryTypes.MediaTypes.IMAGE
}, {
  id: '8b72558253b2502b401bb46e5599f12j',
  index: '12',
  mediaUrl: 'https://static.wixstatic.com/media/ea3d98_f8cbfc1edb064789860e1d27a97b5bc7~mv2.jpg',
  metaData: {
    width: 699,
    height: 1054,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://static.wixstatic.com/media/ea3d98_f8cbfc1edb064789860e1d27a97b5bc7~mv2.jpg',
      target: '_self'
    }
  },
  type: _galleryTypes.MediaTypes.IMAGE
}, {
  id: '8b72558253b2502b401bb46e5599f13j',
  index: '13',
  mediaUrl: 'https://static.wixstatic.com/media/ea3d98_1109b284e62b42c68241884544eaa02f~mv2.jpg',
  metaData: {
    width: 1054,
    height: 1054,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://static.wixstatic.com/media/ea3d98_1109b284e62b42c68241884544eaa02f~mv2.jpg',
      target: '_self'
    }
  },
  type: _galleryTypes.MediaTypes.IMAGE
}, {
  id: '8b72558253b2502b401bb46e5599f14j',
  index: '14',
  mediaUrl: 'https://static.wixstatic.com/media/ea3d98_66e004a53e10416ab01728e3a59eada1~mv2.jpg',
  metaData: {
    width: 699,
    height: 1054,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://static.wixstatic.com/media/ea3d98_66e004a53e10416ab01728e3a59eada1~mv2.jpg',
      target: '_self'
    }
  },
  type: _galleryTypes.MediaTypes.IMAGE
}, {
  id: '8b72558253b2502b401bb46e5599f15j',
  index: '15',
  mediaUrl: 'https://static.wixstatic.com/media/ea3d98_74901b1ec6584e3f9afe4785471c3d39~mv2.jpg',
  metaData: {
    width: 699,
    height: 1054,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://static.wixstatic.com/media/ea3d98_74901b1ec6584e3f9afe4785471c3d39~mv2.jpg',
      target: '_self'
    }
  },
  type: _galleryTypes.MediaTypes.IMAGE
}, {
  id: '8b72558253b2502b401bb46e5599f16j',
  index: '16',
  mediaUrl: 'https://static.wixstatic.com/media/ea3d98_deb70f106d944619ae5e586764c66631~mv2.jpg',
  metaData: {
    width: 953,
    height: 1054,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://static.wixstatic.com/media/ea3d98_deb70f106d944619ae5e586764c66631~mv2.jpg',
      target: '_self'
    }
  },
  type: _galleryTypes.MediaTypes.IMAGE
}, {
  id: '8b72558253b2502b401bb46e5599f17j',
  index: '17',
  mediaUrl: 'https://static.wixstatic.com/media/ea3d98_a918ad76122f42869c7fcf7602803c61~mv2.jpg',
  metaData: {
    width: 699,
    height: 1054,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://static.wixstatic.com/media/ea3d98_a918ad76122f42869c7fcf7602803c61~mv2.jpg',
      target: '_self'
    }
  },
  type: _galleryTypes.MediaTypes.IMAGE
}, {
  id: '8b72558253b2502b401bb46e5599f18j',
  index: '18',
  mediaUrl: 'https://static.wixstatic.com/media/ea3d98_88e9bb3945084b568573b83bf73894f0~mv2.jpg',
  metaData: {
    width: 1870,
    height: 1054,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://static.wixstatic.com/media/ea3d98_88e9bb3945084b568573b83bf73894f0~mv2.jpg',
      target: '_self'
    }
  },
  type: _galleryTypes.MediaTypes.IMAGE
}, {
  id: '8b72558253b2502b401bb46e5599f19j',
  index: '19',
  mediaUrl: 'https://static.wixstatic.com/media/ea3d98_fd3366ade12a408bbec40e6601873dd6~mv2.jpg',
  metaData: {
    width: 1054,
    height: 1054,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://static.wixstatic.com/media/ea3d98_fd3366ade12a408bbec40e6601873dd6~mv2.jpg',
      target: '_self'
    }
  },
  type: _galleryTypes.MediaTypes.IMAGE
}, {
  id: '8b72558253b2502b401bb46e5599f20j',
  index: '20',
  mediaUrl: 'https://static.wixstatic.com/media/ea3d98_4826539ce49740a4b721f23fd4d171df~mv2.jpg',
  metaData: {
    width: 699,
    height: 1054,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://static.wixstatic.com/media/ea3d98_4826539ce49740a4b721f23fd4d171df~mv2.jpg',
      target: '_self'
    }
  },
  type: _galleryTypes.MediaTypes.IMAGE
}, {
  id: '8b72558253b2502b401bb46e5599f21j',
  index: '20',
  mediaUrl: 'https://static.wixstatic.com/media/ea3d98_a1b0a426d2d5436c8c8db22d674c4f3e~mv2.jpg',
  metaData: {
    width: 1870,
    height: 1054,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://static.wixstatic.com/media/ea3d98_a1b0a426d2d5436c8c8db22d674c4f3e~mv2.jpg',
      target: '_self'
    }
  },
  type: _galleryTypes.MediaTypes.IMAGE
}, {
  id: '8b72558253b2502b401bb46e5599f22s',
  index: '21',
  mediaUrl: 'https://static.wixstatic.com/media/ea3d98_659dfe75ad7948d68684be27db0e4e58~mv2.jpg',
  metaData: {
    width: 1870,
    height: 1054,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://static.wixstatic.com/media/ea3d98_659dfe75ad7948d68684be27db0e4e58~mv2.jpg',
      target: '_self'
    }
  },
  type: _galleryTypes.MediaTypes.IMAGE
}];
const testVideos = [{
  id: 'd19570fa-e97f-4b3d-b8ad-4e24b3a9a3ec',
  index: '10',
  mediaUrl: 'https://video.wixstatic.com/video/0d72ac_31569a98fd8d436f98ae9c827c216443/1080p/mp4/file.mp4',
  metaData: {
    width: 1920,
    height: 1080,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://video.wixstatic.com/video/0d72ac_31569a98fd8d436f98ae9c827c216443/1080p/mp4/file.mp4',
      target: '_self'
    },
    posters: [{
      url: 'https://static.wixstatic.com/media/0d72ac_31569a98fd8d436f98ae9c827c216443f000.jpg',
      width: 1920,
      height: 1080
    }],
    duration: 14914
  },
  type: _galleryTypes.MediaTypes.VIDEO
}, {
  id: '5f51db65-4104-447b-b675-94657671a73f',
  index: '11',
  mediaUrl: 'https://video.wixstatic.com/video/0d72ac_3e5c5b13a90749a6a02c845e8d1fa4dc/1080p/mp4/file.mp4',
  metaData: {
    width: 1920,
    height: 1080,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://video.wixstatic.com/video/0d72ac_3e5c5b13a90749a6a02c845e8d1fa4dc/1080p/mp4/file.mp4',
      target: '_self'
    },
    posters: [{
      url: 'https://static.wixstatic.com/media/0d72ac_3e5c5b13a90749a6a02c845e8d1fa4dcf000.jpg',
      width: 1920,
      height: 1080
    }],
    duration: 44878
  },
  type: _galleryTypes.MediaTypes.VIDEO
}, {
  id: '98c5601c-3486-452b-933c-49cb451d676f',
  index: '12',
  mediaUrl: 'https://video.wixstatic.com/video/0d72ac_8cebf3ded81a496cb5a1dde4d3ebf0db/1080p/mp4/file.mp4',
  metaData: {
    width: 1920,
    height: 1080,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://video.wixstatic.com/video/0d72ac_8cebf3ded81a496cb5a1dde4d3ebf0db/1080p/mp4/file.mp4',
      target: '_self'
    },
    posters: [{
      url: 'https://static.wixstatic.com/media/0d72ac_8cebf3ded81a496cb5a1dde4d3ebf0dbf000.jpg',
      width: 1920,
      height: 1080
    }],
    duration: 24290
  },
  type: _galleryTypes.MediaTypes.VIDEO
}, {
  id: '8dc4f30b-85cc-4795-9aa5-5ccb1bdb4ef1',
  index: '13',
  mediaUrl: 'https://video.wixstatic.com/video/0d72ac_11bbc686975e49caa89013d03e05d554/1080p/mp4/file.mp4',
  metaData: {
    width: 1920,
    height: 1080,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://video.wixstatic.com/video/0d72ac_11bbc686975e49caa89013d03e05d554/1080p/mp4/file.mp4',
      target: '_self'
    },
    posters: [{
      url: 'https://static.wixstatic.com/media/0d72ac_11bbc686975e49caa89013d03e05d554f000.jpg',
      width: 1920,
      height: 1080
    }],
    duration: 767
  },
  type: _galleryTypes.MediaTypes.VIDEO
}, {
  id: '1fc26b40-8e53-4de2-afa7-eefa8be40a01',
  index: '14',
  mediaUrl: 'https://video.wixstatic.com/video/0d72ac_ec7acce34f9a48e4adcca80aa7f46827/1080p/mp4/file.mp4',
  metaData: {
    width: 1920,
    height: 1080,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://video.wixstatic.com/video/0d72ac_ec7acce34f9a48e4adcca80aa7f46827/1080p/mp4/file.mp4',
      target: '_self'
    },
    posters: [{
      url: 'https://static.wixstatic.com/media/0d72ac_ec7acce34f9a48e4adcca80aa7f46827f000.jpg',
      width: 1920,
      height: 1080
    }],
    duration: 15048
  },
  type: _galleryTypes.MediaTypes.VIDEO
}, {
  id: '97143d61-94ef-42af-83b2-dc91ade1b5ed',
  index: '15',
  mediaUrl: 'https://video.wixstatic.com/video/0d72ac_8473aaf089cf4f78ad4e037c9340db47/1080p/mp4/file.mp4',
  metaData: {
    width: 1920,
    height: 1080,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://video.wixstatic.com/video/0d72ac_8473aaf089cf4f78ad4e037c9340db47/1080p/mp4/file.mp4',
      target: '_self'
    },
    posters: [{
      url: 'https://static.wixstatic.com/media/0d72ac_8473aaf089cf4f78ad4e037c9340db47f000.jpg',
      width: 1920,
      height: 1080
    }],
    duration: 56656
  },
  type: _galleryTypes.MediaTypes.VIDEO
}, {
  id: 'b801aac0-90d0-439e-9bdf-ad01491f75ae',
  index: '16',
  mediaUrl: 'https://video.wixstatic.com/video/0d72ac_648a202922414ffa836f639dee9ef389/1080p/mp4/file.mp4',
  metaData: {
    width: 1920,
    height: 1080,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://video.wixstatic.com/video/0d72ac_648a202922414ffa836f639dee9ef389/1080p/mp4/file.mp4',
      target: '_self'
    },
    posters: [{
      url: 'https://static.wixstatic.com/media/0d72ac_648a202922414ffa836f639dee9ef389f000.jpg',
      width: 1920,
      height: 1080
    }],
    duration: 14614
  },
  type: _galleryTypes.MediaTypes.VIDEO
}, {
  id: 'd7c33ff4-b737-41b2-84b4-d5ea320dd1b1',
  index: '17',
  mediaUrl: 'https://video.wixstatic.com/video/0d72ac_7ebf5b4a9b23486fa1296c0050fbb4ae/1080p/mp4/file.mp4',
  metaData: {
    width: 1920,
    height: 1080,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://video.wixstatic.com/video/0d72ac_7ebf5b4a9b23486fa1296c0050fbb4ae/1080p/mp4/file.mp4',
      target: '_self'
    },
    posters: [{
      url: 'https://static.wixstatic.com/media/0d72ac_7ebf5b4a9b23486fa1296c0050fbb4aef000.jpg',
      width: 1920,
      height: 1080
    }],
    duration: 12046
  },
  type: _galleryTypes.MediaTypes.VIDEO
}, {
  id: 'c2e7f5d8-1e87-40ed-8d79-62adfe7e7f91',
  index: '18',
  mediaUrl: 'https://video.wixstatic.com/video/0d72ac_4f5e604d0a16487a8a8d493b88ec85e0/1080p/mp4/file.mp4',
  metaData: {
    width: 1920,
    height: 1080,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://video.wixstatic.com/video/0d72ac_4f5e604d0a16487a8a8d493b88ec85e0/1080p/mp4/file.mp4',
      target: '_self'
    },
    posters: [{
      url: 'https://static.wixstatic.com/media/0d72ac_4f5e604d0a16487a8a8d493b88ec85e0f000.jpg',
      width: 1920,
      height: 1080
    }],
    duration: 27496
  },
  type: _galleryTypes.MediaTypes.VIDEO
}, {
  id: '9b9a2f2f-4b56-4c7d-bdd8-fc2a1abfb98f',
  index: '19',
  mediaUrl: 'https://video.wixstatic.com/video/0d72ac_90a1c03adf4e46c7b6d7526dcf2f26b4/1080p/mp4/file.mp4',
  metaData: {
    width: 1920,
    height: 1080,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://video.wixstatic.com/video/0d72ac_90a1c03adf4e46c7b6d7526dcf2f26b4/1080p/mp4/file.mp4',
      target: '_self'
    },
    posters: [{
      url: 'https://static.wixstatic.com/media/0d72ac_90a1c03adf4e46c7b6d7526dcf2f26b4f000.jpg',
      width: 1920,
      height: 1080
    }],
    duration: 16100
  },
  type: _galleryTypes.MediaTypes.VIDEO
}, {
  id: '9bd8d960-e470-476a-b3b6-bf8d580e07f8',
  index: '20',
  mediaUrl: 'https://video.wixstatic.com/video/0d72ac_91b8e09fa11c4f1cb0eb23a2e618a589/1080p/mp4/file.mp4',
  metaData: {
    width: 1920,
    height: 1080,
    title: '',
    description: '',
    alt: '',
    link: {
      url: 'https://video.wixstatic.com/video/0d72ac_91b8e09fa11c4f1cb0eb23a2e618a589/1080p/mp4/file.mp4',
      target: '_self'
    },
    posters: [{
      url: 'https://static.wixstatic.com/media/0d72ac_91b8e09fa11c4f1cb0eb23a2e618a589f000.jpg',
      width: 1920,
      height: 1080
    }],
    duration: 8323
  },
  type: _galleryTypes.MediaTypes.VIDEO
}];
const imagesAndVideos = exports.imagesAndVideos = [...testImages, ...testVideos];
const images = exports.images = testImages.map((image, index) => ({
  ...image,
  metaData: {
    ...image.metaData,
    title: 'Title ' + index,
    description: 'Description ' + index
  }
}));
const imagesWithSecondMedia = exports.imagesWithSecondMedia = testImages.map((image, index) => ({
  firstMedia: image,
  secondMedia: testVideos[index % testVideos.length],
  title: 'Title ' + index,
  description: 'Description ' + index,
  focalPoint: {
    x: 0.5,
    y: 0.5
  }
}));
