// @wix/wix-ui-test-utils/dist/cjs/puppeteer/puppeteer.js
"use strict";

exports.__esModule = true;
exports.puppeteerTestkitFactoryCreator = puppeteerTestkitFactoryCreator;
exports.puppeteerUniTestkitFactoryCreator = puppeteerUniTestkitFactoryCreator;
var _unidriverPuppeteer = require("@wix/unidriver-puppeteer");
/** @DEPRECATED please use `puppeteerUniTestkitFactoryCreator` */
function puppeteerTestkitFactoryCreator(driverFactory) {
  return async obj => driverFactory(await obj.page.$(`[data-hook='${obj.dataHook}']`), obj.page, {
    dataHook: obj.dataHook
  });
}
function puppeteerUniTestkitFactoryCreator(driverFactory) {
  return async obj => {
    const {
      dataHook,
      wrapper: element,
      page
    } = obj; // destructuring `wrapper` so that type inference works well
    const selector = `[data-hook='${dataHook}']`;
    const base = element ? (0, _unidriverPuppeteer.pupUniDriver)(async () => ({
      element: await element.$(selector),
      page,
      selector
    })) : (0, _unidriverPuppeteer.pupUniDriver)({
      page,
      selector
    });
    const body = (0, _unidriverPuppeteer.pupUniDriver)({
      page,
      selector: 'body'
    });
    return driverFactory(base, body, {
      dataHook
    });
  };
}
