// @wix/astro/build/dependencies/astro-auth/backend-runtime/middleware/auth.js
import {
  saveSessionTokensToCookie
} from "../chunk-HPW4ZAEJ.js";
import {
  authAsyncLocalStorage
} from "../chunk-RTCWF7UX.js";
import "../chunk-MLKGABMK.js";

// src/middleware/auth.ts
import { headlessNode } from "@wix/headless-node";

// src/utils/generateVisitorTokens.ts
import { OAuthStrategy } from "@wix/sdk";
import { WIX_CLIENT_ID } from "astro:env/client";
var auth = OAuthStrategy({
  clientId: WIX_CLIENT_ID
});
async function generateVisitorTokens() {
  return auth.generateVisitorTokens();
}

// src/utils/getSessionTokensFromCookie.ts
import { TokenRole } from "@wix/sdk";
import { z } from "astro/zod";
import { WIX_CLIENT_ID as WIX_CLIENT_ID2 } from "astro:env/client";
var tokensSchema = z.object({
  clientId: z.string(),
  tokens: z.object({
    accessToken: z.object({
      expiresAt: z.number(),
      value: z.string()
    }),
    refreshToken: z.object({
      role: z.nativeEnum(TokenRole),
      value: z.string()
    })
  })
});
function getSessionTokensFromCookie(context) {
  if (!context.isPrerendered) {
    const rawCookie = context.cookies.get("wixSession")?.json();
    if (rawCookie != null) {
      const tokensParseResult = tokensSchema.safeParse(rawCookie);
      if (tokensParseResult.success && tokensParseResult.data.clientId === WIX_CLIENT_ID2) {
        return tokensParseResult.data.tokens;
      }
    }
  }
  return null;
}

// src/middleware/auth.ts
var onRequest = async (context, next) => {
  const existingTokens = getExistingTokens(context);
  const usedTokens = existingTokens ?? await generateVisitorTokens();
  const store = {
    sdkHost: headlessNode.host({ req: context.request }),
    sessionTokens: usedTokens
  };
  const response = await authAsyncLocalStorage.run(store, async () => {
    return next();
  });
  const contentType = response.headers.get("Content-Type");
  if (contentType !== "text/html") {
    return response;
  }
  if (context.isPrerendered) {
    return response;
  }
  if (existingTokens === usedTokens) {
    return response;
  }
  saveSessionTokensToCookie(context, usedTokens);
  return response;
};
function getExistingTokens(context) {
  const existingTokens = getSessionTokensFromCookie(context);
  if (existingTokens && existingTokens.accessToken.expiresAt + 600 > Date.now() / 1e3) {
    return existingTokens;
  }
  return null;
}
export {
  onRequest
};
