// @wix/auto_sdk_members_authentication/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  RoleOriginal: () => Role,
  SiteMemberPrivacyStatusOriginal: () => SiteMemberPrivacyStatus,
  SiteMemberStatusOriginal: () => SiteMemberStatus,
  SortingOriginal: () => Sorting,
  SourceOriginal: () => Source,
  VerificationFailureReasonOriginal: () => VerificationFailureReason,
  approve: () => approve2,
  block: () => block2,
  changeLoginEmail: () => changeLoginEmail2,
  login: () => login2,
  register: () => register2,
  sendSetPasswordEmail: () => sendSetPasswordEmail2
});
module.exports = __toCommonJS(meta_exports);

// src/identity-members-v1-member-authentication.http.ts
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
function resolveWixMembersApiV1AuthServiceUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/wix-sm/api/v1/auth",
        destPath: ""
      },
      {
        srcPath: "/members/v1/wix-sm-webapp-proxy/v1/auth",
        destPath: "/api/v1/auth"
      },
      {
        srcPath: "/identity/members/v1/auth/members",
        destPath: "/v1/auth/members"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/identity/members/api/v1/auth/login",
        destPath: "/api/v1/auth/login"
      },
      {
        srcPath: "/identity/members/v1/auth/members",
        destPath: "/v1/auth/members"
      }
    ],
    _: [
      {
        srcPath: "/identity/members/api/v1/auth/login",
        destPath: "/api/v1/auth/login"
      },
      {
        srcPath: "/identity/members/v1/auth/members",
        destPath: "/v1/auth/members"
      }
    ]
  };
  return (0, import_rest_modules.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
function resolveWixMembersApiV1MembersServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/wix-sm-members-proxy",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/api/wix-sm/v1/members",
        destPath: "/api/v1/members"
      },
      {
        srcPath: "/_api/wix-sm-webapp/v1/members",
        destPath: "/api/v1/members"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/wix-sm-webapp/v1/members",
        destPath: "/api/v1/members"
      }
    ],
    _: [
      {
        srcPath: "/_api/wix-sm-webapp/v1/members",
        destPath: "/api/v1/members"
      },
      {
        srcPath: "/api/wix-sm/v1/members",
        destPath: "/api/v1/members"
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/wow/data-hub/api/wix-sm/v1/members",
        destPath: "/api/v1/members"
      },
      {
        srcPath: "/wow/bulx/api/wix-sm/v1/members",
        destPath: "/api/v1/members"
      },
      {
        srcPath: "/_api/wix-sm-webapp/v1/members",
        destPath: "/api/v1/members"
      },
      {
        srcPath: "/_api/wix-sm/v1/members",
        destPath: "/api/v1/members"
      },
      {
        srcPath: "/api/wix-sm/v1/members",
        destPath: "/api/v1/members"
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/wow/data-hub/api/wix-sm/v1/members",
        destPath: "/api/v1/members"
      },
      {
        srcPath: "/wow/bulx/api/wix-sm/v1/members",
        destPath: "/api/v1/members"
      },
      {
        srcPath: "/_api/wix-sm-webapp/v1/members",
        destPath: "/api/v1/members"
      },
      {
        srcPath: "/_api/wix-sm/v1/members",
        destPath: "/api/v1/members"
      },
      {
        srcPath: "/api/wix-sm/v1/members",
        destPath: "/api/v1/members"
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/wow/data-hub/api/wix-sm/v1/members",
        destPath: "/api/v1/members"
      },
      {
        srcPath: "/wow/bulx/api/wix-sm/v1/members",
        destPath: "/api/v1/members"
      },
      {
        srcPath: "/_api/wix-sm-webapp/v1/members",
        destPath: "/api/v1/members"
      },
      {
        srcPath: "/_api/wix-sm/v1/members",
        destPath: "/api/v1/members"
      },
      {
        srcPath: "/api/wix-sm/v1/members",
        destPath: "/api/v1/members"
      }
    ],
    "dev._base_domain_": [
      {
        srcPath: "/_api/wix-sm-webapp/v1/members",
        destPath: "/api/v1/members"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/wix-sm-webapp/v1/members",
        destPath: "/api/v1/members"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/wix-sm-webapp/v1/members",
        destPath: "/api/v1/members"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/wix-sm-webapp/v1/members",
        destPath: "/api/v1/members"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/api/wix-sm/v1/members",
        destPath: "/api/v1/members"
      }
    ],
    "forums._base_domain_": [
      {
        srcPath: "/_api/wix-sm-webapp/v1/members",
        destPath: "/api/v1/members"
      }
    ],
    "groups.wixapps.net": [
      {
        srcPath: "/_api/wix-sm-webapp/v1/members",
        destPath: "/api/v1/members"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/corvid/api/wix-sm/v1/members",
        destPath: "/api/v1/members"
      },
      {
        srcPath: "/_api/wix-sm-webapp/v1/members",
        destPath: "/api/v1/members"
      }
    ],
    "sslusers._base_domain_": [
      {
        srcPath: "/_api/wix-sm-webapp/v1/members",
        destPath: "/api/v1/members"
      },
      {
        srcPath: "/wix-sm/api/v1/members",
        destPath: "/api/v1/members"
      }
    ],
    "users._base_domain_": [
      {
        srcPath: "/_api/wix-sm-webapp/v1/members",
        destPath: "/api/v1/members"
      },
      {
        srcPath: "/wix-sm/api/v1/members",
        destPath: "/api/v1/members"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/blacklivesmatter/api/wix-sm/v1/members",
        destPath: "/api/v1/members"
      },
      {
        srcPath: "/corvid/api/wix-sm/v1/members",
        destPath: "/api/v1/members"
      },
      {
        srcPath: "/_api/wix-sm-webapp/v1/members",
        destPath: "/api/v1/members"
      },
      {
        srcPath: "/_api/wix-sm/v1/members",
        destPath: "/api/v1/members"
      },
      {
        srcPath: "/api/wix-sm/v1/members",
        destPath: "/api/v1/members"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/api/v1/members",
        destPath: "/api/v1/members"
      }
    ]
  };
  return (0, import_rest_modules.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_authentication";
function changeLoginEmail(payload) {
  function __changeLoginEmail({ host }) {
    const metadata = {
      entityFqdn: "wix.identity.members.v1.member",
      method: "PATCH",
      methodFqn: "wix.members.api.v1.MembersService.ChangeLoginEmail",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixMembersApiV1MembersServiceUrl({
        protoPath: "/api/v1/members/{id}/change-login-email",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "member.creationDate" },
            { path: "member.lastUpdateDate" },
            { path: "member.lastLoginDate" },
            { path: "member.customFields.dateValue" },
            { path: "member.picture.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "member.customFields.numValue" },
            { path: "member.picture.focalPoint.x" },
            { path: "member.picture.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __changeLoginEmail;
}
function approve(payload) {
  function __approve({ host }) {
    const metadata = {
      entityFqdn: "wix.identity.members.v1.member",
      method: "POST",
      methodFqn: "wix.members.api.v1.MembersService.Approve",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixMembersApiV1MembersServiceUrl({
        protoPath: "/api/v1/members/approve",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __approve;
}
function block(payload) {
  function __block({ host }) {
    const metadata = {
      entityFqdn: "wix.identity.members.v1.member",
      method: "POST",
      methodFqn: "wix.members.api.v1.MembersService.Block",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixMembersApiV1MembersServiceUrl({
        protoPath: "/api/v1/members/block",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __block;
}
function register(payload) {
  function __register({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [{ path: "contactInfo.customFields.numValue" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "contactInfo.customFields.dateValue" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.identity.members.v1.member",
      method: "POST",
      methodFqn: "wix.members.api.v1.AuthService.Register",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixMembersApiV1AuthServiceUrl({
        protoPath: "/api/v1/auth/register",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "member.creationDate" },
            { path: "member.lastUpdateDate" },
            { path: "member.lastLoginDate" },
            { path: "member.customFields.dateValue" },
            { path: "member.picture.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "member.customFields.numValue" },
            { path: "member.picture.focalPoint.x" },
            { path: "member.picture.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __register;
}
function login(payload) {
  function __login({ host }) {
    const metadata = {
      entityFqdn: "wix.identity.members.v1.member",
      method: "POST",
      methodFqn: "wix.members.api.v1.AuthService.Login",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixMembersApiV1AuthServiceUrl({
        protoPath: "/api/v1/auth/login",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "member.creationDate" },
            { path: "member.lastUpdateDate" },
            { path: "member.lastLoginDate" },
            { path: "member.customFields.dateValue" },
            { path: "member.picture.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "member.customFields.numValue" },
            { path: "member.picture.focalPoint.x" },
            { path: "member.picture.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __login;
}
function sendSetPasswordEmail(payload) {
  function __sendSetPasswordEmail({ host }) {
    const metadata = {
      entityFqdn: "wix.identity.members.v1.member",
      method: "POST",
      methodFqn: "wix.members.api.v1.AuthService.SendSetPasswordEmail",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixMembersApiV1AuthServiceUrl({
        protoPath: "/v1/auth/members/send-set-password-email",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __sendSetPasswordEmail;
}

// src/identity-members-v1-member-authentication.types.ts
var Role = /* @__PURE__ */ ((Role2) => {
  Role2["UNDEFINED_ROLE"] = "UNDEFINED_ROLE";
  Role2["MEMBER"] = "MEMBER";
  Role2["OWNER"] = "OWNER";
  Role2["CONTRIBUTOR"] = "CONTRIBUTOR";
  return Role2;
})(Role || {});
var SiteMemberPrivacyStatus = /* @__PURE__ */ ((SiteMemberPrivacyStatus2) => {
  SiteMemberPrivacyStatus2["UNDEFINED"] = "UNDEFINED";
  SiteMemberPrivacyStatus2["PUBLIC"] = "PUBLIC";
  SiteMemberPrivacyStatus2["PRIVATE"] = "PRIVATE";
  SiteMemberPrivacyStatus2["COMMUNITY"] = "COMMUNITY";
  return SiteMemberPrivacyStatus2;
})(SiteMemberPrivacyStatus || {});
var SiteMemberStatus = /* @__PURE__ */ ((SiteMemberStatus2) => {
  SiteMemberStatus2["UNDEFINED_STATUS"] = "UNDEFINED_STATUS";
  SiteMemberStatus2["APPLICANT"] = "APPLICANT";
  SiteMemberStatus2["ACTIVE"] = "ACTIVE";
  SiteMemberStatus2["INACTIVE"] = "INACTIVE";
  SiteMemberStatus2["BLOCKED"] = "BLOCKED";
  SiteMemberStatus2["OFFLINE_ONLY"] = "OFFLINE_ONLY";
  return SiteMemberStatus2;
})(SiteMemberStatus || {});
var Sorting = /* @__PURE__ */ ((Sorting2) => {
  Sorting2["DESC"] = "DESC";
  Sorting2["ASC"] = "ASC";
  return Sorting2;
})(Sorting || {});
var Source = /* @__PURE__ */ ((Source2) => {
  Source2["UNKNOWN"] = "UNKNOWN";
  Source2["HANDLING_SPAM"] = "HANDLING_SPAM";
  return Source2;
})(Source || {});
var VerificationFailureReason = /* @__PURE__ */ ((VerificationFailureReason2) => {
  VerificationFailureReason2["UNSPECIFIED"] = "UNSPECIFIED";
  VerificationFailureReason2["BAD_CODE"] = "BAD_CODE";
  VerificationFailureReason2["NOT_FOUND"] = "NOT_FOUND";
  VerificationFailureReason2["SEND_CODE_ERROR"] = "SEND_CODE_ERROR";
  return VerificationFailureReason2;
})(VerificationFailureReason || {});

// src/identity-members-v1-member-authentication.meta.ts
function changeLoginEmail2() {
  const payload = { id: ":id" };
  const getRequestOptions = changeLoginEmail(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/api/v1/members/{id}/change-login-email",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function approve2() {
  const payload = {};
  const getRequestOptions = approve(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/api/v1/members/approve",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function block2() {
  const payload = {};
  const getRequestOptions = block(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/api/v1/members/block",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function register2() {
  const payload = {};
  const getRequestOptions = register(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/api/v1/auth/register",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function login2() {
  const payload = {};
  const getRequestOptions = login(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/api/v1/auth/login",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function sendSetPasswordEmail2() {
  const payload = {};
  const getRequestOptions = sendSetPasswordEmail(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/auth/members/send-set-password-email",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  RoleOriginal,
  SiteMemberPrivacyStatusOriginal,
  SiteMemberStatusOriginal,
  SortingOriginal,
  SourceOriginal,
  VerificationFailureReasonOriginal,
  approve,
  block,
  changeLoginEmail,
  login,
  register,
  sendSetPasswordEmail
});
