// @wix/fast-gallery-core/dist/cjs/lib/common/window/viewModeWrapper.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.viewModeWrapper = exports.isSiteMode = exports.isSEOMode = exports.isPreviewMode = exports.isEditMode = exports.isDeviceTypeTouch = exports.isDeviceTypeTablet = exports.isDeviceTypeMobile = exports.isDeviceTypeDesktop = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _viewMode = _interopRequireDefault(require("../constants/viewMode"));
var _deviceType = _interopRequireDefault(require("../constants/deviceType"));
class ViewModeWrapper {
  constructor() {
    (0, _defineProperty2.default)(this, "_viewMode", void 0);
    (0, _defineProperty2.default)(this, "_deviceType", void 0);
    this.setViewMode = this.setViewMode.bind(this);
    this.isSiteMode = this.isSiteMode.bind(this);
    this.isEditMode = this.isEditMode.bind(this);
    this.isPreviewMode = this.isPreviewMode.bind(this);
    this.isSEOMode = this.isSEOMode.bind(this);
    this.setDeviceType = this.setDeviceType.bind(this);
    this.isDeviceTypeDesktop = this.isDeviceTypeDesktop.bind(this);
    this.isDeviceTypeMobile = this.isDeviceTypeMobile.bind(this);
    this.isDeviceTypeTablet = this.isDeviceTypeTablet.bind(this);
    this.isDeviceTypeTouch = this.isDeviceTypeTouch.bind(this);
    this._viewMode = _viewMode.default.SITE;
    this._deviceType = _deviceType.default.DESKTOP;
  }
  setViewMode(viewMode) {
    this._viewMode = viewMode;
  }
  setDeviceType(forceVal) {
    this._deviceType = forceVal;
  }
  isDeviceTypeMobile() {
    return this._deviceType === _deviceType.default.MOBILE;
  }
  isDeviceTypeTablet() {
    return this._deviceType === _deviceType.default.TABLET;
  }
  isDeviceTypeDesktop() {
    return this._deviceType === _deviceType.default.DESKTOP;
  }
  isDeviceTypeTouch() {
    return this.isDeviceTypeMobile() || this.isDeviceTypeTablet();
  }
  isSiteMode() {
    return this._viewMode === _viewMode.default.SITE;
  }
  isEditMode() {
    return this._viewMode === _viewMode.default.EDIT;
  }
  isPreviewMode() {
    return this._viewMode === _viewMode.default.PREVIEW;
  }
  isSEOMode() {
    return this._viewMode === _viewMode.default.SEO;
  }
}
const viewModeWrapper = exports.viewModeWrapper = new ViewModeWrapper();
const isSiteMode = exports.isSiteMode = viewModeWrapper.isSiteMode;
const isEditMode = exports.isEditMode = viewModeWrapper.isEditMode;
const isPreviewMode = exports.isPreviewMode = viewModeWrapper.isPreviewMode;
const isSEOMode = exports.isSEOMode = viewModeWrapper.isSEOMode;
const isDeviceTypeMobile = exports.isDeviceTypeMobile = viewModeWrapper.isDeviceTypeMobile;
const isDeviceTypeTablet = exports.isDeviceTypeTablet = viewModeWrapper.isDeviceTypeTablet;
const isDeviceTypeDesktop = exports.isDeviceTypeDesktop = viewModeWrapper.isDeviceTypeDesktop;
const isDeviceTypeTouch = exports.isDeviceTypeTouch = viewModeWrapper.isDeviceTypeTouch;
