// @wix/fast-gallery-core/dist/cjs/lib/common/window/window.mock.js
"use strict";

exports.__esModule = true;
exports.scrollMock = exports.hydrateMockMap = exports.dimsMock = exports.default = void 0;
const noop = () => ({});
const width = 2560;
const height = 1440;
const dimsMock = exports.dimsMock = {
  y: 0,
  x: 0,
  width,
  height,
  innerWidth: width,
  innerHeight: height,
  outerWidth: width,
  outerHeight: height,
  clientWidth: width,
  clientHeight: height
};
const elem = {
  ...dimsMock,
  getBoundingClientRect: () => dimsMock
};
const documentMock = {
  addEventListener: noop,
  removeEventListener: noop,
  createEvent: noop,
  getElementById: () => elem,
  getElementsByClassName: () => [elem],
  getElementsByTagName: () => [elem],
  querySelector: () => [elem],
  documentElement: elem,
  activeElement: elem,
  style: dimsMock,
  ...dimsMock
};
documentMock.body = documentMock;
const locationMock = {
  href: 'http://mock.wix.com/',
  protocol: 'http:',
  host: 'mock.wix.com',
  hostname: 'mock.wix.com',
  port: '',
  pathname: '/',
  search: '',
  hash: ''
};
const scrollMock = exports.scrollMock = {
  scrollTop: 0,
  scrollY: 0
};
const windowHydrateMock = {
  ...dimsMock,
  ...scrollMock
};
const windowMock = {
  isMock: true,
  isSSR: true,
  orientation: 0,
  devicePixelRatio: 1,
  addEventListener: noop,
  removeEventListener: noop,
  createEvent: noop,
  CustomEvent: noop,
  screen: dimsMock,
  open: noop,
  petri: {},
  search: {},
  location: locationMock,
  postMessage: noop,
  requestAnimationFrame: noop,
  dispatchEvent: noop,
  document: documentMock,
  getComputedStyle: noop,
  localStorage: {},
  frames: [],
  ...windowHydrateMock
};
const hydrateMockMap = exports.hydrateMockMap = new Map(Object.keys(windowHydrateMock).map(key => [key, windowHydrateMock[key]]));
windowMock.parent = windowMock;
var _default = exports.default = windowMock;
