// @wix/design-system/dist/cjs/Avatar/AvatarCore/AvatarCore.uni.driver.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.avatarCoreDriverFactory = void 0;
var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _unidriver = require("../../utils/test-utils/unidriver");
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2.default)(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var avatarCoreDriverFactory = base => {
  var byDataHook = dataHook => dataHook ? base.$("[data-hook=\"".concat(dataHook, "\"]")) : base;
  return _objectSpread(_objectSpread({}, (0, _unidriver.baseUniDriverFactory)(base)), {}, {
    getAttribute: (attrName, dataHook) => {
      if (dataHook) {
        return base.$("[data-hook=\"".concat(dataHook, "\"")).attr(attrName);
      }
      return base.attr(attrName);
    },
    getProperty: propName => base._prop(propName),
    getContentType: () => base.attr('data-content-type'),
    getTextContent: dataHook => byDataHook(dataHook).text(),
    hasClass: (className, dataHook) => {
      return byDataHook(dataHook || 'content').hasClass(className);
    },
    isImageLoaded: function () {
      var _isImageLoaded = (0, _asyncToGenerator2.default)(function* () {
        return (yield base.attr('data-img-loaded')) === 'true';
      });
      function isImageLoaded() {
        return _isImageLoaded.apply(this, arguments);
      }
      return isImageLoaded;
    }()
  });
};
exports.avatarCoreDriverFactory = avatarCoreDriverFactory;
