// @wix/bex-core/dist/cjs/util/showToastCompat.js
"use strict";

exports.__esModule = true;
exports.showToastCompat = showToastCompat;
exports.toastConfigMaps = void 0;
const toastConfigMaps = exports.toastConfigMaps = {
  priority: {
    LOW: 'low',
    HIGH: 'high',
    NORMAL: 'normal'
  },
  timeout: {
    NORMAL: 'normal',
    NONE: 'none'
  },
  type: {
    STANDARD: 'standard',
    ERROR: 'error',
    PREMIUM: 'premium',
    SUCCESS: 'success',
    WARNING: 'warning'
  },
  uiType: {
    BUTTON: 'button',
    LINK: 'link'
  }
};
function showToastCompat(showToast) {
  return ({
    message,
    type,
    timeout,
    priority,
    action,
    onCloseClick,
    onToastSeen,
    ...rest
  }) => {
    const showToastResult = showToast({
      message,
      priority: priority ? toastConfigMaps.priority[priority] : undefined,
      timeout: timeout ? toastConfigMaps.timeout[timeout] : undefined,
      type: type ? toastConfigMaps.type[type] : undefined,
      action: action && {
        text: action.text,
        onClick: action.onClick,
        removeToastOnClick: action.removeToastOnClick,
        uiType: action.uiType ? toastConfigMaps.uiType[action.uiType] : undefined
      },
      onCloseClick,
      onToastSeen,
      ...rest
    });
    return {
      remove: () => {
        (async () => {
          (await showToastResult).remove();
        })();
      }
    };
  };
}
