// @wix/auto_sdk_pricing-plans_plans/build/internal/es/index.typings.mjs
// src/pricing-plans-v2-plan-plans.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/pricing-plans-v2-plan-plans.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressMembershipV2PlansPlansServiceV2Url(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/plan-proxy",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "*.wixapps.net": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    _: [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "editor-flow.wixapps.net": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "forums._base_domain_": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "forums.wixapps.net": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      },
      {
        srcPath: "/_api/pricing-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/pricing-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_pricing-plans_plans";
function listPublicPlans(payload) {
  function __listPublicPlans({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "GET",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.ListPublicPlans",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans/public",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "plans.createdDate" },
            { path: "plans.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listPublicPlans;
}
function queryPublicPlans(payload) {
  function __queryPublicPlans({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.QueryPublicPlans",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans/public/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "plans.createdDate" },
            { path: "plans.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryPublicPlans;
}
function getPlan(payload) {
  function __getPlan({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "GET",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.GetPlan",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans/{id}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __getPlan;
}
function listPlans(payload) {
  function __listPlans({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "GET",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.ListPlans",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "plans.createdDate" },
            { path: "plans.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listPlans;
}
function getPlanStats(payload) {
  function __getPlanStats({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "GET",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.GetPlanStats",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans/stats",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getPlanStats;
}
function createPlan(payload) {
  function __createPlan({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.CreatePlan",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __createPlan;
}
function updatePlan(payload) {
  function __updatePlan({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "PATCH",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.UpdatePlan",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans/{plan.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __updatePlan;
}
function setPlanVisibility(payload) {
  function __setPlanVisibility({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "PUT",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.SetPlanVisibility",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans/{id}/visibility",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __setPlanVisibility;
}
function makePlanPrimary(payload) {
  function __makePlanPrimary({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.MakePlanPrimary",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans/{id}/make-primary",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __makePlanPrimary;
}
function clearPrimary(payload) {
  function __clearPrimary({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.ClearPrimary",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans/clear-primary",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __clearPrimary;
}
function archivePlan(payload) {
  function __archivePlan({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.ArchivePlan",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans/{id}/archive",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __archivePlan;
}

// src/pricing-plans-v2-plan-plans.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var PeriodUnit = /* @__PURE__ */ ((PeriodUnit2) => {
  PeriodUnit2["UNDEFINED"] = "UNDEFINED";
  PeriodUnit2["DAY"] = "DAY";
  PeriodUnit2["WEEK"] = "WEEK";
  PeriodUnit2["MONTH"] = "MONTH";
  PeriodUnit2["YEAR"] = "YEAR";
  return PeriodUnit2;
})(PeriodUnit || {});
var AppliedAt = /* @__PURE__ */ ((AppliedAt2) => {
  AppliedAt2["UNKNOWN_CHARGE_EVENT"] = "UNKNOWN_CHARGE_EVENT";
  AppliedAt2["FIRST_PAYMENT"] = "FIRST_PAYMENT";
  return AppliedAt2;
})(AppliedAt || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var ArchivedFilter = /* @__PURE__ */ ((ArchivedFilter2) => {
  ArchivedFilter2["ACTIVE"] = "ACTIVE";
  ArchivedFilter2["ARCHIVED"] = "ARCHIVED";
  ArchivedFilter2["ARCHIVED_AND_ACTIVE"] = "ARCHIVED_AND_ACTIVE";
  return ArchivedFilter2;
})(ArchivedFilter || {});
var PublicFilter = /* @__PURE__ */ ((PublicFilter2) => {
  PublicFilter2["PUBLIC_AND_HIDDEN"] = "PUBLIC_AND_HIDDEN";
  PublicFilter2["PUBLIC"] = "PUBLIC";
  PublicFilter2["HIDDEN"] = "HIDDEN";
  return PublicFilter2;
})(PublicFilter || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function listPublicPlans2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    limit: options?.limit,
    offset: options?.offset,
    planIds: options?.planIds
  });
  const reqOpts = listPublicPlans(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          limit: "$[0].limit",
          offset: "$[0].offset",
          planIds: "$[0].planIds"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryPublicPlans2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryPublicPlans(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [])
      );
      return {
        items: transformedData?.plans,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "OFFSET",
    transformationPaths: {}
  });
}
async function getPlan2(_id) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ id: _id });
  const reqOpts = getPlan(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.plan;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { id: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listPlans2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    archived: options?.archived,
    public: options?.public,
    limit: options?.limit,
    offset: options?.offset,
    planIds: options?.planIds
  });
  const reqOpts = listPlans(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          archived: "$[0].archived",
          public: "$[0].public",
          limit: "$[0].limit",
          offset: "$[0].offset",
          planIds: "$[0].planIds"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getPlanStats2() {
  const { httpClient, sideEffects } = arguments[0];
  const payload = renameKeysFromSDKRequestToRESTRequest({});
  const reqOpts = getPlanStats(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {},
        singleArgumentUnchanged: false
      },
      []
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function createPlan2(plan) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ plan });
  const reqOpts = createPlan(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.plan;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { plan: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["plan"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updatePlan2(_id, plan) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    plan: { ...plan, id: _id }
  });
  const reqOpts = updatePlan(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.plan;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { plan: "$[1]" },
        explicitPathsToArguments: { "plan.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "plan"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function setPlanVisibility2(_id, visible) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    id: _id,
    visible
  });
  const reqOpts = setPlanVisibility(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { id: "$[0]", visible: "$[1]" },
        singleArgumentUnchanged: false
      },
      ["_id", "visible"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function makePlanPrimary2(_id) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ id: _id });
  const reqOpts = makePlanPrimary(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { id: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function clearPrimary2() {
  const { httpClient, sideEffects } = arguments[0];
  const payload = renameKeysFromSDKRequestToRESTRequest({});
  const reqOpts = clearPrimary(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {},
        singleArgumentUnchanged: false
      },
      []
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function archivePlan2(_id) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ id: _id });
  const reqOpts = archivePlan(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { id: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
export {
  AppliedAt,
  ArchivedFilter,
  PeriodUnit,
  PublicFilter,
  SortOrder,
  WebhookIdentityType,
  archivePlan2 as archivePlan,
  clearPrimary2 as clearPrimary,
  createPlan2 as createPlan,
  getPlan2 as getPlan,
  getPlanStats2 as getPlanStats,
  listPlans2 as listPlans,
  listPublicPlans2 as listPublicPlans,
  makePlanPrimary2 as makePlanPrimary,
  queryPublicPlans2 as queryPublicPlans,
  setPlanVisibility2 as setPlanVisibility,
  updatePlan2 as updatePlan
};
