// @wix/auto_sdk_pricing-plans_plans/build/internal/es/meta.mjs
// src/pricing-plans-v2-plan-plans.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressMembershipV2PlansPlansServiceV2Url(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/plan-proxy",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "*.wixapps.net": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    _: [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "editor-flow.wixapps.net": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "forums._base_domain_": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "forums.wixapps.net": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      },
      {
        srcPath: "/_api/pricing-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/pricing-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_pricing-plans_plans";
function listPublicPlans(payload) {
  function __listPublicPlans({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "GET",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.ListPublicPlans",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans/public",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "plans.createdDate" },
            { path: "plans.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listPublicPlans;
}
function queryPublicPlans(payload) {
  function __queryPublicPlans({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.QueryPublicPlans",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans/public/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "plans.createdDate" },
            { path: "plans.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryPublicPlans;
}
function getPlan(payload) {
  function __getPlan({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "GET",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.GetPlan",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans/{id}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __getPlan;
}
function listPlans(payload) {
  function __listPlans({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "GET",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.ListPlans",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "plans.createdDate" },
            { path: "plans.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listPlans;
}
function getPlanStats(payload) {
  function __getPlanStats({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "GET",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.GetPlanStats",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans/stats",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getPlanStats;
}
function createPlan(payload) {
  function __createPlan({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.CreatePlan",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __createPlan;
}
function updatePlan(payload) {
  function __updatePlan({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "PATCH",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.UpdatePlan",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans/{plan.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __updatePlan;
}
function setPlanVisibility(payload) {
  function __setPlanVisibility({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "PUT",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.SetPlanVisibility",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans/{id}/visibility",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __setPlanVisibility;
}
function makePlanPrimary(payload) {
  function __makePlanPrimary({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.MakePlanPrimary",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans/{id}/make-primary",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __makePlanPrimary;
}
function clearPrimary(payload) {
  function __clearPrimary({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.ClearPrimary",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans/clear-primary",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __clearPrimary;
}
function archivePlan(payload) {
  function __archivePlan({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.ArchivePlan",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans/{id}/archive",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __archivePlan;
}

// src/pricing-plans-v2-plan-plans.types.ts
var PeriodUnit = /* @__PURE__ */ ((PeriodUnit2) => {
  PeriodUnit2["UNDEFINED"] = "UNDEFINED";
  PeriodUnit2["DAY"] = "DAY";
  PeriodUnit2["WEEK"] = "WEEK";
  PeriodUnit2["MONTH"] = "MONTH";
  PeriodUnit2["YEAR"] = "YEAR";
  return PeriodUnit2;
})(PeriodUnit || {});
var AppliedAt = /* @__PURE__ */ ((AppliedAt2) => {
  AppliedAt2["UNKNOWN_CHARGE_EVENT"] = "UNKNOWN_CHARGE_EVENT";
  AppliedAt2["FIRST_PAYMENT"] = "FIRST_PAYMENT";
  return AppliedAt2;
})(AppliedAt || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var ArchivedFilter = /* @__PURE__ */ ((ArchivedFilter2) => {
  ArchivedFilter2["ACTIVE"] = "ACTIVE";
  ArchivedFilter2["ARCHIVED"] = "ARCHIVED";
  ArchivedFilter2["ARCHIVED_AND_ACTIVE"] = "ARCHIVED_AND_ACTIVE";
  return ArchivedFilter2;
})(ArchivedFilter || {});
var PublicFilter = /* @__PURE__ */ ((PublicFilter2) => {
  PublicFilter2["PUBLIC_AND_HIDDEN"] = "PUBLIC_AND_HIDDEN";
  PublicFilter2["PUBLIC"] = "PUBLIC";
  PublicFilter2["HIDDEN"] = "HIDDEN";
  return PublicFilter2;
})(PublicFilter || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/pricing-plans-v2-plan-plans.meta.ts
function listPublicPlans2() {
  const payload = {};
  const getRequestOptions = listPublicPlans(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/plans/public",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryPublicPlans2() {
  const payload = {};
  const getRequestOptions = queryPublicPlans(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/plans/public/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getPlan2() {
  const payload = { id: ":id" };
  const getRequestOptions = getPlan(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/plans/{id}",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listPlans2() {
  const payload = {};
  const getRequestOptions = listPlans(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/plans",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getPlanStats2() {
  const payload = {};
  const getRequestOptions = getPlanStats(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/plans/stats",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createPlan2() {
  const payload = {};
  const getRequestOptions = createPlan(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/plans",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updatePlan2() {
  const payload = { plan: { id: ":planId" } };
  const getRequestOptions = updatePlan(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v2/plans/{plan.id}",
    pathParams: { planId: "planId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function setPlanVisibility2() {
  const payload = { id: ":id" };
  const getRequestOptions = setPlanVisibility(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PUT",
    path: "/v2/plans/{id}/visibility",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function makePlanPrimary2() {
  const payload = { id: ":id" };
  const getRequestOptions = makePlanPrimary(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/plans/{id}/make-primary",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function clearPrimary2() {
  const payload = {};
  const getRequestOptions = clearPrimary(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/plans/clear-primary",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function archivePlan2() {
  const payload = { id: ":id" };
  const getRequestOptions = archivePlan(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/plans/{id}/archive",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  AppliedAt as AppliedAtOriginal,
  ArchivedFilter as ArchivedFilterOriginal,
  PeriodUnit as PeriodUnitOriginal,
  PublicFilter as PublicFilterOriginal,
  SortOrder as SortOrderOriginal,
  WebhookIdentityType as WebhookIdentityTypeOriginal,
  archivePlan2 as archivePlan,
  clearPrimary2 as clearPrimary,
  createPlan2 as createPlan,
  getPlan2 as getPlan,
  getPlanStats2 as getPlanStats,
  listPlans2 as listPlans,
  listPublicPlans2 as listPublicPlans,
  makePlanPrimary2 as makePlanPrimary,
  queryPublicPlans2 as queryPublicPlans,
  setPlanVisibility2 as setPlanVisibility,
  updatePlan2 as updatePlan
};
