// react-module-container/dist/src/demo/EventListener.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.EventsListener = void 0;
var _inheritsLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/inheritsLoose"));
var _react = _interopRequireDefault(require("react"));
var _moduleRegistry = _interopRequireDefault(require("../module-registry"));
var EventsListener = exports.EventsListener = /*#__PURE__*/function (_React$Component) {
  (0, _inheritsLoose2["default"])(EventsListener, _React$Component);
  function EventsListener(props) {
    var _this;
    _this = _React$Component.call(this, props) || this;
    _this.state = {
      gotStartLoadingEvent: false,
      gotComponentReady: false,
      gotComponentWillUnmount: false
    };
    return _this;
  }
  var _proto = EventsListener.prototype;
  _proto.UNSAFE_componentWillMount = function UNSAFE_componentWillMount() {
    var _this2 = this;
    this.unSubscribeStartLoading = _moduleRegistry["default"].addListener('reactModuleContainer.componentStartLoading', function () {
      _this2.setState({
        gotStartLoadingEvent: true
      });
    });
    this.unSubscribeComponentReady = _moduleRegistry["default"].addListener('reactModuleContainer.componentReady', function () {
      _this2.setState({
        gotComponentReady: true
      });
    });
    this.unSubscribeComponentWillUnmount = _moduleRegistry["default"].addListener('reactModuleContainer.componentWillUnmount', function () {
      _this2.setState({
        gotComponentWillUnmount: true
      });
    });
  };
  _proto.componentWillUnmount = function componentWillUnmount() {
    this.unSubscribeStartLoading.remove();
    this.unSubscribeComponentReady.remove();
    this.unSubscribeComponentWillUnmount.remove();
  };
  _proto.render = function render() {
    return /*#__PURE__*/_react["default"].createElement("div", null, /*#__PURE__*/_react["default"].createElement("div", null, "gotStartLoadingEvent:", /*#__PURE__*/_react["default"].createElement("span", {
      id: "got-start-loading"
    }, this.state.gotStartLoadingEvent ? 'true' : 'false')), /*#__PURE__*/_react["default"].createElement("div", null, "gotComponentReady: ", /*#__PURE__*/_react["default"].createElement("span", {
      id: "got-component-ready"
    }, this.state.gotComponentReady ? 'true' : 'false')), /*#__PURE__*/_react["default"].createElement("div", null, "gotComponentWillUnmount:", /*#__PURE__*/_react["default"].createElement("span", {
      id: "got-component-will-unmount"
    }, this.state.gotComponentWillUnmount ? 'true' : 'false')));
  };
  return EventsListener;
}(_react["default"].Component);
