// react-module-container/dist/src/demo/module.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.MyReactCompCrossOrigin = exports.MyReactComp8 = exports.MyReactComp7 = exports.MyReactComp = exports.MyNgComp5 = exports.MyNgComp4 = exports.MyNgComp2 = exports.MyNgComp = exports.MyNgApp5NoUnloadCss = void 0;
var _inheritsLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/inheritsLoose"));
var _reactRouter = require("react-router");
/* global React, AngularLazyComponent, ReactLazyComponent, ModuleRegistry */
var MyNgComp = exports.MyNgComp = /*#__PURE__*/function (_AngularLazyComponent) {
  (0, _inheritsLoose2["default"])(MyNgComp, _AngularLazyComponent);
  function MyNgComp(props) {
    return _AngularLazyComponent.call(this, props, {
      files: [props.topology.staticsUrl + "angular-module.bundle.js"],
      module: 'myApp',
      component: 'my-comp'
    }) || this;
  }
  return MyNgComp;
}(AngularLazyComponent);
var MyNgComp2 = exports.MyNgComp2 = /*#__PURE__*/function (_AngularLazyComponent2) {
  (0, _inheritsLoose2["default"])(MyNgComp2, _AngularLazyComponent2);
  function MyNgComp2(props) {
    return _AngularLazyComponent2.call(this, props, {
      files: [props.topology.staticsUrl + "angular-module.bundle.js"],
      resolve: function resolve() {
        var experimentsPromise = Promise.resolve({
          'specs.fed.ReactModuleContainerWithResolve': true
        });
        var customDataPromise = Promise.resolve({
          user: 'xiw@wix.com'
        });
        return Promise.all([experimentsPromise, customDataPromise]).then(function (results) {
          return {
            experiments: results[0],
            customData: results[1]
          };
        });
      },
      prepare: function prepare() {
        window.__prepare_was_called__ = true;
      },
      module: 'myApp2',
      component: 'my-comp'
    }) || this;
  }
  return MyNgComp2;
}(AngularLazyComponent);
var MyReactComp = exports.MyReactComp = /*#__PURE__*/function (_ReactLazyComponent) {
  (0, _inheritsLoose2["default"])(MyReactComp, _ReactLazyComponent);
  function MyReactComp(props) {
    return _ReactLazyComponent.call(this, props, {
      files: [props.topology.staticsUrl + "react-module.bundle.js"],
      resolve: function resolve() {
        var experimentsPromise = Promise.resolve({
          'specs.fed.ReactModuleContainerWithResolve': true
        });
        var customDataPromise = Promise.resolve({
          user: 'xiw@wix.com'
        });
        return Promise.all([experimentsPromise, customDataPromise]).then(function (results) {
          return {
            experiments: results[0],
            customData: results[1]
          };
        });
      },
      prepare: function prepare() {
        window.__prepare_was_called__ = true;
      },
      component: 'MyApp3.RealReactComp'
    }) || this;
  }
  return MyReactComp;
}(ReactLazyComponent);
var MyReactCompCrossOrigin = exports.MyReactCompCrossOrigin = /*#__PURE__*/function (_ReactLazyComponent2) {
  (0, _inheritsLoose2["default"])(MyReactCompCrossOrigin, _ReactLazyComponent2);
  function MyReactCompCrossOrigin(props) {
    return _ReactLazyComponent2.call(this, props, {
      files: [props.topology.staticsUrl + "react-module.bundle.js"],
      crossorigin: true,
      component: 'MyApp6.RealReactCompCrossOrigin'
    }) || this;
  }
  return MyReactCompCrossOrigin;
}(ReactLazyComponent);
var Hello = /*#__PURE__*/function (_React$Component) {
  (0, _inheritsLoose2["default"])(Hello, _React$Component);
  function Hello(props) {
    var _this;
    _this = _React$Component.call(this, props) || this;
    _this.state = {
      counter: 0
    };
    return _this;
  }
  var _proto = Hello.prototype;
  _proto.handleClick = function handleClick() {
    this.setState({
      counter: this.state.counter + 1
    });
  };
  _proto.render = function render() {
    var _this2 = this;
    return /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("div", {
      onClick: function onClick() {
        return _this2.handleClick();
      }
    }, /*#__PURE__*/React.createElement("div", null, "React Counter (click me): ", /*#__PURE__*/React.createElement("span", {
      id: "counter"
    }, this.state.counter), "!!!"), /*#__PURE__*/React.createElement("div", {
      id: "value-in-react"
    }, this.props.value)), /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement(_reactRouter.Link, {
      className: "react-link",
      to: "/ng-router-app/a"
    }, "ng-route-app"), "\xA0", /*#__PURE__*/React.createElement(_reactRouter.Link, {
      className: "react-link",
      to: "/ui-router-app/"
    }, "ui-route-app"), "\xA0"));
  };
  return Hello;
}(React.Component);
var MyNgComp4 = exports.MyNgComp4 = /*#__PURE__*/function (_AngularLazyComponent3) {
  (0, _inheritsLoose2["default"])(MyNgComp4, _AngularLazyComponent3);
  function MyNgComp4(props) {
    return _AngularLazyComponent3.call(this, props, {
      files: [props.topology.staticsUrl + "angular-module.bundle.js", props.topology.baseUrl + "demo-shared.css", props.topology.baseUrl + "demo-4.css"],
      module: 'myApp4',
      component: 'my-comp'
    }) || this;
  }
  return MyNgComp4;
}(AngularLazyComponent);
var MyNgComp5 = exports.MyNgComp5 = /*#__PURE__*/function (_AngularLazyComponent4) {
  (0, _inheritsLoose2["default"])(MyNgComp5, _AngularLazyComponent4);
  function MyNgComp5(props) {
    return _AngularLazyComponent4.call(this, props, {
      unloadStylesOnDestroy: true,
      files: [props.topology.staticsUrl + "angular-module.bundle.js", props.topology.baseUrl + "demo-shared.css", props.topology.baseUrl + "demo-5.css"],
      module: 'myApp5',
      component: 'my-comp'
    }) || this;
  }
  return MyNgComp5;
}(AngularLazyComponent);
var MyNgApp5NoUnloadCss = exports.MyNgApp5NoUnloadCss = /*#__PURE__*/function (_MyNgComp) {
  (0, _inheritsLoose2["default"])(MyNgApp5NoUnloadCss, _MyNgComp);
  function MyNgApp5NoUnloadCss(props) {
    var _this3;
    _this3 = _MyNgComp.call(this, props) || this;
    _this3.manifest.unloadStylesOnDestroy = false;
    return _this3;
  }
  return MyNgApp5NoUnloadCss;
}(MyNgComp5);
var MyReactComp7 = exports.MyReactComp7 = /*#__PURE__*/function (_ReactLazyComponent3) {
  (0, _inheritsLoose2["default"])(MyReactComp7, _ReactLazyComponent3);
  function MyReactComp7(props) {
    return _ReactLazyComponent3.call(this, props, {
      files: [props.topology.staticsUrl + "react-module.bundle.js", props.topology.baseUrl + "demo-shared.css", props.topology.baseUrl + "demo-4.css"],
      component: 'MyApp7.RealReactComp'
    }) || this;
  }
  return MyReactComp7;
}(ReactLazyComponent);
var MyReactComp8 = exports.MyReactComp8 = /*#__PURE__*/function (_ReactLazyComponent4) {
  (0, _inheritsLoose2["default"])(MyReactComp8, _ReactLazyComponent4);
  function MyReactComp8(props) {
    return _ReactLazyComponent4.call(this, props, {
      files: [props.topology.staticsUrl + "react-module.bundle.js", props.topology.baseUrl + "demo-shared.css", props.topology.baseUrl + "demo-5.css"],
      component: 'MyApp7.RealReactComp',
      unloadStylesOnDestroy: false
    }) || this;
  }
  return MyReactComp8;
}(ReactLazyComponent);
ModuleRegistry.registerComponent('MyApp.MyNgComp', function () {
  return MyNgComp;
});
ModuleRegistry.registerComponent('MyApp2.MyNgComp', function () {
  return MyNgComp2;
});
ModuleRegistry.registerComponent('MyApp3.MyReactComp', function () {
  return MyReactComp;
});
ModuleRegistry.registerComponent('Hello', function () {
  return Hello;
});
ModuleRegistry.registerComponent('MyApp4.MyNgComp', function () {
  return MyNgComp4;
});
ModuleRegistry.registerComponent('MyApp5.MyNgComp', function () {
  return MyNgComp5;
});
ModuleRegistry.registerComponent('MyApp5NoUnloadCss.MyNgComp', function () {
  return MyNgApp5NoUnloadCss;
});
ModuleRegistry.registerComponent('MyApp6.MyReactCompCrossOrigin', function () {
  return MyReactCompCrossOrigin;
});
ModuleRegistry.registerComponent('MyApp7.MyReactComp', function () {
  return MyReactComp7;
});
ModuleRegistry.registerComponent('MyApp8.MyReactComp', function () {
  return MyReactComp8;
});
