// @wix/patterns/dist/cjs/components/CustomFieldModal/CustomFieldModal.uni.driver.js
"use strict";

exports.__esModule = true;
exports.CustomFieldModalUniDriver = CustomFieldModalUniDriver;
var _unidriver = require("@wix/design-system/dist/testkit/unidriver");
var _unidriver2 = require("../../unidriver");
function CustomFieldModalUniDriver(base, body) {
  const getCustomFieldModal = () => (0, _unidriver.MessageModalLayoutUniDriver)(body.$('[data-hook="custom-field-modal"]'), body);
  const getReachedMaxLimitModal = () => (0, _unidriver.MessageModalLayoutUniDriver)(body.$('[data-hook="collection-action-modal"]'), body);
  const getSettingsButton = () => (0, _unidriver.TextButtonUniDriver)(body.$('[data-hook="custom-field-modal-settings-button"]'), body);
  const getErrorStateNotification = () => (0, _unidriver.FloatingNotificationUniDriver)(body.$('[data-hook="custom-field-modal-error-state"]'), body);
  const getLoader = () => (0, _unidriver.LoaderUniDriver)(body.$('[data-hook="custom-field-modal-status-submitting"]'), body);
  const getFieldKeyTooltip = () => (0, _unidriver.TextUniDriver)(body.$('[data-hook="custom-field-modal-fieldKey-tooltip"]'), body);
  const exists = () => getCustomFieldModal().exists();
  const clickConfirm = () => getCustomFieldModal().getPrimaryButtonDriver().then(driver => driver.click());
  const clickClose = () => getCustomFieldModal().clickCloseButton();
  const clickCancel = () => getCustomFieldModal().getSecondaryButtonDriver().then(driver => driver.click());
  const clickAdvancedSettings = () => getSettingsButton().click();
  const isSubmitting = () => getLoader().exists();
  const getFormTypeDriver = () => (0, _unidriver.DropdownUniDriver)(body.$('[data-hook="custom-field-modal-type"]'), body);
  const getFormNameDriver = () => (0, _unidriver.InputUniDriver)(body.$('[data-hook="custom-field-modal-name"]'), body);
  const getFormKeyDriver = () => (0, _unidriver.InputUniDriver)(body.$('[data-hook="custom-field-modal-key"]'), body);
  const getInstalledAppsSection = () => body.$('[data-hook="custom-field-modal-apps-section"]');
  const getSiteVisitorsSection = () => body.$('[data-hook="custom-field-modal-users-section"]');
  const getFormReadApps = () => (0, _unidriver.CheckboxUniDriver)(body.$('[data-hook="custom-field-modal-read-apps"]'), body);
  const getFormWriteApps = () => (0, _unidriver.CheckboxUniDriver)(body.$('[data-hook="custom-field-modal-write-apps"]'), body);
  const getFormReadUsers = () => (0, _unidriver.CheckboxUniDriver)(body.$('[data-hook="custom-field-modal-read-users"]'), body);
  const getFormWriteUsers = () => (0, _unidriver.CheckboxUniDriver)(body.$('[data-hook="custom-field-modal-write-users"]'), body);
  const getTypeFormFieldDriver = () => (0, _unidriver.FormFieldUniDriver)(body.$('[data-hook="custom-field-modal-field-type"]'), body);
  const getNameFormFieldDriver = () => (0, _unidriver.FormFieldUniDriver)(body.$('[data-hook="custom-field-modal-field-name"]'), body);
  const getKeyFormFieldDriver = () => (0, _unidriver.FormFieldUniDriver)(body.$('[data-hook="custom-field-modal-field-key"]'), body);
  const getSubtitle = () => {
    return (0, _unidriver.TextUniDriver)(body.$('[data-hook="baseModalLayout-subtitle"]'), body);
  };
  const getPermissionNote = () => {
    return (0, _unidriver.TextUniDriver)(body.$('[data-hook="permissions-note"]'), body);
  };
  const getEditPiiNote = () => {
    return (0, _unidriver.TextUniDriver)(body.$('[data-hook="pii-toggle-edit-note"]'), body);
  };
  const getPiiToggleText = () => {
    return (0, _unidriver.TextUniDriver)(body.$('[data-hook="pii-toggle-name"]'), body);
  };
  const getPiiFootnote = () => {
    return (0, _unidriver.TextUniDriver)(body.$('[data-hook="pii-footnote"]'), body);
  };
  const getPiiToggleSwitch = () => {
    return (0, _unidriver.ToggleSwitchUniDriver)(body.$('[data-hook="pii-toggle-switch"]'), body);
  };
  const reachedMaxLimitModalExists = () => {
    return getReachedMaxLimitModal().exists();
  };
  const clickReachedMaxPrimaryButton = () => getReachedMaxLimitModal().getPrimaryButtonDriver().then(driver => driver.click());
  const getFormDrivers = () => ({
    getFormTypeDriver,
    getFormNameDriver,
    getFormKeyDriver,
    getFormReadApps,
    getFormWriteApps,
    getFormReadUsers,
    getFormWriteUsers,
    getTypeFormFieldDriver,
    getNameFormFieldDriver,
    getKeyFormFieldDriver,
    getFieldKeyTooltip,
    getSubtitle,
    getPermissionNote,
    getEditPiiNote,
    getPiiToggleText,
    getPiiFootnote,
    getPiiToggleSwitch
  });
  const fillInCustomFieldForm = async ({
    type,
    name,
    key,
    permissions
  }) => {
    if (type) {
      await getFormTypeDriver().dropdownLayoutDriver.clickAtOptionByDataHook(`custom-field-modal-type-${type}`);
    }
    if (name) {
      await getFormNameDriver().enterText(name);
    }
    if (key) {
      await getFormKeyDriver().enterText(key);
    }
    if (permissions) {
      var _permissions$read, _permissions$write, _permissions$read2, _permissions$write2;
      [[getFormReadApps(), (permissions == null || (_permissions$read = permissions.read) == null ? void 0 : _permissions$read.includes('apps')) ?? false], [getFormWriteApps(), (permissions == null || (_permissions$write = permissions.write) == null ? void 0 : _permissions$write.includes('apps')) ?? false], [getFormReadUsers(), (permissions == null || (_permissions$read2 = permissions.read) == null ? void 0 : _permissions$read2.includes('users')) ?? false], [getFormWriteUsers(), (permissions == null || (_permissions$write2 = permissions.write) == null ? void 0 : _permissions$write2.includes('users')) ?? false]].forEach(async ([driver, value]) => {
        const isChecked = await driver.isChecked();
        if (value && !isChecked || !value && isChecked) {
          await driver.click();
        }
      });
    }
  };
  const getCustomFieldFormValues = async () => {
    return {
      type: await getFormTypeDriver().inputDriver.getValue(),
      name: await getFormNameDriver().getValue(),
      key: await getFormKeyDriver().getValue(),
      permissions: {
        read: {
          apps: await getFormReadApps().isChecked(),
          users: await getFormReadUsers().isChecked()
        },
        write: {
          apps: await getFormWriteApps().isChecked(),
          users: await getFormWriteUsers().isChecked()
        }
      }
    };
  };
  return {
    ...(0, _unidriver2.baseUniDriverFactory)(base),
    exists,
    clickConfirm,
    clickClose,
    clickCancel,
    clickAdvancedSettings,
    isInstalledAppsSectionVisible: () => getInstalledAppsSection().exists(),
    isSiteVisitorsSectionVisible: () => getSiteVisitorsSection().exists(),
    fillInCustomFieldForm,
    getCustomFieldFormValues,
    getFormDrivers,
    getErrorStateNotification,
    isSubmitting,
    reachedMaxLimitModalExists,
    clickReachedMaxPrimaryButton
  };
}
