// prosemirror-dev-tools/dist/cjs/tabs/schema.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = SchemaTab;
exports.postprocessValue = postprocessValue;
var _react = _interopRequireDefault(require("react"));
var _splitView = require("../components/split-view");
var _jsonTree = _interopRequireDefault(require("../components/json-tree"));
var _heading = require("../components/heading");
var _jotai = require("jotai");
var _schema = require("../state/schema");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var ignoreFields = ["schema", "contentExpr", "schema", "parseDOM", "toDOM"];
function postprocessValue(ignore, data) {
  if (!data || Object.prototype.toString.call(data) !== "[object Object]") {
    return data;
  }
  return Object.keys(data).filter(function (key) {
    return ignore.indexOf(key) === -1;
  }).reduce(function (res, key) {
    res[key] = data[key];
    return res;
  }, {});
}
function SchemaTab() {
  var schema = (0, _jotai.useAtomValue)(_schema.schemaAtom);
  if (!schema) return null;
  return /*#__PURE__*/_react["default"].createElement(_splitView.SplitView, {
    testId: "__prosemirror_devtools_tabs_schema__"
  }, /*#__PURE__*/_react["default"].createElement(_splitView.SplitViewCol, {
    grow: true
  }, /*#__PURE__*/_react["default"].createElement(_heading.Heading, null, "Nodes"), /*#__PURE__*/_react["default"].createElement(_jsonTree["default"], {
    data: schema.nodes,
    postprocessValue: postprocessValue.bind(null, ignoreFields)
  })), /*#__PURE__*/_react["default"].createElement(_splitView.SplitViewCol, {
    grow: true,
    sep: true
  }, /*#__PURE__*/_react["default"].createElement(_heading.Heading, null, "Marks"), /*#__PURE__*/_react["default"].createElement(_jsonTree["default"], {
    data: schema.marks,
    postprocessValue: postprocessValue.bind(null, ignoreFields)
  })));
}
