// @wix/bex-utils/@wix/ambassador-data-extensions-v1-data-extension-schema/build/cjs/builders.impl.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.aWebhookIdentityType = exports.aIdentificationData = exports.aMessageEnvelope = exports.aVersionStatus = exports.aAppVersionArchived = exports.aAppVersionDraftChanged = exports.aAppVersionApproved = exports.aAppVersionReleased = exports.aAppVersionDeclined = exports.aAppVersionPublished = exports.aAppVersionSubmitted = exports.aAppVersionCreated = exports.aAppVersionStateChanged = exports.aEmpty = exports.aActionEvent = exports.aEntityDeletedEvent = exports.aEntityUpdatedEvent = exports.aRestoreInfo = exports.aEntityCreatedEvent = exports.aDomainEvent = exports.aDeleteByWhiteListedMetaSiteResponse = exports.aDeleteByWhiteListedMetaSiteRequest = exports.aDeleteGlobalExtensionSchemaResponse = exports.aDeleteGlobalExtensionSchemaRequest = exports.aDeleteDemoDataExtensionSchemaResponse = exports.aDeleteDemoDataExtensionSchemaRequest = exports.aUpsertDataExtensionGlobalSchemaResponse = exports.aUpsertDataExtensionGlobalSchemaRequest = exports.aListDataExtensionSchemasResponse = exports.aRequestedField = exports.aListDataExtensionSchemasRequest = exports.aUpdateDataExtensionSchemaResponse = exports.aUpdateDataExtensionSchemaRequest = exports.aCreateDataExtensionSchemaResponse = exports.aCreateDataExtensionSchemaRequest = exports.aReindexField = exports.aReindexEvent = exports.aDataExtensionSchema = void 0;
var metro_common_builders_1 = require("@wix/metro-common-builders");
var types_impl_1 = require("./types.impl");
// @ts-ignore
var callsite_1 = __importDefault(require("callsite"));
function aDataExtensionSchema(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        id: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'FORMAT', value: 'GUID' },
                { type: 'READ_ONLY', value: 'true' },
                { type: 'IMMUTABLE', value: 'true' },
            ],
        }),
        fqdn: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'IMMUTABLE', value: 'true' },
                { type: 'MAX_LENGTH', value: '255' },
            ],
        }),
        namespace: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'IMMUTABLE', value: 'true' },
                { type: 'MAX_LENGTH', value: '164' },
            ],
        }),
        jsonSchema: {},
        updatedDate: (0, metro_common_builders_1.someDate)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        createdDate: (0, metro_common_builders_1.someDate)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        revision: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'DECIMAL_VALUE' },
            ],
        }),
        appDefId: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'FORMAT', value: 'GUID' },
                { type: 'READ_ONLY', value: 'true' },
            ],
        }),
        maxLimitBytes: (0, metro_common_builders_1.someNumber)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        currentSizeBytes: (0, metro_common_builders_1.someNumber)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        extensionPoint: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'IMMUTABLE', value: 'true' },
                { type: 'MAX_LENGTH', value: '20' },
            ],
        }),
    }, overrideObject);
}
exports.aDataExtensionSchema = aDataExtensionSchema;
function aReindexEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        fqdn: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '255' }],
        }),
        fields: [
            aReindexField(),
            aReindexField(),
            aReindexField(),
            aReindexField(),
            aReindexField(),
            aReindexField(),
            aReindexField(),
            aReindexField(),
            aReindexField(),
            aReindexField(),
        ],
    }, overrideObject);
}
exports.aReindexEvent = aReindexEvent;
function aReindexField(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        fieldPath: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '2000' }],
        }),
        reindexSince: (0, metro_common_builders_1.someDate)(),
    }, overrideObject);
}
exports.aReindexField = aReindexField;
function aCreateDataExtensionSchemaRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        dataExtensionSchema: aDataExtensionSchema(),
    }, overrideObject);
}
exports.aCreateDataExtensionSchemaRequest = aCreateDataExtensionSchemaRequest;
function aCreateDataExtensionSchemaResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        dataExtensionSchema: aDataExtensionSchema(),
    }, overrideObject);
}
exports.aCreateDataExtensionSchemaResponse = aCreateDataExtensionSchemaResponse;
function aUpdateDataExtensionSchemaRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        dataExtensionSchema: aDataExtensionSchema(),
    }, overrideObject);
}
exports.aUpdateDataExtensionSchemaRequest = aUpdateDataExtensionSchemaRequest;
function aUpdateDataExtensionSchemaResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        dataExtensionSchema: aDataExtensionSchema(),
    }, overrideObject);
}
exports.aUpdateDataExtensionSchemaResponse = aUpdateDataExtensionSchemaResponse;
function aListDataExtensionSchemasRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        fqdn: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '255' }],
        }),
        namespaces: [
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '164' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '164' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '164' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '164' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '164' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '164' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '164' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '164' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '164' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '164' },
                ],
            }),
        ],
        fields: [
            aRequestedField(),
            aRequestedField(),
            aRequestedField(),
            aRequestedField(),
            aRequestedField(),
        ],
        extensionPoints: [
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '20' },
                    { type: 'MAX_LENGTH', value: '10' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '20' },
                    { type: 'MAX_LENGTH', value: '10' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '20' },
                    { type: 'MAX_LENGTH', value: '10' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '20' },
                    { type: 'MAX_LENGTH', value: '10' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '20' },
                    { type: 'MAX_LENGTH', value: '10' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '20' },
                    { type: 'MAX_LENGTH', value: '10' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '20' },
                    { type: 'MAX_LENGTH', value: '10' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '20' },
                    { type: 'MAX_LENGTH', value: '10' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '20' },
                    { type: 'MAX_LENGTH', value: '10' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '20' },
                    { type: 'MAX_LENGTH', value: '10' },
                ],
            }),
        ],
    }, overrideObject);
}
exports.aListDataExtensionSchemasRequest = aListDataExtensionSchemasRequest;
function aRequestedField(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.RequestedField.UNKNOWN_REQUESTED_FIELD; }
    return enumFirstItem;
}
exports.aRequestedField = aRequestedField;
function aListDataExtensionSchemasResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        dataExtensionSchemas: [aDataExtensionSchema()],
    }, overrideObject);
}
exports.aListDataExtensionSchemasResponse = aListDataExtensionSchemasResponse;
function aUpsertDataExtensionGlobalSchemaRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        dataExtensionSchema: aDataExtensionSchema(),
    }, overrideObject);
}
exports.aUpsertDataExtensionGlobalSchemaRequest = aUpsertDataExtensionGlobalSchemaRequest;
function aUpsertDataExtensionGlobalSchemaResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        dataExtensionSchema: aDataExtensionSchema(),
    }, overrideObject);
}
exports.aUpsertDataExtensionGlobalSchemaResponse = aUpsertDataExtensionGlobalSchemaResponse;
function aDeleteDemoDataExtensionSchemaRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        dataExtensionSchemaId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
exports.aDeleteDemoDataExtensionSchemaRequest = aDeleteDemoDataExtensionSchemaRequest;
function aDeleteDemoDataExtensionSchemaResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({}, overrideObject);
}
exports.aDeleteDemoDataExtensionSchemaResponse = aDeleteDemoDataExtensionSchemaResponse;
function aDeleteGlobalExtensionSchemaRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        fqdn: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '255' }],
        }),
        namespace: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '164' }],
        }),
    }, overrideObject);
}
exports.aDeleteGlobalExtensionSchemaRequest = aDeleteGlobalExtensionSchemaRequest;
function aDeleteGlobalExtensionSchemaResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({}, overrideObject);
}
exports.aDeleteGlobalExtensionSchemaResponse = aDeleteGlobalExtensionSchemaResponse;
function aDeleteByWhiteListedMetaSiteRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        metaSiteId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
exports.aDeleteByWhiteListedMetaSiteRequest = aDeleteByWhiteListedMetaSiteRequest;
function aDeleteByWhiteListedMetaSiteResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        hasMore: true,
    }, overrideObject);
}
exports.aDeleteByWhiteListedMetaSiteResponse = aDeleteByWhiteListedMetaSiteResponse;
function aDomainEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        createdEvent: aEntityCreatedEvent(),
        id: (0, metro_common_builders_1.someString)(),
        entityFqdn: (0, metro_common_builders_1.someString)(),
        slug: (0, metro_common_builders_1.someString)(),
        entityId: (0, metro_common_builders_1.someString)(),
        eventTime: (0, metro_common_builders_1.someDate)(),
        triggeredByAnonymizeRequest: true,
        originatedFrom: (0, metro_common_builders_1.someString)(),
        entityEventSequence: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'DECIMAL_VALUE' }],
        }),
    }, overrideObject);
}
exports.aDomainEvent = aDomainEvent;
function aEntityCreatedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        entityAsJson: (0, metro_common_builders_1.someString)(),
        triggeredByUndelete: true,
        restoreInfo: aRestoreInfo(),
        additionalMetadataAsJson: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aEntityCreatedEvent = aEntityCreatedEvent;
function aRestoreInfo(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        deletedDate: (0, metro_common_builders_1.someDate)(),
    }, overrideObject);
}
exports.aRestoreInfo = aRestoreInfo;
function aEntityUpdatedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        currentEntityAsJson: (0, metro_common_builders_1.someString)(),
        previousEntityAsJson: (0, metro_common_builders_1.someString)(),
        modifiedFields: { modifiedFields: (0, metro_common_builders_1.someAny)() },
        additionalMetadataAsJson: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aEntityUpdatedEvent = aEntityUpdatedEvent;
function aEntityDeletedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        movedToTrash: true,
        deletedEntityAsJson: (0, metro_common_builders_1.someString)(),
        additionalMetadataAsJson: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aEntityDeletedEvent = aEntityDeletedEvent;
function aActionEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        bodyAsJson: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aActionEvent = aActionEvent;
function aEmpty(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({}, overrideObject);
}
exports.aEmpty = aEmpty;
function aAppVersionStateChanged(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        created: aAppVersionCreated(),
        appId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        version: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        previousState: aVersionStatus(),
    }, overrideObject);
}
exports.aAppVersionStateChanged = aAppVersionStateChanged;
function aAppVersionCreated(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({}, overrideObject);
}
exports.aAppVersionCreated = aAppVersionCreated;
function aAppVersionSubmitted(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({}, overrideObject);
}
exports.aAppVersionSubmitted = aAppVersionSubmitted;
function aAppVersionPublished(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({}, overrideObject);
}
exports.aAppVersionPublished = aAppVersionPublished;
function aAppVersionDeclined(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({}, overrideObject);
}
exports.aAppVersionDeclined = aAppVersionDeclined;
function aAppVersionReleased(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({}, overrideObject);
}
exports.aAppVersionReleased = aAppVersionReleased;
function aAppVersionApproved(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({}, overrideObject);
}
exports.aAppVersionApproved = aAppVersionApproved;
function aAppVersionDraftChanged(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({}, overrideObject);
}
exports.aAppVersionDraftChanged = aAppVersionDraftChanged;
function aAppVersionArchived(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({}, overrideObject);
}
exports.aAppVersionArchived = aAppVersionArchived;
function aVersionStatus(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.VersionStatus.NONE_STATUS; }
    return enumFirstItem;
}
exports.aVersionStatus = aVersionStatus;
function aMessageEnvelope(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        instanceId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        eventType: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '150' }],
        }),
        identity: aIdentificationData(),
        data: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aMessageEnvelope = aMessageEnvelope;
function aIdentificationData(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        anonymousVisitorId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        identityType: aWebhookIdentityType(),
    }, overrideObject);
}
exports.aIdentificationData = aIdentificationData;
function aWebhookIdentityType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.WebhookIdentityType.UNKNOWN; }
    return enumFirstItem;
}
exports.aWebhookIdentityType = aWebhookIdentityType;
function isCurrentBuilderWasPreviouslyCalled() {
    var callChainFunctionNames = (0, callsite_1.default)().map(function (item) {
        return item.getFunctionName();
    });
    var callChainWithoutCurrentBuilder = callChainFunctionNames.splice(2);
    var currentBuilderName = callChainFunctionNames[1];
    return callChainWithoutCurrentBuilder.some(function (functionName) { return functionName === currentBuilderName; });
}
__exportStar(require("@wix/metro-common-builders"), exports);
