// @wix/bex-utils/@wix/ambassador-data-extensions-v1-data-extension-schema/build/cjs/meta.impl.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.deleteByWhiteListedMetaSite = exports.listDataExtensionSchemas = exports.updateDataExtensionSchema = exports.createDataExtensionSchema = void 0;
var httpWixDataExtensionsV1DataExtensionSchema = __importStar(require("./http.impl"));
function createDataExtensionSchema() {
    var payload = {};
    var getRequestOptions = httpWixDataExtensionsV1DataExtensionSchema.createDataExtensionSchema(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/schemas',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createDataExtensionSchema = createDataExtensionSchema;
function updateDataExtensionSchema() {
    var payload = {};
    var getRequestOptions = httpWixDataExtensionsV1DataExtensionSchema.updateDataExtensionSchema(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'PUT',
        path: '/v1/schemas',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateDataExtensionSchema = updateDataExtensionSchema;
function listDataExtensionSchemas() {
    var payload = {};
    var getRequestOptions = httpWixDataExtensionsV1DataExtensionSchema.listDataExtensionSchemas(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/v1/schemas',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listDataExtensionSchemas = listDataExtensionSchemas;
function deleteByWhiteListedMetaSite() {
    var payload = {};
    var getRequestOptions = httpWixDataExtensionsV1DataExtensionSchema.deleteByWhiteListedMetaSite(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/schemas/delete-by-white-listed-meta-site',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteByWhiteListedMetaSite = deleteByWhiteListedMetaSite;
