// @wix/bex-utils/dist/cjs/PromiseState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.PromiseState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _reactive = require("./reactive");
class PromiseState {
  constructor(params) {
    (0, _defineProperty2.default)(this, "status", (0, _reactive.reactive)({
      type: 'idle'
    }));
    (0, _defineProperty2.default)(this, "onError", void 0);
    (0, _defineProperty2.default)(this, "fn", void 0);
    this.fn = params.fn;
    this.onError = params.onError;
  }
  async once() {
    const {
      status: {
        value: {
          type,
          promise
        }
      }
    } = this;
    if (type === 'loading') {
      await promise;
      return;
    }
    if (type === 'success') {
      return;
    }
    return this.run();
  }
  async run() {
    const {
      fn,
      status,
      onError
    } = this;
    const promise = fn();
    status.value = {
      type: 'loading',
      promise
    };
    try {
      const data = await promise;
      if (promise === status.value.promise) {
        status.value = {
          type: 'success',
          data,
          promise
        };
      }
    } catch (error) {
      onError == null || onError(error);
      if (promise === status.value.promise) {
        status.value = {
          type: 'error',
          error
        };
      }
    }
  }
}
exports.PromiseState = PromiseState;
