// @wix/restaurants/build/internal/cjs/index.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  availabilityExceptions: () => availabilityExceptions,
  catalogs: () => catalogs,
  discount: () => discount,
  fulfillemtMethods: () => fulfillemtMethods,
  fulfillmentMethods: () => fulfillmentMethods,
  itemLabels: () => itemLabels,
  itemModifierGroups: () => itemModifierGroups,
  itemModifiers: () => itemModifiers,
  itemVariants: () => itemVariants,
  items: () => items,
  localDeliverySpi: () => localDeliverySpi,
  menuOrderingSettings: () => menuOrderingSettings,
  menus: () => menus,
  operationGroups: () => operationGroups,
  operations: () => operations,
  orderPacingOrder: () => orderPacingOrder,
  orderPacingPolicy: () => orderPacingPolicy,
  orderSettings: () => orderSettings,
  orders: () => orders,
  posTpa: () => posTpa,
  recipients: () => recipients,
  restaurantLocations: () => restaurantLocations,
  sections: () => sections,
  serviceFees: () => serviceFees
});
module.exports = __toCommonJS(index_exports);
var localDeliverySpi = __toESM(require("@wix/auto_sdk_restaurants_local-delivery-spi"));
var catalogs = __toESM(require("@wix/auto_sdk_restaurants_catalogs"));
var discount = __toESM(require("@wix/auto_sdk_restaurants_discount"));
var fulfillemtMethods = __toESM(require("@wix/auto_sdk_restaurants_fulfillemt-methods"));
var menuOrderingSettings = __toESM(require("@wix/auto_sdk_restaurants_menu-ordering-settings"));
var items = __toESM(require("@wix/auto_sdk_restaurants_items"));
var itemLabels = __toESM(require("@wix/auto_sdk_restaurants_item-labels"));
var itemModifiers = __toESM(require("@wix/auto_sdk_restaurants_item-modifiers"));
var itemModifierGroups = __toESM(require("@wix/auto_sdk_restaurants_item-modifier-groups"));
var itemVariants = __toESM(require("@wix/auto_sdk_restaurants_item-variants"));
var menus = __toESM(require("@wix/auto_sdk_restaurants_menus"));
var sections = __toESM(require("@wix/auto_sdk_restaurants_sections"));
var operations = __toESM(require("@wix/auto_sdk_restaurants_operations"));
var orderPacingOrder = __toESM(require("@wix/auto_sdk_restaurants_order-pacing-order"));
var orderPacingPolicy = __toESM(require("@wix/auto_sdk_restaurants_order-pacing-policy"));
var orderSettings = __toESM(require("@wix/auto_sdk_restaurants_order-settings"));
var posTpa = __toESM(require("@wix/auto_sdk_restaurants_pos-tpa"));
var recipients = __toESM(require("@wix/auto_sdk_restaurants_recipients"));
var restaurantLocations = __toESM(require("@wix/auto_sdk_restaurants_restaurant-locations"));
var availabilityExceptions = __toESM(require("@wix/auto_sdk_restaurants_availability-exceptions"));
var fulfillmentMethods = __toESM(require("@wix/auto_sdk_restaurants_fulfillment-methods"));
var operationGroups = __toESM(require("@wix/auto_sdk_restaurants_operation-groups"));
var orders = __toESM(require("@wix/auto_sdk_restaurants_orders"));
var serviceFees = __toESM(require("@wix/auto_sdk_restaurants_service-fees"));
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  availabilityExceptions,
  catalogs,
  discount,
  fulfillemtMethods,
  fulfillmentMethods,
  itemLabels,
  itemModifierGroups,
  itemModifiers,
  itemVariants,
  items,
  localDeliverySpi,
  menuOrderingSettings,
  menus,
  operationGroups,
  operations,
  orderPacingOrder,
  orderPacingPolicy,
  orderSettings,
  orders,
  posTpa,
  recipients,
  restaurantLocations,
  sections,
  serviceFees
});
