// @wix/patterns/dist/cjs/components/AppliedFiltersTagList/AppliedFiltersTagList.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.AppliedFiltersTagList = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireWildcard(require("react"));
var _designSystem = require("@wix/design-system");
var _mobxReactLite = require("mobx-react-lite");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/AppliedFiltersTagList/AppliedFiltersTagList.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _AppliedFiltersTagList(props) {
  const {
    state,
    onClearButtonClicked,
    ...rest
  } = props;
  const {
    table,
    tags
  } = state;
  const {
    collection
  } = table;
  const {
    translate: t
  } = collection;
  const tagsMap = (0, _react.useMemo)(() => new Map(tags.map(tag => [tag.id, tag])), [tags]);
  return /*#__PURE__*/_react.default.createElement(_designSystem.TagList, (0, _extends2.default)({
    dataHook: "active-filter-tags",
    tags: tags,
    onTagRemove: tagId => {
      const tag = tagsMap.get(tagId);
      if (tag != null) {
        const {
          values,
          filter,
          filterProps
        } = tag;
        state._reportFilterToggleBiOnNextResult(filter, values);
        filter.remove(values, {
          emitEvents: ['change', 'refresh']
        });
        values.forEach(value => {
          filterProps == null || filterProps.onAppliedFilterTagRemove == null || filterProps.onAppliedFilterTagRemove(value);
        });
      }
    },
    size: "small",
    maxVisibleTags: Infinity,
    actionButton: {
      label: t('cairo.clear'),
      onClick: () => {
        state.onClearButtonClick();
        onClearButtonClicked == null || onClearButtonClicked();
      }
    }
  }, rest, {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 28,
      columnNumber: 5
    }
  }));
}
const AppliedFiltersTagList = exports.AppliedFiltersTagList = (0, _mobxReactLite.observer)(_AppliedFiltersTagList);
