// @wix/patterns/dist/cjs/components/AppliedFiltersTagList/FilterTagContent.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.FilterTagContent = FilterTagContent;
var _react = _interopRequireDefault(require("react"));
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/AppliedFiltersTagList/FilterTagContent.tsx";
function FilterTagContent({
  filterProps,
  values
}) {
  var _filterProps$toolbarT, _filterProps$accordio;
  const filterName = (filterProps == null || (_filterProps$toolbarT = filterProps.toolbarTagProps) == null ? void 0 : _filterProps$toolbarT.label) ?? (filterProps == null || (_filterProps$accordio = filterProps.accordionItemProps) == null ? void 0 : _filterProps$accordio.label);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, filterName && /*#__PURE__*/_react.default.createElement("span", {
    style: {
      fontWeight: 530
    },
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 19,
      columnNumber: 9
    }
  }, `${filterName}: `), values.map(({
    filter,
    value
  }) => filter.itemName(value)).join(', '));
}
