// @wix/auto_sdk_online-programs_sections/build/internal/es/meta.mjs
// src/online-programs-v3-section-sections.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKFloatToRESTFloat } from "@wix/sdk-runtime/transformations/float";
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixOnlineprogramsSectionsV3SectionsServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/program-sections-service",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/program-sections-service",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/program-sections-service",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/program-sections-service",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/program-sections-service",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/program-sections-service",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/program-sections-service",
        destPath: ""
      },
      {
        srcPath: "/program-sections-service",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/program-sections-service",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/program-sections-service",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/program-sections-service",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/program-sections-service",
        destPath: ""
      },
      {
        srcPath: "/program-sections-service",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/program-sections-service",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/program-sections-service",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_online-programs_sections";
function createSection(payload) {
  function __createSection({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "section.createdDate" },
          { path: "section.updatedDate" },
          { path: "section.description.image.urlExpirationDate" },
          { path: "section.description.video.urlExpirationDate" },
          { path: "section.description.video.resolutions.urlExpirationDate" },
          {
            path: "section.description.video.resolutions.poster.urlExpirationDate"
          },
          { path: "section.description.video.posters.urlExpirationDate" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "section.ordering" },
          { path: "section.description.image.focalPoint.x" },
          { path: "section.description.image.focalPoint.y" },
          { path: "section.description.video.resolutions.poster.focalPoint.x" },
          { path: "section.description.video.resolutions.poster.focalPoint.y" },
          { path: "section.description.video.posters.focalPoint.x" },
          { path: "section.description.video.posters.focalPoint.y" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "POST",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.CreateSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/sections",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "section.createdDate" },
            { path: "section.updatedDate" },
            { path: "section.description.image.urlExpirationDate" },
            { path: "section.description.video.urlExpirationDate" },
            {
              path: "section.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "section.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "section.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "section.ordering" },
            { path: "section.description.image.focalPoint.x" },
            { path: "section.description.image.focalPoint.y" },
            {
              path: "section.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "section.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "section.description.video.posters.focalPoint.x" },
            { path: "section.description.video.posters.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createSection;
}
function bulkCreateSection(payload) {
  function __bulkCreateSection({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "sections.createdDate" },
          { path: "sections.updatedDate" },
          { path: "sections.description.image.urlExpirationDate" },
          { path: "sections.description.video.urlExpirationDate" },
          { path: "sections.description.video.resolutions.urlExpirationDate" },
          {
            path: "sections.description.video.resolutions.poster.urlExpirationDate"
          },
          { path: "sections.description.video.posters.urlExpirationDate" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "sections.ordering" },
          { path: "sections.description.image.focalPoint.x" },
          { path: "sections.description.image.focalPoint.y" },
          {
            path: "sections.description.video.resolutions.poster.focalPoint.x"
          },
          {
            path: "sections.description.video.resolutions.poster.focalPoint.y"
          },
          { path: "sections.description.video.posters.focalPoint.x" },
          { path: "sections.description.video.posters.focalPoint.y" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "POST",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.BulkCreateSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/bulk/sections/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.description.image.urlExpirationDate" },
            { path: "results.item.description.video.urlExpirationDate" },
            {
              path: "results.item.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.description.video.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.description.video.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.ordering" },
            { path: "results.item.description.image.focalPoint.x" },
            { path: "results.item.description.image.focalPoint.y" },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "results.item.description.video.posters.focalPoint.x" },
            { path: "results.item.description.video.posters.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateSection;
}
function bulkCreateSectionMigration(payload) {
  function __bulkCreateSectionMigration({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "sections.createdDate" },
          { path: "sections.updatedDate" },
          { path: "sections.description.image.urlExpirationDate" },
          { path: "sections.description.video.urlExpirationDate" },
          { path: "sections.description.video.resolutions.urlExpirationDate" },
          {
            path: "sections.description.video.resolutions.poster.urlExpirationDate"
          },
          { path: "sections.description.video.posters.urlExpirationDate" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "sections.ordering" },
          { path: "sections.description.image.focalPoint.x" },
          { path: "sections.description.image.focalPoint.y" },
          {
            path: "sections.description.video.resolutions.poster.focalPoint.x"
          },
          {
            path: "sections.description.video.resolutions.poster.focalPoint.y"
          },
          { path: "sections.description.video.posters.focalPoint.x" },
          { path: "sections.description.video.posters.focalPoint.y" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "POST",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.BulkCreateSectionMigration",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/bulk/sections/create/migration",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.description.image.urlExpirationDate" },
            { path: "results.item.description.video.urlExpirationDate" },
            {
              path: "results.item.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.description.video.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.description.video.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.ordering" },
            { path: "results.item.description.image.focalPoint.x" },
            { path: "results.item.description.image.focalPoint.y" },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "results.item.description.video.posters.focalPoint.x" },
            { path: "results.item.description.video.posters.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateSectionMigration;
}
function getSection(payload) {
  function __getSection({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "GET",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.GetSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/sections/{sectionId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "section.createdDate" },
            { path: "section.updatedDate" },
            { path: "section.description.image.urlExpirationDate" },
            { path: "section.description.video.urlExpirationDate" },
            {
              path: "section.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "section.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "section.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "section.ordering" },
            { path: "section.description.image.focalPoint.x" },
            { path: "section.description.image.focalPoint.y" },
            {
              path: "section.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "section.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "section.description.video.posters.focalPoint.x" },
            { path: "section.description.video.posters.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getSection;
}
function updateSection(payload) {
  function __updateSection({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "section.createdDate" },
          { path: "section.updatedDate" },
          { path: "section.description.image.urlExpirationDate" },
          { path: "section.description.video.urlExpirationDate" },
          { path: "section.description.video.resolutions.urlExpirationDate" },
          {
            path: "section.description.video.resolutions.poster.urlExpirationDate"
          },
          { path: "section.description.video.posters.urlExpirationDate" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "section.ordering" },
          { path: "section.description.image.focalPoint.x" },
          { path: "section.description.image.focalPoint.y" },
          { path: "section.description.video.resolutions.poster.focalPoint.x" },
          { path: "section.description.video.resolutions.poster.focalPoint.y" },
          { path: "section.description.video.posters.focalPoint.x" },
          { path: "section.description.video.posters.focalPoint.y" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "PATCH",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.UpdateSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/sections/{section.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "section.createdDate" },
            { path: "section.updatedDate" },
            { path: "section.description.image.urlExpirationDate" },
            { path: "section.description.video.urlExpirationDate" },
            {
              path: "section.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "section.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "section.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "section.ordering" },
            { path: "section.description.image.focalPoint.x" },
            { path: "section.description.image.focalPoint.y" },
            {
              path: "section.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "section.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "section.description.video.posters.focalPoint.x" },
            { path: "section.description.video.posters.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateSection;
}
function deleteSection(payload) {
  function __deleteSection({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "DELETE",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.DeleteSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/sections/{sectionId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteSection;
}
function querySections(payload) {
  function __querySections({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "POST",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.QuerySections",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/sections/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "sections.createdDate" },
            { path: "sections.updatedDate" },
            { path: "sections.description.image.urlExpirationDate" },
            { path: "sections.description.video.urlExpirationDate" },
            {
              path: "sections.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "sections.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "sections.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "sections.ordering" },
            { path: "sections.description.image.focalPoint.x" },
            { path: "sections.description.image.focalPoint.y" },
            {
              path: "sections.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "sections.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "sections.description.video.posters.focalPoint.x" },
            { path: "sections.description.video.posters.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __querySections;
}
function listSections(payload) {
  function __listSections({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "GET",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.ListSections",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/sections",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "sections.createdDate" },
            { path: "sections.updatedDate" },
            { path: "sections.description.image.urlExpirationDate" },
            { path: "sections.description.video.urlExpirationDate" },
            {
              path: "sections.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "sections.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "sections.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "sections.ordering" },
            { path: "sections.description.image.focalPoint.x" },
            { path: "sections.description.image.focalPoint.y" },
            {
              path: "sections.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "sections.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "sections.description.video.posters.focalPoint.x" },
            { path: "sections.description.video.posters.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listSections;
}
function cloneSection(payload) {
  function __cloneSection({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "POST",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.CloneSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/sections/{sectionId}/clone",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "section.createdDate" },
            { path: "section.updatedDate" },
            { path: "section.description.image.urlExpirationDate" },
            { path: "section.description.video.urlExpirationDate" },
            {
              path: "section.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "section.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "section.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "section.ordering" },
            { path: "section.description.image.focalPoint.x" },
            { path: "section.description.image.focalPoint.y" },
            {
              path: "section.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "section.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "section.description.video.posters.focalPoint.x" },
            { path: "section.description.video.posters.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __cloneSection;
}
function moveSection(payload) {
  function __moveSection({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "PATCH",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.MoveSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/sections/move",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "section.createdDate" },
            { path: "section.updatedDate" },
            { path: "section.description.image.urlExpirationDate" },
            { path: "section.description.video.urlExpirationDate" },
            {
              path: "section.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "section.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "section.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "section.ordering" },
            { path: "section.description.image.focalPoint.x" },
            { path: "section.description.image.focalPoint.y" },
            {
              path: "section.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "section.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "section.description.video.posters.focalPoint.x" },
            { path: "section.description.video.posters.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __moveSection;
}
function publishSection(payload) {
  function __publishSection({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "PATCH",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.PublishSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/sections/publish",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "section.createdDate" },
            { path: "section.updatedDate" },
            { path: "section.description.image.urlExpirationDate" },
            { path: "section.description.video.urlExpirationDate" },
            {
              path: "section.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "section.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "section.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "section.ordering" },
            { path: "section.description.image.focalPoint.x" },
            { path: "section.description.image.focalPoint.y" },
            {
              path: "section.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "section.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "section.description.video.posters.focalPoint.x" },
            { path: "section.description.video.posters.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __publishSection;
}

// src/online-programs-v3-section-sections.types.ts
var State = /* @__PURE__ */ ((State2) => {
  State2["UNKNOWN_STATE"] = "UNKNOWN_STATE";
  State2["DRAFT"] = "DRAFT";
  State2["PUBLISHED"] = "PUBLISHED";
  return State2;
})(State || {});
var Enum = /* @__PURE__ */ ((Enum2) => {
  Enum2["UNKNOWN"] = "UNKNOWN";
  Enum2["PROTECTED"] = "PROTECTED";
  Enum2["PRIVATE"] = "PRIVATE";
  Enum2["PUBLIC"] = "PUBLIC";
  Enum2["UNCHANGED"] = "UNCHANGED";
  Enum2["GROUP_PROTECTED"] = "GROUP_PROTECTED";
  return Enum2;
})(Enum || {});
var RequestedFields = /* @__PURE__ */ ((RequestedFields2) => {
  RequestedFields2["UNKNOWN_REQUESTED_FIELD"] = "UNKNOWN_REQUESTED_FIELD";
  RequestedFields2["TOTAL_STEPS"] = "TOTAL_STEPS";
  return RequestedFields2;
})(RequestedFields || {});
var BulkActionType = /* @__PURE__ */ ((BulkActionType2) => {
  BulkActionType2["UNKNOWN_ACTION_TYPE"] = "UNKNOWN_ACTION_TYPE";
  BulkActionType2["INSERT"] = "INSERT";
  BulkActionType2["UPDATE"] = "UPDATE";
  BulkActionType2["DELETE"] = "DELETE";
  return BulkActionType2;
})(BulkActionType || {});
var DescriptionFieldSet = /* @__PURE__ */ ((DescriptionFieldSet2) => {
  DescriptionFieldSet2["STANDARD"] = "STANDARD";
  DescriptionFieldSet2["EXTENDED"] = "EXTENDED";
  return DescriptionFieldSet2;
})(DescriptionFieldSet || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var SearchIndexingNotificationState = /* @__PURE__ */ ((SearchIndexingNotificationState2) => {
  SearchIndexingNotificationState2["Unknown"] = "Unknown";
  SearchIndexingNotificationState2["Off"] = "Off";
  SearchIndexingNotificationState2["On"] = "On";
  return SearchIndexingNotificationState2;
})(SearchIndexingNotificationState || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/online-programs-v3-section-sections.meta.ts
function createSection2() {
  const payload = {};
  const getRequestOptions = createSection(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/sections",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkCreateSection2() {
  const payload = {};
  const getRequestOptions = bulkCreateSection(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/bulk/sections/create",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkCreateSectionMigration2() {
  const payload = {};
  const getRequestOptions = bulkCreateSectionMigration(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/bulk/sections/create/migration",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getSection2() {
  const payload = { sectionId: ":sectionId" };
  const getRequestOptions = getSection(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/sections/{sectionId}",
    pathParams: { sectionId: "sectionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateSection2() {
  const payload = { section: { id: ":sectionId" } };
  const getRequestOptions = updateSection(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v3/sections/{section.id}",
    pathParams: { sectionId: "sectionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteSection2() {
  const payload = { sectionId: ":sectionId" };
  const getRequestOptions = deleteSection(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v3/sections/{sectionId}",
    pathParams: { sectionId: "sectionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function querySections2() {
  const payload = {};
  const getRequestOptions = querySections(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/sections/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listSections2() {
  const payload = {};
  const getRequestOptions = listSections(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/sections",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function cloneSection2() {
  const payload = { sectionId: ":sectionId" };
  const getRequestOptions = cloneSection(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/sections/{sectionId}/clone",
    pathParams: { sectionId: "sectionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function moveSection2() {
  const payload = {};
  const getRequestOptions = moveSection(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v3/sections/move",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function publishSection2() {
  const payload = {};
  const getRequestOptions = publishSection(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v3/sections/publish",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  BulkActionType as BulkActionTypeOriginal,
  DescriptionFieldSet as DescriptionFieldSetOriginal,
  Enum as EnumOriginal,
  RequestedFields as RequestedFieldsOriginal,
  SearchIndexingNotificationState as SearchIndexingNotificationStateOriginal,
  SortOrder as SortOrderOriginal,
  State as StateOriginal,
  WebhookIdentityType as WebhookIdentityTypeOriginal,
  bulkCreateSection2 as bulkCreateSection,
  bulkCreateSectionMigration2 as bulkCreateSectionMigration,
  cloneSection2 as cloneSection,
  createSection2 as createSection,
  deleteSection2 as deleteSection,
  getSection2 as getSection,
  listSections2 as listSections,
  moveSection2 as moveSection,
  publishSection2 as publishSection,
  querySections2 as querySections,
  updateSection2 as updateSection
};
