// @wix/design-system/dist/cjs/CircularProgressBar/CircularProgressBarCore/Arc.js
"use strict";

exports.__esModule = true;
exports.Arc = void 0;
var React = _interopRequireWildcard(require("react"));
var _jsxFileName = "/home/builduser/work/57e038ea7326c1ec/packages/wix-design-system/dist/cjs/CircularProgressBar/CircularProgressBarCore/Arc.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function _interopRequireWildcard(e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (var _t in e) "default" !== _t && {}.hasOwnProperty.call(e, _t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, _t)) && (i.get || i.set) ? o(f, _t, i) : f[_t] = e[_t]); return f; })(e, t); }
var arc = (r, value) => {
  value /= 100;
  var angle = 2 * Math.PI * (value - 1 / 4);
  var x = r * Math.cos(angle);
  var y = r * Math.sin(angle);
  var sweep = Math.round(value);
  return "M .01 ".concat(-r, " A ").concat(r, " ").concat(r, " 0 ").concat(sweep, " 1 ").concat(x, " ").concat(y);
};
var Arc = props => {
  var {
    value,
    strokeWidth,
    size,
    className
  } = props;
  var viewBox = "".concat(-size / 2, " ").concat(-size / 2, " ").concat(size, " ").concat(size);
  var d = arc((size - strokeWidth) / 2, value);
  return /*#__PURE__*/React.createElement("svg", {
    className: className,
    width: size,
    height: size,
    viewBox: viewBox,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 25,
      columnNumber: 5
    }
  }, /*#__PURE__*/React.createElement("path", {
    d: d,
    strokeWidth: value !== 0 ? strokeWidth : 0,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 26,
      columnNumber: 7
    }
  }));
};
exports.Arc = Arc;
Arc.displayName = 'Arc';
