// @wix/bex-core/dist/esm/services/AccountPreferencesMergeStrategy.js
import merge from 'lodash/merge';
export class AccountPreferencesMergeStrategy {
    constructor(params) {
        this.storage = params.storage;
    }
    extendScope(scope) {
        return scope;
    }
    async setItem(key, value, options) {
        const { storage } = this;
        const data = (await this.getItem(key, options).catch(() => undefined));
        const partialData = value;
        if (data?.views?.list && partialData?.views?.list) {
            for (const [key, value] of Object.entries(partialData.views.list)) {
                if (value === null) {
                    // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
                    delete data.views.list[key];
                    // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
                    delete partialData.views.list[key];
                }
            }
        }
        return storage.setItem(key, merge(data, value), options);
    }
    async removeItem(key, options) {
        const { storage } = this;
        const data = await this.getItem(key, options).catch(() => undefined);
        return storage.setItem(key, merge(data, { [key]: null }), options);
    }
    async getItem(key, options) {
        return this.storage.getItem(key, options);
    }
    async getAllItems(options) {
        return this.storage.getAllItems(options);
    }
}
