// @wix/bex-core/dist/esm/services/AppSettingsCapsule.js
import { getGlobalSettings, setGlobalSettings, } from '@wix/bex-utils/@wix/ambassador-app-settings-v1-settings/http';
import { QueryClient } from 'react-query/core';
const queryName = 'app-settings-capsule';
export class AppSettingsCapsule {
    constructor(params) {
        this.httpClient = params.httpClient;
        this.context = params.context;
        this.queryCache = params.queryCache;
        this.queryClient = new QueryClient({
            queryCache: this.queryCache,
        });
    }
    _buildAppSettingsPayload(payload) {
        const { context: { componentId }, } = this;
        return {
            ...payload,
            ...(componentId && { componentId }),
        };
    }
    _getQueryKey() {
        return [queryName, this.context];
    }
    async getItem(key) {
        const { context, httpClient, queryClient } = this;
        const { data } = await queryClient.fetchQuery({
            queryFn: () => httpClient.request(getGlobalSettings({
                host: context.host,
            })),
            queryKey: this._getQueryKey(),
            staleTime: 120000,
        });
        return data.settings?.[context.componentId]?.[context.namespace]?.[key];
    }
    async setItem(key, value) {
        const { httpClient, queryCache, context } = this;
        queryCache
            .findAll({ queryKey: this._getQueryKey() })
            .forEach((q) => q.invalidate());
        await httpClient.request(setGlobalSettings({
            host: context.host,
            settings: {
                [context.componentId]: {
                    [context.namespace]: {
                        [key]: value,
                    },
                },
            },
        }));
    }
    async removeItem(key) {
        await this.setItem(key, null);
    }
}
