// @wix/bex-core/dist/esm/services/cairoMonitor.js
import { isWixPatternsError } from './WixPatternsError';
export class WixPatternsMonitor {
    constructor({ internalErrorMonitor, errorMonitor, internalFedops, environment, csid, }) {
        this._reports = {};
        this._internalErrorMonitor = internalErrorMonitor;
        this._errorMonitor = errorMonitor;
        this._internalFedops = internalFedops;
        this._environment = environment;
        this._csid = csid;
    }
    /**
     * This method is used to report errors that occur within the application. It takes in an error object with the following properties:
     * @param errorCode A WixPatternsErrorCode enum value representing the error that occurred.
     * @param exception The exception that was thrown.
     * @param extraData (optional) Additional data related to the error.
     * @param propagate (optional) Propagates report to vertical error monitor.
     * @default propagate true
     */
    reportError({ error, propagate = true, category, }) {
        const options = { tags: { ...this.extraInteractionParams } };
        if (category) {
            options.tags.category = category;
        }
        this._internalErrorMonitor.captureException(error, options);
        if (this._errorMonitor && propagate) {
            const originalException = isWixPatternsError(error)
                ? error.originalException
                : error;
            this._errorMonitor.captureException(originalException);
        }
    }
    get extraInteractionParams() {
        return {
            cairoVersion: this._environment.uiLibVersion,
            componentId: this._environment.componentId,
            csid: this._csid,
        };
    }
    interactionStarted(interactionName) {
        this._internalFedops?.interactionStarted(interactionName, {
            customParams: this.extraInteractionParams,
        });
    }
    interactionEnded(interactionName) {
        this._internalFedops?.interactionEnded(interactionName, {
            customParams: this.extraInteractionParams,
        });
    }
    reportInteractionStart(componentName, event, customParams) {
        const interaction = `${componentName.toLowerCase()}-${event}`;
        const reportKey = `${interaction}-start`;
        if (!this._reports[reportKey]) {
            this._reports[reportKey] = true;
            this._internalFedops?.interactionStarted(interaction, {
                customParams: {
                    ...this.extraInteractionParams,
                    ...customParams,
                },
            });
        }
    }
    reportInteractionEnd(componentName, event, customParams) {
        const interaction = `${componentName.toLowerCase()}-${event}`;
        const reportKey = `${interaction}-end`;
        if (!this._reports[reportKey]) {
            this._reports[reportKey] = true;
            this._internalFedops?.interactionEnded(interaction, {
                customParams: {
                    ...this.extraInteractionParams,
                    ...customParams,
                },
            });
        }
    }
    reportErrorThrownOncePerFlow() {
        this._internalFedops?.reportErrorThrownOncePerFlow();
    }
}
