// @wix/bex-core/dist/esm/services/createAccountComponentSettings.js
import { AccountPreferencesStrategy, } from './AccountPreferencesStrategy';
import { AccountPreferencesMergeStrategy } from './AccountPreferencesMergeStrategy';
function isDataCapsuleStorage(storageStrategy) {
    return 'storageStrategy' in storageStrategy;
}
export const createAccountComponentSettings = ({ httpClient, environment, createEssentials, }) => {
    return (options) => {
        const scope = environment.componentId || environment.artifactId;
        const { dataCapsule: _dataCapsule } = createEssentials({
            dataCapsule: {
                storageStrategy: 'userPreferences',
                namespace: [scope, options.namespace].join('.'),
            },
        });
        const { dataCapsule: oldLocalStorageComponentSettingsDataCapsule, fedopsLogger, } = createEssentials({
            dataCapsule: {
                namespace: options.namespace,
                scope,
            },
            fedops: {
                appName: 'cairo',
            },
        });
        const dataCapsule = _dataCapsule;
        if (isDataCapsuleStorage(dataCapsule.storageStrategy)) {
            dataCapsule.storageStrategy = new AccountPreferencesMergeStrategy({
                storage: new AccountPreferencesStrategy({
                    httpClient,
                }),
            });
        }
        _dataCapsule.migrateItem =
            async (key) => {
                const localData = await oldLocalStorageComponentSettingsDataCapsule
                    .getItem(key)
                    .catch(() => null);
                if (!localData) {
                    return;
                }
                fedopsLogger?.interactionStarted('account-component-settings-local-storage-migration');
                const data = await dataCapsule.getItem(key).catch(() => null);
                await dataCapsule.setItem(key, {
                    views: {
                        list: {
                            ...localData?.views?.list,
                            ...data?.views?.list,
                        },
                    },
                });
                await oldLocalStorageComponentSettingsDataCapsule.removeItem(key);
                fedopsLogger?.interactionEnded('account-component-settings-local-storage-migration');
            };
        return _dataCapsule;
    };
};
