// @wix/bex-core/dist/esm/services/DataCapsuleMergeStrategy.js
import merge from 'lodash/merge';
// This "merge" layer allows to mimic the automatic "merge" behavior of app-settings also in user-preferences storage strategy.
// https://github.com/wix-private/app-settings/blob/64eef870a0958a117a8325511f57df9d73857359/app-settings-service/src/services/mysql-storage.ts#L273
export class DataCapsuleMergeStrategy {
    constructor(params) {
        this.storage = params.storage;
    }
    extendScope(scope) {
        return scope;
    }
    async setItem(key, value, options) {
        const { storage } = this;
        const data = await this.getItem(key, options).catch(() => undefined);
        return storage.setItem(key, merge(data, value), options);
    }
    async removeItem(key, options) {
        const { storage } = this;
        const data = await this.getItem(key, options).catch(() => undefined);
        return storage.setItem(key, merge(data, { [key]: null }), options);
    }
    async getItem(key, options) {
        return this.storage.getItem(key, options);
    }
    async getAllItems(options) {
        return this.storage.getAllItems(options);
    }
}
