// @wix/bex-core/dist/esm/services/navigationManager.js
import EventEmitter from 'events';
export class NavigationManager {
    constructor(window) {
        this.events = new EventEmitter();
        this.window = window;
    }
    navigate(url) {
        const state = {
            prevented: false,
            pause: () => {
                return {
                    resume: () => {
                        this.window.location.href = url;
                    },
                };
            },
        };
        for (const navigation of this.events.listeners('navigation')) {
            navigation(state);
        }
        if (!state.prevented) {
            this.window.location.href = url;
        }
    }
    navigateToPage({ pageId }) {
        const url = new URL(this.window.location.href);
        url.pathname = `/page/${pageId}`;
        this.navigate(url.href);
    }
    confirmLeavePage() {
        this.events.emit('navigationPromptResult', true);
    }
    subscribe(onNavigation) {
        this.events.on('navigation', onNavigation);
        return () => {
            this.events.off('navigation', onNavigation);
        };
    }
}
