// @wix/bex-core/dist/esm/services/translationsImportsMap.js
export const translationsImportsMap = {
    ar: () => import(
    /* webpackChunkName: "cairo-locale-messages_ar-json" */ '../assets/locale/messages_ar.json'),
    bg: () => import(
    /* webpackChunkName: "cairo-locale-messages_bg-json" */ '../assets/locale/messages_bg.json'),
    ca: () => import(
    /* webpackChunkName: "cairo-locale-messages_ca-json" */ '../assets/locale/messages_ca.json'),
    cs: () => import(
    /* webpackChunkName: "cairo-locale-messages_cs-json" */ '../assets/locale/messages_cs.json'),
    da: () => import(
    /* webpackChunkName: "cairo-locale-messages_da-json" */ '../assets/locale/messages_da.json'),
    de: () => import(
    /* webpackChunkName: "cairo-locale-messages_de-json" */ '../assets/locale/messages_de.json'),
    el: () => import(
    /* webpackChunkName: "cairo-locale-messages_el-json" */ '../assets/locale/messages_el.json'),
    en: () => import(
    /* webpackChunkName: "cairo-locale-messages_en-json" */ '../assets/locale/messages_en.json'),
    es: () => import(
    /* webpackChunkName: "cairo-locale-messages_es-json" */ '../assets/locale/messages_es.json'),
    fi: () => import(
    /* webpackChunkName: "cairo-locale-messages_fi-json" */ '../assets/locale/messages_fi.json'),
    fr: () => import(
    /* webpackChunkName: "cairo-locale-messages_fr-json" */ '../assets/locale/messages_fr.json'),
    he: () => import(
    /* webpackChunkName: "cairo-locale-messages_he-json" */ '../assets/locale/messages_he.json'),
    hi: () => import(
    /* webpackChunkName: "cairo-locale-messages_hi-json" */ '../assets/locale/messages_hi.json'),
    hu: () => import(
    /* webpackChunkName: "cairo-locale-messages_hu-json" */ '../assets/locale/messages_hu.json'),
    id: () => import(
    /* webpackChunkName: "cairo-locale-messages_id-json" */ '../assets/locale/messages_id.json'),
    it: () => import(
    /* webpackChunkName: "cairo-locale-messages_it-json" */ '../assets/locale/messages_it.json'),
    ja: () => import(
    /* webpackChunkName: "cairo-locale-messages_ja-json" */ '../assets/locale/messages_ja.json'),
    ko: () => import(
    /* webpackChunkName: "cairo-locale-messages_ko-json" */ '../assets/locale/messages_ko.json'),
    lt: () => import(
    /* webpackChunkName: "cairo-locale-messages_lt-json" */ '../assets/locale/messages_lt.json'),
    ms: () => import(
    /* webpackChunkName: "cairo-locale-messages_ms-json" */ '../assets/locale/messages_ms.json'),
    nl: () => import(
    /* webpackChunkName: "cairo-locale-messages_nl-json" */ '../assets/locale/messages_nl.json'),
    no: () => import(
    /* webpackChunkName: "cairo-locale-messages_no-json" */ '../assets/locale/messages_no.json'),
    pl: () => import(
    /* webpackChunkName: "cairo-locale-messages_pl-json" */ '../assets/locale/messages_pl.json'),
    pt: () => import(
    /* webpackChunkName: "cairo-locale-messages_pt-json" */ '../assets/locale/messages_pt.json'),
    ro: () => import(
    /* webpackChunkName: "cairo-locale-messages_ro-json" */ '../assets/locale/messages_ro.json'),
    ru: () => import(
    /* webpackChunkName: "cairo-locale-messages_ru-json" */ '../assets/locale/messages_ru.json'),
    sk: () => import(
    /* webpackChunkName: "cairo-locale-messages_sk-json" */ '../assets/locale/messages_sk.json'),
    sl: () => import(
    /* webpackChunkName: "cairo-locale-messages_sl-json" */ '../assets/locale/messages_sl.json'),
    sv: () => import(
    /* webpackChunkName: "cairo-locale-messages_sv-json" */ '../assets/locale/messages_sv.json'),
    th: () => import(
    /* webpackChunkName: "cairo-locale-messages_th-json" */ '../assets/locale/messages_th.json'),
    tl: () => import(
    /* webpackChunkName: "cairo-locale-messages_tl-json" */ '../assets/locale/messages_tl.json'),
    tr: () => import(
    /* webpackChunkName: "cairo-locale-messages_tr-json" */ '../assets/locale/messages_tr.json'),
    uk: () => import(
    /* webpackChunkName: "cairo-locale-messages_uk-json" */ '../assets/locale/messages_uk.json'),
    vi: () => import(
    /* webpackChunkName: "cairo-locale-messages_vi-json" */ '../assets/locale/messages_vi.json'),
    visuale2e: () => import(
    /* webpackChunkName: "cairo-locale-messages_visuale2e-json" */ '../assets/locale/messages_visuale2e.json'),
    zh: () => import(
    /* webpackChunkName: "cairo-locale-messages_zh-json" */ '../assets/locale/messages_zh.json'),
};
