// @wix/auto_sdk_events_ticket-definitions-v-2/build/internal/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  Alignment: () => Alignment,
  AppType: () => AppType,
  AspectRatio: () => AspectRatio,
  BackgroundType: () => BackgroundType,
  ButtonDataType: () => ButtonDataType,
  CardStylesType: () => CardStylesType,
  CategoryStateState: () => CategoryStateState,
  CheckoutType: () => CheckoutType,
  ConferenceType: () => ConferenceType,
  Crop: () => Crop,
  DecorationType: () => DecorationType,
  DeleteStatus: () => DeleteStatus,
  Direction: () => Direction,
  DividerDataAlignment: () => DividerDataAlignment,
  EventStatus: () => EventStatus,
  EventType: () => EventType,
  EventsRecurrenceStatusStatus: () => EventsRecurrenceStatusStatus,
  FeeTypeEnumType: () => FeeTypeEnumType,
  Field: () => Field,
  FontType: () => FontType,
  GIFType: () => GIFType,
  ImagePosition: () => ImagePosition,
  InitialExpandedItems: () => InitialExpandedItems,
  InputControlType: () => InputControlType,
  Layout: () => Layout,
  LayoutType: () => LayoutType,
  LineStyle: () => LineStyle,
  LocationLocationType: () => LocationLocationType,
  LocationType: () => LocationType,
  MapType: () => MapType,
  Namespace: () => Namespace,
  NodeType: () => NodeType,
  NullValue: () => NullValue,
  Orientation: () => Orientation,
  Placement: () => Placement,
  PluginContainerDataAlignment: () => PluginContainerDataAlignment,
  PollLayoutDirection: () => PollLayoutDirection,
  PollLayoutType: () => PollLayoutType,
  Position: () => Position,
  PricingTypeEnumType: () => PricingTypeEnumType,
  RecurrenceStatusStatus: () => RecurrenceStatusStatus,
  RegistrationStatus: () => RegistrationStatus,
  Resizing: () => Resizing,
  ResponsivenessBehaviour: () => ResponsivenessBehaviour,
  RsvpStatusOptions: () => RsvpStatusOptions,
  SaleStatusEnumStatus: () => SaleStatusEnumStatus,
  Scaling: () => Scaling,
  SiteCreatedContext: () => SiteCreatedContext,
  SortOrder: () => SortOrder,
  Source: () => Source,
  State: () => State,
  Status: () => Status,
  StylesPosition: () => StylesPosition,
  SubdivisionSubdivisionType: () => SubdivisionSubdivisionType,
  SubdivisionType: () => SubdivisionType,
  Target: () => Target,
  TaxType: () => TaxType,
  TextAlignment: () => TextAlignment,
  ThumbnailsAlignment: () => ThumbnailsAlignment,
  Type: () => Type,
  ValueType: () => ValueType,
  VerticalAlignment: () => VerticalAlignment,
  VerticalAlignmentAlignment: () => VerticalAlignmentAlignment,
  ViewMode: () => ViewMode,
  ViewRole: () => ViewRole,
  VisitorType: () => VisitorType,
  VoteRole: () => VoteRole,
  WebhookIdentityType: () => WebhookIdentityType,
  Width: () => Width,
  WidthType: () => WidthType,
  bulkDeleteTicketDefinitionsByFilter: () => bulkDeleteTicketDefinitionsByFilter2,
  changeCurrency: () => changeCurrency2,
  countAvailableTicketDefinitions: () => countAvailableTicketDefinitions2,
  countTicketDefinitions: () => countTicketDefinitions2,
  createTicketDefinition: () => createTicketDefinition2,
  deleteTicketDefinition: () => deleteTicketDefinition2,
  getTicketDefinition: () => getTicketDefinition2,
  queryAvailableTicketDefinitions: () => queryAvailableTicketDefinitions2,
  queryTicketDefinitions: () => queryTicketDefinitions2,
  reorderTicketDefinitions: () => reorderTicketDefinitions2,
  updateTicketDefinition: () => updateTicketDefinition2
});
module.exports = __toCommonJS(index_typings_exports);

// src/events-v3-ticket-definition-ticket-definitions-v-2.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/events-v3-ticket-definition-ticket-definitions-v-2.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEventsTicketdefV3TicketDefinitionManagementUrl(opts) {
  const domainToMappings = {
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/events-ticket-definitions",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/events-ticket-definitions",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/events-ticket-definitions",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/events-ticket-definitions",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/events-ticket-definitions",
        destPath: ""
      },
      {
        srcPath: "/events/v3/ticket-definitions",
        destPath: "/v3/ticket-definitions"
      },
      {
        srcPath: "/events/v3/bulk/ticket-definitions",
        destPath: "/v3/bulk/ticket-definitions"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/events-ticket-definitions",
        destPath: ""
      }
    ],
    "events.wixapps.net": [
      {
        srcPath: "/_api/events-ticket-definitions",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/events-ticket-definitions",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_events_ticket-definitions-v-2";
function createTicketDefinition(payload) {
  function __createTicketDefinition({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "ticketDefinition.createdDate" },
          { path: "ticketDefinition.updatedDate" },
          { path: "ticketDefinition.eventCreatedDate" },
          { path: "ticketDefinition.salePeriod.startDate" },
          { path: "ticketDefinition.salePeriod.endDate" },
          {
            path: "ticketDefinition.eventDetails.dateAndTimeSettings.startDate"
          },
          { path: "ticketDefinition.eventDetails.dateAndTimeSettings.endDate" },
          {
            path: "ticketDefinition.eventDetails.dateAndTimeSettings.recurringEvents.individualEventDates.startDate"
          },
          {
            path: "ticketDefinition.eventDetails.dateAndTimeSettings.recurringEvents.individualEventDates.endDate"
          }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          {
            path: "ticketDefinition.eventDetails.location.address.geocode.latitude"
          },
          {
            path: "ticketDefinition.eventDetails.location.address.geocode.longitude"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v3.ticket_definition",
      method: "POST",
      methodFqn: "wix.events.ticketdef.v3.TicketDefinitionManagement.CreateTicketDefinition",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketdefV3TicketDefinitionManagementUrl({
        protoPath: "/v3/ticket-definitions",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "ticketDefinition.createdDate" },
            { path: "ticketDefinition.updatedDate" },
            { path: "ticketDefinition.eventCreatedDate" },
            { path: "ticketDefinition.salePeriod.startDate" },
            { path: "ticketDefinition.salePeriod.endDate" },
            {
              path: "ticketDefinition.eventDetails.dateAndTimeSettings.startDate"
            },
            {
              path: "ticketDefinition.eventDetails.dateAndTimeSettings.endDate"
            },
            {
              path: "ticketDefinition.eventDetails.dateAndTimeSettings.recurringEvents.individualEventDates.startDate"
            },
            {
              path: "ticketDefinition.eventDetails.dateAndTimeSettings.recurringEvents.individualEventDates.endDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "ticketDefinition.eventDetails.location.address.geocode.latitude"
            },
            {
              path: "ticketDefinition.eventDetails.location.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createTicketDefinition;
}
function updateTicketDefinition(payload) {
  function __updateTicketDefinition({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "ticketDefinition.createdDate" },
          { path: "ticketDefinition.updatedDate" },
          { path: "ticketDefinition.eventCreatedDate" },
          { path: "ticketDefinition.salePeriod.startDate" },
          { path: "ticketDefinition.salePeriod.endDate" },
          {
            path: "ticketDefinition.eventDetails.dateAndTimeSettings.startDate"
          },
          { path: "ticketDefinition.eventDetails.dateAndTimeSettings.endDate" },
          {
            path: "ticketDefinition.eventDetails.dateAndTimeSettings.recurringEvents.individualEventDates.startDate"
          },
          {
            path: "ticketDefinition.eventDetails.dateAndTimeSettings.recurringEvents.individualEventDates.endDate"
          }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          {
            path: "ticketDefinition.eventDetails.location.address.geocode.latitude"
          },
          {
            path: "ticketDefinition.eventDetails.location.address.geocode.longitude"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v3.ticket_definition",
      method: "PATCH",
      methodFqn: "wix.events.ticketdef.v3.TicketDefinitionManagement.UpdateTicketDefinition",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketdefV3TicketDefinitionManagementUrl({
        protoPath: "/v3/ticket-definitions/{ticketDefinition.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "ticketDefinition.createdDate" },
            { path: "ticketDefinition.updatedDate" },
            { path: "ticketDefinition.eventCreatedDate" },
            { path: "ticketDefinition.salePeriod.startDate" },
            { path: "ticketDefinition.salePeriod.endDate" },
            {
              path: "ticketDefinition.eventDetails.dateAndTimeSettings.startDate"
            },
            {
              path: "ticketDefinition.eventDetails.dateAndTimeSettings.endDate"
            },
            {
              path: "ticketDefinition.eventDetails.dateAndTimeSettings.recurringEvents.individualEventDates.startDate"
            },
            {
              path: "ticketDefinition.eventDetails.dateAndTimeSettings.recurringEvents.individualEventDates.endDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "ticketDefinition.eventDetails.location.address.geocode.latitude"
            },
            {
              path: "ticketDefinition.eventDetails.location.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateTicketDefinition;
}
function getTicketDefinition(payload) {
  function __getTicketDefinition({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v3.ticket_definition",
      method: "GET",
      methodFqn: "wix.events.ticketdef.v3.TicketDefinitionManagement.GetTicketDefinition",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketdefV3TicketDefinitionManagementUrl({
        protoPath: "/v3/ticket-definitions/{ticketDefinitionId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "ticketDefinition.createdDate" },
            { path: "ticketDefinition.updatedDate" },
            { path: "ticketDefinition.eventCreatedDate" },
            { path: "ticketDefinition.salePeriod.startDate" },
            { path: "ticketDefinition.salePeriod.endDate" },
            {
              path: "ticketDefinition.eventDetails.dateAndTimeSettings.startDate"
            },
            {
              path: "ticketDefinition.eventDetails.dateAndTimeSettings.endDate"
            },
            {
              path: "ticketDefinition.eventDetails.dateAndTimeSettings.recurringEvents.individualEventDates.startDate"
            },
            {
              path: "ticketDefinition.eventDetails.dateAndTimeSettings.recurringEvents.individualEventDates.endDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "ticketDefinition.eventDetails.location.address.geocode.latitude"
            },
            {
              path: "ticketDefinition.eventDetails.location.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getTicketDefinition;
}
function deleteTicketDefinition(payload) {
  function __deleteTicketDefinition({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v3.ticket_definition",
      method: "DELETE",
      methodFqn: "wix.events.ticketdef.v3.TicketDefinitionManagement.DeleteTicketDefinition",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketdefV3TicketDefinitionManagementUrl({
        protoPath: "/v3/ticket-definitions/{ticketDefinitionId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteTicketDefinition;
}
function reorderTicketDefinitions(payload) {
  function __reorderTicketDefinitions({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v3.ticket_definition",
      method: "POST",
      methodFqn: "wix.events.ticketdef.v3.TicketDefinitionManagement.ReorderTicketDefinitions",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketdefV3TicketDefinitionManagementUrl({
        protoPath: "/v3/ticket-definitions/reorder",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __reorderTicketDefinitions;
}
function queryTicketDefinitions(payload) {
  function __queryTicketDefinitions({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v3.ticket_definition",
      method: "POST",
      methodFqn: "wix.events.ticketdef.v3.TicketDefinitionManagement.QueryTicketDefinitions",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketdefV3TicketDefinitionManagementUrl({
        protoPath: "/v3/ticket-definitions/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "ticketDefinitions.createdDate" },
            { path: "ticketDefinitions.updatedDate" },
            { path: "ticketDefinitions.eventCreatedDate" },
            { path: "ticketDefinitions.salePeriod.startDate" },
            { path: "ticketDefinitions.salePeriod.endDate" },
            {
              path: "ticketDefinitions.eventDetails.dateAndTimeSettings.startDate"
            },
            {
              path: "ticketDefinitions.eventDetails.dateAndTimeSettings.endDate"
            },
            {
              path: "ticketDefinitions.eventDetails.dateAndTimeSettings.recurringEvents.individualEventDates.startDate"
            },
            {
              path: "ticketDefinitions.eventDetails.dateAndTimeSettings.recurringEvents.individualEventDates.endDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "ticketDefinitions.eventDetails.location.address.geocode.latitude"
            },
            {
              path: "ticketDefinitions.eventDetails.location.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryTicketDefinitions;
}
function queryAvailableTicketDefinitions(payload) {
  function __queryAvailableTicketDefinitions({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v3.ticket_definition",
      method: "POST",
      methodFqn: "wix.events.ticketdef.v3.TicketDefinitionManagement.QueryAvailableTicketDefinitions",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketdefV3TicketDefinitionManagementUrl({
        protoPath: "/v3/ticket-definitions/available/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "ticketDefinitions.createdDate" },
            { path: "ticketDefinitions.updatedDate" },
            { path: "ticketDefinitions.eventCreatedDate" },
            { path: "ticketDefinitions.salePeriod.startDate" },
            { path: "ticketDefinitions.salePeriod.endDate" },
            {
              path: "ticketDefinitions.eventDetails.dateAndTimeSettings.startDate"
            },
            {
              path: "ticketDefinitions.eventDetails.dateAndTimeSettings.endDate"
            },
            {
              path: "ticketDefinitions.eventDetails.dateAndTimeSettings.recurringEvents.individualEventDates.startDate"
            },
            {
              path: "ticketDefinitions.eventDetails.dateAndTimeSettings.recurringEvents.individualEventDates.endDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "ticketDefinitions.eventDetails.location.address.geocode.latitude"
            },
            {
              path: "ticketDefinitions.eventDetails.location.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryAvailableTicketDefinitions;
}
function countTicketDefinitions(payload) {
  function __countTicketDefinitions({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v3.ticket_definition",
      method: "POST",
      methodFqn: "wix.events.ticketdef.v3.TicketDefinitionManagement.CountTicketDefinitions",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketdefV3TicketDefinitionManagementUrl({
        protoPath: "/v3/ticket-definitions/count",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __countTicketDefinitions;
}
function countAvailableTicketDefinitions(payload) {
  function __countAvailableTicketDefinitions({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v3.ticket_definition",
      method: "POST",
      methodFqn: "wix.events.ticketdef.v3.TicketDefinitionManagement.CountAvailableTicketDefinitions",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketdefV3TicketDefinitionManagementUrl({
        protoPath: "/v3/ticket-definitions/available/count",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __countAvailableTicketDefinitions;
}
function bulkDeleteTicketDefinitionsByFilter(payload) {
  function __bulkDeleteTicketDefinitionsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v3.ticket_definition",
      method: "POST",
      methodFqn: "wix.events.ticketdef.v3.TicketDefinitionManagement.BulkDeleteTicketDefinitionsByFilter",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketdefV3TicketDefinitionManagementUrl({
        protoPath: "/v3/bulk/ticket-definitions/delete",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkDeleteTicketDefinitionsByFilter;
}
function changeCurrency(payload) {
  function __changeCurrency({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v3.ticket_definition",
      method: "POST",
      methodFqn: "wix.events.ticketdef.v3.TicketDefinitionManagement.ChangeCurrency",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketdefV3TicketDefinitionManagementUrl({
        protoPath: "/v3/ticket-definitions/currency",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __changeCurrency;
}

// src/events-v3-ticket-definition-ticket-definitions-v-2.universal.ts
var import_address = require("@wix/sdk-runtime/transformations/address");
var import_address2 = require("@wix/sdk-runtime/transformations/address");
var import_page_url = require("@wix/sdk-runtime/transformations/page-url");
var import_page_url2 = require("@wix/sdk-runtime/transformations/page-url");
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var PricingTypeEnumType = /* @__PURE__ */ ((PricingTypeEnumType2) => {
  PricingTypeEnumType2["UNKNOWN_PRICING_TYPE"] = "UNKNOWN_PRICING_TYPE";
  PricingTypeEnumType2["STANDARD"] = "STANDARD";
  PricingTypeEnumType2["DONATION"] = "DONATION";
  return PricingTypeEnumType2;
})(PricingTypeEnumType || {});
var FeeTypeEnumType = /* @__PURE__ */ ((FeeTypeEnumType2) => {
  FeeTypeEnumType2["UNKNOWN_FEE_TYPE"] = "UNKNOWN_FEE_TYPE";
  FeeTypeEnumType2["FEE_INCLUDED"] = "FEE_INCLUDED";
  FeeTypeEnumType2["FEE_ADDED_AT_CHECKOUT"] = "FEE_ADDED_AT_CHECKOUT";
  FeeTypeEnumType2["NO_FEE"] = "NO_FEE";
  return FeeTypeEnumType2;
})(FeeTypeEnumType || {});
var SaleStatusEnumStatus = /* @__PURE__ */ ((SaleStatusEnumStatus2) => {
  SaleStatusEnumStatus2["UNKNOWN_SALE_STATUS"] = "UNKNOWN_SALE_STATUS";
  SaleStatusEnumStatus2["SALE_SCHEDULED"] = "SALE_SCHEDULED";
  SaleStatusEnumStatus2["SALE_STARTED"] = "SALE_STARTED";
  SaleStatusEnumStatus2["SALE_ENDED"] = "SALE_ENDED";
  return SaleStatusEnumStatus2;
})(SaleStatusEnumStatus || {});
var LocationType = /* @__PURE__ */ ((LocationType2) => {
  LocationType2["UNKNOWN_LOCATION"] = "UNKNOWN_LOCATION";
  LocationType2["VENUE"] = "VENUE";
  LocationType2["ONLINE"] = "ONLINE";
  return LocationType2;
})(LocationType || {});
var SubdivisionSubdivisionType = /* @__PURE__ */ ((SubdivisionSubdivisionType2) => {
  SubdivisionSubdivisionType2["UNKNOWN_SUBDIVISION_TYPE"] = "UNKNOWN_SUBDIVISION_TYPE";
  SubdivisionSubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_1"] = "ADMINISTRATIVE_AREA_LEVEL_1";
  SubdivisionSubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_2"] = "ADMINISTRATIVE_AREA_LEVEL_2";
  SubdivisionSubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_3"] = "ADMINISTRATIVE_AREA_LEVEL_3";
  SubdivisionSubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_4"] = "ADMINISTRATIVE_AREA_LEVEL_4";
  SubdivisionSubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_5"] = "ADMINISTRATIVE_AREA_LEVEL_5";
  SubdivisionSubdivisionType2["COUNTRY"] = "COUNTRY";
  return SubdivisionSubdivisionType2;
})(SubdivisionSubdivisionType || {});
var RecurrenceStatusStatus = /* @__PURE__ */ ((RecurrenceStatusStatus2) => {
  RecurrenceStatusStatus2["UNKNOWN_STATUS"] = "UNKNOWN_STATUS";
  RecurrenceStatusStatus2["ONE_TIME"] = "ONE_TIME";
  RecurrenceStatusStatus2["RECURRING"] = "RECURRING";
  RecurrenceStatusStatus2["RECURRING_UPCOMING"] = "RECURRING_UPCOMING";
  RecurrenceStatusStatus2["RECURRING_RECENTLY_ENDED"] = "RECURRING_RECENTLY_ENDED";
  RecurrenceStatusStatus2["RECURRING_RECENTLY_CANCELED"] = "RECURRING_RECENTLY_CANCELED";
  return RecurrenceStatusStatus2;
})(RecurrenceStatusStatus || {});
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["UNKNOWN_EVENT_STATUS"] = "UNKNOWN_EVENT_STATUS";
  Status2["UPCOMING"] = "UPCOMING";
  Status2["STARTED"] = "STARTED";
  Status2["ENDED"] = "ENDED";
  Status2["CANCELED"] = "CANCELED";
  Status2["DRAFT"] = "DRAFT";
  return Status2;
})(Status || {});
var Field = /* @__PURE__ */ ((Field2) => {
  Field2["UNKNOWN_REQUESTED_FIELD"] = "UNKNOWN_REQUESTED_FIELD";
  Field2["SALES_DETAILS"] = "SALES_DETAILS";
  Field2["EVENT_DETAILS"] = "EVENT_DETAILS";
  Field2["SEATING_DETAILS"] = "SEATING_DETAILS";
  return Field2;
})(Field || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var LocationLocationType = /* @__PURE__ */ ((LocationLocationType2) => {
  LocationLocationType2["VENUE"] = "VENUE";
  LocationLocationType2["ONLINE"] = "ONLINE";
  return LocationLocationType2;
})(LocationLocationType || {});
var SubdivisionType = /* @__PURE__ */ ((SubdivisionType2) => {
  SubdivisionType2["UNKNOWN_SUBDIVISION_TYPE"] = "UNKNOWN_SUBDIVISION_TYPE";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_1"] = "ADMINISTRATIVE_AREA_LEVEL_1";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_2"] = "ADMINISTRATIVE_AREA_LEVEL_2";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_3"] = "ADMINISTRATIVE_AREA_LEVEL_3";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_4"] = "ADMINISTRATIVE_AREA_LEVEL_4";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_5"] = "ADMINISTRATIVE_AREA_LEVEL_5";
  SubdivisionType2["COUNTRY"] = "COUNTRY";
  return SubdivisionType2;
})(SubdivisionType || {});
var EventsRecurrenceStatusStatus = /* @__PURE__ */ ((EventsRecurrenceStatusStatus2) => {
  EventsRecurrenceStatusStatus2["ONE_TIME"] = "ONE_TIME";
  EventsRecurrenceStatusStatus2["RECURRING"] = "RECURRING";
  EventsRecurrenceStatusStatus2["RECURRING_NEXT"] = "RECURRING_NEXT";
  EventsRecurrenceStatusStatus2["RECURRING_LAST_ENDED"] = "RECURRING_LAST_ENDED";
  EventsRecurrenceStatusStatus2["RECURRING_LAST_CANCELED"] = "RECURRING_LAST_CANCELED";
  return EventsRecurrenceStatusStatus2;
})(EventsRecurrenceStatusStatus || {});
var EventStatus = /* @__PURE__ */ ((EventStatus2) => {
  EventStatus2["SCHEDULED"] = "SCHEDULED";
  EventStatus2["STARTED"] = "STARTED";
  EventStatus2["ENDED"] = "ENDED";
  EventStatus2["CANCELED"] = "CANCELED";
  return EventStatus2;
})(EventStatus || {});
var EventType = /* @__PURE__ */ ((EventType2) => {
  EventType2["NA_EVENT_TYPE"] = "NA_EVENT_TYPE";
  EventType2["RSVP"] = "RSVP";
  EventType2["TICKETS"] = "TICKETS";
  EventType2["EXTERNAL"] = "EXTERNAL";
  EventType2["NO_REGISTRATION"] = "NO_REGISTRATION";
  return EventType2;
})(EventType || {});
var RegistrationStatus = /* @__PURE__ */ ((RegistrationStatus2) => {
  RegistrationStatus2["NA_REGISTRATION_STATUS"] = "NA_REGISTRATION_STATUS";
  RegistrationStatus2["CLOSED"] = "CLOSED";
  RegistrationStatus2["CLOSED_MANUALLY"] = "CLOSED_MANUALLY";
  RegistrationStatus2["OPEN_RSVP"] = "OPEN_RSVP";
  RegistrationStatus2["OPEN_RSVP_WAITLIST"] = "OPEN_RSVP_WAITLIST";
  RegistrationStatus2["OPEN_TICKETS"] = "OPEN_TICKETS";
  RegistrationStatus2["OPEN_EXTERNAL"] = "OPEN_EXTERNAL";
  RegistrationStatus2["SCHEDULED_RSVP"] = "SCHEDULED_RSVP";
  return RegistrationStatus2;
})(RegistrationStatus || {});
var RsvpStatusOptions = /* @__PURE__ */ ((RsvpStatusOptions2) => {
  RsvpStatusOptions2["YES_ONLY"] = "YES_ONLY";
  RsvpStatusOptions2["YES_AND_NO"] = "YES_AND_NO";
  return RsvpStatusOptions2;
})(RsvpStatusOptions || {});
var TaxType = /* @__PURE__ */ ((TaxType2) => {
  TaxType2["INCLUDED"] = "INCLUDED";
  TaxType2["ADDED"] = "ADDED";
  TaxType2["ADDED_AT_CHECKOUT"] = "ADDED_AT_CHECKOUT";
  return TaxType2;
})(TaxType || {});
var CheckoutType = /* @__PURE__ */ ((CheckoutType2) => {
  CheckoutType2["UNKNOWN_CHECKOUT_TYPE"] = "UNKNOWN_CHECKOUT_TYPE";
  CheckoutType2["EVENTS_APP"] = "EVENTS_APP";
  CheckoutType2["ECOMM_PLATFORM"] = "ECOMM_PLATFORM";
  return CheckoutType2;
})(CheckoutType || {});
var VisitorType = /* @__PURE__ */ ((VisitorType2) => {
  VisitorType2["VISITOR"] = "VISITOR";
  VisitorType2["MEMBER"] = "MEMBER";
  VisitorType2["VISITOR_OR_MEMBER"] = "VISITOR_OR_MEMBER";
  return VisitorType2;
})(VisitorType || {});
var InputControlType = /* @__PURE__ */ ((InputControlType2) => {
  InputControlType2["INPUT"] = "INPUT";
  InputControlType2["TEXTAREA"] = "TEXTAREA";
  InputControlType2["DROPDOWN"] = "DROPDOWN";
  InputControlType2["RADIO"] = "RADIO";
  InputControlType2["CHECKBOX"] = "CHECKBOX";
  InputControlType2["NAME"] = "NAME";
  InputControlType2["GUEST_CONTROL"] = "GUEST_CONTROL";
  InputControlType2["ADDRESS_SHORT"] = "ADDRESS_SHORT";
  InputControlType2["ADDRESS_FULL"] = "ADDRESS_FULL";
  InputControlType2["DATE"] = "DATE";
  return InputControlType2;
})(InputControlType || {});
var ValueType = /* @__PURE__ */ ((ValueType2) => {
  ValueType2["TEXT"] = "TEXT";
  ValueType2["NUMBER"] = "NUMBER";
  ValueType2["TEXT_ARRAY"] = "TEXT_ARRAY";
  ValueType2["DATE_TIME"] = "DATE_TIME";
  ValueType2["ADDRESS"] = "ADDRESS";
  return ValueType2;
})(ValueType || {});
var ConferenceType = /* @__PURE__ */ ((ConferenceType2) => {
  ConferenceType2["MEETING"] = "MEETING";
  ConferenceType2["WEBINAR"] = "WEBINAR";
  return ConferenceType2;
})(ConferenceType || {});
var CategoryStateState = /* @__PURE__ */ ((CategoryStateState2) => {
  CategoryStateState2["MANUAL"] = "MANUAL";
  CategoryStateState2["AUTO"] = "AUTO";
  CategoryStateState2["RECURRING_EVENT"] = "RECURRING_EVENT";
  CategoryStateState2["HIDDEN"] = "HIDDEN";
  return CategoryStateState2;
})(CategoryStateState || {});
var NodeType = /* @__PURE__ */ ((NodeType2) => {
  NodeType2["PARAGRAPH"] = "PARAGRAPH";
  NodeType2["TEXT"] = "TEXT";
  NodeType2["HEADING"] = "HEADING";
  NodeType2["BULLETED_LIST"] = "BULLETED_LIST";
  NodeType2["ORDERED_LIST"] = "ORDERED_LIST";
  NodeType2["LIST_ITEM"] = "LIST_ITEM";
  NodeType2["BLOCKQUOTE"] = "BLOCKQUOTE";
  NodeType2["CODE_BLOCK"] = "CODE_BLOCK";
  NodeType2["VIDEO"] = "VIDEO";
  NodeType2["DIVIDER"] = "DIVIDER";
  NodeType2["FILE"] = "FILE";
  NodeType2["GALLERY"] = "GALLERY";
  NodeType2["GIF"] = "GIF";
  NodeType2["HTML"] = "HTML";
  NodeType2["IMAGE"] = "IMAGE";
  NodeType2["LINK_PREVIEW"] = "LINK_PREVIEW";
  NodeType2["MAP"] = "MAP";
  NodeType2["POLL"] = "POLL";
  NodeType2["APP_EMBED"] = "APP_EMBED";
  NodeType2["BUTTON"] = "BUTTON";
  NodeType2["COLLAPSIBLE_LIST"] = "COLLAPSIBLE_LIST";
  NodeType2["TABLE"] = "TABLE";
  NodeType2["EMBED"] = "EMBED";
  NodeType2["COLLAPSIBLE_ITEM"] = "COLLAPSIBLE_ITEM";
  NodeType2["COLLAPSIBLE_ITEM_TITLE"] = "COLLAPSIBLE_ITEM_TITLE";
  NodeType2["COLLAPSIBLE_ITEM_BODY"] = "COLLAPSIBLE_ITEM_BODY";
  NodeType2["TABLE_CELL"] = "TABLE_CELL";
  NodeType2["TABLE_ROW"] = "TABLE_ROW";
  NodeType2["EXTERNAL"] = "EXTERNAL";
  NodeType2["AUDIO"] = "AUDIO";
  NodeType2["CAPTION"] = "CAPTION";
  NodeType2["LAYOUT"] = "LAYOUT";
  NodeType2["LAYOUT_CELL"] = "LAYOUT_CELL";
  return NodeType2;
})(NodeType || {});
var WidthType = /* @__PURE__ */ ((WidthType2) => {
  WidthType2["CONTENT"] = "CONTENT";
  WidthType2["SMALL"] = "SMALL";
  WidthType2["ORIGINAL"] = "ORIGINAL";
  WidthType2["FULL_WIDTH"] = "FULL_WIDTH";
  return WidthType2;
})(WidthType || {});
var PluginContainerDataAlignment = /* @__PURE__ */ ((PluginContainerDataAlignment2) => {
  PluginContainerDataAlignment2["CENTER"] = "CENTER";
  PluginContainerDataAlignment2["LEFT"] = "LEFT";
  PluginContainerDataAlignment2["RIGHT"] = "RIGHT";
  return PluginContainerDataAlignment2;
})(PluginContainerDataAlignment || {});
var ButtonDataType = /* @__PURE__ */ ((ButtonDataType2) => {
  ButtonDataType2["LINK"] = "LINK";
  ButtonDataType2["ACTION"] = "ACTION";
  return ButtonDataType2;
})(ButtonDataType || {});
var Target = /* @__PURE__ */ ((Target2) => {
  Target2["SELF"] = "SELF";
  Target2["BLANK"] = "BLANK";
  Target2["PARENT"] = "PARENT";
  Target2["TOP"] = "TOP";
  return Target2;
})(Target || {});
var TextAlignment = /* @__PURE__ */ ((TextAlignment2) => {
  TextAlignment2["AUTO"] = "AUTO";
  TextAlignment2["LEFT"] = "LEFT";
  TextAlignment2["RIGHT"] = "RIGHT";
  TextAlignment2["CENTER"] = "CENTER";
  TextAlignment2["JUSTIFY"] = "JUSTIFY";
  return TextAlignment2;
})(TextAlignment || {});
var LineStyle = /* @__PURE__ */ ((LineStyle2) => {
  LineStyle2["SINGLE"] = "SINGLE";
  LineStyle2["DOUBLE"] = "DOUBLE";
  LineStyle2["DASHED"] = "DASHED";
  LineStyle2["DOTTED"] = "DOTTED";
  return LineStyle2;
})(LineStyle || {});
var Width = /* @__PURE__ */ ((Width2) => {
  Width2["LARGE"] = "LARGE";
  Width2["MEDIUM"] = "MEDIUM";
  Width2["SMALL"] = "SMALL";
  return Width2;
})(Width || {});
var DividerDataAlignment = /* @__PURE__ */ ((DividerDataAlignment2) => {
  DividerDataAlignment2["CENTER"] = "CENTER";
  DividerDataAlignment2["LEFT"] = "LEFT";
  DividerDataAlignment2["RIGHT"] = "RIGHT";
  return DividerDataAlignment2;
})(DividerDataAlignment || {});
var ViewMode = /* @__PURE__ */ ((ViewMode2) => {
  ViewMode2["NONE"] = "NONE";
  ViewMode2["FULL"] = "FULL";
  ViewMode2["MINI"] = "MINI";
  return ViewMode2;
})(ViewMode || {});
var LayoutType = /* @__PURE__ */ ((LayoutType2) => {
  LayoutType2["COLLAGE"] = "COLLAGE";
  LayoutType2["MASONRY"] = "MASONRY";
  LayoutType2["GRID"] = "GRID";
  LayoutType2["THUMBNAIL"] = "THUMBNAIL";
  LayoutType2["SLIDER"] = "SLIDER";
  LayoutType2["SLIDESHOW"] = "SLIDESHOW";
  LayoutType2["PANORAMA"] = "PANORAMA";
  LayoutType2["COLUMN"] = "COLUMN";
  LayoutType2["MAGIC"] = "MAGIC";
  LayoutType2["FULLSIZE"] = "FULLSIZE";
  return LayoutType2;
})(LayoutType || {});
var Orientation = /* @__PURE__ */ ((Orientation2) => {
  Orientation2["ROWS"] = "ROWS";
  Orientation2["COLUMNS"] = "COLUMNS";
  return Orientation2;
})(Orientation || {});
var Crop = /* @__PURE__ */ ((Crop2) => {
  Crop2["FILL"] = "FILL";
  Crop2["FIT"] = "FIT";
  return Crop2;
})(Crop || {});
var ThumbnailsAlignment = /* @__PURE__ */ ((ThumbnailsAlignment2) => {
  ThumbnailsAlignment2["TOP"] = "TOP";
  ThumbnailsAlignment2["RIGHT"] = "RIGHT";
  ThumbnailsAlignment2["BOTTOM"] = "BOTTOM";
  ThumbnailsAlignment2["LEFT"] = "LEFT";
  ThumbnailsAlignment2["NONE"] = "NONE";
  return ThumbnailsAlignment2;
})(ThumbnailsAlignment || {});
var GIFType = /* @__PURE__ */ ((GIFType2) => {
  GIFType2["NORMAL"] = "NORMAL";
  GIFType2["STICKER"] = "STICKER";
  return GIFType2;
})(GIFType || {});
var Source = /* @__PURE__ */ ((Source2) => {
  Source2["HTML"] = "HTML";
  Source2["ADSENSE"] = "ADSENSE";
  Source2["AI"] = "AI";
  return Source2;
})(Source || {});
var StylesPosition = /* @__PURE__ */ ((StylesPosition2) => {
  StylesPosition2["START"] = "START";
  StylesPosition2["END"] = "END";
  StylesPosition2["TOP"] = "TOP";
  StylesPosition2["HIDDEN"] = "HIDDEN";
  return StylesPosition2;
})(StylesPosition || {});
var MapType = /* @__PURE__ */ ((MapType2) => {
  MapType2["ROADMAP"] = "ROADMAP";
  MapType2["SATELITE"] = "SATELITE";
  MapType2["HYBRID"] = "HYBRID";
  MapType2["TERRAIN"] = "TERRAIN";
  return MapType2;
})(MapType || {});
var ViewRole = /* @__PURE__ */ ((ViewRole2) => {
  ViewRole2["CREATOR"] = "CREATOR";
  ViewRole2["VOTERS"] = "VOTERS";
  ViewRole2["EVERYONE"] = "EVERYONE";
  return ViewRole2;
})(ViewRole || {});
var VoteRole = /* @__PURE__ */ ((VoteRole2) => {
  VoteRole2["SITE_MEMBERS"] = "SITE_MEMBERS";
  VoteRole2["ALL"] = "ALL";
  return VoteRole2;
})(VoteRole || {});
var PollLayoutType = /* @__PURE__ */ ((PollLayoutType2) => {
  PollLayoutType2["LIST"] = "LIST";
  PollLayoutType2["GRID"] = "GRID";
  return PollLayoutType2;
})(PollLayoutType || {});
var PollLayoutDirection = /* @__PURE__ */ ((PollLayoutDirection2) => {
  PollLayoutDirection2["LTR"] = "LTR";
  PollLayoutDirection2["RTL"] = "RTL";
  return PollLayoutDirection2;
})(PollLayoutDirection || {});
var BackgroundType = /* @__PURE__ */ ((BackgroundType2) => {
  BackgroundType2["COLOR"] = "COLOR";
  BackgroundType2["IMAGE"] = "IMAGE";
  BackgroundType2["GRADIENT"] = "GRADIENT";
  return BackgroundType2;
})(BackgroundType || {});
var DecorationType = /* @__PURE__ */ ((DecorationType2) => {
  DecorationType2["BOLD"] = "BOLD";
  DecorationType2["ITALIC"] = "ITALIC";
  DecorationType2["UNDERLINE"] = "UNDERLINE";
  DecorationType2["SPOILER"] = "SPOILER";
  DecorationType2["ANCHOR"] = "ANCHOR";
  DecorationType2["MENTION"] = "MENTION";
  DecorationType2["LINK"] = "LINK";
  DecorationType2["COLOR"] = "COLOR";
  DecorationType2["FONT_SIZE"] = "FONT_SIZE";
  DecorationType2["EXTERNAL"] = "EXTERNAL";
  DecorationType2["STRIKETHROUGH"] = "STRIKETHROUGH";
  DecorationType2["SUPERSCRIPT"] = "SUPERSCRIPT";
  DecorationType2["SUBSCRIPT"] = "SUBSCRIPT";
  return DecorationType2;
})(DecorationType || {});
var FontType = /* @__PURE__ */ ((FontType2) => {
  FontType2["PX"] = "PX";
  FontType2["EM"] = "EM";
  return FontType2;
})(FontType || {});
var Position = /* @__PURE__ */ ((Position2) => {
  Position2["START"] = "START";
  Position2["END"] = "END";
  Position2["TOP"] = "TOP";
  return Position2;
})(Position || {});
var AspectRatio = /* @__PURE__ */ ((AspectRatio2) => {
  AspectRatio2["SQUARE"] = "SQUARE";
  AspectRatio2["RECTANGLE"] = "RECTANGLE";
  return AspectRatio2;
})(AspectRatio || {});
var Resizing = /* @__PURE__ */ ((Resizing2) => {
  Resizing2["FILL"] = "FILL";
  Resizing2["FIT"] = "FIT";
  return Resizing2;
})(Resizing || {});
var Placement = /* @__PURE__ */ ((Placement2) => {
  Placement2["IMAGE"] = "IMAGE";
  Placement2["PRODUCT_INFO"] = "PRODUCT_INFO";
  return Placement2;
})(Placement || {});
var CardStylesType = /* @__PURE__ */ ((CardStylesType2) => {
  CardStylesType2["CONTAINED"] = "CONTAINED";
  CardStylesType2["FRAMELESS"] = "FRAMELESS";
  return CardStylesType2;
})(CardStylesType || {});
var Alignment = /* @__PURE__ */ ((Alignment2) => {
  Alignment2["START"] = "START";
  Alignment2["CENTER"] = "CENTER";
  Alignment2["END"] = "END";
  return Alignment2;
})(Alignment || {});
var Layout = /* @__PURE__ */ ((Layout2) => {
  Layout2["STACKED"] = "STACKED";
  Layout2["SIDE_BY_SIDE"] = "SIDE_BY_SIDE";
  return Layout2;
})(Layout || {});
var AppType = /* @__PURE__ */ ((AppType2) => {
  AppType2["PRODUCT"] = "PRODUCT";
  AppType2["EVENT"] = "EVENT";
  AppType2["BOOKING"] = "BOOKING";
  return AppType2;
})(AppType || {});
var InitialExpandedItems = /* @__PURE__ */ ((InitialExpandedItems2) => {
  InitialExpandedItems2["FIRST"] = "FIRST";
  InitialExpandedItems2["ALL"] = "ALL";
  InitialExpandedItems2["NONE"] = "NONE";
  return InitialExpandedItems2;
})(InitialExpandedItems || {});
var Direction = /* @__PURE__ */ ((Direction2) => {
  Direction2["LTR"] = "LTR";
  Direction2["RTL"] = "RTL";
  return Direction2;
})(Direction || {});
var VerticalAlignment = /* @__PURE__ */ ((VerticalAlignment2) => {
  VerticalAlignment2["TOP"] = "TOP";
  VerticalAlignment2["MIDDLE"] = "MIDDLE";
  VerticalAlignment2["BOTTOM"] = "BOTTOM";
  return VerticalAlignment2;
})(VerticalAlignment || {});
var NullValue = /* @__PURE__ */ ((NullValue2) => {
  NullValue2["NULL_VALUE"] = "NULL_VALUE";
  return NullValue2;
})(NullValue || {});
var Scaling = /* @__PURE__ */ ((Scaling2) => {
  Scaling2["AUTO"] = "AUTO";
  Scaling2["CONTAIN"] = "CONTAIN";
  Scaling2["COVER"] = "COVER";
  return Scaling2;
})(Scaling || {});
var ImagePosition = /* @__PURE__ */ ((ImagePosition2) => {
  ImagePosition2["CENTER"] = "CENTER";
  ImagePosition2["CENTER_LEFT"] = "CENTER_LEFT";
  ImagePosition2["CENTER_RIGHT"] = "CENTER_RIGHT";
  ImagePosition2["TOP"] = "TOP";
  ImagePosition2["TOP_LEFT"] = "TOP_LEFT";
  ImagePosition2["TOP_RIGHT"] = "TOP_RIGHT";
  ImagePosition2["BOTTOM"] = "BOTTOM";
  ImagePosition2["BOTTOM_LEFT"] = "BOTTOM_LEFT";
  ImagePosition2["BOTTOM_RIGHT"] = "BOTTOM_RIGHT";
  return ImagePosition2;
})(ImagePosition || {});
var VerticalAlignmentAlignment = /* @__PURE__ */ ((VerticalAlignmentAlignment2) => {
  VerticalAlignmentAlignment2["TOP"] = "TOP";
  VerticalAlignmentAlignment2["MIDDLE"] = "MIDDLE";
  VerticalAlignmentAlignment2["BOTTOM"] = "BOTTOM";
  return VerticalAlignmentAlignment2;
})(VerticalAlignmentAlignment || {});
var ResponsivenessBehaviour = /* @__PURE__ */ ((ResponsivenessBehaviour2) => {
  ResponsivenessBehaviour2["STACK"] = "STACK";
  ResponsivenessBehaviour2["WRAP"] = "WRAP";
  return ResponsivenessBehaviour2;
})(ResponsivenessBehaviour || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNKNOWN_BADGE_TYPE"] = "UNKNOWN_BADGE_TYPE";
  Type2["FIRST_PRIORITY"] = "FIRST_PRIORITY";
  Type2["SECOND_PRIORITY"] = "SECOND_PRIORITY";
  Type2["THIRD_PRIORITY"] = "THIRD_PRIORITY";
  return Type2;
})(Type || {});
var State = /* @__PURE__ */ ((State2) => {
  State2["UNKNOWN"] = "UNKNOWN";
  State2["ENABLED"] = "ENABLED";
  State2["DISABLED"] = "DISABLED";
  State2["PENDING"] = "PENDING";
  State2["DEMO"] = "DEMO";
  return State2;
})(State || {});
var SiteCreatedContext = /* @__PURE__ */ ((SiteCreatedContext2) => {
  SiteCreatedContext2["OTHER"] = "OTHER";
  SiteCreatedContext2["FROM_TEMPLATE"] = "FROM_TEMPLATE";
  SiteCreatedContext2["DUPLICATE_BY_SITE_TRANSFER"] = "DUPLICATE_BY_SITE_TRANSFER";
  SiteCreatedContext2["DUPLICATE"] = "DUPLICATE";
  SiteCreatedContext2["OLD_SITE_TRANSFER"] = "OLD_SITE_TRANSFER";
  SiteCreatedContext2["FLASH"] = "FLASH";
  return SiteCreatedContext2;
})(SiteCreatedContext || {});
var Namespace = /* @__PURE__ */ ((Namespace2) => {
  Namespace2["UNKNOWN_NAMESPACE"] = "UNKNOWN_NAMESPACE";
  Namespace2["WIX"] = "WIX";
  Namespace2["SHOUT_OUT"] = "SHOUT_OUT";
  Namespace2["ALBUMS"] = "ALBUMS";
  Namespace2["WIX_STORES_TEST_DRIVE"] = "WIX_STORES_TEST_DRIVE";
  Namespace2["HOTELS"] = "HOTELS";
  Namespace2["CLUBS"] = "CLUBS";
  Namespace2["ONBOARDING_DRAFT"] = "ONBOARDING_DRAFT";
  Namespace2["DEV_SITE"] = "DEV_SITE";
  Namespace2["LOGOS"] = "LOGOS";
  Namespace2["VIDEO_MAKER"] = "VIDEO_MAKER";
  Namespace2["PARTNER_DASHBOARD"] = "PARTNER_DASHBOARD";
  Namespace2["DEV_CENTER_COMPANY"] = "DEV_CENTER_COMPANY";
  Namespace2["HTML_DRAFT"] = "HTML_DRAFT";
  Namespace2["SITELESS_BUSINESS"] = "SITELESS_BUSINESS";
  Namespace2["CREATOR_ECONOMY"] = "CREATOR_ECONOMY";
  Namespace2["DASHBOARD_FIRST"] = "DASHBOARD_FIRST";
  Namespace2["ANYWHERE"] = "ANYWHERE";
  Namespace2["HEADLESS"] = "HEADLESS";
  Namespace2["ACCOUNT_MASTER_CMS"] = "ACCOUNT_MASTER_CMS";
  Namespace2["RISE"] = "RISE";
  Namespace2["BRANDED_FIRST"] = "BRANDED_FIRST";
  Namespace2["NOWNIA"] = "NOWNIA";
  Namespace2["UGC_TEMPLATE"] = "UGC_TEMPLATE";
  Namespace2["CODUX"] = "CODUX";
  Namespace2["MEDIA_DESIGN_CREATOR"] = "MEDIA_DESIGN_CREATOR";
  Namespace2["SHARED_BLOG_ENTERPRISE"] = "SHARED_BLOG_ENTERPRISE";
  Namespace2["STANDALONE_FORMS"] = "STANDALONE_FORMS";
  Namespace2["STANDALONE_EVENTS"] = "STANDALONE_EVENTS";
  Namespace2["MIMIR"] = "MIMIR";
  Namespace2["TWINS"] = "TWINS";
  Namespace2["NANO"] = "NANO";
  return Namespace2;
})(Namespace || {});
var DeleteStatus = /* @__PURE__ */ ((DeleteStatus2) => {
  DeleteStatus2["UNKNOWN"] = "UNKNOWN";
  DeleteStatus2["TRASH"] = "TRASH";
  DeleteStatus2["DELETED"] = "DELETED";
  DeleteStatus2["PENDING_PURGE"] = "PENDING_PURGE";
  DeleteStatus2["PURGED_EXTERNALLY"] = "PURGED_EXTERNALLY";
  return DeleteStatus2;
})(DeleteStatus || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createTicketDefinition2(ticketDefinition, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      ticketDefinition,
      fields: options?.fields
    }),
    [
      {
        transformFn: import_page_url.transformSDKPageURLToRESTPageURL,
        paths: [{ path: "ticketDefinition.eventDetails.eventPageUrl" }]
      },
      {
        transformFn: import_address.transformSDKAddressToRESTAddress,
        paths: [{ path: "ticketDefinition.eventDetails.location.address" }]
      }
    ]
  );
  const reqOpts = createTicketDefinition(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [{ path: "ticketDefinition.eventDetails.eventPageUrl" }]
        },
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [{ path: "ticketDefinition.eventDetails.location.address" }]
        }
      ])
    )?.ticketDefinition;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          ticketDefinition: "$[0]",
          fields: "$[1].fields"
        },
        singleArgumentUnchanged: false
      },
      ["ticketDefinition", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateTicketDefinition2(_id, ticketDefinition, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      ticketDefinition: { ...ticketDefinition, id: _id },
      fields: options?.fields
    }),
    [
      {
        transformFn: import_page_url.transformSDKPageURLToRESTPageURL,
        paths: [{ path: "ticketDefinition.eventDetails.eventPageUrl" }]
      },
      {
        transformFn: import_address.transformSDKAddressToRESTAddress,
        paths: [{ path: "ticketDefinition.eventDetails.location.address" }]
      }
    ]
  );
  const reqOpts = updateTicketDefinition(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [{ path: "ticketDefinition.eventDetails.eventPageUrl" }]
        },
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [{ path: "ticketDefinition.eventDetails.location.address" }]
        }
      ])
    )?.ticketDefinition;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { ticketDefinition: "$[1]" },
        explicitPathsToArguments: {
          "ticketDefinition.id": "$[0]",
          fields: "$[2].fields"
        },
        singleArgumentUnchanged: false
      },
      ["_id", "ticketDefinition", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getTicketDefinition2(ticketDefinitionId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    ticketDefinitionId,
    fields: options?.fields
  });
  const reqOpts = getTicketDefinition(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [{ path: "ticketDefinition.eventDetails.eventPageUrl" }]
        },
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [{ path: "ticketDefinition.eventDetails.location.address" }]
        }
      ])
    )?.ticketDefinition;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          ticketDefinitionId: "$[0]",
          fields: "$[1].fields"
        },
        singleArgumentUnchanged: false
      },
      ["ticketDefinitionId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteTicketDefinition2(ticketDefinitionId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    ticketDefinitionId
  });
  const reqOpts = deleteTicketDefinition(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { ticketDefinitionId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["ticketDefinitionId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function reorderTicketDefinitions2(eventId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    eventId,
    ticketDefinitionId: options?.ticketDefinitionId,
    beforeDefinitionId: options?.beforeDefinitionId,
    afterDefinitionId: options?.afterDefinitionId
  });
  const reqOpts = reorderTicketDefinitions(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          eventId: "$[0]",
          ticketDefinitionId: "$[1].ticketDefinitionId",
          beforeDefinitionId: "$[1].beforeDefinitionId",
          afterDefinitionId: "$[1].afterDefinitionId"
        },
        singleArgumentUnchanged: false
      },
      ["eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryTicketDefinitions2(options) {
  const { httpClient, sideEffects } = arguments[1];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryTicketDefinitions({
        ...payload,
        ...options ?? {}
      });
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, options];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [
          {
            transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
            paths: [{ path: "ticketDefinitions.eventDetails.eventPageUrl" }]
          },
          {
            transformFn: import_address2.transformRESTAddressToSDKAddress,
            paths: [
              { path: "ticketDefinitions.eventDetails.location.address" }
            ]
          }
        ])
      );
      return {
        items: transformedData?.ticketDefinitions,
        pagingMetadata: transformedData?.metadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function queryAvailableTicketDefinitions2(query, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    query,
    fields: options?.fields
  });
  const reqOpts = queryAvailableTicketDefinitions(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [{ path: "ticketDefinitions.eventDetails.eventPageUrl" }]
        },
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [{ path: "ticketDefinitions.eventDetails.location.address" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]", fields: "$[1].fields" },
        singleArgumentUnchanged: false
      },
      ["query", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function countTicketDefinitions2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    filter: options?.filter,
    facet: options?.facet
  });
  const reqOpts = countTicketDefinitions(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          filter: "$[0].filter",
          facet: "$[0].facet"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function countAvailableTicketDefinitions2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    filter: options?.filter
  });
  const reqOpts = countAvailableTicketDefinitions(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { filter: "$[0].filter" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkDeleteTicketDefinitionsByFilter2(filter) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ filter });
  const reqOpts = bulkDeleteTicketDefinitionsByFilter(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { filter: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["filter"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function changeCurrency2(eventId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    eventId,
    currency: options?.currency
  });
  const reqOpts = changeCurrency(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          eventId: "$[0]",
          currency: "$[1].currency"
        },
        singleArgumentUnchanged: false
      },
      ["eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Alignment,
  AppType,
  AspectRatio,
  BackgroundType,
  ButtonDataType,
  CardStylesType,
  CategoryStateState,
  CheckoutType,
  ConferenceType,
  Crop,
  DecorationType,
  DeleteStatus,
  Direction,
  DividerDataAlignment,
  EventStatus,
  EventType,
  EventsRecurrenceStatusStatus,
  FeeTypeEnumType,
  Field,
  FontType,
  GIFType,
  ImagePosition,
  InitialExpandedItems,
  InputControlType,
  Layout,
  LayoutType,
  LineStyle,
  LocationLocationType,
  LocationType,
  MapType,
  Namespace,
  NodeType,
  NullValue,
  Orientation,
  Placement,
  PluginContainerDataAlignment,
  PollLayoutDirection,
  PollLayoutType,
  Position,
  PricingTypeEnumType,
  RecurrenceStatusStatus,
  RegistrationStatus,
  Resizing,
  ResponsivenessBehaviour,
  RsvpStatusOptions,
  SaleStatusEnumStatus,
  Scaling,
  SiteCreatedContext,
  SortOrder,
  Source,
  State,
  Status,
  StylesPosition,
  SubdivisionSubdivisionType,
  SubdivisionType,
  Target,
  TaxType,
  TextAlignment,
  ThumbnailsAlignment,
  Type,
  ValueType,
  VerticalAlignment,
  VerticalAlignmentAlignment,
  ViewMode,
  ViewRole,
  VisitorType,
  VoteRole,
  WebhookIdentityType,
  Width,
  WidthType,
  bulkDeleteTicketDefinitionsByFilter,
  changeCurrency,
  countAvailableTicketDefinitions,
  countTicketDefinitions,
  createTicketDefinition,
  deleteTicketDefinition,
  getTicketDefinition,
  queryAvailableTicketDefinitions,
  queryTicketDefinitions,
  reorderTicketDefinitions,
  updateTicketDefinition
});
