// @wix/wix-style-react-incubator/dist/esm/drag-and-drop/table/DragHandle.js
import _extends from "@babel/runtime/helpers/extends";
import React, { useEffect, useRef } from 'react';
import { DragAndDropLarge } from '@wix/wix-ui-icons-common/system';
import { classes, st } from './DragHandle.st.css.js';
import { useDragAndDropContext } from './DragAndDropContext';
import { useSortableTableRow } from './useSortableTableRow';
export function DragHandle(props) {
  const {
    id,
    className
  } = props;
  const {
    setActivatorNodeRef
  } = useSortableTableRow({
    id
  });
  const {
    eventHandlers,
    announcements,
    dragHandleRef: dragHandleRefProp,
    onDropAnimationEnd
  } = useDragAndDropContext();
  const ariaDescribedBy = `draggable-table-row-description-${id}`;
  const dragHandleRef = useRef();
  useEffect(() => {
    if (onDropAnimationEnd) {
      return onDropAnimationEnd(id, dragHandleRef);
    }
    return;
  }, [id, onDropAnimationEnd]);
  return /*#__PURE__*/React.createElement("div", _extends({}, eventHandlers == null || eventHandlers.handle == null ? void 0 : eventHandlers.handle({
    id
  }), {
    className: st(classes.root, {}, className)
  }), /*#__PURE__*/React.createElement("div", {
    id: ariaDescribedBy,
    style: {
      display: 'none'
    }
  }, announcements == null || announcements.itemDescription == null ? void 0 : announcements.itemDescription(id)), /*#__PURE__*/React.createElement("div", {
    ref: e => {
      dragHandleRef.current = e;
      setActivatorNodeRef(e);
      dragHandleRefProp == null || dragHandleRefProp(id, e);
    },
    role: "button",
    "aria-describedby": ariaDescribedBy,
    "aria-label": announcements == null || announcements.dragHandleLabel == null ? void 0 : announcements.dragHandleLabel(id),
    className: classes.triggerIcon,
    tabIndex: 0
  }, /*#__PURE__*/React.createElement(DragAndDropLarge, null)));
}
export function isDragHandleNode(node) {
  return node.classList.contains(classes.triggerIcon);
}
