// @wix/wix-style-react-incubator/dist/esm/drag-and-drop/table/TableRowHorizontalScrollDragOverlay.js
import _extends from "@babel/runtime/helpers/extends";
import React, { useCallback, useMemo, useRef } from 'react';
import { DragOverlay, useDndContext } from '@dnd-kit/core';
import { useDragAndDropContext } from './DragAndDropContext';
import { CSS } from '@dnd-kit/utilities';
import { restrictContainerToWindowEdges } from './restrictContainerToWindowEdges';
import { restrictKeyboardToVerticalAxis } from './restrictKeyboardToVerticalAxis';
import { useDefaultDragOverlayProps } from './useDefaultDragOverlayProps';
export function TableRowHorizontalScrollDragOverlay(_ref) {
  let {
    children
  } = _ref;
  const {
    dropAnimation: dropAnimationProp,
    DragOverlayComponentType = DragOverlay
  } = useDragAndDropContext();
  const {
    scrollableAncestorRects,
    activeNodeRect
  } = useDndContext();
  const containerNodeRect = scrollableAncestorRects[0];
  const containerWidth = containerNodeRect == null ? void 0 : containerNodeRect.width;
  const widthRef = useRef(containerWidth);
  if (containerWidth != null) {
    widthRef.current = containerWidth;
  }
  const containerLeft = containerNodeRect == null ? void 0 : containerNodeRect.left;
  const leftRef = useRef(containerLeft);
  if (containerNodeRect != null) {
    leftRef.current = containerLeft;
  }
  const style = useMemo(() => ({
    left: leftRef.current,
    width: widthRef.current != null && (activeNodeRect == null ? void 0 : activeNodeRect.width) != null ? Math.min(widthRef.current, activeNodeRect.width) : widthRef.current
  }), [widthRef.current, leftRef.current, activeNodeRect]);
  const keyframes = useCallback(_ref2 => {
    let {
      transform: {
        initial,
        final
      }
    } = _ref2;
    return [{
      transform: CSS.Transform.toString(initial)
    }, {
      transform: CSS.Transform.toString({
        ...final,
        x: initial.x
      })
    }];
  }, []);
  const dropAnimation = useMemo(() => ({
    ...dropAnimationProp,
    keyframes
  }), [dropAnimationProp, keyframes]);
  const defaultDragOverlayProps = useDefaultDragOverlayProps();
  return /*#__PURE__*/React.createElement(DragOverlayComponentType, _extends({}, defaultDragOverlayProps, {
    dropAnimation: dropAnimation,
    modifiers: [restrictContainerToWindowEdges, restrictKeyboardToVerticalAxis],
    style: {
      ...defaultDragOverlayProps.style,
      ...style
    }
  }), children);
}
