// @wix/wix-style-react-incubator/dist/esm/drag-and-drop/table/TestWrappers.js
import _extends from "@babel/runtime/helpers/extends";
import React, { useEffect } from 'react';
import { useSortable } from '@dnd-kit/sortable';
const dndContext = {};
export const TableTestWrapper = props => {
  const {
    children,
    sortableId
  } = props;
  const {
    props: {
      onDragStart,
      onDragEnd,
      onDragCancel
    }
  } = children;
  useEffect(() => {
    dndContext[sortableId] = {
      onDragStart,
      onDragEnd: onDragEnd,
      onDragCancel: onDragCancel,
      activeId: null,
      activeIndex: null,
      overId: null,
      overIndex: null
    };
    return () => {
      dndContext[sortableId] = null;
    };
  }, [onDragStart, onDragEnd, onDragCancel, sortableId]);
  return children;
};
export const TableRowTestWrapper = function _DraggableTableRow(props) {
  const {
    children,
    index,
    id,
    ...rest
  } = props;
  const {
    type: DraggableTableRow,
    props: draggableTableRowProps
  } = children;
  const {
    data
  } = useSortable({
    id
  });
  const containerId = data.sortable.containerId;
  return /*#__PURE__*/React.createElement(DraggableTableRow, _extends({}, draggableTableRowProps, {
    onPointerDown: () => {
      const dnd = dndContext[containerId];
      if (dnd) {
        dnd.activeId = id;
        dnd.activeIndex = index;
        dnd.onDragStart == null || dnd.onDragStart({
          active: {
            id: dnd.activeId,
            data: {
              current: {
                sortable: {
                  containerId,
                  items: [],
                  index
                }
              }
            }
          }
        });
      }
    },
    onPointerUp: () => {
      const dnd = dndContext[containerId];
      if (dnd) {
        dnd.overId = id;
        dnd.overIndex = index;
        dnd.onDragEnd == null || dnd.onDragEnd({
          active: {
            id: dnd.activeId
          },
          over: {
            id: dnd.overId
          }
        });
      }
      setTimeout(() => {
        dndContext[containerId] = null;
      });
    }
  }, rest, {
    style: {
      ...draggableTableRowProps.style,
      ...rest.style,
      transition: undefined
    }
  }));
};
