// @wix/auto_sdk_payments_payment-acceptance-configurations/build/internal/es/index.mjs
// src/payments-payment-acceptance-configurations-v1-payment-acceptance-configuration-payment-acceptance-configurations.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/payments-payment-acceptance-configurations-v1-payment-acceptance-configuration-payment-acceptance-configurations.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixPaymentsPaymentAcceptanceConfigurationsV1PaymentAcceptanceConfigurationServiceUrl(opts) {
  const domainToMappings = {};
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_payment-acceptance-configurations";
function createPaymentAcceptanceConfiguration(payload) {
  function __createPaymentAcceptanceConfiguration({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "paymentAcceptanceConfiguration.createdDate" },
          { path: "paymentAcceptanceConfiguration.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.payments.payment_acceptance_configurations.v1.payment_acceptance_configuration",
      method: "POST",
      methodFqn: "wix.payments.payment_acceptance_configurations.v1.PaymentAcceptanceConfigurationService.CreatePaymentAcceptanceConfiguration",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPaymentAcceptanceConfigurationsV1PaymentAcceptanceConfigurationServiceUrl(
        {
          protoPath: "/v1/payment-acceptance-configurations",
          data: serializedData,
          host
        }
      ),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "paymentAcceptanceConfiguration.createdDate" },
            { path: "paymentAcceptanceConfiguration.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createPaymentAcceptanceConfiguration;
}
function getPaymentAcceptanceConfiguration(payload) {
  function __getPaymentAcceptanceConfiguration({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.payment_acceptance_configurations.v1.payment_acceptance_configuration",
      method: "GET",
      methodFqn: "wix.payments.payment_acceptance_configurations.v1.PaymentAcceptanceConfigurationService.GetPaymentAcceptanceConfiguration",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPaymentAcceptanceConfigurationsV1PaymentAcceptanceConfigurationServiceUrl(
        {
          protoPath: "/v1/payment-acceptance-configurations/{paymentAcceptanceConfigurationId}",
          data: payload,
          host
        }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "paymentAcceptanceConfiguration.createdDate" },
            { path: "paymentAcceptanceConfiguration.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getPaymentAcceptanceConfiguration;
}
function updatePaymentAcceptanceConfiguration(payload) {
  function __updatePaymentAcceptanceConfiguration({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "paymentAcceptanceConfiguration.createdDate" },
          { path: "paymentAcceptanceConfiguration.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.payments.payment_acceptance_configurations.v1.payment_acceptance_configuration",
      method: "PATCH",
      methodFqn: "wix.payments.payment_acceptance_configurations.v1.PaymentAcceptanceConfigurationService.UpdatePaymentAcceptanceConfiguration",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPaymentAcceptanceConfigurationsV1PaymentAcceptanceConfigurationServiceUrl(
        {
          protoPath: "/v1/payment-acceptance-configurations/{paymentAcceptanceConfiguration.id}",
          data: serializedData,
          host
        }
      ),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "paymentAcceptanceConfiguration.createdDate" },
            { path: "paymentAcceptanceConfiguration.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updatePaymentAcceptanceConfiguration;
}
function deletePaymentAcceptanceConfiguration(payload) {
  function __deletePaymentAcceptanceConfiguration({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.payment_acceptance_configurations.v1.payment_acceptance_configuration",
      method: "DELETE",
      methodFqn: "wix.payments.payment_acceptance_configurations.v1.PaymentAcceptanceConfigurationService.DeletePaymentAcceptanceConfiguration",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPaymentAcceptanceConfigurationsV1PaymentAcceptanceConfigurationServiceUrl(
        {
          protoPath: "/v1/payment-acceptance-configurations/{paymentAcceptanceConfigurationId}",
          data: payload,
          host
        }
      ),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deletePaymentAcceptanceConfiguration;
}
function queryPaymentAcceptanceConfigurations(payload) {
  function __queryPaymentAcceptanceConfigurations({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.payment_acceptance_configurations.v1.payment_acceptance_configuration",
      method: "POST",
      methodFqn: "wix.payments.payment_acceptance_configurations.v1.PaymentAcceptanceConfigurationService.QueryPaymentAcceptanceConfigurations",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPaymentAcceptanceConfigurationsV1PaymentAcceptanceConfigurationServiceUrl(
        {
          protoPath: "/v1/payment-acceptance-configurations/query",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "paymentAcceptanceConfigurations.createdDate" },
            { path: "paymentAcceptanceConfigurations.updatedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixPaymentsPaymentAcceptanceConfigurationsV1PaymentAcceptanceConfigurationServiceUrl(
            {
              protoPath: "/v1/payment-acceptance-configurations/query",
              data: payload,
              host
            }
          ),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryPaymentAcceptanceConfigurations;
}
function bulkUpdatePaymentAcceptanceConfigurationTags(payload) {
  function __bulkUpdatePaymentAcceptanceConfigurationTags({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.payment_acceptance_configurations.v1.payment_acceptance_configuration",
      method: "POST",
      methodFqn: "wix.payments.payment_acceptance_configurations.v1.PaymentAcceptanceConfigurationService.BulkUpdatePaymentAcceptanceConfigurationTags",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPaymentAcceptanceConfigurationsV1PaymentAcceptanceConfigurationServiceUrl(
        {
          protoPath: "/v1/bulk/payment-acceptance-configurations/update-tags",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdatePaymentAcceptanceConfigurationTags;
}
function bulkUpdatePaymentAcceptanceConfigurationTagsByFilter(payload) {
  function __bulkUpdatePaymentAcceptanceConfigurationTagsByFilter({
    host
  }) {
    const metadata = {
      entityFqdn: "wix.payments.payment_acceptance_configurations.v1.payment_acceptance_configuration",
      method: "POST",
      methodFqn: "wix.payments.payment_acceptance_configurations.v1.PaymentAcceptanceConfigurationService.BulkUpdatePaymentAcceptanceConfigurationTagsByFilter",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPaymentAcceptanceConfigurationsV1PaymentAcceptanceConfigurationServiceUrl(
        {
          protoPath: "/v1/bulk/payment-acceptance-configurations/update-tags-by-filter",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdatePaymentAcceptanceConfigurationTagsByFilter;
}

// src/payments-payment-acceptance-configurations-v1-payment-acceptance-configuration-payment-acceptance-configurations.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createPaymentAcceptanceConfiguration2(paymentAcceptanceConfiguration) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    paymentAcceptanceConfiguration
  });
  const reqOpts = createPaymentAcceptanceConfiguration(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.paymentAcceptanceConfiguration;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { paymentAcceptanceConfiguration: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["paymentAcceptanceConfiguration"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getPaymentAcceptanceConfiguration2(paymentAcceptanceConfigurationId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    paymentAcceptanceConfigurationId
  });
  const reqOpts = getPaymentAcceptanceConfiguration(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.paymentAcceptanceConfiguration;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { paymentAcceptanceConfigurationId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["paymentAcceptanceConfigurationId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updatePaymentAcceptanceConfiguration2(_id, paymentAcceptanceConfiguration) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    paymentAcceptanceConfiguration: {
      ...paymentAcceptanceConfiguration,
      id: _id
    }
  });
  const reqOpts = updatePaymentAcceptanceConfiguration(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.paymentAcceptanceConfiguration;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { paymentAcceptanceConfiguration: "$[1]" },
        explicitPathsToArguments: {
          "paymentAcceptanceConfiguration.id": "$[0]"
        },
        singleArgumentUnchanged: false
      },
      ["_id", "paymentAcceptanceConfiguration"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deletePaymentAcceptanceConfiguration2(paymentAcceptanceConfigurationId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    paymentAcceptanceConfigurationId
  });
  const reqOpts = deletePaymentAcceptanceConfiguration(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { paymentAcceptanceConfigurationId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["paymentAcceptanceConfigurationId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryPaymentAcceptanceConfigurations2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryPaymentAcceptanceConfigurations(
        payload
      );
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [])
      );
      return {
        items: transformedData?.paymentAcceptanceConfigurations,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function bulkUpdatePaymentAcceptanceConfigurationTags2(ids, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    ids,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdatePaymentAcceptanceConfigurationTags(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          ids: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["ids", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdatePaymentAcceptanceConfigurationTagsByFilter2(filter, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    filter,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdatePaymentAcceptanceConfigurationTagsByFilter(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          filter: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["filter", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/payments-payment-acceptance-configurations-v1-payment-acceptance-configuration-payment-acceptance-configurations.public.ts
function createPaymentAcceptanceConfiguration3(httpClient) {
  return (paymentAcceptanceConfiguration) => createPaymentAcceptanceConfiguration2(
    paymentAcceptanceConfiguration,
    // @ts-ignore
    { httpClient }
  );
}
function getPaymentAcceptanceConfiguration3(httpClient) {
  return (paymentAcceptanceConfigurationId) => getPaymentAcceptanceConfiguration2(
    paymentAcceptanceConfigurationId,
    // @ts-ignore
    { httpClient }
  );
}
function updatePaymentAcceptanceConfiguration3(httpClient) {
  return (_id, paymentAcceptanceConfiguration) => updatePaymentAcceptanceConfiguration2(
    _id,
    paymentAcceptanceConfiguration,
    // @ts-ignore
    { httpClient }
  );
}
function deletePaymentAcceptanceConfiguration3(httpClient) {
  return (paymentAcceptanceConfigurationId) => deletePaymentAcceptanceConfiguration2(
    paymentAcceptanceConfigurationId,
    // @ts-ignore
    { httpClient }
  );
}
function queryPaymentAcceptanceConfigurations3(httpClient) {
  return () => queryPaymentAcceptanceConfigurations2(
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdatePaymentAcceptanceConfigurationTags3(httpClient) {
  return (ids, options) => bulkUpdatePaymentAcceptanceConfigurationTags2(
    ids,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdatePaymentAcceptanceConfigurationTagsByFilter3(httpClient) {
  return (filter, options) => bulkUpdatePaymentAcceptanceConfigurationTagsByFilter2(
    filter,
    options,
    // @ts-ignore
    { httpClient }
  );
}

// src/payments-payment-acceptance-configurations-v1-payment-acceptance-configuration-payment-acceptance-configurations.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
var createPaymentAcceptanceConfiguration4 = /* @__PURE__ */ createRESTModule(createPaymentAcceptanceConfiguration3);
var getPaymentAcceptanceConfiguration4 = /* @__PURE__ */ createRESTModule(getPaymentAcceptanceConfiguration3);
var updatePaymentAcceptanceConfiguration4 = /* @__PURE__ */ createRESTModule(updatePaymentAcceptanceConfiguration3);
var deletePaymentAcceptanceConfiguration4 = /* @__PURE__ */ createRESTModule(deletePaymentAcceptanceConfiguration3);
var queryPaymentAcceptanceConfigurations4 = /* @__PURE__ */ createRESTModule(queryPaymentAcceptanceConfigurations3);
var bulkUpdatePaymentAcceptanceConfigurationTags4 = /* @__PURE__ */ createRESTModule(
  bulkUpdatePaymentAcceptanceConfigurationTags3
);
var bulkUpdatePaymentAcceptanceConfigurationTagsByFilter4 = /* @__PURE__ */ createRESTModule(
  bulkUpdatePaymentAcceptanceConfigurationTagsByFilter3
);
export {
  SortOrder,
  WebhookIdentityType,
  bulkUpdatePaymentAcceptanceConfigurationTags4 as bulkUpdatePaymentAcceptanceConfigurationTags,
  bulkUpdatePaymentAcceptanceConfigurationTagsByFilter4 as bulkUpdatePaymentAcceptanceConfigurationTagsByFilter,
  createPaymentAcceptanceConfiguration4 as createPaymentAcceptanceConfiguration,
  deletePaymentAcceptanceConfiguration4 as deletePaymentAcceptanceConfiguration,
  getPaymentAcceptanceConfiguration4 as getPaymentAcceptanceConfiguration,
  queryPaymentAcceptanceConfigurations4 as queryPaymentAcceptanceConfigurations,
  updatePaymentAcceptanceConfiguration4 as updatePaymentAcceptanceConfiguration
};
