// @wix/patterns/dist/esm/components/CustomFieldFilters/buildCustomFieldFilterElement.js
import React from 'react';
import { DateCustomFieldFilter } from './DateCustomFieldFilter';
import { DateTimeCustomFieldFilter } from './DateTimeCustomFieldFilter';
import { CheckboxCustomFieldFilter } from './CheckboxCustomFieldFilter';
import { NumberRangeFilter } from '../NumberRangeFilter';
export const buildCustomFieldFilterElement = (state, customField) => {
    const customFieldFilterComponents = {
        date: DateCustomFieldFilter,
        dateTime: DateTimeCustomFieldFilter,
        checkbox: CheckboxCustomFieldFilter,
        integer: NumberRangeFilter,
        decimal: ({ filter }) => React.createElement(NumberRangeFilter, { filter: filter, decimal: true }),
    };
    const CustomFieldFilterComponent = customFieldFilterComponents[customField.type];
    if (!CustomFieldFilterComponent) {
        return null;
    }
    const label = customField.label || customField.id;
    return (React.createElement(CustomFieldFilterComponent, { key: customField.id, filter: state.collection.filters[customField.id], accordionItemProps: {
            label,
        }, toolbarItemProps: {
            label,
        } }));
};
