// @wix/design-system/dist/cjs/AddressInputItem/AddressInputItem.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.default = exports.addressInputItemBuilder = void 0;
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireDefault(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _ListItemSelect = _interopRequireDefault(require("../ListItemSelect"));
var _Text = _interopRequireDefault(require("../Text"));
var _Box = _interopRequireDefault(require("../Box"));
var _wixUiIconsCommon = require("@wix/wix-ui-icons-common");
var _AddressInputItemSt = require("./AddressInputItem.st.css.js");
var _constants = require("./constants");
var _excluded = ["selected", "hovered"];
var _jsxFileName = "/home/builduser/work/57e038ea7326c1ec/packages/wix-design-system/dist/cjs/AddressInputItem/AddressInputItem.tsx",
  _AddressInputItem;
/** This component is used to display an address item mainly in <AddressInput/> component. */
class AddressInputItem extends _react.default.PureComponent {
  render() {
    var {
      dataHook,
      className,
      secondaryLabel,
      optionLayout,
      mainLabel,
      suffix,
      prefix,
      disabled,
      selected,
      highlighted,
      onClick
    } = this.props;
    var textProps = {
      tagName: 'div',
      ellipsis: true,
      skin: disabled ? 'disabled' : 'standard',
      light: selected
    };
    return /*#__PURE__*/_react.default.createElement(_ListItemSelect.default, {
      dataHook: dataHook,
      className: (0, _AddressInputItemSt.st)(_AddressInputItemSt.classes.root, {
        optionLayout
      }, className),
      subtitle: secondaryLabel,
      title: /*#__PURE__*/_react.default.createElement(_Text.default, (0, _extends2.default)({}, textProps, {
        __self: this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 66,
          columnNumber: 16
        }
      }), mainLabel),
      suffix: /*#__PURE__*/_react.default.createElement(_Box.default, {
        __self: this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 67,
          columnNumber: 17
        }
      }, suffix),
      prefix: /*#__PURE__*/_react.default.createElement(_Box.default, {
        __self: this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 68,
          columnNumber: 17
        }
      }, prefix),
      ellipsis: true,
      disabled: disabled,
      selected: selected,
      highlighted: highlighted,
      onClick: onClick,
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 62,
        columnNumber: 7
      }
    });
  }
}
_AddressInputItem = AddressInputItem;
AddressInputItem.displayName = 'AddressInputItem';
AddressInputItem.defaultProps = {
  optionLayout: 'single-line',
  prefix: /*#__PURE__*/_react.default.createElement(_wixUiIconsCommon.Location, {
    __self: _AddressInputItem,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 22,
      columnNumber: 13
    }
  })
};
AddressInputItem.propTypes = {
  dataHook: _propTypes.default.string,
  className: _propTypes.default.string,
  mainLabel: _propTypes.default.string,
  secondaryLabel: _propTypes.default.string,
  optionLayout: _propTypes.default.oneOf(Object.values(_constants.OPTION_LAYOUT)),
  prefix: _propTypes.default.node,
  suffix: _propTypes.default.node,
  selected: _propTypes.default.bool,
  disabled: _propTypes.default.bool,
  highlighted: _propTypes.default.bool,
  onClick: _propTypes.default.func
};
var addressInputItemBuilder = _ref => {
  var {
    id,
    className,
    prefix,
    mainLabel,
    secondaryLabel,
    suffix,
    disabled,
    dataHook,
    optionLayout,
    onClick,
    displayLabel
  } = _ref;
  return {
    id,
    disabled,
    overrideOptionStyle: true,
    label: displayLabel,
    value: _ref2 => {
      var {
          selected,
          hovered
        } = _ref2,
        rest = (0, _objectWithoutProperties2.default)(_ref2, _excluded);
      return /*#__PURE__*/_react.default.createElement(AddressInputItem, (0, _extends2.default)({
        optionLayout: optionLayout,
        dataHook: dataHook,
        className: className,
        prefix: prefix,
        mainLabel: mainLabel,
        secondaryLabel: secondaryLabel,
        suffix: suffix,
        selected: selected,
        highlighted: hovered,
        onClick: onClick
      }, rest, {
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 97,
          columnNumber: 5
        }
      }));
    }
  };
};
exports.addressInputItemBuilder = addressInputItemBuilder;
var _default = exports.default = AddressInputItem;
