// @wix/auto_sdk_events_schedule-bookmarks/build/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  ScheduleStatus: () => ScheduleStatus,
  SortOrder: () => SortOrder,
  StateFilter: () => StateFilter,
  createBookmark: () => createBookmark4,
  deleteBookmark: () => deleteBookmark4,
  listBookmarks: () => listBookmarks4
});
module.exports = __toCommonJS(index_exports);

// src/events-schedule-v1-schedule-item-schedule-bookmarks.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/events-schedule-v1-schedule-item-schedule-bookmarks.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEventsScheduleBookmarkManagementUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/wix-events-agenda",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    _: [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "editor-flow.wixapps.net": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/events/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "api._base_domain_": [
      {
        srcPath: "/events/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_events_schedule-bookmarks";
function listBookmarks(payload) {
  function __listBookmarks({ host }) {
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "GET",
      methodFqn: "wix.events.schedule.BookmarkManagement.ListBookmarks",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleBookmarkManagementUrl({
        protoPath: "/v1/schedule/bookmarks",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "items.createdDate" },
            { path: "items.updatedDate" },
            { path: "items.timeSlot.start" },
            { path: "items.timeSlot.end" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listBookmarks;
}
function createBookmark(payload) {
  function __createBookmark({ host }) {
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "POST",
      methodFqn: "wix.events.schedule.BookmarkManagement.CreateBookmark",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleBookmarkManagementUrl({
        protoPath: "/v1/schedule/{itemId}/bookmark",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __createBookmark;
}
function deleteBookmark(payload) {
  function __deleteBookmark({ host }) {
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "DELETE",
      methodFqn: "wix.events.schedule.BookmarkManagement.DeleteBookmark",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleBookmarkManagementUrl({
        protoPath: "/v1/schedule/{itemId}/bookmark",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteBookmark;
}

// src/events-schedule-v1-schedule-item-schedule-bookmarks.universal.ts
var ScheduleStatus = /* @__PURE__ */ ((ScheduleStatus2) => {
  ScheduleStatus2["SCHEDULED"] = "SCHEDULED";
  ScheduleStatus2["CANCELED"] = "CANCELED";
  return ScheduleStatus2;
})(ScheduleStatus || {});
var StateFilter = /* @__PURE__ */ ((StateFilter2) => {
  StateFilter2["PUBLISHED"] = "PUBLISHED";
  StateFilter2["DRAFT"] = "DRAFT";
  StateFilter2["VISIBLE"] = "VISIBLE";
  StateFilter2["HIDDEN"] = "HIDDEN";
  return StateFilter2;
})(StateFilter || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
async function listBookmarks2(eventId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ eventId });
  const reqOpts = listBookmarks(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { eventId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["eventId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function createBookmark2(itemId, eventId) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    itemId,
    eventId
  });
  const reqOpts = createBookmark(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { itemId: "$[0]", eventId: "$[1]" },
        singleArgumentUnchanged: false
      },
      ["itemId", "eventId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteBookmark2(itemId, eventId) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    itemId,
    eventId
  });
  const reqOpts = deleteBookmark(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { itemId: "$[0]", eventId: "$[1]" },
        singleArgumentUnchanged: false
      },
      ["itemId", "eventId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/events-schedule-v1-schedule-item-schedule-bookmarks.public.ts
function listBookmarks3(httpClient) {
  return (eventId) => listBookmarks2(
    eventId,
    // @ts-ignore
    { httpClient }
  );
}
function createBookmark3(httpClient) {
  return (itemId, eventId) => createBookmark2(
    itemId,
    eventId,
    // @ts-ignore
    { httpClient }
  );
}
function deleteBookmark3(httpClient) {
  return (itemId, eventId) => deleteBookmark2(
    itemId,
    eventId,
    // @ts-ignore
    { httpClient }
  );
}

// src/events-schedule-v1-schedule-item-schedule-bookmarks.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var listBookmarks4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(listBookmarks3);
var createBookmark4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(createBookmark3);
var deleteBookmark4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(deleteBookmark3);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ScheduleStatus,
  SortOrder,
  StateFilter,
  createBookmark,
  deleteBookmark,
  listBookmarks
});
