// @wix/auto_sdk_events_schedule-bookmarks/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  ScheduleStatusOriginal: () => ScheduleStatus,
  SortOrderOriginal: () => SortOrder,
  StateFilterOriginal: () => StateFilter,
  createBookmark: () => createBookmark2,
  deleteBookmark: () => deleteBookmark2,
  listBookmarks: () => listBookmarks2
});
module.exports = __toCommonJS(meta_exports);

// src/events-schedule-v1-schedule-item-schedule-bookmarks.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEventsScheduleBookmarkManagementUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/wix-events-agenda",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    _: [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "editor-flow.wixapps.net": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/events/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "api._base_domain_": [
      {
        srcPath: "/events/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_events_schedule-bookmarks";
function listBookmarks(payload) {
  function __listBookmarks({ host }) {
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "GET",
      methodFqn: "wix.events.schedule.BookmarkManagement.ListBookmarks",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleBookmarkManagementUrl({
        protoPath: "/v1/schedule/bookmarks",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "items.createdDate" },
            { path: "items.updatedDate" },
            { path: "items.timeSlot.start" },
            { path: "items.timeSlot.end" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listBookmarks;
}
function createBookmark(payload) {
  function __createBookmark({ host }) {
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "POST",
      methodFqn: "wix.events.schedule.BookmarkManagement.CreateBookmark",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleBookmarkManagementUrl({
        protoPath: "/v1/schedule/{itemId}/bookmark",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __createBookmark;
}
function deleteBookmark(payload) {
  function __deleteBookmark({ host }) {
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "DELETE",
      methodFqn: "wix.events.schedule.BookmarkManagement.DeleteBookmark",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleBookmarkManagementUrl({
        protoPath: "/v1/schedule/{itemId}/bookmark",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteBookmark;
}

// src/events-schedule-v1-schedule-item-schedule-bookmarks.types.ts
var ScheduleStatus = /* @__PURE__ */ ((ScheduleStatus2) => {
  ScheduleStatus2["SCHEDULED"] = "SCHEDULED";
  ScheduleStatus2["CANCELED"] = "CANCELED";
  return ScheduleStatus2;
})(ScheduleStatus || {});
var StateFilter = /* @__PURE__ */ ((StateFilter2) => {
  StateFilter2["PUBLISHED"] = "PUBLISHED";
  StateFilter2["DRAFT"] = "DRAFT";
  StateFilter2["VISIBLE"] = "VISIBLE";
  StateFilter2["HIDDEN"] = "HIDDEN";
  return StateFilter2;
})(StateFilter || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});

// src/events-schedule-v1-schedule-item-schedule-bookmarks.meta.ts
function listBookmarks2() {
  const payload = {};
  const getRequestOptions = listBookmarks(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/schedule/bookmarks",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createBookmark2() {
  const payload = { itemId: ":itemId" };
  const getRequestOptions = createBookmark(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/schedule/{itemId}/bookmark",
    pathParams: { itemId: "itemId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteBookmark2() {
  const payload = { itemId: ":itemId" };
  const getRequestOptions = deleteBookmark(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/schedule/{itemId}/bookmark",
    pathParams: { itemId: "itemId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ScheduleStatusOriginal,
  SortOrderOriginal,
  StateFilterOriginal,
  createBookmark,
  deleteBookmark,
  listBookmarks
});
