// @wix/auto_sdk_faq_question-entry/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  bulkDeleteQuestionEntries: () => bulkDeleteQuestionEntries2,
  bulkUpdateQuestionEntry: () => bulkUpdateQuestionEntry2,
  createQuestionEntry: () => createQuestionEntry2,
  deleteQuestionEntry: () => deleteQuestionEntry2,
  getQuestionEntry: () => getQuestionEntry2,
  listQuestionEntries: () => listQuestionEntries2,
  queryQuestionEntries: () => queryQuestionEntries2,
  setQuestionEntryLabels: () => setQuestionEntryLabels2,
  updateExtendedFields: () => updateExtendedFields2,
  updateQuestionEntry: () => updateQuestionEntry2
});
module.exports = __toCommonJS(meta_exports);

// src/faq-question-entry-v2-question-entry-question-entry.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixFaqQuestionentryV2QuestionEntryServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/question-entry",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/question-entry",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/question-entry",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/faq-server/v2/question-entries",
        destPath: "/v2/question-entries"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-question-entries",
        destPath: "/v2/draft-question-entries"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/faq-server/v2/question-entries",
        destPath: "/v2/question-entries"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-question-entries",
        destPath: "/v2/draft-question-entries"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/faq/v2/question-entries",
        destPath: "/v2/question-entries"
      },
      {
        srcPath: "/faq/v2/draft-question-entries",
        destPath: "/v2/draft-question-entries"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/faq/v2/question-entries",
        destPath: "/v2/question-entries"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/faq-server/v2/question-entries",
        destPath: "/v2/question-entries"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-question-entries",
        destPath: "/v2/draft-question-entries"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/faq-server/v2/question-entries",
        destPath: "/v2/question-entries"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-question-entries",
        destPath: "/v2/draft-question-entries"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/faq-server/v2/question-entries",
        destPath: "/v2/question-entries"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-question-entries",
        destPath: "/v2/draft-question-entries"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/faq-server/v2/question-entries",
        destPath: "/v2/question-entries"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-question-entries",
        destPath: "/v2/draft-question-entries"
      }
    ],
    _: [
      {
        srcPath: "/_api/faq-server/v2/question-entries",
        destPath: "/v2/question-entries"
      }
    ],
    "dev._base_domain_": [
      {
        srcPath: "/_api/faq-server/v2/question-entries",
        destPath: "/v2/question-entries"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-question-entries",
        destPath: "/v2/draft-question-entries"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/faq-server/v2/question-entries",
        destPath: "/v2/question-entries"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-question-entries",
        destPath: "/v2/draft-question-entries"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_faq_question-entry";
function createQuestionEntry(payload) {
  function __createQuestionEntry({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "questionEntry.createdDate" },
          { path: "questionEntry.updatedDate" },
          { path: "questionEntry.richContent.metadata.createdTimestamp" },
          { path: "questionEntry.richContent.metadata.updatedTimestamp" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "questionEntry.sortOrder" },
          {
            path: "questionEntry.richContent.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "questionEntry.richContent.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "questionEntry.richContent.nodes.galleryData.items.video.thumbnail.duration"
          },
          {
            path: "questionEntry.richContent.nodes.galleryData.options.item.ratio"
          },
          { path: "questionEntry.richContent.nodes.imageData.image.duration" },
          { path: "questionEntry.richContent.nodes.mapData.mapSettings.lat" },
          { path: "questionEntry.richContent.nodes.mapData.mapSettings.lng" },
          {
            path: "questionEntry.richContent.nodes.pollData.poll.image.duration"
          },
          {
            path: "questionEntry.richContent.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "questionEntry.richContent.nodes.pollData.design.poll.background.image.duration"
          },
          {
            path: "questionEntry.richContent.nodes.appEmbedData.image.duration"
          },
          { path: "questionEntry.richContent.nodes.videoData.video.duration" },
          {
            path: "questionEntry.richContent.nodes.videoData.thumbnail.duration"
          },
          { path: "questionEntry.richContent.nodes.audioData.audio.duration" },
          {
            path: "questionEntry.richContent.nodes.audioData.coverImage.duration"
          },
          { path: "questionEntry.labels.sortOrder" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.faq.question_entry.v2.question_entry",
      method: "POST",
      methodFqn: "wix.faq.questionentry.v2.QuestionEntryService.CreateQuestionEntry",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqQuestionentryV2QuestionEntryServiceUrl({
        protoPath: "/v2/question-entries",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "questionEntry.createdDate" },
            { path: "questionEntry.updatedDate" },
            { path: "questionEntry.richContent.metadata.createdTimestamp" },
            { path: "questionEntry.richContent.metadata.updatedTimestamp" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "questionEntry.sortOrder" },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "questionEntry.richContent.nodes.imageData.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "questionEntry.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.videoData.video.duration"
            },
            {
              path: "questionEntry.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "questionEntry.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "questionEntry.richContent.nodes.audioData.coverImage.duration"
            },
            { path: "questionEntry.labels.sortOrder" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createQuestionEntry;
}
function getQuestionEntry(payload) {
  function __getQuestionEntry({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.question_entry.v2.question_entry",
      method: "GET",
      methodFqn: "wix.faq.questionentry.v2.QuestionEntryService.GetQuestionEntry",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqQuestionentryV2QuestionEntryServiceUrl({
        protoPath: "/v2/question-entries/{questionEntryId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "questionEntry.createdDate" },
            { path: "questionEntry.updatedDate" },
            { path: "questionEntry.richContent.metadata.createdTimestamp" },
            { path: "questionEntry.richContent.metadata.updatedTimestamp" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "questionEntry.sortOrder" },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "questionEntry.richContent.nodes.imageData.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "questionEntry.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.videoData.video.duration"
            },
            {
              path: "questionEntry.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "questionEntry.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "questionEntry.richContent.nodes.audioData.coverImage.duration"
            },
            { path: "questionEntry.labels.sortOrder" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getQuestionEntry;
}
function updateQuestionEntry(payload) {
  function __updateQuestionEntry({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "questionEntry.createdDate" },
          { path: "questionEntry.updatedDate" },
          { path: "questionEntry.richContent.metadata.createdTimestamp" },
          { path: "questionEntry.richContent.metadata.updatedTimestamp" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "questionEntry.sortOrder" },
          {
            path: "questionEntry.richContent.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "questionEntry.richContent.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "questionEntry.richContent.nodes.galleryData.items.video.thumbnail.duration"
          },
          {
            path: "questionEntry.richContent.nodes.galleryData.options.item.ratio"
          },
          { path: "questionEntry.richContent.nodes.imageData.image.duration" },
          { path: "questionEntry.richContent.nodes.mapData.mapSettings.lat" },
          { path: "questionEntry.richContent.nodes.mapData.mapSettings.lng" },
          {
            path: "questionEntry.richContent.nodes.pollData.poll.image.duration"
          },
          {
            path: "questionEntry.richContent.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "questionEntry.richContent.nodes.pollData.design.poll.background.image.duration"
          },
          {
            path: "questionEntry.richContent.nodes.appEmbedData.image.duration"
          },
          { path: "questionEntry.richContent.nodes.videoData.video.duration" },
          {
            path: "questionEntry.richContent.nodes.videoData.thumbnail.duration"
          },
          { path: "questionEntry.richContent.nodes.audioData.audio.duration" },
          {
            path: "questionEntry.richContent.nodes.audioData.coverImage.duration"
          },
          { path: "questionEntry.labels.sortOrder" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.faq.question_entry.v2.question_entry",
      method: "PATCH",
      methodFqn: "wix.faq.questionentry.v2.QuestionEntryService.UpdateQuestionEntry",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqQuestionentryV2QuestionEntryServiceUrl({
        protoPath: "/v2/question-entries/{questionEntry.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "questionEntry.createdDate" },
            { path: "questionEntry.updatedDate" },
            { path: "questionEntry.richContent.metadata.createdTimestamp" },
            { path: "questionEntry.richContent.metadata.updatedTimestamp" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "questionEntry.sortOrder" },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "questionEntry.richContent.nodes.imageData.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "questionEntry.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.videoData.video.duration"
            },
            {
              path: "questionEntry.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "questionEntry.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "questionEntry.richContent.nodes.audioData.coverImage.duration"
            },
            { path: "questionEntry.labels.sortOrder" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateQuestionEntry;
}
function deleteQuestionEntry(payload) {
  function __deleteQuestionEntry({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.question_entry.v2.question_entry",
      method: "DELETE",
      methodFqn: "wix.faq.questionentry.v2.QuestionEntryService.DeleteQuestionEntry",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqQuestionentryV2QuestionEntryServiceUrl({
        protoPath: "/v2/question-entries/{questionEntryId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteQuestionEntry;
}
function listQuestionEntries(payload) {
  function __listQuestionEntries({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.question_entry.v2.question_entry",
      method: "GET",
      methodFqn: "wix.faq.questionentry.v2.QuestionEntryService.ListQuestionEntries",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqQuestionentryV2QuestionEntryServiceUrl({
        protoPath: "/v2/question-entries",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "questionEntries.createdDate" },
            { path: "questionEntries.updatedDate" },
            { path: "questionEntries.richContent.metadata.createdTimestamp" },
            { path: "questionEntries.richContent.metadata.updatedTimestamp" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "questionEntries.sortOrder" },
            {
              path: "questionEntries.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "questionEntries.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "questionEntries.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "questionEntries.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "questionEntries.richContent.nodes.imageData.image.duration"
            },
            {
              path: "questionEntries.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "questionEntries.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "questionEntries.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "questionEntries.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "questionEntries.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "questionEntries.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "questionEntries.richContent.nodes.videoData.video.duration"
            },
            {
              path: "questionEntries.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "questionEntries.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "questionEntries.richContent.nodes.audioData.coverImage.duration"
            },
            { path: "questionEntries.labels.sortOrder" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listQuestionEntries;
}
function queryQuestionEntries(payload) {
  function __queryQuestionEntries({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.question_entry.v2.question_entry",
      method: "POST",
      methodFqn: "wix.faq.questionentry.v2.QuestionEntryService.QueryQuestionEntries",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqQuestionentryV2QuestionEntryServiceUrl({
        protoPath: "/v2/question-entries/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "questionEntries.createdDate" },
            { path: "questionEntries.updatedDate" },
            { path: "questionEntries.richContent.metadata.createdTimestamp" },
            { path: "questionEntries.richContent.metadata.updatedTimestamp" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "questionEntries.sortOrder" },
            {
              path: "questionEntries.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "questionEntries.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "questionEntries.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "questionEntries.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "questionEntries.richContent.nodes.imageData.image.duration"
            },
            {
              path: "questionEntries.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "questionEntries.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "questionEntries.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "questionEntries.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "questionEntries.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "questionEntries.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "questionEntries.richContent.nodes.videoData.video.duration"
            },
            {
              path: "questionEntries.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "questionEntries.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "questionEntries.richContent.nodes.audioData.coverImage.duration"
            },
            { path: "questionEntries.labels.sortOrder" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryQuestionEntries;
}
function bulkDeleteQuestionEntries(payload) {
  function __bulkDeleteQuestionEntries({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.question_entry.v2.question_entry",
      method: "POST",
      methodFqn: "wix.faq.questionentry.v2.QuestionEntryService.BulkDeleteQuestionEntries",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqQuestionentryV2QuestionEntryServiceUrl({
        protoPath: "/v2/bulk/question-entries/delete",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkDeleteQuestionEntries;
}
function updateExtendedFields(payload) {
  function __updateExtendedFields({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.question_entry.v2.question_entry",
      method: "POST",
      methodFqn: "wix.faq.questionentry.v2.QuestionEntryService.UpdateExtendedFields",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqQuestionentryV2QuestionEntryServiceUrl({
        protoPath: "/v2/question-entries/{id}/update-extended-fields",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "questionEntry.createdDate" },
            { path: "questionEntry.updatedDate" },
            { path: "questionEntry.richContent.metadata.createdTimestamp" },
            { path: "questionEntry.richContent.metadata.updatedTimestamp" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "questionEntry.sortOrder" },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "questionEntry.richContent.nodes.imageData.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "questionEntry.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.videoData.video.duration"
            },
            {
              path: "questionEntry.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "questionEntry.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "questionEntry.richContent.nodes.audioData.coverImage.duration"
            },
            { path: "questionEntry.labels.sortOrder" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateExtendedFields;
}
function setQuestionEntryLabels(payload) {
  function __setQuestionEntryLabels({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [{ path: "labels.sortOrder" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.faq.question_entry.v2.question_entry",
      method: "PATCH",
      methodFqn: "wix.faq.questionentry.v2.QuestionEntryService.SetQuestionEntryLabels",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqQuestionentryV2QuestionEntryServiceUrl({
        protoPath: "/v2/question-entries/{questionEntryId}/labels",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "questionEntry.createdDate" },
            { path: "questionEntry.updatedDate" },
            { path: "questionEntry.richContent.metadata.createdTimestamp" },
            { path: "questionEntry.richContent.metadata.updatedTimestamp" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "questionEntry.sortOrder" },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "questionEntry.richContent.nodes.imageData.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "questionEntry.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.videoData.video.duration"
            },
            {
              path: "questionEntry.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "questionEntry.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "questionEntry.richContent.nodes.audioData.coverImage.duration"
            },
            { path: "questionEntry.labels.sortOrder" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __setQuestionEntryLabels;
}
function bulkUpdateQuestionEntry(payload) {
  function __bulkUpdateQuestionEntry({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "questionEntries.fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "questionEntries.questionEntry.createdDate" },
          { path: "questionEntries.questionEntry.updatedDate" },
          {
            path: "questionEntries.questionEntry.richContent.metadata.createdTimestamp"
          },
          {
            path: "questionEntries.questionEntry.richContent.metadata.updatedTimestamp"
          }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "questionEntries.questionEntry.sortOrder" },
          {
            path: "questionEntries.questionEntry.richContent.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.galleryData.items.video.thumbnail.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.galleryData.options.item.ratio"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.imageData.image.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.mapData.mapSettings.lat"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.mapData.mapSettings.lng"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.pollData.poll.image.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.pollData.design.poll.background.image.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.appEmbedData.image.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.videoData.video.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.videoData.thumbnail.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.audioData.audio.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.audioData.coverImage.duration"
          },
          { path: "questionEntries.questionEntry.labels.sortOrder" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.faq.question_entry.v2.question_entry",
      method: "POST",
      methodFqn: "wix.faq.questionentry.v2.QuestionEntryService.BulkUpdateQuestionEntry",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqQuestionentryV2QuestionEntryServiceUrl({
        protoPath: "/v2/bulk/question-entries/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.questionEntry.createdDate" },
            { path: "results.questionEntry.updatedDate" },
            {
              path: "results.questionEntry.richContent.metadata.createdTimestamp"
            },
            {
              path: "results.questionEntry.richContent.metadata.updatedTimestamp"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.questionEntry.sortOrder" },
            {
              path: "results.questionEntry.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "results.questionEntry.richContent.nodes.imageData.image.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "results.questionEntry.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "results.questionEntry.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.videoData.video.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.audioData.coverImage.duration"
            },
            { path: "results.questionEntry.labels.sortOrder" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateQuestionEntry;
}

// src/faq-question-entry-v2-question-entry-question-entry.meta.ts
function createQuestionEntry2() {
  const payload = {};
  const getRequestOptions = createQuestionEntry(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/question-entries",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getQuestionEntry2() {
  const payload = { questionEntryId: ":questionEntryId" };
  const getRequestOptions = getQuestionEntry(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/question-entries/{questionEntryId}",
    pathParams: { questionEntryId: "questionEntryId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateQuestionEntry2() {
  const payload = { questionEntry: { id: ":questionEntryId" } };
  const getRequestOptions = updateQuestionEntry(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v2/question-entries/{questionEntry.id}",
    pathParams: { questionEntryId: "questionEntryId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteQuestionEntry2() {
  const payload = { questionEntryId: ":questionEntryId" };
  const getRequestOptions = deleteQuestionEntry(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v2/question-entries/{questionEntryId}",
    pathParams: { questionEntryId: "questionEntryId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listQuestionEntries2() {
  const payload = {};
  const getRequestOptions = listQuestionEntries(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/question-entries",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryQuestionEntries2() {
  const payload = {};
  const getRequestOptions = queryQuestionEntries(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/question-entries/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkDeleteQuestionEntries2() {
  const payload = {};
  const getRequestOptions = bulkDeleteQuestionEntries(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/bulk/question-entries/delete",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateExtendedFields2() {
  const payload = { id: ":id" };
  const getRequestOptions = updateExtendedFields(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/question-entries/{id}/update-extended-fields",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function setQuestionEntryLabels2() {
  const payload = { questionEntryId: ":questionEntryId" };
  const getRequestOptions = setQuestionEntryLabels(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v2/question-entries/{questionEntryId}/labels",
    pathParams: { questionEntryId: "questionEntryId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateQuestionEntry2() {
  const payload = {};
  const getRequestOptions = bulkUpdateQuestionEntry(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/bulk/question-entries/update",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  bulkDeleteQuestionEntries,
  bulkUpdateQuestionEntry,
  createQuestionEntry,
  deleteQuestionEntry,
  getQuestionEntry,
  listQuestionEntries,
  queryQuestionEntries,
  setQuestionEntryLabels,
  updateExtendedFields,
  updateQuestionEntry
});
