// @wix/auto-patterns/dist/cjs/components/modals/actions/BulkDeleteModal.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.BulkDeleteModal = void 0;
var _designSystem = require("@wix/design-system");
var _react = _interopRequireDefault(require("react"));
var _jsxFileName = "/home/builduser/work/73d19ce3378ce4dd/packages/auto-patterns/dist/cjs/components/modals/actions/BulkDeleteModal.tsx";
const BulkDeleteModal = ({
  sdk,
  modalParams,
  actionParams
}) => {
  var _modalParams$title, _modalParams$actions, _modalParams$actions2, _modalParams$descript;
  const {
    getOptimisticActions,
    collectionId,
    closeModal
  } = sdk;
  const optimisticActions = getOptimisticActions(collectionId);
  return /*#__PURE__*/_react.default.createElement(_designSystem.Modal, {
    isOpen: true,
    onRequestClose: closeModal,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 20,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.CustomModalLayout, {
    title: ((_modalParams$title = modalParams.title) == null ? void 0 : _modalParams$title.text) || 'Delete item?',
    theme: "destructive",
    onCloseButtonClick: () => {
      closeModal();
    },
    primaryButtonOnClick: () => {
      var _modalParams$feedback;
      const schema = sdk.getSchema(sdk.collectionId);
      if (!schema) {
        throw new Error('Schema not found');
      }
      const selectedIds = actionParams.selectedValues.map(value => value[schema.idField]);
      optimisticActions == null || optimisticActions.deleteMany(actionParams.selectedValues, {
        successToast: ((_modalParams$feedback = modalParams.feedback) == null || (_modalParams$feedback = _modalParams$feedback.successToast) == null ? void 0 : _modalParams$feedback.text) || 'Items deleted successfully',
        errorToast: () => {
          var _modalParams$feedback2;
          return ((_modalParams$feedback2 = modalParams.feedback) == null || (_modalParams$feedback2 = _modalParams$feedback2.errorToast) == null ? void 0 : _modalParams$feedback2.text) || 'Failed to delete items';
        },
        submit: async () => {
          const result = await schema.actions.bulkDelete(selectedIds);
          return [result];
        }
      });
      closeModal();
    },
    secondaryButtonOnClick: () => {
      closeModal();
    },
    primaryButtonText: ((_modalParams$actions = modalParams.actions) == null || (_modalParams$actions = _modalParams$actions.submit) == null ? void 0 : _modalParams$actions.text) || 'Save',
    secondaryButtonText: ((_modalParams$actions2 = modalParams.actions) == null || (_modalParams$actions2 = _modalParams$actions2.cancel) == null ? void 0 : _modalParams$actions2.text) || 'Cancel',
    content: /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
      direction: "vertical",
      width: "552px",
      gap: "SP2",
      paddingTop: "SP1",
      paddingBottom: "SP1",
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 55,
        columnNumber: 11
      }
    }, /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 62,
        columnNumber: 13
      }
    }, /*#__PURE__*/_react.default.createElement(_designSystem.Text, {
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 63,
        columnNumber: 15
      }
    }, ((_modalParams$descript = modalParams.description) == null ? void 0 : _modalParams$descript.text) || 'This items will be deleted'))),
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 21,
      columnNumber: 7
    }
  }));
};
exports.BulkDeleteModal = BulkDeleteModal;
