// @wix/auto-patterns/dist/cjs/components/modals/actions/EditModal.js
"use strict";

exports.__esModule = true;
exports.EditModal = void 0;
var _designSystem = require("@wix/design-system");
var _react = _interopRequireWildcard(require("react"));
var _jsxFileName = "/home/builduser/work/73d19ce3378ce4dd/packages/auto-patterns/dist/cjs/components/modals/actions/EditModal.tsx";
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
const EditModal = ({
  sdk,
  modalParams,
  actionParams
}) => {
  var _modalParams$title, _modalParams$actions, _modalParams$actions2;
  const [values, setValues] = (0, _react.useState)(modalParams.fields.reduce((acc, field) => {
    acc[field.id] = actionParams.item[field.id];
    return acc;
  }, {}));
  const {
    closeModal
  } = sdk;
  const {
    getOptimisticActions,
    collectionId
  } = sdk;
  const optimisticActions = getOptimisticActions(collectionId);
  return /*#__PURE__*/_react.default.createElement(_designSystem.Modal, {
    isOpen: true,
    onRequestClose: closeModal,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 27,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.CustomModalLayout, {
    title: ((_modalParams$title = modalParams.title) == null ? void 0 : _modalParams$title.text) || 'Edit',
    onCloseButtonClick: () => {
      closeModal();
    },
    primaryButtonOnClick: () => {
      var _modalParams$feedback;
      const schema = sdk.getSchema(sdk.collectionId);
      const updatedItem = {
        ...actionParams.item,
        ...values
      };
      optimisticActions == null || optimisticActions.updateOne(updatedItem, {
        successToast: ((_modalParams$feedback = modalParams.feedback) == null || (_modalParams$feedback = _modalParams$feedback.successToast) == null ? void 0 : _modalParams$feedback.text) || 'Item updated successfully',
        errorToast: () => {
          var _modalParams$feedback2;
          return ((_modalParams$feedback2 = modalParams.feedback) == null || (_modalParams$feedback2 = _modalParams$feedback2.errorToast) == null ? void 0 : _modalParams$feedback2.text) || 'Failed to update item';
        },
        submit: async () => {
          if (!schema) {
            throw new Error('Schema not found');
          }
          const result = await schema.actions.update(updatedItem);
          return [result];
        }
      });
      closeModal();
    },
    secondaryButtonOnClick: () => {
      closeModal();
    },
    primaryButtonText: ((_modalParams$actions = modalParams.actions) == null || (_modalParams$actions = _modalParams$actions.submit) == null ? void 0 : _modalParams$actions.text) || 'Save',
    secondaryButtonText: ((_modalParams$actions2 = modalParams.actions) == null || (_modalParams$actions2 = _modalParams$actions2.cancel) == null ? void 0 : _modalParams$actions2.text) || 'Cancel',
    content: /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
      direction: "vertical",
      width: "552px",
      gap: "SP2",
      paddingTop: "SP1",
      paddingBottom: "SP1",
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 59,
        columnNumber: 11
      }
    }, modalParams.fields.map(field => {
      return /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
        key: field.id,
        direction: "vertical",
        width: "100%",
        gap: "SP2",
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 68,
          columnNumber: 17
        }
      }, /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 69,
          columnNumber: 19
        }
      }, /*#__PURE__*/_react.default.createElement(_designSystem.Text, {
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 70,
          columnNumber: 21
        }
      }, field.label)), /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 72,
          columnNumber: 19
        }
      }, /*#__PURE__*/_react.default.createElement(_designSystem.Input, {
        value: values[field.id],
        onChange: e => {
          setValues(prev => ({
            ...prev,
            [field.id]: e.target.value
          }));
        },
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 73,
          columnNumber: 21
        }
      })));
    })),
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 28,
      columnNumber: 7
    }
  }));
};
exports.EditModal = EditModal;
