// @wix/auto_sdk_pricing-plans_pricing-plans-settings/build/internal/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  TaxableAddress: () => TaxableAddress,
  WebhookIdentityType: () => WebhookIdentityType,
  getPricingPlansSettings: () => getPricingPlansSettings2,
  updatePricingPlansSettings: () => updatePricingPlansSettings2
});
module.exports = __toCommonJS(index_typings_exports);

// src/pricing-plans-settings-v1-pricing-plans-settings-pricing-plans-settings.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/pricing-plans-settings-v1-pricing-plans-settings-pricing-plans-settings.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixPricingPlansSettingsV1PricingPlansSettingsServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/pricing-plans-settings",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_pricing-plans_pricing-plans-settings";
function getPricingPlansSettings(payload) {
  function __getPricingPlansSettings({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.settings.v1.pricing_plans_settings",
      method: "GET",
      methodFqn: "wix.pricing_plans.settings.v1.PricingPlansSettingsService.GetPricingPlansSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixPricingPlansSettingsV1PricingPlansSettingsServiceUrl({
        protoPath: "/v1/pricing-plans-settings",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "pricingPlansSettings.createdDate" },
            { path: "pricingPlansSettings.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getPricingPlansSettings;
}
function updatePricingPlansSettings(payload) {
  function __updatePricingPlansSettings({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "pricingPlansSettings.createdDate" },
          { path: "pricingPlansSettings.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.pricing_plans.settings.v1.pricing_plans_settings",
      method: "PATCH",
      methodFqn: "wix.pricing_plans.settings.v1.PricingPlansSettingsService.UpdatePricingPlansSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixPricingPlansSettingsV1PricingPlansSettingsServiceUrl({
        protoPath: "/v1/pricing-plans-settings",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "pricingPlansSettings.createdDate" },
            { path: "pricingPlansSettings.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updatePricingPlansSettings;
}

// src/pricing-plans-settings-v1-pricing-plans-settings-pricing-plans-settings.universal.ts
var TaxableAddress = /* @__PURE__ */ ((TaxableAddress2) => {
  TaxableAddress2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  TaxableAddress2["BILLING"] = "BILLING";
  TaxableAddress2["BUSINESS"] = "BUSINESS";
  return TaxableAddress2;
})(TaxableAddress || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function getPricingPlansSettings2() {
  const { httpClient, sideEffects } = arguments[0];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({});
  const reqOpts = getPricingPlansSettings(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.pricingPlansSettings;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {},
        singleArgumentUnchanged: false
      },
      []
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updatePricingPlansSettings2(pricingPlansSettings) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    pricingPlansSettings
  });
  const reqOpts = updatePricingPlansSettings(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.pricingPlansSettings;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { pricingPlansSettings: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["pricingPlansSettings"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  TaxableAddress,
  WebhookIdentityType,
  getPricingPlansSettings,
  updatePricingPlansSettings
});
