// pro-gallery/dist/cjs/components/item/itemViewStyleProvider.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getInnerInfoStyle = exports.getOuterInfoStyle = exports.getImageStyle = exports.getContainerStyle = void 0;
const pro_gallery_lib_1 = require("pro-gallery-lib");
function getContainerStyle(options) {
    return Object.assign({}, ((options[pro_gallery_lib_1.optionsMap.layoutParams.info.layout] ===
        pro_gallery_lib_1.GALLERY_CONSTS[pro_gallery_lib_1.optionsMap.layoutParams.info.layout].ATTACHED_BACKGROUND ||
        pro_gallery_lib_1.GALLERY_CONSTS.hasHoverPlacement(options[pro_gallery_lib_1.optionsMap.layoutParams.info.placement])) && Object.assign(Object.assign({}, getBorderStyle(options[pro_gallery_lib_1.optionsMap.stylingParams.itemBorderRadius], options[pro_gallery_lib_1.optionsMap.stylingParams.itemBorderWidth], options[pro_gallery_lib_1.optionsMap.stylingParams.itemBorderColor])), boxShadow(options))));
}
exports.getContainerStyle = getContainerStyle;
function boxShadow(options) {
    let _boxShadow = {};
    if (options[pro_gallery_lib_1.optionsMap.stylingParams.itemEnableShadow]) {
        const itemShadowBlur = options[pro_gallery_lib_1.optionsMap.stylingParams.itemShadowBlur];
        const itemShadowDirection = options[pro_gallery_lib_1.optionsMap.stylingParams.itemShadowDirection];
        const itemShadowSize = options[pro_gallery_lib_1.optionsMap.stylingParams.itemShadowSize];
        const alpha = ((-1 * (Number(itemShadowDirection) - 90)) / 360) * 2 * Math.PI;
        const shadowX = Math.round(itemShadowSize * Math.cos(alpha));
        const shadowY = Math.round(-1 * itemShadowSize * Math.sin(alpha));
        _boxShadow = {
            boxShadow: `${shadowX}px ${shadowY}px ${itemShadowBlur}px ${pro_gallery_lib_1.utils.formatColor(options[pro_gallery_lib_1.optionsMap.stylingParams.itemShadowOpacityAndColor])}`,
        };
    }
    return _boxShadow;
}
function getImageStyle(options) {
    return Object.assign({}, (!pro_gallery_lib_1.GALLERY_CONSTS.hasHoverPlacement(options[pro_gallery_lib_1.optionsMap.layoutParams.info.placement]) &&
        (options[pro_gallery_lib_1.optionsMap.layoutParams.info.layout] ===
            pro_gallery_lib_1.GALLERY_CONSTS[pro_gallery_lib_1.optionsMap.layoutParams.info.layout].NO_BACKGROUND ||
            options[pro_gallery_lib_1.optionsMap.layoutParams.info.layout] ===
                pro_gallery_lib_1.GALLERY_CONSTS[pro_gallery_lib_1.optionsMap.layoutParams.info.layout].SEPARATED_BACKGROUND) && Object.assign({}, getBorderStyle(options[pro_gallery_lib_1.optionsMap.stylingParams.itemBorderRadius], options[pro_gallery_lib_1.optionsMap.stylingParams.itemBorderWidth], options[pro_gallery_lib_1.optionsMap.stylingParams.itemBorderColor]))));
}
exports.getImageStyle = getImageStyle;
function getBorderStyle(borderRadius, borderWidth, borderColor) {
    return Object.assign(Object.assign({ overflow: 'hidden' }, (borderRadius > 0 && { borderRadius })), (borderWidth > 0 && {
        borderWidth: borderWidth + 'px',
        borderColor: pro_gallery_lib_1.utils.formatColor(borderColor),
        borderStyle: 'solid',
    }));
}
function getOuterInfoStyle(placement, options, mediaHeight, infoHeight) {
    const styles = Object.assign(Object.assign({}, (pro_gallery_lib_1.GALLERY_CONSTS.hasExternalHorizontalPlacement(placement) && {
        height: mediaHeight,
        float: pro_gallery_lib_1.GALLERY_CONSTS.isExternalRightPlacement(placement) ? 'right' : 'left',
    })), (pro_gallery_lib_1.GALLERY_CONSTS.hasExternalVerticalPlacement(placement) && {
        height: infoHeight,
        boxSizing: 'content-box',
    }));
    if (options[pro_gallery_lib_1.optionsMap.layoutParams.info.layout] ===
        pro_gallery_lib_1.GALLERY_CONSTS[pro_gallery_lib_1.optionsMap.layoutParams.info.layout].SEPARATED_BACKGROUND) {
        return Object.assign(Object.assign(Object.assign(Object.assign({}, styles), getBorderStyle(options[pro_gallery_lib_1.optionsMap.layoutParams.info.border.radius], options[pro_gallery_lib_1.optionsMap.layoutParams.info.border.width], options[pro_gallery_lib_1.optionsMap.layoutParams.info.border.color])), (pro_gallery_lib_1.GALLERY_CONSTS.hasExternalAbovePlacement(placement) && {
            marginBottom: options[pro_gallery_lib_1.optionsMap.layoutParams.info.spacing],
        })), (pro_gallery_lib_1.GALLERY_CONSTS.hasExternalBelowPlacement(placement) && {
            marginTop: options[pro_gallery_lib_1.optionsMap.layoutParams.info.spacing],
        }));
    }
    return styles;
}
exports.getOuterInfoStyle = getOuterInfoStyle;
function getInnerInfoStylesAboveOrBelow(options, infoHeight) {
    return {
        width: '100%',
        height: infoHeight,
    };
}
function getInnerInfoStylesRightOrLeft(options, infoWidth) {
    return {
        height: '100%',
        width: infoWidth,
    };
}
function getInnerInfoStyle(placement, options, infoHeight, infoWidth) {
    const commonStyles = Object.assign(Object.assign({}, ((options[pro_gallery_lib_1.optionsMap.layoutParams.info.layout] ===
        pro_gallery_lib_1.GALLERY_CONSTS[pro_gallery_lib_1.optionsMap.layoutParams.info.layout].SEPARATED_BACKGROUND ||
        options[pro_gallery_lib_1.optionsMap.layoutParams.info.layout] ===
            pro_gallery_lib_1.GALLERY_CONSTS[pro_gallery_lib_1.optionsMap.layoutParams.info.layout].ATTACHED_BACKGROUND) &&
        options.textBoxFillColor &&
        options.textBoxFillColor.value && {
        backgroundColor: options.textBoxFillColor.value,
    })), { overflow: 'hidden', boxSizing: 'border-box' });
    const infoAboveOrBelow = pro_gallery_lib_1.GALLERY_CONSTS.hasExternalVerticalPlacement(placement);
    const infoRightOrLeft = pro_gallery_lib_1.GALLERY_CONSTS.hasExternalHorizontalPlacement(placement);
    return Object.assign(Object.assign(Object.assign({}, commonStyles), (infoAboveOrBelow && getInnerInfoStylesAboveOrBelow(options, infoHeight))), (infoRightOrLeft && getInnerInfoStylesRightOrLeft(options, infoWidth)));
}
exports.getInnerInfoStyle = getInnerInfoStyle;
