// @wix/auto_sdk_online-programs_steps/build/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  BulkActionType: () => BulkActionType,
  DescriptionFieldSet: () => DescriptionFieldSet,
  Enum: () => Enum,
  SortOrder: () => SortOrder,
  State: () => State,
  StepType: () => StepType,
  WebhookIdentityType: () => WebhookIdentityType,
  bulkCreateStep: () => bulkCreateStep2,
  bulkCreateStepMigration: () => bulkCreateStepMigration2,
  bulkUpdateSteps: () => bulkUpdateSteps2,
  cloneStep: () => cloneStep2,
  createStep: () => createStep2,
  deleteStep: () => deleteStep2,
  getStep: () => getStep2,
  moveStep: () => moveStep2,
  querySteps: () => querySteps2,
  updateStep: () => updateStep2
});
module.exports = __toCommonJS(index_typings_exports);

// src/online-programs-v3-step-steps.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/online-programs-v3-step-steps.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixOnlineprogramsStepsV3StepsServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/program-steps-service",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/program-steps-service",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/program-steps-service",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/program-steps-service",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/program-steps-service",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/program-steps-service",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/program-steps-service",
        destPath: ""
      },
      {
        srcPath: "/program-steps-service",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/program-steps-service",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/program-steps-service",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/program-steps-service",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/program-steps-service",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/program-steps-service",
        destPath: ""
      },
      {
        srcPath: "/_api/program-steps-service",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/program-steps-service",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_online-programs_steps";
function createStep(payload) {
  function __createStep({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "step.createdDate" },
          { path: "step.updatedDate" },
          { path: "step.videoOptions.video.urlExpirationDate" },
          { path: "step.videoOptions.video.resolutions.urlExpirationDate" },
          {
            path: "step.videoOptions.video.resolutions.poster.urlExpirationDate"
          },
          { path: "step.videoOptions.video.posters.urlExpirationDate" },
          { path: "step.videoOptions.coverImage.urlExpirationDate" },
          { path: "step.description.image.urlExpirationDate" },
          { path: "step.description.video.urlExpirationDate" },
          { path: "step.description.video.resolutions.urlExpirationDate" },
          {
            path: "step.description.video.resolutions.poster.urlExpirationDate"
          },
          { path: "step.description.video.posters.urlExpirationDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "step.ordering" },
          { path: "step.videoOptions.requiredCompletionPercentage" },
          { path: "step.videoOptions.video.resolutions.poster.focalPoint.x" },
          { path: "step.videoOptions.video.resolutions.poster.focalPoint.y" },
          { path: "step.videoOptions.video.posters.focalPoint.x" },
          { path: "step.videoOptions.video.posters.focalPoint.y" },
          { path: "step.videoOptions.coverImage.focalPoint.x" },
          { path: "step.videoOptions.coverImage.focalPoint.y" },
          { path: "step.description.image.focalPoint.x" },
          { path: "step.description.image.focalPoint.y" },
          { path: "step.description.video.resolutions.poster.focalPoint.x" },
          { path: "step.description.video.resolutions.poster.focalPoint.y" },
          { path: "step.description.video.posters.focalPoint.x" },
          { path: "step.description.video.posters.focalPoint.y" },
          { path: "step.quizOptions.passingGrade" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.online_programs.v3.step",
      method: "POST",
      methodFqn: "wix.onlineprograms.steps.v3.StepsService.CreateStep",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsStepsV3StepsServiceUrl({
        protoPath: "/v3/steps",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "step.createdDate" },
            { path: "step.updatedDate" },
            { path: "step.videoOptions.video.urlExpirationDate" },
            { path: "step.videoOptions.video.resolutions.urlExpirationDate" },
            {
              path: "step.videoOptions.video.resolutions.poster.urlExpirationDate"
            },
            { path: "step.videoOptions.video.posters.urlExpirationDate" },
            { path: "step.videoOptions.coverImage.urlExpirationDate" },
            { path: "step.description.image.urlExpirationDate" },
            { path: "step.description.video.urlExpirationDate" },
            { path: "step.description.video.resolutions.urlExpirationDate" },
            {
              path: "step.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "step.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "step.ordering" },
            { path: "step.videoOptions.requiredCompletionPercentage" },
            {
              path: "step.videoOptions.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "step.videoOptions.video.resolutions.poster.focalPoint.y"
            },
            { path: "step.videoOptions.video.posters.focalPoint.x" },
            { path: "step.videoOptions.video.posters.focalPoint.y" },
            { path: "step.videoOptions.coverImage.focalPoint.x" },
            { path: "step.videoOptions.coverImage.focalPoint.y" },
            { path: "step.description.image.focalPoint.x" },
            { path: "step.description.image.focalPoint.y" },
            {
              path: "step.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "step.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "step.description.video.posters.focalPoint.x" },
            { path: "step.description.video.posters.focalPoint.y" },
            { path: "step.quizOptions.passingGrade" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createStep;
}
function bulkCreateStep(payload) {
  function __bulkCreateStep({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "steps.createdDate" },
          { path: "steps.updatedDate" },
          { path: "steps.videoOptions.video.urlExpirationDate" },
          { path: "steps.videoOptions.video.resolutions.urlExpirationDate" },
          {
            path: "steps.videoOptions.video.resolutions.poster.urlExpirationDate"
          },
          { path: "steps.videoOptions.video.posters.urlExpirationDate" },
          { path: "steps.videoOptions.coverImage.urlExpirationDate" },
          { path: "steps.description.image.urlExpirationDate" },
          { path: "steps.description.video.urlExpirationDate" },
          { path: "steps.description.video.resolutions.urlExpirationDate" },
          {
            path: "steps.description.video.resolutions.poster.urlExpirationDate"
          },
          { path: "steps.description.video.posters.urlExpirationDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "steps.ordering" },
          { path: "steps.videoOptions.requiredCompletionPercentage" },
          { path: "steps.videoOptions.video.resolutions.poster.focalPoint.x" },
          { path: "steps.videoOptions.video.resolutions.poster.focalPoint.y" },
          { path: "steps.videoOptions.video.posters.focalPoint.x" },
          { path: "steps.videoOptions.video.posters.focalPoint.y" },
          { path: "steps.videoOptions.coverImage.focalPoint.x" },
          { path: "steps.videoOptions.coverImage.focalPoint.y" },
          { path: "steps.description.image.focalPoint.x" },
          { path: "steps.description.image.focalPoint.y" },
          { path: "steps.description.video.resolutions.poster.focalPoint.x" },
          { path: "steps.description.video.resolutions.poster.focalPoint.y" },
          { path: "steps.description.video.posters.focalPoint.x" },
          { path: "steps.description.video.posters.focalPoint.y" },
          { path: "steps.quizOptions.passingGrade" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.online_programs.v3.step",
      method: "POST",
      methodFqn: "wix.onlineprograms.steps.v3.StepsService.BulkCreateStep",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsStepsV3StepsServiceUrl({
        protoPath: "/v3/bulk/steps/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.videoOptions.video.urlExpirationDate" },
            {
              path: "results.item.videoOptions.video.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.videoOptions.video.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.videoOptions.video.posters.urlExpirationDate"
            },
            {
              path: "results.item.videoOptions.coverImage.urlExpirationDate"
            },
            { path: "results.item.description.image.urlExpirationDate" },
            { path: "results.item.description.video.urlExpirationDate" },
            {
              path: "results.item.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.description.video.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.description.video.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.ordering" },
            {
              path: "results.item.videoOptions.requiredCompletionPercentage"
            },
            {
              path: "results.item.videoOptions.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.videoOptions.video.resolutions.poster.focalPoint.y"
            },
            { path: "results.item.videoOptions.video.posters.focalPoint.x" },
            { path: "results.item.videoOptions.video.posters.focalPoint.y" },
            { path: "results.item.videoOptions.coverImage.focalPoint.x" },
            { path: "results.item.videoOptions.coverImage.focalPoint.y" },
            { path: "results.item.description.image.focalPoint.x" },
            { path: "results.item.description.image.focalPoint.y" },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "results.item.description.video.posters.focalPoint.x" },
            { path: "results.item.description.video.posters.focalPoint.y" },
            { path: "results.item.quizOptions.passingGrade" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateStep;
}
function bulkCreateStepMigration(payload) {
  function __bulkCreateStepMigration({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "steps.createdDate" },
          { path: "steps.updatedDate" },
          { path: "steps.videoOptions.video.urlExpirationDate" },
          { path: "steps.videoOptions.video.resolutions.urlExpirationDate" },
          {
            path: "steps.videoOptions.video.resolutions.poster.urlExpirationDate"
          },
          { path: "steps.videoOptions.video.posters.urlExpirationDate" },
          { path: "steps.videoOptions.coverImage.urlExpirationDate" },
          { path: "steps.description.image.urlExpirationDate" },
          { path: "steps.description.video.urlExpirationDate" },
          { path: "steps.description.video.resolutions.urlExpirationDate" },
          {
            path: "steps.description.video.resolutions.poster.urlExpirationDate"
          },
          { path: "steps.description.video.posters.urlExpirationDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "steps.ordering" },
          { path: "steps.videoOptions.requiredCompletionPercentage" },
          { path: "steps.videoOptions.video.resolutions.poster.focalPoint.x" },
          { path: "steps.videoOptions.video.resolutions.poster.focalPoint.y" },
          { path: "steps.videoOptions.video.posters.focalPoint.x" },
          { path: "steps.videoOptions.video.posters.focalPoint.y" },
          { path: "steps.videoOptions.coverImage.focalPoint.x" },
          { path: "steps.videoOptions.coverImage.focalPoint.y" },
          { path: "steps.description.image.focalPoint.x" },
          { path: "steps.description.image.focalPoint.y" },
          { path: "steps.description.video.resolutions.poster.focalPoint.x" },
          { path: "steps.description.video.resolutions.poster.focalPoint.y" },
          { path: "steps.description.video.posters.focalPoint.x" },
          { path: "steps.description.video.posters.focalPoint.y" },
          { path: "steps.quizOptions.passingGrade" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.online_programs.v3.step",
      method: "POST",
      methodFqn: "wix.onlineprograms.steps.v3.StepsService.BulkCreateStepMigration",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsStepsV3StepsServiceUrl({
        protoPath: "/v3/bulk/steps/create/migration",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.videoOptions.video.urlExpirationDate" },
            {
              path: "results.item.videoOptions.video.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.videoOptions.video.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.videoOptions.video.posters.urlExpirationDate"
            },
            {
              path: "results.item.videoOptions.coverImage.urlExpirationDate"
            },
            { path: "results.item.description.image.urlExpirationDate" },
            { path: "results.item.description.video.urlExpirationDate" },
            {
              path: "results.item.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.description.video.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.description.video.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.ordering" },
            {
              path: "results.item.videoOptions.requiredCompletionPercentage"
            },
            {
              path: "results.item.videoOptions.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.videoOptions.video.resolutions.poster.focalPoint.y"
            },
            { path: "results.item.videoOptions.video.posters.focalPoint.x" },
            { path: "results.item.videoOptions.video.posters.focalPoint.y" },
            { path: "results.item.videoOptions.coverImage.focalPoint.x" },
            { path: "results.item.videoOptions.coverImage.focalPoint.y" },
            { path: "results.item.description.image.focalPoint.x" },
            { path: "results.item.description.image.focalPoint.y" },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "results.item.description.video.posters.focalPoint.x" },
            { path: "results.item.description.video.posters.focalPoint.y" },
            { path: "results.item.quizOptions.passingGrade" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateStepMigration;
}
function getStep(payload) {
  function __getStep({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.step",
      method: "GET",
      methodFqn: "wix.onlineprograms.steps.v3.StepsService.GetStep",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsStepsV3StepsServiceUrl({
        protoPath: "/v3/steps/{stepId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "step.createdDate" },
            { path: "step.updatedDate" },
            { path: "step.videoOptions.video.urlExpirationDate" },
            { path: "step.videoOptions.video.resolutions.urlExpirationDate" },
            {
              path: "step.videoOptions.video.resolutions.poster.urlExpirationDate"
            },
            { path: "step.videoOptions.video.posters.urlExpirationDate" },
            { path: "step.videoOptions.coverImage.urlExpirationDate" },
            { path: "step.description.image.urlExpirationDate" },
            { path: "step.description.video.urlExpirationDate" },
            { path: "step.description.video.resolutions.urlExpirationDate" },
            {
              path: "step.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "step.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "step.ordering" },
            { path: "step.videoOptions.requiredCompletionPercentage" },
            {
              path: "step.videoOptions.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "step.videoOptions.video.resolutions.poster.focalPoint.y"
            },
            { path: "step.videoOptions.video.posters.focalPoint.x" },
            { path: "step.videoOptions.video.posters.focalPoint.y" },
            { path: "step.videoOptions.coverImage.focalPoint.x" },
            { path: "step.videoOptions.coverImage.focalPoint.y" },
            { path: "step.description.image.focalPoint.x" },
            { path: "step.description.image.focalPoint.y" },
            {
              path: "step.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "step.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "step.description.video.posters.focalPoint.x" },
            { path: "step.description.video.posters.focalPoint.y" },
            { path: "step.quizOptions.passingGrade" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getStep;
}
function updateStep(payload) {
  function __updateStep({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "step.createdDate" },
          { path: "step.updatedDate" },
          { path: "step.videoOptions.video.urlExpirationDate" },
          { path: "step.videoOptions.video.resolutions.urlExpirationDate" },
          {
            path: "step.videoOptions.video.resolutions.poster.urlExpirationDate"
          },
          { path: "step.videoOptions.video.posters.urlExpirationDate" },
          { path: "step.videoOptions.coverImage.urlExpirationDate" },
          { path: "step.description.image.urlExpirationDate" },
          { path: "step.description.video.urlExpirationDate" },
          { path: "step.description.video.resolutions.urlExpirationDate" },
          {
            path: "step.description.video.resolutions.poster.urlExpirationDate"
          },
          { path: "step.description.video.posters.urlExpirationDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "step.ordering" },
          { path: "step.videoOptions.requiredCompletionPercentage" },
          { path: "step.videoOptions.video.resolutions.poster.focalPoint.x" },
          { path: "step.videoOptions.video.resolutions.poster.focalPoint.y" },
          { path: "step.videoOptions.video.posters.focalPoint.x" },
          { path: "step.videoOptions.video.posters.focalPoint.y" },
          { path: "step.videoOptions.coverImage.focalPoint.x" },
          { path: "step.videoOptions.coverImage.focalPoint.y" },
          { path: "step.description.image.focalPoint.x" },
          { path: "step.description.image.focalPoint.y" },
          { path: "step.description.video.resolutions.poster.focalPoint.x" },
          { path: "step.description.video.resolutions.poster.focalPoint.y" },
          { path: "step.description.video.posters.focalPoint.x" },
          { path: "step.description.video.posters.focalPoint.y" },
          { path: "step.quizOptions.passingGrade" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.online_programs.v3.step",
      method: "PATCH",
      methodFqn: "wix.onlineprograms.steps.v3.StepsService.UpdateStep",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsStepsV3StepsServiceUrl({
        protoPath: "/v3/steps/{step.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "step.createdDate" },
            { path: "step.updatedDate" },
            { path: "step.videoOptions.video.urlExpirationDate" },
            { path: "step.videoOptions.video.resolutions.urlExpirationDate" },
            {
              path: "step.videoOptions.video.resolutions.poster.urlExpirationDate"
            },
            { path: "step.videoOptions.video.posters.urlExpirationDate" },
            { path: "step.videoOptions.coverImage.urlExpirationDate" },
            { path: "step.description.image.urlExpirationDate" },
            { path: "step.description.video.urlExpirationDate" },
            { path: "step.description.video.resolutions.urlExpirationDate" },
            {
              path: "step.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "step.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "step.ordering" },
            { path: "step.videoOptions.requiredCompletionPercentage" },
            {
              path: "step.videoOptions.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "step.videoOptions.video.resolutions.poster.focalPoint.y"
            },
            { path: "step.videoOptions.video.posters.focalPoint.x" },
            { path: "step.videoOptions.video.posters.focalPoint.y" },
            { path: "step.videoOptions.coverImage.focalPoint.x" },
            { path: "step.videoOptions.coverImage.focalPoint.y" },
            { path: "step.description.image.focalPoint.x" },
            { path: "step.description.image.focalPoint.y" },
            {
              path: "step.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "step.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "step.description.video.posters.focalPoint.x" },
            { path: "step.description.video.posters.focalPoint.y" },
            { path: "step.quizOptions.passingGrade" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateStep;
}
function bulkUpdateSteps(payload) {
  function __bulkUpdateSteps({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "steps.fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "steps.step.createdDate" },
          { path: "steps.step.updatedDate" },
          { path: "steps.step.videoOptions.video.urlExpirationDate" },
          {
            path: "steps.step.videoOptions.video.resolutions.urlExpirationDate"
          },
          {
            path: "steps.step.videoOptions.video.resolutions.poster.urlExpirationDate"
          },
          { path: "steps.step.videoOptions.video.posters.urlExpirationDate" },
          { path: "steps.step.videoOptions.coverImage.urlExpirationDate" },
          { path: "steps.step.description.image.urlExpirationDate" },
          { path: "steps.step.description.video.urlExpirationDate" },
          {
            path: "steps.step.description.video.resolutions.urlExpirationDate"
          },
          {
            path: "steps.step.description.video.resolutions.poster.urlExpirationDate"
          },
          { path: "steps.step.description.video.posters.urlExpirationDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "steps.step.ordering" },
          { path: "steps.step.videoOptions.requiredCompletionPercentage" },
          {
            path: "steps.step.videoOptions.video.resolutions.poster.focalPoint.x"
          },
          {
            path: "steps.step.videoOptions.video.resolutions.poster.focalPoint.y"
          },
          { path: "steps.step.videoOptions.video.posters.focalPoint.x" },
          { path: "steps.step.videoOptions.video.posters.focalPoint.y" },
          { path: "steps.step.videoOptions.coverImage.focalPoint.x" },
          { path: "steps.step.videoOptions.coverImage.focalPoint.y" },
          { path: "steps.step.description.image.focalPoint.x" },
          { path: "steps.step.description.image.focalPoint.y" },
          {
            path: "steps.step.description.video.resolutions.poster.focalPoint.x"
          },
          {
            path: "steps.step.description.video.resolutions.poster.focalPoint.y"
          },
          { path: "steps.step.description.video.posters.focalPoint.x" },
          { path: "steps.step.description.video.posters.focalPoint.y" },
          { path: "steps.step.quizOptions.passingGrade" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.online_programs.v3.step",
      method: "POST",
      methodFqn: "wix.onlineprograms.steps.v3.StepsService.BulkUpdateSteps",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsStepsV3StepsServiceUrl({
        protoPath: "/v3/bulk/steps/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.videoOptions.video.urlExpirationDate" },
            {
              path: "results.item.videoOptions.video.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.videoOptions.video.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.videoOptions.video.posters.urlExpirationDate"
            },
            {
              path: "results.item.videoOptions.coverImage.urlExpirationDate"
            },
            { path: "results.item.description.image.urlExpirationDate" },
            { path: "results.item.description.video.urlExpirationDate" },
            {
              path: "results.item.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.description.video.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.description.video.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.ordering" },
            {
              path: "results.item.videoOptions.requiredCompletionPercentage"
            },
            {
              path: "results.item.videoOptions.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.videoOptions.video.resolutions.poster.focalPoint.y"
            },
            { path: "results.item.videoOptions.video.posters.focalPoint.x" },
            { path: "results.item.videoOptions.video.posters.focalPoint.y" },
            { path: "results.item.videoOptions.coverImage.focalPoint.x" },
            { path: "results.item.videoOptions.coverImage.focalPoint.y" },
            { path: "results.item.description.image.focalPoint.x" },
            { path: "results.item.description.image.focalPoint.y" },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "results.item.description.video.posters.focalPoint.x" },
            { path: "results.item.description.video.posters.focalPoint.y" },
            { path: "results.item.quizOptions.passingGrade" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateSteps;
}
function deleteStep(payload) {
  function __deleteStep({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.step",
      method: "DELETE",
      methodFqn: "wix.onlineprograms.steps.v3.StepsService.DeleteStep",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsStepsV3StepsServiceUrl({
        protoPath: "/v3/steps/{stepId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteStep;
}
function querySteps(payload) {
  function __querySteps({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.step",
      method: "POST",
      methodFqn: "wix.onlineprograms.steps.v3.StepsService.QuerySteps",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsStepsV3StepsServiceUrl({
        protoPath: "/v3/steps/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "steps.createdDate" },
            { path: "steps.updatedDate" },
            { path: "steps.videoOptions.video.urlExpirationDate" },
            {
              path: "steps.videoOptions.video.resolutions.urlExpirationDate"
            },
            {
              path: "steps.videoOptions.video.resolutions.poster.urlExpirationDate"
            },
            { path: "steps.videoOptions.video.posters.urlExpirationDate" },
            { path: "steps.videoOptions.coverImage.urlExpirationDate" },
            { path: "steps.description.image.urlExpirationDate" },
            { path: "steps.description.video.urlExpirationDate" },
            { path: "steps.description.video.resolutions.urlExpirationDate" },
            {
              path: "steps.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "steps.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "steps.ordering" },
            { path: "steps.videoOptions.requiredCompletionPercentage" },
            {
              path: "steps.videoOptions.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "steps.videoOptions.video.resolutions.poster.focalPoint.y"
            },
            { path: "steps.videoOptions.video.posters.focalPoint.x" },
            { path: "steps.videoOptions.video.posters.focalPoint.y" },
            { path: "steps.videoOptions.coverImage.focalPoint.x" },
            { path: "steps.videoOptions.coverImage.focalPoint.y" },
            { path: "steps.description.image.focalPoint.x" },
            { path: "steps.description.image.focalPoint.y" },
            {
              path: "steps.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "steps.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "steps.description.video.posters.focalPoint.x" },
            { path: "steps.description.video.posters.focalPoint.y" },
            { path: "steps.quizOptions.passingGrade" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __querySteps;
}
function moveStep(payload) {
  function __moveStep({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.step",
      method: "PATCH",
      methodFqn: "wix.onlineprograms.steps.v3.StepsService.MoveStep",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsStepsV3StepsServiceUrl({
        protoPath: "/v3/steps/{stepId}/move",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "step.createdDate" },
            { path: "step.updatedDate" },
            { path: "step.videoOptions.video.urlExpirationDate" },
            { path: "step.videoOptions.video.resolutions.urlExpirationDate" },
            {
              path: "step.videoOptions.video.resolutions.poster.urlExpirationDate"
            },
            { path: "step.videoOptions.video.posters.urlExpirationDate" },
            { path: "step.videoOptions.coverImage.urlExpirationDate" },
            { path: "step.description.image.urlExpirationDate" },
            { path: "step.description.video.urlExpirationDate" },
            { path: "step.description.video.resolutions.urlExpirationDate" },
            {
              path: "step.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "step.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "step.ordering" },
            { path: "step.videoOptions.requiredCompletionPercentage" },
            {
              path: "step.videoOptions.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "step.videoOptions.video.resolutions.poster.focalPoint.y"
            },
            { path: "step.videoOptions.video.posters.focalPoint.x" },
            { path: "step.videoOptions.video.posters.focalPoint.y" },
            { path: "step.videoOptions.coverImage.focalPoint.x" },
            { path: "step.videoOptions.coverImage.focalPoint.y" },
            { path: "step.description.image.focalPoint.x" },
            { path: "step.description.image.focalPoint.y" },
            {
              path: "step.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "step.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "step.description.video.posters.focalPoint.x" },
            { path: "step.description.video.posters.focalPoint.y" },
            { path: "step.quizOptions.passingGrade" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __moveStep;
}
function cloneStep(payload) {
  function __cloneStep({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.step",
      method: "POST",
      methodFqn: "wix.onlineprograms.steps.v3.StepsService.CloneStep",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsStepsV3StepsServiceUrl({
        protoPath: "/v3/steps/{stepId}/clone",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "step.createdDate" },
            { path: "step.updatedDate" },
            { path: "step.videoOptions.video.urlExpirationDate" },
            { path: "step.videoOptions.video.resolutions.urlExpirationDate" },
            {
              path: "step.videoOptions.video.resolutions.poster.urlExpirationDate"
            },
            { path: "step.videoOptions.video.posters.urlExpirationDate" },
            { path: "step.videoOptions.coverImage.urlExpirationDate" },
            { path: "step.description.image.urlExpirationDate" },
            { path: "step.description.video.urlExpirationDate" },
            { path: "step.description.video.resolutions.urlExpirationDate" },
            {
              path: "step.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "step.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "step.ordering" },
            { path: "step.videoOptions.requiredCompletionPercentage" },
            {
              path: "step.videoOptions.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "step.videoOptions.video.resolutions.poster.focalPoint.y"
            },
            { path: "step.videoOptions.video.posters.focalPoint.x" },
            { path: "step.videoOptions.video.posters.focalPoint.y" },
            { path: "step.videoOptions.coverImage.focalPoint.x" },
            { path: "step.videoOptions.coverImage.focalPoint.y" },
            { path: "step.description.image.focalPoint.x" },
            { path: "step.description.image.focalPoint.y" },
            {
              path: "step.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "step.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "step.description.video.posters.focalPoint.x" },
            { path: "step.description.video.posters.focalPoint.y" },
            { path: "step.quizOptions.passingGrade" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __cloneStep;
}

// src/online-programs-v3-step-steps.universal.ts
var import_image = require("@wix/sdk-runtime/transformations/image");
var import_image2 = require("@wix/sdk-runtime/transformations/image");
var import_video_v2 = require("@wix/sdk-runtime/transformations/video-v2");
var import_video_v22 = require("@wix/sdk-runtime/transformations/video-v2");
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var StepType = /* @__PURE__ */ ((StepType2) => {
  StepType2["ARTICLE"] = "ARTICLE";
  StepType2["VIDEO"] = "VIDEO";
  StepType2["QUIZ"] = "QUIZ";
  return StepType2;
})(StepType || {});
var Enum = /* @__PURE__ */ ((Enum2) => {
  Enum2["UNKNOWN"] = "UNKNOWN";
  Enum2["PROTECTED"] = "PROTECTED";
  Enum2["PRIVATE"] = "PRIVATE";
  Enum2["PUBLIC"] = "PUBLIC";
  Enum2["UNCHANGED"] = "UNCHANGED";
  Enum2["GROUP_PROTECTED"] = "GROUP_PROTECTED";
  return Enum2;
})(Enum || {});
var BulkActionType = /* @__PURE__ */ ((BulkActionType2) => {
  BulkActionType2["UNKNOWN_ACTION_TYPE"] = "UNKNOWN_ACTION_TYPE";
  BulkActionType2["INSERT"] = "INSERT";
  BulkActionType2["UPDATE"] = "UPDATE";
  BulkActionType2["DELETE"] = "DELETE";
  return BulkActionType2;
})(BulkActionType || {});
var DescriptionFieldSet = /* @__PURE__ */ ((DescriptionFieldSet2) => {
  DescriptionFieldSet2["STANDARD"] = "STANDARD";
  DescriptionFieldSet2["EXTENDED"] = "EXTENDED";
  return DescriptionFieldSet2;
})(DescriptionFieldSet || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var State = /* @__PURE__ */ ((State2) => {
  State2["Unknown"] = "Unknown";
  State2["Off"] = "Off";
  State2["On"] = "On";
  return State2;
})(State || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createStep2(step, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      step,
      programId: options?.programId
    }),
    [
      {
        transformFn: import_video_v2.transformSDKVideoV2ToRESTVideoV2,
        paths: [
          { path: "step.videoOptions.video" },
          { path: "step.description.video" }
        ]
      },
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [
          { path: "step.videoOptions.coverImage" },
          { path: "step.description.image" }
        ]
      }
    ]
  );
  const reqOpts = createStep(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [
            { path: "step.videoOptions.video" },
            { path: "step.description.video" }
          ]
        },
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "step.videoOptions.coverImage" },
            { path: "step.description.image" }
          ]
        }
      ])
    )?.step;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { step: "$[0]", programId: "$[1].programId" },
        singleArgumentUnchanged: false
      },
      ["step", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCreateStep2(steps, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      steps,
      returnEntity: options?.returnEntity,
      programId: options?.programId
    }),
    [
      {
        transformFn: import_video_v2.transformSDKVideoV2ToRESTVideoV2,
        paths: [
          { path: "steps.videoOptions.video" },
          { path: "steps.description.video" }
        ]
      },
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [
          { path: "steps.videoOptions.coverImage" },
          { path: "steps.description.image" }
        ]
      }
    ]
  );
  const reqOpts = bulkCreateStep(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [
            { path: "results.item.videoOptions.video" },
            { path: "results.item.description.video" }
          ]
        },
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "results.item.videoOptions.coverImage" },
            { path: "results.item.description.image" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          steps: "$[0]",
          returnEntity: "$[1].returnEntity",
          programId: "$[1].programId"
        },
        singleArgumentUnchanged: false
      },
      ["steps", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCreateStepMigration2(steps, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      steps,
      programId: options?.programId
    }),
    [
      {
        transformFn: import_video_v2.transformSDKVideoV2ToRESTVideoV2,
        paths: [
          { path: "steps.videoOptions.video" },
          { path: "steps.description.video" }
        ]
      },
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [
          { path: "steps.videoOptions.coverImage" },
          { path: "steps.description.image" }
        ]
      }
    ]
  );
  const reqOpts = bulkCreateStepMigration(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [
            { path: "results.item.videoOptions.video" },
            { path: "results.item.description.video" }
          ]
        },
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "results.item.videoOptions.coverImage" },
            { path: "results.item.description.image" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          steps: "$[0]",
          programId: "$[1].programId"
        },
        singleArgumentUnchanged: false
      },
      ["steps", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getStep2(stepId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    stepId,
    descriptionFieldSet: options?.descriptionFieldSet
  });
  const reqOpts = getStep(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [
            { path: "step.videoOptions.video" },
            { path: "step.description.video" }
          ]
        },
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "step.videoOptions.coverImage" },
            { path: "step.description.image" }
          ]
        }
      ])
    )?.step;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          stepId: "$[0]",
          descriptionFieldSet: "$[1].descriptionFieldSet"
        },
        singleArgumentUnchanged: false
      },
      ["stepId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateStep2(_id, step) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ step: { ...step, id: _id } }),
    [
      {
        transformFn: import_video_v2.transformSDKVideoV2ToRESTVideoV2,
        paths: [
          { path: "step.videoOptions.video" },
          { path: "step.description.video" }
        ]
      },
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [
          { path: "step.videoOptions.coverImage" },
          { path: "step.description.image" }
        ]
      }
    ]
  );
  const reqOpts = updateStep(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [
            { path: "step.videoOptions.video" },
            { path: "step.description.video" }
          ]
        },
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "step.videoOptions.coverImage" },
            { path: "step.description.image" }
          ]
        }
      ])
    )?.step;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { step: "$[1]" },
        explicitPathsToArguments: { "step.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "step"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateSteps2(steps, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      steps,
      returnEntity: options?.returnEntity
    }),
    [
      {
        transformFn: import_video_v2.transformSDKVideoV2ToRESTVideoV2,
        paths: [
          { path: "steps.step.videoOptions.video" },
          { path: "steps.step.description.video" }
        ]
      },
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [
          { path: "steps.step.videoOptions.coverImage" },
          { path: "steps.step.description.image" }
        ]
      }
    ]
  );
  const reqOpts = bulkUpdateSteps(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [
            { path: "results.item.videoOptions.video" },
            { path: "results.item.description.video" }
          ]
        },
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "results.item.videoOptions.coverImage" },
            { path: "results.item.description.image" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          steps: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["steps", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteStep2(stepId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ stepId });
  const reqOpts = deleteStep(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { stepId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["stepId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function querySteps2(options) {
  const { httpClient, sideEffects } = arguments[1];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = querySteps({
        ...payload,
        ...options ?? {}
      });
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, options];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [
          {
            transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
            paths: [
              { path: "steps.videoOptions.video" },
              { path: "steps.description.video" }
            ]
          },
          {
            transformFn: import_image2.transformRESTImageToSDKImage,
            paths: [
              { path: "steps.videoOptions.coverImage" },
              { path: "steps.description.image" }
            ]
          }
        ])
      );
      return {
        items: transformedData?.steps,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function moveStep2(stepId, sectionId, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    stepId,
    sectionId,
    afterStepId: options?.afterStepId,
    revision: options?.revision
  });
  const reqOpts = moveStep(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [
            { path: "step.videoOptions.video" },
            { path: "step.description.video" }
          ]
        },
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "step.videoOptions.coverImage" },
            { path: "step.description.image" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          stepId: "$[0]",
          sectionId: "$[1]",
          afterStepId: "$[2].afterStepId",
          revision: "$[2].revision"
        },
        singleArgumentUnchanged: false
      },
      ["stepId", "sectionId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function cloneStep2(stepId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ stepId });
  const reqOpts = cloneStep(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [
            { path: "step.videoOptions.video" },
            { path: "step.description.video" }
          ]
        },
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "step.videoOptions.coverImage" },
            { path: "step.description.image" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { stepId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["stepId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  BulkActionType,
  DescriptionFieldSet,
  Enum,
  SortOrder,
  State,
  StepType,
  WebhookIdentityType,
  bulkCreateStep,
  bulkCreateStepMigration,
  bulkUpdateSteps,
  cloneStep,
  createStep,
  deleteStep,
  getStep,
  moveStep,
  querySteps,
  updateStep
});
