// @wix/auto_sdk_online-programs_steps/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  BulkActionTypeOriginal: () => BulkActionType,
  DescriptionFieldSetOriginal: () => DescriptionFieldSet,
  EnumOriginal: () => Enum,
  SortOrderOriginal: () => SortOrder,
  StateOriginal: () => State,
  StepTypeOriginal: () => StepType,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  bulkCreateStep: () => bulkCreateStep2,
  bulkCreateStepMigration: () => bulkCreateStepMigration2,
  bulkUpdateSteps: () => bulkUpdateSteps2,
  cloneStep: () => cloneStep2,
  createStep: () => createStep2,
  deleteStep: () => deleteStep2,
  getStep: () => getStep2,
  moveStep: () => moveStep2,
  querySteps: () => querySteps2,
  updateStep: () => updateStep2
});
module.exports = __toCommonJS(meta_exports);

// src/online-programs-v3-step-steps.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixOnlineprogramsStepsV3StepsServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/program-steps-service",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/program-steps-service",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/program-steps-service",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/program-steps-service",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/program-steps-service",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/program-steps-service",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/program-steps-service",
        destPath: ""
      },
      {
        srcPath: "/program-steps-service",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/program-steps-service",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/program-steps-service",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/program-steps-service",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/program-steps-service",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/program-steps-service",
        destPath: ""
      },
      {
        srcPath: "/_api/program-steps-service",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/program-steps-service",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_online-programs_steps";
function createStep(payload) {
  function __createStep({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "step.createdDate" },
          { path: "step.updatedDate" },
          { path: "step.videoOptions.video.urlExpirationDate" },
          { path: "step.videoOptions.video.resolutions.urlExpirationDate" },
          {
            path: "step.videoOptions.video.resolutions.poster.urlExpirationDate"
          },
          { path: "step.videoOptions.video.posters.urlExpirationDate" },
          { path: "step.videoOptions.coverImage.urlExpirationDate" },
          { path: "step.description.image.urlExpirationDate" },
          { path: "step.description.video.urlExpirationDate" },
          { path: "step.description.video.resolutions.urlExpirationDate" },
          {
            path: "step.description.video.resolutions.poster.urlExpirationDate"
          },
          { path: "step.description.video.posters.urlExpirationDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "step.ordering" },
          { path: "step.videoOptions.requiredCompletionPercentage" },
          { path: "step.videoOptions.video.resolutions.poster.focalPoint.x" },
          { path: "step.videoOptions.video.resolutions.poster.focalPoint.y" },
          { path: "step.videoOptions.video.posters.focalPoint.x" },
          { path: "step.videoOptions.video.posters.focalPoint.y" },
          { path: "step.videoOptions.coverImage.focalPoint.x" },
          { path: "step.videoOptions.coverImage.focalPoint.y" },
          { path: "step.description.image.focalPoint.x" },
          { path: "step.description.image.focalPoint.y" },
          { path: "step.description.video.resolutions.poster.focalPoint.x" },
          { path: "step.description.video.resolutions.poster.focalPoint.y" },
          { path: "step.description.video.posters.focalPoint.x" },
          { path: "step.description.video.posters.focalPoint.y" },
          { path: "step.quizOptions.passingGrade" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.online_programs.v3.step",
      method: "POST",
      methodFqn: "wix.onlineprograms.steps.v3.StepsService.CreateStep",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsStepsV3StepsServiceUrl({
        protoPath: "/v3/steps",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "step.createdDate" },
            { path: "step.updatedDate" },
            { path: "step.videoOptions.video.urlExpirationDate" },
            { path: "step.videoOptions.video.resolutions.urlExpirationDate" },
            {
              path: "step.videoOptions.video.resolutions.poster.urlExpirationDate"
            },
            { path: "step.videoOptions.video.posters.urlExpirationDate" },
            { path: "step.videoOptions.coverImage.urlExpirationDate" },
            { path: "step.description.image.urlExpirationDate" },
            { path: "step.description.video.urlExpirationDate" },
            { path: "step.description.video.resolutions.urlExpirationDate" },
            {
              path: "step.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "step.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "step.ordering" },
            { path: "step.videoOptions.requiredCompletionPercentage" },
            {
              path: "step.videoOptions.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "step.videoOptions.video.resolutions.poster.focalPoint.y"
            },
            { path: "step.videoOptions.video.posters.focalPoint.x" },
            { path: "step.videoOptions.video.posters.focalPoint.y" },
            { path: "step.videoOptions.coverImage.focalPoint.x" },
            { path: "step.videoOptions.coverImage.focalPoint.y" },
            { path: "step.description.image.focalPoint.x" },
            { path: "step.description.image.focalPoint.y" },
            {
              path: "step.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "step.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "step.description.video.posters.focalPoint.x" },
            { path: "step.description.video.posters.focalPoint.y" },
            { path: "step.quizOptions.passingGrade" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createStep;
}
function bulkCreateStep(payload) {
  function __bulkCreateStep({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "steps.createdDate" },
          { path: "steps.updatedDate" },
          { path: "steps.videoOptions.video.urlExpirationDate" },
          { path: "steps.videoOptions.video.resolutions.urlExpirationDate" },
          {
            path: "steps.videoOptions.video.resolutions.poster.urlExpirationDate"
          },
          { path: "steps.videoOptions.video.posters.urlExpirationDate" },
          { path: "steps.videoOptions.coverImage.urlExpirationDate" },
          { path: "steps.description.image.urlExpirationDate" },
          { path: "steps.description.video.urlExpirationDate" },
          { path: "steps.description.video.resolutions.urlExpirationDate" },
          {
            path: "steps.description.video.resolutions.poster.urlExpirationDate"
          },
          { path: "steps.description.video.posters.urlExpirationDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "steps.ordering" },
          { path: "steps.videoOptions.requiredCompletionPercentage" },
          { path: "steps.videoOptions.video.resolutions.poster.focalPoint.x" },
          { path: "steps.videoOptions.video.resolutions.poster.focalPoint.y" },
          { path: "steps.videoOptions.video.posters.focalPoint.x" },
          { path: "steps.videoOptions.video.posters.focalPoint.y" },
          { path: "steps.videoOptions.coverImage.focalPoint.x" },
          { path: "steps.videoOptions.coverImage.focalPoint.y" },
          { path: "steps.description.image.focalPoint.x" },
          { path: "steps.description.image.focalPoint.y" },
          { path: "steps.description.video.resolutions.poster.focalPoint.x" },
          { path: "steps.description.video.resolutions.poster.focalPoint.y" },
          { path: "steps.description.video.posters.focalPoint.x" },
          { path: "steps.description.video.posters.focalPoint.y" },
          { path: "steps.quizOptions.passingGrade" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.online_programs.v3.step",
      method: "POST",
      methodFqn: "wix.onlineprograms.steps.v3.StepsService.BulkCreateStep",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsStepsV3StepsServiceUrl({
        protoPath: "/v3/bulk/steps/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.videoOptions.video.urlExpirationDate" },
            {
              path: "results.item.videoOptions.video.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.videoOptions.video.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.videoOptions.video.posters.urlExpirationDate"
            },
            {
              path: "results.item.videoOptions.coverImage.urlExpirationDate"
            },
            { path: "results.item.description.image.urlExpirationDate" },
            { path: "results.item.description.video.urlExpirationDate" },
            {
              path: "results.item.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.description.video.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.description.video.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.ordering" },
            {
              path: "results.item.videoOptions.requiredCompletionPercentage"
            },
            {
              path: "results.item.videoOptions.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.videoOptions.video.resolutions.poster.focalPoint.y"
            },
            { path: "results.item.videoOptions.video.posters.focalPoint.x" },
            { path: "results.item.videoOptions.video.posters.focalPoint.y" },
            { path: "results.item.videoOptions.coverImage.focalPoint.x" },
            { path: "results.item.videoOptions.coverImage.focalPoint.y" },
            { path: "results.item.description.image.focalPoint.x" },
            { path: "results.item.description.image.focalPoint.y" },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "results.item.description.video.posters.focalPoint.x" },
            { path: "results.item.description.video.posters.focalPoint.y" },
            { path: "results.item.quizOptions.passingGrade" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateStep;
}
function bulkCreateStepMigration(payload) {
  function __bulkCreateStepMigration({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "steps.createdDate" },
          { path: "steps.updatedDate" },
          { path: "steps.videoOptions.video.urlExpirationDate" },
          { path: "steps.videoOptions.video.resolutions.urlExpirationDate" },
          {
            path: "steps.videoOptions.video.resolutions.poster.urlExpirationDate"
          },
          { path: "steps.videoOptions.video.posters.urlExpirationDate" },
          { path: "steps.videoOptions.coverImage.urlExpirationDate" },
          { path: "steps.description.image.urlExpirationDate" },
          { path: "steps.description.video.urlExpirationDate" },
          { path: "steps.description.video.resolutions.urlExpirationDate" },
          {
            path: "steps.description.video.resolutions.poster.urlExpirationDate"
          },
          { path: "steps.description.video.posters.urlExpirationDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "steps.ordering" },
          { path: "steps.videoOptions.requiredCompletionPercentage" },
          { path: "steps.videoOptions.video.resolutions.poster.focalPoint.x" },
          { path: "steps.videoOptions.video.resolutions.poster.focalPoint.y" },
          { path: "steps.videoOptions.video.posters.focalPoint.x" },
          { path: "steps.videoOptions.video.posters.focalPoint.y" },
          { path: "steps.videoOptions.coverImage.focalPoint.x" },
          { path: "steps.videoOptions.coverImage.focalPoint.y" },
          { path: "steps.description.image.focalPoint.x" },
          { path: "steps.description.image.focalPoint.y" },
          { path: "steps.description.video.resolutions.poster.focalPoint.x" },
          { path: "steps.description.video.resolutions.poster.focalPoint.y" },
          { path: "steps.description.video.posters.focalPoint.x" },
          { path: "steps.description.video.posters.focalPoint.y" },
          { path: "steps.quizOptions.passingGrade" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.online_programs.v3.step",
      method: "POST",
      methodFqn: "wix.onlineprograms.steps.v3.StepsService.BulkCreateStepMigration",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsStepsV3StepsServiceUrl({
        protoPath: "/v3/bulk/steps/create/migration",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.videoOptions.video.urlExpirationDate" },
            {
              path: "results.item.videoOptions.video.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.videoOptions.video.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.videoOptions.video.posters.urlExpirationDate"
            },
            {
              path: "results.item.videoOptions.coverImage.urlExpirationDate"
            },
            { path: "results.item.description.image.urlExpirationDate" },
            { path: "results.item.description.video.urlExpirationDate" },
            {
              path: "results.item.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.description.video.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.description.video.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.ordering" },
            {
              path: "results.item.videoOptions.requiredCompletionPercentage"
            },
            {
              path: "results.item.videoOptions.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.videoOptions.video.resolutions.poster.focalPoint.y"
            },
            { path: "results.item.videoOptions.video.posters.focalPoint.x" },
            { path: "results.item.videoOptions.video.posters.focalPoint.y" },
            { path: "results.item.videoOptions.coverImage.focalPoint.x" },
            { path: "results.item.videoOptions.coverImage.focalPoint.y" },
            { path: "results.item.description.image.focalPoint.x" },
            { path: "results.item.description.image.focalPoint.y" },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "results.item.description.video.posters.focalPoint.x" },
            { path: "results.item.description.video.posters.focalPoint.y" },
            { path: "results.item.quizOptions.passingGrade" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateStepMigration;
}
function getStep(payload) {
  function __getStep({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.step",
      method: "GET",
      methodFqn: "wix.onlineprograms.steps.v3.StepsService.GetStep",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsStepsV3StepsServiceUrl({
        protoPath: "/v3/steps/{stepId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "step.createdDate" },
            { path: "step.updatedDate" },
            { path: "step.videoOptions.video.urlExpirationDate" },
            { path: "step.videoOptions.video.resolutions.urlExpirationDate" },
            {
              path: "step.videoOptions.video.resolutions.poster.urlExpirationDate"
            },
            { path: "step.videoOptions.video.posters.urlExpirationDate" },
            { path: "step.videoOptions.coverImage.urlExpirationDate" },
            { path: "step.description.image.urlExpirationDate" },
            { path: "step.description.video.urlExpirationDate" },
            { path: "step.description.video.resolutions.urlExpirationDate" },
            {
              path: "step.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "step.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "step.ordering" },
            { path: "step.videoOptions.requiredCompletionPercentage" },
            {
              path: "step.videoOptions.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "step.videoOptions.video.resolutions.poster.focalPoint.y"
            },
            { path: "step.videoOptions.video.posters.focalPoint.x" },
            { path: "step.videoOptions.video.posters.focalPoint.y" },
            { path: "step.videoOptions.coverImage.focalPoint.x" },
            { path: "step.videoOptions.coverImage.focalPoint.y" },
            { path: "step.description.image.focalPoint.x" },
            { path: "step.description.image.focalPoint.y" },
            {
              path: "step.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "step.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "step.description.video.posters.focalPoint.x" },
            { path: "step.description.video.posters.focalPoint.y" },
            { path: "step.quizOptions.passingGrade" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getStep;
}
function updateStep(payload) {
  function __updateStep({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "step.createdDate" },
          { path: "step.updatedDate" },
          { path: "step.videoOptions.video.urlExpirationDate" },
          { path: "step.videoOptions.video.resolutions.urlExpirationDate" },
          {
            path: "step.videoOptions.video.resolutions.poster.urlExpirationDate"
          },
          { path: "step.videoOptions.video.posters.urlExpirationDate" },
          { path: "step.videoOptions.coverImage.urlExpirationDate" },
          { path: "step.description.image.urlExpirationDate" },
          { path: "step.description.video.urlExpirationDate" },
          { path: "step.description.video.resolutions.urlExpirationDate" },
          {
            path: "step.description.video.resolutions.poster.urlExpirationDate"
          },
          { path: "step.description.video.posters.urlExpirationDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "step.ordering" },
          { path: "step.videoOptions.requiredCompletionPercentage" },
          { path: "step.videoOptions.video.resolutions.poster.focalPoint.x" },
          { path: "step.videoOptions.video.resolutions.poster.focalPoint.y" },
          { path: "step.videoOptions.video.posters.focalPoint.x" },
          { path: "step.videoOptions.video.posters.focalPoint.y" },
          { path: "step.videoOptions.coverImage.focalPoint.x" },
          { path: "step.videoOptions.coverImage.focalPoint.y" },
          { path: "step.description.image.focalPoint.x" },
          { path: "step.description.image.focalPoint.y" },
          { path: "step.description.video.resolutions.poster.focalPoint.x" },
          { path: "step.description.video.resolutions.poster.focalPoint.y" },
          { path: "step.description.video.posters.focalPoint.x" },
          { path: "step.description.video.posters.focalPoint.y" },
          { path: "step.quizOptions.passingGrade" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.online_programs.v3.step",
      method: "PATCH",
      methodFqn: "wix.onlineprograms.steps.v3.StepsService.UpdateStep",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsStepsV3StepsServiceUrl({
        protoPath: "/v3/steps/{step.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "step.createdDate" },
            { path: "step.updatedDate" },
            { path: "step.videoOptions.video.urlExpirationDate" },
            { path: "step.videoOptions.video.resolutions.urlExpirationDate" },
            {
              path: "step.videoOptions.video.resolutions.poster.urlExpirationDate"
            },
            { path: "step.videoOptions.video.posters.urlExpirationDate" },
            { path: "step.videoOptions.coverImage.urlExpirationDate" },
            { path: "step.description.image.urlExpirationDate" },
            { path: "step.description.video.urlExpirationDate" },
            { path: "step.description.video.resolutions.urlExpirationDate" },
            {
              path: "step.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "step.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "step.ordering" },
            { path: "step.videoOptions.requiredCompletionPercentage" },
            {
              path: "step.videoOptions.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "step.videoOptions.video.resolutions.poster.focalPoint.y"
            },
            { path: "step.videoOptions.video.posters.focalPoint.x" },
            { path: "step.videoOptions.video.posters.focalPoint.y" },
            { path: "step.videoOptions.coverImage.focalPoint.x" },
            { path: "step.videoOptions.coverImage.focalPoint.y" },
            { path: "step.description.image.focalPoint.x" },
            { path: "step.description.image.focalPoint.y" },
            {
              path: "step.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "step.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "step.description.video.posters.focalPoint.x" },
            { path: "step.description.video.posters.focalPoint.y" },
            { path: "step.quizOptions.passingGrade" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateStep;
}
function bulkUpdateSteps(payload) {
  function __bulkUpdateSteps({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "steps.fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "steps.step.createdDate" },
          { path: "steps.step.updatedDate" },
          { path: "steps.step.videoOptions.video.urlExpirationDate" },
          {
            path: "steps.step.videoOptions.video.resolutions.urlExpirationDate"
          },
          {
            path: "steps.step.videoOptions.video.resolutions.poster.urlExpirationDate"
          },
          { path: "steps.step.videoOptions.video.posters.urlExpirationDate" },
          { path: "steps.step.videoOptions.coverImage.urlExpirationDate" },
          { path: "steps.step.description.image.urlExpirationDate" },
          { path: "steps.step.description.video.urlExpirationDate" },
          {
            path: "steps.step.description.video.resolutions.urlExpirationDate"
          },
          {
            path: "steps.step.description.video.resolutions.poster.urlExpirationDate"
          },
          { path: "steps.step.description.video.posters.urlExpirationDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "steps.step.ordering" },
          { path: "steps.step.videoOptions.requiredCompletionPercentage" },
          {
            path: "steps.step.videoOptions.video.resolutions.poster.focalPoint.x"
          },
          {
            path: "steps.step.videoOptions.video.resolutions.poster.focalPoint.y"
          },
          { path: "steps.step.videoOptions.video.posters.focalPoint.x" },
          { path: "steps.step.videoOptions.video.posters.focalPoint.y" },
          { path: "steps.step.videoOptions.coverImage.focalPoint.x" },
          { path: "steps.step.videoOptions.coverImage.focalPoint.y" },
          { path: "steps.step.description.image.focalPoint.x" },
          { path: "steps.step.description.image.focalPoint.y" },
          {
            path: "steps.step.description.video.resolutions.poster.focalPoint.x"
          },
          {
            path: "steps.step.description.video.resolutions.poster.focalPoint.y"
          },
          { path: "steps.step.description.video.posters.focalPoint.x" },
          { path: "steps.step.description.video.posters.focalPoint.y" },
          { path: "steps.step.quizOptions.passingGrade" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.online_programs.v3.step",
      method: "POST",
      methodFqn: "wix.onlineprograms.steps.v3.StepsService.BulkUpdateSteps",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsStepsV3StepsServiceUrl({
        protoPath: "/v3/bulk/steps/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.videoOptions.video.urlExpirationDate" },
            {
              path: "results.item.videoOptions.video.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.videoOptions.video.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.videoOptions.video.posters.urlExpirationDate"
            },
            {
              path: "results.item.videoOptions.coverImage.urlExpirationDate"
            },
            { path: "results.item.description.image.urlExpirationDate" },
            { path: "results.item.description.video.urlExpirationDate" },
            {
              path: "results.item.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.description.video.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.description.video.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.ordering" },
            {
              path: "results.item.videoOptions.requiredCompletionPercentage"
            },
            {
              path: "results.item.videoOptions.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.videoOptions.video.resolutions.poster.focalPoint.y"
            },
            { path: "results.item.videoOptions.video.posters.focalPoint.x" },
            { path: "results.item.videoOptions.video.posters.focalPoint.y" },
            { path: "results.item.videoOptions.coverImage.focalPoint.x" },
            { path: "results.item.videoOptions.coverImage.focalPoint.y" },
            { path: "results.item.description.image.focalPoint.x" },
            { path: "results.item.description.image.focalPoint.y" },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "results.item.description.video.posters.focalPoint.x" },
            { path: "results.item.description.video.posters.focalPoint.y" },
            { path: "results.item.quizOptions.passingGrade" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateSteps;
}
function deleteStep(payload) {
  function __deleteStep({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.step",
      method: "DELETE",
      methodFqn: "wix.onlineprograms.steps.v3.StepsService.DeleteStep",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsStepsV3StepsServiceUrl({
        protoPath: "/v3/steps/{stepId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteStep;
}
function querySteps(payload) {
  function __querySteps({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.step",
      method: "POST",
      methodFqn: "wix.onlineprograms.steps.v3.StepsService.QuerySteps",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsStepsV3StepsServiceUrl({
        protoPath: "/v3/steps/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "steps.createdDate" },
            { path: "steps.updatedDate" },
            { path: "steps.videoOptions.video.urlExpirationDate" },
            {
              path: "steps.videoOptions.video.resolutions.urlExpirationDate"
            },
            {
              path: "steps.videoOptions.video.resolutions.poster.urlExpirationDate"
            },
            { path: "steps.videoOptions.video.posters.urlExpirationDate" },
            { path: "steps.videoOptions.coverImage.urlExpirationDate" },
            { path: "steps.description.image.urlExpirationDate" },
            { path: "steps.description.video.urlExpirationDate" },
            { path: "steps.description.video.resolutions.urlExpirationDate" },
            {
              path: "steps.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "steps.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "steps.ordering" },
            { path: "steps.videoOptions.requiredCompletionPercentage" },
            {
              path: "steps.videoOptions.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "steps.videoOptions.video.resolutions.poster.focalPoint.y"
            },
            { path: "steps.videoOptions.video.posters.focalPoint.x" },
            { path: "steps.videoOptions.video.posters.focalPoint.y" },
            { path: "steps.videoOptions.coverImage.focalPoint.x" },
            { path: "steps.videoOptions.coverImage.focalPoint.y" },
            { path: "steps.description.image.focalPoint.x" },
            { path: "steps.description.image.focalPoint.y" },
            {
              path: "steps.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "steps.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "steps.description.video.posters.focalPoint.x" },
            { path: "steps.description.video.posters.focalPoint.y" },
            { path: "steps.quizOptions.passingGrade" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __querySteps;
}
function moveStep(payload) {
  function __moveStep({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.step",
      method: "PATCH",
      methodFqn: "wix.onlineprograms.steps.v3.StepsService.MoveStep",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsStepsV3StepsServiceUrl({
        protoPath: "/v3/steps/{stepId}/move",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "step.createdDate" },
            { path: "step.updatedDate" },
            { path: "step.videoOptions.video.urlExpirationDate" },
            { path: "step.videoOptions.video.resolutions.urlExpirationDate" },
            {
              path: "step.videoOptions.video.resolutions.poster.urlExpirationDate"
            },
            { path: "step.videoOptions.video.posters.urlExpirationDate" },
            { path: "step.videoOptions.coverImage.urlExpirationDate" },
            { path: "step.description.image.urlExpirationDate" },
            { path: "step.description.video.urlExpirationDate" },
            { path: "step.description.video.resolutions.urlExpirationDate" },
            {
              path: "step.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "step.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "step.ordering" },
            { path: "step.videoOptions.requiredCompletionPercentage" },
            {
              path: "step.videoOptions.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "step.videoOptions.video.resolutions.poster.focalPoint.y"
            },
            { path: "step.videoOptions.video.posters.focalPoint.x" },
            { path: "step.videoOptions.video.posters.focalPoint.y" },
            { path: "step.videoOptions.coverImage.focalPoint.x" },
            { path: "step.videoOptions.coverImage.focalPoint.y" },
            { path: "step.description.image.focalPoint.x" },
            { path: "step.description.image.focalPoint.y" },
            {
              path: "step.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "step.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "step.description.video.posters.focalPoint.x" },
            { path: "step.description.video.posters.focalPoint.y" },
            { path: "step.quizOptions.passingGrade" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __moveStep;
}
function cloneStep(payload) {
  function __cloneStep({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.step",
      method: "POST",
      methodFqn: "wix.onlineprograms.steps.v3.StepsService.CloneStep",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsStepsV3StepsServiceUrl({
        protoPath: "/v3/steps/{stepId}/clone",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "step.createdDate" },
            { path: "step.updatedDate" },
            { path: "step.videoOptions.video.urlExpirationDate" },
            { path: "step.videoOptions.video.resolutions.urlExpirationDate" },
            {
              path: "step.videoOptions.video.resolutions.poster.urlExpirationDate"
            },
            { path: "step.videoOptions.video.posters.urlExpirationDate" },
            { path: "step.videoOptions.coverImage.urlExpirationDate" },
            { path: "step.description.image.urlExpirationDate" },
            { path: "step.description.video.urlExpirationDate" },
            { path: "step.description.video.resolutions.urlExpirationDate" },
            {
              path: "step.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "step.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "step.ordering" },
            { path: "step.videoOptions.requiredCompletionPercentage" },
            {
              path: "step.videoOptions.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "step.videoOptions.video.resolutions.poster.focalPoint.y"
            },
            { path: "step.videoOptions.video.posters.focalPoint.x" },
            { path: "step.videoOptions.video.posters.focalPoint.y" },
            { path: "step.videoOptions.coverImage.focalPoint.x" },
            { path: "step.videoOptions.coverImage.focalPoint.y" },
            { path: "step.description.image.focalPoint.x" },
            { path: "step.description.image.focalPoint.y" },
            {
              path: "step.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "step.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "step.description.video.posters.focalPoint.x" },
            { path: "step.description.video.posters.focalPoint.y" },
            { path: "step.quizOptions.passingGrade" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __cloneStep;
}

// src/online-programs-v3-step-steps.types.ts
var StepType = /* @__PURE__ */ ((StepType2) => {
  StepType2["ARTICLE"] = "ARTICLE";
  StepType2["VIDEO"] = "VIDEO";
  StepType2["QUIZ"] = "QUIZ";
  return StepType2;
})(StepType || {});
var Enum = /* @__PURE__ */ ((Enum2) => {
  Enum2["UNKNOWN"] = "UNKNOWN";
  Enum2["PROTECTED"] = "PROTECTED";
  Enum2["PRIVATE"] = "PRIVATE";
  Enum2["PUBLIC"] = "PUBLIC";
  Enum2["UNCHANGED"] = "UNCHANGED";
  Enum2["GROUP_PROTECTED"] = "GROUP_PROTECTED";
  return Enum2;
})(Enum || {});
var BulkActionType = /* @__PURE__ */ ((BulkActionType2) => {
  BulkActionType2["UNKNOWN_ACTION_TYPE"] = "UNKNOWN_ACTION_TYPE";
  BulkActionType2["INSERT"] = "INSERT";
  BulkActionType2["UPDATE"] = "UPDATE";
  BulkActionType2["DELETE"] = "DELETE";
  return BulkActionType2;
})(BulkActionType || {});
var DescriptionFieldSet = /* @__PURE__ */ ((DescriptionFieldSet2) => {
  DescriptionFieldSet2["STANDARD"] = "STANDARD";
  DescriptionFieldSet2["EXTENDED"] = "EXTENDED";
  return DescriptionFieldSet2;
})(DescriptionFieldSet || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var State = /* @__PURE__ */ ((State2) => {
  State2["Unknown"] = "Unknown";
  State2["Off"] = "Off";
  State2["On"] = "On";
  return State2;
})(State || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/online-programs-v3-step-steps.meta.ts
function createStep2() {
  const payload = {};
  const getRequestOptions = createStep(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/steps",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkCreateStep2() {
  const payload = {};
  const getRequestOptions = bulkCreateStep(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/bulk/steps/create",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkCreateStepMigration2() {
  const payload = {};
  const getRequestOptions = bulkCreateStepMigration(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/bulk/steps/create/migration",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getStep2() {
  const payload = { stepId: ":stepId" };
  const getRequestOptions = getStep(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/steps/{stepId}",
    pathParams: { stepId: "stepId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateStep2() {
  const payload = { step: { id: ":stepId" } };
  const getRequestOptions = updateStep(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v3/steps/{step.id}",
    pathParams: { stepId: "stepId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateSteps2() {
  const payload = {};
  const getRequestOptions = bulkUpdateSteps(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/bulk/steps/update",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteStep2() {
  const payload = { stepId: ":stepId" };
  const getRequestOptions = deleteStep(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v3/steps/{stepId}",
    pathParams: { stepId: "stepId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function querySteps2() {
  const payload = {};
  const getRequestOptions = querySteps(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/steps/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function moveStep2() {
  const payload = { stepId: ":stepId" };
  const getRequestOptions = moveStep(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v3/steps/{stepId}/move",
    pathParams: { stepId: "stepId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function cloneStep2() {
  const payload = { stepId: ":stepId" };
  const getRequestOptions = cloneStep(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/steps/{stepId}/clone",
    pathParams: { stepId: "stepId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  BulkActionTypeOriginal,
  DescriptionFieldSetOriginal,
  EnumOriginal,
  SortOrderOriginal,
  StateOriginal,
  StepTypeOriginal,
  WebhookIdentityTypeOriginal,
  bulkCreateStep,
  bulkCreateStepMigration,
  bulkUpdateSteps,
  cloneStep,
  createStep,
  deleteStep,
  getStep,
  moveStep,
  querySteps,
  updateStep
});
