// @wix/auto_sdk_payments_dispute-history-records/build/internal/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  DisputeChannel: () => DisputeChannel,
  DisputeStage: () => DisputeStage,
  DisputeStatus: () => DisputeStatus,
  SortOrder: () => SortOrder,
  WebhookIdentityType: () => WebhookIdentityType,
  bulkUpdateDisputeHistoryRecordTags: () => bulkUpdateDisputeHistoryRecordTags2,
  bulkUpdateDisputeHistoryRecordTagsByFilter: () => bulkUpdateDisputeHistoryRecordTagsByFilter2,
  getDisputeHistoryRecord: () => getDisputeHistoryRecord2,
  queryDisputeHistoryRecords: () => queryDisputeHistoryRecords2
});
module.exports = __toCommonJS(index_typings_exports);

// src/payments-dispute-history-records-v1-dispute-history-record-dispute-history-records.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/payments-dispute-history-records-v1-dispute-history-record-dispute-history-records.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixPaymentsDisputeHistoryRecordsV1DisputeHistoryRecordServiceUrl(opts) {
  const domainToMappings = {};
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_dispute-history-records";
function getDisputeHistoryRecord(payload) {
  function __getDisputeHistoryRecord({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_history_records.v1.dispute_history_record",
      method: "GET",
      methodFqn: "wix.payments.dispute_history_records.v1.DisputeHistoryRecordService.GetDisputeHistoryRecord",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeHistoryRecordsV1DisputeHistoryRecordServiceUrl(
        {
          protoPath: "/v1/dispute-history-records/{disputeHistoryRecordId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "disputeHistoryRecord.createdDate" },
            { path: "disputeHistoryRecord.updatedDate" },
            { path: "disputeHistoryRecord.dueDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getDisputeHistoryRecord;
}
function queryDisputeHistoryRecords(payload) {
  function __queryDisputeHistoryRecords({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_history_records.v1.dispute_history_record",
      method: "POST",
      methodFqn: "wix.payments.dispute_history_records.v1.DisputeHistoryRecordService.QueryDisputeHistoryRecords",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeHistoryRecordsV1DisputeHistoryRecordServiceUrl(
        { protoPath: "/v1/dispute-history-records/query", data: payload, host }
      ),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "disputeHistoryRecords.createdDate" },
            { path: "disputeHistoryRecords.updatedDate" },
            { path: "disputeHistoryRecords.dueDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixPaymentsDisputeHistoryRecordsV1DisputeHistoryRecordServiceUrl(
            {
              protoPath: "/v1/dispute-history-records/query",
              data: payload,
              host
            }
          ),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryDisputeHistoryRecords;
}
function bulkUpdateDisputeHistoryRecordTags(payload) {
  function __bulkUpdateDisputeHistoryRecordTags({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_history_records.v1.dispute_history_record",
      method: "POST",
      methodFqn: "wix.payments.dispute_history_records.v1.DisputeHistoryRecordService.BulkUpdateDisputeHistoryRecordTags",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeHistoryRecordsV1DisputeHistoryRecordServiceUrl(
        {
          protoPath: "/v1/bulk/dispute-history-records/update-tags",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateDisputeHistoryRecordTags;
}
function bulkUpdateDisputeHistoryRecordTagsByFilter(payload) {
  function __bulkUpdateDisputeHistoryRecordTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_history_records.v1.dispute_history_record",
      method: "POST",
      methodFqn: "wix.payments.dispute_history_records.v1.DisputeHistoryRecordService.BulkUpdateDisputeHistoryRecordTagsByFilter",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeHistoryRecordsV1DisputeHistoryRecordServiceUrl(
        {
          protoPath: "/v1/bulk/dispute-history-records/update-tags-by-filter",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateDisputeHistoryRecordTagsByFilter;
}

// src/payments-dispute-history-records-v1-dispute-history-record-dispute-history-records.universal.ts
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var DisputeStage = /* @__PURE__ */ ((DisputeStage2) => {
  DisputeStage2["UNKNOWN_DISPUTE_STAGE"] = "UNKNOWN_DISPUTE_STAGE";
  DisputeStage2["RFI"] = "RFI";
  DisputeStage2["CHARGEBACK"] = "CHARGEBACK";
  return DisputeStage2;
})(DisputeStage || {});
var DisputeStatus = /* @__PURE__ */ ((DisputeStatus2) => {
  DisputeStatus2["UNKNOWN_DISPUTE_STATUS"] = "UNKNOWN_DISPUTE_STATUS";
  DisputeStatus2["WAITING_MERCHANT"] = "WAITING_MERCHANT";
  DisputeStatus2["UNDER_REVIEW"] = "UNDER_REVIEW";
  DisputeStatus2["WAITING_BUYER"] = "WAITING_BUYER";
  DisputeStatus2["WON"] = "WON";
  DisputeStatus2["LOST"] = "LOST";
  return DisputeStatus2;
})(DisputeStatus || {});
var DisputeChannel = /* @__PURE__ */ ((DisputeChannel2) => {
  DisputeChannel2["UNKNOWN_DISPUTE_CHANNEL"] = "UNKNOWN_DISPUTE_CHANNEL";
  DisputeChannel2["INTERNAL"] = "INTERNAL";
  DisputeChannel2["EXTERNAL"] = "EXTERNAL";
  return DisputeChannel2;
})(DisputeChannel || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function getDisputeHistoryRecord2(disputeHistoryRecordId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    disputeHistoryRecordId
  });
  const reqOpts = getDisputeHistoryRecord(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.disputeHistoryRecord;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { disputeHistoryRecordId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["disputeHistoryRecordId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryDisputeHistoryRecords2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryDisputeHistoryRecords(
        payload
      );
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [])
      );
      return {
        items: transformedData?.disputeHistoryRecords,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function bulkUpdateDisputeHistoryRecordTags2(ids, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    ids,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateDisputeHistoryRecordTags(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          ids: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["ids", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateDisputeHistoryRecordTagsByFilter2(filter, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    filter,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateDisputeHistoryRecordTagsByFilter(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          filter: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["filter", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  DisputeChannel,
  DisputeStage,
  DisputeStatus,
  SortOrder,
  WebhookIdentityType,
  bulkUpdateDisputeHistoryRecordTags,
  bulkUpdateDisputeHistoryRecordTagsByFilter,
  getDisputeHistoryRecord,
  queryDisputeHistoryRecords
});
