// @wix/auto_sdk_payments_dispute-history-records/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  bulkUpdateDisputeHistoryRecordTags: () => bulkUpdateDisputeHistoryRecordTags2,
  bulkUpdateDisputeHistoryRecordTagsByFilter: () => bulkUpdateDisputeHistoryRecordTagsByFilter2,
  getDisputeHistoryRecord: () => getDisputeHistoryRecord2,
  queryDisputeHistoryRecords: () => queryDisputeHistoryRecords2
});
module.exports = __toCommonJS(meta_exports);

// src/payments-dispute-history-records-v1-dispute-history-record-dispute-history-records.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixPaymentsDisputeHistoryRecordsV1DisputeHistoryRecordServiceUrl(opts) {
  const domainToMappings = {};
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_dispute-history-records";
function getDisputeHistoryRecord(payload) {
  function __getDisputeHistoryRecord({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_history_records.v1.dispute_history_record",
      method: "GET",
      methodFqn: "wix.payments.dispute_history_records.v1.DisputeHistoryRecordService.GetDisputeHistoryRecord",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeHistoryRecordsV1DisputeHistoryRecordServiceUrl(
        {
          protoPath: "/v1/dispute-history-records/{disputeHistoryRecordId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "disputeHistoryRecord.createdDate" },
            { path: "disputeHistoryRecord.updatedDate" },
            { path: "disputeHistoryRecord.dueDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getDisputeHistoryRecord;
}
function queryDisputeHistoryRecords(payload) {
  function __queryDisputeHistoryRecords({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_history_records.v1.dispute_history_record",
      method: "POST",
      methodFqn: "wix.payments.dispute_history_records.v1.DisputeHistoryRecordService.QueryDisputeHistoryRecords",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeHistoryRecordsV1DisputeHistoryRecordServiceUrl(
        { protoPath: "/v1/dispute-history-records/query", data: payload, host }
      ),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "disputeHistoryRecords.createdDate" },
            { path: "disputeHistoryRecords.updatedDate" },
            { path: "disputeHistoryRecords.dueDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixPaymentsDisputeHistoryRecordsV1DisputeHistoryRecordServiceUrl(
            {
              protoPath: "/v1/dispute-history-records/query",
              data: payload,
              host
            }
          ),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryDisputeHistoryRecords;
}
function bulkUpdateDisputeHistoryRecordTags(payload) {
  function __bulkUpdateDisputeHistoryRecordTags({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_history_records.v1.dispute_history_record",
      method: "POST",
      methodFqn: "wix.payments.dispute_history_records.v1.DisputeHistoryRecordService.BulkUpdateDisputeHistoryRecordTags",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeHistoryRecordsV1DisputeHistoryRecordServiceUrl(
        {
          protoPath: "/v1/bulk/dispute-history-records/update-tags",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateDisputeHistoryRecordTags;
}
function bulkUpdateDisputeHistoryRecordTagsByFilter(payload) {
  function __bulkUpdateDisputeHistoryRecordTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_history_records.v1.dispute_history_record",
      method: "POST",
      methodFqn: "wix.payments.dispute_history_records.v1.DisputeHistoryRecordService.BulkUpdateDisputeHistoryRecordTagsByFilter",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeHistoryRecordsV1DisputeHistoryRecordServiceUrl(
        {
          protoPath: "/v1/bulk/dispute-history-records/update-tags-by-filter",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateDisputeHistoryRecordTagsByFilter;
}

// src/payments-dispute-history-records-v1-dispute-history-record-dispute-history-records.meta.ts
function getDisputeHistoryRecord2() {
  const payload = { disputeHistoryRecordId: ":disputeHistoryRecordId" };
  const getRequestOptions = getDisputeHistoryRecord(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/dispute-history-records/{disputeHistoryRecordId}",
    pathParams: { disputeHistoryRecordId: "disputeHistoryRecordId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryDisputeHistoryRecords2() {
  const payload = {};
  const getRequestOptions = queryDisputeHistoryRecords(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/dispute-history-records/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateDisputeHistoryRecordTags2() {
  const payload = {};
  const getRequestOptions = bulkUpdateDisputeHistoryRecordTags(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/dispute-history-records/update-tags",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateDisputeHistoryRecordTagsByFilter2() {
  const payload = {};
  const getRequestOptions = bulkUpdateDisputeHistoryRecordTagsByFilter(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/dispute-history-records/update-tags-by-filter",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  bulkUpdateDisputeHistoryRecordTags,
  bulkUpdateDisputeHistoryRecordTagsByFilter,
  getDisputeHistoryRecord,
  queryDisputeHistoryRecords
});
