// @wix/headless-faq/cjs/dist/react/Faq.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Answer = exports.Item = exports.Name = void 0;
exports.Root = Root;
const jsx_runtime_1 = require("react/jsx-runtime");
const react_1 = __importDefault(require("react"));
const react_2 = require("@wix/headless-utils/react");
const CoreFaq = __importStar(require("./core/Faq.js"));
var TestIds;
(function (TestIds) {
    // Entity Root
    TestIds["faqRoot"] = "faq-root";
    TestIds["faqName"] = "faq-name";
    TestIds["faqAnswer"] = "faq-answer";
    TestIds["faqItem"] = "faq-item";
})(TestIds || (TestIds = {}));
/**
 * Root component that provides FAQ service context for rendering a single FAQ entry.
 * Follows the services pattern like Product.Root does.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Faq.Root faq={faqEntry}>
 *   <Faq.Name />
 *   <Faq.Answer />
 * </Faq.Root>
 * ```
 */
function Root(props) {
    const { children, faq, ...attrs } = props;
    const serviceConfig = {
        faqs: [faq],
        categoryId: faq.categoryId,
    };
    return ((0, jsx_runtime_1.jsx)(CoreFaq.Root, { faqServiceConfig: serviceConfig, children: (0, jsx_runtime_1.jsx)(react_2.AsChildSlot, { ...attrs, "data-testid": TestIds.faqRoot, "data-faq-id": faq._id, children: children }) }));
}
/**
 * Displays the FAQ question/name.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Faq.Name className="text-lg font-semibold" />
 *
 * // asChild with primitive
 * <Faq.Name asChild>
 *   <h3 className="text-lg font-semibold" />
 * </Faq.Name>
 *
 * // asChild with react component
 * <Faq.Name asChild>
 *   {React.forwardRef(({question, ...props}, ref) => (
 *     <h3 ref={ref} {...props} className="text-lg font-semibold">
 *       Q: {question}
 *     </h3>
 *   ))}
 * </Faq.Name>
 * ```
 */
exports.Name = react_1.default.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return ((0, jsx_runtime_1.jsx)(CoreFaq.Name, { children: ({ question }) => {
            return ((0, jsx_runtime_1.jsx)(react_2.AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.faqName, customElement: children, customElementProps: { question }, content: question, ...otherProps, children: (0, jsx_runtime_1.jsx)("div", { children: question }) }));
        } }));
});
/**
 * FAQ item wrapper component that provides access to the FAQ unique identifier.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Faq.Item className="faq-item" />
 *
 * // asChild with AccordionItem
 * <Faq.Item asChild>
 *   <AccordionItem className="border-b border-white/10 last:border-b-0" />
 * </Faq.Item>
 *
 * // asChild with custom render function
 * <Faq.Item asChild>
 *   {React.forwardRef(({id, ...props}, ref) => (
 *     <AccordionItem ref={ref} {...props} value={id}>
 *       {children}
 *     </AccordionItem>
 *   ))}
 * </Faq.Item>
 * ```
 */
exports.Item = react_1.default.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return ((0, jsx_runtime_1.jsx)(CoreFaq.Item, { children: ({ id }) => {
            return ((0, jsx_runtime_1.jsx)(react_2.AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.faqItem, "data-faq-id": id, customElement: children, customElementProps: { id }, content: id, ...otherProps, children: (0, jsx_runtime_1.jsx)("div", { children: id }) }));
        } }));
});
/**
 * Displays the FAQ answer.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Faq.Answer className="text-base text-content-secondary" />
 *
 * // asChild with primitive
 * <Faq.Answer asChild>
 *   <p className="text-base text-content-secondary" />
 * </Faq.Answer>
 *
 * // asChild with react component
 * <Faq.Answer asChild>
 *   {React.forwardRef(({answer, ...props}, ref) => (
 *     <p ref={ref} {...props} className="text-base text-content-secondary">
 *       A: {answer}
 *     </p>
 *   ))}
 * </Faq.Answer>
 * ```
 */
exports.Answer = react_1.default.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return ((0, jsx_runtime_1.jsx)(CoreFaq.Answer, { children: ({ answer }) => {
            return ((0, jsx_runtime_1.jsx)(react_2.AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.faqAnswer, customElement: children, customElementProps: { answer }, content: answer, ...otherProps, children: (0, jsx_runtime_1.jsx)("div", { children: answer }) }));
        } }));
});
