// @wix/headless-faq/cjs/dist/react/FaqCategories.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CategoryRepeater = exports.Categories = void 0;
exports.Root = Root;
const jsx_runtime_1 = require("react/jsx-runtime");
const react_1 = __importDefault(require("react"));
const react_2 = require("@wix/headless-utils/react");
const CoreFaqCategories = __importStar(require("./core/FaqCategories.js"));
const FaqCategoryComponents = __importStar(require("./FaqList.js"));
var TestIds;
(function (TestIds) {
    // Container Level
    TestIds["faqCategoriesRoot"] = "faq-categories-root";
    // List Container Level
    TestIds["faqCategories"] = "faq-categories";
    // Repeater Level (individual items)
    TestIds["faqCategory"] = "faq-category";
})(TestIds || (TestIds = {}));
/**
 * Root component that provides FAQ categories service context and conditional rendering.
 * Does not render if there are no categories.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <FaqCategories.Root faqCategoriesConfig={faqCategoriesConfig}>
 *   <FaqCategories.Categories>
 *     <FaqCategories.CategoryRepeater>
 *       <FaqCategory.Name />
 *       <FaqCategory.Faqs>
 *         <FaqCategory.FaqRepeater>
 *           <Faq.Name />
 *           <Faq.Answer />
 *         </FaqCategory.FaqRepeater>
 *       </FaqCategory.Faqs>
 *     </FaqCategories.CategoryRepeater>
 *   </FaqCategories.Categories>
 * </FaqCategories.Root>
 * ```
 */
function Root(props) {
    const { children, faqCategoriesConfig, className, ...attrs } = props;
    const hasCategories = faqCategoriesConfig.categories.length > 0;
    // Don't render if no categories (following Container Level pattern)
    if (!hasCategories)
        return null;
    return ((0, jsx_runtime_1.jsx)(CoreFaqCategories.Root, { faqCategoriesServiceConfig: faqCategoriesConfig, children: (0, jsx_runtime_1.jsx)(react_2.AsChildSlot, { ...attrs, className: className, "data-testid": TestIds.faqCategoriesRoot, "data-categories-count": faqCategoriesConfig.categories.length, children: children }) }));
}
/**
 * Container for the list of categories with empty state support.
 * Follows List Container Level pattern.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <FaqCategories.Categories emptyState={<div>No categories available</div>}>
 *   <FaqCategories.CategoryRepeater>
 *     <FaqCategory.Name />
 *     <FaqCategory.Faqs>
 *       <FaqCategory.FaqRepeater>
 *         <Faq.Name />
 *         <Faq.Answer />
 *       </FaqCategory.FaqRepeater>
 *     </FaqCategory.Faqs>
 *   </FaqCategories.CategoryRepeater>
 * </FaqCategories.Categories>
 *
 * // Simple container usage
 * <FaqCategories.Categories emptyState={<div>No categories</div>}>
 *   <div className="categories-container">
 *     <FaqCategories.CategoryRepeater>
 *       <FaqCategory.Name />
 *     </FaqCategories.CategoryRepeater>
 *   </div>
 * </FaqCategories.Categories>
 * ```
 */
exports.Categories = react_1.default.forwardRef((props, ref) => {
    const { children, emptyState } = props;
    return ((0, jsx_runtime_1.jsx)(CoreFaqCategories.Categories, { children: ({ hasCategories }) => {
            if (!hasCategories) {
                return emptyState || null;
            }
            const attributes = {
                'data-testid': TestIds.faqCategories,
            };
            return ((0, jsx_runtime_1.jsx)("div", { ...attributes, ref: ref, children: children }));
        } }));
});
/**
 * Repeater component that renders FaqCategory.Root for each category.
 * Follows Repeater Level pattern.
 *
 * @component
 * @example
 * ```tsx
 * <FaqCategories.CategoryRepeater>
 *   <FaqCategory.Name />
 *   <FaqCategory.Faqs>
 *     <FaqCategory.FaqRepeater>
 *       <Faq.Name />
 *       <Faq.Answer />
 *     </FaqCategory.FaqRepeater>
 *   </FaqCategory.Faqs>
 * </FaqCategories.CategoryRepeater>
 *
 * // With FAQ configurations per category
 * <FaqCategories.CategoryRepeater faqConfigsByCategory={faqConfigsByCategory}>
 *   <FaqCategory.Name />
 *   <FaqCategory.Faqs>
 *     <FaqCategory.FaqRepeater>
 *       <Faq.Name />
 *       <Faq.Answer />
 *     </FaqCategory.FaqRepeater>
 *   </FaqCategory.Faqs>
 * </FaqCategories.CategoryRepeater>
 * ```
 */
exports.CategoryRepeater = react_1.default.forwardRef((props, _ref) => {
    const { children, faqConfigsByCategory } = props;
    return ((0, jsx_runtime_1.jsx)(CoreFaqCategories.Categories, { children: ({ hasCategories, categories }) => {
            if (!hasCategories)
                return null;
            return ((0, jsx_runtime_1.jsx)(jsx_runtime_1.Fragment, { children: categories.map((category) => ((0, jsx_runtime_1.jsx)(FaqCategoryComponents.Root, { category: category, faqConfig: faqConfigsByCategory?.[category._id || ''] || {
                        faqs: [],
                        categoryId: category._id,
                    }, "data-testid": TestIds.faqCategory, children: children }, category._id))) }));
        } }));
});
