// @wix/headless-faq/cjs/dist/react/FaqList.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FaqRepeater = exports.Items = exports.Name = void 0;
exports.Root = Root;
const jsx_runtime_1 = require("react/jsx-runtime");
const react_1 = __importDefault(require("react"));
const react_2 = require("@wix/headless-utils/react");
const CoreFaqList = __importStar(require("./core/FaqList.js"));
const Faq = __importStar(require("./Faq.js"));
var TestIds;
(function (TestIds) {
    // Entity Root
    TestIds["faqListRoot"] = "faq-list-root";
    TestIds["faqListName"] = "faq-list-name";
    // List Container Level
    TestIds["faqListFaqs"] = "faq-list-faqs";
    // Repeater Level (individual items)
    TestIds["faqListFaq"] = "faq-list-faq";
})(TestIds || (TestIds = {}));
/**
 * Root component that provides service context for a FAQ category and its FAQs.
 * Automatically loads FAQs for the category.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <FaqList.Root category={category}>
 *   <FaqList.Name />
 *   <FaqList.Items>
 *     <FaqList.FaqRepeater>
 *       <Faq.Name />
 *       <Faq.Answer />
 *     </FaqList.FaqRepeater>
 *   </FaqList.Items>
 * </FaqList.Root>
 * ```
 */
function Root(props) {
    const { children, category, className, faqConfig, ...attrs } = props;
    // Default FAQ config for this category
    const defaultFaqConfig = react_1.default.useMemo(() => faqConfig || {
        faqs: [],
        categoryId: category._id,
    }, [faqConfig, category._id]);
    return ((0, jsx_runtime_1.jsx)(CoreFaqList.Root, { faqServiceConfig: defaultFaqConfig, children: (0, jsx_runtime_1.jsx)(CategoryContext, { category: category, children: (0, jsx_runtime_1.jsx)("div", { ...attrs, className: className, "data-testid": TestIds.faqListRoot, "data-category-id": category._id, "data-category-name": category.title, children: children }) }) }));
}
const CategoryDataContext = react_1.default.createContext(null);
function CategoryContext({ category, children }) {
    return ((0, jsx_runtime_1.jsx)(CategoryDataContext.Provider, { value: category, children: children }));
}
function useCategoryData() {
    const category = react_1.default.useContext(CategoryDataContext);
    if (!category) {
        throw new Error('useCategoryData must be used within a FaqList.Root component');
    }
    return category;
}
/**
 * Displays the FAQ category name.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <FaqList.Name className="text-2xl font-bold" />
 *
 * // asChild with primitive
 * <FaqList.Name asChild>
 *   <h2 className="text-2xl font-bold" />
 * </FaqList.Name>
 *
 * // asChild with react component
 * <FaqList.Name asChild>
 *   {React.forwardRef(({name, ...props}, ref) => (
 *     <h2 ref={ref} {...props} className="text-2xl font-bold">
 *       {name}
 *     </h2>
 *   ))}
 * </FaqList.Name>
 * ```
 */
exports.Name = react_1.default.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    const category = useCategoryData();
    const name = category.title || '';
    return ((0, jsx_runtime_1.jsx)(react_2.AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.faqListName, customElement: children, customElementProps: { name }, content: name, ...otherProps, children: (0, jsx_runtime_1.jsx)("div", { children: name }) }));
});
/**
 * Container for the list of FAQs with empty state support.
 * Follows List Container Level pattern.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <FaqList.Items emptyState={<div>No FAQs available</div>}>
 *   <FaqList.FaqRepeater>
 *     <Faq.Name />
 *     <Faq.Answer />
 *   </FaqList.FaqRepeater>
 * </FaqList.Items>
 *
 * // Simple container usage
 * <FaqList.Items emptyState={<div>No FAQs</div>}>
 *   <div className="faqs-container">
 *     <FaqList.FaqRepeater>
 *       <Faq.Name />
 *       <Faq.Answer />
 *     </FaqList.FaqRepeater>
 *   </div>
 * </FaqList.Items>
 * ```
 */
exports.Items = react_1.default.forwardRef((props, ref) => {
    const { children, emptyState } = props;
    return ((0, jsx_runtime_1.jsx)(CoreFaqList.Items, { children: ({ hasFaqs }) => {
            if (!hasFaqs) {
                return emptyState || null;
            }
            const attributes = {
                'data-testid': TestIds.faqListFaqs,
            };
            return ((0, jsx_runtime_1.jsx)("div", { ...attributes, ref: ref, children: children }));
        } }));
});
/**
 * Repeater component that renders Faq.Root for each FAQ entry.
 * Follows Repeater Level pattern.
 *
 * @component
 * @example
 * ```tsx
 * <FaqList.FaqRepeater>
 *   <Faq.Name />
 *   <Faq.Answer />
 * </FaqList.FaqRepeater>
 * ```
 */
exports.FaqRepeater = react_1.default.forwardRef((props, _ref) => {
    const { children } = props;
    return ((0, jsx_runtime_1.jsx)(CoreFaqList.Items, { children: ({ hasFaqs, faqs }) => {
            if (!hasFaqs)
                return null;
            return ((0, jsx_runtime_1.jsx)(jsx_runtime_1.Fragment, { children: faqs.map((faq) => ((0, jsx_runtime_1.jsx)(Faq.Root, { faq: faq, "data-testid": TestIds.faqListFaq, children: children }, faq._id))) }));
        } }));
});
