// @wix/ricos/dist/cjs/ricos-styles/decorations/bold-decoration.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var bold_decoration_exports = {};
__export(bold_decoration_exports, {
  BoldDecoration: () => BoldDecoration
});
module.exports = __toCommonJS(bold_decoration_exports);
var import_v1 = require("./../../ricos-schema/legacy/wix/rich_content/v1/index.js");
var import_empty_decoration = require("./empty-decoration.js");
const parseFontWeight = (fontWeight) => {
  if (fontWeight === "bold") {
    return 700;
  }
  if (fontWeight === "normal") {
    return 400;
  }
  return parseInt(fontWeight, 10);
};
class BoldDecoration {
  constructor(customStyle) {
    this.type = import_v1.Decoration_Type.BOLD;
    this.customStyle = customStyle;
  }
  static of(decoration) {
    if (decoration.type !== import_v1.Decoration_Type.BOLD) {
      throw new TypeError(`invalid decoration initializer ${decoration}`);
    }
    return new BoldDecoration({ fontWeight: decoration.fontWeightValue });
  }
  getDecoration() {
    return {
      type: this.type,
      fontWeightValue: parseFontWeight(this.customStyle.fontWeight)
    };
  }
  static fromCustomStyle(customStyle) {
    return new BoldDecoration(customStyle);
  }
  fromCustomStyle(customStyle) {
    return BoldDecoration.fromCustomStyle(customStyle);
  }
  toCustomStyle() {
    return this.customStyle;
  }
  overrideWith(decoration) {
    if (!(decoration instanceof BoldDecoration || decoration instanceof import_empty_decoration.EmptyDecoration)) {
      throw new TypeError(`invalid merge decoration ${decoration}`);
    }
    const customStyle = { ...this.customStyle, ...decoration.toCustomStyle() };
    return new BoldDecoration(customStyle);
  }
}
