// @wix/auto_sdk_restaurants_local-delivery-spi/build/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  BusinessStatus: () => BusinessStatus,
  businessStatus: () => businessStatus2,
  createDelivery: () => createDelivery2,
  estimateDelivery: () => estimateDelivery2,
  listAccountIds: () => listAccountIds2
});
module.exports = __toCommonJS(index_typings_exports);

// src/localdelivery-spi-v1-tpa-local-delivery-spi.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/localdelivery-spi-v1-tpa-local-delivery-spi.http.ts
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressLocaldeliverySpiV1LocalDeliveryProviderUrl(opts) {
  const domainToMappings = {};
  return (0, import_rest_modules.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_local-delivery-spi";
function listAccountIds(payload) {
  function __listAccountIds({ host }) {
    const metadata = {
      entityFqdn: "wix.localdelivery.spi.v1.tpa",
      method: "POST",
      methodFqn: "com.wixpress.localdelivery.spi.v1.LocalDeliveryProvider.ListAccountIds",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressLocaldeliverySpiV1LocalDeliveryProviderUrl({
        protoPath: "/account-ids",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __listAccountIds;
}
function businessStatus(payload) {
  function __businessStatus({ host }) {
    const metadata = {
      entityFqdn: "wix.localdelivery.spi.v1.tpa",
      method: "POST",
      methodFqn: "com.wixpress.localdelivery.spi.v1.LocalDeliveryProvider.BusinessStatus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressLocaldeliverySpiV1LocalDeliveryProviderUrl({
        protoPath: "/business-status",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __businessStatus;
}
function estimateDelivery(payload) {
  function __estimateDelivery({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "pickupTime" }, { path: "dropoffTime" }]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "pickupAddress.geocode.latitude" },
          { path: "pickupAddress.geocode.longitude" },
          { path: "dropoffAddress.geocode.latitude" },
          { path: "dropoffAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.localdelivery.spi.v1.tpa",
      method: "POST",
      methodFqn: "com.wixpress.localdelivery.spi.v1.LocalDeliveryProvider.EstimateDelivery",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressLocaldeliverySpiV1LocalDeliveryProviderUrl({
        protoPath: "/estimate-delivery",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "promisedPickupTime" },
            { path: "promisedDropoffTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __estimateDelivery;
}
function createDelivery(payload) {
  function __createDelivery({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "pickupTime" }, { path: "dropoffTime" }]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "pickupAddress.geocode.latitude" },
          { path: "pickupAddress.geocode.longitude" },
          { path: "dropoffAddress.geocode.latitude" },
          { path: "dropoffAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.localdelivery.spi.v1.tpa",
      method: "POST",
      methodFqn: "com.wixpress.localdelivery.spi.v1.LocalDeliveryProvider.CreateDelivery",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressLocaldeliverySpiV1LocalDeliveryProviderUrl({
        protoPath: "/create-delivery",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "pickupWindowStartTime" },
            { path: "pickupWindowEndTime" },
            { path: "dropoffWindowStartTime" },
            { path: "dropoffWindowEndTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createDelivery;
}

// src/localdelivery-spi-v1-tpa-local-delivery-spi.universal.ts
var import_address = require("@wix/sdk-runtime/transformations/address");
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var BusinessStatus = /* @__PURE__ */ ((BusinessStatus2) => {
  BusinessStatus2["NONE"] = "NONE";
  BusinessStatus2["READY"] = "READY";
  BusinessStatus2["PENDING"] = "PENDING";
  BusinessStatus2["ERROR"] = "ERROR";
  BusinessStatus2["NOT_STARTED"] = "NOT_STARTED";
  return BusinessStatus2;
})(BusinessStatus || {});
async function listAccountIds2(instanceId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    instanceId
  });
  const reqOpts = listAccountIds(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { instanceId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["instanceId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function businessStatus2(instanceId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    instanceId,
    accountId: options?.accountId
  });
  const reqOpts = businessStatus(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          instanceId: "$[0]",
          accountId: "$[1].accountId"
        },
        singleArgumentUnchanged: false
      },
      ["instanceId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function estimateDelivery2(instanceId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      instanceId,
      accountId: options?.accountId,
      pickupAddress: options?.pickupAddress,
      dropoffAddress: options?.dropoffAddress,
      pickupTime: options?.pickupTime,
      dropoffTime: options?.dropoffTime,
      storeLocationId: options?.storeLocationId,
      orderTotal: options?.orderTotal
    }),
    [
      {
        transformFn: import_address.transformSDKAddressToRESTAddress,
        paths: [{ path: "pickupAddress" }, { path: "dropoffAddress" }]
      }
    ]
  );
  const reqOpts = estimateDelivery(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          instanceId: "$[0]",
          accountId: "$[1].accountId",
          pickupAddress: "$[1].pickupAddress",
          dropoffAddress: "$[1].dropoffAddress",
          pickupTime: "$[1].pickupTime",
          dropoffTime: "$[1].dropoffTime",
          storeLocationId: "$[1].storeLocationId",
          orderTotal: "$[1].orderTotal"
        },
        singleArgumentUnchanged: false
      },
      ["instanceId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function createDelivery2(instanceId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      instanceId,
      accountId: options?.accountId,
      storeLocationId: options?.storeLocationId,
      pickupBusinessName: options?.pickupBusinessName,
      estimateId: options?.estimateId,
      pickupTime: options?.pickupTime,
      dropoffTime: options?.dropoffTime,
      pickupAddress: options?.pickupAddress,
      dropoffAddress: options?.dropoffAddress,
      pickupPhoneNumber: options?.pickupPhoneNumber,
      customer: options?.customer,
      pickupInstructions: options?.pickupInstructions,
      dropoffInstructions: options?.dropoffInstructions,
      contactlessDelivery: options?.contactlessDelivery,
      orderId: options?.orderId,
      orderTotal: options?.orderTotal,
      orderTip: options?.orderTip
    }),
    [
      {
        transformFn: import_address.transformSDKAddressToRESTAddress,
        paths: [{ path: "pickupAddress" }, { path: "dropoffAddress" }]
      }
    ]
  );
  const reqOpts = createDelivery(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          instanceId: "$[0]",
          accountId: "$[1].accountId",
          storeLocationId: "$[1].storeLocationId",
          pickupBusinessName: "$[1].pickupBusinessName",
          estimateId: "$[1].estimateId",
          pickupTime: "$[1].pickupTime",
          dropoffTime: "$[1].dropoffTime",
          pickupAddress: "$[1].pickupAddress",
          dropoffAddress: "$[1].dropoffAddress",
          pickupPhoneNumber: "$[1].pickupPhoneNumber",
          customer: "$[1].customer",
          pickupInstructions: "$[1].pickupInstructions",
          dropoffInstructions: "$[1].dropoffInstructions",
          contactlessDelivery: "$[1].contactlessDelivery",
          orderId: "$[1].orderId",
          orderTotal: "$[1].orderTotal",
          orderTip: "$[1].orderTip"
        },
        singleArgumentUnchanged: false
      },
      ["instanceId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  BusinessStatus,
  businessStatus,
  createDelivery,
  estimateDelivery,
  listAccountIds
});
