// @wix/auto_sdk_restaurants_fulfillment-methods/build/es/index.typings.mjs
// src/restaurants-v1-fulfillment-method-fulfillment-methods.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/restaurants-v1-fulfillment-method-fulfillment-methods.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKFloatToRESTFloat } from "@wix/sdk-runtime/transformations/float";
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixRestaurantsFulfillmentV1FulfillmentMethodsServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/fulfillment-methods",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/fulfillment-methods/v1",
        destPath: "/v1"
      }
    ],
    _: [
      {
        srcPath: "/_api/fulfillment-methods/v1",
        destPath: "/v1"
      }
    ],
    _base_domain_: [
      {
        srcPath: "/_api/fulfillment-methods/v1",
        destPath: "/v1"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/fulfillment-methods/v1",
        destPath: "/v1"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/fulfillment-methods/v1",
        destPath: "/v1"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/fulfillment-methods/v1",
        destPath: "/v1"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/fulfillment-methods/v1",
        destPath: "/v1"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/fulfillment-methods/v1",
        destPath: "/v1"
      }
    ],
    "users._base_domain_": [
      {
        srcPath: "/_api/fulfillment-methods/v1",
        destPath: "/v1"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/fulfillment-methods/v1",
        destPath: "/v1"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/fulfillment-methods/v1",
        destPath: "/v1"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/fulfillment-methods",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_fulfillment-methods";
function createFulfillmentMethod(payload) {
  function __createFulfillmentMethod({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "fulfillmentMethod.createdDate" },
          { path: "fulfillmentMethod.updatedDate" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "fulfillmentMethod.pickupOptions.address.geocode.latitude" },
          { path: "fulfillmentMethod.pickupOptions.address.geocode.longitude" },
          {
            path: "fulfillmentMethod.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.latitude"
          },
          {
            path: "fulfillmentMethod.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.longitude"
          },
          {
            path: "fulfillmentMethod.deliveryOptions.deliveryArea.customOptions.geocodes.latitude"
          },
          {
            path: "fulfillmentMethod.deliveryOptions.deliveryArea.customOptions.geocodes.longitude"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.fulfillment_method",
      method: "POST",
      methodFqn: "wix.restaurants.fulfillment.v1.FulfillmentMethodsService.CreateFulfillmentMethod",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsFulfillmentV1FulfillmentMethodsServiceUrl({
        protoPath: "/v1/fulfillment-methods",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "fulfillmentMethod.createdDate" },
            { path: "fulfillmentMethod.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "fulfillmentMethod.pickupOptions.address.geocode.latitude"
            },
            {
              path: "fulfillmentMethod.pickupOptions.address.geocode.longitude"
            },
            {
              path: "fulfillmentMethod.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.latitude"
            },
            {
              path: "fulfillmentMethod.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.longitude"
            },
            {
              path: "fulfillmentMethod.deliveryOptions.deliveryArea.customOptions.geocodes.latitude"
            },
            {
              path: "fulfillmentMethod.deliveryOptions.deliveryArea.customOptions.geocodes.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createFulfillmentMethod;
}
function bulkCreateFulfillmentMethods(payload) {
  function __bulkCreateFulfillmentMethods({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "fulfillmentMethods.createdDate" },
          { path: "fulfillmentMethods.updatedDate" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "fulfillmentMethods.pickupOptions.address.geocode.latitude" },
          {
            path: "fulfillmentMethods.pickupOptions.address.geocode.longitude"
          },
          {
            path: "fulfillmentMethods.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.latitude"
          },
          {
            path: "fulfillmentMethods.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.longitude"
          },
          {
            path: "fulfillmentMethods.deliveryOptions.deliveryArea.customOptions.geocodes.latitude"
          },
          {
            path: "fulfillmentMethods.deliveryOptions.deliveryArea.customOptions.geocodes.longitude"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.fulfillment_method",
      method: "POST",
      methodFqn: "wix.restaurants.fulfillment.v1.FulfillmentMethodsService.BulkCreateFulfillmentMethods",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsFulfillmentV1FulfillmentMethodsServiceUrl({
        protoPath: "/v1/bulk/fulfillment-methods/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.fulfillmentMethod.createdDate" },
            { path: "results.fulfillmentMethod.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "results.fulfillmentMethod.pickupOptions.address.geocode.latitude"
            },
            {
              path: "results.fulfillmentMethod.pickupOptions.address.geocode.longitude"
            },
            {
              path: "results.fulfillmentMethod.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.latitude"
            },
            {
              path: "results.fulfillmentMethod.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.longitude"
            },
            {
              path: "results.fulfillmentMethod.deliveryOptions.deliveryArea.customOptions.geocodes.latitude"
            },
            {
              path: "results.fulfillmentMethod.deliveryOptions.deliveryArea.customOptions.geocodes.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateFulfillmentMethods;
}
function getFulfillmentMethod(payload) {
  function __getFulfillmentMethod({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.fulfillment_method",
      method: "GET",
      methodFqn: "wix.restaurants.fulfillment.v1.FulfillmentMethodsService.GetFulfillmentMethod",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsFulfillmentV1FulfillmentMethodsServiceUrl({
        protoPath: "/v1/fulfillment-methods/{fulfillmentMethodId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "fulfillmentMethod.createdDate" },
            { path: "fulfillmentMethod.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "fulfillmentMethod.pickupOptions.address.geocode.latitude"
            },
            {
              path: "fulfillmentMethod.pickupOptions.address.geocode.longitude"
            },
            {
              path: "fulfillmentMethod.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.latitude"
            },
            {
              path: "fulfillmentMethod.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.longitude"
            },
            {
              path: "fulfillmentMethod.deliveryOptions.deliveryArea.customOptions.geocodes.latitude"
            },
            {
              path: "fulfillmentMethod.deliveryOptions.deliveryArea.customOptions.geocodes.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getFulfillmentMethod;
}
function updateFulfillmentMethod(payload) {
  function __updateFulfillmentMethod({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "fulfillmentMethod.createdDate" },
          { path: "fulfillmentMethod.updatedDate" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "fulfillmentMethod.pickupOptions.address.geocode.latitude" },
          { path: "fulfillmentMethod.pickupOptions.address.geocode.longitude" },
          {
            path: "fulfillmentMethod.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.latitude"
          },
          {
            path: "fulfillmentMethod.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.longitude"
          },
          {
            path: "fulfillmentMethod.deliveryOptions.deliveryArea.customOptions.geocodes.latitude"
          },
          {
            path: "fulfillmentMethod.deliveryOptions.deliveryArea.customOptions.geocodes.longitude"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.fulfillment_method",
      method: "PATCH",
      methodFqn: "wix.restaurants.fulfillment.v1.FulfillmentMethodsService.UpdateFulfillmentMethod",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsFulfillmentV1FulfillmentMethodsServiceUrl({
        protoPath: "/v1/fulfillment-methods/{fulfillmentMethod.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "fulfillmentMethod.createdDate" },
            { path: "fulfillmentMethod.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "fulfillmentMethod.pickupOptions.address.geocode.latitude"
            },
            {
              path: "fulfillmentMethod.pickupOptions.address.geocode.longitude"
            },
            {
              path: "fulfillmentMethod.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.latitude"
            },
            {
              path: "fulfillmentMethod.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.longitude"
            },
            {
              path: "fulfillmentMethod.deliveryOptions.deliveryArea.customOptions.geocodes.latitude"
            },
            {
              path: "fulfillmentMethod.deliveryOptions.deliveryArea.customOptions.geocodes.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateFulfillmentMethod;
}
function deleteFulfillmentMethod(payload) {
  function __deleteFulfillmentMethod({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.fulfillment_method",
      method: "DELETE",
      methodFqn: "wix.restaurants.fulfillment.v1.FulfillmentMethodsService.DeleteFulfillmentMethod",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsFulfillmentV1FulfillmentMethodsServiceUrl({
        protoPath: "/v1/fulfillment-methods/{fulfillmentMethodId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteFulfillmentMethod;
}
function queryFulfillmentMethods(payload) {
  function __queryFulfillmentMethods({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "projectionMask" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.fulfillment_method",
      method: "POST",
      methodFqn: "wix.restaurants.fulfillment.v1.FulfillmentMethodsService.QueryFulfillmentMethods",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsFulfillmentV1FulfillmentMethodsServiceUrl({
        protoPath: "/v1/fulfillment-methods/query",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "fulfillmentMethods.createdDate" },
            { path: "fulfillmentMethods.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "fulfillmentMethods.pickupOptions.address.geocode.latitude"
            },
            {
              path: "fulfillmentMethods.pickupOptions.address.geocode.longitude"
            },
            {
              path: "fulfillmentMethods.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.latitude"
            },
            {
              path: "fulfillmentMethods.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.longitude"
            },
            {
              path: "fulfillmentMethods.deliveryOptions.deliveryArea.customOptions.geocodes.latitude"
            },
            {
              path: "fulfillmentMethods.deliveryOptions.deliveryArea.customOptions.geocodes.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryFulfillmentMethods;
}
function listFulfillmentMethods(payload) {
  function __listFulfillmentMethods({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "address.geocode.latitude" },
          { path: "address.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.fulfillment_method",
      method: "GET",
      methodFqn: "wix.restaurants.fulfillment.v1.FulfillmentMethodsService.ListFulfillmentMethods",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsFulfillmentV1FulfillmentMethodsServiceUrl({
        protoPath: "/v1/fulfillment-methods",
        data: serializedData,
        host
      }),
      params: toURLSearchParams(serializedData),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "fulfillmentMethods.createdDate" },
            { path: "fulfillmentMethods.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "fulfillmentMethods.pickupOptions.address.geocode.latitude"
            },
            {
              path: "fulfillmentMethods.pickupOptions.address.geocode.longitude"
            },
            {
              path: "fulfillmentMethods.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.latitude"
            },
            {
              path: "fulfillmentMethods.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.longitude"
            },
            {
              path: "fulfillmentMethods.deliveryOptions.deliveryArea.customOptions.geocodes.latitude"
            },
            {
              path: "fulfillmentMethods.deliveryOptions.deliveryArea.customOptions.geocodes.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listFulfillmentMethods;
}
function listAvailableFulfillmentMethodsForAddress(payload) {
  function __listAvailableFulfillmentMethodsForAddress({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "address.geocode.latitude" },
          { path: "address.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.fulfillment_method",
      method: "POST",
      methodFqn: "wix.restaurants.fulfillment.v1.FulfillmentMethodsService.ListAvailableFulfillmentMethodsForAddress",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsFulfillmentV1FulfillmentMethodsServiceUrl({
        protoPath: "/v1/fulfillment-methods/available-for-address",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "fulfillmentMethods.createdDate" },
            { path: "fulfillmentMethods.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "fulfillmentMethods.pickupOptions.address.geocode.latitude"
            },
            {
              path: "fulfillmentMethods.pickupOptions.address.geocode.longitude"
            },
            {
              path: "fulfillmentMethods.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.latitude"
            },
            {
              path: "fulfillmentMethods.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.longitude"
            },
            {
              path: "fulfillmentMethods.deliveryOptions.deliveryArea.customOptions.geocodes.latitude"
            },
            {
              path: "fulfillmentMethods.deliveryOptions.deliveryArea.customOptions.geocodes.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listAvailableFulfillmentMethodsForAddress;
}
function getAccumulatedFulfillmentMethodsAvailability(payload) {
  function __getAccumulatedFulfillmentMethodsAvailability({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.fulfillment_method",
      method: "GET",
      methodFqn: "wix.restaurants.fulfillment.v1.FulfillmentMethodsService.GetAccumulatedFulfillmentMethodsAvailability",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsFulfillmentV1FulfillmentMethodsServiceUrl({
        protoPath: "/v1/fulfillment-methods/accumulated-availability",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getAccumulatedFulfillmentMethodsAvailability;
}
function getCombinedMethodAvailability(payload) {
  function __getCombinedMethodAvailability({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.fulfillment_method",
      method: "GET",
      methodFqn: "wix.restaurants.fulfillment.v1.FulfillmentMethodsService.GetCombinedMethodAvailability",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsFulfillmentV1FulfillmentMethodsServiceUrl({
        protoPath: "/v1/fulfillment-methods/combined-availability",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getCombinedMethodAvailability;
}
function getAggregatedMethodAvailability(payload) {
  function __getAggregatedMethodAvailability({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.fulfillment_method",
      method: "POST",
      methodFqn: "wix.restaurants.fulfillment.v1.FulfillmentMethodsService.GetAggregatedMethodAvailability",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsFulfillmentV1FulfillmentMethodsServiceUrl({
        protoPath: "/v1/fulfillment-methods/aggregated-availability",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __getAggregatedMethodAvailability;
}
function bulkUpdateFulfillmentMethodTags(payload) {
  function __bulkUpdateFulfillmentMethodTags({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.fulfillment_method",
      method: "POST",
      methodFqn: "wix.restaurants.fulfillment.v1.FulfillmentMethodsService.BulkUpdateFulfillmentMethodTags",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsFulfillmentV1FulfillmentMethodsServiceUrl({
        protoPath: "/v1/bulk/fulfillment-methods/update-tags",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateFulfillmentMethodTags;
}
function bulkUpdateFulfillmentMethodTagsByFilter(payload) {
  function __bulkUpdateFulfillmentMethodTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.fulfillment_method",
      method: "POST",
      methodFqn: "wix.restaurants.fulfillment.v1.FulfillmentMethodsService.BulkUpdateFulfillmentMethodTagsByFilter",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsFulfillmentV1FulfillmentMethodsServiceUrl({
        protoPath: "/v1/bulk/fulfillment-methods/update-tags-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateFulfillmentMethodTagsByFilter;
}

// src/restaurants-v1-fulfillment-method-fulfillment-methods.universal.ts
import { transformSDKAddressToRESTAddress } from "@wix/sdk-runtime/transformations/address";
import { transformRESTAddressToSDKAddress } from "@wix/sdk-runtime/transformations/address";
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var FulfillmentMethodType = /* @__PURE__ */ ((FulfillmentMethodType2) => {
  FulfillmentMethodType2["UNKNOWN_FULFILLMENT_TYPE"] = "UNKNOWN_FULFILLMENT_TYPE";
  FulfillmentMethodType2["PICKUP"] = "PICKUP";
  FulfillmentMethodType2["DELIVERY"] = "DELIVERY";
  return FulfillmentMethodType2;
})(FulfillmentMethodType || {});
var SubdivisionType = /* @__PURE__ */ ((SubdivisionType2) => {
  SubdivisionType2["UNKNOWN_SUBDIVISION_TYPE"] = "UNKNOWN_SUBDIVISION_TYPE";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_1"] = "ADMINISTRATIVE_AREA_LEVEL_1";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_2"] = "ADMINISTRATIVE_AREA_LEVEL_2";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_3"] = "ADMINISTRATIVE_AREA_LEVEL_3";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_4"] = "ADMINISTRATIVE_AREA_LEVEL_4";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_5"] = "ADMINISTRATIVE_AREA_LEVEL_5";
  SubdivisionType2["COUNTRY"] = "COUNTRY";
  return SubdivisionType2;
})(SubdivisionType || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNKNOWN_DELIVERY_AREA"] = "UNKNOWN_DELIVERY_AREA";
  Type2["RADIUS"] = "RADIUS";
  Type2["POSTAL_CODE"] = "POSTAL_CODE";
  Type2["CUSTOM"] = "CUSTOM";
  Type2["PROVIDER_DEFINED"] = "PROVIDER_DEFINED";
  return Type2;
})(Type || {});
var Unit = /* @__PURE__ */ ((Unit2) => {
  Unit2["UNKNOWN_UNIT"] = "UNKNOWN_UNIT";
  Unit2["MILES"] = "MILES";
  Unit2["KILOMETERS"] = "KILOMETERS";
  return Unit2;
})(Unit || {});
var EntitiesDayOfWeek = /* @__PURE__ */ ((EntitiesDayOfWeek2) => {
  EntitiesDayOfWeek2["MON"] = "MON";
  EntitiesDayOfWeek2["TUE"] = "TUE";
  EntitiesDayOfWeek2["WED"] = "WED";
  EntitiesDayOfWeek2["THU"] = "THU";
  EntitiesDayOfWeek2["FRI"] = "FRI";
  EntitiesDayOfWeek2["SAT"] = "SAT";
  EntitiesDayOfWeek2["SUN"] = "SUN";
  return EntitiesDayOfWeek2;
})(EntitiesDayOfWeek || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var PlacementType = /* @__PURE__ */ ((PlacementType2) => {
  PlacementType2["BEFORE"] = "BEFORE";
  PlacementType2["AFTER"] = "AFTER";
  PlacementType2["REPLACE"] = "REPLACE";
  return PlacementType2;
})(PlacementType || {});
var DayOfWeek = /* @__PURE__ */ ((DayOfWeek2) => {
  DayOfWeek2["MONDAY"] = "MONDAY";
  DayOfWeek2["TUESDAY"] = "TUESDAY";
  DayOfWeek2["WEDNESDAY"] = "WEDNESDAY";
  DayOfWeek2["THURSDAY"] = "THURSDAY";
  DayOfWeek2["FRIDAY"] = "FRIDAY";
  DayOfWeek2["SATURDAY"] = "SATURDAY";
  DayOfWeek2["SUNDAY"] = "SUNDAY";
  return DayOfWeek2;
})(DayOfWeek || {});
var ResolutionMethod = /* @__PURE__ */ ((ResolutionMethod2) => {
  ResolutionMethod2["QUERY_PARAM"] = "QUERY_PARAM";
  ResolutionMethod2["SUBDOMAIN"] = "SUBDOMAIN";
  ResolutionMethod2["SUBDIRECTORY"] = "SUBDIRECTORY";
  return ResolutionMethod2;
})(ResolutionMethod || {});
async function createFulfillmentMethod2(fulfillmentMethod) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({
      fulfillmentMethod
    }),
    [
      {
        transformFn: transformSDKAddressToRESTAddress,
        paths: [
          { path: "fulfillmentMethod.pickupOptions.address" },
          {
            path: "fulfillmentMethod.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress"
          }
        ]
      }
    ]
  );
  const reqOpts = createFulfillmentMethod(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTAddressToSDKAddress,
          paths: [
            { path: "fulfillmentMethod.pickupOptions.address" },
            {
              path: "fulfillmentMethod.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress"
            }
          ]
        }
      ])
    )?.fulfillmentMethod;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { fulfillmentMethod: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["fulfillmentMethod"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCreateFulfillmentMethods2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({
      fulfillmentMethods: options?.fulfillmentMethods,
      returnEntity: options?.returnEntity
    }),
    [
      {
        transformFn: transformSDKAddressToRESTAddress,
        paths: [
          { path: "fulfillmentMethods.pickupOptions.address" },
          {
            path: "fulfillmentMethods.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress"
          }
        ]
      }
    ]
  );
  const reqOpts = bulkCreateFulfillmentMethods(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTAddressToSDKAddress,
          paths: [
            { path: "results.fulfillmentMethod.pickupOptions.address" },
            {
              path: "results.fulfillmentMethod.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress"
            }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          fulfillmentMethods: "$[0].fulfillmentMethods",
          returnEntity: "$[0].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getFulfillmentMethod2(fulfillmentMethodId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    fulfillmentMethodId
  });
  const reqOpts = getFulfillmentMethod(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTAddressToSDKAddress,
          paths: [
            { path: "fulfillmentMethod.pickupOptions.address" },
            {
              path: "fulfillmentMethod.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress"
            }
          ]
        }
      ])
    )?.fulfillmentMethod;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { fulfillmentMethodId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["fulfillmentMethodId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateFulfillmentMethod2(_id, fulfillmentMethod) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({
      fulfillmentMethod: { ...fulfillmentMethod, id: _id }
    }),
    [
      {
        transformFn: transformSDKAddressToRESTAddress,
        paths: [
          { path: "fulfillmentMethod.pickupOptions.address" },
          {
            path: "fulfillmentMethod.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress"
          }
        ]
      }
    ]
  );
  const reqOpts = updateFulfillmentMethod(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTAddressToSDKAddress,
          paths: [
            { path: "fulfillmentMethod.pickupOptions.address" },
            {
              path: "fulfillmentMethod.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress"
            }
          ]
        }
      ])
    )?.fulfillmentMethod;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { fulfillmentMethod: "$[1]" },
        explicitPathsToArguments: { "fulfillmentMethod.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "fulfillmentMethod"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteFulfillmentMethod2(fulfillmentMethodId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    fulfillmentMethodId
  });
  const reqOpts = deleteFulfillmentMethod(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { fulfillmentMethodId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["fulfillmentMethodId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryFulfillmentMethods2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryFulfillmentMethods(
        payload
      );
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [
          {
            transformFn: transformRESTAddressToSDKAddress,
            paths: [
              { path: "fulfillmentMethods.pickupOptions.address" },
              {
                path: "fulfillmentMethods.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress"
              }
            ]
          }
        ])
      );
      return {
        items: transformedData?.fulfillmentMethods,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function listFulfillmentMethods2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({
      cursorPaging: options?.cursorPaging
    }),
    [
      {
        transformFn: transformSDKAddressToRESTAddress,
        paths: [{ path: "address" }]
      }
    ]
  );
  const reqOpts = listFulfillmentMethods(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTAddressToSDKAddress,
          paths: [
            { path: "fulfillmentMethods.pickupOptions.address" },
            {
              path: "fulfillmentMethods.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress"
            }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { cursorPaging: "$[0].cursorPaging" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listAvailableFulfillmentMethodsForAddress2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({
      address: options?.address,
      cursorPaging: options?.cursorPaging,
      fulfillmentMethodIds: options?.fulfillmentMethodIds
    }),
    [
      {
        transformFn: transformSDKAddressToRESTAddress,
        paths: [{ path: "address" }]
      }
    ]
  );
  const reqOpts = listAvailableFulfillmentMethodsForAddress(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTAddressToSDKAddress,
          paths: [
            { path: "fulfillmentMethods.pickupOptions.address" },
            {
              path: "fulfillmentMethods.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress"
            }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          address: "$[0].address",
          cursorPaging: "$[0].cursorPaging",
          fulfillmentMethodIds: "$[0].fulfillmentMethodIds"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getAccumulatedFulfillmentMethodsAvailability2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    fulfillmentMethodIds: options?.fulfillmentMethodIds
  });
  const reqOpts = getAccumulatedFulfillmentMethodsAvailability(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          fulfillmentMethodIds: "$[0].fulfillmentMethodIds"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getCombinedMethodAvailability2(fulfillmentMethodIds) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    fulfillmentMethodIds
  });
  const reqOpts = getCombinedMethodAvailability(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { fulfillmentMethodIds: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["fulfillmentMethodIds"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getAggregatedMethodAvailability2(fulfillmentMethodIds) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    fulfillmentMethodIds
  });
  const reqOpts = getAggregatedMethodAvailability(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { fulfillmentMethodIds: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["fulfillmentMethodIds"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateFulfillmentMethodTags2(fulfillmentMethodIds, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    fulfillmentMethodIds,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateFulfillmentMethodTags(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          fulfillmentMethodIds: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["fulfillmentMethodIds", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateFulfillmentMethodTagsByFilter2(filter, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    filter,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateFulfillmentMethodTagsByFilter(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          filter: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["filter", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
export {
  DayOfWeek,
  EntitiesDayOfWeek,
  FulfillmentMethodType,
  PlacementType,
  ResolutionMethod,
  SortOrder,
  SubdivisionType,
  Type,
  Unit,
  WebhookIdentityType,
  bulkCreateFulfillmentMethods2 as bulkCreateFulfillmentMethods,
  bulkUpdateFulfillmentMethodTags2 as bulkUpdateFulfillmentMethodTags,
  bulkUpdateFulfillmentMethodTagsByFilter2 as bulkUpdateFulfillmentMethodTagsByFilter,
  createFulfillmentMethod2 as createFulfillmentMethod,
  deleteFulfillmentMethod2 as deleteFulfillmentMethod,
  getAccumulatedFulfillmentMethodsAvailability2 as getAccumulatedFulfillmentMethodsAvailability,
  getAggregatedMethodAvailability2 as getAggregatedMethodAvailability,
  getCombinedMethodAvailability2 as getCombinedMethodAvailability,
  getFulfillmentMethod2 as getFulfillmentMethod,
  listAvailableFulfillmentMethodsForAddress2 as listAvailableFulfillmentMethodsForAddress,
  listFulfillmentMethods2 as listFulfillmentMethods,
  queryFulfillmentMethods2 as queryFulfillmentMethods,
  updateFulfillmentMethod2 as updateFulfillmentMethod
};
