// @wix/auto-patterns/dist/cjs/utils/actions/bulkDeleteAction.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.bulkDeleteAction = void 0;
var _wixUiIconsCommon = require("@wix/wix-ui-icons-common");
var _react = _interopRequireDefault(require("react"));
var _jsxFileName = "/home/builduser/work/73d19ce3378ce4dd/packages/auto-patterns/dist/cjs/utils/actions/bulkDeleteAction.tsx";
const bulkDeleteAction = params => {
  const {
    action,
    sdk,
    actionParams
  } = params;
  return {
    onClick: () => {
      if (action.bulkDelete.mode === 'modal') {
        sdk.openModal('BulkDeleteModal', {
          modalParams: action.bulkDelete.modal,
          actionParams,
          sdk
        });
      }
    },
    label: 'Delete',
    icon: /*#__PURE__*/_react.default.createElement(_wixUiIconsCommon.Delete, {
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 18,
        columnNumber: 11
      }
    })
  };
};
exports.bulkDeleteAction = bulkDeleteAction;
