// @wix/auto-patterns/dist/cjs/utils/actions/createAction.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.createAction = void 0;
var _react = _interopRequireDefault(require("react"));
var _wixUiIconsCommon = require("@wix/wix-ui-icons-common");
var _jsxFileName = "/home/builduser/work/73d19ce3378ce4dd/packages/auto-patterns/dist/cjs/utils/actions/createAction.tsx";
const createAction = params => {
  const {
    action,
    sdk
  } = params;
  const {
    appConfig
  } = sdk;
  return {
    onClick: () => {
      if (action.create.mode === 'page') {
        const pageId = action.create.page.id;
        const pageToNavigate = appConfig.pages.find(page => page.id === pageId);
        if (pageToNavigate) {
          if (pageToNavigate.type === 'entityPage') {
            const {
              navigateToEntityPage
            } = sdk;
            if (navigateToEntityPage) {
              navigateToEntityPage({
                entityPageId: pageId,
                item: null
              });
            }
          }
        }
      }
    },
    label: 'Create',
    icon: /*#__PURE__*/_react.default.createElement(_wixUiIconsCommon.Add, {
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 31,
        columnNumber: 11
      }
    })
  };
};
exports.createAction = createAction;
