// @wix/auto-patterns/dist/cjs/utils/actions/deleteAction.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.deleteAction = void 0;
var _wixUiIconsCommon = require("@wix/wix-ui-icons-common");
var _react = _interopRequireDefault(require("react"));
var _jsxFileName = "/home/builduser/work/73d19ce3378ce4dd/packages/auto-patterns/dist/cjs/utils/actions/deleteAction.tsx";
const deleteAction = params => {
  const {
    action,
    sdk,
    actionParams
  } = params;
  const {
    modal: modalParams
  } = action.delete;
  const {
    getOptimisticActions,
    collectionId
  } = sdk;
  const {
    api
  } = actionParams;
  const optimisticActions = getOptimisticActions(collectionId);
  return {
    onClick: () => {
      var _modalParams$actions, _modalParams$actions2;
      api.openConfirmDeleteModal({
        ...(modalParams.title ? {
          title: modalParams.title.text
        } : {}),
        primaryButtonOnClick: () => {
          var _modalParams$feedback;
          const schema = sdk.getSchema(sdk.collectionId);
          if (!schema) {
            throw new Error('Schema not found');
          }
          optimisticActions == null || optimisticActions.deleteOne(actionParams.item, {
            successToast: ((_modalParams$feedback = modalParams.feedback) == null || (_modalParams$feedback = _modalParams$feedback.successToast) == null ? void 0 : _modalParams$feedback.text) || 'Item deleted successfully',
            errorToast: () => {
              var _modalParams$feedback2;
              return ((_modalParams$feedback2 = modalParams.feedback) == null || (_modalParams$feedback2 = _modalParams$feedback2.errorToast) == null ? void 0 : _modalParams$feedback2.text) || 'Failed to delete item';
            },
            submit: async () => {
              const result = await schema.actions.delete(actionParams.item[schema.idField]);
              return [result];
            }
          });
        },
        ...((_modalParams$actions = modalParams.actions) != null && _modalParams$actions.submit ? {
          primaryButtonText: modalParams.actions.submit.text
        } : {}),
        ...((_modalParams$actions2 = modalParams.actions) != null && _modalParams$actions2.cancel ? {
          secondaryButtonText: modalParams.actions.cancel.text
        } : {}),
        ...(modalParams.description ? {
          content: modalParams.description.text
        } : {})
      });
    },
    label: 'Delete',
    icon: /*#__PURE__*/_react.default.createElement(_wixUiIconsCommon.Delete, {
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 51,
        columnNumber: 11
      }
    })
  };
};
exports.deleteAction = deleteAction;
