// @wix/auto-patterns/dist/cjs/utils/actions/updateAction.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.updateAction = void 0;
var _wixUiIconsCommon = require("@wix/wix-ui-icons-common");
var _react = _interopRequireDefault(require("react"));
var _jsxFileName = "/home/builduser/work/73d19ce3378ce4dd/packages/auto-patterns/dist/cjs/utils/actions/updateAction.tsx";
const updateAction = params => {
  const {
    action,
    sdk,
    actionParams
  } = params;
  const {
    appConfig
  } = sdk;
  return {
    onClick: () => {
      if (action.update.mode === 'page') {
        const pageId = action.update.page.id;
        const pageToNavigate = appConfig.pages.find(page => page.id === pageId);
        if (pageToNavigate) {
          if (pageToNavigate.type === 'entityPage') {
            const {
              navigateToEntityPage
            } = sdk;
            const item = actionParams.item;
            if (navigateToEntityPage && item) {
              navigateToEntityPage({
                item,
                entityPageId: pageId
              });
            }
          }
        }
      }
    },
    label: 'Update',
    icon: /*#__PURE__*/_react.default.createElement(_wixUiIconsCommon.Edit, {
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 32,
        columnNumber: 11
      }
    })
  };
};
exports.updateAction = updateAction;
