// @wix/design-system/dist/cjs/Accordion/Accordion.private.uni.driver.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.accordionPrivateDriverFactory = exports.accordionItemPrivateDriverFactory = void 0;
var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _AccordionUni = _interopRequireDefault(require("./Accordion.uni.driver"));
var _constants = require("./constants");
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2.default)(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var accordionPrivateDriverFactory = base => {
  var getItemAt = idx => base.$$("[data-hook=\"".concat(_constants.dataHooks.item, "\"]")).at(idx);
  return _objectSpread(_objectSpread({}, (0, _AccordionUni.default)(base)), {}, {
    getAmountOfItems: function () {
      var _getAmountOfItems = (0, _asyncToGenerator2.default)(function* () {
        return base.$$("[data-hook=\"".concat(_constants.dataHooks.item, "\"]")).count();
      });
      function getAmountOfItems() {
        return _getAmountOfItems.apply(this, arguments);
      }
      return getAmountOfItems;
    }(),
    hoverOnItem: function () {
      var _hoverOnItem = (0, _asyncToGenerator2.default)(function* (idx) {
        return yield getItemAt(idx).$('[data-hook="header"]').hover();
      });
      function hoverOnItem(_x) {
        return _hoverOnItem.apply(this, arguments);
      }
      return hoverOnItem;
    }(),
    getAccordionItemAt: function () {
      var _getAccordionItemAt = (0, _asyncToGenerator2.default)(function* (idx) {
        var accordionItem = yield getItemAt(idx);
        return accordionItemPrivateDriverFactory(accordionItem);
      });
      function getAccordionItemAt(_x2) {
        return _getAccordionItemAt.apply(this, arguments);
      }
      return getAccordionItemAt;
    }()
  });
};
exports.accordionPrivateDriverFactory = accordionPrivateDriverFactory;
var accordionItemPrivateDriverFactory = base => ({
  getToggleIconPlacement: function () {
    var _getToggleIconPlacement = (0, _asyncToGenerator2.default)(function* () {
      var _headerElement$lastEl, _headerElement$lastEl2;
      var header = yield base.get("[data-hook=\"".concat(_constants.dataHooks.header, "\"]"));
      var headerElement = yield header.getNative();
      var isAtStart = headerElement.children[0].getAttribute('data-hook') || headerElement.children[0].querySelector("[data-hook=\"".concat(_constants.dataHooks.toggleButton, "\"]"));
      var isAtEnd = ((_headerElement$lastEl = headerElement.lastElementChild) == null ? void 0 : _headerElement$lastEl.getAttribute('data-hook')) || ((_headerElement$lastEl2 = headerElement.lastElementChild) == null ? void 0 : _headerElement$lastEl2.querySelector("[data-hook=\"".concat(_constants.dataHooks.toggleButton, "\"]")));
      return isAtStart ? 'start' : isAtEnd ? 'end' : 'none';
    });
    function getToggleIconPlacement() {
      return _getToggleIconPlacement.apply(this, arguments);
    }
    return getToggleIconPlacement;
  }()
});
exports.accordionItemPrivateDriverFactory = accordionItemPrivateDriverFactory;
