// @wix/design-system/dist/cjs/Accordion/Accordion.uni.driver.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _unidriver = require("../utils/test-utils/unidriver");
var _constants = require("./constants");
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2.default)(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var accordionDriverFactory = base => _objectSpread(_objectSpread({}, (0, _unidriver.baseUniDriverFactory)(base)), {}, {
  // Gets the title by item index
  getItemTitleAt: idx => (0, _unidriver.findByHookAtIndex)(base, _constants.dataHooks.item, idx).$('[data-hook="title"]').text(),
  // Gets the subtitle by item index
  getItemSubtitleAt: idx => (0, _unidriver.findByHookAtIndex)(base, _constants.dataHooks.item, idx).$('[data-hook="accordion-item-subtitle"]').text(),
  // Checks whether an icon exist at item index
  /**
   * @deprecated
   * use 'isPrefixExistsAt' instead
   */
  isIconExistsAt: idx => (0, _unidriver.findByHookAtIndex)(base, _constants.dataHooks.item, idx).$('[data-hook="icon"]').exists(),
  // Checks whether prefix exists at item index
  isPrefixExistsAt: idx => (0, _unidriver.findByHookAtIndex)(base, _constants.dataHooks.item, idx).$('[data-hook="icon"]').exists(),
  // Checks whether suffix exists at item index
  isSuffixExistsAt: idx => (0, _unidriver.findByHookAtIndex)(base, _constants.dataHooks.item, idx).$('[data-hook="suffix"]').exists(),
  // Checks whether an item is expanded at index
  isItemExpandedAt: idx => (0, _unidriver.findByHookAtIndex)(base, _constants.dataHooks.item, idx).$('[data-hook="children"]').exists(),
  // Clicks toggle button at item index
  clickToggleButtonAt: idx => (0, _unidriver.findByHookAtIndex)(base, _constants.dataHooks.item, idx).$("[data-hook=\"".concat(_constants.dataHooks.toggleButton, "\"]")).click(),
  // Clicks item at index
  clickItemAt: idx => (0, _unidriver.findByHookAtIndex)(base, _constants.dataHooks.item, idx).$('[data-hook="header"]').click(),
  // Gets toggle button label at item index
  getToggleButtonLabelAt: idx => (0, _unidriver.findByHookAtIndex)(base, _constants.dataHooks.item, idx).$("[data-hook=\"".concat(_constants.dataHooks.toggleButton, "\"]")).text(),
  // Gets transition speed
  getTransitionSpeed: () => base.attr('data-transition-speed'),
  // Returns skin
  getSkin: () => base.attr('data-skin'),
  // Returns size of horizontal padding
  getHorizontalPadding: () => base.attr('data-horizontal-padding')
});
var _default = exports.default = accordionDriverFactory;
