// @wix/auto-patterns/dist/cjs/dataSourceAdapters/cms/cmsAdapter.js
"use strict";

exports.__esModule = true;
exports.CmsFieldTypeToPatternsFieldType = void 0;
exports.fetchCmsSchema = fetchCmsSchema;
var _data = require("@wix/data");
var _fetchCmsData = require("./fetchCmsData");
const CmsFieldTypeToPatternsFieldType = exports.CmsFieldTypeToPatternsFieldType = {
  [_data.collections.Type.TEXT]: 'SHORT_TEXT',
  [_data.collections.Type.DATE]: 'DATE',
  [_data.collections.Type.DATETIME]: 'DATETIME',
  [_data.collections.Type.NUMBER]: 'NUMBER',
  [_data.collections.Type.BOOLEAN]: 'BOOLEAN',
  [_data.collections.Type.URL]: 'URL',
  [_data.collections.Type.ADDRESS]: 'SHORT_TEXT',
  [_data.collections.Type.OBJECT]: 'LONG_TEXT',
  [_data.collections.Type.RICH_TEXT]: 'LONG_TEXT',
  [_data.collections.Type.RICH_CONTENT]: 'SHORT_TEXT',
  [_data.collections.Type.ARRAY]: 'ARRAY',
  [_data.collections.Type.ARRAY_STRING]: 'ARRAY',
  [_data.collections.Type.REFERENCE]: 'REFERENCE',
  [_data.collections.Type.IMAGE]: 'IMAGE'
};
async function fetchCmsSchema(collectionId) {
  var _schema$fields$find;
  const schema = await _data.collections.getDataCollection(collectionId);
  const transformedFields = schema.fields.reduce((acc, field) => {
    var _field$capabilities, _field$capabilities2, _field$typeMetadata;
    if (!field.key) {
      return acc;
    }
    const fieldType = field.type ? CmsFieldTypeToPatternsFieldType[field.type] || 'SHORT_TEXT' : 'SHORT_TEXT';
    const baseField = {
      id: field.key,
      displayName: field.displayName || '',
      validation: {
        numberRange: field.numberRange ? {
          min: field.numberRange.min ?? undefined,
          max: field.numberRange.max ?? undefined
        } : undefined,
        stringLengthRange: field.stringLengthRange ? {
          minLength: field.stringLengthRange.minLength ?? undefined,
          maxLength: field.stringLengthRange.maxLength ?? undefined
        } : undefined,
        required: field.required ?? false
      },
      capabilities: {
        supportedQueryOperators: ((_field$capabilities = field.capabilities) == null ? void 0 : _field$capabilities.queryOperators) || [],
        sortable: ((_field$capabilities2 = field.capabilities) == null ? void 0 : _field$capabilities2.sortable) || false
      }
    };
    if (fieldType === 'REFERENCE' && (_field$typeMetadata = field.typeMetadata) != null && (_field$typeMetadata = _field$typeMetadata.reference) != null && _field$typeMetadata.referencedCollectionId) {
      const referenceField = {
        ...baseField,
        type: 'REFERENCE',
        referenceMetadata: {
          referencedCollectionId: field.typeMetadata.reference.referencedCollectionId
        }
      };
      acc[field.key] = referenceField;
    } else {
      const nonReferenceType = fieldType === 'REFERENCE' ? 'SHORT_TEXT' : fieldType;
      const nonReferenceField = {
        ...baseField,
        type: nonReferenceType
      };
      acc[field.key] = nonReferenceField;
    }
    return acc;
  }, {});
  const schemaConfig = {
    id: collectionId,
    displayField: schema.displayField ?? '_id',
    idField: '_id',
    imageField: (_schema$fields$find = schema.fields.find(field => field.type === 'IMAGE')) == null ? void 0 : _schema$fields$find.key,
    fields: transformedFields,
    actions: {
      get: entityId => {
        return _data.items.get(collectionId, entityId);
      },
      create: entity => {
        return _data.items.insert(collectionId, entity);
      },
      update: entity => {
        return _data.items.update(collectionId, entity);
      },
      delete: entityId => {
        return _data.items.remove(collectionId, entityId);
      },
      bulkDelete: entityIds => {
        return _data.items.bulkRemove(collectionId, entityIds);
      },
      find: async (query, options = {}) => {
        return (0, _fetchCmsData.fetchCmsData)(collectionId, query, transformedFields, options);
      }
    }
  };
  return schemaConfig;
}
