// @wix/bex-core/dist/cjs/hooks/createCollectionConfig.js
"use strict";

exports.__esModule = true;
exports.createCollectionConfig = createCollectionConfig;
var _assertKey = require("../util/assertKey");
function createCollectionConfig(config) {
  return {
    itemKey: item => {
      (0, _assertKey.assertOneOfKeys)(item, ['id', '_id'], {
        errorMessage: keys => `Collection config must specify itemKey, or item must have one of properties: ${keys.join(', ')}`
      });
      return item.id ?? item._id;
    },
    itemName: item => {
      (0, _assertKey.assertStringKey)(item, 'name', {
        errorMessage: key => `Collection config must specify itemName, or item must have property: ${key}`
      });
      return item.name;
    },
    toExtendedFields: item => {
      (0, _assertKey.assertKey)(item, 'extendedFields', {
        errorMessage: key => `Collection config must specify toExtendedFields, or item must have property: ${key}`
      });
      return item.extendedFields;
    },
    ...config,
    fetchData: config.fetchData,
    filters: config.filters ?? {}
  };
}
