// @wix/auto_sdk_events_guests/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  AttendanceStatusOriginal: () => AttendanceStatus,
  EventTypeOriginal: () => EventType,
  FeeNameOriginal: () => FeeName,
  FeeTypeOriginal: () => FeeType,
  GuestTypeOriginal: () => GuestType,
  LocationTypeOriginal: () => LocationType,
  MemberEventStatusUpdatedEventTypeOriginal: () => MemberEventStatusUpdatedEventType,
  NotifyActionTypeOriginal: () => NotifyActionType,
  OrderStatusOriginal: () => OrderStatus,
  OrderTypeOriginal: () => OrderType,
  RecurrenceStatusStatusOriginal: () => RecurrenceStatusStatus,
  RequestedFieldsEnumRequestedFieldsOriginal: () => RequestedFieldsEnumRequestedFields,
  RequestedFieldsOriginal: () => RequestedFields,
  SortOrderOriginal: () => SortOrder,
  StatusOriginal: () => Status,
  SubdivisionSubdivisionTypeOriginal: () => SubdivisionSubdivisionType,
  SubdivisionTypeOriginal: () => SubdivisionType,
  TaxTypeOriginal: () => TaxType,
  TimingOriginal: () => Timing,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  queryGuests: () => queryGuests
});
module.exports = __toCommonJS(meta_exports);

// src/events-guests-v1-guest-guests.http.ts
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
function resolveWixEventsGuestsV1EventGuestsServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/events-guests",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/events-guests",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/events-guests",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/events-guests",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/events-guests",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/events-guests",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/events/v2/guests",
        destPath: "/v2/guests"
      },
      {
        srcPath: "/events-guests/v2/guests",
        destPath: "/v2/guests"
      }
    ]
  };
  return (0, import_rest_modules.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_events_guests";
function queryEventGuests(payload) {
  function __queryEventGuests({ host }) {
    const metadata = {
      entityFqdn: "wix.events.guests.v1.guest",
      method: "POST",
      methodFqn: "wix.events.guests.v1.EventGuestsService.QueryEventGuests",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsGuestsV1EventGuestsServiceUrl({
        protoPath: "/v2/guests/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "guests.createdDate" },
            { path: "guests.updatedDate" },
            { path: "guests.attendanceStatusUpdatedDate" },
            { path: "guests.guestDetails.formResponse.inputValues.dateTime" }
          ]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [
            { path: "guests.guestDetails.formResponse.inputValues.number" },
            {
              path: "guests.guestDetails.formResponse.inputValues.address.address.geocode.latitude"
            },
            {
              path: "guests.guestDetails.formResponse.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryEventGuests;
}

// src/events-guests-v1-guest-guests.types.ts
var SubdivisionType = /* @__PURE__ */ ((SubdivisionType2) => {
  SubdivisionType2["UNKNOWN_SUBDIVISION_TYPE"] = "UNKNOWN_SUBDIVISION_TYPE";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_1"] = "ADMINISTRATIVE_AREA_LEVEL_1";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_2"] = "ADMINISTRATIVE_AREA_LEVEL_2";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_3"] = "ADMINISTRATIVE_AREA_LEVEL_3";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_4"] = "ADMINISTRATIVE_AREA_LEVEL_4";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_5"] = "ADMINISTRATIVE_AREA_LEVEL_5";
  SubdivisionType2["COUNTRY"] = "COUNTRY";
  return SubdivisionType2;
})(SubdivisionType || {});
var AttendanceStatus = /* @__PURE__ */ ((AttendanceStatus2) => {
  AttendanceStatus2["NOT_ATTENDING"] = "NOT_ATTENDING";
  AttendanceStatus2["ATTENDING"] = "ATTENDING";
  AttendanceStatus2["IN_WAITLIST"] = "IN_WAITLIST";
  return AttendanceStatus2;
})(AttendanceStatus || {});
var GuestType = /* @__PURE__ */ ((GuestType2) => {
  GuestType2["RSVP"] = "RSVP";
  GuestType2["BUYER"] = "BUYER";
  GuestType2["TICKET_HOLDER"] = "TICKET_HOLDER";
  return GuestType2;
})(GuestType || {});
var MemberEventStatusUpdatedEventType = /* @__PURE__ */ ((MemberEventStatusUpdatedEventType2) => {
  MemberEventStatusUpdatedEventType2["MEMBER_JOINED"] = "MEMBER_JOINED";
  MemberEventStatusUpdatedEventType2["MEMBER_LEFT"] = "MEMBER_LEFT";
  MemberEventStatusUpdatedEventType2["ATTENDING_MEMBER_JOINED"] = "ATTENDING_MEMBER_JOINED";
  MemberEventStatusUpdatedEventType2["LAST_ATTENDING_MEMBER_LEFT"] = "LAST_ATTENDING_MEMBER_LEFT";
  return MemberEventStatusUpdatedEventType2;
})(MemberEventStatusUpdatedEventType || {});
var EventType = /* @__PURE__ */ ((EventType2) => {
  EventType2["CONTACT_JOINED"] = "CONTACT_JOINED";
  EventType2["CONTACT_LEFT"] = "CONTACT_LEFT";
  EventType2["LAST_CONTACT_LEFT"] = "LAST_CONTACT_LEFT";
  return EventType2;
})(EventType || {});
var LocationType = /* @__PURE__ */ ((LocationType2) => {
  LocationType2["UNKNOWN_LOCATION"] = "UNKNOWN_LOCATION";
  LocationType2["VENUE"] = "VENUE";
  LocationType2["ONLINE"] = "ONLINE";
  return LocationType2;
})(LocationType || {});
var SubdivisionSubdivisionType = /* @__PURE__ */ ((SubdivisionSubdivisionType2) => {
  SubdivisionSubdivisionType2["UNKNOWN_SUBDIVISION_TYPE"] = "UNKNOWN_SUBDIVISION_TYPE";
  SubdivisionSubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_1"] = "ADMINISTRATIVE_AREA_LEVEL_1";
  SubdivisionSubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_2"] = "ADMINISTRATIVE_AREA_LEVEL_2";
  SubdivisionSubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_3"] = "ADMINISTRATIVE_AREA_LEVEL_3";
  SubdivisionSubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_4"] = "ADMINISTRATIVE_AREA_LEVEL_4";
  SubdivisionSubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_5"] = "ADMINISTRATIVE_AREA_LEVEL_5";
  SubdivisionSubdivisionType2["COUNTRY"] = "COUNTRY";
  return SubdivisionSubdivisionType2;
})(SubdivisionSubdivisionType || {});
var RecurrenceStatusStatus = /* @__PURE__ */ ((RecurrenceStatusStatus2) => {
  RecurrenceStatusStatus2["UNKNOWN_STATUS"] = "UNKNOWN_STATUS";
  RecurrenceStatusStatus2["ONE_TIME"] = "ONE_TIME";
  RecurrenceStatusStatus2["RECURRING"] = "RECURRING";
  RecurrenceStatusStatus2["RECURRING_UPCOMING"] = "RECURRING_UPCOMING";
  RecurrenceStatusStatus2["RECURRING_RECENTLY_ENDED"] = "RECURRING_RECENTLY_ENDED";
  RecurrenceStatusStatus2["RECURRING_RECENTLY_CANCELED"] = "RECURRING_RECENTLY_CANCELED";
  return RecurrenceStatusStatus2;
})(RecurrenceStatusStatus || {});
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["UNKNOWN_EVENT_STATUS"] = "UNKNOWN_EVENT_STATUS";
  Status2["UPCOMING"] = "UPCOMING";
  Status2["STARTED"] = "STARTED";
  Status2["ENDED"] = "ENDED";
  Status2["CANCELED"] = "CANCELED";
  Status2["DRAFT"] = "DRAFT";
  return Status2;
})(Status || {});
var NotifyActionType = /* @__PURE__ */ ((NotifyActionType2) => {
  NotifyActionType2["UNKNOWN"] = "UNKNOWN";
  NotifyActionType2["EMAIL"] = "EMAIL";
  NotifyActionType2["AUTOMATION_TRIGGER"] = "AUTOMATION_TRIGGER";
  NotifyActionType2["PUSH"] = "PUSH";
  NotifyActionType2["EVENT_CANCELED"] = "EVENT_CANCELED";
  NotifyActionType2["EVENT_STARTS"] = "EVENT_STARTS";
  NotifyActionType2["ORDER_CANCELED"] = "ORDER_CANCELED";
  return NotifyActionType2;
})(NotifyActionType || {});
var Timing = /* @__PURE__ */ ((Timing2) => {
  Timing2["UNKNOWN_TIMING"] = "UNKNOWN_TIMING";
  Timing2["NOW"] = "NOW";
  Timing2["STARTS_IN_1_DAY"] = "STARTS_IN_1_DAY";
  Timing2["STARTS_IN_3_DAYS"] = "STARTS_IN_3_DAYS";
  Timing2["STARTS_IN_1_WEEK"] = "STARTS_IN_1_WEEK";
  Timing2["STARTS_IN_1_HOUR"] = "STARTS_IN_1_HOUR";
  Timing2["STARTS_IN_30_MINUTES"] = "STARTS_IN_30_MINUTES";
  Timing2["STARTS_IN_2_HOURS"] = "STARTS_IN_2_HOURS";
  return Timing2;
})(Timing || {});
var OrderStatus = /* @__PURE__ */ ((OrderStatus2) => {
  OrderStatus2["NA_ORDER_STATUS"] = "NA_ORDER_STATUS";
  OrderStatus2["FREE"] = "FREE";
  OrderStatus2["PENDING"] = "PENDING";
  OrderStatus2["PAID"] = "PAID";
  OrderStatus2["OFFLINE_PENDING"] = "OFFLINE_PENDING";
  OrderStatus2["INITIATED"] = "INITIATED";
  OrderStatus2["CANCELED"] = "CANCELED";
  OrderStatus2["DECLINED"] = "DECLINED";
  OrderStatus2["AUTHORIZED"] = "AUTHORIZED";
  OrderStatus2["VOIDED"] = "VOIDED";
  OrderStatus2["PARTIALLY_PAID"] = "PARTIALLY_PAID";
  return OrderStatus2;
})(OrderStatus || {});
var TaxType = /* @__PURE__ */ ((TaxType2) => {
  TaxType2["INCLUDED"] = "INCLUDED";
  TaxType2["ADDED"] = "ADDED";
  TaxType2["ADDED_AT_CHECKOUT"] = "ADDED_AT_CHECKOUT";
  return TaxType2;
})(TaxType || {});
var FeeName = /* @__PURE__ */ ((FeeName2) => {
  FeeName2["WIX_FEE"] = "WIX_FEE";
  return FeeName2;
})(FeeName || {});
var FeeType = /* @__PURE__ */ ((FeeType2) => {
  FeeType2["FEE_ADDED"] = "FEE_ADDED";
  FeeType2["FEE_INCLUDED"] = "FEE_INCLUDED";
  FeeType2["FEE_ADDED_AT_CHECKOUT"] = "FEE_ADDED_AT_CHECKOUT";
  return FeeType2;
})(FeeType || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var RequestedFieldsEnumRequestedFields = /* @__PURE__ */ ((RequestedFieldsEnumRequestedFields2) => {
  RequestedFieldsEnumRequestedFields2["UNKNOWN_REQUESTED_FIELD"] = "UNKNOWN_REQUESTED_FIELD";
  RequestedFieldsEnumRequestedFields2["GUEST_DETAILS"] = "GUEST_DETAILS";
  RequestedFieldsEnumRequestedFields2["GUEST_TOTAL"] = "GUEST_TOTAL";
  return RequestedFieldsEnumRequestedFields2;
})(RequestedFieldsEnumRequestedFields || {});
var OrderType = /* @__PURE__ */ ((OrderType2) => {
  OrderType2["UNASSIGNED_TICKETS"] = "UNASSIGNED_TICKETS";
  OrderType2["ASSIGNED_TICKETS"] = "ASSIGNED_TICKETS";
  return OrderType2;
})(OrderType || {});
var RequestedFields = /* @__PURE__ */ ((RequestedFields2) => {
  RequestedFields2["UNKNOWN_REQUESTED_FIELD"] = "UNKNOWN_REQUESTED_FIELD";
  RequestedFields2["MEMBER_GUESTS"] = "MEMBER_GUESTS";
  RequestedFields2["WAITLIST_COUNT"] = "WAITLIST_COUNT";
  return RequestedFields2;
})(RequestedFields || {});

// src/events-guests-v1-guest-guests.meta.ts
function queryGuests() {
  const payload = {};
  const getRequestOptions = queryEventGuests(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/guests/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AttendanceStatusOriginal,
  EventTypeOriginal,
  FeeNameOriginal,
  FeeTypeOriginal,
  GuestTypeOriginal,
  LocationTypeOriginal,
  MemberEventStatusUpdatedEventTypeOriginal,
  NotifyActionTypeOriginal,
  OrderStatusOriginal,
  OrderTypeOriginal,
  RecurrenceStatusStatusOriginal,
  RequestedFieldsEnumRequestedFieldsOriginal,
  RequestedFieldsOriginal,
  SortOrderOriginal,
  StatusOriginal,
  SubdivisionSubdivisionTypeOriginal,
  SubdivisionTypeOriginal,
  TaxTypeOriginal,
  TimingOriginal,
  WebhookIdentityTypeOriginal,
  queryGuests
});
