// @wix/auto_sdk_headless-site-assets_scripts/build/es/index.mjs
// src/assets-v1-site-script-scripts.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/assets-v1-site-script-scripts.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixAssetsScriptsV1SiteScriptsServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/_api/site-scripts",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/site-scripts",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/site-scripts",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/assets/scripts/v1/site-scripts",
        destPath: "/v1/site-scripts"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_headless-site-assets_scripts";
function listSiteScripts(payload) {
  function __listSiteScripts({ host }) {
    const metadata = {
      entityFqdn: "wix.assets.v1.site_script",
      method: "GET",
      methodFqn: "wix.assets.scripts.v1.SiteScriptsService.ListSiteScripts",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAssetsScriptsV1SiteScriptsServiceUrl({
        protoPath: "/v1/site-scripts",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __listSiteScripts;
}
function getEssentialProperties(payload) {
  function __getEssentialProperties({ host }) {
    const metadata = {
      entityFqdn: "wix.assets.v1.site_script",
      method: "GET",
      methodFqn: "wix.assets.scripts.v1.SiteScriptsService.GetEssentialProperties",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAssetsScriptsV1SiteScriptsServiceUrl({
        protoPath: "/v1/site-scripts/essentials",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getEssentialProperties;
}

// src/assets-v1-site-script-scripts.universal.ts
var PositionOnPage = /* @__PURE__ */ ((PositionOnPage2) => {
  PositionOnPage2["UNKNOWN_POSITION"] = "UNKNOWN_POSITION";
  PositionOnPage2["HEAD"] = "HEAD";
  PositionOnPage2["BODY_START"] = "BODY_START";
  PositionOnPage2["BODY_END"] = "BODY_END";
  return PositionOnPage2;
})(PositionOnPage || {});
async function listSiteScripts2() {
  const { httpClient, sideEffects } = arguments[0];
  const payload = renameKeysFromSDKRequestToRESTRequest({});
  const reqOpts = listSiteScripts(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {},
        singleArgumentUnchanged: false
      },
      []
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getEssentialProperties2() {
  const { httpClient, sideEffects } = arguments[0];
  const payload = renameKeysFromSDKRequestToRESTRequest({});
  const reqOpts = getEssentialProperties(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {},
        singleArgumentUnchanged: false
      },
      []
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/assets-v1-site-script-scripts.public.ts
function listSiteScripts3(httpClient) {
  return () => listSiteScripts2(
    // @ts-ignore
    { httpClient }
  );
}
function getEssentialProperties3(httpClient) {
  return () => getEssentialProperties2(
    // @ts-ignore
    { httpClient }
  );
}

// src/assets-v1-site-script-scripts.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
var listSiteScripts4 = /* @__PURE__ */ createRESTModule(listSiteScripts3);
var getEssentialProperties4 = /* @__PURE__ */ createRESTModule(getEssentialProperties3);
export {
  PositionOnPage,
  getEssentialProperties4 as getEssentialProperties,
  listSiteScripts4 as listSiteScripts
};
