// @wix/auto_sdk_headless-site-assets_scripts/build/es/meta.mjs
// src/assets-v1-site-script-scripts.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixAssetsScriptsV1SiteScriptsServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/_api/site-scripts",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/site-scripts",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/site-scripts",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/assets/scripts/v1/site-scripts",
        destPath: "/v1/site-scripts"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_headless-site-assets_scripts";
function listSiteScripts(payload) {
  function __listSiteScripts({ host }) {
    const metadata = {
      entityFqdn: "wix.assets.v1.site_script",
      method: "GET",
      methodFqn: "wix.assets.scripts.v1.SiteScriptsService.ListSiteScripts",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAssetsScriptsV1SiteScriptsServiceUrl({
        protoPath: "/v1/site-scripts",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __listSiteScripts;
}
function getEssentialProperties(payload) {
  function __getEssentialProperties({ host }) {
    const metadata = {
      entityFqdn: "wix.assets.v1.site_script",
      method: "GET",
      methodFqn: "wix.assets.scripts.v1.SiteScriptsService.GetEssentialProperties",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAssetsScriptsV1SiteScriptsServiceUrl({
        protoPath: "/v1/site-scripts/essentials",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getEssentialProperties;
}

// src/assets-v1-site-script-scripts.types.ts
var PositionOnPage = /* @__PURE__ */ ((PositionOnPage2) => {
  PositionOnPage2["UNKNOWN_POSITION"] = "UNKNOWN_POSITION";
  PositionOnPage2["HEAD"] = "HEAD";
  PositionOnPage2["BODY_START"] = "BODY_START";
  PositionOnPage2["BODY_END"] = "BODY_END";
  return PositionOnPage2;
})(PositionOnPage || {});

// src/assets-v1-site-script-scripts.meta.ts
function listSiteScripts2() {
  const payload = {};
  const getRequestOptions = listSiteScripts(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/site-scripts",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getEssentialProperties2() {
  const payload = {};
  const getRequestOptions = getEssentialProperties(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/site-scripts/essentials",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  PositionOnPage as PositionOnPageOriginal,
  getEssentialProperties2 as getEssentialProperties,
  listSiteScripts2 as listSiteScripts
};
