// @wix/auto_sdk_forms_chat-settings/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  SortOrderOriginal: () => SortOrder,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  createChatSettings: () => createChatSettings2,
  deleteChatSettings: () => deleteChatSettings2,
  getChatSettings: () => getChatSettings2,
  queryChatSettings: () => queryChatSettings2,
  updateChatSettings: () => updateChatSettings2
});
module.exports = __toCommonJS(meta_exports);

// src/forms-ai-v1-chat-settings-chat-settings.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixFormsAiV1ChatSettingsServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/_api/form-ai-assistant",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/form-ai-assistant",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/form-ai-assistant",
        destPath: ""
      }
    ],
    "*.wixforms.com": [
      {
        srcPath: "/_api/form-ai-assistant",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/form-ai-assistant",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/form-ai-assistant",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/forms/ai/v1/chat-settings",
        destPath: "/v1/chat-settings"
      },
      {
        srcPath: "/forms/ai/v1/interactive-form-sessions",
        destPath: "/v1/interactive-form-sessions"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/form-ai-assistant",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_forms_chat-settings";
function createChatSettings(payload) {
  function __createChatSettings({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "chatSettings.createdDate" },
          { path: "chatSettings.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.forms.ai.v1.chat_settings",
      method: "POST",
      methodFqn: "wix.forms.ai.v1.ChatSettingsService.CreateChatSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsAiV1ChatSettingsServiceUrl({
        protoPath: "/v1/chat-settings",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "chatSettings.createdDate" },
            { path: "chatSettings.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createChatSettings;
}
function updateChatSettings(payload) {
  function __updateChatSettings({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "chatSettings.createdDate" },
          { path: "chatSettings.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.forms.ai.v1.chat_settings",
      method: "PATCH",
      methodFqn: "wix.forms.ai.v1.ChatSettingsService.UpdateChatSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsAiV1ChatSettingsServiceUrl({
        protoPath: "/v1/chat-settings/{chatSettings.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "chatSettings.createdDate" },
            { path: "chatSettings.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateChatSettings;
}
function deleteChatSettings(payload) {
  function __deleteChatSettings({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.ai.v1.chat_settings",
      method: "DELETE",
      methodFqn: "wix.forms.ai.v1.ChatSettingsService.DeleteChatSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsAiV1ChatSettingsServiceUrl({
        protoPath: "/v1/chat-settings/{chatSettingsId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteChatSettings;
}
function getChatSettings(payload) {
  function __getChatSettings({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.ai.v1.chat_settings",
      method: "GET",
      methodFqn: "wix.forms.ai.v1.ChatSettingsService.GetChatSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsAiV1ChatSettingsServiceUrl({
        protoPath: "/v1/chat-settings/{chatSettingsId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "chatSettings.createdDate" },
            { path: "chatSettings.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getChatSettings;
}
function queryChatSettings(payload) {
  function __queryChatSettings({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.ai.v1.chat_settings",
      method: "GET",
      methodFqn: "wix.forms.ai.v1.ChatSettingsService.QueryChatSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsAiV1ChatSettingsServiceUrl({
        protoPath: "/v1/chat-settings/query",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload, true),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "chatSettings.createdDate" },
            { path: "chatSettings.updatedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixFormsAiV1ChatSettingsServiceUrl({
            protoPath: "/v1/chat-settings/query",
            data: payload,
            host
          }),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryChatSettings;
}

// src/forms-ai-v1-chat-settings-chat-settings.types.ts
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/forms-ai-v1-chat-settings-chat-settings.meta.ts
function createChatSettings2() {
  const payload = {};
  const getRequestOptions = createChatSettings(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/chat-settings",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateChatSettings2() {
  const payload = { chatSettings: { id: ":chatSettingsId" } };
  const getRequestOptions = updateChatSettings(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/chat-settings/{chatSettings.id}",
    pathParams: { chatSettingsId: "chatSettingsId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteChatSettings2() {
  const payload = { chatSettingsId: ":chatSettingsId" };
  const getRequestOptions = deleteChatSettings(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/chat-settings/{chatSettingsId}",
    pathParams: { chatSettingsId: "chatSettingsId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getChatSettings2() {
  const payload = { chatSettingsId: ":chatSettingsId" };
  const getRequestOptions = getChatSettings(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/chat-settings/{chatSettingsId}",
    pathParams: { chatSettingsId: "chatSettingsId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryChatSettings2() {
  const payload = {};
  const getRequestOptions = queryChatSettings(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/chat-settings/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  SortOrderOriginal,
  WebhookIdentityTypeOriginal,
  createChatSettings,
  deleteChatSettings,
  getChatSettings,
  queryChatSettings,
  updateChatSettings
});
