// @wix/ricos/dist/es/ricos-common-plugins/bold/bold.js
import { Bold } from "@wix/wix-ui-icons-common";
import { Decoration_Type } from "./../../ricos-schema/legacy/wix/rich_content/v1/index.js";
import { createShortcut } from "./../../ricos-types/index.js";
import { isAlwaysVisible } from "./../../_shared/resolvers/is-always-visible.js";
import { isBoldInText } from "./../../_shared/resolvers/is-bold-in-text.js";
import { isNodeSelected } from "./../../_shared/resolvers/is-node-selected.js";
import { boldConverter } from "./bold-converter.js";
import { TEXT_BUTTON_BOLD } from "./data-hooks.js";
import { bold } from "./extension.js";
const pluginBold = {
  type: Decoration_Type.BOLD,
  config: {},
  pmNodeConverters: [],
  pmMarkConverters: [boldConverter],
  tiptapExtensions: [bold],
  shortcuts: [
    createShortcut({
      name: Decoration_Type.BOLD,
      description: "ShortcutsSheet_Formatting_Bold",
      keys: { macOs: "Meta+B", windows: "Ctrl+B" },
      command({ publicCommands }) {
        publicCommands.toggleInlineStyle("bold");
        return "continue";
      },
      group: "ShortcutsSheet_Section_Formatting",
      enabled: true,
      showInShortcutSheet: true
    })
  ],
  textButtons: [
    {
      id: Decoration_Type.BOLD,
      type: "toggle",
      presentation: {
        dataHook: TEXT_BUTTON_BOLD,
        tooltip: "BoldButton_Tooltip",
        icon: Bold
      },
      attributes: {
        visible: isAlwaysVisible,
        active: isBoldInText,
        disabled: isNodeSelected
      },
      command: ({ publicCommands }) => () => {
        publicCommands.toggleInlineStyle("bold");
        return true;
      }
    }
  ]
};
export {
  pluginBold
};
