// @wix/auto_sdk_benefit-programs_items/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  bulkCreateItems: () => bulkCreateItems2,
  bulkDeleteItems: () => bulkDeleteItems2,
  bulkDeleteItemsByFilter: () => bulkDeleteItemsByFilter2,
  bulkUpdateItems: () => bulkUpdateItems2,
  countItems: () => countItems2,
  createItem: () => createItem2,
  deleteItem: () => deleteItem2,
  getItem: () => getItem2,
  listItems: () => listItems2,
  queryItems: () => queryItems2,
  updateItem: () => updateItem2
});
module.exports = __toCommonJS(meta_exports);

// src/benefit-programs-v1-item-items.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixBenefitProgramsV1ItemItemServiceUrl(opts) {
  const domainToMappings = {
    _: [
      {
        srcPath: "/_api/benefit-programs-items",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-items",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-items",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/benefit-programs-items",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/benefit-programs-items",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/items",
        destPath: "/v1/bulk/items"
      },
      {
        srcPath: "/_api/benefit-programs/v1/items",
        destPath: "/v1/items"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-items",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-items",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/benefit-programs-items",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/benefit-programs/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/benefit-programs/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/benefit-programs/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_benefit-programs_items";
function createItem(payload) {
  function __createItem({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "item.createdDate" }, { path: "item.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.CreateItem",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/items",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "item.createdDate" }, { path: "item.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __createItem;
}
function bulkCreateItems(payload) {
  function __bulkCreateItems({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "items.createdDate" }, { path: "items.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.BulkCreateItems",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/bulk/items/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateItems;
}
function deleteItem(payload) {
  function __deleteItem({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.DeleteItem",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/items/{itemId}/delete",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __deleteItem;
}
function bulkDeleteItems(payload) {
  function __bulkDeleteItems({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.BulkDeleteItems",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/bulk/items/delete",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkDeleteItems;
}
function bulkDeleteItemsByFilter(payload) {
  function __bulkDeleteItemsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.BulkDeleteItemsByFilter",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/bulk/items/delete-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkDeleteItemsByFilter;
}
function updateItem(payload) {
  function __updateItem({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "item.createdDate" }, { path: "item.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "PATCH",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.UpdateItem",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/items/{item.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "item.createdDate" }, { path: "item.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __updateItem;
}
function bulkUpdateItems(payload) {
  function __bulkUpdateItems({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "items.fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "items.item.createdDate" },
          { path: "items.item.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.BulkUpdateItems",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/bulk/items/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateItems;
}
function getItem(payload) {
  function __getItem({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.GetItem",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/items/{itemId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "item.createdDate" }, { path: "item.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __getItem;
}
function listItems(payload) {
  function __listItems({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.ListItems",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/items",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "items.createdDate" },
            { path: "items.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listItems;
}
function queryItems(payload) {
  function __queryItems({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.QueryItems",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/items/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "items.createdDate" },
            { path: "items.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryItems;
}
function countItems(payload) {
  function __countItems({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.CountItems",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/items/count",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __countItems;
}

// src/benefit-programs-v1-item-items.meta.ts
function createItem2() {
  const payload = {};
  const getRequestOptions = createItem(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/items",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkCreateItems2() {
  const payload = {};
  const getRequestOptions = bulkCreateItems(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/items/create",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteItem2() {
  const payload = { itemId: ":itemId" };
  const getRequestOptions = deleteItem(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/items/{itemId}/delete",
    pathParams: { itemId: "itemId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkDeleteItems2() {
  const payload = {};
  const getRequestOptions = bulkDeleteItems(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/items/delete",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkDeleteItemsByFilter2() {
  const payload = {};
  const getRequestOptions = bulkDeleteItemsByFilter(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/items/delete-by-filter",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateItem2() {
  const payload = { item: { id: ":itemId" } };
  const getRequestOptions = updateItem(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/items/{item.id}",
    pathParams: { itemId: "itemId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateItems2() {
  const payload = {};
  const getRequestOptions = bulkUpdateItems(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/items/update",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getItem2() {
  const payload = { itemId: ":itemId" };
  const getRequestOptions = getItem(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/items/{itemId}",
    pathParams: { itemId: "itemId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listItems2() {
  const payload = {};
  const getRequestOptions = listItems(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/items",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryItems2() {
  const payload = {};
  const getRequestOptions = queryItems(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/items/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function countItems2() {
  const payload = {};
  const getRequestOptions = countItems(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/items/count",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  bulkCreateItems,
  bulkDeleteItems,
  bulkDeleteItemsByFilter,
  bulkUpdateItems,
  countItems,
  createItem,
  deleteItem,
  getItem,
  listItems,
  queryItems,
  updateItem
});
