// @wix/ricos/dist/cjs/plugin-table/tiptap/decorations/addColDecoration.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var addColDecoration_exports = {};
__export(addColDecoration_exports, {
  addColDecoration: () => addColDecoration
});
module.exports = __toCommonJS(addColDecoration_exports);
var import_view = require("@tiptap/pm/view");
var import_consts = require("../../consts.js");
var import_types = require("../../types.js");
var import_TableQuery = require("../TableQuery.js");
var import_addRowCol_scss = __toESM(require("./addRowCol-scss.js"));
var import_svgs = require("./svgs.js");
const addColDecoration = (state, editor, services) => {
  const table = import_TableQuery.TableQuery.of(state.selection);
  const colNum = table.getWidth();
  const addColDiv = document.createElement("div");
  addColDiv.innerHTML = import_svgs.addSvg;
  addColDiv.classList.add(import_addRowCol_scss.default.tableAddCol);
  addColDiv.dataset.hook = "addCol";
  addColDiv.addEventListener("mousedown", (e) => {
    e.stopPropagation();
    e.preventDefault();
    editor.chain().addColumnAtIndex(colNum).selectColumnAtIndex(colNum).run();
    services.pluginsEvents.publishTableAddRowCol({
      pluginId: import_types.TABLE_TYPE,
      nodeId: table.getNode().attrs.id,
      category: import_consts.CATEGORY.COLUMN,
      location: import_consts.LOCATION.RIGHT,
      source: import_consts.SOURCE.PLUS_BUTTON
    });
  });
  const colDecoration = import_view.Decoration.widget(table.getStartPos(), addColDiv);
  return colDecoration;
};
