// @wix/patterns/dist/cjs/fields/Dropdown/Dropdown.js
"use strict";

exports.__esModule = true;
exports.Dropdown = void 0;
var _react = _interopRequireWildcard(require("react"));
var _designSystem = require("@wix/design-system");
var _Field = require("../Field");
var _dataHooks = require("../dataHooks");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/fields/Dropdown/Dropdown.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
const Dropdown = ({
  dataHook,
  schema,
  initialValue,
  onChange
}) => {
  const options = (0, _react.useMemo)(() => {
    var _schema$options;
    return ((_schema$options = schema.options) == null ? void 0 : _schema$options.map(o => ({
      id: o,
      value: o
    }))) || [];
  }, [schema]);
  return /*#__PURE__*/_react.default.createElement(_Field.Field, {
    dataHook: dataHook,
    type: schema.type,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 25,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.FormField, {
    label: schema.label,
    dataHook: _dataHooks.dataHooks.Label,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 26,
      columnNumber: 7
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.Dropdown, {
    dataHook: _dataHooks.dataHooks.Input,
    options: options,
    initialSelectedId: initialValue,
    onSelect: ({
      id
    }) => onChange(id),
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 27,
      columnNumber: 9
    }
  })));
};
exports.Dropdown = Dropdown;
