// @wix/auto_sdk_payments_refunds/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  createRefund: () => createRefund2,
  getRefund: () => getRefund2,
  queryRefunds: () => queryRefunds2
});
module.exports = __toCommonJS(meta_exports);

// src/payments-refunds-v1-refund-refunds.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixPaymentsRefundsV1RefundServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/payments/refunds/v1",
        destPath: "/v1"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/payments/refunds/v1/refunds",
        destPath: "/v1/refunds"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/payments/refunds/v1/refunds",
        destPath: "/v1/refunds"
      }
    ],
    _: [
      {
        srcPath: "/payments/refunds/v1/refunds",
        destPath: "/v1/refunds"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_refunds";
function createRefund(payload) {
  function __createRefund({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "refund.createdDate" }, { path: "refund.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.payments.refunds.v1.refund",
      method: "POST",
      methodFqn: "wix.payments.refunds.v1.RefundService.CreateRefund",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsRefundsV1RefundServiceUrl({
        protoPath: "/v1/refunds",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "refund.createdDate" },
            { path: "refund.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createRefund;
}
function getRefund(payload) {
  function __getRefund({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.refunds.v1.refund",
      method: "GET",
      methodFqn: "wix.payments.refunds.v1.RefundService.GetRefund",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsRefundsV1RefundServiceUrl({
        protoPath: "/v1/refunds/{refundId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "refund.createdDate" },
            { path: "refund.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getRefund;
}
function queryRefunds(payload) {
  function __queryRefunds({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.refunds.v1.refund",
      method: "POST",
      methodFqn: "wix.payments.refunds.v1.RefundService.QueryRefunds",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsRefundsV1RefundServiceUrl({
        protoPath: "/v1/refunds/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "refunds.createdDate" },
            { path: "refunds.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryRefunds;
}

// src/payments-refunds-v1-refund-refunds.meta.ts
function createRefund2() {
  const payload = {};
  const getRequestOptions = createRefund(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/refunds",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getRefund2() {
  const payload = { refundId: ":refundId" };
  const getRequestOptions = getRefund(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/refunds/{refundId}",
    pathParams: { refundId: "refundId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryRefunds2() {
  const payload = {};
  const getRequestOptions = queryRefunds(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/refunds/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  createRefund,
  getRefund,
  queryRefunds
});
