// @wix/headless-events/dist/react/core/EventList.js
import { jsx as _jsx } from "react/jsx-runtime";
import { useService, WixServices } from '@wix/services-manager-react';
import { createServicesMap } from '@wix/services-manager';
import { EventListService, EventListServiceDefinition, } from '../../services/event-list-service.js';
import { ALL_CATEGORIES, CATEGORIES_FILTER_KEY, ALL_EVENTS, STATUS_FILTER_KEY, UPCOMING_EVENTS, PAST_EVENTS, } from '../../constants.js';
/**
 * EventList Root core component that provides event list service context.
 *
 * @component
 */
export function Root(props) {
    const { children, eventListServiceConfig } = props;
    return (_jsx(WixServices, { servicesMap: createServicesMap().addService(EventListServiceDefinition, EventListService, eventListServiceConfig), children: children }));
}
/**
 * EventList Events core component that provides event list data.
 *
 * @component
 */
export function Events(props) {
    const eventListService = useService(EventListServiceDefinition);
    const isLoading = eventListService.isLoading.get();
    const events = eventListService.events.get();
    const hasEvents = !!events.length;
    return props.children({ events, hasEvents, isLoading });
}
/**
 * EventList EventRepeater core component that provides event list. Not rendered if there are no events.
 *
 * @component
 */
export function EventRepeater(props) {
    const eventListService = useService(EventListServiceDefinition);
    const events = eventListService.events.get();
    const hasEvents = !!events.length;
    if (!hasEvents) {
        return null;
    }
    return props.children({ events });
}
/**
 * EventList LoadMoreTrigger core component that provides load more trigger data.
 *
 * @component
 */
export function LoadMoreTrigger(props) {
    const eventListService = useService(EventListServiceDefinition);
    const isLoading = eventListService.isLoading.get();
    const isLoadingMore = eventListService.isLoadingMore.get();
    const hasMoreEvents = eventListService.hasMoreEvents.get();
    if (isLoading || !hasMoreEvents) {
        return null;
    }
    return props.children({
        isLoading: isLoadingMore,
        loadMoreEvents: eventListService.loadMoreEvents,
    });
}
/**
 * EventList Error core component that provides event list error. Not rendered if there is no error.
 *
 * @component
 */
export function Error(props) {
    const eventListService = useService(EventListServiceDefinition);
    const isLoading = eventListService.isLoading.get();
    const error = eventListService.error.get();
    if (isLoading || !error) {
        return null;
    }
    return props.children({ error });
}
/**
 * CategoryFilter core component that provides event list filters data.
 *
 * @component
 */
export function CategoryFilter(props) {
    const eventListService = useService(EventListServiceDefinition);
    const categories = eventListService.categories.get();
    const selectedCategoryId = eventListService.selectedCategoryId.get() || ALL_CATEGORIES;
    if (!categories.length) {
        return null;
    }
    const onChange = async (value) => {
        const categoryId = value['categoryId'];
        await eventListService.loadEvents({
            categoryId: categoryId === ALL_CATEGORIES ? null : categoryId,
        });
    };
    const { filterOptions, value } = buildCategoryFilterProps(categories, props.allCategoriesLabel, selectedCategoryId);
    return props.children({
        filterOptions,
        value,
        onChange,
    });
}
/**
 * StatusFilter core component that provides event list status filters data.
 *
 * @component
 */
export function StatusFilter(props) {
    const eventListService = useService(EventListServiceDefinition);
    const selectedStatusId = eventListService.selectedStatusId.get();
    const onChange = async (value) => {
        const statusId = value['statusId'];
        await eventListService.loadEvents({ statusId });
    };
    const { filterOptions, value } = buildStatusFilterProps(props.allEventsLabel, props.upcomingEventsLabel, props.pastEventsLabel, selectedStatusId);
    return props.children({
        filterOptions,
        value,
        onChange,
    });
}
const buildCategoryFilterProps = (categories, allCategoriesLabel, selectedCategoryId) => {
    const FILTER_BASE = {
        key: CATEGORIES_FILTER_KEY,
        label: '',
        type: 'single',
        displayType: 'text',
        fieldName: 'categoryId',
    };
    const filterOptions = [
        {
            ...FILTER_BASE,
            validValues: [
                ALL_CATEGORIES,
                ...categories.map((category) => category._id),
            ],
            valueFormatter: (value) => value === ALL_CATEGORIES
                ? allCategoriesLabel
                : categories.find((category) => category._id === value).name,
        },
    ];
    const value = {
        ...FILTER_BASE,
        categoryId: selectedCategoryId,
    };
    return { filterOptions, value };
};
const buildStatusFilterProps = (allEventsLabel, upcomingEventsLabel, pastEventsLabel, selectedStatusId) => {
    const FILTER_BASE = {
        key: STATUS_FILTER_KEY,
        label: '',
        type: 'single',
        displayType: 'text',
        fieldName: 'statusId',
    };
    const filterOptions = [
        {
            ...FILTER_BASE,
            validValues: [ALL_EVENTS, UPCOMING_EVENTS, PAST_EVENTS],
            valueFormatter: (value) => {
                switch (value) {
                    case ALL_EVENTS:
                        return allEventsLabel;
                    case UPCOMING_EVENTS:
                        return upcomingEventsLabel;
                    case PAST_EVENTS:
                        return pastEventsLabel;
                    default:
                        return upcomingEventsLabel;
                }
            },
        },
    ];
    const value = {
        ...FILTER_BASE,
        statusId: selectedStatusId,
    };
    return { filterOptions, value };
};
