// @wix/headless-events/dist/react/core/OccurrenceList.js
import { jsx as _jsx } from "react/jsx-runtime";
import { useService, WixServices } from '@wix/services-manager-react';
import { createServicesMap } from '@wix/services-manager';
import { OccurrenceListService, OccurrenceListServiceDefinition, } from '../../services/occurrence-list-service.js';
/**
 * OccurrenceList Root core component that provides occurrence list service context.
 *
 * @component
 */
export function Root(props) {
    const { children, occurrenceListServiceConfig } = props;
    return (_jsx(WixServices, { servicesMap: createServicesMap().addService(OccurrenceListServiceDefinition, OccurrenceListService, occurrenceListServiceConfig), children: children }));
}
/**
 * OccurrenceList Occurrences core component that provides occurrence list data.
 *
 * @component
 */
export function Occurrences(props) {
    const occurrenceListService = useService(OccurrenceListServiceDefinition);
    const occurrences = occurrenceListService.occurrences.get();
    const hasOccurrences = !!occurrences.length;
    return props.children({ occurrences, hasOccurrences });
}
/**
 * OccurrenceList OccurrenceRepeater core component that provides occurrence list. Not rendered if there are no occurrences.
 *
 * @component
 */
export function OccurrenceRepeater(props) {
    const occurrenceListService = useService(OccurrenceListServiceDefinition);
    const occurrences = occurrenceListService.occurrences.get();
    const hasOccurrences = !!occurrences.length;
    if (!hasOccurrences) {
        return null;
    }
    return props.children({ occurrences });
}
/**
 * OccurrenceList LoadMoreTrigger core component that provides load more trigger data.
 *
 * @component
 */
export function LoadMoreTrigger(props) {
    const occurrenceListService = useService(OccurrenceListServiceDefinition);
    const isLoading = occurrenceListService.isLoadingMore.get();
    const hasMoreOccurrences = occurrenceListService.hasMoreOccurrences.get();
    if (!hasMoreOccurrences) {
        return null;
    }
    return props.children({
        isLoading,
        loadMoreOccurrences: occurrenceListService.loadMoreOccurrences,
    });
}
/**
 * OccurrenceList Error core component that provides occurrence list error. Not rendered if there is no error.
 *
 * @component
 */
export function Error(props) {
    const occurrenceListService = useService(OccurrenceListServiceDefinition);
    const error = occurrenceListService.error.get();
    if (!error) {
        return null;
    }
    return props.children({ error });
}
