// @wix/ricos/dist/cjs/plugin-audio/modals/audio-embed.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var audio_embed_exports = {};
__export(audio_embed_exports, {
  AudioEmbed: () => AudioEmbed
});
module.exports = __toCommonJS(audio_embed_exports);
var import_design_system = require("@wix/design-system");
var import_react = __toESM(require("react"));
var import_fetchEmbedData = require("./../../wix-ricos-common/fetchEmbedData.js");
var import_ricos_content = require("./../../ricos-content/index.js");
var import_DevContext = require("./../../ricos-context/DevContext.js");
var import_ricos_context = require("./../../ricos-context/index.js");
var import_contains_url = require("./../../_shared/link/contains-url.js");
var import_request_service_provider = require("./../../_shared/request-service-provider/request-service-provider.js");
var import_wix_react_player = require("./../../wix-react-player/index.js");
var import_wix_rich_content_common = require("./../../wix-rich-content-common/index.js");
var import_data_hooks = require("../data-hooks.js");
var import_audio_embed_scss = __toESM(require("../statics/styles/audio-embed-scss.js"));
var import_types = require("../types.js");
var import_utils = require("./utils.js");
const AudioEmbed = ({
  componentData,
  closeModal,
  nodeId,
  logBi
}) => {
  const { t } = (0, import_react.useContext)(import_ricos_context.RicosContext);
  const { getPublicCommands } = (0, import_react.useContext)(import_ricos_context.EditorContext);
  const pluginsEvents = (0, import_react.useContext)(import_ricos_context.PluginsEventsContext);
  const { fetchOembedWithAuthFromDevContext } = (0, import_react.useContext)(import_DevContext.DevContext);
  const { makeRequest } = (0, import_request_service_provider.useRequestService)();
  const [url, setUrl] = (0, import_react.useState)(componentData?.audio?.src?.url || "");
  const [submittedInvalidUrl, setSubmittedInvalidUrl] = (0, import_react.useState)(false);
  const inputRef = (0, import_react.useRef)(null);
  const onEmbedClick = async () => {
    pluginsEvents.publishPluginPopoverClick?.({
      pluginId: import_types.AUDIO_TYPE,
      buttonName: import_wix_rich_content_common.MEDIA_POPOVERS_BUTTONS_NAMES_BI.embed
    });
    logBi({ dataHook: import_data_hooks.EMBED_TAB_SUBMIT_BUTTON });
    const { audio, name, coverImage, authorName, disableDownload, ...rest } = componentData;
    if (url && (!(0, import_contains_url.containsUrl)(url) || !hasValidFilePath(url))) {
      setSubmittedInvalidUrl(true);
      return;
    }
    if ((0, import_utils.isSoundcloudUrl)(url)) {
      await fetchAndEmbedAudio(url, rest);
    } else if (import_wix_react_player.ReactPlayer.canPlay(url)) {
      updateRicosDocument({ audio: { src: { url } } });
      closeModal();
    } else {
      await fetchAndEmbedAudio(url, rest);
    }
  };
  const fetchAndEmbedAudio = async (url2, data) => {
    try {
      const { html } = await fetchOembedWithAuthFromDevContext?.(url2) || await (0, import_fetchEmbedData.fetchOembedWithAuth)?.(url2, makeRequest) || {};
      if (html) {
        updateRicosDocument({ ...data, audio: { src: { url: url2 } }, html });
        closeModal();
      } else {
        setSubmittedInvalidUrl(true);
        inputRef.current?.focus();
      }
    } catch {
      setSubmittedInvalidUrl(true);
      inputRef.current?.focus();
    }
  };
  const updateRicosDocument = (audio) => queueMicrotask(
    () => nodeId ? getPublicCommands().setBlock(nodeId, import_types.AUDIO_TYPE, audio) : getPublicCommands().insertBlockWithBlankLines(import_types.AUDIO_TYPE, { ...audio, id: (0, import_ricos_content.generateId)() })
  );
  return /* @__PURE__ */ import_react.default.createElement(import_react.default.Fragment, null, /* @__PURE__ */ import_react.default.createElement(
    import_design_system.FormField,
    {
      label: t("AudioModal_Embed_Title"),
      maxLines: 2,
      status: submittedInvalidUrl ? "error" : void 0,
      statusMessage: submittedInvalidUrl ? t("SoundCloudUploadModal_Input_InvalidUrl") : void 0
    },
    /* @__PURE__ */ import_react.default.createElement(
      import_design_system.Input,
      {
        inputRef: (ref) => {
          inputRef.current = ref;
        },
        className: import_audio_embed_scss.default.input,
        type: "url",
        autocomplete: "off",
        dataHook: import_data_hooks.EMBED_TAB_URL_INPUT,
        value: url,
        textOverflow: "ellipsis",
        onChange: (e) => {
          setSubmittedInvalidUrl(false);
          setUrl(e.target.value);
          logBi({ dataHook: import_data_hooks.EMBED_TAB_URL_INPUT, value: e.target.value });
        },
        placeholder: t("EmbedURL_Placeholder"),
        onBlur: () => {
          if (url && !(0, import_contains_url.containsUrl)(url)) {
            setSubmittedInvalidUrl(true);
          }
        },
        onKeyDown: (e) => {
          if (e.key === "Enter") {
            onEmbedClick();
          }
        }
      }
    )
  ), /* @__PURE__ */ import_react.default.createElement(import_design_system.Button, { dataHook: import_data_hooks.EMBED_TAB_SUBMIT_BUTTON, onClick: onEmbedClick, size: "small" }, t("AudioModal_Embed_ButtonText")));
};
const hasValidFilePath = (url) => {
  const urlObj = new URL(url);
  return urlObj.pathname.length > 1;
};
