// @wix/auto_sdk_ecom_manual-tax-mappings/build/internal/es/index.mjs
// src/billing-v1-manual-tax-mapping-manual-tax-mappings.public.ts
import { renameKeysFromRESTResponseToSDKResponse as renameKeysFromRESTResponseToSDKResponse2 } from "@wix/sdk-runtime/rename-all-nested-keys";
import { transformRESTTimestampToSDKTimestamp as transformRESTTimestampToSDKTimestamp2 } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths as transformPaths3 } from "@wix/sdk-runtime/transformations/transform-paths";
import { EventDefinition } from "@wix/sdk-types";

// src/billing-v1-manual-tax-mapping-manual-tax-mappings.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/billing-v1-manual-tax-mapping-manual-tax-mappings.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixBillingV1ManualTaxMappingServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/manual-tax-mappings",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/manual-tax-mappings",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/billing/v1/manual-tax-mappings",
        destPath: "/v1/manual-tax-mappings"
      },
      {
        srcPath: "/billing/v1/bulk/manual-tax-mappings",
        destPath: "/v1/bulk/manual-tax-mappings"
      }
    ],
    _: [
      {
        srcPath: "/billing/v1/manual-tax-mappings",
        destPath: "/v1/manual-tax-mappings"
      },
      {
        srcPath: "/billing/v1/bulk/manual-tax-mappings",
        destPath: "/v1/bulk/manual-tax-mappings"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/billing/v1/manual-tax-mappings",
        destPath: "/v1/manual-tax-mappings"
      },
      {
        srcPath: "/billing/v1/bulk/manual-tax-mappings",
        destPath: "/v1/bulk/manual-tax-mappings"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_manual-tax-mappings";
function createManualTaxMapping(payload) {
  function __createManualTaxMapping({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "manualTaxMapping.createdDate" },
          { path: "manualTaxMapping.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.billing.v1.manual_tax_mapping",
      method: "POST",
      methodFqn: "wix.billing.v1.ManualTaxMappingService.CreateManualTaxMapping",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBillingV1ManualTaxMappingServiceUrl({
        protoPath: "/v1/manual-tax-mappings",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "manualTaxMapping.createdDate" },
            { path: "manualTaxMapping.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createManualTaxMapping;
}
function bulkCreateManualTaxMappings(payload) {
  function __bulkCreateManualTaxMappings({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "manualTaxMappings.createdDate" },
          { path: "manualTaxMappings.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.billing.v1.manual_tax_mapping",
      method: "POST",
      methodFqn: "wix.billing.v1.ManualTaxMappingService.BulkCreateManualTaxMappings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBillingV1ManualTaxMappingServiceUrl({
        protoPath: "/v1/bulk/manual-tax-mappings/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateManualTaxMappings;
}
function getManualTaxMapping(payload) {
  function __getManualTaxMapping({ host }) {
    const metadata = {
      entityFqdn: "wix.billing.v1.manual_tax_mapping",
      method: "GET",
      methodFqn: "wix.billing.v1.ManualTaxMappingService.GetManualTaxMapping",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBillingV1ManualTaxMappingServiceUrl({
        protoPath: "/v1/manual-tax-mappings/{manualTaxMappingId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "manualTaxMapping.createdDate" },
            { path: "manualTaxMapping.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getManualTaxMapping;
}
function updateManualTaxMapping(payload) {
  function __updateManualTaxMapping({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "manualTaxMapping.createdDate" },
          { path: "manualTaxMapping.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.billing.v1.manual_tax_mapping",
      method: "PATCH",
      methodFqn: "wix.billing.v1.ManualTaxMappingService.UpdateManualTaxMapping",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBillingV1ManualTaxMappingServiceUrl({
        protoPath: "/v1/manual-tax-mappings/{manualTaxMapping.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "manualTaxMapping.createdDate" },
            { path: "manualTaxMapping.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateManualTaxMapping;
}
function deleteManualTaxMapping(payload) {
  function __deleteManualTaxMapping({ host }) {
    const metadata = {
      entityFqdn: "wix.billing.v1.manual_tax_mapping",
      method: "DELETE",
      methodFqn: "wix.billing.v1.ManualTaxMappingService.DeleteManualTaxMapping",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBillingV1ManualTaxMappingServiceUrl({
        protoPath: "/v1/manual-tax-mappings/{manualTaxMappingId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteManualTaxMapping;
}
function queryManualTaxMappings(payload) {
  function __queryManualTaxMappings({ host }) {
    const metadata = {
      entityFqdn: "wix.billing.v1.manual_tax_mapping",
      method: "GET",
      methodFqn: "wix.billing.v1.ManualTaxMappingService.QueryManualTaxMappings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBillingV1ManualTaxMappingServiceUrl({
        protoPath: "/v1/manual-tax-mappings/query",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload, true),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "manualTaxMappings.createdDate" },
            { path: "manualTaxMappings.updatedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixBillingV1ManualTaxMappingServiceUrl({
            protoPath: "/v1/manual-tax-mappings/query",
            data: payload,
            host
          }),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryManualTaxMappings;
}
function updateExtendedFields(payload) {
  function __updateExtendedFields({ host }) {
    const metadata = {
      entityFqdn: "wix.billing.v1.manual_tax_mapping",
      method: "POST",
      methodFqn: "wix.billing.v1.ManualTaxMappingService.UpdateExtendedFields",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBillingV1ManualTaxMappingServiceUrl({
        protoPath: "/v1/manual-tax-mappings/{id}/update-extended-fields",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "manualTaxMapping.createdDate" },
            { path: "manualTaxMapping.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateExtendedFields;
}
function bulkAssignExemptOverrides(payload) {
  function __bulkAssignExemptOverrides({ host }) {
    const metadata = {
      entityFqdn: "wix.billing.v1.manual_tax_mapping",
      method: "POST",
      methodFqn: "wix.billing.v1.ManualTaxMappingService.BulkAssignExemptOverrides",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBillingV1ManualTaxMappingServiceUrl({
        protoPath: "/v1/manual-tax-mappings/bulk-assign-exempt-overrides",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkAssignExemptOverrides;
}
function bulkUnassignExemptOverrides(payload) {
  function __bulkUnassignExemptOverrides({ host }) {
    const metadata = {
      entityFqdn: "wix.billing.v1.manual_tax_mapping",
      method: "POST",
      methodFqn: "wix.billing.v1.ManualTaxMappingService.BulkUnassignExemptOverrides",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBillingV1ManualTaxMappingServiceUrl({
        protoPath: "/v1/manual-tax-mappings/bulk-unassign-exempt-overrides",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUnassignExemptOverrides;
}
function listManualTaxMappings(payload) {
  function __listManualTaxMappings({ host }) {
    const metadata = {
      entityFqdn: "wix.billing.v1.manual_tax_mapping",
      method: "GET",
      methodFqn: "wix.billing.v1.ManualTaxMappingService.ListManualTaxMappings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBillingV1ManualTaxMappingServiceUrl({
        protoPath: "/v1/manual-tax-mappings",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "manualTaxMappings.createdDate" },
            { path: "manualTaxMappings.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listManualTaxMappings;
}

// src/billing-v1-manual-tax-mapping-manual-tax-mappings.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var JurisdictionType = /* @__PURE__ */ ((JurisdictionType2) => {
  JurisdictionType2["UNDEFINED"] = "UNDEFINED";
  JurisdictionType2["COUNTRY"] = "COUNTRY";
  JurisdictionType2["STATE"] = "STATE";
  JurisdictionType2["COUNTY"] = "COUNTY";
  JurisdictionType2["CITY"] = "CITY";
  JurisdictionType2["SPECIAL"] = "SPECIAL";
  return JurisdictionType2;
})(JurisdictionType || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createManualTaxMapping2(manualTaxMapping) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    manualTaxMapping
  });
  const reqOpts = createManualTaxMapping(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.manualTaxMapping;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { manualTaxMapping: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["manualTaxMapping"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCreateManualTaxMappings2(manualTaxMappings, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    manualTaxMappings,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkCreateManualTaxMappings(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          manualTaxMappings: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["manualTaxMappings", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getManualTaxMapping2(manualTaxMappingId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    manualTaxMappingId
  });
  const reqOpts = getManualTaxMapping(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.manualTaxMapping;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { manualTaxMappingId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["manualTaxMappingId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateManualTaxMapping2(_id, manualTaxMapping) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    manualTaxMapping: { ...manualTaxMapping, id: _id }
  });
  const reqOpts = updateManualTaxMapping(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.manualTaxMapping;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { manualTaxMapping: "$[1]" },
        explicitPathsToArguments: { "manualTaxMapping.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "manualTaxMapping"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteManualTaxMapping2(manualTaxMappingId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    manualTaxMappingId
  });
  const reqOpts = deleteManualTaxMapping(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { manualTaxMappingId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["manualTaxMappingId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryManualTaxMappings2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryManualTaxMappings(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [])
      );
      return {
        items: transformedData?.manualTaxMappings,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function typedQueryManualTaxMappings(query) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ query });
  const reqOpts = queryManualTaxMappings(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["query"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateExtendedFields2(_id, namespace, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    id: _id,
    namespace,
    namespaceData: options?.namespaceData
  });
  const reqOpts = updateExtendedFields(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          id: "$[0]",
          namespace: "$[1]",
          namespaceData: "$[2].namespaceData"
        },
        singleArgumentUnchanged: false
      },
      ["_id", "namespace", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkAssignExemptOverrides2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    assignOverrides: options?.assignOverrides,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkAssignExemptOverrides(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          assignOverrides: "$[0].assignOverrides",
          returnEntity: "$[0].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUnassignExemptOverrides2(unassignOverrides, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    unassignOverrides,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkUnassignExemptOverrides(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          unassignOverrides: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["unassignOverrides", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listManualTaxMappings2() {
  const { httpClient, sideEffects } = arguments[0];
  const payload = renameKeysFromSDKRequestToRESTRequest({});
  const reqOpts = listManualTaxMappings(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {},
        singleArgumentUnchanged: false
      },
      []
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/billing-v1-manual-tax-mapping-manual-tax-mappings.public.ts
function createManualTaxMapping3(httpClient) {
  return (manualTaxMapping) => createManualTaxMapping2(
    manualTaxMapping,
    // @ts-ignore
    { httpClient }
  );
}
function bulkCreateManualTaxMappings3(httpClient) {
  return (manualTaxMappings, options) => bulkCreateManualTaxMappings2(
    manualTaxMappings,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function getManualTaxMapping3(httpClient) {
  return (manualTaxMappingId) => getManualTaxMapping2(
    manualTaxMappingId,
    // @ts-ignore
    { httpClient }
  );
}
function updateManualTaxMapping3(httpClient) {
  return (_id, manualTaxMapping) => updateManualTaxMapping2(
    _id,
    manualTaxMapping,
    // @ts-ignore
    { httpClient }
  );
}
function deleteManualTaxMapping3(httpClient) {
  return (manualTaxMappingId) => deleteManualTaxMapping2(
    manualTaxMappingId,
    // @ts-ignore
    { httpClient }
  );
}
function queryManualTaxMappings3(httpClient) {
  return () => queryManualTaxMappings2(
    // @ts-ignore
    { httpClient }
  );
}
function typedQueryManualTaxMappings2(httpClient) {
  return (query) => typedQueryManualTaxMappings(
    query,
    // @ts-ignore
    { httpClient }
  );
}
function updateExtendedFields3(httpClient) {
  return (_id, namespace, options) => updateExtendedFields2(
    _id,
    namespace,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkAssignExemptOverrides3(httpClient) {
  return (options) => bulkAssignExemptOverrides2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkUnassignExemptOverrides3(httpClient) {
  return (unassignOverrides, options) => bulkUnassignExemptOverrides2(
    unassignOverrides,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function listManualTaxMappings3(httpClient) {
  return () => listManualTaxMappings2(
    // @ts-ignore
    { httpClient }
  );
}
var onManualTaxMappingCreated = EventDefinition(
  "wix.billing.v1.manual_tax_mapping_created",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onManualTaxMappingDeleted = EventDefinition(
  "wix.billing.v1.manual_tax_mapping_deleted",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onManualTaxMappingUpdated = EventDefinition(
  "wix.billing.v1.manual_tax_mapping_updated",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();

// src/billing-v1-manual-tax-mapping-manual-tax-mappings.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
import { createEventModule } from "@wix/sdk-runtime/event-definition-modules";
import { createQueryOverloadRouter } from "@wix/sdk-runtime/query-method-router";
function customQueryManualTaxMappings(httpClient) {
  const router = createQueryOverloadRouter({
    builderQueryFunction: () => queryManualTaxMappings3(httpClient)(),
    typedQueryFunction: (query) => typedQueryManualTaxMappings2(httpClient)(query),
    hasOptionsParameter: false
  });
  function overloadedQuery(query) {
    return router(...arguments);
  }
  return overloadedQuery;
}
var createManualTaxMapping4 = /* @__PURE__ */ createRESTModule(createManualTaxMapping3);
var bulkCreateManualTaxMappings4 = /* @__PURE__ */ createRESTModule(bulkCreateManualTaxMappings3);
var getManualTaxMapping4 = /* @__PURE__ */ createRESTModule(getManualTaxMapping3);
var updateManualTaxMapping4 = /* @__PURE__ */ createRESTModule(updateManualTaxMapping3);
var deleteManualTaxMapping4 = /* @__PURE__ */ createRESTModule(deleteManualTaxMapping3);
var updateExtendedFields4 = /* @__PURE__ */ createRESTModule(updateExtendedFields3);
var bulkAssignExemptOverrides4 = /* @__PURE__ */ createRESTModule(bulkAssignExemptOverrides3);
var bulkUnassignExemptOverrides4 = /* @__PURE__ */ createRESTModule(bulkUnassignExemptOverrides3);
var listManualTaxMappings4 = /* @__PURE__ */ createRESTModule(listManualTaxMappings3);
var queryManualTaxMappings4 = /* @__PURE__ */ createRESTModule(customQueryManualTaxMappings);
var onManualTaxMappingCreated2 = createEventModule(
  onManualTaxMappingCreated
);
var onManualTaxMappingDeleted2 = createEventModule(
  onManualTaxMappingDeleted
);
var onManualTaxMappingUpdated2 = createEventModule(
  onManualTaxMappingUpdated
);
export {
  JurisdictionType,
  SortOrder,
  WebhookIdentityType,
  bulkAssignExemptOverrides4 as bulkAssignExemptOverrides,
  bulkCreateManualTaxMappings4 as bulkCreateManualTaxMappings,
  bulkUnassignExemptOverrides4 as bulkUnassignExemptOverrides,
  createManualTaxMapping4 as createManualTaxMapping,
  deleteManualTaxMapping4 as deleteManualTaxMapping,
  getManualTaxMapping4 as getManualTaxMapping,
  listManualTaxMappings4 as listManualTaxMappings,
  onManualTaxMappingCreated2 as onManualTaxMappingCreated,
  onManualTaxMappingDeleted2 as onManualTaxMappingDeleted,
  onManualTaxMappingUpdated2 as onManualTaxMappingUpdated,
  queryManualTaxMappings4 as queryManualTaxMappings,
  updateExtendedFields4 as updateExtendedFields,
  updateManualTaxMapping4 as updateManualTaxMapping
};
