// @wix/auto_sdk_payments_dispute-evidence-documents/build/es/index.typings.mjs
// src/payments-dispute-evidence-documents-v1-dispute-evidence-document-dispute-evidence-documents.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/payments-dispute-evidence-documents-v1-dispute-evidence-document-dispute-evidence-documents.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(opts) {
  const domainToMappings = {};
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_dispute-evidence-documents";
function createDisputeEvidenceDocument(payload) {
  function __createDisputeEvidenceDocument({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "disputeEvidenceDocument.createdDate" },
          { path: "disputeEvidenceDocument.updatedDate" },
          { path: "disputeEvidenceDocument.submittedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.CreateDisputeEvidenceDocument",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents",
          data: serializedData,
          host
        }
      ),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "disputeEvidenceDocument.createdDate" },
            { path: "disputeEvidenceDocument.updatedDate" },
            { path: "disputeEvidenceDocument.submittedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createDisputeEvidenceDocument;
}
function generateDisputeEvidenceDocumentUploadUrl(payload) {
  function __generateDisputeEvidenceDocumentUploadUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.GenerateDisputeEvidenceDocumentUploadUrl",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}/generate-upload-url",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __generateDisputeEvidenceDocumentUploadUrl;
}
function attachDisputeEvidenceDocumentFile(payload) {
  function __attachDisputeEvidenceDocumentFile({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.AttachDisputeEvidenceDocumentFile",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}/attach-file",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "disputeEvidenceDocument.createdDate" },
            { path: "disputeEvidenceDocument.updatedDate" },
            { path: "disputeEvidenceDocument.submittedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __attachDisputeEvidenceDocumentFile;
}
function getDisputeEvidenceDocument(payload) {
  function __getDisputeEvidenceDocument({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "GET",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.GetDisputeEvidenceDocument",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}",
          data: payload,
          host
        }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "disputeEvidenceDocument.createdDate" },
            { path: "disputeEvidenceDocument.updatedDate" },
            { path: "disputeEvidenceDocument.submittedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getDisputeEvidenceDocument;
}
function generateDisputeEvidenceDocumentDownloadUrl(payload) {
  function __generateDisputeEvidenceDocumentDownloadUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.GenerateDisputeEvidenceDocumentDownloadUrl",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}/generate-download-url",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "urlExpirationDate" }]
        }
      ])
    };
    return metadata;
  }
  return __generateDisputeEvidenceDocumentDownloadUrl;
}
function markDisputeEvidenceDocumentSubmitted(payload) {
  function __markDisputeEvidenceDocumentSubmitted({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.MarkDisputeEvidenceDocumentSubmitted",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}/mark-submitted",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __markDisputeEvidenceDocumentSubmitted;
}
function deleteDisputeEvidenceDocument(payload) {
  function __deleteDisputeEvidenceDocument({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "DELETE",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.DeleteDisputeEvidenceDocument",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}",
          data: payload,
          host
        }
      ),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteDisputeEvidenceDocument;
}
function queryDisputeEvidenceDocuments(payload) {
  function __queryDisputeEvidenceDocuments({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.QueryDisputeEvidenceDocuments",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents/query",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "disputeEvidenceDocuments.createdDate" },
            { path: "disputeEvidenceDocuments.updatedDate" },
            { path: "disputeEvidenceDocuments.submittedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
            {
              protoPath: "/v1/dispute-evidence-documents/query",
              data: payload,
              host
            }
          ),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryDisputeEvidenceDocuments;
}
function bulkUpdateDisputeEvidenceDocumentTags(payload) {
  function __bulkUpdateDisputeEvidenceDocumentTags({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.BulkUpdateDisputeEvidenceDocumentTags",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/bulk/dispute-evidence-documents/update-tags",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateDisputeEvidenceDocumentTags;
}
function bulkUpdateDisputeEvidenceDocumentTagsByFilter(payload) {
  function __bulkUpdateDisputeEvidenceDocumentTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.BulkUpdateDisputeEvidenceDocumentTagsByFilter",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/bulk/dispute-evidence-documents/update-tags-by-filter",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateDisputeEvidenceDocumentTagsByFilter;
}

// src/payments-dispute-evidence-documents-v1-dispute-evidence-document-dispute-evidence-documents.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createDisputeEvidenceDocument2(disputeEvidenceDocument) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    disputeEvidenceDocument
  });
  const reqOpts = createDisputeEvidenceDocument(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.disputeEvidenceDocument;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { disputeEvidenceDocument: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["disputeEvidenceDocument"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function generateDisputeEvidenceDocumentUploadUrl2(disputeEvidenceDocumentId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    disputeEvidenceDocumentId,
    filename: options?.filename
  });
  const reqOpts = generateDisputeEvidenceDocumentUploadUrl(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          disputeEvidenceDocumentId: "$[0]",
          filename: "$[1].filename"
        },
        singleArgumentUnchanged: false
      },
      ["disputeEvidenceDocumentId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function attachDisputeEvidenceDocumentFile2(disputeEvidenceDocumentId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    disputeEvidenceDocumentId
  });
  const reqOpts = attachDisputeEvidenceDocumentFile(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { disputeEvidenceDocumentId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["disputeEvidenceDocumentId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getDisputeEvidenceDocument2(disputeEvidenceDocumentId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    disputeEvidenceDocumentId
  });
  const reqOpts = getDisputeEvidenceDocument(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.disputeEvidenceDocument;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { disputeEvidenceDocumentId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["disputeEvidenceDocumentId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function generateDisputeEvidenceDocumentDownloadUrl2(disputeEvidenceDocumentId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    disputeEvidenceDocumentId
  });
  const reqOpts = generateDisputeEvidenceDocumentDownloadUrl(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { disputeEvidenceDocumentId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["disputeEvidenceDocumentId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function markDisputeEvidenceDocumentSubmitted2(disputeEvidenceDocumentId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    disputeEvidenceDocumentId
  });
  const reqOpts = markDisputeEvidenceDocumentSubmitted(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { disputeEvidenceDocumentId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["disputeEvidenceDocumentId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteDisputeEvidenceDocument2(disputeEvidenceDocumentId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    disputeEvidenceDocumentId
  });
  const reqOpts = deleteDisputeEvidenceDocument(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { disputeEvidenceDocumentId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["disputeEvidenceDocumentId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryDisputeEvidenceDocuments2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryDisputeEvidenceDocuments(
        payload
      );
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [])
      );
      return {
        items: transformedData?.disputeEvidenceDocuments,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function bulkUpdateDisputeEvidenceDocumentTags2(ids, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    ids,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateDisputeEvidenceDocumentTags(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          ids: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["ids", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateDisputeEvidenceDocumentTagsByFilter2(filter, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    filter,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateDisputeEvidenceDocumentTagsByFilter(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          filter: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["filter", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
export {
  SortOrder,
  WebhookIdentityType,
  attachDisputeEvidenceDocumentFile2 as attachDisputeEvidenceDocumentFile,
  bulkUpdateDisputeEvidenceDocumentTags2 as bulkUpdateDisputeEvidenceDocumentTags,
  bulkUpdateDisputeEvidenceDocumentTagsByFilter2 as bulkUpdateDisputeEvidenceDocumentTagsByFilter,
  createDisputeEvidenceDocument2 as createDisputeEvidenceDocument,
  deleteDisputeEvidenceDocument2 as deleteDisputeEvidenceDocument,
  generateDisputeEvidenceDocumentDownloadUrl2 as generateDisputeEvidenceDocumentDownloadUrl,
  generateDisputeEvidenceDocumentUploadUrl2 as generateDisputeEvidenceDocumentUploadUrl,
  getDisputeEvidenceDocument2 as getDisputeEvidenceDocument,
  markDisputeEvidenceDocumentSubmitted2 as markDisputeEvidenceDocumentSubmitted,
  queryDisputeEvidenceDocuments2 as queryDisputeEvidenceDocuments
};
