// @wix/auto_sdk_restaurants_restaurant-locations/build/internal/es/index.mjs
// src/restaurants-restaurantlocations-v3-restaurant-location-restaurant-locations.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/restaurants-restaurantlocations-v3-restaurant-location-restaurant-locations.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressRestaurantsRestaurantlocationsV3RestaurantLocationsServiceUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      },
      {
        srcPath: "/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      },
      {
        srcPath: "/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      }
    ],
    _base_domain_: [
      {
        srcPath: "/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      }
    ],
    "users._base_domain_": [
      {
        srcPath: "/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurant-locations",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_restaurant-locations";
function getRestaurantLocation(payload) {
  function __getRestaurantLocation({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.restaurantlocations.v3.restaurant_location",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.restaurantlocations.v3.RestaurantLocationsService.GetRestaurantLocation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsRestaurantlocationsV3RestaurantLocationsServiceUrl(
        {
          protoPath: "/v3/restaurant-locations/{restaurantLocationId}",
          data: payload,
          host
        }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "restaurantLocation.location.address.geocode.latitude" },
            { path: "restaurantLocation.location.address.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getRestaurantLocation;
}
function listRestaurantLocations(payload) {
  function __listRestaurantLocations({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.restaurantlocations.v3.restaurant_location",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.restaurantlocations.v3.RestaurantLocationsService.ListRestaurantLocations",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsRestaurantlocationsV3RestaurantLocationsServiceUrl(
        { protoPath: "/v3/restaurant-locations", data: payload, host }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "restaurantLocations.location.address.geocode.latitude" },
            {
              path: "restaurantLocations.location.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listRestaurantLocations;
}

// src/restaurants-restaurantlocations-v3-restaurant-location-restaurant-locations.universal.ts
import { transformRESTAddressToSDKAddress } from "@wix/sdk-runtime/transformations/address";
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var SubdivisionType = /* @__PURE__ */ ((SubdivisionType2) => {
  SubdivisionType2["UNKNOWN_SUBDIVISION_TYPE"] = "UNKNOWN_SUBDIVISION_TYPE";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_1"] = "ADMINISTRATIVE_AREA_LEVEL_1";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_2"] = "ADMINISTRATIVE_AREA_LEVEL_2";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_3"] = "ADMINISTRATIVE_AREA_LEVEL_3";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_4"] = "ADMINISTRATIVE_AREA_LEVEL_4";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_5"] = "ADMINISTRATIVE_AREA_LEVEL_5";
  SubdivisionType2["COUNTRY"] = "COUNTRY";
  return SubdivisionType2;
})(SubdivisionType || {});
var DayOfWeek = /* @__PURE__ */ ((DayOfWeek2) => {
  DayOfWeek2["MON"] = "MON";
  DayOfWeek2["TUE"] = "TUE";
  DayOfWeek2["WED"] = "WED";
  DayOfWeek2["THU"] = "THU";
  DayOfWeek2["FRI"] = "FRI";
  DayOfWeek2["SAT"] = "SAT";
  DayOfWeek2["SUN"] = "SUN";
  return DayOfWeek2;
})(DayOfWeek || {});
async function getRestaurantLocation2(restaurantLocationId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    restaurantLocationId
  });
  const reqOpts = getRestaurantLocation(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTAddressToSDKAddress,
          paths: [{ path: "restaurantLocation.location.address" }]
        }
      ])
    )?.restaurantLocation;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { restaurantLocationId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["restaurantLocationId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listRestaurantLocations2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    locationIds: options?.locationIds,
    includeArchived: options?.includeArchived,
    default: options?.default
  });
  const reqOpts = listRestaurantLocations(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTAddressToSDKAddress,
          paths: [{ path: "restaurantLocations.location.address" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          locationIds: "$[0].locationIds",
          includeArchived: "$[0].includeArchived",
          default: "$[0].default"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/restaurants-restaurantlocations-v3-restaurant-location-restaurant-locations.public.ts
function getRestaurantLocation3(httpClient) {
  return (restaurantLocationId) => getRestaurantLocation2(
    restaurantLocationId,
    // @ts-ignore
    { httpClient }
  );
}
function listRestaurantLocations3(httpClient) {
  return (options) => listRestaurantLocations2(
    options,
    // @ts-ignore
    { httpClient }
  );
}

// src/restaurants-restaurantlocations-v3-restaurant-location-restaurant-locations.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
var getRestaurantLocation4 = /* @__PURE__ */ createRESTModule(getRestaurantLocation3);
var listRestaurantLocations4 = /* @__PURE__ */ createRESTModule(listRestaurantLocations3);
export {
  DayOfWeek,
  SubdivisionType,
  getRestaurantLocation4 as getRestaurantLocation,
  listRestaurantLocations4 as listRestaurantLocations
};
