// @wix/auto_sdk_restaurants_restaurant-locations/build/internal/es/meta.mjs
// src/restaurants-restaurantlocations-v3-restaurant-location-restaurant-locations.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressRestaurantsRestaurantlocationsV3RestaurantLocationsServiceUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      },
      {
        srcPath: "/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      },
      {
        srcPath: "/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      }
    ],
    _base_domain_: [
      {
        srcPath: "/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      }
    ],
    "users._base_domain_": [
      {
        srcPath: "/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants/v3/restaurant-locations",
        destPath: "/v3/restaurant-locations"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurant-locations",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_restaurant-locations";
function getRestaurantLocation(payload) {
  function __getRestaurantLocation({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.restaurantlocations.v3.restaurant_location",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.restaurantlocations.v3.RestaurantLocationsService.GetRestaurantLocation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsRestaurantlocationsV3RestaurantLocationsServiceUrl(
        {
          protoPath: "/v3/restaurant-locations/{restaurantLocationId}",
          data: payload,
          host
        }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "restaurantLocation.location.address.geocode.latitude" },
            { path: "restaurantLocation.location.address.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getRestaurantLocation;
}
function listRestaurantLocations(payload) {
  function __listRestaurantLocations({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.restaurantlocations.v3.restaurant_location",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.restaurantlocations.v3.RestaurantLocationsService.ListRestaurantLocations",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsRestaurantlocationsV3RestaurantLocationsServiceUrl(
        { protoPath: "/v3/restaurant-locations", data: payload, host }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "restaurantLocations.location.address.geocode.latitude" },
            {
              path: "restaurantLocations.location.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listRestaurantLocations;
}

// src/restaurants-restaurantlocations-v3-restaurant-location-restaurant-locations.types.ts
var SubdivisionType = /* @__PURE__ */ ((SubdivisionType2) => {
  SubdivisionType2["UNKNOWN_SUBDIVISION_TYPE"] = "UNKNOWN_SUBDIVISION_TYPE";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_1"] = "ADMINISTRATIVE_AREA_LEVEL_1";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_2"] = "ADMINISTRATIVE_AREA_LEVEL_2";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_3"] = "ADMINISTRATIVE_AREA_LEVEL_3";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_4"] = "ADMINISTRATIVE_AREA_LEVEL_4";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_5"] = "ADMINISTRATIVE_AREA_LEVEL_5";
  SubdivisionType2["COUNTRY"] = "COUNTRY";
  return SubdivisionType2;
})(SubdivisionType || {});
var DayOfWeek = /* @__PURE__ */ ((DayOfWeek2) => {
  DayOfWeek2["MON"] = "MON";
  DayOfWeek2["TUE"] = "TUE";
  DayOfWeek2["WED"] = "WED";
  DayOfWeek2["THU"] = "THU";
  DayOfWeek2["FRI"] = "FRI";
  DayOfWeek2["SAT"] = "SAT";
  DayOfWeek2["SUN"] = "SUN";
  return DayOfWeek2;
})(DayOfWeek || {});

// src/restaurants-restaurantlocations-v3-restaurant-location-restaurant-locations.meta.ts
function getRestaurantLocation2() {
  const payload = { restaurantLocationId: ":restaurantLocationId" };
  const getRequestOptions = getRestaurantLocation(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/restaurant-locations/{restaurantLocationId}",
    pathParams: { restaurantLocationId: "restaurantLocationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listRestaurantLocations2() {
  const payload = {};
  const getRequestOptions = listRestaurantLocations(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/restaurant-locations",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  DayOfWeek as DayOfWeekOriginal,
  SubdivisionType as SubdivisionTypeOriginal,
  getRestaurantLocation2 as getRestaurantLocation,
  listRestaurantLocations2 as listRestaurantLocations
};
