// @wix/auto_sdk_payments_payins-configurations/build/internal/es/index.mjs
// src/payments-payins-configurations-v1-payins-configuration-payins-configurations.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/payments-payins-configurations-v1-payins-configuration-payins-configurations.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(opts) {
  const domainToMappings = {};
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_payins-configurations";
function createPayinsConfiguration(payload) {
  function __createPayinsConfiguration({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "payinsConfiguration.createdDate" },
          { path: "payinsConfiguration.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.payments.payins_configurations.v1.payins_configuration",
      method: "POST",
      methodFqn: "wix.payments.payins_configurations.v1.PayinsConfigurationService.CreatePayinsConfiguration",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(
        { protoPath: "/v1/payins-configurations", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "payinsConfiguration.createdDate" },
            { path: "payinsConfiguration.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createPayinsConfiguration;
}
function getPayinsConfiguration(payload) {
  function __getPayinsConfiguration({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.payins_configurations.v1.payins_configuration",
      method: "GET",
      methodFqn: "wix.payments.payins_configurations.v1.PayinsConfigurationService.GetPayinsConfiguration",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(
        {
          protoPath: "/v1/payins-configurations/{payinsConfigurationId}",
          data: payload,
          host
        }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "payinsConfiguration.createdDate" },
            { path: "payinsConfiguration.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getPayinsConfiguration;
}
function updatePayinsConfiguration(payload) {
  function __updatePayinsConfiguration({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "payinsConfiguration.createdDate" },
          { path: "payinsConfiguration.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.payments.payins_configurations.v1.payins_configuration",
      method: "PATCH",
      methodFqn: "wix.payments.payins_configurations.v1.PayinsConfigurationService.UpdatePayinsConfiguration",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(
        {
          protoPath: "/v1/payins-configurations/{payinsConfiguration.id}",
          data: serializedData,
          host
        }
      ),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "payinsConfiguration.createdDate" },
            { path: "payinsConfiguration.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updatePayinsConfiguration;
}
function queryPayinsConfigurations(payload) {
  function __queryPayinsConfigurations({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.payins_configurations.v1.payins_configuration",
      method: "POST",
      methodFqn: "wix.payments.payins_configurations.v1.PayinsConfigurationService.QueryPayinsConfigurations",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(
        { protoPath: "/v1/payins-configurations/query", data: payload, host }
      ),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "payinsConfigurations.createdDate" },
            { path: "payinsConfigurations.updatedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(
            {
              protoPath: "/v1/payins-configurations/query",
              data: payload,
              host
            }
          ),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryPayinsConfigurations;
}
function bulkUpdatePayinsConfigurationTags(payload) {
  function __bulkUpdatePayinsConfigurationTags({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.payins_configurations.v1.payins_configuration",
      method: "POST",
      methodFqn: "wix.payments.payins_configurations.v1.PayinsConfigurationService.BulkUpdatePayinsConfigurationTags",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(
        {
          protoPath: "/v1/bulk/payins-configurations/update-tags",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdatePayinsConfigurationTags;
}
function bulkUpdatePayinsConfigurationTagsByFilter(payload) {
  function __bulkUpdatePayinsConfigurationTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.payins_configurations.v1.payins_configuration",
      method: "POST",
      methodFqn: "wix.payments.payins_configurations.v1.PayinsConfigurationService.BulkUpdatePayinsConfigurationTagsByFilter",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(
        {
          protoPath: "/v1/bulk/payins-configurations/update-tags-by-filter",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdatePayinsConfigurationTagsByFilter;
}

// src/payments-payins-configurations-v1-payins-configuration-payins-configurations.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createPayinsConfiguration2(payinsConfiguration) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    payinsConfiguration
  });
  const reqOpts = createPayinsConfiguration(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.payinsConfiguration;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { payinsConfiguration: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["payinsConfiguration"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getPayinsConfiguration2(payinsConfigurationId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    payinsConfigurationId
  });
  const reqOpts = getPayinsConfiguration(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.payinsConfiguration;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { payinsConfigurationId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["payinsConfigurationId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updatePayinsConfiguration2(_id, payinsConfiguration) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    payinsConfiguration: { ...payinsConfiguration, id: _id }
  });
  const reqOpts = updatePayinsConfiguration(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.payinsConfiguration;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { payinsConfiguration: "$[1]" },
        explicitPathsToArguments: { "payinsConfiguration.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "payinsConfiguration"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryPayinsConfigurations2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryPayinsConfigurations(
        payload
      );
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [])
      );
      return {
        items: transformedData?.payinsConfigurations,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function bulkUpdatePayinsConfigurationTags2(ids, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    ids,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdatePayinsConfigurationTags(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          ids: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["ids", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdatePayinsConfigurationTagsByFilter2(filter, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    filter,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdatePayinsConfigurationTagsByFilter(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          filter: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["filter", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/payments-payins-configurations-v1-payins-configuration-payins-configurations.public.ts
function createPayinsConfiguration3(httpClient) {
  return (payinsConfiguration) => createPayinsConfiguration2(
    payinsConfiguration,
    // @ts-ignore
    { httpClient }
  );
}
function getPayinsConfiguration3(httpClient) {
  return (payinsConfigurationId) => getPayinsConfiguration2(
    payinsConfigurationId,
    // @ts-ignore
    { httpClient }
  );
}
function updatePayinsConfiguration3(httpClient) {
  return (_id, payinsConfiguration) => updatePayinsConfiguration2(
    _id,
    payinsConfiguration,
    // @ts-ignore
    { httpClient }
  );
}
function queryPayinsConfigurations3(httpClient) {
  return () => queryPayinsConfigurations2(
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdatePayinsConfigurationTags3(httpClient) {
  return (ids, options) => bulkUpdatePayinsConfigurationTags2(
    ids,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdatePayinsConfigurationTagsByFilter3(httpClient) {
  return (filter, options) => bulkUpdatePayinsConfigurationTagsByFilter2(
    filter,
    options,
    // @ts-ignore
    { httpClient }
  );
}

// src/payments-payins-configurations-v1-payins-configuration-payins-configurations.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
var createPayinsConfiguration4 = /* @__PURE__ */ createRESTModule(createPayinsConfiguration3);
var getPayinsConfiguration4 = /* @__PURE__ */ createRESTModule(getPayinsConfiguration3);
var updatePayinsConfiguration4 = /* @__PURE__ */ createRESTModule(updatePayinsConfiguration3);
var queryPayinsConfigurations4 = /* @__PURE__ */ createRESTModule(queryPayinsConfigurations3);
var bulkUpdatePayinsConfigurationTags4 = /* @__PURE__ */ createRESTModule(bulkUpdatePayinsConfigurationTags3);
var bulkUpdatePayinsConfigurationTagsByFilter4 = /* @__PURE__ */ createRESTModule(
  bulkUpdatePayinsConfigurationTagsByFilter3
);
export {
  SortOrder,
  WebhookIdentityType,
  bulkUpdatePayinsConfigurationTags4 as bulkUpdatePayinsConfigurationTags,
  bulkUpdatePayinsConfigurationTagsByFilter4 as bulkUpdatePayinsConfigurationTagsByFilter,
  createPayinsConfiguration4 as createPayinsConfiguration,
  getPayinsConfiguration4 as getPayinsConfiguration,
  queryPayinsConfigurations4 as queryPayinsConfigurations,
  updatePayinsConfiguration4 as updatePayinsConfiguration
};
