// @wix/auto_sdk_payments_payins-configurations/build/internal/es/meta.mjs
// src/payments-payins-configurations-v1-payins-configuration-payins-configurations.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(opts) {
  const domainToMappings = {};
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_payins-configurations";
function createPayinsConfiguration(payload) {
  function __createPayinsConfiguration({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "payinsConfiguration.createdDate" },
          { path: "payinsConfiguration.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.payments.payins_configurations.v1.payins_configuration",
      method: "POST",
      methodFqn: "wix.payments.payins_configurations.v1.PayinsConfigurationService.CreatePayinsConfiguration",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(
        { protoPath: "/v1/payins-configurations", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "payinsConfiguration.createdDate" },
            { path: "payinsConfiguration.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createPayinsConfiguration;
}
function getPayinsConfiguration(payload) {
  function __getPayinsConfiguration({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.payins_configurations.v1.payins_configuration",
      method: "GET",
      methodFqn: "wix.payments.payins_configurations.v1.PayinsConfigurationService.GetPayinsConfiguration",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(
        {
          protoPath: "/v1/payins-configurations/{payinsConfigurationId}",
          data: payload,
          host
        }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "payinsConfiguration.createdDate" },
            { path: "payinsConfiguration.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getPayinsConfiguration;
}
function updatePayinsConfiguration(payload) {
  function __updatePayinsConfiguration({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "payinsConfiguration.createdDate" },
          { path: "payinsConfiguration.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.payments.payins_configurations.v1.payins_configuration",
      method: "PATCH",
      methodFqn: "wix.payments.payins_configurations.v1.PayinsConfigurationService.UpdatePayinsConfiguration",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(
        {
          protoPath: "/v1/payins-configurations/{payinsConfiguration.id}",
          data: serializedData,
          host
        }
      ),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "payinsConfiguration.createdDate" },
            { path: "payinsConfiguration.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updatePayinsConfiguration;
}
function queryPayinsConfigurations(payload) {
  function __queryPayinsConfigurations({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.payins_configurations.v1.payins_configuration",
      method: "POST",
      methodFqn: "wix.payments.payins_configurations.v1.PayinsConfigurationService.QueryPayinsConfigurations",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(
        { protoPath: "/v1/payins-configurations/query", data: payload, host }
      ),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "payinsConfigurations.createdDate" },
            { path: "payinsConfigurations.updatedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(
            {
              protoPath: "/v1/payins-configurations/query",
              data: payload,
              host
            }
          ),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryPayinsConfigurations;
}
function bulkUpdatePayinsConfigurationTags(payload) {
  function __bulkUpdatePayinsConfigurationTags({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.payins_configurations.v1.payins_configuration",
      method: "POST",
      methodFqn: "wix.payments.payins_configurations.v1.PayinsConfigurationService.BulkUpdatePayinsConfigurationTags",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(
        {
          protoPath: "/v1/bulk/payins-configurations/update-tags",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdatePayinsConfigurationTags;
}
function bulkUpdatePayinsConfigurationTagsByFilter(payload) {
  function __bulkUpdatePayinsConfigurationTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.payins_configurations.v1.payins_configuration",
      method: "POST",
      methodFqn: "wix.payments.payins_configurations.v1.PayinsConfigurationService.BulkUpdatePayinsConfigurationTagsByFilter",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(
        {
          protoPath: "/v1/bulk/payins-configurations/update-tags-by-filter",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdatePayinsConfigurationTagsByFilter;
}

// src/payments-payins-configurations-v1-payins-configuration-payins-configurations.meta.ts
function createPayinsConfiguration2() {
  const payload = {};
  const getRequestOptions = createPayinsConfiguration(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/payins-configurations",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getPayinsConfiguration2() {
  const payload = { payinsConfigurationId: ":payinsConfigurationId" };
  const getRequestOptions = getPayinsConfiguration(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/payins-configurations/{payinsConfigurationId}",
    pathParams: { payinsConfigurationId: "payinsConfigurationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updatePayinsConfiguration2() {
  const payload = {
    payinsConfiguration: { id: ":payinsConfigurationId" }
  };
  const getRequestOptions = updatePayinsConfiguration(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/payins-configurations/{payinsConfiguration.id}",
    pathParams: { payinsConfigurationId: "payinsConfigurationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryPayinsConfigurations2() {
  const payload = {};
  const getRequestOptions = queryPayinsConfigurations(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/payins-configurations/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdatePayinsConfigurationTags2() {
  const payload = {};
  const getRequestOptions = bulkUpdatePayinsConfigurationTags(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/payins-configurations/update-tags",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdatePayinsConfigurationTagsByFilter2() {
  const payload = {};
  const getRequestOptions = bulkUpdatePayinsConfigurationTagsByFilter(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/payins-configurations/update-tags-by-filter",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  bulkUpdatePayinsConfigurationTags2 as bulkUpdatePayinsConfigurationTags,
  bulkUpdatePayinsConfigurationTagsByFilter2 as bulkUpdatePayinsConfigurationTagsByFilter,
  createPayinsConfiguration2 as createPayinsConfiguration,
  getPayinsConfiguration2 as getPayinsConfiguration,
  queryPayinsConfigurations2 as queryPayinsConfigurations,
  updatePayinsConfiguration2 as updatePayinsConfiguration
};
