// @wix/auto_sdk_benefit-programs_balances/build/internal/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  ChangeBalanceRequestType: () => ChangeBalanceRequestType,
  IdentityType: () => IdentityType,
  LimitRule: () => LimitRule,
  PoolStatus: () => PoolStatus,
  SortOrder: () => SortOrder,
  TransactionReversibility: () => TransactionReversibility,
  Type: () => Type,
  WebhookIdentityType: () => WebhookIdentityType,
  changeBalance: () => changeBalance2,
  getBalance: () => getBalance2,
  listBalances: () => listBalances2,
  queryBalances: () => queryBalances2,
  revertBalanceChange: () => revertBalanceChange2
});
module.exports = __toCommonJS(index_typings_exports);

// src/benefit-programs-v1-balance-balances.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/benefit-programs-v1-balance-balances.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixBenefitProgramsV1BalanceBalanceServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/pool-accounts",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/benefit-programs-pool-accounts",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pool-accounts",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pool-accounts",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/benefit-programs-pool-accounts",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/benefit-programs-pool-accounts",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pool-accounts",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pool-accounts",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_benefit-programs_balances";
function getBalance(payload) {
  function __getBalance({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.balance",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.balance.BalanceService.GetBalance",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1BalanceBalanceServiceUrl({
        protoPath: "/v1/balances/{poolId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "balance.createdDate" },
            { path: "balance.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getBalance;
}
function listBalances(payload) {
  function __listBalances({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.balance",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.balance.BalanceService.ListBalances",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1BalanceBalanceServiceUrl({
        protoPath: "/v1/balances",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "balances.createdDate" },
            { path: "balances.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listBalances;
}
function queryBalances(payload) {
  function __queryBalances({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.balance",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.balance.BalanceService.QueryBalances",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1BalanceBalanceServiceUrl({
        protoPath: "/v1/balances/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "balances.createdDate" },
            { path: "balances.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryBalances;
}
function changeBalance(payload) {
  function __changeBalance({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "transactionDetails.effectiveDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.balance",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.balance.BalanceService.ChangeBalance",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1BalanceBalanceServiceUrl({
        protoPath: "/v1/balances/{poolId}/change",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "balance.createdDate" },
            { path: "balance.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __changeBalance;
}
function revertBalanceChange(payload) {
  function __revertBalanceChange({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.balance",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.balance.BalanceService.RevertBalanceChange",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1BalanceBalanceServiceUrl({
        protoPath: "/v1/balances/changes/{transactionId}/revert",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __revertBalanceChange;
}

// src/benefit-programs-v1-balance-balances.universal.ts
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var IdentityType = /* @__PURE__ */ ((IdentityType2) => {
  IdentityType2["UNKNOWN"] = "UNKNOWN";
  IdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  IdentityType2["MEMBER"] = "MEMBER";
  IdentityType2["WIX_USER"] = "WIX_USER";
  return IdentityType2;
})(IdentityType || {});
var PoolStatus = /* @__PURE__ */ ((PoolStatus2) => {
  PoolStatus2["UNDEFINED"] = "UNDEFINED";
  PoolStatus2["ACTIVE"] = "ACTIVE";
  PoolStatus2["PAUSED"] = "PAUSED";
  PoolStatus2["ENDED"] = "ENDED";
  PoolStatus2["PROVISIONING"] = "PROVISIONING";
  PoolStatus2["RENEWING"] = "RENEWING";
  return PoolStatus2;
})(PoolStatus || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var ChangeBalanceRequestType = /* @__PURE__ */ ((ChangeBalanceRequestType2) => {
  ChangeBalanceRequestType2["UNKNOWN_OPERATION"] = "UNKNOWN_OPERATION";
  ChangeBalanceRequestType2["ADJUST"] = "ADJUST";
  ChangeBalanceRequestType2["SET"] = "SET";
  return ChangeBalanceRequestType2;
})(ChangeBalanceRequestType || {});
var LimitRule = /* @__PURE__ */ ((LimitRule2) => {
  LimitRule2["UNKNOWN_LIMIT_RULE"] = "UNKNOWN_LIMIT_RULE";
  LimitRule2["FAIL"] = "FAIL";
  LimitRule2["SOFT_ENFORCE"] = "SOFT_ENFORCE";
  LimitRule2["HARD_ENFORCE"] = "HARD_ENFORCE";
  return LimitRule2;
})(LimitRule || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNKNOWN_OPERATION"] = "UNKNOWN_OPERATION";
  return Type2;
})(Type || {});
var TransactionReversibility = /* @__PURE__ */ ((TransactionReversibility2) => {
  TransactionReversibility2["TRANSACTION_IS_REVERSIBLE"] = "TRANSACTION_IS_REVERSIBLE";
  TransactionReversibility2["TRANSACTION_ALREADY_REVERSED"] = "TRANSACTION_ALREADY_REVERSED";
  return TransactionReversibility2;
})(TransactionReversibility || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function getBalance2(poolId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ poolId });
  const reqOpts = getBalance(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.balance;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { poolId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["poolId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listBalances2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    poolIds: options?.poolIds,
    cursorPaging: options?.cursorPaging
  });
  const reqOpts = listBalances(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          poolIds: "$[0].poolIds",
          cursorPaging: "$[0].cursorPaging"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryBalances2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryBalances(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [])
      );
      return {
        items: transformedData?.balances,
        pagingMetadata: transformedData?.metadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function changeBalance2(poolId, idempotencyKey, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    poolId,
    idempotencyKey,
    instructingParty: options?.instructingParty,
    type: options?.type,
    adjustOptions: options?.adjustOptions,
    setOptions: options?.setOptions,
    transactionDetails: options?.transactionDetails
  });
  const reqOpts = changeBalance(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          poolId: "$[0]",
          idempotencyKey: "$[1]",
          instructingParty: "$[2].instructingParty",
          type: "$[2].type",
          adjustOptions: "$[2].adjustOptions",
          setOptions: "$[2].setOptions",
          transactionDetails: "$[2].transactionDetails"
        },
        singleArgumentUnchanged: false
      },
      ["poolId", "idempotencyKey", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function revertBalanceChange2(transactionId, idempotencyKey, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    transactionId,
    idempotencyKey,
    instructingParty: options?.instructingParty
  });
  const reqOpts = revertBalanceChange(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          transactionId: "$[0]",
          idempotencyKey: "$[1]",
          instructingParty: "$[2].instructingParty"
        },
        singleArgumentUnchanged: false
      },
      ["transactionId", "idempotencyKey", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ChangeBalanceRequestType,
  IdentityType,
  LimitRule,
  PoolStatus,
  SortOrder,
  TransactionReversibility,
  Type,
  WebhookIdentityType,
  changeBalance,
  getBalance,
  listBalances,
  queryBalances,
  revertBalanceChange
});
