// @wix/auto_sdk_benefit-programs_balances/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  changeBalance: () => changeBalance2,
  getBalance: () => getBalance2,
  listBalances: () => listBalances2,
  queryBalances: () => queryBalances2,
  revertBalanceChange: () => revertBalanceChange2
});
module.exports = __toCommonJS(meta_exports);

// src/benefit-programs-v1-balance-balances.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixBenefitProgramsV1BalanceBalanceServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/pool-accounts",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/benefit-programs-pool-accounts",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pool-accounts",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pool-accounts",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/benefit-programs-pool-accounts",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/benefit-programs-pool-accounts",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pool-accounts",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pool-accounts",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_benefit-programs_balances";
function getBalance(payload) {
  function __getBalance({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.balance",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.balance.BalanceService.GetBalance",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1BalanceBalanceServiceUrl({
        protoPath: "/v1/balances/{poolId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "balance.createdDate" },
            { path: "balance.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getBalance;
}
function listBalances(payload) {
  function __listBalances({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.balance",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.balance.BalanceService.ListBalances",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1BalanceBalanceServiceUrl({
        protoPath: "/v1/balances",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "balances.createdDate" },
            { path: "balances.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listBalances;
}
function queryBalances(payload) {
  function __queryBalances({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.balance",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.balance.BalanceService.QueryBalances",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1BalanceBalanceServiceUrl({
        protoPath: "/v1/balances/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "balances.createdDate" },
            { path: "balances.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryBalances;
}
function changeBalance(payload) {
  function __changeBalance({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "transactionDetails.effectiveDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.balance",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.balance.BalanceService.ChangeBalance",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1BalanceBalanceServiceUrl({
        protoPath: "/v1/balances/{poolId}/change",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "balance.createdDate" },
            { path: "balance.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __changeBalance;
}
function revertBalanceChange(payload) {
  function __revertBalanceChange({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.balance",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.balance.BalanceService.RevertBalanceChange",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1BalanceBalanceServiceUrl({
        protoPath: "/v1/balances/changes/{transactionId}/revert",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __revertBalanceChange;
}

// src/benefit-programs-v1-balance-balances.meta.ts
function getBalance2() {
  const payload = { poolId: ":poolId" };
  const getRequestOptions = getBalance(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/balances/{poolId}",
    pathParams: { poolId: "poolId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listBalances2() {
  const payload = {};
  const getRequestOptions = listBalances(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/balances",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryBalances2() {
  const payload = {};
  const getRequestOptions = queryBalances(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/balances/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function changeBalance2() {
  const payload = { poolId: ":poolId" };
  const getRequestOptions = changeBalance(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/balances/{poolId}/change",
    pathParams: { poolId: "poolId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function revertBalanceChange2() {
  const payload = { transactionId: ":transactionId" };
  const getRequestOptions = revertBalanceChange(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/balances/changes/{transactionId}/revert",
    pathParams: { transactionId: "transactionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  changeBalance,
  getBalance,
  listBalances,
  queryBalances,
  revertBalanceChange
});
