// @wix/auto_sdk_pricing-plans_plans-v-3/build/internal/es/meta.mjs
// src/pricing-plans-v3-plan-plans-v-3.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKFloatToRESTFloat } from "@wix/sdk-runtime/transformations/float";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixPricingPlansPlansV3PlanServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/pricing-plans/v3",
        destPath: "/v3"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/pricing-plans/v3",
        destPath: "/v3"
      }
    ],
    _: [
      {
        srcPath: "/_api/pricing-plans/v3",
        destPath: "/v3"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/pricing-plans/v3",
        destPath: "/v3"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/pricing-plans",
        destPath: "/v3"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_pricing-plans_plans-v-3";
function createPlan(payload) {
  function __createPlan({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.pricing_plans.v3.plan",
      method: "POST",
      methodFqn: "wix.pricing_plans.plans.v3.PlanService.CreatePlan",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixPricingPlansPlansV3PlanServiceUrl({
        protoPath: "/v3/plans",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __createPlan;
}
function getPlan(payload) {
  function __getPlan({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v3.plan",
      method: "GET",
      methodFqn: "wix.pricing_plans.plans.v3.PlanService.GetPlan",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixPricingPlansPlansV3PlanServiceUrl({
        protoPath: "/v3/plans/{planId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __getPlan;
}
function updatePlan(payload) {
  function __updatePlan({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.pricing_plans.v3.plan",
      method: "PATCH",
      methodFqn: "wix.pricing_plans.plans.v3.PlanService.UpdatePlan",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixPricingPlansPlansV3PlanServiceUrl({
        protoPath: "/v3/plans/{plan.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __updatePlan;
}
function bulkUpdatePlans(payload) {
  function __bulkUpdatePlans({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "plans.fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "plans.plan.createdDate" },
          { path: "plans.plan.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.pricing_plans.v3.plan",
      method: "POST",
      methodFqn: "wix.pricing_plans.plans.v3.PlanService.BulkUpdatePlans",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixPricingPlansPlansV3PlanServiceUrl({
        protoPath: "/v3/bulk/plans/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdatePlans;
}
function deletePlan(payload) {
  function __deletePlan({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v3.plan",
      method: "DELETE",
      methodFqn: "wix.pricing_plans.plans.v3.PlanService.DeletePlan",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixPricingPlansPlansV3PlanServiceUrl({
        protoPath: "/v3/plans/{planId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deletePlan;
}
function queryPlans(payload) {
  function __queryPlans({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v3.plan",
      method: "GET",
      methodFqn: "wix.pricing_plans.plans.v3.PlanService.QueryPlans",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixPricingPlansPlansV3PlanServiceUrl({
        protoPath: "/v3/plans/query",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload, true),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "plans.createdDate" },
            { path: "plans.updatedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixPricingPlansPlansV3PlanServiceUrl({
            protoPath: "/v3/plans/query",
            data: payload,
            host
          }),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryPlans;
}
function searchPlans(payload) {
  function __searchPlans({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "search.aggregations.range.buckets.from" },
          { path: "search.aggregations.range.buckets.to" },
          {
            path: "search.aggregations.nested.nestedAggregations.range.buckets.from"
          },
          {
            path: "search.aggregations.nested.nestedAggregations.range.buckets.to"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.pricing_plans.v3.plan",
      method: "GET",
      methodFqn: "wix.pricing_plans.plans.v3.PlanService.SearchPlans",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixPricingPlansPlansV3PlanServiceUrl({
        protoPath: "/v3/plans/search",
        data: serializedData,
        host
      }),
      params: toURLSearchParams(serializedData, true),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "plans.createdDate" },
            { path: "plans.updatedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixPricingPlansPlansV3PlanServiceUrl({
            protoPath: "/v3/plans/search",
            data: serializedData,
            host
          }),
          data: serializedData
        }
      ]
    };
    return metadata;
  }
  return __searchPlans;
}
function countPlans(payload) {
  function __countPlans({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v3.plan",
      method: "POST",
      methodFqn: "wix.pricing_plans.plans.v3.PlanService.CountPlans",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixPricingPlansPlansV3PlanServiceUrl({
        protoPath: "/v3/plans/count",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __countPlans;
}
function changePrice(payload) {
  function __changePrice({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v3.plan",
      method: "POST",
      methodFqn: "wix.pricing_plans.plans.v3.PlanService.ChangePrice",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixPricingPlansPlansV3PlanServiceUrl({
        protoPath: "/v3/plans/{planId}/change-price",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __changePrice;
}

// src/pricing-plans-v3-plan-plans-v-3.types.ts
var PriceType = /* @__PURE__ */ ((PriceType2) => {
  PriceType2["UNKNOWN_PRICE_TYPE"] = "UNKNOWN_PRICE_TYPE";
  PriceType2["FIXED_AMOUNT"] = "FIXED_AMOUNT";
  return PriceType2;
})(PriceType || {});
var AppliedAt = /* @__PURE__ */ ((AppliedAt2) => {
  AppliedAt2["UNKNOWN_APPLIED_AT"] = "UNKNOWN_APPLIED_AT";
  AppliedAt2["FIRST_PAYMENT"] = "FIRST_PAYMENT";
  return AppliedAt2;
})(AppliedAt || {});
var Period = /* @__PURE__ */ ((Period2) => {
  Period2["UNKNOWN_PERIOD"] = "UNKNOWN_PERIOD";
  Period2["DAY"] = "DAY";
  Period2["WEEK"] = "WEEK";
  Period2["MONTH"] = "MONTH";
  Period2["YEAR"] = "YEAR";
  return Period2;
})(Period || {});
var StartType = /* @__PURE__ */ ((StartType2) => {
  StartType2["UNKNOWN_START_TYPE"] = "UNKNOWN_START_TYPE";
  StartType2["ON_PURCHASE"] = "ON_PURCHASE";
  StartType2["CUSTOM"] = "CUSTOM";
  return StartType2;
})(StartType || {});
var EndType = /* @__PURE__ */ ((EndType2) => {
  EndType2["UNKNOWN_END_TYPE"] = "UNKNOWN_END_TYPE";
  EndType2["UNTIL_CANCELLED"] = "UNTIL_CANCELLED";
  EndType2["CYCLES_COMPLETED"] = "CYCLES_COMPLETED";
  return EndType2;
})(EndType || {});
var Visibility = /* @__PURE__ */ ((Visibility2) => {
  Visibility2["UNKNOWN_VISIBILITY"] = "UNKNOWN_VISIBILITY";
  Visibility2["PUBLIC"] = "PUBLIC";
  Visibility2["PRIVATE"] = "PRIVATE";
  return Visibility2;
})(Visibility || {});
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["UNKNOWN_STATUS"] = "UNKNOWN_STATUS";
  Status2["ACTIVE"] = "ACTIVE";
  return Status2;
})(Status || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  Type2["PLAN"] = "PLAN";
  Type2["ADD_ON"] = "ADD_ON";
  return Type2;
})(Type || {});
var TaxableAddress = /* @__PURE__ */ ((TaxableAddress2) => {
  TaxableAddress2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  TaxableAddress2["BILLING"] = "BILLING";
  TaxableAddress2["BUSINESS"] = "BUSINESS";
  return TaxableAddress2;
})(TaxableAddress || {});
var ValidationStrategy = /* @__PURE__ */ ((ValidationStrategy2) => {
  ValidationStrategy2["UNKNOWN_VALIDATION_STRATEGY"] = "UNKNOWN_VALIDATION_STRATEGY";
  ValidationStrategy2["ON_BEHALF"] = "ON_BEHALF";
  ValidationStrategy2["PREVIEW"] = "PREVIEW";
  return ValidationStrategy2;
})(ValidationStrategy || {});
var RulesType = /* @__PURE__ */ ((RulesType2) => {
  RulesType2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  RulesType2["AVAILABLE_IMMEDIATELY"] = "AVAILABLE_IMMEDIATELY";
  RulesType2["FIXED_DATES"] = "FIXED_DATES";
  RulesType2["RECURRENCE"] = "RECURRENCE";
  return RulesType2;
})(RulesType || {});
var RecurrenceType = /* @__PURE__ */ ((RecurrenceType2) => {
  RecurrenceType2["UNKNOWN_RECURRENCE_TYPE"] = "UNKNOWN_RECURRENCE_TYPE";
  RecurrenceType2["WEEKLY"] = "WEEKLY";
  RecurrenceType2["MONTHLY"] = "MONTHLY";
  return RecurrenceType2;
})(RecurrenceType || {});
var V1DayOfWeek = /* @__PURE__ */ ((V1DayOfWeek2) => {
  V1DayOfWeek2["UNKNOWN_DAY_OF_WEEK"] = "UNKNOWN_DAY_OF_WEEK";
  V1DayOfWeek2["SUNDAY"] = "SUNDAY";
  V1DayOfWeek2["MONDAY"] = "MONDAY";
  V1DayOfWeek2["TUESDAY"] = "TUESDAY";
  V1DayOfWeek2["WEDNESDAY"] = "WEDNESDAY";
  V1DayOfWeek2["THURSDAY"] = "THURSDAY";
  V1DayOfWeek2["FRIDAY"] = "FRIDAY";
  V1DayOfWeek2["SATURDAY"] = "SATURDAY";
  return V1DayOfWeek2;
})(V1DayOfWeek || {});
var PlacementType = /* @__PURE__ */ ((PlacementType2) => {
  PlacementType2["BEFORE"] = "BEFORE";
  PlacementType2["AFTER"] = "AFTER";
  PlacementType2["REPLACE"] = "REPLACE";
  return PlacementType2;
})(PlacementType || {});
var DayOfWeek = /* @__PURE__ */ ((DayOfWeek2) => {
  DayOfWeek2["MONDAY"] = "MONDAY";
  DayOfWeek2["TUESDAY"] = "TUESDAY";
  DayOfWeek2["WEDNESDAY"] = "WEDNESDAY";
  DayOfWeek2["THURSDAY"] = "THURSDAY";
  DayOfWeek2["FRIDAY"] = "FRIDAY";
  DayOfWeek2["SATURDAY"] = "SATURDAY";
  DayOfWeek2["SUNDAY"] = "SUNDAY";
  return DayOfWeek2;
})(DayOfWeek || {});
var ResolutionMethod = /* @__PURE__ */ ((ResolutionMethod2) => {
  ResolutionMethod2["QUERY_PARAM"] = "QUERY_PARAM";
  ResolutionMethod2["SUBDOMAIN"] = "SUBDOMAIN";
  ResolutionMethod2["SUBDIRECTORY"] = "SUBDIRECTORY";
  return ResolutionMethod2;
})(ResolutionMethod || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
var Cascade = /* @__PURE__ */ ((Cascade2) => {
  Cascade2["NEW_SUBSCRIPTIONS"] = "NEW_SUBSCRIPTIONS";
  Cascade2["ALL_SUBSCRIPTIONS"] = "ALL_SUBSCRIPTIONS";
  return Cascade2;
})(Cascade || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var SortType = /* @__PURE__ */ ((SortType2) => {
  SortType2["COUNT"] = "COUNT";
  SortType2["VALUE"] = "VALUE";
  return SortType2;
})(SortType || {});
var SortDirection = /* @__PURE__ */ ((SortDirection2) => {
  SortDirection2["DESC"] = "DESC";
  SortDirection2["ASC"] = "ASC";
  return SortDirection2;
})(SortDirection || {});
var MissingValues = /* @__PURE__ */ ((MissingValues2) => {
  MissingValues2["EXCLUDE"] = "EXCLUDE";
  MissingValues2["INCLUDE"] = "INCLUDE";
  return MissingValues2;
})(MissingValues || {});
var ScalarType = /* @__PURE__ */ ((ScalarType2) => {
  ScalarType2["UNKNOWN_SCALAR_TYPE"] = "UNKNOWN_SCALAR_TYPE";
  ScalarType2["COUNT_DISTINCT"] = "COUNT_DISTINCT";
  ScalarType2["MIN"] = "MIN";
  ScalarType2["MAX"] = "MAX";
  return ScalarType2;
})(ScalarType || {});
var NestedAggregationType = /* @__PURE__ */ ((NestedAggregationType2) => {
  NestedAggregationType2["UNKNOWN_AGGREGATION_TYPE"] = "UNKNOWN_AGGREGATION_TYPE";
  NestedAggregationType2["VALUE"] = "VALUE";
  NestedAggregationType2["RANGE"] = "RANGE";
  NestedAggregationType2["SCALAR"] = "SCALAR";
  NestedAggregationType2["DATE_HISTOGRAM"] = "DATE_HISTOGRAM";
  return NestedAggregationType2;
})(NestedAggregationType || {});
var Interval = /* @__PURE__ */ ((Interval2) => {
  Interval2["UNKNOWN_INTERVAL"] = "UNKNOWN_INTERVAL";
  Interval2["YEAR"] = "YEAR";
  Interval2["MONTH"] = "MONTH";
  Interval2["WEEK"] = "WEEK";
  Interval2["DAY"] = "DAY";
  Interval2["HOUR"] = "HOUR";
  Interval2["MINUTE"] = "MINUTE";
  Interval2["SECOND"] = "SECOND";
  return Interval2;
})(Interval || {});
var AggregationType = /* @__PURE__ */ ((AggregationType2) => {
  AggregationType2["UNKNOWN_AGGREGATION_TYPE"] = "UNKNOWN_AGGREGATION_TYPE";
  AggregationType2["VALUE"] = "VALUE";
  AggregationType2["RANGE"] = "RANGE";
  AggregationType2["SCALAR"] = "SCALAR";
  AggregationType2["DATE_HISTOGRAM"] = "DATE_HISTOGRAM";
  AggregationType2["NESTED"] = "NESTED";
  return AggregationType2;
})(AggregationType || {});
var Mode = /* @__PURE__ */ ((Mode2) => {
  Mode2["OR"] = "OR";
  Mode2["AND"] = "AND";
  return Mode2;
})(Mode || {});

// src/pricing-plans-v3-plan-plans-v-3.meta.ts
function createPlan2() {
  const payload = {};
  const getRequestOptions = createPlan(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/plans",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getPlan2() {
  const payload = { planId: ":planId" };
  const getRequestOptions = getPlan(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/plans/{planId}",
    pathParams: { planId: "planId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updatePlan2() {
  const payload = { plan: { id: ":planId" } };
  const getRequestOptions = updatePlan(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v3/plans/{plan.id}",
    pathParams: { planId: "planId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdatePlans2() {
  const payload = {};
  const getRequestOptions = bulkUpdatePlans(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/bulk/plans/update",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deletePlan2() {
  const payload = { planId: ":planId" };
  const getRequestOptions = deletePlan(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v3/plans/{planId}",
    pathParams: { planId: "planId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryPlans2() {
  const payload = {};
  const getRequestOptions = queryPlans(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/plans/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function searchPlans2() {
  const payload = {};
  const getRequestOptions = searchPlans(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/plans/search",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function countPlans2() {
  const payload = {};
  const getRequestOptions = countPlans(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/plans/count",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function changePrice2() {
  const payload = { planId: ":planId" };
  const getRequestOptions = changePrice(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/plans/{planId}/change-price",
    pathParams: { planId: "planId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  AggregationType as AggregationTypeOriginal,
  AppliedAt as AppliedAtOriginal,
  Cascade as CascadeOriginal,
  DayOfWeek as DayOfWeekOriginal,
  EndType as EndTypeOriginal,
  Interval as IntervalOriginal,
  MissingValues as MissingValuesOriginal,
  Mode as ModeOriginal,
  NestedAggregationType as NestedAggregationTypeOriginal,
  Period as PeriodOriginal,
  PlacementType as PlacementTypeOriginal,
  PriceType as PriceTypeOriginal,
  RecurrenceType as RecurrenceTypeOriginal,
  ResolutionMethod as ResolutionMethodOriginal,
  RulesType as RulesTypeOriginal,
  ScalarType as ScalarTypeOriginal,
  SortDirection as SortDirectionOriginal,
  SortOrder as SortOrderOriginal,
  SortType as SortTypeOriginal,
  StartType as StartTypeOriginal,
  Status as StatusOriginal,
  TaxableAddress as TaxableAddressOriginal,
  Type as TypeOriginal,
  V1DayOfWeek as V1DayOfWeekOriginal,
  ValidationStrategy as ValidationStrategyOriginal,
  Visibility as VisibilityOriginal,
  WebhookIdentityType as WebhookIdentityTypeOriginal,
  bulkUpdatePlans2 as bulkUpdatePlans,
  changePrice2 as changePrice,
  countPlans2 as countPlans,
  createPlan2 as createPlan,
  deletePlan2 as deletePlan,
  getPlan2 as getPlan,
  queryPlans2 as queryPlans,
  searchPlans2 as searchPlans,
  updatePlan2 as updatePlan
};
