// @wix/auto_sdk_online-programs_quizzes/build/es/index.mjs
// src/achievements-quizzes-v1-quiz-quizzes.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/achievements-quizzes-v1-quiz-quizzes.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKFloatToRESTFloat } from "@wix/sdk-runtime/transformations/float";
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixAchievementsQuizzesV1QuizServiceUrl(opts) {
  const domainToMappings = {};
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_online-programs_quizzes";
function createQuiz(payload) {
  function __createQuiz({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "quiz.createdDate" }, { path: "quiz.updatedDate" }]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "quiz.fields.score" },
          { path: "quiz.fields.numeric.rightAnswer" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.achievements.quizzes.v1.quiz",
      method: "POST",
      methodFqn: "wix.achievements.quizzes.v1.QuizService.CreateQuiz",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAchievementsQuizzesV1QuizServiceUrl({
        protoPath: "/v1/quizzes",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "quiz.createdDate" }, { path: "quiz.updatedDate" }]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "quiz.fields.score" },
            { path: "quiz.fields.numeric.rightAnswer" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createQuiz;
}
function cloneQuiz(payload) {
  function __cloneQuiz({ host }) {
    const metadata = {
      entityFqdn: "wix.achievements.quizzes.v1.quiz",
      method: "POST",
      methodFqn: "wix.achievements.quizzes.v1.QuizService.CloneQuiz",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAchievementsQuizzesV1QuizServiceUrl({
        protoPath: "/v1/quizzes/{quizId}/clone",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "quiz.createdDate" }, { path: "quiz.updatedDate" }]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "quiz.fields.score" },
            { path: "quiz.fields.numeric.rightAnswer" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __cloneQuiz;
}
function bulkCreateQuiz(payload) {
  function __bulkCreateQuiz({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "quizzes.createdDate" },
          { path: "quizzes.updatedDate" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "quizzes.fields.score" },
          { path: "quizzes.fields.numeric.rightAnswer" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.achievements.quizzes.v1.quiz",
      method: "POST",
      methodFqn: "wix.achievements.quizzes.v1.QuizService.BulkCreateQuiz",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAchievementsQuizzesV1QuizServiceUrl({
        protoPath: "/v1/bulk/quizzes/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.fields.score" },
            { path: "results.item.fields.numeric.rightAnswer" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateQuiz;
}
function bulkCloneQuiz(payload) {
  function __bulkCloneQuiz({ host }) {
    const metadata = {
      entityFqdn: "wix.achievements.quizzes.v1.quiz",
      method: "POST",
      methodFqn: "wix.achievements.quizzes.v1.QuizService.BulkCloneQuiz",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAchievementsQuizzesV1QuizServiceUrl({
        protoPath: "/v1/bulk/quizzes/clone",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.fields.score" },
            { path: "results.item.fields.numeric.rightAnswer" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCloneQuiz;
}
function getQuiz(payload) {
  function __getQuiz({ host }) {
    const metadata = {
      entityFqdn: "wix.achievements.quizzes.v1.quiz",
      method: "GET",
      methodFqn: "wix.achievements.quizzes.v1.QuizService.GetQuiz",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAchievementsQuizzesV1QuizServiceUrl({
        protoPath: "/v1/quizzes/{quizId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "quiz.createdDate" }, { path: "quiz.updatedDate" }]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "quiz.fields.score" },
            { path: "quiz.fields.numeric.rightAnswer" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getQuiz;
}
function deleteQuiz(payload) {
  function __deleteQuiz({ host }) {
    const metadata = {
      entityFqdn: "wix.achievements.quizzes.v1.quiz",
      method: "DELETE",
      methodFqn: "wix.achievements.quizzes.v1.QuizService.DeleteQuiz",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAchievementsQuizzesV1QuizServiceUrl({
        protoPath: "/v1/quizzes/{quizId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "quiz.createdDate" }, { path: "quiz.updatedDate" }]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "quiz.fields.score" },
            { path: "quiz.fields.numeric.rightAnswer" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __deleteQuiz;
}

// src/achievements-quizzes-v1-quiz-quizzes.universal.ts
var FileFormat = /* @__PURE__ */ ((FileFormat2) => {
  FileFormat2["IMAGE"] = "IMAGE";
  FileFormat2["VIDEO"] = "VIDEO";
  FileFormat2["AUDIO"] = "AUDIO";
  FileFormat2["DOCUMENT"] = "DOCUMENT";
  FileFormat2["ARCHIVE"] = "ARCHIVE";
  return FileFormat2;
})(FileFormat || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createQuiz2(quiz) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ quiz });
  const reqOpts = createQuiz(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.quiz;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { quiz: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["quiz"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function cloneQuiz2(quizId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ quizId });
  const reqOpts = cloneQuiz(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { quizId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["quizId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCreateQuiz2(quizzes, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    quizzes,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkCreateQuiz(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          quizzes: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["quizzes", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCloneQuiz2(quizIds, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    quizIds,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkCloneQuiz(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          quizIds: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["quizIds", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getQuiz2(quizId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ quizId });
  const reqOpts = getQuiz(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.quiz;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { quizId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["quizId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteQuiz2(quizId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ quizId });
  const reqOpts = deleteQuiz(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { quizId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["quizId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/achievements-quizzes-v1-quiz-quizzes.public.ts
function createQuiz3(httpClient) {
  return (quiz) => createQuiz2(
    quiz,
    // @ts-ignore
    { httpClient }
  );
}
function cloneQuiz3(httpClient) {
  return (quizId) => cloneQuiz2(
    quizId,
    // @ts-ignore
    { httpClient }
  );
}
function bulkCreateQuiz3(httpClient) {
  return (quizzes, options) => bulkCreateQuiz2(
    quizzes,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkCloneQuiz3(httpClient) {
  return (quizIds, options) => bulkCloneQuiz2(
    quizIds,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function getQuiz3(httpClient) {
  return (quizId) => getQuiz2(
    quizId,
    // @ts-ignore
    { httpClient }
  );
}
function deleteQuiz3(httpClient) {
  return (quizId) => deleteQuiz2(
    quizId,
    // @ts-ignore
    { httpClient }
  );
}

// src/achievements-quizzes-v1-quiz-quizzes.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
var createQuiz4 = /* @__PURE__ */ createRESTModule(createQuiz3);
var cloneQuiz4 = /* @__PURE__ */ createRESTModule(cloneQuiz3);
var bulkCreateQuiz4 = /* @__PURE__ */ createRESTModule(bulkCreateQuiz3);
var bulkCloneQuiz4 = /* @__PURE__ */ createRESTModule(bulkCloneQuiz3);
var getQuiz4 = /* @__PURE__ */ createRESTModule(getQuiz3);
var deleteQuiz4 = /* @__PURE__ */ createRESTModule(deleteQuiz3);
export {
  FileFormat,
  WebhookIdentityType,
  bulkCloneQuiz4 as bulkCloneQuiz,
  bulkCreateQuiz4 as bulkCreateQuiz,
  cloneQuiz4 as cloneQuiz,
  createQuiz4 as createQuiz,
  deleteQuiz4 as deleteQuiz,
  getQuiz4 as getQuiz
};
