// @wix/auto_sdk_online-programs_quizzes/build/es/meta.mjs
// src/achievements-quizzes-v1-quiz-quizzes.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKFloatToRESTFloat } from "@wix/sdk-runtime/transformations/float";
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixAchievementsQuizzesV1QuizServiceUrl(opts) {
  const domainToMappings = {};
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_online-programs_quizzes";
function createQuiz(payload) {
  function __createQuiz({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "quiz.createdDate" }, { path: "quiz.updatedDate" }]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "quiz.fields.score" },
          { path: "quiz.fields.numeric.rightAnswer" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.achievements.quizzes.v1.quiz",
      method: "POST",
      methodFqn: "wix.achievements.quizzes.v1.QuizService.CreateQuiz",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAchievementsQuizzesV1QuizServiceUrl({
        protoPath: "/v1/quizzes",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "quiz.createdDate" }, { path: "quiz.updatedDate" }]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "quiz.fields.score" },
            { path: "quiz.fields.numeric.rightAnswer" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createQuiz;
}
function cloneQuiz(payload) {
  function __cloneQuiz({ host }) {
    const metadata = {
      entityFqdn: "wix.achievements.quizzes.v1.quiz",
      method: "POST",
      methodFqn: "wix.achievements.quizzes.v1.QuizService.CloneQuiz",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAchievementsQuizzesV1QuizServiceUrl({
        protoPath: "/v1/quizzes/{quizId}/clone",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "quiz.createdDate" }, { path: "quiz.updatedDate" }]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "quiz.fields.score" },
            { path: "quiz.fields.numeric.rightAnswer" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __cloneQuiz;
}
function bulkCreateQuiz(payload) {
  function __bulkCreateQuiz({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "quizzes.createdDate" },
          { path: "quizzes.updatedDate" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "quizzes.fields.score" },
          { path: "quizzes.fields.numeric.rightAnswer" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.achievements.quizzes.v1.quiz",
      method: "POST",
      methodFqn: "wix.achievements.quizzes.v1.QuizService.BulkCreateQuiz",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAchievementsQuizzesV1QuizServiceUrl({
        protoPath: "/v1/bulk/quizzes/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.fields.score" },
            { path: "results.item.fields.numeric.rightAnswer" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateQuiz;
}
function bulkCloneQuiz(payload) {
  function __bulkCloneQuiz({ host }) {
    const metadata = {
      entityFqdn: "wix.achievements.quizzes.v1.quiz",
      method: "POST",
      methodFqn: "wix.achievements.quizzes.v1.QuizService.BulkCloneQuiz",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAchievementsQuizzesV1QuizServiceUrl({
        protoPath: "/v1/bulk/quizzes/clone",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.fields.score" },
            { path: "results.item.fields.numeric.rightAnswer" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCloneQuiz;
}
function getQuiz(payload) {
  function __getQuiz({ host }) {
    const metadata = {
      entityFqdn: "wix.achievements.quizzes.v1.quiz",
      method: "GET",
      methodFqn: "wix.achievements.quizzes.v1.QuizService.GetQuiz",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAchievementsQuizzesV1QuizServiceUrl({
        protoPath: "/v1/quizzes/{quizId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "quiz.createdDate" }, { path: "quiz.updatedDate" }]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "quiz.fields.score" },
            { path: "quiz.fields.numeric.rightAnswer" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getQuiz;
}
function deleteQuiz(payload) {
  function __deleteQuiz({ host }) {
    const metadata = {
      entityFqdn: "wix.achievements.quizzes.v1.quiz",
      method: "DELETE",
      methodFqn: "wix.achievements.quizzes.v1.QuizService.DeleteQuiz",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAchievementsQuizzesV1QuizServiceUrl({
        protoPath: "/v1/quizzes/{quizId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "quiz.createdDate" }, { path: "quiz.updatedDate" }]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "quiz.fields.score" },
            { path: "quiz.fields.numeric.rightAnswer" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __deleteQuiz;
}

// src/achievements-quizzes-v1-quiz-quizzes.types.ts
var FileFormat = /* @__PURE__ */ ((FileFormat2) => {
  FileFormat2["IMAGE"] = "IMAGE";
  FileFormat2["VIDEO"] = "VIDEO";
  FileFormat2["AUDIO"] = "AUDIO";
  FileFormat2["DOCUMENT"] = "DOCUMENT";
  FileFormat2["ARCHIVE"] = "ARCHIVE";
  return FileFormat2;
})(FileFormat || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/achievements-quizzes-v1-quiz-quizzes.meta.ts
function createQuiz2() {
  const payload = {};
  const getRequestOptions = createQuiz(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/quizzes",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function cloneQuiz2() {
  const payload = { quizId: ":quizId" };
  const getRequestOptions = cloneQuiz(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/quizzes/{quizId}/clone",
    pathParams: { quizId: "quizId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkCreateQuiz2() {
  const payload = {};
  const getRequestOptions = bulkCreateQuiz(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/quizzes/create",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkCloneQuiz2() {
  const payload = {};
  const getRequestOptions = bulkCloneQuiz(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/quizzes/clone",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getQuiz2() {
  const payload = { quizId: ":quizId" };
  const getRequestOptions = getQuiz(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/quizzes/{quizId}",
    pathParams: { quizId: "quizId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteQuiz2() {
  const payload = { quizId: ":quizId" };
  const getRequestOptions = deleteQuiz(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/quizzes/{quizId}",
    pathParams: { quizId: "quizId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  FileFormat as FileFormatOriginal,
  WebhookIdentityType as WebhookIdentityTypeOriginal,
  bulkCloneQuiz2 as bulkCloneQuiz,
  bulkCreateQuiz2 as bulkCreateQuiz,
  cloneQuiz2 as cloneQuiz,
  createQuiz2 as createQuiz,
  deleteQuiz2 as deleteQuiz,
  getQuiz2 as getQuiz
};
