// @wix/bex-utils/@wix/wix-i18n-config/dist/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.withTranslation = exports.useTranslation = exports.initI18nWithoutICU = exports.initI18nBase = exports.initI18n = exports.i18n = exports.createRtfFormatter = exports.WithTranslationProps = exports.WithTranslation = exports.UseTranslationResponse = exports.TransProps = exports.Trans = exports.TFunction = exports.I18nextProvider = void 0;
var _icu = _interopRequireDefault(require("./icu/icu"));
exports.ICU = _icu.default;
var _i18next = require("i18next");
exports.i18n = _i18next.i18n;
exports.TFunction = _i18next.TFunction;
var _reactI18next = require("react-i18next");
exports.withTranslation = _reactI18next.withTranslation;
exports.WithTranslation = _reactI18next.WithTranslation;
exports.WithTranslationProps = _reactI18next.WithTranslationProps;
exports.useTranslation = _reactI18next.useTranslation;
exports.UseTranslationResponse = _reactI18next.UseTranslationResponse;
exports.I18nextProvider = _reactI18next.I18nextProvider;
exports.Trans = _reactI18next.Trans;
exports.TransProps = _reactI18next.TransProps;
var _initI18n = require("./initI18n");
exports.initI18n = _initI18n.initI18n;
var _initI18nWithoutICU = require("./initI18nWithoutICU");
exports.initI18nWithoutICU = _initI18nWithoutICU.initI18nWithoutICU;
var _relativeTimeFormatting = require("./relative-time-formatting");
exports.createRtfFormatter = _relativeTimeFormatting.createRtfFormatter;
var _i18n = require("./i18n");
exports.initI18nBase = _i18n.i18n;
