// @wix/bookings/build/internal/cjs/meta.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  attendance: () => attendance,
  availabilityCalendar: () => availabilityCalendar,
  availabilityTimeSlots: () => availabilityTimeSlots,
  bookingFees: () => bookingFees,
  bookingPolicies: () => bookingPolicies,
  bookingPolicySnapshots: () => bookingPolicySnapshots,
  bookings: () => bookings,
  categories: () => categories,
  categoriesV2: () => categoriesV2,
  eventTimeSlots: () => eventTimeSlots,
  extendedBookings: () => extendedBookings,
  externalCalendars: () => externalCalendars,
  multiServiceAvailabilityTimeSlots: () => multiServiceAvailabilityTimeSlots,
  pricing: () => pricing,
  resourceTypes: () => resourceTypes,
  resources: () => resources,
  serviceOptionsAndVariants: () => serviceOptionsAndVariants,
  services: () => services,
  staffMembers: () => staffMembers
});
module.exports = __toCommonJS(meta_exports);
var availabilityCalendar = __toESM(require("@wix/auto_sdk_bookings_availability-calendar/meta"));
var availabilityTimeSlots = __toESM(require("@wix/auto_sdk_bookings_availability-time-slots/meta"));
var eventTimeSlots = __toESM(require("@wix/auto_sdk_bookings_event-time-slots/meta"));
var multiServiceAvailabilityTimeSlots = __toESM(require("@wix/auto_sdk_bookings_multi-service-availability-time-slots/meta"));
var externalCalendars = __toESM(require("@wix/auto_sdk_bookings_external-calendars/meta"));
var serviceOptionsAndVariants = __toESM(require("@wix/auto_sdk_bookings_service-options-and-variants/meta"));
var categoriesV2 = __toESM(require("@wix/auto_sdk_bookings_categories-v-2/meta"));
var bookingFees = __toESM(require("@wix/auto_sdk_bookings_booking-fees/meta"));
var bookingPolicySnapshots = __toESM(require("@wix/auto_sdk_bookings_booking-policy-snapshots/meta"));
var extendedBookings = __toESM(require("@wix/auto_sdk_bookings_extended-bookings/meta"));
var resources = __toESM(require("@wix/auto_sdk_bookings_resources/meta"));
var resourceTypes = __toESM(require("@wix/auto_sdk_bookings_resource-types/meta"));
var services = __toESM(require("@wix/auto_sdk_bookings_services/meta"));
var staffMembers = __toESM(require("@wix/auto_sdk_bookings_staff-members/meta"));
var bookingPolicies = __toESM(require("@wix/auto_sdk_bookings_booking-policies/meta"));
var categories = __toESM(require("@wix/auto_sdk_bookings_categories/meta"));
var attendance = __toESM(require("@wix/auto_sdk_bookings_attendance/meta"));
var bookings = __toESM(require("@wix/auto_sdk_bookings_bookings/meta"));
var pricing = __toESM(require("@wix/auto_sdk_bookings_pricing/meta"));
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  attendance,
  availabilityCalendar,
  availabilityTimeSlots,
  bookingFees,
  bookingPolicies,
  bookingPolicySnapshots,
  bookings,
  categories,
  categoriesV2,
  eventTimeSlots,
  extendedBookings,
  externalCalendars,
  multiServiceAvailabilityTimeSlots,
  pricing,
  resourceTypes,
  resources,
  serviceOptionsAndVariants,
  services,
  staffMembers
});
