// @wix/ricos/dist/es/plugin-button/testkit/button.testkit.js
import {
  ALIGNMENT_DROPDOWN_OPTION_LEFT,
  PLUGIN_TOOLBAR_ALIGNMENT_BUTTON,
  PLUGIN_TOOLBAR_DELETE_BUTTON,
  PLUGIN_TOOLBAR_SETTINGS_BUTTON
} from "./../../_shared/data-hooks.js";
import { dataHook } from "./../../wix-rich-content-common/index.js";
import { BUTTON_LINK_SETTINGS_BUTTON, INSERT_BUTTON_BUTTON, ROOT } from "../data-hooks.js";
import { toolbarsToAddIn } from "../getAddButtons.js";
import { createButtonSettingsUniDriver } from "./button-settings.uni.driver.js";
const createButtonTestkit = (buttonType, tag) => (getBase) => {
  const getButton = async (index, options) => {
    const button = await getBase().getAll(`${tag}${dataHook(ROOT)}`, options);
    return button.at(index);
  };
  return {
    content: {
      atIndex: (index) => {
        return {
          exists: async (options) => {
            try {
              const button = await getButton(index, options);
              return button.exists();
            } catch (error) {
              return false;
            }
          },
          click: async () => (await getButton(index)).click(),
          getText: async () => (await getButton(index)).text(),
          getLink: async () => (await getButton(index)).attr("href"),
          getStyles: async () => {
            const btn = await getButton(index);
            return await btn.attr("style");
          }
        };
      }
    },
    toolbars: {
      createAddButtonToolbars: (getAddButtonToolbarBase) => ({
        click: async () => {
          const buttonPluginButton = await getAddButtonToolbarBase().get(
            dataHook(INSERT_BUTTON_BUTTON)
          );
          await buttonPluginButton.click();
        },
        toolbarsToAddIn
      }),
      createPluginToolbar: (getPluginToolbarBase) => ({
        openSettingsModal: async () => {
          const settings = await getPluginToolbarBase(buttonType).get(
            dataHook(PLUGIN_TOOLBAR_SETTINGS_BUTTON)
          );
          await settings.click();
        },
        openLinkModal: async () => {
          const link = await getPluginToolbarBase(buttonType).get(
            dataHook(BUTTON_LINK_SETTINGS_BUTTON)
          );
          await link.click();
        },
        clickDelete: async () => {
          await (await getPluginToolbarBase(buttonType).get(dataHook(PLUGIN_TOOLBAR_DELETE_BUTTON))).click();
        },
        clickAlignmentDropdown: async () => {
          await (await getPluginToolbarBase(buttonType).get(dataHook(PLUGIN_TOOLBAR_ALIGNMENT_BUTTON))).click();
        },
        alignment: {
          clickLeft: async () => {
            await (await getBase().get(dataHook(ALIGNMENT_DROPDOWN_OPTION_LEFT), {
              timeout: 5e3
            })).click();
          },
          toggleTextWrap: async () => {
            await (await getBase().get(dataHook("toggle-switch-input"), {
              timeout: 5e3
            })).click();
          },
          isToggleTextWrapDisabled: async () => {
            return await (await getBase().get(dataHook("toggle-switch-input"), {
              timeout: 5e3
            })).attr("readonly") !== null;
          }
        },
        exists: async () => {
          try {
            return getPluginToolbarBase(buttonType).exists();
          } catch (error) {
            return false;
          }
        }
      })
    },
    ...createButtonSettingsUniDriver(getBase)
  };
};
export {
  createButtonTestkit
};
