// @wix/ricos/dist/es/plugin-button/testkit/button-settings.uni.driver.js
import { InputUniDriver, TabsUniDriver } from "@wix/design-system/dist/testkit/unidriver.js";
import { createLinkModalUnidriver } from "./../../_shared/link-modal/create-link-modal-unidriver.js";
import { dataHook } from "./../../wix-rich-content-common/index.js";
import {
  createButtonColorSwatchesUniDriver,
  createNewColorPickerUniDriver
} from "./../../ui-components/testkit.js";
import {
  DESIGN_SETTINGS_BACKGROUND_COLOR_INPUT,
  DESIGN_SETTINGS_BORDER_COLOR_INPUT,
  DESIGN_SETTINGS_TEXT_COLOR_INPUT,
  GENERAL_SETTINGS_BUTTON_TEXT_INPUT,
  SETTINGS_MODAL,
  SETTINGS_TABS
} from "../data-hooks.js";
const createButtonSettingsUniDriver = (getBase) => {
  return {
    settingsModal: {
      waitForElement: async () => {
        const modal = await getBase().get(`div${dataHook(SETTINGS_MODAL)}`);
        await modal.wait();
      },
      exists: async () => {
        try {
          const modal = await getBase().get(`div${dataHook(SETTINGS_MODAL)}`);
          return modal.exists();
        } catch (error) {
          return false;
        }
      },
      settings: {
        tabs: {
          ...TabsUniDriver(getBase(), getBase()),
          waitForElement: async () => {
            const tabs = await getBase().get(dataHook(SETTINGS_TABS));
            await tabs.wait();
          }
        },
        textInput: {
          waitForElement: async () => {
            await getBase().get(dataHook(GENERAL_SETTINGS_BUTTON_TEXT_INPUT));
          },
          ...InputUniDriver(getBase().$(dataHook(GENERAL_SETTINGS_BUTTON_TEXT_INPUT)), getBase())
        }
      },
      design: {
        textColor: createButtonColorSwatchesUniDriver(
          () => getBase().$(dataHook(DESIGN_SETTINGS_TEXT_COLOR_INPUT))
        ),
        borderColor: createButtonColorSwatchesUniDriver(
          () => getBase().$(dataHook(DESIGN_SETTINGS_BORDER_COLOR_INPUT))
        ),
        backgroundColor: createButtonColorSwatchesUniDriver(
          () => getBase().$(dataHook(DESIGN_SETTINGS_BACKGROUND_COLOR_INPUT))
        )
      }
    },
    linkModal: createLinkModalUnidriver(getBase),
    colorPickerModal: createNewColorPickerUniDriver(getBase)
  };
};
export {
  createButtonSettingsUniDriver
};
