// @wix/auto_sdk_headless-site-assets_scripts/build/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  PositionOnPage: () => PositionOnPage,
  getEssentialProperties: () => getEssentialProperties2,
  listSiteScripts: () => listSiteScripts2
});
module.exports = __toCommonJS(index_typings_exports);

// src/assets-v1-site-script-scripts.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/assets-v1-site-script-scripts.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixAssetsScriptsV1SiteScriptsServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/_api/site-scripts",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/site-scripts",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/site-scripts",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/assets/scripts/v1/site-scripts",
        destPath: "/v1/site-scripts"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_headless-site-assets_scripts";
function listSiteScripts(payload) {
  function __listSiteScripts({ host }) {
    const metadata = {
      entityFqdn: "wix.assets.v1.site_script",
      method: "GET",
      methodFqn: "wix.assets.scripts.v1.SiteScriptsService.ListSiteScripts",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAssetsScriptsV1SiteScriptsServiceUrl({
        protoPath: "/v1/site-scripts",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __listSiteScripts;
}
function getEssentialProperties(payload) {
  function __getEssentialProperties({ host }) {
    const metadata = {
      entityFqdn: "wix.assets.v1.site_script",
      method: "GET",
      methodFqn: "wix.assets.scripts.v1.SiteScriptsService.GetEssentialProperties",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAssetsScriptsV1SiteScriptsServiceUrl({
        protoPath: "/v1/site-scripts/essentials",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getEssentialProperties;
}

// src/assets-v1-site-script-scripts.universal.ts
var PositionOnPage = /* @__PURE__ */ ((PositionOnPage2) => {
  PositionOnPage2["UNKNOWN_POSITION"] = "UNKNOWN_POSITION";
  PositionOnPage2["HEAD"] = "HEAD";
  PositionOnPage2["BODY_START"] = "BODY_START";
  PositionOnPage2["BODY_END"] = "BODY_END";
  return PositionOnPage2;
})(PositionOnPage || {});
async function listSiteScripts2() {
  const { httpClient, sideEffects } = arguments[0];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({});
  const reqOpts = listSiteScripts(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {},
        singleArgumentUnchanged: false
      },
      []
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getEssentialProperties2() {
  const { httpClient, sideEffects } = arguments[0];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({});
  const reqOpts = getEssentialProperties(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {},
        singleArgumentUnchanged: false
      },
      []
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  PositionOnPage,
  getEssentialProperties,
  listSiteScripts
});
