// @wix/patterns/dist/cjs/components/DragAndDrop/DragOverlayTestComponent.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.DragOverlayTestComponent = void 0;
var _react = _interopRequireDefault(require("react"));
var _mobxReactLite = require("mobx-react-lite");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/DragAndDrop/DragOverlayTestComponent.tsx";
function _DragOverlayTestComponent(props) {
  const {
    wrapperElement: Wrapper,
    style,
    children,
    className,
    zIndex,
    state
  } = props;
  if (state.activeId == null) {
    return null;
  }
  let child = children;
  if (Wrapper) {
    child = /*#__PURE__*/_react.default.createElement(Wrapper, {
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 31,
        columnNumber: 13
      }
    }, children);
  }
  return /*#__PURE__*/_react.default.createElement("div", {
    className: className,
    style: {
      position: 'fixed',
      ...style,
      zIndex
    },
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 35,
      columnNumber: 5
    }
  }, child);
}
const DragOverlayTestComponent = exports.DragOverlayTestComponent = (0, _mobxReactLite.observer)(_DragOverlayTestComponent);
