// @wix/wix-style-react-incubator/dist/cjs/drag-and-drop/table/DraggableTableRow.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.DraggableTableRow = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireWildcard(require("react"));
var _utilities = require("@dnd-kit/utilities");
var _TestWrappers = require("./TestWrappers");
var _DraggableTableRowStCss = require("./DraggableTableRow.st.css.js");
var _DragAndDropContext = require("./DragAndDropContext");
var _useSortableTableRow = require("./useSortableTableRow");
var _DraggableRowContext = require("./DraggableRowContext");
var _jsxFileName = "/home/builduser/work/b60d7defec63c7cc/packages/wix-style-react-incubator/dist/cjs/drag-and-drop/table/DraggableTableRow.tsx";
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
const DraggableTableRow = exports.DraggableTableRow = /*#__PURE__*/_react.default.memo(function _DraggableTableRow(props) {
  const {
    children,
    id,
    index,
    isDragOverlay,
    isDragAndDropDisabled,
    forwardedRef
  } = props;
  const {
    type: Tr,
    key,
    props: trProps
  } = children;
  const {
    onDragAndDropDisabledRowPointerDown,
    eventHandlers,
    onDropAnimationEnd,
    hideOverlayBottomShadow,
    useIsDropDisabled
  } = (0, _DragAndDropContext.useDragAndDropContext)();
  const dragHandleRef = (0, _react.useRef)();
  (0, _react.useEffect)(() => {
    if (onDropAnimationEnd) {
      return onDropAnimationEnd(id, dragHandleRef);
    }
    return;
  }, [id, onDropAnimationEnd]);
  const isDropDisabled = useIsDropDisabled == null ? void 0 : useIsDropDisabled(id);
  const disabled = {
    draggable: false,
    droppable: isDropDisabled
  };
  const {
    attributes,
    listeners,
    transform,
    transition,
    setNodeRef,
    isDragging
  } = (0, _useSortableTableRow.useSortableTableRow)({
    id,
    disabled
  });
  const draggableRowContextValue = (0, _react.useMemo)(() => ({
    isDragOverlay
  }), [isDragOverlay]);
  const ref = (0, _react.useCallback)(el => {
    forwardedRef == null || forwardedRef(el);
    setNodeRef(el);
  }, [forwardedRef, setNodeRef]);
  if (isDragAndDropDisabled) {
    return /*#__PURE__*/_react.default.createElement(Tr, (0, _extends2.default)({
      key: key
    }, trProps, {
      onPointerDown: e => {
        trProps.onPointerDown == null || trProps.onPointerDown(e);
        onDragAndDropDisabledRowPointerDown == null || onDragAndDropDisabledRowPointerDown(id, e.nativeEvent);
      },
      ref: el => {
        forwardedRef == null || forwardedRef(el);
      },
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 94,
        columnNumber: 7
      }
    }));
  }
  const style = {
    transform: _utilities.CSS.Transform.toString(transform),
    transition
  };
  const rowEvents = eventHandlers == null || eventHandlers.row == null ? void 0 : eventHandlers.row({
    id
  });
  let tr = /*#__PURE__*/_react.default.createElement(Tr, (0, _extends2.default)({
    key: key
  }, trProps, {
    className: (0, _DraggableTableRowStCss.st)(_DraggableTableRowStCss.classes.root, {
      dragging: isDragging,
      overlay: isDragOverlay,
      hideOverlayBottomShadow
    }, trProps.className),
    style: {
      ...trProps.style,
      ...style
    }
  }, attributes, listeners, {
    ref: ref,
    tabIndex: undefined
  }, rowEvents, {
    onPointerDown: e => {
      trProps == null || trProps.onPointerDown == null || trProps.onPointerDown(e);
      listeners == null || listeners.onPointerDown == null || listeners.onPointerDown(e);
      rowEvents == null || rowEvents.onPointerDown == null || rowEvents.onPointerDown(e);
    },
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 116,
      columnNumber: 5
    }
  }));
  if (process.env.NODE_ENV === 'test') {
    tr = /*#__PURE__*/_react.default.createElement(_TestWrappers.TableRowTestWrapper, (0, _extends2.default)({}, eventHandlers == null || eventHandlers.row == null ? void 0 : eventHandlers.row({
      id,
      disabled
    }), {
      id: id,
      index: index,
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 147,
        columnNumber: 7
      }
    }), tr);
  }
  return /*#__PURE__*/_react.default.createElement(_DraggableRowContext.DraggableRowContext.Provider, {
    value: draggableRowContextValue,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 158,
      columnNumber: 5
    }
  }, tr);
});
