// @wix/wix-style-react-incubator/dist/cjs/drag-and-drop/table/DragHandle.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.DragHandle = DragHandle;
exports.isDragHandleNode = isDragHandleNode;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireWildcard(require("react"));
var _system = require("@wix/wix-ui-icons-common/system");
var _DragHandleStCss = require("./DragHandle.st.css.js");
var _DragAndDropContext = require("./DragAndDropContext");
var _useSortableTableRow = require("./useSortableTableRow");
var _jsxFileName = "/home/builduser/work/b60d7defec63c7cc/packages/wix-style-react-incubator/dist/cjs/drag-and-drop/table/DragHandle.tsx";
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function DragHandle(props) {
  const {
    id,
    className
  } = props;
  const {
    setActivatorNodeRef
  } = (0, _useSortableTableRow.useSortableTableRow)({
    id
  });
  const {
    eventHandlers,
    announcements,
    dragHandleRef: dragHandleRefProp,
    onDropAnimationEnd
  } = (0, _DragAndDropContext.useDragAndDropContext)();
  const ariaDescribedBy = `draggable-table-row-description-${id}`;
  const dragHandleRef = (0, _react.useRef)();
  (0, _react.useEffect)(() => {
    if (onDropAnimationEnd) {
      return onDropAnimationEnd(id, dragHandleRef);
    }
    return;
  }, [id, onDropAnimationEnd]);
  return /*#__PURE__*/_react.default.createElement("div", (0, _extends2.default)({}, eventHandlers == null || eventHandlers.handle == null ? void 0 : eventHandlers.handle({
    id
  }), {
    className: (0, _DragHandleStCss.st)(_DragHandleStCss.classes.root, {}, className),
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 37,
      columnNumber: 5
    }
  }), /*#__PURE__*/_react.default.createElement("div", {
    id: ariaDescribedBy,
    style: {
      display: 'none'
    },
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 41,
      columnNumber: 7
    }
  }, announcements == null || announcements.itemDescription == null ? void 0 : announcements.itemDescription(id)), /*#__PURE__*/_react.default.createElement("div", {
    ref: e => {
      dragHandleRef.current = e;
      setActivatorNodeRef(e);
      dragHandleRefProp == null || dragHandleRefProp(id, e);
    },
    role: "button",
    "aria-describedby": ariaDescribedBy,
    "aria-label": announcements == null || announcements.dragHandleLabel == null ? void 0 : announcements.dragHandleLabel(id),
    className: _DragHandleStCss.classes.triggerIcon,
    tabIndex: 0,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 44,
      columnNumber: 7
    }
  }, /*#__PURE__*/_react.default.createElement(_system.DragAndDropLarge, {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 56,
      columnNumber: 9
    }
  })));
}
function isDragHandleNode(node) {
  return node.classList.contains(_DragHandleStCss.classes.triggerIcon);
}
