// @wix/wix-style-react-incubator/dist/cjs/drag-and-drop/table/TableRowDragOverlay.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.TableRowDragOverlay = TableRowDragOverlay;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireDefault(require("react"));
var _core = require("@dnd-kit/core");
var _DragAndDropContext = require("./DragAndDropContext");
var _modifiers = require("@dnd-kit/modifiers");
var _useDefaultDragOverlayProps = require("./useDefaultDragOverlayProps");
var _jsxFileName = "/home/builduser/work/b60d7defec63c7cc/packages/wix-style-react-incubator/dist/cjs/drag-and-drop/table/TableRowDragOverlay.tsx";
function TableRowDragOverlay({
  children
}) {
  const {
    dropAnimation,
    DragOverlayComponentType = _core.DragOverlay
  } = (0, _DragAndDropContext.useDragAndDropContext)();
  const defaultProps = (0, _useDefaultDragOverlayProps.useDefaultDragOverlayProps)();
  return /*#__PURE__*/_react.default.createElement(DragOverlayComponentType, (0, _extends2.default)({}, defaultProps, {
    dropAnimation: dropAnimation,
    modifiers: [_modifiers.restrictToWindowEdges],
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 14,
      columnNumber: 5
    }
  }), children);
}
