// @wix/wix-style-react-incubator/dist/cjs/drag-and-drop/table/TableRowHorizontalScrollDragOverlay.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.TableRowHorizontalScrollDragOverlay = TableRowHorizontalScrollDragOverlay;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireWildcard(require("react"));
var _core = require("@dnd-kit/core");
var _DragAndDropContext = require("./DragAndDropContext");
var _utilities = require("@dnd-kit/utilities");
var _restrictContainerToWindowEdges = require("./restrictContainerToWindowEdges");
var _restrictKeyboardToVerticalAxis = require("./restrictKeyboardToVerticalAxis");
var _useDefaultDragOverlayProps = require("./useDefaultDragOverlayProps");
var _jsxFileName = "/home/builduser/work/b60d7defec63c7cc/packages/wix-style-react-incubator/dist/cjs/drag-and-drop/table/TableRowHorizontalScrollDragOverlay.tsx";
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function TableRowHorizontalScrollDragOverlay({
  children
}) {
  const {
    dropAnimation: dropAnimationProp,
    DragOverlayComponentType = _core.DragOverlay
  } = (0, _DragAndDropContext.useDragAndDropContext)();
  const {
    scrollableAncestorRects,
    activeNodeRect
  } = (0, _core.useDndContext)();
  const containerNodeRect = scrollableAncestorRects[0];
  const containerWidth = containerNodeRect == null ? void 0 : containerNodeRect.width;
  const widthRef = (0, _react.useRef)(containerWidth);
  if (containerWidth != null) {
    widthRef.current = containerWidth;
  }
  const containerLeft = containerNodeRect == null ? void 0 : containerNodeRect.left;
  const leftRef = (0, _react.useRef)(containerLeft);
  if (containerNodeRect != null) {
    leftRef.current = containerLeft;
  }
  const style = (0, _react.useMemo)(() => ({
    left: leftRef.current,
    width: widthRef.current != null && (activeNodeRect == null ? void 0 : activeNodeRect.width) != null ? Math.min(widthRef.current, activeNodeRect.width) : widthRef.current
  }), [widthRef.current, leftRef.current, activeNodeRect]);
  const keyframes = (0, _react.useCallback)(({
    transform: {
      initial,
      final
    }
  }) => {
    return [{
      transform: _utilities.CSS.Transform.toString(initial)
    }, {
      transform: _utilities.CSS.Transform.toString({
        ...final,
        x: initial.x
      })
    }];
  }, []);
  const dropAnimation = (0, _react.useMemo)(() => ({
    ...dropAnimationProp,
    keyframes
  }), [dropAnimationProp, keyframes]);
  const defaultDragOverlayProps = (0, _useDefaultDragOverlayProps.useDefaultDragOverlayProps)();
  return /*#__PURE__*/_react.default.createElement(DragOverlayComponentType, (0, _extends2.default)({}, defaultDragOverlayProps, {
    dropAnimation: dropAnimation,
    modifiers: [_restrictContainerToWindowEdges.restrictContainerToWindowEdges, _restrictKeyboardToVerticalAxis.restrictKeyboardToVerticalAxis],
    style: {
      ...defaultDragOverlayProps.style,
      ...style
    },
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 80,
      columnNumber: 5
    }
  }), children);
}
