// @wix/wix-style-react-incubator/dist/cjs/drag-and-drop/table/TestWrappers.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.TableTestWrapper = exports.TableRowTestWrapper = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireWildcard(require("react"));
var _sortable = require("@dnd-kit/sortable");
var _jsxFileName = "/home/builduser/work/b60d7defec63c7cc/packages/wix-style-react-incubator/dist/cjs/drag-and-drop/table/TestWrappers.tsx";
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
const dndContext = {};
const TableTestWrapper = props => {
  const {
    children,
    sortableId
  } = props;
  const {
    props: {
      onDragStart,
      onDragEnd,
      onDragCancel
    }
  } = children;
  (0, _react.useEffect)(() => {
    dndContext[sortableId] = {
      onDragStart,
      onDragEnd: onDragEnd,
      onDragCancel: onDragCancel,
      activeId: null,
      activeIndex: null,
      overId: null,
      overIndex: null
    };
    return () => {
      dndContext[sortableId] = null;
    };
  }, [onDragStart, onDragEnd, onDragCancel, sortableId]);
  return children;
};
exports.TableTestWrapper = TableTestWrapper;
const TableRowTestWrapper = exports.TableRowTestWrapper = function _DraggableTableRow(props) {
  const {
    children,
    index,
    id,
    ...rest
  } = props;
  const {
    type: DraggableTableRow,
    props: draggableTableRowProps
  } = children;
  const {
    data
  } = (0, _sortable.useSortable)({
    id
  });
  const containerId = data.sortable.containerId;
  return /*#__PURE__*/_react.default.createElement(DraggableTableRow, (0, _extends2.default)({}, draggableTableRowProps, {
    onPointerDown: () => {
      const dnd = dndContext[containerId];
      if (dnd) {
        dnd.activeId = id;
        dnd.activeIndex = index;
        dnd.onDragStart == null || dnd.onDragStart({
          active: {
            id: dnd.activeId,
            data: {
              current: {
                sortable: {
                  containerId,
                  items: [],
                  index
                }
              }
            }
          }
        });
      }
    },
    onPointerUp: () => {
      const dnd = dndContext[containerId];
      if (dnd) {
        dnd.overId = id;
        dnd.overIndex = index;
        dnd.onDragEnd == null || dnd.onDragEnd({
          active: {
            id: dnd.activeId
          },
          over: {
            id: dnd.overId
          }
        });
      }
      setTimeout(() => {
        dndContext[containerId] = null;
      });
    }
  }, rest, {
    style: {
      ...draggableTableRowProps.style,
      ...rest.style,
      transition: undefined
    },
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 72,
      columnNumber: 5
    }
  }));
};
