// @wix/auto_sdk_bookings_service-options-and-variants/build/internal/es/index.mjs
// src/bookings-catalog-v1-service-options-and-variants-service-options-and-variants.public.ts
import { renameKeysFromRESTResponseToSDKResponse as renameKeysFromRESTResponseToSDKResponse2 } from "@wix/sdk-runtime/rename-all-nested-keys";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths as transformPaths3 } from "@wix/sdk-runtime/transformations/transform-paths";
import { EventDefinition } from "@wix/sdk-types";

// src/bookings-catalog-v1-service-options-and-variants-service-options-and-variants.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/bookings-catalog-v1-service-options-and-variants-service-options-and-variants.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixBookingsCatalogV1ServiceOptionsAndVariantsServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/v1/service-options-and-variants",
        destPath: "/v1/serviceOptionsAndVariants"
      },
      {
        srcPath: "/_api/v1/serviceOptionsAndVariants",
        destPath: "/v1/serviceOptionsAndVariants"
      },
      {
        srcPath: "/_api/bookings/v1/service-options-and-variants",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/v1/service-options-and-variants",
        destPath: "/v1/serviceOptionsAndVariants"
      },
      {
        srcPath: "/_api/v1/serviceOptionsAndVariants",
        destPath: "/v1/serviceOptionsAndVariants"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/v1/service-options-and-variants",
        destPath: "/v1/serviceOptionsAndVariants"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/v1/service-options-and-variants",
        destPath: "/v1/serviceOptionsAndVariants"
      },
      {
        srcPath: "/_api/bookings/v1/service-options-and-variants",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/v1/service-options-and-variants",
        destPath: "/v1/serviceOptionsAndVariants"
      },
      {
        srcPath: "/_api/bookings/v1/service-options-and-variants",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/v1/service-options-and-variants",
        destPath: "/v1/serviceOptionsAndVariants"
      },
      {
        srcPath: "/_api/bookings/v1/service-options-and-variants",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/bookings/v1/serviceOptionsAndVariants",
        destPath: "/v1/serviceOptionsAndVariants"
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/_api/v1/service-options-and-variants",
        destPath: "/v1/serviceOptionsAndVariants"
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/v1/service-options-and-variants",
        destPath: "/v1/serviceOptionsAndVariants"
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/v1/service-options-and-variants",
        destPath: "/v1/serviceOptionsAndVariants"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_bookings_service-options-and-variants";
function createServiceOptionsAndVariants(payload) {
  function __createServiceOptionsAndVariants({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.catalog.v1.service_options_and_variants",
      method: "POST",
      methodFqn: "wix.bookings.catalog.v1.ServiceOptionsAndVariantsService.CreateServiceOptionsAndVariants",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsCatalogV1ServiceOptionsAndVariantsServiceUrl({
        protoPath: "/v1/serviceOptionsAndVariants",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __createServiceOptionsAndVariants;
}
function cloneServiceOptionsAndVariants(payload) {
  function __cloneServiceOptionsAndVariants({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.catalog.v1.service_options_and_variants",
      method: "POST",
      methodFqn: "wix.bookings.catalog.v1.ServiceOptionsAndVariantsService.CloneServiceOptionsAndVariants",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsCatalogV1ServiceOptionsAndVariantsServiceUrl({
        protoPath: "/v1/serviceOptionsAndVariants/{cloneFromId}/clone",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __cloneServiceOptionsAndVariants;
}
function getServiceOptionsAndVariants(payload) {
  function __getServiceOptionsAndVariants({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.catalog.v1.service_options_and_variants",
      method: "GET",
      methodFqn: "wix.bookings.catalog.v1.ServiceOptionsAndVariantsService.GetServiceOptionsAndVariants",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsCatalogV1ServiceOptionsAndVariantsServiceUrl({
        protoPath: "/v1/serviceOptionsAndVariants/{serviceOptionsAndVariantsId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      fallback: [
        {
          method: "GET",
          url: resolveWixBookingsCatalogV1ServiceOptionsAndVariantsServiceUrl({
            protoPath: "/v1/serviceOptionsAndVariants/{serviceOptionsAndVariantsId}",
            data: payload,
            host
          }),
          params: toURLSearchParams(payload)
        }
      ]
    };
    return metadata;
  }
  return __getServiceOptionsAndVariants;
}
function getServiceOptionsAndVariantsByServiceId(payload) {
  function __getServiceOptionsAndVariantsByServiceId({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.catalog.v1.service_options_and_variants",
      method: "GET",
      methodFqn: "wix.bookings.catalog.v1.ServiceOptionsAndVariantsService.GetServiceOptionsAndVariantsByServiceId",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsCatalogV1ServiceOptionsAndVariantsServiceUrl({
        protoPath: "/v1/serviceOptionsAndVariants/service_id/{serviceId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      fallback: [
        {
          method: "GET",
          url: resolveWixBookingsCatalogV1ServiceOptionsAndVariantsServiceUrl({
            protoPath: "/v1/serviceOptionsAndVariants/service_id/{serviceId}",
            data: payload,
            host
          }),
          params: toURLSearchParams(payload)
        }
      ]
    };
    return metadata;
  }
  return __getServiceOptionsAndVariantsByServiceId;
}
function updateServiceOptionsAndVariants(payload) {
  function __updateServiceOptionsAndVariants({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.bookings.catalog.v1.service_options_and_variants",
      method: "PATCH",
      methodFqn: "wix.bookings.catalog.v1.ServiceOptionsAndVariantsService.UpdateServiceOptionsAndVariants",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsCatalogV1ServiceOptionsAndVariantsServiceUrl({
        protoPath: "/v1/serviceOptionsAndVariants/{serviceOptionsAndVariants.id}",
        data: serializedData,
        host
      }),
      data: serializedData
    };
    return metadata;
  }
  return __updateServiceOptionsAndVariants;
}
function deleteServiceOptionsAndVariants(payload) {
  function __deleteServiceOptionsAndVariants({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.catalog.v1.service_options_and_variants",
      method: "DELETE",
      methodFqn: "wix.bookings.catalog.v1.ServiceOptionsAndVariantsService.DeleteServiceOptionsAndVariants",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsCatalogV1ServiceOptionsAndVariantsServiceUrl({
        protoPath: "/v1/serviceOptionsAndVariants/{serviceOptionsAndVariantsId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteServiceOptionsAndVariants;
}
function queryServiceOptionsAndVariants(payload) {
  function __queryServiceOptionsAndVariants({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.catalog.v1.service_options_and_variants",
      method: "POST",
      methodFqn: "wix.bookings.catalog.v1.ServiceOptionsAndVariantsService.QueryServiceOptionsAndVariants",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsCatalogV1ServiceOptionsAndVariantsServiceUrl({
        protoPath: "/v1/serviceOptionsAndVariants/query",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __queryServiceOptionsAndVariants;
}

// src/bookings-catalog-v1-service-options-and-variants-service-options-and-variants.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var ServiceOptionType = /* @__PURE__ */ ((ServiceOptionType2) => {
  ServiceOptionType2["UNKNOWN"] = "UNKNOWN";
  ServiceOptionType2["CUSTOM"] = "CUSTOM";
  ServiceOptionType2["STAFF_MEMBER"] = "STAFF_MEMBER";
  ServiceOptionType2["DURATION"] = "DURATION";
  return ServiceOptionType2;
})(ServiceOptionType || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var PlacementType = /* @__PURE__ */ ((PlacementType2) => {
  PlacementType2["BEFORE"] = "BEFORE";
  PlacementType2["AFTER"] = "AFTER";
  PlacementType2["REPLACE"] = "REPLACE";
  return PlacementType2;
})(PlacementType || {});
var DayOfWeek = /* @__PURE__ */ ((DayOfWeek2) => {
  DayOfWeek2["MONDAY"] = "MONDAY";
  DayOfWeek2["TUESDAY"] = "TUESDAY";
  DayOfWeek2["WEDNESDAY"] = "WEDNESDAY";
  DayOfWeek2["THURSDAY"] = "THURSDAY";
  DayOfWeek2["FRIDAY"] = "FRIDAY";
  DayOfWeek2["SATURDAY"] = "SATURDAY";
  DayOfWeek2["SUNDAY"] = "SUNDAY";
  return DayOfWeek2;
})(DayOfWeek || {});
var ResolutionMethod = /* @__PURE__ */ ((ResolutionMethod2) => {
  ResolutionMethod2["QUERY_PARAM"] = "QUERY_PARAM";
  ResolutionMethod2["SUBDOMAIN"] = "SUBDOMAIN";
  ResolutionMethod2["SUBDIRECTORY"] = "SUBDIRECTORY";
  return ResolutionMethod2;
})(ResolutionMethod || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createServiceOptionsAndVariants2(serviceOptionsAndVariants) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    serviceOptionsAndVariants
  });
  const reqOpts = createServiceOptionsAndVariants(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.serviceOptionsAndVariants;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { serviceOptionsAndVariants: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["serviceOptionsAndVariants"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function cloneServiceOptionsAndVariants2(cloneFromId, targetServiceId) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    cloneFromId,
    targetServiceId
  });
  const reqOpts = cloneServiceOptionsAndVariants(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          cloneFromId: "$[0]",
          targetServiceId: "$[1]"
        },
        singleArgumentUnchanged: false
      },
      ["cloneFromId", "targetServiceId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getServiceOptionsAndVariants2(serviceOptionsAndVariantsId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    serviceOptionsAndVariantsId
  });
  const reqOpts = getServiceOptionsAndVariants(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.serviceOptionsAndVariants;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { serviceOptionsAndVariantsId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["serviceOptionsAndVariantsId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getServiceOptionsAndVariantsByServiceId2(serviceId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    serviceId
  });
  const reqOpts = getServiceOptionsAndVariantsByServiceId(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { serviceId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["serviceId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateServiceOptionsAndVariants2(_id, serviceOptionsAndVariants) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    serviceOptionsAndVariants: { ...serviceOptionsAndVariants, id: _id }
  });
  const reqOpts = updateServiceOptionsAndVariants(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.serviceOptionsAndVariants;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { serviceOptionsAndVariants: "$[1]" },
        explicitPathsToArguments: { "serviceOptionsAndVariants.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "serviceOptionsAndVariants"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteServiceOptionsAndVariants2(serviceOptionsAndVariantsId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    serviceOptionsAndVariantsId,
    revision: options?.revision
  });
  const reqOpts = deleteServiceOptionsAndVariants(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          serviceOptionsAndVariantsId: "$[0]",
          revision: "$[1].revision"
        },
        singleArgumentUnchanged: false
      },
      ["serviceOptionsAndVariantsId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryServiceOptionsAndVariants2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryServiceOptionsAndVariants(
        payload
      );
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [])
      );
      return {
        items: transformedData?.serviceOptionsAndVariantsList,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}

// src/bookings-catalog-v1-service-options-and-variants-service-options-and-variants.public.ts
function createServiceOptionsAndVariants3(httpClient) {
  return (serviceOptionsAndVariants) => createServiceOptionsAndVariants2(
    serviceOptionsAndVariants,
    // @ts-ignore
    { httpClient }
  );
}
function cloneServiceOptionsAndVariants3(httpClient) {
  return (cloneFromId, targetServiceId) => cloneServiceOptionsAndVariants2(
    cloneFromId,
    targetServiceId,
    // @ts-ignore
    { httpClient }
  );
}
function getServiceOptionsAndVariants3(httpClient) {
  return (serviceOptionsAndVariantsId) => getServiceOptionsAndVariants2(
    serviceOptionsAndVariantsId,
    // @ts-ignore
    { httpClient }
  );
}
function getServiceOptionsAndVariantsByServiceId3(httpClient) {
  return (serviceId) => getServiceOptionsAndVariantsByServiceId2(
    serviceId,
    // @ts-ignore
    { httpClient }
  );
}
function updateServiceOptionsAndVariants3(httpClient) {
  return (_id, serviceOptionsAndVariants) => updateServiceOptionsAndVariants2(
    _id,
    serviceOptionsAndVariants,
    // @ts-ignore
    { httpClient }
  );
}
function deleteServiceOptionsAndVariants3(httpClient) {
  return (serviceOptionsAndVariantsId, options) => deleteServiceOptionsAndVariants2(
    serviceOptionsAndVariantsId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function queryServiceOptionsAndVariants3(httpClient) {
  return () => queryServiceOptionsAndVariants2(
    // @ts-ignore
    { httpClient }
  );
}
var onServiceOptionsAndVariantsCreated = EventDefinition(
  "wix.bookings.catalog.v1.service_options_and_variants_created",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [{ path: "metadata.eventTime" }]
      }
    ])
  )
)();
var onServiceOptionsAndVariantsDeleted = EventDefinition(
  "wix.bookings.catalog.v1.service_options_and_variants_deleted",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [{ path: "metadata.eventTime" }]
      }
    ])
  )
)();
var onServiceOptionsAndVariantsUpdated = EventDefinition(
  "wix.bookings.catalog.v1.service_options_and_variants_updated",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [{ path: "metadata.eventTime" }]
      }
    ])
  )
)();

// src/bookings-catalog-v1-service-options-and-variants-service-options-and-variants.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
import { createEventModule } from "@wix/sdk-runtime/event-definition-modules";
var createServiceOptionsAndVariants4 = /* @__PURE__ */ createRESTModule(createServiceOptionsAndVariants3);
var cloneServiceOptionsAndVariants4 = /* @__PURE__ */ createRESTModule(cloneServiceOptionsAndVariants3);
var getServiceOptionsAndVariants4 = /* @__PURE__ */ createRESTModule(getServiceOptionsAndVariants3);
var getServiceOptionsAndVariantsByServiceId4 = /* @__PURE__ */ createRESTModule(
  getServiceOptionsAndVariantsByServiceId3
);
var updateServiceOptionsAndVariants4 = /* @__PURE__ */ createRESTModule(updateServiceOptionsAndVariants3);
var deleteServiceOptionsAndVariants4 = /* @__PURE__ */ createRESTModule(deleteServiceOptionsAndVariants3);
var queryServiceOptionsAndVariants4 = /* @__PURE__ */ createRESTModule(queryServiceOptionsAndVariants3);
var onServiceOptionsAndVariantsCreated2 = createEventModule(
  onServiceOptionsAndVariantsCreated
);
var onServiceOptionsAndVariantsDeleted2 = createEventModule(
  onServiceOptionsAndVariantsDeleted
);
var onServiceOptionsAndVariantsUpdated2 = createEventModule(
  onServiceOptionsAndVariantsUpdated
);
export {
  DayOfWeek,
  PlacementType,
  ResolutionMethod,
  ServiceOptionType,
  SortOrder,
  WebhookIdentityType,
  cloneServiceOptionsAndVariants4 as cloneServiceOptionsAndVariants,
  createServiceOptionsAndVariants4 as createServiceOptionsAndVariants,
  deleteServiceOptionsAndVariants4 as deleteServiceOptionsAndVariants,
  getServiceOptionsAndVariants4 as getServiceOptionsAndVariants,
  getServiceOptionsAndVariantsByServiceId4 as getServiceOptionsAndVariantsByServiceId,
  onServiceOptionsAndVariantsCreated2 as onServiceOptionsAndVariantsCreated,
  onServiceOptionsAndVariantsDeleted2 as onServiceOptionsAndVariantsDeleted,
  onServiceOptionsAndVariantsUpdated2 as onServiceOptionsAndVariantsUpdated,
  queryServiceOptionsAndVariants4 as queryServiceOptionsAndVariants,
  updateServiceOptionsAndVariants4 as updateServiceOptionsAndVariants
};
