// @wix/ricos/dist/cjs/plugin-html/toolbar/insert-modal/AdsenseSourceInput.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var AdsenseSourceInput_exports = {};
__export(AdsenseSourceInput_exports, {
  AdsenseSourceInput: () => AdsenseSourceInput
});
module.exports = __toCommonJS(AdsenseSourceInput_exports);
var import_design_system = require("@wix/design-system");
var import_react = __toESM(require("react"));
var import_ricos_context = require("./../../../ricos-context/index.js");
var import_ui_components = require("./../../../ui-components/index.js");
var import_HtmlInputArea = require("./HtmlInputArea.js");
const AdsenseSourceInput = ({
  htmlNodeAttrs,
  onHtmlInputChange
}) => {
  const { t } = (0, import_react.useContext)(import_ricos_context.RicosContext);
  const zIndexService = (0, import_react.useContext)(import_ricos_context.ZIndexContext);
  const containerEl = (0, import_ui_components.usePanelContainerElement)();
  return /* @__PURE__ */ import_react.default.createElement(import_design_system.Box, { direction: "vertical", gap: "SP3" }, /* @__PURE__ */ import_react.default.createElement(import_design_system.Box, { verticalAlign: "middle", gap: "SP1" }, /* @__PURE__ */ import_react.default.createElement(import_design_system.Text, { size: "small" }, t("HtmlEditPanel_HtmlInput_AdSense_Title"), " "), /* @__PURE__ */ import_react.default.createElement(
    import_design_system.InfoIcon,
    {
      size: "small",
      content: t("HtmlEditPanel_HtmlInput_AdSense_Tooltip"),
      tooltipProps: {
        appendTo: containerEl,
        zIndex: zIndexService.getZIndex("TOOLTIP")
      }
    }
  )), /* @__PURE__ */ import_react.default.createElement(
    import_HtmlInputArea.HtmlInputArea,
    {
      value: htmlNodeAttrs?.html ?? "",
      onChange: (value) => {
        onHtmlInputChange?.(value);
      },
      placeholder: t("HtmlEditPanel_HtmlInput_AdSense_Placeholder")
    }
  ));
};
