// @wix/fast-gallery-core/dist/cjs/lib/utils/commonUtils.js
"use strict";

exports.__esModule = true;
exports.resetItemCustomStyleSheet = exports.resetImageCustomStyleSheet = exports.resetContainerCustomStyleSheet = exports.itemSelector = exports.imageSelector = exports.cssStringToBuilderReturnType = exports.containerSelector = void 0;
var _serialize = require("@emotion/serialize");
const FastGalleryContainerSelector = 'container';
const FastGalleryItemSelector = 'item';
const FastGalleryImageSelector = 'image';
const containerSelector = exports.containerSelector = ` .${FastGalleryContainerSelector}`;
const itemSelector = exports.itemSelector = ` .${FastGalleryItemSelector}`;
const imageSelector = exports.imageSelector = ` .${FastGalleryImageSelector}`;
const cssToJsObject = cssString => {
  const cssObject = {};
  const cssRules = cssString.match(/([^{]+)\{([^}]*)\}/g) || [];
  cssRules.forEach(rule => {
    const [selector, properties] = rule.split('{');
    const propertyPairs = properties.replace('}', '').trim().split(';').filter(Boolean);
    const styleObject = propertyPairs.reduce((acc, pair) => {
      const [property, value] = pair.split(':');
      if (property && value) {
        acc[property.trim()] = value.trim();
      }
      return acc;
    }, {});
    cssObject[selector.trim()] = styleObject;
  });
  return cssObject;
};
const cssStringToBuilderReturnType = cssString => {
  return {
    cssString,
    cssObject: cssToJsObject(cssString)
  };
};
exports.cssStringToBuilderReturnType = cssStringToBuilderReturnType;
const resetContainerCustomStyleSheet = (selector, additionalCss) => {
  const cssReset = (0, _serialize.serializeStyles)([{
    [selector]: {
      display: 'block',
      gridTemplateColumns: 'unset',
      gridTemplateRows: 'unset',
      gridAutoFlow: 'unset',
      gridAutoColumns: 'unset',
      gridAutoRows: 'unset',
      rowGap: 'unset',
      flexWrap: 'nowrap',
      scrollSnapType: 'none',
      overflow: 'visible',
      scrollbarWidth: 'auto',
      gap: 0,
      height: 'auto',
      width: 'auto',
      columnGap: 'normal',
      ...additionalCss
    }
  }]).styles;
  return cssStringToBuilderReturnType(cssReset);
};
exports.resetContainerCustomStyleSheet = resetContainerCustomStyleSheet;
const resetItemCustomStyleSheet = (selector, additionalCss) => {
  const cssReset = (0, _serialize.serializeStyles)([{
    [selector]: {
      gridColumn: 'auto',
      gridRow: 'auto',
      height: 'auto',
      width: 'auto',
      flex: 'unset',
      margin: 0,
      scrollSnapAlign: 'none',
      marginRight: 0,
      ...additionalCss
    },
    [`${selector}:nth-child(n)`]: {
      gridColumn: 'auto',
      gridRow: 'auto',
      height: 'auto',
      width: 'auto',
      flex: 'unset',
      margin: 0,
      scrollSnapAlign: 'none',
      marginRight: 0,
      ...additionalCss
    }
  }]).styles;
  return cssStringToBuilderReturnType(cssReset);
};
exports.resetItemCustomStyleSheet = resetItemCustomStyleSheet;
const resetImageCustomStyleSheet = (selector, additionalCss) => {
  const cssReset = (0, _serialize.serializeStyles)([{
    [selector]: {
      display: 'block',
      width: 'auto !important',
      height: 'auto !important',
      img: {
        width: 'auto !important',
        height: 'auto !important',
        objectFit: 'fill',
        verticalAlign: 'baseline !important'
      },
      ...additionalCss
    }
  }]).styles;
  return cssStringToBuilderReturnType(cssReset);
};
exports.resetImageCustomStyleSheet = resetImageCustomStyleSheet;
