// @wix/fast-gallery-core/dist/cjs/lib/utils/itemBuilder.js
"use strict";

exports.__esModule = true;
exports.buildItemStyles = void 0;
var _serialize = require("@emotion/serialize");
var _types = require("../types");
var _commonUtils = require("./commonUtils");
const getContentCss = (infoPosition, contentItemsAlignment) => {
  const isFlexDirRow = ['LEFT', 'RIGHT', 'ON_IMAGE'].includes(infoPosition);
  const flexDirectionMap = {
    ABOVE: 'column-reverse',
    BELOW: 'column',
    LEFT: 'row-reverse',
    RIGHT: 'row',
    ON_IMAGE: 'row'
  };
  return {
    '.content': {
      justifyContent: isFlexDirRow ? 'center' : contentItemsAlignment,
      alignItems: isFlexDirRow ? contentItemsAlignment : 'center',
      flexDirection: flexDirectionMap[infoPosition] || 'column'
    },
    '.info': {
      // display: enableTitle ? 'block' : 'none',
    },
    ...(infoPosition === 'ON_IMAGE' ? {
      '.content .info, .content .counter': {
        position: 'absolute',
        textAlign: 'center',
        width: '100%',
        color: '#000000',
        top: '50%',
        left: '50%',
        transform: 'translate(-50%, -50%)'
      }
    } : {})
  };
};
const getMediaCss = (imageFit // fix type
) => {
  return {
    '.media': {
      height: imageFit === _types.ImageFit.crop ? '100%' : 'auto'
    },
    '.image': {
      objectFit: imageFit === _types.ImageFit.crop ? 'cover' : 'contain'
    }
  };
};
const getDetailsCss = infoPosition => {
  if (infoPosition === 'ON_IMAGE') {
    return {
      '.details': {
        position: 'absolute',
        height: '100%'
      }
    };
  } else {
    return {};
  }
};
const buildItemStyles = (styles = {}) => {
  const {
    infoPosition = 'ABOVE',
    // enableTitle = true,
    imageFit = _types.ImageFit.crop,
    contentItemsAlignment = 'flex-start'
  } = {
    ...styles
  };
  const contentRule = (0, _serialize.serializeStyles)([getContentCss(infoPosition, contentItemsAlignment)]).styles;
  const mediaRule = (0, _serialize.serializeStyles)([getMediaCss(imageFit)]).styles;
  const detailsRule = (0, _serialize.serializeStyles)([getDetailsCss(infoPosition)]).styles;
  return (0, _commonUtils.cssStringToBuilderReturnType)(contentRule + mediaRule + detailsRule);
};
exports.buildItemStyles = buildItemStyles;
