// @wix/ricos/dist/cjs/wix-rich-content-common/Components/AccessibilityListener.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var AccessibilityListener_exports = {};
__export(AccessibilityListener_exports, {
  default: () => AccessibilityListener
});
module.exports = __toCommonJS(AccessibilityListener_exports);
var import_react = require("react");
const CLASS_NAME = "no-outline";
class AccessibilityListener extends import_react.Component {
  constructor() {
    super(...arguments);
    this.handleTabKeyUp = (e) => {
      if (e.which === 9 && document.body.classList.contains(CLASS_NAME)) {
        document.body.classList.remove(CLASS_NAME);
      }
    };
    this.handleClick = () => {
      if (!document.body.classList.contains(CLASS_NAME)) {
        document.body.classList.add(CLASS_NAME);
      }
    };
    this.render = () => null;
  }
  componentDidMount() {
    document.body.classList.add(CLASS_NAME);
    if (!this.props.isMobile) {
      document.addEventListener("keyup", this.handleTabKeyUp);
      document.addEventListener("click", this.handleClick);
    }
  }
  componentWillUnmount() {
    if (!this.props.isMobile) {
      document.removeEventListener("keyup", this.handleTabKeyUp);
      document.removeEventListener("click", this.handleClick);
    }
  }
}
