// @wix/auto_sdk_online-programs_programs/build/internal/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  AccessType: () => AccessType,
  AggregationType: () => AggregationType,
  BulkActionType: () => BulkActionType,
  DayOfWeek: () => DayOfWeek,
  DeleteStatus: () => DeleteStatus,
  Enum: () => Enum,
  Interval: () => Interval,
  MissingValues: () => MissingValues,
  Mode: () => Mode,
  Namespace: () => Namespace,
  NestedAggregationType: () => NestedAggregationType,
  PlacementType: () => PlacementType,
  ResolutionMethod: () => ResolutionMethod,
  RestrictionsAccessType: () => RestrictionsAccessType,
  ScalarType: () => ScalarType,
  SearchIndexingNotificationState: () => SearchIndexingNotificationState,
  SiteCreatedContext: () => SiteCreatedContext,
  SortDirection: () => SortDirection,
  SortOrder: () => SortOrder,
  SortType: () => SortType,
  State: () => State,
  Status: () => Status,
  Trigger: () => Trigger,
  V3State: () => V3State,
  WebhookIdentityType: () => WebhookIdentityType,
  archiveProgram: () => archiveProgram2,
  bulkUpdatePrograms: () => bulkUpdatePrograms2,
  countPrograms: () => countPrograms2,
  createProgram: () => createProgram2,
  deleteProgram: () => deleteProgram2,
  duplicateProgram: () => duplicateProgram2,
  endProgram: () => endProgram2,
  getProgram: () => getProgram2,
  listSamplePrograms: () => listSamplePrograms2,
  publishProgram: () => publishProgram2,
  queryPrograms: () => queryPrograms2,
  searchPrograms: () => searchPrograms2,
  updateProgram: () => updateProgram2
});
module.exports = __toCommonJS(index_typings_exports);

// src/online-programs-v3-program-programs.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/online-programs-v3-program-programs.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixOnlineprogramsProgramsV3ProgramsServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/programs-service",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/programs-service",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/programs-service",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/_api/programs-service",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_online-programs_programs";
function createProgram(payload) {
  function __createProgram({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "program.createdDate" },
          { path: "program.updatedDate" },
          { path: "program.stateUpdatedDate" },
          { path: "program.statusUpdatedDate" },
          { path: "program.description.image.urlExpirationDate" },
          { path: "program.description.video.urlExpirationDate" },
          { path: "program.description.video.resolutions.urlExpirationDate" },
          {
            path: "program.description.video.resolutions.poster.urlExpirationDate"
          },
          { path: "program.description.video.posters.urlExpirationDate" },
          { path: "program.rewards.certificate.connectedDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "program.description.image.focalPoint.x" },
          { path: "program.description.image.focalPoint.y" },
          { path: "program.description.video.resolutions.poster.focalPoint.x" },
          { path: "program.description.video.resolutions.poster.focalPoint.y" },
          { path: "program.description.video.posters.focalPoint.x" },
          { path: "program.description.video.posters.focalPoint.y" },
          { path: "program.videoStepsSettings.requiredCompletionPercentage" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.online_programs.v3.program",
      method: "POST",
      methodFqn: "wix.onlineprograms.programs.v3.ProgramsService.CreateProgram",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsProgramsV3ProgramsServiceUrl({
        protoPath: "/v3/programs",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "program.createdDate" },
            { path: "program.updatedDate" },
            { path: "program.stateUpdatedDate" },
            { path: "program.statusUpdatedDate" },
            { path: "program.description.image.urlExpirationDate" },
            { path: "program.description.video.urlExpirationDate" },
            {
              path: "program.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "program.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "program.description.video.posters.urlExpirationDate" },
            { path: "program.rewards.certificate.connectedDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "program.description.image.focalPoint.x" },
            { path: "program.description.image.focalPoint.y" },
            {
              path: "program.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "program.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "program.description.video.posters.focalPoint.x" },
            { path: "program.description.video.posters.focalPoint.y" },
            {
              path: "program.videoStepsSettings.requiredCompletionPercentage"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createProgram;
}
function duplicateProgram(payload) {
  function __duplicateProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.program",
      method: "POST",
      methodFqn: "wix.onlineprograms.programs.v3.ProgramsService.DuplicateProgram",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsProgramsV3ProgramsServiceUrl({
        protoPath: "/v3/programs/{programId}/duplicate",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "program.createdDate" },
            { path: "program.updatedDate" },
            { path: "program.stateUpdatedDate" },
            { path: "program.statusUpdatedDate" },
            { path: "program.description.image.urlExpirationDate" },
            { path: "program.description.video.urlExpirationDate" },
            {
              path: "program.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "program.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "program.description.video.posters.urlExpirationDate" },
            { path: "program.rewards.certificate.connectedDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "program.description.image.focalPoint.x" },
            { path: "program.description.image.focalPoint.y" },
            {
              path: "program.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "program.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "program.description.video.posters.focalPoint.x" },
            { path: "program.description.video.posters.focalPoint.y" },
            {
              path: "program.videoStepsSettings.requiredCompletionPercentage"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __duplicateProgram;
}
function getProgram(payload) {
  function __getProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.program",
      method: "GET",
      methodFqn: "wix.onlineprograms.programs.v3.ProgramsService.GetProgram",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsProgramsV3ProgramsServiceUrl({
        protoPath: "/v3/programs/{programId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "program.createdDate" },
            { path: "program.updatedDate" },
            { path: "program.stateUpdatedDate" },
            { path: "program.statusUpdatedDate" },
            { path: "program.description.image.urlExpirationDate" },
            { path: "program.description.video.urlExpirationDate" },
            {
              path: "program.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "program.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "program.description.video.posters.urlExpirationDate" },
            { path: "program.rewards.certificate.connectedDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "program.description.image.focalPoint.x" },
            { path: "program.description.image.focalPoint.y" },
            {
              path: "program.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "program.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "program.description.video.posters.focalPoint.x" },
            { path: "program.description.video.posters.focalPoint.y" },
            {
              path: "program.videoStepsSettings.requiredCompletionPercentage"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getProgram;
}
function updateProgram(payload) {
  function __updateProgram({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "program.createdDate" },
          { path: "program.updatedDate" },
          { path: "program.stateUpdatedDate" },
          { path: "program.statusUpdatedDate" },
          { path: "program.description.image.urlExpirationDate" },
          { path: "program.description.video.urlExpirationDate" },
          { path: "program.description.video.resolutions.urlExpirationDate" },
          {
            path: "program.description.video.resolutions.poster.urlExpirationDate"
          },
          { path: "program.description.video.posters.urlExpirationDate" },
          { path: "program.rewards.certificate.connectedDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "program.description.image.focalPoint.x" },
          { path: "program.description.image.focalPoint.y" },
          { path: "program.description.video.resolutions.poster.focalPoint.x" },
          { path: "program.description.video.resolutions.poster.focalPoint.y" },
          { path: "program.description.video.posters.focalPoint.x" },
          { path: "program.description.video.posters.focalPoint.y" },
          { path: "program.videoStepsSettings.requiredCompletionPercentage" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.online_programs.v3.program",
      method: "PATCH",
      methodFqn: "wix.onlineprograms.programs.v3.ProgramsService.UpdateProgram",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsProgramsV3ProgramsServiceUrl({
        protoPath: "/v3/programs/{program.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "program.createdDate" },
            { path: "program.updatedDate" },
            { path: "program.stateUpdatedDate" },
            { path: "program.statusUpdatedDate" },
            { path: "program.description.image.urlExpirationDate" },
            { path: "program.description.video.urlExpirationDate" },
            {
              path: "program.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "program.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "program.description.video.posters.urlExpirationDate" },
            { path: "program.rewards.certificate.connectedDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "program.description.image.focalPoint.x" },
            { path: "program.description.image.focalPoint.y" },
            {
              path: "program.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "program.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "program.description.video.posters.focalPoint.x" },
            { path: "program.description.video.posters.focalPoint.y" },
            {
              path: "program.videoStepsSettings.requiredCompletionPercentage"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateProgram;
}
function deleteProgram(payload) {
  function __deleteProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.program",
      method: "DELETE",
      methodFqn: "wix.onlineprograms.programs.v3.ProgramsService.DeleteProgram",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsProgramsV3ProgramsServiceUrl({
        protoPath: "/v3/programs/{programId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteProgram;
}
function queryPrograms(payload) {
  function __queryPrograms({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.program",
      method: "POST",
      methodFqn: "wix.onlineprograms.programs.v3.ProgramsService.QueryPrograms",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsProgramsV3ProgramsServiceUrl({
        protoPath: "/v3/programs/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "programs.createdDate" },
            { path: "programs.updatedDate" },
            { path: "programs.stateUpdatedDate" },
            { path: "programs.statusUpdatedDate" },
            { path: "programs.description.image.urlExpirationDate" },
            { path: "programs.description.video.urlExpirationDate" },
            {
              path: "programs.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "programs.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "programs.description.video.posters.urlExpirationDate" },
            { path: "programs.rewards.certificate.connectedDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "programs.description.image.focalPoint.x" },
            { path: "programs.description.image.focalPoint.y" },
            {
              path: "programs.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "programs.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "programs.description.video.posters.focalPoint.x" },
            { path: "programs.description.video.posters.focalPoint.y" },
            {
              path: "programs.videoStepsSettings.requiredCompletionPercentage"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryPrograms;
}
function searchPrograms(payload) {
  function __searchPrograms({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "search.aggregations.range.buckets.from" },
          { path: "search.aggregations.range.buckets.to" },
          {
            path: "search.aggregations.nested.nestedAggregations.range.buckets.from"
          },
          {
            path: "search.aggregations.nested.nestedAggregations.range.buckets.to"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.online_programs.v3.program",
      method: "POST",
      methodFqn: "wix.onlineprograms.programs.v3.ProgramsService.SearchPrograms",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsProgramsV3ProgramsServiceUrl({
        protoPath: "/v3/programs/search",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "programs.createdDate" },
            { path: "programs.updatedDate" },
            { path: "programs.stateUpdatedDate" },
            { path: "programs.statusUpdatedDate" },
            { path: "programs.description.image.urlExpirationDate" },
            { path: "programs.description.video.urlExpirationDate" },
            {
              path: "programs.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "programs.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "programs.description.video.posters.urlExpirationDate" },
            { path: "programs.rewards.certificate.connectedDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "programs.description.image.focalPoint.x" },
            { path: "programs.description.image.focalPoint.y" },
            {
              path: "programs.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "programs.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "programs.description.video.posters.focalPoint.x" },
            { path: "programs.description.video.posters.focalPoint.y" },
            {
              path: "programs.videoStepsSettings.requiredCompletionPercentage"
            },
            { path: "aggregationData.results.scalar.value" },
            {
              path: "aggregationData.results.groupedByValue.results.nestedResults.scalar.value"
            },
            {
              path: "aggregationData.results.nested.results.results.*.scalar.value"
            },
            { path: "aggregationData.results.ranges.results.from" },
            { path: "aggregationData.results.ranges.results.to" },
            {
              path: "aggregationData.results.groupedByValue.results.nestedResults.ranges.results.from"
            },
            {
              path: "aggregationData.results.groupedByValue.results.nestedResults.ranges.results.to"
            },
            {
              path: "aggregationData.results.nested.results.results.*.range.from"
            },
            {
              path: "aggregationData.results.nested.results.results.*.range.to"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __searchPrograms;
}
function publishProgram(payload) {
  function __publishProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.program",
      method: "POST",
      methodFqn: "wix.onlineprograms.programs.v3.ProgramsService.PublishProgram",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsProgramsV3ProgramsServiceUrl({
        protoPath: "/v3/programs/{programId}/publish",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "program.createdDate" },
            { path: "program.updatedDate" },
            { path: "program.stateUpdatedDate" },
            { path: "program.statusUpdatedDate" },
            { path: "program.description.image.urlExpirationDate" },
            { path: "program.description.video.urlExpirationDate" },
            {
              path: "program.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "program.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "program.description.video.posters.urlExpirationDate" },
            { path: "program.rewards.certificate.connectedDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "program.description.image.focalPoint.x" },
            { path: "program.description.image.focalPoint.y" },
            {
              path: "program.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "program.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "program.description.video.posters.focalPoint.x" },
            { path: "program.description.video.posters.focalPoint.y" },
            {
              path: "program.videoStepsSettings.requiredCompletionPercentage"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __publishProgram;
}
function endProgram(payload) {
  function __endProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.program",
      method: "POST",
      methodFqn: "wix.onlineprograms.programs.v3.ProgramsService.EndProgram",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsProgramsV3ProgramsServiceUrl({
        protoPath: "/v3/programs/{programId}/end",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "program.createdDate" },
            { path: "program.updatedDate" },
            { path: "program.stateUpdatedDate" },
            { path: "program.statusUpdatedDate" },
            { path: "program.description.image.urlExpirationDate" },
            { path: "program.description.video.urlExpirationDate" },
            {
              path: "program.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "program.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "program.description.video.posters.urlExpirationDate" },
            { path: "program.rewards.certificate.connectedDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "program.description.image.focalPoint.x" },
            { path: "program.description.image.focalPoint.y" },
            {
              path: "program.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "program.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "program.description.video.posters.focalPoint.x" },
            { path: "program.description.video.posters.focalPoint.y" },
            {
              path: "program.videoStepsSettings.requiredCompletionPercentage"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __endProgram;
}
function archiveProgram(payload) {
  function __archiveProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.program",
      method: "POST",
      methodFqn: "wix.onlineprograms.programs.v3.ProgramsService.ArchiveProgram",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsProgramsV3ProgramsServiceUrl({
        protoPath: "/v3/programs/{programId}/archive",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "program.createdDate" },
            { path: "program.updatedDate" },
            { path: "program.stateUpdatedDate" },
            { path: "program.statusUpdatedDate" },
            { path: "program.description.image.urlExpirationDate" },
            { path: "program.description.video.urlExpirationDate" },
            {
              path: "program.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "program.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "program.description.video.posters.urlExpirationDate" },
            { path: "program.rewards.certificate.connectedDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "program.description.image.focalPoint.x" },
            { path: "program.description.image.focalPoint.y" },
            {
              path: "program.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "program.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "program.description.video.posters.focalPoint.x" },
            { path: "program.description.video.posters.focalPoint.y" },
            {
              path: "program.videoStepsSettings.requiredCompletionPercentage"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __archiveProgram;
}
function bulkUpdatePrograms(payload) {
  function __bulkUpdatePrograms({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "programs.fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "programs.program.createdDate" },
          { path: "programs.program.updatedDate" },
          { path: "programs.program.stateUpdatedDate" },
          { path: "programs.program.statusUpdatedDate" },
          { path: "programs.program.description.image.urlExpirationDate" },
          { path: "programs.program.description.video.urlExpirationDate" },
          {
            path: "programs.program.description.video.resolutions.urlExpirationDate"
          },
          {
            path: "programs.program.description.video.resolutions.poster.urlExpirationDate"
          },
          {
            path: "programs.program.description.video.posters.urlExpirationDate"
          },
          { path: "programs.program.rewards.certificate.connectedDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "programs.program.description.image.focalPoint.x" },
          { path: "programs.program.description.image.focalPoint.y" },
          {
            path: "programs.program.description.video.resolutions.poster.focalPoint.x"
          },
          {
            path: "programs.program.description.video.resolutions.poster.focalPoint.y"
          },
          { path: "programs.program.description.video.posters.focalPoint.x" },
          { path: "programs.program.description.video.posters.focalPoint.y" },
          {
            path: "programs.program.videoStepsSettings.requiredCompletionPercentage"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.online_programs.v3.program",
      method: "POST",
      methodFqn: "wix.onlineprograms.programs.v3.ProgramsService.BulkUpdatePrograms",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsProgramsV3ProgramsServiceUrl({
        protoPath: "/v3/bulk/programs/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.stateUpdatedDate" },
            { path: "results.item.statusUpdatedDate" },
            { path: "results.item.description.image.urlExpirationDate" },
            { path: "results.item.description.video.urlExpirationDate" },
            {
              path: "results.item.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.description.video.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.description.video.posters.urlExpirationDate"
            },
            { path: "results.item.rewards.certificate.connectedDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.description.image.focalPoint.x" },
            { path: "results.item.description.image.focalPoint.y" },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "results.item.description.video.posters.focalPoint.x" },
            { path: "results.item.description.video.posters.focalPoint.y" },
            {
              path: "results.item.videoStepsSettings.requiredCompletionPercentage"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdatePrograms;
}
function countPrograms(payload) {
  function __countPrograms({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.program",
      method: "POST",
      methodFqn: "wix.onlineprograms.programs.v3.ProgramsService.CountPrograms",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsProgramsV3ProgramsServiceUrl({
        protoPath: "/v3/programs/count",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __countPrograms;
}
function listSamplePrograms(payload) {
  function __listSamplePrograms({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.program",
      method: "GET",
      methodFqn: "wix.onlineprograms.programs.v3.ProgramsService.ListSamplePrograms",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsProgramsV3ProgramsServiceUrl({
        protoPath: "/v3/programs/samples",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "programs.program.createdDate" },
            { path: "programs.program.updatedDate" },
            { path: "programs.program.stateUpdatedDate" },
            { path: "programs.program.statusUpdatedDate" },
            { path: "programs.program.description.image.urlExpirationDate" },
            { path: "programs.program.description.video.urlExpirationDate" },
            {
              path: "programs.program.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "programs.program.description.video.resolutions.poster.urlExpirationDate"
            },
            {
              path: "programs.program.description.video.posters.urlExpirationDate"
            },
            { path: "programs.program.rewards.certificate.connectedDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "programs.program.description.image.focalPoint.x" },
            { path: "programs.program.description.image.focalPoint.y" },
            {
              path: "programs.program.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "programs.program.description.video.resolutions.poster.focalPoint.y"
            },
            {
              path: "programs.program.description.video.posters.focalPoint.x"
            },
            {
              path: "programs.program.description.video.posters.focalPoint.y"
            },
            {
              path: "programs.program.videoStepsSettings.requiredCompletionPercentage"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listSamplePrograms;
}

// src/online-programs-v3-program-programs.universal.ts
var import_image = require("@wix/sdk-runtime/transformations/image");
var import_image2 = require("@wix/sdk-runtime/transformations/image");
var import_video_v2 = require("@wix/sdk-runtime/transformations/video-v2");
var import_video_v22 = require("@wix/sdk-runtime/transformations/video-v2");
var import_page_url = require("@wix/sdk-runtime/transformations/page-url");
var import_page_url2 = require("@wix/sdk-runtime/transformations/page-url");
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var V3State = /* @__PURE__ */ ((V3State2) => {
  V3State2["DRAFT"] = "DRAFT";
  V3State2["PUBLISHED"] = "PUBLISHED";
  V3State2["ENDED"] = "ENDED";
  V3State2["ARCHIVED"] = "ARCHIVED";
  return V3State2;
})(V3State || {});
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["UNKNOWN_STATUS"] = "UNKNOWN_STATUS";
  Status2["DRAFT"] = "DRAFT";
  Status2["PUBLISHED"] = "PUBLISHED";
  Status2["ENDED"] = "ENDED";
  Status2["ARCHIVED"] = "ARCHIVED";
  return Status2;
})(Status || {});
var RestrictionsAccessType = /* @__PURE__ */ ((RestrictionsAccessType2) => {
  RestrictionsAccessType2["UNKNOWN_ACCESS_TYPE"] = "UNKNOWN_ACCESS_TYPE";
  RestrictionsAccessType2["PUBLIC"] = "PUBLIC";
  RestrictionsAccessType2["PRIVATE"] = "PRIVATE";
  RestrictionsAccessType2["SECRET"] = "SECRET";
  return RestrictionsAccessType2;
})(RestrictionsAccessType || {});
var Trigger = /* @__PURE__ */ ((Trigger2) => {
  Trigger2["JOINED_TO_PROGRAM"] = "JOINED_TO_PROGRAM";
  Trigger2["STEP_COMPLETED"] = "STEP_COMPLETED";
  Trigger2["ALL_STEPS_COMPLETED"] = "ALL_STEPS_COMPLETED";
  return Trigger2;
})(Trigger || {});
var AccessType = /* @__PURE__ */ ((AccessType2) => {
  AccessType2["PUBLIC"] = "PUBLIC";
  AccessType2["PRIVATE"] = "PRIVATE";
  AccessType2["SECRET"] = "SECRET";
  return AccessType2;
})(AccessType || {});
var Enum = /* @__PURE__ */ ((Enum2) => {
  Enum2["UNKNOWN"] = "UNKNOWN";
  Enum2["PROTECTED"] = "PROTECTED";
  Enum2["PRIVATE"] = "PRIVATE";
  Enum2["PUBLIC"] = "PUBLIC";
  Enum2["UNCHANGED"] = "UNCHANGED";
  Enum2["GROUP_PROTECTED"] = "GROUP_PROTECTED";
  return Enum2;
})(Enum || {});
var BulkActionType = /* @__PURE__ */ ((BulkActionType2) => {
  BulkActionType2["UNKNOWN_ACTION_TYPE"] = "UNKNOWN_ACTION_TYPE";
  BulkActionType2["INSERT"] = "INSERT";
  BulkActionType2["UPDATE"] = "UPDATE";
  BulkActionType2["DELETE"] = "DELETE";
  return BulkActionType2;
})(BulkActionType || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var SortType = /* @__PURE__ */ ((SortType2) => {
  SortType2["COUNT"] = "COUNT";
  SortType2["VALUE"] = "VALUE";
  return SortType2;
})(SortType || {});
var SortDirection = /* @__PURE__ */ ((SortDirection2) => {
  SortDirection2["DESC"] = "DESC";
  SortDirection2["ASC"] = "ASC";
  return SortDirection2;
})(SortDirection || {});
var MissingValues = /* @__PURE__ */ ((MissingValues2) => {
  MissingValues2["EXCLUDE"] = "EXCLUDE";
  MissingValues2["INCLUDE"] = "INCLUDE";
  return MissingValues2;
})(MissingValues || {});
var ScalarType = /* @__PURE__ */ ((ScalarType2) => {
  ScalarType2["UNKNOWN_SCALAR_TYPE"] = "UNKNOWN_SCALAR_TYPE";
  ScalarType2["COUNT_DISTINCT"] = "COUNT_DISTINCT";
  ScalarType2["MIN"] = "MIN";
  ScalarType2["MAX"] = "MAX";
  return ScalarType2;
})(ScalarType || {});
var NestedAggregationType = /* @__PURE__ */ ((NestedAggregationType2) => {
  NestedAggregationType2["UNKNOWN_AGGREGATION_TYPE"] = "UNKNOWN_AGGREGATION_TYPE";
  NestedAggregationType2["VALUE"] = "VALUE";
  NestedAggregationType2["RANGE"] = "RANGE";
  NestedAggregationType2["SCALAR"] = "SCALAR";
  NestedAggregationType2["DATE_HISTOGRAM"] = "DATE_HISTOGRAM";
  return NestedAggregationType2;
})(NestedAggregationType || {});
var Interval = /* @__PURE__ */ ((Interval2) => {
  Interval2["UNKNOWN_INTERVAL"] = "UNKNOWN_INTERVAL";
  Interval2["YEAR"] = "YEAR";
  Interval2["MONTH"] = "MONTH";
  Interval2["WEEK"] = "WEEK";
  Interval2["DAY"] = "DAY";
  Interval2["HOUR"] = "HOUR";
  Interval2["MINUTE"] = "MINUTE";
  Interval2["SECOND"] = "SECOND";
  return Interval2;
})(Interval || {});
var AggregationType = /* @__PURE__ */ ((AggregationType2) => {
  AggregationType2["UNKNOWN_AGGREGATION_TYPE"] = "UNKNOWN_AGGREGATION_TYPE";
  AggregationType2["VALUE"] = "VALUE";
  AggregationType2["RANGE"] = "RANGE";
  AggregationType2["SCALAR"] = "SCALAR";
  AggregationType2["DATE_HISTOGRAM"] = "DATE_HISTOGRAM";
  AggregationType2["NESTED"] = "NESTED";
  return AggregationType2;
})(AggregationType || {});
var Mode = /* @__PURE__ */ ((Mode2) => {
  Mode2["OR"] = "OR";
  Mode2["AND"] = "AND";
  return Mode2;
})(Mode || {});
var State = /* @__PURE__ */ ((State2) => {
  State2["UNKNOWN"] = "UNKNOWN";
  State2["ENABLED"] = "ENABLED";
  State2["DISABLED"] = "DISABLED";
  State2["PENDING"] = "PENDING";
  State2["DEMO"] = "DEMO";
  return State2;
})(State || {});
var SiteCreatedContext = /* @__PURE__ */ ((SiteCreatedContext2) => {
  SiteCreatedContext2["OTHER"] = "OTHER";
  SiteCreatedContext2["FROM_TEMPLATE"] = "FROM_TEMPLATE";
  SiteCreatedContext2["DUPLICATE_BY_SITE_TRANSFER"] = "DUPLICATE_BY_SITE_TRANSFER";
  SiteCreatedContext2["DUPLICATE"] = "DUPLICATE";
  SiteCreatedContext2["OLD_SITE_TRANSFER"] = "OLD_SITE_TRANSFER";
  SiteCreatedContext2["FLASH"] = "FLASH";
  return SiteCreatedContext2;
})(SiteCreatedContext || {});
var Namespace = /* @__PURE__ */ ((Namespace2) => {
  Namespace2["UNKNOWN_NAMESPACE"] = "UNKNOWN_NAMESPACE";
  Namespace2["WIX"] = "WIX";
  Namespace2["SHOUT_OUT"] = "SHOUT_OUT";
  Namespace2["ALBUMS"] = "ALBUMS";
  Namespace2["WIX_STORES_TEST_DRIVE"] = "WIX_STORES_TEST_DRIVE";
  Namespace2["HOTELS"] = "HOTELS";
  Namespace2["CLUBS"] = "CLUBS";
  Namespace2["ONBOARDING_DRAFT"] = "ONBOARDING_DRAFT";
  Namespace2["DEV_SITE"] = "DEV_SITE";
  Namespace2["LOGOS"] = "LOGOS";
  Namespace2["VIDEO_MAKER"] = "VIDEO_MAKER";
  Namespace2["PARTNER_DASHBOARD"] = "PARTNER_DASHBOARD";
  Namespace2["DEV_CENTER_COMPANY"] = "DEV_CENTER_COMPANY";
  Namespace2["HTML_DRAFT"] = "HTML_DRAFT";
  Namespace2["SITELESS_BUSINESS"] = "SITELESS_BUSINESS";
  Namespace2["CREATOR_ECONOMY"] = "CREATOR_ECONOMY";
  Namespace2["DASHBOARD_FIRST"] = "DASHBOARD_FIRST";
  Namespace2["ANYWHERE"] = "ANYWHERE";
  Namespace2["HEADLESS"] = "HEADLESS";
  Namespace2["ACCOUNT_MASTER_CMS"] = "ACCOUNT_MASTER_CMS";
  Namespace2["RISE"] = "RISE";
  Namespace2["BRANDED_FIRST"] = "BRANDED_FIRST";
  Namespace2["NOWNIA"] = "NOWNIA";
  Namespace2["UGC_TEMPLATE"] = "UGC_TEMPLATE";
  Namespace2["CODUX"] = "CODUX";
  Namespace2["MEDIA_DESIGN_CREATOR"] = "MEDIA_DESIGN_CREATOR";
  Namespace2["SHARED_BLOG_ENTERPRISE"] = "SHARED_BLOG_ENTERPRISE";
  Namespace2["STANDALONE_FORMS"] = "STANDALONE_FORMS";
  Namespace2["STANDALONE_EVENTS"] = "STANDALONE_EVENTS";
  Namespace2["MIMIR"] = "MIMIR";
  Namespace2["TWINS"] = "TWINS";
  Namespace2["NANO"] = "NANO";
  return Namespace2;
})(Namespace || {});
var DeleteStatus = /* @__PURE__ */ ((DeleteStatus2) => {
  DeleteStatus2["UNKNOWN"] = "UNKNOWN";
  DeleteStatus2["TRASH"] = "TRASH";
  DeleteStatus2["DELETED"] = "DELETED";
  DeleteStatus2["PENDING_PURGE"] = "PENDING_PURGE";
  DeleteStatus2["PURGED_EXTERNALLY"] = "PURGED_EXTERNALLY";
  return DeleteStatus2;
})(DeleteStatus || {});
var SearchIndexingNotificationState = /* @__PURE__ */ ((SearchIndexingNotificationState2) => {
  SearchIndexingNotificationState2["Unknown"] = "Unknown";
  SearchIndexingNotificationState2["Off"] = "Off";
  SearchIndexingNotificationState2["On"] = "On";
  return SearchIndexingNotificationState2;
})(SearchIndexingNotificationState || {});
var PlacementType = /* @__PURE__ */ ((PlacementType2) => {
  PlacementType2["BEFORE"] = "BEFORE";
  PlacementType2["AFTER"] = "AFTER";
  PlacementType2["REPLACE"] = "REPLACE";
  return PlacementType2;
})(PlacementType || {});
var DayOfWeek = /* @__PURE__ */ ((DayOfWeek2) => {
  DayOfWeek2["MONDAY"] = "MONDAY";
  DayOfWeek2["TUESDAY"] = "TUESDAY";
  DayOfWeek2["WEDNESDAY"] = "WEDNESDAY";
  DayOfWeek2["THURSDAY"] = "THURSDAY";
  DayOfWeek2["FRIDAY"] = "FRIDAY";
  DayOfWeek2["SATURDAY"] = "SATURDAY";
  DayOfWeek2["SUNDAY"] = "SUNDAY";
  return DayOfWeek2;
})(DayOfWeek || {});
var ResolutionMethod = /* @__PURE__ */ ((ResolutionMethod2) => {
  ResolutionMethod2["QUERY_PARAM"] = "QUERY_PARAM";
  ResolutionMethod2["SUBDOMAIN"] = "SUBDOMAIN";
  ResolutionMethod2["SUBDIRECTORY"] = "SUBDIRECTORY";
  return ResolutionMethod2;
})(ResolutionMethod || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createProgram2(program) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ program }),
    [
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [{ path: "program.description.image" }]
      },
      {
        transformFn: import_video_v2.transformSDKVideoV2ToRESTVideoV2,
        paths: [{ path: "program.description.video" }]
      },
      {
        transformFn: import_page_url.transformSDKPageURLToRESTPageURL,
        paths: [{ path: "program.seo.url" }]
      }
    ]
  );
  const reqOpts = createProgram(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "program.description.image" }]
        },
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "program.description.video" }]
        },
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [{ path: "program.seo.url" }]
        }
      ])
    )?.program;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { program: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["program"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function duplicateProgram2(programId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    programId
  });
  const reqOpts = duplicateProgram(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "program.description.image" }]
        },
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "program.description.video" }]
        },
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [{ path: "program.seo.url" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { programId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["programId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getProgram2(programId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    programId
  });
  const reqOpts = getProgram(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "program.description.image" }]
        },
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "program.description.video" }]
        },
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [{ path: "program.seo.url" }]
        }
      ])
    )?.program;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { programId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["programId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateProgram2(_id, program) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ program: { ...program, id: _id } }),
    [
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [{ path: "program.description.image" }]
      },
      {
        transformFn: import_video_v2.transformSDKVideoV2ToRESTVideoV2,
        paths: [{ path: "program.description.video" }]
      },
      {
        transformFn: import_page_url.transformSDKPageURLToRESTPageURL,
        paths: [{ path: "program.seo.url" }]
      }
    ]
  );
  const reqOpts = updateProgram(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "program.description.image" }]
        },
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "program.description.video" }]
        },
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [{ path: "program.seo.url" }]
        }
      ])
    )?.program;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { program: "$[1]" },
        explicitPathsToArguments: { "program.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "program"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteProgram2(programId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    programId
  });
  const reqOpts = deleteProgram(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { programId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["programId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryPrograms2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryPrograms(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [
          {
            transformFn: import_image2.transformRESTImageToSDKImage,
            paths: [{ path: "programs.description.image" }]
          },
          {
            transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
            paths: [{ path: "programs.description.video" }]
          },
          {
            transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
            paths: [{ path: "programs.seo.url" }]
          }
        ])
      );
      return {
        items: transformedData?.programs,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function searchPrograms2(search) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ search });
  const reqOpts = searchPrograms(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "programs.description.image" }]
        },
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "programs.description.video" }]
        },
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [{ path: "programs.seo.url" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { search: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["search"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function publishProgram2(programId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    programId
  });
  const reqOpts = publishProgram(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "program.description.image" }]
        },
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "program.description.video" }]
        },
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [{ path: "program.seo.url" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { programId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["programId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function endProgram2(programId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    programId
  });
  const reqOpts = endProgram(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "program.description.image" }]
        },
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "program.description.video" }]
        },
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [{ path: "program.seo.url" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { programId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["programId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function archiveProgram2(programId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    programId
  });
  const reqOpts = archiveProgram(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "program.description.image" }]
        },
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "program.description.video" }]
        },
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [{ path: "program.seo.url" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { programId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["programId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdatePrograms2(programs, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      programs,
      returnEntity: options?.returnEntity
    }),
    [
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [{ path: "programs.program.description.image" }]
      },
      {
        transformFn: import_video_v2.transformSDKVideoV2ToRESTVideoV2,
        paths: [{ path: "programs.program.description.video" }]
      },
      {
        transformFn: import_page_url.transformSDKPageURLToRESTPageURL,
        paths: [{ path: "programs.program.seo.url" }]
      }
    ]
  );
  const reqOpts = bulkUpdatePrograms(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "results.item.description.image" }]
        },
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "results.item.description.video" }]
        },
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [{ path: "results.item.seo.url" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          programs: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["programs", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function countPrograms2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    filter: options?.filter
  });
  const reqOpts = countPrograms(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { filter: "$[0].filter" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listSamplePrograms2() {
  const { httpClient, sideEffects } = arguments[0];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({});
  const reqOpts = listSamplePrograms(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "programs.program.description.image" }]
        },
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "programs.program.description.video" }]
        },
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [{ path: "programs.program.seo.url" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {},
        singleArgumentUnchanged: false
      },
      []
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AccessType,
  AggregationType,
  BulkActionType,
  DayOfWeek,
  DeleteStatus,
  Enum,
  Interval,
  MissingValues,
  Mode,
  Namespace,
  NestedAggregationType,
  PlacementType,
  ResolutionMethod,
  RestrictionsAccessType,
  ScalarType,
  SearchIndexingNotificationState,
  SiteCreatedContext,
  SortDirection,
  SortOrder,
  SortType,
  State,
  Status,
  Trigger,
  V3State,
  WebhookIdentityType,
  archiveProgram,
  bulkUpdatePrograms,
  countPrograms,
  createProgram,
  deleteProgram,
  duplicateProgram,
  endProgram,
  getProgram,
  listSamplePrograms,
  publishProgram,
  queryPrograms,
  searchPrograms,
  updateProgram
});
