// @wix/auto_sdk_members_member-privacy-settings/build/es/meta.mjs
// src/members-v1-member-privacy-settings-member-privacy-settings.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressMembersMemberprivacysettingsMemberPrivacySettingsServiceUrl(opts) {
  const domainToMappings = {
    "editor.wixapps.net": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/members/v1/privacy-settings",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    _: [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_member-privacy-settings";
function getMemberPrivacySettings(payload) {
  function __getMemberPrivacySettings({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_privacy_settings",
      method: "GET",
      methodFqn: "com.wixpress.members.memberprivacysettings.MemberPrivacySettingsService.GetMemberPrivacySettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembersMemberprivacysettingsMemberPrivacySettingsServiceUrl(
        { protoPath: "/v1/privacy-settings", data: payload, host }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "memberPrivacySettings.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __getMemberPrivacySettings;
}
function setMemberPrivacySettings(payload) {
  function __setMemberPrivacySettings({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "memberPrivacySettings.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.members.v1.member_privacy_settings",
      method: "POST",
      methodFqn: "com.wixpress.members.memberprivacysettings.MemberPrivacySettingsService.SetMemberPrivacySettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembersMemberprivacysettingsMemberPrivacySettingsServiceUrl(
        { protoPath: "/v1/privacy-settings", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "memberPrivacySettings.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __setMemberPrivacySettings;
}

// src/members-v1-member-privacy-settings-member-privacy-settings.types.ts
var Visibility = /* @__PURE__ */ ((Visibility2) => {
  Visibility2["UNKNOWN"] = "UNKNOWN";
  Visibility2["VISIBLE_TO_MEMBERS_ONLY"] = "VISIBLE_TO_MEMBERS_ONLY";
  Visibility2["VISIBLE_TO_ALL"] = "VISIBLE_TO_ALL";
  return Visibility2;
})(Visibility || {});
var Candidates = /* @__PURE__ */ ((Candidates2) => {
  Candidates2["UNKNOWN"] = "UNKNOWN";
  Candidates2["NO_ONE"] = "NO_ONE";
  Candidates2["ANYONE"] = "ANYONE";
  return Candidates2;
})(Candidates || {});
var State = /* @__PURE__ */ ((State2) => {
  State2["UNKNOWN"] = "UNKNOWN";
  State2["ENABLED"] = "ENABLED";
  State2["DISABLED"] = "DISABLED";
  State2["PENDING"] = "PENDING";
  State2["DEMO"] = "DEMO";
  return State2;
})(State || {});
var SiteCreatedContext = /* @__PURE__ */ ((SiteCreatedContext2) => {
  SiteCreatedContext2["OTHER"] = "OTHER";
  SiteCreatedContext2["FROM_TEMPLATE"] = "FROM_TEMPLATE";
  SiteCreatedContext2["DUPLICATE_BY_SITE_TRANSFER"] = "DUPLICATE_BY_SITE_TRANSFER";
  SiteCreatedContext2["DUPLICATE"] = "DUPLICATE";
  SiteCreatedContext2["OLD_SITE_TRANSFER"] = "OLD_SITE_TRANSFER";
  SiteCreatedContext2["FLASH"] = "FLASH";
  return SiteCreatedContext2;
})(SiteCreatedContext || {});
var Namespace = /* @__PURE__ */ ((Namespace2) => {
  Namespace2["UNKNOWN_NAMESPACE"] = "UNKNOWN_NAMESPACE";
  Namespace2["WIX"] = "WIX";
  Namespace2["SHOUT_OUT"] = "SHOUT_OUT";
  Namespace2["ALBUMS"] = "ALBUMS";
  Namespace2["WIX_STORES_TEST_DRIVE"] = "WIX_STORES_TEST_DRIVE";
  Namespace2["HOTELS"] = "HOTELS";
  Namespace2["CLUBS"] = "CLUBS";
  Namespace2["ONBOARDING_DRAFT"] = "ONBOARDING_DRAFT";
  Namespace2["DEV_SITE"] = "DEV_SITE";
  Namespace2["LOGOS"] = "LOGOS";
  Namespace2["VIDEO_MAKER"] = "VIDEO_MAKER";
  Namespace2["PARTNER_DASHBOARD"] = "PARTNER_DASHBOARD";
  Namespace2["DEV_CENTER_COMPANY"] = "DEV_CENTER_COMPANY";
  Namespace2["HTML_DRAFT"] = "HTML_DRAFT";
  Namespace2["SITELESS_BUSINESS"] = "SITELESS_BUSINESS";
  Namespace2["CREATOR_ECONOMY"] = "CREATOR_ECONOMY";
  Namespace2["DASHBOARD_FIRST"] = "DASHBOARD_FIRST";
  Namespace2["ANYWHERE"] = "ANYWHERE";
  Namespace2["HEADLESS"] = "HEADLESS";
  Namespace2["ACCOUNT_MASTER_CMS"] = "ACCOUNT_MASTER_CMS";
  Namespace2["RISE"] = "RISE";
  Namespace2["BRANDED_FIRST"] = "BRANDED_FIRST";
  Namespace2["NOWNIA"] = "NOWNIA";
  Namespace2["UGC_TEMPLATE"] = "UGC_TEMPLATE";
  Namespace2["CODUX"] = "CODUX";
  Namespace2["MEDIA_DESIGN_CREATOR"] = "MEDIA_DESIGN_CREATOR";
  Namespace2["SHARED_BLOG_ENTERPRISE"] = "SHARED_BLOG_ENTERPRISE";
  Namespace2["STANDALONE_FORMS"] = "STANDALONE_FORMS";
  Namespace2["STANDALONE_EVENTS"] = "STANDALONE_EVENTS";
  Namespace2["MIMIR"] = "MIMIR";
  Namespace2["TWINS"] = "TWINS";
  Namespace2["NANO"] = "NANO";
  return Namespace2;
})(Namespace || {});
var DeleteStatus = /* @__PURE__ */ ((DeleteStatus2) => {
  DeleteStatus2["UNKNOWN"] = "UNKNOWN";
  DeleteStatus2["TRASH"] = "TRASH";
  DeleteStatus2["DELETED"] = "DELETED";
  DeleteStatus2["PENDING_PURGE"] = "PENDING_PURGE";
  DeleteStatus2["PURGED_EXTERNALLY"] = "PURGED_EXTERNALLY";
  return DeleteStatus2;
})(DeleteStatus || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/members-v1-member-privacy-settings-member-privacy-settings.meta.ts
function getMemberPrivacySettings2() {
  const payload = {};
  const getRequestOptions = getMemberPrivacySettings(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/privacy-settings",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function setMemberPrivacySettings2() {
  const payload = {};
  const getRequestOptions = setMemberPrivacySettings(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/privacy-settings",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  Candidates as CandidatesOriginal,
  DeleteStatus as DeleteStatusOriginal,
  Namespace as NamespaceOriginal,
  SiteCreatedContext as SiteCreatedContextOriginal,
  State as StateOriginal,
  Visibility as VisibilityOriginal,
  WebhookIdentityType as WebhookIdentityTypeOriginal,
  getMemberPrivacySettings2 as getMemberPrivacySettings,
  setMemberPrivacySettings2 as setMemberPrivacySettings
};
