// @wix/unidriver-playwright/dist/esm/stories/TestApp.stories.js
import { TestApp, itemCreator } from '@wix/unidriver-common/testing';
const meta = {
    title: 'TestApp',
    component: TestApp,
};
export default meta;
export const Base = {
    name: 'base',
    args: { items: [] },
};
export const BaseEmptyInitialText = {
    name: 'base-empty-initial-text',
    args: { items: [], initialText: '' },
};
export const WithInitialText = {
    name: 'with-initial-text',
    args: { items: [], initialText: 'hello' },
};
export const EmptyLabel = {
    name: 'empty-label',
    args: { items: [itemCreator({ label: '' })], initialText: '' },
};
export const Label = {
    name: 'label',
    args: {
        items: [itemCreator({ label: 'hello' })],
        initialText: '',
    },
};
export const WithItem = {
    name: 'with-item',
    args: { items: [itemCreator({ label: 'Bob' })] },
};
export const WithTwoItems = {
    name: 'with-two-items',
    args: {
        items: [itemCreator({ label: 'Bob', completed: false }), itemCreator({ label: 'David', completed: true })],
    },
};
export const DisabledInput = {
    name: 'disabled-input',
    args: { items: [], initialText: 'other text', inputDisabled: true },
};
export const ReadonlyInput = {
    name: 'readonly-input',
    args: { items: [], initialText: 'other text', inputReadOnly: true },
};
export const ItemsWithIds = {
    name: 'items-with-ids',
    args: {
        items: [
            itemCreator({ label: 'Bob', completed: false, id: 'bob' }),
            itemCreator({ label: 'Alice', completed: false, id: 'alice' }),
        ],
    },
};
export const ItemWithEmptyId = {
    name: 'item-with-empty-id',
    args: {
        items: [itemCreator({ label: 'Bob', completed: false, id: '' })],
    },
};
export const WithThreeItems = {
    name: 'with-three-items',
    args: {
        items: [
            itemCreator({ label: 'Bob', completed: false }),
            itemCreator({ id: 'david', label: 'David' }),
            itemCreator({ id: 'jacob', label: 'Jacob', completed: true }),
        ],
    },
};
export const WithThreeItemsNotCompleted = {
    name: 'with-three-items-not-completed',
    args: {
        items: [
            itemCreator({ label: 'Bob', completed: false }),
            itemCreator({ label: 'David' }),
            itemCreator({ label: 'Jacob', completed: false }),
        ],
    },
};
export const WithThreeItemsAllCompleted = {
    name: 'with-three-items-all-completed',
    args: {
        items: [
            itemCreator({ label: 'Bob', completed: true }),
            itemCreator({ label: 'David', completed: true }),
            itemCreator({ label: 'Jacob', completed: true }),
        ],
    },
};
export const Get = {
    name: 'get',
    args: {
        items: [itemCreator({ label: 'Bob' })],
    },
};
export const GetAll = {
    name: 'get-all',
    args: {
        items: [itemCreator({ label: 'Bob' }), itemCreator({ label: 'David' })],
    },
};
