// pro-gallery/dist/cjs/components/gallery/typeValidator/typeErrorsUI.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
function default_1(typeErrors) {
    window.typeErrors = typeErrors;
    return (react_1.default.createElement("div", null,
        react_1.default.createElement("ol", null, typeErrors.map((errObject, i) => (react_1.default.createElement(Error, { errObject: errObject, key: i })))),
        JSON.stringify(typeErrors, null, 4)));
}
exports.default = default_1;
function Error({ errObject }) {
    return (react_1.default.createElement("li", null,
        react_1.default.createElement(ErrorDescription, { errObject: errObject })));
}
class ErrorDescription extends react_1.default.Component {
    constructor(props) {
        super(props);
        this.state = {
            showFull: false,
        };
        this.toggle = this.toggle.bind(this); // PUKE
    }
    render() {
        const CompToRender = this.state.showFull ? FullError : Text;
        return (react_1.default.createElement("a", { onClick: this.toggle },
            react_1.default.createElement(CompToRender, { errObject: this.props.errObject })));
    }
    toggle() {
        this.setState({ showFull: !this.state.showFull });
    }
}
function FullError({ errObject }) {
    return (react_1.default.createElement("ul", null, Object.entries(errObject).map((entry, i) => (react_1.default.createElement(Item, { entry: entry, key: i })))));
}
function Item({ entry }) {
    const key = entry[0];
    const value = entry[1];
    const valueText = typeof value === 'string' ? value : JSON.stringify(value, null, 4);
    return react_1.default.createElement("li", null, `${key}: ${valueText}`);
}
function Text({ errObject }) {
    return errObject.message;
}
