// @wix/auto_sdk_members_user-member/build/es/meta.mjs
// src/members-usermembers-v1-user-member-user-member.http.ts
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixMembersUsermembersV1UserMembersUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/user-members-api",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/members/v1/user-members",
        destPath: "/v1/user-members"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_user-member";
function queryUserMembers(payload) {
  function __queryUserMembers({ host }) {
    const metadata = {
      entityFqdn: "wix.members.usermembers.v1.user_member",
      method: "POST",
      methodFqn: "wix.members.usermembers.v1.UserMembers.QueryUserMembers",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixMembersUsermembersV1UserMembersUrl({
        protoPath: "/v1/user-members/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "userMembers.createdDate" },
            { path: "userMembers.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryUserMembers;
}

// src/members-usermembers-v1-user-member-user-member.types.ts
var PrivacyStatusStatus = /* @__PURE__ */ ((PrivacyStatusStatus2) => {
  PrivacyStatusStatus2["UNKNOWN"] = "UNKNOWN";
  PrivacyStatusStatus2["PRIVATE"] = "PRIVATE";
  PrivacyStatusStatus2["PUBLIC"] = "PUBLIC";
  return PrivacyStatusStatus2;
})(PrivacyStatusStatus || {});
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["UNKNOWN"] = "UNKNOWN";
  Status2["PENDING"] = "PENDING";
  Status2["APPROVED"] = "APPROVED";
  Status2["BLOCKED"] = "BLOCKED";
  Status2["OFFLINE"] = "OFFLINE";
  return Status2;
})(Status || {});
var ActivityStatusStatus = /* @__PURE__ */ ((ActivityStatusStatus2) => {
  ActivityStatusStatus2["UNKNOWN"] = "UNKNOWN";
  ActivityStatusStatus2["ACTIVE"] = "ACTIVE";
  ActivityStatusStatus2["MUTED"] = "MUTED";
  return ActivityStatusStatus2;
})(ActivityStatusStatus || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var State = /* @__PURE__ */ ((State2) => {
  State2["UNKNOWN"] = "UNKNOWN";
  State2["ENABLED"] = "ENABLED";
  State2["DISABLED"] = "DISABLED";
  State2["PENDING"] = "PENDING";
  State2["DEMO"] = "DEMO";
  return State2;
})(State || {});
var SiteCreatedContext = /* @__PURE__ */ ((SiteCreatedContext2) => {
  SiteCreatedContext2["OTHER"] = "OTHER";
  SiteCreatedContext2["FROM_TEMPLATE"] = "FROM_TEMPLATE";
  SiteCreatedContext2["DUPLICATE_BY_SITE_TRANSFER"] = "DUPLICATE_BY_SITE_TRANSFER";
  SiteCreatedContext2["DUPLICATE"] = "DUPLICATE";
  SiteCreatedContext2["OLD_SITE_TRANSFER"] = "OLD_SITE_TRANSFER";
  SiteCreatedContext2["FLASH"] = "FLASH";
  return SiteCreatedContext2;
})(SiteCreatedContext || {});
var Namespace = /* @__PURE__ */ ((Namespace2) => {
  Namespace2["UNKNOWN_NAMESPACE"] = "UNKNOWN_NAMESPACE";
  Namespace2["WIX"] = "WIX";
  Namespace2["SHOUT_OUT"] = "SHOUT_OUT";
  Namespace2["ALBUMS"] = "ALBUMS";
  Namespace2["WIX_STORES_TEST_DRIVE"] = "WIX_STORES_TEST_DRIVE";
  Namespace2["HOTELS"] = "HOTELS";
  Namespace2["CLUBS"] = "CLUBS";
  Namespace2["ONBOARDING_DRAFT"] = "ONBOARDING_DRAFT";
  Namespace2["DEV_SITE"] = "DEV_SITE";
  Namespace2["LOGOS"] = "LOGOS";
  Namespace2["VIDEO_MAKER"] = "VIDEO_MAKER";
  Namespace2["PARTNER_DASHBOARD"] = "PARTNER_DASHBOARD";
  Namespace2["DEV_CENTER_COMPANY"] = "DEV_CENTER_COMPANY";
  Namespace2["HTML_DRAFT"] = "HTML_DRAFT";
  Namespace2["SITELESS_BUSINESS"] = "SITELESS_BUSINESS";
  Namespace2["CREATOR_ECONOMY"] = "CREATOR_ECONOMY";
  Namespace2["DASHBOARD_FIRST"] = "DASHBOARD_FIRST";
  Namespace2["ANYWHERE"] = "ANYWHERE";
  Namespace2["HEADLESS"] = "HEADLESS";
  Namespace2["ACCOUNT_MASTER_CMS"] = "ACCOUNT_MASTER_CMS";
  Namespace2["RISE"] = "RISE";
  Namespace2["BRANDED_FIRST"] = "BRANDED_FIRST";
  Namespace2["NOWNIA"] = "NOWNIA";
  Namespace2["UGC_TEMPLATE"] = "UGC_TEMPLATE";
  Namespace2["CODUX"] = "CODUX";
  Namespace2["MEDIA_DESIGN_CREATOR"] = "MEDIA_DESIGN_CREATOR";
  Namespace2["SHARED_BLOG_ENTERPRISE"] = "SHARED_BLOG_ENTERPRISE";
  Namespace2["STANDALONE_FORMS"] = "STANDALONE_FORMS";
  Namespace2["STANDALONE_EVENTS"] = "STANDALONE_EVENTS";
  Namespace2["MIMIR"] = "MIMIR";
  Namespace2["TWINS"] = "TWINS";
  Namespace2["NANO"] = "NANO";
  return Namespace2;
})(Namespace || {});
var DeleteStatus = /* @__PURE__ */ ((DeleteStatus2) => {
  DeleteStatus2["UNKNOWN"] = "UNKNOWN";
  DeleteStatus2["TRASH"] = "TRASH";
  DeleteStatus2["DELETED"] = "DELETED";
  DeleteStatus2["PENDING_PURGE"] = "PENDING_PURGE";
  DeleteStatus2["PURGED_EXTERNALLY"] = "PURGED_EXTERNALLY";
  return DeleteStatus2;
})(DeleteStatus || {});

// src/members-usermembers-v1-user-member-user-member.meta.ts
function queryUserMembers2() {
  const payload = {};
  const getRequestOptions = queryUserMembers(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/user-members/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  ActivityStatusStatus as ActivityStatusStatusOriginal,
  DeleteStatus as DeleteStatusOriginal,
  Namespace as NamespaceOriginal,
  PrivacyStatusStatus as PrivacyStatusStatusOriginal,
  SiteCreatedContext as SiteCreatedContextOriginal,
  SortOrder as SortOrderOriginal,
  State as StateOriginal,
  Status as StatusOriginal,
  queryUserMembers2 as queryUserMembers
};
