// @wix/ricos/dist/cjs/_shared/bi-context-provider/bi-context-provider.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var bi_context_provider_exports = {};
__export(bi_context_provider_exports, {
  BiContextProvider: () => BiContextProvider,
  useBiNext: () => useBiNext
});
module.exports = __toCommonJS(bi_context_provider_exports);
var import_react = __toESM(require("react"));
var import_ricos_context = require("./../../ricos-context/index.js");
const BiContext = import_react.default.createContext(null);
const BiContextProvider = ({
  children,
  biLogger
}) => {
  return /* @__PURE__ */ import_react.default.createElement(BiContext.Provider, { value: import_react.default.useMemo(() => ({ biLogger }), [biLogger]) }, children);
};
const useBiNext = (params) => {
  const devLogBi = (0, import_ricos_context.useDevLogBi)();
  if (devLogBi) {
    return (params2) => devLogBi({ ...params, ...params2 });
  }
  return useDebouncedLogBi(params);
};
const useDebouncedLogBi = (params, delay = 500) => {
  const { biLogger } = import_react.default.useContext(BiContext) ?? {};
  const timeoutsRef = import_react.default.useRef(/* @__PURE__ */ new Map());
  return import_react.default.useCallback(
    ({ dataHook, value }) => {
      if (!dataHook || !biLogger) {
        return;
      }
      if (timeoutsRef.current.has(dataHook)) {
        clearTimeout(timeoutsRef.current.get(dataHook));
      }
      const timeout = setTimeout(() => {
        biLogger.logV2?.({
          element_id: dataHook,
          value: value?.toString(),
          plugin_id: params?.pluginId,
          modal_id: params?.modalId,
          node_id: params?.nodeId
        });
        timeoutsRef.current.delete(dataHook);
      }, delay);
      timeoutsRef.current.set(dataHook, timeout);
    },
    [biLogger, delay, params?.pluginId, params?.modalId]
  );
};
