// pro-gallery-lib/dist/cjs/common/window/viewModeWrapper.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isDeviceTypeTouch = exports.isDeviceTypeDesktop = exports.isDeviceTypeTablet = exports.isDeviceTypeMobile = exports.isSEOMode = exports.isPreviewMode = exports.isEditMode = exports.isSiteMode = exports.viewModeWrapper = void 0;
const tslib_1 = require("tslib");
const viewMode_js_1 = tslib_1.__importDefault(require("../constants/viewMode.js"));
const deviceType_js_1 = tslib_1.__importDefault(require("../constants/deviceType.js"));
class ViewModeWrapper {
    constructor() {
        this.setViewMode = this.setViewMode.bind(this);
        this.isSiteMode = this.isSiteMode.bind(this);
        this.isEditMode = this.isEditMode.bind(this);
        this.isPreviewMode = this.isPreviewMode.bind(this);
        this.isSEOMode = this.isSEOMode.bind(this);
        this.setDeviceType = this.setDeviceType.bind(this);
        this.isDeviceTypeDesktop = this.isDeviceTypeDesktop.bind(this);
        this.isDeviceTypeMobile = this.isDeviceTypeMobile.bind(this);
        this.isDeviceTypeTablet = this.isDeviceTypeTablet.bind(this);
        this.isDeviceTypeTouch = this.isDeviceTypeTouch.bind(this);
        this._viewMode = viewMode_js_1.default.SITE;
        this._deviceType = deviceType_js_1.default.DESKTOP;
    }
    setViewMode(viewMode) {
        this._viewMode = viewMode;
    }
    setDeviceType(forceVal) {
        this._deviceType = forceVal;
    }
    isDeviceTypeMobile() {
        return this._deviceType === deviceType_js_1.default.MOBILE;
    }
    isDeviceTypeTablet() {
        return this._deviceType === deviceType_js_1.default.TABLET;
    }
    isDeviceTypeDesktop() {
        return this._deviceType === deviceType_js_1.default.DESKTOP;
    }
    isDeviceTypeTouch() {
        return this.isDeviceTypeMobile() || this.isDeviceTypeTablet();
    }
    isSiteMode() {
        return this._viewMode === viewMode_js_1.default.SITE;
    }
    isEditMode() {
        return this._viewMode === viewMode_js_1.default.EDIT;
    }
    isPreviewMode() {
        return this._viewMode === viewMode_js_1.default.PREVIEW;
    }
    isSEOMode() {
        return this._viewMode === viewMode_js_1.default.SEO;
    }
}
const viewModeWrapper = new ViewModeWrapper();
exports.viewModeWrapper = viewModeWrapper;
const isSiteMode = viewModeWrapper.isSiteMode;
exports.isSiteMode = isSiteMode;
const isEditMode = viewModeWrapper.isEditMode;
exports.isEditMode = isEditMode;
const isPreviewMode = viewModeWrapper.isPreviewMode;
exports.isPreviewMode = isPreviewMode;
const isSEOMode = viewModeWrapper.isSEOMode;
exports.isSEOMode = isSEOMode;
const isDeviceTypeMobile = viewModeWrapper.isDeviceTypeMobile;
exports.isDeviceTypeMobile = isDeviceTypeMobile;
const isDeviceTypeTablet = viewModeWrapper.isDeviceTypeTablet;
exports.isDeviceTypeTablet = isDeviceTypeTablet;
const isDeviceTypeDesktop = viewModeWrapper.isDeviceTypeDesktop;
exports.isDeviceTypeDesktop = isDeviceTypeDesktop;
const isDeviceTypeTouch = viewModeWrapper.isDeviceTypeTouch;
exports.isDeviceTypeTouch = isDeviceTypeTouch;
