// pro-gallery-lib/dist/cjs/common/window/window.mock.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.hydrateMockMap = exports.scrollMock = exports.dimsMock = void 0;
const noop = () => ({});
const width = 2560;
const height = 1440;
exports.dimsMock = {
    y: 0,
    x: 0,
    width,
    height,
    innerWidth: width,
    innerHeight: height,
    outerWidth: width,
    outerHeight: height,
    clientWidth: width,
    clientHeight: height,
};
const elem = Object.assign(Object.assign({}, exports.dimsMock), { getBoundingClientRect: () => exports.dimsMock });
const documentMock = Object.assign({ addEventListener: noop, removeEventListener: noop, createEvent: noop, getElementById: () => elem, getElementsByClassName: () => [elem], getElementsByTagName: () => [elem], querySelector: () => [elem], documentElement: elem, activeElement: elem, style: exports.dimsMock }, exports.dimsMock);
documentMock.body = documentMock;
const locationMock = {
    href: 'http://mock.wix.com/',
    protocol: 'http:',
    host: 'mock.wix.com',
    hostname: 'mock.wix.com',
    port: '',
    pathname: '/',
    search: '',
    hash: '',
};
exports.scrollMock = {
    scrollTop: 0,
    scrollY: 0,
};
const windowHydrateMock = Object.assign(Object.assign({}, exports.dimsMock), exports.scrollMock);
const windowMock = Object.assign({ isMock: true, isSSR: true, orientation: 0, devicePixelRatio: 1, addEventListener: noop, removeEventListener: noop, createEvent: noop, CustomEvent: noop, screen: exports.dimsMock, open: noop, petri: {}, search: {}, location: locationMock, postMessage: noop, requestAnimationFrame: noop, dispatchEvent: noop, document: documentMock, getComputedStyle: noop, localStorage: {}, frames: [] }, windowHydrateMock);
exports.hydrateMockMap = new Map(Object.keys(windowHydrateMock).map((key) => [key, windowHydrateMock[key]]));
windowMock.parent = windowMock;
exports.default = windowMock;
