// @wix/wix-data-items-common/dist/cjs/api/PipelineBuilder.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.pipelineBuilder = pipelineBuilder;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
/**
 * @builder
 */

class PipelineBuilderImpl {
  constructor() {
    (0, _defineProperty2.default)(this, "stages", []);
    (0, _defineProperty2.default)(this, "offsetPaging", void 0);
    (0, _defineProperty2.default)(this, "cursorPaging", void 0);
  }
  withStages(stages) {
    this.stages = stages;
    return this;
  }
  withPaging(paging) {
    const isCursor = 'cursor' in paging;
    this.cursorPaging = isCursor ? paging : undefined;
    this.offsetPaging = isCursor ? undefined : paging;
    return this;
  }
  build() {
    return {
      stages: this.stages.map(stage => stage.build()),
      ...(this.offsetPaging && {
        paging: this.offsetPaging
      }),
      ...(this.cursorPaging && {
        cursorPaging: this.cursorPaging
      })
    };
  }
}

/**
 * Builds an aggregation pipeline for use with [`aggregatePipeline()`](https://dev.wix.com/docs/sdk/backend-modules/data/items/aggregate-pipeline).
 *
 * Configure the pipeline by calling `pipelineBuilder()` and chaining [`PipelineBuilder`](https://dev.wix.com/docs/sdk/backend-modules/data/items/pipeline-builder/introduction) methods. Then, specify the pipeline as the `pipeline` parameter when calling the [`aggregatePipeline()`](https://dev.wix.com/docs/sdk/backend-modules/data/items/aggregate-pipeline) method.
 *
 * > **Note**: You can only build an aggregation pipeline for collections [created in the CMS](https://support.wix.com/en/article/cms-formerly-content-manager-creating-a-collection) or with the [Data Collections API](https://dev.wix.com/docs/sdk/backend-modules/data/collections/introduction). They cannot be used on [Wix app collections](https://support.wix.com/en/article/cms-formerly-content-manager-working-with-wix-app-collections) or [external collections](https://dev.wix.com/docs/develop-websites/articles/databases/external-databases/overview/integrating-external-databases-with-your-wix-site).
 *
 * @public
 * @documentationMaturity preview
 * @returns PipelineBuilder object
 */
function pipelineBuilder() {
  return new PipelineBuilderImpl();
}
