// @wix/astro/build/dependencies/astro-auth/browser-runtime/setup.js
// src/runtime/client/setup.ts
import { resolveContext } from "@wix/sdk-runtime/context";

// src/runtime/client/utils.ts
import { headlessSite } from "@wix/headless-site";
import { createClient } from "@wix/sdk";
import { SiteSessionAuth } from "@wix/sdk/auth/site-session";
function getSessionClient(options) {
  return createClient({
    auth: SiteSessionAuth({
      tokens: getCookieAsJson("wixSession")?.tokens,
      ...options
    }),
    host: headlessSite.host()
  });
}
function getCookieAsJson(name) {
  const cookies = document.cookie.split("; ");
  const cookie = cookies.find((row) => row.startsWith(`${name}=`));
  if (cookie == null) {
    return null;
  }
  try {
    const jsonString = decodeURIComponent(cookie.split("=")[1] ?? "");
    const cookieValue = JSON.parse(jsonString);
    if (isCookieValue(cookieValue)) {
      return cookieValue;
    }
    throw new Error("Invalid cookie value");
  } catch {
    return null;
  }
}
function isCookieValue(value) {
  if (value == null || typeof value !== "object") {
    return false;
  }
  if (!("tokens" in value)) {
    return false;
  }
  const tokens = value.tokens;
  if (tokens == null || typeof tokens !== "object") {
    return false;
  }
  if (!("accessToken" in tokens) || !("refreshToken" in tokens)) {
    return false;
  }
  return true;
}

// src/runtime/client/setup.ts
function setup(options) {
  const context = resolveContext();
  if (context != null) {
    return;
  }
  if (!location.pathname.startsWith("/_wix/extensions/backoffice")) {
    getSessionClient(options).enableContext("global");
  }
}
export {
  setup
};
