// pro-gallery/dist/cjs/components/gallery/proGallery/galleryHelpers.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isGalleryInViewport = exports.Deferred = void 0;
const pro_gallery_lib_1 = require("pro-gallery-lib");
// Promise-wrapping class
class Deferred {
    constructor() {
        this.promise = new Promise((resolve, reject) => {
            this.reject = (...args) => {
                this.isPending = false;
                reject(...args);
            };
            this.resolve = (...args) => {
                this.isPending = false;
                resolve(...args);
            };
            this.isPending = true;
        });
    }
}
exports.Deferred = Deferred;
function isGalleryInViewport({ container, scrollTop }) {
    try {
        const isTopVisible = container.scrollBase < scrollTop + pro_gallery_lib_1.window.innerHeight;
        const isBottomVisible = container.scrollBase + container.galleryHeight > scrollTop;
        return isTopVisible && isBottomVisible;
    }
    catch (e) {
        console.warn('Could not calculate viewport', e);
        return true;
    }
}
exports.isGalleryInViewport = isGalleryInViewport;
