// pro-gallery/dist/cjs/components/gallery/proGallery/navigationPanel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getCustomNavigationPanelInlineStyles = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const pro_gallery_lib_1 = require("pro-gallery-lib");
const GalleryUI_js_1 = require("../../item/media/GalleryUI.js");
const textItem_js_1 = tslib_1.__importDefault(require("../../item/textItem.js"));
const ENTER_KEY = 'Enter';
class NavigationPanel extends react_1.default.Component {
    constructor(props) {
        super(props);
        this.scrollToThumbnail = this.scrollToThumbnail.bind(this);
    }
    scrollToThumbnail(itemIdx) {
        this.props.navigationToIdxCB(itemIdx);
    }
    createThumbnails({ navigationPanelPosition, thumbnailAlignment, options, galleryStructure, settings }) {
        const clearedGalleryItems = pro_gallery_lib_1.thumbnailsLogic.clearGalleryItems(this.props.items, this.props.galleryStructure.galleryItems);
        const activeIndex = pro_gallery_lib_1.utils.inRange(this.props.activeIndex, clearedGalleryItems.length);
        const { horizontalThumbnails, items, thumbnailsMargins, thumbnailsStyle, activeIndexOffsetMemory } = pro_gallery_lib_1.thumbnailsLogic.getThumbnailsData({
            items: this.props.items,
            activeIndex,
            options,
            galleryStructure,
            thumbnailAlignment,
            containerHeight: this.props.container.height,
            containerWidth: this.props.container.width,
            activeIndexOffsetMemory: this.activeIndexOffsetMemory,
            prevActiveIndex: this.prevActiveIndex,
        });
        this.prevActiveIndex = activeIndex;
        this.activeIndexOffsetMemory = activeIndexOffsetMemory;
        return (react_1.default.createElement("div", { className: 'pro-gallery inline-styles thumbnails-gallery ' +
                (this.props.domOrder ? 'thumbnails-gallery-' + this.props.domOrder : '') +
                (horizontalThumbnails ? ' one-row hide-scrollbars ' : '') +
                (options[pro_gallery_lib_1.optionsMap.behaviourParams.gallery.layoutDirection] ===
                    pro_gallery_lib_1.GALLERY_CONSTS[pro_gallery_lib_1.optionsMap.behaviourParams.gallery.layoutDirection].RIGHT_TO_LEFT
                    ? ' rtl '
                    : ' ltr ') +
                ((settings === null || settings === void 0 ? void 0 : settings.isAccessible) ? ' accessible ' : ''), style: Object.assign(Object.assign(Object.assign(Object.assign({}, (this.props.isPrerenderMode ? { display: 'block' } : {})), { width: thumbnailsStyle.width, height: thumbnailsStyle.height }), thumbnailsMargins), getNavigationPanelOnGalleryPositionStyles({
                galleryThumbnailsAlignment: thumbnailAlignment,
                navigationPanelPosition,
            })), "data-hook": "gallery-thumbnails" },
            react_1.default.createElement("div", { "data-hook": "gallery-thumbnails-column", className: 'galleryColumn', key: 'thumbnails-column', style: Object.assign({}, thumbnailsStyle) }, items.map(({ thumbnailItem, location, idx }) => {
                const highlighted = idx === activeIndex % clearedGalleryItems.length;
                const itemStyle = Object.assign(Object.assign({ width: options[pro_gallery_lib_1.optionsMap.layoutParams.thumbnails.size], height: options[pro_gallery_lib_1.optionsMap.layoutParams.thumbnails.size], overflow: 'hidden', backgroundImage: `url(${thumbnailItem.createUrl(pro_gallery_lib_1.GALLERY_CONSTS.urlSizes.THUMBNAIL, pro_gallery_lib_1.GALLERY_CONSTS.urlTypes.HIGH_RES)})` }, location), (this.props.isPrerenderMode ? { opacity: 0 } : {}));
                return (react_1.default.createElement("div", { key: 'thumbnail-' + thumbnailItem.id + (Number.isInteger(idx) ? '-' + idx : ''), className: 'thumbnailItem ' +
                        'thumbnail-' +
                        thumbnailItem.idx +
                        (highlighted
                            ? ' pro-gallery-thumbnails-highlighted pro-gallery-highlight' +
                                (pro_gallery_lib_1.utils.isMobile() ? ' pro-gallery-mobile-indicator' : '')
                            : ''), "data-key": thumbnailItem.id, style: itemStyle, onClick: () => this.scrollToThumbnail(idx), tabIndex: 0, onKeyDown: (e) => {
                        if (e.key === ENTER_KEY) {
                            this.scrollToThumbnail(idx);
                        }
                    } },
                    thumbnailItem.type === 'video' &&
                        options[pro_gallery_lib_1.optionsMap.behaviourParams.item.video.enableThumbnailsPlayButton] && (react_1.default.createElement(GalleryUI_js_1.GalleryUI, { type: 'videoPlayButton', size: '28' })),
                    thumbnailItem.type === '3d' &&
                        options[pro_gallery_lib_1.optionsMap.behaviourParams.item.threeDimensionalScene.enableThumbnailsPlayButton] && (react_1.default.createElement(GalleryUI_js_1.GalleryUI, { type: 'rotateArrow', size: '28' })),
                    thumbnailItem.type === 'text' ? (react_1.default.createElement(textItem_js_1.default, Object.assign({}, this.props, thumbnailItem.renderProps(), { options: Object.assign(Object.assign({}, options), { [pro_gallery_lib_1.optionsMap.layoutParams.crop.method]: pro_gallery_lib_1.GALLERY_CONSTS[pro_gallery_lib_1.optionsMap.layoutParams.crop.method].FILL, [pro_gallery_lib_1.optionsMap.layoutParams.crop.enable]: true }), actions: {}, imageDimensions: Object.assign(Object.assign({}, itemStyle), { marginTop: 0, marginLeft: 0 }), style: Object.assign(Object.assign({}, thumbnailItem.renderProps().style), itemStyle) }))) : null));
            }))));
    }
    //-----------------------------------------| RENDER |--------------------------------------------//
    render() {
        const navigationRelevantProps = {
            navigationPanelPosition: this.props.options[pro_gallery_lib_1.optionsMap.layoutParams.thumbnails.position],
            thumbnailAlignment: this.props.options[pro_gallery_lib_1.optionsMap.layoutParams.thumbnails.alignment],
            options: this.props.options,
            galleryStructure: this.props.galleryStructure,
            settings: this.props.settings,
        };
        return this.createThumbnails(navigationRelevantProps);
    }
}
const getHorizontalNavigationPanelDimensions = ({ width, height, galleryHeight, navigationPanelPosition }) => {
    if (navigationPanelPosition === pro_gallery_lib_1.GALLERY_CONSTS[pro_gallery_lib_1.optionsMap.layoutParams.thumbnails.position].ON_GALLERY) {
        return {};
    }
    else {
        return { width: width, height: height - galleryHeight };
    }
};
const getVerticalNavigationPanelDimensions = ({ width, height, galleryWidth, navigationPanelPosition }) => {
    if (navigationPanelPosition === pro_gallery_lib_1.GALLERY_CONSTS[pro_gallery_lib_1.optionsMap.layoutParams.thumbnails.position].ON_GALLERY) {
        return {};
    }
    else {
        return { width: width - galleryWidth, height: height };
    }
};
const getCustomNavigationPanelDimensions = ({ galleryHeight, galleryWidth, height, width, galleryThumbnailsAlignment, navigationPanelPosition, }) => {
    switch (galleryThumbnailsAlignment) {
        case pro_gallery_lib_1.GALLERY_CONSTS[pro_gallery_lib_1.optionsMap.layoutParams.thumbnails.alignment].TOP:
            return getHorizontalNavigationPanelDimensions({ galleryHeight, galleryWidth, height, width, navigationPanelPosition }, false);
        case pro_gallery_lib_1.GALLERY_CONSTS[pro_gallery_lib_1.optionsMap.layoutParams.thumbnails.alignment].BOTTOM:
            return getHorizontalNavigationPanelDimensions({ galleryHeight, galleryWidth, height, width, navigationPanelPosition }, true);
        case pro_gallery_lib_1.GALLERY_CONSTS[pro_gallery_lib_1.optionsMap.layoutParams.thumbnails.alignment].RIGHT:
        case pro_gallery_lib_1.GALLERY_CONSTS[pro_gallery_lib_1.optionsMap.layoutParams.thumbnails.alignment].LEFT:
            return getVerticalNavigationPanelDimensions({
                galleryHeight,
                galleryWidth,
                height,
                width,
                navigationPanelPosition,
            });
        default:
            return false;
    }
};
const getCustomNavigationPanelInlineStyles = ({ galleryHeight, galleryWidth, height, width, galleryThumbnailsAlignment, navigationPanelPosition, }) => {
    return Object.assign(Object.assign(Object.assign({}, getCustomNavigationPanelDimensions({
        galleryHeight,
        galleryWidth,
        height,
        width,
        galleryThumbnailsAlignment,
        navigationPanelPosition,
    })), getNavigationPanelOnGalleryPositionStyles({
        galleryThumbnailsAlignment,
        navigationPanelPosition,
    })), { overflow: 'hidden', float: 'left' });
};
exports.getCustomNavigationPanelInlineStyles = getCustomNavigationPanelInlineStyles;
const getNavigationPanelOnGalleryPositionStyles = ({ galleryThumbnailsAlignment, navigationPanelPosition }) => {
    if (navigationPanelPosition === pro_gallery_lib_1.GALLERY_CONSTS[pro_gallery_lib_1.optionsMap.layoutParams.thumbnails.position].ON_GALLERY) {
        let onGalleryStyles = { position: 'absolute' };
        onGalleryStyles[galleryThumbnailsAlignment.toLowerCase()] = 0;
        return onGalleryStyles;
    }
};
exports.default = NavigationPanel;
