// pro-gallery/dist/cjs/components/gallery/proGallery/navigationArrows.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ArrowsContainer = exports.ArrowButtonWithCursorController = exports.ArrowButton = exports.NavigationArrows = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const ReactDOM = tslib_1.__importStar(require("react-dom"));
const pro_gallery_lib_1 = require("pro-gallery-lib");
const mouseCursorPosition_js_1 = require("../../helpers/mouseCursorPosition.js");
const navigationArrowUtils_js_1 = require("../../helpers/navigationArrowUtils.js");
function NavigationArrows({ options, hideLeftArrow, hideRightArrow, container, customNavArrowsRenderer, next, id, }) {
    const itemSpacing = options[pro_gallery_lib_1.optionsMap.layoutParams.structure.itemSpacing];
    const isRTL = options[pro_gallery_lib_1.optionsMap.behaviourParams.gallery.layoutDirection] ===
        pro_gallery_lib_1.GALLERY_CONSTS[pro_gallery_lib_1.optionsMap.behaviourParams.gallery.layoutDirection].RIGHT_TO_LEFT;
    const arrowsPosition = options[pro_gallery_lib_1.optionsMap.layoutParams.navigationArrows.position];
    const arrowsPadding = options[pro_gallery_lib_1.optionsMap.layoutParams.navigationArrows.padding];
    const scrollDirection = options[pro_gallery_lib_1.optionsMap.layoutParams.structure.scrollDirection];
    const mouseCursorContainerMaxWidth = options[pro_gallery_lib_1.optionsMap.layoutParams.navigationArrows.mouseCursorContainerMaxWidth];
    const { arrowRenderer: renderArrowSvg, navArrowsContainerWidth, navArrowsContainerHeight, } = (0, navigationArrowUtils_js_1.getArrowsRenderData)({
        customNavArrowsRenderer,
        arrowsColor: options[pro_gallery_lib_1.optionsMap.stylingParams.arrowsColor],
        arrowsSize: options[pro_gallery_lib_1.optionsMap.layoutParams.navigationArrows.size],
        arrowsType: options[pro_gallery_lib_1.optionsMap.layoutParams.navigationArrows.type],
        containerStyleType: options[pro_gallery_lib_1.optionsMap.layoutParams.navigationArrows.container.type],
    });
    const mouseCursorEnabled = arrowsPosition === pro_gallery_lib_1.GALLERY_CONSTS[pro_gallery_lib_1.optionsMap.layoutParams.navigationArrows.position].MOUSE_CURSOR;
    const { galleryHeight } = container;
    const infoHeight = options[pro_gallery_lib_1.optionsMap.layoutParams.info.height];
    const imageHeight = galleryHeight - infoHeight;
    // the nav arrows parent container top edge is itemSpacing/2 ABOVE the actual view, that calculates the middle point of gallery
    const galleryVerticalCenter = `50% + ${itemSpacing / 4}px`;
    // Determines the direction fix, the direction in which we move the nav arrows 'vertical position fix' pixels
    let directionFix;
    if (pro_gallery_lib_1.GALLERY_CONSTS.hasExternalAbovePlacement(options[pro_gallery_lib_1.optionsMap.layoutParams.info.placement])) {
        directionFix = -1;
    }
    else if (pro_gallery_lib_1.GALLERY_CONSTS.hasExternalBelowPlacement(options[pro_gallery_lib_1.optionsMap.layoutParams.info.placement])) {
        directionFix = 1;
    }
    else {
        // if we got here, we should be ITEM_CENTER, taken care of in layoutHelper.ts
    }
    const verticalPositionFix = {
        [pro_gallery_lib_1.GALLERY_CONSTS[pro_gallery_lib_1.optionsMap.layoutParams.navigationArrows.verticalAlignment].ITEM_CENTER]: 0,
        [pro_gallery_lib_1.GALLERY_CONSTS[pro_gallery_lib_1.optionsMap.layoutParams.navigationArrows.verticalAlignment].IMAGE_CENTER]: infoHeight * directionFix,
        [pro_gallery_lib_1.GALLERY_CONSTS[pro_gallery_lib_1.optionsMap.layoutParams.navigationArrows.verticalAlignment].INFO_CENTER]: -imageHeight * directionFix,
    }[options[pro_gallery_lib_1.optionsMap.layoutParams.navigationArrows.verticalAlignment]];
    const arrowBoxStyle = (0, navigationArrowUtils_js_1.getArrowBoxStyle)({
        type: options[pro_gallery_lib_1.optionsMap.layoutParams.navigationArrows.container.type],
        backgroundColor: options[pro_gallery_lib_1.optionsMap.layoutParams.navigationArrows.container.backgroundColor],
        borderRadius: options[pro_gallery_lib_1.optionsMap.layoutParams.navigationArrows.container.borderRadius],
    });
    const containerStyle = mouseCursorEnabled
        ? Object.assign({ width: `${navArrowsContainerWidth}px`, height: `${navArrowsContainerHeight}px` }, arrowBoxStyle) : Object.assign({ width: `${navArrowsContainerWidth}px`, height: `${navArrowsContainerHeight}px`, padding: 0, top: `calc(${galleryVerticalCenter} - ${navArrowsContainerHeight / 2}px - 
        ${verticalPositionFix / 2}px)` }, arrowBoxStyle);
    const arrowsPos = scrollDirection === pro_gallery_lib_1.GALLERY_CONSTS[pro_gallery_lib_1.optionsMap.layoutParams.structure.scrollDirection].HORIZONTAL &&
        arrowsPosition === pro_gallery_lib_1.GALLERY_CONSTS[pro_gallery_lib_1.optionsMap.layoutParams.navigationArrows.position].OUTSIDE_GALLERY
        ? `-${20 + navArrowsContainerWidth}px`
        : `${itemSpacing / 2 + (arrowsPadding ? arrowsPadding : 0)}px`;
    // itemSpacing effect the margin of the main div ('pro-gallery-parent-container') that SlideshowView is rendering, so the arrows should be places accordingly
    // arrowsPadding relevant only for arrowsPosition.ON_GALLERY
    const prevContainerStyle = { left: mouseCursorEnabled ? 0 : arrowsPos };
    const nextContainerStyle = { right: mouseCursorEnabled ? 0 : arrowsPos };
    const containerStylesByType = {
        BOX: 'box',
        SHADOW: 'shadow',
        NONE: 'only-arrows',
    };
    const containerStylingClass = containerStylesByType[options[pro_gallery_lib_1.optionsMap.layoutParams.navigationArrows.container.type]] || '';
    const arrowsBaseClasses = [
        'nav-arrows-container',
        containerStylingClass,
        pro_gallery_lib_1.utils.isMobile() ? ' pro-gallery-mobile-indicator' : '',
    ];
    const navigationArrowPortalId = `arrow-portal-container-${id}`;
    const ArrowRenderHandler = mouseCursorEnabled ? ArrowButtonWithCursorController : ArrowButton;
    const renderArrow = (directionIsLeft) => {
        return (react_1.default.createElement(ArrowRenderHandler, Object.assign({}, {
            renderArrowSvg,
            next,
            directionIsLeft,
            arrowsBaseClasses,
            containerStyle,
            prevContainerStyle,
            nextContainerStyle,
            isRTL,
            hideLeftArrow,
            hideRightArrow,
            arrowBoxStyle,
            navArrowsContainerWidth,
            navArrowsContainerHeight,
            navigationArrowPortalId,
            mouseCursorContainerMaxWidth,
            id,
        })));
    };
    const provideMouseFollower = (children) => {
        if (mouseCursorEnabled) {
            return react_1.default.createElement(mouseCursorPosition_js_1.MouseFollowerProvider, { id: id }, children);
        }
        return children;
    };
    return (react_1.default.createElement(ArrowsContainer, { hideLeftArrow: hideLeftArrow, hideRightArrow: hideRightArrow, mouseCursorEnabled: mouseCursorEnabled, isRTL: isRTL, navigationArrowPortalId: navigationArrowPortalId }, provideMouseFollower(react_1.default.createElement(react_1.default.Fragment, null,
        hideLeftArrow ? null : renderArrow(true),
        hideRightArrow ? null : renderArrow(false)))));
}
exports.NavigationArrows = NavigationArrows;
function ArrowButton({ cursor, arrowsBaseClasses, containerStyle, prevContainerStyle, nextContainerStyle, isRTL, directionIsLeft, renderArrowSvg, next, arrowBoxStyle, navArrowsContainerWidth, navArrowsContainerHeight, navigationArrowPortalId, }) {
    const isNext = (directionIsLeft && isRTL) || (!directionIsLeft && !isRTL);
    const nextAction = () => next({ direction: directionIsLeft ? -1 : 1 });
    const buttonProps = {
        className: arrowsBaseClasses.join(' '),
        onClick: () => setTimeout(nextAction, 0),
        ['aria-label']: `${isNext ? 'Next' : 'Previous'} Item`,
        tabIndex: '0',
        key: !isNext ? 'nav-arrow-back' : 'nav-arrow-next',
        ['data-hook']: !isNext ? 'nav-arrow-back' : 'nav-arrow-next',
        style: Object.assign(Object.assign({}, containerStyle), (directionIsLeft ? prevContainerStyle : nextContainerStyle)),
    };
    if (cursor) {
        const { containerRef, isMouseEnter, position } = cursor;
        return (react_1.default.createElement("button", Object.assign({ ref: (ref) => (containerRef.current = ref) }, buttonProps), isMouseEnter &&
            ReactDOM.createPortal(react_1.default.createElement("span", { style: Object.assign(Object.assign({ top: position.y - navArrowsContainerHeight / 2, left: position.x - navArrowsContainerWidth / 2 }, arrowBoxStyle), { width: navArrowsContainerWidth, height: navArrowsContainerHeight }) }, renderArrowSvg(directionIsLeft ? 'left' : 'right')), window.document.getElementById(navigationArrowPortalId))));
    }
    return react_1.default.createElement("button", Object.assign({}, buttonProps), renderArrowSvg(directionIsLeft ? 'left' : 'right'));
}
exports.ArrowButton = ArrowButton;
function ArrowButtonWithCursorController(props) {
    const { directionIsLeft, next, mouseCursorContainerMaxWidth, hideLeftArrow, hideRightArrow, renderArrowSvg, containerStyle, arrowsBaseClasses, navArrowsContainerWidth, navArrowsContainerHeight, } = props;
    const isTheOnlyArrow = hideLeftArrow || hideRightArrow;
    return (react_1.default.createElement(mouseCursorPosition_js_1.ArrowFollower, { id: props.id, mouseCursorContainerMaxWidth: mouseCursorContainerMaxWidth, navArrowsContainerWidth: navArrowsContainerWidth, navArrowsContainerHeight: navArrowsContainerHeight, onNavigate: () => next({ direction: directionIsLeft ? -1 : 1 }), direction: directionIsLeft ? 'left' : 'right', isTheOnlyArrow: isTheOnlyArrow }, (x, y) => (react_1.default.createElement("div", { style: {
            top: y,
            left: x,
        }, className: "follow-mouse-cursor" },
        react_1.default.createElement("div", { className: arrowsBaseClasses.join(' '), style: Object.assign(Object.assign({}, containerStyle), { top: -navArrowsContainerHeight / 2, left: -navArrowsContainerWidth / 2 }) }, renderArrowSvg(directionIsLeft ? 'left' : 'right'))))));
}
exports.ArrowButtonWithCursorController = ArrowButtonWithCursorController;
function ArrowsContainer({ hideRightArrow, hideLeftArrow, mouseCursorEnabled, isRTL, children, navigationArrowPortalId, }) {
    if (mouseCursorEnabled) {
        const styleForMouseCursor = {
            justifyContent: hideLeftArrow ? 'flex-end' : hideRightArrow ? 'flex-start' : 'space-between',
            flexDirection: isRTL ? 'row-reverse' : 'row',
        };
        return (react_1.default.createElement("div", { className: "mouse-cursor", style: Object.assign({}, styleForMouseCursor) },
            react_1.default.createElement("div", { id: navigationArrowPortalId, className: "arrow-portal-container" }),
            children));
    }
    return react_1.default.Fragment ? react_1.default.createElement(react_1.default.Fragment, null, children) : react_1.default.createElement("div", null, children);
}
exports.ArrowsContainer = ArrowsContainer;
