// pro-gallery/dist/cjs/components/gallery/proGallery/proGallery.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
require("../../../common/utils/polyfills.js");
const react_1 = tslib_1.__importDefault(require("react"));
const pro_gallery_lib_1 = require("pro-gallery-lib");
const galleryContainer_js_1 = tslib_1.__importDefault(require("./galleryContainer.js"));
require("../../../versionLogger.js");
const GalleryContext_js_1 = require("../../../context/GalleryContext.js");
class ProGallery extends react_1.default.Component {
    constructor() {
        super();
        if (pro_gallery_lib_1.utils.isLocal() && !pro_gallery_lib_1.utils.isTest()) {
            console.log('PRO GALLERY DEV');
        }
    }
    renderProps() {
        var _a;
        return Object.assign(Object.assign({}, this.props), { id: this.props.id, items: this.props.items || [], settings: this.props.settings || {}, offsetTop: this.props.offsetTop, proGalleryRegionLabel: this.props.proGalleryRegionLabel, isInDisplay: (_a = this.props.isInDisplay) !== null && _a !== void 0 ? _a : true });
    }
    containerProps() {
        return {
            id: `pro-gallery-${this.props.id}`,
            className: 'pro-gallery',
        };
    }
    render() {
        return (react_1.default.createElement(GalleryContext_js_1.GalleryProvider, Object.assign({}, this.props),
            react_1.default.createElement("div", Object.assign({}, this.containerProps()),
                react_1.default.createElement(galleryContainer_js_1.default, Object.assign({}, this.renderProps())))));
    }
}
exports.default = ProGallery;
