// @wix/metro-common-builders/dist/__tests__/merge-deep.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var src_1 = require("../src");
describe('Merge deep', function () {
    it('should merge two object', function () {
        var firstPerson = {
            name: 'eli',
            address: {
                city: 'jerusalem',
                building: {
                    apt: 10,
                    floor: 3,
                    entrance: 'a',
                    some: {
                        nested: true,
                        array: [1, 2],
                    },
                },
            },
            age: 35,
        };
        var secondPerson = {
            name: 'moshe',
            address: {
                building: {
                    apt: 11,
                    some: {
                        nested: false,
                        array: [3, 2],
                    },
                },
            },
        };
        var expectedPerson = {
            name: 'moshe',
            address: {
                city: 'jerusalem',
                building: {
                    apt: 11,
                    floor: 3,
                    entrance: 'a',
                    some: {
                        nested: false,
                        array: [3, 2],
                    },
                },
            },
            age: 35,
        };
        var actualPerson = (0, src_1.mergeDeep)(firstPerson, secondPerson);
        expect(actualPerson).toEqual(expectedPerson);
    });
    it('should merge objects with date properly', function () {
        var overrideDate = new Date(12345);
        var initialObject = {
            some: {
                date: new Date(),
                type: 'a',
                extraField: true,
            },
        };
        var overrideObject = {
            some: {
                date: overrideDate,
                type: 'a',
            },
        };
        var mergeResult = (0, src_1.mergeDeep)(initialObject, overrideObject);
        expect(mergeResult.some.date instanceof Date).toBe(true);
        expect(mergeResult.some.date).toEqual(overrideDate);
        expect(mergeResult.some.type).toEqual(overrideObject.some.type);
        expect(mergeResult.some.extraField).toBe(true);
    });
});
