// @wix/auto_sdk_payments_dispute-evidence-documents/build/internal/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  SortOrder: () => SortOrder,
  WebhookIdentityType: () => WebhookIdentityType,
  attachDisputeEvidenceDocumentFile: () => attachDisputeEvidenceDocumentFile2,
  bulkUpdateDisputeEvidenceDocumentTags: () => bulkUpdateDisputeEvidenceDocumentTags2,
  bulkUpdateDisputeEvidenceDocumentTagsByFilter: () => bulkUpdateDisputeEvidenceDocumentTagsByFilter2,
  createDisputeEvidenceDocument: () => createDisputeEvidenceDocument2,
  deleteDisputeEvidenceDocument: () => deleteDisputeEvidenceDocument2,
  generateDisputeEvidenceDocumentDownloadUrl: () => generateDisputeEvidenceDocumentDownloadUrl2,
  generateDisputeEvidenceDocumentUploadUrl: () => generateDisputeEvidenceDocumentUploadUrl2,
  getDisputeEvidenceDocument: () => getDisputeEvidenceDocument2,
  markDisputeEvidenceDocumentSubmitted: () => markDisputeEvidenceDocumentSubmitted2,
  queryDisputeEvidenceDocuments: () => queryDisputeEvidenceDocuments2
});
module.exports = __toCommonJS(index_typings_exports);

// src/payments-dispute-evidence-documents-v1-dispute-evidence-document-dispute-evidence-documents.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/payments-dispute-evidence-documents-v1-dispute-evidence-document-dispute-evidence-documents.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(opts) {
  const domainToMappings = {};
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_dispute-evidence-documents";
function createDisputeEvidenceDocument(payload) {
  function __createDisputeEvidenceDocument({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "disputeEvidenceDocument.createdDate" },
          { path: "disputeEvidenceDocument.updatedDate" },
          { path: "disputeEvidenceDocument.submittedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.CreateDisputeEvidenceDocument",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents",
          data: serializedData,
          host
        }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "disputeEvidenceDocument.createdDate" },
            { path: "disputeEvidenceDocument.updatedDate" },
            { path: "disputeEvidenceDocument.submittedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createDisputeEvidenceDocument;
}
function generateDisputeEvidenceDocumentUploadUrl(payload) {
  function __generateDisputeEvidenceDocumentUploadUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.GenerateDisputeEvidenceDocumentUploadUrl",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}/generate-upload-url",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __generateDisputeEvidenceDocumentUploadUrl;
}
function attachDisputeEvidenceDocumentFile(payload) {
  function __attachDisputeEvidenceDocumentFile({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.AttachDisputeEvidenceDocumentFile",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}/attach-file",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "disputeEvidenceDocument.createdDate" },
            { path: "disputeEvidenceDocument.updatedDate" },
            { path: "disputeEvidenceDocument.submittedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __attachDisputeEvidenceDocumentFile;
}
function getDisputeEvidenceDocument(payload) {
  function __getDisputeEvidenceDocument({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "GET",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.GetDisputeEvidenceDocument",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "disputeEvidenceDocument.createdDate" },
            { path: "disputeEvidenceDocument.updatedDate" },
            { path: "disputeEvidenceDocument.submittedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getDisputeEvidenceDocument;
}
function generateDisputeEvidenceDocumentDownloadUrl(payload) {
  function __generateDisputeEvidenceDocumentDownloadUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.GenerateDisputeEvidenceDocumentDownloadUrl",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}/generate-download-url",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "urlExpirationDate" }]
        }
      ])
    };
    return metadata;
  }
  return __generateDisputeEvidenceDocumentDownloadUrl;
}
function markDisputeEvidenceDocumentSubmitted(payload) {
  function __markDisputeEvidenceDocumentSubmitted({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.MarkDisputeEvidenceDocumentSubmitted",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}/mark-submitted",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __markDisputeEvidenceDocumentSubmitted;
}
function deleteDisputeEvidenceDocument(payload) {
  function __deleteDisputeEvidenceDocument({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "DELETE",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.DeleteDisputeEvidenceDocument",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteDisputeEvidenceDocument;
}
function queryDisputeEvidenceDocuments(payload) {
  function __queryDisputeEvidenceDocuments({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.QueryDisputeEvidenceDocuments",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents/query",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "disputeEvidenceDocuments.createdDate" },
            { path: "disputeEvidenceDocuments.updatedDate" },
            { path: "disputeEvidenceDocuments.submittedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
            {
              protoPath: "/v1/dispute-evidence-documents/query",
              data: payload,
              host
            }
          ),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryDisputeEvidenceDocuments;
}
function bulkUpdateDisputeEvidenceDocumentTags(payload) {
  function __bulkUpdateDisputeEvidenceDocumentTags({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.BulkUpdateDisputeEvidenceDocumentTags",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/bulk/dispute-evidence-documents/update-tags",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateDisputeEvidenceDocumentTags;
}
function bulkUpdateDisputeEvidenceDocumentTagsByFilter(payload) {
  function __bulkUpdateDisputeEvidenceDocumentTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.BulkUpdateDisputeEvidenceDocumentTagsByFilter",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/bulk/dispute-evidence-documents/update-tags-by-filter",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateDisputeEvidenceDocumentTagsByFilter;
}

// src/payments-dispute-evidence-documents-v1-dispute-evidence-document-dispute-evidence-documents.universal.ts
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createDisputeEvidenceDocument2(disputeEvidenceDocument) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    disputeEvidenceDocument
  });
  const reqOpts = createDisputeEvidenceDocument(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.disputeEvidenceDocument;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { disputeEvidenceDocument: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["disputeEvidenceDocument"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function generateDisputeEvidenceDocumentUploadUrl2(disputeEvidenceDocumentId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    disputeEvidenceDocumentId,
    filename: options?.filename
  });
  const reqOpts = generateDisputeEvidenceDocumentUploadUrl(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          disputeEvidenceDocumentId: "$[0]",
          filename: "$[1].filename"
        },
        singleArgumentUnchanged: false
      },
      ["disputeEvidenceDocumentId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function attachDisputeEvidenceDocumentFile2(disputeEvidenceDocumentId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    disputeEvidenceDocumentId
  });
  const reqOpts = attachDisputeEvidenceDocumentFile(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { disputeEvidenceDocumentId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["disputeEvidenceDocumentId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getDisputeEvidenceDocument2(disputeEvidenceDocumentId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    disputeEvidenceDocumentId
  });
  const reqOpts = getDisputeEvidenceDocument(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.disputeEvidenceDocument;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { disputeEvidenceDocumentId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["disputeEvidenceDocumentId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function generateDisputeEvidenceDocumentDownloadUrl2(disputeEvidenceDocumentId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    disputeEvidenceDocumentId
  });
  const reqOpts = generateDisputeEvidenceDocumentDownloadUrl(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { disputeEvidenceDocumentId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["disputeEvidenceDocumentId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function markDisputeEvidenceDocumentSubmitted2(disputeEvidenceDocumentId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    disputeEvidenceDocumentId
  });
  const reqOpts = markDisputeEvidenceDocumentSubmitted(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { disputeEvidenceDocumentId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["disputeEvidenceDocumentId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteDisputeEvidenceDocument2(disputeEvidenceDocumentId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    disputeEvidenceDocumentId
  });
  const reqOpts = deleteDisputeEvidenceDocument(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { disputeEvidenceDocumentId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["disputeEvidenceDocumentId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryDisputeEvidenceDocuments2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryDisputeEvidenceDocuments(
        payload
      );
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [])
      );
      return {
        items: transformedData?.disputeEvidenceDocuments,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function bulkUpdateDisputeEvidenceDocumentTags2(ids, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    ids,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateDisputeEvidenceDocumentTags(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          ids: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["ids", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateDisputeEvidenceDocumentTagsByFilter2(filter, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    filter,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateDisputeEvidenceDocumentTagsByFilter(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          filter: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["filter", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  SortOrder,
  WebhookIdentityType,
  attachDisputeEvidenceDocumentFile,
  bulkUpdateDisputeEvidenceDocumentTags,
  bulkUpdateDisputeEvidenceDocumentTagsByFilter,
  createDisputeEvidenceDocument,
  deleteDisputeEvidenceDocument,
  generateDisputeEvidenceDocumentDownloadUrl,
  generateDisputeEvidenceDocumentUploadUrl,
  getDisputeEvidenceDocument,
  markDisputeEvidenceDocumentSubmitted,
  queryDisputeEvidenceDocuments
});
