// @wix/auto_sdk_payments_dispute-evidence-documents/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  attachDisputeEvidenceDocumentFile: () => attachDisputeEvidenceDocumentFile2,
  bulkUpdateDisputeEvidenceDocumentTags: () => bulkUpdateDisputeEvidenceDocumentTags2,
  bulkUpdateDisputeEvidenceDocumentTagsByFilter: () => bulkUpdateDisputeEvidenceDocumentTagsByFilter2,
  createDisputeEvidenceDocument: () => createDisputeEvidenceDocument2,
  deleteDisputeEvidenceDocument: () => deleteDisputeEvidenceDocument2,
  generateDisputeEvidenceDocumentDownloadUrl: () => generateDisputeEvidenceDocumentDownloadUrl2,
  generateDisputeEvidenceDocumentUploadUrl: () => generateDisputeEvidenceDocumentUploadUrl2,
  getDisputeEvidenceDocument: () => getDisputeEvidenceDocument2,
  markDisputeEvidenceDocumentSubmitted: () => markDisputeEvidenceDocumentSubmitted2,
  queryDisputeEvidenceDocuments: () => queryDisputeEvidenceDocuments2
});
module.exports = __toCommonJS(meta_exports);

// src/payments-dispute-evidence-documents-v1-dispute-evidence-document-dispute-evidence-documents.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(opts) {
  const domainToMappings = {};
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_dispute-evidence-documents";
function createDisputeEvidenceDocument(payload) {
  function __createDisputeEvidenceDocument({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "disputeEvidenceDocument.createdDate" },
          { path: "disputeEvidenceDocument.updatedDate" },
          { path: "disputeEvidenceDocument.submittedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.CreateDisputeEvidenceDocument",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents",
          data: serializedData,
          host
        }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "disputeEvidenceDocument.createdDate" },
            { path: "disputeEvidenceDocument.updatedDate" },
            { path: "disputeEvidenceDocument.submittedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createDisputeEvidenceDocument;
}
function generateDisputeEvidenceDocumentUploadUrl(payload) {
  function __generateDisputeEvidenceDocumentUploadUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.GenerateDisputeEvidenceDocumentUploadUrl",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}/generate-upload-url",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __generateDisputeEvidenceDocumentUploadUrl;
}
function attachDisputeEvidenceDocumentFile(payload) {
  function __attachDisputeEvidenceDocumentFile({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.AttachDisputeEvidenceDocumentFile",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}/attach-file",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "disputeEvidenceDocument.createdDate" },
            { path: "disputeEvidenceDocument.updatedDate" },
            { path: "disputeEvidenceDocument.submittedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __attachDisputeEvidenceDocumentFile;
}
function getDisputeEvidenceDocument(payload) {
  function __getDisputeEvidenceDocument({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "GET",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.GetDisputeEvidenceDocument",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "disputeEvidenceDocument.createdDate" },
            { path: "disputeEvidenceDocument.updatedDate" },
            { path: "disputeEvidenceDocument.submittedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getDisputeEvidenceDocument;
}
function generateDisputeEvidenceDocumentDownloadUrl(payload) {
  function __generateDisputeEvidenceDocumentDownloadUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.GenerateDisputeEvidenceDocumentDownloadUrl",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}/generate-download-url",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "urlExpirationDate" }]
        }
      ])
    };
    return metadata;
  }
  return __generateDisputeEvidenceDocumentDownloadUrl;
}
function markDisputeEvidenceDocumentSubmitted(payload) {
  function __markDisputeEvidenceDocumentSubmitted({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.MarkDisputeEvidenceDocumentSubmitted",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}/mark-submitted",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __markDisputeEvidenceDocumentSubmitted;
}
function deleteDisputeEvidenceDocument(payload) {
  function __deleteDisputeEvidenceDocument({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "DELETE",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.DeleteDisputeEvidenceDocument",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteDisputeEvidenceDocument;
}
function queryDisputeEvidenceDocuments(payload) {
  function __queryDisputeEvidenceDocuments({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.QueryDisputeEvidenceDocuments",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/dispute-evidence-documents/query",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "disputeEvidenceDocuments.createdDate" },
            { path: "disputeEvidenceDocuments.updatedDate" },
            { path: "disputeEvidenceDocuments.submittedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
            {
              protoPath: "/v1/dispute-evidence-documents/query",
              data: payload,
              host
            }
          ),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryDisputeEvidenceDocuments;
}
function bulkUpdateDisputeEvidenceDocumentTags(payload) {
  function __bulkUpdateDisputeEvidenceDocumentTags({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.BulkUpdateDisputeEvidenceDocumentTags",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/bulk/dispute-evidence-documents/update-tags",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateDisputeEvidenceDocumentTags;
}
function bulkUpdateDisputeEvidenceDocumentTagsByFilter(payload) {
  function __bulkUpdateDisputeEvidenceDocumentTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_evidence_documents.v1.dispute_evidence_document",
      method: "POST",
      methodFqn: "wix.payments.dispute_evidence_documents.v1.DisputeEvidenceDocumentService.BulkUpdateDisputeEvidenceDocumentTagsByFilter",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeEvidenceDocumentsV1DisputeEvidenceDocumentServiceUrl(
        {
          protoPath: "/v1/bulk/dispute-evidence-documents/update-tags-by-filter",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateDisputeEvidenceDocumentTagsByFilter;
}

// src/payments-dispute-evidence-documents-v1-dispute-evidence-document-dispute-evidence-documents.meta.ts
function createDisputeEvidenceDocument2() {
  const payload = {};
  const getRequestOptions = createDisputeEvidenceDocument(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/dispute-evidence-documents",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function generateDisputeEvidenceDocumentUploadUrl2() {
  const payload = {
    disputeEvidenceDocumentId: ":disputeEvidenceDocumentId"
  };
  const getRequestOptions = generateDisputeEvidenceDocumentUploadUrl(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}/generate-upload-url",
    pathParams: { disputeEvidenceDocumentId: "disputeEvidenceDocumentId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function attachDisputeEvidenceDocumentFile2() {
  const payload = {
    disputeEvidenceDocumentId: ":disputeEvidenceDocumentId"
  };
  const getRequestOptions = attachDisputeEvidenceDocumentFile(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}/attach-file",
    pathParams: { disputeEvidenceDocumentId: "disputeEvidenceDocumentId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getDisputeEvidenceDocument2() {
  const payload = {
    disputeEvidenceDocumentId: ":disputeEvidenceDocumentId"
  };
  const getRequestOptions = getDisputeEvidenceDocument(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}",
    pathParams: { disputeEvidenceDocumentId: "disputeEvidenceDocumentId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function generateDisputeEvidenceDocumentDownloadUrl2() {
  const payload = {
    disputeEvidenceDocumentId: ":disputeEvidenceDocumentId"
  };
  const getRequestOptions = generateDisputeEvidenceDocumentDownloadUrl(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}/generate-download-url",
    pathParams: { disputeEvidenceDocumentId: "disputeEvidenceDocumentId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function markDisputeEvidenceDocumentSubmitted2() {
  const payload = {
    disputeEvidenceDocumentId: ":disputeEvidenceDocumentId"
  };
  const getRequestOptions = markDisputeEvidenceDocumentSubmitted(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}/mark-submitted",
    pathParams: { disputeEvidenceDocumentId: "disputeEvidenceDocumentId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteDisputeEvidenceDocument2() {
  const payload = {
    disputeEvidenceDocumentId: ":disputeEvidenceDocumentId"
  };
  const getRequestOptions = deleteDisputeEvidenceDocument(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/dispute-evidence-documents/{disputeEvidenceDocumentId}",
    pathParams: { disputeEvidenceDocumentId: "disputeEvidenceDocumentId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryDisputeEvidenceDocuments2() {
  const payload = {};
  const getRequestOptions = queryDisputeEvidenceDocuments(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/dispute-evidence-documents/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateDisputeEvidenceDocumentTags2() {
  const payload = {};
  const getRequestOptions = bulkUpdateDisputeEvidenceDocumentTags(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/dispute-evidence-documents/update-tags",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateDisputeEvidenceDocumentTagsByFilter2() {
  const payload = {};
  const getRequestOptions = bulkUpdateDisputeEvidenceDocumentTagsByFilter(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/dispute-evidence-documents/update-tags-by-filter",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  attachDisputeEvidenceDocumentFile,
  bulkUpdateDisputeEvidenceDocumentTags,
  bulkUpdateDisputeEvidenceDocumentTagsByFilter,
  createDisputeEvidenceDocument,
  deleteDisputeEvidenceDocument,
  generateDisputeEvidenceDocumentDownloadUrl,
  generateDisputeEvidenceDocumentUploadUrl,
  getDisputeEvidenceDocument,
  markDisputeEvidenceDocumentSubmitted,
  queryDisputeEvidenceDocuments
});
