// @wix/auto_sdk_restaurants_item-modifiers/build/es/index.mjs
// src/restaurants-menus-v1-item-modifier-item-modifiers.public.ts
import { renameKeysFromRESTResponseToSDKResponse as renameKeysFromRESTResponseToSDKResponse2 } from "@wix/sdk-runtime/rename-all-nested-keys";
import { transformRESTTimestampToSDKTimestamp as transformRESTTimestampToSDKTimestamp2 } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths as transformPaths3 } from "@wix/sdk-runtime/transformations/transform-paths";
import { EventDefinition } from "@wix/sdk-types";

// src/restaurants-menus-v1-item-modifier-item-modifiers.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/restaurants-menus-v1-item-modifier-item-modifiers.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixRestaurantsMenusItemModifierV1ModifiersServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-modifier",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-modifier",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-modifier",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-menus-item-modifier",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-menus-item-modifier",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-menus-item-modifier",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers/delete",
        destPath: "/v1/bulk/modifiers/delete"
      },
      {
        srcPath: "/restaurants/menus/v1/modifiers/count",
        destPath: "/v1/modifiers/count"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers/create",
        destPath: "/v1/bulk/modifiers/create"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-menus-item-modifier",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers/delete",
        destPath: "/v1/bulk/modifiers/delete"
      },
      {
        srcPath: "/restaurants/menus/v1/modifiers/count",
        destPath: "/v1/modifiers/count"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers/create",
        destPath: "/v1/bulk/modifiers/create"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-modifier",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-menus-item-modifier",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/item-modifiers",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers/create",
        destPath: "/v1/bulk/modifiers/create"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers/update",
        destPath: "/v1/bulk/modifiers/update"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers/delete",
        destPath: "/v1/bulk/modifiers/delete"
      },
      {
        srcPath: "/restaurants/menus/v1/modifiers/count",
        destPath: "/v1/modifiers/count"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-modifier",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_item-modifiers";
function createModifier(payload) {
  function __createModifier({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "modifier.createdDate" },
          { path: "modifier.updatedDate" },
          { path: "modifier.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier",
      method: "POST",
      methodFqn: "wix.restaurants.menus_item_modifier.v1.ModifiersService.CreateModifier",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemModifierV1ModifiersServiceUrl({
        protoPath: "/v1/modifiers",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "modifier.createdDate" },
            { path: "modifier.updatedDate" },
            { path: "modifier.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createModifier;
}
function getModifier(payload) {
  function __getModifier({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier",
      method: "GET",
      methodFqn: "wix.restaurants.menus_item_modifier.v1.ModifiersService.GetModifier",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemModifierV1ModifiersServiceUrl({
        protoPath: "/v1/modifiers/{modifierId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "modifier.createdDate" },
            { path: "modifier.updatedDate" },
            { path: "modifier.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getModifier;
}
function listModifiers(payload) {
  function __listModifiers({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier",
      method: "GET",
      methodFqn: "wix.restaurants.menus_item_modifier.v1.ModifiersService.ListModifiers",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemModifierV1ModifiersServiceUrl({
        protoPath: "/v1/modifiers",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "modifiers.createdDate" },
            { path: "modifiers.updatedDate" },
            { path: "modifiers.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listModifiers;
}
function updateModifier(payload) {
  function __updateModifier({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "modifier.createdDate" },
          { path: "modifier.updatedDate" },
          { path: "modifier.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier",
      method: "PATCH",
      methodFqn: "wix.restaurants.menus_item_modifier.v1.ModifiersService.UpdateModifier",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemModifierV1ModifiersServiceUrl({
        protoPath: "/v1/modifiers/{modifier.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "modifier.createdDate" },
            { path: "modifier.updatedDate" },
            { path: "modifier.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateModifier;
}
function bulkCreateModifiers(payload) {
  function __bulkCreateModifiers({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "modifiers.createdDate" },
          { path: "modifiers.updatedDate" },
          { path: "modifiers.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier",
      method: "POST",
      methodFqn: "wix.restaurants.menus_item_modifier.v1.ModifiersService.BulkCreateModifiers",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemModifierV1ModifiersServiceUrl({
        protoPath: "/v1/bulk/modifiers/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.modifier.createdDate" },
            { path: "results.modifier.updatedDate" },
            { path: "results.modifier.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateModifiers;
}
function bulkUpdateModifiers(payload) {
  function __bulkUpdateModifiers({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "modifiers.mask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "modifiers.modifier.createdDate" },
          { path: "modifiers.modifier.updatedDate" },
          { path: "modifiers.modifier.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier",
      method: "POST",
      methodFqn: "wix.restaurants.menus_item_modifier.v1.ModifiersService.BulkUpdateModifiers",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemModifierV1ModifiersServiceUrl({
        protoPath: "/v1/bulk/modifiers/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.modifier.createdDate" },
            { path: "results.modifier.updatedDate" },
            { path: "results.modifier.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateModifiers;
}
function deleteModifier(payload) {
  function __deleteModifier({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier",
      method: "DELETE",
      methodFqn: "wix.restaurants.menus_item_modifier.v1.ModifiersService.DeleteModifier",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemModifierV1ModifiersServiceUrl({
        protoPath: "/v1/modifiers/{modifierId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteModifier;
}
function queryModifiers(payload) {
  function __queryModifiers({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier",
      method: "POST",
      methodFqn: "wix.restaurants.menus_item_modifier.v1.ModifiersService.QueryModifiers",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemModifierV1ModifiersServiceUrl({
        protoPath: "/v1/modifiers/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "modifiers.createdDate" },
            { path: "modifiers.updatedDate" },
            { path: "modifiers.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryModifiers;
}
function countModifiers(payload) {
  function __countModifiers({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier",
      method: "POST",
      methodFqn: "wix.restaurants.menus_item_modifier.v1.ModifiersService.CountModifiers",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemModifierV1ModifiersServiceUrl({
        protoPath: "/v1/modifiers/count",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __countModifiers;
}
function bulkDeleteModifiers(payload) {
  function __bulkDeleteModifiers({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier",
      method: "DELETE",
      methodFqn: "wix.restaurants.menus_item_modifier.v1.ModifiersService.BulkDeleteModifiers",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemModifierV1ModifiersServiceUrl({
        protoPath: "/v1/bulk/modifiers/delete",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __bulkDeleteModifiers;
}

// src/restaurants-menus-v1-item-modifier-item-modifiers.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createModifier2(modifier) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ modifier });
  const reqOpts = createModifier(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.modifier;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { modifier: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["modifier"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getModifier2(modifierId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    modifierId
  });
  const reqOpts = getModifier(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.modifier;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { modifierId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["modifierId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listModifiers2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    modifierIds: options?.modifierIds,
    paging: options?.paging
  });
  const reqOpts = listModifiers(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          modifierIds: "$[0].modifierIds",
          paging: "$[0].paging"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateModifier2(_id, modifier) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    modifier: { ...modifier, id: _id }
  });
  const reqOpts = updateModifier(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.modifier;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { modifier: "$[1]" },
        explicitPathsToArguments: { "modifier.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "modifier"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCreateModifiers2(modifiers, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    modifiers,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkCreateModifiers(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          modifiers: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["modifiers", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateModifiers2(modifiers, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    modifiers,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkUpdateModifiers(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          modifiers: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["modifiers", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteModifier2(modifierId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    modifierId
  });
  const reqOpts = deleteModifier(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { modifierId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["modifierId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryModifiers2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryModifiers(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [])
      );
      return {
        items: transformedData?.modifiers,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function countModifiers2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    filter: options?.filter
  });
  const reqOpts = countModifiers(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { filter: "$[0].filter" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkDeleteModifiers2(ids) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ ids });
  const reqOpts = bulkDeleteModifiers(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { ids: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["ids"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/restaurants-menus-v1-item-modifier-item-modifiers.public.ts
function createModifier3(httpClient) {
  return (modifier) => createModifier2(
    modifier,
    // @ts-ignore
    { httpClient }
  );
}
function getModifier3(httpClient) {
  return (modifierId) => getModifier2(
    modifierId,
    // @ts-ignore
    { httpClient }
  );
}
function listModifiers3(httpClient) {
  return (options) => listModifiers2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function updateModifier3(httpClient) {
  return (_id, modifier) => updateModifier2(
    _id,
    modifier,
    // @ts-ignore
    { httpClient }
  );
}
function bulkCreateModifiers3(httpClient) {
  return (modifiers, options) => bulkCreateModifiers2(
    modifiers,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdateModifiers3(httpClient) {
  return (modifiers, options) => bulkUpdateModifiers2(
    modifiers,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function deleteModifier3(httpClient) {
  return (modifierId) => deleteModifier2(
    modifierId,
    // @ts-ignore
    { httpClient }
  );
}
function queryModifiers3(httpClient) {
  return () => queryModifiers2(
    // @ts-ignore
    { httpClient }
  );
}
function countModifiers3(httpClient) {
  return (options) => countModifiers2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkDeleteModifiers3(httpClient) {
  return (ids) => bulkDeleteModifiers2(
    ids,
    // @ts-ignore
    { httpClient }
  );
}
var onItemModifierCreated = EventDefinition(
  "wix.restaurants.menus.v1.item_modifier_created",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.externalReferenceInfo.lastSyncDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onItemModifierDeleted = EventDefinition(
  "wix.restaurants.menus.v1.item_modifier_deleted",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "undefined.createdDate" },
          { path: "undefined.updatedDate" },
          { path: "undefined.externalReferenceInfo.lastSyncDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onItemModifierUpdated = EventDefinition(
  "wix.restaurants.menus.v1.item_modifier_updated",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.externalReferenceInfo.lastSyncDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();

// src/restaurants-menus-v1-item-modifier-item-modifiers.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
import { createEventModule } from "@wix/sdk-runtime/event-definition-modules";
var createModifier4 = /* @__PURE__ */ createRESTModule(createModifier3);
var getModifier4 = /* @__PURE__ */ createRESTModule(getModifier3);
var listModifiers4 = /* @__PURE__ */ createRESTModule(listModifiers3);
var updateModifier4 = /* @__PURE__ */ createRESTModule(updateModifier3);
var bulkCreateModifiers4 = /* @__PURE__ */ createRESTModule(bulkCreateModifiers3);
var bulkUpdateModifiers4 = /* @__PURE__ */ createRESTModule(bulkUpdateModifiers3);
var deleteModifier4 = /* @__PURE__ */ createRESTModule(deleteModifier3);
var queryModifiers4 = /* @__PURE__ */ createRESTModule(queryModifiers3);
var countModifiers4 = /* @__PURE__ */ createRESTModule(countModifiers3);
var bulkDeleteModifiers4 = /* @__PURE__ */ createRESTModule(bulkDeleteModifiers3);
var onItemModifierCreated2 = createEventModule(
  onItemModifierCreated
);
var onItemModifierDeleted2 = createEventModule(
  onItemModifierDeleted
);
var onItemModifierUpdated2 = createEventModule(
  onItemModifierUpdated
);
export {
  SortOrder,
  WebhookIdentityType,
  bulkCreateModifiers4 as bulkCreateModifiers,
  bulkDeleteModifiers4 as bulkDeleteModifiers,
  bulkUpdateModifiers4 as bulkUpdateModifiers,
  countModifiers4 as countModifiers,
  createModifier4 as createModifier,
  deleteModifier4 as deleteModifier,
  getModifier4 as getModifier,
  listModifiers4 as listModifiers,
  onItemModifierCreated2 as onItemModifierCreated,
  onItemModifierDeleted2 as onItemModifierDeleted,
  onItemModifierUpdated2 as onItemModifierUpdated,
  queryModifiers4 as queryModifiers,
  updateModifier4 as updateModifier
};
