// @wix/patterns/dist/cjs/components/CollectionViewsDropdown/modals/RenameView/CollectionRenameViewModal.uni.driver.js
"use strict";

exports.__esModule = true;
exports.CollectionRenameViewModalUniDriver = void 0;
var _unidriver = require("@wix/design-system/dist/testkit/unidriver");
var _unidriver2 = require("../../../../unidriver");
const CollectionRenameViewModalUniDriver = (base, body) => {
  const getModalBase = () => body.$('[data-hook="rename-view-modal"]');
  const renameViewModalDriver = () => (0, _unidriver.CustomModalLayoutUniDriver)(getModalBase(), body);
  const viewNameFormFiledDriver = () => (0, _unidriver.FormFieldUniDriver)(body.$('[data-hook="rename-view-modal-form-field"]'), body);
  const viewNameInputDriver = () => (0, _unidriver.InputUniDriver)(body.$('[data-hook="rename-view-modal-input"]'), body);
  const getSaveButton = () => {
    return renameViewModalDriver().getPrimaryButtonDriver();
  };
  const getCancelButton = () => {
    return renameViewModalDriver().getSecondaryButtonDriver();
  };
  return {
    ...(0, _unidriver2.baseUniDriverFactory)(base),
    exists: () => renameViewModalDriver().exists(),
    getTitle: () => renameViewModalDriver().getTitleText(),
    getSubTitle: () => renameViewModalDriver().getSubtitleText(),
    getSaveButtonText: async () => {
      const saveButton = await getSaveButton();
      return saveButton.getButtonTextContent();
    },
    isSaveButtonDisabled: async () => {
      const saveButton = await getSaveButton();
      return saveButton.isButtonDisabled();
    },
    clickSaveButton: async () => {
      const saveButton = await getSaveButton();
      await saveButton.click();
    },
    getCancelButtonText: async () => {
      const cancelButton = await getCancelButton();
      return cancelButton.getButtonTextContent();
    },
    clickCancelButton: async () => {
      const cancelButton = await getCancelButton();
      await cancelButton.click();
    },
    clickCloseButton: () => renameViewModalDriver().clickCloseButton(),
    viewNameFormFieldExists: () => viewNameFormFiledDriver().exists(),
    getViewNameInputValue: () => viewNameInputDriver().getValue(),
    getViewNameFormFieldTitle: () => viewNameFormFiledDriver().getLabel(),
    getViewNameInputPlaceholder: () => viewNameInputDriver().getPlaceholder(),
    fillViewNameInput: value => viewNameInputDriver().enterText(value),
    getViewNameInputStatusMessage: () => viewNameInputDriver().getStatusMessage(),
    getViewNameHasErrorStatus: () => viewNameInputDriver().hasStatus('error')
  };
};
exports.CollectionRenameViewModalUniDriver = CollectionRenameViewModalUniDriver;
