// @wix/ricos/dist/cjs/plugin-audio/converters/audio-converter.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var audio_converter_exports = {};
__export(audio_converter_exports, {
  audioConverter: () => audioConverter
});
module.exports = __toCommonJS(audio_converter_exports);
var import_v1 = require("./../../ricos-schema/legacy/wix/rich_content/v1/index.js");
const audioConverter = {
  toPm: {
    types: [import_v1.Node_Type.AUDIO],
    // @ts-ignore $TSFixMe
    convert: (node) => ({
      type: import_v1.Node_Type.AUDIO,
      attrs: {
        ...node.audioData,
        id: node.id
      }
    })
  },
  fromPm: {
    types: [import_v1.Node_Type.AUDIO],
    convert: (node) => {
      const { id, ...data } = node.attrs || {};
      return {
        type: import_v1.Node_Type.AUDIO,
        id,
        nodes: [],
        audioData: {
          ...data,
          disableDownload: data.disableDownload || void 0,
          name: data.name || void 0,
          authorName: data.authorName || void 0
        }
      };
    }
  }
};
