// @wix/bex-utils/@wix/ambassador-infra-asyncjobs-v1-async-job/build/cjs/builders.impl.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.aWebhookIdentityType = exports.aIdentificationData = exports.aMessageEnvelope = exports.aActionEvent = exports.aEntityDeletedEvent = exports.aEntityUpdatedEvent = exports.aRestoreInfo = exports.aEntityCreatedEvent = exports.aDomainEvent = exports.aCursors = exports.aPagingMetadataV2 = exports.aListAsyncJobItemsResponse = exports.aStatusFilter = exports.aCursorPaging = exports.aListAsyncJobItemsRequest = exports.aAsyncJobItemAddedEvent = exports.aReportAsyncJobProgressResponse = exports.aReportAsyncJobProgressRequest = exports.aCompleteAsyncJobResponse = exports.aCompleteStatus = exports.aCompleteAsyncJobRequest = exports.aDeleteAsyncJobResponse = exports.aDeleteAsyncJobRequest = exports.aUpdateAsyncJobResponse = exports.aUpdateAsyncJobRequest = exports.aGetAsyncJobResponse = exports.aGetAsyncJobRequest = exports.aCreateAsyncJobResponse = exports.aCreateAsyncJobRequest = exports.aApplicationError = exports.aAsyncJobItem = exports.aAsyncJobItemFailedEvent = exports.aAsyncJobFailedEvent = exports.aAsyncJobFinishedEvent = exports.aCounts = exports.aCreatedBy = exports.aOwner = exports.aCounters = exports.aStatus = exports.aAppIdOptions = exports.aWixUserIdOptions = exports.aMemberIdOptions = exports.aAnonymousVisitorIdOptions = exports.aType = exports.aAsyncJob = void 0;
var metro_common_builders_1 = require("@wix/metro-common-builders");
var types_impl_1 = require("./types.impl");
// @ts-ignore
var callsite_1 = __importDefault(require("callsite"));
function aAsyncJob(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        id: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'FORMAT', value: 'GUID' },
                { type: 'READ_ONLY', value: 'true' },
            ],
        }),
        metadata: {},
        status: aStatus(),
        counters: aCounters(),
        createdDate: (0, metro_common_builders_1.someDate)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        updatedDate: (0, metro_common_builders_1.someDate)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        owner: aOwner(),
        createdBy: aCreatedBy(),
        counts: aCounts(),
    }, overrideObject);
}
exports.aAsyncJob = aAsyncJob;
function aType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.Type.UNKNOWN_TYPE; }
    return enumFirstItem;
}
exports.aType = aType;
function aAnonymousVisitorIdOptions(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        anonymousVisitorId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
exports.aAnonymousVisitorIdOptions = aAnonymousVisitorIdOptions;
function aMemberIdOptions(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        memberId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
exports.aMemberIdOptions = aMemberIdOptions;
function aWixUserIdOptions(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        wixUserId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
exports.aWixUserIdOptions = aWixUserIdOptions;
function aAppIdOptions(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        appId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
exports.aAppIdOptions = aAppIdOptions;
function aStatus(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.Status.UNKNOWN; }
    return enumFirstItem;
}
exports.aStatus = aStatus;
function aCounters(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        itemsToProcess: (0, metro_common_builders_1.someNumber)(),
        itemsSucceeded: (0, metro_common_builders_1.someNumber)(),
        itemsFailed: (0, metro_common_builders_1.someNumber)(),
        errorCountByCode: { errorCountByCode: (0, metro_common_builders_1.someNumber)() },
    }, overrideObject);
}
exports.aCounters = aCounters;
function aOwner(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        anonymousVisitorId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
exports.aOwner = aOwner;
function aCreatedBy(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        anonymousVisitorIdOptions: aAnonymousVisitorIdOptions(),
        type: aType(),
    }, overrideObject);
}
exports.aCreatedBy = aCreatedBy;
function aCounts(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        total: (0, metro_common_builders_1.someNumber)(),
        successCount: (0, metro_common_builders_1.someNumber)(),
        failCount: (0, metro_common_builders_1.someNumber)(),
        errorByCodeCount: { errorByCodeCount: (0, metro_common_builders_1.someNumber)() },
    }, overrideObject);
}
exports.aCounts = aCounts;
function aAsyncJobFinishedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        jobId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
exports.aAsyncJobFinishedEvent = aAsyncJobFinishedEvent;
function aAsyncJobFailedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        jobId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
exports.aAsyncJobFailedEvent = aAsyncJobFailedEvent;
function aAsyncJobItemFailedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        jobId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        jobItem: aAsyncJobItem(),
    }, overrideObject);
}
exports.aAsyncJobItemFailedEvent = aAsyncJobItemFailedEvent;
function aAsyncJobItem(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        id: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'FORMAT', value: 'GUID' },
                { type: 'READ_ONLY', value: 'true' },
            ],
        }),
        data: {},
        entityId: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '32' },
                { type: 'IMMUTABLE', value: 'true' },
            ],
        }),
        success: true,
        error: aApplicationError(),
    }, overrideObject);
}
exports.aAsyncJobItem = aAsyncJobItem;
function aApplicationError(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        code: (0, metro_common_builders_1.someString)(),
        description: (0, metro_common_builders_1.someString)(),
        data: {},
    }, overrideObject);
}
exports.aApplicationError = aApplicationError;
function aCreateAsyncJobRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        job: aAsyncJob(),
    }, overrideObject);
}
exports.aCreateAsyncJobRequest = aCreateAsyncJobRequest;
function aCreateAsyncJobResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        job: aAsyncJob(),
    }, overrideObject);
}
exports.aCreateAsyncJobResponse = aCreateAsyncJobResponse;
function aGetAsyncJobRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        jobId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
exports.aGetAsyncJobRequest = aGetAsyncJobRequest;
function aGetAsyncJobResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        job: aAsyncJob(),
    }, overrideObject);
}
exports.aGetAsyncJobResponse = aGetAsyncJobResponse;
function aUpdateAsyncJobRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        job: aAsyncJob(),
        mask: (0, metro_common_builders_1.someRepeated)(),
    }, overrideObject);
}
exports.aUpdateAsyncJobRequest = aUpdateAsyncJobRequest;
function aUpdateAsyncJobResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        job: aAsyncJob(),
    }, overrideObject);
}
exports.aUpdateAsyncJobResponse = aUpdateAsyncJobResponse;
function aDeleteAsyncJobRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        jobId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
exports.aDeleteAsyncJobRequest = aDeleteAsyncJobRequest;
function aDeleteAsyncJobResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({}, overrideObject);
}
exports.aDeleteAsyncJobResponse = aDeleteAsyncJobResponse;
function aCompleteAsyncJobRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        jobId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        status: aCompleteStatus(),
    }, overrideObject);
}
exports.aCompleteAsyncJobRequest = aCompleteAsyncJobRequest;
function aCompleteStatus(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.CompleteStatus.FINISHED; }
    return enumFirstItem;
}
exports.aCompleteStatus = aCompleteStatus;
function aCompleteAsyncJobResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({}, overrideObject);
}
exports.aCompleteAsyncJobResponse = aCompleteAsyncJobResponse;
function aReportAsyncJobProgressRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        jobId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        items: [
            aAsyncJobItem(),
            aAsyncJobItem(),
            aAsyncJobItem(),
            aAsyncJobItem(),
            aAsyncJobItem(),
            aAsyncJobItem(),
            aAsyncJobItem(),
            aAsyncJobItem(),
            aAsyncJobItem(),
            aAsyncJobItem(),
        ],
    }, overrideObject);
}
exports.aReportAsyncJobProgressRequest = aReportAsyncJobProgressRequest;
function aReportAsyncJobProgressResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({}, overrideObject);
}
exports.aReportAsyncJobProgressResponse = aReportAsyncJobProgressResponse;
function aAsyncJobItemAddedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        jobId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        jobItem: aAsyncJobItem(),
    }, overrideObject);
}
exports.aAsyncJobItemAddedEvent = aAsyncJobItemAddedEvent;
function aListAsyncJobItemsRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        jobId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        paging: aCursorPaging(),
        hasError: true,
        statusFilter: aStatusFilter(),
    }, overrideObject);
}
exports.aListAsyncJobItemsRequest = aListAsyncJobItemsRequest;
function aCursorPaging(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        limit: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'MIN' },
                { type: 'MAX', value: '100' },
            ],
        }),
        cursor: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MAX_LENGTH', value: '16000' },
            ],
        }),
    }, overrideObject);
}
exports.aCursorPaging = aCursorPaging;
function aStatusFilter(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.StatusFilter.ALL; }
    return enumFirstItem;
}
exports.aStatusFilter = aStatusFilter;
function aListAsyncJobItemsResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        results: [aAsyncJobItem()],
        pagingMetadata: aPagingMetadataV2(),
    }, overrideObject);
}
exports.aListAsyncJobItemsResponse = aListAsyncJobItemsResponse;
function aPagingMetadataV2(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        count: (0, metro_common_builders_1.someNumber)(),
        cursors: aCursors(),
        hasNext: true,
    }, overrideObject);
}
exports.aPagingMetadataV2 = aPagingMetadataV2;
function aCursors(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        next: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MAX_LENGTH', value: '16000' },
            ],
        }),
        prev: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MAX_LENGTH', value: '16000' },
            ],
        }),
    }, overrideObject);
}
exports.aCursors = aCursors;
function aDomainEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        createdEvent: aEntityCreatedEvent(),
        id: (0, metro_common_builders_1.someString)(),
        entityFqdn: (0, metro_common_builders_1.someString)(),
        slug: (0, metro_common_builders_1.someString)(),
        entityId: (0, metro_common_builders_1.someString)(),
        eventTime: (0, metro_common_builders_1.someDate)(),
        triggeredByAnonymizeRequest: true,
        originatedFrom: (0, metro_common_builders_1.someString)(),
        entityEventSequence: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'DECIMAL_VALUE' }],
        }),
    }, overrideObject);
}
exports.aDomainEvent = aDomainEvent;
function aEntityCreatedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        entityAsJson: (0, metro_common_builders_1.someString)(),
        triggeredByUndelete: true,
        restoreInfo: aRestoreInfo(),
        additionalMetadataAsJson: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aEntityCreatedEvent = aEntityCreatedEvent;
function aRestoreInfo(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        deletedDate: (0, metro_common_builders_1.someDate)(),
    }, overrideObject);
}
exports.aRestoreInfo = aRestoreInfo;
function aEntityUpdatedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        currentEntityAsJson: (0, metro_common_builders_1.someString)(),
        previousEntityAsJson: (0, metro_common_builders_1.someString)(),
        modifiedFields: { modifiedFields: (0, metro_common_builders_1.someAny)() },
        additionalMetadataAsJson: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aEntityUpdatedEvent = aEntityUpdatedEvent;
function aEntityDeletedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        movedToTrash: true,
        deletedEntityAsJson: (0, metro_common_builders_1.someString)(),
        additionalMetadataAsJson: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aEntityDeletedEvent = aEntityDeletedEvent;
function aActionEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        bodyAsJson: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aActionEvent = aActionEvent;
function aMessageEnvelope(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        instanceId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        eventType: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '150' }],
        }),
        identity: aIdentificationData(),
        data: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aMessageEnvelope = aMessageEnvelope;
function aIdentificationData(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        anonymousVisitorId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        identityType: aWebhookIdentityType(),
    }, overrideObject);
}
exports.aIdentificationData = aIdentificationData;
function aWebhookIdentityType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.WebhookIdentityType.UNKNOWN; }
    return enumFirstItem;
}
exports.aWebhookIdentityType = aWebhookIdentityType;
function isCurrentBuilderWasPreviouslyCalled() {
    var callChainFunctionNames = (0, callsite_1.default)().map(function (item) {
        return item.getFunctionName();
    });
    var callChainWithoutCurrentBuilder = callChainFunctionNames.splice(2);
    var currentBuilderName = callChainFunctionNames[1];
    return callChainWithoutCurrentBuilder.some(function (functionName) { return functionName === currentBuilderName; });
}
__exportStar(require("@wix/metro-common-builders"), exports);
