// @wix/bex-utils/@wix/ambassador-infra-asyncjobs-v1-async-job/build/cjs/http.impl.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.listAsyncJobItems = exports.reportAsyncJobProgress = exports.completeAsyncJob = exports.deleteAsyncJob = exports.updateAsyncJob = exports.getAsyncJob = exports.createAsyncJob = void 0;
var metro_runtime_1 = require("@wix/metro-runtime");
var ambassador_1 = require("@wix/metro-runtime/ambassador");
var metro_runtime_2 = require("@wix/metro-runtime");
var _asyncJob = {
    createdDate: 'google.protobuf.Timestamp',
    updatedDate: 'google.protobuf.Timestamp',
};
var _completeAsyncJobRequest = {};
var _completeAsyncJobResponse = {};
var _createAsyncJobRequest = { job: '_asyncJob' };
var _createAsyncJobResponse = { job: '_asyncJob' };
var _deleteAsyncJobRequest = {};
var _deleteAsyncJobResponse = {};
var _getAsyncJobRequest = {};
var _getAsyncJobResponse = { job: '_asyncJob' };
var _listAsyncJobItemsRequest = {};
var _listAsyncJobItemsResponse = {};
var _reportAsyncJobProgressRequest = {};
var _reportAsyncJobProgressResponse = {};
var _updateAsyncJobRequest = {
    mask: 'google.protobuf.FieldMask',
    job: '_asyncJob',
};
var _updateAsyncJobResponse = { job: '_asyncJob' };
function resolveWixInfraAsyncjobsV1AsyncJobServiceUrl(opts) {
    var domainToMappings = {
        'api._api_base_domain_': [
            {
                srcPath: '/async-job-service',
                destPath: '',
            },
        ],
        'manage._base_domain_': [
            {
                srcPath: '/_api/async-jobs',
                destPath: '/v1/async-jobs',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/async-jobs',
                destPath: '',
            },
        ],
        _: [
            {
                srcPath: '/_api/async-jobs',
                destPath: '',
            },
        ],
        '*.dev.wix-code.com': [
            {
                srcPath: '/_api/async-jobs',
                destPath: '',
            },
        ],
    };
    return (0, metro_runtime_2.resolveUrl)(Object.assign(opts, { domainToMappings: domainToMappings }));
}
/** Creates an async job. */
function createAsyncJob(payload) {
    var _a = (0, ambassador_1.serializer)(_createAsyncJobRequest, { _asyncJob: _asyncJob }), toReq = _a.toJSON, fromReq = _a.fromJSON;
    var fromRes = (0, ambassador_1.serializer)(_createAsyncJobResponse, {
        _asyncJob: _asyncJob,
    }).fromJSON;
    function __createAsyncJob(_a) {
        var host = _a.host;
        var serializedData = toReq(payload);
        var metadata = {
            entityFqdn: 'wix.infra.asyncjobs.v1.async_job',
            method: 'POST',
            methodFqn: 'wix.infra.asyncjobs.v1.AsyncJobService.CreateAsyncJob',
            url: resolveWixInfraAsyncjobsV1AsyncJobServiceUrl({
                protoPath: '/v1/async-jobs',
                data: serializedData,
                host: host,
            }),
            data: serializedData,
            transformResponse: fromRes,
        };
        return metadata;
    }
    __createAsyncJob.fromReq = fromReq;
    __createAsyncJob.__isAmbassador = true;
    return __createAsyncJob;
}
exports.createAsyncJob = createAsyncJob;
/** Retrieves a job. */
function getAsyncJob(payload) {
    var _a = (0, ambassador_1.serializer)(_getAsyncJobRequest, {}), toReq = _a.toJSON, fromReq = _a.fromJSON;
    var fromRes = (0, ambassador_1.serializer)(_getAsyncJobResponse, { _asyncJob: _asyncJob }).fromJSON;
    function __getAsyncJob(_a) {
        var host = _a.host;
        var serializedData = toReq(payload);
        var metadata = {
            entityFqdn: 'wix.infra.asyncjobs.v1.async_job',
            method: 'GET',
            methodFqn: 'wix.infra.asyncjobs.v1.AsyncJobService.GetAsyncJob',
            url: resolveWixInfraAsyncjobsV1AsyncJobServiceUrl({
                protoPath: '/v1/async-jobs/{jobId}',
                data: serializedData,
                host: host,
            }),
            params: (0, metro_runtime_1.toURLSearchParams)(serializedData),
            transformResponse: fromRes,
        };
        return metadata;
    }
    __getAsyncJob.fromReq = fromReq;
    __getAsyncJob.__isAmbassador = true;
    return __getAsyncJob;
}
exports.getAsyncJob = getAsyncJob;
/** Updates a job. */
function updateAsyncJob(payload) {
    var _a = (0, ambassador_1.serializer)(_updateAsyncJobRequest, { _asyncJob: _asyncJob }), toReq = _a.toJSON, fromReq = _a.fromJSON;
    var fromRes = (0, ambassador_1.serializer)(_updateAsyncJobResponse, {
        _asyncJob: _asyncJob,
    }).fromJSON;
    function __updateAsyncJob(_a) {
        var host = _a.host;
        var serializedData = toReq(payload);
        var metadata = {
            entityFqdn: 'wix.infra.asyncjobs.v1.async_job',
            method: 'PATCH',
            methodFqn: 'wix.infra.asyncjobs.v1.AsyncJobService.UpdateAsyncJob',
            url: resolveWixInfraAsyncjobsV1AsyncJobServiceUrl({
                protoPath: '/v1/async-jobs/{job.id}',
                data: serializedData,
                host: host,
            }),
            data: serializedData,
            transformResponse: fromRes,
        };
        return metadata;
    }
    __updateAsyncJob.fromReq = fromReq;
    __updateAsyncJob.__isAmbassador = true;
    return __updateAsyncJob;
}
exports.updateAsyncJob = updateAsyncJob;
/** Deletes a job. */
function deleteAsyncJob(payload) {
    var _a = (0, ambassador_1.serializer)(_deleteAsyncJobRequest, {}), toReq = _a.toJSON, fromReq = _a.fromJSON;
    var fromRes = (0, ambassador_1.serializer)(_deleteAsyncJobResponse, {}).fromJSON;
    function __deleteAsyncJob(_a) {
        var host = _a.host;
        var serializedData = toReq(payload);
        var metadata = {
            entityFqdn: 'wix.infra.asyncjobs.v1.async_job',
            method: 'DELETE',
            methodFqn: 'wix.infra.asyncjobs.v1.AsyncJobService.DeleteAsyncJob',
            url: resolveWixInfraAsyncjobsV1AsyncJobServiceUrl({
                protoPath: '/v1/async-jobs/{jobId}',
                data: serializedData,
                host: host,
            }),
            params: (0, metro_runtime_1.toURLSearchParams)(serializedData),
            transformResponse: fromRes,
        };
        return metadata;
    }
    __deleteAsyncJob.fromReq = fromReq;
    __deleteAsyncJob.__isAmbassador = true;
    return __deleteAsyncJob;
}
exports.deleteAsyncJob = deleteAsyncJob;
/** Completes a job. */
function completeAsyncJob(payload) {
    var _a = (0, ambassador_1.serializer)(_completeAsyncJobRequest, {}), toReq = _a.toJSON, fromReq = _a.fromJSON;
    var fromRes = (0, ambassador_1.serializer)(_completeAsyncJobResponse, {}).fromJSON;
    function __completeAsyncJob(_a) {
        var host = _a.host;
        var serializedData = toReq(payload);
        var metadata = {
            entityFqdn: 'wix.infra.asyncjobs.v1.async_job',
            method: 'PUT',
            methodFqn: 'wix.infra.asyncjobs.v1.AsyncJobService.CompleteAsyncJob',
            url: resolveWixInfraAsyncjobsV1AsyncJobServiceUrl({
                protoPath: '/v1/async-jobs/{jobId}/complete',
                data: serializedData,
                host: host,
            }),
            data: serializedData,
            transformResponse: fromRes,
        };
        return metadata;
    }
    __completeAsyncJob.fromReq = fromReq;
    __completeAsyncJob.__isAmbassador = true;
    return __completeAsyncJob;
}
exports.completeAsyncJob = completeAsyncJob;
/** Reports job progress. */
function reportAsyncJobProgress(payload) {
    var _a = (0, ambassador_1.serializer)(_reportAsyncJobProgressRequest, {}), toReq = _a.toJSON, fromReq = _a.fromJSON;
    var fromRes = (0, ambassador_1.serializer)(_reportAsyncJobProgressResponse, {}).fromJSON;
    function __reportAsyncJobProgress(_a) {
        var host = _a.host;
        var serializedData = toReq(payload);
        var metadata = {
            entityFqdn: 'wix.infra.asyncjobs.v1.async_job',
            method: 'POST',
            methodFqn: 'wix.infra.asyncjobs.v1.AsyncJobService.ReportAsyncJobProgress',
            url: resolveWixInfraAsyncjobsV1AsyncJobServiceUrl({
                protoPath: '/v1/async-jobs/{jobId}/items/report',
                data: serializedData,
                host: host,
            }),
            data: serializedData,
            transformResponse: fromRes,
        };
        return metadata;
    }
    __reportAsyncJobProgress.fromReq = fromReq;
    __reportAsyncJobProgress.__isAmbassador = true;
    return __reportAsyncJobProgress;
}
exports.reportAsyncJobProgress = reportAsyncJobProgress;
/** Retrieves a list of job items. */
function listAsyncJobItems(payload) {
    var _a = (0, ambassador_1.serializer)(_listAsyncJobItemsRequest, {}), toReq = _a.toJSON, fromReq = _a.fromJSON;
    var fromRes = (0, ambassador_1.serializer)(_listAsyncJobItemsResponse, {}).fromJSON;
    function __listAsyncJobItems(_a) {
        var host = _a.host;
        var serializedData = toReq(payload);
        var metadata = {
            entityFqdn: 'wix.infra.asyncjobs.v1.async_job',
            method: 'GET',
            methodFqn: 'wix.infra.asyncjobs.v1.AsyncJobService.ListAsyncJobItems',
            url: resolveWixInfraAsyncjobsV1AsyncJobServiceUrl({
                protoPath: '/v1/async-jobs/{jobId}/items',
                data: serializedData,
                host: host,
            }),
            params: (0, metro_runtime_1.toURLSearchParams)(serializedData),
            transformResponse: fromRes,
        };
        return metadata;
    }
    __listAsyncJobItems.fromReq = fromReq;
    __listAsyncJobItems.__isAmbassador = true;
    return __listAsyncJobItems;
}
exports.listAsyncJobItems = listAsyncJobItems;
