// @wix/bex-utils/@wix/ambassador-infra-asyncjobs-v1-async-job/build/cjs/meta.impl.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listAsyncJobItems = exports.reportAsyncJobProgress = exports.completeAsyncJob = exports.deleteAsyncJob = exports.updateAsyncJob = exports.getAsyncJob = exports.createAsyncJob = void 0;
var httpWixInfraAsyncjobsV1AsyncJob = __importStar(require("./http.impl"));
function createAsyncJob() {
    var payload = {};
    var getRequestOptions = httpWixInfraAsyncjobsV1AsyncJob.createAsyncJob(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/async-jobs',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createAsyncJob = createAsyncJob;
function getAsyncJob() {
    var payload = { jobId: ':jobId' };
    var getRequestOptions = httpWixInfraAsyncjobsV1AsyncJob.getAsyncJob(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/v1/async-jobs/{jobId}',
        pathParams: { jobId: 'jobId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getAsyncJob = getAsyncJob;
function updateAsyncJob() {
    var payload = { job: { id: ':jobId' } };
    var getRequestOptions = httpWixInfraAsyncjobsV1AsyncJob.updateAsyncJob(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'PATCH',
        path: '/v1/async-jobs/{job.id}',
        pathParams: { jobId: 'jobId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateAsyncJob = updateAsyncJob;
function deleteAsyncJob() {
    var payload = { jobId: ':jobId' };
    var getRequestOptions = httpWixInfraAsyncjobsV1AsyncJob.deleteAsyncJob(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'DELETE',
        path: '/v1/async-jobs/{jobId}',
        pathParams: { jobId: 'jobId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteAsyncJob = deleteAsyncJob;
function completeAsyncJob() {
    var payload = { jobId: ':jobId' };
    var getRequestOptions = httpWixInfraAsyncjobsV1AsyncJob.completeAsyncJob(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'PUT',
        path: '/v1/async-jobs/{jobId}/complete',
        pathParams: { jobId: 'jobId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.completeAsyncJob = completeAsyncJob;
function reportAsyncJobProgress() {
    var payload = { jobId: ':jobId' };
    var getRequestOptions = httpWixInfraAsyncjobsV1AsyncJob.reportAsyncJobProgress(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/async-jobs/{jobId}/items/report',
        pathParams: { jobId: 'jobId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.reportAsyncJobProgress = reportAsyncJobProgress;
function listAsyncJobItems() {
    var payload = { jobId: ':jobId' };
    var getRequestOptions = httpWixInfraAsyncjobsV1AsyncJob.listAsyncJobItems(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/v1/async-jobs/{jobId}/items',
        pathParams: { jobId: 'jobId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listAsyncJobItems = listAsyncJobItems;
