// @wix/auto_sdk_online-programs_steps/build/internal/es/index.typings.mjs
// src/online-programs-v3-step-steps.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/online-programs-v3-step-steps.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKFloatToRESTFloat } from "@wix/sdk-runtime/transformations/float";
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixOnlineprogramsStepsV3StepsServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/program-steps-service",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/program-steps-service",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/program-steps-service",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/program-steps-service",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/program-steps-service",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/program-steps-service",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/program-steps-service",
        destPath: ""
      },
      {
        srcPath: "/program-steps-service",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/program-steps-service",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/program-steps-service",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/program-steps-service",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/program-steps-service",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/program-steps-service",
        destPath: ""
      },
      {
        srcPath: "/_api/program-steps-service",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/program-steps-service",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_online-programs_steps";
function createStep(payload) {
  function __createStep({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "step.createdDate" },
          { path: "step.updatedDate" },
          { path: "step.videoOptions.video.urlExpirationDate" },
          { path: "step.videoOptions.video.resolutions.urlExpirationDate" },
          {
            path: "step.videoOptions.video.resolutions.poster.urlExpirationDate"
          },
          { path: "step.videoOptions.video.posters.urlExpirationDate" },
          { path: "step.videoOptions.coverImage.urlExpirationDate" },
          { path: "step.description.image.urlExpirationDate" },
          { path: "step.description.video.urlExpirationDate" },
          { path: "step.description.video.resolutions.urlExpirationDate" },
          {
            path: "step.description.video.resolutions.poster.urlExpirationDate"
          },
          { path: "step.description.video.posters.urlExpirationDate" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "step.ordering" },
          { path: "step.videoOptions.requiredCompletionPercentage" },
          { path: "step.videoOptions.video.resolutions.poster.focalPoint.x" },
          { path: "step.videoOptions.video.resolutions.poster.focalPoint.y" },
          { path: "step.videoOptions.video.posters.focalPoint.x" },
          { path: "step.videoOptions.video.posters.focalPoint.y" },
          { path: "step.videoOptions.coverImage.focalPoint.x" },
          { path: "step.videoOptions.coverImage.focalPoint.y" },
          { path: "step.description.image.focalPoint.x" },
          { path: "step.description.image.focalPoint.y" },
          { path: "step.description.video.resolutions.poster.focalPoint.x" },
          { path: "step.description.video.resolutions.poster.focalPoint.y" },
          { path: "step.description.video.posters.focalPoint.x" },
          { path: "step.description.video.posters.focalPoint.y" },
          { path: "step.quizOptions.passingGrade" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.online_programs.v3.step",
      method: "POST",
      methodFqn: "wix.onlineprograms.steps.v3.StepsService.CreateStep",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsStepsV3StepsServiceUrl({
        protoPath: "/v3/steps",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "step.createdDate" },
            { path: "step.updatedDate" },
            { path: "step.videoOptions.video.urlExpirationDate" },
            { path: "step.videoOptions.video.resolutions.urlExpirationDate" },
            {
              path: "step.videoOptions.video.resolutions.poster.urlExpirationDate"
            },
            { path: "step.videoOptions.video.posters.urlExpirationDate" },
            { path: "step.videoOptions.coverImage.urlExpirationDate" },
            { path: "step.description.image.urlExpirationDate" },
            { path: "step.description.video.urlExpirationDate" },
            { path: "step.description.video.resolutions.urlExpirationDate" },
            {
              path: "step.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "step.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "step.ordering" },
            { path: "step.videoOptions.requiredCompletionPercentage" },
            {
              path: "step.videoOptions.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "step.videoOptions.video.resolutions.poster.focalPoint.y"
            },
            { path: "step.videoOptions.video.posters.focalPoint.x" },
            { path: "step.videoOptions.video.posters.focalPoint.y" },
            { path: "step.videoOptions.coverImage.focalPoint.x" },
            { path: "step.videoOptions.coverImage.focalPoint.y" },
            { path: "step.description.image.focalPoint.x" },
            { path: "step.description.image.focalPoint.y" },
            {
              path: "step.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "step.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "step.description.video.posters.focalPoint.x" },
            { path: "step.description.video.posters.focalPoint.y" },
            { path: "step.quizOptions.passingGrade" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createStep;
}
function bulkCreateStep(payload) {
  function __bulkCreateStep({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "steps.createdDate" },
          { path: "steps.updatedDate" },
          { path: "steps.videoOptions.video.urlExpirationDate" },
          { path: "steps.videoOptions.video.resolutions.urlExpirationDate" },
          {
            path: "steps.videoOptions.video.resolutions.poster.urlExpirationDate"
          },
          { path: "steps.videoOptions.video.posters.urlExpirationDate" },
          { path: "steps.videoOptions.coverImage.urlExpirationDate" },
          { path: "steps.description.image.urlExpirationDate" },
          { path: "steps.description.video.urlExpirationDate" },
          { path: "steps.description.video.resolutions.urlExpirationDate" },
          {
            path: "steps.description.video.resolutions.poster.urlExpirationDate"
          },
          { path: "steps.description.video.posters.urlExpirationDate" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "steps.ordering" },
          { path: "steps.videoOptions.requiredCompletionPercentage" },
          { path: "steps.videoOptions.video.resolutions.poster.focalPoint.x" },
          { path: "steps.videoOptions.video.resolutions.poster.focalPoint.y" },
          { path: "steps.videoOptions.video.posters.focalPoint.x" },
          { path: "steps.videoOptions.video.posters.focalPoint.y" },
          { path: "steps.videoOptions.coverImage.focalPoint.x" },
          { path: "steps.videoOptions.coverImage.focalPoint.y" },
          { path: "steps.description.image.focalPoint.x" },
          { path: "steps.description.image.focalPoint.y" },
          { path: "steps.description.video.resolutions.poster.focalPoint.x" },
          { path: "steps.description.video.resolutions.poster.focalPoint.y" },
          { path: "steps.description.video.posters.focalPoint.x" },
          { path: "steps.description.video.posters.focalPoint.y" },
          { path: "steps.quizOptions.passingGrade" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.online_programs.v3.step",
      method: "POST",
      methodFqn: "wix.onlineprograms.steps.v3.StepsService.BulkCreateStep",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsStepsV3StepsServiceUrl({
        protoPath: "/v3/bulk/steps/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.videoOptions.video.urlExpirationDate" },
            {
              path: "results.item.videoOptions.video.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.videoOptions.video.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.videoOptions.video.posters.urlExpirationDate"
            },
            {
              path: "results.item.videoOptions.coverImage.urlExpirationDate"
            },
            { path: "results.item.description.image.urlExpirationDate" },
            { path: "results.item.description.video.urlExpirationDate" },
            {
              path: "results.item.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.description.video.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.description.video.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.ordering" },
            {
              path: "results.item.videoOptions.requiredCompletionPercentage"
            },
            {
              path: "results.item.videoOptions.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.videoOptions.video.resolutions.poster.focalPoint.y"
            },
            { path: "results.item.videoOptions.video.posters.focalPoint.x" },
            { path: "results.item.videoOptions.video.posters.focalPoint.y" },
            { path: "results.item.videoOptions.coverImage.focalPoint.x" },
            { path: "results.item.videoOptions.coverImage.focalPoint.y" },
            { path: "results.item.description.image.focalPoint.x" },
            { path: "results.item.description.image.focalPoint.y" },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "results.item.description.video.posters.focalPoint.x" },
            { path: "results.item.description.video.posters.focalPoint.y" },
            { path: "results.item.quizOptions.passingGrade" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateStep;
}
function bulkCreateStepMigration(payload) {
  function __bulkCreateStepMigration({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "steps.createdDate" },
          { path: "steps.updatedDate" },
          { path: "steps.videoOptions.video.urlExpirationDate" },
          { path: "steps.videoOptions.video.resolutions.urlExpirationDate" },
          {
            path: "steps.videoOptions.video.resolutions.poster.urlExpirationDate"
          },
          { path: "steps.videoOptions.video.posters.urlExpirationDate" },
          { path: "steps.videoOptions.coverImage.urlExpirationDate" },
          { path: "steps.description.image.urlExpirationDate" },
          { path: "steps.description.video.urlExpirationDate" },
          { path: "steps.description.video.resolutions.urlExpirationDate" },
          {
            path: "steps.description.video.resolutions.poster.urlExpirationDate"
          },
          { path: "steps.description.video.posters.urlExpirationDate" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "steps.ordering" },
          { path: "steps.videoOptions.requiredCompletionPercentage" },
          { path: "steps.videoOptions.video.resolutions.poster.focalPoint.x" },
          { path: "steps.videoOptions.video.resolutions.poster.focalPoint.y" },
          { path: "steps.videoOptions.video.posters.focalPoint.x" },
          { path: "steps.videoOptions.video.posters.focalPoint.y" },
          { path: "steps.videoOptions.coverImage.focalPoint.x" },
          { path: "steps.videoOptions.coverImage.focalPoint.y" },
          { path: "steps.description.image.focalPoint.x" },
          { path: "steps.description.image.focalPoint.y" },
          { path: "steps.description.video.resolutions.poster.focalPoint.x" },
          { path: "steps.description.video.resolutions.poster.focalPoint.y" },
          { path: "steps.description.video.posters.focalPoint.x" },
          { path: "steps.description.video.posters.focalPoint.y" },
          { path: "steps.quizOptions.passingGrade" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.online_programs.v3.step",
      method: "POST",
      methodFqn: "wix.onlineprograms.steps.v3.StepsService.BulkCreateStepMigration",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsStepsV3StepsServiceUrl({
        protoPath: "/v3/bulk/steps/create/migration",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.videoOptions.video.urlExpirationDate" },
            {
              path: "results.item.videoOptions.video.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.videoOptions.video.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.videoOptions.video.posters.urlExpirationDate"
            },
            {
              path: "results.item.videoOptions.coverImage.urlExpirationDate"
            },
            { path: "results.item.description.image.urlExpirationDate" },
            { path: "results.item.description.video.urlExpirationDate" },
            {
              path: "results.item.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.description.video.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.description.video.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.ordering" },
            {
              path: "results.item.videoOptions.requiredCompletionPercentage"
            },
            {
              path: "results.item.videoOptions.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.videoOptions.video.resolutions.poster.focalPoint.y"
            },
            { path: "results.item.videoOptions.video.posters.focalPoint.x" },
            { path: "results.item.videoOptions.video.posters.focalPoint.y" },
            { path: "results.item.videoOptions.coverImage.focalPoint.x" },
            { path: "results.item.videoOptions.coverImage.focalPoint.y" },
            { path: "results.item.description.image.focalPoint.x" },
            { path: "results.item.description.image.focalPoint.y" },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "results.item.description.video.posters.focalPoint.x" },
            { path: "results.item.description.video.posters.focalPoint.y" },
            { path: "results.item.quizOptions.passingGrade" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateStepMigration;
}
function getStep(payload) {
  function __getStep({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.step",
      method: "GET",
      methodFqn: "wix.onlineprograms.steps.v3.StepsService.GetStep",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsStepsV3StepsServiceUrl({
        protoPath: "/v3/steps/{stepId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "step.createdDate" },
            { path: "step.updatedDate" },
            { path: "step.videoOptions.video.urlExpirationDate" },
            { path: "step.videoOptions.video.resolutions.urlExpirationDate" },
            {
              path: "step.videoOptions.video.resolutions.poster.urlExpirationDate"
            },
            { path: "step.videoOptions.video.posters.urlExpirationDate" },
            { path: "step.videoOptions.coverImage.urlExpirationDate" },
            { path: "step.description.image.urlExpirationDate" },
            { path: "step.description.video.urlExpirationDate" },
            { path: "step.description.video.resolutions.urlExpirationDate" },
            {
              path: "step.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "step.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "step.ordering" },
            { path: "step.videoOptions.requiredCompletionPercentage" },
            {
              path: "step.videoOptions.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "step.videoOptions.video.resolutions.poster.focalPoint.y"
            },
            { path: "step.videoOptions.video.posters.focalPoint.x" },
            { path: "step.videoOptions.video.posters.focalPoint.y" },
            { path: "step.videoOptions.coverImage.focalPoint.x" },
            { path: "step.videoOptions.coverImage.focalPoint.y" },
            { path: "step.description.image.focalPoint.x" },
            { path: "step.description.image.focalPoint.y" },
            {
              path: "step.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "step.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "step.description.video.posters.focalPoint.x" },
            { path: "step.description.video.posters.focalPoint.y" },
            { path: "step.quizOptions.passingGrade" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getStep;
}
function updateStep(payload) {
  function __updateStep({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "step.createdDate" },
          { path: "step.updatedDate" },
          { path: "step.videoOptions.video.urlExpirationDate" },
          { path: "step.videoOptions.video.resolutions.urlExpirationDate" },
          {
            path: "step.videoOptions.video.resolutions.poster.urlExpirationDate"
          },
          { path: "step.videoOptions.video.posters.urlExpirationDate" },
          { path: "step.videoOptions.coverImage.urlExpirationDate" },
          { path: "step.description.image.urlExpirationDate" },
          { path: "step.description.video.urlExpirationDate" },
          { path: "step.description.video.resolutions.urlExpirationDate" },
          {
            path: "step.description.video.resolutions.poster.urlExpirationDate"
          },
          { path: "step.description.video.posters.urlExpirationDate" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "step.ordering" },
          { path: "step.videoOptions.requiredCompletionPercentage" },
          { path: "step.videoOptions.video.resolutions.poster.focalPoint.x" },
          { path: "step.videoOptions.video.resolutions.poster.focalPoint.y" },
          { path: "step.videoOptions.video.posters.focalPoint.x" },
          { path: "step.videoOptions.video.posters.focalPoint.y" },
          { path: "step.videoOptions.coverImage.focalPoint.x" },
          { path: "step.videoOptions.coverImage.focalPoint.y" },
          { path: "step.description.image.focalPoint.x" },
          { path: "step.description.image.focalPoint.y" },
          { path: "step.description.video.resolutions.poster.focalPoint.x" },
          { path: "step.description.video.resolutions.poster.focalPoint.y" },
          { path: "step.description.video.posters.focalPoint.x" },
          { path: "step.description.video.posters.focalPoint.y" },
          { path: "step.quizOptions.passingGrade" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.online_programs.v3.step",
      method: "PATCH",
      methodFqn: "wix.onlineprograms.steps.v3.StepsService.UpdateStep",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsStepsV3StepsServiceUrl({
        protoPath: "/v3/steps/{step.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "step.createdDate" },
            { path: "step.updatedDate" },
            { path: "step.videoOptions.video.urlExpirationDate" },
            { path: "step.videoOptions.video.resolutions.urlExpirationDate" },
            {
              path: "step.videoOptions.video.resolutions.poster.urlExpirationDate"
            },
            { path: "step.videoOptions.video.posters.urlExpirationDate" },
            { path: "step.videoOptions.coverImage.urlExpirationDate" },
            { path: "step.description.image.urlExpirationDate" },
            { path: "step.description.video.urlExpirationDate" },
            { path: "step.description.video.resolutions.urlExpirationDate" },
            {
              path: "step.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "step.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "step.ordering" },
            { path: "step.videoOptions.requiredCompletionPercentage" },
            {
              path: "step.videoOptions.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "step.videoOptions.video.resolutions.poster.focalPoint.y"
            },
            { path: "step.videoOptions.video.posters.focalPoint.x" },
            { path: "step.videoOptions.video.posters.focalPoint.y" },
            { path: "step.videoOptions.coverImage.focalPoint.x" },
            { path: "step.videoOptions.coverImage.focalPoint.y" },
            { path: "step.description.image.focalPoint.x" },
            { path: "step.description.image.focalPoint.y" },
            {
              path: "step.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "step.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "step.description.video.posters.focalPoint.x" },
            { path: "step.description.video.posters.focalPoint.y" },
            { path: "step.quizOptions.passingGrade" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateStep;
}
function bulkUpdateSteps(payload) {
  function __bulkUpdateSteps({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "steps.fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "steps.step.createdDate" },
          { path: "steps.step.updatedDate" },
          { path: "steps.step.videoOptions.video.urlExpirationDate" },
          {
            path: "steps.step.videoOptions.video.resolutions.urlExpirationDate"
          },
          {
            path: "steps.step.videoOptions.video.resolutions.poster.urlExpirationDate"
          },
          { path: "steps.step.videoOptions.video.posters.urlExpirationDate" },
          { path: "steps.step.videoOptions.coverImage.urlExpirationDate" },
          { path: "steps.step.description.image.urlExpirationDate" },
          { path: "steps.step.description.video.urlExpirationDate" },
          {
            path: "steps.step.description.video.resolutions.urlExpirationDate"
          },
          {
            path: "steps.step.description.video.resolutions.poster.urlExpirationDate"
          },
          { path: "steps.step.description.video.posters.urlExpirationDate" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "steps.step.ordering" },
          { path: "steps.step.videoOptions.requiredCompletionPercentage" },
          {
            path: "steps.step.videoOptions.video.resolutions.poster.focalPoint.x"
          },
          {
            path: "steps.step.videoOptions.video.resolutions.poster.focalPoint.y"
          },
          { path: "steps.step.videoOptions.video.posters.focalPoint.x" },
          { path: "steps.step.videoOptions.video.posters.focalPoint.y" },
          { path: "steps.step.videoOptions.coverImage.focalPoint.x" },
          { path: "steps.step.videoOptions.coverImage.focalPoint.y" },
          { path: "steps.step.description.image.focalPoint.x" },
          { path: "steps.step.description.image.focalPoint.y" },
          {
            path: "steps.step.description.video.resolutions.poster.focalPoint.x"
          },
          {
            path: "steps.step.description.video.resolutions.poster.focalPoint.y"
          },
          { path: "steps.step.description.video.posters.focalPoint.x" },
          { path: "steps.step.description.video.posters.focalPoint.y" },
          { path: "steps.step.quizOptions.passingGrade" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.online_programs.v3.step",
      method: "POST",
      methodFqn: "wix.onlineprograms.steps.v3.StepsService.BulkUpdateSteps",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsStepsV3StepsServiceUrl({
        protoPath: "/v3/bulk/steps/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.videoOptions.video.urlExpirationDate" },
            {
              path: "results.item.videoOptions.video.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.videoOptions.video.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.videoOptions.video.posters.urlExpirationDate"
            },
            {
              path: "results.item.videoOptions.coverImage.urlExpirationDate"
            },
            { path: "results.item.description.image.urlExpirationDate" },
            { path: "results.item.description.video.urlExpirationDate" },
            {
              path: "results.item.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.description.video.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.description.video.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.ordering" },
            {
              path: "results.item.videoOptions.requiredCompletionPercentage"
            },
            {
              path: "results.item.videoOptions.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.videoOptions.video.resolutions.poster.focalPoint.y"
            },
            { path: "results.item.videoOptions.video.posters.focalPoint.x" },
            { path: "results.item.videoOptions.video.posters.focalPoint.y" },
            { path: "results.item.videoOptions.coverImage.focalPoint.x" },
            { path: "results.item.videoOptions.coverImage.focalPoint.y" },
            { path: "results.item.description.image.focalPoint.x" },
            { path: "results.item.description.image.focalPoint.y" },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "results.item.description.video.posters.focalPoint.x" },
            { path: "results.item.description.video.posters.focalPoint.y" },
            { path: "results.item.quizOptions.passingGrade" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateSteps;
}
function deleteStep(payload) {
  function __deleteStep({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.step",
      method: "DELETE",
      methodFqn: "wix.onlineprograms.steps.v3.StepsService.DeleteStep",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsStepsV3StepsServiceUrl({
        protoPath: "/v3/steps/{stepId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteStep;
}
function querySteps(payload) {
  function __querySteps({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.step",
      method: "POST",
      methodFqn: "wix.onlineprograms.steps.v3.StepsService.QuerySteps",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsStepsV3StepsServiceUrl({
        protoPath: "/v3/steps/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "steps.createdDate" },
            { path: "steps.updatedDate" },
            { path: "steps.videoOptions.video.urlExpirationDate" },
            {
              path: "steps.videoOptions.video.resolutions.urlExpirationDate"
            },
            {
              path: "steps.videoOptions.video.resolutions.poster.urlExpirationDate"
            },
            { path: "steps.videoOptions.video.posters.urlExpirationDate" },
            { path: "steps.videoOptions.coverImage.urlExpirationDate" },
            { path: "steps.description.image.urlExpirationDate" },
            { path: "steps.description.video.urlExpirationDate" },
            { path: "steps.description.video.resolutions.urlExpirationDate" },
            {
              path: "steps.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "steps.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "steps.ordering" },
            { path: "steps.videoOptions.requiredCompletionPercentage" },
            {
              path: "steps.videoOptions.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "steps.videoOptions.video.resolutions.poster.focalPoint.y"
            },
            { path: "steps.videoOptions.video.posters.focalPoint.x" },
            { path: "steps.videoOptions.video.posters.focalPoint.y" },
            { path: "steps.videoOptions.coverImage.focalPoint.x" },
            { path: "steps.videoOptions.coverImage.focalPoint.y" },
            { path: "steps.description.image.focalPoint.x" },
            { path: "steps.description.image.focalPoint.y" },
            {
              path: "steps.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "steps.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "steps.description.video.posters.focalPoint.x" },
            { path: "steps.description.video.posters.focalPoint.y" },
            { path: "steps.quizOptions.passingGrade" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __querySteps;
}
function moveStep(payload) {
  function __moveStep({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.step",
      method: "PATCH",
      methodFqn: "wix.onlineprograms.steps.v3.StepsService.MoveStep",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsStepsV3StepsServiceUrl({
        protoPath: "/v3/steps/{stepId}/move",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "step.createdDate" },
            { path: "step.updatedDate" },
            { path: "step.videoOptions.video.urlExpirationDate" },
            { path: "step.videoOptions.video.resolutions.urlExpirationDate" },
            {
              path: "step.videoOptions.video.resolutions.poster.urlExpirationDate"
            },
            { path: "step.videoOptions.video.posters.urlExpirationDate" },
            { path: "step.videoOptions.coverImage.urlExpirationDate" },
            { path: "step.description.image.urlExpirationDate" },
            { path: "step.description.video.urlExpirationDate" },
            { path: "step.description.video.resolutions.urlExpirationDate" },
            {
              path: "step.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "step.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "step.ordering" },
            { path: "step.videoOptions.requiredCompletionPercentage" },
            {
              path: "step.videoOptions.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "step.videoOptions.video.resolutions.poster.focalPoint.y"
            },
            { path: "step.videoOptions.video.posters.focalPoint.x" },
            { path: "step.videoOptions.video.posters.focalPoint.y" },
            { path: "step.videoOptions.coverImage.focalPoint.x" },
            { path: "step.videoOptions.coverImage.focalPoint.y" },
            { path: "step.description.image.focalPoint.x" },
            { path: "step.description.image.focalPoint.y" },
            {
              path: "step.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "step.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "step.description.video.posters.focalPoint.x" },
            { path: "step.description.video.posters.focalPoint.y" },
            { path: "step.quizOptions.passingGrade" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __moveStep;
}
function cloneStep(payload) {
  function __cloneStep({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.step",
      method: "POST",
      methodFqn: "wix.onlineprograms.steps.v3.StepsService.CloneStep",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsStepsV3StepsServiceUrl({
        protoPath: "/v3/steps/{stepId}/clone",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "step.createdDate" },
            { path: "step.updatedDate" },
            { path: "step.videoOptions.video.urlExpirationDate" },
            { path: "step.videoOptions.video.resolutions.urlExpirationDate" },
            {
              path: "step.videoOptions.video.resolutions.poster.urlExpirationDate"
            },
            { path: "step.videoOptions.video.posters.urlExpirationDate" },
            { path: "step.videoOptions.coverImage.urlExpirationDate" },
            { path: "step.description.image.urlExpirationDate" },
            { path: "step.description.video.urlExpirationDate" },
            { path: "step.description.video.resolutions.urlExpirationDate" },
            {
              path: "step.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "step.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "step.ordering" },
            { path: "step.videoOptions.requiredCompletionPercentage" },
            {
              path: "step.videoOptions.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "step.videoOptions.video.resolutions.poster.focalPoint.y"
            },
            { path: "step.videoOptions.video.posters.focalPoint.x" },
            { path: "step.videoOptions.video.posters.focalPoint.y" },
            { path: "step.videoOptions.coverImage.focalPoint.x" },
            { path: "step.videoOptions.coverImage.focalPoint.y" },
            { path: "step.description.image.focalPoint.x" },
            { path: "step.description.image.focalPoint.y" },
            {
              path: "step.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "step.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "step.description.video.posters.focalPoint.x" },
            { path: "step.description.video.posters.focalPoint.y" },
            { path: "step.quizOptions.passingGrade" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __cloneStep;
}

// src/online-programs-v3-step-steps.universal.ts
import { transformSDKImageToRESTImage } from "@wix/sdk-runtime/transformations/image";
import { transformRESTImageToSDKImage } from "@wix/sdk-runtime/transformations/image";
import { transformSDKVideoV2ToRESTVideoV2 } from "@wix/sdk-runtime/transformations/video-v2";
import { transformRESTVideoV2ToSDKVideoV2 } from "@wix/sdk-runtime/transformations/video-v2";
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var StepType = /* @__PURE__ */ ((StepType2) => {
  StepType2["ARTICLE"] = "ARTICLE";
  StepType2["VIDEO"] = "VIDEO";
  StepType2["QUIZ"] = "QUIZ";
  return StepType2;
})(StepType || {});
var Enum = /* @__PURE__ */ ((Enum2) => {
  Enum2["UNKNOWN"] = "UNKNOWN";
  Enum2["PROTECTED"] = "PROTECTED";
  Enum2["PRIVATE"] = "PRIVATE";
  Enum2["PUBLIC"] = "PUBLIC";
  Enum2["UNCHANGED"] = "UNCHANGED";
  Enum2["GROUP_PROTECTED"] = "GROUP_PROTECTED";
  return Enum2;
})(Enum || {});
var BulkActionType = /* @__PURE__ */ ((BulkActionType2) => {
  BulkActionType2["UNKNOWN_ACTION_TYPE"] = "UNKNOWN_ACTION_TYPE";
  BulkActionType2["INSERT"] = "INSERT";
  BulkActionType2["UPDATE"] = "UPDATE";
  BulkActionType2["DELETE"] = "DELETE";
  return BulkActionType2;
})(BulkActionType || {});
var DescriptionFieldSet = /* @__PURE__ */ ((DescriptionFieldSet2) => {
  DescriptionFieldSet2["STANDARD"] = "STANDARD";
  DescriptionFieldSet2["EXTENDED"] = "EXTENDED";
  return DescriptionFieldSet2;
})(DescriptionFieldSet || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var State = /* @__PURE__ */ ((State2) => {
  State2["Unknown"] = "Unknown";
  State2["Off"] = "Off";
  State2["On"] = "On";
  return State2;
})(State || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createStep2(step, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({
      step,
      programId: options?.programId
    }),
    [
      {
        transformFn: transformSDKVideoV2ToRESTVideoV2,
        paths: [
          { path: "step.videoOptions.video" },
          { path: "step.description.video" }
        ]
      },
      {
        transformFn: transformSDKImageToRESTImage,
        paths: [
          { path: "step.videoOptions.coverImage" },
          { path: "step.description.image" }
        ]
      }
    ]
  );
  const reqOpts = createStep(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTVideoV2ToSDKVideoV2,
          paths: [
            { path: "step.videoOptions.video" },
            { path: "step.description.video" }
          ]
        },
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [
            { path: "step.videoOptions.coverImage" },
            { path: "step.description.image" }
          ]
        }
      ])
    )?.step;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { step: "$[0]", programId: "$[1].programId" },
        singleArgumentUnchanged: false
      },
      ["step", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCreateStep2(steps, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({
      steps,
      returnEntity: options?.returnEntity,
      programId: options?.programId
    }),
    [
      {
        transformFn: transformSDKVideoV2ToRESTVideoV2,
        paths: [
          { path: "steps.videoOptions.video" },
          { path: "steps.description.video" }
        ]
      },
      {
        transformFn: transformSDKImageToRESTImage,
        paths: [
          { path: "steps.videoOptions.coverImage" },
          { path: "steps.description.image" }
        ]
      }
    ]
  );
  const reqOpts = bulkCreateStep(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTVideoV2ToSDKVideoV2,
          paths: [
            { path: "results.item.videoOptions.video" },
            { path: "results.item.description.video" }
          ]
        },
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [
            { path: "results.item.videoOptions.coverImage" },
            { path: "results.item.description.image" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          steps: "$[0]",
          returnEntity: "$[1].returnEntity",
          programId: "$[1].programId"
        },
        singleArgumentUnchanged: false
      },
      ["steps", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCreateStepMigration2(steps, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({
      steps,
      programId: options?.programId
    }),
    [
      {
        transformFn: transformSDKVideoV2ToRESTVideoV2,
        paths: [
          { path: "steps.videoOptions.video" },
          { path: "steps.description.video" }
        ]
      },
      {
        transformFn: transformSDKImageToRESTImage,
        paths: [
          { path: "steps.videoOptions.coverImage" },
          { path: "steps.description.image" }
        ]
      }
    ]
  );
  const reqOpts = bulkCreateStepMigration(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTVideoV2ToSDKVideoV2,
          paths: [
            { path: "results.item.videoOptions.video" },
            { path: "results.item.description.video" }
          ]
        },
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [
            { path: "results.item.videoOptions.coverImage" },
            { path: "results.item.description.image" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          steps: "$[0]",
          programId: "$[1].programId"
        },
        singleArgumentUnchanged: false
      },
      ["steps", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getStep2(stepId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    stepId,
    descriptionFieldSet: options?.descriptionFieldSet
  });
  const reqOpts = getStep(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTVideoV2ToSDKVideoV2,
          paths: [
            { path: "step.videoOptions.video" },
            { path: "step.description.video" }
          ]
        },
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [
            { path: "step.videoOptions.coverImage" },
            { path: "step.description.image" }
          ]
        }
      ])
    )?.step;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          stepId: "$[0]",
          descriptionFieldSet: "$[1].descriptionFieldSet"
        },
        singleArgumentUnchanged: false
      },
      ["stepId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateStep2(_id, step) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({ step: { ...step, id: _id } }),
    [
      {
        transformFn: transformSDKVideoV2ToRESTVideoV2,
        paths: [
          { path: "step.videoOptions.video" },
          { path: "step.description.video" }
        ]
      },
      {
        transformFn: transformSDKImageToRESTImage,
        paths: [
          { path: "step.videoOptions.coverImage" },
          { path: "step.description.image" }
        ]
      }
    ]
  );
  const reqOpts = updateStep(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTVideoV2ToSDKVideoV2,
          paths: [
            { path: "step.videoOptions.video" },
            { path: "step.description.video" }
          ]
        },
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [
            { path: "step.videoOptions.coverImage" },
            { path: "step.description.image" }
          ]
        }
      ])
    )?.step;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { step: "$[1]" },
        explicitPathsToArguments: { "step.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "step"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateSteps2(steps, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({
      steps,
      returnEntity: options?.returnEntity
    }),
    [
      {
        transformFn: transformSDKVideoV2ToRESTVideoV2,
        paths: [
          { path: "steps.step.videoOptions.video" },
          { path: "steps.step.description.video" }
        ]
      },
      {
        transformFn: transformSDKImageToRESTImage,
        paths: [
          { path: "steps.step.videoOptions.coverImage" },
          { path: "steps.step.description.image" }
        ]
      }
    ]
  );
  const reqOpts = bulkUpdateSteps(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTVideoV2ToSDKVideoV2,
          paths: [
            { path: "results.item.videoOptions.video" },
            { path: "results.item.description.video" }
          ]
        },
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [
            { path: "results.item.videoOptions.coverImage" },
            { path: "results.item.description.image" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          steps: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["steps", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteStep2(stepId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ stepId });
  const reqOpts = deleteStep(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { stepId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["stepId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function querySteps2(options) {
  const { httpClient, sideEffects } = arguments[1];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = querySteps({
        ...payload,
        ...options ?? {}
      });
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, options];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [
          {
            transformFn: transformRESTVideoV2ToSDKVideoV2,
            paths: [
              { path: "steps.videoOptions.video" },
              { path: "steps.description.video" }
            ]
          },
          {
            transformFn: transformRESTImageToSDKImage,
            paths: [
              { path: "steps.videoOptions.coverImage" },
              { path: "steps.description.image" }
            ]
          }
        ])
      );
      return {
        items: transformedData?.steps,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function moveStep2(stepId, sectionId, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    stepId,
    sectionId,
    afterStepId: options?.afterStepId,
    revision: options?.revision
  });
  const reqOpts = moveStep(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTVideoV2ToSDKVideoV2,
          paths: [
            { path: "step.videoOptions.video" },
            { path: "step.description.video" }
          ]
        },
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [
            { path: "step.videoOptions.coverImage" },
            { path: "step.description.image" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          stepId: "$[0]",
          sectionId: "$[1]",
          afterStepId: "$[2].afterStepId",
          revision: "$[2].revision"
        },
        singleArgumentUnchanged: false
      },
      ["stepId", "sectionId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function cloneStep2(stepId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ stepId });
  const reqOpts = cloneStep(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTVideoV2ToSDKVideoV2,
          paths: [
            { path: "step.videoOptions.video" },
            { path: "step.description.video" }
          ]
        },
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [
            { path: "step.videoOptions.coverImage" },
            { path: "step.description.image" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { stepId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["stepId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
export {
  BulkActionType,
  DescriptionFieldSet,
  Enum,
  SortOrder,
  State,
  StepType,
  WebhookIdentityType,
  bulkCreateStep2 as bulkCreateStep,
  bulkCreateStepMigration2 as bulkCreateStepMigration,
  bulkUpdateSteps2 as bulkUpdateSteps,
  cloneStep2 as cloneStep,
  createStep2 as createStep,
  deleteStep2 as deleteStep,
  getStep2 as getStep,
  moveStep2 as moveStep,
  querySteps2 as querySteps,
  updateStep2 as updateStep
};
