// @wix/auto_sdk_ecom_manual-tax-mappings/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  JurisdictionTypeOriginal: () => JurisdictionType,
  SortOrderOriginal: () => SortOrder,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  bulkAssignExemptOverrides: () => bulkAssignExemptOverrides2,
  bulkCreateManualTaxMappings: () => bulkCreateManualTaxMappings2,
  bulkUnassignExemptOverrides: () => bulkUnassignExemptOverrides2,
  createManualTaxMapping: () => createManualTaxMapping2,
  deleteManualTaxMapping: () => deleteManualTaxMapping2,
  getManualTaxMapping: () => getManualTaxMapping2,
  listManualTaxMappings: () => listManualTaxMappings2,
  queryManualTaxMappings: () => queryManualTaxMappings2,
  updateExtendedFields: () => updateExtendedFields2,
  updateManualTaxMapping: () => updateManualTaxMapping2
});
module.exports = __toCommonJS(meta_exports);

// src/billing-v1-manual-tax-mapping-manual-tax-mappings.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixBillingV1ManualTaxMappingServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/manual-tax-mappings",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/manual-tax-mappings",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/billing/v1/manual-tax-mappings",
        destPath: "/v1/manual-tax-mappings"
      },
      {
        srcPath: "/billing/v1/bulk/manual-tax-mappings",
        destPath: "/v1/bulk/manual-tax-mappings"
      }
    ],
    _: [
      {
        srcPath: "/billing/v1/manual-tax-mappings",
        destPath: "/v1/manual-tax-mappings"
      },
      {
        srcPath: "/billing/v1/bulk/manual-tax-mappings",
        destPath: "/v1/bulk/manual-tax-mappings"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/billing/v1/manual-tax-mappings",
        destPath: "/v1/manual-tax-mappings"
      },
      {
        srcPath: "/billing/v1/bulk/manual-tax-mappings",
        destPath: "/v1/bulk/manual-tax-mappings"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_manual-tax-mappings";
function createManualTaxMapping(payload) {
  function __createManualTaxMapping({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "manualTaxMapping.createdDate" },
          { path: "manualTaxMapping.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.billing.v1.manual_tax_mapping",
      method: "POST",
      methodFqn: "wix.billing.v1.ManualTaxMappingService.CreateManualTaxMapping",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBillingV1ManualTaxMappingServiceUrl({
        protoPath: "/v1/manual-tax-mappings",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "manualTaxMapping.createdDate" },
            { path: "manualTaxMapping.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createManualTaxMapping;
}
function bulkCreateManualTaxMappings(payload) {
  function __bulkCreateManualTaxMappings({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "manualTaxMappings.createdDate" },
          { path: "manualTaxMappings.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.billing.v1.manual_tax_mapping",
      method: "POST",
      methodFqn: "wix.billing.v1.ManualTaxMappingService.BulkCreateManualTaxMappings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBillingV1ManualTaxMappingServiceUrl({
        protoPath: "/v1/bulk/manual-tax-mappings/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateManualTaxMappings;
}
function getManualTaxMapping(payload) {
  function __getManualTaxMapping({ host }) {
    const metadata = {
      entityFqdn: "wix.billing.v1.manual_tax_mapping",
      method: "GET",
      methodFqn: "wix.billing.v1.ManualTaxMappingService.GetManualTaxMapping",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBillingV1ManualTaxMappingServiceUrl({
        protoPath: "/v1/manual-tax-mappings/{manualTaxMappingId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "manualTaxMapping.createdDate" },
            { path: "manualTaxMapping.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getManualTaxMapping;
}
function updateManualTaxMapping(payload) {
  function __updateManualTaxMapping({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "manualTaxMapping.createdDate" },
          { path: "manualTaxMapping.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.billing.v1.manual_tax_mapping",
      method: "PATCH",
      methodFqn: "wix.billing.v1.ManualTaxMappingService.UpdateManualTaxMapping",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBillingV1ManualTaxMappingServiceUrl({
        protoPath: "/v1/manual-tax-mappings/{manualTaxMapping.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "manualTaxMapping.createdDate" },
            { path: "manualTaxMapping.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateManualTaxMapping;
}
function deleteManualTaxMapping(payload) {
  function __deleteManualTaxMapping({ host }) {
    const metadata = {
      entityFqdn: "wix.billing.v1.manual_tax_mapping",
      method: "DELETE",
      methodFqn: "wix.billing.v1.ManualTaxMappingService.DeleteManualTaxMapping",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBillingV1ManualTaxMappingServiceUrl({
        protoPath: "/v1/manual-tax-mappings/{manualTaxMappingId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteManualTaxMapping;
}
function queryManualTaxMappings(payload) {
  function __queryManualTaxMappings({ host }) {
    const metadata = {
      entityFqdn: "wix.billing.v1.manual_tax_mapping",
      method: "GET",
      methodFqn: "wix.billing.v1.ManualTaxMappingService.QueryManualTaxMappings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBillingV1ManualTaxMappingServiceUrl({
        protoPath: "/v1/manual-tax-mappings/query",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload, true),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "manualTaxMappings.createdDate" },
            { path: "manualTaxMappings.updatedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixBillingV1ManualTaxMappingServiceUrl({
            protoPath: "/v1/manual-tax-mappings/query",
            data: payload,
            host
          }),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryManualTaxMappings;
}
function updateExtendedFields(payload) {
  function __updateExtendedFields({ host }) {
    const metadata = {
      entityFqdn: "wix.billing.v1.manual_tax_mapping",
      method: "POST",
      methodFqn: "wix.billing.v1.ManualTaxMappingService.UpdateExtendedFields",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBillingV1ManualTaxMappingServiceUrl({
        protoPath: "/v1/manual-tax-mappings/{id}/update-extended-fields",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "manualTaxMapping.createdDate" },
            { path: "manualTaxMapping.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateExtendedFields;
}
function bulkAssignExemptOverrides(payload) {
  function __bulkAssignExemptOverrides({ host }) {
    const metadata = {
      entityFqdn: "wix.billing.v1.manual_tax_mapping",
      method: "POST",
      methodFqn: "wix.billing.v1.ManualTaxMappingService.BulkAssignExemptOverrides",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBillingV1ManualTaxMappingServiceUrl({
        protoPath: "/v1/manual-tax-mappings/bulk-assign-exempt-overrides",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkAssignExemptOverrides;
}
function bulkUnassignExemptOverrides(payload) {
  function __bulkUnassignExemptOverrides({ host }) {
    const metadata = {
      entityFqdn: "wix.billing.v1.manual_tax_mapping",
      method: "POST",
      methodFqn: "wix.billing.v1.ManualTaxMappingService.BulkUnassignExemptOverrides",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBillingV1ManualTaxMappingServiceUrl({
        protoPath: "/v1/manual-tax-mappings/bulk-unassign-exempt-overrides",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUnassignExemptOverrides;
}
function listManualTaxMappings(payload) {
  function __listManualTaxMappings({ host }) {
    const metadata = {
      entityFqdn: "wix.billing.v1.manual_tax_mapping",
      method: "GET",
      methodFqn: "wix.billing.v1.ManualTaxMappingService.ListManualTaxMappings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBillingV1ManualTaxMappingServiceUrl({
        protoPath: "/v1/manual-tax-mappings",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "manualTaxMappings.createdDate" },
            { path: "manualTaxMappings.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listManualTaxMappings;
}

// src/billing-v1-manual-tax-mapping-manual-tax-mappings.types.ts
var JurisdictionType = /* @__PURE__ */ ((JurisdictionType2) => {
  JurisdictionType2["UNDEFINED"] = "UNDEFINED";
  JurisdictionType2["COUNTRY"] = "COUNTRY";
  JurisdictionType2["STATE"] = "STATE";
  JurisdictionType2["COUNTY"] = "COUNTY";
  JurisdictionType2["CITY"] = "CITY";
  JurisdictionType2["SPECIAL"] = "SPECIAL";
  return JurisdictionType2;
})(JurisdictionType || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/billing-v1-manual-tax-mapping-manual-tax-mappings.meta.ts
function createManualTaxMapping2() {
  const payload = {};
  const getRequestOptions = createManualTaxMapping(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/manual-tax-mappings",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkCreateManualTaxMappings2() {
  const payload = {};
  const getRequestOptions = bulkCreateManualTaxMappings(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/manual-tax-mappings/create",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getManualTaxMapping2() {
  const payload = { manualTaxMappingId: ":manualTaxMappingId" };
  const getRequestOptions = getManualTaxMapping(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/manual-tax-mappings/{manualTaxMappingId}",
    pathParams: { manualTaxMappingId: "manualTaxMappingId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateManualTaxMapping2() {
  const payload = { manualTaxMapping: { id: ":manualTaxMappingId" } };
  const getRequestOptions = updateManualTaxMapping(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/manual-tax-mappings/{manualTaxMapping.id}",
    pathParams: { manualTaxMappingId: "manualTaxMappingId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteManualTaxMapping2() {
  const payload = { manualTaxMappingId: ":manualTaxMappingId" };
  const getRequestOptions = deleteManualTaxMapping(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/manual-tax-mappings/{manualTaxMappingId}",
    pathParams: { manualTaxMappingId: "manualTaxMappingId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryManualTaxMappings2() {
  const payload = {};
  const getRequestOptions = queryManualTaxMappings(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/manual-tax-mappings/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateExtendedFields2() {
  const payload = { id: ":id" };
  const getRequestOptions = updateExtendedFields(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/manual-tax-mappings/{id}/update-extended-fields",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkAssignExemptOverrides2() {
  const payload = {};
  const getRequestOptions = bulkAssignExemptOverrides(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/manual-tax-mappings/bulk-assign-exempt-overrides",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUnassignExemptOverrides2() {
  const payload = {};
  const getRequestOptions = bulkUnassignExemptOverrides(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/manual-tax-mappings/bulk-unassign-exempt-overrides",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listManualTaxMappings2() {
  const payload = {};
  const getRequestOptions = listManualTaxMappings(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/manual-tax-mappings",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  JurisdictionTypeOriginal,
  SortOrderOriginal,
  WebhookIdentityTypeOriginal,
  bulkAssignExemptOverrides,
  bulkCreateManualTaxMappings,
  bulkUnassignExemptOverrides,
  createManualTaxMapping,
  deleteManualTaxMapping,
  getManualTaxMapping,
  listManualTaxMappings,
  queryManualTaxMappings,
  updateExtendedFields,
  updateManualTaxMapping
});
