// @wix/patterns/dist/esm/giza/buildWixPatternsEssentials.js
import { Trans, useExperiments, useTranslation } from '@wix/yoshi-flow-bm';
import { createAccountComponentSettings, getWixPatternsWindowExperiments, getExperimentsCompat, } from '@wix/bex-core';
export function buildWixPatternsEssentials(essentials, { overrides } = {}) {
    const { errorMonitor, httpClient, environment: essentialsEnvironment, errorHandler, } = essentials;
    const environment = {
        ...essentialsEnvironment,
        ...overrides?.environment,
        providerType: 'giza',
    };
    return {
        createEssentials: (options) => essentials.createChildEssentials({
            ...options,
            experiments: {
                bag: getWixPatternsWindowExperiments() ?? {},
            },
        }),
        errorMonitor: errorMonitor ?? { captureException: () => { } },
        errorHandler,
        useTranslation,
        biLogger: process.env.NODE_ENV !== 'test'
            ? essentials.asPartial.biLogger
            : undefined,
        Trans,
        useExperiments: () => useExperiments({ readOnly: true }),
        showToast: () => ({ remove: () => { } }),
        httpClient,
        experiments: getExperimentsCompat(essentials),
        createComponentSettings: createAccountComponentSettings,
        ...overrides,
        user: {
            permissions: [],
            ...overrides?.user,
        },
        environment,
        environmentBI: {
            artifactId: essentials.environment.artifactId,
            componentId: essentials.environment.componentId,
        },
    };
}
