// @wix/wix-ui-test-utils/dist/esm/vanilla/vanilla.js
import * as React from 'react';
import * as ReactDom from 'react-dom';
import { renderIntoDocument } from 'react-dom/test-utils';
import { reactEventTrigger } from '../react-helpers';
import { jsdomReactUniDriver } from '@wix/unidriver-jsdom-react-legacy';
const getElement = ({ wrapper, dataHook }) => {
    const domInstance = ReactDom.findDOMNode(wrapper);
    if (domInstance) {
        const dataHookOnInstance = domInstance.attributes.getNamedItem('data-hook') || { value: '' };
        return dataHook === dataHookOnInstance.value
            ? domInstance
            : domInstance.querySelector(`[data-hook='${dataHook}']`);
    }
    return;
};
export function testkitFactoryCreator(driverFactory) {
    return (testkitArgs) => driverFactory({
        element: getElement(testkitArgs),
        wrapper: testkitArgs.wrapper,
        eventTrigger: reactEventTrigger(),
        dataHook: testkitArgs.dataHook,
    });
}
export function uniTestkitFactoryCreator(driverFactory) {
    return (testkitArgs) => {
        const element = getElement(testkitArgs);
        return driverFactory(jsdomReactUniDriver(element), jsdomReactUniDriver(document.body), { dataHook: testkitArgs.dataHook });
    };
}
export function isTestkitExists(Element, testkitFactory, options) {
    const div = document.createElement('div');
    const dataHook = 'myDataHook';
    const dataHookPropName = options && options.dataHookPropName;
    const extraProps = dataHookPropName
        ? { [dataHookPropName]: dataHook }
        : { 'data-hook': dataHook, dataHook }; // For backward compatibility add dataHook which is used in Wix-Style-React
    const elementToRender = React.cloneElement(Element, extraProps);
    const renderedElement = renderIntoDocument(React.createElement("div", null, elementToRender));
    const wrapper = div.appendChild(renderedElement);
    const testkit = testkitFactory({ wrapper, dataHook });
    return testkit.exists();
}
export async function isUniTestkitExists(Element, testkitFactory, options) {
    const div = document.createElement('div');
    const dataHook = 'myDataHook';
    const dataHookPropName = options && options.dataHookPropName;
    const extraProps = dataHookPropName
        ? { [dataHookPropName]: dataHook }
        : { 'data-hook': dataHook, dataHook };
    const elementToRender = React.cloneElement(Element, extraProps);
    const renderedElement = renderIntoDocument(React.createElement("div", null, elementToRender));
    const wrapper = div.appendChild(renderedElement);
    const testkit = testkitFactory({ wrapper, dataHook });
    return testkit.exists();
}
