// @wix/patterns/dist/esm/components/BulkActionToolbar/BulkActionToolbar.uni.driver.js
import { ButtonUniDriver, TooltipUniDriver, PopoverMenuUniDriver, } from '@wix/design-system/dist/testkit/unidriver';
import { baseUniDriverFactory } from '../../unidriver';
export function BulkActionToolbarUniDriver(base, body) {
    const button = (dataHook = 'bulk-action-button') => ButtonUniDriver(base.$(`[data-hook="${dataHook}"]`), body);
    const tooltip = TooltipUniDriver(base.$('[data-hook="bulk-action-button-tooltip"]'), body);
    const popover = (dataHook) => PopoverMenuUniDriver(base.$(`[data-hook="${dataHook}"]`), body);
    return {
        ...baseUniDriverFactory(base),
        getButton: (dataHook) => button(dataHook),
        clickButton: (dataHook) => button(dataHook).click(),
        isButtonExists: (dataHook) => button(dataHook).exists(),
        isButtonDisabled: (dataHook) => button(dataHook).isButtonDisabled(),
        getButtonTextContent: (dataHook) => button(dataHook).getButtonTextContent(),
        getPopover: (dataHook) => popover(dataHook),
        isPopoverExists: (dataHook) => popover(dataHook).exists(),
        isTooltipExists: tooltip.exists,
        getTooltipText: tooltip.getTooltipText,
        clickOutsideTooltip: tooltip.clickOutside,
        getSelectedCountText: () => base.$('[data-hook="selected-count"]').text(),
    };
}
