// @wix/auto_sdk_benefit-programs_programs/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  endProgram: () => endProgram2,
  getProgram: () => getProgram2,
  getProgramByExternalIdAndNamespace: () => getProgramByExternalIdAndNamespace2,
  pauseProgram: () => pauseProgram2,
  provisionProgram: () => provisionProgram2,
  queryPrograms: () => queryPrograms2,
  renewProgram: () => renewProgram2,
  resumeProgram: () => resumeProgram2,
  updateProgram: () => updateProgram2
});
module.exports = __toCommonJS(meta_exports);

// src/benefit-programs-v1-program-programs.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixBenefitProgramsV1ProgramProgramServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/programs",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/benefit-programs-programs",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-programs",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/programs",
        destPath: "/v1/programs"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-programs",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/programs",
        destPath: "/v1/programs"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/benefit-programs-programs",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/programs",
        destPath: "/v1/programs"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/benefit-programs-programs",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/programs",
        destPath: "/v1/programs"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-programs",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/programs",
        destPath: "/v1/programs"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-programs",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/programs",
        destPath: "/v1/programs"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/benefit-programs/v1/programs",
        destPath: "/v1/programs"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/benefit-programs/v1/programs",
        destPath: "/v1/programs"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_benefit-programs_programs";
function provisionProgram(payload) {
  function __provisionProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.program.ProgramService.ProvisionProgram",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramProgramServiceUrl({
        protoPath: "/v1/programs/provision",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "program.createdDate" },
            { path: "program.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __provisionProgram;
}
function renewProgram(payload) {
  function __renewProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.program.ProgramService.RenewProgram",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramProgramServiceUrl({
        protoPath: "/v1/programs/renew",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __renewProgram;
}
function pauseProgram(payload) {
  function __pauseProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.program.ProgramService.PauseProgram",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramProgramServiceUrl({
        protoPath: "/v1/programs/pause",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __pauseProgram;
}
function resumeProgram(payload) {
  function __resumeProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.program.ProgramService.ResumeProgram",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramProgramServiceUrl({
        protoPath: "/v1/programs/resume",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __resumeProgram;
}
function endProgram(payload) {
  function __endProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.program.ProgramService.EndProgram",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramProgramServiceUrl({
        protoPath: "/v1/programs/end",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __endProgram;
}
function updateProgram(payload) {
  function __updateProgram({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "program.createdDate" },
          { path: "program.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program",
      method: "PATCH",
      methodFqn: "wix.benefit_programs.v1.program.ProgramService.UpdateProgram",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramProgramServiceUrl({
        protoPath: "/v1/programs/{program.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "program.createdDate" },
            { path: "program.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateProgram;
}
function getProgram(payload) {
  function __getProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.program.ProgramService.GetProgram",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramProgramServiceUrl({
        protoPath: "/v1/programs/{programId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "program.createdDate" },
            { path: "program.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getProgram;
}
function queryPrograms(payload) {
  function __queryPrograms({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.program.ProgramService.QueryPrograms",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramProgramServiceUrl({
        protoPath: "/v1/programs/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "programs.createdDate" },
            { path: "programs.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryPrograms;
}
function getProgramByExternalIdAndNamespace(payload) {
  function __getProgramByExternalIdAndNamespace({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.program.ProgramService.GetProgramByExternalIdAndNamespace",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramProgramServiceUrl({
        protoPath: "/v1/programs/by-namespace-and-external-id",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "program.createdDate" },
            { path: "program.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getProgramByExternalIdAndNamespace;
}

// src/benefit-programs-v1-program-programs.meta.ts
function provisionProgram2() {
  const payload = {};
  const getRequestOptions = provisionProgram(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/programs/provision",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function renewProgram2() {
  const payload = {};
  const getRequestOptions = renewProgram(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/programs/renew",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function pauseProgram2() {
  const payload = {};
  const getRequestOptions = pauseProgram(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/programs/pause",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function resumeProgram2() {
  const payload = {};
  const getRequestOptions = resumeProgram(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/programs/resume",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function endProgram2() {
  const payload = {};
  const getRequestOptions = endProgram(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/programs/end",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateProgram2() {
  const payload = { program: { id: ":programId" } };
  const getRequestOptions = updateProgram(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/programs/{program.id}",
    pathParams: { programId: "programId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getProgram2() {
  const payload = { programId: ":programId" };
  const getRequestOptions = getProgram(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/programs/{programId}",
    pathParams: { programId: "programId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryPrograms2() {
  const payload = {};
  const getRequestOptions = queryPrograms(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/programs/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getProgramByExternalIdAndNamespace2() {
  const payload = {};
  const getRequestOptions = getProgramByExternalIdAndNamespace(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/programs/by-namespace-and-external-id",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  endProgram,
  getProgram,
  getProgramByExternalIdAndNamespace,
  pauseProgram,
  provisionProgram,
  queryPrograms,
  renewProgram,
  resumeProgram,
  updateProgram
});
