// @wix/fast-gallery-core/dist/cjs/components/gallery/Gallery.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _clsx = _interopRequireDefault(require("clsx"));
var _navigator = require("../../navigator");
var _Item = _interopRequireDefault(require("../item/Item"));
var _lib = require("../../lib");
var _generalConstants = require("../generalConstants");
var _GalleryModule = _interopRequireDefault(require("./Gallery.module.scss"));
var _useArrowNavigation = require("../hooks/useArrowNavigation");
var _jsxFileName = "/home/builduser/work/d3ecddb38d0a3df1/packages/fast-gallery-packages/fast-gallery-core/dist/cjs/components/gallery/Gallery.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
const Gallery = ({
  dataList,
  selectors: gallerySelectors,
  customRenderers,
  eventsListener = () => {},
  createResizedMediaUrl,
  options = {
    disableContextMenu: true
  },
  isAccessible = false,
  wrapperRef,
  getItemCustomAttributes
}) => {
  const [navigator, setNavigator] = (0, _react.useState)(null);
  const scrollerRef = (0, _react.useRef)(null);
  const defaultScrollerWrapperRef = (0, _react.useRef)(null);
  const scrollerWrapperRef = wrapperRef ? wrapperRef : defaultScrollerWrapperRef;
  const containerRef = (0, _react.useRef)(null);
  const [currentIndex, setCurrentIndex] = (0, _react.useState)(0);
  const {
    handleKeyDown,
    registerItemRef
  } = (0, _useArrowNavigation.useArrowsNavigation)(navigator, isAccessible);
  const [isInHover, setIsInHover] = (0, _react.useState)(false);
  const [currentHoveredIndex, setCurrentHoveredIndex] = (0, _react.useState)(-1);
  const selectors = {
    ..._generalConstants.defaultGallerySelectors,
    ...gallerySelectors
  };
  const galleryEventsListener = (0, _lib.makeEventsListener)().CURRENT_ITEM_CHANGED(eventData => {
    setCurrentIndex(eventData.index);
  }).HOVER_SET(eventData => {
    setCurrentHoveredIndex(eventData.index ?? -1);
  }).done(eventsListener);
  (0, _react.useEffect)(() => {
    const loadNavigator = async () => {
      const {
        Navigator
      } = await Promise.resolve().then(() => _interopRequireWildcard(require(/* webpackChunkName: "Navigator" */
      '../../navigator')));
      setNavigator(new Navigator({
        initSettings: {},
        eventsListener: galleryEventsListener
      }));
    };
    loadNavigator();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  (0, _react.useEffect)(() => {
    if (navigator === null) {
      return;
    }
    navigator.setScrollerRef(scrollerRef);
    navigator.setWrapperRef(wrapperRef);
    navigator.setContainerRef(containerRef);
    navigator.updateTotalItemsCount(5);
    const scrollerElement = scrollerRef.current;
    if (scrollerElement) {
      scrollerElement.addEventListener('keydown', handleKeyDown);
    }
    return () => {
      if (navigator === null) {
        return;
      }
      navigator.removeScrollerRef(scrollerRef);
      navigator.removeWrapperRef(wrapperRef);
      navigator.removeContainerRef(containerRef);
      if (scrollerElement) {
        scrollerElement.removeEventListener('keydown', handleKeyDown);
      }
    };
  }, [navigator, scrollerRef, wrapperRef, handleKeyDown]);
  const handleScrollerMeasure = newMeasurements => {
    navigator == null || navigator.updateScrollerMeasurements(newMeasurements);
  };
  const handleWrapperMeasure = newMeasurements => {
    navigator == null || navigator.updateWrapperMeasurements(newMeasurements);
  };
  const handleContainerMeasure = newMeasurements => {
    navigator == null || navigator.updateContainerMeasurements(newMeasurements);
  };
  const handleItemMeasure = (index, newMeasurements, itemRef) => {
    navigator == null || navigator.updateItemMeasurements(newMeasurements, index, itemRef);
  };
  (0, _navigator.useMeasure)({
    callback: measurements => handleScrollerMeasure(measurements),
    ref: scrollerRef
  });
  (0, _navigator.useMeasure)({
    callback: measurements => handleWrapperMeasure(measurements),
    ref: scrollerWrapperRef
  });
  (0, _navigator.useMeasure)({
    callback: measurements => handleContainerMeasure(measurements),
    ref: containerRef
  });
  const onMouseEnter = () => {
    var _containerRef$current;
    // the commented code is copied from pro-gallery and is not used in this component currently
    // if (
    //   this.currentScrollData?.isContinuousScrolling &&
    //   this.state.options[optionsMap.behaviourParams.gallery.horizontal.autoSlide.pauseOnHover]
    // ) {
    //   haltScroll(this.currentScrollData);
    // }
    setIsInHover(true);
    (_containerRef$current = containerRef.current) == null || _containerRef$current.classList.add(_generalConstants.hoverClasses.containerHover);
  };
  const onMouseLeave = () => {
    var _containerRef$current2;
    setIsInHover(false);
    (_containerRef$current2 = containerRef.current) == null || _containerRef$current2.classList.remove(_generalConstants.hoverClasses.containerHover);
  };
  const onContainerFocus = () => {
    var _containerRef$current3;
    setIsInHover(true);
    (_containerRef$current3 = containerRef.current) == null || _containerRef$current3.classList.add(_generalConstants.hoverClasses.containerHover);
  };
  const onContainerBlur = () => {
    var _containerRef$current4;
    setIsInHover(false);
    (_containerRef$current4 = containerRef.current) == null || _containerRef$current4.classList.remove(_generalConstants.hoverClasses.containerHover);
  };
  return /*#__PURE__*/_react.default.createElement("div", {
    ref: scrollerRef,
    id: "scroller",
    className: (0, _clsx.default)(_GalleryModule.default.scroller, 'scroller', selectors == null ? void 0 : selectors.scroller, isInHover && _GalleryModule.default.hover),
    onMouseEnter: onMouseEnter,
    onMouseLeave: onMouseLeave,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 177,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement("div", {
    ref: containerRef,
    className: (0, _clsx.default)(_GalleryModule.default.container, selectors == null ? void 0 : selectors.container, isInHover ? 'container-hover' : ''),
    role: "list",
    onFocus: onContainerFocus,
    onBlur: onContainerBlur,
    tabIndex: isAccessible ? 0 : -1,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 189,
      columnNumber: 7
    }
  }, dataList.map((data, index) => /*#__PURE__*/_react.default.createElement(_Item.default, {
    getItemCustomAttributes: getItemCustomAttributes,
    key: data.id,
    index: index,
    data: data,
    selectors: selectors,
    customRenderers: customRenderers,
    eventsListener: galleryEventsListener,
    createResizedMediaUrl: createResizedMediaUrl,
    onMeasure: handleItemMeasure,
    activeIndex: currentIndex,
    isCurrentlyHovered: index === currentHoveredIndex,
    options: options,
    isAccessible: isAccessible,
    registerItemRef: registerItemRef,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 202,
      columnNumber: 11
    }
  }))));
};
var _default = exports.default = Gallery;
