// @wix/auto_sdk_events_ticket-reservations/build/internal/es/meta.mjs
// src/events-v1-ticket-reservation-ticket-reservations.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixEventsTicketreservationsV1TicketReservationManagementUrl(opts) {
  const domainToMappings = {
    _: [
      {
        srcPath: "/_api/events-ticket-reservations",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/events-ticket-reservations",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/events-ticket-reservations",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/events/v1/ticket-reservations",
        destPath: "/v1/ticket-reservations"
      },
      {
        srcPath: "/events/v1/bulk/ticket-reservations",
        destPath: "/v1/bulk/ticket-reservations"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_events_ticket-reservations";
function createTicketReservation(payload) {
  function __createTicketReservation({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "ticketReservation.createdDate" },
          { path: "ticketReservation.updatedDate" },
          { path: "ticketReservation.expirationDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v1.ticket_reservation",
      method: "POST",
      methodFqn: "wix.events.ticketreservations.v1.TicketReservationManagement.CreateTicketReservation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketreservationsV1TicketReservationManagementUrl({
        protoPath: "/v1/ticket-reservations",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "ticketReservation.createdDate" },
            { path: "ticketReservation.updatedDate" },
            { path: "ticketReservation.expirationDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createTicketReservation;
}
function getTicketReservation(payload) {
  function __getTicketReservation({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.ticket_reservation",
      method: "GET",
      methodFqn: "wix.events.ticketreservations.v1.TicketReservationManagement.GetTicketReservation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketreservationsV1TicketReservationManagementUrl({
        protoPath: "/v1/ticket-reservations/{ticketReservationId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "ticketReservation.createdDate" },
            { path: "ticketReservation.updatedDate" },
            { path: "ticketReservation.expirationDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getTicketReservation;
}
function deleteTicketReservation(payload) {
  function __deleteTicketReservation({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.ticket_reservation",
      method: "DELETE",
      methodFqn: "wix.events.ticketreservations.v1.TicketReservationManagement.DeleteTicketReservation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketreservationsV1TicketReservationManagementUrl({
        protoPath: "/v1/ticket-reservations/{ticketReservationId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "ticketReservation.createdDate" },
            { path: "ticketReservation.updatedDate" },
            { path: "ticketReservation.expirationDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __deleteTicketReservation;
}
function bulkUpdateTicketReservationTags(payload) {
  function __bulkUpdateTicketReservationTags({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.ticket_reservation",
      method: "POST",
      methodFqn: "wix.events.ticketreservations.v1.TicketReservationManagement.BulkUpdateTicketReservationTags",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketreservationsV1TicketReservationManagementUrl({
        protoPath: "/v1/bulk/ticket-reservations/update-tags",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateTicketReservationTags;
}
function bulkUpdateTicketReservationTagsByFilter(payload) {
  function __bulkUpdateTicketReservationTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.ticket_reservation",
      method: "POST",
      methodFqn: "wix.events.ticketreservations.v1.TicketReservationManagement.BulkUpdateTicketReservationTagsByFilter",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketreservationsV1TicketReservationManagementUrl({
        protoPath: "/v1/bulk/ticket-reservations/update-tags-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateTicketReservationTagsByFilter;
}
function cancelTicketReservation(payload) {
  function __cancelTicketReservation({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.ticket_reservation",
      method: "POST",
      methodFqn: "wix.events.ticketreservations.v1.TicketReservationManagement.CancelTicketReservation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketreservationsV1TicketReservationManagementUrl({
        protoPath: "/v1/ticket-reservations/{ticketReservationId}/cancel",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "ticketReservation.createdDate" },
            { path: "ticketReservation.updatedDate" },
            { path: "ticketReservation.expirationDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __cancelTicketReservation;
}

// src/events-v1-ticket-reservation-ticket-reservations.types.ts
var ReservationStatus = /* @__PURE__ */ ((ReservationStatus2) => {
  ReservationStatus2["PENDING"] = "PENDING";
  ReservationStatus2["CONFIRMED"] = "CONFIRMED";
  ReservationStatus2["CANCELED"] = "CANCELED";
  ReservationStatus2["CANCELED_MANUALLY"] = "CANCELED_MANUALLY";
  ReservationStatus2["EXPIRED"] = "EXPIRED";
  return ReservationStatus2;
})(ReservationStatus || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["STANDARD"] = "STANDARD";
  Type2["DONATION"] = "DONATION";
  return Type2;
})(Type || {});
var CheckoutType = /* @__PURE__ */ ((CheckoutType2) => {
  CheckoutType2["UNKNOWN_CHECKOUT_TYPE"] = "UNKNOWN_CHECKOUT_TYPE";
  CheckoutType2["EVENTS_APP"] = "EVENTS_APP";
  CheckoutType2["ECOMM_PLATFORM"] = "ECOMM_PLATFORM";
  return CheckoutType2;
})(CheckoutType || {});
var FeeType = /* @__PURE__ */ ((FeeType2) => {
  FeeType2["FEE_INCLUDED"] = "FEE_INCLUDED";
  FeeType2["FEE_ADDED_AT_CHECKOUT"] = "FEE_ADDED_AT_CHECKOUT";
  return FeeType2;
})(FeeType || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
var PromotionType = /* @__PURE__ */ ((PromotionType2) => {
  PromotionType2["UNKNOWN_PROMOTION_TYPE"] = "UNKNOWN_PROMOTION_TYPE";
  PromotionType2["OTHER"] = "OTHER";
  PromotionType2["COUPON_CODE"] = "COUPON_CODE";
  PromotionType2["PAID_PLAN_BENEFIT"] = "PAID_PLAN_BENEFIT";
  PromotionType2["GIFT_CARD"] = "GIFT_CARD";
  return PromotionType2;
})(PromotionType || {});
var OrderStatus = /* @__PURE__ */ ((OrderStatus2) => {
  OrderStatus2["NA_ORDER_STATUS"] = "NA_ORDER_STATUS";
  OrderStatus2["FREE"] = "FREE";
  OrderStatus2["PENDING"] = "PENDING";
  OrderStatus2["PAID"] = "PAID";
  OrderStatus2["OFFLINE_PENDING"] = "OFFLINE_PENDING";
  OrderStatus2["INITIATED"] = "INITIATED";
  OrderStatus2["CANCELED"] = "CANCELED";
  OrderStatus2["DECLINED"] = "DECLINED";
  OrderStatus2["AUTHORIZED"] = "AUTHORIZED";
  OrderStatus2["VOIDED"] = "VOIDED";
  OrderStatus2["PARTIALLY_PAID"] = "PARTIALLY_PAID";
  return OrderStatus2;
})(OrderStatus || {});
var SubdivisionType = /* @__PURE__ */ ((SubdivisionType2) => {
  SubdivisionType2["UNKNOWN_SUBDIVISION_TYPE"] = "UNKNOWN_SUBDIVISION_TYPE";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_1"] = "ADMINISTRATIVE_AREA_LEVEL_1";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_2"] = "ADMINISTRATIVE_AREA_LEVEL_2";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_3"] = "ADMINISTRATIVE_AREA_LEVEL_3";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_4"] = "ADMINISTRATIVE_AREA_LEVEL_4";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_5"] = "ADMINISTRATIVE_AREA_LEVEL_5";
  SubdivisionType2["COUNTRY"] = "COUNTRY";
  return SubdivisionType2;
})(SubdivisionType || {});

// src/events-v1-ticket-reservation-ticket-reservations.meta.ts
function createTicketReservation2() {
  const payload = {};
  const getRequestOptions = createTicketReservation(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/ticket-reservations",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getTicketReservation2() {
  const payload = { ticketReservationId: ":ticketReservationId" };
  const getRequestOptions = getTicketReservation(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/ticket-reservations/{ticketReservationId}",
    pathParams: { ticketReservationId: "ticketReservationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteTicketReservation2() {
  const payload = { ticketReservationId: ":ticketReservationId" };
  const getRequestOptions = deleteTicketReservation(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/ticket-reservations/{ticketReservationId}",
    pathParams: { ticketReservationId: "ticketReservationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateTicketReservationTags2() {
  const payload = {};
  const getRequestOptions = bulkUpdateTicketReservationTags(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/ticket-reservations/update-tags",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateTicketReservationTagsByFilter2() {
  const payload = {};
  const getRequestOptions = bulkUpdateTicketReservationTagsByFilter(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/ticket-reservations/update-tags-by-filter",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function cancelTicketReservation2() {
  const payload = { ticketReservationId: ":ticketReservationId" };
  const getRequestOptions = cancelTicketReservation(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/ticket-reservations/{ticketReservationId}/cancel",
    pathParams: { ticketReservationId: "ticketReservationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  CheckoutType as CheckoutTypeOriginal,
  FeeType as FeeTypeOriginal,
  OrderStatus as OrderStatusOriginal,
  PromotionType as PromotionTypeOriginal,
  ReservationStatus as ReservationStatusOriginal,
  SubdivisionType as SubdivisionTypeOriginal,
  Type as TypeOriginal,
  WebhookIdentityType as WebhookIdentityTypeOriginal,
  bulkUpdateTicketReservationTags2 as bulkUpdateTicketReservationTags,
  bulkUpdateTicketReservationTagsByFilter2 as bulkUpdateTicketReservationTagsByFilter,
  cancelTicketReservation2 as cancelTicketReservation,
  createTicketReservation2 as createTicketReservation,
  deleteTicketReservation2 as deleteTicketReservation,
  getTicketReservation2 as getTicketReservation
};
