// @wix/patterns/dist/cjs/components/CollectionEmptyStateBase/CollectionEmptyStateBase.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.CollectionEmptyStateBase = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireDefault(require("react"));
var _designSystem = require("@wix/design-system");
var _mobxReactLite = require("mobx-react-lite");
var _hooks = require("../../hooks");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CollectionEmptyStateBase/CollectionEmptyStateBase.tsx";
function _CollectionEmptyStateBase(props) {
  const {
    component: Component = _designSystem.EmptyState
  } = props;
  const {
    title,
    subtitle
  } = (0, _hooks.useEmptyStateTexts)(props);
  return /*#__PURE__*/_react.default.createElement(Component, (0, _extends2.default)({
    title: title,
    subtitle: subtitle
  }, props, {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 19,
      columnNumber: 10
    }
  }));
}
const CollectionEmptyStateBase = exports.CollectionEmptyStateBase = (0, _mobxReactLite.observer)(_CollectionEmptyStateBase);
