// @wix/design-system/dist/esm/Accordion/Accordion.private.uni.driver.js
import publicDriverFactory from './Accordion.uni.driver';
import { dataHooks } from './constants';
export const accordionPrivateDriverFactory = (base) => {
    const getItemAt = (idx) => base.$$(`[data-hook="${dataHooks.item}"]`).at(idx);
    return {
        ...publicDriverFactory(base),
        getAmountOfItems: async () => base.$$(`[data-hook="${dataHooks.item}"]`).count(),
        hoverOnItem: async (idx) => await getItemAt(idx).$('[data-hook="header"]').hover(),
        getAccordionItemAt: async (idx) => {
            const accordionItem = await getItemAt(idx);
            return accordionItemPrivateDriverFactory(accordionItem);
        },
    };
};
export const accordionItemPrivateDriverFactory = (base) => ({
    getToggleIconPlacement: async () => {
        const header = await base.get(`[data-hook="${dataHooks.header}"]`);
        const headerElement = (await header.getNative());
        const isAtStart = headerElement.children[0].getAttribute('data-hook') ||
            headerElement.children[0].querySelector(`[data-hook="${dataHooks.toggleButton}"]`);
        const isAtEnd = headerElement.lastElementChild?.getAttribute('data-hook') ||
            headerElement.lastElementChild?.querySelector(`[data-hook="${dataHooks.toggleButton}"]`);
        return isAtStart ? 'start' : isAtEnd ? 'end' : 'none';
    },
});
