// @wix/patterns/dist/cjs/components/CollectionPagination.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.CollectionPagination = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireWildcard(require("react"));
var _designSystem = require("@wix/design-system");
var _mobxReactLite = require("mobx-react-lite");
var _providers = require("../providers");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CollectionPagination.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
const CollectionPagination = exports.CollectionPagination = (0, _mobxReactLite.observer)(function PaginationTable({
  state,
  ...rest
}) {
  const {
    result: {
      status
    },
    totalPages,
    query: {
      pagination
    }
  } = state;
  const {
    scrollableContentRef
  } = (0, _providers.useScrollableContentContext)();
  const showPagination = totalPages > 0 && !status.isLoading;
  const onPageChange = (0, _react.useCallback)(async ({
    page
  }) => {
    var _scrollableContentRef;
    scrollableContentRef == null || (_scrollableContentRef = scrollableContentRef.current) == null || _scrollableContentRef.scrollTo({
      top: 0
    });
    await state.clearAndMoveToPage(page);
  }, []);
  return /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
    align: "center",
    padding: "SP8",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 31,
      columnNumber: 5
    }
  }, showPagination && /*#__PURE__*/_react.default.createElement(_designSystem.Pagination, (0, _extends2.default)({
    totalPages: totalPages,
    currentPage: pagination.page,
    onChange: onPageChange
  }, rest, {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 33,
      columnNumber: 9
    }
  })));
});
