// @wix/patterns/dist/cjs/components/CollectionModal.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.ModalWrapper = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _designSystem = require("@wix/design-system");
var _react = _interopRequireDefault(require("react"));
var _mobxReactLite = require("mobx-react-lite");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CollectionModal.tsx";
function _ModalWrapper({
  state,
  ...rest
}) {
  return /*#__PURE__*/_react.default.createElement("div", {
    onClick: e => e.stopPropagation(),
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 12,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.Modal, (0, _extends2.default)({
    isOpen: state.isOpen,
    onRequestClose: state.close
  }, rest, {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 13,
      columnNumber: 7
    }
  })));
}
const ModalWrapper = exports.ModalWrapper = (0, _mobxReactLite.observer)(_ModalWrapper);
