// @wix/patterns/dist/cjs/components/DiscardChangesModal.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.DiscardChangesModal = DiscardChangesModal;
var _ToolbarCollectionContext = require("./ToolbarCollectionContext");
var _react = _interopRequireDefault(require("react"));
var _ActionModal = require("./ActionModal");
var _react2 = require("@wix/bex-core/react");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/DiscardChangesModal.tsx";
function DiscardChangesModal() {
  const {
    translate: t
  } = (0, _react2.useWixPatternsContainer)();
  const {
    pendingQuery
  } = (0, _ToolbarCollectionContext.useToolbarCollectionContext)();
  if (pendingQuery == null) {
    return null;
  }
  return /*#__PURE__*/_react.default.createElement(_ActionModal.MessageActionModal, {
    dataHook: "discard-changes-modal",
    state: pendingQuery.discardChangesModal,
    title: t('cairo.appliedFilters.discardChanges.title'),
    content: t('cairo.appliedFilters.discardChanges.description'),
    primaryButtonText: t('cairo.appliedFilters.discardChanges.CTA'),
    secondaryButtonText: t('cairo.appliedFilters.applyAndClose.CTA'),
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 16,
      columnNumber: 5
    }
  });
}
