// @wix/patterns/dist/cjs/components/ExpandableSearch.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.ExpandableSearch = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireDefault(require("react"));
var _CollectionSearch = require("./CollectionSearch");
var _mobxReactLite = require("mobx-react-lite");
var _usehooksTs = require("usehooks-ts");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/ExpandableSearch.tsx";
function _ExpandableSearch({
  visibility,
  ...rest
}) {
  const [containerRef, {
    width
  }] = (0, _usehooksTs.useElementSize)();
  return /*#__PURE__*/_react.default.createElement("div", {
    style: {
      width: '100%',
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'flex-end'
    },
    ref: containerRef,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 14,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      marginLeft: 'auto'
    },
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 23,
      columnNumber: 7
    }
  }, /*#__PURE__*/_react.default.createElement(_CollectionSearch.CollectionSearch, (0, _extends2.default)({}, rest, {
    expandWidth: width,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 24,
      columnNumber: 9
    }
  }))));
}
const ExpandableSearch = exports.ExpandableSearch = (0, _mobxReactLite.observer)(_ExpandableSearch);
