// @wix/auto_sdk_data_sharing/build/es/index.typings.mjs
// src/data-v1-data-sharing-policy-sharing.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/data-v1-data-sharing-policy-sharing.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressCloudDataApiShareDataCollectionSharingServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/cloud-data",
        destPath: ""
      }
    ],
    "code._base_domain_": [
      {
        srcPath: "/_api/cloud-data/v1/data-settings",
        destPath: "/v1/data-settings"
      },
      {
        srcPath: "/_api/cloud-data/v1/schemas",
        destPath: "/v1/schemas"
      },
      {
        srcPath: "/_api/cloud-data/v1/connector",
        destPath: "/v1/connector"
      },
      {
        srcPath: "/_api/data/v1/data-settings",
        destPath: "/v1/data-settings"
      },
      {
        srcPath: "/_api/data/v1/schemas",
        destPath: "/v1/schemas"
      },
      {
        srcPath: "/_api/data/v1/connector",
        destPath: "/v1/connector"
      },
      {
        srcPath: "/_api/cloud-data/v1/wix-data",
        destPath: "/v1/wix-data"
      }
    ],
    "cloud-data.wix-code.com": [
      {
        srcPath: "",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/cloud-data",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/cloud-data",
        destPath: ""
      },
      {
        srcPath: "/_api/data",
        destPath: ""
      },
      {
        srcPath: "/_api/cloud-data",
        destPath: ""
      }
    ],
    "dev._base_domain_": [
      {
        srcPath: "/_api/cloud-data",
        destPath: ""
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/_api/cloud-data/v1",
        destPath: "/v1"
      },
      {
        srcPath: "/_api/cloud-data/v2",
        destPath: "/v2"
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/cloud-data/v1",
        destPath: "/v1"
      },
      {
        srcPath: "/_api/cloud-data/v2",
        destPath: "/v2"
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/cloud-data/v1",
        destPath: "/v1"
      },
      {
        srcPath: "/_api/cloud-data/v2",
        destPath: "/v2"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/cloud-data",
        destPath: ""
      },
      {
        srcPath: "/_api/data",
        destPath: ""
      },
      {
        srcPath: "/_api/cloud-data",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/cloud-data/v1/schemas",
        destPath: "/v1/schemas"
      },
      {
        srcPath: "/_api/cloud-data/dbs/tasks",
        destPath: "/dbs/tasks"
      },
      {
        srcPath: "/_api/data/v1/schemas",
        destPath: "/v1/schemas"
      },
      {
        srcPath: "/_api/data/dbs/tasks",
        destPath: "/dbs/tasks"
      },
      {
        srcPath: "/_api/cloud-data/v1/wix-data",
        destPath: "/v1/wix-data"
      },
      {
        srcPath: "/_api/cloud-data/v1/data-settings",
        destPath: "/v1/data-settings"
      },
      {
        srcPath: "/_api/data/v2/indexes",
        destPath: "/v2/indexes"
      },
      {
        srcPath: "/_api/cloud-data/v1/external-database-connections",
        destPath: "/v1/external-database-connections"
      },
      {
        srcPath: "/_api/data/v1/data-collection-sharing",
        destPath: "/v1/data-collection-sharing"
      },
      {
        srcPath: "/_api/cloud-data/v2/collections",
        destPath: "/v2/collections"
      },
      {
        srcPath: "/_api/cloud-data/v2/bulk",
        destPath: "/v2/bulk"
      },
      {
        srcPath: "/_api/cloud-data/v2/items",
        destPath: "/v2/items"
      },
      {
        srcPath: "/_api/cloud-data/v2/dbs/tasks",
        destPath: "/v2/dbs/tasks"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/cloud-data/v1/schemas",
        destPath: "/v1/schemas"
      },
      {
        srcPath: "/_api/cloud-data/dbs/tasks",
        destPath: "/dbs/tasks"
      },
      {
        srcPath: "/_api/data/v1/schemas",
        destPath: "/v1/schemas"
      },
      {
        srcPath: "/_api/data/dbs/tasks",
        destPath: "/dbs/tasks"
      },
      {
        srcPath: "/_api/cloud-data/v1/wix-data",
        destPath: "/v1/wix-data"
      },
      {
        srcPath: "/_api/cloud-data/v1/data-settings",
        destPath: "/v1/data-settings"
      },
      {
        srcPath: "/_api/data/v2/indexes",
        destPath: "/v2/indexes"
      },
      {
        srcPath: "/_api/cloud-data/v1/external-database-connections",
        destPath: "/v1/external-database-connections"
      },
      {
        srcPath: "/_api/data/v1/data-collection-sharing",
        destPath: "/v1/data-collection-sharing"
      },
      {
        srcPath: "/_api/cloud-data/v2/collections",
        destPath: "/v2/collections"
      },
      {
        srcPath: "/_api/cloud-data/v2/bulk",
        destPath: "/v2/bulk"
      },
      {
        srcPath: "/_api/cloud-data/v2/items",
        destPath: "/v2/items"
      },
      {
        srcPath: "/_api/cloud-data/v2/dbs/tasks",
        destPath: "/v2/dbs/tasks"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/cloud-data/v1/schemas",
        destPath: "/v1/schemas"
      },
      {
        srcPath: "/_api/cloud-data/dbs/tasks",
        destPath: "/dbs/tasks"
      },
      {
        srcPath: "/_api/data/v1/schemas",
        destPath: "/v1/schemas"
      },
      {
        srcPath: "/_api/data/dbs/tasks",
        destPath: "/dbs/tasks"
      },
      {
        srcPath: "/_api/cloud-data/v1/wix-data",
        destPath: "/v1/wix-data"
      },
      {
        srcPath: "/_api/cloud-data/v1/data-settings",
        destPath: "/v1/data-settings"
      },
      {
        srcPath: "/_api/data/v2/indexes",
        destPath: "/v2/indexes"
      },
      {
        srcPath: "/_api/cloud-data/v1/external-database-connections",
        destPath: "/v1/external-database-connections"
      },
      {
        srcPath: "/_api/data/v1/data-collection-sharing",
        destPath: "/v1/data-collection-sharing"
      },
      {
        srcPath: "/_api/cloud-data/v2/collections",
        destPath: "/v2/collections"
      },
      {
        srcPath: "/_api/cloud-data/v2/bulk",
        destPath: "/v2/bulk"
      },
      {
        srcPath: "/_api/cloud-data/v2/items",
        destPath: "/v2/items"
      },
      {
        srcPath: "/_api/cloud-data/v2/dbs/tasks",
        destPath: "/v2/dbs/tasks"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/wix-data/v1/collections",
        destPath: "/v1/wix-data/collections"
      },
      {
        srcPath: "/wix-data/v1/external-database-connections",
        destPath: "/v1/external-database-connections"
      },
      {
        srcPath: "/wix-data/v2/indexes",
        destPath: "/v2/indexes"
      },
      {
        srcPath: "/wix-data/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/wix-data/v1/bulk",
        destPath: "/v1/bulk"
      },
      {
        srcPath: "/wix-data/v1/external-databases",
        destPath: "/v1/external-databases"
      },
      {
        srcPath: "/wix-data/v2",
        destPath: "/v2"
      },
      {
        srcPath: "/data/v2/collections",
        destPath: "/v2/collections"
      },
      {
        srcPath: "/data/v2/items",
        destPath: "/v2/items"
      },
      {
        srcPath: "/data/v2/bulk/items",
        destPath: "/v2/bulk/items"
      },
      {
        srcPath: "/data/v2/indexes",
        destPath: "/v2/indexes"
      },
      {
        srcPath: "/data/v1/external-database-connections",
        destPath: "/v1/external-database-connections"
      },
      {
        srcPath: "/data/v1/data-collection-sharing",
        destPath: "/v1/data-collection-sharing"
      }
    ],
    "www.wixgateway.com": [
      {
        srcPath: "/wix-data/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/wix-data/v1/bulk",
        destPath: "/v1/bulk"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/cloud-data/v2",
        destPath: "/v2"
      },
      {
        srcPath: "/_api/cloud-data/v1/external-database-connections",
        destPath: "/v1/external-database-connections"
      },
      {
        srcPath: "/data/v1/data-collection-sharing",
        destPath: "/v1/data-collection-sharing"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_data_sharing";
function createDataSharingPolicy(payload) {
  function __createDataSharingPolicy({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "dataSharingPolicy.createdDate" },
          { path: "dataSharingPolicy.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.data.v1.data_sharing_policy",
      method: "POST",
      methodFqn: "com.wixpress.cloud.data.api.share.DataCollectionSharingService.CreateDataSharingPolicy",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressCloudDataApiShareDataCollectionSharingServiceUrl({
        protoPath: "/v1/data-collection-sharing/policies",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "dataSharingPolicy.createdDate" },
            { path: "dataSharingPolicy.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createDataSharingPolicy;
}
function updateDataSharingPolicy(payload) {
  function __updateDataSharingPolicy({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "dataSharingPolicy.createdDate" },
          { path: "dataSharingPolicy.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.data.v1.data_sharing_policy",
      method: "POST",
      methodFqn: "com.wixpress.cloud.data.api.share.DataCollectionSharingService.UpdateDataSharingPolicy",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressCloudDataApiShareDataCollectionSharingServiceUrl({
        protoPath: "/v1/data-collection-sharing/policies/{dataSharingPolicy.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "dataSharingPolicy.createdDate" },
            { path: "dataSharingPolicy.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateDataSharingPolicy;
}
function deleteDataSharingPolicy(payload) {
  function __deleteDataSharingPolicy({ host }) {
    const metadata = {
      entityFqdn: "wix.data.v1.data_sharing_policy",
      method: "DELETE",
      methodFqn: "com.wixpress.cloud.data.api.share.DataCollectionSharingService.DeleteDataSharingPolicy",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressCloudDataApiShareDataCollectionSharingServiceUrl({
        protoPath: "/v1/data-collection-sharing/policies/{dataSharingPolicyId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteDataSharingPolicy;
}
function getDataSharingPolicy(payload) {
  function __getDataSharingPolicy({ host }) {
    const metadata = {
      entityFqdn: "wix.data.v1.data_sharing_policy",
      method: "GET",
      methodFqn: "com.wixpress.cloud.data.api.share.DataCollectionSharingService.GetDataSharingPolicy",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressCloudDataApiShareDataCollectionSharingServiceUrl({
        protoPath: "/v1/data-collection-sharing/policies/{dataSharingPolicyId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "dataSharingPolicy.createdDate" },
            { path: "dataSharingPolicy.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getDataSharingPolicy;
}
function listDataSharingPolicies(payload) {
  function __listDataSharingPolicies({ host }) {
    const metadata = {
      entityFqdn: "wix.data.v1.data_sharing_policy",
      method: "GET",
      methodFqn: "com.wixpress.cloud.data.api.share.DataCollectionSharingService.ListDataSharingPolicies",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressCloudDataApiShareDataCollectionSharingServiceUrl({
        protoPath: "/v1/data-collection-sharing/policies",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "dataSharingPolicies.createdDate" },
            { path: "dataSharingPolicies.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listDataSharingPolicies;
}
function listSharedDataCollections(payload) {
  function __listSharedDataCollections({ host }) {
    const metadata = {
      entityFqdn: "wix.data.v1.data_sharing_policy",
      method: "GET",
      methodFqn: "com.wixpress.cloud.data.api.share.DataCollectionSharingService.ListSharedDataCollections",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressCloudDataApiShareDataCollectionSharingServiceUrl({
        protoPath: "/v1/data-collection-sharing/shared",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "connections.createdDate" },
            { path: "connections.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listSharedDataCollections;
}
function connectToSharedCollection(payload) {
  function __connectToSharedCollection({ host }) {
    const metadata = {
      entityFqdn: "wix.data.v1.data_sharing_policy",
      method: "POST",
      methodFqn: "com.wixpress.cloud.data.api.share.DataCollectionSharingService.ConnectToSharedCollection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressCloudDataApiShareDataCollectionSharingServiceUrl({
        protoPath: "/v1/data-collection-sharing/connect-to-shared-collection",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "connection.createdDate" },
            { path: "connection.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __connectToSharedCollection;
}
function disconnectFromSharedCollection(payload) {
  function __disconnectFromSharedCollection({ host }) {
    const metadata = {
      entityFqdn: "wix.data.v1.data_sharing_policy",
      method: "POST",
      methodFqn: "com.wixpress.cloud.data.api.share.DataCollectionSharingService.DisconnectFromSharedCollection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressCloudDataApiShareDataCollectionSharingServiceUrl({
        protoPath: "/v1/data-collection-sharing/disconnect-from-shared-collection",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __disconnectFromSharedCollection;
}

// src/data-v1-data-sharing-policy-sharing.universal.ts
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createDataSharingPolicy2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    dataSharingPolicy: options?.dataSharingPolicy
  });
  const reqOpts = createDataSharingPolicy(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.dataSharingPolicy;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          dataSharingPolicy: "$[0].dataSharingPolicy"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateDataSharingPolicy2(_id, dataSharingPolicy) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    dataSharingPolicy: { ...dataSharingPolicy, id: _id }
  });
  const reqOpts = updateDataSharingPolicy(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.dataSharingPolicy;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { dataSharingPolicy: "$[1]" },
        explicitPathsToArguments: { "dataSharingPolicy.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "dataSharingPolicy"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteDataSharingPolicy2(dataSharingPolicyId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    dataSharingPolicyId
  });
  const reqOpts = deleteDataSharingPolicy(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { dataSharingPolicyId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["dataSharingPolicyId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getDataSharingPolicy2(dataSharingPolicyId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    dataSharingPolicyId
  });
  const reqOpts = getDataSharingPolicy(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.dataSharingPolicy;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { dataSharingPolicyId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["dataSharingPolicyId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listDataSharingPolicies2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    dataCollectionIds: options?.dataCollectionIds,
    paging: options?.paging
  });
  const reqOpts = listDataSharingPolicies(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          dataCollectionIds: "$[0].dataCollectionIds",
          paging: "$[0].paging"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listSharedDataCollections2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    dataCollectionIds: options?.dataCollectionIds,
    sharedWithCurrent: options?.sharedWithCurrent,
    paging: options?.paging,
    sharedWithCurrentSite: options?.sharedWithCurrentSite
  });
  const reqOpts = listSharedDataCollections(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          dataCollectionIds: "$[0].dataCollectionIds",
          sharedWithCurrent: "$[0].sharedWithCurrent",
          paging: "$[0].paging",
          sharedWithCurrentSite: "$[0].sharedWithCurrentSite"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function connectToSharedCollection2(namespace, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    namespace,
    displayNamespace: options?.displayNamespace,
    dataSharingPolicyId: options?.dataSharingPolicyId
  });
  const reqOpts = connectToSharedCollection(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          namespace: "$[0]",
          displayNamespace: "$[1].displayNamespace",
          dataSharingPolicyId: "$[1].dataSharingPolicyId"
        },
        singleArgumentUnchanged: false
      },
      ["namespace", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function disconnectFromSharedCollection2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    dataCollectionId: options?.dataCollectionId,
    targetDataCollectionId: options?.targetDataCollectionId
  });
  const reqOpts = disconnectFromSharedCollection(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          dataCollectionId: "$[0].dataCollectionId",
          targetDataCollectionId: "$[0].targetDataCollectionId"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
export {
  WebhookIdentityType,
  connectToSharedCollection2 as connectToSharedCollection,
  createDataSharingPolicy2 as createDataSharingPolicy,
  deleteDataSharingPolicy2 as deleteDataSharingPolicy,
  disconnectFromSharedCollection2 as disconnectFromSharedCollection,
  getDataSharingPolicy2 as getDataSharingPolicy,
  listDataSharingPolicies2 as listDataSharingPolicies,
  listSharedDataCollections2 as listSharedDataCollections,
  updateDataSharingPolicy2 as updateDataSharingPolicy
};
