// @wix/bex-utils/@wix/ambassador-media-site-media-v1-file-descriptor/build/cjs/builders.impl.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.aUnsupportedRequestValueError = exports.aUpdateFileDescriptorResponse = exports.aUpdateFileDescriptorRequest = exports.aGetFileDescriptorsResponse = exports.aGetFileDescriptorsRequest = exports.aGetFileDescriptorResponse = exports.aGetFileDescriptorRequest = exports.aServiceError = exports.aDownloadUrl = exports.aGenerateFileDownloadUrlResponse = exports.aContentDisposition = exports.aGenerateFileDownloadUrlRequest = exports.aGenerateFilesDownloadUrlResponse = exports.aGenerateFilesDownloadUrlRequest = exports.aBulkActionMetadata = exports.aItemMetadata = exports.aBulkAnnotateImageResult = exports.aBulkAnnotateImagesResponse = exports.aImageAnnotationType = exports.aBulkAnnotateImagesRequest = exports.aDraftFilePublished = exports.aApplicationError = exports.aFileFailed = exports.aExternalInfo = exports.aFileReady = exports.aIdentityType = exports.aIdentityInfo = exports.aNamespace = exports.aState = exports.aOperationStatus = exports.aFontAsset = exports.aFontMedia = exports.aOtherMedia = exports.aModel3D = exports.aArchive = exports.aDocument = exports.aAudio = exports.aAudioV2 = exports.aVideoResolution = exports.aVideoV2 = exports.aFaceRecognition = exports.aColorRGB = exports.aColor = exports.aColors = exports.aFocalPoint = exports.aImage = exports.aImageMedia = exports.aFileMedia = exports.aMediaType = exports.aFileDescriptor = void 0;
exports.aMessageEnvelope = exports.aActionEvent = exports.aEntityDeletedEvent = exports.aEntityUpdatedEvent = exports.aRestoreInfo = exports.aEntityCreatedEvent = exports.aDomainEvent = exports.aUpdateFileResponse = exports.aUpdateFileRequest = exports.aBulkPublishDraftFileResult = exports.aBulkPublishDraftFilesResponse = exports.aBulkPublishDraftFilesRequest = exports.aListDeletedFilesResponse = exports.aListDeletedFilesRequest = exports.aBulkRestoreFilesFromTrashBinResponse = exports.aBulkRestoreFilesFromTrashBinRequest = exports.aBulkDeleteFilesResponse = exports.aBulkDeleteFilesRequest = exports.aGenerateWebSocketTokenResponse = exports.aGenerateWebSocketTokenRequest = exports.aGenerateVideoStreamingUrlResponse = exports.aStreamFormat = exports.aGenerateVideoStreamingUrlRequest = exports.aSearchFilesResponse = exports.aRootFolder = exports.aSearchFilesRequest = exports.aCursors = exports.aPagingMetadataV2 = exports.aListFilesResponse = exports.aCursorPaging = exports.aSortOrder = exports.aSorting = exports.aListFilesRequest = exports.aBulkImportFileResult = exports.aBulkImportFileResponse = exports.aBulkImportFileRequest = exports.aBulkImportFilesResponse = exports.aBulkImportFilesRequest = exports.aImportFileResponse = exports.aImportFileRequest = exports.aGenerateFileResumableUploadUrlResponse = exports.aUploadProtocol = exports.aGenerateFileResumableUploadUrlRequest = exports.aPlan = exports.aPlans = exports.aTotalQuota = exports.aSiteQuotaExceededError = exports.aGenerateFileUploadUrlResponse = exports.aImageAnnotationConfigurations = exports.aGenerateFileUploadUrlRequest = void 0;
exports.aWebhookIdentityType = exports.aIdentificationData = void 0;
var metro_common_builders_1 = require("@wix/metro-common-builders");
var types_impl_1 = require("./types.impl");
// @ts-ignore
var callsite_1 = __importDefault(require("callsite"));
function aFileDescriptor(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        id: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MAX_LENGTH', value: '1000' },
            ],
        }),
        displayName: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'READ_ONLY' },
                { type: 'MAX_LENGTH', value: '200' },
            ],
        }),
        url: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'WEB_URL' },
            ],
        }),
        parentFolderId: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'READ_ONLY' },
                { type: 'MAX_LENGTH', value: '100' },
            ],
        }),
        hash: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MAX_LENGTH', value: '100' },
            ],
        }),
        sizeInBytes: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
                { type: 'DECIMAL_VALUE', value: { maxScale: 0 } },
            ],
        }),
        private: true,
        mediaType: aMediaType(),
        media: aFileMedia(),
        operationStatus: aOperationStatus(),
        sourceUrl: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'WEB_URL' },
            ],
        }),
        thumbnailUrl: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'WEB_URL' },
            ],
        }),
        labels: [
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'READ_ONLY' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'READ_ONLY' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'READ_ONLY' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'READ_ONLY' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'READ_ONLY' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'READ_ONLY' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'READ_ONLY' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'READ_ONLY' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'READ_ONLY' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'READ_ONLY' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
        ],
        createdDate: (0, metro_common_builders_1.someDate)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        updatedDate: (0, metro_common_builders_1.someDate)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        siteId: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MAX_LENGTH', value: '100' },
            ],
        }),
        state: aState(),
        internalTags: [
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
        ],
        namespace: aNamespace(),
        addedBy: aIdentityInfo(),
        rawNamespace: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MAX_LENGTH', value: '200' },
            ],
        }),
        lastUsedDate: (0, metro_common_builders_1.someDate)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        mimeType: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MAX_LENGTH', value: '200' },
            ],
        }),
    }, overrideObject);
}
exports.aFileDescriptor = aFileDescriptor;
function aMediaType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.MediaType.UNKNOWN; }
    return enumFirstItem;
}
exports.aMediaType = aMediaType;
function aFileMedia(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        image: aImageMedia(),
    }, overrideObject);
}
exports.aFileMedia = aFileMedia;
function aImageMedia(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        image: aImage(),
        colors: aColors(),
        faces: [
            aFaceRecognition(),
            aFaceRecognition(),
            aFaceRecognition(),
            aFaceRecognition(),
            aFaceRecognition(),
            aFaceRecognition(),
            aFaceRecognition(),
            aFaceRecognition(),
            aFaceRecognition(),
            aFaceRecognition(),
        ],
        previewImage: aImage(),
        caption: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MAX_LENGTH', value: '300' },
            ],
        }),
        containsText: true,
        animated: true,
    }, overrideObject);
}
exports.aImageMedia = aImageMedia;
function aImage(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        id: (0, metro_common_builders_1.someString)(),
        url: (0, metro_common_builders_1.someString)(),
        height: (0, metro_common_builders_1.someNumber)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        width: (0, metro_common_builders_1.someNumber)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        altText: (0, metro_common_builders_1.someString)(),
        urlExpirationDate: (0, metro_common_builders_1.someDate)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        filename: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        sizeInBytes: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
            ],
        }),
        focalPoint: aFocalPoint(),
    }, overrideObject);
}
exports.aImage = aImage;
function aFocalPoint(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        x: (0, metro_common_builders_1.someNumber)(),
        y: (0, metro_common_builders_1.someNumber)(),
        height: (0, metro_common_builders_1.someNumber)(),
        width: (0, metro_common_builders_1.someNumber)(),
    }, overrideObject);
}
exports.aFocalPoint = aFocalPoint;
function aColors(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        prominent: aColor(),
        palette: [
            aColor(),
            aColor(),
            aColor(),
            aColor(),
            aColor(),
            aColor(),
            aColor(),
            aColor(),
            aColor(),
            aColor(),
        ],
    }, overrideObject);
}
exports.aColors = aColors;
function aColor(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        hex: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'FORMAT', value: 'COLOR_HEX' },
            ],
        }),
        rgb: aColorRGB(),
    }, overrideObject);
}
exports.aColor = aColor;
function aColorRGB(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        r: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'MIN' },
                { type: 'MAX', value: '255' },
            ],
        }),
        g: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'MIN' },
                { type: 'MAX', value: '255' },
            ],
        }),
        b: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'MIN' },
                { type: 'MAX', value: '255' },
            ],
        }),
    }, overrideObject);
}
exports.aColorRGB = aColorRGB;
function aFaceRecognition(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        confidence: (0, metro_common_builders_1.someNumber)(),
        x: (0, metro_common_builders_1.someNumber)(),
        y: (0, metro_common_builders_1.someNumber)(),
        height: (0, metro_common_builders_1.someNumber)(),
        width: (0, metro_common_builders_1.someNumber)(),
    }, overrideObject);
}
exports.aFaceRecognition = aFaceRecognition;
function aVideoV2(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        id: (0, metro_common_builders_1.someString)(),
        url: (0, metro_common_builders_1.someString)(),
        resolutions: [
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
        ],
        filename: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        posters: [
            aImage(),
            aImage(),
            aImage(),
            aImage(),
            aImage(),
            aImage(),
            aImage(),
            aImage(),
            aImage(),
            aImage(),
        ],
        sizeInBytes: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
            ],
        }),
        urlExpirationDate: (0, metro_common_builders_1.someDate)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        durationInMilliseconds: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MIN' },
            ],
        }),
        title: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '300' }],
        }),
        description: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '2000' }],
        }),
        height: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MIN' },
            ],
        }),
        width: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MIN' },
            ],
        }),
    }, overrideObject);
}
exports.aVideoV2 = aVideoV2;
function aVideoResolution(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        url: (0, metro_common_builders_1.someString)(),
        height: (0, metro_common_builders_1.someNumber)(),
        width: (0, metro_common_builders_1.someNumber)(),
        poster: aImage(),
        format: (0, metro_common_builders_1.someString)(),
        urlExpirationDate: (0, metro_common_builders_1.someDate)(),
        sizeInBytes: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
            ],
        }),
        quality: (0, metro_common_builders_1.someString)(),
        filename: (0, metro_common_builders_1.someString)(),
        durationInSeconds: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MIN' },
            ],
        }),
        durationInMilliseconds: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MIN' },
            ],
        }),
        private: true,
        assetKey: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
    }, overrideObject);
}
exports.aVideoResolution = aVideoResolution;
function aAudioV2(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        id: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
        assets: [
            aAudio(),
            aAudio(),
            aAudio(),
            aAudio(),
            aAudio(),
            aAudio(),
            aAudio(),
            aAudio(),
            aAudio(),
            aAudio(),
        ],
        bitrate: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MIN' },
            ],
        }),
        format: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MAX_LENGTH', value: '100' },
            ],
        }),
        duration: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MIN' },
            ],
        }),
        sizeInBytes: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
                { type: 'DECIMAL_VALUE', value: { maxScale: 0 } },
            ],
        }),
    }, overrideObject);
}
exports.aAudioV2 = aAudioV2;
function aAudio(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        id: (0, metro_common_builders_1.someString)(),
        url: (0, metro_common_builders_1.someString)(),
        urlExpirationDate: (0, metro_common_builders_1.someDate)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        sizeInBytes: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
            ],
        }),
        filename: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        duration: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MIN' },
            ],
        }),
        private: true,
        assetKey: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        format: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        quality: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
    }, overrideObject);
}
exports.aAudio = aAudio;
function aDocument(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        id: (0, metro_common_builders_1.someString)(),
        url: (0, metro_common_builders_1.someString)(),
        urlExpirationDate: (0, metro_common_builders_1.someDate)(),
        sizeInBytes: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
            ],
        }),
        filename: (0, metro_common_builders_1.someString)(),
        thumbnail: aImage(),
    }, overrideObject);
}
exports.aDocument = aDocument;
function aArchive(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        id: (0, metro_common_builders_1.someString)(),
        url: (0, metro_common_builders_1.someString)(),
        urlExpirationDate: (0, metro_common_builders_1.someDate)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        sizeInBytes: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
            ],
        }),
        filename: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aArchive = aArchive;
function aModel3D(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        id: (0, metro_common_builders_1.someString)(),
        url: (0, metro_common_builders_1.someString)(),
        thumbnail: aImage(),
        altText: (0, metro_common_builders_1.someString)(),
        urlExpirationDate: (0, metro_common_builders_1.someDate)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        filename: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        sizeInBytes: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
            ],
        }),
    }, overrideObject);
}
exports.aModel3D = aModel3D;
function aOtherMedia(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        id: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
        internalMediaType: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MAX_LENGTH', value: '100' },
            ],
        }),
        sizeInBytes: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
                {
                    type: 'DECIMAL_VALUE',
                    value: { gte: '0', maxScale: 0 },
                },
            ],
        }),
        url: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'WEB_URL' }],
        }),
    }, overrideObject);
}
exports.aOtherMedia = aOtherMedia;
function aFontMedia(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        id: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
        sizeInBytes: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
                {
                    type: 'DECIMAL_VALUE',
                    value: { gte: '0', maxScale: 0 },
                },
            ],
        }),
        format: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
        family: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        fontName: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        assets: [
            aFontAsset(),
            aFontAsset(),
            aFontAsset(),
            aFontAsset(),
            aFontAsset(),
        ],
    }, overrideObject);
}
exports.aFontMedia = aFontMedia;
function aFontAsset(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        assetKey: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        url: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
        format: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
    }, overrideObject);
}
exports.aFontAsset = aFontAsset;
function aOperationStatus(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.OperationStatus.FAILED; }
    return enumFirstItem;
}
exports.aOperationStatus = aOperationStatus;
function aState(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.State.OK; }
    return enumFirstItem;
}
exports.aState = aState;
function aNamespace(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.Namespace.NO_NAMESPACE; }
    return enumFirstItem;
}
exports.aNamespace = aNamespace;
function aIdentityInfo(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        identityType: aIdentityType(),
        identityId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
exports.aIdentityInfo = aIdentityInfo;
function aIdentityType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.IdentityType.UNKNOWN; }
    return enumFirstItem;
}
exports.aIdentityType = aIdentityType;
function aFileReady(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        file: aFileDescriptor(),
        externalInfo: aExternalInfo(),
        triggeredByUndelete: true,
    }, overrideObject);
}
exports.aFileReady = aFileReady;
function aExternalInfo(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        origin: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '1000' }],
        }),
        externalIds: [
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '30' },
                    { type: 'MAX_LENGTH', value: '4000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '30' },
                    { type: 'MAX_LENGTH', value: '4000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '30' },
                    { type: 'MAX_LENGTH', value: '4000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '30' },
                    { type: 'MAX_LENGTH', value: '4000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '30' },
                    { type: 'MAX_LENGTH', value: '4000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '30' },
                    { type: 'MAX_LENGTH', value: '4000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '30' },
                    { type: 'MAX_LENGTH', value: '4000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '30' },
                    { type: 'MAX_LENGTH', value: '4000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '30' },
                    { type: 'MAX_LENGTH', value: '4000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '30' },
                    { type: 'MAX_LENGTH', value: '4000' },
                ],
            }),
        ],
    }, overrideObject);
}
exports.aExternalInfo = aExternalInfo;
function aFileFailed(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        externalInfo: aExternalInfo(),
        error: aApplicationError(),
    }, overrideObject);
}
exports.aFileFailed = aFileFailed;
function aApplicationError(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        code: (0, metro_common_builders_1.someString)(),
        description: (0, metro_common_builders_1.someString)(),
        data: {},
    }, overrideObject);
}
exports.aApplicationError = aApplicationError;
function aDraftFilePublished(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        file: aFileDescriptor(),
        namespace: aNamespace(),
    }, overrideObject);
}
exports.aDraftFilePublished = aDraftFilePublished;
function aBulkAnnotateImagesRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        fileIds: [
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MIN_SIZE', value: '1' },
                    { type: 'MAX_SIZE', value: '10' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
        ],
        annotationTypes: [aImageAnnotationType()],
        overwrite: true,
        returnEntity: true,
    }, overrideObject);
}
exports.aBulkAnnotateImagesRequest = aBulkAnnotateImagesRequest;
function aImageAnnotationType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.ImageAnnotationType.UNKNOWN_IMAGE_ANNOTATION_TYPE; }
    return enumFirstItem;
}
exports.aImageAnnotationType = aImageAnnotationType;
function aBulkAnnotateImagesResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        results: [aBulkAnnotateImageResult()],
        bulkActionMetadata: aBulkActionMetadata(),
    }, overrideObject);
}
exports.aBulkAnnotateImagesResponse = aBulkAnnotateImagesResponse;
function aBulkAnnotateImageResult(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        itemMetadata: aItemMetadata(),
        item: aFileDescriptor(),
    }, overrideObject);
}
exports.aBulkAnnotateImageResult = aBulkAnnotateImageResult;
function aItemMetadata(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        id: (0, metro_common_builders_1.someString)(),
        originalIndex: (0, metro_common_builders_1.someNumber)(),
        success: true,
        error: aApplicationError(),
    }, overrideObject);
}
exports.aItemMetadata = aItemMetadata;
function aBulkActionMetadata(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        totalSuccesses: (0, metro_common_builders_1.someNumber)(),
        totalFailures: (0, metro_common_builders_1.someNumber)(),
        undetailedFailures: (0, metro_common_builders_1.someNumber)(),
    }, overrideObject);
}
exports.aBulkActionMetadata = aBulkActionMetadata;
function aGenerateFilesDownloadUrlRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        fileIds: [
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MIN_SIZE', value: '1' },
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
        ],
    }, overrideObject);
}
exports.aGenerateFilesDownloadUrlRequest = aGenerateFilesDownloadUrlRequest;
function aGenerateFilesDownloadUrlResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        downloadUrl: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'WEB_URL' }],
        }),
    }, overrideObject);
}
exports.aGenerateFilesDownloadUrlResponse = aGenerateFilesDownloadUrlResponse;
function aGenerateFileDownloadUrlRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        fileId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '1000' }],
        }),
        downloadFileName: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        expirationInMinutes: (0, metro_common_builders_1.someNumber)({
            validations: [{ type: 'MIN', value: '1' }],
        }),
        expirationRedirectUrl: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'WEB_URL' }],
        }),
        assetKeys: [
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '100' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '100' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '100' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '100' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '100' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '100' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '100' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '100' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '100' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '100' },
                ],
            }),
        ],
        contentDisposition: aContentDisposition(),
    }, overrideObject);
}
exports.aGenerateFileDownloadUrlRequest = aGenerateFileDownloadUrlRequest;
function aContentDisposition(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.ContentDisposition.ATTACHMENT; }
    return enumFirstItem;
}
exports.aContentDisposition = aContentDisposition;
function aGenerateFileDownloadUrlResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        downloadUrls: [
            aDownloadUrl(),
            aDownloadUrl(),
            aDownloadUrl(),
            aDownloadUrl(),
            aDownloadUrl(),
            aDownloadUrl(),
            aDownloadUrl(),
            aDownloadUrl(),
            aDownloadUrl(),
            aDownloadUrl(),
        ],
    }, overrideObject);
}
exports.aGenerateFileDownloadUrlResponse = aGenerateFileDownloadUrlResponse;
function aDownloadUrl(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        url: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'WEB_URL' }],
        }),
        assetKey: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
    }, overrideObject);
}
exports.aDownloadUrl = aDownloadUrl;
function aServiceError(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        internalCode: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
        internalHttpStatusCode: (0, metro_common_builders_1.someNumber)(),
        internalKey: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
    }, overrideObject);
}
exports.aServiceError = aServiceError;
function aGetFileDescriptorRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        fileId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '1000' }],
        }),
    }, overrideObject);
}
exports.aGetFileDescriptorRequest = aGetFileDescriptorRequest;
function aGetFileDescriptorResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        file: aFileDescriptor(),
    }, overrideObject);
}
exports.aGetFileDescriptorResponse = aGetFileDescriptorResponse;
function aGetFileDescriptorsRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        fileIds: [
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MIN_SIZE', value: '1' },
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
        ],
    }, overrideObject);
}
exports.aGetFileDescriptorsRequest = aGetFileDescriptorsRequest;
function aGetFileDescriptorsResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        files: [
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
        ],
    }, overrideObject);
}
exports.aGetFileDescriptorsResponse = aGetFileDescriptorsResponse;
function aUpdateFileDescriptorRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        file: aFileDescriptor(),
        fieldMask: (0, metro_common_builders_1.someRepeated)(),
    }, overrideObject);
}
exports.aUpdateFileDescriptorRequest = aUpdateFileDescriptorRequest;
function aUpdateFileDescriptorResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        file: aFileDescriptor(),
    }, overrideObject);
}
exports.aUpdateFileDescriptorResponse = aUpdateFileDescriptorResponse;
function aUnsupportedRequestValueError(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        allowedValues: [
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
        ],
        requestValue: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
    }, overrideObject);
}
exports.aUnsupportedRequestValueError = aUnsupportedRequestValueError;
function aGenerateFileUploadUrlRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        mimeType: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        fileName: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
        sizeInBytes: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
                {
                    type: 'DECIMAL_VALUE',
                    value: { gte: '0', maxScale: 0 },
                },
            ],
        }),
        parentFolderId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        private: true,
        labels: [
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '50' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '50' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '50' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '50' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '50' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '50' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '50' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '50' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '50' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '50' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
        ],
        externalInfo: aExternalInfo(),
        rawNamespace: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        namespace: aNamespace(),
        filePath: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        internalTags: [
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
        ],
        draftTtlInHours: (0, metro_common_builders_1.someNumber)({
            validations: [{ type: 'MIN', value: '1' }],
        }),
        imageAnnotationConfigurations: aImageAnnotationConfigurations(),
    }, overrideObject);
}
exports.aGenerateFileUploadUrlRequest = aGenerateFileUploadUrlRequest;
function aImageAnnotationConfigurations(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        skipDetection: true,
        annotationTypes: [
            aImageAnnotationType(),
            aImageAnnotationType(),
            aImageAnnotationType(),
            aImageAnnotationType(),
            aImageAnnotationType(),
            aImageAnnotationType(),
            aImageAnnotationType(),
            aImageAnnotationType(),
            aImageAnnotationType(),
            aImageAnnotationType(),
        ],
    }, overrideObject);
}
exports.aImageAnnotationConfigurations = aImageAnnotationConfigurations;
function aGenerateFileUploadUrlResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        uploadUrl: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'WEB_URL' }],
        }),
    }, overrideObject);
}
exports.aGenerateFileUploadUrlResponse = aGenerateFileUploadUrlResponse;
function aSiteQuotaExceededError(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        internalCode: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
        internalHttpStatusCode: (0, metro_common_builders_1.someNumber)(),
        internalKey: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
        quota: aTotalQuota(),
    }, overrideObject);
}
exports.aSiteQuotaExceededError = aSiteQuotaExceededError;
function aTotalQuota(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        storage: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'DECIMAL_VALUE' }],
        }),
        plans: aPlans(),
        timestamp: (0, metro_common_builders_1.someDate)(),
    }, overrideObject);
}
exports.aTotalQuota = aTotalQuota;
function aPlans(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        premium: aPlan(),
        upgradeUrl: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'WEB_URL' }],
        }),
    }, overrideObject);
}
exports.aPlans = aPlans;
function aPlan(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        id: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'GUID' },
            ],
        }),
        name: (0, metro_common_builders_1.someString)(),
        createdAt: (0, metro_common_builders_1.someDate)(),
        expiresAt: (0, metro_common_builders_1.someDate)(),
    }, overrideObject);
}
exports.aPlan = aPlan;
function aGenerateFileResumableUploadUrlRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        mimeType: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        fileName: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
        sizeInBytes: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
                {
                    type: 'DECIMAL_VALUE',
                    value: { gte: '0', maxScale: 0 },
                },
            ],
        }),
        parentFolderId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        private: true,
        labels: [
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
        ],
        uploadProtocol: aUploadProtocol(),
        rawNamespace: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        namespace: aNamespace(),
        filePath: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        internalTags: [
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
        ],
        draftTtlInHours: (0, metro_common_builders_1.someNumber)({
            validations: [{ type: 'MIN', value: '1' }],
        }),
        imageAnnotationConfigurations: aImageAnnotationConfigurations(),
    }, overrideObject);
}
exports.aGenerateFileResumableUploadUrlRequest = aGenerateFileResumableUploadUrlRequest;
function aUploadProtocol(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.UploadProtocol.TUS; }
    return enumFirstItem;
}
exports.aUploadProtocol = aUploadProtocol;
function aGenerateFileResumableUploadUrlResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        uploadProtocol: aUploadProtocol(),
        uploadUrl: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'WEB_URL' }],
        }),
        uploadToken: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '2000' }],
        }),
    }, overrideObject);
}
exports.aGenerateFileResumableUploadUrlResponse = aGenerateFileResumableUploadUrlResponse;
function aImportFileRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        url: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'WEB_URL' }],
        }),
        mediaType: aMediaType(),
        displayName: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
        parentFolderId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        private: true,
        labels: [
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
        ],
        mimeType: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        externalInfo: aExternalInfo(),
        urlParams: {},
        urlHeaders: {},
        rawNamespace: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        namespace: aNamespace(),
        filePath: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        internalTags: [
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
        ],
        draftTtlInHours: (0, metro_common_builders_1.someNumber)({
            validations: [{ type: 'MIN', value: '1' }],
        }),
        imageAnnotationConfigurations: aImageAnnotationConfigurations(),
    }, overrideObject);
}
exports.aImportFileRequest = aImportFileRequest;
function aImportFileResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        file: aFileDescriptor(),
    }, overrideObject);
}
exports.aImportFileResponse = aImportFileResponse;
function aBulkImportFilesRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        importFileRequests: [aImportFileRequest()],
    }, overrideObject);
}
exports.aBulkImportFilesRequest = aBulkImportFilesRequest;
function aBulkImportFilesResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        files: [
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
        ],
    }, overrideObject);
}
exports.aBulkImportFilesResponse = aBulkImportFilesResponse;
function aBulkImportFileRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        importFileRequests: [aImportFileRequest()],
        returnEntity: true,
    }, overrideObject);
}
exports.aBulkImportFileRequest = aBulkImportFileRequest;
function aBulkImportFileResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        results: [aBulkImportFileResult()],
        bulkActionMetadata: aBulkActionMetadata(),
    }, overrideObject);
}
exports.aBulkImportFileResponse = aBulkImportFileResponse;
function aBulkImportFileResult(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        itemMetadata: aItemMetadata(),
        item: aFileDescriptor(),
    }, overrideObject);
}
exports.aBulkImportFileResult = aBulkImportFileResult;
function aListFilesRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        parentFolderId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        mediaTypes: [
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
        ],
        private: true,
        sort: aSorting(),
        paging: aCursorPaging(),
        rawNamespace: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        namespace: aNamespace(),
    }, overrideObject);
}
exports.aListFilesRequest = aListFilesRequest;
function aSorting(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        fieldName: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '512' }],
        }),
        order: aSortOrder(),
        selectItemsBy: [{}, {}, {}, {}, {}, {}, {}, {}, {}, {}],
    }, overrideObject);
}
exports.aSorting = aSorting;
function aSortOrder(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.SortOrder.ASC; }
    return enumFirstItem;
}
exports.aSortOrder = aSortOrder;
function aCursorPaging(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        limit: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'MIN' },
                { type: 'MAX', value: '100' },
            ],
        }),
        cursor: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MAX_LENGTH', value: '16000' },
            ],
        }),
    }, overrideObject);
}
exports.aCursorPaging = aCursorPaging;
function aListFilesResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        files: [
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
        ],
        nextCursor: aPagingMetadataV2(),
    }, overrideObject);
}
exports.aListFilesResponse = aListFilesResponse;
function aPagingMetadataV2(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        total: (0, metro_common_builders_1.someNumber)(),
        cursors: aCursors(),
        hasNext: true,
    }, overrideObject);
}
exports.aPagingMetadataV2 = aPagingMetadataV2;
function aCursors(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        next: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MAX_LENGTH', value: '16000' },
            ],
        }),
    }, overrideObject);
}
exports.aCursors = aCursors;
function aSearchFilesRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        search: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
        rootFolder: aRootFolder(),
        mediaTypes: [
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
        ],
        private: true,
        sort: aSorting(),
        paging: aCursorPaging(),
        internalTag: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
        rawNamespace: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        namespace: aNamespace(),
    }, overrideObject);
}
exports.aSearchFilesRequest = aSearchFilesRequest;
function aRootFolder(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.RootFolder.MEDIA_ROOT; }
    return enumFirstItem;
}
exports.aRootFolder = aRootFolder;
function aSearchFilesResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        files: [
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
        ],
        nextCursor: aPagingMetadataV2(),
    }, overrideObject);
}
exports.aSearchFilesResponse = aSearchFilesResponse;
function aGenerateVideoStreamingUrlRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        fileId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '1000' }],
        }),
        format: aStreamFormat(),
    }, overrideObject);
}
exports.aGenerateVideoStreamingUrlRequest = aGenerateVideoStreamingUrlRequest;
function aStreamFormat(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.StreamFormat.UNKNOWN; }
    return enumFirstItem;
}
exports.aStreamFormat = aStreamFormat;
function aGenerateVideoStreamingUrlResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        downloadUrl: aDownloadUrl(),
    }, overrideObject);
}
exports.aGenerateVideoStreamingUrlResponse = aGenerateVideoStreamingUrlResponse;
function aGenerateWebSocketTokenRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({}, overrideObject);
}
exports.aGenerateWebSocketTokenRequest = aGenerateWebSocketTokenRequest;
function aGenerateWebSocketTokenResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        token: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '2000' }],
        }),
    }, overrideObject);
}
exports.aGenerateWebSocketTokenResponse = aGenerateWebSocketTokenResponse;
function aBulkDeleteFilesRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        fileIds: [
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
        ],
        permanent: true,
    }, overrideObject);
}
exports.aBulkDeleteFilesRequest = aBulkDeleteFilesRequest;
function aBulkDeleteFilesResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({}, overrideObject);
}
exports.aBulkDeleteFilesResponse = aBulkDeleteFilesResponse;
function aBulkRestoreFilesFromTrashBinRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        fileIds: [
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
        ],
    }, overrideObject);
}
exports.aBulkRestoreFilesFromTrashBinRequest = aBulkRestoreFilesFromTrashBinRequest;
function aBulkRestoreFilesFromTrashBinResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({}, overrideObject);
}
exports.aBulkRestoreFilesFromTrashBinResponse = aBulkRestoreFilesFromTrashBinResponse;
function aListDeletedFilesRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        parentFolderId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        mediaTypes: [
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
        ],
        private: true,
        sort: aSorting(),
        paging: aCursorPaging(),
        rawNamespace: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        namespace: aNamespace(),
    }, overrideObject);
}
exports.aListDeletedFilesRequest = aListDeletedFilesRequest;
function aListDeletedFilesResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        files: [
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
        ],
        nextCursor: aPagingMetadataV2(),
    }, overrideObject);
}
exports.aListDeletedFilesResponse = aListDeletedFilesResponse;
function aBulkPublishDraftFilesRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        fileIds: [
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
        ],
        namespace: aNamespace(),
        parentFolderId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        filePath: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        returnEntity: true,
    }, overrideObject);
}
exports.aBulkPublishDraftFilesRequest = aBulkPublishDraftFilesRequest;
function aBulkPublishDraftFilesResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        results: [aBulkPublishDraftFileResult()],
        bulkActionMetadata: aBulkActionMetadata(),
    }, overrideObject);
}
exports.aBulkPublishDraftFilesResponse = aBulkPublishDraftFilesResponse;
function aBulkPublishDraftFileResult(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        itemMetadata: aItemMetadata(),
        item: aFileDescriptor(),
    }, overrideObject);
}
exports.aBulkPublishDraftFileResult = aBulkPublishDraftFileResult;
function aUpdateFileRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        fileId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '1000' }],
        }),
        displayName: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
        parentFolderId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        labels: [
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
        ],
    }, overrideObject);
}
exports.aUpdateFileRequest = aUpdateFileRequest;
function aUpdateFileResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        file: aFileDescriptor(),
    }, overrideObject);
}
exports.aUpdateFileResponse = aUpdateFileResponse;
function aDomainEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        createdEvent: aEntityCreatedEvent(),
        id: (0, metro_common_builders_1.someString)(),
        entityFqdn: (0, metro_common_builders_1.someString)(),
        slug: (0, metro_common_builders_1.someString)(),
        entityId: (0, metro_common_builders_1.someString)(),
        eventTime: (0, metro_common_builders_1.someDate)(),
        triggeredByAnonymizeRequest: true,
        originatedFrom: (0, metro_common_builders_1.someString)(),
        entityEventSequence: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'DECIMAL_VALUE' }],
        }),
    }, overrideObject);
}
exports.aDomainEvent = aDomainEvent;
function aEntityCreatedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        entityAsJson: (0, metro_common_builders_1.someString)(),
        triggeredByUndelete: true,
        restoreInfo: aRestoreInfo(),
        additionalMetadataAsJson: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aEntityCreatedEvent = aEntityCreatedEvent;
function aRestoreInfo(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        deletedDate: (0, metro_common_builders_1.someDate)(),
    }, overrideObject);
}
exports.aRestoreInfo = aRestoreInfo;
function aEntityUpdatedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        currentEntityAsJson: (0, metro_common_builders_1.someString)(),
        previousEntityAsJson: (0, metro_common_builders_1.someString)(),
        modifiedFields: { modifiedFields: (0, metro_common_builders_1.someAny)() },
        additionalMetadataAsJson: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aEntityUpdatedEvent = aEntityUpdatedEvent;
function aEntityDeletedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        movedToTrash: true,
        deletedEntityAsJson: (0, metro_common_builders_1.someString)(),
        additionalMetadataAsJson: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aEntityDeletedEvent = aEntityDeletedEvent;
function aActionEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        bodyAsJson: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aActionEvent = aActionEvent;
function aMessageEnvelope(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        instanceId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        eventType: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '150' }],
        }),
        identity: aIdentificationData(),
        data: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aMessageEnvelope = aMessageEnvelope;
function aIdentificationData(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        anonymousVisitorId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        identityType: aWebhookIdentityType(),
    }, overrideObject);
}
exports.aIdentificationData = aIdentificationData;
function aWebhookIdentityType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.WebhookIdentityType.UNKNOWN; }
    return enumFirstItem;
}
exports.aWebhookIdentityType = aWebhookIdentityType;
function isCurrentBuilderWasPreviouslyCalled() {
    var callChainFunctionNames = (0, callsite_1.default)().map(function (item) {
        return item.getFunctionName();
    });
    var callChainWithoutCurrentBuilder = callChainFunctionNames.splice(2);
    var currentBuilderName = callChainFunctionNames[1];
    return callChainWithoutCurrentBuilder.some(function (functionName) { return functionName === currentBuilderName; });
}
__exportStar(require("@wix/metro-common-builders"), exports);
