// @wix/bex-utils/@wix/ambassador-media-site-media-v1-file-descriptor/build/cjs/meta.impl.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateFile = exports.bulkPublishDraftFiles = exports.listDeletedFiles = exports.bulkRestoreFilesFromTrashBin = exports.bulkDeleteFiles = exports.generateWebSocketToken = exports.generateVideoStreamingUrl = exports.searchFiles = exports.listFiles = exports.bulkImportFile = exports.bulkImportFiles = exports.importFile = exports.generateFileResumableUploadUrl = exports.generateFileUploadUrl = exports.updateFileDescriptor = exports.getFileDescriptors = exports.getFileDescriptor = exports.generateFileDownloadUrl = exports.generateFilesDownloadUrl = exports.bulkAnnotateImages = void 0;
var httpWixMediaSiteMediaV1FileDescriptor = __importStar(require("./http.impl"));
function bulkAnnotateImages() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.bulkAnnotateImages(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/files/images/annotate',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkAnnotateImages = bulkAnnotateImages;
function generateFilesDownloadUrl() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.generateFilesDownloadUrl(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/files/generate-download-url',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.generateFilesDownloadUrl = generateFilesDownloadUrl;
function generateFileDownloadUrl() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.generateFileDownloadUrl(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/files/generate-file-download-url',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.generateFileDownloadUrl = generateFileDownloadUrl;
function getFileDescriptor() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.getFileDescriptor(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/v1/files/get-file-by-id',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getFileDescriptor = getFileDescriptor;
function getFileDescriptors() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.getFileDescriptors(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/files/get-files',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getFileDescriptors = getFileDescriptors;
function updateFileDescriptor() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.updateFileDescriptor(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'PATCH',
        path: '/v1/files/update-file-descriptor',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateFileDescriptor = updateFileDescriptor;
function generateFileUploadUrl() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.generateFileUploadUrl(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/files/generate-upload-url',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.generateFileUploadUrl = generateFileUploadUrl;
function generateFileResumableUploadUrl() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.generateFileResumableUploadUrl(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/files/generate-resumable-upload-url',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.generateFileResumableUploadUrl = generateFileResumableUploadUrl;
function importFile() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.importFile(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/files/import',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.importFile = importFile;
function bulkImportFiles() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.bulkImportFiles(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/files/import',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkImportFiles = bulkImportFiles;
function bulkImportFile() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.bulkImportFile(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/files/import-v2',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkImportFile = bulkImportFile;
function listFiles() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.listFiles(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/v1/files',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listFiles = listFiles;
function searchFiles() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.searchFiles(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/files/search',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.searchFiles = searchFiles;
function generateVideoStreamingUrl() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.generateVideoStreamingUrl(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/files/generate-video-stream-url',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.generateVideoStreamingUrl = generateVideoStreamingUrl;
function generateWebSocketToken() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.generateWebSocketToken(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/files/generate-web-socket-token',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.generateWebSocketToken = generateWebSocketToken;
function bulkDeleteFiles() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.bulkDeleteFiles(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/files/delete',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkDeleteFiles = bulkDeleteFiles;
function bulkRestoreFilesFromTrashBin() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.bulkRestoreFilesFromTrashBin(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/trash-bin/files/restore',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkRestoreFilesFromTrashBin = bulkRestoreFilesFromTrashBin;
function listDeletedFiles() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.listDeletedFiles(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/v1/trash-bin/files',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listDeletedFiles = listDeletedFiles;
function bulkPublishDraftFiles() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.bulkPublishDraftFiles(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/files/publish-drafts',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkPublishDraftFiles = bulkPublishDraftFiles;
function updateFile() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.updateFile(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'PATCH',
        path: '/v1/files/update-file',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateFile = updateFile;
