// @wix/design-system/dist/cjs/Badge/Badge.uni.driver.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.badgeUniDriverFactory = void 0;
var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _unidriver = require("../utils/test-utils/unidriver");
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2.default)(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var badgeUniDriverFactory = base => _objectSpread(_objectSpread({}, (0, _unidriver.baseUniDriverFactory)(base)), {}, {
  /**
   * Gets badge text
   */
  text: () => base.text(),
  /**
   * Gets badge type
   */
  getType: () => base.attr('data-type'),
  /**
   * Gets badge type
   */
  getSkin: () => base.attr('data-skin'),
  /**
   * Gets badge size
   */
  getSize: () => base.attr('data-size'),
  /**
   * Checks whether the text is uppercase
   */
  isUppercase: function () {
    var _isUppercase = (0, _asyncToGenerator2.default)(function* () {
      return (yield base.attr('data-uppercase')) === 'true';
    });
    function isUppercase() {
      return _isUppercase.apply(this, arguments);
    }
    return isUppercase;
  }(),
  /**
   * Checks whether badge is clickable
   */
  hasClickCursor: function () {
    var _hasClickCursor = (0, _asyncToGenerator2.default)(function* () {
      return (yield base.attr('data-clickable')) === 'true';
    });
    function hasClickCursor() {
      return _hasClickCursor.apply(this, arguments);
    }
    return hasClickCursor;
  }()
});
exports.badgeUniDriverFactory = badgeUniDriverFactory;
