// @wix/bex-utils/@wix/ambassador-os-tags-v1-tag/build/es/meta.impl.js
import * as httpWixOsTagsV1Tag from './http.impl';
export function createTagForAccount() {
    var payload = {};
    var getRequestOptions = httpWixOsTagsV1Tag.createTagForAccount(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/account/tags',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getTagForAccount() {
    var payload = { tagId: ':tagId' };
    var getRequestOptions = httpWixOsTagsV1Tag.getTagForAccount(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/v1/account/tags/{tagId}',
        pathParams: { tagId: 'tagId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateTagForAccount() {
    var payload = { tag: { id: ':tagId' } };
    var getRequestOptions = httpWixOsTagsV1Tag.updateTagForAccount(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'PATCH',
        path: '/v1/account/tags/{tag.id}',
        pathParams: { tagId: 'tagId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteTagForAccount() {
    var payload = { tagId: ':tagId' };
    var getRequestOptions = httpWixOsTagsV1Tag.deleteTagForAccount(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'DELETE',
        path: '/v1/account/tags/{tagId}',
        pathParams: { tagId: 'tagId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listTagsForAccount() {
    var payload = {};
    var getRequestOptions = httpWixOsTagsV1Tag.listTagsForAccount(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/v1/account/tags',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function createTag() {
    var payload = {};
    var getRequestOptions = httpWixOsTagsV1Tag.createTag(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/tags',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getTag() {
    var payload = { tagId: ':tagId' };
    var getRequestOptions = httpWixOsTagsV1Tag.getTag(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/v1/tags/{tagId}',
        pathParams: { tagId: 'tagId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateTag() {
    var payload = { tag: { id: ':tagId' } };
    var getRequestOptions = httpWixOsTagsV1Tag.updateTag(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'PATCH',
        path: '/v1/tags/{tag.id}',
        pathParams: { tagId: 'tagId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteTag() {
    var payload = { tagId: ':tagId' };
    var getRequestOptions = httpWixOsTagsV1Tag.deleteTag(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'DELETE',
        path: '/v1/tags/{tagId}',
        pathParams: { tagId: 'tagId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listTags() {
    var payload = {};
    var getRequestOptions = httpWixOsTagsV1Tag.listTags(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/v1/tags',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
