// @wix/auto_sdk_media_enterprise-media-categories/build/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  MediaType: () => MediaType,
  PublishStatus: () => PublishStatus,
  WebhookIdentityType: () => WebhookIdentityType,
  createCategory: () => createCategory2,
  deleteCategory: () => deleteCategory2,
  enterpriseOnboarding: () => enterpriseOnboarding2,
  getCategory: () => getCategory2,
  getMediaManagerCategories: () => getMediaManagerCategories2,
  updateCategory: () => updateCategory2
});
module.exports = __toCommonJS(index_typings_exports);

// src/media-enterprise-public-media-v1-enterprise-category-enterprise-media-categories.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/media-enterprise-public-media-v1-enterprise-category-enterprise-media-categories.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixMediaEnterprisePublicMediaV1EnterpriseCategoriesServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/enterprise-public-media-server",
        destPath: ""
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/_api/enterprise-public-media-server",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/enterprise-public-media-server",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/enterprise-public-media-server",
        destPath: ""
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/enterprise-public-media-server",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_media_enterprise-media-categories";
function createCategory(payload) {
  function __createCategory({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [{ path: "category.sortOrder" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "category.createdDate" },
          { path: "category.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_category",
      method: "POST",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseCategoriesService.CreateCategory",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseCategoriesServiceUrl(
        { protoPath: "/v1/enterprise/categories", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [{ path: "category.sortOrder" }]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "category.createdDate" },
            { path: "category.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createCategory;
}
function deleteCategory(payload) {
  function __deleteCategory({ host }) {
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_category",
      method: "DELETE",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseCategoriesService.DeleteCategory",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseCategoriesServiceUrl(
        {
          protoPath: "/v1/enterprise/categories/{categoryId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteCategory;
}
function updateCategory(payload) {
  function __updateCategory({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [{ path: "category.sortOrder" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "category.createdDate" },
          { path: "category.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_category",
      method: "PATCH",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseCategoriesService.UpdateCategory",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseCategoriesServiceUrl(
        {
          protoPath: "/v1/enterprise/categories/{category.id}",
          data: serializedData,
          host
        }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [{ path: "category.sortOrder" }]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "category.createdDate" },
            { path: "category.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateCategory;
}
function getCategory(payload) {
  function __getCategory({ host }) {
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_category",
      method: "GET",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseCategoriesService.GetCategory",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseCategoriesServiceUrl(
        {
          protoPath: "/v1/enterprise/categories/{categoryId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [{ path: "category.category.sortOrder" }]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "category.category.createdDate" },
            { path: "category.category.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getCategory;
}
function enterpriseOnboarding(payload) {
  function __enterpriseOnboarding({ host }) {
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_category",
      method: "POST",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseCategoriesService.EnterpriseOnboarding",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseCategoriesServiceUrl(
        { protoPath: "/v1/enterprise/onboarding", data: payload, host }
      ),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [{ path: "category.sortOrder" }]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "category.createdDate" },
            { path: "category.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __enterpriseOnboarding;
}
function getMediaManagerCategories(payload) {
  function __getMediaManagerCategories({ host }) {
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_category",
      method: "GET",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseCategoriesService.GetMediaManagerCategories",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseCategoriesServiceUrl(
        {
          protoPath: "/v1/enterprise/media-manager/categories",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [{ path: "category.category.sortOrder" }]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "category.category.createdDate" },
            { path: "category.category.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getMediaManagerCategories;
}

// src/media-enterprise-public-media-v1-enterprise-category-enterprise-media-categories.universal.ts
var PublishStatus = /* @__PURE__ */ ((PublishStatus2) => {
  PublishStatus2["UNDEFINED"] = "UNDEFINED";
  PublishStatus2["UNPUBLISHED"] = "UNPUBLISHED";
  PublishStatus2["PUBLISHED"] = "PUBLISHED";
  PublishStatus2["WIX_ONLY"] = "WIX_ONLY";
  return PublishStatus2;
})(PublishStatus || {});
var MediaType = /* @__PURE__ */ ((MediaType2) => {
  MediaType2["MIXED"] = "MIXED";
  MediaType2["IMAGE"] = "IMAGE";
  MediaType2["VIDEO"] = "VIDEO";
  MediaType2["AUDIO"] = "AUDIO";
  MediaType2["DOCUMENT"] = "DOCUMENT";
  MediaType2["VECTOR"] = "VECTOR";
  MediaType2["ARCHIVE"] = "ARCHIVE";
  MediaType2["MODEL3D"] = "MODEL3D";
  return MediaType2;
})(MediaType || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createCategory2(category) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ category });
  const reqOpts = createCategory(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.category;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { category: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["category"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteCategory2(categoryId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    categoryId
  });
  const reqOpts = deleteCategory(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { categoryId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["categoryId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateCategory2(_id, category) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    category: { ...category, id: _id }
  });
  const reqOpts = updateCategory(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.category;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { category: "$[1]" },
        explicitPathsToArguments: { "category.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "category"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getCategory2(categoryId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    categoryId,
    levels: options?.levels,
    publishStatus: options?.publishStatus
  });
  const reqOpts = getCategory(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.category;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          categoryId: "$[0]",
          levels: "$[1].levels",
          publishStatus: "$[1].publishStatus"
        },
        singleArgumentUnchanged: false
      },
      ["categoryId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function enterpriseOnboarding2(accountId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    accountId,
    accountName: options?.accountName
  });
  const reqOpts = enterpriseOnboarding(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          accountId: "$[0]",
          accountName: "$[1].accountName"
        },
        singleArgumentUnchanged: false
      },
      ["accountId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getMediaManagerCategories2() {
  const { httpClient, sideEffects } = arguments[0];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({});
  const reqOpts = getMediaManagerCategories(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {},
        singleArgumentUnchanged: false
      },
      []
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  MediaType,
  PublishStatus,
  WebhookIdentityType,
  createCategory,
  deleteCategory,
  enterpriseOnboarding,
  getCategory,
  getMediaManagerCategories,
  updateCategory
});
