// @wix/auto_sdk_media_enterprise-media-categories/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  createCategory: () => createCategory2,
  deleteCategory: () => deleteCategory2,
  enterpriseOnboarding: () => enterpriseOnboarding2,
  getCategory: () => getCategory2,
  getMediaManagerCategories: () => getMediaManagerCategories2,
  updateCategory: () => updateCategory2
});
module.exports = __toCommonJS(meta_exports);

// src/media-enterprise-public-media-v1-enterprise-category-enterprise-media-categories.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixMediaEnterprisePublicMediaV1EnterpriseCategoriesServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/enterprise-public-media-server",
        destPath: ""
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/_api/enterprise-public-media-server",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/enterprise-public-media-server",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/enterprise-public-media-server",
        destPath: ""
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/enterprise-public-media-server",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_media_enterprise-media-categories";
function createCategory(payload) {
  function __createCategory({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [{ path: "category.sortOrder" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "category.createdDate" },
          { path: "category.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_category",
      method: "POST",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseCategoriesService.CreateCategory",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseCategoriesServiceUrl(
        { protoPath: "/v1/enterprise/categories", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [{ path: "category.sortOrder" }]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "category.createdDate" },
            { path: "category.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createCategory;
}
function deleteCategory(payload) {
  function __deleteCategory({ host }) {
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_category",
      method: "DELETE",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseCategoriesService.DeleteCategory",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseCategoriesServiceUrl(
        {
          protoPath: "/v1/enterprise/categories/{categoryId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteCategory;
}
function updateCategory(payload) {
  function __updateCategory({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [{ path: "category.sortOrder" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "category.createdDate" },
          { path: "category.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_category",
      method: "PATCH",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseCategoriesService.UpdateCategory",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseCategoriesServiceUrl(
        {
          protoPath: "/v1/enterprise/categories/{category.id}",
          data: serializedData,
          host
        }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [{ path: "category.sortOrder" }]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "category.createdDate" },
            { path: "category.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateCategory;
}
function getCategory(payload) {
  function __getCategory({ host }) {
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_category",
      method: "GET",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseCategoriesService.GetCategory",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseCategoriesServiceUrl(
        {
          protoPath: "/v1/enterprise/categories/{categoryId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [{ path: "category.category.sortOrder" }]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "category.category.createdDate" },
            { path: "category.category.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getCategory;
}
function enterpriseOnboarding(payload) {
  function __enterpriseOnboarding({ host }) {
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_category",
      method: "POST",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseCategoriesService.EnterpriseOnboarding",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseCategoriesServiceUrl(
        { protoPath: "/v1/enterprise/onboarding", data: payload, host }
      ),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [{ path: "category.sortOrder" }]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "category.createdDate" },
            { path: "category.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __enterpriseOnboarding;
}
function getMediaManagerCategories(payload) {
  function __getMediaManagerCategories({ host }) {
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_category",
      method: "GET",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseCategoriesService.GetMediaManagerCategories",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseCategoriesServiceUrl(
        {
          protoPath: "/v1/enterprise/media-manager/categories",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [{ path: "category.category.sortOrder" }]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "category.category.createdDate" },
            { path: "category.category.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getMediaManagerCategories;
}

// src/media-enterprise-public-media-v1-enterprise-category-enterprise-media-categories.meta.ts
function createCategory2() {
  const payload = {};
  const getRequestOptions = createCategory(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/enterprise/categories",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteCategory2() {
  const payload = { categoryId: ":categoryId" };
  const getRequestOptions = deleteCategory(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/enterprise/categories/{categoryId}",
    pathParams: { categoryId: "categoryId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateCategory2() {
  const payload = { category: { id: ":categoryId" } };
  const getRequestOptions = updateCategory(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/enterprise/categories/{category.id}",
    pathParams: { categoryId: "categoryId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getCategory2() {
  const payload = { categoryId: ":categoryId" };
  const getRequestOptions = getCategory(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/enterprise/categories/{categoryId}",
    pathParams: { categoryId: "categoryId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function enterpriseOnboarding2() {
  const payload = {};
  const getRequestOptions = enterpriseOnboarding(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/enterprise/onboarding",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getMediaManagerCategories2() {
  const payload = {};
  const getRequestOptions = getMediaManagerCategories(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/enterprise/media-manager/categories",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  createCategory,
  deleteCategory,
  enterpriseOnboarding,
  getCategory,
  getMediaManagerCategories,
  updateCategory
});
