// @wix/auto_sdk_comments_widgets/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  SortOrderOriginal: () => SortOrder,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  createCommentsWidget: () => createCommentsWidget2,
  deleteCommentsWidget: () => deleteCommentsWidget2,
  getCommentsWidget: () => getCommentsWidget2,
  queryCommentsWidget: () => queryCommentsWidget2,
  queryCommentsWidgetResources: () => queryCommentsWidgetResources2,
  registerCommentsWidget: () => registerCommentsWidget2,
  updateCommentsWidget: () => updateCommentsWidget2
});
module.exports = __toCommonJS(meta_exports);

// src/comments-categories-v1-comments-widget-widgets.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixCommentsCategoriesV1CommentsWidgetRegistryUrl(opts) {
  const domainToMappings = {
    _: [
      {
        srcPath: "/_api/comments-widget-registry",
        destPath: "/v1/comments/widgets"
      }
    ],
    "dev._base_domain_": [
      {
        srcPath: "/_api/comments-widget-registry/v1/comments/widgets",
        destPath: "/v1/comments/widgets"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/comments-widget-registry/v1/comments/widgets",
        destPath: "/v1/comments/widgets"
      },
      {
        srcPath: "/_api/comments/widgets/v1",
        destPath: "/v1/comments/widgets"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/comments-widget-registry/v1/comments/widgets",
        destPath: "/v1/comments/widgets"
      },
      {
        srcPath: "/_api/comments/widgets/v1",
        destPath: "/v1/comments/widgets"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/comments-widget-registry/v1/comments/widgets",
        destPath: "/v1/comments/widgets"
      },
      {
        srcPath: "/_api/comments/widgets/v1",
        destPath: "/v1/comments/widgets"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/comments-widget-registry/v1/comments/widgets",
        destPath: "/v1/comments/widgets"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/comments-widget-registry/v1/comments/widgets",
        destPath: "/v1/comments/widgets"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/api/v1/comments/widgets",
        destPath: "/v1/comments/widgets"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/comments-widget-registry",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_comments_widgets";
function registerCommentsWidget(payload) {
  function __registerCommentsWidget({ host }) {
    const metadata = {
      entityFqdn: "wix.comments.categories.v1.comments_widget",
      method: "POST",
      methodFqn: "wix.comments.categories.v1.CommentsWidgetRegistry.RegisterCommentsWidget",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixCommentsCategoriesV1CommentsWidgetRegistryUrl({
        protoPath: "/v1/comments/widgets/register",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "commentsWidget.createdDate" },
            { path: "commentsWidget.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __registerCommentsWidget;
}
function queryCommentsWidgetResources(payload) {
  function __queryCommentsWidgetResources({ host }) {
    const metadata = {
      entityFqdn: "wix.comments.categories.v1.comments_widget",
      method: "POST",
      methodFqn: "wix.comments.categories.v1.CommentsWidgetRegistry.QueryCommentsWidgetResources",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixCommentsCategoriesV1CommentsWidgetRegistryUrl({
        protoPath: "/v1/comments/widgets/resources/query",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __queryCommentsWidgetResources;
}
function createCommentsWidget(payload) {
  function __createCommentsWidget({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "commentsWidget.createdDate" },
          { path: "commentsWidget.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.comments.categories.v1.comments_widget",
      method: "POST",
      methodFqn: "wix.comments.categories.v1.CommentsWidgetRegistry.CreateCommentsWidget",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixCommentsCategoriesV1CommentsWidgetRegistryUrl({
        protoPath: "/v1/comments/widgets",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "commentsWidget.createdDate" },
            { path: "commentsWidget.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createCommentsWidget;
}
function getCommentsWidget(payload) {
  function __getCommentsWidget({ host }) {
    const metadata = {
      entityFqdn: "wix.comments.categories.v1.comments_widget",
      method: "GET",
      methodFqn: "wix.comments.categories.v1.CommentsWidgetRegistry.GetCommentsWidget",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixCommentsCategoriesV1CommentsWidgetRegistryUrl({
        protoPath: "/v1/comments/widgets/{commentsWidgetId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "commentsWidget.createdDate" },
            { path: "commentsWidget.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getCommentsWidget;
}
function updateCommentsWidget(payload) {
  function __updateCommentsWidget({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "commentsWidget.createdDate" },
          { path: "commentsWidget.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.comments.categories.v1.comments_widget",
      method: "PATCH",
      methodFqn: "wix.comments.categories.v1.CommentsWidgetRegistry.UpdateCommentsWidget",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixCommentsCategoriesV1CommentsWidgetRegistryUrl({
        protoPath: "/v1/comments/widgets/{commentsWidget.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "commentsWidget.createdDate" },
            { path: "commentsWidget.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateCommentsWidget;
}
function deleteCommentsWidget(payload) {
  function __deleteCommentsWidget({ host }) {
    const metadata = {
      entityFqdn: "wix.comments.categories.v1.comments_widget",
      method: "DELETE",
      methodFqn: "wix.comments.categories.v1.CommentsWidgetRegistry.DeleteCommentsWidget",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixCommentsCategoriesV1CommentsWidgetRegistryUrl({
        protoPath: "/v1/comments/widgets/{commentsWidgetId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteCommentsWidget;
}
function queryCommentsWidget(payload) {
  function __queryCommentsWidget({ host }) {
    const metadata = {
      entityFqdn: "wix.comments.categories.v1.comments_widget",
      method: "POST",
      methodFqn: "wix.comments.categories.v1.CommentsWidgetRegistry.QueryCommentsWidget",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixCommentsCategoriesV1CommentsWidgetRegistryUrl({
        protoPath: "/v1/comments/widgets/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "commentsWidgets.createdDate" },
            { path: "commentsWidgets.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryCommentsWidget;
}

// src/comments-categories-v1-comments-widget-widgets.types.ts
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/comments-categories-v1-comments-widget-widgets.meta.ts
function registerCommentsWidget2() {
  const payload = {};
  const getRequestOptions = registerCommentsWidget(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/comments/widgets/register",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryCommentsWidgetResources2() {
  const payload = {};
  const getRequestOptions = queryCommentsWidgetResources(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/comments/widgets/resources/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createCommentsWidget2() {
  const payload = {};
  const getRequestOptions = createCommentsWidget(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/comments/widgets",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getCommentsWidget2() {
  const payload = { commentsWidgetId: ":commentsWidgetId" };
  const getRequestOptions = getCommentsWidget(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/comments/widgets/{commentsWidgetId}",
    pathParams: { commentsWidgetId: "commentsWidgetId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateCommentsWidget2() {
  const payload = { commentsWidget: { id: ":commentsWidgetId" } };
  const getRequestOptions = updateCommentsWidget(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/comments/widgets/{commentsWidget.id}",
    pathParams: { commentsWidgetId: "commentsWidgetId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteCommentsWidget2() {
  const payload = { commentsWidgetId: ":commentsWidgetId" };
  const getRequestOptions = deleteCommentsWidget(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/comments/widgets/{commentsWidgetId}",
    pathParams: { commentsWidgetId: "commentsWidgetId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryCommentsWidget2() {
  const payload = {};
  const getRequestOptions = queryCommentsWidget(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/comments/widgets/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  SortOrderOriginal,
  WebhookIdentityTypeOriginal,
  createCommentsWidget,
  deleteCommentsWidget,
  getCommentsWidget,
  queryCommentsWidget,
  queryCommentsWidgetResources,
  registerCommentsWidget,
  updateCommentsWidget
});
