// @wix/auto_sdk_restaurants_order-pacing-policy/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  SyncLocationsSourceOriginal: () => SyncLocationsSource,
  TypeOriginal: () => Type,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  createPolicy: () => createPolicy2,
  getPolicy: () => getPolicy2,
  updatePolicy: () => updatePolicy2
});
module.exports = __toCommonJS(meta_exports);

// src/restaurants-order-pacing-v1-policy-order-pacing-policy.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressRestaurantsPacingPoliciesOrderPacingPoliciesUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/order-pacing/v1/policies",
        destPath: "/v1/policies"
      },
      {
        srcPath: "/restaurants/v1/order-pacing-policies",
        destPath: "/v1"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-order-pacing-policies",
        destPath: "/api"
      },
      {
        srcPath: "/order-pacing/v1/policies",
        destPath: "/v1/policies"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-order-pacing-policies",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-order-pacing-policies",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_order-pacing-policy";
function createPolicy(payload) {
  function __createPolicy({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.order_pacing.v1.policy",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.pacing.policies.OrderPacingPolicies.CreatePolicy",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsPacingPoliciesOrderPacingPoliciesUrl({
        protoPath: "/v1/policies",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __createPolicy;
}
function getPolicy(payload) {
  function __getPolicy({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.order_pacing.v1.policy",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.pacing.policies.OrderPacingPolicies.GetPolicy",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsPacingPoliciesOrderPacingPoliciesUrl({
        protoPath: "/v1/policies",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getPolicy;
}
function updatePolicy(payload) {
  function __updatePolicy({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.order_pacing.v1.policy",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.pacing.policies.OrderPacingPolicies.UpdatePolicy",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsPacingPoliciesOrderPacingPoliciesUrl({
        protoPath: "/v1/policies/{policy.id}",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __updatePolicy;
}

// src/restaurants-order-pacing-v1-policy-order-pacing-policy.types.ts
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNKNOWN"] = "UNKNOWN";
  Type2["FROM_ZERO_LOCATIONS_TO_ONE_LOCATION"] = "FROM_ZERO_LOCATIONS_TO_ONE_LOCATION";
  Type2["FROM_ONE_LOCATION_TO_ZERO_LOCATIONS"] = "FROM_ONE_LOCATION_TO_ZERO_LOCATIONS";
  return Type2;
})(Type || {});
var SyncLocationsSource = /* @__PURE__ */ ((SyncLocationsSource2) => {
  SyncLocationsSource2["API"] = "API";
  SyncLocationsSource2["LOCATION_EVENT"] = "LOCATION_EVENT";
  SyncLocationsSource2["RESTAURANT_PROVISIONED_EVENT"] = "RESTAURANT_PROVISIONED_EVENT";
  return SyncLocationsSource2;
})(SyncLocationsSource || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/restaurants-order-pacing-v1-policy-order-pacing-policy.meta.ts
function createPolicy2() {
  const payload = {};
  const getRequestOptions = createPolicy(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/policies",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getPolicy2() {
  const payload = {};
  const getRequestOptions = getPolicy(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/policies",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updatePolicy2() {
  const payload = { policy: { id: ":policyId" } };
  const getRequestOptions = updatePolicy(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/policies/{policy.id}",
    pathParams: { policyId: "policyId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  SyncLocationsSourceOriginal,
  TypeOriginal,
  WebhookIdentityTypeOriginal,
  createPolicy,
  getPolicy,
  updatePolicy
});
