// @wix/ricos/dist/es/wix-ricos-common/utils/biDefaultParamsMock.js
const biLogger = {
  biLoggerClientFactoryInstance: {
    options: { useBatch: true, publishMethod: "auto" },
    nonEssentialContext: {},
    initialized: true,
    muted: false,
    loggerClientFactory: {
      _publishers: [null],
      _validators: [],
      _defaults: {
        _msid: "83fa8f6f-de43-42ee-ab3c-c58d0cc74144",
        vsi: "eb5a0bf9-7df5-4ad4-a644-bb33cd0e6082",
        rid: "1617192230.649856859644113146",
        _av: "thunderbolt-1.6045.0",
        isb: false,
        _appId: "3e1dc7e8-3444-45ba-a54b-884466c7da0e",
        _instanceId: "285e5939-b8ff-4273-aa70-24bd04fac523",
        _siteOwnerId: "6287ae85-25b0-482a-87ca-6714ed292e10",
        _viewMode: "site",
        src: 11,
        visitor_id: "9e4f431b-e160-401c-b779-a7cdb0d0f984",
        origin: "viewer",
        bsi: () => "88f4a792-e069-4613-8bce-b2fab7fdc1c9|1",
        appName: "RICOS DEMO TPA",
        projectName: "ricos-tpa"
      },
      _events: {}
    }
  },
  util: {
    log: async () => {
      return void 0;
    },
    updateDefaults: () => ""
  },
  bi: {}
};
const biLoggerFactory = () => ({
  loggerClientFactory: {
    _publishers: [],
    _validators: [],
    _defaults: {
      _msid: "83fa8f6f-de43-42ee-ab3c-c58d0cc74144",
      bsi: () => "88f4a792-e069-4613-8bce-b2fab7fdc1c9|1",
      vsi: "eb5a0bf9-7df5-4ad4-a644-bb33cd0e6082",
      rid: "1617192230.649856859644113146",
      _av: "thunderbolt-1.6045.0",
      isb: false,
      _appId: "3e1dc7e8-3444-45ba-a54b-884466c7da0e",
      _instanceId: "285e5939-b8ff-4273-aa70-24bd04fac523",
      _siteOwnerId: "6287ae85-25b0-482a-87ca-6714ed292e10",
      _viewMode: "site"
    },
    _events: {}
  }
});
const bi = {
  viewerSessionId: "eb5a0bf9-7df5-4ad4-a644-bb33cd0e6082",
  visitorId: "9e4f431b-e160-401c-b779-a7cdb0d0f984",
  requestId: "1617192230.649856859644113146",
  siteMemberId: "6287ae85-25b0-482a-87ca-6714ed292e10",
  isCached: false,
  is_rollout: 0,
  dc: "84",
  isjp: false,
  btype: "",
  pageLoadStart: 1617192231260,
  networkPageLoadStart: 1617192230328,
  pageNumber: 1,
  metaSiteId: "83fa8f6f-de43-42ee-ab3c-c58d0cc74144",
  ownerId: "6287ae85-25b0-482a-87ca-6714ed292e10",
  isMobileFriendly: true,
  viewMode: "site",
  isPreview: false,
  svSession: "3c1c0ebfef9de0c68fb275915df19a0f403b6fe4163c677e514715da95e020109d715f8cc8ac55f85778b8de9bf628df1e60994d53964e647acf431e4f798bcd04fc489cdf8552d0ec58d87b94a0414760fd2dc2af4a5ac40a3cabef17b6e0e0",
  pageId: "c1dmp",
  pageUrl: "https://barakl1.wixsite.com/ricos-tpa?petri_ovr=specs.ExcludeSiteFromSsr:true&tpaWidgetUrlOverride=e5e16f5f-936d-4545-8d6c-205ee84416f2=https://localhost:3000/editor/RICOS&tpaMobileUrlOverride=e5e16f5f-936d-4545-8d6c-205ee84416f2=https://localhost:3000/editor/RICOS&tpaSettingsUrlOverride=e5e16f5f-936d-4545-8d6c-205ee84416f2=https://localhost:3000/settings/RICOS&widgetsUrlOverride=e5e16f5f-936d-4545-8d6c-205ee84416f2=https://localhost:3200/RICOSViewerWidget.bundle.js&viewerPlatformOverrides=3e1dc7e8-3444-45ba-a54b-884466c7da0e=https://localhost:3200/viewerScript.bundle.js&overridePlatformBaseUrls=3e1dc7e8-3444-45ba-a54b-884466c7da0e=%7B%22staticsBaseUrl%22:%22https://localhost:3200/%22%7D",
  isServerSide: false,
  viewerName: "thunderbolt",
  artifactVersion: "thunderbolt-1.6045.0",
  isPopup: false,
  muteBi: false
};
export {
  bi,
  biLogger,
  biLoggerFactory
};
