// @wix/fast-gallery-core/dist/cjs/navigator/asyncUtils.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.TrackablePromise = exports.Deferred = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
// Define Deferred Class

class Deferred {
  constructor() {
    (0, _defineProperty2.default)(this, "promise", void 0);
    (0, _defineProperty2.default)(this, "resolve", void 0);
    (0, _defineProperty2.default)(this, "reject", void 0);
    (0, _defineProperty2.default)(this, "isPending", void 0);
    this.isPending = true;
    this.promise = new Promise((resolve, reject) => {
      this.resolve = value => {
        this.isPending = false;
        resolve(value);
      };
      this.reject = reason => {
        this.isPending = false;
        reject(reason);
      };
    });
  }
}

// Custom Promise wrapper to track state
exports.Deferred = Deferred;
class TrackablePromise {
  constructor(promise) {
    (0, _defineProperty2.default)(this, "promise", void 0);
    (0, _defineProperty2.default)(this, "_state", 'pending');
    this.promise = promise;
    this.promise.then(() => {
      this._state = 'fulfilled';
    }).catch(() => {
      this._state = 'rejected';
    });
  }
  get state() {
    return this._state;
  }
  then(onfulfilled, onrejected) {
    return this.promise.then(onfulfilled, onrejected);
  }
  catch(onrejected) {
    return this.promise.catch(onrejected);
  }
}
exports.TrackablePromise = TrackablePromise;
