// @wix/headless-media/dist/react/MediaGallery.js
import { jsx as _jsx, jsxs as _jsxs, Fragment as _Fragment } from "react/jsx-runtime";
/**
 * @fileoverview MediaGallery Primitive Components
 *
 * This module provides unstyled, composable components for building media galleries.
 * These components follow the Radix UI primitive pattern, offering:
 *
 * - **Unstyled**: No default styling, only functional behavior
 * - **Composable**: Support for the `asChild` pattern for flexible DOM structure
 * - **Accessible**: Built-in keyboard navigation and ARIA attributes
 * - **Flexible**: Render props pattern for maximum customization
 *
 * ## Architecture
 *
 * These components are the **primitive layer** that sits between:
 * 1. **Core components** (pure logic, no DOM)
 * 2. **Styled components** (project-specific styling)
 *
 * ## Usage
 *
 * ```tsx
 * import { MediaGallery } from '@wix/headless-media/react';
 *
 * function ProductGallery({ productMedia }) {
 *   return (
 *     <MediaGallery.Root mediaGalleryServiceConfig={{ media: productMedia }}>
 *       <MediaGallery.Viewport />
 *       <MediaGallery.Previous />
 *       <MediaGallery.Next />
 *       <MediaGallery.Indicator />
 *       <MediaGallery.Thumbnails>
 *         <MediaGallery.ThumbnailRepeater>
 *           <MediaGallery.ThumbnailItem />
 *         </MediaGallery.ThumbnailRepeater>
 *       </MediaGallery.Thumbnails>
 *     </MediaGallery.Root>
 *   );
 * }
 * ```
 *
 * @module MediaGallery
 */
import { Root as CoreRoot, Next as CoreNext, Previous as CorePrevious, Viewport as CoreViewport, Indicator as CoreIndicator, ThumbnailList as CoreThumbnailList, ThumbnailItem as CoreThumbnailItem, } from './core/MediaGallery.js';
import React, { createContext, useContext } from 'react';
import { AsChildSlot } from '@wix/headless-utils/react';
import { Slot } from '@radix-ui/react-slot';
import { WixMediaImage } from './WixMediaImage.js';
// Components that render actual DOM elements get test IDs on their rendered elements
// Components that only provide context/logic don't introduce new DOM elements
export var TestIds;
(function (TestIds) {
    TestIds["mediaGalleryRoot"] = "media-gallery-root";
    TestIds["mediaGalleryNext"] = "media-gallery-next";
    TestIds["mediaGalleryPrevious"] = "media-gallery-previous";
    TestIds["mediaGalleryViewport"] = "media-gallery-viewport";
    TestIds["mediaGalleryIndicator"] = "media-gallery-indicator";
    TestIds["mediaGalleryThumbnailItem"] = "media-gallery-thumbnail-item";
})(TestIds || (TestIds = {}));
/**
 * Root component that provides media gallery service context to its children.
 * This is a primitive wrapper around the core Root component that maintains
 * the same API while providing a foundation for composition patterns.
 *
 * @component
 * @example
 * ```tsx
 * import { MediaGallery } from '@wix/headless-media/react';
 *
 * function ProductGallery({ productMedia }) {
 *   return (
 *     <MediaGallery.Root mediaGalleryServiceConfig={{ media: productMedia }}>
 *       <MediaGallery.Viewport className="main-viewer" />
 *       <MediaGallery.Thumbnails>
 *         <MediaGallery.ThumbnailRepeater>
 *           <MediaGallery.ThumbnailItem />
 *         </MediaGallery.ThumbnailRepeater>
 *       </MediaGallery.Thumbnails>
 *     </MediaGallery.Root>
 *   );
 * }
 * ```
 */
export const Root = ({ children, mediaGalleryServiceConfig }) => {
    return (_jsx(CoreRoot, { mediaGalleryServiceConfig: mediaGalleryServiceConfig, "data-testid": TestIds.mediaGalleryRoot, children: children }));
};
/**
 * Next button component that navigates to the next media item.
 * Supports the asChild pattern for flexible composition.
 *
 * @component
 * @example
 * ```tsx
 * // Default button
 * <MediaGallery.Next />
 *
 * // Custom button with asChild
 * <MediaGallery.Next asChild>
 *   <button className="custom-next-btn">
 *     <ChevronRightIcon />
 *   </button>
 * </MediaGallery.Next>
 *
 * // With custom content
 * <MediaGallery.Next className="nav-button">
 *   Next Image →
 * </MediaGallery.Next>
 * ```
 */
export const Next = React.forwardRef(({ children, ...props }, ref) => {
    const Comp = props.asChild ? Slot : 'button';
    return (_jsx(CoreNext, { children: ({ next, canGoNext }) => (_jsx(Comp, { ref: ref, onClick: next, disabled: !canGoNext, "data-testid": TestIds.mediaGalleryNext, ...props, children: children })) }));
});
/**
 * Previous button component that navigates to the previous media item.
 * Supports the asChild pattern for flexible composition.
 *
 * @component
 * @example
 * ```tsx
 * // Default button
 * <MediaGallery.Previous />
 *
 * // Custom button with asChild
 * <MediaGallery.Previous asChild>
 *   <button className="custom-prev-btn">
 *     <ChevronLeftIcon />
 *   </button>
 * </MediaGallery.Previous>
 *
 * // With custom content
 * <MediaGallery.Previous className="nav-button">
 *   ← Previous Image
 * </MediaGallery.Previous>
 * ```
 */
export const Previous = React.forwardRef(({ children, ...props }, ref) => {
    const Comp = props.asChild ? Slot : 'button';
    return (_jsx(CorePrevious, { children: ({ previous, canGoPrevious }) => (_jsx(Comp, { ref: ref, onClick: previous, disabled: !canGoPrevious, "data-testid": TestIds.mediaGalleryPrevious, ...props, children: children })) }));
});
/**
 * Viewport component that displays the currently selected media item.
 * Automatically renders the active media using WixMediaImage for optimization.
 * Supports the asChild pattern and custom empty states.
 *
 * @component
 * @example
 * ```tsx
 * // Default viewport
 * <MediaGallery.Viewport />
 *
 * // With custom styling
 * <MediaGallery.Viewport className="rounded-lg border" />
 *
 * // With custom empty state
 * <MediaGallery.Viewport
 *   emptyState={<div>No images available</div>}
 * />
 *
 * // Using asChild for custom wrapper
 * <MediaGallery.Viewport asChild>
 *   <div className="custom-viewport-wrapper">
 *     Content will be rendered here
 *   </div>
 * </MediaGallery.Viewport>
 *
 * // With completely custom children
 * <MediaGallery.Viewport>
 *   <CustomImageComponent />
 * </MediaGallery.Viewport>
 * ```
 */
export const Viewport = React.forwardRef(({ children, asChild, emptyState, ...otherProps }, ref) => {
    return (_jsx(CoreViewport, { children: ({ src, alt }) => {
            const defaultContent = children ??
                (src ? (_jsx(WixMediaImage, { media: { image: src }, alt: alt })) : ((emptyState ?? _jsx("div", { children: "No image" }))));
            return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, "data-src": src, "data-alt": alt, "data-testid": TestIds.mediaGalleryViewport, customElement: children, customElementProps: { src, alt }, content: defaultContent, ...otherProps, children: _jsx("div", { children: React.isValidElement(defaultContent) ? defaultContent : null }) }));
        } }));
});
/**
 * Indicator component that displays the current media position (e.g., "1 / 5").
 * Automatically tracks the current and total media count.
 * Supports the asChild pattern for flexible styling.
 *
 * @component
 * @example
 * ```tsx
 * // Default indicator
 * <MediaGallery.Indicator />
 *
 * // With custom styling
 * <MediaGallery.Indicator className="text-sm opacity-80" />
 *
 * // Using asChild for custom wrapper
 * <MediaGallery.Indicator asChild>
 *   <span className="badge">
 *    "1 / 5" will be rendered here
 *   </span>
 * </MediaGallery.Indicator>
 *
 * // With completely custom children
 * <MediaGallery.Indicator>
 *   <CustomIndicatorComponent />
 * </MediaGallery.Indicator>
 * ```
 */
export const Indicator = React.forwardRef(({ children, asChild, ...props }, ref) => {
    return (_jsx(CoreIndicator, { children: ({ current, total }) => {
            const Comp = asChild ? Slot : 'div';
            return (_jsx(Comp, { ref: ref, "data-current": current, "data-total": total, "data-testid": TestIds.mediaGalleryIndicator, ...props, children: children ?? (_jsxs("div", { children: [current, " / ", total] })) }));
        } }));
});
const ThumbnailsContext = createContext(null);
const ThumbnailItemContext = createContext(null);
/**
 * Thumbnails container component that provides thumbnail context to its children.
 * Only renders when there are multiple media items to display.
 *
 * @component
 * @example
 * ```tsx
 * <MediaGallery.Thumbnails>
 *   <MediaGallery.ThumbnailRepeater>
 *     <MediaGallery.ThumbnailItem />
 *   </MediaGallery.ThumbnailRepeater>
 * </MediaGallery.Thumbnails>
 * ```
 */
export const Thumbnails = ({ children }) => (_jsx(CoreThumbnailList, { children: ({ items }) => (_jsx(ThumbnailsContext.Provider, { value: { items: items }, children: children })) }));
/**
 * ThumbnailRepeater component that renders a template for each media item.
 * Provides index context to each thumbnail item. Only renders when there are
 * multiple media items available.
 *
 * @component
 * @example
 * ```tsx
 * <MediaGallery.ThumbnailRepeater>
 *   <MediaGallery.ThumbnailItem className="thumbnail" />
 * </MediaGallery.ThumbnailRepeater>
 * ```
 */
export const ThumbnailRepeater = ({ children }) => {
    const ctx = useContext(ThumbnailsContext);
    if (!ctx || !ctx.items || ctx.items.length <= 1)
        return null;
    return (_jsx(_Fragment, { children: ctx.items.map((_, i) => (_jsx(ThumbnailItemContext.Provider, { value: { index: i }, children: children }, i))) }));
};
/**
 * ThumbnailItem component that renders an individual thumbnail for media selection.
 * Automatically handles selection state, click events, and renders the thumbnail image.
 * Must be used within a ThumbnailRepeater context.
 *
 * @component
 * @example
 * ```tsx
 * // Default thumbnail item
 * <MediaGallery.ThumbnailItem />
 *
 * // With custom styling
 * <MediaGallery.ThumbnailItem className="border rounded-lg hover:shadow-lg" />
 *
 * // With custom empty state
 * <MediaGallery.ThumbnailItem
 *   emptyState={<div className="placeholder">No preview</div>}
 * />
 *
 * // Using asChild for custom wrapper
 * <MediaGallery.ThumbnailItem asChild>
 *   <button className="custom-thumbnail-button">
 *    Thumbnail content will be rendered here
 *   </button>
 * </MediaGallery.ThumbnailItem>
 *
 * // With completely custom children
 * <MediaGallery.ThumbnailItem>
 *   <CustomThumbnailComponent />
 * </MediaGallery.ThumbnailItem>
 * ```
 */
export const ThumbnailItem = React.forwardRef(({ children, asChild, emptyState, ...otherProps }, ref) => {
    const itemCtx = useContext(ThumbnailItemContext);
    if (!itemCtx)
        return null;
    const { index } = itemCtx;
    return (_jsx(CoreThumbnailItem, { index: index, children: ({ src, isActive, select, alt }) => {
            const defaultContent = children ??
                (src ? (_jsx(WixMediaImage, { media: { image: src }, alt: alt })) : ((emptyState ?? _jsx("div", { children: "No image" }))));
            return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, onClick: select, "data-active": isActive, "data-src": src, "data-alt": alt, "data-index": index, "data-available": true, "data-testid": TestIds.mediaGalleryThumbnailItem, customElement: children, customElementProps: { src, alt, isActive, select, index }, content: defaultContent, ...otherProps, children: _jsx("div", { children: React.isValidElement(defaultContent) ? defaultContent : null }) }));
        } }));
});
