// @wix/ricos/dist/es/plugin-ai/ai-image-modal/ai-image-modal.js
import { Box, Heading } from "@wix/design-system";
import { SparklesFilled } from "@wix/wix-ui-icons-common";
import React from "react";
import { EditorContext, PluginsEventsContext, RicosContext } from "./../../ricos-context/index.js";
import {
  Panel,
  PanelHeader,
  useClickOutside,
  usePanelContainerElement
} from "./../../ui-components/index.js";
import { AI_IMAGE_MODAL_ROOT } from "../data-hooks.js";
import { isOfType } from "../ts-utils.js";
import { AI_IMAGE_MODAL_ID, AI_TYPE } from "../types.js";
import { getSelectionAsText } from "../utils/get-selection-as-text.js";
import { Feed } from "./feed.js";
import { Footer } from "./footer.js";
import { useAiImageModalApi } from "./use-ai-image-modal-api.js";
const AiImageModal = ({
  closeModal,
  openDiscardModal,
  config,
  modalSessionId,
  zIndex,
  convertSelectionToPrompt,
  insertActions
}) => {
  const { t } = React.useContext(RicosContext);
  const { adapter } = React.useContext(EditorContext);
  const footerRef = React.useRef(null);
  const selection = React.useMemo(() => getSelectionAsText(adapter.tiptapEditor), []);
  const containerEl = usePanelContainerElement();
  const resolvedZIndex = typeof zIndex === "number" ? zIndex : 5500;
  const pluginsEvents = React.useContext(PluginsEventsContext);
  const publishModalAction = React.useMemo(() => {
    let counter = 0;
    return (actionName, value, imageStyle) => {
      counter++;
      pluginsEvents.publishModalAction({
        pluginId: AI_TYPE,
        modalId: AI_IMAGE_MODAL_ID,
        modalSessionId,
        actionName,
        value,
        actionNumber: counter,
        initialSelection: selection,
        style: imageStyle
      });
    };
  }, [pluginsEvents.publishModalAction, modalSessionId]);
  const {
    state,
    goBack,
    retryPrompt,
    setPromptValue,
    setImageStyleValue,
    tryConvertingSelectionToPrompt,
    submitPrompt,
    cancelPrompt,
    insertImage,
    isConvertedPromptHelperVisible,
    closeOptimisedPromptHelper,
    tryCloseModal,
    isDiscardModalOpen
  } = useAiImageModalApi({
    ...config,
    selection,
    closeModal,
    openDiscardModal,
    focusInputArea: () => footerRef.current?.focus(),
    modalRef: containerEl,
    convertSelectionToPrompt,
    publishModalAction
  });
  const closeOnClickOutside = React.useCallback(() => {
    tryCloseModal();
    publishModalAction("close_on_click_outside");
  }, [tryCloseModal]);
  useClickOutside({
    containerRef: { current: containerEl },
    onClickOutside: closeOnClickOutside,
    disable: isDiscardModalOpen
  });
  React.useEffect(() => {
    if (isOfType(["INITIAL"], state)) {
      setTimeout(() => {
        footerRef.current?.focus();
      }, 0);
    }
  }, [state.type]);
  React.useEffect(() => {
    if (convertSelectionToPrompt && state.type === "INITIALIZING") {
      tryConvertingSelectionToPrompt(selection);
    }
  }, []);
  return /* @__PURE__ */ React.createElement(
    Panel,
    {
      closeButtonProps: { onClick: () => closeModal() },
      width: 540,
      dataHook: AI_IMAGE_MODAL_ROOT
    },
    /* @__PURE__ */ React.createElement(
      PanelHeader,
      {
        title: /* @__PURE__ */ React.createElement(Box, { gap: "SP1" }, /* @__PURE__ */ React.createElement(Box, { color: "B10" }, /* @__PURE__ */ React.createElement(SparklesFilled, null)), /* @__PURE__ */ React.createElement(Heading, { size: "medium" }, t("AIPlugin_Modal_ImageGeneratorTitle")))
      }
    ),
    /* @__PURE__ */ React.createElement(Feed, { state, insertImage, insertActions }),
    /* @__PURE__ */ React.createElement(
      Footer,
      {
        ref: footerRef,
        state,
        zIndex: resolvedZIndex,
        modalRef: containerEl,
        setPromptValue,
        cancelPrompt,
        setImageStyleValue,
        retryPrompt,
        goBack,
        submitPrompt,
        isConvertedPromptHelperVisible,
        closeOptimisedPromptHelper,
        publishModalAction
      }
    )
  );
};
export {
  AiImageModal
};
