// @wix/auto-patterns/dist/cjs/hooks/useBulkActionToolbar.js
"use strict";

exports.__esModule = true;
exports.useBulkActionToolbar = void 0;
var _react = _interopRequireWildcard(require("react"));
var _useActionsSDK = require("./useActionsSDK");
var _patterns = require("@wix/patterns");
var _actions = require("../utils/actions");
var _jsxFileName = "/home/builduser/work/73d19ce3378ce4dd/packages/auto-patterns/dist/cjs/hooks/useBulkActionToolbar.tsx";
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
const useBulkActionToolbar = ({
  config,
  collectionId
}) => {
  const sdk = (0, _useActionsSDK.useActionsSDK)({
    collectionId
  });
  const bulkActionToolbar = (0, _react.useCallback)(param => {
    const buildAction = actionConfig => {
      let actionToResolve = null;
      const baseParams = {
        actionParams: param,
        sdk
      };
      switch (actionConfig.type) {
        case 'bulkDelete':
          actionToResolve = (0, _actions.bulkDeleteAction)({
            action: actionConfig,
            ...baseParams
          });
          break;
        case 'custom':
          actionToResolve = (0, _actions.customAction)({
            action: actionConfig,
            ...baseParams
          });
          break;
        default:
          return null;
      }
      if (!actionToResolve) {
        return null;
      }
      const resolvedAction = (0, _actions.resolveAction)(actionConfig, actionToResolve);
      const {
        icon,
        ...resolvedActionParams
      } = resolvedAction;
      return {
        prefixIcon: /*#__PURE__*/_react.default.isValidElement(icon) ? icon : undefined,
        dataHook: actionConfig.id,
        ...resolvedActionParams
      };
    };
    const buildPrimaryActions = () => {
      if (!(config != null && config.primaryActions)) {
        return [];
      }
      return config.primaryActions.map((primaryItem, index) => {
        if (primaryItem.type === 'action') {
          return buildAction(primaryItem.action.item);
        }
        if (primaryItem.type === 'menu') {
          const actionGroups = (0, _actions.groupActions)(primaryItem.menu.items);
          const subItems = actionGroups.map(group => group.map(buildAction).filter(action => Boolean(action))).filter(group => group.length > 0);
          if (subItems.length === 0) {
            return null;
          }
          return {
            label: primaryItem.menu.label,
            subItems,
            dataHook: `bulk-multi-action-${index}`
          };
        }
        return null;
      }).filter(action => Boolean(action));
    };
    const primaryActionItems = buildPrimaryActions();
    const secondaryActionItems = config != null && config.secondaryActions ? (0, _actions.groupActions)(config.secondaryActions).map(group => group.map(buildAction).filter(action => Boolean(action))).filter(group => group.length > 0) : [];
    return /*#__PURE__*/_react.default.createElement(_patterns.MultiBulkActionToolbar, {
      primaryActionItems: primaryActionItems,
      secondaryActionItems: secondaryActionItems,
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 134,
        columnNumber: 9
      }
    });
  }, [config, sdk]);
  if (!config) {
    return undefined;
  }
  const hasPrimaryActions = config.primaryActions && config.primaryActions.length > 0;
  const hasSecondaryActions = config.secondaryActions && config.secondaryActions.length > 0;
  if (!hasPrimaryActions && !hasSecondaryActions) {
    return undefined;
  }
  return bulkActionToolbar;
};
exports.useBulkActionToolbar = useBulkActionToolbar;
