// @wix/auto-patterns/dist/cjs/hooks/useBaseGridFeatures.js
"use strict";

exports.__esModule = true;
exports.useBaseGridFeatures = void 0;
var _react = require("react");
var _getImageUrl = require("../utils/media/getImageUrl");
var _providers = require("../providers");
const useBaseGridFeatures = configuration => {
  const {
    item
  } = configuration;
  const schema = (0, _providers.useSchema)();
  const {
    titleFieldId,
    subtitleFieldId,
    imageFieldId,
    imagePlacement,
    cardContentMode: preset
  } = item;
  const visibleFieldIds = (0, _react.useMemo)(() => [titleFieldId, subtitleFieldId, imageFieldId].filter(Boolean), [titleFieldId, subtitleFieldId, imageFieldId]);
  const renderItem = (0, _react.useMemo)(() => gridItem => ({
    id: gridItem[schema.idField],
    title: `${gridItem[titleFieldId]}`,
    ...(subtitleFieldId ? {
      subtitle: `${gridItem[subtitleFieldId] ?? ''}`
    } : {}),
    ...(imageFieldId ? {
      backgroundImageUrl: (0, _getImageUrl.getImageUrl)(gridItem[imageFieldId])
    } : {})
  }), [titleFieldId, subtitleFieldId, imageFieldId, schema.idField]);
  return {
    imagePlacement,
    preset,
    renderItem,
    visibleFieldIds
  };
};
exports.useBaseGridFeatures = useBaseGridFeatures;
