// @wix/auto-patterns/dist/cjs/hooks/useBaseTableFeatures.js
"use strict";

exports.__esModule = true;
exports.useBaseTableFeatures = void 0;
var _react = _interopRequireWildcard(require("react"));
var _useColumns = require("./useColumns");
var _patterns = require("@wix/patterns");
var _useCollectionPageOnRowClickActions = require("./useCollectionPageOnRowClickActions");
var _useNavigationUtils = require("./useNavigationUtils");
var _jsxFileName = "/home/builduser/work/73d19ce3378ce4dd/packages/auto-patterns/dist/cjs/hooks/useBaseTableFeatures.tsx";
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
const useBaseTableFeatures = configuration => {
  var _configuration$custom;
  const {
    navigateToEntityPage
  } = (0, _useNavigationUtils.useNavigation)();
  const columns = (0, _useColumns.useColumns)(configuration.columns);
  const visibleFieldIds = (0, _react.useMemo)(() => configuration.columns.map(column => column.id), [configuration]);
  const entityPageId = configuration.entityPageId;
  const onRowClickOverride = (0, _useCollectionPageOnRowClickActions.useCollectionPageOnRowClickActions)({
    collectionPageOnRowClickActions: configuration.onRowClick,
    collectionId: configuration.collection.collectionId
  });
  const onRowClick = configuration.onRowClick ? onRowClickOverride : item => navigateToEntityPage({
    item,
    entityPageId
  });
  const stickyColumns = configuration.stickyColumns;
  const stickySelectionColumn = !!stickyColumns; // Sticky selection column is true if stickyColumns is positive number

  return {
    columns,
    visibleFieldIds,
    onRowClick,
    customColumns: (_configuration$custom = configuration.customColumns) != null && _configuration$custom.enabled ? /*#__PURE__*/_react.default.createElement(_patterns.CustomColumns, {
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 39,
        columnNumber: 7
      }
    }) : undefined,
    stickyColumns,
    stickySelectionColumn
  };
};
exports.useBaseTableFeatures = useBaseTableFeatures;
