// @wix/auto-patterns/dist/cjs/hooks/useColumns.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.useColumns = void 0;
var _react = _interopRequireDefault(require("react"));
var _providers = require("../providers");
var _lodash = require("lodash");
var _designSystem = require("@wix/design-system");
var _getImageUrl = require("../utils/media/getImageUrl");
var _jsxFileName = "/home/builduser/work/73d19ce3378ce4dd/packages/auto-patterns/dist/cjs/hooks/useColumns.tsx";
const useColumns = columns => {
  const {
    fields
  } = (0, _providers.useSchema)();
  const {
    getSchema
  } = (0, _providers.useSchemaRegistry)();
  const overrides = (0, _providers.usePatternsWizardOverridesContext)();
  const existingColumns = columns.filter(column => {
    var _overrides$columns;
    return !!fields[column.id] || (overrides == null || (_overrides$columns = overrides.columns) == null ? void 0 : _overrides$columns[column.id]);
  });
  const createRenderer = (fieldDefinition, itemValue) => {
    if (itemValue == null) {
      return '';
    }
    switch (fieldDefinition.type) {
      case 'BOOLEAN':
        return itemValue ? '✔' : '✖';
      case 'DATE':
        if (itemValue instanceof Date) {
          return itemValue.toLocaleDateString();
        }
        if (typeof itemValue === 'string') {
          const date = new Date(itemValue);
          return date.toLocaleDateString();
        }
        return itemValue;
      case 'DATETIME':
        if (itemValue instanceof Date) {
          return itemValue.toLocaleString();
        }
        if (typeof itemValue === 'string') {
          const date = new Date(itemValue);
          return date.toLocaleString();
        }
        return itemValue;
      case 'ARRAY':
        // We assume that the array contains strings
        // If the array contains objects, we need to handle it differently
        return itemValue.join(', ');
      case 'REFERENCE':
        if (fieldDefinition.referenceMetadata && getSchema) {
          const {
            referencedCollectionId
          } = fieldDefinition.referenceMetadata;
          const refSchema = getSchema(referencedCollectionId);
          if (refSchema) {
            let nameValue;
            let imageValue;
            const displayField = refSchema.displayField;
            if (typeof itemValue === 'object' && itemValue !== null) {
              nameValue = itemValue[displayField] || '';
              imageValue = refSchema.imageField && itemValue[refSchema.imageField];
            }
            const imageUrl = imageValue && (0, _getImageUrl.getImageUrl)(imageValue);
            return /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
              gap: "SP2",
              verticalAlign: "middle",
              __self: void 0,
              __source: {
                fileName: _jsxFileName,
                lineNumber: 69,
                columnNumber: 15
              }
            }, imageUrl && /*#__PURE__*/_react.default.createElement(_designSystem.Avatar, {
              size: "size24",
              name: nameValue,
              imgProps: {
                src: imageUrl
              },
              __self: void 0,
              __source: {
                fileName: _jsxFileName,
                lineNumber: 71,
                columnNumber: 19
              }
            }), /*#__PURE__*/_react.default.createElement("div", {
              __self: void 0,
              __source: {
                fileName: _jsxFileName,
                lineNumber: 77,
                columnNumber: 17
              }
            }, nameValue));
          }
        }
        return itemValue;
      case 'IMAGE':
        const imageUrl = (0, _getImageUrl.getImageUrl)(itemValue);
        return /*#__PURE__*/_react.default.createElement(_designSystem.Image, {
          src: imageUrl,
          width: "48px",
          dataHook: "auto-patterns-image",
          __self: void 0,
          __source: {
            fileName: _jsxFileName,
            lineNumber: 88,
            columnNumber: 11
          }
        });
      default:
        return itemValue;
    }
  };
  return existingColumns.map(column => {
    var _overrides$columns2, _fieldDefinition$capa;
    const fieldDefinition = fields[column.id];
    if (!fieldDefinition && !(overrides != null && (_overrides$columns2 = overrides.columns) != null && _overrides$columns2[column.id])) {
      return null;
    }
    const sortable = column.sortable && (fieldDefinition == null || (_fieldDefinition$capa = fieldDefinition.capabilities) == null ? void 0 : _fieldDefinition$capa.sortable);
    const baseColumnConfig = {
      id: column.id,
      title: column.name || (fieldDefinition == null ? void 0 : fieldDefinition.displayName),
      width: column.width,
      ...createSortConfiguration(column, sortable),
      ...createCustomColumnConfiguration(column),
      render: row => {
        var _overrides$columns3;
        const itemValue = (0, _lodash.get)(row, column.id);
        if (overrides != null && (_overrides$columns3 = overrides.columns) != null && _overrides$columns3[(0, _lodash.camelCase)(column.id)]) {
          var _overrides$columns4, _overrides$columns4$c;
          return overrides == null || (_overrides$columns4 = overrides.columns) == null || (_overrides$columns4$c = _overrides$columns4[(0, _lodash.camelCase)(column.id)]) == null ? void 0 : _overrides$columns4$c.call(_overrides$columns4, {
            value: itemValue,
            row
          });
        }
        if (fieldDefinition) {
          return createRenderer(fieldDefinition, itemValue);
        }
        return null; // Shouldn't really happen
      }
    };
    return baseColumnConfig;
  }).filter(Boolean);
};
exports.useColumns = useColumns;
const createSortConfiguration = (column, sortable) => {
  if (!sortable) {
    return {};
  }
  const sortableColumnConfig = {
    sortable,
    defaultSortOrder: column.defaultSortOrder
  };

  // default sort mode is ['asc', 'desc', undefined]
  // so if sort mode is 'desc' -> we want to override the default
  // and set it to ['desc', 'asc', undefined]
  if (column.sortMode && column.sortMode === 'desc') {
    return {
      ...sortableColumnConfig,
      sortMode: ['desc', 'asc', undefined]
    };
  }
  return {
    ...sortableColumnConfig
  };
};
const createCustomColumnConfiguration = column => {
  const reorderDisabled = !!column.reorderDisabled;
  if (column.hiddenFromCustomColumnsSelection) {
    return {
      hiddenFromCustomColumnsSelection: true
    };
  }
  if (column.hideable === false) {
    return {
      hideable: false,
      reorderDisabled
    };
  }
  return {
    defaultHidden: !!column.defaultHidden,
    reorderDisabled
  };
};
