// @wix/auto-patterns/dist/cjs/hooks/useCommonCollectionFeatures.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.useCommonCollectionFeatures = void 0;
var _react = _interopRequireDefault(require("react"));
var _useEmptyStates = require("./useEmptyStates");
var _patterns = require("@wix/patterns");
var _designSystem = require("@wix/design-system");
var _useActionCell = require("./useActionCell");
var _useBulkActionToolbar = require("./useBulkActionToolbar");
var _jsxFileName = "/home/builduser/work/73d19ce3378ce4dd/packages/auto-patterns/dist/cjs/hooks/useCommonCollectionFeatures.tsx";
const useCommonCollectionFeatures = ({
  actionCell: actionCellConfig,
  bulkActionToolbar: bulkActionToolbarConfig,
  toolbarTitle,
  search,
  emptyState: emptyStateConfig,
  collection
}) => {
  var _toolbarTitle$subtitl, _toolbarTitle$subtitl2, _toolbarTitle$subtitl3, _toolbarTitle$subtitl4, _toolbarTitle$subtitl5, _toolbarTitle$subtitl6, _toolbarTitle$subtitl7;
  const {
    emptyState
  } = (0, _useEmptyStates.useEmptyStates)(emptyStateConfig);
  const actionCell = (0, _useActionCell.useActionCell)({
    config: actionCellConfig,
    collectionId: collection.collectionId
  });
  const bulkActionToolbar = (0, _useBulkActionToolbar.useBulkActionToolbar)({
    config: bulkActionToolbarConfig,
    collectionId: collection.collectionId
  });
  const title = toolbarTitle && /*#__PURE__*/_react.default.createElement(_patterns.ToolbarTitle, {
    title: toolbarTitle.title,
    subtitle: {
      text: ((_toolbarTitle$subtitl = toolbarTitle.subtitle) == null ? void 0 : _toolbarTitle$subtitl.text) ?? '',
      learnMore: (_toolbarTitle$subtitl2 = toolbarTitle.subtitle) == null ? void 0 : _toolbarTitle$subtitl2.learnMore,
      info: ((_toolbarTitle$subtitl3 = toolbarTitle.subtitle) == null ? void 0 : _toolbarTitle$subtitl3.info) && {
        content: /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
          direction: "vertical",
          gap: "6px",
          __self: void 0,
          __source: {
            fileName: _jsxFileName,
            lineNumber: 51,
            columnNumber: 13
          }
        }, /*#__PURE__*/_react.default.createElement(_designSystem.Text, {
          dataHook: "toolbar-title-subtitle-info-content",
          light: true,
          size: "small",
          __self: void 0,
          __source: {
            fileName: _jsxFileName,
            lineNumber: 52,
            columnNumber: 15
          }
        }, (_toolbarTitle$subtitl4 = toolbarTitle.subtitle) == null || (_toolbarTitle$subtitl4 = _toolbarTitle$subtitl4.info) == null ? void 0 : _toolbarTitle$subtitl4.content.text), ((_toolbarTitle$subtitl5 = toolbarTitle.subtitle) == null || (_toolbarTitle$subtitl5 = _toolbarTitle$subtitl5.info) == null ? void 0 : _toolbarTitle$subtitl5.content.link) && /*#__PURE__*/_react.default.createElement(_designSystem.TextButton, {
          dataHook: "toolbar-title-subtitle-info-link",
          size: "small",
          underline: "always",
          skin: "light",
          as: "a",
          rel: "noopener",
          target: "_blank",
          href: (_toolbarTitle$subtitl6 = toolbarTitle.subtitle) == null || (_toolbarTitle$subtitl6 = _toolbarTitle$subtitl6.info) == null ? void 0 : _toolbarTitle$subtitl6.content.link.url,
          __self: void 0,
          __source: {
            fileName: _jsxFileName,
            lineNumber: 60,
            columnNumber: 17
          }
        }, (_toolbarTitle$subtitl7 = toolbarTitle.subtitle) == null || (_toolbarTitle$subtitl7 = _toolbarTitle$subtitl7.info) == null ? void 0 : _toolbarTitle$subtitl7.content.link.label))
      }
    },
    itemsLimit: toolbarTitle.itemsLimit,
    showTotal: toolbarTitle.showTotal,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 44,
      columnNumber: 5
    }
  });
  return {
    search: (search == null ? void 0 : search.shown) !== false,
    title,
    emptyState,
    actionCell,
    bulkActionToolbar
  };
};
exports.useCommonCollectionFeatures = useCommonCollectionFeatures;
