// @wix/auto-patterns/dist/cjs/hooks/useCollectionPageActions.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.useCollectionPageActions = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireWildcard(require("react"));
var _useActionsSDK = require("./useActionsSDK");
var _actions = require("../utils/actions");
var _patterns = require("@wix/patterns");
var _jsxFileName = "/home/builduser/work/73d19ce3378ce4dd/packages/auto-patterns/dist/cjs/hooks/useCollectionPageActions.tsx";
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
const useCollectionPageActions = ({
  collectionPageActions
}) => {
  const sdk = (0, _useActionsSDK.useActionsSDK)({
    collectionId: ''
  });
  const buildCollectionPageActions = (0, _react.useCallback)(() => {
    if (!collectionPageActions) {
      return;
    }
    const buildAction = actionConfig => {
      let actionToResolve = null;
      const baseParams = {
        actionParams: {
          collectionId: actionConfig.collection.collectionId
        },
        sdk
      };
      switch (actionConfig.type) {
        case 'create':
          actionToResolve = (0, _actions.createAction)({
            action: actionConfig,
            ...baseParams
          });
          break;
        case 'custom':
          actionToResolve = (0, _actions.customAction)({
            action: actionConfig,
            ...baseParams
          });
          break;
        default:
          return null;
      }
      if (!actionToResolve) {
        return null;
      }
      const resolvedAction = (0, _actions.resolveAction)(actionConfig, actionToResolve);
      const {
        tooltip,
        icon,
        ...resolvedActionProps
      } = resolvedAction;
      return {
        ...resolvedActionProps,
        prefixIcon: icon,
        disabledDescription: resolvedAction.tooltip
      };
    };
    const handleActionsType = actions => {
      if (actions.type === 'action') {
        return buildAction(actions.action.item);
      }
      if (actions.type === 'menu') {
        const actionGroups = (0, _actions.groupActions)(actions.menu.items);
        const subItems = actionGroups.map(group => group.map(buildAction).filter(action => action !== null)).filter(group => group.length > 0);
        if (!subItems.length) {
          return null;
        }
        return {
          label: actions.menu.label,
          subItems
        };
      }
      return null;
    };
    const primaryActions = (collectionPageActions == null ? void 0 : collectionPageActions.primaryActions) && handleActionsType(collectionPageActions.primaryActions);
    const secondaryActions = (collectionPageActions == null ? void 0 : collectionPageActions.secondaryActions) && handleActionsType(collectionPageActions.secondaryActions);
    if (!primaryActions && !secondaryActions) {
      return;
    }
    return {
      primaryActions: primaryActions ? /*#__PURE__*/_react.default.createElement(_patterns.PrimaryActions, (0, _extends2.default)({}, primaryActions, {
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 120,
          columnNumber: 9
        }
      })) : undefined,
      secondaryActions: secondaryActions ? /*#__PURE__*/_react.default.createElement(_patterns.SecondaryActions, (0, _extends2.default)({}, secondaryActions, {
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 123,
          columnNumber: 9
        }
      })) : undefined
    };
  }, [collectionPageActions, sdk]);
  const actions = (0, _react.useMemo)(() => {
    return {
      primaryActions: undefined,
      secondaryActions: undefined
    };
  }, []);
  if (!(collectionPageActions != null && collectionPageActions.primaryActions) && !(collectionPageActions != null && collectionPageActions.secondaryActions)) {
    return {
      ...actions
    };
  }
  return buildCollectionPageActions() ?? {
    ...actions
  };
};
exports.useCollectionPageActions = useCollectionPageActions;
