// @wix/auto-patterns/dist/cjs/hooks/useEntityPageMoreActions.js
"use strict";

exports.__esModule = true;
exports.useEntityPageMoreActions = useEntityPageMoreActions;
var _actions = require("../utils/actions");
var _react = require("react");
function useEntityPageMoreActions(moreActions, entity, sdk, form) {
  const buildEntityPageActions = (0, _react.useCallback)(() => {
    if (!moreActions) {
      return;
    }
    const buildAction = actionConfig => {
      let actionToResolve = null;
      const baseParams = {
        actionParams: {
          entity,
          form
        },
        sdk
      };
      switch (actionConfig.type) {
        case 'custom':
          actionToResolve = (0, _actions.customAction)({
            action: actionConfig,
            ...baseParams
          });
          break;
        default:
          return null;
      }
      if (!actionToResolve) {
        return null;
      }
      const resolved = (0, _actions.resolveAction)(actionConfig, actionToResolve);
      return {
        onClick: resolved.onClick,
        text: resolved.label,
        prefixIcon: resolved.icon,
        disabled: resolved.disabled
      };
    };
    const actionGroups = (0, _actions.groupActions)(moreActions);
    return actionGroups.map(group => group.map(buildAction).filter(action => Boolean(action))).filter(group => group.length > 0);
  }, [moreActions, sdk, entity, form]);
  return buildEntityPageActions() ?? [];
}
