// @wix/auto-patterns/dist/cjs/hooks/useFilters.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.useFilters = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireDefault(require("react"));
var _patterns = require("@wix/patterns");
var _SchemaContext = require("../providers/SchemaContext");
var _providers = require("../providers");
var _filters = require("../components/filters");
var _filterCreators = require("../utils/filterCreators");
var _jsxFileName = "/home/builduser/work/73d19ce3378ce4dd/packages/auto-patterns/dist/cjs/hooks/useFilters.tsx";
const useFilters = configuration => {
  const schema = (0, _SchemaContext.useSchema)();
  const schemaRegistry = (0, _providers.useSchemaRegistry)();
  const filters = _react.default.useMemo(() => {
    const currFilters = [];
    configuration == null || configuration.items.forEach(filter => {
      const {
        id,
        fieldId
      } = filter;
      const fieldDefinition = schema.fields[fieldId];
      if (!fieldDefinition) {
        return;
      }
      const label = filter.displayName || fieldDefinition.displayName;
      const {
        openByDefault,
        tagLabel,
        sectionTitle
      } = filter;
      const baseParams = {
        id,
        fieldId,
        commonFilterProps: {
          sectionTitle,
          initiallyOpen: openByDefault,
          accordionItemProps: {
            label
          },
          toolbarItemProps: {
            label
          },
          toolbarTagProps: {
            label: tagLabel
          }
        }
      };
      switch (fieldDefinition.type) {
        case 'DATE':
        case 'DATETIME':
          currFilters.push((0, _filterCreators.createDateTimeFilter)({
            ...baseParams,
            config: filter.dateConfig
          }));
          break;
        case 'BOOLEAN':
          currFilters.push((0, _filterCreators.createBooleanFilter)({
            ...baseParams,
            config: filter.booleanConfig
          }));
          break;
        case 'NUMBER':
          currFilters.push((0, _filterCreators.createNumberFilter)({
            ...baseParams,
            config: filter.numberConfig
          }));
          break;
        case 'REFERENCE':
          {
            const {
              referencedCollectionId
            } = fieldDefinition.referenceMetadata;
            const referencedSchema = schemaRegistry.getSchema(referencedCollectionId);
            if (referencedSchema) {
              currFilters.push((0, _filterCreators.createReferenceFilter)({
                ...baseParams,
                config: filter.dynamicOptionsConfig,
                additionalParams: {
                  referencedSchema
                }
              }));
            }
            break;
          }
        default:
          if (filter != null && filter.enumConfig) {
            currFilters.push((0, _filterCreators.createEnumFilter)({
              ...baseParams,
              config: filter.enumConfig
            }));
          }
          break;
      }
    });
    return currFilters;
  }, [configuration == null ? void 0 : configuration.items, schema.fields, schemaRegistry]);
  const filterFieldMapping = _react.default.useMemo(() => {
    return configuration == null ? void 0 : configuration.items.reduce((acc, filter) => {
      acc[filter.id] = filter;
      return acc;
    }, {});
  }, [configuration == null ? void 0 : configuration.items]);
  const collectionToolbarFiltersProps = _react.default.useMemo(() => ({
    inline: (configuration == null ? void 0 : configuration.maxInlineFilters) || 0,
    panelTitle: configuration == null ? void 0 : configuration.panelTitle
  }), [configuration]);
  const filtersData = filters.reduce((acc, filter) => {
    acc.filters[filter.key] = filter.filter;
    acc.components.push(filter.collectionData ? /*#__PURE__*/_react.default.createElement(_filters.StaticCollectionFilter, (0, _extends2.default)({
      filter: filter.filter,
      key: filter.key,
      component: filter.component,
      collectionData: filter.collectionData
    }, filter.props, {
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 155,
        columnNumber: 11
      }
    })) : filter.collectionId ? /*#__PURE__*/_react.default.createElement(_filters.DynamicCollectionFilter, (0, _extends2.default)({
      filter: filter.filter,
      key: filter.key,
      component: filter.component,
      collectionId: filter.collectionId,
      componentName: filter.props.componentName
    }, filter.props, {
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 163,
        columnNumber: 11
      }
    })) : /*#__PURE__*/_react.default.createElement(filter.component, (0, _extends2.default)({
      key: filter.key,
      filter: filter.filter
    }, filter.props, {
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 172,
        columnNumber: 11
      }
    })));
    return acc;
  }, {
    filters: {},
    components: [],
    collectionToolbarFiltersProps
  });
  return {
    filterFieldMapping,
    filtersObject: filtersData.filters,
    filterComponent: filtersData.components.length > 0 ? /*#__PURE__*/_react.default.createElement(_patterns.CollectionToolbarFilters, (0, _extends2.default)({}, collectionToolbarFiltersProps, {
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 190,
        columnNumber: 9
      }
    }), filtersData.components) : undefined
  };
};
exports.useFilters = useFilters;
