// @wix/auto-patterns/dist/cjs/hooks/useGridFeatures.js
"use strict";

exports.__esModule = true;
exports.useGridFeatures = void 0;
var _useFetchData = require("./useFetchData");
var _patterns = require("@wix/patterns");
var _useCommonCollectionFeatures = require("./useCommonCollectionFeatures");
var _useFilters = require("./useFilters");
var _SchemaContext = require("../providers/SchemaContext");
var _useAutoPatternsOptimisticActions = require("./useAutoPatternsOptimisticActions");
var _useBaseGridFeatures = require("./useBaseGridFeatures");
const useGridFeatures = configuration => {
  const {
    collectionId,
    limit,
    reflectQueryInUrl,
    selectAllScope,
    selectionUpdateMode
  } = configuration.collection;
  const {
    renderItem,
    imagePlacement,
    preset,
    visibleFieldIds
  } = (0, _useBaseGridFeatures.useBaseGridFeatures)(configuration);
  const {
    filterComponent,
    filtersObject,
    filterFieldMapping
  } = (0, _useFilters.useFilters)(configuration.filters);
  const {
    fetchData
  } = (0, _useFetchData.useFetchData)({
    searchableFieldIds: visibleFieldIds,
    filterFieldMapping
  });
  const schema = (0, _SchemaContext.useSchema)();
  const state = (0, _patterns.useGridCollection)({
    queryName: `${collectionId}`,
    itemKey: gridItem => gridItem[schema.idField],
    itemName: gridItem => gridItem[schema.displayField],
    filters: filtersObject,
    toExtendedFields: gridItem => gridItem.extendedFields,
    fetchErrorMessage: ({
      err
    }) => String(err),
    fetchData,
    limit,
    persistQueryToUrl: reflectQueryInUrl,
    disableAutoSelectAllCount: selectAllScope === 'page',
    selectionConsistencyMode: selectionUpdateMode
  });
  const optimisticActions = (0, _useAutoPatternsOptimisticActions.useAutoPatternsOptimisticActions)(collectionId, state.collection);
  const commonFeatures = (0, _useCommonCollectionFeatures.useCommonCollectionFeatures)({
    ...configuration,
    optimisticActions
  });
  return {
    ...commonFeatures,
    state,
    imagePlacement,
    preset,
    optimisticActions,
    renderItem,
    filters: filterComponent
  };
};
exports.useGridFeatures = useGridFeatures;
