// @wix/auto-patterns/dist/cjs/hooks/useTableFeatures.js
"use strict";

exports.__esModule = true;
exports.useTableFeatures = void 0;
var _useFetchData = require("./useFetchData");
var _patterns = require("@wix/patterns");
var _useCommonCollectionFeatures = require("./useCommonCollectionFeatures");
var _useFilters = require("./useFilters");
var _SchemaContext = require("../providers/SchemaContext");
var _useAutoPatternsOptimisticActions = require("./useAutoPatternsOptimisticActions");
var _useBaseTableFeatures = require("./useBaseTableFeatures");
const useTableFeatures = configuration => {
  const {
    collectionId,
    limit,
    reflectQueryInUrl,
    selectAllScope,
    selectionUpdateMode
  } = configuration.collection;
  const {
    filterComponent,
    filtersObject,
    filterFieldMapping
  } = (0, _useFilters.useFilters)(configuration.filters);
  const {
    customColumns,
    visibleFieldIds,
    columns,
    onRowClick,
    stickyColumns,
    stickySelectionColumn
  } = (0, _useBaseTableFeatures.useBaseTableFeatures)(configuration);
  const {
    fetchData
  } = (0, _useFetchData.useFetchData)({
    searchableFieldIds: visibleFieldIds,
    filterFieldMapping
  });
  const schema = (0, _SchemaContext.useSchema)();
  const state = (0, _patterns.useTableCollection)({
    queryName: `${collectionId}`,
    itemKey: item => item[schema.idField],
    itemName: item => item[schema.displayField],
    filters: filtersObject,
    toExtendedFields: item => item.extendedFields,
    fetchErrorMessage: ({
      err
    }) => String(err),
    fetchData,
    limit,
    persistQueryToUrl: reflectQueryInUrl,
    disableAutoSelectAllCount: selectAllScope === 'page',
    selectionConsistencyMode: selectionUpdateMode
  });
  const optimisticActions = (0, _useAutoPatternsOptimisticActions.useAutoPatternsOptimisticActions)(collectionId, state.collection);
  const commonFeatures = (0, _useCommonCollectionFeatures.useCommonCollectionFeatures)({
    ...configuration,
    optimisticActions
  });
  return {
    ...commonFeatures,
    state,
    columns,
    filters: filterComponent,
    optimisticActions,
    customColumns,
    onRowClick,
    stickyColumns,
    stickySelectionColumn
  };
};
exports.useTableFeatures = useTableFeatures;
