// @wix/auto-patterns/dist/cjs/hooks/useTableGridSwitchFeatures.js
"use strict";

exports.__esModule = true;
exports.useTableGridSwitchFeatures = void 0;
var _useFetchData = require("./useFetchData");
var _patterns = require("@wix/patterns");
var _useCommonCollectionFeatures = require("./useCommonCollectionFeatures");
var _useFilters = require("./useFilters");
var _SchemaContext = require("../providers/SchemaContext");
var _useAutoPatternsOptimisticActions = require("./useAutoPatternsOptimisticActions");
var _useBaseGridFeatures = require("./useBaseGridFeatures");
var _useBaseTableFeatures = require("./useBaseTableFeatures");
const useTableGridSwitchFeatures = config => {
  const schema = (0, _SchemaContext.useSchema)();
  const {
    collectionId,
    limit,
    reflectQueryInUrl,
    selectAllScope,
    selectionUpdateMode
  } = config.collection;
  const {
    filterComponent,
    filtersObject,
    filterFieldMapping
  } = (0, _useFilters.useFilters)(config.filters);
  const {
    preset,
    imagePlacement,
    visibleFieldIds: gridVisibleFieldIds,
    renderItem
  } = (0, _useBaseGridFeatures.useBaseGridFeatures)(config);
  const {
    customColumns,
    visibleFieldIds: tableVisibleFieldIds,
    columns: tableColumns,
    onRowClick,
    stickyColumns,
    stickySelectionColumn
  } = (0, _useBaseTableFeatures.useBaseTableFeatures)(config);
  const {
    fetchData: tableFetchData
  } = (0, _useFetchData.useFetchData)({
    searchableFieldIds: Array.from(new Set([...tableVisibleFieldIds, ...gridVisibleFieldIds])),
    filterFieldMapping
  });
  const state = (0, _patterns.useTableGridSwitchCollection)({
    queryName: `${collectionId}`,
    itemKey: item => item[schema.idField],
    itemName: item => item[schema.displayField],
    filters: filtersObject,
    toExtendedFields: item => item.extendedFields,
    fetchErrorMessage: ({
      err
    }) => String(err),
    fetchData: tableFetchData,
    limit,
    persistQueryToUrl: reflectQueryInUrl,
    disableAutoSelectAllCount: selectAllScope === 'page',
    selectionConsistencyMode: selectionUpdateMode
  });
  const optimisticActions = (0, _useAutoPatternsOptimisticActions.useAutoPatternsOptimisticActions)(collectionId, state.collection);
  const commonFeatures = (0, _useCommonCollectionFeatures.useCommonCollectionFeatures)({
    ...config,
    optimisticActions
  });
  return {
    ...commonFeatures,
    state,
    columns: tableColumns,
    filters: filterComponent,
    optimisticActions,
    customColumns,
    onRowClick,
    stickyColumns,
    stickySelectionColumn,
    horizontalScroll: true,
    // Grid specific features
    imagePlacement,
    preset,
    renderItem
  };
};
exports.useTableGridSwitchFeatures = useTableGridSwitchFeatures;
