// @wix/ricos/dist/cjs/ricos-common-plugins/breakout/breakout.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var breakout_exports = {};
__export(breakout_exports, {
  pluginBreakout: () => pluginBreakout
});
module.exports = __toCommonJS(breakout_exports);
var import_extension = require("./extension.js");
const pluginBreakout = {
  type: "BREAKOUT",
  config: {},
  tiptapExtensions: [import_extension.breakout],
  shortcuts: [],
  textButtons: []
};
