// @wix/auto_sdk_events_schedule-bookmarks/build/internal/es/meta.mjs
// src/events-schedule-v1-schedule-item-schedule-bookmarks.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixEventsScheduleBookmarkManagementUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/wix-events-agenda",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    _: [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "editor-flow.wixapps.net": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/events/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "api._base_domain_": [
      {
        srcPath: "/events/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_events_schedule-bookmarks";
function listBookmarks(payload) {
  function __listBookmarks({ host }) {
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "GET",
      methodFqn: "wix.events.schedule.BookmarkManagement.ListBookmarks",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleBookmarkManagementUrl({
        protoPath: "/v1/schedule/bookmarks",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "items.createdDate" },
            { path: "items.updatedDate" },
            { path: "items.timeSlot.start" },
            { path: "items.timeSlot.end" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listBookmarks;
}
function createBookmark(payload) {
  function __createBookmark({ host }) {
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "POST",
      methodFqn: "wix.events.schedule.BookmarkManagement.CreateBookmark",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleBookmarkManagementUrl({
        protoPath: "/v1/schedule/{itemId}/bookmark",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __createBookmark;
}
function deleteBookmark(payload) {
  function __deleteBookmark({ host }) {
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "DELETE",
      methodFqn: "wix.events.schedule.BookmarkManagement.DeleteBookmark",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleBookmarkManagementUrl({
        protoPath: "/v1/schedule/{itemId}/bookmark",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteBookmark;
}

// src/events-schedule-v1-schedule-item-schedule-bookmarks.types.ts
var ScheduleStatus = /* @__PURE__ */ ((ScheduleStatus2) => {
  ScheduleStatus2["SCHEDULED"] = "SCHEDULED";
  ScheduleStatus2["CANCELED"] = "CANCELED";
  return ScheduleStatus2;
})(ScheduleStatus || {});
var StateFilter = /* @__PURE__ */ ((StateFilter2) => {
  StateFilter2["PUBLISHED"] = "PUBLISHED";
  StateFilter2["DRAFT"] = "DRAFT";
  StateFilter2["VISIBLE"] = "VISIBLE";
  StateFilter2["HIDDEN"] = "HIDDEN";
  return StateFilter2;
})(StateFilter || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});

// src/events-schedule-v1-schedule-item-schedule-bookmarks.meta.ts
function listBookmarks2() {
  const payload = {};
  const getRequestOptions = listBookmarks(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/schedule/bookmarks",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createBookmark2() {
  const payload = { itemId: ":itemId" };
  const getRequestOptions = createBookmark(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/schedule/{itemId}/bookmark",
    pathParams: { itemId: "itemId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteBookmark2() {
  const payload = { itemId: ":itemId" };
  const getRequestOptions = deleteBookmark(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/schedule/{itemId}/bookmark",
    pathParams: { itemId: "itemId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  ScheduleStatus as ScheduleStatusOriginal,
  SortOrder as SortOrderOriginal,
  StateFilter as StateFilterOriginal,
  createBookmark2 as createBookmark,
  deleteBookmark2 as deleteBookmark,
  listBookmarks2 as listBookmarks
};
