// @wix/patterns/dist/cjs/components/FormPageAdditionalContent/FormPageAdditionalContent.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.FormPageAdditionalContent = FormPageAdditionalContent;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireDefault(require("react"));
var _designSystem = require("@wix/design-system");
var _useIsMobile = require("../../hooks/useIsMobile");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/FormPageAdditionalContent/FormPageAdditionalContent.tsx";
function FormPageAdditionalContent(props) {
  const isMobile = (0, _useIsMobile.useIsMobile)();
  const {
    children,
    ...rest
  } = props;
  return /*#__PURE__*/_react.default.createElement(_designSystem.Cell, (0, _extends2.default)({
    span: isMobile ? 12 : 4
  }, rest, {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 19,
      columnNumber: 5
    }
  }), /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
    direction: "vertical",
    gap: isMobile ? 'SP2' : 'SP4',
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 20,
      columnNumber: 7
    }
  }, children));
}
FormPageAdditionalContent.displayName = 'FormPageAdditionalContent';
