// @wix/patterns/dist/cjs/components/GridDragAndDropDndKit/DraggableCardOverlay.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.DraggableCardOverlay = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireWildcard(require("react"));
var _core = require("@wix/wix-style-react-incubator/dnd-kit/core");
var _reactDom = require("react-dom");
var _modifiers = require("@wix/wix-style-react-incubator/dnd-kit/modifiers");
var _GridDragOverlayCardItem = require("./GridDragOverlayCardItem");
var _mobxReactLite = require("mobx-react-lite");
var _react2 = require("@wix/bex-core/react");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/GridDragAndDropDndKit/DraggableCardOverlay.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
const DraggableCardOverlayWrapper = /*#__PURE__*/(0, _react.forwardRef)((props, ref) => /*#__PURE__*/_react.default.createElement("div", (0, _extends2.default)({
  "data-hook": "draggable-card-overlay",
  ref: ref
}, props, {
  __self: void 0,
  __source: {
    fileName: _jsxFileName,
    lineNumber: 18,
    columnNumber: 3
  }
})));
function _DraggableCardOverlay(props) {
  const {
    state,
    renderItem
  } = props;
  const {
    _dnd
  } = state;
  const {
    modalsContainerRef
  } = (0, _react2.usePageContext)();
  return /*#__PURE__*/(0, _reactDom.createPortal)(/*#__PURE__*/_react.default.createElement(_core.DragOverlay, {
    dropAnimation: _dnd == null ? void 0 : _dnd.dropAnimation,
    modifiers: [_modifiers.restrictToWindowEdges],
    wrapperElement: DraggableCardOverlayWrapper,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 31,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(_GridDragOverlayCardItem.GridDragOverlayCardItem, {
    renderItem: renderItem,
    state: state,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 40,
      columnNumber: 7
    }
  })), (modalsContainerRef == null ? void 0 : modalsContainerRef.current) ?? document.body);
}
const DraggableCardOverlay = exports.DraggableCardOverlay = (0, _mobxReactLite.observer)(_DraggableCardOverlay);
