// @wix/design-system/dist/cjs/BadgeSelect/BadgeSelect.private.driver.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.badgeSelectPrivateDriverFactory = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _Badge = _interopRequireDefault(require("../Badge/Badge.driver"));
var _Popover = require("../Popover/Popover.driver");
var _BadgeSelect = _interopRequireDefault(require("./BadgeSelect.driver"));
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2.default)(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var badgeSelectPrivateDriverFactory = _ref => {
  var {
    element,
    eventTrigger
  } = _ref;
  var popoverDriver = (0, _Popover.popoverLegacyDriverFactory)({
    element,
    eventTrigger
  });
  var badgeDriver = (0, _Badge.default)({
    element: popoverDriver.getTargetElement().childNodes[0],
    eventTrigger
  });
  return _objectSpread(_objectSpread({}, (0, _BadgeSelect.default)({
    element,
    eventTrigger
  })), {}, {
    /** Clicks on the element */
    click: () => element.click(),
    /** Clicks on the badge */
    clickOnBadge: () => badgeDriver.click(),
    /** Performs a click outside the component */
    clickOutside: () => popoverDriver.clickOutside(),
    /** Whether the dropdown is shown */
    isDropdownShown: () => popoverDriver.isContentElementExists(),
    /** Gets badge type */
    getType: badgeDriver.getType,
    /** Gets badge type */
    getSkin: badgeDriver.getSkin,
    /** Gets badge size */
    getSize: badgeDriver.getSize,
    /** Checks whether the text is uppercase */
    isUppercase: badgeDriver.isUppercase
  });
};
exports.badgeSelectPrivateDriverFactory = badgeSelectPrivateDriverFactory;
