// @wix/design-system/dist/cjs/BadgeSelect/BadgeSelect.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.default = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = _interopRequireDefault(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _wixUiIconsCommon = require("@wix/wix-ui-icons-common");
var _BadgeSelectSt = require("./BadgeSelect.st.css.js");
var _DropdownLayout = _interopRequireDefault(require("../DropdownLayout"));
var _Popover = _interopRequireDefault(require("../Popover"));
var _Badge = _interopRequireDefault(require("../Badge"));
var _BadgeSelectItem = require("../BadgeSelectItem");
var DATA_ATTR = _interopRequireWildcard(require("./DataAttr"));
var _PopoverCommon = require("../common/PropTypes/PopoverCommon");
var _jsxFileName = "/home/builduser/work/57e038ea7326c1ec/packages/wix-design-system/dist/cjs/BadgeSelect/BadgeSelect.js";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function _interopRequireWildcard(e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (var _t in e) "default" !== _t && {}.hasOwnProperty.call(e, _t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, _t)) && (i.get || i.set) ? o(f, _t, i) : f[_t] = e[_t]); return f; })(e, t); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2.default)(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
class BadgeSelect extends _react.default.Component {
  constructor(_props) {
    super(_props);
    this._isControlled = () => {
      return typeof this.props.selectedId !== 'undefined';
    };
    /**
     * Determine if a certain key should open the DropdownLayout
     *
     * @param {KeyboardEvent.key} key - The key name
     * @return {boolean} - Whether the key should cause the DropdownLayout to open
     */
    this._isOpenKey = key => {
      return ['Enter', 'Spacebar', ' ', 'ArrowDown'].includes(key);
    };
    this._getBadgeOptionById = (options, wantedId) => {
      return options.find(_ref => {
        var {
          id
        } = _ref;
        return id === wantedId;
      });
    };
    this._handleSelect = _ref2 => {
      var {
        id: selectedId
      } = _ref2;
      var {
        onSelect,
        options
      } = this.props;
      var selectedBadge = this._getBadgeOptionById(options, selectedId);
      var newState = {
        visible: false
      };
      if (!this._isControlled()) {
        newState.selectedBadge = selectedBadge;
      }
      this.setState(newState);
      onSelect && onSelect(selectedBadge);
    };
    this._onKeyDown = event => {
      // Delegate the event to the DropdownLayout. It'll handle the navigation,
      // option selection and closing of the dropdown.
      var isHandledByDropdownLayout = this.dropdownLayout ? this.dropdownLayout._onSelectListKeyDown(event) : false;

      // If the event wasn't handled by the DropdownLayout correctly, check if
      // the pressed key should open the dropdown and act accordingly.
      if (!isHandledByDropdownLayout) {
        if (this._isOpenKey(event.key)) {
          this.showDropdown();
          event.preventDefault();
        }
      }
    };
    this.getSelectedOption = props => {
      return this._getBadgeOptionById(props.options, props.selectedId) || props.options[0];
    };
    this.hideDropdown = () => {
      this.setState({
        visible: false
      });
    };
    this.showDropdown = () => {
      this.setState({
        visible: true
      });
    };
    this.toggleDropdown = () => {
      this.setState({
        visible: !this.state.visible
      });
    };
    this.state = {
      visible: false,
      selectedBadge: this.getSelectedOption(_props)
    };
  }
  get options() {
    var {
      options
    } = this.props;
    return Array.isArray(options) ? options.map(option => (0, _BadgeSelectItem.badgeSelectItemBuilder)(_objectSpread({}, option))) : [];
  }
  UNSAFE_componentWillReceiveProps(nextProps) {
    if (this.props.selectedId !== nextProps.selectedId) {
      this.setState({
        selectedBadge: this.getSelectedOption(nextProps)
      });
    }
  }
  render() {
    var {
      type,
      size,
      uppercase,
      dataHook,
      popoverProps
    } = this.props;
    var {
      visible,
      selectedBadge
    } = this.state;
    return /*#__PURE__*/_react.default.createElement(_Popover.default, (0, _extends2.default)({
      shown: visible,
      dataHook: dataHook,
      placement: "bottom",
      onKeyDown: this._onKeyDown,
      onClickOutside: this.hideDropdown
    }, popoverProps, {
      className: _BadgeSelectSt.classes.root,
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 111,
        columnNumber: 7
      }
    }), /*#__PURE__*/_react.default.createElement(_Popover.default.Element, {
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 120,
        columnNumber: 9
      }
    }, /*#__PURE__*/_react.default.createElement(_Badge.default, {
      ref: badge => this.badge = badge,
      type: type,
      size: size,
      uppercase: uppercase,
      suffixIcon: /*#__PURE__*/_react.default.createElement(_wixUiIconsCommon.ChevronDownSmall, {
        __self: this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 126,
          columnNumber: 25
        }
      }),
      skin: this.state.selectedBadge.skin,
      onClick: this.toggleDropdown,
      dataHook: DATA_ATTR.DATA_BADGE,
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 121,
        columnNumber: 11
      }
    }, this.state.selectedBadge.text)), /*#__PURE__*/_react.default.createElement(_Popover.default.Content, {
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 135,
        columnNumber: 9
      }
    }, /*#__PURE__*/_react.default.createElement(_DropdownLayout.default, {
      ref: r => this.dropdownLayout = r,
      dataHook: DATA_ATTR.DATA_DROPDOWN,
      visible: true,
      selectedId: selectedBadge.id,
      options: this.options,
      onSelect: this._handleSelect,
      onClose: this.hideDropdown,
      inContainer: true,
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 136,
        columnNumber: 11
      }
    })));
  }
}
BadgeSelect.displayName = 'BadgeSelect';
BadgeSelect.propTypes = {
  /** An array of options. Each option must have a unique `id`, a `text` and a `skin` whose value should match one of `<Badge/>`'s skin values */
  options: _propTypes.default.arrayOf(_propTypes.default.shape({
    id: _propTypes.default.string.isRequired,
    skin: _propTypes.default.oneOf(['general', 'standard', 'danger', 'success', 'neutral', 'neutralLight', 'warning', 'warningLight', 'urgent', 'neutralStandard', 'neutralSuccess', 'neutralDanger', 'premium']).isRequired,
    text: _propTypes.default.string.isRequired,
    subtitle: _propTypes.default.string,
    ellipsis: _propTypes.default.bool
  })).isRequired,
  /** The id of the selected option in the list */
  selectedId: _propTypes.default.string,
  /** Callback function called whenever the user selects a different option in the list */
  onSelect: _propTypes.default.func,
  /** The size of the `<Badge/>` */
  size: _propTypes.default.oneOf(['small', 'medium']),
  /** The type of the `<Badge/>` */
  type: _propTypes.default.oneOf(['solid', 'outlined', 'transparent']),
  /** Whether the text of the `<Badge/>` should be uppercase */
  uppercase: _propTypes.default.bool,
  /** Applied as data-hook HTML attribute that can be used to create driver in testing */
  dataHook: _propTypes.default.string,
  /** common popover props */
  popoverProps: _propTypes.default.shape(_PopoverCommon.PopoverCommonProps)
};
BadgeSelect.defaultProps = {
  size: 'medium',
  type: 'solid',
  uppercase: true
};
var _default = exports.default = BadgeSelect;
