// @wix/ricos/dist/es/toolbars-v3/components/buttons/AddPluginMenuButtons/AddPluginMenuMobileButton.js
import React, { useContext } from "react";
import { ModalServiceContext, RicosContext } from "./../../../../ricos-context/index.js";
import { ToolbarButton } from "./../../../../_shared/toolbar-button/toolbar-button.js";
import { toggleQuickActionsModal } from "./../../../../ricos-common-plugins/quick-actions/toggle-quick-actions-modal/toggle-quick-actions-modal.js";
import classes from "./AddPluginMenuMobileButton-scss.js";
const AddPluginMenuMobileButton = ({
  toolbarItem
}) => {
  const { isMobile } = useContext(RicosContext);
  const modalService = useContext(ModalServiceContext) || {};
  return /* @__PURE__ */ React.createElement(
    ToolbarButton,
    {
      dataHook: "add-plugin-button-mobile",
      onClick: () => {
        toggleQuickActionsModal({ modalService, referenceElement: null, isMobile });
      },
      className: classes.toggleButton,
      prefix: toolbarItem.presentation?.icon,
      tabIndex: toolbarItem.presentation.tabIndex
    }
  );
};
export {
  AddPluginMenuMobileButton
};
