// @wix/patterns/dist/cjs/components/Extensions/Extensions.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.Extensions = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireDefault(require("react"));
var _mobxReactLite = require("mobx-react-lite");
var _designSystem = require("@wix/design-system");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/Extensions/Extensions.tsx";
function _Extensions(params) {
  const {
    state,
    dataHook,
    ...rest
  } = params;
  const {
    items
  } = state;
  return /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
    direction: "vertical",
    gap: "SP4",
    dataHook: dataHook,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 18,
      columnNumber: 5
    }
  }, items.map(({
    Extension: _Extension,
    componentId
  }) => {
    const Extension = _Extension;
    return /*#__PURE__*/_react.default.createElement(Extension, (0, _extends2.default)({}, rest, {
      key: componentId,
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 21,
        columnNumber: 16
      }
    }));
  }));
}
const Extensions = exports.Extensions = (0, _mobxReactLite.observer)(_Extensions);
