// @wix/patterns/dist/cjs/test-utils/AppsMocks.js
"use strict";

exports.__esModule = true;
exports.appsAPIHttpClientMocks = void 0;
exports.createApp = createApp;
exports.createBackend = createBackend;
var _client = require("@wix/fe-essentials/http-client/testkit/client");
var _http = require("@wix/bex-utils/@wix/ambassador-devcenter-app-market-listing-v1-market-listing/http");
var _backend = require("../testkit/backend");
function createApp({
  createdDate = new Date(Date.now() - 365 * 24 * 60 * 60 * 1000),
  updatedDate = createdDate,
  ...overrides
} = {}) {
  return {
    createdDate,
    updatedDate,
    ...overrides
  };
}
function createBackend(apps) {
  return new _backend.InMemoryBackend({
    total: apps.length,
    paginationMode: 'offset',
    delay: {
      min: 100,
      max: 2000
    },
    createOne: index => createApp(apps[index]),
    predicate: query => {
      const filters = query.filters ?? {};
      const {
        appId,
        languageCode
      } = filters;
      return item => appId.includes(item.appId) && languageCode.includes(item.languageCode);
    },
    orderBy: query => {
      return [...(query.sort ?? [])];
    },
    itemKey: item => item.id
  });
}
const appsAPIHttpClientMocks = apps => {
  const appsBackend = createBackend(apps);
  return [(0, _client.whenRequest)(_http.queryMarketListing).reply(200, async ({
    query: {
      filter
    }
  }) => {
    return appsBackend.fetchData({
      filters: {
        appId: filter == null ? void 0 : filter.appId,
        languageCode: filter == null ? void 0 : filter.languageCode
      },
      limit: 100
    }).then(({
      items
    }) => {
      return {
        marketListing: items
      };
    });
  }).persist()];
};
exports.appsAPIHttpClientMocks = appsAPIHttpClientMocks;
