// @wix/patterns/dist/cjs/test-utils/CairoTestApp.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.CairoTestApp = CairoTestApp;
var _react = _interopRequireDefault(require("react"));
var _components = require("../components");
var _designSystem = require("@wix/design-system");
var _react2 = require("@wix/bex-core/react");
var _react3 = require("@wix/fe-essentials/react");
var _EntityPage = require("../components/EntityPage");
var _CollectionPageNew = require("../components/CollectionPageNew");
var _PageWrapper = require("../components/PageWrapper/PageWrapper");
var _Router = require("../components/Router");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/test-utils/CairoTestApp.tsx";
function CairoTestApp({
  children,
  driver,
  page,
  scrollElement,
  entityPageState,
  collectionPageState,
  routerParams
}) {
  let child = children;
  switch (page) {
    case 'wsr':
      child = /*#__PURE__*/_react.default.createElement(_designSystem.Page, {
        __self: this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 46,
          columnNumber: 9
        }
      }, /*#__PURE__*/_react.default.createElement(_designSystem.Page.Header, {
        title: "Page Title",
        __self: this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 47,
          columnNumber: 11
        }
      }), /*#__PURE__*/_react.default.createElement(_designSystem.Page.Content, {
        __self: this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 48,
          columnNumber: 11
        }
      }, children));
      break;
    case 'entity':
      child = /*#__PURE__*/_react.default.createElement(_designSystem.Page, {
        __self: this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 54,
          columnNumber: 9
        }
      }, /*#__PURE__*/_react.default.createElement(_designSystem.Page.Content, {
        __self: this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 55,
          columnNumber: 11
        }
      }, children));
      break;
    case 'none':
      child = /*#__PURE__*/_react.default.createElement("div", {
        ref: scrollElement,
        "data-hook": "custom-scroll-container",
        style: {
          height: '200px',
          overflow: 'auto'
        },
        __self: this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 61,
          columnNumber: 9
        }
      }, child);
      break;
    case 'entity-page':
      if (!entityPageState) {
        throw new Error('Page state was not initialized');
      }
      child = /*#__PURE__*/_react.default.createElement(_EntityPage.EntityPage, {
        state: entityPageState,
        dataHook: page,
        __self: this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 75,
          columnNumber: 9
        }
      }, children);
      break;
    case 'collection-page':
      if (!collectionPageState) {
        throw new Error('Page state was not initialized');
      }
      child = /*#__PURE__*/_react.default.createElement(_CollectionPageNew.CollectionPage, {
        dataHook: page,
        __self: this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 85,
          columnNumber: 9
        }
      }, /*#__PURE__*/_react.default.createElement(_CollectionPageNew.CollectionPage.Header, {
        title: {
          text: collectionPageState.pageTitle || 'Title',
          hideTotal: collectionPageState.hideHeaderTotal
        },
        __self: this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 86,
          columnNumber: 11
        }
      }), /*#__PURE__*/_react.default.createElement(_CollectionPageNew.CollectionPage.Content, {
        __self: this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 92,
          columnNumber: 11
        }
      }, children));
      break;
    case 'router':
      if (!routerParams) {
        throw new Error('Router params were not initialized');
      }
      _react.default.Children.forEach(children, currentChild => {
        if (! /*#__PURE__*/_react.default.isValidElement(currentChild)) {
          throw new Error('Each child needs to be a Route element');
        }
      });
      child = /*#__PURE__*/_react.default.createElement(_Router.PatternsReactRouter, {
        __self: this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 108,
          columnNumber: 9
        }
      }, children);
      break;
    default:
      child = /*#__PURE__*/_react.default.createElement(_PageWrapper.PageWrapper, {
        __self: this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 115,
          columnNumber: 9
        }
      }, /*#__PURE__*/_react.default.createElement(_designSystem.Page, {
        __self: this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 116,
          columnNumber: 11
        }
      }, /*#__PURE__*/_react.default.createElement(_designSystem.Page.Content, {
        __self: this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 117,
          columnNumber: 13
        }
      }, children)));
      break;
  }
  return /*#__PURE__*/_react.default.createElement(_react3.EssentialsProvider, {
    essentials: driver.testContext,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 125,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.WixStyleReactEnvironmentProvider, {
    locale: driver.testContext.environment.language,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 126,
      columnNumber: 7
    }
  }, /*#__PURE__*/_react.default.createElement(_components.WixPatternsTestProvider, {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 129,
      columnNumber: 9
    }
  }, /*#__PURE__*/_react.default.createElement(_react2.WixPatternsContainerProvider, {
    value: driver.container,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 130,
      columnNumber: 11
    }
  }, child))));
}
