// @wix/patterns/dist/cjs/test-utils/dataExtensionServiceMocks.js
"use strict";

exports.__esModule = true;
exports.aDataExtensionField = void 0;
exports.dataExtensionServiceMocks = dataExtensionServiceMocks;
exports.resolveDataExtensions = void 0;
var _http = require("@wix/bex-utils/@wix/ambassador-data-extensions-v1-data-extension-schema/http");
var _chance = require("chance");
function dataExtensionServiceMocks(whenRequest) {
  return {
    mocks: [whenRequest(_http.listDataExtensionSchemas).reply(200, async () => {
      const res = {
        dataExtensionSchemas: []
      };
      return res;
    }).persist()]
  };
}
const aDataExtensionField = (overrides = {}) => {
  const chance = new _chance.Chance();
  return {
    id: `${chance.word()}_${chance.integer({
      min: 0,
      max: 1000
    })}`,
    title: chance.name(),
    type: 'string',
    'x-wix-permissions': {
      read: ['apps', 'users'],
      write: ['apps']
    },
    'x-wix-created-date': '2023-06-05T11:15:39+00:00',
    ...overrides
  };
};
exports.aDataExtensionField = aDataExtensionField;
const resolveDataExtensions = ({
  whenRequest,
  fqdn,
  fields,
  extensionPoint,
  persist = true
}) => {
  const chance = new _chance.Chance();
  const request = whenRequest(_http.listDataExtensionSchemas).reply(200, async () => {
    return {
      dataExtensionSchemas: [{
        id: chance.guid(),
        fqdn,
        extensionPoint,
        namespace: '_user_fields',
        jsonSchema: {
          properties: fields.reduce((acc, curr) => {
            acc[curr.id] = curr;
            return acc;
          }, {})
        }
      }]
    };
  });
  if (persist) {
    return request.persist();
  }
  return request;
};
exports.resolveDataExtensions = resolveDataExtensions;
