// @wix/patterns/dist/cjs/test-utils/DataExtensionMocks.js
"use strict";

exports.__esModule = true;
exports.createBackend = createBackend;
exports.createSchema = createSchema;
exports.dataExtensionAPIHttpClientMocks = void 0;
var _client = require("@wix/fe-essentials/http-client/testkit/client");
var _http = require("@wix/bex-utils/@wix/ambassador-data-extensions-v1-data-extension-schema/http");
var _backend = require("../testkit/backend");
var _chance = require("chance");
function createSchema(chance) {
  return ({
    createdDate = chance.date({
      min: new Date(Date.now() - 365 * 24 * 60 * 60 * 1000),
      max: new Date()
    }),
    updatedDate = createdDate,
    fqdn = 'wix.patterns.dummyservice.v1.dummy_entity',
    namespace = '_user_fields',
    ...overrides
  } = {}) => ({
    id: chance.guid(),
    createdDate,
    updatedDate,
    namespace,
    fqdn,
    ...overrides
  });
}
function createBackend(initialSchema, appSchemas) {
  const chance = new _chance.Chance();
  return new _backend.InMemoryBackend({
    total: 1 + appSchemas.length,
    paginationMode: 'offset',
    delay: {
      min: 100,
      max: 2000
    },
    createOne: i => createSchema(chance)(i === 0 ? initialSchema : appSchemas[i - 1]),
    predicate: query => {
      const filters = query.filters ?? {};
      const {
        fqdn
      } = filters;
      return item => item.fqdn === fqdn;
    },
    orderBy: query => {
      return [...(query.sort ?? [])];
    },
    itemKey: item => item.id
  });
}
const dataExtensionAPIHttpClientMocks = (initialSchema = {}, appSchemas = []) => {
  const dataExtensionBackend = createBackend(initialSchema, appSchemas);
  return [(0, _client.whenRequest)(_http.listDataExtensionSchemas).reply(200, async ({
    fqdn
  }) => {
    return dataExtensionBackend.fetchData({
      limit: 1 + appSchemas.length,
      filters: {
        fqdn
      }
    }).then(({
      items
    }) => {
      return {
        dataExtensionSchemas: items
      };
    });
  }).persist(), (0, _client.whenRequest)(_http.updateDataExtensionSchema).reply(200, async updatedSchema => {
    return dataExtensionBackend.update(updatedSchema.dataExtensionSchema).then(() => {
      return updatedSchema;
    });
  }).persist()];
};
exports.dataExtensionAPIHttpClientMocks = dataExtensionAPIHttpClientMocks;
