// @wix/patterns/dist/cjs/test-utils/ExportMocks.js
"use strict";

exports.__esModule = true;
exports.exportMocks = void 0;
var _client = require("@wix/fe-essentials/http-client/testkit/client");
var _http = require("@wix/bex-utils/@wix/ambassador-fedinfra-exportservice-v1-export-async-job/http");
let resultsDelayCounter = 0;
const exportMocks = exports.exportMocks = [(0, _client.whenRequest)(_http.createExportAsyncJob).reply(200, {
  __applicationErrorsType: {},
  job: {
    id: 'XXX'
  }
}).persist(), (0, _client.whenRequest)(_http.getExportAsyncJob).reply(200, () => {
  if (resultsDelayCounter < 5) {
    resultsDelayCounter++;
    return {
      __applicationErrorsType: {},
      job: {
        id: 'XXX',
        data: {
          pagingMetadata: {
            offset: 20 * resultsDelayCounter
          }
        }
      }
    };
  }
  resultsDelayCounter = 0;
  return {
    __applicationErrorsType: {},
    job: {
      id: 'XXX',
      data: {
        pagingMetadata: {
          offset: 10
        },
        signedUrl: './export.csv'
      }
    }
  };
}).persist()];
