// @wix/online-programs/build/internal/cjs/meta.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  categories: () => categories,
  instructors: () => instructors,
  onlineProgramParticipants: () => onlineProgramParticipants,
  participants: () => participants,
  programs: () => programs,
  quizzes: () => quizzes,
  sections: () => sections,
  steps: () => steps
});
module.exports = __toCommonJS(meta_exports);
var categories = __toESM(require("@wix/auto_sdk_online-programs_categories/meta"));
var instructors = __toESM(require("@wix/auto_sdk_online-programs_instructors/meta"));
var quizzes = __toESM(require("@wix/auto_sdk_online-programs_quizzes/meta"));
var onlineProgramParticipants = __toESM(require("@wix/auto_sdk_online-programs_online-program-participants/meta"));
var participants = __toESM(require("@wix/auto_sdk_online-programs_participants/meta"));
var programs = __toESM(require("@wix/auto_sdk_online-programs_programs/meta"));
var sections = __toESM(require("@wix/auto_sdk_online-programs_sections/meta"));
var steps = __toESM(require("@wix/auto_sdk_online-programs_steps/meta"));
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  categories,
  instructors,
  onlineProgramParticipants,
  participants,
  programs,
  quizzes,
  sections,
  steps
});
