// @wix/patterns/dist/esm/components/CollectionViewsDropdown/CollectionViewsDropdown.uni.driver.js
import { DropdownUniDriver, ListItemSectionUniDriver, } from '@wix/design-system/dist/testkit/unidriver';
import { baseUniDriverFactory } from '../../unidriver';
import { ViewsDropdownListItemUniDriver } from './ViewsDropdownListItem.uni.driver';
import { ManageViewPopoverUniDriver } from './ManageViewPopover.uni.driver';
import { CollectionSaveViewModalUniDriver } from './modals/SaveView/CollectionSaveViewModal.uni.driver';
import { CollectionRenameViewModalUniDriver } from './modals/RenameView/CollectionRenameViewModal.uni.driver';
export const CollectionViewsDropdownUniDriver = (base, body) => {
    const viewsDropdownDriverBase = () => base.$('[data-hook="collection-views-dropdown"]');
    const manageViewPopover = () => ManageViewPopoverUniDriver(base, body);
    const getSaveViewModal = () => CollectionSaveViewModalUniDriver(base, body);
    const getRenameViewModal = () => CollectionRenameViewModalUniDriver(base, body);
    const getDropdownLayoutBase = async () => body.$(`[data-content-element="${await viewsDropdownDriverBase().attr('data-content-hook')}"]`);
    const viewsDropdownDriver = () => DropdownUniDriver(viewsDropdownDriverBase(), body);
    const getViewListItemById = (id) => {
        return ViewsDropdownListItemUniDriver(viewsDropdownDriverBase().$(`[data-hook="dropdown-item-${id}"]`), body);
    };
    const getViewListItemAt = (index) => {
        return ViewsDropdownListItemUniDriver(viewsDropdownDriverBase().$$('[data-hook="view-item"]').get(index), body);
    };
    const getCategoryListItemById = (id) => {
        return ListItemSectionUniDriver(viewsDropdownDriverBase().$(`[data-hook="dropdown-item-${id}"]`), body);
    };
    const getCategoryListItemAt = (index) => {
        return ListItemSectionUniDriver(viewsDropdownDriverBase().$$('[data-hook="view-category"]').get(index), body);
    };
    const openManageViewPopover = () => manageViewPopover().open();
    const saveNewViewActionClick = () => manageViewPopover().clickSaveNewView();
    const deleteViewActionClick = () => manageViewPopover().clickDeleteView();
    const saveViewChangesActionClick = () => manageViewPopover().clickSaveChanges();
    const renameViewActionClick = () => manageViewPopover().clickRenameView();
    const setAsDefaultViewActionClick = () => manageViewPopover().clickSetAsDefaultView();
    return {
        ...baseUniDriverFactory(base),
        _input: () => viewsDropdownDriverBase().$('input'),
        exists: () => viewsDropdownDriver().exists(),
        initialLoaderExists: async () => (await getDropdownLayoutBase())
            .$('[data-hook="initial-loader"]')
            .exists(),
        /** Get the currently selected view name */
        getSelected: () => viewsDropdownDriver().inputDriver.getValue(),
        /** Get the dropdown input text */
        getInputValue: () => viewsDropdownDriver().inputDriver.getValue(),
        /** Indicates input is ready-only */
        getInputReadOnly: () => viewsDropdownDriver().inputDriver.getReadOnly(),
        /** Indicates input is focused */
        isInputFocused: () => viewsDropdownDriver().inputDriver.isFocus(),
        /** Get the currently selected view id */
        getSelectedId: () => viewsDropdownDriver().dropdownLayoutDriver.getSelectedOptionId(),
        /** Checks if total is shown */
        hasTotal: () => viewsDropdownDriver().inputDriver.suffixComponentExists('.total-count'),
        /** Opens views dropdown */
        open: () => viewsDropdownDriver().inputDriver.click(),
        /** Enter dropdown input text */
        enterText: (value) => viewsDropdownDriver().inputDriver.enterText(value),
        /** Click dropdown input clear button */
        clickInputClearButton: () => viewsDropdownDriver().inputDriver.clickClear(),
        /** Indicates input clear button exists  */
        inputClearButtonExists: () => viewsDropdownDriver().inputDriver.hasClearButton(),
        /** Get number of views */
        getViewsCount: async () => viewsDropdownDriverBase().$$('[data-hook="view-item"]').count(),
        /** Get array of all view names in the dropdown */
        getViewsContent: async () => viewsDropdownDriverBase().$$('[data-hook="view-item"]').text(),
        /** Get array popover content , like empty state */
        getViewsPopoverContent: async () => viewsDropdownDriverBase().$$('[data-hook="popover-content"]').text(),
        /** Indicates whether the dropdown is open */
        isOpen: () => viewsDropdownDriver().dropdownLayoutDriver.isShown(),
        /** Get the placeholder test of the dropdown */
        getPlaceholderText: () => viewsDropdownDriver().inputDriver.getPlaceholder(),
        /** Get modal driver for "Save View" */
        getSaveViewModal,
        /** Get modal driver for "Rename View" */
        getRenameViewModal,
        /** Get view list item driver by view id */
        getViewListItemById,
        /** Get view list item driver by view index */
        getViewListItemAt,
        /** Get category list item driver by category id */
        getCategoryListItemById,
        /** Get category list item driver by category index */
        getCategoryListItemAt,
        /** Get "all items" view list item */
        getAllItemsView: () => getCategoryListItemById('all-items-view'),
        /** Get number of categories */
        getCategoriesCount: async () => viewsDropdownDriverBase().$$('[data-hook="view-category"]').count(),
        /** Get array of all category names in the dropdown */
        getCategoriesContent: async () => viewsDropdownDriverBase().$$('[data-hook="view-category"]').text(),
        /** Check if view pending changes */
        isViewPendingChanges: async () => viewsDropdownDriver().inputDriver.prefixComponentExists('[data-hook="view-pending-changes"]'),
        isDefaultView: async () => viewsDropdownDriver().inputDriver.prefixComponentExists('[data-hook="view-default"]'),
        /** Indicates whether the Manage View popover exists */
        manageViewPopoverExists: () => manageViewPopover().exists(),
        /** Indicates whether the Manage View popover opened */
        isManageViewPopoverOpen: () => manageViewPopover().isOpen(),
        /** Opens the Manage View popover */
        openManageViewPopover,
        /** Clicks the Save Changes action in the Manage View popover */
        saveViewChangesActionClick,
        /** Clicks the Save as New View action in the Manage View popover */
        saveNewViewActionClick,
        /** Clicks the Rename action in the Manage View popover */
        renameViewActionClick,
        /** Clicks the Set as Default View action in the Manage View popover */
        setAsDefaultViewActionClick,
        /** Clicks the Delete action in the Manage View popover */
        deleteViewActionClick,
        /** Get DropdownLayoutOptionUniDriver of a Save Changes option */
        getSaveViewChangesBtn: () => manageViewPopover().getSaveChangesBtn(),
        /** Get DropdownLayoutOptionUniDriver of a Save as New View option */
        getSaveNewViewBtn: () => manageViewPopover().getSaveNewViewBtn(),
        /** Get DropdownLayoutOptionUniDriver of a Delete View option */
        getDeleteViewBtn: () => manageViewPopover().getDeleteViewBtn(),
        /** Get DropdownLayoutOptionUniDriver of a Rename option */
        getRenameViewBtn: () => manageViewPopover().getRenameViewBtn(),
        /** Get DropdownLayoutOptionUniDriver of a Set as Default View option */
        getSetAsDefaultViewBtn: () => manageViewPopover().getSetAsDefaultViewBtn(),
        /** Saves new view with a defined name and current filters / sort / columns */
        saveNewView: async (name) => {
            await openManageViewPopover();
            await saveNewViewActionClick();
            const saveViewModal = getSaveViewModal();
            await saveViewModal.fillViewNameInput(name);
            await saveViewModal.clickSaveButton();
        },
        /** Deletes the view with a given index */
        deleteView: async (ind) => {
            await open();
            await getViewListItemAt(ind).select();
            await openManageViewPopover();
            await deleteViewActionClick();
        },
        /** Saves changes to the current view */
        saveViewChanges: async () => {
            await openManageViewPopover();
            await saveViewChangesActionClick();
        },
        /** Renames the current view with a given name */
        renameView: async (name) => {
            await openManageViewPopover();
            await renameViewActionClick();
            const renameViewModal = getRenameViewModal();
            await renameViewModal.fillViewNameInput(name);
            await renameViewModal.clickSaveButton();
        },
        /** Sets a default view with a given index */
        setAsDefaultView: async (ind) => {
            await open();
            await getViewListItemAt(ind).select();
            await openManageViewPopover();
            await setAsDefaultViewActionClick();
        },
    };
};
