// @wix/auto_sdk_seo_seo-tags/build/internal/es/index.typings.mjs
// src/promote-seo-v1-seo-tags-seo-tags.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/promote-seo-v1-seo-tags-seo-tags.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixPromoteSeoTagsServiceV1SeoTagsServiceUrl(opts) {
  const domainToMappings = {
    _: [
      {
        srcPath: "/_api/seo-tags-service/v1/seo-tags",
        destPath: "/v1/seo-tags"
      },
      {
        srcPath: "/_api/seo-tags-service/v1/resolve-seo-tags",
        destPath: "/v1/resolve-seo-tags"
      },
      {
        srcPath: "/_api/seo-tags-service/v1/resolve-static-page-seo-tags",
        destPath: "/v1/resolve-static-page-seo-tags"
      },
      {
        srcPath: "/_api/seo-tags-service/v1/resolve-item-seo-tags",
        destPath: "/v1/resolve-item-seo-tags"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/seo-tags-service/v1/resolve-seo-tags",
        destPath: "/v1/resolve-seo-tags"
      },
      {
        srcPath: "/_api/seo-tags-service/v1/resolve-static-page-seo-tags",
        destPath: "/v1/resolve-static-page-seo-tags"
      },
      {
        srcPath: "/_api/seo-tags-service/v1/resolve-item-seo-tags",
        destPath: "/v1/resolve-item-seo-tags"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/promote/seo/v1/resolve-seo-tags",
        destPath: "/v1/resolve-seo-tags"
      },
      {
        srcPath: "/promote/seo/v1/resolve-static-page-seo-tags",
        destPath: "/v1/resolve-static-page-seo-tags"
      },
      {
        srcPath: "/promote/seo/v1/resolve-item-seo-tags",
        destPath: "/v1/resolve-item-seo-tags"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_seo_seo-tags";
function resolveSeoTags(payload) {
  function __resolveSeoTags({ host }) {
    const metadata = {
      entityFqdn: "wix.promote.seo.v1.seo_tags",
      method: "GET",
      methodFqn: "wix.promote.seo.tags.service.v1.SeoTagsService.ResolveSeoTags",
      packageName: PACKAGE_NAME,
      url: resolveWixPromoteSeoTagsServiceV1SeoTagsServiceUrl({
        protoPath: "/v1/resolve-seo-tags",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload, true)
    };
    return metadata;
  }
  return __resolveSeoTags;
}
function resolveStaticPageSeoTags(payload) {
  function __resolveStaticPageSeoTags({ host }) {
    const metadata = {
      entityFqdn: "wix.promote.seo.v1.seo_tags",
      method: "GET",
      methodFqn: "wix.promote.seo.tags.service.v1.SeoTagsService.ResolveStaticPageSeoTags",
      packageName: PACKAGE_NAME,
      url: resolveWixPromoteSeoTagsServiceV1SeoTagsServiceUrl({
        protoPath: "/v1/resolve-static-page-seo-tags",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload, true)
    };
    return metadata;
  }
  return __resolveStaticPageSeoTags;
}
function resolveItemSeoTags(payload) {
  function __resolveItemSeoTags({ host }) {
    const metadata = {
      entityFqdn: "wix.promote.seo.v1.seo_tags",
      method: "GET",
      methodFqn: "wix.promote.seo.tags.service.v1.SeoTagsService.ResolveItemSeoTags",
      packageName: PACKAGE_NAME,
      url: resolveWixPromoteSeoTagsServiceV1SeoTagsServiceUrl({
        protoPath: "/v1/resolve-item-seo-tags",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __resolveItemSeoTags;
}

// src/promote-seo-v1-seo-tags-seo-tags.universal.ts
var ItemType = /* @__PURE__ */ ((ItemType2) => {
  ItemType2["UNKNOWN_ITEM_TYPE"] = "UNKNOWN_ITEM_TYPE";
  ItemType2["STORES_PRODUCT"] = "STORES_PRODUCT";
  ItemType2["BOOKINGS_SERVICE"] = "BOOKINGS_SERVICE";
  ItemType2["STORES_CATEGORY"] = "STORES_CATEGORY";
  return ItemType2;
})(ItemType || {});
async function resolveSeoTags2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    itemType: options?.itemType,
    pageNameData: options?.pageNameData,
    slugData: options?.slugData,
    pageUrl: options?.pageUrl
  });
  const reqOpts = resolveSeoTags(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          itemType: "$[0].itemType",
          pageNameData: "$[0].pageNameData",
          slugData: "$[0].slugData",
          pageUrl: "$[0].pageUrl"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function resolveStaticPageSeoTags2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    pageUrl: options?.pageUrl,
    pageName: options?.pageName,
    seoData: options?.seoData
  });
  const reqOpts = resolveStaticPageSeoTags(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          pageUrl: "$[0].pageUrl",
          pageName: "$[0].pageName",
          seoData: "$[0].seoData"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function resolveItemSeoTags2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    pageUrl: options?.pageUrl,
    itemType: options?.itemType,
    slug: options?.slug,
    pagination: options?.pagination
  });
  const reqOpts = resolveItemSeoTags(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          pageUrl: "$[0].pageUrl",
          itemType: "$[0].itemType",
          slug: "$[0].slug",
          pagination: "$[0].pagination"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
export {
  ItemType,
  resolveItemSeoTags2 as resolveItemSeoTags,
  resolveSeoTags2 as resolveSeoTags,
  resolveStaticPageSeoTags2 as resolveStaticPageSeoTags
};
