// @wix/auto_sdk_pricing-plans_orders/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  CancellationCauseOriginal: () => CancellationCause,
  CancellationEffectiveAtOriginal: () => CancellationEffectiveAt,
  OrderMethodOriginal: () => OrderMethod,
  OrderStatusOriginal: () => OrderStatus,
  OrderTypeOriginal: () => OrderType,
  PaymentStatusOriginal: () => PaymentStatus,
  PeriodUnitOriginal: () => PeriodUnit,
  ReasonNotSuspendableOriginal: () => ReasonNotSuspendable,
  SetOriginal: () => Set,
  SortOrderOriginal: () => SortOrder,
  StatusOriginal: () => Status,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  cancelOrder: () => cancelOrder2,
  createOfflineOrder: () => createOfflineOrder2,
  createOnlineOrder: () => createOnlineOrder2,
  getOfflineOrderPreview: () => getOfflineOrderPreview2,
  getOnlineOrderPreview: () => getOnlineOrderPreview2,
  getPricePreview: () => getPricePreview2,
  managementGetOrder: () => managementGetOrder2,
  managementListOrders: () => managementListOrders2,
  markAsPaid: () => markAsPaid2,
  memberGetOrder: () => memberGetOrder2,
  memberListOrders: () => memberListOrders2,
  pauseOrder: () => pauseOrder2,
  postponeEndDate: () => postponeEndDate2,
  requestCancellation: () => requestCancellation2,
  resumeOrder: () => resumeOrder2
});
module.exports = __toCommonJS(meta_exports);

// src/pricing-plans-v2-order-orders.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressMembershipV2OrdersCheckoutServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/pricing-plans-checkout-proxy",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/paid-plans/v2/checkout",
        destPath: "/v2/checkout"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/paid-plans/v2/checkout",
        destPath: "/v2/checkout"
      }
    ],
    "*.wixapps.net": [
      {
        srcPath: "/_api/paid-plans/v2/checkout",
        destPath: "/v2/checkout"
      }
    ],
    _: [
      {
        srcPath: "/pricing-plans/v2/checkout",
        destPath: "/v2/checkout"
      },
      {
        srcPath: "/_api/paid-plans/v2/checkout",
        destPath: "/v2/checkout"
      },
      {
        srcPath: "/_api/pricing-plans/v2/checkout",
        destPath: "/v2/checkout"
      }
    ],
    "editor-flow.wixapps.net": [
      {
        srcPath: "/_api/paid-plans/v2/checkout",
        destPath: "/v2/checkout"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/paid-plans/v2/checkout",
        destPath: "/v2/checkout"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/paid-plans/v2/checkout",
        destPath: "/v2/checkout"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/paid-plans/v2/checkout",
        destPath: "/v2/checkout"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/paid-plans/v2/checkout",
        destPath: "/v2/checkout"
      }
    ],
    "forums._base_domain_": [
      {
        srcPath: "/_api/paid-plans/v2/checkout",
        destPath: "/v2/checkout"
      }
    ],
    "forums.wixapps.net": [
      {
        srcPath: "/_api/paid-plans/v2/checkout",
        destPath: "/v2/checkout"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/pricing-plans/v2/checkout",
        destPath: "/v2/checkout"
      },
      {
        srcPath: "/_api/paid-plans/v2/checkout",
        destPath: "/v2/checkout"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/paid-plans/v2/checkout",
        destPath: "/v2/checkout"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/pricing-plans/v2/checkout",
        destPath: "/v2/checkout"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
function resolveComWixpressMembershipV2OrdersMemberMemberOrdersServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/member-orders-service-proxy",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/pricing-plans-member-orders/v2/member/orders",
        destPath: "/v2/member/orders"
      },
      {
        srcPath: "/_api/pricing-plans/v2/member/orders",
        destPath: "/v2/member/orders"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/pricing-plans-member-orders/v2/member/orders",
        destPath: "/v2/member/orders"
      },
      {
        srcPath: "/_api/pricing-plans/v2/member/orders",
        destPath: "/v2/member/orders"
      }
    ],
    _: [
      {
        srcPath: "/_api/pricing-plans/v2/member/orders",
        destPath: "/v2/member/orders"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/pricing-plans/v2/member/orders",
        destPath: "/v2/member/orders"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/pricing-plans/v2/member/orders",
        destPath: "/v2/member/orders"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/pricing-plans/v2/member/orders",
        destPath: "/v2/member/orders"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/pricing-plans/v2/member/orders",
        destPath: "/v2/member/orders"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
function resolveComWixpressMembershipV2OrdersOrderManagementServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/order-management-service-proxy",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/paid-plans/v2/orders",
        destPath: "/v2/orders"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/paid-plans/v2/orders",
        destPath: "/v2/orders"
      }
    ],
    "*.wixapps.net": [
      {
        srcPath: "/_api/paid-plans/v2/orders",
        destPath: "/v2/orders"
      }
    ],
    _: [
      {
        srcPath: "/_api/paid-plans/v2/orders",
        destPath: "/v2/orders"
      }
    ],
    "editor-flow.wixapps.net": [
      {
        srcPath: "/_api/paid-plans/v2/orders",
        destPath: "/v2/orders"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/paid-plans/v2/orders",
        destPath: "/v2/orders"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/paid-plans/v2/orders",
        destPath: "/v2/orders"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/paid-plans/v2/orders",
        destPath: "/v2/orders"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/paid-plans/v2/orders",
        destPath: "/v2/orders"
      }
    ],
    "forums._base_domain_": [
      {
        srcPath: "/_api/paid-plans/v2/orders",
        destPath: "/v2/orders"
      }
    ],
    "forums.wixapps.net": [
      {
        srcPath: "/_api/paid-plans/v2/orders",
        destPath: "/v2/orders"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/pricing-plans/v2/orders",
        destPath: "/v2/orders"
      },
      {
        srcPath: "/_api/paid-plans/v2/orders",
        destPath: "/v2/orders"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/paid-plans/v2/orders",
        destPath: "/v2/orders"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/pricing-plans/v2/orders",
        destPath: "/v2/orders"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_pricing-plans_orders";
function managementGetOrder(payload) {
  function __managementGetOrder({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.order",
      method: "GET",
      methodFqn: "com.wixpress.membership.v2.orders.OrderManagementService.GetOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2OrdersOrderManagementServiceUrl({
        protoPath: "/v2/orders/{id}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.startDate" },
            { path: "order.endDate" },
            { path: "order.earliestEndDate" },
            { path: "order.createdDate" },
            { path: "order.updatedDate" },
            { path: "order.cancellation.requestedDate" },
            { path: "order.pausePeriods.pauseDate" },
            { path: "order.pausePeriods.resumeDate" },
            { path: "order.currentCycle.startedDate" },
            { path: "order.currentCycle.endedDate" },
            { path: "order.cycles.startedDate" },
            { path: "order.cycles.endedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __managementGetOrder;
}
function managementListOrders(payload) {
  function __managementListOrders({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.order",
      method: "GET",
      methodFqn: "com.wixpress.membership.v2.orders.OrderManagementService.ListOrders",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2OrdersOrderManagementServiceUrl({
        protoPath: "/v2/orders",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload, true),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "orders.startDate" },
            { path: "orders.endDate" },
            { path: "orders.earliestEndDate" },
            { path: "orders.createdDate" },
            { path: "orders.updatedDate" },
            { path: "orders.cancellation.requestedDate" },
            { path: "orders.pausePeriods.pauseDate" },
            { path: "orders.pausePeriods.resumeDate" },
            { path: "orders.currentCycle.startedDate" },
            { path: "orders.currentCycle.endedDate" },
            { path: "orders.cycles.startedDate" },
            { path: "orders.cycles.endedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __managementListOrders;
}
function postponeEndDate(payload) {
  function __postponeEndDate({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "endDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.order",
      method: "PATCH",
      methodFqn: "com.wixpress.membership.v2.orders.OrderManagementService.PostponeEndDate",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2OrdersOrderManagementServiceUrl({
        protoPath: "/v2/orders/{id}",
        data: serializedData,
        host
      }),
      data: serializedData
    };
    return metadata;
  }
  return __postponeEndDate;
}
function cancelOrder(payload) {
  function __cancelOrder({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.order",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.orders.OrderManagementService.CancelOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2OrdersOrderManagementServiceUrl({
        protoPath: "/v2/orders/{id}/cancel",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __cancelOrder;
}
function markAsPaid(payload) {
  function __markAsPaid({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.order",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.orders.OrderManagementService.MarkAsPaid",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2OrdersOrderManagementServiceUrl({
        protoPath: "/v2/orders/{id}/mark-as-paid",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __markAsPaid;
}
function pauseOrder(payload) {
  function __pauseOrder({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.order",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.orders.OrderManagementService.PauseOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2OrdersOrderManagementServiceUrl({
        protoPath: "/v2/orders/{id}/pause",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __pauseOrder;
}
function resumeOrder(payload) {
  function __resumeOrder({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.order",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.orders.OrderManagementService.ResumeOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2OrdersOrderManagementServiceUrl({
        protoPath: "/v2/orders/{id}/resume",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __resumeOrder;
}
function memberGetOrder(payload) {
  function __memberGetOrder({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.order",
      method: "GET",
      methodFqn: "com.wixpress.membership.v2.orders.member.MemberOrdersService.GetOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2OrdersMemberMemberOrdersServiceUrl({
        protoPath: "/v2/member/orders/{id}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.startDate" },
            { path: "order.endDate" },
            { path: "order.earliestEndDate" },
            { path: "order.createdDate" },
            { path: "order.updatedDate" },
            { path: "order.cancellation.requestedDate" },
            { path: "order.pausePeriods.pauseDate" },
            { path: "order.pausePeriods.resumeDate" },
            { path: "order.currentCycle.startedDate" },
            { path: "order.currentCycle.endedDate" },
            { path: "order.cycles.startedDate" },
            { path: "order.cycles.endedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __memberGetOrder;
}
function memberListOrders(payload) {
  function __memberListOrders({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.order",
      method: "GET",
      methodFqn: "com.wixpress.membership.v2.orders.member.MemberOrdersService.ListOrders",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2OrdersMemberMemberOrdersServiceUrl({
        protoPath: "/v2/member/orders",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload, true),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "orders.startDate" },
            { path: "orders.endDate" },
            { path: "orders.earliestEndDate" },
            { path: "orders.createdDate" },
            { path: "orders.updatedDate" },
            { path: "orders.cancellation.requestedDate" },
            { path: "orders.pausePeriods.pauseDate" },
            { path: "orders.pausePeriods.resumeDate" },
            { path: "orders.currentCycle.startedDate" },
            { path: "orders.currentCycle.endedDate" },
            { path: "orders.cycles.startedDate" },
            { path: "orders.cycles.endedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __memberListOrders;
}
function requestCancellation(payload) {
  function __requestCancellation({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.order",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.orders.member.MemberOrdersService.RequestCancellation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2OrdersMemberMemberOrdersServiceUrl({
        protoPath: "/v2/member/orders/{id}/cancel",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __requestCancellation;
}
function createOnlineOrder(payload) {
  function __createOnlineOrder({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "startDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.order",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.orders.CheckoutService.CreateOnlineOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2OrdersCheckoutServiceUrl({
        protoPath: "/v2/checkout/orders/online",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.startDate" },
            { path: "order.endDate" },
            { path: "order.earliestEndDate" },
            { path: "order.createdDate" },
            { path: "order.updatedDate" },
            { path: "order.cancellation.requestedDate" },
            { path: "order.pausePeriods.pauseDate" },
            { path: "order.pausePeriods.resumeDate" },
            { path: "order.currentCycle.startedDate" },
            { path: "order.currentCycle.endedDate" },
            { path: "order.cycles.startedDate" },
            { path: "order.cycles.endedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createOnlineOrder;
}
function createOfflineOrder(payload) {
  function __createOfflineOrder({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "startDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.order",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.orders.CheckoutService.CreateOfflineOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2OrdersCheckoutServiceUrl({
        protoPath: "/v2/checkout/orders/offline",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.startDate" },
            { path: "order.endDate" },
            { path: "order.earliestEndDate" },
            { path: "order.createdDate" },
            { path: "order.updatedDate" },
            { path: "order.cancellation.requestedDate" },
            { path: "order.pausePeriods.pauseDate" },
            { path: "order.pausePeriods.resumeDate" },
            { path: "order.currentCycle.startedDate" },
            { path: "order.currentCycle.endedDate" },
            { path: "order.cycles.startedDate" },
            { path: "order.cycles.endedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createOfflineOrder;
}
function getOnlineOrderPreview(payload) {
  function __getOnlineOrderPreview({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "startDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.order",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.orders.CheckoutService.GetOnlineOrderPreview",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2OrdersCheckoutServiceUrl({
        protoPath: "/v2/checkout/orders/preview-online",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.startDate" },
            { path: "order.endDate" },
            { path: "order.earliestEndDate" },
            { path: "order.createdDate" },
            { path: "order.updatedDate" },
            { path: "order.cancellation.requestedDate" },
            { path: "order.pausePeriods.pauseDate" },
            { path: "order.pausePeriods.resumeDate" },
            { path: "order.currentCycle.startedDate" },
            { path: "order.currentCycle.endedDate" },
            { path: "order.cycles.startedDate" },
            { path: "order.cycles.endedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getOnlineOrderPreview;
}
function getOfflineOrderPreview(payload) {
  function __getOfflineOrderPreview({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "startDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.order",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.orders.CheckoutService.GetOfflineOrderPreview",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2OrdersCheckoutServiceUrl({
        protoPath: "/v2/checkout/orders/preview-offline",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.startDate" },
            { path: "order.endDate" },
            { path: "order.earliestEndDate" },
            { path: "order.createdDate" },
            { path: "order.updatedDate" },
            { path: "order.cancellation.requestedDate" },
            { path: "order.pausePeriods.pauseDate" },
            { path: "order.pausePeriods.resumeDate" },
            { path: "order.currentCycle.startedDate" },
            { path: "order.currentCycle.endedDate" },
            { path: "order.cycles.startedDate" },
            { path: "order.cycles.endedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getOfflineOrderPreview;
}
function getPricePreview(payload) {
  function __getPricePreview({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.order",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.orders.CheckoutService.GetPricePreview",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2OrdersCheckoutServiceUrl({
        protoPath: "/v2/checkout/orders/price-preview",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __getPricePreview;
}

// src/pricing-plans-v2-order-orders.types.ts
var PeriodUnit = /* @__PURE__ */ ((PeriodUnit2) => {
  PeriodUnit2["UNDEFINED"] = "UNDEFINED";
  PeriodUnit2["DAY"] = "DAY";
  PeriodUnit2["WEEK"] = "WEEK";
  PeriodUnit2["MONTH"] = "MONTH";
  PeriodUnit2["YEAR"] = "YEAR";
  return PeriodUnit2;
})(PeriodUnit || {});
var OrderType = /* @__PURE__ */ ((OrderType2) => {
  OrderType2["UNDEFINED"] = "UNDEFINED";
  OrderType2["ONLINE"] = "ONLINE";
  OrderType2["OFFLINE"] = "OFFLINE";
  OrderType2["EXTERNAL"] = "EXTERNAL";
  return OrderType2;
})(OrderType || {});
var OrderMethod = /* @__PURE__ */ ((OrderMethod2) => {
  OrderMethod2["UNKNOWN"] = "UNKNOWN";
  OrderMethod2["MOTO"] = "MOTO";
  OrderMethod2["POS"] = "POS";
  return OrderMethod2;
})(OrderMethod || {});
var OrderStatus = /* @__PURE__ */ ((OrderStatus2) => {
  OrderStatus2["UNDEFINED"] = "UNDEFINED";
  OrderStatus2["DRAFT"] = "DRAFT";
  OrderStatus2["PENDING"] = "PENDING";
  OrderStatus2["ACTIVE"] = "ACTIVE";
  OrderStatus2["PAUSED"] = "PAUSED";
  OrderStatus2["ENDED"] = "ENDED";
  OrderStatus2["CANCELED"] = "CANCELED";
  return OrderStatus2;
})(OrderStatus || {});
var CancellationCause = /* @__PURE__ */ ((CancellationCause2) => {
  CancellationCause2["UNDEFINED"] = "UNDEFINED";
  CancellationCause2["OWNER_ACTION"] = "OWNER_ACTION";
  CancellationCause2["MEMBER_ACTION"] = "MEMBER_ACTION";
  CancellationCause2["PAYMENT_FAILURE"] = "PAYMENT_FAILURE";
  CancellationCause2["PAYMENT_SETUP_FAILURE"] = "PAYMENT_SETUP_FAILURE";
  CancellationCause2["UNKNOWN"] = "UNKNOWN";
  return CancellationCause2;
})(CancellationCause || {});
var CancellationEffectiveAt = /* @__PURE__ */ ((CancellationEffectiveAt2) => {
  CancellationEffectiveAt2["UNDEFINED"] = "UNDEFINED";
  CancellationEffectiveAt2["IMMEDIATELY"] = "IMMEDIATELY";
  CancellationEffectiveAt2["NEXT_PAYMENT_DATE"] = "NEXT_PAYMENT_DATE";
  return CancellationEffectiveAt2;
})(CancellationEffectiveAt || {});
var PaymentStatus = /* @__PURE__ */ ((PaymentStatus2) => {
  PaymentStatus2["UNDEFINED"] = "UNDEFINED";
  PaymentStatus2["PAID"] = "PAID";
  PaymentStatus2["REFUNDED"] = "REFUNDED";
  PaymentStatus2["FAILED"] = "FAILED";
  PaymentStatus2["UNPAID"] = "UNPAID";
  PaymentStatus2["PENDING"] = "PENDING";
  PaymentStatus2["NOT_APPLICABLE"] = "NOT_APPLICABLE";
  return PaymentStatus2;
})(PaymentStatus || {});
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["UNDEFINED"] = "UNDEFINED";
  Status2["ACTIVE"] = "ACTIVE";
  Status2["ENDED"] = "ENDED";
  return Status2;
})(Status || {});
var Set = /* @__PURE__ */ ((Set2) => {
  Set2["UNKNOWN_SET"] = "UNKNOWN_SET";
  Set2["BASIC"] = "BASIC";
  Set2["FULL"] = "FULL";
  return Set2;
})(Set || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var ReasonNotSuspendable = /* @__PURE__ */ ((ReasonNotSuspendable2) => {
  ReasonNotSuspendable2["UNDEFINED"] = "UNDEFINED";
  ReasonNotSuspendable2["PENDING"] = "PENDING";
  ReasonNotSuspendable2["TRIAL"] = "TRIAL";
  ReasonNotSuspendable2["CANCELED"] = "CANCELED";
  ReasonNotSuspendable2["ENDED"] = "ENDED";
  ReasonNotSuspendable2["NOT_STARTED"] = "NOT_STARTED";
  ReasonNotSuspendable2["ALREADY_SUSPENDED"] = "ALREADY_SUSPENDED";
  ReasonNotSuspendable2["OLD_STRIPE"] = "OLD_STRIPE";
  return ReasonNotSuspendable2;
})(ReasonNotSuspendable || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/pricing-plans-v2-order-orders.meta.ts
function managementGetOrder2() {
  const payload = { id: ":id" };
  const getRequestOptions = managementGetOrder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/orders/{id}",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function managementListOrders2() {
  const payload = {};
  const getRequestOptions = managementListOrders(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/orders",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function postponeEndDate2() {
  const payload = { id: ":id" };
  const getRequestOptions = postponeEndDate(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v2/orders/{id}",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function cancelOrder2() {
  const payload = { id: ":id" };
  const getRequestOptions = cancelOrder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/orders/{id}/cancel",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function markAsPaid2() {
  const payload = { id: ":id" };
  const getRequestOptions = markAsPaid(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/orders/{id}/mark-as-paid",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function pauseOrder2() {
  const payload = { id: ":id" };
  const getRequestOptions = pauseOrder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/orders/{id}/pause",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function resumeOrder2() {
  const payload = { id: ":id" };
  const getRequestOptions = resumeOrder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/orders/{id}/resume",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function memberGetOrder2() {
  const payload = { id: ":id" };
  const getRequestOptions = memberGetOrder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/member/orders/{id}",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function memberListOrders2() {
  const payload = {};
  const getRequestOptions = memberListOrders(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/member/orders",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function requestCancellation2() {
  const payload = { id: ":id" };
  const getRequestOptions = requestCancellation(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/member/orders/{id}/cancel",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createOnlineOrder2() {
  const payload = {};
  const getRequestOptions = createOnlineOrder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/checkout/orders/online",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createOfflineOrder2() {
  const payload = {};
  const getRequestOptions = createOfflineOrder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/checkout/orders/offline",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getOnlineOrderPreview2() {
  const payload = {};
  const getRequestOptions = getOnlineOrderPreview(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/checkout/orders/preview-online",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getOfflineOrderPreview2() {
  const payload = {};
  const getRequestOptions = getOfflineOrderPreview(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/checkout/orders/preview-offline",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getPricePreview2() {
  const payload = {};
  const getRequestOptions = getPricePreview(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/checkout/orders/price-preview",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  CancellationCauseOriginal,
  CancellationEffectiveAtOriginal,
  OrderMethodOriginal,
  OrderStatusOriginal,
  OrderTypeOriginal,
  PaymentStatusOriginal,
  PeriodUnitOriginal,
  ReasonNotSuspendableOriginal,
  SetOriginal,
  SortOrderOriginal,
  StatusOriginal,
  WebhookIdentityTypeOriginal,
  cancelOrder,
  createOfflineOrder,
  createOnlineOrder,
  getOfflineOrderPreview,
  getOnlineOrderPreview,
  getPricePreview,
  managementGetOrder,
  managementListOrders,
  markAsPaid,
  memberGetOrder,
  memberListOrders,
  pauseOrder,
  postponeEndDate,
  requestCancellation,
  resumeOrder
});
