// fp-ts-esm/lib/Applicative.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getApplicativeMonoid = getApplicativeMonoid;
exports.getApplicativeComposition = getApplicativeComposition;
/**
 * The `Applicative` type class extends the `Apply` type class with a `of` function, which can be used to create values
 * of type `f a` from values of type `a`.
 *
 * Where `Apply` provides the ability to lift functions of two or more arguments to functions whose arguments are
 * wrapped using `f`, and `Functor` provides the ability to lift functions of one argument, `pure` can be seen as the
 * function which lifts functions of _zero_ arguments. That is, `Applicative` functors support a lifting operation for
 * any number of function arguments.
 *
 * Instances must satisfy the following laws in addition to the `Apply` laws:
 *
 * 1. Identity: `A.ap(A.of(a => a), fa) <-> fa`
 * 2. Homomorphism: `A.ap(A.of(ab), A.of(a)) <-> A.of(ab(a))`
 * 3. Interchange: `A.ap(fab, A.of(a)) <-> A.ap(A.of(ab => ab(a)), fab)`
 *
 * Note. `Functor`'s `map` can be derived: `A.map(x, f) = A.ap(A.of(f), x)`
 *
 * @since 2.0.0
 */
const Apply_js_1 = require("./Apply.js");
const function_js_1 = require("./function.js");
const Functor_js_1 = require("./Functor.js");
function getApplicativeMonoid(F) {
    const f = (0, Apply_js_1.getApplySemigroup)(F);
    return (M) => ({
        concat: f(M).concat,
        empty: F.of(M.empty)
    });
}
/** @deprecated */
function getApplicativeComposition(F, G) {
    const map = (0, Functor_js_1.getFunctorComposition)(F, G).map;
    const _ap = (0, Apply_js_1.ap)(F, G);
    return {
        map,
        of: (a) => F.of(G.of(a)),
        ap: (fgab, fga) => (0, function_js_1.pipe)(fgab, _ap(fga))
    };
}
