// fp-ts-esm/lib/Const.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.const_ = exports.Bifunctor = exports.Contravariant = exports.flap = exports.Functor = exports.URI = exports.mapLeft = exports.bimap = exports.map = exports.contramap = exports.getBooleanAlgebra = exports.getHeytingAlgebra = exports.getRing = exports.getSemiring = exports.getMonoid = exports.getSemigroup = exports.getBounded = exports.getOrd = exports.getEq = exports.make = void 0;
exports.getShow = getShow;
exports.getApply = getApply;
exports.getApplicative = getApplicative;
const function_js_1 = require("./function.js");
const Functor_js_1 = require("./Functor.js");
/**
 * @category constructors
 * @since 2.0.0
 */
exports.make = function_js_1.unsafeCoerce;
/**
 * @category instances
 * @since 2.0.0
 */
function getShow(S) {
    return {
        show: (c) => `make(${S.show(c)})`
    };
}
/**
 * @category instances
 * @since 2.0.0
 */
exports.getEq = function_js_1.identity;
/**
 * @category instances
 * @since 2.6.0
 */
exports.getOrd = function_js_1.identity;
/**
 * @category instances
 * @since 2.6.0
 */
exports.getBounded = function_js_1.identity;
/**
 * @category instances
 * @since 2.6.0
 */
exports.getSemigroup = function_js_1.identity;
/**
 * @category instances
 * @since 2.6.0
 */
exports.getMonoid = function_js_1.identity;
/**
 * @category instances
 * @since 2.6.0
 */
exports.getSemiring = function_js_1.identity;
/**
 * @category instances
 * @since 2.6.0
 */
exports.getRing = function_js_1.identity;
/**
 * @category instances
 * @since 2.6.0
 */
exports.getHeytingAlgebra = function_js_1.identity;
/**
 * @category instances
 * @since 2.6.0
 */
exports.getBooleanAlgebra = function_js_1.identity;
/**
 * @category instances
 * @since 2.0.0
 */
function getApply(S) {
    return {
        URI: exports.URI,
        _E: undefined,
        map: _map,
        ap: (fab, fa) => (0, exports.make)(S.concat(fab, fa))
    };
}
/**
 * @category instances
 * @since 2.0.0
 */
function getApplicative(M) {
    const A = getApply(M);
    return {
        URI: exports.URI,
        _E: undefined,
        map: A.map,
        ap: A.ap,
        of: () => (0, exports.make)(M.empty)
    };
}
const _contramap = (fa, f) => (0, function_js_1.pipe)(fa, (0, exports.contramap)(f));
/* istanbul ignore next */
const _map = (fa, f) => (0, function_js_1.pipe)(fa, (0, exports.map)(f));
/* istanbul ignore next */
const _bimap = (fa, f, g) => (0, function_js_1.pipe)(fa, (0, exports.bimap)(f, g));
/* istanbul ignore next */
const _mapLeft = (fa, f) => (0, function_js_1.pipe)(fa, (0, exports.mapLeft)(f));
/**
 * @since 2.0.0
 */
const contramap = () => function_js_1.unsafeCoerce;
exports.contramap = contramap;
/**
 * `map` can be used to turn functions `(a: A) => B` into functions `(fa: F<A>) => F<B>` whose argument and return types
 * use the type constructor `F` to represent some computational context.
 *
 * @category mapping
 * @since 2.0.0
 */
const map = () => function_js_1.unsafeCoerce;
exports.map = map;
/**
 * Map a pair of functions over the two type arguments of the bifunctor.
 *
 * @category mapping
 * @since 2.6.2
 */
const bimap = (f) => (fa) => (0, exports.make)(f(fa));
exports.bimap = bimap;
/**
 * Map a function over the first type argument of a bifunctor.
 *
 * @category error handling
 * @since 2.6.2
 */
const mapLeft = (f) => (fa) => (0, exports.make)(f(fa));
exports.mapLeft = mapLeft;
/**
 * @category type lambdas
 * @since 2.0.0
 */
exports.URI = 'Const';
/**
 * @category instances
 * @since 2.7.0
 */
exports.Functor = {
    URI: exports.URI,
    map: _map
};
/**
 * @category mapping
 * @since 2.10.0
 */
exports.flap = (0, Functor_js_1.flap)(exports.Functor);
/**
 * @category instances
 * @since 2.7.0
 */
exports.Contravariant = {
    URI: exports.URI,
    contramap: _contramap
};
/**
 * @category instances
 * @since 2.7.0
 */
exports.Bifunctor = {
    URI: exports.URI,
    bimap: _bimap,
    mapLeft: _mapLeft
};
// -------------------------------------------------------------------------------------
// deprecated
// -------------------------------------------------------------------------------------
/**
 * This instance is deprecated, use small, specific instances instead.
 * For example if a function needs a `Functor` instance, pass `C.Functor` instead of `C.const_`
 * (where `C` is from `import C from 'fp-ts/Const'`)
 *
 * @category zone of death
 * @since 2.0.0
 * @deprecated
 */
exports.const_ = {
    URI: exports.URI,
    map: _map,
    contramap: _contramap,
    bimap: _bimap,
    mapLeft: _mapLeft
};
