// fp-ts-esm/lib/EitherT.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.right = right;
exports.left = left;
exports.rightF = rightF;
exports.leftF = leftF;
exports.fromNullable = fromNullable;
exports.fromNullableK = fromNullableK;
exports.chainNullableK = chainNullableK;
exports.map = map;
exports.ap = ap;
exports.chain = chain;
exports.flatMap = flatMap;
exports.alt = alt;
exports.bimap = bimap;
exports.mapBoth = mapBoth;
exports.mapLeft = mapLeft;
exports.mapError = mapError;
exports.altValidation = altValidation;
exports.match = match;
exports.matchE = matchE;
exports.getOrElse = getOrElse;
exports.orElse = orElse;
exports.orElseFirst = orElseFirst;
exports.tapError = tapError;
exports.orLeft = orLeft;
exports.swap = swap;
exports.toUnion = toUnion;
exports.getEitherM = getEitherM;
const Apply_js_1 = require("./Apply.js");
const E = __importStar(require("./Either.js"));
const function_js_1 = require("./function.js");
const Functor_js_1 = require("./Functor.js");
function right(F) {
    return (0, function_js_1.flow)(E.right, F.of);
}
function left(F) {
    return (0, function_js_1.flow)(E.left, F.of);
}
function rightF(F) {
    return (fa) => F.map(fa, E.right);
}
function leftF(F) {
    return (fe) => F.map(fe, E.left);
}
function fromNullable(F) {
    return (e) => (0, function_js_1.flow)(E.fromNullable(e), F.of);
}
function fromNullableK(F) {
    const fromNullableF = fromNullable(F);
    return (e) => {
        const fromNullableFE = fromNullableF(e);
        return (f) => (0, function_js_1.flow)(f, fromNullableFE);
    };
}
function chainNullableK(M) {
    const chainM = chain(M);
    const fromNullableKM = fromNullableK(M);
    return (e) => {
        const fromNullableKMe = fromNullableKM(e);
        return (f) => chainM(fromNullableKMe(f));
    };
}
function map(F) {
    return (0, Functor_js_1.map)(F, E.Functor);
}
function ap(F) {
    return (0, Apply_js_1.ap)(F, E.Apply);
}
function chain(M) {
    const flatMapM = flatMap(M);
    return (f) => (ma) => flatMapM(ma, f);
}
/** @internal */
function flatMap(M) {
    return (ma, f) => M.chain(ma, (e) => (E.isLeft(e) ? M.of(e) : f(e.right)));
}
function alt(M) {
    return (second) => (first) => M.chain(first, (e) => (E.isLeft(e) ? second() : M.of(e)));
}
function bimap(F) {
    const mapBothF = mapBoth(F);
    return (f, g) => (self) => mapBothF(self, f, g);
}
/** @internal */
function mapBoth(F) {
    return (self, f, g) => F.map(self, E.bimap(f, g));
}
function mapLeft(F) {
    const mapErrorF = mapError(F);
    return (f) => (self) => mapErrorF(self, f);
}
/** @internal */
function mapError(F) {
    return (self, f) => F.map(self, E.mapLeft(f));
}
function altValidation(M, S) {
    return (second) => (first) => M.chain(first, E.match((e1) => M.map(second(), E.mapLeft((e2) => S.concat(e1, e2))), right(M)));
}
function match(F) {
    return (onLeft, onRight) => (ma) => F.map(ma, E.match(onLeft, onRight));
}
function matchE(M) {
    return (onLeft, onRight) => (ma) => M.chain(ma, E.match(onLeft, onRight));
}
function getOrElse(M) {
    return (onLeft) => (ma) => M.chain(ma, E.match(onLeft, M.of));
}
function orElse(M) {
    return (onLeft) => (ma) => M.chain(ma, (e) => (E.isLeft(e) ? onLeft(e.left) : M.of(e)));
}
function orElseFirst(M) {
    const tapErrorM = tapError(M);
    return (onLeft) => (ma) => tapErrorM(ma, onLeft);
}
/** @internal */
function tapError(M) {
    const orElseM = orElse(M);
    return (ma, onLeft) => (0, function_js_1.pipe)(ma, orElseM((e) => M.map(onLeft(e), (eb) => (E.isLeft(eb) ? eb : E.left(e)))));
}
function orLeft(M) {
    return (onLeft) => (ma) => M.chain(ma, E.match((e) => M.map(onLeft(e), E.left), (a) => M.of(E.right(a))));
}
function swap(F) {
    return (ma) => F.map(ma, E.swap);
}
function toUnion(F) {
    return (fa) => F.map(fa, E.toUnion);
}
/** @deprecated  */
/* istanbul ignore next */
function getEitherM(M) {
    const _ap = ap(M);
    const _map = map(M);
    const _chain = chain(M);
    const _alt = alt(M);
    const _bimap = bimap(M);
    const _mapLeft = mapLeft(M);
    const _fold = matchE(M);
    const _getOrElse = getOrElse(M);
    const _orElse = orElse(M);
    return {
        map: (fa, f) => (0, function_js_1.pipe)(fa, _map(f)),
        ap: (fab, fa) => (0, function_js_1.pipe)(fab, _ap(fa)),
        of: right(M),
        chain: (ma, f) => (0, function_js_1.pipe)(ma, _chain(f)),
        alt: (fa, that) => (0, function_js_1.pipe)(fa, _alt(that)),
        bimap: (fea, f, g) => (0, function_js_1.pipe)(fea, _bimap(f, g)),
        mapLeft: (fea, f) => (0, function_js_1.pipe)(fea, _mapLeft(f)),
        fold: (fa, onLeft, onRight) => (0, function_js_1.pipe)(fa, _fold(onLeft, onRight)),
        getOrElse: (fa, onLeft) => (0, function_js_1.pipe)(fa, _getOrElse(onLeft)),
        orElse: (fa, f) => (0, function_js_1.pipe)(fa, _orElse(f)),
        swap: swap(M),
        rightM: rightF(M),
        leftM: leftF(M),
        left: left(M)
    };
}
