// fp-ts-esm/lib/Filterable.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.filter = filter;
exports.filterMap = filterMap;
exports.partition = partition;
exports.partitionMap = partitionMap;
exports.getFilterableComposition = getFilterableComposition;
/**
 * `Filterable` represents data structures which can be _partitioned_/_filtered_.
 *
 * Adapted from https://github.com/LiamGoodacre/purescript-filterable/blob/master/src/Data/Filterable.purs
 *
 * @since 2.0.0
 */
const Compactable_js_1 = require("./Compactable.js");
const function_js_1 = require("./function.js");
const Functor_js_1 = require("./Functor.js");
const Option_js_1 = require("./Option.js");
const Predicate_js_1 = require("./Predicate.js");
const Separated_js_1 = require("./Separated.js");
function filter(F, G) {
    return (predicate) => (fga) => F.map(fga, (ga) => G.filter(ga, predicate));
}
function filterMap(F, G) {
    return (f) => (fga) => F.map(fga, (ga) => G.filterMap(ga, f));
}
function partition(F, G) {
    const _filter = filter(F, G);
    return (predicate) => {
        const left = _filter((0, Predicate_js_1.not)(predicate));
        const right = _filter(predicate);
        return (fgb) => (0, Separated_js_1.separated)(left(fgb), right(fgb));
    };
}
function partitionMap(F, G) {
    const _filterMap = filterMap(F, G);
    return (f) => (fga) => (0, Separated_js_1.separated)((0, function_js_1.pipe)(fga, _filterMap((a) => (0, Option_js_1.getLeft)(f(a)))), (0, function_js_1.pipe)(fga, _filterMap((a) => (0, Option_js_1.getRight)(f(a)))));
}
/** @deprecated */
function getFilterableComposition(F, G) {
    const map = (0, Functor_js_1.getFunctorComposition)(F, G).map;
    const _compact = (0, Compactable_js_1.compact)(F, G);
    const _separate = (0, Compactable_js_1.separate)(F, G, G);
    const _filter = filter(F, G);
    const _filterMap = filterMap(F, G);
    const _partition = partition(F, G);
    const _partitionMap = partitionMap(F, G);
    return {
        map,
        compact: _compact,
        separate: _separate,
        filter: (fga, f) => (0, function_js_1.pipe)(fga, _filter(f)),
        filterMap: (fga, f) => (0, function_js_1.pipe)(fga, _filterMap(f)),
        partition: (fga, p) => (0, function_js_1.pipe)(fga, _partition(p)),
        partitionMap: (fga, f) => (0, function_js_1.pipe)(fga, _partitionMap(f))
    };
}
