// fp-ts-esm/lib/FromEither.js
"use strict";
/**
 * The `FromEither` type class represents those data types which support errors.
 *
 * @since 2.10.0
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.fromOption = fromOption;
exports.fromPredicate = fromPredicate;
exports.fromOptionK = fromOptionK;
exports.chainOptionK = chainOptionK;
exports.fromEitherK = fromEitherK;
exports.chainEitherK = chainEitherK;
exports.chainFirstEitherK = chainFirstEitherK;
exports.filterOrElse = filterOrElse;
exports.tapEither = tapEither;
const Chain_js_1 = require("./Chain.js");
const function_js_1 = require("./function.js");
const _ = __importStar(require("./internal.js"));
function fromOption(F) {
    return (onNone) => (ma) => F.fromEither(_.isNone(ma) ? _.left(onNone()) : _.right(ma.value));
}
function fromPredicate(F) {
    return (predicate, onFalse) => (a) => F.fromEither(predicate(a) ? _.right(a) : _.left(onFalse(a)));
}
function fromOptionK(F) {
    const fromOptionF = fromOption(F);
    return (onNone) => {
        const from = fromOptionF(onNone);
        return (f) => (0, function_js_1.flow)(f, from);
    };
}
function chainOptionK(F, M) {
    const fromOptionKF = fromOptionK(F);
    return (onNone) => {
        const from = fromOptionKF(onNone);
        return (f) => (ma) => M.chain(ma, from(f));
    };
}
function fromEitherK(F) {
    return (f) => (0, function_js_1.flow)(f, F.fromEither);
}
function chainEitherK(F, M) {
    const fromEitherKF = fromEitherK(F);
    return (f) => (ma) => M.chain(ma, fromEitherKF(f));
}
function chainFirstEitherK(F, M) {
    const tapEitherM = tapEither(F, M);
    return (f) => (ma) => tapEitherM(ma, f);
}
function filterOrElse(F, M) {
    return (predicate, onFalse) => (ma) => M.chain(ma, (a) => F.fromEither(predicate(a) ? _.right(a) : _.left(onFalse(a))));
}
/** @internal */
function tapEither(F, M) {
    const fromEither = fromEitherK(F);
    const tapM = (0, Chain_js_1.tap)(M);
    return (self, f) => tapM(self, fromEither(f));
}
