// fp-ts-esm/lib/FromReader.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ask = ask;
exports.asks = asks;
exports.fromReaderK = fromReaderK;
exports.chainReaderK = chainReaderK;
exports.chainFirstReaderK = chainFirstReaderK;
exports.tapReader = tapReader;
/**
 * Lift a computation from the `Reader` monad.
 *
 * @since 2.11.0
 */
const Chain_js_1 = require("./Chain.js");
const function_js_1 = require("./function.js");
const R = __importStar(require("./Reader.js"));
function ask(F) {
    return () => F.fromReader(R.ask());
}
function asks(F) {
    return F.fromReader;
}
function fromReaderK(F) {
    return (f) => (0, function_js_1.flow)(f, F.fromReader);
}
function chainReaderK(F, M) {
    const fromReaderKF = fromReaderK(F);
    return (f) => (ma) => M.chain(ma, fromReaderKF(f));
}
function chainFirstReaderK(F, M) {
    const tapM = tapReader(F, M);
    return (f) => (self) => tapM(self, f);
}
/** @internal */
function tapReader(F, M) {
    const tapM = (0, Chain_js_1.tap)(M);
    return (self, f) => tapM(self, (0, function_js_1.flow)(f, F.fromReader));
}
