// fp-ts-esm/lib/Functor.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.map = map;
exports.flap = flap;
exports.bindTo = bindTo;
exports.let = let_;
exports.getFunctorComposition = getFunctorComposition;
exports.as = as;
exports.asUnit = asUnit;
/**
 * A `Functor` is a type constructor which supports a mapping operation `map`.
 *
 * `map` can be used to turn functions `a -> b` into functions `f a -> f b` whose argument and return types use the type
 * constructor `f` to represent some computational context.
 *
 * Instances must satisfy the following laws:
 *
 * 1. Identity: `F.map(fa, a => a) <-> fa`
 * 2. Composition: `F.map(fa, a => bc(ab(a))) <-> F.map(F.map(fa, ab), bc)`
 *
 * @since 2.0.0
 */
const function_js_1 = require("./function.js");
function map(F, G) {
    return (f) => (fa) => F.map(fa, (ga) => G.map(ga, f));
}
function flap(F) {
    return (a) => (fab) => F.map(fab, (f) => f(a));
}
function bindTo(F) {
    return (name) => (fa) => F.map(fa, (a) => ({ [name]: a }));
}
function let_(F) {
    return (name, f) => (fa) => F.map(fa, (a) => Object.assign({}, a, { [name]: f(a) }));
}
/** @deprecated */
function getFunctorComposition(F, G) {
    const _map = map(F, G);
    return {
        map: (fga, f) => (0, function_js_1.pipe)(fga, _map(f))
    };
}
/** @internal */
function as(F) {
    return (self, b) => F.map(self, () => b);
}
/** @internal */
function asUnit(F) {
    const asM = as(F);
    return (self) => asM(self, undefined);
}
