// fp-ts-esm/lib/index.js
"use strict";
/**
 * @since 2.0.0
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ioOption = exports.ioEither = exports.io = exports.invariant = exports.identity = exports.hkt = exports.heytingAlgebra = exports.group = exports.functorWithIndex = exports.functor = exports.function = exports.fromThese = exports.fromTask = exports.fromState = exports.fromReader = exports.fromIO = exports.fromEither = exports.foldableWithIndex = exports.foldable = exports.filterableWithIndex = exports.filterable = exports.field = exports.extend = exports.endomorphism = exports.eitherT = exports.either = exports.distributiveLattice = exports.date = exports.contravariant = exports.const = exports.console = exports.compactable = exports.comonad = exports.choice = exports.chainRec = exports.chain = exports.category = exports.boundedMeetSemilattice = exports.boundedLattice = exports.boundedJoinSemilattice = exports.boundedDistributiveLattice = exports.bounded = exports.booleanAlgebra = exports.boolean = exports.bifunctor = exports.array = exports.apply = exports.applicative = exports.alternative = exports.alt = void 0;
exports.store = exports.stateT = exports.stateReaderTaskEither = exports.state = exports.show = exports.eq = exports.set = exports.separated = exports.semiring = exports.semigroupoid = exports.semigroup = exports.ring = exports.refinement = exports.record = exports.readerTask = exports.readonlyTuple = exports.readonlySet = exports.readonlyRecord = exports.readonlyNonEmptyArray = exports.readonlyMap = exports.readonlyArray = exports.readerTaskEither = exports.readerT = exports.readerIO = exports.readerEither = exports.reader = exports.random = exports.profunctor = exports.predicate = exports.pointed = exports.pipeable = exports.ordering = exports.ord = exports.optionT = exports.option = exports.number = exports.nonEmptyArray = exports.naturalTransformation = exports.monoid = exports.monadThrow = exports.monadTask = exports.monadIO = exports.monad = exports.meetSemilattice = exports.map = exports.magma = exports.lattice = exports.json = exports.joinSemilattice = exports.ioRef = void 0;
exports.zero = exports.writerT = exports.writer = exports.witherable = exports.void = exports.validationT = exports.unfoldable = exports.tuple = exports.tree = exports.traversableWithIndex = exports.traversable = exports.traced = exports.theseT = exports.these = exports.taskThese = exports.taskOption = exports.taskEither = exports.task = exports.struct = exports.strong = exports.string = void 0;
const alt = __importStar(require("./Alt.js"));
exports.alt = alt;
const alternative = __importStar(require("./Alternative.js"));
exports.alternative = alternative;
const applicative = __importStar(require("./Applicative.js"));
exports.applicative = applicative;
const apply = __importStar(require("./Apply.js"));
exports.apply = apply;
const array = __importStar(require("./Array.js"));
exports.array = array;
const bifunctor = __importStar(require("./Bifunctor.js"));
exports.bifunctor = bifunctor;
const boolean = __importStar(require("./boolean.js"));
exports.boolean = boolean;
const booleanAlgebra = __importStar(require("./BooleanAlgebra.js"));
exports.booleanAlgebra = booleanAlgebra;
const bounded = __importStar(require("./Bounded.js"));
exports.bounded = bounded;
const boundedDistributiveLattice = __importStar(require("./BoundedDistributiveLattice.js"));
exports.boundedDistributiveLattice = boundedDistributiveLattice;
const boundedJoinSemilattice = __importStar(require("./BoundedJoinSemilattice.js"));
exports.boundedJoinSemilattice = boundedJoinSemilattice;
const boundedLattice = __importStar(require("./BoundedLattice.js"));
exports.boundedLattice = boundedLattice;
const boundedMeetSemilattice = __importStar(require("./BoundedMeetSemilattice.js"));
exports.boundedMeetSemilattice = boundedMeetSemilattice;
const category = __importStar(require("./Category.js"));
exports.category = category;
const chain = __importStar(require("./Chain.js"));
exports.chain = chain;
const chainRec = __importStar(require("./ChainRec.js"));
exports.chainRec = chainRec;
const choice = __importStar(require("./Choice.js"));
exports.choice = choice;
const comonad = __importStar(require("./Comonad.js"));
exports.comonad = comonad;
const compactable = __importStar(require("./Compactable.js"));
exports.compactable = compactable;
const console = __importStar(require("./Console.js"));
exports.console = console;
const const_ = __importStar(require("./Const.js"));
exports.const = const_;
const contravariant = __importStar(require("./Contravariant.js"));
exports.contravariant = contravariant;
const date = __importStar(require("./Date.js"));
exports.date = date;
const distributiveLattice = __importStar(require("./DistributiveLattice.js"));
exports.distributiveLattice = distributiveLattice;
const either = __importStar(require("./Either.js"));
exports.either = either;
const eitherT = __importStar(require("./EitherT.js"));
exports.eitherT = eitherT;
const endomorphism = __importStar(require("./Endomorphism.js"));
exports.endomorphism = endomorphism;
const eq = __importStar(require("./Eq.js"));
exports.eq = eq;
const extend = __importStar(require("./Extend.js"));
exports.extend = extend;
const field = __importStar(require("./Field.js"));
exports.field = field;
const filterable = __importStar(require("./Filterable.js"));
exports.filterable = filterable;
const filterableWithIndex = __importStar(require("./FilterableWithIndex.js"));
exports.filterableWithIndex = filterableWithIndex;
const foldable = __importStar(require("./Foldable.js"));
exports.foldable = foldable;
const foldableWithIndex = __importStar(require("./FoldableWithIndex.js"));
exports.foldableWithIndex = foldableWithIndex;
const fromEither = __importStar(require("./FromEither.js"));
exports.fromEither = fromEither;
const fromIO = __importStar(require("./FromIO.js"));
exports.fromIO = fromIO;
const fromReader = __importStar(require("./FromReader.js"));
exports.fromReader = fromReader;
const fromState = __importStar(require("./FromState.js"));
exports.fromState = fromState;
const fromTask = __importStar(require("./FromTask.js"));
exports.fromTask = fromTask;
const fromThese = __importStar(require("./FromThese.js"));
exports.fromThese = fromThese;
const function_ = __importStar(require("./function.js"));
exports.function = function_;
const functor = __importStar(require("./Functor.js"));
exports.functor = functor;
const functorWithIndex = __importStar(require("./FunctorWithIndex.js"));
exports.functorWithIndex = functorWithIndex;
const group = __importStar(require("./Group.js"));
exports.group = group;
const heytingAlgebra = __importStar(require("./HeytingAlgebra.js"));
exports.heytingAlgebra = heytingAlgebra;
const hkt = __importStar(require("./HKT.js"));
exports.hkt = hkt;
const identity = __importStar(require("./Identity.js"));
exports.identity = identity;
const invariant = __importStar(require("./Invariant.js"));
exports.invariant = invariant;
const io = __importStar(require("./IO.js"));
exports.io = io;
const ioEither = __importStar(require("./IOEither.js"));
exports.ioEither = ioEither;
const ioOption = __importStar(require("./IOOption.js"));
exports.ioOption = ioOption;
const ioRef = __importStar(require("./IORef.js"));
exports.ioRef = ioRef;
const joinSemilattice = __importStar(require("./JoinSemilattice.js"));
exports.joinSemilattice = joinSemilattice;
const json = __importStar(require("./Json.js"));
exports.json = json;
const lattice = __importStar(require("./Lattice.js"));
exports.lattice = lattice;
const magma = __importStar(require("./Magma.js"));
exports.magma = magma;
const map = __importStar(require("./Map.js"));
exports.map = map;
const meetSemilattice = __importStar(require("./MeetSemilattice.js"));
exports.meetSemilattice = meetSemilattice;
const monad = __importStar(require("./Monad.js"));
exports.monad = monad;
const monadIO = __importStar(require("./MonadIO.js"));
exports.monadIO = monadIO;
const monadTask = __importStar(require("./MonadTask.js"));
exports.monadTask = monadTask;
const monadThrow = __importStar(require("./MonadThrow.js"));
exports.monadThrow = monadThrow;
const monoid = __importStar(require("./Monoid.js"));
exports.monoid = monoid;
const naturalTransformation = __importStar(require("./NaturalTransformation.js"));
exports.naturalTransformation = naturalTransformation;
const nonEmptyArray = __importStar(require("./NonEmptyArray.js"));
exports.nonEmptyArray = nonEmptyArray;
const number = __importStar(require("./number.js"));
exports.number = number;
const option = __importStar(require("./Option.js"));
exports.option = option;
const optionT = __importStar(require("./OptionT.js"));
exports.optionT = optionT;
const ord = __importStar(require("./Ord.js"));
exports.ord = ord;
const ordering = __importStar(require("./Ordering.js"));
exports.ordering = ordering;
const pipeable = __importStar(require("./pipeable.js"));
exports.pipeable = pipeable;
const pointed = __importStar(require("./Pointed.js"));
exports.pointed = pointed;
const predicate = __importStar(require("./Predicate.js"));
exports.predicate = predicate;
const profunctor = __importStar(require("./Profunctor.js"));
exports.profunctor = profunctor;
const random = __importStar(require("./Random.js"));
exports.random = random;
const reader = __importStar(require("./Reader.js"));
exports.reader = reader;
const readerEither = __importStar(require("./ReaderEither.js"));
exports.readerEither = readerEither;
const readerIO = __importStar(require("./ReaderIO.js"));
exports.readerIO = readerIO;
const readerT = __importStar(require("./ReaderT.js"));
exports.readerT = readerT;
const readerTask = __importStar(require("./ReaderTask.js"));
exports.readerTask = readerTask;
const readerTaskEither = __importStar(require("./ReaderTaskEither.js"));
exports.readerTaskEither = readerTaskEither;
const readonlyArray = __importStar(require("./ReadonlyArray.js"));
exports.readonlyArray = readonlyArray;
const readonlyMap = __importStar(require("./ReadonlyMap.js"));
exports.readonlyMap = readonlyMap;
const readonlyNonEmptyArray = __importStar(require("./ReadonlyNonEmptyArray.js"));
exports.readonlyNonEmptyArray = readonlyNonEmptyArray;
const readonlyRecord = __importStar(require("./ReadonlyRecord.js"));
exports.readonlyRecord = readonlyRecord;
const readonlySet = __importStar(require("./ReadonlySet.js"));
exports.readonlySet = readonlySet;
const readonlyTuple = __importStar(require("./ReadonlyTuple.js"));
exports.readonlyTuple = readonlyTuple;
const record = __importStar(require("./Record.js"));
exports.record = record;
const refinement = __importStar(require("./Refinement.js"));
exports.refinement = refinement;
const ring = __importStar(require("./Ring.js"));
exports.ring = ring;
const semigroup = __importStar(require("./Semigroup.js"));
exports.semigroup = semigroup;
const semigroupoid = __importStar(require("./Semigroupoid.js"));
exports.semigroupoid = semigroupoid;
const semiring = __importStar(require("./Semiring.js"));
exports.semiring = semiring;
const separated = __importStar(require("./Separated.js"));
exports.separated = separated;
const set = __importStar(require("./Set.js"));
exports.set = set;
const show = __importStar(require("./Show.js"));
exports.show = show;
const state = __importStar(require("./State.js"));
exports.state = state;
const stateReaderTaskEither = __importStar(require("./StateReaderTaskEither.js"));
exports.stateReaderTaskEither = stateReaderTaskEither;
const stateT = __importStar(require("./StateT.js"));
exports.stateT = stateT;
const store = __importStar(require("./Store.js"));
exports.store = store;
const string = __importStar(require("./string.js"));
exports.string = string;
const strong = __importStar(require("./Strong.js"));
exports.strong = strong;
const struct = __importStar(require("./struct.js"));
exports.struct = struct;
const task = __importStar(require("./Task.js"));
exports.task = task;
const taskEither = __importStar(require("./TaskEither.js"));
exports.taskEither = taskEither;
const taskOption = __importStar(require("./TaskOption.js"));
exports.taskOption = taskOption;
const taskThese = __importStar(require("./TaskThese.js"));
exports.taskThese = taskThese;
const these = __importStar(require("./These.js"));
exports.these = these;
const theseT = __importStar(require("./TheseT.js"));
exports.theseT = theseT;
const traced = __importStar(require("./Traced.js"));
exports.traced = traced;
const traversable = __importStar(require("./Traversable.js"));
exports.traversable = traversable;
const traversableWithIndex = __importStar(require("./TraversableWithIndex.js"));
exports.traversableWithIndex = traversableWithIndex;
const tree = __importStar(require("./Tree.js"));
exports.tree = tree;
const tuple = __importStar(require("./Tuple.js"));
exports.tuple = tuple;
const unfoldable = __importStar(require("./Unfoldable.js"));
exports.unfoldable = unfoldable;
const validationT = __importStar(require("./ValidationT.js"));
exports.validationT = validationT;
const void_ = __importStar(require("./void.js"));
exports.void = void_;
const witherable = __importStar(require("./Witherable.js"));
exports.witherable = witherable;
const writer = __importStar(require("./Writer.js"));
exports.writer = writer;
const writerT = __importStar(require("./WriterT.js"));
exports.writerT = writerT;
const zero = __importStar(require("./Zero.js"));
exports.zero = zero;
