// fp-ts-esm/lib/internal.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.flatMapReader = exports.flatMapTask = exports.flatMapIO = exports.flatMapEither = exports.flatMapOption = exports.flatMapNullable = exports.liftOption = exports.liftNullable = exports.fromReadonlyNonEmptyArray = exports.has = exports.emptyRecord = exports.emptyReadonlyArray = exports.tail = exports.head = exports.isNonEmpty = exports.singleton = exports.right = exports.left = exports.isRight = exports.isLeft = exports.some = exports.none = exports.isSome = exports.isNone = void 0;
const function_js_1 = require("./function.js");
// -------------------------------------------------------------------------------------
// Option
// -------------------------------------------------------------------------------------
/** @internal */
const isNone = (fa) => fa._tag === 'None';
exports.isNone = isNone;
/** @internal */
const isSome = (fa) => fa._tag === 'Some';
exports.isSome = isSome;
/** @internal */
exports.none = { _tag: 'None' };
/** @internal */
const some = (a) => ({ _tag: 'Some', value: a });
exports.some = some;
// -------------------------------------------------------------------------------------
// Either
// -------------------------------------------------------------------------------------
/** @internal */
const isLeft = (ma) => ma._tag === 'Left';
exports.isLeft = isLeft;
/** @internal */
const isRight = (ma) => ma._tag === 'Right';
exports.isRight = isRight;
/** @internal */
const left = (e) => ({ _tag: 'Left', left: e });
exports.left = left;
/** @internal */
const right = (a) => ({ _tag: 'Right', right: a });
exports.right = right;
// -------------------------------------------------------------------------------------
// ReadonlyNonEmptyArray
// -------------------------------------------------------------------------------------
/** @internal */
const singleton = (a) => [a];
exports.singleton = singleton;
/** @internal */
const isNonEmpty = (as) => as.length > 0;
exports.isNonEmpty = isNonEmpty;
/** @internal */
const head = (as) => as[0];
exports.head = head;
/** @internal */
const tail = (as) => as.slice(1);
exports.tail = tail;
// -------------------------------------------------------------------------------------
// empty
// -------------------------------------------------------------------------------------
/** @internal */
exports.emptyReadonlyArray = [];
/** @internal */
exports.emptyRecord = {};
// -------------------------------------------------------------------------------------
// Record
// -------------------------------------------------------------------------------------
/** @internal */
exports.has = Object.prototype.hasOwnProperty;
// -------------------------------------------------------------------------------------
// NonEmptyArray
// -------------------------------------------------------------------------------------
/** @internal */
const fromReadonlyNonEmptyArray = (as) => [as[0], ...as.slice(1)];
exports.fromReadonlyNonEmptyArray = fromReadonlyNonEmptyArray;
/** @internal */
const liftNullable = (F) => (f, onNullable) => (...a) => {
    const o = f(...a);
    return F.fromEither(o == null ? (0, exports.left)(onNullable(...a)) : (0, exports.right)(o));
};
exports.liftNullable = liftNullable;
/** @internal */
const liftOption = (F) => (f, onNone) => (...a) => {
    const o = f(...a);
    return F.fromEither((0, exports.isNone)(o) ? (0, exports.left)(onNone(...a)) : (0, exports.right)(o.value));
};
exports.liftOption = liftOption;
/** @internal */
const flatMapNullable = (F, M) => {
    return /*#__PURE__*/ (0, function_js_1.dual)(3, (self, f, onNullable) => M.flatMap(self, (0, exports.liftNullable)(F)(f, onNullable)));
};
exports.flatMapNullable = flatMapNullable;
/** @internal */
const flatMapOption = (F, M) => {
    return /*#__PURE__*/ (0, function_js_1.dual)(3, (self, f, onNone) => M.flatMap(self, (0, exports.liftOption)(F)(f, onNone)));
};
exports.flatMapOption = flatMapOption;
/** @internal */
const flatMapEither = (F, M) => {
    return /*#__PURE__*/ (0, function_js_1.dual)(2, (self, f) => M.flatMap(self, (a) => F.fromEither(f(a))));
};
exports.flatMapEither = flatMapEither;
/** @internal */
const flatMapIO = (F, M) => {
    return /*#__PURE__*/ (0, function_js_1.dual)(2, (self, f) => M.flatMap(self, (a) => F.fromIO(f(a))));
};
exports.flatMapIO = flatMapIO;
/** @internal */
const flatMapTask = (F, M) => {
    return /*#__PURE__*/ (0, function_js_1.dual)(2, (self, f) => M.flatMap(self, (a) => F.fromTask(f(a))));
};
exports.flatMapTask = flatMapTask;
/** @internal */
const flatMapReader = (F, M) => {
    return /*#__PURE__*/ (0, function_js_1.dual)(2, (self, f) => M.flatMap(self, (a) => F.fromReader(f(a))));
};
exports.flatMapReader = flatMapReader;
