// fp-ts-esm/lib/OptionT.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.some = some;
exports.zero = zero;
exports.fromF = fromF;
exports.fromNullable = fromNullable;
exports.fromNullableK = fromNullableK;
exports.chainNullableK = chainNullableK;
exports.fromOptionK = fromOptionK;
exports.chainOptionK = chainOptionK;
exports.fromPredicate = fromPredicate;
exports.fromEither = fromEither;
exports.match = match;
exports.matchE = matchE;
exports.getOrElse = getOrElse;
exports.map = map;
exports.ap = ap;
exports.chain = chain;
exports.flatMap = flatMap;
exports.alt = alt;
exports.getOptionM = getOptionM;
const Apply_js_1 = require("./Apply.js");
const function_js_1 = require("./function.js");
const Functor_js_1 = require("./Functor.js");
const O = __importStar(require("./Option.js"));
function some(F) {
    return (0, function_js_1.flow)(O.some, F.of);
}
function zero(F) {
    return (0, function_js_1.constant)(F.of(O.none));
}
function fromF(F) {
    return (ma) => F.map(ma, O.some);
}
function fromNullable(F) {
    return (0, function_js_1.flow)(O.fromNullable, F.of);
}
function fromNullableK(F) {
    const fromNullableF = fromNullable(F);
    return (f) => (0, function_js_1.flow)(f, fromNullableF);
}
function chainNullableK(M) {
    const chainM = chain(M);
    const fromNullableKM = fromNullableK(M);
    return (f) => chainM(fromNullableKM(f));
}
function fromOptionK(F) {
    return (f) => (0, function_js_1.flow)(f, F.of);
}
function chainOptionK(M) {
    const chainM = chain(M);
    const fromOptionKM = fromOptionK(M);
    return (f) => chainM(fromOptionKM(f));
}
function fromPredicate(F) {
    return (predicate) => (a) => F.of(O.fromPredicate(predicate)(a));
}
function fromEither(F) {
    return (0, function_js_1.flow)(O.fromEither, F.of);
}
function match(F) {
    return (onNone, onSome) => (ma) => F.map(ma, O.match(onNone, onSome));
}
function matchE(M) {
    return (onNone, onSome) => (ma) => M.chain(ma, O.match(onNone, onSome));
}
function getOrElse(M) {
    return (onNone) => (fa) => M.chain(fa, O.match(onNone, M.of));
}
function map(F) {
    return (0, Functor_js_1.map)(F, O.Functor);
}
function ap(F) {
    return (0, Apply_js_1.ap)(F, O.Apply);
}
function chain(M) {
    const flatMapM = flatMap(M);
    return (f) => (ma) => flatMapM(ma, f);
}
/** @internal */
function flatMap(M) {
    const zeroM = zero(M);
    return (ma, f) => M.chain(ma, O.match(() => zeroM(), f));
}
function alt(M) {
    const _some = some(M);
    return (second) => (first) => M.chain(first, O.match(second, _some));
}
/** @deprecated */
function getOptionM(M) {
    const apM = ap(M);
    const mapM = map(M);
    const chainM = chain(M);
    const altM = alt(M);
    const foldM = matchE(M);
    const getOrElseM = getOrElse(M);
    const zeroM = zero(M);
    return {
        map: (fa, f) => (0, function_js_1.pipe)(fa, mapM(f)),
        ap: (fab, fa) => (0, function_js_1.pipe)(fab, apM(fa)),
        of: some(M),
        chain: (ma, f) => (0, function_js_1.pipe)(ma, chainM(f)),
        alt: (fa, that) => (0, function_js_1.pipe)(fa, altM(that)),
        fold: (fa, onNone, onSome) => (0, function_js_1.pipe)(fa, foldM(onNone, onSome)),
        getOrElse: (fa, onNone) => (0, function_js_1.pipe)(fa, getOrElseM(onNone)),
        fromM: fromF(M),
        none: () => zeroM()
    };
}
