// fp-ts-esm/lib/ReaderT.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.of = of;
exports.map = map;
exports.ap = ap;
exports.chain = chain;
exports.flatMap = flatMap;
exports.fromReader = fromReader;
exports.fromNaturalTransformation = fromNaturalTransformation;
exports.getReaderM = getReaderM;
const function_js_1 = require("./function.js");
function of(F) {
    return (a) => () => F.of(a);
}
function map(F) {
    return (f) => (fa) => (r) => F.map(fa(r), f);
}
function ap(F) {
    return (fa) => (fab) => (r) => F.ap(fab(r), fa(r));
}
function chain(M) {
    const flatMapM = flatMap(M);
    return (f) => (ma) => flatMapM(ma, f);
}
/** @internal */
function flatMap(M) {
    return (ma, f) => (r) => M.chain(ma(r), (a) => f(a)(r));
}
function fromReader(F) {
    return (ma) => (0, function_js_1.flow)(ma, F.of);
}
function fromNaturalTransformation(nt) {
    return (f) => (0, function_js_1.flow)(f, nt);
}
/** @deprecated */
/* istanbul ignore next */
function getReaderM(M) {
    const _ap = ap(M);
    const _map = map(M);
    const _chain = chain(M);
    return {
        map: (fa, f) => (0, function_js_1.pipe)(fa, _map(f)),
        ap: (fab, fa) => (0, function_js_1.pipe)(fab, _ap(fa)),
        of: of(M),
        chain: (ma, f) => (0, function_js_1.pipe)(ma, _chain(f)),
        ask: () => M.of,
        asks: (f) => (0, function_js_1.flow)(f, M.of),
        local: (ma, f) => (q) => ma(f(q)),
        fromReader: fromReader(M),
        fromM: (ma) => () => ma
    };
}
