// fp-ts-esm/lib/ReadonlySet.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.fromArray = exports.getDifferenceMagma = exports.getIntersectionSemigroup = exports.getUnionMonoid = exports.getUnionSemigroup = exports.URI = exports.toReadonlyArray = exports.some = exports.size = exports.isEmpty = exports.empty = exports.compact = exports.toggle = exports.remove = exports.reduceRight = exports.fromReadonlyArray = exports.singleton = exports.fromSet = void 0;
exports.toSet = toSet;
exports.map = map;
exports.chain = chain;
exports.filter = filter;
exports.partition = partition;
exports.union = union;
exports.intersection = intersection;
exports.partitionMap = partitionMap;
exports.difference = difference;
exports.reduce = reduce;
exports.foldMap = foldMap;
exports.insert = insert;
exports.separate = separate;
exports.filterMap = filterMap;
exports.every = every;
exports.isSubset = isSubset;
exports.elem = elem;
exports.getShow = getShow;
exports.getEq = getEq;
const Eq_js_1 = require("./Eq.js");
const function_js_1 = require("./function.js");
const Predicate_js_1 = require("./Predicate.js");
const Separated_js_1 = require("./Separated.js");
/**
 * @category conversions
 * @since 2.5.0
 */
const fromSet = (s) => new Set(s);
exports.fromSet = fromSet;
// -------------------------------------------------------------------------------------
// constructors
// -------------------------------------------------------------------------------------
/**
 * Create a set with one element
 *
 * @category constructors
 * @since 2.5.0
 */
const singleton = (a) => new Set([a]);
exports.singleton = singleton;
/**
 * Create a `ReadonlySet` from a `ReadonlyArray`
 *
 * @category conversions
 * @since 2.10.0
 */
const fromReadonlyArray = (E) => (as) => {
    const len = as.length;
    const out = new Set();
    const has = elem(E);
    for (let i = 0; i < len; i++) {
        const a = as[i];
        if (!has(a, out)) {
            out.add(a);
        }
    }
    return out;
};
exports.fromReadonlyArray = fromReadonlyArray;
/**
 * @category conversions
 * @since 2.5.0
 */
function toSet(s) {
    return new Set(s);
}
/**
 * Projects a Set through a function
 *
 * @since 2.5.0
 */
function map(E) {
    const elemE = elem(E);
    return (f) => (set) => {
        const r = new Set();
        set.forEach((e) => {
            const v = f(e);
            if (!elemE(v, r)) {
                r.add(v);
            }
        });
        return r;
    };
}
/**
 * @since 2.5.0
 */
function chain(E) {
    const elemE = elem(E);
    return (f) => (set) => {
        const r = new Set();
        set.forEach((e) => {
            f(e).forEach((e) => {
                if (!elemE(e, r)) {
                    r.add(e);
                }
            });
        });
        return r;
    };
}
function filter(predicate) {
    return (set) => {
        const values = set.values();
        let e;
        const r = new Set();
        while (!(e = values.next()).done) {
            const a = e.value;
            if (predicate(a)) {
                r.add(a);
            }
        }
        return r;
    };
}
function partition(predicate) {
    return (set) => {
        const values = set.values();
        let e;
        const right = new Set();
        const left = new Set();
        while (!(e = values.next()).done) {
            const a = e.value;
            if (predicate(a)) {
                right.add(a);
            }
            else {
                left.add(a);
            }
        }
        return (0, Separated_js_1.separated)(left, right);
    };
}
function union(E) {
    const elemE = elem(E);
    return (me, that) => {
        if (that === undefined) {
            const unionE = union(E);
            return (that) => unionE(me, that);
        }
        if ((0, exports.isEmpty)(me)) {
            return that;
        }
        if ((0, exports.isEmpty)(that)) {
            return me;
        }
        const r = new Set(me);
        that.forEach((e) => {
            if (!elemE(e, r)) {
                r.add(e);
            }
        });
        return r;
    };
}
function intersection(E) {
    const elemE = elem(E);
    return (me, that) => {
        if (that === undefined) {
            const intersectionE = intersection(E);
            return (that) => intersectionE(that, me);
        }
        if ((0, exports.isEmpty)(me) || (0, exports.isEmpty)(that)) {
            return exports.empty;
        }
        const r = new Set();
        me.forEach((e) => {
            if (elemE(e, that)) {
                r.add(e);
            }
        });
        return r;
    };
}
/**
 * @since 2.5.0
 */
function partitionMap(EB, EC) {
    return (f) => (set) => {
        const values = set.values();
        let e;
        const left = new Set();
        const right = new Set();
        const hasB = elem(EB);
        const hasC = elem(EC);
        while (!(e = values.next()).done) {
            const v = f(e.value);
            switch (v._tag) {
                case 'Left':
                    if (!hasB(v.left, left)) {
                        left.add(v.left);
                    }
                    break;
                case 'Right':
                    if (!hasC(v.right, right)) {
                        right.add(v.right);
                    }
                    break;
            }
        }
        return (0, Separated_js_1.separated)(left, right);
    };
}
function difference(E) {
    const elemE = elem(E);
    return (me, that) => {
        if (that === undefined) {
            const differenceE = difference(E);
            return (that) => differenceE(that, me);
        }
        return filter((a) => !elemE(a, that))(me);
    };
}
/**
 * @since 2.5.0
 */
function reduce(O) {
    const toReadonlyArrayO = (0, exports.toReadonlyArray)(O);
    return (b, f) => (fa) => toReadonlyArrayO(fa).reduce(f, b);
}
/**
 * @since 2.5.0
 */
function foldMap(O, M) {
    const toReadonlyArrayO = (0, exports.toReadonlyArray)(O);
    return (f) => (fa) => toReadonlyArrayO(fa).reduce((b, a) => M.concat(b, f(a)), M.empty);
}
/**
 * @category folding
 * @since 2.11.0
 */
const reduceRight = (O) => {
    const toReadonlyArrayO = (0, exports.toReadonlyArray)(O);
    return (b, f) => (fa) => toReadonlyArrayO(fa).reduceRight((b, a) => f(a, b), b);
};
exports.reduceRight = reduceRight;
/**
 * Insert a value into a set
 *
 * @since 2.5.0
 */
function insert(E) {
    const elemE = elem(E);
    return (a) => (set) => {
        if (!elemE(a)(set)) {
            const r = new Set(set);
            r.add(a);
            return r;
        }
        else {
            return set;
        }
    };
}
/**
 * Delete a value from a set
 *
 * @since 2.5.0
 */
const remove = (E) => (a) => (set) => filter((ax) => !E.equals(a, ax))(set);
exports.remove = remove;
/**
 * Checks an element is a member of a set;
 * If yes, removes the value from the set
 * If no, inserts the value to the set
 *
 * @since 2.10.0
 */
const toggle = (E) => {
    const elemE = elem(E);
    const removeE = (0, exports.remove)(E);
    const insertE = insert(E);
    return (a) => (set) => (elemE(a, set) ? removeE : insertE)(a)(set);
};
exports.toggle = toggle;
/**
 * @since 2.5.0
 */
const compact = (E) => filterMap(E)(function_js_1.identity);
exports.compact = compact;
/**
 * @since 2.5.0
 */
function separate(EE, EA) {
    return (fa) => {
        const elemEE = elem(EE);
        const elemEA = elem(EA);
        const left = new Set();
        const right = new Set();
        fa.forEach((e) => {
            switch (e._tag) {
                case 'Left':
                    if (!elemEE(e.left, left)) {
                        left.add(e.left);
                    }
                    break;
                case 'Right':
                    if (!elemEA(e.right, right)) {
                        right.add(e.right);
                    }
                    break;
            }
        });
        return (0, Separated_js_1.separated)(left, right);
    };
}
/**
 * @since 2.5.0
 */
function filterMap(E) {
    const elemE = elem(E);
    return (f) => (fa) => {
        const r = new Set();
        fa.forEach((a) => {
            const ob = f(a);
            if (ob._tag === 'Some' && !elemE(ob.value, r)) {
                r.add(ob.value);
            }
        });
        return r;
    };
}
// -------------------------------------------------------------------------------------
// utils
// -------------------------------------------------------------------------------------
/**
 * @since 2.5.0
 */
exports.empty = new Set();
/**
 * Test whether a `ReadonlySet` is empty.
 *
 * @since 2.10.0
 */
const isEmpty = (set) => set.size === 0;
exports.isEmpty = isEmpty;
/**
 * Calculate the number of elements in a `ReadonlySet`.
 *
 * @since 2.10.0
 */
const size = (set) => set.size;
exports.size = size;
/**
 * @since 2.5.0
 */
const some = (predicate) => (set) => {
    const values = set.values();
    let e;
    let found = false;
    while (!found && !(e = values.next()).done) {
        found = predicate(e.value);
    }
    return found;
};
exports.some = some;
function every(predicate) {
    return (0, Predicate_js_1.not)((0, exports.some)((0, Predicate_js_1.not)(predicate)));
}
function isSubset(E) {
    const elemE = elem(E);
    return (me, that) => {
        if (that === undefined) {
            const isSubsetE = isSubset(E);
            return (that) => isSubsetE(that, me);
        }
        return every((a) => elemE(a, that))(me);
    };
}
function elem(E) {
    return (a, set) => {
        if (set === undefined) {
            const elemE = elem(E);
            return (set) => elemE(a, set);
        }
        const values = set.values();
        let e;
        let found = false;
        while (!found && !(e = values.next()).done) {
            found = E.equals(a, e.value);
        }
        return found;
    };
}
/**
 * Get a sorted `ReadonlyArray` of the values contained in a `ReadonlySet`.
 *
 * @category conversions
 * @since 2.5.0
 */
const toReadonlyArray = (O) => (set) => {
    const out = [];
    set.forEach((e) => out.push(e));
    return out.sort(O.compare);
};
exports.toReadonlyArray = toReadonlyArray;
/**
 * @category type lambdas
 * @since 2.11.0
 */
exports.URI = 'ReadonlySet';
/**
 * @category instances
 * @since 2.5.0
 */
function getShow(S) {
    return {
        show: (s) => {
            const entries = [];
            s.forEach((a) => {
                entries.push(S.show(a));
            });
            return `new Set([${entries.sort().join(', ')}])`;
        }
    };
}
/**
 * @category instances
 * @since 2.5.0
 */
function getEq(E) {
    const subsetE = isSubset(E);
    return (0, Eq_js_1.fromEquals)((x, y) => subsetE(x, y) && subsetE(y, x));
}
/**
 * @category instances
 * @since 2.11.0
 */
const getUnionSemigroup = (E) => ({
    concat: union(E)
});
exports.getUnionSemigroup = getUnionSemigroup;
/**
 * @category instances
 * @since 2.5.0
 */
const getUnionMonoid = (E) => ({
    concat: (0, exports.getUnionSemigroup)(E).concat,
    empty: exports.empty
});
exports.getUnionMonoid = getUnionMonoid;
/**
 * @category instances
 * @since 2.5.0
 */
const getIntersectionSemigroup = (E) => ({
    concat: intersection(E)
});
exports.getIntersectionSemigroup = getIntersectionSemigroup;
/**
 * @category instances
 * @since 2.11.0
 */
const getDifferenceMagma = (E) => ({
    concat: difference(E)
});
exports.getDifferenceMagma = getDifferenceMagma;
// -------------------------------------------------------------------------------------
// deprecated
// -------------------------------------------------------------------------------------
/**
 * Use [`fromReadonlyArray`](#fromreadonlyarray) instead.
 *
 * @category zone of death
 * @since 2.5.0
 * @deprecated
 */
exports.fromArray = exports.fromReadonlyArray;
