// fp-ts-esm/lib/StateT.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.of = of;
exports.map = map;
exports.ap = ap;
exports.chain = chain;
exports.flatMap = flatMap;
exports.fromState = fromState;
exports.fromF = fromF;
exports.evaluate = evaluate;
exports.execute = execute;
exports.getStateM = getStateM;
const function_js_1 = require("./function.js");
const ReadonlyTuple_js_1 = require("./ReadonlyTuple.js");
function of(F) {
    return (a) => (s) => F.of([a, s]);
}
function map(F) {
    return (f) => (fa) => (s) => F.map(fa(s), ([a, s1]) => [f(a), s1]);
}
function ap(M) {
    return (fa) => (fab) => (s) => M.chain(fab(s), ([f, s]) => M.map(fa(s), ([a, s]) => [f(a), s]));
}
function chain(M) {
    const flatMapM = flatMap(M);
    return (f) => (ma) => flatMapM(ma, f);
}
/** @internal */
function flatMap(M) {
    return (ma, f) => (s) => M.chain(ma(s), ([a, s1]) => f(a)(s1));
}
function fromState(F) {
    return (sa) => (s) => F.of(sa(s));
}
function fromF(F) {
    return (ma) => (s) => F.map(ma, (a) => [a, s]);
}
function evaluate(F) {
    return (s) => (ma) => F.map(ma(s), ([a]) => a);
}
function execute(F) {
    return (s) => (ma) => F.map(ma(s), ReadonlyTuple_js_1.snd);
}
/** @deprecated */
/* istanbul ignore next */
function getStateM(M) {
    const _ap = ap(M);
    const _map = map(M);
    const _chain = chain(M);
    const _evaluate = evaluate(M);
    const _execute = execute(M);
    return {
        map: (fa, f) => (0, function_js_1.pipe)(fa, _map(f)),
        ap: (fab, fa) => (0, function_js_1.pipe)(fab, _ap(fa)),
        of: of(M),
        chain: (ma, f) => (0, function_js_1.pipe)(ma, _chain(f)),
        get: () => (s) => M.of([s, s]),
        put: (s) => () => M.of([undefined, s]),
        modify: (f) => (s) => M.of([undefined, f(s)]),
        gets: (f) => (s) => M.of([f(s), s]),
        fromState: fromState(M),
        fromM: fromF(M),
        evalState: (fa, s) => (0, function_js_1.pipe)(fa, _evaluate(s)),
        execState: (fa, s) => (0, function_js_1.pipe)(fa, _execute(s))
    };
}
