// fp-ts-esm/lib/Store.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.store = exports.Comonad = exports.flap = exports.Functor = exports.URI = exports.map = exports.duplicate = exports.extract = exports.extend = void 0;
exports.seek = seek;
exports.seeks = seeks;
exports.peeks = peeks;
exports.experiment = experiment;
const function_js_1 = require("./function.js");
const Functor_js_1 = require("./Functor.js");
/**
 * Reposition the focus at the specified position
 *
 * @since 2.0.0
 */
function seek(s) {
    return (wa) => ({ peek: wa.peek, pos: s });
}
/**
 * Reposition the focus at the specified position, which depends on the current position
 *
 * @since 2.0.0
 */
function seeks(f) {
    return (wa) => ({ peek: wa.peek, pos: f(wa.pos) });
}
/**
 * Extract a value from a position which depends on the current position
 *
 * @since 2.0.0
 */
function peeks(f) {
    return (wa) => wa.peek(f(wa.pos));
}
function experiment(F) {
    return (f) => (wa) => F.map(f(wa.pos), (s) => wa.peek(s));
}
/* istanbul ignore next */
const _map = (fa, f) => (0, function_js_1.pipe)(fa, (0, exports.map)(f));
/* istanbul ignore next */
const _extend = (wa, f) => (0, function_js_1.pipe)(wa, (0, exports.extend)(f));
/**
 * @since 2.0.0
 */
const extend = (f) => (wa) => ({
    peek: (s) => f({ peek: wa.peek, pos: s }),
    pos: wa.pos
});
exports.extend = extend;
/**
 * @category Extract
 * @since 2.6.2
 */
const extract = (wa) => wa.peek(wa.pos);
exports.extract = extract;
/**
 * @since 2.0.0
 */
exports.duplicate = (0, exports.extend)(function_js_1.identity);
/**
 * `map` can be used to turn functions `(a: A) => B` into functions `(fa: F<A>) => F<B>` whose argument and return types
 * use the type constructor `F` to represent some computational context.
 *
 * @category mapping
 * @since 2.0.0
 */
const map = (f) => (fa) => ({
    peek: (s) => f(fa.peek(s)),
    pos: fa.pos
});
exports.map = map;
/**
 * @category type lambdas
 * @since 2.0.0
 */
exports.URI = 'Store';
/**
 * @category instances
 * @since 2.7.0
 */
exports.Functor = {
    URI: exports.URI,
    map: _map
};
/**
 * @category mapping
 * @since 2.10.0
 */
exports.flap = (0, Functor_js_1.flap)(exports.Functor);
/**
 * @category instances
 * @since 2.7.0
 */
exports.Comonad = {
    URI: exports.URI,
    map: _map,
    extend: _extend,
    extract: exports.extract
};
// -------------------------------------------------------------------------------------
// deprecated
// -------------------------------------------------------------------------------------
/**
 * This instance is deprecated, use small, specific instances instead.
 * For example if a function needs a `Comonad` instance, pass `S.Comonad` instead of `S.store`
 * (where `S` is from `import S from 'fp-ts/Store'`)
 *
 * @category zone of death
 * @since 2.0.0
 * @deprecated
 */
exports.store = exports.Comonad;
