// fp-ts-esm/lib/TheseT.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.right = right;
exports.left = left;
exports.both = both;
exports.rightF = rightF;
exports.leftF = leftF;
exports.map = map;
exports.ap = ap;
exports.chain = chain;
exports.bimap = bimap;
exports.mapLeft = mapLeft;
exports.match = match;
exports.matchE = matchE;
exports.swap = swap;
exports.toTuple2 = toTuple2;
exports.getTheseM = getTheseM;
/**
 * @since 2.4.0
 */
const Apply_js_1 = require("./Apply.js");
const function_js_1 = require("./function.js");
const Functor_js_1 = require("./Functor.js");
const T = __importStar(require("./These.js"));
function right(F) {
    return (0, function_js_1.flow)(T.right, F.of);
}
function left(F) {
    return (0, function_js_1.flow)(T.left, F.of);
}
function both(F) {
    return (0, function_js_1.flow)(T.both, F.of);
}
function rightF(F) {
    return (fa) => F.map(fa, T.right);
}
function leftF(F) {
    return (fe) => F.map(fe, T.left);
}
function map(F) {
    return (0, Functor_js_1.map)(F, T.Functor);
}
function ap(F, S) {
    return (0, Apply_js_1.ap)(F, T.getApply(S));
}
function chain(M, S) {
    const _left = left(M);
    return (f) => (ma) => M.chain(ma, T.match(_left, f, (e1, a) => M.map(f(a), T.match((e2) => T.left(S.concat(e1, e2)), (b) => T.both(e1, b), (e2, b) => T.both(S.concat(e1, e2), b)))));
}
function bimap(F) {
    return (f, g) => (fea) => F.map(fea, T.bimap(f, g));
}
function mapLeft(F) {
    return (f) => (fea) => F.map(fea, T.mapLeft(f));
}
function match(F) {
    return (onLeft, onRight, onBoth) => (ma) => F.map(ma, T.match(onLeft, onRight, onBoth));
}
function matchE(M) {
    return (onLeft, onRight, onBoth) => (ma) => M.chain(ma, T.match(onLeft, onRight, onBoth));
}
function swap(F) {
    return (ma) => F.map(ma, T.swap);
}
function toTuple2(F) {
    return (e, a) => (fa) => F.map(fa, T.toTuple2(e, a));
}
/** @deprecated */
/* istanbul ignore next */
function getTheseM(M) {
    const _map = map(M);
    const _bimap = bimap(M);
    const _mapLeft = mapLeft(M);
    const _fold = matchE(M);
    const _toTuple2 = toTuple2(M);
    const of = right(M);
    const mapT = (fa, f) => (0, function_js_1.pipe)(fa, _map(f));
    return {
        map: mapT,
        bimap: (fea, f, g) => (0, function_js_1.pipe)(fea, _bimap(f, g)),
        mapLeft: (fea, f) => (0, function_js_1.pipe)(fea, _mapLeft(f)),
        fold: (fa, onLeft, onRight, onBoth) => (0, function_js_1.pipe)(fa, _fold(onLeft, onRight, onBoth)),
        swap: swap(M),
        rightM: rightF(M),
        leftM: leftF(M),
        left: left(M),
        right: right(M),
        both: both(M),
        toTuple: (fa, e, a) => (0, function_js_1.pipe)(fa, _toTuple2(() => e, () => a)),
        getMonad: (S) => {
            const _ap = ap(M, S);
            const _chain = chain(M, S);
            return {
                _E: undefined,
                map: mapT,
                of,
                ap: (fab, fa) => (0, function_js_1.pipe)(fab, _ap(fa)),
                chain: (ma, f) => (0, function_js_1.pipe)(ma, _chain(f))
            };
        }
    };
}
