// fp-ts-esm/lib/Traced.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.traced = exports.flap = exports.Functor = exports.URI = exports.map = void 0;
exports.tracks = tracks;
exports.listen = listen;
exports.listens = listens;
exports.censor = censor;
exports.getComonad = getComonad;
const function_js_1 = require("./function.js");
const Functor_js_1 = require("./Functor.js");
// TODO: curry in v3
/**
 * Extracts a value at a relative position which depends on the current value.
 *
 * @since 2.0.0
 */
function tracks(M, f) {
    return (wa) => wa(f(wa(M.empty)));
}
/**
 * Get the current position
 *
 * @since 2.0.0
 */
function listen(wa) {
    return (e) => [wa(e), e];
}
/**
 * Get a value which depends on the current position
 *
 * @since 2.0.0
 */
function listens(f) {
    return (wa) => (e) => [wa(e), f(e)];
}
/**
 * Apply a function to the current position
 *
 * @since 2.0.0
 */
function censor(f) {
    return (wa) => (e) => wa(f(e));
}
/**
 * @category instances
 * @since 2.0.0
 */
function getComonad(monoid) {
    function extend(wa, f) {
        return (p1) => f((p2) => wa(monoid.concat(p1, p2)));
    }
    function extract(wa) {
        return wa(monoid.empty);
    }
    return {
        URI: exports.URI,
        _E: undefined,
        map: _map,
        extend,
        extract
    };
}
/* istanbul ignore next */
const _map = (fa, f) => (0, function_js_1.pipe)(fa, (0, exports.map)(f));
/**
 * `map` can be used to turn functions `(a: A) => B` into functions `(fa: F<A>) => F<B>` whose argument and return types
 * use the type constructor `F` to represent some computational context.
 *
 * @category mapping
 * @since 2.0.0
 */
const map = (f) => (fa) => (p) => f(fa(p));
exports.map = map;
/**
 * @category type lambdas
 * @since 2.0.0
 */
exports.URI = 'Traced';
/**
 * @category instances
 * @since 2.7.0
 */
exports.Functor = {
    URI: exports.URI,
    map: _map
};
/**
 * @category mapping
 * @since 2.10.0
 */
exports.flap = (0, Functor_js_1.flap)(exports.Functor);
// -------------------------------------------------------------------------------------
// deprecated
// -------------------------------------------------------------------------------------
/**
 * Use [`Functor`](#functor) instead.
 *
 * @category zone of death
 * @since 2.0.0
 * @deprecated
 */
exports.traced = exports.Functor;
