// fp-ts-esm/lib/Traversable.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.traverse = traverse;
exports.sequence = sequence;
exports.getTraversableComposition = getTraversableComposition;
const Foldable_js_1 = require("./Foldable.js");
const function_js_1 = require("./function.js");
const Functor_js_1 = require("./Functor.js");
function traverse(T, G) {
    return (F) => {
        const traverseT = T.traverse(F);
        const traverseG = G.traverse(F);
        return (f) => (fga) => traverseT(fga, (ga) => traverseG(ga, f));
    };
}
function sequence(T, G) {
    return (F) => {
        const sequenceT = T.sequence(F);
        const sequenceG = G.sequence(F);
        return (fgha) => sequenceT(T.map(fgha, sequenceG));
    };
}
/** @deprecated */
function getTraversableComposition(F, G) {
    const map = (0, Functor_js_1.getFunctorComposition)(F, G).map;
    const FC = (0, Foldable_js_1.getFoldableComposition)(F, G);
    const _traverse = traverse(F, G);
    const _sequence = sequence(F, G);
    return {
        map,
        reduce: FC.reduce,
        foldMap: FC.foldMap,
        reduceRight: FC.reduceRight,
        traverse: (H) => {
            const traverseH = _traverse(H);
            return (fga, f) => (0, function_js_1.pipe)(fga, traverseH(f));
        },
        sequence: (H) => {
            const sequenceH = _sequence(H);
            return (fgha) => (0, function_js_1.pipe)(fgha, sequenceH);
        }
    };
}
