// fp-ts-esm/lib/Tree.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.tree = exports.chain = exports.exists = exports.apS = exports.bind = exports.let = exports.bindTo = exports.Do = exports.Comonad = exports.Traversable = exports.Foldable = exports.chainFirst = exports.Monad = exports.Chain = exports.Applicative = exports.apSecond = exports.apFirst = exports.Apply = exports.Pointed = exports.flap = exports.Functor = exports.URI = exports.of = exports.sequence = exports.traverse = exports.extract = exports.reduceRight = exports.foldMap = exports.reduce = exports.map = exports.flatten = exports.duplicate = exports.extend = exports.flatMap = exports.ap = void 0;
exports.make = make;
exports.getShow = getShow;
exports.getEq = getEq;
exports.drawForest = drawForest;
exports.drawTree = drawTree;
exports.unfoldTree = unfoldTree;
exports.unfoldForest = unfoldForest;
exports.unfoldTreeM = unfoldTreeM;
exports.unfoldForestM = unfoldForestM;
exports.fold = fold;
exports.elem = elem;
const Apply_js_1 = require("./Apply.js");
const A = __importStar(require("./Array.js"));
const Chain_js_1 = require("./Chain.js");
const Eq_js_1 = require("./Eq.js");
const function_js_1 = require("./function.js");
const Functor_js_1 = require("./Functor.js");
const _ = __importStar(require("./internal.js"));
/**
 * @category constructors
 * @since 2.0.0
 */
function make(value, forest = []) {
    return {
        value,
        forest
    };
}
/**
 * @category instances
 * @since 2.0.0
 */
function getShow(S) {
    const show = (t) => {
        return A.isEmpty(t.forest)
            ? `make(${S.show(t.value)})`
            : `make(${S.show(t.value)}, [${t.forest.map(show).join(', ')}])`;
    };
    return {
        show
    };
}
/**
 * @category instances
 * @since 2.0.0
 */
function getEq(E) {
    // eslint-disable-next-line prefer-const
    let SA;
    const R = (0, Eq_js_1.fromEquals)((x, y) => E.equals(x.value, y.value) && SA.equals(x.forest, y.forest));
    SA = A.getEq(R);
    return R;
}
const draw = (indentation, forest) => {
    let r = '';
    const len = forest.length;
    let tree;
    for (let i = 0; i < len; i++) {
        tree = forest[i];
        const isLast = i === len - 1;
        r += indentation + (isLast ? '└' : '├') + '─ ' + tree.value;
        r += draw(indentation + (len > 1 && !isLast ? '│  ' : '   '), tree.forest);
    }
    return r;
};
/**
 * Neat 2-dimensional drawing of a forest
 *
 * @since 2.0.0
 */
function drawForest(forest) {
    return draw('\n', forest);
}
/**
 * Neat 2-dimensional drawing of a tree
 *
 * @example
 * import { make, drawTree } from 'fp-ts/Tree'
 *
 * const fa = make('a', [
 *   make('b'),
 *   make('c'),
 *   make('d', [make('e'), make('f')])
 * ])
 *
 * assert.strictEqual(drawTree(fa), `a
 * ├─ b
 * ├─ c
 * └─ d
 *    ├─ e
 *    └─ f`)
 *
 *
 * @since 2.0.0
 */
function drawTree(tree) {
    return tree.value + drawForest(tree.forest);
}
/**
 * Build a (possibly infinite) tree from a seed value in breadth-first order.
 *
 * @category constructors
 * @since 2.0.0
 */
function unfoldTree(b, f) {
    const [a, bs] = f(b);
    return { value: a, forest: unfoldForest(bs, f) };
}
/**
 * Build a (possibly infinite) forest from a list of seed values in breadth-first order.
 *
 * @category constructors
 * @since 2.0.0
 */
function unfoldForest(bs, f) {
    return bs.map((b) => unfoldTree(b, f));
}
function unfoldTreeM(M) {
    const unfoldForestMM = unfoldForestM(M);
    return (b, f) => M.chain(f(b), ([a, bs]) => M.map(unfoldForestMM(bs, f), (ts) => ({ value: a, forest: ts })));
}
function unfoldForestM(M) {
    const traverseM = A.traverse(M);
    return (bs, f) => (0, function_js_1.pipe)(bs, traverseM((b) => unfoldTreeM(M)(b, f)));
}
/**
 * Fold a tree into a "summary" value in depth-first order.
 *
 * For each node in the tree, apply `f` to the `value` and the result of applying `f` to each `forest`.
 *
 * This is also known as the catamorphism on trees.
 *
 * @example
 * import { fold, make } from 'fp-ts/Tree'
 * import { concatAll } from 'fp-ts/Monoid'
 * import { MonoidSum } from 'fp-ts/number'
 *
 * const t = make(1, [make(2), make(3)])
 *
 * const sum = concatAll(MonoidSum)
 *
 * // Sum the values in a tree:
 * assert.deepStrictEqual(fold((a: number, bs: Array<number>) => a + sum(bs))(t), 6)
 *
 * // Find the maximum value in the tree:
 * assert.deepStrictEqual(fold((a: number, bs: Array<number>) => bs.reduce((b, acc) => Math.max(b, acc), a))(t), 3)
 *
 * // Count the number of leaves in the tree:
 * assert.deepStrictEqual(fold((_: number, bs: Array<number>) => (bs.length === 0 ? 1 : sum(bs)))(t), 2)
 *
 * @category folding
 * @since 2.6.0
 */
function fold(f) {
    const go = (tree) => f(tree.value, tree.forest.map(go));
    return go;
}
/* istanbul ignore next */
const _map = (fa, f) => (0, function_js_1.pipe)(fa, (0, exports.map)(f));
const _ap = (fab, fa) => (0, exports.flatMap)(fab, (f) => (0, function_js_1.pipe)(fa, (0, exports.map)(f)));
/* istanbul ignore next */
const _reduce = (fa, b, f) => (0, function_js_1.pipe)(fa, (0, exports.reduce)(b, f));
/* istanbul ignore next */
const _foldMap = (M) => {
    const foldMapM = (0, exports.foldMap)(M);
    return (fa, f) => (0, function_js_1.pipe)(fa, foldMapM(f));
};
/* istanbul ignore next */
const _reduceRight = (fa, b, f) => (0, function_js_1.pipe)(fa, (0, exports.reduceRight)(b, f));
/* istanbul ignore next */
const _extend = (wa, f) => (0, function_js_1.pipe)(wa, (0, exports.extend)(f));
/* istanbul ignore next */
const _traverse = (F) => {
    const traverseF = (0, exports.traverse)(F);
    return (ta, f) => (0, function_js_1.pipe)(ta, traverseF(f));
};
/**
 * @since 2.0.0
 */
const ap = (fa) => (fab) => _ap(fab, fa);
exports.ap = ap;
/**
 * @category sequencing
 * @since 2.14.0
 */
exports.flatMap = (0, function_js_1.dual)(2, (ma, f) => {
    const { value, forest } = f(ma.value);
    const concat = A.getMonoid().concat;
    return {
        value,
        forest: concat(forest, ma.forest.map((0, exports.flatMap)(f)))
    };
});
/**
 * @since 2.0.0
 */
const extend = (f) => (wa) => ({
    value: f(wa),
    forest: wa.forest.map((0, exports.extend)(f))
});
exports.extend = extend;
/**
 * @since 2.0.0
 */
exports.duplicate = (0, exports.extend)(function_js_1.identity);
/**
 * @category sequencing
 * @since 2.0.0
 */
exports.flatten = (0, exports.flatMap)(function_js_1.identity);
/**
 * `map` can be used to turn functions `(a: A) => B` into functions `(fa: F<A>) => F<B>` whose argument and return types
 * use the type constructor `F` to represent some computational context.
 *
 * @category mapping
 * @since 2.0.0
 */
const map = (f) => (fa) => ({
    value: f(fa.value),
    forest: fa.forest.map((0, exports.map)(f))
});
exports.map = map;
/**
 * @category folding
 * @since 2.0.0
 */
const reduce = (b, f) => (fa) => {
    let r = f(b, fa.value);
    const len = fa.forest.length;
    for (let i = 0; i < len; i++) {
        r = (0, function_js_1.pipe)(fa.forest[i], (0, exports.reduce)(r, f));
    }
    return r;
};
exports.reduce = reduce;
/**
 * @category folding
 * @since 2.0.0
 */
const foldMap = (M) => (f) => (0, exports.reduce)(M.empty, (acc, a) => M.concat(acc, f(a)));
exports.foldMap = foldMap;
/**
 * @category folding
 * @since 2.0.0
 */
const reduceRight = (b, f) => (fa) => {
    let r = b;
    const len = fa.forest.length;
    for (let i = len - 1; i >= 0; i--) {
        r = (0, function_js_1.pipe)(fa.forest[i], (0, exports.reduceRight)(r, f));
    }
    return f(fa.value, r);
};
exports.reduceRight = reduceRight;
/**
 * @category Extract
 * @since 2.6.2
 */
const extract = (wa) => wa.value;
exports.extract = extract;
/**
 * @category traversing
 * @since 2.6.3
 */
const traverse = (F) => {
    const traverseF = A.traverse(F);
    const out = (f) => (ta) => F.ap(F.map(f(ta.value), (value) => (forest) => ({
        value,
        forest
    })), (0, function_js_1.pipe)(ta.forest, traverseF(out(f))));
    return out;
};
exports.traverse = traverse;
/**
 * @category traversing
 * @since 2.6.3
 */
const sequence = (F) => (0, exports.traverse)(F)(function_js_1.identity);
exports.sequence = sequence;
/**
 * @category constructors
 * @since 2.7.0
 */
const of = (a) => make(a);
exports.of = of;
/**
 * @category type lambdas
 * @since 2.0.0
 */
exports.URI = 'Tree';
/**
 * @category instances
 * @since 2.7.0
 */
exports.Functor = {
    URI: exports.URI,
    map: _map
};
/**
 * @category mapping
 * @since 2.10.0
 */
exports.flap = (0, Functor_js_1.flap)(exports.Functor);
/**
 * @category instances
 * @since 2.10.0
 */
exports.Pointed = {
    URI: exports.URI,
    of: exports.of
};
/**
 * @category instances
 * @since 2.10.0
 */
exports.Apply = {
    URI: exports.URI,
    map: _map,
    ap: _ap
};
/**
 * Combine two effectful actions, keeping only the result of the first.
 *
 * @since 2.0.0
 */
exports.apFirst = (0, Apply_js_1.apFirst)(exports.Apply);
/**
 * Combine two effectful actions, keeping only the result of the second.
 *
 * @since 2.0.0
 */
exports.apSecond = (0, Apply_js_1.apSecond)(exports.Apply);
/**
 * @category instances
 * @since 2.7.0
 */
exports.Applicative = {
    URI: exports.URI,
    map: _map,
    ap: _ap,
    of: exports.of
};
/**
 * @category instances
 * @since 2.10.0
 */
exports.Chain = {
    URI: exports.URI,
    map: _map,
    ap: _ap,
    chain: exports.flatMap
};
/**
 * @category instances
 * @since 2.7.0
 */
exports.Monad = {
    URI: exports.URI,
    map: _map,
    ap: _ap,
    of: exports.of,
    chain: exports.flatMap
};
/**
 * Composes computations in sequence, using the return value of one computation to determine the next computation and
 * keeping only the result of the first.
 *
 * @since 2.0.0
 */
exports.chainFirst = (0, Chain_js_1.chainFirst)(exports.Chain);
/**
 * @category instances
 * @since 2.7.0
 */
exports.Foldable = {
    URI: exports.URI,
    reduce: _reduce,
    foldMap: _foldMap,
    reduceRight: _reduceRight
};
/**
 * @category instances
 * @since 2.7.0
 */
exports.Traversable = {
    URI: exports.URI,
    map: _map,
    reduce: _reduce,
    foldMap: _foldMap,
    reduceRight: _reduceRight,
    traverse: _traverse,
    sequence: exports.sequence
};
/**
 * @category instances
 * @since 2.7.0
 */
exports.Comonad = {
    URI: exports.URI,
    map: _map,
    extend: _extend,
    extract: exports.extract
};
// -------------------------------------------------------------------------------------
// do notation
// -------------------------------------------------------------------------------------
/**
 * @category do notation
 * @since 2.9.0
 */
exports.Do = (0, exports.of)(_.emptyRecord);
/**
 * @category do notation
 * @since 2.8.0
 */
exports.bindTo = (0, Functor_js_1.bindTo)(exports.Functor);
const let_ = /*#__PURE__*/ (0, Functor_js_1.let)(exports.Functor);
exports.let = let_;
/**
 * @category do notation
 * @since 2.8.0
 */
exports.bind = (0, Chain_js_1.bind)(exports.Chain);
/**
 * @category do notation
 * @since 2.8.0
 */
exports.apS = (0, Apply_js_1.apS)(exports.Apply);
// -------------------------------------------------------------------------------------
// utils
// -------------------------------------------------------------------------------------
/**
 * @since 2.0.0
 */
function elem(E) {
    const go = (a, fa) => E.equals(a, fa.value) || fa.forest.some((tree) => go(a, tree));
    return go;
}
/**
 * @since 2.11.0
 */
const exists = (predicate) => (ma) => predicate(ma.value) || ma.forest.some((0, exports.exists)(predicate));
exports.exists = exists;
// -------------------------------------------------------------------------------------
// legacy
// -------------------------------------------------------------------------------------
/**
 * Alias of `flatMap`.
 *
 * @category legacy
 * @since 2.0.0
 */
exports.chain = exports.flatMap;
// -------------------------------------------------------------------------------------
// deprecated
// -------------------------------------------------------------------------------------
/**
 * This instance is deprecated, use small, specific instances instead.
 * For example if a function needs a `Functor` instance, pass `T.Functor` instead of `T.tree`
 * (where `T` is from `import T from 'fp-ts/Tree'`)
 *
 * @category zone of death
 * @since 2.0.0
 * @deprecated
 */
exports.tree = {
    URI: exports.URI,
    map: _map,
    of: exports.of,
    ap: _ap,
    chain: exports.flatMap,
    reduce: _reduce,
    foldMap: _foldMap,
    reduceRight: _reduceRight,
    traverse: _traverse,
    sequence: exports.sequence,
    extract: exports.extract,
    extend: _extend
};
