// fp-ts-esm/lib/Writer.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.writer = exports.execWriter = exports.evalWriter = exports.execute = exports.evaluate = exports.flap = exports.Functor = exports.getApplicative = exports.getApply = exports.getPointed = exports.URI = exports.map = exports.censor = exports.listens = exports.pass = exports.listen = exports.tell = void 0;
exports.getChain = getChain;
exports.getMonad = getMonad;
const function_js_1 = require("./function.js");
const Functor_js_1 = require("./Functor.js");
// -------------------------------------------------------------------------------------
// constructors
// -------------------------------------------------------------------------------------
/**
 * Appends a value to the accumulator
 *
 * @category constructors
 * @since 2.0.0
 */
const tell = (w) => () => [undefined, w];
exports.tell = tell;
// -------------------------------------------------------------------------------------
// combinators
// -------------------------------------------------------------------------------------
/**
 * Modifies the result to include the changes to the accumulator
 *
 * @since 2.0.0
 */
const listen = (fa) => () => {
    const [a, w] = fa();
    return [[a, w], w];
};
exports.listen = listen;
/**
 * Applies the returned function to the accumulator
 *
 * @since 2.0.0
 */
const pass = (fa) => () => {
    const [[a, f], w] = fa();
    return [a, f(w)];
};
exports.pass = pass;
/**
 * Projects a value from modifications made to the accumulator during an action
 *
 * @since 2.0.0
 */
const listens = (f) => (fa) => () => {
    const [a, w] = fa();
    return [[a, f(w)], w];
};
exports.listens = listens;
/**
 * Modify the final accumulator value by applying a function
 *
 * @since 2.0.0
 */
const censor = (f) => (fa) => () => {
    const [a, w] = fa();
    return [a, f(w)];
};
exports.censor = censor;
/* istanbul ignore next */
const _map = (fa, f) => (0, function_js_1.pipe)(fa, (0, exports.map)(f));
/**
 * `map` can be used to turn functions `(a: A) => B` into functions `(fa: F<A>) => F<B>` whose argument and return types
 * use the type constructor `F` to represent some computational context.
 *
 * @category mapping
 * @since 2.0.0
 */
const map = (f) => (fa) => () => {
    const [a, w] = fa();
    return [f(a), w];
};
exports.map = map;
/**
 * @category type lambdas
 * @since 2.0.0
 */
exports.URI = 'Writer';
/**
 * @category instances
 * @since 2.10.0
 */
const getPointed = (M) => ({
    URI: exports.URI,
    _E: undefined,
    of: (a) => () => [a, M.empty]
});
exports.getPointed = getPointed;
/**
 * @category instances
 * @since 2.10.0
 */
const getApply = (S) => ({
    URI: exports.URI,
    _E: undefined,
    map: _map,
    ap: (fab, fa) => () => {
        const [f, w1] = fab();
        const [a, w2] = fa();
        return [f(a), S.concat(w1, w2)];
    }
});
exports.getApply = getApply;
/**
 * @category instances
 * @since 2.10.0
 */
const getApplicative = (M) => {
    const A = (0, exports.getApply)(M);
    const P = (0, exports.getPointed)(M);
    return {
        URI: exports.URI,
        _E: undefined,
        map: _map,
        ap: A.ap,
        of: P.of
    };
};
exports.getApplicative = getApplicative;
/**
 * @category instances
 * @since 2.10.0
 */
function getChain(S) {
    const A = (0, exports.getApply)(S);
    return {
        URI: exports.URI,
        _E: undefined,
        map: _map,
        ap: A.ap,
        chain: (fa, f) => () => {
            const [a, w1] = fa();
            const [b, w2] = f(a)();
            return [b, S.concat(w1, w2)];
        }
    };
}
/**
 * @category instances
 * @since 2.0.0
 */
function getMonad(M) {
    const A = (0, exports.getApplicative)(M);
    const C = getChain(M);
    return {
        URI: exports.URI,
        _E: undefined,
        map: _map,
        ap: A.ap,
        of: A.of,
        chain: C.chain
    };
}
/**
 * @category instances
 * @since 2.7.0
 */
exports.Functor = {
    URI: exports.URI,
    map: _map
};
/**
 * @category mapping
 * @since 2.10.0
 */
exports.flap = (0, Functor_js_1.flap)(exports.Functor);
// -------------------------------------------------------------------------------------
// utils
// -------------------------------------------------------------------------------------
/**
 * @since 2.8.0
 */
const evaluate = (fa) => fa()[0];
exports.evaluate = evaluate;
/**
 * @since 2.8.0
 */
const execute = (fa) => fa()[1];
exports.execute = execute;
// -------------------------------------------------------------------------------------
// deprecated
// -------------------------------------------------------------------------------------
/**
 * Use [`evaluate`](#evaluate) instead
 *
 * @category zone of death
 * @since 2.0.0
 * @deprecated
 */
const evalWriter = (fa) => fa()[0];
exports.evalWriter = evalWriter;
/**
 * Use [`execute`](#execute) instead
 *
 * @category zone of death
 * @since 2.0.0
 * @deprecated
 */
const execWriter = (fa) => fa()[1];
exports.execWriter = execWriter;
/**
 * Use [`Functor`](#functor) instead.
 *
 * @category zone of death
 * @since 2.0.0
 * @deprecated
 */
exports.writer = exports.Functor;
