// @wix/auto_sdk_payments_payment-acceptance-configurations/build/internal/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  SortOrder: () => SortOrder,
  WebhookIdentityType: () => WebhookIdentityType,
  bulkUpdatePaymentAcceptanceConfigurationTags: () => bulkUpdatePaymentAcceptanceConfigurationTags4,
  bulkUpdatePaymentAcceptanceConfigurationTagsByFilter: () => bulkUpdatePaymentAcceptanceConfigurationTagsByFilter4,
  createPaymentAcceptanceConfiguration: () => createPaymentAcceptanceConfiguration4,
  deletePaymentAcceptanceConfiguration: () => deletePaymentAcceptanceConfiguration4,
  getPaymentAcceptanceConfiguration: () => getPaymentAcceptanceConfiguration4,
  queryPaymentAcceptanceConfigurations: () => queryPaymentAcceptanceConfigurations4,
  updatePaymentAcceptanceConfiguration: () => updatePaymentAcceptanceConfiguration4
});
module.exports = __toCommonJS(index_exports);

// src/payments-payment-acceptance-configurations-v1-payment-acceptance-configuration-payment-acceptance-configurations.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/payments-payment-acceptance-configurations-v1-payment-acceptance-configuration-payment-acceptance-configurations.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixPaymentsPaymentAcceptanceConfigurationsV1PaymentAcceptanceConfigurationServiceUrl(opts) {
  const domainToMappings = {};
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_payment-acceptance-configurations";
function createPaymentAcceptanceConfiguration(payload) {
  function __createPaymentAcceptanceConfiguration({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "paymentAcceptanceConfiguration.createdDate" },
          { path: "paymentAcceptanceConfiguration.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.payments.payment_acceptance_configurations.v1.payment_acceptance_configuration",
      method: "POST",
      methodFqn: "wix.payments.payment_acceptance_configurations.v1.PaymentAcceptanceConfigurationService.CreatePaymentAcceptanceConfiguration",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPaymentAcceptanceConfigurationsV1PaymentAcceptanceConfigurationServiceUrl(
        {
          protoPath: "/v1/payment-acceptance-configurations",
          data: serializedData,
          host
        }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "paymentAcceptanceConfiguration.createdDate" },
            { path: "paymentAcceptanceConfiguration.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createPaymentAcceptanceConfiguration;
}
function getPaymentAcceptanceConfiguration(payload) {
  function __getPaymentAcceptanceConfiguration({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.payment_acceptance_configurations.v1.payment_acceptance_configuration",
      method: "GET",
      methodFqn: "wix.payments.payment_acceptance_configurations.v1.PaymentAcceptanceConfigurationService.GetPaymentAcceptanceConfiguration",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPaymentAcceptanceConfigurationsV1PaymentAcceptanceConfigurationServiceUrl(
        {
          protoPath: "/v1/payment-acceptance-configurations/{paymentAcceptanceConfigurationId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "paymentAcceptanceConfiguration.createdDate" },
            { path: "paymentAcceptanceConfiguration.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getPaymentAcceptanceConfiguration;
}
function updatePaymentAcceptanceConfiguration(payload) {
  function __updatePaymentAcceptanceConfiguration({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "paymentAcceptanceConfiguration.createdDate" },
          { path: "paymentAcceptanceConfiguration.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.payments.payment_acceptance_configurations.v1.payment_acceptance_configuration",
      method: "PATCH",
      methodFqn: "wix.payments.payment_acceptance_configurations.v1.PaymentAcceptanceConfigurationService.UpdatePaymentAcceptanceConfiguration",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPaymentAcceptanceConfigurationsV1PaymentAcceptanceConfigurationServiceUrl(
        {
          protoPath: "/v1/payment-acceptance-configurations/{paymentAcceptanceConfiguration.id}",
          data: serializedData,
          host
        }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "paymentAcceptanceConfiguration.createdDate" },
            { path: "paymentAcceptanceConfiguration.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updatePaymentAcceptanceConfiguration;
}
function deletePaymentAcceptanceConfiguration(payload) {
  function __deletePaymentAcceptanceConfiguration({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.payment_acceptance_configurations.v1.payment_acceptance_configuration",
      method: "DELETE",
      methodFqn: "wix.payments.payment_acceptance_configurations.v1.PaymentAcceptanceConfigurationService.DeletePaymentAcceptanceConfiguration",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPaymentAcceptanceConfigurationsV1PaymentAcceptanceConfigurationServiceUrl(
        {
          protoPath: "/v1/payment-acceptance-configurations/{paymentAcceptanceConfigurationId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deletePaymentAcceptanceConfiguration;
}
function queryPaymentAcceptanceConfigurations(payload) {
  function __queryPaymentAcceptanceConfigurations({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.payment_acceptance_configurations.v1.payment_acceptance_configuration",
      method: "POST",
      methodFqn: "wix.payments.payment_acceptance_configurations.v1.PaymentAcceptanceConfigurationService.QueryPaymentAcceptanceConfigurations",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPaymentAcceptanceConfigurationsV1PaymentAcceptanceConfigurationServiceUrl(
        {
          protoPath: "/v1/payment-acceptance-configurations/query",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "paymentAcceptanceConfigurations.createdDate" },
            { path: "paymentAcceptanceConfigurations.updatedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixPaymentsPaymentAcceptanceConfigurationsV1PaymentAcceptanceConfigurationServiceUrl(
            {
              protoPath: "/v1/payment-acceptance-configurations/query",
              data: payload,
              host
            }
          ),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryPaymentAcceptanceConfigurations;
}
function bulkUpdatePaymentAcceptanceConfigurationTags(payload) {
  function __bulkUpdatePaymentAcceptanceConfigurationTags({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.payment_acceptance_configurations.v1.payment_acceptance_configuration",
      method: "POST",
      methodFqn: "wix.payments.payment_acceptance_configurations.v1.PaymentAcceptanceConfigurationService.BulkUpdatePaymentAcceptanceConfigurationTags",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPaymentAcceptanceConfigurationsV1PaymentAcceptanceConfigurationServiceUrl(
        {
          protoPath: "/v1/bulk/payment-acceptance-configurations/update-tags",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdatePaymentAcceptanceConfigurationTags;
}
function bulkUpdatePaymentAcceptanceConfigurationTagsByFilter(payload) {
  function __bulkUpdatePaymentAcceptanceConfigurationTagsByFilter({
    host
  }) {
    const metadata = {
      entityFqdn: "wix.payments.payment_acceptance_configurations.v1.payment_acceptance_configuration",
      method: "POST",
      methodFqn: "wix.payments.payment_acceptance_configurations.v1.PaymentAcceptanceConfigurationService.BulkUpdatePaymentAcceptanceConfigurationTagsByFilter",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPaymentAcceptanceConfigurationsV1PaymentAcceptanceConfigurationServiceUrl(
        {
          protoPath: "/v1/bulk/payment-acceptance-configurations/update-tags-by-filter",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdatePaymentAcceptanceConfigurationTagsByFilter;
}

// src/payments-payment-acceptance-configurations-v1-payment-acceptance-configuration-payment-acceptance-configurations.universal.ts
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createPaymentAcceptanceConfiguration2(paymentAcceptanceConfiguration) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    paymentAcceptanceConfiguration
  });
  const reqOpts = createPaymentAcceptanceConfiguration(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.paymentAcceptanceConfiguration;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { paymentAcceptanceConfiguration: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["paymentAcceptanceConfiguration"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getPaymentAcceptanceConfiguration2(paymentAcceptanceConfigurationId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    paymentAcceptanceConfigurationId
  });
  const reqOpts = getPaymentAcceptanceConfiguration(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.paymentAcceptanceConfiguration;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { paymentAcceptanceConfigurationId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["paymentAcceptanceConfigurationId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updatePaymentAcceptanceConfiguration2(_id, paymentAcceptanceConfiguration) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    paymentAcceptanceConfiguration: {
      ...paymentAcceptanceConfiguration,
      id: _id
    }
  });
  const reqOpts = updatePaymentAcceptanceConfiguration(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.paymentAcceptanceConfiguration;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { paymentAcceptanceConfiguration: "$[1]" },
        explicitPathsToArguments: {
          "paymentAcceptanceConfiguration.id": "$[0]"
        },
        singleArgumentUnchanged: false
      },
      ["_id", "paymentAcceptanceConfiguration"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deletePaymentAcceptanceConfiguration2(paymentAcceptanceConfigurationId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    paymentAcceptanceConfigurationId
  });
  const reqOpts = deletePaymentAcceptanceConfiguration(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { paymentAcceptanceConfigurationId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["paymentAcceptanceConfigurationId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryPaymentAcceptanceConfigurations2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryPaymentAcceptanceConfigurations(
        payload
      );
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [])
      );
      return {
        items: transformedData?.paymentAcceptanceConfigurations,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function bulkUpdatePaymentAcceptanceConfigurationTags2(ids, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    ids,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdatePaymentAcceptanceConfigurationTags(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          ids: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["ids", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdatePaymentAcceptanceConfigurationTagsByFilter2(filter, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    filter,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdatePaymentAcceptanceConfigurationTagsByFilter(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          filter: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["filter", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/payments-payment-acceptance-configurations-v1-payment-acceptance-configuration-payment-acceptance-configurations.public.ts
function createPaymentAcceptanceConfiguration3(httpClient) {
  return (paymentAcceptanceConfiguration) => createPaymentAcceptanceConfiguration2(
    paymentAcceptanceConfiguration,
    // @ts-ignore
    { httpClient }
  );
}
function getPaymentAcceptanceConfiguration3(httpClient) {
  return (paymentAcceptanceConfigurationId) => getPaymentAcceptanceConfiguration2(
    paymentAcceptanceConfigurationId,
    // @ts-ignore
    { httpClient }
  );
}
function updatePaymentAcceptanceConfiguration3(httpClient) {
  return (_id, paymentAcceptanceConfiguration) => updatePaymentAcceptanceConfiguration2(
    _id,
    paymentAcceptanceConfiguration,
    // @ts-ignore
    { httpClient }
  );
}
function deletePaymentAcceptanceConfiguration3(httpClient) {
  return (paymentAcceptanceConfigurationId) => deletePaymentAcceptanceConfiguration2(
    paymentAcceptanceConfigurationId,
    // @ts-ignore
    { httpClient }
  );
}
function queryPaymentAcceptanceConfigurations3(httpClient) {
  return () => queryPaymentAcceptanceConfigurations2(
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdatePaymentAcceptanceConfigurationTags3(httpClient) {
  return (ids, options) => bulkUpdatePaymentAcceptanceConfigurationTags2(
    ids,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdatePaymentAcceptanceConfigurationTagsByFilter3(httpClient) {
  return (filter, options) => bulkUpdatePaymentAcceptanceConfigurationTagsByFilter2(
    filter,
    options,
    // @ts-ignore
    { httpClient }
  );
}

// src/payments-payment-acceptance-configurations-v1-payment-acceptance-configuration-payment-acceptance-configurations.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var createPaymentAcceptanceConfiguration4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(createPaymentAcceptanceConfiguration3);
var getPaymentAcceptanceConfiguration4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getPaymentAcceptanceConfiguration3);
var updatePaymentAcceptanceConfiguration4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(updatePaymentAcceptanceConfiguration3);
var deletePaymentAcceptanceConfiguration4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(deletePaymentAcceptanceConfiguration3);
var queryPaymentAcceptanceConfigurations4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(queryPaymentAcceptanceConfigurations3);
var bulkUpdatePaymentAcceptanceConfigurationTags4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(
  bulkUpdatePaymentAcceptanceConfigurationTags3
);
var bulkUpdatePaymentAcceptanceConfigurationTagsByFilter4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(
  bulkUpdatePaymentAcceptanceConfigurationTagsByFilter3
);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  SortOrder,
  WebhookIdentityType,
  bulkUpdatePaymentAcceptanceConfigurationTags,
  bulkUpdatePaymentAcceptanceConfigurationTagsByFilter,
  createPaymentAcceptanceConfiguration,
  deletePaymentAcceptanceConfiguration,
  getPaymentAcceptanceConfiguration,
  queryPaymentAcceptanceConfigurations,
  updatePaymentAcceptanceConfiguration
});
