// @wix/auto_sdk_seo_seo-tags/build/es/meta.mjs
// src/promote-seo-v1-seo-tags-seo-tags.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixPromoteSeoTagsServiceV1SeoTagsServiceUrl(opts) {
  const domainToMappings = {
    _: [
      {
        srcPath: "/_api/seo-tags-service/v1/seo-tags",
        destPath: "/v1/seo-tags"
      },
      {
        srcPath: "/_api/seo-tags-service/v1/resolve-seo-tags",
        destPath: "/v1/resolve-seo-tags"
      },
      {
        srcPath: "/_api/seo-tags-service/v1/resolve-static-page-seo-tags",
        destPath: "/v1/resolve-static-page-seo-tags"
      },
      {
        srcPath: "/_api/seo-tags-service/v1/resolve-item-seo-tags",
        destPath: "/v1/resolve-item-seo-tags"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/seo-tags-service/v1/resolve-seo-tags",
        destPath: "/v1/resolve-seo-tags"
      },
      {
        srcPath: "/_api/seo-tags-service/v1/resolve-static-page-seo-tags",
        destPath: "/v1/resolve-static-page-seo-tags"
      },
      {
        srcPath: "/_api/seo-tags-service/v1/resolve-item-seo-tags",
        destPath: "/v1/resolve-item-seo-tags"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/promote/seo/v1/resolve-seo-tags",
        destPath: "/v1/resolve-seo-tags"
      },
      {
        srcPath: "/promote/seo/v1/resolve-static-page-seo-tags",
        destPath: "/v1/resolve-static-page-seo-tags"
      },
      {
        srcPath: "/promote/seo/v1/resolve-item-seo-tags",
        destPath: "/v1/resolve-item-seo-tags"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_seo_seo-tags";
function resolveSeoTags(payload) {
  function __resolveSeoTags({ host }) {
    const metadata = {
      entityFqdn: "wix.promote.seo.v1.seo_tags",
      method: "GET",
      methodFqn: "wix.promote.seo.tags.service.v1.SeoTagsService.ResolveSeoTags",
      packageName: PACKAGE_NAME,
      url: resolveWixPromoteSeoTagsServiceV1SeoTagsServiceUrl({
        protoPath: "/v1/resolve-seo-tags",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload, true)
    };
    return metadata;
  }
  return __resolveSeoTags;
}
function resolveStaticPageSeoTags(payload) {
  function __resolveStaticPageSeoTags({ host }) {
    const metadata = {
      entityFqdn: "wix.promote.seo.v1.seo_tags",
      method: "GET",
      methodFqn: "wix.promote.seo.tags.service.v1.SeoTagsService.ResolveStaticPageSeoTags",
      packageName: PACKAGE_NAME,
      url: resolveWixPromoteSeoTagsServiceV1SeoTagsServiceUrl({
        protoPath: "/v1/resolve-static-page-seo-tags",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload, true)
    };
    return metadata;
  }
  return __resolveStaticPageSeoTags;
}
function resolveItemSeoTags(payload) {
  function __resolveItemSeoTags({ host }) {
    const metadata = {
      entityFqdn: "wix.promote.seo.v1.seo_tags",
      method: "GET",
      methodFqn: "wix.promote.seo.tags.service.v1.SeoTagsService.ResolveItemSeoTags",
      packageName: PACKAGE_NAME,
      url: resolveWixPromoteSeoTagsServiceV1SeoTagsServiceUrl({
        protoPath: "/v1/resolve-item-seo-tags",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __resolveItemSeoTags;
}

// src/promote-seo-v1-seo-tags-seo-tags.meta.ts
function resolveSeoTags2() {
  const payload = {};
  const getRequestOptions = resolveSeoTags(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/resolve-seo-tags",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function resolveStaticPageSeoTags2() {
  const payload = {};
  const getRequestOptions = resolveStaticPageSeoTags(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/resolve-static-page-seo-tags",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function resolveItemSeoTags2() {
  const payload = {};
  const getRequestOptions = resolveItemSeoTags(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/resolve-item-seo-tags",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  resolveItemSeoTags2 as resolveItemSeoTags,
  resolveSeoTags2 as resolveSeoTags,
  resolveStaticPageSeoTags2 as resolveStaticPageSeoTags
};
