// @wix/auto_sdk_restaurants_order-pacing-order/build/internal/es/meta.mjs
// src/restaurants-order-pacing-v1-order-order-pacing-order.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressRestaurantsOrderPacingOrderPacingServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-order-pacing-pacer",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-order-pacing-pacer",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-order-pacing-pacer",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/order-pacing-pacer",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/v1/order-pacing-pacer",
        destPath: "/v1"
      },
      {
        srcPath: "/restaurants/order-pacing-pacer/v1",
        destPath: "/v1"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-order-pacing-pacer",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/order-pacing-pacer/v1",
        destPath: "/v1"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/order-pacing-pacer/v1",
        destPath: "/v1"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/order-pacing-pacer/v1",
        destPath: "/v1"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_order-pacing-order";
function getCurrentDelay(payload) {
  function __getCurrentDelay({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.order_pacing.v1.order",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.order.pacing.OrderPacingService.GetCurrentDelay",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsOrderPacingOrderPacingServiceUrl({
        protoPath: "/v1/current-delay",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getCurrentDelay;
}
function getCurrentDelayWithValidityPeriod(payload) {
  function __getCurrentDelayWithValidityPeriod({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.order_pacing.v1.order",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.order.pacing.OrderPacingService.GetCurrentDelayWithValidityPeriod",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsOrderPacingOrderPacingServiceUrl({
        protoPath: "/v1/current-delay-with-validity-period",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getCurrentDelayWithValidityPeriod;
}
function listBusyBlocks(payload) {
  function __listBusyBlocks({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "from" }, { path: "until" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.order_pacing.v1.order",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.order.pacing.OrderPacingService.ListBusyBlocks",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsOrderPacingOrderPacingServiceUrl({
        protoPath: "/v1/busy-blocks",
        data: serializedData,
        host
      }),
      params: toURLSearchParams(serializedData),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "busyBlocks.startTime" },
            { path: "busyBlocks.endTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listBusyBlocks;
}

// src/restaurants-order-pacing-v1-order-order-pacing-order.meta.ts
function getCurrentDelay2() {
  const payload = {};
  const getRequestOptions = getCurrentDelay(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/current-delay",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getCurrentDelayWithValidityPeriod2() {
  const payload = {};
  const getRequestOptions = getCurrentDelayWithValidityPeriod(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/current-delay-with-validity-period",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listBusyBlocks2() {
  const payload = {};
  const getRequestOptions = listBusyBlocks(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/busy-blocks",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  getCurrentDelay2 as getCurrentDelay,
  getCurrentDelayWithValidityPeriod2 as getCurrentDelayWithValidityPeriod,
  listBusyBlocks2 as listBusyBlocks
};
