// @wix/auto_sdk_restaurants_item-variants/build/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  SortOrder: () => SortOrder,
  WebhookIdentityType: () => WebhookIdentityType,
  bulkCreateVariants: () => bulkCreateVariants2,
  bulkDeleteVariants: () => bulkDeleteVariants2,
  bulkUpdateVariants: () => bulkUpdateVariants2,
  countVariants: () => countVariants2,
  createVariant: () => createVariant2,
  deleteVariant: () => deleteVariant2,
  getVariant: () => getVariant2,
  listVariants: () => listVariants2,
  queryVariants: () => queryVariants2,
  updateVariant: () => updateVariant2
});
module.exports = __toCommonJS(index_typings_exports);

// src/restaurants-menus-v1-item-variant-item-variants.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/restaurants-menus-v1-item-variant-item-variants.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-variant",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-variant",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-variant",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-menus-item-variant",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-menus-item-variant",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-menus-item-variant",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/variants",
        destPath: "/v1/variants"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/variants/create",
        destPath: "/v1/bulk/variants/create"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/variants/update",
        destPath: "/v1/bulk/variants/update"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/variants/delete",
        destPath: "/v1/bulk/variants/delete"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-variant",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-menus-item-variant",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/variants",
        destPath: "/v1/variants"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/variants/create",
        destPath: "/v1/bulk/variants/create"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/variants/update",
        destPath: "/v1/bulk/variants/update"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/variants/delete",
        destPath: "/v1/bulk/variants/delete"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/item-variants",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/variants/create",
        destPath: "/v1/bulk/variants/create"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/variants/update",
        destPath: "/v1/bulk/variants/update"
      },
      {
        srcPath: "/restaurants/menus/v1/variants",
        destPath: "/v1/variants"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/variants/delete",
        destPath: "/v1/bulk/variants/delete"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-menus-item-variant",
        destPath: ""
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-variant",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_item-variants";
function createVariant(payload) {
  function __createVariant({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "variant.createdDate" },
          { path: "variant.updatedDate" },
          { path: "variant.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_variant",
      method: "POST",
      methodFqn: "wix.restaurants.menus_item_variant.v1.VariantsService.CreateVariant",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl({
        protoPath: "/v1/variants",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "variant.createdDate" },
            { path: "variant.updatedDate" },
            { path: "variant.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createVariant;
}
function getVariant(payload) {
  function __getVariant({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_variant",
      method: "GET",
      methodFqn: "wix.restaurants.menus_item_variant.v1.VariantsService.GetVariant",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl({
        protoPath: "/v1/variants/{variantId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "variant.createdDate" },
            { path: "variant.updatedDate" },
            { path: "variant.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getVariant;
}
function listVariants(payload) {
  function __listVariants({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_variant",
      method: "GET",
      methodFqn: "wix.restaurants.menus_item_variant.v1.VariantsService.ListVariants",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl({
        protoPath: "/v1/variants",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "variants.createdDate" },
            { path: "variants.updatedDate" },
            { path: "variants.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listVariants;
}
function updateVariant(payload) {
  function __updateVariant({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "variant.createdDate" },
          { path: "variant.updatedDate" },
          { path: "variant.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_variant",
      method: "PATCH",
      methodFqn: "wix.restaurants.menus_item_variant.v1.VariantsService.UpdateVariant",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl({
        protoPath: "/v1/variants/{variant.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "variant.createdDate" },
            { path: "variant.updatedDate" },
            { path: "variant.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateVariant;
}
function bulkCreateVariants(payload) {
  function __bulkCreateVariants({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "variants.createdDate" },
          { path: "variants.updatedDate" },
          { path: "variants.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_variant",
      method: "POST",
      methodFqn: "wix.restaurants.menus_item_variant.v1.VariantsService.BulkCreateVariants",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl({
        protoPath: "/v1/bulk/variants/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.variant.createdDate" },
            { path: "results.variant.updatedDate" },
            { path: "results.variant.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateVariants;
}
function bulkUpdateVariants(payload) {
  function __bulkUpdateVariants({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "variants.fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "variants.variant.createdDate" },
          { path: "variants.variant.updatedDate" },
          { path: "variants.variant.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_variant",
      method: "POST",
      methodFqn: "wix.restaurants.menus_item_variant.v1.VariantsService.BulkUpdateVariants",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl({
        protoPath: "/v1/bulk/variants/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.variant.createdDate" },
            { path: "results.variant.updatedDate" },
            { path: "results.variant.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateVariants;
}
function deleteVariant(payload) {
  function __deleteVariant({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_variant",
      method: "DELETE",
      methodFqn: "wix.restaurants.menus_item_variant.v1.VariantsService.DeleteVariant",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl({
        protoPath: "/v1/variants/{variantId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteVariant;
}
function queryVariants(payload) {
  function __queryVariants({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_variant",
      method: "POST",
      methodFqn: "wix.restaurants.menus_item_variant.v1.VariantsService.QueryVariants",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl({
        protoPath: "/v1/variants/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "variants.createdDate" },
            { path: "variants.updatedDate" },
            { path: "variants.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryVariants;
}
function countVariants(payload) {
  function __countVariants({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_variant",
      method: "POST",
      methodFqn: "wix.restaurants.menus_item_variant.v1.VariantsService.CountVariants",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl({
        protoPath: "/v1/variants/count",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __countVariants;
}
function bulkDeleteVariants(payload) {
  function __bulkDeleteVariants({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_variant",
      method: "DELETE",
      methodFqn: "wix.restaurants.menus_item_variant.v1.VariantsService.BulkDeleteVariants",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl({
        protoPath: "/v1/bulk/variants/delete",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __bulkDeleteVariants;
}

// src/restaurants-menus-v1-item-variant-item-variants.universal.ts
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createVariant2(variant) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ variant });
  const reqOpts = createVariant(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.variant;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { variant: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["variant"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getVariant2(variantId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    variantId
  });
  const reqOpts = getVariant(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.variant;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { variantId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["variantId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listVariants2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    variantIds: options?.variantIds,
    paging: options?.paging
  });
  const reqOpts = listVariants(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          variantIds: "$[0].variantIds",
          paging: "$[0].paging"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateVariant2(_id, variant) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    variant: { ...variant, id: _id }
  });
  const reqOpts = updateVariant(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.variant;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { variant: "$[1]" },
        explicitPathsToArguments: { "variant.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "variant"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCreateVariants2(variants, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    variants,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkCreateVariants(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          variants: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["variants", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateVariants2(variants, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    variants,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkUpdateVariants(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          variants: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["variants", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteVariant2(variantId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    variantId
  });
  const reqOpts = deleteVariant(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { variantId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["variantId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryVariants2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryVariants(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [])
      );
      return {
        items: transformedData?.variants,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function countVariants2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    filter: options?.filter
  });
  const reqOpts = countVariants(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { filter: "$[0].filter" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkDeleteVariants2(ids) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ ids });
  const reqOpts = bulkDeleteVariants(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { ids: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["ids"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  SortOrder,
  WebhookIdentityType,
  bulkCreateVariants,
  bulkDeleteVariants,
  bulkUpdateVariants,
  countVariants,
  createVariant,
  deleteVariant,
  getVariant,
  listVariants,
  queryVariants,
  updateVariant
});
