// @wix/auto_sdk_payments_saved-payment-methods/build/internal/es/index.mjs
// src/payments-saved-payment-methods-v1-saved-payment-method-saved-payment-methods.public.ts
import { renameKeysFromRESTResponseToSDKResponse as renameKeysFromRESTResponseToSDKResponse2 } from "@wix/sdk-runtime/rename-all-nested-keys";
import { transformRESTTimestampToSDKTimestamp as transformRESTTimestampToSDKTimestamp2 } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
import { EventDefinition } from "@wix/sdk-types";

// src/payments-saved-payment-methods-v1-saved-payment-method-saved-payment-methods.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/payments-saved-payment-methods-v1-saved-payment-method-saved-payment-methods.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixPaymentsSavedPaymentMethodsV1SavedPaymentMethodServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/payments/saved-payment-methods",
        destPath: ""
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/payments/saved-payment-methods",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/payments/saved-payment-methods",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_saved-payment-methods";
function upsertSavedPaymentMethod(payload) {
  function __upsertSavedPaymentMethod({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "savedPaymentMethod.createdDate" },
          { path: "savedPaymentMethod.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.payments.saved_payment_methods.v1.saved_payment_method",
      method: "POST",
      methodFqn: "wix.payments.saved_payment_methods.v1.SavedPaymentMethodService.UpsertSavedPaymentMethod",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsSavedPaymentMethodsV1SavedPaymentMethodServiceUrl({
        protoPath: "/v1/saved-payment-methods/upsert",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "savedPaymentMethod.createdDate" },
            { path: "savedPaymentMethod.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __upsertSavedPaymentMethod;
}
function getSavedPaymentMethod(payload) {
  function __getSavedPaymentMethod({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.saved_payment_methods.v1.saved_payment_method",
      method: "GET",
      methodFqn: "wix.payments.saved_payment_methods.v1.SavedPaymentMethodService.GetSavedPaymentMethod",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsSavedPaymentMethodsV1SavedPaymentMethodServiceUrl({
        protoPath: "/v1/saved-payment-methods/{savedPaymentMethodId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "savedPaymentMethod.createdDate" },
            { path: "savedPaymentMethod.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getSavedPaymentMethod;
}
function deleteSavedPaymentMethod(payload) {
  function __deleteSavedPaymentMethod({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.saved_payment_methods.v1.saved_payment_method",
      method: "DELETE",
      methodFqn: "wix.payments.saved_payment_methods.v1.SavedPaymentMethodService.DeleteSavedPaymentMethod",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsSavedPaymentMethodsV1SavedPaymentMethodServiceUrl({
        protoPath: "/v1/saved-payment-methods/{savedPaymentMethodId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteSavedPaymentMethod;
}
function listSavedPaymentMethods(payload) {
  function __listSavedPaymentMethods({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.saved_payment_methods.v1.saved_payment_method",
      method: "GET",
      methodFqn: "wix.payments.saved_payment_methods.v1.SavedPaymentMethodService.ListSavedPaymentMethods",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsSavedPaymentMethodsV1SavedPaymentMethodServiceUrl({
        protoPath: "/v1/saved-payment-methods",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "savedPaymentMethods.createdDate" },
            { path: "savedPaymentMethods.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listSavedPaymentMethods;
}
function updateSavedPaymentMethod(payload) {
  function __updateSavedPaymentMethod({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "savedPaymentMethod.createdDate" },
          { path: "savedPaymentMethod.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.payments.saved_payment_methods.v1.saved_payment_method",
      method: "PATCH",
      methodFqn: "wix.payments.saved_payment_methods.v1.SavedPaymentMethodService.UpdateSavedPaymentMethod",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsSavedPaymentMethodsV1SavedPaymentMethodServiceUrl({
        protoPath: "/v1/saved-payment-methods/{savedPaymentMethod.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "savedPaymentMethod.createdDate" },
            { path: "savedPaymentMethod.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateSavedPaymentMethod;
}
function markSavedPaymentMethodPrimary(payload) {
  function __markSavedPaymentMethodPrimary({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.saved_payment_methods.v1.saved_payment_method",
      method: "GET",
      methodFqn: "wix.payments.saved_payment_methods.v1.SavedPaymentMethodService.MarkSavedPaymentMethodPrimary",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsSavedPaymentMethodsV1SavedPaymentMethodServiceUrl({
        protoPath: "/v1/saved-payment-methods/{savedPaymentMethodId}/mark-primary",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __markSavedPaymentMethodPrimary;
}

// src/payments-saved-payment-methods-v1-saved-payment-method-saved-payment-methods.universal.ts
var CardBrand = /* @__PURE__ */ ((CardBrand2) => {
  CardBrand2["UNKNOWN_CARD_BRAND"] = "UNKNOWN_CARD_BRAND";
  CardBrand2["AMEX"] = "AMEX";
  CardBrand2["DANKORT"] = "DANKORT";
  CardBrand2["DINERS"] = "DINERS";
  CardBrand2["DISCOVER"] = "DISCOVER";
  CardBrand2["ISRACARD"] = "ISRACARD";
  CardBrand2["JCB"] = "JCB";
  CardBrand2["MAESTRO"] = "MAESTRO";
  CardBrand2["MASTERCARD"] = "MASTERCARD";
  CardBrand2["UNIONPAY"] = "UNIONPAY";
  CardBrand2["VISA"] = "VISA";
  CardBrand2["RUPAY"] = "RUPAY";
  return CardBrand2;
})(CardBrand || {});
var RequestedFields = /* @__PURE__ */ ((RequestedFields2) => {
  RequestedFields2["UNKNOWN_REQUESTED_FIELD"] = "UNKNOWN_REQUESTED_FIELD";
  RequestedFields2["SENSITIVE_INFO"] = "SENSITIVE_INFO";
  return RequestedFields2;
})(RequestedFields || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function upsertSavedPaymentMethod2(savedPaymentMethod, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    savedPaymentMethod,
    temporaryCardToken: options?.temporaryCardToken
  });
  const reqOpts = upsertSavedPaymentMethod(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          savedPaymentMethod: "$[0]",
          temporaryCardToken: "$[1].temporaryCardToken"
        },
        singleArgumentUnchanged: false
      },
      ["savedPaymentMethod", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getSavedPaymentMethod2(savedPaymentMethodId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    savedPaymentMethodId,
    fields: options?.fields
  });
  const reqOpts = getSavedPaymentMethod(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.savedPaymentMethod;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          savedPaymentMethodId: "$[0]",
          fields: "$[1].fields"
        },
        singleArgumentUnchanged: false
      },
      ["savedPaymentMethodId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteSavedPaymentMethod2(savedPaymentMethodId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    savedPaymentMethodId
  });
  const reqOpts = deleteSavedPaymentMethod(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { savedPaymentMethodId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["savedPaymentMethodId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listSavedPaymentMethods2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    siteMemberId: options?.siteMemberId,
    paymentMethodTypeId: options?.paymentMethodTypeId,
    merchantUseAllowed: options?.merchantUseAllowed,
    fields: options?.fields,
    paging: options?.paging
  });
  const reqOpts = listSavedPaymentMethods(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          siteMemberId: "$[0].siteMemberId",
          paymentMethodTypeId: "$[0].paymentMethodTypeId",
          merchantUseAllowed: "$[0].merchantUseAllowed",
          fields: "$[0].fields",
          paging: "$[0].paging"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateSavedPaymentMethod2(_id, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    savedPaymentMethod: { ...options?.savedPaymentMethod, id: _id }
  });
  const reqOpts = updateSavedPaymentMethod(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.savedPaymentMethod;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {
          savedPaymentMethod: "$[1].savedPaymentMethod"
        },
        explicitPathsToArguments: { "savedPaymentMethod.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function markSavedPaymentMethodPrimary2(savedPaymentMethodId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    savedPaymentMethodId
  });
  const reqOpts = markSavedPaymentMethodPrimary(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { savedPaymentMethodId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["savedPaymentMethodId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/payments-saved-payment-methods-v1-saved-payment-method-saved-payment-methods.public.ts
function upsertSavedPaymentMethod3(httpClient) {
  return (savedPaymentMethod, options) => upsertSavedPaymentMethod2(
    savedPaymentMethod,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function getSavedPaymentMethod3(httpClient) {
  return (savedPaymentMethodId, options) => getSavedPaymentMethod2(
    savedPaymentMethodId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function deleteSavedPaymentMethod3(httpClient) {
  return (savedPaymentMethodId) => deleteSavedPaymentMethod2(
    savedPaymentMethodId,
    // @ts-ignore
    { httpClient }
  );
}
function listSavedPaymentMethods3(httpClient) {
  return (options) => listSavedPaymentMethods2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function updateSavedPaymentMethod3(httpClient) {
  return (_id, options) => updateSavedPaymentMethod2(
    _id,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function markSavedPaymentMethodPrimary3(httpClient) {
  return (savedPaymentMethodId) => markSavedPaymentMethodPrimary2(
    savedPaymentMethodId,
    // @ts-ignore
    { httpClient }
  );
}
var onSavedPaymentMethodCreated = EventDefinition(
  "wix.payments.saved_payment_methods.v1.saved_payment_method_created",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths2(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onSavedPaymentMethodDeleted = EventDefinition(
  "wix.payments.saved_payment_methods.v1.saved_payment_method_deleted",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths2(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "undefined.createdDate" },
          { path: "undefined.updatedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onSavedPaymentMethodUpdated = EventDefinition(
  "wix.payments.saved_payment_methods.v1.saved_payment_method_updated",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths2(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();

// src/payments-saved-payment-methods-v1-saved-payment-method-saved-payment-methods.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
import { createEventModule } from "@wix/sdk-runtime/event-definition-modules";
var upsertSavedPaymentMethod4 = /* @__PURE__ */ createRESTModule(upsertSavedPaymentMethod3);
var getSavedPaymentMethod4 = /* @__PURE__ */ createRESTModule(getSavedPaymentMethod3);
var deleteSavedPaymentMethod4 = /* @__PURE__ */ createRESTModule(deleteSavedPaymentMethod3);
var listSavedPaymentMethods4 = /* @__PURE__ */ createRESTModule(listSavedPaymentMethods3);
var updateSavedPaymentMethod4 = /* @__PURE__ */ createRESTModule(updateSavedPaymentMethod3);
var markSavedPaymentMethodPrimary4 = /* @__PURE__ */ createRESTModule(markSavedPaymentMethodPrimary3);
var onSavedPaymentMethodCreated2 = createEventModule(onSavedPaymentMethodCreated);
var onSavedPaymentMethodDeleted2 = createEventModule(onSavedPaymentMethodDeleted);
var onSavedPaymentMethodUpdated2 = createEventModule(onSavedPaymentMethodUpdated);
export {
  CardBrand,
  RequestedFields,
  WebhookIdentityType,
  deleteSavedPaymentMethod4 as deleteSavedPaymentMethod,
  getSavedPaymentMethod4 as getSavedPaymentMethod,
  listSavedPaymentMethods4 as listSavedPaymentMethods,
  markSavedPaymentMethodPrimary4 as markSavedPaymentMethodPrimary,
  onSavedPaymentMethodCreated2 as onSavedPaymentMethodCreated,
  onSavedPaymentMethodDeleted2 as onSavedPaymentMethodDeleted,
  onSavedPaymentMethodUpdated2 as onSavedPaymentMethodUpdated,
  updateSavedPaymentMethod4 as updateSavedPaymentMethod,
  upsertSavedPaymentMethod4 as upsertSavedPaymentMethod
};
