// @wix/patterns/dist/cjs/providers/DropdownTranslationProvider.js
"use strict";

exports.__esModule = true;
exports.useDropdownTranslationContext = exports.DropdownTranslationProvider = exports.DropdownTranslationContext = void 0;
var _react = _interopRequireWildcard(require("react"));
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/providers/DropdownTranslationProvider.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
const DropdownTranslationContext = exports.DropdownTranslationContext = /*#__PURE__*/(0, _react.createContext)(undefined);
const DropdownTranslationProvider = ({
  children,
  value
}) => {
  return /*#__PURE__*/_react.default.createElement(DropdownTranslationContext.Provider, {
    value: value,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 21,
      columnNumber: 5
    }
  }, children);
};
exports.DropdownTranslationProvider = DropdownTranslationProvider;
const useDropdownTranslationContext = () => {
  const context = (0, _react.useContext)(DropdownTranslationContext);
  if (!context) {
    throw new Error('useDropdownTranslationContext must be used within a DropdownTranslationProvider');
  }
  const translate = (0, _react.useCallback)((key, options) => {
    const valueFuncOrString = context.translations[key];
    if (typeof valueFuncOrString === 'function') {
      return valueFuncOrString(options);
    }
    return valueFuncOrString;
  }, [context]);
  return {
    translate
  };
};
exports.useDropdownTranslationContext = useDropdownTranslationContext;
