// @wix/patterns/dist/cjs/components/CollectionTable/CollectionTableContentCard.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.CollectionTableContentCard = CollectionTableContentCard;
var _designSystem = require("@wix/design-system");
var _react = _interopRequireDefault(require("react"));
var _ToolbarCollectionContentAutoSizer = require("../ToolbarCollectionContentAutoSizer");
var _commonStCss = require("../common.st.css.js");
var _useTableRoundedCornersEnabled = require("../../hooks/useTableRoundedCornersEnabled");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CollectionTable/CollectionTableContentCard.tsx";
function CollectionTableContentCard(props) {
  const {
    children
  } = props;
  const roundedCornersEnabled = (0, _useTableRoundedCornersEnabled.useTableRoundedCornersEnabled)();
  return (
    /*#__PURE__*/
    // wrapping with Card is WSR way to make the sticky table header not to add extra border between the header and the content https://www.docs.wixdesignsystem.com/?path=/story/components-lists-table--table#Sticky_Toolbar_&_Sub-toolbar
    _react.default.createElement(_designSystem.Box, {
      direction: "vertical",
      backgroundColor: "D80",
      borderRadius: "0px 0px 8px 8px",
      overflow: roundedCornersEnabled ? 'hidden' : undefined,
      className: (0, _commonStCss.st)(_commonStCss.classes.column, _commonStCss.classes.fullHeight),
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 20,
        columnNumber: 5
      }
    }, /*#__PURE__*/_react.default.createElement(_ToolbarCollectionContentAutoSizer.ToolbarCollectionContentAutoSizer, {
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 27,
        columnNumber: 7
      }
    }, /*#__PURE__*/_react.default.createElement(_designSystem.Table.Content, {
      titleBarVisible: false,
      dataHook: "table-content",
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 28,
        columnNumber: 9
      }
    }), children))
  );
}
