// @wix/patterns/dist/cjs/components/CollectionTable/CollectionTableContent.js
"use strict";

exports.__esModule = true;
exports.CollectionTableContent = void 0;
var _react = _interopRequireWildcard(require("react"));
var _mobxReactLite = require("mobx-react-lite");
var _CollectionTableContentCard = require("./CollectionTableContentCard");
var _designSystem = require("@wix/design-system");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CollectionTable/CollectionTableContent.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _CollectionTableContent(props) {
  var _state$container$inte;
  const {
    state: tableState,
    children
  } = props;
  const state = tableState.toolbar;
  const {
    showInitialLoader,
    showEmptyState,
    multi: {
      showRefreshLoader,
      showErrorState
    },
    internalScroll
  } = state;
  const floatingScrollBarEnabled = (_state$container$inte = state.container.internalExperiments) == null ? void 0 : _state$container$inte.enabled('specs.cairo.HorizontalScrollWithScrollBarNewUX');
  const showTableContent = !showInitialLoader && !showRefreshLoader && !showErrorState && (!internalScroll || !showEmptyState);
  const [onFullRenderEnd] = (0, _react.useState)(() => state.fedopsReporter.onFullRenderStart());
  (0, _react.useEffect)(() => {
    if (showTableContent) {
      onFullRenderEnd();
    }
  }, [showTableContent]);
  if (!showTableContent) {
    return null;
  }
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, floatingScrollBarEnabled && /*#__PURE__*/_react.default.createElement(_designSystem.TableFloatingScrollBar, {
    zIndex: 1,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 57,
      columnNumber: 36
    }
  }), /*#__PURE__*/_react.default.createElement(_CollectionTableContentCard.CollectionTableContentCard, {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 58,
      columnNumber: 7
    }
  }, children));
}
const CollectionTableContent = exports.CollectionTableContent = (0, _mobxReactLite.observer)(_CollectionTableContent);
