// @wix/auto_sdk_restaurants_local-delivery-spi/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  BusinessStatusOriginal: () => BusinessStatus,
  businessStatus: () => businessStatus2,
  createDelivery: () => createDelivery2,
  estimateDelivery: () => estimateDelivery2,
  listAccountIds: () => listAccountIds2
});
module.exports = __toCommonJS(meta_exports);

// src/localdelivery-spi-v1-tpa-local-delivery-spi.http.ts
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressLocaldeliverySpiV1LocalDeliveryProviderUrl(opts) {
  const domainToMappings = {};
  return (0, import_rest_modules.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_local-delivery-spi";
function listAccountIds(payload) {
  function __listAccountIds({ host }) {
    const metadata = {
      entityFqdn: "wix.localdelivery.spi.v1.tpa",
      method: "POST",
      methodFqn: "com.wixpress.localdelivery.spi.v1.LocalDeliveryProvider.ListAccountIds",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressLocaldeliverySpiV1LocalDeliveryProviderUrl({
        protoPath: "/account-ids",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __listAccountIds;
}
function businessStatus(payload) {
  function __businessStatus({ host }) {
    const metadata = {
      entityFqdn: "wix.localdelivery.spi.v1.tpa",
      method: "POST",
      methodFqn: "com.wixpress.localdelivery.spi.v1.LocalDeliveryProvider.BusinessStatus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressLocaldeliverySpiV1LocalDeliveryProviderUrl({
        protoPath: "/business-status",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __businessStatus;
}
function estimateDelivery(payload) {
  function __estimateDelivery({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "pickupTime" }, { path: "dropoffTime" }]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "pickupAddress.geocode.latitude" },
          { path: "pickupAddress.geocode.longitude" },
          { path: "dropoffAddress.geocode.latitude" },
          { path: "dropoffAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.localdelivery.spi.v1.tpa",
      method: "POST",
      methodFqn: "com.wixpress.localdelivery.spi.v1.LocalDeliveryProvider.EstimateDelivery",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressLocaldeliverySpiV1LocalDeliveryProviderUrl({
        protoPath: "/estimate-delivery",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "promisedPickupTime" },
            { path: "promisedDropoffTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __estimateDelivery;
}
function createDelivery(payload) {
  function __createDelivery({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "pickupTime" }, { path: "dropoffTime" }]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "pickupAddress.geocode.latitude" },
          { path: "pickupAddress.geocode.longitude" },
          { path: "dropoffAddress.geocode.latitude" },
          { path: "dropoffAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.localdelivery.spi.v1.tpa",
      method: "POST",
      methodFqn: "com.wixpress.localdelivery.spi.v1.LocalDeliveryProvider.CreateDelivery",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressLocaldeliverySpiV1LocalDeliveryProviderUrl({
        protoPath: "/create-delivery",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "pickupWindowStartTime" },
            { path: "pickupWindowEndTime" },
            { path: "dropoffWindowStartTime" },
            { path: "dropoffWindowEndTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createDelivery;
}

// src/localdelivery-spi-v1-tpa-local-delivery-spi.types.ts
var BusinessStatus = /* @__PURE__ */ ((BusinessStatus2) => {
  BusinessStatus2["NONE"] = "NONE";
  BusinessStatus2["READY"] = "READY";
  BusinessStatus2["PENDING"] = "PENDING";
  BusinessStatus2["ERROR"] = "ERROR";
  BusinessStatus2["NOT_STARTED"] = "NOT_STARTED";
  return BusinessStatus2;
})(BusinessStatus || {});

// src/localdelivery-spi-v1-tpa-local-delivery-spi.meta.ts
function listAccountIds2() {
  const payload = {};
  const getRequestOptions = listAccountIds(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/account-ids",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function businessStatus2() {
  const payload = {};
  const getRequestOptions = businessStatus(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/business-status",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function estimateDelivery2() {
  const payload = {};
  const getRequestOptions = estimateDelivery(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/estimate-delivery",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createDelivery2() {
  const payload = {};
  const getRequestOptions = createDelivery(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/create-delivery",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  BusinessStatusOriginal,
  businessStatus,
  createDelivery,
  estimateDelivery,
  listAccountIds
});
