// @wix/astro/build/dependencies/astro-auth/backend-runtime/routes/callback.js
import {
  saveSessionTokensToCookie
} from "../chunk-HPW4ZAEJ.js";
import {
  object,
  string
} from "../chunk-BIARYLOZ.js";
import {
  auth,
  oAuthStateCookieName
} from "../chunk-YHKPE45V.js";
import "../chunk-MLKGABMK.js";

// src/routes/callback.ts
var oauthCookieSchema = object({
  codeChallenge: string(),
  codeVerifier: string(),
  originalUri: string(),
  redirectUri: string(),
  state: string()
});
var GET = async (context) => {
  const oauthStateCookie = context.cookies.get(oAuthStateCookieName);
  if (oauthStateCookie == null) {
    throw new Error(`Missing \`${oAuthStateCookieName}\` cookie`);
  }
  const oauthData = oauthCookieSchema.parse(JSON.parse(oauthStateCookie.value));
  if (!oauthData.originalUri.startsWith("/")) {
    throw new Error(
      "Invalid `originalUri` cookie param, only relative URLs are allowed"
    );
  }
  const { code, error, errorDescription, state } = auth.parseFromUrl(
    context.url.toString(),
    "query"
  );
  if (error != null) {
    throw new Error(`Error while authenticating: \`${errorDescription}\``);
  }
  const memberTokens = await auth.getMemberTokens(code, state, oauthData);
  context.cookies.delete(oAuthStateCookieName, {
    httpOnly: true,
    path: "/",
    sameSite: "lax",
    secure: true
  });
  saveSessionTokensToCookie(context, memberTokens);
  return context.redirect(oauthData.originalUri);
};
export {
  GET
};
