// @wix/astro/build/dependencies/astro-auth/backend-runtime/routes/login.js
import {
  _enum,
  object,
  string
} from "../chunk-BIARYLOZ.js";
import {
  auth,
  oAuthStateCookieName
} from "../chunk-YHKPE45V.js";
import "../chunk-MLKGABMK.js";

// src/routes/login.ts
var loginSearchParams = object({
  prompt: _enum(["login", "none"]).optional(),
  returnToUrl: string().optional()
});
var GET = async ({ request, url }) => {
  const { prompt, returnToUrl = "/" } = loginSearchParams.parse(
    Object.fromEntries(url.searchParams.entries())
  );
  const callbackUrl = new URL("/api/auth/callback", url);
  const referer = request.headers.get("referer");
  if (referer != null) {
    const originalUrl = new URL(referer);
    callbackUrl.protocol = originalUrl.protocol;
  }
  const oauthData = auth.generateOAuthData(callbackUrl.toString(), returnToUrl);
  const { authUrl } = await auth.getAuthUrl(oauthData, {
    prompt,
    responseMode: "query"
  });
  const sameSite = import.meta.env.DEV ? "None" : "Lax";
  return new Response(null, {
    headers: {
      Location: authUrl,
      "Set-Cookie": `${oAuthStateCookieName}=${JSON.stringify(
        oauthData
      )}; Max-Age=1800; Path=/; HttpOnly; Secure; SameSite=${sameSite}`
    },
    status: 302
  });
};
export {
  GET
};
