// @wix/bex-utils/@wix/ambassador-app-settings-v1-settings/build/es/builders.impl.js
import { mergeDeep, someString, someRepeated, } from '@wix/metro-common-builders';
import { Host, Scope, State, } from './types.impl';
// @ts-ignore
import stack from 'callsite';
export function aAppSettings(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        state: aState(),
        host: aHost(),
        settings: {},
        translations: { translations: someString() },
    }, overrideObject);
}
export function aState(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = State.NR; }
    return enumFirstItem;
}
export function aHost(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = Host.NA; }
    return enumFirstItem;
}
export function aSetComponentSettingsRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        componentId: someString({
            validations: [{ type: 'MAX_LENGTH', value: '36' }],
        }),
        settings: {},
        state: aState(),
        host: aHost(),
        translations: [aTranslatedLanguage()],
        states: [aState()],
    }, overrideObject);
}
export function aTranslatedLanguage(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        languageKey: aLanguageKey(),
        translations: { translations: someString() },
    }, overrideObject);
}
export function aLanguageKey(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        languageCode: someString(),
        regionCode: someString(),
        scriptVariant: someString(),
    }, overrideObject);
}
export function aSetComponentSettingsResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        settings: {},
        host: aHost(),
        translations: [aTranslatedLanguage()],
        states: [aState()],
    }, overrideObject);
}
export function aSetComponentSettingsForMigrationRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        componentId: someString({
            validations: [{ type: 'MAX_LENGTH', value: '36' }],
        }),
        settings: {},
        state: aState(),
        host: aHost(),
        translations: [aTranslatedLanguage()],
        states: [aState()],
        appDefId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        instanceId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
export function aSetComponentSettingsForMigrationResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        settings: {},
        host: aHost(),
        translations: [aTranslatedLanguage()],
        states: [aState()],
    }, overrideObject);
}
export function aGetComponentSettingsRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        componentId: someString({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '36' },
            ],
        }),
        state: aState(),
        host: aHost(),
        languageKey: aLanguageKey(),
        includeGlobalSettings: true,
    }, overrideObject);
}
export function aGetComponentSettingsResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        settings: {},
        state: aState(),
        host: aHost(),
        translations: { translations: someString() },
        globalSettings: aAppSettings(),
    }, overrideObject);
}
export function aBulkGetComponentSettingsRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        componentIds: [
            someString({
                validations: [
                    { type: 'MIN_LENGTH', value: '1' },
                    { type: 'MAX_LENGTH', value: '36' },
                    { type: 'MIN_SIZE', value: '1' },
                    { type: 'MAX_SIZE', value: '100' },
                ],
            }),
        ],
        state: aState(),
        host: aHost(),
        languageKey: aLanguageKey(),
    }, overrideObject);
}
export function aBulkGetComponentSettingsResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        componentSettings: [aComponentSettings()],
    }, overrideObject);
}
export function aComponentSettings(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        componentId: someString({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '36' },
            ],
        }),
        settings: {},
        state: aState(),
        host: aHost(),
        translations: { translations: someString() },
    }, overrideObject);
}
export function aSetGlobalSettingsRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        settings: {},
        componentSettings: [aSetComponentSettingsItem()],
        state: aState(),
        host: aHost(),
        translations: [aTranslatedLanguage()],
        states: [aState()],
    }, overrideObject);
}
export function aSetComponentSettingsItem(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        state: aState(),
        host: aHost(),
        componentId: someString({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '36' },
            ],
        }),
        settings: {},
        translations: [aTranslatedLanguage()],
    }, overrideObject);
}
export function aSetGlobalSettingsResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        settings: {},
        componentSettings: [aSetComponentSettingsItem()],
        host: aHost(),
        translations: [aTranslatedLanguage()],
        states: [aState()],
    }, overrideObject);
}
export function aGetGlobalSettingsRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        state: aState(),
        host: aHost(),
        includeComponentSettings: true,
        languageKey: aLanguageKey(),
    }, overrideObject);
}
export function aGetGlobalSettingsResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        state: aState(),
        host: aHost(),
        settings: {},
        componentSettings: [aGetComponentSettingsItem()],
        translations: { translations: someString() },
    }, overrideObject);
}
export function aGetComponentSettingsItem(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        state: aState(),
        host: aHost(),
        componentId: someString({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '36' },
            ],
        }),
        settings: {},
        translations: { translations: someString() },
    }, overrideObject);
}
export function aSetGlobalSettingsForMigrationRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        settings: {},
        componentSettings: [aSetComponentSettingsItem()],
        state: aState(),
        host: aHost(),
        translations: [aTranslatedLanguage()],
        states: [aState()],
        appDefId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        instanceId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
export function aSetGlobalSettingsForMigrationResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        settings: {},
        componentSettings: [aSetComponentSettingsItem()],
        host: aHost(),
        translations: [aTranslatedLanguage()],
        states: [aState()],
    }, overrideObject);
}
export function aGetGlobalSettingsForMigrationRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        state: aState(),
        host: aHost(),
        includeComponentSettings: true,
        languageKey: aLanguageKey(),
        appDefId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        instanceId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
export function aGetGlobalSettingsForMigrationResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        state: aState(),
        host: aHost(),
        settings: {},
        componentSettings: [aGetComponentSettingsItem()],
        translations: [aTranslatedLanguage()],
    }, overrideObject);
}
export function aGetRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        externalAppId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        externalComponentId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        fields: someRepeated(),
    }, overrideObject);
}
export function aGetResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        appSettings: {},
        componentSettings: {},
    }, overrideObject);
}
export function aSetRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        externalId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        scope: aScope(),
        data: {},
    }, overrideObject);
}
export function aScope(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = Scope.APP; }
    return enumFirstItem;
}
export function aSetResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({}, overrideObject);
}
export function aUpdateRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        externalId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        scope: aScope(),
        fields: someRepeated(),
        data: {},
    }, overrideObject);
}
export function aUpdateResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        newExternalId: someString(),
    }, overrideObject);
}
function isCurrentBuilderWasPreviouslyCalled() {
    var callChainFunctionNames = stack().map(function (item) {
        return item.getFunctionName();
    });
    var callChainWithoutCurrentBuilder = callChainFunctionNames.splice(2);
    var currentBuilderName = callChainFunctionNames[1];
    return callChainWithoutCurrentBuilder.some(function (functionName) { return functionName === currentBuilderName; });
}
export * from '@wix/metro-common-builders';
