// @wix/bex-utils/@wix/ambassador-app-settings-v1-settings/build/es/meta.impl.js
import * as httpWixAppSettingsV1Settings from './http.impl';
export function setComponentSettings() {
    var payload = { componentId: ':componentId' };
    var getRequestOptions = httpWixAppSettingsV1Settings.setComponentSettings(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'PATCH',
        path: '/api/v1/settings/components/{componentId}',
        pathParams: { componentId: 'componentId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getComponentSettings() {
    var payload = { componentId: ':componentId' };
    var getRequestOptions = httpWixAppSettingsV1Settings.getComponentSettings(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/api/v1/settings/components/{componentId}',
        pathParams: { componentId: 'componentId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkGetComponentSettings() {
    var payload = {};
    var getRequestOptions = httpWixAppSettingsV1Settings.bulkGetComponentSettings(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/api/v1/settings/components/bulk',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function setGlobalSettings() {
    var payload = {};
    var getRequestOptions = httpWixAppSettingsV1Settings.setGlobalSettings(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'PATCH',
        path: '/api/v1/settings/instances',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getGlobalSettings() {
    var payload = {};
    var getRequestOptions = httpWixAppSettingsV1Settings.getGlobalSettings(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/api/v1/settings/instances',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function get() {
    var payload = {};
    var getRequestOptions = httpWixAppSettingsV1Settings.get(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/api/v2/settings',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function set() {
    var payload = { scope: ':scope', externalId: ':externalId' };
    var getRequestOptions = httpWixAppSettingsV1Settings.set(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'PUT',
        path: '/api/v2/settings/{scope}/{externalId}',
        pathParams: { scope: 'scope', externalId: 'externalId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function update() {
    var payload = { scope: ':scope', externalId: ':externalId' };
    var getRequestOptions = httpWixAppSettingsV1Settings.update(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'PATCH',
        path: '/api/v2/settings/{scope}/{externalId}',
        pathParams: { scope: 'scope', externalId: 'externalId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
