// @wix/ricos/dist/es/plugin-ai/exports/image-alt-text-input/alt-text-input.js
import { Box, FormField, InputArea, Text, TextButton } from "@wix/design-system";
import { SparklesSmall } from "@wix/wix-ui-icons-common";
import React, { useContext } from "react";
import { RicosContext } from "./../../../ricos-context/index.js";
import {
  AI_EXPORTS_ALT_TEXT_GENERATE_BUTTON,
  AI_EXPORTS_ALT_TEXT_INPUT_AREA
} from "../../data-hooks.js";
import { getImageUrl } from "../../utils/get-image-url.js";
import { useAiRequestService } from "../../utils/use-ai-request-service.js";
const AltTextInput = ({
  image,
  altText,
  onChange,
  config,
  tooltipAppendTo,
  logBi,
  disabled
}) => {
  const { t, localeContent } = useContext(RicosContext);
  const { fetchPrompt } = useAiRequestService(config);
  const [requestState, setRequestState] = React.useState({ type: "INIT" });
  const inputRef = React.useRef(null);
  const isDestroyed = React.useRef(false);
  React.useEffect(() => {
    isDestroyed.current = false;
    return () => {
      isDestroyed.current = true;
    };
  }, []);
  const generateAltText = async () => {
    if (!image) {
      return;
    }
    setRequestState({ type: "LOADING" });
    inputRef.current?.calculateComputedRows();
    const result = await fetchPrompt({
      promptSlug: "generate-alt-text",
      promptParams: {
        imageUrl: getImageUrl({ image }),
        siteLanguage: localeContent
      },
      additionalTags: "image alt text generation"
    });
    if (isDestroyed.current) {
      return;
    }
    if (result.type === "SUCCESS") {
      setRequestState({ type: "INIT" });
      logBi({
        dataHook: AI_EXPORTS_ALT_TEXT_GENERATE_BUTTON,
        value: result.results?.[0] ?? altText
      });
      onChange(result.results?.[0] ?? altText);
      inputRef.current?.calculateComputedRows();
      return;
    }
    setRequestState({ type: "ERROR", error: t("AIPlugin_Prompt_GenerateAltText_Error") });
  };
  return /* @__PURE__ */ React.createElement(
    FormField,
    {
      label: t("ImageSettings_Alt_Label"),
      infoContent: /* @__PURE__ */ React.createElement(Box, { direction: "vertical", gap: "SP1" }, /* @__PURE__ */ React.createElement(Text, { light: true, size: "small" }, t("ImageSettings_Alt_Label_Tooltip"))),
      infoTooltipProps: {
        appendTo: tooltipAppendTo ?? ((el) => Boolean(el.getAttribute("data-hook")?.includes("ricos-portal")))
      },
      suffix: image ? /* @__PURE__ */ React.createElement(Box, { direction: "vertical", gap: "SP1" }, /* @__PURE__ */ React.createElement(
        TextButton,
        {
          dataHook: AI_EXPORTS_ALT_TEXT_GENERATE_BUTTON,
          disabled: requestState.type === "LOADING" || disabled,
          size: "tiny",
          prefixIcon: /* @__PURE__ */ React.createElement(SparklesSmall, null),
          onClick: () => {
            logBi({ dataHook: AI_EXPORTS_ALT_TEXT_GENERATE_BUTTON });
            generateAltText();
          }
        },
        t("AIPlugin_ImagePlugin_AltText_Generate_Text")
      )) : void 0
    },
    /* @__PURE__ */ React.createElement(
      InputArea,
      {
        ref: inputRef,
        dataHook: AI_EXPORTS_ALT_TEXT_INPUT_AREA,
        placeholder: t("ImageSettings_Alt_Input_Placeholder"),
        onChange: (e) => {
          logBi({ dataHook: AI_EXPORTS_ALT_TEXT_INPUT_AREA, value: e.target.value });
          onChange(e.target.value);
        },
        value: requestState.type === "LOADING" ? t("AIPlugin_ImagePlugin_AltText_InProgress") : altText,
        rows: 3,
        readOnly: requestState.type === "LOADING",
        status: requestState.type === "ERROR" ? "error" : requestState.type === "LOADING" ? "loading" : void 0,
        statusMessage: requestState.type === "ERROR" ? requestState.error : void 0,
        tooltipPlacement: "top-end",
        size: "small",
        disabled
      }
    )
  );
};
const createAltTextInput = (config) => {
  return (props) => /* @__PURE__ */ React.createElement(AltTextInput, { ...props, config });
};
export {
  createAltTextInput
};
