// @wix/auto_sdk_faq_category/build/internal/es/index.typings.mjs
// src/faq-category-v2-category-category.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/faq-category-v2-category-category.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKFloatToRESTFloat } from "@wix/sdk-runtime/transformations/float";
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixFaqCategoryV2CategoryServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/category",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/category",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/category",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/faq-server/v2/categories",
        destPath: "/v2/categories"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-categories",
        destPath: "/v2/draft-categories"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/faq-server/v2/categories",
        destPath: "/v2/categories"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-categories",
        destPath: "/v2/draft-categories"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/faq/v2/categories",
        destPath: "/v2/categories"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-categories",
        destPath: "/v2/draft-categories"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/faq/v2/categories",
        destPath: "/v2/categories"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/faq-server/v2/categories",
        destPath: "/v2/categories"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-categories",
        destPath: "/v2/draft-categories"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/faq-server/v2/categories",
        destPath: "/v2/categories"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-categories",
        destPath: "/v2/draft-categories"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/faq-server/v2/categories",
        destPath: "/v2/categories"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-categories",
        destPath: "/v2/draft-categories"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/faq-server/v2/categories",
        destPath: "/v2/categories"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-categories",
        destPath: "/v2/draft-categories"
      }
    ],
    _: [
      {
        srcPath: "/_api/faq-server/v2/categories",
        destPath: "/v2/categories"
      }
    ],
    "dev._base_domain_": [
      {
        srcPath: "/_api/faq-server/v2/categories",
        destPath: "/v2/categories"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-categories",
        destPath: "/v2/draft-categories"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/faq-server/v2/draft-categories",
        destPath: "/v2/draft-categories"
      },
      {
        srcPath: "/_api/faq-server/v2/categories",
        destPath: "/v2/categories"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_faq_category";
function createCategory(payload) {
  function __createCategory({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "category.createdDate" },
          { path: "category.updatedDate" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [{ path: "category.sortOrder" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.faq.category.v2.category",
      method: "POST",
      methodFqn: "wix.faq.category.v2.CategoryService.CreateCategory",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqCategoryV2CategoryServiceUrl({
        protoPath: "/v2/categories",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "category.createdDate" },
            { path: "category.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [{ path: "category.sortOrder" }]
        }
      ])
    };
    return metadata;
  }
  return __createCategory;
}
function getCategory(payload) {
  function __getCategory({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.category.v2.category",
      method: "GET",
      methodFqn: "wix.faq.category.v2.CategoryService.GetCategory",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqCategoryV2CategoryServiceUrl({
        protoPath: "/v2/categories/{categoryId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "category.createdDate" },
            { path: "category.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [{ path: "category.sortOrder" }]
        }
      ])
    };
    return metadata;
  }
  return __getCategory;
}
function updateCategory(payload) {
  function __updateCategory({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "category.createdDate" },
          { path: "category.updatedDate" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [{ path: "category.sortOrder" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.faq.category.v2.category",
      method: "PATCH",
      methodFqn: "wix.faq.category.v2.CategoryService.UpdateCategory",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqCategoryV2CategoryServiceUrl({
        protoPath: "/v2/categories/{category.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "category.createdDate" },
            { path: "category.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [{ path: "category.sortOrder" }]
        }
      ])
    };
    return metadata;
  }
  return __updateCategory;
}
function deleteCategory(payload) {
  function __deleteCategory({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.category.v2.category",
      method: "DELETE",
      methodFqn: "wix.faq.category.v2.CategoryService.DeleteCategory",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqCategoryV2CategoryServiceUrl({
        protoPath: "/v2/categories/{categoryId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteCategory;
}
function listCategories(payload) {
  function __listCategories({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.category.v2.category",
      method: "GET",
      methodFqn: "wix.faq.category.v2.CategoryService.ListCategories",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqCategoryV2CategoryServiceUrl({
        protoPath: "/v2/categories",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "categories.createdDate" },
            { path: "categories.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [{ path: "categories.sortOrder" }]
        }
      ])
    };
    return metadata;
  }
  return __listCategories;
}
function queryCategories(payload) {
  function __queryCategories({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.category.v2.category",
      method: "POST",
      methodFqn: "wix.faq.category.v2.CategoryService.QueryCategories",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqCategoryV2CategoryServiceUrl({
        protoPath: "/v2/categories/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "categories.createdDate" },
            { path: "categories.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [{ path: "categories.sortOrder" }]
        }
      ])
    };
    return metadata;
  }
  return __queryCategories;
}
function updateExtendedFields(payload) {
  function __updateExtendedFields({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.category.v2.category",
      method: "POST",
      methodFqn: "wix.faq.category.v2.CategoryService.UpdateExtendedFields",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqCategoryV2CategoryServiceUrl({
        protoPath: "/v2/categories/{id}/update-extended-fields",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "category.createdDate" },
            { path: "category.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [{ path: "category.sortOrder" }]
        }
      ])
    };
    return metadata;
  }
  return __updateExtendedFields;
}

// src/faq-category-v2-category-category.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createCategory2(category) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ category });
  const reqOpts = createCategory(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.category;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { category: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["category"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getCategory2(categoryId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    categoryId
  });
  const reqOpts = getCategory(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.category;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { categoryId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["categoryId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateCategory2(_id, category) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    category: { ...category, id: _id }
  });
  const reqOpts = updateCategory(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.category;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { category: "$[1]" },
        explicitPathsToArguments: { "category.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "category"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteCategory2(categoryId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    categoryId
  });
  const reqOpts = deleteCategory(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { categoryId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["categoryId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listCategories2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    paging: options?.paging
  });
  const reqOpts = listCategories(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { paging: "$[0].paging" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryCategories2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryCategories(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [])
      );
      return {
        items: transformedData?.categories,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function updateExtendedFields2(_id, namespace, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    id: _id,
    namespace,
    namespaceData: options?.namespaceData
  });
  const reqOpts = updateExtendedFields(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          id: "$[0]",
          namespace: "$[1]",
          namespaceData: "$[2].namespaceData"
        },
        singleArgumentUnchanged: false
      },
      ["_id", "namespace", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
export {
  SortOrder,
  WebhookIdentityType,
  createCategory2 as createCategory,
  deleteCategory2 as deleteCategory,
  getCategory2 as getCategory,
  listCategories2 as listCategories,
  queryCategories2 as queryCategories,
  updateCategory2 as updateCategory,
  updateExtendedFields2 as updateExtendedFields
};
