// @wix/auto_sdk_restaurants_item-modifier-groups/build/es/meta.mjs
// src/restaurants-menus-v1-item-modifier-group-item-modifier-groups.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl(opts) {
  const domainToMappings = {
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-modifier-group",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers-groups/delete",
        destPath: "/v1/bulk/modifiers-groups/delete"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers-groups/update",
        destPath: "/v1/bulk/modifiers-groups/update"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifier-groups/create",
        destPath: "/v1/bulk/modifier-groups/create"
      },
      {
        srcPath: "/restaurants/menus/v1/modifier-groups",
        destPath: "/v1/modifier-groups"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-modifier-group",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-modifier-group",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-modifier-group",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-modifier-group",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-modifier-group",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-modifier-group",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers-groups/delete",
        destPath: "/v1/bulk/modifiers-groups/delete"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers-groups/update",
        destPath: "/v1/bulk/modifiers-groups/update"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifier-groups/create",
        destPath: "/v1/bulk/modifier-groups/create"
      },
      {
        srcPath: "/restaurants/menus/v1/modifier-groups",
        destPath: "/v1/modifier-groups"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-modifier-group",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-modifier-group",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/item-modifier-group",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifier-groups/create",
        destPath: "/v1/bulk/modifier-groups/create"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers-groups/delete",
        destPath: "/v1/bulk/modifiers-groups/delete"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers-groups/update",
        destPath: "/v1/bulk/modifiers-groups/update"
      },
      {
        srcPath: "/restaurants/menus/v1/modifier-groups",
        destPath: "/v1/modifier-groups"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-modifier-group",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_item-modifier-groups";
function createModifierGroup(payload) {
  function __createModifierGroup({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "modifierGroup.createdDate" },
          { path: "modifierGroup.updatedDate" },
          { path: "modifierGroup.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier_group",
      method: "POST",
      methodFqn: "wix.restaurants.menus.v1.ModifierGroupsService.CreateModifierGroup",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl({
        protoPath: "/v1/modifier-groups",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "modifierGroup.createdDate" },
            { path: "modifierGroup.updatedDate" },
            { path: "modifierGroup.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createModifierGroup;
}
function getModifierGroup(payload) {
  function __getModifierGroup({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier_group",
      method: "GET",
      methodFqn: "wix.restaurants.menus.v1.ModifierGroupsService.GetModifierGroup",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl({
        protoPath: "/v1/modifier-groups/{modifierGroupId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "modifierGroup.createdDate" },
            { path: "modifierGroup.updatedDate" },
            { path: "modifierGroup.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getModifierGroup;
}
function listModifierGroups(payload) {
  function __listModifierGroups({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier_group",
      method: "GET",
      methodFqn: "wix.restaurants.menus.v1.ModifierGroupsService.ListModifierGroups",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl({
        protoPath: "/v1/modifier-groups",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "modifierGroups.createdDate" },
            { path: "modifierGroups.updatedDate" },
            { path: "modifierGroups.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listModifierGroups;
}
function queryModifierGroups(payload) {
  function __queryModifierGroups({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier_group",
      method: "POST",
      methodFqn: "wix.restaurants.menus.v1.ModifierGroupsService.QueryModifierGroups",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl({
        protoPath: "/v1/modifier-groups/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "modifierGroups.createdDate" },
            { path: "modifierGroups.updatedDate" },
            { path: "modifierGroups.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryModifierGroups;
}
function countModifierGroups(payload) {
  function __countModifierGroups({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier_group",
      method: "POST",
      methodFqn: "wix.restaurants.menus.v1.ModifierGroupsService.CountModifierGroups",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl({
        protoPath: "/v1/modifier-groups/count",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __countModifierGroups;
}
function updateModifierGroup(payload) {
  function __updateModifierGroup({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "modifierGroup.createdDate" },
          { path: "modifierGroup.updatedDate" },
          { path: "modifierGroup.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier_group",
      method: "PATCH",
      methodFqn: "wix.restaurants.menus.v1.ModifierGroupsService.UpdateModifierGroup",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl({
        protoPath: "/v1/modifier-groups/{modifierGroup.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "modifierGroup.createdDate" },
            { path: "modifierGroup.updatedDate" },
            { path: "modifierGroup.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateModifierGroup;
}
function deleteModifierGroup(payload) {
  function __deleteModifierGroup({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier_group",
      method: "DELETE",
      methodFqn: "wix.restaurants.menus.v1.ModifierGroupsService.DeleteModifierGroup",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl({
        protoPath: "/v1/modifier-groups/{modifierGroupId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteModifierGroup;
}
function bulkCreateModifierGroups(payload) {
  function __bulkCreateModifierGroups({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "modifierGroups.createdDate" },
          { path: "modifierGroups.updatedDate" },
          { path: "modifierGroups.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier_group",
      method: "POST",
      methodFqn: "wix.restaurants.menus.v1.ModifierGroupsService.BulkCreateModifierGroups",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl({
        protoPath: "/v1/bulk/modifier-groups/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.modifierGroup.createdDate" },
            { path: "results.modifierGroup.updatedDate" },
            {
              path: "results.modifierGroup.externalReferenceInfo.lastSyncDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateModifierGroups;
}
function bulkUpdateModifierGroups(payload) {
  function __bulkUpdateModifierGroups({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "modifierGroups.mask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "modifierGroups.modifierGroup.createdDate" },
          { path: "modifierGroups.modifierGroup.updatedDate" },
          {
            path: "modifierGroups.modifierGroup.externalReferenceInfo.lastSyncDate"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier_group",
      method: "POST",
      methodFqn: "wix.restaurants.menus.v1.ModifierGroupsService.BulkUpdateModifierGroups",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl({
        protoPath: "/v1/bulk/modifiers-groups/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.modifierGroup.createdDate" },
            { path: "results.modifierGroup.updatedDate" },
            {
              path: "results.modifierGroup.externalReferenceInfo.lastSyncDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateModifierGroups;
}
function bulkDeleteModifierGroups(payload) {
  function __bulkDeleteModifierGroups({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier_group",
      method: "DELETE",
      methodFqn: "wix.restaurants.menus.v1.ModifierGroupsService.BulkDeleteModifierGroups",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl({
        protoPath: "/v1/bulk/modifiers-groups/delete",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __bulkDeleteModifierGroups;
}

// src/restaurants-menus-v1-item-modifier-group-item-modifier-groups.types.ts
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/restaurants-menus-v1-item-modifier-group-item-modifier-groups.meta.ts
function createModifierGroup2() {
  const payload = {};
  const getRequestOptions = createModifierGroup(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/modifier-groups",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getModifierGroup2() {
  const payload = { modifierGroupId: ":modifierGroupId" };
  const getRequestOptions = getModifierGroup(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/modifier-groups/{modifierGroupId}",
    pathParams: { modifierGroupId: "modifierGroupId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listModifierGroups2() {
  const payload = {};
  const getRequestOptions = listModifierGroups(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/modifier-groups",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryModifierGroups2() {
  const payload = {};
  const getRequestOptions = queryModifierGroups(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/modifier-groups/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function countModifierGroups2() {
  const payload = {};
  const getRequestOptions = countModifierGroups(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/modifier-groups/count",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateModifierGroup2() {
  const payload = { modifierGroup: { id: ":modifierGroupId" } };
  const getRequestOptions = updateModifierGroup(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/modifier-groups/{modifierGroup.id}",
    pathParams: { modifierGroupId: "modifierGroupId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteModifierGroup2() {
  const payload = { modifierGroupId: ":modifierGroupId" };
  const getRequestOptions = deleteModifierGroup(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/modifier-groups/{modifierGroupId}",
    pathParams: { modifierGroupId: "modifierGroupId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkCreateModifierGroups2() {
  const payload = {};
  const getRequestOptions = bulkCreateModifierGroups(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/modifier-groups/create",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateModifierGroups2() {
  const payload = {};
  const getRequestOptions = bulkUpdateModifierGroups(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/modifiers-groups/update",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkDeleteModifierGroups2() {
  const payload = {};
  const getRequestOptions = bulkDeleteModifierGroups(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/bulk/modifiers-groups/delete",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  SortOrder as SortOrderOriginal,
  WebhookIdentityType as WebhookIdentityTypeOriginal,
  bulkCreateModifierGroups2 as bulkCreateModifierGroups,
  bulkDeleteModifierGroups2 as bulkDeleteModifierGroups,
  bulkUpdateModifierGroups2 as bulkUpdateModifierGroups,
  countModifierGroups2 as countModifierGroups,
  createModifierGroup2 as createModifierGroup,
  deleteModifierGroup2 as deleteModifierGroup,
  getModifierGroup2 as getModifierGroup,
  listModifierGroups2 as listModifierGroups,
  queryModifierGroups2 as queryModifierGroups,
  updateModifierGroup2 as updateModifierGroup
};
