// @wix/auto_sdk_restaurants_fulfillemt-methods/build/es/index.typings.mjs
// src/restaurants-fulfillments-v3-fulfillment-method-fulfillemt-methods.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/restaurants-fulfillments-v3-fulfillment-method-fulfillemt-methods.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKFloatToRESTFloat } from "@wix/sdk-runtime/transformations/float";
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressRestaurantsFulfillmentsV3FulfillmentMethodsUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/v3/fulfillment-methods",
        destPath: "/v3/fulfillment-methods"
      }
    ],
    _: [
      {
        srcPath: "/_api/fulfillment-methods",
        destPath: "/v3/fulfillment-methods"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/fulfillment-methods",
        destPath: "/v3/fulfillment-methods"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/fulfillment-methods",
        destPath: "/v3/fulfillment-methods"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_fulfillemt-methods";
function createFulfillmentMethod(payload) {
  function __createFulfillmentMethod({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          {
            path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.radiusDetails.geocode.latitude"
          },
          {
            path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.radiusDetails.geocode.longitude"
          },
          {
            path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.customDetails.geocodes.latitude"
          },
          {
            path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.customDetails.geocodes.longitude"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.fulfillments.v3.fulfillment_method",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.fulfillments.v3.FulfillmentMethods.CreateFulfillmentMethod",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsFulfillmentsV3FulfillmentMethodsUrl({
        protoPath: "/v3/fulfillment-methods",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.radiusDetails.geocode.latitude"
            },
            {
              path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.radiusDetails.geocode.longitude"
            },
            {
              path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.customDetails.geocodes.latitude"
            },
            {
              path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.customDetails.geocodes.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createFulfillmentMethod;
}
function getFulfillmentMethod(payload) {
  function __getFulfillmentMethod({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.fulfillments.v3.fulfillment_method",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.fulfillments.v3.FulfillmentMethods.GetFulfillmentMethod",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsFulfillmentsV3FulfillmentMethodsUrl({
        protoPath: "/v3/fulfillment-methods/{fulfillmentMethodId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.radiusDetails.geocode.latitude"
            },
            {
              path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.radiusDetails.geocode.longitude"
            },
            {
              path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.customDetails.geocodes.latitude"
            },
            {
              path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.customDetails.geocodes.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getFulfillmentMethod;
}
function updateFulfillmentMethod(payload) {
  function __updateFulfillmentMethod({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          {
            path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.radiusDetails.geocode.latitude"
          },
          {
            path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.radiusDetails.geocode.longitude"
          },
          {
            path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.customDetails.geocodes.latitude"
          },
          {
            path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.customDetails.geocodes.longitude"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.fulfillments.v3.fulfillment_method",
      method: "PATCH",
      methodFqn: "com.wixpress.restaurants.fulfillments.v3.FulfillmentMethods.UpdateFulfillmentMethod",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsFulfillmentsV3FulfillmentMethodsUrl({
        protoPath: "/v3/fulfillment-methods/{fulfillmentMethod.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.radiusDetails.geocode.latitude"
            },
            {
              path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.radiusDetails.geocode.longitude"
            },
            {
              path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.customDetails.geocodes.latitude"
            },
            {
              path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.customDetails.geocodes.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateFulfillmentMethod;
}
function deleteFulfillmentMethod(payload) {
  function __deleteFulfillmentMethod({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.fulfillments.v3.fulfillment_method",
      method: "DELETE",
      methodFqn: "com.wixpress.restaurants.fulfillments.v3.FulfillmentMethods.DeleteFulfillmentMethod",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsFulfillmentsV3FulfillmentMethodsUrl({
        protoPath: "/v3/fulfillment-methods/{fulfillmentMethodId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteFulfillmentMethod;
}
function listFulfillmentMethods(payload) {
  function __listFulfillmentMethods({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.fulfillments.v3.fulfillment_method",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.fulfillments.v3.FulfillmentMethods.ListFulfillmentMethods",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsFulfillmentsV3FulfillmentMethodsUrl({
        protoPath: "/v3/fulfillment-methods",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "fulfillmentMethods.deliveryInfo.restaurant.deliveryRange.radiusDetails.geocode.latitude"
            },
            {
              path: "fulfillmentMethods.deliveryInfo.restaurant.deliveryRange.radiusDetails.geocode.longitude"
            },
            {
              path: "fulfillmentMethods.deliveryInfo.restaurant.deliveryRange.customDetails.geocodes.latitude"
            },
            {
              path: "fulfillmentMethods.deliveryInfo.restaurant.deliveryRange.customDetails.geocodes.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listFulfillmentMethods;
}

// src/restaurants-fulfillments-v3-fulfillment-method-fulfillemt-methods.universal.ts
var FulfillmentMethodType = /* @__PURE__ */ ((FulfillmentMethodType2) => {
  FulfillmentMethodType2["UNSPECIFIED_FULFILLMENT_TYPE"] = "UNSPECIFIED_FULFILLMENT_TYPE";
  FulfillmentMethodType2["PICKUP"] = "PICKUP";
  FulfillmentMethodType2["DELIVERY"] = "DELIVERY";
  FulfillmentMethodType2["DINE_IN"] = "DINE_IN";
  return FulfillmentMethodType2;
})(FulfillmentMethodType || {});
var DayOfWeek = /* @__PURE__ */ ((DayOfWeek2) => {
  DayOfWeek2["MON"] = "MON";
  DayOfWeek2["TUE"] = "TUE";
  DayOfWeek2["WED"] = "WED";
  DayOfWeek2["THU"] = "THU";
  DayOfWeek2["FRI"] = "FRI";
  DayOfWeek2["SAT"] = "SAT";
  DayOfWeek2["SUN"] = "SUN";
  return DayOfWeek2;
})(DayOfWeek || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNDEFINED"] = "UNDEFINED";
  Type2["RADIUS"] = "RADIUS";
  Type2["POSTAL_CODE"] = "POSTAL_CODE";
  Type2["CUSTOM"] = "CUSTOM";
  return Type2;
})(Type || {});
var TimeDisplayUnit = /* @__PURE__ */ ((TimeDisplayUnit2) => {
  TimeDisplayUnit2["UNDEFINED"] = "UNDEFINED";
  TimeDisplayUnit2["MINUTES"] = "MINUTES";
  TimeDisplayUnit2["HOURS"] = "HOURS";
  TimeDisplayUnit2["DAYS"] = "DAYS";
  return TimeDisplayUnit2;
})(TimeDisplayUnit || {});
var BestDeliveryByCategory = /* @__PURE__ */ ((BestDeliveryByCategory2) => {
  BestDeliveryByCategory2["DELIVERY_FEE"] = "DELIVERY_FEE";
  return BestDeliveryByCategory2;
})(BestDeliveryByCategory || {});
async function createFulfillmentMethod2(fulfillmentMethod) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    fulfillmentMethod
  });
  const reqOpts = createFulfillmentMethod(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.fulfillmentMethod;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { fulfillmentMethod: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["fulfillmentMethod"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getFulfillmentMethod2(fulfillmentMethodId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    fulfillmentMethodId
  });
  const reqOpts = getFulfillmentMethod(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.fulfillmentMethod;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { fulfillmentMethodId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["fulfillmentMethodId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateFulfillmentMethod2(_id, fulfillmentMethod) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    fulfillmentMethod: { ...fulfillmentMethod, id: _id }
  });
  const reqOpts = updateFulfillmentMethod(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.fulfillmentMethod;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { fulfillmentMethod: "$[1]" },
        explicitPathsToArguments: { "fulfillmentMethod.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "fulfillmentMethod"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteFulfillmentMethod2(fulfillmentMethodId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    fulfillmentMethodId
  });
  const reqOpts = deleteFulfillmentMethod(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { fulfillmentMethodId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["fulfillmentMethodId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listFulfillmentMethods2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    restaurantLocationIds: options?.restaurantLocationIds
  });
  const reqOpts = listFulfillmentMethods(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          restaurantLocationIds: "$[0].restaurantLocationIds"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
export {
  BestDeliveryByCategory,
  DayOfWeek,
  FulfillmentMethodType,
  TimeDisplayUnit,
  Type,
  createFulfillmentMethod2 as createFulfillmentMethod,
  deleteFulfillmentMethod2 as deleteFulfillmentMethod,
  getFulfillmentMethod2 as getFulfillmentMethod,
  listFulfillmentMethods2 as listFulfillmentMethods,
  updateFulfillmentMethod2 as updateFulfillmentMethod
};
