// @wix/metro-common-builders/dist/src/value-generator/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FunctionNames = exports.GeneratorType = void 0;
var GeneratorType;
(function (GeneratorType) {
    GeneratorType["NUMBER"] = "number";
    GeneratorType["LONG"] = "long";
    GeneratorType["FLOAT"] = "float";
    GeneratorType["STRING"] = "string";
    GeneratorType["REPEATED"] = "repeated";
    GeneratorType["DATE"] = "date";
    GeneratorType["DURATION"] = "duration";
    GeneratorType["BYTES"] = "bytes";
    GeneratorType["OTHER"] = "other";
    GeneratorType["ANY"] = "any";
})(GeneratorType = exports.GeneratorType || (exports.GeneratorType = {}));
var FunctionNames;
(function (FunctionNames) {
    FunctionNames["MERGE_DEEP"] = "mergeDeep";
    FunctionNames["SOME_LONG"] = "someLong";
    FunctionNames["SOME_NUMBER"] = "someNumber";
    FunctionNames["SOME_FLOAT"] = "someFloat";
    FunctionNames["SOME_STRING"] = "someString";
    FunctionNames["SOME_ANY"] = "someAny";
    FunctionNames["SOME_REPEATED"] = "someRepeated";
    FunctionNames["SOME_DATE"] = "someDate";
    FunctionNames["SOME_DURATION"] = "someDuration";
    FunctionNames["SOME_BYTES"] = "someBytes";
    FunctionNames["GET_GENERATOR_TYPE"] = "getGeneratorTypeForField";
})(FunctionNames = exports.FunctionNames || (exports.FunctionNames = {}));
