// @wix/metro-common-builders/dist/src/value-generator/validation-extractor.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValidationExtractor = void 0;
var domain_1 = require("../domain");
var value_generator_utils_1 = require("./value-generator-utils");
var ValidationExtractor = /** @class */ (function () {
    function ValidationExtractor(validations) {
        this.validations = validations;
    }
    ValidationExtractor.prototype.getValidationOfType = function (validationType) {
        return this.validations.find(function (validation) { return validation.type === validationType; });
    };
    ValidationExtractor.prototype.getValidationValueOfType = function (validationType) {
        var _a;
        return (_a = this.getValidationOfType(validationType)) === null || _a === void 0 ? void 0 : _a.value;
    };
    ValidationExtractor.prototype.getNumberValidations = function () {
        var max = this.getValidationValueOfType(domain_1.ValidationType.MAX);
        var min = this.getValidationValueOfType(domain_1.ValidationType.MIN);
        var _a = this.getValidationValueOfType(domain_1.ValidationType.DECIMAL_VALUE) || {}, lowerThanEqual = _a.lte, lowerThan = _a.lt, greaterThanEqual = _a.gte, greaterThan = _a.gt, maxScale = _a.maxScale;
        return {
            min: (0, value_generator_utils_1.toNumber)(min),
            max: (0, value_generator_utils_1.toNumber)(max),
            lowerThanEqual: (0, value_generator_utils_1.toNumber)(lowerThanEqual),
            greaterThanEqual: (0, value_generator_utils_1.toNumber)(greaterThanEqual),
            lowerThan: (0, value_generator_utils_1.toNumber)(lowerThan),
            greaterThan: (0, value_generator_utils_1.toNumber)(greaterThan),
            maxScale: (0, value_generator_utils_1.toNumber)(maxScale),
        };
    };
    ValidationExtractor.prototype.getRepeatedValidations = function () {
        var maxSize = this.getValidationValueOfType(domain_1.ValidationType.MAX_SIZE);
        var minSize = this.getValidationValueOfType(domain_1.ValidationType.MIN_SIZE);
        return { minSize: minSize, maxSize: maxSize };
    };
    ValidationExtractor.prototype.getStringValidations = function () {
        var maxLength = this.getValidationValueOfType(domain_1.ValidationType.MAX_LENGTH);
        var minLength = this.getValidationValueOfType(domain_1.ValidationType.MIN_LENGTH);
        var format = this.getValidationValueOfType(domain_1.ValidationType.FORMAT);
        return { minLength: minLength, maxLength: maxLength, format: format };
    };
    return ValidationExtractor;
}());
exports.ValidationExtractor = ValidationExtractor;
