// @wix/metro-common-builders/dist/src/value-generator/value-generator-utils.js
"use strict";
var _a, _b;
Object.defineProperty(exports, "__esModule", { value: true });
exports.wrapperToBusinessSchemaType = exports.typeToGeneratorType = exports.generatorTypeToFunctionName = exports.toNumber = exports.padArray = exports.getAdditionOrSubtractionValue = exports.decrement = exports.increment = exports.applyScale = void 0;
var domain_1 = require("../domain");
var constants_1 = require("./constants");
function applyScale(maxScale, num) {
    return maxScale ? parseFloat(num.toFixed(maxScale)) : num;
}
exports.applyScale = applyScale;
function increment(maxScale, num) {
    return num + getAdditionOrSubtractionValue(maxScale);
}
exports.increment = increment;
function decrement(maxScale, num) {
    return num - getAdditionOrSubtractionValue(maxScale);
}
exports.decrement = decrement;
function getAdditionOrSubtractionValue(scale) {
    // if float - generate according to scale, eg, 0.01, 0.001... else return 1
    return scale ? parseFloat("0.".concat('0'.repeat(scale - 1), "1")) : 1;
}
exports.getAdditionOrSubtractionValue = getAdditionOrSubtractionValue;
function padArray(array, minSize, fillValue) {
    return Object.assign(new Array(minSize).fill(fillValue), array);
}
exports.padArray = padArray;
function toNumber(number) {
    return typeof number === 'string' ? parseFloat(number) : number;
}
exports.toNumber = toNumber;
function generatorTypeToFunctionName(generatorType) {
    return GENERATOR_TYPE_TO_FUNCTION[generatorType];
}
exports.generatorTypeToFunctionName = generatorTypeToFunctionName;
var GENERATOR_TYPE_TO_FUNCTION = (_a = {},
    _a[constants_1.GeneratorType.NUMBER] = constants_1.FunctionNames.SOME_NUMBER,
    _a[constants_1.GeneratorType.LONG] = constants_1.FunctionNames.SOME_LONG,
    _a[constants_1.GeneratorType.FLOAT] = constants_1.FunctionNames.SOME_FLOAT,
    _a[constants_1.GeneratorType.STRING] = constants_1.FunctionNames.SOME_STRING,
    _a[constants_1.GeneratorType.REPEATED] = constants_1.FunctionNames.SOME_REPEATED,
    _a[constants_1.GeneratorType.DATE] = constants_1.FunctionNames.SOME_DATE,
    _a[constants_1.GeneratorType.DURATION] = constants_1.FunctionNames.SOME_DURATION,
    _a[constants_1.GeneratorType.BYTES] = constants_1.FunctionNames.SOME_BYTES,
    _a[constants_1.GeneratorType.ANY] = constants_1.FunctionNames.SOME_ANY,
    _a);
function typeToGeneratorType(type) {
    return TYPE_TO_GENERATOR_TYPE[type];
}
exports.typeToGeneratorType = typeToGeneratorType;
var TYPE_TO_GENERATOR_TYPE = (_b = {},
    _b[domain_1.Type.DOUBLE] = constants_1.GeneratorType.NUMBER,
    _b[domain_1.Type.FLOAT] = constants_1.GeneratorType.FLOAT,
    _b[domain_1.Type.INT32] = constants_1.GeneratorType.NUMBER,
    _b[domain_1.Type.UINT32] = constants_1.GeneratorType.NUMBER,
    _b[domain_1.Type.SINT32] = constants_1.GeneratorType.NUMBER,
    _b[domain_1.Type.FIXED32] = constants_1.GeneratorType.FLOAT,
    _b[domain_1.Type.SFIXED32] = constants_1.GeneratorType.FLOAT,
    _b[domain_1.Type.INT64] = constants_1.GeneratorType.STRING,
    _b[domain_1.Type.FIXED64] = constants_1.GeneratorType.STRING,
    _b[domain_1.Type.SFIXED64] = constants_1.GeneratorType.STRING,
    _b[domain_1.Type.SINT64] = constants_1.GeneratorType.STRING,
    _b[domain_1.Type.UINT64] = constants_1.GeneratorType.STRING,
    _b[domain_1.Type.STRING] = constants_1.GeneratorType.STRING,
    _b.FIELD_MASK = constants_1.GeneratorType.REPEATED,
    _b.LIST_VALUE = constants_1.GeneratorType.REPEATED,
    _b[domain_1.Type.BOOL] = constants_1.GeneratorType.OTHER,
    _b[domain_1.Type.BYTES] = constants_1.GeneratorType.BYTES,
    _b.NULL = constants_1.GeneratorType.OTHER,
    _b.DURATION = constants_1.GeneratorType.DURATION,
    _b.DATE = constants_1.GeneratorType.DATE,
    _b.ANY = constants_1.GeneratorType.ANY,
    _b);
function wrapperToBusinessSchemaType(protoType) {
    return WRAPPER_TO_BUSINESS_SCHEMA[protoType];
}
exports.wrapperToBusinessSchemaType = wrapperToBusinessSchemaType;
var WRAPPER_TO_BUSINESS_SCHEMA = {
    'google.protobuf.DoubleValue': domain_1.Type.DOUBLE,
    'google.protobuf.FloatValue': domain_1.Type.FLOAT,
    'google.protobuf.Int32Value': domain_1.Type.INT32,
    'google.protobuf.UInt32Value': domain_1.Type.UINT32,
    'google.protobuf.Int64Value': domain_1.Type.INT64,
    'google.protobuf.UInt64Value': domain_1.Type.UINT64,
    'google.protobuf.BoolValue': domain_1.Type.BOOL,
    'google.protobuf.StringValue': domain_1.Type.STRING,
    'google.protobuf.BytesValue': domain_1.Type.BYTES,
    'google.protobuf.NullValue': 'NULL',
    'google.protobuf.Struct': 'ANY',
    // TODO - remove when rich content create V2 without copy/paste google types
    'wix.rich_content.v1.Struct': 'ANY',
    'wix.rich_content.v1.Value': 'ANY',
    'google.protobuf.Duration': 'DURATION',
    'google.protobuf.Timestamp': 'DATE',
    'google.protobuf.Value': 'ANY',
    'google.protobuf.ListValue': 'LIST_VALUE',
    'google.protobuf.FieldMask': 'FIELD_MASK',
};
