// @wix/metro-common-builders/dist/src/value-generator/value-generator.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getGeneratorFunctionNameForGeneratorType = exports.someAny = exports.someBytes = exports.someDate = exports.someDuration = exports.someRepeated = exports.someFloat = exports.someLong = exports.someNumber = exports.someString = exports.getGeneratorTypeForFieldType = exports.getGeneratorTypeForField = void 0;
var tslib_1 = require("tslib");
var long_1 = tslib_1.__importDefault(require("long"));
var domain_1 = require("../domain");
var constant_values_1 = require("./constant-values");
var validation_extractor_1 = require("./validation-extractor");
var value_generator_utils_1 = require("./value-generator-utils");
function getGeneratorTypeForField(field) {
    return (0, value_generator_utils_1.typeToGeneratorType)(getTypeForGenerator(field));
}
exports.getGeneratorTypeForField = getGeneratorTypeForField;
function getGeneratorTypeForFieldType(filedType) {
    return (0, value_generator_utils_1.typeToGeneratorType)(filedType);
}
exports.getGeneratorTypeForFieldType = getGeneratorTypeForFieldType;
function someString(_a) {
    var _b;
    var _c = _a === void 0 ? {
        validations: [],
        isUnsigned: false,
    } : _a, validations = _c.validations, isUnsigned = _c.isUnsigned;
    if (!(validations === null || validations === void 0 ? void 0 : validations.length)) {
        return constant_values_1.CONSTANT_VALUES.STRING;
    }
    var validationExtractor = new validation_extractor_1.ValidationExtractor(validations);
    var _d = validationExtractor.getStringValidations(), minLength = _d.minLength, maxLength = _d.maxLength, format = _d.format;
    var decimalValidations = validationExtractor.getValidationOfType(domain_1.ValidationType.DECIMAL_VALUE);
    if (decimalValidations) {
        var scale = ((_b = decimalValidations.value) === null || _b === void 0 ? void 0 : _b.maxScale) || 0;
        return someNumber({ validations: validations, isUnsigned: isUnsigned }).toFixed(scale);
    }
    if (format) {
        return constant_values_1.FORMAT_VALUES[format];
    }
    var shouldPadStringEndWithValues = minLength && constant_values_1.CONSTANT_VALUES.STRING.length < minLength;
    if (shouldPadStringEndWithValues) {
        return constant_values_1.CONSTANT_VALUES.STRING.padEnd(minLength, constant_values_1.CONSTANT_VALUES.STRING);
    }
    var shouldShortenString = maxLength && constant_values_1.CONSTANT_VALUES.STRING.length > maxLength;
    if (shouldShortenString) {
        return constant_values_1.CONSTANT_VALUES.STRING.substring(0, maxLength);
    }
    return constant_values_1.CONSTANT_VALUES.STRING;
}
exports.someString = someString;
function someNumber(_a) {
    var _b = _a === void 0 ? {
        validations: [],
        isUnsigned: false,
    } : _a, validations = _b.validations, isUnsigned = _b.isUnsigned, defaultValue = _b.defaultValue;
    var actualDefaultValue = defaultValue || (isUnsigned ? constant_values_1.CONSTANT_VALUES.UINT : constant_values_1.CONSTANT_VALUES.INT);
    if (!(validations === null || validations === void 0 ? void 0 : validations.length)) {
        return actualDefaultValue;
    }
    var validationExtractor = new validation_extractor_1.ValidationExtractor(validations);
    var _c = validationExtractor.getNumberValidations(), min = _c.min, max = _c.max, lowerThanEqual = _c.lowerThanEqual, lowerThan = _c.lowerThan, greaterThanEqual = _c.greaterThanEqual, greaterThan = _c.greaterThan, maxScale = _c.maxScale;
    var minOrMax = min || greaterThanEqual || max || lowerThanEqual;
    var numberValue = applyNumberValidations({
        greaterThan: greaterThan,
        minOrMax: minOrMax,
        lowerThan: lowerThan,
        maxScale: maxScale,
    }, actualDefaultValue);
    return (0, value_generator_utils_1.applyScale)(maxScale, numberValue);
}
exports.someNumber = someNumber;
function someLong(_a) {
    var _b = _a === void 0 ? {
        validations: [],
        isUnsigned: false,
    } : _a, validations = _b.validations, isUnsigned = _b.isUnsigned;
    var number = someNumber({ validations: validations, isUnsigned: isUnsigned });
    // new Long(min || max, 0, true) in case of unsigned
    // new Long(min || max, 0) otherwise
    return new long_1.default(number, 0, isUnsigned);
}
exports.someLong = someLong;
function someFloat(_a) {
    var _b = _a === void 0 ? { validations: [] } : _a, validations = _b.validations;
    return someNumber({ validations: validations, defaultValue: constant_values_1.CONSTANT_VALUES.FLOAT });
}
exports.someFloat = someFloat;
function someRepeated(_a, onlyStringValues) {
    var _b = _a === void 0 ? { validations: [] } : _a, validations = _b.validations;
    var repeatedValue = onlyStringValues
        ? constant_values_1.CONSTANT_VALUES.FIELD_MASK
        : constant_values_1.CONSTANT_VALUES.LIST_VALUE;
    if (!(validations === null || validations === void 0 ? void 0 : validations.length)) {
        return repeatedValue;
    }
    var validationExtractor = new validation_extractor_1.ValidationExtractor(validations);
    var _c = validationExtractor.getRepeatedValidations(), minSize = _c.minSize, maxSize = _c.maxSize;
    var shouldPadArrayWithValue = minSize && repeatedValue.length < minSize;
    if (shouldPadArrayWithValue) {
        return (0, value_generator_utils_1.padArray)(repeatedValue, minSize, constant_values_1.CONSTANT_VALUES.STRING);
    }
    var shouldReduceArraySize = maxSize && repeatedValue.length > maxSize;
    if (shouldReduceArraySize) {
        return repeatedValue.slice(0, maxSize);
    }
    return repeatedValue;
}
exports.someRepeated = someRepeated;
function someDuration(_a) {
    var _b = _a === void 0 ? { validations: [] } : _a, validations = _b.validations;
    return constant_values_1.CONSTANT_VALUES.DURATION;
}
exports.someDuration = someDuration;
function someDate(_a) {
    var _b = _a === void 0 ? { validations: [] } : _a, validations = _b.validations;
    return constant_values_1.CONSTANT_VALUES.DATE;
}
exports.someDate = someDate;
function someBytes(_a) {
    var _b = _a === void 0 ? { validations: [] } : _a, validations = _b.validations;
    return constant_values_1.CONSTANT_VALUES.BYTES;
}
exports.someBytes = someBytes;
function someAny(_a) {
    var _b = _a === void 0 ? { validations: [] } : _a, validations = _b.validations;
    return constant_values_1.CONSTANT_VALUES.EMPTY_OBJECT;
}
exports.someAny = someAny;
function getGeneratorFunctionNameForGeneratorType(generatorType) {
    return (0, value_generator_utils_1.generatorTypeToFunctionName)(generatorType);
}
exports.getGeneratorFunctionNameForGeneratorType = getGeneratorFunctionNameForGeneratorType;
function applyNumberValidations(_a, defaultValue) {
    var greaterThan = _a.greaterThan, minOrMax = _a.minOrMax, lowerThan = _a.lowerThan, maxScale = _a.maxScale;
    if (greaterThan) {
        return (0, value_generator_utils_1.increment)(maxScale, greaterThan);
    }
    else if (minOrMax) {
        return minOrMax;
    }
    else if (lowerThan) {
        return (0, value_generator_utils_1.decrement)(maxScale, lowerThan);
    }
    else {
        return defaultValue;
    }
}
function getTypeForGenerator(field) {
    if (field.mapValueType && field.mapValueType !== 'UNKNOWN_TYPE') {
        return (0, value_generator_utils_1.wrapperToBusinessSchemaType)(field.schemaName) || field.mapValueType;
    }
    return getFieldType(field);
}
function getFieldType(field) {
    return (0, value_generator_utils_1.wrapperToBusinessSchemaType)(field.schemaName) || field.type;
}
