// @wix/auto_sdk_forms_interactive-form-sessions/build/es/index.typings.mjs
// src/forms-ai-v1-interactive-form-session-interactive-form-sessions.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/forms-ai-v1-interactive-form-session-interactive-form-sessions.http.ts
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixFormsAiV1InteractiveFormSessionsServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/_api/form-ai-assistant",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/form-ai-assistant",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/form-ai-assistant",
        destPath: ""
      }
    ],
    "*.wixforms.com": [
      {
        srcPath: "/_api/form-ai-assistant",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/form-ai-assistant",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/form-ai-assistant",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/forms/ai/v1/chat-settings",
        destPath: "/v1/chat-settings"
      },
      {
        srcPath: "/forms/ai/v1/interactive-form-sessions",
        destPath: "/v1/interactive-form-sessions"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/form-ai-assistant",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_forms_interactive-form-sessions";
function createInteractiveFormSession(payload) {
  function __createInteractiveFormSession({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.ai.v1.interactive_form_session",
      method: "POST",
      methodFqn: "wix.forms.ai.v1.InteractiveFormSessionsService.CreateInteractiveFormSession",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsAiV1InteractiveFormSessionsServiceUrl({
        protoPath: "/v1/interactive-form-sessions",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "responseChunks.numberInputDetails.multipleOf" },
            {
              path: "responseChunks.numberInputDetails.rangeLimit.maxInclusiveValue"
            },
            {
              path: "responseChunks.numberInputDetails.rangeLimit.maxExclusiveValue"
            },
            {
              path: "responseChunks.numberInputDetails.rangeLimit.minInclusiveValue"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createInteractiveFormSession;
}
function createInteractiveFormSessionStreamed(payload) {
  function __createInteractiveFormSessionStreamed({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.ai.v1.interactive_form_session",
      method: "POST",
      methodFqn: "wix.forms.ai.v1.InteractiveFormSessionsService.CreateInteractiveFormSessionStreamed",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsAiV1InteractiveFormSessionsServiceUrl({
        protoPath: "/v1/interactive-form-sessions/create-streamed",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "responseChunk.numberInputDetails.multipleOf" },
            {
              path: "responseChunk.numberInputDetails.rangeLimit.maxInclusiveValue"
            },
            {
              path: "responseChunk.numberInputDetails.rangeLimit.maxExclusiveValue"
            },
            {
              path: "responseChunk.numberInputDetails.rangeLimit.minInclusiveValue"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createInteractiveFormSessionStreamed;
}
function sendUserMessage(payload) {
  function __sendUserMessage({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.ai.v1.interactive_form_session",
      method: "POST",
      methodFqn: "wix.forms.ai.v1.InteractiveFormSessionsService.SendUserMessage",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsAiV1InteractiveFormSessionsServiceUrl({
        protoPath: "/v1/interactive-form-sessions/{interactiveFormSessionId}/send-user-message",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "responseChunks.numberInputDetails.multipleOf" },
            {
              path: "responseChunks.numberInputDetails.rangeLimit.maxInclusiveValue"
            },
            {
              path: "responseChunks.numberInputDetails.rangeLimit.maxExclusiveValue"
            },
            {
              path: "responseChunks.numberInputDetails.rangeLimit.minInclusiveValue"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __sendUserMessage;
}
function sendUserMessageStreamed(payload) {
  function __sendUserMessageStreamed({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.ai.v1.interactive_form_session",
      method: "POST",
      methodFqn: "wix.forms.ai.v1.InteractiveFormSessionsService.SendUserMessageStreamed",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsAiV1InteractiveFormSessionsServiceUrl({
        protoPath: "/v1/interactive-form-sessions/{interactiveFormSessionId}/send-user-message-streamed",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "responseChunk.numberInputDetails.multipleOf" },
            {
              path: "responseChunk.numberInputDetails.rangeLimit.maxInclusiveValue"
            },
            {
              path: "responseChunk.numberInputDetails.rangeLimit.maxExclusiveValue"
            },
            {
              path: "responseChunk.numberInputDetails.rangeLimit.minInclusiveValue"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __sendUserMessageStreamed;
}

// src/forms-ai-v1-interactive-form-session-interactive-form-sessions.universal.ts
var ChunkType = /* @__PURE__ */ ((ChunkType2) => {
  ChunkType2["UNKNOWN_CHUNK_TYPE"] = "UNKNOWN_CHUNK_TYPE";
  ChunkType2["TEXT"] = "TEXT";
  ChunkType2["TEXT_DATA"] = "TEXT_DATA";
  ChunkType2["MULTI_SELECT_INPUT"] = "MULTI_SELECT_INPUT";
  ChunkType2["NUMBER_INPUT"] = "NUMBER_INPUT";
  ChunkType2["SEPARATOR"] = "SEPARATOR";
  ChunkType2["SINGLE_SELECT_INPUT"] = "SINGLE_SELECT_INPUT";
  ChunkType2["ERROR"] = "ERROR";
  ChunkType2["SUBMISSION"] = "SUBMISSION";
  ChunkType2["IMPORTANT_TEXT"] = "IMPORTANT_TEXT";
  ChunkType2["DEBUG"] = "DEBUG";
  ChunkType2["END_OF_RESPONSE"] = "END_OF_RESPONSE";
  ChunkType2["FILE_UPLOAD"] = "FILE_UPLOAD";
  ChunkType2["SIGNATURE"] = "SIGNATURE";
  return ChunkType2;
})(ChunkType || {});
var Style = /* @__PURE__ */ ((Style2) => {
  Style2["UNKNOWN_STYLE"] = "UNKNOWN_STYLE";
  Style2["NORMAL"] = "NORMAL";
  Style2["BOLD"] = "BOLD";
  return Style2;
})(Style || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  Type2["PARAGRAPH"] = "PARAGRAPH";
  return Type2;
})(Type || {});
var Role = /* @__PURE__ */ ((Role2) => {
  Role2["UNKNOWN_ROLE"] = "UNKNOWN_ROLE";
  Role2["USER"] = "USER";
  Role2["ASSISTANT"] = "ASSISTANT";
  Role2["FUNCTION_CALL"] = "FUNCTION_CALL";
  Role2["FUNCTION_CALL_OUTPUT"] = "FUNCTION_CALL_OUTPUT";
  Role2["DEVELOPER"] = "DEVELOPER";
  return Role2;
})(Role || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createInteractiveFormSession2(formId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    formId,
    currentValues: options?.currentValues,
    dryRun: options?.dryRun,
    clientTime: options?.clientTime
  });
  const reqOpts = createInteractiveFormSession(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.interactiveFormSession;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          formId: "$[0]",
          currentValues: "$[1].currentValues",
          dryRun: "$[1].dryRun",
          clientTime: "$[1].clientTime"
        },
        singleArgumentUnchanged: false
      },
      ["formId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function createInteractiveFormSessionStreamed2(formId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    formId,
    currentValues: options?.currentValues,
    dryRun: options?.dryRun,
    clientTime: options?.clientTime
  });
  const reqOpts = createInteractiveFormSessionStreamed(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          formId: "$[0]",
          currentValues: "$[1].currentValues",
          dryRun: "$[1].dryRun",
          clientTime: "$[1].clientTime"
        },
        singleArgumentUnchanged: false
      },
      ["formId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function sendUserMessage2(interactiveFormSessionId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    interactiveFormSessionId,
    input: options?.input,
    currentValues: options?.currentValues
  });
  const reqOpts = sendUserMessage(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          interactiveFormSessionId: "$[0]",
          input: "$[1].input",
          currentValues: "$[1].currentValues"
        },
        singleArgumentUnchanged: false
      },
      ["interactiveFormSessionId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function sendUserMessageStreamed2(interactiveFormSessionId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    interactiveFormSessionId,
    input: options?.input,
    currentValues: options?.currentValues
  });
  const reqOpts = sendUserMessageStreamed(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          interactiveFormSessionId: "$[0]",
          input: "$[1].input",
          currentValues: "$[1].currentValues"
        },
        singleArgumentUnchanged: false
      },
      ["interactiveFormSessionId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
export {
  ChunkType,
  Role,
  Style,
  Type,
  WebhookIdentityType,
  createInteractiveFormSession2 as createInteractiveFormSession,
  createInteractiveFormSessionStreamed2 as createInteractiveFormSessionStreamed,
  sendUserMessage2 as sendUserMessage,
  sendUserMessageStreamed2 as sendUserMessageStreamed
};
