// @wix/bookings_app-extensions/build/cjs/site.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// site.ts
var site_exports = {};
__export(site_exports, {
  plugins: () => editor_pages_plugin_props_exports
});
module.exports = __toCommonJS(site_exports);

// src/extensions/editor-pages-plugin-props.ts
var editor_pages_plugin_props_exports = {};
__export(editor_pages_plugin_props_exports, {
  BookingCalendar: () => d912c5_52cb_4657_b8fa_e1a4cda8ed01_exports,
  ServicePage: () => a91a0543_d4bd_4e6b_b315_9410aa27bcde_exports
});

// src/extensions/editor-page-plugins/54d912c5-52cb-4657-b8fa-e1a4cda8ed01.ts
var d912c5_52cb_4657_b8fa_e1a4cda8ed01_exports = {};
__export(d912c5_52cb_4657_b8fa_e1a4cda8ed01_exports, {
  LocationType: () => LocationType,
  ServiceType: () => ServiceType,
  SlotIds: () => SlotIds
});
var LocationType = /* @__PURE__ */ ((LocationType2) => {
  LocationType2["UNKNOWN_LOCATION_TYPE"] = "UNKNOWN_LOCATION_TYPE";
  LocationType2["CUSTOM"] = "CUSTOM";
  LocationType2["BUSINESS"] = "BUSINESS";
  LocationType2["CUSTOMER"] = "CUSTOMER";
  return LocationType2;
})(LocationType || {});
var ServiceType = /* @__PURE__ */ ((ServiceType2) => {
  ServiceType2["UNKNOWN_SERVICE_TYPE"] = "UNKNOWN_SERVICE_TYPE";
  ServiceType2["APPOINTMENT"] = "APPOINTMENT";
  ServiceType2["CLASS"] = "CLASS";
  ServiceType2["COURSE"] = "COURSE";
  return ServiceType2;
})(ServiceType || {});
var SlotIds = {
  calendarAvailabilitySlot: "74c4c90d-39a4-4ee4-bfd3-a025857db32a"
};

// src/extensions/editor-page-plugins/a91a0543-d4bd-4e6b-b315-9410aa27bcde.ts
var a91a0543_d4bd_4e6b_b315_9410aa27bcde_exports = {};
__export(a91a0543_d4bd_4e6b_b315_9410aa27bcde_exports, {
  SlotIds: () => SlotIds2
});
var SlotIds2 = {
  slot1: "1c36d1de-3ede-43cf-adb5-4148276ad0a7"
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  plugins
});
