// pro-gallery-lib/dist/cjs/core/presets/presets.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isInPreset = exports.getLayoutName = exports.NEW_PRESETS = exports.addPresetOptions = void 0;
const tslib_1 = require("tslib");
//#region Imports Layouts
const alternateGallery_js_1 = require("./alternateGallery.js");
const bricksGallery_js_1 = require("./bricksGallery.js");
const collageGallery_js_1 = require("./collageGallery.js");
const fullsizeGallery_js_1 = require("./fullsizeGallery.js");
const columnGallery_js_1 = require("./columnGallery.js");
const gridGallery_js_1 = require("./gridGallery.js");
const masonryGallery_js_1 = require("./masonryGallery.js");
const mixGallery_js_1 = require("./mixGallery.js");
const panoramaGallery_js_1 = require("./panoramaGallery.js");
const sliderGallery_js_1 = require("./sliderGallery.js");
const slideshowGallery_js_1 = require("./slideshowGallery.js");
const thumbnailsGallery_js_1 = require("./thumbnailsGallery.js");
const emptyGallery_js_1 = require("./emptyGallery.js");
const designedPresetGallery_js_1 = require("./designedPresetGallery.js");
const optionsMap_js_1 = tslib_1.__importDefault(require("../helpers/optionsMap.js"));
const index_js_1 = require("../../index.js");
//#endregion Imports
const addPresetOptions = (options) => {
    // new layouts
    const LAYOUTS = index_js_1.GALLERY_CONSTS[optionsMap_js_1.default.layoutParams.structure.galleryLayout];
    switch (options[optionsMap_js_1.default.layoutParams.structure.galleryLayout]) {
        case LAYOUTS.MASONRY:
            return (0, masonryGallery_js_1.createOptions)(options);
        case LAYOUTS.GRID:
            return (0, gridGallery_js_1.createOptions)(options);
        case LAYOUTS.THUMBNAIL:
            return (0, thumbnailsGallery_js_1.createOptions)(options);
        case LAYOUTS.SLIDER:
            return (0, sliderGallery_js_1.createOptions)(options);
        case LAYOUTS.SLIDESHOW:
            return (0, slideshowGallery_js_1.createOptions)(options);
        case LAYOUTS.PANORAMA:
            return (0, panoramaGallery_js_1.createOptions)(options);
        case LAYOUTS.COLUMN:
            return (0, columnGallery_js_1.createOptions)(options);
        case LAYOUTS.MAGIC:
            console.error('A layout generated by the magicLayoutSeed is no longer supported as of Pro Gallery v5');
            return options;
        case LAYOUTS.FULLSIZE:
            return (0, fullsizeGallery_js_1.createOptions)(options);
        case LAYOUTS.BRICKS:
            return (0, bricksGallery_js_1.createOptions)(options);
        case LAYOUTS.MIX:
            return (0, mixGallery_js_1.createOptions)(options);
        case LAYOUTS.ALTERNATE:
            return (0, alternateGallery_js_1.createOptions)(options);
        case LAYOUTS.EMPTY:
            return (0, emptyGallery_js_1.createOptions)(options);
        case LAYOUTS.JSON_FIXED:
            return (0, designedPresetGallery_js_1.createOptions)(options);
        case LAYOUTS.COLLAGE:
        default:
            return (0, collageGallery_js_1.createOptions)(options);
    }
};
exports.addPresetOptions = addPresetOptions;
const NEW_PRESETS = {
    alternate: alternateGallery_js_1.fixedOptions,
    bricks: bricksGallery_js_1.fixedOptions,
    collage: collageGallery_js_1.fixedOptions,
    fullsize: fullsizeGallery_js_1.fixedOptions,
    column: columnGallery_js_1.fixedOptions,
    grid: gridGallery_js_1.fixedOptions,
    masonry: masonryGallery_js_1.fixedOptions,
    mix: mixGallery_js_1.fixedOptions,
    panorama: panoramaGallery_js_1.fixedOptions,
    slider: sliderGallery_js_1.fixedOptions,
    slideshow: slideshowGallery_js_1.fixedOptions,
    thumbnails: thumbnailsGallery_js_1.fixedOptions,
    empty: emptyGallery_js_1.fixedOptions,
};
exports.NEW_PRESETS = NEW_PRESETS;
const getLayoutName = (galleryLayout) => {
    const galleyLayoutList = [
        'empty',
        'collage',
        'masonry',
        'grid',
        'thumbnails',
        'slider',
        'slideshow',
        'panorama',
        'column',
        'magic-deprecated',
        'fullsize',
        'bricks',
        'mix',
        'alternate', // 12
    ];
    return galleyLayoutList[galleryLayout + 1];
};
exports.getLayoutName = getLayoutName;
// returns true if the given param is in the current layout preset
const isInPreset = (galleryLayout, paramToCheck) => {
    const layoutName = getLayoutName(galleryLayout) || 'empty'; // empty for when there is no layout given
    return Object.keys(NEW_PRESETS[layoutName]).includes(paramToCheck);
};
exports.isInPreset = isInPreset;
