// @wix/ricos/dist/cjs/ui-components/color-picker/button-color-swatches/button-color-swatches.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var button_color_swatches_exports = {};
__export(button_color_swatches_exports, {
  ButtonColorSwatches: () => ButtonColorSwatches
});
module.exports = __toCommonJS(button_color_swatches_exports);
var import_design_system = require("@wix/design-system");
var import_wix_ui_icons_common = require("@wix/wix-ui-icons-common");
var import_react = __toESM(require("react"));
var import_data_hooks = require("./../../color-picker/button-color-swatches/data-hooks.js");
var import_single_color_swatch = require("../new-color-picker/components/single-color-swatch/index.js");
var import_button_color_swatches_scss = __toESM(require("./button-color-swatches-scss.js"));
const SWATCH_COLORS = ["#000624", "#25A55A", "#FFB700", "#E62214", "#116DFF"];
const ButtonColorSwatches = ({
  label,
  color,
  onChange,
  dataHook,
  defaultColor,
  onMorePress,
  moreButtonRef
}) => {
  return /* @__PURE__ */ import_react.default.createElement(import_design_system.Box, { gap: "SP3", direction: "vertical", dataHook }, /* @__PURE__ */ import_react.default.createElement(import_design_system.Text, { size: "small" }, label), /* @__PURE__ */ import_react.default.createElement(import_design_system.Box, { display: "flex", width: "100%", alignContent: "stretch", gap: "SP3" }, SWATCH_COLORS.map((c) => /* @__PURE__ */ import_react.default.createElement(
    import_single_color_swatch.SingleColorSwatch,
    {
      key: c,
      fill: c,
      selectedColor: color || defaultColor,
      onClick: onChange,
      className: import_button_color_swatches_scss.default.fillPreview
    }
  )), /* @__PURE__ */ import_react.default.createElement(import_design_system.Divider, { direction: "vertical", className: import_button_color_swatches_scss.default.divider }), /* @__PURE__ */ import_react.default.createElement(
    import_design_system.FillPreview,
    {
      fill: /* @__PURE__ */ import_react.default.createElement(import_wix_ui_icons_common.MoreSmall, null),
      className: import_button_color_swatches_scss.default.fillPreview,
      onClick: onMorePress,
      ref: moreButtonRef,
      dataHook: import_data_hooks.SWATCH_TOGGLE
    }
  )));
};
