// @wix/headless-stores/cjs/dist/react/core/SelectedVariant.js
import { jsx as _jsx } from "react/jsx-runtime";
import { useService, WixServices } from '@wix/services-manager-react';
import { SelectedVariantServiceDefinition, SelectedVariantService, } from '../../services/selected-variant-service.js';
import { ProductModifiersServiceDefinition } from '../../services/product-modifiers-service.js';
import { createServicesMap } from '@wix/services-manager';
import { Commerce } from '@wix/headless-ecom/react';
import { CheckoutServiceDefinition, CurrentCartServiceDefinition, } from '@wix/headless-ecom/services';
/**
 * Root component that provides the SelectedVariant service context to its children.
 * This component sets up the necessary services for rendering and managing selected variant data.
 *
 * @order 1
 * @component
 * @example
 * ```tsx
 * import { SelectedVariant } from '@wix/stores/components';
 *
 * function ProductVariantDisplay() {
 *   return (
 *     <SelectedVariant.Root selectedVariantServiceConfig={{ fetchInventoryData: true }}>
 *       <div>
 *         <SelectedVariant.Price>
 *           {({ price, compareAtPrice, currency }) => (
 *             <div className="price-display">
 *               <span className="current-price">{price}</span>
 *               {compareAtPrice && (
 *                 <span className="compare-price">
 *                   <s>{compareAtPrice}</s>
 *                 </span>
 *               )}
 *               <span className="currency">{currency}</span>
 *             </div>
 *           )}
 *         </SelectedVariant.Price>
 *       </div>
 *     </SelectedVariant.Root>
 *   );
 * }
 * ```
 */
export function Root(props) {
    return (_jsx(WixServices, { servicesMap: createServicesMap().addService(SelectedVariantServiceDefinition, SelectedVariantService, props.selectedVariantServiceConfig), children: props.children }));
}
/**
 * Headless component for selected variant details display
 *
 * @component
 * @example
 * ```tsx
 * import { SelectedVariant } from '@wix/stores/components';
 *
 * function VariantDetails() {
 *   return (
 *     <SelectedVariant.Details>
 *       {({ sku, weight }) => (
 *         <div>
 *           {sku && <div>SKU: {sku}</div>}
 *           {weight && <div>Weight: {weight}</div>}
 *         </div>
 *       )}
 *     </SelectedVariant.Details>
 *   );
 * }
 * ```
 */
export function Details(props) {
    const selectedVariantService = useService(SelectedVariantServiceDefinition);
    const selectedVariant = selectedVariantService.currentVariant?.get();
    let sku = selectedVariant?.sku || null;
    let weight = selectedVariant?.physicalProperties?.weight?.toString() || null;
    return props.children({
        sku,
        weight,
    });
}
/**
 * Headless component for product price display
 *
 * @component
 * @example
 * ```tsx
 * import { SelectedVariant } from '@wix/stores/components';
 *
 * function ProductPrice() {
 *   return (
 *     <SelectedVariant.Price>
 *       {({ price, compareAtPrice, currency }) => (
 *         <div className="price-display">
 *           <span className="current-price">{price}</span>
 *           {compareAtPrice && (
 *             <span className="compare-price">
 *               <s>{compareAtPrice}</s>
 *             </span>
 *           )}
 *           <span className="currency">{currency}</span>
 *         </div>
 *       )}
 *     </SelectedVariant.Price>
 *   );
 * }
 * ```
 */
export function Price(props) {
    const variantService = useService(SelectedVariantServiceDefinition);
    const price = variantService.currentPrice.get();
    const compareAtPrice = variantService.currentCompareAtPrice.get();
    const currency = variantService.currency.get();
    return props.children({
        price,
        compareAtPrice,
        currency,
    });
}
/**
 * Headless component for product SKU display
 *
 * @component
 * @example
 * ```tsx
 * import { SelectedVariant } from '@wix/stores/components';
 *
 * function ProductSKU() {
 *   return (
 *     <SelectedVariant.SKU>
 *       {({ sku }) => (
 *         <div>
 *           {sku && (
 *             <div className="product-sku">
 *               <strong>SKU:</strong> {sku}
 *             </div>
 *           )}
 *         </div>
 *       )}
 *     </SelectedVariant.SKU>
 *   );
 * }
 * ```
 */
export function SKU(props) {
    const selectedVariantService = useService(SelectedVariantServiceDefinition);
    const selectedVariant = selectedVariantService.currentVariant?.get();
    const sku = selectedVariant?.sku || null;
    return props.children({
        sku,
    });
}
/**
 * Headless component for product actions (add to cart, buy now)
 *
 * @component
 * @example
 * ```tsx
 * import { SelectedVariant } from '@wix/stores/components';
 *
 * function AddToCartButton() {
 *   return (
 *     <SelectedVariant.Actions>
 *       {({ addToCart, buyNow, canAddToCart, isLoading, inStock, error }) => (
 *         <div>
 *           {error && <div className="error">{error}</div>}
 *           {!inStock && <div>Out of stock</div>}
 *           <button
 *             onClick={addToCart}
 *             disabled={!canAddToCart || isLoading}
 *           >
 *             {isLoading ? 'Adding...' : 'Add to Cart'}
 *           </button>
 *           <button
 *             onClick={buyNow}
 *             disabled={!canAddToCart || isLoading}
 *           >
 *             Buy Now
 *           </button>
 *         </div>
 *       )}
 *     </SelectedVariant.Actions>
 *   );
 * }
 * ```
 */
export function Actions(props) {
    const variantService = useService(SelectedVariantServiceDefinition);
    const cartService = useService(CurrentCartServiceDefinition);
    // Try to get checkout service - it may not be available
    let checkoutService = null;
    try {
        checkoutService = useService(CheckoutServiceDefinition);
    }
    catch {
        // Checkout service not available
        checkoutService = null;
    }
    // Try to get modifiers service - it may not exist for all products
    let modifiersService = null;
    try {
        modifiersService = useService(ProductModifiersServiceDefinition);
    }
    catch {
        // Modifiers service not available for this product
        modifiersService = null;
    }
    const inStock = variantService.isInStock.get();
    const isPreOrderEnabled = variantService.isPreOrderEnabled.get() && !inStock;
    const preOrderMessage = variantService.preOrderMessage.get();
    const isLoading = variantService.isLoading.get() ||
        (checkoutService ? checkoutService.isLoading.get() : false);
    const error = variantService.error.get() ||
        (checkoutService ? checkoutService.error.get() : null);
    const quantity = variantService.selectedQuantity.get();
    // Check if all required modifiers are filled
    const areAllRequiredModifiersFilled = modifiersService
        ? modifiersService.areAllRequiredModifiersFilled()
        : true; // If no modifiers service, assume no required modifiers
    const canAddToCart = (inStock || isPreOrderEnabled) &&
        !isLoading &&
        areAllRequiredModifiersFilled;
    const getModifiersData = () => {
        if (modifiersService) {
            const selectedModifiers = modifiersService.selectedModifiers.get();
            if (Object.keys(selectedModifiers).length > 0) {
                return selectedModifiers;
            }
        }
    };
    const addToCart = async () => {
        const modifiersData = getModifiersData();
        await variantService.addToCart(quantity, modifiersData);
    };
    const buyNowFallback = async () => {
        try {
            // Clear the cart first
            await cartService.clearCart();
            // Add the product to cart
            await addToCart();
            // Proceed to checkout
            await cartService.proceedToCheckout();
        }
        catch (error) {
            console.error('Buy now failed:', error);
            throw error;
        }
    };
    const getLineItems = () => {
        const modifiersData = getModifiersData();
        return variantService.createLineItems(quantity, modifiersData);
    };
    const commonProps = {
        lineItems: getLineItems(),
        addToCart,
        canAddToCart,
        isLoading,
        inStock,
        isPreOrderEnabled,
        preOrderMessage,
        error,
    };
    if (checkoutService) {
        return (_jsx(Commerce.CheckoutTrigger, { children: ({ createCheckout, isLoading: checkoutLoading, error: checkoutError, }) => props.children({
                ...commonProps,
                isLoading: isLoading || checkoutLoading,
                error: error || checkoutError,
                buyNow: () => createCheckout(getLineItems()),
            }) }));
    }
    return props.children({
        ...commonProps,
        buyNow: buyNowFallback,
    });
}
