// @wix/ricos/dist/es/plugin-table/tiptap/plugins/addRowColPlugin.js
import { DecorationSet } from "@tiptap/pm/view";
import { findTable } from "prosemirror-utils";
import { addColDecoration, addRowDecoration } from "../decorations/index.js";
const addRowColPlugin = (Plugin, PluginKey, editor, services) => {
  const key = new PluginKey("table-add-row-col");
  return new Plugin({
    key,
    state: {
      init: () => {
        return {};
      },
      apply(tr, prev, oldState, newState) {
        const newParentTable = findTable(newState.selection);
        const oldParentTable = findTable(oldState.selection);
        if (!newParentTable) {
          return {};
        }
        if (oldParentTable?.start === newParentTable.start) {
          return prev;
        }
        const decorations = DecorationSet.create(newState.doc, [
          addRowDecoration(newState, editor, services),
          addColDecoration(newState, editor, services)
        ]);
        return { decorations };
      }
    },
    props: {
      decorations(state) {
        return key.getState(state).decorations;
      }
    }
  });
};
export {
  addRowColPlugin
};
