// @wix/auto_sdk_benefit-programs_pool-definitions/build/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  Cascade: () => Cascade,
  PolicyExpressionType: () => PolicyExpressionType,
  RateLimitedPolicyType: () => RateLimitedPolicyType,
  SortOrder: () => SortOrder,
  Type: () => Type,
  WebhookIdentityType: () => WebhookIdentityType,
  WeekDay: () => WeekDay,
  addPoolDefinitionToProgramDefinition: () => addPoolDefinitionToProgramDefinition2,
  bulkCreatePoolDefinitions: () => bulkCreatePoolDefinitions2,
  bulkDeletePoolDefinitions: () => bulkDeletePoolDefinitions2,
  bulkUpdatePoolDefinitions: () => bulkUpdatePoolDefinitions2,
  createPoolDefinition: () => createPoolDefinition2,
  deletePoolDefinition: () => deletePoolDefinition2,
  findPoolDefinitionsByProgramDefinition: () => findPoolDefinitionsByProgramDefinition2,
  getPoolDefinition: () => getPoolDefinition2,
  queryPoolDefinitions: () => queryPoolDefinitions2,
  removePoolDefinitionFromProgramDefinition: () => removePoolDefinitionFromProgramDefinition2,
  updatePoolDefinition: () => updatePoolDefinition2
});
module.exports = __toCommonJS(index_typings_exports);

// src/benefit-programs-v1-pool-definition-pool-definitions.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/benefit-programs-v1-pool-definition-pool-definitions.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixBenefitProgramsV1PoolDefinitionPoolDefinitionServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/pool-definitions",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/benefit-programs-pool-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/pool-definitions",
        destPath: "/v1/pool-definitions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/pool-definitions",
        destPath: "/v1/bulk/pool-definitions"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/benefit-programs-pool-definitions",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/benefit-programs-pool-definitions",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pool-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/pool-definitions",
        destPath: "/v1/pool-definitions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/pool-definitions",
        destPath: "/v1/bulk/pool-definitions"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pool-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/pool-definitions",
        destPath: "/v1/pool-definitions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/pool-definitions",
        destPath: "/v1/bulk/pool-definitions"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/benefit-programs-pool-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/pool-definitions",
        destPath: "/v1/pool-definitions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/pool-definitions",
        destPath: "/v1/bulk/pool-definitions"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/benefit-programs-pool-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/pool-definitions",
        destPath: "/v1/pool-definitions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/pool-definitions",
        destPath: "/v1/bulk/pool-definitions"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pool-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/pool-definitions",
        destPath: "/v1/pool-definitions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/pool-definitions",
        destPath: "/v1/bulk/pool-definitions"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pool-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/pool-definitions",
        destPath: "/v1/pool-definitions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/pool-definitions",
        destPath: "/v1/bulk/pool-definitions"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/_api/benefit-programs/v1/pool-definitions",
        destPath: "/v1/pool-definitions"
      },
      {
        srcPath: "/benefit-programs/v1/pool-definitions",
        destPath: "/v1/pool-definitions"
      },
      {
        srcPath: "/benefit-programs/v1/bulk/pool-definitions",
        destPath: "/v1/bulk/pool-definitions"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_benefit-programs_pool-definitions";
function createPoolDefinition(payload) {
  function __createPoolDefinition({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "poolDefinition.createdDate" },
          { path: "poolDefinition.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool_definition",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.pool_definition.PoolDefinitionService.CreatePoolDefinition",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolDefinitionPoolDefinitionServiceUrl({
        protoPath: "/v1/pool-definitions",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "poolDefinition.createdDate" },
            { path: "poolDefinition.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createPoolDefinition;
}
function bulkCreatePoolDefinitions(payload) {
  function __bulkCreatePoolDefinitions({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "poolDefinitions.createdDate" },
          { path: "poolDefinitions.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool_definition",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.pool_definition.PoolDefinitionService.BulkCreatePoolDefinitions",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolDefinitionPoolDefinitionServiceUrl({
        protoPath: "/v1/bulk/pool-definitions/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.poolDefinition.createdDate" },
            { path: "results.poolDefinition.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreatePoolDefinitions;
}
function updatePoolDefinition(payload) {
  function __updatePoolDefinition({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "poolDefinition.createdDate" },
          { path: "poolDefinition.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool_definition",
      method: "PATCH",
      methodFqn: "wix.benefit_programs.v1.pool_definition.PoolDefinitionService.UpdatePoolDefinition",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolDefinitionPoolDefinitionServiceUrl({
        protoPath: "/v1/pool-definitions/{poolDefinition.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "poolDefinition.createdDate" },
            { path: "poolDefinition.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updatePoolDefinition;
}
function bulkUpdatePoolDefinitions(payload) {
  function __bulkUpdatePoolDefinitions({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "poolDefinitions.fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "poolDefinitions.poolDefinition.createdDate" },
          { path: "poolDefinitions.poolDefinition.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool_definition",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.pool_definition.PoolDefinitionService.BulkUpdatePoolDefinitions",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolDefinitionPoolDefinitionServiceUrl({
        protoPath: "/v1/bulk/pool-definitions/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.poolDefinition.createdDate" },
            { path: "results.poolDefinition.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdatePoolDefinitions;
}
function deletePoolDefinition(payload) {
  function __deletePoolDefinition({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool_definition",
      method: "DELETE",
      methodFqn: "wix.benefit_programs.v1.pool_definition.PoolDefinitionService.DeletePoolDefinition",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolDefinitionPoolDefinitionServiceUrl({
        protoPath: "/v1/pool-definitions/{poolDefinitionId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deletePoolDefinition;
}
function bulkDeletePoolDefinitions(payload) {
  function __bulkDeletePoolDefinitions({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool_definition",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.pool_definition.PoolDefinitionService.BulkDeletePoolDefinitions",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolDefinitionPoolDefinitionServiceUrl({
        protoPath: "/v1/bulk/pool-definitions/delete",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.poolDefinition.createdDate" },
            { path: "results.poolDefinition.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkDeletePoolDefinitions;
}
function getPoolDefinition(payload) {
  function __getPoolDefinition({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool_definition",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.pool_definition.PoolDefinitionService.GetPoolDefinition",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolDefinitionPoolDefinitionServiceUrl({
        protoPath: "/v1/pool-definitions/{poolDefinitionId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "poolDefinition.createdDate" },
            { path: "poolDefinition.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getPoolDefinition;
}
function queryPoolDefinitions(payload) {
  function __queryPoolDefinitions({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool_definition",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.pool_definition.PoolDefinitionService.QueryPoolDefinitions",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolDefinitionPoolDefinitionServiceUrl({
        protoPath: "/v1/pool-definitions/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "poolDefinitions.createdDate" },
            { path: "poolDefinitions.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryPoolDefinitions;
}
function addPoolDefinitionToProgramDefinition(payload) {
  function __addPoolDefinitionToProgramDefinition({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool_definition",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.pool_definition.PoolDefinitionService.AddPoolDefinitionToProgramDefinition",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolDefinitionPoolDefinitionServiceUrl({
        protoPath: "/v1/pool-definitions/program-definition/assign",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "poolDefinition.createdDate" },
            { path: "poolDefinition.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __addPoolDefinitionToProgramDefinition;
}
function removePoolDefinitionFromProgramDefinition(payload) {
  function __removePoolDefinitionFromProgramDefinition({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool_definition",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.pool_definition.PoolDefinitionService.RemovePoolDefinitionFromProgramDefinition",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolDefinitionPoolDefinitionServiceUrl({
        protoPath: "/v1/pool-definitions/program-definition/remove",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "poolDefinition.createdDate" },
            { path: "poolDefinition.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __removePoolDefinitionFromProgramDefinition;
}
function findPoolDefinitionsByProgramDefinition(payload) {
  function __findPoolDefinitionsByProgramDefinition({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool_definition",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.pool_definition.PoolDefinitionService.FindPoolDefinitionsByProgramDefinition",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolDefinitionPoolDefinitionServiceUrl({
        protoPath: "/v1/pool-definitions/find-pool-definitions-by-program-definition",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "poolDefinitions.createdDate" },
            { path: "poolDefinitions.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __findPoolDefinitionsByProgramDefinition;
}

// src/benefit-programs-v1-pool-definition-pool-definitions.universal.ts
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var PolicyExpressionType = /* @__PURE__ */ ((PolicyExpressionType2) => {
  PolicyExpressionType2["UNKNOWN"] = "UNKNOWN";
  PolicyExpressionType2["OPERATOR_NOT"] = "OPERATOR_NOT";
  PolicyExpressionType2["OPERATOR_AND"] = "OPERATOR_AND";
  PolicyExpressionType2["OPERATOR_OR"] = "OPERATOR_OR";
  PolicyExpressionType2["POLICY"] = "POLICY";
  return PolicyExpressionType2;
})(PolicyExpressionType || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNKNOWN"] = "UNKNOWN";
  Type2["FIXED_INTERVAL"] = "FIXED_INTERVAL";
  Type2["RATE_LIMITED"] = "RATE_LIMITED";
  Type2["CUSTOM"] = "CUSTOM";
  return Type2;
})(Type || {});
var WeekDay = /* @__PURE__ */ ((WeekDay2) => {
  WeekDay2["UNKNOWN"] = "UNKNOWN";
  WeekDay2["MONDAY"] = "MONDAY";
  WeekDay2["TUESDAY"] = "TUESDAY";
  WeekDay2["WEDNESDAY"] = "WEDNESDAY";
  WeekDay2["THURSDAY"] = "THURSDAY";
  WeekDay2["FRIDAY"] = "FRIDAY";
  WeekDay2["SATURDAY"] = "SATURDAY";
  WeekDay2["SUNDAY"] = "SUNDAY";
  return WeekDay2;
})(WeekDay || {});
var RateLimitedPolicyType = /* @__PURE__ */ ((RateLimitedPolicyType2) => {
  RateLimitedPolicyType2["UNKNOWN"] = "UNKNOWN";
  RateLimitedPolicyType2["FIXED_INTERVAL"] = "FIXED_INTERVAL";
  RateLimitedPolicyType2["PER_CYCLE"] = "PER_CYCLE";
  return RateLimitedPolicyType2;
})(RateLimitedPolicyType || {});
var Cascade = /* @__PURE__ */ ((Cascade2) => {
  Cascade2["UNKNOWN_CASCADE"] = "UNKNOWN_CASCADE";
  Cascade2["NEXT_RENEWAL"] = "NEXT_RENEWAL";
  Cascade2["IMMEDIATELY"] = "IMMEDIATELY";
  Cascade2["FUTURE_PROVISIONS"] = "FUTURE_PROVISIONS";
  return Cascade2;
})(Cascade || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createPoolDefinition2(poolDefinition, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    poolDefinition,
    cascade: options?.cascade
  });
  const reqOpts = createPoolDefinition(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.poolDefinition;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          poolDefinition: "$[0]",
          cascade: "$[1].cascade"
        },
        singleArgumentUnchanged: false
      },
      ["poolDefinition", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCreatePoolDefinitions2(poolDefinitions, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    poolDefinitions,
    cascade: options?.cascade,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkCreatePoolDefinitions(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          poolDefinitions: "$[0]",
          cascade: "$[1].cascade",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["poolDefinitions", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updatePoolDefinition2(_id, poolDefinition, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    poolDefinition: { ...poolDefinition, id: _id },
    cascade: options?.cascade
  });
  const reqOpts = updatePoolDefinition(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.poolDefinition;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { poolDefinition: "$[1]" },
        explicitPathsToArguments: {
          "poolDefinition.id": "$[0]",
          cascade: "$[2].cascade"
        },
        singleArgumentUnchanged: false
      },
      ["_id", "poolDefinition", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdatePoolDefinitions2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    poolDefinitions: options?.poolDefinitions,
    cascade: options?.cascade,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkUpdatePoolDefinitions(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          poolDefinitions: "$[0].poolDefinitions",
          cascade: "$[0].cascade",
          returnEntity: "$[0].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deletePoolDefinition2(poolDefinitionId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    poolDefinitionId,
    cascade: options?.cascade
  });
  const reqOpts = deletePoolDefinition(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          poolDefinitionId: "$[0]",
          cascade: "$[1].cascade"
        },
        singleArgumentUnchanged: false
      },
      ["poolDefinitionId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkDeletePoolDefinitions2(poolDefinitionIds, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    poolDefinitionIds,
    cascade: options?.cascade
  });
  const reqOpts = bulkDeletePoolDefinitions(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          poolDefinitionIds: "$[0]",
          cascade: "$[1].cascade"
        },
        singleArgumentUnchanged: false
      },
      ["poolDefinitionIds", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getPoolDefinition2(poolDefinitionId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    poolDefinitionId
  });
  const reqOpts = getPoolDefinition(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.poolDefinition;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { poolDefinitionId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["poolDefinitionId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryPoolDefinitions2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryPoolDefinitions(
        payload
      );
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [])
      );
      return {
        items: transformedData?.poolDefinitions,
        pagingMetadata: transformedData?.metadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function addPoolDefinitionToProgramDefinition2(poolDefinitionId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    poolDefinitionId,
    programDefinitionId: options?.programDefinitionId,
    cascade: options?.cascade
  });
  const reqOpts = addPoolDefinitionToProgramDefinition(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          poolDefinitionId: "$[0]",
          programDefinitionId: "$[1].programDefinitionId",
          cascade: "$[1].cascade"
        },
        singleArgumentUnchanged: false
      },
      ["poolDefinitionId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function removePoolDefinitionFromProgramDefinition2(poolDefinitionId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    poolDefinitionId,
    programDefinitionId: options?.programDefinitionId,
    cascade: options?.cascade
  });
  const reqOpts = removePoolDefinitionFromProgramDefinition(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          poolDefinitionId: "$[0]",
          programDefinitionId: "$[1].programDefinitionId",
          cascade: "$[1].cascade"
        },
        singleArgumentUnchanged: false
      },
      ["poolDefinitionId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function findPoolDefinitionsByProgramDefinition2(programDefinitionId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    programDefinitionId,
    namespace: options?.namespace
  });
  const reqOpts = findPoolDefinitionsByProgramDefinition(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          programDefinitionId: "$[0]",
          namespace: "$[1].namespace"
        },
        singleArgumentUnchanged: false
      },
      ["programDefinitionId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Cascade,
  PolicyExpressionType,
  RateLimitedPolicyType,
  SortOrder,
  Type,
  WebhookIdentityType,
  WeekDay,
  addPoolDefinitionToProgramDefinition,
  bulkCreatePoolDefinitions,
  bulkDeletePoolDefinitions,
  bulkUpdatePoolDefinitions,
  createPoolDefinition,
  deletePoolDefinition,
  findPoolDefinitionsByProgramDefinition,
  getPoolDefinition,
  queryPoolDefinitions,
  removePoolDefinitionFromProgramDefinition,
  updatePoolDefinition
});
