// @wix/auto_sdk_benefit-programs_pool-definitions/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  addPoolDefinitionToProgramDefinition: () => addPoolDefinitionToProgramDefinition2,
  bulkCreatePoolDefinitions: () => bulkCreatePoolDefinitions2,
  bulkDeletePoolDefinitions: () => bulkDeletePoolDefinitions2,
  bulkUpdatePoolDefinitions: () => bulkUpdatePoolDefinitions2,
  createPoolDefinition: () => createPoolDefinition2,
  deletePoolDefinition: () => deletePoolDefinition2,
  findPoolDefinitionsByProgramDefinition: () => findPoolDefinitionsByProgramDefinition2,
  getPoolDefinition: () => getPoolDefinition2,
  queryPoolDefinitions: () => queryPoolDefinitions2,
  removePoolDefinitionFromProgramDefinition: () => removePoolDefinitionFromProgramDefinition2,
  updatePoolDefinition: () => updatePoolDefinition2
});
module.exports = __toCommonJS(meta_exports);

// src/benefit-programs-v1-pool-definition-pool-definitions.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixBenefitProgramsV1PoolDefinitionPoolDefinitionServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/pool-definitions",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/benefit-programs-pool-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/pool-definitions",
        destPath: "/v1/pool-definitions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/pool-definitions",
        destPath: "/v1/bulk/pool-definitions"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/benefit-programs-pool-definitions",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/benefit-programs-pool-definitions",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pool-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/pool-definitions",
        destPath: "/v1/pool-definitions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/pool-definitions",
        destPath: "/v1/bulk/pool-definitions"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pool-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/pool-definitions",
        destPath: "/v1/pool-definitions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/pool-definitions",
        destPath: "/v1/bulk/pool-definitions"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/benefit-programs-pool-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/pool-definitions",
        destPath: "/v1/pool-definitions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/pool-definitions",
        destPath: "/v1/bulk/pool-definitions"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/benefit-programs-pool-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/pool-definitions",
        destPath: "/v1/pool-definitions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/pool-definitions",
        destPath: "/v1/bulk/pool-definitions"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pool-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/pool-definitions",
        destPath: "/v1/pool-definitions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/pool-definitions",
        destPath: "/v1/bulk/pool-definitions"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pool-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/pool-definitions",
        destPath: "/v1/pool-definitions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/pool-definitions",
        destPath: "/v1/bulk/pool-definitions"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/_api/benefit-programs/v1/pool-definitions",
        destPath: "/v1/pool-definitions"
      },
      {
        srcPath: "/benefit-programs/v1/pool-definitions",
        destPath: "/v1/pool-definitions"
      },
      {
        srcPath: "/benefit-programs/v1/bulk/pool-definitions",
        destPath: "/v1/bulk/pool-definitions"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_benefit-programs_pool-definitions";
function createPoolDefinition(payload) {
  function __createPoolDefinition({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "poolDefinition.createdDate" },
          { path: "poolDefinition.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool_definition",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.pool_definition.PoolDefinitionService.CreatePoolDefinition",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolDefinitionPoolDefinitionServiceUrl({
        protoPath: "/v1/pool-definitions",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "poolDefinition.createdDate" },
            { path: "poolDefinition.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createPoolDefinition;
}
function bulkCreatePoolDefinitions(payload) {
  function __bulkCreatePoolDefinitions({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "poolDefinitions.createdDate" },
          { path: "poolDefinitions.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool_definition",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.pool_definition.PoolDefinitionService.BulkCreatePoolDefinitions",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolDefinitionPoolDefinitionServiceUrl({
        protoPath: "/v1/bulk/pool-definitions/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.poolDefinition.createdDate" },
            { path: "results.poolDefinition.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreatePoolDefinitions;
}
function updatePoolDefinition(payload) {
  function __updatePoolDefinition({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "poolDefinition.createdDate" },
          { path: "poolDefinition.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool_definition",
      method: "PATCH",
      methodFqn: "wix.benefit_programs.v1.pool_definition.PoolDefinitionService.UpdatePoolDefinition",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolDefinitionPoolDefinitionServiceUrl({
        protoPath: "/v1/pool-definitions/{poolDefinition.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "poolDefinition.createdDate" },
            { path: "poolDefinition.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updatePoolDefinition;
}
function bulkUpdatePoolDefinitions(payload) {
  function __bulkUpdatePoolDefinitions({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "poolDefinitions.fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "poolDefinitions.poolDefinition.createdDate" },
          { path: "poolDefinitions.poolDefinition.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool_definition",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.pool_definition.PoolDefinitionService.BulkUpdatePoolDefinitions",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolDefinitionPoolDefinitionServiceUrl({
        protoPath: "/v1/bulk/pool-definitions/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.poolDefinition.createdDate" },
            { path: "results.poolDefinition.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdatePoolDefinitions;
}
function deletePoolDefinition(payload) {
  function __deletePoolDefinition({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool_definition",
      method: "DELETE",
      methodFqn: "wix.benefit_programs.v1.pool_definition.PoolDefinitionService.DeletePoolDefinition",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolDefinitionPoolDefinitionServiceUrl({
        protoPath: "/v1/pool-definitions/{poolDefinitionId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deletePoolDefinition;
}
function bulkDeletePoolDefinitions(payload) {
  function __bulkDeletePoolDefinitions({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool_definition",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.pool_definition.PoolDefinitionService.BulkDeletePoolDefinitions",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolDefinitionPoolDefinitionServiceUrl({
        protoPath: "/v1/bulk/pool-definitions/delete",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.poolDefinition.createdDate" },
            { path: "results.poolDefinition.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkDeletePoolDefinitions;
}
function getPoolDefinition(payload) {
  function __getPoolDefinition({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool_definition",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.pool_definition.PoolDefinitionService.GetPoolDefinition",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolDefinitionPoolDefinitionServiceUrl({
        protoPath: "/v1/pool-definitions/{poolDefinitionId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "poolDefinition.createdDate" },
            { path: "poolDefinition.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getPoolDefinition;
}
function queryPoolDefinitions(payload) {
  function __queryPoolDefinitions({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool_definition",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.pool_definition.PoolDefinitionService.QueryPoolDefinitions",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolDefinitionPoolDefinitionServiceUrl({
        protoPath: "/v1/pool-definitions/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "poolDefinitions.createdDate" },
            { path: "poolDefinitions.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryPoolDefinitions;
}
function addPoolDefinitionToProgramDefinition(payload) {
  function __addPoolDefinitionToProgramDefinition({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool_definition",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.pool_definition.PoolDefinitionService.AddPoolDefinitionToProgramDefinition",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolDefinitionPoolDefinitionServiceUrl({
        protoPath: "/v1/pool-definitions/program-definition/assign",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "poolDefinition.createdDate" },
            { path: "poolDefinition.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __addPoolDefinitionToProgramDefinition;
}
function removePoolDefinitionFromProgramDefinition(payload) {
  function __removePoolDefinitionFromProgramDefinition({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool_definition",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.pool_definition.PoolDefinitionService.RemovePoolDefinitionFromProgramDefinition",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolDefinitionPoolDefinitionServiceUrl({
        protoPath: "/v1/pool-definitions/program-definition/remove",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "poolDefinition.createdDate" },
            { path: "poolDefinition.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __removePoolDefinitionFromProgramDefinition;
}
function findPoolDefinitionsByProgramDefinition(payload) {
  function __findPoolDefinitionsByProgramDefinition({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool_definition",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.pool_definition.PoolDefinitionService.FindPoolDefinitionsByProgramDefinition",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolDefinitionPoolDefinitionServiceUrl({
        protoPath: "/v1/pool-definitions/find-pool-definitions-by-program-definition",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "poolDefinitions.createdDate" },
            { path: "poolDefinitions.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __findPoolDefinitionsByProgramDefinition;
}

// src/benefit-programs-v1-pool-definition-pool-definitions.meta.ts
function createPoolDefinition2() {
  const payload = {};
  const getRequestOptions = createPoolDefinition(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/pool-definitions",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkCreatePoolDefinitions2() {
  const payload = {};
  const getRequestOptions = bulkCreatePoolDefinitions(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/pool-definitions/create",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updatePoolDefinition2() {
  const payload = { poolDefinition: { id: ":poolDefinitionId" } };
  const getRequestOptions = updatePoolDefinition(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/pool-definitions/{poolDefinition.id}",
    pathParams: { poolDefinitionId: "poolDefinitionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdatePoolDefinitions2() {
  const payload = {};
  const getRequestOptions = bulkUpdatePoolDefinitions(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/pool-definitions/update",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deletePoolDefinition2() {
  const payload = { poolDefinitionId: ":poolDefinitionId" };
  const getRequestOptions = deletePoolDefinition(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/pool-definitions/{poolDefinitionId}",
    pathParams: { poolDefinitionId: "poolDefinitionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkDeletePoolDefinitions2() {
  const payload = {};
  const getRequestOptions = bulkDeletePoolDefinitions(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/pool-definitions/delete",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getPoolDefinition2() {
  const payload = { poolDefinitionId: ":poolDefinitionId" };
  const getRequestOptions = getPoolDefinition(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/pool-definitions/{poolDefinitionId}",
    pathParams: { poolDefinitionId: "poolDefinitionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryPoolDefinitions2() {
  const payload = {};
  const getRequestOptions = queryPoolDefinitions(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/pool-definitions/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function addPoolDefinitionToProgramDefinition2() {
  const payload = {};
  const getRequestOptions = addPoolDefinitionToProgramDefinition(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/pool-definitions/program-definition/assign",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function removePoolDefinitionFromProgramDefinition2() {
  const payload = {};
  const getRequestOptions = removePoolDefinitionFromProgramDefinition(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/pool-definitions/program-definition/remove",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function findPoolDefinitionsByProgramDefinition2() {
  const payload = {};
  const getRequestOptions = findPoolDefinitionsByProgramDefinition(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/pool-definitions/find-pool-definitions-by-program-definition",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  addPoolDefinitionToProgramDefinition,
  bulkCreatePoolDefinitions,
  bulkDeletePoolDefinitions,
  bulkUpdatePoolDefinitions,
  createPoolDefinition,
  deletePoolDefinition,
  findPoolDefinitionsByProgramDefinition,
  getPoolDefinition,
  queryPoolDefinitions,
  removePoolDefinitionFromProgramDefinition,
  updatePoolDefinition
});
