// @wix/ricos/dist/es/ricos-content/builders/button.builder.js
import {
  ButtonData_Type,
  Node_Type,
  PluginContainerData_Alignment,
  PluginContainerData_Width_Type
} from "./../../ricos-schema/legacy/wix/rich_content/v1/index.js";
import { isDefined } from "./utils.js";
const button = (...args) => {
  const buttonData = args.find(
    (a) => a?.type === ButtonData_Type.LINK || a?.type === ButtonData_Type.ACTION
  );
  return {
    type: "block_node",
    // @ts-ignore
    node: (getId) => {
      const nodes = args.map((arg) => "type" in arg && arg.type === "block_node" ? arg.node(getId) : void 0).filter(isDefined);
      return {
        id: getId(),
        type: Node_Type.BUTTON,
        ...nodes?.length && { nodes },
        buttonData: buttonData ? buttonData : {
          type: ButtonData_Type.LINK,
          containerData: {
            alignment: PluginContainerData_Alignment.CENTER,
            width: { size: PluginContainerData_Width_Type.ORIGINAL }
          },
          styles: {},
          text: "Click Me"
        }
      };
    }
  };
};
export {
  button
};
