// @wix/headless-media/dist/react/core/FileUpload.js
import { jsx as _jsx } from "react/jsx-runtime";
import { useService, WixServices } from '@wix/services-manager-react';
import { FileUploadServiceDefinition, FileUploadService, } from '../../services/index.js';
import { createServicesMap } from '@wix/services-manager';
/**
 * Root component that provides the FileUpload service context to its children.
 * This component sets up the necessary services for managing file upload functionality.
 *
 * @order 1
 * @component
 * @example
 * ```tsx
 * import { FileUpload } from '@wix/media/components';
 *
 * function FileUploadForm() {
 *   const uploadAction = async (file: File) => {
 *     // Your upload implementation
 *     return { url: 'uploaded-file-url' };
 *   };
 *
 *   return (
 *     <FileUpload.Root fileUploadServiceConfig={{
 *       maxFileSize: 10 * 1024 * 1024, // 10MB
 *       allowedTypes: ['image/*'],
 *       allowedExtensions: ['.jpg', '.png', '.gif'],
 *       uploadAction,
 *       onUploadSuccess: (result) => console.log('Upload success:', result),
 *       onUploadError: (error) => console.error('Upload error:', error)
 *     }}>
 *       <div>
 *         <FileUpload.FileSelector>
 *           {({ selectedFile, selectFile, clearFile, handleFileSelect }) => (
 *             <div>
 *               <input
 *                 type="file"
 *                 onChange={handleFileSelect}
 *                 accept="image/*"
 *               />
 *               {selectedFile && (
 *                 <p>Selected: {selectedFile.name}</p>
 *               )}
 *             </div>
 *           )}
 *         </FileUpload.FileSelector>
 *       </div>
 *     </FileUpload.Root>
 *   );
 * }
 * ```
 */
export function Root(props) {
    return (_jsx(WixServices, { servicesMap: createServicesMap().addService(FileUploadServiceDefinition, FileUploadService, props.fileUploadServiceConfig), children: props.children }));
}
/**
 * FileSelector - Handles file selection via drag & drop or file input
 *
 * @component
 * @example
 * ```tsx
 * import { FileUpload } from '@wix/media/components';
 *
 * function FileDropZone() {
 *   return (
 *     <FileUpload.FileSelector>
 *       {({ dragOver, handleDragOver, handleDragLeave, handleDrop, handleFileSelect, selectedFile, clearFile }) => (
 *         <div
 *           className={`drop-zone ${dragOver ? 'drag-over' : ''}`}
 *           onDragOver={handleDragOver}
 *           onDragLeave={handleDragLeave}
 *           onDrop={handleDrop}
 *         >
 *           {selectedFile ? (
 *             <div>
 *               Selected: {selectedFile.name}
 *               <button onClick={clearFile}>Remove</button>
 *             </div>
 *           ) : (
 *             <div>
 *               <p>Drag & drop a file here, or click to select</p>
 *               <input
 *                 type="file"
 *                 onChange={handleFileSelect}
 *                 style={{ display: 'none' }}
 *                 id="file-input"
 *               />
 *               <label htmlFor="file-input">Choose File</label>
 *             </div>
 *           )}
 *         </div>
 *       )}
 *     </FileUpload.FileSelector>
 *   );
 * }
 * ```
 */
export const FileSelector = (props) => {
    const service = useService(FileUploadServiceDefinition);
    const selectedFile = service.selectedFile.get();
    const previewUrl = service.previewUrl.get();
    const dragOver = service.dragOver.get();
    const handleDragOver = (event) => {
        event.preventDefault();
        service.setDragOver(true);
    };
    const handleDragLeave = (event) => {
        event.preventDefault();
        service.setDragOver(false);
    };
    const handleDrop = (event) => {
        event.preventDefault();
        service.setDragOver(false);
        const files = event.dataTransfer?.files;
        if (files && files.length > 0) {
            service.selectFile(files[0]);
        }
    };
    const handleFileSelect = (event) => {
        const files = event.target.files;
        if (files && files.length > 0) {
            service.selectFile(files[0]);
        }
    };
    const canPreview = selectedFile ? service.canPreview(selectedFile) : false;
    return props.children({
        selectedFile,
        previewUrl,
        dragOver,
        selectFile: service.selectFile,
        clearFile: service.clearFile,
        handleDragOver,
        handleDragLeave,
        handleDrop,
        handleFileSelect,
        canPreview,
    });
};
/**
 * UploadProgress - Shows upload status and progress
 *
 * @component
 * @example
 * ```tsx
 * import { FileUpload } from '@wix/media/components';
 *
 * function UploadStatusIndicator() {
 *   return (
 *     <FileUpload.UploadProgress>
 *       {({ uploadState, isLoading, isSuccess, isError, hasMessage }) => (
 *         <div className="upload-status">
 *           {isLoading && (
 *             <div className="loading">
 *               <div className="spinner" />
 *               Uploading...
 *             </div>
 *           )}
 *           {isSuccess && (
 *             <div className="success">
 *               ✅ Upload completed successfully!
 *             </div>
 *           )}
 *           {isError && (
 *             <div className="error">
 *               ❌ Upload failed
 *               {hasMessage && <p>{uploadState.message}</p>}
 *             </div>
 *           )}
 *         </div>
 *       )}
 *     </FileUpload.UploadProgress>
 *   );
 * }
 * ```
 */
export const UploadProgress = (props) => {
    const service = useService(FileUploadServiceDefinition);
    const uploadState = service.uploadState.get();
    return props.children({
        uploadState,
        isLoading: uploadState.type === 'loading',
        isSuccess: uploadState.type === 'success',
        isError: uploadState.type === 'error',
        hasError: uploadState.type === 'error',
        hasMessage: uploadState.message !== '',
    });
};
/**
 * UploadTrigger - Handles file upload action
 *
 * @component
 * @example
 * ```tsx
 * import { FileUpload } from '@wix/media/components';
 *
 * function UploadButton() {
 *   return (
 *     <FileUpload.UploadTrigger>
 *       {({ uploadFile, canUpload, isUploading }) => (
 *         <button
 *           onClick={uploadFile}
 *           disabled={!canUpload}
 *           className="upload-btn"
 *         >
 *           {isUploading ? 'Uploading...' : 'Upload File'}
 *         </button>
 *       )}
 *     </FileUpload.UploadTrigger>
 *   );
 * }
 * ```
 */
export const UploadTrigger = (props) => {
    const service = useService(FileUploadServiceDefinition);
    const selectedFile = service.selectedFile.get();
    const uploadState = service.uploadState.get();
    const canUpload = selectedFile !== null && uploadState.type !== 'loading';
    const isUploading = uploadState.type === 'loading';
    return props.children({
        uploadFile: service.uploadFile,
        canUpload,
        isUploading,
    });
};
/**
 * FilePreview - Displays preview information for the selected file
 *
 * @component
 * @example
 * ```tsx
 * import { FileUpload } from '@wix/media/components';
 *
 * function FilePreviewCard() {
 *   return (
 *     <FileUpload.FilePreview>
 *       {({ selectedFile, previewUrl, hasPreview, canPreview, fileName, formattedFileSize, fileType }) => (
 *         selectedFile && (
 *           <div className="file-preview">
 *             <div className="file-info">
 *               <h3>{fileName}</h3>
 *               <p>Size: {formattedFileSize}</p>
 *               <p>Type: {fileType}</p>
 *             </div>
 *             {canPreview && hasPreview && (
 *               <div className="preview">
 *                 <img src={previewUrl} alt="File preview" />
 *               </div>
 *             )}
 *           </div>
 *         )
 *       )}
 *     </FileUpload.FilePreview>
 *   );
 * }
 * ```
 */
export const FilePreview = (props) => {
    const service = useService(FileUploadServiceDefinition);
    const selectedFile = service.selectedFile.get();
    const previewUrl = service.previewUrl.get();
    const canPreview = selectedFile ? service.canPreview(selectedFile) : false;
    const hasPreview = previewUrl !== null;
    const formatFileSize = (bytes) => {
        if (bytes === 0)
            return '0 Bytes';
        const k = 1024;
        const sizes = ['Bytes', 'KB', 'MB', 'GB'];
        const i = Math.floor(Math.log(bytes) / Math.log(k));
        return parseFloat((bytes / Math.pow(k, i)).toFixed(2)) + ' ' + sizes[i];
    };
    return props.children({
        selectedFile,
        previewUrl,
        hasPreview,
        canPreview,
        fileName: selectedFile?.name || '',
        fileSize: selectedFile?.size || 0,
        formattedFileSize: selectedFile ? formatFileSize(selectedFile.size) : '',
        fileType: selectedFile?.type || '',
    });
};
/**
 * ValidationStatus - Shows file validation status and rules
 *
 * @component
 * @example
 * ```tsx
 * import { FileUpload } from '@wix/media/components';
 *
 * function FileValidationMessage() {
 *   return (
 *     <FileUpload.ValidationStatus>
 *       {({ isValid, error, validationRules }) => (
 *         <div className="validation-status">
 *           {!isValid && error && (
 *             <div className="error-message">
 *               ⚠️ {error}
 *             </div>
 *           )}
 *           <div className="rules">
 *             <h4>Upload Requirements:</h4>
 *             {validationRules.maxFileSize && (
 *               <p>Max size: {validationRules.maxFileSize / 1024 / 1024}MB</p>
 *             )}
 *             {validationRules.allowedTypes && (
 *               <p>Allowed types: {validationRules.allowedTypes.join(', ')}</p>
 *             )}
 *           </div>
 *         </div>
 *       )}
 *     </FileUpload.ValidationStatus>
 *   );
 * }
 * ```
 */
export const ValidationStatus = (props) => {
    const service = useService(FileUploadServiceDefinition);
    const selectedFile = service.selectedFile.get();
    const fileToValidate = props.file || selectedFile;
    const validationResult = fileToValidate
        ? service.validateFile(fileToValidate)
        : { isValid: true, error: undefined };
    return props.children({
        isValid: validationResult.isValid,
        error: validationResult.error,
        validationRules: service.validationRules,
    });
};
