// @wix/auto_sdk_faq_question-entry/build/internal/es/index.typings.mjs
// src/faq-question-entry-v2-question-entry-question-entry.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/faq-question-entry-v2-question-entry-question-entry.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKFloatToRESTFloat } from "@wix/sdk-runtime/transformations/float";
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixFaqQuestionentryV2QuestionEntryServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/question-entry",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/question-entry",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/question-entry",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/faq-server/v2/question-entries",
        destPath: "/v2/question-entries"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-question-entries",
        destPath: "/v2/draft-question-entries"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/faq-server/v2/question-entries",
        destPath: "/v2/question-entries"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-question-entries",
        destPath: "/v2/draft-question-entries"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/faq/v2/question-entries",
        destPath: "/v2/question-entries"
      },
      {
        srcPath: "/faq/v2/draft-question-entries",
        destPath: "/v2/draft-question-entries"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/faq/v2/question-entries",
        destPath: "/v2/question-entries"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/faq-server/v2/question-entries",
        destPath: "/v2/question-entries"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-question-entries",
        destPath: "/v2/draft-question-entries"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/faq-server/v2/question-entries",
        destPath: "/v2/question-entries"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-question-entries",
        destPath: "/v2/draft-question-entries"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/faq-server/v2/question-entries",
        destPath: "/v2/question-entries"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-question-entries",
        destPath: "/v2/draft-question-entries"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/faq-server/v2/question-entries",
        destPath: "/v2/question-entries"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-question-entries",
        destPath: "/v2/draft-question-entries"
      }
    ],
    _: [
      {
        srcPath: "/_api/faq-server/v2/question-entries",
        destPath: "/v2/question-entries"
      }
    ],
    "dev._base_domain_": [
      {
        srcPath: "/_api/faq-server/v2/question-entries",
        destPath: "/v2/question-entries"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-question-entries",
        destPath: "/v2/draft-question-entries"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/faq-server/v2/question-entries",
        destPath: "/v2/question-entries"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-question-entries",
        destPath: "/v2/draft-question-entries"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_faq_question-entry";
function createQuestionEntry(payload) {
  function __createQuestionEntry({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "questionEntry.createdDate" },
          { path: "questionEntry.updatedDate" },
          { path: "questionEntry.richContent.metadata.createdTimestamp" },
          { path: "questionEntry.richContent.metadata.updatedTimestamp" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "questionEntry.sortOrder" },
          {
            path: "questionEntry.richContent.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "questionEntry.richContent.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "questionEntry.richContent.nodes.galleryData.items.video.thumbnail.duration"
          },
          {
            path: "questionEntry.richContent.nodes.galleryData.options.item.ratio"
          },
          { path: "questionEntry.richContent.nodes.imageData.image.duration" },
          { path: "questionEntry.richContent.nodes.mapData.mapSettings.lat" },
          { path: "questionEntry.richContent.nodes.mapData.mapSettings.lng" },
          {
            path: "questionEntry.richContent.nodes.pollData.poll.image.duration"
          },
          {
            path: "questionEntry.richContent.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "questionEntry.richContent.nodes.pollData.design.poll.background.image.duration"
          },
          {
            path: "questionEntry.richContent.nodes.appEmbedData.image.duration"
          },
          { path: "questionEntry.richContent.nodes.videoData.video.duration" },
          {
            path: "questionEntry.richContent.nodes.videoData.thumbnail.duration"
          },
          { path: "questionEntry.richContent.nodes.audioData.audio.duration" },
          {
            path: "questionEntry.richContent.nodes.audioData.coverImage.duration"
          },
          { path: "questionEntry.labels.sortOrder" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.faq.question_entry.v2.question_entry",
      method: "POST",
      methodFqn: "wix.faq.questionentry.v2.QuestionEntryService.CreateQuestionEntry",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqQuestionentryV2QuestionEntryServiceUrl({
        protoPath: "/v2/question-entries",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "questionEntry.createdDate" },
            { path: "questionEntry.updatedDate" },
            { path: "questionEntry.richContent.metadata.createdTimestamp" },
            { path: "questionEntry.richContent.metadata.updatedTimestamp" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "questionEntry.sortOrder" },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "questionEntry.richContent.nodes.imageData.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "questionEntry.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.videoData.video.duration"
            },
            {
              path: "questionEntry.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "questionEntry.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "questionEntry.richContent.nodes.audioData.coverImage.duration"
            },
            { path: "questionEntry.labels.sortOrder" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createQuestionEntry;
}
function getQuestionEntry(payload) {
  function __getQuestionEntry({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.question_entry.v2.question_entry",
      method: "GET",
      methodFqn: "wix.faq.questionentry.v2.QuestionEntryService.GetQuestionEntry",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqQuestionentryV2QuestionEntryServiceUrl({
        protoPath: "/v2/question-entries/{questionEntryId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "questionEntry.createdDate" },
            { path: "questionEntry.updatedDate" },
            { path: "questionEntry.richContent.metadata.createdTimestamp" },
            { path: "questionEntry.richContent.metadata.updatedTimestamp" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "questionEntry.sortOrder" },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "questionEntry.richContent.nodes.imageData.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "questionEntry.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.videoData.video.duration"
            },
            {
              path: "questionEntry.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "questionEntry.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "questionEntry.richContent.nodes.audioData.coverImage.duration"
            },
            { path: "questionEntry.labels.sortOrder" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getQuestionEntry;
}
function updateQuestionEntry(payload) {
  function __updateQuestionEntry({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "questionEntry.createdDate" },
          { path: "questionEntry.updatedDate" },
          { path: "questionEntry.richContent.metadata.createdTimestamp" },
          { path: "questionEntry.richContent.metadata.updatedTimestamp" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "questionEntry.sortOrder" },
          {
            path: "questionEntry.richContent.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "questionEntry.richContent.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "questionEntry.richContent.nodes.galleryData.items.video.thumbnail.duration"
          },
          {
            path: "questionEntry.richContent.nodes.galleryData.options.item.ratio"
          },
          { path: "questionEntry.richContent.nodes.imageData.image.duration" },
          { path: "questionEntry.richContent.nodes.mapData.mapSettings.lat" },
          { path: "questionEntry.richContent.nodes.mapData.mapSettings.lng" },
          {
            path: "questionEntry.richContent.nodes.pollData.poll.image.duration"
          },
          {
            path: "questionEntry.richContent.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "questionEntry.richContent.nodes.pollData.design.poll.background.image.duration"
          },
          {
            path: "questionEntry.richContent.nodes.appEmbedData.image.duration"
          },
          { path: "questionEntry.richContent.nodes.videoData.video.duration" },
          {
            path: "questionEntry.richContent.nodes.videoData.thumbnail.duration"
          },
          { path: "questionEntry.richContent.nodes.audioData.audio.duration" },
          {
            path: "questionEntry.richContent.nodes.audioData.coverImage.duration"
          },
          { path: "questionEntry.labels.sortOrder" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.faq.question_entry.v2.question_entry",
      method: "PATCH",
      methodFqn: "wix.faq.questionentry.v2.QuestionEntryService.UpdateQuestionEntry",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqQuestionentryV2QuestionEntryServiceUrl({
        protoPath: "/v2/question-entries/{questionEntry.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "questionEntry.createdDate" },
            { path: "questionEntry.updatedDate" },
            { path: "questionEntry.richContent.metadata.createdTimestamp" },
            { path: "questionEntry.richContent.metadata.updatedTimestamp" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "questionEntry.sortOrder" },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "questionEntry.richContent.nodes.imageData.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "questionEntry.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.videoData.video.duration"
            },
            {
              path: "questionEntry.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "questionEntry.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "questionEntry.richContent.nodes.audioData.coverImage.duration"
            },
            { path: "questionEntry.labels.sortOrder" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateQuestionEntry;
}
function deleteQuestionEntry(payload) {
  function __deleteQuestionEntry({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.question_entry.v2.question_entry",
      method: "DELETE",
      methodFqn: "wix.faq.questionentry.v2.QuestionEntryService.DeleteQuestionEntry",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqQuestionentryV2QuestionEntryServiceUrl({
        protoPath: "/v2/question-entries/{questionEntryId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteQuestionEntry;
}
function listQuestionEntries(payload) {
  function __listQuestionEntries({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.question_entry.v2.question_entry",
      method: "GET",
      methodFqn: "wix.faq.questionentry.v2.QuestionEntryService.ListQuestionEntries",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqQuestionentryV2QuestionEntryServiceUrl({
        protoPath: "/v2/question-entries",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "questionEntries.createdDate" },
            { path: "questionEntries.updatedDate" },
            { path: "questionEntries.richContent.metadata.createdTimestamp" },
            { path: "questionEntries.richContent.metadata.updatedTimestamp" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "questionEntries.sortOrder" },
            {
              path: "questionEntries.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "questionEntries.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "questionEntries.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "questionEntries.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "questionEntries.richContent.nodes.imageData.image.duration"
            },
            {
              path: "questionEntries.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "questionEntries.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "questionEntries.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "questionEntries.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "questionEntries.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "questionEntries.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "questionEntries.richContent.nodes.videoData.video.duration"
            },
            {
              path: "questionEntries.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "questionEntries.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "questionEntries.richContent.nodes.audioData.coverImage.duration"
            },
            { path: "questionEntries.labels.sortOrder" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listQuestionEntries;
}
function queryQuestionEntries(payload) {
  function __queryQuestionEntries({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.question_entry.v2.question_entry",
      method: "POST",
      methodFqn: "wix.faq.questionentry.v2.QuestionEntryService.QueryQuestionEntries",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqQuestionentryV2QuestionEntryServiceUrl({
        protoPath: "/v2/question-entries/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "questionEntries.createdDate" },
            { path: "questionEntries.updatedDate" },
            { path: "questionEntries.richContent.metadata.createdTimestamp" },
            { path: "questionEntries.richContent.metadata.updatedTimestamp" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "questionEntries.sortOrder" },
            {
              path: "questionEntries.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "questionEntries.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "questionEntries.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "questionEntries.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "questionEntries.richContent.nodes.imageData.image.duration"
            },
            {
              path: "questionEntries.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "questionEntries.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "questionEntries.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "questionEntries.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "questionEntries.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "questionEntries.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "questionEntries.richContent.nodes.videoData.video.duration"
            },
            {
              path: "questionEntries.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "questionEntries.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "questionEntries.richContent.nodes.audioData.coverImage.duration"
            },
            { path: "questionEntries.labels.sortOrder" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryQuestionEntries;
}
function bulkDeleteQuestionEntries(payload) {
  function __bulkDeleteQuestionEntries({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.question_entry.v2.question_entry",
      method: "POST",
      methodFqn: "wix.faq.questionentry.v2.QuestionEntryService.BulkDeleteQuestionEntries",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqQuestionentryV2QuestionEntryServiceUrl({
        protoPath: "/v2/bulk/question-entries/delete",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkDeleteQuestionEntries;
}
function updateExtendedFields(payload) {
  function __updateExtendedFields({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.question_entry.v2.question_entry",
      method: "POST",
      methodFqn: "wix.faq.questionentry.v2.QuestionEntryService.UpdateExtendedFields",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqQuestionentryV2QuestionEntryServiceUrl({
        protoPath: "/v2/question-entries/{id}/update-extended-fields",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "questionEntry.createdDate" },
            { path: "questionEntry.updatedDate" },
            { path: "questionEntry.richContent.metadata.createdTimestamp" },
            { path: "questionEntry.richContent.metadata.updatedTimestamp" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "questionEntry.sortOrder" },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "questionEntry.richContent.nodes.imageData.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "questionEntry.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.videoData.video.duration"
            },
            {
              path: "questionEntry.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "questionEntry.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "questionEntry.richContent.nodes.audioData.coverImage.duration"
            },
            { path: "questionEntry.labels.sortOrder" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateExtendedFields;
}
function setQuestionEntryLabels(payload) {
  function __setQuestionEntryLabels({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [{ path: "labels.sortOrder" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.faq.question_entry.v2.question_entry",
      method: "PATCH",
      methodFqn: "wix.faq.questionentry.v2.QuestionEntryService.SetQuestionEntryLabels",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqQuestionentryV2QuestionEntryServiceUrl({
        protoPath: "/v2/question-entries/{questionEntryId}/labels",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "questionEntry.createdDate" },
            { path: "questionEntry.updatedDate" },
            { path: "questionEntry.richContent.metadata.createdTimestamp" },
            { path: "questionEntry.richContent.metadata.updatedTimestamp" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "questionEntry.sortOrder" },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "questionEntry.richContent.nodes.imageData.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "questionEntry.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.videoData.video.duration"
            },
            {
              path: "questionEntry.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "questionEntry.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "questionEntry.richContent.nodes.audioData.coverImage.duration"
            },
            { path: "questionEntry.labels.sortOrder" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __setQuestionEntryLabels;
}
function bulkUpdateQuestionEntry(payload) {
  function __bulkUpdateQuestionEntry({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "questionEntries.fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "questionEntries.questionEntry.createdDate" },
          { path: "questionEntries.questionEntry.updatedDate" },
          {
            path: "questionEntries.questionEntry.richContent.metadata.createdTimestamp"
          },
          {
            path: "questionEntries.questionEntry.richContent.metadata.updatedTimestamp"
          }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "questionEntries.questionEntry.sortOrder" },
          {
            path: "questionEntries.questionEntry.richContent.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.galleryData.items.video.thumbnail.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.galleryData.options.item.ratio"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.imageData.image.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.mapData.mapSettings.lat"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.mapData.mapSettings.lng"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.pollData.poll.image.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.pollData.design.poll.background.image.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.appEmbedData.image.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.videoData.video.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.videoData.thumbnail.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.audioData.audio.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.audioData.coverImage.duration"
          },
          { path: "questionEntries.questionEntry.labels.sortOrder" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.faq.question_entry.v2.question_entry",
      method: "POST",
      methodFqn: "wix.faq.questionentry.v2.QuestionEntryService.BulkUpdateQuestionEntry",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqQuestionentryV2QuestionEntryServiceUrl({
        protoPath: "/v2/bulk/question-entries/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.questionEntry.createdDate" },
            { path: "results.questionEntry.updatedDate" },
            {
              path: "results.questionEntry.richContent.metadata.createdTimestamp"
            },
            {
              path: "results.questionEntry.richContent.metadata.updatedTimestamp"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.questionEntry.sortOrder" },
            {
              path: "results.questionEntry.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "results.questionEntry.richContent.nodes.imageData.image.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "results.questionEntry.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "results.questionEntry.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.videoData.video.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.audioData.coverImage.duration"
            },
            { path: "results.questionEntry.labels.sortOrder" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateQuestionEntry;
}

// src/faq-question-entry-v2-question-entry-question-entry.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var NodeType = /* @__PURE__ */ ((NodeType2) => {
  NodeType2["PARAGRAPH"] = "PARAGRAPH";
  NodeType2["TEXT"] = "TEXT";
  NodeType2["HEADING"] = "HEADING";
  NodeType2["BULLETED_LIST"] = "BULLETED_LIST";
  NodeType2["ORDERED_LIST"] = "ORDERED_LIST";
  NodeType2["LIST_ITEM"] = "LIST_ITEM";
  NodeType2["BLOCKQUOTE"] = "BLOCKQUOTE";
  NodeType2["CODE_BLOCK"] = "CODE_BLOCK";
  NodeType2["VIDEO"] = "VIDEO";
  NodeType2["DIVIDER"] = "DIVIDER";
  NodeType2["FILE"] = "FILE";
  NodeType2["GALLERY"] = "GALLERY";
  NodeType2["GIF"] = "GIF";
  NodeType2["HTML"] = "HTML";
  NodeType2["IMAGE"] = "IMAGE";
  NodeType2["LINK_PREVIEW"] = "LINK_PREVIEW";
  NodeType2["MAP"] = "MAP";
  NodeType2["POLL"] = "POLL";
  NodeType2["APP_EMBED"] = "APP_EMBED";
  NodeType2["BUTTON"] = "BUTTON";
  NodeType2["COLLAPSIBLE_LIST"] = "COLLAPSIBLE_LIST";
  NodeType2["TABLE"] = "TABLE";
  NodeType2["EMBED"] = "EMBED";
  NodeType2["COLLAPSIBLE_ITEM"] = "COLLAPSIBLE_ITEM";
  NodeType2["COLLAPSIBLE_ITEM_TITLE"] = "COLLAPSIBLE_ITEM_TITLE";
  NodeType2["COLLAPSIBLE_ITEM_BODY"] = "COLLAPSIBLE_ITEM_BODY";
  NodeType2["TABLE_CELL"] = "TABLE_CELL";
  NodeType2["TABLE_ROW"] = "TABLE_ROW";
  NodeType2["EXTERNAL"] = "EXTERNAL";
  NodeType2["AUDIO"] = "AUDIO";
  NodeType2["CAPTION"] = "CAPTION";
  NodeType2["LAYOUT"] = "LAYOUT";
  NodeType2["LAYOUT_CELL"] = "LAYOUT_CELL";
  return NodeType2;
})(NodeType || {});
var WidthType = /* @__PURE__ */ ((WidthType2) => {
  WidthType2["CONTENT"] = "CONTENT";
  WidthType2["SMALL"] = "SMALL";
  WidthType2["ORIGINAL"] = "ORIGINAL";
  WidthType2["FULL_WIDTH"] = "FULL_WIDTH";
  return WidthType2;
})(WidthType || {});
var PluginContainerDataAlignment = /* @__PURE__ */ ((PluginContainerDataAlignment2) => {
  PluginContainerDataAlignment2["CENTER"] = "CENTER";
  PluginContainerDataAlignment2["LEFT"] = "LEFT";
  PluginContainerDataAlignment2["RIGHT"] = "RIGHT";
  return PluginContainerDataAlignment2;
})(PluginContainerDataAlignment || {});
var ButtonDataType = /* @__PURE__ */ ((ButtonDataType2) => {
  ButtonDataType2["LINK"] = "LINK";
  ButtonDataType2["ACTION"] = "ACTION";
  return ButtonDataType2;
})(ButtonDataType || {});
var Target = /* @__PURE__ */ ((Target2) => {
  Target2["SELF"] = "SELF";
  Target2["BLANK"] = "BLANK";
  Target2["PARENT"] = "PARENT";
  Target2["TOP"] = "TOP";
  return Target2;
})(Target || {});
var TextAlignment = /* @__PURE__ */ ((TextAlignment2) => {
  TextAlignment2["AUTO"] = "AUTO";
  TextAlignment2["LEFT"] = "LEFT";
  TextAlignment2["RIGHT"] = "RIGHT";
  TextAlignment2["CENTER"] = "CENTER";
  TextAlignment2["JUSTIFY"] = "JUSTIFY";
  return TextAlignment2;
})(TextAlignment || {});
var LineStyle = /* @__PURE__ */ ((LineStyle2) => {
  LineStyle2["SINGLE"] = "SINGLE";
  LineStyle2["DOUBLE"] = "DOUBLE";
  LineStyle2["DASHED"] = "DASHED";
  LineStyle2["DOTTED"] = "DOTTED";
  return LineStyle2;
})(LineStyle || {});
var Width = /* @__PURE__ */ ((Width2) => {
  Width2["LARGE"] = "LARGE";
  Width2["MEDIUM"] = "MEDIUM";
  Width2["SMALL"] = "SMALL";
  return Width2;
})(Width || {});
var DividerDataAlignment = /* @__PURE__ */ ((DividerDataAlignment2) => {
  DividerDataAlignment2["CENTER"] = "CENTER";
  DividerDataAlignment2["LEFT"] = "LEFT";
  DividerDataAlignment2["RIGHT"] = "RIGHT";
  return DividerDataAlignment2;
})(DividerDataAlignment || {});
var ViewMode = /* @__PURE__ */ ((ViewMode2) => {
  ViewMode2["NONE"] = "NONE";
  ViewMode2["FULL"] = "FULL";
  ViewMode2["MINI"] = "MINI";
  return ViewMode2;
})(ViewMode || {});
var LayoutType = /* @__PURE__ */ ((LayoutType2) => {
  LayoutType2["COLLAGE"] = "COLLAGE";
  LayoutType2["MASONRY"] = "MASONRY";
  LayoutType2["GRID"] = "GRID";
  LayoutType2["THUMBNAIL"] = "THUMBNAIL";
  LayoutType2["SLIDER"] = "SLIDER";
  LayoutType2["SLIDESHOW"] = "SLIDESHOW";
  LayoutType2["PANORAMA"] = "PANORAMA";
  LayoutType2["COLUMN"] = "COLUMN";
  LayoutType2["MAGIC"] = "MAGIC";
  LayoutType2["FULLSIZE"] = "FULLSIZE";
  return LayoutType2;
})(LayoutType || {});
var Orientation = /* @__PURE__ */ ((Orientation2) => {
  Orientation2["ROWS"] = "ROWS";
  Orientation2["COLUMNS"] = "COLUMNS";
  return Orientation2;
})(Orientation || {});
var Crop = /* @__PURE__ */ ((Crop2) => {
  Crop2["FILL"] = "FILL";
  Crop2["FIT"] = "FIT";
  return Crop2;
})(Crop || {});
var ThumbnailsAlignment = /* @__PURE__ */ ((ThumbnailsAlignment2) => {
  ThumbnailsAlignment2["TOP"] = "TOP";
  ThumbnailsAlignment2["RIGHT"] = "RIGHT";
  ThumbnailsAlignment2["BOTTOM"] = "BOTTOM";
  ThumbnailsAlignment2["LEFT"] = "LEFT";
  ThumbnailsAlignment2["NONE"] = "NONE";
  return ThumbnailsAlignment2;
})(ThumbnailsAlignment || {});
var GIFType = /* @__PURE__ */ ((GIFType2) => {
  GIFType2["NORMAL"] = "NORMAL";
  GIFType2["STICKER"] = "STICKER";
  return GIFType2;
})(GIFType || {});
var Source = /* @__PURE__ */ ((Source2) => {
  Source2["HTML"] = "HTML";
  Source2["ADSENSE"] = "ADSENSE";
  return Source2;
})(Source || {});
var StylesPosition = /* @__PURE__ */ ((StylesPosition2) => {
  StylesPosition2["START"] = "START";
  StylesPosition2["END"] = "END";
  StylesPosition2["TOP"] = "TOP";
  StylesPosition2["HIDDEN"] = "HIDDEN";
  return StylesPosition2;
})(StylesPosition || {});
var MapType = /* @__PURE__ */ ((MapType2) => {
  MapType2["ROADMAP"] = "ROADMAP";
  MapType2["SATELITE"] = "SATELITE";
  MapType2["HYBRID"] = "HYBRID";
  MapType2["TERRAIN"] = "TERRAIN";
  return MapType2;
})(MapType || {});
var ViewRole = /* @__PURE__ */ ((ViewRole2) => {
  ViewRole2["CREATOR"] = "CREATOR";
  ViewRole2["VOTERS"] = "VOTERS";
  ViewRole2["EVERYONE"] = "EVERYONE";
  return ViewRole2;
})(ViewRole || {});
var VoteRole = /* @__PURE__ */ ((VoteRole2) => {
  VoteRole2["SITE_MEMBERS"] = "SITE_MEMBERS";
  VoteRole2["ALL"] = "ALL";
  return VoteRole2;
})(VoteRole || {});
var PollLayoutType = /* @__PURE__ */ ((PollLayoutType2) => {
  PollLayoutType2["LIST"] = "LIST";
  PollLayoutType2["GRID"] = "GRID";
  return PollLayoutType2;
})(PollLayoutType || {});
var PollLayoutDirection = /* @__PURE__ */ ((PollLayoutDirection2) => {
  PollLayoutDirection2["LTR"] = "LTR";
  PollLayoutDirection2["RTL"] = "RTL";
  return PollLayoutDirection2;
})(PollLayoutDirection || {});
var BackgroundType = /* @__PURE__ */ ((BackgroundType2) => {
  BackgroundType2["COLOR"] = "COLOR";
  BackgroundType2["IMAGE"] = "IMAGE";
  BackgroundType2["GRADIENT"] = "GRADIENT";
  return BackgroundType2;
})(BackgroundType || {});
var DecorationType = /* @__PURE__ */ ((DecorationType2) => {
  DecorationType2["BOLD"] = "BOLD";
  DecorationType2["ITALIC"] = "ITALIC";
  DecorationType2["UNDERLINE"] = "UNDERLINE";
  DecorationType2["SPOILER"] = "SPOILER";
  DecorationType2["ANCHOR"] = "ANCHOR";
  DecorationType2["MENTION"] = "MENTION";
  DecorationType2["LINK"] = "LINK";
  DecorationType2["COLOR"] = "COLOR";
  DecorationType2["FONT_SIZE"] = "FONT_SIZE";
  DecorationType2["EXTERNAL"] = "EXTERNAL";
  DecorationType2["STRIKETHROUGH"] = "STRIKETHROUGH";
  DecorationType2["SUPERSCRIPT"] = "SUPERSCRIPT";
  DecorationType2["SUBSCRIPT"] = "SUBSCRIPT";
  return DecorationType2;
})(DecorationType || {});
var FontType = /* @__PURE__ */ ((FontType2) => {
  FontType2["PX"] = "PX";
  FontType2["EM"] = "EM";
  return FontType2;
})(FontType || {});
var Position = /* @__PURE__ */ ((Position2) => {
  Position2["START"] = "START";
  Position2["END"] = "END";
  Position2["TOP"] = "TOP";
  return Position2;
})(Position || {});
var AspectRatio = /* @__PURE__ */ ((AspectRatio2) => {
  AspectRatio2["SQUARE"] = "SQUARE";
  AspectRatio2["RECTANGLE"] = "RECTANGLE";
  return AspectRatio2;
})(AspectRatio || {});
var Resizing = /* @__PURE__ */ ((Resizing2) => {
  Resizing2["FILL"] = "FILL";
  Resizing2["FIT"] = "FIT";
  return Resizing2;
})(Resizing || {});
var Placement = /* @__PURE__ */ ((Placement2) => {
  Placement2["IMAGE"] = "IMAGE";
  Placement2["PRODUCT_INFO"] = "PRODUCT_INFO";
  return Placement2;
})(Placement || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["CONTAINED"] = "CONTAINED";
  Type2["FRAMELESS"] = "FRAMELESS";
  return Type2;
})(Type || {});
var Alignment = /* @__PURE__ */ ((Alignment2) => {
  Alignment2["START"] = "START";
  Alignment2["CENTER"] = "CENTER";
  Alignment2["END"] = "END";
  return Alignment2;
})(Alignment || {});
var Layout = /* @__PURE__ */ ((Layout2) => {
  Layout2["STACKED"] = "STACKED";
  Layout2["SIDE_BY_SIDE"] = "SIDE_BY_SIDE";
  return Layout2;
})(Layout || {});
var AppType = /* @__PURE__ */ ((AppType2) => {
  AppType2["PRODUCT"] = "PRODUCT";
  AppType2["EVENT"] = "EVENT";
  AppType2["BOOKING"] = "BOOKING";
  return AppType2;
})(AppType || {});
var InitialExpandedItems = /* @__PURE__ */ ((InitialExpandedItems2) => {
  InitialExpandedItems2["FIRST"] = "FIRST";
  InitialExpandedItems2["ALL"] = "ALL";
  InitialExpandedItems2["NONE"] = "NONE";
  return InitialExpandedItems2;
})(InitialExpandedItems || {});
var Direction = /* @__PURE__ */ ((Direction2) => {
  Direction2["LTR"] = "LTR";
  Direction2["RTL"] = "RTL";
  return Direction2;
})(Direction || {});
var VerticalAlignment = /* @__PURE__ */ ((VerticalAlignment2) => {
  VerticalAlignment2["TOP"] = "TOP";
  VerticalAlignment2["MIDDLE"] = "MIDDLE";
  VerticalAlignment2["BOTTOM"] = "BOTTOM";
  return VerticalAlignment2;
})(VerticalAlignment || {});
var NullValue = /* @__PURE__ */ ((NullValue2) => {
  NullValue2["NULL_VALUE"] = "NULL_VALUE";
  return NullValue2;
})(NullValue || {});
var QuestionStatus = /* @__PURE__ */ ((QuestionStatus2) => {
  QuestionStatus2["UNKNOWN"] = "UNKNOWN";
  QuestionStatus2["VISIBLE"] = "VISIBLE";
  QuestionStatus2["HIDDEN"] = "HIDDEN";
  return QuestionStatus2;
})(QuestionStatus || {});
var ContentFormat = /* @__PURE__ */ ((ContentFormat2) => {
  ContentFormat2["DRAFTJS"] = "DRAFTJS";
  ContentFormat2["PLAIN_TEXT"] = "PLAIN_TEXT";
  ContentFormat2["RICH_CONTENT"] = "RICH_CONTENT";
  return ContentFormat2;
})(ContentFormat || {});
var FieldSet = /* @__PURE__ */ ((FieldSet2) => {
  FieldSet2["UNKNOWN"] = "UNKNOWN";
  FieldSet2["SHARE_LINKS"] = "SHARE_LINKS";
  return FieldSet2;
})(FieldSet || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var QueryQuestionEntriesRequestContentFormat = /* @__PURE__ */ ((QueryQuestionEntriesRequestContentFormat2) => {
  QueryQuestionEntriesRequestContentFormat2["DRAFTJS"] = "DRAFTJS";
  QueryQuestionEntriesRequestContentFormat2["PLAIN_TEXT"] = "PLAIN_TEXT";
  QueryQuestionEntriesRequestContentFormat2["RICH_CONTENT"] = "RICH_CONTENT";
  return QueryQuestionEntriesRequestContentFormat2;
})(QueryQuestionEntriesRequestContentFormat || {});
var QueryQuestionEntriesRequestFieldSet = /* @__PURE__ */ ((QueryQuestionEntriesRequestFieldSet2) => {
  QueryQuestionEntriesRequestFieldSet2["UNKNOWN"] = "UNKNOWN";
  QueryQuestionEntriesRequestFieldSet2["SHARE_LINKS"] = "SHARE_LINKS";
  return QueryQuestionEntriesRequestFieldSet2;
})(QueryQuestionEntriesRequestFieldSet || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createQuestionEntry2(questionEntry) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest(
    { questionEntry },
    ["questionEntry.richContent"]
  );
  const reqOpts = createQuestionEntry(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data, [
      "questionEntry.richContent"
    ])?.questionEntry;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { questionEntry: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["questionEntry"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getQuestionEntry2(questionEntryId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest(
    { questionEntryId },
    []
  );
  const reqOpts = getQuestionEntry(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data, [
      "questionEntry.richContent"
    ])?.questionEntry;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { questionEntryId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["questionEntryId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateQuestionEntry2(_id, questionEntry) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest(
    { questionEntry: { ...questionEntry, id: _id } },
    ["questionEntry.richContent"]
  );
  const reqOpts = updateQuestionEntry(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data, [
      "questionEntry.richContent"
    ])?.questionEntry;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { questionEntry: "$[1]" },
        explicitPathsToArguments: { "questionEntry.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "questionEntry"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteQuestionEntry2(questionEntryId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest(
    { questionEntryId },
    []
  );
  const reqOpts = deleteQuestionEntry(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { questionEntryId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["questionEntryId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listQuestionEntries2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest(
    {
      categoryId: options?.categoryId,
      questionEntryIds: options?.questionEntryIds,
      questionEntryLabel: options?.questionEntryLabel,
      paging: options?.paging,
      contentFormat: options?.contentFormat,
      fieldSet: options?.fieldSet
    },
    []
  );
  const reqOpts = listQuestionEntries(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data, [
      "questionEntries.richContent"
    ]);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          categoryId: "$[0].categoryId",
          questionEntryIds: "$[0].questionEntryIds",
          questionEntryLabel: "$[0].questionEntryLabel",
          paging: "$[0].paging",
          contentFormat: "$[0].contentFormat",
          fieldSet: "$[0].fieldSet"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryQuestionEntries2(options) {
  const { httpClient, sideEffects } = arguments[1];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryQuestionEntries({
        ...payload,
        ...options ?? {}
      });
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, options];
      return renameKeysFromSDKRequestToRESTRequest(
        { ...args?.[1], query: args?.[0] },
        []
      );
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, []),
        ["questionEntries.richContent"]
      );
      return {
        items: transformedData?.questionEntries,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function bulkDeleteQuestionEntries2(questionEntryIds) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest(
    { questionEntryIds },
    []
  );
  const reqOpts = bulkDeleteQuestionEntries(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data, []);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { questionEntryIds: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["questionEntryIds"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateExtendedFields2(_id, namespace, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = renameKeysFromSDKRequestToRESTRequest(
    { id: _id, namespace, namespaceData: options?.namespaceData },
    []
  );
  const reqOpts = updateExtendedFields(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data, [
      "questionEntry.richContent"
    ]);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          id: "$[0]",
          namespace: "$[1]",
          namespaceData: "$[2].namespaceData"
        },
        singleArgumentUnchanged: false
      },
      ["_id", "namespace", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function setQuestionEntryLabels2(questionEntryId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest(
    { questionEntryId, labels: options?.labels },
    []
  );
  const reqOpts = setQuestionEntryLabels(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data, [
      "questionEntry.richContent"
    ]);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          questionEntryId: "$[0]",
          labels: "$[1].labels"
        },
        singleArgumentUnchanged: false
      },
      ["questionEntryId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateQuestionEntry2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest(
    {
      questionEntries: options?.questionEntries,
      returnFullEntity: options?.returnFullEntity
    },
    ["questionEntries.questionEntry.richContent"]
  );
  const reqOpts = bulkUpdateQuestionEntry(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data, [
      "results.questionEntry.richContent"
    ]);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          questionEntries: "$[0].questionEntries",
          returnFullEntity: "$[0].returnFullEntity"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
export {
  Alignment,
  AppType,
  AspectRatio,
  BackgroundType,
  ButtonDataType,
  ContentFormat,
  Crop,
  DecorationType,
  Direction,
  DividerDataAlignment,
  FieldSet,
  FontType,
  GIFType,
  InitialExpandedItems,
  Layout,
  LayoutType,
  LineStyle,
  MapType,
  NodeType,
  NullValue,
  Orientation,
  Placement,
  PluginContainerDataAlignment,
  PollLayoutDirection,
  PollLayoutType,
  Position,
  QueryQuestionEntriesRequestContentFormat,
  QueryQuestionEntriesRequestFieldSet,
  QuestionStatus,
  Resizing,
  SortOrder,
  Source,
  StylesPosition,
  Target,
  TextAlignment,
  ThumbnailsAlignment,
  Type,
  VerticalAlignment,
  ViewMode,
  ViewRole,
  VoteRole,
  WebhookIdentityType,
  Width,
  WidthType,
  bulkDeleteQuestionEntries2 as bulkDeleteQuestionEntries,
  bulkUpdateQuestionEntry2 as bulkUpdateQuestionEntry,
  createQuestionEntry2 as createQuestionEntry,
  deleteQuestionEntry2 as deleteQuestionEntry,
  getQuestionEntry2 as getQuestionEntry,
  listQuestionEntries2 as listQuestionEntries,
  queryQuestionEntries2 as queryQuestionEntries,
  setQuestionEntryLabels2 as setQuestionEntryLabels,
  updateExtendedFields2 as updateExtendedFields,
  updateQuestionEntry2 as updateQuestionEntry
};
