// @wix/auto_sdk_faq_question-entry/build/internal/es/meta.mjs
// src/faq-question-entry-v2-question-entry-question-entry.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKFloatToRESTFloat } from "@wix/sdk-runtime/transformations/float";
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixFaqQuestionentryV2QuestionEntryServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/question-entry",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/question-entry",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/question-entry",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/faq-server/v2/question-entries",
        destPath: "/v2/question-entries"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-question-entries",
        destPath: "/v2/draft-question-entries"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/faq-server/v2/question-entries",
        destPath: "/v2/question-entries"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-question-entries",
        destPath: "/v2/draft-question-entries"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/faq/v2/question-entries",
        destPath: "/v2/question-entries"
      },
      {
        srcPath: "/faq/v2/draft-question-entries",
        destPath: "/v2/draft-question-entries"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/faq/v2/question-entries",
        destPath: "/v2/question-entries"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/faq-server/v2/question-entries",
        destPath: "/v2/question-entries"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-question-entries",
        destPath: "/v2/draft-question-entries"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/faq-server/v2/question-entries",
        destPath: "/v2/question-entries"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-question-entries",
        destPath: "/v2/draft-question-entries"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/faq-server/v2/question-entries",
        destPath: "/v2/question-entries"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-question-entries",
        destPath: "/v2/draft-question-entries"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/faq-server/v2/question-entries",
        destPath: "/v2/question-entries"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-question-entries",
        destPath: "/v2/draft-question-entries"
      }
    ],
    _: [
      {
        srcPath: "/_api/faq-server/v2/question-entries",
        destPath: "/v2/question-entries"
      }
    ],
    "dev._base_domain_": [
      {
        srcPath: "/_api/faq-server/v2/question-entries",
        destPath: "/v2/question-entries"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-question-entries",
        destPath: "/v2/draft-question-entries"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/faq-server/v2/question-entries",
        destPath: "/v2/question-entries"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-question-entries",
        destPath: "/v2/draft-question-entries"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_faq_question-entry";
function createQuestionEntry(payload) {
  function __createQuestionEntry({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "questionEntry.createdDate" },
          { path: "questionEntry.updatedDate" },
          { path: "questionEntry.richContent.metadata.createdTimestamp" },
          { path: "questionEntry.richContent.metadata.updatedTimestamp" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "questionEntry.sortOrder" },
          {
            path: "questionEntry.richContent.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "questionEntry.richContent.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "questionEntry.richContent.nodes.galleryData.items.video.thumbnail.duration"
          },
          {
            path: "questionEntry.richContent.nodes.galleryData.options.item.ratio"
          },
          { path: "questionEntry.richContent.nodes.imageData.image.duration" },
          { path: "questionEntry.richContent.nodes.mapData.mapSettings.lat" },
          { path: "questionEntry.richContent.nodes.mapData.mapSettings.lng" },
          {
            path: "questionEntry.richContent.nodes.pollData.poll.image.duration"
          },
          {
            path: "questionEntry.richContent.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "questionEntry.richContent.nodes.pollData.design.poll.background.image.duration"
          },
          {
            path: "questionEntry.richContent.nodes.appEmbedData.image.duration"
          },
          { path: "questionEntry.richContent.nodes.videoData.video.duration" },
          {
            path: "questionEntry.richContent.nodes.videoData.thumbnail.duration"
          },
          { path: "questionEntry.richContent.nodes.audioData.audio.duration" },
          {
            path: "questionEntry.richContent.nodes.audioData.coverImage.duration"
          },
          { path: "questionEntry.labels.sortOrder" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.faq.question_entry.v2.question_entry",
      method: "POST",
      methodFqn: "wix.faq.questionentry.v2.QuestionEntryService.CreateQuestionEntry",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqQuestionentryV2QuestionEntryServiceUrl({
        protoPath: "/v2/question-entries",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "questionEntry.createdDate" },
            { path: "questionEntry.updatedDate" },
            { path: "questionEntry.richContent.metadata.createdTimestamp" },
            { path: "questionEntry.richContent.metadata.updatedTimestamp" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "questionEntry.sortOrder" },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "questionEntry.richContent.nodes.imageData.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "questionEntry.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.videoData.video.duration"
            },
            {
              path: "questionEntry.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "questionEntry.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "questionEntry.richContent.nodes.audioData.coverImage.duration"
            },
            { path: "questionEntry.labels.sortOrder" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createQuestionEntry;
}
function getQuestionEntry(payload) {
  function __getQuestionEntry({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.question_entry.v2.question_entry",
      method: "GET",
      methodFqn: "wix.faq.questionentry.v2.QuestionEntryService.GetQuestionEntry",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqQuestionentryV2QuestionEntryServiceUrl({
        protoPath: "/v2/question-entries/{questionEntryId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "questionEntry.createdDate" },
            { path: "questionEntry.updatedDate" },
            { path: "questionEntry.richContent.metadata.createdTimestamp" },
            { path: "questionEntry.richContent.metadata.updatedTimestamp" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "questionEntry.sortOrder" },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "questionEntry.richContent.nodes.imageData.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "questionEntry.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.videoData.video.duration"
            },
            {
              path: "questionEntry.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "questionEntry.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "questionEntry.richContent.nodes.audioData.coverImage.duration"
            },
            { path: "questionEntry.labels.sortOrder" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getQuestionEntry;
}
function updateQuestionEntry(payload) {
  function __updateQuestionEntry({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "questionEntry.createdDate" },
          { path: "questionEntry.updatedDate" },
          { path: "questionEntry.richContent.metadata.createdTimestamp" },
          { path: "questionEntry.richContent.metadata.updatedTimestamp" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "questionEntry.sortOrder" },
          {
            path: "questionEntry.richContent.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "questionEntry.richContent.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "questionEntry.richContent.nodes.galleryData.items.video.thumbnail.duration"
          },
          {
            path: "questionEntry.richContent.nodes.galleryData.options.item.ratio"
          },
          { path: "questionEntry.richContent.nodes.imageData.image.duration" },
          { path: "questionEntry.richContent.nodes.mapData.mapSettings.lat" },
          { path: "questionEntry.richContent.nodes.mapData.mapSettings.lng" },
          {
            path: "questionEntry.richContent.nodes.pollData.poll.image.duration"
          },
          {
            path: "questionEntry.richContent.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "questionEntry.richContent.nodes.pollData.design.poll.background.image.duration"
          },
          {
            path: "questionEntry.richContent.nodes.appEmbedData.image.duration"
          },
          { path: "questionEntry.richContent.nodes.videoData.video.duration" },
          {
            path: "questionEntry.richContent.nodes.videoData.thumbnail.duration"
          },
          { path: "questionEntry.richContent.nodes.audioData.audio.duration" },
          {
            path: "questionEntry.richContent.nodes.audioData.coverImage.duration"
          },
          { path: "questionEntry.labels.sortOrder" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.faq.question_entry.v2.question_entry",
      method: "PATCH",
      methodFqn: "wix.faq.questionentry.v2.QuestionEntryService.UpdateQuestionEntry",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqQuestionentryV2QuestionEntryServiceUrl({
        protoPath: "/v2/question-entries/{questionEntry.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "questionEntry.createdDate" },
            { path: "questionEntry.updatedDate" },
            { path: "questionEntry.richContent.metadata.createdTimestamp" },
            { path: "questionEntry.richContent.metadata.updatedTimestamp" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "questionEntry.sortOrder" },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "questionEntry.richContent.nodes.imageData.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "questionEntry.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.videoData.video.duration"
            },
            {
              path: "questionEntry.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "questionEntry.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "questionEntry.richContent.nodes.audioData.coverImage.duration"
            },
            { path: "questionEntry.labels.sortOrder" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateQuestionEntry;
}
function deleteQuestionEntry(payload) {
  function __deleteQuestionEntry({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.question_entry.v2.question_entry",
      method: "DELETE",
      methodFqn: "wix.faq.questionentry.v2.QuestionEntryService.DeleteQuestionEntry",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqQuestionentryV2QuestionEntryServiceUrl({
        protoPath: "/v2/question-entries/{questionEntryId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteQuestionEntry;
}
function listQuestionEntries(payload) {
  function __listQuestionEntries({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.question_entry.v2.question_entry",
      method: "GET",
      methodFqn: "wix.faq.questionentry.v2.QuestionEntryService.ListQuestionEntries",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqQuestionentryV2QuestionEntryServiceUrl({
        protoPath: "/v2/question-entries",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "questionEntries.createdDate" },
            { path: "questionEntries.updatedDate" },
            { path: "questionEntries.richContent.metadata.createdTimestamp" },
            { path: "questionEntries.richContent.metadata.updatedTimestamp" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "questionEntries.sortOrder" },
            {
              path: "questionEntries.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "questionEntries.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "questionEntries.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "questionEntries.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "questionEntries.richContent.nodes.imageData.image.duration"
            },
            {
              path: "questionEntries.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "questionEntries.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "questionEntries.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "questionEntries.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "questionEntries.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "questionEntries.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "questionEntries.richContent.nodes.videoData.video.duration"
            },
            {
              path: "questionEntries.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "questionEntries.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "questionEntries.richContent.nodes.audioData.coverImage.duration"
            },
            { path: "questionEntries.labels.sortOrder" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listQuestionEntries;
}
function queryQuestionEntries(payload) {
  function __queryQuestionEntries({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.question_entry.v2.question_entry",
      method: "POST",
      methodFqn: "wix.faq.questionentry.v2.QuestionEntryService.QueryQuestionEntries",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqQuestionentryV2QuestionEntryServiceUrl({
        protoPath: "/v2/question-entries/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "questionEntries.createdDate" },
            { path: "questionEntries.updatedDate" },
            { path: "questionEntries.richContent.metadata.createdTimestamp" },
            { path: "questionEntries.richContent.metadata.updatedTimestamp" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "questionEntries.sortOrder" },
            {
              path: "questionEntries.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "questionEntries.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "questionEntries.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "questionEntries.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "questionEntries.richContent.nodes.imageData.image.duration"
            },
            {
              path: "questionEntries.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "questionEntries.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "questionEntries.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "questionEntries.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "questionEntries.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "questionEntries.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "questionEntries.richContent.nodes.videoData.video.duration"
            },
            {
              path: "questionEntries.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "questionEntries.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "questionEntries.richContent.nodes.audioData.coverImage.duration"
            },
            { path: "questionEntries.labels.sortOrder" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryQuestionEntries;
}
function bulkDeleteQuestionEntries(payload) {
  function __bulkDeleteQuestionEntries({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.question_entry.v2.question_entry",
      method: "POST",
      methodFqn: "wix.faq.questionentry.v2.QuestionEntryService.BulkDeleteQuestionEntries",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqQuestionentryV2QuestionEntryServiceUrl({
        protoPath: "/v2/bulk/question-entries/delete",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkDeleteQuestionEntries;
}
function updateExtendedFields(payload) {
  function __updateExtendedFields({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.question_entry.v2.question_entry",
      method: "POST",
      methodFqn: "wix.faq.questionentry.v2.QuestionEntryService.UpdateExtendedFields",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqQuestionentryV2QuestionEntryServiceUrl({
        protoPath: "/v2/question-entries/{id}/update-extended-fields",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "questionEntry.createdDate" },
            { path: "questionEntry.updatedDate" },
            { path: "questionEntry.richContent.metadata.createdTimestamp" },
            { path: "questionEntry.richContent.metadata.updatedTimestamp" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "questionEntry.sortOrder" },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "questionEntry.richContent.nodes.imageData.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "questionEntry.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.videoData.video.duration"
            },
            {
              path: "questionEntry.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "questionEntry.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "questionEntry.richContent.nodes.audioData.coverImage.duration"
            },
            { path: "questionEntry.labels.sortOrder" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateExtendedFields;
}
function setQuestionEntryLabels(payload) {
  function __setQuestionEntryLabels({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [{ path: "labels.sortOrder" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.faq.question_entry.v2.question_entry",
      method: "PATCH",
      methodFqn: "wix.faq.questionentry.v2.QuestionEntryService.SetQuestionEntryLabels",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqQuestionentryV2QuestionEntryServiceUrl({
        protoPath: "/v2/question-entries/{questionEntryId}/labels",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "questionEntry.createdDate" },
            { path: "questionEntry.updatedDate" },
            { path: "questionEntry.richContent.metadata.createdTimestamp" },
            { path: "questionEntry.richContent.metadata.updatedTimestamp" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "questionEntry.sortOrder" },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "questionEntry.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "questionEntry.richContent.nodes.imageData.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "questionEntry.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "questionEntry.richContent.nodes.videoData.video.duration"
            },
            {
              path: "questionEntry.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "questionEntry.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "questionEntry.richContent.nodes.audioData.coverImage.duration"
            },
            { path: "questionEntry.labels.sortOrder" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __setQuestionEntryLabels;
}
function bulkUpdateQuestionEntry(payload) {
  function __bulkUpdateQuestionEntry({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "questionEntries.fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "questionEntries.questionEntry.createdDate" },
          { path: "questionEntries.questionEntry.updatedDate" },
          {
            path: "questionEntries.questionEntry.richContent.metadata.createdTimestamp"
          },
          {
            path: "questionEntries.questionEntry.richContent.metadata.updatedTimestamp"
          }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "questionEntries.questionEntry.sortOrder" },
          {
            path: "questionEntries.questionEntry.richContent.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.galleryData.items.video.thumbnail.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.galleryData.options.item.ratio"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.imageData.image.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.mapData.mapSettings.lat"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.mapData.mapSettings.lng"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.pollData.poll.image.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.pollData.design.poll.background.image.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.appEmbedData.image.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.videoData.video.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.videoData.thumbnail.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.audioData.audio.duration"
          },
          {
            path: "questionEntries.questionEntry.richContent.nodes.audioData.coverImage.duration"
          },
          { path: "questionEntries.questionEntry.labels.sortOrder" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.faq.question_entry.v2.question_entry",
      method: "POST",
      methodFqn: "wix.faq.questionentry.v2.QuestionEntryService.BulkUpdateQuestionEntry",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqQuestionentryV2QuestionEntryServiceUrl({
        protoPath: "/v2/bulk/question-entries/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.questionEntry.createdDate" },
            { path: "results.questionEntry.updatedDate" },
            {
              path: "results.questionEntry.richContent.metadata.createdTimestamp"
            },
            {
              path: "results.questionEntry.richContent.metadata.updatedTimestamp"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.questionEntry.sortOrder" },
            {
              path: "results.questionEntry.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "results.questionEntry.richContent.nodes.imageData.image.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "results.questionEntry.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "results.questionEntry.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.videoData.video.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "results.questionEntry.richContent.nodes.audioData.coverImage.duration"
            },
            { path: "results.questionEntry.labels.sortOrder" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateQuestionEntry;
}

// src/faq-question-entry-v2-question-entry-question-entry.meta.ts
function createQuestionEntry2() {
  const payload = {};
  const getRequestOptions = createQuestionEntry(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/question-entries",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getQuestionEntry2() {
  const payload = { questionEntryId: ":questionEntryId" };
  const getRequestOptions = getQuestionEntry(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/question-entries/{questionEntryId}",
    pathParams: { questionEntryId: "questionEntryId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateQuestionEntry2() {
  const payload = { questionEntry: { id: ":questionEntryId" } };
  const getRequestOptions = updateQuestionEntry(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v2/question-entries/{questionEntry.id}",
    pathParams: { questionEntryId: "questionEntryId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteQuestionEntry2() {
  const payload = { questionEntryId: ":questionEntryId" };
  const getRequestOptions = deleteQuestionEntry(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v2/question-entries/{questionEntryId}",
    pathParams: { questionEntryId: "questionEntryId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listQuestionEntries2() {
  const payload = {};
  const getRequestOptions = listQuestionEntries(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/question-entries",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryQuestionEntries2() {
  const payload = {};
  const getRequestOptions = queryQuestionEntries(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/question-entries/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkDeleteQuestionEntries2() {
  const payload = {};
  const getRequestOptions = bulkDeleteQuestionEntries(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/bulk/question-entries/delete",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateExtendedFields2() {
  const payload = { id: ":id" };
  const getRequestOptions = updateExtendedFields(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/question-entries/{id}/update-extended-fields",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function setQuestionEntryLabels2() {
  const payload = { questionEntryId: ":questionEntryId" };
  const getRequestOptions = setQuestionEntryLabels(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v2/question-entries/{questionEntryId}/labels",
    pathParams: { questionEntryId: "questionEntryId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateQuestionEntry2() {
  const payload = {};
  const getRequestOptions = bulkUpdateQuestionEntry(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/bulk/question-entries/update",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  bulkDeleteQuestionEntries2 as bulkDeleteQuestionEntries,
  bulkUpdateQuestionEntry2 as bulkUpdateQuestionEntry,
  createQuestionEntry2 as createQuestionEntry,
  deleteQuestionEntry2 as deleteQuestionEntry,
  getQuestionEntry2 as getQuestionEntry,
  listQuestionEntries2 as listQuestionEntries,
  queryQuestionEntries2 as queryQuestionEntries,
  setQuestionEntryLabels2 as setQuestionEntryLabels,
  updateExtendedFields2 as updateExtendedFields,
  updateQuestionEntry2 as updateQuestionEntry
};
