// @wix/bex-core/dist/esm/services/tagsService/createTagErrors.js
export const createTagErrors = ({ translate: t, visualType = 'toast', retryFn, }) => {
    return {
        applicationError: {
            FQDN_NOT_FOUND: () => ({
                message: visualType === 'toast'
                    ? t('cairo.tags.fqdn-not-found.error')
                    : t('cairo.tags.fqdn-not-found.manageTagsModal.error'),
                action: {
                    text: t('cairo.error.contactSupport.cta'),
                    onClick: () => {
                        // TODO: contact support
                    },
                },
            }),
            TAG_NAME_ALREADY_EXISTS: () => ({
                message: t('cairo.tags.manageTags.modal.list.input.validation.exists'),
            }),
            TAGS_REACHED_LIMIT: () => ({
                message: getTagsReachedLimitErrorMessage({
                    visualType,
                    translate: t,
                }),
            }),
            CREATE_ID_EXHAUSTED_RETRY_ATTEMPTS: () => ({
                message: visualType === 'toast'
                    ? t('cairo.tags.collectionPage.toast.create.error')
                    : t('cairo.tags.manageTags.modal.toast.create.error'),
                action: {
                    text: t('cairo.error.tryAgain.cta'),
                    onClick: retryFn,
                },
            }),
            EXPOSURE_NOT_SUPPORTED: false,
        },
    };
};
export const getTagsReachedLimitErrorMessage = ({ visualType, translate: t, }) => visualType === 'toast'
    ? t('cairo.tags.popover.creationLimitReached.error')
    : t('cairo.tags.creationLimitReached.error');
