// @wix/bex-core/dist/esm/services/tagsService/tagsService.js
import { listTags, listTagsForAccount, createTag, createTagForAccount, updateTag, updateTagForAccount, deleteTag, deleteTagForAccount, } from '@wix/bex-utils/@wix/ambassador-os-tags-v1-tag/http';
export class TagsService {
    constructor({ container }) {
        this.container = container;
    }
    get httpClient() {
        return this.container.httpClient;
    }
    getContextFunction(accountFn, siteFn) {
        return this.container.contextType === 'ACCOUNT' ? accountFn : siteFn;
    }
    async listTags(request, errorCodesMap) {
        const fn = this.getContextFunction(listTagsForAccount, listTags);
        const { data: { tags = [] }, } = await this.container.errorHandler.withErrorHandler(() => this.httpClient.request(fn(request)), { errorCodesMap });
        return tags.filter(({ id, name }) => id && name);
    }
    async createTag(request, errorCodesMap) {
        // We cannot use getContextFunction because createTagForAccount and createTag have different return signatures due to error handling
        if (this.container.contextType === 'SITE') {
            const { data: { tag }, } = await this.container.errorHandler.withErrorHandler(() => this.httpClient.request(createTag(request)), { errorCodesMap });
            return tag;
        }
        else {
            const { data: { tag }, } = await this.container.errorHandler.withErrorHandler(() => this.httpClient.request(createTagForAccount(request)), { errorCodesMap });
            return tag;
        }
    }
    async updateTag(request) {
        const fn = this.getContextFunction(updateTagForAccount, updateTag);
        const { data: { tag }, } = await this.container.errorHandler.withErrorHandler(() => this.httpClient.request(fn(request)), { errorCodesMap: {} });
        return tag;
    }
    async deleteTag(request) {
        const fn = this.getContextFunction(deleteTagForAccount, deleteTag);
        return this.container.errorHandler.withErrorHandler(() => this.httpClient.request(fn(request)), { errorCodesMap: {} });
    }
}
