// @wix/patterns/dist/cjs/components/Fade/Fade.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.Fade = Fade;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _reactTransitionGroup = require("react-transition-group");
var _react = _interopRequireWildcard(require("react"));
var _FadeStCss = require("./Fade.st.css.js");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/Fade/Fade.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function Fade({
  children,
  timeout,
  ...rest
}) {
  const {
    type: Component,
    props: childProps
  } = children;
  const style = (0, _react.useMemo)(() => ({
    transitionDuration: `${timeout}ms`,
    ...childProps.style
  }), [timeout, childProps.style]);
  return /*#__PURE__*/_react.default.createElement(_reactTransitionGroup.Transition, (0, _extends2.default)({
    timeout: timeout
  }, rest, {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 21,
      columnNumber: 5
    }
  }), status => {
    const className = (0, _FadeStCss.st)(_FadeStCss.classes.root, {
      exit: status === 'exited' || status === 'exiting'
    }, childProps.className, `fade-${status}`);
    return /*#__PURE__*/_react.default.createElement(Component, (0, _extends2.default)({}, childProps, {
      className: className,
      style: style,
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 33,
        columnNumber: 11
      }
    }));
  });
}
