// @wix/bex-core/dist/esm/state/TagsState/TagsWidgetState.js
import { action, computed, makeObservable, observable } from 'mobx';
import { TagsManagementState, } from './TagsManagementState';
import { TagsBIReporter } from './TagsBIReporter';
import { TaskState } from '../TaskState';
import { WixPatternsError } from '../../services';
export class TagsWidgetState {
    constructor({ container, origin, ...tagsManagementStateParams }) {
        this.tagsFormState = null;
        this.initTask = new TaskState();
        this.retry = () => {
            this.tagsBIReporter.onTryAgain({
                actionName: 'load tags widget',
                loaction: 'Component error state',
            });
            this.initTask.runOnce();
        };
        this.container = container;
        this.origin = origin;
        this.tagsManagementState = new TagsManagementState({
            ...tagsManagementStateParams,
            container,
        });
        this.tagsBIReporter = new TagsBIReporter({
            collection: this.tagsManagementState.collection,
            reportBi: this.container.createBILogger({
                componentType: 'TagsWidget',
                ...tagsManagementStateParams.fqdn,
            }),
            location: 'Tags widget',
        });
        makeObservable(this, {
            init: action,
            tagsFormState: observable,
            isDirty: computed,
            retry: action.bound,
        });
    }
    init() {
        const { initTask } = this;
        const disposers = [this.tagsManagementState.init()];
        initTask.runOnce(async () => {
            this.tagsManagementState.initTask.runOnce();
            await this.tagsManagementState.initTask.status.promise?.catch((error) => {
                this.tagsBIReporter.onError({
                    actionName: 'Load tags widget',
                    origin: this.origin,
                }, error);
                this.container.internalMonitor.reportError({
                    error: new WixPatternsError({
                        errorCode: 'GetTagsFailed',
                        originalException: error,
                    }),
                    category: 'Tags',
                });
                throw error;
            });
        });
        return () => {
            disposers.forEach((disposer) => disposer?.());
        };
    }
    get isLoading() {
        return this.initTask.status.isLoading;
    }
    get isError() {
        return this.initTask.status.isError;
    }
    async validate() {
        const data = await this.tagsFormState?.validate();
        return data || { isValid: true };
    }
    get isDirty() {
        return this.tagsFormState?.isDirty || false;
    }
}
