// @wix/bex-core/dist/esm/state/TagsState/TagsBIReporter.js
import { cairoSearchForTags, cairoTagsAddDeleteTag, cairoTagsModalEndProcess, cairoTagsUnsuccessfulUpdateInServer, cairoTryAgainClicked, cairoUndoClicked, } from '@wix/bex-utils/@wix/bi-logger-os-data/v2';
import EventEmitter from 'events';
import { withoutIrrelevant } from '../../util';
import { isErrorDetails, isHttpError } from '../../util/isHttpError';
const ACTIONS = {
    'Add tag': 'Add',
    'Rename tag': 'Rename',
    'Delete tag': 'Delete',
};
const withoutDefaults = withoutIrrelevant();
export class TagsBIReporter {
    constructor(params) {
        this.events = new EventEmitter();
        this.actionsMade = {
            Add: false,
            Delete: false,
            Rename: false,
        };
        this.collection = params.collection;
        this.reportBi = params.reportBi;
        this.location = params.location;
    }
    set setActionsMade(actionName) {
        if (!(actionName in ACTIONS)) {
            throw Error(`Unknown action ${actionName} was invoked`);
        }
        const action = ACTIONS[actionName];
        this.actionsMade = {
            ...this.actionsMade,
            [action]: true,
        };
    }
    resetActionsMade() {
        this.actionsMade = {
            Add: false,
            Delete: false,
            Rename: false,
        };
    }
    _clearAllEvents() {
        const { events } = this;
        events.removeAllListeners('select');
        events.removeAllListeners('blur');
        events.removeAllListeners('add');
        events.removeAllListeners('clear');
    }
    _getErrorType(e) {
        let isNetworkError;
        const details = isHttpError(e) && isErrorDetails(e?.response?.data?.details)
            ? e?.response?.data?.details
            : null;
        if (details) {
            isNetworkError = details.applicationError?.code === 'ERR_NETWORK';
        }
        else {
            isNetworkError = e?.code === 'ERR_NETWORK';
        }
        return isNetworkError ? 'Network error' : 'Technical issue';
    }
    onSearch(searchValue) {
        const { events } = this;
        const afterSearchParams = {
            searchResultsCnt: this.collection.result.items.filter((item) => item.name.toLowerCase().includes(searchValue.toLowerCase())).length,
            searchTerm: searchValue,
        };
        const reportTagSelectedSearchBI = () => {
            this._clearAllEvents();
            this.reportBi(withoutDefaults(cairoSearchForTags)({
                ...afterSearchParams,
                location: this.location,
                endType: 'Tag selected',
            }));
        };
        const reportSearchDismissedSearchBI = () => {
            this._clearAllEvents();
            this.reportBi(withoutDefaults(cairoSearchForTags)({
                ...afterSearchParams,
                location: this.location,
                endType: 'Search dismissed',
            }));
        };
        const reportAddNewFromSearchBI = () => {
            this._clearAllEvents();
            this.reportBi(withoutDefaults(cairoSearchForTags)({
                ...afterSearchParams,
                location: this.location,
                endType: 'Add new tag',
            }));
        };
        this._clearAllEvents();
        events.once('select', reportTagSelectedSearchBI);
        events.once('blur', reportSearchDismissedSearchBI);
        events.once('clear', reportSearchDismissedSearchBI);
        events.once('add', reportAddNewFromSearchBI);
    }
    onAction(params) {
        if (params.actionName) {
            this.setActionsMade = params.actionName;
        }
        const afterActionParams = {
            numTags: this.collection.result.items.length,
        };
        this.reportBi(withoutDefaults(cairoTagsAddDeleteTag)({
            ...params,
            ...afterActionParams,
        }));
    }
    onError(params, error) {
        const errorParams = {
            numTags: this.collection.result.items.length,
            errorType: this._getErrorType(error),
        };
        this.reportBi(withoutDefaults(cairoTagsUnsuccessfulUpdateInServer)({
            ...errorParams,
            ...params,
        }));
    }
    onTryAgain(params) {
        const tryAgainParams = {
            loaction: 'Toast',
        };
        this.reportBi(withoutDefaults(cairoTryAgainClicked)({
            ...tryAgainParams,
            ...params,
        }));
    }
    onUndo(params) {
        this.reportBi(withoutDefaults(cairoUndoClicked)(params));
    }
    getActionsMadeChangesString() {
        const actionsMadeArray = [];
        this.actionsMade.Add && actionsMadeArray.push('Add');
        this.actionsMade.Delete && actionsMadeArray.push('Delete');
        this.actionsMade.Rename && actionsMadeArray.push('Rename');
        return actionsMadeArray.join('/');
    }
    onModalClose(params) {
        let endType = this.getActionsMadeChangesString();
        if (endType) {
            endType = `Close with changes(${endType})`;
        }
        else {
            endType = 'Close without changes';
        }
        this.reportBi(withoutDefaults(cairoTagsModalEndProcess)({
            ...params,
            numTags: this.collection.result.items.length,
            endType,
        }));
        this.resetActionsMade();
    }
}
