// @wix/bex-core/dist/esm/state/TagsState/TagsSearchState.js
import { TAG_MAX_LENGTH } from './TagsManagementState';
import { action, makeObservable, observable, computed } from 'mobx';
export class TagsSearchState {
    constructor({ tagsManagementState }) {
        this.searchValue = '';
        this.setSearchValue = (newSearchValue) => {
            this.searchValue = newSearchValue;
        };
        this.tagsManagementState = tagsManagementState;
        makeObservable(this, {
            searchValue: observable,
            setSearchValue: action,
            filteredItems: computed,
            hasSameTagAsSearch: computed,
        });
    }
    get filteredItems() {
        if (!this.searchValue) {
            return this.tagsManagementState.items;
        }
        const filteredTags = this.tagsManagementState.items.filter((tag) => {
            return tag.name.toLowerCase().includes(this.searchValue.toLowerCase());
        });
        if (filteredTags.length === 0 && this.searchValue.length > TAG_MAX_LENGTH) {
            const truncatedTag = this.tagsManagementState.items.find((tag) => tag.name.toLowerCase() ===
                this.searchValue.substring(0, TAG_MAX_LENGTH).toLowerCase());
            if (truncatedTag) {
                return [truncatedTag];
            }
        }
        return filteredTags;
    }
    get hasSameTagAsSearch() {
        return this.tagsManagementState.items.some((tag) => {
            return (tag.name.toLowerCase() ===
                this.searchValue.substring(0, TAG_MAX_LENGTH).toLowerCase());
        });
    }
}
