// @wix/unidriver-jsdom-react-legacy/dist/cjs/jsdomReactUniDriver.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.jsdomReactUniDriver = void 0;
const internal_1 = require("@wix/unidriver-core/internal");
const test_utils_1 = require("react-dom/test-utils");
const actCompat_1 = require("./actCompat");
const contextToWaitError_1 = require("./contextToWaitError");
const jsdomReactUniDriverList_1 = require("./jsdomReactUniDriverList");
const enterValue_1 = require("./utils/enterValue");
const waitForLegacy_1 = require("./waitForLegacy");
const errors_1 = require("./utils/errors");
const keyMeta_1 = require("./keyMeta");
const getSelectorPath_1 = require("./utils/getSelectorPath");
const jsdomReactUniDriver = (adapterParams, ctx = { selector: 'Root React driver' }) => {
    const getBaseElement = async () => {
        const container = typeof adapterParams === 'function' ? adapterParams() : adapterParams;
        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
        if (!container) {
            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access,@typescript-eslint/no-unnecessary-condition
            throw new errors_1.NoElementWithSelectorError({ selector: ctx.selector ?? '' });
        }
        return container;
    };
    const exists = async () => {
        try {
            await getBaseElement();
            return true;
        }
        catch (e) {
            if ((0, errors_1.isMultipleElementsWithSelectorError)(e)) {
                throw e;
            }
            else {
                return false;
            }
        }
    };
    const getElementBySelector = async (selector) => {
        const container = await getBaseElement();
        const elements = container.querySelectorAll(selector);
        if (!elements.length) {
            throw new errors_1.NoElementWithSelectorError({ selector });
        }
        else if (elements.length > 1) {
            throw new errors_1.MultipleElementsWithSelectorError(elements.length, selector);
        }
        return elements[0];
    };
    const getElementsBySelector = async (selector) => {
        const base = await getBaseElement();
        return Array.from(base.querySelectorAll(selector));
    };
    const getDefinitionForKeyType = (keyType) => {
        const definition = keyMeta_1.keyMeta[keyType];
        return {
            key: definition.key,
            keyCode: 'keyCode' in definition ? Number(definition.keyCode) : undefined,
        };
    };
    const api = {
        getType: () => {
            throw new Error(`Not implemented. Deprecated adapter - migrate to "@wix/unidriver-jsdom-react"`);
        },
        awaited: (timeout) => {
            return (0, exports.jsdomReactUniDriver)(async () => {
                try {
                    await (0, internal_1.eventually)(async () => {
                        if (!(await api.exists())) {
                            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access,@typescript-eslint/no-unnecessary-condition
                            throw new errors_1.NoElementWithSelectorError({ selector: ctx.selector ?? '' });
                        }
                    }, { timeout });
                    // eslint-disable-next-line @typescript-eslint/return-await
                    return getBaseElement();
                }
                catch (e) {
                    throw e.cause ?? e;
                }
            }, {
                parent: ctx.parent,
                selector: ctx.selector,
            });
        },
        // @ts-expect-error necessary type hack
        unwrap: () => getBaseElement(),
        prop: async (name) => {
            const base = await getBaseElement();
            return base[name];
        },
        s: (selector) => {
            return (0, exports.jsdomReactUniDriver)(async () => {
                try {
                    return await getElementBySelector(selector);
                }
                catch (e) {
                    if ((0, errors_1.isNoElementWithSelectorError)(e)) {
                        throw new errors_1.NoElementWithSelectorError({
                            selector,
                            failedPath: e.failedPath,
                            fullPath: [...(await (0, getSelectorPath_1.getSelectorsPath)(ctx)), selector],
                        });
                    }
                    throw e;
                }
            }, {
                parent: ctx,
                selector,
            });
        },
        ss: (selector) => (0, jsdomReactUniDriverList_1.jsdomReactUniDriverList)(async () => {
            const e = await getBaseElement();
            return Array.from(e.querySelectorAll(selector));
        }, { parent: ctx, selector }),
        $: (selector) => {
            return api.s(internal_1.unidriverConfig.queryByAttribute(selector));
        },
        $$: (selector) => {
            return api.ss(internal_1.unidriverConfig.queryByAttribute(selector));
        },
        text: async () => getBaseElement().then((e) => {
            return e.textContent ?? '';
        }),
        value: async () => {
            const base = (await getBaseElement());
            const value = base.value;
            return typeof value === 'number' ? String(value) : (value ?? '');
        },
        click: async (options) => {
            const elementIsFocusable = (el) => {
                return (el.tagName === 'INPUT' ||
                    el.tagName === 'SELECT' ||
                    el.tagName === 'TEXTAREA' ||
                    el.tagName === 'BUTTON' ||
                    el.tagName === 'svg');
            };
            const el = await getBaseElement();
            const eventData = { button: 0 }; // 0 - Main Button (Left)
            // setting button 0 is now needed in React 16+ as it's not set by react anymore
            // 15 - https://github.com/facebook/react/blob/v15.6.1/src/renderers/dom/client/syntheticEvents/SyntheticMouseEvent.js#L45
            // 16 - https://github.com/facebook/react/blob/master/packages/react-dom/src/events/SyntheticMouseEvent.js#L33
            await (0, actCompat_1.act)(async () => {
                test_utils_1.Simulate.mouseDown(el, eventData);
            });
            if (elementIsFocusable(el)) {
                if (document.activeElement !== el) {
                    if (document.activeElement) {
                        await (0, actCompat_1.act)(async () => {
                            test_utils_1.Simulate.blur(document.activeElement);
                        });
                    }
                    if (!el.disabled) {
                        await (0, actCompat_1.act)(async () => {
                            el.focus();
                        });
                        await (0, actCompat_1.act)(async () => {
                            test_utils_1.Simulate.focus(el);
                        });
                    }
                }
            }
            await (0, actCompat_1.act)(async () => {
                test_utils_1.Simulate.mouseUp(el, eventData);
            });
            await (0, actCompat_1.act)(async () => {
                if (options?.type === 'double') {
                    test_utils_1.Simulate.doubleClick(el, eventData);
                }
                else {
                    test_utils_1.Simulate.click(el, eventData);
                }
            });
            const isCheckable = () => {
                return (el.tagName === 'INPUT' &&
                    (el.type === 'checkbox' || el.type === 'radio'));
            };
            if (isCheckable()) {
                // copying value, because original input el get mutated and this value changes
                const invertedCheck = !el.checked;
                await (0, actCompat_1.act)(async () => {
                    test_utils_1.Simulate.input(el, {
                        target: {
                            checked: invertedCheck,
                        },
                    });
                });
                await (0, actCompat_1.act)(async () => {
                    test_utils_1.Simulate.change(el, {
                        target: {
                            checked: invertedCheck,
                        },
                    });
                });
            }
        },
        mouse: {
            hover: async () => {
                const el = await getBaseElement();
                await (0, actCompat_1.act)(async () => {
                    test_utils_1.Simulate.mouseOver(el);
                });
                await (0, actCompat_1.act)(async () => {
                    test_utils_1.Simulate.mouseEnter(el);
                });
            },
            press: async () => {
                const el = await getBaseElement();
                await (0, actCompat_1.act)(async () => {
                    test_utils_1.Simulate.mouseDown(el);
                });
            },
            release: async () => {
                const el = await getBaseElement();
                await (0, actCompat_1.act)(async () => {
                    test_utils_1.Simulate.mouseUp(el);
                });
            },
            moveTo: async (to) => {
                const el = await getBaseElement();
                const { left, top } = (await to.unwrap()).getBoundingClientRect();
                await (0, actCompat_1.act)(async () => {
                    test_utils_1.Simulate.mouseMove(el, { clientX: left, clientY: top });
                });
            },
            leave: async () => {
                const el = await getBaseElement();
                await (0, actCompat_1.act)(async () => {
                    test_utils_1.Simulate.mouseLeave(el);
                });
            },
        },
        pressKey: async (key) => {
            const el = await getBaseElement();
            const def = getDefinitionForKeyType(key);
            await (0, actCompat_1.act)(async () => {
                test_utils_1.Simulate.keyDown(el, def);
            });
            await (0, actCompat_1.act)(async () => {
                test_utils_1.Simulate.keyUp(el, def);
            });
        },
        hasClass: async (className) => (await getBaseElement()).classList.contains(className),
        enterValue: async (value, options) => {
            const base = (await getBaseElement());
            return (0, enterValue_1.enterValue)(base, value, options);
        },
        enterText: async (value, options) => {
            return api.enterValue(value, {
                ...options,
                shouldClear: options?.clear,
            });
        },
        attr: async (name) => {
            const el = await getBaseElement();
            return el.getAttribute(name);
        },
        exists,
        get: async (selector, options) => {
            await (0, waitForLegacy_1.waitForLegacy)(async () => {
                try {
                    const element = await getElementBySelector(selector);
                    return !!element;
                }
                catch {
                    return false;
                }
            }, {
                timeout: options?.timeout,
            });
            return (0, exports.jsdomReactUniDriver)(() => getElementBySelector(selector), {
                parent: ctx,
                selector,
            });
        },
        getAll: async (selector, options) => {
            await (0, waitForLegacy_1.waitForLegacy)(async () => {
                const elements = await getElementsBySelector(selector);
                return !!elements.length;
            }, {
                timeout: options?.timeout,
            });
            return (0, jsdomReactUniDriverList_1.jsdomReactUniDriverList)(() => getElementsBySelector(selector), {
                parent: ctx,
                selector,
            });
        },
        hover: async () => {
            return api.mouse.hover();
        },
        wait: async (timeout) => {
            return (0, waitForLegacy_1.waitForLegacy)(exists, {
                timeout,
                message: (0, contextToWaitError_1.contextToWaitError)(ctx),
            });
        },
        type: 'react',
        // eslint-disable-next-line @typescript-eslint/no-unsafe-return
        getNative: () => getBaseElement(),
        _prop: async (name) => {
            // eslint-disable-next-line @typescript-eslint/no-unsafe-return
            return api.prop(name);
        },
    };
    return api;
};
exports.jsdomReactUniDriver = jsdomReactUniDriver;
