// @wix/patterns/dist/cjs/components/CustomColumnsPanel/CustomColumnsPanelEntry.js
"use strict";

exports.__esModule = true;
exports.CustomColumnsPanelEntry = void 0;
var _react = _interopRequireWildcard(require("react"));
var _ToolbarCollectionContext = require("../ToolbarCollectionContext");
var _CustomColumnsPanel = require("./CustomColumnsPanel");
var _state = require("../../state");
var _minimalRequiredWdsRuntimeCheck = require("../../minimalRequiredWdsRuntimeCheck");
var _react2 = require("@wix/bex-core/react");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CustomColumnsPanel/CustomColumnsPanelEntry.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _CustomColumnsPanelEntry() {
  const toolbar = (0, _ToolbarCollectionContext.useToolbarCollectionContext)();
  const {
    modalsContainerRef
  } = (0, _react2.usePageContext)();
  if (process.env.NODE_ENV !== 'production') {
    (0, _minimalRequiredWdsRuntimeCheck.minimalRequiredWdsRuntimeCheck)({
      required: '^1.68.0',
      requiredBy: 'WixPatterns/CustomColumns'
    });
  }
  const [state] = (0, _react.useState)(() => toolbar.customColumnsState ?? (toolbar.customColumnsState = new _state.CustomColumnsState({
    table: toolbar,
    container: toolbar.container,
    dragAndDrop: CustomColumnsPanelEntry.CustomColumnsDragAndDrop,
    modalsContainerRef
  })));
  return /*#__PURE__*/_react.default.createElement(_CustomColumnsPanel.CustomColumnsPanel, {
    state: state,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 30,
      columnNumber: 10
    }
  });
}
const CustomColumnsPanelEntry = exports.CustomColumnsPanelEntry = _CustomColumnsPanelEntry;
