// @wix/patterns/dist/cjs/components/CustomColumnsPanel/CustomColumnsPanel.js
"use strict";

exports.__esModule = true;
exports.CustomColumnsPanel = void 0;
exports._CustomColumnsPanel = _CustomColumnsPanel;
var _designSystem = require("@wix/design-system");
var _react = _interopRequireWildcard(require("react"));
var _bexCore = require("@wix/bex-core");
var _mobxReactLite = require("mobx-react-lite");
var _react2 = require("@wix/bex-core/react");
var _ColumnsList = require("./ColumnsList");
var _useRafRender = require("../../hooks/useRafRender");
var _bi = require("@wix/bex-core/bi");
var _ScrollableContent = require("../ScrollableContent");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CustomColumnsPanel/CustomColumnsPanel.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _CustomColumnsPanel({
  state
}) {
  var _collectionDataExtens;
  const container = (0, _react2.useWixPatternsContainer)();
  const {
    translate: t
  } = container;
  const entered = (0, _useRafRender.useRafRender)();
  const {
    table,
    dragAndDrop
  } = state;
  const {
    sidePanel,
    collectionDataExtension
  } = table;
  (0, _react.useEffect)(() => state.init(), []);
  if (!entered) {
    return null;
  }
  return /*#__PURE__*/_react.default.createElement(_designSystem.SidePanel, {
    dataHook: "custom-columns-panel",
    width: "100%",
    onCloseButtonClick: () => {
      table.reportBi((0, _bexCore.withoutDefaults)(_bi.cairoFiltersPanelUsed)({
        isOpened: false,
        feature: 'Custom Columns',
        origin: 'X Button'
      }));
      sidePanel.close();
    },
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 34,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.SidePanel.Header, {
    title: t('cairo.table.custom-columns.title'),
    subtitle: dragAndDrop ? t('cairo.table.custom-columns.dragAndDrop.subtitle') : t('cairo.table.custom-columns.subtitle'),
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 48,
      columnNumber: 7
    }
  }), /*#__PURE__*/_react.default.createElement(_ScrollableContent.ScrollableContent, {
    overflowY: true,
    dataHook: "custom-columns-panel-scroll-container",
    ref: e => {
      if (state.dragAndDrop) {
        state.dragAndDrop.listRect.element = e;
      }
    },
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 56,
      columnNumber: 7
    }
  }, /*#__PURE__*/_react.default.createElement(_ColumnsList.ColumnsList, {
    state: state,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 65,
      columnNumber: 9
    }
  })), (collectionDataExtension == null || (_collectionDataExtens = collectionDataExtension.DataExtension) == null ? void 0 : _collectionDataExtens.CustomColumnsAddCustomFieldCTAButton) && /*#__PURE__*/_react.default.createElement(collectionDataExtension.DataExtension.CustomColumnsAddCustomFieldCTAButton, {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 69,
      columnNumber: 9
    }
  }));
}
const CustomColumnsPanel = exports.CustomColumnsPanel = (0, _mobxReactLite.observer)(_CustomColumnsPanel);
