// pro-gallery-lib/dist/cjs/common/constants/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const viewMode_js_1 = tslib_1.__importDefault(require("./viewMode.js"));
const urlTypes_js_1 = require("./urlTypes.js");
const socialNetworks_js_1 = tslib_1.__importDefault(require("./socialNetworks.js"));
const resizeMethods_js_1 = tslib_1.__importDefault(require("./resizeMethods.js"));
const deviceType_js_1 = tslib_1.__importDefault(require("./deviceType.js"));
const events_js_1 = tslib_1.__importDefault(require("./events.js"));
const layoutParams_crop_method_js_1 = tslib_1.__importDefault(require("./layoutParams_crop_method.js"));
const layoutParams_crop_alignment_js_1 = tslib_1.__importDefault(require("./layoutParams_crop_alignment.js"));
const layoutParams_structure_galleryLayout_js_1 = tslib_1.__importStar(require("./layoutParams_structure_galleryLayout.js"));
const layoutParams_structure_groupsOrder_js_1 = tslib_1.__importDefault(require("./layoutParams_structure_groupsOrder.js"));
const layoutParams_structure_layoutOrientation_js_1 = tslib_1.__importDefault(require("./layoutParams_structure_layoutOrientation.js"));
const layoutParams_structure_responsiveMode_js_1 = tslib_1.__importDefault(require("./layoutParams_structure_responsiveMode.js"));
const layoutParams_structure_scrollDirection_js_1 = tslib_1.__importDefault(require("./layoutParams_structure_scrollDirection.js"));
const layoutParams_groups_allowedGroupTypes_js_1 = tslib_1.__importDefault(require("./layoutParams_groups_allowedGroupTypes.js"));
const layoutParams_thumbnails_alignment_js_1 = tslib_1.__importDefault(require("./layoutParams_thumbnails_alignment.js"));
const layoutParams_navigationArrows_verticalAlignment_js_1 = tslib_1.__importDefault(require("./layoutParams_navigationArrows_verticalAlignment.js"));
const layoutParams_navigationArrows_position_js_1 = tslib_1.__importDefault(require("./layoutParams_navigationArrows_position.js"));
const layoutParams_navigationArrows_type_js_1 = tslib_1.__importDefault(require("./layoutParams_navigationArrows_type.js"));
const layoutParams_navigationArrows_container_type_js_1 = tslib_1.__importDefault(require("./layoutParams_navigationArrows_container_type.js"));
const layoutParams_info_layout_js_1 = tslib_1.__importDefault(require("./layoutParams_info_layout.js"));
const layoutParams_info_sizeUnits_js_1 = tslib_1.__importDefault(require("./layoutParams_info_sizeUnits.js"));
const layoutParams_info_placement_js_1 = tslib_1.__importStar(require("./layoutParams_info_placement.js"));
const layoutParams_targetItemSize_unit_js_1 = tslib_1.__importDefault(require("./layoutParams_targetItemSize_unit.js"));
const behaviourParams_item_clickAction_js_1 = tslib_1.__importDefault(require("./behaviourParams_item_clickAction.js"));
const playTrigger_js_1 = tslib_1.__importDefault(require("./playTrigger.js"));
const behaviourParams_item_overlay_hoveringBehaviour_js_1 = tslib_1.__importDefault(require("./behaviourParams_item_overlay_hoveringBehaviour.js"));
const behaviourParams_item_overlay_hoverAnimation_js_1 = tslib_1.__importDefault(require("./behaviourParams_item_overlay_hoverAnimation.js"));
const behaviourParams_item_overlay_position_js_1 = tslib_1.__importDefault(require("./behaviourParams_item_overlay_position.js"));
const behaviourParams_item_overlay_sizeUnits_js_1 = tslib_1.__importDefault(require("./behaviourParams_item_overlay_sizeUnits.js"));
const behaviourParams_item_content_hoverAnimation_js_1 = tslib_1.__importDefault(require("./behaviourParams_item_content_hoverAnimation.js"));
const behaviourParams_item_content_placementAnimation_js_1 = tslib_1.__importDefault(require("./behaviourParams_item_content_placementAnimation.js"));
const behaviourParams_item_content_loader_js_1 = tslib_1.__importDefault(require("./behaviourParams_item_content_loader.js"));
const behaviourParams_item_secondaryMedia_trigger_js_1 = tslib_1.__importDefault(require("./behaviourParams_item_secondaryMedia_trigger.js"));
const behaviourParams_item_secondaryMedia_behaviour_js_1 = tslib_1.__importDefault(require("./behaviourParams_item_secondaryMedia_behaviour.js"));
const behaviourParams_gallery_layoutDirection_js_1 = tslib_1.__importDefault(require("./behaviourParams_gallery_layoutDirection.js"));
const behaviourParams_gallery_scrollAnimation_js_1 = tslib_1.__importDefault(require("./behaviourParams_gallery_scrollAnimation.js"));
const behaviourParams_gallery_vertical_loadMore_amount_js_1 = tslib_1.__importDefault(require("./behaviourParams_gallery_vertical_loadMore_amount.js"));
const behaviourParams_gallery_horizontal_slideAnimation_js_1 = tslib_1.__importDefault(require("./behaviourParams_gallery_horizontal_slideAnimation.js"));
const behaviourParams_gallery_horizontal_slideTransition_js_1 = tslib_1.__importDefault(require("./behaviourParams_gallery_horizontal_slideTransition.js"));
const behaviourParams_gallery_horizontal_autoSlide_behaviour_js_1 = tslib_1.__importDefault(require("./behaviourParams_gallery_horizontal_autoSlide_behaviour.js"));
const behaviourParams_gallery_horizontal_slideshowInfo_buttonsAlignment_js_1 = tslib_1.__importDefault(require("./behaviourParams_gallery_horizontal_slideshowInfo_buttonsAlignment.js"));
const stylingParams_itemResolutionMode_js_1 = tslib_1.__importDefault(require("./stylingParams_itemResolutionMode.js"));
const layoutParams_thumbnails_position_js_1 = tslib_1.__importDefault(require("./layoutParams_thumbnails_position.js"));
const behaviourParams_item_threeDimensionalScene_controls_js_1 = require("./behaviourParams_item_threeDimensionalScene_controls.js");
//NEW STYPEPARAMS METHOD
exports.default = {
    events: events_js_1.default,
    deviceType: deviceType_js_1.default,
    isLayout: layoutParams_structure_galleryLayout_js_1.isLayout,
    hasExternalAbovePlacement: layoutParams_info_placement_js_1.hasExternalAbovePlacement,
    hasExternalBelowPlacement: layoutParams_info_placement_js_1.hasExternalBelowPlacement,
    hasHoverPlacement: layoutParams_info_placement_js_1.hasHoverPlacement,
    hasExternalRightPlacement: layoutParams_info_placement_js_1.hasExternalRightPlacement,
    hasExternalLeftPlacement: layoutParams_info_placement_js_1.hasExternalLeftPlacement,
    hasExternalVerticalPlacement: layoutParams_info_placement_js_1.hasExternalVerticalPlacement,
    hasExternalHorizontalPlacement: layoutParams_info_placement_js_1.hasExternalHorizontalPlacement,
    isExternalAbovePlacement: layoutParams_info_placement_js_1.isExternalAbovePlacement,
    isExternalBelowPlacement: layoutParams_info_placement_js_1.isExternalBelowPlacement,
    isHoverPlacement: layoutParams_info_placement_js_1.isHoverPlacement,
    isExternalRightPlacement: layoutParams_info_placement_js_1.isExternalRightPlacement,
    isExternalLeftPlacement: layoutParams_info_placement_js_1.isExternalLeftPlacement,
    isExternalVerticalPlacement: layoutParams_info_placement_js_1.isExternalVerticalPlacement,
    isExternalHorizontalPlacement: layoutParams_info_placement_js_1.isExternalHorizontalPlacement,
    isConstantVerticalPlacement: layoutParams_info_placement_js_1.isConstantVerticalPlacement,
    resizeMethods: resizeMethods_js_1.default,
    socialNetworks: socialNetworks_js_1.default,
    urlSizes: urlTypes_js_1.URL_SIZES,
    urlTypes: urlTypes_js_1.URL_TYPES,
    viewMode: viewMode_js_1.default,
    layoutParams_crop_method: layoutParams_crop_method_js_1.default,
    layoutParams_crop_alignment: layoutParams_crop_alignment_js_1.default,
    layoutParams_structure_galleryLayout: layoutParams_structure_galleryLayout_js_1.default,
    layoutParams_structure_groupsOrder: layoutParams_structure_groupsOrder_js_1.default,
    layoutParams_structure_layoutOrientation: layoutParams_structure_layoutOrientation_js_1.default,
    layoutParams_structure_responsiveMode: layoutParams_structure_responsiveMode_js_1.default,
    layoutParams_structure_scrollDirection: layoutParams_structure_scrollDirection_js_1.default,
    layoutParams_groups_allowedGroupTypes: layoutParams_groups_allowedGroupTypes_js_1.default,
    layoutParams_thumbnails_alignment: layoutParams_thumbnails_alignment_js_1.default,
    layoutParams_navigationArrows_verticalAlignment: layoutParams_navigationArrows_verticalAlignment_js_1.default,
    layoutParams_navigationArrows_position: layoutParams_navigationArrows_position_js_1.default,
    layoutParams_navigationArrows_type: layoutParams_navigationArrows_type_js_1.default,
    layoutParams_navigationArrows_container_type: layoutParams_navigationArrows_container_type_js_1.default,
    layoutParams_info_layout: layoutParams_info_layout_js_1.default,
    layoutParams_info_sizeUnits: layoutParams_info_sizeUnits_js_1.default,
    layoutParams_info_placement: layoutParams_info_placement_js_1.default,
    layoutParams_targetItemSize_unit: layoutParams_targetItemSize_unit_js_1.default,
    behaviourParams_item_clickAction: behaviourParams_item_clickAction_js_1.default,
    behaviourParams_item_video_playTrigger: playTrigger_js_1.default,
    behaviourParams_item_overlay_hoveringBehaviour: behaviourParams_item_overlay_hoveringBehaviour_js_1.default,
    behaviourParams_item_overlay_hoverAnimation: behaviourParams_item_overlay_hoverAnimation_js_1.default,
    behaviourParams_item_overlay_position: behaviourParams_item_overlay_position_js_1.default,
    behaviourParams_item_overlay_sizeUnits: behaviourParams_item_overlay_sizeUnits_js_1.default,
    behaviourParams_item_content_hoverAnimation: behaviourParams_item_content_hoverAnimation_js_1.default,
    behaviourParams_item_content_placementAnimation: behaviourParams_item_content_placementAnimation_js_1.default,
    behaviourParams_item_content_loader: behaviourParams_item_content_loader_js_1.default,
    behaviourParams_item_secondaryMedia_trigger: behaviourParams_item_secondaryMedia_trigger_js_1.default,
    behaviourParams_item_secondaryMedia_behaviour: behaviourParams_item_secondaryMedia_behaviour_js_1.default,
    behaviourParams_gallery_layoutDirection: behaviourParams_gallery_layoutDirection_js_1.default,
    behaviourParams_gallery_scrollAnimation: behaviourParams_gallery_scrollAnimation_js_1.default,
    behaviourParams_gallery_vertical_loadMore_amount: behaviourParams_gallery_vertical_loadMore_amount_js_1.default,
    behaviourParams_gallery_horizontal_slideAnimation: behaviourParams_gallery_horizontal_slideAnimation_js_1.default,
    behaviourParams_gallery_horizontal_slideTransition: behaviourParams_gallery_horizontal_slideTransition_js_1.default,
    behaviourParams_gallery_horizontal_autoSlide_behaviour: behaviourParams_gallery_horizontal_autoSlide_behaviour_js_1.default,
    behaviourParams_gallery_horizontal_slideshowInfo_buttonsAlignment: behaviourParams_gallery_horizontal_slideshowInfo_buttonsAlignment_js_1.default,
    behaviourParams_item_threeDimensionalScene_playTrigger: playTrigger_js_1.default,
    parse3DDimensions: behaviourParams_item_threeDimensionalScene_controls_js_1.parse3DDimensions,
    stylingParams_itemResolutionMode: stylingParams_itemResolutionMode_js_1.default,
    layoutParams_thumbnails_position: layoutParams_thumbnails_position_js_1.default,
    //NEW STYPEPARAMS METHOD
};
