// pro-gallery-lib/dist/cjs/common/constants/layoutParams_info_placement.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isConstantVerticalPlacement = exports.isExternalLeftPlacement = exports.isExternalRightPlacement = exports.isHoverPlacement = exports.isExternalAboveOrBelowSinglePlacement = exports.isExternalBelowPlacement = exports.isExternalAbovePlacement = exports.isExternalHorizontalPlacement = exports.isExternalVerticalPlacement = exports.hasExternalHorizontalPlacement = exports.hasExternalVerticalPlacement = exports.hasExternalLeftPlacement = exports.hasExternalRightPlacement = exports.hasExternalBelowPlacement = exports.hasExternalAbovePlacement = exports.hasHoverPlacement = void 0;
const PLACEMENTS = {
    OVERLAY: 'OVERLAY',
    ABOVE: 'ABOVE',
    BELOW: 'BELOW',
    LEFT: 'LEFT',
    RIGHT: 'RIGHT',
    ALTERNATE_HORIZONTALLY: 'ALTERNATE_HORIZONTALLY',
    ALTERNATE_VERTICALLY: 'ALTERNATE_VERTICALLY',
};
const hasHoverPlacement = (placement) => String(placement).includes(PLACEMENTS.OVERLAY);
exports.hasHoverPlacement = hasHoverPlacement;
const hasExternalAbovePlacement = (placement, idx) => String(placement).includes(PLACEMENTS.ABOVE) ||
    (idx % 2 === 0 && String(placement).includes(PLACEMENTS.ALTERNATE_VERTICALLY));
exports.hasExternalAbovePlacement = hasExternalAbovePlacement;
const hasExternalBelowPlacement = (placement, idx) => String(placement).includes(PLACEMENTS.BELOW) ||
    (idx % 2 === 1 && String(placement).includes(PLACEMENTS.ALTERNATE_VERTICALLY));
exports.hasExternalBelowPlacement = hasExternalBelowPlacement;
const hasExternalRightPlacement = (placement, idx) => String(placement).includes(PLACEMENTS.RIGHT) ||
    (idx % 2 === 0 && String(placement).includes(PLACEMENTS.ALTERNATE_HORIZONTALLY));
exports.hasExternalRightPlacement = hasExternalRightPlacement;
const hasExternalLeftPlacement = (placement, idx) => String(placement).includes(PLACEMENTS.LEFT) ||
    (idx % 2 === 1 && String(placement).includes(PLACEMENTS.ALTERNATE_HORIZONTALLY));
exports.hasExternalLeftPlacement = hasExternalLeftPlacement;
const hasExternalVerticalPlacement = (placement) => (0, exports.hasExternalAbovePlacement)(placement, 0) || (0, exports.hasExternalBelowPlacement)(placement, 1);
exports.hasExternalVerticalPlacement = hasExternalVerticalPlacement;
const hasExternalHorizontalPlacement = (placement) => (0, exports.hasExternalRightPlacement)(placement, 0) || (0, exports.hasExternalLeftPlacement)(placement, 1);
exports.hasExternalHorizontalPlacement = hasExternalHorizontalPlacement;
const isExternalVerticalPlacement = (placement) => (0, exports.hasExternalVerticalPlacement)(placement) &&
    !(0, exports.hasExternalHorizontalPlacement)(placement) &&
    !(0, exports.hasHoverPlacement)(placement);
exports.isExternalVerticalPlacement = isExternalVerticalPlacement;
const isExternalHorizontalPlacement = (placement) => (0, exports.hasExternalHorizontalPlacement)(placement) &&
    !(0, exports.hasExternalVerticalPlacement)(placement) &&
    !(0, exports.hasHoverPlacement)(placement);
exports.isExternalHorizontalPlacement = isExternalHorizontalPlacement;
const isExternalAbovePlacement = (placement) => String(placement) === PLACEMENTS.ABOVE;
exports.isExternalAbovePlacement = isExternalAbovePlacement;
const isExternalBelowPlacement = (placement) => String(placement) === PLACEMENTS.BELOW;
exports.isExternalBelowPlacement = isExternalBelowPlacement;
const isExternalAboveOrBelowSinglePlacement = (placement) => (0, exports.isExternalAbovePlacement)(placement) || (0, exports.isExternalBelowPlacement)(placement);
exports.isExternalAboveOrBelowSinglePlacement = isExternalAboveOrBelowSinglePlacement;
const isHoverPlacement = (placement) => String(placement) === PLACEMENTS.OVERLAY;
exports.isHoverPlacement = isHoverPlacement;
const isExternalRightPlacement = (placement) => String(placement) === PLACEMENTS.RIGHT;
exports.isExternalRightPlacement = isExternalRightPlacement;
const isExternalLeftPlacement = (placement) => String(placement) === PLACEMENTS.LEFT;
exports.isExternalLeftPlacement = isExternalLeftPlacement;
const isConstantVerticalPlacement = (placement) => {
    const placementArray = placement.split(',');
    const firstPlacement = placementArray.shift();
    return ((0, exports.isExternalAboveOrBelowSinglePlacement)(firstPlacement) &&
        placementArray.every((placement) => placement === firstPlacement));
};
exports.isConstantVerticalPlacement = isConstantVerticalPlacement;
exports.default = PLACEMENTS;
