// @wix/auto_sdk_events_policies/build/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  EventStatus: () => EventStatus,
  LocationType: () => LocationType,
  SortOrder: () => SortOrder,
  Status: () => Status,
  SubdivisionType: () => SubdivisionType,
  WebhookIdentityType: () => WebhookIdentityType,
  createPolicy: () => createPolicy4,
  deletePolicy: () => deletePolicy4,
  getPolicy: () => getPolicy4,
  onPolicyCreated: () => onPolicyCreated2,
  onPolicyDeleted: () => onPolicyDeleted2,
  onPolicyUpdated: () => onPolicyUpdated2,
  queryPolicies: () => queryPolicies4,
  reorderEventPolicies: () => reorderEventPolicies4,
  updatePolicy: () => updatePolicy4
});
module.exports = __toCommonJS(index_exports);

// src/events-v2-policy-policies.public.ts
var import_rename_all_nested_keys2 = require("@wix/sdk-runtime/rename-all-nested-keys");
var import_timestamp3 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths3 = require("@wix/sdk-runtime/transformations/transform-paths");
var import_sdk_types = require("@wix/sdk-types");

// src/events-v2-policy-policies.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/events-v2-policy-policies.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEventsV2PolicyManagementUrl(opts) {
  const domainToMappings = {
    _: [
      {
        srcPath: "/_api/events-policies",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/events-policies",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/events-policies",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/events-policies",
        destPath: ""
      },
      {
        srcPath: "/events/v2/policies",
        destPath: "/v2/policies"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/events-policies",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/events-policies",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/events-policies",
        destPath: ""
      }
    ],
    "events.wixapps.net": [
      {
        srcPath: "/_api/events-policies",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_events_policies";
function createPolicy(payload) {
  function __createPolicy({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "policy.createdDate" }, { path: "policy.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v2.policy",
      method: "POST",
      methodFqn: "wix.events.v2.PolicyManagement.CreatePolicy",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsV2PolicyManagementUrl({
        protoPath: "/v2/policies",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "policy.createdDate" },
            { path: "policy.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createPolicy;
}
function updatePolicy(payload) {
  function __updatePolicy({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "policy.createdDate" }, { path: "policy.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v2.policy",
      method: "PATCH",
      methodFqn: "wix.events.v2.PolicyManagement.UpdatePolicy",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsV2PolicyManagementUrl({
        protoPath: "/v2/policies/{policy.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "policy.createdDate" },
            { path: "policy.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updatePolicy;
}
function deletePolicy(payload) {
  function __deletePolicy({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v2.policy",
      method: "DELETE",
      methodFqn: "wix.events.v2.PolicyManagement.DeletePolicy",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsV2PolicyManagementUrl({
        protoPath: "/v2/policies/{policyId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deletePolicy;
}
function queryPolicies(payload) {
  function __queryPolicies({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v2.policy",
      method: "POST",
      methodFqn: "wix.events.v2.PolicyManagement.QueryPolicies",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsV2PolicyManagementUrl({
        protoPath: "/v2/policies/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "policies.createdDate" },
            { path: "policies.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryPolicies;
}
function reorderEventPolicies(payload) {
  function __reorderEventPolicies({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v2.policy",
      method: "POST",
      methodFqn: "wix.events.v2.PolicyManagement.ReorderEventPolicies",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsV2PolicyManagementUrl({
        protoPath: "/v2/policies/reorder",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "policies.createdDate" },
            { path: "policies.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __reorderEventPolicies;
}
function getPolicy(payload) {
  function __getPolicy({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v2.policy",
      method: "GET",
      methodFqn: "wix.events.v2.PolicyManagement.GetPolicy",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsV2PolicyManagementUrl({
        protoPath: "/v2/policies/{policyId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "policy.createdDate" },
            { path: "policy.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getPolicy;
}

// src/events-v2-policy-policies.universal.ts
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var LocationType = /* @__PURE__ */ ((LocationType2) => {
  LocationType2["VENUE"] = "VENUE";
  LocationType2["ONLINE"] = "ONLINE";
  return LocationType2;
})(LocationType || {});
var SubdivisionType = /* @__PURE__ */ ((SubdivisionType2) => {
  SubdivisionType2["UNKNOWN_SUBDIVISION_TYPE"] = "UNKNOWN_SUBDIVISION_TYPE";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_1"] = "ADMINISTRATIVE_AREA_LEVEL_1";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_2"] = "ADMINISTRATIVE_AREA_LEVEL_2";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_3"] = "ADMINISTRATIVE_AREA_LEVEL_3";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_4"] = "ADMINISTRATIVE_AREA_LEVEL_4";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_5"] = "ADMINISTRATIVE_AREA_LEVEL_5";
  SubdivisionType2["COUNTRY"] = "COUNTRY";
  return SubdivisionType2;
})(SubdivisionType || {});
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["ONE_TIME"] = "ONE_TIME";
  Status2["RECURRING"] = "RECURRING";
  Status2["RECURRING_NEXT"] = "RECURRING_NEXT";
  Status2["RECURRING_LAST_ENDED"] = "RECURRING_LAST_ENDED";
  Status2["RECURRING_LAST_CANCELED"] = "RECURRING_LAST_CANCELED";
  return Status2;
})(Status || {});
var EventStatus = /* @__PURE__ */ ((EventStatus2) => {
  EventStatus2["SCHEDULED"] = "SCHEDULED";
  EventStatus2["STARTED"] = "STARTED";
  EventStatus2["ENDED"] = "ENDED";
  EventStatus2["CANCELED"] = "CANCELED";
  return EventStatus2;
})(EventStatus || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createPolicy2(policy) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ policy });
  const reqOpts = createPolicy(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.policy;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { policy: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["policy"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updatePolicy2(_id, policy) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    policy: { ...policy, id: _id }
  });
  const reqOpts = updatePolicy(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.policy;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { policy: "$[1]" },
        explicitPathsToArguments: { "policy.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "policy"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deletePolicy2(policyId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ policyId });
  const reqOpts = deletePolicy(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { policyId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["policyId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryPolicies2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryPolicies(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [])
      );
      return {
        items: transformedData?.policies,
        pagingMetadata: transformedData?.metadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function reorderEventPolicies2(policyId, eventId, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    policyId,
    eventId,
    beforePolicyId: options?.beforePolicyId,
    afterPolicyId: options?.afterPolicyId
  });
  const reqOpts = reorderEventPolicies(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          policyId: "$[0]",
          eventId: "$[1]",
          beforePolicyId: "$[2].beforePolicyId",
          afterPolicyId: "$[2].afterPolicyId"
        },
        singleArgumentUnchanged: false
      },
      ["policyId", "eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getPolicy2(policyId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ policyId });
  const reqOpts = getPolicy(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.policy;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { policyId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["policyId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/events-v2-policy-policies.public.ts
function createPolicy3(httpClient) {
  return (policy) => createPolicy2(
    policy,
    // @ts-ignore
    { httpClient }
  );
}
function updatePolicy3(httpClient) {
  return (_id, policy) => updatePolicy2(
    _id,
    policy,
    // @ts-ignore
    { httpClient }
  );
}
function deletePolicy3(httpClient) {
  return (policyId) => deletePolicy2(
    policyId,
    // @ts-ignore
    { httpClient }
  );
}
function queryPolicies3(httpClient) {
  return () => queryPolicies2(
    // @ts-ignore
    { httpClient }
  );
}
function reorderEventPolicies3(httpClient) {
  return (policyId, eventId, options) => reorderEventPolicies2(
    policyId,
    eventId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function getPolicy3(httpClient) {
  return (policyId) => getPolicy2(
    policyId,
    // @ts-ignore
    { httpClient }
  );
}
var onPolicyCreated = (0, import_sdk_types.EventDefinition)(
  "wix.events.v2.policy_created",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onPolicyDeleted = (0, import_sdk_types.EventDefinition)(
  "wix.events.v2.policy_deleted",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onPolicyUpdated = (0, import_sdk_types.EventDefinition)(
  "wix.events.v2.policy_updated",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();

// src/events-v2-policy-policies.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var import_event_definition_modules = require("@wix/sdk-runtime/event-definition-modules");
var createPolicy4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(createPolicy3);
var updatePolicy4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(updatePolicy3);
var deletePolicy4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(deletePolicy3);
var queryPolicies4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(queryPolicies3);
var reorderEventPolicies4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(reorderEventPolicies3);
var getPolicy4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getPolicy3);
var onPolicyCreated2 = (0, import_event_definition_modules.createEventModule)(onPolicyCreated);
var onPolicyDeleted2 = (0, import_event_definition_modules.createEventModule)(onPolicyDeleted);
var onPolicyUpdated2 = (0, import_event_definition_modules.createEventModule)(onPolicyUpdated);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  EventStatus,
  LocationType,
  SortOrder,
  Status,
  SubdivisionType,
  WebhookIdentityType,
  createPolicy,
  deletePolicy,
  getPolicy,
  onPolicyCreated,
  onPolicyDeleted,
  onPolicyUpdated,
  queryPolicies,
  reorderEventPolicies,
  updatePolicy
});
