// @wix/patterns/dist/cjs/components/Grid/DebugItemKeyRecycler.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.DebugItemKeyRecycler = DebugItemKeyRecycler;
var _react = _interopRequireDefault(require("react"));
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/Grid/DebugItemKeyRecycler.tsx";
// this helps rendering the item key inside the card instead of inspecting in in react-dev-tools
function DebugItemKeyRecycler({
  children
}) {
  return /*#__PURE__*/_react.default.createElement("div", {
    style: {
      position: 'absolute',
      border: 'solid 1px #fff',
      top: 6,
      left: 6,
      padding: 6,
      backgroundColor: '#000',
      zIndex: 2000
    },
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 6,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement("span", {
    style: {
      color: '#fff'
    },
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 17,
      columnNumber: 7
    }
  }, children));
}
