// @wix/auto_sdk_bookings_attendance/build/es/index.typings.mjs
// src/bookings-v2-attendance-attendance.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/bookings-v2-attendance-attendance.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressBookingsAttendanceV2AttendanceServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/bookings-attendance",
        destPath: ""
      },
      {
        srcPath: "/bookings-attendance",
        destPath: ""
      },
      {
        srcPath: "/_api/bookings-attendance/v2/bulk/attendance",
        destPath: "/v2/bulk/attendance"
      },
      {
        srcPath: "/bookings-attendance/v2/bulk/attendance",
        destPath: "/v2/bulk/attendance"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/bookings-attendance",
        destPath: "/v2/attendance"
      },
      {
        srcPath: "/bookings/attendance/v2/bulk/attendance",
        destPath: "/v2/bulk/attendance"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/bookings/bookings-attendance",
        destPath: "/v2/attendance"
      },
      {
        srcPath: "/_api/bookings-attendance",
        destPath: "/v2/attendance"
      },
      {
        srcPath: "/bookings/attendance",
        destPath: "/v2/attendance"
      },
      {
        srcPath: "/bookings/attendance/v2/attendance",
        destPath: "/v2/attendance"
      },
      {
        srcPath: "/bookings/v2/attendance",
        destPath: "/v2/attendance"
      },
      {
        srcPath: "/bookings/v2/bulk/attendance",
        destPath: "/v2/bulk/attendance"
      },
      {
        srcPath: "/bookings-attendance/v2/bulk/attendance",
        destPath: "/v2/bulk/attendance"
      },
      {
        srcPath: "/_api/bookings-attendance/v2/bulk/attendance",
        destPath: "/v2/bulk/attendance"
      },
      {
        srcPath: "/bookings/attendance/v2/bulk/attendance",
        destPath: "/v2/bulk/attendance"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/bookings-attendance",
        destPath: "/v2/attendance"
      },
      {
        srcPath: "/_api/bookings-attendance/v2/bulk/attendance",
        destPath: "/v2/bulk/attendance"
      }
    ],
    _: [
      {
        srcPath: "/bookings-attendance",
        destPath: "/v2/attendance"
      },
      {
        srcPath: "/_api/bookings-attendance/v2/bulk/attendance",
        destPath: "/v2/bulk/attendance"
      },
      {
        srcPath: "/bookings/attendance/v2/bulk/attendance",
        destPath: "/v2/bulk/attendance"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/bookings-attendance",
        destPath: "/v2/attendance"
      },
      {
        srcPath: "/api/bookings-attendance/v2/bulk/attendance",
        destPath: "/v2/bulk/attendance"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_bookings_attendance";
function getAttendance(payload) {
  function __getAttendance({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.v2.attendance",
      method: "GET",
      methodFqn: "com.wixpress.bookings.attendance.v2.AttendanceService.GetAttendance",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressBookingsAttendanceV2AttendanceServiceUrl({
        protoPath: "/v2/attendance/{attendanceId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "attendance.updatedDate" },
            { path: "attendance.statusUpdatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getAttendance;
}
function setAttendance(payload) {
  function __setAttendance({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "attendance.updatedDate" },
          { path: "attendance.statusUpdatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.bookings.v2.attendance",
      method: "POST",
      methodFqn: "com.wixpress.bookings.attendance.v2.AttendanceService.SetAttendance",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressBookingsAttendanceV2AttendanceServiceUrl({
        protoPath: "/v2/attendance/set",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "attendance.updatedDate" },
            { path: "attendance.statusUpdatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __setAttendance;
}
function bulkSetAttendance(payload) {
  function __bulkSetAttendance({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "attendanceList.updatedDate" },
          { path: "attendanceList.statusUpdatedDate" },
          { path: "attendanceDetails.attendance.updatedDate" },
          { path: "attendanceDetails.attendance.statusUpdatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.bookings.v2.attendance",
      method: "POST",
      methodFqn: "com.wixpress.bookings.attendance.v2.AttendanceService.BulkSetAttendance",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressBookingsAttendanceV2AttendanceServiceUrl({
        protoPath: "/v2/bulk/attendance/set",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.updatedDate" },
            { path: "results.item.statusUpdatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkSetAttendance;
}
function queryAttendance(payload) {
  function __queryAttendance({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.v2.attendance",
      method: "POST",
      methodFqn: "com.wixpress.bookings.attendance.v2.AttendanceService.QueryAttendance",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressBookingsAttendanceV2AttendanceServiceUrl({
        protoPath: "/v2/attendance/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "attendances.updatedDate" },
            { path: "attendances.statusUpdatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryAttendance;
}
function countAttendances(payload) {
  function __countAttendances({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.v2.attendance",
      method: "POST",
      methodFqn: "com.wixpress.bookings.attendance.v2.AttendanceService.CountAttendances",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressBookingsAttendanceV2AttendanceServiceUrl({
        protoPath: "/v2/attendance/count",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __countAttendances;
}

// src/bookings-v2-attendance-attendance.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var AttendanceStatus = /* @__PURE__ */ ((AttendanceStatus2) => {
  AttendanceStatus2["NOT_SET"] = "NOT_SET";
  AttendanceStatus2["ATTENDED"] = "ATTENDED";
  AttendanceStatus2["NOT_ATTENDED"] = "NOT_ATTENDED";
  return AttendanceStatus2;
})(AttendanceStatus || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function getAttendance2(attendanceId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    attendanceId
  });
  const reqOpts = getAttendance(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.attendance;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { attendanceId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["attendanceId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function setAttendance2(attendance, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    attendance,
    participantNotification: options?.participantNotification
  });
  const reqOpts = setAttendance(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          attendance: "$[0]",
          participantNotification: "$[1].participantNotification"
        },
        singleArgumentUnchanged: false
      },
      ["attendance", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkSetAttendance2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    returnFullEntity: options?.returnFullEntity,
    attendanceDetails: options?.attendanceDetails
  });
  const reqOpts = bulkSetAttendance(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          returnFullEntity: "$[0].returnFullEntity",
          attendanceDetails: "$[0].attendanceDetails"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryAttendance2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryAttendance(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [])
      );
      return {
        items: transformedData?.attendances,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function countAttendances2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    filter: options?.filter
  });
  const reqOpts = countAttendances(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { filter: "$[0].filter" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
export {
  AttendanceStatus,
  SortOrder,
  WebhookIdentityType,
  bulkSetAttendance2 as bulkSetAttendance,
  countAttendances2 as countAttendances,
  getAttendance2 as getAttendance,
  queryAttendance2 as queryAttendance,
  setAttendance2 as setAttendance
};
