// @wix/headless-seo/cjs/dist/services/seo-tags-service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SEOTagsService = exports.SEOTagsServiceDefinition = void 0;
exports.loadSEOTagsServiceConfig = loadSEOTagsServiceConfig;
const services_definitions_1 = require("@wix/services-definitions");
const signals_1 = require("@wix/services-definitions/core-services/signals");
const auto_sdk_seo_seo_tags_1 = require("@wix/auto_sdk_seo_seo-tags");
exports.SEOTagsServiceDefinition = (0, services_definitions_1.defineService)('seoTagsService');
exports.SEOTagsService = services_definitions_1.implementService.withConfig()(exports.SEOTagsServiceDefinition, ({ getService, config }) => {
    const signalsService = getService(signals_1.SignalsServiceDefinition);
    const { tags: initialTags } = config;
    const tags = signalsService.signal(initialTags);
    const updateSeoTags = async (itemType = auto_sdk_seo_seo_tags_1.ItemType.UNKNOWN_ITEM_TYPE, itemData = {
        pageName: '',
        seoData: { tags: [] },
    }) => {
        const pageURL = window.location.href;
        const { tags: updatedTags } = await loadSEOTagsServiceConfig({
            pageUrl: pageURL,
            itemType,
            itemData,
        });
        tags.set(updatedTags);
        appendNewTags(updatedTags);
    };
    return { seoTags: tags, updateSeoTags };
});
async function resolveDynamicSeoTags(pageUrl, itemType, itemData) {
    const res = await (0, auto_sdk_seo_seo_tags_1.resolveItemSeoTags)({
        pageUrl,
        itemType,
        slug: itemData.slug,
    });
    return res.seoTags?.tags || [];
}
async function resolveStaticSeoTags(pageUrl, itemData) {
    const { pageName, seoData } = itemData;
    const res = await (0, auto_sdk_seo_seo_tags_1.resolveStaticPageSeoTags)({
        pageUrl,
        pageName,
        seoData,
    });
    return res.seoTags?.tags || [];
}
async function loadSEOTagsServiceConfig(params) {
    const { itemType, pageUrl, itemData } = params;
    const isStaticPage = !itemType || itemType === auto_sdk_seo_seo_tags_1.ItemType.UNKNOWN_ITEM_TYPE;
    let tags = [];
    try {
        if (isStaticPage) {
            tags = await resolveStaticSeoTags(pageUrl, itemData);
        }
        else {
            tags = await resolveDynamicSeoTags(pageUrl, itemType, itemData);
        }
    }
    catch (error) {
        console.error('Error loading SEO tags service config', error);
    }
    return {
        tags,
    };
}
function appendNewTags(tags) {
    if (typeof window === 'undefined')
        return;
    const newTagElements = [];
    try {
        tags.forEach((tag) => {
            const el = createTagElement(tag);
            if (el)
                newTagElements.push(el);
        });
        document.head
            .querySelectorAll('[wix-seo-tags="true"]')
            .forEach((el) => el.remove());
        newTagElements.forEach((el) => document.head.appendChild(el));
    }
    catch (err) {
        console.error('SEO tag update failed', err);
    }
    function createTagElement(tag) {
        let el = null;
        if (tag.type === 'title') {
            el = document.createElement('title');
            el.textContent = tag.children || '';
        }
        else if (tag.type === 'meta') {
            el = document.createElement('meta');
            setAttributes(el, tag.props);
        }
        else if (tag.type === 'link') {
            el = document.createElement('link');
            setAttributes(el, tag.props);
        }
        else if (tag.type === 'script') {
            el = document.createElement('script');
            setAttributes(el, tag.props);
            setAttributes(el, tag.meta);
            if (tag.children)
                el.textContent = tag.children;
        }
        if (el)
            el.setAttribute('wix-seo-tags', 'true');
        return el;
    }
    function setAttributes(el, attrs) {
        if (!attrs)
            return;
        Object.entries(attrs).forEach(([k, v]) => {
            if (v !== undefined)
                el.setAttribute(k, v);
        });
    }
}
