// fp-ts-esm/es6/IOOption.js
import { apFirst as apFirst_, apS as apS_, apSecond as apSecond_ } from './Apply.js';
import * as chainable from './Chain.js';
import { compact as compact_, separate as separate_ } from './Compactable.js';
import { filter as filter_, filterMap as filterMap_, partition as partition_, partitionMap as partitionMap_ } from './Filterable.js';
import { fromEitherK as fromEitherK_, tapEither as tapEither_ } from './FromEither.js';
import { fromIOK as fromIOK_, tapIO as tapIO_ } from './FromIO.js';
import { dual, flow, identity, pipe, SK } from './function.js';
import { as as as_, asUnit as asUnit_, bindTo as bindTo_, flap as flap_, let as let__ } from './Functor.js';
import * as _ from './internal.js';
import * as I from './IO.js';
import * as O from './Option.js';
import * as OT from './OptionT.js';
import { guard as guard_ } from './Zero.js';
// -------------------------------------------------------------------------------------
// constructors
// -------------------------------------------------------------------------------------
/**
 * @category constructors
 * @since 2.12.0
 */
export const some = /*#__PURE__*/ OT.some(I.Pointed);
/**
 * @category lifting
 * @since 2.12.0
 */
export const fromPredicate = /*#__PURE__*/ OT.fromPredicate(I.Pointed);
// -------------------------------------------------------------------------------------
// conversions
// -------------------------------------------------------------------------------------
/**
 * @category conversions
 * @since 2.12.0
 */
export const fromOption = I.of;
/**
 * @category conversions
 * @since 2.12.0
 */
export const fromEither = /*#__PURE__*/ OT.fromEither(I.Pointed);
/**
 * @category conversions
 * @since 2.12.0
 */
export const fromIO = /*#__PURE__*/ OT.fromF(I.Functor);
/**
 * @category conversions
 * @since 2.12.0
 */
export const fromIOEither = /*#__PURE__*/ I.map(O.fromEither);
/**
 * @category pattern matching
 * @since 2.12.0
 */
export const match = /*#__PURE__*/ OT.match(I.Functor);
/**
 * Less strict version of [`match`](#match).
 *
 * The `W` suffix (short for **W**idening) means that the handler return types will be merged.
 *
 * @category pattern matching
 * @since 2.12.0
 */
export const matchW = match;
/**
 * The `E` suffix (short for **E**ffect) means that the handlers return an effect (`IO`).
 *
 * @category pattern matching
 * @since 2.12.0
 */
export const matchE = 
/*#__PURE__*/ OT.matchE(I.Chain);
/**
 * Alias of [`matchE`](#matche).
 *
 * @category pattern matching
 * @since 2.12.0
 */
export const fold = matchE;
/**
 * Less strict version of [`matchE`](#matche).
 *
 * The `W` suffix (short for **W**idening) means that the handler return types will be merged.
 *
 * @category pattern matching
 * @since 2.12.0
 */
export const matchEW = matchE;
/**
 * @category error handling
 * @since 2.12.0
 */
export const getOrElse = /*#__PURE__*/ OT.getOrElse(I.Monad);
/**
 * Less strict version of [`getOrElse`](#getorelse).
 *
 * The `W` suffix (short for **W**idening) means that the handler return type will be merged.
 *
 * @category error handling
 * @since 2.12.0
 */
export const getOrElseW = getOrElse;
/**
 * @category conversions
 * @since 2.12.0
 */
export const toUndefined = I.map(O.toUndefined);
/**
 * @category conversions
 * @since 2.12.0
 */
export const toNullable = I.map(O.toNullable);
/**
 * @category conversions
 * @since 2.12.0
 */
export const fromNullable = /*#__PURE__*/ OT.fromNullable(I.Pointed);
/**
 * @category lifting
 * @since 2.12.0
 */
export const fromNullableK = /*#__PURE__*/ OT.fromNullableK(I.Pointed);
/**
 * Alias of `flatMapNullable`.
 *
 * @category legacy
 * @since 2.12.0
 */
export const chainNullableK = /*#__PURE__*/ OT.chainNullableK(I.Monad);
// -------------------------------------------------------------------------------------
// combinators
// -------------------------------------------------------------------------------------
/**
 * @category lifting
 * @since 2.12.0
 */
export const fromOptionK = 
/*#__PURE__*/ OT.fromOptionK(I.Pointed);
/**
 * `map` can be used to turn functions `(a: A) => B` into functions `(fa: F<A>) => F<B>` whose argument and return types
 * use the type constructor `F` to represent some computational context.
 *
 * @category mapping
 * @since 2.12.0
 */
export const map = /*#__PURE__*/ OT.map(I.Functor);
/**
 * @since 2.12.0
 */
export const ap = /*#__PURE__*/ OT.ap(I.Apply);
/**
 * @category constructors
 * @since 2.12.0
 */
export const of = some;
/**
 * @category sequencing
 * @since 2.14.0
 */
export const flatMap = /*#__PURE__*/ dual(2, OT.flatMap(I.Monad));
/**
 * @category sequencing
 * @since 2.12.0
 */
export const flatten = /*#__PURE__*/ flatMap(identity);
/**
 * @category error handling
 * @since 2.12.0
 */
export const alt = /*#__PURE__*/ OT.alt(I.Monad);
/**
 * Less strict version of [`alt`](#alt).
 *
 * The `W` suffix (short for **W**idening) means that the return types will be merged.
 *
 * @category error handling
 * @since 2.12.0
 */
export const altW = alt;
/**
 * @since 2.12.0
 */
export const zero = /*#__PURE__*/ OT.zero(I.Pointed);
/**
 * @category constructors
 * @since 2.12.0
 */
export const none = /*#__PURE__*/ zero();
/**
 * @category filtering
 * @since 2.12.0
 */
export const compact = /*#__PURE__*/ compact_(I.Functor, O.Compactable);
/**
 * @category filtering
 * @since 2.12.0
 */
export const separate = /*#__PURE__*/ separate_(I.Functor, O.Compactable, O.Functor);
/**
 * @category filtering
 * @since 2.12.0
 */
export const filter = /*#__PURE__*/ filter_(I.Functor, O.Filterable);
/**
 * @category filtering
 * @since 2.12.0
 */
export const filterMap = /*#__PURE__*/ filterMap_(I.Functor, O.Filterable);
/**
 * @category filtering
 * @since 2.12.0
 */
export const partition = /*#__PURE__*/ partition_(I.Functor, O.Filterable);
/**
 * @category filtering
 * @since 2.12.0
 */
export const partitionMap = /*#__PURE__*/ partitionMap_(I.Functor, O.Filterable);
// -------------------------------------------------------------------------------------
// instances
// -------------------------------------------------------------------------------------
const _map = (fa, f) => pipe(fa, map(f));
/* istanbul ignore next */
const _ap = (fab, fa) => pipe(fab, ap(fa));
/* istanbul ignore next */
const _alt = (fa, that) => pipe(fa, alt(that));
/* istanbul ignore next */
const _filter = (fa, predicate) => pipe(fa, filter(predicate));
/* istanbul ignore next */
const _filterMap = (fa, f) => pipe(fa, filterMap(f));
/* istanbul ignore next */
const _partition = (fa, predicate) => pipe(fa, partition(predicate));
/* istanbul ignore next */
const _partitionMap = (fa, f) => pipe(fa, partitionMap(f));
/**
 * @category type lambdas
 * @since 2.12.0
 */
export const URI = 'IOOption';
/**
 * @category instances
 * @since 2.12.0
 */
export const Functor = {
    URI,
    map: _map
};
/**
 * Maps the `Some` value of this `IOOption` to the specified constant value.
 *
 * @category mapping
 * @since 2.16.0
 */
export const as = dual(2, as_(Functor));
/**
 * Maps the `Some` value of this `IOOption` to the void constant value.
 *
 * @category mapping
 * @since 2.16.0
 */
export const asUnit = asUnit_(Functor);
/**
 * @category mapping
 * @since 2.12.0
 */
export const flap = /*#__PURE__*/ flap_(Functor);
/**
 * @category instances
 * @since 2.12.0
 */
export const Pointed = {
    URI,
    of
};
/**
 * @category instances
 * @since 2.12.0
 */
export const Apply = {
    URI,
    map: _map,
    ap: _ap
};
/**
 * Combine two effectful actions, keeping only the result of the first.
 *
 * @since 2.12.0
 */
export const apFirst = /*#__PURE__*/ apFirst_(Apply);
/**
 * Combine two effectful actions, keeping only the result of the second.
 *
 * @since 2.12.0
 */
export const apSecond = /*#__PURE__*/ apSecond_(Apply);
/**
 * @category instances
 * @since 2.12.0
 */
export const Applicative = {
    URI,
    map: _map,
    ap: _ap,
    of
};
/**
 * @category instances
 * @since 2.12.0
 */
export const Chain = {
    URI,
    map: _map,
    ap: _ap,
    chain: flatMap
};
/**
 * @category instances
 * @since 2.12.0
 */
export const FromEither = {
    URI,
    fromEither
};
/**
 * @category instances
 * @since 2.12.0
 */
export const FromIO = {
    URI,
    fromIO
};
/**
 * Composes computations in sequence, using the return value of one computation to determine the next computation and
 * keeping only the result of the first.
 *
 * @category combinators
 * @since 2.15.0
 */
export const tap = /*#__PURE__*/ dual(2, chainable.tap(Chain));
/**
 * Composes computations in sequence, using the return value of one computation to determine the next computation and
 * keeping only the result of the first.
 *
 * @example
 * import { pipe } from 'fp-ts/function'
 * import * as IOO from 'fp-ts/IOOption'
 * import * as O from 'fp-ts/Option'
 * import * as E from 'fp-ts/Either'
 *
 * const compute = (value: number) => pipe(
 *   IOO.of(value),
 *   IOO.tapEither((value) => value > 0 ? E.right('ok') : E.left('error')),
 * )
 *
 * assert.deepStrictEqual(compute(1)(), O.of(1))
 * assert.deepStrictEqual(compute(-1)(), O.none)
 *
 * @category combinators
 * @since 2.16.0
 */
export const tapEither = /*#__PURE__*/ dual(2, tapEither_(FromEither, Chain));
/**
 * Composes computations in sequence, using the return value of one computation to determine the next computation and
 * keeping only the result of the first.
 *
 * @example
 * import { pipe } from 'fp-ts/function'
 * import * as IOO from 'fp-ts/IOOption'
 * import * as O from 'fp-ts/Option'
 * import * as Console from 'fp-ts/Console'
 *
 * // Will produce `Hello, fp-ts` to the stdout
 * const effectA = pipe(
 *   IOO.of('fp-ts'),
 *   IOO.tapIO((value) => Console.log(`Hello, ${value}`)),
 * )
 *
 * // No output to the stdout
 * const effectB = pipe(
 *   IOO.none as IOO.IOOption<string>,
 *   IOO.tapIO((value) => Console.log(`Hello, ${value}`)),
 * )
 *
 * async function test() {
 *   assert.deepStrictEqual(effectA(), O.of('fp-ts'))
 *   assert.deepStrictEqual(effectB(), O.none)
 * }
 *
 * test()
 *
 * @category combinators
 * @since 2.16.0
 */
export const tapIO = /*#__PURE__*/ dual(2, tapIO_(FromIO, Chain));
/**
 * @category instances
 * @since 2.12.0
 */
export const Alt = {
    URI,
    map: _map,
    alt: _alt
};
/**
 * @category instances
 * @since 2.12.0
 */
export const Zero = {
    URI,
    zero
};
/**
 * @category do notation
 * @since 2.12.0
 */
export const guard = /*#__PURE__*/ guard_(Zero, Pointed);
/**
 * @category instances
 * @since 2.12.0
 */
export const Alternative = {
    URI,
    map: _map,
    ap: _ap,
    of,
    alt: _alt,
    zero
};
/**
 * @category instances
 * @since 2.12.0
 */
export const Monad = {
    URI,
    map: _map,
    ap: _ap,
    of,
    chain: flatMap
};
/**
 * @category instances
 * @since 2.12.0
 */
export const MonadIO = {
    URI,
    map: _map,
    ap: _ap,
    of,
    chain: flatMap,
    fromIO
};
/**
 * @category instances
 * @since 2.12.0
 */
export const Compactable = {
    URI,
    compact,
    separate
};
/**
 * @category instances
 * @since 2.12.0
 */
export const Filterable = {
    URI,
    map: _map,
    compact,
    separate,
    filter: _filter,
    filterMap: _filterMap,
    partition: _partition,
    partitionMap: _partitionMap
};
/** @internal */
const _FlatMap = {
    flatMap
};
/** @internal */
const _FromIO = {
    fromIO: FromIO.fromIO
};
/** @internal */
const _FromEither = {
    fromEither
};
/**
 * @category sequencing
 * @since 2.16.0
 */
export const flatMapIO = _.flatMapIO(_FromIO, _FlatMap);
/**
 * @category sequencing
 * @since 2.16.0
 */
export const flatMapOption = /*#__PURE__*/ dual(2, (self, f) => flatMap(self, fromOptionK(f)));
/**
 * @category sequencing
 * @since 2.16.0
 */
export const flatMapEither = /*#__PURE__*/ _.flatMapEither(_FromEither, _FlatMap);
/**
 * @category sequencing
 * @since 2.16.0
 */
export const flatMapNullable = /*#__PURE__*/ dual(2, (self, f) => flatMap(self, fromNullableK(f)));
/**
 * @category lifting
 * @since 2.12.0
 */
export const fromIOK = 
/*#__PURE__*/ fromIOK_(FromIO);
/**
 * Alias of `flatMapIO`.
 *
 * @category legacy
 * @since 2.12.0
 */
export const chainIOK = flatMapIO;
/**
 * Alias of `tapIO`.
 *
 * @category legacy
 * @since 2.12.0
 */
export const chainFirstIOK = tapIO;
/**
 * @category lifting
 * @since 2.12.0
 */
export const fromEitherK = /*#__PURE__*/ fromEitherK_(FromEither);
/**
 * Alias of `flatMapEither`.
 *
 * @category legacy
 * @since 2.12.0
 */
export const chainEitherK = flatMapEither;
/**
 * Alias of `tapEither`.
 *
 * @category legacy
 * @since 2.12.0
 */
export const chainFirstEitherK = tapEither;
/**
 * Alias of `flatMapOption`.
 *
 * @category legacy
 * @since 2.12.0
 */
export const chainOptionK = flatMapOption;
// -------------------------------------------------------------------------------------
// do notation
// -------------------------------------------------------------------------------------
/**
 * @category do notation
 * @since 2.12.0
 */
export const Do = /*#__PURE__*/ of(_.emptyRecord);
/**
 * @category do notation
 * @since 2.12.0
 */
export const bindTo = /*#__PURE__*/ bindTo_(Functor);
const let_ = /*#__PURE__*/ let__(Functor);
export { 
/**
 * @category do notation
 * @since 2.13.0
 */
let_ as let };
/**
 * @category do notation
 * @since 2.12.0
 */
export const bind = /*#__PURE__*/ chainable.bind(Chain);
/**
 * @category do notation
 * @since 2.12.0
 */
export const apS = /*#__PURE__*/ apS_(Apply);
/**
 * @since 2.12.0
 */
export const ApT = /*#__PURE__*/ of(_.emptyReadonlyArray);
// -------------------------------------------------------------------------------------
// array utils
// -------------------------------------------------------------------------------------
/**
 * Equivalent to `ReadonlyNonEmptyArray#traverseWithIndex(Applicative)`.
 *
 * @category traversing
 * @since 2.12.0
 */
export const traverseReadonlyNonEmptyArrayWithIndex = (f) => flow(I.traverseReadonlyNonEmptyArrayWithIndex(f), I.map(O.traverseReadonlyNonEmptyArrayWithIndex(SK)));
/**
 * Equivalent to `ReadonlyArray#traverseWithIndex(Applicative)`.
 *
 * @category traversing
 * @since 2.12.0
 */
export const traverseReadonlyArrayWithIndex = (f) => {
    const g = traverseReadonlyNonEmptyArrayWithIndex(f);
    return (as) => (_.isNonEmpty(as) ? g(as) : ApT);
};
// -------------------------------------------------------------------------------------
// legacy
// -------------------------------------------------------------------------------------
/**
 * Alias of `flatMap`.
 *
 * @category legacy
 * @since 2.12.0
 */
export const chain = flatMap;
/**
 * Alias of `tap`.
 *
 * @category legacy
 * @since 2.12.0
 */
export const chainFirst = tap;
