// @wix/auto_sdk_payments_site-payment-method-types/build/internal/es/index.typings.mjs
// src/payments-site-payment-method-types-v1-site-payment-method-type-site-payment-method-types.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/payments-site-payment-method-types-v1-site-payment-method-type-site-payment-method-types.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixPaymentsSitePaymentMethodTypesV1SitePaymentMethodTypeServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/payment-method-types",
        destPath: ""
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      },
      {
        srcPath: "/_api/payments/payment-method-types",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/payments/site-payment-method-types",
        destPath: ""
      },
      {
        srcPath: "/_api/payments/payment-method-types",
        destPath: ""
      }
    ],
    "cashier._base_domain_": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "cashier.wixapps.net": [
      {
        srcPath: "/_api/payments/payment-method-types",
        destPath: ""
      },
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "cashier-services._base_domain_": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "events._base_domain_": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "events.wixapps.net": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "invoices._base_domain_": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "premium._base_domain_": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "pricing-plans.wixapps.net": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "vod-server._base_domain_": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "www.wixhotels.com": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/payments/payment-method-types",
        destPath: ""
      }
    ],
    "tpa.wixapps.net": [
      {
        srcPath: "/_api/payments/payment-method-types",
        destPath: ""
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/payments/payment-method-types",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/payments/payment-method-types",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/payments/payment-method-types",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/payments/payment-method-types",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/payments/payment-method-types",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/payments/site-payment-method-types/v1/site-payment-method-types",
        destPath: "/v1/site-payment-method-types"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/payments/site-payment-method-types/v1/site-payment-method-types",
        destPath: "/v1/site-payment-method-types"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_site-payment-method-types";
function getSitePaymentMethodType(payload) {
  function __getSitePaymentMethodType({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.site_payment_method_types.v1.site_payment_method_type",
      method: "GET",
      methodFqn: "wix.payments.site_payment_method_types.v1.SitePaymentMethodTypeService.GetSitePaymentMethodType",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsSitePaymentMethodTypesV1SitePaymentMethodTypeServiceUrl(
        {
          protoPath: "/v1/site-payment-method-types/{sitePaymentMethodTypeId}",
          data: payload,
          host
        }
      ),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getSitePaymentMethodType;
}
function listSitePaymentMethodTypes(payload) {
  function __listSitePaymentMethodTypes({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.site_payment_method_types.v1.site_payment_method_type",
      method: "GET",
      methodFqn: "wix.payments.site_payment_method_types.v1.SitePaymentMethodTypeService.ListSitePaymentMethodTypes",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsSitePaymentMethodTypesV1SitePaymentMethodTypeServiceUrl(
        { protoPath: "/v1/site-payment-method-types", data: payload, host }
      ),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __listSitePaymentMethodTypes;
}

// src/payments-site-payment-method-types-v1-site-payment-method-type-site-payment-method-types.universal.ts
import { transformRESTImageToSDKImage } from "@wix/sdk-runtime/transformations/image";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
var Format = /* @__PURE__ */ ((Format2) => {
  Format2["UNKNOWN_FORMAT"] = "UNKNOWN_FORMAT";
  Format2["SVG"] = "SVG";
  Format2["PNG"] = "PNG";
  return Format2;
})(Format || {});
var RequestedFields = /* @__PURE__ */ ((RequestedFields2) => {
  RequestedFields2["UNKNOWN_REQUESTED_FIELD"] = "UNKNOWN_REQUESTED_FIELD";
  RequestedFields2["ALLOWED_CURRENCIES"] = "ALLOWED_CURRENCIES";
  return RequestedFields2;
})(RequestedFields || {});
async function getSitePaymentMethodType2(sitePaymentMethodTypeId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    sitePaymentMethodTypeId,
    languageCode: options?.languageCode,
    countryCode: options?.countryCode,
    currencyCode: options?.currencyCode,
    requestedFields: options?.requestedFields
  });
  const reqOpts = getSitePaymentMethodType(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths(result.data, [
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [
            { path: "sitePaymentMethodType.merchantFlow.icons.image" },
            { path: "sitePaymentMethodType.buyerFlow.icons.image" }
          ]
        }
      ])
    )?.sitePaymentMethodType;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          sitePaymentMethodTypeId: "$[0]",
          languageCode: "$[1].languageCode",
          countryCode: "$[1].countryCode",
          currencyCode: "$[1].currencyCode",
          requestedFields: "$[1].requestedFields"
        },
        singleArgumentUnchanged: false
      },
      ["sitePaymentMethodTypeId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listSitePaymentMethodTypes2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    ids: options?.ids,
    languageCode: options?.languageCode,
    countryCode: options?.countryCode,
    currencyCode: options?.currencyCode,
    requestedFields: options?.requestedFields
  });
  const reqOpts = listSitePaymentMethodTypes(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths(result.data, [
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [
            { path: "sitePaymentMethodTypes.merchantFlow.icons.image" },
            { path: "sitePaymentMethodTypes.buyerFlow.icons.image" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          ids: "$[0].ids",
          languageCode: "$[0].languageCode",
          countryCode: "$[0].countryCode",
          currencyCode: "$[0].currencyCode",
          requestedFields: "$[0].requestedFields"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
export {
  Format,
  RequestedFields,
  getSitePaymentMethodType2 as getSitePaymentMethodType,
  listSitePaymentMethodTypes2 as listSitePaymentMethodTypes
};
