// @wix/auto_sdk_events_ticket-reservations/build/internal/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  CheckoutType: () => CheckoutType,
  FeeType: () => FeeType,
  OrderStatus: () => OrderStatus,
  PromotionType: () => PromotionType,
  ReservationStatus: () => ReservationStatus,
  SubdivisionType: () => SubdivisionType,
  Type: () => Type,
  WebhookIdentityType: () => WebhookIdentityType,
  bulkUpdateTicketReservationTags: () => bulkUpdateTicketReservationTags4,
  bulkUpdateTicketReservationTagsByFilter: () => bulkUpdateTicketReservationTagsByFilter4,
  cancelTicketReservation: () => cancelTicketReservation4,
  createTicketReservation: () => createTicketReservation4,
  deleteTicketReservation: () => deleteTicketReservation4,
  getTicketReservation: () => getTicketReservation4,
  onTicketReservationCreated: () => onTicketReservationCreated2,
  onTicketReservationDeleted: () => onTicketReservationDeleted2,
  onTicketReservationUpdated: () => onTicketReservationUpdated2
});
module.exports = __toCommonJS(index_exports);

// src/events-v1-ticket-reservation-ticket-reservations.public.ts
var import_rename_all_nested_keys2 = require("@wix/sdk-runtime/rename-all-nested-keys");
var import_timestamp3 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var import_sdk_types = require("@wix/sdk-types");

// src/events-v1-ticket-reservation-ticket-reservations.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/events-v1-ticket-reservation-ticket-reservations.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEventsTicketreservationsV1TicketReservationManagementUrl(opts) {
  const domainToMappings = {
    _: [
      {
        srcPath: "/_api/events-ticket-reservations",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/events-ticket-reservations",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/events-ticket-reservations",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/events/v1/ticket-reservations",
        destPath: "/v1/ticket-reservations"
      },
      {
        srcPath: "/events/v1/bulk/ticket-reservations",
        destPath: "/v1/bulk/ticket-reservations"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_events_ticket-reservations";
function createTicketReservation(payload) {
  function __createTicketReservation({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "ticketReservation.createdDate" },
          { path: "ticketReservation.updatedDate" },
          { path: "ticketReservation.expirationDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v1.ticket_reservation",
      method: "POST",
      methodFqn: "wix.events.ticketreservations.v1.TicketReservationManagement.CreateTicketReservation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketreservationsV1TicketReservationManagementUrl({
        protoPath: "/v1/ticket-reservations",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "ticketReservation.createdDate" },
            { path: "ticketReservation.updatedDate" },
            { path: "ticketReservation.expirationDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createTicketReservation;
}
function getTicketReservation(payload) {
  function __getTicketReservation({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.ticket_reservation",
      method: "GET",
      methodFqn: "wix.events.ticketreservations.v1.TicketReservationManagement.GetTicketReservation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketreservationsV1TicketReservationManagementUrl({
        protoPath: "/v1/ticket-reservations/{ticketReservationId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "ticketReservation.createdDate" },
            { path: "ticketReservation.updatedDate" },
            { path: "ticketReservation.expirationDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getTicketReservation;
}
function deleteTicketReservation(payload) {
  function __deleteTicketReservation({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.ticket_reservation",
      method: "DELETE",
      methodFqn: "wix.events.ticketreservations.v1.TicketReservationManagement.DeleteTicketReservation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketreservationsV1TicketReservationManagementUrl({
        protoPath: "/v1/ticket-reservations/{ticketReservationId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "ticketReservation.createdDate" },
            { path: "ticketReservation.updatedDate" },
            { path: "ticketReservation.expirationDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __deleteTicketReservation;
}
function bulkUpdateTicketReservationTags(payload) {
  function __bulkUpdateTicketReservationTags({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.ticket_reservation",
      method: "POST",
      methodFqn: "wix.events.ticketreservations.v1.TicketReservationManagement.BulkUpdateTicketReservationTags",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketreservationsV1TicketReservationManagementUrl({
        protoPath: "/v1/bulk/ticket-reservations/update-tags",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateTicketReservationTags;
}
function bulkUpdateTicketReservationTagsByFilter(payload) {
  function __bulkUpdateTicketReservationTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.ticket_reservation",
      method: "POST",
      methodFqn: "wix.events.ticketreservations.v1.TicketReservationManagement.BulkUpdateTicketReservationTagsByFilter",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketreservationsV1TicketReservationManagementUrl({
        protoPath: "/v1/bulk/ticket-reservations/update-tags-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateTicketReservationTagsByFilter;
}
function cancelTicketReservation(payload) {
  function __cancelTicketReservation({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.ticket_reservation",
      method: "POST",
      methodFqn: "wix.events.ticketreservations.v1.TicketReservationManagement.CancelTicketReservation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketreservationsV1TicketReservationManagementUrl({
        protoPath: "/v1/ticket-reservations/{ticketReservationId}/cancel",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "ticketReservation.createdDate" },
            { path: "ticketReservation.updatedDate" },
            { path: "ticketReservation.expirationDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __cancelTicketReservation;
}

// src/events-v1-ticket-reservation-ticket-reservations.universal.ts
var ReservationStatus = /* @__PURE__ */ ((ReservationStatus2) => {
  ReservationStatus2["PENDING"] = "PENDING";
  ReservationStatus2["CONFIRMED"] = "CONFIRMED";
  ReservationStatus2["CANCELED"] = "CANCELED";
  ReservationStatus2["CANCELED_MANUALLY"] = "CANCELED_MANUALLY";
  ReservationStatus2["EXPIRED"] = "EXPIRED";
  return ReservationStatus2;
})(ReservationStatus || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["STANDARD"] = "STANDARD";
  Type2["DONATION"] = "DONATION";
  return Type2;
})(Type || {});
var CheckoutType = /* @__PURE__ */ ((CheckoutType2) => {
  CheckoutType2["UNKNOWN_CHECKOUT_TYPE"] = "UNKNOWN_CHECKOUT_TYPE";
  CheckoutType2["EVENTS_APP"] = "EVENTS_APP";
  CheckoutType2["ECOMM_PLATFORM"] = "ECOMM_PLATFORM";
  return CheckoutType2;
})(CheckoutType || {});
var FeeType = /* @__PURE__ */ ((FeeType2) => {
  FeeType2["FEE_INCLUDED"] = "FEE_INCLUDED";
  FeeType2["FEE_ADDED_AT_CHECKOUT"] = "FEE_ADDED_AT_CHECKOUT";
  return FeeType2;
})(FeeType || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
var PromotionType = /* @__PURE__ */ ((PromotionType2) => {
  PromotionType2["UNKNOWN_PROMOTION_TYPE"] = "UNKNOWN_PROMOTION_TYPE";
  PromotionType2["OTHER"] = "OTHER";
  PromotionType2["COUPON_CODE"] = "COUPON_CODE";
  PromotionType2["PAID_PLAN_BENEFIT"] = "PAID_PLAN_BENEFIT";
  PromotionType2["GIFT_CARD"] = "GIFT_CARD";
  return PromotionType2;
})(PromotionType || {});
var OrderStatus = /* @__PURE__ */ ((OrderStatus2) => {
  OrderStatus2["NA_ORDER_STATUS"] = "NA_ORDER_STATUS";
  OrderStatus2["FREE"] = "FREE";
  OrderStatus2["PENDING"] = "PENDING";
  OrderStatus2["PAID"] = "PAID";
  OrderStatus2["OFFLINE_PENDING"] = "OFFLINE_PENDING";
  OrderStatus2["INITIATED"] = "INITIATED";
  OrderStatus2["CANCELED"] = "CANCELED";
  OrderStatus2["DECLINED"] = "DECLINED";
  OrderStatus2["AUTHORIZED"] = "AUTHORIZED";
  OrderStatus2["VOIDED"] = "VOIDED";
  OrderStatus2["PARTIALLY_PAID"] = "PARTIALLY_PAID";
  return OrderStatus2;
})(OrderStatus || {});
var SubdivisionType = /* @__PURE__ */ ((SubdivisionType2) => {
  SubdivisionType2["UNKNOWN_SUBDIVISION_TYPE"] = "UNKNOWN_SUBDIVISION_TYPE";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_1"] = "ADMINISTRATIVE_AREA_LEVEL_1";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_2"] = "ADMINISTRATIVE_AREA_LEVEL_2";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_3"] = "ADMINISTRATIVE_AREA_LEVEL_3";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_4"] = "ADMINISTRATIVE_AREA_LEVEL_4";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_5"] = "ADMINISTRATIVE_AREA_LEVEL_5";
  SubdivisionType2["COUNTRY"] = "COUNTRY";
  return SubdivisionType2;
})(SubdivisionType || {});
async function createTicketReservation2(ticketReservation) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    ticketReservation
  });
  const reqOpts = createTicketReservation(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.ticketReservation;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { ticketReservation: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["ticketReservation"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getTicketReservation2(ticketReservationId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    ticketReservationId
  });
  const reqOpts = getTicketReservation(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.ticketReservation;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { ticketReservationId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["ticketReservationId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteTicketReservation2(ticketReservationId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    ticketReservationId
  });
  const reqOpts = deleteTicketReservation(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { ticketReservationId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["ticketReservationId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateTicketReservationTags2(ids, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    ids,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateTicketReservationTags(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          ids: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["ids", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateTicketReservationTagsByFilter2(filter, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    filter,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateTicketReservationTagsByFilter(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          filter: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["filter", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function cancelTicketReservation2(ticketReservationId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    ticketReservationId
  });
  const reqOpts = cancelTicketReservation(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { ticketReservationId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["ticketReservationId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/events-v1-ticket-reservation-ticket-reservations.public.ts
function createTicketReservation3(httpClient) {
  return (ticketReservation) => createTicketReservation2(
    ticketReservation,
    // @ts-ignore
    { httpClient }
  );
}
function getTicketReservation3(httpClient) {
  return (ticketReservationId) => getTicketReservation2(
    ticketReservationId,
    // @ts-ignore
    { httpClient }
  );
}
function deleteTicketReservation3(httpClient) {
  return (ticketReservationId) => deleteTicketReservation2(
    ticketReservationId,
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdateTicketReservationTags3(httpClient) {
  return (ids, options) => bulkUpdateTicketReservationTags2(
    ids,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdateTicketReservationTagsByFilter3(httpClient) {
  return (filter, options) => bulkUpdateTicketReservationTagsByFilter2(
    filter,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function cancelTicketReservation3(httpClient) {
  return (ticketReservationId) => cancelTicketReservation2(
    ticketReservationId,
    // @ts-ignore
    { httpClient }
  );
}
var onTicketReservationCreated = (0, import_sdk_types.EventDefinition)(
  "wix.events.v1.ticket_reservation_created",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths2.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.expirationDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onTicketReservationDeleted = (0, import_sdk_types.EventDefinition)(
  "wix.events.v1.ticket_reservation_deleted",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths2.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.expirationDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onTicketReservationUpdated = (0, import_sdk_types.EventDefinition)(
  "wix.events.v1.ticket_reservation_updated",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths2.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.expirationDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();

// src/events-v1-ticket-reservation-ticket-reservations.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var import_event_definition_modules = require("@wix/sdk-runtime/event-definition-modules");
var createTicketReservation4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(createTicketReservation3);
var getTicketReservation4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getTicketReservation3);
var deleteTicketReservation4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(deleteTicketReservation3);
var bulkUpdateTicketReservationTags4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkUpdateTicketReservationTags3);
var bulkUpdateTicketReservationTagsByFilter4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(
  bulkUpdateTicketReservationTagsByFilter3
);
var cancelTicketReservation4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(cancelTicketReservation3);
var onTicketReservationCreated2 = (0, import_event_definition_modules.createEventModule)(
  onTicketReservationCreated
);
var onTicketReservationDeleted2 = (0, import_event_definition_modules.createEventModule)(
  onTicketReservationDeleted
);
var onTicketReservationUpdated2 = (0, import_event_definition_modules.createEventModule)(
  onTicketReservationUpdated
);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  CheckoutType,
  FeeType,
  OrderStatus,
  PromotionType,
  ReservationStatus,
  SubdivisionType,
  Type,
  WebhookIdentityType,
  bulkUpdateTicketReservationTags,
  bulkUpdateTicketReservationTagsByFilter,
  cancelTicketReservation,
  createTicketReservation,
  deleteTicketReservation,
  getTicketReservation,
  onTicketReservationCreated,
  onTicketReservationDeleted,
  onTicketReservationUpdated
});
