// @wix/auto_sdk_events_ticket-definitions/build/es/index.typings.mjs
// src/events-v1-ticket-definition-ticket-definitions.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/events-v1-ticket-definition-ticket-definitions.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixEventsTicketingTicketDefinitionManagementUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/events/v2/ticket-definitions",
        destPath: "/v2/ticket-definitions"
      },
      {
        srcPath: "/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v2/ticket-definitions",
        destPath: "/v2/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/events-ticketing-v1-proxy",
        destPath: "/api"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/events/v2/ticket-definitions",
        destPath: "/v2/ticket-definitions"
      },
      {
        srcPath: "/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    _: [
      {
        srcPath: "/_api/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v2/ticket-definition",
        destPath: "/v2/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/events-ticketing-v1-proxy/v1",
        destPath: "/v1"
      },
      {
        srcPath: "/_api/events-ticketing-v1-proxy/v2",
        destPath: "/v2"
      },
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v2/ticket-definitions",
        destPath: "/v2/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v2/ticket-definitions",
        destPath: "/v2/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v2/ticket-definitions",
        destPath: "/v2/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    "api._base_domain_": [
      {
        srcPath: "/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    "editor-flow.wixapps.net": [
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_events_ticket-definitions";
function queryTicketDefinitions(payload) {
  function __queryTicketDefinitions({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.ticket_definition",
      method: "POST",
      methodFqn: "wix.events.ticketing.TicketDefinitionManagement.QueryTicketDefinitions",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingTicketDefinitionManagementUrl({
        protoPath: "/v1/ticket-definitions/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "definitions.salePeriod.startDate" },
            { path: "definitions.salePeriod.endDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryTicketDefinitions;
}
function queryTicketDefinitionsV2(payload) {
  function __queryTicketDefinitionsV2({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.ticket_definition",
      method: "POST",
      methodFqn: "wix.events.ticketing.TicketDefinitionManagement.QueryTicketDefinitionsV2",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingTicketDefinitionManagementUrl({
        protoPath: "/v2/ticket-definitions/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "definitions.salePeriod.startDate" },
            { path: "definitions.salePeriod.endDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryTicketDefinitionsV2;
}
function listTicketDefinitions(payload) {
  function __listTicketDefinitions({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.ticket_definition",
      method: "GET",
      methodFqn: "wix.events.ticketing.TicketDefinitionManagement.ListTicketDefinitions",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingTicketDefinitionManagementUrl({
        protoPath: "/v1/ticket-definitions",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "definitions.salePeriod.startDate" },
            { path: "definitions.salePeriod.endDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listTicketDefinitions;
}
function getTicketDefinition(payload) {
  function __getTicketDefinition({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.ticket_definition",
      method: "GET",
      methodFqn: "wix.events.ticketing.TicketDefinitionManagement.GetTicketDefinition",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingTicketDefinitionManagementUrl({
        protoPath: "/v1/ticket-definitions/{definitionId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "definition.salePeriod.startDate" },
            { path: "definition.salePeriod.endDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getTicketDefinition;
}
function createTicketDefinition(payload) {
  function __createTicketDefinition({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "definition.salePeriod.startDate" },
          { path: "definition.salePeriod.endDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v1.ticket_definition",
      method: "POST",
      methodFqn: "wix.events.ticketing.TicketDefinitionManagement.CreateTicketDefinition",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingTicketDefinitionManagementUrl({
        protoPath: "/v1/ticket-definitions",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "definition.salePeriod.startDate" },
            { path: "definition.salePeriod.endDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createTicketDefinition;
}
function updateTicketDefinition(payload) {
  function __updateTicketDefinition({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fields" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "definition.salePeriod.startDate" },
          { path: "definition.salePeriod.endDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v1.ticket_definition",
      method: "PATCH",
      methodFqn: "wix.events.ticketing.TicketDefinitionManagement.UpdateTicketDefinition",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingTicketDefinitionManagementUrl({
        protoPath: "/v1/ticket-definitions/{definitionId}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "definition.salePeriod.startDate" },
            { path: "definition.salePeriod.endDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateTicketDefinition;
}
function deleteTicketDefinition(payload) {
  function __deleteTicketDefinition({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.ticket_definition",
      method: "DELETE",
      methodFqn: "wix.events.ticketing.TicketDefinitionManagement.DeleteTicketDefinition",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingTicketDefinitionManagementUrl({
        protoPath: "/v1/ticket-definitions",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteTicketDefinition;
}
function changeCurrency(payload) {
  function __changeCurrency({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.ticket_definition",
      method: "PUT",
      methodFqn: "wix.events.ticketing.TicketDefinitionManagement.ChangeCurrency",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingTicketDefinitionManagementUrl({
        protoPath: "/v1/ticket-definitions/currency",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __changeCurrency;
}

// src/events-v1-ticket-definition-ticket-definitions.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var FeeType = /* @__PURE__ */ ((FeeType2) => {
  FeeType2["FEE_ADDED"] = "FEE_ADDED";
  FeeType2["FEE_INCLUDED"] = "FEE_INCLUDED";
  FeeType2["FEE_ADDED_AT_CHECKOUT"] = "FEE_ADDED_AT_CHECKOUT";
  return FeeType2;
})(FeeType || {});
var TicketSaleStatus = /* @__PURE__ */ ((TicketSaleStatus2) => {
  TicketSaleStatus2["SALE_SCHEDULED"] = "SALE_SCHEDULED";
  TicketSaleStatus2["SALE_STARTED"] = "SALE_STARTED";
  TicketSaleStatus2["SALE_ENDED"] = "SALE_ENDED";
  return TicketSaleStatus2;
})(TicketSaleStatus || {});
var TicketDefinitionStateEnumState = /* @__PURE__ */ ((TicketDefinitionStateEnumState2) => {
  TicketDefinitionStateEnumState2["INCLUDE_HIDDEN_NOT_ON_SALE"] = "INCLUDE_HIDDEN_NOT_ON_SALE";
  return TicketDefinitionStateEnumState2;
})(TicketDefinitionStateEnumState || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["STANDARD"] = "STANDARD";
  Type2["DONATION"] = "DONATION";
  return Type2;
})(Type || {});
var TicketDefinitionFieldset = /* @__PURE__ */ ((TicketDefinitionFieldset2) => {
  TicketDefinitionFieldset2["POLICY"] = "POLICY";
  TicketDefinitionFieldset2["DASHBOARD"] = "DASHBOARD";
  return TicketDefinitionFieldset2;
})(TicketDefinitionFieldset || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var State = /* @__PURE__ */ ((State2) => {
  State2["VISIBLE"] = "VISIBLE";
  State2["HIDDEN"] = "HIDDEN";
  State2["FREE"] = "FREE";
  State2["PAID"] = "PAID";
  return State2;
})(State || {});
var GroupBy = /* @__PURE__ */ ((GroupBy2) => {
  GroupBy2["NONE"] = "NONE";
  GroupBy2["EVENT_CREATED_DESC"] = "EVENT_CREATED_DESC";
  GroupBy2["EVENT_CREATED_ASC"] = "EVENT_CREATED_ASC";
  return GroupBy2;
})(GroupBy || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function queryTicketDefinitions2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    offset: options?.offset,
    limit: options?.limit,
    fieldset: options?.fieldset,
    filter: options?.filter,
    sort: options?.sort,
    facet: options?.facet
  });
  const reqOpts = queryTicketDefinitions(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          offset: "$[0].offset",
          limit: "$[0].limit",
          fieldset: "$[0].fieldset",
          filter: "$[0].filter",
          sort: "$[0].sort",
          facet: "$[0].facet"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryTicketDefinitionsV22() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryTicketDefinitionsV2(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [])
      );
      return {
        items: transformedData?.definitions,
        pagingMetadata: transformedData?.metadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function listTicketDefinitions2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    eventId: options?.eventId,
    offset: options?.offset,
    limit: options?.limit,
    fieldset: options?.fieldset,
    eventCreatorId: options?.eventCreatorId,
    state: options?.state,
    sort: options?.sort,
    saleStatus: options?.saleStatus,
    facet: options?.facet
  });
  const reqOpts = listTicketDefinitions(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          eventId: "$[0].eventId",
          offset: "$[0].offset",
          limit: "$[0].limit",
          fieldset: "$[0].fieldset",
          eventCreatorId: "$[0].eventCreatorId",
          state: "$[0].state",
          sort: "$[0].sort",
          saleStatus: "$[0].saleStatus",
          facet: "$[0].facet"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getTicketDefinition2(definitionId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    definitionId,
    fieldset: options?.fieldset
  });
  const reqOpts = getTicketDefinition(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.definition;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          definitionId: "$[0]",
          fieldset: "$[1].fieldset"
        },
        singleArgumentUnchanged: false
      },
      ["definitionId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function createTicketDefinition2(eventId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    eventId,
    definition: options?.definition
  });
  const reqOpts = createTicketDefinition(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          eventId: "$[0]",
          definition: "$[1].definition"
        },
        singleArgumentUnchanged: false
      },
      ["eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateTicketDefinition2(definitionId, eventId, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    definitionId,
    eventId,
    definition: options?.definition,
    fields: options?.fields
  });
  const reqOpts = updateTicketDefinition(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          definitionId: "$[0]",
          eventId: "$[1]",
          definition: "$[2].definition",
          fields: "$[2].fields"
        },
        singleArgumentUnchanged: false
      },
      ["definitionId", "eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteTicketDefinition2(eventId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    eventId,
    byId: options?.byId,
    all: options?.all
  });
  const reqOpts = deleteTicketDefinition(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          eventId: "$[0]",
          byId: "$[1].byId",
          all: "$[1].all"
        },
        singleArgumentUnchanged: false
      },
      ["eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function changeCurrency2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    eventId: options?.eventId,
    currency: options?.currency
  });
  const reqOpts = changeCurrency(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          eventId: "$[0].eventId",
          currency: "$[0].currency"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
export {
  FeeType,
  GroupBy,
  SortOrder,
  State,
  TicketDefinitionFieldset,
  TicketDefinitionStateEnumState,
  TicketSaleStatus,
  Type,
  WebhookIdentityType,
  changeCurrency2 as changeCurrency,
  createTicketDefinition2 as createTicketDefinition,
  deleteTicketDefinition2 as deleteTicketDefinition,
  getTicketDefinition2 as getTicketDefinition,
  listTicketDefinitions2 as listTicketDefinitions,
  queryTicketDefinitions2 as queryTicketDefinitions,
  queryTicketDefinitionsV22 as queryTicketDefinitionsV2,
  updateTicketDefinition2 as updateTicketDefinition
};
