// @wix/bex-utils/@wix/ambassador-items-selection-spi-host-v1-provider/build/es/http.impl.js
import { toURLSearchParams } from '@wix/metro-runtime';
import { serializer } from '@wix/metro-runtime/ambassador';
import { resolveUrl } from '@wix/metro-runtime';
var _listInstalledProvidersRequest = {};
var _listInstalledProvidersResponse = {};
var _listProvidersRequest = {};
var _listProvidersResponse = {};
var _queryItemsRequest = {};
var _queryItemsResponse = {};
function resolveWixItemsSelectionSpiHostV1ItemsSelectionServiceUrl(opts) {
    var domainToMappings = {
        'manage._base_domain_': [
            {
                srcPath: '/_api/items-selection-service',
                destPath: '',
            },
        ],
        'dev._base_domain_': [
            {
                srcPath: '/_api/items-selection-service',
                destPath: '',
            },
        ],
        'platform.rise.ai': [
            {
                srcPath: '/v1/items-selection',
                destPath: '/v1/items-selection',
            },
        ],
    };
    return resolveUrl(Object.assign(opts, { domainToMappings: domainToMappings }));
}
/** Returns list of installed app providers for a specific site */
export function listInstalledProviders(payload) {
    var _a = serializer(_listInstalledProvidersRequest, {}), toReq = _a.toJSON, fromReq = _a.fromJSON;
    var fromRes = serializer(_listInstalledProvidersResponse, {}).fromJSON;
    function __listInstalledProviders(_a) {
        var host = _a.host;
        var serializedData = toReq(payload);
        var metadata = {
            entityFqdn: 'wix.items_selection.spi_host.v1.provider',
            method: 'GET',
            methodFqn: 'wix.items_selection.spi_host.v1.ItemsSelectionService.ListInstalledProviders',
            url: resolveWixItemsSelectionSpiHostV1ItemsSelectionServiceUrl({
                protoPath: '/v1/items-selection/installed-providers',
                data: serializedData,
                host: host,
            }),
            params: toURLSearchParams(serializedData),
            transformResponse: fromRes,
        };
        return metadata;
    }
    __listInstalledProviders.fromReq = fromReq;
    __listInstalledProviders.__isAmbassador = true;
    return __listInstalledProviders;
}
/** Returns list of all existing providers */
export function listProviders(payload) {
    var _a = serializer(_listProvidersRequest, {}), toReq = _a.toJSON, fromReq = _a.fromJSON;
    var fromRes = serializer(_listProvidersResponse, {}).fromJSON;
    function __listProviders(_a) {
        var host = _a.host;
        var serializedData = toReq(payload);
        var metadata = {
            entityFqdn: 'wix.items_selection.spi_host.v1.provider',
            method: 'POST',
            methodFqn: 'wix.items_selection.spi_host.v1.ItemsSelectionService.ListProviders',
            url: resolveWixItemsSelectionSpiHostV1ItemsSelectionServiceUrl({
                protoPath: '/v1/items-selection/providers',
                data: serializedData,
                host: host,
            }),
            data: serializedData,
            transformResponse: fromRes,
        };
        return metadata;
    }
    __listProviders.fromReq = fromReq;
    __listProviders.__isAmbassador = true;
    return __listProviders;
}
/**
 * Retrieves a list of items that match the specified criteria.
 *
 * Returns:
 * - A list of items that match the search criteria.
 * - Metadata about the search results, such as the total number of items and pagination information.
 */
export function queryItems(payload) {
    var _a = serializer(_queryItemsRequest, {}), toReq = _a.toJSON, fromReq = _a.fromJSON;
    var fromRes = serializer(_queryItemsResponse, {}).fromJSON;
    function __queryItems(_a) {
        var host = _a.host;
        var serializedData = toReq(payload);
        var metadata = {
            entityFqdn: 'wix.items_selection.spi_host.v1.provider',
            method: 'POST',
            methodFqn: 'wix.items_selection.spi_host.v1.ItemsSelectionService.QueryItems',
            url: resolveWixItemsSelectionSpiHostV1ItemsSelectionServiceUrl({
                protoPath: '/v1/items-selection/{providerKey}/items',
                data: serializedData,
                host: host,
            }),
            data: serializedData,
            transformResponse: fromRes,
        };
        return metadata;
    }
    __queryItems.fromReq = fromReq;
    __queryItems.__isAmbassador = true;
    return __queryItems;
}
