// @wix/bex-utils/@wix/ambassador-items-selection-spi-host-v1-provider/build/es/meta.impl.js
import * as httpWixItemsSelectionSpiHostV1Provider from './http.impl';
export function listInstalledProviders() {
    var payload = {};
    var getRequestOptions = httpWixItemsSelectionSpiHostV1Provider.listInstalledProviders(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/v1/items-selection/installed-providers',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listProviders() {
    var payload = {};
    var getRequestOptions = httpWixItemsSelectionSpiHostV1Provider.listProviders(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/items-selection/providers',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryItems() {
    var payload = { providerKey: ':providerKey' };
    var getRequestOptions = httpWixItemsSelectionSpiHostV1Provider.queryItems(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/items-selection/{providerKey}/items',
        pathParams: { providerKey: 'providerKey' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
