// @wix/dashboard/dist/cjs/hostPlatform.js
"use strict";

exports.__esModule = true;
exports.createHost = void 0;
var _communicationChannel = require("@wix/communication-channel");
var _monitoringBrowserSdkHost = require("@wix/monitoring-browser-sdk-host");
var _registerHeightReporter = require("./helpers/registerHeightReporter");
var _isWixOrigin = require("./helpers/isWixOrigin");
var _registerOnBeforeUnload = require("./helpers/registerOnBeforeUnload");
var _essentials = require("./helpers/essentials");
const createHost = function (config = {}) {
  // when full url is provided (worker case) use it and ignore own location search
  const searchString = config.fullUrl && new URL(config.fullUrl).search || typeof window !== 'undefined' && window.location.search || '';
  const queryParams = new URLSearchParams(searchString);
  const origin = config.origin || queryParams.get('origin') || '';
  const postMessage = config.postMessage || window.parent.postMessage.bind(window.parent);
  if (!(0, _isWixOrigin.isWixOrigin)(origin)) {
    throw new Error(`Wix Dashboard SDK: Unable to establish a connection with the Wix dashboard. Please ensure that you are running your app within the Wix dashboard.`);
  }
  const {
    channel,
    close
  } = (0, _communicationChannel.open)({
    postMessage,
    origin
  });
  const closeMethods = [close];
  if (config.autoHeightUpdate === undefined || config.autoHeightUpdate) {
    const element = config.autoHeightElement || document.documentElement;
    const clearHeightReport = (0, _registerHeightReporter.registerHeightReporter)(channel, element);
    closeMethods.push(() => clearHeightReport());
  }
  const {
    onBeforeUnload,
    clearBeforeUnloadHandler
  } = (0, _registerOnBeforeUnload.registerBeforeUnloadHandler)(channel);
  closeMethods.push(clearBeforeUnloadHandler);
  const siteInfo = parseSiteInfo(queryParams.get('siteInfo') || '');
  const getSiteInfoSync = () => siteInfo;
  const channelProxy = new Proxy(channel, {
    get(target, prop, receiver) {
      if (prop === 'onBeforeUnload') {
        return onBeforeUnload;
      }
      if (prop === 'getSiteInfo') {
        return getSiteInfoSync;
      }
      return Reflect.get(target, prop, receiver);
    }
  });
  const {
    monitoring,
    ...essentials
  } = (0, _essentials.getEssentialsFromQueryParams)(queryParams) || {};
  const getMonitoringClient = (0, _monitoringBrowserSdkHost.getMonitoringClientFunction)({
    hostContext: monitoring == null ? void 0 : monitoring.context,
    monitoringConfig: monitoring == null ? void 0 : monitoring.config,
    sentrySDK: typeof window !== 'undefined' ? window.Sentry : undefined
  });
  return {
    channel: channelProxy,
    close: () => closeMethods.forEach(method => method()),
    getMonitoringClient,
    essentials
  };
};
exports.createHost = createHost;
const parseSiteInfo = siteInfo => {
  try {
    return JSON.parse(decodeURIComponent(siteInfo));
  } catch (e) {
    console.warn('Failed to parse "siteInfo" query parameter. Make sure to run the code inside a supported Wix App extension.', e);
    return {};
  }
};
