// @wix/dashboard/dist/cjs/internal.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.withContextualWixClient = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireWildcard(require("react"));
var _sdk = require("@wix/sdk");
var _index = require("./index");
var _jsxFileName = "/home/builduser/work/299b793a8321398/giza/giza-dashboard-sdk/dist/cjs/internal.tsx"; // eslint-disable-next-line import/no-extraneous-dependencies
// eslint-disable-next-line import/no-extraneous-dependencies
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
const useBeforeRenderEffect = (effect, effectCondition) => {
  const [lastEffectResult, setState] = (0, _react.useState)(() => effect());
  if (effectCondition(lastEffectResult)) {
    setState(effect());
  }
  return lastEffectResult;
};
const withContextualWixClient = Component => {
  return ownPropsWithHost => {
    const {
      host,
      compProps
    } = useHostAndProps(ownPropsWithHost);
    useBeforeRenderEffect(() => {
      (0, _sdk.createClient)({
        auth: _index.dashboard.auth(),
        host
      }).enableContext('module');
      return host;
    }, lastHost => lastHost !== host);
    return compProps ? /*#__PURE__*/_react.default.createElement(Component, (0, _extends2.default)({
      host: host
    }, compProps, {
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 35,
        columnNumber: 24
      }
    })) : null;
  };
};
exports.withContextualWixClient = withContextualWixClient;
const useHostAndProps = ownPropsWithHost => {
  const {
    host: maybeLocalHost,
    ...ownProps
  } = ownPropsWithHost;
  const {
    host,
    hostType
  } = useBeforeRenderEffect(() => maybeLocalHost ? {
    host: maybeLocalHost,
    hostType: 'local'
  } : {
    host: _index.dashboard.host(),
    hostType: 'remote'
  }, lastResult => !!maybeLocalHost && maybeLocalHost !== lastResult.host);
  const [hostProps, setHostProps] = (0, _react.useState)(undefined);
  (0, _react.useEffect)(() => {
    if (hostType === 'local') {
      return;
    }
    const observeResult = host.channel.observeState(newPropsFromHost => {
      setHostProps(newPropsFromHost || {});
    });
    const cleanup = async () => {
      (await observeResult).disconnect();
      host.close();
    };
    return () => {
      cleanup();
    };
  }, [host, hostType]);
  return hostType === 'local' ? {
    host,
    compProps: ownProps
  } : {
    host,
    compProps: hostProps ? {
      ...hostProps,
      ...ownProps
    } : undefined
  };
};
