// @wix/auto_sdk_bookings_services/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  ActionOriginal: () => Action,
  AddOnPaymentOptionsOriginal: () => AddOnPaymentOptions,
  AggregationTypeOriginal: () => AggregationType,
  BenefitTypeOriginal: () => BenefitType,
  CategoryNotificationEventOriginal: () => CategoryNotificationEvent,
  CloneErrorsOriginal: () => CloneErrors,
  CrudTypeOriginal: () => CrudType,
  DayOfWeekOriginal: () => DayOfWeek,
  EventOriginal: () => Event,
  IntervalOriginal: () => Interval,
  InvalidSlugErrorOriginal: () => InvalidSlugError,
  LocationTypeOriginal: () => LocationType,
  MissingValuesOriginal: () => MissingValues,
  ModeOriginal: () => Mode,
  NestedAggregationTypeOriginal: () => NestedAggregationType,
  PlacementTypeOriginal: () => PlacementType,
  RankingOrderOriginal: () => RankingOrder,
  RateTypeOriginal: () => RateType,
  RequestedFieldsOriginal: () => RequestedFields,
  ResolutionMethodOriginal: () => ResolutionMethod,
  ScalarTypeOriginal: () => ScalarType,
  ServiceTypeOriginal: () => ServiceType,
  SortDirectionOriginal: () => SortDirection,
  SortOrderOriginal: () => SortOrder,
  SortTypeOriginal: () => SortType,
  SortingMethodTypeOriginal: () => SortingMethodType,
  StatusOriginal: () => Status,
  TaxableAddressTypeOriginal: () => TaxableAddressType,
  V2RequestedFieldsOriginal: () => V2RequestedFields,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  bulkCreateServices: () => bulkCreateServices2,
  bulkDeleteServices: () => bulkDeleteServices2,
  bulkDeleteServicesByFilter: () => bulkDeleteServicesByFilter2,
  bulkUpdateServices: () => bulkUpdateServices2,
  bulkUpdateServicesByFilter: () => bulkUpdateServicesByFilter2,
  cloneService: () => cloneService2,
  countServices: () => countServices2,
  createAddOnGroup: () => createAddOnGroup2,
  createService: () => createService2,
  deleteAddOnGroup: () => deleteAddOnGroup2,
  deleteService: () => deleteService2,
  disablePricingPlansForService: () => disablePricingPlansForService2,
  enablePricingPlansForService: () => enablePricingPlansForService2,
  getService: () => getService2,
  listAddOnGroupsByServiceId: () => listAddOnGroupsByServiceId2,
  queryBookingForms: () => queryBookingForms2,
  queryCategories: () => queryCategories2,
  queryLocations: () => queryLocations2,
  queryPolicies: () => queryPolicies2,
  queryServices: () => queryServices2,
  searchServices: () => searchServices2,
  setAddOnsForGroup: () => setAddOnsForGroup2,
  setCustomSlug: () => setCustomSlug2,
  setServiceLocations: () => setServiceLocations2,
  updateAddOnGroup: () => updateAddOnGroup2,
  updateService: () => updateService2,
  validateSlug: () => validateSlug2
});
module.exports = __toCommonJS(meta_exports);

// src/bookings-services-v2-service-services.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixBookingsServicesV2ServicesServiceUrl(opts) {
  const domainToMappings = {
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/services-2",
        destPath: ""
      },
      {
        srcPath: "/_api/bookings/v2/services",
        destPath: "/v2/services"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/_api/bookings/v2/services",
        destPath: "/v2/services"
      },
      {
        srcPath: "/bookings/v2/services",
        destPath: "/v2/services"
      },
      {
        srcPath: "/bookings/v2/bulk/services",
        destPath: "/v2/bulk/services"
      },
      {
        srcPath: "/bookings/services/v2/bulk/services",
        destPath: "/v2/bulk/services"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/bookings/v2/services",
        destPath: "/v2/services"
      },
      {
        srcPath: "/_api/bookings/v2/bulk/services",
        destPath: "/v2/bulk/services"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/bookings/v2/services",
        destPath: "/v2/services"
      },
      {
        srcPath: "/_api/bookings/v2/bulk/services",
        destPath: "/v2/bulk/services"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/bookings/v2/services",
        destPath: "/v2/services"
      },
      {
        srcPath: "/_api/bookings/v2/bulk/services",
        destPath: "/v2/bulk/services"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/bookings/v2/services",
        destPath: "/v2/services"
      },
      {
        srcPath: "/_api/bookings/v2/bulk/services",
        destPath: "/v2/bulk/services"
      }
    ],
    _: [
      {
        srcPath: "/_api/bookings/v2/services",
        destPath: "/v2/services"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/bookings/v2/services",
        destPath: "/v2/services"
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/_api/bookings/v2/services",
        destPath: "/v2/services"
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/bookings/v2/services",
        destPath: "/v2/services"
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/bookings/v2/services",
        destPath: "/v2/services"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/bookings/v2/services",
        destPath: "/v2/services"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/services-2",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
function resolveWixBookingsServicesV2AddOnGroupsServiceUrl(opts) {
  const domainToMappings = {
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/services-2",
        destPath: ""
      },
      {
        srcPath: "/_api/bookings/v2/services",
        destPath: "/v2/services"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/_api/bookings/v2/services",
        destPath: "/v2/services"
      },
      {
        srcPath: "/bookings/v2/services",
        destPath: "/v2/services"
      },
      {
        srcPath: "/bookings/v2/bulk/services",
        destPath: "/v2/bulk/services"
      },
      {
        srcPath: "/bookings/services/v2/bulk/services",
        destPath: "/v2/bulk/services"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/bookings/v2/services",
        destPath: "/v2/services"
      },
      {
        srcPath: "/_api/bookings/v2/bulk/services",
        destPath: "/v2/bulk/services"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/bookings/v2/services",
        destPath: "/v2/services"
      },
      {
        srcPath: "/_api/bookings/v2/bulk/services",
        destPath: "/v2/bulk/services"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/bookings/v2/services",
        destPath: "/v2/services"
      },
      {
        srcPath: "/_api/bookings/v2/bulk/services",
        destPath: "/v2/bulk/services"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/bookings/v2/services",
        destPath: "/v2/services"
      },
      {
        srcPath: "/_api/bookings/v2/bulk/services",
        destPath: "/v2/bulk/services"
      }
    ],
    _: [
      {
        srcPath: "/_api/bookings/v2/services",
        destPath: "/v2/services"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/bookings/v2/services",
        destPath: "/v2/services"
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/_api/bookings/v2/services",
        destPath: "/v2/services"
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/bookings/v2/services",
        destPath: "/v2/services"
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/bookings/v2/services",
        destPath: "/v2/services"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/bookings/v2/services",
        destPath: "/v2/services"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/services-2",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_bookings_services";
function createAddOnGroup(payload) {
  function __createAddOnGroup({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.services.v2.service",
      method: "POST",
      methodFqn: "wix.bookings.services.v2.AddOnGroupsService.CreateAddOnGroup",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsServicesV2AddOnGroupsServiceUrl({
        protoPath: "/v2/services/add-on-groups/create",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __createAddOnGroup;
}
function deleteAddOnGroup(payload) {
  function __deleteAddOnGroup({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.services.v2.service",
      method: "POST",
      methodFqn: "wix.bookings.services.v2.AddOnGroupsService.DeleteAddOnGroup",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsServicesV2AddOnGroupsServiceUrl({
        protoPath: "/v2/services/add-on-groups/delete",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __deleteAddOnGroup;
}
function updateAddOnGroup(payload) {
  function __updateAddOnGroup({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.bookings.services.v2.service",
      method: "POST",
      methodFqn: "wix.bookings.services.v2.AddOnGroupsService.UpdateAddOnGroup",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsServicesV2AddOnGroupsServiceUrl({
        protoPath: "/v2/services/add-on-groups/update",
        data: serializedData,
        host
      }),
      data: serializedData
    };
    return metadata;
  }
  return __updateAddOnGroup;
}
function listAddOnGroupsByServiceId(payload) {
  function __listAddOnGroupsByServiceId({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.services.v2.service",
      method: "POST",
      methodFqn: "wix.bookings.services.v2.AddOnGroupsService.ListAddOnGroupsByServiceId",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsServicesV2AddOnGroupsServiceUrl({
        protoPath: "/v2/services/add-on-groups/list-add-on-groups-by-service-id",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __listAddOnGroupsByServiceId;
}
function setAddOnsForGroup(payload) {
  function __setAddOnsForGroup({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.services.v2.service",
      method: "POST",
      methodFqn: "wix.bookings.services.v2.AddOnGroupsService.SetAddOnsForGroup",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsServicesV2AddOnGroupsServiceUrl({
        protoPath: "/v2/services/add-on-groups/set-add-ons-for-group",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __setAddOnsForGroup;
}
function createService(payload) {
  function __createService({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "service.createdDate" },
          { path: "service.updatedDate" },
          { path: "service.media.items.image.urlExpirationDate" },
          { path: "service.media.mainMedia.image.urlExpirationDate" },
          { path: "service.media.coverMedia.image.urlExpirationDate" },
          { path: "service.bookingPolicy.createdDate" },
          { path: "service.bookingPolicy.updatedDate" },
          { path: "service.schedule.firstSessionStart" },
          { path: "service.schedule.lastSessionEnd" },
          { path: "service.staffMembers.mainMedia.image.urlExpirationDate" },
          {
            path: "service.staffMemberDetails.staffMembers.mainMedia.image.urlExpirationDate"
          },
          { path: "service.supportedSlugs.createdDate" },
          { path: "service.mainSlug.createdDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "service.locations.business.address.geocode.latitude" },
          { path: "service.locations.business.address.geocode.longitude" },
          { path: "service.locations.custom.address.geocode.latitude" },
          { path: "service.locations.custom.address.geocode.longitude" },
          { path: "service.locations.calculatedAddress.geocode.latitude" },
          { path: "service.locations.calculatedAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.bookings.services.v2.service",
      method: "POST",
      methodFqn: "wix.bookings.services.v2.ServicesService.CreateService",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsServicesV2ServicesServiceUrl({
        protoPath: "/v2/services",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "service.createdDate" },
            { path: "service.updatedDate" },
            { path: "service.media.items.image.urlExpirationDate" },
            { path: "service.media.mainMedia.image.urlExpirationDate" },
            { path: "service.media.coverMedia.image.urlExpirationDate" },
            { path: "service.bookingPolicy.createdDate" },
            { path: "service.bookingPolicy.updatedDate" },
            { path: "service.schedule.firstSessionStart" },
            { path: "service.schedule.lastSessionEnd" },
            {
              path: "service.staffMembers.mainMedia.image.urlExpirationDate"
            },
            {
              path: "service.staffMemberDetails.staffMembers.mainMedia.image.urlExpirationDate"
            },
            { path: "service.supportedSlugs.createdDate" },
            { path: "service.mainSlug.createdDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "service.locations.business.address.geocode.latitude" },
            { path: "service.locations.business.address.geocode.longitude" },
            { path: "service.locations.custom.address.geocode.latitude" },
            { path: "service.locations.custom.address.geocode.longitude" },
            { path: "service.locations.calculatedAddress.geocode.latitude" },
            { path: "service.locations.calculatedAddress.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createService;
}
function bulkCreateServices(payload) {
  function __bulkCreateServices({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "services.createdDate" },
          { path: "services.updatedDate" },
          { path: "services.media.items.image.urlExpirationDate" },
          { path: "services.media.mainMedia.image.urlExpirationDate" },
          { path: "services.media.coverMedia.image.urlExpirationDate" },
          { path: "services.bookingPolicy.createdDate" },
          { path: "services.bookingPolicy.updatedDate" },
          { path: "services.schedule.firstSessionStart" },
          { path: "services.schedule.lastSessionEnd" },
          { path: "services.staffMembers.mainMedia.image.urlExpirationDate" },
          {
            path: "services.staffMemberDetails.staffMembers.mainMedia.image.urlExpirationDate"
          },
          { path: "services.supportedSlugs.createdDate" },
          { path: "services.mainSlug.createdDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "services.locations.business.address.geocode.latitude" },
          { path: "services.locations.business.address.geocode.longitude" },
          { path: "services.locations.custom.address.geocode.latitude" },
          { path: "services.locations.custom.address.geocode.longitude" },
          { path: "services.locations.calculatedAddress.geocode.latitude" },
          { path: "services.locations.calculatedAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.bookings.services.v2.service",
      method: "POST",
      methodFqn: "wix.bookings.services.v2.ServicesService.BulkCreateServices",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsServicesV2ServicesServiceUrl({
        protoPath: "/v2/bulk/services/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.media.items.image.urlExpirationDate" },
            { path: "results.item.media.mainMedia.image.urlExpirationDate" },
            { path: "results.item.media.coverMedia.image.urlExpirationDate" },
            { path: "results.item.bookingPolicy.createdDate" },
            { path: "results.item.bookingPolicy.updatedDate" },
            { path: "results.item.schedule.firstSessionStart" },
            { path: "results.item.schedule.lastSessionEnd" },
            {
              path: "results.item.staffMembers.mainMedia.image.urlExpirationDate"
            },
            {
              path: "results.item.staffMemberDetails.staffMembers.mainMedia.image.urlExpirationDate"
            },
            { path: "results.item.supportedSlugs.createdDate" },
            { path: "results.item.mainSlug.createdDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "results.item.locations.business.address.geocode.latitude"
            },
            {
              path: "results.item.locations.business.address.geocode.longitude"
            },
            {
              path: "results.item.locations.custom.address.geocode.latitude"
            },
            {
              path: "results.item.locations.custom.address.geocode.longitude"
            },
            {
              path: "results.item.locations.calculatedAddress.geocode.latitude"
            },
            {
              path: "results.item.locations.calculatedAddress.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateServices;
}
function getService(payload) {
  function __getService({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.services.v2.service",
      method: "GET",
      methodFqn: "wix.bookings.services.v2.ServicesService.GetService",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsServicesV2ServicesServiceUrl({
        protoPath: "/v2/services/{serviceId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "service.createdDate" },
            { path: "service.updatedDate" },
            { path: "service.media.items.image.urlExpirationDate" },
            { path: "service.media.mainMedia.image.urlExpirationDate" },
            { path: "service.media.coverMedia.image.urlExpirationDate" },
            { path: "service.bookingPolicy.createdDate" },
            { path: "service.bookingPolicy.updatedDate" },
            { path: "service.schedule.firstSessionStart" },
            { path: "service.schedule.lastSessionEnd" },
            {
              path: "service.staffMembers.mainMedia.image.urlExpirationDate"
            },
            {
              path: "service.staffMemberDetails.staffMembers.mainMedia.image.urlExpirationDate"
            },
            { path: "service.supportedSlugs.createdDate" },
            { path: "service.mainSlug.createdDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "service.locations.business.address.geocode.latitude" },
            { path: "service.locations.business.address.geocode.longitude" },
            { path: "service.locations.custom.address.geocode.latitude" },
            { path: "service.locations.custom.address.geocode.longitude" },
            { path: "service.locations.calculatedAddress.geocode.latitude" },
            { path: "service.locations.calculatedAddress.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getService;
}
function updateService(payload) {
  function __updateService({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "service.createdDate" },
          { path: "service.updatedDate" },
          { path: "service.media.items.image.urlExpirationDate" },
          { path: "service.media.mainMedia.image.urlExpirationDate" },
          { path: "service.media.coverMedia.image.urlExpirationDate" },
          { path: "service.bookingPolicy.createdDate" },
          { path: "service.bookingPolicy.updatedDate" },
          { path: "service.schedule.firstSessionStart" },
          { path: "service.schedule.lastSessionEnd" },
          { path: "service.staffMembers.mainMedia.image.urlExpirationDate" },
          {
            path: "service.staffMemberDetails.staffMembers.mainMedia.image.urlExpirationDate"
          },
          { path: "service.supportedSlugs.createdDate" },
          { path: "service.mainSlug.createdDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "service.locations.business.address.geocode.latitude" },
          { path: "service.locations.business.address.geocode.longitude" },
          { path: "service.locations.custom.address.geocode.latitude" },
          { path: "service.locations.custom.address.geocode.longitude" },
          { path: "service.locations.calculatedAddress.geocode.latitude" },
          { path: "service.locations.calculatedAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.bookings.services.v2.service",
      method: "PATCH",
      methodFqn: "wix.bookings.services.v2.ServicesService.UpdateService",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsServicesV2ServicesServiceUrl({
        protoPath: "/v2/services/{service.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "service.createdDate" },
            { path: "service.updatedDate" },
            { path: "service.media.items.image.urlExpirationDate" },
            { path: "service.media.mainMedia.image.urlExpirationDate" },
            { path: "service.media.coverMedia.image.urlExpirationDate" },
            { path: "service.bookingPolicy.createdDate" },
            { path: "service.bookingPolicy.updatedDate" },
            { path: "service.schedule.firstSessionStart" },
            { path: "service.schedule.lastSessionEnd" },
            {
              path: "service.staffMembers.mainMedia.image.urlExpirationDate"
            },
            {
              path: "service.staffMemberDetails.staffMembers.mainMedia.image.urlExpirationDate"
            },
            { path: "service.supportedSlugs.createdDate" },
            { path: "service.mainSlug.createdDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "service.locations.business.address.geocode.latitude" },
            { path: "service.locations.business.address.geocode.longitude" },
            { path: "service.locations.custom.address.geocode.latitude" },
            { path: "service.locations.custom.address.geocode.longitude" },
            { path: "service.locations.calculatedAddress.geocode.latitude" },
            { path: "service.locations.calculatedAddress.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateService;
}
function bulkUpdateServices(payload) {
  function __bulkUpdateServices({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "services.mask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "services.service.createdDate" },
          { path: "services.service.updatedDate" },
          { path: "services.service.media.items.image.urlExpirationDate" },
          { path: "services.service.media.mainMedia.image.urlExpirationDate" },
          { path: "services.service.media.coverMedia.image.urlExpirationDate" },
          { path: "services.service.bookingPolicy.createdDate" },
          { path: "services.service.bookingPolicy.updatedDate" },
          { path: "services.service.schedule.firstSessionStart" },
          { path: "services.service.schedule.lastSessionEnd" },
          {
            path: "services.service.staffMembers.mainMedia.image.urlExpirationDate"
          },
          {
            path: "services.service.staffMemberDetails.staffMembers.mainMedia.image.urlExpirationDate"
          },
          { path: "services.service.supportedSlugs.createdDate" },
          { path: "services.service.mainSlug.createdDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          {
            path: "services.service.locations.business.address.geocode.latitude"
          },
          {
            path: "services.service.locations.business.address.geocode.longitude"
          },
          {
            path: "services.service.locations.custom.address.geocode.latitude"
          },
          {
            path: "services.service.locations.custom.address.geocode.longitude"
          },
          {
            path: "services.service.locations.calculatedAddress.geocode.latitude"
          },
          {
            path: "services.service.locations.calculatedAddress.geocode.longitude"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.bookings.services.v2.service",
      method: "POST",
      methodFqn: "wix.bookings.services.v2.ServicesService.BulkUpdateServices",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsServicesV2ServicesServiceUrl({
        protoPath: "/v2/bulk/services/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.media.items.image.urlExpirationDate" },
            { path: "results.item.media.mainMedia.image.urlExpirationDate" },
            { path: "results.item.media.coverMedia.image.urlExpirationDate" },
            { path: "results.item.bookingPolicy.createdDate" },
            { path: "results.item.bookingPolicy.updatedDate" },
            { path: "results.item.schedule.firstSessionStart" },
            { path: "results.item.schedule.lastSessionEnd" },
            {
              path: "results.item.staffMembers.mainMedia.image.urlExpirationDate"
            },
            {
              path: "results.item.staffMemberDetails.staffMembers.mainMedia.image.urlExpirationDate"
            },
            { path: "results.item.supportedSlugs.createdDate" },
            { path: "results.item.mainSlug.createdDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "results.item.locations.business.address.geocode.latitude"
            },
            {
              path: "results.item.locations.business.address.geocode.longitude"
            },
            {
              path: "results.item.locations.custom.address.geocode.latitude"
            },
            {
              path: "results.item.locations.custom.address.geocode.longitude"
            },
            {
              path: "results.item.locations.calculatedAddress.geocode.latitude"
            },
            {
              path: "results.item.locations.calculatedAddress.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateServices;
}
function bulkUpdateServicesByFilter(payload) {
  function __bulkUpdateServicesByFilter({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "service.createdDate" },
          { path: "service.updatedDate" },
          { path: "service.media.items.image.urlExpirationDate" },
          { path: "service.media.mainMedia.image.urlExpirationDate" },
          { path: "service.media.coverMedia.image.urlExpirationDate" },
          { path: "service.bookingPolicy.createdDate" },
          { path: "service.bookingPolicy.updatedDate" },
          { path: "service.schedule.firstSessionStart" },
          { path: "service.schedule.lastSessionEnd" },
          { path: "service.staffMembers.mainMedia.image.urlExpirationDate" },
          {
            path: "service.staffMemberDetails.staffMembers.mainMedia.image.urlExpirationDate"
          },
          { path: "service.supportedSlugs.createdDate" },
          { path: "service.mainSlug.createdDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "service.locations.business.address.geocode.latitude" },
          { path: "service.locations.business.address.geocode.longitude" },
          { path: "service.locations.custom.address.geocode.latitude" },
          { path: "service.locations.custom.address.geocode.longitude" },
          { path: "service.locations.calculatedAddress.geocode.latitude" },
          { path: "service.locations.calculatedAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.bookings.services.v2.service",
      method: "POST",
      methodFqn: "wix.bookings.services.v2.ServicesService.BulkUpdateServicesByFilter",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsServicesV2ServicesServiceUrl({
        protoPath: "/v2/bulk/services/update-by-filter",
        data: serializedData,
        host
      }),
      data: serializedData
    };
    return metadata;
  }
  return __bulkUpdateServicesByFilter;
}
function deleteService(payload) {
  function __deleteService({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.services.v2.service",
      method: "DELETE",
      methodFqn: "wix.bookings.services.v2.ServicesService.DeleteService",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsServicesV2ServicesServiceUrl({
        protoPath: "/v2/services/{serviceId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteService;
}
function bulkDeleteServices(payload) {
  function __bulkDeleteServices({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.services.v2.service",
      method: "POST",
      methodFqn: "wix.bookings.services.v2.ServicesService.BulkDeleteServices",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsServicesV2ServicesServiceUrl({
        protoPath: "/v2/bulk/services/delete",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.media.items.image.urlExpirationDate" },
            { path: "results.item.media.mainMedia.image.urlExpirationDate" },
            { path: "results.item.media.coverMedia.image.urlExpirationDate" },
            { path: "results.item.bookingPolicy.createdDate" },
            { path: "results.item.bookingPolicy.updatedDate" },
            { path: "results.item.schedule.firstSessionStart" },
            { path: "results.item.schedule.lastSessionEnd" },
            {
              path: "results.item.staffMembers.mainMedia.image.urlExpirationDate"
            },
            {
              path: "results.item.staffMemberDetails.staffMembers.mainMedia.image.urlExpirationDate"
            },
            { path: "results.item.supportedSlugs.createdDate" },
            { path: "results.item.mainSlug.createdDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "results.item.locations.business.address.geocode.latitude"
            },
            {
              path: "results.item.locations.business.address.geocode.longitude"
            },
            {
              path: "results.item.locations.custom.address.geocode.latitude"
            },
            {
              path: "results.item.locations.custom.address.geocode.longitude"
            },
            {
              path: "results.item.locations.calculatedAddress.geocode.latitude"
            },
            {
              path: "results.item.locations.calculatedAddress.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkDeleteServices;
}
function bulkDeleteServicesByFilter(payload) {
  function __bulkDeleteServicesByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.services.v2.service",
      method: "POST",
      methodFqn: "wix.bookings.services.v2.ServicesService.BulkDeleteServicesByFilter",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsServicesV2ServicesServiceUrl({
        protoPath: "/v2/bulk/services/delete-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkDeleteServicesByFilter;
}
function queryServices(payload) {
  function __queryServices({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.services.v2.service",
      method: "POST",
      methodFqn: "wix.bookings.services.v2.ServicesService.QueryServices",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsServicesV2ServicesServiceUrl({
        protoPath: "/v2/services/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "services.createdDate" },
            { path: "services.updatedDate" },
            { path: "services.media.items.image.urlExpirationDate" },
            { path: "services.media.mainMedia.image.urlExpirationDate" },
            { path: "services.media.coverMedia.image.urlExpirationDate" },
            { path: "services.bookingPolicy.createdDate" },
            { path: "services.bookingPolicy.updatedDate" },
            { path: "services.schedule.firstSessionStart" },
            { path: "services.schedule.lastSessionEnd" },
            {
              path: "services.staffMembers.mainMedia.image.urlExpirationDate"
            },
            {
              path: "services.staffMemberDetails.staffMembers.mainMedia.image.urlExpirationDate"
            },
            { path: "services.supportedSlugs.createdDate" },
            { path: "services.mainSlug.createdDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "services.locations.business.address.geocode.latitude" },
            { path: "services.locations.business.address.geocode.longitude" },
            { path: "services.locations.custom.address.geocode.latitude" },
            { path: "services.locations.custom.address.geocode.longitude" },
            { path: "services.locations.calculatedAddress.geocode.latitude" },
            {
              path: "services.locations.calculatedAddress.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryServices;
}
function searchServices(payload) {
  function __searchServices({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "search.aggregations.range.buckets.from" },
          { path: "search.aggregations.range.buckets.to" },
          {
            path: "search.aggregations.nested.nestedAggregations.range.buckets.from"
          },
          {
            path: "search.aggregations.nested.nestedAggregations.range.buckets.to"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.bookings.services.v2.service",
      method: "POST",
      methodFqn: "wix.bookings.services.v2.ServicesService.SearchServices",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsServicesV2ServicesServiceUrl({
        protoPath: "/v2/services/search",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "services.createdDate" },
            { path: "services.updatedDate" },
            { path: "services.media.items.image.urlExpirationDate" },
            { path: "services.media.mainMedia.image.urlExpirationDate" },
            { path: "services.media.coverMedia.image.urlExpirationDate" },
            { path: "services.bookingPolicy.createdDate" },
            { path: "services.bookingPolicy.updatedDate" },
            { path: "services.schedule.firstSessionStart" },
            { path: "services.schedule.lastSessionEnd" },
            {
              path: "services.staffMembers.mainMedia.image.urlExpirationDate"
            },
            {
              path: "services.staffMemberDetails.staffMembers.mainMedia.image.urlExpirationDate"
            },
            { path: "services.supportedSlugs.createdDate" },
            { path: "services.mainSlug.createdDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "services.locations.business.address.geocode.latitude" },
            { path: "services.locations.business.address.geocode.longitude" },
            { path: "services.locations.custom.address.geocode.latitude" },
            { path: "services.locations.custom.address.geocode.longitude" },
            { path: "services.locations.calculatedAddress.geocode.latitude" },
            {
              path: "services.locations.calculatedAddress.geocode.longitude"
            },
            { path: "aggregationData.results.ranges.results.from" },
            { path: "aggregationData.results.ranges.results.to" },
            {
              path: "aggregationData.results.groupedByValue.results.nestedResults.ranges.results.from"
            },
            {
              path: "aggregationData.results.groupedByValue.results.nestedResults.ranges.results.to"
            },
            {
              path: "aggregationData.results.nested.results.results.*.range.from"
            },
            {
              path: "aggregationData.results.nested.results.results.*.range.to"
            },
            { path: "aggregationData.results.scalar.value" },
            {
              path: "aggregationData.results.groupedByValue.results.nestedResults.scalar.value"
            },
            {
              path: "aggregationData.results.nested.results.results.*.scalar.value"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __searchServices;
}
function queryPolicies(payload) {
  function __queryPolicies({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.services.v2.service",
      method: "POST",
      methodFqn: "wix.bookings.services.v2.ServicesService.QueryPolicies",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsServicesV2ServicesServiceUrl({
        protoPath: "/v2/services/policies/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "bookingPolicies.bookingPolicy.createdDate" },
            { path: "bookingPolicies.bookingPolicy.updatedDate" },
            { path: "bookingPolicies.services.createdDate" },
            { path: "bookingPolicies.services.updatedDate" },
            {
              path: "bookingPolicies.services.media.items.image.urlExpirationDate"
            },
            {
              path: "bookingPolicies.services.media.mainMedia.image.urlExpirationDate"
            },
            {
              path: "bookingPolicies.services.media.coverMedia.image.urlExpirationDate"
            },
            { path: "bookingPolicies.services.bookingPolicy.createdDate" },
            { path: "bookingPolicies.services.bookingPolicy.updatedDate" },
            { path: "bookingPolicies.services.schedule.firstSessionStart" },
            { path: "bookingPolicies.services.schedule.lastSessionEnd" },
            {
              path: "bookingPolicies.services.staffMembers.mainMedia.image.urlExpirationDate"
            },
            {
              path: "bookingPolicies.services.staffMemberDetails.staffMembers.mainMedia.image.urlExpirationDate"
            },
            { path: "bookingPolicies.services.supportedSlugs.createdDate" },
            { path: "bookingPolicies.services.mainSlug.createdDate" },
            { path: "bookingPolicies.connectedServices.createdDate" },
            { path: "bookingPolicies.connectedServices.updatedDate" },
            {
              path: "bookingPolicies.connectedServices.media.items.image.urlExpirationDate"
            },
            {
              path: "bookingPolicies.connectedServices.media.mainMedia.image.urlExpirationDate"
            },
            {
              path: "bookingPolicies.connectedServices.media.coverMedia.image.urlExpirationDate"
            },
            {
              path: "bookingPolicies.connectedServices.bookingPolicy.createdDate"
            },
            {
              path: "bookingPolicies.connectedServices.bookingPolicy.updatedDate"
            },
            {
              path: "bookingPolicies.connectedServices.schedule.firstSessionStart"
            },
            {
              path: "bookingPolicies.connectedServices.schedule.lastSessionEnd"
            },
            {
              path: "bookingPolicies.connectedServices.staffMembers.mainMedia.image.urlExpirationDate"
            },
            {
              path: "bookingPolicies.connectedServices.staffMemberDetails.staffMembers.mainMedia.image.urlExpirationDate"
            },
            {
              path: "bookingPolicies.connectedServices.supportedSlugs.createdDate"
            },
            {
              path: "bookingPolicies.connectedServices.mainSlug.createdDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "bookingPolicies.services.locations.business.address.geocode.latitude"
            },
            {
              path: "bookingPolicies.services.locations.business.address.geocode.longitude"
            },
            {
              path: "bookingPolicies.services.locations.custom.address.geocode.latitude"
            },
            {
              path: "bookingPolicies.services.locations.custom.address.geocode.longitude"
            },
            {
              path: "bookingPolicies.services.locations.calculatedAddress.geocode.latitude"
            },
            {
              path: "bookingPolicies.services.locations.calculatedAddress.geocode.longitude"
            },
            {
              path: "bookingPolicies.connectedServices.locations.business.address.geocode.latitude"
            },
            {
              path: "bookingPolicies.connectedServices.locations.business.address.geocode.longitude"
            },
            {
              path: "bookingPolicies.connectedServices.locations.custom.address.geocode.latitude"
            },
            {
              path: "bookingPolicies.connectedServices.locations.custom.address.geocode.longitude"
            },
            {
              path: "bookingPolicies.connectedServices.locations.calculatedAddress.geocode.latitude"
            },
            {
              path: "bookingPolicies.connectedServices.locations.calculatedAddress.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryPolicies;
}
function queryBookingForms(payload) {
  function __queryBookingForms({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.services.v2.service",
      method: "POST",
      methodFqn: "wix.bookings.services.v2.ServicesService.QueryBookingForms",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsServicesV2ServicesServiceUrl({
        protoPath: "/v2/services/booking-forms/query",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __queryBookingForms;
}
function countServices(payload) {
  function __countServices({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.services.v2.service",
      method: "POST",
      methodFqn: "wix.bookings.services.v2.ServicesService.CountServices",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsServicesV2ServicesServiceUrl({
        protoPath: "/v2/services/count",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __countServices;
}
function queryLocations(payload) {
  function __queryLocations({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.services.v2.service",
      method: "POST",
      methodFqn: "wix.bookings.services.v2.ServicesService.QueryLocations",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsServicesV2ServicesServiceUrl({
        protoPath: "/v2/services/locations/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "businessLocations.locations.business.address.geocode.latitude"
            },
            {
              path: "businessLocations.locations.business.address.geocode.longitude"
            },
            {
              path: "businessLocations.locations.custom.address.geocode.latitude"
            },
            {
              path: "businessLocations.locations.custom.address.geocode.longitude"
            },
            {
              path: "businessLocations.locations.calculatedAddress.geocode.latitude"
            },
            {
              path: "businessLocations.locations.calculatedAddress.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryLocations;
}
function queryCategories(payload) {
  function __queryCategories({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.services.v2.service",
      method: "POST",
      methodFqn: "wix.bookings.services.v2.ServicesService.QueryCategories",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsServicesV2ServicesServiceUrl({
        protoPath: "/v2/services/categories/query",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __queryCategories;
}
function setServiceLocations(payload) {
  function __setServiceLocations({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "locations.business.address.geocode.latitude" },
          { path: "locations.business.address.geocode.longitude" },
          { path: "locations.custom.address.geocode.latitude" },
          { path: "locations.custom.address.geocode.longitude" },
          { path: "locations.calculatedAddress.geocode.latitude" },
          { path: "locations.calculatedAddress.geocode.longitude" },
          {
            path: "removedLocationSessionsAction.moveToLocationOptions.newLocation.business.address.geocode.latitude"
          },
          {
            path: "removedLocationSessionsAction.moveToLocationOptions.newLocation.business.address.geocode.longitude"
          },
          {
            path: "removedLocationSessionsAction.moveToLocationOptions.newLocation.custom.address.geocode.latitude"
          },
          {
            path: "removedLocationSessionsAction.moveToLocationOptions.newLocation.custom.address.geocode.longitude"
          },
          {
            path: "removedLocationSessionsAction.moveToLocationOptions.newLocation.calculatedAddress.geocode.latitude"
          },
          {
            path: "removedLocationSessionsAction.moveToLocationOptions.newLocation.calculatedAddress.geocode.longitude"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.bookings.services.v2.service",
      method: "POST",
      methodFqn: "wix.bookings.services.v2.ServicesService.SetServiceLocations",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsServicesV2ServicesServiceUrl({
        protoPath: "/v2/services/{serviceId}/locations",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "service.createdDate" },
            { path: "service.updatedDate" },
            { path: "service.media.items.image.urlExpirationDate" },
            { path: "service.media.mainMedia.image.urlExpirationDate" },
            { path: "service.media.coverMedia.image.urlExpirationDate" },
            { path: "service.bookingPolicy.createdDate" },
            { path: "service.bookingPolicy.updatedDate" },
            { path: "service.schedule.firstSessionStart" },
            { path: "service.schedule.lastSessionEnd" },
            {
              path: "service.staffMembers.mainMedia.image.urlExpirationDate"
            },
            {
              path: "service.staffMemberDetails.staffMembers.mainMedia.image.urlExpirationDate"
            },
            { path: "service.supportedSlugs.createdDate" },
            { path: "service.mainSlug.createdDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "service.locations.business.address.geocode.latitude" },
            { path: "service.locations.business.address.geocode.longitude" },
            { path: "service.locations.custom.address.geocode.latitude" },
            { path: "service.locations.custom.address.geocode.longitude" },
            { path: "service.locations.calculatedAddress.geocode.latitude" },
            { path: "service.locations.calculatedAddress.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __setServiceLocations;
}
function enablePricingPlansForService(payload) {
  function __enablePricingPlansForService({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.services.v2.service",
      method: "POST",
      methodFqn: "wix.bookings.services.v2.ServicesService.EnablePricingPlansForService",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsServicesV2ServicesServiceUrl({
        protoPath: "/v2/services/{serviceId}/pricing-plans/add",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "service.createdDate" },
            { path: "service.updatedDate" },
            { path: "service.media.items.image.urlExpirationDate" },
            { path: "service.media.mainMedia.image.urlExpirationDate" },
            { path: "service.media.coverMedia.image.urlExpirationDate" },
            { path: "service.bookingPolicy.createdDate" },
            { path: "service.bookingPolicy.updatedDate" },
            { path: "service.schedule.firstSessionStart" },
            { path: "service.schedule.lastSessionEnd" },
            {
              path: "service.staffMembers.mainMedia.image.urlExpirationDate"
            },
            {
              path: "service.staffMemberDetails.staffMembers.mainMedia.image.urlExpirationDate"
            },
            { path: "service.supportedSlugs.createdDate" },
            { path: "service.mainSlug.createdDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "service.locations.business.address.geocode.latitude" },
            { path: "service.locations.business.address.geocode.longitude" },
            { path: "service.locations.custom.address.geocode.latitude" },
            { path: "service.locations.custom.address.geocode.longitude" },
            { path: "service.locations.calculatedAddress.geocode.latitude" },
            { path: "service.locations.calculatedAddress.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __enablePricingPlansForService;
}
function disablePricingPlansForService(payload) {
  function __disablePricingPlansForService({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.services.v2.service",
      method: "POST",
      methodFqn: "wix.bookings.services.v2.ServicesService.DisablePricingPlansForService",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsServicesV2ServicesServiceUrl({
        protoPath: "/v2/services/{serviceId}/pricing-plans/remove",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "service.createdDate" },
            { path: "service.updatedDate" },
            { path: "service.media.items.image.urlExpirationDate" },
            { path: "service.media.mainMedia.image.urlExpirationDate" },
            { path: "service.media.coverMedia.image.urlExpirationDate" },
            { path: "service.bookingPolicy.createdDate" },
            { path: "service.bookingPolicy.updatedDate" },
            { path: "service.schedule.firstSessionStart" },
            { path: "service.schedule.lastSessionEnd" },
            {
              path: "service.staffMembers.mainMedia.image.urlExpirationDate"
            },
            {
              path: "service.staffMemberDetails.staffMembers.mainMedia.image.urlExpirationDate"
            },
            { path: "service.supportedSlugs.createdDate" },
            { path: "service.mainSlug.createdDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "service.locations.business.address.geocode.latitude" },
            { path: "service.locations.business.address.geocode.longitude" },
            { path: "service.locations.custom.address.geocode.latitude" },
            { path: "service.locations.custom.address.geocode.longitude" },
            { path: "service.locations.calculatedAddress.geocode.latitude" },
            { path: "service.locations.calculatedAddress.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __disablePricingPlansForService;
}
function setCustomSlug(payload) {
  function __setCustomSlug({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.services.v2.service",
      method: "POST",
      methodFqn: "wix.bookings.services.v2.ServicesService.SetCustomSlug",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsServicesV2ServicesServiceUrl({
        protoPath: "/v2/services/{serviceId}/slugs/custom",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "slug.createdDate" },
            { path: "service.createdDate" },
            { path: "service.updatedDate" },
            { path: "service.media.items.image.urlExpirationDate" },
            { path: "service.media.mainMedia.image.urlExpirationDate" },
            { path: "service.media.coverMedia.image.urlExpirationDate" },
            { path: "service.bookingPolicy.createdDate" },
            { path: "service.bookingPolicy.updatedDate" },
            { path: "service.schedule.firstSessionStart" },
            { path: "service.schedule.lastSessionEnd" },
            {
              path: "service.staffMembers.mainMedia.image.urlExpirationDate"
            },
            {
              path: "service.staffMemberDetails.staffMembers.mainMedia.image.urlExpirationDate"
            },
            { path: "service.supportedSlugs.createdDate" },
            { path: "service.mainSlug.createdDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "service.locations.business.address.geocode.latitude" },
            { path: "service.locations.business.address.geocode.longitude" },
            { path: "service.locations.custom.address.geocode.latitude" },
            { path: "service.locations.custom.address.geocode.longitude" },
            { path: "service.locations.calculatedAddress.geocode.latitude" },
            { path: "service.locations.calculatedAddress.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __setCustomSlug;
}
function validateSlug(payload) {
  function __validateSlug({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.services.v2.service",
      method: "POST",
      methodFqn: "wix.bookings.services.v2.ServicesService.ValidateSlug",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsServicesV2ServicesServiceUrl({
        protoPath: "/v2/services/slugs/validate",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __validateSlug;
}
function cloneService(payload) {
  function __cloneService({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.services.v2.service",
      method: "POST",
      methodFqn: "wix.bookings.services.v2.ServicesService.CloneService",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBookingsServicesV2ServicesServiceUrl({
        protoPath: "/v2/services/clone",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "service.createdDate" },
            { path: "service.updatedDate" },
            { path: "service.media.items.image.urlExpirationDate" },
            { path: "service.media.mainMedia.image.urlExpirationDate" },
            { path: "service.media.coverMedia.image.urlExpirationDate" },
            { path: "service.bookingPolicy.createdDate" },
            { path: "service.bookingPolicy.updatedDate" },
            { path: "service.schedule.firstSessionStart" },
            { path: "service.schedule.lastSessionEnd" },
            {
              path: "service.staffMembers.mainMedia.image.urlExpirationDate"
            },
            {
              path: "service.staffMemberDetails.staffMembers.mainMedia.image.urlExpirationDate"
            },
            { path: "service.supportedSlugs.createdDate" },
            { path: "service.mainSlug.createdDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "service.locations.business.address.geocode.latitude" },
            { path: "service.locations.business.address.geocode.longitude" },
            { path: "service.locations.custom.address.geocode.latitude" },
            { path: "service.locations.custom.address.geocode.longitude" },
            { path: "service.locations.calculatedAddress.geocode.latitude" },
            { path: "service.locations.calculatedAddress.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __cloneService;
}

// src/bookings-services-v2-service-services.types.ts
var ServiceType = /* @__PURE__ */ ((ServiceType2) => {
  ServiceType2["APPOINTMENT"] = "APPOINTMENT";
  ServiceType2["CLASS"] = "CLASS";
  ServiceType2["COURSE"] = "COURSE";
  return ServiceType2;
})(ServiceType || {});
var RateType = /* @__PURE__ */ ((RateType2) => {
  RateType2["UNKNOWN_RATE_TYPE"] = "UNKNOWN_RATE_TYPE";
  RateType2["FIXED"] = "FIXED";
  RateType2["CUSTOM"] = "CUSTOM";
  RateType2["VARIED"] = "VARIED";
  RateType2["NO_FEE"] = "NO_FEE";
  return RateType2;
})(RateType || {});
var AddOnPaymentOptions = /* @__PURE__ */ ((AddOnPaymentOptions2) => {
  AddOnPaymentOptions2["UNKNOWN_ADD_ON_PAYMENT_OPTIONS"] = "UNKNOWN_ADD_ON_PAYMENT_OPTIONS";
  AddOnPaymentOptions2["ONLINE"] = "ONLINE";
  AddOnPaymentOptions2["IN_PERSON"] = "IN_PERSON";
  return AddOnPaymentOptions2;
})(AddOnPaymentOptions || {});
var LocationType = /* @__PURE__ */ ((LocationType2) => {
  LocationType2["UNKNOWN_LOCATION_TYPE"] = "UNKNOWN_LOCATION_TYPE";
  LocationType2["CUSTOM"] = "CUSTOM";
  LocationType2["BUSINESS"] = "BUSINESS";
  LocationType2["CUSTOMER"] = "CUSTOMER";
  return LocationType2;
})(LocationType || {});
var RankingOrder = /* @__PURE__ */ ((RankingOrder2) => {
  RankingOrder2["UNKNOWN_RANKING_ORDER"] = "UNKNOWN_RANKING_ORDER";
  RankingOrder2["LOWEST_TO_HIGHEST"] = "LOWEST_TO_HIGHEST";
  RankingOrder2["HIGHEST_TO_LOWEST"] = "HIGHEST_TO_LOWEST";
  return RankingOrder2;
})(RankingOrder || {});
var SortingMethodType = /* @__PURE__ */ ((SortingMethodType2) => {
  SortingMethodType2["UNKNOWN_SORTING_METHOD_TYPE"] = "UNKNOWN_SORTING_METHOD_TYPE";
  SortingMethodType2["RANDOM"] = "RANDOM";
  SortingMethodType2["RANKING"] = "RANKING";
  SortingMethodType2["CUSTOM"] = "CUSTOM";
  return SortingMethodType2;
})(SortingMethodType || {});
var TaxableAddressType = /* @__PURE__ */ ((TaxableAddressType2) => {
  TaxableAddressType2["UNKNOWN_TAXABLE_ADDRESS_TYPE"] = "UNKNOWN_TAXABLE_ADDRESS_TYPE";
  TaxableAddressType2["BUSINESS"] = "BUSINESS";
  TaxableAddressType2["BILLING"] = "BILLING";
  return TaxableAddressType2;
})(TaxableAddressType || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
var V2RequestedFields = /* @__PURE__ */ ((V2RequestedFields2) => {
  V2RequestedFields2["UNKNOWN_REQUESTED_FIELD"] = "UNKNOWN_REQUESTED_FIELD";
  V2RequestedFields2["STAFF_MEMBER_DETAILS"] = "STAFF_MEMBER_DETAILS";
  V2RequestedFields2["RESOURCE_TYPE_DETAILS"] = "RESOURCE_TYPE_DETAILS";
  return V2RequestedFields2;
})(V2RequestedFields || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var SortType = /* @__PURE__ */ ((SortType2) => {
  SortType2["COUNT"] = "COUNT";
  SortType2["VALUE"] = "VALUE";
  return SortType2;
})(SortType || {});
var SortDirection = /* @__PURE__ */ ((SortDirection2) => {
  SortDirection2["DESC"] = "DESC";
  SortDirection2["ASC"] = "ASC";
  return SortDirection2;
})(SortDirection || {});
var MissingValues = /* @__PURE__ */ ((MissingValues2) => {
  MissingValues2["EXCLUDE"] = "EXCLUDE";
  MissingValues2["INCLUDE"] = "INCLUDE";
  return MissingValues2;
})(MissingValues || {});
var ScalarType = /* @__PURE__ */ ((ScalarType2) => {
  ScalarType2["UNKNOWN_SCALAR_TYPE"] = "UNKNOWN_SCALAR_TYPE";
  ScalarType2["COUNT_DISTINCT"] = "COUNT_DISTINCT";
  ScalarType2["MIN"] = "MIN";
  ScalarType2["MAX"] = "MAX";
  return ScalarType2;
})(ScalarType || {});
var NestedAggregationType = /* @__PURE__ */ ((NestedAggregationType2) => {
  NestedAggregationType2["UNKNOWN_AGGREGATION_TYPE"] = "UNKNOWN_AGGREGATION_TYPE";
  NestedAggregationType2["VALUE"] = "VALUE";
  NestedAggregationType2["RANGE"] = "RANGE";
  NestedAggregationType2["SCALAR"] = "SCALAR";
  NestedAggregationType2["DATE_HISTOGRAM"] = "DATE_HISTOGRAM";
  return NestedAggregationType2;
})(NestedAggregationType || {});
var Interval = /* @__PURE__ */ ((Interval2) => {
  Interval2["UNKNOWN_INTERVAL"] = "UNKNOWN_INTERVAL";
  Interval2["YEAR"] = "YEAR";
  Interval2["MONTH"] = "MONTH";
  Interval2["WEEK"] = "WEEK";
  Interval2["DAY"] = "DAY";
  Interval2["HOUR"] = "HOUR";
  Interval2["MINUTE"] = "MINUTE";
  Interval2["SECOND"] = "SECOND";
  return Interval2;
})(Interval || {});
var AggregationType = /* @__PURE__ */ ((AggregationType2) => {
  AggregationType2["UNKNOWN_AGGREGATION_TYPE"] = "UNKNOWN_AGGREGATION_TYPE";
  AggregationType2["VALUE"] = "VALUE";
  AggregationType2["RANGE"] = "RANGE";
  AggregationType2["SCALAR"] = "SCALAR";
  AggregationType2["DATE_HISTOGRAM"] = "DATE_HISTOGRAM";
  AggregationType2["NESTED"] = "NESTED";
  return AggregationType2;
})(AggregationType || {});
var Mode = /* @__PURE__ */ ((Mode2) => {
  Mode2["OR"] = "OR";
  Mode2["AND"] = "AND";
  return Mode2;
})(Mode || {});
var RequestedFields = /* @__PURE__ */ ((RequestedFields2) => {
  RequestedFields2["UNKNOWN_REQUESTED_FIELD"] = "UNKNOWN_REQUESTED_FIELD";
  RequestedFields2["DEFAULT_BOOKING_FORM"] = "DEFAULT_BOOKING_FORM";
  return RequestedFields2;
})(RequestedFields || {});
var Action = /* @__PURE__ */ ((Action2) => {
  Action2["UNKNOWN_ACTION_TYPE"] = "UNKNOWN_ACTION_TYPE";
  Action2["KEEP_AT_CURRENT_LOCATION"] = "KEEP_AT_CURRENT_LOCATION";
  Action2["MOVE_TO_LOCATION"] = "MOVE_TO_LOCATION";
  Action2["DELETE"] = "DELETE";
  return Action2;
})(Action || {});
var InvalidSlugError = /* @__PURE__ */ ((InvalidSlugError2) => {
  InvalidSlugError2["UNKNOWN_SLUG_ERROR"] = "UNKNOWN_SLUG_ERROR";
  InvalidSlugError2["SLUG_CONTAINS_ILLEGAL_CHARACTERS"] = "SLUG_CONTAINS_ILLEGAL_CHARACTERS";
  InvalidSlugError2["SLUG_ALREADY_EXISTS"] = "SLUG_ALREADY_EXISTS";
  return InvalidSlugError2;
})(InvalidSlugError || {});
var CloneErrors = /* @__PURE__ */ ((CloneErrors2) => {
  CloneErrors2["OPTIONS_AND_VARIANTS"] = "OPTIONS_AND_VARIANTS";
  CloneErrors2["FORM"] = "FORM";
  return CloneErrors2;
})(CloneErrors || {});
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["CREATED"] = "CREATED";
  Status2["DELETED"] = "DELETED";
  return Status2;
})(Status || {});
var CategoryNotificationEvent = /* @__PURE__ */ ((CategoryNotificationEvent2) => {
  CategoryNotificationEvent2["Updated"] = "Updated";
  CategoryNotificationEvent2["Deleted"] = "Deleted";
  CategoryNotificationEvent2["Created"] = "Created";
  return CategoryNotificationEvent2;
})(CategoryNotificationEvent || {});
var BenefitType = /* @__PURE__ */ ((BenefitType2) => {
  BenefitType2["UNDEFINED"] = "UNDEFINED";
  BenefitType2["LIMITED"] = "LIMITED";
  BenefitType2["UNLIMITED"] = "UNLIMITED";
  return BenefitType2;
})(BenefitType || {});
var Event = /* @__PURE__ */ ((Event2) => {
  Event2["Updated"] = "Updated";
  Event2["Deleted"] = "Deleted";
  Event2["Created"] = "Created";
  return Event2;
})(Event || {});
var CrudType = /* @__PURE__ */ ((CrudType2) => {
  CrudType2["INVALID_CRUD_TYPE"] = "INVALID_CRUD_TYPE";
  CrudType2["CREATE"] = "CREATE";
  CrudType2["UPDATE"] = "UPDATE";
  CrudType2["DELETE"] = "DELETE";
  CrudType2["CREATE_OR_UPDATE"] = "CREATE_OR_UPDATE";
  return CrudType2;
})(CrudType || {});
var PlacementType = /* @__PURE__ */ ((PlacementType2) => {
  PlacementType2["BEFORE"] = "BEFORE";
  PlacementType2["AFTER"] = "AFTER";
  PlacementType2["REPLACE"] = "REPLACE";
  return PlacementType2;
})(PlacementType || {});
var DayOfWeek = /* @__PURE__ */ ((DayOfWeek2) => {
  DayOfWeek2["MONDAY"] = "MONDAY";
  DayOfWeek2["TUESDAY"] = "TUESDAY";
  DayOfWeek2["WEDNESDAY"] = "WEDNESDAY";
  DayOfWeek2["THURSDAY"] = "THURSDAY";
  DayOfWeek2["FRIDAY"] = "FRIDAY";
  DayOfWeek2["SATURDAY"] = "SATURDAY";
  DayOfWeek2["SUNDAY"] = "SUNDAY";
  return DayOfWeek2;
})(DayOfWeek || {});
var ResolutionMethod = /* @__PURE__ */ ((ResolutionMethod2) => {
  ResolutionMethod2["QUERY_PARAM"] = "QUERY_PARAM";
  ResolutionMethod2["SUBDOMAIN"] = "SUBDOMAIN";
  ResolutionMethod2["SUBDIRECTORY"] = "SUBDIRECTORY";
  return ResolutionMethod2;
})(ResolutionMethod || {});

// src/bookings-services-v2-service-services.meta.ts
function createAddOnGroup2() {
  const payload = {};
  const getRequestOptions = createAddOnGroup(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/services/add-on-groups/create",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteAddOnGroup2() {
  const payload = {};
  const getRequestOptions = deleteAddOnGroup(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/services/add-on-groups/delete",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateAddOnGroup2() {
  const payload = {};
  const getRequestOptions = updateAddOnGroup(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/services/add-on-groups/update",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listAddOnGroupsByServiceId2() {
  const payload = {};
  const getRequestOptions = listAddOnGroupsByServiceId(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/services/add-on-groups/list-add-on-groups-by-service-id",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function setAddOnsForGroup2() {
  const payload = {};
  const getRequestOptions = setAddOnsForGroup(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/services/add-on-groups/set-add-ons-for-group",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createService2() {
  const payload = {};
  const getRequestOptions = createService(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/services",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkCreateServices2() {
  const payload = {};
  const getRequestOptions = bulkCreateServices(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/bulk/services/create",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getService2() {
  const payload = { serviceId: ":serviceId" };
  const getRequestOptions = getService(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/services/{serviceId}",
    pathParams: { serviceId: "serviceId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateService2() {
  const payload = { service: { id: ":serviceId" } };
  const getRequestOptions = updateService(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v2/services/{service.id}",
    pathParams: { serviceId: "serviceId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateServices2() {
  const payload = {};
  const getRequestOptions = bulkUpdateServices(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/bulk/services/update",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateServicesByFilter2() {
  const payload = {};
  const getRequestOptions = bulkUpdateServicesByFilter(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/bulk/services/update-by-filter",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteService2() {
  const payload = { serviceId: ":serviceId" };
  const getRequestOptions = deleteService(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v2/services/{serviceId}",
    pathParams: { serviceId: "serviceId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkDeleteServices2() {
  const payload = {};
  const getRequestOptions = bulkDeleteServices(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/bulk/services/delete",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkDeleteServicesByFilter2() {
  const payload = {};
  const getRequestOptions = bulkDeleteServicesByFilter(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/bulk/services/delete-by-filter",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryServices2() {
  const payload = {};
  const getRequestOptions = queryServices(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/services/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function searchServices2() {
  const payload = {};
  const getRequestOptions = searchServices(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/services/search",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryPolicies2() {
  const payload = {};
  const getRequestOptions = queryPolicies(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/services/policies/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryBookingForms2() {
  const payload = {};
  const getRequestOptions = queryBookingForms(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/services/booking-forms/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function countServices2() {
  const payload = {};
  const getRequestOptions = countServices(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/services/count",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryLocations2() {
  const payload = {};
  const getRequestOptions = queryLocations(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/services/locations/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryCategories2() {
  const payload = {};
  const getRequestOptions = queryCategories(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/services/categories/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function setServiceLocations2() {
  const payload = { serviceId: ":serviceId" };
  const getRequestOptions = setServiceLocations(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/services/{serviceId}/locations",
    pathParams: { serviceId: "serviceId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function enablePricingPlansForService2() {
  const payload = { serviceId: ":serviceId" };
  const getRequestOptions = enablePricingPlansForService(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/services/{serviceId}/pricing-plans/add",
    pathParams: { serviceId: "serviceId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function disablePricingPlansForService2() {
  const payload = { serviceId: ":serviceId" };
  const getRequestOptions = disablePricingPlansForService(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/services/{serviceId}/pricing-plans/remove",
    pathParams: { serviceId: "serviceId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function setCustomSlug2() {
  const payload = { serviceId: ":serviceId" };
  const getRequestOptions = setCustomSlug(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/services/{serviceId}/slugs/custom",
    pathParams: { serviceId: "serviceId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function validateSlug2() {
  const payload = {};
  const getRequestOptions = validateSlug(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/services/slugs/validate",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function cloneService2() {
  const payload = {};
  const getRequestOptions = cloneService(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/services/clone",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ActionOriginal,
  AddOnPaymentOptionsOriginal,
  AggregationTypeOriginal,
  BenefitTypeOriginal,
  CategoryNotificationEventOriginal,
  CloneErrorsOriginal,
  CrudTypeOriginal,
  DayOfWeekOriginal,
  EventOriginal,
  IntervalOriginal,
  InvalidSlugErrorOriginal,
  LocationTypeOriginal,
  MissingValuesOriginal,
  ModeOriginal,
  NestedAggregationTypeOriginal,
  PlacementTypeOriginal,
  RankingOrderOriginal,
  RateTypeOriginal,
  RequestedFieldsOriginal,
  ResolutionMethodOriginal,
  ScalarTypeOriginal,
  ServiceTypeOriginal,
  SortDirectionOriginal,
  SortOrderOriginal,
  SortTypeOriginal,
  SortingMethodTypeOriginal,
  StatusOriginal,
  TaxableAddressTypeOriginal,
  V2RequestedFieldsOriginal,
  WebhookIdentityTypeOriginal,
  bulkCreateServices,
  bulkDeleteServices,
  bulkDeleteServicesByFilter,
  bulkUpdateServices,
  bulkUpdateServicesByFilter,
  cloneService,
  countServices,
  createAddOnGroup,
  createService,
  deleteAddOnGroup,
  deleteService,
  disablePricingPlansForService,
  enablePricingPlansForService,
  getService,
  listAddOnGroupsByServiceId,
  queryBookingForms,
  queryCategories,
  queryLocations,
  queryPolicies,
  queryServices,
  searchServices,
  setAddOnsForGroup,
  setCustomSlug,
  setServiceLocations,
  updateAddOnGroup,
  updateService,
  validateSlug
});
