// @wix/ricos/dist/cjs/wix-rich-content-common/devTools/components/Button.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var Button_exports = {};
__export(Button_exports, {
  Button: () => Button
});
module.exports = __toCommonJS(Button_exports);
var import_react = __toESM(require("react"));
const Button = (props) => {
  return /* @__PURE__ */ import_react.default.createElement(import_react.default.Fragment, null, /* @__PURE__ */ import_react.default.createElement("style", null, `.ricos-dev-tools-button:hover {
        background-color: rgba(203, 166, 247, .4);
        color: #cdd6f4;
    }`), /* @__PURE__ */ import_react.default.createElement(
    "button",
    {
      type: "button",
      ...props,
      style: {
        border: "none",
        textTransform: "uppercase",
        fontWeight: 400,
        letterSpacing: "1px",
        padding: "6px 10px",
        margin: "0 0 0 8px",
        fontSize: "11px",
        color: "rgba(205, 214, 244, .8)",
        transition: "background .3s, color .3s",
        borderRadius: "2px",
        background: "transparent",
        cursor: "pointer"
      },
      className: "ricos-dev-tools-button"
    }
  ));
};
