// @wix/auto_sdk_restaurants_availability-exceptions/build/es/index.mjs
// src/restaurants-v1-availability-exception-availability-exceptions.public.ts
import { renameKeysFromRESTResponseToSDKResponse as renameKeysFromRESTResponseToSDKResponse2 } from "@wix/sdk-runtime/rename-all-nested-keys";
import { transformRESTTimestampToSDKTimestamp as transformRESTTimestampToSDKTimestamp2 } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths as transformPaths3 } from "@wix/sdk-runtime/transformations/transform-paths";
import { EventDefinition } from "@wix/sdk-types";

// src/restaurants-v1-availability-exception-availability-exceptions.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/restaurants-v1-availability-exception-availability-exceptions.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    _base_domain_: [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "users._base_domain_": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants-availability-exceptions",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_availability-exceptions";
function createAvailabilityException(payload) {
  function __createAvailabilityException({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "availabilityException.createdDate" },
          { path: "availabilityException.updatedDate" },
          { path: "availabilityException.startTime" },
          { path: "availabilityException.endTime" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "POST",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.CreateAvailabilityException",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/availability-exceptions",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "availabilityException.createdDate" },
            { path: "availabilityException.updatedDate" },
            { path: "availabilityException.startTime" },
            { path: "availabilityException.endTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createAvailabilityException;
}
function getAvailabilityException(payload) {
  function __getAvailabilityException({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "GET",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.GetAvailabilityException",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/availability-exceptions/{availabilityExceptionId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "availabilityException.createdDate" },
            { path: "availabilityException.updatedDate" },
            { path: "availabilityException.startTime" },
            { path: "availabilityException.endTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getAvailabilityException;
}
function updateAvailabilityException(payload) {
  function __updateAvailabilityException({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "availabilityException.createdDate" },
          { path: "availabilityException.updatedDate" },
          { path: "availabilityException.startTime" },
          { path: "availabilityException.endTime" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "PATCH",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.UpdateAvailabilityException",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/availability-exceptions/{availabilityException.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "availabilityException.createdDate" },
            { path: "availabilityException.updatedDate" },
            { path: "availabilityException.startTime" },
            { path: "availabilityException.endTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateAvailabilityException;
}
function deleteAvailabilityException(payload) {
  function __deleteAvailabilityException({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "DELETE",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.DeleteAvailabilityException",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/availability-exceptions/{availabilityExceptionId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteAvailabilityException;
}
function queryAvailabilityExceptions(payload) {
  function __queryAvailabilityExceptions({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "POST",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.QueryAvailabilityExceptions",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/availability-exceptions/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "availabilityExceptions.createdDate" },
            { path: "availabilityExceptions.updatedDate" },
            { path: "availabilityExceptions.startTime" },
            { path: "availabilityExceptions.endTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryAvailabilityExceptions;
}
function bulkUpdateAvailabilityExceptions(payload) {
  function __bulkUpdateAvailabilityExceptions({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "availabilityExceptions.mask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "availabilityExceptions.availabilityException.createdDate" },
          { path: "availabilityExceptions.availabilityException.updatedDate" },
          { path: "availabilityExceptions.availabilityException.startTime" },
          { path: "availabilityExceptions.availabilityException.endTime" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "POST",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.BulkUpdateAvailabilityExceptions",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/bulk/availability-exceptions/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.startTime" },
            { path: "results.item.endTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateAvailabilityExceptions;
}
function bulkUpdateAvailabilityExceptionTags(payload) {
  function __bulkUpdateAvailabilityExceptionTags({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "POST",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.BulkUpdateAvailabilityExceptionTags",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/bulk/availability-exceptions/update-tags",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateAvailabilityExceptionTags;
}
function bulkUpdateAvailabilityExceptionTagsByFilter(payload) {
  function __bulkUpdateAvailabilityExceptionTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "POST",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.BulkUpdateAvailabilityExceptionTagsByFilter",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/bulk/availability-exceptions/update-tags-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateAvailabilityExceptionTagsByFilter;
}

// src/restaurants-v1-availability-exception-availability-exceptions.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var AffectedMethods = /* @__PURE__ */ ((AffectedMethods2) => {
  AffectedMethods2["UNKNOWN_AFFECTED_METHODS"] = "UNKNOWN_AFFECTED_METHODS";
  AffectedMethods2["ALL_FULFILLMENT_METHODS"] = "ALL_FULFILLMENT_METHODS";
  AffectedMethods2["SPECIFIC_FULFILLMENT_METHODS"] = "SPECIFIC_FULFILLMENT_METHODS";
  AffectedMethods2["ALL_PICKUP_FULFILLMENT_METHODS"] = "ALL_PICKUP_FULFILLMENT_METHODS";
  AffectedMethods2["ALL_DELIVERY_FULFILLMENT_METHODS"] = "ALL_DELIVERY_FULFILLMENT_METHODS";
  return AffectedMethods2;
})(AffectedMethods || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNSPECIFIED_AVAILABILITY_EXCEPTION_TYPE"] = "UNSPECIFIED_AVAILABILITY_EXCEPTION_TYPE";
  Type2["ORDERING_PAUSED"] = "ORDERING_PAUSED";
  Type2["CLOSURE"] = "CLOSURE";
  return Type2;
})(Type || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createAvailabilityException2(availabilityException) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    availabilityException
  });
  const reqOpts = createAvailabilityException(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.availabilityException;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { availabilityException: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["availabilityException"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getAvailabilityException2(availabilityExceptionId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    availabilityExceptionId
  });
  const reqOpts = getAvailabilityException(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.availabilityException;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { availabilityExceptionId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["availabilityExceptionId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateAvailabilityException2(_id, availabilityException) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    availabilityException: { ...availabilityException, id: _id }
  });
  const reqOpts = updateAvailabilityException(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.availabilityException;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { availabilityException: "$[1]" },
        explicitPathsToArguments: { "availabilityException.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "availabilityException"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteAvailabilityException2(availabilityExceptionId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    availabilityExceptionId
  });
  const reqOpts = deleteAvailabilityException(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { availabilityExceptionId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["availabilityExceptionId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryAvailabilityExceptions2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryAvailabilityExceptions(
        payload
      );
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [])
      );
      return {
        items: transformedData?.availabilityExceptions,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function bulkUpdateAvailabilityExceptions2(availabilityExceptions, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    availabilityExceptions,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkUpdateAvailabilityExceptions(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          availabilityExceptions: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["availabilityExceptions", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateAvailabilityExceptionTags2(availabilityExceptionIds, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    availabilityExceptionIds,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateAvailabilityExceptionTags(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          availabilityExceptionIds: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["availabilityExceptionIds", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateAvailabilityExceptionTagsByFilter2(filter, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    filter,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateAvailabilityExceptionTagsByFilter(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          filter: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["filter", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/restaurants-v1-availability-exception-availability-exceptions.public.ts
function createAvailabilityException3(httpClient) {
  return (availabilityException) => createAvailabilityException2(
    availabilityException,
    // @ts-ignore
    { httpClient }
  );
}
function getAvailabilityException3(httpClient) {
  return (availabilityExceptionId) => getAvailabilityException2(
    availabilityExceptionId,
    // @ts-ignore
    { httpClient }
  );
}
function updateAvailabilityException3(httpClient) {
  return (_id, availabilityException) => updateAvailabilityException2(
    _id,
    availabilityException,
    // @ts-ignore
    { httpClient }
  );
}
function deleteAvailabilityException3(httpClient) {
  return (availabilityExceptionId) => deleteAvailabilityException2(
    availabilityExceptionId,
    // @ts-ignore
    { httpClient }
  );
}
function queryAvailabilityExceptions3(httpClient) {
  return () => queryAvailabilityExceptions2(
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdateAvailabilityExceptions3(httpClient) {
  return (availabilityExceptions, options) => bulkUpdateAvailabilityExceptions2(
    availabilityExceptions,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdateAvailabilityExceptionTags3(httpClient) {
  return (availabilityExceptionIds, options) => bulkUpdateAvailabilityExceptionTags2(
    availabilityExceptionIds,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdateAvailabilityExceptionTagsByFilter3(httpClient) {
  return (filter, options) => bulkUpdateAvailabilityExceptionTagsByFilter2(
    filter,
    options,
    // @ts-ignore
    { httpClient }
  );
}
var onAvailabilityExceptionCreated = EventDefinition(
  "wix.restaurants.v1.availability_exception_created",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.startTime" },
          { path: "entity.endTime" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onAvailabilityExceptionDeleted = EventDefinition(
  "wix.restaurants.v1.availability_exception_deleted",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.startTime" },
          { path: "entity.endTime" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onAvailabilityExceptionUpdated = EventDefinition(
  "wix.restaurants.v1.availability_exception_updated",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.startTime" },
          { path: "entity.endTime" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();

// src/restaurants-v1-availability-exception-availability-exceptions.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
import { createEventModule } from "@wix/sdk-runtime/event-definition-modules";
var createAvailabilityException4 = /* @__PURE__ */ createRESTModule(createAvailabilityException3);
var getAvailabilityException4 = /* @__PURE__ */ createRESTModule(getAvailabilityException3);
var updateAvailabilityException4 = /* @__PURE__ */ createRESTModule(updateAvailabilityException3);
var deleteAvailabilityException4 = /* @__PURE__ */ createRESTModule(deleteAvailabilityException3);
var queryAvailabilityExceptions4 = /* @__PURE__ */ createRESTModule(queryAvailabilityExceptions3);
var bulkUpdateAvailabilityExceptions4 = /* @__PURE__ */ createRESTModule(bulkUpdateAvailabilityExceptions3);
var bulkUpdateAvailabilityExceptionTags4 = /* @__PURE__ */ createRESTModule(bulkUpdateAvailabilityExceptionTags3);
var bulkUpdateAvailabilityExceptionTagsByFilter4 = /* @__PURE__ */ createRESTModule(
  bulkUpdateAvailabilityExceptionTagsByFilter3
);
var onAvailabilityExceptionCreated2 = createEventModule(
  onAvailabilityExceptionCreated
);
var onAvailabilityExceptionDeleted2 = createEventModule(
  onAvailabilityExceptionDeleted
);
var onAvailabilityExceptionUpdated2 = createEventModule(
  onAvailabilityExceptionUpdated
);
export {
  AffectedMethods,
  SortOrder,
  Type,
  WebhookIdentityType,
  bulkUpdateAvailabilityExceptionTags4 as bulkUpdateAvailabilityExceptionTags,
  bulkUpdateAvailabilityExceptionTagsByFilter4 as bulkUpdateAvailabilityExceptionTagsByFilter,
  bulkUpdateAvailabilityExceptions4 as bulkUpdateAvailabilityExceptions,
  createAvailabilityException4 as createAvailabilityException,
  deleteAvailabilityException4 as deleteAvailabilityException,
  getAvailabilityException4 as getAvailabilityException,
  onAvailabilityExceptionCreated2 as onAvailabilityExceptionCreated,
  onAvailabilityExceptionDeleted2 as onAvailabilityExceptionDeleted,
  onAvailabilityExceptionUpdated2 as onAvailabilityExceptionUpdated,
  queryAvailabilityExceptions4 as queryAvailabilityExceptions,
  updateAvailabilityException4 as updateAvailabilityException
};
