// @wix/design-system/dist/cjs/BarChart/BarChart.uni.driver.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));
var _unidriver = require("../utils/test-utils/unidriver");
var _dataHooks = _interopRequireDefault(require("./dataHooks"));
var _TooltipUni = require("../Tooltip/Tooltip.uni.driver");
var _AdaptiveHeadingUni = require("../common/AdaptiveHeading/AdaptiveHeading.uni.driver");
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2.default)(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var barChartDriverFactory = (base, body) => {
  var getTooltipDriver = /*#__PURE__*/function () {
    var _ref = (0, _asyncToGenerator2.default)(function* (index) {
      var item = yield (0, _unidriver.findByHookAtIndex)(base, _dataHooks.default.bar, index);
      var tooltip = yield (0, _unidriver.findByHook)(item, _dataHooks.default.tooltip);
      return (0, _TooltipUni.tooltipDriverFactory)(tooltip, body);
    });
    return function getTooltipDriver(_x) {
      return _ref.apply(this, arguments);
    };
  }();
  var getAdaptiveHeadingDriver = /*#__PURE__*/function () {
    var _ref2 = (0, _asyncToGenerator2.default)(function* (index) {
      var item = yield (0, _unidriver.findByHookAtIndex)(base, _dataHooks.default.bar, index);
      var heading = yield (0, _unidriver.findByHook)(item, _dataHooks.default.value);
      return (0, _AdaptiveHeadingUni.adaptiveHeadingDriverFactory)(heading);
    });
    return function getAdaptiveHeadingDriver(_x2) {
      return _ref2.apply(this, arguments);
    };
  }();
  return _objectSpread(_objectSpread({}, (0, _unidriver.baseUniDriverFactory)(base)), {}, {
    /**
     * Gets number of items
     * @return {Promise<number>}
     */
    getItemsCount: () => (0, _unidriver.countByHook)(base, _dataHooks.default.bar),
    /**
     * Gets item value at index
     * @param {number} index Item index
     * @return {Promise<string>}
     */
    getValue: function () {
      var _getValue = (0, _asyncToGenerator2.default)(function* (index) {
        return (yield getAdaptiveHeadingDriver(index)).getText();
      });
      function getValue(_x3) {
        return _getValue.apply(this, arguments);
      }
      return getValue;
    }(),
    /**
     * Gets item short value by index
     * @param index Item index
     * @return {Promise<string>}
     */
    getValueInShort: function () {
      var _getValueInShort = (0, _asyncToGenerator2.default)(function* (index) {
        return (yield getAdaptiveHeadingDriver(index)).getShortText();
      });
      function getValueInShort(_x4) {
        return _getValueInShort.apply(this, arguments);
      }
      return getValueInShort;
    }(),
    /**
     * Gets item description by index
     * @param index Item index
     * @return {Promise<string>}
     */
    getDescription: function () {
      var _getDescription = (0, _asyncToGenerator2.default)(function* (index) {
        return (0, _unidriver.findByHook)(yield (0, _unidriver.findByHookAtIndex)(base, _dataHooks.default.bar, index), _dataHooks.default.description).text();
      });
      function getDescription(_x5) {
        return _getDescription.apply(this, arguments);
      }
      return getDescription;
    }(),
    /**
     * Gets item tooltip text by index
     * @param index Item index
     * @return {Promise<string>}
     */
    getDescriptionInfo: function () {
      var _getDescriptionInfo = (0, _asyncToGenerator2.default)(function* (index) {
        var tooltip = yield getTooltipDriver(index);
        if (!(yield tooltip.exists())) {
          return null;
        }
        yield tooltip.mouseEnter();
        var text = yield tooltip.getTooltipText();
        yield tooltip.mouseLeave();
        return text;
      });
      function getDescriptionInfo(_x6) {
        return _getDescriptionInfo.apply(this, arguments);
      }
      return getDescriptionInfo;
    }()
  });
};
var _default = exports.default = barChartDriverFactory;
