// @wix/auto_sdk_ecom_tax-calculation/build/internal/es/meta.mjs
// src/gateways-billing-v1-tax-calculation-gateway-tax-calculation.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixBillingV1TaxCalculationUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/billing/v1/calculate-tax",
        destPath: "/v1/calculate-tax"
      },
      {
        srcPath: "/billing/v1/list-tax-calculators",
        destPath: "/v1/list-tax-calculators"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/billing/v1/calculate-tax",
        destPath: "/v1/calculate-tax"
      },
      {
        srcPath: "/billing/v1/list-tax-calculators",
        destPath: "/v1/list-tax-calculators"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/gateways/billing/v1/calculate-tax",
        destPath: "/v1/calculate-tax"
      }
    ],
    _: [
      {
        srcPath: "/gateways/billing/v1/calculate-tax",
        destPath: "/v1/calculate-tax"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_tax-calculation";
function calculateTax(payload) {
  function __calculateTax({ host }) {
    const metadata = {
      entityFqdn: "wix.gateways.billing.v1.tax_calculation_gateway",
      method: "POST",
      methodFqn: "wix.billing.v1.TaxCalculation.CalculateTax",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBillingV1TaxCalculationUrl({
        protoPath: "/v1/calculate-tax",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __calculateTax;
}
function listTaxCalculators(payload) {
  function __listTaxCalculators({ host }) {
    const metadata = {
      entityFqdn: "wix.gateways.billing.v1.tax_calculation_gateway",
      method: "GET",
      methodFqn: "wix.billing.v1.TaxCalculation.ListTaxCalculators",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixBillingV1TaxCalculationUrl({
        protoPath: "/v1/list-tax-calculators",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __listTaxCalculators;
}

// src/gateways-billing-v1-tax-calculation-gateway-tax-calculation.types.ts
var ExemptBy = /* @__PURE__ */ ((ExemptBy2) => {
  ExemptBy2["UNKNOWN_EXEMPT_BY"] = "UNKNOWN_EXEMPT_BY";
  ExemptBy2["EXEMPT_GROUP"] = "EXEMPT_GROUP";
  ExemptBy2["CONTACT"] = "CONTACT";
  return ExemptBy2;
})(ExemptBy || {});
var JurisdictionType = /* @__PURE__ */ ((JurisdictionType2) => {
  JurisdictionType2["UNDEFINED"] = "UNDEFINED";
  JurisdictionType2["COUNTRY"] = "COUNTRY";
  JurisdictionType2["STATE"] = "STATE";
  JurisdictionType2["COUNTY"] = "COUNTY";
  JurisdictionType2["CITY"] = "CITY";
  JurisdictionType2["SPECIAL"] = "SPECIAL";
  return JurisdictionType2;
})(JurisdictionType || {});

// src/gateways-billing-v1-tax-calculation-gateway-tax-calculation.meta.ts
function calculateTax2() {
  const payload = {};
  const getRequestOptions = calculateTax(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/calculate-tax",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listTaxCalculators2() {
  const payload = {};
  const getRequestOptions = listTaxCalculators(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/list-tax-calculators",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  ExemptBy as ExemptByOriginal,
  JurisdictionType as JurisdictionTypeOriginal,
  calculateTax2 as calculateTax,
  listTaxCalculators2 as listTaxCalculators
};
