// @wix/auto_sdk_events_rsvp/build/internal/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  RsvpFieldset: () => RsvpFieldset,
  RsvpStatus: () => RsvpStatus,
  RsvpTag: () => RsvpTag,
  SubdivisionType: () => SubdivisionType,
  WebhookIdentityType: () => WebhookIdentityType,
  bulkUpdateRsvp: () => bulkUpdateRsvp2,
  checkInRsvp: () => checkInRsvp2,
  createRsvp: () => createRsvp2,
  deleteRsvp: () => deleteRsvp2,
  deleteRsvpCheckIn: () => deleteRsvpCheckIn2,
  getRsvp: () => getRsvp2,
  listRsvp: () => listRsvp2,
  queryRsvp: () => queryRsvp2,
  updateRsvp: () => updateRsvp2
});
module.exports = __toCommonJS(index_typings_exports);

// src/events-v1-rsvp-rsvp.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/events-v1-rsvp-rsvp.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEventsRsvpRsvpManagementUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/events/v1/rsvp",
        destPath: "/v1/rsvp"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/events/v1/rsvp",
        destPath: "/v1/rsvp"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/events-rsvp-management-v1-proxy",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/wix-events-web/v1/rsvp",
        destPath: "/v1/rsvp"
      }
    ],
    _: [
      {
        srcPath: "/_api/wix-events-web/v1/rsvp",
        destPath: "/v1/rsvp"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/wix-events-web/v1/rsvp",
        destPath: "/v1/rsvp"
      }
    ],
    "api._base_domain_": [
      {
        srcPath: "/events/v1/rsvp",
        destPath: "/v1/rsvp"
      }
    ],
    "editor-flow.wixapps.net": [
      {
        srcPath: "/_api/wix-events-web/v1/rsvp",
        destPath: "/v1/rsvp"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/wix-events-web/v1/rsvp",
        destPath: "/v1/rsvp"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/wix-events-web/v1/rsvp",
        destPath: "/v1/rsvp"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/wix-events-web/v1/rsvp",
        destPath: "/v1/rsvp"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/wix-events-web/v1/rsvp",
        destPath: "/v1/rsvp"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/wix-events-web/v1/rsvp",
        destPath: "/v1/rsvp"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_events_rsvp";
function listRsvp(payload) {
  function __listRsvp({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.rsvp",
      method: "GET",
      methodFqn: "wix.events.rsvp.RsvpManagement.ListRsvp",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsRsvpRsvpManagementUrl({
        protoPath: "/v1/rsvp",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "rsvps.created" },
            { path: "rsvps.modified" },
            { path: "rsvps.rsvpForm.inputValues.dateTime" },
            { path: "rsvps.guests.checkIn.created" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "rsvps.rsvpForm.inputValues.number" },
            {
              path: "rsvps.rsvpForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "rsvps.rsvpForm.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listRsvp;
}
function queryRsvp(payload) {
  function __queryRsvp({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.rsvp",
      method: "POST",
      methodFqn: "wix.events.rsvp.RsvpManagement.QueryRsvp",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsRsvpRsvpManagementUrl({
        protoPath: "/v1/rsvp/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "rsvps.created" },
            { path: "rsvps.modified" },
            { path: "rsvps.rsvpForm.inputValues.dateTime" },
            { path: "rsvps.guests.checkIn.created" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "rsvps.rsvpForm.inputValues.number" },
            {
              path: "rsvps.rsvpForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "rsvps.rsvpForm.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryRsvp;
}
function getRsvp(payload) {
  function __getRsvp({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.rsvp",
      method: "GET",
      methodFqn: "wix.events.rsvp.RsvpManagement.GetRsvp",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsRsvpRsvpManagementUrl({
        protoPath: "/v1/rsvp/{rsvpId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "rsvp.created" },
            { path: "rsvp.modified" },
            { path: "rsvp.rsvpForm.inputValues.dateTime" },
            { path: "rsvp.guests.checkIn.created" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "rsvp.rsvpForm.inputValues.number" },
            {
              path: "rsvp.rsvpForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "rsvp.rsvpForm.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getRsvp;
}
function createRsvp(payload) {
  function __createRsvp({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "form.inputValues.number" },
          { path: "form.inputValues.address.address.geocode.latitude" },
          { path: "form.inputValues.address.address.geocode.longitude" }
        ]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "form.inputValues.dateTime" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v1.rsvp",
      method: "POST",
      methodFqn: "wix.events.rsvp.RsvpManagement.CreateRsvp",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsRsvpRsvpManagementUrl({
        protoPath: "/v1/rsvp",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "rsvp.created" },
            { path: "rsvp.modified" },
            { path: "rsvp.rsvpForm.inputValues.dateTime" },
            { path: "rsvp.guests.checkIn.created" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "rsvp.rsvpForm.inputValues.number" },
            {
              path: "rsvp.rsvpForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "rsvp.rsvpForm.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createRsvp;
}
function updateRsvp(payload) {
  function __updateRsvp({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fields" }]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "rsvpForm.inputValues.number" },
          { path: "rsvpForm.inputValues.address.address.geocode.latitude" },
          { path: "rsvpForm.inputValues.address.address.geocode.longitude" }
        ]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "rsvpForm.inputValues.dateTime" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v1.rsvp",
      method: "PATCH",
      methodFqn: "wix.events.rsvp.RsvpManagement.UpdateRsvp",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsRsvpRsvpManagementUrl({
        protoPath: "/v1/rsvp/{rsvpId}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "rsvp.created" },
            { path: "rsvp.modified" },
            { path: "rsvp.rsvpForm.inputValues.dateTime" },
            { path: "rsvp.guests.checkIn.created" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "rsvp.rsvpForm.inputValues.number" },
            {
              path: "rsvp.rsvpForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "rsvp.rsvpForm.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateRsvp;
}
function bulkUpdateRsvp(payload) {
  function __bulkUpdateRsvp({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fields" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v1.rsvp",
      method: "PATCH",
      methodFqn: "wix.events.rsvp.RsvpManagement.BulkUpdateRsvp",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsRsvpRsvpManagementUrl({
        protoPath: "/v1/rsvp",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "rsvps.created" },
            { path: "rsvps.modified" },
            { path: "rsvps.rsvpForm.inputValues.dateTime" },
            { path: "rsvps.guests.checkIn.created" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "rsvps.rsvpForm.inputValues.number" },
            {
              path: "rsvps.rsvpForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "rsvps.rsvpForm.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateRsvp;
}
function deleteRsvp(payload) {
  function __deleteRsvp({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.rsvp",
      method: "DELETE",
      methodFqn: "wix.events.rsvp.RsvpManagement.DeleteRsvp",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsRsvpRsvpManagementUrl({
        protoPath: "/v1/rsvp",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteRsvp;
}
function checkInRsvp(payload) {
  function __checkInRsvp({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.rsvp",
      method: "POST",
      methodFqn: "wix.events.rsvp.RsvpManagement.CheckInRsvp",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsRsvpRsvpManagementUrl({
        protoPath: "/v1/rsvp/check-in",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "rsvp.created" },
            { path: "rsvp.modified" },
            { path: "rsvp.rsvpForm.inputValues.dateTime" },
            { path: "rsvp.guests.checkIn.created" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "rsvp.rsvpForm.inputValues.number" },
            {
              path: "rsvp.rsvpForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "rsvp.rsvpForm.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __checkInRsvp;
}
function deleteRsvpCheckIn(payload) {
  function __deleteRsvpCheckIn({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.rsvp",
      method: "DELETE",
      methodFqn: "wix.events.rsvp.RsvpManagement.DeleteRsvpCheckIn",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsRsvpRsvpManagementUrl({
        protoPath: "/v1/rsvp/check-in",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "rsvp.created" },
            { path: "rsvp.modified" },
            { path: "rsvp.rsvpForm.inputValues.dateTime" },
            { path: "rsvp.guests.checkIn.created" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "rsvp.rsvpForm.inputValues.number" },
            {
              path: "rsvp.rsvpForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "rsvp.rsvpForm.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __deleteRsvpCheckIn;
}

// src/events-v1-rsvp-rsvp.universal.ts
var import_address = require("@wix/sdk-runtime/transformations/address");
var import_address2 = require("@wix/sdk-runtime/transformations/address");
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var SubdivisionType = /* @__PURE__ */ ((SubdivisionType2) => {
  SubdivisionType2["UNKNOWN_SUBDIVISION_TYPE"] = "UNKNOWN_SUBDIVISION_TYPE";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_1"] = "ADMINISTRATIVE_AREA_LEVEL_1";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_2"] = "ADMINISTRATIVE_AREA_LEVEL_2";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_3"] = "ADMINISTRATIVE_AREA_LEVEL_3";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_4"] = "ADMINISTRATIVE_AREA_LEVEL_4";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_5"] = "ADMINISTRATIVE_AREA_LEVEL_5";
  SubdivisionType2["COUNTRY"] = "COUNTRY";
  return SubdivisionType2;
})(SubdivisionType || {});
var RsvpStatus = /* @__PURE__ */ ((RsvpStatus2) => {
  RsvpStatus2["YES"] = "YES";
  RsvpStatus2["NO"] = "NO";
  RsvpStatus2["WAITING"] = "WAITING";
  return RsvpStatus2;
})(RsvpStatus || {});
var RsvpFieldset = /* @__PURE__ */ ((RsvpFieldset2) => {
  RsvpFieldset2["DETAILS"] = "DETAILS";
  RsvpFieldset2["FORM"] = "FORM";
  RsvpFieldset2["CONTACT_DETAILS"] = "CONTACT_DETAILS";
  return RsvpFieldset2;
})(RsvpFieldset || {});
var RsvpTag = /* @__PURE__ */ ((RsvpTag2) => {
  RsvpTag2["FULLY_CHECKED_IN"] = "FULLY_CHECKED_IN";
  RsvpTag2["NOT_FULLY_CHECKED_IN"] = "NOT_FULLY_CHECKED_IN";
  RsvpTag2["MEMBER"] = "MEMBER";
  return RsvpTag2;
})(RsvpTag || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function listRsvp2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    offset: options?.offset,
    limit: options?.limit,
    fieldset: options?.fieldset,
    eventId: options?.eventId,
    rsvpId: options?.rsvpId,
    status: options?.status,
    memberId: options?.memberId,
    facet: options?.facet,
    searchPhrase: options?.searchPhrase,
    eventCreatorId: options?.eventCreatorId,
    sort: options?.sort,
    contactId: options?.contactId,
    tag: options?.tag
  });
  const reqOpts = listRsvp(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [{ path: "rsvps.rsvpForm.inputValues.address.address" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          offset: "$[0].offset",
          limit: "$[0].limit",
          fieldset: "$[0].fieldset",
          eventId: "$[0].eventId",
          rsvpId: "$[0].rsvpId",
          status: "$[0].status",
          memberId: "$[0].memberId",
          facet: "$[0].facet",
          searchPhrase: "$[0].searchPhrase",
          eventCreatorId: "$[0].eventCreatorId",
          sort: "$[0].sort",
          contactId: "$[0].contactId",
          tag: "$[0].tag"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function queryRsvp2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    offset: options?.offset,
    limit: options?.limit,
    fieldset: options?.fieldset,
    filter: options?.filter,
    memberId: options?.memberId,
    facet: options?.facet,
    searchPhrase: options?.searchPhrase,
    eventCreatorId: options?.eventCreatorId,
    sort: options?.sort,
    contactId: options?.contactId,
    tag: options?.tag
  });
  const reqOpts = queryRsvp(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [{ path: "rsvps.rsvpForm.inputValues.address.address" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          offset: "$[0].offset",
          limit: "$[0].limit",
          fieldset: "$[0].fieldset",
          filter: "$[0].filter",
          memberId: "$[0].memberId",
          facet: "$[0].facet",
          searchPhrase: "$[0].searchPhrase",
          eventCreatorId: "$[0].eventCreatorId",
          sort: "$[0].sort",
          contactId: "$[0].contactId",
          tag: "$[0].tag"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getRsvp2(rsvpId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    rsvpId,
    eventId: options?.eventId,
    fieldset: options?.fieldset
  });
  const reqOpts = getRsvp(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [{ path: "rsvp.rsvpForm.inputValues.address.address" }]
        }
      ])
    )?.rsvp;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          rsvpId: "$[0]",
          eventId: "$[1].eventId",
          fieldset: "$[1].fieldset"
        },
        singleArgumentUnchanged: false
      },
      ["rsvpId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function createRsvp2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      eventId: options?.eventId,
      form: options?.form,
      status: options?.status,
      memberId: options?.memberId,
      options: options?.options,
      marketingConsent: options?.marketingConsent
    }),
    [
      {
        transformFn: import_address.transformSDKAddressToRESTAddress,
        paths: [{ path: "form.inputValues.address.address" }]
      }
    ]
  );
  const reqOpts = createRsvp(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [{ path: "rsvp.rsvpForm.inputValues.address.address" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          eventId: "$[0].eventId",
          form: "$[0].form",
          status: "$[0].status",
          memberId: "$[0].memberId",
          options: "$[0].options",
          marketingConsent: "$[0].marketingConsent"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateRsvp2(rsvpId, eventId, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      rsvpId,
      eventId,
      fields: options?.fields,
      rsvpForm: options?.rsvpForm,
      status: options?.status,
      options: options?.options
    }),
    [
      {
        transformFn: import_address.transformSDKAddressToRESTAddress,
        paths: [{ path: "rsvpForm.inputValues.address.address" }]
      }
    ]
  );
  const reqOpts = updateRsvp(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [{ path: "rsvp.rsvpForm.inputValues.address.address" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          rsvpId: "$[0]",
          eventId: "$[1]",
          fields: "$[2].fields",
          rsvpForm: "$[2].rsvpForm",
          status: "$[2].status",
          options: "$[2].options"
        },
        singleArgumentUnchanged: false
      },
      ["rsvpId", "eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateRsvp2(eventId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    eventId,
    rsvpId: options?.rsvpId,
    fields: options?.fields,
    status: options?.status
  });
  const reqOpts = bulkUpdateRsvp(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [{ path: "rsvps.rsvpForm.inputValues.address.address" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          eventId: "$[0]",
          rsvpId: "$[1].rsvpId",
          fields: "$[1].fields",
          status: "$[1].status"
        },
        singleArgumentUnchanged: false
      },
      ["eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteRsvp2(eventId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    eventId,
    rsvps: options?.rsvps
  });
  const reqOpts = deleteRsvp(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { eventId: "$[0]", rsvps: "$[1].rsvps" },
        singleArgumentUnchanged: false
      },
      ["eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function checkInRsvp2(eventId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    eventId,
    rsvpId: options?.rsvpId,
    guestId: options?.guestId
  });
  const reqOpts = checkInRsvp(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [{ path: "rsvp.rsvpForm.inputValues.address.address" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          eventId: "$[0]",
          rsvpId: "$[1].rsvpId",
          guestId: "$[1].guestId"
        },
        singleArgumentUnchanged: false
      },
      ["eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteRsvpCheckIn2(eventId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    eventId,
    rsvpId: options?.rsvpId,
    guestId: options?.guestId
  });
  const reqOpts = deleteRsvpCheckIn(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [{ path: "rsvp.rsvpForm.inputValues.address.address" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          eventId: "$[0]",
          rsvpId: "$[1].rsvpId",
          guestId: "$[1].guestId"
        },
        singleArgumentUnchanged: false
      },
      ["eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  RsvpFieldset,
  RsvpStatus,
  RsvpTag,
  SubdivisionType,
  WebhookIdentityType,
  bulkUpdateRsvp,
  checkInRsvp,
  createRsvp,
  deleteRsvp,
  deleteRsvpCheckIn,
  getRsvp,
  listRsvp,
  queryRsvp,
  updateRsvp
});
