// @wix/auto_sdk_events_rsvp/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  RsvpFieldsetOriginal: () => RsvpFieldset,
  RsvpStatusOriginal: () => RsvpStatus,
  RsvpTagOriginal: () => RsvpTag,
  SubdivisionTypeOriginal: () => SubdivisionType,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  bulkUpdateRsvp: () => bulkUpdateRsvp2,
  checkInRsvp: () => checkInRsvp2,
  createRsvp: () => createRsvp2,
  deleteRsvp: () => deleteRsvp2,
  deleteRsvpCheckIn: () => deleteRsvpCheckIn2,
  getRsvp: () => getRsvp2,
  listRsvp: () => listRsvp2,
  queryRsvp: () => queryRsvp2,
  updateRsvp: () => updateRsvp2
});
module.exports = __toCommonJS(meta_exports);

// src/events-v1-rsvp-rsvp.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEventsRsvpRsvpManagementUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/events/v1/rsvp",
        destPath: "/v1/rsvp"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/events/v1/rsvp",
        destPath: "/v1/rsvp"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/events-rsvp-management-v1-proxy",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/wix-events-web/v1/rsvp",
        destPath: "/v1/rsvp"
      }
    ],
    _: [
      {
        srcPath: "/_api/wix-events-web/v1/rsvp",
        destPath: "/v1/rsvp"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/wix-events-web/v1/rsvp",
        destPath: "/v1/rsvp"
      }
    ],
    "api._base_domain_": [
      {
        srcPath: "/events/v1/rsvp",
        destPath: "/v1/rsvp"
      }
    ],
    "editor-flow.wixapps.net": [
      {
        srcPath: "/_api/wix-events-web/v1/rsvp",
        destPath: "/v1/rsvp"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/wix-events-web/v1/rsvp",
        destPath: "/v1/rsvp"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/wix-events-web/v1/rsvp",
        destPath: "/v1/rsvp"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/wix-events-web/v1/rsvp",
        destPath: "/v1/rsvp"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/wix-events-web/v1/rsvp",
        destPath: "/v1/rsvp"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/wix-events-web/v1/rsvp",
        destPath: "/v1/rsvp"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_events_rsvp";
function listRsvp(payload) {
  function __listRsvp({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.rsvp",
      method: "GET",
      methodFqn: "wix.events.rsvp.RsvpManagement.ListRsvp",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsRsvpRsvpManagementUrl({
        protoPath: "/v1/rsvp",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "rsvps.created" },
            { path: "rsvps.modified" },
            { path: "rsvps.rsvpForm.inputValues.dateTime" },
            { path: "rsvps.guests.checkIn.created" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "rsvps.rsvpForm.inputValues.number" },
            {
              path: "rsvps.rsvpForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "rsvps.rsvpForm.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listRsvp;
}
function queryRsvp(payload) {
  function __queryRsvp({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.rsvp",
      method: "POST",
      methodFqn: "wix.events.rsvp.RsvpManagement.QueryRsvp",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsRsvpRsvpManagementUrl({
        protoPath: "/v1/rsvp/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "rsvps.created" },
            { path: "rsvps.modified" },
            { path: "rsvps.rsvpForm.inputValues.dateTime" },
            { path: "rsvps.guests.checkIn.created" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "rsvps.rsvpForm.inputValues.number" },
            {
              path: "rsvps.rsvpForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "rsvps.rsvpForm.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryRsvp;
}
function getRsvp(payload) {
  function __getRsvp({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.rsvp",
      method: "GET",
      methodFqn: "wix.events.rsvp.RsvpManagement.GetRsvp",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsRsvpRsvpManagementUrl({
        protoPath: "/v1/rsvp/{rsvpId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "rsvp.created" },
            { path: "rsvp.modified" },
            { path: "rsvp.rsvpForm.inputValues.dateTime" },
            { path: "rsvp.guests.checkIn.created" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "rsvp.rsvpForm.inputValues.number" },
            {
              path: "rsvp.rsvpForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "rsvp.rsvpForm.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getRsvp;
}
function createRsvp(payload) {
  function __createRsvp({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "form.inputValues.number" },
          { path: "form.inputValues.address.address.geocode.latitude" },
          { path: "form.inputValues.address.address.geocode.longitude" }
        ]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "form.inputValues.dateTime" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v1.rsvp",
      method: "POST",
      methodFqn: "wix.events.rsvp.RsvpManagement.CreateRsvp",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsRsvpRsvpManagementUrl({
        protoPath: "/v1/rsvp",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "rsvp.created" },
            { path: "rsvp.modified" },
            { path: "rsvp.rsvpForm.inputValues.dateTime" },
            { path: "rsvp.guests.checkIn.created" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "rsvp.rsvpForm.inputValues.number" },
            {
              path: "rsvp.rsvpForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "rsvp.rsvpForm.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createRsvp;
}
function updateRsvp(payload) {
  function __updateRsvp({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fields" }]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "rsvpForm.inputValues.number" },
          { path: "rsvpForm.inputValues.address.address.geocode.latitude" },
          { path: "rsvpForm.inputValues.address.address.geocode.longitude" }
        ]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "rsvpForm.inputValues.dateTime" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v1.rsvp",
      method: "PATCH",
      methodFqn: "wix.events.rsvp.RsvpManagement.UpdateRsvp",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsRsvpRsvpManagementUrl({
        protoPath: "/v1/rsvp/{rsvpId}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "rsvp.created" },
            { path: "rsvp.modified" },
            { path: "rsvp.rsvpForm.inputValues.dateTime" },
            { path: "rsvp.guests.checkIn.created" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "rsvp.rsvpForm.inputValues.number" },
            {
              path: "rsvp.rsvpForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "rsvp.rsvpForm.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateRsvp;
}
function bulkUpdateRsvp(payload) {
  function __bulkUpdateRsvp({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fields" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v1.rsvp",
      method: "PATCH",
      methodFqn: "wix.events.rsvp.RsvpManagement.BulkUpdateRsvp",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsRsvpRsvpManagementUrl({
        protoPath: "/v1/rsvp",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "rsvps.created" },
            { path: "rsvps.modified" },
            { path: "rsvps.rsvpForm.inputValues.dateTime" },
            { path: "rsvps.guests.checkIn.created" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "rsvps.rsvpForm.inputValues.number" },
            {
              path: "rsvps.rsvpForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "rsvps.rsvpForm.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateRsvp;
}
function deleteRsvp(payload) {
  function __deleteRsvp({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.rsvp",
      method: "DELETE",
      methodFqn: "wix.events.rsvp.RsvpManagement.DeleteRsvp",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsRsvpRsvpManagementUrl({
        protoPath: "/v1/rsvp",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteRsvp;
}
function checkInRsvp(payload) {
  function __checkInRsvp({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.rsvp",
      method: "POST",
      methodFqn: "wix.events.rsvp.RsvpManagement.CheckInRsvp",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsRsvpRsvpManagementUrl({
        protoPath: "/v1/rsvp/check-in",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "rsvp.created" },
            { path: "rsvp.modified" },
            { path: "rsvp.rsvpForm.inputValues.dateTime" },
            { path: "rsvp.guests.checkIn.created" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "rsvp.rsvpForm.inputValues.number" },
            {
              path: "rsvp.rsvpForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "rsvp.rsvpForm.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __checkInRsvp;
}
function deleteRsvpCheckIn(payload) {
  function __deleteRsvpCheckIn({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.rsvp",
      method: "DELETE",
      methodFqn: "wix.events.rsvp.RsvpManagement.DeleteRsvpCheckIn",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsRsvpRsvpManagementUrl({
        protoPath: "/v1/rsvp/check-in",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "rsvp.created" },
            { path: "rsvp.modified" },
            { path: "rsvp.rsvpForm.inputValues.dateTime" },
            { path: "rsvp.guests.checkIn.created" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "rsvp.rsvpForm.inputValues.number" },
            {
              path: "rsvp.rsvpForm.inputValues.address.address.geocode.latitude"
            },
            {
              path: "rsvp.rsvpForm.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __deleteRsvpCheckIn;
}

// src/events-v1-rsvp-rsvp.types.ts
var SubdivisionType = /* @__PURE__ */ ((SubdivisionType2) => {
  SubdivisionType2["UNKNOWN_SUBDIVISION_TYPE"] = "UNKNOWN_SUBDIVISION_TYPE";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_1"] = "ADMINISTRATIVE_AREA_LEVEL_1";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_2"] = "ADMINISTRATIVE_AREA_LEVEL_2";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_3"] = "ADMINISTRATIVE_AREA_LEVEL_3";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_4"] = "ADMINISTRATIVE_AREA_LEVEL_4";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_5"] = "ADMINISTRATIVE_AREA_LEVEL_5";
  SubdivisionType2["COUNTRY"] = "COUNTRY";
  return SubdivisionType2;
})(SubdivisionType || {});
var RsvpStatus = /* @__PURE__ */ ((RsvpStatus2) => {
  RsvpStatus2["YES"] = "YES";
  RsvpStatus2["NO"] = "NO";
  RsvpStatus2["WAITING"] = "WAITING";
  return RsvpStatus2;
})(RsvpStatus || {});
var RsvpFieldset = /* @__PURE__ */ ((RsvpFieldset2) => {
  RsvpFieldset2["DETAILS"] = "DETAILS";
  RsvpFieldset2["FORM"] = "FORM";
  RsvpFieldset2["CONTACT_DETAILS"] = "CONTACT_DETAILS";
  return RsvpFieldset2;
})(RsvpFieldset || {});
var RsvpTag = /* @__PURE__ */ ((RsvpTag2) => {
  RsvpTag2["FULLY_CHECKED_IN"] = "FULLY_CHECKED_IN";
  RsvpTag2["NOT_FULLY_CHECKED_IN"] = "NOT_FULLY_CHECKED_IN";
  RsvpTag2["MEMBER"] = "MEMBER";
  return RsvpTag2;
})(RsvpTag || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/events-v1-rsvp-rsvp.meta.ts
function listRsvp2() {
  const payload = {};
  const getRequestOptions = listRsvp(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/rsvp",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryRsvp2() {
  const payload = {};
  const getRequestOptions = queryRsvp(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/rsvp/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getRsvp2() {
  const payload = { rsvpId: ":rsvpId" };
  const getRequestOptions = getRsvp(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/rsvp/{rsvpId}",
    pathParams: { rsvpId: "rsvpId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createRsvp2() {
  const payload = {};
  const getRequestOptions = createRsvp(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/rsvp",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateRsvp2() {
  const payload = { rsvpId: ":rsvpId" };
  const getRequestOptions = updateRsvp(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/rsvp/{rsvpId}",
    pathParams: { rsvpId: "rsvpId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateRsvp2() {
  const payload = {};
  const getRequestOptions = bulkUpdateRsvp(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/rsvp",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteRsvp2() {
  const payload = {};
  const getRequestOptions = deleteRsvp(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/rsvp",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function checkInRsvp2() {
  const payload = {};
  const getRequestOptions = checkInRsvp(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/rsvp/check-in",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteRsvpCheckIn2() {
  const payload = {};
  const getRequestOptions = deleteRsvpCheckIn(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/rsvp/check-in",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  RsvpFieldsetOriginal,
  RsvpStatusOriginal,
  RsvpTagOriginal,
  SubdivisionTypeOriginal,
  WebhookIdentityTypeOriginal,
  bulkUpdateRsvp,
  checkInRsvp,
  createRsvp,
  deleteRsvp,
  deleteRsvpCheckIn,
  getRsvp,
  listRsvp,
  queryRsvp,
  updateRsvp
});
