// @wix/auto_sdk_media_files/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  bulkDeleteFiles: () => bulkDeleteFiles2,
  bulkImportFile: () => bulkImportFile2,
  bulkImportFiles: () => bulkImportFiles2,
  bulkRestoreFilesFromTrashBin: () => bulkRestoreFilesFromTrashBin2,
  generateFileDownloadUrl: () => generateFileDownloadUrl2,
  generateFileResumableUploadUrl: () => generateFileResumableUploadUrl2,
  generateFileUploadUrl: () => generateFileUploadUrl2,
  generateFilesDownloadUrl: () => generateFilesDownloadUrl2,
  generateVideoStreamingUrl: () => generateVideoStreamingUrl2,
  getFileDescriptor: () => getFileDescriptor2,
  getFileDescriptors: () => getFileDescriptors2,
  importFile: () => importFile2,
  listDeletedFiles: () => listDeletedFiles2,
  listFiles: () => listFiles2,
  searchFiles: () => searchFiles2,
  updateFileDescriptor: () => updateFileDescriptor2
});
module.exports = __toCommonJS(meta_exports);

// src/media-site-media-v1-file-descriptor-files.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixMediaSiteMediaV1FilesServiceUrl(opts) {
  const domainToMappings = {
    "apps._base_domain_": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: "/api"
      },
      {
        srcPath: "/_api/wix-privatemedia-proxy-server/v1",
        destPath: "/v1"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/site-media/v1",
        destPath: "/v1"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "platform.rise.ai": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_media_files";
function generateFilesDownloadUrl(payload) {
  function __generateFilesDownloadUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FilesService.GenerateFilesDownloadUrl",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/files/generate-download-url",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __generateFilesDownloadUrl;
}
function generateFileDownloadUrl(payload) {
  function __generateFileDownloadUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FilesService.GenerateFileDownloadUrl",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/files/generate-file-download-url",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __generateFileDownloadUrl;
}
function getFileDescriptor(payload) {
  function __getFileDescriptor({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "GET",
      methodFqn: "com.wix.media.site_media.v1.FilesService.GetFileDescriptor",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/files/get-file-by-id",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "file.createdDate" },
            { path: "file.updatedDate" },
            { path: "file.lastUsedDate" },
            { path: "file.media.image.image.urlExpirationDate" },
            { path: "file.media.image.previewImage.urlExpirationDate" },
            { path: "file.media.video.urlExpirationDate" },
            { path: "file.media.video.resolutions.urlExpirationDate" },
            { path: "file.media.video.resolutions.poster.urlExpirationDate" },
            { path: "file.media.video.posters.urlExpirationDate" },
            { path: "file.media.audio.assets.urlExpirationDate" },
            { path: "file.media.document.urlExpirationDate" },
            { path: "file.media.document.thumbnail.urlExpirationDate" },
            { path: "file.media.vector.image.urlExpirationDate" },
            { path: "file.media.vector.previewImage.urlExpirationDate" },
            { path: "file.media.archive.urlExpirationDate" },
            { path: "file.media.model3d.urlExpirationDate" },
            { path: "file.media.model3d.thumbnail.urlExpirationDate" },
            { path: "file.media.icon.image.urlExpirationDate" },
            { path: "file.media.icon.previewImage.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "file.media.image.image.focalPoint.x" },
            { path: "file.media.image.image.focalPoint.y" },
            { path: "file.media.image.faces.confidence" },
            { path: "file.media.image.previewImage.focalPoint.x" },
            { path: "file.media.image.previewImage.focalPoint.y" },
            { path: "file.media.video.resolutions.poster.focalPoint.x" },
            { path: "file.media.video.resolutions.poster.focalPoint.y" },
            { path: "file.media.video.posters.focalPoint.x" },
            { path: "file.media.video.posters.focalPoint.y" },
            { path: "file.media.document.thumbnail.focalPoint.x" },
            { path: "file.media.document.thumbnail.focalPoint.y" },
            { path: "file.media.vector.image.focalPoint.x" },
            { path: "file.media.vector.image.focalPoint.y" },
            { path: "file.media.vector.faces.confidence" },
            { path: "file.media.vector.previewImage.focalPoint.x" },
            { path: "file.media.vector.previewImage.focalPoint.y" },
            { path: "file.media.model3d.thumbnail.focalPoint.x" },
            { path: "file.media.model3d.thumbnail.focalPoint.y" },
            { path: "file.media.icon.image.focalPoint.x" },
            { path: "file.media.icon.image.focalPoint.y" },
            { path: "file.media.icon.faces.confidence" },
            { path: "file.media.icon.previewImage.focalPoint.x" },
            { path: "file.media.icon.previewImage.focalPoint.y" }
          ]
        }
      ]),
      fallback: [
        {
          method: "GET",
          url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
            protoPath: "/v1/files/get-file-by-id",
            data: payload,
            host
          }),
          params: (0, import_rest_modules.toURLSearchParams)(payload)
        }
      ]
    };
    return metadata;
  }
  return __getFileDescriptor;
}
function getFileDescriptors(payload) {
  function __getFileDescriptors({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FilesService.GetFileDescriptors",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/files/get-files",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "files.createdDate" },
            { path: "files.updatedDate" },
            { path: "files.lastUsedDate" },
            { path: "files.media.image.image.urlExpirationDate" },
            { path: "files.media.image.previewImage.urlExpirationDate" },
            { path: "files.media.video.urlExpirationDate" },
            { path: "files.media.video.resolutions.urlExpirationDate" },
            {
              path: "files.media.video.resolutions.poster.urlExpirationDate"
            },
            { path: "files.media.video.posters.urlExpirationDate" },
            { path: "files.media.audio.assets.urlExpirationDate" },
            { path: "files.media.document.urlExpirationDate" },
            { path: "files.media.document.thumbnail.urlExpirationDate" },
            { path: "files.media.vector.image.urlExpirationDate" },
            { path: "files.media.vector.previewImage.urlExpirationDate" },
            { path: "files.media.archive.urlExpirationDate" },
            { path: "files.media.model3d.urlExpirationDate" },
            { path: "files.media.model3d.thumbnail.urlExpirationDate" },
            { path: "files.media.icon.image.urlExpirationDate" },
            { path: "files.media.icon.previewImage.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "files.media.image.image.focalPoint.x" },
            { path: "files.media.image.image.focalPoint.y" },
            { path: "files.media.image.faces.confidence" },
            { path: "files.media.image.previewImage.focalPoint.x" },
            { path: "files.media.image.previewImage.focalPoint.y" },
            { path: "files.media.video.resolutions.poster.focalPoint.x" },
            { path: "files.media.video.resolutions.poster.focalPoint.y" },
            { path: "files.media.video.posters.focalPoint.x" },
            { path: "files.media.video.posters.focalPoint.y" },
            { path: "files.media.document.thumbnail.focalPoint.x" },
            { path: "files.media.document.thumbnail.focalPoint.y" },
            { path: "files.media.vector.image.focalPoint.x" },
            { path: "files.media.vector.image.focalPoint.y" },
            { path: "files.media.vector.faces.confidence" },
            { path: "files.media.vector.previewImage.focalPoint.x" },
            { path: "files.media.vector.previewImage.focalPoint.y" },
            { path: "files.media.model3d.thumbnail.focalPoint.x" },
            { path: "files.media.model3d.thumbnail.focalPoint.y" },
            { path: "files.media.icon.image.focalPoint.x" },
            { path: "files.media.icon.image.focalPoint.y" },
            { path: "files.media.icon.faces.confidence" },
            { path: "files.media.icon.previewImage.focalPoint.x" },
            { path: "files.media.icon.previewImage.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getFileDescriptors;
}
function updateFileDescriptor(payload) {
  function __updateFileDescriptor({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "file.createdDate" },
          { path: "file.updatedDate" },
          { path: "file.lastUsedDate" },
          { path: "file.media.image.image.urlExpirationDate" },
          { path: "file.media.image.previewImage.urlExpirationDate" },
          { path: "file.media.video.urlExpirationDate" },
          { path: "file.media.video.resolutions.urlExpirationDate" },
          { path: "file.media.video.resolutions.poster.urlExpirationDate" },
          { path: "file.media.video.posters.urlExpirationDate" },
          { path: "file.media.audio.assets.urlExpirationDate" },
          { path: "file.media.document.urlExpirationDate" },
          { path: "file.media.document.thumbnail.urlExpirationDate" },
          { path: "file.media.vector.image.urlExpirationDate" },
          { path: "file.media.vector.previewImage.urlExpirationDate" },
          { path: "file.media.archive.urlExpirationDate" },
          { path: "file.media.model3d.urlExpirationDate" },
          { path: "file.media.model3d.thumbnail.urlExpirationDate" },
          { path: "file.media.icon.image.urlExpirationDate" },
          { path: "file.media.icon.previewImage.urlExpirationDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "file.media.image.image.focalPoint.x" },
          { path: "file.media.image.image.focalPoint.y" },
          { path: "file.media.image.faces.confidence" },
          { path: "file.media.image.previewImage.focalPoint.x" },
          { path: "file.media.image.previewImage.focalPoint.y" },
          { path: "file.media.video.resolutions.poster.focalPoint.x" },
          { path: "file.media.video.resolutions.poster.focalPoint.y" },
          { path: "file.media.video.posters.focalPoint.x" },
          { path: "file.media.video.posters.focalPoint.y" },
          { path: "file.media.document.thumbnail.focalPoint.x" },
          { path: "file.media.document.thumbnail.focalPoint.y" },
          { path: "file.media.vector.image.focalPoint.x" },
          { path: "file.media.vector.image.focalPoint.y" },
          { path: "file.media.vector.faces.confidence" },
          { path: "file.media.vector.previewImage.focalPoint.x" },
          { path: "file.media.vector.previewImage.focalPoint.y" },
          { path: "file.media.model3d.thumbnail.focalPoint.x" },
          { path: "file.media.model3d.thumbnail.focalPoint.y" },
          { path: "file.media.icon.image.focalPoint.x" },
          { path: "file.media.icon.image.focalPoint.y" },
          { path: "file.media.icon.faces.confidence" },
          { path: "file.media.icon.previewImage.focalPoint.x" },
          { path: "file.media.icon.previewImage.focalPoint.y" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "PATCH",
      methodFqn: "com.wix.media.site_media.v1.FilesService.UpdateFileDescriptor",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/files/update-file-descriptor",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "file.createdDate" },
            { path: "file.updatedDate" },
            { path: "file.lastUsedDate" },
            { path: "file.media.image.image.urlExpirationDate" },
            { path: "file.media.image.previewImage.urlExpirationDate" },
            { path: "file.media.video.urlExpirationDate" },
            { path: "file.media.video.resolutions.urlExpirationDate" },
            { path: "file.media.video.resolutions.poster.urlExpirationDate" },
            { path: "file.media.video.posters.urlExpirationDate" },
            { path: "file.media.audio.assets.urlExpirationDate" },
            { path: "file.media.document.urlExpirationDate" },
            { path: "file.media.document.thumbnail.urlExpirationDate" },
            { path: "file.media.vector.image.urlExpirationDate" },
            { path: "file.media.vector.previewImage.urlExpirationDate" },
            { path: "file.media.archive.urlExpirationDate" },
            { path: "file.media.model3d.urlExpirationDate" },
            { path: "file.media.model3d.thumbnail.urlExpirationDate" },
            { path: "file.media.icon.image.urlExpirationDate" },
            { path: "file.media.icon.previewImage.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "file.media.image.image.focalPoint.x" },
            { path: "file.media.image.image.focalPoint.y" },
            { path: "file.media.image.faces.confidence" },
            { path: "file.media.image.previewImage.focalPoint.x" },
            { path: "file.media.image.previewImage.focalPoint.y" },
            { path: "file.media.video.resolutions.poster.focalPoint.x" },
            { path: "file.media.video.resolutions.poster.focalPoint.y" },
            { path: "file.media.video.posters.focalPoint.x" },
            { path: "file.media.video.posters.focalPoint.y" },
            { path: "file.media.document.thumbnail.focalPoint.x" },
            { path: "file.media.document.thumbnail.focalPoint.y" },
            { path: "file.media.vector.image.focalPoint.x" },
            { path: "file.media.vector.image.focalPoint.y" },
            { path: "file.media.vector.faces.confidence" },
            { path: "file.media.vector.previewImage.focalPoint.x" },
            { path: "file.media.vector.previewImage.focalPoint.y" },
            { path: "file.media.model3d.thumbnail.focalPoint.x" },
            { path: "file.media.model3d.thumbnail.focalPoint.y" },
            { path: "file.media.icon.image.focalPoint.x" },
            { path: "file.media.icon.image.focalPoint.y" },
            { path: "file.media.icon.faces.confidence" },
            { path: "file.media.icon.previewImage.focalPoint.x" },
            { path: "file.media.icon.previewImage.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateFileDescriptor;
}
function generateFileUploadUrl(payload) {
  function __generateFileUploadUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FilesService.GenerateFileUploadUrl",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/files/generate-upload-url",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __generateFileUploadUrl;
}
function generateFileResumableUploadUrl(payload) {
  function __generateFileResumableUploadUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FilesService.GenerateFileResumableUploadUrl",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/files/generate-resumable-upload-url",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __generateFileResumableUploadUrl;
}
function importFile(payload) {
  function __importFile({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FilesService.ImportFile",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/files/import",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "file.createdDate" },
            { path: "file.updatedDate" },
            { path: "file.lastUsedDate" },
            { path: "file.media.image.image.urlExpirationDate" },
            { path: "file.media.image.previewImage.urlExpirationDate" },
            { path: "file.media.video.urlExpirationDate" },
            { path: "file.media.video.resolutions.urlExpirationDate" },
            { path: "file.media.video.resolutions.poster.urlExpirationDate" },
            { path: "file.media.video.posters.urlExpirationDate" },
            { path: "file.media.audio.assets.urlExpirationDate" },
            { path: "file.media.document.urlExpirationDate" },
            { path: "file.media.document.thumbnail.urlExpirationDate" },
            { path: "file.media.vector.image.urlExpirationDate" },
            { path: "file.media.vector.previewImage.urlExpirationDate" },
            { path: "file.media.archive.urlExpirationDate" },
            { path: "file.media.model3d.urlExpirationDate" },
            { path: "file.media.model3d.thumbnail.urlExpirationDate" },
            { path: "file.media.icon.image.urlExpirationDate" },
            { path: "file.media.icon.previewImage.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "file.media.image.image.focalPoint.x" },
            { path: "file.media.image.image.focalPoint.y" },
            { path: "file.media.image.faces.confidence" },
            { path: "file.media.image.previewImage.focalPoint.x" },
            { path: "file.media.image.previewImage.focalPoint.y" },
            { path: "file.media.video.resolutions.poster.focalPoint.x" },
            { path: "file.media.video.resolutions.poster.focalPoint.y" },
            { path: "file.media.video.posters.focalPoint.x" },
            { path: "file.media.video.posters.focalPoint.y" },
            { path: "file.media.document.thumbnail.focalPoint.x" },
            { path: "file.media.document.thumbnail.focalPoint.y" },
            { path: "file.media.vector.image.focalPoint.x" },
            { path: "file.media.vector.image.focalPoint.y" },
            { path: "file.media.vector.faces.confidence" },
            { path: "file.media.vector.previewImage.focalPoint.x" },
            { path: "file.media.vector.previewImage.focalPoint.y" },
            { path: "file.media.model3d.thumbnail.focalPoint.x" },
            { path: "file.media.model3d.thumbnail.focalPoint.y" },
            { path: "file.media.icon.image.focalPoint.x" },
            { path: "file.media.icon.image.focalPoint.y" },
            { path: "file.media.icon.faces.confidence" },
            { path: "file.media.icon.previewImage.focalPoint.x" },
            { path: "file.media.icon.previewImage.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __importFile;
}
function bulkImportFiles(payload) {
  function __bulkImportFiles({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FilesService.BulkImportFiles",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/bulk/files/import",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "files.createdDate" },
            { path: "files.updatedDate" },
            { path: "files.lastUsedDate" },
            { path: "files.media.image.image.urlExpirationDate" },
            { path: "files.media.image.previewImage.urlExpirationDate" },
            { path: "files.media.video.urlExpirationDate" },
            { path: "files.media.video.resolutions.urlExpirationDate" },
            {
              path: "files.media.video.resolutions.poster.urlExpirationDate"
            },
            { path: "files.media.video.posters.urlExpirationDate" },
            { path: "files.media.audio.assets.urlExpirationDate" },
            { path: "files.media.document.urlExpirationDate" },
            { path: "files.media.document.thumbnail.urlExpirationDate" },
            { path: "files.media.vector.image.urlExpirationDate" },
            { path: "files.media.vector.previewImage.urlExpirationDate" },
            { path: "files.media.archive.urlExpirationDate" },
            { path: "files.media.model3d.urlExpirationDate" },
            { path: "files.media.model3d.thumbnail.urlExpirationDate" },
            { path: "files.media.icon.image.urlExpirationDate" },
            { path: "files.media.icon.previewImage.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "files.media.image.image.focalPoint.x" },
            { path: "files.media.image.image.focalPoint.y" },
            { path: "files.media.image.faces.confidence" },
            { path: "files.media.image.previewImage.focalPoint.x" },
            { path: "files.media.image.previewImage.focalPoint.y" },
            { path: "files.media.video.resolutions.poster.focalPoint.x" },
            { path: "files.media.video.resolutions.poster.focalPoint.y" },
            { path: "files.media.video.posters.focalPoint.x" },
            { path: "files.media.video.posters.focalPoint.y" },
            { path: "files.media.document.thumbnail.focalPoint.x" },
            { path: "files.media.document.thumbnail.focalPoint.y" },
            { path: "files.media.vector.image.focalPoint.x" },
            { path: "files.media.vector.image.focalPoint.y" },
            { path: "files.media.vector.faces.confidence" },
            { path: "files.media.vector.previewImage.focalPoint.x" },
            { path: "files.media.vector.previewImage.focalPoint.y" },
            { path: "files.media.model3d.thumbnail.focalPoint.x" },
            { path: "files.media.model3d.thumbnail.focalPoint.y" },
            { path: "files.media.icon.image.focalPoint.x" },
            { path: "files.media.icon.image.focalPoint.y" },
            { path: "files.media.icon.faces.confidence" },
            { path: "files.media.icon.previewImage.focalPoint.x" },
            { path: "files.media.icon.previewImage.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkImportFiles;
}
function bulkImportFile(payload) {
  function __bulkImportFile({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FilesService.BulkImportFile",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/bulk/files/import-v2",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.lastUsedDate" },
            { path: "results.item.media.image.image.urlExpirationDate" },
            {
              path: "results.item.media.image.previewImage.urlExpirationDate"
            },
            { path: "results.item.media.video.urlExpirationDate" },
            {
              path: "results.item.media.video.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.media.video.resolutions.poster.urlExpirationDate"
            },
            { path: "results.item.media.video.posters.urlExpirationDate" },
            { path: "results.item.media.audio.assets.urlExpirationDate" },
            { path: "results.item.media.document.urlExpirationDate" },
            {
              path: "results.item.media.document.thumbnail.urlExpirationDate"
            },
            { path: "results.item.media.vector.image.urlExpirationDate" },
            {
              path: "results.item.media.vector.previewImage.urlExpirationDate"
            },
            { path: "results.item.media.archive.urlExpirationDate" },
            { path: "results.item.media.model3d.urlExpirationDate" },
            {
              path: "results.item.media.model3d.thumbnail.urlExpirationDate"
            },
            { path: "results.item.media.icon.image.urlExpirationDate" },
            {
              path: "results.item.media.icon.previewImage.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.media.image.image.focalPoint.x" },
            { path: "results.item.media.image.image.focalPoint.y" },
            { path: "results.item.media.image.faces.confidence" },
            { path: "results.item.media.image.previewImage.focalPoint.x" },
            { path: "results.item.media.image.previewImage.focalPoint.y" },
            {
              path: "results.item.media.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.media.video.resolutions.poster.focalPoint.y"
            },
            { path: "results.item.media.video.posters.focalPoint.x" },
            { path: "results.item.media.video.posters.focalPoint.y" },
            { path: "results.item.media.document.thumbnail.focalPoint.x" },
            { path: "results.item.media.document.thumbnail.focalPoint.y" },
            { path: "results.item.media.vector.image.focalPoint.x" },
            { path: "results.item.media.vector.image.focalPoint.y" },
            { path: "results.item.media.vector.faces.confidence" },
            { path: "results.item.media.vector.previewImage.focalPoint.x" },
            { path: "results.item.media.vector.previewImage.focalPoint.y" },
            { path: "results.item.media.model3d.thumbnail.focalPoint.x" },
            { path: "results.item.media.model3d.thumbnail.focalPoint.y" },
            { path: "results.item.media.icon.image.focalPoint.x" },
            { path: "results.item.media.icon.image.focalPoint.y" },
            { path: "results.item.media.icon.faces.confidence" },
            { path: "results.item.media.icon.previewImage.focalPoint.x" },
            { path: "results.item.media.icon.previewImage.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkImportFile;
}
function listFiles(payload) {
  function __listFiles({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "GET",
      methodFqn: "com.wix.media.site_media.v1.FilesService.ListFiles",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/files",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload, true),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "files.createdDate" },
            { path: "files.updatedDate" },
            { path: "files.lastUsedDate" },
            { path: "files.media.image.image.urlExpirationDate" },
            { path: "files.media.image.previewImage.urlExpirationDate" },
            { path: "files.media.video.urlExpirationDate" },
            { path: "files.media.video.resolutions.urlExpirationDate" },
            {
              path: "files.media.video.resolutions.poster.urlExpirationDate"
            },
            { path: "files.media.video.posters.urlExpirationDate" },
            { path: "files.media.audio.assets.urlExpirationDate" },
            { path: "files.media.document.urlExpirationDate" },
            { path: "files.media.document.thumbnail.urlExpirationDate" },
            { path: "files.media.vector.image.urlExpirationDate" },
            { path: "files.media.vector.previewImage.urlExpirationDate" },
            { path: "files.media.archive.urlExpirationDate" },
            { path: "files.media.model3d.urlExpirationDate" },
            { path: "files.media.model3d.thumbnail.urlExpirationDate" },
            { path: "files.media.icon.image.urlExpirationDate" },
            { path: "files.media.icon.previewImage.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "files.media.image.image.focalPoint.x" },
            { path: "files.media.image.image.focalPoint.y" },
            { path: "files.media.image.faces.confidence" },
            { path: "files.media.image.previewImage.focalPoint.x" },
            { path: "files.media.image.previewImage.focalPoint.y" },
            { path: "files.media.video.resolutions.poster.focalPoint.x" },
            { path: "files.media.video.resolutions.poster.focalPoint.y" },
            { path: "files.media.video.posters.focalPoint.x" },
            { path: "files.media.video.posters.focalPoint.y" },
            { path: "files.media.document.thumbnail.focalPoint.x" },
            { path: "files.media.document.thumbnail.focalPoint.y" },
            { path: "files.media.vector.image.focalPoint.x" },
            { path: "files.media.vector.image.focalPoint.y" },
            { path: "files.media.vector.faces.confidence" },
            { path: "files.media.vector.previewImage.focalPoint.x" },
            { path: "files.media.vector.previewImage.focalPoint.y" },
            { path: "files.media.model3d.thumbnail.focalPoint.x" },
            { path: "files.media.model3d.thumbnail.focalPoint.y" },
            { path: "files.media.icon.image.focalPoint.x" },
            { path: "files.media.icon.image.focalPoint.y" },
            { path: "files.media.icon.faces.confidence" },
            { path: "files.media.icon.previewImage.focalPoint.x" },
            { path: "files.media.icon.previewImage.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listFiles;
}
function searchFiles(payload) {
  function __searchFiles({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FilesService.SearchFiles",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/files/search",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "files.createdDate" },
            { path: "files.updatedDate" },
            { path: "files.lastUsedDate" },
            { path: "files.media.image.image.urlExpirationDate" },
            { path: "files.media.image.previewImage.urlExpirationDate" },
            { path: "files.media.video.urlExpirationDate" },
            { path: "files.media.video.resolutions.urlExpirationDate" },
            {
              path: "files.media.video.resolutions.poster.urlExpirationDate"
            },
            { path: "files.media.video.posters.urlExpirationDate" },
            { path: "files.media.audio.assets.urlExpirationDate" },
            { path: "files.media.document.urlExpirationDate" },
            { path: "files.media.document.thumbnail.urlExpirationDate" },
            { path: "files.media.vector.image.urlExpirationDate" },
            { path: "files.media.vector.previewImage.urlExpirationDate" },
            { path: "files.media.archive.urlExpirationDate" },
            { path: "files.media.model3d.urlExpirationDate" },
            { path: "files.media.model3d.thumbnail.urlExpirationDate" },
            { path: "files.media.icon.image.urlExpirationDate" },
            { path: "files.media.icon.previewImage.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "files.media.image.image.focalPoint.x" },
            { path: "files.media.image.image.focalPoint.y" },
            { path: "files.media.image.faces.confidence" },
            { path: "files.media.image.previewImage.focalPoint.x" },
            { path: "files.media.image.previewImage.focalPoint.y" },
            { path: "files.media.video.resolutions.poster.focalPoint.x" },
            { path: "files.media.video.resolutions.poster.focalPoint.y" },
            { path: "files.media.video.posters.focalPoint.x" },
            { path: "files.media.video.posters.focalPoint.y" },
            { path: "files.media.document.thumbnail.focalPoint.x" },
            { path: "files.media.document.thumbnail.focalPoint.y" },
            { path: "files.media.vector.image.focalPoint.x" },
            { path: "files.media.vector.image.focalPoint.y" },
            { path: "files.media.vector.faces.confidence" },
            { path: "files.media.vector.previewImage.focalPoint.x" },
            { path: "files.media.vector.previewImage.focalPoint.y" },
            { path: "files.media.model3d.thumbnail.focalPoint.x" },
            { path: "files.media.model3d.thumbnail.focalPoint.y" },
            { path: "files.media.icon.image.focalPoint.x" },
            { path: "files.media.icon.image.focalPoint.y" },
            { path: "files.media.icon.faces.confidence" },
            { path: "files.media.icon.previewImage.focalPoint.x" },
            { path: "files.media.icon.previewImage.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __searchFiles;
}
function generateVideoStreamingUrl(payload) {
  function __generateVideoStreamingUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FilesService.GenerateVideoStreamingUrl",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/files/generate-video-stream-url",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __generateVideoStreamingUrl;
}
function bulkDeleteFiles(payload) {
  function __bulkDeleteFiles({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FilesService.BulkDeleteFiles",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/bulk/files/delete",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkDeleteFiles;
}
function bulkRestoreFilesFromTrashBin(payload) {
  function __bulkRestoreFilesFromTrashBin({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FilesService.BulkRestoreFilesFromTrashBin",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/bulk/trash-bin/files/restore",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkRestoreFilesFromTrashBin;
}
function listDeletedFiles(payload) {
  function __listDeletedFiles({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "GET",
      methodFqn: "com.wix.media.site_media.v1.FilesService.ListDeletedFiles",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/trash-bin/files",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload, true),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "files.createdDate" },
            { path: "files.updatedDate" },
            { path: "files.lastUsedDate" },
            { path: "files.media.image.image.urlExpirationDate" },
            { path: "files.media.image.previewImage.urlExpirationDate" },
            { path: "files.media.video.urlExpirationDate" },
            { path: "files.media.video.resolutions.urlExpirationDate" },
            {
              path: "files.media.video.resolutions.poster.urlExpirationDate"
            },
            { path: "files.media.video.posters.urlExpirationDate" },
            { path: "files.media.audio.assets.urlExpirationDate" },
            { path: "files.media.document.urlExpirationDate" },
            { path: "files.media.document.thumbnail.urlExpirationDate" },
            { path: "files.media.vector.image.urlExpirationDate" },
            { path: "files.media.vector.previewImage.urlExpirationDate" },
            { path: "files.media.archive.urlExpirationDate" },
            { path: "files.media.model3d.urlExpirationDate" },
            { path: "files.media.model3d.thumbnail.urlExpirationDate" },
            { path: "files.media.icon.image.urlExpirationDate" },
            { path: "files.media.icon.previewImage.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "files.media.image.image.focalPoint.x" },
            { path: "files.media.image.image.focalPoint.y" },
            { path: "files.media.image.faces.confidence" },
            { path: "files.media.image.previewImage.focalPoint.x" },
            { path: "files.media.image.previewImage.focalPoint.y" },
            { path: "files.media.video.resolutions.poster.focalPoint.x" },
            { path: "files.media.video.resolutions.poster.focalPoint.y" },
            { path: "files.media.video.posters.focalPoint.x" },
            { path: "files.media.video.posters.focalPoint.y" },
            { path: "files.media.document.thumbnail.focalPoint.x" },
            { path: "files.media.document.thumbnail.focalPoint.y" },
            { path: "files.media.vector.image.focalPoint.x" },
            { path: "files.media.vector.image.focalPoint.y" },
            { path: "files.media.vector.faces.confidence" },
            { path: "files.media.vector.previewImage.focalPoint.x" },
            { path: "files.media.vector.previewImage.focalPoint.y" },
            { path: "files.media.model3d.thumbnail.focalPoint.x" },
            { path: "files.media.model3d.thumbnail.focalPoint.y" },
            { path: "files.media.icon.image.focalPoint.x" },
            { path: "files.media.icon.image.focalPoint.y" },
            { path: "files.media.icon.faces.confidence" },
            { path: "files.media.icon.previewImage.focalPoint.x" },
            { path: "files.media.icon.previewImage.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listDeletedFiles;
}

// src/media-site-media-v1-file-descriptor-files.meta.ts
function generateFilesDownloadUrl2() {
  const payload = {};
  const getRequestOptions = generateFilesDownloadUrl(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/files/generate-download-url",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function generateFileDownloadUrl2() {
  const payload = {};
  const getRequestOptions = generateFileDownloadUrl(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/files/generate-file-download-url",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getFileDescriptor2() {
  const payload = {};
  const getRequestOptions = getFileDescriptor(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/files/get-file-by-id",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getFileDescriptors2() {
  const payload = {};
  const getRequestOptions = getFileDescriptors(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/files/get-files",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateFileDescriptor2() {
  const payload = {};
  const getRequestOptions = updateFileDescriptor(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/files/update-file-descriptor",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function generateFileUploadUrl2() {
  const payload = {};
  const getRequestOptions = generateFileUploadUrl(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/files/generate-upload-url",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function generateFileResumableUploadUrl2() {
  const payload = {};
  const getRequestOptions = generateFileResumableUploadUrl(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/files/generate-resumable-upload-url",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function importFile2() {
  const payload = {};
  const getRequestOptions = importFile(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/files/import",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkImportFiles2() {
  const payload = {};
  const getRequestOptions = bulkImportFiles(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/files/import",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkImportFile2() {
  const payload = {};
  const getRequestOptions = bulkImportFile(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/files/import-v2",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listFiles2() {
  const payload = {};
  const getRequestOptions = listFiles(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/files",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function searchFiles2() {
  const payload = {};
  const getRequestOptions = searchFiles(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/files/search",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function generateVideoStreamingUrl2() {
  const payload = {};
  const getRequestOptions = generateVideoStreamingUrl(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/files/generate-video-stream-url",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkDeleteFiles2() {
  const payload = {};
  const getRequestOptions = bulkDeleteFiles(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/files/delete",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkRestoreFilesFromTrashBin2() {
  const payload = {};
  const getRequestOptions = bulkRestoreFilesFromTrashBin(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/trash-bin/files/restore",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listDeletedFiles2() {
  const payload = {};
  const getRequestOptions = listDeletedFiles(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/trash-bin/files",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  bulkDeleteFiles,
  bulkImportFile,
  bulkImportFiles,
  bulkRestoreFilesFromTrashBin,
  generateFileDownloadUrl,
  generateFileResumableUploadUrl,
  generateFileUploadUrl,
  generateFilesDownloadUrl,
  generateVideoStreamingUrl,
  getFileDescriptor,
  getFileDescriptors,
  importFile,
  listDeletedFiles,
  listFiles,
  searchFiles,
  updateFileDescriptor
});
