// @wix/headless-blog/dist/services/blog-categories-service.js
import { categories } from '@wix/blog';
import { media } from '@wix/sdk';
import { defineService, implementService } from '@wix/services-definitions';
import { SignalsServiceDefinition } from '@wix/services-definitions/core-services/signals';
export const BlogCategoriesServiceDefinition = defineService('blogCategoriesService');
export const BlogCategoriesService = implementService.withConfig()(BlogCategoriesServiceDefinition, ({ getService, config }) => {
    const signalsService = getService(SignalsServiceDefinition);
    const categoriesSignal = signalsService.signal(config.initialCategories || []);
    return {
        categories: categoriesSignal,
    };
});
export async function loadBlogCategoriesServiceConfig() {
    try {
        const result = await categories
            .queryCategories()
            .gt('postCount', 0)
            .limit(100)
            .ascending('displayPosition')
            .find();
        return {
            initialCategories: enhanceCategories(result.items || []),
        };
    }
    catch (error) {
        console.error('Error loading blog categories service config', error);
        return {
            initialCategories: [],
        };
    }
}
export function enhanceCategories(categories) {
    return categories.map((category) => {
        const imageUrl = category.coverImage ? media.getImageUrl(category.coverImage).url : null;
        return {
            ...category,
            imageUrl,
            isCustom: false,
        };
    });
}
