// @wix/auto_sdk_restaurants_availability-exceptions/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  AffectedMethodsOriginal: () => AffectedMethods,
  SortOrderOriginal: () => SortOrder,
  TypeOriginal: () => Type,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  bulkUpdateAvailabilityExceptionTags: () => bulkUpdateAvailabilityExceptionTags2,
  bulkUpdateAvailabilityExceptionTagsByFilter: () => bulkUpdateAvailabilityExceptionTagsByFilter2,
  bulkUpdateAvailabilityExceptions: () => bulkUpdateAvailabilityExceptions2,
  createAvailabilityException: () => createAvailabilityException2,
  deleteAvailabilityException: () => deleteAvailabilityException2,
  getAvailabilityException: () => getAvailabilityException2,
  queryAvailabilityExceptions: () => queryAvailabilityExceptions2,
  updateAvailabilityException: () => updateAvailabilityException2
});
module.exports = __toCommonJS(meta_exports);

// src/restaurants-v1-availability-exception-availability-exceptions.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    _base_domain_: [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "users._base_domain_": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants-availability-exceptions",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_availability-exceptions";
function createAvailabilityException(payload) {
  function __createAvailabilityException({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "availabilityException.createdDate" },
          { path: "availabilityException.updatedDate" },
          { path: "availabilityException.startTime" },
          { path: "availabilityException.endTime" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "POST",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.CreateAvailabilityException",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/availability-exceptions",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "availabilityException.createdDate" },
            { path: "availabilityException.updatedDate" },
            { path: "availabilityException.startTime" },
            { path: "availabilityException.endTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createAvailabilityException;
}
function getAvailabilityException(payload) {
  function __getAvailabilityException({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "GET",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.GetAvailabilityException",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/availability-exceptions/{availabilityExceptionId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "availabilityException.createdDate" },
            { path: "availabilityException.updatedDate" },
            { path: "availabilityException.startTime" },
            { path: "availabilityException.endTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getAvailabilityException;
}
function updateAvailabilityException(payload) {
  function __updateAvailabilityException({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "availabilityException.createdDate" },
          { path: "availabilityException.updatedDate" },
          { path: "availabilityException.startTime" },
          { path: "availabilityException.endTime" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "PATCH",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.UpdateAvailabilityException",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/availability-exceptions/{availabilityException.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "availabilityException.createdDate" },
            { path: "availabilityException.updatedDate" },
            { path: "availabilityException.startTime" },
            { path: "availabilityException.endTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateAvailabilityException;
}
function deleteAvailabilityException(payload) {
  function __deleteAvailabilityException({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "DELETE",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.DeleteAvailabilityException",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/availability-exceptions/{availabilityExceptionId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteAvailabilityException;
}
function queryAvailabilityExceptions(payload) {
  function __queryAvailabilityExceptions({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "POST",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.QueryAvailabilityExceptions",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/availability-exceptions/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "availabilityExceptions.createdDate" },
            { path: "availabilityExceptions.updatedDate" },
            { path: "availabilityExceptions.startTime" },
            { path: "availabilityExceptions.endTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryAvailabilityExceptions;
}
function bulkUpdateAvailabilityExceptions(payload) {
  function __bulkUpdateAvailabilityExceptions({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "availabilityExceptions.mask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "availabilityExceptions.availabilityException.createdDate" },
          { path: "availabilityExceptions.availabilityException.updatedDate" },
          { path: "availabilityExceptions.availabilityException.startTime" },
          { path: "availabilityExceptions.availabilityException.endTime" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "POST",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.BulkUpdateAvailabilityExceptions",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/bulk/availability-exceptions/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.startTime" },
            { path: "results.item.endTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateAvailabilityExceptions;
}
function bulkUpdateAvailabilityExceptionTags(payload) {
  function __bulkUpdateAvailabilityExceptionTags({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "POST",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.BulkUpdateAvailabilityExceptionTags",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/bulk/availability-exceptions/update-tags",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateAvailabilityExceptionTags;
}
function bulkUpdateAvailabilityExceptionTagsByFilter(payload) {
  function __bulkUpdateAvailabilityExceptionTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "POST",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.BulkUpdateAvailabilityExceptionTagsByFilter",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/bulk/availability-exceptions/update-tags-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateAvailabilityExceptionTagsByFilter;
}

// src/restaurants-v1-availability-exception-availability-exceptions.types.ts
var AffectedMethods = /* @__PURE__ */ ((AffectedMethods2) => {
  AffectedMethods2["UNKNOWN_AFFECTED_METHODS"] = "UNKNOWN_AFFECTED_METHODS";
  AffectedMethods2["ALL_FULFILLMENT_METHODS"] = "ALL_FULFILLMENT_METHODS";
  AffectedMethods2["SPECIFIC_FULFILLMENT_METHODS"] = "SPECIFIC_FULFILLMENT_METHODS";
  AffectedMethods2["ALL_PICKUP_FULFILLMENT_METHODS"] = "ALL_PICKUP_FULFILLMENT_METHODS";
  AffectedMethods2["ALL_DELIVERY_FULFILLMENT_METHODS"] = "ALL_DELIVERY_FULFILLMENT_METHODS";
  return AffectedMethods2;
})(AffectedMethods || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNSPECIFIED_AVAILABILITY_EXCEPTION_TYPE"] = "UNSPECIFIED_AVAILABILITY_EXCEPTION_TYPE";
  Type2["ORDERING_PAUSED"] = "ORDERING_PAUSED";
  Type2["CLOSURE"] = "CLOSURE";
  return Type2;
})(Type || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/restaurants-v1-availability-exception-availability-exceptions.meta.ts
function createAvailabilityException2() {
  const payload = {};
  const getRequestOptions = createAvailabilityException(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/availability-exceptions",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getAvailabilityException2() {
  const payload = {
    availabilityExceptionId: ":availabilityExceptionId"
  };
  const getRequestOptions = getAvailabilityException(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/availability-exceptions/{availabilityExceptionId}",
    pathParams: { availabilityExceptionId: "availabilityExceptionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateAvailabilityException2() {
  const payload = {
    availabilityException: { id: ":availabilityExceptionId" }
  };
  const getRequestOptions = updateAvailabilityException(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/availability-exceptions/{availabilityException.id}",
    pathParams: { availabilityExceptionId: "availabilityExceptionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteAvailabilityException2() {
  const payload = {
    availabilityExceptionId: ":availabilityExceptionId"
  };
  const getRequestOptions = deleteAvailabilityException(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/availability-exceptions/{availabilityExceptionId}",
    pathParams: { availabilityExceptionId: "availabilityExceptionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryAvailabilityExceptions2() {
  const payload = {};
  const getRequestOptions = queryAvailabilityExceptions(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/availability-exceptions/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateAvailabilityExceptions2() {
  const payload = {};
  const getRequestOptions = bulkUpdateAvailabilityExceptions(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/availability-exceptions/update",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateAvailabilityExceptionTags2() {
  const payload = {};
  const getRequestOptions = bulkUpdateAvailabilityExceptionTags(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/availability-exceptions/update-tags",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateAvailabilityExceptionTagsByFilter2() {
  const payload = {};
  const getRequestOptions = bulkUpdateAvailabilityExceptionTagsByFilter(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/availability-exceptions/update-tags-by-filter",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AffectedMethodsOriginal,
  SortOrderOriginal,
  TypeOriginal,
  WebhookIdentityTypeOriginal,
  bulkUpdateAvailabilityExceptionTags,
  bulkUpdateAvailabilityExceptionTagsByFilter,
  bulkUpdateAvailabilityExceptions,
  createAvailabilityException,
  deleteAvailabilityException,
  getAvailabilityException,
  queryAvailabilityExceptions,
  updateAvailabilityException
});
