// @wix/patterns/dist/esm/components/CollectionFilter/CollectionFilterState.js
import { addEventListener } from '@wix/bex-core/util';
import { withoutDefaults, } from '@wix/bex-core';
import { cairoFilterToggled } from '@wix/bex-core/bi';
import { EventEmitter } from 'events';
import { action } from 'mobx';
export class CollectionFilterState {
    constructor(params) {
        this.events = new EventEmitter();
        this.filter = params.filter;
        this.toolbarFiltersState = params.toolbarFiltersState;
        this.location = params.location;
    }
    _reportFilterToggleBiOnNextResult(collection, params) {
        const { filter, toolbarFiltersState, location, events } = this;
        const { toolbar } = toolbarFiltersState;
        const onFetch = () => {
            events.off('fetch', onFetch);
            // const currentFilters = ??? remember filters before query result?
            return action(() => {
                const { actionType, clickedValueKey } = params ?? {};
                toolbar.reportBi(withoutDefaults(cairoFilterToggled)({
                    ...toolbar._commonDynamicBiParams(),
                    ...collection._commonDynamicBiParams(),
                    origin: location,
                    filterValue: clickedValueKey,
                    filterName: filter.name,
                    actionType,
                    numFiltersActive: toolbar.collection.query.activeFiltersCount,
                    filteredListSize: toolbar.collection.result.total,
                    isCustomField: !!filter.isCustomField,
                }));
            });
        };
        events.on('fetch', onFetch);
    }
    init() {
        const { filter, toolbarFiltersState } = this;
        const { toolbar } = toolbarFiltersState;
        const { multi: { collections }, } = toolbar;
        const disposers = [
            ...collections
                .filter(({ query: { filtersArray } }) => filtersArray.includes(filter))
                .flatMap((collection) => [
                addEventListener(filter.events, 'beforeRefresh', (params) => {
                    this._reportFilterToggleBiOnNextResult(collection, params);
                }),
                addEventListener(collection.emitter, 'fetch', () => {
                    const ls = this.events.listeners('fetch').map((l) => l());
                    return () => {
                        ls.forEach((l) => l());
                    };
                }),
            ]),
        ];
        return () => {
            disposers.forEach((d) => d());
        };
    }
}
