// @wix/unidriver-jsdom-react/dist/cjs/UniDriverClass.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UniDriverClass = void 0;
const react_1 = require("@testing-library/react");
const user_event_1 = require("@testing-library/user-event");
const unidriver_common_1 = require("@wix/unidriver-common");
const UniDriverClassList_1 = require("./UniDriverClassList");
const enterText_1 = require("./utils/enterText");
const enterValue_1 = require("./utils/enterValue");
class UniDriverClass extends unidriver_common_1.UniDriverBase {
    constructor(ctx) {
        super(ctx);
        Object.defineProperty(this, "type", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: 'react'
        });
        Object.defineProperty(this, "get", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: async (selector, options) => {
                return (0, react_1.waitFor)(() => {
                    return super.get(selector, options);
                }, { timeout: this.getTimeBudget(options?.timeout) });
            }
        });
        Object.defineProperty(this, "getAll", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: async (selector, options) => {
                return (0, react_1.waitFor)(() => {
                    return super.getAll(selector, options);
                }, { timeout: this.getTimeBudget(options?.timeout) });
            }
        });
        Object.defineProperty(this, "api", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: {
                Class: UniDriverClass,
                ListClass: UniDriverClassList_1.UniDriverClassList,
                unwrap: async (base) => base,
                getType: () => 'jsdom-react',
                getElementsBySelector: async (base, selector) => [...base.querySelectorAll(selector)],
                click: async (base, options) => {
                    const clickParams = {
                        delay: null,
                        // We don't check pointer events for jsdom as we can't compute styles
                        pointerEventsCheck: user_event_1.PointerEventsCheckLevel.Never,
                    };
                    if (options?.type === 'double') {
                        return user_event_1.userEvent.dblClick(base, clickParams);
                    }
                    return user_event_1.userEvent.click(base, clickParams);
                },
                text: async (base) => base.textContent ?? '',
                value: async (base) => {
                    const value = base.value;
                    return typeof value === 'number' ? String(value) : (value ?? '');
                },
                enterText: async (base, value, options) => {
                    await (0, enterText_1.enterText)(base, value, options);
                },
                enterValue: async (base, value, options) => {
                    await (0, enterValue_1.enterValue)(base, value, options);
                },
                pressKey: async (base, keyMeta) => {
                    await (0, react_1.act)(async () => {
                        react_1.fireEvent.keyDown(base, keyMeta);
                        react_1.fireEvent.keyUp(base, keyMeta);
                    });
                },
                attr: async (base, attr) => base.getAttribute(attr),
                hasClass: async (base, className) => base.classList.contains(className),
                prop: async (base, prop) => base[prop],
                mouse: {
                    press: async (base) => {
                        await (0, react_1.act)(async () => {
                            react_1.fireEvent.mouseDown(base);
                        });
                    },
                    release: async (base) => {
                        await (0, react_1.act)(async () => {
                            react_1.fireEvent.mouseUp(base);
                        });
                    },
                    hover: async (base) => {
                        await (0, react_1.act)(async () => {
                            react_1.fireEvent.mouseEnter(base);
                        });
                    },
                    leave: async (base) => {
                        await (0, react_1.act)(async () => {
                            react_1.fireEvent.mouseLeave(base);
                        });
                    },
                    moveTo: async (base, to) => {
                        const { left, top } = (await to.unwrap()).getBoundingClientRect();
                        await (0, react_1.act)(async () => {
                            react_1.fireEvent.mouseMove(base, { clientX: left, clientY: top });
                        });
                    },
                },
            }
        });
        (0, unidriver_common_1.bindUniDriverApiMethods)(this, ['get', 'getAll']);
    }
}
exports.UniDriverClass = UniDriverClass;
