// @wix/auto_sdk_payments_wix-payments-account/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  connectWixPaymentsAccount: () => connectWixPaymentsAccount2
});
module.exports = __toCommonJS(meta_exports);

// src/payments-mcp-v1-wix-payments-account-wix-payments-account.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
function resolveWixPaymentsMcpV1WixPaymentsAccountServiceUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/payments/v1/wix-payments-account",
        destPath: "/v1/wix-payments-account"
      },
      {
        srcPath: "/payments/mcp/v1/wix-payments-account",
        destPath: "/v1/wix-payments-account"
      }
    ]
  };
  return (0, import_rest_modules.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_wix-payments-account";
function connectWixPaymentsAccount(payload) {
  function __connectWixPaymentsAccount({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.mcp.v1.wix_payments_account",
      method: "POST",
      methodFqn: "wix.payments.mcp.v1.WixPaymentsAccountService.ConnectWixPaymentsAccount",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsMcpV1WixPaymentsAccountServiceUrl({
        protoPath: "/v1/wix-payments-account/connect",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __connectWixPaymentsAccount;
}

// src/payments-mcp-v1-wix-payments-account-wix-payments-account.meta.ts
function connectWixPaymentsAccount2() {
  const payload = {};
  const getRequestOptions = connectWixPaymentsAccount(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/wix-payments-account/connect",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  connectWixPaymentsAccount
});
