// @wix/bex-core/dist/cjs/state/ArchiveFieldModalState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.ArchiveFieldModalState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _mobx = require("mobx");
var _ConditionalModalState = require("./ConditionalModalState");
var _DataExtensionStateBIReporter = require("./DataExtensionStateBIReporter");
class ArchiveFieldModalState {
  constructor(params) {
    (0, _defineProperty2.default)(this, "biReporter", void 0);
    (0, _defineProperty2.default)(this, "dataExtension", void 0);
    (0, _defineProperty2.default)(this, "modal", void 0);
    this.biReporter = new _DataExtensionStateBIReporter.DataExtensionStateBIReporter({
      reportBi: params.reportBi
    });
    this.modal = new _ConditionalModalState.ConditionalModalState({
      onConfirm: async () => {
        if (!this.modal.value) {
          return;
        }
        this.biReporter.reportArchiveFieldModalCtaClicked('Archive');
        await this.dataExtension.archiveField({
          ...this.modal.value,
          reportBi: this.biReporter.reportBi
        });
      },
      onClose: () => {
        this.biReporter.reportArchiveFieldModalCtaClicked('Cancel');
      }
    });
    this.dataExtension = params.dataExtension;
    (0, _mobx.makeAutoObservable)(this);
  }
}
exports.ArchiveFieldModalState = ArchiveFieldModalState;
