// @wix/bex-core/dist/cjs/state/BulkActionModalState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.BulkActionModalState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _ConditionalModalState = require("./ConditionalModalState");
class BulkActionModalState {
  constructor(params) {
    (0, _defineProperty2.default)(this, "translate", void 0);
    (0, _defineProperty2.default)(this, "scrollToTop", void 0);
    (0, _defineProperty2.default)(this, "actionModal", new _ConditionalModalState.ConditionalModalState());
    (0, _defineProperty2.default)(this, "openModal", () => {
      this.actionModal.modal.open();
    });
    (0, _defineProperty2.default)(this, "closeModal", () => {
      this.actionModal.modal.close();
    });
    (0, _defineProperty2.default)(this, "closeModalAndScrollTop", () => {
      this.closeModal();
      this.scrollToTop();
    });
    (0, _defineProperty2.default)(this, "openConfirmModal", props => {
      const {
        translate: t
      } = this;
      this.actionModal.open({
        title: t('cairo.confirmModal.title'),
        primaryButtonText: t('cairo.confirm.button'),
        secondaryButtonText: t('cairo.cancel.button'),
        ...props
      });
    });
    this.translate = params.translate;
    this.scrollToTop = params.scrollToTop;
  }
  get isModalOpen() {
    return this.actionModal.isOpen;
  }
}
exports.BulkActionModalState = BulkActionModalState;
