// @wix/bex-core/dist/cjs/state/CollectionBIReporter.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.CollectionBIReporter = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _util = require("../util");
var _v = require("@wix/bex-utils/@wix/bi-logger-os-data/v2");
var _mobx = require("mobx");
var _events = require("events");
const withoutDefaults = (0, _util.withoutIrrelevant)();
class CollectionBIReporter {
  constructor(params) {
    (0, _defineProperty2.default)(this, "collection", void 0);
    (0, _defineProperty2.default)(this, "reportBi", void 0);
    (0, _defineProperty2.default)(this, "events", new _events.EventEmitter());
    (0, _defineProperty2.default)(this, "sortedColumnId", '');
    this.collection = params.collection;
    this.reportBi = params.reportBi;
  }
  init() {
    const {
      collection,
      events
    } = this;
    const {
      emitter,
      query
    } = collection;
    const disposers = [(0, _util.addEventListener)(emitter, 'beforeClearFilters', (0, _mobx.action)(({
      origin
    }) => {
      this.reportBi(withoutDefaults(_v.cairoAllFiltersCleared)({
        filterNames: collection.query.activeFilters.map(filter => filter.name).join(','),
        filteredListSize: collection.result.total,
        numFiltersCleared: collection.query.activeFiltersCount,
        origin
      }));
    })), (0, _util.addEventListener)(emitter, 'beforeSortStart', (0, _mobx.action)(({
      origin
    }) => {
      const prevSortedColumnId = this.sortedColumnId;
      const prevSortedIds = new Set(query.sort.value.map(({
        field
      }) => field));
      const prevSortOrder = this._getSortOrder();
      let prevSortingOrder = this._getSortDirection(prevSortedColumnId);
      events.once('sortStart', ({
        col
      }) => {
        this.sortedColumnId = col.id;
        if (prevSortedColumnId && col.id !== prevSortedColumnId) {
          prevSortingOrder = this._getSortDirection(prevSortedColumnId);
        } else if (!prevSortedColumnId) {
          prevSortingOrder = '';
        }
        const itemIndex = this.collection.query.sort.value.findIndex(({
          field
        }) => field === col.id);
        this.reportBi(withoutDefaults(_v.cairoComponentSorted)({
          newSortedColumn: col.id,
          newSortingOrder: this._getSortDirection(col.id),
          prevSortedColumn: prevSortedColumnId,
          prevSortingOrder,
          origin,
          prevSortOrder,
          newSortOrder: this._getSortOrder(),
          itemIndex,
          isAddNewSort: !prevSortedIds.has(col.id),
          prevNumSortingColumns: prevSortedIds.size,
          numSortingColumns: query.sort.value.length
        }));
      });
    })), (0, _util.addEventListener)(emitter, 'sortStart', params => {
      events.emit('sortStart', params);
    }), (0, _util.addEventListener)(emitter, 'newPageStart', () => {
      const startTime = performance.now();
      return ({
        totalNewItems
      }) => {
        this.reportBi(withoutDefaults(_v.loadMore)({
          loadedItems: totalNewItems,
          loadingTime: Math.round(performance.now() - startTime)
        }));
      };
    }), (0, _util.addEventListener)(emitter, 'searchStart', () => {
      const startTime = performance.now();
      const searchTerm = collection.query.search.value;
      return () => {
        this.reportBi(withoutDefaults(_v.cairoSearchResults)({
          searchTerm,
          searchResultsCnt: collection.result.asComputed.total,
          loadingTime: Math.round(performance.now() - startTime)
        }));
      };
    }), (0, _util.addEventListener)(emitter, 'search', (0, _mobx.action)(() => {
      this.reportBi(withoutDefaults(_v.cairoFilterToggled)({
        filterName: 'search',
        origin: 'toolbar',
        numFiltersActive: collection.query.activeFiltersCount,
        actionType: 'add',
        filterValue: collection.query.search.value,
        filteredListSize: collection.result.total
      }));
    }))];
    return () => {
      disposers.forEach(d => d());
    };
  }
  _getSortDirection(columndId) {
    const {
      query: {
        sort: {
          value
        }
      }
    } = this.collection;
    const sort = value.find(({
      field
    }) => field === columndId);
    if (sort) {
      return sort.direction;
    }
    return 'removed';
  }
  _getSortOrder() {
    const {
      query: {
        sort
      }
    } = this.collection;
    return JSON.stringify(sort.value.map(({
      field,
      direction
    }) => ({
      fieldName: field,
      order: direction
    })));
  }
}
exports.CollectionBIReporter = CollectionBIReporter;
