// @wix/bex-core/dist/cjs/state/CollectionOptimisticHelpers.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.CollectionOptimisticProcessorHelpers = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _mobx = require("mobx");
class CollectionOptimisticProcessorHelpers {
  constructor(collection) {
    (0, _defineProperty2.default)(this, "collection", void 0);
    this.collection = collection;
    (0, _mobx.makeObservable)(this, {
      reprocess: _mobx.action.bound,
      commit: _mobx.action
    });
  }
  reprocess() {
    const {
      collection
    } = this;
    const {
      result
    } = collection;
    const {
      _pages: {
        unprocessed: unprocessedPages
      },
      _totals: {
        unprocessed: unprocessedTotal
      }
    } = result;
    const {
      processedTotal,
      processedPages
    } = collection._getProcessedResult({
      unprocessedPages,
      unprocessedTotal,
      currentItemsMap: new Map()
    });
    result.setTotals({
      unprocessed: unprocessedTotal,
      processed: processedTotal
    });
    result.setPages({
      unprocessed: unprocessedPages,
      processed: processedPages
    });
  }
  commit() {
    const {
      collection
    } = this;
    const {
      result
    } = collection;
    const {
      _pages: {
        processed: processedPages
      },
      _totals: {
        processed: processedTotal
      }
    } = result;
    result._totals.unprocessed = processedTotal;
    result._pages.unprocessed = processedPages;
  }
}
exports.CollectionOptimisticProcessorHelpers = CollectionOptimisticProcessorHelpers;
