// @wix/bex-core/dist/cjs/state/ConditionalModalState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.ConditionalModalState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _mobx = require("mobx");
var _ModalState = require("./ModalState");
class ConditionalModalState {
  constructor(params) {
    (0, _defineProperty2.default)(this, "modal", void 0);
    (0, _defineProperty2.default)(this, "value", null);
    this.modal = new _ModalState.ModalState(params);
    (0, _mobx.makeObservable)(this, {
      value: _mobx.observable.ref,
      afterClose: _mobx.action.bound,
      open: _mobx.action.bound,
      close: _mobx.action.bound
    });
  }
  get isOpen() {
    return this.modal.isOpen;
  }
  close() {
    this.modal.close();
  }
  afterClose() {
    this.value = null;
  }
  open(value) {
    this.value = value;
    this.modal.open();
  }
}
exports.ConditionalModalState = ConditionalModalState;
