// @wix/bex-core/dist/cjs/state/CustomFieldsViewWidgetState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.CustomFieldsViewWidgetState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _TaskState = require("./TaskState");
var _DataExtensionState = require("./DataExtensionState");
var _services = require("../services");
var _v = require("@wix/bex-utils/@wix/bi-logger-os-data/v2");
var _AppsState = require("./AppsState");
var _mobx = require("mobx");
class CustomFieldsViewWidgetState {
  constructor(params) {
    (0, _defineProperty2.default)(this, "container", void 0);
    (0, _defineProperty2.default)(this, "dataExtension", void 0);
    (0, _defineProperty2.default)(this, "apps", void 0);
    (0, _defineProperty2.default)(this, "initTask", new _TaskState.TaskState());
    (0, _defineProperty2.default)(this, "reportBi", void 0);
    (0, _defineProperty2.default)(this, "startTime", performance.now());
    (0, _defineProperty2.default)(this, "url", void 0);
    (0, _defineProperty2.default)(this, "route", void 0);
    (0, _defineProperty2.default)(this, "theme", void 0);
    (0, _defineProperty2.default)(this, "userFieldsNamespace", void 0);
    const {
      container,
      fqdn,
      url,
      theme,
      route,
      extensionPoint
    } = params;
    this.container = container;
    this.url = url;
    this.route = route;
    this.theme = theme;
    this.reportBi = container.createBILogger({
      componentType: 'CustomFieldsViewWidget',
      ...fqdn
    });
    this.dataExtension = new _DataExtensionState.DataExtensionState({
      container,
      fqdn,
      extensionPoint
    });
    this.userFieldsNamespace = this.dataExtension.dataExtensionService.userFieldsNamespace;
    this.apps = new _AppsState.AppsState({
      container,
      dataExtension: this.dataExtension
    });
    this.reportBi((0, _v.loadStart)({
      url: this.url,
      theme
    }));
    (0, _mobx.makeObservable)(this, {
      isLoading: _mobx.computed
    });
  }
  init() {
    const {
      initTask,
      apps,
      dataExtension
    } = this;
    dataExtension.init();
    apps.init();
    initTask.runOnce(async () => {
      var _dataExtension$initTa, _apps$initTask$status;
      dataExtension.initTask.runOnce();
      apps.initTask.runOnce();
      await Promise.all([(_dataExtension$initTa = dataExtension.initTask.status.promise) == null ? void 0 : _dataExtension$initTa.catch(this.onInitError.bind(this)), (_apps$initTask$status = apps.initTask.status.promise) == null ? void 0 : _apps$initTask$status.catch(() => null)]);
      this.reportBi((0, _v.loadEnd)({
        url: this.url,
        route: this.route,
        loadingTime: Math.round(performance.now() - this.startTime),
        theme: this.theme
      }));
    });
  }
  onInitError(err) {
    const cairoError = new _services.WixPatternsError({
      errorCode: 'InitCustomFieldsDataExtensionFailed',
      originalException: err
    });
    this.container.internalMonitor.reportError({
      error: cairoError
    });
    this.reportBi((0, _v.cairoErrorInLoadingAComponent)({
      errorType: cairoError.name
    }));
    throw cairoError;
  }
  retry() {
    this.reportBi((0, _v.cairoTryAgainClicked)({
      filteredListSize: this.dataExtension.availableUserCustomFields.length,
      actionName: 'Load custom fields',
      loaction: 'View only widget'
    }));
    this.initTask.runOnce();
  }
  get isLoading() {
    const {
      status
    } = this.initTask;
    return status.isIdle || status.isLoading;
  }
  get isError() {
    return this.initTask.status.isError;
  }
  getExtendedFieldsToDisplayForNamespace(namespace, extendedFields) {
    var _this$dataExtension$a;
    const isUserNamespace = namespace === this.userFieldsNamespace;
    if (isUserNamespace) {
      var _extendedFields$names;
      const userCustomFields = this.dataExtension.availableUserCustomFields;
      const fields = (extendedFields == null || (_extendedFields$names = extendedFields.namespaces) == null ? void 0 : _extendedFields$names[namespace]) ?? {};
      return userCustomFields.filter(field => {
        const fieldValue = fields[field.id];
        const isEmpty = fieldValue === null || fieldValue === undefined;
        return !isEmpty;
      });
    }
    const appSchema = (_this$dataExtension$a = this.dataExtension.appSchemas) == null ? void 0 : _this$dataExtension$a.find(schema => schema.namespace === namespace);
    const appFields = appSchema ? this.dataExtension.getAvailableCustomFieldsForApp(appSchema) : [];
    return appFields;
  }
  getAppName(namespace) {
    var _this$dataExtension$a2;
    const appSchema = (_this$dataExtension$a2 = this.dataExtension.appSchemas) == null ? void 0 : _this$dataExtension$a2.find(schema => schema.namespace === namespace);
    return this.apps.appNames[(appSchema == null ? void 0 : appSchema.appDefId) || ''];
  }
}
exports.CustomFieldsViewWidgetState = CustomFieldsViewWidgetState;
