// @wix/bex-core/dist/cjs/state/DataExtensionState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.MAX_CUSTOM_FIELDS = exports.DataExtensionState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _services = require("../services");
var _TaskState = require("./TaskState");
var _mobx = require("mobx");
var _DataExtensionStateBIReporter = require("./DataExtensionStateBIReporter");
var _util = require("../util");
var _DataExtensionField = require("./DataExtension/DataExtensionField");
const MAX_CUSTOM_FIELDS = exports.MAX_CUSTOM_FIELDS = 50;
class DataExtensionState {
  get fqdnString() {
    return (0, _util.fqdnToString)(this.fqdn);
  }
  get isFqdnValid() {
    return this.fqdn.resource !== undefined;
  }
  constructor(params) {
    (0, _defineProperty2.default)(this, "dataExtensionService", void 0);
    (0, _defineProperty2.default)(this, "initTask", new _TaskState.TaskState());
    (0, _defineProperty2.default)(this, "container", void 0);
    (0, _defineProperty2.default)(this, "_hasWritePermissions", void 0);
    (0, _defineProperty2.default)(this, "closeSidePanel", void 0);
    (0, _defineProperty2.default)(this, "fqdn", void 0);
    (0, _defineProperty2.default)(this, "_dataExtensionSchemasBase", []);
    (0, _defineProperty2.default)(this, "containsPii", void 0);
    (0, _defineProperty2.default)(this, "defaultPermissions", void 0);
    (0, _defineProperty2.default)(this, "disableUserDefinedWriting", void 0);
    (0, _defineProperty2.default)(this, "filterable", void 0);
    (0, _defineProperty2.default)(this, "_extensionPoint", void 0);
    this.container = params.container;
    this.disableUserDefinedWriting = params.disableUserDefinedWriting;
    this.filterable = params.filterable;
    this.closeSidePanel = params.closeSidePanel;
    this.fqdn = params.fqdn;
    this._extensionPoint = params.extensionPoint;
    this.dataExtensionService = new _services.DataExtensionSchemaService(params.container);
    this.containsPii = params.containsPii;
    this.defaultPermissions = params.defaultPermissions;
    (0, _mobx.makeObservable)(this, {
      allUserCustomFields: _mobx.computed,
      allAppsCustomFields: _mobx.computed,
      allCustomFields: _mobx.computed,
      filterableCustomFields: _mobx.computed,
      availableUserCustomFields: _mobx.computed,
      availableAppsCustomFields: _mobx.computed,
      availableCustomFields: _mobx.computed,
      isReachedMaxCustomFields: _mobx.computed,
      disableAddCustomField: _mobx.computed,
      disableAddCustomFieldContent: _mobx.computed,
      _dataExtensionSchemasBase: _mobx.observable.ref,
      _dataExtensionSchemas: _mobx.computed,
      userSchema: _mobx.computed,
      appSchemas: _mobx.computed,
      _upsertSchema: _mobx.action,
      containsPii: _mobx.observable,
      defaultPermissions: _mobx.observable
    });
  }
  get hasWritePermissions() {
    if (this._hasWritePermissions == null) {
      this._hasWritePermissions = this.container.user.permissions.includes('DATA_EXTENSION_SCHEMA.WRITE');
    }
    return this._hasWritePermissions;
  }
  get disableAddCustomField() {
    return !this.hasWritePermissions || this.isReachedMaxCustomFields;
  }
  get disableAddCustomFieldContent() {
    if (!this.hasWritePermissions) {
      return this.container.translate('cairo.customFields.writePermission.tooltip');
    }
    if (this.isReachedMaxCustomFields) {
      return this.container.translate('cairo.customFields.maxFields.tooltip', {
        max: MAX_CUSTOM_FIELDS
      });
    }
    return '';
  }
  async archiveField({
    id,
    origin,
    reportBi
  }) {
    const biReporter = new _DataExtensionStateBIReporter.DataExtensionStateBIReporter({
      reportBi
    });
    if (id == null) {
      return;
    }
    try {
      var _this$container$showT, _this$container;
      const {
        dataExtensionSchema
      } = await this.dataExtensionService.archiveCustomField({
        schema: this.userSchema,
        fqdn: this.fqdnString,
        id
      });
      if (dataExtensionSchema) {
        this._upsertSchema(dataExtensionSchema);
      }
      (_this$container$showT = (_this$container = this.container).showToast) == null || _this$container$showT.call(_this$container, {
        message: this.container.translate('cairo.customFields.archive.success.toast'),
        type: 'SUCCESS',
        biName: 'cairo-archive-custom-field-success-toast'
      });
    } catch (e) {
      var _this$container$showT2, _this$container2;
      (_this$container$showT2 = (_this$container2 = this.container).showToast) == null || _this$container$showT2.call(_this$container2, {
        message: this.container.translate('cairo.customFields.archive.technicalError'),
        action: {
          uiType: 'LINK',
          text: this.container.translate('cairo.toast.retry'),
          onClick: () => {
            biReporter.reportArchiveTryAgain();
            this.archiveField({
              id,
              origin,
              reportBi
            });
          },
          removeToastOnClick: true
        },
        type: 'ERROR',
        biName: 'cairo-archive-custom-field-error-toast'
      });
      biReporter.reportArchiveFieldFailed({
        formValues: this.extractFieldValuesFromSchema(id),
        origin
      });
    }
  }
  extractFieldValuesFromSchema(fieldId) {
    var _customField$xWixPi;
    if (!this.userSchema) {
      return;
    }
    const customField = _services.DataExtensionSchemaService.toCustomField(this.userSchema, fieldId);
    if (!customField || !customField.type || !customField.title) {
      return;
    }
    const {
      read,
      write
    } = customField['x-wix-permissions'];
    const isPiiField = (_customField$xWixPi = customField['x-wix-pii']) == null ? void 0 : _customField$xWixPi.enabled;
    const readApps = read == null ? void 0 : read.includes('apps');
    const readUsers = read == null ? void 0 : read.includes('users-of-users');
    const writeApps = write == null ? void 0 : write.includes('apps');
    const writeUsers = write == null ? void 0 : write.includes('users-of-users');
    return {
      type: _services.DataExtensionSchemaService.toCustomType(customField),
      name: customField.title,
      key: customField.id,
      readApps,
      readUsers,
      writeApps,
      writeUsers,
      containsPii: Boolean(isPiiField)
    };
  }
  _toCustomField({
    key,
    schema,
    editable,
    owner,
    namespace
  }) {
    if (!_services.DataExtensionSchemaService.toCustomType(schema)) {
      return;
    }
    return {
      id: key,
      label: (schema == null ? void 0 : schema.title) || key,
      maxLength: schema == null ? void 0 : schema.maxLength,
      options: schema == null ? void 0 : schema.enum,
      type: _services.DataExtensionSchemaService.toCustomType(schema),
      editable,
      filterable: schema['x-wix-filterable'],
      owner,
      properties: (schema == null ? void 0 : schema.properties) ?? {},
      namespace,
      permissions: schema['x-wix-permissions']
    };
  }
  isSupportedFieldType(type) {
    if (!type) {
      return false;
    }
    return ['string', 'number', 'integer', 'boolean', 'array'].includes(type);
  }
  getAvailableCustomFieldsInSchema(properties, {
    editable,
    requiredPermissions,
    owner,
    namespace
  }) {
    const filteredFields = Object.keys(properties).filter(key => {
      const field = properties[key];
      const isAllowedType = this.isSupportedFieldType(field.type) || field.type === 'object' && field['x-wix-object-type'] === 'FILE/V1';
      const isPermitted = requiredPermissions == null || (0, _DataExtensionField.includesAllReadPermissions)(field, requiredPermissions);
      return !field['x-wix-archived'] && isAllowedType && isPermitted;
    });
    const sorted = filteredFields.sort((a, b) => {
      const aVal = new Date(properties[a]['x-wix-created-date']);
      const bVal = new Date(properties[b]['x-wix-created-date']);
      if (isNaN(bVal.valueOf()) || isNaN(aVal.valueOf())) {
        return 0;
      }
      return aVal.valueOf() - bVal.valueOf();
    });
    return sorted.map(key => this._toCustomField({
      key,
      schema: properties[key],
      editable,
      owner,
      namespace
    })).filter(Boolean);
  }

  /**
   * Available custom fields, excluding archived fields
   */
  get availableUserCustomFields() {
    var _this$userSchema;
    const properties = ((_this$userSchema = this.userSchema) == null || (_this$userSchema = _this$userSchema.jsonSchema) == null ? void 0 : _this$userSchema.properties) || {};
    return this.getAvailableCustomFieldsInSchema(properties, {
      editable: true,
      owner: 'user',
      namespace: '_user_fields'
    });
  }
  get availableAppsCustomFields() {
    var _this$appSchemas;
    return ((_this$appSchemas = this.appSchemas) == null ? void 0 : _this$appSchemas.reduce((prev, schema) => {
      return [...prev, ...this.getAvailableCustomFieldsForApp(schema)];
    }, [])) ?? [];
  }
  getSchemaAndFieldsByNamespace(namespace) {
    var _schema$jsonSchema;
    const {
      _dataExtensionSchemas,
      dataExtensionService: {
        userFieldsNamespace
      }
    } = this;
    const schema = _dataExtensionSchemas.find(s => s.namespace === namespace);
    if (!schema) {
      return {};
    }
    const fields = this.getAvailableCustomFieldsInSchema(((_schema$jsonSchema = schema.jsonSchema) == null ? void 0 : _schema$jsonSchema.properties) || {}, {
      editable: false,
      owner: namespace === userFieldsNamespace ? 'user' : 'app',
      namespace
    });
    return {
      schema,
      fields
    };
  }
  getAvailableCustomFieldsForApp(appSchema) {
    var _appSchema$jsonSchema;
    const properties = ((_appSchema$jsonSchema = appSchema.jsonSchema) == null ? void 0 : _appSchema$jsonSchema.properties) || {};
    return this.getAvailableCustomFieldsInSchema(properties, {
      editable: false,
      requiredPermissions: ['users'],
      owner: 'app',
      namespace: appSchema.namespace
    });
  }
  get availableCustomFields() {
    return [...this.availableUserCustomFields, ...this.availableAppsCustomFields];
  }
  get allUserCustomFields() {
    var _this$userSchema2;
    const properties = ((_this$userSchema2 = this.userSchema) == null || (_this$userSchema2 = _this$userSchema2.jsonSchema) == null ? void 0 : _this$userSchema2.properties) || {};
    return Object.keys(properties).map(key => this._toCustomField({
      key,
      schema: properties[key],
      editable: true,
      owner: 'user',
      namespace: '_user_fields'
    })).filter(Boolean);
  }
  get allAppsCustomFields() {
    var _this$appSchemas2;
    return ((_this$appSchemas2 = this.appSchemas) == null ? void 0 : _this$appSchemas2.reduce((prev, appSchema) => {
      var _appSchema$jsonSchema2;
      const properties = (appSchema == null || (_appSchema$jsonSchema2 = appSchema.jsonSchema) == null ? void 0 : _appSchema$jsonSchema2.properties) || {};
      return prev.concat(Object.keys(properties).map(key => this._toCustomField({
        key,
        schema: properties[key],
        editable: false,
        owner: 'app',
        namespace: appSchema.namespace
      })).filter(Boolean));
    }, [])) ?? [];
  }
  get allCustomFields() {
    return [...this.allUserCustomFields, ...this.allAppsCustomFields];
  }
  get filterableCustomFields() {
    if (!this.filterable) {
      return [];
    }
    return this.availableCustomFields.filter(field => field.filterable);
  }
  get isReachedMaxCustomFields() {
    return this.allUserCustomFields.length >= MAX_CUSTOM_FIELDS;
  }
  get allNamespaces() {
    var _this$userSchema3;
    return [(_this$userSchema3 = this.userSchema) == null ? void 0 : _this$userSchema3.namespace, ...(this.appSchemas || []).map(schema => schema.namespace)].filter(Boolean);
  }
  init() {
    const {
      initTask
    } = this;
    initTask.runOnce(async () => {
      await this.fetchSchemas();
    });
    return () => {};
  }
  _upsertSchema(newSchema) {
    const {
      _dataExtensionSchemasBase
    } = this;
    const index = _dataExtensionSchemasBase.findIndex(schema => schema.namespace === (newSchema == null ? void 0 : newSchema.namespace));
    if (index === -1) {
      this._dataExtensionSchemasBase = [..._dataExtensionSchemasBase, newSchema];
    } else {
      this._dataExtensionSchemasBase = _dataExtensionSchemasBase.map(schema => schema.namespace === newSchema.namespace ? newSchema : schema);
    }
  }
  get userSchema() {
    const {
      _dataExtensionSchemas,
      dataExtensionService: {
        userFieldsNamespace
      }
    } = this;
    return _dataExtensionSchemas.find(s => s.namespace === userFieldsNamespace);
  }
  get _dataExtensionSchemas() {
    const {
      _extensionPoint,
      _dataExtensionSchemasBase
    } = this;
    return _dataExtensionSchemasBase.filter(s => _extensionPoint ? s.extensionPoint === _extensionPoint : s.extensionPoint == null || s.extensionPoint === 'ROOT');
  }
  get appSchemas() {
    const {
      _dataExtensionSchemas,
      dataExtensionService: {
        userFieldsNamespace
      }
    } = this;
    return _dataExtensionSchemas.filter(s => s.namespace !== userFieldsNamespace);
  }
  async fetchSchemas() {
    const {
      dataExtensionService
    } = this;
    if (this.isFqdnValid) {
      const {
        dataExtensionSchemas
      } = await dataExtensionService.fetchSchemas(this.fqdnString);
      (0, _mobx.runInAction)(() => {
        this._dataExtensionSchemasBase = dataExtensionSchemas ?? [];
      });
    }
  }
}
exports.DataExtensionState = DataExtensionState;
