// @wix/bex-core/dist/cjs/state/FormPageState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.FormPageState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _mobx = require("mobx");
var _TaskState = require("./TaskState");
var _WidgetsFormState = require("./WidgetsFormState");
class FormPageState {
  constructor(props) {
    (0, _defineProperty2.default)(this, "_widgets", void 0);
    (0, _defineProperty2.default)(this, "_form", void 0);
    (0, _defineProperty2.default)(this, "initTask", new _TaskState.TaskState());
    (0, _defineProperty2.default)(this, "submitTask", new _TaskState.TaskState());
    (0, _defineProperty2.default)(this, "_isSubmitSuccessful", undefined);
    (0, _defineProperty2.default)(this, "_isFormDirty", false);
    this._form = props.form;
    this._widgets = props.widgets ?? new _WidgetsFormState.WidgetsFormState();
    (0, _mobx.makeObservable)(this, {
      isFetching: _mobx.computed,
      _isFormDirty: _mobx.observable.ref,
      isDirty: _mobx.computed
    });
  }
  set isFormDirty(flag) {
    this._isFormDirty = flag;
  }
  get isDirty() {
    return this._isFormDirty || this._widgets.isDirty;
  }
  get form() {
    return this._form;
  }
  get isFetching() {
    return this.initTask.status.isLoading || this.initTask.status.isIdle;
  }
  set isSubmitSuccessful(flag) {
    this._isSubmitSuccessful = flag;
  }
  retryFetch() {
    this.initTask.runOnce();
  }

  /**
   * For backward compatibility, keep this as validate, but it only validates widgets.
   */
  validate() {
    return this._widgets.validate();
  }
  registerWidget(key, state) {
    this._widgets.registerWidget(key, state);
  }
  getWidgetState(key) {
    return this._widgets.get(key);
  }
}
exports.FormPageState = FormPageState;
