// @wix/bex-core/dist/cjs/state/ModalState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.ModalState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _ToggleState = require("./ToggleState");
var _mobx = require("mobx");
var _model = require("../model");
var _events = require("events");
class ModalState {
  constructor(params = {}) {
    (0, _defineProperty2.default)(this, "openToggle", new _ToggleState.ToggleState());
    (0, _defineProperty2.default)(this, "toggle", this.openToggle.toggle);
    (0, _defineProperty2.default)(this, "onConfirm", void 0);
    (0, _defineProperty2.default)(this, "onClose", void 0);
    (0, _defineProperty2.default)(this, "confirming", _model.queryStatus.idle);
    (0, _defineProperty2.default)(this, "events", new _events.EventEmitter());
    (0, _defineProperty2.default)(this, "requestClose", () => {
      this.events.emit('requestClose');
    });
    (0, _mobx.makeObservable)(this, {
      confirming: _mobx.observable,
      open: _mobx.action.bound,
      confirm: _mobx.action.bound,
      close: _mobx.action.bound,
      isOpen: _mobx.computed
    });
    this.onConfirm = params.onConfirm;
    this.onClose = params.onClose;
  }
  open() {
    this.events.emit('open');
    this.openToggle.setOn();
  }
  promise() {
    return new Promise(resolve => {
      this.events.once('close', options => {
        resolve(options);
      });
    });
  }
  async confirm() {
    try {
      var _this$onConfirm;
      this.confirming = _model.queryStatus.loading;
      await ((_this$onConfirm = this.onConfirm) == null ? void 0 : _this$onConfirm.call(this));
      (0, _mobx.runInAction)(() => {
        this.confirming = _model.queryStatus.success;
        this.events.emit('close', {
          result: 'confirmed'
        });
        return this.openToggle.setOff();
      });
    } catch (e) {
      (0, _mobx.runInAction)(() => {
        this.confirming = _model.queryStatus.error;
      });
    }
  }
  close() {
    var _this$onClose;
    (_this$onClose = this.onClose) == null || _this$onClose.call(this);
    this.events.emit('close', {
      result: 'dismissed'
    });
    return this.openToggle.setOff();
  }
  decline() {
    var _this$onClose2;
    (_this$onClose2 = this.onClose) == null || _this$onClose2.call(this);
    this.events.emit('close', {
      result: 'declined'
    });
    return this.openToggle.setOff();
  }
  get isOpen() {
    return this.openToggle.isOn;
  }
}
exports.ModalState = ModalState;
