// @wix/bex-core/dist/cjs/state/MultiCollectionSupport.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.MultiCollectionSupport = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _mobx = require("mobx");
var _TaskState = require("./TaskState");
class MultiCollectionSupport {
  constructor(params) {
    (0, _defineProperty2.default)(this, "collections", void 0);
    (0, _defineProperty2.default)(this, "initTask", new _TaskState.TaskState());
    (0, _defineProperty2.default)(this, "retryErrorState", (0, _mobx.action)(() => {
      const {
        initTask,
        collections
      } = this;
      if (initTask.status.isError) {
        initTask.runOnce();
      } else {
        collections.forEach(c => c.retryErrorState());
      }
    }));
    this.collections = params.collections;
    (0, _mobx.makeObservable)(this, {
      showInitialLoader: _mobx.computed,
      showEmptyState: _mobx.computed,
      showErrorState: _mobx.computed,
      showRefreshLoader: _mobx.computed,
      errorStatus: _mobx.computed,
      hasAvailableItems: _mobx.computed,
      hasNonPersistentActiveFilters: _mobx.computed,
      hasNonPersistentCustomFilters: _mobx.computed
    });
  }
  get showInitialLoader() {
    return this.collections.some(c => c.showInitialLoader);
  }
  get showEmptyState() {
    return this.collections.every(collection => collection.showEmptyState);
  }
  get showErrorState() {
    return this.collections.some(collection => collection.showErrorState);
  }
  get showRefreshLoader() {
    return this.collections.some(collection => collection.showRefreshLoader);
  }
  get showLoadingState() {
    return this.showInitialLoader || this.showRefreshLoader;
  }
  get errorStatus() {
    return this.collections.map(c => c.result.status).find(status => status.isError);
  }
  get hasAvailableItems() {
    return this.collections.some(c => c.result.hasAvailableItems);
  }
  get hasNonPersistentActiveFilters() {
    return this.collections.some(c => c.query.hasNonPersistentActiveFilters);
  }
  get hasNonPersistentCustomFilters() {
    return this.collections.some(c => c.query.hasNonPersistentCustomFilters);
  }
  get resultOriginQuerySearch() {
    var _this$collections$;
    return (_this$collections$ = this.collections[0]) == null ? void 0 : _this$collections$.result.originQuery.search;
  }
  get resultOriginQueryRawSearch() {
    var _this$collections$2;
    return (_this$collections$2 = this.collections[0]) == null ? void 0 : _this$collections$2.result.originQuery.rawSearch;
  }
  init({
    skipFilterListeners = false
  } = {}) {
    const {
      collections,
      initTask
    } = this;
    const disposers = [...collections.map(c => c.init({
      skipFilterListeners
    }))];
    initTask.runOnce(async () => {
      collections.forEach(c => c.initTask.runOnce());
      await Promise.all(collections.map(c => c.initTask.status.promise));
    });
    return () => {
      disposers.forEach(d => d());
    };
  }
}
exports.MultiCollectionSupport = MultiCollectionSupport;
