// @wix/bex-core/dist/cjs/state/OnlineState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.OnlineState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _events = require("events");
class OnlineState {
  constructor(params) {
    (0, _defineProperty2.default)(this, "onlineManager", void 0);
    (0, _defineProperty2.default)(this, "focusManager", void 0);
    (0, _defineProperty2.default)(this, "showToast", void 0);
    (0, _defineProperty2.default)(this, "translate", void 0);
    (0, _defineProperty2.default)(this, "events", new _events.EventEmitter());
    (0, _defineProperty2.default)(this, "_removeOfflineToastTimeoutId", null);
    (0, _defineProperty2.default)(this, "_removeOfflineToast", null);
    this.onlineManager = params.onlineManager;
    this.focusManager = params.focusManager;
    this.showToast = params.showToast;
    this.translate = params.translate;
  }
  get isOnline() {
    return this.onlineManager.isOnline();
  }
  showOfflineToast() {
    var _this$_removeOfflineT;
    const {
      showToast,
      translate: t
    } = this;
    if (showToast == null) {
      return;
    }
    (_this$_removeOfflineT = this._removeOfflineToast) == null || _this$_removeOfflineT.call(this);
    this._removeOfflineToast = showToast({
      type: 'ERROR',
      message: t('cairo.offline'),
      biName: 'cairo-offline-error',
      timeout: 'NONE'
    }).remove;

    // this._removeOfflineToastTimeoutId = setTimeout(() => {
    //   this._removeOfflineToastTimeoutId = null;
    //   this._removeOfflineToast?.();
    // }, 6000);
  }
  init() {
    const {
      focusManager,
      onlineManager,
      events
    } = this;
    const disposers = [onlineManager.subscribe(() => {
      const {
        isOnline
      } = this;
      events.emit('change', isOnline);
      if (!isOnline) {
        this.showOfflineToast();
      } else {
        var _this$_removeOfflineT2;
        (_this$_removeOfflineT2 = this._removeOfflineToast) == null || _this$_removeOfflineT2.call(this);
      }
    }), focusManager.subscribe(() => {
      const {
        isOnline
      } = this;
      if (focusManager.isFocused() && isOnline && this._removeOfflineToast) {
        this._removeOfflineToast();
      }
    })];
    return () => {
      disposers.forEach(d => d());
      if (this._removeOfflineToast) {
        this._removeOfflineToast();
      }
      if (this._removeOfflineToastTimeoutId != null) {
        clearTimeout(this._removeOfflineToastTimeoutId);
      }
    };
  }
}
exports.OnlineState = OnlineState;
