// @wix/bex-core/dist/cjs/state/PageState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.PageState = exports.MOBILE_BREAKPOINT = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _bexUtils = require("@wix/bex-utils");
var _mobx = require("mobx");
var _resizeObserverCompat = require("./resizeObserverCompat");
const MOBILE_BREAKPOINT = exports.MOBILE_BREAKPOINT = 768;
class PageState {
  constructor() {
    (0, _defineProperty2.default)(this, "modalsContainerRef", void 0);
    (0, _defineProperty2.default)(this, "stackingRootRef", void 0);
    (0, _defineProperty2.default)(this, "scrollableContentRef", void 0);
    (0, _defineProperty2.default)(this, "_sidePanelOpen", (0, _bexUtils.reactive)(false));
    (0, _defineProperty2.default)(this, "_isMobile", false);
    this.modalsContainerRef = (0, _mobx.makeObservable)({
      current: null
    }, {
      current: _mobx.observable.ref
    });
    this.stackingRootRef = {
      current: null
    };
    this.scrollableContentRef = {
      current: null
    };
    (0, _mobx.makeObservable)(this, {
      _isMobile: _mobx.observable.ref,
      _updateIsMobile: _mobx.action.bound
    });
    this._updateIsMobile();
  }
  _updateIsMobile() {
    if (typeof window !== 'undefined') {
      this._isMobile = window.innerWidth <= MOBILE_BREAKPOINT;
    }
  }
  get isMobile() {
    return this._isMobile;
  }
  set sidePanelOpen(value) {
    this._sidePanelOpen.value = value;
  }
  init() {
    if (typeof window === 'undefined') {
      return;
    }
    this._updateIsMobile();
    const container = this.stackingRootRef.current ?? document.body;
    const disposers = [(0, _resizeObserverCompat.addResizeObserver)(container, this._updateIsMobile)];
    return () => {
      disposers.forEach(d => d == null ? void 0 : d());
    };
  }
}
exports.PageState = PageState;
