// @wix/bex-core/dist/cjs/state/PaginationModeCursor.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.PaginationModeCursor = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _mobx = require("mobx");
class PaginationModeCursor {
  constructor(params) {
    (0, _defineProperty2.default)(this, "collection", void 0);
    this.collection = params.collection;
    (0, _mobx.makeObservable)(this, {
      fetchedLastPage: _mobx.computed,
      setCursor: _mobx.action
    });
  }
  get fetchedLastPage() {
    const {
      query: {
        pagination: {
          cursor
        }
      }
    } = this.collection;
    return cursor == null;
  }
  setCursor(cursor) {
    const {
      query
    } = this.collection;
    query.pagination.cursor = cursor;
  }
  get id() {
    return PaginationModeCursor.id;
  }
}
exports.PaginationModeCursor = PaginationModeCursor;
(0, _defineProperty2.default)(PaginationModeCursor, "id", 'cursor');
