// @wix/bex-core/dist/cjs/state/PaginationMode.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.PaginationMode = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _PaginationModeCursor = require("./PaginationModeCursor");
var _PaginationModeOffset = require("./PaginationModeOffset");
class PaginationMode {
  constructor(collection) {
    (0, _defineProperty2.default)(this, "collection", void 0);
    (0, _defineProperty2.default)(this, "cursorMode", void 0);
    (0, _defineProperty2.default)(this, "offsetMode", void 0);
    this.collection = collection;
    this.cursorMode = new _PaginationModeCursor.PaginationModeCursor({
      collection
    });
    this.offsetMode = new _PaginationModeOffset.PaginationModeOffset({
      collection
    });
  }
  get mode() {
    return this.collection._paginationModeRef.current === 'cursor' ? this.cursorMode : this.offsetMode;
  }
  get fetchedLastPage() {
    return this.mode.fetchedLastPage;
  }
  setCursor(cursor) {
    this.mode.setCursor(cursor);
  }
  get id() {
    return this.mode.id;
  }
}
exports.PaginationMode = PaginationMode;
