// @wix/bex-core/dist/cjs/state/Pagination.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.initialCursor = exports.Pagination = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _mobx = require("mobx");
const initialCursor = exports.initialCursor = {};
class Pagination {
  constructor(params) {
    (0, _defineProperty2.default)(this, "limit", void 0);
    (0, _defineProperty2.default)(this, "page", 1);
    (0, _defineProperty2.default)(this, "cursor", initialCursor);
    this.limit = params.limit ?? 100;
    (0, _mobx.makeObservable)(this, {
      limit: _mobx.observable.ref,
      page: _mobx.observable.ref,
      cursor: _mobx.observable.ref,
      offset: _mobx.computed,
      changeLimit: _mobx.action,
      nextPage: _mobx.action,
      prevPage: _mobx.action,
      moveToPage: _mobx.action
    });
  }
  changeOffset(offset) {
    this.page = Math.floor(offset / this.limit) + 1;
  }
  effectiveLimit(maxLimit = Infinity) {
    const {
      limit
    } = this;
    return limit > 0 ? limit : maxLimit;
  }
  eagerLimitQuery() {
    const limit = this.effectiveLimit();
    return limit === Infinity ? {} : this.limitQuery();
  }
  limitQuery(maxLimit) {
    const {
      page
    } = this;
    const limit = this.effectiveLimit(maxLimit);
    return {
      limit,
      offset: limit * (page - 1)
    };
  }
  toQueryString() {
    return {
      page: String(this.page),
      limit: String(this.limit)
    };
  }
  changeLimit(limit) {
    this.limit = limit;
  }
  nextPage() {
    this.page += 1;
  }
  prevPage() {
    this.page -= 1;
  }
  moveToPage(page = 0) {
    this.page = page;
  }
  get offset() {
    const {
      limit,
      page
    } = this;
    return limit * (page - 1);
  }
  pageBounds(page) {
    const {
      limit
    } = this;
    const from = limit * (page - 1);
    return {
      from,
      to: from + limit
    };
  }
  pageOf(index) {
    return Math.floor(index / this.limit) + 1;
  }
}
exports.Pagination = Pagination;
