// @wix/bex-core/dist/cjs/state/PromiseState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.PromiseState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _model = require("../model");
var _mobx = require("mobx");
var _once = _interopRequireDefault(require("lodash/once"));
class PromiseState {
  constructor(params) {
    (0, _defineProperty2.default)(this, "createPromise", void 0);
    (0, _defineProperty2.default)(this, "status", {
      ..._model.queryStatus.idle,
      error: null,
      isFetching: false
    });
    (0, _defineProperty2.default)(this, "fetchOnce", (0, _once.default)(() => this.fetch()));
    this.createPromise = params.createPromise;
    (0, _mobx.makeObservable)(this, {
      status: _mobx.observable.ref,
      fetch: _mobx.action
    });
  }
  async fetch() {
    try {
      this.status = {
        ..._model.queryStatus.loading,
        error: null,
        isFetching: true
      };
      const data = await this.createPromise();
      (0, _mobx.runInAction)(() => {
        this.status = {
          ..._model.queryStatus.success,
          error: null,
          isFetching: false,
          data
        };
      });
    } catch (error) {
      (0, _mobx.runInAction)(() => {
        this.status = {
          ..._model.queryStatus.error,
          error,
          isFetching: false
        };
      });
    }
  }
}
exports.PromiseState = PromiseState;
