// @wix/bex-core/dist/cjs/state/SelectMapState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.SelectMapState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _mobx = require("mobx");
var _events = require("events");
class SelectMapState {
  constructor() {
    (0, _defineProperty2.default)(this, "map", void 0);
    (0, _defineProperty2.default)(this, "events", new _events.EventEmitter());
    this.map = _mobx.observable.map([], {
      deep: false
    });
  }
  set(key, value) {
    if (this.events.listeners('isDisabled').some(listener => listener(value))) {
      return;
    }
    (0, _mobx.runInAction)(() => {
      this.map.set(key, value);
    });
  }
  forceSet(key, value) {
    (0, _mobx.runInAction)(() => {
      this.map.set(key, value);
    });
  }
  get(key) {
    return this.map.get(key);
  }
  values() {
    return this.map.values();
  }
  entries() {
    return this.map.entries();
  }
  clear() {
    return this.map.clear();
  }
}
exports.SelectMapState = SelectMapState;
