// @wix/bex-core/dist/cjs/state/SelectionStatusWixPatternsPreset.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.selectionStatusWixPatternsPreset = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
class SelectionStatusWixPatternsPreset {
  constructor() {
    (0, _defineProperty2.default)(this, "unchecked", {
      id: 'unchecked',
      toggle: () => this.checked,
      start: () => this.unchecked,
      final: () => this.checked,
      isChecked: false,
      isIndeterminate: false
    });
    (0, _defineProperty2.default)(this, "checked", {
      id: 'checked',
      toggle: () => this.unchecked,
      start: () => this.unchecked,
      final: () => this.checked,
      isChecked: true,
      isIndeterminate: false
    });
    (0, _defineProperty2.default)(this, "indeterminateUnchecked", {
      id: 'indeterminateUnchecked',
      toggle: () => this.indeterminate,
      start: () => this.indeterminateUnchecked,
      final: () => this.indeterminateChecked,
      isChecked: false,
      isIndeterminate: false
    });
    (0, _defineProperty2.default)(this, "indeterminate", {
      id: 'indeterminate',
      toggle: () => this.indeterminateChecked,
      start: () => this.indeterminateUnchecked,
      final: () => this.indeterminateChecked,
      isChecked: true,
      isIndeterminate: true
    });
    (0, _defineProperty2.default)(this, "indeterminateChecked", {
      id: 'indeterminateChecked',
      toggle: () => this.indeterminateUnchecked,
      start: () => this.indeterminateUnchecked,
      final: () => this.indeterminateChecked,
      isChecked: true,
      isIndeterminate: false
    });
  }
}
const selectionStatusWixPatternsPreset = exports.selectionStatusWixPatternsPreset = new SelectionStatusWixPatternsPreset();
