// @wix/bex-core/dist/cjs/state/SidePanelState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.SidePanelWidth = exports.SidePanelState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _TransitionState = require("./TransitionState");
var _ReturnFocus = require("./ReturnFocus");
// FIXME: Move to the class property once the state could be read in PageWrapper.
const SidePanelWidth = exports.SidePanelWidth = 420;
class SidePanelState {
  constructor() {
    (0, _defineProperty2.default)(this, "transition", new _TransitionState.TransitionState());
    (0, _defineProperty2.default)(this, "focus", new _ReturnFocus.ReturnFocus());
  }
  get isOpen() {
    return this.transition.show;
  }
  get currentPanelId() {
    return this.transition.data;
  }
  close() {
    this.transition.clickClose();
    this.focus.onClose();
  }
  _open(id) {
    this.transition.setSync(id);
    this.focus.onOpen();
  }
  init() {
    const disposers = [this.focus.init()];
    return () => {
      disposers.forEach(d => d());
    };
  }
}
exports.SidePanelState = SidePanelState;
