// @wix/bex-core/dist/cjs/state/snapshot.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.Snapshot = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _ValueBox = require("./ValueBox");
var _mobx = require("mobx");
class Snapshot {
  constructor(params) {
    (0, _defineProperty2.default)(this, "box", void 0);
    (0, _defineProperty2.default)(this, "expression", void 0);
    (0, _defineProperty2.default)(this, "effect", void 0);
    this.box = new _ValueBox.ValueBox((0, _mobx.runInAction)(() => params.effect()));
    this.expression = params.expression;
    this.effect = params.effect;
    (0, _mobx.makeObservable)(this, {
      init: _mobx.action,
      forceUpdate: _mobx.action
    });
  }
  get value() {
    return this.box.value;
  }
  set value(v) {
    this.box.value = v;
  }
  forceUpdate() {
    this.value = this.effect();
  }
  init() {
    const {
      box,
      effect
    } = this;
    const dispose = (0, _mobx.reaction)(this.expression, () => {
      box.set(effect());
    });
    return () => {
      dispose();
    };
  }
}
exports.Snapshot = Snapshot;
