// @wix/bex-core/dist/cjs/state/TaskState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.TaskState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _model = require("../model");
var _mobx = require("mobx");
class TaskState {
  constructor() {
    (0, _defineProperty2.default)(this, "status", {
      ..._model.queryStatus.idle
    });
    (0, _defineProperty2.default)(this, "_fn", void 0);
    (0, _defineProperty2.default)(this, "_onError", void 0);
    (0, _mobx.makeObservable)(this, {
      status: _mobx.observable.ref,
      run: _mobx.action,
      runOnce: _mobx.action,
      reset: _mobx.action
    });
  }
  runOnce(fn, onError) {
    if (fn) {
      this._fn = fn;
    }
    if (onError) {
      this._onError = onError;
    }
    if (!this._fn) {
      return;
    }
    const {
      status
    } = this;
    if (status.isLoading || status.isSuccess) {
      return;
    }
    return this.run(this._fn, this._onError);
  }
  get errorStatus() {
    const {
      status
    } = this;
    return status.isError ? status : undefined;
  }
  reset() {
    this.status = {
      ..._model.queryStatus.idle
    };
  }
  async run(fn, onError) {
    const promise = fn();
    this.status = {
      ..._model.queryStatus.loading,
      promise
    };
    try {
      const data = await promise;
      (0, _mobx.runInAction)(() => {
        if (promise === this.status.promise) {
          this.status = {
            ..._model.queryStatus.success,
            error: null,
            data
          };
        }
      });
    } catch (err) {
      onError == null || onError(err);
      const error = err;
      (0, _mobx.runInAction)(() => {
        if (promise === this.status.promise) {
          this.status = {
            ..._model.queryStatus.error,
            error
          };
        }
      });
    }
  }
}
exports.TaskState = TaskState;
