// @wix/bex-core/dist/cjs/state/ToggleState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.ToggleState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _mobx = require("mobx");
var _events = require("events");
class ToggleState {
  constructor(initialValue = false) {
    (0, _defineProperty2.default)(this, "value", void 0);
    (0, _defineProperty2.default)(this, "events", new _events.EventEmitter());
    this.value = initialValue;
    (0, _mobx.makeObservable)(this, {
      value: _mobx.observable,
      isOn: _mobx.computed,
      isOff: _mobx.computed,
      toggle: _mobx.action.bound,
      setOff: _mobx.action.bound,
      setOn: _mobx.action.bound
    });
  }
  get isOn() {
    return this.value;
  }
  get isOff() {
    return !this.value;
  }
  toggle(value = !this.value) {
    if (this.value !== value) {
      this.value = value;
      this.events.emit('toggle');
    }
  }
  setOff() {
    this.toggle(false);
    // return new Promise<void>((resolve) => {
    //   this.events.once('afterClose', resolve);
    // });
  }
  setOn() {
    this.toggle(true);
    this.events.emit('open');
  }
}
exports.ToggleState = ToggleState;
