// @wix/bex-core/dist/cjs/state/TransitionState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.TransitionState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _events = require("events");
var _mobx = require("mobx");
class TransitionState {
  constructor(initialValue = false) {
    (0, _defineProperty2.default)(this, "status", void 0);
    (0, _defineProperty2.default)(this, "show", void 0);
    (0, _defineProperty2.default)(this, "data", void 0);
    (0, _defineProperty2.default)(this, "nextData", void 0);
    (0, _defineProperty2.default)(this, "events", new _events.EventEmitter());
    this.show = initialValue;
    this.status = initialValue ? 'entered' : 'exited';
    this.data = null;
    this.nextData = null;
    this.events.on('exited', () => {
      const {
        events
      } = this;
      events.emit('exitedAfterNewData');
      events.emit('exitedAfterClickClose');
    });
    (0, _mobx.makeObservable)(this, {
      show: _mobx.observable.ref,
      status: _mobx.observable.ref,
      data: _mobx.observable.ref,
      setShow: _mobx.action,
      setStatus: _mobx.action,
      clickClose: _mobx.action,
      set: _mobx.action.bound,
      setSync: _mobx.action.bound
    });
  }
  setShow(value) {
    this.show = value;
  }
  setStatus(status) {
    this.status = status;
    this.events.emit(status);
  }
  set(data) {
    if (data == null) {
      this.events.emit('dismissed', {
        dismissMode: 'automatic'
      });
      this.show = false;
      return;
    }
    this.transitionToNewData(data);
  }

  // does not transition if data already exists
  setSync(data) {
    if (this.show && data != null) {
      this.data = data;
      return;
    }
    this.transitionToNewData(data);
  }
  clickClose(onAfterClose) {
    if (!this.show) {
      return;
    }
    const {
      events
    } = this;
    this.show = false;
    this.nextData = undefined;
    events.removeAllListeners('exitedAfterClickClose');
    events.once('exitedAfterClickClose', (0, _mobx.action)(() => {
      events.emit('dismissed', {
        dismissMode: 'manual'
      });
      if (!this.show && typeof this.nextData === 'undefined') {
        this.data = null;
        onAfterClose == null || onAfterClose();
      }
    }));
  }
  transitionToNewData(data) {
    const {
      events
    } = this;
    this.nextData = data;
    if (this.show) {
      this.show = false;
      events.removeAllListeners('exitedAfterNewData');
      events.once('exitedAfterNewData', (0, _mobx.action)(() => {
        events.emit('dismissed', {
          dismissMode: 'automatic'
        });
        if (!this.show && this.nextData === data) {
          this.show = true;
          this.data = data;
        }
      }));
    } else {
      this.show = true;
      this.data = data;
    }
  }
}
exports.TransitionState = TransitionState;
