// @wix/bex-core/dist/cjs/state/warmUpPickerStandalone.js
"use strict";

exports.__esModule = true;
exports.warmUpPickerStandalone = warmUpPickerStandalone;
var _services = require("../services");
var _model = require("../model");
var _v = require("@wix/bex-utils/@wix/bi-logger-os-data/v2");
var _util = require("../util");
function warmUpPickerStandalone(wixPatternsEssentials, params) {
  const {
    errorMonitor,
    createEssentials
  } = wixPatternsEssentials;
  const environment = (0, _model.wixPatternsEnvironment)(wixPatternsEssentials.environment);
  (0, _services.prefetchTranslations)({
    errorMonitor
  })(environment.language);
  const {
    queryName
  } = params;
  const fqdn = (0, _util.parseFqdn)(queryName);
  const {
    biLoggerFactory
  } = createEssentials({});
  if (biLoggerFactory) {
    const bi = biLoggerFactory().updateDefaults({
      artifactId: environment.artifactId,
      componentName: environment.componentName,
      cairoVersion: environment.uiLibVersion,
      ...fqdn,
      type: 'standalone',
      componentType: 'Picker Standalone'
    }).logger();
    bi == null || bi.report((0, _v.loadStart)({
      ...{
        maxItems: 1
      },
      url: (0, _util.fixDomainForBI)(window.location.href),
      featuresAvailability: JSON.stringify({}),
      ...(environment.providerType === 'bm' && {
        _hostingPlatform: 'business-manager'
      })
    }));
  }
}
