// @wix/bex-core/dist/cjs/state/WidgetsFormState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.WidgetsFormState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _mobx = require("mobx");
class WidgetsFormState {
  constructor() {
    (0, _defineProperty2.default)(this, "_widgets", _mobx.observable.map(new Map()));
    (0, _mobx.makeObservable)(this, {
      registerWidget: _mobx.action.bound,
      isDirty: _mobx.computed
    });
  }
  registerWidget(key, state) {
    this._widgets.set(key, state);
  }
  async validate() {
    const data = {};
    let isValid = true;
    for (const [key, widget] of this._widgets.entries()) {
      const widgetData = await widget.validate();
      if (!widgetData.isValid) {
        isValid = false;
        return {
          isValid: false,
          data: {}
        };
      }
      data[key] = widgetData.values;
    }
    return {
      isValid,
      data
    };
  }
  get isDirty() {
    return [...this._widgets.values()].some(w => w.isDirty);
  }
  get(key) {
    return this._widgets.get(key);
  }
}
exports.WidgetsFormState = WidgetsFormState;
