// @wix/auto_sdk_events_schedule/build/es/index.typings.mjs
// src/events-schedule-v1-schedule-item-schedule.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/events-schedule-v1-schedule-item-schedule.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKDurationToRESTDuration } from "@wix/sdk-runtime/transformations/duration";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixEventsScheduleScheduleManagementUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/wix-events-agenda",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    _: [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "editor-flow.wixapps.net": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/events/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "api._base_domain_": [
      {
        srcPath: "/events/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_events_schedule";
function listScheduleItems(payload) {
  function __listScheduleItems({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "startingFrom" }, { path: "startingBefore" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "GET",
      methodFqn: "wix.events.schedule.ScheduleManagement.ListScheduleItems",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleScheduleManagementUrl({
        protoPath: "/v1/schedule",
        data: serializedData,
        host
      }),
      params: toURLSearchParams(serializedData),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "items.createdDate" },
            { path: "items.updatedDate" },
            { path: "items.timeSlot.start" },
            { path: "items.timeSlot.end" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listScheduleItems;
}
function queryScheduleItems(payload) {
  function __queryScheduleItems({ host }) {
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "POST",
      methodFqn: "wix.events.schedule.ScheduleManagement.QueryScheduleItems",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleScheduleManagementUrl({
        protoPath: "/v1/schedule/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "items.createdDate" },
            { path: "items.updatedDate" },
            { path: "items.timeSlot.start" },
            { path: "items.timeSlot.end" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryScheduleItems;
}
function getScheduleItem(payload) {
  function __getScheduleItem({ host }) {
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "GET",
      methodFqn: "wix.events.schedule.ScheduleManagement.GetScheduleItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleScheduleManagementUrl({
        protoPath: "/v1/schedule/{itemId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "item.createdDate" },
            { path: "item.updatedDate" },
            { path: "item.timeSlot.start" },
            { path: "item.timeSlot.end" },
            { path: "draft.createdDate" },
            { path: "draft.updatedDate" },
            { path: "draft.timeSlot.start" },
            { path: "draft.timeSlot.end" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getScheduleItem;
}
function addScheduleItem(payload) {
  function __addScheduleItem({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "item.timeSlot.start" }, { path: "item.timeSlot.end" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "POST",
      methodFqn: "wix.events.schedule.ScheduleManagement.AddScheduleItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleScheduleManagementUrl({
        protoPath: "/v1/schedule/draft",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "item.createdDate" },
            { path: "item.updatedDate" },
            { path: "item.timeSlot.start" },
            { path: "item.timeSlot.end" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __addScheduleItem;
}
function updateScheduleItem(payload) {
  function __updateScheduleItem({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fields" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "item.timeSlot.start" }, { path: "item.timeSlot.end" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "PATCH",
      methodFqn: "wix.events.schedule.ScheduleManagement.UpdateScheduleItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleScheduleManagementUrl({
        protoPath: "/v1/schedule/draft/{itemId}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "item.createdDate" },
            { path: "item.updatedDate" },
            { path: "item.timeSlot.start" },
            { path: "item.timeSlot.end" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateScheduleItem;
}
function deleteScheduleItem(payload) {
  function __deleteScheduleItem({ host }) {
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "DELETE",
      methodFqn: "wix.events.schedule.ScheduleManagement.DeleteScheduleItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleScheduleManagementUrl({
        protoPath: "/v1/schedule/draft/items",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteScheduleItem;
}
function discardDraft(payload) {
  function __discardDraft({ host }) {
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "DELETE",
      methodFqn: "wix.events.schedule.ScheduleManagement.DiscardDraft",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleScheduleManagementUrl({
        protoPath: "/v1/schedule/draft",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __discardDraft;
}
function publishDraft(payload) {
  function __publishDraft({ host }) {
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "POST",
      methodFqn: "wix.events.schedule.ScheduleManagement.PublishDraft",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleScheduleManagementUrl({
        protoPath: "/v1/schedule/publish",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __publishDraft;
}
function rescheduleDraft(payload) {
  function __rescheduleDraft({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKDurationToRESTDuration,
        paths: [{ path: "timeSlotOffset" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "POST",
      methodFqn: "wix.events.schedule.ScheduleManagement.RescheduleDraft",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleScheduleManagementUrl({
        protoPath: "/v1/schedule/draft/reschedule",
        data: serializedData,
        host
      }),
      data: serializedData
    };
    return metadata;
  }
  return __rescheduleDraft;
}

// src/events-schedule-v1-schedule-item-schedule.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var ScheduleStatus = /* @__PURE__ */ ((ScheduleStatus2) => {
  ScheduleStatus2["SCHEDULED"] = "SCHEDULED";
  ScheduleStatus2["CANCELED"] = "CANCELED";
  return ScheduleStatus2;
})(ScheduleStatus || {});
var StateFilter = /* @__PURE__ */ ((StateFilter2) => {
  StateFilter2["PUBLISHED"] = "PUBLISHED";
  StateFilter2["DRAFT"] = "DRAFT";
  StateFilter2["VISIBLE"] = "VISIBLE";
  StateFilter2["HIDDEN"] = "HIDDEN";
  return StateFilter2;
})(StateFilter || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
async function listScheduleItems2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    eventId: options?.eventId,
    state: options?.state,
    startingFrom: options?.startingFrom,
    startingBefore: options?.startingBefore,
    offset: options?.offset,
    limit: options?.limit,
    facet: options?.facet,
    itemId: options?.itemId,
    tag: options?.tag,
    stageName: options?.stageName,
    paging: options?.paging
  });
  const reqOpts = listScheduleItems(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          eventId: "$[0].eventId",
          state: "$[0].state",
          startingFrom: "$[0].startingFrom",
          startingBefore: "$[0].startingBefore",
          offset: "$[0].offset",
          limit: "$[0].limit",
          facet: "$[0].facet",
          itemId: "$[0].itemId",
          tag: "$[0].tag",
          stageName: "$[0].stageName",
          paging: "$[0].paging"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryScheduleItems2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryScheduleItems(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [])
      );
      return {
        items: transformedData?.items,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function getScheduleItem2(itemId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    itemId,
    eventId: options?.eventId,
    includeDraft: options?.includeDraft
  });
  const reqOpts = getScheduleItem(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.item;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          itemId: "$[0]",
          eventId: "$[1].eventId",
          includeDraft: "$[1].includeDraft"
        },
        singleArgumentUnchanged: false
      },
      ["itemId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function addScheduleItem2(eventId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    eventId,
    item: options?.item
  });
  const reqOpts = addScheduleItem(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { eventId: "$[0]", item: "$[1].item" },
        singleArgumentUnchanged: false
      },
      ["eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateScheduleItem2(itemId, eventId, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    itemId,
    eventId,
    item: options?.item,
    fields: options?.fields
  });
  const reqOpts = updateScheduleItem(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          itemId: "$[0]",
          eventId: "$[1]",
          item: "$[2].item",
          fields: "$[2].fields"
        },
        singleArgumentUnchanged: false
      },
      ["itemId", "eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteScheduleItem2(eventId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    eventId,
    itemIds: options?.itemIds
  });
  const reqOpts = deleteScheduleItem(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { eventId: "$[0]", itemIds: "$[1].itemIds" },
        singleArgumentUnchanged: false
      },
      ["eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function discardDraft2(eventId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ eventId });
  const reqOpts = discardDraft(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { eventId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["eventId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function publishDraft2(eventId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ eventId });
  const reqOpts = publishDraft(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { eventId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["eventId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function rescheduleDraft2(eventId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    eventId,
    timeZoneId: options?.timeZoneId,
    timeSlotOffset: options?.timeSlotOffset
  });
  const reqOpts = rescheduleDraft(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          eventId: "$[0]",
          timeZoneId: "$[1].timeZoneId",
          timeSlotOffset: "$[1].timeSlotOffset"
        },
        singleArgumentUnchanged: false
      },
      ["eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
export {
  ScheduleStatus,
  SortOrder,
  StateFilter,
  addScheduleItem2 as addScheduleItem,
  deleteScheduleItem2 as deleteScheduleItem,
  discardDraft2 as discardDraft,
  getScheduleItem2 as getScheduleItem,
  listScheduleItems2 as listScheduleItems,
  publishDraft2 as publishDraft,
  queryScheduleItems2 as queryScheduleItems,
  rescheduleDraft2 as rescheduleDraft,
  updateScheduleItem2 as updateScheduleItem
};
