// @wix/auto-patterns/dist/cjs/components/AutoPatternsGrid/AutoPatternsGrid.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.AutoPatternsGrid = void 0;
var _react = _interopRequireDefault(require("react"));
var _patterns = require("@wix/patterns");
var _hooks = require("../../hooks");
var _jsxFileName = "/home/builduser/work/73d19ce3378ce4dd/packages/auto-patterns/dist/cjs/components/AutoPatternsGrid/AutoPatternsGrid.tsx";
const AutoPatternsGrid = ({
  configuration
}) => {
  const {
    state,
    search,
    title,
    emptyState,
    filters,
    actionCell,
    renderItem,
    imagePlacement,
    preset
  } = (0, _hooks.useGridFeatures)(configuration);
  return /*#__PURE__*/_react.default.createElement(_patterns.Grid, {
    dataHook: "auto-patterns-grid",
    search: search,
    title: title,
    state: state,
    actionCell: actionCell,
    filters: filters,
    emptyState: emptyState,
    imagePlacement: imagePlacement,
    preset: preset,
    renderItem: renderItem,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 26,
      columnNumber: 5
    }
  });
};
exports.AutoPatternsGrid = AutoPatternsGrid;
