// @wix/patterns/dist/cjs/state/NestedTableState/createNestedOptimisticActions.js
"use strict";

exports.__esModule = true;
exports.createNestedOptimisticActions = createNestedOptimisticActions;
var _bexCore = require("@wix/bex-core");
var _createParentIdPredicate = require("./createParentIdPredicate");
function createNestedOptimisticActions(container, collection, state, params) {
  const parentIdPredicate = (0, _createParentIdPredicate.createParentIdPredicate)(state);
  const optimisticActions = collection._optimisticActions ?? new _bexCore.CollectionOptimisticActions({
    container,
    collection,
    orderBy: () => [],
    ...params,
    predicate: query => {
      var _state$levelDescripto;
      const originalPredicateForQuery = (params == null || params.predicate == null ? void 0 : params.predicate(query)) ?? (state == null || (_state$levelDescripto = state.levelDescriptor) == null || _state$levelDescripto.optimisticActionsPredicate == null ? void 0 : _state$levelDescripto.optimisticActionsPredicate(query));
      const parentIdPredicateForQuery = parentIdPredicate(query);
      return item => parentIdPredicateForQuery(item) && ((originalPredicateForQuery == null ? void 0 : originalPredicateForQuery(item)) ?? true);
    },
    cacheNamespace: state == null ? void 0 : state.parentId,
    sequences: state == null ? void 0 : state.sequences,
    query: state == null ? void 0 : state.query
  });
  optimisticActions._commitMode = false;
  return optimisticActions;
}
