// @wix/wix-ui-test-utils/dist/esm/react-helpers/helpers.js
import * as React from 'react';
import { Simulate } from 'react-dom/test-utils';
export const isClassExists = (element, className) => !!element && !!element.className.match(new RegExp('\\b' + className + '\\b'));
export const sleep = (ms) => {
    return new Promise((resolve) => setTimeout(resolve, ms));
};
// HOC that makes underlying component "controlled"
/** @deprecated */
export function makeControlled(Component) {
    var _a;
    return _a = class ControlledComponent extends React.Component {
            constructor() {
                super(...arguments);
                this.state = { value: this.props.value };
                this._onChange = (e) => {
                    const { onChange } = this.props;
                    this.setState({
                        value: e.target.value,
                    });
                    onChange && onChange(e);
                };
            }
            render() {
                const bindedPropMethods = {};
                for (const propName of Object.keys(this.props)) {
                    const propValue = this.props[propName];
                    if (typeof propValue === 'function') {
                        bindedPropMethods[propName] = this.props[propName].bind(this);
                    }
                }
                return (React.createElement(Component, { ...this.props, ...bindedPropMethods, value: this.state.value, onChange: this._onChange }));
            }
        },
        _a.defaultProps = {
            value: '',
        },
        _a;
}
export const reactEventTrigger = () => Simulate;
