// pro-gallery-lib/dist/cjs/common/defaultOptions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.populateWithDefaultOptions = void 0;
const tslib_1 = require("tslib");
const optionsMap_js_1 = tslib_1.__importDefault(require("../core/helpers/optionsMap.js"));
const optionsUtils_js_1 = require("../core/helpers/optionsUtils.js");
const index_js_1 = tslib_1.__importDefault(require("./constants/index.js"));
const defaultOptions = (0, optionsUtils_js_1.flattenObject)({
    layoutParams: {
        crop: {
            ratios: [1],
            method: index_js_1.default[optionsMap_js_1.default.layoutParams.crop.method].FILL,
            enable: false,
            enableSmartCrop: false,
            cropOnlyFill: false,
            alignment: index_js_1.default[optionsMap_js_1.default.layoutParams.crop.alignment].CENTER,
        },
        structure: {
            galleryLayout: -1,
            galleryRatio: { value: 0, includeExternalInfo: false },
            scrollDirection: index_js_1.default[optionsMap_js_1.default.layoutParams.structure.scrollDirection].VERTICAL,
            gallerySpacing: 0,
            itemSpacing: 10,
            enableStreching: true,
            responsiveMode: index_js_1.default[optionsMap_js_1.default.layoutParams.structure.responsiveMode].FIT_TO_SCREEN,
            scatter: {
                randomScatter: 0,
                manualScatter: '',
            },
            layoutOrientation: index_js_1.default[optionsMap_js_1.default.layoutParams.structure.layoutOrientation].HORIZONTAL,
            groupsOrder: index_js_1.default[optionsMap_js_1.default.layoutParams.structure.groupsOrder].BY_HEIGHT,
            numberOfGridRows: 1,
            numberOfColumns: 3,
            columnRatios: [],
        },
        groups: {
            groupByOrientation: true,
            numberOfGroupsPerRow: 0,
            density: 0.8,
            groupSize: 3,
            allowedGroupTypes: [
                index_js_1.default[optionsMap_js_1.default.layoutParams.groups.allowedGroupTypes]['1'],
                index_js_1.default[optionsMap_js_1.default.layoutParams.groups.allowedGroupTypes]['2h'],
                index_js_1.default[optionsMap_js_1.default.layoutParams.groups.allowedGroupTypes]['2v'],
                index_js_1.default[optionsMap_js_1.default.layoutParams.groups.allowedGroupTypes]['3t'],
                index_js_1.default[optionsMap_js_1.default.layoutParams.groups.allowedGroupTypes]['3b'],
                index_js_1.default[optionsMap_js_1.default.layoutParams.groups.allowedGroupTypes]['3l'],
                index_js_1.default[optionsMap_js_1.default.layoutParams.groups.allowedGroupTypes]['3r'],
            ],
            repeatingGroupTypes: [],
        },
        thumbnails: {
            size: 120,
            spacing: 4,
            marginToGallery: 0,
            enable: false,
            position: index_js_1.default[optionsMap_js_1.default.layoutParams.thumbnails.position].OUTSIDE_GALLERY,
            alignment: index_js_1.default[optionsMap_js_1.default.layoutParams.thumbnails.alignment].BOTTOM,
        },
        navigationArrows: {
            enable: true,
            position: index_js_1.default[optionsMap_js_1.default.layoutParams.navigationArrows.position].ON_GALLERY,
            padding: 23,
            size: 23,
            verticalAlignment: index_js_1.default[optionsMap_js_1.default.layoutParams.navigationArrows.verticalAlignment].ITEM_CENTER,
            type: index_js_1.default[optionsMap_js_1.default.layoutParams.navigationArrows.type].DEFAULT_ARROW,
            container: {
                type: index_js_1.default[optionsMap_js_1.default.layoutParams.navigationArrows.container.type].SHADOW,
                backgroundColor: '',
                borderRadius: -1,
            },
        },
        targetItemSize: {
            unit: index_js_1.default[optionsMap_js_1.default.layoutParams.targetItemSize.unit].SMART,
            value: 30,
            minimum: 120,
        },
        info: {
            sizeUnits: index_js_1.default[optionsMap_js_1.default.layoutParams.info.sizeUnits].PERCENT,
            width: 50,
            height: 200,
            spacing: 10,
            layout: index_js_1.default[optionsMap_js_1.default.layoutParams.info.layout].NO_BACKGROUND,
            // widthByPercent: 50, //I want this to be in the width just like we did the overlaySize
            placement: index_js_1.default[optionsMap_js_1.default.layoutParams.info.placement].OVERLAY,
            border: {
                width: 0,
                radius: 0,
                color: '',
            },
        },
    },
    behaviourParams: {
        item: {
            clickAction: index_js_1.default[optionsMap_js_1.default.behaviourParams.item.clickAction].NOTHING,
            video: {
                playTrigger: index_js_1.default[optionsMap_js_1.default.behaviourParams.item.video.playTrigger].HOVER,
                loop: true,
                volume: 0,
                enableControls: false,
                speed: 1,
                enablePlayButton: false,
                enableThumbnailsPlayButton: false,
                enablePlaceholder: true,
            },
            overlay: {
                hoveringBehaviour: index_js_1.default[optionsMap_js_1.default.behaviourParams.item.overlay.hoveringBehaviour].APPEARS,
                hoverAnimation: index_js_1.default[optionsMap_js_1.default.behaviourParams.item.overlay.hoverAnimation].NO_EFFECT,
                position: index_js_1.default[optionsMap_js_1.default.behaviourParams.item.overlay.position].LEFT,
                size: 100,
                sizeUnits: index_js_1.default[optionsMap_js_1.default.behaviourParams.item.overlay.sizeUnits].PERCENT,
                padding: 0,
            },
            content: {
                hoverAnimation: index_js_1.default[optionsMap_js_1.default.behaviourParams.item.content.hoverAnimation].NO_EFFECT,
                loader: index_js_1.default[optionsMap_js_1.default.behaviourParams.item.content.loader].BLUR,
                placementAnimation: index_js_1.default[optionsMap_js_1.default.behaviourParams.item.content.placementAnimation].NO_EFFECT,
                magnificationValue: 2,
                tiltAngleValue: -4,
            },
            secondaryMedia: {
                trigger: index_js_1.default[optionsMap_js_1.default.behaviourParams.item.secondaryMedia.trigger].OFF,
                behaviour: index_js_1.default[optionsMap_js_1.default.behaviourParams.item.secondaryMedia.behaviour].APPEARS,
            },
            threeDimensionalScene: {
                controls: {
                    enableAutoRotate: false,
                    enablePan: true,
                    enableRotate: true,
                    enableZoom: false,
                },
                transform: {
                    rotation: 'x0y0z0',
                    scale: 'x1y1z1',
                    position: 'x0y0z0',
                },
                playTrigger: index_js_1.default[optionsMap_js_1.default.behaviourParams.item.threeDimensionalScene.playTrigger].HOVER,
                enablePlayButton: false,
                enableThumbnailsPlayButton: false,
                enablePlaceholder: true,
            },
        },
        gallery: {
            blockContextMenu: true,
            layoutDirection: index_js_1.default[optionsMap_js_1.default.behaviourParams.gallery.layoutDirection].LEFT_TO_RIGHT,
            scrollAnimation: index_js_1.default[optionsMap_js_1.default.behaviourParams.gallery.scrollAnimation].NO_EFFECT,
            enableIndexingShareLinks: true,
            horizontal: {
                enableScrollSnap: false,
                navigationDuration: 400,
                blockScroll: false,
                loop: false,
                slideTransition: index_js_1.default[optionsMap_js_1.default.behaviourParams.gallery.horizontal.slideTransition].EASE,
                slideAnimation: index_js_1.default[optionsMap_js_1.default.behaviourParams.gallery.horizontal.slideAnimation].SCROLL,
                autoSlide: {
                    behaviour: index_js_1.default[optionsMap_js_1.default.behaviourParams.gallery.horizontal.autoSlide.behaviour].OFF,
                    interval: 4,
                    pauseOnHover: true,
                    speed: 200,
                },
                slideshowInfo: {
                    buttonsAlignment: index_js_1.default[optionsMap_js_1.default.behaviourParams.gallery.horizontal.slideshowInfo.buttonsAlignment].CENTER,
                    enablePlayButton: false,
                    enableCounter: false,
                },
            },
            vertical: {
                loadMore: {
                    enable: false,
                    text: '',
                    amount: index_js_1.default[optionsMap_js_1.default.behaviourParams.gallery.vertical.loadMore.amount].ALL,
                },
            },
        },
    },
    stylingParams: {
        itemBorderWidth: 0,
        itemBorderRadius: 0,
        itemEnableShadow: false,
        itemBorderColor: '',
        itemShadowBlur: 20,
        itemShadowDirection: 135,
        itemShadowSize: 10,
        itemShadowOpacityAndColor: '',
        arrowsColor: '',
    },
    //layouter API params
    fixedColumns: 0, // determine the number of columns regardless of the screen size (use 0 to ignore)
});
function populateWithDefaultOptions(options) {
    const mergedOptions = Object.assign({}, defaultOptions, options);
    Object.keys(mergedOptions).forEach((key) => {
        if (typeof mergedOptions[key] === 'undefined') {
            mergedOptions[key] = defaultOptions[key];
        }
    });
    return mergedOptions;
}
exports.populateWithDefaultOptions = populateWithDefaultOptions;
exports.default = defaultOptions;
