// @wix/headless-stores/cjs/dist/react/CategoryList.js
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import React from 'react';
import * as CoreCategoryList from './core/CategoryList.js';
import * as Category from './Category.js';
import { AsChildSlot } from '@wix/headless-utils/react';
import { DataComponentTags } from '../data-component-tags.js';
var TestIds;
(function (TestIds) {
    TestIds["categoryListRoot"] = "category-list";
    TestIds["categoryRepeater"] = "category-repeater";
})(TestIds || (TestIds = {}));
/**
 * Root container that provides category list context to all child components.
 * This component sets up the necessary services for managing categories list state.
 * Automatically connects to ProductList filtering when available.
 *
 * @order 1
 * @component
 * @example
 * ```tsx
 * import { CategoryList } from '@wix/headless-stores/react';
 *
 * function CategoriesPage() {
 *   return (
 *     <CategoryList.Root categoriesListConfig={categoriesListConfig}>
 *       <CategoryList.Loading>Loading...</CategoryList.Loading>
 *       <CategoryList.CategoryRepeater>
 *         <Category.Label />
 *       </CategoryList.CategoryRepeater>
 *     </CategoryList.Root>
 *   );
 * }
 * ```
 */
export const Root = (props) => {
    const { categories, categoriesListConfig, children, emptyState } = props;
    // Create service config, prioritizing categoriesListConfig, then categories prop
    const serviceConfig = categoriesListConfig || (categories ? { categories } : { categories: [] });
    return (_jsx(CoreCategoryList.Root, { categoriesListConfig: serviceConfig, children: _jsxs(AsChildSlot, { "data-testid": TestIds.categoryListRoot, "data-component-tag": DataComponentTags.categoryListRoot, children: [children, emptyState && (_jsx(CoreCategoryList.EmptyState, { children: emptyState }))] }) }));
};
Root.displayName = 'CategoryList.Root';
/**
 * Displays loading state while category data is being fetched.
 * Only displays its children when the categories list is currently loading.
 *
 * @component
 * @example
 * ```tsx
 * import { CategoryList } from '@wix/headless-stores/react';
 *
 * function CategoriesLoading() {
 *   return (
 *     <CategoryList.Loading>
 *       <div className="loading-spinner">Loading categories...</div>
 *     </CategoryList.Loading>
 *   );
 * }
 * ```
 */
export const Loading = React.forwardRef((props, ref) => {
    const { asChild, children, className } = props;
    return (_jsx(CoreCategoryList.Loading, { children: _jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, customElement: children, children: _jsx("h1", { children: "Loading..." }) }) }));
});
/**
 * Repeats for each category in the list, providing individual category context.
 * Maps over all categories and provides each category through a service context.
 * Only renders when categories are successfully loaded.
 *
 * @component
 * @example
 * ```tsx
 * import { CategoryList } from '@wix/headless-stores/react';
 *
 * function CategoriesGrid() {
 *   return (
 *     <CategoryList.CategoryRepeater maxDepth={3}>
 *       <Category.Label />
 *       <Category.ID />
 *     </CategoryList.CategoryRepeater>
 *   );
 * }
 * ```
 */
export function CategoryRepeater(props) {
    // const { children, asChild = false, className } = props;
    const { children } = props;
    // Note: maxDepth is not implemented yet as it depends on category hierarchy structure
    return (_jsx(CoreCategoryList.ItemContent, { children: ({ category }) => {
            return (_jsx(Category.Root, { category: category, children: children }, category._id));
        } }));
}
