// @wix/headless-stores/cjs/dist/react/Choice.js
import { jsx as _jsx } from "react/jsx-runtime";
import React from 'react';
import { FreeText as FreeTextPrimitive } from './core/ProductModifiers.js';
import { AsChildSlot } from '@wix/headless-utils/react';
import { DataComponentTags } from '../data-component-tags.js';
var TestIds;
(function (TestIds) {
    TestIds["choiceRoot"] = "choice-root";
    TestIds["choiceText"] = "choice-text";
    TestIds["choiceColor"] = "choice-color";
    TestIds["choiceFreetext"] = "choice-freetext";
})(TestIds || (TestIds = {}));
// Create a context for individual choices
export const ChoiceContext = React.createContext(null);
/**
 * Hook to access ChoiceContext with proper error handling
 * @throws {Error} When used outside of a Choice context provider
 * @returns {ChoiceContextValue} The choice context value
 */
export function useChoiceContext() {
    const context = React.useContext(ChoiceContext);
    if (!context) {
        throw new Error('useChoiceContext must be used within a Choice context provider (Option.ChoiceRepeater)');
    }
    return context;
}
/**
 * Root component that consumes ChoiceContext provided by Option.ChoiceRepeater.
 *
 * @component
 * @example
 * ```tsx
 * <Choice.Root>
 *   <Choice.Text />
 *   <Choice.Color />
 * </Choice.Root>
 * ```
 */
export const Root = React.forwardRef((props, ref) => {
    const { children } = props;
    const choiceContext = React.useContext(ChoiceContext);
    if (!choiceContext) {
        return null; // Should be used within Option.ChoiceRepeater
    }
    const { choice } = choiceContext;
    const attributes = {
        'data-component-tag': DataComponentTags.choiceRoot,
        'data-testid': TestIds.choiceRoot,
        'data-type': choice?.type,
    };
    return (_jsx("div", { ...attributes, ref: ref, children: children }));
});
Root.displayName = 'Choice.Root';
/**
 * Text-based choice button.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Choice.Text className="px-4 py-2 border rounded-lg" />
 *
 * // asChild with primitive
 * <Choice.Text asChild>
 *   <button className="px-4 py-2 border rounded-lg" />
 * </Choice.Text>
 *
 * // asChild with react component
 * <Choice.Text asChild>
 *   {React.forwardRef(({id, value, ...props}, ref) => (
 *     <button ref={ref} {...props} className="px-4 py-2 border rounded-lg">
 *       {value}
 *     </button>
 *   ))}
 * </Choice.Text>
 * ```
 *
 * Data Attributes:
 * - `data-testid="choice-text"` - Applied to choice buttons
 * - `data-selected` - Is Choice selected
 * - `disabled` - Is Choice disabled (not in stock)
 */
export const Text = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    const choiceContext = React.useContext(ChoiceContext);
    if (!choiceContext) {
        return null; // Should be used within Option.ChoiceRepeater
    }
    const { choice, shouldRenderAsText, value, isSelected, select, isVisible, isInStock, isPreOrderEnabled, } = choiceContext;
    // Only render if this should be rendered as text
    if (!shouldRenderAsText)
        return null;
    // Don't render if not visible (handled by ProductVariantSelector in Root)
    if (!isVisible)
        return null;
    const choiceId = choice?.choiceId || '';
    return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.choiceText, "data-selected": isSelected ? 'true' : 'false', disabled: !isInStock && !isPreOrderEnabled, onClick: select, customElement: children, customElementProps: { id: choiceId, value }, content: value, ...otherProps, children: _jsx("button", { children: value }) }));
});
/**
 * Color swatch choice.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Choice.Color className="w-10 h-10 rounded-full border-4" />
 *
 * // asChild with primitive
 * <Choice.Color asChild>
 *   <button className="w-10 h-10 rounded-full border-4" />
 * </Choice.Color>
 *
 * // asChild with react component
 * <Choice.Color asChild>
 *   {React.forwardRef(({colorCode, name, id, ...props}, ref) => (
 *     <button
 *       ref={ref}
 *       {...props}
 *       className="w-10 h-10 rounded-full border-4"
 *       style={{ backgroundColor: colorCode }}
 *       title={name}
 *     />
 *   ))}
 * </Choice.Color>
 * ```
 *
 * Data Attributes:
 * - `data-testid="choice-color"` - Applied to color swatches
 * - `data-selected` - Is Choice selected
 * - `disabled` - Is Choice disabled (not in stock)
 */
export const Color = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    const choiceContext = React.useContext(ChoiceContext);
    if (!choiceContext) {
        return null; // Should be used within Option.ChoiceRepeater
    }
    const { choice, shouldRenderAsColor, value, isSelected, select, isVisible, isInStock, isPreOrderEnabled, } = choiceContext;
    // Only render if this should be rendered as color
    if (!shouldRenderAsColor)
        return null;
    // Don't render if not visible (handled by ProductVariantSelector in Root)
    if (!isVisible)
        return null;
    const { colorCode, choiceId } = choice;
    return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.choiceColor, "data-selected": isSelected ? 'true' : 'false', disabled: !isInStock && !isPreOrderEnabled, onClick: select, customElement: children, customElementProps: {
            colorCode: colorCode || '',
            name: value,
            id: choiceId || '',
        }, ...otherProps, children: _jsx("button", { style: { backgroundColor: colorCode }, title: value }) }));
});
/**
 * Provides a free text input for variant selection.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Choice.FreeText className="p-3 border rounded-lg" />
 *
 * // asChild with primitive
 * <Choice.FreeText asChild>
 *   <textarea className="p-3 border rounded-lg" />
 * </Choice.FreeText>
 *
 * // asChild with react component
 * <Choice.FreeText asChild>
 *   {React.forwardRef(({value, onChange, title, ...props}, ref) => (
 *     <textarea
 *       ref={ref}
 *       {...props}
 *       className="p-3 border rounded-lg"
 *       value={value}
 *       onChange={onChange}
 *       placeholder={`Enter custom ${title}...`}
 *     />
 *   ))}
 * </Choice.FreeText>
 * ```
 *
 * Data Attributes:
 * - `data-testid="choice-free-text"` - Applied to free text input
 * - `data-selected` - Is Choice selected
 */
export const FreeText = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    const choiceContext = React.useContext(ChoiceContext);
    if (!choiceContext) {
        return null; // Should be used within Option.ChoiceRepeater
    }
    const { choice, shouldRenderAsFreeText, onValueChange, isVisible, isSelected, } = choiceContext;
    // Only render if this should be rendered as free text
    if (!shouldRenderAsFreeText)
        return null;
    // Don't render if not visible (handled by ProductVariantSelector in Root)
    if (!isVisible)
        return null;
    return (_jsx(FreeTextPrimitive, { modifier: choice, children: ({ value, setText, placeholder, maxChars }) => {
            const handleChange = (e) => {
                setText(e.target.value);
                if (onValueChange) {
                    onValueChange(e.target.value);
                }
            };
            return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.choiceFreetext, "data-selected": isSelected ? 'true' : 'false', customElement: children, customElementProps: {
                    minCharCount: choice?.minCharCount,
                    maxCharCount: choice?.maxCharCount,
                    defaultAddedPrice: choice?.addedPrice || undefined,
                    title: choice?.name || undefined,
                    onChange: handleChange,
                }, ...otherProps, children: _jsx("textarea", { placeholder: placeholder, rows: 3, value: value, maxLength: maxChars, onChange: handleChange }) }));
        } }));
});
