// @wix/auto_sdk_events_categories/build/internal/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  CategoryFieldset: () => CategoryFieldset,
  SortOrder: () => SortOrder,
  State: () => State,
  WebhookIdentityType: () => WebhookIdentityType,
  assignEvents: () => assignEvents2,
  bulkAssignEvents: () => bulkAssignEvents2,
  bulkAssignEventsAsync: () => bulkAssignEventsAsync2,
  bulkCreateCategory: () => bulkCreateCategory2,
  bulkUnassignEvents: () => bulkUnassignEvents2,
  bulkUnassignEventsAsync: () => bulkUnassignEventsAsync2,
  createCategory: () => createCategory2,
  deleteCategory: () => deleteCategory2,
  listEventCategories: () => listEventCategories2,
  queryCategories: () => queryCategories2,
  reorderCategoryEvents: () => reorderCategoryEvents2,
  unassignEvents: () => unassignEvents2,
  updateCategory: () => updateCategory2
});
module.exports = __toCommonJS(index_typings_exports);

// src/events-v1-category-categories.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/events-v1-category-categories.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEventsCategoriesCategoryManagementUrl(opts) {
  const domainToMappings = {
    "events._base_domain_": [
      {
        srcPath: "",
        destPath: ""
      },
      {
        srcPath: "/_api/wix-events-web",
        destPath: "/api"
      }
    ],
    "www.wixevents.com": [
      {
        srcPath: "/_api/wix-events-web",
        destPath: "/api"
      },
      {
        srcPath: "/api",
        destPath: "/api"
      },
      {
        srcPath: "/viewEvent",
        destPath: "/viewEvent"
      },
      {
        srcPath: "/viewRsvpDashboard",
        destPath: "/viewRsvpDashboard"
      },
      {
        srcPath: "/viewOrderDashboard",
        destPath: "/viewOrderDashboard"
      },
      {
        srcPath: "/viewSummarySubscription",
        destPath: "/viewSummarySubscription"
      },
      {
        srcPath: "/check-in",
        destPath: "/check-in"
      },
      {
        srcPath: "/v2",
        destPath: "/v2"
      },
      {
        srcPath: "/export",
        destPath: "/export"
      },
      {
        srcPath: "/tickets",
        destPath: "/tickets"
      },
      {
        srcPath: "/payment-redirect",
        destPath: "/payment-redirect"
      },
      {
        srcPath: "/admin",
        destPath: "/admin"
      },
      {
        srcPath: "/ticket",
        destPath: "/ticket"
      },
      {
        srcPath: "/media",
        destPath: "/media"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/wix-events-web",
        destPath: ""
      }
    ],
    "events.wixapps.net": [
      {
        srcPath: "/_api/wix-events-web",
        destPath: "/api"
      },
      {
        srcPath: "/events",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/wix-events-web",
        destPath: "/api"
      },
      {
        srcPath: "/_api/wix-events-web/v2",
        destPath: "/v2"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/wix-events-web",
        destPath: "/api"
      },
      {
        srcPath: "/events",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/wix-events-web",
        destPath: "/api"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v2/events",
        destPath: "/v2/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/events",
        destPath: "/v1/bulk/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      },
      {
        srcPath: "/events/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/events/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/events/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/events/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/events/v1/bulk/events",
        destPath: "/v1/bulk/events"
      },
      {
        srcPath: "/events/v1/reports",
        destPath: "/v1/reports"
      },
      {
        srcPath: "/events/v1/tickets",
        destPath: "/v1/tickets"
      },
      {
        srcPath: "/events/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/events/v1/web",
        destPath: "/v1/web"
      },
      {
        srcPath: "/events/v2/tickets",
        destPath: "/v2/tickets"
      },
      {
        srcPath: "/events/v2/events",
        destPath: "/v2/events"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/events/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/events/v1/tickets",
        destPath: "/v1/tickets"
      },
      {
        srcPath: "/events/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/events/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/events/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/events/v2/events",
        destPath: "/v2/events"
      },
      {
        srcPath: "/events/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      }
    ],
    "api._base_domain_": [
      {
        srcPath: "/events/v2",
        destPath: "/v2"
      },
      {
        srcPath: "/events/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/events/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/events/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/events/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/events/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/events/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    _: [
      {
        srcPath: "/_api/wix-events-web/v2",
        destPath: "/v2"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/events/v2/events",
        destPath: "/v2/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/wix-events-web/v2",
        destPath: "/v2"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "editor-flow.wixapps.net": [
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_events_categories";
function createCategory(payload) {
  function __createCategory({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "category.createdDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v1.category",
      method: "POST",
      methodFqn: "wix.events.categories.CategoryManagement.CreateCategory",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsCategoriesCategoryManagementUrl({
        protoPath: "/v1/categories",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "category.createdDate" }]
        }
      ])
    };
    return metadata;
  }
  return __createCategory;
}
function bulkCreateCategory(payload) {
  function __bulkCreateCategory({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "categories.createdDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v1.category",
      method: "POST",
      methodFqn: "wix.events.categories.CategoryManagement.BulkCreateCategory",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsCategoriesCategoryManagementUrl({
        protoPath: "/v1/bulk/categories/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "results.item.createdDate" }]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateCategory;
}
function updateCategory(payload) {
  function __updateCategory({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "category.createdDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v1.category",
      method: "PATCH",
      methodFqn: "wix.events.categories.CategoryManagement.UpdateCategory",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsCategoriesCategoryManagementUrl({
        protoPath: "/v1/categories/{category.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "category.createdDate" }]
        }
      ])
    };
    return metadata;
  }
  return __updateCategory;
}
function deleteCategory(payload) {
  function __deleteCategory({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.category",
      method: "DELETE",
      methodFqn: "wix.events.categories.CategoryManagement.DeleteCategory",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsCategoriesCategoryManagementUrl({
        protoPath: "/v1/categories/{categoryId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteCategory;
}
function queryCategories(payload) {
  function __queryCategories({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.category",
      method: "POST",
      methodFqn: "wix.events.categories.CategoryManagement.QueryCategories",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsCategoriesCategoryManagementUrl({
        protoPath: "/v1/categories/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "categories.createdDate" }]
        }
      ])
    };
    return metadata;
  }
  return __queryCategories;
}
function assignEvents(payload) {
  function __assignEvents({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.category",
      method: "POST",
      methodFqn: "wix.events.categories.CategoryManagement.AssignEvents",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsCategoriesCategoryManagementUrl({
        protoPath: "/v1/categories/{categoryId}/events",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __assignEvents;
}
function bulkAssignEvents(payload) {
  function __bulkAssignEvents({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.category",
      method: "POST",
      methodFqn: "wix.events.categories.CategoryManagement.BulkAssignEvents",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsCategoriesCategoryManagementUrl({
        protoPath: "/v1/bulk/categories/events",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "results.item.createdDate" }]
        }
      ])
    };
    return metadata;
  }
  return __bulkAssignEvents;
}
function bulkAssignEventsAsync(payload) {
  function __bulkAssignEventsAsync({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.category",
      method: "POST",
      methodFqn: "wix.events.categories.CategoryManagement.BulkAssignEventsAsync",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsCategoriesCategoryManagementUrl({
        protoPath: "/v1/bulk-async/categories/assign-events-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkAssignEventsAsync;
}
function unassignEvents(payload) {
  function __unassignEvents({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.category",
      method: "DELETE",
      methodFqn: "wix.events.categories.CategoryManagement.UnassignEvents",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsCategoriesCategoryManagementUrl({
        protoPath: "/v1/categories/{categoryId}/events",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __unassignEvents;
}
function bulkUnassignEvents(payload) {
  function __bulkUnassignEvents({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.category",
      method: "DELETE",
      methodFqn: "wix.events.categories.CategoryManagement.BulkUnassignEvents",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsCategoriesCategoryManagementUrl({
        protoPath: "/v1/bulk/categories/events",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "results.item.createdDate" }]
        }
      ])
    };
    return metadata;
  }
  return __bulkUnassignEvents;
}
function bulkUnassignEventsAsync(payload) {
  function __bulkUnassignEventsAsync({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.category",
      method: "POST",
      methodFqn: "wix.events.categories.CategoryManagement.BulkUnassignEventsAsync",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsCategoriesCategoryManagementUrl({
        protoPath: "/v1/bulk-async/categories/unassign-events-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUnassignEventsAsync;
}
function listEventCategories(payload) {
  function __listEventCategories({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.category",
      method: "GET",
      methodFqn: "wix.events.categories.CategoryManagement.ListEventCategories",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsCategoriesCategoryManagementUrl({
        protoPath: "/v1/categories/{eventId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "categories.createdDate" }]
        }
      ])
    };
    return metadata;
  }
  return __listEventCategories;
}
function reorderCategoryEvents(payload) {
  function __reorderCategoryEvents({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.category",
      method: "POST",
      methodFqn: "wix.events.categories.CategoryManagement.ReorderCategoryEvents",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsCategoriesCategoryManagementUrl({
        protoPath: "/v1/categories/{categoryId}/reorder",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __reorderCategoryEvents;
}

// src/events-v1-category-categories.universal.ts
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var State = /* @__PURE__ */ ((State2) => {
  State2["MANUAL"] = "MANUAL";
  State2["AUTO"] = "AUTO";
  State2["RECURRING_EVENT"] = "RECURRING_EVENT";
  State2["HIDDEN"] = "HIDDEN";
  return State2;
})(State || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var CategoryFieldset = /* @__PURE__ */ ((CategoryFieldset2) => {
  CategoryFieldset2["COUNTS"] = "COUNTS";
  return CategoryFieldset2;
})(CategoryFieldset || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createCategory2(category) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ category });
  const reqOpts = createCategory(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.category;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { category: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["category"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCreateCategory2(categories) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    categories
  });
  const reqOpts = bulkCreateCategory(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { categories: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["categories"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateCategory2(_id, category) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    category: { ...category, id: _id }
  });
  const reqOpts = updateCategory(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.category;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { category: "$[1]" },
        explicitPathsToArguments: { "category.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "category"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteCategory2(categoryId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    categoryId
  });
  const reqOpts = deleteCategory(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { categoryId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["categoryId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryCategories2(options) {
  const { httpClient, sideEffects } = arguments[1];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryCategories({
        ...payload,
        ...options ?? {}
      });
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, options];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [])
      );
      return {
        items: transformedData?.categories,
        pagingMetadata: transformedData?.metaData
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "OFFSET",
    transformationPaths: {}
  });
}
async function assignEvents2(categoryId, eventId) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    categoryId,
    eventId
  });
  const reqOpts = assignEvents(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { categoryId: "$[0]", eventId: "$[1]" },
        singleArgumentUnchanged: false
      },
      ["categoryId", "eventId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkAssignEvents2(categoryId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    categoryId,
    eventId: options?.eventId
  });
  const reqOpts = bulkAssignEvents(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          categoryId: "$[0]",
          eventId: "$[1].eventId"
        },
        singleArgumentUnchanged: false
      },
      ["categoryId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkAssignEventsAsync2(categoryId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    categoryId,
    filter: options?.filter
  });
  const reqOpts = bulkAssignEventsAsync(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { categoryId: "$[0]", filter: "$[1].filter" },
        singleArgumentUnchanged: false
      },
      ["categoryId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function unassignEvents2(categoryId, eventId) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    categoryId,
    eventId
  });
  const reqOpts = unassignEvents(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { categoryId: "$[0]", eventId: "$[1]" },
        singleArgumentUnchanged: false
      },
      ["categoryId", "eventId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUnassignEvents2(categoryId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    categoryId,
    eventId: options?.eventId
  });
  const reqOpts = bulkUnassignEvents(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          categoryId: "$[0]",
          eventId: "$[1].eventId"
        },
        singleArgumentUnchanged: false
      },
      ["categoryId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUnassignEventsAsync2(categoryId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    categoryId,
    filter: options?.filter
  });
  const reqOpts = bulkUnassignEventsAsync(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { categoryId: "$[0]", filter: "$[1].filter" },
        singleArgumentUnchanged: false
      },
      ["categoryId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listEventCategories2(eventId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ eventId });
  const reqOpts = listEventCategories(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { eventId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["eventId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function reorderCategoryEvents2(categoryId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    categoryId,
    eventId: options?.eventId,
    beforeEventId: options?.beforeEventId,
    afterEventId: options?.afterEventId
  });
  const reqOpts = reorderCategoryEvents(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          categoryId: "$[0]",
          eventId: "$[1].eventId",
          beforeEventId: "$[1].beforeEventId",
          afterEventId: "$[1].afterEventId"
        },
        singleArgumentUnchanged: false
      },
      ["categoryId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  CategoryFieldset,
  SortOrder,
  State,
  WebhookIdentityType,
  assignEvents,
  bulkAssignEvents,
  bulkAssignEventsAsync,
  bulkCreateCategory,
  bulkUnassignEvents,
  bulkUnassignEventsAsync,
  createCategory,
  deleteCategory,
  listEventCategories,
  queryCategories,
  reorderCategoryEvents,
  unassignEvents,
  updateCategory
});
