// @wix/patterns/dist/cjs/components/CustomFieldsPanel/ArchiveFieldModal.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports._ArchiveFieldModal = exports.ArchiveFieldModal = void 0;
var _designSystem = require("@wix/design-system");
var _react = _interopRequireDefault(require("react"));
var _react2 = require("@wix/bex-core/react");
var _ActionModal = require("../ActionModal");
var _mobxReactLite = require("mobx-react-lite");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CustomFieldsPanel/ArchiveFieldModal.tsx";
const _ArchiveFieldModal = ({
  state
}) => {
  const container = (0, _react2.useWixPatternsContainer)();
  const {
    translate: t
  } = container;
  return /*#__PURE__*/_react.default.createElement(_ActionModal.CustomActionModal, {
    dataHook: "archive-custom-field-modal",
    state: state.modal,
    title: t('cairo.customFields.archive.modal.title'),
    primaryButtonText: t('cairo.customFields.archive.modal.CTA'),
    secondaryButtonText: t('cairo.customFields.archive.modal.cancel.CTA'),
    content: /*#__PURE__*/_react.default.createElement(_designSystem.Text, {
      dataHook: "archive-custom-field-body",
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 24,
        columnNumber: 9
      }
    }, t('cairo.customFields.archive.modal.description')),
    theme: "destructive",
    width: "510px",
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 17,
      columnNumber: 5
    }
  });
};
exports._ArchiveFieldModal = _ArchiveFieldModal;
const ArchiveFieldModal = exports.ArchiveFieldModal = (0, _mobxReactLite.observer)(_ArchiveFieldModal);
