// @wix/patterns/dist/cjs/components/CustomFieldsPanel/CustomFieldsPanel.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.CustomFieldsPanel = void 0;
exports._CustomFieldsPanel = _CustomFieldsPanel;
var _bexCore = require("@wix/bex-core");
var _react = require("@wix/bex-core/react");
var _useRafRender = require("../../hooks/useRafRender");
var _react2 = _interopRequireDefault(require("react"));
var _designSystem = require("@wix/design-system");
var _ScrollableContent = require("../ScrollableContent");
var _mobxReactLite = require("mobx-react-lite");
var _CustomFieldsList = require("./CustomFieldsList");
var _AddCustomFieldCTAButton = require("./AddCustomFieldCTAButton");
var _bi = require("@wix/bex-core/bi");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CustomFieldsPanel/CustomFieldsPanel.tsx";
function _CustomFieldsPanel({
  state
}) {
  const container = (0, _react.useWixPatternsContainer)();
  const {
    translate: t
  } = container;
  const origin = 'Custom fields panel';
  const entered = (0, _useRafRender.useRafRender)();
  if (!entered) {
    return null;
  }
  return /*#__PURE__*/_react2.default.createElement(_designSystem.SidePanel, {
    dataHook: "custom-fields-panel",
    width: "100%",
    onCloseButtonClick: () => {
      state.reportBi((0, _bexCore.withoutDefaults)(_bi.cairoFiltersPanelUsed)({
        isOpened: false,
        feature: 'Manage custom fields',
        origin: 'X Button'
      }));
      state.closeSidePanel == null || state.closeSidePanel();
    },
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 28,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react2.default.createElement(_designSystem.SidePanel.Header, {
    title: t('cairo.customFields.manageFields.title'),
    subtitle: t('cairo.customFields.manageFields.subtitle'),
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 42,
      columnNumber: 7
    }
  }), /*#__PURE__*/_react2.default.createElement(_ScrollableContent.ScrollableContent, {
    overflowY: true,
    dataHook: "custom-fields-panel-scroll-container",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 46,
      columnNumber: 7
    }
  }, /*#__PURE__*/_react2.default.createElement(_CustomFieldsList.CustomFieldsList, {
    state: state,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 50,
      columnNumber: 9
    }
  })), /*#__PURE__*/_react2.default.createElement(_designSystem.Box, {
    direction: "vertical",
    width: "100%",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 53,
      columnNumber: 7
    }
  }, /*#__PURE__*/_react2.default.createElement(_designSystem.Divider, {
    direction: "horizontal",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 54,
      columnNumber: 9
    }
  }), /*#__PURE__*/_react2.default.createElement(_designSystem.Box, {
    padding: "SP4",
    direction: "vertical",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 55,
      columnNumber: 9
    }
  }, /*#__PURE__*/_react2.default.createElement(_AddCustomFieldCTAButton.AddCustomFieldCTAButton, {
    state: state,
    onClick: () => {
      state.openCustomFieldModal({
        origin
      });
    },
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 56,
      columnNumber: 11
    }
  }))));
}
const CustomFieldsPanel = exports.CustomFieldsPanel = (0, _mobxReactLite.observer)(_CustomFieldsPanel);
