// @wix/patterns/dist/cjs/components/CustomFieldsPanel/CustomFieldItem.js
"use strict";

exports.__esModule = true;
exports.CustomFieldItem = void 0;
var _designSystem = require("@wix/design-system");
var _react = _interopRequireWildcard(require("react"));
var _mobxReactLite = require("mobx-react-lite");
var _ManageFieldsPopoverMenu = require("./ManageFieldsPopoverMenu");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CustomFieldsPanel/CustomFieldItem.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _CustomFieldItem(props) {
  const {
    field,
    showDivider,
    state
  } = props;
  const style = (0, _react.useMemo)(() => ({
    display: 'flex',
    flexDirection: 'column',
    position: 'relative'
  }), []);
  return /*#__PURE__*/_react.default.createElement("div", {
    style: style,
    "data-hook": `custom-field-${field.id}`,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 27,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.TableListItem, {
    showDivider: showDivider,
    verticalPadding: "tiny",
    options: [{
      value: /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
        __self: this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 34,
          columnNumber: 15
        }
      }, /*#__PURE__*/_react.default.createElement("div", {
        style: {
          padding: '0 6px',
          wordBreak: 'break-word'
        },
        __self: this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 35,
          columnNumber: 17
        }
      }, /*#__PURE__*/_react.default.createElement(_designSystem.Text, {
        size: "small",
        __self: this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 41,
          columnNumber: 19
        }
      }, field.label || field.id)))
    }, {
      value: /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
        height: "24px",
        __self: this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 48,
          columnNumber: 15
        }
      }, /*#__PURE__*/_react.default.createElement(_ManageFieldsPopoverMenu.ManageFieldsPopoverMenu, {
        origin: "Custom fields panel",
        state: state,
        id: field.id,
        location: "Manage custom fields action cell",
        __self: this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 49,
          columnNumber: 17
        }
      })),
      align: 'right'
    }],
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 28,
      columnNumber: 7
    }
  }));
}
const CustomFieldItem = exports.CustomFieldItem = (0, _mobxReactLite.observer)(_CustomFieldItem);
