// @wix/patterns/dist/cjs/components/CustomFieldsPanel/CustomFieldsList.js
"use strict";

exports.__esModule = true;
exports.CustomFieldsList = void 0;
var _mobxReactLite = require("mobx-react-lite");
var _react = _interopRequireWildcard(require("react"));
var _CustomFieldItem = require("./CustomFieldItem");
var _designSystem = require("@wix/design-system");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CustomFieldsPanel/CustomFieldsList.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _CustomFieldsList(props) {
  const {
    state
  } = props;
  const renderItem = (0, _react.useCallback)((field, index, array) => {
    return /*#__PURE__*/_react.default.createElement(_CustomFieldItem.CustomFieldItem, {
      state: state,
      key: field.id,
      field: field,
      showDivider: index < array.length - 1,
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 18,
        columnNumber: 9
      }
    });
  }, []);
  return /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
    direction: "vertical",
    dataHook: "custom-fields-list",
    backgroundColor: "D60",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 30,
      columnNumber: 5
    }
  }, state.dataExtension.availableUserCustomFields.map(renderItem));
}
const CustomFieldsList = exports.CustomFieldsList = (0, _mobxReactLite.observer)(_CustomFieldsList);
