// @wix/patterns/dist/cjs/components/CustomFieldsPanel/CustomFieldPanel.uni.driver.js
"use strict";

exports.__esModule = true;
exports.CustomFieldPanelUniDriver = CustomFieldPanelUniDriver;
var _unidriver = require("@wix/design-system/dist/testkit/unidriver");
var _SlidingModalUni = require("../SlidingModal/SlidingModal.uni.driver");
var _CustomFieldPopoverUni = require("./CustomFieldPopover.uni.driver");
var _CustomFieldsDisabledHintUni = require("../CustomFields/CustomFieldsDisabledHint.uni.driver");
function CustomFieldPanelUniDriver(base, body) {
  const panelBase = () => base.$('[data-hook="custom-fields-panel"]');
  const panel = () => (0, _unidriver.SidePanelUniDriver)(panelBase(), body);
  const slidingModal = () => (0, _SlidingModalUni.SlidingModalUniDriver)(base.$('[data-hook="cairo-side-panel-modal"]'), body);
  const addItem = () => (0, _CustomFieldsDisabledHintUni.CustomFieldsDisabledHintUniDriver)(base, body, {
    dataHook: 'data-extension-add-new'
  });
  const customFieldPanel = i => (0, _CustomFieldPopoverUni.CustomFieldPopoverUniDriver)(getItemAtBase(i), body);
  const list = () => panelBase().$('[data-hook="custom-fields-list"]');
  const items = () => panelBase().$$('[data-hook="custom-fields-list"] > div');
  const getItemAtBase = i => items().get(i);
  const getItemByIdBase = id => list().$(`[data-hook="custom-field-${id}"]`);
  return {
    exists: () => panelBase().exists(),
    isOpen: () => Promise.all([slidingModal().isOpen(), panel().exists()]).then(arr => arr.every(Boolean)),
    getTitle: () => panel().getTitleText(),
    getSubtitle: () => panel().getSubtitleText(),
    clickCloseButton: () => panel().clickClose(),
    closeButtonExists: () => panel().isCloseButtonExists(),
    getItemLabelById: id => getItemByIdBase(id).text(),
    getItemLabelByIndex: i => getItemAtBase(i).text(),
    getItemsCount: () => items().count(),
    getAddNewFieldButton: addItem,
    getCustomFieldOptionsAt: i => customFieldPanel(i).getCustomFieldOptionsAt()
  };
}
