// @wix/patterns/dist/cjs/components/CustomFieldsPanel/CustomFieldPopover.uni.driver.js
"use strict";

exports.__esModule = true;
exports.CustomFieldPopoverUniDriver = CustomFieldPopoverUniDriver;
var _unidriver = require("@wix/design-system/dist/testkit/unidriver");
var _DisabledTooltipButtonUni = require("../DisabledTooltip/DisabledTooltipButton.uni.driver");
function CustomFieldPopoverUniDriver(base, body) {
  const customFieldPopoverDriverBase = () => base.$('[data-hook="custom-field-options"]');
  const customFieldPopoverDriver = () => (0, _unidriver.PopoverMenuUniDriver)(customFieldPopoverDriverBase(), body);
  const customFieldOptionsTrigger = () => (0, _DisabledTooltipButtonUni.DisabledTooltipButtonUniDriver)(base, body, {
    dataHook: 'custom-field-options-trigger'
  });
  const getCustomFieldOptionsAt = () => {
    return {
      open: () => customFieldPopoverDriver().getTriggerElement('custom-field-options-trigger').click(),
      clickEdit: () => customFieldPopoverDriver().clickAtChildByDataHook('custom-field-options-edit'),
      clickArchive: () => customFieldPopoverDriver().clickAtChildByDataHook('custom-field-options-archive')
    };
  };
  return {
    exists: () => customFieldPopoverDriverBase().exists(),
    isDisabled: () => customFieldOptionsTrigger().isButtonDisabled(),
    getTooltipText: () => customFieldOptionsTrigger().getTooltipText(),
    getCustomFieldOptionsAt
  };
}
