// @wix/bex-core/dist/cjs/state/ArrayFilterState/ArrayFilterState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.stringsArrayFilter = exports.idNameArrayFilterNum = exports.idNameArrayFilter = exports.idArrayFilter = exports.arrayFilter = exports.ArrayFilterState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _events = require("events");
var _mobx = require("mobx");
var _refreshFilter = require("../Filter/refreshFilter");
class ArrayFilterState {
  constructor(params) {
    (0, _defineProperty2.default)(this, "_value", void 0);
    (0, _defineProperty2.default)(this, "name", void 0);
    (0, _defineProperty2.default)(this, "persistent", void 0);
    (0, _defineProperty2.default)(this, "defaultValue", void 0);
    (0, _defineProperty2.default)(this, "parse", void 0);
    (0, _defineProperty2.default)(this, "simplify", void 0);
    (0, _defineProperty2.default)(this, "itemKey", void 0);
    (0, _defineProperty2.default)(this, "itemName", void 0);
    (0, _defineProperty2.default)(this, "equals", void 0);
    (0, _defineProperty2.default)(this, "isCustomField", void 0);
    (0, _defineProperty2.default)(this, "events", new _events.EventEmitter());
    (0, _defineProperty2.default)(this, "matches", void 0);
    (0, _defineProperty2.default)(this, "_toArray", value => {
      return value;
    });
    this.name = params.name ?? '';
    this.persistent = params.persistent;
    this.defaultValue = params.defaultValue ?? [];
    this.parse = params.parse;
    this.simplify = params.simplify ?? (value => value);
    this.itemKey = params.itemKey;
    this.itemName = params.itemName;
    this._value = params.initialValue ?? this.defaultValue;
    this.matches = params.matches ?? ((item, value) => {
      const {
        itemKey
      } = this;
      const key = typeof item === 'string' ? item : itemKey(item);
      return value == null || !value.length || value.some(v => itemKey(v) === key);
    });
    this.equals = params.equals;
    this.isCustomField = params.isCustomField;
    (0, _mobx.makeObservable)(this, {
      _value: _mobx.observable.ref,
      defaultValue: _mobx.observable.ref,
      persistent: _mobx.observable.ref,
      isEmpty: _mobx.computed,
      size: _mobx.computed,
      value: _mobx.computed,
      reset: _mobx.action,
      setValue: _mobx.action,
      changeValue: _mobx.action,
      scheduleRefresh: _mobx.action,
      clone: _mobx.action
    });
  }
  get value() {
    return this._value;
  }
  get toArray() {
    return this._value;
  }
  get isEmpty() {
    return !this.size;
  }
  get size() {
    return this._value.length;
  }
  get toQueryString() {
    return JSON.stringify(this._value.map(this.simplify));
  }
  applyFromQueryString(str) {
    try {
      const maybeArr = JSON.parse(str);
      if (Array.isArray(maybeArr)) {
        this.setValue(maybeArr.map(this.parse).filter(value => value != null));
      }
    } catch (_) {}
  }
  encode(value) {
    return value.map(this.simplify);
  }
  decode(raw) {
    if (Array.isArray(raw)) {
      return raw.map(this.parse).filter(value => value != null);
    }
    return [];
  }
  setValue(value, {
    emitEvents,
    ...options
  } = {}) {
    this._value = value;
    if (emitEvents) {
      for (const event of emitEvents) {
        this.events.emit(event, options);
      }
    }
  }
  changeValue(value, {
    emitEvents = [],
    ...options
  } = {}) {
    this.setValue(value, {
      ...options,
      emitEvents: ['change', ...emitEvents]
    });
  }
  refresh(value, options) {
    (0, _refreshFilter.refreshFilter)(this, value, options);
  }
  scheduleRefresh(value) {
    this.changeValue(value, {
      emitEvents: ['scheduleRefresh']
    });
  }
  reset(options = {}) {
    this.setValue(this.defaultValue, options);
  }
  hasDiff(items) {
    const {
      value
    } = this;
    return value.length !== items.length || value.some(item => items.every(i => this.itemKey(i) !== this.itemKey(item)));
  }
  remove(items, options = {}) {
    const {
      itemKey
    } = this;
    const keysToRemove = new Set(items.map(itemKey));
    this.setValue(this._value.filter(item => !keysToRemove.has(itemKey(item))), options);
  }
  clone(params) {
    const {
      itemName,
      itemKey,
      simplify,
      parse,
      defaultValue,
      value: initialValue,
      matches,
      persistent,
      name
    } = this;
    return new ArrayFilterState({
      itemName,
      itemKey,
      simplify,
      parse,
      defaultValue,
      initialValue,
      matches,
      persistent,
      name,
      ...params
    });
  }
}
exports.ArrayFilterState = ArrayFilterState;
const stringsArrayFilter = params => new ArrayFilterState({
  itemKey: item => item,
  itemName: item => item,
  parse: value => {
    if (value == null || typeof value !== 'string') {
      return null;
    }
    return value;
  },
  simplify: value => value,
  ...params
});
exports.stringsArrayFilter = stringsArrayFilter;
const arrayFilter = params => new ArrayFilterState(params);
exports.arrayFilter = arrayFilter;
const idNameArrayFilter = params => new ArrayFilterState({
  itemKey: item => item.id,
  itemName: item => item.name,
  parse: value => {
    if (value == null || typeof value === 'string' || typeof value.id !== 'string' || typeof value.name !== 'string') {
      return null;
    }
    return value;
  },
  simplify: value => ({
    id: value.id,
    name: value.name
  }),
  ...params
});
exports.idNameArrayFilter = idNameArrayFilter;
const idArrayFilter = params => new ArrayFilterState({
  itemKey: item => item.id,
  itemName: () => '',
  parse: value => {
    if (value == null || typeof value === 'string' || typeof value.id !== 'string') {
      return null;
    }
    return value;
  },
  simplify: value => ({
    id: value.id
  }),
  ...params
});
exports.idArrayFilter = idArrayFilter;
const idNameArrayFilterNum = params => new ArrayFilterState({
  itemKey: item => String(item.id),
  itemName: item => item.name,
  parse: value => {
    if (value == null || typeof value === 'string' || typeof value.id !== 'number' || typeof value.name !== 'string') {
      return null;
    }
    return value;
  },
  simplify: value => ({
    id: value.id,
    name: value.name
  }),
  ...params
});
exports.idNameArrayFilterNum = idNameArrayFilterNum;
