// @wix/auto_sdk_media_enterprise-media-items/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  bulkImportFiles: () => bulkImportFiles2,
  bulkUpdateItem: () => bulkUpdateItem2,
  generateFileUploadUrl: () => generateFileUploadUrl2,
  getItem: () => getItem2,
  importFile: () => importFile2,
  itemUploadCallback: () => itemUploadCallback2,
  linkItemToCategories: () => linkItemToCategories2,
  overwriteItemCategories: () => overwriteItemCategories2,
  queryItems: () => queryItems2,
  searchItems: () => searchItems2,
  unlinkItemFromCategories: () => unlinkItemFromCategories2,
  updateItem: () => updateItem2
});
module.exports = __toCommonJS(meta_exports);

// src/media-enterprise-public-media-v1-enterprise-item-enterprise-media-items.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixMediaEnterprisePublicMediaV1EnterpriseItemsServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/enterprise-public-media-server",
        destPath: ""
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/_api/enterprise-public-media-server",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/enterprise-public-media-server",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/enterprise-public-media-server",
        destPath: ""
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/enterprise-public-media-server",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_media_enterprise-media-items";
function itemUploadCallback(payload) {
  function __itemUploadCallback({ host }) {
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_item",
      method: "POST",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseItemsService.ItemUploadCallback",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseItemsServiceUrl({
        protoPath: "/v1/enterprise/items/upload/callback",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __itemUploadCallback;
}
function generateFileUploadUrl(payload) {
  function __generateFileUploadUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_item",
      method: "POST",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseItemsService.GenerateFileUploadUrl",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseItemsServiceUrl({
        protoPath: "/v1/enterprise/items/upload/generate-url",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __generateFileUploadUrl;
}
function importFile(payload) {
  function __importFile({ host }) {
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_item",
      method: "POST",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseItemsService.ImportFile",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseItemsServiceUrl({
        protoPath: "/v1/enterprise/items/upload/import",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "item.createdDate" },
            { path: "item.updatedDate" },
            { path: "item.assets.image.urlExpirationDate" },
            { path: "item.assets.video.urlExpirationDate" },
            { path: "item.assets.video.resolutions.urlExpirationDate" },
            {
              path: "item.assets.video.resolutions.poster.urlExpirationDate"
            },
            { path: "item.assets.video.posters.urlExpirationDate" },
            { path: "item.assets.vector.urlExpirationDate" },
            { path: "item.assets.audio.urlExpirationDate" },
            { path: "item.assets.document.urlExpirationDate" },
            { path: "item.assets.archive.urlExpirationDate" },
            { path: "item.assets.model3d.urlExpirationDate" },
            { path: "item.assets.model3d.thumbnail.urlExpirationDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __importFile;
}
function bulkImportFiles(payload) {
  function __bulkImportFiles({ host }) {
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_item",
      method: "POST",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseItemsService.BulkImportFiles",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseItemsServiceUrl({
        protoPath: "/v1/bulk/files/import",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.assets.image.urlExpirationDate" },
            { path: "results.item.assets.video.urlExpirationDate" },
            {
              path: "results.item.assets.video.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.assets.video.resolutions.poster.urlExpirationDate"
            },
            { path: "results.item.assets.video.posters.urlExpirationDate" },
            { path: "results.item.assets.vector.urlExpirationDate" },
            { path: "results.item.assets.audio.urlExpirationDate" },
            { path: "results.item.assets.document.urlExpirationDate" },
            { path: "results.item.assets.archive.urlExpirationDate" },
            { path: "results.item.assets.model3d.urlExpirationDate" },
            {
              path: "results.item.assets.model3d.thumbnail.urlExpirationDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkImportFiles;
}
function searchItems(payload) {
  function __searchItems({ host }) {
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_item",
      method: "POST",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseItemsService.SearchItems",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseItemsServiceUrl({
        protoPath: "/v1/enterprise/items/search",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "items.createdDate" },
            { path: "items.updatedDate" },
            { path: "items.assets.image.urlExpirationDate" },
            { path: "items.assets.video.urlExpirationDate" },
            { path: "items.assets.video.resolutions.urlExpirationDate" },
            {
              path: "items.assets.video.resolutions.poster.urlExpirationDate"
            },
            { path: "items.assets.video.posters.urlExpirationDate" },
            { path: "items.assets.vector.urlExpirationDate" },
            { path: "items.assets.audio.urlExpirationDate" },
            { path: "items.assets.document.urlExpirationDate" },
            { path: "items.assets.archive.urlExpirationDate" },
            { path: "items.assets.model3d.urlExpirationDate" },
            { path: "items.assets.model3d.thumbnail.urlExpirationDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __searchItems;
}
function queryItems(payload) {
  function __queryItems({ host }) {
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_item",
      method: "POST",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseItemsService.QueryItems",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseItemsServiceUrl({
        protoPath: "/v1/enterprise/items/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "items.createdDate" },
            { path: "items.updatedDate" },
            { path: "items.assets.image.urlExpirationDate" },
            { path: "items.assets.video.urlExpirationDate" },
            { path: "items.assets.video.resolutions.urlExpirationDate" },
            {
              path: "items.assets.video.resolutions.poster.urlExpirationDate"
            },
            { path: "items.assets.video.posters.urlExpirationDate" },
            { path: "items.assets.vector.urlExpirationDate" },
            { path: "items.assets.audio.urlExpirationDate" },
            { path: "items.assets.document.urlExpirationDate" },
            { path: "items.assets.archive.urlExpirationDate" },
            { path: "items.assets.model3d.urlExpirationDate" },
            { path: "items.assets.model3d.thumbnail.urlExpirationDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryItems;
}
function updateItem(payload) {
  function __updateItem({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "item.createdDate" },
          { path: "item.updatedDate" },
          { path: "item.assets.image.urlExpirationDate" },
          { path: "item.assets.video.urlExpirationDate" },
          { path: "item.assets.video.resolutions.urlExpirationDate" },
          { path: "item.assets.video.resolutions.poster.urlExpirationDate" },
          { path: "item.assets.video.posters.urlExpirationDate" },
          { path: "item.assets.vector.urlExpirationDate" },
          { path: "item.assets.audio.urlExpirationDate" },
          { path: "item.assets.document.urlExpirationDate" },
          { path: "item.assets.archive.urlExpirationDate" },
          { path: "item.assets.model3d.urlExpirationDate" },
          { path: "item.assets.model3d.thumbnail.urlExpirationDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_item",
      method: "PATCH",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseItemsService.UpdateItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseItemsServiceUrl({
        protoPath: "/v1/enterprise/items/{item.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "item.createdDate" },
            { path: "item.updatedDate" },
            { path: "item.assets.image.urlExpirationDate" },
            { path: "item.assets.video.urlExpirationDate" },
            { path: "item.assets.video.resolutions.urlExpirationDate" },
            {
              path: "item.assets.video.resolutions.poster.urlExpirationDate"
            },
            { path: "item.assets.video.posters.urlExpirationDate" },
            { path: "item.assets.vector.urlExpirationDate" },
            { path: "item.assets.audio.urlExpirationDate" },
            { path: "item.assets.document.urlExpirationDate" },
            { path: "item.assets.archive.urlExpirationDate" },
            { path: "item.assets.model3d.urlExpirationDate" },
            { path: "item.assets.model3d.thumbnail.urlExpirationDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateItem;
}
function bulkUpdateItem(payload) {
  function __bulkUpdateItem({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "updateItemRequests.fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "updateItemRequests.item.createdDate" },
          { path: "updateItemRequests.item.updatedDate" },
          { path: "updateItemRequests.item.assets.image.urlExpirationDate" },
          { path: "updateItemRequests.item.assets.video.urlExpirationDate" },
          {
            path: "updateItemRequests.item.assets.video.resolutions.urlExpirationDate"
          },
          {
            path: "updateItemRequests.item.assets.video.resolutions.poster.urlExpirationDate"
          },
          {
            path: "updateItemRequests.item.assets.video.posters.urlExpirationDate"
          },
          { path: "updateItemRequests.item.assets.vector.urlExpirationDate" },
          { path: "updateItemRequests.item.assets.audio.urlExpirationDate" },
          { path: "updateItemRequests.item.assets.document.urlExpirationDate" },
          { path: "updateItemRequests.item.assets.archive.urlExpirationDate" },
          { path: "updateItemRequests.item.assets.model3d.urlExpirationDate" },
          {
            path: "updateItemRequests.item.assets.model3d.thumbnail.urlExpirationDate"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_item",
      method: "PATCH",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseItemsService.BulkUpdateItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseItemsServiceUrl({
        protoPath: "/v1/bulk/enterprise/items/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.assets.image.urlExpirationDate" },
            { path: "results.item.assets.video.urlExpirationDate" },
            {
              path: "results.item.assets.video.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.assets.video.resolutions.poster.urlExpirationDate"
            },
            { path: "results.item.assets.video.posters.urlExpirationDate" },
            { path: "results.item.assets.vector.urlExpirationDate" },
            { path: "results.item.assets.audio.urlExpirationDate" },
            { path: "results.item.assets.document.urlExpirationDate" },
            { path: "results.item.assets.archive.urlExpirationDate" },
            { path: "results.item.assets.model3d.urlExpirationDate" },
            {
              path: "results.item.assets.model3d.thumbnail.urlExpirationDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateItem;
}
function getItem(payload) {
  function __getItem({ host }) {
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_item",
      method: "GET",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseItemsService.GetItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseItemsServiceUrl({
        protoPath: "/v1/enterprise/items/{itemId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "item.createdDate" },
            { path: "item.updatedDate" },
            { path: "item.assets.image.urlExpirationDate" },
            { path: "item.assets.video.urlExpirationDate" },
            { path: "item.assets.video.resolutions.urlExpirationDate" },
            {
              path: "item.assets.video.resolutions.poster.urlExpirationDate"
            },
            { path: "item.assets.video.posters.urlExpirationDate" },
            { path: "item.assets.vector.urlExpirationDate" },
            { path: "item.assets.audio.urlExpirationDate" },
            { path: "item.assets.document.urlExpirationDate" },
            { path: "item.assets.archive.urlExpirationDate" },
            { path: "item.assets.model3d.urlExpirationDate" },
            { path: "item.assets.model3d.thumbnail.urlExpirationDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getItem;
}
function linkItemToCategories(payload) {
  function __linkItemToCategories({ host }) {
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_item",
      method: "POST",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseItemsService.LinkItemToCategories",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseItemsServiceUrl({
        protoPath: "/v1/enterprise/items/{itemId}/link-categories",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __linkItemToCategories;
}
function unlinkItemFromCategories(payload) {
  function __unlinkItemFromCategories({ host }) {
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_item",
      method: "POST",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseItemsService.UnlinkItemFromCategories",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseItemsServiceUrl({
        protoPath: "/v1/enterprise/items/{itemId}/unlink-categories",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __unlinkItemFromCategories;
}
function overwriteItemCategories(payload) {
  function __overwriteItemCategories({ host }) {
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_item",
      method: "POST",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseItemsService.OverwriteItemCategories",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseItemsServiceUrl({
        protoPath: "/v1/enterprise/items/{itemId}/overwrite-categories",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __overwriteItemCategories;
}

// src/media-enterprise-public-media-v1-enterprise-item-enterprise-media-items.meta.ts
function itemUploadCallback2() {
  const payload = {};
  const getRequestOptions = itemUploadCallback(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/enterprise/items/upload/callback",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function generateFileUploadUrl2() {
  const payload = {};
  const getRequestOptions = generateFileUploadUrl(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/enterprise/items/upload/generate-url",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function importFile2() {
  const payload = {};
  const getRequestOptions = importFile(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/enterprise/items/upload/import",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkImportFiles2() {
  const payload = {};
  const getRequestOptions = bulkImportFiles(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/files/import",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function searchItems2() {
  const payload = {};
  const getRequestOptions = searchItems(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/enterprise/items/search",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryItems2() {
  const payload = {};
  const getRequestOptions = queryItems(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/enterprise/items/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateItem2() {
  const payload = { item: { id: ":itemId" } };
  const getRequestOptions = updateItem(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/enterprise/items/{item.id}",
    pathParams: { itemId: "itemId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateItem2() {
  const payload = {};
  const getRequestOptions = bulkUpdateItem(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/bulk/enterprise/items/update",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getItem2() {
  const payload = { itemId: ":itemId" };
  const getRequestOptions = getItem(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/enterprise/items/{itemId}",
    pathParams: { itemId: "itemId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function linkItemToCategories2() {
  const payload = { itemId: ":itemId" };
  const getRequestOptions = linkItemToCategories(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/enterprise/items/{itemId}/link-categories",
    pathParams: { itemId: "itemId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function unlinkItemFromCategories2() {
  const payload = { itemId: ":itemId" };
  const getRequestOptions = unlinkItemFromCategories(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/enterprise/items/{itemId}/unlink-categories",
    pathParams: { itemId: "itemId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function overwriteItemCategories2() {
  const payload = { itemId: ":itemId" };
  const getRequestOptions = overwriteItemCategories(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/enterprise/items/{itemId}/overwrite-categories",
    pathParams: { itemId: "itemId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  bulkImportFiles,
  bulkUpdateItem,
  generateFileUploadUrl,
  getItem,
  importFile,
  itemUploadCallback,
  linkItemToCategories,
  overwriteItemCategories,
  queryItems,
  searchItems,
  unlinkItemFromCategories,
  updateItem
});
