// @wix/auto_sdk_restaurants_service-fees/build/internal/es/index.typings.mjs
// src/service-fees-v1-rule-service-fees.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/service-fees-v1-rule-service-fees.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKFloatToRESTFloat } from "@wix/sdk-runtime/transformations/float";
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressServiceFeesV1ServiceFeesRulesUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "users._base_domain_": [
      {
        srcPath: "/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      }
    ],
    _base_domain_: [
      {
        srcPath: "/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/_api/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/_api/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/_api/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/_api/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/_api/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/service-fees-rules",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
function resolveComWixpressServiceFeesV1ServiceFeesCalculateUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "users._base_domain_": [
      {
        srcPath: "/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      }
    ],
    _base_domain_: [
      {
        srcPath: "/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/_api/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/_api/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/_api/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/_api/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/_api/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/service-fees-rules",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_service-fees";
function calculateServiceFees(payload) {
  function __calculateServiceFees({ host }) {
    const metadata = {
      entityFqdn: "wix.service.fees.v1.rule",
      method: "POST",
      methodFqn: "com.wixpress.service.fees.v1.ServiceFeesCalculate.CalculateServiceFees",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressServiceFeesV1ServiceFeesCalculateUrl({
        protoPath: "/v1/calculate",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __calculateServiceFees;
}
function createRule(payload) {
  function __createRule({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "rule.createdDate" }, { path: "rule.updatedDate" }]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "rule.condition.number.value" },
          { path: "rule.conditionTree.leftCondition.number.value" },
          { path: "rule.conditionTree.rightCondition.number.value" },
          { path: "rule.conditionOptions.number.value" },
          { path: "rule.conditionTreeOptions.leftCondition.number.value" },
          { path: "rule.conditionTreeOptions.rightCondition.number.value" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.service.fees.v1.rule",
      method: "POST",
      methodFqn: "com.wixpress.service.fees.v1.ServiceFeesRules.CreateRule",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressServiceFeesV1ServiceFeesRulesUrl({
        protoPath: "/v1/rules",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "rule.createdDate" }, { path: "rule.updatedDate" }]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "rule.condition.number.value" },
            { path: "rule.conditionTree.leftCondition.number.value" },
            { path: "rule.conditionTree.rightCondition.number.value" },
            { path: "rule.conditionOptions.number.value" },
            { path: "rule.conditionTreeOptions.leftCondition.number.value" },
            { path: "rule.conditionTreeOptions.rightCondition.number.value" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createRule;
}
function getRule(payload) {
  function __getRule({ host }) {
    const metadata = {
      entityFqdn: "wix.service.fees.v1.rule",
      method: "GET",
      methodFqn: "com.wixpress.service.fees.v1.ServiceFeesRules.GetRule",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressServiceFeesV1ServiceFeesRulesUrl({
        protoPath: "/v1/rules/{ruleId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "rule.createdDate" }, { path: "rule.updatedDate" }]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "rule.condition.number.value" },
            { path: "rule.conditionTree.leftCondition.number.value" },
            { path: "rule.conditionTree.rightCondition.number.value" },
            { path: "rule.conditionOptions.number.value" },
            { path: "rule.conditionTreeOptions.leftCondition.number.value" },
            { path: "rule.conditionTreeOptions.rightCondition.number.value" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getRule;
}
function updateRule(payload) {
  function __updateRule({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "rule.createdDate" }, { path: "rule.updatedDate" }]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "rule.condition.number.value" },
          { path: "rule.conditionTree.leftCondition.number.value" },
          { path: "rule.conditionTree.rightCondition.number.value" },
          { path: "rule.conditionOptions.number.value" },
          { path: "rule.conditionTreeOptions.leftCondition.number.value" },
          { path: "rule.conditionTreeOptions.rightCondition.number.value" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.service.fees.v1.rule",
      method: "PATCH",
      methodFqn: "com.wixpress.service.fees.v1.ServiceFeesRules.UpdateRule",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressServiceFeesV1ServiceFeesRulesUrl({
        protoPath: "/v1/rules/{rule.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "rule.createdDate" }, { path: "rule.updatedDate" }]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "rule.condition.number.value" },
            { path: "rule.conditionTree.leftCondition.number.value" },
            { path: "rule.conditionTree.rightCondition.number.value" },
            { path: "rule.conditionOptions.number.value" },
            { path: "rule.conditionTreeOptions.leftCondition.number.value" },
            { path: "rule.conditionTreeOptions.rightCondition.number.value" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateRule;
}
function deleteRule(payload) {
  function __deleteRule({ host }) {
    const metadata = {
      entityFqdn: "wix.service.fees.v1.rule",
      method: "DELETE",
      methodFqn: "com.wixpress.service.fees.v1.ServiceFeesRules.DeleteRule",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressServiceFeesV1ServiceFeesRulesUrl({
        protoPath: "/v1/rules/{ruleId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteRule;
}
function listRules(payload) {
  function __listRules({ host }) {
    const metadata = {
      entityFqdn: "wix.service.fees.v1.rule",
      method: "GET",
      methodFqn: "com.wixpress.service.fees.v1.ServiceFeesRules.ListRules",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressServiceFeesV1ServiceFeesRulesUrl({
        protoPath: "/v1/rules",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "rules.createdDate" },
            { path: "rules.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "rules.condition.number.value" },
            { path: "rules.conditionTree.leftCondition.number.value" },
            { path: "rules.conditionTree.rightCondition.number.value" },
            { path: "rules.conditionOptions.number.value" },
            { path: "rules.conditionTreeOptions.leftCondition.number.value" },
            {
              path: "rules.conditionTreeOptions.rightCondition.number.value"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listRules;
}
function queryRules(payload) {
  function __queryRules({ host }) {
    const metadata = {
      entityFqdn: "wix.service.fees.v1.rule",
      method: "POST",
      methodFqn: "com.wixpress.service.fees.v1.ServiceFeesRules.QueryRules",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressServiceFeesV1ServiceFeesRulesUrl({
        protoPath: "/v1/rules/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "rules.createdDate" },
            { path: "rules.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "rules.condition.number.value" },
            { path: "rules.conditionTree.leftCondition.number.value" },
            { path: "rules.conditionTree.rightCondition.number.value" },
            { path: "rules.conditionOptions.number.value" },
            { path: "rules.conditionTreeOptions.leftCondition.number.value" },
            {
              path: "rules.conditionTreeOptions.rightCondition.number.value"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryRules;
}
function bulkCreateRules(payload) {
  function __bulkCreateRules({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "rules.createdDate" }, { path: "rules.updatedDate" }]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "rules.condition.number.value" },
          { path: "rules.conditionTree.leftCondition.number.value" },
          { path: "rules.conditionTree.rightCondition.number.value" },
          { path: "rules.conditionOptions.number.value" },
          { path: "rules.conditionTreeOptions.leftCondition.number.value" },
          { path: "rules.conditionTreeOptions.rightCondition.number.value" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.service.fees.v1.rule",
      method: "POST",
      methodFqn: "com.wixpress.service.fees.v1.ServiceFeesRules.BulkCreateRules",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressServiceFeesV1ServiceFeesRulesUrl({
        protoPath: "/v1/bulk/rules/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.rule.createdDate" },
            { path: "results.rule.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.rule.condition.number.value" },
            { path: "results.rule.conditionTree.leftCondition.number.value" },
            {
              path: "results.rule.conditionTree.rightCondition.number.value"
            },
            { path: "results.rule.conditionOptions.number.value" },
            {
              path: "results.rule.conditionTreeOptions.leftCondition.number.value"
            },
            {
              path: "results.rule.conditionTreeOptions.rightCondition.number.value"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateRules;
}
function bulkUpdateRules(payload) {
  function __bulkUpdateRules({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "rules.mask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "rules.rule.createdDate" },
          { path: "rules.rule.updatedDate" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "rules.rule.condition.number.value" },
          { path: "rules.rule.conditionTree.leftCondition.number.value" },
          { path: "rules.rule.conditionTree.rightCondition.number.value" },
          { path: "rules.rule.conditionOptions.number.value" },
          {
            path: "rules.rule.conditionTreeOptions.leftCondition.number.value"
          },
          {
            path: "rules.rule.conditionTreeOptions.rightCondition.number.value"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.service.fees.v1.rule",
      method: "PATCH",
      methodFqn: "com.wixpress.service.fees.v1.ServiceFeesRules.BulkUpdateRules",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressServiceFeesV1ServiceFeesRulesUrl({
        protoPath: "/v1/bulk/rules/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.rule.createdDate" },
            { path: "results.rule.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.rule.condition.number.value" },
            { path: "results.rule.conditionTree.leftCondition.number.value" },
            {
              path: "results.rule.conditionTree.rightCondition.number.value"
            },
            { path: "results.rule.conditionOptions.number.value" },
            {
              path: "results.rule.conditionTreeOptions.leftCondition.number.value"
            },
            {
              path: "results.rule.conditionTreeOptions.rightCondition.number.value"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateRules;
}
function bulkDeleteRules(payload) {
  function __bulkDeleteRules({ host }) {
    const metadata = {
      entityFqdn: "wix.service.fees.v1.rule",
      method: "DELETE",
      methodFqn: "com.wixpress.service.fees.v1.ServiceFeesRules.BulkDeleteRules",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressServiceFeesV1ServiceFeesRulesUrl({
        protoPath: "/v1/bulk/rules/delete",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.rule.createdDate" },
            { path: "results.rule.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.rule.condition.number.value" },
            { path: "results.rule.conditionTree.leftCondition.number.value" },
            {
              path: "results.rule.conditionTree.rightCondition.number.value"
            },
            { path: "results.rule.conditionOptions.number.value" },
            {
              path: "results.rule.conditionTreeOptions.leftCondition.number.value"
            },
            {
              path: "results.rule.conditionTreeOptions.rightCondition.number.value"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkDeleteRules;
}
function bulkUpdateRuleTags(payload) {
  function __bulkUpdateRuleTags({ host }) {
    const metadata = {
      entityFqdn: "wix.service.fees.v1.rule",
      method: "POST",
      methodFqn: "com.wixpress.service.fees.v1.ServiceFeesRules.BulkUpdateRuleTags",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressServiceFeesV1ServiceFeesRulesUrl({
        protoPath: "/v1/bulk/rules/update-tags",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateRuleTags;
}
function bulkUpdateRuleTagsByFilter(payload) {
  function __bulkUpdateRuleTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.service.fees.v1.rule",
      method: "POST",
      methodFqn: "com.wixpress.service.fees.v1.ServiceFeesRules.BulkUpdateRuleTagsByFilter",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressServiceFeesV1ServiceFeesRulesUrl({
        protoPath: "/v1/bulk/rules/update-tags-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateRuleTagsByFilter;
}

// src/service-fees-v1-rule-service-fees.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var Value = /* @__PURE__ */ ((Value2) => {
  Value2["NUMBER"] = "NUMBER";
  Value2["LIST"] = "LIST";
  Value2["STRING"] = "STRING";
  return Value2;
})(Value || {});
var ExpectedFieldType = /* @__PURE__ */ ((ExpectedFieldType2) => {
  ExpectedFieldType2["UNKNOWN_EXPECTED_FIELD_TYPE"] = "UNKNOWN_EXPECTED_FIELD_TYPE";
  ExpectedFieldType2["NUMBER"] = "NUMBER";
  ExpectedFieldType2["LIST"] = "LIST";
  ExpectedFieldType2["STRING"] = "STRING";
  return ExpectedFieldType2;
})(ExpectedFieldType || {});
var Operation = /* @__PURE__ */ ((Operation2) => {
  Operation2["EQ"] = "EQ";
  Operation2["LT"] = "LT";
  Operation2["LE"] = "LE";
  Operation2["GT"] = "GT";
  Operation2["GE"] = "GE";
  return Operation2;
})(Operation || {});
var Operator = /* @__PURE__ */ ((Operator2) => {
  Operator2["AND"] = "AND";
  Operator2["OR"] = "OR";
  return Operator2;
})(Operator || {});
var ConditionsType = /* @__PURE__ */ ((ConditionsType2) => {
  ConditionsType2["NO_CONDITIONS"] = "NO_CONDITIONS";
  ConditionsType2["CONDITION"] = "CONDITION";
  ConditionsType2["CONDITION_TREE"] = "CONDITION_TREE";
  return ConditionsType2;
})(ConditionsType || {});
var ConditionType = /* @__PURE__ */ ((ConditionType2) => {
  ConditionType2["UNDEFINED_CONDITION_TYPE"] = "UNDEFINED_CONDITION_TYPE";
  ConditionType2["CONDITION"] = "CONDITION";
  ConditionType2["CONDITION_TREE"] = "CONDITION_TREE";
  return ConditionType2;
})(ConditionType || {});
var RoundingStrategy = /* @__PURE__ */ ((RoundingStrategy2) => {
  RoundingStrategy2["UNKNOWN_ROUNDING_STRATEGY"] = "UNKNOWN_ROUNDING_STRATEGY";
  RoundingStrategy2["HALF_UP"] = "HALF_UP";
  RoundingStrategy2["HALF_EVEN"] = "HALF_EVEN";
  return RoundingStrategy2;
})(RoundingStrategy || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNSPECIFIED_FULFILLMENT_TYPE"] = "UNSPECIFIED_FULFILLMENT_TYPE";
  Type2["PICKUP"] = "PICKUP";
  Type2["DELIVERY"] = "DELIVERY";
  Type2["DINE_IN"] = "DINE_IN";
  Type2["CURBSIDE_PICKUP"] = "CURBSIDE_PICKUP";
  return Type2;
})(Type || {});
var PlatformValue = /* @__PURE__ */ ((PlatformValue2) => {
  PlatformValue2["SITE"] = "SITE";
  PlatformValue2["MOBILE_SITE"] = "MOBILE_SITE";
  PlatformValue2["MOBILE_APP"] = "MOBILE_APP";
  return PlatformValue2;
})(PlatformValue || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function calculateServiceFees2(order, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    order,
    label: options?.label,
    appId: options?.appId
  });
  const reqOpts = calculateServiceFees(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          order: "$[0]",
          label: "$[1].label",
          appId: "$[1].appId"
        },
        singleArgumentUnchanged: false
      },
      ["order", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function createRule2(rule) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ rule });
  const reqOpts = createRule(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.rule;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { rule: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["rule"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getRule2(ruleId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ ruleId });
  const reqOpts = getRule(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.rule;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { ruleId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["ruleId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateRule2(_id, rule) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    rule: { ...rule, id: _id }
  });
  const reqOpts = updateRule(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.rule;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { rule: "$[1]" },
        explicitPathsToArguments: { "rule.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "rule"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteRule2(ruleId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ ruleId });
  const reqOpts = deleteRule(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { ruleId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["ruleId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listRules2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    locationId: options?.locationId,
    label: options?.label,
    appId: options?.appId
  });
  const reqOpts = listRules(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          locationId: "$[0].locationId",
          label: "$[0].label",
          appId: "$[0].appId"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryRules2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryRules(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [])
      );
      return {
        items: transformedData?.rules,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function bulkCreateRules2(rules, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    rules,
    returnFullEntity: options?.returnFullEntity
  });
  const reqOpts = bulkCreateRules(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          rules: "$[0]",
          returnFullEntity: "$[1].returnFullEntity"
        },
        singleArgumentUnchanged: false
      },
      ["rules", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateRules2(rules, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    rules,
    returnFullEntity: options?.returnFullEntity
  });
  const reqOpts = bulkUpdateRules(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          rules: "$[0]",
          returnFullEntity: "$[1].returnFullEntity"
        },
        singleArgumentUnchanged: false
      },
      ["rules", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkDeleteRules2(ruleIds) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ ruleIds });
  const reqOpts = bulkDeleteRules(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { ruleIds: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["ruleIds"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateRuleTags2(ruleIds, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    ruleIds,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateRuleTags(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          ruleIds: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["ruleIds", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateRuleTagsByFilter2(filter, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    filter,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateRuleTagsByFilter(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          filter: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["filter", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
export {
  ConditionType,
  ConditionsType,
  ExpectedFieldType,
  Operation,
  Operator,
  PlatformValue,
  RoundingStrategy,
  SortOrder,
  Type,
  Value,
  WebhookIdentityType,
  bulkCreateRules2 as bulkCreateRules,
  bulkDeleteRules2 as bulkDeleteRules,
  bulkUpdateRuleTags2 as bulkUpdateRuleTags,
  bulkUpdateRuleTagsByFilter2 as bulkUpdateRuleTagsByFilter,
  bulkUpdateRules2 as bulkUpdateRules,
  calculateServiceFees2 as calculateServiceFees,
  createRule2 as createRule,
  deleteRule2 as deleteRule,
  getRule2 as getRule,
  listRules2 as listRules,
  queryRules2 as queryRules,
  updateRule2 as updateRule
};
