// @wix/ricos/dist/cjs/ricos-common-plugins/bulleted-list/bullet-list.testkit.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var bullet_list_testkit_exports = {};
__export(bullet_list_testkit_exports, {
  createBulletListTestkit: () => createBulletListTestkit
});
module.exports = __toCommonJS(bullet_list_testkit_exports);
var import_wix_rich_content_common = require("./../../wix-rich-content-common/index.js");
var import_data_hooks = require("./data-hooks.js");
const createBulletListTestkit = (getBase) => {
  return {
    content: {
      atIndex: (index) => {
        return {
          getStyles: async () => {
            const el = (await getBase().getAll("li")).at(index);
            const style = await el._prop("style");
            return style;
          }
        };
      }
    },
    toolbars: {
      createTextButtonToolbars: (getTextButtonBase) => ({
        click: async () => {
          (await getTextButtonBase().get((0, import_wix_rich_content_common.dataHook)(import_data_hooks.TEXT_BUTTON_BULLETED_LIST))).click();
        }
      })
    }
  };
};
