// @wix/ricos/dist/cjs/ricos-common-plugins/bulleted-list/bulleted-list-converter.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var bulleted_list_converter_exports = {};
__export(bulleted_list_converter_exports, {
  bulletedListConverter: () => bulletedListConverter
});
module.exports = __toCommonJS(bulleted_list_converter_exports);
var import_v1 = require("./../../ricos-schema/legacy/wix/rich_content/v1/index.js");
const bulletedListDataDefaults = { indentation: 0, offset: 0 };
const bulletedListConverter = {
  toPm: {
    types: [import_v1.Node_Type.BULLETED_LIST],
    // @ts-expect-error $TSFixMe
    convert: (node, visit) => {
      const dataWithDefaults = Object.entries(bulletedListDataDefaults).reduce((acc, [key, defaultValue]) => {
        acc[key] = node.bulletedListData?.[key] ?? defaultValue;
        return acc;
      }, {});
      return {
        type: import_v1.Node_Type.BULLETED_LIST,
        attrs: {
          ...dataWithDefaults,
          id: node.id
        },
        content: visit(node)
      };
    }
  },
  fromPm: {
    types: [import_v1.Node_Type.BULLETED_LIST],
    // @ts-expect-error $TSFixMe
    convert: (node, visit) => {
      const { attrs = {} } = node;
      const { id, ...data } = attrs;
      return {
        type: import_v1.Node_Type.BULLETED_LIST,
        id,
        nodes: visit(node),
        bulletedListData: Object.keys(data).reduce((acc, key) => {
          if (data[key] !== bulletedListDataDefaults[key]) {
            acc[key] = data[key];
          }
          return acc;
        }, {})
      };
    }
  }
};
