// @wix/headless-stores/dist/services/product-service.js
import { defineService, implementService } from '@wix/services-definitions';
import { SignalsServiceDefinition, } from '@wix/services-definitions/core-services/signals';
import * as productsV3 from '@wix/auto_sdk_stores_products-v-3';
/**
 * Service definition for the Product service.
 * This defines the contract that the ProductService must implement.
 *
 * @constant
 */
export const ProductServiceDefinition = defineService('product');
/**
 * Implementation of the Product service that manages reactive product data.
 * This service provides signals for product data, loading state, and error handling,
 * along with methods to dynamically load products.
 *
 * @example
 * ```tsx
 * import { ProductService, ProductServiceDefinition } from '@wix/stores/services';
 * import { useService } from '@wix/services-manager-react';
 *
 * function ProductComponent({ productConfig }) {
 *   return (
 *     <ServiceProvider services={createServicesMap([
 *       [ProductServiceDefinition, ProductService.withConfig(productConfig)]
 *     ])}>
 *       <ProductDisplay />
 *     </ServiceProvider>
 *   );
 * }
 *
 * function ProductDisplay() {
 *   const productService = useService(ProductServiceDefinition);
 *   const product = productService.product.get();
 *   const isLoading = productService.isLoading.get();
 *   const error = productService.error.get();
 *
 *   if (isLoading) return <div>Loading...</div>;
 *   if (error) return <div>Error: {error}</div>;
 *
 *   return <h1>{product.name}</h1>;
 * }
 * ```
 */
export const ProductService = implementService.withConfig()(ProductServiceDefinition, ({ getService, config }) => {
    const signalsService = getService(SignalsServiceDefinition);
    const product = signalsService.signal(config.product);
    const isLoading = signalsService.signal(!!config.productSlug);
    const error = signalsService.signal(null);
    const loadProduct = async (slug) => {
        isLoading.set(true);
        const productResponse = await loadProductBySlug(slug);
        if (!productResponse.product) {
            error.set('Product not found');
        }
        else {
            product.set(productResponse.product);
            error.set(null);
        }
        isLoading.set(false);
    };
    if (config.productSlug) {
        loadProduct(config.productSlug);
    }
    return {
        product,
        isLoading,
        error,
        loadProduct,
    };
});
/**
 * Internal helper function to load a product by its slug from the Wix Products API.
 * Fetches comprehensive product data including description, categories, media, etc.
 *
 * @private
 * @param {string} slug - The product slug to load
 * @returns {Promise} Product response from the API
 */
const loadProductBySlug = async (slug) => {
    const productResponse = await productsV3.getProductBySlug(slug, {
        fields: [
            'DESCRIPTION',
            'DIRECT_CATEGORIES_INFO',
            'BREADCRUMBS_INFO',
            'INFO_SECTION',
            'MEDIA_ITEMS_INFO',
            'PLAIN_DESCRIPTION',
            'THUMBNAIL',
            'URL',
            'VARIANT_OPTION_CHOICE_NAMES',
            'WEIGHT_MEASUREMENT_UNIT_INFO',
        ],
    });
    return productResponse;
};
/**
 * Loads product service configuration from the Wix Products API for SSR initialization.
 * This function is designed to be used during Server-Side Rendering (SSR) to preload
 * a specific product by slug that will be used to configure the ProductService.
 *
 * @param productSlug The product slug to load
 * @returns Promise that resolves to ProductServiceConfigResult (success with config or notFound)
 *
 * @example
 * ```astro
 * ---
 * // Astro page example - pages/product/[slug].astro
 * import { loadProductServiceConfig } from '@wix/stores/services';
 * import { Product } from '@wix/stores/components';
 *
 * // Get product slug from URL params
 * const { slug } = Astro.params;
 *
 * // Load product data during SSR
 * const productResult = await loadProductServiceConfig(slug);
 *
 * // Handle not found case
 * if (productResult.type === 'notFound') {
 *   return Astro.redirect('/404');
 * }
 * ---
 *
 * <Product.Root productConfig={productResult.config}>
 *   <Product.Name>
 *     {({ name }) => <h1>{name}</h1>}
 *   </Product.Name>
 * </Product.Root>
 * ```
 *
 * @example
 * ```tsx
 * // Next.js page example - pages/product/[slug].tsx
 * import { GetServerSideProps } from 'next';
 * import { loadProductServiceConfig } from '@wix/stores/services';
 * import { Product } from '@wix/stores/components';
 *
 * interface ProductPageProps {
 *   productConfig: Extract<Awaited<ReturnType<typeof loadProductServiceConfig>>, { type: 'success' }>['config'];
 * }
 *
 * export const getServerSideProps: GetServerSideProps<ProductPageProps> = async ({ params }) => {
 *   const slug = params?.slug as string;
 *
 *   // Load product data during SSR
 *   const productResult = await loadProductServiceConfig(slug);
 *
 *   // Handle not found case
 *   if (productResult.type === 'notFound') {
 *     return {
 *       notFound: true,
 *     };
 *   }
 *
 *   return {
 *     props: {
 *       productConfig: productResult.config,
 *     },
 *   };
 * };
 *
 * export default function ProductPage({ productConfig }: ProductPageProps) {
 *   return (
 *     <Product.Root productConfig={productConfig}>
 *       <Product.Name>
 *         {({ name }) => <h1>{name}</h1>}
 *       </Product.Name>
 *     </Product.Root>
 *   );
 * }
 * ```
 */
export async function loadProductServiceConfig(productSlug) {
    try {
        // Use getProductBySlug directly - single API call with comprehensive fields
        const productResponse = await loadProductBySlug(productSlug);
        if (!productResponse.product) {
            return { type: 'notFound' };
        }
        return {
            type: 'success',
            config: {
                product: productResponse.product,
            },
        };
    }
    catch (error) {
        console.error(`Failed to load product for slug "${productSlug}":`, error);
        return { type: 'notFound' };
    }
}
