// @wix/auto-patterns/dist/esm/components/AutoPatternsRoute/AutoPatternsRoutes.js
import React from 'react';
import { usePagePath } from '../../hooks';
import { PatternsReactRoute, PatternsReactRouter } from '@wix/patterns/router';
import { AutoPatternsPage } from './AutoPatternsPage';
import { Navigate } from 'react-router-dom';
const pageTypeToRouteType = {
  collectionPage: 'collection',
  entityPage: 'editEntity'
};
export const AutoPatternsRoutes = _ref => {
  let {
    config
  } = _ref;
  const {
    getPagePath
  } = usePagePath(config);
  const mainPage = config.pages.find(page => page.appMainPage);
  const routes = config.pages.map(page => {
    const routePath = getPagePath(page.id);
    if (!routePath) {
      return null;
    }
    return /*#__PURE__*/React.createElement(PatternsReactRoute, {
      key: page.id,
      type: pageTypeToRouteType[page.type] ?? 'other',
      path: routePath,
      element: /*#__PURE__*/React.createElement(AutoPatternsPage, {
        pageConfig: page
      })
    });
  }).filter(Boolean);
  if (mainPage) {
    const mainPagePath = getPagePath(mainPage.id);
    if (mainPagePath) {
      routes.push(/*#__PURE__*/React.createElement(PatternsReactRoute, {
        key: "main-redirect",
        type: "other",
        path: "/",
        element: /*#__PURE__*/React.createElement(Navigate, {
          to: mainPagePath
        })
      }));
    }
  }
  return /*#__PURE__*/React.createElement(PatternsReactRouter, null, routes);
};
