// @wix/runtime-fetch-adapter/dist/adapter.js
import { fileURLToPath } from 'node:url';
import { join } from 'node:path';
import { writeFile, mkdir } from 'node:fs/promises';
/**
 * Wix Runtime Fetch Adapter integration.
 * Configures SSR output, sets the adapter, and emits a stable manifest re-export.
 */
export function createIntegration(options = {}) {
    const normalizedOptions = {
        namespacedEnv: options.namespacedEnv ?? false,
        ...options
    };
    return {
        name: 'wix-runtime-fetch-adapter',
        hooks: {
            /** Configure Astro/Vite for SSR with fully bundled server output. */
            'astro:config:setup': ({ updateConfig, logger }) => {
                logger.info('🚀 Setting up Wix Runtime Fetch Adapter...');
                updateConfig({
                    output: 'server',
                    base: '/',
                    site: undefined,
                    build: { assets: '_astro' },
                    vite: {
                        root: process.cwd(),
                        ssr: { noExternal: true },
                        build: { ssr: true, emptyOutDir: false, rollupOptions: { external: [], output: {} } },
                        resolve: { preserveSymlinks: false }
                    }
                });
                logger.info('✅ Vite configuration updated for platform-agnostic bundling');
            },
            /** Set adapter capabilities. */
            'astro:config:done': ({ setAdapter, logger }) => {
                logger.info('⚙️ Setting adapter configuration...');
                logger.info(`🔧 Environment access pattern: ${normalizedOptions.namespacedEnv ? 'namespaced (locals.env.VAR)' : 'direct (locals.VAR)'}`);
                setAdapter(getAdapter(normalizedOptions));
                logger.info('✅ Adapter configuration complete');
            },
            /** Prune path-bearing fields from manifest before serialization. */
            'astro:build:ssr': async ({ manifest, logger }) => {
                logger.info('🔧 Intercepting SSR manifest to sanitize paths...');
                if (manifest && typeof manifest === 'object') {
                    const m = manifest;
                    m.hrefRoot = 'file:///';
                    for (const key of ['cacheDir', 'outDir', 'srcDir', 'publicDir', 'buildClientDir', 'buildServerDir']) {
                        if (key in m)
                            delete m[key];
                    }
                    if ('componentMetadata' in m)
                        delete m.componentMetadata;
                    logger.info('✅ Pruned local path fields from manifest');
                }
            },
            /** Emit a stable server/manifest.mjs that re-exports the virtual manifest. */
            'astro:build:done': async ({ dir, logger, routes }) => {
                try {
                    const outDir = fileURLToPath(dir);
                    const serverDir = join(outDir, '..', 'server');
                    await mkdir(serverDir, { recursive: true });
                    const stableManifestPath = join(serverDir, 'manifest.mjs');
                    await writeFile(stableManifestPath, 'export { manifest } from "@astrojs/manifest";\n');
                    logger.info('✅ Created stable manifest: manifest.mjs');
                    logger.info('📦 Self-contained server bundle created');
                    logger.info(`📍 Output directory: ${outDir}`);
                    logger.info(`🚏 Routes processed: ${routes?.length || 0}`);
                }
                catch (error) {
                    logger.error('❌ Failed to complete build process');
                    logger.error(`Error: ${error instanceof Error ? error.message : 'Unknown error'}`);
                    throw new Error(`Wix Runtime Fetch Adapter: Failed to complete build process - ${error instanceof Error ? error.message : 'Unknown error'}`);
                }
            }
        }
    };
}
/**
 * Adapter descriptor provided to Astro.
 */
function getAdapter(options) {
    return {
        name: 'wix-runtime-fetch-adapter',
        serverEntrypoint: '@wix/runtime-fetch-adapter/entry/server',
        exports: ['default'],
        adapterFeatures: { edgeMiddleware: false, buildOutput: 'server' },
        supportedAstroFeatures: {
            hybridOutput: 'stable',
            staticOutput: 'stable',
            serverOutput: 'stable',
            envGetSecret: 'stable',
            sharpImageService: {
                support: 'limited',
                message: 'Sharp image service is not supported'
            }
        }
    };
}
