// @wix/auto_sdk_faq_category/build/es/meta.mjs
// src/faq-category-v2-category-category.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKFloatToRESTFloat } from "@wix/sdk-runtime/transformations/float";
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixFaqCategoryV2CategoryServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/category",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/category",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/category",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/faq-server/v2/categories",
        destPath: "/v2/categories"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-categories",
        destPath: "/v2/draft-categories"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/faq-server/v2/categories",
        destPath: "/v2/categories"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-categories",
        destPath: "/v2/draft-categories"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/faq/v2/categories",
        destPath: "/v2/categories"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-categories",
        destPath: "/v2/draft-categories"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/faq/v2/categories",
        destPath: "/v2/categories"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/faq-server/v2/categories",
        destPath: "/v2/categories"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-categories",
        destPath: "/v2/draft-categories"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/faq-server/v2/categories",
        destPath: "/v2/categories"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-categories",
        destPath: "/v2/draft-categories"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/faq-server/v2/categories",
        destPath: "/v2/categories"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-categories",
        destPath: "/v2/draft-categories"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/faq-server/v2/categories",
        destPath: "/v2/categories"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-categories",
        destPath: "/v2/draft-categories"
      }
    ],
    _: [
      {
        srcPath: "/_api/faq-server/v2/categories",
        destPath: "/v2/categories"
      }
    ],
    "dev._base_domain_": [
      {
        srcPath: "/_api/faq-server/v2/categories",
        destPath: "/v2/categories"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-categories",
        destPath: "/v2/draft-categories"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/faq-server/v2/draft-categories",
        destPath: "/v2/draft-categories"
      },
      {
        srcPath: "/_api/faq-server/v2/categories",
        destPath: "/v2/categories"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_faq_category";
function createCategory(payload) {
  function __createCategory({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "category.createdDate" },
          { path: "category.updatedDate" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [{ path: "category.sortOrder" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.faq.category.v2.category",
      method: "POST",
      methodFqn: "wix.faq.category.v2.CategoryService.CreateCategory",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqCategoryV2CategoryServiceUrl({
        protoPath: "/v2/categories",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "category.createdDate" },
            { path: "category.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [{ path: "category.sortOrder" }]
        }
      ])
    };
    return metadata;
  }
  return __createCategory;
}
function getCategory(payload) {
  function __getCategory({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.category.v2.category",
      method: "GET",
      methodFqn: "wix.faq.category.v2.CategoryService.GetCategory",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqCategoryV2CategoryServiceUrl({
        protoPath: "/v2/categories/{categoryId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "category.createdDate" },
            { path: "category.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [{ path: "category.sortOrder" }]
        }
      ])
    };
    return metadata;
  }
  return __getCategory;
}
function updateCategory(payload) {
  function __updateCategory({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "category.createdDate" },
          { path: "category.updatedDate" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [{ path: "category.sortOrder" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.faq.category.v2.category",
      method: "PATCH",
      methodFqn: "wix.faq.category.v2.CategoryService.UpdateCategory",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqCategoryV2CategoryServiceUrl({
        protoPath: "/v2/categories/{category.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "category.createdDate" },
            { path: "category.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [{ path: "category.sortOrder" }]
        }
      ])
    };
    return metadata;
  }
  return __updateCategory;
}
function deleteCategory(payload) {
  function __deleteCategory({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.category.v2.category",
      method: "DELETE",
      methodFqn: "wix.faq.category.v2.CategoryService.DeleteCategory",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqCategoryV2CategoryServiceUrl({
        protoPath: "/v2/categories/{categoryId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteCategory;
}
function listCategories(payload) {
  function __listCategories({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.category.v2.category",
      method: "GET",
      methodFqn: "wix.faq.category.v2.CategoryService.ListCategories",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqCategoryV2CategoryServiceUrl({
        protoPath: "/v2/categories",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "categories.createdDate" },
            { path: "categories.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [{ path: "categories.sortOrder" }]
        }
      ])
    };
    return metadata;
  }
  return __listCategories;
}
function queryCategories(payload) {
  function __queryCategories({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.category.v2.category",
      method: "POST",
      methodFqn: "wix.faq.category.v2.CategoryService.QueryCategories",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqCategoryV2CategoryServiceUrl({
        protoPath: "/v2/categories/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "categories.createdDate" },
            { path: "categories.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [{ path: "categories.sortOrder" }]
        }
      ])
    };
    return metadata;
  }
  return __queryCategories;
}
function updateExtendedFields(payload) {
  function __updateExtendedFields({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.category.v2.category",
      method: "POST",
      methodFqn: "wix.faq.category.v2.CategoryService.UpdateExtendedFields",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqCategoryV2CategoryServiceUrl({
        protoPath: "/v2/categories/{id}/update-extended-fields",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "category.createdDate" },
            { path: "category.updatedDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [{ path: "category.sortOrder" }]
        }
      ])
    };
    return metadata;
  }
  return __updateExtendedFields;
}

// src/faq-category-v2-category-category.meta.ts
function createCategory2() {
  const payload = {};
  const getRequestOptions = createCategory(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/categories",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getCategory2() {
  const payload = { categoryId: ":categoryId" };
  const getRequestOptions = getCategory(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/categories/{categoryId}",
    pathParams: { categoryId: "categoryId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateCategory2() {
  const payload = { category: { id: ":categoryId" } };
  const getRequestOptions = updateCategory(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v2/categories/{category.id}",
    pathParams: { categoryId: "categoryId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteCategory2() {
  const payload = { categoryId: ":categoryId" };
  const getRequestOptions = deleteCategory(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v2/categories/{categoryId}",
    pathParams: { categoryId: "categoryId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listCategories2() {
  const payload = {};
  const getRequestOptions = listCategories(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/categories",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryCategories2() {
  const payload = {};
  const getRequestOptions = queryCategories(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/categories/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateExtendedFields2() {
  const payload = { id: ":id" };
  const getRequestOptions = updateExtendedFields(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/categories/{id}/update-extended-fields",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  createCategory2 as createCategory,
  deleteCategory2 as deleteCategory,
  getCategory2 as getCategory,
  listCategories2 as listCategories,
  queryCategories2 as queryCategories,
  updateCategory2 as updateCategory,
  updateExtendedFields2 as updateExtendedFields
};
