// @wix/auto_sdk_forms_chat-settings/build/es/index.typings.mjs
// src/forms-ai-v1-chat-settings-chat-settings.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/forms-ai-v1-chat-settings-chat-settings.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixFormsAiV1ChatSettingsServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/_api/form-ai-assistant",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/form-ai-assistant",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/form-ai-assistant",
        destPath: ""
      }
    ],
    "*.wixforms.com": [
      {
        srcPath: "/_api/form-ai-assistant",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/form-ai-assistant",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/form-ai-assistant",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/forms/ai/v1/chat-settings",
        destPath: "/v1/chat-settings"
      },
      {
        srcPath: "/forms/ai/v1/interactive-form-sessions",
        destPath: "/v1/interactive-form-sessions"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/form-ai-assistant",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_forms_chat-settings";
function createChatSettings(payload) {
  function __createChatSettings({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "chatSettings.createdDate" },
          { path: "chatSettings.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.forms.ai.v1.chat_settings",
      method: "POST",
      methodFqn: "wix.forms.ai.v1.ChatSettingsService.CreateChatSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsAiV1ChatSettingsServiceUrl({
        protoPath: "/v1/chat-settings",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "chatSettings.createdDate" },
            { path: "chatSettings.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createChatSettings;
}
function updateChatSettings(payload) {
  function __updateChatSettings({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "chatSettings.createdDate" },
          { path: "chatSettings.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.forms.ai.v1.chat_settings",
      method: "PATCH",
      methodFqn: "wix.forms.ai.v1.ChatSettingsService.UpdateChatSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsAiV1ChatSettingsServiceUrl({
        protoPath: "/v1/chat-settings/{chatSettings.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "chatSettings.createdDate" },
            { path: "chatSettings.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateChatSettings;
}
function deleteChatSettings(payload) {
  function __deleteChatSettings({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.ai.v1.chat_settings",
      method: "DELETE",
      methodFqn: "wix.forms.ai.v1.ChatSettingsService.DeleteChatSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsAiV1ChatSettingsServiceUrl({
        protoPath: "/v1/chat-settings/{chatSettingsId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteChatSettings;
}
function getChatSettings(payload) {
  function __getChatSettings({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.ai.v1.chat_settings",
      method: "GET",
      methodFqn: "wix.forms.ai.v1.ChatSettingsService.GetChatSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsAiV1ChatSettingsServiceUrl({
        protoPath: "/v1/chat-settings/{chatSettingsId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "chatSettings.createdDate" },
            { path: "chatSettings.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getChatSettings;
}
function queryChatSettings(payload) {
  function __queryChatSettings({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.ai.v1.chat_settings",
      method: "GET",
      methodFqn: "wix.forms.ai.v1.ChatSettingsService.QueryChatSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsAiV1ChatSettingsServiceUrl({
        protoPath: "/v1/chat-settings/query",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload, true),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "chatSettings.createdDate" },
            { path: "chatSettings.updatedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixFormsAiV1ChatSettingsServiceUrl({
            protoPath: "/v1/chat-settings/query",
            data: payload,
            host
          }),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryChatSettings;
}

// src/forms-ai-v1-chat-settings-chat-settings.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createChatSettings2(chatSettings) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    chatSettings
  });
  const reqOpts = createChatSettings(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.chatSettings;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { chatSettings: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["chatSettings"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateChatSettings2(_id, chatSettings) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    chatSettings: { ...chatSettings, id: _id }
  });
  const reqOpts = updateChatSettings(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.chatSettings;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { chatSettings: "$[1]" },
        explicitPathsToArguments: { "chatSettings.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "chatSettings"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteChatSettings2(chatSettingsId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    chatSettingsId
  });
  const reqOpts = deleteChatSettings(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { chatSettingsId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["chatSettingsId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getChatSettings2(chatSettingsId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    chatSettingsId
  });
  const reqOpts = getChatSettings(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.chatSettings;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { chatSettingsId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["chatSettingsId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryChatSettings2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryChatSettings(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [])
      );
      return {
        items: transformedData?.chatSettings,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
export {
  SortOrder,
  WebhookIdentityType,
  createChatSettings2 as createChatSettings,
  deleteChatSettings2 as deleteChatSettings,
  getChatSettings2 as getChatSettings,
  queryChatSettings2 as queryChatSettings,
  updateChatSettings2 as updateChatSettings
};
