// @wix/astro/build/integration/index.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  createIntegration,
  createIntegration2,
  createIntegration3,
  createIntegration4,
  createIntegration5,
  createIntegration6,
  createIntegration7,
  defaultOutdent,
  init_esm_shims,
  normalizePath
} from "./chunk-JFJIM45R.js";

// src/index.ts
init_esm_shims();

// ../astro-auth/src/index.ts
init_esm_shims();
import { fileURLToPath } from "url";

// ../astro-auth/src/plugins/patchAstroInlineScripts.ts
init_esm_shims();
function patchAstroInlineScripts() {
  return {
    name: "patch-astro-inline-scripts",
    transform(code, id) {
      const [_filename, rawQuery] = id.split(`?`, 2);
      const query = Object.fromEntries(new URLSearchParams(rawQuery).entries());
      if (query.astro == null || query.type !== "script" || query["lang.ts"] == null || code === "") {
        return null;
      }
      return defaultOutdent`
        import 'astro:scripts/page.js';

        ${code}
      `;
    }
  };
}

// ../astro-auth/src/plugins/setupContentCollectionContext.ts
init_esm_shims();
import { relative } from "path";
var contentCollectionFilePaths = [
  "content.config.mjs",
  "content.config.js",
  "content.config.mts",
  "content.config.ts",
  "content/config.mjs",
  "content/config.js",
  "content/config.mts",
  "content/config.ts"
];
var setupContextualClientUrl = new URL(
  "../dependencies/astro-auth/backend-runtime/runtime/backend/setupContextualClient.js",
  import.meta.url
);
function setupContentCollectionContext(srcDir) {
  return {
    name: "setup-content-collection-context",
    applyToEnvironment(environment) {
      return environment.name === "ssr";
    },
    transform(code, id) {
      const relativeId = relative(srcDir, id);
      const isContentCollectionFile = contentCollectionFilePaths.includes(relativeId);
      if (isContentCollectionFile) {
        return defaultOutdent`
          import '${normalizePath(setupContextualClientUrl)}';

          ${code}
        `;
      }
      return null;
    }
  };
}

// ../astro-auth/src/index.ts
var setupContextUrl = new URL(
  "../dependencies/astro-auth/browser-runtime/setup.js",
  import.meta.url
);
var createIntegration8 = ({
  enableAuthRoutes
}) => {
  return {
    name: "@wix/astro/auth",
    hooks: {
      "astro:config:setup"({
        addMiddleware,
        config,
        injectRoute,
        injectScript,
        updateConfig
      }) {
        const srcDir = fileURLToPath(config.srcDir);
        updateConfig({
          vite: {
            plugins: [
              setupContentCollectionContext(srcDir),
              patchAstroInlineScripts()
            ]
          }
        });
        const setupContextCode = defaultOutdent`
          import { WIX_CLIENT_ID } from 'astro:env/client';
          import { setup } from '${normalizePath(setupContextUrl)}';

          setup({ clientId: WIX_CLIENT_ID });
        `;
        injectScript("before-hydration", setupContextCode);
        injectScript("page", setupContextCode);
        addMiddleware({
          entrypoint: new URL(
            "../dependencies/astro-auth/backend-runtime/middleware/auth.js",
            import.meta.url
          ),
          order: "pre"
        });
        if (enableAuthRoutes) {
          injectRoute({
            entrypoint: new URL(
              "../dependencies/astro-auth/backend-runtime/routes/login.js",
              import.meta.url
            ),
            pattern: "/api/auth/login",
            prerender: false
          });
          injectRoute({
            entrypoint: new URL(
              "../dependencies/astro-auth/backend-runtime/routes/logout.js",
              import.meta.url
            ),
            pattern: "/api/auth/logout",
            prerender: false
          });
          injectRoute({
            entrypoint: new URL(
              "../dependencies/astro-auth/backend-runtime/routes/callback.js",
              import.meta.url
            ),
            pattern: "/api/auth/callback",
            prerender: false
          });
          injectRoute({
            entrypoint: new URL(
              "../dependencies/astro-auth/backend-runtime/routes/logout-callback.js",
              import.meta.url
            ),
            pattern: "/api/auth/logout-callback",
            prerender: false
          });
        }
      }
    }
  };
};
var src_default = createIntegration8;

// ../astro-html-embeds/src/index.ts
init_esm_shims();
var createIntegration9 = () => {
  return {
    name: "@wix/astro/html-embeds",
    hooks: {
      "astro:config:setup"({ addMiddleware }) {
        addMiddleware({
          entrypoint: new URL(
            "../dependencies/astro-html-embeds/backend-runtime/html-embeds.js",
            import.meta.url
          ),
          order: "post"
        });
      }
    }
  };
};
var src_default2 = createIntegration9;

// ../astro-payment-links/src/index.ts
init_esm_shims();
var createIntegration10 = () => {
  return {
    name: "@wix/astro/payment-links",
    hooks: {
      "astro:config:setup"({ injectRoute }) {
        injectRoute({
          entrypoint: new URL(
            "../dependencies/astro-payment-links/backend-runtime/payment-link.js",
            import.meta.url
          ),
          pattern: "/_paylink/[id]",
          prerender: false
        });
        injectRoute({
          entrypoint: new URL(
            "../dependencies/astro-payment-links/backend-runtime/payment-checkout.js",
            import.meta.url
          ),
          pattern: "/checkout",
          prerender: false
        });
        injectRoute({
          entrypoint: new URL(
            "../dependencies/astro-payment-links/backend-runtime/payment-checkout.js",
            import.meta.url
          ),
          pattern: "/__ecom/checkout",
          prerender: false
        });
      }
    }
  };
};
var src_default3 = createIntegration10;

// ../astro-robots/src/index.ts
init_esm_shims();
var createIntegration11 = () => {
  return {
    name: "@wix/astro/robots",
    hooks: {
      "astro:config:setup"({ injectRoute }) {
        injectRoute({
          entrypoint: new URL(
            "../dependencies/astro-robots/backend-runtime/robots.js",
            import.meta.url
          ),
          pattern: "/robots.txt",
          prerender: false
        });
      }
    }
  };
};
var src_default4 = createIntegration11;

// ../astro-viewer-api/src/index.ts
init_esm_shims();

// ../astro-viewer-api/src/integration.ts
init_esm_shims();
var createIntegration12 = () => {
  return {
    name: "@wix/astro/viewer-api",
    hooks: {
      "astro:config:setup"({ injectRoute }) {
        injectRoute({
          entrypoint: new URL(
            "../dependencies/astro-viewer-api/backend-runtime/api.js",
            import.meta.url
          ),
          pattern: "/_api/[...path]",
          prerender: false
        });
      }
    }
  };
};

// src/index.ts
function isEnabled(value) {
  if (typeof value === "boolean") {
    return value;
  }
  return value?.enable ?? true;
}
var createIntegration13 = (config) => {
  const enableAuthRoutes = isEnabled(config?.auth);
  const enableHtmlEmbeds = isEnabled(config?.htmlEmbeds);
  const enableRobots = isEnabled(config?.robots);
  return {
    name: "@wix/astro",
    hooks: {
      async "astro:config:setup"({ updateConfig }) {
        updateConfig({
          integrations: [
            createIntegration3({
              integrations: [
                src_default({ enableAuthRoutes }),
                createIntegration(),
                createIntegration2(),
                createIntegration7(),
                createIntegration4(),
                ...enableHtmlEmbeds ? [src_default2()] : [],
                src_default3(),
                createIntegration5(),
                createIntegration6(),
                ...enableRobots ? [src_default4()] : [],
                createIntegration12()
              ]
            })
          ]
        });
      }
    }
  };
};
var index_default = createIntegration13;
export {
  index_default as default
};
