// pro-gallery/dist/esm/components/item/3d/3dItem.js
import React, { useEffect } from 'react';
import { use3DItem } from './hooks.js';
import { GalleryUI } from '../media/GalleryUI.js';
export default function ThreeDItem(props) {
    const { canvasRef, isLoaded } = use3DItem(props);
    const canvas = (React.createElement("canvas", { key: 'canvas', width: '100%', height: '100%', ref: canvasRef, style: {
            background: 'transparent',
            opacity: isLoaded ? 1 : 0,
            transition: 'opacity 200ms',
        } }));
    useOrbitControlsClickPropagation(props);
    return (React.createElement(React.Fragment, null,
        canvas,
        props.thumbnailWithOverride({
            extraClasses: [props.placeholderExtraClasses, isLoaded ? 'three-d-loaded' : 'three-d-loading'].join(' '),
            overlay: React.createElement(GalleryUI, { type: 'rotateArrow', size: 60 }),
        }),
        props.hover));
}
function useOrbitControlsClickPropagation(props) {
    useEffect(() => {
        const container = props.itemContainer.current;
        if (!props.shouldPlay && container) {
            const onMouseDown = (e) => {
                e.stopImmediatePropagation();
                const itemWrapper = container.querySelector('.gallery-item-wrapper');
                if (itemWrapper) {
                    itemWrapper.click();
                }
            };
            container.addEventListener('mousedown', onMouseDown);
            return () => container.removeEventListener('mousedown', onMouseDown);
        }
        return;
    }, [props.shouldPlay]);
}
