// pro-gallery/dist/esm/components/item/3d/hooks.js
import { GALLERY_CONSTS } from 'pro-gallery-lib';
import { useSceneManager } from '../../helpers/3dManager.js';
import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
function mapSceneToStyleParams(scene, options) {
    var _a, _b, _c, _d, _e, _f, _g;
    return {
        behaviourParams_item_threeDimensionalScene_transform_rotation: ((_a = scene.transform) === null || _a === void 0 ? void 0 : _a.rotation) || options.behaviourParams_item_threeDimensionalScene_transform_rotation,
        behaviourParams_item_threeDimensionalScene_transform_scale: ((_b = scene.transform) === null || _b === void 0 ? void 0 : _b.scale) || options.behaviourParams_item_threeDimensionalScene_transform_scale,
        behaviourParams_item_threeDimensionalScene_transform_position: ((_c = scene.transform) === null || _c === void 0 ? void 0 : _c.position) || options.behaviourParams_item_threeDimensionalScene_transform_position,
        behaviourParams_item_threeDimensionalScene_controls_enablePan: ((_d = scene.controls) === null || _d === void 0 ? void 0 : _d.enablePan) || options.behaviourParams_item_threeDimensionalScene_controls_enablePan,
        behaviourParams_item_threeDimensionalScene_controls_enableRotate: ((_e = scene.controls) === null || _e === void 0 ? void 0 : _e.enableRotate) || options.behaviourParams_item_threeDimensionalScene_controls_enableRotate,
        behaviourParams_item_threeDimensionalScene_controls_enableZoom: ((_f = scene.controls) === null || _f === void 0 ? void 0 : _f.enableZoom) || options.behaviourParams_item_threeDimensionalScene_controls_enableZoom,
        behaviourParams_item_threeDimensionalScene_controls_enableAutoRotate: ((_g = scene.controls) === null || _g === void 0 ? void 0 : _g.enableAutoRotate) || options.behaviourParams_item_threeDimensionalScene_controls_enableAutoRotate,
    };
}
export function use3DItem(props) {
    const canvasRef = useRef(null);
    const { sceneManager, render } = useSceneManager();
    const [isLoaded, setIsLoaded] = useState(false);
    const sceneParams = mapSceneToStyleParams(props.scene || {}, props.options);
    const loadedManager = useMemo(() => {
        if (!props.shouldPlay) {
            return sceneManager;
        }
        if (sceneManager) {
            return sceneManager;
        }
        if (!props.itemContainer.current || !canvasRef.current) {
            return sceneManager;
        }
        const manager = render(props.itemContainer.current, canvasRef.current);
        if (!manager) {
            throw new Error('Could not create scene manager post play');
        }
        manager.opacity = 0;
        manager.model
            .load3DModel(props.createUrl(GALLERY_CONSTS.urlSizes.RESIZED, GALLERY_CONSTS.urlTypes.THREE_D))
            .then(({ punctualLights }) => {
            props.actions.setItemLoaded();
            manager.opacity = 1;
            manager.model.addGround();
            if (punctualLights) {
                manager.environment.addAmbientLight();
                manager.environment.sun();
            }
            requestAnimationFrame(() => {
                setIsLoaded(true);
            });
        });
        return manager;
    }, [render, sceneManager, props.shouldPlay, props.itemContainer.current, canvasRef.current]);
    const postLoadUpdate = useCallback(({ behaviourParams_item_threeDimensionalScene_transform_rotation, behaviourParams_item_threeDimensionalScene_transform_scale, behaviourParams_item_threeDimensionalScene_transform_position, behaviourParams_item_threeDimensionalScene_controls_enablePan, behaviourParams_item_threeDimensionalScene_controls_enableRotate, behaviourParams_item_threeDimensionalScene_controls_enableZoom, behaviourParams_item_threeDimensionalScene_controls_enableAutoRotate, }) => {
        if (!loadedManager) {
            return;
        }
        const rotation = GALLERY_CONSTS.parse3DDimensions(behaviourParams_item_threeDimensionalScene_transform_rotation);
        const scale = GALLERY_CONSTS.parse3DDimensions(behaviourParams_item_threeDimensionalScene_transform_scale);
        const position = GALLERY_CONSTS.parse3DDimensions(behaviourParams_item_threeDimensionalScene_transform_position);
        loadedManager.model.transform.setPosition(position.x, position.y, position.z);
        loadedManager.model.transform.setRotation(rotation.x, rotation.y, rotation.z);
        loadedManager.camera.enablePan = behaviourParams_item_threeDimensionalScene_controls_enablePan;
        loadedManager.camera.enableRotate = behaviourParams_item_threeDimensionalScene_controls_enableRotate;
        loadedManager.camera.enableZoom = behaviourParams_item_threeDimensionalScene_controls_enableZoom;
        loadedManager.camera.enableAutoRotate = behaviourParams_item_threeDimensionalScene_controls_enableAutoRotate;
        loadedManager.model.transform.setScale(scale.x, scale.y, scale.z);
    }, [loadedManager]);
    useEffect(() => {
        if (isLoaded && loadedManager && props.shouldPlay === loadedManager.stop) {
            loadedManager.stop = !props.shouldPlay;
            return;
        }
    }, [props.shouldPlay, loadedManager, isLoaded]);
    postLoadUpdate(sceneParams);
    useEffect(() => {
        return () => {
            if (loadedManager) {
                loadedManager.dispose();
            }
        };
    }, [loadedManager]);
    return {
        canvasRef,
        isLoaded,
    };
}
