// @wix/auto_sdk_restaurants_order-settings/build/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  ApproximateAddresses: () => ApproximateAddresses,
  ChargeRoundingStrategy: () => ChargeRoundingStrategy,
  DayOfWeek: () => DayOfWeek,
  TipsType: () => TipsType,
  Type: () => Type,
  getOrderSettings: () => getOrderSettings2,
  listOrderSettings: () => listOrderSettings2,
  updateOrderSettings: () => updateOrderSettings2
});
module.exports = __toCommonJS(index_typings_exports);

// src/restaurants-ordersettings-v3-order-settings-order-settings.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/restaurants-ordersettings-v3-order-settings-order-settings.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressRestaurantsOrdersettingsV3OrderSettingsApiUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/v3/order-settings",
        destPath: "/v3/order-settings"
      }
    ],
    _: [
      {
        srcPath: "/_api/order-settings",
        destPath: "/v3/order-settings"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/order-settings",
        destPath: "/v3/order-settings"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/order-settings",
        destPath: "/v3/order-settings"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants/v3/order-settings",
        destPath: "/v3/order-settings"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_order-settings";
function getOrderSettings(payload) {
  function __getOrderSettings({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.ordersettings.v3.order_settings",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.ordersettings.v3.OrderSettingsApi.GetOrderSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsOrdersettingsV3OrderSettingsApiUrl({
        protoPath: "/v3/order-settings/{orderSettingsId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getOrderSettings;
}
function updateOrderSettings(payload) {
  function __updateOrderSettings({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.ordersettings.v3.order_settings",
      method: "PATCH",
      methodFqn: "com.wixpress.restaurants.ordersettings.v3.OrderSettingsApi.UpdateOrderSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsOrdersettingsV3OrderSettingsApiUrl({
        protoPath: "/v3/order-settings/{orderSettings.id}",
        data: serializedData,
        host
      }),
      data: serializedData
    };
    return metadata;
  }
  return __updateOrderSettings;
}
function listOrderSettings(payload) {
  function __listOrderSettings({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.ordersettings.v3.order_settings",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.ordersettings.v3.OrderSettingsApi.ListOrderSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsOrdersettingsV3OrderSettingsApiUrl({
        protoPath: "/v3/order-settings",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __listOrderSettings;
}

// src/restaurants-ordersettings-v3-order-settings-order-settings.universal.ts
var ApproximateAddresses = /* @__PURE__ */ ((ApproximateAddresses2) => {
  ApproximateAddresses2["EMPTY"] = "EMPTY";
  ApproximateAddresses2["BLOCK"] = "BLOCK";
  ApproximateAddresses2["ALLOW"] = "ALLOW";
  return ApproximateAddresses2;
})(ApproximateAddresses || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["URL"] = "URL";
  Type2["TEXT"] = "TEXT";
  return Type2;
})(Type || {});
var TipsType = /* @__PURE__ */ ((TipsType2) => {
  TipsType2["AMOUNT"] = "AMOUNT";
  TipsType2["PERCENTAGE"] = "PERCENTAGE";
  return TipsType2;
})(TipsType || {});
var DayOfWeek = /* @__PURE__ */ ((DayOfWeek2) => {
  DayOfWeek2["MON"] = "MON";
  DayOfWeek2["TUE"] = "TUE";
  DayOfWeek2["WED"] = "WED";
  DayOfWeek2["THU"] = "THU";
  DayOfWeek2["FRI"] = "FRI";
  DayOfWeek2["SAT"] = "SAT";
  DayOfWeek2["SUN"] = "SUN";
  return DayOfWeek2;
})(DayOfWeek || {});
var ChargeRoundingStrategy = /* @__PURE__ */ ((ChargeRoundingStrategy2) => {
  ChargeRoundingStrategy2["UNDEFINED"] = "UNDEFINED";
  ChargeRoundingStrategy2["HALF_AWAY_FROM_ZERO"] = "HALF_AWAY_FROM_ZERO";
  ChargeRoundingStrategy2["HALF_EVEN"] = "HALF_EVEN";
  return ChargeRoundingStrategy2;
})(ChargeRoundingStrategy || {});
async function getOrderSettings2(orderSettingsId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    orderSettingsId
  });
  const reqOpts = getOrderSettings(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.orderSettings;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { orderSettingsId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["orderSettingsId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateOrderSettings2(_id, orderSettings) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    orderSettings: { ...orderSettings, id: _id }
  });
  const reqOpts = updateOrderSettings(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.orderSettings;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { orderSettings: "$[1]" },
        explicitPathsToArguments: { "orderSettings.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "orderSettings"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listOrderSettings2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    restaurantLocationIds: options?.restaurantLocationIds
  });
  const reqOpts = listOrderSettings(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          restaurantLocationIds: "$[0].restaurantLocationIds"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ApproximateAddresses,
  ChargeRoundingStrategy,
  DayOfWeek,
  TipsType,
  Type,
  getOrderSettings,
  listOrderSettings,
  updateOrderSettings
});
