// @wix/auto_sdk_restaurants_fulfillment-methods/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  DayOfWeekOriginal: () => DayOfWeek,
  EntitiesDayOfWeekOriginal: () => EntitiesDayOfWeek,
  FulfillmentMethodTypeOriginal: () => FulfillmentMethodType,
  PlacementTypeOriginal: () => PlacementType,
  ResolutionMethodOriginal: () => ResolutionMethod,
  SortOrderOriginal: () => SortOrder,
  SubdivisionTypeOriginal: () => SubdivisionType,
  TypeOriginal: () => Type,
  UnitOriginal: () => Unit,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  bulkCreateFulfillmentMethods: () => bulkCreateFulfillmentMethods2,
  bulkUpdateFulfillmentMethodTags: () => bulkUpdateFulfillmentMethodTags2,
  bulkUpdateFulfillmentMethodTagsByFilter: () => bulkUpdateFulfillmentMethodTagsByFilter2,
  createFulfillmentMethod: () => createFulfillmentMethod2,
  deleteFulfillmentMethod: () => deleteFulfillmentMethod2,
  getAccumulatedFulfillmentMethodsAvailability: () => getAccumulatedFulfillmentMethodsAvailability2,
  getAggregatedMethodAvailability: () => getAggregatedMethodAvailability2,
  getCombinedMethodAvailability: () => getCombinedMethodAvailability2,
  getFulfillmentMethod: () => getFulfillmentMethod2,
  listAvailableFulfillmentMethodsForAddress: () => listAvailableFulfillmentMethodsForAddress2,
  listFulfillmentMethods: () => listFulfillmentMethods2,
  queryFulfillmentMethods: () => queryFulfillmentMethods2,
  updateFulfillmentMethod: () => updateFulfillmentMethod2
});
module.exports = __toCommonJS(meta_exports);

// src/restaurants-v1-fulfillment-method-fulfillment-methods.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixRestaurantsFulfillmentV1FulfillmentMethodsServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/fulfillment-methods",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/fulfillment-methods/v1",
        destPath: "/v1"
      }
    ],
    _: [
      {
        srcPath: "/_api/fulfillment-methods/v1",
        destPath: "/v1"
      }
    ],
    _base_domain_: [
      {
        srcPath: "/_api/fulfillment-methods/v1",
        destPath: "/v1"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/fulfillment-methods/v1",
        destPath: "/v1"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/fulfillment-methods/v1",
        destPath: "/v1"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/fulfillment-methods/v1",
        destPath: "/v1"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/fulfillment-methods/v1",
        destPath: "/v1"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/fulfillment-methods/v1",
        destPath: "/v1"
      }
    ],
    "users._base_domain_": [
      {
        srcPath: "/_api/fulfillment-methods/v1",
        destPath: "/v1"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/fulfillment-methods/v1",
        destPath: "/v1"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/fulfillment-methods/v1",
        destPath: "/v1"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/fulfillment-methods",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_fulfillment-methods";
function createFulfillmentMethod(payload) {
  function __createFulfillmentMethod({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "fulfillmentMethod.createdDate" },
          { path: "fulfillmentMethod.updatedDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "fulfillmentMethod.pickupOptions.address.geocode.latitude" },
          { path: "fulfillmentMethod.pickupOptions.address.geocode.longitude" },
          {
            path: "fulfillmentMethod.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.latitude"
          },
          {
            path: "fulfillmentMethod.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.longitude"
          },
          {
            path: "fulfillmentMethod.deliveryOptions.deliveryArea.customOptions.geocodes.latitude"
          },
          {
            path: "fulfillmentMethod.deliveryOptions.deliveryArea.customOptions.geocodes.longitude"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.fulfillment_method",
      method: "POST",
      methodFqn: "wix.restaurants.fulfillment.v1.FulfillmentMethodsService.CreateFulfillmentMethod",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsFulfillmentV1FulfillmentMethodsServiceUrl({
        protoPath: "/v1/fulfillment-methods",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "fulfillmentMethod.createdDate" },
            { path: "fulfillmentMethod.updatedDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "fulfillmentMethod.pickupOptions.address.geocode.latitude"
            },
            {
              path: "fulfillmentMethod.pickupOptions.address.geocode.longitude"
            },
            {
              path: "fulfillmentMethod.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.latitude"
            },
            {
              path: "fulfillmentMethod.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.longitude"
            },
            {
              path: "fulfillmentMethod.deliveryOptions.deliveryArea.customOptions.geocodes.latitude"
            },
            {
              path: "fulfillmentMethod.deliveryOptions.deliveryArea.customOptions.geocodes.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createFulfillmentMethod;
}
function bulkCreateFulfillmentMethods(payload) {
  function __bulkCreateFulfillmentMethods({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "fulfillmentMethods.createdDate" },
          { path: "fulfillmentMethods.updatedDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "fulfillmentMethods.pickupOptions.address.geocode.latitude" },
          {
            path: "fulfillmentMethods.pickupOptions.address.geocode.longitude"
          },
          {
            path: "fulfillmentMethods.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.latitude"
          },
          {
            path: "fulfillmentMethods.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.longitude"
          },
          {
            path: "fulfillmentMethods.deliveryOptions.deliveryArea.customOptions.geocodes.latitude"
          },
          {
            path: "fulfillmentMethods.deliveryOptions.deliveryArea.customOptions.geocodes.longitude"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.fulfillment_method",
      method: "POST",
      methodFqn: "wix.restaurants.fulfillment.v1.FulfillmentMethodsService.BulkCreateFulfillmentMethods",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsFulfillmentV1FulfillmentMethodsServiceUrl({
        protoPath: "/v1/bulk/fulfillment-methods/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.fulfillmentMethod.createdDate" },
            { path: "results.fulfillmentMethod.updatedDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "results.fulfillmentMethod.pickupOptions.address.geocode.latitude"
            },
            {
              path: "results.fulfillmentMethod.pickupOptions.address.geocode.longitude"
            },
            {
              path: "results.fulfillmentMethod.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.latitude"
            },
            {
              path: "results.fulfillmentMethod.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.longitude"
            },
            {
              path: "results.fulfillmentMethod.deliveryOptions.deliveryArea.customOptions.geocodes.latitude"
            },
            {
              path: "results.fulfillmentMethod.deliveryOptions.deliveryArea.customOptions.geocodes.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateFulfillmentMethods;
}
function getFulfillmentMethod(payload) {
  function __getFulfillmentMethod({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.fulfillment_method",
      method: "GET",
      methodFqn: "wix.restaurants.fulfillment.v1.FulfillmentMethodsService.GetFulfillmentMethod",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsFulfillmentV1FulfillmentMethodsServiceUrl({
        protoPath: "/v1/fulfillment-methods/{fulfillmentMethodId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "fulfillmentMethod.createdDate" },
            { path: "fulfillmentMethod.updatedDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "fulfillmentMethod.pickupOptions.address.geocode.latitude"
            },
            {
              path: "fulfillmentMethod.pickupOptions.address.geocode.longitude"
            },
            {
              path: "fulfillmentMethod.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.latitude"
            },
            {
              path: "fulfillmentMethod.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.longitude"
            },
            {
              path: "fulfillmentMethod.deliveryOptions.deliveryArea.customOptions.geocodes.latitude"
            },
            {
              path: "fulfillmentMethod.deliveryOptions.deliveryArea.customOptions.geocodes.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getFulfillmentMethod;
}
function updateFulfillmentMethod(payload) {
  function __updateFulfillmentMethod({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "fulfillmentMethod.createdDate" },
          { path: "fulfillmentMethod.updatedDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "fulfillmentMethod.pickupOptions.address.geocode.latitude" },
          { path: "fulfillmentMethod.pickupOptions.address.geocode.longitude" },
          {
            path: "fulfillmentMethod.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.latitude"
          },
          {
            path: "fulfillmentMethod.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.longitude"
          },
          {
            path: "fulfillmentMethod.deliveryOptions.deliveryArea.customOptions.geocodes.latitude"
          },
          {
            path: "fulfillmentMethod.deliveryOptions.deliveryArea.customOptions.geocodes.longitude"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.fulfillment_method",
      method: "PATCH",
      methodFqn: "wix.restaurants.fulfillment.v1.FulfillmentMethodsService.UpdateFulfillmentMethod",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsFulfillmentV1FulfillmentMethodsServiceUrl({
        protoPath: "/v1/fulfillment-methods/{fulfillmentMethod.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "fulfillmentMethod.createdDate" },
            { path: "fulfillmentMethod.updatedDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "fulfillmentMethod.pickupOptions.address.geocode.latitude"
            },
            {
              path: "fulfillmentMethod.pickupOptions.address.geocode.longitude"
            },
            {
              path: "fulfillmentMethod.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.latitude"
            },
            {
              path: "fulfillmentMethod.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.longitude"
            },
            {
              path: "fulfillmentMethod.deliveryOptions.deliveryArea.customOptions.geocodes.latitude"
            },
            {
              path: "fulfillmentMethod.deliveryOptions.deliveryArea.customOptions.geocodes.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateFulfillmentMethod;
}
function deleteFulfillmentMethod(payload) {
  function __deleteFulfillmentMethod({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.fulfillment_method",
      method: "DELETE",
      methodFqn: "wix.restaurants.fulfillment.v1.FulfillmentMethodsService.DeleteFulfillmentMethod",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsFulfillmentV1FulfillmentMethodsServiceUrl({
        protoPath: "/v1/fulfillment-methods/{fulfillmentMethodId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteFulfillmentMethod;
}
function queryFulfillmentMethods(payload) {
  function __queryFulfillmentMethods({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "projectionMask" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.fulfillment_method",
      method: "POST",
      methodFqn: "wix.restaurants.fulfillment.v1.FulfillmentMethodsService.QueryFulfillmentMethods",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsFulfillmentV1FulfillmentMethodsServiceUrl({
        protoPath: "/v1/fulfillment-methods/query",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "fulfillmentMethods.createdDate" },
            { path: "fulfillmentMethods.updatedDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "fulfillmentMethods.pickupOptions.address.geocode.latitude"
            },
            {
              path: "fulfillmentMethods.pickupOptions.address.geocode.longitude"
            },
            {
              path: "fulfillmentMethods.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.latitude"
            },
            {
              path: "fulfillmentMethods.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.longitude"
            },
            {
              path: "fulfillmentMethods.deliveryOptions.deliveryArea.customOptions.geocodes.latitude"
            },
            {
              path: "fulfillmentMethods.deliveryOptions.deliveryArea.customOptions.geocodes.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryFulfillmentMethods;
}
function listFulfillmentMethods(payload) {
  function __listFulfillmentMethods({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "address.geocode.latitude" },
          { path: "address.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.fulfillment_method",
      method: "GET",
      methodFqn: "wix.restaurants.fulfillment.v1.FulfillmentMethodsService.ListFulfillmentMethods",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsFulfillmentV1FulfillmentMethodsServiceUrl({
        protoPath: "/v1/fulfillment-methods",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "fulfillmentMethods.createdDate" },
            { path: "fulfillmentMethods.updatedDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "fulfillmentMethods.pickupOptions.address.geocode.latitude"
            },
            {
              path: "fulfillmentMethods.pickupOptions.address.geocode.longitude"
            },
            {
              path: "fulfillmentMethods.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.latitude"
            },
            {
              path: "fulfillmentMethods.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.longitude"
            },
            {
              path: "fulfillmentMethods.deliveryOptions.deliveryArea.customOptions.geocodes.latitude"
            },
            {
              path: "fulfillmentMethods.deliveryOptions.deliveryArea.customOptions.geocodes.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listFulfillmentMethods;
}
function listAvailableFulfillmentMethodsForAddress(payload) {
  function __listAvailableFulfillmentMethodsForAddress({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "address.geocode.latitude" },
          { path: "address.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.fulfillment_method",
      method: "POST",
      methodFqn: "wix.restaurants.fulfillment.v1.FulfillmentMethodsService.ListAvailableFulfillmentMethodsForAddress",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsFulfillmentV1FulfillmentMethodsServiceUrl({
        protoPath: "/v1/fulfillment-methods/available-for-address",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "fulfillmentMethods.createdDate" },
            { path: "fulfillmentMethods.updatedDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "fulfillmentMethods.pickupOptions.address.geocode.latitude"
            },
            {
              path: "fulfillmentMethods.pickupOptions.address.geocode.longitude"
            },
            {
              path: "fulfillmentMethods.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.latitude"
            },
            {
              path: "fulfillmentMethods.deliveryOptions.deliveryArea.radiusOptions.centerPointAddress.geocode.longitude"
            },
            {
              path: "fulfillmentMethods.deliveryOptions.deliveryArea.customOptions.geocodes.latitude"
            },
            {
              path: "fulfillmentMethods.deliveryOptions.deliveryArea.customOptions.geocodes.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listAvailableFulfillmentMethodsForAddress;
}
function getAccumulatedFulfillmentMethodsAvailability(payload) {
  function __getAccumulatedFulfillmentMethodsAvailability({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.fulfillment_method",
      method: "GET",
      methodFqn: "wix.restaurants.fulfillment.v1.FulfillmentMethodsService.GetAccumulatedFulfillmentMethodsAvailability",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsFulfillmentV1FulfillmentMethodsServiceUrl({
        protoPath: "/v1/fulfillment-methods/accumulated-availability",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getAccumulatedFulfillmentMethodsAvailability;
}
function getCombinedMethodAvailability(payload) {
  function __getCombinedMethodAvailability({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.fulfillment_method",
      method: "GET",
      methodFqn: "wix.restaurants.fulfillment.v1.FulfillmentMethodsService.GetCombinedMethodAvailability",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsFulfillmentV1FulfillmentMethodsServiceUrl({
        protoPath: "/v1/fulfillment-methods/combined-availability",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getCombinedMethodAvailability;
}
function getAggregatedMethodAvailability(payload) {
  function __getAggregatedMethodAvailability({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.fulfillment_method",
      method: "POST",
      methodFqn: "wix.restaurants.fulfillment.v1.FulfillmentMethodsService.GetAggregatedMethodAvailability",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsFulfillmentV1FulfillmentMethodsServiceUrl({
        protoPath: "/v1/fulfillment-methods/aggregated-availability",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __getAggregatedMethodAvailability;
}
function bulkUpdateFulfillmentMethodTags(payload) {
  function __bulkUpdateFulfillmentMethodTags({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.fulfillment_method",
      method: "POST",
      methodFqn: "wix.restaurants.fulfillment.v1.FulfillmentMethodsService.BulkUpdateFulfillmentMethodTags",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsFulfillmentV1FulfillmentMethodsServiceUrl({
        protoPath: "/v1/bulk/fulfillment-methods/update-tags",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateFulfillmentMethodTags;
}
function bulkUpdateFulfillmentMethodTagsByFilter(payload) {
  function __bulkUpdateFulfillmentMethodTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.fulfillment_method",
      method: "POST",
      methodFqn: "wix.restaurants.fulfillment.v1.FulfillmentMethodsService.BulkUpdateFulfillmentMethodTagsByFilter",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsFulfillmentV1FulfillmentMethodsServiceUrl({
        protoPath: "/v1/bulk/fulfillment-methods/update-tags-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateFulfillmentMethodTagsByFilter;
}

// src/restaurants-v1-fulfillment-method-fulfillment-methods.types.ts
var FulfillmentMethodType = /* @__PURE__ */ ((FulfillmentMethodType2) => {
  FulfillmentMethodType2["UNKNOWN_FULFILLMENT_TYPE"] = "UNKNOWN_FULFILLMENT_TYPE";
  FulfillmentMethodType2["PICKUP"] = "PICKUP";
  FulfillmentMethodType2["DELIVERY"] = "DELIVERY";
  return FulfillmentMethodType2;
})(FulfillmentMethodType || {});
var SubdivisionType = /* @__PURE__ */ ((SubdivisionType2) => {
  SubdivisionType2["UNKNOWN_SUBDIVISION_TYPE"] = "UNKNOWN_SUBDIVISION_TYPE";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_1"] = "ADMINISTRATIVE_AREA_LEVEL_1";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_2"] = "ADMINISTRATIVE_AREA_LEVEL_2";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_3"] = "ADMINISTRATIVE_AREA_LEVEL_3";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_4"] = "ADMINISTRATIVE_AREA_LEVEL_4";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_5"] = "ADMINISTRATIVE_AREA_LEVEL_5";
  SubdivisionType2["COUNTRY"] = "COUNTRY";
  return SubdivisionType2;
})(SubdivisionType || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNKNOWN_DELIVERY_AREA"] = "UNKNOWN_DELIVERY_AREA";
  Type2["RADIUS"] = "RADIUS";
  Type2["POSTAL_CODE"] = "POSTAL_CODE";
  Type2["CUSTOM"] = "CUSTOM";
  Type2["PROVIDER_DEFINED"] = "PROVIDER_DEFINED";
  return Type2;
})(Type || {});
var Unit = /* @__PURE__ */ ((Unit2) => {
  Unit2["UNKNOWN_UNIT"] = "UNKNOWN_UNIT";
  Unit2["MILES"] = "MILES";
  Unit2["KILOMETERS"] = "KILOMETERS";
  return Unit2;
})(Unit || {});
var EntitiesDayOfWeek = /* @__PURE__ */ ((EntitiesDayOfWeek2) => {
  EntitiesDayOfWeek2["MON"] = "MON";
  EntitiesDayOfWeek2["TUE"] = "TUE";
  EntitiesDayOfWeek2["WED"] = "WED";
  EntitiesDayOfWeek2["THU"] = "THU";
  EntitiesDayOfWeek2["FRI"] = "FRI";
  EntitiesDayOfWeek2["SAT"] = "SAT";
  EntitiesDayOfWeek2["SUN"] = "SUN";
  return EntitiesDayOfWeek2;
})(EntitiesDayOfWeek || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var PlacementType = /* @__PURE__ */ ((PlacementType2) => {
  PlacementType2["BEFORE"] = "BEFORE";
  PlacementType2["AFTER"] = "AFTER";
  PlacementType2["REPLACE"] = "REPLACE";
  return PlacementType2;
})(PlacementType || {});
var DayOfWeek = /* @__PURE__ */ ((DayOfWeek2) => {
  DayOfWeek2["MONDAY"] = "MONDAY";
  DayOfWeek2["TUESDAY"] = "TUESDAY";
  DayOfWeek2["WEDNESDAY"] = "WEDNESDAY";
  DayOfWeek2["THURSDAY"] = "THURSDAY";
  DayOfWeek2["FRIDAY"] = "FRIDAY";
  DayOfWeek2["SATURDAY"] = "SATURDAY";
  DayOfWeek2["SUNDAY"] = "SUNDAY";
  return DayOfWeek2;
})(DayOfWeek || {});
var ResolutionMethod = /* @__PURE__ */ ((ResolutionMethod2) => {
  ResolutionMethod2["QUERY_PARAM"] = "QUERY_PARAM";
  ResolutionMethod2["SUBDOMAIN"] = "SUBDOMAIN";
  ResolutionMethod2["SUBDIRECTORY"] = "SUBDIRECTORY";
  return ResolutionMethod2;
})(ResolutionMethod || {});

// src/restaurants-v1-fulfillment-method-fulfillment-methods.meta.ts
function createFulfillmentMethod2() {
  const payload = {};
  const getRequestOptions = createFulfillmentMethod(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/fulfillment-methods",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkCreateFulfillmentMethods2() {
  const payload = {};
  const getRequestOptions = bulkCreateFulfillmentMethods(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/fulfillment-methods/create",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getFulfillmentMethod2() {
  const payload = { fulfillmentMethodId: ":fulfillmentMethodId" };
  const getRequestOptions = getFulfillmentMethod(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/fulfillment-methods/{fulfillmentMethodId}",
    pathParams: { fulfillmentMethodId: "fulfillmentMethodId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateFulfillmentMethod2() {
  const payload = { fulfillmentMethod: { id: ":fulfillmentMethodId" } };
  const getRequestOptions = updateFulfillmentMethod(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/fulfillment-methods/{fulfillmentMethod.id}",
    pathParams: { fulfillmentMethodId: "fulfillmentMethodId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteFulfillmentMethod2() {
  const payload = { fulfillmentMethodId: ":fulfillmentMethodId" };
  const getRequestOptions = deleteFulfillmentMethod(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/fulfillment-methods/{fulfillmentMethodId}",
    pathParams: { fulfillmentMethodId: "fulfillmentMethodId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryFulfillmentMethods2() {
  const payload = {};
  const getRequestOptions = queryFulfillmentMethods(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/fulfillment-methods/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listFulfillmentMethods2() {
  const payload = {};
  const getRequestOptions = listFulfillmentMethods(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/fulfillment-methods",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listAvailableFulfillmentMethodsForAddress2() {
  const payload = {};
  const getRequestOptions = listAvailableFulfillmentMethodsForAddress(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/fulfillment-methods/available-for-address",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getAccumulatedFulfillmentMethodsAvailability2() {
  const payload = {};
  const getRequestOptions = getAccumulatedFulfillmentMethodsAvailability(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/fulfillment-methods/accumulated-availability",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getCombinedMethodAvailability2() {
  const payload = {};
  const getRequestOptions = getCombinedMethodAvailability(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/fulfillment-methods/combined-availability",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getAggregatedMethodAvailability2() {
  const payload = {};
  const getRequestOptions = getAggregatedMethodAvailability(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/fulfillment-methods/aggregated-availability",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateFulfillmentMethodTags2() {
  const payload = {};
  const getRequestOptions = bulkUpdateFulfillmentMethodTags(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/fulfillment-methods/update-tags",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateFulfillmentMethodTagsByFilter2() {
  const payload = {};
  const getRequestOptions = bulkUpdateFulfillmentMethodTagsByFilter(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/fulfillment-methods/update-tags-by-filter",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  DayOfWeekOriginal,
  EntitiesDayOfWeekOriginal,
  FulfillmentMethodTypeOriginal,
  PlacementTypeOriginal,
  ResolutionMethodOriginal,
  SortOrderOriginal,
  SubdivisionTypeOriginal,
  TypeOriginal,
  UnitOriginal,
  WebhookIdentityTypeOriginal,
  bulkCreateFulfillmentMethods,
  bulkUpdateFulfillmentMethodTags,
  bulkUpdateFulfillmentMethodTagsByFilter,
  createFulfillmentMethod,
  deleteFulfillmentMethod,
  getAccumulatedFulfillmentMethodsAvailability,
  getAggregatedMethodAvailability,
  getCombinedMethodAvailability,
  getFulfillmentMethod,
  listAvailableFulfillmentMethodsForAddress,
  listFulfillmentMethods,
  queryFulfillmentMethods,
  updateFulfillmentMethod
});
