// @wix/auto_sdk_seo_robots-txt/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  getRobotsTxt: () => getRobotsTxt2,
  updateRobotsTxt: () => updateRobotsTxt2
});
module.exports = __toCommonJS(meta_exports);

// src/promote-seo-v2-robots-txt-robots-txt.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressPromoteSeoRobotsServerV2RobotsServiceV2Url(opts) {
  const domainToMappings = {
    "www._base_domain_": [
      {
        srcPath: "/_api/promote-seo-robots-server",
        destPath: "/api"
      },
      {
        srcPath: "/_api/promote-seo-robots-server/v2/robots",
        destPath: "/v2/robots"
      },
      {
        srcPath: "/_api/promote-seo-robots-server/v2/ads",
        destPath: "/v2/ads"
      },
      {
        srcPath: "/_api/promote-seo-robots-server/v2/llms",
        destPath: "/v2/llms"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/promote-seo-robots-server",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/promote-seo-robots-server",
        destPath: "/api"
      },
      {
        srcPath: "/_api/promote-seo-robots-server/v2/robots",
        destPath: "/v2/robots"
      },
      {
        srcPath: "/_api/promote-seo-robots-server/v2/ads",
        destPath: "/v2/ads"
      },
      {
        srcPath: "/_api/promote-seo-robots-server/v2/llms",
        destPath: "/v2/llms"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/promote-seo-robots-server/v2/robots",
        destPath: "/v2/robots"
      },
      {
        srcPath: "/promote-seo-robots-server/v2/ads",
        destPath: "/v2/ads"
      },
      {
        srcPath: "/promote-seo-txt-file-server/v2/ads",
        destPath: "/v2/ads"
      },
      {
        srcPath: "/promote-seo-txt-file-server/v2/robots",
        destPath: "/v2/robots"
      },
      {
        srcPath: "/promote-seo-robots-server/v2/llms",
        destPath: "/v2/llms"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/promote/marketing/v2/ads",
        destPath: "/v2/ads"
      },
      {
        srcPath: "/promote/seo/v2/llms",
        destPath: "/v2/llms"
      },
      {
        srcPath: "/promote/seo/v2/robots",
        destPath: "/v2/robots"
      }
    ],
    _: [
      {
        srcPath: "/promote/marketing/v2/ads",
        destPath: "/v2/ads"
      },
      {
        srcPath: "/promote/seo/v2/llms",
        destPath: "/v2/llms"
      },
      {
        srcPath: "/promote/seo/v2/robots",
        destPath: "/v2/robots"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_seo_robots-txt";
function getRobotsTxt(payload) {
  function __getRobotsTxt({ host }) {
    const metadata = {
      entityFqdn: "wix.promote.seo.v2.robots_txt",
      method: "GET",
      methodFqn: "com.wixpress.promote.seo.robots.server.v2.RobotsServiceV2.GetRobotsTxt",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressPromoteSeoRobotsServerV2RobotsServiceV2Url({
        protoPath: "/v2/robots",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getRobotsTxt;
}
function updateRobotsTxt(payload) {
  function __updateRobotsTxt({ host }) {
    const metadata = {
      entityFqdn: "wix.promote.seo.v2.robots_txt",
      method: "PUT",
      methodFqn: "com.wixpress.promote.seo.robots.server.v2.RobotsServiceV2.UpdateRobotsTxt",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressPromoteSeoRobotsServerV2RobotsServiceV2Url({
        protoPath: "/v2/robots",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __updateRobotsTxt;
}

// src/promote-seo-v2-robots-txt-robots-txt.meta.ts
function getRobotsTxt2() {
  const payload = {};
  const getRequestOptions = getRobotsTxt(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/robots",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateRobotsTxt2() {
  const payload = {};
  const getRequestOptions = updateRobotsTxt(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PUT",
    path: "/v2/robots",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getRobotsTxt,
  updateRobotsTxt
});
