// @wix/auto_sdk_restaurants_item-modifiers/build/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  SortOrder: () => SortOrder,
  WebhookIdentityType: () => WebhookIdentityType,
  bulkCreateModifiers: () => bulkCreateModifiers2,
  bulkDeleteModifiers: () => bulkDeleteModifiers2,
  bulkUpdateModifiers: () => bulkUpdateModifiers2,
  countModifiers: () => countModifiers2,
  createModifier: () => createModifier2,
  deleteModifier: () => deleteModifier2,
  getModifier: () => getModifier2,
  listModifiers: () => listModifiers2,
  queryModifiers: () => queryModifiers2,
  updateModifier: () => updateModifier2
});
module.exports = __toCommonJS(index_typings_exports);

// src/restaurants-menus-v1-item-modifier-item-modifiers.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/restaurants-menus-v1-item-modifier-item-modifiers.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixRestaurantsMenusItemModifierV1ModifiersServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-modifier",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-modifier",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-modifier",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-menus-item-modifier",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-menus-item-modifier",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-menus-item-modifier",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers/delete",
        destPath: "/v1/bulk/modifiers/delete"
      },
      {
        srcPath: "/restaurants/menus/v1/modifiers/count",
        destPath: "/v1/modifiers/count"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers/create",
        destPath: "/v1/bulk/modifiers/create"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-menus-item-modifier",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers/delete",
        destPath: "/v1/bulk/modifiers/delete"
      },
      {
        srcPath: "/restaurants/menus/v1/modifiers/count",
        destPath: "/v1/modifiers/count"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers/create",
        destPath: "/v1/bulk/modifiers/create"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-modifier",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-menus-item-modifier",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/item-modifiers",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers/create",
        destPath: "/v1/bulk/modifiers/create"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers/update",
        destPath: "/v1/bulk/modifiers/update"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers/delete",
        destPath: "/v1/bulk/modifiers/delete"
      },
      {
        srcPath: "/restaurants/menus/v1/modifiers/count",
        destPath: "/v1/modifiers/count"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-modifier",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_item-modifiers";
function createModifier(payload) {
  function __createModifier({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "modifier.createdDate" },
          { path: "modifier.updatedDate" },
          { path: "modifier.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier",
      method: "POST",
      methodFqn: "wix.restaurants.menus_item_modifier.v1.ModifiersService.CreateModifier",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemModifierV1ModifiersServiceUrl({
        protoPath: "/v1/modifiers",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "modifier.createdDate" },
            { path: "modifier.updatedDate" },
            { path: "modifier.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createModifier;
}
function getModifier(payload) {
  function __getModifier({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier",
      method: "GET",
      methodFqn: "wix.restaurants.menus_item_modifier.v1.ModifiersService.GetModifier",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemModifierV1ModifiersServiceUrl({
        protoPath: "/v1/modifiers/{modifierId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "modifier.createdDate" },
            { path: "modifier.updatedDate" },
            { path: "modifier.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getModifier;
}
function listModifiers(payload) {
  function __listModifiers({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier",
      method: "GET",
      methodFqn: "wix.restaurants.menus_item_modifier.v1.ModifiersService.ListModifiers",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemModifierV1ModifiersServiceUrl({
        protoPath: "/v1/modifiers",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "modifiers.createdDate" },
            { path: "modifiers.updatedDate" },
            { path: "modifiers.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listModifiers;
}
function updateModifier(payload) {
  function __updateModifier({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "modifier.createdDate" },
          { path: "modifier.updatedDate" },
          { path: "modifier.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier",
      method: "PATCH",
      methodFqn: "wix.restaurants.menus_item_modifier.v1.ModifiersService.UpdateModifier",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemModifierV1ModifiersServiceUrl({
        protoPath: "/v1/modifiers/{modifier.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "modifier.createdDate" },
            { path: "modifier.updatedDate" },
            { path: "modifier.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateModifier;
}
function bulkCreateModifiers(payload) {
  function __bulkCreateModifiers({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "modifiers.createdDate" },
          { path: "modifiers.updatedDate" },
          { path: "modifiers.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier",
      method: "POST",
      methodFqn: "wix.restaurants.menus_item_modifier.v1.ModifiersService.BulkCreateModifiers",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemModifierV1ModifiersServiceUrl({
        protoPath: "/v1/bulk/modifiers/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.modifier.createdDate" },
            { path: "results.modifier.updatedDate" },
            { path: "results.modifier.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateModifiers;
}
function bulkUpdateModifiers(payload) {
  function __bulkUpdateModifiers({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "modifiers.mask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "modifiers.modifier.createdDate" },
          { path: "modifiers.modifier.updatedDate" },
          { path: "modifiers.modifier.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier",
      method: "POST",
      methodFqn: "wix.restaurants.menus_item_modifier.v1.ModifiersService.BulkUpdateModifiers",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemModifierV1ModifiersServiceUrl({
        protoPath: "/v1/bulk/modifiers/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.modifier.createdDate" },
            { path: "results.modifier.updatedDate" },
            { path: "results.modifier.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateModifiers;
}
function deleteModifier(payload) {
  function __deleteModifier({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier",
      method: "DELETE",
      methodFqn: "wix.restaurants.menus_item_modifier.v1.ModifiersService.DeleteModifier",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemModifierV1ModifiersServiceUrl({
        protoPath: "/v1/modifiers/{modifierId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteModifier;
}
function queryModifiers(payload) {
  function __queryModifiers({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier",
      method: "POST",
      methodFqn: "wix.restaurants.menus_item_modifier.v1.ModifiersService.QueryModifiers",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemModifierV1ModifiersServiceUrl({
        protoPath: "/v1/modifiers/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "modifiers.createdDate" },
            { path: "modifiers.updatedDate" },
            { path: "modifiers.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryModifiers;
}
function countModifiers(payload) {
  function __countModifiers({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier",
      method: "POST",
      methodFqn: "wix.restaurants.menus_item_modifier.v1.ModifiersService.CountModifiers",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemModifierV1ModifiersServiceUrl({
        protoPath: "/v1/modifiers/count",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __countModifiers;
}
function bulkDeleteModifiers(payload) {
  function __bulkDeleteModifiers({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier",
      method: "DELETE",
      methodFqn: "wix.restaurants.menus_item_modifier.v1.ModifiersService.BulkDeleteModifiers",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemModifierV1ModifiersServiceUrl({
        protoPath: "/v1/bulk/modifiers/delete",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __bulkDeleteModifiers;
}

// src/restaurants-menus-v1-item-modifier-item-modifiers.universal.ts
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createModifier2(modifier) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ modifier });
  const reqOpts = createModifier(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.modifier;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { modifier: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["modifier"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getModifier2(modifierId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    modifierId
  });
  const reqOpts = getModifier(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.modifier;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { modifierId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["modifierId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listModifiers2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    modifierIds: options?.modifierIds,
    paging: options?.paging
  });
  const reqOpts = listModifiers(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          modifierIds: "$[0].modifierIds",
          paging: "$[0].paging"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateModifier2(_id, modifier) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    modifier: { ...modifier, id: _id }
  });
  const reqOpts = updateModifier(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.modifier;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { modifier: "$[1]" },
        explicitPathsToArguments: { "modifier.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "modifier"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCreateModifiers2(modifiers, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    modifiers,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkCreateModifiers(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          modifiers: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["modifiers", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateModifiers2(modifiers, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    modifiers,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkUpdateModifiers(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          modifiers: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["modifiers", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteModifier2(modifierId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    modifierId
  });
  const reqOpts = deleteModifier(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { modifierId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["modifierId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryModifiers2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryModifiers(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [])
      );
      return {
        items: transformedData?.modifiers,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function countModifiers2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    filter: options?.filter
  });
  const reqOpts = countModifiers(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { filter: "$[0].filter" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkDeleteModifiers2(ids) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ ids });
  const reqOpts = bulkDeleteModifiers(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { ids: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["ids"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  SortOrder,
  WebhookIdentityType,
  bulkCreateModifiers,
  bulkDeleteModifiers,
  bulkUpdateModifiers,
  countModifiers,
  createModifier,
  deleteModifier,
  getModifier,
  listModifiers,
  queryModifiers,
  updateModifier
});
