// @wix/ricos/dist/es/ricos-common-plugins/bulleted-list/bulleted-list.js
import { BulletListThin, BulletListThinSmall } from "@wix/wix-ui-icons-common";
import { BULLET_LIST_TYPE } from "./../../ricos-content/index.js";
import { Node_Type } from "./../../ricos-schema/legacy/wix/rich_content/v1/index.js";
import { createShortcut, quickActionsModalSections } from "./../../ricos-types/index.js";
import { isAlwaysVisible } from "./../../_shared/resolvers/is-always-visible.js";
import { isBulletedListInSelection } from "./../../_shared/resolvers/is-bulleted-list-in-selection.js";
import { isNodeOrCaptionSelected } from "./../../_shared/resolvers/is-node-or-caption-selected.js";
import { bulletedListConverter } from "./bulleted-list-converter.js";
import { TEXT_BUTTON_BULLETED_LIST } from "./data-hooks.js";
import { bulletedList } from "./extension.js";
const pluginBulletedList = {
  type: Node_Type.BULLETED_LIST,
  config: {},
  tiptapExtensions: [bulletedList],
  pmMarkConverters: [],
  pmNodeConverters: [bulletedListConverter],
  shortcuts: [
    createShortcut({
      name: Node_Type.BULLETED_LIST,
      description: "ShortcutsSheet_Formatting_BulletedList",
      keys: { macOs: "Meta+Shift+8", windows: "Ctrl+Shift+8" },
      command({ publicCommands }) {
        publicCommands.setBlockType(BULLET_LIST_TYPE);
        return "continue";
      },
      group: "ShortcutsSheet_Section_Formatting",
      enabled: true,
      showInShortcutSheet: true
    })
  ],
  textButtons: [
    {
      id: Node_Type.BULLETED_LIST,
      type: "toggle",
      presentation: {
        dataHook: TEXT_BUTTON_BULLETED_LIST,
        tooltip: "UnorderedListButton_Tooltip",
        icon: BulletListThin
      },
      attributes: {
        visible: isAlwaysVisible,
        active: isBulletedListInSelection,
        disabled: isNodeOrCaptionSelected
      },
      command: ({ publicCommands }) => () => {
        publicCommands.setBlockType(BULLET_LIST_TYPE);
        return true;
      }
    }
  ],
  getQuickActions: ({ pluginServices }) => [
    {
      id: "bulletedList",
      title: "QuickActionsPlugin_ActionTitle_BulletedList",
      icon: pluginServices?.context.isMobile ? BulletListThin : BulletListThinSmall,
      section: quickActionsModalSections.FormatLayout,
      tags: ["unordered", "points", "items", "dots"],
      command: ({ publicCommands }) => publicCommands.setBlockType(BULLET_LIST_TYPE)
    }
  ]
};
export {
  pluginBulletedList
};
