// @wix/ricos/dist/es/ricos-common-plugins/bulleted-list/bulleted-list-converter.js
import { Node_Type } from "./../../ricos-schema/legacy/wix/rich_content/v1/index.js";
const bulletedListDataDefaults = { indentation: 0, offset: 0 };
const bulletedListConverter = {
  toPm: {
    types: [Node_Type.BULLETED_LIST],
    // @ts-expect-error $TSFixMe
    convert: (node, visit) => {
      const dataWithDefaults = Object.entries(bulletedListDataDefaults).reduce((acc, [key, defaultValue]) => {
        acc[key] = node.bulletedListData?.[key] ?? defaultValue;
        return acc;
      }, {});
      return {
        type: Node_Type.BULLETED_LIST,
        attrs: {
          ...dataWithDefaults,
          id: node.id
        },
        content: visit(node)
      };
    }
  },
  fromPm: {
    types: [Node_Type.BULLETED_LIST],
    // @ts-expect-error $TSFixMe
    convert: (node, visit) => {
      const { attrs = {} } = node;
      const { id, ...data } = attrs;
      return {
        type: Node_Type.BULLETED_LIST,
        id,
        nodes: visit(node),
        bulletedListData: Object.keys(data).reduce((acc, key) => {
          if (data[key] !== bulletedListDataDefaults[key]) {
            acc[key] = data[key];
          }
          return acc;
        }, {})
      };
    }
  }
};
export {
  bulletedListConverter
};
