// @wix/auto_sdk_benefit-programs_transactions/build/internal/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  BalanceType: () => BalanceType,
  IdentityType: () => IdentityType,
  SortOrder: () => SortOrder,
  TransactionStatus: () => TransactionStatus,
  WebhookIdentityType: () => WebhookIdentityType,
  getTransaction: () => getTransaction2,
  queryTransactions: () => queryTransactions2
});
module.exports = __toCommonJS(index_typings_exports);

// src/benefit-programs-v1-transaction-transactions.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/benefit-programs-v1-transaction-transactions.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixBenefitProgramsV1TransactionTransactionServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/pool-transactions",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ],
    _: [
      {
        srcPath: "/_api/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_benefit-programs_transactions";
function getTransaction(payload) {
  function __getTransaction({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.transaction",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.transaction.TransactionService.GetTransaction",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1TransactionTransactionServiceUrl({
        protoPath: "/v1/transactions/{transactionId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "transaction.createdDate" },
            { path: "transaction.updatedDate" },
            { path: "transaction.details.effectiveDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getTransaction;
}
function queryTransactions(payload) {
  function __queryTransactions({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.transaction",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.transaction.TransactionService.QueryTransactions",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1TransactionTransactionServiceUrl({
        protoPath: "/v1/transactions/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "transactions.createdDate" },
            { path: "transactions.updatedDate" },
            { path: "transactions.details.effectiveDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryTransactions;
}

// src/benefit-programs-v1-transaction-transactions.universal.ts
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var BalanceType = /* @__PURE__ */ ((BalanceType2) => {
  BalanceType2["UNDEFINED"] = "UNDEFINED";
  BalanceType2["AVAILABLE"] = "AVAILABLE";
  BalanceType2["EXTERNAL"] = "EXTERNAL";
  return BalanceType2;
})(BalanceType || {});
var IdentityType = /* @__PURE__ */ ((IdentityType2) => {
  IdentityType2["UNKNOWN"] = "UNKNOWN";
  IdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  IdentityType2["MEMBER"] = "MEMBER";
  IdentityType2["WIX_USER"] = "WIX_USER";
  return IdentityType2;
})(IdentityType || {});
var TransactionStatus = /* @__PURE__ */ ((TransactionStatus2) => {
  TransactionStatus2["UNDEFINED"] = "UNDEFINED";
  TransactionStatus2["PENDING"] = "PENDING";
  TransactionStatus2["COMPLETED"] = "COMPLETED";
  TransactionStatus2["FAILED"] = "FAILED";
  return TransactionStatus2;
})(TransactionStatus || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function getTransaction2(transactionId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    transactionId
  });
  const reqOpts = getTransaction(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.transaction;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { transactionId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["transactionId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryTransactions2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryTransactions(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [])
      );
      return {
        items: transformedData?.transactions,
        pagingMetadata: transformedData?.metadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  BalanceType,
  IdentityType,
  SortOrder,
  TransactionStatus,
  WebhookIdentityType,
  getTransaction,
  queryTransactions
});
