// @wix/auto_sdk_benefit-programs_transactions/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  getTransaction: () => getTransaction2,
  queryTransactions: () => queryTransactions2
});
module.exports = __toCommonJS(meta_exports);

// src/benefit-programs-v1-transaction-transactions.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixBenefitProgramsV1TransactionTransactionServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/pool-transactions",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ],
    _: [
      {
        srcPath: "/_api/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_benefit-programs_transactions";
function getTransaction(payload) {
  function __getTransaction({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.transaction",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.transaction.TransactionService.GetTransaction",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1TransactionTransactionServiceUrl({
        protoPath: "/v1/transactions/{transactionId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "transaction.createdDate" },
            { path: "transaction.updatedDate" },
            { path: "transaction.details.effectiveDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getTransaction;
}
function queryTransactions(payload) {
  function __queryTransactions({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.transaction",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.transaction.TransactionService.QueryTransactions",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1TransactionTransactionServiceUrl({
        protoPath: "/v1/transactions/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "transactions.createdDate" },
            { path: "transactions.updatedDate" },
            { path: "transactions.details.effectiveDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryTransactions;
}

// src/benefit-programs-v1-transaction-transactions.meta.ts
function getTransaction2() {
  const payload = { transactionId: ":transactionId" };
  const getRequestOptions = getTransaction(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/transactions/{transactionId}",
    pathParams: { transactionId: "transactionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryTransactions2() {
  const payload = {};
  const getRequestOptions = queryTransactions(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/transactions/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getTransaction,
  queryTransactions
});
