// @wix/payments/build/cjs/meta.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  disputeEvidenceDocuments: () => disputeEvidenceDocuments,
  disputeHistoryRecords: () => disputeHistoryRecords,
  disputes: () => disputes,
  onboardingAvailability: () => onboardingAvailability,
  payinsConfigurations: () => payinsConfigurations,
  paymentAcceptanceConfigurations: () => paymentAcceptanceConfigurations,
  pspCallbacks: () => pspCallbacks,
  refunds: () => refunds,
  savedPaymentMethods: () => savedPaymentMethods,
  sitePaymentMethodTypes: () => sitePaymentMethodTypes,
  wixPaymentsAccount: () => wixPaymentsAccount
});
module.exports = __toCommonJS(meta_exports);
var onboardingAvailability = __toESM(require("@wix/auto_sdk_payments_onboarding-availability/meta"));
var disputeEvidenceDocuments = __toESM(require("@wix/auto_sdk_payments_dispute-evidence-documents/meta"));
var disputeHistoryRecords = __toESM(require("@wix/auto_sdk_payments_dispute-history-records/meta"));
var disputes = __toESM(require("@wix/auto_sdk_payments_disputes/meta"));
var wixPaymentsAccount = __toESM(require("@wix/auto_sdk_payments_wix-payments-account/meta"));
var payinsConfigurations = __toESM(require("@wix/auto_sdk_payments_payins-configurations/meta"));
var paymentAcceptanceConfigurations = __toESM(require("@wix/auto_sdk_payments_payment-acceptance-configurations/meta"));
var pspCallbacks = __toESM(require("@wix/auto_sdk_payments_psp-callbacks/meta"));
var refunds = __toESM(require("@wix/auto_sdk_payments_refunds/meta"));
var savedPaymentMethods = __toESM(require("@wix/auto_sdk_payments_saved-payment-methods/meta"));
var sitePaymentMethodTypes = __toESM(require("@wix/auto_sdk_payments_site-payment-method-types/meta"));
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  disputeEvidenceDocuments,
  disputeHistoryRecords,
  disputes,
  onboardingAvailability,
  payinsConfigurations,
  paymentAcceptanceConfigurations,
  pspCallbacks,
  refunds,
  savedPaymentMethods,
  sitePaymentMethodTypes,
  wixPaymentsAccount
});
