// @wix/ricos/dist/es/plugin-ai/ai-flow-selection-modal/ai-flow-selection-modal.js
import { DropdownLayout } from "@wix/design-system";
import { EditSparklesSmall, ImageSparklesSmall } from "@wix/wix-ui-icons-common";
import React, { useContext } from "react";
import { EditorContext, ModalServiceContext, RicosContext } from "./../../ricos-context/index.js";
import {
  AI_FLOW_ITEM_CONTENT,
  AI_FLOW_ITEM_IMAGE,
  AI_FLOW_SELECTION_POPOVER
} from "../data-hooks.js";
import styles from "./ai-flow-selection-modal-scss.js";
const AiFlowSelectionModal = ({
  toggleAiContentModal,
  toggleAiImageModal
}) => {
  const { t } = useContext(RicosContext);
  const modalService = useContext(ModalServiceContext);
  const { adapter } = useContext(EditorContext);
  const options = React.useMemo(() => {
    const commonOptions = {
      as: "button",
      ellipsis: false,
      size: "small",
      skin: "standard",
      title: false,
      value: "action",
      shouldFocusWithoutScroll: true
    };
    return [
      {
        ...commonOptions,
        dataHook: AI_FLOW_ITEM_CONTENT,
        id: "ai-content",
        onClick: () => toggleAiContentModal(adapter.tiptapEditor, modalService),
        optionTitle: t("AIPlugin_Flow_ContentAssistant"),
        prefixIcon: /* @__PURE__ */ React.createElement(EditSparklesSmall, null)
      },
      {
        ...commonOptions,
        dataHook: AI_FLOW_ITEM_IMAGE,
        id: "ai-image",
        onClick: () => toggleAiImageModal(adapter.tiptapEditor, modalService),
        optionTitle: t("AIPlugin_Flow_ImageGenerator"),
        prefixIcon: /* @__PURE__ */ React.createElement(ImageSparklesSmall, null)
      }
    ];
  }, []);
  return /* @__PURE__ */ React.createElement(
    DropdownLayout,
    {
      dataHook: AI_FLOW_SELECTION_POPOVER,
      className: styles.dropdown,
      maxHeightPixels: 250,
      minWidthPixels: 206,
      visible: true,
      options
    }
  );
};
export {
  AiFlowSelectionModal
};
