// @wix/unidriver-puppeteer/dist/cjs/stories/TestApp.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetAll = exports.Get = exports.WithDelayedItem = exports.WithThreeItemsAllCompleted = exports.WithThreeItemsNotCompleted = exports.WithThreeItems = exports.ItemWithEmptyId = exports.ItemsWithIds = exports.ReadonlyInput = exports.DisabledInput = exports.WithTwoItems = exports.WithItem = exports.Label = exports.EmptyLabel = exports.WithInitialText = exports.BaseEmptyInitialText = exports.Base = void 0;
const react_1 = require("react");
const testing_1 = require("@wix/unidriver-common/testing");
const meta = {
    title: 'TestApp',
    component: testing_1.TestApp,
};
exports.default = meta;
exports.Base = {
    name: 'base',
    args: { items: [] },
};
exports.BaseEmptyInitialText = {
    name: 'base-empty-initial-text',
    args: { items: [], initialText: '' },
};
exports.WithInitialText = {
    name: 'with-initial-text',
    args: { items: [], initialText: 'hello' },
};
exports.EmptyLabel = {
    name: 'empty-label',
    args: { items: [(0, testing_1.itemCreator)({ label: '' })], initialText: '' },
};
exports.Label = {
    name: 'label',
    args: {
        items: [(0, testing_1.itemCreator)({ label: 'hello' })],
        initialText: '',
    },
};
exports.WithItem = {
    name: 'with-item',
    args: { items: [(0, testing_1.itemCreator)({ label: 'Bob' })] },
};
exports.WithTwoItems = {
    name: 'with-two-items',
    args: {
        items: [(0, testing_1.itemCreator)({ label: 'Bob', completed: false }), (0, testing_1.itemCreator)({ label: 'David', completed: true })],
    },
};
exports.DisabledInput = {
    name: 'disabled-input',
    args: { items: [], initialText: 'other text', inputDisabled: true },
};
exports.ReadonlyInput = {
    name: 'readonly-input',
    args: { items: [], initialText: 'other text', inputReadOnly: true },
};
exports.ItemsWithIds = {
    name: 'items-with-ids',
    args: {
        items: [
            (0, testing_1.itemCreator)({ label: 'Bob', completed: false, id: 'bob' }),
            (0, testing_1.itemCreator)({ label: 'Alice', completed: false, id: 'alice' }),
        ],
    },
};
exports.ItemWithEmptyId = {
    name: 'item-with-empty-id',
    args: {
        items: [(0, testing_1.itemCreator)({ label: 'Bob', completed: false, id: '' })],
    },
};
exports.WithThreeItems = {
    name: 'with-three-items',
    args: {
        items: [
            (0, testing_1.itemCreator)({ label: 'Bob', completed: false }),
            (0, testing_1.itemCreator)({ id: 'david', label: 'David' }),
            (0, testing_1.itemCreator)({ id: 'jacob', label: 'Jacob', completed: true }),
        ],
    },
};
exports.WithThreeItemsNotCompleted = {
    name: 'with-three-items-not-completed',
    args: {
        items: [
            (0, testing_1.itemCreator)({ label: 'Bob', completed: false }),
            (0, testing_1.itemCreator)({ label: 'David' }),
            (0, testing_1.itemCreator)({ label: 'Jacob', completed: false }),
        ],
    },
};
exports.WithThreeItemsAllCompleted = {
    name: 'with-three-items-all-completed',
    args: {
        items: [
            (0, testing_1.itemCreator)({ label: 'Bob', completed: true }),
            (0, testing_1.itemCreator)({ label: 'David', completed: true }),
            (0, testing_1.itemCreator)({ label: 'Jacob', completed: true }),
        ],
    },
};
exports.WithDelayedItem = {
    name: 'with-delayed-item',
    render: () => {
        return (0, react_1.createElement)(testing_1.AsyncChildrenApp, {
            timeout: 2000,
            children: (0, react_1.createElement)('span', {
                className: 'delayed-item',
            }),
        });
    },
};
exports.Get = {
    name: 'get',
    args: {
        items: [(0, testing_1.itemCreator)({ label: 'Bob' })],
    },
};
exports.GetAll = {
    name: 'get-all',
    args: {
        items: [(0, testing_1.itemCreator)({ label: 'Bob' }), (0, testing_1.itemCreator)({ label: 'David' })],
    },
};
