// @wix/unidriver-puppeteer/dist/esm/__tests__/global-setup.js
import { eventually } from '@wix/unidriver-core/internal';
import { delay } from '@wix/unidriver-common';
import { execSync } from 'node:child_process';
import { isCI } from 'ci-info';
const DOCKER_CONTAINER_NAME = `puppeteerRunner_${Date.now().toString()}`;
const CHROME_PORT = '9222';
// We don't need teardown because --rm is used in the docker command
export async function setup() {
    if (isCI) {
        try {
            await startDockerChrome();
        }
        catch (e) {
            console.error('setup failed:', e);
            stopDockerChrome();
            throw e;
        }
    }
}
function runCommand(command) {
    console.log(`Running command: ${command}`);
    return execSync(command, { stdio: 'inherit' });
}
async function waitForChrome() {
    await eventually(async () => {
        // Wait 1 second before first attempt, need some time to start the server / wait between attempts
        await delay(1_000);
        const response = await fetch(`http://localhost:${CHROME_PORT}/json/version`);
        if (!response.ok) {
            throw new Error('Chrome failed to start');
        }
    }, {
        timeout: 20_000,
    });
    console.log('Chrome is ready!');
}
async function startDockerChrome() {
    runCommand(`docker run --detach --init --rm ` +
        `--network host ` +
        `--name ${DOCKER_CONTAINER_NAME} ` +
        `zenika/alpine-chrome:latest ` +
        `--remote-debugging-port=${CHROME_PORT} ` +
        `--no-sandbox ` +
        `--disable-dev-shm-usage ` +
        `--headless`);
    await waitForChrome();
}
function stopDockerChrome() {
    try {
        runCommand(`docker stop ${DOCKER_CONTAINER_NAME}`);
    }
    catch (e) {
        console.log('Failed to stop Docker runner:', e);
    }
}
