// @wix/unidriver-puppeteer/dist/esm/__tests__/setup.js
import { puppeteerUniDriver } from '../puppeteerUniDriver';
import puppeteer from 'puppeteer';
import { isCI } from 'ci-info';
import { eventually } from '@wix/unidriver-core/internal';
export const setupWithAwaiter = async (params) => {
    const { driver, tearDown, page } = await setup(params);
    const APP_SELECTOR = '[class="todo-app"]';
    await eventually(async () => {
        if (!(await driver.$(APP_SELECTOR).exists())) {
            throw new Error(`element with selector ${APP_SELECTOR} not found`);
        }
    }, 1_000);
    return { driver, tearDown, page };
};
export const setup = async ({ story }) => {
    const browser = isCI
        ? await puppeteer.connect({
            browserURL: 'http://localhost:9222',
        })
        : await puppeteer.launch({
            headless: process.env.DEBUG !== 'true',
        });
    const page = await browser.newPage();
    const url = `http://localhost:6006/iframe.html?id=testapp--${story}`;
    await page.goto(url, {
        waitUntil: 'networkidle0',
        timeout: 10_000,
    });
    const driver = puppeteerUniDriver({
        page,
        selector: 'body',
    });
    return {
        page,
        driver,
        tearDown: () => {
            return isCI ? page.close() : browser.close();
        },
    };
};
