// @wix/auto_sdk_restaurants_catalogs/build/internal/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  DayOfWeek: () => DayOfWeek,
  DiscountPlatform: () => DiscountPlatform,
  DiscountType: () => DiscountType,
  FulfillmentType: () => FulfillmentType,
  ItemType: () => ItemType,
  Platform: () => Platform,
  SortOrder: () => SortOrder,
  Type: () => Type,
  WebhookIdentityType: () => WebhookIdentityType,
  archiveMenu: () => archiveMenu4,
  bulkArchiveMenus: () => bulkArchiveMenus4,
  bulkCreateDishes: () => bulkCreateDishes4,
  bulkCreateMenus: () => bulkCreateMenus4,
  bulkCreateSections: () => bulkCreateSections4,
  bulkCreateVariations: () => bulkCreateVariations4,
  bulkUnarchiveMenus: () => bulkUnarchiveMenus4,
  bulkUpdateItems: () => bulkUpdateItems4,
  bulkUpdateMenus: () => bulkUpdateMenus4,
  bulkUpdateSections: () => bulkUpdateSections4,
  createDish: () => createDish4,
  createDraftCatalog: () => createDraftCatalog4,
  createMenu: () => createMenu4,
  createSection: () => createSection4,
  createVariation: () => createVariation4,
  discardDraftCatalog: () => discardDraftCatalog4,
  getItem: () => getItem4,
  getMenu: () => getMenu4,
  getSection: () => getSection4,
  listCatalogs: () => listCatalogs4,
  listItems: () => listItems4,
  listMenus: () => listMenus4,
  listSections: () => listSections4,
  onCatalogChanged: () => onCatalogChanged2,
  onCatalogCreated: () => onCatalogCreated2,
  publishDraftCatalog: () => publishDraftCatalog4,
  unarchiveMenu: () => unarchiveMenu4,
  updateItem: () => updateItem4,
  updateMenu: () => updateMenu4,
  updateSection: () => updateSection4
});
module.exports = __toCommonJS(index_exports);

// src/restaurants-catalogs-v3-catalog-catalogs.public.ts
var import_rename_all_nested_keys2 = require("@wix/sdk-runtime/rename-all-nested-keys");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var import_sdk_types = require("@wix/sdk-types");

// src/restaurants-catalogs-v3-catalog-catalogs.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/restaurants-catalogs-v3-catalog-catalogs.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      },
      {
        srcPath: "/restaurants/v3/catalog",
        destPath: "/v3/catalog"
      },
      {
        srcPath: "/restaurants/v3/draft-catalogs",
        destPath: "/v3/draft-catalogs"
      },
      {
        srcPath: "/restaurants/v3/catalogs_test",
        destPath: "/v3/catalogs"
      },
      {
        srcPath: "/_api/restaurants/v3/catalogs_test",
        destPath: "/v3/catalogs"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    "users._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      },
      {
        srcPath: "/restaurants/v3/catalog",
        destPath: "/v3/catalog"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
function resolveComWixpressRestaurantsCatalogsV3DraftCatalogsApiUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      },
      {
        srcPath: "/restaurants/v3/catalog",
        destPath: "/v3/catalog"
      },
      {
        srcPath: "/restaurants/v3/draft-catalogs",
        destPath: "/v3/draft-catalogs"
      },
      {
        srcPath: "/restaurants/v3/catalogs_test",
        destPath: "/v3/catalogs"
      },
      {
        srcPath: "/_api/restaurants/v3/catalogs_test",
        destPath: "/v3/catalogs"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    "users._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      },
      {
        srcPath: "/restaurants/v3/catalog",
        destPath: "/v3/catalog"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_catalogs";
function createDraftCatalog(payload) {
  function __createDraftCatalog({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.DraftCatalogsApi.CreateDraftCatalog",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3DraftCatalogsApiUrl({
        protoPath: "/v3/draft-catalogs/{catalogId}/create",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "catalog.draftPublishedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __createDraftCatalog;
}
function publishDraftCatalog(payload) {
  function __publishDraftCatalog({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.DraftCatalogsApi.PublishDraftCatalog",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3DraftCatalogsApiUrl({
        protoPath: "/v3/draft-catalogs/{catalogId}/publish",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "catalog.draftPublishedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __publishDraftCatalog;
}
function discardDraftCatalog(payload) {
  function __discardDraftCatalog({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "DELETE",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.DraftCatalogsApi.DiscardDraftCatalog",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3DraftCatalogsApiUrl({
        protoPath: "/v3/draft-catalogs/{catalogId}/discard",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __discardDraftCatalog;
}
function bulkCreateVariations(payload) {
  function __bulkCreateVariations({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.DraftCatalogsApi.BulkCreateVariations",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3DraftCatalogsApiUrl({
        protoPath: "/v3/draft-catalogs/{catalogId}/bulk/variations/create",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkCreateVariations;
}
function bulkUpdateItems(payload) {
  function __bulkUpdateItems({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.DraftCatalogsApi.BulkUpdateItems",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3DraftCatalogsApiUrl({
        protoPath: "/v3/draft-catalogs/{catalogId}/bulk/items/update",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateItems;
}
function bulkCreateDishes(payload) {
  function __bulkCreateDishes({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.DraftCatalogsApi.BulkCreateDishes",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3DraftCatalogsApiUrl({
        protoPath: "/v3/draft-catalogs/{catalogId}/bulk/dishes/create",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkCreateDishes;
}
function bulkCreateMenus(payload) {
  function __bulkCreateMenus({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.DraftCatalogsApi.BulkCreateMenus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3DraftCatalogsApiUrl({
        protoPath: "/v3/draft-catalogs/{catalogId}/bulk/menus/create",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkCreateMenus;
}
function bulkUpdateMenus(payload) {
  function __bulkUpdateMenus({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.DraftCatalogsApi.BulkUpdateMenus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3DraftCatalogsApiUrl({
        protoPath: "/v3/draft-catalogs/{catalogId}/bulk/menus/update",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateMenus;
}
function bulkCreateSections(payload) {
  function __bulkCreateSections({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.DraftCatalogsApi.BulkCreateSections",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3DraftCatalogsApiUrl({
        protoPath: "/v3/draft-catalogs/{catalogId}/bulk/sections/create",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkCreateSections;
}
function bulkUpdateSections(payload) {
  function __bulkUpdateSections({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.DraftCatalogsApi.BulkUpdateSections",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3DraftCatalogsApiUrl({
        protoPath: "/v3/draft-catalogs/{catalogId}/bulk/sections/update",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateSections;
}
function bulkArchiveMenus(payload) {
  function __bulkArchiveMenus({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.DraftCatalogsApi.BulkArchiveMenus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3DraftCatalogsApiUrl({
        protoPath: "/v3/draft-catalogs/{catalogId}/bulk/menus/archive",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkArchiveMenus;
}
function bulkUnarchiveMenus(payload) {
  function __bulkUnarchiveMenus({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.DraftCatalogsApi.BulkUnarchiveMenus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3DraftCatalogsApiUrl({
        protoPath: "/v3/draft-catalogs/{catalogId}/bulk/menus/unarchive",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUnarchiveMenus;
}
function listCatalogs(payload) {
  function __listCatalogs({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.ListCatalogs",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload, true),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "catalogs.draftPublishedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __listCatalogs;
}
function getMenu(payload) {
  function __getMenu({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.GetMenu",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/menus/{menuId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getMenu;
}
function getSection(payload) {
  function __getSection({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.GetSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/menus/{menuId}/sections/{sectionId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getSection;
}
function updateMenu(payload) {
  function __updateMenu({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "PATCH",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.UpdateMenu",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/menus/{menu.id}",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __updateMenu;
}
function updateSection(payload) {
  function __updateSection({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "PATCH",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.UpdateSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/menus/{menuId}/sections/{section.id}",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __updateSection;
}
function listMenus(payload) {
  function __listMenus({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.ListMenus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/menus",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData)
    };
    return metadata;
  }
  return __listMenus;
}
function listSections(payload) {
  function __listSections({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.ListSections",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/sections",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData)
    };
    return metadata;
  }
  return __listSections;
}
function createMenu(payload) {
  function __createMenu({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.CreateMenu",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/menus",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __createMenu;
}
function archiveMenu(payload) {
  function __archiveMenu({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.ArchiveMenu",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/menus/{menuId}/archive",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __archiveMenu;
}
function unarchiveMenu(payload) {
  function __unarchiveMenu({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.UnarchiveMenu",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/menus/{menuId}/unarchive",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __unarchiveMenu;
}
function createSection(payload) {
  function __createSection({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.CreateSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/menus/{menuId}/sections",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __createSection;
}
function getItem(payload) {
  function __getItem({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.GetItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/items/{itemId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getItem;
}
function listItems(payload) {
  function __listItems({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.ListItems",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/items",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData)
    };
    return metadata;
  }
  return __listItems;
}
function updateItem(payload) {
  function __updateItem({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "PATCH",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.UpdateItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/items/{item.id}",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __updateItem;
}
function createDish(payload) {
  function __createDish({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.CreateDish",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/menus/{menuId}/sections/{sectionId}/items",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __createDish;
}
function createVariation(payload) {
  function __createVariation({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.CreateVariation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/variations",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __createVariation;
}

// src/restaurants-catalogs-v3-catalog-catalogs.universal.ts
var FulfillmentType = /* @__PURE__ */ ((FulfillmentType2) => {
  FulfillmentType2["UNSPECIFIED_FULFILLMENT_TYPE"] = "UNSPECIFIED_FULFILLMENT_TYPE";
  FulfillmentType2["DELIVERY"] = "DELIVERY";
  FulfillmentType2["PICKUP_OR_DINE_IN"] = "PICKUP_OR_DINE_IN";
  return FulfillmentType2;
})(FulfillmentType || {});
var Platform = /* @__PURE__ */ ((Platform2) => {
  Platform2["UNSPECIFIED_PLATFORM"] = "UNSPECIFIED_PLATFORM";
  Platform2["SITE"] = "SITE";
  Platform2["MOBILE_SITE"] = "MOBILE_SITE";
  Platform2["WIX_APP"] = "WIX_APP";
  Platform2["CALL_CENTER"] = "CALL_CENTER";
  Platform2["CHAT_BOT"] = "CHAT_BOT";
  return Platform2;
})(Platform || {});
var DayOfWeek = /* @__PURE__ */ ((DayOfWeek2) => {
  DayOfWeek2["UNDEFINED"] = "UNDEFINED";
  DayOfWeek2["SUN"] = "SUN";
  DayOfWeek2["MON"] = "MON";
  DayOfWeek2["TUE"] = "TUE";
  DayOfWeek2["WED"] = "WED";
  DayOfWeek2["THU"] = "THU";
  DayOfWeek2["FRI"] = "FRI";
  DayOfWeek2["SAT"] = "SAT";
  return DayOfWeek2;
})(DayOfWeek || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNSPECIFIED_DISH_OPTION_TYPE"] = "UNSPECIFIED_DISH_OPTION_TYPE";
  Type2["SELECTION"] = "SELECTION";
  Type2["EXTRAS"] = "EXTRAS";
  Type2["DESELECTION"] = "DESELECTION";
  return Type2;
})(Type || {});
var ItemType = /* @__PURE__ */ ((ItemType2) => {
  ItemType2["UNSPECIFIED_ITEM_TYPE"] = "UNSPECIFIED_ITEM_TYPE";
  ItemType2["DISH"] = "DISH";
  ItemType2["VARIATION"] = "VARIATION";
  return ItemType2;
})(ItemType || {});
var DiscountType = /* @__PURE__ */ ((DiscountType2) => {
  DiscountType2["UNSPECIFIED_TYPE"] = "UNSPECIFIED_TYPE";
  DiscountType2["OFF_ITEM"] = "OFF_ITEM";
  DiscountType2["OFF_ORDER"] = "OFF_ORDER";
  return DiscountType2;
})(DiscountType || {});
var DiscountPlatform = /* @__PURE__ */ ((DiscountPlatform2) => {
  DiscountPlatform2["UNSPECIFIED_PLATFORM"] = "UNSPECIFIED_PLATFORM";
  DiscountPlatform2["SITE"] = "SITE";
  DiscountPlatform2["MOBILE_SITE"] = "MOBILE_SITE";
  DiscountPlatform2["CALL_CENTER"] = "CALL_CENTER";
  return DiscountPlatform2;
})(DiscountPlatform || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createDraftCatalog2(catalogId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogId
  });
  const reqOpts = createDraftCatalog(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { catalogId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["catalogId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function publishDraftCatalog2(catalogId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogId
  });
  const reqOpts = publishDraftCatalog(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { catalogId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["catalogId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function discardDraftCatalog2(catalogId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogId
  });
  const reqOpts = discardDraftCatalog(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { catalogId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["catalogId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCreateVariations2(catalogId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogId,
    variations: options?.variations,
    returnFullEntity: options?.returnFullEntity
  });
  const reqOpts = bulkCreateVariations(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          catalogId: "$[0]",
          variations: "$[1].variations",
          returnFullEntity: "$[1].returnFullEntity"
        },
        singleArgumentUnchanged: false
      },
      ["catalogId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateItems2(catalogId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogId,
    items: options?.items,
    returnFullEntity: options?.returnFullEntity
  });
  const reqOpts = bulkUpdateItems(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          catalogId: "$[0]",
          items: "$[1].items",
          returnFullEntity: "$[1].returnFullEntity"
        },
        singleArgumentUnchanged: false
      },
      ["catalogId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCreateDishes2(catalogId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogId,
    createDishRequests: options?.createDishRequests,
    returnFullEntity: options?.returnFullEntity
  });
  const reqOpts = bulkCreateDishes(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          catalogId: "$[0]",
          createDishRequests: "$[1].createDishRequests",
          returnFullEntity: "$[1].returnFullEntity"
        },
        singleArgumentUnchanged: false
      },
      ["catalogId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCreateMenus2(catalogId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogId,
    menus: options?.menus,
    returnFullEntity: options?.returnFullEntity
  });
  const reqOpts = bulkCreateMenus(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          catalogId: "$[0]",
          menus: "$[1].menus",
          returnFullEntity: "$[1].returnFullEntity"
        },
        singleArgumentUnchanged: false
      },
      ["catalogId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateMenus2(catalogId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogId,
    menus: options?.menus,
    returnFullEntity: options?.returnFullEntity
  });
  const reqOpts = bulkUpdateMenus(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          catalogId: "$[0]",
          menus: "$[1].menus",
          returnFullEntity: "$[1].returnFullEntity"
        },
        singleArgumentUnchanged: false
      },
      ["catalogId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCreateSections2(catalogId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogId,
    createSectionRequests: options?.createSectionRequests,
    returnFullEntity: options?.returnFullEntity
  });
  const reqOpts = bulkCreateSections(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          catalogId: "$[0]",
          createSectionRequests: "$[1].createSectionRequests",
          returnFullEntity: "$[1].returnFullEntity"
        },
        singleArgumentUnchanged: false
      },
      ["catalogId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateSections2(catalogId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogId,
    updateSectionsRequests: options?.updateSectionsRequests,
    returnFullEntity: options?.returnFullEntity
  });
  const reqOpts = bulkUpdateSections(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          catalogId: "$[0]",
          updateSectionsRequests: "$[1].updateSectionsRequests",
          returnFullEntity: "$[1].returnFullEntity"
        },
        singleArgumentUnchanged: false
      },
      ["catalogId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkArchiveMenus2(catalogId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogId,
    ids: options?.ids,
    returnFullEntity: options?.returnFullEntity
  });
  const reqOpts = bulkArchiveMenus(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          catalogId: "$[0]",
          ids: "$[1].ids",
          returnFullEntity: "$[1].returnFullEntity"
        },
        singleArgumentUnchanged: false
      },
      ["catalogId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUnarchiveMenus2(catalogId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogId,
    ids: options?.ids,
    returnFullEntity: options?.returnFullEntity
  });
  const reqOpts = bulkUnarchiveMenus(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          catalogId: "$[0]",
          ids: "$[1].ids",
          returnFullEntity: "$[1].returnFullEntity"
        },
        singleArgumentUnchanged: false
      },
      ["catalogId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listCatalogs2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    sort: options?.sort,
    paging: options?.paging,
    archived: options?.archived,
    locationIds: options?.locationIds
  });
  const reqOpts = listCatalogs(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          sort: "$[0].sort",
          paging: "$[0].paging",
          archived: "$[0].archived",
          locationIds: "$[0].locationIds"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getMenu2(identifiers) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogId: identifiers?.catalogId,
    menuId: identifiers?.menuId
  });
  const reqOpts = getMenu(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          catalogId: "$[0].catalogId",
          menuId: "$[0].menuId"
        },
        singleArgumentUnchanged: false
      },
      ["identifiers"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getSection2(identifiers) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogId: identifiers?.catalogId,
    menuId: identifiers?.menuId,
    sectionId: identifiers?.sectionId
  });
  const reqOpts = getSection(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          catalogId: "$[0].catalogId",
          menuId: "$[0].menuId",
          sectionId: "$[0].sectionId"
        },
        singleArgumentUnchanged: false
      },
      ["identifiers"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateMenu2(identifiers, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogId: identifiers?.catalogId,
    menu: { ...options?.menu, id: identifiers?.menuId }
  });
  const reqOpts = updateMenu(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { menu: "$[1].menu" },
        explicitPathsToArguments: {
          catalogId: "$[0].catalogId",
          "menu.id": "$[0].menuId"
        },
        singleArgumentUnchanged: false
      },
      ["identifiers", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateSection2(identifiers, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogId: identifiers?.catalogId,
    menuId: identifiers?.menuId,
    section: { ...options?.section, id: identifiers?.sectionId }
  });
  const reqOpts = updateSection(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { section: "$[1].section" },
        explicitPathsToArguments: {
          catalogId: "$[0].catalogId",
          menuId: "$[0].menuId",
          "section.id": "$[0].sectionId"
        },
        singleArgumentUnchanged: false
      },
      ["identifiers", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listMenus2(catalogId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogId,
    fieldMask: options?.fieldMask,
    archived: options?.archived
  });
  const reqOpts = listMenus(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          catalogId: "$[0]",
          fieldMask: "$[1].fieldMask",
          archived: "$[1].archived"
        },
        singleArgumentUnchanged: false
      },
      ["catalogId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listSections2(catalogId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogId,
    fieldMask: options?.fieldMask,
    archived: options?.archived
  });
  const reqOpts = listSections(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          catalogId: "$[0]",
          fieldMask: "$[1].fieldMask",
          archived: "$[1].archived"
        },
        singleArgumentUnchanged: false
      },
      ["catalogId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function createMenu2(catalogId, menu) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogId,
    menu
  });
  const reqOpts = createMenu(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { catalogId: "$[0]", menu: "$[1]" },
        singleArgumentUnchanged: false
      },
      ["catalogId", "menu"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function archiveMenu2(identifiers) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogId: identifiers?.catalogId,
    menuId: identifiers?.menuId
  });
  const reqOpts = archiveMenu(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          catalogId: "$[0].catalogId",
          menuId: "$[0].menuId"
        },
        singleArgumentUnchanged: false
      },
      ["identifiers"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function unarchiveMenu2(identifiers) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogId: identifiers?.catalogId,
    menuId: identifiers?.menuId
  });
  const reqOpts = unarchiveMenu(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          catalogId: "$[0].catalogId",
          menuId: "$[0].menuId"
        },
        singleArgumentUnchanged: false
      },
      ["identifiers"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function createSection2(identifiers, section) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogId: identifiers?.catalogId,
    menuId: identifiers?.menuId,
    section
  });
  const reqOpts = createSection(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          catalogId: "$[0].catalogId",
          menuId: "$[0].menuId",
          section: "$[1]"
        },
        singleArgumentUnchanged: false
      },
      ["identifiers", "section"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getItem2(identifiers, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogId: identifiers?.catalogId,
    itemId: identifiers?.itemId,
    includeVisibilityCriteria: options?.includeVisibilityCriteria
  });
  const reqOpts = getItem(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          catalogId: "$[0].catalogId",
          itemId: "$[0].itemId",
          includeVisibilityCriteria: "$[1].includeVisibilityCriteria"
        },
        singleArgumentUnchanged: false
      },
      ["identifiers", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listItems2(catalogId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogId,
    fieldMask: options?.fieldMask,
    archived: options?.archived
  });
  const reqOpts = listItems(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          catalogId: "$[0]",
          fieldMask: "$[1].fieldMask",
          archived: "$[1].archived"
        },
        singleArgumentUnchanged: false
      },
      ["catalogId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateItem2(identifiers, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogId: identifiers?.catalogId,
    item: { ...options?.item, id: identifiers?.itemId }
  });
  const reqOpts = updateItem(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { item: "$[1].item" },
        explicitPathsToArguments: {
          catalogId: "$[0].catalogId",
          "item.id": "$[0].itemId"
        },
        singleArgumentUnchanged: false
      },
      ["identifiers", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function createDish2(identifiers, dish) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogId: identifiers?.catalogId,
    menuId: identifiers?.menuId,
    sectionId: identifiers?.sectionId,
    dish
  });
  const reqOpts = createDish(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          catalogId: "$[0].catalogId",
          menuId: "$[0].menuId",
          sectionId: "$[0].sectionId",
          dish: "$[1]"
        },
        singleArgumentUnchanged: false
      },
      ["identifiers", "dish"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function createVariation2(catalogId, name) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogId,
    name
  });
  const reqOpts = createVariation(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { catalogId: "$[0]", name: "$[1]" },
        singleArgumentUnchanged: false
      },
      ["catalogId", "name"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/restaurants-catalogs-v3-catalog-catalogs.public.ts
function createDraftCatalog3(httpClient) {
  return (catalogId) => createDraftCatalog2(
    catalogId,
    // @ts-ignore
    { httpClient }
  );
}
function publishDraftCatalog3(httpClient) {
  return (catalogId) => publishDraftCatalog2(
    catalogId,
    // @ts-ignore
    { httpClient }
  );
}
function discardDraftCatalog3(httpClient) {
  return (catalogId) => discardDraftCatalog2(
    catalogId,
    // @ts-ignore
    { httpClient }
  );
}
function bulkCreateVariations3(httpClient) {
  return (catalogId, options) => bulkCreateVariations2(
    catalogId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdateItems3(httpClient) {
  return (catalogId, options) => bulkUpdateItems2(
    catalogId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkCreateDishes3(httpClient) {
  return (catalogId, options) => bulkCreateDishes2(
    catalogId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkCreateMenus3(httpClient) {
  return (catalogId, options) => bulkCreateMenus2(
    catalogId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdateMenus3(httpClient) {
  return (catalogId, options) => bulkUpdateMenus2(
    catalogId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkCreateSections3(httpClient) {
  return (catalogId, options) => bulkCreateSections2(
    catalogId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdateSections3(httpClient) {
  return (catalogId, options) => bulkUpdateSections2(
    catalogId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkArchiveMenus3(httpClient) {
  return (catalogId, options) => bulkArchiveMenus2(
    catalogId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkUnarchiveMenus3(httpClient) {
  return (catalogId, options) => bulkUnarchiveMenus2(
    catalogId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function listCatalogs3(httpClient) {
  return (options) => listCatalogs2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function getMenu3(httpClient) {
  return (identifiers) => getMenu2(
    identifiers,
    // @ts-ignore
    { httpClient }
  );
}
function getSection3(httpClient) {
  return (identifiers) => getSection2(
    identifiers,
    // @ts-ignore
    { httpClient }
  );
}
function updateMenu3(httpClient) {
  return (identifiers, options) => updateMenu2(
    identifiers,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function updateSection3(httpClient) {
  return (identifiers, options) => updateSection2(
    identifiers,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function listMenus3(httpClient) {
  return (catalogId, options) => listMenus2(
    catalogId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function listSections3(httpClient) {
  return (catalogId, options) => listSections2(
    catalogId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function createMenu3(httpClient) {
  return (catalogId, menu) => createMenu2(
    catalogId,
    menu,
    // @ts-ignore
    { httpClient }
  );
}
function archiveMenu3(httpClient) {
  return (identifiers) => archiveMenu2(
    identifiers,
    // @ts-ignore
    { httpClient }
  );
}
function unarchiveMenu3(httpClient) {
  return (identifiers) => unarchiveMenu2(
    identifiers,
    // @ts-ignore
    { httpClient }
  );
}
function createSection3(httpClient) {
  return (identifiers, section) => createSection2(
    identifiers,
    section,
    // @ts-ignore
    { httpClient }
  );
}
function getItem3(httpClient) {
  return (identifiers, options) => getItem2(
    identifiers,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function listItems3(httpClient) {
  return (catalogId, options) => listItems2(
    catalogId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function updateItem3(httpClient) {
  return (identifiers, options) => updateItem2(
    identifiers,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function createDish3(httpClient) {
  return (identifiers, dish) => createDish2(
    identifiers,
    dish,
    // @ts-ignore
    { httpClient }
  );
}
function createVariation3(httpClient) {
  return (catalogId, name) => createVariation2(
    catalogId,
    name,
    // @ts-ignore
    { httpClient }
  );
}
var onCatalogChanged = (0, import_sdk_types.EventDefinition)(
  "wix.restaurants.catalogs.v3.catalog_catalog_changed",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths2.transformPaths)(event, [
      {
        transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "data.catalog.draftPublishedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onCatalogCreated = (0, import_sdk_types.EventDefinition)(
  "wix.restaurants.catalogs.v3.catalog_created",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths2.transformPaths)(event, [
      {
        transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.draftPublishedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();

// src/restaurants-catalogs-v3-catalog-catalogs.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var import_event_definition_modules = require("@wix/sdk-runtime/event-definition-modules");
var createDraftCatalog4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(createDraftCatalog3);
var publishDraftCatalog4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(publishDraftCatalog3);
var discardDraftCatalog4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(discardDraftCatalog3);
var bulkCreateVariations4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkCreateVariations3);
var bulkUpdateItems4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkUpdateItems3);
var bulkCreateDishes4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkCreateDishes3);
var bulkCreateMenus4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkCreateMenus3);
var bulkUpdateMenus4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkUpdateMenus3);
var bulkCreateSections4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkCreateSections3);
var bulkUpdateSections4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkUpdateSections3);
var bulkArchiveMenus4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkArchiveMenus3);
var bulkUnarchiveMenus4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkUnarchiveMenus3);
var listCatalogs4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(listCatalogs3);
var getMenu4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getMenu3);
var getSection4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getSection3);
var updateMenu4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(updateMenu3);
var updateSection4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(updateSection3);
var listMenus4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(listMenus3);
var listSections4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(listSections3);
var createMenu4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(createMenu3);
var archiveMenu4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(archiveMenu3);
var unarchiveMenu4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(unarchiveMenu3);
var createSection4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(createSection3);
var getItem4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getItem3);
var listItems4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(listItems3);
var updateItem4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(updateItem3);
var createDish4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(createDish3);
var createVariation4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(createVariation3);
var onCatalogChanged2 = (0, import_event_definition_modules.createEventModule)(onCatalogChanged);
var onCatalogCreated2 = (0, import_event_definition_modules.createEventModule)(onCatalogCreated);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  DayOfWeek,
  DiscountPlatform,
  DiscountType,
  FulfillmentType,
  ItemType,
  Platform,
  SortOrder,
  Type,
  WebhookIdentityType,
  archiveMenu,
  bulkArchiveMenus,
  bulkCreateDishes,
  bulkCreateMenus,
  bulkCreateSections,
  bulkCreateVariations,
  bulkUnarchiveMenus,
  bulkUpdateItems,
  bulkUpdateMenus,
  bulkUpdateSections,
  createDish,
  createDraftCatalog,
  createMenu,
  createSection,
  createVariation,
  discardDraftCatalog,
  getItem,
  getMenu,
  getSection,
  listCatalogs,
  listItems,
  listMenus,
  listSections,
  onCatalogChanged,
  onCatalogCreated,
  publishDraftCatalog,
  unarchiveMenu,
  updateItem,
  updateMenu,
  updateSection
});
