// @wix/auto_sdk_restaurants_catalogs/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  DayOfWeekOriginal: () => DayOfWeek,
  DiscountPlatformOriginal: () => DiscountPlatform,
  DiscountTypeOriginal: () => DiscountType,
  FulfillmentTypeOriginal: () => FulfillmentType,
  ItemTypeOriginal: () => ItemType,
  PlatformOriginal: () => Platform,
  SortOrderOriginal: () => SortOrder,
  TypeOriginal: () => Type,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  archiveMenu: () => archiveMenu2,
  bulkArchiveMenus: () => bulkArchiveMenus2,
  bulkCreateDishes: () => bulkCreateDishes2,
  bulkCreateMenus: () => bulkCreateMenus2,
  bulkCreateSections: () => bulkCreateSections2,
  bulkCreateVariations: () => bulkCreateVariations2,
  bulkUnarchiveMenus: () => bulkUnarchiveMenus2,
  bulkUpdateItems: () => bulkUpdateItems2,
  bulkUpdateMenus: () => bulkUpdateMenus2,
  bulkUpdateSections: () => bulkUpdateSections2,
  createDish: () => createDish2,
  createDraftCatalog: () => createDraftCatalog2,
  createMenu: () => createMenu2,
  createSection: () => createSection2,
  createVariation: () => createVariation2,
  discardDraftCatalog: () => discardDraftCatalog2,
  getItem: () => getItem2,
  getMenu: () => getMenu2,
  getSection: () => getSection2,
  listCatalogs: () => listCatalogs2,
  listItems: () => listItems2,
  listMenus: () => listMenus2,
  listSections: () => listSections2,
  publishDraftCatalog: () => publishDraftCatalog2,
  unarchiveMenu: () => unarchiveMenu2,
  updateItem: () => updateItem2,
  updateMenu: () => updateMenu2,
  updateSection: () => updateSection2
});
module.exports = __toCommonJS(meta_exports);

// src/restaurants-catalogs-v3-catalog-catalogs.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      },
      {
        srcPath: "/restaurants/v3/catalog",
        destPath: "/v3/catalog"
      },
      {
        srcPath: "/restaurants/v3/draft-catalogs",
        destPath: "/v3/draft-catalogs"
      },
      {
        srcPath: "/restaurants/v3/catalogs_test",
        destPath: "/v3/catalogs"
      },
      {
        srcPath: "/_api/restaurants/v3/catalogs_test",
        destPath: "/v3/catalogs"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    "users._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      },
      {
        srcPath: "/restaurants/v3/catalog",
        destPath: "/v3/catalog"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
function resolveComWixpressRestaurantsCatalogsV3DraftCatalogsApiUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      },
      {
        srcPath: "/restaurants/v3/catalog",
        destPath: "/v3/catalog"
      },
      {
        srcPath: "/restaurants/v3/draft-catalogs",
        destPath: "/v3/draft-catalogs"
      },
      {
        srcPath: "/restaurants/v3/catalogs_test",
        destPath: "/v3/catalogs"
      },
      {
        srcPath: "/_api/restaurants/v3/catalogs_test",
        destPath: "/v3/catalogs"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    "users._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      },
      {
        srcPath: "/restaurants/v3/catalog",
        destPath: "/v3/catalog"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/catalogs",
        destPath: "/v3/catalogs"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_catalogs";
function createDraftCatalog(payload) {
  function __createDraftCatalog({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.DraftCatalogsApi.CreateDraftCatalog",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3DraftCatalogsApiUrl({
        protoPath: "/v3/draft-catalogs/{catalogId}/create",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "catalog.draftPublishedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __createDraftCatalog;
}
function publishDraftCatalog(payload) {
  function __publishDraftCatalog({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.DraftCatalogsApi.PublishDraftCatalog",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3DraftCatalogsApiUrl({
        protoPath: "/v3/draft-catalogs/{catalogId}/publish",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "catalog.draftPublishedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __publishDraftCatalog;
}
function discardDraftCatalog(payload) {
  function __discardDraftCatalog({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "DELETE",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.DraftCatalogsApi.DiscardDraftCatalog",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3DraftCatalogsApiUrl({
        protoPath: "/v3/draft-catalogs/{catalogId}/discard",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __discardDraftCatalog;
}
function bulkCreateVariations(payload) {
  function __bulkCreateVariations({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.DraftCatalogsApi.BulkCreateVariations",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3DraftCatalogsApiUrl({
        protoPath: "/v3/draft-catalogs/{catalogId}/bulk/variations/create",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkCreateVariations;
}
function bulkUpdateItems(payload) {
  function __bulkUpdateItems({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.DraftCatalogsApi.BulkUpdateItems",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3DraftCatalogsApiUrl({
        protoPath: "/v3/draft-catalogs/{catalogId}/bulk/items/update",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateItems;
}
function bulkCreateDishes(payload) {
  function __bulkCreateDishes({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.DraftCatalogsApi.BulkCreateDishes",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3DraftCatalogsApiUrl({
        protoPath: "/v3/draft-catalogs/{catalogId}/bulk/dishes/create",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkCreateDishes;
}
function bulkCreateMenus(payload) {
  function __bulkCreateMenus({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.DraftCatalogsApi.BulkCreateMenus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3DraftCatalogsApiUrl({
        protoPath: "/v3/draft-catalogs/{catalogId}/bulk/menus/create",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkCreateMenus;
}
function bulkUpdateMenus(payload) {
  function __bulkUpdateMenus({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.DraftCatalogsApi.BulkUpdateMenus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3DraftCatalogsApiUrl({
        protoPath: "/v3/draft-catalogs/{catalogId}/bulk/menus/update",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateMenus;
}
function bulkCreateSections(payload) {
  function __bulkCreateSections({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.DraftCatalogsApi.BulkCreateSections",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3DraftCatalogsApiUrl({
        protoPath: "/v3/draft-catalogs/{catalogId}/bulk/sections/create",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkCreateSections;
}
function bulkUpdateSections(payload) {
  function __bulkUpdateSections({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.DraftCatalogsApi.BulkUpdateSections",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3DraftCatalogsApiUrl({
        protoPath: "/v3/draft-catalogs/{catalogId}/bulk/sections/update",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateSections;
}
function bulkArchiveMenus(payload) {
  function __bulkArchiveMenus({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.DraftCatalogsApi.BulkArchiveMenus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3DraftCatalogsApiUrl({
        protoPath: "/v3/draft-catalogs/{catalogId}/bulk/menus/archive",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkArchiveMenus;
}
function bulkUnarchiveMenus(payload) {
  function __bulkUnarchiveMenus({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.DraftCatalogsApi.BulkUnarchiveMenus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3DraftCatalogsApiUrl({
        protoPath: "/v3/draft-catalogs/{catalogId}/bulk/menus/unarchive",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUnarchiveMenus;
}
function listCatalogs(payload) {
  function __listCatalogs({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.ListCatalogs",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload, true),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "catalogs.draftPublishedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __listCatalogs;
}
function getMenu(payload) {
  function __getMenu({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.GetMenu",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/menus/{menuId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getMenu;
}
function getSection(payload) {
  function __getSection({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.GetSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/menus/{menuId}/sections/{sectionId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getSection;
}
function updateMenu(payload) {
  function __updateMenu({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "PATCH",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.UpdateMenu",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/menus/{menu.id}",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __updateMenu;
}
function updateSection(payload) {
  function __updateSection({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "PATCH",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.UpdateSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/menus/{menuId}/sections/{section.id}",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __updateSection;
}
function listMenus(payload) {
  function __listMenus({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.ListMenus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/menus",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData)
    };
    return metadata;
  }
  return __listMenus;
}
function listSections(payload) {
  function __listSections({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.ListSections",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/sections",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData)
    };
    return metadata;
  }
  return __listSections;
}
function createMenu(payload) {
  function __createMenu({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.CreateMenu",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/menus",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __createMenu;
}
function archiveMenu(payload) {
  function __archiveMenu({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.ArchiveMenu",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/menus/{menuId}/archive",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __archiveMenu;
}
function unarchiveMenu(payload) {
  function __unarchiveMenu({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.UnarchiveMenu",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/menus/{menuId}/unarchive",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __unarchiveMenu;
}
function createSection(payload) {
  function __createSection({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.CreateSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/menus/{menuId}/sections",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __createSection;
}
function getItem(payload) {
  function __getItem({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.GetItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/items/{itemId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getItem;
}
function listItems(payload) {
  function __listItems({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.ListItems",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/items",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData)
    };
    return metadata;
  }
  return __listItems;
}
function updateItem(payload) {
  function __updateItem({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "PATCH",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.UpdateItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/items/{item.id}",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __updateItem;
}
function createDish(payload) {
  function __createDish({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.CreateDish",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/menus/{menuId}/sections/{sectionId}/items",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __createDish;
}
function createVariation(payload) {
  function __createVariation({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.catalogs.v3.catalog",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.catalogs.v3.CatalogsApi.CreateVariation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsCatalogsV3CatalogsApiUrl({
        protoPath: "/v3/catalogs/{catalogId}/variations",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __createVariation;
}

// src/restaurants-catalogs-v3-catalog-catalogs.types.ts
var FulfillmentType = /* @__PURE__ */ ((FulfillmentType2) => {
  FulfillmentType2["UNSPECIFIED_FULFILLMENT_TYPE"] = "UNSPECIFIED_FULFILLMENT_TYPE";
  FulfillmentType2["DELIVERY"] = "DELIVERY";
  FulfillmentType2["PICKUP_OR_DINE_IN"] = "PICKUP_OR_DINE_IN";
  return FulfillmentType2;
})(FulfillmentType || {});
var Platform = /* @__PURE__ */ ((Platform2) => {
  Platform2["UNSPECIFIED_PLATFORM"] = "UNSPECIFIED_PLATFORM";
  Platform2["SITE"] = "SITE";
  Platform2["MOBILE_SITE"] = "MOBILE_SITE";
  Platform2["WIX_APP"] = "WIX_APP";
  Platform2["CALL_CENTER"] = "CALL_CENTER";
  Platform2["CHAT_BOT"] = "CHAT_BOT";
  return Platform2;
})(Platform || {});
var DayOfWeek = /* @__PURE__ */ ((DayOfWeek2) => {
  DayOfWeek2["UNDEFINED"] = "UNDEFINED";
  DayOfWeek2["SUN"] = "SUN";
  DayOfWeek2["MON"] = "MON";
  DayOfWeek2["TUE"] = "TUE";
  DayOfWeek2["WED"] = "WED";
  DayOfWeek2["THU"] = "THU";
  DayOfWeek2["FRI"] = "FRI";
  DayOfWeek2["SAT"] = "SAT";
  return DayOfWeek2;
})(DayOfWeek || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNSPECIFIED_DISH_OPTION_TYPE"] = "UNSPECIFIED_DISH_OPTION_TYPE";
  Type2["SELECTION"] = "SELECTION";
  Type2["EXTRAS"] = "EXTRAS";
  Type2["DESELECTION"] = "DESELECTION";
  return Type2;
})(Type || {});
var ItemType = /* @__PURE__ */ ((ItemType2) => {
  ItemType2["UNSPECIFIED_ITEM_TYPE"] = "UNSPECIFIED_ITEM_TYPE";
  ItemType2["DISH"] = "DISH";
  ItemType2["VARIATION"] = "VARIATION";
  return ItemType2;
})(ItemType || {});
var DiscountType = /* @__PURE__ */ ((DiscountType2) => {
  DiscountType2["UNSPECIFIED_TYPE"] = "UNSPECIFIED_TYPE";
  DiscountType2["OFF_ITEM"] = "OFF_ITEM";
  DiscountType2["OFF_ORDER"] = "OFF_ORDER";
  return DiscountType2;
})(DiscountType || {});
var DiscountPlatform = /* @__PURE__ */ ((DiscountPlatform2) => {
  DiscountPlatform2["UNSPECIFIED_PLATFORM"] = "UNSPECIFIED_PLATFORM";
  DiscountPlatform2["SITE"] = "SITE";
  DiscountPlatform2["MOBILE_SITE"] = "MOBILE_SITE";
  DiscountPlatform2["CALL_CENTER"] = "CALL_CENTER";
  return DiscountPlatform2;
})(DiscountPlatform || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/restaurants-catalogs-v3-catalog-catalogs.meta.ts
function createDraftCatalog2() {
  const payload = { catalogId: ":catalogId" };
  const getRequestOptions = createDraftCatalog(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/draft-catalogs/{catalogId}/create",
    pathParams: { catalogId: "catalogId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function publishDraftCatalog2() {
  const payload = { catalogId: ":catalogId" };
  const getRequestOptions = publishDraftCatalog(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/draft-catalogs/{catalogId}/publish",
    pathParams: { catalogId: "catalogId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function discardDraftCatalog2() {
  const payload = { catalogId: ":catalogId" };
  const getRequestOptions = discardDraftCatalog(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v3/draft-catalogs/{catalogId}/discard",
    pathParams: { catalogId: "catalogId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkCreateVariations2() {
  const payload = { catalogId: ":catalogId" };
  const getRequestOptions = bulkCreateVariations(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/draft-catalogs/{catalogId}/bulk/variations/create",
    pathParams: { catalogId: "catalogId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateItems2() {
  const payload = { catalogId: ":catalogId" };
  const getRequestOptions = bulkUpdateItems(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/draft-catalogs/{catalogId}/bulk/items/update",
    pathParams: { catalogId: "catalogId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkCreateDishes2() {
  const payload = { catalogId: ":catalogId" };
  const getRequestOptions = bulkCreateDishes(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/draft-catalogs/{catalogId}/bulk/dishes/create",
    pathParams: { catalogId: "catalogId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkCreateMenus2() {
  const payload = { catalogId: ":catalogId" };
  const getRequestOptions = bulkCreateMenus(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/draft-catalogs/{catalogId}/bulk/menus/create",
    pathParams: { catalogId: "catalogId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateMenus2() {
  const payload = { catalogId: ":catalogId" };
  const getRequestOptions = bulkUpdateMenus(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/draft-catalogs/{catalogId}/bulk/menus/update",
    pathParams: { catalogId: "catalogId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkCreateSections2() {
  const payload = { catalogId: ":catalogId" };
  const getRequestOptions = bulkCreateSections(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/draft-catalogs/{catalogId}/bulk/sections/create",
    pathParams: { catalogId: "catalogId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateSections2() {
  const payload = { catalogId: ":catalogId" };
  const getRequestOptions = bulkUpdateSections(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/draft-catalogs/{catalogId}/bulk/sections/update",
    pathParams: { catalogId: "catalogId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkArchiveMenus2() {
  const payload = { catalogId: ":catalogId" };
  const getRequestOptions = bulkArchiveMenus(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/draft-catalogs/{catalogId}/bulk/menus/archive",
    pathParams: { catalogId: "catalogId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUnarchiveMenus2() {
  const payload = { catalogId: ":catalogId" };
  const getRequestOptions = bulkUnarchiveMenus(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/draft-catalogs/{catalogId}/bulk/menus/unarchive",
    pathParams: { catalogId: "catalogId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listCatalogs2() {
  const payload = {};
  const getRequestOptions = listCatalogs(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/catalogs",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getMenu2() {
  const payload = { catalogId: ":catalogId", menuId: ":menuId" };
  const getRequestOptions = getMenu(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/catalogs/{catalogId}/menus/{menuId}",
    pathParams: { catalogId: "catalogId", menuId: "menuId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getSection2() {
  const payload = {
    catalogId: ":catalogId",
    menuId: ":menuId",
    sectionId: ":sectionId"
  };
  const getRequestOptions = getSection(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/catalogs/{catalogId}/menus/{menuId}/sections/{sectionId}",
    pathParams: {
      catalogId: "catalogId",
      menuId: "menuId",
      sectionId: "sectionId"
    },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateMenu2() {
  const payload = { catalogId: ":catalogId", menu: { id: ":menuId" } };
  const getRequestOptions = updateMenu(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v3/catalogs/{catalogId}/menus/{menu.id}",
    pathParams: { catalogId: "catalogId", menuId: "menuId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateSection2() {
  const payload = {
    catalogId: ":catalogId",
    menuId: ":menuId",
    section: { id: ":sectionId" }
  };
  const getRequestOptions = updateSection(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v3/catalogs/{catalogId}/menus/{menuId}/sections/{section.id}",
    pathParams: {
      catalogId: "catalogId",
      menuId: "menuId",
      sectionId: "sectionId"
    },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listMenus2() {
  const payload = { catalogId: ":catalogId" };
  const getRequestOptions = listMenus(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/catalogs/{catalogId}/menus",
    pathParams: { catalogId: "catalogId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listSections2() {
  const payload = { catalogId: ":catalogId" };
  const getRequestOptions = listSections(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/catalogs/{catalogId}/sections",
    pathParams: { catalogId: "catalogId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createMenu2() {
  const payload = { catalogId: ":catalogId" };
  const getRequestOptions = createMenu(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/catalogs/{catalogId}/menus",
    pathParams: { catalogId: "catalogId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function archiveMenu2() {
  const payload = { catalogId: ":catalogId", menuId: ":menuId" };
  const getRequestOptions = archiveMenu(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/catalogs/{catalogId}/menus/{menuId}/archive",
    pathParams: { catalogId: "catalogId", menuId: "menuId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function unarchiveMenu2() {
  const payload = { catalogId: ":catalogId", menuId: ":menuId" };
  const getRequestOptions = unarchiveMenu(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/catalogs/{catalogId}/menus/{menuId}/unarchive",
    pathParams: { catalogId: "catalogId", menuId: "menuId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createSection2() {
  const payload = { catalogId: ":catalogId", menuId: ":menuId" };
  const getRequestOptions = createSection(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/catalogs/{catalogId}/menus/{menuId}/sections",
    pathParams: { catalogId: "catalogId", menuId: "menuId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getItem2() {
  const payload = { catalogId: ":catalogId", itemId: ":itemId" };
  const getRequestOptions = getItem(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/catalogs/{catalogId}/items/{itemId}",
    pathParams: { catalogId: "catalogId", itemId: "itemId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listItems2() {
  const payload = { catalogId: ":catalogId" };
  const getRequestOptions = listItems(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/catalogs/{catalogId}/items",
    pathParams: { catalogId: "catalogId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateItem2() {
  const payload = { catalogId: ":catalogId", item: { id: ":itemId" } };
  const getRequestOptions = updateItem(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v3/catalogs/{catalogId}/items/{item.id}",
    pathParams: { catalogId: "catalogId", itemId: "itemId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createDish2() {
  const payload = {
    catalogId: ":catalogId",
    menuId: ":menuId",
    sectionId: ":sectionId"
  };
  const getRequestOptions = createDish(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/catalogs/{catalogId}/menus/{menuId}/sections/{sectionId}/items",
    pathParams: {
      catalogId: "catalogId",
      menuId: "menuId",
      sectionId: "sectionId"
    },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createVariation2() {
  const payload = { catalogId: ":catalogId" };
  const getRequestOptions = createVariation(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/catalogs/{catalogId}/variations",
    pathParams: { catalogId: "catalogId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  DayOfWeekOriginal,
  DiscountPlatformOriginal,
  DiscountTypeOriginal,
  FulfillmentTypeOriginal,
  ItemTypeOriginal,
  PlatformOriginal,
  SortOrderOriginal,
  TypeOriginal,
  WebhookIdentityTypeOriginal,
  archiveMenu,
  bulkArchiveMenus,
  bulkCreateDishes,
  bulkCreateMenus,
  bulkCreateSections,
  bulkCreateVariations,
  bulkUnarchiveMenus,
  bulkUpdateItems,
  bulkUpdateMenus,
  bulkUpdateSections,
  createDish,
  createDraftCatalog,
  createMenu,
  createSection,
  createVariation,
  discardDraftCatalog,
  getItem,
  getMenu,
  getSection,
  listCatalogs,
  listItems,
  listMenus,
  listSections,
  publishDraftCatalog,
  unarchiveMenu,
  updateItem,
  updateMenu,
  updateSection
});
