// @wix/auto_sdk_restaurants_pos-tpa/build/internal/es/index.mjs
// src/restaurants-postpaapi-v1-configuration-pos-tpa.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/restaurants-postpaapi-v1-configuration-pos-tpa.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressRestaurantsPostpaapiRestaurantsPosTpaApiUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/restaurants/v1/pos-spi-host",
        destPath: "/v1/pos-spi-host"
      }
    ],
    "com.wixpress.restaurants.restaurants-pos-spi-host.wixapps.net": [
      {
        srcPath: "/restaurants-pos-spi-host",
        destPath: "/v1/pos-spi-host"
      }
    ],
    "www.wixapps.com": [
      {
        srcPath: "/restaurants/v1/pos-integration",
        destPath: "/v1/pos-integration"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/v1/pos-integration",
        destPath: "/v1/pos-integration"
      }
    ],
    "users._base_domain": [
      {
        srcPath: "/_api/restaurants/v1/pos-spi-host",
        destPath: "/v1/pos-spi-host"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_pos-tpa";
function getConfiguration(payload) {
  function __getConfiguration({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.postpaapi.v1.configuration",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.postpaapi.RestaurantsPosTpaApi.GetConfiguration",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsPostpaapiRestaurantsPosTpaApiUrl({
        protoPath: "/v1/pos-integration/configurations/{id}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getConfiguration;
}
function listConfigurations(payload) {
  function __listConfigurations({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.postpaapi.v1.configuration",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.postpaapi.RestaurantsPosTpaApi.ListConfigurations",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsPostpaapiRestaurantsPosTpaApiUrl({
        protoPath: "/v1/pos-integration/configurations",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __listConfigurations;
}
function disconnectAccount(payload) {
  function __disconnectAccount({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.postpaapi.v1.configuration",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.postpaapi.RestaurantsPosTpaApi.DisconnectAccount",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsPostpaapiRestaurantsPosTpaApiUrl({
        protoPath: "/v1/pos-integration/accounts/{accountId}/disconnect",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __disconnectAccount;
}
function disconnectConfiguration(payload) {
  function __disconnectConfiguration({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.postpaapi.v1.configuration",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.postpaapi.RestaurantsPosTpaApi.DisconnectConfiguration",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsPostpaapiRestaurantsPosTpaApiUrl({
        protoPath: "/v1/pos-integration/configurations/{configurationId}/disconnect",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __disconnectConfiguration;
}

// src/restaurants-postpaapi-v1-configuration-pos-tpa.universal.ts
async function getConfiguration2(_id) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ id: _id });
  const reqOpts = getConfiguration(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.configuration;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { id: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listConfigurations2() {
  const { httpClient, sideEffects } = arguments[0];
  const payload = renameKeysFromSDKRequestToRESTRequest({});
  const reqOpts = listConfigurations(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {},
        singleArgumentUnchanged: false
      },
      []
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function disconnectAccount2(accountId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    accountId,
    appDefId: options?.appDefId
  });
  const reqOpts = disconnectAccount(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          accountId: "$[0]",
          appDefId: "$[1].appDefId"
        },
        singleArgumentUnchanged: false
      },
      ["accountId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function disconnectConfiguration2(configurationId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    configurationId
  });
  const reqOpts = disconnectConfiguration(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { configurationId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["configurationId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/restaurants-postpaapi-v1-configuration-pos-tpa.public.ts
function getConfiguration3(httpClient) {
  return (_id) => getConfiguration2(
    _id,
    // @ts-ignore
    { httpClient }
  );
}
function listConfigurations3(httpClient) {
  return () => listConfigurations2(
    // @ts-ignore
    { httpClient }
  );
}
function disconnectAccount3(httpClient) {
  return (accountId, options) => disconnectAccount2(
    accountId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function disconnectConfiguration3(httpClient) {
  return (configurationId) => disconnectConfiguration2(
    configurationId,
    // @ts-ignore
    { httpClient }
  );
}

// src/restaurants-postpaapi-v1-configuration-pos-tpa.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
var getConfiguration4 = /* @__PURE__ */ createRESTModule(getConfiguration3);
var listConfigurations4 = /* @__PURE__ */ createRESTModule(listConfigurations3);
var disconnectAccount4 = /* @__PURE__ */ createRESTModule(disconnectAccount3);
var disconnectConfiguration4 = /* @__PURE__ */ createRESTModule(disconnectConfiguration3);
export {
  disconnectAccount4 as disconnectAccount,
  disconnectConfiguration4 as disconnectConfiguration,
  getConfiguration4 as getConfiguration,
  listConfigurations4 as listConfigurations
};
