// @wix/auto_sdk_bookings_event-time-slots/build/internal/es/meta.mjs
// src/bookings-availability-v2-time-slot-event-time-slots.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressBookingsAvailabilityV2EventTimeSlotsUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/_api/service-availability/v2/time-slots",
        destPath: "/v2/time-slots"
      },
      {
        srcPath: "/_api/service-availability/v2/multi-service-time-slots",
        destPath: "/v2/multi-service-time-slots"
      },
      {
        srcPath: "/service-availability/v2/time-slots",
        destPath: "/v2/time-slots"
      },
      {
        srcPath: "/service-availability/v2/multi-service-time-slots",
        destPath: "/v2/multi-service-time-slots"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/service-availability/v2/time-slots",
        destPath: "/v2/time-slots"
      },
      {
        srcPath: "/_api/service-availability/v2/multi-service-time-slots",
        destPath: "/v2/multi-service-time-slots"
      },
      {
        srcPath: "/service-availability/v2/time-slots",
        destPath: "/v2/time-slots"
      },
      {
        srcPath: "/service-availability/v2/multi-service-time-slots",
        destPath: "/v2/multi-service-time-slots"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/service-availability/v2/time-slots",
        destPath: "/v2/time-slots"
      },
      {
        srcPath: "/_api/service-availability/v2/multi-service-time-slots",
        destPath: "/v2/multi-service-time-slots"
      },
      {
        srcPath: "/service-availability/v2/multi-service-time-slots",
        destPath: "/v2/multi-service-time-slots"
      },
      {
        srcPath: "/service-availability/v2/time-slots",
        destPath: "/v2/time-slots"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/service-availability/v2/time-slots",
        destPath: "/v2/time-slots"
      },
      {
        srcPath: "/_api/service-availability/v2/multi-service-time-slots",
        destPath: "/v2/multi-service-time-slots"
      }
    ],
    _: [
      {
        srcPath: "/_api/service-availability/v2/time-slots",
        destPath: "/v2/time-slots"
      },
      {
        srcPath: "/_api/service-availability/v2/multi-service-time-slots",
        destPath: "/v2/multi-service-time-slots"
      },
      {
        srcPath: "/_api/service-availability/v2/time-slots/event",
        destPath: "/v2/time-slots/event"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/service-availability/v2/multi-service-time-slots",
        destPath: "/v2/multi-service-time-slots"
      },
      {
        srcPath: "/_api/service-availability/v2/time-slots",
        destPath: "/v2/time-slots"
      },
      {
        srcPath: "/_api/service-availability/v2/time-slots/event",
        destPath: "/v2/time-slots/event"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/service-availability/v2/multi-service-time-slots",
        destPath: "/v2/multi-service-time-slots"
      },
      {
        srcPath: "/_api/service-availability/v2/time-slots",
        destPath: "/v2/time-slots"
      },
      {
        srcPath: "/_api/service-availability/v2/time-slots/event",
        destPath: "/v2/time-slots/event"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/service-availability/v2/multi-service-time-slots",
        destPath: "/v2/multi-service-time-slots"
      },
      {
        srcPath: "/_api/service-availability/v2/time-slots",
        destPath: "/v2/time-slots"
      },
      {
        srcPath: "/_api/service-availability/v2/time-slots/event",
        destPath: "/v2/time-slots/event"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/service-availability/v2/time-slots/event",
        destPath: "/v2/time-slots/event"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_bookings_event-time-slots";
function listEventTimeSlots(payload) {
  function __listEventTimeSlots({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "bookingPolicyViolations.earliestBookingDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.bookings.availability.v2.time_slot",
      method: "POST",
      methodFqn: "com.wixpress.bookings.availability.v2.EventTimeSlots.ListEventTimeSlots",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressBookingsAvailabilityV2EventTimeSlotsUrl({
        protoPath: "/v2/time-slots/event",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "timeSlots.bookingPolicyViolations.earliestBookingDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listEventTimeSlots;
}
function getEventTimeSlot(payload) {
  function __getEventTimeSlot({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.availability.v2.time_slot",
      method: "GET",
      methodFqn: "com.wixpress.bookings.availability.v2.EventTimeSlots.GetEventTimeSlot",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressBookingsAvailabilityV2EventTimeSlotsUrl({
        protoPath: "/v2/time-slots/event/{eventId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "timeSlot.bookingPolicyViolations.earliestBookingDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getEventTimeSlot;
}

// src/bookings-availability-v2-time-slot-event-time-slots.types.ts
var LocationType = /* @__PURE__ */ ((LocationType2) => {
  LocationType2["UNKNOWN_LOCATION_TYPE"] = "UNKNOWN_LOCATION_TYPE";
  LocationType2["BUSINESS"] = "BUSINESS";
  LocationType2["CUSTOM"] = "CUSTOM";
  LocationType2["CUSTOMER"] = "CUSTOMER";
  return LocationType2;
})(LocationType || {});

// src/bookings-availability-v2-time-slot-event-time-slots.meta.ts
function listEventTimeSlots2() {
  const payload = {};
  const getRequestOptions = listEventTimeSlots(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/time-slots/event",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getEventTimeSlot2() {
  const payload = { eventId: ":eventId" };
  const getRequestOptions = getEventTimeSlot(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/time-slots/event/{eventId}",
    pathParams: { eventId: "eventId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  LocationType as LocationTypeOriginal,
  getEventTimeSlot2 as getEventTimeSlot,
  listEventTimeSlots2 as listEventTimeSlots
};
