// @wix/headless-events/cjs/dist/react/core/InvoiceItem.js
import { jsx as _jsx } from "react/jsx-runtime";
import { useService, WixServices } from '@wix/services-manager-react';
import { createServicesMap } from '@wix/services-manager';
import { InvoiceItemService, InvoiceItemServiceDefinition, } from '../../services/invoice-item-service.js';
import { formatPrice } from '../../utils/price.js';
/**
 * InvoiceItem Root core component that provides invoice item service context.
 *
 * @component
 */
export function Root(props) {
    const { children, invoiceItem } = props;
    const invoiceItemServiceConfig = {
        invoiceItem,
    };
    return (_jsx(WixServices, { servicesMap: createServicesMap().addService(InvoiceItemServiceDefinition, InvoiceItemService, invoiceItemServiceConfig), children: children }));
}
/**
 * InvoiceItem Name core component that provides invoice item name.
 *
 * @component
 */
export function Name(props) {
    const invoiceItemService = useService(InvoiceItemServiceDefinition);
    const invoiceItem = invoiceItemService.invoiceItem.get();
    const name = invoiceItem.name;
    return props.children({ name });
}
/**
 * InvoiceItem Price core component that provides invoice item price.
 *
 * @component
 */
export function Price(props) {
    const invoiceItemService = useService(InvoiceItemServiceDefinition);
    const invoiceItem = invoiceItemService.invoiceItem.get();
    const price = invoiceItem.price;
    const value = Number(price.value);
    const currency = price.currency;
    return props.children({
        value,
        currency,
        formattedValue: formatPrice(value, currency, props.locale),
    });
}
/**
 * InvoiceItem Quantity core component that provides invoice item quantity.
 *
 * @component
 */
export function Quantity(props) {
    const invoiceItemService = useService(InvoiceItemServiceDefinition);
    const invoiceItem = invoiceItemService.invoiceItem.get();
    const quantity = invoiceItem.quantity;
    return props.children({ quantity });
}
/**
 * InvoiceItem Total core component that provides invoice item total.
 *
 * @component
 */
export function Total(props) {
    const invoiceItemService = useService(InvoiceItemServiceDefinition);
    const invoiceItem = invoiceItemService.invoiceItem.get();
    const total = invoiceItem.total;
    const value = Number(total.value);
    const currency = total.currency;
    return props.children({
        value,
        currency,
        formattedValue: formatPrice(value, currency, props.locale),
    });
}
