// @wix/headless-events/cjs/dist/react/core/Order.js
import { jsx as _jsx } from "react/jsx-runtime";
import { useService, WixServices } from '@wix/services-manager-react';
import { createServicesMap } from '@wix/services-manager';
import { isOrderReady, OrderService, OrderServiceDefinition, } from '../../services/order-service.js';
import { formatMonthDayYear } from '../../utils/date.js';
import { formatPrice } from '../../utils/price.js';
/**
 * Order Root core component that provides order service context.
 *
 * @component
 */
export function Root(props) {
    const { children, orderServiceConfig } = props;
    return (_jsx(WixServices, { servicesMap: createServicesMap().addService(OrderServiceDefinition, OrderService, orderServiceConfig), children: children }));
}
/**
 * Order OrderNumber core component that provides order number.
 *
 * @component
 */
export function OrderNumber(props) {
    const orderService = useService(OrderServiceDefinition);
    const order = orderService.order.get();
    if (!order) {
        return null;
    }
    const orderNumber = order.orderNumber;
    return props.children({ orderNumber });
}
/**
 * Order GuestEmail core component that provides guest email.
 *
 * @component
 */
export function GuestEmail(props) {
    const orderService = useService(OrderServiceDefinition);
    const order = orderService.order.get();
    if (!order) {
        return null;
    }
    const guestEmail = order.email;
    return props.children({ guestEmail });
}
/**
 * Order CreatedDate core component that provides created date.
 *
 * @component
 */
export function CreatedDate(props) {
    const orderService = useService(OrderServiceDefinition);
    const order = orderService.order.get();
    if (!order) {
        return null;
    }
    const createdDate = new Date(order.created);
    const formattedDate = formatMonthDayYear(createdDate, undefined, props.locale);
    return props.children({ createdDate, formattedDate });
}
/**
 * Order DownloadTicketsButton core component that provides tickets PDF URL.
 *
 * @component
 */
export function DownloadTicketsButton(props) {
    const orderService = useService(OrderServiceDefinition);
    const order = orderService.order.get();
    if (!order) {
        return null;
    }
    const ticketsPdfUrl = order.ticketsPdf;
    const isPolling = orderService.isPolling.get();
    const isReady = isOrderReady(order);
    if (!isReady || isPolling) {
        return null;
    }
    return props.children({ ticketsPdfUrl });
}
/**
 * Order InvoiceItems core component that provides invoice items.
 *
 * @component
 */
export function InvoiceItems(props) {
    const orderService = useService(OrderServiceDefinition);
    const order = orderService.order.get();
    if (!order) {
        return null;
    }
    const invoiceItems = order.invoice.items;
    return props.children({ invoiceItems });
}
/**
 * Order InvoiceItemRepeater core component that provides invoice items.
 *
 * @component
 */
export function InvoiceItemRepeater(props) {
    const orderService = useService(OrderServiceDefinition);
    const order = orderService.order.get();
    if (!order) {
        return null;
    }
    const invoiceItems = order.invoice.items;
    return props.children({ invoiceItems });
}
/**
 * Order Subtotal core component that provides order subtotal.
 *
 * @component
 */
export function Subtotal(props) {
    const orderService = useService(OrderServiceDefinition);
    const order = orderService.order.get();
    if (!order) {
        return null;
    }
    const subtotal = order.invoice.subTotal;
    const value = Number(subtotal.value);
    const currency = subtotal.currency;
    return props.children({
        value,
        currency,
        formattedValue: formatPrice(value, currency, props.locale),
    });
}
/**
 * Order PaidPlanDiscount core component that provides paid plan discount.
 *
 * @component
 */
export function PaidPlanDiscount(props) {
    const orderService = useService(OrderServiceDefinition);
    const order = orderService.order.get();
    if (!order) {
        return null;
    }
    const discounts = order.invoice.discount?.discounts;
    const paidPlanDiscount = discounts?.find((item) => item.paidPlan);
    if (!paidPlanDiscount) {
        return null;
    }
    const value = Number(paidPlanDiscount.amount.value);
    const currency = paidPlanDiscount.amount.currency;
    return props.children({
        value,
        currency,
        formattedValue: formatPrice(value, currency, props.locale),
        rate: Number(paidPlanDiscount.paidPlan.percentDiscount.rate),
    });
}
/**
 * Order CouponDiscount core component that provides coupon discount.
 *
 * @component
 */
export function CouponDiscount(props) {
    const orderService = useService(OrderServiceDefinition);
    const order = orderService.order.get();
    if (!order) {
        return null;
    }
    const discounts = order.invoice.discount?.discounts;
    const couponDiscount = discounts?.find((item) => item.coupon);
    if (!couponDiscount) {
        return null;
    }
    const value = Number(couponDiscount.amount.value);
    const currency = couponDiscount.amount.currency;
    return props.children({
        value,
        currency,
        formattedValue: formatPrice(value, currency, props.locale),
    });
}
/**
 * Order Tax core component that provides order tax.
 *
 * @component
 */
export function Tax(props) {
    const orderService = useService(OrderServiceDefinition);
    const order = orderService.order.get();
    if (!order) {
        return null;
    }
    const tax = order.invoice.tax;
    if (!tax) {
        return null;
    }
    const value = Number(tax.amount.value);
    const currency = tax.amount.currency;
    return props.children({
        value,
        currency,
        formattedValue: formatPrice(value, currency, props.locale),
        rate: Number(tax.rate),
        name: tax.name,
    });
}
/**
 * Order Fee core component that provides order fee.
 *
 * @component
 */
export function Fee(props) {
    const orderService = useService(OrderServiceDefinition);
    const order = orderService.order.get();
    if (!order) {
        return null;
    }
    const fees = order.invoice.fees;
    const addedFee = fees?.find((fee) => fee.type === 'FEE_ADDED_AT_CHECKOUT');
    if (!addedFee) {
        return null;
    }
    const value = Number(addedFee.amount.value);
    const currency = addedFee.amount.currency;
    return props.children({
        value,
        currency,
        formattedValue: formatPrice(value, currency, props.locale),
        rate: Number(addedFee.rate),
    });
}
/**
 * Order Total core component that provides order total.
 *
 * @component
 */
export function Total(props) {
    const orderService = useService(OrderServiceDefinition);
    const order = orderService.order.get();
    if (!order) {
        return null;
    }
    const total = order.invoice.grandTotal;
    const value = Number(total.value);
    const currency = total.currency;
    return props.children({
        value,
        currency,
        formattedValue: formatPrice(value, currency, props.locale),
    });
}
