// @wix/headless-events/cjs/dist/react/core/ScheduleList.js
import { jsx as _jsx } from "react/jsx-runtime";
import { useService, WixServices } from '@wix/services-manager-react';
import { createServicesMap } from '@wix/services-manager';
import { ScheduleListService, ScheduleListServiceDefinition, } from '../../services/schedule-list-service.js';
import { STAGES_FILTER_KEY, TAGS_FILTER_KEY } from '../../constants.js';
/**
 * ScheduleList Root core component that provides schedule list service data.
 *
 * @component
 */
export function Root(props) {
    const { children, scheduleListServiceConfig } = props;
    return (_jsx(WixServices, { servicesMap: createServicesMap().addService(ScheduleListServiceDefinition, ScheduleListService, scheduleListServiceConfig), children: children }));
}
/**
 * ScheduleList Items core component that provides schedule list data.
 *
 * @component
 */
export function Items(props) {
    const scheduleListService = useService(ScheduleListServiceDefinition);
    const items = scheduleListService.items.get();
    if (!items.length) {
        return null;
    }
    return props.children({ items });
}
/**
 * ScheduleList ItemRepeater core component that provides schedule item list data. Not rendered if there are no items.
 *
 * @component
 */
export function ItemRepeater(props) {
    const scheduleListService = useService(ScheduleListServiceDefinition);
    const items = scheduleListService.items.get();
    if (!items.length) {
        return null;
    }
    return props.children({ items });
}
/**
 * ScheduleList Groups core component that provides grouped schedule items data.
 *
 * @component
 */
export function Groups(props) {
    const scheduleListService = useService(ScheduleListServiceDefinition);
    const itemsGroups = scheduleListService.itemsGroups.get();
    return props.children({ itemsGroups });
}
/**
 * ScheduleList GroupRepeater core component that provides grouped schedule items data.
 * Not rendered if there are no groups.
 *
 * @component
 */
export function GroupRepeater(props) {
    const scheduleListService = useService(ScheduleListServiceDefinition);
    const itemsGroups = scheduleListService.itemsGroups.get();
    if (!itemsGroups.length) {
        return null;
    }
    return props.children({ itemsGroups });
}
/**
 * ScheduleList Filters core component that provides schedule list filter data.
 *
 * @component
 */
export function Filters(props) {
    const scheduleListService = useService(ScheduleListServiceDefinition);
    const tags = scheduleListService.tags.get();
    const stageNames = scheduleListService.stageNames.get();
    const currentTagFilters = scheduleListService.tagFilters.get();
    const currentStageFilter = scheduleListService.stageFilter.get();
    const onChange = async (value) => {
        const stageFilter = value['stage'] !== props.allStagesLabel ? value['stage'] : null;
        const tagFilters = value['tag']?.$in || value['tag'] || [];
        scheduleListService.setStageFilter(stageFilter);
        scheduleListService.setTagFilters(tagFilters);
    };
    const { filterOptions, value } = buildFilterProps(tags, currentTagFilters, currentStageFilter, stageNames, props.allStagesLabel);
    if (!stageNames.length && !tags.length) {
        return null;
    }
    return props.children({
        filterOptions,
        value,
        onChange,
    });
}
const buildFilterProps = (tags, currentTagFilters, currentStageFilter, stageNames, allStagesLabel) => {
    const FILTER_BASE = {
        label: '',
        displayType: 'text',
    };
    const filterOptions = [
        {
            ...FILTER_BASE,
            key: STAGES_FILTER_KEY,
            type: 'single',
            fieldName: 'stage',
            validValues: [allStagesLabel, ...stageNames],
        },
        {
            ...FILTER_BASE,
            key: TAGS_FILTER_KEY,
            type: 'multi',
            fieldName: 'tag',
            validValues: tags,
        },
    ];
    const value = {
        tag: currentTagFilters,
        stage: currentStageFilter || allStagesLabel,
    };
    return {
        filterOptions,
        value,
    };
};
