// @wix/headless-events/cjs/dist/react/core/TicketsPicker.js
import { jsx as _jsx } from "react/jsx-runtime";
import { useService, WixServices } from '@wix/services-manager-react';
import { createServicesMap } from '@wix/services-manager';
import { getTicketReservationTotals, TicketDefinitionListService, TicketDefinitionListServiceDefinition, } from '../../services/ticket-definition-list-service.js';
import { EventServiceDefinition, EventService, } from '../../services/event-service.js';
import { CheckoutServiceDefinition, CheckoutService, } from '../../services/checkout-service.js';
import { WIX_FEE_RATE } from '../../constants.js';
/**
 * TicketsPicker Root core component that provides necessary services context.
 *
 * @component
 */
export function Root(props) {
    const { children, eventServiceConfig, ticketDefinitionListServiceConfig, checkoutServiceConfig, } = props;
    return (_jsx(WixServices, { servicesMap: createServicesMap()
            .addService(EventServiceDefinition, EventService, eventServiceConfig)
            .addService(TicketDefinitionListServiceDefinition, TicketDefinitionListService, ticketDefinitionListServiceConfig)
            .addService(CheckoutServiceDefinition, CheckoutService, checkoutServiceConfig), children: children }));
}
/**
 * TicketsPicker TicketDefinitions core component that provides ticket definitions data.
 *
 * @component
 */
export function TicketDefinitions(props) {
    const ticketDefinitionListService = useService(TicketDefinitionListServiceDefinition);
    const ticketDefinitions = ticketDefinitionListService.ticketDefinitions.get();
    const hasTicketDefinitions = !!ticketDefinitions.length;
    return props.children({ ticketDefinitions, hasTicketDefinitions });
}
/**
 * TicketsPicker TicketDefinitionRepeater core component that provides ticket definitions. Not rendered if there are no ticket definitions.
 *
 * @component
 */
export function TicketDefinitionRepeater(props) {
    const ticketDefinitionListService = useService(TicketDefinitionListServiceDefinition);
    const ticketDefinitions = ticketDefinitionListService.ticketDefinitions.get();
    const hasTicketDefinitions = !!ticketDefinitions.length;
    if (!hasTicketDefinitions) {
        return null;
    }
    return props.children({ ticketDefinitions });
}
/**
 * TicketsPicker Totals core component that provides totals data.
 *
 * @component
 */
export function Totals(props) {
    const eventService = useService(EventServiceDefinition);
    const ticketDefinitionListService = useService(TicketDefinitionListServiceDefinition);
    const event = eventService.event.get();
    const ticketDefinitions = ticketDefinitionListService.ticketDefinitions.get();
    const selectedQuantities = ticketDefinitionListService.selectedQuantities.get();
    if (!ticketDefinitions.length) {
        return null;
    }
    const { total, subtotal, tax, fee, currency, formattedTotal, formattedSubtotal, formattedTax, formattedFee, } = getTicketReservationTotals(event, ticketDefinitions, selectedQuantities, props.locale);
    const taxSettings = event.registration?.tickets?.taxSettings;
    return props.children({
        total,
        subtotal,
        tax,
        fee,
        currency,
        formattedTotal,
        formattedSubtotal,
        formattedTax,
        formattedFee,
        taxName: taxSettings ? taxSettings.name : null,
        taxRate: taxSettings ? Number(taxSettings.rate) : null,
        taxIncluded: taxSettings?.type === 'INCLUDED_IN_PRICE',
        feeRate: WIX_FEE_RATE,
    });
}
/**
 * TicketsPicker CheckoutError core component that provides checkout error. Not rendered if there is no error.
 *
 * @component
 */
export function CheckoutError(props) {
    const checkoutService = useService(CheckoutServiceDefinition);
    const error = checkoutService.error.get();
    if (!error) {
        return null;
    }
    return props.children({ error });
}
/**
 * TicketsPicker CheckoutTrigger core component that provides checkout functionality. Not rendered if there are no ticket definitions.
 *
 * @component
 */
export function CheckoutTrigger(props) {
    const ticketDefinitionListService = useService(TicketDefinitionListServiceDefinition);
    const eventService = useService(EventServiceDefinition);
    const checkoutService = useService(CheckoutServiceDefinition);
    const event = eventService.event.get();
    const ticketDefinitions = ticketDefinitionListService.ticketDefinitions.get();
    const selectedQuantities = ticketDefinitionListService.selectedQuantities.get();
    const isLoading = checkoutService.isLoading.get();
    const error = checkoutService.error.get();
    const filteredSelectedQuantities = selectedQuantities.filter((selectedQuantity) => !!selectedQuantity.quantity);
    const hasTicketDefinitions = !!ticketDefinitions.length;
    const hasSelectedTicketDefinitions = !!filteredSelectedQuantities.length;
    const checkout = () => checkoutService.checkout(event._id, event.slug, filteredSelectedQuantities);
    if (!hasTicketDefinitions) {
        return null;
    }
    return props.children({
        isLoading,
        error,
        hasSelectedTicketDefinitions,
        checkout,
    });
}
