// @wix/auto_sdk_media_files/build/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  ContentDisposition: () => ContentDisposition,
  IdentityType: () => IdentityType,
  ImageAnnotationType: () => ImageAnnotationType,
  MediaType: () => MediaType,
  Namespace: () => Namespace,
  OperationStatus: () => OperationStatus,
  RootFolder: () => RootFolder,
  SortOrder: () => SortOrder,
  State: () => State,
  StreamFormat: () => StreamFormat,
  UploadProtocol: () => UploadProtocol,
  WebhookIdentityType: () => WebhookIdentityType,
  bulkDeleteFiles: () => bulkDeleteFiles4,
  bulkImportFile: () => bulkImportFile4,
  bulkImportFiles: () => bulkImportFiles4,
  bulkRestoreFilesFromTrashBin: () => bulkRestoreFilesFromTrashBin4,
  generateFileDownloadUrl: () => generateFileDownloadUrl4,
  generateFileResumableUploadUrl: () => generateFileResumableUploadUrl4,
  generateFileUploadUrl: () => generateFileUploadUrl4,
  generateFilesDownloadUrl: () => generateFilesDownloadUrl4,
  generateVideoStreamingUrl: () => generateVideoStreamingUrl4,
  getFileDescriptor: () => getFileDescriptor4,
  getFileDescriptors: () => getFileDescriptors4,
  importFile: () => importFile4,
  listDeletedFiles: () => listDeletedFiles4,
  listFiles: () => listFiles4,
  onFileDescriptorDeleted: () => onFileDescriptorDeleted2,
  onFileDescriptorFileFailed: () => onFileDescriptorFileFailed2,
  onFileDescriptorFileReady: () => onFileDescriptorFileReady2,
  onFileDescriptorUpdated: () => onFileDescriptorUpdated2,
  searchFiles: () => searchFiles4,
  updateFileDescriptor: () => updateFileDescriptor4
});
module.exports = __toCommonJS(index_exports);

// src/media-site-media-v1-file-descriptor-files.public.ts
var import_rename_all_nested_keys2 = require("@wix/sdk-runtime/rename-all-nested-keys");
var import_audio3 = require("@wix/sdk-runtime/transformations/audio");
var import_document3 = require("@wix/sdk-runtime/transformations/document");
var import_float3 = require("@wix/sdk-runtime/transformations/float");
var import_image3 = require("@wix/sdk-runtime/transformations/image");
var import_timestamp3 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths3 = require("@wix/sdk-runtime/transformations/transform-paths");
var import_video_v23 = require("@wix/sdk-runtime/transformations/video-v2");
var import_sdk_types = require("@wix/sdk-types");

// src/media-site-media-v1-file-descriptor-files.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/media-site-media-v1-file-descriptor-files.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixMediaSiteMediaV1FilesServiceUrl(opts) {
  const domainToMappings = {
    "apps._base_domain_": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: "/api"
      },
      {
        srcPath: "/_api/wix-privatemedia-proxy-server/v1",
        destPath: "/v1"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/site-media/v1",
        destPath: "/v1"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "platform.rise.ai": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_media_files";
function generateFilesDownloadUrl(payload) {
  function __generateFilesDownloadUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FilesService.GenerateFilesDownloadUrl",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/files/generate-download-url",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __generateFilesDownloadUrl;
}
function generateFileDownloadUrl(payload) {
  function __generateFileDownloadUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FilesService.GenerateFileDownloadUrl",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/files/generate-file-download-url",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __generateFileDownloadUrl;
}
function getFileDescriptor(payload) {
  function __getFileDescriptor({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "GET",
      methodFqn: "com.wix.media.site_media.v1.FilesService.GetFileDescriptor",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/files/get-file-by-id",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "file.createdDate" },
            { path: "file.updatedDate" },
            { path: "file.lastUsedDate" },
            { path: "file.media.image.image.urlExpirationDate" },
            { path: "file.media.image.previewImage.urlExpirationDate" },
            { path: "file.media.video.urlExpirationDate" },
            { path: "file.media.video.resolutions.urlExpirationDate" },
            { path: "file.media.video.resolutions.poster.urlExpirationDate" },
            { path: "file.media.video.posters.urlExpirationDate" },
            { path: "file.media.audio.assets.urlExpirationDate" },
            { path: "file.media.document.urlExpirationDate" },
            { path: "file.media.document.thumbnail.urlExpirationDate" },
            { path: "file.media.vector.image.urlExpirationDate" },
            { path: "file.media.vector.previewImage.urlExpirationDate" },
            { path: "file.media.archive.urlExpirationDate" },
            { path: "file.media.model3d.urlExpirationDate" },
            { path: "file.media.model3d.thumbnail.urlExpirationDate" },
            { path: "file.media.icon.image.urlExpirationDate" },
            { path: "file.media.icon.previewImage.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "file.media.image.image.focalPoint.x" },
            { path: "file.media.image.image.focalPoint.y" },
            { path: "file.media.image.faces.confidence" },
            { path: "file.media.image.previewImage.focalPoint.x" },
            { path: "file.media.image.previewImage.focalPoint.y" },
            { path: "file.media.video.resolutions.poster.focalPoint.x" },
            { path: "file.media.video.resolutions.poster.focalPoint.y" },
            { path: "file.media.video.posters.focalPoint.x" },
            { path: "file.media.video.posters.focalPoint.y" },
            { path: "file.media.document.thumbnail.focalPoint.x" },
            { path: "file.media.document.thumbnail.focalPoint.y" },
            { path: "file.media.vector.image.focalPoint.x" },
            { path: "file.media.vector.image.focalPoint.y" },
            { path: "file.media.vector.faces.confidence" },
            { path: "file.media.vector.previewImage.focalPoint.x" },
            { path: "file.media.vector.previewImage.focalPoint.y" },
            { path: "file.media.model3d.thumbnail.focalPoint.x" },
            { path: "file.media.model3d.thumbnail.focalPoint.y" },
            { path: "file.media.icon.image.focalPoint.x" },
            { path: "file.media.icon.image.focalPoint.y" },
            { path: "file.media.icon.faces.confidence" },
            { path: "file.media.icon.previewImage.focalPoint.x" },
            { path: "file.media.icon.previewImage.focalPoint.y" }
          ]
        }
      ]),
      fallback: [
        {
          method: "GET",
          url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
            protoPath: "/v1/files/get-file-by-id",
            data: payload,
            host
          }),
          params: (0, import_rest_modules.toURLSearchParams)(payload)
        }
      ]
    };
    return metadata;
  }
  return __getFileDescriptor;
}
function getFileDescriptors(payload) {
  function __getFileDescriptors({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FilesService.GetFileDescriptors",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/files/get-files",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "files.createdDate" },
            { path: "files.updatedDate" },
            { path: "files.lastUsedDate" },
            { path: "files.media.image.image.urlExpirationDate" },
            { path: "files.media.image.previewImage.urlExpirationDate" },
            { path: "files.media.video.urlExpirationDate" },
            { path: "files.media.video.resolutions.urlExpirationDate" },
            {
              path: "files.media.video.resolutions.poster.urlExpirationDate"
            },
            { path: "files.media.video.posters.urlExpirationDate" },
            { path: "files.media.audio.assets.urlExpirationDate" },
            { path: "files.media.document.urlExpirationDate" },
            { path: "files.media.document.thumbnail.urlExpirationDate" },
            { path: "files.media.vector.image.urlExpirationDate" },
            { path: "files.media.vector.previewImage.urlExpirationDate" },
            { path: "files.media.archive.urlExpirationDate" },
            { path: "files.media.model3d.urlExpirationDate" },
            { path: "files.media.model3d.thumbnail.urlExpirationDate" },
            { path: "files.media.icon.image.urlExpirationDate" },
            { path: "files.media.icon.previewImage.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "files.media.image.image.focalPoint.x" },
            { path: "files.media.image.image.focalPoint.y" },
            { path: "files.media.image.faces.confidence" },
            { path: "files.media.image.previewImage.focalPoint.x" },
            { path: "files.media.image.previewImage.focalPoint.y" },
            { path: "files.media.video.resolutions.poster.focalPoint.x" },
            { path: "files.media.video.resolutions.poster.focalPoint.y" },
            { path: "files.media.video.posters.focalPoint.x" },
            { path: "files.media.video.posters.focalPoint.y" },
            { path: "files.media.document.thumbnail.focalPoint.x" },
            { path: "files.media.document.thumbnail.focalPoint.y" },
            { path: "files.media.vector.image.focalPoint.x" },
            { path: "files.media.vector.image.focalPoint.y" },
            { path: "files.media.vector.faces.confidence" },
            { path: "files.media.vector.previewImage.focalPoint.x" },
            { path: "files.media.vector.previewImage.focalPoint.y" },
            { path: "files.media.model3d.thumbnail.focalPoint.x" },
            { path: "files.media.model3d.thumbnail.focalPoint.y" },
            { path: "files.media.icon.image.focalPoint.x" },
            { path: "files.media.icon.image.focalPoint.y" },
            { path: "files.media.icon.faces.confidence" },
            { path: "files.media.icon.previewImage.focalPoint.x" },
            { path: "files.media.icon.previewImage.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getFileDescriptors;
}
function updateFileDescriptor(payload) {
  function __updateFileDescriptor({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "file.createdDate" },
          { path: "file.updatedDate" },
          { path: "file.lastUsedDate" },
          { path: "file.media.image.image.urlExpirationDate" },
          { path: "file.media.image.previewImage.urlExpirationDate" },
          { path: "file.media.video.urlExpirationDate" },
          { path: "file.media.video.resolutions.urlExpirationDate" },
          { path: "file.media.video.resolutions.poster.urlExpirationDate" },
          { path: "file.media.video.posters.urlExpirationDate" },
          { path: "file.media.audio.assets.urlExpirationDate" },
          { path: "file.media.document.urlExpirationDate" },
          { path: "file.media.document.thumbnail.urlExpirationDate" },
          { path: "file.media.vector.image.urlExpirationDate" },
          { path: "file.media.vector.previewImage.urlExpirationDate" },
          { path: "file.media.archive.urlExpirationDate" },
          { path: "file.media.model3d.urlExpirationDate" },
          { path: "file.media.model3d.thumbnail.urlExpirationDate" },
          { path: "file.media.icon.image.urlExpirationDate" },
          { path: "file.media.icon.previewImage.urlExpirationDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "file.media.image.image.focalPoint.x" },
          { path: "file.media.image.image.focalPoint.y" },
          { path: "file.media.image.faces.confidence" },
          { path: "file.media.image.previewImage.focalPoint.x" },
          { path: "file.media.image.previewImage.focalPoint.y" },
          { path: "file.media.video.resolutions.poster.focalPoint.x" },
          { path: "file.media.video.resolutions.poster.focalPoint.y" },
          { path: "file.media.video.posters.focalPoint.x" },
          { path: "file.media.video.posters.focalPoint.y" },
          { path: "file.media.document.thumbnail.focalPoint.x" },
          { path: "file.media.document.thumbnail.focalPoint.y" },
          { path: "file.media.vector.image.focalPoint.x" },
          { path: "file.media.vector.image.focalPoint.y" },
          { path: "file.media.vector.faces.confidence" },
          { path: "file.media.vector.previewImage.focalPoint.x" },
          { path: "file.media.vector.previewImage.focalPoint.y" },
          { path: "file.media.model3d.thumbnail.focalPoint.x" },
          { path: "file.media.model3d.thumbnail.focalPoint.y" },
          { path: "file.media.icon.image.focalPoint.x" },
          { path: "file.media.icon.image.focalPoint.y" },
          { path: "file.media.icon.faces.confidence" },
          { path: "file.media.icon.previewImage.focalPoint.x" },
          { path: "file.media.icon.previewImage.focalPoint.y" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "PATCH",
      methodFqn: "com.wix.media.site_media.v1.FilesService.UpdateFileDescriptor",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/files/update-file-descriptor",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "file.createdDate" },
            { path: "file.updatedDate" },
            { path: "file.lastUsedDate" },
            { path: "file.media.image.image.urlExpirationDate" },
            { path: "file.media.image.previewImage.urlExpirationDate" },
            { path: "file.media.video.urlExpirationDate" },
            { path: "file.media.video.resolutions.urlExpirationDate" },
            { path: "file.media.video.resolutions.poster.urlExpirationDate" },
            { path: "file.media.video.posters.urlExpirationDate" },
            { path: "file.media.audio.assets.urlExpirationDate" },
            { path: "file.media.document.urlExpirationDate" },
            { path: "file.media.document.thumbnail.urlExpirationDate" },
            { path: "file.media.vector.image.urlExpirationDate" },
            { path: "file.media.vector.previewImage.urlExpirationDate" },
            { path: "file.media.archive.urlExpirationDate" },
            { path: "file.media.model3d.urlExpirationDate" },
            { path: "file.media.model3d.thumbnail.urlExpirationDate" },
            { path: "file.media.icon.image.urlExpirationDate" },
            { path: "file.media.icon.previewImage.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "file.media.image.image.focalPoint.x" },
            { path: "file.media.image.image.focalPoint.y" },
            { path: "file.media.image.faces.confidence" },
            { path: "file.media.image.previewImage.focalPoint.x" },
            { path: "file.media.image.previewImage.focalPoint.y" },
            { path: "file.media.video.resolutions.poster.focalPoint.x" },
            { path: "file.media.video.resolutions.poster.focalPoint.y" },
            { path: "file.media.video.posters.focalPoint.x" },
            { path: "file.media.video.posters.focalPoint.y" },
            { path: "file.media.document.thumbnail.focalPoint.x" },
            { path: "file.media.document.thumbnail.focalPoint.y" },
            { path: "file.media.vector.image.focalPoint.x" },
            { path: "file.media.vector.image.focalPoint.y" },
            { path: "file.media.vector.faces.confidence" },
            { path: "file.media.vector.previewImage.focalPoint.x" },
            { path: "file.media.vector.previewImage.focalPoint.y" },
            { path: "file.media.model3d.thumbnail.focalPoint.x" },
            { path: "file.media.model3d.thumbnail.focalPoint.y" },
            { path: "file.media.icon.image.focalPoint.x" },
            { path: "file.media.icon.image.focalPoint.y" },
            { path: "file.media.icon.faces.confidence" },
            { path: "file.media.icon.previewImage.focalPoint.x" },
            { path: "file.media.icon.previewImage.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateFileDescriptor;
}
function generateFileUploadUrl(payload) {
  function __generateFileUploadUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FilesService.GenerateFileUploadUrl",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/files/generate-upload-url",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __generateFileUploadUrl;
}
function generateFileResumableUploadUrl(payload) {
  function __generateFileResumableUploadUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FilesService.GenerateFileResumableUploadUrl",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/files/generate-resumable-upload-url",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __generateFileResumableUploadUrl;
}
function importFile(payload) {
  function __importFile({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FilesService.ImportFile",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/files/import",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "file.createdDate" },
            { path: "file.updatedDate" },
            { path: "file.lastUsedDate" },
            { path: "file.media.image.image.urlExpirationDate" },
            { path: "file.media.image.previewImage.urlExpirationDate" },
            { path: "file.media.video.urlExpirationDate" },
            { path: "file.media.video.resolutions.urlExpirationDate" },
            { path: "file.media.video.resolutions.poster.urlExpirationDate" },
            { path: "file.media.video.posters.urlExpirationDate" },
            { path: "file.media.audio.assets.urlExpirationDate" },
            { path: "file.media.document.urlExpirationDate" },
            { path: "file.media.document.thumbnail.urlExpirationDate" },
            { path: "file.media.vector.image.urlExpirationDate" },
            { path: "file.media.vector.previewImage.urlExpirationDate" },
            { path: "file.media.archive.urlExpirationDate" },
            { path: "file.media.model3d.urlExpirationDate" },
            { path: "file.media.model3d.thumbnail.urlExpirationDate" },
            { path: "file.media.icon.image.urlExpirationDate" },
            { path: "file.media.icon.previewImage.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "file.media.image.image.focalPoint.x" },
            { path: "file.media.image.image.focalPoint.y" },
            { path: "file.media.image.faces.confidence" },
            { path: "file.media.image.previewImage.focalPoint.x" },
            { path: "file.media.image.previewImage.focalPoint.y" },
            { path: "file.media.video.resolutions.poster.focalPoint.x" },
            { path: "file.media.video.resolutions.poster.focalPoint.y" },
            { path: "file.media.video.posters.focalPoint.x" },
            { path: "file.media.video.posters.focalPoint.y" },
            { path: "file.media.document.thumbnail.focalPoint.x" },
            { path: "file.media.document.thumbnail.focalPoint.y" },
            { path: "file.media.vector.image.focalPoint.x" },
            { path: "file.media.vector.image.focalPoint.y" },
            { path: "file.media.vector.faces.confidence" },
            { path: "file.media.vector.previewImage.focalPoint.x" },
            { path: "file.media.vector.previewImage.focalPoint.y" },
            { path: "file.media.model3d.thumbnail.focalPoint.x" },
            { path: "file.media.model3d.thumbnail.focalPoint.y" },
            { path: "file.media.icon.image.focalPoint.x" },
            { path: "file.media.icon.image.focalPoint.y" },
            { path: "file.media.icon.faces.confidence" },
            { path: "file.media.icon.previewImage.focalPoint.x" },
            { path: "file.media.icon.previewImage.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __importFile;
}
function bulkImportFiles(payload) {
  function __bulkImportFiles({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FilesService.BulkImportFiles",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/bulk/files/import",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "files.createdDate" },
            { path: "files.updatedDate" },
            { path: "files.lastUsedDate" },
            { path: "files.media.image.image.urlExpirationDate" },
            { path: "files.media.image.previewImage.urlExpirationDate" },
            { path: "files.media.video.urlExpirationDate" },
            { path: "files.media.video.resolutions.urlExpirationDate" },
            {
              path: "files.media.video.resolutions.poster.urlExpirationDate"
            },
            { path: "files.media.video.posters.urlExpirationDate" },
            { path: "files.media.audio.assets.urlExpirationDate" },
            { path: "files.media.document.urlExpirationDate" },
            { path: "files.media.document.thumbnail.urlExpirationDate" },
            { path: "files.media.vector.image.urlExpirationDate" },
            { path: "files.media.vector.previewImage.urlExpirationDate" },
            { path: "files.media.archive.urlExpirationDate" },
            { path: "files.media.model3d.urlExpirationDate" },
            { path: "files.media.model3d.thumbnail.urlExpirationDate" },
            { path: "files.media.icon.image.urlExpirationDate" },
            { path: "files.media.icon.previewImage.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "files.media.image.image.focalPoint.x" },
            { path: "files.media.image.image.focalPoint.y" },
            { path: "files.media.image.faces.confidence" },
            { path: "files.media.image.previewImage.focalPoint.x" },
            { path: "files.media.image.previewImage.focalPoint.y" },
            { path: "files.media.video.resolutions.poster.focalPoint.x" },
            { path: "files.media.video.resolutions.poster.focalPoint.y" },
            { path: "files.media.video.posters.focalPoint.x" },
            { path: "files.media.video.posters.focalPoint.y" },
            { path: "files.media.document.thumbnail.focalPoint.x" },
            { path: "files.media.document.thumbnail.focalPoint.y" },
            { path: "files.media.vector.image.focalPoint.x" },
            { path: "files.media.vector.image.focalPoint.y" },
            { path: "files.media.vector.faces.confidence" },
            { path: "files.media.vector.previewImage.focalPoint.x" },
            { path: "files.media.vector.previewImage.focalPoint.y" },
            { path: "files.media.model3d.thumbnail.focalPoint.x" },
            { path: "files.media.model3d.thumbnail.focalPoint.y" },
            { path: "files.media.icon.image.focalPoint.x" },
            { path: "files.media.icon.image.focalPoint.y" },
            { path: "files.media.icon.faces.confidence" },
            { path: "files.media.icon.previewImage.focalPoint.x" },
            { path: "files.media.icon.previewImage.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkImportFiles;
}
function bulkImportFile(payload) {
  function __bulkImportFile({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FilesService.BulkImportFile",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/bulk/files/import-v2",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.lastUsedDate" },
            { path: "results.item.media.image.image.urlExpirationDate" },
            {
              path: "results.item.media.image.previewImage.urlExpirationDate"
            },
            { path: "results.item.media.video.urlExpirationDate" },
            {
              path: "results.item.media.video.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.media.video.resolutions.poster.urlExpirationDate"
            },
            { path: "results.item.media.video.posters.urlExpirationDate" },
            { path: "results.item.media.audio.assets.urlExpirationDate" },
            { path: "results.item.media.document.urlExpirationDate" },
            {
              path: "results.item.media.document.thumbnail.urlExpirationDate"
            },
            { path: "results.item.media.vector.image.urlExpirationDate" },
            {
              path: "results.item.media.vector.previewImage.urlExpirationDate"
            },
            { path: "results.item.media.archive.urlExpirationDate" },
            { path: "results.item.media.model3d.urlExpirationDate" },
            {
              path: "results.item.media.model3d.thumbnail.urlExpirationDate"
            },
            { path: "results.item.media.icon.image.urlExpirationDate" },
            {
              path: "results.item.media.icon.previewImage.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.media.image.image.focalPoint.x" },
            { path: "results.item.media.image.image.focalPoint.y" },
            { path: "results.item.media.image.faces.confidence" },
            { path: "results.item.media.image.previewImage.focalPoint.x" },
            { path: "results.item.media.image.previewImage.focalPoint.y" },
            {
              path: "results.item.media.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.media.video.resolutions.poster.focalPoint.y"
            },
            { path: "results.item.media.video.posters.focalPoint.x" },
            { path: "results.item.media.video.posters.focalPoint.y" },
            { path: "results.item.media.document.thumbnail.focalPoint.x" },
            { path: "results.item.media.document.thumbnail.focalPoint.y" },
            { path: "results.item.media.vector.image.focalPoint.x" },
            { path: "results.item.media.vector.image.focalPoint.y" },
            { path: "results.item.media.vector.faces.confidence" },
            { path: "results.item.media.vector.previewImage.focalPoint.x" },
            { path: "results.item.media.vector.previewImage.focalPoint.y" },
            { path: "results.item.media.model3d.thumbnail.focalPoint.x" },
            { path: "results.item.media.model3d.thumbnail.focalPoint.y" },
            { path: "results.item.media.icon.image.focalPoint.x" },
            { path: "results.item.media.icon.image.focalPoint.y" },
            { path: "results.item.media.icon.faces.confidence" },
            { path: "results.item.media.icon.previewImage.focalPoint.x" },
            { path: "results.item.media.icon.previewImage.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkImportFile;
}
function listFiles(payload) {
  function __listFiles({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "GET",
      methodFqn: "com.wix.media.site_media.v1.FilesService.ListFiles",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/files",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload, true),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "files.createdDate" },
            { path: "files.updatedDate" },
            { path: "files.lastUsedDate" },
            { path: "files.media.image.image.urlExpirationDate" },
            { path: "files.media.image.previewImage.urlExpirationDate" },
            { path: "files.media.video.urlExpirationDate" },
            { path: "files.media.video.resolutions.urlExpirationDate" },
            {
              path: "files.media.video.resolutions.poster.urlExpirationDate"
            },
            { path: "files.media.video.posters.urlExpirationDate" },
            { path: "files.media.audio.assets.urlExpirationDate" },
            { path: "files.media.document.urlExpirationDate" },
            { path: "files.media.document.thumbnail.urlExpirationDate" },
            { path: "files.media.vector.image.urlExpirationDate" },
            { path: "files.media.vector.previewImage.urlExpirationDate" },
            { path: "files.media.archive.urlExpirationDate" },
            { path: "files.media.model3d.urlExpirationDate" },
            { path: "files.media.model3d.thumbnail.urlExpirationDate" },
            { path: "files.media.icon.image.urlExpirationDate" },
            { path: "files.media.icon.previewImage.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "files.media.image.image.focalPoint.x" },
            { path: "files.media.image.image.focalPoint.y" },
            { path: "files.media.image.faces.confidence" },
            { path: "files.media.image.previewImage.focalPoint.x" },
            { path: "files.media.image.previewImage.focalPoint.y" },
            { path: "files.media.video.resolutions.poster.focalPoint.x" },
            { path: "files.media.video.resolutions.poster.focalPoint.y" },
            { path: "files.media.video.posters.focalPoint.x" },
            { path: "files.media.video.posters.focalPoint.y" },
            { path: "files.media.document.thumbnail.focalPoint.x" },
            { path: "files.media.document.thumbnail.focalPoint.y" },
            { path: "files.media.vector.image.focalPoint.x" },
            { path: "files.media.vector.image.focalPoint.y" },
            { path: "files.media.vector.faces.confidence" },
            { path: "files.media.vector.previewImage.focalPoint.x" },
            { path: "files.media.vector.previewImage.focalPoint.y" },
            { path: "files.media.model3d.thumbnail.focalPoint.x" },
            { path: "files.media.model3d.thumbnail.focalPoint.y" },
            { path: "files.media.icon.image.focalPoint.x" },
            { path: "files.media.icon.image.focalPoint.y" },
            { path: "files.media.icon.faces.confidence" },
            { path: "files.media.icon.previewImage.focalPoint.x" },
            { path: "files.media.icon.previewImage.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listFiles;
}
function searchFiles(payload) {
  function __searchFiles({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FilesService.SearchFiles",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/files/search",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "files.createdDate" },
            { path: "files.updatedDate" },
            { path: "files.lastUsedDate" },
            { path: "files.media.image.image.urlExpirationDate" },
            { path: "files.media.image.previewImage.urlExpirationDate" },
            { path: "files.media.video.urlExpirationDate" },
            { path: "files.media.video.resolutions.urlExpirationDate" },
            {
              path: "files.media.video.resolutions.poster.urlExpirationDate"
            },
            { path: "files.media.video.posters.urlExpirationDate" },
            { path: "files.media.audio.assets.urlExpirationDate" },
            { path: "files.media.document.urlExpirationDate" },
            { path: "files.media.document.thumbnail.urlExpirationDate" },
            { path: "files.media.vector.image.urlExpirationDate" },
            { path: "files.media.vector.previewImage.urlExpirationDate" },
            { path: "files.media.archive.urlExpirationDate" },
            { path: "files.media.model3d.urlExpirationDate" },
            { path: "files.media.model3d.thumbnail.urlExpirationDate" },
            { path: "files.media.icon.image.urlExpirationDate" },
            { path: "files.media.icon.previewImage.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "files.media.image.image.focalPoint.x" },
            { path: "files.media.image.image.focalPoint.y" },
            { path: "files.media.image.faces.confidence" },
            { path: "files.media.image.previewImage.focalPoint.x" },
            { path: "files.media.image.previewImage.focalPoint.y" },
            { path: "files.media.video.resolutions.poster.focalPoint.x" },
            { path: "files.media.video.resolutions.poster.focalPoint.y" },
            { path: "files.media.video.posters.focalPoint.x" },
            { path: "files.media.video.posters.focalPoint.y" },
            { path: "files.media.document.thumbnail.focalPoint.x" },
            { path: "files.media.document.thumbnail.focalPoint.y" },
            { path: "files.media.vector.image.focalPoint.x" },
            { path: "files.media.vector.image.focalPoint.y" },
            { path: "files.media.vector.faces.confidence" },
            { path: "files.media.vector.previewImage.focalPoint.x" },
            { path: "files.media.vector.previewImage.focalPoint.y" },
            { path: "files.media.model3d.thumbnail.focalPoint.x" },
            { path: "files.media.model3d.thumbnail.focalPoint.y" },
            { path: "files.media.icon.image.focalPoint.x" },
            { path: "files.media.icon.image.focalPoint.y" },
            { path: "files.media.icon.faces.confidence" },
            { path: "files.media.icon.previewImage.focalPoint.x" },
            { path: "files.media.icon.previewImage.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __searchFiles;
}
function generateVideoStreamingUrl(payload) {
  function __generateVideoStreamingUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FilesService.GenerateVideoStreamingUrl",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/files/generate-video-stream-url",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __generateVideoStreamingUrl;
}
function bulkDeleteFiles(payload) {
  function __bulkDeleteFiles({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FilesService.BulkDeleteFiles",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/bulk/files/delete",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkDeleteFiles;
}
function bulkRestoreFilesFromTrashBin(payload) {
  function __bulkRestoreFilesFromTrashBin({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FilesService.BulkRestoreFilesFromTrashBin",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/bulk/trash-bin/files/restore",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkRestoreFilesFromTrashBin;
}
function listDeletedFiles(payload) {
  function __listDeletedFiles({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.file_descriptor",
      method: "GET",
      methodFqn: "com.wix.media.site_media.v1.FilesService.ListDeletedFiles",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FilesServiceUrl({
        protoPath: "/v1/trash-bin/files",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload, true),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "files.createdDate" },
            { path: "files.updatedDate" },
            { path: "files.lastUsedDate" },
            { path: "files.media.image.image.urlExpirationDate" },
            { path: "files.media.image.previewImage.urlExpirationDate" },
            { path: "files.media.video.urlExpirationDate" },
            { path: "files.media.video.resolutions.urlExpirationDate" },
            {
              path: "files.media.video.resolutions.poster.urlExpirationDate"
            },
            { path: "files.media.video.posters.urlExpirationDate" },
            { path: "files.media.audio.assets.urlExpirationDate" },
            { path: "files.media.document.urlExpirationDate" },
            { path: "files.media.document.thumbnail.urlExpirationDate" },
            { path: "files.media.vector.image.urlExpirationDate" },
            { path: "files.media.vector.previewImage.urlExpirationDate" },
            { path: "files.media.archive.urlExpirationDate" },
            { path: "files.media.model3d.urlExpirationDate" },
            { path: "files.media.model3d.thumbnail.urlExpirationDate" },
            { path: "files.media.icon.image.urlExpirationDate" },
            { path: "files.media.icon.previewImage.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "files.media.image.image.focalPoint.x" },
            { path: "files.media.image.image.focalPoint.y" },
            { path: "files.media.image.faces.confidence" },
            { path: "files.media.image.previewImage.focalPoint.x" },
            { path: "files.media.image.previewImage.focalPoint.y" },
            { path: "files.media.video.resolutions.poster.focalPoint.x" },
            { path: "files.media.video.resolutions.poster.focalPoint.y" },
            { path: "files.media.video.posters.focalPoint.x" },
            { path: "files.media.video.posters.focalPoint.y" },
            { path: "files.media.document.thumbnail.focalPoint.x" },
            { path: "files.media.document.thumbnail.focalPoint.y" },
            { path: "files.media.vector.image.focalPoint.x" },
            { path: "files.media.vector.image.focalPoint.y" },
            { path: "files.media.vector.faces.confidence" },
            { path: "files.media.vector.previewImage.focalPoint.x" },
            { path: "files.media.vector.previewImage.focalPoint.y" },
            { path: "files.media.model3d.thumbnail.focalPoint.x" },
            { path: "files.media.model3d.thumbnail.focalPoint.y" },
            { path: "files.media.icon.image.focalPoint.x" },
            { path: "files.media.icon.image.focalPoint.y" },
            { path: "files.media.icon.faces.confidence" },
            { path: "files.media.icon.previewImage.focalPoint.x" },
            { path: "files.media.icon.previewImage.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listDeletedFiles;
}

// src/media-site-media-v1-file-descriptor-files.universal.ts
var import_audio = require("@wix/sdk-runtime/transformations/audio");
var import_audio2 = require("@wix/sdk-runtime/transformations/audio");
var import_image = require("@wix/sdk-runtime/transformations/image");
var import_image2 = require("@wix/sdk-runtime/transformations/image");
var import_video_v2 = require("@wix/sdk-runtime/transformations/video-v2");
var import_video_v22 = require("@wix/sdk-runtime/transformations/video-v2");
var import_document = require("@wix/sdk-runtime/transformations/document");
var import_document2 = require("@wix/sdk-runtime/transformations/document");
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var MediaType = /* @__PURE__ */ ((MediaType2) => {
  MediaType2["UNKNOWN"] = "UNKNOWN";
  MediaType2["IMAGE"] = "IMAGE";
  MediaType2["VIDEO"] = "VIDEO";
  MediaType2["AUDIO"] = "AUDIO";
  MediaType2["DOCUMENT"] = "DOCUMENT";
  MediaType2["VECTOR"] = "VECTOR";
  MediaType2["ARCHIVE"] = "ARCHIVE";
  MediaType2["MODEL3D"] = "MODEL3D";
  MediaType2["OTHER"] = "OTHER";
  return MediaType2;
})(MediaType || {});
var OperationStatus = /* @__PURE__ */ ((OperationStatus2) => {
  OperationStatus2["FAILED"] = "FAILED";
  OperationStatus2["READY"] = "READY";
  OperationStatus2["PENDING"] = "PENDING";
  return OperationStatus2;
})(OperationStatus || {});
var State = /* @__PURE__ */ ((State2) => {
  State2["OK"] = "OK";
  State2["DELETED"] = "DELETED";
  return State2;
})(State || {});
var Namespace = /* @__PURE__ */ ((Namespace2) => {
  Namespace2["NO_NAMESPACE"] = "NO_NAMESPACE";
  Namespace2["OTHERS"] = "OTHERS";
  Namespace2["WIX_VIDEO"] = "WIX_VIDEO";
  Namespace2["WIX_MUSIC"] = "WIX_MUSIC";
  Namespace2["ALBUMS_AND_ART_STORE"] = "ALBUMS_AND_ART_STORE";
  Namespace2["WIX_ECOM"] = "WIX_ECOM";
  Namespace2["PHOTO_SHARE_APP"] = "PHOTO_SHARE_APP";
  Namespace2["SHARING_APP"] = "SHARING_APP";
  Namespace2["CHAT"] = "CHAT";
  Namespace2["LOGO_BUILDER"] = "LOGO_BUILDER";
  Namespace2["ALBUMS_OLD"] = "ALBUMS_OLD";
  Namespace2["CHAT_MOBILE"] = "CHAT_MOBILE";
  Namespace2["WIX_FORMS"] = "WIX_FORMS";
  Namespace2["WIX_LEGENDS"] = "WIX_LEGENDS";
  Namespace2["PORTFOLIO"] = "PORTFOLIO";
  return Namespace2;
})(Namespace || {});
var IdentityType = /* @__PURE__ */ ((IdentityType2) => {
  IdentityType2["UNKNOWN"] = "UNKNOWN";
  IdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  IdentityType2["MEMBER"] = "MEMBER";
  IdentityType2["WIX_USER"] = "WIX_USER";
  IdentityType2["APP"] = "APP";
  return IdentityType2;
})(IdentityType || {});
var ImageAnnotationType = /* @__PURE__ */ ((ImageAnnotationType2) => {
  ImageAnnotationType2["UNKNOWN_IMAGE_ANNOTATION_TYPE"] = "UNKNOWN_IMAGE_ANNOTATION_TYPE";
  ImageAnnotationType2["CONTAINS_TEXT"] = "CONTAINS_TEXT";
  ImageAnnotationType2["IS_ANIMATED"] = "IS_ANIMATED";
  ImageAnnotationType2["FACES"] = "FACES";
  ImageAnnotationType2["LABELS"] = "LABELS";
  ImageAnnotationType2["COLORS"] = "COLORS";
  ImageAnnotationType2["CAPTION"] = "CAPTION";
  return ImageAnnotationType2;
})(ImageAnnotationType || {});
var ContentDisposition = /* @__PURE__ */ ((ContentDisposition2) => {
  ContentDisposition2["ATTACHMENT"] = "ATTACHMENT";
  ContentDisposition2["INLINE"] = "INLINE";
  return ContentDisposition2;
})(ContentDisposition || {});
var UploadProtocol = /* @__PURE__ */ ((UploadProtocol2) => {
  UploadProtocol2["TUS"] = "TUS";
  return UploadProtocol2;
})(UploadProtocol || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var RootFolder = /* @__PURE__ */ ((RootFolder2) => {
  RootFolder2["MEDIA_ROOT"] = "MEDIA_ROOT";
  RootFolder2["TRASH_ROOT"] = "TRASH_ROOT";
  RootFolder2["VISITOR_UPLOADS_ROOT"] = "VISITOR_UPLOADS_ROOT";
  return RootFolder2;
})(RootFolder || {});
var StreamFormat = /* @__PURE__ */ ((StreamFormat2) => {
  StreamFormat2["UNKNOWN"] = "UNKNOWN";
  StreamFormat2["HLS"] = "HLS";
  StreamFormat2["DASH"] = "DASH";
  return StreamFormat2;
})(StreamFormat || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function generateFilesDownloadUrl2(fileIds) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ fileIds });
  const reqOpts = generateFilesDownloadUrl(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { fileIds: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["fileIds"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function generateFileDownloadUrl2(fileId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    fileId,
    downloadFileName: options?.downloadFileName,
    expirationInMinutes: options?.expirationInMinutes,
    expirationRedirectUrl: options?.expirationRedirectUrl,
    assetKeys: options?.assetKeys,
    contentDisposition: options?.contentDisposition
  });
  const reqOpts = generateFileDownloadUrl(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          fileId: "$[0]",
          downloadFileName: "$[1].downloadFileName",
          expirationInMinutes: "$[1].expirationInMinutes",
          expirationRedirectUrl: "$[1].expirationRedirectUrl",
          assetKeys: "$[1].assetKeys",
          contentDisposition: "$[1].contentDisposition"
        },
        singleArgumentUnchanged: false
      },
      ["fileId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getFileDescriptor2(fileId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ fileId });
  const reqOpts = getFileDescriptor(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "file.media.video" }]
        },
        {
          transformFn: import_document2.transformRESTDocumentToSDKDocument,
          paths: [{ path: "file.media.document" }]
        },
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "file.media.image.image" },
            { path: "file.media.image.previewImage" },
            { path: "file.media.vector.image" },
            { path: "file.media.vector.previewImage" },
            { path: "file.media.model3d.thumbnail" },
            { path: "file.media.icon.image" },
            { path: "file.media.icon.previewImage" }
          ]
        },
        {
          transformFn: import_audio2.transformRESTAudioToSDKAudio,
          paths: [{ path: "file.media.audio.assets", isRepeated: true }]
        }
      ])
    )?.file;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { fileId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["fileId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getFileDescriptors2(fileIds) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ fileIds });
  const reqOpts = getFileDescriptors(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "files.media.video" }]
        },
        {
          transformFn: import_document2.transformRESTDocumentToSDKDocument,
          paths: [{ path: "files.media.document" }]
        },
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "files.media.image.image" },
            { path: "files.media.image.previewImage" },
            { path: "files.media.vector.image" },
            { path: "files.media.vector.previewImage" },
            { path: "files.media.model3d.thumbnail" },
            { path: "files.media.icon.image" },
            { path: "files.media.icon.previewImage" }
          ]
        },
        {
          transformFn: import_audio2.transformRESTAudioToSDKAudio,
          paths: [{ path: "files.media.audio.assets", isRepeated: true }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { fileIds: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["fileIds"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateFileDescriptor2(file) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ file }),
    [
      {
        transformFn: import_video_v2.transformSDKVideoV2ToRESTVideoV2,
        paths: [{ path: "file.media.video" }]
      },
      {
        transformFn: import_document.transformSDKDocumentToRESTDocument,
        paths: [{ path: "file.media.document" }]
      },
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [
          { path: "file.media.image.image" },
          { path: "file.media.image.previewImage" },
          { path: "file.media.vector.image" },
          { path: "file.media.vector.previewImage" },
          { path: "file.media.model3d.thumbnail" },
          { path: "file.media.icon.image" },
          { path: "file.media.icon.previewImage" }
        ]
      },
      {
        transformFn: import_audio.transformSDKAudioToRESTAudio,
        paths: [{ path: "file.media.audio.assets", isRepeated: true }]
      }
    ]
  );
  const reqOpts = updateFileDescriptor(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "file.media.video" }]
        },
        {
          transformFn: import_document2.transformRESTDocumentToSDKDocument,
          paths: [{ path: "file.media.document" }]
        },
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "file.media.image.image" },
            { path: "file.media.image.previewImage" },
            { path: "file.media.vector.image" },
            { path: "file.media.vector.previewImage" },
            { path: "file.media.model3d.thumbnail" },
            { path: "file.media.icon.image" },
            { path: "file.media.icon.previewImage" }
          ]
        },
        {
          transformFn: import_audio2.transformRESTAudioToSDKAudio,
          paths: [{ path: "file.media.audio.assets", isRepeated: true }]
        }
      ])
    )?.file;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { file: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["file"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function generateFileUploadUrl2(mimeType, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    mimeType,
    fileName: options?.fileName,
    sizeInBytes: options?.sizeInBytes,
    parentFolderId: options?.parentFolderId,
    private: options?.private,
    labels: options?.labels,
    externalInfo: options?.externalInfo,
    filePath: options?.filePath
  });
  const reqOpts = generateFileUploadUrl(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          mimeType: "$[0]",
          fileName: "$[1].fileName",
          sizeInBytes: "$[1].sizeInBytes",
          parentFolderId: "$[1].parentFolderId",
          private: "$[1].private",
          labels: "$[1].labels",
          externalInfo: "$[1].externalInfo",
          filePath: "$[1].filePath"
        },
        singleArgumentUnchanged: false
      },
      ["mimeType", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function generateFileResumableUploadUrl2(mimeType, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    mimeType,
    fileName: options?.fileName,
    sizeInBytes: options?.sizeInBytes,
    parentFolderId: options?.parentFolderId,
    private: options?.private,
    labels: options?.labels,
    uploadProtocol: options?.uploadProtocol,
    filePath: options?.filePath
  });
  const reqOpts = generateFileResumableUploadUrl(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          mimeType: "$[0]",
          fileName: "$[1].fileName",
          sizeInBytes: "$[1].sizeInBytes",
          parentFolderId: "$[1].parentFolderId",
          private: "$[1].private",
          labels: "$[1].labels",
          uploadProtocol: "$[1].uploadProtocol",
          filePath: "$[1].filePath"
        },
        singleArgumentUnchanged: false
      },
      ["mimeType", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function importFile2(url, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    url,
    mediaType: options?.mediaType,
    displayName: options?.displayName,
    parentFolderId: options?.parentFolderId,
    private: options?.private,
    labels: options?.labels,
    mimeType: options?.mimeType,
    externalInfo: options?.externalInfo,
    urlParams: options?.urlParams,
    urlHeaders: options?.urlHeaders,
    filePath: options?.filePath
  });
  const reqOpts = importFile(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "file.media.video" }]
        },
        {
          transformFn: import_document2.transformRESTDocumentToSDKDocument,
          paths: [{ path: "file.media.document" }]
        },
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "file.media.image.image" },
            { path: "file.media.image.previewImage" },
            { path: "file.media.vector.image" },
            { path: "file.media.vector.previewImage" },
            { path: "file.media.model3d.thumbnail" },
            { path: "file.media.icon.image" },
            { path: "file.media.icon.previewImage" }
          ]
        },
        {
          transformFn: import_audio2.transformRESTAudioToSDKAudio,
          paths: [{ path: "file.media.audio.assets", isRepeated: true }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          url: "$[0]",
          mediaType: "$[1].mediaType",
          displayName: "$[1].displayName",
          parentFolderId: "$[1].parentFolderId",
          private: "$[1].private",
          labels: "$[1].labels",
          mimeType: "$[1].mimeType",
          externalInfo: "$[1].externalInfo",
          urlParams: "$[1].urlParams",
          urlHeaders: "$[1].urlHeaders",
          filePath: "$[1].filePath"
        },
        singleArgumentUnchanged: false
      },
      ["url", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkImportFiles2(importFileRequests) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    importFileRequests
  });
  const reqOpts = bulkImportFiles(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "files.media.video" }]
        },
        {
          transformFn: import_document2.transformRESTDocumentToSDKDocument,
          paths: [{ path: "files.media.document" }]
        },
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "files.media.image.image" },
            { path: "files.media.image.previewImage" },
            { path: "files.media.vector.image" },
            { path: "files.media.vector.previewImage" },
            { path: "files.media.model3d.thumbnail" },
            { path: "files.media.icon.image" },
            { path: "files.media.icon.previewImage" }
          ]
        },
        {
          transformFn: import_audio2.transformRESTAudioToSDKAudio,
          paths: [{ path: "files.media.audio.assets", isRepeated: true }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { importFileRequests: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["importFileRequests"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkImportFile2(importFileRequests, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    importFileRequests,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkImportFile(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "results.item.media.video" }]
        },
        {
          transformFn: import_document2.transformRESTDocumentToSDKDocument,
          paths: [{ path: "results.item.media.document" }]
        },
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "results.item.media.image.image" },
            { path: "results.item.media.image.previewImage" },
            { path: "results.item.media.vector.image" },
            { path: "results.item.media.vector.previewImage" },
            { path: "results.item.media.model3d.thumbnail" },
            { path: "results.item.media.icon.image" },
            { path: "results.item.media.icon.previewImage" }
          ]
        },
        {
          transformFn: import_audio2.transformRESTAudioToSDKAudio,
          paths: [
            { path: "results.item.media.audio.assets", isRepeated: true }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          importFileRequests: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["importFileRequests", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listFiles2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    parentFolderId: options?.parentFolderId,
    mediaTypes: options?.mediaTypes,
    private: options?.private,
    sort: options?.sort,
    paging: options?.paging
  });
  const reqOpts = listFiles(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "files.media.video" }]
        },
        {
          transformFn: import_document2.transformRESTDocumentToSDKDocument,
          paths: [{ path: "files.media.document" }]
        },
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "files.media.image.image" },
            { path: "files.media.image.previewImage" },
            { path: "files.media.vector.image" },
            { path: "files.media.vector.previewImage" },
            { path: "files.media.model3d.thumbnail" },
            { path: "files.media.icon.image" },
            { path: "files.media.icon.previewImage" }
          ]
        },
        {
          transformFn: import_audio2.transformRESTAudioToSDKAudio,
          paths: [{ path: "files.media.audio.assets", isRepeated: true }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          parentFolderId: "$[0].parentFolderId",
          mediaTypes: "$[0].mediaTypes",
          private: "$[0].private",
          sort: "$[0].sort",
          paging: "$[0].paging"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function searchFiles2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    search: options?.search,
    rootFolder: options?.rootFolder,
    mediaTypes: options?.mediaTypes,
    private: options?.private,
    sort: options?.sort,
    paging: options?.paging
  });
  const reqOpts = searchFiles(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "files.media.video" }]
        },
        {
          transformFn: import_document2.transformRESTDocumentToSDKDocument,
          paths: [{ path: "files.media.document" }]
        },
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "files.media.image.image" },
            { path: "files.media.image.previewImage" },
            { path: "files.media.vector.image" },
            { path: "files.media.vector.previewImage" },
            { path: "files.media.model3d.thumbnail" },
            { path: "files.media.icon.image" },
            { path: "files.media.icon.previewImage" }
          ]
        },
        {
          transformFn: import_audio2.transformRESTAudioToSDKAudio,
          paths: [{ path: "files.media.audio.assets", isRepeated: true }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          search: "$[0].search",
          rootFolder: "$[0].rootFolder",
          mediaTypes: "$[0].mediaTypes",
          private: "$[0].private",
          sort: "$[0].sort",
          paging: "$[0].paging"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function generateVideoStreamingUrl2(fileId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    fileId,
    format: options?.format
  });
  const reqOpts = generateVideoStreamingUrl(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { fileId: "$[0]", format: "$[1].format" },
        singleArgumentUnchanged: false
      },
      ["fileId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkDeleteFiles2(fileIds, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    fileIds,
    permanent: options?.permanent
  });
  const reqOpts = bulkDeleteFiles(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          fileIds: "$[0]",
          permanent: "$[1].permanent"
        },
        singleArgumentUnchanged: false
      },
      ["fileIds", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkRestoreFilesFromTrashBin2(fileIds) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ fileIds });
  const reqOpts = bulkRestoreFilesFromTrashBin(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { fileIds: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["fileIds"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listDeletedFiles2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    parentFolderId: options?.parentFolderId,
    mediaTypes: options?.mediaTypes,
    private: options?.private,
    sort: options?.sort,
    paging: options?.paging
  });
  const reqOpts = listDeletedFiles(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "files.media.video" }]
        },
        {
          transformFn: import_document2.transformRESTDocumentToSDKDocument,
          paths: [{ path: "files.media.document" }]
        },
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "files.media.image.image" },
            { path: "files.media.image.previewImage" },
            { path: "files.media.vector.image" },
            { path: "files.media.vector.previewImage" },
            { path: "files.media.model3d.thumbnail" },
            { path: "files.media.icon.image" },
            { path: "files.media.icon.previewImage" }
          ]
        },
        {
          transformFn: import_audio2.transformRESTAudioToSDKAudio,
          paths: [{ path: "files.media.audio.assets", isRepeated: true }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          parentFolderId: "$[0].parentFolderId",
          mediaTypes: "$[0].mediaTypes",
          private: "$[0].private",
          sort: "$[0].sort",
          paging: "$[0].paging"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/media-site-media-v1-file-descriptor-files.public.ts
function generateFilesDownloadUrl3(httpClient) {
  return (fileIds) => generateFilesDownloadUrl2(
    fileIds,
    // @ts-ignore
    { httpClient }
  );
}
function generateFileDownloadUrl3(httpClient) {
  return (fileId, options) => generateFileDownloadUrl2(
    fileId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function getFileDescriptor3(httpClient) {
  return (fileId) => getFileDescriptor2(
    fileId,
    // @ts-ignore
    { httpClient }
  );
}
function getFileDescriptors3(httpClient) {
  return (fileIds) => getFileDescriptors2(
    fileIds,
    // @ts-ignore
    { httpClient }
  );
}
function updateFileDescriptor3(httpClient) {
  return (file) => updateFileDescriptor2(
    file,
    // @ts-ignore
    { httpClient }
  );
}
function generateFileUploadUrl3(httpClient) {
  return (mimeType, options) => generateFileUploadUrl2(
    mimeType,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function generateFileResumableUploadUrl3(httpClient) {
  return (mimeType, options) => generateFileResumableUploadUrl2(
    mimeType,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function importFile3(httpClient) {
  return (url, options) => importFile2(
    url,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkImportFiles3(httpClient) {
  return (importFileRequests) => bulkImportFiles2(
    importFileRequests,
    // @ts-ignore
    { httpClient }
  );
}
function bulkImportFile3(httpClient) {
  return (importFileRequests, options) => bulkImportFile2(
    importFileRequests,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function listFiles3(httpClient) {
  return (options) => listFiles2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function searchFiles3(httpClient) {
  return (options) => searchFiles2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function generateVideoStreamingUrl3(httpClient) {
  return (fileId, options) => generateVideoStreamingUrl2(
    fileId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkDeleteFiles3(httpClient) {
  return (fileIds, options) => bulkDeleteFiles2(
    fileIds,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkRestoreFilesFromTrashBin3(httpClient) {
  return (fileIds) => bulkRestoreFilesFromTrashBin2(
    fileIds,
    // @ts-ignore
    { httpClient }
  );
}
function listDeletedFiles3(httpClient) {
  return (options) => listDeletedFiles2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
var onFileDescriptorDeleted = (0, import_sdk_types.EventDefinition)(
  "wix.media.site_media.v1.file_descriptor_deleted",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "undefined.createdDate" },
          { path: "undefined.updatedDate" },
          { path: "undefined.lastUsedDate" },
          { path: "undefined.media.archive.urlExpirationDate" },
          { path: "undefined.media.model3d.urlExpirationDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_video_v23.transformRESTVideoV2ToSDKVideoV2,
        paths: [{ path: "undefined.media.video" }]
      },
      {
        transformFn: import_document3.transformRESTDocumentToSDKDocument,
        paths: [{ path: "undefined.media.document" }]
      },
      {
        transformFn: import_image3.transformRESTImageToSDKImage,
        paths: [
          { path: "undefined.media.image.image" },
          { path: "undefined.media.image.previewImage" },
          { path: "undefined.media.vector.image" },
          { path: "undefined.media.vector.previewImage" },
          { path: "undefined.media.model3d.thumbnail" },
          { path: "undefined.media.icon.image" },
          { path: "undefined.media.icon.previewImage" }
        ]
      },
      {
        transformFn: import_float3.transformRESTFloatToSDKFloat,
        paths: [
          { path: "undefined.media.image.faces.confidence" },
          { path: "undefined.media.vector.faces.confidence" },
          { path: "undefined.media.icon.faces.confidence" }
        ]
      },
      {
        transformFn: import_audio3.transformRESTAudioToSDKAudio,
        paths: [{ path: "undefined.media.audio.assets", isRepeated: true }]
      }
    ])
  )
)();
var onFileDescriptorFileFailed = (0, import_sdk_types.EventDefinition)(
  "wix.media.site_media.v1.file_descriptor_file_failed",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [{ path: "metadata.eventTime" }]
      }
    ])
  )
)();
var onFileDescriptorFileReady = (0, import_sdk_types.EventDefinition)(
  "wix.media.site_media.v1.file_descriptor_file_ready",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "data.file.createdDate" },
          { path: "data.file.updatedDate" },
          { path: "data.file.lastUsedDate" },
          { path: "data.file.media.archive.urlExpirationDate" },
          { path: "data.file.media.model3d.urlExpirationDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_video_v23.transformRESTVideoV2ToSDKVideoV2,
        paths: [{ path: "data.file.media.video" }]
      },
      {
        transformFn: import_document3.transformRESTDocumentToSDKDocument,
        paths: [{ path: "data.file.media.document" }]
      },
      {
        transformFn: import_image3.transformRESTImageToSDKImage,
        paths: [
          { path: "data.file.media.image.image" },
          { path: "data.file.media.image.previewImage" },
          { path: "data.file.media.vector.image" },
          { path: "data.file.media.vector.previewImage" },
          { path: "data.file.media.model3d.thumbnail" },
          { path: "data.file.media.icon.image" },
          { path: "data.file.media.icon.previewImage" }
        ]
      },
      {
        transformFn: import_float3.transformRESTFloatToSDKFloat,
        paths: [
          { path: "data.file.media.image.faces.confidence" },
          { path: "data.file.media.vector.faces.confidence" },
          { path: "data.file.media.icon.faces.confidence" }
        ]
      },
      {
        transformFn: import_audio3.transformRESTAudioToSDKAudio,
        paths: [{ path: "data.file.media.audio.assets", isRepeated: true }]
      }
    ])
  )
)();
var onFileDescriptorUpdated = (0, import_sdk_types.EventDefinition)(
  "wix.media.site_media.v1.file_descriptor_updated",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.lastUsedDate" },
          { path: "entity.media.archive.urlExpirationDate" },
          { path: "entity.media.model3d.urlExpirationDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_video_v23.transformRESTVideoV2ToSDKVideoV2,
        paths: [{ path: "entity.media.video" }]
      },
      {
        transformFn: import_document3.transformRESTDocumentToSDKDocument,
        paths: [{ path: "entity.media.document" }]
      },
      {
        transformFn: import_image3.transformRESTImageToSDKImage,
        paths: [
          { path: "entity.media.image.image" },
          { path: "entity.media.image.previewImage" },
          { path: "entity.media.vector.image" },
          { path: "entity.media.vector.previewImage" },
          { path: "entity.media.model3d.thumbnail" },
          { path: "entity.media.icon.image" },
          { path: "entity.media.icon.previewImage" }
        ]
      },
      {
        transformFn: import_float3.transformRESTFloatToSDKFloat,
        paths: [
          { path: "entity.media.image.faces.confidence" },
          { path: "entity.media.vector.faces.confidence" },
          { path: "entity.media.icon.faces.confidence" }
        ]
      },
      {
        transformFn: import_audio3.transformRESTAudioToSDKAudio,
        paths: [{ path: "entity.media.audio.assets", isRepeated: true }]
      }
    ])
  )
)();

// src/media-site-media-v1-file-descriptor-files.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var import_event_definition_modules = require("@wix/sdk-runtime/event-definition-modules");
var generateFilesDownloadUrl4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(generateFilesDownloadUrl3);
var generateFileDownloadUrl4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(generateFileDownloadUrl3);
var getFileDescriptor4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getFileDescriptor3);
var getFileDescriptors4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getFileDescriptors3);
var updateFileDescriptor4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(updateFileDescriptor3);
var generateFileUploadUrl4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(generateFileUploadUrl3);
var generateFileResumableUploadUrl4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(generateFileResumableUploadUrl3);
var importFile4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(importFile3);
var bulkImportFiles4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkImportFiles3);
var bulkImportFile4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkImportFile3);
var listFiles4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(listFiles3);
var searchFiles4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(searchFiles3);
var generateVideoStreamingUrl4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(generateVideoStreamingUrl3);
var bulkDeleteFiles4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkDeleteFiles3);
var bulkRestoreFilesFromTrashBin4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkRestoreFilesFromTrashBin3);
var listDeletedFiles4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(listDeletedFiles3);
var onFileDescriptorDeleted2 = (0, import_event_definition_modules.createEventModule)(
  onFileDescriptorDeleted
);
var onFileDescriptorFileFailed2 = (0, import_event_definition_modules.createEventModule)(
  onFileDescriptorFileFailed
);
var onFileDescriptorFileReady2 = (0, import_event_definition_modules.createEventModule)(
  onFileDescriptorFileReady
);
var onFileDescriptorUpdated2 = (0, import_event_definition_modules.createEventModule)(
  onFileDescriptorUpdated
);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ContentDisposition,
  IdentityType,
  ImageAnnotationType,
  MediaType,
  Namespace,
  OperationStatus,
  RootFolder,
  SortOrder,
  State,
  StreamFormat,
  UploadProtocol,
  WebhookIdentityType,
  bulkDeleteFiles,
  bulkImportFile,
  bulkImportFiles,
  bulkRestoreFilesFromTrashBin,
  generateFileDownloadUrl,
  generateFileResumableUploadUrl,
  generateFileUploadUrl,
  generateFilesDownloadUrl,
  generateVideoStreamingUrl,
  getFileDescriptor,
  getFileDescriptors,
  importFile,
  listDeletedFiles,
  listFiles,
  onFileDescriptorDeleted,
  onFileDescriptorFileFailed,
  onFileDescriptorFileReady,
  onFileDescriptorUpdated,
  searchFiles,
  updateFileDescriptor
});
