// @wix/headless-faq/dist/react/Faq.js
import { jsx as _jsx } from "react/jsx-runtime";
import React from 'react';
import { AsChildSlot } from '@wix/headless-utils/react';
import * as CoreFaq from './core/Faq.js';
var TestIds;
(function (TestIds) {
    // Entity Root
    TestIds["faqRoot"] = "faq-root";
    TestIds["faqName"] = "faq-name";
    TestIds["faqAnswer"] = "faq-answer";
    TestIds["faqItem"] = "faq-item";
})(TestIds || (TestIds = {}));
/**
 * Root component that provides FAQ service context for rendering a single FAQ entry.
 * Follows the services pattern like Product.Root does.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Faq.Root faq={faqEntry}>
 *   <Faq.Name />
 *   <Faq.Answer />
 * </Faq.Root>
 * ```
 */
export function Root(props) {
    const { children, faq, ...attrs } = props;
    const serviceConfig = {
        faqs: [faq],
        categoryId: faq.categoryId,
    };
    return (_jsx(CoreFaq.Root, { faqServiceConfig: serviceConfig, children: _jsx(AsChildSlot, { ...attrs, "data-testid": TestIds.faqRoot, "data-faq-id": faq._id, children: children }) }));
}
/**
 * Displays the FAQ question/name.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Faq.Name className="text-lg font-semibold" />
 *
 * // asChild with primitive
 * <Faq.Name asChild>
 *   <h3 className="text-lg font-semibold" />
 * </Faq.Name>
 *
 * // asChild with react component
 * <Faq.Name asChild>
 *   {React.forwardRef(({question, ...props}, ref) => (
 *     <h3 ref={ref} {...props} className="text-lg font-semibold">
 *       Q: {question}
 *     </h3>
 *   ))}
 * </Faq.Name>
 * ```
 */
export const Name = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreFaq.Name, { children: ({ question }) => {
            return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.faqName, customElement: children, customElementProps: { question }, content: question, ...otherProps, children: _jsx("div", { children: question }) }));
        } }));
});
/**
 * FAQ item wrapper component that provides access to the FAQ unique identifier.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Faq.Item className="faq-item" />
 *
 * // asChild with AccordionItem
 * <Faq.Item asChild>
 *   <AccordionItem className="border-b border-white/10 last:border-b-0" />
 * </Faq.Item>
 *
 * // asChild with custom render function
 * <Faq.Item asChild>
 *   {React.forwardRef(({id, ...props}, ref) => (
 *     <AccordionItem ref={ref} {...props} value={id}>
 *       {children}
 *     </AccordionItem>
 *   ))}
 * </Faq.Item>
 * ```
 */
export const Item = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreFaq.Item, { children: ({ id }) => {
            return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.faqItem, "data-faq-id": id, customElement: children, customElementProps: { id }, content: id, ...otherProps, children: _jsx("div", { children: id }) }));
        } }));
});
/**
 * Displays the FAQ answer.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Faq.Answer className="text-base text-content-secondary" />
 *
 * // asChild with primitive
 * <Faq.Answer asChild>
 *   <p className="text-base text-content-secondary" />
 * </Faq.Answer>
 *
 * // asChild with react component
 * <Faq.Answer asChild>
 *   {React.forwardRef(({answer, ...props}, ref) => (
 *     <p ref={ref} {...props} className="text-base text-content-secondary">
 *       A: {answer}
 *     </p>
 *   ))}
 * </Faq.Answer>
 * ```
 */
export const Answer = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreFaq.Answer, { children: ({ answer }) => {
            return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.faqAnswer, customElement: children, customElementProps: { answer }, content: answer, ...otherProps, children: _jsx("div", { children: answer }) }));
        } }));
});
