// @wix/headless-faq/dist/react/FaqCategories.js
import { jsx as _jsx, Fragment as _Fragment } from "react/jsx-runtime";
import React from 'react';
import { AsChildSlot } from '@wix/headless-utils/react';
import * as CoreFaqCategories from './core/FaqCategories.js';
import * as FaqCategoryComponents from './FaqList.js';
var TestIds;
(function (TestIds) {
    // Container Level
    TestIds["faqCategoriesRoot"] = "faq-categories-root";
    // List Container Level
    TestIds["faqCategories"] = "faq-categories";
    // Repeater Level (individual items)
    TestIds["faqCategory"] = "faq-category";
})(TestIds || (TestIds = {}));
/**
 * Root component that provides FAQ categories service context and conditional rendering.
 * Does not render if there are no categories.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <FaqCategories.Root faqCategoriesConfig={faqCategoriesConfig}>
 *   <FaqCategories.Categories>
 *     <FaqCategories.CategoryRepeater>
 *       <FaqCategory.Name />
 *       <FaqCategory.Faqs>
 *         <FaqCategory.FaqRepeater>
 *           <Faq.Name />
 *           <Faq.Answer />
 *         </FaqCategory.FaqRepeater>
 *       </FaqCategory.Faqs>
 *     </FaqCategories.CategoryRepeater>
 *   </FaqCategories.Categories>
 * </FaqCategories.Root>
 * ```
 */
export function Root(props) {
    const { children, faqCategoriesConfig, className, ...attrs } = props;
    const hasCategories = faqCategoriesConfig.categories.length > 0;
    // Don't render if no categories (following Container Level pattern)
    if (!hasCategories)
        return null;
    return (_jsx(CoreFaqCategories.Root, { faqCategoriesServiceConfig: faqCategoriesConfig, children: _jsx(AsChildSlot, { ...attrs, className: className, "data-testid": TestIds.faqCategoriesRoot, "data-categories-count": faqCategoriesConfig.categories.length, children: children }) }));
}
/**
 * Container for the list of categories with empty state support.
 * Follows List Container Level pattern.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <FaqCategories.Categories emptyState={<div>No categories available</div>}>
 *   <FaqCategories.CategoryRepeater>
 *     <FaqCategory.Name />
 *     <FaqCategory.Faqs>
 *       <FaqCategory.FaqRepeater>
 *         <Faq.Name />
 *         <Faq.Answer />
 *       </FaqCategory.FaqRepeater>
 *     </FaqCategory.Faqs>
 *   </FaqCategories.CategoryRepeater>
 * </FaqCategories.Categories>
 *
 * // Simple container usage
 * <FaqCategories.Categories emptyState={<div>No categories</div>}>
 *   <div className="categories-container">
 *     <FaqCategories.CategoryRepeater>
 *       <FaqCategory.Name />
 *     </FaqCategories.CategoryRepeater>
 *   </div>
 * </FaqCategories.Categories>
 * ```
 */
export const Categories = React.forwardRef((props, ref) => {
    const { children, emptyState } = props;
    return (_jsx(CoreFaqCategories.Categories, { children: ({ hasCategories }) => {
            if (!hasCategories) {
                return emptyState || null;
            }
            const attributes = {
                'data-testid': TestIds.faqCategories,
            };
            return (_jsx("div", { ...attributes, ref: ref, children: children }));
        } }));
});
/**
 * Repeater component that renders FaqCategory.Root for each category.
 * Follows Repeater Level pattern.
 *
 * @component
 * @example
 * ```tsx
 * <FaqCategories.CategoryRepeater>
 *   <FaqCategory.Name />
 *   <FaqCategory.Faqs>
 *     <FaqCategory.FaqRepeater>
 *       <Faq.Name />
 *       <Faq.Answer />
 *     </FaqCategory.FaqRepeater>
 *   </FaqCategory.Faqs>
 * </FaqCategories.CategoryRepeater>
 *
 * // With FAQ configurations per category
 * <FaqCategories.CategoryRepeater faqConfigsByCategory={faqConfigsByCategory}>
 *   <FaqCategory.Name />
 *   <FaqCategory.Faqs>
 *     <FaqCategory.FaqRepeater>
 *       <Faq.Name />
 *       <Faq.Answer />
 *     </FaqCategory.FaqRepeater>
 *   </FaqCategory.Faqs>
 * </FaqCategories.CategoryRepeater>
 * ```
 */
export const CategoryRepeater = React.forwardRef((props, _ref) => {
    const { children, faqConfigsByCategory } = props;
    return (_jsx(CoreFaqCategories.Categories, { children: ({ hasCategories, categories }) => {
            if (!hasCategories)
                return null;
            return (_jsx(_Fragment, { children: categories.map((category) => (_jsx(FaqCategoryComponents.Root, { category: category, faqConfig: faqConfigsByCategory?.[category._id || ''] || {
                        faqs: [],
                        categoryId: category._id,
                    }, "data-testid": TestIds.faqCategory, children: children }, category._id))) }));
        } }));
});
