// @wix/headless-faq/dist/react/FaqList.js
import { jsx as _jsx, Fragment as _Fragment } from "react/jsx-runtime";
import React from 'react';
import { AsChildSlot } from '@wix/headless-utils/react';
import * as CoreFaqList from './core/FaqList.js';
import * as Faq from './Faq.js';
var TestIds;
(function (TestIds) {
    // Entity Root
    TestIds["faqListRoot"] = "faq-list-root";
    TestIds["faqListName"] = "faq-list-name";
    // List Container Level
    TestIds["faqListFaqs"] = "faq-list-faqs";
    // Repeater Level (individual items)
    TestIds["faqListFaq"] = "faq-list-faq";
})(TestIds || (TestIds = {}));
/**
 * Root component that provides service context for a FAQ category and its FAQs.
 * Automatically loads FAQs for the category.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <FaqList.Root category={category}>
 *   <FaqList.Name />
 *   <FaqList.Items>
 *     <FaqList.FaqRepeater>
 *       <Faq.Name />
 *       <Faq.Answer />
 *     </FaqList.FaqRepeater>
 *   </FaqList.Items>
 * </FaqList.Root>
 * ```
 */
export function Root(props) {
    const { children, category, className, faqConfig, ...attrs } = props;
    // Default FAQ config for this category
    const defaultFaqConfig = React.useMemo(() => faqConfig || {
        faqs: [],
        categoryId: category._id,
    }, [faqConfig, category._id]);
    return (_jsx(CoreFaqList.Root, { faqServiceConfig: defaultFaqConfig, children: _jsx(CategoryContext, { category: category, children: _jsx("div", { ...attrs, className: className, "data-testid": TestIds.faqListRoot, "data-category-id": category._id, "data-category-name": category.title, children: children }) }) }));
}
const CategoryDataContext = React.createContext(null);
function CategoryContext({ category, children }) {
    return (_jsx(CategoryDataContext.Provider, { value: category, children: children }));
}
function useCategoryData() {
    const category = React.useContext(CategoryDataContext);
    if (!category) {
        throw new Error('useCategoryData must be used within a FaqList.Root component');
    }
    return category;
}
/**
 * Displays the FAQ category name.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <FaqList.Name className="text-2xl font-bold" />
 *
 * // asChild with primitive
 * <FaqList.Name asChild>
 *   <h2 className="text-2xl font-bold" />
 * </FaqList.Name>
 *
 * // asChild with react component
 * <FaqList.Name asChild>
 *   {React.forwardRef(({name, ...props}, ref) => (
 *     <h2 ref={ref} {...props} className="text-2xl font-bold">
 *       {name}
 *     </h2>
 *   ))}
 * </FaqList.Name>
 * ```
 */
export const Name = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    const category = useCategoryData();
    const name = category.title || '';
    return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.faqListName, customElement: children, customElementProps: { name }, content: name, ...otherProps, children: _jsx("div", { children: name }) }));
});
/**
 * Container for the list of FAQs with empty state support.
 * Follows List Container Level pattern.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <FaqList.Items emptyState={<div>No FAQs available</div>}>
 *   <FaqList.FaqRepeater>
 *     <Faq.Name />
 *     <Faq.Answer />
 *   </FaqList.FaqRepeater>
 * </FaqList.Items>
 *
 * // Simple container usage
 * <FaqList.Items emptyState={<div>No FAQs</div>}>
 *   <div className="faqs-container">
 *     <FaqList.FaqRepeater>
 *       <Faq.Name />
 *       <Faq.Answer />
 *     </FaqList.FaqRepeater>
 *   </div>
 * </FaqList.Items>
 * ```
 */
export const Items = React.forwardRef((props, ref) => {
    const { children, emptyState } = props;
    return (_jsx(CoreFaqList.Items, { children: ({ hasFaqs }) => {
            if (!hasFaqs) {
                return emptyState || null;
            }
            const attributes = {
                'data-testid': TestIds.faqListFaqs,
            };
            return (_jsx("div", { ...attributes, ref: ref, children: children }));
        } }));
});
/**
 * Repeater component that renders Faq.Root for each FAQ entry.
 * Follows Repeater Level pattern.
 *
 * @component
 * @example
 * ```tsx
 * <FaqList.FaqRepeater>
 *   <Faq.Name />
 *   <Faq.Answer />
 * </FaqList.FaqRepeater>
 * ```
 */
export const FaqRepeater = React.forwardRef((props, _ref) => {
    const { children } = props;
    return (_jsx(CoreFaqList.Items, { children: ({ hasFaqs, faqs }) => {
            if (!hasFaqs)
                return null;
            return (_jsx(_Fragment, { children: faqs.map((faq) => (_jsx(Faq.Root, { faq: faq, "data-testid": TestIds.faqListFaq, children: children }, faq._id))) }));
        } }));
});
