// @wix/auto_sdk_restaurants_menus/build/internal/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  SortOrder: () => SortOrder,
  WebhookIdentityType: () => WebhookIdentityType,
  bulkCreateMenus: () => bulkCreateMenus2,
  bulkUpdateMenu: () => bulkUpdateMenu2,
  createMenu: () => createMenu2,
  deleteMenu: () => deleteMenu2,
  duplicateMenu: () => duplicateMenu2,
  getMenu: () => getMenu2,
  getMenuSiteUrl: () => getMenuSiteUrl2,
  listMenus: () => listMenus2,
  queryMenus: () => queryMenus2,
  queryMenusSiteUrl: () => queryMenusSiteUrl2,
  updateExtendedFields: () => updateExtendedFields2,
  updateMenu: () => updateMenu2
});
module.exports = __toCommonJS(index_typings_exports);

// src/restaurants-menus-v1-menu-menus.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/restaurants-menus-v1-menu-menus.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressRestaurantsMenusMenuV1RestaurantsDuplicateMenuRestaurantsDuplicateMenuUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/menus",
        destPath: "/v1/menus"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/menus",
        destPath: "/v1/bulk/menus"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/menus",
        destPath: "/v1/menus"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/menus",
        destPath: "/v1/bulk/menus"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/menus-menu",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/menus",
        destPath: "/v1/bulk/menus"
      },
      {
        srcPath: "/restaurants/menus/v1/menus",
        destPath: "/v1/menus"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
function resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusSiteUrlRestaurantsMenusSiteUrlUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/menus",
        destPath: "/v1/menus"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/menus",
        destPath: "/v1/bulk/menus"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/menus",
        destPath: "/v1/menus"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/menus",
        destPath: "/v1/bulk/menus"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/menus-menu",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/menus",
        destPath: "/v1/bulk/menus"
      },
      {
        srcPath: "/restaurants/menus/v1/menus",
        destPath: "/v1/menus"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
function resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusMenuUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/menus",
        destPath: "/v1/menus"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/menus",
        destPath: "/v1/bulk/menus"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/menus",
        destPath: "/v1/menus"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/menus",
        destPath: "/v1/bulk/menus"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/menus-menu",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/menus",
        destPath: "/v1/bulk/menus"
      },
      {
        srcPath: "/restaurants/menus/v1/menus",
        destPath: "/v1/menus"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_menus";
function createMenu(payload) {
  function __createMenu({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "menu.createdDate" },
          { path: "menu.updatedDate" },
          { path: "menu.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.RestaurantsMenusMenu.CreateMenu",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusMenuUrl({
        protoPath: "/v1/menus",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "menu.createdDate" },
            { path: "menu.updatedDate" },
            { path: "menu.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createMenu;
}
function bulkCreateMenus(payload) {
  function __bulkCreateMenus({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "menus.createdDate" },
          { path: "menus.updatedDate" },
          { path: "menus.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.RestaurantsMenusMenu.BulkCreateMenus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusMenuUrl({
        protoPath: "/v1/bulk/menus/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.menu.createdDate" },
            { path: "results.menu.updatedDate" },
            { path: "results.menu.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateMenus;
}
function getMenu(payload) {
  function __getMenu({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.RestaurantsMenusMenu.GetMenu",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusMenuUrl({
        protoPath: "/v1/menus/{menuId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "menu.createdDate" },
            { path: "menu.updatedDate" },
            { path: "menu.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getMenu;
}
function listMenus(payload) {
  function __listMenus({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.RestaurantsMenusMenu.ListMenus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusMenuUrl({
        protoPath: "/v1/menus",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "menus.createdDate" },
            { path: "menus.updatedDate" },
            { path: "menus.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listMenus;
}
function queryMenus(payload) {
  function __queryMenus({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.RestaurantsMenusMenu.QueryMenus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusMenuUrl({
        protoPath: "/v1/menus/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "menus.createdDate" },
            { path: "menus.updatedDate" },
            { path: "menus.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryMenus;
}
function updateMenu(payload) {
  function __updateMenu({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "menu.createdDate" },
          { path: "menu.updatedDate" },
          { path: "menu.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "PATCH",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.RestaurantsMenusMenu.UpdateMenu",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusMenuUrl({
        protoPath: "/v1/menus/{menu.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "menu.createdDate" },
            { path: "menu.updatedDate" },
            { path: "menu.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateMenu;
}
function bulkUpdateMenu(payload) {
  function __bulkUpdateMenu({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "menus.mask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "menus.menu.createdDate" },
          { path: "menus.menu.updatedDate" },
          { path: "menus.menu.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.RestaurantsMenusMenu.BulkUpdateMenu",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusMenuUrl({
        protoPath: "/v1/bulk/menus/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.menu.createdDate" },
            { path: "results.menu.updatedDate" },
            { path: "results.menu.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateMenu;
}
function updateExtendedFields(payload) {
  function __updateExtendedFields({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.RestaurantsMenusMenu.UpdateExtendedFields",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusMenuUrl({
        protoPath: "/v1/menus/{id}/updateExtendedFields",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __updateExtendedFields;
}
function deleteMenu(payload) {
  function __deleteMenu({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "DELETE",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.RestaurantsMenusMenu.DeleteMenu",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusMenuUrl({
        protoPath: "/v1/menus/{menuId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteMenu;
}
function getMenuSiteUrl(payload) {
  function __getMenuSiteUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.restaurants_menus_site_url.RestaurantsMenusSiteUrl.GetMenuSiteUrl",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusSiteUrlRestaurantsMenusSiteUrlUrl(
        { protoPath: "/v1/menus-site-url/menus/{id}", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getMenuSiteUrl;
}
function queryMenusSiteUrl(payload) {
  function __queryMenusSiteUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.restaurants_menus_site_url.RestaurantsMenusSiteUrl.QueryMenusSiteUrl",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusSiteUrlRestaurantsMenusSiteUrlUrl(
        { protoPath: "/v1/menus-site-url/menus/query", data: payload, host }
      ),
      data: payload
    };
    return metadata;
  }
  return __queryMenusSiteUrl;
}
function duplicateMenu(payload) {
  function __duplicateMenu({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.restaurants_duplicate_menu.RestaurantsDuplicateMenu.DuplicateMenu",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsDuplicateMenuRestaurantsDuplicateMenuUrl(
        { protoPath: "/v1/menus/{id}/duplicate", data: payload, host }
      ),
      data: payload
    };
    return metadata;
  }
  return __duplicateMenu;
}

// src/restaurants-menus-v1-menu-menus.universal.ts
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createMenu2(menu) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ menu });
  const reqOpts = createMenu(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.menu;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { menu: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["menu"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCreateMenus2(menus, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    menus,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkCreateMenus(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          menus: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["menus", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getMenu2(menuId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ menuId });
  const reqOpts = getMenu(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.menu;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { menuId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["menuId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listMenus2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    menuIds: options?.menuIds,
    paging: options?.paging,
    onlyVisible: options?.onlyVisible
  });
  const reqOpts = listMenus(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          menuIds: "$[0].menuIds",
          paging: "$[0].paging",
          onlyVisible: "$[0].onlyVisible"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryMenus2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryMenus(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [])
      );
      return {
        items: transformedData?.menus,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function updateMenu2(_id, menu) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    menu: { ...menu, id: _id }
  });
  const reqOpts = updateMenu(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.menu;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { menu: "$[1]" },
        explicitPathsToArguments: { "menu.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "menu"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateMenu2(menus, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    menus,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkUpdateMenu(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          menus: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["menus", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateExtendedFields2(_id, namespace, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    id: _id,
    namespace,
    namespaceData: options?.namespaceData
  });
  const reqOpts = updateExtendedFields(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          id: "$[0]",
          namespace: "$[1]",
          namespaceData: "$[2].namespaceData"
        },
        singleArgumentUnchanged: false
      },
      ["_id", "namespace", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteMenu2(menuId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ menuId });
  const reqOpts = deleteMenu(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { menuId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["menuId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getMenuSiteUrl2(_id) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ id: _id });
  const reqOpts = getMenuSiteUrl(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { id: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function queryMenusSiteUrl2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    query: options?.query
  });
  const reqOpts = queryMenusSiteUrl(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0].query" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function duplicateMenu2(_id, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    id: _id,
    businessLocationIds: options?.businessLocationIds,
    duplicateSubEntities: options?.duplicateSubEntities,
    menuName: options?.menuName
  });
  const reqOpts = duplicateMenu(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          id: "$[0]",
          businessLocationIds: "$[1].businessLocationIds",
          duplicateSubEntities: "$[1].duplicateSubEntities",
          menuName: "$[1].menuName"
        },
        singleArgumentUnchanged: false
      },
      ["_id", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  SortOrder,
  WebhookIdentityType,
  bulkCreateMenus,
  bulkUpdateMenu,
  createMenu,
  deleteMenu,
  duplicateMenu,
  getMenu,
  getMenuSiteUrl,
  listMenus,
  queryMenus,
  queryMenusSiteUrl,
  updateExtendedFields,
  updateMenu
});
