// @wix/auto_sdk_restaurants_menus/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  SortOrderOriginal: () => SortOrder,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  bulkCreateMenus: () => bulkCreateMenus2,
  bulkUpdateMenu: () => bulkUpdateMenu2,
  createMenu: () => createMenu2,
  deleteMenu: () => deleteMenu2,
  duplicateMenu: () => duplicateMenu2,
  getMenu: () => getMenu2,
  getMenuSiteUrl: () => getMenuSiteUrl2,
  listMenus: () => listMenus2,
  queryMenus: () => queryMenus2,
  queryMenusSiteUrl: () => queryMenusSiteUrl2,
  updateExtendedFields: () => updateExtendedFields2,
  updateMenu: () => updateMenu2
});
module.exports = __toCommonJS(meta_exports);

// src/restaurants-menus-v1-menu-menus.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressRestaurantsMenusMenuV1RestaurantsDuplicateMenuRestaurantsDuplicateMenuUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/menus",
        destPath: "/v1/menus"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/menus",
        destPath: "/v1/bulk/menus"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/menus",
        destPath: "/v1/menus"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/menus",
        destPath: "/v1/bulk/menus"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/menus-menu",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/menus",
        destPath: "/v1/bulk/menus"
      },
      {
        srcPath: "/restaurants/menus/v1/menus",
        destPath: "/v1/menus"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
function resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusSiteUrlRestaurantsMenusSiteUrlUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/menus",
        destPath: "/v1/menus"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/menus",
        destPath: "/v1/bulk/menus"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/menus",
        destPath: "/v1/menus"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/menus",
        destPath: "/v1/bulk/menus"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/menus-menu",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/menus",
        destPath: "/v1/bulk/menus"
      },
      {
        srcPath: "/restaurants/menus/v1/menus",
        destPath: "/v1/menus"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
function resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusMenuUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/menus",
        destPath: "/v1/menus"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/menus",
        destPath: "/v1/bulk/menus"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/menus",
        destPath: "/v1/menus"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/menus",
        destPath: "/v1/bulk/menus"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-menus-menu",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/menus-menu",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/menus",
        destPath: "/v1/bulk/menus"
      },
      {
        srcPath: "/restaurants/menus/v1/menus",
        destPath: "/v1/menus"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-menu",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_menus";
function createMenu(payload) {
  function __createMenu({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "menu.createdDate" },
          { path: "menu.updatedDate" },
          { path: "menu.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.RestaurantsMenusMenu.CreateMenu",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusMenuUrl({
        protoPath: "/v1/menus",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "menu.createdDate" },
            { path: "menu.updatedDate" },
            { path: "menu.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createMenu;
}
function bulkCreateMenus(payload) {
  function __bulkCreateMenus({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "menus.createdDate" },
          { path: "menus.updatedDate" },
          { path: "menus.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.RestaurantsMenusMenu.BulkCreateMenus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusMenuUrl({
        protoPath: "/v1/bulk/menus/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.menu.createdDate" },
            { path: "results.menu.updatedDate" },
            { path: "results.menu.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateMenus;
}
function getMenu(payload) {
  function __getMenu({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.RestaurantsMenusMenu.GetMenu",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusMenuUrl({
        protoPath: "/v1/menus/{menuId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "menu.createdDate" },
            { path: "menu.updatedDate" },
            { path: "menu.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getMenu;
}
function listMenus(payload) {
  function __listMenus({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.RestaurantsMenusMenu.ListMenus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusMenuUrl({
        protoPath: "/v1/menus",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "menus.createdDate" },
            { path: "menus.updatedDate" },
            { path: "menus.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listMenus;
}
function queryMenus(payload) {
  function __queryMenus({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.RestaurantsMenusMenu.QueryMenus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusMenuUrl({
        protoPath: "/v1/menus/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "menus.createdDate" },
            { path: "menus.updatedDate" },
            { path: "menus.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryMenus;
}
function updateMenu(payload) {
  function __updateMenu({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "menu.createdDate" },
          { path: "menu.updatedDate" },
          { path: "menu.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "PATCH",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.RestaurantsMenusMenu.UpdateMenu",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusMenuUrl({
        protoPath: "/v1/menus/{menu.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "menu.createdDate" },
            { path: "menu.updatedDate" },
            { path: "menu.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateMenu;
}
function bulkUpdateMenu(payload) {
  function __bulkUpdateMenu({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "menus.mask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "menus.menu.createdDate" },
          { path: "menus.menu.updatedDate" },
          { path: "menus.menu.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.RestaurantsMenusMenu.BulkUpdateMenu",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusMenuUrl({
        protoPath: "/v1/bulk/menus/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.menu.createdDate" },
            { path: "results.menu.updatedDate" },
            { path: "results.menu.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateMenu;
}
function updateExtendedFields(payload) {
  function __updateExtendedFields({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.RestaurantsMenusMenu.UpdateExtendedFields",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusMenuUrl({
        protoPath: "/v1/menus/{id}/updateExtendedFields",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __updateExtendedFields;
}
function deleteMenu(payload) {
  function __deleteMenu({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "DELETE",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.RestaurantsMenusMenu.DeleteMenu",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusMenuUrl({
        protoPath: "/v1/menus/{menuId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteMenu;
}
function getMenuSiteUrl(payload) {
  function __getMenuSiteUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.restaurants_menus_site_url.RestaurantsMenusSiteUrl.GetMenuSiteUrl",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusSiteUrlRestaurantsMenusSiteUrlUrl(
        { protoPath: "/v1/menus-site-url/menus/{id}", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getMenuSiteUrl;
}
function queryMenusSiteUrl(payload) {
  function __queryMenusSiteUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.restaurants_menus_site_url.RestaurantsMenusSiteUrl.QueryMenusSiteUrl",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsMenusSiteUrlRestaurantsMenusSiteUrlUrl(
        { protoPath: "/v1/menus-site-url/menus/query", data: payload, host }
      ),
      data: payload
    };
    return metadata;
  }
  return __queryMenusSiteUrl;
}
function duplicateMenu(payload) {
  function __duplicateMenu({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.menu",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_menu.v1.restaurants_duplicate_menu.RestaurantsDuplicateMenu.DuplicateMenu",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusMenuV1RestaurantsDuplicateMenuRestaurantsDuplicateMenuUrl(
        { protoPath: "/v1/menus/{id}/duplicate", data: payload, host }
      ),
      data: payload
    };
    return metadata;
  }
  return __duplicateMenu;
}

// src/restaurants-menus-v1-menu-menus.types.ts
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/restaurants-menus-v1-menu-menus.meta.ts
function createMenu2() {
  const payload = {};
  const getRequestOptions = createMenu(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/menus",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkCreateMenus2() {
  const payload = {};
  const getRequestOptions = bulkCreateMenus(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/menus/create",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getMenu2() {
  const payload = { menuId: ":menuId" };
  const getRequestOptions = getMenu(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/menus/{menuId}",
    pathParams: { menuId: "menuId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listMenus2() {
  const payload = {};
  const getRequestOptions = listMenus(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/menus",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryMenus2() {
  const payload = {};
  const getRequestOptions = queryMenus(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/menus/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateMenu2() {
  const payload = { menu: { id: ":menuId" } };
  const getRequestOptions = updateMenu(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/menus/{menu.id}",
    pathParams: { menuId: "menuId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateMenu2() {
  const payload = {};
  const getRequestOptions = bulkUpdateMenu(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/menus/update",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateExtendedFields2() {
  const payload = { id: ":id" };
  const getRequestOptions = updateExtendedFields(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/menus/{id}/updateExtendedFields",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteMenu2() {
  const payload = { menuId: ":menuId" };
  const getRequestOptions = deleteMenu(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/menus/{menuId}",
    pathParams: { menuId: "menuId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getMenuSiteUrl2() {
  const payload = { id: ":id" };
  const getRequestOptions = getMenuSiteUrl(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/menus-site-url/menus/{id}",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryMenusSiteUrl2() {
  const payload = {};
  const getRequestOptions = queryMenusSiteUrl(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/menus-site-url/menus/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function duplicateMenu2() {
  const payload = { id: ":id" };
  const getRequestOptions = duplicateMenu(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/menus/{id}/duplicate",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  SortOrderOriginal,
  WebhookIdentityTypeOriginal,
  bulkCreateMenus,
  bulkUpdateMenu,
  createMenu,
  deleteMenu,
  duplicateMenu,
  getMenu,
  getMenuSiteUrl,
  listMenus,
  queryMenus,
  queryMenusSiteUrl,
  updateExtendedFields,
  updateMenu
});
