// @wix/auto_sdk_faq_category/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  createCategory: () => createCategory2,
  deleteCategory: () => deleteCategory2,
  getCategory: () => getCategory2,
  listCategories: () => listCategories2,
  queryCategories: () => queryCategories2,
  updateCategory: () => updateCategory2,
  updateExtendedFields: () => updateExtendedFields2
});
module.exports = __toCommonJS(meta_exports);

// src/faq-category-v2-category-category.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixFaqCategoryV2CategoryServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/category",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/category",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/category",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/faq-server/v2/categories",
        destPath: "/v2/categories"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-categories",
        destPath: "/v2/draft-categories"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/faq-server/v2/categories",
        destPath: "/v2/categories"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-categories",
        destPath: "/v2/draft-categories"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/faq/v2/categories",
        destPath: "/v2/categories"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-categories",
        destPath: "/v2/draft-categories"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/faq/v2/categories",
        destPath: "/v2/categories"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/faq-server/v2/categories",
        destPath: "/v2/categories"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-categories",
        destPath: "/v2/draft-categories"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/faq-server/v2/categories",
        destPath: "/v2/categories"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-categories",
        destPath: "/v2/draft-categories"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/faq-server/v2/categories",
        destPath: "/v2/categories"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-categories",
        destPath: "/v2/draft-categories"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/faq-server/v2/categories",
        destPath: "/v2/categories"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-categories",
        destPath: "/v2/draft-categories"
      }
    ],
    _: [
      {
        srcPath: "/_api/faq-server/v2/categories",
        destPath: "/v2/categories"
      }
    ],
    "dev._base_domain_": [
      {
        srcPath: "/_api/faq-server/v2/categories",
        destPath: "/v2/categories"
      },
      {
        srcPath: "/_api/faq-server/v2/draft-categories",
        destPath: "/v2/draft-categories"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/faq-server/v2/draft-categories",
        destPath: "/v2/draft-categories"
      },
      {
        srcPath: "/_api/faq-server/v2/categories",
        destPath: "/v2/categories"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_faq_category";
function createCategory(payload) {
  function __createCategory({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "category.createdDate" },
          { path: "category.updatedDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [{ path: "category.sortOrder" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.faq.category.v2.category",
      method: "POST",
      methodFqn: "wix.faq.category.v2.CategoryService.CreateCategory",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqCategoryV2CategoryServiceUrl({
        protoPath: "/v2/categories",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "category.createdDate" },
            { path: "category.updatedDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [{ path: "category.sortOrder" }]
        }
      ])
    };
    return metadata;
  }
  return __createCategory;
}
function getCategory(payload) {
  function __getCategory({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.category.v2.category",
      method: "GET",
      methodFqn: "wix.faq.category.v2.CategoryService.GetCategory",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqCategoryV2CategoryServiceUrl({
        protoPath: "/v2/categories/{categoryId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "category.createdDate" },
            { path: "category.updatedDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [{ path: "category.sortOrder" }]
        }
      ])
    };
    return metadata;
  }
  return __getCategory;
}
function updateCategory(payload) {
  function __updateCategory({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "category.createdDate" },
          { path: "category.updatedDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [{ path: "category.sortOrder" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.faq.category.v2.category",
      method: "PATCH",
      methodFqn: "wix.faq.category.v2.CategoryService.UpdateCategory",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqCategoryV2CategoryServiceUrl({
        protoPath: "/v2/categories/{category.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "category.createdDate" },
            { path: "category.updatedDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [{ path: "category.sortOrder" }]
        }
      ])
    };
    return metadata;
  }
  return __updateCategory;
}
function deleteCategory(payload) {
  function __deleteCategory({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.category.v2.category",
      method: "DELETE",
      methodFqn: "wix.faq.category.v2.CategoryService.DeleteCategory",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqCategoryV2CategoryServiceUrl({
        protoPath: "/v2/categories/{categoryId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteCategory;
}
function listCategories(payload) {
  function __listCategories({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.category.v2.category",
      method: "GET",
      methodFqn: "wix.faq.category.v2.CategoryService.ListCategories",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqCategoryV2CategoryServiceUrl({
        protoPath: "/v2/categories",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "categories.createdDate" },
            { path: "categories.updatedDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [{ path: "categories.sortOrder" }]
        }
      ])
    };
    return metadata;
  }
  return __listCategories;
}
function queryCategories(payload) {
  function __queryCategories({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.category.v2.category",
      method: "POST",
      methodFqn: "wix.faq.category.v2.CategoryService.QueryCategories",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqCategoryV2CategoryServiceUrl({
        protoPath: "/v2/categories/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "categories.createdDate" },
            { path: "categories.updatedDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [{ path: "categories.sortOrder" }]
        }
      ])
    };
    return metadata;
  }
  return __queryCategories;
}
function updateExtendedFields(payload) {
  function __updateExtendedFields({ host }) {
    const metadata = {
      entityFqdn: "wix.faq.category.v2.category",
      method: "POST",
      methodFqn: "wix.faq.category.v2.CategoryService.UpdateExtendedFields",
      packageName: PACKAGE_NAME,
      url: resolveWixFaqCategoryV2CategoryServiceUrl({
        protoPath: "/v2/categories/{id}/update-extended-fields",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "category.createdDate" },
            { path: "category.updatedDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [{ path: "category.sortOrder" }]
        }
      ])
    };
    return metadata;
  }
  return __updateExtendedFields;
}

// src/faq-category-v2-category-category.meta.ts
function createCategory2() {
  const payload = {};
  const getRequestOptions = createCategory(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/categories",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getCategory2() {
  const payload = { categoryId: ":categoryId" };
  const getRequestOptions = getCategory(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/categories/{categoryId}",
    pathParams: { categoryId: "categoryId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateCategory2() {
  const payload = { category: { id: ":categoryId" } };
  const getRequestOptions = updateCategory(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v2/categories/{category.id}",
    pathParams: { categoryId: "categoryId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteCategory2() {
  const payload = { categoryId: ":categoryId" };
  const getRequestOptions = deleteCategory(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v2/categories/{categoryId}",
    pathParams: { categoryId: "categoryId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listCategories2() {
  const payload = {};
  const getRequestOptions = listCategories(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/categories",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryCategories2() {
  const payload = {};
  const getRequestOptions = queryCategories(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/categories/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateExtendedFields2() {
  const payload = { id: ":id" };
  const getRequestOptions = updateExtendedFields(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/categories/{id}/update-extended-fields",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  createCategory,
  deleteCategory,
  getCategory,
  listCategories,
  queryCategories,
  updateCategory,
  updateExtendedFields
});
