// @wix/auto_sdk_events_notifications/build/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  Alignment: () => Alignment,
  AppType: () => AppType,
  AspectRatio: () => AspectRatio,
  BackgroundType: () => BackgroundType,
  ButtonDataType: () => ButtonDataType,
  CardStylesType: () => CardStylesType,
  CheckoutType: () => CheckoutType,
  ConferenceType: () => ConferenceType,
  Crop: () => Crop,
  DecorationType: () => DecorationType,
  Direction: () => Direction,
  DividerDataAlignment: () => DividerDataAlignment,
  EmailNotificationReminderTime: () => EmailNotificationReminderTime,
  EmailNotificationType: () => EmailNotificationType,
  EventStatus: () => EventStatus,
  EventType: () => EventType,
  FeeName: () => FeeName,
  FeeType: () => FeeType,
  FontType: () => FontType,
  GIFType: () => GIFType,
  ImagePosition: () => ImagePosition,
  InitialExpandedItems: () => InitialExpandedItems,
  InputControlType: () => InputControlType,
  Layout: () => Layout,
  LayoutType: () => LayoutType,
  LineStyle: () => LineStyle,
  LocationType: () => LocationType,
  MapType: () => MapType,
  NodeType: () => NodeType,
  NullValue: () => NullValue,
  OrderStatus: () => OrderStatus,
  Orientation: () => Orientation,
  Placement: () => Placement,
  PluginContainerDataAlignment: () => PluginContainerDataAlignment,
  PollLayoutDirection: () => PollLayoutDirection,
  PollLayoutType: () => PollLayoutType,
  Position: () => Position,
  PushNotificationType: () => PushNotificationType,
  RegistrationStatus: () => RegistrationStatus,
  Resizing: () => Resizing,
  ResponsivenessBehaviour: () => ResponsivenessBehaviour,
  RsvpStatusOptions: () => RsvpStatusOptions,
  Scaling: () => Scaling,
  Source: () => Source,
  State: () => State,
  Status: () => Status,
  StylesPosition: () => StylesPosition,
  SubdivisionType: () => SubdivisionType,
  Target: () => Target,
  TaxType: () => TaxType,
  TextAlignment: () => TextAlignment,
  ThumbnailsAlignment: () => ThumbnailsAlignment,
  Type: () => Type,
  ValueType: () => ValueType,
  VerticalAlignment: () => VerticalAlignment,
  VerticalAlignmentAlignment: () => VerticalAlignmentAlignment,
  ViewMode: () => ViewMode,
  ViewRole: () => ViewRole,
  VisitorType: () => VisitorType,
  VoteRole: () => VoteRole,
  WebhookIdentityType: () => WebhookIdentityType,
  Width: () => Width,
  WidthType: () => WidthType,
  resolveNotificationConfig: () => resolveNotificationConfig2,
  triggerNotification: () => triggerNotification2,
  upsertNotificationConfig: () => upsertNotificationConfig2
});
module.exports = __toCommonJS(index_typings_exports);

// src/events-notifications-v2-notification-config-notifications.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/events-notifications-v2-notification-config-notifications.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEventsNotificationsV2NotificationConfigManagementUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/events-notifications",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/events-notifications",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/events-notifications",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/events-notifications",
        destPath: ""
      },
      {
        srcPath: "/events/v2/notifications",
        destPath: "/v2/notifications"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
function resolveWixEventsNotificationsV2NotificationServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/events-notifications",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/events-notifications",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/events-notifications",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/events-notifications",
        destPath: ""
      },
      {
        srcPath: "/events/v2/notifications",
        destPath: "/v2/notifications"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_events_notifications";
function triggerNotification(payload) {
  function __triggerNotification({ host }) {
    const metadata = {
      entityFqdn: "wix.events.notifications.v2.notification_config",
      method: "POST",
      methodFqn: "wix.events.notifications.v2.NotificationService.TriggerNotification",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsNotificationsV2NotificationServiceUrl({
        protoPath: "/v2/notifications/trigger",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __triggerNotification;
}
function resolveNotificationConfig(payload) {
  function __resolveNotificationConfig({ host }) {
    const metadata = {
      entityFqdn: "wix.events.notifications.v2.notification_config",
      method: "GET",
      methodFqn: "wix.events.notifications.v2.NotificationConfigManagement.ResolveNotificationConfig",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsNotificationsV2NotificationConfigManagementUrl({
        protoPath: "/v2/notification-configs/resolve/{notificationConfigId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "notificationConfig.createdDate" },
            { path: "notificationConfig.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __resolveNotificationConfig;
}
function upsertNotificationConfig(payload) {
  function __upsertNotificationConfig({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "notificationConfig.createdDate" },
          { path: "notificationConfig.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.notifications.v2.notification_config",
      method: "POST",
      methodFqn: "wix.events.notifications.v2.NotificationConfigManagement.UpsertNotificationConfig",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsNotificationsV2NotificationConfigManagementUrl({
        protoPath: "/v2/notification-configs/{notificationConfig.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "notificationConfig.createdDate" },
            { path: "notificationConfig.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __upsertNotificationConfig;
}

// src/events-notifications-v2-notification-config-notifications.universal.ts
var EmailNotificationType = /* @__PURE__ */ ((EmailNotificationType2) => {
  EmailNotificationType2["UNKNOWN_NOTIFICATION_TYPE"] = "UNKNOWN_NOTIFICATION_TYPE";
  EmailNotificationType2["EMAIL_RSVP_CONFIRMATION"] = "EMAIL_RSVP_CONFIRMATION";
  EmailNotificationType2["EMAIL_NEW_SPOTS_AVAILABLE"] = "EMAIL_NEW_SPOTS_AVAILABLE";
  EmailNotificationType2["EMAIL_EVENT_CANCELATION"] = "EMAIL_EVENT_CANCELATION";
  EmailNotificationType2["EMAIL_UPCOMING_EVENT_REMINDER"] = "EMAIL_UPCOMING_EVENT_REMINDER";
  EmailNotificationType2["EMAIL_ORDER_CONFIRMATION"] = "EMAIL_ORDER_CONFIRMATION";
  EmailNotificationType2["EMAIL_TICKET_CONFIRMATION"] = "EMAIL_TICKET_CONFIRMATION";
  EmailNotificationType2["EMAIL_INVOICE"] = "EMAIL_INVOICE";
  return EmailNotificationType2;
})(EmailNotificationType || {});
var EmailNotificationReminderTime = /* @__PURE__ */ ((EmailNotificationReminderTime2) => {
  EmailNotificationReminderTime2["UNKNOWN_REMINDER_TIME"] = "UNKNOWN_REMINDER_TIME";
  EmailNotificationReminderTime2["INSTANT"] = "INSTANT";
  EmailNotificationReminderTime2["BEFORE_1_DAY"] = "BEFORE_1_DAY";
  EmailNotificationReminderTime2["BEFORE_3_DAYS"] = "BEFORE_3_DAYS";
  EmailNotificationReminderTime2["BEFORE_1_WEEK"] = "BEFORE_1_WEEK";
  EmailNotificationReminderTime2["BEFORE_1_HOUR"] = "BEFORE_1_HOUR";
  EmailNotificationReminderTime2["BEFORE_30_MINUTES"] = "BEFORE_30_MINUTES";
  return EmailNotificationReminderTime2;
})(EmailNotificationReminderTime || {});
var PushNotificationType = /* @__PURE__ */ ((PushNotificationType2) => {
  PushNotificationType2["UNKNOWN_PUSH_TYPE"] = "UNKNOWN_PUSH_TYPE";
  PushNotificationType2["PUSH_EVENT_UPDATED"] = "PUSH_EVENT_UPDATED";
  PushNotificationType2["PUSH_EVENT_CANCELATION"] = "PUSH_EVENT_CANCELATION";
  PushNotificationType2["PUSH_UPCOMING_EVENT_REMINDER"] = "PUSH_UPCOMING_EVENT_REMINDER";
  PushNotificationType2["PUSH_EVENT_CREATED"] = "PUSH_EVENT_CREATED";
  return PushNotificationType2;
})(PushNotificationType || {});
var LocationType = /* @__PURE__ */ ((LocationType2) => {
  LocationType2["VENUE"] = "VENUE";
  LocationType2["ONLINE"] = "ONLINE";
  return LocationType2;
})(LocationType || {});
var SubdivisionType = /* @__PURE__ */ ((SubdivisionType2) => {
  SubdivisionType2["UNKNOWN_SUBDIVISION_TYPE"] = "UNKNOWN_SUBDIVISION_TYPE";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_1"] = "ADMINISTRATIVE_AREA_LEVEL_1";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_2"] = "ADMINISTRATIVE_AREA_LEVEL_2";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_3"] = "ADMINISTRATIVE_AREA_LEVEL_3";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_4"] = "ADMINISTRATIVE_AREA_LEVEL_4";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_5"] = "ADMINISTRATIVE_AREA_LEVEL_5";
  SubdivisionType2["COUNTRY"] = "COUNTRY";
  return SubdivisionType2;
})(SubdivisionType || {});
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["ONE_TIME"] = "ONE_TIME";
  Status2["RECURRING"] = "RECURRING";
  Status2["RECURRING_NEXT"] = "RECURRING_NEXT";
  Status2["RECURRING_LAST_ENDED"] = "RECURRING_LAST_ENDED";
  Status2["RECURRING_LAST_CANCELED"] = "RECURRING_LAST_CANCELED";
  return Status2;
})(Status || {});
var EventStatus = /* @__PURE__ */ ((EventStatus2) => {
  EventStatus2["SCHEDULED"] = "SCHEDULED";
  EventStatus2["STARTED"] = "STARTED";
  EventStatus2["ENDED"] = "ENDED";
  EventStatus2["CANCELED"] = "CANCELED";
  return EventStatus2;
})(EventStatus || {});
var EventType = /* @__PURE__ */ ((EventType2) => {
  EventType2["NA_EVENT_TYPE"] = "NA_EVENT_TYPE";
  EventType2["RSVP"] = "RSVP";
  EventType2["TICKETS"] = "TICKETS";
  EventType2["EXTERNAL"] = "EXTERNAL";
  EventType2["NO_REGISTRATION"] = "NO_REGISTRATION";
  return EventType2;
})(EventType || {});
var RegistrationStatus = /* @__PURE__ */ ((RegistrationStatus2) => {
  RegistrationStatus2["NA_REGISTRATION_STATUS"] = "NA_REGISTRATION_STATUS";
  RegistrationStatus2["CLOSED"] = "CLOSED";
  RegistrationStatus2["CLOSED_MANUALLY"] = "CLOSED_MANUALLY";
  RegistrationStatus2["OPEN_RSVP"] = "OPEN_RSVP";
  RegistrationStatus2["OPEN_RSVP_WAITLIST"] = "OPEN_RSVP_WAITLIST";
  RegistrationStatus2["OPEN_TICKETS"] = "OPEN_TICKETS";
  RegistrationStatus2["OPEN_EXTERNAL"] = "OPEN_EXTERNAL";
  RegistrationStatus2["SCHEDULED_RSVP"] = "SCHEDULED_RSVP";
  return RegistrationStatus2;
})(RegistrationStatus || {});
var RsvpStatusOptions = /* @__PURE__ */ ((RsvpStatusOptions2) => {
  RsvpStatusOptions2["YES_ONLY"] = "YES_ONLY";
  RsvpStatusOptions2["YES_AND_NO"] = "YES_AND_NO";
  return RsvpStatusOptions2;
})(RsvpStatusOptions || {});
var TaxType = /* @__PURE__ */ ((TaxType2) => {
  TaxType2["INCLUDED"] = "INCLUDED";
  TaxType2["ADDED"] = "ADDED";
  TaxType2["ADDED_AT_CHECKOUT"] = "ADDED_AT_CHECKOUT";
  return TaxType2;
})(TaxType || {});
var CheckoutType = /* @__PURE__ */ ((CheckoutType2) => {
  CheckoutType2["UNKNOWN_CHECKOUT_TYPE"] = "UNKNOWN_CHECKOUT_TYPE";
  CheckoutType2["EVENTS_APP"] = "EVENTS_APP";
  CheckoutType2["ECOMM_PLATFORM"] = "ECOMM_PLATFORM";
  return CheckoutType2;
})(CheckoutType || {});
var VisitorType = /* @__PURE__ */ ((VisitorType2) => {
  VisitorType2["VISITOR"] = "VISITOR";
  VisitorType2["MEMBER"] = "MEMBER";
  VisitorType2["VISITOR_OR_MEMBER"] = "VISITOR_OR_MEMBER";
  return VisitorType2;
})(VisitorType || {});
var InputControlType = /* @__PURE__ */ ((InputControlType2) => {
  InputControlType2["INPUT"] = "INPUT";
  InputControlType2["TEXTAREA"] = "TEXTAREA";
  InputControlType2["DROPDOWN"] = "DROPDOWN";
  InputControlType2["RADIO"] = "RADIO";
  InputControlType2["CHECKBOX"] = "CHECKBOX";
  InputControlType2["NAME"] = "NAME";
  InputControlType2["GUEST_CONTROL"] = "GUEST_CONTROL";
  InputControlType2["ADDRESS_SHORT"] = "ADDRESS_SHORT";
  InputControlType2["ADDRESS_FULL"] = "ADDRESS_FULL";
  InputControlType2["DATE"] = "DATE";
  return InputControlType2;
})(InputControlType || {});
var ValueType = /* @__PURE__ */ ((ValueType2) => {
  ValueType2["TEXT"] = "TEXT";
  ValueType2["NUMBER"] = "NUMBER";
  ValueType2["TEXT_ARRAY"] = "TEXT_ARRAY";
  ValueType2["DATE_TIME"] = "DATE_TIME";
  ValueType2["ADDRESS"] = "ADDRESS";
  return ValueType2;
})(ValueType || {});
var ConferenceType = /* @__PURE__ */ ((ConferenceType2) => {
  ConferenceType2["MEETING"] = "MEETING";
  ConferenceType2["WEBINAR"] = "WEBINAR";
  return ConferenceType2;
})(ConferenceType || {});
var State = /* @__PURE__ */ ((State2) => {
  State2["MANUAL"] = "MANUAL";
  State2["AUTO"] = "AUTO";
  State2["RECURRING_EVENT"] = "RECURRING_EVENT";
  State2["HIDDEN"] = "HIDDEN";
  return State2;
})(State || {});
var NodeType = /* @__PURE__ */ ((NodeType2) => {
  NodeType2["PARAGRAPH"] = "PARAGRAPH";
  NodeType2["TEXT"] = "TEXT";
  NodeType2["HEADING"] = "HEADING";
  NodeType2["BULLETED_LIST"] = "BULLETED_LIST";
  NodeType2["ORDERED_LIST"] = "ORDERED_LIST";
  NodeType2["LIST_ITEM"] = "LIST_ITEM";
  NodeType2["BLOCKQUOTE"] = "BLOCKQUOTE";
  NodeType2["CODE_BLOCK"] = "CODE_BLOCK";
  NodeType2["VIDEO"] = "VIDEO";
  NodeType2["DIVIDER"] = "DIVIDER";
  NodeType2["FILE"] = "FILE";
  NodeType2["GALLERY"] = "GALLERY";
  NodeType2["GIF"] = "GIF";
  NodeType2["HTML"] = "HTML";
  NodeType2["IMAGE"] = "IMAGE";
  NodeType2["LINK_PREVIEW"] = "LINK_PREVIEW";
  NodeType2["MAP"] = "MAP";
  NodeType2["POLL"] = "POLL";
  NodeType2["APP_EMBED"] = "APP_EMBED";
  NodeType2["BUTTON"] = "BUTTON";
  NodeType2["COLLAPSIBLE_LIST"] = "COLLAPSIBLE_LIST";
  NodeType2["TABLE"] = "TABLE";
  NodeType2["EMBED"] = "EMBED";
  NodeType2["COLLAPSIBLE_ITEM"] = "COLLAPSIBLE_ITEM";
  NodeType2["COLLAPSIBLE_ITEM_TITLE"] = "COLLAPSIBLE_ITEM_TITLE";
  NodeType2["COLLAPSIBLE_ITEM_BODY"] = "COLLAPSIBLE_ITEM_BODY";
  NodeType2["TABLE_CELL"] = "TABLE_CELL";
  NodeType2["TABLE_ROW"] = "TABLE_ROW";
  NodeType2["EXTERNAL"] = "EXTERNAL";
  NodeType2["AUDIO"] = "AUDIO";
  NodeType2["CAPTION"] = "CAPTION";
  NodeType2["LAYOUT"] = "LAYOUT";
  NodeType2["LAYOUT_CELL"] = "LAYOUT_CELL";
  return NodeType2;
})(NodeType || {});
var WidthType = /* @__PURE__ */ ((WidthType2) => {
  WidthType2["CONTENT"] = "CONTENT";
  WidthType2["SMALL"] = "SMALL";
  WidthType2["ORIGINAL"] = "ORIGINAL";
  WidthType2["FULL_WIDTH"] = "FULL_WIDTH";
  return WidthType2;
})(WidthType || {});
var PluginContainerDataAlignment = /* @__PURE__ */ ((PluginContainerDataAlignment2) => {
  PluginContainerDataAlignment2["CENTER"] = "CENTER";
  PluginContainerDataAlignment2["LEFT"] = "LEFT";
  PluginContainerDataAlignment2["RIGHT"] = "RIGHT";
  return PluginContainerDataAlignment2;
})(PluginContainerDataAlignment || {});
var ButtonDataType = /* @__PURE__ */ ((ButtonDataType2) => {
  ButtonDataType2["LINK"] = "LINK";
  ButtonDataType2["ACTION"] = "ACTION";
  return ButtonDataType2;
})(ButtonDataType || {});
var Target = /* @__PURE__ */ ((Target2) => {
  Target2["SELF"] = "SELF";
  Target2["BLANK"] = "BLANK";
  Target2["PARENT"] = "PARENT";
  Target2["TOP"] = "TOP";
  return Target2;
})(Target || {});
var TextAlignment = /* @__PURE__ */ ((TextAlignment2) => {
  TextAlignment2["AUTO"] = "AUTO";
  TextAlignment2["LEFT"] = "LEFT";
  TextAlignment2["RIGHT"] = "RIGHT";
  TextAlignment2["CENTER"] = "CENTER";
  TextAlignment2["JUSTIFY"] = "JUSTIFY";
  return TextAlignment2;
})(TextAlignment || {});
var LineStyle = /* @__PURE__ */ ((LineStyle2) => {
  LineStyle2["SINGLE"] = "SINGLE";
  LineStyle2["DOUBLE"] = "DOUBLE";
  LineStyle2["DASHED"] = "DASHED";
  LineStyle2["DOTTED"] = "DOTTED";
  return LineStyle2;
})(LineStyle || {});
var Width = /* @__PURE__ */ ((Width2) => {
  Width2["LARGE"] = "LARGE";
  Width2["MEDIUM"] = "MEDIUM";
  Width2["SMALL"] = "SMALL";
  return Width2;
})(Width || {});
var DividerDataAlignment = /* @__PURE__ */ ((DividerDataAlignment2) => {
  DividerDataAlignment2["CENTER"] = "CENTER";
  DividerDataAlignment2["LEFT"] = "LEFT";
  DividerDataAlignment2["RIGHT"] = "RIGHT";
  return DividerDataAlignment2;
})(DividerDataAlignment || {});
var ViewMode = /* @__PURE__ */ ((ViewMode2) => {
  ViewMode2["NONE"] = "NONE";
  ViewMode2["FULL"] = "FULL";
  ViewMode2["MINI"] = "MINI";
  return ViewMode2;
})(ViewMode || {});
var LayoutType = /* @__PURE__ */ ((LayoutType2) => {
  LayoutType2["COLLAGE"] = "COLLAGE";
  LayoutType2["MASONRY"] = "MASONRY";
  LayoutType2["GRID"] = "GRID";
  LayoutType2["THUMBNAIL"] = "THUMBNAIL";
  LayoutType2["SLIDER"] = "SLIDER";
  LayoutType2["SLIDESHOW"] = "SLIDESHOW";
  LayoutType2["PANORAMA"] = "PANORAMA";
  LayoutType2["COLUMN"] = "COLUMN";
  LayoutType2["MAGIC"] = "MAGIC";
  LayoutType2["FULLSIZE"] = "FULLSIZE";
  return LayoutType2;
})(LayoutType || {});
var Orientation = /* @__PURE__ */ ((Orientation2) => {
  Orientation2["ROWS"] = "ROWS";
  Orientation2["COLUMNS"] = "COLUMNS";
  return Orientation2;
})(Orientation || {});
var Crop = /* @__PURE__ */ ((Crop2) => {
  Crop2["FILL"] = "FILL";
  Crop2["FIT"] = "FIT";
  return Crop2;
})(Crop || {});
var ThumbnailsAlignment = /* @__PURE__ */ ((ThumbnailsAlignment2) => {
  ThumbnailsAlignment2["TOP"] = "TOP";
  ThumbnailsAlignment2["RIGHT"] = "RIGHT";
  ThumbnailsAlignment2["BOTTOM"] = "BOTTOM";
  ThumbnailsAlignment2["LEFT"] = "LEFT";
  ThumbnailsAlignment2["NONE"] = "NONE";
  return ThumbnailsAlignment2;
})(ThumbnailsAlignment || {});
var GIFType = /* @__PURE__ */ ((GIFType2) => {
  GIFType2["NORMAL"] = "NORMAL";
  GIFType2["STICKER"] = "STICKER";
  return GIFType2;
})(GIFType || {});
var Source = /* @__PURE__ */ ((Source2) => {
  Source2["HTML"] = "HTML";
  Source2["ADSENSE"] = "ADSENSE";
  Source2["AI"] = "AI";
  return Source2;
})(Source || {});
var StylesPosition = /* @__PURE__ */ ((StylesPosition2) => {
  StylesPosition2["START"] = "START";
  StylesPosition2["END"] = "END";
  StylesPosition2["TOP"] = "TOP";
  StylesPosition2["HIDDEN"] = "HIDDEN";
  return StylesPosition2;
})(StylesPosition || {});
var MapType = /* @__PURE__ */ ((MapType2) => {
  MapType2["ROADMAP"] = "ROADMAP";
  MapType2["SATELITE"] = "SATELITE";
  MapType2["HYBRID"] = "HYBRID";
  MapType2["TERRAIN"] = "TERRAIN";
  return MapType2;
})(MapType || {});
var ViewRole = /* @__PURE__ */ ((ViewRole2) => {
  ViewRole2["CREATOR"] = "CREATOR";
  ViewRole2["VOTERS"] = "VOTERS";
  ViewRole2["EVERYONE"] = "EVERYONE";
  return ViewRole2;
})(ViewRole || {});
var VoteRole = /* @__PURE__ */ ((VoteRole2) => {
  VoteRole2["SITE_MEMBERS"] = "SITE_MEMBERS";
  VoteRole2["ALL"] = "ALL";
  return VoteRole2;
})(VoteRole || {});
var PollLayoutType = /* @__PURE__ */ ((PollLayoutType2) => {
  PollLayoutType2["LIST"] = "LIST";
  PollLayoutType2["GRID"] = "GRID";
  return PollLayoutType2;
})(PollLayoutType || {});
var PollLayoutDirection = /* @__PURE__ */ ((PollLayoutDirection2) => {
  PollLayoutDirection2["LTR"] = "LTR";
  PollLayoutDirection2["RTL"] = "RTL";
  return PollLayoutDirection2;
})(PollLayoutDirection || {});
var BackgroundType = /* @__PURE__ */ ((BackgroundType2) => {
  BackgroundType2["COLOR"] = "COLOR";
  BackgroundType2["IMAGE"] = "IMAGE";
  BackgroundType2["GRADIENT"] = "GRADIENT";
  return BackgroundType2;
})(BackgroundType || {});
var DecorationType = /* @__PURE__ */ ((DecorationType2) => {
  DecorationType2["BOLD"] = "BOLD";
  DecorationType2["ITALIC"] = "ITALIC";
  DecorationType2["UNDERLINE"] = "UNDERLINE";
  DecorationType2["SPOILER"] = "SPOILER";
  DecorationType2["ANCHOR"] = "ANCHOR";
  DecorationType2["MENTION"] = "MENTION";
  DecorationType2["LINK"] = "LINK";
  DecorationType2["COLOR"] = "COLOR";
  DecorationType2["FONT_SIZE"] = "FONT_SIZE";
  DecorationType2["EXTERNAL"] = "EXTERNAL";
  DecorationType2["STRIKETHROUGH"] = "STRIKETHROUGH";
  DecorationType2["SUPERSCRIPT"] = "SUPERSCRIPT";
  DecorationType2["SUBSCRIPT"] = "SUBSCRIPT";
  return DecorationType2;
})(DecorationType || {});
var FontType = /* @__PURE__ */ ((FontType2) => {
  FontType2["PX"] = "PX";
  FontType2["EM"] = "EM";
  return FontType2;
})(FontType || {});
var Position = /* @__PURE__ */ ((Position2) => {
  Position2["START"] = "START";
  Position2["END"] = "END";
  Position2["TOP"] = "TOP";
  return Position2;
})(Position || {});
var AspectRatio = /* @__PURE__ */ ((AspectRatio2) => {
  AspectRatio2["SQUARE"] = "SQUARE";
  AspectRatio2["RECTANGLE"] = "RECTANGLE";
  return AspectRatio2;
})(AspectRatio || {});
var Resizing = /* @__PURE__ */ ((Resizing2) => {
  Resizing2["FILL"] = "FILL";
  Resizing2["FIT"] = "FIT";
  return Resizing2;
})(Resizing || {});
var Placement = /* @__PURE__ */ ((Placement2) => {
  Placement2["IMAGE"] = "IMAGE";
  Placement2["PRODUCT_INFO"] = "PRODUCT_INFO";
  return Placement2;
})(Placement || {});
var CardStylesType = /* @__PURE__ */ ((CardStylesType2) => {
  CardStylesType2["CONTAINED"] = "CONTAINED";
  CardStylesType2["FRAMELESS"] = "FRAMELESS";
  return CardStylesType2;
})(CardStylesType || {});
var Alignment = /* @__PURE__ */ ((Alignment2) => {
  Alignment2["START"] = "START";
  Alignment2["CENTER"] = "CENTER";
  Alignment2["END"] = "END";
  return Alignment2;
})(Alignment || {});
var Layout = /* @__PURE__ */ ((Layout2) => {
  Layout2["STACKED"] = "STACKED";
  Layout2["SIDE_BY_SIDE"] = "SIDE_BY_SIDE";
  return Layout2;
})(Layout || {});
var AppType = /* @__PURE__ */ ((AppType2) => {
  AppType2["PRODUCT"] = "PRODUCT";
  AppType2["EVENT"] = "EVENT";
  AppType2["BOOKING"] = "BOOKING";
  return AppType2;
})(AppType || {});
var InitialExpandedItems = /* @__PURE__ */ ((InitialExpandedItems2) => {
  InitialExpandedItems2["FIRST"] = "FIRST";
  InitialExpandedItems2["ALL"] = "ALL";
  InitialExpandedItems2["NONE"] = "NONE";
  return InitialExpandedItems2;
})(InitialExpandedItems || {});
var Direction = /* @__PURE__ */ ((Direction2) => {
  Direction2["LTR"] = "LTR";
  Direction2["RTL"] = "RTL";
  return Direction2;
})(Direction || {});
var VerticalAlignment = /* @__PURE__ */ ((VerticalAlignment2) => {
  VerticalAlignment2["TOP"] = "TOP";
  VerticalAlignment2["MIDDLE"] = "MIDDLE";
  VerticalAlignment2["BOTTOM"] = "BOTTOM";
  return VerticalAlignment2;
})(VerticalAlignment || {});
var NullValue = /* @__PURE__ */ ((NullValue2) => {
  NullValue2["NULL_VALUE"] = "NULL_VALUE";
  return NullValue2;
})(NullValue || {});
var Scaling = /* @__PURE__ */ ((Scaling2) => {
  Scaling2["AUTO"] = "AUTO";
  Scaling2["CONTAIN"] = "CONTAIN";
  Scaling2["COVER"] = "COVER";
  return Scaling2;
})(Scaling || {});
var ImagePosition = /* @__PURE__ */ ((ImagePosition2) => {
  ImagePosition2["CENTER"] = "CENTER";
  ImagePosition2["CENTER_LEFT"] = "CENTER_LEFT";
  ImagePosition2["CENTER_RIGHT"] = "CENTER_RIGHT";
  ImagePosition2["TOP"] = "TOP";
  ImagePosition2["TOP_LEFT"] = "TOP_LEFT";
  ImagePosition2["TOP_RIGHT"] = "TOP_RIGHT";
  ImagePosition2["BOTTOM"] = "BOTTOM";
  ImagePosition2["BOTTOM_LEFT"] = "BOTTOM_LEFT";
  ImagePosition2["BOTTOM_RIGHT"] = "BOTTOM_RIGHT";
  return ImagePosition2;
})(ImagePosition || {});
var VerticalAlignmentAlignment = /* @__PURE__ */ ((VerticalAlignmentAlignment2) => {
  VerticalAlignmentAlignment2["TOP"] = "TOP";
  VerticalAlignmentAlignment2["MIDDLE"] = "MIDDLE";
  VerticalAlignmentAlignment2["BOTTOM"] = "BOTTOM";
  return VerticalAlignmentAlignment2;
})(VerticalAlignmentAlignment || {});
var ResponsivenessBehaviour = /* @__PURE__ */ ((ResponsivenessBehaviour2) => {
  ResponsivenessBehaviour2["STACK"] = "STACK";
  ResponsivenessBehaviour2["WRAP"] = "WRAP";
  return ResponsivenessBehaviour2;
})(ResponsivenessBehaviour || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNKNOWN_BADGE_TYPE"] = "UNKNOWN_BADGE_TYPE";
  Type2["FIRST_PRIORITY"] = "FIRST_PRIORITY";
  Type2["SECOND_PRIORITY"] = "SECOND_PRIORITY";
  Type2["THIRD_PRIORITY"] = "THIRD_PRIORITY";
  return Type2;
})(Type || {});
var OrderStatus = /* @__PURE__ */ ((OrderStatus2) => {
  OrderStatus2["NA_ORDER_STATUS"] = "NA_ORDER_STATUS";
  OrderStatus2["FREE"] = "FREE";
  OrderStatus2["PENDING"] = "PENDING";
  OrderStatus2["PAID"] = "PAID";
  OrderStatus2["OFFLINE_PENDING"] = "OFFLINE_PENDING";
  OrderStatus2["INITIATED"] = "INITIATED";
  OrderStatus2["CANCELED"] = "CANCELED";
  OrderStatus2["DECLINED"] = "DECLINED";
  OrderStatus2["AUTHORIZED"] = "AUTHORIZED";
  OrderStatus2["VOIDED"] = "VOIDED";
  OrderStatus2["PARTIALLY_PAID"] = "PARTIALLY_PAID";
  return OrderStatus2;
})(OrderStatus || {});
var FeeName = /* @__PURE__ */ ((FeeName2) => {
  FeeName2["WIX_FEE"] = "WIX_FEE";
  return FeeName2;
})(FeeName || {});
var FeeType = /* @__PURE__ */ ((FeeType2) => {
  FeeType2["FEE_ADDED"] = "FEE_ADDED";
  FeeType2["FEE_INCLUDED"] = "FEE_INCLUDED";
  FeeType2["FEE_ADDED_AT_CHECKOUT"] = "FEE_ADDED_AT_CHECKOUT";
  return FeeType2;
})(FeeType || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function triggerNotification2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    emailNotificationType: options?.emailNotificationType,
    pushNotificationType: options?.pushNotificationType,
    rsvpGuest: options?.rsvpGuest,
    orderGuest: options?.orderGuest,
    eventGuests: options?.eventGuests,
    ticketGuest: options?.ticketGuest
  });
  const reqOpts = triggerNotification(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          emailNotificationType: "$[0].emailNotificationType",
          pushNotificationType: "$[0].pushNotificationType",
          rsvpGuest: "$[0].rsvpGuest",
          orderGuest: "$[0].orderGuest",
          eventGuests: "$[0].eventGuests",
          ticketGuest: "$[0].ticketGuest"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function resolveNotificationConfig2(notificationConfigId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    notificationConfigId
  });
  const reqOpts = resolveNotificationConfig(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { notificationConfigId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["notificationConfigId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function upsertNotificationConfig2(_id, notificationConfig) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    notificationConfig: { ...notificationConfig, id: _id }
  });
  const reqOpts = upsertNotificationConfig(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { notificationConfig: "$[1]" },
        explicitPathsToArguments: { "notificationConfig.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "notificationConfig"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Alignment,
  AppType,
  AspectRatio,
  BackgroundType,
  ButtonDataType,
  CardStylesType,
  CheckoutType,
  ConferenceType,
  Crop,
  DecorationType,
  Direction,
  DividerDataAlignment,
  EmailNotificationReminderTime,
  EmailNotificationType,
  EventStatus,
  EventType,
  FeeName,
  FeeType,
  FontType,
  GIFType,
  ImagePosition,
  InitialExpandedItems,
  InputControlType,
  Layout,
  LayoutType,
  LineStyle,
  LocationType,
  MapType,
  NodeType,
  NullValue,
  OrderStatus,
  Orientation,
  Placement,
  PluginContainerDataAlignment,
  PollLayoutDirection,
  PollLayoutType,
  Position,
  PushNotificationType,
  RegistrationStatus,
  Resizing,
  ResponsivenessBehaviour,
  RsvpStatusOptions,
  Scaling,
  Source,
  State,
  Status,
  StylesPosition,
  SubdivisionType,
  Target,
  TaxType,
  TextAlignment,
  ThumbnailsAlignment,
  Type,
  ValueType,
  VerticalAlignment,
  VerticalAlignmentAlignment,
  ViewMode,
  ViewRole,
  VisitorType,
  VoteRole,
  WebhookIdentityType,
  Width,
  WidthType,
  resolveNotificationConfig,
  triggerNotification,
  upsertNotificationConfig
});
