// @wix/ricos/dist/cjs/plugin-link/anchor-converter.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var anchor_converter_exports = {};
__export(anchor_converter_exports, {
  anchorConverter: () => anchorConverter
});
module.exports = __toCommonJS(anchor_converter_exports);
var import_v1 = require("./../ricos-schema/legacy/wix/rich_content/v1/index.js");
const anchorConverter = {
  fromPm: {
    type: import_v1.Decoration_Type.ANCHOR,
    convert: (mark) => {
      const { attrs } = mark;
      return [
        {
          type: import_v1.Decoration_Type.ANCHOR,
          anchorData: { ...attrs }
        }
      ];
    }
  },
  toPm: {
    type: import_v1.Decoration_Type.ANCHOR,
    convert: (decoration) => {
      const { anchorData } = decoration;
      return {
        type: import_v1.Decoration_Type.ANCHOR,
        attrs: { ...anchorData }
      };
    }
  }
};
