// @wix/design-system/dist/cjs/AreaChart/AreaChart.st.css.js

require("../Foundation/stylable/colors.st.css.js");
require("../Foundation/stylable/easing.st.css.js");
const { classesRuntime, statesRuntime } = require("./../stylable-cjs-runtime.js");



var _namespace_ = "wds_1_226_0_AreaChart";
var _style_ = classesRuntime.bind(null, _namespace_);

module.exports.cssStates = statesRuntime.bind(null, _namespace_);
module.exports.style = _style_;
module.exports.st = _style_;

module.exports.namespace = _namespace_;
module.exports.classes = {"root":"wds_1_226_0_AreaChart__root","lineChart":"wds_1_226_0_AreaChart__lineChart","lineChartCursor":"wds_1_226_0_AreaChart__lineChartCursor","lineChartAxis":"wds_1_226_0_AreaChart__lineChartAxis"};
module.exports.keyframes = {}; 
module.exports.layers = {};
module.exports.containers = {};
module.exports.stVars = {"canvas-top-gap":"6px","canvas-bottom-gap":"31px","datasetBg":"rgba(51, 112, 251, 0.06)","pointBorderColor":"rgba(51, 112, 251, 0)","pointBackgroundColor":"rgba(51, 112, 251, 0)","datasetLineColor":"#3370FB","pointHoverBackgroundColor":"#FFFFFF","pointHoverBorderColor":"#3370FB","gridLineZeroLineColor":"#7A92A5","gridLineFontColor":"#CBD3DC","tooltipBg":"#162D3D"}; 
module.exports.vars = {};
