// @wix/design-system/dist/cjs/TopBanner/components/ActionLink/ActionLink.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.ActionLink = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _react = _interopRequireDefault(require("react"));
var _TextButton = _interopRequireDefault(require("../../../TextButton"));
var _excluded = ["dataHook", "children"];
var _jsxFileName = "/home/builduser/work/57e038ea7326c1ec/packages/wix-design-system/dist/cjs/TopBanner/components/ActionLink/ActionLink.tsx";
var ActionLink = _ref => {
  var {
      dataHook,
      children
    } = _ref,
    rest = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  return /*#__PURE__*/_react.default.createElement(_TextButton.default, (0, _extends2.default)({
    skin: "light",
    underline: "always",
    dataHook: dataHook
  }, rest, {
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 10,
      columnNumber: 3
    }
  }), children);
};
exports.ActionLink = ActionLink;
