// @wix/headless-pricing-plans/dist/services/plan-service.js
import { defineService, implementService } from '@wix/services-definitions';
import { SignalsServiceDefinition, } from '@wix/services-definitions/core-services/signals';
import { plansV3 } from '@wix/pricing-plans';
export const PRICING_PLANS_APP_ID = '1522827f-c56c-a5c9-2ac9-00f9e6ae12d3';
export const PlanServiceDefinition = defineService('planService');
export const PlanService = implementService.withConfig()(PlanServiceDefinition, ({ getService, config }) => {
    const signalsService = getService(SignalsServiceDefinition);
    const isLoadingSignal = signalsService.signal(false);
    const errorSignal = signalsService.signal(null);
    const configHasPlan = 'plan' in config;
    const planSignal = signalsService.signal(configHasPlan ? config.plan : null);
    if (!configHasPlan) {
        loadPlan(config.planId);
    }
    async function loadPlan(planId) {
        isLoadingSignal.set(true);
        errorSignal.set(null);
        try {
            const plan = await fetchAndEnhancePlan(planId);
            planSignal.set(plan);
        }
        catch (error) {
            errorSignal.set(error instanceof Error ? error : new Error(error));
        }
        finally {
            isLoadingSignal.set(false);
        }
    }
    return {
        planSignal: planSignal,
        isLoadingSignal: isLoadingSignal,
        errorSignal: errorSignal,
    };
});
export function enhancePlanData(plan) {
    return {
        ...plan,
        enhancedData: {
            price: getPlanPriceData(plan),
            additionalFees: getFormattedAdditionalFeesData(plan),
            recurrence: getPlanRecurrence(plan),
            duration: getPlanDuration(plan),
            freeTrialDays: getPlanFreeTrialDays(plan),
        },
    };
}
function getPlanPriceData(plan) {
    const pricingVariant = plan.pricingVariants?.[0];
    if (!pricingVariant) {
        throw new Error('No pricing variant found');
    }
    const priceAmount = Number(pricingVariant?.pricingStrategies?.[0]?.flatRate?.amount);
    if (Number.isNaN(priceAmount)) {
        throw new Error('No price amount found');
    }
    // TODO: Remove this once we have a generic price formatting component
    const formattedPrice = new Intl.NumberFormat('en-US', {
        style: 'currency',
        currency: plan.currency,
    }).format(priceAmount);
    return {
        pricingVariantId: pricingVariant._id,
        amount: priceAmount,
        currency: plan.currency,
        formattedPrice,
    };
}
function getFormattedAdditionalFeesData(plan) {
    const additionalFees = plan.pricingVariants?.[0]?.fees;
    if (!additionalFees) {
        return [];
    }
    return additionalFees
        .map((additionalFee) => {
        const amount = Number(additionalFee.fixedAmountOptions?.amount);
        if (Number.isNaN(amount)) {
            return null;
        }
        const formattedFee = new Intl.NumberFormat('en-US', {
            style: 'currency',
            currency: plan.currency,
        }).format(amount);
        return {
            name: additionalFee.name ?? '',
            amount: amount,
            currency: plan.currency,
            formattedFee,
        };
    })
        .filter((fee) => !!fee);
}
function getPlanRecurrence(plan) {
    if (!isPlanRecurring(plan)) {
        return null;
    }
    const billingTerms = plan.pricingVariants?.[0]?.billingTerms;
    if (!billingTerms) {
        return null;
    }
    const count = Number(billingTerms.billingCycle?.count);
    const period = billingTerms.billingCycle?.period;
    return { count, period };
}
function getPlanFreeTrialDays(plan) {
    const pricingVariant = plan.pricingVariants?.[0];
    if (!pricingVariant) {
        throw new Error('No pricing variant found');
    }
    return pricingVariant.freeTrialDays ?? null;
}
function isPlanRecurring(plan) {
    const billingTerms = plan.pricingVariants?.[0]?.billingTerms;
    if (billingTerms?.endType === 'UNTIL_CANCELLED') {
        return !!billingTerms?.billingCycle;
    }
    if (billingTerms?.endType === 'CYCLES_COMPLETED') {
        return billingTerms?.cyclesCompletedDetails?.billingCycleCount !== '1';
    }
    return false;
}
function getPlanDuration(plan) {
    const billingTerms = plan.pricingVariants?.[0]?.billingTerms;
    if (!billingTerms || billingTerms.endType === 'UNTIL_CANCELLED') {
        return null;
    }
    if (isPlanRecurring(plan)) {
        const recurrence = getPlanRecurrence(plan);
        const totalCycleCount = Number(billingTerms.cyclesCompletedDetails?.billingCycleCount);
        return {
            count: recurrence.count * totalCycleCount,
            period: recurrence.period,
        };
    }
    return {
        count: Number(billingTerms.billingCycle?.count),
        period: billingTerms.billingCycle?.period,
    };
}
async function fetchAndEnhancePlan(planId) {
    try {
        const result = await plansV3
            .queryPlans()
            .eq('_id', planId)
            .eq('visibility', 'PUBLIC')
            .find();
        const [plan] = result.items;
        if (!plan) {
            // TODO: Is there an HttpError class where we could set status code?
            throw new Error(`Plan ${planId} not found`);
        }
        return enhancePlanData(plan);
    }
    catch (error) {
        console.log('Error fetching and enhancing plan data:', error);
        throw error;
    }
}
export async function loadPlanServiceConfig(planId) {
    const plan = await fetchAndEnhancePlan(planId);
    return { plan };
}
