// @wix/ricos/dist/cjs/plugin-link/toolbar/anchor-link-button.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var anchor_link_button_exports = {};
__export(anchor_link_button_exports, {
  AnchorLinkButton: () => AnchorLinkButton
});
module.exports = __toCommonJS(anchor_link_button_exports);
var import_classnames = __toESM(require("classnames"));
var import_react = __toESM(require("react"));
var import_ricos_context = require("./../../ricos-context/index.js");
var import_url_link_button_scss = __toESM(require("./url-link-button-scss.js"));
const AnchorLinkButton = ({ toolbarItem }) => {
  const { isMobile, t } = (0, import_react.useContext)(import_ricos_context.RicosContext);
  const dataHook = toolbarItem.presentation?.dataHook;
  return /* @__PURE__ */ import_react.default.createElement("div", { className: import_url_link_button_scss.default.toolbarUrlContainer }, /* @__PURE__ */ import_react.default.createElement(
    "button",
    {
      "data-hook": dataHook,
      className: (0, import_classnames.default)(import_url_link_button_scss.default.toolbarUrl, import_url_link_button_scss.default.toolbarUrlAnchor, isMobile && import_url_link_button_scss.default.isMobile),
      onClick: (e) => {
        e.preventDefault();
        toolbarItem.commands.click();
      },
      onMouseDown: (e) => {
        e.preventDefault();
      },
      type: "button",
      tabIndex: toolbarItem.presentation.tabIndex
    },
    t("LinkTo_Toolbar_GoTo")
  ));
};
