// @wix/auto_sdk_benefit-programs_items/build/internal/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  SortOrder: () => SortOrder,
  Type: () => Type,
  WebhookIdentityType: () => WebhookIdentityType,
  bulkCreateItems: () => bulkCreateItems2,
  bulkDeleteItems: () => bulkDeleteItems2,
  bulkDeleteItemsByFilter: () => bulkDeleteItemsByFilter2,
  bulkUpdateItems: () => bulkUpdateItems2,
  countItems: () => countItems2,
  createItem: () => createItem2,
  deleteItem: () => deleteItem2,
  getItem: () => getItem2,
  listItems: () => listItems2,
  queryItems: () => queryItems2,
  updateItem: () => updateItem2
});
module.exports = __toCommonJS(index_typings_exports);

// src/benefit-programs-v1-item-items.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/benefit-programs-v1-item-items.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixBenefitProgramsV1ItemItemServiceUrl(opts) {
  const domainToMappings = {
    _: [
      {
        srcPath: "/_api/benefit-programs-items",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-items",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-items",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/benefit-programs-items",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/benefit-programs-items",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/items",
        destPath: "/v1/bulk/items"
      },
      {
        srcPath: "/_api/benefit-programs/v1/items",
        destPath: "/v1/items"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-items",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-items",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/benefit-programs-items",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/benefit-programs/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/benefit-programs/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/benefit-programs/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_benefit-programs_items";
function createItem(payload) {
  function __createItem({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "item.createdDate" }, { path: "item.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.CreateItem",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/items",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "item.createdDate" }, { path: "item.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __createItem;
}
function bulkCreateItems(payload) {
  function __bulkCreateItems({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "items.createdDate" }, { path: "items.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.BulkCreateItems",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/bulk/items/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateItems;
}
function deleteItem(payload) {
  function __deleteItem({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.DeleteItem",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/items/{itemId}/delete",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __deleteItem;
}
function bulkDeleteItems(payload) {
  function __bulkDeleteItems({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.BulkDeleteItems",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/bulk/items/delete",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkDeleteItems;
}
function bulkDeleteItemsByFilter(payload) {
  function __bulkDeleteItemsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.BulkDeleteItemsByFilter",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/bulk/items/delete-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkDeleteItemsByFilter;
}
function updateItem(payload) {
  function __updateItem({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "item.createdDate" }, { path: "item.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "PATCH",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.UpdateItem",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/items/{item.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "item.createdDate" }, { path: "item.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __updateItem;
}
function bulkUpdateItems(payload) {
  function __bulkUpdateItems({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "items.fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "items.item.createdDate" },
          { path: "items.item.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.BulkUpdateItems",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/bulk/items/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateItems;
}
function getItem(payload) {
  function __getItem({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.GetItem",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/items/{itemId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "item.createdDate" }, { path: "item.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __getItem;
}
function listItems(payload) {
  function __listItems({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.ListItems",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/items",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "items.createdDate" },
            { path: "items.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listItems;
}
function queryItems(payload) {
  function __queryItems({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.QueryItems",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/items/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "items.createdDate" },
            { path: "items.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryItems;
}
function countItems(payload) {
  function __countItems({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.item",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.item.ItemService.CountItems",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ItemItemServiceUrl({
        protoPath: "/v1/items/count",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __countItems;
}

// src/benefit-programs-v1-item-items.universal.ts
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNKNOWN_FILTER"] = "UNKNOWN_FILTER";
  Type2["BY_ITEM_SET_ID_AND_REFERENCE"] = "BY_ITEM_SET_ID_AND_REFERENCE";
  Type2["BY_REFERENCE"] = "BY_REFERENCE";
  return Type2;
})(Type || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createItem2(item) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ item });
  const reqOpts = createItem(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.item;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { item: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["item"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCreateItems2(items, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    items,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkCreateItems(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          items: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["items", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteItem2(itemId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ itemId });
  const reqOpts = deleteItem(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { itemId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["itemId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkDeleteItems2(itemIds) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ itemIds });
  const reqOpts = bulkDeleteItems(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { itemIds: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["itemIds"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkDeleteItemsByFilter2(namespace, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    namespace,
    filter: options?.filter
  });
  const reqOpts = bulkDeleteItemsByFilter(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { namespace: "$[0]", filter: "$[1].filter" },
        singleArgumentUnchanged: false
      },
      ["namespace", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateItem2(_id, item) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    item: { ...item, id: _id }
  });
  const reqOpts = updateItem(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.item;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { item: "$[1]" },
        explicitPathsToArguments: { "item.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "item"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateItems2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    items: options?.items,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkUpdateItems(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          items: "$[0].items",
          returnEntity: "$[0].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getItem2(itemId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ itemId });
  const reqOpts = getItem(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.item;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { itemId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["itemId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listItems2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    filter: options?.filter,
    cursorPaging: options?.cursorPaging
  });
  const reqOpts = listItems(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          filter: "$[0].filter",
          cursorPaging: "$[0].cursorPaging"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryItems2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryItems(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [])
      );
      return {
        items: transformedData?.items,
        pagingMetadata: transformedData?.metadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function countItems2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    filter: options?.filter
  });
  const reqOpts = countItems(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { filter: "$[0].filter" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  SortOrder,
  Type,
  WebhookIdentityType,
  bulkCreateItems,
  bulkDeleteItems,
  bulkDeleteItemsByFilter,
  bulkUpdateItems,
  countItems,
  createItem,
  deleteItem,
  getItem,
  listItems,
  queryItems,
  updateItem
});
