// @wix/headless-restaurants-olo/dist/services/olo-settings-service.js
import { defineService, implementService } from '@wix/services-definitions';
import * as operationGroupsApi from '@wix/auto_sdk_restaurants_operation-groups';
import * as operationsApi from '@wix/auto_sdk_restaurants_operations';
import { SignalsServiceDefinition, } from '@wix/services-definitions/core-services/signals';
export const OLOSettingsServiceDefinition = defineService('oloSettings');
export const OLOSettingsService = implementService.withConfig()(OLOSettingsServiceDefinition, ({ getService, config }) => {
    const signalsService = getService(SignalsServiceDefinition);
    const operationGroup = signalsService.signal(config.operationGroup);
    const operation = signalsService.signal(config.operation);
    const selectedItem = signalsService.signal(null);
    const isLoading = signalsService.signal(false);
    const error = signalsService.signal(null);
    return {
        operationGroup,
        operation,
        isLoading,
        error,
        selectedItem,
    };
});
export async function loadOLOSettingsServiceConfig() {
    try {
        // Fetch operation groups and operations in parallel
        const [operationGroupsResponse, operationsResponse] = await Promise.all([
            operationGroupsApi.queryOperationGroups().find(),
            operationsApi.queryOperation().find(),
        ]);
        return {
            operationGroup: operationGroupsResponse.items[0] || undefined,
            operation: operationsResponse.items[0] || undefined,
        };
    }
    catch (error) {
        console.error('Failed to load OLO settings service config:', error);
        return {
            operationGroup: undefined,
            operation: undefined,
            isLoading: false,
            error: error?.message ||
                'Failed to load OLO settings service config',
        };
    }
}
