// @wix/auto_sdk_benefit-programs_balances/build/es/index.mjs
// src/benefit-programs-v1-balance-balances.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/benefit-programs-v1-balance-balances.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixBenefitProgramsV1BalanceBalanceServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/pool-accounts",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/benefit-programs-pool-accounts",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pool-accounts",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pool-accounts",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/benefit-programs-pool-accounts",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/benefit-programs-pool-accounts",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pool-accounts",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pool-accounts",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_benefit-programs_balances";
function getBalance(payload) {
  function __getBalance({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.balance",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.balance.BalanceService.GetBalance",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1BalanceBalanceServiceUrl({
        protoPath: "/v1/balances/{poolId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "balance.createdDate" },
            { path: "balance.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getBalance;
}
function listBalances(payload) {
  function __listBalances({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.balance",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.balance.BalanceService.ListBalances",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1BalanceBalanceServiceUrl({
        protoPath: "/v1/balances",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "balances.createdDate" },
            { path: "balances.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listBalances;
}
function queryBalances(payload) {
  function __queryBalances({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.balance",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.balance.BalanceService.QueryBalances",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1BalanceBalanceServiceUrl({
        protoPath: "/v1/balances/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "balances.createdDate" },
            { path: "balances.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryBalances;
}
function changeBalance(payload) {
  function __changeBalance({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "transactionDetails.effectiveDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.balance",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.balance.BalanceService.ChangeBalance",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1BalanceBalanceServiceUrl({
        protoPath: "/v1/balances/{poolId}/change",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "balance.createdDate" },
            { path: "balance.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __changeBalance;
}
function revertBalanceChange(payload) {
  function __revertBalanceChange({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.balance",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.balance.BalanceService.RevertBalanceChange",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1BalanceBalanceServiceUrl({
        protoPath: "/v1/balances/changes/{transactionId}/revert",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __revertBalanceChange;
}

// src/benefit-programs-v1-balance-balances.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var IdentityType = /* @__PURE__ */ ((IdentityType2) => {
  IdentityType2["UNKNOWN"] = "UNKNOWN";
  IdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  IdentityType2["MEMBER"] = "MEMBER";
  IdentityType2["WIX_USER"] = "WIX_USER";
  return IdentityType2;
})(IdentityType || {});
var PoolStatus = /* @__PURE__ */ ((PoolStatus2) => {
  PoolStatus2["UNDEFINED"] = "UNDEFINED";
  PoolStatus2["ACTIVE"] = "ACTIVE";
  PoolStatus2["PAUSED"] = "PAUSED";
  PoolStatus2["ENDED"] = "ENDED";
  PoolStatus2["PROVISIONING"] = "PROVISIONING";
  PoolStatus2["RENEWING"] = "RENEWING";
  return PoolStatus2;
})(PoolStatus || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var ChangeBalanceRequestType = /* @__PURE__ */ ((ChangeBalanceRequestType2) => {
  ChangeBalanceRequestType2["UNKNOWN_OPERATION"] = "UNKNOWN_OPERATION";
  ChangeBalanceRequestType2["ADJUST"] = "ADJUST";
  ChangeBalanceRequestType2["SET"] = "SET";
  return ChangeBalanceRequestType2;
})(ChangeBalanceRequestType || {});
var LimitRule = /* @__PURE__ */ ((LimitRule2) => {
  LimitRule2["UNKNOWN_LIMIT_RULE"] = "UNKNOWN_LIMIT_RULE";
  LimitRule2["FAIL"] = "FAIL";
  LimitRule2["SOFT_ENFORCE"] = "SOFT_ENFORCE";
  LimitRule2["HARD_ENFORCE"] = "HARD_ENFORCE";
  return LimitRule2;
})(LimitRule || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNKNOWN_OPERATION"] = "UNKNOWN_OPERATION";
  return Type2;
})(Type || {});
var TransactionReversibility = /* @__PURE__ */ ((TransactionReversibility2) => {
  TransactionReversibility2["TRANSACTION_IS_REVERSIBLE"] = "TRANSACTION_IS_REVERSIBLE";
  TransactionReversibility2["TRANSACTION_ALREADY_REVERSED"] = "TRANSACTION_ALREADY_REVERSED";
  return TransactionReversibility2;
})(TransactionReversibility || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function getBalance2(poolId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ poolId });
  const reqOpts = getBalance(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.balance;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { poolId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["poolId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listBalances2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    poolIds: options?.poolIds,
    cursorPaging: options?.cursorPaging
  });
  const reqOpts = listBalances(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          poolIds: "$[0].poolIds",
          cursorPaging: "$[0].cursorPaging"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryBalances2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryBalances(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [])
      );
      return {
        items: transformedData?.balances,
        pagingMetadata: transformedData?.metadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function changeBalance2(poolId, idempotencyKey, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    poolId,
    idempotencyKey,
    instructingParty: options?.instructingParty,
    type: options?.type,
    adjustOptions: options?.adjustOptions,
    setOptions: options?.setOptions,
    transactionDetails: options?.transactionDetails
  });
  const reqOpts = changeBalance(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          poolId: "$[0]",
          idempotencyKey: "$[1]",
          instructingParty: "$[2].instructingParty",
          type: "$[2].type",
          adjustOptions: "$[2].adjustOptions",
          setOptions: "$[2].setOptions",
          transactionDetails: "$[2].transactionDetails"
        },
        singleArgumentUnchanged: false
      },
      ["poolId", "idempotencyKey", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function revertBalanceChange2(transactionId, idempotencyKey, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    transactionId,
    idempotencyKey,
    instructingParty: options?.instructingParty
  });
  const reqOpts = revertBalanceChange(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          transactionId: "$[0]",
          idempotencyKey: "$[1]",
          instructingParty: "$[2].instructingParty"
        },
        singleArgumentUnchanged: false
      },
      ["transactionId", "idempotencyKey", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/benefit-programs-v1-balance-balances.public.ts
function getBalance3(httpClient) {
  return (poolId) => getBalance2(
    poolId,
    // @ts-ignore
    { httpClient }
  );
}
function listBalances3(httpClient) {
  return (options) => listBalances2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function queryBalances3(httpClient) {
  return () => queryBalances2(
    // @ts-ignore
    { httpClient }
  );
}
function changeBalance3(httpClient) {
  return (poolId, idempotencyKey, options) => changeBalance2(
    poolId,
    idempotencyKey,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function revertBalanceChange3(httpClient) {
  return (transactionId, idempotencyKey, options) => revertBalanceChange2(
    transactionId,
    idempotencyKey,
    options,
    // @ts-ignore
    { httpClient }
  );
}

// src/benefit-programs-v1-balance-balances.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
var getBalance4 = /* @__PURE__ */ createRESTModule(getBalance3);
var listBalances4 = /* @__PURE__ */ createRESTModule(listBalances3);
var queryBalances4 = /* @__PURE__ */ createRESTModule(queryBalances3);
var changeBalance4 = /* @__PURE__ */ createRESTModule(changeBalance3);
var revertBalanceChange4 = /* @__PURE__ */ createRESTModule(revertBalanceChange3);
export {
  ChangeBalanceRequestType,
  IdentityType,
  LimitRule,
  PoolStatus,
  SortOrder,
  TransactionReversibility,
  Type,
  WebhookIdentityType,
  changeBalance4 as changeBalance,
  getBalance4 as getBalance,
  listBalances4 as listBalances,
  queryBalances4 as queryBalances,
  revertBalanceChange4 as revertBalanceChange
};
