// @wix/auto-patterns/dist/esm/components/filters/DynamicCollectionFilter.js
import _extends from "@babel/runtime/helpers/extends";
import React from 'react';
import { useFilterCollection } from '@wix/patterns';
import { useSchemaRegistry } from '../../providers';
import { fetchCmsData } from '../../dataSourceAdapters/cms/fetchCmsData';
export const DynamicCollectionFilter = _ref => {
  let {
    filter,
    collectionId,
    component: FilterComponent,
    ...filterProps
  } = _ref;
  const {
    getSchema
  } = useSchemaRegistry();
  const referencedSchema = getSchema(collectionId);
  const fetchData = React.useCallback(async query => {
    if (!referencedSchema) {
      return {
        items: [],
        total: 0
      };
    }
    const result = await fetchCmsData(collectionId, query, referencedSchema.fields, {
      searchableFieldIds: [referencedSchema.displayField],
      filterFieldMapping: {}
    });
    return {
      items: result.items.map(item => ({
        id: item[referencedSchema.idField],
        name: item[referencedSchema.displayField]
      }))
    };
  }, [collectionId, referencedSchema]);
  const collection = useFilterCollection(filter, {
    queryName: `filter-collection-${collectionId}`,
    fetchData
  });
  return /*#__PURE__*/React.createElement(FilterComponent, _extends({
    filter: filter,
    collection: collection
  }, filterProps));
};
