// @wix/bex-utils/@wix/ambassador-media-site-media-v1-file-descriptor/build/es/builders.impl.js
import { mergeDeep, someNumber, someString, someAny, someRepeated, someDate, } from '@wix/metro-common-builders';
import { StreamFormat, MediaType, OperationStatus, UploadProtocol, State, RootFolder, IdentityType, Namespace, ContentDisposition, WebhookIdentityType, SortOrder, ImageAnnotationType, } from './types.impl';
// @ts-ignore
import stack from 'callsite';
export function aFileDescriptor(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        id: someString({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MAX_LENGTH', value: '1000' },
            ],
        }),
        displayName: someString({
            validations: [
                { type: 'READ_ONLY' },
                { type: 'MAX_LENGTH', value: '200' },
            ],
        }),
        url: someString({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'WEB_URL' },
            ],
        }),
        parentFolderId: someString({
            validations: [
                { type: 'READ_ONLY' },
                { type: 'MAX_LENGTH', value: '100' },
            ],
        }),
        hash: someString({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MAX_LENGTH', value: '100' },
            ],
        }),
        sizeInBytes: someString({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
                { type: 'DECIMAL_VALUE', value: { maxScale: 0 } },
            ],
        }),
        private: true,
        mediaType: aMediaType(),
        media: aFileMedia(),
        operationStatus: aOperationStatus(),
        sourceUrl: someString({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'WEB_URL' },
            ],
        }),
        thumbnailUrl: someString({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'WEB_URL' },
            ],
        }),
        labels: [
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'READ_ONLY' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'READ_ONLY' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'READ_ONLY' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'READ_ONLY' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'READ_ONLY' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'READ_ONLY' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'READ_ONLY' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'READ_ONLY' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'READ_ONLY' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'READ_ONLY' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
        ],
        createdDate: someDate({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        updatedDate: someDate({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        siteId: someString({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MAX_LENGTH', value: '100' },
            ],
        }),
        state: aState(),
        internalTags: [
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
        ],
        namespace: aNamespace(),
        addedBy: aIdentityInfo(),
        rawNamespace: someString({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MAX_LENGTH', value: '200' },
            ],
        }),
        lastUsedDate: someDate({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        mimeType: someString({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MAX_LENGTH', value: '200' },
            ],
        }),
    }, overrideObject);
}
export function aMediaType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = MediaType.UNKNOWN; }
    return enumFirstItem;
}
export function aFileMedia(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        image: aImageMedia(),
    }, overrideObject);
}
export function aImageMedia(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        image: aImage(),
        colors: aColors(),
        faces: [
            aFaceRecognition(),
            aFaceRecognition(),
            aFaceRecognition(),
            aFaceRecognition(),
            aFaceRecognition(),
            aFaceRecognition(),
            aFaceRecognition(),
            aFaceRecognition(),
            aFaceRecognition(),
            aFaceRecognition(),
        ],
        previewImage: aImage(),
        caption: someString({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MAX_LENGTH', value: '300' },
            ],
        }),
        containsText: true,
        animated: true,
    }, overrideObject);
}
export function aImage(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        id: someString(),
        url: someString(),
        height: someNumber({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        width: someNumber({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        altText: someString(),
        urlExpirationDate: someDate({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        filename: someString({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        sizeInBytes: someString({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
            ],
        }),
        focalPoint: aFocalPoint(),
    }, overrideObject);
}
export function aFocalPoint(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        x: someNumber(),
        y: someNumber(),
        height: someNumber(),
        width: someNumber(),
    }, overrideObject);
}
export function aColors(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        prominent: aColor(),
        palette: [
            aColor(),
            aColor(),
            aColor(),
            aColor(),
            aColor(),
            aColor(),
            aColor(),
            aColor(),
            aColor(),
            aColor(),
        ],
    }, overrideObject);
}
export function aColor(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        hex: someString({
            validations: [
                { type: 'FORMAT', value: 'COLOR_HEX' },
            ],
        }),
        rgb: aColorRGB(),
    }, overrideObject);
}
export function aColorRGB(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        r: someNumber({
            validations: [
                { type: 'MIN' },
                { type: 'MAX', value: '255' },
            ],
        }),
        g: someNumber({
            validations: [
                { type: 'MIN' },
                { type: 'MAX', value: '255' },
            ],
        }),
        b: someNumber({
            validations: [
                { type: 'MIN' },
                { type: 'MAX', value: '255' },
            ],
        }),
    }, overrideObject);
}
export function aFaceRecognition(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        confidence: someNumber(),
        x: someNumber(),
        y: someNumber(),
        height: someNumber(),
        width: someNumber(),
    }, overrideObject);
}
export function aVideoV2(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        id: someString(),
        url: someString(),
        resolutions: [
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
        ],
        filename: someString({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        posters: [
            aImage(),
            aImage(),
            aImage(),
            aImage(),
            aImage(),
            aImage(),
            aImage(),
            aImage(),
            aImage(),
            aImage(),
        ],
        sizeInBytes: someString({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
            ],
        }),
        urlExpirationDate: someDate({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        durationInMilliseconds: someNumber({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MIN' },
            ],
        }),
        title: someString({
            validations: [{ type: 'MAX_LENGTH', value: '300' }],
        }),
        description: someString({
            validations: [{ type: 'MAX_LENGTH', value: '2000' }],
        }),
        height: someNumber({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MIN' },
            ],
        }),
        width: someNumber({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MIN' },
            ],
        }),
    }, overrideObject);
}
export function aVideoResolution(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        url: someString(),
        height: someNumber(),
        width: someNumber(),
        poster: aImage(),
        format: someString(),
        urlExpirationDate: someDate(),
        sizeInBytes: someString({
            validations: [
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
            ],
        }),
        quality: someString(),
        filename: someString(),
        durationInSeconds: someNumber({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MIN' },
            ],
        }),
        durationInMilliseconds: someNumber({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MIN' },
            ],
        }),
        private: true,
        assetKey: someString({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
    }, overrideObject);
}
export function aAudioV2(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        id: someString({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
        assets: [
            aAudio(),
            aAudio(),
            aAudio(),
            aAudio(),
            aAudio(),
            aAudio(),
            aAudio(),
            aAudio(),
            aAudio(),
            aAudio(),
        ],
        bitrate: someNumber({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MIN' },
            ],
        }),
        format: someString({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MAX_LENGTH', value: '100' },
            ],
        }),
        duration: someNumber({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MIN' },
            ],
        }),
        sizeInBytes: someString({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
                { type: 'DECIMAL_VALUE', value: { maxScale: 0 } },
            ],
        }),
    }, overrideObject);
}
export function aAudio(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        id: someString(),
        url: someString(),
        urlExpirationDate: someDate({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        sizeInBytes: someString({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
            ],
        }),
        filename: someString({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        duration: someNumber({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MIN' },
            ],
        }),
        private: true,
        assetKey: someString({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        format: someString({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        quality: someString({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
    }, overrideObject);
}
export function aDocument(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        id: someString(),
        url: someString(),
        urlExpirationDate: someDate(),
        sizeInBytes: someString({
            validations: [
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
            ],
        }),
        filename: someString(),
        thumbnail: aImage(),
    }, overrideObject);
}
export function aArchive(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        id: someString(),
        url: someString(),
        urlExpirationDate: someDate({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        sizeInBytes: someString({
            validations: [
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
            ],
        }),
        filename: someString(),
    }, overrideObject);
}
export function aModel3D(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        id: someString(),
        url: someString(),
        thumbnail: aImage(),
        altText: someString(),
        urlExpirationDate: someDate({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        filename: someString({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        sizeInBytes: someString({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
            ],
        }),
    }, overrideObject);
}
export function aOtherMedia(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        id: someString({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
        internalMediaType: someString({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MAX_LENGTH', value: '100' },
            ],
        }),
        sizeInBytes: someString({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
                {
                    type: 'DECIMAL_VALUE',
                    value: { gte: '0', maxScale: 0 },
                },
            ],
        }),
        url: someString({
            validations: [{ type: 'FORMAT', value: 'WEB_URL' }],
        }),
    }, overrideObject);
}
export function aFontMedia(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        id: someString({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
        sizeInBytes: someString({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
                {
                    type: 'DECIMAL_VALUE',
                    value: { gte: '0', maxScale: 0 },
                },
            ],
        }),
        format: someString({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
        family: someString({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        fontName: someString({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        assets: [
            aFontAsset(),
            aFontAsset(),
            aFontAsset(),
            aFontAsset(),
            aFontAsset(),
        ],
    }, overrideObject);
}
export function aFontAsset(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        assetKey: someString({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        url: someString({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
        format: someString({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
    }, overrideObject);
}
export function aOperationStatus(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = OperationStatus.FAILED; }
    return enumFirstItem;
}
export function aState(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = State.OK; }
    return enumFirstItem;
}
export function aNamespace(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = Namespace.NO_NAMESPACE; }
    return enumFirstItem;
}
export function aIdentityInfo(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        identityType: aIdentityType(),
        identityId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
export function aIdentityType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = IdentityType.UNKNOWN; }
    return enumFirstItem;
}
export function aFileReady(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        file: aFileDescriptor(),
        externalInfo: aExternalInfo(),
        triggeredByUndelete: true,
    }, overrideObject);
}
export function aExternalInfo(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        origin: someString({
            validations: [{ type: 'MAX_LENGTH', value: '1000' }],
        }),
        externalIds: [
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '30' },
                    { type: 'MAX_LENGTH', value: '4000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '30' },
                    { type: 'MAX_LENGTH', value: '4000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '30' },
                    { type: 'MAX_LENGTH', value: '4000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '30' },
                    { type: 'MAX_LENGTH', value: '4000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '30' },
                    { type: 'MAX_LENGTH', value: '4000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '30' },
                    { type: 'MAX_LENGTH', value: '4000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '30' },
                    { type: 'MAX_LENGTH', value: '4000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '30' },
                    { type: 'MAX_LENGTH', value: '4000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '30' },
                    { type: 'MAX_LENGTH', value: '4000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '30' },
                    { type: 'MAX_LENGTH', value: '4000' },
                ],
            }),
        ],
    }, overrideObject);
}
export function aFileFailed(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        externalInfo: aExternalInfo(),
        error: aApplicationError(),
    }, overrideObject);
}
export function aApplicationError(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        code: someString(),
        description: someString(),
        data: {},
    }, overrideObject);
}
export function aDraftFilePublished(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        file: aFileDescriptor(),
        namespace: aNamespace(),
    }, overrideObject);
}
export function aBulkAnnotateImagesRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        fileIds: [
            someString({
                validations: [
                    { type: 'MIN_SIZE', value: '1' },
                    { type: 'MAX_SIZE', value: '10' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
        ],
        annotationTypes: [aImageAnnotationType()],
        overwrite: true,
        returnEntity: true,
    }, overrideObject);
}
export function aImageAnnotationType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = ImageAnnotationType.UNKNOWN_IMAGE_ANNOTATION_TYPE; }
    return enumFirstItem;
}
export function aBulkAnnotateImagesResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        results: [aBulkAnnotateImageResult()],
        bulkActionMetadata: aBulkActionMetadata(),
    }, overrideObject);
}
export function aBulkAnnotateImageResult(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        itemMetadata: aItemMetadata(),
        item: aFileDescriptor(),
    }, overrideObject);
}
export function aItemMetadata(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        id: someString(),
        originalIndex: someNumber(),
        success: true,
        error: aApplicationError(),
    }, overrideObject);
}
export function aBulkActionMetadata(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        totalSuccesses: someNumber(),
        totalFailures: someNumber(),
        undetailedFailures: someNumber(),
    }, overrideObject);
}
export function aGenerateFilesDownloadUrlRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        fileIds: [
            someString({
                validations: [
                    { type: 'MIN_SIZE', value: '1' },
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
        ],
    }, overrideObject);
}
export function aGenerateFilesDownloadUrlResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        downloadUrl: someString({
            validations: [{ type: 'FORMAT', value: 'WEB_URL' }],
        }),
    }, overrideObject);
}
export function aGenerateFileDownloadUrlRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        fileId: someString({
            validations: [{ type: 'MAX_LENGTH', value: '1000' }],
        }),
        downloadFileName: someString({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        expirationInMinutes: someNumber({
            validations: [{ type: 'MIN', value: '1' }],
        }),
        expirationRedirectUrl: someString({
            validations: [{ type: 'FORMAT', value: 'WEB_URL' }],
        }),
        assetKeys: [
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '100' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '100' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '100' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '100' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '100' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '100' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '100' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '100' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '100' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '100' },
                ],
            }),
        ],
        contentDisposition: aContentDisposition(),
    }, overrideObject);
}
export function aContentDisposition(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = ContentDisposition.ATTACHMENT; }
    return enumFirstItem;
}
export function aGenerateFileDownloadUrlResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        downloadUrls: [
            aDownloadUrl(),
            aDownloadUrl(),
            aDownloadUrl(),
            aDownloadUrl(),
            aDownloadUrl(),
            aDownloadUrl(),
            aDownloadUrl(),
            aDownloadUrl(),
            aDownloadUrl(),
            aDownloadUrl(),
        ],
    }, overrideObject);
}
export function aDownloadUrl(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        url: someString({
            validations: [{ type: 'FORMAT', value: 'WEB_URL' }],
        }),
        assetKey: someString({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
    }, overrideObject);
}
export function aServiceError(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        internalCode: someString({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
        internalHttpStatusCode: someNumber(),
        internalKey: someString({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
    }, overrideObject);
}
export function aGetFileDescriptorRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        fileId: someString({
            validations: [{ type: 'MAX_LENGTH', value: '1000' }],
        }),
    }, overrideObject);
}
export function aGetFileDescriptorResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        file: aFileDescriptor(),
    }, overrideObject);
}
export function aGetFileDescriptorsRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        fileIds: [
            someString({
                validations: [
                    { type: 'MIN_SIZE', value: '1' },
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
        ],
    }, overrideObject);
}
export function aGetFileDescriptorsResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        files: [
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
        ],
    }, overrideObject);
}
export function aUpdateFileDescriptorRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        file: aFileDescriptor(),
        fieldMask: someRepeated(),
    }, overrideObject);
}
export function aUpdateFileDescriptorResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        file: aFileDescriptor(),
    }, overrideObject);
}
export function aUnsupportedRequestValueError(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        allowedValues: [
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
        ],
        requestValue: someString({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
    }, overrideObject);
}
export function aGenerateFileUploadUrlRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        mimeType: someString({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        fileName: someString({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
        sizeInBytes: someString({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
                {
                    type: 'DECIMAL_VALUE',
                    value: { gte: '0', maxScale: 0 },
                },
            ],
        }),
        parentFolderId: someString({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        private: true,
        labels: [
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '50' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '50' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '50' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '50' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '50' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '50' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '50' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '50' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '50' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '50' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
        ],
        externalInfo: aExternalInfo(),
        rawNamespace: someString({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        namespace: aNamespace(),
        filePath: someString({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        internalTags: [
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
        ],
        draftTtlInHours: someNumber({
            validations: [{ type: 'MIN', value: '1' }],
        }),
        imageAnnotationConfigurations: aImageAnnotationConfigurations(),
    }, overrideObject);
}
export function aImageAnnotationConfigurations(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        skipDetection: true,
        annotationTypes: [
            aImageAnnotationType(),
            aImageAnnotationType(),
            aImageAnnotationType(),
            aImageAnnotationType(),
            aImageAnnotationType(),
            aImageAnnotationType(),
            aImageAnnotationType(),
            aImageAnnotationType(),
            aImageAnnotationType(),
            aImageAnnotationType(),
        ],
    }, overrideObject);
}
export function aGenerateFileUploadUrlResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        uploadUrl: someString({
            validations: [{ type: 'FORMAT', value: 'WEB_URL' }],
        }),
    }, overrideObject);
}
export function aSiteQuotaExceededError(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        internalCode: someString({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
        internalHttpStatusCode: someNumber(),
        internalKey: someString({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
        quota: aTotalQuota(),
    }, overrideObject);
}
export function aTotalQuota(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        storage: someString({
            validations: [{ type: 'DECIMAL_VALUE' }],
        }),
        plans: aPlans(),
        timestamp: someDate(),
    }, overrideObject);
}
export function aPlans(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        premium: aPlan(),
        upgradeUrl: someString({
            validations: [{ type: 'FORMAT', value: 'WEB_URL' }],
        }),
    }, overrideObject);
}
export function aPlan(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        id: someString({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'GUID' },
            ],
        }),
        name: someString(),
        createdAt: someDate(),
        expiresAt: someDate(),
    }, overrideObject);
}
export function aGenerateFileResumableUploadUrlRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        mimeType: someString({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        fileName: someString({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
        sizeInBytes: someString({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
                {
                    type: 'DECIMAL_VALUE',
                    value: { gte: '0', maxScale: 0 },
                },
            ],
        }),
        parentFolderId: someString({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        private: true,
        labels: [
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
        ],
        uploadProtocol: aUploadProtocol(),
        rawNamespace: someString({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        namespace: aNamespace(),
        filePath: someString({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        internalTags: [
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
        ],
        draftTtlInHours: someNumber({
            validations: [{ type: 'MIN', value: '1' }],
        }),
        imageAnnotationConfigurations: aImageAnnotationConfigurations(),
    }, overrideObject);
}
export function aUploadProtocol(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = UploadProtocol.TUS; }
    return enumFirstItem;
}
export function aGenerateFileResumableUploadUrlResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        uploadProtocol: aUploadProtocol(),
        uploadUrl: someString({
            validations: [{ type: 'FORMAT', value: 'WEB_URL' }],
        }),
        uploadToken: someString({
            validations: [{ type: 'MAX_LENGTH', value: '2000' }],
        }),
    }, overrideObject);
}
export function aImportFileRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        url: someString({
            validations: [{ type: 'FORMAT', value: 'WEB_URL' }],
        }),
        mediaType: aMediaType(),
        displayName: someString({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
        parentFolderId: someString({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        private: true,
        labels: [
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
        ],
        mimeType: someString({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        externalInfo: aExternalInfo(),
        urlParams: {},
        urlHeaders: {},
        rawNamespace: someString({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        namespace: aNamespace(),
        filePath: someString({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        internalTags: [
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
        ],
        draftTtlInHours: someNumber({
            validations: [{ type: 'MIN', value: '1' }],
        }),
        imageAnnotationConfigurations: aImageAnnotationConfigurations(),
    }, overrideObject);
}
export function aImportFileResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        file: aFileDescriptor(),
    }, overrideObject);
}
export function aBulkImportFilesRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        importFileRequests: [aImportFileRequest()],
    }, overrideObject);
}
export function aBulkImportFilesResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        files: [
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
        ],
    }, overrideObject);
}
export function aBulkImportFileRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        importFileRequests: [aImportFileRequest()],
        returnEntity: true,
    }, overrideObject);
}
export function aBulkImportFileResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        results: [aBulkImportFileResult()],
        bulkActionMetadata: aBulkActionMetadata(),
    }, overrideObject);
}
export function aBulkImportFileResult(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        itemMetadata: aItemMetadata(),
        item: aFileDescriptor(),
    }, overrideObject);
}
export function aListFilesRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        parentFolderId: someString({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        mediaTypes: [
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
        ],
        private: true,
        sort: aSorting(),
        paging: aCursorPaging(),
        rawNamespace: someString({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        namespace: aNamespace(),
    }, overrideObject);
}
export function aSorting(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        fieldName: someString({
            validations: [{ type: 'MAX_LENGTH', value: '512' }],
        }),
        order: aSortOrder(),
        selectItemsBy: [{}, {}, {}, {}, {}, {}, {}, {}, {}, {}],
    }, overrideObject);
}
export function aSortOrder(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = SortOrder.ASC; }
    return enumFirstItem;
}
export function aCursorPaging(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        limit: someNumber({
            validations: [
                { type: 'MIN' },
                { type: 'MAX', value: '100' },
            ],
        }),
        cursor: someString({
            validations: [
                { type: 'MAX_LENGTH', value: '16000' },
            ],
        }),
    }, overrideObject);
}
export function aListFilesResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        files: [
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
        ],
        nextCursor: aPagingMetadataV2(),
    }, overrideObject);
}
export function aPagingMetadataV2(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        total: someNumber(),
        cursors: aCursors(),
        hasNext: true,
    }, overrideObject);
}
export function aCursors(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        next: someString({
            validations: [
                { type: 'MAX_LENGTH', value: '16000' },
            ],
        }),
    }, overrideObject);
}
export function aSearchFilesRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        search: someString({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
        rootFolder: aRootFolder(),
        mediaTypes: [
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
        ],
        private: true,
        sort: aSorting(),
        paging: aCursorPaging(),
        internalTag: someString({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
        rawNamespace: someString({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        namespace: aNamespace(),
    }, overrideObject);
}
export function aRootFolder(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = RootFolder.MEDIA_ROOT; }
    return enumFirstItem;
}
export function aSearchFilesResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        files: [
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
        ],
        nextCursor: aPagingMetadataV2(),
    }, overrideObject);
}
export function aGenerateVideoStreamingUrlRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        fileId: someString({
            validations: [{ type: 'MAX_LENGTH', value: '1000' }],
        }),
        format: aStreamFormat(),
    }, overrideObject);
}
export function aStreamFormat(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = StreamFormat.UNKNOWN; }
    return enumFirstItem;
}
export function aGenerateVideoStreamingUrlResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        downloadUrl: aDownloadUrl(),
    }, overrideObject);
}
export function aGenerateWebSocketTokenRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({}, overrideObject);
}
export function aGenerateWebSocketTokenResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        token: someString({
            validations: [{ type: 'MAX_LENGTH', value: '2000' }],
        }),
    }, overrideObject);
}
export function aBulkDeleteFilesRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        fileIds: [
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
        ],
        permanent: true,
    }, overrideObject);
}
export function aBulkDeleteFilesResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({}, overrideObject);
}
export function aBulkRestoreFilesFromTrashBinRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        fileIds: [
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '1000' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
        ],
    }, overrideObject);
}
export function aBulkRestoreFilesFromTrashBinResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({}, overrideObject);
}
export function aListDeletedFilesRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        parentFolderId: someString({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        mediaTypes: [
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
            aMediaType(),
        ],
        private: true,
        sort: aSorting(),
        paging: aCursorPaging(),
        rawNamespace: someString({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        namespace: aNamespace(),
    }, overrideObject);
}
export function aListDeletedFilesResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        files: [
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
            aFileDescriptor(),
        ],
        nextCursor: aPagingMetadataV2(),
    }, overrideObject);
}
export function aBulkPublishDraftFilesRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        fileIds: [
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '1000' },
                ],
            }),
        ],
        namespace: aNamespace(),
        parentFolderId: someString({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        filePath: someString({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        returnEntity: true,
    }, overrideObject);
}
export function aBulkPublishDraftFilesResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        results: [aBulkPublishDraftFileResult()],
        bulkActionMetadata: aBulkActionMetadata(),
    }, overrideObject);
}
export function aBulkPublishDraftFileResult(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        itemMetadata: aItemMetadata(),
        item: aFileDescriptor(),
    }, overrideObject);
}
export function aUpdateFileRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        fileId: someString({
            validations: [{ type: 'MAX_LENGTH', value: '1000' }],
        }),
        displayName: someString({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
        parentFolderId: someString({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        labels: [
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '500' },
                    { type: 'MAX_LENGTH', value: '200' },
                ],
            }),
        ],
    }, overrideObject);
}
export function aUpdateFileResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        file: aFileDescriptor(),
    }, overrideObject);
}
export function aDomainEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        createdEvent: aEntityCreatedEvent(),
        id: someString(),
        entityFqdn: someString(),
        slug: someString(),
        entityId: someString(),
        eventTime: someDate(),
        triggeredByAnonymizeRequest: true,
        originatedFrom: someString(),
        entityEventSequence: someString({
            validations: [{ type: 'DECIMAL_VALUE' }],
        }),
    }, overrideObject);
}
export function aEntityCreatedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        entityAsJson: someString(),
        triggeredByUndelete: true,
        restoreInfo: aRestoreInfo(),
        additionalMetadataAsJson: someString(),
    }, overrideObject);
}
export function aRestoreInfo(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        deletedDate: someDate(),
    }, overrideObject);
}
export function aEntityUpdatedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        currentEntityAsJson: someString(),
        previousEntityAsJson: someString(),
        modifiedFields: { modifiedFields: someAny() },
        additionalMetadataAsJson: someString(),
    }, overrideObject);
}
export function aEntityDeletedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        movedToTrash: true,
        deletedEntityAsJson: someString(),
        additionalMetadataAsJson: someString(),
    }, overrideObject);
}
export function aActionEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        bodyAsJson: someString(),
    }, overrideObject);
}
export function aMessageEnvelope(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        instanceId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        eventType: someString({
            validations: [{ type: 'MAX_LENGTH', value: '150' }],
        }),
        identity: aIdentificationData(),
        data: someString(),
    }, overrideObject);
}
export function aIdentificationData(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        anonymousVisitorId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        identityType: aWebhookIdentityType(),
    }, overrideObject);
}
export function aWebhookIdentityType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = WebhookIdentityType.UNKNOWN; }
    return enumFirstItem;
}
function isCurrentBuilderWasPreviouslyCalled() {
    var callChainFunctionNames = stack().map(function (item) {
        return item.getFunctionName();
    });
    var callChainWithoutCurrentBuilder = callChainFunctionNames.splice(2);
    var currentBuilderName = callChainFunctionNames[1];
    return callChainWithoutCurrentBuilder.some(function (functionName) { return functionName === currentBuilderName; });
}
export * from '@wix/metro-common-builders';
