// @wix/bex-utils/@wix/ambassador-media-site-media-v1-file-descriptor/build/es/meta.impl.js
import * as httpWixMediaSiteMediaV1FileDescriptor from './http.impl';
export function bulkAnnotateImages() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.bulkAnnotateImages(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/files/images/annotate',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function generateFilesDownloadUrl() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.generateFilesDownloadUrl(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/files/generate-download-url',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function generateFileDownloadUrl() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.generateFileDownloadUrl(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/files/generate-file-download-url',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getFileDescriptor() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.getFileDescriptor(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/v1/files/get-file-by-id',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getFileDescriptors() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.getFileDescriptors(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/files/get-files',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateFileDescriptor() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.updateFileDescriptor(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'PATCH',
        path: '/v1/files/update-file-descriptor',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function generateFileUploadUrl() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.generateFileUploadUrl(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/files/generate-upload-url',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function generateFileResumableUploadUrl() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.generateFileResumableUploadUrl(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/files/generate-resumable-upload-url',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function importFile() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.importFile(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/files/import',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkImportFiles() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.bulkImportFiles(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/files/import',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkImportFile() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.bulkImportFile(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/files/import-v2',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listFiles() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.listFiles(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/v1/files',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function searchFiles() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.searchFiles(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/files/search',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function generateVideoStreamingUrl() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.generateVideoStreamingUrl(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/files/generate-video-stream-url',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function generateWebSocketToken() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.generateWebSocketToken(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/files/generate-web-socket-token',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkDeleteFiles() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.bulkDeleteFiles(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/files/delete',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkRestoreFilesFromTrashBin() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.bulkRestoreFilesFromTrashBin(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/trash-bin/files/restore',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listDeletedFiles() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.listDeletedFiles(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/v1/trash-bin/files',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkPublishDraftFiles() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.bulkPublishDraftFiles(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/files/publish-drafts',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateFile() {
    var payload = {};
    var getRequestOptions = httpWixMediaSiteMediaV1FileDescriptor.updateFile(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'PATCH',
        path: '/v1/files/update-file',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
