// @wix/auto_sdk_payments_onboarding-availability/build/internal/es/index.typings.mjs
// src/cashier-onboarding-availability-v1-onboarding-availability-onboarding-availability.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/cashier-onboarding-availability-v1-onboarding-availability-onboarding-availability.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixCashierOnboardingAvailabilityV1OnboardingAvailabilityServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/v1/onboarding-availability",
        destPath: "/v1/onboarding-availability"
      },
      {
        srcPath: "/ambassador/v1/onboarding-availability",
        destPath: "/v1/onboarding-availability"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_onboarding-availability";
function getOnboardingAvailability(payload) {
  function __getOnboardingAvailability({ host }) {
    const metadata = {
      entityFqdn: "wix.cashier.onboarding_availability.v1.onboarding_availability",
      method: "GET",
      methodFqn: "wix.cashier.onboarding_availability.v1.OnboardingAvailabilityService.GetOnboardingAvailability",
      packageName: PACKAGE_NAME,
      url: resolveWixCashierOnboardingAvailabilityV1OnboardingAvailabilityServiceUrl(
        { protoPath: "/v1/onboarding-availability", data: payload, host }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            {
              path: "onboardingAvailability.cbdFlow.attestationInfo.attestationFormSignedDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getOnboardingAvailability;
}
function updateCbdFlow(payload) {
  function __updateCbdFlow({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "cbdFlow.attestationInfo.attestationFormSignedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.cashier.onboarding_availability.v1.onboarding_availability",
      method: "PATCH",
      methodFqn: "wix.cashier.onboarding_availability.v1.OnboardingAvailabilityService.UpdateCbdFlow",
      packageName: PACKAGE_NAME,
      url: resolveWixCashierOnboardingAvailabilityV1OnboardingAvailabilityServiceUrl(
        {
          protoPath: "/v1/onboarding-availability/cbd-flow",
          data: serializedData,
          host
        }
      ),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            {
              path: "onboardingAvailability.cbdFlow.attestationInfo.attestationFormSignedDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateCbdFlow;
}
function updateRestrictedGoodsFlow(payload) {
  function __updateRestrictedGoodsFlow({ host }) {
    const metadata = {
      entityFqdn: "wix.cashier.onboarding_availability.v1.onboarding_availability",
      method: "PATCH",
      methodFqn: "wix.cashier.onboarding_availability.v1.OnboardingAvailabilityService.UpdateRestrictedGoodsFlow",
      packageName: PACKAGE_NAME,
      url: resolveWixCashierOnboardingAvailabilityV1OnboardingAvailabilityServiceUrl(
        {
          protoPath: "/v1/onboarding-availability/restricted-goods-flow",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            {
              path: "onboardingAvailability.cbdFlow.attestationInfo.attestationFormSignedDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateRestrictedGoodsFlow;
}
function updatePartnerFlow(payload) {
  function __updatePartnerFlow({ host }) {
    const metadata = {
      entityFqdn: "wix.cashier.onboarding_availability.v1.onboarding_availability",
      method: "PATCH",
      methodFqn: "wix.cashier.onboarding_availability.v1.OnboardingAvailabilityService.UpdatePartnerFlow",
      packageName: PACKAGE_NAME,
      url: resolveWixCashierOnboardingAvailabilityV1OnboardingAvailabilityServiceUrl(
        {
          protoPath: "/v1/onboarding-availability/partner-flow",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            {
              path: "onboardingAvailability.cbdFlow.attestationInfo.attestationFormSignedDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updatePartnerFlow;
}

// src/cashier-onboarding-availability-v1-onboarding-availability-onboarding-availability.universal.ts
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["UNDEFINED"] = "UNDEFINED";
  Status2["DECLINED"] = "DECLINED";
  Status2["POSSIBLE"] = "POSSIBLE";
  Status2["CONFIRMED"] = "CONFIRMED";
  return Status2;
})(Status || {});
var RestrictedGoodsFlowStatus = /* @__PURE__ */ ((RestrictedGoodsFlowStatus2) => {
  RestrictedGoodsFlowStatus2["UNDEFINED"] = "UNDEFINED";
  RestrictedGoodsFlowStatus2["DECLINED"] = "DECLINED";
  RestrictedGoodsFlowStatus2["CONFIRMED"] = "CONFIRMED";
  return RestrictedGoodsFlowStatus2;
})(RestrictedGoodsFlowStatus || {});
var RestrictedGoodsCategory = /* @__PURE__ */ ((RestrictedGoodsCategory2) => {
  RestrictedGoodsCategory2["UNDEFINED"] = "UNDEFINED";
  RestrictedGoodsCategory2["TOBACCO_ALCOHOL"] = "TOBACCO_ALCOHOL";
  RestrictedGoodsCategory2["FIREARMS_WEAPONS"] = "FIREARMS_WEAPONS";
  RestrictedGoodsCategory2["ADULT"] = "ADULT";
  RestrictedGoodsCategory2["MEDICAL"] = "MEDICAL";
  RestrictedGoodsCategory2["FINANCIAL"] = "FINANCIAL";
  RestrictedGoodsCategory2["TRAVEL_AGENCIES"] = "TRAVEL_AGENCIES";
  RestrictedGoodsCategory2["GAMBLING_LOTTERIES_SKILL_GAMES"] = "GAMBLING_LOTTERIES_SKILL_GAMES";
  RestrictedGoodsCategory2["BINARY_OPTIONS_CRYPTOCURRENCIES"] = "BINARY_OPTIONS_CRYPTOCURRENCIES";
  RestrictedGoodsCategory2["MARKETPLACES"] = "MARKETPLACES";
  RestrictedGoodsCategory2["OTHER"] = "OTHER";
  RestrictedGoodsCategory2["CBD"] = "CBD";
  RestrictedGoodsCategory2["TOBACCO_E_CIGARETTES"] = "TOBACCO_E_CIGARETTES";
  RestrictedGoodsCategory2["ALCOHOL"] = "ALCOHOL";
  RestrictedGoodsCategory2["NUTRACEUTICALS"] = "NUTRACEUTICALS";
  RestrictedGoodsCategory2["MISLEADING_MARKETING"] = "MISLEADING_MARKETING";
  RestrictedGoodsCategory2["INTELLECTUAL_PROPERTY"] = "INTELLECTUAL_PROPERTY";
  return RestrictedGoodsCategory2;
})(RestrictedGoodsCategory || {});
var PartnerFlowStatus = /* @__PURE__ */ ((PartnerFlowStatus2) => {
  PartnerFlowStatus2["UNDEFINED"] = "UNDEFINED";
  PartnerFlowStatus2["DECLINED"] = "DECLINED";
  PartnerFlowStatus2["CONFIRMED"] = "CONFIRMED";
  return PartnerFlowStatus2;
})(PartnerFlowStatus || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function getOnboardingAvailability2() {
  const { httpClient, sideEffects } = arguments[0];
  const payload = renameKeysFromSDKRequestToRESTRequest({});
  const reqOpts = getOnboardingAvailability(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {},
        singleArgumentUnchanged: false
      },
      []
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateCbdFlow2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    cbdFlow: options?.cbdFlow
  });
  const reqOpts = updateCbdFlow(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { cbdFlow: "$[0].cbdFlow" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateRestrictedGoodsFlow2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    restrictedGoods: options?.restrictedGoods
  });
  const reqOpts = updateRestrictedGoodsFlow(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { restrictedGoods: "$[0].restrictedGoods" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updatePartnerFlow2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    partnerFlow: options?.partnerFlow
  });
  const reqOpts = updatePartnerFlow(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { partnerFlow: "$[0].partnerFlow" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
export {
  PartnerFlowStatus,
  RestrictedGoodsCategory,
  RestrictedGoodsFlowStatus,
  Status,
  WebhookIdentityType,
  getOnboardingAvailability2 as getOnboardingAvailability,
  updateCbdFlow2 as updateCbdFlow,
  updatePartnerFlow2 as updatePartnerFlow,
  updateRestrictedGoodsFlow2 as updateRestrictedGoodsFlow
};
