// @wix/auto-patterns/dist/cjs/components/AutoPatternsEntityPage/Fields/DateTime.js
"use strict";

exports.__esModule = true;
exports.DateTime = void 0;
var _react = _interopRequireWildcard(require("react"));
var _designSystem = require("@wix/design-system");
var _patterns = require("@wix/patterns");
var _form = require("@wix/patterns/form");
var _jsxFileName = "/home/builduser/work/73d19ce3378ce4dd/packages/auto-patterns/dist/cjs/components/AutoPatternsEntityPage/Fields/DateTime.tsx";
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
const DateTime = props => {
  var _pageState$entity, _field$validation2;
  const {
    field,
    dataHook,
    inputRef
  } = props;
  const {
    translate: t,
    ...container
  } = (0, _patterns.useWixPatternsContainer)();
  (0, _patterns.useSelector)(() => container.initTask.status);
  const pageState = (0, _patterns.useEntityPageContext)();
  const controller = (0, _form.useController)({
    name: (field == null ? void 0 : field.id) ?? '',
    control: pageState.form.control,
    defaultValue: (_pageState$entity = pageState.entity) == null ? void 0 : _pageState$entity[(field == null ? void 0 : field.id) ?? ''],
    rules: {
      validate: v => {
        var _field$validation;
        if (field != null && (_field$validation = field.validation) != null && _field$validation.required && (v === null || v === undefined || v === '')) {
          return t('cairo.fieldValidation.requiredField');
        }
        return true;
      }
    }
  });
  const date = (0, _react.useMemo)(() => {
    if (!controller.field.value) {
      return undefined;
    }
    const d = new Date(controller.field.value);
    return isNaN(d.getTime()) ? undefined : d;
  }, [controller.field.value]);
  const [dateStatusMessage, setDateStatusMessage] = (0, _react.useState)('');
  const [timeStatusMessage, setTimeStatusMessage] = (0, _react.useState)('');
  const ref = (0, _react.useRef)({});
  const dateRef = (0, _react.useRef)({});
  const timeRef = (0, _react.useRef)({});
  const [dateValue, setDateValue] = (0, _react.useState)(date == null ? void 0 : date.getDate());
  const [timeValue, setTimeValue] = (0, _react.useState)(date == null ? void 0 : date.getTime());
  (0, _react.useEffect)(() => {
    ref.current.focus = () => {
      if (dateRef.current.invalid) {
        return dateRef.current.focus == null ? void 0 : dateRef.current.focus();
      }
      return timeRef.current.focus == null ? void 0 : timeRef.current.focus();
    };
    inputRef == null || inputRef(ref.current);
  }, [inputRef]);
  return /*#__PURE__*/_react.default.createElement(_designSystem.FormField, {
    label: field.displayName,
    required: (_field$validation2 = field.validation) == null ? void 0 : _field$validation2.required,
    dataHook: dataHook,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 76,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.Layout, {
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 81,
      columnNumber: 7
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.Cell, {
    span: 6,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 82,
      columnNumber: 9
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.DatePicker, {
    ref: internalRef => {
      dateRef.current.focus = () => {
        var _internalRef$state;
        return (// TODO: PR to WSR needed to expose as public API
          // @ts-expect-error
          internalRef == null || (_internalRef$state = internalRef.state) == null || (_internalRef$state = _internalRef$state.inputRef) == null || _internalRef$state.focus == null ? void 0 : _internalRef$state.focus()
        );
      };
    },
    width: "100%",
    value: date
    // onChange is triggered only when you have valid, non-empty value. and onValidate is triggered on every change so we use onValidate
    ,
    onChange: () => {},
    onValidate: ({
      validationType,
      value: newValue,
      format
    }) => {
      if (validationType === 'valid') {
        const dateWithNewDate = date ? new Date(date) : new Date(newValue);
        const _value = new Date(newValue);
        dateWithNewDate.setFullYear(_value.getFullYear());
        dateWithNewDate.setMonth(_value.getMonth());
        dateWithNewDate.setDate(_value.getDate());
        controller.field.onChange(dateWithNewDate.toISOString());
        setDateValue(new Date(newValue).getDate());
        setDateStatusMessage('');
        dateRef.current.invalid = false;
      } else if (validationType === 'formatError' && newValue === '') {
        var _field$validation3;
        if ((_field$validation3 = field.validation) != null && _field$validation3.required) {
          setDateStatusMessage(t('cairo.fieldValidation.requiredField'));
          dateRef.current.invalid = true;
        } else {
          setDateStatusMessage('');
          dateRef.current.invalid = false;
        }
        setDateValue(newValue);
        if (!timeValue) {
          controller.field.onChange(null);
        }
      } else {
        setDateStatusMessage(t('cairo.customFields.fieldType.date.invalid.value', {
          dateFormat: format ?? ''
        }));
        dateRef.current.invalid = true;
      }
      ref.current.invalid = dateRef.current.invalid || timeRef.current.invalid;
    },
    status: controller.fieldState.error || dateStatusMessage ? 'error' : undefined,
    statusMessage: controller.fieldState.error || dateStatusMessage,
    dataHook: `datetime-date-${field.id}`,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 83,
      columnNumber: 11
    }
  })), /*#__PURE__*/_react.default.createElement(_designSystem.Cell, {
    span: 6,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 142,
      columnNumber: 9
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.TimeInput, {
    ref: internalRef => {
      timeRef.current.focus = () => internalRef == null ? void 0 : internalRef.focus();
    },
    dataHook: `datetime-time-${field.id}`,
    value: date ?? null,
    status: controller.fieldState.error || timeStatusMessage ? 'error' : undefined,
    statusMessage: controller.fieldState.error || timeStatusMessage,
    onInvalid: () => {
      timeRef.current.invalid = true;
      setTimeStatusMessage(t('cairo.customFields.fieldType.dateAndTime.invalid.value'));
      ref.current.invalid = dateRef.current.invalid || timeRef.current.invalid;
    },
    onChange: ({
      date: newValue
    }) => {
      timeRef.current.invalid = false;
      ref.current.invalid = dateRef.current.invalid || timeRef.current.invalid;
      if (newValue) {
        const dateWithNewTime = date ? new Date(date) : newValue;
        dateWithNewTime.setHours(newValue.getHours());
        dateWithNewTime.setMinutes(newValue.getMinutes());
        dateWithNewTime.setSeconds(newValue.getSeconds());
        dateWithNewTime.setMilliseconds(newValue.getMilliseconds());
        controller.field.onChange(dateWithNewTime.toISOString());
        setTimeStatusMessage('');
        setDateStatusMessage('');
      } else {
        if (!newValue) {
          var _field$validation4;
          setTimeValue('');
          if ((_field$validation4 = field.validation) != null && _field$validation4.required) {
            setTimeStatusMessage(t('cairo.fieldValidation.requiredField'));
            timeRef.current.invalid = true;
          } else {
            setTimeStatusMessage('');
            timeRef.current.invalid = false;
          }
          if (!ref.current.invalid) {
            if (!dateValue) {
              controller.field.onChange(null);
            }
          }
        }
      }
    },
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 143,
      columnNumber: 11
    }
  }))));
};
exports.DateTime = DateTime;
