// @wix/auto-patterns/dist/cjs/components/AutoPatternsEntityPage/Fields/FormFieldInput.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.FormFieldInput = exports.EDITABLE_INPUT_ELEMENTS = void 0;
var _react = _interopRequireDefault(require("react"));
var _ShortText = require("./ShortText");
var _patterns = require("@wix/patterns");
var _form = require("@wix/patterns/form");
var _Checkbox = require("./Checkbox");
var _DateInput = require("./DateInput");
var _DateTime = require("./DateTime");
var _Number = require("./Number");
var _LongText = require("./LongText");
var _Url = require("./Url");
var _ImageInput = require("./ImageInput");
var _jsxFileName = "/home/builduser/work/73d19ce3378ce4dd/packages/auto-patterns/dist/cjs/components/AutoPatternsEntityPage/Fields/FormFieldInput.tsx";
const EDITABLE_INPUT_ELEMENTS = exports.EDITABLE_INPUT_ELEMENTS = {
  SHORT_TEXT: _ShortText.ShortText,
  LONG_TEXT: _LongText.LongText,
  BOOLEAN: _Checkbox.Checkbox,
  DATE: _DateInput.DateInput,
  DATETIME: _DateTime.DateTime,
  NUMBER: _Number.Number,
  URL: _Url.Url,
  IMAGE: _ImageInput.ImageInput
};
const FormFieldInput = props => {
  var _pageState$entity;
  const {
    field,
    inputRef
  } = props;
  const pageState = (0, _patterns.useEntityPageContext)();
  const controller = (0, _form.useController)({
    name: (field == null ? void 0 : field.id) ?? '',
    control: pageState.form.control,
    defaultValue: (_pageState$entity = pageState.entity) == null ? void 0 : _pageState$entity[(field == null ? void 0 : field.id) ?? '']
  });
  if (!EDITABLE_INPUT_ELEMENTS.hasOwnProperty(field.type) || !pageState) {
    return null;
  }
  const FieldInput = EDITABLE_INPUT_ELEMENTS[field.type];
  return /*#__PURE__*/_react.default.createElement(FieldInput, {
    field: field,
    value: controller.field.value,
    onChange: controller.field.onChange,
    dataHook: `form-field-${field.id}`,
    inputRef: inputRef,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 51,
      columnNumber: 5
    }
  });
};
exports.FormFieldInput = FormFieldInput;
