// @wix/auto-patterns/dist/cjs/components/AutoPatternsEntityPage/Fields/ImageInput.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.ImageInput = void 0;
var _react = _interopRequireDefault(require("react"));
var _designSystem = require("@wix/design-system");
var _patterns = require("@wix/patterns");
var _form = require("@wix/patterns/form");
var _getImageUrl = require("../../../utils/media/getImageUrl");
var _dashboard = require("@wix/dashboard");
var _jsxFileName = "/home/builduser/work/73d19ce3378ce4dd/packages/auto-patterns/dist/cjs/components/AutoPatternsEntityPage/Fields/ImageInput.tsx";
const ImageInput = props => {
  var _pageState$entity, _field$validation2, _controller$fieldStat;
  const {
    field,
    dataHook
  } = props;
  const {
    translate: t,
    ...container
  } = (0, _patterns.useWixPatternsContainer)();
  (0, _patterns.useSelector)(() => container.initTask.status);
  const pageState = (0, _patterns.useEntityPageContext)();
  async function openMedia() {
    try {
      var _chosenMediaItems$ite;
      const chosenMediaItems = await _dashboard.dashboard.openMediaManager({
        multiSelect: false,
        category: 'IMAGE'
      });
      const newImage = chosenMediaItems == null || (_chosenMediaItems$ite = chosenMediaItems.items[0]) == null || (_chosenMediaItems$ite = _chosenMediaItems$ite.media) == null || (_chosenMediaItems$ite = _chosenMediaItems$ite.image) == null ? void 0 : _chosenMediaItems$ite.image;
      controller.field.onChange(newImage);
    } catch (error) {
      console.error('Error opening Media Manager or no files selected:', error);
    }
  }
  const controller = (0, _form.useController)({
    name: (field == null ? void 0 : field.id) ?? '',
    control: pageState.form.control,
    defaultValue: (_pageState$entity = pageState.entity) == null ? void 0 : _pageState$entity[(field == null ? void 0 : field.id) ?? ''],
    rules: {
      validate: value => {
        var _field$validation;
        if ((_field$validation = field.validation) != null && _field$validation.required && (value === null || value === undefined || value === '')) {
          return t('cairo.fieldValidation.requiredField');
        }
        return true;
      }
    }
  });
  const imageUrl = controller.field.value && (0, _getImageUrl.getImageUrl)(controller.field.value);
  return /*#__PURE__*/_react.default.createElement(_designSystem.FormField, {
    label: field.displayName,
    required: (_field$validation2 = field.validation) == null ? void 0 : _field$validation2.required,
    status: controller.fieldState.error ? 'error' : undefined,
    statusMessage: (_controller$fieldStat = controller.fieldState.error) == null ? void 0 : _controller$fieldStat.message,
    dataHook: dataHook,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 60,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.ImageViewer, {
    imageUrl: imageUrl,
    dataHook: `image-${field.id}`,
    onAddImage: () => openMedia(),
    onUpdateImage: () => openMedia(),
    updateImageInfo: t('cairo.replace-image.tooltip'),
    removeImageInfo: t('cairo.remove-image.tooltip'),
    addImageInfo: t('cairo.add-image.tooltip'),
    height: "120px",
    width: "25%",
    fit: "cover",
    onRemoveImage: () => controller.field.onChange(null),
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 67,
      columnNumber: 7
    }
  }));
};
exports.ImageInput = ImageInput;
