// @wix/auto-patterns/dist/cjs/components/AutoPatternsEntityPage/Fields/Number.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.Number = void 0;
var _react = _interopRequireDefault(require("react"));
var _designSystem = require("@wix/design-system");
var _patterns = require("@wix/patterns");
var _form = require("@wix/patterns/form");
var _jsxFileName = "/home/builduser/work/73d19ce3378ce4dd/packages/auto-patterns/dist/cjs/components/AutoPatternsEntityPage/Fields/Number.tsx";
const Number = props => {
  var _field$validation, _pageState$entity, _field$validation3, _controller$fieldStat;
  const {
    field,
    dataHook
  } = props;
  const {
    translate: t,
    ...container
  } = (0, _patterns.useWixPatternsContainer)();
  (0, _patterns.useSelector)(() => container.initTask.status);
  const pageState = (0, _patterns.useEntityPageContext)();
  const numberRange = (_field$validation = field.validation) == null ? void 0 : _field$validation.numberRange;
  const min = numberRange == null ? void 0 : numberRange.min;
  const max = numberRange == null ? void 0 : numberRange.max;
  const controller = (0, _form.useController)({
    name: (field == null ? void 0 : field.id) ?? '',
    control: pageState.form.control,
    defaultValue: (_pageState$entity = pageState.entity) == null ? void 0 : _pageState$entity[(field == null ? void 0 : field.id) ?? ''],
    rules: {
      validate: value => {
        var _field$validation2;
        if ((_field$validation2 = field.validation) != null && _field$validation2.required && (value === null || value === undefined || value === '')) {
          return t('cairo.fieldValidation.requiredField');
        }
        if (value !== null && value !== undefined) {
          if (min !== undefined && value < min) {
            return t('cairo.fieldValidation.minNumber', {
              minAmount: min
            });
          }
          if (max !== undefined && value > max) {
            return t('cairo.fieldValidation.maxNumber', {
              maxAmount: max
            });
          }
        }
        return true;
      }
    }
  });
  return /*#__PURE__*/_react.default.createElement(_designSystem.FormField, {
    label: field.displayName,
    required: (_field$validation3 = field.validation) == null ? void 0 : _field$validation3.required,
    status: controller.fieldState.error ? 'error' : undefined,
    statusMessage: (_controller$fieldStat = controller.fieldState.error) == null ? void 0 : _controller$fieldStat.message,
    dataHook: dataHook,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 57,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.NumberInput, {
    value: controller.field.value ?? undefined,
    onChange: e => controller.field.onChange(e),
    onBlur: controller.field.onBlur,
    inputRef: controller.field.ref,
    dataHook: `number-${field.id}`,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 64,
      columnNumber: 7
    }
  }));
};
exports.Number = Number;
