// @wix/auto-patterns/dist/cjs/components/AutoPatternsEntityPage/Fields/Url.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.Url = void 0;
var _react = _interopRequireDefault(require("react"));
var _designSystem = require("@wix/design-system");
var _wixUiIconsCommon = require("@wix/wix-ui-icons-common");
var _patterns = require("@wix/patterns");
var _form = require("@wix/patterns/form");
var _jsxFileName = "/home/builduser/work/73d19ce3378ce4dd/packages/auto-patterns/dist/cjs/components/AutoPatternsEntityPage/Fields/Url.tsx";
const Url = props => {
  var _pageState$entity, _field$validation2, _controller$fieldStat;
  const {
    field,
    dataHook
  } = props;
  const {
    translate: t,
    ...container
  } = (0, _patterns.useWixPatternsContainer)();
  (0, _patterns.useSelector)(() => container.initTask.status);
  const pageState = (0, _patterns.useEntityPageContext)();

  // Same regex as data-extensions https://github.com/wix-private/server-infra/blob/a6c3009d39bbb0312c929f68ee81c559e1eb5b4e/iptf/dataextensions/src/main/scala/com/wixpress/infra/dataextensions/runtime/validator/Formats.scala#L31
  const uriPattern =
  // eslint-disable-next-line no-useless-escape
  /^[A-Za-z][A-Za-z0-9+.-]*:(\/\/([A-Za-z0-9._~\-%!$&'()*+,;=:]*@)?[A-Za-z0-9._~\-!$&'()*+,;=%:\[\]]*(:[0-9]*)?)?[A-Za-z0-9._~\-%!$&'()*+,;=:@\/]*([?][A-Za-z0-9._~\-%!$&'()*+,;=:@\/?]*)?([#][A-Za-z0-9._~\-%!$&'()*+,;=:@\/?]*)?$/;
  const controller = (0, _form.useController)({
    name: (field == null ? void 0 : field.id) ?? '',
    control: pageState.form.control,
    defaultValue: (_pageState$entity = pageState.entity) == null ? void 0 : _pageState$entity[(field == null ? void 0 : field.id) ?? ''],
    rules: {
      validate: value => {
        var _field$validation;
        if ((_field$validation = field.validation) != null && _field$validation.required && (value === null || value === undefined || value === '')) {
          return t('cairo.fieldValidation.requiredField');
        } else if (value) {
          return uriPattern.test(value) || t('cairo.customFields.url-input.error');
        }
        return true;
      }
    }
  });
  return /*#__PURE__*/_react.default.createElement(_designSystem.FormField, {
    label: field.displayName,
    required: (_field$validation2 = field.validation) == null ? void 0 : _field$validation2.required,
    dataHook: dataHook,
    status: controller.fieldState.error ? 'error' : undefined,
    statusMessage: (_controller$fieldStat = controller.fieldState.error) == null ? void 0 : _controller$fieldStat.message,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 47,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.Input, {
    prefix: /*#__PURE__*/_react.default.createElement(_designSystem.Input.IconAffix, {
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 56,
        columnNumber: 11
      }
    }, /*#__PURE__*/_react.default.createElement(_wixUiIconsCommon.Link, {
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 57,
        columnNumber: 13
      }
    })),
    placeholder: t('cairo.customFields.url-input.placeholder'),
    dataHook: `url-${field.id}`,
    value: controller.field.value,
    onChange: e => controller.field.onChange == null ? void 0 : controller.field.onChange(e.target.value),
    onBlur: controller.field.onBlur,
    ref: controller.field.ref,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 54,
      columnNumber: 7
    }
  }));
};
exports.Url = Url;
