// @wix/bex-utils/@wix/ambassador-devcenter-app-market-listing-v1-market-listing/build/es/builders.impl.js
import { mergeDeep, someNumber, someString, someAny, someRepeated, someDate, } from '@wix/metro-common-builders';
import { MediaType, SortOrder, Status, VersionStatus, Classification, WebhookIdentityType, } from './types.impl';
// @ts-ignore
import stack from 'callsite';
export function aMarketListing(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        id: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        appId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        languageCode: someString({
            validations: [{ type: 'FORMAT', value: 'LANGUAGE' }],
        }),
        basicInfo: aBasicInfo(),
        assetsMap: { assetsMap: aAssets() },
        contactInfo: aContactInfo(),
        installationRequirement: aInstallationRequirement(),
        revision: someString({
            validations: [{ type: 'DECIMAL_VALUE' }],
            isUnsigned: true,
        }),
        createdDate: someDate({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        updatedDate: someDate({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        status: aStatus(),
        classification: aClassification(),
        pricingData: aPricingData(),
    }, overrideObject);
}
export function aBasicInfo(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        name: someString(),
        teaser: someString(),
        longDescription: someString(),
        benefits: [aBenefits()],
        searchKeywords: [someString()],
        demoUrl: someString({
            validations: [{ type: 'FORMAT', value: 'WEB_URL' }],
        }),
        categories: [someString()],
    }, overrideObject);
}
export function aBenefits(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        title: someString(),
    }, overrideObject);
}
export function aAssets(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        assets: [aAsset()],
    }, overrideObject);
}
export function aAsset(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        url: someString({
            validations: [{ type: 'FORMAT', value: 'WEB_URL' }],
        }),
        mediaType: aMediaType(),
    }, overrideObject);
}
export function aMediaType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = MediaType.VIDEO; }
    return enumFirstItem;
}
export function aContactInfo(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        website: someString({
            validations: [{ type: 'FORMAT', value: 'WEB_URL' }],
        }),
        developerMetaSiteId: someString(),
        supportEmail: someString({
            validations: [{ type: 'FORMAT', value: 'EMAIL' }],
        }),
        privacyPolicyUrl: someString({
            validations: [{ type: 'FORMAT', value: 'WEB_URL' }],
        }),
        termsOfUseUrl: someString({
            validations: [{ type: 'FORMAT', value: 'WEB_URL' }],
        }),
    }, overrideObject);
}
export function aInstallationRequirement(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        requiredApps: [aRequiredApp()],
        includeCountries: [
            someString({
                validations: [
                    { type: 'FORMAT', value: 'COUNTRY' },
                ],
            }),
        ],
        excludeCountries: [
            someString({
                validations: [
                    { type: 'FORMAT', value: 'COUNTRY' },
                ],
            }),
        ],
        includeLanguages: [
            someString({
                validations: [
                    { type: 'FORMAT', value: 'LANGUAGE' },
                ],
            }),
        ],
        excludeLanguages: [
            someString({
                validations: [
                    { type: 'FORMAT', value: 'LANGUAGE' },
                ],
            }),
        ],
        allowLanguageCustomization: true,
        requiredWixPremium: true,
    }, overrideObject);
}
export function aRequiredApp(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        id: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        slug: someString(),
    }, overrideObject);
}
export function aStatus(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = Status.DRAFT; }
    return enumFirstItem;
}
export function aClassification(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = Classification.SITE_APP; }
    return enumFirstItem;
}
export function aPricingData(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        externalPricingPageUrl: someString(),
        externalCheckout: true,
        freeTrialDays: someNumber(),
        internalFreeTrialDays: someNumber({
            validations: [
                { type: 'MIN' },
                { type: 'MAX', value: '30' },
            ],
        }),
    }, overrideObject);
}
export function aCreateMarketListingRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        marketListing: aMarketListing(),
    }, overrideObject);
}
export function aCreateMarketListingResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        marketListing: aMarketListing(),
    }, overrideObject);
}
export function aGetMarketListingRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        id: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
export function aGetMarketListingResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        marketListing: aMarketListing(),
    }, overrideObject);
}
export function aUpdateMarketListingRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        marketListing: aMarketListing(),
        mask: someRepeated(),
    }, overrideObject);
}
export function aUpdateMarketListingResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        marketListing: aMarketListing(),
    }, overrideObject);
}
export function aOldListingChanged(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        appId: someString(),
        marketListing: aApiMarketListing(),
        languageCode: someString({
            validations: [{ type: 'FORMAT', value: 'LANGUAGE' }],
        }),
        name: someString(),
        iconUrl: someString(),
    }, overrideObject);
}
export function aApiMarketListing(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        basicInfo: aApiBasicInfo(),
        assetsMap: { assetsMap: aAssets() },
        contactInfo: aApiContactInfo(),
        trackingAndAnalytics: aTracking(),
        languageCode: someString({
            validations: [{ type: 'FORMAT', value: 'LANGUAGE' }],
        }),
        installationRequirement: aApiInstallationRequirement(),
    }, overrideObject);
}
export function aApiBasicInfo(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        appId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        name: someString(),
        teaser: someString(),
        longDescription: someString(),
        benefits: [aApiBenefits()],
        searchKeywords: [someString()],
        demoUrl: someString({
            validations: [{ type: 'FORMAT', value: 'URI' }],
        }),
        categories: [someString()],
    }, overrideObject);
}
export function aApiBenefits(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        title: someString(),
    }, overrideObject);
}
export function aApiContactInfo(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        website: someString({
            validations: [{ type: 'FORMAT', value: 'URI' }],
        }),
        developerMetaSiteId: someString(),
        supportEmail: someString({
            validations: [{ type: 'FORMAT', value: 'EMAIL' }],
        }),
        privacyPolicyUrl: someString({
            validations: [{ type: 'FORMAT', value: 'URI' }],
        }),
        termsOfUseUrl: someString({
            validations: [{ type: 'FORMAT', value: 'URI' }],
        }),
    }, overrideObject);
}
export function aTracking(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        googleAnalyticsTrackingId: someString(),
        googleConversionId: someString(),
        adRolllAdvId: someString(),
        adRolllPixId: someString(),
    }, overrideObject);
}
export function aApiInstallationRequirement(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        requiredApps: [aApiRequiredApp()],
        includeCountries: [
            someString({
                validations: [
                    { type: 'FORMAT', value: 'COUNTRY' },
                ],
            }),
        ],
        excludeCountries: [
            someString({
                validations: [
                    { type: 'FORMAT', value: 'COUNTRY' },
                ],
            }),
        ],
        includeLanguages: [
            someString({
                validations: [
                    { type: 'FORMAT', value: 'LANGUAGE' },
                ],
            }),
        ],
        excludeLanguages: [
            someString({
                validations: [
                    { type: 'FORMAT', value: 'LANGUAGE' },
                ],
            }),
        ],
        allowLanguageCustomization: true,
    }, overrideObject);
}
export function aApiRequiredApp(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        id: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        slug: someString(),
    }, overrideObject);
}
export function aEmpty(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({}, overrideObject);
}
export function aAppVersionStateChanged(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        created: aAppVersionCreated(),
        appId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        version: someString({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        previousState: aVersionStatus(),
        partialVersionOptions: aPartialVersionOptions(),
    }, overrideObject);
}
export function aAppVersionCreated(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({}, overrideObject);
}
export function aAppVersionSubmitted(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({}, overrideObject);
}
export function aAppVersionPublished(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({}, overrideObject);
}
export function aAppVersionDeclined(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({}, overrideObject);
}
export function aAppVersionReleased(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({}, overrideObject);
}
export function aAppVersionApproved(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({}, overrideObject);
}
export function aAppVersionDraftChanged(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({}, overrideObject);
}
export function aAppVersionArchived(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({}, overrideObject);
}
export function aVersionStatus(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = VersionStatus.NONE_STATUS; }
    return enumFirstItem;
}
export function aPartialVersionOptions(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        includeDraftDevCenterData: true,
        includeDraftMarketListingData: true,
        includeDraftPricingInfoData: true,
    }, overrideObject);
}
export function aQueryMarketListingRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        query: aQueryV2(),
    }, overrideObject);
}
export function aQueryV2(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        paging: aPaging(),
        filter: {},
        sort: [aSorting()],
        fields: [someString()],
        fieldsets: [someString()],
    }, overrideObject);
}
export function aSorting(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        fieldName: someString({
            validations: [{ type: 'MAX_LENGTH', value: '512' }],
        }),
        order: aSortOrder(),
        selectItemsBy: [{}, {}, {}, {}, {}, {}, {}, {}, {}, {}],
    }, overrideObject);
}
export function aSortOrder(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = SortOrder.ASC; }
    return enumFirstItem;
}
export function aPaging(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        limit: someNumber({
            validations: [{ type: 'MIN' }],
        }),
        offset: someNumber({
            validations: [{ type: 'MIN' }],
        }),
    }, overrideObject);
}
export function aCursorPaging(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        limit: someNumber({
            validations: [
                { type: 'MIN' },
                { type: 'MAX', value: '100' },
            ],
        }),
        cursor: someString({
            validations: [
                { type: 'MAX_LENGTH', value: '16000' },
            ],
        }),
    }, overrideObject);
}
export function aQueryMarketListingResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        marketListing: [
            aMarketListing(),
            aMarketListing(),
            aMarketListing(),
            aMarketListing(),
            aMarketListing(),
            aMarketListing(),
            aMarketListing(),
            aMarketListing(),
            aMarketListing(),
            aMarketListing(),
        ],
        metadata: aPagingMetadataV2(),
    }, overrideObject);
}
export function aPagingMetadataV2(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        count: someNumber(),
        offset: someNumber(),
        total: someNumber(),
        tooManyToCount: true,
        cursors: aCursors(),
        hasNext: true,
    }, overrideObject);
}
export function aCursors(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        next: someString({
            validations: [
                { type: 'MAX_LENGTH', value: '16000' },
            ],
        }),
        prev: someString({
            validations: [
                { type: 'MAX_LENGTH', value: '16000' },
            ],
        }),
    }, overrideObject);
}
export function aDomainEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        createdEvent: aEntityCreatedEvent(),
        id: someString(),
        entityFqdn: someString(),
        slug: someString(),
        entityId: someString(),
        eventTime: someDate(),
        triggeredByAnonymizeRequest: true,
        originatedFrom: someString(),
        entityEventSequence: someString({
            validations: [{ type: 'DECIMAL_VALUE' }],
        }),
    }, overrideObject);
}
export function aEntityCreatedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        entityAsJson: someString(),
        triggeredByUndelete: true,
        restoreInfo: aRestoreInfo(),
        additionalMetadataAsJson: someString(),
    }, overrideObject);
}
export function aRestoreInfo(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        deletedDate: someDate(),
    }, overrideObject);
}
export function aEntityUpdatedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        currentEntityAsJson: someString(),
        previousEntityAsJson: someString(),
        modifiedFields: { modifiedFields: someAny() },
        additionalMetadataAsJson: someString(),
    }, overrideObject);
}
export function aEntityDeletedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        movedToTrash: true,
        deletedEntityAsJson: someString(),
        additionalMetadataAsJson: someString(),
    }, overrideObject);
}
export function aActionEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        bodyAsJson: someString(),
    }, overrideObject);
}
export function aPromoteRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        appId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        from: aStatus(),
        to: aStatus(),
    }, overrideObject);
}
export function aPromoteResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        appId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        status: aStatus(),
    }, overrideObject);
}
export function aCompareRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        appId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        from: aStatus(),
        to: aStatus(),
    }, overrideObject);
}
export function aCompareResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        equals: true,
    }, overrideObject);
}
export function aMessageEnvelope(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        instanceId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        eventType: someString({
            validations: [{ type: 'MAX_LENGTH', value: '150' }],
        }),
        identity: aIdentificationData(),
        data: someString(),
    }, overrideObject);
}
export function aIdentificationData(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        anonymousVisitorId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        identityType: aWebhookIdentityType(),
    }, overrideObject);
}
export function aWebhookIdentityType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = WebhookIdentityType.UNKNOWN; }
    return enumFirstItem;
}
function isCurrentBuilderWasPreviouslyCalled() {
    var callChainFunctionNames = stack().map(function (item) {
        return item.getFunctionName();
    });
    var callChainWithoutCurrentBuilder = callChainFunctionNames.splice(2);
    var currentBuilderName = callChainFunctionNames[1];
    return callChainWithoutCurrentBuilder.some(function (functionName) { return functionName === currentBuilderName; });
}
export * from '@wix/metro-common-builders';
