// @wix/bex-utils/@wix/ambassador-devcenter-app-market-listing-v1-market-listing/build/es/meta.impl.js
import * as httpWixDevcenterAppMarketListingV1MarketListing from './http.impl';
export function createMarketListing() {
    var payload = {};
    var getRequestOptions = httpWixDevcenterAppMarketListingV1MarketListing.createMarketListing(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/marketListings',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getMarketListing() {
    var payload = { id: ':id' };
    var getRequestOptions = httpWixDevcenterAppMarketListingV1MarketListing.getMarketListing(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/v1/marketListings/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateMarketListing() {
    var payload = { marketListing: { id: ':marketListingId' } };
    var getRequestOptions = httpWixDevcenterAppMarketListingV1MarketListing.updateMarketListing(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'PUT',
        path: '/v1/marketListings/{marketListing.id}',
        pathParams: { marketListingId: 'marketListingId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryMarketListing() {
    var payload = {};
    var getRequestOptions = httpWixDevcenterAppMarketListingV1MarketListing.queryMarketListing(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/marketListings/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function compare() {
    var payload = {};
    var getRequestOptions = httpWixDevcenterAppMarketListingV1MarketListing.compare(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/market-listings/compare',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
