// @wix/headless-ecom/cjs/dist/react/Cart.js
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
/**
 * @fileoverview Cart Primitive Components
 *
 * This module provides unstyled, composable components for building cart functionality.
 * These components follow the Radix UI primitive pattern, offering:
 *
 * - **Unstyled**: No default styling, only functional behavior
 * - **Composable**: Support for the `asChild` pattern for flexible DOM structure
 * - **Accessible**: Built-in ARIA attributes and proper semantics
 * - **Flexible**: Render props pattern for maximum customization
 *
 * ## Architecture
 *
 * These components are the **primitive layer** that sits between:
 * 1. **Core components** (pure logic, no DOM)
 * 2. **Styled components** (project-specific styling)
 *
 * ## Usage
 *
 * ```tsx
 * import { Cart } from '@wix/ecom/react';
 *
 * function CartProvider() {
 *   return (
 *     <Cart.Root currentCartServiceConfig={config}>
 *       <Cart.OpenTrigger>
 *         {({ totalItems, open }) => (
 *           <button onClick={open}>Cart ({totalItems})</button>
 *         )}
 *       </Cart.OpenTrigger>
 *       <Cart.Content>
 *         {({ cart, isLoading }) => (
 *           <div>
 *             {isLoading ? 'Loading...' : 'Cart content'}
 *           </div>
 *         )}
 *       </Cart.Content>
 *     </Cart.Root>
 *   );
 * }
 * ```
 *
 * @module Cart
 */
import { EmptyState as CoreEmptyState, LineItemsList as CoreLineItemsList, Summary as CoreSummary, Content as CoreContent, Clear as CoreClear, Checkout as CoreCheckout, Notes as CoreNotes, LineItemAdded as CoreLineItemAdded, } from './core/CurrentCart.js';
import { useService } from '@wix/services-manager-react';
import React from 'react';
import { CurrentCartServiceDefinition } from '../services/current-cart-service.js';
import { AsChildSlot } from '@wix/headless-utils/react';
import * as LineItemComponent from './LineItem.js';
import * as CouponComponents from './CartCoupon.js';
import { DataComponentTags } from '../data-component-tags.js';
// Components that render actual DOM elements get test IDs on their rendered elements
// Components that only provide context/logic don't introduce new DOM elements
var TestIds;
(function (TestIds) {
    TestIds["cartRoot"] = "cart-root";
    TestIds["cartEmptyState"] = "cart-empty-state";
    TestIds["cartOpenTrigger"] = "cart-open-trigger";
    TestIds["cartLineItemsList"] = "cart-line-items-list";
    TestIds["cartItem"] = "cart-item";
    TestIds["cartSummary"] = "cart-summary";
    TestIds["cartClear"] = "cart-clear";
    TestIds["cartCheckout"] = "cart-checkout";
    TestIds["cartErrors"] = "cart-errors";
    TestIds["cartNotes"] = "cart-notes";
    TestIds["cartCoupon"] = "cart-coupon";
    TestIds["cartNoteInput"] = "cart-note-input";
    TestIds["cartLineItems"] = "cart-line-items";
    TestIds["cartLineItemRepeater"] = "cart-line-item-repeater";
})(TestIds || (TestIds = {}));
/**
 * Root component that provides cart service context to its children.
 * This is a primitive wrapper around the core Root component that maintains
 * the same API while providing a foundation for composition patterns.
 *
 * @component
 * @example
 * ```tsx
 * import { Cart } from '@wix/ecom/react';
 *
 * function CartProvider() {
 *   return (
 *     <Cart.Root currentCartServiceConfig={config}>
 *       <Cart.OpenTrigger>
 *         {({ totalItems, open }) => (
 *           <button onClick={open}>Cart ({totalItems})</button>
 *         )}
 *       </Cart.OpenTrigger>
 *     </Cart.Root>
 *   );
 * }
 * ```
 */
export const Root = ({ children }) => {
    return (_jsx(AsChildSlot, { "data-component-tag": DataComponentTags.cartRoot, "data-testid": TestIds.cartRoot, children: children }));
};
Root.displayName = 'Cart.Root';
/**
 * Component that renders content when the cart is empty.
 * Only displays its children when there are no items in cart, no loading state, and no errors.
 * Supports the asChild pattern for flexible composition.
 *
 * @component
 * @example
 * ```tsx
 * // Default rendering with default label
 * <Cart.EmptyState />
 *
 * // Custom label
 * <Cart.EmptyState label="Your cart is empty" />
 *
 * // Custom content with render function
 * <Cart.EmptyState>
 *   {() => (
 *     <div>
 *       <h3>No items in cart</h3>
 *       <p>Items will appear here once they are added</p>
 *     </div>
 *   )}
 * </Cart.EmptyState>
 *
 * // Using asChild for custom wrapper
 * <Cart.EmptyState asChild>
 *   <section className="empty-cart-section">
 *     Content will be rendered here
 *   </section>
 * </Cart.EmptyState>
 * ```
 */
export const EmptyState = React.forwardRef(({ asChild, children, label = 'No items in cart' }, ref) => {
    return (_jsx(CoreEmptyState, { children: _jsx(AsChildSlot, { asChild: asChild, customElement: children, customElementProps: {}, ref: ref, "data-testid": TestIds.cartEmptyState, children: label }) }));
});
/**
 * Headless component for cart items collection.
 * Supports the asChild pattern for flexible composition.
 *
 * @component
 * @example
 * ```tsx
 * // Default div wrapper
 * <Cart.LineItemsList>
 *   {({ items, totalItems }) => (
 *     <div>
 *       <h1>Cart ({totalItems} items)</h1>
 *       {items.map(item => <div key={item._id}>{item.productName?.original}</div>)}
 *     </div>
 *   )}
 * </Cart.LineItemsList>
 *
 * // Using asChild for custom wrapper
 * <Cart.LineItemsList asChild>
 *   <ul className="cart-items-list">
 *     Items content will be rendered here
 *   </ul>
 * </Cart.LineItemsList>
 * ```
 */
export const LineItemsList = React.forwardRef(({ asChild, children, ...props }, ref) => {
    return (_jsx(CoreLineItemsList, { children: (renderProps) => {
            if (!renderProps.totalItems) {
                return props.emptyState || null;
            }
            return (_jsxs(AsChildSlot, { asChild: asChild, customElement: children, customElementProps: renderProps, ref: ref, "data-testid": TestIds.cartLineItemsList, ...props, children: [children ? children : null, _jsx(EmptyState, { asChild: asChild, children: props.emptyState })] }));
        } }));
});
/**
 * Headless component for cart items collection (alternative to LineItemsList)
 *
 * @example
 * ```tsx
 * <Cart.LineItems
 *   emptyState={<div>Your cart is empty</div>}
 * >
 *   <Cart.LineItemRepeater>
 *     <LineItem.Image />
 *     <LineItem.Title />
 *     <LineItem.SelectedOptions />
 *   </Cart.LineItemRepeater>
 * </Cart.LineItems>
 * ```
 */
export const LineItems = React.forwardRef((props, ref) => {
    const { asChild, children, emptyState, ...otherProps } = props;
    const service = useService(CurrentCartServiceDefinition);
    const cart = service.cart.get();
    // Show empty state if cart is null or has no items
    if (!cart?.lineItems?.length && emptyState) {
        return emptyState;
    }
    return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, "data-testid": TestIds.cartLineItems, customElement: children, ...otherProps, children: _jsx("div", { children: children }) }));
});
/**
 * Component that repeats its children for each line item in the cart.
 * Provides context for each line item.
 *
 * @component
 * @example
 * ```tsx
 * import { Cart } from '@wix/ecom/components';
 *
 * function CartItemsList() {
 *   return (
 *     <Cart.LineItems>
 *       <Cart.LineItemRepeater>
 *         <LineItem.Title />
 *         <LineItem.Image />
 *         <LineItem.Quantity />
 *       </Cart.LineItemRepeater>
 *     </Cart.LineItems>
 *   );
 * }
 * ```
 */
export const LineItemRepeater = React.forwardRef((props, ref) => {
    const service = useService(CurrentCartServiceDefinition);
    const { asChild, children, className, ...otherProps } = props;
    const cart = service.cart.get();
    const items = cart?.lineItems || [];
    const content = items.map((item, index) => (_jsx(LineItemComponent.Root, { item: item, children: React.isValidElement(children) ? children : null }, item._id || index)));
    return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.cartLineItemRepeater, customElement: children, customElementProps: {}, ...otherProps, children: _jsx("div", { children: content }) }));
});
export const Content = (props) => {
    return _jsx(CoreContent, { children: props.children });
};
/**
 * Headless component for cart summary/totals.
 * Supports the asChild pattern for flexible composition.
 *
 * @component
 * @example
 * ```tsx
 * // Default div wrapper
 * <Cart.Summary>
 *   {({ subtotal, total, totalItems, isTotalsLoading }) => (
 *     <div>
 *       <p>Subtotal ({totalItems} items): {subtotal}</p>
 *       <p>Total: {isTotalsLoading ? 'Calculating...' : total}</p>
 *     </div>
 *   )}
 * </Cart.Summary>
 *
 * // With custom labels
 * <Cart.Summary
 *   labels={{
 *     subtotal: 'Items ({totalItems})',
 *     total: 'Grand Total',
 *     calculating: 'Computing totals...'
 *   }}
 * />
 *
 * // Using asChild for custom wrapper
 * <Cart.Summary asChild>
 *   <section className="cart-summary">
 *     Summary content will be rendered here
 *   </section>
 * </Cart.Summary>
 * ```
 */
export const Summary = React.forwardRef(({ asChild, children, labels, ...props }, ref) => {
    const defaultLabels = {
        subtotal: 'Subtotal ({totalItems} items)',
        total: 'Total',
        calculating: 'Calculating...',
    };
    const mergedLabels = { ...defaultLabels, ...labels };
    return (_jsx(CoreSummary, { children: (renderProps) => {
            const subtotalLabel = mergedLabels.subtotal.replace('{totalItems}', renderProps.totalItems.toString());
            return (_jsx(AsChildSlot, { asChild: asChild, customElement: children, customElementProps: renderProps, ref: ref, "data-testid": TestIds.cartSummary, ...props, children: _jsxs("div", { children: [_jsxs("p", { children: [subtotalLabel, ": ", renderProps.subtotal] }), _jsxs("p", { children: [mergedLabels.total, ":", ' ', renderProps.isTotalsLoading
                                    ? mergedLabels.calculating
                                    : renderProps.total] })] }) }));
        } }));
});
/**
 * Headless component for clearing the cart.
 * Supports the asChild pattern for flexible composition.
 *
 * @component
 * @example
 * ```tsx
 * // Default button
 * <Cart.Clear>
 *   {({ clear, totalItems, isLoading }) => (
 *     <button onClick={clear} disabled={isLoading || totalItems === 0}>
 *       {isLoading ? 'Clearing...' : `Clear Cart (${totalItems})`}
 *     </button>
 *   )}
 * </Cart.Clear>
 *
 * // With custom labels
 * <Cart.Clear
 *   labels={{
 *     clearing: 'Removing items...',
 *     clear: 'Empty Cart ({totalItems} items)'
 *   }}
 * />
 *
 * // Using asChild for custom button
 * <Cart.Clear asChild>
 *   <button className="danger-button">
 *     Clear action content will be rendered here
 *   </button>
 * </Cart.Clear>
 * ```
 */
export const Clear = React.forwardRef(({ children, asChild, labels, ...props }, ref) => {
    const defaultLabels = {
        clearing: 'Clearing...',
        clear: 'Clear Cart ({totalItems})',
    };
    const mergedLabels = { ...defaultLabels, ...labels };
    return (_jsx(CoreClear, { children: (renderProps) => {
            if (renderProps.totalItems === 0) {
                return null;
            }
            const clearingLabel = mergedLabels.clearing;
            const clearLabel = mergedLabels.clear.replace('{totalItems}', renderProps.totalItems.toString());
            return (_jsx(AsChildSlot, { asChild: asChild, customElement: children, customElementProps: renderProps, ref: ref, onClick: renderProps.clear, disabled: renderProps.isLoading, "data-testid": TestIds.cartClear, ...props, children: _jsx("button", { onClick: renderProps.clear, disabled: renderProps.isLoading || renderProps.totalItems === 0, children: renderProps.isLoading ? clearingLabel : clearLabel }) }));
        } }));
});
/**
 * Headless component for checkout action.
 * Supports the asChild pattern for flexible composition.
 *
 * @component
 * @example
 * ```tsx
 * // Default button
 * <Cart.Checkout>
 *   {({ proceedToCheckout, canCheckout, isLoading, error }) => (
 *     <div>
 *       {error && <p>Error: {error}</p>}
 *       <button onClick={proceedToCheckout} disabled={!canCheckout || isLoading}>
 *         {isLoading ? 'Processing...' : 'Proceed to Checkout'}
 *       </button>
 *     </div>
 *   )}
 * </Cart.Checkout>
 *
 * // Using asChild for custom button
 * <Cart.Checkout asChild>
 *   <button className="primary-button">
 *     Checkout content will be rendered here
 *   </button>
 * </Cart.Checkout>
 * ```
 */
export const Checkout = React.forwardRef(({ children, asChild, ...props }, ref) => {
    return (_jsx(CoreCheckout, { children: (renderProps) => {
            return (_jsx(AsChildSlot, { asChild: asChild, customElement: children, customElementProps: renderProps, ref: ref, onClick: renderProps.proceedToCheckout, disabled: !renderProps.canCheckout || renderProps.isLoading, "data-testid": TestIds.cartCheckout, ...props, children: _jsx("button", { onClick: renderProps.proceedToCheckout, disabled: !renderProps.canCheckout || renderProps.isLoading, children: renderProps.isLoading
                        ? 'Processing...'
                        : 'Proceed to Checkout' }) }));
        } }));
});
/**
 * Headless component for notes.
 * Supports the asChild pattern for flexible composition.
 *
 * @component
 * @example
 * ```tsx
 * // Default div wrapper
 * <Cart.Notes>
 *   {({ notes, updateNotes }) => (
 *     <textarea
 *       value={notes}
 *       onChange={e => updateNotes(e.target.value)}
 *       placeholder="Special instructions for your order"
 *     />
 *   )}
 * </Cart.Notes>
 *
 * // Using asChild for custom wrapper
 * <Cart.Notes asChild>
 *   <div className="notes-section">
 *     Notes content will be rendered here
 *   </div>
 * </Cart.Notes>
 * ```
 */
export const Notes = React.forwardRef(({ asChild, children, ...props }, ref) => {
    return (_jsx(CoreNotes, { children: (renderProps) => {
            return (_jsx(AsChildSlot, { asChild: asChild, customElement: children, customElementProps: renderProps, ref: ref, "data-testid": TestIds.cartNotes, ...props, children: _jsxs("div", { children: [_jsx("label", { className: "block text-sm font-medium text-content-primary mb-2", children: "Notes:" }), _jsx("textarea", { value: renderProps.notes, onChange: (e) => renderProps.updateNotes(e.target.value), placeholder: "Special instructions for your order (e.g., gift wrap, delivery notes)", rows: 3, className: "w-full px-3 py-2 bg-surface-interactive border border-surface-interactive rounded-lg text-content-primary placeholder:text-content-muted focus:border-brand-primary focus:ring-1 focus:ring-brand-primary transition-colors duration-200 resize-vertical" })] }) }));
        } }));
});
/**
 * Headless component for line item added event subscription.
 * This component does not render any DOM elements - it only provides event subscription functionality.
 *
 * @component
 * @example
 * ```tsx
 * <Cart.LineItemAdded>
 *   {({ onAddedToCart }) => {
 *     useEffect(() => {
 *       return onAddedToCart((lineItems) => {
 *         console.log('Items added:', lineItems);
 *         setShowNotification(true);
 *       });
 *     }, [onAddedToCart]);
 *     return null;
 *   }}
 * </Cart.LineItemAdded>
 * ```
 */
export const LineItemAdded = (props) => {
    return _jsx(CoreLineItemAdded, { children: props.children });
};
/**
 * Order notes input field for customers to add special instructions or comments.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Cart.Note.Input placeholder="Add special instructions..." className="w-full px-3 py-2 border rounded-lg" />
 *
 * // Custom rendering with forwardRef
 * <Cart.Note.Input asChild>
 *   {React.forwardRef(({value, onChange, ...props}, ref) => (
 *     <textarea
 *       ref={ref}
 *       {...props}
 *       value={value}
 *       onChange={(e) => onChange(e.target.value)}
 *       className="w-full px-3 py-2 border rounded-lg resize-none focus:ring-2"
 *       rows={3}
 *     />
 *   ))}
 * </Cart.Note.Input>
 * ```
 */
const NoteInput = React.forwardRef(({ asChild, children, placeholder = 'Add special instructions...', maxLength = 500, ...props }, ref) => {
    return (_jsx(CoreNotes, { children: (renderProps) => {
            const { notes, updateNotes } = renderProps;
            return (_jsx(AsChildSlot, { asChild: asChild, customElement: children, customElementProps: renderProps, ref: ref, "data-testid": TestIds.cartNoteInput, ...props, children: _jsx("textarea", { value: notes || '', onChange: (e) => updateNotes(e.target.value), placeholder: placeholder, maxLength: maxLength, "data-testid": TestIds.cartNoteInput, ...props }) }));
        } }));
});
// ===== NESTED NAMESPACE EXPORTS =====
/**
 * Coupon-related components namespace
 *
 * @example
 * ```tsx
 * // Complete coupon management with new Root structure
 * <Cart.Coupon.Root>
 *   <div className="flex gap-2">
 *     <Cart.Coupon.Input
 *       placeholder="Enter coupon code"
 *       className="flex-1 px-3 py-2 border rounded"
 *     />
 *     <Cart.Coupon.Trigger className="px-4 py-2 bg-blue-500 text-white rounded">
 *       Apply
 *     </Cart.Coupon.Trigger>
 *   </div>
 *   <Cart.Coupon.Clear className="text-sm text-red-500 underline mt-2">
 *     Remove coupon
 *   </Cart.Coupon.Clear>
 * </Cart.Coupon.Root>
 *
 * // Raw access to all coupon functionality
 * <Cart.Coupon.Raw>
 *   {({ appliedCoupon, apply, remove, isLoading, error }) => (
 *     <div>
 *       {appliedCoupon ? (
 *         <span>Applied: {appliedCoupon}</span>
 *       ) : (
 *         <input onBlur={(e) => apply(e.target.value)} />
 *       )}
 *     </div>
 *   )}
 * </Cart.Coupon.Raw>
 * ```
 */
export const Coupon = {
    Root: CouponComponents.Root,
    Input: CouponComponents.Input,
    Trigger: CouponComponents.Trigger,
    Clear: CouponComponents.Clear,
    Raw: CouponComponents.Raw,
};
/**
 * Note-related components namespace
 */
export const Note = {
    Input: NoteInput,
};
/**
 * Factory function that creates price component variants for different cart pricing elements.
 * This internal factory generates components for subtotal, discount, shipping, tax, and total prices
 * with consistent behavior and styling patterns.
 *
 * @internal
 * @param config - Configuration object for the price component
 * @param config.priceProp - The property key from cart summary to display (e.g., 'subtotal', 'total')
 * @param config.shouldRender - Function to determine if the component should render based on cart state
 * @param config.isDiscount - Whether this represents a discount amount (will be displayed with negative sign)
 * @returns A React component that displays the specified price information
 */
const PricePartFactory = ({ priceProp, shouldRender = () => true, isDiscount = false, }) => React.forwardRef(({ asChild, children, className, label, ...props }, ref) => {
    return (_jsx(CoreSummary, { children: (renderProps) => {
            const { amountValues, currency, isTotalsLoading } = renderProps;
            const total = renderProps[priceProp];
            if (!total || !shouldRender(renderProps)) {
                return null;
            }
            const price = {
                amount: amountValues[priceProp],
                currency: currency,
            };
            return (_jsx(AsChildSlot, { asChild: asChild, customElement: children, customElementProps: {
                    price,
                    formattedPrice: total,
                    isLoading: isTotalsLoading,
                    label,
                }, ref: ref, className: className, ...props, children: _jsxs("div", { children: [label && _jsx("span", { children: label }), isTotalsLoading ? (_jsx("span", { children: "..." })) : (_jsx("span", { children: isDiscount ? `-${total}` : total }))] }) }));
        } }));
});
/**
 * Cart subtotal price component that displays the sum of all items before taxes, shipping, and discounts.
 * Supports the asChild pattern for flexible composition.
 *
 * @component
 * @example
 * ```tsx
 * // Default rendering
 * <Cart.Totals.Price label="Subtotal:" />
 *
 * // Custom rendering with asChild
 * <Cart.Totals.Price asChild>
 *   {({ price, formattedPrice, isLoading }, ref) => (
 *     <div ref={ref} className="flex justify-between">
 *       <span>Subtotal</span>
 *       <span>{isLoading ? '...' : formattedPrice}</span>
 *     </div>
 *   )}
 * </Cart.Totals.Price>
 * ```
 */
const Price = PricePartFactory({ priceProp: 'subtotal' });
/**
 * Cart discount amount component that displays the discount applied from coupons.
 * Only renders when a coupon is applied to the cart.
 *
 * @component
 * @example
 * ```tsx
 * // Default rendering (only shows when coupon applied)
 * <Cart.Totals.Discount label="Discount:" />
 *
 * // Custom rendering with asChild
 * <Cart.Totals.Discount asChild>
 *   {({ price, formattedPrice, isLoading }, ref) => (
 *     <div ref={ref} className="text-green-600">
 *       <span>Savings: </span>
 *       <span>{isLoading ? '...' : formattedPrice}</span>
 *     </div>
 *   )}
 * </Cart.Totals.Discount>
 * ```
 */
const Discount = PricePartFactory({
    priceProp: 'discount',
    shouldRender: (renderProps) => !!renderProps.appliedCoupon,
    isDiscount: true,
});
/**
 * Cart shipping cost component that displays the calculated shipping fee.
 * Not rendered when there is no shipping cost for the cart.
 *
 * @component
 * @example
 * ```tsx
 * // Default rendering
 * <Cart.Totals.Shipping label="Shipping:" />
 *
 * // Custom rendering with conditional display
 * <Cart.Totals.Shipping asChild>
 *   {({ price, formattedPrice, isLoading }, ref) => (
 *     <div ref={ref}>
 *       <span>Shipping: </span>
 *       <span>{price.amount === 0 ? 'FREE' : (isLoading ? '...' : formattedPrice)}</span>
 *     </div>
 *   )}
 * </Cart.Totals.Shipping>
 * ```
 */
const Shipping = PricePartFactory({ priceProp: 'shipping' });
/**
 * Cart tax amount component that displays the calculated tax for the order.
 *
 * @component
 * @example
 * ```tsx
 * // Default rendering
 * <Cart.Totals.Tax label="Tax:" />
 *
 * // Custom rendering with percentage display
 * <Cart.Totals.Tax asChild>
 *   {({ price, formattedPrice, isLoading }, ref) => (
 *     <div ref={ref} className="text-sm text-gray-600">
 *       <span>Tax: </span>
 *       <span>{isLoading ? '...' : formattedPrice}</span>
 *     </div>
 *   )}
 * </Cart.Totals.Tax>
 * ```
 */
const Tax = PricePartFactory({ priceProp: 'tax' });
/**
 * Cart total price component that displays the final amount to be charged.
 * This includes all items, taxes, shipping, and applies any discounts.
 *
 * @component
 * @example
 * ```tsx
 * // Default rendering
 * <Cart.Totals.Total label="Total:" />
 *
 * // Custom rendering with emphasis
 * <Cart.Totals.Total asChild>
 *   {({ price, formattedPrice, isLoading }, ref) => (
 *     <div ref={ref} className="font-bold text-lg border-t pt-2">
 *       <span>Total: </span>
 *       <span>{isLoading ? 'Calculating...' : formattedPrice}</span>
 *     </div>
 *   )}
 * </Cart.Totals.Total>
 * ```
 */
const Total = PricePartFactory({ priceProp: 'total' });
/**
 * Namespace containing all cart pricing/totals components.
 * These components provide a consistent way to display different aspects of cart pricing
 * including subtotal, discounts, shipping, tax, and final total.
 *
 * All components support the asChild pattern for flexible styling and composition.
 * They automatically handle loading states and conditional rendering based on cart state.
 *
 * @namespace
 * @example
 * ```tsx
 * // Basic cart totals display
 * function CartSummary() {
 *   return (
 *     <div className="cart-totals">
 *       <Cart.Totals.Price label="Subtotal:" />
 *       <Cart.Totals.Discount label="Discount:" />
 *       <Cart.Totals.Shipping label="Shipping:" />
 *       <Cart.Totals.Tax label="Tax:" />
 *       <hr />
 *       <Cart.Totals.Total label="Total:" />
 *     </div>
 *   );
 * }
 *
 * // Custom styled totals with asChild
 * function StyledCartTotals() {
 *   return (
 *     <div className="space-y-2">
 *       <Cart.Totals.Price asChild>
 *         {({ formattedPrice, isLoading }, ref) => (
 *           <div ref={ref} className="flex justify-between">
 *             <span>Items:</span>
 *             <span>{isLoading ? '...' : formattedPrice}</span>
 *           </div>
 *         )}
 *       </Cart.Totals.Price>
 *
 *       <Cart.Totals.Total asChild>
 *         {({ formattedPrice, isLoading }, ref) => (
 *           <div ref={ref} className="flex justify-between font-bold text-lg">
 *             <span>Total:</span>
 *             <span>{isLoading ? 'Calculating...' : formattedPrice}</span>
 *           </div>
 *         )}
 *       </Cart.Totals.Total>
 *     </div>
 *   );
 * }
 * ```
 */
export const Totals = {
    /** Cart subtotal before taxes, shipping, and discounts */
    Price,
    /** Discount amount from applied coupons (only shows when coupon applied) */
    Discount,
    /** Shipping cost for the order */
    Shipping,
    /** Tax amount for the order */
    Tax,
    /** Final total including all charges and discounts */
    Total,
};
export const Errors = React.forwardRef(({ asChild, children, className, ...props }, ref) => {
    return (_jsx(CoreCheckout, { children: (renderProps) => {
            if (!renderProps.error) {
                return null;
            }
            return (_jsx(AsChildSlot, { asChild: asChild, customElement: children, customElementProps: {
                    error: renderProps.error,
                }, ref: ref, className: className, "data-testid": TestIds.cartErrors, ...props, children: renderProps.error }));
        } }));
});
