// @wix/dashboard/dist/esm/hostPlatform.js
import { open } from '@wix/communication-channel';
import { getMonitoringClientFunction } from '@wix/monitoring-browser-sdk-host';
import { registerHeightReporter } from './helpers/registerHeightReporter';
import { isWixOrigin } from './helpers/isWixOrigin';
import { registerBeforeUnloadHandler } from './helpers/registerOnBeforeUnload';
import { getEssentialsFromQueryParams } from './helpers/essentials';
export const createHost = function (config) {
  if (config === void 0) {
    config = {};
  }
  // when full url is provided (worker case) use it and ignore own location search
  const searchString = config.fullUrl && new URL(config.fullUrl).search || typeof window !== 'undefined' && window.location.search || '';
  const queryParams = new URLSearchParams(searchString);
  const origin = config.origin || queryParams.get('origin') || '';
  const postMessage = config.postMessage || window.parent.postMessage.bind(window.parent);
  if (!isWixOrigin(origin)) {
    throw new Error(`Wix Dashboard SDK: Unable to establish a connection with the Wix dashboard. Please ensure that you are running your app within the Wix dashboard.`);
  }
  const {
    channel,
    close
  } = open({
    postMessage,
    origin
  });
  const closeMethods = [close];
  if (config.autoHeightUpdate === undefined || config.autoHeightUpdate) {
    const element = config.autoHeightElement || document.documentElement;
    const clearHeightReport = registerHeightReporter(channel, element);
    closeMethods.push(() => clearHeightReport());
  }
  const {
    onBeforeUnload,
    clearBeforeUnloadHandler
  } = registerBeforeUnloadHandler(channel);
  closeMethods.push(clearBeforeUnloadHandler);
  const siteInfo = parseSiteInfo(queryParams.get('siteInfo') || '');
  const getSiteInfoSync = () => siteInfo;
  const channelProxy = new Proxy(channel, {
    get(target, prop, receiver) {
      if (prop === 'onBeforeUnload') {
        return onBeforeUnload;
      }
      if (prop === 'getSiteInfo') {
        return getSiteInfoSync;
      }
      return Reflect.get(target, prop, receiver);
    }
  });
  const {
    monitoring,
    ...essentials
  } = getEssentialsFromQueryParams(queryParams) || {};
  const getMonitoringClient = getMonitoringClientFunction({
    hostContext: monitoring == null ? void 0 : monitoring.context,
    monitoringConfig: monitoring == null ? void 0 : monitoring.config,
    sentrySDK: typeof window !== 'undefined' ? window.Sentry : undefined
  });
  return {
    channel: channelProxy,
    close: () => closeMethods.forEach(method => method()),
    getMonitoringClient,
    essentials
  };
};
const parseSiteInfo = siteInfo => {
  try {
    return JSON.parse(decodeURIComponent(siteInfo));
  } catch (e) {
    console.warn('Failed to parse "siteInfo" query parameter. Make sure to run the code inside a supported Wix App extension.', e);
    return {};
  }
};
