// @wix/ricos/dist/cjs/_shared/link-modal/open-link-modal/attributes/attributes-desktop.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var attributes_desktop_exports = {};
__export(attributes_desktop_exports, {
  AttributesDesktop: () => AttributesDesktop
});
module.exports = __toCommonJS(attributes_desktop_exports);
var import_design_system = require("@wix/design-system");
var import_react = __toESM(require("react"));
var import_v1 = require("./../../../../ricos-schema/legacy/wix/rich_content/v1/index.js");
var import_data_hooks = require("../../data-hooks/data-hooks.js");
var import_use_attributes = require("./use-attributes.js");
const AttributesDesktop = ({
  link,
  onChange,
  t,
  tooltipProps
}) => {
  const { onToggleNewTab, onToggleNofollow, onToggleNoreferrer, onToggleSponsored } = (0, import_use_attributes.useAttributes)(
    link,
    onChange
  );
  return /* @__PURE__ */ import_react.default.createElement(import_design_system.Box, { direction: "vertical", gap: "SP3" }, /* @__PURE__ */ import_react.default.createElement(
    import_design_system.FormField,
    {
      label: t("LinkSettings_Settings_NewTab_Label"),
      labelPlacement: "left",
      labelWidth: "1fr"
    },
    /* @__PURE__ */ import_react.default.createElement(
      import_design_system.ToggleSwitch,
      {
        checked: link.target === import_v1.Link_Target.BLANK,
        dataHook: import_data_hooks.ATTRIBUTES_NEW_TAB,
        onChange: onToggleNewTab,
        size: "small"
      }
    )
  ), /* @__PURE__ */ import_react.default.createElement(
    import_design_system.FormField,
    {
      infoContent: t("LinkSettings_Settings_Noreferrer_Tooltip"),
      infoTooltipProps: tooltipProps,
      label: t("LinkSettings_Settings_Noreferrer_Label"),
      labelPlacement: "left",
      labelWidth: "1fr"
    },
    /* @__PURE__ */ import_react.default.createElement(
      import_design_system.ToggleSwitch,
      {
        checked: link.rel?.noreferrer,
        dataHook: import_data_hooks.ATTRIBUTES_NOREFERRER,
        onChange: onToggleNoreferrer,
        size: "small"
      }
    )
  ), /* @__PURE__ */ import_react.default.createElement(
    import_design_system.FormField,
    {
      infoContent: t("LinkSettings_Settings_Nofollow_Tooltip"),
      infoTooltipProps: tooltipProps,
      label: t("LinkSettings_Settings_Nofollow_Label"),
      labelPlacement: "left",
      labelWidth: "1fr"
    },
    /* @__PURE__ */ import_react.default.createElement(
      import_design_system.ToggleSwitch,
      {
        checked: link.rel?.nofollow,
        dataHook: import_data_hooks.ATTRIBUTES_NOFOLLOW,
        onChange: onToggleNofollow,
        size: "small"
      }
    )
  ), /* @__PURE__ */ import_react.default.createElement(
    import_design_system.FormField,
    {
      infoContent: t("LinkSettings_Settings_Sponsored_Tooltip"),
      infoTooltipProps: tooltipProps,
      label: t("LinkSettings_Settings_Sponsored_Label"),
      labelPlacement: "left",
      labelWidth: "1fr"
    },
    /* @__PURE__ */ import_react.default.createElement(
      import_design_system.ToggleSwitch,
      {
        checked: link.rel?.sponsored,
        dataHook: import_data_hooks.ATTRIBUTES_SPONSORED,
        onChange: onToggleSponsored,
        size: "small"
      }
    )
  ));
};
