// @wix/headless-blog/cjs/dist/react/Categories.js
import { jsx as _jsx, Fragment as _Fragment } from "react/jsx-runtime";
import { AsChildSlot } from '@wix/headless-utils/react';
import { createServicesMap } from '@wix/services-manager';
import { WixServices } from '@wix/services-manager-react';
import React from 'react';
import { BlogCategoriesService, BlogCategoriesServiceDefinition, enhanceCategories, } from '../services/blog-categories-service.js';
import * as CoreCategories from './core/Categories.js';
import { isActiveCategory, isValidChildren } from './helpers.js';
/** https://manage.wix.com/apps/14bcded7-0066-7c35-14d7-466cb3f09103/extensions/dynamic/wix-vibe-component?component-id=833bf7b1-2ad5-45f7-b2ac-89c207bcdc42 */
const HTML_CODE_TAG = 'blog.categories';
const CategoriesContext = React.createContext(null);
CategoriesContext.displayName = 'Blog.Categories.Context';
function useCategoriesContext() {
    const context = React.useContext(CategoriesContext);
    if (!context) {
        throw new Error('useCategoriesContext must be used within a BlogCategories.Root component');
    }
    return context;
}
/**
 * Root container for blog categories that provides categories context to all child components.
 * Supports both real categories from the service and custom categories.
 * Follows Container Level pattern from architecture rules.
 *
 * @component
 * @example
 * ```tsx
 * import { Blog } from '@wix/blog/components';
 *
 * function CategoryNavigation() {
 *   return (
 *     <Blog.Categories.Root customCategories={[createCustomCategory('All', '/')]}>
 *       <Blog.Categories.CategoryItems emptyState={<div>No categories found</div>}>
 *         <Blog.Categories.CategoryItemRepeater>
 *           <Blog.Category.Link baseUrl="/category/">
 *             <Blog.Category.Label />
 *           </Blog.Category.Link>
 *         </Blog.Categories.CategoryItemRepeater>
 *       </Blog.Categories.CategoryItems>
 *     </Blog.Categories.Root>
 *   );
 * }
 * ```
 */
export const Root = React.forwardRef((props, ref) => {
    const { asChild, children, className, categories: categoriesProp, customCategoriesToPrepend = [], blogCategoriesServiceConfig, fallbackImageUrl, } = props;
    const renderRoot = (categories, hasCategories) => {
        const contextValue = {
            categories,
            fallbackImageUrl,
            hasCategories,
        };
        const attributes = {
            'data-component-tag': asChild ? undefined : HTML_CODE_TAG,
            'data-testid': "blog-categories-root" /* TestIds.blogCategoriesRoot */,
            'data-has-categories': hasCategories,
        };
        return (_jsx(CategoriesContext.Provider, { value: contextValue, children: _jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, ...attributes, customElement: children, customElementProps: { hasCategories }, children: _jsx("div", { children: isValidChildren(children) ? children : null }) }) }));
    };
    if (categoriesProp) {
        return renderRoot(enhanceCategories(categoriesProp), categoriesProp.length > 0);
    }
    return (_jsx(WixServices, { servicesMap: createServicesMap().addService(BlogCategoriesServiceDefinition, BlogCategoriesService, blogCategoriesServiceConfig), children: _jsx(CoreCategories.Categories, { children: ({ categories: realCategories, hasCategories: hasRealCategories }) => {
                const allCategories = [
                    ...customCategoriesToPrepend,
                    ...realCategories,
                ];
                const hasCategories = hasRealCategories;
                return renderRoot(allCategories, hasCategories);
            } }) }));
});
Root.displayName = 'Blog.Categories.Root';
/**
 * Container for the categories list with empty state support.
 * Follows List Container Level pattern from architecture rules.
 *
 * @component
 * @example
 * ```tsx
 * <Blog.Categories.CategoryItems emptyState={<div>No categories found</div>}>
 *   <Blog.Categories.CategoryItemRepeater>
 *     <Blog.Category.Label />
 *   </Blog.Categories.CategoryItemRepeater>
 * </Blog.Categories.CategoryItems>
 * ```
 */
export const CategoryItems = React.forwardRef((props, ref) => {
    const { children, emptyState } = props;
    const { hasCategories } = useCategoriesContext();
    if (!hasCategories) {
        return emptyState || null;
    }
    const attributes = {
        'data-testid': "blog-categories" /* TestIds.blogCategories */,
    };
    return (_jsx("div", { ...attributes, ref: ref, children: children }));
});
CategoryItems.displayName = 'Blog.Categories.CategoryItems';
/**
 * Repeater component that renders Category for each category.
 * Follows Repeater Level pattern from architecture rules.
 * Note: Repeater components do NOT support asChild as per architecture rules.
 *
 * @component
 * @example
 * ```tsx
 * <Blog.Categories.CategoryItems>
 *   <Blog.Categories.CategoryItemRepeater>
 *     <Blog.Categories.Link baseUrl="/category/">
 *       <Blog.Categories.Label />
 *     </Blog.Categories.Link>
 *   </Blog.Categories.CategoryItemRepeater>
 * </Blog.Categories.CategoryItems>
 * ```
 */
export const CategoryItemRepeater = React.forwardRef((props, _ref) => {
    const { children, offset = 0, limit = Infinity } = props;
    const { categories, fallbackImageUrl } = useCategoriesContext();
    const categoriesSlice = categories.slice(offset, offset + limit);
    return (_jsx(_Fragment, { children: categoriesSlice.map((category) => {
            const imageUrl = category.imageUrl || fallbackImageUrl;
            const contextValue = {
                category,
                imageUrl,
            };
            return (_jsx(CategoryItemRepeaterContext.Provider, { value: contextValue, children: children }, category._id));
        }) }));
});
CategoryItemRepeater.displayName = 'Blog.Categories.CategoryItemRepeater';
const CategoryItemRepeaterContext = React.createContext(null);
CategoryItemRepeaterContext.displayName = 'Blog.Categories.CategoryContext';
export function useCategoryItemRepeaterContext() {
    const context = React.useContext(CategoryItemRepeaterContext);
    if (!context) {
        throw new Error('useCategoryContext must be used within a Category.Root component');
    }
    return context;
}
/**
 * Component that finds and displays the currently active category.
 * Provides the active category through CategoryContext to its children.
 *
 * @component
 * @example
 * ```tsx
 * <Blog.Categories.ActiveCategory pathname={location.pathname} baseUrl="/category/">
 *   {({ category }) => (
 *     <div className="hero-section">
 *       <Blog.Categories.Image />
 *       <div className="overlay">
 *         <Blog.Categories.Label className="text-white text-3xl" />
 *         <Blog.Categories.Description className="text-white/90" />
 *       </div>
 *     </div>
 *   )}
 * </Blog.Categories.ActiveCategory>
 * ```
 */
export const ActiveCategory = React.forwardRef((props, ref) => {
    const { asChild, className, pathname, baseUrl = '', children } = props;
    const { categories, fallbackImageUrl } = useCategoriesContext();
    const activeCategory = categories.find((category) => isActiveCategory(pathname, baseUrl, category));
    if (!activeCategory)
        return null;
    const contextValue = {
        category: activeCategory,
        imageUrl: activeCategory.imageUrl || fallbackImageUrl,
    };
    const attributes = {
        'data-testid': "blog-active-category" /* TestIds.blogActiveCategory */,
        'data-has-image': !!contextValue.imageUrl,
    };
    return (_jsx(CategoryItemRepeaterContext.Provider, { value: contextValue, children: _jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, ...attributes, customElement: children, customElementProps: { category: activeCategory }, children: _jsx("div", { children: isValidChildren(children) ? children : null }) }) }));
});
ActiveCategory.displayName = 'Blog.Categories.ActiveCategory';
