// @wix/headless-blog/cjs/dist/react/Comments.js
import { jsx as _jsx } from "react/jsx-runtime";
import { Sort as SortPrimitive } from '@wix/headless-components/react';
import { AsChildSlot } from '@wix/headless-utils/react';
import { createServicesMap } from '@wix/services-manager';
import { WixServices, useService } from '@wix/services-manager-react';
import React from 'react';
import { CommentsService, CommentsServiceDefinition, } from '../services/comments-service.js';
import * as Comment from './Comment.js';
import * as CoreComments from './core/Comments.js';
import { isValidChildren, useIntersectionObserver } from './helpers.js';
const CurrentMemberContext = React.createContext(undefined);
const useCurrentMemberId = () => {
    const currentMember = React.useContext(CurrentMemberContext);
    const currentMemberId = typeof currentMember === 'object' &&
        currentMember !== null &&
        '_id' in currentMember &&
        typeof currentMember._id === 'string'
        ? currentMember._id
        : undefined;
    return currentMemberId;
};
/**
 * Root container for comments that provides comments context to all child components.
 * Uses IntersectionObserver for lazy loading - comments are loaded when the container becomes visible.
 *
 * @component
 * @example
 * ```tsx
 * import { Blog, Comment } from '@wix/blog/components';
 *
 * function CommentsSection() {
 *   return (
 *     <Comments.Root commentsServiceConfig={{ contextId, resourceId }}>
 *       <Comments.CommentItemRepeater>
 *         <Comment.Author />
 *         <Comment.Content />
 *       </Comments.CommentItemRepeater>
 *     </Comments.Root>
 *   );
 * }
 * ```
 */
export const Root = React.forwardRef((props, forwardedRef) => {
    const { asChild, children, className, currentMember, commentsServiceConfig } = props;
    const { ref, isVisible } = useIntersectionObserver(forwardedRef);
    return (_jsx(WixServices
    // key: Ensure we re-render the component when the comments config changes
    , { servicesMap: createServicesMap().addService(CommentsServiceDefinition, CommentsService, commentsServiceConfig), children: _jsx(CurrentMemberContext.Provider, { value: currentMember, children: _jsx(CoreComments.Comments, { children: ({ initialLoad, isLoading }) => {
                    // Trigger initial load when component becomes visible
                    React.useEffect(() => {
                        if (isVisible) {
                            initialLoad();
                        }
                    }, [isVisible, initialLoad]);
                    const attributes = {
                        'data-testid': "comments-root" /* TestIds.root */,
                        'data-visible': isVisible,
                        'data-loading': isLoading() === 'initial',
                    };
                    return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, ...attributes, customElement: children, children: _jsx("div", { children: isValidChildren(children) ? children : null }) }));
                } }) }) }, `${commentsServiceConfig.contextId}-${commentsServiceConfig.resourceId}`));
});
Root.displayName = 'Comments.Root';
/**
 * Container for the comments list with empty state support.
 * Follows List Container Level pattern from architecture rules.
 *
 * @component
 * @example
 * ```tsx
 * <Comments.CommentItems emptyState={<div>No comments yet</div>}>
 *   <Comments.CommentItemRepeater>
 *     <Comment.Author />
 *     <Comment.Content />
 *   </Comments.CommentItemRepeater>
 * </Comments.CommentItems>
 * ```
 */
export const CommentItems = React.forwardRef((props, ref) => {
    const { children, emptyState, loadingState, className } = props;
    const service = useService(CommentsServiceDefinition);
    const comments = service.getComments();
    const isLoading = service.isLoading() === 'initial';
    if (isLoading && loadingState) {
        return loadingState;
    }
    if (comments.length === 0) {
        return emptyState || null;
    }
    const attributes = {
        'data-testid': "comments-items" /* TestIds.items */,
    };
    return (_jsx("div", { ...attributes, ref: ref, className: className, children: children }));
});
CommentItems.displayName = 'Comments.CommentItems';
/**
 * Sort component for comments that provides sorting functionality.
 *
 * @component
 * @example
 * ```tsx
 * // Default select dropdown
 * <Comments.Sort />
 *
 * // As list of clickable options
 * <Comments.Sort as="list" />
 *
 * // With custom styling
 * <Comments.Sort
 *   as="select"
 *   className="custom-sort-select"
 * />
 * ```
 */
export const Sort = React.forwardRef(({ children, className, as, asChild }, ref) => {
    return (_jsx(CoreComments.Sort, { children: ({ currentSort, sortOptions, setSort }) => {
            if (asChild && children) {
                return children({ currentSort, sortOptions, setSort });
            }
            return (_jsx(SortPrimitive.Root, { ref: ref, value: currentSort, onChange: (value) => setSort(value), sortOptions: sortOptions, as: as, className: className, "data-testid": "comments-sort" /* TestIds.sort */, children: sortOptions.map((option) => {
                    if ('fieldName' in option) {
                        return (_jsx(SortPrimitive.Option, { fieldName: option.fieldName, order: 'order' in option ? option.order : 'ASC', label: option.label }, option.label));
                    }
                    return null;
                }) }));
        } }));
});
Sort.displayName = 'Comments.Sort';
/**
 * Repeater component that creates individual comment contexts for each comment.
 * Follows Repeater Level pattern from architecture rules.
 * Note: Repeater components do NOT support asChild as per architecture rules.
 *
 * @component
 * @example
 * ```tsx
 * <Comments.CommentItemRepeater>
 *   <Comment.Author />
 *   <Comment.Content />
 *   <Comment.CommentDate />
 * </Comments.CommentItemRepeater>
 * ```
 */
export const CommentItemRepeater = React.forwardRef((props, _ref) => {
    const { children } = props;
    const service = useService(CommentsServiceDefinition);
    const comments = service.getComments();
    const currentMemberId = useCurrentMemberId();
    if (comments.length === 0)
        return null;
    return comments.map((comment) => {
        return (_jsx(CoreComments.TopLevelCommentRoot, { comment: comment, children: _jsx(Comment.Root, { comment: comment, asChild: true, currentMemberId: currentMemberId, children: children }) }, comment._id));
    });
});
CommentItemRepeater.displayName = 'Comments.CommentItemRepeater';
/**
 * Load more trigger component for pagination.
 *
 * @component
 * @example
 * ```tsx
 * <Comments.LoadMore asChild>
 *   {({ hasNextPage, isLoading, loadNextPage }) => (
 *     <button
 *       onClick={loadNextPage}
 *       disabled={!hasNextPage || isLoading}
 *     >
 *       {isLoading ? 'Loading...' : 'Load More Comments'}
 *     </button>
 *   )}
 * </Comments.LoadMore>
 * ```
 */
export const LoadMore = React.forwardRef((props, ref) => {
    const { asChild, children, className, loadingState } = props;
    return (_jsx(CoreComments.Comments, { children: ({ hasNextPage: getHasNextPage, isLoading: getIsLoading, loadMore }) => {
            const hasNextPage = getHasNextPage();
            if (!hasNextPage)
                return null;
            const isLoading = getIsLoading() === 'more';
            const handleClick = () => {
                if (isLoading)
                    return;
                loadMore();
            };
            const dataAttributes = {
                'data-testid': "comments-load-more" /* TestIds.loadMore */,
                'data-loading': isLoading,
                'data-has-next-page': hasNextPage,
            };
            const buttonAttributes = {
                onClick: handleClick,
                type: 'button',
            };
            return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, ...dataAttributes, ...buttonAttributes, customElement: children, customElementProps: { isLoading, loadNextPage: handleClick }, content: isLoading && loadingState ? loadingState : undefined, children: _jsx("button", { children: isValidChildren(children) ? children : null }) }));
        } }));
});
LoadMore.displayName = 'Comments.LoadMore';
