// @wix/headless-events/dist/react/Event.js
import { jsx as _jsx } from "react/jsx-runtime";
import { Form as FormPrimitive } from '@wix/headless-forms/react';
import { WixMediaImage } from '@wix/headless-media/react';
import { AsChildSlot } from '@wix/headless-utils/react';
import React from 'react';
import { hasDescription } from '../utils/event.js';
import * as CoreEvent from './core/Event.js';
import * as CoreEventList from './core/EventList.js';
import * as CoreOccurrenceList from './core/OccurrenceList.js';
var TestIds;
(function (TestIds) {
    TestIds["eventRoot"] = "event-root";
    TestIds["eventSlug"] = "event-slug";
    TestIds["eventImage"] = "event-image";
    TestIds["eventTitle"] = "event-title";
    TestIds["eventDate"] = "event-date";
    TestIds["eventLocation"] = "event-location";
    TestIds["eventShortDescription"] = "event-short-description";
    TestIds["eventDescription"] = "event-description";
    TestIds["eventRsvpButton"] = "event-rsvp-button";
    TestIds["eventFacebookShare"] = "event-facebook-share";
    TestIds["eventLinkedInShare"] = "event-linked-in-share";
    TestIds["eventXShare"] = "event-x-share";
    TestIds["eventAddToGoogleCalendar"] = "event-add-to-google-calendar";
    TestIds["eventAddToIcsCalendar"] = "event-add-to-ics-calendar";
    TestIds["eventOccurrences"] = "event-occurrences";
    TestIds["eventOtherEvents"] = "event-other-events";
    TestIds["eventForm"] = "event-form";
})(TestIds || (TestIds = {}));
const DATA_COMPONENT_TAG = 'events.event';
/**
 * Root container that provides event context to all child components.
 * Must be used as the top-level Event component.
 *
 * @order 1
 * @component
 * @example
 * ```tsx
 * import { Event } from '@wix/events/components';
 *
 * function EventPage({ event }) {
 *   return (
 *     <Event.Root event={event}>
 *       <Event.Image />
 *       <Event.Title />
 *       <Event.Date />
 *       <Event.Location />
 *       <Event.Description />
 *       <Event.RsvpButton label="RSVP" />
 *     </Event.Root>
 *   );
 * }
 * ```
 */
export const Root = React.forwardRef((props, ref) => {
    const { event, eventListServiceConfig, occurrenceListServiceConfig, asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreEvent.Root, { event: event, children: _jsx(CoreEventList.Root, { eventListServiceConfig: eventListServiceConfig ?? {
                events: [],
                categories: [],
                pageSize: 0,
                currentPage: 0,
                totalPages: 0,
            }, children: _jsx(CoreOccurrenceList.Root, { occurrenceListServiceConfig: occurrenceListServiceConfig ?? {
                    recurringCategoryId: undefined,
                    occurrences: [],
                    pageSize: 0,
                    currentPage: 0,
                    totalPages: 0,
                }, children: _jsx(RootContent, { ref: ref, asChild: asChild, className: className, ...otherProps, children: children }) }) }) }));
});
/**
 * Internal Event RootContent component.
 *
 * @component
 * @internal
 */
const RootContent = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreEvent.Raw, { children: ({ event }) => (_jsx(CoreOccurrenceList.Occurrences, { children: ({ hasOccurrences }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-component-tag": DATA_COMPONENT_TAG, "data-testid": TestIds.eventRoot, "data-upcoming": event.status === 'UPCOMING', "data-started": event.status === 'STARTED', "data-ended": event.status === 'ENDED', "data-sold-out": !!event.registration?.tickets?.soldOut, "data-registration-closed": event.registration?.status === 'CLOSED_MANUALLY' ||
                    event.registration?.status === 'CLOSED_AUTOMATICALLY', "data-has-image": !!event.mainImage, "data-has-description": hasDescription(event), "data-has-occurrences": hasOccurrences, customElement: children, customElementProps: {}, ...otherProps, children: _jsx("div", { children: children }) })) })) }));
});
/**
 * Displays the event slug.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Event.Slug className="text-sm font-medium" />
 *
 * // asChild with primitive
 * <Event.Slug asChild>
 *   <span className="text-sm font-medium" />
 * </Event.Slug>
 *
 * // asChild with react component
 * <Event.Slug>
 *   {React.forwardRef(({ slug, ...props }, ref) => (
 *     <a ref={ref} {...props} href={`/events/${slug}`}>
 *       Event Details
 *     </a>
 *   ))}
 * </Event.Slug>
 * ```
 */
export const Slug = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreEvent.Slug, { children: ({ slug }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.eventSlug, customElement: children, customElementProps: { slug }, content: slug, ...otherProps, children: _jsx("span", { children: slug }) })) }));
});
/**
 * Provides event type information.
 *
 * @component
 * @example
 * ```tsx
 * // asChild with react component
 * <Event.Type asChild>
 *   {React.forwardRef(({ ticketed, rsvp, external, ...props }, ref) => (
 *     <span ref={ref} {...props}>
 *       {ticketed ? 'Ticketed' : rsvp ? 'RSVP' : external ? 'External' : ''}
 *     </span>
 *   ))}
 * </Event.Type>
 * ```
 */
export const Type = React.forwardRef((props, ref) => {
    const { children, ...otherProps } = props;
    return (_jsx(CoreEvent.Type, { children: ({ ticketed, rsvp, external }) => (_jsx(AsChildSlot, { asChild: true, ref: ref, customElement: children, customElementProps: { ticketed, rsvp, external }, ...otherProps })) }));
});
/**
 * Displays the event image using WixMediaImage component.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Event.Image className="w-full h-full object-cover" />
 *
 * // asChild with primitive
 * <Event.Image asChild>
 *   <img className="w-full h-full object-cover" />
 * </Event.Image>
 *
 * // asChild with react component
 * <Event.Image asChild>
 *   {React.forwardRef(({ src, alt, width, height, ...props }, ref) => (
 *     <img ref={ref} src={src} alt={alt} width={width} height={height} {...props} />
 *   ))}
 * </Event.Image>
 * ```
 */
export const Image = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreEvent.Image, { children: ({ image }) => (_jsx(WixMediaImage, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.eventImage, media: { image }, children: children, ...otherProps })) }));
});
/**
 * Displays the event title.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Event.Title className="text-4xl font-bold" />
 *
 * // asChild with primitive
 * <Event.Title asChild>
 *   <h1 className="text-4xl font-bold" />
 * </Event.Title>
 *
 * // asChild with react component
 * <Event.Title asChild>
 *   {React.forwardRef(({ title, ...props }, ref) => (
 *     <h1 ref={ref} {...props} className="text-4xl font-bold">
 *       {title}
 *     </h1>
 *   ))}
 * </Event.Title>
 * ```
 */
export const Title = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreEvent.Title, { children: ({ title }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.eventTitle, customElement: children, customElementProps: { title }, content: title, ...otherProps, children: _jsx("span", { children: title }) })) }));
});
/**
 * Displays the event date.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Event.Date className="text-sm font-medium" format="full" />
 *
 * // asChild with primitive
 * <Event.Date asChild format="full">
 *   <span className="text-sm font-medium" />
 * </Event.Date>
 *
 * // asChild with react component
 * <Event.Date asChild>
 *   {React.forwardRef(({ startDate, endDate, timeZoneId, dateAndTimeTbd, dateAndTimeTbdMessage, hideEndDate, showTimeZone, formattedDate, ...props }, ref) => (
 *     <span ref={ref} {...props} className="text-sm font-medium">
 *       {formattedDate}
 *     </span>
 *   ))}
 * </Event.Date>
 * ```
 */
export const Date = React.forwardRef((props, ref) => {
    const { asChild, children, className, format = 'short', locale, ...otherProps } = props;
    return (_jsx(CoreEvent.DateCore, { format: format, locale: locale, children: ({ startDate, endDate, timeZoneId, dateAndTimeTbd, dateAndTimeTbdMessage, hideEndDate, showTimeZone, formattedDate, }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.eventDate, customElement: children, customElementProps: {
                startDate,
                endDate,
                timeZoneId,
                dateAndTimeTbd,
                dateAndTimeTbdMessage,
                hideEndDate,
                showTimeZone,
                formattedDate,
            }, content: formattedDate, ...otherProps, children: _jsx("span", { children: formattedDate }) })) }));
});
/**
 * Displays the event location.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Event.Location className="text-sm font-medium" format="full" />
 *
 * // asChild with primitive
 * <Event.Location asChild format="full">
 *   <span className="text-sm font-medium" />
 * </Event.Location>
 *
 * // asChild with react component
 * <Event.Location asChild>
 *   {React.forwardRef(({ formattedLocation, latitude, longitude, ...props }, ref) => (
 *     <span ref={ref} {...props} className="text-sm font-medium">
 *       {formattedLocation}
 *     </span>
 *   ))}
 * </Event.Location>
 * ```
 */
export const Location = React.forwardRef((props, ref) => {
    const { asChild, children, className, format = 'short', ...otherProps } = props;
    return (_jsx(CoreEvent.Location, { format: format, children: ({ formattedLocation, latitude, longitude }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.eventLocation, customElement: children, customElementProps: { formattedLocation, latitude, longitude }, content: formattedLocation, ...otherProps, children: _jsx("span", { children: formattedLocation }) })) }));
});
/**
 * Displays the event short description.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Event.ShortDescription className="text-sm font-medium" />
 *
 * // asChild with primitive
 * <Event.ShortDescription asChild>
 *   <span className="text-sm font-medium" />
 * </Event.ShortDescription>
 *
 * // asChild with react component
 * <Event.ShortDescription asChild>
 *   {React.forwardRef(({ shortDescription, ...props }, ref) => (
 *     <span ref={ref} {...props} className="text-sm font-medium">
 *       {shortDescription}
 *     </span>
 *   ))}
 * </Event.ShortDescription>
 * ```
 */
export const ShortDescription = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreEvent.ShortDescription, { children: ({ shortDescription }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.eventShortDescription, customElement: children, customElementProps: { shortDescription }, content: shortDescription, ...otherProps, children: _jsx("span", { children: shortDescription }) })) }));
});
/**
 * Provides the event description. RicosViewer should be used to render the description.
 *
 * @component
 * @example
 * ```tsx
 * <Event.Description>
 *   {React.forwardRef(({ description, ...props }, ref) => (
 *     <RicosViewer ref={ref} content={description} />
 *   ))}
 * </Event.Description>
 * ```
 */
export const Description = React.forwardRef((props, ref) => {
    const { children, ...otherProps } = props;
    return (_jsx(CoreEvent.Description, { children: ({ description }) => (_jsx(AsChildSlot, { asChild: true, ref: ref, "data-testid": TestIds.eventDescription, customElement: children, customElementProps: { description }, ...otherProps })) }));
});
/**
 * Displays the event RSVP button.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Event.RsvpButton className="w-full" label="RSVP" />
 *
 * // asChild with primitive
 * <Event.RsvpButton asChild>
 *   <button className="w-full">RSVP</button>
 * </Event.RsvpButton>
 *
 * // asChild with react component
 * <Event.RsvpButton asChild>
 *   {React.forwardRef(({ slug, ticketed, ...props }, ref) => (
 *     <button ref={ref} {...props}>
 *       {ticketed ? 'Buy Tickets' : 'RSVP'}
 *     </button>
 *   ))}
 * </Event.RsvpButton>
 * ```
 */
export const RsvpButton = React.forwardRef((props, ref) => {
    const { asChild, children, className, label, ...otherProps } = props;
    return (_jsx(CoreEvent.RsvpButton, { children: ({ slug, ticketed }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.eventRsvpButton, "data-ticketed": ticketed, customElement: children, customElementProps: { slug, ticketed }, ...otherProps, children: _jsx("button", { children: label }) })) }));
});
/**
 * Displays Facebook share element.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Event.FacebookShare eventPageUrl={eventPageUrl} />
 *
 * // asChild with primitive
 * <Event.FacebookShare asChild eventPageUrl={eventPageUrl}>
 *   <a />
 * </Event.FacebookShare>
 *
 * // asChild with react component
 * <Event.FacebookShare asChild eventPageUrl={eventPageUrl}>
 *   {React.forwardRef(({ shareUrl, ...props }, ref) => (
 *     <button ref={ref} onClick={() => window.open(shareUrl, '_blank')} />
 *   ))}
 * </Event.FacebookShare>
 * ```
 */
export const FacebookShare = React.forwardRef((props, ref) => {
    const { eventPageUrl, asChild, children, className, ...otherProps } = props;
    const shareUrl = `https://www.facebook.com/sharer/sharer.php?u=${encodeURIComponent(eventPageUrl)}`;
    return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.eventFacebookShare, customElement: children, customElementProps: { shareUrl }, href: shareUrl, target: "_blank", rel: "noreferrer", ...otherProps, children: _jsx("a", {}) }));
});
/**
 * Displays LinkedIn share element.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Event.LinkedInShare eventPageUrl={eventPageUrl} />
 *
 * // asChild with primitive
 * <Event.LinkedInShare asChild eventPageUrl={eventPageUrl}>
 *   <a />
 * </Event.LinkedInShare>
 *
 * // asChild with react component
 * <Event.LinkedInShare asChild>
 *   {React.forwardRef(({ shareUrl, ...props }, ref) => (
 *     <button ref={ref} onClick={() => window.open(shareUrl, '_blank')} />
 *   ))}
 * </Event.LinkedInShare>
 * ```
 */
export const LinkedInShare = React.forwardRef((props, ref) => {
    const { eventPageUrl, asChild, children, className, ...otherProps } = props;
    const shareUrl = `https://www.linkedin.com/shareArticle?mini=true&url=${encodeURIComponent(eventPageUrl)}`;
    return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.eventLinkedInShare, customElement: children, customElementProps: { shareUrl }, href: shareUrl, target: "_blank", rel: "noreferrer", ...otherProps, children: _jsx("a", {}) }));
});
/**
 * Displays X share element.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Event.XShare eventPageUrl={eventPageUrl} />
 *
 * // asChild with primitive
 * <Event.XShare asChild eventPageUrl={eventPageUrl}>
 *   <a />
 * </Event.XShare>
 *
 * // asChild with react component
 * <Event.XShare asChild>
 *   {React.forwardRef(({ shareUrl, ...props }, ref) => (
 *     <button ref={ref} onClick={() => window.open(shareUrl, '_blank')} />
 *   ))}
 * </Event.XShare>
 * ```
 */
export const XShare = React.forwardRef((props, ref) => {
    const { eventPageUrl, asChild, children, className, ...otherProps } = props;
    const shareUrl = `https://x.com/intent/post?url=${encodeURIComponent(eventPageUrl)}`;
    return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.eventXShare, customElement: children, customElementProps: { shareUrl }, href: shareUrl, target: "_blank", rel: "noreferrer", ...otherProps, children: _jsx("a", {}) }));
});
/**
 * Displays link to add the event to Google calendar.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Event.AddToGoogleCalendar />
 *
 * // asChild with primitive
 * <Event.AddToGoogleCalendar asChild>
 *   <a />
 * </Event.AddToGoogleCalendar>
 *
 * // asChild with react component
 * <Event.AddToGoogleCalendar asChild>
 *   {React.forwardRef(({ url, ...props }, ref) => (
 *     <button ref={ref} onClick={() => window.open(url, '_blank')} />
 *   ))}
 * </Event.AddToGoogleCalendar>
 * ```
 */
export const AddToGoogleCalendar = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreEvent.AddToGoogleCalendar, { children: ({ url }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.eventAddToGoogleCalendar, customElement: children, customElementProps: { url }, href: url, target: "_blank", rel: "noreferrer", ...otherProps, children: _jsx("a", {}) })) }));
});
/**
 * Displays link to add the event to ICS calendar.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Event.AddToIcsCalendar />
 *
 * // asChild with primitive
 * <Event.AddToIcsCalendar asChild>
 *   <a />
 * </Event.AddToIcsCalendar>
 *
 * // asChild with react component
 * <Event.AddToIcsCalendar asChild>
 *   {React.forwardRef(({ url, ...props }, ref) => (
 *     <button ref={ref} onClick={() => window.open(url, '_blank')} />
 *   ))}
 * </Event.AddToIcsCalendar>
 * ```
 */
export const AddToIcsCalendar = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreEvent.AddToIcsCalendar, { children: ({ url }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.eventAddToIcsCalendar, customElement: children, customElementProps: { url }, href: url, target: "_blank", rel: "noreferrer", ...otherProps, children: _jsx("a", {}) })) }));
});
/**
 * Container for other events.
 * Follows List Container Level pattern.
 *
 * @component
 * @example
 * ```tsx
 * <Event.OtherEvents count={5}>
 *   <Event.OtherEventRepeater>
 *     <Event.Image />
 *     <Event.Title />
 *   </Event.OtherEventRepeater>
 * </Event.OtherEvents>
 * ```
 */
export const OtherEvents = React.forwardRef((props, ref) => {
    const { count = 3, asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreEvent.OtherEvents, { count: count, children: ({ events }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.eventOtherEvents, customElement: children, customElementProps: { events }, ...otherProps, children: _jsx("div", { children: children }) })) }));
});
/**
 * Repeater component that renders Event.Root for each event.
 * Follows Repeater Level pattern.
 * Note: Repeater components do NOT support asChild as per architecture rules.
 *
 * @component
 * @example
 * ```tsx
 * <Event.OtherEventRepeater>
 *   <Event.Image />
 *   <Event.Title />
 * </Event.OtherEventRepeater>
 * ```
 */
export const OtherEventRepeater = (props) => {
    const { count = 3, children, className } = props;
    return (_jsx(CoreEvent.OtherEvents, { count: count, children: ({ events }) => events.map((event) => (_jsx(Root, { event: event, className: className, children: children }, event._id))) }));
};
/**
 * Displays the event form.
 *
 * @component
 * @example
 * ```tsx
 * import { Form } from '@wix/headless-forms/react';
 *
 * <Event.Form>
 *   <Form.Loading />
 *   <Form.LoadingError />
 *   <Form.Fields fieldMap={FIELD_MAP} />
 * </Event.Form>
 * ```
 */
export const Form = React.forwardRef((props, ref) => {
    const { asChild, children, className, thankYouPageUrl, ...otherProps } = props;
    return (_jsx(CoreEvent.Form, { thankYouPageUrl: thankYouPageUrl, children: ({ formId, onSubmit }) => (_jsx(FormPrimitive.Root, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.eventForm, formServiceConfig: { formId, onSubmit }, ...otherProps, children: children })) }));
});
