// @wix/headless-events/dist/react/InvoiceItem.js
import { jsx as _jsx } from "react/jsx-runtime";
import { AsChildSlot } from '@wix/headless-utils/react';
import React from 'react';
import * as CoreInvoiceItem from './core/InvoiceItem.js';
var TestIds;
(function (TestIds) {
    TestIds["invoiceItemRoot"] = "invoice-item-root";
    TestIds["invoiceItemName"] = "invoice-item-name";
    TestIds["invoiceItemPrice"] = "invoice-item-price";
    TestIds["invoiceItemQuantity"] = "invoice-item-quantity";
    TestIds["invoiceItemTotal"] = "invoice-item-total";
})(TestIds || (TestIds = {}));
/**
 * Root container that provides invoice item service context to all child components.
 * Must be used as the top-level InvoiceItem component.
 *
 * @order 1
 * @component
 * @example
 * ```tsx
 * import { InvoiceItem } from '@wix/events/components';
 *
 * function InvoiceItemComponent({ invoiceItem }) {
 *   return (
 *     <InvoiceItem.Root invoiceItem={invoiceItem}>
 *       <InvoiceItem.Name />
 *       <InvoiceItem.Price />
 *       <InvoiceItem.Quantity />
 *       <InvoiceItem.Total />
 *     </InvoiceItem.Root>
 *   );
 * }
 * ```
 */
export const Root = React.forwardRef((props, ref) => {
    const { invoiceItem, asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreInvoiceItem.Root, { invoiceItem: invoiceItem, children: _jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.invoiceItemRoot, customElement: children, customElementProps: {}, ...otherProps, children: _jsx("div", { children: children }) }) }));
});
/**
 * Displays the invoice item name.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <InvoiceItem.Name className="font-medium" />
 *
 * // asChild with primitive
 * <InvoiceItem.Name asChild>
 *   <h3 className="text-lg font-bold" />
 * </InvoiceItem.Name>
 *
 * // asChild with React component
 * <InvoiceItem.Name asChild>
 *   {React.forwardRef(({ name, ...props }, ref) => (
 *     <h3 ref={ref} {...props} className="text-lg font-bold">
 *       {name}
 *     </h3>
 *   ))}
 * </InvoiceItem.Name>
 * ```
 */
export const Name = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreInvoiceItem.Name, { children: ({ name }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.invoiceItemName, customElement: children, customElementProps: { name }, content: name, ...otherProps, children: _jsx("span", { children: name }) })) }));
});
/**
 * Displays the invoice item price.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <InvoiceItem.Price className="text-green-600 font-semibold" />
 *
 * // asChild with primitive
 * <InvoiceItem.Price asChild>
 *   <p className="text-lg text-green-600" />
 * </InvoiceItem.Price>
 *
 * // asChild with React component
 * <InvoiceItem.Price asChild>
 *   {React.forwardRef(({ value, currency, formattedValue, ...props }, ref) => (
 *     <p ref={ref} {...props} className="text-lg text-green-600">
 *       Price: {formattedValue}
 *     </p>
 *   ))}
 * </InvoiceItem.Price>
 * ```
 */
export const Price = React.forwardRef((props, ref) => {
    const { asChild, children, className, locale, ...otherProps } = props;
    return (_jsx(CoreInvoiceItem.Price, { locale: locale, children: ({ value, currency, formattedValue }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.invoiceItemPrice, customElement: children, customElementProps: { value, currency, formattedValue }, content: formattedValue, ...otherProps, children: _jsx("span", { children: formattedValue }) })) }));
});
/**
 * Displays the invoice item quantity.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <InvoiceItem.Quantity className="text-gray-600" />
 *
 * // asChild with primitive
 * <InvoiceItem.Quantity asChild>
 *   <span className="badge" />
 * </InvoiceItem.Quantity>
 *
 * // asChild with React component
 * <InvoiceItem.Quantity asChild>
 *   {React.forwardRef(({ quantity, ...props }, ref) => (
 *     <span ref={ref} {...props} className="badge">
 *       Qty: {quantity}
 *     </span>
 *   ))}
 * </InvoiceItem.Quantity>
 * ```
 */
export const Quantity = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreInvoiceItem.Quantity, { children: ({ quantity }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.invoiceItemQuantity, customElement: children, customElementProps: { quantity }, content: quantity, ...otherProps, children: _jsx("span", { children: quantity }) })) }));
});
/**
 * Displays the invoice item total amount.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <InvoiceItem.Total className="font-bold text-xl" />
 *
 * // asChild with primitive
 * <InvoiceItem.Total asChild>
 *   <p className="font-bold text-xl text-green-600" />
 * </InvoiceItem.Total>
 *
 * // asChild with React component
 * <InvoiceItem.Total asChild>
 *   {React.forwardRef(({ value, currency, formattedValue, ...props }, ref) => (
 *     <p ref={ref} {...props} className="font-bold text-xl text-green-600">
 *       Total: {formattedValue}
 *     </p>
 *   ))}
 * </InvoiceItem.Total>
 * ```
 */
export const Total = React.forwardRef((props, ref) => {
    const { asChild, children, className, locale, ...otherProps } = props;
    return (_jsx(CoreInvoiceItem.Total, { locale: locale, children: ({ value, currency, formattedValue }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.invoiceItemTotal, customElement: children, customElementProps: { value, currency, formattedValue }, content: formattedValue, ...otherProps, children: _jsx("span", { children: formattedValue }) })) }));
});
