// @wix/headless-events/dist/react/OccurrenceList.js
import { jsx as _jsx } from "react/jsx-runtime";
import { AsChildSlot } from '@wix/headless-utils/react';
import React from 'react';
import * as Event from './Event.js';
import * as CoreOccurrenceList from './core/OccurrenceList.js';
var TestIds;
(function (TestIds) {
    TestIds["occurrenceListOccurrences"] = "occurrence-list-occurrences";
    TestIds["occurrenceListLoadMore"] = "occurrence-list-load-more";
    TestIds["occurrenceListError"] = "occurrence-list-error";
})(TestIds || (TestIds = {}));
/**
 * Root container that provides occurrence list service context to all child components.
 * Must be used as the top-level component for occurrence list functionality.
 *
 * @order 1
 * @component
 * @example
 * ```tsx
 * import { OccurrenceList } from '@wix/events/components';
 *
 * function OccurrenceListPage({ occurrenceListServiceConfig }) {
 *   return (
 *     <OccurrenceList.Root occurrenceListServiceConfig={occurrenceListServiceConfig}>
 *       <OccurrenceList.Occurrences>
 *         <OccurrenceList.OccurrenceRepeater>
 *           <Event.Image />
 *           <Event.Title />
 *           <Event.Date />
 *           <Event.Location />
 *           <Event.Description />
 *           <Event.RsvpButton label="RSVP" />
 *         </OccurrenceList.OccurrenceRepeater>
 *       </OccurrenceList.Occurrences>
 *     </OccurrenceList.Root>
 *   );
 * }
 * ```
 */
export const Root = (props) => {
    const { children, occurrenceListServiceConfig } = props;
    return (_jsx(CoreOccurrenceList.Root, { occurrenceListServiceConfig: occurrenceListServiceConfig, children: children }));
};
/**
 * Container for the occurrence list.
 * Follows List Container Level pattern.
 *
 * @component
 * @example
 * ```tsx
 * <OccurrenceList.Occurrences>
 *   <OccurrenceList.OccurrenceRepeater>
 *     <Event.Image />
 *     <Event.Title />
 *   </OccurrenceList.OccurrenceRepeater>
 * </OccurrenceList.Occurrences>
 * ```
 */
export const Occurrences = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreOccurrenceList.Occurrences, { children: ({ occurrences, hasOccurrences }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.occurrenceListOccurrences, customElement: children, customElementProps: { occurrences, hasOccurrences }, ...otherProps, children: _jsx("div", { children: children }) })) }));
});
/**
 * Repeater component that renders Event.Root for each occurrence.
 * Follows Repeater Level pattern.
 * Note: Repeater components do NOT support asChild as per architecture rules.
 *
 * @component
 * @example
 * ```tsx
 * <OccurrenceList.OccurrenceRepeater>
 *   <Event.Image />
 *   <Event.Title />
 * </OccurrenceList.OccurrenceRepeater>
 * ```
 */
export const OccurrenceRepeater = (props) => {
    const { children, className } = props;
    return (_jsx(CoreOccurrenceList.OccurrenceRepeater, { children: ({ occurrences }) => occurrences.map((occurrence) => (_jsx(Event.Root, { event: occurrence, className: className, children: children }, occurrence._id))) }));
};
/**
 * Displays a button to load more occurrences. Not rendered if no occurrences are left to load.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <OccurrenceList.LoadMoreTrigger className="bg-blue-600 hover:bg-blue-700 text-white" label="Load More" loadingState="Loading..." />
 *
 * // asChild with primitive
 * <OccurrenceList.LoadMoreTrigger asChild className="bg-blue-600 hover:bg-blue-700 text-white">
 *   <button>Load More</button>
 * </OccurrenceList.LoadMoreTrigger>
 *
 * // asChild with react component
 * <OccurrenceList.LoadMoreTrigger asChild className="bg-blue-600 hover:bg-blue-700 text-white">
 *   {React.forwardRef(({ isLoading, loadMoreOccurrences, ...props }, ref) => (
 *     <button ref={ref} {...props}>
 *       {isLoading ? 'Loading...' : 'Load More'}
 *     </button>
 *   ))}
 * </OccurrenceList.LoadMoreTrigger>
 * ```
 */
export const LoadMoreTrigger = React.forwardRef((props, ref) => {
    const { asChild, children, className, label, loadingState = label, ...otherProps } = props;
    return (_jsx(CoreOccurrenceList.LoadMoreTrigger, { children: ({ isLoading, loadMoreOccurrences }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.occurrenceListLoadMore, customElement: children, customElementProps: { isLoading, loadMoreOccurrences }, disabled: isLoading, onClick: loadMoreOccurrences, ...otherProps, children: _jsx("button", { children: isLoading ? loadingState : label }) })) }));
});
/**
 * Displays an error message when the occurrence list fails to load.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <OccurrenceList.Error className="text-red-500" />
 *
 * // asChild with primitive
 * <OccurrenceList.Error asChild className="text-red-500">
 *   <span />
 * </OccurrenceList.Error>
 *
 * // asChild with react component
 * <OccurrenceList.Error asChild className="text-red-500">
 *   {React.forwardRef(({ error, ...props }, ref) => (
 *     <span ref={ref} {...props}>
 *       {error}
 *     </span>
 *   ))}
 * </OccurrenceList.Error>
 * ```
 */
export const Error = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreOccurrenceList.Error, { children: ({ error }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.occurrenceListError, customElement: children, customElementProps: { error }, content: error, ...otherProps, children: _jsx("span", { children: error }) })) }));
});
