// @wix/headless-events/dist/react/Order.js
import { jsx as _jsx } from "react/jsx-runtime";
import { AsChildSlot } from '@wix/headless-utils/react';
import React from 'react';
import * as InvoiceItem from './InvoiceItem.js';
import * as CoreOrder from './core/Order.js';
var TestIds;
(function (TestIds) {
    TestIds["orderNumber"] = "order-number";
    TestIds["orderGuestEmail"] = "order-guest-email";
    TestIds["orderCreatedDate"] = "order-created-date";
    TestIds["orderDownloadTicketsButton"] = "order-download-tickets-button";
    TestIds["orderInvoiceItems"] = "order-invoice-items";
    TestIds["orderSubtotal"] = "order-subtotal";
    TestIds["orderTax"] = "order-tax";
    TestIds["orderFee"] = "order-fee";
    TestIds["orderTotal"] = "order-total";
    TestIds["orderPaidPlanDiscount"] = "order-paid-plan-discount";
    TestIds["orderCouponDiscount"] = "order-coupon-discount";
})(TestIds || (TestIds = {}));
/**
 * Order Root component that provides order service context to child components.
 * Must be used as the top-level component for order functionality.
 *
 * @order 1
 * @component
 * @example
 * ```tsx
 * <Order.Root orderServiceConfig={orderServiceConfig}>
 *   <Order.InvoiceItems className="px-6 border-b border-gray-300">
 *     <Order.InvoiceItemRepeater className="flex border-b border-gray-200 py-4 font-light text-gray-700">
 *       <InvoiceItem.Name className="w-[35%]" />
 *       <InvoiceItem.Price className="w-[25%]" />
 *       <InvoiceItem.Quantity className="w-[15%]" />
 *       <InvoiceItem.Total className="w-[25%] text-right" />
 *     </Order.InvoiceItemRepeater>
 *   </Order.InvoiceItems>
 * </Order.Root>
 * ```
 */
export const Root = (props) => {
    const { children, orderServiceConfig } = props;
    return (_jsx(CoreOrder.Root, { orderServiceConfig: orderServiceConfig, children: children }));
};
/**
 * Displays the order number.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Order.OrderNumber className="font-light text-lg text-gray-700" />
 *
 * // asChild with primitive
 * <Order.OrderNumber asChild>
 *   <h2 className="text-xl font-bold text-gray-900" />
 * </Order.OrderNumber>
 *
 * // asChild with react component
 * <Order.OrderNumber
 *   asChild
 *   className="font-light text-sm text-gray-700"
 * >
 *   {({ orderNumber }) => (
 *     <span>Order No. #{orderNumber}</span>
 *   )}
 * </Order.OrderNumber>
 * ```
 */
export const OrderNumber = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreOrder.OrderNumber, { children: ({ orderNumber }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.orderNumber, customElement: children, customElementProps: { orderNumber }, content: orderNumber, ...otherProps, children: _jsx("span", { children: orderNumber }) })) }));
});
/**
 * Displays the guest email.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Order.GuestEmail className="text-gray-600 font-light" />
 *
 * // asChild with primitive
 * <Order.GuestEmail asChild>
 *   <p className="text-sm text-gray-500 font-light" />
 * </Order.GuestEmail>
 *
 * // asChild with react component
 * <Order.GuestEmail asChild>
 *   {({ guestEmail }) => (
 *     <span>Email: {guestEmail}</span>
 *   )}
 * </Order.GuestEmail>
 * ```
 */
export const GuestEmail = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreOrder.GuestEmail, { children: ({ guestEmail }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.orderGuestEmail, customElement: children, customElementProps: { guestEmail }, content: guestEmail, ...otherProps, children: _jsx("span", { children: guestEmail }) })) }));
});
/**
 * Displays the order creation date.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Order.CreatedDate className="text-gray-600 font-light" />
 *
 * // asChild with primitive
 * <Order.CreatedDate asChild>
 *   <p className="text-sm text-gray-500 font-light" />
 * </Order.CreatedDate>
 *
 * // asChild with react component
 * <Order.CreatedDate
 *   asChild
 *   className="font-light text-sm text-gray-700"
 * >
 *   {({ createdDate, formattedDate }) => (
 *     <span>Placed on: {formattedDate}</span>
 *   )}
 * </Order.CreatedDate>
 * ```
 */
export const CreatedDate = React.forwardRef((props, ref) => {
    const { asChild, children, className, locale, ...otherProps } = props;
    return (_jsx(CoreOrder.CreatedDate, { locale: locale, children: ({ createdDate, formattedDate }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.orderCreatedDate, customElement: children, customElementProps: { createdDate, formattedDate }, content: formattedDate, ...otherProps, children: _jsx("span", { children: formattedDate }) })) }));
});
/**
 * Download tickets button that provides access to event tickets PDF.
 * Only shows when tickets are available for download.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Order.DownloadTicketsButton
 *   label="Download Tickets"
 *   className="bg-blue-500 text-white px-4 py-2 rounded"
 * />
 *
 * // asChild with primitive
 * <Order.DownloadTicketsButton asChild>
 *   <button className="bg-blue-500 text-white px-4 py-2 rounded hover:bg-blue-600">
 *     Download Tickets
 *   </button>
 * </Order.DownloadTicketsButton>
 *
 * // asChild with react component
 * <Order.DownloadTicketsButton asChild>
 *   {({ ticketsPdfUrl }) =>
 *     <button
 *       className="bg-blue-500 text-white px-4 py-2 rounded hover:bg-blue-600"
 *       onClick={() => window.open(ticketsPdfUrl, '_blank')}
 *     >
 *       Download Tickets
 *     </button>
 *   }
 * </Order.DownloadTicketsButton>
 * ```
 */
export const DownloadTicketsButton = React.forwardRef((props, ref) => {
    const { asChild, children, className, label, ...otherProps } = props;
    return (_jsx(CoreOrder.DownloadTicketsButton, { children: ({ ticketsPdfUrl }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.orderDownloadTicketsButton, customElement: children, customElementProps: { ticketsPdfUrl }, onClick: () => window.open(ticketsPdfUrl, '_blank'), ...otherProps, children: _jsx("button", { children: label }) })) }));
});
/**
 * Container for order invoice items.
 *
 * @component
 * @example
 * ```tsx
 * <Order.InvoiceItems className="px-6 border-b border-gray-300">
 *   <div className="flex border-b border-gray-200 py-4 font-light text-gray-700">
 *     <div className="w-[35%]">Ticket type</div>
 *     <div className="w-[25%]">Price</div>
 *     <div className="w-[15%]">Quantity</div>
 *     <div className="w-[25%] text-right">Total</div>
 *   </div>
 *   <Order.InvoiceItemRepeater className="flex border-b border-gray-200 py-4 font-light text-gray-700">
 *     <InvoiceItem.Name className="w-[35%]" />
 *     <InvoiceItem.Price className="w-[25%]" />
 *     <InvoiceItem.Quantity className="w-[15%]" />
 *     <InvoiceItem.Total className="w-[25%] text-right" />
 *   </Order.InvoiceItemRepeater>
 * </Order.InvoiceItems>
 * ```
 */
export const InvoiceItems = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreOrder.InvoiceItems, { children: ({ invoiceItems }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.orderInvoiceItems, customElement: children, customElementProps: { invoiceItems }, ...otherProps, children: _jsx("div", { children: children }) })) }));
});
/**
 * Repeater component that renders InvoiceItem.Root for each invoice item in the order.
 * Must be used within Order.InvoiceItems component.
 *
 * @component
 * @example
 * ```tsx
 * <Order.InvoiceItems>
 *   <Order.InvoiceItemRepeater className="flex border-b border-gray-200 py-4 font-light text-gray-700">
 *     <InvoiceItem.Name className="w-[35%]" />
 *     <InvoiceItem.Price className="w-[25%]" />
 *     <InvoiceItem.Quantity className="w-[15%]" />
 *     <InvoiceItem.Total className="w-[25%] text-right" />
 *   </Order.InvoiceItemRepeater>
 * </Order.InvoiceItems>
 * ```
 */
export const InvoiceItemRepeater = (props) => {
    const { children, className } = props;
    return (_jsx(CoreOrder.InvoiceItemRepeater, { children: ({ invoiceItems }) => invoiceItems.map((invoiceItem) => (_jsx(InvoiceItem.Root, { invoiceItem: invoiceItem, className: className, children: children }, invoiceItem._id))) }));
};
/**
 * Displays the order subtotal amount.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Order.Subtotal className="text-lg font-light text-gray-700" />
 *
 * // asChild with primitive
 * <Order.Subtotal asChild>
 *   <p className="text-lg font-light text-gray-700" />
 * </Order.Subtotal>
 *
 * // asChild with react component
 * <Order.Subtotal
 *   asChild
 *   className="font-light text-gray-700 justify-between flex"
 * >
 *   {({ value, currency, formattedValue }) => (
 *     <div>
 *       <span>Subtotal:</span>
 *       <span>{formattedValue}</span>
 *     </div>
 *   )}
 * </Order.Subtotal>
 * ```
 */
export const Subtotal = React.forwardRef((props, ref) => {
    const { asChild, children, className, locale, ...otherProps } = props;
    return (_jsx(CoreOrder.Subtotal, { locale: locale, children: ({ value, currency, formattedValue }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.orderSubtotal, customElement: children, customElementProps: { value, currency, formattedValue }, content: formattedValue, ...otherProps, children: _jsx("span", { children: formattedValue }) })) }));
});
/**
 * Displays the order paid plan discount.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Order.PaidPlanDiscount className="text-gray-600 font-light" />
 *
 * // asChild with primitive
 * <Order.PaidPlanDiscount asChild>
 *   <p className="text-sm text-gray-600 font-light" />
 * </Order.PaidPlanDiscount>
 *
 * // asChild with react component
 * <Order.PaidPlanDiscount asChild>
 *   {({ value, currency, formattedValue, rate }) => (
 *     <div>
 *       <span>Paid Plan Discount ({rate}%)</span>
 *       <span>{formattedValue}</span>
 *     </div>
 *   )}
 * </Order.PaidPlanDiscount>
 * ```
 */
export const PaidPlanDiscount = React.forwardRef((props, ref) => {
    const { asChild, children, className, locale, ...otherProps } = props;
    return (_jsx(CoreOrder.PaidPlanDiscount, { locale: locale, children: ({ value, currency, formattedValue, rate }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.orderPaidPlanDiscount, customElement: children, customElementProps: { value, currency, formattedValue, rate }, content: formattedValue, ...otherProps, children: _jsx("span", { children: formattedValue }) })) }));
});
/**
 * Displays the order coupon discount.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Order.CouponDiscount className="text-gray-600 font-light" />
 *
 * // asChild with primitive
 * <Order.CouponDiscount asChild>
 *   <p className="text-sm text-gray-600 font-light" />
 * </Order.CouponDiscount>
 *
 * // asChild with react component
 * <Order.CouponDiscount asChild>
 *   {({ value, currency, formattedValue }) => (
 *     <div>
 *       <span>Coupon Discount</span>
 *       <span>{formattedValue}</span>
 *     </div>
 *   )}
 * </Order.CouponDiscount>
 * ```
 */
export const CouponDiscount = React.forwardRef((props, ref) => {
    const { asChild, children, className, locale, ...otherProps } = props;
    return (_jsx(CoreOrder.CouponDiscount, { locale: locale, children: ({ value, currency, formattedValue }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.orderCouponDiscount, customElement: children, customElementProps: { value, currency, formattedValue }, content: formattedValue, ...otherProps, children: _jsx("span", { children: formattedValue }) })) }));
});
/**
 * Displays the order tax.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Order.Tax className="text-gray-600 font-light" />
 *
 * // asChild with primitive
 * <Order.Tax asChild>
 *   <p className="text-sm text-gray-600 font-light" />
 * </Order.Tax>
 *
 * // asChild with react component
 * <Order.Tax
 *   asChild
 *   className="font-light text-gray-700 justify-between flex"
 * >
 *   {({ value, currency, formattedValue, rate, name }) => (
 *     <div>
 *       <span>{name} ({rate}%)</span>
 *       <span>{formattedValue}</span>
 *     </div>
 *   )}
 * </Order.Tax>
 * ```
 */
export const Tax = React.forwardRef((props, ref) => {
    const { asChild, children, className, locale, ...otherProps } = props;
    return (_jsx(CoreOrder.Tax, { locale: locale, children: ({ value, currency, formattedValue, rate, name }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.orderTax, customElement: children, customElementProps: { value, currency, formattedValue, rate, name }, content: formattedValue, ...otherProps, children: _jsx("span", { children: formattedValue }) })) }));
});
/**
 * Displays the order fee.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Order.Fee className="text-gray-600 font-light" />
 *
 * // asChild with primitive
 * <Order.Fee asChild>
 *   <p className="text-sm text-gray-600 font-light" />
 * </Order.Fee>
 *
 * // asChild with react component
 * <Order.Fee
 *   asChild
 *   className="font-light text-gray-700 justify-between flex"
 * >
 *   {({ value, currency, formattedValue, rate }) => (
 *     <div>
 *       <span>Service Fee ({rate}%)</span>
 *       <span>{formattedValue}</span>
 *     </div>
 *   )}
 * </Order.Fee>
 * ```
 */
export const Fee = React.forwardRef((props, ref) => {
    const { asChild, children, className, locale, ...otherProps } = props;
    return (_jsx(CoreOrder.Fee, { locale: locale, children: ({ value, currency, formattedValue, rate }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.orderFee, customElement: children, customElementProps: { value, currency, formattedValue, rate }, content: formattedValue, ...otherProps, children: _jsx("span", { children: formattedValue }) })) }));
});
/**
 * Displays the order total amount.
 * This represents the final amount to be paid including all fees and taxes.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Order.Total className="text-xl font-light text-gray-700" />
 *
 * // asChild with primitive
 * <Order.Total asChild>
 *   <p className="text-2xl font-bold text-gray-900" />
 * </Order.Total>
 *
 * // asChild with react component
 * <Order.Total
 *   asChild
 *   className="font-light text-gray-700 justify-between flex py-5"
 * >
 *   {({ value, currency, formattedValue }) => (
 *     <div>
 *       <span>Total:</span>
 *       <span>{formattedValue}</span>
 *     </div>
 *   )}
 * </Order.Total>
 * ```
 */
export const Total = React.forwardRef((props, ref) => {
    const { asChild, children, className, locale, ...otherProps } = props;
    return (_jsx(CoreOrder.Total, { locale: locale, children: ({ value, currency, formattedValue }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.orderTotal, customElement: children, customElementProps: { value, currency, formattedValue }, content: formattedValue, ...otherProps, children: _jsx("span", { children: formattedValue }) })) }));
});
