// @wix/headless-events/dist/react/ScheduleItemsGroup.js
import { jsx as _jsx } from "react/jsx-runtime";
import { AsChildSlot } from '@wix/headless-utils/react';
import React from 'react';
import * as ScheduleItem from './ScheduleItem.js';
import * as CoreScheduleItemsGroup from './core/ScheduleItemsGroup.js';
var TestIds;
(function (TestIds) {
    TestIds["scheduleItemsGroupRoot"] = "schedule-items-group-root";
    TestIds["scheduleItemsGroupDateLabel"] = "schedule-items-group-date-label";
    TestIds["scheduleItemsGroupItems"] = "schedule-items-group-items";
})(TestIds || (TestIds = {}));
/**
 * Root container that provides schedule items group service context to all child components.
 * Must be used as the top-level ScheduleItemsGroup component.
 *
 * @order 1
 * @component
 * @example
 * ```tsx
 * import { ScheduleItemsGroup } from '@wix/events/components';
 *
 * function ScheduleItemsGroupComponent({ itemsGroup }) {
 *   return (
 *     <ScheduleItemsGroup.Root itemsGroup={itemsGroup}>
 *       <ScheduleItemsGroup.DateLabel />
 *       <ScheduleItemsGroup.Items>
 *         <ScheduleItemsGroup.ItemRepeater>
 *           <ScheduleItem.Name />
 *           <ScheduleItem.TimeSlot />
 *         </ScheduleItemsGroup.ItemRepeater>
 *       </ScheduleItemsGroup.Items>
 *     </ScheduleItemsGroup.Root>
 *   );
 * }
 * ```
 */
export const Root = React.forwardRef((props, ref) => {
    const { itemsGroup, asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreScheduleItemsGroup.Root, { itemsGroup: itemsGroup, children: _jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.scheduleItemsGroupRoot, customElement: children, customElementProps: {}, ...otherProps, children: _jsx("div", { children: children }) }) }));
});
/**
 * Displays the date label for a schedule group (e.g., "Mon, 07 Jul").
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <ScheduleItemsGroup.DateLabel className="text-lg font-semibold" />
 *
 * // asChild with primitive
 * <ScheduleItemsGroup.DateLabel asChild>
 *   <h3 className="text-lg font-semibold" />
 * </ScheduleItemsGroup.DateLabel>
 *
 * // asChild with react component
 * <ScheduleItemsGroup.DateLabel asChild>
 *   {React.forwardRef(({ date, timeZoneId, formattedDate, ...props }, ref) => (
 *     <h3 ref={ref} {...props} className="text-lg font-semibold">
 *       {formattedDate}
 *     </h3>
 *   ))}
 * </ScheduleItemsGroup.DateLabel>
 * ```
 */
export const DateLabel = React.forwardRef((props, ref) => {
    const { asChild, children, className, locale, ...otherProps } = props;
    return (_jsx(CoreScheduleItemsGroup.DateLabel, { locale: locale, children: ({ date, timeZoneId, formattedDate }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.scheduleItemsGroupDateLabel, customElement: children, customElementProps: { date, timeZoneId, formattedDate }, content: formattedDate, ...otherProps, children: _jsx("span", { children: formattedDate }) })) }));
});
/**
 * Container for the schedule items in the group.
 *
 * @component
 * @example
 * ```tsx
 * <ScheduleItemsGroup.Items>
 *   <ScheduleItemsGroup.ItemRepeater>
 *     <ScheduleItem.Name />
 *     <ScheduleItem.TimeSlot />
 *   </ScheduleItemsGroup.ItemRepeater>
 * </ScheduleItemsGroup.Items>
 * ```
 */
export const Items = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreScheduleItemsGroup.Items, { children: ({ items }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.scheduleItemsGroupItems, customElement: children, customElementProps: { items }, ...otherProps, children: _jsx("div", { children: children }) })) }));
});
/**
 * Repeater component that renders ScheduleItem.Root for each schedule item in the group.
 *
 * @component
 * @example
 * ```tsx
 * <ScheduleItemsGroup.ItemRepeater>
 *   <ScheduleItem.Name />
 *   <ScheduleItem.TimeSlot />
 *   <ScheduleItem.Duration />
 * </ScheduleItemsGroup.ItemRepeater>
 * ```
 */
export const ItemRepeater = (props) => {
    const { children, className } = props;
    return (_jsx(CoreScheduleItemsGroup.Items, { children: ({ items }) => items.map((item) => (_jsx(ScheduleItem.Root, { item: item, className: className, children: children }, item._id))) }));
};
