// @wix/headless-events/dist/react/ScheduleList.js
import { jsx as _jsx } from "react/jsx-runtime";
import { Filter as FilterPrimitive } from '@wix/headless-components/react';
import { AsChildSlot } from '@wix/headless-utils/react';
import React from 'react';
import * as ScheduleItem from './ScheduleItem.js';
import * as ScheduleItemsGroup from './ScheduleItemsGroup.js';
import * as CoreScheduleList from './core/ScheduleList.js';
var TestIds;
(function (TestIds) {
    TestIds["scheduleListItems"] = "schedule-list-items";
    TestIds["scheduleListGroups"] = "schedule-list-groups";
})(TestIds || (TestIds = {}));
/**
 * Root container that provides schedule list service context to all child components.
 * Must be used as the top-level component for schedule list functionality.
 *
 * @order 1
 * @component
 * @example
 * ```tsx
 * import { ScheduleList } from '@wix/events/components';
 *
 * function ScheduleListPage({ scheduleListServiceConfig }) {
 *   return (
 *     <ScheduleList.Root scheduleListServiceConfig={scheduleListServiceConfig}>
 *       <ScheduleList.Items>
 *         <ScheduleList.ItemRepeater>
 *           <ScheduleItem.Name />
 *           <ScheduleItem.TimeSlot />
 *           <ScheduleItem.Duration />
 *           <ScheduleItem.Description />
 *           <ScheduleItem.Stage />
 *           <ScheduleItem.Tags>
 *             <ScheduleItem.TagRepeater>
 *              <ScheduleItemTag.Label />
 *             </ScheduleItem.TagRepeater>
 *           </ScheduleItem.Tags>
 *         </ScheduleList.ItemRepeater>
 *       </ScheduleList.Items>
 *     </ScheduleList.Root>
 *   );
 * }
 * ```
 */
export const Root = (props) => {
    const { children, scheduleListServiceConfig } = props;
    return (_jsx(CoreScheduleList.Root, { scheduleListServiceConfig: scheduleListServiceConfig, children: children }));
};
/**
 * Container for the schedule list items.
 *
 * @component
 * @example
 * ```tsx
 * <ScheduleList.Items>
 *   <ScheduleList.ItemRepeater>
 *     <ScheduleItem.Name />
 *     <ScheduleItem.TimeSlot />
 *     <ScheduleItem.Duration />
 *     <ScheduleItem.Description />
 *     <ScheduleItem.Stage />
 *     <ScheduleItem.Tags>
 *       <ScheduleItem.TagRepeater>
 *         <ScheduleItemTag.Label />
 *       </ScheduleItem.TagRepeater>
 *     </ScheduleItem.Tags>
 *   </ScheduleList.ItemRepeater>
 * </ScheduleList.Items>
 * ```
 */
export const Items = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreScheduleList.Items, { children: ({ items }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.scheduleListItems, customElement: children, customElementProps: { items }, ...otherProps, children: _jsx("div", { children: children }) })) }));
});
/**
 * Repeater component that renders ScheduleItem.Root for each schedule item.
 *
 * @component
 * @example
 * ```tsx
 * <ScheduleList.ItemRepeater>
 *   <ScheduleItem.Name />
 *   <ScheduleItem.TimeSlot />
 *   <ScheduleItem.Duration />
 *   <ScheduleItem.Description />
 *   <ScheduleItem.Stage />
 *   <ScheduleItem.Tags>
 *     <ScheduleItem.TagRepeater>
 *       <ScheduleItemTag.Label />
 *     </ScheduleItem.TagRepeater>
 *   </ScheduleItem.Tags>
 * </ScheduleList.ItemRepeater>
 * ```
 */
export const ItemRepeater = (props) => {
    const { children, className } = props;
    return (_jsx(CoreScheduleList.ItemRepeater, { children: ({ items }) => items.map((item) => (_jsx(ScheduleItem.Root, { item: item, className: className, children: children }, item._id))) }));
};
/**
 * Container for the grouped schedule items.
 * Follows List Container Level pattern - supports empty state rendering.
 *
 * @component
 * @example
 * ```tsx
 * <ScheduleList.Groups emptyState={<div>No schedule items available</div>}>
 *   <ScheduleList.GroupRepeater>
 *     <ScheduleItemsGroup.GroupDateLabel />
 *     <ScheduleItemsGroup.GroupItems>
 *       <ScheduleItemsGroup.GroupItemRepeater>
 *         <ScheduleItem.Name />
 *         <ScheduleItem.TimeSlot />
 *       </ScheduleItemsGroup.GroupItemRepeater>
 *     </ScheduleItemsGroup.GroupItems>
 *   </ScheduleList.GroupRepeater>
 * </ScheduleList.Groups>
 * ```
 */
export const Groups = React.forwardRef((props, ref) => {
    const { asChild, children, className, emptyState, ...otherProps } = props;
    return (_jsx(CoreScheduleList.Groups, { children: ({ itemsGroups }) => {
            if (!itemsGroups.length) {
                return emptyState || null;
            }
            return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.scheduleListGroups, customElement: children, customElementProps: { itemsGroups }, ...otherProps, children: _jsx("div", { children: children }) }));
        } }));
});
/**
 * Repeater component that renders ScheduleItemsGroup.Root for each schedule items group.
 *
 * @component
 * @example
 * ```tsx
 * <ScheduleList.GroupRepeater>
 *   <ScheduleItemsGroup.DateLabel />
 *   <ScheduleItemsGroup.Items>
 *     <ScheduleItemsGroup.ItemRepeater>
 *       <ScheduleItem.Name />
 *       <ScheduleItem.TimeSlot />
 *       <ScheduleItem.Duration />
 *       <ScheduleItem.Description />
 *       <ScheduleItem.Stage />
 *       <ScheduleItem.Tags>
 *         <ScheduleItem.TagRepeater>
 *           <ScheduleItemTag.Label />
 *         </ScheduleItem.TagRepeater>
 *       </ScheduleItem.Tags>
 *     </ScheduleItemsGroup.ItemRepeater>
 *   </ScheduleItemsGroup.Items>
 * </ScheduleList.GroupRepeater>
 * ```
 */
export const GroupRepeater = (props) => {
    const { children, className } = props;
    return (_jsx(CoreScheduleList.GroupRepeater, { children: ({ itemsGroups }) => itemsGroups.map((itemsGroup) => (_jsx(ScheduleItemsGroup.Root, { itemsGroup: itemsGroup, className: className, children: children }, itemsGroup.id))) }));
};
/**
 * Container for the schedule list filters.
 *
 * @component
 * @example
 * ```tsx
 * <ScheduleList.Filters allStagesLabel="All">
 *   <Filter.FilterOptions className="border-b border-gray-500 mb-6">
 *     <Filter.FilterOptionRepeater>
 *       <Filter.FilterOption.SingleFilter className="flex gap-2" />
 *     </Filter.FilterOptionRepeater>
 *   </Filter.FilterOptions>
 * </ScheduleList.Filters>
 * ```
 */
export const Filters = React.forwardRef((props, ref) => {
    const { asChild, children, className, allStagesLabel, ...otherProps } = props;
    return (_jsx(CoreScheduleList.Filters, { allStagesLabel: allStagesLabel, children: ({ filterOptions, value, onChange }) => (_jsx(FilterPrimitive.Root, { ref: ref, asChild: asChild, className: className, filterOptions: filterOptions, value: value, onChange: onChange, ...otherProps, children: children })) }));
});
