// @wix/unidriver-jsdom-react/dist/esm/UniDriverClass.js
import { act, fireEvent, waitFor } from '@testing-library/react';
import { PointerEventsCheckLevel, userEvent } from '@testing-library/user-event';
import { bindUniDriverApiMethods, UniDriverBase } from '@wix/unidriver-common';
import { UniDriverClassList } from './UniDriverClassList';
import { enterText } from './utils/enterText';
import { enterValue } from './utils/enterValue';
export class UniDriverClass extends UniDriverBase {
    constructor(ctx) {
        super(ctx);
        Object.defineProperty(this, "type", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: 'react'
        });
        Object.defineProperty(this, "get", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: async (selector, options) => {
                return waitFor(() => {
                    return super.get(selector, options);
                }, { timeout: this.getTimeBudget(options?.timeout) });
            }
        });
        Object.defineProperty(this, "getAll", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: async (selector, options) => {
                return waitFor(() => {
                    return super.getAll(selector, options);
                }, { timeout: this.getTimeBudget(options?.timeout) });
            }
        });
        Object.defineProperty(this, "api", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: {
                Class: UniDriverClass,
                ListClass: UniDriverClassList,
                unwrap: async (base) => base,
                getType: () => 'jsdom-react',
                getElementsBySelector: async (base, selector) => [...base.querySelectorAll(selector)],
                click: async (base, options) => {
                    const clickParams = {
                        delay: null,
                        // We don't check pointer events for jsdom as we can't compute styles
                        pointerEventsCheck: PointerEventsCheckLevel.Never,
                    };
                    if (options?.type === 'double') {
                        return userEvent.dblClick(base, clickParams);
                    }
                    return userEvent.click(base, clickParams);
                },
                text: async (base) => base.textContent ?? '',
                value: async (base) => {
                    const value = base.value;
                    return typeof value === 'number' ? String(value) : (value ?? '');
                },
                enterText: async (base, value, options) => {
                    await enterText(base, value, options);
                },
                enterValue: async (base, value, options) => {
                    await enterValue(base, value, options);
                },
                pressKey: async (base, keyMeta) => {
                    await act(async () => {
                        fireEvent.keyDown(base, keyMeta);
                        fireEvent.keyUp(base, keyMeta);
                    });
                },
                attr: async (base, attr) => base.getAttribute(attr),
                hasClass: async (base, className) => base.classList.contains(className),
                prop: async (base, prop) => base[prop],
                mouse: {
                    press: async (base) => {
                        await act(async () => {
                            fireEvent.mouseDown(base);
                        });
                    },
                    release: async (base) => {
                        await act(async () => {
                            fireEvent.mouseUp(base);
                        });
                    },
                    hover: async (base) => {
                        await act(async () => {
                            fireEvent.mouseEnter(base);
                        });
                    },
                    leave: async (base) => {
                        await act(async () => {
                            fireEvent.mouseLeave(base);
                        });
                    },
                    moveTo: async (base, to) => {
                        const { left, top } = (await to.unwrap()).getBoundingClientRect();
                        await act(async () => {
                            fireEvent.mouseMove(base, { clientX: left, clientY: top });
                        });
                    },
                },
            }
        });
        bindUniDriverApiMethods(this, ['get', 'getAll']);
    }
}
