// @wix/design-system/dist/esm/Page/test/examples/BMPageContainer.st.css.js

import "../../../Foundation/stylable/colors.st.css.js";
import { classesRuntime, statesRuntime } from "./../../../stylable-esm-runtime.js";



var _namespace_ = "wds_1_226_0_BMPageContainer";
var _style_ = classesRuntime.bind(null, _namespace_);

export var cssStates = statesRuntime.bind(null, _namespace_);
export var style = _style_;
export var st = _style_;

export var namespace = _namespace_;
export var classes = {"root":"wds_1_226_0_BMPageContainer__root","header":"wds_1_226_0_BMPageContainer__header","body":"wds_1_226_0_BMPageContainer__body","sideBar":"wds_1_226_0_BMPageContainer__sideBar","sideBarContent":"wds_1_226_0_BMPageContainer__sideBarContent","rightSide":"wds_1_226_0_BMPageContainer__rightSide","mainContentContainer":"wds_1_226_0_BMPageContainer__mainContentContainer","mainContent":"wds_1_226_0_BMPageContainer__mainContent","withLoadingIndicatorWrapper":"wds_1_226_0_BMPageContainer__withLoadingIndicatorWrapper","withLoadingIndicatorContainer":"wds_1_226_0_BMPageContainer__withLoadingIndicatorContainer"};
export var keyframes = {}; 
export var layers = {};
export var containers = {};
export var stVars = {}; 
export var vars = {};
