// @wix/auto_sdk_payments_payins-configurations/build/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  SortOrder: () => SortOrder,
  WebhookIdentityType: () => WebhookIdentityType,
  bulkUpdatePayinsConfigurationTags: () => bulkUpdatePayinsConfigurationTags2,
  bulkUpdatePayinsConfigurationTagsByFilter: () => bulkUpdatePayinsConfigurationTagsByFilter2,
  createPayinsConfiguration: () => createPayinsConfiguration2,
  getPayinsConfiguration: () => getPayinsConfiguration2,
  queryPayinsConfigurations: () => queryPayinsConfigurations2,
  updatePayinsConfiguration: () => updatePayinsConfiguration2
});
module.exports = __toCommonJS(index_typings_exports);

// src/payments-payins-configurations-v1-payins-configuration-payins-configurations.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/payments-payins-configurations-v1-payins-configuration-payins-configurations.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(opts) {
  const domainToMappings = {};
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_payins-configurations";
function createPayinsConfiguration(payload) {
  function __createPayinsConfiguration({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "payinsConfiguration.createdDate" },
          { path: "payinsConfiguration.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.payments.payins_configurations.v1.payins_configuration",
      method: "POST",
      methodFqn: "wix.payments.payins_configurations.v1.PayinsConfigurationService.CreatePayinsConfiguration",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(
        { protoPath: "/v1/payins-configurations", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "payinsConfiguration.createdDate" },
            { path: "payinsConfiguration.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createPayinsConfiguration;
}
function getPayinsConfiguration(payload) {
  function __getPayinsConfiguration({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.payins_configurations.v1.payins_configuration",
      method: "GET",
      methodFqn: "wix.payments.payins_configurations.v1.PayinsConfigurationService.GetPayinsConfiguration",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(
        {
          protoPath: "/v1/payins-configurations/{payinsConfigurationId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "payinsConfiguration.createdDate" },
            { path: "payinsConfiguration.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getPayinsConfiguration;
}
function updatePayinsConfiguration(payload) {
  function __updatePayinsConfiguration({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "payinsConfiguration.createdDate" },
          { path: "payinsConfiguration.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.payments.payins_configurations.v1.payins_configuration",
      method: "PATCH",
      methodFqn: "wix.payments.payins_configurations.v1.PayinsConfigurationService.UpdatePayinsConfiguration",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(
        {
          protoPath: "/v1/payins-configurations/{payinsConfiguration.id}",
          data: serializedData,
          host
        }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "payinsConfiguration.createdDate" },
            { path: "payinsConfiguration.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updatePayinsConfiguration;
}
function queryPayinsConfigurations(payload) {
  function __queryPayinsConfigurations({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.payins_configurations.v1.payins_configuration",
      method: "POST",
      methodFqn: "wix.payments.payins_configurations.v1.PayinsConfigurationService.QueryPayinsConfigurations",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(
        { protoPath: "/v1/payins-configurations/query", data: payload, host }
      ),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "payinsConfigurations.createdDate" },
            { path: "payinsConfigurations.updatedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(
            {
              protoPath: "/v1/payins-configurations/query",
              data: payload,
              host
            }
          ),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryPayinsConfigurations;
}
function bulkUpdatePayinsConfigurationTags(payload) {
  function __bulkUpdatePayinsConfigurationTags({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.payins_configurations.v1.payins_configuration",
      method: "POST",
      methodFqn: "wix.payments.payins_configurations.v1.PayinsConfigurationService.BulkUpdatePayinsConfigurationTags",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(
        {
          protoPath: "/v1/bulk/payins-configurations/update-tags",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdatePayinsConfigurationTags;
}
function bulkUpdatePayinsConfigurationTagsByFilter(payload) {
  function __bulkUpdatePayinsConfigurationTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.payins_configurations.v1.payins_configuration",
      method: "POST",
      methodFqn: "wix.payments.payins_configurations.v1.PayinsConfigurationService.BulkUpdatePayinsConfigurationTagsByFilter",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(
        {
          protoPath: "/v1/bulk/payins-configurations/update-tags-by-filter",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdatePayinsConfigurationTagsByFilter;
}

// src/payments-payins-configurations-v1-payins-configuration-payins-configurations.universal.ts
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createPayinsConfiguration2(payinsConfiguration) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    payinsConfiguration
  });
  const reqOpts = createPayinsConfiguration(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.payinsConfiguration;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { payinsConfiguration: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["payinsConfiguration"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getPayinsConfiguration2(payinsConfigurationId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    payinsConfigurationId
  });
  const reqOpts = getPayinsConfiguration(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.payinsConfiguration;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { payinsConfigurationId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["payinsConfigurationId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updatePayinsConfiguration2(_id, payinsConfiguration) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    payinsConfiguration: { ...payinsConfiguration, id: _id }
  });
  const reqOpts = updatePayinsConfiguration(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.payinsConfiguration;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { payinsConfiguration: "$[1]" },
        explicitPathsToArguments: { "payinsConfiguration.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "payinsConfiguration"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryPayinsConfigurations2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryPayinsConfigurations(
        payload
      );
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [])
      );
      return {
        items: transformedData?.payinsConfigurations,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function bulkUpdatePayinsConfigurationTags2(ids, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    ids,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdatePayinsConfigurationTags(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          ids: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["ids", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdatePayinsConfigurationTagsByFilter2(filter, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    filter,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdatePayinsConfigurationTagsByFilter(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          filter: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["filter", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  SortOrder,
  WebhookIdentityType,
  bulkUpdatePayinsConfigurationTags,
  bulkUpdatePayinsConfigurationTagsByFilter,
  createPayinsConfiguration,
  getPayinsConfiguration,
  queryPayinsConfigurations,
  updatePayinsConfiguration
});
