// @wix/auto_sdk_payments_payins-configurations/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  bulkUpdatePayinsConfigurationTags: () => bulkUpdatePayinsConfigurationTags2,
  bulkUpdatePayinsConfigurationTagsByFilter: () => bulkUpdatePayinsConfigurationTagsByFilter2,
  createPayinsConfiguration: () => createPayinsConfiguration2,
  getPayinsConfiguration: () => getPayinsConfiguration2,
  queryPayinsConfigurations: () => queryPayinsConfigurations2,
  updatePayinsConfiguration: () => updatePayinsConfiguration2
});
module.exports = __toCommonJS(meta_exports);

// src/payments-payins-configurations-v1-payins-configuration-payins-configurations.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(opts) {
  const domainToMappings = {};
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_payins-configurations";
function createPayinsConfiguration(payload) {
  function __createPayinsConfiguration({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "payinsConfiguration.createdDate" },
          { path: "payinsConfiguration.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.payments.payins_configurations.v1.payins_configuration",
      method: "POST",
      methodFqn: "wix.payments.payins_configurations.v1.PayinsConfigurationService.CreatePayinsConfiguration",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(
        { protoPath: "/v1/payins-configurations", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "payinsConfiguration.createdDate" },
            { path: "payinsConfiguration.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createPayinsConfiguration;
}
function getPayinsConfiguration(payload) {
  function __getPayinsConfiguration({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.payins_configurations.v1.payins_configuration",
      method: "GET",
      methodFqn: "wix.payments.payins_configurations.v1.PayinsConfigurationService.GetPayinsConfiguration",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(
        {
          protoPath: "/v1/payins-configurations/{payinsConfigurationId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "payinsConfiguration.createdDate" },
            { path: "payinsConfiguration.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getPayinsConfiguration;
}
function updatePayinsConfiguration(payload) {
  function __updatePayinsConfiguration({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "payinsConfiguration.createdDate" },
          { path: "payinsConfiguration.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.payments.payins_configurations.v1.payins_configuration",
      method: "PATCH",
      methodFqn: "wix.payments.payins_configurations.v1.PayinsConfigurationService.UpdatePayinsConfiguration",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(
        {
          protoPath: "/v1/payins-configurations/{payinsConfiguration.id}",
          data: serializedData,
          host
        }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "payinsConfiguration.createdDate" },
            { path: "payinsConfiguration.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updatePayinsConfiguration;
}
function queryPayinsConfigurations(payload) {
  function __queryPayinsConfigurations({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.payins_configurations.v1.payins_configuration",
      method: "POST",
      methodFqn: "wix.payments.payins_configurations.v1.PayinsConfigurationService.QueryPayinsConfigurations",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(
        { protoPath: "/v1/payins-configurations/query", data: payload, host }
      ),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "payinsConfigurations.createdDate" },
            { path: "payinsConfigurations.updatedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(
            {
              protoPath: "/v1/payins-configurations/query",
              data: payload,
              host
            }
          ),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryPayinsConfigurations;
}
function bulkUpdatePayinsConfigurationTags(payload) {
  function __bulkUpdatePayinsConfigurationTags({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.payins_configurations.v1.payins_configuration",
      method: "POST",
      methodFqn: "wix.payments.payins_configurations.v1.PayinsConfigurationService.BulkUpdatePayinsConfigurationTags",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(
        {
          protoPath: "/v1/bulk/payins-configurations/update-tags",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdatePayinsConfigurationTags;
}
function bulkUpdatePayinsConfigurationTagsByFilter(payload) {
  function __bulkUpdatePayinsConfigurationTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.payins_configurations.v1.payins_configuration",
      method: "POST",
      methodFqn: "wix.payments.payins_configurations.v1.PayinsConfigurationService.BulkUpdatePayinsConfigurationTagsByFilter",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsPayinsConfigurationsV1PayinsConfigurationServiceUrl(
        {
          protoPath: "/v1/bulk/payins-configurations/update-tags-by-filter",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdatePayinsConfigurationTagsByFilter;
}

// src/payments-payins-configurations-v1-payins-configuration-payins-configurations.meta.ts
function createPayinsConfiguration2() {
  const payload = {};
  const getRequestOptions = createPayinsConfiguration(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/payins-configurations",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getPayinsConfiguration2() {
  const payload = { payinsConfigurationId: ":payinsConfigurationId" };
  const getRequestOptions = getPayinsConfiguration(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/payins-configurations/{payinsConfigurationId}",
    pathParams: { payinsConfigurationId: "payinsConfigurationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updatePayinsConfiguration2() {
  const payload = {
    payinsConfiguration: { id: ":payinsConfigurationId" }
  };
  const getRequestOptions = updatePayinsConfiguration(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/payins-configurations/{payinsConfiguration.id}",
    pathParams: { payinsConfigurationId: "payinsConfigurationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryPayinsConfigurations2() {
  const payload = {};
  const getRequestOptions = queryPayinsConfigurations(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/payins-configurations/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdatePayinsConfigurationTags2() {
  const payload = {};
  const getRequestOptions = bulkUpdatePayinsConfigurationTags(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/payins-configurations/update-tags",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdatePayinsConfigurationTagsByFilter2() {
  const payload = {};
  const getRequestOptions = bulkUpdatePayinsConfigurationTagsByFilter(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/payins-configurations/update-tags-by-filter",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  bulkUpdatePayinsConfigurationTags,
  bulkUpdatePayinsConfigurationTagsByFilter,
  createPayinsConfiguration,
  getPayinsConfiguration,
  queryPayinsConfigurations,
  updatePayinsConfiguration
});
