// @wix/headless-restaurants-menus/dist/react/Item.js
import { jsx as _jsx, jsxs as _jsxs, Fragment as _Fragment } from "react/jsx-runtime";
import React from 'react';
import { Item as CoreItem, ItemName, ItemDescription, Price as CorePrice, Images as CoreImages, Featured as CoreFeatured, useItemContext, } from './core/index.js';
import { AsChildSlot } from '@wix/headless-utils/react';
import { TestIds } from './TestIds.js';
import { WixMediaImage, MediaGallery } from '@wix/headless-media/react';
import { StarFilledIcon, ChevronLeftIcon, ChevronRightIcon, } from '@radix-ui/react-icons';
import * as VariantComponent from './Variant.js';
import * as LabelComponent from './Label.js';
import * as ModifierGroupComponent from './ModifierGroup.js';
export function Root(props) {
    if (!props.item) {
        return null;
    }
    return (_jsx(CoreItem, { item: props.item, children: _jsx("div", { "data-featured": props.item.featured ? 'true' : 'false', className: "group", children: props.children }) }));
}
/**
 * Displays the item name with customizable rendering following the documented API.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Item.Name className="text-lg font-semibold" />
 *
 * // asChild with primitive
 * <Item.Name asChild>
 *   <h2 className="text-lg font-semibold" />
 * </Item.Name>
 *
 * // asChild with react component
 * <Item.Name asChild>
 *   {React.forwardRef(({name, ...props}, ref) => (
 *     <h2 ref={ref} {...props} className="text-lg font-semibold">
 *       {name}
 *     </h2>
 *   ))}
 * </Item.Name>
 * ```
 */
export const Name = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(ItemName, { children: ({ name }) => {
            return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.itemName, customElement: children, customElementProps: { name }, content: name, ...otherProps, children: _jsx("p", { children: name }) }));
        } }));
});
/**
 * Displays the item description with customizable rendering following the documented API.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Item.Description className="text-sm text-muted-foreground" />
 *
 * // asChild with primitive
 * <Item.Description asChild>
 *   <p className="text-sm text-muted-foreground" />
 * </Item.Description>
 *
 * // asChild with react component
 * <Item.Description asChild>
 *   {React.forwardRef(({description, ...props}, ref) => (
 *     <p ref={ref} {...props} className="text-sm text-muted-foreground">
 *       {description}
 *     </p>
 *   ))}
 * </Item.Description>
 * ```
 */
export const Description = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(ItemDescription, { children: ({ description }) => {
            return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.itemDescription, customElement: children, customElementProps: { description }, content: description, ...otherProps, children: _jsx("p", { children: description }) }));
        } }));
});
/**
 * Displays the item price with customizable rendering following the documented API.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Item.Price className="text-lg font-bold text-primary" />
 *
 * // asChild with primitive
 * <Item.Price asChild>
 *   <span className="text-lg font-bold text-primary" />
 * </Item.Price>
 *
 * // asChild with react component
 * <Item.Price asChild>
 *   {React.forwardRef(({price, currency, formattedPrice, ...props}, ref) => (
 *     <span ref={ref} {...props} className="text-lg font-bold text-primary">
 *       {formattedPrice || `${currency} ${price}`}
 *     </span>
 *   ))}
 * </Item.Price>
 * ```
 */
export const Price = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CorePrice, { children: ({ price, formattedPrice }) => {
            return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.itemPrice, customElement: children, customElementProps: { price, formattedPrice }, content: formattedPrice || price, ...otherProps, children: _jsx("p", { children: formattedPrice || price }) }));
        } }));
});
/**
 * Displays item images intelligently based on available images.
 * - No images: returns null
 * - Single image: renders WixMediaImage
 * - Multiple images: renders MediaGallery
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Item.Images className="w-full h-48" />
 *
 * // With custom gallery controls styling
 * <Item.Images
 *   className="w-full h-48"
 *   previousClassName="absolute left-2 top-1/2 -translate-y-1/2 bg-black/50 text-white p-2 rounded-full hover:bg-black/70"
 *   nextClassName="absolute right-2 top-1/2 -translate-y-1/2 bg-black/50 text-white p-2 rounded-full hover:bg-black/70"
 *   indicatorClassName="absolute bottom-2 left-1/2 -translate-x-1/2 flex gap-1"
 *   previousIconClassName="text-white drop-shadow-sm"
 *   nextIconClassName="text-white drop-shadow-sm"
 * />
 *
 * // asChild with primitive
 * <Item.Images asChild>
 *   <div className="w-full h-48" />
 * </Item.Images>
 *
 * // asChild with custom component
 * <Item.Images asChild>
 *   {React.forwardRef(({images, altText, ...props}, ref) => (
 *     <div ref={ref} {...props} className="w-full h-48">
 *       {images.length > 0 && (
 *         images.length === 1 ? (
 *           <WixMediaImage media={{ image: images[0] }} alt={altText} />
 *         ) : (
 *           <MediaGallery.Root mediaGalleryServiceConfig={{ media: images.map(img => ({ image: img })) }}>
 *             <MediaGallery.Viewport />
 *             <MediaGallery.Previous />
 *             <MediaGallery.Next />
 *           </MediaGallery.Root>
 *         )
 *       )}
 *     </div>
 *   ))}
 * </Item.Images>
 *
 * // Custom render function
 * <Item.Images>
 *   {({ images, altText }) => (
 *     <div className="w-full h-48">
 *       {images.length > 0 && (
 *         images.length === 1 ? (
 *           <WixMediaImage media={{ image: images[0] }} alt={altText} />
 *         ) : (
 *           <MediaGallery.Root mediaGalleryServiceConfig={{ media: images.map(img => ({ image: img })) }}>
 *             <MediaGallery.Viewport />
 *             <MediaGallery.Previous />
 *             <MediaGallery.Next />
 *             <MediaGallery.Indicator />
 *           </MediaGallery.Root>
 *         )
 *       )}
 *     </div>
 *   )}
 * </Item.Images>
 * ```
 */
export const Images = React.forwardRef((props, ref) => {
    const { asChild, children, className, previousClassName, nextClassName, indicatorClassName, previousIconClassName, nextIconClassName, ...otherProps } = props;
    return (_jsx(CoreImages, { children: ({ images, altText }) => {
            if (asChild && children) {
                return children({ images, altText }, ref);
            }
            if (images.length === 0) {
                return null;
            }
            if (images.length === 1) {
                return (_jsx(WixMediaImage, { media: { image: images[0] }, alt: altText, "data-testid": TestIds.itemImages, className: className, ...otherProps }));
            }
            const mediaItems = images.map((image) => ({ image }));
            return (_jsxs(MediaGallery.Root, { mediaGalleryServiceConfig: { media: mediaItems }, children: [_jsx(MediaGallery.Viewport, { asChild: true, children: ({ src, alt }) => (_jsx(WixMediaImage, { media: { image: src }, alt: alt, className: className, "data-testid": TestIds.itemImages, ...otherProps })) }), _jsx(MediaGallery.Previous, { className: previousClassName, children: _jsx(ChevronLeftIcon, { className: previousIconClassName }) }), _jsx(MediaGallery.Next, { className: nextClassName, children: _jsx(ChevronRightIcon, { className: nextIconClassName }) }), _jsx(MediaGallery.Indicator, { className: indicatorClassName })] }));
        } }));
});
/**
 * Displays the item featured status with customizable rendering.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Item.Featured className="w-4 h-4 text-blue-500" />
 *
 * // asChild with primitive
 * <Item.Featured asChild>
 *   <StarFilledIcon className="w-4 h-4 text-blue-500" />
 * </Item.Featured>
 *
 * // asChild with react component
 * <Item.Featured asChild>
 *   {React.forwardRef(({featured, ...props}, ref) => (
 *     <StarFilledIcon ref={ref} {...props} className="w-4 h-4 text-blue-500" />
 *   ))}
 * </Item.Featured>
 *
 * // Custom render function
 * <Item.Featured>
 *   {({ featured }) => (
 *     <StarFilledIcon className="w-4 h-4 text-blue-500" />
 *   )}
 * </Item.Featured>
 * ```
 */
export const Featured = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreFeatured, { children: ({ featured }) => {
            if (!featured) {
                return null;
            }
            return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, "data-testid": TestIds.itemFeatured, className: className, ...otherProps, children: _jsx(StarFilledIcon, { className: className }) }));
        } }));
});
/**
 * Repeater component for rendering individual variants.
 *
 * @component
 * @example
 * ```tsx
 * <Item.VariantsRepeater>
 *   {({ variant, index }) => (
 *     <div key={variant._id} className="flex justify-between p-2 border rounded">
 *       <p className="font-medium">{variant.name}</p>
 *       <p className="text-primary">
 *         {variant.priceInfo?.formattedPrice || 'No price'}
 *       </p>
 *     </div>
 *   )}
 * </Item.VariantsRepeater>
 * ```
 */
export const VariantsRepeater = React.forwardRef((props, _ref) => {
    const { children } = props;
    const { item } = useItemContext();
    const variants = item.priceVariants || [];
    const hasVariants = !!(variants && variants.length > 0);
    if (!hasVariants) {
        return null;
    }
    return (_jsx(_Fragment, { children: variants.map((variant) => (_jsx(VariantComponent.Root, { variant: variant, "data-testid": TestIds.itemName, "data-variant-id": variant._id, children: children }, variant._id))) }));
});
Name.displayName = 'Item.Name';
Description.displayName = 'Item.Description';
Price.displayName = 'Item.Price';
Images.displayName = 'Item.Images';
VariantsRepeater.displayName = 'Item.VariantsRepeater';
export const LabelsRepeater = (props) => {
    const { children } = props;
    const { item } = useItemContext();
    const hasLabels = !!(item.labels && item.labels.length > 0);
    if (!hasLabels) {
        return null;
    }
    const itemLabels = item.labels;
    return (_jsx(_Fragment, { children: itemLabels.map((label) => (_jsx(LabelComponent.Root, { label: label, children: children }, label._id))) }));
};
export const ModifierGroupsRepeater = (props) => {
    const { children } = props;
    const { item } = useItemContext();
    const hasModifierGroups = !!(item.modifierGroups && item.modifierGroups.length > 0);
    if (!hasModifierGroups) {
        return null;
    }
    const itemModifierGroups = item.modifierGroups;
    return (_jsx(_Fragment, { children: itemModifierGroups.map((modifierGroup) => (_jsx(ModifierGroupComponent.Root, { modifierGroup: modifierGroup, children: children }, modifierGroup._id))) }));
};
LabelsRepeater.displayName = 'Item.LabelsRepeater';
ModifierGroupsRepeater.displayName = 'Item.ModifierGroupsRepeater';
/**
 * Item namespace containing all item components
 * following the compound component pattern: Item.Root, Item.Name, Item.Description, etc.
 */
export const Item = {
    /** Item root component */
    Root,
    /** Item name component */
    Name,
    /** Item description component */
    Description,
    /** Item price component */
    Price,
    /** Item images component (intelligent single/multiple) */
    Images,
    /** Item featured component */
    Featured,
    /** Item variants repeater component */
    VariantsRepeater,
    /** Item labels repeater component */
    LabelsRepeater,
    /** Item modifier groups repeater component */
    ModifierGroupsRepeater,
};
