// @wix/ricos/dist/cjs/plugin-ai/exports/image-caption-integration/caption-trigger.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var caption_trigger_exports = {};
__export(caption_trigger_exports, {
  createCaptionTrigger: () => createCaptionTrigger
});
module.exports = __toCommonJS(caption_trigger_exports);
var import_design_system = require("@wix/design-system");
var import_wix_ui_icons_common = require("@wix/wix-ui-icons-common");
var import_react = __toESM(require("react"));
var import_ricos_context = require("./../../../ricos-context/index.js");
var import_ai_plus_button = require("./../../ai-plus-button/ai-plus-button.js");
var import_ai_content_modal = require("../../ai-content-modal/ai-content-modal.js");
var import_data_hooks = require("../../data-hooks.js");
var import_types = require("../../types.js");
var import_get_image_url = require("../../utils/get-image-url.js");
const CaptionTrigger = ({ node, onCaptionChange, appDefId, openDiscardModal }) => {
  const modalsService = (0, import_react.useContext)(import_ricos_context.ModalServiceContext);
  const { localeContent, isMobile } = (0, import_react.useContext)(import_ricos_context.RicosContext);
  const [buttonRef, setButtonRef] = import_react.default.useState(null);
  const openAiModal = () => {
    const initialPrompt = {
      type: "prompt",
      title: "AIPlugin_Prompt_GenerateCaption",
      promptSlug: "generate-caption",
      promptParams: {
        imageUrl: (0, import_get_image_url.getImageUrl)(node.imageData),
        siteLanguage: localeContent
      },
      additionalTags: "image caption generation"
    };
    modalsService.togglePluginPanel({
      id: import_types.AI_MODAL_ID,
      Component: import_ai_content_modal.AiContentModal,
      componentProps: {
        closeModal: () => modalsService.closeModal(import_types.AI_MODAL_ID),
        openDiscardModal: () => openDiscardModal(modalsService, buttonRef),
        config: {
          configurePrompts: () => ({ selectionPrompts: [], floatingToolbarButtonPrompts: [] }),
          appDefId
        },
        feedUserEntryFormatter: (prompt, t) => {
          if (typeof prompt === "object" && "promptSlug" in prompt && prompt.promptSlug === initialPrompt.promptSlug) {
            return prompt.promptParams?.context ?? t(prompt.title);
          }
          return typeof prompt === "string" ? prompt : t(prompt.title);
        },
        initialPromptOptions: {
          emptyStateTitle: "AIPlugin_ImagePlugin_NoSelection_Title",
          emptyStateText: "AIPlugin_ImagePlugin_NoSelection_Text",
          placeholder: "AIPlugin_ImagePlugin_Input_Placeholder_Text",
          contextualResultActions: [
            {
              name: "insert",
              type: "action",
              title: "AIPlugin_Action_Insert",
              icon: import_wix_ui_icons_common.StatusCompleteSmall,
              onClick: async (entry) => {
                onCaptionChange(entry.response.plainContent);
                modalsService.closeModal(import_types.AI_MODAL_ID);
              }
            }
          ],
          prompt: initialPrompt,
          autoTrigger: false,
          promptPayloadFormatter: (prompt, textualPrompt) => {
            return {
              ...prompt,
              promptParams: {
                ...prompt.promptParams ?? {},
                context: textualPrompt
              }
            };
          },
          isSubmitButtonVisible: () => true
        }
      },
      positioning: {
        referenceElement: buttonRef,
        placement: "bottom-end"
      },
      onClickOutside: () => {
      },
      entryType: "image-caption",
      isMobile
    });
  };
  return /* @__PURE__ */ import_react.default.createElement(import_design_system.WixDesignSystemProvider, { features: { newColorsBranding: true } }, /* @__PURE__ */ import_react.default.createElement(
    import_ai_plus_button.AiPlusButton,
    {
      ref: setButtonRef,
      onClick: openAiModal,
      dataHook: import_data_hooks.AI_EXPORTS_CAPTION_TRIGGER_BUTTON
    }
  ));
};
const createCaptionTrigger = (appDefId, openDiscardModal) => {
  return (props) => /* @__PURE__ */ import_react.default.createElement(CaptionTrigger, { ...props, appDefId, openDiscardModal });
};
