// @wix/fast-gallery-vibe/dist/esm/components/GalleryWrapper.js
import * as React from 'react';
import { Gallery, MediaTypes, } from '@wix/fast-gallery-core';
import { clsx } from 'clsx';
import { galleryVariants } from './variants.js';
import './gallery.css';
export const GalleryWrapper = React.forwardRef((props, ref) => {
    const { items = [], variant = 'grid', className, itemRenderer, emptyState, children, ...otherProps } = props;
    const hasItems = items.length > 0;
    // Convert items to gallery format
    const convertItemToGalleryItem = React.useCallback((item, index) => {
        return {
            id: item.id || item._id || `item-${index}`,
            index: index.toString(),
            mediaUrl: '',
            type: MediaTypes.IMAGE,
            metaData: {
                width: 1,
                height: 1,
                title: '',
                description: '',
                alt: '',
                link: { url: '', target: '_self' },
            },
        };
    }, []);
    const galleryItems = React.useMemo(() => items.map((item, index) => convertItemToGalleryItem(item, index)), [items, convertItemToGalleryItem]);
    // Create custom item renderer component
    const ItemRendererComponent = React.useMemo(() => {
        const Component = (itemProps) => {
            if (itemRenderer) {
                const originalItem = items[parseInt(itemProps.data.index, 10)];
                if (originalItem) {
                    const rendered = itemRenderer(originalItem, parseInt(itemProps.data.index, 10));
                    return React.createElement(React.Fragment, null, rendered);
                }
            }
            // Development warning if no itemRenderer is provided
            if (process.env.NODE_ENV === 'development' && !itemRenderer) {
                console.warn('GalleryWrapper: No itemRenderer provided. Gallery will render empty items. ' +
                    'Please provide an itemRenderer prop to render your content.');
            }
            return null;
        };
        return Component;
    }, [itemRenderer, items]);
    const customRenderers = React.useMemo(() => ({
        itemRenderer: ItemRendererComponent,
    }), [ItemRendererComponent]);
    const finalClassName = clsx(galleryVariants({ variant }), className);
    // Show empty state if no items
    if (!hasItems) {
        return emptyState || null;
    }
    return (React.createElement("div", { ref: ref, className: finalClassName, "data-gallery-variant": variant, ...otherProps },
        React.createElement(Gallery, { dataList: galleryItems, customRenderers: customRenderers, options: {} }),
        children));
});
GalleryWrapper.displayName = 'GalleryWrapper';
