// @wix/auto_sdk_payments_onboarding-availability/build/es/meta.mjs
// src/cashier-onboarding-availability-v1-onboarding-availability-onboarding-availability.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixCashierOnboardingAvailabilityV1OnboardingAvailabilityServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/v1/onboarding-availability",
        destPath: "/v1/onboarding-availability"
      },
      {
        srcPath: "/ambassador/v1/onboarding-availability",
        destPath: "/v1/onboarding-availability"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_onboarding-availability";
function getOnboardingAvailability(payload) {
  function __getOnboardingAvailability({ host }) {
    const metadata = {
      entityFqdn: "wix.cashier.onboarding_availability.v1.onboarding_availability",
      method: "GET",
      methodFqn: "wix.cashier.onboarding_availability.v1.OnboardingAvailabilityService.GetOnboardingAvailability",
      packageName: PACKAGE_NAME,
      url: resolveWixCashierOnboardingAvailabilityV1OnboardingAvailabilityServiceUrl(
        { protoPath: "/v1/onboarding-availability", data: payload, host }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            {
              path: "onboardingAvailability.cbdFlow.attestationInfo.attestationFormSignedDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getOnboardingAvailability;
}
function updateCbdFlow(payload) {
  function __updateCbdFlow({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "cbdFlow.attestationInfo.attestationFormSignedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.cashier.onboarding_availability.v1.onboarding_availability",
      method: "PATCH",
      methodFqn: "wix.cashier.onboarding_availability.v1.OnboardingAvailabilityService.UpdateCbdFlow",
      packageName: PACKAGE_NAME,
      url: resolveWixCashierOnboardingAvailabilityV1OnboardingAvailabilityServiceUrl(
        {
          protoPath: "/v1/onboarding-availability/cbd-flow",
          data: serializedData,
          host
        }
      ),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            {
              path: "onboardingAvailability.cbdFlow.attestationInfo.attestationFormSignedDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateCbdFlow;
}
function updateRestrictedGoodsFlow(payload) {
  function __updateRestrictedGoodsFlow({ host }) {
    const metadata = {
      entityFqdn: "wix.cashier.onboarding_availability.v1.onboarding_availability",
      method: "PATCH",
      methodFqn: "wix.cashier.onboarding_availability.v1.OnboardingAvailabilityService.UpdateRestrictedGoodsFlow",
      packageName: PACKAGE_NAME,
      url: resolveWixCashierOnboardingAvailabilityV1OnboardingAvailabilityServiceUrl(
        {
          protoPath: "/v1/onboarding-availability/restricted-goods-flow",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            {
              path: "onboardingAvailability.cbdFlow.attestationInfo.attestationFormSignedDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateRestrictedGoodsFlow;
}
function updatePartnerFlow(payload) {
  function __updatePartnerFlow({ host }) {
    const metadata = {
      entityFqdn: "wix.cashier.onboarding_availability.v1.onboarding_availability",
      method: "PATCH",
      methodFqn: "wix.cashier.onboarding_availability.v1.OnboardingAvailabilityService.UpdatePartnerFlow",
      packageName: PACKAGE_NAME,
      url: resolveWixCashierOnboardingAvailabilityV1OnboardingAvailabilityServiceUrl(
        {
          protoPath: "/v1/onboarding-availability/partner-flow",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            {
              path: "onboardingAvailability.cbdFlow.attestationInfo.attestationFormSignedDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updatePartnerFlow;
}

// src/cashier-onboarding-availability-v1-onboarding-availability-onboarding-availability.meta.ts
function getOnboardingAvailability2() {
  const payload = {};
  const getRequestOptions = getOnboardingAvailability(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/onboarding-availability",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateCbdFlow2() {
  const payload = {};
  const getRequestOptions = updateCbdFlow(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/onboarding-availability/cbd-flow",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateRestrictedGoodsFlow2() {
  const payload = {};
  const getRequestOptions = updateRestrictedGoodsFlow(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/onboarding-availability/restricted-goods-flow",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updatePartnerFlow2() {
  const payload = {};
  const getRequestOptions = updatePartnerFlow(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/onboarding-availability/partner-flow",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  getOnboardingAvailability2 as getOnboardingAvailability,
  updateCbdFlow2 as updateCbdFlow,
  updatePartnerFlow2 as updatePartnerFlow,
  updateRestrictedGoodsFlow2 as updateRestrictedGoodsFlow
};
