// @wix/headless-events/cjs/dist/react/EventList.js
import { jsx as _jsx } from "react/jsx-runtime";
import { AsChildSlot } from '@wix/headless-utils/react';
import { Filter as FilterPrimitive } from '@wix/headless-components/react';
import React from 'react';
import * as CoreEventList from './core/EventList.js';
import * as Event from './Event.js';
var TestIds;
(function (TestIds) {
    TestIds["eventListEvents"] = "event-list-events";
    TestIds["eventListLoadMore"] = "event-list-load-more";
    TestIds["eventListError"] = "event-list-error";
})(TestIds || (TestIds = {}));
const DATA_COMPONENT_TAG = 'events.event-list';
/**
 * Root container that provides event list service context to all child components.
 * Must be used as the top-level component for event list functionality.
 *
 * @order 1
 * @component
 * @example
 * ```tsx
 * import { EventList } from '@wix/events/components';
 *
 * function EventListPage({ eventListServiceConfig }) {
 *   return (
 *     <EventList.Root eventListServiceConfig={eventListServiceConfig}>
 *       <EventList.Filters allCategoriesLabel="All">
 *         <Filter.FilterOptions>
 *           <Filter.FilterOptionRepeater>
 *             <Filter.FilterOption.SingleFilter />
 *           </Filter.FilterOptionRepeater>
 *         </Filter.FilterOptions>
 *       </EventList.Filters>
 *       <EventList.Events>
 *         <EventList.EventRepeater>
 *           <Event.Image />
 *           <Event.Title />
 *           <Event.Date />
 *           <Event.Location />
 *           <Event.Description />
 *           <Event.RsvpButton label="RSVP" />
 *         </EventList.EventRepeater>
 *       </EventList.Events>
 *     </EventList.Root>
 *   );
 * }
 * ```
 */
export const Root = (props) => {
    const { children, eventListServiceConfig } = props;
    return (_jsx(CoreEventList.Root, { eventListServiceConfig: eventListServiceConfig, children: children }));
};
/**
 * Container for the event list with support for empty state.
 * Follows List Container Level pattern.
 *
 * @component
 * @example
 * ```tsx
 * <EventList.Events emptyState={<div>No events found</div>}>
 *   <EventList.EventRepeater>
 *     <Event.Image />
 *     <Event.Title />
 *   </EventList.EventRepeater>
 * </EventList.Events>
 * ```
 */
export const Events = React.forwardRef((props, ref) => {
    const { asChild, children, emptyState, className, ...otherProps } = props;
    return (_jsx(CoreEventList.Events, { children: ({ events, hasEvents, isLoading }) => {
            if (!hasEvents && !isLoading) {
                return emptyState || null;
            }
            return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-component-tag": DATA_COMPONENT_TAG, "data-testid": TestIds.eventListEvents, "data-loading": isLoading, customElement: children, customElementProps: { events, isLoading }, ...otherProps, children: _jsx("div", { children: children }) }));
        } }));
});
/**
 * Repeater component that renders Event.Root for each event.
 * Follows Repeater Level pattern.
 * Note: Repeater components do NOT support asChild as per architecture rules.
 *
 * @component
 * @example
 * ```tsx
 * <EventList.EventRepeater>
 *   <Event.Image />
 *   <Event.Title />
 * </EventList.EventRepeater>
 * ```
 */
export const EventRepeater = (props) => {
    const { children, className } = props;
    return (_jsx(CoreEventList.EventRepeater, { children: ({ events }) => events.map((event) => (_jsx(Event.Root, { event: event, className: className, children: children }, event._id))) }));
};
/**
 * Displays a button to load more events. Not rendered if no events are left to load.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <EventList.LoadMoreTrigger className="bg-blue-600 hover:bg-blue-700 text-white" label="Load More" loadingState="Loading..." />
 *
 * // asChild with primitive
 * <EventList.LoadMoreTrigger asChild className="bg-blue-600 hover:bg-blue-700 text-white">
 *   <button>Load More</button>
 * </EventList.LoadMoreTrigger>
 *
 * // asChild with react component
 * <EventList.LoadMoreTrigger asChild className="bg-blue-600 hover:bg-blue-700 text-white">
 *   {React.forwardRef(({ isLoading, loadMoreEvents, ...props }, ref) => (
 *     <button ref={ref} {...props}>
 *       {isLoading ? 'Loading...' : 'Load More'}
 *     </button>
 *   ))}
 * </EventList.LoadMoreTrigger>
 * ```
 */
export const LoadMoreTrigger = React.forwardRef((props, ref) => {
    const { asChild, children, className, label, loadingState = label, ...otherProps } = props;
    return (_jsx(CoreEventList.LoadMoreTrigger, { children: ({ isLoading, loadMoreEvents }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.eventListLoadMore, customElement: children, customElementProps: { isLoading, loadMoreEvents }, disabled: isLoading, onClick: loadMoreEvents, ...otherProps, children: _jsx("button", { children: isLoading ? loadingState : label }) })) }));
});
/**
 * Displays an error message when the event list fails to load.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <EventList.Error className="text-red-500" />
 *
 * // asChild with primitive
 * <EventList.Error asChild className="text-red-500">
 *   <span />
 * </EventList.Error>
 *
 * // asChild with react component
 * <EventList.Error asChild className="text-red-500">
 *   {React.forwardRef(({ error, ...props }, ref) => (
 *     <span ref={ref} {...props}>
 *       {error}
 *     </span>
 *   ))}
 * </EventList.Error>
 * ```
 */
export const Error = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreEventList.Error, { children: ({ error }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.eventListError, customElement: children, customElementProps: { error }, content: error, ...otherProps, children: _jsx("span", { children: error }) })) }));
});
/**
 * Container for the event list category filters. Not rendered if there are no categories.
 *
 * @component
 * @example
 * ```tsx
 * <EventList.CategoryFilter allCategoriesLabel="All">
 *   <Filter.FilterOptions className="border-b border-gray-500 mb-6">
 *     <Filter.FilterOptionRepeater>
 *       <Filter.FilterOption.SingleFilter className="flex gap-2" />
 *     </Filter.FilterOptionRepeater>
 *   </Filter.FilterOptions>
 * </EventList.CategoryFilter>
 * ```
 */
export const CategoryFilter = React.forwardRef((props, ref) => {
    const { asChild, children, className, allCategoriesLabel, ...otherProps } = props;
    return (_jsx(CoreEventList.CategoryFilter, { allCategoriesLabel: allCategoriesLabel, children: ({ filterOptions, value, onChange }) => (_jsx(FilterPrimitive.Root, { ref: ref, asChild: asChild, className: className, filterOptions: filterOptions, value: value, onChange: onChange, ...otherProps, children: children })) }));
});
/**
 * Container for the event list status filters.
 *
 * @component
 * @example
 * ```tsx
 * <EventList.StatusFilter
 *   allEventsLabel="All Events"
 *   upcomingEventsLabel="Upcoming"
 *   pastEventsLabel="Past"
 * >
 *   <Filter.FilterOptions>
 *     <Filter.FilterOptionRepeater>
 *       <Filter.FilterOption.SingleFilter />
 *     </Filter.FilterOptionRepeater>
 *   </Filter.FilterOptions>
 * </EventList.StatusFilter>
 * ```
 */
export const StatusFilter = React.forwardRef((props, ref) => {
    const { asChild, children, className, allEventsLabel, upcomingEventsLabel, pastEventsLabel, ...otherProps } = props;
    return (_jsx(CoreEventList.StatusFilter, { allEventsLabel: allEventsLabel, upcomingEventsLabel: upcomingEventsLabel, pastEventsLabel: pastEventsLabel, children: ({ filterOptions, value, onChange }) => (_jsx(FilterPrimitive.Root, { ref: ref, asChild: asChild, className: className, filterOptions: filterOptions, value: value, onChange: onChange, ...otherProps, children: children })) }));
});
