// @wix/headless-events/cjs/dist/react/PricingOption.js
import { jsx as _jsx } from "react/jsx-runtime";
import { AsChildSlot } from '@wix/headless-utils/react';
import React from 'react';
import * as CorePricingOption from './core/PricingOption.js';
var TestIds;
(function (TestIds) {
    TestIds["pricingOptionRoot"] = "pricing-option-root";
    TestIds["pricingOptionName"] = "pricing-option-name";
    TestIds["pricingOptionPricing"] = "pricing-option-pricing";
    TestIds["pricingOptionTax"] = "pricing-option-tax";
    TestIds["pricingOptionFee"] = "pricing-option-fee";
    TestIds["pricingOptionQuantity"] = "pricing-option-quantity";
})(TestIds || (TestIds = {}));
/**
 * Root container that provides pricing option context to all child components.
 * Must be used as the top-level PricingOption component.
 *
 * @order 1
 * @component
 * @example
 * ```tsx
 * import { PricingOption } from '@wix/events/components';
 *
 * function PricingOptionComponent({ pricingOption }) {
 *   return (
 *     <PricingOption.Root pricingOption={pricingOption}>
 *       <PricingOption.Name />
 *       <PricingOption.Pricing />
 *       <PricingOption.Quantity />
 *     </PricingOption.Root>
 *   );
 * }
 * ```
 */
export const Root = React.forwardRef((props, ref) => {
    const { pricingOption, asChild, children, className, ...otherProps } = props;
    return (_jsx(CorePricingOption.Root, { pricingOption: pricingOption, children: _jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.pricingOptionRoot, customElement: children, customElementProps: {}, ...otherProps, children: _jsx("div", { children: children }) }) }));
});
/**
 * Displays the pricing option name.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <PricingOption.Name className="text-xl font-bold" />
 *
 * // asChild with primitive
 * <PricingOption.Name asChild>
 *   <h2 className="text-xl font-bold" />
 * </PricingOption.Name>
 *
 * // asChild with react component
 * <PricingOption.Name asChild>
 *   {React.forwardRef(({ name, ...props }, ref) => (
 *     <h2 ref={ref} {...props} className="text-xl font-bold">
 *       {name}
 *     </h2>
 *   ))}
 * </PricingOption.Name>
 * ```
 */
export const Name = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CorePricingOption.Name, { children: ({ name }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.pricingOptionName, customElement: children, customElementProps: { name }, content: name, ...otherProps, children: _jsx("span", { children: name }) })) }));
});
/**
 * Displays the pricing option price.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <PricingOption.Pricing className="text-lg font-semibold" />
 *
 * // asChild with primitive
 * <PricingOption.Pricing asChild>
 *   <span className="text-lg font-semibold" />
 * </PricingOption.Pricing>
 *
 * // asChild with react component
 * <PricingOption.Pricing asChild>
 *   {React.forwardRef(({ value, currency, formattedValue, ...props }, ref) => (
 *     <span ref={ref} {...props} className="text-lg font-semibold">
 *       {formattedValue}
 *     </span>
 *   ))}
 * </PricingOption.Pricing>
 * ```
 */
export const Pricing = React.forwardRef((props, ref) => {
    const { asChild, children, className, locale, ...otherProps } = props;
    return (_jsx(CorePricingOption.Pricing, { locale: locale, children: ({ value, currency, formattedValue }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.pricingOptionPricing, customElement: children, customElementProps: {
                value,
                currency,
                formattedValue,
            }, content: formattedValue, ...otherProps, children: _jsx("span", { children: formattedValue }) })) }));
});
/**
 * Displays the tax for the pricing option. Only renders when event has tax settings.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <PricingOption.Tax className="text-sm text-gray-500" />
 *
 * // asChild with primitive
 * <PricingOption.Tax asChild>
 *   <span className="text-sm text-gray-500" />
 * </PricingOption.Tax>
 *
 * // asChild with react component
 * <PricingOption.Tax asChild>
 *   {React.forwardRef(({ name, rate, included, taxableValue, taxValue, currency, formattedTaxValue, ...props }, ref) => (
 *     <span ref={ref} {...props} className="text-sm text-gray-500">
 *       {included ? `${name} included` : `+${formattedTaxValue} ${name}`}
 *     </span>
 *   ))}
 * </PricingOption.Tax>
 * ```
 */
export const Tax = React.forwardRef((props, ref) => {
    const { asChild, children, className, locale, ...otherProps } = props;
    return (_jsx(CorePricingOption.Tax, { locale: locale, children: ({ name, rate, included, taxableValue, taxValue, currency, formattedTaxValue, }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.pricingOptionTax, customElement: children, customElementProps: {
                name,
                rate,
                included,
                taxableValue,
                taxValue,
                currency,
                formattedTaxValue,
            }, content: formattedTaxValue, ...otherProps, children: _jsx("span", { children: formattedTaxValue }) })) }));
});
/**
 * Displays the fee for the pricing option. Only renders when ticket definition has fee enabled.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <PricingOption.Fee className="text-sm text-gray-500" />
 *
 * // asChild with primitive
 * <PricingOption.Fee asChild>
 *   <span className="text-sm text-gray-500" />
 * </PricingOption.Fee>
 *
 * // asChild with react component
 * <PricingOption.Fee asChild>
 *   {React.forwardRef(({ rate, value, currency, formattedValue, ...props }, ref) => (
 *     <span ref={ref} {...props} className="text-sm text-gray-500">
 *       +{formattedValue} service fee
 *     </span>
 *   ))}
 * </PricingOption.Fee>
 * ```
 */
export const Fee = React.forwardRef((props, ref) => {
    const { asChild, children, className, locale, ...otherProps } = props;
    return (_jsx(CorePricingOption.Fee, { locale: locale, children: ({ rate, value, currency, formattedValue }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.pricingOptionFee, customElement: children, customElementProps: {
                rate,
                value,
                currency,
                formattedValue,
            }, content: formattedValue, ...otherProps, children: _jsx("span", { children: formattedValue }) })) }));
});
/**
 * Displays a quantity selector for the pricing option. Only renders when the sale has started and the ticket definition is not sold out.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <PricingOption.Quantity className="border rounded px-3 py-2" />
 *
 * // asChild with primitive
 * <PricingOption.Quantity asChild>
 *   <select className="border rounded px-3 py-2" />
 * </PricingOption.Quantity>
 *
 * // asChild with react component
 * <PricingOption.Quantity asChild>
 *   {React.forwardRef(({ options, quantity, maxQuantity, increment, decrement, setQuantity, ...props }, ref) => (
 *     <div ref={ref} {...props} className="flex items-center space-x-2">
 *       <button disabled={quantity === 0} onClick={decrement}>-</button>
 *       <span>{quantity}</span>
 *       <button disabled={quantity >= maxQuantity} onClick={increment}>+</button>
 *     </div>
 *   ))}
 * </PricingOption.Quantity>
 * ```
 */
export const Quantity = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CorePricingOption.Quantity, { children: ({ options, quantity, maxQuantity, increment, decrement, setQuantity, }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.pricingOptionQuantity, customElement: children, customElementProps: {
                options,
                quantity,
                maxQuantity,
                increment,
                decrement,
                setQuantity,
            }, value: quantity, onChange: (event) => setQuantity(Number(event.target.value)), ...otherProps, children: _jsx("select", { children: options.map((option) => (_jsx("option", { value: option, children: option }, option))) }) })) }));
});
