// @wix/headless-events/cjs/dist/react/ScheduleItemTag.js
import { jsx as _jsx } from "react/jsx-runtime";
import { AsChildSlot } from '@wix/headless-utils/react';
import React from 'react';
import * as CoreTag from './core/ScheduleItemTag.js';
var TestIds;
(function (TestIds) {
    TestIds["scheduleItemTagRoot"] = "schedule-item-tag-root";
    TestIds["scheduleItemTagLabel"] = "schedule-item-tag-label";
    TestIds["scheduleItemTagButton"] = "schedule-item-tag-button";
})(TestIds || (TestIds = {}));
/**
 * Root container that provides schedule item tag context to all child components.
 * Must be used as the top-level ScheduleItemTag component.
 *
 * @order 1
 * @component
 * @example
 * ```tsx
 * import { ScheduleItemTag } from '@wix/events/components';
 *
 * function ScheduleItemTagComponent({ tag }) {
 *   return (
 *     <ScheduleItemTag.Root tag={tag}>
 *       <ScheduleItemTag.Label />
 *     </ScheduleItemTag.Root>
 *   );
 * }
 * ```
 */
export const Root = React.forwardRef((props, ref) => {
    const { tag, asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreTag.Root, { tag: tag, children: _jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.scheduleItemTagRoot, customElement: children, customElementProps: {}, ...otherProps, children: _jsx("div", { children: children }) }) }));
});
/**
 * Displays the schedule item tag label.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <ScheduleItemTag.Label className="px-2 py-1 bg-gray-100 rounded" />
 *
 * // asChild with primitive
 * <ScheduleItemTag.Label asChild>
 *   <span className="px-2 py-1 bg-gray-100 rounded" />
 * </ScheduleItemTag.Label>
 *
 * // asChild with react component
 * <ScheduleItemTag.Label asChild>
 *   {React.forwardRef(({ label, ...props }, ref) => (
 *     <span ref={ref} {...props} className="px-2 py-1 bg-gray-100 rounded">
 *       {label}
 *     </span>
 *   ))}
 * </ScheduleItemTag.Label>
 * ```
 */
export const Label = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreTag.Tag, { children: ({ tag }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.scheduleItemTagLabel, customElement: children, customElementProps: { label: tag }, content: tag, ...otherProps, children: _jsx("span", { children: tag }) })) }));
});
