// @wix/headless-events/cjs/dist/react/ScheduleItem.js
import { jsx as _jsx } from "react/jsx-runtime";
import { AsChildSlot } from '@wix/headless-utils/react';
import React from 'react';
import * as ScheduleItemTag from './ScheduleItemTag.js';
import * as CoreScheduleItem from './core/ScheduleItem.js';
var TestIds;
(function (TestIds) {
    TestIds["scheduleItemRoot"] = "schedule-item-root";
    TestIds["scheduleItemName"] = "schedule-item-name";
    TestIds["scheduleItemTimeSlot"] = "schedule-item-time-slot";
    TestIds["scheduleItemDuration"] = "schedule-item-duration";
    TestIds["scheduleItemDescription"] = "schedule-item-description";
    TestIds["scheduleItemStage"] = "schedule-item-stage";
    TestIds["scheduleItemTags"] = "schedule-item-tags";
})(TestIds || (TestIds = {}));
/**
 * Root container that provides schedule item service context to all child components.
 * Must be used as the top-level ScheduleItem component.
 *
 * @order 1
 * @component
 * @example
 * ```tsx
 * import { ScheduleItem } from '@wix/events/components';
 *
 * function ScheduleItemPage({ item }) {
 *   return (
 *     <ScheduleItem.Root item={item}>
 *       <ScheduleItem.Name />
 *       <ScheduleItem.TimeSlot />
 *       <ScheduleItem.Duration />
 *       <ScheduleItem.Description />
 *       <ScheduleItem.Stage />
 *     </ScheduleItem.Root>
 *   );
 * }
 * ```
 */
export const Root = React.forwardRef((props, ref) => {
    const { asChild, children, item, className, ...otherProps } = props;
    return (_jsx(CoreScheduleItem.Root, { item: item, children: _jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.scheduleItemRoot, "data-has-description": !!item.description, customElement: children, customElementProps: {}, ...otherProps, children: _jsx("div", { children: children }) }) }));
});
/**
 * Displays the schedule item name.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <ScheduleItem.Name className="font-bold text-lg" />
 *
 * // asChild with primitive
 * <ScheduleItem.Name asChild>
 *   <h2 className="font-bold text-lg" />
 * </ScheduleItem.Name>
 *
 * // asChild with react component
 * <ScheduleItem.Name asChild>
 *   {React.forwardRef(({ name, ...props }, ref) => (
 *     <h2 ref={ref} className="font-bold text-lg" {...props}>
 *       {name}
 *     </h2>
 *   ))}
 * </ScheduleItem.Name>
 * ```
 */
export const Name = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreScheduleItem.Name, { children: ({ name }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.scheduleItemName, customElement: children, customElementProps: { name }, content: name, ...otherProps, children: _jsx("span", { children: name }) })) }));
});
/**
 * Displays the schedule item time slot information.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <ScheduleItem.TimeSlot className="text-gray-600 font-medium" />
 *
 * // asChild with primitive
 * <ScheduleItem.TimeSlot asChild>
 *   <div className="text-gray-600" />
 * </ScheduleItem.TimeSlot>
 *
 * // asChild with react component
 * <ScheduleItem.TimeSlot asChild>
 *   {React.forwardRef(({ startTime, endTime, timeZoneId, formattedTimeRange, ...props }, ref) => (
 *     <time ref={ref} className="text-gray-600" {...props} dateTime={startTime.toISOString()}>
 *       {formattedTimeRange}
 *     </time>
 *   ))}
 * </ScheduleItem.TimeSlot>
 * ```
 */
export const TimeSlot = React.forwardRef((props, ref) => {
    const { asChild, children, className, locale, ...otherProps } = props;
    return (_jsx(CoreScheduleItem.TimeSlot, { locale: locale, children: ({ startTime, endTime, timeZoneId, formattedTimeRange }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.scheduleItemTimeSlot, customElement: children, customElementProps: {
                startTime,
                endTime,
                timeZoneId,
                formattedTimeRange,
            }, content: formattedTimeRange, ...otherProps, children: _jsx("span", { children: formattedTimeRange }) })) }));
});
/**
 * Displays the schedule item duration.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <ScheduleItem.Duration className="text-sm text-gray-500" />
 *
 * // asChild with primitive
 * <ScheduleItem.Duration asChild>
 *   <div className="text-sm text-gray-500" />
 * </ScheduleItem.Duration>
 *
 * // asChild with react component
 * <ScheduleItem.Duration asChild>
 *   {React.forwardRef(({ durationMinutes, ...props }, ref) => (
 *     <span ref={ref} className="text-sm text-gray-500" {...props}>
 *       {durationMinutes > 0 ? `${durationMinutes} minutes` : ''}
 *     </span>
 *   ))}
 * </ScheduleItem.Duration>
 * ```
 */
export const Duration = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreScheduleItem.Duration, { children: ({ durationMinutes }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.scheduleItemDuration, customElement: children, customElementProps: { durationMinutes }, content: durationMinutes, ...otherProps, children: _jsx("span", { children: durationMinutes }) })) }));
});
/**
 * Provides the schedule item description in rich text format, e.g. <p><strong>Description</strong></p>.
 *
 * @component
 * @example
 * ```tsx
 * // Usage with react component
 * // If using RicosViewer, use fromRichTextHtml from @wix/ricos to convert the description to RichContent
 * <ScheduleItem.Description>
 *   {React.forwardRef(({ description, ...props }, ref) => (
 *     <RicosViewer ref={ref} content={fromRichTextHtml(description)} />
 *   ))}
 * </ScheduleItem.Description>
 * ```
 */
export const Description = React.forwardRef((props, ref) => {
    const { children, ...otherProps } = props;
    return (_jsx(CoreScheduleItem.Description, { children: ({ description }) => (_jsx(AsChildSlot, { asChild: true, ref: ref, "data-testid": TestIds.scheduleItemDescription, customElement: children, customElementProps: { description }, ...otherProps })) }));
});
/**
 * Displays the schedule item stage.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <ScheduleItem.Stage className="text-blue-600 font-medium" />
 *
 * // asChild with primitive
 * <ScheduleItem.Stage asChild>
 *   <span className="text-blue-600 font-medium" />
 * </ScheduleItem.Stage>
 *
 * // asChild with react component
 * <ScheduleItem.Stage asChild>
 *   {React.forwardRef(({ stageName, ...props }, ref) => (
 *     <span ref={ref} className="text-blue-600 font-medium" {...props}>
 *       {stageName}
 *     </span>
 *   ))}
 * </ScheduleItem.Stage>
 * ```
 */
export const Stage = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreScheduleItem.Stage, { children: ({ stageName }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.scheduleItemStage, customElement: children, customElementProps: { stageName }, content: stageName, ...otherProps, children: _jsx("span", { children: stageName }) })) }));
});
/**
 * Container for the schedule item tags.
 *
 * @component
 * @example
 * ```tsx
 * <ScheduleItem.Tags>
 *   <ScheduleItem.TagRepeater>
 *     <ScheduleItemTag.Label />
 *   </ScheduleItem.TagRepeater>
 * </ScheduleItem.Tags>
 * ```
 */
export const Tags = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreScheduleItem.Tags, { children: ({ tags }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.scheduleItemTags, customElement: children, customElementProps: { tags }, ...otherProps, children: _jsx("div", { children: children }) })) }));
});
/**
 * Repeater component that renders ScheduleItemTag.Root for each tag.
 *
 * @component
 * @example
 * ```tsx
 * <ScheduleItem.TagRepeater>
 *   <ScheduleItemTag.Label />
 * </ScheduleItem.TagRepeater>
 * ```
 */
export const TagRepeater = (props) => {
    const { children, className } = props;
    return (_jsx(CoreScheduleItem.Tags, { children: ({ tags }) => tags.map((tag, index) => (_jsx(ScheduleItemTag.Root, { tag: tag, className: className, children: children }, index))) }));
};
