// @wix/headless-events/cjs/dist/react/TicketsPicker.js
import { jsx as _jsx } from "react/jsx-runtime";
import { AsChildSlot } from '@wix/headless-utils/react';
import React from 'react';
import * as TicketDefinition from './TicketDefinition.js';
import * as CoreTicketsPicker from './core/TicketsPicker.js';
var TestIds;
(function (TestIds) {
    TestIds["ticketsPickerTicketDefinitions"] = "tickets-picker-ticket-definitions";
    TestIds["ticketsPickerTotals"] = "tickets-picker-totals";
    TestIds["ticketsPickerCheckoutError"] = "tickets-picker-checkout-error";
    TestIds["ticketsPickerCheckoutTrigger"] = "tickets-picker-checkout-trigger";
})(TestIds || (TestIds = {}));
/**
 * Root container that provides necessary services context to all child components.
 * Must be used as the top-level component for tickets picker functionality.
 *
 * @order 1
 * @component
 * @example
 * ```tsx
 * import { TicketsPicker } from '@wix/events/components';
 *
 * function TicketsPickerComponent({ eventServiceConfig, ticketDefinitionListServiceConfig, checkoutServiceConfig }) {
 *   return (
 *     <TicketsPicker.Root
 *       eventServiceConfig={eventServiceConfig}
 *       ticketDefinitionListServiceConfig={ticketDefinitionListServiceConfig}
 *       checkoutServiceConfig={checkoutServiceConfig}
 *     >
 *       <TicketsPicker.TicketDefinitions>
 *         <TicketsPicker.TicketDefinitionRepeater>
 *           <TicketDefinition.Name />
 *           <TicketDefinition.Description />
 *           <TicketDefinition.FixedPricing />
 *           <TicketDefinition.GuestPricing />
 *           <TicketDefinition.PricingOptions>
 *             <TicketDefinition.PricingOptionRepeater>
 *               <PricingOption.Name />
 *               <PricingOption.Pricing />
 *               <PricingOption.Quantity />
 *             </TicketDefinition.PricingOptionRepeater>
 *           </TicketDefinition.PricingOptions>
 *           <TicketDefinition.Remaining />
 *           <TicketDefinition.SaleStartDate />
 *           <TicketDefinition.SaleEndDate />
 *           <TicketDefinition.Quantity />
 *         </TicketsPicker.TicketDefinitionRepeater>
 *       </TicketsPicker.TicketDefinitions>
 *     </TicketsPicker.Root>
 *   );
 * }
 * ```
 */
export const Root = (props) => {
    const { children, eventServiceConfig, ticketDefinitionListServiceConfig, checkoutServiceConfig, } = props;
    return (_jsx(CoreTicketsPicker.Root, { eventServiceConfig: eventServiceConfig, ticketDefinitionListServiceConfig: ticketDefinitionListServiceConfig, checkoutServiceConfig: checkoutServiceConfig, children: children }));
};
/**
 * Container for the ticket definition list with empty state support.
 * Follows List Container Level pattern.
 *
 * @component
 * @example
 * ```tsx
 * <TicketsPicker.TicketDefinitions emptyState={<div>No tickets found</div>}>
 *   <TicketsPicker.TicketDefinitionRepeater>
 *     <TicketDefinition.Name />
 *     <TicketDefinition.Description />
 *   </TicketsPicker.TicketDefinitionRepeater>
 * </TicketsPicker.TicketDefinitions>
 * ```
 */
export const TicketDefinitions = React.forwardRef((props, ref) => {
    const { asChild, children, emptyState, className, ...otherProps } = props;
    return (_jsx(CoreTicketsPicker.TicketDefinitions, { children: ({ ticketDefinitions, hasTicketDefinitions }) => {
            if (!hasTicketDefinitions) {
                return emptyState || null;
            }
            return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.ticketsPickerTicketDefinitions, customElement: children, customElementProps: { ticketDefinitions }, ...otherProps, children: _jsx("div", { children: children }) }));
        } }));
});
/**
 * Repeater component that renders TicketDefinition.Root for each ticket definition.
 * Follows Repeater Level pattern.
 * Note: Repeater components do NOT support asChild as per architecture rules.
 *
 * @component
 * @example
 * ```tsx
 * <TicketsPicker.TicketDefinitionRepeater>
 *   <TicketDefinition.Name />
 *   <TicketDefinition.Description />
 * </TicketsPicker.TicketDefinitionRepeater>
 * ```
 */
export const TicketDefinitionRepeater = (props) => {
    const { children, className } = props;
    return (_jsx(CoreTicketsPicker.TicketDefinitionRepeater, { children: ({ ticketDefinitions }) => ticketDefinitions.map((ticketDefinition) => (_jsx(TicketDefinition.Root, { ticketDefinition: ticketDefinition, className: className, children: children }, ticketDefinition._id))) }));
};
/**
 * Provides totals data for the tickets picker.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <TicketsPicker.Totals className="text-gray-500" />
 *
 * // asChild with primitive
 * <TicketsPicker.Totals asChild className="text-gray-500">
 *   <span />
 * </TicketsPicker.Totals>
 *
 * // asChild with react component
 * <TicketsPicker.Totals asChild className="text-gray-500">
 *   {React.forwardRef(({total, subtotal, tax, fee, currency, formattedTotal, formattedSubtotal, formattedTax, formattedFee, taxName, taxRate, taxIncluded, feeRate, ...props }, ref) => (
 *     <span ref={ref} {...props}>
 *       Subtotal: {formattedSubtotal}
 *       Tax: {formattedTax}
 *       Fee: {formattedFee}
 *       Total: {formattedTotal}
 *     </span>
 *   ))}
 * </TicketsPicker.Totals>
 * ```
 */
export const Totals = React.forwardRef((props, ref) => {
    const { asChild, children, className, locale, ...otherProps } = props;
    return (_jsx(CoreTicketsPicker.Totals, { locale: locale, children: ({ total, subtotal, tax, fee, currency, formattedTotal, formattedSubtotal, formattedTax, formattedFee, taxName, taxRate, taxIncluded, feeRate, }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.ticketsPickerTotals, customElement: children, customElementProps: {
                total,
                subtotal,
                tax,
                fee,
                currency,
                formattedTotal,
                formattedSubtotal,
                formattedTax,
                formattedFee,
                taxName,
                taxRate,
                taxIncluded,
                feeRate,
            }, content: formattedTotal, ...otherProps, children: _jsx("span", { children: formattedTotal }) })) }));
});
/**
 * Displays an error message when the checkout fails.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <TicketsPicker.CheckoutError className="text-red-500" />
 *
 * // asChild with primitive
 * <TicketsPicker.CheckoutError asChild className="text-red-500">
 *   <span />
 * </TicketsPicker.CheckoutError>
 *
 * // asChild with react component
 * <TicketsPicker.CheckoutError asChild className="text-red-500">
 *   {React.forwardRef(({ error, ...props }, ref) => (
 *     <span ref={ref} {...props}>
 *       {error}
 *     </span>
 *   ))}
 * </TicketsPicker.CheckoutError>
 * ```
 */
export const CheckoutError = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreTicketsPicker.CheckoutError, { children: ({ error }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.ticketsPickerCheckoutError, customElement: children, customElementProps: { error }, content: error, ...otherProps, children: _jsx("span", { children: error }) })) }));
});
/**
 * Displays a button for checkout functionality.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <TicketsPicker.CheckoutTrigger className="bg-blue-600 hover:bg-blue-700 text-white" label="Checkout" loadingState="Processing..." />
 *
 * // asChild with primitive
 * <TicketsPicker.CheckoutTrigger asChild className="bg-blue-600 hover:bg-blue-700 text-white">
 *   <button>Checkout</button>
 * </TicketsPicker.CheckoutTrigger>
 *
 * // asChild with react component
 * <TicketsPicker.CheckoutTrigger asChild className="bg-blue-600 hover:bg-blue-700 text-white">
 *   {React.forwardRef(({ isLoading, error, hasSelectedTicketDefinitions, checkout, ...props }, ref) => (
 *     <button ref={ref} {...props}>
 *       {isLoading ? 'Processing...' : 'Checkout'}
 *     </button>
 *   ))}
 * </TicketsPicker.CheckoutTrigger>
 * ```
 */
export const CheckoutTrigger = React.forwardRef((props, ref) => {
    const { asChild, children, className, label, loadingState = label, ...otherProps } = props;
    return (_jsx(CoreTicketsPicker.CheckoutTrigger, { children: ({ isLoading, error, hasSelectedTicketDefinitions, checkout }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.ticketsPickerCheckoutTrigger, "data-in-progress": isLoading, customElement: children, customElementProps: {
                isLoading,
                error,
                hasSelectedTicketDefinitions,
                checkout,
            }, disabled: isLoading, onClick: checkout, ...otherProps, children: _jsx("button", { children: isLoading ? loadingState : label }) })) }));
});
