// @wix/headless-events/cjs/dist/react/TicketDefinition.js
import { jsx as _jsx } from "react/jsx-runtime";
import { AsChildSlot } from '@wix/headless-utils/react';
import React from 'react';
import { isTicketDefinitionAvailable } from '../utils/ticket-definition.js';
import * as PricingOption from './PricingOption.js';
import * as CoreTicketDefinition from './core/TicketDefinition.js';
var TestIds;
(function (TestIds) {
    TestIds["ticketDefinitionRoot"] = "ticket-definition-root";
    TestIds["ticketDefinitionName"] = "ticket-definition-name";
    TestIds["ticketDefinitionDescription"] = "ticket-definition-description";
    TestIds["ticketDefinitionFixedPricing"] = "ticket-definition-fixed-pricing";
    TestIds["ticketDefinitionGuestPricing"] = "ticket-definition-guest-pricing";
    TestIds["ticketDefinitionPricingRange"] = "ticket-definition-pricing-range";
    TestIds["ticketDefinitionTax"] = "ticket-definition-tax";
    TestIds["ticketDefinitionFee"] = "ticket-definition-fee";
    TestIds["ticketDefinitionRemaining"] = "ticket-definition-remaining";
    TestIds["ticketDefinitionSaleStartDate"] = "ticket-definition-sale-start-date";
    TestIds["ticketDefinitionSaleEndDate"] = "ticket-definition-sale-end-date";
    TestIds["ticketDefinitionQuantity"] = "ticket-definition-quantity";
    TestIds["ticketDefinitionPricingOptions"] = "ticket-definition-pricing-options";
})(TestIds || (TestIds = {}));
/**
 * Root container that provides ticket definition context to all child components.
 * Must be used as the top-level TicketDefinition component.
 *
 * @order 1
 * @component
 * @example
 * ```tsx
 * import { TicketDefinition } from '@wix/events/components';
 *
 * function TicketDefinitionComponent({ ticketDefinition }) {
 *   return (
 *     <TicketDefinition.Root ticketDefinition={ticketDefinition}>
 *       <TicketDefinition.Name />
 *       <TicketDefinition.Description />
 *       <TicketDefinition.FixedPricing />
 *       <TicketDefinition.GuestPricing />
 *       <TicketDefinition.PricingOptions>
 *         <TicketDefinition.PricingOptionRepeater>
 *           <PricingOption.Name />
 *           <PricingOption.Pricing />
 *           <PricingOption.Quantity />
 *         </TicketDefinition.PricingOptionRepeater>
 *       </TicketDefinition.PricingOptions>
 *       <TicketDefinition.Remaining />
 *       <TicketDefinition.SaleStartDate />
 *       <TicketDefinition.SaleEndDate />
 *       <TicketDefinition.Quantity />
 *     </TicketDefinition.Root>
 *   );
 * }
 * ```
 */
export const Root = React.forwardRef((props, ref) => {
    const { ticketDefinition, asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreTicketDefinition.Root, { ticketDefinition: ticketDefinition, children: _jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.ticketDefinitionRoot, "data-sold-out": ticketDefinition.limitPerCheckout === 0, "data-sale-started": ticketDefinition.saleStatus === 'SALE_STARTED', "data-sale-ended": ticketDefinition.saleStatus === 'SALE_ENDED', "data-free": !!ticketDefinition.pricingMethod?.free, "data-fixed-pricing": !!ticketDefinition.pricingMethod?.fixedPrice, "data-guest-pricing": !!ticketDefinition.pricingMethod?.guestPrice, "data-pricing-options": !!ticketDefinition.pricingMethod?.pricingOptions, "data-has-description": !!ticketDefinition.description, "data-available": isTicketDefinitionAvailable(ticketDefinition), customElement: children, customElementProps: {}, ...otherProps, children: _jsx("div", { children: children }) }) }));
});
/**
 * Displays the ticket definition name.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <TicketDefinition.Name className="text-xl font-bold" />
 *
 * // asChild with primitive
 * <TicketDefinition.Name asChild>
 *   <h2 className="text-xl font-bold" />
 * </TicketDefinition.Name>
 *
 * // asChild with react component
 * <TicketDefinition.Name asChild>
 *   {React.forwardRef(({ name, ...props }, ref) => (
 *     <h2 ref={ref} {...props} className="text-xl font-bold">
 *       {name}
 *     </h2>
 *   ))}
 * </TicketDefinition.Name>
 * ```
 */
export const Name = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreTicketDefinition.Name, { children: ({ name }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.ticketDefinitionName, customElement: children, customElementProps: { name }, content: name, ...otherProps, children: _jsx("span", { children: name }) })) }));
});
/**
 * Displays the ticket definition description. Not rendered if there is no description.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <TicketDefinition.Description className="text-sm text-gray-600" />
 *
 * // asChild with primitive
 * <TicketDefinition.Description asChild>
 *   <p className="text-sm text-gray-600" />
 * </TicketDefinition.Description>
 *
 * // asChild with react component
 * <TicketDefinition.Description asChild>
 *   {React.forwardRef(({ description, ...props }, ref) => (
 *     <p ref={ref} {...props} className="text-sm text-gray-600">
 *       {description}
 *     </p>
 *   ))}
 * </TicketDefinition.Description>
 * ```
 */
export const Description = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreTicketDefinition.Description, { children: ({ description }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.ticketDefinitionDescription, customElement: children, customElementProps: { description }, content: description, ...otherProps, children: _jsx("span", { children: description }) })) }));
});
/**
 * Displays the fixed pricing for the ticket definition. Only renders when ticket definition has fixed price pricing method.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <TicketDefinition.FixedPricing className="text-lg font-semibold" />
 *
 * // asChild with primitive
 * <TicketDefinition.FixedPricing asChild>
 *   <span className="text-lg font-semibold" />
 * </TicketDefinition.FixedPricing>
 *
 * // asChild with react component
 * <TicketDefinition.FixedPricing asChild>
 *   {React.forwardRef(({ value, currency, formattedValue, free, ...props }, ref) => (
 *     <span ref={ref} {...props} className="text-lg font-semibold">
 *       {free ? 'Free' : `${formattedValue}`}
 *     </span>
 *   ))}
 * </TicketDefinition.FixedPricing>
 * ```
 */
export const FixedPricing = React.forwardRef((props, ref) => {
    const { asChild, children, className, locale, ...otherProps } = props;
    return (_jsx(CoreTicketDefinition.FixedPricing, { locale: locale, children: ({ value, currency, formattedValue, free }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.ticketDefinitionFixedPricing, customElement: children, customElementProps: {
                value,
                currency,
                formattedValue,
                free,
            }, content: formattedValue, ...otherProps, children: _jsx("span", { children: formattedValue }) })) }));
});
/**
 * Displays an input for guest pricing (pay-what-you-want). Only renders when ticket definition has guest price pricing method.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <TicketDefinition.GuestPricing className="border rounded px-3 py-2" />
 *
 * // asChild with primitive
 * <TicketDefinition.GuestPricing asChild>
 *   <input className="border rounded px-3 py-2" />
 * </TicketDefinition.GuestPricing>
 *
 * // asChild with react component
 * <TicketDefinition.GuestPricing asChild>
 *   {React.forwardRef(({ price, minPrice, currency, formattedMinPrice, setPrice, ...props }, ref) => (
 *     <input ref={ref} {...props} className="border rounded px-3 py-2" />
 *   ))}
 * </TicketDefinition.GuestPricing>
 * ```
 */
export const GuestPricing = React.forwardRef((props, ref) => {
    const { asChild, children, className, locale, ...otherProps } = props;
    return (_jsx(CoreTicketDefinition.GuestPricing, { locale: locale, children: ({ price, minPrice, currency, formattedMinPrice, setPrice }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.ticketDefinitionGuestPricing, customElement: children, customElementProps: {
                price,
                minPrice,
                currency,
                formattedMinPrice,
                setPrice,
            }, type: "number", placeholder: formattedMinPrice, value: price, min: minPrice, onChange: (event) => setPrice(event.target.value), ...otherProps, children: _jsx("input", {}) })) }));
});
/**
 * Displays the pricing range for the ticket definition. Only renders when ticket definition has pricing options.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <TicketDefinition.PricingRange className="text-sm text-gray-500" />
 *
 * // asChild with primitive
 * <TicketDefinition.PricingRange asChild>
 *   <span className="text-sm text-gray-500" />
 * </TicketDefinition.PricingRange>
 *
 * // asChild with react component
 * <TicketDefinition.PricingRange asChild>
 *   {React.forwardRef(({ minPrice, maxPrice, currency, formattedMinPrice, formattedMaxPrice, formattedPriceRange, ...props }, ref) => (
 *     <span ref={ref} {...props} className="text-sm text-gray-500">
 *       From {formattedMinPrice} to {formattedMaxPrice}
 *     </span>
 *   ))}
 * </TicketDefinition.PricingRange>
 * ```
 */
export const PricingRange = React.forwardRef((props, ref) => {
    const { asChild, children, className, locale, ...otherProps } = props;
    return (_jsx(CoreTicketDefinition.PricingRange, { locale: locale, children: ({ minPrice, maxPrice, currency, formattedMinPrice, formattedMaxPrice, formattedPriceRange, }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.ticketDefinitionPricingRange, customElement: children, customElementProps: {
                minPrice,
                maxPrice,
                currency,
                formattedMinPrice,
                formattedMaxPrice,
                formattedPriceRange,
            }, content: formattedPriceRange, ...otherProps, children: _jsx("span", { children: formattedPriceRange }) })) }));
});
/**
 * Displays the tax for the ticket definition. Only renders when event has tax settings, ticket definition is not free and has no pricing options, or ticket definition has guest pricing and tax is applied to donations.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <TicketDefinition.Tax className="text-sm text-gray-500" />
 *
 * // asChild with primitive
 * <TicketDefinition.Tax asChild>
 *   <span className="text-sm text-gray-500" />
 * </TicketDefinition.Tax>
 *
 * // asChild with react component
 * <TicketDefinition.Tax asChild>
 *   {React.forwardRef(({ name, rate, included, taxableValue, taxValue, currency, formattedTaxValue, ...props }, ref) => (
 *     <span ref={ref} {...props} className="text-sm text-gray-500">
 *       {included ? `${name} included` : `+${formattedTaxValue} ${name}`}
 *     </span>
 *   ))}
 * </TicketDefinition.Tax>
 * ```
 */
export const Tax = React.forwardRef((props, ref) => {
    const { asChild, children, className, locale, ...otherProps } = props;
    return (_jsx(CoreTicketDefinition.Tax, { locale: locale, children: ({ name, rate, included, taxableValue, taxValue, currency, formattedTaxValue, }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.ticketDefinitionTax, customElement: children, customElementProps: {
                name,
                rate,
                included,
                taxableValue,
                taxValue,
                currency,
                formattedTaxValue,
            }, content: formattedTaxValue, ...otherProps, children: _jsx("span", { children: formattedTaxValue }) })) }));
});
/**
 * Displays the fee for the ticket definition. Only renders when ticket definition has fee enabled, is not free and has no pricing options.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <TicketDefinition.Fee className="text-sm text-gray-500" />
 *
 * // asChild with primitive
 * <TicketDefinition.Fee asChild>
 *   <span className="text-sm text-gray-500" />
 * </TicketDefinition.Fee>
 *
 * // asChild with react component
 * <TicketDefinition.Fee asChild>
 *   {React.forwardRef(({ rate, value, currency, formattedValue, ...props }, ref) => (
 *     <span ref={ref} {...props} className="text-sm text-gray-500">
 *       +{formattedValue} service fee
 *     </span>
 *   ))}
 * </TicketDefinition.Fee>
 * ```
 */
export const Fee = React.forwardRef((props, ref) => {
    const { asChild, children, className, locale, ...otherProps } = props;
    return (_jsx(CoreTicketDefinition.Fee, { locale: locale, children: ({ rate, value, currency, formattedValue }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.ticketDefinitionFee, customElement: children, customElementProps: {
                rate,
                value,
                currency,
                formattedValue,
            }, content: formattedValue, ...otherProps, children: _jsx("span", { children: formattedValue }) })) }));
});
/**
 * Displays the remaining count.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <TicketDefinition.Remaining className="text-sm text-gray-500" />
 *
 * // asChild with primitive
 * <TicketDefinition.Remaining asChild>
 *   <span className="text-sm text-gray-500" />
 * </TicketDefinition.Remaining>
 *
 * // asChild with react component
 * <TicketDefinition.Remaining asChild>
 *   {React.forwardRef(({ remaining, ...props }, ref) => (
 *     <span ref={ref} {...props} className="text-sm text-gray-500">
 *       {remaining} remaining
 *     </span>
 *   ))}
 * </TicketDefinition.Remaining>
 * ```
 */
export const Remaining = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreTicketDefinition.Remaining, { children: ({ remaining }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.ticketDefinitionRemaining, customElement: children, customElementProps: { remaining }, content: remaining, ...otherProps, children: _jsx("span", { children: remaining }) })) }));
});
/**
 * Displays the sale start date. Only renders when the sale is scheduled to start in the future.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <TicketDefinition.SaleStartDate className="text-sm" />
 *
 * // asChild with primitive
 * <TicketDefinition.SaleStartDate asChild>
 *   <span className="text-sm" />
 * </TicketDefinition.SaleStartDate>
 *
 * // asChild with react component
 * <TicketDefinition.SaleStartDate asChild>
 *   {React.forwardRef(({ startDate, startDateFormatted, ...props }, ref) => (
 *     <span ref={ref} {...props} className="text-sm">
 *       Sale starts: {startDateFormatted}
 *     </span>
 *   ))}
 * </TicketDefinition.SaleStartDate>
 * ```
 */
export const SaleStartDate = React.forwardRef((props, ref) => {
    const { asChild, children, className, locale, ...otherProps } = props;
    return (_jsx(CoreTicketDefinition.SaleStartDate, { locale: locale, children: ({ startDate, startDateFormatted }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.ticketDefinitionSaleStartDate, customElement: children, customElementProps: { startDate, startDateFormatted }, content: startDateFormatted, ...otherProps, children: _jsx("span", { children: startDateFormatted }) })) }));
});
/**
 * Displays the sale end date. Only renders when the sale has started or ended.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <TicketDefinition.SaleEndDate className="text-sm text-red-600" />
 *
 * // asChild with primitive
 * <TicketDefinition.SaleEndDate asChild>
 *   <span className="text-sm text-red-600" />
 * </TicketDefinition.SaleEndDate>
 *
 * // asChild with react component
 * <TicketDefinition.SaleEndDate asChild>
 *   {React.forwardRef(({ endDate, endDateFormatted, saleEnded, ...props }, ref) => (
 *     <span ref={ref} {...props} className="text-sm text-red-600">
 *       {saleEnded ? 'Sale ended' : 'Sale ends'}: {endDateFormatted}
 *     </span>
 *   ))}
 * </TicketDefinition.SaleEndDate>
 * ```
 */
export const SaleEndDate = React.forwardRef((props, ref) => {
    const { asChild, children, className, locale, ...otherProps } = props;
    return (_jsx(CoreTicketDefinition.SaleEndDate, { locale: locale, children: ({ endDate, endDateFormatted, saleEnded }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.ticketDefinitionSaleEndDate, customElement: children, customElementProps: { endDate, endDateFormatted, saleEnded }, content: endDateFormatted, ...otherProps, children: _jsx("span", { children: endDateFormatted }) })) }));
});
/**
 * Displays a quantity selector for the ticket definition. Only renders when there are no pricing options, the sale has started, and the ticket definition is not sold out.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <TicketDefinition.Quantity className="border rounded px-3 py-2" />
 *
 * // asChild with primitive
 * <TicketDefinition.Quantity asChild>
 *   <select className="border rounded px-3 py-2" />
 * </TicketDefinition.Quantity>
 *
 * // asChild with react component
 * <TicketDefinition.Quantity asChild>
 *   {React.forwardRef(({ options, quantity, maxQuantity, increment, decrement, setQuantity, ...props }, ref) => (
 *     <div ref={ref} {...props} className="flex items-center space-x-2">
 *       <button disabled={quantity === 0} onClick={decrement}>-</button>
 *       <span>{quantity}</span>
 *       <button disabled={quantity >= maxQuantity} onClick={increment}>+</button>
 *     </div>
 *   ))}
 * </TicketDefinition.Quantity>
 * ```
 */
export const Quantity = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreTicketDefinition.Quantity, { children: ({ options, quantity, maxQuantity, increment, decrement, setQuantity, }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.ticketDefinitionQuantity, customElement: children, customElementProps: {
                options,
                quantity,
                maxQuantity,
                increment,
                decrement,
                setQuantity,
            }, value: quantity, onChange: (event) => setQuantity(Number(event.target.value)), ...otherProps, children: _jsx("select", { children: options.map((option) => (_jsx("option", { value: option, children: option }, option))) }) })) }));
});
/**
 * Container for pricing options. Only renders when there are pricing options available.
 * Follows List Container Level pattern.
 *
 * @component
 * @example
 * ```tsx
 * <TicketDefinition.PricingOptions>
 *   <TicketDefinition.PricingOptionRepeater>
 *     <PricingOption.Name />
 *     <PricingOption.Pricing />
 *     <PricingOption.Quantity />
 *   </TicketDefinition.PricingOptionRepeater>
 * </TicketDefinition.PricingOptions>
 * ```
 */
export const PricingOptions = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreTicketDefinition.PricingOptions, { children: ({ pricingOptions }) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.ticketDefinitionPricingOptions, customElement: children, customElementProps: { pricingOptions }, ...otherProps, children: _jsx("div", { children: children }) })) }));
});
/**
 * Repeater component that renders PricingOption.Root for each pricing option.
 * Follows Repeater Level pattern.
 * Note: Repeater components do NOT support asChild as per architecture rules.
 *
 * @component
 * @example
 * ```tsx
 * <TicketDefinition.PricingOptionRepeater>
 *   <PricingOption.Name />
 *   <PricingOption.Pricing />
 *   <PricingOption.Quantity />
 * </TicketDefinition.PricingOptionRepeater>
 * ```
 */
export const PricingOptionRepeater = (props) => {
    const { children, className } = props;
    return (_jsx(CoreTicketDefinition.PricingOptionRepeater, { children: ({ pricingOptions }) => pricingOptions.map((pricingOption) => (_jsx(PricingOption.Root, { pricingOption: pricingOption, className: className, children: children }, pricingOption.optionId))) }));
};
