// @wix/headless-node/dist/cjs/host-module.js
"use strict";

exports.__esModule = true;
exports.createHostModule = void 0;
var _monitoringVelo = require("@wix/monitoring-velo");
var _headers = require("./headers");
var _authUtils = require("./utils/auth-utils");
const createHostModule = () => {
  return {
    __type: 'host',
    create: () => {
      return {};
    },
    host: ({
      req,
      waitUntil,
      essentials
    } = {}) => {
      const passThroughHeaders = (0, _headers.getPassThroughHeaders)(req);
      return {
        essentials: {
          passThroughHeaders,
          language: (essentials == null ? void 0 : essentials.language) ?? 'en',
          locale: (essentials == null ? void 0 : essentials.locale) ?? 'en-US',
          timezone: (essentials == null ? void 0 : essentials.timezone) ?? 'UTC'
        },
        getMonitoringClient: () => {
          return (0, _monitoringVelo.createVeloMonitoringClient)({
            req: req,
            namespace: 'Headless',
            producer: 'backend',
            getAuthToken: async () => (0, _authUtils.resolveAuthToken)(),
            onBatchQueueCreated: batchFlushPromise => waitUntil == null ? void 0 : waitUntil(batchFlushPromise)
          });
        }
      };
    }
  };
};
exports.createHostModule = createHostModule;
