// @wix/auto_sdk_restaurants_order-pacing-order/build/internal/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  getCurrentDelay: () => getCurrentDelay4,
  getCurrentDelayWithValidityPeriod: () => getCurrentDelayWithValidityPeriod4,
  listBusyBlocks: () => listBusyBlocks4
});
module.exports = __toCommonJS(index_exports);

// src/restaurants-order-pacing-v1-order-order-pacing-order.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/restaurants-order-pacing-v1-order-order-pacing-order.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressRestaurantsOrderPacingOrderPacingServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-order-pacing-pacer",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-order-pacing-pacer",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-order-pacing-pacer",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/order-pacing-pacer",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/v1/order-pacing-pacer",
        destPath: "/v1"
      },
      {
        srcPath: "/restaurants/order-pacing-pacer/v1",
        destPath: "/v1"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-order-pacing-pacer",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/order-pacing-pacer/v1",
        destPath: "/v1"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/order-pacing-pacer/v1",
        destPath: "/v1"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/order-pacing-pacer/v1",
        destPath: "/v1"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_order-pacing-order";
function getCurrentDelay(payload) {
  function __getCurrentDelay({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.order_pacing.v1.order",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.order.pacing.OrderPacingService.GetCurrentDelay",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsOrderPacingOrderPacingServiceUrl({
        protoPath: "/v1/current-delay",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getCurrentDelay;
}
function getCurrentDelayWithValidityPeriod(payload) {
  function __getCurrentDelayWithValidityPeriod({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.order_pacing.v1.order",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.order.pacing.OrderPacingService.GetCurrentDelayWithValidityPeriod",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsOrderPacingOrderPacingServiceUrl({
        protoPath: "/v1/current-delay-with-validity-period",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getCurrentDelayWithValidityPeriod;
}
function listBusyBlocks(payload) {
  function __listBusyBlocks({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "from" }, { path: "until" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.order_pacing.v1.order",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.order.pacing.OrderPacingService.ListBusyBlocks",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsOrderPacingOrderPacingServiceUrl({
        protoPath: "/v1/busy-blocks",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "busyBlocks.startTime" },
            { path: "busyBlocks.endTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listBusyBlocks;
}

// src/restaurants-order-pacing-v1-order-order-pacing-order.universal.ts
async function getCurrentDelay2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    locationId: options?.locationId,
    maxAcceptableDelayTimeInMinutes: options?.maxAcceptableDelayTimeInMinutes
  });
  const reqOpts = getCurrentDelay(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          locationId: "$[0].locationId",
          maxAcceptableDelayTimeInMinutes: "$[0].maxAcceptableDelayTimeInMinutes"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getCurrentDelayWithValidityPeriod2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    locationId: options?.locationId,
    maxAcceptableDelayTimeInMinutes: options?.maxAcceptableDelayTimeInMinutes
  });
  const reqOpts = getCurrentDelayWithValidityPeriod(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          locationId: "$[0].locationId",
          maxAcceptableDelayTimeInMinutes: "$[0].maxAcceptableDelayTimeInMinutes"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listBusyBlocks2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    locationId: options?.locationId,
    from: options?.from,
    until: options?.until,
    orderFulfillmentTimeInMinutes: options?.orderFulfillmentTimeInMinutes
  });
  const reqOpts = listBusyBlocks(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          locationId: "$[0].locationId",
          from: "$[0].from",
          until: "$[0].until",
          orderFulfillmentTimeInMinutes: "$[0].orderFulfillmentTimeInMinutes"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/restaurants-order-pacing-v1-order-order-pacing-order.public.ts
function getCurrentDelay3(httpClient) {
  return (options) => getCurrentDelay2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function getCurrentDelayWithValidityPeriod3(httpClient) {
  return (options) => getCurrentDelayWithValidityPeriod2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function listBusyBlocks3(httpClient) {
  return (options) => listBusyBlocks2(
    options,
    // @ts-ignore
    { httpClient }
  );
}

// src/restaurants-order-pacing-v1-order-order-pacing-order.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var getCurrentDelay4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getCurrentDelay3);
var getCurrentDelayWithValidityPeriod4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getCurrentDelayWithValidityPeriod3);
var listBusyBlocks4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(listBusyBlocks3);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getCurrentDelay,
  getCurrentDelayWithValidityPeriod,
  listBusyBlocks
});
