// @wix/auto_sdk_restaurants_order-pacing-order/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  getCurrentDelay: () => getCurrentDelay2,
  getCurrentDelayWithValidityPeriod: () => getCurrentDelayWithValidityPeriod2,
  listBusyBlocks: () => listBusyBlocks2
});
module.exports = __toCommonJS(meta_exports);

// src/restaurants-order-pacing-v1-order-order-pacing-order.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressRestaurantsOrderPacingOrderPacingServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-order-pacing-pacer",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-order-pacing-pacer",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-order-pacing-pacer",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/order-pacing-pacer",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/v1/order-pacing-pacer",
        destPath: "/v1"
      },
      {
        srcPath: "/restaurants/order-pacing-pacer/v1",
        destPath: "/v1"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-order-pacing-pacer",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/order-pacing-pacer/v1",
        destPath: "/v1"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/order-pacing-pacer/v1",
        destPath: "/v1"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/order-pacing-pacer/v1",
        destPath: "/v1"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_order-pacing-order";
function getCurrentDelay(payload) {
  function __getCurrentDelay({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.order_pacing.v1.order",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.order.pacing.OrderPacingService.GetCurrentDelay",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsOrderPacingOrderPacingServiceUrl({
        protoPath: "/v1/current-delay",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getCurrentDelay;
}
function getCurrentDelayWithValidityPeriod(payload) {
  function __getCurrentDelayWithValidityPeriod({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.order_pacing.v1.order",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.order.pacing.OrderPacingService.GetCurrentDelayWithValidityPeriod",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsOrderPacingOrderPacingServiceUrl({
        protoPath: "/v1/current-delay-with-validity-period",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getCurrentDelayWithValidityPeriod;
}
function listBusyBlocks(payload) {
  function __listBusyBlocks({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "from" }, { path: "until" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.order_pacing.v1.order",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.order.pacing.OrderPacingService.ListBusyBlocks",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsOrderPacingOrderPacingServiceUrl({
        protoPath: "/v1/busy-blocks",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "busyBlocks.startTime" },
            { path: "busyBlocks.endTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listBusyBlocks;
}

// src/restaurants-order-pacing-v1-order-order-pacing-order.meta.ts
function getCurrentDelay2() {
  const payload = {};
  const getRequestOptions = getCurrentDelay(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/current-delay",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getCurrentDelayWithValidityPeriod2() {
  const payload = {};
  const getRequestOptions = getCurrentDelayWithValidityPeriod(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/current-delay-with-validity-period",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listBusyBlocks2() {
  const payload = {};
  const getRequestOptions = listBusyBlocks(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/busy-blocks",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getCurrentDelay,
  getCurrentDelayWithValidityPeriod,
  listBusyBlocks
});
