// @wix/patterns/dist/cjs/components/FileCard/FileCardList.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.FileCardList = void 0;
var _react = _interopRequireDefault(require("react"));
var _FilePreview = require("./FilePreview");
var _designSystem = require("@wix/design-system");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/FileCard/FileCardList.tsx";
const FileCardList = ({
  files,
  showDownloadCTA,
  showDetails
}) => {
  if (!files) {
    return null;
  }
  return /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
    direction: "vertical",
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 21,
      columnNumber: 5
    }
  }, files.map(file => /*#__PURE__*/_react.default.createElement(_FilePreview.FilePreview, {
    file: file,
    key: file.mediaId,
    showDownloadCTA: showDownloadCTA,
    showDetails: showDetails,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 23,
      columnNumber: 9
    }
  })));
};
exports.FileCardList = FileCardList;
