// @wix/patterns/dist/cjs/components/FileCard/FileCard.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.FileCard = void 0;
var _react = _interopRequireDefault(require("react"));
var _designSystem = require("@wix/design-system");
var _Thumbnail = require("./Thumbnail");
var _FileDetails = require("./FileDetails");
var _DownloadButton = require("./DownloadButton");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/FileCard/FileCard.tsx";
const FileCard = ({
  file,
  showDownloadCTA = false,
  showDetails = false
}) => {
  // const downloadIconWidth = showDownloadCTA ? 36 : 0;

  return /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
    verticalAlign: "middle",
    align: "left",
    gap: showDetails ? 'SP2' : 'SP1',
    paddingBlockEnd: 1,
    maxWidth: "100%",
    width: "100%",
    height: "100%",
    dataHook: `file-card-${file.mediaId}`,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 22,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(_Thumbnail.Thumbnail, {
    file: file,
    showDetails: showDetails,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 32,
      columnNumber: 7
    }
  }), /*#__PURE__*/_react.default.createElement(_FileDetails.FileDetails, {
    file: file,
    showDetails: showDetails,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 33,
      columnNumber: 7
    }
  }), showDownloadCTA && !file.private ? /*#__PURE__*/_react.default.createElement(_DownloadButton.DownloadButton, {
    file: file,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 34,
      columnNumber: 43
    }
  }) : null);
};
exports.FileCard = FileCard;
