// @wix/patterns/dist/cjs/components/BulkActionButton/BulkActionButton.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.BulkActionButton = BulkActionButton;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _designSystem = require("@wix/design-system");
var _react = _interopRequireDefault(require("react"));
var _DisabledTooltip = require("../DisabledTooltip");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/BulkActionButton/BulkActionButton.tsx";
function BulkActionButton(props) {
  const {
    tooltipText,
    tooltipProps,
    ...rest
  } = props;
  return /*#__PURE__*/_react.default.createElement(_DisabledTooltip.DisabledTooltip, (0, _extends2.default)({
    content: tooltipText,
    dataHook: "bulk-action-button-tooltip"
  }, tooltipProps, {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 13,
      columnNumber: 5
    }
  }), /*#__PURE__*/_react.default.createElement(_designSystem.Button, (0, _extends2.default)({
    dataHook: "bulk-action-button",
    skin: "standard",
    priority: "secondary"
  }, rest, {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 18,
      columnNumber: 7
    }
  })));
}
