// @wix/patterns/dist/cjs/components/MoreActions/ExploreAppsModal.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.ExploreAppsModal = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactModuleContainer = require("react-module-container");
var _mobxReactLite = require("mobx-react-lite");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/MoreActions/ExploreAppsModal.tsx";
function _ExploreAppsModal({
  state
}) {
  if (!state.isModalOpen) {
    return null;
  }
  const InContextAppsModal = _reactModuleContainer.ModuleRegistry.component('in-context-apps-modal');
  return /*#__PURE__*/_react.default.createElement(InContextAppsModal, {
    title: state.modalProps.title,
    subTitle: state.modalProps.subtitle,
    tagIds: [state.modalProps.tag],
    origin: "patterns-more-action",
    dataHook: "explore-apps-modal",
    isOpen: state.isModalOpen,
    onClose: () => state.closeExploreAppsModal(),
    onInstallationSuccess: ({
      appId
    }) => state.onInstallationSuccess({
      appId
    }),
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 12,
      columnNumber: 5
    }
  });
}
const ExploreAppsModal = exports.ExploreAppsModal = (0, _mobxReactLite.observer)(_ExploreAppsModal);
