// @wix/ricos/dist/cjs/plugin-code-block/code-block-converter.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var code_block_converter_exports = {};
__export(code_block_converter_exports, {
  codeBlockConverter: () => codeBlockConverter
});
module.exports = __toCommonJS(code_block_converter_exports);
var import_v1 = require("./../ricos-schema/legacy/wix/rich_content/v1/index.js");
const codeBlockConverter = {
  toPm: {
    types: [import_v1.Node_Type.CODE_BLOCK],
    // @ts-ignore $TSFixMe
    convert: (node, visit) => ({
      type: import_v1.Node_Type.CODE_BLOCK,
      attrs: {
        ...node.codeBlockData,
        ...node.style ? { style: node.style } : {},
        id: node.id
      },
      content: visit(node)
    })
  },
  fromPm: {
    types: [import_v1.Node_Type.CODE_BLOCK],
    // @ts-ignore $TSFixMe
    convert: (node, visit) => {
      const { id, style, ...data } = node.attrs || {};
      return {
        type: import_v1.Node_Type.CODE_BLOCK,
        id,
        nodes: visit(node),
        ...style ? { style } : {},
        codeBlockData: {
          ...data
        }
      };
    }
  }
};
