// @wix/patterns/dist/cjs/components/CounterBadgeButton/CounterBadgeButton.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.CounterBadgeButton = void 0;
var _react = _interopRequireDefault(require("react"));
var _designSystem = require("@wix/design-system");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CounterBadgeButton/CounterBadgeButton.tsx";
const CounterBadgeButton = props => {
  const {
    children,
    count = 0,
    dataHook = 'counter-badge-count'
  } = props;
  return /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
    position: "relative",
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 14,
      columnNumber: 5
    }
  }, children, count > 0 && /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
    position: "absolute",
    top: 3,
    right: 3,
    transform: "translate3d(50%, -50%, 0)",
    pointerEvents: "none",
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 17,
      columnNumber: 9
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.CounterBadge, {
    dataHook: dataHook,
    skin: "neutralStandard",
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 24,
      columnNumber: 11
    }
  }, count)));
};
exports.CounterBadgeButton = CounterBadgeButton;
