// @wix/ricos/dist/es/plugin-table/toolbar/buttons/BackgroundColorButton.js
import { ColorBucket } from "@wix/wix-ui-icons-common";
import React, { useContext, useState } from "react";
import { RicosContext } from "./../../../ricos-context/index.js";
import { ToolbarButton } from "./../../../_shared/toolbar-button/toolbar-button.js";
import { useColorPickerModal } from "../../modals/use-color-picker-modal.js";
const BackgroundColorButton = ({ dataHook, toolbarItem }) => {
  const { t } = useContext(RicosContext);
  const [referenceElement, setReferenceElement] = useState(null);
  const colorPickerModal = useColorPickerModal();
  return /* @__PURE__ */ React.createElement(
    ToolbarButton,
    {
      ref: setReferenceElement,
      prefix: ColorBucket,
      onClick: () => colorPickerModal.toggleModal(referenceElement, {
        nodeId: toolbarItem?.attributes?.selectedNode?.attrs?.id,
        color: toolbarItem.attributes.backgroundColor,
        onChange: (color) => toolbarItem.commands.click({ backgroundColor: color })
      }),
      dataHook,
      tooltip: t("TablePlugin_Toolbar_BGColor_Tooltip"),
      tabIndex: toolbarItem.presentation.tabIndex
    }
  );
};
export {
  BackgroundColorButton
};
