// @wix/ricos/dist/es/plugin-table/toolbar/buttons/BorderButton.js
import { AllBorders, SquareRatio } from "@wix/wix-ui-icons-common";
import React, { useContext, useState } from "react";
import { RicosContext } from "./../../../ricos-context/index.js";
import { DropdownButton } from "./../../../_shared/dropdown-button.js";
import { ToolbarButton } from "./../../../_shared/toolbar-button/toolbar-button.js";
import { BORDER_TYPES, CATEGORY } from "../../consts.js";
import { useColorPickerModal } from "../../modals/use-color-picker-modal.js";
import { useBorderPanel } from "../modals/use-border-panel.js";
const BorderButton = ({ dataHook, toolbarItem }) => {
  const { t } = useContext(RicosContext);
  const [referenceElement, setReferenceElement] = useState(null);
  const isMultipleSelection = !(toolbarItem.attributes?.selectedCategory === CATEGORY.CELL_BORDER);
  const colorPickerModal = useColorPickerModal();
  const borderPanel = useBorderPanel({
    onClick: (borderType) => {
      toggleColorPickerModal(borderType);
    },
    getSelectedStyle: () => ""
  });
  const toggleColorPickerModal = (borderType) => {
    colorPickerModal.toggleModal(referenceElement, {
      nodeId: toolbarItem.attributes?.selectedNode?.attrs?.id,
      color: toolbarItem.attributes?.cellBorderColor,
      onChange: (color) => toolbarItem.commands?.click({ type: borderType, color })
    });
  };
  return isMultipleSelection ? /* @__PURE__ */ React.createElement(
    DropdownButton,
    {
      dataHook,
      active: borderPanel.isModalOpen,
      onClick: () => borderPanel.toggleModal(referenceElement),
      ref: setReferenceElement,
      icon: AllBorders,
      tooltip: t("TablePlugin_Toolbar_Border_Color"),
      tabIndex: toolbarItem.presentation.tabIndex
    }
  ) : /* @__PURE__ */ React.createElement(
    ToolbarButton,
    {
      ref: setReferenceElement,
      prefix: SquareRatio,
      onClick: () => {
        toggleColorPickerModal(BORDER_TYPES.borders);
      },
      dataHook,
      tooltip: t("TablePlugin_Toolbar_Border_Color"),
      tabIndex: toolbarItem.presentation.tabIndex
    }
  );
};
export {
  BorderButton
};
