// @wix/design-system/dist/cjs/AnnouncementModalLayout/AnnouncementModalLayout.st.css.js

require("../Foundation/stylable/spacing.st.css.js");
require("../BaseModalLayout/BaseModalLayout.st.css.js");
require("../Heading/Heading.st.css.js");
require("../Text/Text.st.css.js");
require("../Foundation/stylable/typography.st.css.js");
require("../Foundation/stylable/border.st.css.js");

const { classesRuntime, statesRuntime } = require("./../stylable-cjs-runtime.js");



var _namespace_ = "wds_1_226_0_AnnouncementModalLayout";
var _style_ = classesRuntime.bind(null, _namespace_);

module.exports.cssStates = statesRuntime.bind(null, _namespace_);
module.exports.style = _style_;
module.exports.st = _style_;

module.exports.namespace = _namespace_;
module.exports.classes = {"root":"wds_1_226_0_AnnouncementModalLayout__root","announcementModalLayout":"wds_1_226_0_AnnouncementModalLayout__announcementModalLayout","link":"wds_1_226_0_AnnouncementModalLayout__link","bottomSpacing":"wds_1_226_0_AnnouncementModalLayout__bottomSpacing"};
module.exports.keyframes = {}; 
module.exports.layers = {};
module.exports.containers = {};
module.exports.stVars = {}; 
module.exports.vars = {"wsr-heading-font-size-m":"--wsr-heading-font-size-m","wsr-heading-font-size-xl":"--wsr-heading-font-size-xl","wsr-text-line-height-medium":"--wsr-text-line-height-medium","wds-announcement-modal-actions-padding":"--wds-announcement-modal-actions-padding","wds-announcement-modal-border-radius":"--wds-announcement-modal-border-radius","wds-announcement-modal-content-gap":"--wds-announcement-modal-content-gap","wds-announcement-modal-padding-horizontal":"--wds-announcement-modal-padding-horizontal","wds-announcement-modal-padding-vertical":"--wds-announcement-modal-padding-vertical","wds-announcement-modal-text-font-line-height":"--wds-announcement-modal-text-font-line-height","wds-announcement-modal-text-font-size":"--wds-announcement-modal-text-font-size","wds-announcement-modal-title-font-line-height":"--wds-announcement-modal-title-font-line-height","wds-announcement-modal-title-font-size":"--wds-announcement-modal-title-font-size","wds-space-200":"--wds-space-200","wds-space-500":"--wds-space-500","wds-space-700":"--wds-space-700","wds-space-800":"--wds-space-800","wds-space-1300":"--wds-space-1300"};
