// @wix/auto_sdk_restaurants_item-modifier-groups/build/internal/es/index.typings.mjs
// src/restaurants-menus-v1-item-modifier-group-item-modifier-groups.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/restaurants-menus-v1-item-modifier-group-item-modifier-groups.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl(opts) {
  const domainToMappings = {
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-modifier-group",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers-groups/delete",
        destPath: "/v1/bulk/modifiers-groups/delete"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers-groups/update",
        destPath: "/v1/bulk/modifiers-groups/update"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifier-groups/create",
        destPath: "/v1/bulk/modifier-groups/create"
      },
      {
        srcPath: "/restaurants/menus/v1/modifier-groups",
        destPath: "/v1/modifier-groups"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-modifier-group",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-modifier-group",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-modifier-group",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-modifier-group",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-modifier-group",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-modifier-group",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers-groups/delete",
        destPath: "/v1/bulk/modifiers-groups/delete"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers-groups/update",
        destPath: "/v1/bulk/modifiers-groups/update"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifier-groups/create",
        destPath: "/v1/bulk/modifier-groups/create"
      },
      {
        srcPath: "/restaurants/menus/v1/modifier-groups",
        destPath: "/v1/modifier-groups"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-modifier-group",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-modifier-group",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/item-modifier-group",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifier-groups/create",
        destPath: "/v1/bulk/modifier-groups/create"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers-groups/delete",
        destPath: "/v1/bulk/modifiers-groups/delete"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/modifiers-groups/update",
        destPath: "/v1/bulk/modifiers-groups/update"
      },
      {
        srcPath: "/restaurants/menus/v1/modifier-groups",
        destPath: "/v1/modifier-groups"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-modifier-group",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_item-modifier-groups";
function createModifierGroup(payload) {
  function __createModifierGroup({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "modifierGroup.createdDate" },
          { path: "modifierGroup.updatedDate" },
          { path: "modifierGroup.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier_group",
      method: "POST",
      methodFqn: "wix.restaurants.menus.v1.ModifierGroupsService.CreateModifierGroup",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl({
        protoPath: "/v1/modifier-groups",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "modifierGroup.createdDate" },
            { path: "modifierGroup.updatedDate" },
            { path: "modifierGroup.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createModifierGroup;
}
function getModifierGroup(payload) {
  function __getModifierGroup({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier_group",
      method: "GET",
      methodFqn: "wix.restaurants.menus.v1.ModifierGroupsService.GetModifierGroup",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl({
        protoPath: "/v1/modifier-groups/{modifierGroupId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "modifierGroup.createdDate" },
            { path: "modifierGroup.updatedDate" },
            { path: "modifierGroup.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getModifierGroup;
}
function listModifierGroups(payload) {
  function __listModifierGroups({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier_group",
      method: "GET",
      methodFqn: "wix.restaurants.menus.v1.ModifierGroupsService.ListModifierGroups",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl({
        protoPath: "/v1/modifier-groups",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "modifierGroups.createdDate" },
            { path: "modifierGroups.updatedDate" },
            { path: "modifierGroups.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listModifierGroups;
}
function queryModifierGroups(payload) {
  function __queryModifierGroups({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier_group",
      method: "POST",
      methodFqn: "wix.restaurants.menus.v1.ModifierGroupsService.QueryModifierGroups",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl({
        protoPath: "/v1/modifier-groups/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "modifierGroups.createdDate" },
            { path: "modifierGroups.updatedDate" },
            { path: "modifierGroups.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryModifierGroups;
}
function countModifierGroups(payload) {
  function __countModifierGroups({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier_group",
      method: "POST",
      methodFqn: "wix.restaurants.menus.v1.ModifierGroupsService.CountModifierGroups",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl({
        protoPath: "/v1/modifier-groups/count",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __countModifierGroups;
}
function updateModifierGroup(payload) {
  function __updateModifierGroup({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "modifierGroup.createdDate" },
          { path: "modifierGroup.updatedDate" },
          { path: "modifierGroup.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier_group",
      method: "PATCH",
      methodFqn: "wix.restaurants.menus.v1.ModifierGroupsService.UpdateModifierGroup",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl({
        protoPath: "/v1/modifier-groups/{modifierGroup.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "modifierGroup.createdDate" },
            { path: "modifierGroup.updatedDate" },
            { path: "modifierGroup.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateModifierGroup;
}
function deleteModifierGroup(payload) {
  function __deleteModifierGroup({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier_group",
      method: "DELETE",
      methodFqn: "wix.restaurants.menus.v1.ModifierGroupsService.DeleteModifierGroup",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl({
        protoPath: "/v1/modifier-groups/{modifierGroupId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteModifierGroup;
}
function bulkCreateModifierGroups(payload) {
  function __bulkCreateModifierGroups({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "modifierGroups.createdDate" },
          { path: "modifierGroups.updatedDate" },
          { path: "modifierGroups.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier_group",
      method: "POST",
      methodFqn: "wix.restaurants.menus.v1.ModifierGroupsService.BulkCreateModifierGroups",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl({
        protoPath: "/v1/bulk/modifier-groups/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.modifierGroup.createdDate" },
            { path: "results.modifierGroup.updatedDate" },
            {
              path: "results.modifierGroup.externalReferenceInfo.lastSyncDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateModifierGroups;
}
function bulkUpdateModifierGroups(payload) {
  function __bulkUpdateModifierGroups({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "modifierGroups.mask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "modifierGroups.modifierGroup.createdDate" },
          { path: "modifierGroups.modifierGroup.updatedDate" },
          {
            path: "modifierGroups.modifierGroup.externalReferenceInfo.lastSyncDate"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier_group",
      method: "POST",
      methodFqn: "wix.restaurants.menus.v1.ModifierGroupsService.BulkUpdateModifierGroups",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl({
        protoPath: "/v1/bulk/modifiers-groups/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.modifierGroup.createdDate" },
            { path: "results.modifierGroup.updatedDate" },
            {
              path: "results.modifierGroup.externalReferenceInfo.lastSyncDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateModifierGroups;
}
function bulkDeleteModifierGroups(payload) {
  function __bulkDeleteModifierGroups({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_modifier_group",
      method: "DELETE",
      methodFqn: "wix.restaurants.menus.v1.ModifierGroupsService.BulkDeleteModifierGroups",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusV1ModifierGroupsServiceUrl({
        protoPath: "/v1/bulk/modifiers-groups/delete",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __bulkDeleteModifierGroups;
}

// src/restaurants-menus-v1-item-modifier-group-item-modifier-groups.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createModifierGroup2(modifierGroup) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    modifierGroup
  });
  const reqOpts = createModifierGroup(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.modifierGroup;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { modifierGroup: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["modifierGroup"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getModifierGroup2(modifierGroupId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    modifierGroupId
  });
  const reqOpts = getModifierGroup(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.modifierGroup;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { modifierGroupId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["modifierGroupId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listModifierGroups2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    modifierGroupIds: options?.modifierGroupIds,
    paging: options?.paging
  });
  const reqOpts = listModifierGroups(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          modifierGroupIds: "$[0].modifierGroupIds",
          paging: "$[0].paging"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryModifierGroups2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryModifierGroups(
        payload
      );
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [])
      );
      return {
        items: transformedData?.modifierGroups,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function countModifierGroups2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    filter: options?.filter
  });
  const reqOpts = countModifierGroups(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { filter: "$[0].filter" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateModifierGroup2(_id, modifierGroup) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    modifierGroup: { ...modifierGroup, id: _id }
  });
  const reqOpts = updateModifierGroup(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.modifierGroup;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { modifierGroup: "$[1]" },
        explicitPathsToArguments: { "modifierGroup.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "modifierGroup"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteModifierGroup2(modifierGroupId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    modifierGroupId
  });
  const reqOpts = deleteModifierGroup(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { modifierGroupId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["modifierGroupId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCreateModifierGroups2(modifierGroups, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    modifierGroups,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkCreateModifierGroups(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          modifierGroups: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["modifierGroups", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateModifierGroups2(modifierGroups, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    modifierGroups,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkUpdateModifierGroups(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          modifierGroups: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["modifierGroups", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkDeleteModifierGroups2(ids) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ ids });
  const reqOpts = bulkDeleteModifierGroups(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { ids: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["ids"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
export {
  SortOrder,
  WebhookIdentityType,
  bulkCreateModifierGroups2 as bulkCreateModifierGroups,
  bulkDeleteModifierGroups2 as bulkDeleteModifierGroups,
  bulkUpdateModifierGroups2 as bulkUpdateModifierGroups,
  countModifierGroups2 as countModifierGroups,
  createModifierGroup2 as createModifierGroup,
  deleteModifierGroup2 as deleteModifierGroup,
  getModifierGroup2 as getModifierGroup,
  listModifierGroups2 as listModifierGroups,
  queryModifierGroups2 as queryModifierGroups,
  updateModifierGroup2 as updateModifierGroup
};
