// @wix/auto_sdk_restaurants_menu-ordering-settings/build/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  AvailabilityStatus: () => AvailabilityStatus,
  AvailabilityType: () => AvailabilityType,
  DayOfWeek: () => DayOfWeek,
  EntitiesDayOfWeek: () => EntitiesDayOfWeek,
  PlacementType: () => PlacementType,
  ResolutionMethod: () => ResolutionMethod,
  SortOrder: () => SortOrder,
  WebhookIdentityType: () => WebhookIdentityType,
  bulkUpdateMenuOrderingSettings: () => bulkUpdateMenuOrderingSettings4,
  bulkUpdateMenuOrderingSettingsTags: () => bulkUpdateMenuOrderingSettingsTags4,
  bulkUpdateMenuOrderingSettingsTagsByFilter: () => bulkUpdateMenuOrderingSettingsTagsByFilter4,
  getMenuOrderingSettings: () => getMenuOrderingSettings4,
  listMenusAvailabilityStatus: () => listMenusAvailabilityStatus4,
  onMenuOrderingSettingsCreated: () => onMenuOrderingSettingsCreated2,
  onMenuOrderingSettingsDeleted: () => onMenuOrderingSettingsDeleted2,
  onMenuOrderingSettingsUpdated: () => onMenuOrderingSettingsUpdated2,
  queryMenuOrderingSettings: () => queryMenuOrderingSettings4,
  updateExtendedFields: () => updateExtendedFields4,
  updateMenuOrderingSettings: () => updateMenuOrderingSettings4,
  upsertMenuOrderingSettingsByMenuId: () => upsertMenuOrderingSettingsByMenuId4
});
module.exports = __toCommonJS(index_exports);

// src/restaurants-menu-settings-v1-menu-ordering-settings-menu-ordering-settings.public.ts
var import_rename_all_nested_keys2 = require("@wix/sdk-runtime/rename-all-nested-keys");
var import_timestamp3 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths3 = require("@wix/sdk-runtime/transformations/transform-paths");
var import_sdk_types = require("@wix/sdk-types");

// src/restaurants-menu-settings-v1-menu-ordering-settings-menu-ordering-settings.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/restaurants-menu-settings-v1-menu-ordering-settings-menu-ordering-settings.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixRestaurantsMenuSettingsV1MenuOrderingSettingsServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/_api/menu-ordering-settings",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/menu-ordering-settings",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/menu-ordering-settings",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/menu-ordering-settings",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/menu-ordering-settings",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/menu-ordering-settings",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/menu-ordering-settings",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/menu-ordering-settings",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/menu-ordering-settings",
        destPath: ""
      }
    ],
    _base_domain_: [
      {
        srcPath: "/_api/menu-ordering-settings",
        destPath: ""
      }
    ],
    "users._base_domain_": [
      {
        srcPath: "/_api/menu-ordering-settings",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/menu-ordering-settings",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/menu-ordering-settings",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/menu-ordering-settings",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/menu-ordering-settings",
        destPath: ""
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/menu-ordering-settings",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_menu-ordering-settings";
function getMenuOrderingSettings(payload) {
  function __getMenuOrderingSettings({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menu_settings.v1.menu_ordering_settings",
      method: "GET",
      methodFqn: "wix.restaurants.menu_settings.v1.MenuOrderingSettingsService.GetMenuOrderingSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenuSettingsV1MenuOrderingSettingsServiceUrl({
        protoPath: "/v1/menu-ordering-settings/{menuOrderingSettingsId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "menuOrderingSettings.createdDate" },
            { path: "menuOrderingSettings.updatedDate" },
            {
              path: "menuOrderingSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "menuOrderingSettings.availability.timestampRangesOptions.ranges.endTime"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getMenuOrderingSettings;
}
function upsertMenuOrderingSettingsByMenuId(payload) {
  function __upsertMenuOrderingSettingsByMenuId({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "menuOrderingSettings.createdDate" },
          { path: "menuOrderingSettings.updatedDate" },
          {
            path: "menuOrderingSettings.availability.timestampRangesOptions.ranges.startTime"
          },
          {
            path: "menuOrderingSettings.availability.timestampRangesOptions.ranges.endTime"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menu_settings.v1.menu_ordering_settings",
      method: "POST",
      methodFqn: "wix.restaurants.menu_settings.v1.MenuOrderingSettingsService.UpsertMenuOrderingSettingsByMenuId",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenuSettingsV1MenuOrderingSettingsServiceUrl({
        protoPath: "/v1/menu-ordering-settings/upsert/menu-id/{menuOrderingSettings.menuId}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "menuOrderingSettings.createdDate" },
            { path: "menuOrderingSettings.updatedDate" },
            {
              path: "menuOrderingSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "menuOrderingSettings.availability.timestampRangesOptions.ranges.endTime"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __upsertMenuOrderingSettingsByMenuId;
}
function updateMenuOrderingSettings(payload) {
  function __updateMenuOrderingSettings({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "menuOrderingSettings.createdDate" },
          { path: "menuOrderingSettings.updatedDate" },
          {
            path: "menuOrderingSettings.availability.timestampRangesOptions.ranges.startTime"
          },
          {
            path: "menuOrderingSettings.availability.timestampRangesOptions.ranges.endTime"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menu_settings.v1.menu_ordering_settings",
      method: "PATCH",
      methodFqn: "wix.restaurants.menu_settings.v1.MenuOrderingSettingsService.UpdateMenuOrderingSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenuSettingsV1MenuOrderingSettingsServiceUrl({
        protoPath: "/v1/menu-ordering-settings/{menuOrderingSettings.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "menuOrderingSettings.createdDate" },
            { path: "menuOrderingSettings.updatedDate" },
            {
              path: "menuOrderingSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "menuOrderingSettings.availability.timestampRangesOptions.ranges.endTime"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateMenuOrderingSettings;
}
function queryMenuOrderingSettings(payload) {
  function __queryMenuOrderingSettings({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menu_settings.v1.menu_ordering_settings",
      method: "POST",
      methodFqn: "wix.restaurants.menu_settings.v1.MenuOrderingSettingsService.QueryMenuOrderingSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenuSettingsV1MenuOrderingSettingsServiceUrl({
        protoPath: "/v1/menu-ordering-settings/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "menuOrderingSettings.createdDate" },
            { path: "menuOrderingSettings.updatedDate" },
            {
              path: "menuOrderingSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "menuOrderingSettings.availability.timestampRangesOptions.ranges.endTime"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryMenuOrderingSettings;
}
function bulkUpdateMenuOrderingSettings(payload) {
  function __bulkUpdateMenuOrderingSettings({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "menusOrderingSettings.mask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "menusOrderingSettings.menuOrderingSettings.createdDate" },
          { path: "menusOrderingSettings.menuOrderingSettings.updatedDate" },
          {
            path: "menusOrderingSettings.menuOrderingSettings.availability.timestampRangesOptions.ranges.startTime"
          },
          {
            path: "menusOrderingSettings.menuOrderingSettings.availability.timestampRangesOptions.ranges.endTime"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menu_settings.v1.menu_ordering_settings",
      method: "POST",
      methodFqn: "wix.restaurants.menu_settings.v1.MenuOrderingSettingsService.BulkUpdateMenuOrderingSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenuSettingsV1MenuOrderingSettingsServiceUrl({
        protoPath: "/v1/bulk/menu-ordering-settings/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.menuOrderingSettings.createdDate" },
            { path: "results.menuOrderingSettings.updatedDate" },
            {
              path: "results.menuOrderingSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "results.menuOrderingSettings.availability.timestampRangesOptions.ranges.endTime"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateMenuOrderingSettings;
}
function listMenusAvailabilityStatus(payload) {
  function __listMenusAvailabilityStatus({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "timeSlot.startTime" }, { path: "timeSlot.endTime" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menu_settings.v1.menu_ordering_settings",
      method: "GET",
      methodFqn: "wix.restaurants.menu_settings.v1.MenuOrderingSettingsService.ListMenusAvailabilityStatus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenuSettingsV1MenuOrderingSettingsServiceUrl({
        protoPath: "/v1/menu-ordering-settings/menus-availability-status",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData)
    };
    return metadata;
  }
  return __listMenusAvailabilityStatus;
}
function updateExtendedFields(payload) {
  function __updateExtendedFields({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menu_settings.v1.menu_ordering_settings",
      method: "POST",
      methodFqn: "wix.restaurants.menu_settings.v1.MenuOrderingSettingsService.UpdateExtendedFields",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenuSettingsV1MenuOrderingSettingsServiceUrl({
        protoPath: "/v1/menu-ordering-settings/{id}/update-extended-fields",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "menuOrderingSettings.createdDate" },
            { path: "menuOrderingSettings.updatedDate" },
            {
              path: "menuOrderingSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "menuOrderingSettings.availability.timestampRangesOptions.ranges.endTime"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateExtendedFields;
}
function bulkUpdateMenuOrderingSettingsTags(payload) {
  function __bulkUpdateMenuOrderingSettingsTags({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menu_settings.v1.menu_ordering_settings",
      method: "POST",
      methodFqn: "wix.restaurants.menu_settings.v1.MenuOrderingSettingsService.BulkUpdateMenuOrderingSettingsTags",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenuSettingsV1MenuOrderingSettingsServiceUrl({
        protoPath: "/v1/bulk/menu-ordering-settings/update-tags",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateMenuOrderingSettingsTags;
}
function bulkUpdateMenuOrderingSettingsTagsByFilter(payload) {
  function __bulkUpdateMenuOrderingSettingsTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menu_settings.v1.menu_ordering_settings",
      method: "POST",
      methodFqn: "wix.restaurants.menu_settings.v1.MenuOrderingSettingsService.BulkUpdateMenuOrderingSettingsTagsByFilter",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenuSettingsV1MenuOrderingSettingsServiceUrl({
        protoPath: "/v1/bulk/menu-ordering-settings/update-tags-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateMenuOrderingSettingsTagsByFilter;
}

// src/restaurants-menu-settings-v1-menu-ordering-settings-menu-ordering-settings.universal.ts
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var AvailabilityType = /* @__PURE__ */ ((AvailabilityType2) => {
  AvailabilityType2["UNSPECIFIED_AVAILABILITY_OPTION"] = "UNSPECIFIED_AVAILABILITY_OPTION";
  AvailabilityType2["ALWAYS_AVAILABLE"] = "ALWAYS_AVAILABLE";
  AvailabilityType2["WEEKLY_SCHEDULE"] = "WEEKLY_SCHEDULE";
  AvailabilityType2["TIMESTAMP_RANGES"] = "TIMESTAMP_RANGES";
  return AvailabilityType2;
})(AvailabilityType || {});
var EntitiesDayOfWeek = /* @__PURE__ */ ((EntitiesDayOfWeek2) => {
  EntitiesDayOfWeek2["MON"] = "MON";
  EntitiesDayOfWeek2["TUE"] = "TUE";
  EntitiesDayOfWeek2["WED"] = "WED";
  EntitiesDayOfWeek2["THU"] = "THU";
  EntitiesDayOfWeek2["FRI"] = "FRI";
  EntitiesDayOfWeek2["SAT"] = "SAT";
  EntitiesDayOfWeek2["SUN"] = "SUN";
  return EntitiesDayOfWeek2;
})(EntitiesDayOfWeek || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var AvailabilityStatus = /* @__PURE__ */ ((AvailabilityStatus2) => {
  AvailabilityStatus2["UNKNOWN_AVAILABILITY_STATUS"] = "UNKNOWN_AVAILABILITY_STATUS";
  AvailabilityStatus2["AVAILABLE"] = "AVAILABLE";
  AvailabilityStatus2["UNAVAILABLE"] = "UNAVAILABLE";
  return AvailabilityStatus2;
})(AvailabilityStatus || {});
var PlacementType = /* @__PURE__ */ ((PlacementType2) => {
  PlacementType2["BEFORE"] = "BEFORE";
  PlacementType2["AFTER"] = "AFTER";
  PlacementType2["REPLACE"] = "REPLACE";
  return PlacementType2;
})(PlacementType || {});
var DayOfWeek = /* @__PURE__ */ ((DayOfWeek2) => {
  DayOfWeek2["MONDAY"] = "MONDAY";
  DayOfWeek2["TUESDAY"] = "TUESDAY";
  DayOfWeek2["WEDNESDAY"] = "WEDNESDAY";
  DayOfWeek2["THURSDAY"] = "THURSDAY";
  DayOfWeek2["FRIDAY"] = "FRIDAY";
  DayOfWeek2["SATURDAY"] = "SATURDAY";
  DayOfWeek2["SUNDAY"] = "SUNDAY";
  return DayOfWeek2;
})(DayOfWeek || {});
var ResolutionMethod = /* @__PURE__ */ ((ResolutionMethod2) => {
  ResolutionMethod2["QUERY_PARAM"] = "QUERY_PARAM";
  ResolutionMethod2["SUBDOMAIN"] = "SUBDOMAIN";
  ResolutionMethod2["SUBDIRECTORY"] = "SUBDIRECTORY";
  return ResolutionMethod2;
})(ResolutionMethod || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function getMenuOrderingSettings2(menuOrderingSettingsId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    menuOrderingSettingsId
  });
  const reqOpts = getMenuOrderingSettings(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.menuOrderingSettings;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { menuOrderingSettingsId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["menuOrderingSettingsId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function upsertMenuOrderingSettingsByMenuId2(menuId, menuOrderingSettings) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    menuOrderingSettings: { ...menuOrderingSettings, menuId }
  });
  const reqOpts = upsertMenuOrderingSettingsByMenuId(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { menuOrderingSettings: "$[1]" },
        explicitPathsToArguments: { "menuOrderingSettings.menuId": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["menuId", "menuOrderingSettings"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateMenuOrderingSettings2(_id, menuOrderingSettings) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    menuOrderingSettings: { ...menuOrderingSettings, id: _id }
  });
  const reqOpts = updateMenuOrderingSettings(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.menuOrderingSettings;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { menuOrderingSettings: "$[1]" },
        explicitPathsToArguments: { "menuOrderingSettings.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "menuOrderingSettings"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryMenuOrderingSettings2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryMenuOrderingSettings(
        payload
      );
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [])
      );
      return {
        items: transformedData?.menuOrderingSettings,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function bulkUpdateMenuOrderingSettings2(menusOrderingSettings, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    menusOrderingSettings,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkUpdateMenuOrderingSettings(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          menusOrderingSettings: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["menusOrderingSettings", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listMenusAvailabilityStatus2(timeSlot, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    timeSlot,
    operationId: options?.operationId,
    cursorPaging: options?.cursorPaging
  });
  const reqOpts = listMenusAvailabilityStatus(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          timeSlot: "$[0]",
          operationId: "$[1].operationId",
          cursorPaging: "$[1].cursorPaging"
        },
        singleArgumentUnchanged: false
      },
      ["timeSlot", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateExtendedFields2(_id, namespace, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    id: _id,
    namespace,
    namespaceData: options?.namespaceData
  });
  const reqOpts = updateExtendedFields(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          id: "$[0]",
          namespace: "$[1]",
          namespaceData: "$[2].namespaceData"
        },
        singleArgumentUnchanged: false
      },
      ["_id", "namespace", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateMenuOrderingSettingsTags2(menuOrderingSettingsIds, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    menuOrderingSettingsIds,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateMenuOrderingSettingsTags(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          menuOrderingSettingsIds: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["menuOrderingSettingsIds", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateMenuOrderingSettingsTagsByFilter2(filter, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    filter,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateMenuOrderingSettingsTagsByFilter(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          filter: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["filter", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/restaurants-menu-settings-v1-menu-ordering-settings-menu-ordering-settings.public.ts
function getMenuOrderingSettings3(httpClient) {
  return (menuOrderingSettingsId) => getMenuOrderingSettings2(
    menuOrderingSettingsId,
    // @ts-ignore
    { httpClient }
  );
}
function upsertMenuOrderingSettingsByMenuId3(httpClient) {
  return (menuId, menuOrderingSettings) => upsertMenuOrderingSettingsByMenuId2(
    menuId,
    menuOrderingSettings,
    // @ts-ignore
    { httpClient }
  );
}
function updateMenuOrderingSettings3(httpClient) {
  return (_id, menuOrderingSettings) => updateMenuOrderingSettings2(
    _id,
    menuOrderingSettings,
    // @ts-ignore
    { httpClient }
  );
}
function queryMenuOrderingSettings3(httpClient) {
  return () => queryMenuOrderingSettings2(
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdateMenuOrderingSettings3(httpClient) {
  return (menusOrderingSettings, options) => bulkUpdateMenuOrderingSettings2(
    menusOrderingSettings,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function listMenusAvailabilityStatus3(httpClient) {
  return (timeSlot, options) => listMenusAvailabilityStatus2(
    timeSlot,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function updateExtendedFields3(httpClient) {
  return (_id, namespace, options) => updateExtendedFields2(
    _id,
    namespace,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdateMenuOrderingSettingsTags3(httpClient) {
  return (menuOrderingSettingsIds, options) => bulkUpdateMenuOrderingSettingsTags2(
    menuOrderingSettingsIds,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdateMenuOrderingSettingsTagsByFilter3(httpClient) {
  return (filter, options) => bulkUpdateMenuOrderingSettingsTagsByFilter2(
    filter,
    options,
    // @ts-ignore
    { httpClient }
  );
}
var onMenuOrderingSettingsCreated = (0, import_sdk_types.EventDefinition)(
  "wix.restaurants.menu_settings.v1.menu_ordering_settings_created",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          {
            path: "entity.availability.timestampRangesOptions.ranges.startTime"
          },
          {
            path: "entity.availability.timestampRangesOptions.ranges.endTime"
          },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onMenuOrderingSettingsDeleted = (0, import_sdk_types.EventDefinition)(
  "wix.restaurants.menu_settings.v1.menu_ordering_settings_deleted",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          {
            path: "entity.availability.timestampRangesOptions.ranges.startTime"
          },
          {
            path: "entity.availability.timestampRangesOptions.ranges.endTime"
          },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onMenuOrderingSettingsUpdated = (0, import_sdk_types.EventDefinition)(
  "wix.restaurants.menu_settings.v1.menu_ordering_settings_updated",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          {
            path: "entity.availability.timestampRangesOptions.ranges.startTime"
          },
          {
            path: "entity.availability.timestampRangesOptions.ranges.endTime"
          },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();

// src/restaurants-menu-settings-v1-menu-ordering-settings-menu-ordering-settings.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var import_event_definition_modules = require("@wix/sdk-runtime/event-definition-modules");
var getMenuOrderingSettings4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getMenuOrderingSettings3);
var upsertMenuOrderingSettingsByMenuId4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(upsertMenuOrderingSettingsByMenuId3);
var updateMenuOrderingSettings4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(updateMenuOrderingSettings3);
var queryMenuOrderingSettings4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(queryMenuOrderingSettings3);
var bulkUpdateMenuOrderingSettings4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkUpdateMenuOrderingSettings3);
var listMenusAvailabilityStatus4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(listMenusAvailabilityStatus3);
var updateExtendedFields4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(updateExtendedFields3);
var bulkUpdateMenuOrderingSettingsTags4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkUpdateMenuOrderingSettingsTags3);
var bulkUpdateMenuOrderingSettingsTagsByFilter4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(
  bulkUpdateMenuOrderingSettingsTagsByFilter3
);
var onMenuOrderingSettingsCreated2 = (0, import_event_definition_modules.createEventModule)(
  onMenuOrderingSettingsCreated
);
var onMenuOrderingSettingsDeleted2 = (0, import_event_definition_modules.createEventModule)(
  onMenuOrderingSettingsDeleted
);
var onMenuOrderingSettingsUpdated2 = (0, import_event_definition_modules.createEventModule)(
  onMenuOrderingSettingsUpdated
);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AvailabilityStatus,
  AvailabilityType,
  DayOfWeek,
  EntitiesDayOfWeek,
  PlacementType,
  ResolutionMethod,
  SortOrder,
  WebhookIdentityType,
  bulkUpdateMenuOrderingSettings,
  bulkUpdateMenuOrderingSettingsTags,
  bulkUpdateMenuOrderingSettingsTagsByFilter,
  getMenuOrderingSettings,
  listMenusAvailabilityStatus,
  onMenuOrderingSettingsCreated,
  onMenuOrderingSettingsDeleted,
  onMenuOrderingSettingsUpdated,
  queryMenuOrderingSettings,
  updateExtendedFields,
  updateMenuOrderingSettings,
  upsertMenuOrderingSettingsByMenuId
});
