// @wix/auto_sdk_online-programs_sections/build/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  BulkActionType: () => BulkActionType,
  DescriptionFieldSet: () => DescriptionFieldSet,
  Enum: () => Enum,
  RequestedFields: () => RequestedFields,
  SearchIndexingNotificationState: () => SearchIndexingNotificationState,
  SortOrder: () => SortOrder,
  State: () => State,
  WebhookIdentityType: () => WebhookIdentityType,
  bulkCreateSection: () => bulkCreateSection4,
  bulkCreateSectionMigration: () => bulkCreateSectionMigration4,
  cloneSection: () => cloneSection4,
  createSection: () => createSection4,
  deleteSection: () => deleteSection4,
  getSection: () => getSection4,
  listSections: () => listSections4,
  moveSection: () => moveSection4,
  publishSection: () => publishSection4,
  querySections: () => querySections4,
  updateSection: () => updateSection4
});
module.exports = __toCommonJS(index_exports);

// src/online-programs-v3-section-sections.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/online-programs-v3-section-sections.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixOnlineprogramsSectionsV3SectionsServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/program-sections-service",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/program-sections-service",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/program-sections-service",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/program-sections-service",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/program-sections-service",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/program-sections-service",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/program-sections-service",
        destPath: ""
      },
      {
        srcPath: "/program-sections-service",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/program-sections-service",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/program-sections-service",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/program-sections-service",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/program-sections-service",
        destPath: ""
      },
      {
        srcPath: "/program-sections-service",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/program-sections-service",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/program-sections-service",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_online-programs_sections";
function createSection(payload) {
  function __createSection({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "section.createdDate" },
          { path: "section.updatedDate" },
          { path: "section.description.image.urlExpirationDate" },
          { path: "section.description.video.urlExpirationDate" },
          { path: "section.description.video.resolutions.urlExpirationDate" },
          {
            path: "section.description.video.resolutions.poster.urlExpirationDate"
          },
          { path: "section.description.video.posters.urlExpirationDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "section.ordering" },
          { path: "section.description.image.focalPoint.x" },
          { path: "section.description.image.focalPoint.y" },
          { path: "section.description.video.resolutions.poster.focalPoint.x" },
          { path: "section.description.video.resolutions.poster.focalPoint.y" },
          { path: "section.description.video.posters.focalPoint.x" },
          { path: "section.description.video.posters.focalPoint.y" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "POST",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.CreateSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/sections",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "section.createdDate" },
            { path: "section.updatedDate" },
            { path: "section.description.image.urlExpirationDate" },
            { path: "section.description.video.urlExpirationDate" },
            {
              path: "section.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "section.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "section.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "section.ordering" },
            { path: "section.description.image.focalPoint.x" },
            { path: "section.description.image.focalPoint.y" },
            {
              path: "section.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "section.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "section.description.video.posters.focalPoint.x" },
            { path: "section.description.video.posters.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createSection;
}
function bulkCreateSection(payload) {
  function __bulkCreateSection({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "sections.createdDate" },
          { path: "sections.updatedDate" },
          { path: "sections.description.image.urlExpirationDate" },
          { path: "sections.description.video.urlExpirationDate" },
          { path: "sections.description.video.resolutions.urlExpirationDate" },
          {
            path: "sections.description.video.resolutions.poster.urlExpirationDate"
          },
          { path: "sections.description.video.posters.urlExpirationDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "sections.ordering" },
          { path: "sections.description.image.focalPoint.x" },
          { path: "sections.description.image.focalPoint.y" },
          {
            path: "sections.description.video.resolutions.poster.focalPoint.x"
          },
          {
            path: "sections.description.video.resolutions.poster.focalPoint.y"
          },
          { path: "sections.description.video.posters.focalPoint.x" },
          { path: "sections.description.video.posters.focalPoint.y" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "POST",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.BulkCreateSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/bulk/sections/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.description.image.urlExpirationDate" },
            { path: "results.item.description.video.urlExpirationDate" },
            {
              path: "results.item.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.description.video.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.description.video.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.ordering" },
            { path: "results.item.description.image.focalPoint.x" },
            { path: "results.item.description.image.focalPoint.y" },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "results.item.description.video.posters.focalPoint.x" },
            { path: "results.item.description.video.posters.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateSection;
}
function bulkCreateSectionMigration(payload) {
  function __bulkCreateSectionMigration({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "sections.createdDate" },
          { path: "sections.updatedDate" },
          { path: "sections.description.image.urlExpirationDate" },
          { path: "sections.description.video.urlExpirationDate" },
          { path: "sections.description.video.resolutions.urlExpirationDate" },
          {
            path: "sections.description.video.resolutions.poster.urlExpirationDate"
          },
          { path: "sections.description.video.posters.urlExpirationDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "sections.ordering" },
          { path: "sections.description.image.focalPoint.x" },
          { path: "sections.description.image.focalPoint.y" },
          {
            path: "sections.description.video.resolutions.poster.focalPoint.x"
          },
          {
            path: "sections.description.video.resolutions.poster.focalPoint.y"
          },
          { path: "sections.description.video.posters.focalPoint.x" },
          { path: "sections.description.video.posters.focalPoint.y" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "POST",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.BulkCreateSectionMigration",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/bulk/sections/create/migration",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.description.image.urlExpirationDate" },
            { path: "results.item.description.video.urlExpirationDate" },
            {
              path: "results.item.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.description.video.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.description.video.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.ordering" },
            { path: "results.item.description.image.focalPoint.x" },
            { path: "results.item.description.image.focalPoint.y" },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "results.item.description.video.posters.focalPoint.x" },
            { path: "results.item.description.video.posters.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateSectionMigration;
}
function getSection(payload) {
  function __getSection({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "GET",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.GetSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/sections/{sectionId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "section.createdDate" },
            { path: "section.updatedDate" },
            { path: "section.description.image.urlExpirationDate" },
            { path: "section.description.video.urlExpirationDate" },
            {
              path: "section.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "section.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "section.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "section.ordering" },
            { path: "section.description.image.focalPoint.x" },
            { path: "section.description.image.focalPoint.y" },
            {
              path: "section.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "section.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "section.description.video.posters.focalPoint.x" },
            { path: "section.description.video.posters.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getSection;
}
function updateSection(payload) {
  function __updateSection({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "section.createdDate" },
          { path: "section.updatedDate" },
          { path: "section.description.image.urlExpirationDate" },
          { path: "section.description.video.urlExpirationDate" },
          { path: "section.description.video.resolutions.urlExpirationDate" },
          {
            path: "section.description.video.resolutions.poster.urlExpirationDate"
          },
          { path: "section.description.video.posters.urlExpirationDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "section.ordering" },
          { path: "section.description.image.focalPoint.x" },
          { path: "section.description.image.focalPoint.y" },
          { path: "section.description.video.resolutions.poster.focalPoint.x" },
          { path: "section.description.video.resolutions.poster.focalPoint.y" },
          { path: "section.description.video.posters.focalPoint.x" },
          { path: "section.description.video.posters.focalPoint.y" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "PATCH",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.UpdateSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/sections/{section.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "section.createdDate" },
            { path: "section.updatedDate" },
            { path: "section.description.image.urlExpirationDate" },
            { path: "section.description.video.urlExpirationDate" },
            {
              path: "section.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "section.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "section.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "section.ordering" },
            { path: "section.description.image.focalPoint.x" },
            { path: "section.description.image.focalPoint.y" },
            {
              path: "section.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "section.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "section.description.video.posters.focalPoint.x" },
            { path: "section.description.video.posters.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateSection;
}
function deleteSection(payload) {
  function __deleteSection({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "DELETE",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.DeleteSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/sections/{sectionId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteSection;
}
function querySections(payload) {
  function __querySections({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "POST",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.QuerySections",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/sections/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "sections.createdDate" },
            { path: "sections.updatedDate" },
            { path: "sections.description.image.urlExpirationDate" },
            { path: "sections.description.video.urlExpirationDate" },
            {
              path: "sections.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "sections.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "sections.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "sections.ordering" },
            { path: "sections.description.image.focalPoint.x" },
            { path: "sections.description.image.focalPoint.y" },
            {
              path: "sections.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "sections.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "sections.description.video.posters.focalPoint.x" },
            { path: "sections.description.video.posters.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __querySections;
}
function listSections(payload) {
  function __listSections({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "GET",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.ListSections",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/sections",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "sections.createdDate" },
            { path: "sections.updatedDate" },
            { path: "sections.description.image.urlExpirationDate" },
            { path: "sections.description.video.urlExpirationDate" },
            {
              path: "sections.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "sections.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "sections.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "sections.ordering" },
            { path: "sections.description.image.focalPoint.x" },
            { path: "sections.description.image.focalPoint.y" },
            {
              path: "sections.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "sections.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "sections.description.video.posters.focalPoint.x" },
            { path: "sections.description.video.posters.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listSections;
}
function cloneSection(payload) {
  function __cloneSection({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "POST",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.CloneSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/sections/{sectionId}/clone",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "section.createdDate" },
            { path: "section.updatedDate" },
            { path: "section.description.image.urlExpirationDate" },
            { path: "section.description.video.urlExpirationDate" },
            {
              path: "section.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "section.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "section.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "section.ordering" },
            { path: "section.description.image.focalPoint.x" },
            { path: "section.description.image.focalPoint.y" },
            {
              path: "section.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "section.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "section.description.video.posters.focalPoint.x" },
            { path: "section.description.video.posters.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __cloneSection;
}
function moveSection(payload) {
  function __moveSection({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "PATCH",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.MoveSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/sections/move",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "section.createdDate" },
            { path: "section.updatedDate" },
            { path: "section.description.image.urlExpirationDate" },
            { path: "section.description.video.urlExpirationDate" },
            {
              path: "section.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "section.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "section.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "section.ordering" },
            { path: "section.description.image.focalPoint.x" },
            { path: "section.description.image.focalPoint.y" },
            {
              path: "section.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "section.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "section.description.video.posters.focalPoint.x" },
            { path: "section.description.video.posters.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __moveSection;
}
function publishSection(payload) {
  function __publishSection({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "PATCH",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.PublishSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/sections/publish",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "section.createdDate" },
            { path: "section.updatedDate" },
            { path: "section.description.image.urlExpirationDate" },
            { path: "section.description.video.urlExpirationDate" },
            {
              path: "section.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "section.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "section.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "section.ordering" },
            { path: "section.description.image.focalPoint.x" },
            { path: "section.description.image.focalPoint.y" },
            {
              path: "section.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "section.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "section.description.video.posters.focalPoint.x" },
            { path: "section.description.video.posters.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __publishSection;
}

// src/online-programs-v3-section-sections.universal.ts
var import_image = require("@wix/sdk-runtime/transformations/image");
var import_image2 = require("@wix/sdk-runtime/transformations/image");
var import_video_v2 = require("@wix/sdk-runtime/transformations/video-v2");
var import_video_v22 = require("@wix/sdk-runtime/transformations/video-v2");
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var State = /* @__PURE__ */ ((State2) => {
  State2["UNKNOWN_STATE"] = "UNKNOWN_STATE";
  State2["DRAFT"] = "DRAFT";
  State2["PUBLISHED"] = "PUBLISHED";
  return State2;
})(State || {});
var Enum = /* @__PURE__ */ ((Enum2) => {
  Enum2["UNKNOWN"] = "UNKNOWN";
  Enum2["PROTECTED"] = "PROTECTED";
  Enum2["PRIVATE"] = "PRIVATE";
  Enum2["PUBLIC"] = "PUBLIC";
  Enum2["UNCHANGED"] = "UNCHANGED";
  Enum2["GROUP_PROTECTED"] = "GROUP_PROTECTED";
  return Enum2;
})(Enum || {});
var RequestedFields = /* @__PURE__ */ ((RequestedFields2) => {
  RequestedFields2["UNKNOWN_REQUESTED_FIELD"] = "UNKNOWN_REQUESTED_FIELD";
  RequestedFields2["TOTAL_STEPS"] = "TOTAL_STEPS";
  return RequestedFields2;
})(RequestedFields || {});
var BulkActionType = /* @__PURE__ */ ((BulkActionType2) => {
  BulkActionType2["UNKNOWN_ACTION_TYPE"] = "UNKNOWN_ACTION_TYPE";
  BulkActionType2["INSERT"] = "INSERT";
  BulkActionType2["UPDATE"] = "UPDATE";
  BulkActionType2["DELETE"] = "DELETE";
  return BulkActionType2;
})(BulkActionType || {});
var DescriptionFieldSet = /* @__PURE__ */ ((DescriptionFieldSet2) => {
  DescriptionFieldSet2["STANDARD"] = "STANDARD";
  DescriptionFieldSet2["EXTENDED"] = "EXTENDED";
  return DescriptionFieldSet2;
})(DescriptionFieldSet || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var SearchIndexingNotificationState = /* @__PURE__ */ ((SearchIndexingNotificationState2) => {
  SearchIndexingNotificationState2["Unknown"] = "Unknown";
  SearchIndexingNotificationState2["Off"] = "Off";
  SearchIndexingNotificationState2["On"] = "On";
  return SearchIndexingNotificationState2;
})(SearchIndexingNotificationState || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createSection2(section, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      section,
      fields: options?.fields
    }),
    [
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [{ path: "section.description.image" }]
      },
      {
        transformFn: import_video_v2.transformSDKVideoV2ToRESTVideoV2,
        paths: [{ path: "section.description.video" }]
      }
    ]
  );
  const reqOpts = createSection(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "section.description.image" }]
        },
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "section.description.video" }]
        }
      ])
    )?.section;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { section: "$[0]", fields: "$[1].fields" },
        singleArgumentUnchanged: false
      },
      ["section", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCreateSection2(sections, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      sections,
      fields: options?.fields
    }),
    [
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [{ path: "sections.description.image" }]
      },
      {
        transformFn: import_video_v2.transformSDKVideoV2ToRESTVideoV2,
        paths: [{ path: "sections.description.video" }]
      }
    ]
  );
  const reqOpts = bulkCreateSection(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "results.item.description.image" }]
        },
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "results.item.description.video" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { sections: "$[0]", fields: "$[1].fields" },
        singleArgumentUnchanged: false
      },
      ["sections", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCreateSectionMigration2(sections) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ sections }),
    [
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [{ path: "sections.description.image" }]
      },
      {
        transformFn: import_video_v2.transformSDKVideoV2ToRESTVideoV2,
        paths: [{ path: "sections.description.video" }]
      }
    ]
  );
  const reqOpts = bulkCreateSectionMigration(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "results.item.description.image" }]
        },
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "results.item.description.video" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { sections: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["sections"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getSection2(sectionId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    sectionId,
    descriptionFieldSet: options?.descriptionFieldSet,
    fields: options?.fields
  });
  const reqOpts = getSection(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "section.description.image" }]
        },
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "section.description.video" }]
        }
      ])
    )?.section;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          sectionId: "$[0]",
          descriptionFieldSet: "$[1].descriptionFieldSet",
          fields: "$[1].fields"
        },
        singleArgumentUnchanged: false
      },
      ["sectionId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateSection2(_id, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      section: { ...options?.section, id: _id },
      fields: options?.fields
    }),
    [
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [{ path: "section.description.image" }]
      },
      {
        transformFn: import_video_v2.transformSDKVideoV2ToRESTVideoV2,
        paths: [{ path: "section.description.video" }]
      }
    ]
  );
  const reqOpts = updateSection(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "section.description.image" }]
        },
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "section.description.video" }]
        }
      ])
    )?.section;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { section: "$[1].section" },
        explicitPathsToArguments: {
          "section.id": "$[0]",
          fields: "$[1].fields"
        },
        singleArgumentUnchanged: false
      },
      ["_id", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteSection2(sectionId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    sectionId
  });
  const reqOpts = deleteSection(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { sectionId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["sectionId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function querySections2(options) {
  const { httpClient, sideEffects } = arguments[1];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = querySections({
        ...payload,
        ...options ?? {}
      });
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, options];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [
          {
            transformFn: import_image2.transformRESTImageToSDKImage,
            paths: [{ path: "sections.description.image" }]
          },
          {
            transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
            paths: [{ path: "sections.description.video" }]
          }
        ])
      );
      return {
        items: transformedData?.sections,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function listSections2(programId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    programId,
    descriptionFieldSet: options?.descriptionFieldSet,
    fields: options?.fields
  });
  const reqOpts = listSections(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "sections.description.image" }]
        },
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "sections.description.video" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          programId: "$[0]",
          descriptionFieldSet: "$[1].descriptionFieldSet",
          fields: "$[1].fields"
        },
        singleArgumentUnchanged: false
      },
      ["programId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function cloneSection2(sectionId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    sectionId,
    fields: options?.fields
  });
  const reqOpts = cloneSection(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "section.description.image" }]
        },
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "section.description.video" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { sectionId: "$[0]", fields: "$[1].fields" },
        singleArgumentUnchanged: false
      },
      ["sectionId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function moveSection2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    sectionId: options?.sectionId,
    afterSectionId: options?.afterSectionId,
    revision: options?.revision,
    fields: options?.fields
  });
  const reqOpts = moveSection(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "section.description.image" }]
        },
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "section.description.video" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          sectionId: "$[0].sectionId",
          afterSectionId: "$[0].afterSectionId",
          revision: "$[0].revision",
          fields: "$[0].fields"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function publishSection2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    sectionId: options?.sectionId,
    revision: options?.revision,
    fields: options?.fields
  });
  const reqOpts = publishSection(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "section.description.image" }]
        },
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "section.description.video" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          sectionId: "$[0].sectionId",
          revision: "$[0].revision",
          fields: "$[0].fields"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/online-programs-v3-section-sections.public.ts
function createSection3(httpClient) {
  return (section, options) => createSection2(
    section,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkCreateSection3(httpClient) {
  return (sections, options) => bulkCreateSection2(
    sections,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkCreateSectionMigration3(httpClient) {
  return (sections) => bulkCreateSectionMigration2(
    sections,
    // @ts-ignore
    { httpClient }
  );
}
function getSection3(httpClient) {
  return (sectionId, options) => getSection2(
    sectionId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function updateSection3(httpClient) {
  return (_id, options) => updateSection2(
    _id,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function deleteSection3(httpClient) {
  return (sectionId) => deleteSection2(
    sectionId,
    // @ts-ignore
    { httpClient }
  );
}
function querySections3(httpClient) {
  return (options) => querySections2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function listSections3(httpClient) {
  return (programId, options) => listSections2(
    programId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function cloneSection3(httpClient) {
  return (sectionId, options) => cloneSection2(
    sectionId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function moveSection3(httpClient) {
  return (options) => moveSection2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function publishSection3(httpClient) {
  return (options) => publishSection2(
    options,
    // @ts-ignore
    { httpClient }
  );
}

// src/online-programs-v3-section-sections.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var createSection4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(createSection3);
var bulkCreateSection4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkCreateSection3);
var bulkCreateSectionMigration4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkCreateSectionMigration3);
var getSection4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getSection3);
var updateSection4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(updateSection3);
var deleteSection4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(deleteSection3);
var querySections4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(querySections3);
var listSections4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(listSections3);
var cloneSection4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(cloneSection3);
var moveSection4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(moveSection3);
var publishSection4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(publishSection3);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  BulkActionType,
  DescriptionFieldSet,
  Enum,
  RequestedFields,
  SearchIndexingNotificationState,
  SortOrder,
  State,
  WebhookIdentityType,
  bulkCreateSection,
  bulkCreateSectionMigration,
  cloneSection,
  createSection,
  deleteSection,
  getSection,
  listSections,
  moveSection,
  publishSection,
  querySections,
  updateSection
});
