// @wix/unidriver-puppeteer/dist/cjs/__tests__/setup.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.setup = exports.setupWithAwaiter = void 0;
const tslib_1 = require("tslib");
const puppeteerUniDriver_1 = require("../puppeteerUniDriver");
const puppeteer_1 = tslib_1.__importDefault(require("puppeteer"));
const ci_info_1 = require("ci-info");
const internal_1 = require("@wix/unidriver-core/internal");
const setupWithAwaiter = async (params) => {
    const { driver, tearDown, page } = await (0, exports.setup)(params);
    const APP_SELECTOR = '[class="todo-app"]';
    await (0, internal_1.eventually)(async () => {
        if (!(await driver.$(APP_SELECTOR).exists())) {
            throw new Error(`element with selector ${APP_SELECTOR} not found`);
        }
    }, 1_000);
    return { driver, tearDown, page };
};
exports.setupWithAwaiter = setupWithAwaiter;
const setup = async ({ story }) => {
    const browser = ci_info_1.isCI
        ? await puppeteer_1.default.connect({
            browserURL: 'http://localhost:9222',
        })
        : await puppeteer_1.default.launch({
            headless: process.env.DEBUG !== 'true',
        });
    const page = await browser.newPage();
    const url = `http://localhost:6006/iframe.html?id=testapp--${story}`;
    await page.goto(url, {
        waitUntil: 'networkidle0',
        timeout: 10_000,
    });
    const driver = (0, puppeteerUniDriver_1.puppeteerUniDriver)({
        page,
        selector: 'body',
    });
    return {
        page,
        driver,
        tearDown: () => {
            return ci_info_1.isCI ? page.close() : browser.close();
        },
    };
};
exports.setup = setup;
