// @wix/auto_sdk_media_folders/build/es/index.typings.mjs
// src/media-site-media-v1-folder-folders.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/media-site-media-v1-folder-folders.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixMediaSiteMediaV1FoldersServiceUrl(opts) {
  const domainToMappings = {
    "apps._base_domain_": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: "/api"
      },
      {
        srcPath: "/_api/wix-privatemedia-proxy-server/v1",
        destPath: "/v1"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/site-media/v1",
        destPath: "/v1"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "platform.rise.ai": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_media_folders";
function createFolder(payload) {
  function __createFolder({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.CreateFolder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/folders",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "folder.createdDate" },
            { path: "folder.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createFolder;
}
function getFolder(payload) {
  function __getFolder({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "GET",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.GetFolder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/folders/{folderId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "folder.createdDate" },
            { path: "folder.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getFolder;
}
function listFolders(payload) {
  function __listFolders({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "GET",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.ListFolders",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/folders",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload, true),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "folders.createdDate" },
            { path: "folders.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listFolders;
}
function searchFolders(payload) {
  function __searchFolders({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.SearchFolders",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/folders/search",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "folders.createdDate" },
            { path: "folders.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __searchFolders;
}
function updateFolder(payload) {
  function __updateFolder({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "folder.createdDate" }, { path: "folder.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "PATCH",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.UpdateFolder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/folders/{folder.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "folder.createdDate" },
            { path: "folder.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateFolder;
}
function generateFolderDownloadUrl(payload) {
  function __generateFolderDownloadUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.GenerateFolderDownloadUrl",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/folders/{folderId}/generate-download-url",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __generateFolderDownloadUrl;
}
function bulkDeleteFolders(payload) {
  function __bulkDeleteFolders({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.BulkDeleteFolders",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/bulk/folders/delete",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkDeleteFolders;
}
function bulkRestoreFoldersFromTrashBin(payload) {
  function __bulkRestoreFoldersFromTrashBin({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.BulkRestoreFoldersFromTrashBin",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/bulk/trash-bin/folders/restore",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkRestoreFoldersFromTrashBin;
}
function listDeletedFolders(payload) {
  function __listDeletedFolders({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "GET",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.ListDeletedFolders",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/trash-bin/folders",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload, true),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "folders.createdDate" },
            { path: "folders.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listDeletedFolders;
}

// src/media-site-media-v1-folder-folders.universal.ts
var State = /* @__PURE__ */ ((State2) => {
  State2["OK"] = "OK";
  State2["DELETED"] = "DELETED";
  return State2;
})(State || {});
var Namespace = /* @__PURE__ */ ((Namespace2) => {
  Namespace2["NO_NAMESPACE"] = "NO_NAMESPACE";
  Namespace2["OTHERS"] = "OTHERS";
  Namespace2["WIX_VIDEO"] = "WIX_VIDEO";
  Namespace2["WIX_MUSIC"] = "WIX_MUSIC";
  Namespace2["ALBUMS_AND_ART_STORE"] = "ALBUMS_AND_ART_STORE";
  Namespace2["WIX_ECOM"] = "WIX_ECOM";
  Namespace2["PHOTO_SHARE_APP"] = "PHOTO_SHARE_APP";
  Namespace2["SHARING_APP"] = "SHARING_APP";
  Namespace2["CHAT"] = "CHAT";
  Namespace2["LOGO_BUILDER"] = "LOGO_BUILDER";
  Namespace2["ALBUMS_OLD"] = "ALBUMS_OLD";
  Namespace2["CHAT_MOBILE"] = "CHAT_MOBILE";
  Namespace2["WIX_FORMS"] = "WIX_FORMS";
  Namespace2["WIX_LEGENDS"] = "WIX_LEGENDS";
  Namespace2["PORTFOLIO"] = "PORTFOLIO";
  return Namespace2;
})(Namespace || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var RootFolder = /* @__PURE__ */ ((RootFolder2) => {
  RootFolder2["MEDIA_ROOT"] = "MEDIA_ROOT";
  RootFolder2["TRASH_ROOT"] = "TRASH_ROOT";
  RootFolder2["VISITOR_UPLOADS_ROOT"] = "VISITOR_UPLOADS_ROOT";
  return RootFolder2;
})(RootFolder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createFolder2(displayName, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    displayName,
    parentFolderId: options?.parentFolderId
  });
  const reqOpts = createFolder(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          displayName: "$[0]",
          parentFolderId: "$[1].parentFolderId"
        },
        singleArgumentUnchanged: false
      },
      ["displayName", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getFolder2(folderId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ folderId });
  const reqOpts = getFolder(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.folder;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { folderId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["folderId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listFolders2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    parentFolderId: options?.parentFolderId,
    sort: options?.sort,
    paging: options?.paging
  });
  const reqOpts = listFolders(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          parentFolderId: "$[0].parentFolderId",
          sort: "$[0].sort",
          paging: "$[0].paging"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function searchFolders2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    rootFolder: options?.rootFolder,
    sort: options?.sort,
    paging: options?.paging,
    search: options?.search
  });
  const reqOpts = searchFolders(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          rootFolder: "$[0].rootFolder",
          sort: "$[0].sort",
          paging: "$[0].paging",
          search: "$[0].search"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateFolder2(_id, folder) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    folder: { ...folder, id: _id }
  });
  const reqOpts = updateFolder(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.folder;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { folder: "$[1]" },
        explicitPathsToArguments: { "folder.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "folder"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function generateFolderDownloadUrl2(folderId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ folderId });
  const reqOpts = generateFolderDownloadUrl(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { folderId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["folderId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkDeleteFolders2(folderIds, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    folderIds,
    permanent: options?.permanent
  });
  const reqOpts = bulkDeleteFolders(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          folderIds: "$[0]",
          permanent: "$[1].permanent"
        },
        singleArgumentUnchanged: false
      },
      ["folderIds", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkRestoreFoldersFromTrashBin2(folderIds) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    folderIds
  });
  const reqOpts = bulkRestoreFoldersFromTrashBin(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { folderIds: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["folderIds"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listDeletedFolders2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    parentFolderId: options?.parentFolderId,
    sort: options?.sort,
    paging: options?.paging
  });
  const reqOpts = listDeletedFolders(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          parentFolderId: "$[0].parentFolderId",
          sort: "$[0].sort",
          paging: "$[0].paging"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
export {
  Namespace,
  RootFolder,
  SortOrder,
  State,
  WebhookIdentityType,
  bulkDeleteFolders2 as bulkDeleteFolders,
  bulkRestoreFoldersFromTrashBin2 as bulkRestoreFoldersFromTrashBin,
  createFolder2 as createFolder,
  generateFolderDownloadUrl2 as generateFolderDownloadUrl,
  getFolder2 as getFolder,
  listDeletedFolders2 as listDeletedFolders,
  listFolders2 as listFolders,
  searchFolders2 as searchFolders,
  updateFolder2 as updateFolder
};
