// @wix/auto_sdk_media_folders/build/es/meta.mjs
// src/media-site-media-v1-folder-folders.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixMediaSiteMediaV1FoldersServiceUrl(opts) {
  const domainToMappings = {
    "apps._base_domain_": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: "/api"
      },
      {
        srcPath: "/_api/wix-privatemedia-proxy-server/v1",
        destPath: "/v1"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/site-media/v1",
        destPath: "/v1"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "platform.rise.ai": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_media_folders";
function createFolder(payload) {
  function __createFolder({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.CreateFolder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/folders",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "folder.createdDate" },
            { path: "folder.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createFolder;
}
function getFolder(payload) {
  function __getFolder({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "GET",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.GetFolder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/folders/{folderId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "folder.createdDate" },
            { path: "folder.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getFolder;
}
function listFolders(payload) {
  function __listFolders({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "GET",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.ListFolders",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/folders",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload, true),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "folders.createdDate" },
            { path: "folders.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listFolders;
}
function searchFolders(payload) {
  function __searchFolders({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.SearchFolders",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/folders/search",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "folders.createdDate" },
            { path: "folders.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __searchFolders;
}
function updateFolder(payload) {
  function __updateFolder({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "folder.createdDate" }, { path: "folder.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "PATCH",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.UpdateFolder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/folders/{folder.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "folder.createdDate" },
            { path: "folder.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateFolder;
}
function generateFolderDownloadUrl(payload) {
  function __generateFolderDownloadUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.GenerateFolderDownloadUrl",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/folders/{folderId}/generate-download-url",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __generateFolderDownloadUrl;
}
function bulkDeleteFolders(payload) {
  function __bulkDeleteFolders({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.BulkDeleteFolders",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/bulk/folders/delete",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkDeleteFolders;
}
function bulkRestoreFoldersFromTrashBin(payload) {
  function __bulkRestoreFoldersFromTrashBin({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.BulkRestoreFoldersFromTrashBin",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/bulk/trash-bin/folders/restore",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkRestoreFoldersFromTrashBin;
}
function listDeletedFolders(payload) {
  function __listDeletedFolders({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "GET",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.ListDeletedFolders",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/trash-bin/folders",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload, true),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "folders.createdDate" },
            { path: "folders.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listDeletedFolders;
}

// src/media-site-media-v1-folder-folders.meta.ts
function createFolder2() {
  const payload = {};
  const getRequestOptions = createFolder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/folders",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getFolder2() {
  const payload = { folderId: ":folderId" };
  const getRequestOptions = getFolder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/folders/{folderId}",
    pathParams: { folderId: "folderId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listFolders2() {
  const payload = {};
  const getRequestOptions = listFolders(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/folders",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function searchFolders2() {
  const payload = {};
  const getRequestOptions = searchFolders(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/folders/search",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateFolder2() {
  const payload = { folder: { id: ":folderId" } };
  const getRequestOptions = updateFolder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/folders/{folder.id}",
    pathParams: { folderId: "folderId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function generateFolderDownloadUrl2() {
  const payload = { folderId: ":folderId" };
  const getRequestOptions = generateFolderDownloadUrl(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/folders/{folderId}/generate-download-url",
    pathParams: { folderId: "folderId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkDeleteFolders2() {
  const payload = {};
  const getRequestOptions = bulkDeleteFolders(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/folders/delete",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkRestoreFoldersFromTrashBin2() {
  const payload = {};
  const getRequestOptions = bulkRestoreFoldersFromTrashBin(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/trash-bin/folders/restore",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listDeletedFolders2() {
  const payload = {};
  const getRequestOptions = listDeletedFolders(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/trash-bin/folders",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  bulkDeleteFolders2 as bulkDeleteFolders,
  bulkRestoreFoldersFromTrashBin2 as bulkRestoreFoldersFromTrashBin,
  createFolder2 as createFolder,
  generateFolderDownloadUrl2 as generateFolderDownloadUrl,
  getFolder2 as getFolder,
  listDeletedFolders2 as listDeletedFolders,
  listFolders2 as listFolders,
  searchFolders2 as searchFolders,
  updateFolder2 as updateFolder
};
