// @wix/auto_sdk_restaurants_local-delivery-spi/build/es/meta.mjs
// src/localdelivery-spi-v1-tpa-local-delivery-spi.http.ts
import { transformSDKFloatToRESTFloat } from "@wix/sdk-runtime/transformations/float";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressLocaldeliverySpiV1LocalDeliveryProviderUrl(opts) {
  const domainToMappings = {};
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_local-delivery-spi";
function listAccountIds(payload) {
  function __listAccountIds({ host }) {
    const metadata = {
      entityFqdn: "wix.localdelivery.spi.v1.tpa",
      method: "POST",
      methodFqn: "com.wixpress.localdelivery.spi.v1.LocalDeliveryProvider.ListAccountIds",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressLocaldeliverySpiV1LocalDeliveryProviderUrl({
        protoPath: "/account-ids",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __listAccountIds;
}
function businessStatus(payload) {
  function __businessStatus({ host }) {
    const metadata = {
      entityFqdn: "wix.localdelivery.spi.v1.tpa",
      method: "POST",
      methodFqn: "com.wixpress.localdelivery.spi.v1.LocalDeliveryProvider.BusinessStatus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressLocaldeliverySpiV1LocalDeliveryProviderUrl({
        protoPath: "/business-status",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __businessStatus;
}
function estimateDelivery(payload) {
  function __estimateDelivery({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "pickupTime" }, { path: "dropoffTime" }]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "pickupAddress.geocode.latitude" },
          { path: "pickupAddress.geocode.longitude" },
          { path: "dropoffAddress.geocode.latitude" },
          { path: "dropoffAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.localdelivery.spi.v1.tpa",
      method: "POST",
      methodFqn: "com.wixpress.localdelivery.spi.v1.LocalDeliveryProvider.EstimateDelivery",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressLocaldeliverySpiV1LocalDeliveryProviderUrl({
        protoPath: "/estimate-delivery",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "promisedPickupTime" },
            { path: "promisedDropoffTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __estimateDelivery;
}
function createDelivery(payload) {
  function __createDelivery({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "pickupTime" }, { path: "dropoffTime" }]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "pickupAddress.geocode.latitude" },
          { path: "pickupAddress.geocode.longitude" },
          { path: "dropoffAddress.geocode.latitude" },
          { path: "dropoffAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.localdelivery.spi.v1.tpa",
      method: "POST",
      methodFqn: "com.wixpress.localdelivery.spi.v1.LocalDeliveryProvider.CreateDelivery",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressLocaldeliverySpiV1LocalDeliveryProviderUrl({
        protoPath: "/create-delivery",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "pickupWindowStartTime" },
            { path: "pickupWindowEndTime" },
            { path: "dropoffWindowStartTime" },
            { path: "dropoffWindowEndTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createDelivery;
}

// src/localdelivery-spi-v1-tpa-local-delivery-spi.types.ts
var BusinessStatus = /* @__PURE__ */ ((BusinessStatus2) => {
  BusinessStatus2["NONE"] = "NONE";
  BusinessStatus2["READY"] = "READY";
  BusinessStatus2["PENDING"] = "PENDING";
  BusinessStatus2["ERROR"] = "ERROR";
  BusinessStatus2["NOT_STARTED"] = "NOT_STARTED";
  return BusinessStatus2;
})(BusinessStatus || {});

// src/localdelivery-spi-v1-tpa-local-delivery-spi.meta.ts
function listAccountIds2() {
  const payload = {};
  const getRequestOptions = listAccountIds(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/account-ids",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function businessStatus2() {
  const payload = {};
  const getRequestOptions = businessStatus(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/business-status",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function estimateDelivery2() {
  const payload = {};
  const getRequestOptions = estimateDelivery(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/estimate-delivery",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createDelivery2() {
  const payload = {};
  const getRequestOptions = createDelivery(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/create-delivery",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  BusinessStatus as BusinessStatusOriginal,
  businessStatus2 as businessStatus,
  createDelivery2 as createDelivery,
  estimateDelivery2 as estimateDelivery,
  listAccountIds2 as listAccountIds
};
