// @wix/bex-core/dist/cjs/state/TagsState/TagsBIReporter.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.TagsBIReporter = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _v = require("@wix/bex-utils/@wix/bi-logger-os-data/v2");
var _events = _interopRequireDefault(require("events"));
var _util = require("../../util");
var _isHttpError = require("../../util/isHttpError");
const ACTIONS = {
  'Add tag': 'Add',
  'Rename tag': 'Rename',
  'Delete tag': 'Delete'
};
const withoutDefaults = (0, _util.withoutIrrelevant)();
class TagsBIReporter {
  constructor(params) {
    (0, _defineProperty2.default)(this, "collection", void 0);
    (0, _defineProperty2.default)(this, "reportBi", void 0);
    (0, _defineProperty2.default)(this, "location", void 0);
    (0, _defineProperty2.default)(this, "events", new _events.default());
    (0, _defineProperty2.default)(this, "actionsMade", {
      Add: false,
      Delete: false,
      Rename: false
    });
    this.collection = params.collection;
    this.reportBi = params.reportBi;
    this.location = params.location;
  }
  set setActionsMade(actionName) {
    if (!(actionName in ACTIONS)) {
      throw Error(`Unknown action ${actionName} was invoked`);
    }
    const action = ACTIONS[actionName];
    this.actionsMade = {
      ...this.actionsMade,
      [action]: true
    };
  }
  resetActionsMade() {
    this.actionsMade = {
      Add: false,
      Delete: false,
      Rename: false
    };
  }
  _clearAllEvents() {
    const {
      events
    } = this;
    events.removeAllListeners('select');
    events.removeAllListeners('blur');
    events.removeAllListeners('add');
    events.removeAllListeners('clear');
  }
  _getErrorType(e) {
    var _e$response, _e$response2;
    let isNetworkError;
    const details = (0, _isHttpError.isHttpError)(e) && (0, _isHttpError.isErrorDetails)(e == null || (_e$response = e.response) == null || (_e$response = _e$response.data) == null ? void 0 : _e$response.details) ? e == null || (_e$response2 = e.response) == null || (_e$response2 = _e$response2.data) == null ? void 0 : _e$response2.details : null;
    if (details) {
      var _details$applicationE;
      isNetworkError = ((_details$applicationE = details.applicationError) == null ? void 0 : _details$applicationE.code) === 'ERR_NETWORK';
    } else {
      isNetworkError = (e == null ? void 0 : e.code) === 'ERR_NETWORK';
    }
    return isNetworkError ? 'Network error' : 'Technical issue';
  }
  onSearch(searchValue) {
    const {
      events
    } = this;
    const afterSearchParams = {
      searchResultsCnt: this.collection.result.items.filter(item => item.name.toLowerCase().includes(searchValue.toLowerCase())).length,
      searchTerm: searchValue
    };
    const reportTagSelectedSearchBI = () => {
      this._clearAllEvents();
      this.reportBi(withoutDefaults(_v.cairoSearchForTags)({
        ...afterSearchParams,
        location: this.location,
        endType: 'Tag selected'
      }));
    };
    const reportSearchDismissedSearchBI = () => {
      this._clearAllEvents();
      this.reportBi(withoutDefaults(_v.cairoSearchForTags)({
        ...afterSearchParams,
        location: this.location,
        endType: 'Search dismissed'
      }));
    };
    const reportAddNewFromSearchBI = () => {
      this._clearAllEvents();
      this.reportBi(withoutDefaults(_v.cairoSearchForTags)({
        ...afterSearchParams,
        location: this.location,
        endType: 'Add new tag'
      }));
    };
    this._clearAllEvents();
    events.once('select', reportTagSelectedSearchBI);
    events.once('blur', reportSearchDismissedSearchBI);
    events.once('clear', reportSearchDismissedSearchBI);
    events.once('add', reportAddNewFromSearchBI);
  }
  onAction(params) {
    if (params.actionName) {
      this.setActionsMade = params.actionName;
    }
    const afterActionParams = {
      numTags: this.collection.result.items.length
    };
    this.reportBi(withoutDefaults(_v.cairoTagsAddDeleteTag)({
      ...params,
      ...afterActionParams
    }));
  }
  onError(params, error) {
    const errorParams = {
      numTags: this.collection.result.items.length,
      errorType: this._getErrorType(error)
    };
    this.reportBi(withoutDefaults(_v.cairoTagsUnsuccessfulUpdateInServer)({
      ...errorParams,
      ...params
    }));
  }
  onTryAgain(params) {
    const tryAgainParams = {
      loaction: 'Toast'
    };
    this.reportBi(withoutDefaults(_v.cairoTryAgainClicked)({
      ...tryAgainParams,
      ...params
    }));
  }
  onUndo(params) {
    this.reportBi(withoutDefaults(_v.cairoUndoClicked)(params));
  }
  getActionsMadeChangesString() {
    const actionsMadeArray = [];
    this.actionsMade.Add && actionsMadeArray.push('Add');
    this.actionsMade.Delete && actionsMadeArray.push('Delete');
    this.actionsMade.Rename && actionsMadeArray.push('Rename');
    return actionsMadeArray.join('/');
  }
  onModalClose(params) {
    let endType = this.getActionsMadeChangesString();
    if (endType) {
      endType = `Close with changes(${endType})`;
    } else {
      endType = 'Close without changes';
    }
    this.reportBi(withoutDefaults(_v.cairoTagsModalEndProcess)({
      ...params,
      numTags: this.collection.result.items.length,
      endType
    }));
    this.resetActionsMade();
  }
}
exports.TagsBIReporter = TagsBIReporter;
