// @wix/bex-core/dist/cjs/state/TagsState/TagsWidgetState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.TagsWidgetState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _mobx = require("mobx");
var _TagsManagementState = require("./TagsManagementState");
var _TagsBIReporter = require("./TagsBIReporter");
var _TaskState = require("../TaskState");
var _services = require("../../services");
class TagsWidgetState {
  constructor({
    container,
    origin,
    ...tagsManagementStateParams
  }) {
    (0, _defineProperty2.default)(this, "tagsFormState", null);
    (0, _defineProperty2.default)(this, "tagsManagementState", void 0);
    (0, _defineProperty2.default)(this, "initTask", new _TaskState.TaskState());
    (0, _defineProperty2.default)(this, "tagsBIReporter", void 0);
    (0, _defineProperty2.default)(this, "container", void 0);
    (0, _defineProperty2.default)(this, "origin", void 0);
    (0, _defineProperty2.default)(this, "retry", () => {
      this.tagsBIReporter.onTryAgain({
        actionName: 'load tags widget',
        loaction: 'Component error state'
      });
      this.initTask.runOnce();
    });
    this.container = container;
    this.origin = origin;
    this.tagsManagementState = new _TagsManagementState.TagsManagementState({
      ...tagsManagementStateParams,
      container
    });
    this.tagsBIReporter = new _TagsBIReporter.TagsBIReporter({
      collection: this.tagsManagementState.collection,
      reportBi: this.container.createBILogger({
        componentType: 'TagsWidget',
        ...tagsManagementStateParams.fqdn
      }),
      location: 'Tags widget'
    });
    (0, _mobx.makeObservable)(this, {
      init: _mobx.action,
      tagsFormState: _mobx.observable,
      isDirty: _mobx.computed,
      retry: _mobx.action.bound
    });
  }
  init() {
    const {
      initTask
    } = this;
    const disposers = [this.tagsManagementState.init()];
    initTask.runOnce(async () => {
      var _this$tagsManagementS;
      this.tagsManagementState.initTask.runOnce();
      await ((_this$tagsManagementS = this.tagsManagementState.initTask.status.promise) == null ? void 0 : _this$tagsManagementS.catch(error => {
        this.tagsBIReporter.onError({
          actionName: 'Load tags widget',
          origin: this.origin
        }, error);
        this.container.internalMonitor.reportError({
          error: new _services.WixPatternsError({
            errorCode: 'GetTagsFailed',
            originalException: error
          }),
          category: 'Tags'
        });
        throw error;
      }));
    });
    return () => {
      disposers.forEach(disposer => disposer == null ? void 0 : disposer());
    };
  }
  get isLoading() {
    return this.initTask.status.isLoading;
  }
  get isError() {
    return this.initTask.status.isError;
  }
  async validate() {
    var _this$tagsFormState;
    const data = await ((_this$tagsFormState = this.tagsFormState) == null ? void 0 : _this$tagsFormState.validate());
    return data || {
      isValid: true
    };
  }
  get isDirty() {
    var _this$tagsFormState2;
    return ((_this$tagsFormState2 = this.tagsFormState) == null ? void 0 : _this$tagsFormState2.isDirty) || false;
  }
}
exports.TagsWidgetState = TagsWidgetState;
