// @wix/bex-core/dist/cjs/state/TagsState/TagsSearchState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.TagsSearchState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _TagsManagementState = require("./TagsManagementState");
var _mobx = require("mobx");
class TagsSearchState {
  constructor({
    tagsManagementState
  }) {
    (0, _defineProperty2.default)(this, "tagsManagementState", void 0);
    (0, _defineProperty2.default)(this, "searchValue", '');
    (0, _defineProperty2.default)(this, "setSearchValue", newSearchValue => {
      this.searchValue = newSearchValue;
    });
    this.tagsManagementState = tagsManagementState;
    (0, _mobx.makeObservable)(this, {
      searchValue: _mobx.observable,
      setSearchValue: _mobx.action,
      filteredItems: _mobx.computed,
      hasSameTagAsSearch: _mobx.computed
    });
  }
  get filteredItems() {
    if (!this.searchValue) {
      return this.tagsManagementState.items;
    }
    const filteredTags = this.tagsManagementState.items.filter(tag => {
      return tag.name.toLowerCase().includes(this.searchValue.toLowerCase());
    });
    if (filteredTags.length === 0 && this.searchValue.length > _TagsManagementState.TAG_MAX_LENGTH) {
      const truncatedTag = this.tagsManagementState.items.find(tag => tag.name.toLowerCase() === this.searchValue.substring(0, _TagsManagementState.TAG_MAX_LENGTH).toLowerCase());
      if (truncatedTag) {
        return [truncatedTag];
      }
    }
    return filteredTags;
  }
  get hasSameTagAsSearch() {
    return this.tagsManagementState.items.some(tag => {
      return tag.name.toLowerCase() === this.searchValue.substring(0, _TagsManagementState.TAG_MAX_LENGTH).toLowerCase();
    });
  }
}
exports.TagsSearchState = TagsSearchState;
