// @wix/bex-core/dist/cjs/testkit/createCountriesBackend.js
"use strict";

exports.__esModule = true;
exports.createCountriesBackend = void 0;
var _chance = require("chance");
var _lodash = require("lodash");
var _InMemoryBackend = require("../InMemoryBackend");
const createCountriesBackend = ({
  createOne,
  ...params
} = {}) => {
  const chance = new _chance.Chance();
  const countries = chance.unique(chance.country.bind(chance, {
    full: true
  }), chance.countries().length).sort().map(name => ({
    id: (0, _lodash.snakeCase)(name),
    name
  }));
  function createCountry(index, {
    ...rest
  } = {}) {
    return {
      ...countries[index],
      ...rest
    };
  }
  return new _InMemoryBackend.InMemoryBackend({
    enableTotal: params == null ? void 0 : params.enableTotal,
    createOne: createOne ?? (index => createCountry(index)),
    itemKey: item => item.id,
    total: countries.length,
    predicate: ({
      search
    }) => {
      const rgx = new RegExp((0, _lodash.escapeRegExp)(search), 'i');
      return item => {
        if (search && !rgx.test(item.name)) {
          return false;
        }
        return true;
      };
    },
    ...params
  });
};
exports.createCountriesBackend = createCountriesBackend;
