// @wix/bex-core/dist/cjs/testkit/createProductsBackend.js
"use strict";

exports.__esModule = true;
exports.createProduct = createProduct;
exports.createProductsBackend = void 0;
var _chance = require("chance");
var _lodash = require("lodash");
var _InMemoryBackend = require("../InMemoryBackend");
function createProduct(chance, {
  createdAt = chance.date({
    min: new Date(Date.now() - 365 * 24 * 60 * 60 * 1000),
    max: new Date(Date.now() - 2 * 24 * 60 * 60 * 1000)
  }),
  ...rest
} = {}) {
  return {
    id: chance.guid(),
    name: chance.name(),
    description: chance.sentence(),
    price: chance.integer({
      min: 0,
      max: 10000
    }),
    img: chance.url(),
    status: chance.pickone(['available', 'outOfStock']),
    createdAt,
    updatedAt: createdAt,
    country: (0, _lodash.snakeCase)(chance.country({
      full: true
    })),
    ...rest
  };
}
const createProductsBackend = ({
  createOne,
  ...params
} = {}) => {
  const chance = new _chance.Chance();
  return new _InMemoryBackend.InMemoryBackend({
    enableTotal: params == null ? void 0 : params.enableTotal,
    createOne: createOne ?? (() => createProduct(chance)),
    itemKey: item => item.id,
    total: 1000,
    predicate: query => {
      const {
        search
      } = query;
      const filters = query.filters ?? {};
      const rgx = new RegExp((0, _lodash.escapeRegExp)(search), 'i');
      return item => {
        var _filters$status$, _filters$updatedAt, _filters$updatedAt2;
        if (search && !rgx.test(item.name)) {
          return false;
        }
        if (filters.productId && filters.productId.length && filters.productId.indexOf(item.id) === -1) {
          return false;
        }
        if (filters.status && filters.status.length && ((_filters$status$ = filters.status[0]) == null ? void 0 : _filters$status$.id) !== 'all' && !filters.status.some(status => status.id === item.status)) {
          return false;
        }
        if (item.updatedAt && (_filters$updatedAt = filters.updatedAt) != null && _filters$updatedAt.from && filters.updatedAt.from.valueOf() > item.updatedAt.valueOf()) {
          return false;
        }
        if (item.updatedAt && (_filters$updatedAt2 = filters.updatedAt) != null && _filters$updatedAt2.to && filters.updatedAt.to.valueOf() < item.updatedAt.valueOf()) {
          return false;
        }
        if (item.country && filters.country && filters.country.length !== 0 && !filters.country.some(country => country.id === item.country)) {
          return false;
        }
        return true;
      };
    },
    orderBy: query => {
      return [...(query.sort ?? [])];
    },
    ...params
  });
};
exports.createProductsBackend = createProductsBackend;
