// @wix/auto_sdk_seo_seo-tags/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  resolveItemSeoTags: () => resolveItemSeoTags2,
  resolveSeoTags: () => resolveSeoTags2,
  resolveStaticPageSeoTags: () => resolveStaticPageSeoTags2
});
module.exports = __toCommonJS(meta_exports);

// src/promote-seo-v1-seo-tags-seo-tags.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixPromoteSeoTagsServiceV1SeoTagsServiceUrl(opts) {
  const domainToMappings = {
    _: [
      {
        srcPath: "/_api/seo-tags-service/v1/seo-tags",
        destPath: "/v1/seo-tags"
      },
      {
        srcPath: "/_api/seo-tags-service/v1/resolve-seo-tags",
        destPath: "/v1/resolve-seo-tags"
      },
      {
        srcPath: "/_api/seo-tags-service/v1/resolve-static-page-seo-tags",
        destPath: "/v1/resolve-static-page-seo-tags"
      },
      {
        srcPath: "/_api/seo-tags-service/v1/resolve-item-seo-tags",
        destPath: "/v1/resolve-item-seo-tags"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/seo-tags-service/v1/resolve-seo-tags",
        destPath: "/v1/resolve-seo-tags"
      },
      {
        srcPath: "/_api/seo-tags-service/v1/resolve-static-page-seo-tags",
        destPath: "/v1/resolve-static-page-seo-tags"
      },
      {
        srcPath: "/_api/seo-tags-service/v1/resolve-item-seo-tags",
        destPath: "/v1/resolve-item-seo-tags"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/promote/seo/v1/resolve-seo-tags",
        destPath: "/v1/resolve-seo-tags"
      },
      {
        srcPath: "/promote/seo/v1/resolve-static-page-seo-tags",
        destPath: "/v1/resolve-static-page-seo-tags"
      },
      {
        srcPath: "/promote/seo/v1/resolve-item-seo-tags",
        destPath: "/v1/resolve-item-seo-tags"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_seo_seo-tags";
function resolveSeoTags(payload) {
  function __resolveSeoTags({ host }) {
    const metadata = {
      entityFqdn: "wix.promote.seo.v1.seo_tags",
      method: "GET",
      methodFqn: "wix.promote.seo.tags.service.v1.SeoTagsService.ResolveSeoTags",
      packageName: PACKAGE_NAME,
      url: resolveWixPromoteSeoTagsServiceV1SeoTagsServiceUrl({
        protoPath: "/v1/resolve-seo-tags",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload, true)
    };
    return metadata;
  }
  return __resolveSeoTags;
}
function resolveStaticPageSeoTags(payload) {
  function __resolveStaticPageSeoTags({ host }) {
    const metadata = {
      entityFqdn: "wix.promote.seo.v1.seo_tags",
      method: "GET",
      methodFqn: "wix.promote.seo.tags.service.v1.SeoTagsService.ResolveStaticPageSeoTags",
      packageName: PACKAGE_NAME,
      url: resolveWixPromoteSeoTagsServiceV1SeoTagsServiceUrl({
        protoPath: "/v1/resolve-static-page-seo-tags",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload, true)
    };
    return metadata;
  }
  return __resolveStaticPageSeoTags;
}
function resolveItemSeoTags(payload) {
  function __resolveItemSeoTags({ host }) {
    const metadata = {
      entityFqdn: "wix.promote.seo.v1.seo_tags",
      method: "GET",
      methodFqn: "wix.promote.seo.tags.service.v1.SeoTagsService.ResolveItemSeoTags",
      packageName: PACKAGE_NAME,
      url: resolveWixPromoteSeoTagsServiceV1SeoTagsServiceUrl({
        protoPath: "/v1/resolve-item-seo-tags",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __resolveItemSeoTags;
}

// src/promote-seo-v1-seo-tags-seo-tags.meta.ts
function resolveSeoTags2() {
  const payload = {};
  const getRequestOptions = resolveSeoTags(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/resolve-seo-tags",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function resolveStaticPageSeoTags2() {
  const payload = {};
  const getRequestOptions = resolveStaticPageSeoTags(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/resolve-static-page-seo-tags",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function resolveItemSeoTags2() {
  const payload = {};
  const getRequestOptions = resolveItemSeoTags(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/resolve-item-seo-tags",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  resolveItemSeoTags,
  resolveSeoTags,
  resolveStaticPageSeoTags
});
