// @wix/ricos/dist/cjs/plugin-figure/converter/caption-converter.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var caption_converter_exports = {};
__export(caption_converter_exports, {
  captionConverter: () => captionConverter,
  createCaptionTiptapNode: () => createCaptionTiptapNode
});
module.exports = __toCommonJS(caption_converter_exports);
var import_ricos_content = require("./../../ricos-content/index.js");
var import_v1 = require("./../../ricos-schema/legacy/wix/rich_content/v1/index.js");
var import_consts = require("./../../_shared/figure/consts.js");
const captionConverter = {
  toPm: {
    types: [import_v1.Node_Type.CAPTION],
    // @ts-ignore $TSFixMe
    convert: (node, visit) => {
      const { style, captionData } = node || {};
      return {
        type: import_v1.Node_Type.CAPTION,
        attrs: {
          ...captionData,
          ...style ? { style } : {},
          id: node.id ?? (0, import_ricos_content.generateId)()
        },
        content: visit(node)
      };
    }
  },
  fromPm: {
    types: [import_v1.Node_Type.CAPTION],
    // @ts-ignore $TSFixMe
    convert: (node, visit) => {
      const { id, style, dir: _, textStyle } = node.attrs || {};
      const transformedStyle = {
        paddingBottom: excludeDefaultValue(
          style?.paddingBottom,
          import_consts.captionDefaultAttributes.styleDefaults.paddingBottom
        ),
        paddingTop: excludeDefaultValue(
          style?.paddingTop,
          import_consts.captionDefaultAttributes.styleDefaults.paddingTop
        )
      };
      const transformedTextStyle = {
        textAlignment: excludeDefaultValue(
          textStyle?.textAlignment,
          import_consts.captionDefaultAttributes.textStyle.textAlignment
        ),
        lineHeight: excludeDefaultValue(
          textStyle?.lineHeight,
          import_consts.captionDefaultAttributes.textStyle.lineHeight
        )
      };
      return {
        type: import_v1.Node_Type.CAPTION,
        id,
        nodes: visit(node),
        ...Object.values(transformedStyle).some((v) => v !== void 0) && {
          style: transformedStyle
        },
        ...Object.values(transformedTextStyle).some((v) => v !== void 0) && {
          captionData: { textStyle: transformedTextStyle }
        }
      };
    }
  }
};
const createCaptionTiptapNode = (content) => {
  return {
    type: import_v1.Node_Type.CAPTION,
    attrs: {
      id: (0, import_ricos_content.generateId)()
    },
    content: content ?? []
  };
};
function excludeDefaultValue(value, defaultValue) {
  return value === defaultValue ? void 0 : value;
}
