// @wix/headless-online-programs/dist/react/Category.js
import { jsx as _jsx } from "react/jsx-runtime";
import React from 'react';
import { AsChildSlot } from '@wix/headless-utils/react';
var TestIds;
(function (TestIds) {
    TestIds["categoryId"] = "program-category-id";
    TestIds["categoryName"] = "program-category-label";
})(TestIds || (TestIds = {}));
const CategoryContext = React.createContext(null);
function useCategoryContext() {
    const context = React.useContext(CategoryContext);
    if (!context) {
        throw new Error('useCategoryContext must be used within a Category.Root component');
    }
    return context;
}
/**
 * Root component that provides the Category context to its children.
 *
 * @order 1
 * @component
 * @example
 * ```tsx
 * import { Category } from '@wix/online-programs/components';
 *
 * function CategoryCard(props) {
 *  const { category } = props;
 *
 *   return (
 *    <Category.Root category={category}>
 *      <Category.Label />
 *      <Category.Id />
 *    </Category.Root>
 *   );
 * }
 * ```
 */
function Root(props) {
    const { category, children } = props;
    return (_jsx(CategoryContext.Provider, { value: {
            category,
        }, children: children }));
}
Root.displayName = 'Category.Root';
/**
 * Headless component for category id display
 *
 * @component
 * @example
 * ```tsx
 * import { Category } from '@wix/online-programs/components';
 *
 * function CategoryCard(props) {
 *  const { category } = props;
 *
 *   return (
 *    <Category.Root category={category}>
 *      <Category.Id>
 *        {({ id }) => (
 *          <span>Category ID: {id}</span>
 *        )}
 *      </Category.Id>
 *    </Category.Root>
 *   );
 * }
 * ```
 */
const Id = React.forwardRef((props, ref) => {
    const { asChild, children, className } = props;
    const { category } = useCategoryContext();
    const { _id: id } = category;
    if (!id) {
        return null;
    }
    return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.categoryId, customElement: children, customElementProps: { id }, content: id, children: _jsx("span", { children: id }) }));
});
Id.displayName = 'Category.Id';
/**
 * Headless component for category id display
 *
 * @component
 * @example
 * ```tsx
 * import { Category } from '@wix/online-programs/components';
 *
 * function CategoryCard(props) {
 *  const { category } = props;
 *
 *   return (
 *    <Category.Root category={category}>
 *      <Category.Label />
 *    </Category.Root>
 *   );
 * }
 * ```
 */
const Label = React.forwardRef((props, ref) => {
    const { asChild, children, className } = props;
    const { category } = useCategoryContext();
    const { label } = category;
    if (!label) {
        return null;
    }
    return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.categoryName, customElement: children, customElementProps: { label }, content: label, children: _jsx("span", { children: label }) }));
});
Label.displayName = 'Category.Label';
/**
 * Compound component for Category with all sub-components
 */
export const Category = {
    Root,
    Id,
    Label,
};
