// @wix/auto_sdk_members_member-role-definition/build/internal/es/meta.mjs
// src/members-v1-member-role-definition-member-role-definition.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressMembersMemberrolesMemberRoleDefinitionsUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/members/v1/roles",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "members.wixapps.net": [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_member-role-definition";
function createMemberRoleDefinition(payload) {
  function __createMemberRoleDefinition({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_role_definition",
      method: "POST",
      methodFqn: "com.wixpress.members.memberroles.MemberRoleDefinitions.CreateMemberRoleDefinition",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembersMemberrolesMemberRoleDefinitionsUrl({
        protoPath: "/v1/roles/definitions",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __createMemberRoleDefinition;
}
function listMemberRoleDefinitions(payload) {
  function __listMemberRoleDefinitions({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_role_definition",
      method: "GET",
      methodFqn: "com.wixpress.members.memberroles.MemberRoleDefinitions.ListMemberRoleDefinitions",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembersMemberrolesMemberRoleDefinitionsUrl({
        protoPath: "/v1/roles/definitions",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __listMemberRoleDefinitions;
}
function getMemberRoleDefinition(payload) {
  function __getMemberRoleDefinition({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_role_definition",
      method: "GET",
      methodFqn: "com.wixpress.members.memberroles.MemberRoleDefinitions.GetMemberRoleDefinition",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembersMemberrolesMemberRoleDefinitionsUrl({
        protoPath: "/v1/roles/definitions/{roleKey}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getMemberRoleDefinition;
}
function updateMemberRoleDefinition(payload) {
  function __updateMemberRoleDefinition({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_role_definition",
      method: "PATCH",
      methodFqn: "com.wixpress.members.memberroles.MemberRoleDefinitions.UpdateMemberRoleDefinition",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembersMemberrolesMemberRoleDefinitionsUrl({
        protoPath: "/v1/roles/definitions/{key}",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __updateMemberRoleDefinition;
}
function deleteMemberRoleDefinition(payload) {
  function __deleteMemberRoleDefinition({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_role_definition",
      method: "DELETE",
      methodFqn: "com.wixpress.members.memberroles.MemberRoleDefinitions.DeleteMemberRoleDefinition",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembersMemberrolesMemberRoleDefinitionsUrl({
        protoPath: "/v1/roles/definitions/{roleKey}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteMemberRoleDefinition;
}

// src/members-v1-member-role-definition-member-role-definition.types.ts
var State = /* @__PURE__ */ ((State2) => {
  State2["UNKNOWN"] = "UNKNOWN";
  State2["ENABLED"] = "ENABLED";
  State2["DISABLED"] = "DISABLED";
  State2["PENDING"] = "PENDING";
  State2["DEMO"] = "DEMO";
  return State2;
})(State || {});
var SiteCreatedContext = /* @__PURE__ */ ((SiteCreatedContext2) => {
  SiteCreatedContext2["OTHER"] = "OTHER";
  SiteCreatedContext2["FROM_TEMPLATE"] = "FROM_TEMPLATE";
  SiteCreatedContext2["DUPLICATE_BY_SITE_TRANSFER"] = "DUPLICATE_BY_SITE_TRANSFER";
  SiteCreatedContext2["DUPLICATE"] = "DUPLICATE";
  SiteCreatedContext2["OLD_SITE_TRANSFER"] = "OLD_SITE_TRANSFER";
  SiteCreatedContext2["FLASH"] = "FLASH";
  return SiteCreatedContext2;
})(SiteCreatedContext || {});
var Namespace = /* @__PURE__ */ ((Namespace2) => {
  Namespace2["UNKNOWN_NAMESPACE"] = "UNKNOWN_NAMESPACE";
  Namespace2["WIX"] = "WIX";
  Namespace2["SHOUT_OUT"] = "SHOUT_OUT";
  Namespace2["ALBUMS"] = "ALBUMS";
  Namespace2["WIX_STORES_TEST_DRIVE"] = "WIX_STORES_TEST_DRIVE";
  Namespace2["HOTELS"] = "HOTELS";
  Namespace2["CLUBS"] = "CLUBS";
  Namespace2["ONBOARDING_DRAFT"] = "ONBOARDING_DRAFT";
  Namespace2["DEV_SITE"] = "DEV_SITE";
  Namespace2["LOGOS"] = "LOGOS";
  Namespace2["VIDEO_MAKER"] = "VIDEO_MAKER";
  Namespace2["PARTNER_DASHBOARD"] = "PARTNER_DASHBOARD";
  Namespace2["DEV_CENTER_COMPANY"] = "DEV_CENTER_COMPANY";
  Namespace2["HTML_DRAFT"] = "HTML_DRAFT";
  Namespace2["SITELESS_BUSINESS"] = "SITELESS_BUSINESS";
  Namespace2["CREATOR_ECONOMY"] = "CREATOR_ECONOMY";
  Namespace2["DASHBOARD_FIRST"] = "DASHBOARD_FIRST";
  Namespace2["ANYWHERE"] = "ANYWHERE";
  Namespace2["HEADLESS"] = "HEADLESS";
  Namespace2["ACCOUNT_MASTER_CMS"] = "ACCOUNT_MASTER_CMS";
  Namespace2["RISE"] = "RISE";
  Namespace2["BRANDED_FIRST"] = "BRANDED_FIRST";
  Namespace2["NOWNIA"] = "NOWNIA";
  Namespace2["UGC_TEMPLATE"] = "UGC_TEMPLATE";
  Namespace2["CODUX"] = "CODUX";
  Namespace2["MEDIA_DESIGN_CREATOR"] = "MEDIA_DESIGN_CREATOR";
  Namespace2["SHARED_BLOG_ENTERPRISE"] = "SHARED_BLOG_ENTERPRISE";
  Namespace2["STANDALONE_FORMS"] = "STANDALONE_FORMS";
  Namespace2["STANDALONE_EVENTS"] = "STANDALONE_EVENTS";
  Namespace2["MIMIR"] = "MIMIR";
  Namespace2["TWINS"] = "TWINS";
  Namespace2["NANO"] = "NANO";
  return Namespace2;
})(Namespace || {});
var DeleteStatus = /* @__PURE__ */ ((DeleteStatus2) => {
  DeleteStatus2["UNKNOWN"] = "UNKNOWN";
  DeleteStatus2["TRASH"] = "TRASH";
  DeleteStatus2["DELETED"] = "DELETED";
  DeleteStatus2["PENDING_PURGE"] = "PENDING_PURGE";
  DeleteStatus2["PURGED_EXTERNALLY"] = "PURGED_EXTERNALLY";
  return DeleteStatus2;
})(DeleteStatus || {});

// src/members-v1-member-role-definition-member-role-definition.meta.ts
function createMemberRoleDefinition2() {
  const payload = {};
  const getRequestOptions = createMemberRoleDefinition(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/roles/definitions",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listMemberRoleDefinitions2() {
  const payload = {};
  const getRequestOptions = listMemberRoleDefinitions(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/roles/definitions",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getMemberRoleDefinition2() {
  const payload = { roleKey: ":roleKey" };
  const getRequestOptions = getMemberRoleDefinition(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/roles/definitions/{roleKey}",
    pathParams: { roleKey: "roleKey" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateMemberRoleDefinition2() {
  const payload = { key: ":key" };
  const getRequestOptions = updateMemberRoleDefinition(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/roles/definitions/{key}",
    pathParams: { key: "key" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteMemberRoleDefinition2() {
  const payload = { roleKey: ":roleKey" };
  const getRequestOptions = deleteMemberRoleDefinition(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/roles/definitions/{roleKey}",
    pathParams: { roleKey: "roleKey" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  DeleteStatus as DeleteStatusOriginal,
  Namespace as NamespaceOriginal,
  SiteCreatedContext as SiteCreatedContextOriginal,
  State as StateOriginal,
  createMemberRoleDefinition2 as createMemberRoleDefinition,
  deleteMemberRoleDefinition2 as deleteMemberRoleDefinition,
  getMemberRoleDefinition2 as getMemberRoleDefinition,
  listMemberRoleDefinitions2 as listMemberRoleDefinitions,
  updateMemberRoleDefinition2 as updateMemberRoleDefinition
};
