// @wix/auto-patterns/dist/cjs/components/AutoPatternsCollectionPage/AutoPatternsCollectionPage.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.AutoPatternsCollectionPage = void 0;
var _react = _interopRequireDefault(require("react"));
var _page = require("@wix/patterns/page");
var _AutoPatternsCollectionPageContent = require("../AutoPatternsCollectionPageContent");
var _providers = require("../../providers");
var _useCollectionPageActions = require("../../hooks/useCollectionPageActions");
var _jsxFileName = "/home/builduser/work/73d19ce3378ce4dd/packages/auto-patterns/dist/cjs/components/AutoPatternsCollectionPage/AutoPatternsCollectionPage.tsx";
const AutoPatternsCollectionPage = ({
  configuration
}) => {
  const {
    title,
    subtitle
  } = configuration;
  const actions = (0, _useCollectionPageActions.useCollectionPageActions)({
    collectionPageActions: configuration.actions
  });
  const context = (0, _providers.usePatternsWizardOverridesContext)();
  return /*#__PURE__*/_react.default.createElement(_page.CollectionPage, {
    height: "100vh",
    dataHook: "auto-patterns-page",
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 24,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(_page.CollectionPage.Header, {
    title: title,
    subtitle: subtitle,
    primaryAction: actions.primaryActions,
    secondaryActions: actions.secondaryActions,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 25,
      columnNumber: 7
    }
  }), /*#__PURE__*/_react.default.createElement(_page.CollectionPage.Content, {
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 31,
      columnNumber: 7
    }
  }, context != null && context.HeaderAddon ? /*#__PURE__*/_react.default.createElement(context.HeaderAddon, {
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 32,
      columnNumber: 33
    }
  }) : null, /*#__PURE__*/_react.default.createElement(_AutoPatternsCollectionPageContent.AutoPatternsCollectionPageContent, {
    configuration: configuration,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 33,
      columnNumber: 9
    }
  })));
};
exports.AutoPatternsCollectionPage = AutoPatternsCollectionPage;
