// @wix/auto_sdk_events_policies/build/es/index.typings.mjs
// src/events-v2-policy-policies.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/events-v2-policy-policies.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixEventsV2PolicyManagementUrl(opts) {
  const domainToMappings = {
    _: [
      {
        srcPath: "/_api/events-policies",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/events-policies",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/events-policies",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/events-policies",
        destPath: ""
      },
      {
        srcPath: "/events/v2/policies",
        destPath: "/v2/policies"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/events-policies",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/events-policies",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/events-policies",
        destPath: ""
      }
    ],
    "events.wixapps.net": [
      {
        srcPath: "/_api/events-policies",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_events_policies";
function createPolicy(payload) {
  function __createPolicy({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "policy.createdDate" }, { path: "policy.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v2.policy",
      method: "POST",
      methodFqn: "wix.events.v2.PolicyManagement.CreatePolicy",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsV2PolicyManagementUrl({
        protoPath: "/v2/policies",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "policy.createdDate" },
            { path: "policy.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createPolicy;
}
function updatePolicy(payload) {
  function __updatePolicy({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "policy.createdDate" }, { path: "policy.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v2.policy",
      method: "PATCH",
      methodFqn: "wix.events.v2.PolicyManagement.UpdatePolicy",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsV2PolicyManagementUrl({
        protoPath: "/v2/policies/{policy.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "policy.createdDate" },
            { path: "policy.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updatePolicy;
}
function deletePolicy(payload) {
  function __deletePolicy({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v2.policy",
      method: "DELETE",
      methodFqn: "wix.events.v2.PolicyManagement.DeletePolicy",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsV2PolicyManagementUrl({
        protoPath: "/v2/policies/{policyId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deletePolicy;
}
function queryPolicies(payload) {
  function __queryPolicies({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v2.policy",
      method: "POST",
      methodFqn: "wix.events.v2.PolicyManagement.QueryPolicies",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsV2PolicyManagementUrl({
        protoPath: "/v2/policies/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "policies.createdDate" },
            { path: "policies.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryPolicies;
}
function reorderEventPolicies(payload) {
  function __reorderEventPolicies({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v2.policy",
      method: "POST",
      methodFqn: "wix.events.v2.PolicyManagement.ReorderEventPolicies",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsV2PolicyManagementUrl({
        protoPath: "/v2/policies/reorder",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "policies.createdDate" },
            { path: "policies.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __reorderEventPolicies;
}
function getPolicy(payload) {
  function __getPolicy({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v2.policy",
      method: "GET",
      methodFqn: "wix.events.v2.PolicyManagement.GetPolicy",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsV2PolicyManagementUrl({
        protoPath: "/v2/policies/{policyId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "policy.createdDate" },
            { path: "policy.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getPolicy;
}

// src/events-v2-policy-policies.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var LocationType = /* @__PURE__ */ ((LocationType2) => {
  LocationType2["VENUE"] = "VENUE";
  LocationType2["ONLINE"] = "ONLINE";
  return LocationType2;
})(LocationType || {});
var SubdivisionType = /* @__PURE__ */ ((SubdivisionType2) => {
  SubdivisionType2["UNKNOWN_SUBDIVISION_TYPE"] = "UNKNOWN_SUBDIVISION_TYPE";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_1"] = "ADMINISTRATIVE_AREA_LEVEL_1";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_2"] = "ADMINISTRATIVE_AREA_LEVEL_2";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_3"] = "ADMINISTRATIVE_AREA_LEVEL_3";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_4"] = "ADMINISTRATIVE_AREA_LEVEL_4";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_5"] = "ADMINISTRATIVE_AREA_LEVEL_5";
  SubdivisionType2["COUNTRY"] = "COUNTRY";
  return SubdivisionType2;
})(SubdivisionType || {});
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["ONE_TIME"] = "ONE_TIME";
  Status2["RECURRING"] = "RECURRING";
  Status2["RECURRING_NEXT"] = "RECURRING_NEXT";
  Status2["RECURRING_LAST_ENDED"] = "RECURRING_LAST_ENDED";
  Status2["RECURRING_LAST_CANCELED"] = "RECURRING_LAST_CANCELED";
  return Status2;
})(Status || {});
var EventStatus = /* @__PURE__ */ ((EventStatus2) => {
  EventStatus2["SCHEDULED"] = "SCHEDULED";
  EventStatus2["STARTED"] = "STARTED";
  EventStatus2["ENDED"] = "ENDED";
  EventStatus2["CANCELED"] = "CANCELED";
  return EventStatus2;
})(EventStatus || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createPolicy2(policy) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ policy });
  const reqOpts = createPolicy(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.policy;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { policy: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["policy"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updatePolicy2(_id, policy) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    policy: { ...policy, id: _id }
  });
  const reqOpts = updatePolicy(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.policy;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { policy: "$[1]" },
        explicitPathsToArguments: { "policy.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "policy"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deletePolicy2(policyId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ policyId });
  const reqOpts = deletePolicy(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { policyId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["policyId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryPolicies2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryPolicies(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [])
      );
      return {
        items: transformedData?.policies,
        pagingMetadata: transformedData?.metadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function reorderEventPolicies2(policyId, eventId, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    policyId,
    eventId,
    beforePolicyId: options?.beforePolicyId,
    afterPolicyId: options?.afterPolicyId
  });
  const reqOpts = reorderEventPolicies(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          policyId: "$[0]",
          eventId: "$[1]",
          beforePolicyId: "$[2].beforePolicyId",
          afterPolicyId: "$[2].afterPolicyId"
        },
        singleArgumentUnchanged: false
      },
      ["policyId", "eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getPolicy2(policyId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ policyId });
  const reqOpts = getPolicy(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.policy;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { policyId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["policyId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
export {
  EventStatus,
  LocationType,
  SortOrder,
  Status,
  SubdivisionType,
  WebhookIdentityType,
  createPolicy2 as createPolicy,
  deletePolicy2 as deletePolicy,
  getPolicy2 as getPolicy,
  queryPolicies2 as queryPolicies,
  reorderEventPolicies2 as reorderEventPolicies,
  updatePolicy2 as updatePolicy
};
