// @wix/bex-core/dist/esm/state/CollectionPageState/CollectionPageState.js
import { makeObservable, observable } from 'mobx';
import { PageState } from '../PageState';
import { addResizeObserver } from '../resizeObserverCompat';
import { CollectionPageStateBIReporter } from './CollectionPageStateBIReporter';
export class CollectionPageState {
    constructor(params) {
        this.hasHeader = false;
        this.total = false;
        this.hideHeaderTotal = false;
        this.scrollTopBeforeUnmount = null;
        this.startTime = performance.now();
        this.pageState = new PageState();
        this.pageTitle = params.pageTitle;
        this.hasHeader = params.hasHeader;
        this.hideHeaderTotal = params.hideHeaderTotal;
        this.withinRouter = params.withinRouter;
        if (params.reportBI) {
            this.bi = new CollectionPageStateBIReporter({
                reportBI: params.reportBI,
                collectionPageState: this,
            });
        }
        makeObservable(this, {
            pageTitle: observable.ref,
            hasHeader: observable.ref,
            hideHeaderTotal: observable.ref,
            total: observable.ref,
        });
    }
    init() {
        this.bi?.init();
        const scrollableContent = this.pageState.scrollableContentRef.current;
        if (!scrollableContent) {
            return () => { };
        }
        const disposer = addResizeObserver(scrollableContent, () => {
            if (this.scrollTopBeforeUnmount === null) {
                return;
            }
            if (scrollableContent.scrollHeight >= this.scrollTopBeforeUnmount) {
                scrollableContent.scrollTo({
                    top: this.scrollTopBeforeUnmount,
                });
                this.scrollTopBeforeUnmount = null;
            }
        });
        this.bi?.appLoaded?.();
        return () => {
            disposer?.();
        };
    }
}
