// @wix/bex-core/dist/esm/state/CollectionPageState/CollectionPageStateBIReporter.js
import { loadEnd, loadStart } from '../../exports/bi';
import { withoutDefaults } from '../../util';
export class CollectionPageStateBIReporter {
    constructor(params) {
        this.collectionPageState = params.collectionPageState;
        this.reportBI = params.reportBI;
    }
    init() {
        const startTime = performance.now();
        this.appLoaded = this.reportOnLoadStart(startTime);
    }
    reportOnLoadStart(startTime) {
        this.reportBI(withoutDefaults(loadStart)({
            ...this._commonDynamicBiParams(),
        }));
        return () => {
            this.reportOnLoadEnd(Math.round(performance.now() - startTime));
        };
    }
    reportOnLoadEnd(endTime) {
        this.reportBI(withoutDefaults(loadEnd)({
            ...this._commonDynamicBiParams(),
            loadingTime: endTime,
        }));
    }
    _commonDynamicBiParams() {
        return {
            url: window.location.href,
            route: window.location.pathname,
            routerUsage: this.collectionPageState.withinRouter,
        };
    }
}
