// @wix/wix-ui-test-utils/dist/cjs/vanilla/vanilla.js
"use strict";

exports.__esModule = true;
exports.isTestkitExists = isTestkitExists;
exports.isUniTestkitExists = isUniTestkitExists;
exports.testkitFactoryCreator = testkitFactoryCreator;
exports.uniTestkitFactoryCreator = uniTestkitFactoryCreator;
var React = _interopRequireWildcard(require("react"));
var ReactDom = _interopRequireWildcard(require("react-dom"));
var _testUtils = require("react-dom/test-utils");
var _reactHelpers = require("../react-helpers");
var _unidriverJsdomReactLegacy = require("@wix/unidriver-jsdom-react-legacy");
var _jsxFileName = "/home/builduser/work/c98d836f8565ce5f/packages/wix-ui-test-utils/dist/cjs/vanilla/vanilla.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
const getElement = ({
  wrapper,
  dataHook
}) => {
  const domInstance = ReactDom.findDOMNode(wrapper);
  if (domInstance) {
    const dataHookOnInstance = domInstance.attributes.getNamedItem('data-hook') || {
      value: ''
    };
    return dataHook === dataHookOnInstance.value ? domInstance : domInstance.querySelector(`[data-hook='${dataHook}']`);
  }
  return;
};
function testkitFactoryCreator(driverFactory) {
  return testkitArgs => driverFactory({
    element: getElement(testkitArgs),
    wrapper: testkitArgs.wrapper,
    eventTrigger: (0, _reactHelpers.reactEventTrigger)(),
    dataHook: testkitArgs.dataHook
  });
}
function uniTestkitFactoryCreator(driverFactory) {
  return testkitArgs => {
    const element = getElement(testkitArgs);
    return driverFactory((0, _unidriverJsdomReactLegacy.jsdomReactUniDriver)(element), (0, _unidriverJsdomReactLegacy.jsdomReactUniDriver)(document.body), {
      dataHook: testkitArgs.dataHook
    });
  };
}
function isTestkitExists(Element, testkitFactory, options) {
  const div = document.createElement('div');
  const dataHook = 'myDataHook';
  const dataHookPropName = options && options.dataHookPropName;
  const extraProps = dataHookPropName ? {
    [dataHookPropName]: dataHook
  } : {
    'data-hook': dataHook,
    dataHook
  }; // For backward compatibility add dataHook which is used in Wix-Style-React
  const elementToRender = /*#__PURE__*/React.cloneElement(Element, extraProps);
  const renderedElement = (0, _testUtils.renderIntoDocument)(/*#__PURE__*/React.createElement("div", {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 83,
      columnNumber: 46
    }
  }, elementToRender));
  const wrapper = div.appendChild(renderedElement);
  const testkit = testkitFactory({
    wrapper,
    dataHook
  });
  return testkit.exists();
}
async function isUniTestkitExists(Element, testkitFactory, options) {
  const div = document.createElement('div');
  const dataHook = 'myDataHook';
  const dataHookPropName = options && options.dataHookPropName;
  const extraProps = dataHookPropName ? {
    [dataHookPropName]: dataHook
  } : {
    'data-hook': dataHook,
    dataHook
  };
  const elementToRender = /*#__PURE__*/React.cloneElement(Element, extraProps);
  const renderedElement = (0, _testUtils.renderIntoDocument)(/*#__PURE__*/React.createElement("div", {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 101,
      columnNumber: 46
    }
  }, elementToRender));
  const wrapper = div.appendChild(renderedElement);
  const testkit = testkitFactory({
    wrapper,
    dataHook
  });
  return testkit.exists();
}
