// @wix/headless-pricing-plans/dist/react/core/Plan.js
import { jsx as _jsx } from "react/jsx-runtime";
import { createContext, useContext } from 'react';
import { useService, WixServices } from '@wix/services-manager-react';
import { createServicesMap } from '@wix/services-manager';
import { PlanService, PlanServiceDefinition, } from '../../services/index.js';
import { ChannelType, CheckoutService, CheckoutServiceDefinition, } from '@wix/headless-ecom/services';
/**
 * The root container that provides plan context to all child components.
 *
 * @order 1
 * @component
 *
 * @todo: Example
 */
export function Root({ planServiceConfig, children }) {
    return (_jsx(WixServices, { servicesMap: createServicesMap()
            .addService(PlanServiceDefinition, PlanService, planServiceConfig)
            .addService(CheckoutServiceDefinition, CheckoutService, {
            channelType: ChannelType.WEB,
            // TODO: Perhaps we can add postFlowUrl?
            // postFlowUrl: ''
        }), children: children }));
}
export function Plan({ children }) {
    const { isLoadingSignal, planSignal, errorSignal } = useService(PlanServiceDefinition);
    const isLoading = isLoadingSignal.get();
    const error = errorSignal.get();
    if (isLoading) {
        return children({ isLoading: true, error: null, plan: null });
    }
    if (error) {
        return children({ isLoading: false, error, plan: null });
    }
    const plan = planSignal.get();
    return children({ isLoading: false, error: null, plan });
}
export function Image({ children }) {
    const { planSignal } = useService(PlanServiceDefinition);
    const plan = planSignal.get();
    // TODO: How to handle image not being available? Perhaps we need to forward null in case consumer needs to handle it?
    if (!plan || !plan.image) {
        return null;
    }
    return children({ image: plan.image });
}
export function Name({ children }) {
    const { planSignal } = useService(PlanServiceDefinition);
    const plan = planSignal.get();
    if (!plan) {
        return null;
    }
    return children({ name: plan.name });
}
export function Description({ children }) {
    const { planSignal } = useService(PlanServiceDefinition);
    const plan = planSignal.get();
    if (!plan || !plan.description) {
        return null;
    }
    return children({ description: plan.description });
}
export function Price({ children }) {
    const { planSignal } = useService(PlanServiceDefinition);
    const plan = planSignal.get();
    if (!plan) {
        return null;
    }
    return children({ price: plan.enhancedData.price });
}
export function AdditionalFees({ children }) {
    const { planSignal } = useService(PlanServiceDefinition);
    const plan = planSignal.get();
    if (!plan || !plan.enhancedData.additionalFees.length) {
        return null;
    }
    return children({ additionalFees: plan.enhancedData.additionalFees });
}
export function AdditionalFeesRepeater({ children, }) {
    const { planSignal } = useService(PlanServiceDefinition);
    const plan = planSignal.get();
    if (!plan) {
        return null;
    }
    return children({ additionalFees: plan.enhancedData.additionalFees });
}
// Context for sharing current fee data between repeater and individual fee components
const FeeContext = createContext(null);
export function AdditionalFeeRoot({ fee, children }) {
    return _jsx(FeeContext.Provider, { value: fee, children: children });
}
export function AdditionalFeeName({ children }) {
    const fee = useContext(FeeContext);
    if (!fee) {
        return null;
    }
    return children({ name: fee.name });
}
export function AdditionalFeeAmount({ children }) {
    const fee = useContext(FeeContext);
    if (!fee) {
        return null;
    }
    return children({
        amount: fee.amount,
        currency: fee.currency,
        formattedFee: fee.formattedFee,
    });
}
export function Recurrence({ children }) {
    const { planSignal } = useService(PlanServiceDefinition);
    const plan = planSignal.get();
    if (!plan) {
        return null;
    }
    return children({ recurrence: plan.enhancedData.recurrence });
}
export function Duration({ children }) {
    const { planSignal } = useService(PlanServiceDefinition);
    const plan = planSignal.get();
    if (!plan) {
        return null;
    }
    return children({ duration: plan.enhancedData.duration });
}
export function FreeTrialDays({ children }) {
    const { planSignal } = useService(PlanServiceDefinition);
    const plan = planSignal.get();
    if (!plan || !plan.enhancedData.freeTrialDays) {
        return null;
    }
    return children({ freeTrialDays: plan.enhancedData.freeTrialDays });
}
export const PerkItemContext = createContext(null);
export function Perks({ children }) {
    const { planSignal } = useService(PlanServiceDefinition);
    const plan = planSignal.get();
    if (!plan || !plan.perks?.length) {
        return null;
    }
    return children({
        perks: plan.perks?.map((perk) => perk.description) ?? [],
    });
}
export function PerksRepeater({ children }) {
    const { planSignal } = useService(PlanServiceDefinition);
    const plan = planSignal.get();
    if (!plan) {
        return null;
    }
    return children({
        perks: plan.perks?.map((perk) => perk.description) ?? [],
    });
}
export function PerkItem({ children }) {
    const perkItem = useContext(PerkItemContext);
    if (!perkItem) {
        return null;
    }
    return children({ perk: perkItem.perk });
}
