// @wix/auto_sdk_blog_draft-posts/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  ActionOriginal: () => Action,
  AlignmentOriginal: () => Alignment,
  AppTypeOriginal: () => AppType,
  AspectRatioOriginal: () => AspectRatio,
  BackgroundTypeOriginal: () => BackgroundType,
  ButtonDataTypeOriginal: () => ButtonDataType,
  CardStylesTypeOriginal: () => CardStylesType,
  CropOriginal: () => Crop,
  DecorationTypeOriginal: () => DecorationType,
  DirectionOriginal: () => Direction,
  DividerDataAlignmentOriginal: () => DividerDataAlignment,
  FieldOriginal: () => Field,
  FontTypeOriginal: () => FontType,
  GIFTypeOriginal: () => GIFType,
  GetDraftPostsSortOriginal: () => GetDraftPostsSort,
  InitialExpandedItemsOriginal: () => InitialExpandedItems,
  LayoutOriginal: () => Layout,
  LayoutTypeOriginal: () => LayoutType,
  LineStyleOriginal: () => LineStyle,
  MapTypeOriginal: () => MapType,
  ModerationStatusStatusOriginal: () => ModerationStatusStatus,
  NodeTypeOriginal: () => NodeType,
  NullValueOriginal: () => NullValue,
  OrientationOriginal: () => Orientation,
  OriginOriginal: () => Origin,
  PlacementOriginal: () => Placement,
  PluginContainerDataAlignmentOriginal: () => PluginContainerDataAlignment,
  PollLayoutDirectionOriginal: () => PollLayoutDirection,
  PollLayoutTypeOriginal: () => PollLayoutType,
  PositionOriginal: () => Position,
  ResizingOriginal: () => Resizing,
  SortOrderOriginal: () => SortOrder,
  SourceOriginal: () => Source,
  StatusOriginal: () => Status,
  StylesPositionOriginal: () => StylesPosition,
  TargetOriginal: () => Target,
  TextAlignmentOriginal: () => TextAlignment,
  ThumbnailsAlignmentOriginal: () => ThumbnailsAlignment,
  TotalDraftPostsGroupingFieldOriginal: () => TotalDraftPostsGroupingField,
  TypeOriginal: () => Type,
  VerticalAlignmentOriginal: () => VerticalAlignment,
  ViewModeOriginal: () => ViewMode,
  ViewRoleOriginal: () => ViewRole,
  VoteRoleOriginal: () => VoteRole,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  WidthOriginal: () => Width,
  WidthTypeOriginal: () => WidthType,
  bulkCreateDraftPosts: () => bulkCreateDraftPosts2,
  bulkDeleteDraftPosts: () => bulkDeleteDraftPosts2,
  bulkUpdateDraftPosts: () => bulkUpdateDraftPosts2,
  createDraftPost: () => createDraftPost2,
  deleteDraftPost: () => deleteDraftPost2,
  getDeletedDraftPost: () => getDeletedDraftPost2,
  getDraftPost: () => getDraftPost2,
  listDeletedDraftPosts: () => listDeletedDraftPosts2,
  listDraftPosts: () => listDraftPosts2,
  publishDraftPost: () => publishDraftPost2,
  queryDraftPosts: () => queryDraftPosts2,
  removeFromTrashBin: () => removeFromTrashBin2,
  restoreFromTrashBin: () => restoreFromTrashBin2,
  updateDraftPost: () => updateDraftPost2
});
module.exports = __toCommonJS(meta_exports);

// src/blog-v3-draft-draft-posts.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(opts) {
  const domainToMappings = {
    _api_base_domain_: [
      {
        srcPath: "/draft-post-service-api-proxy",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      },
      {
        srcPath: "/blog/v3/bulk/draft-posts/create",
        destPath: "/v3/bulk/draft-posts/create"
      },
      {
        srcPath: "/blog/v3/bulk/draft-posts",
        destPath: "/v3/bulk/draft-posts"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    _: [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      },
      {
        srcPath: "/blog/v3/bulk/draft-posts/create",
        destPath: "/v3/bulk/draft-posts/create"
      },
      {
        srcPath: "/blog/v3/bulk/draft-posts",
        destPath: "/v3/bulk/draft-posts"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      },
      {
        srcPath: "/_api/draft-post-service-api-proxy/v3/bulk/draft-posts",
        destPath: "/v3/bulk/draft-posts"
      }
    ],
    "social-blog._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/blog/v3/draft-posts",
        destPath: "/v3/draft-posts"
      },
      {
        srcPath: "/blog/v3/bulk/draft-posts/create",
        destPath: "/v3/bulk/draft-posts/create"
      },
      {
        srcPath: "/blog/v3/draft-posts/",
        destPath: "/v3/draft-posts"
      },
      {
        srcPath: "/blog/v3/bulk/draft-posts",
        destPath: "/v3/bulk/draft-posts"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_blog_draft-posts";
function createDraftPost(payload) {
  function __createDraftPost({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "draftPost.editedDate" },
          { path: "draftPost.scheduledPublishDate" },
          { path: "draftPost.firstPublishedDate" },
          { path: "draftPost.createdDate" },
          { path: "draftPost.heroImage.urlExpirationDate" },
          { path: "draftPost.richContent.metadata.createdTimestamp" },
          { path: "draftPost.richContent.metadata.updatedTimestamp" },
          { path: "draftPost.moderationDetails.submittedDate" },
          { path: "draftPost.moderationDetails.moderationDate" },
          { path: "draftPost.media.wixMedia.image.urlExpirationDate" },
          { path: "draftPost.media.wixMedia.videoV2.urlExpirationDate" },
          {
            path: "draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate"
          },
          {
            path: "draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
          },
          {
            path: "draftPost.media.wixMedia.videoV2.posters.urlExpirationDate"
          }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "draftPost.heroImage.focalPoint.x" },
          { path: "draftPost.heroImage.focalPoint.y" },
          { path: "draftPost.media.wixMedia.image.focalPoint.x" },
          { path: "draftPost.media.wixMedia.image.focalPoint.y" },
          {
            path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
          },
          {
            path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
          },
          { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.x" },
          { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.y" },
          {
            path: "draftPost.richContent.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "draftPost.richContent.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration"
          },
          {
            path: "draftPost.richContent.nodes.galleryData.options.item.ratio"
          },
          { path: "draftPost.richContent.nodes.imageData.image.duration" },
          { path: "draftPost.richContent.nodes.mapData.mapSettings.lat" },
          { path: "draftPost.richContent.nodes.mapData.mapSettings.lng" },
          { path: "draftPost.richContent.nodes.pollData.poll.image.duration" },
          {
            path: "draftPost.richContent.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "draftPost.richContent.nodes.pollData.design.poll.background.image.duration"
          },
          { path: "draftPost.richContent.nodes.appEmbedData.image.duration" },
          { path: "draftPost.richContent.nodes.videoData.video.duration" },
          { path: "draftPost.richContent.nodes.videoData.thumbnail.duration" },
          { path: "draftPost.richContent.nodes.audioData.audio.duration" },
          { path: "draftPost.richContent.nodes.audioData.coverImage.duration" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "POST",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.CreateDraftPost",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        { protoPath: "/v3/draft-posts", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "draftPost.editedDate" },
            { path: "draftPost.scheduledPublishDate" },
            { path: "draftPost.firstPublishedDate" },
            { path: "draftPost.createdDate" },
            { path: "draftPost.heroImage.urlExpirationDate" },
            { path: "draftPost.richContent.metadata.createdTimestamp" },
            { path: "draftPost.richContent.metadata.updatedTimestamp" },
            { path: "draftPost.moderationDetails.submittedDate" },
            { path: "draftPost.moderationDetails.moderationDate" },
            { path: "draftPost.media.wixMedia.image.urlExpirationDate" },
            { path: "draftPost.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "draftPost.heroImage.focalPoint.x" },
            { path: "draftPost.heroImage.focalPoint.y" },
            { path: "draftPost.media.wixMedia.image.focalPoint.x" },
            { path: "draftPost.media.wixMedia.image.focalPoint.y" },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.x" },
            { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.y" },
            {
              path: "draftPost.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.options.item.ratio"
            },
            { path: "draftPost.richContent.nodes.imageData.image.duration" },
            { path: "draftPost.richContent.nodes.mapData.mapSettings.lat" },
            { path: "draftPost.richContent.nodes.mapData.mapSettings.lng" },
            {
              path: "draftPost.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.appEmbedData.image.duration"
            },
            { path: "draftPost.richContent.nodes.videoData.video.duration" },
            {
              path: "draftPost.richContent.nodes.videoData.thumbnail.duration"
            },
            { path: "draftPost.richContent.nodes.audioData.audio.duration" },
            {
              path: "draftPost.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createDraftPost;
}
function bulkCreateDraftPosts(payload) {
  function __bulkCreateDraftPosts({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "draftPosts.editedDate" },
          { path: "draftPosts.scheduledPublishDate" },
          { path: "draftPosts.firstPublishedDate" },
          { path: "draftPosts.createdDate" },
          { path: "draftPosts.heroImage.urlExpirationDate" },
          { path: "draftPosts.richContent.metadata.createdTimestamp" },
          { path: "draftPosts.richContent.metadata.updatedTimestamp" },
          { path: "draftPosts.moderationDetails.submittedDate" },
          { path: "draftPosts.moderationDetails.moderationDate" },
          { path: "draftPosts.media.wixMedia.image.urlExpirationDate" },
          { path: "draftPosts.media.wixMedia.videoV2.urlExpirationDate" },
          {
            path: "draftPosts.media.wixMedia.videoV2.resolutions.urlExpirationDate"
          },
          {
            path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
          },
          {
            path: "draftPosts.media.wixMedia.videoV2.posters.urlExpirationDate"
          }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "draftPosts.heroImage.focalPoint.x" },
          { path: "draftPosts.heroImage.focalPoint.y" },
          { path: "draftPosts.media.wixMedia.image.focalPoint.x" },
          { path: "draftPosts.media.wixMedia.image.focalPoint.y" },
          {
            path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
          },
          {
            path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
          },
          { path: "draftPosts.media.wixMedia.videoV2.posters.focalPoint.x" },
          { path: "draftPosts.media.wixMedia.videoV2.posters.focalPoint.y" },
          {
            path: "draftPosts.richContent.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "draftPosts.richContent.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "draftPosts.richContent.nodes.galleryData.items.video.thumbnail.duration"
          },
          {
            path: "draftPosts.richContent.nodes.galleryData.options.item.ratio"
          },
          { path: "draftPosts.richContent.nodes.imageData.image.duration" },
          { path: "draftPosts.richContent.nodes.mapData.mapSettings.lat" },
          { path: "draftPosts.richContent.nodes.mapData.mapSettings.lng" },
          { path: "draftPosts.richContent.nodes.pollData.poll.image.duration" },
          {
            path: "draftPosts.richContent.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "draftPosts.richContent.nodes.pollData.design.poll.background.image.duration"
          },
          { path: "draftPosts.richContent.nodes.appEmbedData.image.duration" },
          { path: "draftPosts.richContent.nodes.videoData.video.duration" },
          { path: "draftPosts.richContent.nodes.videoData.thumbnail.duration" },
          { path: "draftPosts.richContent.nodes.audioData.audio.duration" },
          {
            path: "draftPosts.richContent.nodes.audioData.coverImage.duration"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "POST",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.BulkCreateDraftPosts",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        { protoPath: "/v3/bulk/draft-posts/create", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.editedDate" },
            { path: "results.item.scheduledPublishDate" },
            { path: "results.item.firstPublishedDate" },
            { path: "results.item.createdDate" },
            { path: "results.item.heroImage.urlExpirationDate" },
            { path: "results.item.richContent.metadata.createdTimestamp" },
            { path: "results.item.richContent.metadata.updatedTimestamp" },
            { path: "results.item.moderationDetails.submittedDate" },
            { path: "results.item.moderationDetails.moderationDate" },
            { path: "results.item.media.wixMedia.image.urlExpirationDate" },
            { path: "results.item.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.heroImage.focalPoint.x" },
            { path: "results.item.heroImage.focalPoint.y" },
            { path: "results.item.media.wixMedia.image.focalPoint.x" },
            { path: "results.item.media.wixMedia.image.focalPoint.y" },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            {
              path: "results.item.media.wixMedia.videoV2.posters.focalPoint.x"
            },
            {
              path: "results.item.media.wixMedia.videoV2.posters.focalPoint.y"
            },
            {
              path: "results.item.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "results.item.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "results.item.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "results.item.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "results.item.richContent.nodes.imageData.image.duration"
            },
            {
              path: "results.item.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "results.item.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "results.item.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "results.item.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "results.item.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "results.item.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "results.item.richContent.nodes.videoData.video.duration"
            },
            {
              path: "results.item.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "results.item.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "results.item.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateDraftPosts;
}
function bulkUpdateDraftPosts(payload) {
  function __bulkUpdateDraftPosts({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "scheduledPublishDate" },
          { path: "draftPosts.draftPost.editedDate" },
          { path: "draftPosts.draftPost.scheduledPublishDate" },
          { path: "draftPosts.draftPost.firstPublishedDate" },
          { path: "draftPosts.draftPost.createdDate" },
          { path: "draftPosts.draftPost.heroImage.urlExpirationDate" },
          {
            path: "draftPosts.draftPost.richContent.metadata.createdTimestamp"
          },
          {
            path: "draftPosts.draftPost.richContent.metadata.updatedTimestamp"
          },
          { path: "draftPosts.draftPost.moderationDetails.submittedDate" },
          { path: "draftPosts.draftPost.moderationDetails.moderationDate" },
          {
            path: "draftPosts.draftPost.media.wixMedia.image.urlExpirationDate"
          },
          {
            path: "draftPosts.draftPost.media.wixMedia.videoV2.urlExpirationDate"
          },
          {
            path: "draftPosts.draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate"
          },
          {
            path: "draftPosts.draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
          },
          {
            path: "draftPosts.draftPost.media.wixMedia.videoV2.posters.urlExpirationDate"
          }
        ]
      },
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "draftPosts.fieldMask" }]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "draftPosts.draftPost.heroImage.focalPoint.x" },
          { path: "draftPosts.draftPost.heroImage.focalPoint.y" },
          { path: "draftPosts.draftPost.media.wixMedia.image.focalPoint.x" },
          { path: "draftPosts.draftPost.media.wixMedia.image.focalPoint.y" },
          {
            path: "draftPosts.draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
          },
          {
            path: "draftPosts.draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
          },
          {
            path: "draftPosts.draftPost.media.wixMedia.videoV2.posters.focalPoint.x"
          },
          {
            path: "draftPosts.draftPost.media.wixMedia.videoV2.posters.focalPoint.y"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.galleryData.options.item.ratio"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.imageData.image.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.mapData.mapSettings.lat"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.mapData.mapSettings.lng"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.pollData.poll.image.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.pollData.design.poll.background.image.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.appEmbedData.image.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.videoData.video.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.videoData.thumbnail.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.audioData.audio.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.audioData.coverImage.duration"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "PATCH",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.BulkUpdateDraftPosts",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        { protoPath: "/v3/draft-posts/update", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.editedDate" },
            { path: "results.item.scheduledPublishDate" },
            { path: "results.item.firstPublishedDate" },
            { path: "results.item.createdDate" },
            { path: "results.item.heroImage.urlExpirationDate" },
            { path: "results.item.richContent.metadata.createdTimestamp" },
            { path: "results.item.richContent.metadata.updatedTimestamp" },
            { path: "results.item.moderationDetails.submittedDate" },
            { path: "results.item.moderationDetails.moderationDate" },
            { path: "results.item.media.wixMedia.image.urlExpirationDate" },
            { path: "results.item.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.heroImage.focalPoint.x" },
            { path: "results.item.heroImage.focalPoint.y" },
            { path: "results.item.media.wixMedia.image.focalPoint.x" },
            { path: "results.item.media.wixMedia.image.focalPoint.y" },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            {
              path: "results.item.media.wixMedia.videoV2.posters.focalPoint.x"
            },
            {
              path: "results.item.media.wixMedia.videoV2.posters.focalPoint.y"
            },
            {
              path: "results.item.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "results.item.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "results.item.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "results.item.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "results.item.richContent.nodes.imageData.image.duration"
            },
            {
              path: "results.item.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "results.item.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "results.item.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "results.item.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "results.item.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "results.item.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "results.item.richContent.nodes.videoData.video.duration"
            },
            {
              path: "results.item.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "results.item.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "results.item.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateDraftPosts;
}
function listDeletedDraftPosts(payload) {
  function __listDeletedDraftPosts({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "GET",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.ListDeletedDraftPosts",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        { protoPath: "/v3/draft-posts/trash-bin", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "draftPosts.editedDate" },
            { path: "draftPosts.scheduledPublishDate" },
            { path: "draftPosts.firstPublishedDate" },
            { path: "draftPosts.createdDate" },
            { path: "draftPosts.heroImage.urlExpirationDate" },
            { path: "draftPosts.richContent.metadata.createdTimestamp" },
            { path: "draftPosts.richContent.metadata.updatedTimestamp" },
            { path: "draftPosts.moderationDetails.submittedDate" },
            { path: "draftPosts.moderationDetails.moderationDate" },
            { path: "draftPosts.media.wixMedia.image.urlExpirationDate" },
            { path: "draftPosts.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "draftPosts.heroImage.focalPoint.x" },
            { path: "draftPosts.heroImage.focalPoint.y" },
            { path: "draftPosts.media.wixMedia.image.focalPoint.x" },
            { path: "draftPosts.media.wixMedia.image.focalPoint.y" },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.posters.focalPoint.x"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.posters.focalPoint.y"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.options.item.ratio"
            },
            { path: "draftPosts.richContent.nodes.imageData.image.duration" },
            { path: "draftPosts.richContent.nodes.mapData.mapSettings.lat" },
            { path: "draftPosts.richContent.nodes.mapData.mapSettings.lng" },
            {
              path: "draftPosts.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "draftPosts.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "draftPosts.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "draftPosts.richContent.nodes.appEmbedData.image.duration"
            },
            { path: "draftPosts.richContent.nodes.videoData.video.duration" },
            {
              path: "draftPosts.richContent.nodes.videoData.thumbnail.duration"
            },
            { path: "draftPosts.richContent.nodes.audioData.audio.duration" },
            {
              path: "draftPosts.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listDeletedDraftPosts;
}
function getDraftPost(payload) {
  function __getDraftPost({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "GET",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.GetDraftPost",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        { protoPath: "/v3/draft-posts/{draftPostId}", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "draftPost.editedDate" },
            { path: "draftPost.scheduledPublishDate" },
            { path: "draftPost.firstPublishedDate" },
            { path: "draftPost.createdDate" },
            { path: "draftPost.heroImage.urlExpirationDate" },
            { path: "draftPost.richContent.metadata.createdTimestamp" },
            { path: "draftPost.richContent.metadata.updatedTimestamp" },
            { path: "draftPost.moderationDetails.submittedDate" },
            { path: "draftPost.moderationDetails.moderationDate" },
            { path: "draftPost.media.wixMedia.image.urlExpirationDate" },
            { path: "draftPost.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "draftPost.heroImage.focalPoint.x" },
            { path: "draftPost.heroImage.focalPoint.y" },
            { path: "draftPost.media.wixMedia.image.focalPoint.x" },
            { path: "draftPost.media.wixMedia.image.focalPoint.y" },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.x" },
            { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.y" },
            {
              path: "draftPost.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.options.item.ratio"
            },
            { path: "draftPost.richContent.nodes.imageData.image.duration" },
            { path: "draftPost.richContent.nodes.mapData.mapSettings.lat" },
            { path: "draftPost.richContent.nodes.mapData.mapSettings.lng" },
            {
              path: "draftPost.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.appEmbedData.image.duration"
            },
            { path: "draftPost.richContent.nodes.videoData.video.duration" },
            {
              path: "draftPost.richContent.nodes.videoData.thumbnail.duration"
            },
            { path: "draftPost.richContent.nodes.audioData.audio.duration" },
            {
              path: "draftPost.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getDraftPost;
}
function updateDraftPost(payload) {
  function __updateDraftPost({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "scheduledPublishDate" },
          { path: "draftPost.editedDate" },
          { path: "draftPost.scheduledPublishDate" },
          { path: "draftPost.firstPublishedDate" },
          { path: "draftPost.createdDate" },
          { path: "draftPost.heroImage.urlExpirationDate" },
          { path: "draftPost.richContent.metadata.createdTimestamp" },
          { path: "draftPost.richContent.metadata.updatedTimestamp" },
          { path: "draftPost.moderationDetails.submittedDate" },
          { path: "draftPost.moderationDetails.moderationDate" },
          { path: "draftPost.media.wixMedia.image.urlExpirationDate" },
          { path: "draftPost.media.wixMedia.videoV2.urlExpirationDate" },
          {
            path: "draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate"
          },
          {
            path: "draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
          },
          {
            path: "draftPost.media.wixMedia.videoV2.posters.urlExpirationDate"
          }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "draftPost.heroImage.focalPoint.x" },
          { path: "draftPost.heroImage.focalPoint.y" },
          { path: "draftPost.media.wixMedia.image.focalPoint.x" },
          { path: "draftPost.media.wixMedia.image.focalPoint.y" },
          {
            path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
          },
          {
            path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
          },
          { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.x" },
          { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.y" },
          {
            path: "draftPost.richContent.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "draftPost.richContent.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration"
          },
          {
            path: "draftPost.richContent.nodes.galleryData.options.item.ratio"
          },
          { path: "draftPost.richContent.nodes.imageData.image.duration" },
          { path: "draftPost.richContent.nodes.mapData.mapSettings.lat" },
          { path: "draftPost.richContent.nodes.mapData.mapSettings.lng" },
          { path: "draftPost.richContent.nodes.pollData.poll.image.duration" },
          {
            path: "draftPost.richContent.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "draftPost.richContent.nodes.pollData.design.poll.background.image.duration"
          },
          { path: "draftPost.richContent.nodes.appEmbedData.image.duration" },
          { path: "draftPost.richContent.nodes.videoData.video.duration" },
          { path: "draftPost.richContent.nodes.videoData.thumbnail.duration" },
          { path: "draftPost.richContent.nodes.audioData.audio.duration" },
          { path: "draftPost.richContent.nodes.audioData.coverImage.duration" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "PATCH",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.UpdateDraftPost",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        {
          protoPath: "/v3/draft-posts/{draftPost.id}",
          data: serializedData,
          host
        }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "draftPost.editedDate" },
            { path: "draftPost.scheduledPublishDate" },
            { path: "draftPost.firstPublishedDate" },
            { path: "draftPost.createdDate" },
            { path: "draftPost.heroImage.urlExpirationDate" },
            { path: "draftPost.richContent.metadata.createdTimestamp" },
            { path: "draftPost.richContent.metadata.updatedTimestamp" },
            { path: "draftPost.moderationDetails.submittedDate" },
            { path: "draftPost.moderationDetails.moderationDate" },
            { path: "draftPost.media.wixMedia.image.urlExpirationDate" },
            { path: "draftPost.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "draftPost.heroImage.focalPoint.x" },
            { path: "draftPost.heroImage.focalPoint.y" },
            { path: "draftPost.media.wixMedia.image.focalPoint.x" },
            { path: "draftPost.media.wixMedia.image.focalPoint.y" },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.x" },
            { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.y" },
            {
              path: "draftPost.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.options.item.ratio"
            },
            { path: "draftPost.richContent.nodes.imageData.image.duration" },
            { path: "draftPost.richContent.nodes.mapData.mapSettings.lat" },
            { path: "draftPost.richContent.nodes.mapData.mapSettings.lng" },
            {
              path: "draftPost.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.appEmbedData.image.duration"
            },
            { path: "draftPost.richContent.nodes.videoData.video.duration" },
            {
              path: "draftPost.richContent.nodes.videoData.thumbnail.duration"
            },
            { path: "draftPost.richContent.nodes.audioData.audio.duration" },
            {
              path: "draftPost.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateDraftPost;
}
function deleteDraftPost(payload) {
  function __deleteDraftPost({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "DELETE",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.DeleteDraftPost",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        { protoPath: "/v3/draft-posts/{draftPostId}", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteDraftPost;
}
function removeFromTrashBin(payload) {
  function __removeFromTrashBin({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "DELETE",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.RemoveFromTrashBin",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        {
          protoPath: "/v3/draft-posts/trash-bin/{draftPostId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __removeFromTrashBin;
}
function bulkDeleteDraftPosts(payload) {
  function __bulkDeleteDraftPosts({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "DELETE",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.BulkDeleteDraftPosts",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        { protoPath: "/v3/bulk/draft-posts", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.editedDate" },
            { path: "results.item.scheduledPublishDate" },
            { path: "results.item.firstPublishedDate" },
            { path: "results.item.createdDate" },
            { path: "results.item.heroImage.urlExpirationDate" },
            { path: "results.item.richContent.metadata.createdTimestamp" },
            { path: "results.item.richContent.metadata.updatedTimestamp" },
            { path: "results.item.moderationDetails.submittedDate" },
            { path: "results.item.moderationDetails.moderationDate" },
            { path: "results.item.media.wixMedia.image.urlExpirationDate" },
            { path: "results.item.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.heroImage.focalPoint.x" },
            { path: "results.item.heroImage.focalPoint.y" },
            { path: "results.item.media.wixMedia.image.focalPoint.x" },
            { path: "results.item.media.wixMedia.image.focalPoint.y" },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            {
              path: "results.item.media.wixMedia.videoV2.posters.focalPoint.x"
            },
            {
              path: "results.item.media.wixMedia.videoV2.posters.focalPoint.y"
            },
            {
              path: "results.item.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "results.item.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "results.item.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "results.item.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "results.item.richContent.nodes.imageData.image.duration"
            },
            {
              path: "results.item.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "results.item.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "results.item.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "results.item.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "results.item.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "results.item.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "results.item.richContent.nodes.videoData.video.duration"
            },
            {
              path: "results.item.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "results.item.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "results.item.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkDeleteDraftPosts;
}
function listDraftPosts(payload) {
  function __listDraftPosts({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "GET",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.ListDraftPosts",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        { protoPath: "/v3/draft-posts", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "draftPosts.editedDate" },
            { path: "draftPosts.scheduledPublishDate" },
            { path: "draftPosts.firstPublishedDate" },
            { path: "draftPosts.createdDate" },
            { path: "draftPosts.heroImage.urlExpirationDate" },
            { path: "draftPosts.richContent.metadata.createdTimestamp" },
            { path: "draftPosts.richContent.metadata.updatedTimestamp" },
            { path: "draftPosts.moderationDetails.submittedDate" },
            { path: "draftPosts.moderationDetails.moderationDate" },
            { path: "draftPosts.media.wixMedia.image.urlExpirationDate" },
            { path: "draftPosts.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "draftPosts.heroImage.focalPoint.x" },
            { path: "draftPosts.heroImage.focalPoint.y" },
            { path: "draftPosts.media.wixMedia.image.focalPoint.x" },
            { path: "draftPosts.media.wixMedia.image.focalPoint.y" },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.posters.focalPoint.x"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.posters.focalPoint.y"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.options.item.ratio"
            },
            { path: "draftPosts.richContent.nodes.imageData.image.duration" },
            { path: "draftPosts.richContent.nodes.mapData.mapSettings.lat" },
            { path: "draftPosts.richContent.nodes.mapData.mapSettings.lng" },
            {
              path: "draftPosts.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "draftPosts.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "draftPosts.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "draftPosts.richContent.nodes.appEmbedData.image.duration"
            },
            { path: "draftPosts.richContent.nodes.videoData.video.duration" },
            {
              path: "draftPosts.richContent.nodes.videoData.thumbnail.duration"
            },
            { path: "draftPosts.richContent.nodes.audioData.audio.duration" },
            {
              path: "draftPosts.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listDraftPosts;
}
function getDeletedDraftPost(payload) {
  function __getDeletedDraftPost({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "GET",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.GetDeletedDraftPost",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        {
          protoPath: "/v3/draft-posts/trash-bin/{draftPostId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "draftPost.editedDate" },
            { path: "draftPost.scheduledPublishDate" },
            { path: "draftPost.firstPublishedDate" },
            { path: "draftPost.createdDate" },
            { path: "draftPost.heroImage.urlExpirationDate" },
            { path: "draftPost.richContent.metadata.createdTimestamp" },
            { path: "draftPost.richContent.metadata.updatedTimestamp" },
            { path: "draftPost.moderationDetails.submittedDate" },
            { path: "draftPost.moderationDetails.moderationDate" },
            { path: "draftPost.media.wixMedia.image.urlExpirationDate" },
            { path: "draftPost.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "draftPost.heroImage.focalPoint.x" },
            { path: "draftPost.heroImage.focalPoint.y" },
            { path: "draftPost.media.wixMedia.image.focalPoint.x" },
            { path: "draftPost.media.wixMedia.image.focalPoint.y" },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.x" },
            { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.y" },
            {
              path: "draftPost.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.options.item.ratio"
            },
            { path: "draftPost.richContent.nodes.imageData.image.duration" },
            { path: "draftPost.richContent.nodes.mapData.mapSettings.lat" },
            { path: "draftPost.richContent.nodes.mapData.mapSettings.lng" },
            {
              path: "draftPost.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.appEmbedData.image.duration"
            },
            { path: "draftPost.richContent.nodes.videoData.video.duration" },
            {
              path: "draftPost.richContent.nodes.videoData.thumbnail.duration"
            },
            { path: "draftPost.richContent.nodes.audioData.audio.duration" },
            {
              path: "draftPost.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getDeletedDraftPost;
}
function restoreFromTrashBin(payload) {
  function __restoreFromTrashBin({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "POST",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.RestoreFromTrashBin",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        {
          protoPath: "/v3/draft-posts/trash-bin/{draftPostId}/restore",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "draftPost.editedDate" },
            { path: "draftPost.scheduledPublishDate" },
            { path: "draftPost.firstPublishedDate" },
            { path: "draftPost.createdDate" },
            { path: "draftPost.heroImage.urlExpirationDate" },
            { path: "draftPost.richContent.metadata.createdTimestamp" },
            { path: "draftPost.richContent.metadata.updatedTimestamp" },
            { path: "draftPost.moderationDetails.submittedDate" },
            { path: "draftPost.moderationDetails.moderationDate" },
            { path: "draftPost.media.wixMedia.image.urlExpirationDate" },
            { path: "draftPost.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "draftPost.heroImage.focalPoint.x" },
            { path: "draftPost.heroImage.focalPoint.y" },
            { path: "draftPost.media.wixMedia.image.focalPoint.x" },
            { path: "draftPost.media.wixMedia.image.focalPoint.y" },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.x" },
            { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.y" },
            {
              path: "draftPost.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.options.item.ratio"
            },
            { path: "draftPost.richContent.nodes.imageData.image.duration" },
            { path: "draftPost.richContent.nodes.mapData.mapSettings.lat" },
            { path: "draftPost.richContent.nodes.mapData.mapSettings.lng" },
            {
              path: "draftPost.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.appEmbedData.image.duration"
            },
            { path: "draftPost.richContent.nodes.videoData.video.duration" },
            {
              path: "draftPost.richContent.nodes.videoData.thumbnail.duration"
            },
            { path: "draftPost.richContent.nodes.audioData.audio.duration" },
            {
              path: "draftPost.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __restoreFromTrashBin;
}
function queryDraftPosts(payload) {
  function __queryDraftPosts({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "POST",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.QueryDraftPosts",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        { protoPath: "/v3/draft-posts/query", data: payload, host }
      ),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "draftPosts.editedDate" },
            { path: "draftPosts.scheduledPublishDate" },
            { path: "draftPosts.firstPublishedDate" },
            { path: "draftPosts.createdDate" },
            { path: "draftPosts.heroImage.urlExpirationDate" },
            { path: "draftPosts.richContent.metadata.createdTimestamp" },
            { path: "draftPosts.richContent.metadata.updatedTimestamp" },
            { path: "draftPosts.moderationDetails.submittedDate" },
            { path: "draftPosts.moderationDetails.moderationDate" },
            { path: "draftPosts.media.wixMedia.image.urlExpirationDate" },
            { path: "draftPosts.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "draftPosts.heroImage.focalPoint.x" },
            { path: "draftPosts.heroImage.focalPoint.y" },
            { path: "draftPosts.media.wixMedia.image.focalPoint.x" },
            { path: "draftPosts.media.wixMedia.image.focalPoint.y" },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.posters.focalPoint.x"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.posters.focalPoint.y"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.options.item.ratio"
            },
            { path: "draftPosts.richContent.nodes.imageData.image.duration" },
            { path: "draftPosts.richContent.nodes.mapData.mapSettings.lat" },
            { path: "draftPosts.richContent.nodes.mapData.mapSettings.lng" },
            {
              path: "draftPosts.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "draftPosts.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "draftPosts.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "draftPosts.richContent.nodes.appEmbedData.image.duration"
            },
            { path: "draftPosts.richContent.nodes.videoData.video.duration" },
            {
              path: "draftPosts.richContent.nodes.videoData.thumbnail.duration"
            },
            { path: "draftPosts.richContent.nodes.audioData.audio.duration" },
            {
              path: "draftPosts.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryDraftPosts;
}
function publishDraftPost(payload) {
  function __publishDraftPost({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "POST",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.PublishDraftPost",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        {
          protoPath: "/v3/draft-posts/{draftPostId}/publish",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __publishDraftPost;
}

// src/blog-v3-draft-draft-posts.types.ts
var Origin = /* @__PURE__ */ ((Origin2) => {
  Origin2["UNKNOWN"] = "UNKNOWN";
  Origin2["ADMIN"] = "ADMIN";
  Origin2["ADD_CATEGORIES"] = "ADD_CATEGORIES";
  Origin2["AUTO_SAVE"] = "AUTO_SAVE";
  Origin2["COPY_TEMPLATE"] = "COPY_TEMPLATE";
  Origin2["IMPORT"] = "IMPORT";
  Origin2["IMPORT_BULK"] = "IMPORT_BULK";
  Origin2["IMPORT_HTML"] = "IMPORT_HTML";
  Origin2["IMPORT_PATCH"] = "IMPORT_PATCH";
  Origin2["LANGUAGE_CHANGE"] = "LANGUAGE_CHANGE";
  Origin2["MANUAL_SAVE"] = "MANUAL_SAVE";
  Origin2["MIGRATION"] = "MIGRATION";
  Origin2["MODERATION"] = "MODERATION";
  Origin2["MOVE_TO_TRASH"] = "MOVE_TO_TRASH";
  Origin2["PRICING_PLANS_CHANGE"] = "PRICING_PLANS_CHANGE";
  Origin2["PROVISION"] = "PROVISION";
  Origin2["PUBLISH"] = "PUBLISH";
  Origin2["REASSIGN_OWNER"] = "REASSIGN_OWNER";
  Origin2["REBLOG"] = "REBLOG";
  Origin2["RESTORE"] = "RESTORE";
  Origin2["REVERT_TO_DRAFT"] = "REVERT_TO_DRAFT";
  Origin2["TRANSLATION"] = "TRANSLATION";
  Origin2["UNPUBLISH"] = "UNPUBLISH";
  Origin2["UNSCHEDULE"] = "UNSCHEDULE";
  Origin2["NEW_EDIT_SESSION"] = "NEW_EDIT_SESSION";
  Origin2["SCHEDULING_SERVICE_SCHEDULE"] = "SCHEDULING_SERVICE_SCHEDULE";
  Origin2["SCHEDULING_SERVICE_UNSCHEDULE"] = "SCHEDULING_SERVICE_UNSCHEDULE";
  Origin2["SCHEDULING_SERVICE_PUBLISH"] = "SCHEDULING_SERVICE_PUBLISH";
  Origin2["SCHEDULE"] = "SCHEDULE";
  Origin2["REMOVE_FROM_MODERATION"] = "REMOVE_FROM_MODERATION";
  Origin2["REJECT_FROM_MODERATION"] = "REJECT_FROM_MODERATION";
  Origin2["APPROVE_IN_MODERATION"] = "APPROVE_IN_MODERATION";
  Origin2["DELETE_TAG"] = "DELETE_TAG";
  Origin2["PIN"] = "PIN";
  Origin2["UNPIN"] = "UNPIN";
  Origin2["AI_AUTO_SAVE"] = "AI_AUTO_SAVE";
  return Origin2;
})(Origin || {});
var NodeType = /* @__PURE__ */ ((NodeType2) => {
  NodeType2["PARAGRAPH"] = "PARAGRAPH";
  NodeType2["TEXT"] = "TEXT";
  NodeType2["HEADING"] = "HEADING";
  NodeType2["BULLETED_LIST"] = "BULLETED_LIST";
  NodeType2["ORDERED_LIST"] = "ORDERED_LIST";
  NodeType2["LIST_ITEM"] = "LIST_ITEM";
  NodeType2["BLOCKQUOTE"] = "BLOCKQUOTE";
  NodeType2["CODE_BLOCK"] = "CODE_BLOCK";
  NodeType2["VIDEO"] = "VIDEO";
  NodeType2["DIVIDER"] = "DIVIDER";
  NodeType2["FILE"] = "FILE";
  NodeType2["GALLERY"] = "GALLERY";
  NodeType2["GIF"] = "GIF";
  NodeType2["HTML"] = "HTML";
  NodeType2["IMAGE"] = "IMAGE";
  NodeType2["LINK_PREVIEW"] = "LINK_PREVIEW";
  NodeType2["MAP"] = "MAP";
  NodeType2["POLL"] = "POLL";
  NodeType2["APP_EMBED"] = "APP_EMBED";
  NodeType2["BUTTON"] = "BUTTON";
  NodeType2["COLLAPSIBLE_LIST"] = "COLLAPSIBLE_LIST";
  NodeType2["TABLE"] = "TABLE";
  NodeType2["EMBED"] = "EMBED";
  NodeType2["COLLAPSIBLE_ITEM"] = "COLLAPSIBLE_ITEM";
  NodeType2["COLLAPSIBLE_ITEM_TITLE"] = "COLLAPSIBLE_ITEM_TITLE";
  NodeType2["COLLAPSIBLE_ITEM_BODY"] = "COLLAPSIBLE_ITEM_BODY";
  NodeType2["TABLE_CELL"] = "TABLE_CELL";
  NodeType2["TABLE_ROW"] = "TABLE_ROW";
  NodeType2["EXTERNAL"] = "EXTERNAL";
  NodeType2["AUDIO"] = "AUDIO";
  NodeType2["CAPTION"] = "CAPTION";
  NodeType2["LAYOUT"] = "LAYOUT";
  NodeType2["LAYOUT_CELL"] = "LAYOUT_CELL";
  return NodeType2;
})(NodeType || {});
var WidthType = /* @__PURE__ */ ((WidthType2) => {
  WidthType2["CONTENT"] = "CONTENT";
  WidthType2["SMALL"] = "SMALL";
  WidthType2["ORIGINAL"] = "ORIGINAL";
  WidthType2["FULL_WIDTH"] = "FULL_WIDTH";
  return WidthType2;
})(WidthType || {});
var PluginContainerDataAlignment = /* @__PURE__ */ ((PluginContainerDataAlignment2) => {
  PluginContainerDataAlignment2["CENTER"] = "CENTER";
  PluginContainerDataAlignment2["LEFT"] = "LEFT";
  PluginContainerDataAlignment2["RIGHT"] = "RIGHT";
  return PluginContainerDataAlignment2;
})(PluginContainerDataAlignment || {});
var ButtonDataType = /* @__PURE__ */ ((ButtonDataType2) => {
  ButtonDataType2["LINK"] = "LINK";
  ButtonDataType2["ACTION"] = "ACTION";
  return ButtonDataType2;
})(ButtonDataType || {});
var Target = /* @__PURE__ */ ((Target2) => {
  Target2["SELF"] = "SELF";
  Target2["BLANK"] = "BLANK";
  Target2["PARENT"] = "PARENT";
  Target2["TOP"] = "TOP";
  return Target2;
})(Target || {});
var TextAlignment = /* @__PURE__ */ ((TextAlignment2) => {
  TextAlignment2["AUTO"] = "AUTO";
  TextAlignment2["LEFT"] = "LEFT";
  TextAlignment2["RIGHT"] = "RIGHT";
  TextAlignment2["CENTER"] = "CENTER";
  TextAlignment2["JUSTIFY"] = "JUSTIFY";
  return TextAlignment2;
})(TextAlignment || {});
var LineStyle = /* @__PURE__ */ ((LineStyle2) => {
  LineStyle2["SINGLE"] = "SINGLE";
  LineStyle2["DOUBLE"] = "DOUBLE";
  LineStyle2["DASHED"] = "DASHED";
  LineStyle2["DOTTED"] = "DOTTED";
  return LineStyle2;
})(LineStyle || {});
var Width = /* @__PURE__ */ ((Width2) => {
  Width2["LARGE"] = "LARGE";
  Width2["MEDIUM"] = "MEDIUM";
  Width2["SMALL"] = "SMALL";
  return Width2;
})(Width || {});
var DividerDataAlignment = /* @__PURE__ */ ((DividerDataAlignment2) => {
  DividerDataAlignment2["CENTER"] = "CENTER";
  DividerDataAlignment2["LEFT"] = "LEFT";
  DividerDataAlignment2["RIGHT"] = "RIGHT";
  return DividerDataAlignment2;
})(DividerDataAlignment || {});
var ViewMode = /* @__PURE__ */ ((ViewMode2) => {
  ViewMode2["NONE"] = "NONE";
  ViewMode2["FULL"] = "FULL";
  ViewMode2["MINI"] = "MINI";
  return ViewMode2;
})(ViewMode || {});
var LayoutType = /* @__PURE__ */ ((LayoutType2) => {
  LayoutType2["COLLAGE"] = "COLLAGE";
  LayoutType2["MASONRY"] = "MASONRY";
  LayoutType2["GRID"] = "GRID";
  LayoutType2["THUMBNAIL"] = "THUMBNAIL";
  LayoutType2["SLIDER"] = "SLIDER";
  LayoutType2["SLIDESHOW"] = "SLIDESHOW";
  LayoutType2["PANORAMA"] = "PANORAMA";
  LayoutType2["COLUMN"] = "COLUMN";
  LayoutType2["MAGIC"] = "MAGIC";
  LayoutType2["FULLSIZE"] = "FULLSIZE";
  return LayoutType2;
})(LayoutType || {});
var Orientation = /* @__PURE__ */ ((Orientation2) => {
  Orientation2["ROWS"] = "ROWS";
  Orientation2["COLUMNS"] = "COLUMNS";
  return Orientation2;
})(Orientation || {});
var Crop = /* @__PURE__ */ ((Crop2) => {
  Crop2["FILL"] = "FILL";
  Crop2["FIT"] = "FIT";
  return Crop2;
})(Crop || {});
var ThumbnailsAlignment = /* @__PURE__ */ ((ThumbnailsAlignment2) => {
  ThumbnailsAlignment2["TOP"] = "TOP";
  ThumbnailsAlignment2["RIGHT"] = "RIGHT";
  ThumbnailsAlignment2["BOTTOM"] = "BOTTOM";
  ThumbnailsAlignment2["LEFT"] = "LEFT";
  ThumbnailsAlignment2["NONE"] = "NONE";
  return ThumbnailsAlignment2;
})(ThumbnailsAlignment || {});
var GIFType = /* @__PURE__ */ ((GIFType2) => {
  GIFType2["NORMAL"] = "NORMAL";
  GIFType2["STICKER"] = "STICKER";
  return GIFType2;
})(GIFType || {});
var Source = /* @__PURE__ */ ((Source2) => {
  Source2["HTML"] = "HTML";
  Source2["ADSENSE"] = "ADSENSE";
  Source2["AI"] = "AI";
  return Source2;
})(Source || {});
var StylesPosition = /* @__PURE__ */ ((StylesPosition2) => {
  StylesPosition2["START"] = "START";
  StylesPosition2["END"] = "END";
  StylesPosition2["TOP"] = "TOP";
  StylesPosition2["HIDDEN"] = "HIDDEN";
  return StylesPosition2;
})(StylesPosition || {});
var MapType = /* @__PURE__ */ ((MapType2) => {
  MapType2["ROADMAP"] = "ROADMAP";
  MapType2["SATELITE"] = "SATELITE";
  MapType2["HYBRID"] = "HYBRID";
  MapType2["TERRAIN"] = "TERRAIN";
  return MapType2;
})(MapType || {});
var ViewRole = /* @__PURE__ */ ((ViewRole2) => {
  ViewRole2["CREATOR"] = "CREATOR";
  ViewRole2["VOTERS"] = "VOTERS";
  ViewRole2["EVERYONE"] = "EVERYONE";
  return ViewRole2;
})(ViewRole || {});
var VoteRole = /* @__PURE__ */ ((VoteRole2) => {
  VoteRole2["SITE_MEMBERS"] = "SITE_MEMBERS";
  VoteRole2["ALL"] = "ALL";
  return VoteRole2;
})(VoteRole || {});
var PollLayoutType = /* @__PURE__ */ ((PollLayoutType2) => {
  PollLayoutType2["LIST"] = "LIST";
  PollLayoutType2["GRID"] = "GRID";
  return PollLayoutType2;
})(PollLayoutType || {});
var PollLayoutDirection = /* @__PURE__ */ ((PollLayoutDirection2) => {
  PollLayoutDirection2["LTR"] = "LTR";
  PollLayoutDirection2["RTL"] = "RTL";
  return PollLayoutDirection2;
})(PollLayoutDirection || {});
var BackgroundType = /* @__PURE__ */ ((BackgroundType2) => {
  BackgroundType2["COLOR"] = "COLOR";
  BackgroundType2["IMAGE"] = "IMAGE";
  BackgroundType2["GRADIENT"] = "GRADIENT";
  return BackgroundType2;
})(BackgroundType || {});
var DecorationType = /* @__PURE__ */ ((DecorationType2) => {
  DecorationType2["BOLD"] = "BOLD";
  DecorationType2["ITALIC"] = "ITALIC";
  DecorationType2["UNDERLINE"] = "UNDERLINE";
  DecorationType2["SPOILER"] = "SPOILER";
  DecorationType2["ANCHOR"] = "ANCHOR";
  DecorationType2["MENTION"] = "MENTION";
  DecorationType2["LINK"] = "LINK";
  DecorationType2["COLOR"] = "COLOR";
  DecorationType2["FONT_SIZE"] = "FONT_SIZE";
  DecorationType2["EXTERNAL"] = "EXTERNAL";
  DecorationType2["STRIKETHROUGH"] = "STRIKETHROUGH";
  DecorationType2["SUPERSCRIPT"] = "SUPERSCRIPT";
  DecorationType2["SUBSCRIPT"] = "SUBSCRIPT";
  return DecorationType2;
})(DecorationType || {});
var FontType = /* @__PURE__ */ ((FontType2) => {
  FontType2["PX"] = "PX";
  FontType2["EM"] = "EM";
  return FontType2;
})(FontType || {});
var Position = /* @__PURE__ */ ((Position2) => {
  Position2["START"] = "START";
  Position2["END"] = "END";
  Position2["TOP"] = "TOP";
  return Position2;
})(Position || {});
var AspectRatio = /* @__PURE__ */ ((AspectRatio2) => {
  AspectRatio2["SQUARE"] = "SQUARE";
  AspectRatio2["RECTANGLE"] = "RECTANGLE";
  return AspectRatio2;
})(AspectRatio || {});
var Resizing = /* @__PURE__ */ ((Resizing2) => {
  Resizing2["FILL"] = "FILL";
  Resizing2["FIT"] = "FIT";
  return Resizing2;
})(Resizing || {});
var Placement = /* @__PURE__ */ ((Placement2) => {
  Placement2["IMAGE"] = "IMAGE";
  Placement2["PRODUCT_INFO"] = "PRODUCT_INFO";
  return Placement2;
})(Placement || {});
var CardStylesType = /* @__PURE__ */ ((CardStylesType2) => {
  CardStylesType2["CONTAINED"] = "CONTAINED";
  CardStylesType2["FRAMELESS"] = "FRAMELESS";
  return CardStylesType2;
})(CardStylesType || {});
var Alignment = /* @__PURE__ */ ((Alignment2) => {
  Alignment2["START"] = "START";
  Alignment2["CENTER"] = "CENTER";
  Alignment2["END"] = "END";
  return Alignment2;
})(Alignment || {});
var Layout = /* @__PURE__ */ ((Layout2) => {
  Layout2["STACKED"] = "STACKED";
  Layout2["SIDE_BY_SIDE"] = "SIDE_BY_SIDE";
  return Layout2;
})(Layout || {});
var AppType = /* @__PURE__ */ ((AppType2) => {
  AppType2["PRODUCT"] = "PRODUCT";
  AppType2["EVENT"] = "EVENT";
  AppType2["BOOKING"] = "BOOKING";
  return AppType2;
})(AppType || {});
var InitialExpandedItems = /* @__PURE__ */ ((InitialExpandedItems2) => {
  InitialExpandedItems2["FIRST"] = "FIRST";
  InitialExpandedItems2["ALL"] = "ALL";
  InitialExpandedItems2["NONE"] = "NONE";
  return InitialExpandedItems2;
})(InitialExpandedItems || {});
var Direction = /* @__PURE__ */ ((Direction2) => {
  Direction2["LTR"] = "LTR";
  Direction2["RTL"] = "RTL";
  return Direction2;
})(Direction || {});
var VerticalAlignment = /* @__PURE__ */ ((VerticalAlignment2) => {
  VerticalAlignment2["TOP"] = "TOP";
  VerticalAlignment2["MIDDLE"] = "MIDDLE";
  VerticalAlignment2["BOTTOM"] = "BOTTOM";
  return VerticalAlignment2;
})(VerticalAlignment || {});
var NullValue = /* @__PURE__ */ ((NullValue2) => {
  NullValue2["NULL_VALUE"] = "NULL_VALUE";
  return NullValue2;
})(NullValue || {});
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["UNKNOWN"] = "UNKNOWN";
  Status2["PUBLISHED"] = "PUBLISHED";
  Status2["UNPUBLISHED"] = "UNPUBLISHED";
  Status2["SCHEDULED"] = "SCHEDULED";
  Status2["DELETED"] = "DELETED";
  Status2["IN_REVIEW"] = "IN_REVIEW";
  return Status2;
})(Status || {});
var ModerationStatusStatus = /* @__PURE__ */ ((ModerationStatusStatus2) => {
  ModerationStatusStatus2["UNKNOWN"] = "UNKNOWN";
  ModerationStatusStatus2["APPROVED"] = "APPROVED";
  ModerationStatusStatus2["REJECTED"] = "REJECTED";
  return ModerationStatusStatus2;
})(ModerationStatusStatus || {});
var TotalDraftPostsGroupingField = /* @__PURE__ */ ((TotalDraftPostsGroupingField2) => {
  TotalDraftPostsGroupingField2["STATUS"] = "STATUS";
  TotalDraftPostsGroupingField2["LANGUAGE"] = "LANGUAGE";
  return TotalDraftPostsGroupingField2;
})(TotalDraftPostsGroupingField || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNKNOWN"] = "UNKNOWN";
  Type2["MANUAL"] = "MANUAL";
  Type2["AUTO_SAVE"] = "AUTO_SAVE";
  Type2["PROVISIONING"] = "PROVISIONING";
  Type2["IMPORT"] = "IMPORT";
  Type2["AI_AUTO_SAVE"] = "AI_AUTO_SAVE";
  return Type2;
})(Type || {});
var Field = /* @__PURE__ */ ((Field2) => {
  Field2["UNKNOWN"] = "UNKNOWN";
  Field2["URL"] = "URL";
  Field2["CONTENT"] = "CONTENT";
  Field2["RICH_CONTENT"] = "RICH_CONTENT";
  Field2["GENERATED_EXCERPT"] = "GENERATED_EXCERPT";
  return Field2;
})(Field || {});
var Action = /* @__PURE__ */ ((Action2) => {
  Action2["UPDATE"] = "UPDATE";
  Action2["UPDATE_PUBLISH"] = "UPDATE_PUBLISH";
  Action2["UPDATE_SCHEDULE"] = "UPDATE_SCHEDULE";
  Action2["UPDATE_REVERT_TO_DRAFT"] = "UPDATE_REVERT_TO_DRAFT";
  Action2["UPDATE_CANCEL_SCHEDULE"] = "UPDATE_CANCEL_SCHEDULE";
  Action2["UPDATE_REJECT"] = "UPDATE_REJECT";
  Action2["UPDATE_PUBLICATION"] = "UPDATE_PUBLICATION";
  return Action2;
})(Action || {});
var GetDraftPostsSort = /* @__PURE__ */ ((GetDraftPostsSort2) => {
  GetDraftPostsSort2["EDITING_DATE_DESC"] = "EDITING_DATE_DESC";
  GetDraftPostsSort2["EDITING_DATE_ASC"] = "EDITING_DATE_ASC";
  return GetDraftPostsSort2;
})(GetDraftPostsSort || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});

// src/blog-v3-draft-draft-posts.meta.ts
function createDraftPost2() {
  const payload = {};
  const getRequestOptions = createDraftPost(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/draft-posts",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkCreateDraftPosts2() {
  const payload = {};
  const getRequestOptions = bulkCreateDraftPosts(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/bulk/draft-posts/create",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateDraftPosts2() {
  const payload = {};
  const getRequestOptions = bulkUpdateDraftPosts(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v3/draft-posts/update",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listDeletedDraftPosts2() {
  const payload = {};
  const getRequestOptions = listDeletedDraftPosts(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/draft-posts/trash-bin",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getDraftPost2() {
  const payload = { draftPostId: ":draftPostId" };
  const getRequestOptions = getDraftPost(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/draft-posts/{draftPostId}",
    pathParams: { draftPostId: "draftPostId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateDraftPost2() {
  const payload = { draftPost: { id: ":draftPostId" } };
  const getRequestOptions = updateDraftPost(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v3/draft-posts/{draftPost.id}",
    pathParams: { draftPostId: "draftPostId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteDraftPost2() {
  const payload = { draftPostId: ":draftPostId" };
  const getRequestOptions = deleteDraftPost(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v3/draft-posts/{draftPostId}",
    pathParams: { draftPostId: "draftPostId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function removeFromTrashBin2() {
  const payload = { draftPostId: ":draftPostId" };
  const getRequestOptions = removeFromTrashBin(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v3/draft-posts/trash-bin/{draftPostId}",
    pathParams: { draftPostId: "draftPostId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkDeleteDraftPosts2() {
  const payload = {};
  const getRequestOptions = bulkDeleteDraftPosts(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v3/bulk/draft-posts",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listDraftPosts2() {
  const payload = {};
  const getRequestOptions = listDraftPosts(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/draft-posts",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getDeletedDraftPost2() {
  const payload = { draftPostId: ":draftPostId" };
  const getRequestOptions = getDeletedDraftPost(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/draft-posts/trash-bin/{draftPostId}",
    pathParams: { draftPostId: "draftPostId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function restoreFromTrashBin2() {
  const payload = { draftPostId: ":draftPostId" };
  const getRequestOptions = restoreFromTrashBin(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/draft-posts/trash-bin/{draftPostId}/restore",
    pathParams: { draftPostId: "draftPostId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryDraftPosts2() {
  const payload = {};
  const getRequestOptions = queryDraftPosts(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/draft-posts/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function publishDraftPost2() {
  const payload = { draftPostId: ":draftPostId" };
  const getRequestOptions = publishDraftPost(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/draft-posts/{draftPostId}/publish",
    pathParams: { draftPostId: "draftPostId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ActionOriginal,
  AlignmentOriginal,
  AppTypeOriginal,
  AspectRatioOriginal,
  BackgroundTypeOriginal,
  ButtonDataTypeOriginal,
  CardStylesTypeOriginal,
  CropOriginal,
  DecorationTypeOriginal,
  DirectionOriginal,
  DividerDataAlignmentOriginal,
  FieldOriginal,
  FontTypeOriginal,
  GIFTypeOriginal,
  GetDraftPostsSortOriginal,
  InitialExpandedItemsOriginal,
  LayoutOriginal,
  LayoutTypeOriginal,
  LineStyleOriginal,
  MapTypeOriginal,
  ModerationStatusStatusOriginal,
  NodeTypeOriginal,
  NullValueOriginal,
  OrientationOriginal,
  OriginOriginal,
  PlacementOriginal,
  PluginContainerDataAlignmentOriginal,
  PollLayoutDirectionOriginal,
  PollLayoutTypeOriginal,
  PositionOriginal,
  ResizingOriginal,
  SortOrderOriginal,
  SourceOriginal,
  StatusOriginal,
  StylesPositionOriginal,
  TargetOriginal,
  TextAlignmentOriginal,
  ThumbnailsAlignmentOriginal,
  TotalDraftPostsGroupingFieldOriginal,
  TypeOriginal,
  VerticalAlignmentOriginal,
  ViewModeOriginal,
  ViewRoleOriginal,
  VoteRoleOriginal,
  WebhookIdentityTypeOriginal,
  WidthOriginal,
  WidthTypeOriginal,
  bulkCreateDraftPosts,
  bulkDeleteDraftPosts,
  bulkUpdateDraftPosts,
  createDraftPost,
  deleteDraftPost,
  getDeletedDraftPost,
  getDraftPost,
  listDeletedDraftPosts,
  listDraftPosts,
  publishDraftPost,
  queryDraftPosts,
  removeFromTrashBin,
  restoreFromTrashBin,
  updateDraftPost
});
