// @wix/design-system/dist/cjs/VerticalTabsItem/components/ActionItem.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.ActionItem = void 0;
var _react = _interopRequireDefault(require("react"));
var _VerticalTabsItemSt = require("../VerticalTabsItem.st.css.js");
var _Affix = require("./Affix");
var _VerticalTabsItem = require("../VerticalTabsItem.constants");
var _TextButton = _interopRequireDefault(require("../../TextButton"));
var _VerticalTabs = require("../../VerticalTabs/VerticalTabs.constants");
var _jsxFileName = "/home/builduser/work/57e038ea7326c1ec/packages/wix-design-system/dist/cjs/VerticalTabsItem/components/ActionItem.tsx";
var ActionItem = _ref => {
  var {
    children,
    dataHook,
    prefixIcon,
    suffixIcon,
    size,
    baseSkin
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _VerticalTabsItemSt.st)(_VerticalTabsItemSt.classes.root, (0, _VerticalTabsItemSt.cssStates)({
      size,
      baseSkin,
      suffixIcon: !!suffixIcon,
      prefixIcon: !!prefixIcon
    })),
    "data-hook": dataHook,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 26,
      columnNumber: 3
    }
  }, /*#__PURE__*/_react.default.createElement(_Affix.Affix, {
    dataHook: _VerticalTabsItem.DATA_HOOKS.ITEM_PREFIX_ICON,
    size: size,
    className: (0, _VerticalTabsItemSt.st)(_VerticalTabsItemSt.classes.prefixIcon, {
      action: true
    }),
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 38,
      columnNumber: 5
    }
  }, prefixIcon), /*#__PURE__*/_react.default.createElement(_TextButton.default, {
    weight: "thin",
    className: _VerticalTabsItemSt.classes.textButton,
    size: size === _VerticalTabs.SIZE.TINY ? _VerticalTabs.SIZE.SMALL : size,
    dataHook: _VerticalTabsItem.DATA_HOOKS.ITEM_TEXT,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 46,
      columnNumber: 5
    }
  }, children), /*#__PURE__*/_react.default.createElement(_Affix.Affix, {
    dataHook: _VerticalTabsItem.DATA_HOOKS.ITEM_SUFFIX_ICON,
    size: size,
    className: _VerticalTabsItemSt.classes.suffixIcon,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 55,
      columnNumber: 5
    }
  }, suffixIcon));
};
exports.ActionItem = ActionItem;
