// @wix/design-system/dist/cjs/AddItem/AddItem.st.css.js


require("../Foundation/stylable/colors.st.css.js");
require("../Foundation/stylable/easing.st.css.js");
require("../Tooltip/Tooltip.st.css.js");
require("../Foundation/stylable/opacity.st.css.js");
require("../Foundation/stylable/border.st.css.js");
require("../Text/Text.st.css.js");
const { classesRuntime, statesRuntime } = require("./../stylable-cjs-runtime.js");



var _namespace_ = "wds_1_226_0_AddItem";
var _style_ = classesRuntime.bind(null, _namespace_);

module.exports.cssStates = statesRuntime.bind(null, _namespace_);
module.exports.style = _style_;
module.exports.st = _style_;

module.exports.namespace = _namespace_;
module.exports.classes = {"root":"wds_1_226_0_AddItem__root","icon":"wds_1_226_0_AddItem__icon","illustration":"wds_1_226_0_AddItem__illustration","contentContainer":"wds_1_226_0_AddItem__contentContainer","content":"wds_1_226_0_AddItem__content","textWrapper":"wds_1_226_0_AddItem__textWrapper","textContent":"wds_1_226_0_AddItem__textContent","subtitle":"wds_1_226_0_AddItem__subtitle","tooltip":"wds_1_226_0_AddItem__tooltip"};
module.exports.keyframes = {}; 
module.exports.layers = {};
module.exports.containers = {};
module.exports.stVars = {}; 
module.exports.vars = {"wds-add-item-border-color-disabled":"--wds-add-item-border-color-disabled","wds-add-item-border-color-hover":"--wds-add-item-border-color-hover","wds-add-item-border-color":"--wds-add-item-border-color","wds-add-item-border-dashed-radius":"--wds-add-item-border-dashed-radius","wds-add-item-border-radius":"--wds-add-item-border-radius","wds-add-item-fill-pressed":"--wds-add-item-fill-pressed","wds-add-item-fill-disabled":"--wds-add-item-fill-disabled","wds-add-item-fill-hover":"--wds-add-item-fill-hover","wds-add-item-fill":"--wds-add-item-fill","wds-add-item-label-color":"--wds-add-item-label-color","wds-add-item-padding-horizontal-large":"--wds-add-item-padding-horizontal-large","wds-add-item-padding-vertical-large":"--wds-add-item-padding-vertical-large","wds-add-item-padding-vertical-medium":"--wds-add-item-padding-vertical-medium","wds-add-item-padding-vertical-small":"--wds-add-item-padding-vertical-small","wds-add-item-padding-vertical-tiny":"--wds-add-item-padding-vertical-tiny","wds-color-fill-standard-tertiary":"--wds-color-fill-standard-tertiary","wds-color-text-disabled":"--wds-color-text-disabled","wds-color-text-standard-secondary":"--wds-color-text-standard-secondary","wds-shadow-focus-standard":"--wds-shadow-focus-standard","wds-space-100":"--wds-space-100","wds-space-200":"--wds-space-200"};
