// @wix/design-system/dist/cjs/AddItem/AddItem.uni.driver.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.addItemUniDriverFactory = void 0;
var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _unidriver = require("../utils/test-utils/unidriver");
var _TextUni = require("../Text/Text.uni.driver");
var _TooltipUni = require("../Tooltip/Tooltip.uni.driver");
var _constants = require("./constants");
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2.default)(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; } // @ts-ignore
var addItemUniDriverFactory = (base, body) => {
  var tooltipDriver = (0, _TooltipUni.tooltipDriverFactory)((0, _unidriver.findByHook)(base, _constants.dataHooks.itemTooltip), body);
  var textDriver = (0, _TextUni.textUniDriverFactory)((0, _unidriver.findByHook)(base, _constants.dataHooks.itemText));
  var subtitleTextDriver = (0, _TextUni.textUniDriverFactory)((0, _unidriver.findByHook)(base, _constants.dataHooks.itemSubtitle));
  var addItemElement = (0, _unidriver.findByHook)(base, _constants.dataHooks.addItem);
  var reactBaseElement = (0, _unidriver.ReactBase)(addItemElement);
  return _objectSpread(_objectSpread({}, (0, _unidriver.baseUniDriverFactory)(addItemElement)), {}, {
    /**
     * Gets AddItem text
     * @return {Promise<string>}
     */
    getText: function () {
      var _getText = (0, _asyncToGenerator2.default)(function* () {
        return textDriver.getText();
      });
      function getText() {
        return _getText.apply(this, arguments);
      }
      return getText;
    }(),
    /**
     * Checks whether AddItem text exist
     * @returns {Promise<boolean>}
     */
    textExists: function () {
      var _textExists = (0, _asyncToGenerator2.default)(function* () {
        return textDriver.exists();
      });
      function textExists() {
        return _textExists.apply(this, arguments);
      }
      return textExists;
    }(),
    /**
     * Gets tooltip text
     * @return {Promise<string | null>}
     */
    getTooltipContent: function () {
      var _getTooltipContent = (0, _asyncToGenerator2.default)(function* () {
        return tooltipDriver.getTooltipText();
      });
      function getTooltipContent() {
        return _getTooltipContent.apply(this, arguments);
      }
      return getTooltipContent;
    }(),
    /**
     * Gets AddItem subtitle
     * @return {Promise<string>}
     */
    getSubtitle: function () {
      var _getSubtitle = (0, _asyncToGenerator2.default)(function* () {
        return subtitleTextDriver.getText();
      });
      function getSubtitle() {
        return _getSubtitle.apply(this, arguments);
      }
      return getSubtitle;
    }(),
    /**
     * Checks whether AddItem is focused
     * @returns {Promise<boolean>}
     */
    isFocused: function () {
      var _isFocused = (0, _asyncToGenerator2.default)(function* () {
        return (0, _unidriver.isElementFocused)(addItemElement);
      });
      function isFocused() {
        return _isFocused.apply(this, arguments);
      }
      return isFocused;
    }(),
    /**
     * Sets focus on the element
     * @returns {Promise<void>}
     */
    focus: function () {
      var _focus = (0, _asyncToGenerator2.default)(function* () {
        return reactBaseElement.focus();
      });
      function focus() {
        return _focus.apply(this, arguments);
      }
      return focus;
    }(),
    /**
     * Removes focus on the element
     * @returns {Promise<void>}
     */
    blur: function () {
      var _blur = (0, _asyncToGenerator2.default)(function* () {
        return reactBaseElement.blur();
      });
      function blur() {
        return _blur.apply(this, arguments);
      }
      return blur;
    }()
  });
};
exports.addItemUniDriverFactory = addItemUniDriverFactory;
