// @wix/auto_sdk_restaurants_operations/build/internal/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  ActivationType: () => ActivationType,
  AsapFutureHandlingType: () => AsapFutureHandlingType,
  AsapPacingWindowType: () => AsapPacingWindowType,
  AsapPreorderType: () => AsapPreorderType,
  DayOfWeek: () => DayOfWeek,
  DeleteStatus: () => DeleteStatus,
  EntitiesDayOfWeek: () => EntitiesDayOfWeek,
  FulfillmentTimeType: () => FulfillmentTimeType,
  FulfillmentTimesDisplayType: () => FulfillmentTimesDisplayType,
  FulfillmentTimesType: () => FulfillmentTimesType,
  FulfillmentType: () => FulfillmentType,
  MethodType: () => MethodType,
  Namespace: () => Namespace,
  OnlineOrderingStatusType: () => OnlineOrderingStatusType,
  OrderSchedulingType: () => OrderSchedulingType,
  PlacementType: () => PlacementType,
  PreorderPacingWindowType: () => PreorderPacingWindowType,
  PreparationTimePreparationTimeType: () => PreparationTimePreparationTimeType,
  PreparationTimeType: () => PreparationTimeType,
  ResolutionMethod: () => ResolutionMethod,
  SchedulingType: () => SchedulingType,
  SiteCreatedContext: () => SiteCreatedContext,
  SortOrder: () => SortOrder,
  State: () => State,
  SubdivisionType: () => SubdivisionType,
  TimeSlotForFulfillmentTimeSlotStatus: () => TimeSlotForFulfillmentTimeSlotStatus,
  TimeSlotStatus: () => TimeSlotStatus,
  TimeUnit: () => TimeUnit,
  ViolationType: () => ViolationType,
  WebhookIdentityType: () => WebhookIdentityType,
  bulkUpdateOperationTags: () => bulkUpdateOperationTags4,
  bulkUpdateOperationTagsByFilter: () => bulkUpdateOperationTagsByFilter4,
  calculateAvailableDatesInRange: () => calculateAvailableDatesInRange4,
  calculateAvailableTimeSlotsForDate: () => calculateAvailableTimeSlotsForDate4,
  calculateFirstAvailableTimeSlotPerFulfillmentType: () => calculateFirstAvailableTimeSlotPerFulfillmentType4,
  calculateFirstAvailableTimeSlotsForItem: () => calculateFirstAvailableTimeSlotsForItem4,
  calculateFirstAvailableTimeSlotsPerMenu: () => calculateFirstAvailableTimeSlotsPerMenu4,
  calculateFirstAvailableTimeSlotsPerOperation: () => calculateFirstAvailableTimeSlotsPerOperation4,
  createOperation: () => createOperation4,
  deleteOperation: () => deleteOperation4,
  getOperation: () => getOperation4,
  listAvailableDatesInRange: () => listAvailableDatesInRange4,
  listAvailableFulfillmentOptions: () => listAvailableFulfillmentOptions4,
  listAvailableTimeSlotsForDate: () => listAvailableTimeSlotsForDate4,
  listFirstAvailableTimeSlotForFulfillmentTypes: () => listFirstAvailableTimeSlotForFulfillmentTypes4,
  listFirstAvailableTimeSlotsForMenus: () => listFirstAvailableTimeSlotsForMenus4,
  listFirstAvailableTimeSlotsForOperations: () => listFirstAvailableTimeSlotsForOperations4,
  listOperations: () => listOperations4,
  onOperationCreated: () => onOperationCreated2,
  onOperationDeleted: () => onOperationDeleted2,
  onOperationUpdated: () => onOperationUpdated2,
  queryOperation: () => queryOperation4,
  updateOperation: () => updateOperation4,
  validateOperationAddress: () => validateOperationAddress4
});
module.exports = __toCommonJS(index_exports);

// src/restaurants-operations-v1-operation-operations.public.ts
var import_rename_all_nested_keys2 = require("@wix/sdk-runtime/rename-all-nested-keys");
var import_address3 = require("@wix/sdk-runtime/transformations/address");
var import_timestamp3 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths3 = require("@wix/sdk-runtime/transformations/transform-paths");
var import_sdk_types = require("@wix/sdk-types");

// src/restaurants-operations-v1-operation-operations.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/restaurants-operations-v1-operation-operations.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixRestaurantsOperationsV1OperationsServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-operations/v1",
        destPath: "/v1"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-operations/v1",
        destPath: "/v1"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-operations/v1",
        destPath: "/v1"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-operations/v1",
        destPath: "/v1"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants-operations/v1",
        destPath: "/v1"
      },
      {
        srcPath: "/restaurants/operations/v1/operations",
        destPath: "/v1/operations"
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-operations/v1",
        destPath: "/v1"
      }
    ],
    _base_domain_: [
      {
        srcPath: "/_api/restaurants-operations/v1",
        destPath: "/v1"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-operations/v1",
        destPath: "/v1"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-operations/v1",
        destPath: "/v1"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-operations/v1",
        destPath: "/v1"
      }
    ],
    "users._base_domain_": [
      {
        srcPath: "/_api/restaurants-operations/v1",
        destPath: "/v1"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-operations",
        destPath: ""
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-operations",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_operations";
function createOperation(payload) {
  function __createOperation({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "operation.createdDate" },
          { path: "operation.updatedDate" },
          { path: "operation.pausedUntilOptions.time" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          {
            path: "operation.businessLocationDetails.address.geocode.latitude"
          },
          {
            path: "operation.businessLocationDetails.address.geocode.longitude"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "POST",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.CreateOperation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "operation.createdDate" },
            { path: "operation.updatedDate" },
            { path: "operation.pausedUntilOptions.time" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "operation.businessLocationDetails.address.geocode.latitude"
            },
            {
              path: "operation.businessLocationDetails.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createOperation;
}
function getOperation(payload) {
  function __getOperation({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "GET",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.GetOperation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/{operationId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "operation.createdDate" },
            { path: "operation.updatedDate" },
            { path: "operation.pausedUntilOptions.time" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "operation.businessLocationDetails.address.geocode.latitude"
            },
            {
              path: "operation.businessLocationDetails.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getOperation;
}
function updateOperation(payload) {
  function __updateOperation({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "operation.createdDate" },
          { path: "operation.updatedDate" },
          { path: "operation.pausedUntilOptions.time" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          {
            path: "operation.businessLocationDetails.address.geocode.latitude"
          },
          {
            path: "operation.businessLocationDetails.address.geocode.longitude"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "PATCH",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.UpdateOperation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/{operation.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "operation.createdDate" },
            { path: "operation.updatedDate" },
            { path: "operation.pausedUntilOptions.time" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "operation.businessLocationDetails.address.geocode.latitude"
            },
            {
              path: "operation.businessLocationDetails.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateOperation;
}
function deleteOperation(payload) {
  function __deleteOperation({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "DELETE",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.DeleteOperation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/{operationId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteOperation;
}
function queryOperation(payload) {
  function __queryOperation({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "POST",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.QueryOperation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "operations.createdDate" },
            { path: "operations.updatedDate" },
            { path: "operations.pausedUntilOptions.time" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "operations.businessLocationDetails.address.geocode.latitude"
            },
            {
              path: "operations.businessLocationDetails.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryOperation;
}
function listOperations(payload) {
  function __listOperations({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "GET",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.ListOperations",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "operations.createdDate" },
            { path: "operations.updatedDate" },
            { path: "operations.pausedUntilOptions.time" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "operations.businessLocationDetails.address.geocode.latitude"
            },
            {
              path: "operations.businessLocationDetails.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listOperations;
}
function listAvailableFulfillmentOptions(payload) {
  function __listAvailableFulfillmentOptions({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "deliveryAddress.geocode.latitude" },
          { path: "deliveryAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "GET",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.ListAvailableFulfillmentOptions",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/{operationId}/available-fulfillment-options",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "fulfillmentOptions.pickupOptions.address.geocode.latitude"
            },
            {
              path: "fulfillmentOptions.pickupOptions.address.geocode.longitude"
            }
          ]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "fulfillmentOptions.availability.startTime" },
            { path: "fulfillmentOptions.availability.endTime" },
            { path: "fulfillmentOptions.availability.exceptions.startTime" },
            { path: "fulfillmentOptions.availability.exceptions.endTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listAvailableFulfillmentOptions;
}
function listFirstAvailableTimeSlotForFulfillmentTypes(payload) {
  function __listFirstAvailableTimeSlotForFulfillmentTypes({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "deliveryAddress.geocode.latitude" },
          { path: "deliveryAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "GET",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.ListFirstAvailableTimeSlotForFulfillmentTypes",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/{operationId}/first-available-time-slot-for-fulfillment-types",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "timeSlots.startTime" },
            { path: "timeSlots.endTime" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "timeSlots.fulfillmentAddress.address.geocode.latitude" },
            {
              path: "timeSlots.fulfillmentAddress.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listFirstAvailableTimeSlotForFulfillmentTypes;
}
function calculateFirstAvailableTimeSlotPerFulfillmentType(payload) {
  function __calculateFirstAvailableTimeSlotPerFulfillmentType({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "deliveryAddress.geocode.latitude" },
          { path: "deliveryAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "GET",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.CalculateFirstAvailableTimeSlotPerFulfillmentType",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/{operationId}/first-available-time-slot-per-fulfillment-type",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "timeslotsPerFulfillmentType.timeSlot.startTime" },
            { path: "timeslotsPerFulfillmentType.timeSlot.endTime" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "timeslotsPerFulfillmentType.fulfillmentInfo.address.geocode.latitude"
            },
            {
              path: "timeslotsPerFulfillmentType.fulfillmentInfo.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __calculateFirstAvailableTimeSlotPerFulfillmentType;
}
function listFirstAvailableTimeSlotsForOperations(payload) {
  function __listFirstAvailableTimeSlotsForOperations({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "deliveryAddress.geocode.latitude" },
          { path: "deliveryAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "POST",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.ListFirstAvailableTimeSlotsForOperations",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/first-available-time-slots-for-operations",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "timeSlots.timeSlots.startTime" },
            { path: "timeSlots.timeSlots.endTime" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "timeSlots.timeSlots.fulfillmentAddress.address.geocode.latitude"
            },
            {
              path: "timeSlots.timeSlots.fulfillmentAddress.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listFirstAvailableTimeSlotsForOperations;
}
function calculateFirstAvailableTimeSlotsPerOperation(payload) {
  function __calculateFirstAvailableTimeSlotsPerOperation({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "deliveryAddress.geocode.latitude" },
          { path: "deliveryAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "POST",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.CalculateFirstAvailableTimeSlotsPerOperation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/first-available-time-slots-per-operations",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            {
              path: "timeSlotsPerOperation.timeslotsPerFulfillmentType.timeSlot.startTime"
            },
            {
              path: "timeSlotsPerOperation.timeslotsPerFulfillmentType.timeSlot.endTime"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "timeSlotsPerOperation.timeslotsPerFulfillmentType.fulfillmentInfo.address.geocode.latitude"
            },
            {
              path: "timeSlotsPerOperation.timeslotsPerFulfillmentType.fulfillmentInfo.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __calculateFirstAvailableTimeSlotsPerOperation;
}
function listFirstAvailableTimeSlotsForMenus(payload) {
  function __listFirstAvailableTimeSlotsForMenus({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "deliveryAddress.geocode.latitude" },
          { path: "deliveryAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "GET",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.ListFirstAvailableTimeSlotsForMenus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/{operationId}/first-available-time-slots-for-menus",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            {
              path: "timeSlotsPerMenu.timeslotsPerFulfillmentType.startTime"
            },
            { path: "timeSlotsPerMenu.timeslotsPerFulfillmentType.endTime" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "timeSlotsPerMenu.timeslotsPerFulfillmentType.fulfillmentAddress.address.geocode.latitude"
            },
            {
              path: "timeSlotsPerMenu.timeslotsPerFulfillmentType.fulfillmentAddress.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listFirstAvailableTimeSlotsForMenus;
}
function calculateFirstAvailableTimeSlotsPerMenu(payload) {
  function __calculateFirstAvailableTimeSlotsPerMenu({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "deliveryAddress.geocode.latitude" },
          { path: "deliveryAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "GET",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.CalculateFirstAvailableTimeSlotsPerMenu",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/{operationId}/first-available-time-slots-per-menus",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            {
              path: "timeSlotsPerMenu.timeslotsPerFulfillmentType.timeSlot.startTime"
            },
            {
              path: "timeSlotsPerMenu.timeslotsPerFulfillmentType.timeSlot.endTime"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "timeSlotsPerMenu.timeslotsPerFulfillmentType.fulfillmentInfo.address.geocode.latitude"
            },
            {
              path: "timeSlotsPerMenu.timeslotsPerFulfillmentType.fulfillmentInfo.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __calculateFirstAvailableTimeSlotsPerMenu;
}
function calculateFirstAvailableTimeSlotsForItem(payload) {
  function __calculateFirstAvailableTimeSlotsForItem({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "deliveryAddress.geocode.latitude" },
          { path: "deliveryAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "GET",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.CalculateFirstAvailableTimeSlotsForItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/{operationId}/first-available-time-slots-per-item",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "timeslotsPerFulfillmentType.timeSlot.startTime" },
            { path: "timeslotsPerFulfillmentType.timeSlot.endTime" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "timeslotsPerFulfillmentType.fulfillmentInfo.address.geocode.latitude"
            },
            {
              path: "timeslotsPerFulfillmentType.fulfillmentInfo.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __calculateFirstAvailableTimeSlotsForItem;
}
function listAvailableTimeSlotsForDate(payload) {
  function __listAvailableTimeSlotsForDate({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "deliveryAddress.geocode.latitude" },
          { path: "deliveryAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "GET",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.ListAvailableTimeSlotsForDate",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/{operationId}/available-time-slots-for-date",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "timeSlots.startTime" },
            { path: "timeSlots.endTime" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "timeSlots.fulfillmentAddress.address.geocode.latitude" },
            {
              path: "timeSlots.fulfillmentAddress.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listAvailableTimeSlotsForDate;
}
function calculateAvailableTimeSlotsForDate(payload) {
  function __calculateAvailableTimeSlotsForDate({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "deliveryAddress.geocode.latitude" },
          { path: "deliveryAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "GET",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.CalculateAvailableTimeSlotsForDate",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/{operationId}/available-time-slots-per-date",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "timeslotsPerFulfillmentType.timeSlot.startTime" },
            { path: "timeslotsPerFulfillmentType.timeSlot.endTime" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "timeslotsPerFulfillmentType.fulfillmentInfo.address.geocode.latitude"
            },
            {
              path: "timeslotsPerFulfillmentType.fulfillmentInfo.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __calculateAvailableTimeSlotsForDate;
}
function listAvailableDatesInRange(payload) {
  function __listAvailableDatesInRange({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "deliveryAddress.geocode.latitude" },
          { path: "deliveryAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "GET",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.ListAvailableDatesInRange",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/{operationId}/available-dates-in-range",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData)
    };
    return metadata;
  }
  return __listAvailableDatesInRange;
}
function calculateAvailableDatesInRange(payload) {
  function __calculateAvailableDatesInRange({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "deliveryAddress.geocode.latitude" },
          { path: "deliveryAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "GET",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.CalculateAvailableDatesInRange",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/{operationId}/available-dates",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData)
    };
    return metadata;
  }
  return __calculateAvailableDatesInRange;
}
function validateOperationAddress(payload) {
  function __validateOperationAddress({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "GET",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.ValidateOperationAddress",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/operations/{operationId}/validate-address",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __validateOperationAddress;
}
function bulkUpdateOperationTags(payload) {
  function __bulkUpdateOperationTags({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "POST",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.BulkUpdateOperationTags",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/bulk/operations/update-tags",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateOperationTags;
}
function bulkUpdateOperationTagsByFilter(payload) {
  function __bulkUpdateOperationTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.operations.v1.operation",
      method: "POST",
      methodFqn: "wix.restaurants.operations.v1.OperationsService.BulkUpdateOperationTagsByFilter",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsOperationsV1OperationsServiceUrl({
        protoPath: "/v1/bulk/operations/update-tags-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateOperationTagsByFilter;
}

// src/restaurants-operations-v1-operation-operations.universal.ts
var import_address = require("@wix/sdk-runtime/transformations/address");
var import_address2 = require("@wix/sdk-runtime/transformations/address");
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var SchedulingType = /* @__PURE__ */ ((SchedulingType2) => {
  SchedulingType2["UNKNOWN_SCHEDULING"] = "UNKNOWN_SCHEDULING";
  SchedulingType2["ASAP"] = "ASAP";
  SchedulingType2["PREORDER"] = "PREORDER";
  return SchedulingType2;
})(SchedulingType || {});
var PreparationTimeType = /* @__PURE__ */ ((PreparationTimeType2) => {
  PreparationTimeType2["UNKNOWN_PREPARATION_TIME"] = "UNKNOWN_PREPARATION_TIME";
  PreparationTimeType2["MAX"] = "MAX";
  PreparationTimeType2["RANGE"] = "RANGE";
  return PreparationTimeType2;
})(PreparationTimeType || {});
var TimeUnit = /* @__PURE__ */ ((TimeUnit2) => {
  TimeUnit2["UNKNOWN_TIME_UNIT"] = "UNKNOWN_TIME_UNIT";
  TimeUnit2["MINUTES"] = "MINUTES";
  TimeUnit2["HOURS"] = "HOURS";
  TimeUnit2["DAYS"] = "DAYS";
  return TimeUnit2;
})(TimeUnit || {});
var AsapPreorderType = /* @__PURE__ */ ((AsapPreorderType2) => {
  AsapPreorderType2["UNKNOWN_ASAP_PREORDER"] = "UNKNOWN_ASAP_PREORDER";
  AsapPreorderType2["NO_PREORDER"] = "NO_PREORDER";
  AsapPreorderType2["BUSINESS_DAYS_PREORDER"] = "BUSINESS_DAYS_PREORDER";
  return AsapPreorderType2;
})(AsapPreorderType || {});
var AsapPacingWindowType = /* @__PURE__ */ ((AsapPacingWindowType2) => {
  AsapPacingWindowType2["UNKNOWN_ASAP_PACING_DURATION"] = "UNKNOWN_ASAP_PACING_DURATION";
  AsapPacingWindowType2["FIXED_15_MINUTE"] = "FIXED_15_MINUTE";
  return AsapPacingWindowType2;
})(AsapPacingWindowType || {});
var ActivationType = /* @__PURE__ */ ((ActivationType2) => {
  ActivationType2["UNKNOWN_SCHEDULE"] = "UNKNOWN_SCHEDULE";
  ActivationType2["ALWAYS_ACTIVE"] = "ALWAYS_ACTIVE";
  ActivationType2["SCHEDULE"] = "SCHEDULE";
  return ActivationType2;
})(ActivationType || {});
var EntitiesDayOfWeek = /* @__PURE__ */ ((EntitiesDayOfWeek2) => {
  EntitiesDayOfWeek2["MON"] = "MON";
  EntitiesDayOfWeek2["TUE"] = "TUE";
  EntitiesDayOfWeek2["WED"] = "WED";
  EntitiesDayOfWeek2["THU"] = "THU";
  EntitiesDayOfWeek2["FRI"] = "FRI";
  EntitiesDayOfWeek2["SAT"] = "SAT";
  EntitiesDayOfWeek2["SUN"] = "SUN";
  return EntitiesDayOfWeek2;
})(EntitiesDayOfWeek || {});
var MethodType = /* @__PURE__ */ ((MethodType2) => {
  MethodType2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  MethodType2["TIME_BOUNDED"] = "TIME_BOUNDED";
  MethodType2["WEEKLY_SCHEDULE"] = "WEEKLY_SCHEDULE";
  return MethodType2;
})(MethodType || {});
var FulfillmentTimesType = /* @__PURE__ */ ((FulfillmentTimesType2) => {
  FulfillmentTimesType2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  FulfillmentTimesType2["TIME_WINDOWS"] = "TIME_WINDOWS";
  return FulfillmentTimesType2;
})(FulfillmentTimesType || {});
var PreorderPacingWindowType = /* @__PURE__ */ ((PreorderPacingWindowType2) => {
  PreorderPacingWindowType2["UNKNOWN_PREORDER_PACING_WINDOW"] = "UNKNOWN_PREORDER_PACING_WINDOW";
  PreorderPacingWindowType2["DAILY"] = "DAILY";
  return PreorderPacingWindowType2;
})(PreorderPacingWindowType || {});
var OnlineOrderingStatusType = /* @__PURE__ */ ((OnlineOrderingStatusType2) => {
  OnlineOrderingStatusType2["UNDEFINED_ONLINE_ORDERING_STATUS"] = "UNDEFINED_ONLINE_ORDERING_STATUS";
  OnlineOrderingStatusType2["ENABLED"] = "ENABLED";
  OnlineOrderingStatusType2["DISABLED"] = "DISABLED";
  OnlineOrderingStatusType2["PAUSED_UNTIL"] = "PAUSED_UNTIL";
  return OnlineOrderingStatusType2;
})(OnlineOrderingStatusType || {});
var FulfillmentType = /* @__PURE__ */ ((FulfillmentType2) => {
  FulfillmentType2["UNDEFINED_FULFILLMENT_TYPE"] = "UNDEFINED_FULFILLMENT_TYPE";
  FulfillmentType2["PICKUP"] = "PICKUP";
  FulfillmentType2["DELIVERY"] = "DELIVERY";
  return FulfillmentType2;
})(FulfillmentType || {});
var PreparationTimePreparationTimeType = /* @__PURE__ */ ((PreparationTimePreparationTimeType2) => {
  PreparationTimePreparationTimeType2["UNKNOWN_PREPARATION_TIME"] = "UNKNOWN_PREPARATION_TIME";
  PreparationTimePreparationTimeType2["MAX_TIME"] = "MAX_TIME";
  PreparationTimePreparationTimeType2["TIME_RANGE"] = "TIME_RANGE";
  return PreparationTimePreparationTimeType2;
})(PreparationTimePreparationTimeType || {});
var AsapFutureHandlingType = /* @__PURE__ */ ((AsapFutureHandlingType2) => {
  AsapFutureHandlingType2["UNKNOWN_ASAP_FUTURE_HANDLING"] = "UNKNOWN_ASAP_FUTURE_HANDLING";
  AsapFutureHandlingType2["NO_FUTURE_HANDLING"] = "NO_FUTURE_HANDLING";
  AsapFutureHandlingType2["BUSINESS_DAYS_AHEAD_HANDLING"] = "BUSINESS_DAYS_AHEAD_HANDLING";
  return AsapFutureHandlingType2;
})(AsapFutureHandlingType || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var SubdivisionType = /* @__PURE__ */ ((SubdivisionType2) => {
  SubdivisionType2["UNKNOWN_SUBDIVISION_TYPE"] = "UNKNOWN_SUBDIVISION_TYPE";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_1"] = "ADMINISTRATIVE_AREA_LEVEL_1";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_2"] = "ADMINISTRATIVE_AREA_LEVEL_2";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_3"] = "ADMINISTRATIVE_AREA_LEVEL_3";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_4"] = "ADMINISTRATIVE_AREA_LEVEL_4";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_5"] = "ADMINISTRATIVE_AREA_LEVEL_5";
  SubdivisionType2["COUNTRY"] = "COUNTRY";
  return SubdivisionType2;
})(SubdivisionType || {});
var FulfillmentTimeType = /* @__PURE__ */ ((FulfillmentTimeType2) => {
  FulfillmentTimeType2["UNDEFINED_FULFILLMENT_TIME"] = "UNDEFINED_FULFILLMENT_TIME";
  FulfillmentTimeType2["MAX_TIME"] = "MAX_TIME";
  FulfillmentTimeType2["DURATION_RANGE"] = "DURATION_RANGE";
  return FulfillmentTimeType2;
})(FulfillmentTimeType || {});
var FulfillmentTimesDisplayType = /* @__PURE__ */ ((FulfillmentTimesDisplayType2) => {
  FulfillmentTimesDisplayType2["UNDEFINED_FULFILLMENT_TIMES_DISPLAY"] = "UNDEFINED_FULFILLMENT_TIMES_DISPLAY";
  FulfillmentTimesDisplayType2["TIME_WINDOWS"] = "TIME_WINDOWS";
  return FulfillmentTimesDisplayType2;
})(FulfillmentTimesDisplayType || {});
var TimeSlotStatus = /* @__PURE__ */ ((TimeSlotStatus2) => {
  TimeSlotStatus2["AVAILABLE"] = "AVAILABLE";
  TimeSlotStatus2["BLOCKED"] = "BLOCKED";
  return TimeSlotStatus2;
})(TimeSlotStatus || {});
var OrderSchedulingType = /* @__PURE__ */ ((OrderSchedulingType2) => {
  OrderSchedulingType2["ASAP"] = "ASAP";
  OrderSchedulingType2["PREORDER"] = "PREORDER";
  return OrderSchedulingType2;
})(OrderSchedulingType || {});
var TimeSlotForFulfillmentTimeSlotStatus = /* @__PURE__ */ ((TimeSlotForFulfillmentTimeSlotStatus2) => {
  TimeSlotForFulfillmentTimeSlotStatus2["AVAILABLE"] = "AVAILABLE";
  TimeSlotForFulfillmentTimeSlotStatus2["BLOCKED"] = "BLOCKED";
  return TimeSlotForFulfillmentTimeSlotStatus2;
})(TimeSlotForFulfillmentTimeSlotStatus || {});
var ViolationType = /* @__PURE__ */ ((ViolationType2) => {
  ViolationType2["UNKNOWN"] = "UNKNOWN";
  ViolationType2["NO_ADDRESS"] = "NO_ADDRESS";
  ViolationType2["MISSING_FORMATTED_ADDRESS"] = "MISSING_FORMATTED_ADDRESS";
  ViolationType2["INVALID_GEOCODE"] = "INVALID_GEOCODE";
  ViolationType2["MISSING_COUNTRY"] = "MISSING_COUNTRY";
  ViolationType2["MISSING_SUBDIVISION"] = "MISSING_SUBDIVISION";
  return ViolationType2;
})(ViolationType || {});
var PlacementType = /* @__PURE__ */ ((PlacementType2) => {
  PlacementType2["BEFORE"] = "BEFORE";
  PlacementType2["AFTER"] = "AFTER";
  PlacementType2["REPLACE"] = "REPLACE";
  return PlacementType2;
})(PlacementType || {});
var DayOfWeek = /* @__PURE__ */ ((DayOfWeek2) => {
  DayOfWeek2["MONDAY"] = "MONDAY";
  DayOfWeek2["TUESDAY"] = "TUESDAY";
  DayOfWeek2["WEDNESDAY"] = "WEDNESDAY";
  DayOfWeek2["THURSDAY"] = "THURSDAY";
  DayOfWeek2["FRIDAY"] = "FRIDAY";
  DayOfWeek2["SATURDAY"] = "SATURDAY";
  DayOfWeek2["SUNDAY"] = "SUNDAY";
  return DayOfWeek2;
})(DayOfWeek || {});
var ResolutionMethod = /* @__PURE__ */ ((ResolutionMethod2) => {
  ResolutionMethod2["QUERY_PARAM"] = "QUERY_PARAM";
  ResolutionMethod2["SUBDOMAIN"] = "SUBDOMAIN";
  ResolutionMethod2["SUBDIRECTORY"] = "SUBDIRECTORY";
  return ResolutionMethod2;
})(ResolutionMethod || {});
var State = /* @__PURE__ */ ((State2) => {
  State2["UNKNOWN"] = "UNKNOWN";
  State2["ENABLED"] = "ENABLED";
  State2["DISABLED"] = "DISABLED";
  State2["PENDING"] = "PENDING";
  State2["DEMO"] = "DEMO";
  return State2;
})(State || {});
var SiteCreatedContext = /* @__PURE__ */ ((SiteCreatedContext2) => {
  SiteCreatedContext2["OTHER"] = "OTHER";
  SiteCreatedContext2["FROM_TEMPLATE"] = "FROM_TEMPLATE";
  SiteCreatedContext2["DUPLICATE_BY_SITE_TRANSFER"] = "DUPLICATE_BY_SITE_TRANSFER";
  SiteCreatedContext2["DUPLICATE"] = "DUPLICATE";
  SiteCreatedContext2["OLD_SITE_TRANSFER"] = "OLD_SITE_TRANSFER";
  SiteCreatedContext2["FLASH"] = "FLASH";
  return SiteCreatedContext2;
})(SiteCreatedContext || {});
var Namespace = /* @__PURE__ */ ((Namespace2) => {
  Namespace2["UNKNOWN_NAMESPACE"] = "UNKNOWN_NAMESPACE";
  Namespace2["WIX"] = "WIX";
  Namespace2["SHOUT_OUT"] = "SHOUT_OUT";
  Namespace2["ALBUMS"] = "ALBUMS";
  Namespace2["WIX_STORES_TEST_DRIVE"] = "WIX_STORES_TEST_DRIVE";
  Namespace2["HOTELS"] = "HOTELS";
  Namespace2["CLUBS"] = "CLUBS";
  Namespace2["ONBOARDING_DRAFT"] = "ONBOARDING_DRAFT";
  Namespace2["DEV_SITE"] = "DEV_SITE";
  Namespace2["LOGOS"] = "LOGOS";
  Namespace2["VIDEO_MAKER"] = "VIDEO_MAKER";
  Namespace2["PARTNER_DASHBOARD"] = "PARTNER_DASHBOARD";
  Namespace2["DEV_CENTER_COMPANY"] = "DEV_CENTER_COMPANY";
  Namespace2["HTML_DRAFT"] = "HTML_DRAFT";
  Namespace2["SITELESS_BUSINESS"] = "SITELESS_BUSINESS";
  Namespace2["CREATOR_ECONOMY"] = "CREATOR_ECONOMY";
  Namespace2["DASHBOARD_FIRST"] = "DASHBOARD_FIRST";
  Namespace2["ANYWHERE"] = "ANYWHERE";
  Namespace2["HEADLESS"] = "HEADLESS";
  Namespace2["ACCOUNT_MASTER_CMS"] = "ACCOUNT_MASTER_CMS";
  Namespace2["RISE"] = "RISE";
  Namespace2["BRANDED_FIRST"] = "BRANDED_FIRST";
  Namespace2["NOWNIA"] = "NOWNIA";
  Namespace2["UGC_TEMPLATE"] = "UGC_TEMPLATE";
  Namespace2["CODUX"] = "CODUX";
  Namespace2["MEDIA_DESIGN_CREATOR"] = "MEDIA_DESIGN_CREATOR";
  Namespace2["SHARED_BLOG_ENTERPRISE"] = "SHARED_BLOG_ENTERPRISE";
  Namespace2["STANDALONE_FORMS"] = "STANDALONE_FORMS";
  Namespace2["STANDALONE_EVENTS"] = "STANDALONE_EVENTS";
  Namespace2["MIMIR"] = "MIMIR";
  Namespace2["TWINS"] = "TWINS";
  Namespace2["NANO"] = "NANO";
  return Namespace2;
})(Namespace || {});
var DeleteStatus = /* @__PURE__ */ ((DeleteStatus2) => {
  DeleteStatus2["UNKNOWN"] = "UNKNOWN";
  DeleteStatus2["TRASH"] = "TRASH";
  DeleteStatus2["DELETED"] = "DELETED";
  DeleteStatus2["PENDING_PURGE"] = "PENDING_PURGE";
  DeleteStatus2["PURGED_EXTERNALLY"] = "PURGED_EXTERNALLY";
  return DeleteStatus2;
})(DeleteStatus || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createOperation2(operation) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ operation }),
    [
      {
        transformFn: import_address.transformSDKAddressToRESTAddress,
        paths: [{ path: "operation.businessLocationDetails.address" }]
      }
    ]
  );
  const reqOpts = createOperation(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [{ path: "operation.businessLocationDetails.address" }]
        }
      ])
    )?.operation;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { operation: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["operation"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getOperation2(operationId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    operationId
  });
  const reqOpts = getOperation(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [{ path: "operation.businessLocationDetails.address" }]
        }
      ])
    )?.operation;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { operationId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["operationId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateOperation2(_id, operation) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      operation: { ...operation, id: _id }
    }),
    [
      {
        transformFn: import_address.transformSDKAddressToRESTAddress,
        paths: [{ path: "operation.businessLocationDetails.address" }]
      }
    ]
  );
  const reqOpts = updateOperation(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [{ path: "operation.businessLocationDetails.address" }]
        }
      ])
    )?.operation;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { operation: "$[1]" },
        explicitPathsToArguments: { "operation.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "operation"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteOperation2(operationId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    operationId
  });
  const reqOpts = deleteOperation(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { operationId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["operationId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryOperation2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryOperation(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [
          {
            transformFn: import_address2.transformRESTAddressToSDKAddress,
            paths: [{ path: "operations.businessLocationDetails.address" }]
          }
        ])
      );
      return {
        items: transformedData?.operations,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function listOperations2() {
  const { httpClient, sideEffects } = arguments[0];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({});
  const reqOpts = listOperations(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [{ path: "operations.businessLocationDetails.address" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {},
        singleArgumentUnchanged: false
      },
      []
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listAvailableFulfillmentOptions2(operationId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      operationId,
      deliveryAddress: options?.deliveryAddress
    }),
    [
      {
        transformFn: import_address.transformSDKAddressToRESTAddress,
        paths: [{ path: "deliveryAddress" }]
      }
    ]
  );
  const reqOpts = listAvailableFulfillmentOptions(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [{ path: "fulfillmentOptions.pickupOptions.address" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          operationId: "$[0]",
          deliveryAddress: "$[1].deliveryAddress"
        },
        singleArgumentUnchanged: false
      },
      ["operationId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listFirstAvailableTimeSlotForFulfillmentTypes2(operationId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      operationId,
      deliveryAddress: options?.deliveryAddress
    }),
    [
      {
        transformFn: import_address.transformSDKAddressToRESTAddress,
        paths: [{ path: "deliveryAddress" }]
      }
    ]
  );
  const reqOpts = listFirstAvailableTimeSlotForFulfillmentTypes(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [{ path: "timeSlots.fulfillmentAddress.address" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          operationId: "$[0]",
          deliveryAddress: "$[1].deliveryAddress"
        },
        singleArgumentUnchanged: false
      },
      ["operationId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function calculateFirstAvailableTimeSlotPerFulfillmentType2(operationId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      operationId,
      deliveryAddress: options?.deliveryAddress
    }),
    [
      {
        transformFn: import_address.transformSDKAddressToRESTAddress,
        paths: [{ path: "deliveryAddress" }]
      }
    ]
  );
  const reqOpts = calculateFirstAvailableTimeSlotPerFulfillmentType(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [
            { path: "timeslotsPerFulfillmentType.fulfillmentInfo.address" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          operationId: "$[0]",
          deliveryAddress: "$[1].deliveryAddress"
        },
        singleArgumentUnchanged: false
      },
      ["operationId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listFirstAvailableTimeSlotsForOperations2(operationIds, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      operationIds,
      deliveryAddress: options?.deliveryAddress
    }),
    [
      {
        transformFn: import_address.transformSDKAddressToRESTAddress,
        paths: [{ path: "deliveryAddress" }]
      }
    ]
  );
  const reqOpts = listFirstAvailableTimeSlotsForOperations(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [{ path: "timeSlots.timeSlots.fulfillmentAddress.address" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          operationIds: "$[0]",
          deliveryAddress: "$[1].deliveryAddress"
        },
        singleArgumentUnchanged: false
      },
      ["operationIds", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function calculateFirstAvailableTimeSlotsPerOperation2(operationIds, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      operationIds,
      deliveryAddress: options?.deliveryAddress
    }),
    [
      {
        transformFn: import_address.transformSDKAddressToRESTAddress,
        paths: [{ path: "deliveryAddress" }]
      }
    ]
  );
  const reqOpts = calculateFirstAvailableTimeSlotsPerOperation(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [
            {
              path: "timeSlotsPerOperation.timeslotsPerFulfillmentType.fulfillmentInfo.address"
            }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          operationIds: "$[0]",
          deliveryAddress: "$[1].deliveryAddress"
        },
        singleArgumentUnchanged: false
      },
      ["operationIds", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listFirstAvailableTimeSlotsForMenus2(operationId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      operationId,
      deliveryAddress: options?.deliveryAddress,
      cursorPaging: options?.cursorPaging
    }),
    [
      {
        transformFn: import_address.transformSDKAddressToRESTAddress,
        paths: [{ path: "deliveryAddress" }]
      }
    ]
  );
  const reqOpts = listFirstAvailableTimeSlotsForMenus(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [
            {
              path: "timeSlotsPerMenu.timeslotsPerFulfillmentType.fulfillmentAddress.address"
            }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          operationId: "$[0]",
          deliveryAddress: "$[1].deliveryAddress",
          cursorPaging: "$[1].cursorPaging"
        },
        singleArgumentUnchanged: false
      },
      ["operationId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function calculateFirstAvailableTimeSlotsPerMenu2(operationId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      operationId,
      deliveryAddress: options?.deliveryAddress,
      cursorPaging: options?.cursorPaging,
      onlyOnlineOrderingEnabled: options?.onlyOnlineOrderingEnabled
    }),
    [
      {
        transformFn: import_address.transformSDKAddressToRESTAddress,
        paths: [{ path: "deliveryAddress" }]
      }
    ]
  );
  const reqOpts = calculateFirstAvailableTimeSlotsPerMenu(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [
            {
              path: "timeSlotsPerMenu.timeslotsPerFulfillmentType.fulfillmentInfo.address"
            }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          operationId: "$[0]",
          deliveryAddress: "$[1].deliveryAddress",
          cursorPaging: "$[1].cursorPaging",
          onlyOnlineOrderingEnabled: "$[1].onlyOnlineOrderingEnabled"
        },
        singleArgumentUnchanged: false
      },
      ["operationId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function calculateFirstAvailableTimeSlotsForItem2(operationId, menuId, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      operationId,
      menuId,
      sectionId: options?.sectionId,
      itemId: options?.itemId,
      deliveryAddress: options?.deliveryAddress
    }),
    [
      {
        transformFn: import_address.transformSDKAddressToRESTAddress,
        paths: [{ path: "deliveryAddress" }]
      }
    ]
  );
  const reqOpts = calculateFirstAvailableTimeSlotsForItem(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [
            { path: "timeslotsPerFulfillmentType.fulfillmentInfo.address" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          operationId: "$[0]",
          menuId: "$[1]",
          sectionId: "$[2].sectionId",
          itemId: "$[2].itemId",
          deliveryAddress: "$[2].deliveryAddress"
        },
        singleArgumentUnchanged: false
      },
      ["operationId", "menuId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listAvailableTimeSlotsForDate2(operationId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      operationId,
      deliveryAddress: options?.deliveryAddress,
      date: options?.date
    }),
    [
      {
        transformFn: import_address.transformSDKAddressToRESTAddress,
        paths: [{ path: "deliveryAddress" }]
      }
    ]
  );
  const reqOpts = listAvailableTimeSlotsForDate(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [{ path: "timeSlots.fulfillmentAddress.address" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          operationId: "$[0]",
          deliveryAddress: "$[1].deliveryAddress",
          date: "$[1].date"
        },
        singleArgumentUnchanged: false
      },
      ["operationId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function calculateAvailableTimeSlotsForDate2(operationId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      operationId,
      deliveryAddress: options?.deliveryAddress,
      date: options?.date
    }),
    [
      {
        transformFn: import_address.transformSDKAddressToRESTAddress,
        paths: [{ path: "deliveryAddress" }]
      }
    ]
  );
  const reqOpts = calculateAvailableTimeSlotsForDate(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [
            { path: "timeslotsPerFulfillmentType.fulfillmentInfo.address" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          operationId: "$[0]",
          deliveryAddress: "$[1].deliveryAddress",
          date: "$[1].date"
        },
        singleArgumentUnchanged: false
      },
      ["operationId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listAvailableDatesInRange2(operationId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      operationId,
      deliveryAddress: options?.deliveryAddress,
      from: options?.from,
      until: options?.until
    }),
    [
      {
        transformFn: import_address.transformSDKAddressToRESTAddress,
        paths: [{ path: "deliveryAddress" }]
      }
    ]
  );
  const reqOpts = listAvailableDatesInRange(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          operationId: "$[0]",
          deliveryAddress: "$[1].deliveryAddress",
          from: "$[1].from",
          until: "$[1].until"
        },
        singleArgumentUnchanged: false
      },
      ["operationId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function calculateAvailableDatesInRange2(operationId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      operationId,
      deliveryAddress: options?.deliveryAddress,
      from: options?.from,
      until: options?.until
    }),
    [
      {
        transformFn: import_address.transformSDKAddressToRESTAddress,
        paths: [{ path: "deliveryAddress" }]
      }
    ]
  );
  const reqOpts = calculateAvailableDatesInRange(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          operationId: "$[0]",
          deliveryAddress: "$[1].deliveryAddress",
          from: "$[1].from",
          until: "$[1].until"
        },
        singleArgumentUnchanged: false
      },
      ["operationId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function validateOperationAddress2(operationId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    operationId
  });
  const reqOpts = validateOperationAddress(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { operationId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["operationId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateOperationTags2(operationIds, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    operationIds,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateOperationTags(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          operationIds: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["operationIds", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateOperationTagsByFilter2(filter, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    filter,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateOperationTagsByFilter(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          filter: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["filter", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/restaurants-operations-v1-operation-operations.public.ts
function createOperation3(httpClient) {
  return (operation) => createOperation2(
    operation,
    // @ts-ignore
    { httpClient }
  );
}
function getOperation3(httpClient) {
  return (operationId) => getOperation2(
    operationId,
    // @ts-ignore
    { httpClient }
  );
}
function updateOperation3(httpClient) {
  return (_id, operation) => updateOperation2(
    _id,
    operation,
    // @ts-ignore
    { httpClient }
  );
}
function deleteOperation3(httpClient) {
  return (operationId) => deleteOperation2(
    operationId,
    // @ts-ignore
    { httpClient }
  );
}
function queryOperation3(httpClient) {
  return () => queryOperation2(
    // @ts-ignore
    { httpClient }
  );
}
function listOperations3(httpClient) {
  return () => listOperations2(
    // @ts-ignore
    { httpClient }
  );
}
function listAvailableFulfillmentOptions3(httpClient) {
  return (operationId, options) => listAvailableFulfillmentOptions2(
    operationId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function listFirstAvailableTimeSlotForFulfillmentTypes3(httpClient) {
  return (operationId, options) => listFirstAvailableTimeSlotForFulfillmentTypes2(
    operationId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function calculateFirstAvailableTimeSlotPerFulfillmentType3(httpClient) {
  return (operationId, options) => calculateFirstAvailableTimeSlotPerFulfillmentType2(
    operationId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function listFirstAvailableTimeSlotsForOperations3(httpClient) {
  return (operationIds, options) => listFirstAvailableTimeSlotsForOperations2(
    operationIds,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function calculateFirstAvailableTimeSlotsPerOperation3(httpClient) {
  return (operationIds, options) => calculateFirstAvailableTimeSlotsPerOperation2(
    operationIds,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function listFirstAvailableTimeSlotsForMenus3(httpClient) {
  return (operationId, options) => listFirstAvailableTimeSlotsForMenus2(
    operationId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function calculateFirstAvailableTimeSlotsPerMenu3(httpClient) {
  return (operationId, options) => calculateFirstAvailableTimeSlotsPerMenu2(
    operationId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function calculateFirstAvailableTimeSlotsForItem3(httpClient) {
  return (operationId, menuId, options) => calculateFirstAvailableTimeSlotsForItem2(
    operationId,
    menuId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function listAvailableTimeSlotsForDate3(httpClient) {
  return (operationId, options) => listAvailableTimeSlotsForDate2(
    operationId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function calculateAvailableTimeSlotsForDate3(httpClient) {
  return (operationId, options) => calculateAvailableTimeSlotsForDate2(
    operationId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function listAvailableDatesInRange3(httpClient) {
  return (operationId, options) => listAvailableDatesInRange2(
    operationId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function calculateAvailableDatesInRange3(httpClient) {
  return (operationId, options) => calculateAvailableDatesInRange2(
    operationId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function validateOperationAddress3(httpClient) {
  return (operationId) => validateOperationAddress2(
    operationId,
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdateOperationTags3(httpClient) {
  return (operationIds, options) => bulkUpdateOperationTags2(
    operationIds,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdateOperationTagsByFilter3(httpClient) {
  return (filter, options) => bulkUpdateOperationTagsByFilter2(
    filter,
    options,
    // @ts-ignore
    { httpClient }
  );
}
var onOperationCreated = (0, import_sdk_types.EventDefinition)(
  "wix.restaurants.operations.v1.operation_created",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.pausedUntilOptions.time" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_address3.transformRESTAddressToSDKAddress,
        paths: [{ path: "entity.businessLocationDetails.address" }]
      }
    ])
  )
)();
var onOperationDeleted = (0, import_sdk_types.EventDefinition)(
  "wix.restaurants.operations.v1.operation_deleted",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.pausedUntilOptions.time" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_address3.transformRESTAddressToSDKAddress,
        paths: [{ path: "entity.businessLocationDetails.address" }]
      }
    ])
  )
)();
var onOperationUpdated = (0, import_sdk_types.EventDefinition)(
  "wix.restaurants.operations.v1.operation_updated",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.pausedUntilOptions.time" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_address3.transformRESTAddressToSDKAddress,
        paths: [{ path: "entity.businessLocationDetails.address" }]
      }
    ])
  )
)();

// src/restaurants-operations-v1-operation-operations.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var import_event_definition_modules = require("@wix/sdk-runtime/event-definition-modules");
var createOperation4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(createOperation3);
var getOperation4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getOperation3);
var updateOperation4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(updateOperation3);
var deleteOperation4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(deleteOperation3);
var queryOperation4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(queryOperation3);
var listOperations4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(listOperations3);
var listAvailableFulfillmentOptions4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(listAvailableFulfillmentOptions3);
var listFirstAvailableTimeSlotForFulfillmentTypes4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(
  listFirstAvailableTimeSlotForFulfillmentTypes3
);
var calculateFirstAvailableTimeSlotPerFulfillmentType4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(
  calculateFirstAvailableTimeSlotPerFulfillmentType3
);
var listFirstAvailableTimeSlotsForOperations4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(
  listFirstAvailableTimeSlotsForOperations3
);
var calculateFirstAvailableTimeSlotsPerOperation4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(
  calculateFirstAvailableTimeSlotsPerOperation3
);
var listFirstAvailableTimeSlotsForMenus4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(listFirstAvailableTimeSlotsForMenus3);
var calculateFirstAvailableTimeSlotsPerMenu4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(
  calculateFirstAvailableTimeSlotsPerMenu3
);
var calculateFirstAvailableTimeSlotsForItem4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(
  calculateFirstAvailableTimeSlotsForItem3
);
var listAvailableTimeSlotsForDate4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(listAvailableTimeSlotsForDate3);
var calculateAvailableTimeSlotsForDate4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(calculateAvailableTimeSlotsForDate3);
var listAvailableDatesInRange4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(listAvailableDatesInRange3);
var calculateAvailableDatesInRange4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(calculateAvailableDatesInRange3);
var validateOperationAddress4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(validateOperationAddress3);
var bulkUpdateOperationTags4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkUpdateOperationTags3);
var bulkUpdateOperationTagsByFilter4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkUpdateOperationTagsByFilter3);
var onOperationCreated2 = (0, import_event_definition_modules.createEventModule)(onOperationCreated);
var onOperationDeleted2 = (0, import_event_definition_modules.createEventModule)(onOperationDeleted);
var onOperationUpdated2 = (0, import_event_definition_modules.createEventModule)(onOperationUpdated);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ActivationType,
  AsapFutureHandlingType,
  AsapPacingWindowType,
  AsapPreorderType,
  DayOfWeek,
  DeleteStatus,
  EntitiesDayOfWeek,
  FulfillmentTimeType,
  FulfillmentTimesDisplayType,
  FulfillmentTimesType,
  FulfillmentType,
  MethodType,
  Namespace,
  OnlineOrderingStatusType,
  OrderSchedulingType,
  PlacementType,
  PreorderPacingWindowType,
  PreparationTimePreparationTimeType,
  PreparationTimeType,
  ResolutionMethod,
  SchedulingType,
  SiteCreatedContext,
  SortOrder,
  State,
  SubdivisionType,
  TimeSlotForFulfillmentTimeSlotStatus,
  TimeSlotStatus,
  TimeUnit,
  ViolationType,
  WebhookIdentityType,
  bulkUpdateOperationTags,
  bulkUpdateOperationTagsByFilter,
  calculateAvailableDatesInRange,
  calculateAvailableTimeSlotsForDate,
  calculateFirstAvailableTimeSlotPerFulfillmentType,
  calculateFirstAvailableTimeSlotsForItem,
  calculateFirstAvailableTimeSlotsPerMenu,
  calculateFirstAvailableTimeSlotsPerOperation,
  createOperation,
  deleteOperation,
  getOperation,
  listAvailableDatesInRange,
  listAvailableFulfillmentOptions,
  listAvailableTimeSlotsForDate,
  listFirstAvailableTimeSlotForFulfillmentTypes,
  listFirstAvailableTimeSlotsForMenus,
  listFirstAvailableTimeSlotsForOperations,
  listOperations,
  onOperationCreated,
  onOperationDeleted,
  onOperationUpdated,
  queryOperation,
  updateOperation,
  validateOperationAddress
});
