// @wix/headless-forms/dist/services/form-service.js
import { forms, submissions } from '@wix/forms';
import { defineService, implementService } from '@wix/services-definitions';
import { SignalsServiceDefinition, } from '@wix/services-definitions/core-services/signals';
/**
 * Service definition for the Form service.
 * This defines the contract that the FormService must implement.
 *
 * @constant
 */
export const FormServiceDefinition = defineService('formService');
/**
 * Implementation of the Form service that manages reactive form data and submissions.
 * This service provides signals for form data, loading state, error handling, and submission state.
 * It supports both pre-loaded form data and lazy loading with form IDs.
 * Consumers can provide a custom submission handler via config.
 *
 * @example
 * ```tsx
 * // Pre-loaded form data (SSR/SSG)
 * const formService = FormService.withConfig({ form: formData });
 *
 * // Lazy loading with form ID (client-side)
 * const formService = FormService.withConfig({ formId: 'form-123' });
 *
 * // With custom submission handler
 * const formService = FormService.withConfig({
 *   formId: 'form-123',
 *   onSubmit: async (formId, formValues) => {
 *     // Custom submission logic
 *     await fetch('/api/submit', { method: 'POST', body: JSON.stringify({ formId, ...formValues }) });
 *     return { type: 'success', message: 'Form submitted!' };
 *   }
 * });
 * ```
 */
export const FormService = implementService.withConfig()(FormServiceDefinition, ({ getService, config }) => {
    const signalsService = getService(SignalsServiceDefinition);
    const isLoadingSignal = signalsService.signal(false);
    const errorSignal = signalsService.signal(null);
    const submitResponseSignal = signalsService.signal({
        type: 'idle',
    });
    const hasSchema = 'form' in config;
    const formSignal = signalsService.signal(hasSchema ? config.form : null);
    if (!hasSchema) {
        loadForm(config.formId);
    }
    async function loadForm(id) {
        isLoadingSignal.set(true);
        errorSignal.set(null);
        try {
            const result = await fetchForm(id);
            if (result) {
                formSignal.set(result);
            }
            else {
                errorSignal.set('Form not found');
            }
        }
        catch (err) {
            errorSignal.set('Failed to load form');
            throw err;
        }
        finally {
            isLoadingSignal.set(false);
        }
    }
    async function defaultSubmitHandler(formId, formValues) {
        try {
            await submissions.createSubmission({
                formId,
                submissions: formValues,
            });
            // TODO: add message
            return { type: 'success' };
        }
        catch (error) {
            console.error('Form submission failed:', error);
            return { type: 'error', message: 'Failed to submit form' };
        }
    }
    /**
     * Submits the form with the provided values.
     * Uses custom handler if provided in config, otherwise uses default submission.
     */
    async function submitForm(formValues) {
        const form = formSignal.get();
        if (!form) {
            console.error('Cannot submit: form not loaded');
            return;
        }
        // @ts-expect-error
        const formId = form._id ? form._id : form.id;
        submitResponseSignal.set({ type: 'loading' });
        try {
            const handler = config.onSubmit || defaultSubmitHandler;
            const response = await handler(formId, formValues);
            submitResponseSignal.set(response);
        }
        catch (error) {
            console.error('Unexpected error during submission:', error);
            submitResponseSignal.set({
                type: 'error',
                message: 'Unexpected error during submission',
            });
        }
    }
    return {
        formSignal: formSignal,
        isLoadingSignal: isLoadingSignal,
        errorSignal: errorSignal,
        submitResponseSignal: submitResponseSignal,
        submitForm: submitForm,
    };
});
async function fetchForm(id) {
    try {
        const result = await forms.getForm(id);
        if (!result) {
            throw new Error(`Form ${id} not found`);
        }
        return result;
    }
    catch (err) {
        console.error('Failed to load form:', id, err);
        throw err;
    }
}
/**
 * Loads form service configuration from the Wix Forms API for SSR initialization.
 * This function is designed to be used during Server-Side Rendering (SSR) to preload
 * a specific form by ID that will be used to configure the FormService.
 *
 * @param {string} formId - The unique identifier of the form to load
 * @returns {Promise<FormServiceConfig>} Configuration object with pre-loaded form data
 * @throws {Error} When the form cannot be loaded
 *
 * @example
 * ```tsx
 * // In your SSR/SSG setup
 * const formConfig = await loadFormServiceConfig('form-123');
 * const formService = FormService.withConfig(formConfig);
 * ```
 */
export async function loadFormServiceConfig(formId) {
    const form = await fetchForm(formId);
    return { form };
}
