// @wix/bex-utils/@wix/ambassador-data-extensions-v1-data-extension-schema/build/es/builders.impl.js
import { mergeDeep, someNumber, someString, someAny, someDate, } from '@wix/metro-common-builders';
import { RequestedField, WebhookIdentityType, VersionStatus, } from './types.impl';
// @ts-ignore
import stack from 'callsite';
export function aDataExtensionSchema(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        id: someString({
            validations: [
                { type: 'FORMAT', value: 'GUID' },
                { type: 'READ_ONLY', value: 'true' },
                { type: 'IMMUTABLE', value: 'true' },
            ],
        }),
        fqdn: someString({
            validations: [
                { type: 'IMMUTABLE', value: 'true' },
                { type: 'MAX_LENGTH', value: '255' },
            ],
        }),
        namespace: someString({
            validations: [
                { type: 'IMMUTABLE', value: 'true' },
                { type: 'MAX_LENGTH', value: '164' },
            ],
        }),
        jsonSchema: {},
        updatedDate: someDate({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        createdDate: someDate({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        revision: someString({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'DECIMAL_VALUE' },
            ],
        }),
        appDefId: someString({
            validations: [
                { type: 'FORMAT', value: 'GUID' },
                { type: 'READ_ONLY', value: 'true' },
            ],
        }),
        maxLimitBytes: someNumber({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        currentSizeBytes: someNumber({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        extensionPoint: someString({
            validations: [
                { type: 'IMMUTABLE', value: 'true' },
                { type: 'MAX_LENGTH', value: '20' },
            ],
        }),
    }, overrideObject);
}
export function aReindexEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        fqdn: someString({
            validations: [{ type: 'MAX_LENGTH', value: '255' }],
        }),
        fields: [
            aReindexField(),
            aReindexField(),
            aReindexField(),
            aReindexField(),
            aReindexField(),
            aReindexField(),
            aReindexField(),
            aReindexField(),
            aReindexField(),
            aReindexField(),
        ],
    }, overrideObject);
}
export function aReindexField(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        fieldPath: someString({
            validations: [{ type: 'MAX_LENGTH', value: '2000' }],
        }),
        reindexSince: someDate(),
    }, overrideObject);
}
export function aCreateDataExtensionSchemaRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        dataExtensionSchema: aDataExtensionSchema(),
    }, overrideObject);
}
export function aCreateDataExtensionSchemaResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        dataExtensionSchema: aDataExtensionSchema(),
    }, overrideObject);
}
export function aUpdateDataExtensionSchemaRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        dataExtensionSchema: aDataExtensionSchema(),
    }, overrideObject);
}
export function aUpdateDataExtensionSchemaResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        dataExtensionSchema: aDataExtensionSchema(),
    }, overrideObject);
}
export function aListDataExtensionSchemasRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        fqdn: someString({
            validations: [{ type: 'MAX_LENGTH', value: '255' }],
        }),
        namespaces: [
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '164' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '164' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '164' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '164' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '164' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '164' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '164' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '164' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '164' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '100' },
                    { type: 'MAX_LENGTH', value: '164' },
                ],
            }),
        ],
        fields: [
            aRequestedField(),
            aRequestedField(),
            aRequestedField(),
            aRequestedField(),
            aRequestedField(),
        ],
        extensionPoints: [
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '20' },
                    { type: 'MAX_LENGTH', value: '10' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '20' },
                    { type: 'MAX_LENGTH', value: '10' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '20' },
                    { type: 'MAX_LENGTH', value: '10' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '20' },
                    { type: 'MAX_LENGTH', value: '10' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '20' },
                    { type: 'MAX_LENGTH', value: '10' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '20' },
                    { type: 'MAX_LENGTH', value: '10' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '20' },
                    { type: 'MAX_LENGTH', value: '10' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '20' },
                    { type: 'MAX_LENGTH', value: '10' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '20' },
                    { type: 'MAX_LENGTH', value: '10' },
                ],
            }),
            someString({
                validations: [
                    { type: 'MAX_SIZE', value: '20' },
                    { type: 'MAX_LENGTH', value: '10' },
                ],
            }),
        ],
    }, overrideObject);
}
export function aRequestedField(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = RequestedField.UNKNOWN_REQUESTED_FIELD; }
    return enumFirstItem;
}
export function aListDataExtensionSchemasResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        dataExtensionSchemas: [aDataExtensionSchema()],
    }, overrideObject);
}
export function aUpsertDataExtensionGlobalSchemaRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        dataExtensionSchema: aDataExtensionSchema(),
    }, overrideObject);
}
export function aUpsertDataExtensionGlobalSchemaResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        dataExtensionSchema: aDataExtensionSchema(),
    }, overrideObject);
}
export function aDeleteDemoDataExtensionSchemaRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        dataExtensionSchemaId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
export function aDeleteDemoDataExtensionSchemaResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({}, overrideObject);
}
export function aDeleteGlobalExtensionSchemaRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        fqdn: someString({
            validations: [{ type: 'MAX_LENGTH', value: '255' }],
        }),
        namespace: someString({
            validations: [{ type: 'MAX_LENGTH', value: '164' }],
        }),
    }, overrideObject);
}
export function aDeleteGlobalExtensionSchemaResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({}, overrideObject);
}
export function aDeleteByWhiteListedMetaSiteRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        metaSiteId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
export function aDeleteByWhiteListedMetaSiteResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        hasMore: true,
    }, overrideObject);
}
export function aDomainEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        createdEvent: aEntityCreatedEvent(),
        id: someString(),
        entityFqdn: someString(),
        slug: someString(),
        entityId: someString(),
        eventTime: someDate(),
        triggeredByAnonymizeRequest: true,
        originatedFrom: someString(),
        entityEventSequence: someString({
            validations: [{ type: 'DECIMAL_VALUE' }],
        }),
    }, overrideObject);
}
export function aEntityCreatedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        entityAsJson: someString(),
        triggeredByUndelete: true,
        restoreInfo: aRestoreInfo(),
        additionalMetadataAsJson: someString(),
    }, overrideObject);
}
export function aRestoreInfo(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        deletedDate: someDate(),
    }, overrideObject);
}
export function aEntityUpdatedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        currentEntityAsJson: someString(),
        previousEntityAsJson: someString(),
        modifiedFields: { modifiedFields: someAny() },
        additionalMetadataAsJson: someString(),
    }, overrideObject);
}
export function aEntityDeletedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        movedToTrash: true,
        deletedEntityAsJson: someString(),
        additionalMetadataAsJson: someString(),
    }, overrideObject);
}
export function aActionEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        bodyAsJson: someString(),
    }, overrideObject);
}
export function aEmpty(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({}, overrideObject);
}
export function aAppVersionStateChanged(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        created: aAppVersionCreated(),
        appId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        version: someString({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        previousState: aVersionStatus(),
    }, overrideObject);
}
export function aAppVersionCreated(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({}, overrideObject);
}
export function aAppVersionSubmitted(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({}, overrideObject);
}
export function aAppVersionPublished(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({}, overrideObject);
}
export function aAppVersionDeclined(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({}, overrideObject);
}
export function aAppVersionReleased(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({}, overrideObject);
}
export function aAppVersionApproved(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({}, overrideObject);
}
export function aAppVersionDraftChanged(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({}, overrideObject);
}
export function aAppVersionArchived(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({}, overrideObject);
}
export function aVersionStatus(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = VersionStatus.NONE_STATUS; }
    return enumFirstItem;
}
export function aMessageEnvelope(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        instanceId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        eventType: someString({
            validations: [{ type: 'MAX_LENGTH', value: '150' }],
        }),
        identity: aIdentificationData(),
        data: someString(),
    }, overrideObject);
}
export function aIdentificationData(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        anonymousVisitorId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        identityType: aWebhookIdentityType(),
    }, overrideObject);
}
export function aWebhookIdentityType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = WebhookIdentityType.UNKNOWN; }
    return enumFirstItem;
}
function isCurrentBuilderWasPreviouslyCalled() {
    var callChainFunctionNames = stack().map(function (item) {
        return item.getFunctionName();
    });
    var callChainWithoutCurrentBuilder = callChainFunctionNames.splice(2);
    var currentBuilderName = callChainFunctionNames[1];
    return callChainWithoutCurrentBuilder.some(function (functionName) { return functionName === currentBuilderName; });
}
export * from '@wix/metro-common-builders';
