// @wix/bex-utils/@wix/ambassador-data-extensions-v1-data-extension-schema/build/es/meta.impl.js
import * as httpWixDataExtensionsV1DataExtensionSchema from './http.impl';
export function createDataExtensionSchema() {
    var payload = {};
    var getRequestOptions = httpWixDataExtensionsV1DataExtensionSchema.createDataExtensionSchema(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/schemas',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateDataExtensionSchema() {
    var payload = {};
    var getRequestOptions = httpWixDataExtensionsV1DataExtensionSchema.updateDataExtensionSchema(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'PUT',
        path: '/v1/schemas',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listDataExtensionSchemas() {
    var payload = {};
    var getRequestOptions = httpWixDataExtensionsV1DataExtensionSchema.listDataExtensionSchemas(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/v1/schemas',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteByWhiteListedMetaSite() {
    var payload = {};
    var getRequestOptions = httpWixDataExtensionsV1DataExtensionSchema.deleteByWhiteListedMetaSite(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/schemas/delete-by-white-listed-meta-site',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
