// @wix/ricos/dist/es/plugin-table/tiptap/decorations/addColDecoration.js
import { Decoration } from "@tiptap/pm/view";
import { CATEGORY, LOCATION, SOURCE } from "../../consts.js";
import { TABLE_TYPE } from "../../types.js";
import { TableQuery } from "../TableQuery.js";
import styles from "./addRowCol-scss.js";
import { addSvg } from "./svgs.js";
const addColDecoration = (state, editor, services) => {
  const table = TableQuery.of(state.selection);
  const colNum = table.getWidth();
  const addColDiv = document.createElement("div");
  addColDiv.innerHTML = addSvg;
  addColDiv.classList.add(styles.tableAddCol);
  addColDiv.dataset.hook = "addCol";
  addColDiv.addEventListener("mousedown", (e) => {
    e.stopPropagation();
    e.preventDefault();
    editor.chain().addColumnAtIndex(colNum).selectColumnAtIndex(colNum).run();
    services.pluginsEvents.publishTableAddRowCol({
      pluginId: TABLE_TYPE,
      nodeId: table.getNode().attrs.id,
      category: CATEGORY.COLUMN,
      location: LOCATION.RIGHT,
      source: SOURCE.PLUS_BUTTON
    });
  });
  const colDecoration = Decoration.widget(table.getStartPos(), addColDiv);
  return colDecoration;
};
export {
  addColDecoration
};
