// @wix/ricos/dist/es/plugin-table/tiptap/decorations/addRowDecoration.js
import { Decoration } from "@tiptap/pm/view";
import { CATEGORY, LOCATION, SOURCE } from "../../consts.js";
import { TABLE_TYPE } from "../../types.js";
import { TableQuery } from "../TableQuery.js";
import styles from "./addRowCol-scss.js";
import { addSvg } from "./svgs.js";
const addRowDecoration = (state, editor, services) => {
  const table = TableQuery.of(state.selection);
  const rowNum = table.getHeight();
  const addRowDiv = document.createElement("div");
  addRowDiv.dataset.hook = "addRow";
  addRowDiv.innerHTML = addSvg;
  addRowDiv.classList.add(styles.tableAddRow);
  addRowDiv.addEventListener("mousedown", (e) => {
    e.stopPropagation();
    e.preventDefault();
    editor.chain().addRowAtIndex(rowNum).selectRowAtIndex(rowNum).run();
    services.pluginsEvents.publishTableAddRowCol({
      pluginId: TABLE_TYPE,
      nodeId: table.getNode().attrs.id,
      category: CATEGORY.ROW,
      location: LOCATION.BELOW,
      source: SOURCE.PLUS_BUTTON
    });
  });
  const rowDecoration = Decoration.widget(table.getStartPos(), addRowDiv);
  return rowDecoration;
};
export {
  addRowDecoration
};
