// @wix/headless-forms/cjs/dist/react/Form.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Field = exports.FieldInput = exports.FieldLabel = exports.Fields = exports.Submitted = exports.Error = exports.LoadingError = exports.Loading = exports.Root = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const react_1 = __importStar(require("react"));
const react_2 = require("@wix/headless-utils/react");
const form_public_1 = require("@wix/form-public");
const Form_js_1 = require("./core/Form.js");
var TestIds;
(function (TestIds) {
    TestIds["formRoot"] = "form-root";
    TestIds["form"] = "form";
    TestIds["formLoading"] = "form-loading";
    TestIds["formLoadingError"] = "form-loading-error";
    TestIds["formError"] = "form-error";
    TestIds["formSubmitted"] = "form-submitted";
    TestIds["fieldRoot"] = "field-root";
    TestIds["fieldLabel"] = "field-label";
    TestIds["fieldInput"] = "field-input";
})(TestIds || (TestIds = {}));
/**
 * Root component that provides all necessary service contexts for a complete form experience.
 * This component sets up the Form service and provides context to child components.
 * Must be used as the top-level component for all form functionality.
 *
 * @component
 * @param {RootProps} props - The component props
 * @param {React.ReactNode} props.children - Child components that will have access to form context
 * @param {FormServiceConfig} props.formServiceConfig - Form service configuration object
 * @param {boolean} [props.asChild] - Whether to render as a child component
 * @param {string} [props.className] - CSS classes to apply to the root element
 * @example
 * ```tsx
 * import { Form } from '@wix/headless-forms/react';
 * import { loadFormServiceConfig } from '@wix/headless-forms/services';
 *
 * const FIELD_MAP = {
 *   TEXT_INPUT: TextInput,
 *   TEXT_AREA: TextArea,
 *   CHECKBOX: Checkbox,
 *   // ... other field components
 * };
 *
 * // Pattern 1: Pre-loaded form data (SSR/SSG)
 * function FormPage({ formServiceConfig }) {
 *   return (
 *     <Form.Root formServiceConfig={formServiceConfig}>
 *       <Form.Loading className="flex justify-center p-4" />
 *       <Form.LoadingError className="text-destructive px-4 py-3 rounded mb-4" />
 *       <Form.Fields fieldMap={FIELD_MAP} />
 *       <Form.Error className="text-destructive p-4 rounded-lg mb-4" />
 *       <Form.Submitted className="text-green-500 p-4 rounded-lg mb-4" />
 *     </Form.Root>
 *   );
 * }
 *
 * // Pattern 2: Lazy loading with formId (Client-side)
 * function DynamicFormPage({ formId }) {
 *   return (
 *     <Form.Root formServiceConfig={{ formId }}>
 *       <Form.Loading className="flex justify-center p-4" />
 *       <Form.LoadingError className="text-destructive px-4 py-3 rounded mb-4" />
 *       <Form.Fields fieldMap={FIELD_MAP} />
 *       <Form.Error className="text-destructive p-4 rounded-lg mb-4" />
 *       <Form.Submitted className="text-green-500 p-4 rounded-lg mb-4" />
 *     </Form.Root>
 *   );
 * }
 * ```
 */
exports.Root = react_1.default.forwardRef((props, ref) => {
    const { children, formServiceConfig, asChild, ...otherProps } = props;
    return ((0, jsx_runtime_1.jsx)(Form_js_1.Root, { formServiceConfig: formServiceConfig, children: (0, jsx_runtime_1.jsx)(RootContent, { asChild: asChild, ref: ref, ...otherProps, children: children }) }));
});
/**
 * Internal component to handle the Root content with service access.
 * This component wraps the children with the necessary div container and applies styling.
 *
 * @internal
 * @param {RootContentProps} props - Component props
 * @param {React.ReactNode} props.children - Child components to render
 * @param {string} [props.className] - CSS classes to apply to the container
 * @param {boolean} [props.asChild] - Whether to render as a child component
 * @returns {JSX.Element} The wrapped content
 */
const RootContent = react_1.default.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return ((0, jsx_runtime_1.jsx)(react_2.AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.formRoot, customElement: children, customElementProps: {}, ...otherProps, children: (0, jsx_runtime_1.jsx)("div", { children: children }) }));
});
/**
 * Component that renders content during loading state.
 * Only displays its children when the form is currently loading.
 *
 * @component
 * @param {LoadingProps} props - The component props
 * @param {boolean} [props.asChild] - Whether to render as a child component
 * @param {React.ReactNode} [props.children] - Content to display during loading state
 * @param {string} [props.className] - CSS classes to apply to the default element
 * @example
 * ```tsx
 * import { Form } from '@wix/headless-forms/react';
 *
 * // Default usage with className
 * function FormLoading() {
 *   return (
 *     <Form.Loading className="flex justify-center p-4" />
 *   );
 * }
 *
 * // Custom content
 * function CustomFormLoading() {
 *   return (
 *     <Form.Loading>
 *       <div className="flex justify-center items-center p-4">
 *         <div className="animate-spin rounded-full h-8 w-8 border-b-2 border-primary"></div>
 *         <span className="ml-2 text-foreground font-paragraph">Loading form...</span>
 *       </div>
 *     </Form.Loading>
 *   );
 * }
 *
 * // With asChild for custom components
 * function CustomFormLoadingAsChild() {
 *   return (
 *     <Form.Loading asChild>
 *       <div className="custom-loading-container">
 *         <div className="animate-spin rounded-full h-8 w-8 border-b-2 border-primary"></div>
 *         <span className="ml-2 text-foreground font-paragraph">Loading form...</span>
 *       </div>
 *     </Form.Loading>
 *   );
 * }
 * ```
 */
exports.Loading = react_1.default.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return ((0, jsx_runtime_1.jsx)(Form_js_1.Loading, { children: ({ isLoading }) => {
            if (!isLoading)
                return null;
            return ((0, jsx_runtime_1.jsx)(react_2.AsChildSlot, { "data-testid": TestIds.formLoading, ref: ref, asChild: asChild, className: className, customElement: children, content: "Loading form...", ...otherProps, children: (0, jsx_runtime_1.jsx)("div", { children: "Loading form..." }) }));
        } }));
});
/**
 * Component that renders content when there's an error loading the form.
 * Only displays its children when an error has occurred.
 *
 * @component
 * @param {LoadingErrorProps} props - The component props
 * @param {boolean} [props.asChild] - Whether to render as a child component
 * @param {React.ReactNode} [props.children] - Content to display during error state
 * @param {string} [props.className] - CSS classes to apply to the default element
 * @example
 * ```tsx
 * import { Form } from '@wix/headless-forms/react';
 *
 * // Default usage with className
 * function FormLoadingError() {
 *   return (
 *     <Form.LoadingError className="text-destructive px-4 py-3 rounded mb-4" />
 *   );
 * }
 *
 * // Custom content
 * function CustomLoadingError() {
 *   return (
 *     <Form.LoadingError>
 *       <div className="bg-destructive/10 border border-destructive text-destructive px-4 py-3 rounded mb-4">
 *         <h3 className="font-heading text-lg">Error loading form</h3>
 *         <p className="font-paragraph">Something went wrong. Please try again.</p>
 *       </div>
 *     </Form.LoadingError>
 *   );
 * }
 *
 * // With asChild for custom components
 * function CustomLoadingErrorAsChild() {
 *   return (
 *     <Form.LoadingError asChild>
 *       {React.forwardRef<HTMLDivElement, { error: string | null; hasError: boolean }>(
 *         ({ error }, ref) => (
 *           <div ref={ref} className="custom-error-container">
 *             <h3 className="font-heading">Error Loading Form</h3>
 *             <p className="font-paragraph">{error}</p>
 *           </div>
 *         )
 *       )}
 *     </Form.LoadingError>
 *   );
 * }
 * ```
 */
exports.LoadingError = react_1.default.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return ((0, jsx_runtime_1.jsx)(Form_js_1.LoadingError, { children: ({ error, hasError }) => {
            if (!hasError)
                return null;
            const errorData = { error, hasError };
            return ((0, jsx_runtime_1.jsx)(react_2.AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.formLoadingError, customElement: children, customElementProps: errorData, content: error, ...otherProps, children: (0, jsx_runtime_1.jsx)("div", { children: error }) }));
        } }));
});
/**
 * Component that renders content when there's an error during form submission.
 * Only displays its children when a submission error has occurred.
 *
 * @component
 * @param {ErrorProps} props - The component props
 * @param {boolean} [props.asChild] - Whether to render as a child component
 * @param {React.ReactNode} [props.children] - Content to display during submit error state
 * @param {string} [props.className] - CSS classes to apply to the default element
 * @example
 * ```tsx
 * import { Form } from '@wix/headless-forms/react';
 *
 * // Default usage with className
 * function FormError() {
 *   return <Form.Error className="text-destructive p-4 rounded-lg mb-4" />;
 * }
 *
 * // Custom content
 * function CustomFormError() {
 *   return (
 *     <Form.Error>
 *       <div className="bg-destructive/10 border border-destructive text-destructive p-4 rounded-lg mb-4">
 *         <h3 className="font-heading text-lg">Submission Failed</h3>
 *         <p className="font-paragraph">Please check your input and try again.</p>
 *       </div>
 *     </Form.Error>
 *   );
 * }
 *
 * // With asChild for custom components
 * function CustomFormErrorAsChild() {
 *   return (
 *     <Form.Error asChild>
 *       {React.forwardRef<HTMLDivElement, { error: string | null; hasError: boolean }>(
 *         ({ error }, ref) => (
 *           <div ref={ref} className="custom-error-container">
 *             <h3 className="font-heading">Submission Failed</h3>
 *             <p className="font-paragraph">{error}</p>
 *           </div>
 *         )
 *       )}
 *     </Form.Error>
 *   );
 * }
 * ```
 */
exports.Error = react_1.default.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return ((0, jsx_runtime_1.jsx)(Form_js_1.Error, { children: ({ error, hasError }) => {
            if (!hasError)
                return null;
            const errorData = { error, hasError };
            return ((0, jsx_runtime_1.jsx)(react_2.AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.formError, customElement: children, customElementProps: errorData, content: error, ...otherProps, children: (0, jsx_runtime_1.jsx)("div", { className: "text-destructive text-sm sm:text-base", children: error }) }));
        } }));
});
/**
 * Component that renders content after successful form submission.
 * Only displays its children when the form has been successfully submitted.
 *
 * @component
 * @param {SubmittedProps} props - The component props
 * @param {boolean} [props.asChild] - Whether to render as a child component
 * @param {React.ReactNode} [props.children] - Content to display after successful submission
 * @param {string} [props.className] - CSS classes to apply to the default element
 * @example
 * ```tsx
 * import { Form } from '@wix/headless-forms/react';
 *
 * // Default usage with className
 * function FormSubmitted() {
 *   return <Form.Submitted className="text-green-500 p-4 rounded-lg mb-4" />;
 * }
 *
 * // Custom content
 * function CustomFormSubmitted() {
 *   return (
 *     <Form.Submitted>
 *       <div className="bg-green-50 border border-green-200 text-green-800 p-6 rounded-lg mb-4">
 *         <h2 className="font-heading text-xl mb-2">Thank You!</h2>
 *         <p className="font-paragraph">Your form has been submitted successfully.</p>
 *       </div>
 *     </Form.Submitted>
 *   );
 * }
 *
 * // With asChild for custom components
 * function CustomFormSubmittedAsChild() {
 *   return (
 *     <Form.Submitted asChild>
 *       {React.forwardRef<HTMLDivElement, { isSubmitted: boolean; message: string }>(
 *         ({ message }, ref) => (
 *           <div ref={ref} className="custom-success-container">
 *             <h2 className="font-heading">Thank You!</h2>
 *             <p className="font-paragraph">{message}</p>
 *           </div>
 *         )
 *       )}
 *     </Form.Submitted>
 *   );
 * }
 * ```
 */
exports.Submitted = react_1.default.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return ((0, jsx_runtime_1.jsx)(Form_js_1.Submitted, { children: ({ isSubmitted, message }) => {
            if (!isSubmitted)
                return null;
            const submittedData = { isSubmitted, message };
            return ((0, jsx_runtime_1.jsx)(react_2.AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.formSubmitted, customElement: children, customElementProps: submittedData, content: message, ...otherProps, children: (0, jsx_runtime_1.jsx)("div", { className: "text-green-500 text-sm sm:text-base", children: message }) }));
        } }));
});
/**
 * Fields component for rendering a form with custom field renderers.
 * This component handles the rendering of form fields based on the provided fieldMap.
 * Must be used within Form.Root to access form context.
 *
 * @component
 * @param {FieldsProps} props - Component props
 * @param {FieldMap} props.fieldMap - A mapping of field types to their corresponding React components
 * @param {string} props.rowGapClassname - CSS class name for gap between rows
 * @param {string} props.columnGapClassname - CSS class name for gap between columns
 * @example
 * ```tsx
 * import { Form } from '@wix/headless-forms/react';
 * import { TextInput, TextArea, Checkbox } from './field-components';
 *
 * const FIELD_MAP = {
 *   TEXT_INPUT: TextInput,
 *   TEXT_AREA: TextArea,
 *   CHECKBOX: Checkbox,
 *   NUMBER_INPUT: NumberInput,
 *   // ... remaining field components
 * };
 *
 * function ContactForm({ formServiceConfig }) {
 *   return (
 *     <Form.Root formServiceConfig={formServiceConfig}>
 *       <Form.Loading className="flex justify-center p-4" />
 *       <Form.LoadingError className="text-destructive px-4 py-3 rounded mb-4" />
 *       <Form.Fields
 *         fieldMap={FIELD_MAP}
 *         rowGapClassname="gap-y-4"
 *         columnGapClassname="gap-x-2"
 *       />
 *     </Form.Root>
 *   );
 * }
 * ```
 */
/**
 * Fields component for rendering a form with custom field renderers.
 * It maps each field type from the form configuration to its corresponding React component
 * and renders them in the order and layout defined by the form structure.
 *
 * The component automatically handles:
 * - Field validation and error display
 * - Form state management
 * - Field value updates
 * - Grid layout with configurable row and column gaps
 *
 * Must be used within Form.Root to access form context.
 *
 * @component
 * @param {FieldsProps} props - The component props
 * @param {FieldMap} props.fieldMap - A mapping of field types to their corresponding React components. Each key represents a field type (e.g., 'TEXT_INPUT', 'CHECKBOX') and the value is the React component that should render that field type.
 * @param {string} props.rowGapClassname - CSS class name for gap between form rows
 * @param {string} props.columnGapClassname - CSS class name for gap between form columns
 *
 * @example
 * ```tsx
 * import { Form } from '@wix/headless-forms/react';
 * import { loadFormServiceConfig } from '@wix/headless-forms/services';
 * import {
 *   TextInput,
 *   TextArea,
 *   PhoneInput,
 *   MultilineAddress,
 *   DateInput,
 *   DatePicker,
 *   DateTimeInput,
 *   FileUpload,
 *   NumberInput,
 *   Checkbox,
 *   Signature,
 *   RatingInput,
 *   RadioGroup,
 *   CheckboxGroup,
 *   Dropdown,
 *   Tags,
 *   TimeInput,
 *   RichText,
 *   SubmitButton,
 *   ProductList,
 *   FixedPayment,
 *   PaymentInput,
 *   Donation,
 *   Appointment,
 *   ImageChoice
 * } from './components';
 *
 * // Define your field mapping - this tells the Fields component which React component to use for each field type
 * const FIELD_MAP = {
 *   TEXT_INPUT: TextInput,
 *   TEXT_AREA: TextArea,
 *   PHONE_INPUT: PhoneInput,
 *   MULTILINE_ADDRESS: MultilineAddress,
 *   DATE_INPUT: DateInput,
 *   DATE_PICKER: DatePicker,
 *   DATE_TIME_INPUT: DateTimeInput,
 *   FILE_UPLOAD: FileUpload,
 *   NUMBER_INPUT: NumberInput,
 *   CHECKBOX: Checkbox,
 *   SIGNATURE: Signature,
 *   RATING_INPUT: RatingInput,
 *   RADIO_GROUP: RadioGroup,
 *   CHECKBOX_GROUP: CheckboxGroup,
 *   DROPDOWN: Dropdown,
 *   TAGS: Tags,
 *   TIME_INPUT: TimeInput,
 *   TEXT: RichText,
 *   SUBMIT_BUTTON: SubmitButton,
 *   PRODUCT_LIST: ProductList,
 *   FIXED_PAYMENT: FixedPayment,
 *   PAYMENT_INPUT: PaymentInput,
 *   DONATION: Donation,
 *   APPOINTMENT: Appointment,
 *   IMAGE_CHOICE: ImageChoice,
 * };
 *
 * function ContactForm({ formServiceConfig }) {
 *   return (
 *     <Form.Root formServiceConfig={formServiceConfig}>
 *       <Form.Loading className="flex justify-center p-4" />
 *       <Form.LoadingError className="text-destructive px-4 py-3 rounded mb-4" />
 *       <Form.Fields
 *         fieldMap={FIELD_MAP}
 *         rowGapClassname="gap-y-4"
 *         columnGapClassname="gap-x-2"
 *       />
 *       <Form.Error className="text-destructive p-4 rounded-lg mb-4" />
 *       <Form.Submitted className="text-green-500 p-4 rounded-lg mb-4" />
 *     </Form.Root>
 *   );
 * }
 * ```
 *
 * @example
 * ```tsx
 * // Creating custom field components - ALL field components MUST use Form.Field
 * // This example shows the REQUIRED structure for a TEXT_INPUT component
 * import { Form, type TextInputProps } from '@wix/headless-forms/react';
 *
 * const TextInput = (props: TextInputProps) => {
 *   const { id, value, onChange, label, error, required, ...inputProps } = props;
 *
 *   // Form.Field provides automatic grid layout positioning
 *   return (
 *     <Form.Field id={id}>
 *       <Form.Field.Label>
 *         <label className="text-foreground font-paragraph">
 *           {label}
 *           {required && <span className="text-destructive ml-1">*</span>}
 *         </label>
 *       </Form.Field.Label>
 *       <Form.Field.Input
 *         description={error && <span className="text-destructive text-sm">{error}</span>}
 *       >
 *         <input
 *           type="text"
 *           value={value || ''}
 *           onChange={(e) => onChange(e.target.value)}
 *           className="bg-background border-foreground text-foreground"
 *           aria-invalid={!!error}
 *           {...inputProps}
 *         />
 *       </Form.Field.Input>
 *     </Form.Field>
 *   );
 * };
 *
 * const FIELD_MAP = {
 *   TEXT_INPUT: TextInput,
 *   // ... all other field components must also use Form.Field
 * };
 * ```
 */
exports.Fields = react_1.default.forwardRef((props, ref) => {
    const [formValues, setFormValues] = (0, react_1.useState)({});
    const [formErrors, setFormErrors] = (0, react_1.useState)([]);
    const handleFormChange = (0, react_1.useCallback)((values) => {
        setFormValues(values);
    }, []);
    const handleFormValidate = (0, react_1.useCallback)((errors) => {
        setFormErrors(errors);
    }, []);
    return ((0, jsx_runtime_1.jsx)(Form_js_1.Fields, { children: ({ form, submitForm }) => {
            if (!form)
                return null;
            return ((0, jsx_runtime_1.jsx)("div", { ref: ref, children: (0, jsx_runtime_1.jsx)(form_public_1.FormProvider, { children: (0, jsx_runtime_1.jsx)(FieldsWithForm, { form: form, values: formValues, onChange: handleFormChange, errors: formErrors, onValidate: handleFormValidate, fields: props.fieldMap, submitForm: () => submitForm(formValues), rowGapClassname: props.rowGapClassname, columnGapClassname: props.columnGapClassname }) }) }));
        } }));
});
const FieldsWithForm = ({ form, submitForm, values, onChange, errors, onValidate, fields: fieldMap, rowGapClassname, columnGapClassname, }) => {
    const formData = (0, form_public_1.useForm)({
        form,
        values,
        onChange,
        errors,
        onValidate,
        submitForm,
        fieldMap,
    });
    if (!formData)
        return null;
    console.log('formData', formData);
    const { columnCount, fieldElements, fieldsLayout } = formData;
    return (
    // TODO: use readOnly, isDisabled
    // TODO: step title a11y support
    // TODO: mobile support?
    (0, jsx_runtime_1.jsx)(FieldLayoutProvider, { value: fieldsLayout, children: (0, jsx_runtime_1.jsx)("form", { onSubmit: (e) => e.preventDefault(), children: (0, jsx_runtime_1.jsx)("fieldset", { style: { display: 'flex', flexDirection: 'column' }, className: rowGapClassname, children: fieldElements.map((rowElements, index) => {
                    return ((0, jsx_runtime_1.jsx)("div", { style: {
                            display: 'grid',
                            width: '100%',
                            gridTemplateColumns: `repeat(${columnCount}, 1fr)`,
                            gridAutoRows: 'minmax(min-content, max-content)',
                        }, className: columnGapClassname, children: rowElements }, index));
                }) }) }) }));
};
/**
 * Context for sharing field layout data across the form
 * @internal
 */
const FieldLayoutContext = react_1.default.createContext(null);
/**
 * Provider component that makes field layout data available to child components
 * @internal
 */
const FieldLayoutProvider = ({ value, children, }) => {
    return ((0, jsx_runtime_1.jsx)(FieldLayoutContext.Provider, { value: value, children: children }));
};
/**
 * Hook to access layout configuration for a specific field
 * @internal
 * @param {string} fieldId - The unique identifier of the field
 * @returns {Layout | null} The layout configuration for the field, or null if not found
 */
function useFieldLayout(fieldId) {
    const layoutMap = react_1.default.useContext(FieldLayoutContext);
    if (!layoutMap) {
        return null;
    }
    return layoutMap[fieldId] || null;
}
const FieldContext = react_1.default.createContext(null);
/**
 * Hook to access field context
 */
function useFieldContext() {
    const context = react_1.default.useContext(FieldContext);
    if (!context) {
        throw new globalThis.Error('Field components must be used within a Form.Field component');
    }
    return context;
}
/**
 * Container component for a form field with grid layout support.
 * Provides context to Field.Label and Field.Input child components.
 * Based on the default-field-layout functionality.
 *
 * @component
 * @example
 * ```tsx
 * import { Form } from '@wix/headless-forms/react';
 *
 * function FormFields() {
 *   return (
 *     <Form.Field id="username">
 *       <Form.Field.Label>
 *         <label className="text-foreground font-paragraph">Username</label>
 *       </Form.Field.Label>
 *       <Form.Field.Input description={<span className="text-secondary-foreground">Required</span>}>
 *         <input className="bg-background border-foreground text-foreground" />
 *       </Form.Field.Input>
 *     </Form.Field>
 *   );
 * }
 * ```
 */
const FieldRoot = react_1.default.forwardRef((props, ref) => {
    const { id, children, asChild, className, ...otherProps } = props;
    const layout = useFieldLayout(id);
    if (!layout) {
        return null;
    }
    return ((0, jsx_runtime_1.jsx)(Form_js_1.Field, { id: id, layout: layout, children: (fieldData) => {
            const contextValue = {
                id,
                layout: fieldData.layout,
                gridStyles: fieldData.gridStyles,
            };
            return ((0, jsx_runtime_1.jsx)(FieldContext.Provider, { value: contextValue, children: (0, jsx_runtime_1.jsx)(react_2.AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.fieldRoot, customElement: children, customElementProps: {}, ...otherProps, children: children }) }));
        } }));
});
FieldRoot.displayName = 'Form.Field';
/**
 * Label component for a form field with automatic grid positioning.
 * Must be used within a Form.Field component.
 * Renders in the label row of the field's grid layout.
 *
 * @component
 * @example
 * ```tsx
 * import { Form } from '@wix/headless-forms/react';
 *
 * <Form.Field id="email">
 *   <Form.Field.Label>
 *     <label className="text-foreground font-paragraph">Email Address</label>
 *   </Form.Field.Label>
 *   <Form.Field.Input>
 *     <input type="email" className="bg-background border-foreground" />
 *   </Form.Field.Input>
 * </Form.Field>
 * ```
 */
exports.FieldLabel = react_1.default.forwardRef((props, ref) => {
    const { children, asChild, className, ...otherProps } = props;
    const { gridStyles } = useFieldContext();
    return ((0, jsx_runtime_1.jsx)(react_2.AsChildSlot, { ref: ref, asChild: asChild, className: className, style: gridStyles.label, "data-testid": TestIds.fieldLabel, customElement: children, customElementProps: {}, ...otherProps, children: (0, jsx_runtime_1.jsx)("div", { children: children }) }));
});
exports.FieldLabel.displayName = 'Form.Field.Label';
/**
 * Input component for a form field with automatic grid positioning.
 * Must be used within a Form.Field component.
 * Renders in the input row of the field's grid layout with optional description.
 *
 * @component
 * @example
 * ```tsx
 * import { Form } from '@wix/headless-forms/react';
 *
 * <Form.Field id="password">
 *   <Form.Field.Label>
 *     <label className="text-foreground font-paragraph">Password</label>
 *   </Form.Field.Label>
 *   <Form.Field.Input description={<span className="text-secondary-foreground">Min 8 characters</span>}>
 *     <input type="password" className="bg-background border-foreground text-foreground" />
 *   </Form.Field.Input>
 * </Form.Field>
 * ```
 */
exports.FieldInput = react_1.default.forwardRef((props, ref) => {
    const { children, description, asChild, className, ...otherProps } = props;
    const { gridStyles } = useFieldContext();
    return ((0, jsx_runtime_1.jsx)(react_2.AsChildSlot, { ref: ref, asChild: asChild, className: className, style: gridStyles.input, "data-testid": TestIds.fieldInput, customElement: children, customElementProps: {}, ...otherProps, children: (0, jsx_runtime_1.jsx)("div", { children: children }) }));
});
exports.FieldInput.displayName = 'Form.Field.Input';
exports.Field = FieldRoot;
exports.Field.Label = exports.FieldLabel;
exports.Field.Input = exports.FieldInput;
