// @wix/restaurants/build/es/meta.mjs
// meta.ts
import * as localDeliverySpi from "@wix/auto_sdk_restaurants_local-delivery-spi/meta";
import * as catalogs from "@wix/auto_sdk_restaurants_catalogs/meta";
import * as discount from "@wix/auto_sdk_restaurants_discount/meta";
import * as fulfillemtMethods from "@wix/auto_sdk_restaurants_fulfillemt-methods/meta";
import * as menuOrderingSettings from "@wix/auto_sdk_restaurants_menu-ordering-settings/meta";
import * as items from "@wix/auto_sdk_restaurants_items/meta";
import * as itemLabels from "@wix/auto_sdk_restaurants_item-labels/meta";
import * as itemModifiers from "@wix/auto_sdk_restaurants_item-modifiers/meta";
import * as itemModifierGroups from "@wix/auto_sdk_restaurants_item-modifier-groups/meta";
import * as itemVariants from "@wix/auto_sdk_restaurants_item-variants/meta";
import * as menus from "@wix/auto_sdk_restaurants_menus/meta";
import * as sections from "@wix/auto_sdk_restaurants_sections/meta";
import * as operations from "@wix/auto_sdk_restaurants_operations/meta";
import * as orderPacingOrder from "@wix/auto_sdk_restaurants_order-pacing-order/meta";
import * as orderPacingPolicy from "@wix/auto_sdk_restaurants_order-pacing-policy/meta";
import * as orderSettings from "@wix/auto_sdk_restaurants_order-settings/meta";
import * as posTpa from "@wix/auto_sdk_restaurants_pos-tpa/meta";
import * as recipients from "@wix/auto_sdk_restaurants_recipients/meta";
import * as restaurantLocations from "@wix/auto_sdk_restaurants_restaurant-locations/meta";
import * as availabilityExceptions from "@wix/auto_sdk_restaurants_availability-exceptions/meta";
import * as fulfillmentMethods from "@wix/auto_sdk_restaurants_fulfillment-methods/meta";
import * as operationGroups from "@wix/auto_sdk_restaurants_operation-groups/meta";
import * as orders from "@wix/auto_sdk_restaurants_orders/meta";
import * as serviceFees from "@wix/auto_sdk_restaurants_service-fees/meta";
export {
  availabilityExceptions,
  catalogs,
  discount,
  fulfillemtMethods,
  fulfillmentMethods,
  itemLabels,
  itemModifierGroups,
  itemModifiers,
  itemVariants,
  items,
  localDeliverySpi,
  menuOrderingSettings,
  menus,
  operationGroups,
  operations,
  orderPacingOrder,
  orderPacingPolicy,
  orderSettings,
  orders,
  posTpa,
  recipients,
  restaurantLocations,
  sections,
  serviceFees
};
