// @wix/headless-media/cjs/dist/react/core/MediaGallery.js
import { jsx as _jsx } from "react/jsx-runtime";
import { useService, WixServices } from '@wix/services-manager-react';
import { MediaGalleryServiceDefinition, MediaGalleryService, } from '../../services/index.js';
import { createServicesMap } from '@wix/services-manager';
/**
 * Root component that provides the MediaGallery service context to its children.
 * This component sets up the necessary services for rendering and managing media gallery functionality.
 *
 * @order 1
 * @component
 * @example
 * ```tsx
 * import { MediaGallery } from '@wix/media/components';
 *
 * function ProductMediaGallery({ productMedia }) {
 *   return (
 *     <MediaGallery.Root mediaGalleryServiceConfig={{ media: productMedia }}>
 *       <div className="media-gallery">
 *         <MediaGallery.Viewport>
 *           {({ src, alt }) => (
 *             <img
 *               src={src || '/placeholder.jpg'}
 *               alt={alt}
 *               className="main-media"
 *             />
 *           )}
 *         </MediaGallery.Viewport>
 *         <MediaGallery.ThumbnailList>
 *           {({ items }) => (
 *             <div className="thumbnail-grid">
 *               {items.map((item, index) => (
 *                 <MediaGallery.ThumbnailItem key={index} index={index}>
 *                   {({ src, isActive, onSelect, alt }) => (
 *                     <button
 *                       onClick={onSelect}
 *                       className={`thumbnail ${isActive ? 'active' : ''}`}
 *                     >
 *                       <img src={src} alt={alt} />
 *                     </button>
 *                   )}
 *                 </MediaGallery.ThumbnailItem>
 *               ))}
 *             </div>
 *           )}
 *         </MediaGallery.ThumbnailList>
 *       </div>
 *     </MediaGallery.Root>
 *   );
 * }
 * ```
 */
export function Root(props) {
    return (_jsx(WixServices, { servicesMap: createServicesMap().addService(MediaGalleryServiceDefinition, MediaGalleryService, props.mediaGalleryServiceConfig), children: props.children }));
}
/**
 * Headless component for displaying the main viewport media
 *
 * @component
 * @example
 * ```tsx
 * import { MediaGallery } from '@wix/media/components';
 *
 * function MediaViewer() {
 *   return (
 *     <MediaGallery.Viewport>
 *       {({ src, alt }) => (
 *         <img
 *           src={src || '/placeholder.jpg'}
 *           alt={alt}
 *           className="main-media"
 *         />
 *       )}
 *     </MediaGallery.Viewport>
 *   );
 * }
 * ```
 */
export const Viewport = (props) => {
    const mediaService = useService(MediaGalleryServiceDefinition);
    const currentIndex = mediaService.selectedMediaIndex.get();
    const mediaToDisplay = mediaService.mediaToDisplay.get();
    if (mediaToDisplay.length === 0) {
        return null;
    }
    // Get the current media from the relevant media array
    const src = mediaToDisplay[currentIndex].image;
    const alt = mediaToDisplay[currentIndex].altText;
    return props.children({
        src,
        alt,
    });
};
/**
 * Headless component for managing a list of thumbnails
 *
 * @component
 * @example
 * ```tsx
 * import { MediaGallery } from '@wix/media/components';
 *
 * function ThumbnailGrid() {
 *   return (
 *     <MediaGallery.ThumbnailList>
 *       {({ items }) => (
 *         <div className="thumbnail-grid">
 *           {items.map((item, index) => (
 *             <MediaGallery.ThumbnailItem key={index} index={index}>
 *               {({ src, isActive, onSelect, alt }) => (
 *                 <button
 *                   onClick={onSelect}
 *                   className={`thumbnail ${isActive ? 'active' : ''}`}
 *                 >
 *                   <img src={src} alt={alt} />
 *                 </button>
 *               )}
 *             </MediaGallery.ThumbnailItem>
 *           ))}
 *         </div>
 *       )}
 *     </MediaGallery.ThumbnailList>
 *   );
 * }
 * ```
 */
export const ThumbnailList = (props) => {
    const mediaService = useService(MediaGalleryServiceDefinition);
    const mediaToDisplay = mediaService.mediaToDisplay.get();
    if (mediaToDisplay.length <= 1) {
        return null;
    }
    return props.children({
        items: mediaToDisplay,
    });
};
/**
 * Headless component for individual media thumbnail
 *
 * @component
 * @example
 * ```tsx
 * import { MediaGallery } from '@wix/media/components';
 *
 * function ThumbnailButton({ index }) {
 *   return (
 *     <MediaGallery.ThumbnailItem index={index}>
 *       {({ src, isActive, select, alt }) => (
 *         <button
 *           onClick={select}
 *           className={`thumbnail-btn ${isActive ? 'active' : ''}`}
 *         >
 *           <img src={src} alt={alt} />
 *           {isActive && <div className="active-indicator" />}
 *         </button>
 *       )}
 *     </MediaGallery.ThumbnailItem>
 *   );
 * }
 * ```
 */
export const ThumbnailItem = (props) => {
    const mediaService = useService(MediaGalleryServiceDefinition);
    const currentIndex = mediaService.selectedMediaIndex.get();
    const mediaToDisplay = mediaService.mediaToDisplay.get();
    if (mediaToDisplay.length === 0) {
        return null;
    }
    // Get the image source from the centralized relevant images
    const src = mediaToDisplay[props.index].image;
    const alt = mediaToDisplay[props.index].altText;
    const isActive = currentIndex === props.index;
    const select = () => {
        mediaService.setSelectedMediaIndex(props.index);
    };
    return props.children({
        item: mediaToDisplay[props.index],
        src,
        isActive,
        select,
        index: props.index,
        alt,
    });
};
/**
 * Headless component for next media navigation
 *
 * @component
 * @example
 * ```tsx
 * import { MediaGallery } from '@wix/media/components';
 *
 * function NextButton() {
 *   return (
 *     <MediaGallery.Next>
 *       {({ next, canGoNext }) => (
 *         <button
 *           onClick={next}
 *           disabled={!canGoNext}
 *           className="nav-btn next-btn"
 *         >
 *           Next →
 *         </button>
 *       )}
 *     </MediaGallery.Next>
 *   );
 * }
 * ```
 */
export const Next = (props) => {
    const mediaService = useService(MediaGalleryServiceDefinition);
    return props.children({
        next: mediaService.nextMedia,
        canGoNext: mediaService.hasNextMedia(),
    });
};
/**
 * Headless component for previous media navigation
 *
 * @component
 * @example
 * ```tsx
 * import { MediaGallery } from '@wix/media/components';
 *
 * function PreviousButton() {
 *   return (
 *     <MediaGallery.Previous>
 *       {({ previous, canGoPrevious }) => (
 *         <button
 *           onClick={previous}
 *           disabled={!canGoPrevious}
 *           className="nav-btn prev-btn"
 *         >
 *           ← Previous
 *         </button>
 *       )}
 *     </MediaGallery.Previous>
 *   );
 * }
 * ```
 */
export const Previous = (props) => {
    const mediaService = useService(MediaGalleryServiceDefinition);
    return props.children({
        previous: mediaService.previousMedia,
        canGoPrevious: mediaService.hasPreviousMedia(),
    });
};
/**
 * Headless component for media gallery indicator/counter
 *
 * @component
 * @example
 * ```tsx
 * import { MediaGallery } from '@wix/media/components';
 *
 * function MediaCounter() {
 *   return (
 *     <MediaGallery.Indicator>
 *       {({ current, total }) => (
 *         <div className="media-indicator">
 *           {current} of {total}
 *         </div>
 *       )}
 *     </MediaGallery.Indicator>
 *   );
 * }
 * ```
 */
export const Indicator = (props) => {
    const mediaService = useService(MediaGalleryServiceDefinition);
    const currentIndex = mediaService.selectedMediaIndex.get();
    const totalMedia = mediaService.totalMedia.get();
    if (totalMedia <= 1) {
        return null;
    }
    return props.children({
        current: currentIndex + 1,
        total: totalMedia,
    });
};
