// @wix/auto_sdk_restaurants_service-fees/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  ConditionTypeOriginal: () => ConditionType,
  ConditionsTypeOriginal: () => ConditionsType,
  ExpectedFieldTypeOriginal: () => ExpectedFieldType,
  OperationOriginal: () => Operation,
  OperatorOriginal: () => Operator,
  PlatformValueOriginal: () => PlatformValue,
  RoundingStrategyOriginal: () => RoundingStrategy,
  SortOrderOriginal: () => SortOrder,
  TypeOriginal: () => Type,
  ValueOriginal: () => Value,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  bulkCreateRules: () => bulkCreateRules2,
  bulkDeleteRules: () => bulkDeleteRules2,
  bulkUpdateRuleTags: () => bulkUpdateRuleTags2,
  bulkUpdateRuleTagsByFilter: () => bulkUpdateRuleTagsByFilter2,
  bulkUpdateRules: () => bulkUpdateRules2,
  calculateServiceFees: () => calculateServiceFees2,
  createRule: () => createRule2,
  deleteRule: () => deleteRule2,
  getRule: () => getRule2,
  listRules: () => listRules2,
  queryRules: () => queryRules2,
  updateRule: () => updateRule2
});
module.exports = __toCommonJS(meta_exports);

// src/service-fees-v1-rule-service-fees.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressServiceFeesV1ServiceFeesRulesUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "users._base_domain_": [
      {
        srcPath: "/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      }
    ],
    _base_domain_: [
      {
        srcPath: "/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/_api/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/_api/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/_api/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/_api/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/_api/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/service-fees-rules",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
function resolveComWixpressServiceFeesV1ServiceFeesCalculateUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "users._base_domain_": [
      {
        srcPath: "/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      }
    ],
    _base_domain_: [
      {
        srcPath: "/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/_api/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/_api/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/_api/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/_api/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants/v1/service-fees/rules",
        destPath: "/v1/service-fees/rules"
      },
      {
        srcPath: "/_api/restaurants/v1/service-fees/calculate",
        destPath: "/v1/service-fees/calculate"
      },
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/service-fees/v1",
        destPath: "/v1"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/service-fees-rules",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_service-fees";
function calculateServiceFees(payload) {
  function __calculateServiceFees({ host }) {
    const metadata = {
      entityFqdn: "wix.service.fees.v1.rule",
      method: "POST",
      methodFqn: "com.wixpress.service.fees.v1.ServiceFeesCalculate.CalculateServiceFees",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressServiceFeesV1ServiceFeesCalculateUrl({
        protoPath: "/v1/calculate",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __calculateServiceFees;
}
function createRule(payload) {
  function __createRule({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "rule.createdDate" }, { path: "rule.updatedDate" }]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "rule.condition.number.value" },
          { path: "rule.conditionTree.leftCondition.number.value" },
          { path: "rule.conditionTree.rightCondition.number.value" },
          { path: "rule.conditionOptions.number.value" },
          { path: "rule.conditionTreeOptions.leftCondition.number.value" },
          { path: "rule.conditionTreeOptions.rightCondition.number.value" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.service.fees.v1.rule",
      method: "POST",
      methodFqn: "com.wixpress.service.fees.v1.ServiceFeesRules.CreateRule",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressServiceFeesV1ServiceFeesRulesUrl({
        protoPath: "/v1/rules",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "rule.createdDate" }, { path: "rule.updatedDate" }]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "rule.condition.number.value" },
            { path: "rule.conditionTree.leftCondition.number.value" },
            { path: "rule.conditionTree.rightCondition.number.value" },
            { path: "rule.conditionOptions.number.value" },
            { path: "rule.conditionTreeOptions.leftCondition.number.value" },
            { path: "rule.conditionTreeOptions.rightCondition.number.value" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createRule;
}
function getRule(payload) {
  function __getRule({ host }) {
    const metadata = {
      entityFqdn: "wix.service.fees.v1.rule",
      method: "GET",
      methodFqn: "com.wixpress.service.fees.v1.ServiceFeesRules.GetRule",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressServiceFeesV1ServiceFeesRulesUrl({
        protoPath: "/v1/rules/{ruleId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "rule.createdDate" }, { path: "rule.updatedDate" }]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "rule.condition.number.value" },
            { path: "rule.conditionTree.leftCondition.number.value" },
            { path: "rule.conditionTree.rightCondition.number.value" },
            { path: "rule.conditionOptions.number.value" },
            { path: "rule.conditionTreeOptions.leftCondition.number.value" },
            { path: "rule.conditionTreeOptions.rightCondition.number.value" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getRule;
}
function updateRule(payload) {
  function __updateRule({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "rule.createdDate" }, { path: "rule.updatedDate" }]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "rule.condition.number.value" },
          { path: "rule.conditionTree.leftCondition.number.value" },
          { path: "rule.conditionTree.rightCondition.number.value" },
          { path: "rule.conditionOptions.number.value" },
          { path: "rule.conditionTreeOptions.leftCondition.number.value" },
          { path: "rule.conditionTreeOptions.rightCondition.number.value" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.service.fees.v1.rule",
      method: "PATCH",
      methodFqn: "com.wixpress.service.fees.v1.ServiceFeesRules.UpdateRule",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressServiceFeesV1ServiceFeesRulesUrl({
        protoPath: "/v1/rules/{rule.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "rule.createdDate" }, { path: "rule.updatedDate" }]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "rule.condition.number.value" },
            { path: "rule.conditionTree.leftCondition.number.value" },
            { path: "rule.conditionTree.rightCondition.number.value" },
            { path: "rule.conditionOptions.number.value" },
            { path: "rule.conditionTreeOptions.leftCondition.number.value" },
            { path: "rule.conditionTreeOptions.rightCondition.number.value" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateRule;
}
function deleteRule(payload) {
  function __deleteRule({ host }) {
    const metadata = {
      entityFqdn: "wix.service.fees.v1.rule",
      method: "DELETE",
      methodFqn: "com.wixpress.service.fees.v1.ServiceFeesRules.DeleteRule",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressServiceFeesV1ServiceFeesRulesUrl({
        protoPath: "/v1/rules/{ruleId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteRule;
}
function listRules(payload) {
  function __listRules({ host }) {
    const metadata = {
      entityFqdn: "wix.service.fees.v1.rule",
      method: "GET",
      methodFqn: "com.wixpress.service.fees.v1.ServiceFeesRules.ListRules",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressServiceFeesV1ServiceFeesRulesUrl({
        protoPath: "/v1/rules",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "rules.createdDate" },
            { path: "rules.updatedDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "rules.condition.number.value" },
            { path: "rules.conditionTree.leftCondition.number.value" },
            { path: "rules.conditionTree.rightCondition.number.value" },
            { path: "rules.conditionOptions.number.value" },
            { path: "rules.conditionTreeOptions.leftCondition.number.value" },
            {
              path: "rules.conditionTreeOptions.rightCondition.number.value"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listRules;
}
function queryRules(payload) {
  function __queryRules({ host }) {
    const metadata = {
      entityFqdn: "wix.service.fees.v1.rule",
      method: "POST",
      methodFqn: "com.wixpress.service.fees.v1.ServiceFeesRules.QueryRules",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressServiceFeesV1ServiceFeesRulesUrl({
        protoPath: "/v1/rules/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "rules.createdDate" },
            { path: "rules.updatedDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "rules.condition.number.value" },
            { path: "rules.conditionTree.leftCondition.number.value" },
            { path: "rules.conditionTree.rightCondition.number.value" },
            { path: "rules.conditionOptions.number.value" },
            { path: "rules.conditionTreeOptions.leftCondition.number.value" },
            {
              path: "rules.conditionTreeOptions.rightCondition.number.value"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryRules;
}
function bulkCreateRules(payload) {
  function __bulkCreateRules({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "rules.createdDate" }, { path: "rules.updatedDate" }]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "rules.condition.number.value" },
          { path: "rules.conditionTree.leftCondition.number.value" },
          { path: "rules.conditionTree.rightCondition.number.value" },
          { path: "rules.conditionOptions.number.value" },
          { path: "rules.conditionTreeOptions.leftCondition.number.value" },
          { path: "rules.conditionTreeOptions.rightCondition.number.value" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.service.fees.v1.rule",
      method: "POST",
      methodFqn: "com.wixpress.service.fees.v1.ServiceFeesRules.BulkCreateRules",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressServiceFeesV1ServiceFeesRulesUrl({
        protoPath: "/v1/bulk/rules/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.rule.createdDate" },
            { path: "results.rule.updatedDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.rule.condition.number.value" },
            { path: "results.rule.conditionTree.leftCondition.number.value" },
            {
              path: "results.rule.conditionTree.rightCondition.number.value"
            },
            { path: "results.rule.conditionOptions.number.value" },
            {
              path: "results.rule.conditionTreeOptions.leftCondition.number.value"
            },
            {
              path: "results.rule.conditionTreeOptions.rightCondition.number.value"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateRules;
}
function bulkUpdateRules(payload) {
  function __bulkUpdateRules({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "rules.mask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "rules.rule.createdDate" },
          { path: "rules.rule.updatedDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "rules.rule.condition.number.value" },
          { path: "rules.rule.conditionTree.leftCondition.number.value" },
          { path: "rules.rule.conditionTree.rightCondition.number.value" },
          { path: "rules.rule.conditionOptions.number.value" },
          {
            path: "rules.rule.conditionTreeOptions.leftCondition.number.value"
          },
          {
            path: "rules.rule.conditionTreeOptions.rightCondition.number.value"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.service.fees.v1.rule",
      method: "PATCH",
      methodFqn: "com.wixpress.service.fees.v1.ServiceFeesRules.BulkUpdateRules",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressServiceFeesV1ServiceFeesRulesUrl({
        protoPath: "/v1/bulk/rules/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.rule.createdDate" },
            { path: "results.rule.updatedDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.rule.condition.number.value" },
            { path: "results.rule.conditionTree.leftCondition.number.value" },
            {
              path: "results.rule.conditionTree.rightCondition.number.value"
            },
            { path: "results.rule.conditionOptions.number.value" },
            {
              path: "results.rule.conditionTreeOptions.leftCondition.number.value"
            },
            {
              path: "results.rule.conditionTreeOptions.rightCondition.number.value"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateRules;
}
function bulkDeleteRules(payload) {
  function __bulkDeleteRules({ host }) {
    const metadata = {
      entityFqdn: "wix.service.fees.v1.rule",
      method: "DELETE",
      methodFqn: "com.wixpress.service.fees.v1.ServiceFeesRules.BulkDeleteRules",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressServiceFeesV1ServiceFeesRulesUrl({
        protoPath: "/v1/bulk/rules/delete",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.rule.createdDate" },
            { path: "results.rule.updatedDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.rule.condition.number.value" },
            { path: "results.rule.conditionTree.leftCondition.number.value" },
            {
              path: "results.rule.conditionTree.rightCondition.number.value"
            },
            { path: "results.rule.conditionOptions.number.value" },
            {
              path: "results.rule.conditionTreeOptions.leftCondition.number.value"
            },
            {
              path: "results.rule.conditionTreeOptions.rightCondition.number.value"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkDeleteRules;
}
function bulkUpdateRuleTags(payload) {
  function __bulkUpdateRuleTags({ host }) {
    const metadata = {
      entityFqdn: "wix.service.fees.v1.rule",
      method: "POST",
      methodFqn: "com.wixpress.service.fees.v1.ServiceFeesRules.BulkUpdateRuleTags",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressServiceFeesV1ServiceFeesRulesUrl({
        protoPath: "/v1/bulk/rules/update-tags",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateRuleTags;
}
function bulkUpdateRuleTagsByFilter(payload) {
  function __bulkUpdateRuleTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.service.fees.v1.rule",
      method: "POST",
      methodFqn: "com.wixpress.service.fees.v1.ServiceFeesRules.BulkUpdateRuleTagsByFilter",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressServiceFeesV1ServiceFeesRulesUrl({
        protoPath: "/v1/bulk/rules/update-tags-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateRuleTagsByFilter;
}

// src/service-fees-v1-rule-service-fees.types.ts
var Value = /* @__PURE__ */ ((Value2) => {
  Value2["NUMBER"] = "NUMBER";
  Value2["LIST"] = "LIST";
  Value2["STRING"] = "STRING";
  return Value2;
})(Value || {});
var ExpectedFieldType = /* @__PURE__ */ ((ExpectedFieldType2) => {
  ExpectedFieldType2["UNKNOWN_EXPECTED_FIELD_TYPE"] = "UNKNOWN_EXPECTED_FIELD_TYPE";
  ExpectedFieldType2["NUMBER"] = "NUMBER";
  ExpectedFieldType2["LIST"] = "LIST";
  ExpectedFieldType2["STRING"] = "STRING";
  return ExpectedFieldType2;
})(ExpectedFieldType || {});
var Operation = /* @__PURE__ */ ((Operation2) => {
  Operation2["EQ"] = "EQ";
  Operation2["LT"] = "LT";
  Operation2["LE"] = "LE";
  Operation2["GT"] = "GT";
  Operation2["GE"] = "GE";
  return Operation2;
})(Operation || {});
var Operator = /* @__PURE__ */ ((Operator2) => {
  Operator2["AND"] = "AND";
  Operator2["OR"] = "OR";
  return Operator2;
})(Operator || {});
var ConditionsType = /* @__PURE__ */ ((ConditionsType2) => {
  ConditionsType2["NO_CONDITIONS"] = "NO_CONDITIONS";
  ConditionsType2["CONDITION"] = "CONDITION";
  ConditionsType2["CONDITION_TREE"] = "CONDITION_TREE";
  return ConditionsType2;
})(ConditionsType || {});
var ConditionType = /* @__PURE__ */ ((ConditionType2) => {
  ConditionType2["UNDEFINED_CONDITION_TYPE"] = "UNDEFINED_CONDITION_TYPE";
  ConditionType2["CONDITION"] = "CONDITION";
  ConditionType2["CONDITION_TREE"] = "CONDITION_TREE";
  return ConditionType2;
})(ConditionType || {});
var RoundingStrategy = /* @__PURE__ */ ((RoundingStrategy2) => {
  RoundingStrategy2["UNKNOWN_ROUNDING_STRATEGY"] = "UNKNOWN_ROUNDING_STRATEGY";
  RoundingStrategy2["HALF_UP"] = "HALF_UP";
  RoundingStrategy2["HALF_EVEN"] = "HALF_EVEN";
  return RoundingStrategy2;
})(RoundingStrategy || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNSPECIFIED_FULFILLMENT_TYPE"] = "UNSPECIFIED_FULFILLMENT_TYPE";
  Type2["PICKUP"] = "PICKUP";
  Type2["DELIVERY"] = "DELIVERY";
  Type2["DINE_IN"] = "DINE_IN";
  Type2["CURBSIDE_PICKUP"] = "CURBSIDE_PICKUP";
  return Type2;
})(Type || {});
var PlatformValue = /* @__PURE__ */ ((PlatformValue2) => {
  PlatformValue2["SITE"] = "SITE";
  PlatformValue2["MOBILE_SITE"] = "MOBILE_SITE";
  PlatformValue2["MOBILE_APP"] = "MOBILE_APP";
  return PlatformValue2;
})(PlatformValue || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/service-fees-v1-rule-service-fees.meta.ts
function calculateServiceFees2() {
  const payload = {};
  const getRequestOptions = calculateServiceFees(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/calculate",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createRule2() {
  const payload = {};
  const getRequestOptions = createRule(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/rules",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getRule2() {
  const payload = { ruleId: ":ruleId" };
  const getRequestOptions = getRule(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/rules/{ruleId}",
    pathParams: { ruleId: "ruleId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateRule2() {
  const payload = { rule: { id: ":ruleId" } };
  const getRequestOptions = updateRule(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/rules/{rule.id}",
    pathParams: { ruleId: "ruleId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteRule2() {
  const payload = { ruleId: ":ruleId" };
  const getRequestOptions = deleteRule(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/rules/{ruleId}",
    pathParams: { ruleId: "ruleId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listRules2() {
  const payload = {};
  const getRequestOptions = listRules(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/rules",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryRules2() {
  const payload = {};
  const getRequestOptions = queryRules(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/rules/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkCreateRules2() {
  const payload = {};
  const getRequestOptions = bulkCreateRules(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/rules/create",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateRules2() {
  const payload = {};
  const getRequestOptions = bulkUpdateRules(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/bulk/rules/update",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkDeleteRules2() {
  const payload = {};
  const getRequestOptions = bulkDeleteRules(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/bulk/rules/delete",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateRuleTags2() {
  const payload = {};
  const getRequestOptions = bulkUpdateRuleTags(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/rules/update-tags",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateRuleTagsByFilter2() {
  const payload = {};
  const getRequestOptions = bulkUpdateRuleTagsByFilter(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/rules/update-tags-by-filter",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ConditionTypeOriginal,
  ConditionsTypeOriginal,
  ExpectedFieldTypeOriginal,
  OperationOriginal,
  OperatorOriginal,
  PlatformValueOriginal,
  RoundingStrategyOriginal,
  SortOrderOriginal,
  TypeOriginal,
  ValueOriginal,
  WebhookIdentityTypeOriginal,
  bulkCreateRules,
  bulkDeleteRules,
  bulkUpdateRuleTags,
  bulkUpdateRuleTagsByFilter,
  bulkUpdateRules,
  calculateServiceFees,
  createRule,
  deleteRule,
  getRule,
  listRules,
  queryRules,
  updateRule
});
