// @wix/auto-patterns/dist/cjs/components/AutoPatternsEntityPage/AutoPatternsEntityPage.js
"use strict";

exports.__esModule = true;
exports.AutoPatternsEntityPage = void 0;
var _react = _interopRequireWildcard(require("react"));
var _patterns = require("@wix/patterns");
var _form = require("@wix/patterns/form");
var _designSystem = require("@wix/design-system");
var _reactRouterDom = require("react-router-dom");
var _SchemaContext = require("../../providers/SchemaContext");
var _FormFieldInput = require("./Fields/FormFieldInput");
var _hooks = require("../../hooks");
var _providers = require("../../providers");
var _useEntityPageMoreActions = require("../../hooks/useEntityPageMoreActions");
var _useActionsSDK = require("../../hooks/useActionsSDK");
var _jsxFileName = "/home/builduser/work/73d19ce3378ce4dd/packages/auto-patterns/dist/cjs/components/AutoPatternsEntityPage/AutoPatternsEntityPage.tsx";
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
const RenderLayoutContent = ({
  content,
  level,
  sectionId,
  cardIndex,
  contentIndex,
  setInputRef
}) => {
  var _content$container$ch, _overrides$components;
  const schema = (0, _SchemaContext.useSchema)();
  const overrides = (0, _providers.usePatternsWizardOverridesContext)();
  const pageState = (0, _patterns.useEntityPageContext)();
  switch (content.type) {
    case 'field':
      if (!content.field.fieldId) {
        return null;
      }
      const field = schema.fields[content.field.fieldId];
      if (!field) {
        return null;
      }
      return /*#__PURE__*/_react.default.createElement(_designSystem.Cell, {
        span: content.field.span,
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 47,
          columnNumber: 9
        }
      }, /*#__PURE__*/_react.default.createElement(_FormFieldInput.FormFieldInput, {
        key: content.field.fieldId,
        field: field,
        inputRef: ref => setInputRef(content.field.fieldId, ref),
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 48,
          columnNumber: 11
        }
      }));
    case 'container':
      if (!((_content$container$ch = content.container.children) != null && _content$container$ch.length)) {
        return null;
      }
      return /*#__PURE__*/_react.default.createElement(_designSystem.Cell, {
        span: content.container.span,
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 62,
          columnNumber: 9
        }
      }, /*#__PURE__*/_react.default.createElement(_designSystem.Layout, {
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 63,
          columnNumber: 11
        }
      }, content.container.children.map((child, index) => /*#__PURE__*/_react.default.createElement(RenderLayoutContent, {
        key: `${sectionId}-card-${cardIndex}-content-${contentIndex}-level-${level}-child-${index}`,
        content: child,
        level: level + 1,
        sectionId: sectionId,
        cardIndex: cardIndex,
        contentIndex: index,
        setInputRef: setInputRef,
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 65,
          columnNumber: 15
        }
      }))));
    case 'component':
      if (!content.component.componentId) {
        return null;
      }
      const Component = overrides == null || (_overrides$components = overrides.components) == null ? void 0 : _overrides$components[content.component.componentId];
      if (!Component) {
        return null;
      }
      return /*#__PURE__*/_react.default.createElement(_designSystem.Cell, {
        span: content.component.span,
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 91,
          columnNumber: 9
        }
      }, /*#__PURE__*/_react.default.createElement(Component, {
        form: pageState.form,
        entity: pageState.entity,
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 92,
          columnNumber: 11
        }
      }));
    default:
      return null;
  }
};
const RenderLayoutCard = ({
  layout,
  sectionId,
  cardIndex,
  setInputRef
}) => {
  var _layout$card$subtitle;
  switch (layout.type) {
    case 'card':
      return /*#__PURE__*/_react.default.createElement(_patterns.EntityPage.Card, {
        minHeight: "50px",
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 113,
          columnNumber: 9
        }
      }, /*#__PURE__*/_react.default.createElement(_designSystem.Card.Header, {
        title: layout.card.title.text,
        subtitle: (_layout$card$subtitle = layout.card.subtitle) == null ? void 0 : _layout$card$subtitle.text,
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 114,
          columnNumber: 11
        }
      }), /*#__PURE__*/_react.default.createElement(_designSystem.Card.Divider, {
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 118,
          columnNumber: 11
        }
      }), /*#__PURE__*/_react.default.createElement(_designSystem.Card.Content, {
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 119,
          columnNumber: 11
        }
      }, /*#__PURE__*/_react.default.createElement(_designSystem.Layout, {
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 120,
          columnNumber: 13
        }
      }, layout.card.children.map((child, index) => /*#__PURE__*/_react.default.createElement(RenderLayoutContent, {
        key: `${sectionId}-card-${cardIndex}-content-${index}`,
        content: child,
        level: 0,
        sectionId: sectionId,
        cardIndex: cardIndex,
        contentIndex: index,
        setInputRef: setInputRef,
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 122,
          columnNumber: 17
        }
      })))));
    default:
      return null;
  }
};
const AutoPatternsEntityPage = ({
  configuration
}) => {
  var _pageLayout$main, _pageLayout$sidebar;
  const {
    layout: pageLayout,
    parentPageId,
    route,
    moreActions,
    collectionId
  } = configuration;
  const form = (0, _form.useForm)({
    mode: 'onChange'
  });
  const params = (0, _reactRouterDom.useParams)();
  const entityId = params[route.params.id];
  const schema = (0, _SchemaContext.useSchema)();
  const isCreateMode = !entityId;
  const inputRefs = (0, _react.useRef)({});
  const setInputRef = (id, input) => {
    inputRefs.current[id] = input;
  };
  const validate = () => {
    const invalidFieldId = Object.keys(inputRefs.current).find(fieldId => {
      var _inputRefs$current$fi;
      return (_inputRefs$current$fi = inputRefs.current[fieldId]) == null ? void 0 : _inputRefs$current$fi.invalid;
    });
    if (invalidFieldId) {
      var _inputRefs$current$in;
      (_inputRefs$current$in = inputRefs.current[invalidFieldId]) == null || _inputRefs$current$in.focus == null || _inputRefs$current$in.focus();
      throw new Error('Invalid form');
    }
  };
  const {
    getParentPagePath
  } = (0, _hooks.useNavigation)();
  const parentPath = (0, _react.useMemo)(() => getParentPagePath(parentPageId), [parentPageId, getParentPagePath]);
  const state = (0, _patterns.useEntityPage)({
    parentPath,
    form,
    onSave: async () => {
      validate();
      const formValues = form.getValues();
      const baseEntity = state.entity || {};
      const updatedEntity = {
        ...baseEntity,
        ...formValues
      };
      const newEntity = entityId ? await schema.actions.update(updatedEntity) : await schema.actions.create(updatedEntity);
      return {
        updatedEntity: newEntity
      };
    },
    fetch: async () => {
      if (!entityId) {
        return {
          entity: undefined
        };
      }
      const entity = await schema.actions.get(entityId);
      return {
        entity
      };
    }
  });
  const entity = (0, _patterns.useEntity)(state);
  const {
    title,
    subtitle
  } = (0, _hooks.useEntityPageHeaderTexts)({
    config: configuration,
    isCreateMode,
    entityDisplayName: entity == null ? void 0 : entity[schema.displayField]
  });
  const sdk = (0, _useActionsSDK.useActionsSDK)({
    collectionId
  });
  const resolvedMoreActions = (0, _useEntityPageMoreActions.useEntityPageMoreActions)(moreActions, entity, sdk, form);
  return /*#__PURE__*/_react.default.createElement(_patterns.EntityPage, {
    state: state,
    dataHook: "auto-patterns-entity-page",
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 235,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(_patterns.EntityPage.Header, {
    title: title,
    subtitle: subtitle,
    moreActions: resolvedMoreActions.length > 0 ? /*#__PURE__*/_react.default.createElement(_patterns.MoreActions, {
      items: resolvedMoreActions,
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 241,
        columnNumber: 13
      }
    }) : undefined,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 236,
      columnNumber: 7
    }
  }), /*#__PURE__*/_react.default.createElement(_patterns.EntityPage.Content, {
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 245,
      columnNumber: 7
    }
  }, /*#__PURE__*/_react.default.createElement(_patterns.EntityPage.MainContent, {
    dataHook: "entity-page-main-content",
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 246,
      columnNumber: 9
    }
  }, pageLayout == null || (_pageLayout$main = pageLayout.main) == null ? void 0 : _pageLayout$main.map((layout, layoutIndex) => /*#__PURE__*/_react.default.createElement(RenderLayoutCard, {
    key: `main-section-card-${layoutIndex}`,
    layout: layout,
    sectionId: "main",
    cardIndex: layoutIndex,
    setInputRef: setInputRef,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 248,
      columnNumber: 13
    }
  }))), pageLayout != null && pageLayout.sidebar ? /*#__PURE__*/_react.default.createElement(_patterns.EntityPage.AdditionalContent, {
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 258,
      columnNumber: 11
    }
  }, (_pageLayout$sidebar = pageLayout.sidebar) == null ? void 0 : _pageLayout$sidebar.map((layout, layoutIndex) => /*#__PURE__*/_react.default.createElement(RenderLayoutCard, {
    key: `sidebar-section-card-${layoutIndex}`,
    layout: layout,
    sectionId: "sidebar",
    cardIndex: layoutIndex,
    setInputRef: setInputRef,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 260,
      columnNumber: 15
    }
  }))) : null));
};

/*
  Key structure explanation:

  Main cards: main-section-card-{layoutIndex}
  Sidebar cards: sidebar-section-card-{layoutIndex}

  Layout content:
  {sectionId}-card-{cardIndex}-content-{contentIndex}

  Nested children:
  {sectionId}-card-{cardIndex}-content-{parentContentIndex}-level-{level}-child-{childIndex}

  This ensures uniqueness across all levels and sections.
*/
exports.AutoPatternsEntityPage = AutoPatternsEntityPage;
