// @wix/auto-patterns/dist/cjs/components/AutoPatternsEntityPage/SkeletonEntity.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.SkeletonEntity = void 0;
var _patterns = require("@wix/patterns");
var _form = require("../../exports/form");
var _react = _interopRequireDefault(require("react"));
var _designSystem = require("@wix/design-system");
var _jsxFileName = "/home/builduser/work/73d19ce3378ce4dd/packages/auto-patterns/dist/cjs/components/AutoPatternsEntityPage/SkeletonEntity.tsx";
const RenderLayoutContent = ({
  content,
  level,
  sectionId,
  cardIndex,
  contentIndex
}) => {
  var _content$container$ch;
  switch (content.type) {
    case 'container':
      if (!((_content$container$ch = content.container.children) != null && _content$container$ch.length)) {
        return null;
      }
      return /*#__PURE__*/_react.default.createElement(_designSystem.Cell, {
        span: content.container.span,
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 25,
          columnNumber: 9
        }
      }, /*#__PURE__*/_react.default.createElement(_designSystem.Layout, {
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 26,
          columnNumber: 11
        }
      }, content.container.children.map((child, index) => /*#__PURE__*/_react.default.createElement(RenderLayoutContent, {
        key: `${sectionId}-card-${cardIndex}-content-${contentIndex}-level-${level}-child-${index}`,
        content: child,
        level: level + 1,
        sectionId: sectionId,
        cardIndex: cardIndex,
        contentIndex: index,
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 28,
          columnNumber: 15
        }
      }))));
    default:
      return null;
  }
};
const RenderLayoutCard = ({
  layout,
  sectionId,
  cardIndex
}) => {
  var _layout$card$subtitle;
  switch (layout.type) {
    case 'card':
      return /*#__PURE__*/_react.default.createElement(_patterns.EntityPage.Card, {
        minHeight: "50px",
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 54,
          columnNumber: 9
        }
      }, /*#__PURE__*/_react.default.createElement(_designSystem.Card.Header, {
        title: layout.card.title.text,
        subtitle: (_layout$card$subtitle = layout.card.subtitle) == null ? void 0 : _layout$card$subtitle.text,
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 55,
          columnNumber: 11
        }
      }), /*#__PURE__*/_react.default.createElement(_designSystem.Card.Divider, {
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 59,
          columnNumber: 11
        }
      }), /*#__PURE__*/_react.default.createElement(_designSystem.Card.Content, {
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 60,
          columnNumber: 11
        }
      }, /*#__PURE__*/_react.default.createElement(_designSystem.Layout, {
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 61,
          columnNumber: 13
        }
      }, layout.card.children.map((child, index) => /*#__PURE__*/_react.default.createElement(RenderLayoutContent, {
        key: `${sectionId}-card-${cardIndex}-content-${index}`,
        content: child,
        level: 0,
        sectionId: sectionId,
        cardIndex: cardIndex,
        contentIndex: index,
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 63,
          columnNumber: 17
        }
      })))));
    default:
      return null;
  }
};
const SkeletonEntity = ({
  entityPage
}) => {
  var _entityPage$layout, _entityPage$layout2, _entityPage$layout$si;
  const form = (0, _form.useForm)();
  const state = (0, _patterns.useEntityPage)({
    parentPath: '',
    form,
    onSave: async () => ({
      updatedEntity: {}
    }),
    fetch: async () => new Promise(() => {})
  });
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_patterns.EntityPage, {
    state: state,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 93,
      columnNumber: 7
    }
  }, /*#__PURE__*/_react.default.createElement(_patterns.EntityPage.Header, {
    title: {
      text: ''
    },
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 94,
      columnNumber: 9
    }
  }), /*#__PURE__*/_react.default.createElement(_patterns.EntityPage.Content, {
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 95,
      columnNumber: 9
    }
  }, /*#__PURE__*/_react.default.createElement(_patterns.EntityPage.MainContent, {
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 96,
      columnNumber: 11
    }
  }, (_entityPage$layout = entityPage.layout) == null || (_entityPage$layout = _entityPage$layout.main) == null ? void 0 : _entityPage$layout.map((layout, layoutIndex) => /*#__PURE__*/_react.default.createElement(RenderLayoutCard, {
    key: `main-section-card-${layoutIndex}`,
    layout: layout,
    sectionId: "main",
    cardIndex: layoutIndex,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 98,
      columnNumber: 15
    }
  }))), (_entityPage$layout2 = entityPage.layout) != null && _entityPage$layout2.sidebar ? /*#__PURE__*/_react.default.createElement(_patterns.EntityPage.AdditionalContent, {
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 107,
      columnNumber: 13
    }
  }, (_entityPage$layout$si = entityPage.layout.sidebar) == null ? void 0 : _entityPage$layout$si.map((layout, layoutIndex) => /*#__PURE__*/_react.default.createElement(RenderLayoutCard, {
    key: `sidebar-section-card-${layoutIndex}`,
    layout: layout,
    sectionId: "sidebar",
    cardIndex: layoutIndex,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 109,
      columnNumber: 17
    }
  }))) : null)));
};
exports.SkeletonEntity = SkeletonEntity;
