// @wix/fast-gallery-core/dist/cjs/components/item/ItemRenderer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _clsx = _interopRequireDefault(require("clsx"));
var _MediaRenderer = _interopRequireDefault(require("../media/MediaRenderer"));
var _utils = require("../utils");
var _ItemRendererModule = _interopRequireDefault(require("./ItemRenderer.module.scss"));
var _jsxFileName = "/home/builduser/work/d3ecddb38d0a3df1/packages/fast-gallery-packages/fast-gallery-core/dist/cjs/components/item/ItemRenderer.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
const ItemRenderer = ({
  data,
  selectors,
  customRenderers,
  onItemClick,
  shouldUseDirectLink,
  options
}) => {
  const anchorRef = (0, _react.useRef)(null);
  const props = {
    data,
    selectors,
    customRenderers,
    options
  };
  const {
    metaData
  } = data;
  const handleKeyUp = e => {
    switch (e.key) {
      case ' ': // space
      case 'Enter':
        // enter
        e.stopPropagation();
        onItemClick(e, false); // pressing enter or space always behaves as click on main image, even if the click is on a thumbnail
        if (shouldUseDirectLink()) {
          var _anchorRef$current;
          (_anchorRef$current = anchorRef.current) == null || _anchorRef$current.click(); // when directLink, we want to simulate the 'enter' or 'space' press on an <a> element
        }
        return false;
      default:
        return true;
    }
  };
  const innerDiv = /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _clsx.default)(_ItemRendererModule.default.info, selectors == null ? void 0 : selectors.info),
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 36,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _clsx.default)(_ItemRendererModule.default.title, selectors == null ? void 0 : selectors.title),
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 37,
      columnNumber: 7
    }
  }, metaData.title), /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _clsx.default)(_ItemRendererModule.default.description, selectors == null ? void 0 : selectors.description),
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 40,
      columnNumber: 7
    }
  }, metaData.description));
  const linkParams = (0, _utils.getLinkParams)({
    link: metaData.link,
    itemClickAction: options == null ? void 0 : options.itemClickAction
  });
  const currentIndex = Number(props.data.index);
  const counterDiv = /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _clsx.default)(_ItemRendererModule.default.counter, selectors == null ? void 0 : selectors.counter),
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 52,
      columnNumber: 5
    }
  }, currentIndex < 9 ? `0${currentIndex + 1}` : currentIndex + 1);
  return /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _clsx.default)(_ItemRendererModule.default.content, selectors == null ? void 0 : selectors.content),
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 58,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _clsx.default)(_ItemRendererModule.default.media, selectors == null ? void 0 : selectors.media),
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 59,
      columnNumber: 7
    }
  }, (0, _utils.renderCustomOrDefaultComp)(customRenderers == null ? void 0 : customRenderers.mediaRenderer, _MediaRenderer.default, props)), linkParams != null && linkParams.href ? /*#__PURE__*/_react.default.createElement("a", {
    ref: anchorRef,
    className: (0, _clsx.default)(_ItemRendererModule.default.details, selectors == null ? void 0 : selectors.details),
    rel: "playground noreferrer",
    onKeyUp: handleKeyUp,
    href: linkParams.href,
    target: linkParams.target,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 67,
      columnNumber: 9
    }
  }, counterDiv, innerDiv) : /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("span", {
    className: (0, _clsx.default)(_ItemRendererModule.default.details, selectors == null ? void 0 : selectors.details),
    "aria-hidden": true // hide it from assistive technologies for SEO
    ,
    rel: "playground noreferrer",
    onKeyUp: handleKeyUp,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 80,
      columnNumber: 11
    }
  }, counterDiv, innerDiv)));
};
var _default = exports.default = ItemRenderer;
