// @wix/fast-gallery-core/dist/cjs/components/item/Item.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.default = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireWildcard(require("react"));
var _clsx = _interopRequireDefault(require("clsx"));
var _navigator = require("../../navigator");
var _ItemRenderer = _interopRequireDefault(require("./ItemRenderer"));
var _lib = require("../../lib");
var _utils = require("../utils");
var _generalConstants = require("../generalConstants");
var _ItemModule = _interopRequireDefault(require("./Item.module.scss"));
var _jsxFileName = "/home/builduser/work/d3ecddb38d0a3df1/packages/fast-gallery-packages/fast-gallery-core/dist/cjs/components/item/Item.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
const Item = props => {
  var _props$options;
  const {
    index,
    data,
    selectors,
    customRenderers,
    eventsListener,
    createResizedMediaUrl,
    onMeasure,
    activeIndex,
    isCurrentlyHovered,
    options,
    isAccessible,
    getItemCustomAttributes,
    registerItemRef
  } = props;
  const itemRef = (0, _react.useRef)(null);
  const [isActive, setIsActive] = (0, _react.useState)(false);
  const [isInFocus, setIsInFocus] = (0, _react.useState)(false);
  const itemClickAction = props == null || (_props$options = props.options) == null ? void 0 : _props$options.itemClickAction;
  (0, _react.useEffect)(() => {
    setIsActive(index === activeIndex);
  }, [activeIndex, index]);
  (0, _react.useEffect)(() => {
    if (registerItemRef) {
      registerItemRef(index, itemRef.current);
    }
  }, [registerItemRef, index]);
  (0, _navigator.useMeasure)({
    callback: measurements => onMeasure(index, measurements, itemRef),
    index,
    ref: itemRef
  });
  const handleRightClickContextMenu = event => {
    if (options != null && options.disableContextMenu) {
      event.preventDefault();
    }
  };
  const applyItemHover = () => {
    var _itemRef$current;
    eventsListener(_lib.GALLERY_EVENTS.HOVER_SET, {
      index
    });
    (_itemRef$current = itemRef.current) == null || _itemRef$current.classList.add(_generalConstants.hoverClasses.itemHover);
  };
  const clearItemHover = () => {
    var _itemRef$current2;
    (_itemRef$current2 = itemRef.current) == null || _itemRef$current2.classList.remove(_generalConstants.hoverClasses.itemHover);
    eventsListener(_lib.GALLERY_EVENTS.HOVER_SET, {
      index: -1
    });
  };
  const onMouseEnter = () => {
    // if (!utils.isMobile()) { //TODO: add this when isMobile is available
    // }
    applyItemHover();
  };
  const onMouseLeave = () => {
    // if (!utils.isMobile()) { //TODO: add this when isMobile is available
    // }
    clearItemHover();
  };
  const onFocus = () => {
    eventsListener(_lib.GALLERY_EVENTS.ITEM_FOCUSED, {
      index
    });
    applyItemHover();
    setIsInFocus(true);
  };
  const onBlur = e => {
    if (isAccessible && isCurrentlyHovered) {
      eventsListener(_lib.GALLERY_EVENTS.HOVER_SET, {
        index: -1
      });
      clearItemHover();
    } else {
      eventsListener(_lib.GALLERY_EVENTS.ITEM_LOST_FOCUS, {
        index
      });
      e.currentTarget.blur();
      setIsInFocus(false);
    }
  };
  const onKeyDown = e => {
    if (e.key === 'Escape') {
      onBlur(e);
    }
  };
  const itemHasLink = () => {
    const {
      link
    } = data.metaData;
    const itemDoesntHaveLink = !link || !link.url || link.url === ''; // when itemClickAction is 'LINK' but no link was added to this specific item
    return !itemDoesntHaveLink;
  };
  const isItemClickable = () => {
    var _props$options2, _props$options3;
    const itemDoesntHaveLink = !itemHasLink(); // when itemClickAction is 'LINK' but no link was added to this specific item
    return (props == null || (_props$options2 = props.options) == null ? void 0 : _props$options2.itemClickAction) === _lib.CLICK_ACTION.NOTHING || (props == null || (_props$options3 = props.options) == null ? void 0 : _props$options3.itemClickAction) === _lib.CLICK_ACTION.LINK && itemDoesntHaveLink ? false : true;
  };
  const onItemClick = (e, shouldPreventDefault = false) => {
    eventsListener(_lib.GALLERY_EVENTS.ON_ITEM_CLICKED, {
      index
    });
    if (shouldUseDirectLink()) {
      return;
    }
    if (shouldPreventDefault) {
      e.preventDefault();
    }
    if (shouldShowHoverOnMobile()
    // || shouldShowSecondMediaOnMobile()
    ) {
      handleHoverClickOnMobile();
    } else if (shouldHoverWithoutOverlayAndClickOnMobile()) {
      eventsListener(_lib.GALLERY_EVENTS.HOVER_SET, {
        index
      });
    } else {
      handleGalleryItemAction();
    }
  };
  const handleGalleryItemAction = () => {
    if (itemClickAction === _lib.CLICK_ACTION.ACTION) {
      eventsListener(_lib.GALLERY_EVENTS.ITEM_ACTION_TRIGGERED, {
        index
      });
    }
  };
  const shouldUseDirectLink = () => {
    const {
      link
    } = data.metaData;
    const {
      url,
      target
    } = link || {};
    const useDirectLink = !!(url && target && itemClickAction === _lib.CLICK_ACTION.LINK);
    const isHoverEnabledOnMobile = shouldShowHoverOnMobile();
    const shouldUseDirectLinkOnMobile = isHoverEnabledOnMobile &&
    // isClickOnCurrentHoveredItem() &&
    useDirectLink;
    if (shouldUseDirectLinkOnMobile) {
      eventsListener(_lib.GALLERY_EVENTS.HOVER_SET, {
        index: -1
      });
      return true;
    }
    if (useDirectLink && !isHoverEnabledOnMobile) {
      return true;
    }
    return false;
  };

  // const isClickOnCurrentHoveredItem = () =>
  //   isCurrentlyHovered
  // || // this single item was already hovered.
  // props.options?.hoveringBehavior ===
  //   OVERLAY_BEHAVIOR_ON_HOVER.ALWAYS_SHOW;

  const handleHoverClickOnMobile = () => {
    if (isCurrentlyHovered) {
      handleGalleryItemAction();
      eventsListener(_lib.GALLERY_EVENTS.HOVER_SET, {
        index: -1
      });
    } else {
      eventsListener(_lib.GALLERY_EVENTS.HOVER_SET, {
        index
      });
    }
  };
  const shouldShowHoverOnMobile = () => {
    if (_lib.GalleryUtils.isMobile()) {
      // const { hoveringBehavior, alwaysShowHover, previewHover } =
      // props.options?.behaviors?.item?.overlay || {};

      // if (hoveringBehavior === OVERLAY_BEHAVIOR_ON_HOVER.NEVER_SHOW) {
      //   return false;
      // }
      if (itemClickAction === _lib.CLICK_ACTION.NOTHING && data.type !== _lib.MediaTypes.VIDEO) {
        return true;
      }
      // if (
      // this.props.customComponents.customHoverRenderer &&
      // GALLERY_CONSTS.hasHoverPlacement(
      //   this.props.options[optionsMap.layoutParams.info.placement]
      // ) &&
      // hoveringBehaviour !== OVERLAY_BEHAVIOR_ON_HOVER.NEVER_SHOW
      // ) {
      // return true;
      // }
      // TODO Notice this important case
      // if (isEditMode() && previewHover) {
      //   return true;
      // }
      if ((0, _lib.isEditMode)()) {
        return true;
      }
    }
    return false;
  };

  // const shouldShowSecondMediaOnMobile = () => {
  //   if (utils.isMobile()) {
  //     if (
  //       itemClickAction === CLICK_ACTION.NOTHING &&
  //       data.type !== 'video' &&
  //       data.type !== '3d'
  //     ) {
  //       return (
  //         props.options?.secondaryMedia?.trigger ===
  //         SECONDARY_MEDIA_TRIGGER.HOVER
  //       );
  //     } else {
  //       return false;
  //     }
  //   }
  //   return false;
  // };

  const shouldHoverWithoutOverlayAndClickOnMobile = () => {
    return _lib.GalleryUtils.isMobile() &&
    // props.options?.behaviors?.item?.video?.playTrigger ===
    //   PLAY_TRIGGER.HOVER &&
    itemClickAction === _lib.CLICK_ACTION.NOTHING;
  };
  const getItemAriaLabel = () => {
    const {
      type
    } = data;
    const {
      alt,
      description,
      title
    } = data.metaData;
    const label = alt || title || description.substring(0, 120);
    const defaultLabel = `Gallery item ${index + 1}`;
    const mapTypeToLabel = {
      [_lib.MediaTypes.VIDEO]: label || `${defaultLabel} (video)`,
      [_lib.MediaTypes.IMAGE]: label || defaultLabel
    };
    return mapTypeToLabel[type];
  };

  // const getItemContainerTabIndex = () => {
  //   const tabIndex = this.isHighlight()
  //     ? utils.getTabIndex('currentThumbnail')
  //     : this.props.activeIndex === this.props.idx
  //     ? utils.getTabIndex('currentGalleryItem')
  //     : -1;
  //   return tabIndex;
  // }; //TODO: add this when getTabIndex is needed

  const generateResizedMediaUrl = () => {
    const resizeMediaConsts = {
      resizeMethod: 'fit',
      requiredWidth: 1000,
      requiredHeight: 1000,
      createMultiple: true
    };
    const resizeMediaParams = {
      ...resizeMediaConsts,
      item: {
        item: data,
        dto: {
          imageToken: ''
        }
      },
      originalUrl: data.mediaUrl
    };
    if (typeof createResizedMediaUrl === 'function') {
      return createResizedMediaUrl(resizeMediaParams);
    } else {
      return data.mediaUrl;
    }
  };
  const itemRendererProps = {
    data: {
      ...data,
      mediaUrl: generateResizedMediaUrl()
    },
    selectors,
    customRenderers,
    eventsListener,
    onItemClick,
    shouldUseDirectLink,
    options
  };

  // Get custom attributes for the item element
  const itemCustomAttributes = getItemCustomAttributes ? getItemCustomAttributes(index) : {};
  return /*#__PURE__*/_react.default.createElement("div", (0, _extends2.default)({
    ref: itemRef,
    className: (0, _clsx.default)(_ItemModule.default.item, selectors == null ? void 0 : selectors.item, isActive ? 'active' : '', isInFocus ? 'focus' : '', isItemClickable() ? _ItemModule.default.clickable : '', isCurrentlyHovered ? 'item-hover' : '', isAccessible ? 'accessible' : ''),
    tabIndex: index === activeIndex ? 0 : -1,
    "data-index": index,
    role: "listitem",
    onContextMenu: e => {
      handleRightClickContextMenu(e);
    },
    onClick: e => onItemClick(e, false),
    onMouseEnter: onMouseEnter,
    onMouseLeave: onMouseLeave,
    onFocus: onFocus,
    onBlur: onBlur,
    onKeyDown: onKeyDown,
    "aria-label": getItemAriaLabel()
  }, itemCustomAttributes, {
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 349,
      columnNumber: 5
    }
  }), (0, _utils.renderCustomOrDefaultComp)(customRenderers == null ? void 0 : customRenderers.itemRenderer, _ItemRenderer.default, itemRendererProps));
};
var _default = exports.default = Item;
