// @wix/ricos/dist/cjs/plugin-spoiler/Components/BlockSpoilerComponent.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var BlockSpoilerComponent_exports = {};
__export(BlockSpoilerComponent_exports, {
  default: () => BlockSpoilerComponent_default
});
module.exports = __toCommonJS(BlockSpoilerComponent_exports);
var import_classnames = __toESM(require("classnames"));
var import_lodash = __toESM(require("lodash"));
var import_prop_types = __toESM(require("prop-types"));
var import_react = __toESM(require("react"));
var import_data_hooks = require("./../../_shared/data-hooks.js");
var import_spoiler_scss = __toESM(require("../statics/styles/spoiler-scss.js"));
var import_types = require("../types.js");
var import_SpoilerContainer = __toESM(require("./SpoilerContainer.js"));
const Overlay = ({ hideOverlay, pluginType }) => !hideOverlay ? /* @__PURE__ */ import_react.default.createElement(
  "div",
  {
    role: "none",
    className: pluginType === "Gallery" ? import_spoiler_scss.default.overlay_gallery : import_spoiler_scss.default.overlay
  }
) : null;
Overlay.propTypes = {
  hideOverlay: import_prop_types.default.bool,
  pluginType: import_prop_types.default.string
};
class BlockSpoilerComponent extends import_react.default.Component {
  constructor(props) {
    super(props);
    this.onResizeElement = () => {
      this.forceUpdate();
    };
    this.handleClick = (e) => {
      const { onClick } = this.props;
      const { isReveal } = this.state;
      isReveal && onClick?.(e);
    };
    this.onRevealSpoiler = (e) => {
      const { helpers, pluginType } = this.props;
      e.preventDefault();
      this.setState({ isReveal: true });
      helpers.onViewerAction?.(import_types.SPOILER_TYPE, "Click", `reveal_${pluginType}`);
    };
    this.renderSpoilerContainer = () => {
      const { isReveal } = this.state;
      const {
        disabledRevealSpoilerBtn,
        setFocusToBlock,
        componentData,
        pluginType,
        setInPluginEditingMode,
        handleButtonContentChange,
        handleDescriptionChange,
        isMobile,
        t,
        helpers
      } = this.props;
      const width = this.elementRef?.offsetWidth;
      const height = this.elementRef?.offsetHeight;
      return !isReveal && /* @__PURE__ */ import_react.default.createElement(
        import_SpoilerContainer.default,
        {
          styles: import_spoiler_scss.default,
          width,
          height,
          setInPluginEditingMode,
          setFocusToBlock,
          pluginType,
          description: componentData?.config?.spoiler?.description,
          buttonContent: componentData?.config?.spoiler?.buttonContent,
          onRevealSpoiler: !disabledRevealSpoilerBtn ? this.onRevealSpoiler : void 0,
          handleButtonContentChange,
          handleDescriptionChange,
          isMobile,
          t,
          helpers
        }
      );
    };
    this.onKeyDown = (e) => {
      if (e.key === "Enter") {
        this.handleClick(e);
      }
    };
    this.setRef = (ref) => (
      // @ts-expect-error
      this.elementRef = ref
    );
    this.state = {};
  }
  componentDidMount() {
    if (window?.ResizeObserver) {
      this.resizeObserver = new ResizeObserver(import_lodash.default.debounce(this.onResizeElement, 60));
      this.resizeObserver?.observe(this.elementRef);
    }
  }
  componentWillUnmount() {
    if (window?.ResizeObserver) {
      this.resizeObserver?.unobserve(this.elementRef);
    }
  }
  render() {
    const { children, pluginType, width, isMobile } = this.props;
    const { isReveal } = this.state;
    let className = "";
    let blur = "";
    if (!isReveal) {
      const isGallery = pluginType === "Gallery";
      className = isGallery ? import_spoiler_scss.default.hideBlock_gallery : import_spoiler_scss.default.hideBlock;
      if (!isGallery) {
        const width2 = this.elementRef?.offsetWidth;
        blur = width2 < 400 ? import_spoiler_scss.default.smallBlur : width2 < 700 ? import_spoiler_scss.default.mediumBlur : import_spoiler_scss.default.largeBlur;
      }
    }
    return /* @__PURE__ */ import_react.default.createElement(
      "div",
      {
        ref: this.setRef,
        className: (0, import_classnames.default)(import_spoiler_scss.default.spoilerWrapper, this.props.className, {
          [import_spoiler_scss.default.isMobile]: isMobile
        }),
        style: { width },
        "data-hook": import_data_hooks.SPOILER_CONTAINER
      },
      this.renderSpoilerContainer(),
      /* @__PURE__ */ import_react.default.createElement(
        "div",
        {
          className: (0, import_classnames.default)(className, blur),
          onClick: this.handleClick,
          role: "button",
          tabIndex: 0,
          onKeyDown: this.onKeyDown
        },
        children,
        /* @__PURE__ */ import_react.default.createElement(Overlay, { hideOverlay: isReveal, pluginType })
      )
    );
  }
}
BlockSpoilerComponent.propTypes = {
  componentData: import_prop_types.default.object.isRequired,
  children: import_prop_types.default.object.isRequired,
  theme: import_prop_types.default.object,
  disabledRevealSpoilerBtn: import_prop_types.default.bool,
  isEditableText: import_prop_types.default.bool,
  pluginType: import_prop_types.default.string,
  onClick: import_prop_types.default.func,
  handleDescriptionChange: import_prop_types.default.func,
  handleButtonContentChange: import_prop_types.default.func,
  className: import_prop_types.default.string,
  setFocusToBlock: import_prop_types.default.func,
  setInPluginEditingMode: import_prop_types.default.func,
  width: import_prop_types.default.object,
  t: import_prop_types.default.func,
  isMobile: import_prop_types.default.bool,
  helpers: import_prop_types.default.object,
  type: import_prop_types.default.string
};
var BlockSpoilerComponent_default = BlockSpoilerComponent;
