// @wix/bex-utils/@wix/ambassador-infra-asyncjobs-v1-async-job/build/es/builders.impl.js
import { mergeDeep, someNumber, someString, someAny, someRepeated, someDate, } from '@wix/metro-common-builders';
import { Status, CompleteStatus, WebhookIdentityType, Type, StatusFilter, } from './types.impl';
// @ts-ignore
import stack from 'callsite';
export function aAsyncJob(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        id: someString({
            validations: [
                { type: 'FORMAT', value: 'GUID' },
                { type: 'READ_ONLY', value: 'true' },
            ],
        }),
        metadata: {},
        status: aStatus(),
        counters: aCounters(),
        createdDate: someDate({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        updatedDate: someDate({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        owner: aOwner(),
        createdBy: aCreatedBy(),
        counts: aCounts(),
    }, overrideObject);
}
export function aType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = Type.UNKNOWN_TYPE; }
    return enumFirstItem;
}
export function aAnonymousVisitorIdOptions(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        anonymousVisitorId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
export function aMemberIdOptions(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        memberId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
export function aWixUserIdOptions(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        wixUserId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
export function aAppIdOptions(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        appId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
export function aStatus(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = Status.UNKNOWN; }
    return enumFirstItem;
}
export function aCounters(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        itemsToProcess: someNumber(),
        itemsSucceeded: someNumber(),
        itemsFailed: someNumber(),
        errorCountByCode: { errorCountByCode: someNumber() },
    }, overrideObject);
}
export function aOwner(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        anonymousVisitorId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
export function aCreatedBy(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        anonymousVisitorIdOptions: aAnonymousVisitorIdOptions(),
        type: aType(),
    }, overrideObject);
}
export function aCounts(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        total: someNumber(),
        successCount: someNumber(),
        failCount: someNumber(),
        errorByCodeCount: { errorByCodeCount: someNumber() },
    }, overrideObject);
}
export function aAsyncJobFinishedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        jobId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
export function aAsyncJobFailedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        jobId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
export function aAsyncJobItemFailedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        jobId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        jobItem: aAsyncJobItem(),
    }, overrideObject);
}
export function aAsyncJobItem(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        id: someString({
            validations: [
                { type: 'FORMAT', value: 'GUID' },
                { type: 'READ_ONLY', value: 'true' },
            ],
        }),
        data: {},
        entityId: someString({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '32' },
                { type: 'IMMUTABLE', value: 'true' },
            ],
        }),
        success: true,
        error: aApplicationError(),
    }, overrideObject);
}
export function aApplicationError(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        code: someString(),
        description: someString(),
        data: {},
    }, overrideObject);
}
export function aCreateAsyncJobRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        job: aAsyncJob(),
    }, overrideObject);
}
export function aCreateAsyncJobResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        job: aAsyncJob(),
    }, overrideObject);
}
export function aGetAsyncJobRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        jobId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
export function aGetAsyncJobResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        job: aAsyncJob(),
    }, overrideObject);
}
export function aUpdateAsyncJobRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        job: aAsyncJob(),
        mask: someRepeated(),
    }, overrideObject);
}
export function aUpdateAsyncJobResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        job: aAsyncJob(),
    }, overrideObject);
}
export function aDeleteAsyncJobRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        jobId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
export function aDeleteAsyncJobResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({}, overrideObject);
}
export function aCompleteAsyncJobRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        jobId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        status: aCompleteStatus(),
    }, overrideObject);
}
export function aCompleteStatus(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = CompleteStatus.FINISHED; }
    return enumFirstItem;
}
export function aCompleteAsyncJobResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({}, overrideObject);
}
export function aReportAsyncJobProgressRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        jobId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        items: [
            aAsyncJobItem(),
            aAsyncJobItem(),
            aAsyncJobItem(),
            aAsyncJobItem(),
            aAsyncJobItem(),
            aAsyncJobItem(),
            aAsyncJobItem(),
            aAsyncJobItem(),
            aAsyncJobItem(),
            aAsyncJobItem(),
        ],
    }, overrideObject);
}
export function aReportAsyncJobProgressResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({}, overrideObject);
}
export function aAsyncJobItemAddedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        jobId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        jobItem: aAsyncJobItem(),
    }, overrideObject);
}
export function aListAsyncJobItemsRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        jobId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        paging: aCursorPaging(),
        hasError: true,
        statusFilter: aStatusFilter(),
    }, overrideObject);
}
export function aCursorPaging(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        limit: someNumber({
            validations: [
                { type: 'MIN' },
                { type: 'MAX', value: '100' },
            ],
        }),
        cursor: someString({
            validations: [
                { type: 'MAX_LENGTH', value: '16000' },
            ],
        }),
    }, overrideObject);
}
export function aStatusFilter(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = StatusFilter.ALL; }
    return enumFirstItem;
}
export function aListAsyncJobItemsResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        results: [aAsyncJobItem()],
        pagingMetadata: aPagingMetadataV2(),
    }, overrideObject);
}
export function aPagingMetadataV2(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        count: someNumber(),
        cursors: aCursors(),
        hasNext: true,
    }, overrideObject);
}
export function aCursors(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        next: someString({
            validations: [
                { type: 'MAX_LENGTH', value: '16000' },
            ],
        }),
        prev: someString({
            validations: [
                { type: 'MAX_LENGTH', value: '16000' },
            ],
        }),
    }, overrideObject);
}
export function aDomainEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        createdEvent: aEntityCreatedEvent(),
        id: someString(),
        entityFqdn: someString(),
        slug: someString(),
        entityId: someString(),
        eventTime: someDate(),
        triggeredByAnonymizeRequest: true,
        originatedFrom: someString(),
        entityEventSequence: someString({
            validations: [{ type: 'DECIMAL_VALUE' }],
        }),
    }, overrideObject);
}
export function aEntityCreatedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        entityAsJson: someString(),
        triggeredByUndelete: true,
        restoreInfo: aRestoreInfo(),
        additionalMetadataAsJson: someString(),
    }, overrideObject);
}
export function aRestoreInfo(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        deletedDate: someDate(),
    }, overrideObject);
}
export function aEntityUpdatedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        currentEntityAsJson: someString(),
        previousEntityAsJson: someString(),
        modifiedFields: { modifiedFields: someAny() },
        additionalMetadataAsJson: someString(),
    }, overrideObject);
}
export function aEntityDeletedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        movedToTrash: true,
        deletedEntityAsJson: someString(),
        additionalMetadataAsJson: someString(),
    }, overrideObject);
}
export function aActionEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        bodyAsJson: someString(),
    }, overrideObject);
}
export function aMessageEnvelope(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        instanceId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        eventType: someString({
            validations: [{ type: 'MAX_LENGTH', value: '150' }],
        }),
        identity: aIdentificationData(),
        data: someString(),
    }, overrideObject);
}
export function aIdentificationData(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        anonymousVisitorId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        identityType: aWebhookIdentityType(),
    }, overrideObject);
}
export function aWebhookIdentityType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = WebhookIdentityType.UNKNOWN; }
    return enumFirstItem;
}
function isCurrentBuilderWasPreviouslyCalled() {
    var callChainFunctionNames = stack().map(function (item) {
        return item.getFunctionName();
    });
    var callChainWithoutCurrentBuilder = callChainFunctionNames.splice(2);
    var currentBuilderName = callChainFunctionNames[1];
    return callChainWithoutCurrentBuilder.some(function (functionName) { return functionName === currentBuilderName; });
}
export * from '@wix/metro-common-builders';
