// @wix/bex-utils/@wix/ambassador-infra-asyncjobs-v1-async-job/build/es/meta.impl.js
import * as httpWixInfraAsyncjobsV1AsyncJob from './http.impl';
export function createAsyncJob() {
    var payload = {};
    var getRequestOptions = httpWixInfraAsyncjobsV1AsyncJob.createAsyncJob(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/async-jobs',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getAsyncJob() {
    var payload = { jobId: ':jobId' };
    var getRequestOptions = httpWixInfraAsyncjobsV1AsyncJob.getAsyncJob(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/v1/async-jobs/{jobId}',
        pathParams: { jobId: 'jobId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateAsyncJob() {
    var payload = { job: { id: ':jobId' } };
    var getRequestOptions = httpWixInfraAsyncjobsV1AsyncJob.updateAsyncJob(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'PATCH',
        path: '/v1/async-jobs/{job.id}',
        pathParams: { jobId: 'jobId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteAsyncJob() {
    var payload = { jobId: ':jobId' };
    var getRequestOptions = httpWixInfraAsyncjobsV1AsyncJob.deleteAsyncJob(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'DELETE',
        path: '/v1/async-jobs/{jobId}',
        pathParams: { jobId: 'jobId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function completeAsyncJob() {
    var payload = { jobId: ':jobId' };
    var getRequestOptions = httpWixInfraAsyncjobsV1AsyncJob.completeAsyncJob(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'PUT',
        path: '/v1/async-jobs/{jobId}/complete',
        pathParams: { jobId: 'jobId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function reportAsyncJobProgress() {
    var payload = { jobId: ':jobId' };
    var getRequestOptions = httpWixInfraAsyncjobsV1AsyncJob.reportAsyncJobProgress(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/async-jobs/{jobId}/items/report',
        pathParams: { jobId: 'jobId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listAsyncJobItems() {
    var payload = { jobId: ':jobId' };
    var getRequestOptions = httpWixInfraAsyncjobsV1AsyncJob.listAsyncJobItems(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/v1/async-jobs/{jobId}/items',
        pathParams: { jobId: 'jobId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
