// @wix/bex-core/dist/cjs/state/StringFilterState/StringFilterState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.stringFilter = exports.StringFilterState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _events = require("events");
var _mobx = require("mobx");
var _refreshFilter = require("../Filter/refreshFilter");
class StringFilterState {
  constructor(params) {
    (0, _defineProperty2.default)(this, "name", void 0);
    (0, _defineProperty2.default)(this, "persistent", void 0);
    (0, _defineProperty2.default)(this, "defaultValue", void 0);
    (0, _defineProperty2.default)(this, "itemKey", void 0);
    (0, _defineProperty2.default)(this, "itemName", void 0);
    (0, _defineProperty2.default)(this, "matches", void 0);
    (0, _defineProperty2.default)(this, "equals", void 0);
    (0, _defineProperty2.default)(this, "_value", void 0);
    (0, _defineProperty2.default)(this, "events", new _events.EventEmitter());
    (0, _defineProperty2.default)(this, "_toArray", value => {
      return [value];
    });
    this.name = params.name ?? '';
    this.persistent = params.persistent;
    this.defaultValue = params.defaultValue;
    this.itemKey = item => item;
    this.itemName = item => item;
    this._value = params.initialValue ?? this.defaultValue;
    this.matches = params.matches ?? ((item, value) => value == null || value === '' || typeof item === 'string' ? value.toLowerCase().includes(item.toLowerCase()) : false);
    this.equals = params.equals;
    (0, _mobx.makeObservable)(this, {
      _value: _mobx.observable.ref,
      persistent: _mobx.observable.ref,
      isEmpty: _mobx.computed,
      size: _mobx.computed,
      value: _mobx.computed,
      reset: _mobx.action,
      clone: _mobx.action,
      setValue: _mobx.action,
      changeValue: _mobx.action,
      scheduleRefresh: _mobx.action
    });
  }
  get value() {
    return this._value;
  }
  get toArray() {
    return [this._value];
  }
  get isEmpty() {
    return !this._value.length;
  }
  get size() {
    return this.isEmpty ? 0 : 1;
  }
  get toQueryString() {
    return this._value;
  }
  applyFromQueryString(str) {
    this.setValue(str);
  }
  encode(value) {
    return value;
  }
  decode(raw) {
    if (typeof raw === 'string') {
      return raw;
    }
    return '';
  }
  setValue(value, {
    emitEvents,
    ...options
  } = {}) {
    this._value = value;
    if (emitEvents) {
      for (const event of emitEvents) {
        this.events.emit(event, options);
      }
    }
  }
  changeValue(value, {
    emitEvents = [],
    ...options
  } = {}) {
    this.setValue(value, {
      ...options,
      emitEvents: ['change', ...emitEvents]
    });
  }
  refresh(value, options) {
    (0, _refreshFilter.refreshFilter)(this, value, options);
  }
  scheduleRefresh(value) {
    this.changeValue(value, {
      emitEvents: ['scheduleRefresh']
    });
  }
  reset(options = {}) {
    this.setValue(this.defaultValue, options);
  }
  remove(_, options = {}) {
    this.reset(options);
  }
  hasDiff(item) {
    return this.value !== item;
  }
  clone(params) {
    const {
      defaultValue,
      value: initialValue,
      matches,
      persistent,
      name
    } = this;
    return new StringFilterState({
      defaultValue,
      initialValue,
      matches,
      persistent,
      name,
      ...params
    });
  }
}
exports.StringFilterState = StringFilterState;
const stringFilter = (params = {}) => new StringFilterState({
  defaultValue: '',
  ...params
});
exports.stringFilter = stringFilter;
