// @wix/auto_sdk_restaurants_local-delivery-spi/build/internal/es/index.mjs
// src/localdelivery-spi-v1-tpa-local-delivery-spi.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/localdelivery-spi-v1-tpa-local-delivery-spi.http.ts
import { transformSDKFloatToRESTFloat } from "@wix/sdk-runtime/transformations/float";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressLocaldeliverySpiV1LocalDeliveryProviderUrl(opts) {
  const domainToMappings = {};
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_local-delivery-spi";
function listAccountIds(payload) {
  function __listAccountIds({ host }) {
    const metadata = {
      entityFqdn: "wix.localdelivery.spi.v1.tpa",
      method: "POST",
      methodFqn: "com.wixpress.localdelivery.spi.v1.LocalDeliveryProvider.ListAccountIds",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressLocaldeliverySpiV1LocalDeliveryProviderUrl({
        protoPath: "/account-ids",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __listAccountIds;
}
function businessStatus(payload) {
  function __businessStatus({ host }) {
    const metadata = {
      entityFqdn: "wix.localdelivery.spi.v1.tpa",
      method: "POST",
      methodFqn: "com.wixpress.localdelivery.spi.v1.LocalDeliveryProvider.BusinessStatus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressLocaldeliverySpiV1LocalDeliveryProviderUrl({
        protoPath: "/business-status",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __businessStatus;
}
function estimateDelivery(payload) {
  function __estimateDelivery({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "pickupTime" }, { path: "dropoffTime" }]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "pickupAddress.geocode.latitude" },
          { path: "pickupAddress.geocode.longitude" },
          { path: "dropoffAddress.geocode.latitude" },
          { path: "dropoffAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.localdelivery.spi.v1.tpa",
      method: "POST",
      methodFqn: "com.wixpress.localdelivery.spi.v1.LocalDeliveryProvider.EstimateDelivery",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressLocaldeliverySpiV1LocalDeliveryProviderUrl({
        protoPath: "/estimate-delivery",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "promisedPickupTime" },
            { path: "promisedDropoffTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __estimateDelivery;
}
function createDelivery(payload) {
  function __createDelivery({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "pickupTime" }, { path: "dropoffTime" }]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "pickupAddress.geocode.latitude" },
          { path: "pickupAddress.geocode.longitude" },
          { path: "dropoffAddress.geocode.latitude" },
          { path: "dropoffAddress.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.localdelivery.spi.v1.tpa",
      method: "POST",
      methodFqn: "com.wixpress.localdelivery.spi.v1.LocalDeliveryProvider.CreateDelivery",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressLocaldeliverySpiV1LocalDeliveryProviderUrl({
        protoPath: "/create-delivery",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "pickupWindowStartTime" },
            { path: "pickupWindowEndTime" },
            { path: "dropoffWindowStartTime" },
            { path: "dropoffWindowEndTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createDelivery;
}

// src/localdelivery-spi-v1-tpa-local-delivery-spi.universal.ts
import { transformSDKAddressToRESTAddress } from "@wix/sdk-runtime/transformations/address";
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var BusinessStatus = /* @__PURE__ */ ((BusinessStatus2) => {
  BusinessStatus2["NONE"] = "NONE";
  BusinessStatus2["READY"] = "READY";
  BusinessStatus2["PENDING"] = "PENDING";
  BusinessStatus2["ERROR"] = "ERROR";
  BusinessStatus2["NOT_STARTED"] = "NOT_STARTED";
  return BusinessStatus2;
})(BusinessStatus || {});
async function listAccountIds2(instanceId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    instanceId
  });
  const reqOpts = listAccountIds(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { instanceId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["instanceId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function businessStatus2(instanceId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    instanceId,
    accountId: options?.accountId
  });
  const reqOpts = businessStatus(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          instanceId: "$[0]",
          accountId: "$[1].accountId"
        },
        singleArgumentUnchanged: false
      },
      ["instanceId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function estimateDelivery2(instanceId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({
      instanceId,
      accountId: options?.accountId,
      pickupAddress: options?.pickupAddress,
      dropoffAddress: options?.dropoffAddress,
      pickupTime: options?.pickupTime,
      dropoffTime: options?.dropoffTime,
      storeLocationId: options?.storeLocationId,
      orderTotal: options?.orderTotal
    }),
    [
      {
        transformFn: transformSDKAddressToRESTAddress,
        paths: [{ path: "pickupAddress" }, { path: "dropoffAddress" }]
      }
    ]
  );
  const reqOpts = estimateDelivery(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          instanceId: "$[0]",
          accountId: "$[1].accountId",
          pickupAddress: "$[1].pickupAddress",
          dropoffAddress: "$[1].dropoffAddress",
          pickupTime: "$[1].pickupTime",
          dropoffTime: "$[1].dropoffTime",
          storeLocationId: "$[1].storeLocationId",
          orderTotal: "$[1].orderTotal"
        },
        singleArgumentUnchanged: false
      },
      ["instanceId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function createDelivery2(instanceId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({
      instanceId,
      accountId: options?.accountId,
      storeLocationId: options?.storeLocationId,
      pickupBusinessName: options?.pickupBusinessName,
      estimateId: options?.estimateId,
      pickupTime: options?.pickupTime,
      dropoffTime: options?.dropoffTime,
      pickupAddress: options?.pickupAddress,
      dropoffAddress: options?.dropoffAddress,
      pickupPhoneNumber: options?.pickupPhoneNumber,
      customer: options?.customer,
      pickupInstructions: options?.pickupInstructions,
      dropoffInstructions: options?.dropoffInstructions,
      contactlessDelivery: options?.contactlessDelivery,
      orderId: options?.orderId,
      orderTotal: options?.orderTotal,
      orderTip: options?.orderTip
    }),
    [
      {
        transformFn: transformSDKAddressToRESTAddress,
        paths: [{ path: "pickupAddress" }, { path: "dropoffAddress" }]
      }
    ]
  );
  const reqOpts = createDelivery(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          instanceId: "$[0]",
          accountId: "$[1].accountId",
          storeLocationId: "$[1].storeLocationId",
          pickupBusinessName: "$[1].pickupBusinessName",
          estimateId: "$[1].estimateId",
          pickupTime: "$[1].pickupTime",
          dropoffTime: "$[1].dropoffTime",
          pickupAddress: "$[1].pickupAddress",
          dropoffAddress: "$[1].dropoffAddress",
          pickupPhoneNumber: "$[1].pickupPhoneNumber",
          customer: "$[1].customer",
          pickupInstructions: "$[1].pickupInstructions",
          dropoffInstructions: "$[1].dropoffInstructions",
          contactlessDelivery: "$[1].contactlessDelivery",
          orderId: "$[1].orderId",
          orderTotal: "$[1].orderTotal",
          orderTip: "$[1].orderTip"
        },
        singleArgumentUnchanged: false
      },
      ["instanceId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/localdelivery-spi-v1-tpa-local-delivery-spi.public.ts
function listAccountIds3(httpClient) {
  return (instanceId) => listAccountIds2(
    instanceId,
    // @ts-ignore
    { httpClient }
  );
}
function businessStatus3(httpClient) {
  return (instanceId, options) => businessStatus2(
    instanceId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function estimateDelivery3(httpClient) {
  return (instanceId, options) => estimateDelivery2(
    instanceId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function createDelivery3(httpClient) {
  return (instanceId, options) => createDelivery2(
    instanceId,
    options,
    // @ts-ignore
    { httpClient }
  );
}

// src/localdelivery-spi-v1-tpa-local-delivery-spi.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
var listAccountIds4 = /* @__PURE__ */ createRESTModule(listAccountIds3);
var businessStatus4 = /* @__PURE__ */ createRESTModule(businessStatus3);
var estimateDelivery4 = /* @__PURE__ */ createRESTModule(estimateDelivery3);
var createDelivery4 = /* @__PURE__ */ createRESTModule(createDelivery3);
export {
  BusinessStatus,
  businessStatus4 as businessStatus,
  createDelivery4 as createDelivery,
  estimateDelivery4 as estimateDelivery,
  listAccountIds4 as listAccountIds
};
