// @wix/patterns/dist/cjs/components/DragHandle/DragHandle.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.DragHandle = DragHandle;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _DragHandleStCss = require("./DragHandle.st.css.js");
var _react = _interopRequireDefault(require("react"));
var _system = require("@wix/wix-ui-icons-common/system");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/DragHandle/DragHandle.tsx";
function DragHandle(props) {
  const {
    dataHook,
    domRef,
    dragging,
    disabled,
    className,
    ...rest
  } = props;
  const dragHandleIcon = disabled ? /*#__PURE__*/_react.default.createElement(_system.DragAndDropLockedLarge, {
    "data-hook": "dnd-locked",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 18,
      columnNumber: 5
    }
  }) : /*#__PURE__*/_react.default.createElement(_system.DragAndDropLarge, {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 20,
      columnNumber: 5
    }
  });
  return /*#__PURE__*/_react.default.createElement("div", (0, _extends2.default)({
    "data-hook": dataHook,
    ref: domRef
  }, rest, {
    className: (0, _DragHandleStCss.st)(_DragHandleStCss.classes.root, {
      disabled,
      dragging
    }, className),
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 24,
      columnNumber: 5
    }
  }), dragHandleIcon);
}
