// @wix/bex-utils/@wix/ambassador-items-selection-spi-host-v1-provider/build/cjs/builders.impl.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.aQueryItemsEmptyState = exports.aRenderOverrides = exports.aCursors = exports.aPagingMetadataV2 = exports.aAction = exports.aInboxData = exports.aEcomCatalogData = exports.aAdditionalData = exports.aImage = exports.aItem = exports.aQueryItemsResponse = exports.aCursorPaging = exports.aPaging = exports.aSortOrder = exports.aSorting = exports.aQuery = exports.aQueryItemsRequest = exports.aListProvidersResponse = exports.aListProvidersRequest = exports.aListInstalledProvidersResponse = exports.aListInstalledProvidersRequest = exports.aPaginationMode = exports.aAvatarShape = exports.aAvatarConfig = exports.aImageShape = exports.aImageConfig = exports.aThumbnailType = exports.aItemThumbnail = exports.aCreateNewItemInfo = exports.aAdditionalStepInfo = exports.aLearnMore = exports.aContentData = exports.aProviderFilterOptions = exports.aStaticFilterOption = exports.aStaticFilterOptions = exports.aFilterOptionsType = exports.aFilterSelectionType = exports.aTag = exports.aFilterInfo = exports.aSearchField = exports.aSearchParams = exports.aItemsSelectionProvider = void 0;
var metro_common_builders_1 = require("@wix/metro-common-builders");
var types_impl_1 = require("./types.impl");
// @ts-ignore
var callsite_1 = __importDefault(require("callsite"));
function aItemsSelectionProvider(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        key: (0, metro_common_builders_1.someString)(),
        appId: (0, metro_common_builders_1.someString)(),
        searchParams: aSearchParams(),
        contentData: aContentData(),
        additionalStepInfo: aAdditionalStepInfo(),
        supportedTags: [
            aTag(),
            aTag(),
            aTag(),
            aTag(),
            aTag(),
            aTag(),
            aTag(),
            aTag(),
            aTag(),
            aTag(),
        ],
        createNewItemInfo: aCreateNewItemInfo(),
        itemThumbnail: aItemThumbnail(),
        paginationMode: aPaginationMode(),
    }, overrideObject);
}
exports.aItemsSelectionProvider = aItemsSelectionProvider;
function aSearchParams(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        fields: [
            aSearchField(),
            aSearchField(),
            aSearchField(),
            aSearchField(),
            aSearchField(),
            aSearchField(),
            aSearchField(),
            aSearchField(),
            aSearchField(),
            aSearchField(),
        ],
        filters: [
            aFilterInfo(),
            aFilterInfo(),
            aFilterInfo(),
            aFilterInfo(),
            aFilterInfo(),
            aFilterInfo(),
            aFilterInfo(),
            aFilterInfo(),
            aFilterInfo(),
            aFilterInfo(),
        ],
    }, overrideObject);
}
exports.aSearchParams = aSearchParams;
function aSearchField(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        key: (0, metro_common_builders_1.someString)(),
        description: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '100' },
            ],
        }),
    }, overrideObject);
}
exports.aSearchField = aSearchField;
function aFilterInfo(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        staticOptions: aStaticFilterOptions(),
        key: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '100' },
            ],
        }),
        tag: aTag(),
        selectionType: aFilterSelectionType(),
        optionsType: aFilterOptionsType(),
        placeholder: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '500' }],
        }),
    }, overrideObject);
}
exports.aFilterInfo = aFilterInfo;
function aTag(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.Tag.UNKNOWN_TAG; }
    return enumFirstItem;
}
exports.aTag = aTag;
function aFilterSelectionType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.FilterSelectionType.UNKNOWN_SELECTION; }
    return enumFirstItem;
}
exports.aFilterSelectionType = aFilterSelectionType;
function aFilterOptionsType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.FilterOptionsType.UNKNOWN_FILTER_OPTIONS; }
    return enumFirstItem;
}
exports.aFilterOptionsType = aFilterOptionsType;
function aStaticFilterOptions(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        data: [
            aStaticFilterOption(),
            aStaticFilterOption(),
            aStaticFilterOption(),
            aStaticFilterOption(),
            aStaticFilterOption(),
            aStaticFilterOption(),
            aStaticFilterOption(),
            aStaticFilterOption(),
            aStaticFilterOption(),
            aStaticFilterOption(),
        ],
    }, overrideObject);
}
exports.aStaticFilterOptions = aStaticFilterOptions;
function aStaticFilterOption(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        label: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '500' },
            ],
        }),
        value: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '100' },
            ],
        }),
    }, overrideObject);
}
exports.aStaticFilterOption = aStaticFilterOption;
function aProviderFilterOptions(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        key: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MIN_LENGTH', value: '3' },
                { type: 'MAX_LENGTH', value: '250' },
            ],
        }),
    }, overrideObject);
}
exports.aProviderFilterOptions = aProviderFilterOptions;
function aContentData(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        iconKey: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MIN_LENGTH', value: '3' },
                { type: 'MAX_LENGTH', value: '50' },
            ],
        }),
        providerName: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '100' },
            ],
        }),
        title: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '100' },
            ],
        }),
        button: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '50' },
            ],
        }),
        subtitle: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '100' },
            ],
        }),
        learnMore: aLearnMore(),
    }, overrideObject);
}
exports.aContentData = aContentData;
function aLearnMore(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        url: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '2048' }],
        }),
        label: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '500' }],
        }),
    }, overrideObject);
}
exports.aLearnMore = aLearnMore;
function aAdditionalStepInfo(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        widgetId: (0, metro_common_builders_1.someString)(),
        mobileComponentId: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '50' },
            ],
        }),
        title: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '100' },
            ],
        }),
        subtitle: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        hideItemNote: true,
    }, overrideObject);
}
exports.aAdditionalStepInfo = aAdditionalStepInfo;
function aCreateNewItemInfo(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        widgetId: (0, metro_common_builders_1.someString)(),
        mobileComponentId: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '50' },
            ],
        }),
        buttonText: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '50' },
            ],
        }),
    }, overrideObject);
}
exports.aCreateNewItemInfo = aCreateNewItemInfo;
function aItemThumbnail(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        imageConfig: aImageConfig(),
        type: aThumbnailType(),
        hidden: true,
    }, overrideObject);
}
exports.aItemThumbnail = aItemThumbnail;
function aThumbnailType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.ThumbnailType.UNKNOWN_THUMBNAIL_TYPE; }
    return enumFirstItem;
}
exports.aThumbnailType = aThumbnailType;
function aImageConfig(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        shape: aImageShape(),
    }, overrideObject);
}
exports.aImageConfig = aImageConfig;
function aImageShape(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.ImageShape.UNKNOWN_SHAPE; }
    return enumFirstItem;
}
exports.aImageShape = aImageShape;
function aAvatarConfig(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        shape: aAvatarShape(),
    }, overrideObject);
}
exports.aAvatarConfig = aAvatarConfig;
function aAvatarShape(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.AvatarShape.UNKNOWN_SHAPE; }
    return enumFirstItem;
}
exports.aAvatarShape = aAvatarShape;
function aPaginationMode(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.PaginationMode.UNKNOWN_PAGINATION_MODE; }
    return enumFirstItem;
}
exports.aPaginationMode = aPaginationMode;
function aListInstalledProvidersRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        tag: aTag(),
    }, overrideObject);
}
exports.aListInstalledProvidersRequest = aListInstalledProvidersRequest;
function aListInstalledProvidersResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        providers: [aItemsSelectionProvider()],
    }, overrideObject);
}
exports.aListInstalledProvidersResponse = aListInstalledProvidersResponse;
function aListProvidersRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        tag: aTag(),
    }, overrideObject);
}
exports.aListProvidersRequest = aListProvidersRequest;
function aListProvidersResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        providers: [aItemsSelectionProvider()],
    }, overrideObject);
}
exports.aListProvidersResponse = aListProvidersResponse;
function aQueryItemsRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        providerKey: (0, metro_common_builders_1.someString)(),
        query: aQuery(),
        tag: aTag(),
    }, overrideObject);
}
exports.aQueryItemsRequest = aQueryItemsRequest;
function aQuery(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        paging: aPaging(),
        filter: {},
        sort: [aSorting()],
        search: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aQuery = aQuery;
function aSorting(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        fieldName: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '512' }],
        }),
        order: aSortOrder(),
        selectItemsBy: [{}, {}, {}, {}, {}, {}, {}, {}, {}, {}],
    }, overrideObject);
}
exports.aSorting = aSorting;
function aSortOrder(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.SortOrder.ASC; }
    return enumFirstItem;
}
exports.aSortOrder = aSortOrder;
function aPaging(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        limit: (0, metro_common_builders_1.someNumber)({
            validations: [{ type: 'MIN' }],
        }),
        offset: (0, metro_common_builders_1.someNumber)({
            validations: [{ type: 'MIN' }],
        }),
    }, overrideObject);
}
exports.aPaging = aPaging;
function aCursorPaging(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        limit: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'MIN' },
                { type: 'MAX', value: '100' },
            ],
        }),
        cursor: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MAX_LENGTH', value: '16000' },
            ],
        }),
    }, overrideObject);
}
exports.aCursorPaging = aCursorPaging;
function aQueryItemsResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        items: [aItem()],
        pagingMetadata: aPagingMetadataV2(),
        renderOverrides: aRenderOverrides(),
    }, overrideObject);
}
exports.aQueryItemsResponse = aQueryItemsResponse;
function aItem(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        id: (0, metro_common_builders_1.someString)(),
        name: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '250' }],
        }),
        description: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '1000' }],
        }),
        image: aImage(),
        note: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        hasAdditionalStep: true,
        additionalData: aAdditionalData(),
        additionalFields: {},
    }, overrideObject);
}
exports.aItem = aItem;
function aImage(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        url: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '2048' }],
        }),
    }, overrideObject);
}
exports.aImage = aImage;
function aAdditionalData(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        ecomCatalogData: aEcomCatalogData(),
    }, overrideObject);
}
exports.aAdditionalData = aAdditionalData;
function aEcomCatalogData(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        appId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        catalogItemId: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '36' },
            ],
        }),
        price: (0, metro_common_builders_1.someString)({
            validations: [
                {
                    type: 'DECIMAL_VALUE',
                    value: { gte: '0', lte: '1000000000000000' },
                },
            ],
        }),
        options: {},
    }, overrideObject);
}
exports.aEcomCatalogData = aEcomCatalogData;
function aInboxData(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        actions: [
            aAction(),
            aAction(),
            aAction(),
            aAction(),
            aAction(),
            aAction(),
            aAction(),
            aAction(),
            aAction(),
            aAction(),
        ],
    }, overrideObject);
}
exports.aInboxData = aInboxData;
function aAction(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        title: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        url: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '2048' }],
        }),
    }, overrideObject);
}
exports.aAction = aAction;
function aPagingMetadataV2(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        count: (0, metro_common_builders_1.someNumber)(),
        offset: (0, metro_common_builders_1.someNumber)(),
        total: (0, metro_common_builders_1.someNumber)(),
        tooManyToCount: true,
        cursors: aCursors(),
        hasNext: true,
    }, overrideObject);
}
exports.aPagingMetadataV2 = aPagingMetadataV2;
function aCursors(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        next: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MAX_LENGTH', value: '16000' },
            ],
        }),
        prev: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MAX_LENGTH', value: '16000' },
            ],
        }),
    }, overrideObject);
}
exports.aCursors = aCursors;
function aRenderOverrides(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        emptyState: aQueryItemsEmptyState(),
    }, overrideObject);
}
exports.aRenderOverrides = aRenderOverrides;
function aQueryItemsEmptyState(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        title: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
        subtitle: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '200' }],
        }),
    }, overrideObject);
}
exports.aQueryItemsEmptyState = aQueryItemsEmptyState;
function isCurrentBuilderWasPreviouslyCalled() {
    var callChainFunctionNames = (0, callsite_1.default)().map(function (item) {
        return item.getFunctionName();
    });
    var callChainWithoutCurrentBuilder = callChainFunctionNames.splice(2);
    var currentBuilderName = callChainFunctionNames[1];
    return callChainWithoutCurrentBuilder.some(function (functionName) { return functionName === currentBuilderName; });
}
__exportStar(require("@wix/metro-common-builders"), exports);
