// @wix/bex-utils/@wix/ambassador-items-selection-spi-host-v1-provider/build/cjs/meta.impl.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.queryItems = exports.listProviders = exports.listInstalledProviders = void 0;
var httpWixItemsSelectionSpiHostV1Provider = __importStar(require("./http.impl"));
function listInstalledProviders() {
    var payload = {};
    var getRequestOptions = httpWixItemsSelectionSpiHostV1Provider.listInstalledProviders(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/v1/items-selection/installed-providers',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listInstalledProviders = listInstalledProviders;
function listProviders() {
    var payload = {};
    var getRequestOptions = httpWixItemsSelectionSpiHostV1Provider.listProviders(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/items-selection/providers',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listProviders = listProviders;
function queryItems() {
    var payload = { providerKey: ':providerKey' };
    var getRequestOptions = httpWixItemsSelectionSpiHostV1Provider.queryItems(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/items-selection/{providerKey}/items',
        pathParams: { providerKey: 'providerKey' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryItems = queryItems;
