// @wix/bex-utils/@wix/ambassador-devcenter-app-market-listing-v1-market-listing/build/cjs/builders.impl.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.aRestoreInfo = exports.aEntityCreatedEvent = exports.aDomainEvent = exports.aCursors = exports.aPagingMetadataV2 = exports.aQueryMarketListingResponse = exports.aCursorPaging = exports.aPaging = exports.aSortOrder = exports.aSorting = exports.aQueryV2 = exports.aQueryMarketListingRequest = exports.aPartialVersionOptions = exports.aVersionStatus = exports.aAppVersionArchived = exports.aAppVersionDraftChanged = exports.aAppVersionApproved = exports.aAppVersionReleased = exports.aAppVersionDeclined = exports.aAppVersionPublished = exports.aAppVersionSubmitted = exports.aAppVersionCreated = exports.aAppVersionStateChanged = exports.aEmpty = exports.aApiRequiredApp = exports.aApiInstallationRequirement = exports.aTracking = exports.aApiContactInfo = exports.aApiBenefits = exports.aApiBasicInfo = exports.aApiMarketListing = exports.aOldListingChanged = exports.aUpdateMarketListingResponse = exports.aUpdateMarketListingRequest = exports.aGetMarketListingResponse = exports.aGetMarketListingRequest = exports.aCreateMarketListingResponse = exports.aCreateMarketListingRequest = exports.aPricingData = exports.aClassification = exports.aStatus = exports.aRequiredApp = exports.aInstallationRequirement = exports.aContactInfo = exports.aMediaType = exports.aAsset = exports.aAssets = exports.aBenefits = exports.aBasicInfo = exports.aMarketListing = void 0;
exports.aWebhookIdentityType = exports.aIdentificationData = exports.aMessageEnvelope = exports.aCompareResponse = exports.aCompareRequest = exports.aPromoteResponse = exports.aPromoteRequest = exports.aActionEvent = exports.aEntityDeletedEvent = exports.aEntityUpdatedEvent = void 0;
var metro_common_builders_1 = require("@wix/metro-common-builders");
var types_impl_1 = require("./types.impl");
// @ts-ignore
var callsite_1 = __importDefault(require("callsite"));
function aMarketListing(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        id: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        appId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        languageCode: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'LANGUAGE' }],
        }),
        basicInfo: aBasicInfo(),
        assetsMap: { assetsMap: aAssets() },
        contactInfo: aContactInfo(),
        installationRequirement: aInstallationRequirement(),
        revision: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'DECIMAL_VALUE' }],
            isUnsigned: true,
        }),
        createdDate: (0, metro_common_builders_1.someDate)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        updatedDate: (0, metro_common_builders_1.someDate)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        status: aStatus(),
        classification: aClassification(),
        pricingData: aPricingData(),
    }, overrideObject);
}
exports.aMarketListing = aMarketListing;
function aBasicInfo(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        name: (0, metro_common_builders_1.someString)(),
        teaser: (0, metro_common_builders_1.someString)(),
        longDescription: (0, metro_common_builders_1.someString)(),
        benefits: [aBenefits()],
        searchKeywords: [(0, metro_common_builders_1.someString)()],
        demoUrl: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'WEB_URL' }],
        }),
        categories: [(0, metro_common_builders_1.someString)()],
    }, overrideObject);
}
exports.aBasicInfo = aBasicInfo;
function aBenefits(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        title: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aBenefits = aBenefits;
function aAssets(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        assets: [aAsset()],
    }, overrideObject);
}
exports.aAssets = aAssets;
function aAsset(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        url: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'WEB_URL' }],
        }),
        mediaType: aMediaType(),
    }, overrideObject);
}
exports.aAsset = aAsset;
function aMediaType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.MediaType.VIDEO; }
    return enumFirstItem;
}
exports.aMediaType = aMediaType;
function aContactInfo(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        website: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'WEB_URL' }],
        }),
        developerMetaSiteId: (0, metro_common_builders_1.someString)(),
        supportEmail: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'EMAIL' }],
        }),
        privacyPolicyUrl: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'WEB_URL' }],
        }),
        termsOfUseUrl: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'WEB_URL' }],
        }),
    }, overrideObject);
}
exports.aContactInfo = aContactInfo;
function aInstallationRequirement(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        requiredApps: [aRequiredApp()],
        includeCountries: [
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'FORMAT', value: 'COUNTRY' },
                ],
            }),
        ],
        excludeCountries: [
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'FORMAT', value: 'COUNTRY' },
                ],
            }),
        ],
        includeLanguages: [
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'FORMAT', value: 'LANGUAGE' },
                ],
            }),
        ],
        excludeLanguages: [
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'FORMAT', value: 'LANGUAGE' },
                ],
            }),
        ],
        allowLanguageCustomization: true,
        requiredWixPremium: true,
    }, overrideObject);
}
exports.aInstallationRequirement = aInstallationRequirement;
function aRequiredApp(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        id: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        slug: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aRequiredApp = aRequiredApp;
function aStatus(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.Status.DRAFT; }
    return enumFirstItem;
}
exports.aStatus = aStatus;
function aClassification(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.Classification.SITE_APP; }
    return enumFirstItem;
}
exports.aClassification = aClassification;
function aPricingData(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        externalPricingPageUrl: (0, metro_common_builders_1.someString)(),
        externalCheckout: true,
        freeTrialDays: (0, metro_common_builders_1.someNumber)(),
        internalFreeTrialDays: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'MIN' },
                { type: 'MAX', value: '30' },
            ],
        }),
    }, overrideObject);
}
exports.aPricingData = aPricingData;
function aCreateMarketListingRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        marketListing: aMarketListing(),
    }, overrideObject);
}
exports.aCreateMarketListingRequest = aCreateMarketListingRequest;
function aCreateMarketListingResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        marketListing: aMarketListing(),
    }, overrideObject);
}
exports.aCreateMarketListingResponse = aCreateMarketListingResponse;
function aGetMarketListingRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        id: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
exports.aGetMarketListingRequest = aGetMarketListingRequest;
function aGetMarketListingResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        marketListing: aMarketListing(),
    }, overrideObject);
}
exports.aGetMarketListingResponse = aGetMarketListingResponse;
function aUpdateMarketListingRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        marketListing: aMarketListing(),
        mask: (0, metro_common_builders_1.someRepeated)(),
    }, overrideObject);
}
exports.aUpdateMarketListingRequest = aUpdateMarketListingRequest;
function aUpdateMarketListingResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        marketListing: aMarketListing(),
    }, overrideObject);
}
exports.aUpdateMarketListingResponse = aUpdateMarketListingResponse;
function aOldListingChanged(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        appId: (0, metro_common_builders_1.someString)(),
        marketListing: aApiMarketListing(),
        languageCode: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'LANGUAGE' }],
        }),
        name: (0, metro_common_builders_1.someString)(),
        iconUrl: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aOldListingChanged = aOldListingChanged;
function aApiMarketListing(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        basicInfo: aApiBasicInfo(),
        assetsMap: { assetsMap: aAssets() },
        contactInfo: aApiContactInfo(),
        trackingAndAnalytics: aTracking(),
        languageCode: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'LANGUAGE' }],
        }),
        installationRequirement: aApiInstallationRequirement(),
    }, overrideObject);
}
exports.aApiMarketListing = aApiMarketListing;
function aApiBasicInfo(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        appId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        name: (0, metro_common_builders_1.someString)(),
        teaser: (0, metro_common_builders_1.someString)(),
        longDescription: (0, metro_common_builders_1.someString)(),
        benefits: [aApiBenefits()],
        searchKeywords: [(0, metro_common_builders_1.someString)()],
        demoUrl: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'URI' }],
        }),
        categories: [(0, metro_common_builders_1.someString)()],
    }, overrideObject);
}
exports.aApiBasicInfo = aApiBasicInfo;
function aApiBenefits(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        title: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aApiBenefits = aApiBenefits;
function aApiContactInfo(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        website: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'URI' }],
        }),
        developerMetaSiteId: (0, metro_common_builders_1.someString)(),
        supportEmail: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'EMAIL' }],
        }),
        privacyPolicyUrl: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'URI' }],
        }),
        termsOfUseUrl: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'URI' }],
        }),
    }, overrideObject);
}
exports.aApiContactInfo = aApiContactInfo;
function aTracking(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        googleAnalyticsTrackingId: (0, metro_common_builders_1.someString)(),
        googleConversionId: (0, metro_common_builders_1.someString)(),
        adRolllAdvId: (0, metro_common_builders_1.someString)(),
        adRolllPixId: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aTracking = aTracking;
function aApiInstallationRequirement(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        requiredApps: [aApiRequiredApp()],
        includeCountries: [
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'FORMAT', value: 'COUNTRY' },
                ],
            }),
        ],
        excludeCountries: [
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'FORMAT', value: 'COUNTRY' },
                ],
            }),
        ],
        includeLanguages: [
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'FORMAT', value: 'LANGUAGE' },
                ],
            }),
        ],
        excludeLanguages: [
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'FORMAT', value: 'LANGUAGE' },
                ],
            }),
        ],
        allowLanguageCustomization: true,
    }, overrideObject);
}
exports.aApiInstallationRequirement = aApiInstallationRequirement;
function aApiRequiredApp(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        id: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        slug: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aApiRequiredApp = aApiRequiredApp;
function aEmpty(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({}, overrideObject);
}
exports.aEmpty = aEmpty;
function aAppVersionStateChanged(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        created: aAppVersionCreated(),
        appId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        version: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        previousState: aVersionStatus(),
        partialVersionOptions: aPartialVersionOptions(),
    }, overrideObject);
}
exports.aAppVersionStateChanged = aAppVersionStateChanged;
function aAppVersionCreated(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({}, overrideObject);
}
exports.aAppVersionCreated = aAppVersionCreated;
function aAppVersionSubmitted(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({}, overrideObject);
}
exports.aAppVersionSubmitted = aAppVersionSubmitted;
function aAppVersionPublished(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({}, overrideObject);
}
exports.aAppVersionPublished = aAppVersionPublished;
function aAppVersionDeclined(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({}, overrideObject);
}
exports.aAppVersionDeclined = aAppVersionDeclined;
function aAppVersionReleased(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({}, overrideObject);
}
exports.aAppVersionReleased = aAppVersionReleased;
function aAppVersionApproved(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({}, overrideObject);
}
exports.aAppVersionApproved = aAppVersionApproved;
function aAppVersionDraftChanged(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({}, overrideObject);
}
exports.aAppVersionDraftChanged = aAppVersionDraftChanged;
function aAppVersionArchived(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({}, overrideObject);
}
exports.aAppVersionArchived = aAppVersionArchived;
function aVersionStatus(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.VersionStatus.NONE_STATUS; }
    return enumFirstItem;
}
exports.aVersionStatus = aVersionStatus;
function aPartialVersionOptions(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        includeDraftDevCenterData: true,
        includeDraftMarketListingData: true,
        includeDraftPricingInfoData: true,
    }, overrideObject);
}
exports.aPartialVersionOptions = aPartialVersionOptions;
function aQueryMarketListingRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        query: aQueryV2(),
    }, overrideObject);
}
exports.aQueryMarketListingRequest = aQueryMarketListingRequest;
function aQueryV2(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        paging: aPaging(),
        filter: {},
        sort: [aSorting()],
        fields: [(0, metro_common_builders_1.someString)()],
        fieldsets: [(0, metro_common_builders_1.someString)()],
    }, overrideObject);
}
exports.aQueryV2 = aQueryV2;
function aSorting(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        fieldName: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '512' }],
        }),
        order: aSortOrder(),
        selectItemsBy: [{}, {}, {}, {}, {}, {}, {}, {}, {}, {}],
    }, overrideObject);
}
exports.aSorting = aSorting;
function aSortOrder(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.SortOrder.ASC; }
    return enumFirstItem;
}
exports.aSortOrder = aSortOrder;
function aPaging(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        limit: (0, metro_common_builders_1.someNumber)({
            validations: [{ type: 'MIN' }],
        }),
        offset: (0, metro_common_builders_1.someNumber)({
            validations: [{ type: 'MIN' }],
        }),
    }, overrideObject);
}
exports.aPaging = aPaging;
function aCursorPaging(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        limit: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'MIN' },
                { type: 'MAX', value: '100' },
            ],
        }),
        cursor: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MAX_LENGTH', value: '16000' },
            ],
        }),
    }, overrideObject);
}
exports.aCursorPaging = aCursorPaging;
function aQueryMarketListingResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        marketListing: [
            aMarketListing(),
            aMarketListing(),
            aMarketListing(),
            aMarketListing(),
            aMarketListing(),
            aMarketListing(),
            aMarketListing(),
            aMarketListing(),
            aMarketListing(),
            aMarketListing(),
        ],
        metadata: aPagingMetadataV2(),
    }, overrideObject);
}
exports.aQueryMarketListingResponse = aQueryMarketListingResponse;
function aPagingMetadataV2(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        count: (0, metro_common_builders_1.someNumber)(),
        offset: (0, metro_common_builders_1.someNumber)(),
        total: (0, metro_common_builders_1.someNumber)(),
        tooManyToCount: true,
        cursors: aCursors(),
        hasNext: true,
    }, overrideObject);
}
exports.aPagingMetadataV2 = aPagingMetadataV2;
function aCursors(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        next: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MAX_LENGTH', value: '16000' },
            ],
        }),
        prev: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MAX_LENGTH', value: '16000' },
            ],
        }),
    }, overrideObject);
}
exports.aCursors = aCursors;
function aDomainEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        createdEvent: aEntityCreatedEvent(),
        id: (0, metro_common_builders_1.someString)(),
        entityFqdn: (0, metro_common_builders_1.someString)(),
        slug: (0, metro_common_builders_1.someString)(),
        entityId: (0, metro_common_builders_1.someString)(),
        eventTime: (0, metro_common_builders_1.someDate)(),
        triggeredByAnonymizeRequest: true,
        originatedFrom: (0, metro_common_builders_1.someString)(),
        entityEventSequence: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'DECIMAL_VALUE' }],
        }),
    }, overrideObject);
}
exports.aDomainEvent = aDomainEvent;
function aEntityCreatedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        entityAsJson: (0, metro_common_builders_1.someString)(),
        triggeredByUndelete: true,
        restoreInfo: aRestoreInfo(),
        additionalMetadataAsJson: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aEntityCreatedEvent = aEntityCreatedEvent;
function aRestoreInfo(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        deletedDate: (0, metro_common_builders_1.someDate)(),
    }, overrideObject);
}
exports.aRestoreInfo = aRestoreInfo;
function aEntityUpdatedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        currentEntityAsJson: (0, metro_common_builders_1.someString)(),
        previousEntityAsJson: (0, metro_common_builders_1.someString)(),
        modifiedFields: { modifiedFields: (0, metro_common_builders_1.someAny)() },
        additionalMetadataAsJson: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aEntityUpdatedEvent = aEntityUpdatedEvent;
function aEntityDeletedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        movedToTrash: true,
        deletedEntityAsJson: (0, metro_common_builders_1.someString)(),
        additionalMetadataAsJson: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aEntityDeletedEvent = aEntityDeletedEvent;
function aActionEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        bodyAsJson: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aActionEvent = aActionEvent;
function aPromoteRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        appId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        from: aStatus(),
        to: aStatus(),
    }, overrideObject);
}
exports.aPromoteRequest = aPromoteRequest;
function aPromoteResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        appId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        status: aStatus(),
    }, overrideObject);
}
exports.aPromoteResponse = aPromoteResponse;
function aCompareRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        appId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        from: aStatus(),
        to: aStatus(),
    }, overrideObject);
}
exports.aCompareRequest = aCompareRequest;
function aCompareResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        equals: true,
    }, overrideObject);
}
exports.aCompareResponse = aCompareResponse;
function aMessageEnvelope(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        instanceId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        eventType: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '150' }],
        }),
        identity: aIdentificationData(),
        data: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aMessageEnvelope = aMessageEnvelope;
function aIdentificationData(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        anonymousVisitorId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        identityType: aWebhookIdentityType(),
    }, overrideObject);
}
exports.aIdentificationData = aIdentificationData;
function aWebhookIdentityType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.WebhookIdentityType.UNKNOWN; }
    return enumFirstItem;
}
exports.aWebhookIdentityType = aWebhookIdentityType;
function isCurrentBuilderWasPreviouslyCalled() {
    var callChainFunctionNames = (0, callsite_1.default)().map(function (item) {
        return item.getFunctionName();
    });
    var callChainWithoutCurrentBuilder = callChainFunctionNames.splice(2);
    var currentBuilderName = callChainFunctionNames[1];
    return callChainWithoutCurrentBuilder.some(function (functionName) { return functionName === currentBuilderName; });
}
__exportStar(require("@wix/metro-common-builders"), exports);
