// @wix/bex-utils/@wix/ambassador-devcenter-app-market-listing-v1-market-listing/build/cjs/meta.impl.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.compare = exports.queryMarketListing = exports.updateMarketListing = exports.getMarketListing = exports.createMarketListing = void 0;
var httpWixDevcenterAppMarketListingV1MarketListing = __importStar(require("./http.impl"));
function createMarketListing() {
    var payload = {};
    var getRequestOptions = httpWixDevcenterAppMarketListingV1MarketListing.createMarketListing(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/marketListings',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createMarketListing = createMarketListing;
function getMarketListing() {
    var payload = { id: ':id' };
    var getRequestOptions = httpWixDevcenterAppMarketListingV1MarketListing.getMarketListing(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/v1/marketListings/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getMarketListing = getMarketListing;
function updateMarketListing() {
    var payload = { marketListing: { id: ':marketListingId' } };
    var getRequestOptions = httpWixDevcenterAppMarketListingV1MarketListing.updateMarketListing(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'PUT',
        path: '/v1/marketListings/{marketListing.id}',
        pathParams: { marketListingId: 'marketListingId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateMarketListing = updateMarketListing;
function queryMarketListing() {
    var payload = {};
    var getRequestOptions = httpWixDevcenterAppMarketListingV1MarketListing.queryMarketListing(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/marketListings/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryMarketListing = queryMarketListing;
function compare() {
    var payload = {};
    var getRequestOptions = httpWixDevcenterAppMarketListingV1MarketListing.compare(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/market-listings/compare',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.compare = compare;
