// @wix/auto_sdk_payments_refunds/build/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  Initiator: () => Initiator,
  RejectionReason: () => RejectionReason,
  SortOrder: () => SortOrder,
  Status: () => Status,
  WebhookIdentityType: () => WebhookIdentityType,
  createRefund: () => createRefund4,
  getRefund: () => getRefund4,
  onRefundCreated: () => onRefundCreated2,
  onRefundUpdated: () => onRefundUpdated2,
  queryRefunds: () => queryRefunds4
});
module.exports = __toCommonJS(index_exports);

// src/payments-refunds-v1-refund-refunds.public.ts
var import_rename_all_nested_keys2 = require("@wix/sdk-runtime/rename-all-nested-keys");
var import_timestamp3 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths3 = require("@wix/sdk-runtime/transformations/transform-paths");
var import_sdk_types = require("@wix/sdk-types");

// src/payments-refunds-v1-refund-refunds.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/payments-refunds-v1-refund-refunds.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixPaymentsRefundsV1RefundServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/payments/refunds/v1",
        destPath: "/v1"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/payments/refunds/v1/refunds",
        destPath: "/v1/refunds"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/payments/refunds/v1/refunds",
        destPath: "/v1/refunds"
      }
    ],
    _: [
      {
        srcPath: "/payments/refunds/v1/refunds",
        destPath: "/v1/refunds"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_refunds";
function createRefund(payload) {
  function __createRefund({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "refund.createdDate" }, { path: "refund.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.payments.refunds.v1.refund",
      method: "POST",
      methodFqn: "wix.payments.refunds.v1.RefundService.CreateRefund",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsRefundsV1RefundServiceUrl({
        protoPath: "/v1/refunds",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "refund.createdDate" },
            { path: "refund.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createRefund;
}
function getRefund(payload) {
  function __getRefund({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.refunds.v1.refund",
      method: "GET",
      methodFqn: "wix.payments.refunds.v1.RefundService.GetRefund",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsRefundsV1RefundServiceUrl({
        protoPath: "/v1/refunds/{refundId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "refund.createdDate" },
            { path: "refund.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getRefund;
}
function queryRefunds(payload) {
  function __queryRefunds({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.refunds.v1.refund",
      method: "POST",
      methodFqn: "wix.payments.refunds.v1.RefundService.QueryRefunds",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsRefundsV1RefundServiceUrl({
        protoPath: "/v1/refunds/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "refunds.createdDate" },
            { path: "refunds.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryRefunds;
}

// src/payments-refunds-v1-refund-refunds.universal.ts
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["UNKNOWN_STATUS"] = "UNKNOWN_STATUS";
  Status2["PENDING"] = "PENDING";
  Status2["SUCCEEDED"] = "SUCCEEDED";
  Status2["FAILED"] = "FAILED";
  Status2["REVERSED"] = "REVERSED";
  return Status2;
})(Status || {});
var Initiator = /* @__PURE__ */ ((Initiator2) => {
  Initiator2["UNKNOWN_INITIATOR"] = "UNKNOWN_INITIATOR";
  Initiator2["WIX"] = "WIX";
  Initiator2["API"] = "API";
  Initiator2["PROVIDER"] = "PROVIDER";
  return Initiator2;
})(Initiator || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var RejectionReason = /* @__PURE__ */ ((RejectionReason2) => {
  RejectionReason2["UNKNOWN_REJECTION_REASON"] = "UNKNOWN_REJECTION_REASON";
  RejectionReason2["CHARGE_REFUNDED"] = "CHARGE_REFUNDED";
  RejectionReason2["CHARGE_REFUND_IN_PROGRESS"] = "CHARGE_REFUND_IN_PROGRESS";
  RejectionReason2["CHARGE_DISPUTED"] = "CHARGE_DISPUTED";
  RejectionReason2["CHARGE_REFUND_PERIOD_ENDED"] = "CHARGE_REFUND_PERIOD_ENDED";
  RejectionReason2["CHARGE_UNPAID"] = "CHARGE_UNPAID";
  RejectionReason2["PROVIDER_DOWN"] = "PROVIDER_DOWN";
  RejectionReason2["PROVIDER_NOT_SUPPORTED"] = "PROVIDER_NOT_SUPPORTED";
  RejectionReason2["PAYMENT_METHOD_NOT_SUPPORTED"] = "PAYMENT_METHOD_NOT_SUPPORTED";
  RejectionReason2["MERCHANT_BALANCE_INSUFFICIENT"] = "MERCHANT_BALANCE_INSUFFICIENT";
  RejectionReason2["ONLINE_REFUND_NOT_SUPPORTED"] = "ONLINE_REFUND_NOT_SUPPORTED";
  RejectionReason2["NOT_AUTHORIZED"] = "NOT_AUTHORIZED";
  return RejectionReason2;
})(RejectionReason || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createRefund2(refund, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    refund,
    previouslyRefundedAmount: options?.previouslyRefundedAmount
  });
  const reqOpts = createRefund(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.refund;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          refund: "$[0]",
          previouslyRefundedAmount: "$[1].previouslyRefundedAmount"
        },
        singleArgumentUnchanged: false
      },
      ["refund", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getRefund2(refundId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ refundId });
  const reqOpts = getRefund(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.refund;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { refundId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["refundId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryRefunds2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryRefunds(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [])
      );
      return {
        items: transformedData?.refunds,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}

// src/payments-refunds-v1-refund-refunds.public.ts
function createRefund3(httpClient) {
  return (refund, options) => createRefund2(
    refund,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function getRefund3(httpClient) {
  return (refundId) => getRefund2(
    refundId,
    // @ts-ignore
    { httpClient }
  );
}
function queryRefunds3(httpClient) {
  return () => queryRefunds2(
    // @ts-ignore
    { httpClient }
  );
}
var onRefundCreated = (0, import_sdk_types.EventDefinition)(
  "wix.payments.refunds.v1.refund_created",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onRefundUpdated = (0, import_sdk_types.EventDefinition)(
  "wix.payments.refunds.v1.refund_updated",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();

// src/payments-refunds-v1-refund-refunds.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var import_event_definition_modules = require("@wix/sdk-runtime/event-definition-modules");
var createRefund4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(createRefund3);
var getRefund4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getRefund3);
var queryRefunds4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(queryRefunds3);
var onRefundCreated2 = (0, import_event_definition_modules.createEventModule)(onRefundCreated);
var onRefundUpdated2 = (0, import_event_definition_modules.createEventModule)(onRefundUpdated);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Initiator,
  RejectionReason,
  SortOrder,
  Status,
  WebhookIdentityType,
  createRefund,
  getRefund,
  onRefundCreated,
  onRefundUpdated,
  queryRefunds
});
