// @wix/bex-utils/@wix/ambassador-fedinfra-exportservice-v1-export-async-job/build/cjs/builders.impl.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.aCustomTextField = exports.aMediaItemVideo = exports.aMediaItemType = exports.aMediaItemUrlAndSize = exports.aMediaItem = exports.aMedia = exports.aRibbon = exports.aAdditionalInfoSection = exports.aMeasurementUnit = exports.aPricePerUnitData = exports.aCostAndProfitData = exports.aFormattedPrice = exports.aPriceData = exports.aInventoryStatus = exports.aStock = exports.aNumericPropertyRange = exports.aProductType = exports.aProduct = exports.aProductOrVariant = exports.aQueryVariantsExportSpiResponse = exports.aQueryRequestLoose = exports.aGenerateExportAsyncJobDownloadUrlResponse = exports.aGenerateExportAsyncJobDownloadUrlRequest = exports.aCancelExportAsyncJobResponse = exports.aCancelExportAsyncJobRequest = exports.aGetExportAsyncJobResponse = exports.aGetExportAsyncJobRequest = exports.aCreateExportAsyncJobResponse = exports.aCreateExportAsyncJobRequest = exports.aStatus = exports.aSystemError = exports.aFieldViolation = exports.aRuleType = exports.aValidationError = exports.aApplicationError = exports.aDetails = exports.aCursors = exports.aPagingMetadataV2 = exports.aQueryFieldNumber = exports.aMethodSpec = exports.aMethodMetadata = exports.aArrayFieldDelimiter = exports.aFieldDescriptor = exports.aExportCursorPaging = exports.aPaging = exports.aSortOrder = exports.aSorting = exports.aExportQueryV2 = exports.aExportAsyncJobData = exports.aExportAsyncJob = void 0;
exports.aProductOrVariantV2 = exports.aQueryProductsExportSpiResponse = exports.aPreorderInfo = exports.aVideoResolution = exports.aVideoV2 = exports.aFocalPoint = exports.aImage = exports.aPlatformMedia = exports.aStoreVariant = exports.aFileType = exports.aSecuredMedia = exports.aSettings = exports.aTag = exports.aKeyword = exports.aSeoSchema = exports.aVariantStock = exports.aVariantDataWithNoStock = exports.aVariant = exports.aDiscountType = exports.aDiscount = exports.aPageUrl = exports.aChoice = exports.aOptionType = exports.aProductOption = void 0;
var metro_common_builders_1 = require("@wix/metro-common-builders");
var types_impl_1 = require("./types.impl");
// @ts-ignore
var callsite_1 = __importDefault(require("callsite"));
function aExportAsyncJob(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        id: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'FORMAT', value: 'GUID' },
                { type: 'READ_ONLY', value: 'true' },
            ],
        }),
        data: aExportAsyncJobData(),
        status: aStatus(),
        createDate: (0, metro_common_builders_1.someDate)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        updateDate: (0, metro_common_builders_1.someDate)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
    }, overrideObject);
}
exports.aExportAsyncJob = aExportAsyncJob;
function aExportAsyncJobData(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        query: aExportQueryV2(),
        fields: [aFieldDescriptor()],
        arrayFieldDelimiter: aArrayFieldDelimiter(),
        methodMetadata: aMethodMetadata(),
        methodSpec: aMethodSpec(),
        userId: (0, metro_common_builders_1.someString)(),
        subDir: (0, metro_common_builders_1.someString)(),
        format: (0, metro_common_builders_1.someString)(),
        uploadId: (0, metro_common_builders_1.someString)(),
        saveAs: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '255' }],
        }),
        pagingMetadata: aPagingMetadataV2(),
        iterationsCount: (0, metro_common_builders_1.someNumber)(),
        processedItemsCount: (0, metro_common_builders_1.someNumber)(),
        completionStep: (0, metro_common_builders_1.someString)(),
        resetCount: (0, metro_common_builders_1.someNumber)(),
        signedUrl: (0, metro_common_builders_1.someString)(),
        error: aDetails(),
        testParams: {},
    }, overrideObject);
}
exports.aExportAsyncJobData = aExportAsyncJobData;
function aExportQueryV2(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        paging: aPaging(),
        filter: {},
        sort: [aSorting()],
        fields: [(0, metro_common_builders_1.someString)()],
        fieldsets: [(0, metro_common_builders_1.someString)()],
    }, overrideObject);
}
exports.aExportQueryV2 = aExportQueryV2;
function aSorting(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        fieldName: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '512' }],
        }),
        order: aSortOrder(),
        selectItemsBy: [{}, {}, {}, {}, {}, {}, {}, {}, {}, {}],
    }, overrideObject);
}
exports.aSorting = aSorting;
function aSortOrder(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.SortOrder.ASC; }
    return enumFirstItem;
}
exports.aSortOrder = aSortOrder;
function aPaging(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        limit: (0, metro_common_builders_1.someNumber)({
            validations: [{ type: 'MIN' }],
        }),
        offset: (0, metro_common_builders_1.someNumber)({
            validations: [{ type: 'MIN' }],
        }),
    }, overrideObject);
}
exports.aPaging = aPaging;
function aExportCursorPaging(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        limit: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'MIN' },
                { type: 'MAX', value: '10000' },
            ],
        }),
        cursor: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aExportCursorPaging = aExportCursorPaging;
function aFieldDescriptor(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        id: (0, metro_common_builders_1.someString)(),
        header: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aFieldDescriptor = aFieldDescriptor;
function aArrayFieldDelimiter(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.ArrayFieldDelimiter.SEMICOLON; }
    return enumFirstItem;
}
exports.aArrayFieldDelimiter = aArrayFieldDelimiter;
function aMethodMetadata(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        artifact: (0, metro_common_builders_1.someString)(),
        service: (0, metro_common_builders_1.someString)(),
        method: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aMethodMetadata = aMethodMetadata;
function aMethodSpec(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        requestQueryFieldNumber: aQueryFieldNumber(),
        responseRepeatedFieldName: (0, metro_common_builders_1.someString)(),
        responsePagingMetadataFieldName: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aMethodSpec = aMethodSpec;
function aQueryFieldNumber(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.QueryFieldNumber.DEFAULT; }
    return enumFirstItem;
}
exports.aQueryFieldNumber = aQueryFieldNumber;
function aPagingMetadataV2(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        count: (0, metro_common_builders_1.someNumber)(),
        offset: (0, metro_common_builders_1.someNumber)(),
        total: (0, metro_common_builders_1.someNumber)(),
        tooManyToCount: true,
        cursors: aCursors(),
        hasNext: true,
    }, overrideObject);
}
exports.aPagingMetadataV2 = aPagingMetadataV2;
function aCursors(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        next: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MAX_LENGTH', value: '16000' },
            ],
        }),
        prev: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MAX_LENGTH', value: '16000' },
            ],
        }),
    }, overrideObject);
}
exports.aCursors = aCursors;
function aDetails(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        applicationError: aApplicationError(),
        tracing: { tracing: (0, metro_common_builders_1.someString)() },
    }, overrideObject);
}
exports.aDetails = aDetails;
function aApplicationError(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        code: (0, metro_common_builders_1.someString)(),
        description: (0, metro_common_builders_1.someString)(),
        data: {},
    }, overrideObject);
}
exports.aApplicationError = aApplicationError;
function aValidationError(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        fieldViolations: [aFieldViolation()],
    }, overrideObject);
}
exports.aValidationError = aValidationError;
function aRuleType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.RuleType.VALIDATION; }
    return enumFirstItem;
}
exports.aRuleType = aRuleType;
function aFieldViolation(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        field: (0, metro_common_builders_1.someString)(),
        description: (0, metro_common_builders_1.someString)(),
        violatedRule: aRuleType(),
        ruleName: (0, metro_common_builders_1.someString)(),
        data: {},
    }, overrideObject);
}
exports.aFieldViolation = aFieldViolation;
function aSystemError(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        errorCode: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aSystemError = aSystemError;
function aStatus(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.Status.UNKNOWN; }
    return enumFirstItem;
}
exports.aStatus = aStatus;
function aCreateExportAsyncJobRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        query: aExportQueryV2(),
        fields: [aFieldDescriptor()],
        arrayFieldDelimiter: aArrayFieldDelimiter(),
        methodMetadata: aMethodMetadata(),
        methodSpec: aMethodSpec(),
        saveAs: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '255' }],
        }),
        testParams: {},
    }, overrideObject);
}
exports.aCreateExportAsyncJobRequest = aCreateExportAsyncJobRequest;
function aCreateExportAsyncJobResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        job: aExportAsyncJob(),
    }, overrideObject);
}
exports.aCreateExportAsyncJobResponse = aCreateExportAsyncJobResponse;
function aGetExportAsyncJobRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        jobId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
exports.aGetExportAsyncJobRequest = aGetExportAsyncJobRequest;
function aGetExportAsyncJobResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        job: aExportAsyncJob(),
    }, overrideObject);
}
exports.aGetExportAsyncJobResponse = aGetExportAsyncJobResponse;
function aCancelExportAsyncJobRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        jobId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
exports.aCancelExportAsyncJobRequest = aCancelExportAsyncJobRequest;
function aCancelExportAsyncJobResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        job: aExportAsyncJob(),
    }, overrideObject);
}
exports.aCancelExportAsyncJobResponse = aCancelExportAsyncJobResponse;
function aGenerateExportAsyncJobDownloadUrlRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        jobId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
exports.aGenerateExportAsyncJobDownloadUrlRequest = aGenerateExportAsyncJobDownloadUrlRequest;
function aGenerateExportAsyncJobDownloadUrlResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        job: aExportAsyncJob(),
    }, overrideObject);
}
exports.aGenerateExportAsyncJobDownloadUrlResponse = aGenerateExportAsyncJobDownloadUrlResponse;
function aQueryRequestLoose(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        query: aExportQueryV2(),
    }, overrideObject);
}
exports.aQueryRequestLoose = aQueryRequestLoose;
function aQueryVariantsExportSpiResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        items: [aProductOrVariant()],
        pagingMetadata: aPagingMetadataV2(),
    }, overrideObject);
}
exports.aQueryVariantsExportSpiResponse = aQueryVariantsExportSpiResponse;
function aProductOrVariant(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        kind: (0, metro_common_builders_1.someString)(),
        product: aProduct(),
        variant: aStoreVariant(),
    }, overrideObject);
}
exports.aProductOrVariant = aProductOrVariant;
function aProduct(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        id: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        name: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '80' },
            ],
        }),
        slug: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        visible: true,
        productType: aProductType(),
        description: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '8000' }],
        }),
        sku: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '40' }],
        }),
        weight: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'MIN' },
                { type: 'MAX', value: '999999999.99' },
            ],
        }),
        weightRange: aNumericPropertyRange(),
        stock: aStock(),
        price: aPriceData(),
        priceData: aPriceData(),
        convertedPriceData: aPriceData(),
        priceRange: aNumericPropertyRange(),
        costAndProfitData: aCostAndProfitData(),
        costRange: aNumericPropertyRange(),
        pricePerUnitData: aPricePerUnitData(),
        additionalInfoSections: [aAdditionalInfoSection()],
        ribbons: [aRibbon()],
        media: aMedia(),
        customTextFields: [aCustomTextField()],
        manageVariants: true,
        productOptions: [
            aProductOption(),
            aProductOption(),
            aProductOption(),
            aProductOption(),
            aProductOption(),
            aProductOption(),
        ],
        productPageUrl: aPageUrl(),
        numericId: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'DECIMAL_VALUE' },
            ],
            isUnsigned: true,
        }),
        inventoryItemId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        discount: aDiscount(),
        collectionIds: [
            (0, metro_common_builders_1.someString)({
                validations: [
                    { type: 'READ_ONLY', value: 'true' },
                ],
            }),
        ],
        variants: [
            aVariant(),
            aVariant(),
            aVariant(),
            aVariant(),
            aVariant(),
            aVariant(),
            aVariant(),
            aVariant(),
            aVariant(),
            aVariant(),
        ],
        lastUpdated: (0, metro_common_builders_1.someDate)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        createdDate: (0, metro_common_builders_1.someDate)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        seoData: aSeoSchema(),
        ribbon: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '30' }],
        }),
        brand: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '50' },
            ],
        }),
        taxGroupId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        exportProductId: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MAX_LENGTH', value: '100' },
            ],
        }),
        digitalFile: aSecuredMedia(),
    }, overrideObject);
}
exports.aProduct = aProduct;
function aProductType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.ProductType.unspecified_product_type; }
    return enumFirstItem;
}
exports.aProductType = aProductType;
function aNumericPropertyRange(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        minValue: (0, metro_common_builders_1.someNumber)(),
        maxValue: (0, metro_common_builders_1.someNumber)(),
    }, overrideObject);
}
exports.aNumericPropertyRange = aNumericPropertyRange;
function aStock(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        trackInventory: true,
        quantity: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'MIN' },
                { type: 'MAX', value: '99999' },
            ],
        }),
        inStock: true,
        inventoryStatus: aInventoryStatus(),
    }, overrideObject);
}
exports.aStock = aStock;
function aInventoryStatus(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.InventoryStatus.IN_STOCK; }
    return enumFirstItem;
}
exports.aInventoryStatus = aInventoryStatus;
function aPriceData(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        currency: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        price: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'MIN' },
                { type: 'MAX', value: '999999999.99' },
            ],
        }),
        discountedPrice: (0, metro_common_builders_1.someNumber)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        formatted: aFormattedPrice(),
        pricePerUnit: (0, metro_common_builders_1.someNumber)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
    }, overrideObject);
}
exports.aPriceData = aPriceData;
function aFormattedPrice(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        price: (0, metro_common_builders_1.someString)(),
        discountedPrice: (0, metro_common_builders_1.someString)(),
        pricePerUnit: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
    }, overrideObject);
}
exports.aFormattedPrice = aFormattedPrice;
function aCostAndProfitData(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        itemCost: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'MIN' },
                { type: 'MAX', value: '999999999.99' },
            ],
        }),
        formattedItemCost: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        profit: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MIN' },
                { type: 'MAX', value: '999999999.99' },
            ],
        }),
        formattedProfit: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        profitMargin: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MIN' },
                { type: 'MAX', value: '1' },
            ],
        }),
    }, overrideObject);
}
exports.aCostAndProfitData = aCostAndProfitData;
function aPricePerUnitData(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        totalQuantity: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'MIN', value: '0.01' },
                { type: 'MAX', value: '999999999.99' },
            ],
        }),
        totalMeasurementUnit: aMeasurementUnit(),
        baseQuantity: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'MIN', value: '0.01' },
                { type: 'MAX', value: '999999999.99' },
            ],
        }),
        baseMeasurementUnit: aMeasurementUnit(),
    }, overrideObject);
}
exports.aPricePerUnitData = aPricePerUnitData;
function aMeasurementUnit(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.MeasurementUnit.UNSPECIFIED; }
    return enumFirstItem;
}
exports.aMeasurementUnit = aMeasurementUnit;
function aAdditionalInfoSection(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        title: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '50' }],
        }),
        description: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MAX_LENGTH', value: '16000' },
            ],
        }),
    }, overrideObject);
}
exports.aAdditionalInfoSection = aAdditionalInfoSection;
function aRibbon(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        text: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '30' }],
        }),
    }, overrideObject);
}
exports.aRibbon = aRibbon;
function aMedia(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        mainMedia: aMediaItem(),
        items: [aMediaItem()],
    }, overrideObject);
}
exports.aMedia = aMedia;
function aMediaItem(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        image: aMediaItemUrlAndSize(),
        thumbnail: aMediaItemUrlAndSize(),
        mediaType: aMediaItemType(),
        title: (0, metro_common_builders_1.someString)(),
        id: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aMediaItem = aMediaItem;
function aMediaItemUrlAndSize(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        url: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'WEB_URL' }],
        }),
        width: (0, metro_common_builders_1.someNumber)({
            validations: [{ type: 'MIN' }],
        }),
        height: (0, metro_common_builders_1.someNumber)({
            validations: [{ type: 'MIN' }],
        }),
        format: (0, metro_common_builders_1.someString)(),
        altText: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aMediaItemUrlAndSize = aMediaItemUrlAndSize;
function aMediaItemType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.MediaItemType.unspecified_media_item_type; }
    return enumFirstItem;
}
exports.aMediaItemType = aMediaItemType;
function aMediaItemVideo(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        files: [aMediaItemUrlAndSize()],
        stillFrameMediaId: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aMediaItemVideo = aMediaItemVideo;
function aCustomTextField(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        title: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        maxLength: (0, metro_common_builders_1.someNumber)({
            validations: [{ type: 'MAX', value: '500' }],
        }),
        mandatory: true,
    }, overrideObject);
}
exports.aCustomTextField = aCustomTextField;
function aProductOption(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        optionType: aOptionType(),
        name: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '50' },
            ],
        }),
        choices: [aChoice()],
    }, overrideObject);
}
exports.aProductOption = aProductOption;
function aOptionType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.OptionType.unspecified_option_type; }
    return enumFirstItem;
}
exports.aOptionType = aOptionType;
function aChoice(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        value: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '50' },
            ],
        }),
        description: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '50' },
            ],
        }),
        media: aMedia(),
        inStock: true,
        visible: true,
    }, overrideObject);
}
exports.aChoice = aChoice;
function aPageUrl(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        base: (0, metro_common_builders_1.someString)(),
        path: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aPageUrl = aPageUrl;
function aDiscount(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        type: aDiscountType(),
        value: (0, metro_common_builders_1.someNumber)(),
    }, overrideObject);
}
exports.aDiscount = aDiscount;
function aDiscountType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.DiscountType.UNDEFINED; }
    return enumFirstItem;
}
exports.aDiscountType = aDiscountType;
function aVariant(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        id: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        choices: { choices: (0, metro_common_builders_1.someString)() },
        variant: aVariantDataWithNoStock(),
        stock: aVariantStock(),
    }, overrideObject);
}
exports.aVariant = aVariant;
function aVariantDataWithNoStock(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        priceData: aPriceData(),
        convertedPriceData: aPriceData(),
        costAndProfitData: aCostAndProfitData(),
        weight: (0, metro_common_builders_1.someNumber)(),
        sku: (0, metro_common_builders_1.someString)(),
        visible: true,
    }, overrideObject);
}
exports.aVariantDataWithNoStock = aVariantDataWithNoStock;
function aVariantStock(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        trackQuantity: true,
        quantity: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'MIN' },
                { type: 'MAX', value: '99999' },
            ],
        }),
        inStock: true,
        locationId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
exports.aVariantStock = aVariantStock;
function aSeoSchema(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        tags: [aTag()],
        settings: aSettings(),
    }, overrideObject);
}
exports.aSeoSchema = aSeoSchema;
function aKeyword(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        term: (0, metro_common_builders_1.someString)(),
        isMain: true,
        origin: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '1000' }],
        }),
    }, overrideObject);
}
exports.aKeyword = aKeyword;
function aTag(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        type: (0, metro_common_builders_1.someString)(),
        props: {},
        meta: {},
        children: (0, metro_common_builders_1.someString)(),
        custom: true,
        disabled: true,
    }, overrideObject);
}
exports.aTag = aTag;
function aSettings(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        preventAutoRedirect: true,
        keywords: [aKeyword(), aKeyword(), aKeyword(), aKeyword(), aKeyword()],
    }, overrideObject);
}
exports.aSettings = aSettings;
function aSecuredMedia(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        id: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '100' },
            ],
        }),
        fileName: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '1000' },
            ],
        }),
        fileType: aFileType(),
    }, overrideObject);
}
exports.aSecuredMedia = aSecuredMedia;
function aFileType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.FileType.UNSPECIFIED; }
    return enumFirstItem;
}
exports.aFileType = aFileType;
function aStoreVariant(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        id: (0, metro_common_builders_1.someString)(),
        variantId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        productId: (0, metro_common_builders_1.someString)(),
        variantName: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '80' },
            ],
        }),
        productName: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '80' },
            ],
        }),
        managedVariant: true,
        sku: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '40' }],
        }),
        stock: aVariantStock(),
        choices: { choices: (0, metro_common_builders_1.someString)() },
        collectionIds: [
            (0, metro_common_builders_1.someString)({
                validations: [{ type: 'FORMAT', value: 'GUID' }],
            }),
        ],
        media: aPlatformMedia(),
        customFields: { customFields: (0, metro_common_builders_1.someAny)() },
        preorderInfo: aPreorderInfo(),
    }, overrideObject);
}
exports.aStoreVariant = aStoreVariant;
function aPlatformMedia(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        image: aImage(),
    }, overrideObject);
}
exports.aPlatformMedia = aPlatformMedia;
function aImage(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        id: (0, metro_common_builders_1.someString)(),
        url: (0, metro_common_builders_1.someString)(),
        height: (0, metro_common_builders_1.someNumber)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        width: (0, metro_common_builders_1.someNumber)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        altText: (0, metro_common_builders_1.someString)(),
        urlExpirationDate: (0, metro_common_builders_1.someDate)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        filename: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        sizeInBytes: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
            ],
        }),
        focalPoint: aFocalPoint(),
    }, overrideObject);
}
exports.aImage = aImage;
function aFocalPoint(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        x: (0, metro_common_builders_1.someNumber)(),
        y: (0, metro_common_builders_1.someNumber)(),
        height: (0, metro_common_builders_1.someNumber)(),
        width: (0, metro_common_builders_1.someNumber)(),
    }, overrideObject);
}
exports.aFocalPoint = aFocalPoint;
function aVideoV2(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        id: (0, metro_common_builders_1.someString)(),
        url: (0, metro_common_builders_1.someString)(),
        resolutions: [
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
        ],
        filename: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        posters: [aImage()],
        sizeInBytes: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
            ],
        }),
        urlExpirationDate: (0, metro_common_builders_1.someDate)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        durationInMilliseconds: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MIN' },
            ],
        }),
        title: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '300' }],
        }),
        description: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '2000' }],
        }),
        height: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MIN' },
            ],
        }),
        width: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MIN' },
            ],
        }),
    }, overrideObject);
}
exports.aVideoV2 = aVideoV2;
function aVideoResolution(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        url: (0, metro_common_builders_1.someString)(),
        height: (0, metro_common_builders_1.someNumber)(),
        width: (0, metro_common_builders_1.someNumber)(),
        poster: aImage(),
        format: (0, metro_common_builders_1.someString)(),
        urlExpirationDate: (0, metro_common_builders_1.someDate)(),
        sizeInBytes: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
            ],
        }),
        quality: (0, metro_common_builders_1.someString)(),
        filename: (0, metro_common_builders_1.someString)(),
        durationInSeconds: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MIN' },
            ],
        }),
        durationInMilliseconds: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MIN' },
            ],
        }),
        private: true,
        assetKey: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
    }, overrideObject);
}
exports.aVideoResolution = aVideoResolution;
function aPreorderInfo(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        enabled: true,
        message: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '250' },
            ],
        }),
        limit: (0, metro_common_builders_1.someNumber)({
            validations: [
                { type: 'MIN' },
                { type: 'MAX', value: '100000' },
            ],
        }),
    }, overrideObject);
}
exports.aPreorderInfo = aPreorderInfo;
function aQueryProductsExportSpiResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        items: [aProductOrVariantV2()],
        pagingMetadata: aPagingMetadataV2(),
    }, overrideObject);
}
exports.aQueryProductsExportSpiResponse = aQueryProductsExportSpiResponse;
function aProductOrVariantV2(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        kind: (0, metro_common_builders_1.someString)(),
        product: aProduct(),
        variant: aVariant(),
    }, overrideObject);
}
exports.aProductOrVariantV2 = aProductOrVariantV2;
function isCurrentBuilderWasPreviouslyCalled() {
    var callChainFunctionNames = (0, callsite_1.default)().map(function (item) {
        return item.getFunctionName();
    });
    var callChainWithoutCurrentBuilder = callChainFunctionNames.splice(2);
    var currentBuilderName = callChainFunctionNames[1];
    return callChainWithoutCurrentBuilder.some(function (functionName) { return functionName === currentBuilderName; });
}
__exportStar(require("@wix/metro-common-builders"), exports);
