// @wix/bex-utils/@wix/ambassador-fedinfra-exportservice-v1-export-async-job/build/cjs/meta.impl.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.generateExportAsyncJobDownloadUrl = exports.cancelExportAsyncJob = exports.getExportAsyncJob = exports.createExportAsyncJob = void 0;
var httpWixFedinfraExportserviceV1ExportAsyncJob = __importStar(require("./http.impl"));
function createExportAsyncJob() {
    var payload = {};
    var getRequestOptions = httpWixFedinfraExportserviceV1ExportAsyncJob.createExportAsyncJob(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/export-async-job',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createExportAsyncJob = createExportAsyncJob;
function getExportAsyncJob() {
    var payload = { jobId: ':jobId' };
    var getRequestOptions = httpWixFedinfraExportserviceV1ExportAsyncJob.getExportAsyncJob(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/v1/export-async-job/{jobId}',
        pathParams: { jobId: 'jobId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getExportAsyncJob = getExportAsyncJob;
function cancelExportAsyncJob() {
    var payload = { jobId: ':jobId' };
    var getRequestOptions = httpWixFedinfraExportserviceV1ExportAsyncJob.cancelExportAsyncJob(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'PUT',
        path: '/v1/export-async-job/{jobId}/cancel',
        pathParams: { jobId: 'jobId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.cancelExportAsyncJob = cancelExportAsyncJob;
function generateExportAsyncJobDownloadUrl() {
    var payload = { jobId: ':jobId' };
    var getRequestOptions = httpWixFedinfraExportserviceV1ExportAsyncJob.generateExportAsyncJobDownloadUrl(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'PUT',
        path: '/v1/export-async-job/{jobId}/generate-download-url',
        pathParams: { jobId: 'jobId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.generateExportAsyncJobDownloadUrl = generateExportAsyncJobDownloadUrl;
