// @wix/headless-restaurants-menus/cjs/dist/react/Menus.js
import { jsx as _jsx, Fragment as _Fragment, jsxs as _jsxs } from "react/jsx-runtime";
import React from 'react';
import { Loading as CoreLoading, ErrorState as CoreErrorState, Menus, MenuSelector as CoreMenuSelector, LocationSelector as CoreLocationSelector, } from './core/index.js';
import { AsChildSlot } from '@wix/headless-utils/react';
import { TestIds } from './TestIds.js';
import * as Tabs from '@radix-ui/react-tabs';
import * as Select from '@radix-ui/react-select';
import { useService } from '@wix/services-manager-react';
import { MenusServiceDefinition, } from '../services/index.js';
import * as MenuComponent from './Menu.js';
export function Root(props) {
    return (_jsx(Menus, { config: props.config, children: _jsx("div", { "data-component-tag": "menus.root", children: props.children }) }));
}
/**
 * Menu selector component that provides menu selection functionality using tabs.
 * Uses the core MenuSelector component and wraps it with AsChildSlot for flexible rendering.
 * Only renders when there are more than 1 real menu available (excluding "All" option).
 *
 * @component
 * @example
 * ```tsx
 * // Default usage with custom styling (only shows if more than 1 real menu)
 * <Menus.MenuSelector
 *   className="w-full"
 *   listClassName="flex space-x-1"
 *   triggerClassName="px-4 py-2 rounded-md hover:bg-gray-100 data-[state=active]:bg-blue-500 data-[state=active]:text-white"
 * />
 *
 * // Conditional rendering based on menu count
 * {menus.filter(menu => menu._id !== 'all').length > 1 && (
 *   <Menus.MenuSelector className="mb-4" />
 * )}
 *
 * // asChild with custom tabs implementation
 * <Menus.MenuSelector asChild>
 *   {React.forwardRef(({menus, selectedMenu, onMenuSelect, ...props}, ref) => (
 *     <Tabs.Root ref={ref} {...props} value={selectedMenu?._id || ''} onValueChange={(value) => {
 *       const menu = menus.find(m => m._id === value);
 *       if (menu) onMenuSelect(menu);
 *     }}>
 *       <Tabs.List className="flex space-x-1">
 *         {menus.map(menu => (
 *           <Tabs.Trigger key={menu._id} value={menu._id} className="px-4 py-2">
 *             {menu.name}
 *           </Tabs.Trigger>
 *         ))}
 *       </Tabs.List>
 *     </Tabs.Root>
 *   ))}
 * </Menus.MenuSelector>
 *
 * // Custom render function
 * <Menus.MenuSelector>
 *   {({ menus, selectedMenu, onMenuSelect }) => (
 *     <Tabs.Root value={selectedMenu?._id || ''} onValueChange={(value) => {
 *       const menu = menus.find(m => m._id === value);
 *       if (menu) onMenuSelect(menu);
 *     }}>
 *       <Tabs.List className="flex space-x-1">
 *         {menus.map(menu => (
 *           <Tabs.Trigger key={menu._id} value={menu._id} className="px-4 py-2">
 *             {menu.name}
 *           </Tabs.Trigger>
 *         ))}
 *       </Tabs.List>
 *     </Tabs.Root>
 *   )}
 * </Menus.MenuSelector>
 * ```
 */
export const MenuSelector = React.forwardRef((props, ref) => {
    const { asChild, children, className, listClassName, triggerClassName, allText = 'All', showAll = true, ...otherProps } = props;
    return (_jsx(CoreMenuSelector, { allText: allText, showAll: showAll, children: ({ menus, selectedMenu, onMenuSelect, }) => {
            // Only show selector if there are more than 1 real menu (excluding "All" option)
            const realMenus = menus.filter((menu) => menu._id !== 'all');
            if (realMenus.length <= 1) {
                return null;
            }
            if (asChild && children && typeof children === 'function') {
                return children({ menus, selectedMenu, onMenuSelect }, ref);
            }
            return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, "data-testid": TestIds.menuSelector, ...otherProps, children: _jsx(Tabs.Root, { className: className, value: selectedMenu?._id || 'all', onValueChange: (value) => {
                        const menu = menus.find((m) => m._id === value);
                        if (menu)
                            onMenuSelect(menu);
                    }, children: _jsx(Tabs.List, { className: listClassName, children: menus.map((menu) => (_jsx(Tabs.Trigger, { value: menu._id || '', className: triggerClassName, children: menu.name }, menu._id))) }) }) }));
        } }));
});
MenuSelector.displayName = 'Menus.MenuSelector';
export const MenusRepeater = React.forwardRef((props, _ref) => {
    const { children } = props;
    const menusService = useService(MenusServiceDefinition);
    const filteredMenus = menusService.filteredMenus.get();
    const selectedMenu = menusService.selectedMenu.get();
    const hasMenus = filteredMenus.length > 0;
    if (!hasMenus)
        return null;
    const menusToRender = selectedMenu ? [selectedMenu] : filteredMenus;
    return (_jsx(_Fragment, { children: menusToRender.map((menu) => (_jsx(MenuComponent.Root, { menu: menu, "data-testid": TestIds.itemName, "data-menu-id": menu._id, children: children }, menu._id))) }));
});
MenusRepeater.displayName = 'Menus.Repeater';
/**
 * Displays loading state with customizable rendering following the documented API.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Menus.Loading className="text-center" />
 *
 * // asChild with primitive
 * <Menus.Loading asChild>
 *   <p className="text-center" />
 * </Menus.Loading>
 *
 * // asChild with react component
 * <Menus.Loading asChild>
 *   {React.forwardRef(({loading, ...props}, ref) => (
 *     <p ref={ref} {...props} className="text-center">
 *       {loading ? "Loading..." : null}
 *     </p>
 *   ))}
 * </Menus.Loading>
 *
 * // With custom loading text
 * <Menus.Loading loadingText="Please wait..." className="text-center" />
 * ```
 */
export const Loading = React.forwardRef((props, ref) => {
    const { asChild, children, className, loadingText, ...otherProps } = props;
    return (_jsx(CoreLoading, { children: ({ loading }) => {
            const displayContent = loading && loadingText ? loadingText : null;
            return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.menusLoading, customElement: children, customElementProps: { loading }, content: displayContent, ...otherProps, children: _jsx("p", { children: displayContent }) }));
        } }));
});
/**
 * Displays error state with customizable rendering following the documented API.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Menus.Error className="text-red-600" />
 *
 * // asChild with primitive
 * <Menus.Error asChild>
 *   <p className="text-red-600" />
 * </Menus.Error>
 *
 * // asChild with react component
 * <Menus.Error asChild>
 *   {React.forwardRef(({error, ...props}, ref) => (
 *     <p ref={ref} {...props} className="text-red-600">
 *       {error ? `Error: ${error}` : null}
 *     </p>
 *   ))}
 * </Menus.Error>
 *
 * // With custom error prefix
 * <Menus.Error errorPrefix="Error: " className="text-red-600" />
 *
 * // Without prefix (only error message)
 * <Menus.Error className="text-red-600" />
 * ```
 */
export const Error = React.forwardRef((props, ref) => {
    const { asChild, children, className, errorPrefix, ...otherProps } = props;
    return (_jsx(CoreErrorState, { children: ({ error }) => {
            const displayContent = error
                ? errorPrefix
                    ? `${errorPrefix}${error}`
                    : error
                : null;
            return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.menusError, customElement: children, customElementProps: { error }, content: displayContent, ...otherProps, children: _jsx("p", { children: displayContent }) }));
        } }));
});
Loading.displayName = 'Menus.Loading';
Error.displayName = 'Menus.Error';
/**
 * Location selector component that provides location selection functionality using a dropdown.
 * Uses the core LocationSelector component and wraps it with AsChildSlot for flexible rendering.
 * Only renders when there are more than 1 real location available (excluding "All" option).
 *
 * @component
 * @example
 * ```tsx
 * // Default usage with custom styling (only shows if more than 1 real location)
 * <Menus.LocationSelector
 *   className="w-full"
 *   triggerClassName="w-full p-2 border rounded-md bg-white"
 *   contentClassName="bg-white border rounded-md shadow-lg"
 *   viewportClassName="p-1"
 *   optionClassName="p-2 hover:bg-gray-100 cursor-pointer"
 *   scrollUpButtonClassName="p-2 text-gray-500"
 *   scrollDownButtonClassName="p-2 text-gray-500"
 *   placeholder="Choose your location"
 * />
 *
 * // Conditional rendering based on location count
 * {locations.filter(location => location.id !== 'all').length > 1 && (
 *   <Menus.LocationSelector
 *     className="mb-4"
 *     triggerClassName="w-full p-3 border-2 border-blue-500 rounded-lg"
 *     contentClassName="bg-blue-50 border-2 border-blue-500 rounded-lg"
 *     placeholder="Pick a location"
 *   />
 * )}
 *
 * // asChild with custom Radix Select implementation
 * <Menus.LocationSelector asChild>
 *   {React.forwardRef(({locations, selectedLocation, onLocationSelect, ...props}, ref) => (
 *     <Select.Root
 *       ref={ref}
 *       {...props}
 *       value={selectedLocation || ''}
 *       onValueChange={onLocationSelect}
 *     >
 *       <Select.Trigger>
 *         <Select.Value placeholder="Select a location" />
 *         <Select.Icon />
 *       </Select.Trigger>
 *       <Select.Portal>
 *         <Select.Content>
 *           <Select.Viewport>
 *             {locations.map(location => (
 *               <Select.Item key={location.id} value={location.id}>
 *                 <Select.ItemText>{location.name}</Select.ItemText>
 *               </Select.Item>
 *             ))}
 *           </Select.Viewport>
 *         </Select.Content>
 *       </Select.Portal>
 *     </Select.Root>
 *   ))}
 * </Menus.LocationSelector>
 *
 * // Custom render function
 * <Menus.LocationSelector>
 *   {({ locations, selectedLocation, onLocationSelect }) => (
 *     <Select.Root
 *       value={selectedLocation || ''}
 *       onValueChange={onLocationSelect}
 *     >
 *       <Select.Trigger>
 *         <Select.Value placeholder="Select a location" />
 *         <Select.Icon />
 *       </Select.Trigger>
 *       <Select.Portal>
 *         <Select.Content>
 *           <Select.Viewport>
 *             {locations.map(location => (
 *               <Select.Item key={location.id} value={location.id}>
 *                 <Select.ItemText>{location.name}</Select.ItemText>
 *               </Select.Item>
 *             ))}
 *           </Select.Viewport>
 *         </Select.Content>
 *       </Select.Portal>
 *     </Select.Root>
 *   )}
 * </Menus.LocationSelector>
 * ```
 */
export const LocationSelector = React.forwardRef((props, ref) => {
    const { asChild, children, className, triggerClassName, contentClassName, viewportClassName, optionClassName, scrollUpButtonClassName, scrollDownButtonClassName, placeholder = 'Select a location', allText = 'All', showAll = true, ...otherProps } = props;
    return (_jsx(CoreLocationSelector, { allText: allText, showAll: showAll, children: ({ locations, selectedLocation, onLocationSelect, }) => {
            // Only show selector if there are more than 1 real location (excluding "All" option)
            const realLocations = locations.filter((location) => location.id !== 'all');
            if (realLocations.length <= 1) {
                return null;
            }
            if (asChild && children && typeof children === 'function') {
                return children({ locations, selectedLocation, onLocationSelect }, ref);
            }
            return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.locationSelector, ...otherProps, children: _jsxs(Select.Root, { value: selectedLocation || 'all', onValueChange: onLocationSelect, children: [_jsxs(Select.Trigger, { className: triggerClassName, children: [_jsx(Select.Value, { placeholder: placeholder }), _jsx(Select.Icon, {})] }), _jsx(Select.Portal, { children: _jsxs(Select.Content, { className: contentClassName, children: [_jsx(Select.ScrollUpButton, { className: scrollUpButtonClassName }), _jsx(Select.Viewport, { className: viewportClassName, children: locations.map((location) => (_jsx(Select.Item, { value: location.id, className: optionClassName, children: _jsx(Select.ItemText, { children: location.name }) }, location.id))) }), _jsx(Select.ScrollDownButton, { className: scrollDownButtonClassName })] }) })] }) }));
        } }));
});
LocationSelector.displayName = 'Menus.LocationSelector';
/**
 * Menus namespace containing all menus components
 * following the compound component pattern: MenusComponent.Root, MenusComponent.MenusRepeater, MenusComponent.Loading, etc.
 */
export const MenusComponent = {
    /** Menus root component */
    Root,
    /** Menu selector component */
    MenuSelector,
    /** Location selector component */
    LocationSelector,
    /** Menus repeater component */
    MenusRepeater,
    /** Menus loading component */
    Loading,
    /** Menus error component */
    Error,
};
