// @wix/headless-events/dist/services/checkout-service.js
import { defineService, implementService } from '@wix/services-definitions';
import { SignalsServiceDefinition, } from '@wix/services-definitions/core-services/signals';
import { ticketReservations } from '@wix/events';
import { redirects } from '@wix/redirects';
import { getErrorMessage } from '../utils/errors.js';
export const CheckoutServiceDefinition = defineService('checkout');
export const CheckoutService = implementService.withConfig()(CheckoutServiceDefinition, ({ getService, config }) => {
    const signalsService = getService(SignalsServiceDefinition);
    const isLoading = signalsService.signal(false);
    const error = signalsService.signal(null);
    const checkout = async (eventId, eventSlug, ticketReservationQuantities) => {
        if (!ticketReservationQuantities.length) {
            error.set(config.noTicketDefinitionsSelectedError || 'Select a ticket');
            return;
        }
        try {
            isLoading.set(true);
            error.set(null);
            const { _id: reservationId } = await ticketReservations.createTicketReservation({
                tickets: ticketReservationQuantities
                    .map(({ quantity, ticketDefinitionId, priceOverride, pricingOptionId, }) => {
                    if (quantity) {
                        return {
                            eventId,
                            quantity,
                            ticketDefinitionId,
                            ticketInfo: priceOverride || pricingOptionId
                                ? {
                                    pricingOptionId,
                                    guestPrice: priceOverride,
                                }
                                : undefined,
                        };
                    }
                    return null;
                })
                    .filter(Boolean),
            });
            if (!reservationId) {
                throw new Error('Failed to create reservation');
            }
            const { redirectSession } = await redirects.createRedirectSession({
                eventsCheckout: {
                    reservationId,
                    eventSlug,
                },
                callbacks: {
                    thankYouPageUrl: config.thankYouPageUrl || window.location.href,
                },
            });
            if (redirectSession?.fullUrl) {
                window.location.href = redirectSession.fullUrl;
            }
            else {
                throw new Error('Failed to create redirect session');
            }
        }
        catch (err) {
            error.set(getErrorMessage(err));
        }
        finally {
            isLoading.set(false);
        }
    };
    return {
        isLoading,
        error,
        checkout,
    };
});
