// @wix/headless-events/dist/services/event-list-service.js
import { defineService, implementService } from '@wix/services-definitions';
import { SignalsServiceDefinition, } from '@wix/services-definitions/core-services/signals';
import { wixEventsV2, categories } from '@wix/events';
import { getErrorMessage } from '../utils/errors.js';
import { ALL_EVENTS, PAST_EVENTS, UPCOMING_EVENTS } from '../constants.js';
export const EventListServiceDefinition = defineService('eventList');
export const EventListService = implementService.withConfig()(EventListServiceDefinition, ({ getService, config }) => {
    const signalsService = getService(SignalsServiceDefinition);
    const events = signalsService.signal(config.events);
    const categories = signalsService.signal(config.categories);
    const selectedCategoryId = signalsService.signal(null);
    const selectedStatusId = signalsService.signal(UPCOMING_EVENTS);
    const isLoading = signalsService.signal(false);
    const isLoadingMore = signalsService.signal(false);
    const error = signalsService.signal(null);
    const pageSize = signalsService.signal(config.pageSize);
    const currentPage = signalsService.signal(config.currentPage);
    const totalPages = signalsService.signal(config.totalPages);
    const hasMoreEvents = signalsService.computed(() => currentPage.get() + 1 < totalPages.get());
    const loadEvents = async ({ categoryId = selectedCategoryId.get(), statusId = selectedStatusId.get(), }) => {
        const status = getStatusFromId(statusId);
        selectedCategoryId.set(categoryId);
        selectedStatusId.set(statusId);
        isLoading.set(true);
        error.set(null);
        try {
            const response = await queryEvents({ categoryId, status });
            events.set(response.items);
            pageSize.set(response.pageSize);
            currentPage.set(response.currentPage);
            totalPages.set(response.totalPages);
        }
        catch (err) {
            error.set(getErrorMessage(err));
        }
        finally {
            isLoading.set(false);
        }
    };
    const loadMoreEvents = async () => {
        isLoadingMore.set(true);
        error.set(null);
        try {
            const offset = pageSize.get() * (currentPage.get() + 1);
            const status = getStatusFromId(selectedStatusId.get());
            const response = await queryEvents({
                offset,
                categoryId: selectedCategoryId.get(),
                status,
            });
            events.set([...events.get(), ...response.items]);
            pageSize.set(response.pageSize);
            currentPage.set(response.currentPage);
            totalPages.set(response.totalPages);
        }
        catch (err) {
            error.set(getErrorMessage(err));
        }
        finally {
            isLoadingMore.set(false);
        }
    };
    return {
        events,
        categories,
        selectedCategoryId,
        selectedStatusId,
        isLoading,
        isLoadingMore,
        error,
        pageSize,
        currentPage,
        totalPages,
        hasMoreEvents,
        loadEvents,
        loadMoreEvents,
    };
});
export async function loadEventListServiceConfig() {
    const [queryEventsResponse, queryCategoriesResponse] = await Promise.all([
        queryEvents(),
        queryCategories(),
    ]);
    return {
        events: queryEventsResponse.items,
        categories: queryCategoriesResponse.items ?? [],
        pageSize: queryEventsResponse.pageSize,
        currentPage: queryEventsResponse.currentPage,
        totalPages: queryEventsResponse.totalPages,
    };
}
export async function queryEvents({ offset = 0, limit = 20, categoryId, status = [wixEventsV2.Status.UPCOMING, wixEventsV2.Status.STARTED], } = {}) {
    let eventsQuery = wixEventsV2
        .queryEvents({
        fields: [
            wixEventsV2.RequestedFields.DETAILS,
            wixEventsV2.RequestedFields.REGISTRATION,
        ],
    })
        .ascending('dateAndTimeSettings.startDate')
        .descending('_createdDate')
        .in('status', status)
        .limit(limit)
        .skip(offset);
    if (categoryId) {
        // @ts-expect-error
        eventsQuery = eventsQuery.in('categories._id', [categoryId]);
    }
    const response = await eventsQuery.find();
    return {
        items: response.items ?? [],
        pageSize: response.pageSize,
        currentPage: response.currentPage ?? 0,
        totalPages: response.totalPages ?? 0,
    };
}
function queryCategories() {
    return categories
        .queryCategories()
        .hasSome('states', [categories.State.MANUAL])
        .limit(100)
        .find();
}
function getStatusFromId(statusId) {
    switch (statusId) {
        case PAST_EVENTS:
            return [wixEventsV2.Status.ENDED];
        case ALL_EVENTS:
            return [
                wixEventsV2.Status.UPCOMING,
                wixEventsV2.Status.STARTED,
                wixEventsV2.Status.ENDED,
            ];
        case UPCOMING_EVENTS:
        default:
            return [wixEventsV2.Status.UPCOMING, wixEventsV2.Status.STARTED];
    }
}
