// @wix/auto_sdk_members_badges/build/internal/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  SortOrder: () => SortOrder,
  WebhookIdentityType: () => WebhookIdentityType,
  assignBadge: () => assignBadge2,
  createBadge: () => createBadge2,
  deleteBadge: () => deleteBadge2,
  getBadge: () => getBadge2,
  getMemberCountsPerBadge: () => getMemberCountsPerBadge2,
  listBadges: () => listBadges2,
  listBadgesPerMember: () => listBadgesPerMember,
  listMembersByBadge: () => listMembersByBadge,
  unassignBadge: () => unassignBadge2,
  updateBadge: () => updateBadge2,
  updateBadgesDisplayOrder: () => updateBadgesDisplayOrder2
});
module.exports = __toCommonJS(index_typings_exports);

// src/badges-v3-badge-badges.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/badges-v3-badge-badges.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressBadgesV3BadgesServiceV3Url(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/members/v3/badges",
        destPath: "/v3/badges"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/members-badges-server/v3",
        destPath: "/v3"
      },
      {
        srcPath: "/_api/members/v3/badges",
        destPath: "/v3/badges"
      },
      {
        srcPath: "/_api/members-badges-server-test/v3",
        destPath: "/v3"
      },
      {
        srcPath: "/_api/members-badges-server-test/v3/badges/v1/items",
        destPath: "/v3/badges"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/members/v3/badges",
        destPath: "/v3/badges"
      }
    ],
    _: [
      {
        srcPath: "/_api/members/v3/badges",
        destPath: "/v3/badges"
      },
      {
        srcPath: "/badges/v3/badges",
        destPath: "/v3/badges"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/members/v3/badges",
        destPath: "/v3/badges"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/badges/v3/badges",
        destPath: "/v3/badges"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_badges";
function createBadge(payload) {
  function __createBadge({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "badge.createdDate" }, { path: "badge.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.badges.v3.badge",
      method: "POST",
      methodFqn: "com.wixpress.badges.v3.BadgesServiceV3.CreateBadge",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressBadgesV3BadgesServiceV3Url({
        protoPath: "/v3/badges",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "badge.createdDate" },
            { path: "badge.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createBadge;
}
function updateBadge(payload) {
  function __updateBadge({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "badge.createdDate" }, { path: "badge.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.badges.v3.badge",
      method: "PATCH",
      methodFqn: "com.wixpress.badges.v3.BadgesServiceV3.UpdateBadge",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressBadgesV3BadgesServiceV3Url({
        protoPath: "/v3/badges/{badge.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "badge.createdDate" },
            { path: "badge.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateBadge;
}
function listBadges(payload) {
  function __listBadges({ host }) {
    const metadata = {
      entityFqdn: "wix.badges.v3.badge",
      method: "GET",
      methodFqn: "com.wixpress.badges.v3.BadgesServiceV3.ListBadges",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressBadgesV3BadgesServiceV3Url({
        protoPath: "/v3/badges",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "badges.createdDate" },
            { path: "badges.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listBadges;
}
function getBadge(payload) {
  function __getBadge({ host }) {
    const metadata = {
      entityFqdn: "wix.badges.v3.badge",
      method: "GET",
      methodFqn: "com.wixpress.badges.v3.BadgesServiceV3.GetBadge",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressBadgesV3BadgesServiceV3Url({
        protoPath: "/v3/badges/{id}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "badge.createdDate" },
            { path: "badge.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getBadge;
}
function deleteBadge(payload) {
  function __deleteBadge({ host }) {
    const metadata = {
      entityFqdn: "wix.badges.v3.badge",
      method: "DELETE",
      methodFqn: "com.wixpress.badges.v3.BadgesServiceV3.DeleteBadge",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressBadgesV3BadgesServiceV3Url({
        protoPath: "/v3/badges/{id}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteBadge;
}
function assignBadge(payload) {
  function __assignBadge({ host }) {
    const metadata = {
      entityFqdn: "wix.badges.v3.badge",
      method: "POST",
      methodFqn: "com.wixpress.badges.v3.BadgesServiceV3.AssignBadge",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressBadgesV3BadgesServiceV3Url({
        protoPath: "/v3/badges/{id}/members",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __assignBadge;
}
function unassignBadge(payload) {
  function __unassignBadge({ host }) {
    const metadata = {
      entityFqdn: "wix.badges.v3.badge",
      method: "DELETE",
      methodFqn: "com.wixpress.badges.v3.BadgesServiceV3.UnassignBadge",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressBadgesV3BadgesServiceV3Url({
        protoPath: "/v3/badges/{id}/members",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __unassignBadge;
}
function listMembers(payload) {
  function __listMembers({ host }) {
    const metadata = {
      entityFqdn: "wix.badges.v3.badge",
      method: "GET",
      methodFqn: "com.wixpress.badges.v3.BadgesServiceV3.ListMembers",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressBadgesV3BadgesServiceV3Url({
        protoPath: "/v3/badges/{id}/members",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __listMembers;
}
function listMembersBadgeIds(payload) {
  function __listMembersBadgeIds({ host }) {
    const metadata = {
      entityFqdn: "wix.badges.v3.badge",
      method: "POST",
      methodFqn: "com.wixpress.badges.v3.BadgesServiceV3.ListMembersBadgeIds",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressBadgesV3BadgesServiceV3Url({
        protoPath: "/v3/badges/members",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __listMembersBadgeIds;
}
function getMemberCountsPerBadge(payload) {
  function __getMemberCountsPerBadge({ host }) {
    const metadata = {
      entityFqdn: "wix.badges.v3.badge",
      method: "GET",
      methodFqn: "com.wixpress.badges.v3.BadgesServiceV3.GetMemberCountsPerBadge",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressBadgesV3BadgesServiceV3Url({
        protoPath: "/v3/badges/members/count",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getMemberCountsPerBadge;
}
function updateBadgesDisplayOrder(payload) {
  function __updateBadgesDisplayOrder({ host }) {
    const metadata = {
      entityFqdn: "wix.badges.v3.badge",
      method: "PATCH",
      methodFqn: "com.wixpress.badges.v3.BadgesServiceV3.UpdateBadgesDisplayOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressBadgesV3BadgesServiceV3Url({
        protoPath: "/v3/badges/order",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "badges.createdDate" },
            { path: "badges.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateBadgesDisplayOrder;
}

// src/badges-v3-badge-badges.universal.ts
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createBadge2(badge) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ badge });
  const reqOpts = createBadge(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.badge;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { badge: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["badge"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateBadge2(_id, badge) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    badge: { ...badge, id: _id }
  });
  const reqOpts = updateBadge(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.badge;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { badge: "$[1]" },
        explicitPathsToArguments: { "badge.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "badge"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listBadges2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    paging: options?.paging
  });
  const reqOpts = listBadges(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { paging: "$[0].paging" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getBadge2(_id) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ id: _id });
  const reqOpts = getBadge(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.badge;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { id: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteBadge2(_id) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ id: _id });
  const reqOpts = deleteBadge(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { id: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function assignBadge2(_id, memberIds) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    id: _id,
    memberIds
  });
  const reqOpts = assignBadge(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { id: "$[0]", memberIds: "$[1]" },
        singleArgumentUnchanged: false
      },
      ["_id", "memberIds"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function unassignBadge2(_id, memberIds) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    id: _id,
    memberIds
  });
  const reqOpts = unassignBadge(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { id: "$[0]", memberIds: "$[1]" },
        singleArgumentUnchanged: false
      },
      ["_id", "memberIds"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listMembersByBadge(_id, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    id: _id,
    paging: options?.paging
  });
  const reqOpts = listMembers(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { id: "$[0]", paging: "$[1].paging" },
        singleArgumentUnchanged: false
      },
      ["_id", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listBadgesPerMember(memberIds) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    memberIds
  });
  const reqOpts = listMembersBadgeIds(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { memberIds: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["memberIds"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getMemberCountsPerBadge2() {
  const { httpClient, sideEffects } = arguments[0];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({});
  const reqOpts = getMemberCountsPerBadge(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {},
        singleArgumentUnchanged: false
      },
      []
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateBadgesDisplayOrder2(badgeIds) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ badgeIds });
  const reqOpts = updateBadgesDisplayOrder(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { badgeIds: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["badgeIds"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  SortOrder,
  WebhookIdentityType,
  assignBadge,
  createBadge,
  deleteBadge,
  getBadge,
  getMemberCountsPerBadge,
  listBadges,
  listBadgesPerMember,
  listMembersByBadge,
  unassignBadge,
  updateBadge,
  updateBadgesDisplayOrder
});
