// @wix/auto_sdk_payments_saved-payment-methods/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  deleteSavedPaymentMethod: () => deleteSavedPaymentMethod2,
  getSavedPaymentMethod: () => getSavedPaymentMethod2,
  listSavedPaymentMethods: () => listSavedPaymentMethods2,
  markSavedPaymentMethodPrimary: () => markSavedPaymentMethodPrimary2,
  updateSavedPaymentMethod: () => updateSavedPaymentMethod2,
  upsertSavedPaymentMethod: () => upsertSavedPaymentMethod2
});
module.exports = __toCommonJS(meta_exports);

// src/payments-saved-payment-methods-v1-saved-payment-method-saved-payment-methods.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixPaymentsSavedPaymentMethodsV1SavedPaymentMethodServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/payments/saved-payment-methods",
        destPath: ""
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/payments/saved-payment-methods",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/payments/saved-payment-methods",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_saved-payment-methods";
function upsertSavedPaymentMethod(payload) {
  function __upsertSavedPaymentMethod({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "savedPaymentMethod.createdDate" },
          { path: "savedPaymentMethod.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.payments.saved_payment_methods.v1.saved_payment_method",
      method: "POST",
      methodFqn: "wix.payments.saved_payment_methods.v1.SavedPaymentMethodService.UpsertSavedPaymentMethod",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsSavedPaymentMethodsV1SavedPaymentMethodServiceUrl({
        protoPath: "/v1/saved-payment-methods/upsert",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "savedPaymentMethod.createdDate" },
            { path: "savedPaymentMethod.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __upsertSavedPaymentMethod;
}
function getSavedPaymentMethod(payload) {
  function __getSavedPaymentMethod({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.saved_payment_methods.v1.saved_payment_method",
      method: "GET",
      methodFqn: "wix.payments.saved_payment_methods.v1.SavedPaymentMethodService.GetSavedPaymentMethod",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsSavedPaymentMethodsV1SavedPaymentMethodServiceUrl({
        protoPath: "/v1/saved-payment-methods/{savedPaymentMethodId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "savedPaymentMethod.createdDate" },
            { path: "savedPaymentMethod.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getSavedPaymentMethod;
}
function deleteSavedPaymentMethod(payload) {
  function __deleteSavedPaymentMethod({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.saved_payment_methods.v1.saved_payment_method",
      method: "DELETE",
      methodFqn: "wix.payments.saved_payment_methods.v1.SavedPaymentMethodService.DeleteSavedPaymentMethod",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsSavedPaymentMethodsV1SavedPaymentMethodServiceUrl({
        protoPath: "/v1/saved-payment-methods/{savedPaymentMethodId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteSavedPaymentMethod;
}
function listSavedPaymentMethods(payload) {
  function __listSavedPaymentMethods({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.saved_payment_methods.v1.saved_payment_method",
      method: "GET",
      methodFqn: "wix.payments.saved_payment_methods.v1.SavedPaymentMethodService.ListSavedPaymentMethods",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsSavedPaymentMethodsV1SavedPaymentMethodServiceUrl({
        protoPath: "/v1/saved-payment-methods",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "savedPaymentMethods.createdDate" },
            { path: "savedPaymentMethods.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listSavedPaymentMethods;
}
function updateSavedPaymentMethod(payload) {
  function __updateSavedPaymentMethod({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "savedPaymentMethod.createdDate" },
          { path: "savedPaymentMethod.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.payments.saved_payment_methods.v1.saved_payment_method",
      method: "PATCH",
      methodFqn: "wix.payments.saved_payment_methods.v1.SavedPaymentMethodService.UpdateSavedPaymentMethod",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsSavedPaymentMethodsV1SavedPaymentMethodServiceUrl({
        protoPath: "/v1/saved-payment-methods/{savedPaymentMethod.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "savedPaymentMethod.createdDate" },
            { path: "savedPaymentMethod.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateSavedPaymentMethod;
}
function markSavedPaymentMethodPrimary(payload) {
  function __markSavedPaymentMethodPrimary({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.saved_payment_methods.v1.saved_payment_method",
      method: "GET",
      methodFqn: "wix.payments.saved_payment_methods.v1.SavedPaymentMethodService.MarkSavedPaymentMethodPrimary",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsSavedPaymentMethodsV1SavedPaymentMethodServiceUrl({
        protoPath: "/v1/saved-payment-methods/{savedPaymentMethodId}/mark-primary",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __markSavedPaymentMethodPrimary;
}

// src/payments-saved-payment-methods-v1-saved-payment-method-saved-payment-methods.meta.ts
function upsertSavedPaymentMethod2() {
  const payload = {};
  const getRequestOptions = upsertSavedPaymentMethod(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/saved-payment-methods/upsert",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getSavedPaymentMethod2() {
  const payload = { savedPaymentMethodId: ":savedPaymentMethodId" };
  const getRequestOptions = getSavedPaymentMethod(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/saved-payment-methods/{savedPaymentMethodId}",
    pathParams: { savedPaymentMethodId: "savedPaymentMethodId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteSavedPaymentMethod2() {
  const payload = { savedPaymentMethodId: ":savedPaymentMethodId" };
  const getRequestOptions = deleteSavedPaymentMethod(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/saved-payment-methods/{savedPaymentMethodId}",
    pathParams: { savedPaymentMethodId: "savedPaymentMethodId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listSavedPaymentMethods2() {
  const payload = {};
  const getRequestOptions = listSavedPaymentMethods(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/saved-payment-methods",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateSavedPaymentMethod2() {
  const payload = {
    savedPaymentMethod: { id: ":savedPaymentMethodId" }
  };
  const getRequestOptions = updateSavedPaymentMethod(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/saved-payment-methods/{savedPaymentMethod.id}",
    pathParams: { savedPaymentMethodId: "savedPaymentMethodId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function markSavedPaymentMethodPrimary2() {
  const payload = { savedPaymentMethodId: ":savedPaymentMethodId" };
  const getRequestOptions = markSavedPaymentMethodPrimary(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/saved-payment-methods/{savedPaymentMethodId}/mark-primary",
    pathParams: { savedPaymentMethodId: "savedPaymentMethodId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  deleteSavedPaymentMethod,
  getSavedPaymentMethod,
  listSavedPaymentMethods,
  markSavedPaymentMethodPrimary,
  updateSavedPaymentMethod,
  upsertSavedPaymentMethod
});
