// @wix/headless-blog/cjs/dist/services/blog-post-service.js
import { posts } from '@wix/blog';
import { defineService, implementService, } from '@wix/services-definitions';
import { SignalsServiceDefinition } from '@wix/services-definitions/core-services/signals';
import { enhancePosts } from './blog-feed-service.js';
export const BlogPostServiceDefinition = defineService('blogPostService');
export const BlogPostService = implementService.withConfig()(BlogPostServiceDefinition, ({ getService, config }) => {
    const signalsService = getService(SignalsServiceDefinition);
    const postSignal = signalsService.signal(config.post);
    const olderPostSignal = signalsService.signal(config.olderPost);
    const newerPostSignal = signalsService.signal(config.newerPost);
    return {
        post: postSignal,
        olderPost: olderPostSignal,
        newerPost: newerPostSignal,
    };
});
export async function loadBlogPostServiceConfig(params) {
    const { postSlug, includeSiblingPosts = true } = params;
    if (!postSlug) {
        return { type: 'notFound' };
    }
    try {
        const { post } = await posts.getPostBySlug(postSlug, {
            fieldsets: ['RICH_CONTENT', 'SEO', 'REFERENCE_ID'],
        });
        if (!post) {
            return { type: 'notFound' };
        }
        const siblingPosts = includeSiblingPosts
            ? await fetchSiblingPosts(post)
            : { olderPost: undefined, newerPost: undefined };
        const [enhancedPost, olderPost, newerPost] = await enhancePosts([
            post,
            siblingPosts.olderPost,
            siblingPosts.newerPost,
        ]);
        if (!enhancedPost) {
            return { type: 'notFound' };
        }
        return {
            type: 'success',
            config: {
                post: enhancedPost,
                olderPost,
                newerPost,
            },
        };
    }
    catch (error) {
        console.error('Failed to load initial post for slug', postSlug, error);
        return { type: 'notFound' };
    }
}
async function fetchSiblingPosts(post) {
    const [olderPost, newerPost] = await Promise.all([fetchOlderPost(post), fetchNewerPost(post)]);
    return { olderPost, newerPost };
}
// Edge case: using `or()` to fetch older with the same firstPublishedDate using id as tie-breaker
async function fetchOlderPost(post) {
    try {
        const olderPostQuery = posts
            .queryPosts()
            // @ts-expect-error
            .or(posts.queryPosts().lt('firstPublishedDate', post.firstPublishedDate))
            .or(posts.queryPosts().eq('firstPublishedDate', post.firstPublishedDate).lt(
        // @ts-expect-error
        '_id', post._id))
            .descending('firstPublishedDate')
            .limit(1);
        const result = await olderPostQuery.find();
        return result.items.at(0);
    }
    catch (error) {
        console.error('Failed to fetch older post', post, error);
        return undefined;
    }
}
// Edge case: using `or()` to fetch newer with the same firstPublishedDate using id as tie-breaker
async function fetchNewerPost(post) {
    try {
        const newerPostQuery = posts
            .queryPosts()
            // @ts-expect-error
            .or(posts.queryPosts().gt('firstPublishedDate', post.firstPublishedDate))
            .or(posts.queryPosts().eq('firstPublishedDate', post.firstPublishedDate).gt(
        // @ts-expect-error
        '_id', post._id))
            .ascending('firstPublishedDate')
            .limit(1);
        const result = await newerPostQuery.find();
        return result.items.at(0);
    }
    catch (error) {
        console.error('Failed to fetch newer post', post, error);
        return undefined;
    }
}
