// @wix/design-system/dist/cjs/NestableList/Arrow.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.Arrow = Arrow;
var _react = _interopRequireDefault(require("react"));
var _jsxFileName = "/home/builduser/work/57e038ea7326c1ec/packages/wix-design-system/dist/cjs/NestableList/Arrow.js";
function Arrow(_ref) {
  var {
    className = undefined
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("svg", {
    className: className,
    xmlns: "http://www.w3.org/2000/svg",
    width: 24,
    height: 8,
    viewBox: "0 0 24 8",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 5,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement("defs", {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 12,
      columnNumber: 7
    }
  }, /*#__PURE__*/_react.default.createElement("filter", {
    id: "zaxo0jk8xa",
    width: "110.4%",
    height: "125.6%",
    x: "-5.2%",
    y: "-12.8%",
    filterUnits: "objectBoundingBox",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 13,
      columnNumber: 9
    }
  }, /*#__PURE__*/_react.default.createElement("feOffset", {
    dx: 2,
    dy: 2,
    in: "SourceAlpha",
    result: "shadowOffsetOuter1",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 21,
      columnNumber: 11
    }
  }), /*#__PURE__*/_react.default.createElement("feGaussianBlur", {
    in: "shadowOffsetOuter1",
    result: "shadowBlurOuter1",
    stdDeviation: 12,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 27,
      columnNumber: 11
    }
  }), /*#__PURE__*/_react.default.createElement("feColorMatrix", {
    in: "shadowBlurOuter1",
    result: "shadowMatrixOuter1",
    values: "0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.05 0",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 32,
      columnNumber: 11
    }
  }), /*#__PURE__*/_react.default.createElement("feMerge", {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 37,
      columnNumber: 11
    }
  }, /*#__PURE__*/_react.default.createElement("feMergeNode", {
    in: "shadowMatrixOuter1",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 38,
      columnNumber: 13
    }
  }), /*#__PURE__*/_react.default.createElement("feMergeNode", {
    in: "SourceGraphic",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 39,
      columnNumber: 13
    }
  })))), /*#__PURE__*/_react.default.createElement("g", {
    fillRule: "evenodd",
    filter: "url(#zaxo0jk8xa)",
    transform: "translate(-33 -269)",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 43,
      columnNumber: 7
    }
  }, /*#__PURE__*/_react.default.createElement("path", {
    d: "M34 269v.5c0 1.325 1 2.5 2.5 2.5h18.793l-2.147-2.146c-.173-.174-.195-.512 0-.708.196-.195.512-.195.708 0l3 3c.195.196.195.512 0 .708l-3 3c-.196.195-.512.195-.708 0-.173-.174-.2-.503 0-.708L55.293 273H36.5c-2 0-3.5-1.5-3.5-3.5v-.5h1z",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 48,
      columnNumber: 9
    }
  })));
}
