// @wix/auto_sdk_events_policies/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  EventStatusOriginal: () => EventStatus,
  LocationTypeOriginal: () => LocationType,
  SortOrderOriginal: () => SortOrder,
  StatusOriginal: () => Status,
  SubdivisionTypeOriginal: () => SubdivisionType,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  createPolicy: () => createPolicy2,
  deletePolicy: () => deletePolicy2,
  getPolicy: () => getPolicy2,
  queryPolicies: () => queryPolicies2,
  reorderEventPolicies: () => reorderEventPolicies2,
  updatePolicy: () => updatePolicy2
});
module.exports = __toCommonJS(meta_exports);

// src/events-v2-policy-policies.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEventsV2PolicyManagementUrl(opts) {
  const domainToMappings = {
    _: [
      {
        srcPath: "/_api/events-policies",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/events-policies",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/events-policies",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/events-policies",
        destPath: ""
      },
      {
        srcPath: "/events/v2/policies",
        destPath: "/v2/policies"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/events-policies",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/events-policies",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/events-policies",
        destPath: ""
      }
    ],
    "events.wixapps.net": [
      {
        srcPath: "/_api/events-policies",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_events_policies";
function createPolicy(payload) {
  function __createPolicy({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "policy.createdDate" }, { path: "policy.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v2.policy",
      method: "POST",
      methodFqn: "wix.events.v2.PolicyManagement.CreatePolicy",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsV2PolicyManagementUrl({
        protoPath: "/v2/policies",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "policy.createdDate" },
            { path: "policy.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createPolicy;
}
function updatePolicy(payload) {
  function __updatePolicy({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "policy.createdDate" }, { path: "policy.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v2.policy",
      method: "PATCH",
      methodFqn: "wix.events.v2.PolicyManagement.UpdatePolicy",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsV2PolicyManagementUrl({
        protoPath: "/v2/policies/{policy.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "policy.createdDate" },
            { path: "policy.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updatePolicy;
}
function deletePolicy(payload) {
  function __deletePolicy({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v2.policy",
      method: "DELETE",
      methodFqn: "wix.events.v2.PolicyManagement.DeletePolicy",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsV2PolicyManagementUrl({
        protoPath: "/v2/policies/{policyId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deletePolicy;
}
function queryPolicies(payload) {
  function __queryPolicies({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v2.policy",
      method: "POST",
      methodFqn: "wix.events.v2.PolicyManagement.QueryPolicies",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsV2PolicyManagementUrl({
        protoPath: "/v2/policies/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "policies.createdDate" },
            { path: "policies.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryPolicies;
}
function reorderEventPolicies(payload) {
  function __reorderEventPolicies({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v2.policy",
      method: "POST",
      methodFqn: "wix.events.v2.PolicyManagement.ReorderEventPolicies",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsV2PolicyManagementUrl({
        protoPath: "/v2/policies/reorder",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "policies.createdDate" },
            { path: "policies.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __reorderEventPolicies;
}
function getPolicy(payload) {
  function __getPolicy({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v2.policy",
      method: "GET",
      methodFqn: "wix.events.v2.PolicyManagement.GetPolicy",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsV2PolicyManagementUrl({
        protoPath: "/v2/policies/{policyId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "policy.createdDate" },
            { path: "policy.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getPolicy;
}

// src/events-v2-policy-policies.types.ts
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var LocationType = /* @__PURE__ */ ((LocationType2) => {
  LocationType2["VENUE"] = "VENUE";
  LocationType2["ONLINE"] = "ONLINE";
  return LocationType2;
})(LocationType || {});
var SubdivisionType = /* @__PURE__ */ ((SubdivisionType2) => {
  SubdivisionType2["UNKNOWN_SUBDIVISION_TYPE"] = "UNKNOWN_SUBDIVISION_TYPE";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_1"] = "ADMINISTRATIVE_AREA_LEVEL_1";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_2"] = "ADMINISTRATIVE_AREA_LEVEL_2";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_3"] = "ADMINISTRATIVE_AREA_LEVEL_3";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_4"] = "ADMINISTRATIVE_AREA_LEVEL_4";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_5"] = "ADMINISTRATIVE_AREA_LEVEL_5";
  SubdivisionType2["COUNTRY"] = "COUNTRY";
  return SubdivisionType2;
})(SubdivisionType || {});
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["ONE_TIME"] = "ONE_TIME";
  Status2["RECURRING"] = "RECURRING";
  Status2["RECURRING_NEXT"] = "RECURRING_NEXT";
  Status2["RECURRING_LAST_ENDED"] = "RECURRING_LAST_ENDED";
  Status2["RECURRING_LAST_CANCELED"] = "RECURRING_LAST_CANCELED";
  return Status2;
})(Status || {});
var EventStatus = /* @__PURE__ */ ((EventStatus2) => {
  EventStatus2["SCHEDULED"] = "SCHEDULED";
  EventStatus2["STARTED"] = "STARTED";
  EventStatus2["ENDED"] = "ENDED";
  EventStatus2["CANCELED"] = "CANCELED";
  return EventStatus2;
})(EventStatus || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/events-v2-policy-policies.meta.ts
function createPolicy2() {
  const payload = {};
  const getRequestOptions = createPolicy(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/policies",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updatePolicy2() {
  const payload = { policy: { id: ":policyId" } };
  const getRequestOptions = updatePolicy(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v2/policies/{policy.id}",
    pathParams: { policyId: "policyId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deletePolicy2() {
  const payload = { policyId: ":policyId" };
  const getRequestOptions = deletePolicy(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v2/policies/{policyId}",
    pathParams: { policyId: "policyId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryPolicies2() {
  const payload = {};
  const getRequestOptions = queryPolicies(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/policies/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function reorderEventPolicies2() {
  const payload = {};
  const getRequestOptions = reorderEventPolicies(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/policies/reorder",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getPolicy2() {
  const payload = { policyId: ":policyId" };
  const getRequestOptions = getPolicy(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/policies/{policyId}",
    pathParams: { policyId: "policyId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  EventStatusOriginal,
  LocationTypeOriginal,
  SortOrderOriginal,
  StatusOriginal,
  SubdivisionTypeOriginal,
  WebhookIdentityTypeOriginal,
  createPolicy,
  deletePolicy,
  getPolicy,
  queryPolicies,
  reorderEventPolicies,
  updatePolicy
});
