// @wix/design-system/dist/cjs/AnalyticsSummaryCard/AnalyticsSummaryCard.st.css.js

require("../Foundation/stylable/colors.st.css.js");
require("../common/Focusable/Focusable.st.css.js");

const { classesRuntime, statesRuntime } = require("./../stylable-cjs-runtime.js");



var _namespace_ = "wds_1_226_0_AnalyticsSummaryCard";
var _style_ = classesRuntime.bind(null, _namespace_);

module.exports.cssStates = statesRuntime.bind(null, _namespace_);
module.exports.style = _style_;
module.exports.st = _style_;

module.exports.namespace = _namespace_;
module.exports.classes = {"root":"wds_1_226_0_AnalyticsSummaryCard__root","title":"wds_1_226_0_AnalyticsSummaryCard__title","valueLineContainer":"wds_1_226_0_AnalyticsSummaryCard__valueLineContainer","valueAndTrend":"wds_1_226_0_AnalyticsSummaryCard__valueAndTrend","trend":"wds_1_226_0_AnalyticsSummaryCard__trend","loader":"wds_1_226_0_AnalyticsSummaryCard__loader","ctaButton":"wds_1_226_0_AnalyticsSummaryCard__ctaButton","footer":"wds_1_226_0_AnalyticsSummaryCard__footer","sparkline":"wds_1_226_0_AnalyticsSummaryCard__sparkline"};
module.exports.keyframes = {}; 
module.exports.layers = {};
module.exports.containers = {};
module.exports.stVars = {}; 
module.exports.vars = {"wds-space-300":"--wds-space-300","wds-space-400":"--wds-space-400","wds-space-500":"--wds-space-500","wds-border-radius-300":"--wds-border-radius-300","wds-color-border-standard-secondary":"--wds-color-border-standard-secondary","wds-color-border-standard-secondary-hover":"--wds-color-border-standard-secondary-hover"};
