// @wix/design-system/dist/cjs/AnalyticsSummaryCard/AnalyticsSummaryCard.uni.driver.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.analyticsSummaryCardDriverFactory = void 0;
var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _unidriver = require("../utils/test-utils/unidriver");
var _LoaderUni = require("../Loader/Loader.uni.driver");
var _TextUni = require("../Text/Text.uni.driver");
var _TooltipUni = require("../Tooltip/Tooltip.uni.driver");
var _TrendIndicatorUni = require("../TrendIndicator/TrendIndicator.uni.driver");
var _AnalyticsSummaryCard = require("./AnalyticsSummaryCard.constants");
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2.default)(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var analyticsSummaryCardDriverFactory = (base, body) => {
  var getLoader = () => (0, _LoaderUni.loaderUniDriverFactory)((0, _unidriver.findByHook)(base, _AnalyticsSummaryCard.dataHooks.analyticsSummaryCardLoader), body);
  var _getValue = () => (0, _TextUni.textUniDriverFactory)((0, _unidriver.findByHook)(base, _AnalyticsSummaryCard.dataHooks.analyticsSummaryCardValue));
  var _getTitle = () => (0, _TextUni.textUniDriverFactory)((0, _unidriver.findByHook)(base, _AnalyticsSummaryCard.dataHooks.analyticsSummaryCardTitle));
  var getValueTooltip = () => (0, _TooltipUni.tooltipDriverFactory)(base.$("[data-hook=\"".concat(_AnalyticsSummaryCard.dataHooks.analyticsSummaryCardValueTooltip, "\"]")), body);
  var getTitleTooltip = () => (0, _TooltipUni.tooltipDriverFactory)(base.$("[data-hook=\"".concat(_AnalyticsSummaryCard.dataHooks.analyticsSummaryCardTitleTooltip, "\"]")), body);
  var getTrendIndicator = () => (0, _TrendIndicatorUni.trendIndicatorDriverFactory)((0, _unidriver.findByHook)(base, _AnalyticsSummaryCard.dataHooks.analyticsSummaryCardTrend));
  var getTooltipText = tooltip => tooltip.getTooltipText();
  return _objectSpread(_objectSpread({}, (0, _unidriver.baseUniDriverFactory)(base)), {}, {
    /**
     * Click on CTA
     * @returns {Promise<React.MouseEventHandler<HTMLButtonElement>>}
     */
    clickCTA: function () {
      var _clickCTA = (0, _asyncToGenerator2.default)(function* () {
        yield base.hover();
        yield (0, _unidriver.findByHook)(base, _AnalyticsSummaryCard.dataHooks.analyticsSummaryCardCTA).click();
      });
      function clickCTA() {
        return _clickCTA.apply(this, arguments);
      }
      return clickCTA;
    }(),
    /**
     * Check if CTA exists
     * @returns {boolean}
     */
    CTAExists: function () {
      var _CTAExists = (0, _asyncToGenerator2.default)(function* () {
        yield base.hover();
        return (0, _unidriver.findByHook)(base, _AnalyticsSummaryCard.dataHooks.analyticsSummaryCardCTA).exists();
      });
      function CTAExists() {
        return _CTAExists.apply(this, arguments);
      }
      return CTAExists;
    }(),
    /**
     * Check if CTA exists
     * @returns {boolean}
     */
    CTAExistsWithoutHover: function () {
      var _CTAExistsWithoutHover = (0, _asyncToGenerator2.default)(function* () {
        return (0, _unidriver.findByHook)(base, _AnalyticsSummaryCard.dataHooks.analyticsSummaryCardCTA).exists();
      });
      function CTAExistsWithoutHover() {
        return _CTAExistsWithoutHover.apply(this, arguments);
      }
      return CTAExistsWithoutHover;
    }(),
    /**
     * Checks whether the Analytics are loading
     * @returns {Promise<boolean>}
     */
    isLoading: () => getLoader().exists(),
    /**
     * returns the value text
     * @returns {Promise<string>}
     */
    getValue: () => _getValue().getText(),
    /**
     * returns the value tooltip text
     * @returns {Promise<string>}
     */
    getValueTooltipText: function () {
      var _getValueTooltipText = (0, _asyncToGenerator2.default)(function* () {
        return getTooltipText(yield getValueTooltip());
      });
      function getValueTooltipText() {
        return _getValueTooltipText.apply(this, arguments);
      }
      return getValueTooltipText;
    }(),
    /**
     * returns the title text
     * @returns {Promise<string>}
     */
    getTitle: () => _getTitle().getText(),
    /**
     * returns the title tooltip text
     * @returns {Promise<string>}
     */
    getTitleTooltipText: function () {
      var _getTitleTooltipText = (0, _asyncToGenerator2.default)(function* () {
        return getTooltipText(yield getTitleTooltip());
      });
      function getTitleTooltipText() {
        return _getTitleTooltipText.apply(this, arguments);
      }
      return getTitleTooltipText;
    }(),
    /**
     * checks if trend exists
     * @returns {Promise<boolean>}
     */
    trendExists: () => getTrendIndicator().exists(),
    /**
     * returns the trend value
     * @returns {Promise<number>}
     */
    getTrendvalue: () => getTrendIndicator().getTrendValue()
  });
};
exports.analyticsSummaryCardDriverFactory = analyticsSummaryCardDriverFactory;
