// @wix/auto-patterns/dist/cjs/providers/ModalContext.js
"use strict";

exports.__esModule = true;
exports.useModalContext = exports.ModalProvider = void 0;
var _ModalRenderer = require("../components/ModalRenderer");
var _react = _interopRequireWildcard(require("react"));
var _jsxFileName = "/home/builduser/work/73d19ce3378ce4dd/packages/auto-patterns/dist/cjs/providers/ModalContext.tsx";
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
const ModalContext = /*#__PURE__*/(0, _react.createContext)(null);
const ModalProvider = ({
  children
}) => {
  const [modal, setModal] = (0, _react.useState)({
    id: null
  });
  const openModal = (0, _react.useCallback)((id, props) => {
    setModal({
      id,
      props
    });
  }, []);
  const closeModal = (0, _react.useCallback)(() => {
    setModal({
      id: null
    });
  }, []);
  return /*#__PURE__*/_react.default.createElement(ModalContext.Provider, {
    value: {
      modal,
      openModal,
      closeModal
    },
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 41,
      columnNumber: 5
    }
  }, children, /*#__PURE__*/_react.default.createElement(_ModalRenderer.ModalRenderer, {
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 43,
      columnNumber: 7
    }
  }));
};
exports.ModalProvider = ModalProvider;
const useModalContext = () => {
  const context = (0, _react.useContext)(ModalContext);
  if (!context) {
    throw new Error('useModalContext must be used within ModalProvider');
  }
  return context;
};
exports.useModalContext = useModalContext;
