// @wix/auto-patterns/dist/cjs/providers/SchemaRegistryContext.js
"use strict";

exports.__esModule = true;
exports.useSchemaRegistry = exports.SchemaRegistryProvider = void 0;
var _react = _interopRequireWildcard(require("react"));
var _jsxFileName = "/home/builduser/work/73d19ce3378ce4dd/packages/auto-patterns/dist/cjs/providers/SchemaRegistryContext.tsx";
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
const SchemaRegistryContext = /*#__PURE__*/(0, _react.createContext)({
  getSchema: () => undefined,
  addSchema: () => {}
});
const SchemaRegistryProvider = ({
  children
}) => {
  const [registry, setRegistry] = (0, _react.useState)({});
  const addSchema = (0, _react.useCallback)((collectionId, schema) => {
    setRegistry(prevRegistry => ({
      ...prevRegistry,
      [collectionId]: schema
    }));
  }, []);
  const getSchema = (0, _react.useCallback)(collectionId => {
    return registry == null ? void 0 : registry[collectionId];
  }, [registry]);
  const contextValue = (0, _react.useMemo)(() => ({
    getSchema,
    addSchema
  }), [getSchema, addSchema]);
  return /*#__PURE__*/_react.default.createElement(SchemaRegistryContext.Provider, {
    value: contextValue,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 59,
      columnNumber: 5
    }
  }, children);
};
exports.SchemaRegistryProvider = SchemaRegistryProvider;
const useSchemaRegistry = () => {
  const context = (0, _react.useContext)(SchemaRegistryContext);
  if (!context) {
    throw new Error('useSchemaRegistry must be used within a UseRegistryProvider');
  }
  return context;
};
exports.useSchemaRegistry = useSchemaRegistry;
