// @wix/auto_sdk_comments_categories/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  DeleteStatusOriginal: () => DeleteStatus,
  IconTypeTypeOriginal: () => IconTypeType,
  NamespaceOriginal: () => Namespace,
  ReactionTypeTypeOriginal: () => ReactionTypeType,
  RepliesReactionTypeTypeOriginal: () => RepliesReactionTypeType,
  RoleOriginal: () => Role,
  SiteCreatedContextOriginal: () => SiteCreatedContext,
  SortOrderOriginal: () => SortOrder,
  StateOriginal: () => State,
  TypeOriginal: () => Type,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  createCategory: () => createCategory2,
  getCategory: () => getCategory2,
  queryCategory: () => queryCategory2,
  updateCategory: () => updateCategory2
});
module.exports = __toCommonJS(meta_exports);

// src/comments-v1-category-categories.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixCommentsCategoriesV1CommentsCategoriesUrl(opts) {
  const domainToMappings = {
    _: [
      {
        srcPath: "/_api/comments-categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/comments-categories/v1/categories",
        destPath: "/v1/categories"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/comments-categories/v1/categories",
        destPath: "/v1/categories"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/comments-categories/v1/categories",
        destPath: "/v1/categories"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/comments-categories/v1/categories",
        destPath: "/v1/categories"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/comments-categories/v1/categories",
        destPath: "/v1/categories"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/comments-categories/v1/categories",
        destPath: "/v1/categories"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/comments-categories",
        destPath: "/v1/categories"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/comments-categories",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/comments-categories/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/comments-categories",
        destPath: "/v1/categories"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/comments/v1/categories",
        destPath: "/v1/categories"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_comments_categories";
function createCategory(payload) {
  function __createCategory({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "category.createdDate" },
          { path: "category.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.comments.v1.category",
      method: "POST",
      methodFqn: "wix.comments.categories.v1.CommentsCategories.CreateCategory",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixCommentsCategoriesV1CommentsCategoriesUrl({
        protoPath: "/v1/categories",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "category.createdDate" },
            { path: "category.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createCategory;
}
function getCategory(payload) {
  function __getCategory({ host }) {
    const metadata = {
      entityFqdn: "wix.comments.v1.category",
      method: "GET",
      methodFqn: "wix.comments.categories.v1.CommentsCategories.GetCategory",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixCommentsCategoriesV1CommentsCategoriesUrl({
        protoPath: "/v1/categories/{categoryId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "category.createdDate" },
            { path: "category.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getCategory;
}
function updateCategory(payload) {
  function __updateCategory({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "category.createdDate" },
          { path: "category.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.comments.v1.category",
      method: "PATCH",
      methodFqn: "wix.comments.categories.v1.CommentsCategories.UpdateCategory",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixCommentsCategoriesV1CommentsCategoriesUrl({
        protoPath: "/v1/categories/{category.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "category.createdDate" },
            { path: "category.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateCategory;
}
function queryCategory(payload) {
  function __queryCategory({ host }) {
    const metadata = {
      entityFqdn: "wix.comments.v1.category",
      method: "POST",
      methodFqn: "wix.comments.categories.v1.CommentsCategories.QueryCategory",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixCommentsCategoriesV1CommentsCategoriesUrl({
        protoPath: "/v1/categories/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "categories.createdDate" },
            { path: "categories.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryCategory;
}

// src/comments-v1-category-categories.types.ts
var ReactionTypeType = /* @__PURE__ */ ((ReactionTypeType2) => {
  ReactionTypeType2["UNKNOWN"] = "UNKNOWN";
  ReactionTypeType2["VOTE"] = "VOTE";
  ReactionTypeType2["REACTIONS"] = "REACTIONS";
  ReactionTypeType2["MULTIPLE_REACTIONS"] = "MULTIPLE_REACTIONS";
  return ReactionTypeType2;
})(ReactionTypeType || {});
var RepliesReactionTypeType = /* @__PURE__ */ ((RepliesReactionTypeType2) => {
  RepliesReactionTypeType2["UNKNOWN"] = "UNKNOWN";
  RepliesReactionTypeType2["REACTIONS"] = "REACTIONS";
  RepliesReactionTypeType2["MULTIPLE_REACTIONS"] = "MULTIPLE_REACTIONS";
  return RepliesReactionTypeType2;
})(RepliesReactionTypeType || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNKNOWN"] = "UNKNOWN";
  Type2["FEATURED"] = "FEATURED";
  Type2["BEST_ANSWER"] = "BEST_ANSWER";
  Type2["SOLVED"] = "SOLVED";
  Type2["PINNED"] = "PINNED";
  Type2["CUSTOM"] = "CUSTOM";
  return Type2;
})(Type || {});
var IconTypeType = /* @__PURE__ */ ((IconTypeType2) => {
  IconTypeType2["UNKNOWN"] = "UNKNOWN";
  IconTypeType2["LOVE"] = "LOVE";
  IconTypeType2["THUMBS_UP"] = "THUMBS_UP";
  return IconTypeType2;
})(IconTypeType || {});
var Role = /* @__PURE__ */ ((Role2) => {
  Role2["UNKNOWN_ROLE"] = "UNKNOWN_ROLE";
  Role2["ALL"] = "ALL";
  Role2["MEMBER"] = "MEMBER";
  Role2["ADMIN"] = "ADMIN";
  return Role2;
})(Role || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var State = /* @__PURE__ */ ((State2) => {
  State2["UNKNOWN"] = "UNKNOWN";
  State2["ENABLED"] = "ENABLED";
  State2["DISABLED"] = "DISABLED";
  State2["PENDING"] = "PENDING";
  State2["DEMO"] = "DEMO";
  return State2;
})(State || {});
var SiteCreatedContext = /* @__PURE__ */ ((SiteCreatedContext2) => {
  SiteCreatedContext2["OTHER"] = "OTHER";
  SiteCreatedContext2["FROM_TEMPLATE"] = "FROM_TEMPLATE";
  SiteCreatedContext2["DUPLICATE_BY_SITE_TRANSFER"] = "DUPLICATE_BY_SITE_TRANSFER";
  SiteCreatedContext2["DUPLICATE"] = "DUPLICATE";
  SiteCreatedContext2["OLD_SITE_TRANSFER"] = "OLD_SITE_TRANSFER";
  SiteCreatedContext2["FLASH"] = "FLASH";
  return SiteCreatedContext2;
})(SiteCreatedContext || {});
var Namespace = /* @__PURE__ */ ((Namespace2) => {
  Namespace2["UNKNOWN_NAMESPACE"] = "UNKNOWN_NAMESPACE";
  Namespace2["WIX"] = "WIX";
  Namespace2["SHOUT_OUT"] = "SHOUT_OUT";
  Namespace2["ALBUMS"] = "ALBUMS";
  Namespace2["WIX_STORES_TEST_DRIVE"] = "WIX_STORES_TEST_DRIVE";
  Namespace2["HOTELS"] = "HOTELS";
  Namespace2["CLUBS"] = "CLUBS";
  Namespace2["ONBOARDING_DRAFT"] = "ONBOARDING_DRAFT";
  Namespace2["DEV_SITE"] = "DEV_SITE";
  Namespace2["LOGOS"] = "LOGOS";
  Namespace2["VIDEO_MAKER"] = "VIDEO_MAKER";
  Namespace2["PARTNER_DASHBOARD"] = "PARTNER_DASHBOARD";
  Namespace2["DEV_CENTER_COMPANY"] = "DEV_CENTER_COMPANY";
  Namespace2["HTML_DRAFT"] = "HTML_DRAFT";
  Namespace2["SITELESS_BUSINESS"] = "SITELESS_BUSINESS";
  Namespace2["CREATOR_ECONOMY"] = "CREATOR_ECONOMY";
  Namespace2["DASHBOARD_FIRST"] = "DASHBOARD_FIRST";
  Namespace2["ANYWHERE"] = "ANYWHERE";
  Namespace2["HEADLESS"] = "HEADLESS";
  Namespace2["ACCOUNT_MASTER_CMS"] = "ACCOUNT_MASTER_CMS";
  Namespace2["RISE"] = "RISE";
  Namespace2["BRANDED_FIRST"] = "BRANDED_FIRST";
  Namespace2["NOWNIA"] = "NOWNIA";
  Namespace2["UGC_TEMPLATE"] = "UGC_TEMPLATE";
  Namespace2["CODUX"] = "CODUX";
  Namespace2["MEDIA_DESIGN_CREATOR"] = "MEDIA_DESIGN_CREATOR";
  Namespace2["SHARED_BLOG_ENTERPRISE"] = "SHARED_BLOG_ENTERPRISE";
  Namespace2["STANDALONE_FORMS"] = "STANDALONE_FORMS";
  Namespace2["STANDALONE_EVENTS"] = "STANDALONE_EVENTS";
  Namespace2["MIMIR"] = "MIMIR";
  Namespace2["TWINS"] = "TWINS";
  Namespace2["NANO"] = "NANO";
  return Namespace2;
})(Namespace || {});
var DeleteStatus = /* @__PURE__ */ ((DeleteStatus2) => {
  DeleteStatus2["UNKNOWN"] = "UNKNOWN";
  DeleteStatus2["TRASH"] = "TRASH";
  DeleteStatus2["DELETED"] = "DELETED";
  DeleteStatus2["PENDING_PURGE"] = "PENDING_PURGE";
  DeleteStatus2["PURGED_EXTERNALLY"] = "PURGED_EXTERNALLY";
  return DeleteStatus2;
})(DeleteStatus || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/comments-v1-category-categories.meta.ts
function createCategory2() {
  const payload = {};
  const getRequestOptions = createCategory(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/categories",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getCategory2() {
  const payload = { categoryId: ":categoryId" };
  const getRequestOptions = getCategory(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/categories/{categoryId}",
    pathParams: { categoryId: "categoryId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateCategory2() {
  const payload = { category: { id: ":categoryId" } };
  const getRequestOptions = updateCategory(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/categories/{category.id}",
    pathParams: { categoryId: "categoryId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryCategory2() {
  const payload = {};
  const getRequestOptions = queryCategory(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/categories/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  DeleteStatusOriginal,
  IconTypeTypeOriginal,
  NamespaceOriginal,
  ReactionTypeTypeOriginal,
  RepliesReactionTypeTypeOriginal,
  RoleOriginal,
  SiteCreatedContextOriginal,
  SortOrderOriginal,
  StateOriginal,
  TypeOriginal,
  WebhookIdentityTypeOriginal,
  createCategory,
  getCategory,
  queryCategory,
  updateCategory
});
