// @wix/auto_sdk_benefit-programs_balances/build/internal/es/meta.mjs
// src/benefit-programs-v1-balance-balances.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixBenefitProgramsV1BalanceBalanceServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/pool-accounts",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/benefit-programs-pool-accounts",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pool-accounts",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pool-accounts",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/benefit-programs-pool-accounts",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/benefit-programs-pool-accounts",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pool-accounts",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pool-accounts",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/benefit-programs/v1/balances",
        destPath: "/v1/balances"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/balances",
        destPath: "/v1/bulk/balances"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_benefit-programs_balances";
function getBalance(payload) {
  function __getBalance({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.balance",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.balance.BalanceService.GetBalance",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1BalanceBalanceServiceUrl({
        protoPath: "/v1/balances/{poolId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "balance.createdDate" },
            { path: "balance.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getBalance;
}
function listBalances(payload) {
  function __listBalances({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.balance",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.balance.BalanceService.ListBalances",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1BalanceBalanceServiceUrl({
        protoPath: "/v1/balances",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "balances.createdDate" },
            { path: "balances.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listBalances;
}
function queryBalances(payload) {
  function __queryBalances({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.balance",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.balance.BalanceService.QueryBalances",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1BalanceBalanceServiceUrl({
        protoPath: "/v1/balances/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "balances.createdDate" },
            { path: "balances.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryBalances;
}
function changeBalance(payload) {
  function __changeBalance({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "transactionDetails.effectiveDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.balance",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.balance.BalanceService.ChangeBalance",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1BalanceBalanceServiceUrl({
        protoPath: "/v1/balances/{poolId}/change",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "balance.createdDate" },
            { path: "balance.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __changeBalance;
}
function revertBalanceChange(payload) {
  function __revertBalanceChange({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.balance",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.balance.BalanceService.RevertBalanceChange",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1BalanceBalanceServiceUrl({
        protoPath: "/v1/balances/changes/{transactionId}/revert",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __revertBalanceChange;
}

// src/benefit-programs-v1-balance-balances.meta.ts
function getBalance2() {
  const payload = { poolId: ":poolId" };
  const getRequestOptions = getBalance(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/balances/{poolId}",
    pathParams: { poolId: "poolId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listBalances2() {
  const payload = {};
  const getRequestOptions = listBalances(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/balances",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryBalances2() {
  const payload = {};
  const getRequestOptions = queryBalances(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/balances/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function changeBalance2() {
  const payload = { poolId: ":poolId" };
  const getRequestOptions = changeBalance(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/balances/{poolId}/change",
    pathParams: { poolId: "poolId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function revertBalanceChange2() {
  const payload = { transactionId: ":transactionId" };
  const getRequestOptions = revertBalanceChange(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/balances/changes/{transactionId}/revert",
    pathParams: { transactionId: "transactionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  changeBalance2 as changeBalance,
  getBalance2 as getBalance,
  listBalances2 as listBalances,
  queryBalances2 as queryBalances,
  revertBalanceChange2 as revertBalanceChange
};
