// @wix/patterns/dist/cjs/components/NestedTableDragAndDrop/DepthLimitMessage.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.DepthLimitMessage = void 0;
var _react = _interopRequireDefault(require("react"));
var _mobxReactLite = require("mobx-react-lite");
var _designSystem = require("@wix/design-system");
var _wixUiIconsCommon = require("@wix/wix-ui-icons-common");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/NestedTableDragAndDrop/DepthLimitMessage.tsx";
function _DepthLimitMessage(props) {
  const {
    state
  } = props;
  const {
    nestedState
  } = state;
  const {
    levels,
    container: {
      translate: t
    }
  } = nestedState;
  return /*#__PURE__*/_react.default.createElement(_designSystem.Tag, {
    id: "nested-table-dnd-depth-limit-message",
    dataHook: "nested-table-dnd-depth-limit-message",
    theme: "light",
    thumb: /*#__PURE__*/_react.default.createElement(_wixUiIconsCommon.BlockSmall, {
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 27,
        columnNumber: 14
      }
    }),
    removable: false,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 23,
      columnNumber: 5
    }
  }, t('cairo.collectionPage.nestedLimit.tooltip', {
    subCatsAmount: levels.length
  }));
}
const DepthLimitMessage = exports.DepthLimitMessage = (0, _mobxReactLite.observer)(_DepthLimitMessage);
