// @wix/patterns/dist/cjs/components/NestedTableDragAndDrop/DragAndDropDisabledMessage.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.DragAndDropDisabledMessage = DragAndDropDisabledMessage;
var _react = _interopRequireDefault(require("react"));
var _designSystem = require("@wix/design-system");
var _wixUiIconsCommon = require("@wix/wix-ui-icons-common");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/NestedTableDragAndDrop/DragAndDropDisabledMessage.tsx";
function DragAndDropDisabledMessage(props) {
  const {
    message
  } = props;
  return /*#__PURE__*/_react.default.createElement(_designSystem.Tag, {
    id: "nested-table-dnd-disabled-message",
    dataHook: "nested-table-dnd-disabled-message",
    theme: "light",
    thumb: /*#__PURE__*/_react.default.createElement(_wixUiIconsCommon.BlockSmall, {
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 19,
        columnNumber: 14
      }
    }),
    removable: false,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 15,
      columnNumber: 5
    }
  }, message);
}
