// @wix/wix-ui-test-utils/dist/esm/enzyme/enzyme.js
import * as React from 'react';
import { reactEventTrigger } from '../react-helpers';
import { jsdomReactUniDriver } from '@wix/unidriver-jsdom-react-legacy';
export function enzymeTestkitFactoryCreator(driverFactory) {
    return (obj) => {
        const eventTrigger = reactEventTrigger();
        const regexp = new RegExp(`^<[^>]+data-hook="${obj.dataHook}"`);
        const component = obj.wrapper.findWhere((n) => n.length > 0 && typeof n.type() === 'string' && regexp.test(n.html()));
        const element = component.length > 0 ? component.first().getDOMNode() : undefined;
        return driverFactory({
            element,
            wrapper: obj.wrapper,
            eventTrigger,
            dataHook: obj.dataHook,
        });
    };
}
export function enzymeUniTestkitFactoryCreator(driverFactory) {
    return (obj) => {
        const regexp = new RegExp(`^<[^>]+data-hook="${obj.dataHook}"`);
        const component = obj.wrapper.findWhere((n) => n.length > 0 && typeof n.type() === 'string' && regexp.test(n.html()));
        const element = component.length > 0 ? component.first().getDOMNode() : undefined;
        const base = jsdomReactUniDriver(element);
        const body = jsdomReactUniDriver(document.body);
        return driverFactory(base, body, { dataHook: obj.dataHook });
    };
}
/**
 * Checks if the given Element accepts a data hook, and that the testkit factory finds the component's root element using that data hook.
 *
 * This method supports both new snake-case and legacy camelCase data hook prop name (e.g `data-hook` and `dataHook`).
 * The default is to check by both prop name options.
 */
export function isEnzymeTestkitExists(Element, testkitFactory, mount, options = {}) {
    return isEnzymeTestkitExistsInternal({
        Element,
        testkitFactory,
        mount,
        ...options,
    });
}
export async function isUniEnzymeTestkitExists(Element, testkitFactory, mount, { withoutDataHook, dataHookPropName } = {}) {
    const dataHook = withoutDataHook ? '' : 'myDataHook';
    const extraProps = dataHookPropName
        ? { [dataHookPropName]: dataHook }
        : { dataHook, 'data-hook': dataHook };
    const elementToRender = React.cloneElement(Element, extraProps);
    const wrapper = mount(elementToRender);
    const testkit = testkitFactory({ wrapper, dataHook });
    return testkit.exists();
}
/**
 * This internal function is only in order to allow separate defaults to each options.
 */
function isEnzymeTestkitExistsInternal({ Element, testkitFactory, mount, withoutDataHook = false, dataHookPropName, }) {
    const dataHook = withoutDataHook ? '' : 'myDataHook';
    const extraProps = dataHookPropName
        ? { [dataHookPropName]: dataHook }
        : { dataHook, 'data-hook': dataHook };
    const elementToRender = React.cloneElement(Element, extraProps);
    const wrapper = mount(elementToRender);
    const testkit = testkitFactory({ wrapper, dataHook });
    return testkit.exists();
}
