// @wix/events/build/es/meta.mjs
// meta.ts
import * as guests from "@wix/auto_sdk_events_guests/meta";
import * as notifications from "@wix/auto_sdk_events_notifications/meta";
import * as scheduleBookmarks from "@wix/auto_sdk_events_schedule-bookmarks/meta";
import * as schedule from "@wix/auto_sdk_events_schedule/meta";
import * as staffMembers from "@wix/auto_sdk_events_staff-members/meta";
import * as categories from "@wix/auto_sdk_events_categories/meta";
import * as eventsSettings from "@wix/auto_sdk_events_events-settings/meta";
import * as forms from "@wix/auto_sdk_events_forms/meta";
import * as orders from "@wix/auto_sdk_events_orders/meta";
import * as rsvp from "@wix/auto_sdk_events_rsvp/meta";
import * as tickets from "@wix/auto_sdk_events_tickets/meta";
import * as ticketDefinitions from "@wix/auto_sdk_events_ticket-definitions/meta";
import * as ticketReservations from "@wix/auto_sdk_events_ticket-reservations/meta";
import * as policies from "@wix/auto_sdk_events_policies/meta";
import * as rsvpV2 from "@wix/auto_sdk_events_rsvp-v-2/meta";
import * as wixEventsV2 from "@wix/auto_sdk_events_wix-events-v-2/meta";
import * as ticketDefinitionsV2 from "@wix/auto_sdk_events_ticket-definitions-v-2/meta";
export {
  categories,
  eventsSettings,
  forms,
  guests,
  notifications,
  orders,
  policies,
  rsvp,
  rsvpV2,
  schedule,
  scheduleBookmarks,
  staffMembers,
  ticketDefinitions,
  ticketDefinitionsV2,
  ticketReservations,
  tickets,
  wixEventsV2
};
