// @wix/auto_sdk_restaurants_recipients/build/internal/es/index.mjs
// src/restaurants-recipient-v1-recipient-recipients.public.ts
import { renameKeysFromRESTResponseToSDKResponse as renameKeysFromRESTResponseToSDKResponse2 } from "@wix/sdk-runtime/rename-all-nested-keys";
import { transformRESTTimestampToSDKTimestamp as transformRESTTimestampToSDKTimestamp2 } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths as transformPaths3 } from "@wix/sdk-runtime/transformations/transform-paths";
import { EventDefinition } from "@wix/sdk-types";

// src/restaurants-recipient-v1-recipient-recipients.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/restaurants-recipient-v1-recipient-recipients.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/rest-notification-recipients",
        destPath: ""
      }
    ],
    "dev._base_domain_": [
      {
        srcPath: "/_api/rest-notification-recipients",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/rest-notification-recipients/v1",
        destPath: "/v1"
      },
      {
        srcPath: "/restaurants/recipient/v1/recipients",
        destPath: "/v1/recipients"
      }
    ],
    _: [
      {
        srcPath: "/_api/rest-notification-recipients",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/rest-notification-recipients",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_recipients";
function createRecipient(payload) {
  function __createRecipient({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "recipient.createdDate" },
          { path: "recipient.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.recipient.v1.recipient",
      method: "POST",
      methodFqn: "wix.restaurants.recipient.v1.NotificationRecipientsService.CreateRecipient",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl({
        protoPath: "/v1/recipients",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "recipient.createdDate" },
            { path: "recipient.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createRecipient;
}
function getRecipient(payload) {
  function __getRecipient({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.recipient.v1.recipient",
      method: "GET",
      methodFqn: "wix.restaurants.recipient.v1.NotificationRecipientsService.GetRecipient",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl({
        protoPath: "/v1/recipients/{recipientId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "recipient.createdDate" },
            { path: "recipient.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getRecipient;
}
function updateRecipient(payload) {
  function __updateRecipient({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "recipient.createdDate" },
          { path: "recipient.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.recipient.v1.recipient",
      method: "PATCH",
      methodFqn: "wix.restaurants.recipient.v1.NotificationRecipientsService.UpdateRecipient",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl({
        protoPath: "/v1/recipients/{recipient.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "recipient.createdDate" },
            { path: "recipient.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateRecipient;
}
function deleteRecipient(payload) {
  function __deleteRecipient({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.recipient.v1.recipient",
      method: "DELETE",
      methodFqn: "wix.restaurants.recipient.v1.NotificationRecipientsService.DeleteRecipient",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl({
        protoPath: "/v1/recipients/{recipientId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteRecipient;
}
function queryRecipients(payload) {
  function __queryRecipients({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.recipient.v1.recipient",
      method: "GET",
      methodFqn: "wix.restaurants.recipient.v1.NotificationRecipientsService.QueryRecipients",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl({
        protoPath: "/v1/recipients/query",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload, true),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "recipients.createdDate" },
            { path: "recipients.updatedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl(
            { protoPath: "/v1/recipients/query", data: payload, host }
          ),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryRecipients;
}
function bulkUpdateRecipientTags(payload) {
  function __bulkUpdateRecipientTags({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.recipient.v1.recipient",
      method: "POST",
      methodFqn: "wix.restaurants.recipient.v1.NotificationRecipientsService.BulkUpdateRecipientTags",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl({
        protoPath: "/v1/bulk/recipients/update-tags",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateRecipientTags;
}
function bulkUpdateRecipientTagsByFilter(payload) {
  function __bulkUpdateRecipientTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.recipient.v1.recipient",
      method: "POST",
      methodFqn: "wix.restaurants.recipient.v1.NotificationRecipientsService.BulkUpdateRecipientTagsByFilter",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl({
        protoPath: "/v1/bulk/recipients/update-tags-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateRecipientTagsByFilter;
}

// src/restaurants-recipient-v1-recipient-recipients.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var Channel = /* @__PURE__ */ ((Channel2) => {
  Channel2["UNKNOWN_CHANNEL"] = "UNKNOWN_CHANNEL";
  Channel2["SMS"] = "SMS";
  Channel2["PHONE"] = "PHONE";
  return Channel2;
})(Channel || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createRecipient2(recipient) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    recipient
  });
  const reqOpts = createRecipient(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.recipient;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { recipient: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["recipient"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getRecipient2(recipientId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    recipientId
  });
  const reqOpts = getRecipient(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.recipient;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { recipientId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["recipientId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateRecipient2(_id, recipient) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    recipient: { ...recipient, id: _id }
  });
  const reqOpts = updateRecipient(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.recipient;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { recipient: "$[1]" },
        explicitPathsToArguments: { "recipient.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "recipient"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteRecipient2(recipientId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    recipientId
  });
  const reqOpts = deleteRecipient(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { recipientId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["recipientId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryRecipients2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryRecipients(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [])
      );
      return {
        items: transformedData?.recipients,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function bulkUpdateRecipientTags2(recipientIds, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    recipientIds,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateRecipientTags(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          recipientIds: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["recipientIds", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateRecipientTagsByFilter2(filter, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    filter,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateRecipientTagsByFilter(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          filter: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["filter", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/restaurants-recipient-v1-recipient-recipients.public.ts
function createRecipient3(httpClient) {
  return (recipient) => createRecipient2(
    recipient,
    // @ts-ignore
    { httpClient }
  );
}
function getRecipient3(httpClient) {
  return (recipientId) => getRecipient2(
    recipientId,
    // @ts-ignore
    { httpClient }
  );
}
function updateRecipient3(httpClient) {
  return (_id, recipient) => updateRecipient2(
    _id,
    recipient,
    // @ts-ignore
    { httpClient }
  );
}
function deleteRecipient3(httpClient) {
  return (recipientId) => deleteRecipient2(
    recipientId,
    // @ts-ignore
    { httpClient }
  );
}
function queryRecipients3(httpClient) {
  return () => queryRecipients2(
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdateRecipientTags3(httpClient) {
  return (recipientIds, options) => bulkUpdateRecipientTags2(
    recipientIds,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdateRecipientTagsByFilter3(httpClient) {
  return (filter, options) => bulkUpdateRecipientTagsByFilter2(
    filter,
    options,
    // @ts-ignore
    { httpClient }
  );
}
var onRecipientCreated = EventDefinition(
  "wix.restaurants.recipient.v1.recipient_created",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onRecipientDeleted = EventDefinition(
  "wix.restaurants.recipient.v1.recipient_deleted",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onRecipientUpdated = EventDefinition(
  "wix.restaurants.recipient.v1.recipient_updated",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();

// src/restaurants-recipient-v1-recipient-recipients.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
import { createEventModule } from "@wix/sdk-runtime/event-definition-modules";
var createRecipient4 = /* @__PURE__ */ createRESTModule(createRecipient3);
var getRecipient4 = /* @__PURE__ */ createRESTModule(getRecipient3);
var updateRecipient4 = /* @__PURE__ */ createRESTModule(updateRecipient3);
var deleteRecipient4 = /* @__PURE__ */ createRESTModule(deleteRecipient3);
var queryRecipients4 = /* @__PURE__ */ createRESTModule(queryRecipients3);
var bulkUpdateRecipientTags4 = /* @__PURE__ */ createRESTModule(bulkUpdateRecipientTags3);
var bulkUpdateRecipientTagsByFilter4 = /* @__PURE__ */ createRESTModule(bulkUpdateRecipientTagsByFilter3);
var onRecipientCreated2 = createEventModule(onRecipientCreated);
var onRecipientDeleted2 = createEventModule(onRecipientDeleted);
var onRecipientUpdated2 = createEventModule(onRecipientUpdated);
export {
  Channel,
  SortOrder,
  WebhookIdentityType,
  bulkUpdateRecipientTags4 as bulkUpdateRecipientTags,
  bulkUpdateRecipientTagsByFilter4 as bulkUpdateRecipientTagsByFilter,
  createRecipient4 as createRecipient,
  deleteRecipient4 as deleteRecipient,
  getRecipient4 as getRecipient,
  onRecipientCreated2 as onRecipientCreated,
  onRecipientDeleted2 as onRecipientDeleted,
  onRecipientUpdated2 as onRecipientUpdated,
  queryRecipients4 as queryRecipients,
  updateRecipient4 as updateRecipient
};
