// @wix/auto_sdk_auth-management_o-auth-apps/build/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  DeleteStatus: () => DeleteStatus,
  Namespace: () => Namespace,
  OAuthAppType: () => OAuthAppType,
  OAuthTechnologies: () => OAuthTechnologies,
  SiteCreatedContext: () => SiteCreatedContext,
  SortOrder: () => SortOrder,
  State: () => State,
  WebhookIdentityType: () => WebhookIdentityType,
  createOAuthApp: () => createOAuthApp4,
  deleteOAuthApp: () => deleteOAuthApp4,
  generateOAuthAppSecret: () => generateOAuthAppSecret4,
  getOAuthApp: () => getOAuthApp4,
  onOAuthAppCreated: () => onOAuthAppCreated2,
  onOAuthAppDeleted: () => onOAuthAppDeleted2,
  onOAuthAppUpdated: () => onOAuthAppUpdated2,
  queryOAuthApps: () => queryOAuthApps4,
  updateOAuthApp: () => updateOAuthApp4
});
module.exports = __toCommonJS(index_exports);

// src/headless-v1-o-auth-app-o-auth-apps.public.ts
var import_rename_all_nested_keys2 = require("@wix/sdk-runtime/rename-all-nested-keys");
var import_timestamp3 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths3 = require("@wix/sdk-runtime/transformations/transform-paths");
var import_sdk_types = require("@wix/sdk-types");

// src/headless-v1-o-auth-app-o-auth-apps.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/headless-v1-o-auth-app-o-auth-apps.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixHeadlessV1OAuthAppServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/oauth-app-service",
        destPath: ""
      },
      {
        srcPath: "/oauth-app",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/oauth-app",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/oauth-app-service",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_auth-management_o-auth-apps";
function createOAuthApp(payload) {
  function __createOAuthApp({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "oAuthApp.createdDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.headless.v1.o_auth_app",
      method: "POST",
      methodFqn: "wix.headless.v1.OAuthAppService.CreateOAuthApp",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixHeadlessV1OAuthAppServiceUrl({
        protoPath: "/v1/oauth-apps",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "oAuthApp.createdDate" }]
        }
      ])
    };
    return metadata;
  }
  return __createOAuthApp;
}
function getOAuthApp(payload) {
  function __getOAuthApp({ host }) {
    const metadata = {
      entityFqdn: "wix.headless.v1.o_auth_app",
      method: "GET",
      methodFqn: "wix.headless.v1.OAuthAppService.GetOAuthApp",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixHeadlessV1OAuthAppServiceUrl({
        protoPath: "/v1/oauth-apps/{oAuthAppId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "oAuthApp.createdDate" }]
        }
      ])
    };
    return metadata;
  }
  return __getOAuthApp;
}
function updateOAuthApp(payload) {
  function __updateOAuthApp({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "oAuthApp.createdDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.headless.v1.o_auth_app",
      method: "PATCH",
      methodFqn: "wix.headless.v1.OAuthAppService.UpdateOAuthApp",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixHeadlessV1OAuthAppServiceUrl({
        protoPath: "/v1/oauth-apps/{oAuthApp.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "oAuthApp.createdDate" }]
        }
      ])
    };
    return metadata;
  }
  return __updateOAuthApp;
}
function deleteOAuthApp(payload) {
  function __deleteOAuthApp({ host }) {
    const metadata = {
      entityFqdn: "wix.headless.v1.o_auth_app",
      method: "DELETE",
      methodFqn: "wix.headless.v1.OAuthAppService.DeleteOAuthApp",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixHeadlessV1OAuthAppServiceUrl({
        protoPath: "/v1/oauth-apps/{oAuthAppId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteOAuthApp;
}
function queryOAuthApps(payload) {
  function __queryOAuthApps({ host }) {
    const metadata = {
      entityFqdn: "wix.headless.v1.o_auth_app",
      method: "POST",
      methodFqn: "wix.headless.v1.OAuthAppService.QueryOAuthApps",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixHeadlessV1OAuthAppServiceUrl({
        protoPath: "/v1/oauth-apps/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "oAuthApps.createdDate" }]
        }
      ])
    };
    return metadata;
  }
  return __queryOAuthApps;
}
function generateOAuthAppSecret(payload) {
  function __generateOAuthAppSecret({ host }) {
    const metadata = {
      entityFqdn: "wix.headless.v1.o_auth_app",
      method: "POST",
      methodFqn: "wix.headless.v1.OAuthAppService.GenerateOAuthAppSecret",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixHeadlessV1OAuthAppServiceUrl({
        protoPath: "/v1/oauth-apps/{oAuthAppId}/generate-secret",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __generateOAuthAppSecret;
}

// src/headless-v1-o-auth-app-o-auth-apps.universal.ts
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var OAuthAppType = /* @__PURE__ */ ((OAuthAppType2) => {
  OAuthAppType2["OAUTH_APP_TYPE_UNSPECIFIED"] = "OAUTH_APP_TYPE_UNSPECIFIED";
  OAuthAppType2["WEB_APP"] = "WEB_APP";
  OAuthAppType2["MOBILE"] = "MOBILE";
  OAuthAppType2["OTHER"] = "OTHER";
  return OAuthAppType2;
})(OAuthAppType || {});
var OAuthTechnologies = /* @__PURE__ */ ((OAuthTechnologies2) => {
  OAuthTechnologies2["OAUTH_TECHNOLOGY_UNSPECIFIED"] = "OAUTH_TECHNOLOGY_UNSPECIFIED";
  OAuthTechnologies2["JAVASCRIPT"] = "JAVASCRIPT";
  OAuthTechnologies2["ANGULAR"] = "ANGULAR";
  OAuthTechnologies2["VUE"] = "VUE";
  OAuthTechnologies2["REACT"] = "REACT";
  OAuthTechnologies2["REACT_NATIVE"] = "REACT_NATIVE";
  OAuthTechnologies2["IOS"] = "IOS";
  OAuthTechnologies2["ANDROID"] = "ANDROID";
  OAuthTechnologies2["OTHER_TECHNOLOGY"] = "OTHER_TECHNOLOGY";
  return OAuthTechnologies2;
})(OAuthTechnologies || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var State = /* @__PURE__ */ ((State2) => {
  State2["UNKNOWN"] = "UNKNOWN";
  State2["ENABLED"] = "ENABLED";
  State2["DISABLED"] = "DISABLED";
  State2["PENDING"] = "PENDING";
  State2["DEMO"] = "DEMO";
  return State2;
})(State || {});
var SiteCreatedContext = /* @__PURE__ */ ((SiteCreatedContext2) => {
  SiteCreatedContext2["OTHER"] = "OTHER";
  SiteCreatedContext2["FROM_TEMPLATE"] = "FROM_TEMPLATE";
  SiteCreatedContext2["DUPLICATE_BY_SITE_TRANSFER"] = "DUPLICATE_BY_SITE_TRANSFER";
  SiteCreatedContext2["DUPLICATE"] = "DUPLICATE";
  SiteCreatedContext2["OLD_SITE_TRANSFER"] = "OLD_SITE_TRANSFER";
  SiteCreatedContext2["FLASH"] = "FLASH";
  return SiteCreatedContext2;
})(SiteCreatedContext || {});
var Namespace = /* @__PURE__ */ ((Namespace2) => {
  Namespace2["UNKNOWN_NAMESPACE"] = "UNKNOWN_NAMESPACE";
  Namespace2["WIX"] = "WIX";
  Namespace2["SHOUT_OUT"] = "SHOUT_OUT";
  Namespace2["ALBUMS"] = "ALBUMS";
  Namespace2["WIX_STORES_TEST_DRIVE"] = "WIX_STORES_TEST_DRIVE";
  Namespace2["HOTELS"] = "HOTELS";
  Namespace2["CLUBS"] = "CLUBS";
  Namespace2["ONBOARDING_DRAFT"] = "ONBOARDING_DRAFT";
  Namespace2["DEV_SITE"] = "DEV_SITE";
  Namespace2["LOGOS"] = "LOGOS";
  Namespace2["VIDEO_MAKER"] = "VIDEO_MAKER";
  Namespace2["PARTNER_DASHBOARD"] = "PARTNER_DASHBOARD";
  Namespace2["DEV_CENTER_COMPANY"] = "DEV_CENTER_COMPANY";
  Namespace2["HTML_DRAFT"] = "HTML_DRAFT";
  Namespace2["SITELESS_BUSINESS"] = "SITELESS_BUSINESS";
  Namespace2["CREATOR_ECONOMY"] = "CREATOR_ECONOMY";
  Namespace2["DASHBOARD_FIRST"] = "DASHBOARD_FIRST";
  Namespace2["ANYWHERE"] = "ANYWHERE";
  Namespace2["HEADLESS"] = "HEADLESS";
  Namespace2["ACCOUNT_MASTER_CMS"] = "ACCOUNT_MASTER_CMS";
  Namespace2["RISE"] = "RISE";
  Namespace2["BRANDED_FIRST"] = "BRANDED_FIRST";
  Namespace2["NOWNIA"] = "NOWNIA";
  Namespace2["UGC_TEMPLATE"] = "UGC_TEMPLATE";
  Namespace2["CODUX"] = "CODUX";
  Namespace2["MEDIA_DESIGN_CREATOR"] = "MEDIA_DESIGN_CREATOR";
  Namespace2["SHARED_BLOG_ENTERPRISE"] = "SHARED_BLOG_ENTERPRISE";
  Namespace2["STANDALONE_FORMS"] = "STANDALONE_FORMS";
  Namespace2["STANDALONE_EVENTS"] = "STANDALONE_EVENTS";
  Namespace2["MIMIR"] = "MIMIR";
  Namespace2["TWINS"] = "TWINS";
  Namespace2["NANO"] = "NANO";
  return Namespace2;
})(Namespace || {});
var DeleteStatus = /* @__PURE__ */ ((DeleteStatus2) => {
  DeleteStatus2["UNKNOWN"] = "UNKNOWN";
  DeleteStatus2["TRASH"] = "TRASH";
  DeleteStatus2["DELETED"] = "DELETED";
  DeleteStatus2["PENDING_PURGE"] = "PENDING_PURGE";
  DeleteStatus2["PURGED_EXTERNALLY"] = "PURGED_EXTERNALLY";
  return DeleteStatus2;
})(DeleteStatus || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createOAuthApp2(oAuthApp) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ oAuthApp });
  const reqOpts = createOAuthApp(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.oAuthApp;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { oAuthApp: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["oAuthApp"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getOAuthApp2(oAuthAppId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    oAuthAppId
  });
  const reqOpts = getOAuthApp(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.oAuthApp;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { oAuthAppId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["oAuthAppId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateOAuthApp2(_id, oAuthApp, mask) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    oAuthApp: { ...oAuthApp, id: _id },
    mask
  });
  const reqOpts = updateOAuthApp(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.oAuthApp;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { oAuthApp: "$[1]" },
        explicitPathsToArguments: { "oAuthApp.id": "$[0]", mask: "$[2]" },
        singleArgumentUnchanged: false
      },
      ["_id", "oAuthApp", "mask"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteOAuthApp2(oAuthAppId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    oAuthAppId
  });
  const reqOpts = deleteOAuthApp(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { oAuthAppId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["oAuthAppId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryOAuthApps2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryOAuthApps(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [])
      );
      return {
        items: transformedData?.oAuthApps,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "OFFSET",
    transformationPaths: {}
  });
}
async function generateOAuthAppSecret2(oAuthAppId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    oAuthAppId
  });
  const reqOpts = generateOAuthAppSecret(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { oAuthAppId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["oAuthAppId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/headless-v1-o-auth-app-o-auth-apps.public.ts
function createOAuthApp3(httpClient) {
  return (oAuthApp) => createOAuthApp2(
    oAuthApp,
    // @ts-ignore
    { httpClient }
  );
}
function getOAuthApp3(httpClient) {
  return (oAuthAppId) => getOAuthApp2(
    oAuthAppId,
    // @ts-ignore
    { httpClient }
  );
}
function updateOAuthApp3(httpClient) {
  return (_id, oAuthApp, mask) => updateOAuthApp2(
    _id,
    oAuthApp,
    mask,
    // @ts-ignore
    { httpClient }
  );
}
function deleteOAuthApp3(httpClient) {
  return (oAuthAppId) => deleteOAuthApp2(
    oAuthAppId,
    // @ts-ignore
    { httpClient }
  );
}
function queryOAuthApps3(httpClient) {
  return () => queryOAuthApps2(
    // @ts-ignore
    { httpClient }
  );
}
function generateOAuthAppSecret3(httpClient) {
  return (oAuthAppId) => generateOAuthAppSecret2(
    oAuthAppId,
    // @ts-ignore
    { httpClient }
  );
}
var onOAuthAppCreated = (0, import_sdk_types.EventDefinition)(
  "wix.headless.v1.o_auth_app_created",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onOAuthAppDeleted = (0, import_sdk_types.EventDefinition)(
  "wix.headless.v1.o_auth_app_deleted",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "undefined.createdDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onOAuthAppUpdated = (0, import_sdk_types.EventDefinition)(
  "wix.headless.v1.o_auth_app_updated",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();

// src/headless-v1-o-auth-app-o-auth-apps.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var import_event_definition_modules = require("@wix/sdk-runtime/event-definition-modules");
var createOAuthApp4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(createOAuthApp3);
var getOAuthApp4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getOAuthApp3);
var updateOAuthApp4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(updateOAuthApp3);
var deleteOAuthApp4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(deleteOAuthApp3);
var queryOAuthApps4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(queryOAuthApps3);
var generateOAuthAppSecret4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(generateOAuthAppSecret3);
var onOAuthAppCreated2 = (0, import_event_definition_modules.createEventModule)(onOAuthAppCreated);
var onOAuthAppDeleted2 = (0, import_event_definition_modules.createEventModule)(onOAuthAppDeleted);
var onOAuthAppUpdated2 = (0, import_event_definition_modules.createEventModule)(onOAuthAppUpdated);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  DeleteStatus,
  Namespace,
  OAuthAppType,
  OAuthTechnologies,
  SiteCreatedContext,
  SortOrder,
  State,
  WebhookIdentityType,
  createOAuthApp,
  deleteOAuthApp,
  generateOAuthAppSecret,
  getOAuthApp,
  onOAuthAppCreated,
  onOAuthAppDeleted,
  onOAuthAppUpdated,
  queryOAuthApps,
  updateOAuthApp
});
