// @wix/ricos/dist/cjs/plugin-ai/ai-flow-selection-modal/ai-flow-selection-modal.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var ai_flow_selection_modal_exports = {};
__export(ai_flow_selection_modal_exports, {
  AiFlowSelectionModal: () => AiFlowSelectionModal
});
module.exports = __toCommonJS(ai_flow_selection_modal_exports);
var import_design_system = require("@wix/design-system");
var import_wix_ui_icons_common = require("@wix/wix-ui-icons-common");
var import_react = __toESM(require("react"));
var import_ricos_context = require("./../../ricos-context/index.js");
var import_data_hooks = require("../data-hooks.js");
var import_ai_flow_selection_modal_scss = __toESM(require("./ai-flow-selection-modal-scss.js"));
const AiFlowSelectionModal = ({
  toggleAiContentModal,
  toggleAiImageModal
}) => {
  const { t } = (0, import_react.useContext)(import_ricos_context.RicosContext);
  const modalService = (0, import_react.useContext)(import_ricos_context.ModalServiceContext);
  const { adapter } = (0, import_react.useContext)(import_ricos_context.EditorContext);
  const options = import_react.default.useMemo(() => {
    const commonOptions = {
      as: "button",
      ellipsis: false,
      size: "small",
      skin: "standard",
      title: false,
      value: "action",
      shouldFocusWithoutScroll: true
    };
    return [
      {
        ...commonOptions,
        dataHook: import_data_hooks.AI_FLOW_ITEM_CONTENT,
        id: "ai-content",
        onClick: () => toggleAiContentModal(adapter.tiptapEditor, modalService),
        optionTitle: t("AIPlugin_Flow_ContentAssistant"),
        prefixIcon: /* @__PURE__ */ import_react.default.createElement(import_wix_ui_icons_common.EditSparklesSmall, null)
      },
      {
        ...commonOptions,
        dataHook: import_data_hooks.AI_FLOW_ITEM_IMAGE,
        id: "ai-image",
        onClick: () => toggleAiImageModal(adapter.tiptapEditor, modalService),
        optionTitle: t("AIPlugin_Flow_ImageGenerator"),
        prefixIcon: /* @__PURE__ */ import_react.default.createElement(import_wix_ui_icons_common.ImageSparklesSmall, null)
      }
    ];
  }, []);
  return /* @__PURE__ */ import_react.default.createElement(
    import_design_system.DropdownLayout,
    {
      dataHook: import_data_hooks.AI_FLOW_SELECTION_POPOVER,
      className: import_ai_flow_selection_modal_scss.default.dropdown,
      maxHeightPixels: 250,
      minWidthPixels: 206,
      visible: true,
      options
    }
  );
};
