// @wix/patterns/dist/cjs/components/EntityPage/EntityPageActionsBar.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.EntityPageActionsBar = void 0;
exports._EntityPageActionsBar = _EntityPageActionsBar;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireDefault(require("react"));
var _providers = require("../../providers");
var _mobxReactLite = require("mobx-react-lite");
var _FormPage = require("../FormPage");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/EntityPage/EntityPageActionsBar.tsx";
function _EntityPageActionsBar(props) {
  const state = (0, _providers.useEntityPageContext)();
  return /*#__PURE__*/_react.default.createElement(_FormPage.FormPageActionsBar, (0, _extends2.default)({
    onCancel: state.onCancel.bind(state),
    onSave: state.handleSubmit.bind(state),
    actionsBarConfig: state.actionsBarConfig
  }, props, {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 15,
      columnNumber: 5
    }
  }));
}
const EntityPageActionsBar = exports.EntityPageActionsBar = (0, _mobxReactLite.observer)(_EntityPageActionsBar);
