// @wix/auto_sdk_events_staff-members/build/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  Fieldset: () => Fieldset,
  Permission: () => Permission,
  Scope: () => Scope,
  SortOrder: () => SortOrder,
  Status: () => Status,
  WebhookIdentityType: () => WebhookIdentityType,
  accessStaffMember: () => accessStaffMember4,
  createStaffMember: () => createStaffMember4,
  deleteStaffMember: () => deleteStaffMember4,
  getStaffMember: () => getStaffMember4,
  joinStaffMember: () => joinStaffMember4,
  loginStaffMember: () => loginStaffMember4,
  onStaffMemberCreated: () => onStaffMemberCreated2,
  onStaffMemberDeleted: () => onStaffMemberDeleted2,
  onStaffMemberUpdated: () => onStaffMemberUpdated2,
  queryStaffMembers: () => queryStaffMembers4,
  updateStaffMember: () => updateStaffMember4
});
module.exports = __toCommonJS(index_exports);

// src/events-staffmembers-v1-staff-member-staff-members.public.ts
var import_rename_all_nested_keys2 = require("@wix/sdk-runtime/rename-all-nested-keys");
var import_timestamp3 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths3 = require("@wix/sdk-runtime/transformations/transform-paths");
var import_sdk_types = require("@wix/sdk-types");

// src/events-staffmembers-v1-staff-member-staff-members.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/events-staffmembers-v1-staff-member-staff-members.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEventsStaffmembersV1StaffMemberManagementUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/events-staff-members",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/events-staff-members",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/events-staff-members",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/events-staff-members",
        destPath: ""
      },
      {
        srcPath: "/_api/events-staff-members",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/events-staff-members",
        destPath: ""
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/events-staff-members",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/events-staff-members",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/events/v1/events-staff-members",
        destPath: "/v1/events-staff-members"
      },
      {
        srcPath: "/events/v1/staff-members",
        destPath: "/v1/staff-members"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_events_staff-members";
function createStaffMember(payload) {
  function __createStaffMember({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "staffMember.createdDate" },
          { path: "staffMember.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.staffmembers.v1.staff_member",
      method: "POST",
      methodFqn: "wix.events.staffmembers.v1.StaffMemberManagement.CreateStaffMember",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsStaffmembersV1StaffMemberManagementUrl({
        protoPath: "/v1/staff-members",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "staffMember.createdDate" },
            { path: "staffMember.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createStaffMember;
}
function getStaffMember(payload) {
  function __getStaffMember({ host }) {
    const metadata = {
      entityFqdn: "wix.events.staffmembers.v1.staff_member",
      method: "GET",
      methodFqn: "wix.events.staffmembers.v1.StaffMemberManagement.GetStaffMember",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsStaffmembersV1StaffMemberManagementUrl({
        protoPath: "/v1/staff-members/{staffMemberId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "staffMember.createdDate" },
            { path: "staffMember.updatedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "GET",
          url: resolveWixEventsStaffmembersV1StaffMemberManagementUrl({
            protoPath: "/v1/staff-members/{staffMemberId}",
            data: payload,
            host
          }),
          params: (0, import_rest_modules.toURLSearchParams)(payload)
        }
      ]
    };
    return metadata;
  }
  return __getStaffMember;
}
function updateStaffMember(payload) {
  function __updateStaffMember({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "staffMember.createdDate" },
          { path: "staffMember.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.staffmembers.v1.staff_member",
      method: "PATCH",
      methodFqn: "wix.events.staffmembers.v1.StaffMemberManagement.UpdateStaffMember",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsStaffmembersV1StaffMemberManagementUrl({
        protoPath: "/v1/staff-members/{staffMember.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "staffMember.createdDate" },
            { path: "staffMember.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateStaffMember;
}
function deleteStaffMember(payload) {
  function __deleteStaffMember({ host }) {
    const metadata = {
      entityFqdn: "wix.events.staffmembers.v1.staff_member",
      method: "DELETE",
      methodFqn: "wix.events.staffmembers.v1.StaffMemberManagement.DeleteStaffMember",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsStaffmembersV1StaffMemberManagementUrl({
        protoPath: "/v1/staff-members/{staffMemberId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteStaffMember;
}
function queryStaffMembers(payload) {
  function __queryStaffMembers({ host }) {
    const metadata = {
      entityFqdn: "wix.events.staffmembers.v1.staff_member",
      method: "POST",
      methodFqn: "wix.events.staffmembers.v1.StaffMemberManagement.QueryStaffMembers",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsStaffmembersV1StaffMemberManagementUrl({
        protoPath: "/v1/staff-members/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "staffMembers.createdDate" },
            { path: "staffMembers.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryStaffMembers;
}
function joinStaffMember(payload) {
  function __joinStaffMember({ host }) {
    const metadata = {
      entityFqdn: "wix.events.staffmembers.v1.staff_member",
      method: "POST",
      methodFqn: "wix.events.staffmembers.v1.StaffMemberManagement.JoinStaffMember",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsStaffmembersV1StaffMemberManagementUrl({
        protoPath: "/v1/staff-members/join",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "staffMember.createdDate" },
            { path: "staffMember.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __joinStaffMember;
}
function loginStaffMember(payload) {
  function __loginStaffMember({ host }) {
    const metadata = {
      entityFqdn: "wix.events.staffmembers.v1.staff_member",
      method: "POST",
      methodFqn: "wix.events.staffmembers.v1.StaffMemberManagement.LoginStaffMember",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsStaffmembersV1StaffMemberManagementUrl({
        protoPath: "/v1/staff-members/login",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "staffMember.createdDate" },
            { path: "staffMember.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __loginStaffMember;
}
function accessStaffMember(payload) {
  function __accessStaffMember({ host }) {
    const metadata = {
      entityFqdn: "wix.events.staffmembers.v1.staff_member",
      method: "POST",
      methodFqn: "wix.events.staffmembers.v1.StaffMemberManagement.AccessStaffMember",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsStaffmembersV1StaffMemberManagementUrl({
        protoPath: "/v1/staff-members/access",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "staffMember.createdDate" },
            { path: "staffMember.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __accessStaffMember;
}

// src/events-staffmembers-v1-staff-member-staff-members.universal.ts
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var Scope = /* @__PURE__ */ ((Scope2) => {
  Scope2["UNKNOWN_SCOPE"] = "UNKNOWN_SCOPE";
  Scope2["CHECK_IN_GUESTS"] = "CHECK_IN_GUESTS";
  Scope2["SELL_TICKETS"] = "SELL_TICKETS";
  Scope2["ADD_GUESTS"] = "ADD_GUESTS";
  Scope2["READ_ANALYTICS"] = "READ_ANALYTICS";
  return Scope2;
})(Scope || {});
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["PENDING"] = "PENDING";
  Status2["HAS_ACCESS"] = "HAS_ACCESS";
  Status2["ACCESS_PAUSED"] = "ACCESS_PAUSED";
  return Status2;
})(Status || {});
var Permission = /* @__PURE__ */ ((Permission2) => {
  Permission2["UNKNOWN_PERMISSION"] = "UNKNOWN_PERMISSION";
  Permission2["CHECK_IN_GUESTS"] = "CHECK_IN_GUESTS";
  Permission2["SELL_TICKETS"] = "SELL_TICKETS";
  Permission2["ADD_GUESTS"] = "ADD_GUESTS";
  Permission2["READ_ANALYTICS"] = "READ_ANALYTICS";
  Permission2["READ_GUEST_FORM"] = "READ_GUEST_FORM";
  return Permission2;
})(Permission || {});
var Fieldset = /* @__PURE__ */ ((Fieldset2) => {
  Fieldset2["LOGIN_LINK"] = "LOGIN_LINK";
  return Fieldset2;
})(Fieldset || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createStaffMember2(staffMember) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    staffMember
  });
  const reqOpts = createStaffMember(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.staffMember;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { staffMember: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["staffMember"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getStaffMember2(staffMemberId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    staffMemberId,
    fieldsets: options?.fieldsets
  });
  const reqOpts = getStaffMember(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.staffMember;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          staffMemberId: "$[0]",
          fieldsets: "$[1].fieldsets"
        },
        singleArgumentUnchanged: false
      },
      ["staffMemberId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateStaffMember2(_id, staffMember) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    staffMember: { ...staffMember, id: _id }
  });
  const reqOpts = updateStaffMember(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.staffMember;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { staffMember: "$[1]" },
        explicitPathsToArguments: { "staffMember.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "staffMember"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteStaffMember2(staffMemberId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    staffMemberId
  });
  const reqOpts = deleteStaffMember(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { staffMemberId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["staffMemberId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryStaffMembers2(options) {
  const { httpClient, sideEffects } = arguments[1];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryStaffMembers({
        ...payload,
        ...options ?? {}
      });
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, options];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [])
      );
      return {
        items: transformedData?.staffMembers,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function joinStaffMember2(joinToken) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    joinToken
  });
  const reqOpts = joinStaffMember(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { joinToken: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["joinToken"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function loginStaffMember2(loginToken) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    loginToken
  });
  const reqOpts = loginStaffMember(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { loginToken: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["loginToken"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function accessStaffMember2(accessToken) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    accessToken
  });
  const reqOpts = accessStaffMember(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { accessToken: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["accessToken"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/events-staffmembers-v1-staff-member-staff-members.public.ts
function createStaffMember3(httpClient) {
  return (staffMember) => createStaffMember2(
    staffMember,
    // @ts-ignore
    { httpClient }
  );
}
function getStaffMember3(httpClient) {
  return (staffMemberId, options) => getStaffMember2(
    staffMemberId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function updateStaffMember3(httpClient) {
  return (_id, staffMember) => updateStaffMember2(
    _id,
    staffMember,
    // @ts-ignore
    { httpClient }
  );
}
function deleteStaffMember3(httpClient) {
  return (staffMemberId) => deleteStaffMember2(
    staffMemberId,
    // @ts-ignore
    { httpClient }
  );
}
function queryStaffMembers3(httpClient) {
  return (options) => queryStaffMembers2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function joinStaffMember3(httpClient) {
  return (joinToken) => joinStaffMember2(
    joinToken,
    // @ts-ignore
    { httpClient }
  );
}
function loginStaffMember3(httpClient) {
  return (loginToken) => loginStaffMember2(
    loginToken,
    // @ts-ignore
    { httpClient }
  );
}
function accessStaffMember3(httpClient) {
  return (accessToken) => accessStaffMember2(
    accessToken,
    // @ts-ignore
    { httpClient }
  );
}
var onStaffMemberCreated = (0, import_sdk_types.EventDefinition)(
  "wix.events.staffmembers.v1.staff_member_created",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onStaffMemberDeleted = (0, import_sdk_types.EventDefinition)(
  "wix.events.staffmembers.v1.staff_member_deleted",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onStaffMemberUpdated = (0, import_sdk_types.EventDefinition)(
  "wix.events.staffmembers.v1.staff_member_updated",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();

// src/events-staffmembers-v1-staff-member-staff-members.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var import_event_definition_modules = require("@wix/sdk-runtime/event-definition-modules");
var createStaffMember4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(createStaffMember3);
var getStaffMember4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getStaffMember3);
var updateStaffMember4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(updateStaffMember3);
var deleteStaffMember4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(deleteStaffMember3);
var queryStaffMembers4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(queryStaffMembers3);
var joinStaffMember4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(joinStaffMember3);
var loginStaffMember4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(loginStaffMember3);
var accessStaffMember4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(accessStaffMember3);
var onStaffMemberCreated2 = (0, import_event_definition_modules.createEventModule)(
  onStaffMemberCreated
);
var onStaffMemberDeleted2 = (0, import_event_definition_modules.createEventModule)(
  onStaffMemberDeleted
);
var onStaffMemberUpdated2 = (0, import_event_definition_modules.createEventModule)(
  onStaffMemberUpdated
);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Fieldset,
  Permission,
  Scope,
  SortOrder,
  Status,
  WebhookIdentityType,
  accessStaffMember,
  createStaffMember,
  deleteStaffMember,
  getStaffMember,
  joinStaffMember,
  loginStaffMember,
  onStaffMemberCreated,
  onStaffMemberDeleted,
  onStaffMemberUpdated,
  queryStaffMembers,
  updateStaffMember
});
