// @wix/headless-restaurants-olo/dist/react/ItemDetails.js
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import React from 'react';
import { Commerce } from '@wix/ecom/components';
import { AsChildSlot } from '@wix/headless-utils/react';
import { Quantity as QuantityComponent } from '@wix/headless-components/react';
import { Item, Label, Modifier, ModifierGroup, Variant, } from '@wix/restaurants/components';
import * as CoreItemDetails from './core/ItemDetails.js';
var TestIds;
(function (TestIds) {
    TestIds["itemName"] = "item-name";
    TestIds["itemPrice"] = "item-price";
    TestIds["itemDescription"] = "item-description";
    TestIds["itemImage"] = "item-image";
    TestIds["itemAddToCart"] = "item-add-to-cart";
    TestIds["itemSpecialRequest"] = "item-special-request";
    TestIds["itemLabels"] = "item-labels";
    TestIds["itemModifierGroups"] = "item-modifier-groups";
    TestIds["itemVariants"] = "item-variants";
})(TestIds || (TestIds = {}));
export const Root = ({ children, itemDetailsServiceConfig }) => {
    return (_jsx(CoreItemDetails.Root, { itemDetailsServiceConfig: itemDetailsServiceConfig, children: ({ item }) => _jsx(Item.Root, { item: item, children: children }) }));
};
export const Name = React.forwardRef(({ asChild, children, className, ...rest }, ref) => {
    return (_jsx(Item.Name, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.itemName, ...rest, children: children }));
});
Name.displayName = 'ItemDetails.Name';
export const Image = React.forwardRef(({ asChild, children, className, ...rest }) => {
    return (_jsx(Item.Image, { asChild: asChild, className: className, "data-testid": TestIds.itemImage, ...rest, children: children }));
});
Image.displayName = 'ItemDetails.Image';
export const Price = React.forwardRef(({ asChild, className, ...rest }, ref) => {
    return (_jsx(Item.Price, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.itemPrice, ...rest }));
});
Price.displayName = 'ItemDetails.Price';
export const Description = React.forwardRef(({ asChild, className, ...rest }, ref) => {
    return (_jsx(Item.Description, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.itemDescription, ...rest }));
});
Description.displayName = 'ItemDetails.Description';
export const Labels = React.forwardRef(({ children, className, asChild, iconClassName, nameClassName, ...rest }) => {
    return (_jsx(Item.LabelsRepeater, { ...rest, children: _jsxs(AsChildSlot, { asChild: asChild, testId: TestIds.itemLabels, className: className, customElement: children, children: [_jsx(Label.Icon, { className: iconClassName }), _jsx(Label.Name, { className: nameClassName })] }) }));
});
Labels.displayName = 'ItemDetails.Labels';
export const ModifierGroups = React.forwardRef(({ children, className, asChild, modifierNameClassName, modifierPriceClassName, ...rest }) => {
    return (_jsx(Item.ModifierGroupsRepeater, { ...rest, children: _jsx(AsChildSlot, { asChild: asChild, testId: TestIds.itemModifierGroups, className: className, customElement: children, children: _jsxs(ModifierGroup.ModifiersRepeater, { children: [_jsx(Modifier.Name, { className: modifierNameClassName }), _jsx(Modifier.Price, { className: modifierPriceClassName })] }) }) }));
});
export const Variants = React.forwardRef(({ children, className, asChild, variantNameClassName, variantPriceClassName, ...rest }) => {
    return (_jsx(Item.VariantsRepeater, { ...rest, children: _jsxs(AsChildSlot, { asChild: asChild, testId: TestIds.itemVariants, className: className, customElement: children, children: [_jsx(Variant.Name, { className: variantNameClassName }), _jsx(Variant.Price, { className: variantPriceClassName })] }) }));
});
Variants.displayName = 'ItemDetails.Variants';
ModifierGroups.displayName = 'ItemDetails.ModifierGroups';
/**
 * Add to Cart button for the menu item.
 * Triggers the action to add the selected item (and its modifiers/variants) to the cart.
 *
 * @component
 * @example
 * ```tsx
 * <ItemDetails>
 *   <AddToCart
 *     label="Add to Cart"
 *     lineItems={[{ catalogReference: { ... }, quantity: 1 }]}
 *   />
 * </ItemDetails>
 * ```
 */
export const AddToCart = React.forwardRef(({ lineItems, className, label, ...props }, ref) => {
    return (_jsx(Commerce.Actions.AddToCart, { ref: ref, asChild: false, label: label, className: className, lineItems: lineItems, children: props.children }));
});
AddToCart.displayName = 'AddToCart';
export const AddToCartButton = ({ asChild = false, children, className, onClick, label = 'Add to cart', ...props }) => {
    return (_jsx(CoreItemDetails.LineItemComponent, { children: ({ lineItem }) => (_jsx(AsChildSlot, { asChild: asChild, className: className, customElement: children, customElementProps: {
                onClick,
            }, children: _jsx(Commerce.Actions.AddToCart, { asChild: false, label: label, className: className, lineItems: [lineItem], ...props, children: children }) })) }));
};
export const Quantity = ({ children }) => {
    return (_jsx(CoreItemDetails.QuantityComponent, { children: ({ quantity, onValueChange, }) => (_jsx(QuantityComponent.Root, { onValueChange: onValueChange, initialValue: quantity, children: children })) }));
};
Quantity.displayName = 'Quantity';
export const SpecialRequest = React.forwardRef(({ className, labelClassName, placeholder = 'Any special requests or dietary restrictions?', maxLength = 200, rows = 3, label = 'Special Requests', asChild, children, ...props }, ref) => {
    return (_jsx(CoreItemDetails.SpecialRequest, { children: ({ value, onChange, }) => (_jsxs(AsChildSlot, { ref: ref, asChild: asChild, className: className, onChange: onChange, "data-testid": TestIds.itemSpecialRequest, customElement: children, customElementProps: { label, value }, content: value, ...props, children: [label && _jsx("label", { className: labelClassName, children: label }), _jsx("textarea", { value: value, onChange: (e) => onChange(e.target.value), placeholder: placeholder, maxLength: maxLength, rows: rows, className: className, children: value })] })) }));
});
SpecialRequest.displayName = 'SpecialRequest';
