// @wix/headless-restaurants-olo/dist/react/OLO.js
import { jsx as _jsx, Fragment as _Fragment, jsxs as _jsxs } from "react/jsx-runtime";
import React from 'react';
import { CoreOLO } from './core/index.js';
/**
 * Root headless component for OLO service management
 * Wraps CoreOLO.Root and provides service state to children
 *
 * @example
 * ```tsx
 * <OLO.Root itemId="item-123">
 *   {({ isLoading, hasServices, error, retry }) => (
 *     isLoading ? (
 *       <LoadingSpinner />
 *     ) : error ? (
 *       <ErrorMessage error={error} onRetry={retry} />
 *     ) : hasServices ? (
 *       <ItemDetailsComponents />
 *     ) : null
 *   )}
 * </OLO.Root>
 * ```
 */
export const Root = ({ itemId, oloSettingsServiceConfig, children, }) => {
    const [retryKey] = React.useState(0);
    // const retry = React.useCallback(() => {
    //   setRetryKey(prev => prev + 1);
    // }, []);
    return (_jsx(CoreOLO.Root, { itemId: itemId, oloSettingsServiceConfig: oloSettingsServiceConfig, children: children }, retryKey));
};
/**
 * Convenience provider that handles loading and error states automatically
 *
 * @example
 * ```tsx
 * <OLO.Provider
 *   itemId="item-123"
 *   loading={<Spinner />}
 *   error={({ error, retry }) => <ErrorBanner message={error} onRetry={retry} />}
 * >
 *   <ItemDetailsComponents />
 * </OLO.Provider>
 * ```
 */
export const Provider = ({ itemId, configs, 
// loading = (
//   <div className="flex items-center justify-center p-8">
//     <div className="text-secondary-foreground">Loading services...</div>
//   </div>
// ),
// error: errorComponent = ({ error, retry }) => (
//   <div className="flex flex-col items-center justify-center p-8 space-y-4">
//     <div className="text-destructive">Error: {error}</div>
//     <button
//       onClick={retry}
//       className="px-4 py-2 bg-primary text-primary-foreground rounded hover:bg-primary/90"
//     >
//       Retry
//     </button>
//   </div>
// ),
children, }) => {
    return (_jsx(Root, { itemId: itemId, itemServiceConfig: configs?.itemServiceConfig, children: children }));
};
/**
 * Specialized headless component for item details
 * Includes item-specific error handling (like 404 not found)
 *
 * @example
 * ```tsx
 * <OLO.ItemDetails
 *   itemId="item-123"
 *   notFound={<NotFoundPage />}
 * >
 *   <ItemDetailsUI />
 * </OLO.ItemDetails>
 * ```
 */
export const ItemDetails = ({ itemId, configs, loading, error, notFound = (_jsx("div", { className: "flex items-center justify-center p-8", children: _jsx("div", { className: "text-secondary-foreground", children: "Item not found" }) })), children, }) => {
    return (_jsx(Provider, { itemId: itemId, configs: configs, loading: loading, error: ({ error: errorMessage, retry }) => {
            // Handle item not found specifically
            if (errorMessage === 'Item not found') {
                return _jsx(_Fragment, { children: notFound });
            }
            // Use custom error component if provided
            if (error) {
                return _jsx(_Fragment, { children: error({ error: errorMessage, retry }) });
            }
            // Default error handling
            return (_jsxs("div", { className: "flex flex-col items-center justify-center p-8 space-y-4", children: [_jsxs("div", { className: "text-destructive", children: ["Error: ", errorMessage] }), _jsx("button", { onClick: retry, className: "px-4 py-2 bg-primary text-primary-foreground rounded hover:bg-primary/90", children: "Retry" })] }));
        }, children: children }));
};
/**
 * Headless component for cart-only functionality
 * Doesn't load item services, only cart services
 *
 * @example
 * ```tsx
 * <OLO.Cart>
 *   <CartComponents />
 * </OLO.Cart>
 * ```
 */
export const Cart = ({ loading, error, children }) => {
    return (_jsx(Provider, { loading: loading, error: error, children: children }));
};
/**
 * Headless component for checking service status
 * Useful for debugging or conditional rendering
 *
 * @example
 * ```tsx
 * <OLO.ServicesStatus itemId="item-123">
 *   {({ hasItemService, hasCartService, error }) => (
 *     <div>
 *       Item Service: {hasItemService ? '✅' : '❌'}
 *       Cart Service: {hasCartService ? '✅' : '❌'}
 *       {error && <div>Error: {error}</div>}
 *     </div>
 *   )}
 * </OLO.ServicesStatus>
 * ```
 */
export const ServicesStatus = ({ itemId, children, }) => {
    return (_jsx(Root, { itemId: itemId, children: children }));
};
export const Menus = ({ 
// menuServiceConfig,
children, }) => {
    // const [retryKey, setRetryKey] = React.useState(0);
    // const [menus, setMenus] = React.useState<any[]>([]);
    // const [isLoading, setIsLoading] = React.useState(true);
    // const [error, setError] = React.useState<string | undefined>(undefined);
    return _jsx(_Fragment, { children: children });
};
