// @wix/patterns/dist/esm/test-utils/AsyncJobMocks.js
import { Chance } from 'chance';
import { getAsyncJob } from '@wix/bex-utils/@wix/ambassador-infra-asyncjobs-v1-async-job/http';
const chance = new Chance();
export const anAsyncJob = (overrides = {}) => ({
    id: `${chance.word()}_${chance.integer({ min: 0, max: 1000 })}`,
    status: 'FINISHED',
    ...overrides,
});
export const resolveAsyncJob = ({ whenRequest, job, spy, persist = true, }) => {
    const request = whenRequest(getAsyncJob).reply(200, async () => {
        if (spy) {
            spy();
        }
        const data = {
            job,
        };
        return data;
    });
    if (persist) {
        return request.persist();
    }
    return request;
};
export const rejectAsyncJob = ({ whenRequest, persist = true, }) => {
    const request = whenRequest(getAsyncJob).reply(400, async () => {
        return new Promise((_, reject) => window.setTimeout(() => {
            return reject(new Error('Async job failed'));
        }, 0));
    });
    if (persist) {
        return request.persist();
    }
    return request;
};
export function asyncJobMocks(whenRequest) {
    return {
        mocks: [
            whenRequest(getAsyncJob)
                .reply(200, async () => {
                const res = {
                    job: anAsyncJob(),
                };
                return res;
            })
                .persist(),
        ],
    };
}
