// @wix/patterns/dist/esm/test-utils/dataExtensionServiceMocks.js
import { listDataExtensionSchemas } from '@wix/bex-utils/@wix/ambassador-data-extensions-v1-data-extension-schema/http';
import { Chance } from 'chance';
export function dataExtensionServiceMocks(whenRequest) {
    return {
        mocks: [
            whenRequest(listDataExtensionSchemas)
                .reply(200, async () => {
                const res = {
                    dataExtensionSchemas: [],
                };
                return res;
            })
                .persist(),
        ],
    };
}
export const aDataExtensionField = (overrides = {}) => {
    const chance = new Chance();
    return {
        id: `${chance.word()}_${chance.integer({ min: 0, max: 1000 })}`,
        title: chance.name(),
        type: 'string',
        'x-wix-permissions': {
            read: ['apps', 'users'],
            write: ['apps'],
        },
        'x-wix-created-date': '2023-06-05T11:15:39+00:00',
        ...overrides,
    };
};
export const resolveDataExtensions = ({ whenRequest, fqdn, fields, extensionPoint, persist = true, }) => {
    const chance = new Chance();
    const request = whenRequest(listDataExtensionSchemas).reply(200, async () => {
        return {
            dataExtensionSchemas: [
                {
                    id: chance.guid(),
                    fqdn,
                    extensionPoint,
                    namespace: '_user_fields',
                    jsonSchema: {
                        properties: fields.reduce((acc, curr) => {
                            acc[curr.id] = curr;
                            return acc;
                        }, {}),
                    },
                },
            ],
        };
    });
    if (persist) {
        return request.persist();
    }
    return request;
};
