// @wix/ricos/dist/cjs/plugin-table/toolbar/buttons/BackgroundColorButton.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var BackgroundColorButton_exports = {};
__export(BackgroundColorButton_exports, {
  BackgroundColorButton: () => BackgroundColorButton
});
module.exports = __toCommonJS(BackgroundColorButton_exports);
var import_wix_ui_icons_common = require("@wix/wix-ui-icons-common");
var import_react = __toESM(require("react"));
var import_ricos_context = require("./../../../ricos-context/index.js");
var import_toolbar_button = require("./../../../_shared/toolbar-button/toolbar-button.js");
var import_use_color_picker_modal = require("../../modals/use-color-picker-modal.js");
const BackgroundColorButton = ({ dataHook, toolbarItem }) => {
  const { t } = (0, import_react.useContext)(import_ricos_context.RicosContext);
  const [referenceElement, setReferenceElement] = (0, import_react.useState)(null);
  const colorPickerModal = (0, import_use_color_picker_modal.useColorPickerModal)();
  return /* @__PURE__ */ import_react.default.createElement(
    import_toolbar_button.ToolbarButton,
    {
      ref: setReferenceElement,
      prefix: import_wix_ui_icons_common.ColorBucket,
      onClick: () => colorPickerModal.toggleModal(referenceElement, {
        nodeId: toolbarItem?.attributes?.selectedNode?.attrs?.id,
        color: toolbarItem.attributes.backgroundColor,
        onChange: (color) => toolbarItem.commands.click({ backgroundColor: color })
      }),
      dataHook,
      tooltip: t("TablePlugin_Toolbar_BGColor_Tooltip"),
      tabIndex: toolbarItem.presentation.tabIndex
    }
  );
};
