// @wix/patterns/dist/cjs/components/FormPageContent/FormPageContent.uni.driver.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.FormPageContentUniDriver = void 0;
var _unidriver = require("../../unidriver");
var _unidriver2 = require("@wix/design-system/dist/testkit/unidriver");
var _PageErrorStateUni = _interopRequireDefault(require("../PageErrorState/PageErrorState.uni.driver"));
var _FormPageCardUni = _interopRequireDefault(require("../FormPageCard/FormPageCard.uni.driver"));
const FormPageContentUniDriver = (base, body) => {
  const moreActionsPopoverDriver = dataHook => (0, _unidriver2.PopoverMenuUniDriver)(base.$(`[data-hook="${dataHook}"]`), body);
  const getCard = dataHook => (0, _FormPageCardUni.default)(base.$(`[data-hook="${dataHook}"]`), body);
  return {
    /** Returns Cancel button driver */
    cancelButton: () => (0, _unidriver2.ButtonUniDriver)(base.$$(`[data-hook="form-page-cancel-btn"]`).get(0), body),
    /** Returns Save button driver */
    saveButton: () => (0, _unidriver2.ButtonUniDriver)(base.$$(`[data-hook="form-page-save-btn"]`).get(0), body),
    getMoreActions: dataHook => ({
      exists: () => moreActionsPopoverDriver(dataHook).exists(),
      /** Opens the manage view popover */
      open: () => moreActionsPopoverDriver(dataHook).getTriggerElement('more-actions-popover-button').click(),
      /** Indicates whether the manage view popover is open */
      isOpen: () => moreActionsPopoverDriver(dataHook).isMenuOpen(),
      /** Clicks at child by data-hook */
      clickAtChildByDataHook: childDataHook => moreActionsPopoverDriver(dataHook).clickAtChildByDataHook(childDataHook),
      /** Clicks at child by index */
      clickAtChildByIndex: index => moreActionsPopoverDriver(dataHook).clickAtChild(index),
      /** Gets content at index */
      childContentAt: index => moreActionsPopoverDriver(dataHook).itemContentAt(index),
      /** Gets children count */
      childrenCount: () => moreActionsPopoverDriver(dataHook).childrenCount()
    }),
    /** Returns EntityPage.Card driver by dataHook */
    getCard,
    /** Returns error state driver */
    getErrorState: () => (0, _PageErrorStateUni.default)(base, body),
    ...(0, _unidriver.baseUniDriverFactory)(base)
  };
};
exports.FormPageContentUniDriver = FormPageContentUniDriver;
