// @wix/bookings_app-extensions/build/es/dashboard.mjs
var __defProp = Object.defineProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};

// src/extensions/dashboard-page-builders.ts
var dashboard_page_builders_exports = {};
__export(dashboard_page_builders_exports, {
  exportBookingData: () => exportBookingData,
  serviceForm: () => serviceForm,
  servicesList: () => servicesList,
  staffEdit: () => staffEdit,
  staffList: () => staffList,
  syncPersonalCalendar: () => syncPersonalCalendar
});
function serviceForm(options) {
  const { serviceId } = options;
  return {
    pageId: "db6319e8-f8d6-4de3-830e-917d4e5c6d47",
    relativeUrl: `${serviceId}`
  };
}
function exportBookingData() {
  return {
    pageId: "bbbc8c7d-19b9-4a86-bfc5-1bf59208b01b",
    relativeUrl: `?openModal=exportBookingData`
  };
}
function syncPersonalCalendar() {
  return {
    pageId: "bbbc8c7d-19b9-4a86-bfc5-1bf59208b01b",
    relativeUrl: `?openModal=syncPersonalCalendar`
  };
}
function servicesList() {
  return { pageId: "c756378f-3438-41c6-a686-85e9fb0193ac" };
}
function staffEdit(options) {
  const { staffMemberId } = options;
  return {
    pageId: "a6a6a0ee-865f-43cc-a8f2-b50984f74457",
    relativeUrl: `${staffMemberId}`
  };
}
function staffList() {
  return { pageId: "a4f36fac-ea18-483b-8236-095e29fcb726" };
}

// src/extensions/dashboard-pages-plugin-props.ts
var dashboard_pages_plugin_props_exports = {};
__export(dashboard_pages_plugin_props_exports, {
  BookingCalendar: () => bbbc8c7d_19b9_4a86_bfc5_1bf59208b01b_exports,
  BookingsServiceForm: () => db6319e8_f8d6_4de3_830e_917d4e5c6d47_exports,
  BookingsStaffMemberForm: () => a6a6a0ee_865f_43cc_a8f2_b50984f74457_exports,
  Services: () => c756378f_3438_41c6_a686_85e9fb0193ac_exports,
  Staff: () => a4f36fac_ea18_483b_8236_095e29fcb726_exports
});

// src/extensions/dashboard-page-plugins/db6319e8-f8d6-4de3-830e-917d4e5c6d47.ts
var db6319e8_f8d6_4de3_830e_917d4e5c6d47_exports = {};

// src/extensions/dashboard-page-plugins/bbbc8c7d-19b9-4a86-bfc5-1bf59208b01b.ts
var bbbc8c7d_19b9_4a86_bfc5_1bf59208b01b_exports = {};

// src/extensions/dashboard-page-plugins/c756378f-3438-41c6-a686-85e9fb0193ac.ts
var c756378f_3438_41c6_a686_85e9fb0193ac_exports = {};

// src/extensions/dashboard-page-plugins/a6a6a0ee-865f-43cc-a8f2-b50984f74457.ts
var a6a6a0ee_865f_43cc_a8f2_b50984f74457_exports = {};

// src/extensions/dashboard-page-plugins/a4f36fac-ea18-483b-8236-095e29fcb726.ts
var a4f36fac_ea18_483b_8236_095e29fcb726_exports = {};
export {
  dashboard_page_builders_exports as pages,
  dashboard_pages_plugin_props_exports as plugins
};
