// @wix/ricos/dist/cjs/ricos-content/builders/audio.builder.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var audio_builder_exports = {};
__export(audio_builder_exports, {
  audio: () => audio
});
module.exports = __toCommonJS(audio_builder_exports);
var import_v1 = require("./../../ricos-schema/legacy/wix/rich_content/v1/index.js");
const audio = (...args) => {
  const audioData = args.find((a) => !("type" in a));
  const nodeBuilders = args.filter((a) => "type" in a && a.type === "block_node");
  return {
    type: "block_node",
    // @ts-expect-error
    node: (getId) => ({
      id: getId(),
      type: import_v1.Node_Type.AUDIO,
      ...nodeBuilders.length && {
        nodes: nodeBuilders.map((nb) => nb.node(getId))
      },
      audioData: audioData ? audioData : {}
    })
  };
};
