// @wix/ricos/dist/cjs/ricos-content/builders/button.builder.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var button_builder_exports = {};
__export(button_builder_exports, {
  button: () => button
});
module.exports = __toCommonJS(button_builder_exports);
var import_v1 = require("./../../ricos-schema/legacy/wix/rich_content/v1/index.js");
var import_utils = require("./utils.js");
const button = (...args) => {
  const buttonData = args.find(
    (a) => a?.type === import_v1.ButtonData_Type.LINK || a?.type === import_v1.ButtonData_Type.ACTION
  );
  return {
    type: "block_node",
    // @ts-ignore
    node: (getId) => {
      const nodes = args.map((arg) => "type" in arg && arg.type === "block_node" ? arg.node(getId) : void 0).filter(import_utils.isDefined);
      return {
        id: getId(),
        type: import_v1.Node_Type.BUTTON,
        ...nodes?.length && { nodes },
        buttonData: buttonData ? buttonData : {
          type: import_v1.ButtonData_Type.LINK,
          containerData: {
            alignment: import_v1.PluginContainerData_Alignment.CENTER,
            width: { size: import_v1.PluginContainerData_Width_Type.ORIGINAL }
          },
          styles: {},
          text: "Click Me"
        }
      };
    }
  };
};
