// @wix/auto-patterns/dist/esm/components/AutoPatternsEntityPage/SkeletonEntity.js
import { EntityPage, useEntityPage } from '@wix/patterns';
import { useForm } from '../../exports/form';
import React from 'react';
import { Card, Cell, Layout } from '@wix/design-system';
const RenderLayoutContent = _ref => {
  var _content$container$ch;
  let {
    content,
    level,
    sectionId,
    cardIndex,
    contentIndex
  } = _ref;
  switch (content.type) {
    case 'container':
      if (!((_content$container$ch = content.container.children) != null && _content$container$ch.length)) {
        return null;
      }
      return /*#__PURE__*/React.createElement(Cell, {
        span: content.container.span
      }, /*#__PURE__*/React.createElement(Layout, null, content.container.children.map((child, index) => /*#__PURE__*/React.createElement(RenderLayoutContent, {
        key: `${sectionId}-card-${cardIndex}-content-${contentIndex}-level-${level}-child-${index}`,
        content: child,
        level: level + 1,
        sectionId: sectionId,
        cardIndex: cardIndex,
        contentIndex: index
      }))));
    default:
      return null;
  }
};
const RenderLayoutCard = _ref2 => {
  var _layout$card$subtitle;
  let {
    layout,
    sectionId,
    cardIndex
  } = _ref2;
  switch (layout.type) {
    case 'card':
      return /*#__PURE__*/React.createElement(EntityPage.Card, {
        minHeight: "50px"
      }, /*#__PURE__*/React.createElement(Card.Header, {
        title: layout.card.title.text,
        subtitle: (_layout$card$subtitle = layout.card.subtitle) == null ? void 0 : _layout$card$subtitle.text
      }), /*#__PURE__*/React.createElement(Card.Divider, null), /*#__PURE__*/React.createElement(Card.Content, null, /*#__PURE__*/React.createElement(Layout, null, layout.card.children.map((child, index) => /*#__PURE__*/React.createElement(RenderLayoutContent, {
        key: `${sectionId}-card-${cardIndex}-content-${index}`,
        content: child,
        level: 0,
        sectionId: sectionId,
        cardIndex: cardIndex,
        contentIndex: index
      })))));
    default:
      return null;
  }
};
export const SkeletonEntity = _ref3 => {
  var _entityPage$layout, _entityPage$layout2, _entityPage$layout$si;
  let {
    entityPage
  } = _ref3;
  const form = useForm();
  const state = useEntityPage({
    parentPath: '',
    form,
    onSave: async () => ({
      updatedEntity: {}
    }),
    fetch: async () => new Promise(() => {})
  });
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(EntityPage, {
    state: state
  }, /*#__PURE__*/React.createElement(EntityPage.Header, {
    title: {
      text: ''
    }
  }), /*#__PURE__*/React.createElement(EntityPage.Content, null, /*#__PURE__*/React.createElement(EntityPage.MainContent, null, (_entityPage$layout = entityPage.layout) == null || (_entityPage$layout = _entityPage$layout.main) == null ? void 0 : _entityPage$layout.map((layout, layoutIndex) => /*#__PURE__*/React.createElement(RenderLayoutCard, {
    key: `main-section-card-${layoutIndex}`,
    layout: layout,
    sectionId: "main",
    cardIndex: layoutIndex
  }))), (_entityPage$layout2 = entityPage.layout) != null && _entityPage$layout2.sidebar ? /*#__PURE__*/React.createElement(EntityPage.AdditionalContent, null, (_entityPage$layout$si = entityPage.layout.sidebar) == null ? void 0 : _entityPage$layout$si.map((layout, layoutIndex) => /*#__PURE__*/React.createElement(RenderLayoutCard, {
    key: `sidebar-section-card-${layoutIndex}`,
    layout: layout,
    sectionId: "sidebar",
    cardIndex: layoutIndex
  }))) : null)));
};
