// @wix/auto_sdk_restaurants_order-settings/build/es/meta.mjs
// src/restaurants-ordersettings-v3-order-settings-order-settings.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressRestaurantsOrdersettingsV3OrderSettingsApiUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/v3/order-settings",
        destPath: "/v3/order-settings"
      }
    ],
    _: [
      {
        srcPath: "/_api/order-settings",
        destPath: "/v3/order-settings"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/order-settings",
        destPath: "/v3/order-settings"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/order-settings",
        destPath: "/v3/order-settings"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants/v3/order-settings",
        destPath: "/v3/order-settings"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_order-settings";
function getOrderSettings(payload) {
  function __getOrderSettings({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.ordersettings.v3.order_settings",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.ordersettings.v3.OrderSettingsApi.GetOrderSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsOrdersettingsV3OrderSettingsApiUrl({
        protoPath: "/v3/order-settings/{orderSettingsId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getOrderSettings;
}
function updateOrderSettings(payload) {
  function __updateOrderSettings({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.ordersettings.v3.order_settings",
      method: "PATCH",
      methodFqn: "com.wixpress.restaurants.ordersettings.v3.OrderSettingsApi.UpdateOrderSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsOrdersettingsV3OrderSettingsApiUrl({
        protoPath: "/v3/order-settings/{orderSettings.id}",
        data: serializedData,
        host
      }),
      data: serializedData
    };
    return metadata;
  }
  return __updateOrderSettings;
}
function listOrderSettings(payload) {
  function __listOrderSettings({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.ordersettings.v3.order_settings",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.ordersettings.v3.OrderSettingsApi.ListOrderSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsOrdersettingsV3OrderSettingsApiUrl({
        protoPath: "/v3/order-settings",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __listOrderSettings;
}

// src/restaurants-ordersettings-v3-order-settings-order-settings.types.ts
var ApproximateAddresses = /* @__PURE__ */ ((ApproximateAddresses2) => {
  ApproximateAddresses2["EMPTY"] = "EMPTY";
  ApproximateAddresses2["BLOCK"] = "BLOCK";
  ApproximateAddresses2["ALLOW"] = "ALLOW";
  return ApproximateAddresses2;
})(ApproximateAddresses || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["URL"] = "URL";
  Type2["TEXT"] = "TEXT";
  return Type2;
})(Type || {});
var TipsType = /* @__PURE__ */ ((TipsType2) => {
  TipsType2["AMOUNT"] = "AMOUNT";
  TipsType2["PERCENTAGE"] = "PERCENTAGE";
  return TipsType2;
})(TipsType || {});
var DayOfWeek = /* @__PURE__ */ ((DayOfWeek2) => {
  DayOfWeek2["MON"] = "MON";
  DayOfWeek2["TUE"] = "TUE";
  DayOfWeek2["WED"] = "WED";
  DayOfWeek2["THU"] = "THU";
  DayOfWeek2["FRI"] = "FRI";
  DayOfWeek2["SAT"] = "SAT";
  DayOfWeek2["SUN"] = "SUN";
  return DayOfWeek2;
})(DayOfWeek || {});
var ChargeRoundingStrategy = /* @__PURE__ */ ((ChargeRoundingStrategy2) => {
  ChargeRoundingStrategy2["UNDEFINED"] = "UNDEFINED";
  ChargeRoundingStrategy2["HALF_AWAY_FROM_ZERO"] = "HALF_AWAY_FROM_ZERO";
  ChargeRoundingStrategy2["HALF_EVEN"] = "HALF_EVEN";
  return ChargeRoundingStrategy2;
})(ChargeRoundingStrategy || {});

// src/restaurants-ordersettings-v3-order-settings-order-settings.meta.ts
function getOrderSettings2() {
  const payload = { orderSettingsId: ":orderSettingsId" };
  const getRequestOptions = getOrderSettings(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/order-settings/{orderSettingsId}",
    pathParams: { orderSettingsId: "orderSettingsId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateOrderSettings2() {
  const payload = { orderSettings: { id: ":orderSettingsId" } };
  const getRequestOptions = updateOrderSettings(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v3/order-settings/{orderSettings.id}",
    pathParams: { orderSettingsId: "orderSettingsId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listOrderSettings2() {
  const payload = {};
  const getRequestOptions = listOrderSettings(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/order-settings",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  ApproximateAddresses as ApproximateAddressesOriginal,
  ChargeRoundingStrategy as ChargeRoundingStrategyOriginal,
  DayOfWeek as DayOfWeekOriginal,
  TipsType as TipsTypeOriginal,
  Type as TypeOriginal,
  getOrderSettings2 as getOrderSettings,
  listOrderSettings2 as listOrderSettings,
  updateOrderSettings2 as updateOrderSettings
};
