// @wix/patterns/dist/cjs/components/BulkActionToolbar/BulkActionToolbar.uni.driver.js
"use strict";

exports.__esModule = true;
exports.BulkActionToolbarUniDriver = BulkActionToolbarUniDriver;
var _unidriver = require("@wix/design-system/dist/testkit/unidriver");
var _unidriver2 = require("../../unidriver");
function BulkActionToolbarUniDriver(base, body) {
  const button = (dataHook = 'bulk-action-button') => (0, _unidriver.ButtonUniDriver)(base.$(`[data-hook="${dataHook}"]`), body);
  const tooltip = (0, _unidriver.TooltipUniDriver)(base.$('[data-hook="bulk-action-button-tooltip"]'), body);
  const popover = dataHook => (0, _unidriver.PopoverMenuUniDriver)(base.$(`[data-hook="${dataHook}"]`), body);
  return {
    ...(0, _unidriver2.baseUniDriverFactory)(base),
    getButton: dataHook => button(dataHook),
    clickButton: dataHook => button(dataHook).click(),
    isButtonExists: dataHook => button(dataHook).exists(),
    isButtonDisabled: dataHook => button(dataHook).isButtonDisabled(),
    getButtonTextContent: dataHook => button(dataHook).getButtonTextContent(),
    getPopover: dataHook => popover(dataHook),
    isPopoverExists: dataHook => popover(dataHook).exists(),
    isTooltipExists: tooltip.exists,
    getTooltipText: tooltip.getTooltipText,
    clickOutsideTooltip: tooltip.clickOutside,
    getSelectedCountText: () => base.$('[data-hook="selected-count"]').text()
  };
}
