// @wix/ricos/dist/cjs/plugin-button/testkit/button-settings.uni.driver.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var button_settings_uni_driver_exports = {};
__export(button_settings_uni_driver_exports, {
  createButtonSettingsUniDriver: () => createButtonSettingsUniDriver
});
module.exports = __toCommonJS(button_settings_uni_driver_exports);
var import_unidriver = require("@wix/design-system/dist/testkit/unidriver.js");
var import_create_link_modal_unidriver = require("./../../_shared/link-modal/create-link-modal-unidriver.js");
var import_wix_rich_content_common = require("./../../wix-rich-content-common/index.js");
var import_testkit = require("./../../ui-components/testkit.js");
var import_data_hooks = require("../data-hooks.js");
const createButtonSettingsUniDriver = (getBase) => {
  return {
    settingsModal: {
      waitForElement: async () => {
        const modal = await getBase().get(`div${(0, import_wix_rich_content_common.dataHook)(import_data_hooks.SETTINGS_MODAL)}`);
        await modal.wait();
      },
      exists: async () => {
        try {
          const modal = await getBase().get(`div${(0, import_wix_rich_content_common.dataHook)(import_data_hooks.SETTINGS_MODAL)}`);
          return modal.exists();
        } catch (error) {
          return false;
        }
      },
      settings: {
        tabs: {
          ...(0, import_unidriver.TabsUniDriver)(getBase(), getBase()),
          waitForElement: async () => {
            const tabs = await getBase().get((0, import_wix_rich_content_common.dataHook)(import_data_hooks.SETTINGS_TABS));
            await tabs.wait();
          }
        },
        textInput: {
          waitForElement: async () => {
            await getBase().get((0, import_wix_rich_content_common.dataHook)(import_data_hooks.GENERAL_SETTINGS_BUTTON_TEXT_INPUT));
          },
          ...(0, import_unidriver.InputUniDriver)(getBase().$((0, import_wix_rich_content_common.dataHook)(import_data_hooks.GENERAL_SETTINGS_BUTTON_TEXT_INPUT)), getBase())
        }
      },
      design: {
        textColor: (0, import_testkit.createButtonColorSwatchesUniDriver)(
          () => getBase().$((0, import_wix_rich_content_common.dataHook)(import_data_hooks.DESIGN_SETTINGS_TEXT_COLOR_INPUT))
        ),
        borderColor: (0, import_testkit.createButtonColorSwatchesUniDriver)(
          () => getBase().$((0, import_wix_rich_content_common.dataHook)(import_data_hooks.DESIGN_SETTINGS_BORDER_COLOR_INPUT))
        ),
        backgroundColor: (0, import_testkit.createButtonColorSwatchesUniDriver)(
          () => getBase().$((0, import_wix_rich_content_common.dataHook)(import_data_hooks.DESIGN_SETTINGS_BACKGROUND_COLOR_INPUT))
        )
      }
    },
    linkModal: (0, import_create_link_modal_unidriver.createLinkModalUnidriver)(getBase),
    colorPickerModal: (0, import_testkit.createNewColorPickerUniDriver)(getBase)
  };
};
