// @wix/headless-forms/dist/react/core/Form.js
import { jsx as _jsx } from "react/jsx-runtime";
import { useService, WixServices } from '@wix/services-manager-react';
import { createServicesMap } from '@wix/services-manager';
import { FormServiceDefinition, FormService, } from '../../services/form-service.js';
import { calculateGridStyles } from '../utils.js';
const DEFAULT_SUCCESS_MESSAGE = 'Your form has been submitted successfully.';
/**
 * Root component that provides the Form service context to its children.
 * This component sets up the necessary services for rendering and managing form data.
 *
 * @order 1
 * @component
 * @param {RootProps} props - Component props
 * @param {React.ReactNode} props.children - Child components that will have access to form context
 * @param {FormServiceConfig} props.formServiceConfig - Configuration object containing form data
 * @example
 * ```tsx
 * import { Form } from '@wix/headless-forms/react';
 * import { loadFormServiceConfig } from '@wix/headless-forms/services';
 *
 * // Pattern 1: Pre-loaded form data (SSR/SSG)
 * function FormPage({ formServiceConfig }) {
 *   return (
 *     <Form.Root formServiceConfig={formServiceConfig}>
 *       <Form.Loading>
 *         {({ isLoading }) => isLoading ? <div>Loading form...</div> : null}
 *       </Form.Loading>
 *       <Form.LoadingError>
 *         {({ error, hasError }) => hasError ? <div>{error}</div> : null}
 *       </Form.LoadingError>
 *       <Form.Fields fieldMap={FIELD_MAP} />
 *     </Form.Root>
 *   );
 * }
 *
 * // Pattern 2: Lazy loading with formId (Client-side)
 * function DynamicFormPage({ formId }) {
 *   return (
 *     <Form.Root formServiceConfig={{ formId }}>
 *       <Form.Loading>
 *         {({ isLoading }) => isLoading ? <div>Loading form...</div> : null}
 *       </Form.Loading>
 *       <Form.LoadingError>
 *         {({ error, hasError }) => hasError ? <div>{error}</div> : null}
 *       </Form.LoadingError>
 *       <Form.Fields fieldMap={FIELD_MAP} />
 *     </Form.Root>
 *   );
 * }
 * ```
 */
export function Root({ formServiceConfig, children, }) {
    return (_jsx(WixServices, { servicesMap: createServicesMap().addService(FormServiceDefinition, FormService, formServiceConfig), children: children }));
}
/**
 * Headless component for form loading state access
 *
 * @component
 * @param {FormLoadingProps} props - Component props
 * @param {FormLoadingProps['children']} props.children - Render prop function that receives loading state
 * @example
 * ```tsx
 * import { Form } from '@wix/headless-forms/react';
 *
 * function FormLoadingIndicator() {
 *   return (
 *     <Form.Loading>
 *       {({ isLoading }) => (
 *         isLoading ? (
 *           <div>Loading form...</div>
 *         ) : null
 *       )}
 *     </Form.Loading>
 *   );
 * }
 * ```
 */
export function Loading(props) {
    const { isLoadingSignal } = useService(FormServiceDefinition);
    const isLoading = isLoadingSignal.get();
    return props.children({
        isLoading,
    });
}
/**
 * Headless component for form error state access
 *
 * @component
 * @param {FormErrorProps} props - Component props
 * @param {FormErrorProps['children']} props.children - Render prop function that receives error state
 * @example
 * ```tsx
 * import { Form } from '@wix/headless-forms/react';
 *
 * function FormErrorDisplay() {
 *   return (
 *     <Form.LoadingError>
 *       {({ error, hasError }) => (
 *         hasError ? (
 *           <div className="error-message">
 *             {error}
 *           </div>
 *         ) : null
 *       )}
 *     </Form.LoadingError>
 *   );
 * }
 * ```
 */
export function LoadingError(props) {
    const { errorSignal } = useService(FormServiceDefinition);
    const error = errorSignal.get();
    const hasError = !!error;
    return props.children({
        error,
        hasError,
    });
}
/**
 * Headless component for form submit error state access
 *
 * @component
 * @param {FormSubmitErrorProps} props - Component props
 * @param {FormSubmitErrorProps['children']} props.children - Render prop function that receives submit error state
 * @example
 * ```tsx
 * import { Form } from '@wix/headless-forms/react';
 *
 * function FormSubmitErrorDisplay() {
 *   return (
 *     <Form.Error>
 *       {({ error, hasError }) => (
 *         hasError ? (
 *           <div className="error-message">
 *             {error}
 *           </div>
 *         ) : null
 *       )}
 *     </Form.Error>
 *   );
 * }
 * ```
 */
export function Error(props) {
    const { submitResponseSignal } = useService(FormServiceDefinition);
    const submitResponse = submitResponseSignal.get();
    const error = submitResponse.type === 'error' ? submitResponse.message : null;
    const hasError = submitResponse.type === 'error';
    return props.children({
        error,
        hasError,
    });
}
/**
 * Headless component for form submission state access
 *
 * @component
 * @param {FormSubmittedProps} props - Component props
 * @param {FormSubmittedProps['children']} props.children - Render prop function that receives submission state
 * @example
 * ```tsx
 * import { Form } from '@wix/headless-forms/react';
 *
 * function FormSubmittedDisplay() {
 *   return (
 *     <Form.Submitted>
 *       {({ isSubmitted, message }) => (
 *         isSubmitted ? (
 *           <div className="success-message">
 *             <h2>Thank You!</h2>
 *             <p>{message}</p>
 *           </div>
 *         ) : null
 *       )}
 *     </Form.Submitted>
 *   );
 * }
 * ```
 */
export function Submitted(props) {
    const { submitResponseSignal } = useService(FormServiceDefinition);
    const submitResponse = submitResponseSignal.get();
    const isSubmitted = submitResponse.type === 'success';
    const message = submitResponse.type === 'success'
        ? submitResponse.message || DEFAULT_SUCCESS_MESSAGE
        : DEFAULT_SUCCESS_MESSAGE;
    return props.children({
        isSubmitted,
        message,
    });
}
/**
 * Fields component that provides form data and actions to its children.
 * This component accesses the form data and submitForm action from the service
 * and passes them to children via render props.
 *
 * @component
 * @param {FieldsProps} props - Component props
 * @param {FieldsProps['children']} props.children - Render prop function that receives form data and actions
 * @example
 * ```tsx
 * import { Form } from '@wix/headless-forms/react';
 *
 * function FormFields() {
 *   return (
 *     <Form.Fields>
 *       {({ form, submitForm }) => (
 *         form ? (
 *           <div>
 *             <h2>{form.name}</h2>
 *             <p>{form.description}</p>
 *             <button onClick={() => submitForm({ field1: 'value' })}>
 *               Submit
 *             </button>
 *           </div>
 *         ) : null
 *       )}
 *     </Form.Fields>
 *   );
 * }
 * ```
 */
export function Fields(props) {
    const { formSignal, submitForm } = useService(FormServiceDefinition);
    const form = formSignal.get();
    return props.children({
        form,
        submitForm,
    });
}
/**
 * Headless Field component that provides field layout data and grid styles.
 * This component accesses field configuration and calculates grid positioning.
 *
 * @component
 * @param {FieldProps} props - Component props
 * @param {FieldProps['children']} props.children - Render prop function that receives field layout data
 * @example
 * ```tsx
 * import { Form } from '@wix/headless-forms/react';
 *
 * function CustomField({ id, layout }) {
 *   return (
 *     <Form.Field id={id} layout={layout}>
 *       {({ id, layout, gridStyles }) => (
 *         <div data-field-id={id}>
 *           <div style={gridStyles.label}>Label</div>
 *           <div style={gridStyles.input}>Input</div>
 *         </div>
 *       )}
 *     </Form.Field>
 *   );
 * }
 * ```
 */
export function Field(props) {
    const { id, children, layout } = props;
    const { formSignal } = useService(FormServiceDefinition);
    const form = formSignal.get();
    if (!form) {
        return null;
    }
    const gridStyles = calculateGridStyles(layout);
    return children({
        id,
        layout,
        gridStyles,
    });
}
