// @wix/wix-data-items-common/dist/cjs/types/data-item-types.js
"use strict";

exports.__esModule = true;
exports.SortOrder = exports.SortMode = exports.Mode = exports.Environment = exports.BulkActionType = exports.Action = exports.ACTION = void 0;
let Environment = exports.Environment = /*#__PURE__*/function (Environment) {
  Environment["LIVE"] = "LIVE";
  Environment["SANDBOX"] = "SANDBOX";
  Environment["SANDBOX_PREFERRED"] = "SANDBOX_PREFERRED";
  return Environment;
}({});
/** @oneof */
let ACTION = exports.ACTION = /*#__PURE__*/function (ACTION) {
  ACTION["UNKNOWN_ACTION"] = "UNKNOWN_ACTION";
  ACTION["SET_FIELD"] = "SET_FIELD";
  ACTION["REMOVE_FIELD"] = "REMOVE_FIELD";
  ACTION["INCREMENT_FIELD"] = "INCREMENT_FIELD";
  ACTION["APPEND_TO_ARRAY"] = "APPEND_TO_ARRAY";
  ACTION["REMOVE_FROM_ARRAY"] = "REMOVE_FROM_ARRAY";
  return ACTION;
}({});
let BulkActionType = exports.BulkActionType = /*#__PURE__*/function (BulkActionType) {
  BulkActionType["UNKNOWN_ACTION_TYPE"] = "UNKNOWN_ACTION_TYPE";
  BulkActionType["INSERT"] = "INSERT";
  BulkActionType["UPDATE"] = "UPDATE";
  BulkActionType["DELETE"] = "DELETE";
  BulkActionType["PATCH"] = "PATCH";
  return BulkActionType;
}({});
let Action = exports.Action = /*#__PURE__*/function (Action) {
  Action["UNKNOWN_ACTION"] = "UNKNOWN_ACTION";
  Action["INSERTED"] = "INSERTED";
  Action["UPDATED"] = "UPDATED";
  return Action;
}({});
/** @oneof */
/** @oneof */
let SortOrder = exports.SortOrder = /*#__PURE__*/function (SortOrder) {
  SortOrder["ASC"] = "ASC";
  SortOrder["DESC"] = "DESC";
  return SortOrder;
}({});
let SortMode = exports.SortMode = /*#__PURE__*/function (SortMode) {
  /** Placeholder. Falls back to REGULAR. */
  SortMode["UNKNOWN_SORT_MODE"] = "UNKNOWN_SORT_MODE";
  /** Use the sorting specified in query.sort. */
  SortMode["WIX_QUERY"] = "WIX_QUERY";
  /** Return randomly sorted results, disregarding query.sort. */
  SortMode["RANDOM"] = "RANDOM";
  return SortMode;
}({});
/** @oneof */
/** @oneof */
/** @oneof */
/** @oneof */
/** @oneof */
/** @oneof */
let Mode = exports.Mode = /*#__PURE__*/function (Mode) {
  /** Any of the search terms must be present. */
  Mode["OR"] = "OR";
  /** All search terms must be present. */
  Mode["AND"] = "AND";
  return Mode;
}({});
/** @oneof */
/** @oneof */
/** @oneof */
/** @oneof */
/** Transforms collection items by including, excluding, or reshaping fields. */
/** Converts an object to an array. */
/** Unwinds an array. */
/** @oneof */
