// @wix/bex-core/dist/cjs/state/ExportState/ExportStateBIReporter.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.ExportStateBIReporter = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _v = require("@wix/bex-utils/@wix/bi-logger-os-data/v2");
var _util = require("../../util");
var _mobx = require("mobx");
var _uuid = require("uuid");
var _getJobItemsAbsoluteCounter = require("../getJobItemsAbsoluteCounter");
const withoutDefaults = (0, _util.withoutIrrelevant)();
const exportTypeToBiExportType = {
  all: 'All Items',
  filtered: 'Filtered Items',
  selected: 'Selected Items'
};
class ExportStateBIReporter {
  constructor(state) {
    (0, _defineProperty2.default)(this, "state", void 0);
    (0, _defineProperty2.default)(this, "exportBiUniqueId", (0, _uuid.v4)());
    this.state = state;
  }
  get reportBi() {
    return this.state.reportBi;
  }
  async _reportWithDefaultParams(reportProps) {
    const {
      state,
      exportBiUniqueId
    } = this;
    this.reportBi({
      ...reportProps,
      params: {
        ...reportProps.params,
        exportId: exportBiUniqueId,
        supportsTotal: state.collection.result.supportTotal,
        availableExportTypes: JSON.stringify({
          ...(state.disabledRadios.indexOf('all') === -1 && {
            [exportTypeToBiExportType.all]: await state.totalForExportGettersMap.all()
          }),
          ...(state.disabledRadios.indexOf('filtered') === -1 && {
            [exportTypeToBiExportType.filtered]: state.totalForExportGettersMap.filtered()
          }),
          ...(state.disabledRadios.indexOf('selected') === -1 && {
            [exportTypeToBiExportType.selected]: state.totalForExportGettersMap.selected()
          })
        })
      }
    });
  }
  async _reportWithSelectedExportParams(reportProps) {
    const {
      state
    } = this;
    await this._reportWithDefaultParams({
      ...reportProps,
      params: {
        ...reportProps.params,
        selectedExportType: exportTypeToBiExportType[state.selectedExportOption],
        numItems: await state.totalForExportGettersMap[state.selectedExportOption]()
      }
    });
  }
  onExportTypeChange({
    prev,
    next
  }) {
    this._reportWithDefaultParams(withoutDefaults(_v.cairoItemToggledInExportModal)({
      newSelectedExportType: exportTypeToBiExportType[next],
      prevSelectedExportType: exportTypeToBiExportType[prev]
    }));
  }
  clickOpenModal() {
    const {
      state
    } = this;
    this._reportWithDefaultParams(withoutDefaults(_v.cairoExportCtaClicked)({
      defaultExportType: exportTypeToBiExportType[state.selectedExportOption],
      numSelectedItems: state.totalForExportGettersMap.selected()
    }));
  }
  beforeToast() {
    const {
      state,
      reportBi
    } = this;
    reportBi(withoutDefaults(_v.cairoFileDownloadToastDisplayed)({
      exportId: this.exportBiUniqueId,
      exportType: exportTypeToBiExportType[state.selectedExportOption],
      numSelectedItems: state.totalForExportGettersMap.selected(),
      try: state.retries
    }));
    return {
      startTime: performance.now()
    };
  }
  onToastActionClick({
    startTime
  }) {
    const {
      state,
      reportBi
    } = this;
    reportBi(withoutDefaults(_v.cairoFileDownloadToastDownloadButtonClicked)({
      exportId: this.exportBiUniqueId,
      exportType: exportTypeToBiExportType[state.selectedExportOption],
      numSelectedItems: state.totalForExportGettersMap.selected(),
      duration: Math.round(performance.now() - startTime)
    }));
  }
  onRetryExportToClick() {
    this._reportWithSelectedExportParams(withoutDefaults(_v.cairoExportFailedModalCtaClicked)({
      ctaName: 'Try Again',
      try: this.state.retries
    }));
  }
  onFailedCancel() {
    this._reportWithSelectedExportParams(withoutDefaults(_v.cairoExportFailedModalCtaClicked)({
      ctaName: 'Cancel',
      try: this.state.retries
    }));
  }
  onFailedClose() {
    this._reportWithSelectedExportParams(withoutDefaults(_v.cairoExportFailedModalCtaClicked)({
      ctaName: 'X Button',
      try: this.state.retries
    }));
  }
  onExportToClick() {
    this._reportWithSelectedExportParams(withoutDefaults(_v.cairoExportModalCtaClicked)({
      action: 'export',
      origin: 'Export Button'
    }));
  }
  onExportStart() {
    const startTime = performance.now();
    this._reportWithSelectedExportParams(withoutDefaults(_v.cairoExportProcessStart)({
      try: this.state.retries
    }));
    return (0, _mobx.action)(async ({
      job,
      query,
      result,
      saveAs
    }) => {
      const {
        state: {
          retries,
          totalForExport
        }
      } = this;
      this._reportWithSelectedExportParams(withoutDefaults(_v.cairoExportProcessEnd)({
        duration: Math.round(performance.now() - startTime),
        try: retries,
        result,
        fileName: saveAs || '',
        numItemsExported: job != null ? (0, _getJobItemsAbsoluteCounter.getJobItemsAbsoluteCounter)(job, {
          query,
          totalForExport
        }) : 0
      }));
    });
  }
  onCancelClick() {
    this._reportWithSelectedExportParams(withoutDefaults(_v.cairoExportModalCtaClicked)({
      action: 'dismiss',
      origin: 'Cancel Button'
    }));
  }
  onCancelExportButtonClick() {
    const {
      state
    } = this;
    const {
      retries,
      exportStartTime
    } = state;
    const duration = performance.now() - exportStartTime;
    this._reportWithSelectedExportParams(withoutDefaults(_v.cairoCancelExportModalCtaClicked)({
      duration,
      try: retries,
      ctaName: 'Cancel Export',
      modalType: 'Export'
    }));
  }
  onCancelExportGoBackButtonClick() {
    const {
      state
    } = this;
    const {
      retries,
      exportStartTime
    } = state;
    const duration = performance.now() - exportStartTime;
    this._reportWithSelectedExportParams(withoutDefaults(_v.cairoCancelExportModalCtaClicked)({
      duration,
      try: retries,
      ctaName: 'Go Back',
      modalType: 'Export'
    }));
  }
  onCancelExportCloseButtonClick() {
    const {
      state
    } = this;
    const {
      retries,
      exportStartTime
    } = state;
    const duration = performance.now() - exportStartTime;
    this._reportWithSelectedExportParams(withoutDefaults(_v.cairoCancelExportModalCtaClicked)({
      duration,
      try: retries,
      ctaName: 'X Button',
      modalType: 'Export'
    }));
  }
  onExportCloseAttemptClick() {
    const {
      state
    } = this;
    const {
      retries,
      exportStartTime
    } = state;
    const duration = performance.now() - exportStartTime;
    this._reportWithSelectedExportParams(withoutDefaults(_v.cairoUserAttemptsToCancelTheExport)({
      duration,
      origin: 'X Button',
      try: retries
    }));
  }
  onExportCancelAttemptClick() {
    const {
      state
    } = this;
    const {
      retries,
      exportStartTime
    } = state;
    const duration = performance.now() - exportStartTime;
    this._reportWithSelectedExportParams(withoutDefaults(_v.cairoUserAttemptsToCancelTheExport)({
      duration,
      origin: 'Cancel Button',
      try: retries
    }));
  }
  onCloseClick() {
    this._reportWithSelectedExportParams(withoutDefaults(_v.cairoExportModalCtaClicked)({
      action: 'dismiss',
      origin: 'X Button'
    }));
  }
  onToastDismissed({
    startTime,
    by: dismissMode
  }) {
    const {
      state,
      reportBi
    } = this;
    reportBi(withoutDefaults(_v.cairoFileDownloadToastDismissed)({
      exportId: this.exportBiUniqueId,
      exportType: state.selectedExportOption,
      numSelectedItems: state.totalForExportGettersMap.selected(),
      duration: Math.round(performance.now() - startTime),
      dismissMode
    }));
  }
}
exports.ExportStateBIReporter = ExportStateBIReporter;
