// @wix/headless-blog/dist/react/CommentForm.js
import { jsx as _jsx } from "react/jsx-runtime";
import { AsChildSlot } from '@wix/headless-utils/react';
import React from 'react';
import { CommentContext } from './Comment.js';
import * as CoreComments from './core/Comments.js';
import { isValidChildren } from './helpers.js';
const CommentFormContext = React.createContext(null);
CommentFormContext.displayName = 'Comment.CommentFormContext';
function useCommentFormContext() {
    const context = React.useContext(CommentFormContext);
    if (!context) {
        throw new Error('useCommentFormContext must be used within a Comment.CommentForm.Root component');
    }
    return context;
}
/**
 * Form component for creating comments.
 * Provides context for sub-components to share form state.
 * Must contain composable sub-components as children.
 *
 * @component
 * @example
 * ```tsx
 * // Composable form with sub-components
 * <Comment.Form.Root className="space-y-3">
 *   <Comment.Form.Input
 *     className="w-full border rounded-lg p-3"
 *     placeholder="Write your reply..."
 *   />
 *   <Comment.Form.Message className="text-red-600" />
 *   <Comment.Form.SubmitButton className="btn-primary" />
 * </Comment.Form.Root>
 *
 * // Custom rendering with asChild
 * <Comment.Form.Root asChild>
 *   {({ createComment, isLoading, error, commentText, setCommentText, handleSubmit, handleCancel }) => (
 *     <CustomCommentForm
 *       onCommentAdded={onCommentAdded}
 *       loading={isLoading}
 *       error={error}
 *       commentText={commentText}
 *       setCommentText={setCommentText}
 *       handleSubmit={handleSubmit}
 *       handleCancel={handleCancel}
 *     />
 *   )}
 * </Comment.Form.Root>
 * ```
 */
export const Root = React.forwardRef((props, ref) => {
    const { asChild, children, className, maxLength = 500, onCommentAdded } = props;
    const topLevelComment = React.useContext(CoreComments.TopLevelCommentContext);
    const parentComment = React.useContext(CommentContext);
    return (_jsx(CoreComments.CreateComment, { parentCommentId: parentComment?.comment._id, topCommentId: topLevelComment?._id, children: ({ createComment, isLoading, error, clearError }) => {
            const [commentText, setCommentText] = React.useState('');
            const htmlId = `text-field-${React.useId()}`;
            const handleSubmit = React.useCallback(async (e) => {
                e.preventDefault();
                const text = commentText.trim();
                if (!text || isLoading) {
                    return;
                }
                try {
                    // Create reply content structure for Wix Comments API
                    const result = await createComment(text);
                    if (result) {
                        // Clear form on success
                        setCommentText('');
                        onCommentAdded?.();
                    }
                }
                catch (err) {
                    // Error is handled by the service
                    console.error('Failed to create reply:', err);
                }
            }, [commentText, createComment, isLoading]);
            const handleCancel = React.useCallback(() => {
                setCommentText('');
                clearError();
            }, [setCommentText, clearError]);
            const contextValue = {
                commentText,
                setCommentText,
                handleSubmit,
                handleCancel,
                createComment,
                isLoading,
                error,
                maxLength,
                htmlId,
            };
            const attributes = {
                'data-testid': "comment-form-root" /* TestIds.root */,
                'data-loading': isLoading,
            };
            return (_jsx(CommentFormContext.Provider, { value: contextValue, children: _jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, ...attributes, customElement: children, customElementProps: {
                        createComment,
                        isLoading,
                        error,
                        commentText,
                        setCommentText,
                        handleSubmit,
                        handleCancel,
                    }, children: isValidChildren(children) ? (_jsx("form", { onSubmit: handleSubmit, children: children })) : (children) }) }));
        } }));
});
/**
 * Label component for comment form input.
 *
 * @component
 * @example
 * ```tsx
 * // Default label
 * <Comment.Form.Label />
 *
 * // Custom styling
 * <Comment.Form.Label className="text-gray-700 font-medium mb-2" />
 *
 * // Custom rendering with asChild
 * <Comment.Form.Label asChild>
 *   {({ htmlFor, labelText }) => (
 *     <label htmlFor={htmlFor} className="custom-label">
 *       {labelText}
 *     </label>
 *   )}
 * </Comment.Form.Label>
 * ```
 */
export const Label = React.forwardRef((props, ref) => {
    const { asChild, children, className } = props;
    const { htmlId } = useCommentFormContext();
    const attributes = {
        'data-testid': "comment-form-label" /* TestIds.label */,
    };
    return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, ...attributes, customElement: children, customElementProps: {
            htmlFor: htmlId,
            labelText: 'Comment',
        }, children: _jsx("label", { htmlFor: htmlId, children: "Comment" }) }));
});
Label.displayName = 'Comment.Form.Label';
/**
 * Input component for form textarea.
 *
 * @component
 * @example
 * ```tsx
 * // Default textarea
 * <Comment.Form.Input placeholder="Write your comment..." />
 *
 * // Custom styling
 * <Comment.Form.Input
 *   className="w-full h-24 p-3 border border-gray-300 rounded-lg focus:ring-2 focus:ring-blue-500 focus:border-blue-500 resize-none"
 *   placeholder="Write your comment..."
 *   rows={3}
 * />
 *
 * // Custom rendering with asChild
 * <Comment.Form.Input asChild>
 *   {({ value, onChange, disabled, placeholder, maxLength, id }) => (
 *     <RichTextEditor
 *       id={id}
 *       value={value}
 *       onChange={onChange}
 *       disabled={disabled}
 *       placeholder={placeholder}
 *       maxLength={maxLength}
 *     />
 *   )}
 * </Comment.Form.Input>
 * ```
 */
export const Input = React.forwardRef((props, ref) => {
    const { asChild, children, className } = props;
    const { commentText, setCommentText, isLoading, maxLength, htmlId } = useCommentFormContext();
    const handleChange = React.useCallback((e) => {
        setCommentText(e.target.value);
    }, [setCommentText]);
    const inputAttributes = {
        value: commentText,
        onChange: handleChange,
        minLength: 1,
        disabled: isLoading,
        maxLength,
        id: htmlId,
    };
    const dataAttributes = {
        'data-testid': "comment-form-input" /* TestIds.input */,
        'data-loading': isLoading,
    };
    return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, ...dataAttributes, ...inputAttributes, customElement: children, customElementProps: inputAttributes, children: _jsx("textarea", {}) }));
});
Input.displayName = 'Comment.Form.Input';
/**
 * Error message display component for comment form.
 *
 * @component
 * @example
 * ```tsx
 * // Default error display
 * <Comment.Form.Message />
 *
 * // Custom styling
 * <Comment.Form.Message className="text-red-600 text-sm bg-red-50 p-2 rounded" />
 *
 * // Custom rendering with asChild
 * <Comment.Form.Message asChild>
 *   {({ error, hasError }) => (
 *     hasError && <CustomErrorAlert message={error} />
 *   )}
 * </Comment.Form.Message>
 * ```
 */
export const Message = React.forwardRef((props, ref) => {
    const { asChild, children, className } = props;
    const { error } = useCommentFormContext();
    const hasError = Boolean(error);
    if (!hasError)
        return null;
    const attributes = {
        'data-testid': "comment-form-message" /* TestIds.messsage */,
        'data-has-error': hasError,
    };
    return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, ...attributes, customElement: children, customElementProps: { error: error, hasError }, children: _jsx("div", { children: error }) }));
});
Message.displayName = 'Comment.Form.Message';
/**
 * Cancel button for the comment form.
 * Clears the form input when clicked.
 *
 * @component
 * @example
 * ```tsx
 * // Default rendering
 * <Comment.Form.CancelButton />
 *
 * // Custom rendering with asChild
 * <Comment.Form.CancelButton asChild>
 *   {({ handleCancel }) => (
 *     <Button variant="ghost" onClick={handleCancel}>
 *       Cancel
 *     </Button>
 *   )}
 * </Comment.Form.CancelButton>
 * ```
 */
export const CancelButton = React.forwardRef((props, ref) => {
    const { asChild, children, className } = props;
    const { commentText, isLoading, handleCancel } = useCommentFormContext();
    const isDisabled = !commentText.trim();
    const buttonAttributes = {
        type: 'reset',
        onClick: handleCancel,
    };
    const dataAttributes = {
        'data-testid': "comment-form-cancel" /* TestIds.cancel */,
        'data-disabled': isDisabled,
        'data-loading': isLoading,
    };
    return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, ...dataAttributes, ...buttonAttributes, customElement: children, customElementProps: {
            isDisabled,
            isLoading,
            handleCancel,
        }, children: _jsx("button", { children: "Cancel" }) }));
});
CancelButton.displayName = 'Comment.Form.CancelButton';
/**
 * Submit button for the comment form.
 * Automatically disabled when the form is empty or submitting.
 *
 * @component
 * @example
 * ```tsx
 * // Default rendering
 * <Comment.Form.SubmitButton />
 *
 * // Custom rendering with asChild
 * <Comment.Form.SubmitButton asChild>
 *   <Button>Post reply</Button>
 * </Comment.Form.SubmitButton>
 * ```
 */
export const SubmitButton = React.forwardRef((props, ref) => {
    const { asChild, children, className, loadingState } = props;
    const { commentText, isLoading } = useCommentFormContext();
    const isDisabled = !commentText.trim();
    const buttonAttributes = {
        type: 'submit',
        disabled: isDisabled,
    };
    const dataAttributes = {
        'data-testid': "comment-form-submit" /* TestIds.submit */,
        'data-disabled': isDisabled,
        'data-loading': isLoading,
    };
    return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, ...dataAttributes, ...buttonAttributes, customElement: children, customElementProps: {
            isDisabled,
            isLoading,
        }, content: isLoading && loadingState ? loadingState : undefined, children: _jsx("button", { children: "Post comment" }) }));
});
SubmitButton.displayName = 'Comment.Form.SubmitButton';
