// @wix/auto_sdk_restaurants_pos-tpa/build/es/meta.mjs
// src/restaurants-postpaapi-v1-configuration-pos-tpa.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressRestaurantsPostpaapiRestaurantsPosTpaApiUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/restaurants/v1/pos-spi-host",
        destPath: "/v1/pos-spi-host"
      }
    ],
    "com.wixpress.restaurants.restaurants-pos-spi-host.wixapps.net": [
      {
        srcPath: "/restaurants-pos-spi-host",
        destPath: "/v1/pos-spi-host"
      }
    ],
    "www.wixapps.com": [
      {
        srcPath: "/restaurants/v1/pos-integration",
        destPath: "/v1/pos-integration"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/v1/pos-integration",
        destPath: "/v1/pos-integration"
      }
    ],
    "users._base_domain": [
      {
        srcPath: "/_api/restaurants/v1/pos-spi-host",
        destPath: "/v1/pos-spi-host"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_pos-tpa";
function getConfiguration(payload) {
  function __getConfiguration({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.postpaapi.v1.configuration",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.postpaapi.RestaurantsPosTpaApi.GetConfiguration",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsPostpaapiRestaurantsPosTpaApiUrl({
        protoPath: "/v1/pos-integration/configurations/{id}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getConfiguration;
}
function listConfigurations(payload) {
  function __listConfigurations({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.postpaapi.v1.configuration",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.postpaapi.RestaurantsPosTpaApi.ListConfigurations",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsPostpaapiRestaurantsPosTpaApiUrl({
        protoPath: "/v1/pos-integration/configurations",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __listConfigurations;
}
function disconnectAccount(payload) {
  function __disconnectAccount({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.postpaapi.v1.configuration",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.postpaapi.RestaurantsPosTpaApi.DisconnectAccount",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsPostpaapiRestaurantsPosTpaApiUrl({
        protoPath: "/v1/pos-integration/accounts/{accountId}/disconnect",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __disconnectAccount;
}
function disconnectConfiguration(payload) {
  function __disconnectConfiguration({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.postpaapi.v1.configuration",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.postpaapi.RestaurantsPosTpaApi.DisconnectConfiguration",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsPostpaapiRestaurantsPosTpaApiUrl({
        protoPath: "/v1/pos-integration/configurations/{configurationId}/disconnect",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __disconnectConfiguration;
}

// src/restaurants-postpaapi-v1-configuration-pos-tpa.meta.ts
function getConfiguration2() {
  const payload = { id: ":id" };
  const getRequestOptions = getConfiguration(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/pos-integration/configurations/{id}",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listConfigurations2() {
  const payload = {};
  const getRequestOptions = listConfigurations(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/pos-integration/configurations",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function disconnectAccount2() {
  const payload = { accountId: ":accountId" };
  const getRequestOptions = disconnectAccount(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/pos-integration/accounts/{accountId}/disconnect",
    pathParams: { accountId: "accountId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function disconnectConfiguration2() {
  const payload = { configurationId: ":configurationId" };
  const getRequestOptions = disconnectConfiguration(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/pos-integration/configurations/{configurationId}/disconnect",
    pathParams: { configurationId: "configurationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  disconnectAccount2 as disconnectAccount,
  disconnectConfiguration2 as disconnectConfiguration,
  getConfiguration2 as getConfiguration,
  listConfigurations2 as listConfigurations
};
