// @wix/headless-stores/dist/react/core/CategoryList.js
import { jsx as _jsx } from "react/jsx-runtime";
import React from 'react';
import { useService, WixServices } from '@wix/services-manager-react';
import { createServicesMap } from '@wix/services-manager';
import { CategoriesListService, CategoriesListServiceDefinition, } from '../../services/categories-list-service.js';
/**
 * Root component that provides the CategoryList service context to its children.
 * This component sets up the necessary services for managing categories list state.
 * Optionally connects to ProductList filtering when connectToProductFilter is enabled.
 *
 * @order 1
 * @component
 * @example
 * ```tsx
 * import { CategoryListCore } from '@wix/stores/components';
 *
 * function CategoriesPage() {
 *   return (
 *     <CategoryList.Root
 *       categoriesListConfig={{ categories: myCategories }}
 *       connectToProductFilter={true}
 *     >
 *       <CategoryList.ItemContent>
 *         {({ category }) => (
 *           <div key={category._id}>
 *             <h2>{category.name}</h2>
 *           </div>
 *         )}
 *       </CategoryList.ItemContent>
 *     </CategoryList.Root>
 *   );
 * }
 * ```
 */
export function Root(props) {
    const { categoriesListConfig, children } = props;
    return (_jsx(WixServices, { servicesMap: createServicesMap().addService(CategoriesListServiceDefinition, CategoriesListService, categoriesListConfig), children: children }));
}
Root.displayName = 'CategoryList.Root';
/**
 * Component that renders content when the categories list is empty.
 * Only displays its children when there are no categories, no loading state, and no errors.
 *
 * @component
 * @example
 * ```tsx
 * import { CategoryListCore } from '@wix/stores/components';
 *
 * function EmptyCategoriesMessage() {
 *   return (
 *     <CategoryList.EmptyState>
 *       {() => (
 *         <div className="empty-state">
 *           <h3>No categories found</h3>
 *           <p>Categories will appear here once they are created</p>
 *         </div>
 *       )}
 *     </CategoryList.EmptyState>
 *   );
 * }
 * ```
 */
export function EmptyState(props) {
    const { isLoading, error, categories } = useService(CategoriesListServiceDefinition);
    const isLoadingValue = isLoading.get();
    const errorValue = error.get();
    const categoriesValue = categories.get();
    if (!isLoadingValue && !errorValue && categoriesValue.length === 0) {
        return typeof props.children === 'function'
            ? props.children({})
            : props.children;
    }
    return null;
}
/**
 * Component that renders content during loading state.
 * Only displays its children when the categories list is currently loading.
 *
 * @component
 * @example
 * ```tsx
 * import { CategoryListCore } from '@wix/stores/components';
 *
 * function CategoriesLoading() {
 *   return (
 *     <CategoryList.Loading>
 *       {() => (
 *         <div className="loading-spinner">
 *           <div>Loading categories...</div>
 *           <div className="spinner"></div>
 *         </div>
 *       )}
 *     </CategoryList.Loading>
 *   );
 * }
 * ```
 */
export function Loading(props) {
    const { isLoading } = useService(CategoriesListServiceDefinition);
    const isLoadingValue = isLoading.get();
    if (isLoadingValue) {
        return typeof props.children === 'function'
            ? props.children({})
            : props.children;
    }
    return null;
}
/**
 * Component that renders content when there's an error loading categories.
 * Only displays its children when an error has occurred.
 *
 * @component
 * @example
 * ```tsx
 * import { CategoryListCore } from '@wix/stores/components';
 *
 * function CategoriesError() {
 *   return (
 *     <CategoryList.Error>
 *       {({ error }) => (
 *         <div className="error-state">
 *           <h3>Error loading categories</h3>
 *           <p>{error}</p>
 *           <button onClick={() => window.location.reload()}>
 *             Try Again
 *           </button>
 *         </div>
 *       )}
 *     </CategoryList.Error>
 *   );
 * }
 * ```
 */
export function Error(props) {
    const { error } = useService(CategoriesListServiceDefinition);
    const errorValue = error.get();
    if (errorValue) {
        return typeof props.children === 'function'
            ? props.children({ error: errorValue })
            : props.children;
    }
    return null;
}
/**
 * Component that renders content for each category in the list.
 * Maps over all categories and provides each category through a service context.
 * Only renders when categories are successfully loaded (not loading, no error, and has categories).
 *
 * @component
 * @example
 * ```tsx
 * import { CategoryListCore } from '@wix/stores/components';
 *
 * function CategoriesGrid() {
 *   return (
 *     <CategoryList.ItemContent>
 *       {({ category }) => (
 *         <div className="category-card">
 *           <h3>{category.name}</h3>
 *           <p>{category.description}</p>
 *           <a href={`/categories/${category.slug}`}>View Category</a>
 *         </div>
 *       )}
 *     </CategoryList.ItemContent>
 *   );
 * }
 * ```
 */
export function ItemContent(props) {
    const { categories, isLoading, error } = useService(CategoriesListServiceDefinition);
    const categoriesValue = categories.get();
    if (isLoading.get() || error.get() || categoriesValue.length === 0) {
        return null;
    }
    return categoriesValue.map((category) => (_jsx(React.Fragment, { children: typeof props.children === 'function'
            ? props.children({
                category,
            })
            : props.children }, category._id)));
}
