// @wix/headless-stores/dist/react/core/Product.js
import { jsx as _jsx } from "react/jsx-runtime";
import { useService, WixServices } from '@wix/services-manager-react';
import { ProductServiceDefinition, ProductService, } from '../../services/product-service.js';
import { createServicesMap } from '@wix/services-manager';
/**
 * Root component that provides the Product service context to its children.
 * This component sets up the necessary services for rendering and managing a single product's data.
 *
 * @order 1
 * @component
 * @example
 * ```tsx
 * import { Product } from '@wix/stores/components';
 *
 * function ProductPage() {
 *   return (
 *     <Product.Root productServiceConfig={{ product: myProduct }}>
 *       <div>
 *         <Product.Name>
 *           {({ name }) => (
 *             <h1
 *               className="text-4xl font-bold text-content-primary mb-4"
 *               data-testid="product-name"
 *             >
 *               {name}
 *             </h1>
 *           )}
 *         </Product.Name>
 *       </div>
 *     </Product.Root>
 *   );
 * }
 * ```
 */
export function Root(props) {
    return (_jsx(WixServices, { servicesMap: createServicesMap().addService(ProductServiceDefinition, ProductService, props.productServiceConfig), children: props.children }));
}
/**
 * Headless component for product name display
 *
 * @component
 * @example
 * ```tsx
 * import { Product } from '@wix/stores/components';
 *
 * function ProductHeader() {
 *   return (
 *     <Product.Name>
 *       {({ name }) => (
 *         <h1>{name}</h1>
 *       )}
 *     </Product.Name>
 *   );
 * }
 * ```
 */
export function Name(props) {
    const service = useService(ProductServiceDefinition);
    const product = service.product.get();
    const name = product.name;
    return props.children({
        name,
    });
}
/**
 * Headless component for product description display
 *
 * @component
 * @example
 * ```tsx
 * import { Product } from '@wix/stores/components';
 *
 * function ProductDescription() {
 *   return (
 *     <Product.Description>
 *       {({ plainDescription, description }) => (
 *         <div>
 *           {plainDescription && (
 *             <div
 *               dangerouslySetInnerHTML={{
 *                 __html: plainDescription,
 *               }}
 *             />
 *           )}
 *           {description && (
 *             <div>Rich content description available</div>
 *           )}
 *         </div>
 *       )}
 *     </Product.Description>
 *   );
 * }
 * ```
 */
export function Description(props) {
    const service = useService(ProductServiceDefinition);
    const product = service.product.get();
    const descriptionRichContent = product.description;
    const plainDescription = product.plainDescription;
    return props.children({
        description: descriptionRichContent,
        plainDescription: plainDescription,
    });
}
export function Media(props) {
    const service = useService(ProductServiceDefinition);
    const product = service.product.get();
    const mainMedia = product.media?.main;
    const mediaItems = product.media?.itemsInfo?.items ?? [];
    return props.children({
        mediaItems,
        mainMedia,
    });
}
export function Content(props) {
    const service = useService(ProductServiceDefinition);
    const product = service.product.get();
    return props.children({
        product,
    });
}
export function Loading(props) {
    const service = useService(ProductServiceDefinition);
    const isLoading = service.isLoading.get();
    return props.children({
        isLoading,
    });
}
/**
 * Headless component for product slug display
 *
 * @component
 * @example
 * ```tsx
 * import { Product } from '@wix/stores/components';
 *
 * function ProductSlugDisplay() {
 *   return (
 *     <Product.Slug>
 *       {({ slug }) => (
 *         <a href={`/product/${slug}`}>
 *           View Product
 *         </a>
 *       )}
 *     </Product.Slug>
 *   );
 * }
 * ```
 */
export function Slug(props) {
    const service = useService(ProductServiceDefinition);
    const product = service.product.get();
    const slug = product.slug;
    return props.children({
        slug,
    });
}
/**
 * Headless component for product ribbon display
 *
 * @component
 * @example
 * ```tsx
 * import { Product } from '@wix/stores/components';
 *
 * function ProductRibbonDisplay() {
 *   return (
 *     <Product.Ribbon>
 *       {({ ribbon, hasRibbon }) => (
 *         hasRibbon ? (
 *           <span className="ribbon-badge">
 *             {ribbon}
 *           </span>
 *         ) : null
 *       )}
 *     </Product.Ribbon>
 *   );
 * }
 * ```
 */
export function Ribbon(props) {
    const service = useService(ProductServiceDefinition);
    const product = service.product.get();
    const ribbon = product.ribbon?.name || null;
    const hasRibbon = !!ribbon;
    return props.children({
        ribbon,
        hasRibbon,
    });
}
