// @wix/headless-restaurants-menus/dist/react/core/Item.js
import { jsx as _jsx } from "react/jsx-runtime";
import { createContext, useContext } from 'react';
const ItemContext = createContext(null);
export function Item(props) {
    const contextValue = {
        item: props.item,
    };
    return (_jsx(ItemContext.Provider, { value: contextValue, children: props.children }));
}
export function useItemContext() {
    const context = useContext(ItemContext);
    if (!context) {
        throw new Error('useItemContext must be used within Item');
    }
    return context;
}
export function Name(props) {
    const { item } = useItemContext();
    return props.children({ name: item.name ?? '' });
}
export function Description(props) {
    const { item } = useItemContext();
    return props.children({ description: item.description ?? '' });
}
export function Price(props) {
    const { item } = useItemContext();
    const price = item.priceInfo?.price ?? '';
    const formattedPrice = item.priceInfo
        ?.formattedPrice;
    return props.children({ price, formattedPrice });
}
export function Images(props) {
    const { item } = useItemContext();
    const mainImage = item.image;
    const additionalImages = item.additionalImages || [];
    const images = mainImage
        ? [mainImage, ...additionalImages]
        : additionalImages;
    const altText = item.name ?? '';
    return props.children({
        images,
        altText,
    });
}
export function Featured(props) {
    const { item } = useItemContext();
    return props.children({ featured: !!item.featured });
}
