// @wix/auto_sdk_auth-management_o-auth-apps/build/internal/es/meta.mjs
// src/headless-v1-o-auth-app-o-auth-apps.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixHeadlessV1OAuthAppServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/oauth-app-service",
        destPath: ""
      },
      {
        srcPath: "/oauth-app",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/oauth-app",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/oauth-app-service",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_auth-management_o-auth-apps";
function createOAuthApp(payload) {
  function __createOAuthApp({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "oAuthApp.createdDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.headless.v1.o_auth_app",
      method: "POST",
      methodFqn: "wix.headless.v1.OAuthAppService.CreateOAuthApp",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixHeadlessV1OAuthAppServiceUrl({
        protoPath: "/v1/oauth-apps",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "oAuthApp.createdDate" }]
        }
      ])
    };
    return metadata;
  }
  return __createOAuthApp;
}
function getOAuthApp(payload) {
  function __getOAuthApp({ host }) {
    const metadata = {
      entityFqdn: "wix.headless.v1.o_auth_app",
      method: "GET",
      methodFqn: "wix.headless.v1.OAuthAppService.GetOAuthApp",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixHeadlessV1OAuthAppServiceUrl({
        protoPath: "/v1/oauth-apps/{oAuthAppId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "oAuthApp.createdDate" }]
        }
      ])
    };
    return metadata;
  }
  return __getOAuthApp;
}
function updateOAuthApp(payload) {
  function __updateOAuthApp({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "oAuthApp.createdDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.headless.v1.o_auth_app",
      method: "PATCH",
      methodFqn: "wix.headless.v1.OAuthAppService.UpdateOAuthApp",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixHeadlessV1OAuthAppServiceUrl({
        protoPath: "/v1/oauth-apps/{oAuthApp.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "oAuthApp.createdDate" }]
        }
      ])
    };
    return metadata;
  }
  return __updateOAuthApp;
}
function deleteOAuthApp(payload) {
  function __deleteOAuthApp({ host }) {
    const metadata = {
      entityFqdn: "wix.headless.v1.o_auth_app",
      method: "DELETE",
      methodFqn: "wix.headless.v1.OAuthAppService.DeleteOAuthApp",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixHeadlessV1OAuthAppServiceUrl({
        protoPath: "/v1/oauth-apps/{oAuthAppId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteOAuthApp;
}
function queryOAuthApps(payload) {
  function __queryOAuthApps({ host }) {
    const metadata = {
      entityFqdn: "wix.headless.v1.o_auth_app",
      method: "POST",
      methodFqn: "wix.headless.v1.OAuthAppService.QueryOAuthApps",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixHeadlessV1OAuthAppServiceUrl({
        protoPath: "/v1/oauth-apps/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "oAuthApps.createdDate" }]
        }
      ])
    };
    return metadata;
  }
  return __queryOAuthApps;
}
function generateOAuthAppSecret(payload) {
  function __generateOAuthAppSecret({ host }) {
    const metadata = {
      entityFqdn: "wix.headless.v1.o_auth_app",
      method: "POST",
      methodFqn: "wix.headless.v1.OAuthAppService.GenerateOAuthAppSecret",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixHeadlessV1OAuthAppServiceUrl({
        protoPath: "/v1/oauth-apps/{oAuthAppId}/generate-secret",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __generateOAuthAppSecret;
}

// src/headless-v1-o-auth-app-o-auth-apps.types.ts
var OAuthAppType = /* @__PURE__ */ ((OAuthAppType2) => {
  OAuthAppType2["OAUTH_APP_TYPE_UNSPECIFIED"] = "OAUTH_APP_TYPE_UNSPECIFIED";
  OAuthAppType2["WEB_APP"] = "WEB_APP";
  OAuthAppType2["MOBILE"] = "MOBILE";
  OAuthAppType2["OTHER"] = "OTHER";
  return OAuthAppType2;
})(OAuthAppType || {});
var OAuthTechnologies = /* @__PURE__ */ ((OAuthTechnologies2) => {
  OAuthTechnologies2["OAUTH_TECHNOLOGY_UNSPECIFIED"] = "OAUTH_TECHNOLOGY_UNSPECIFIED";
  OAuthTechnologies2["JAVASCRIPT"] = "JAVASCRIPT";
  OAuthTechnologies2["ANGULAR"] = "ANGULAR";
  OAuthTechnologies2["VUE"] = "VUE";
  OAuthTechnologies2["REACT"] = "REACT";
  OAuthTechnologies2["REACT_NATIVE"] = "REACT_NATIVE";
  OAuthTechnologies2["IOS"] = "IOS";
  OAuthTechnologies2["ANDROID"] = "ANDROID";
  OAuthTechnologies2["OTHER_TECHNOLOGY"] = "OTHER_TECHNOLOGY";
  return OAuthTechnologies2;
})(OAuthTechnologies || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var State = /* @__PURE__ */ ((State2) => {
  State2["UNKNOWN"] = "UNKNOWN";
  State2["ENABLED"] = "ENABLED";
  State2["DISABLED"] = "DISABLED";
  State2["PENDING"] = "PENDING";
  State2["DEMO"] = "DEMO";
  return State2;
})(State || {});
var SiteCreatedContext = /* @__PURE__ */ ((SiteCreatedContext2) => {
  SiteCreatedContext2["OTHER"] = "OTHER";
  SiteCreatedContext2["FROM_TEMPLATE"] = "FROM_TEMPLATE";
  SiteCreatedContext2["DUPLICATE_BY_SITE_TRANSFER"] = "DUPLICATE_BY_SITE_TRANSFER";
  SiteCreatedContext2["DUPLICATE"] = "DUPLICATE";
  SiteCreatedContext2["OLD_SITE_TRANSFER"] = "OLD_SITE_TRANSFER";
  SiteCreatedContext2["FLASH"] = "FLASH";
  return SiteCreatedContext2;
})(SiteCreatedContext || {});
var Namespace = /* @__PURE__ */ ((Namespace2) => {
  Namespace2["UNKNOWN_NAMESPACE"] = "UNKNOWN_NAMESPACE";
  Namespace2["WIX"] = "WIX";
  Namespace2["SHOUT_OUT"] = "SHOUT_OUT";
  Namespace2["ALBUMS"] = "ALBUMS";
  Namespace2["WIX_STORES_TEST_DRIVE"] = "WIX_STORES_TEST_DRIVE";
  Namespace2["HOTELS"] = "HOTELS";
  Namespace2["CLUBS"] = "CLUBS";
  Namespace2["ONBOARDING_DRAFT"] = "ONBOARDING_DRAFT";
  Namespace2["DEV_SITE"] = "DEV_SITE";
  Namespace2["LOGOS"] = "LOGOS";
  Namespace2["VIDEO_MAKER"] = "VIDEO_MAKER";
  Namespace2["PARTNER_DASHBOARD"] = "PARTNER_DASHBOARD";
  Namespace2["DEV_CENTER_COMPANY"] = "DEV_CENTER_COMPANY";
  Namespace2["HTML_DRAFT"] = "HTML_DRAFT";
  Namespace2["SITELESS_BUSINESS"] = "SITELESS_BUSINESS";
  Namespace2["CREATOR_ECONOMY"] = "CREATOR_ECONOMY";
  Namespace2["DASHBOARD_FIRST"] = "DASHBOARD_FIRST";
  Namespace2["ANYWHERE"] = "ANYWHERE";
  Namespace2["HEADLESS"] = "HEADLESS";
  Namespace2["ACCOUNT_MASTER_CMS"] = "ACCOUNT_MASTER_CMS";
  Namespace2["RISE"] = "RISE";
  Namespace2["BRANDED_FIRST"] = "BRANDED_FIRST";
  Namespace2["NOWNIA"] = "NOWNIA";
  Namespace2["UGC_TEMPLATE"] = "UGC_TEMPLATE";
  Namespace2["CODUX"] = "CODUX";
  Namespace2["MEDIA_DESIGN_CREATOR"] = "MEDIA_DESIGN_CREATOR";
  Namespace2["SHARED_BLOG_ENTERPRISE"] = "SHARED_BLOG_ENTERPRISE";
  Namespace2["STANDALONE_FORMS"] = "STANDALONE_FORMS";
  Namespace2["STANDALONE_EVENTS"] = "STANDALONE_EVENTS";
  Namespace2["MIMIR"] = "MIMIR";
  Namespace2["TWINS"] = "TWINS";
  Namespace2["NANO"] = "NANO";
  return Namespace2;
})(Namespace || {});
var DeleteStatus = /* @__PURE__ */ ((DeleteStatus2) => {
  DeleteStatus2["UNKNOWN"] = "UNKNOWN";
  DeleteStatus2["TRASH"] = "TRASH";
  DeleteStatus2["DELETED"] = "DELETED";
  DeleteStatus2["PENDING_PURGE"] = "PENDING_PURGE";
  DeleteStatus2["PURGED_EXTERNALLY"] = "PURGED_EXTERNALLY";
  return DeleteStatus2;
})(DeleteStatus || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/headless-v1-o-auth-app-o-auth-apps.meta.ts
function createOAuthApp2() {
  const payload = {};
  const getRequestOptions = createOAuthApp(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/oauth-apps",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getOAuthApp2() {
  const payload = { oAuthAppId: ":oAuthAppId" };
  const getRequestOptions = getOAuthApp(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/oauth-apps/{oAuthAppId}",
    pathParams: { oAuthAppId: "oAuthAppId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateOAuthApp2() {
  const payload = { oAuthApp: { id: ":oAuthAppId" } };
  const getRequestOptions = updateOAuthApp(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/oauth-apps/{oAuthApp.id}",
    pathParams: { oAuthAppId: "oAuthAppId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteOAuthApp2() {
  const payload = { oAuthAppId: ":oAuthAppId" };
  const getRequestOptions = deleteOAuthApp(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/oauth-apps/{oAuthAppId}",
    pathParams: { oAuthAppId: "oAuthAppId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryOAuthApps2() {
  const payload = {};
  const getRequestOptions = queryOAuthApps(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/oauth-apps/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function generateOAuthAppSecret2() {
  const payload = { oAuthAppId: ":oAuthAppId" };
  const getRequestOptions = generateOAuthAppSecret(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/oauth-apps/{oAuthAppId}/generate-secret",
    pathParams: { oAuthAppId: "oAuthAppId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  DeleteStatus as DeleteStatusOriginal,
  Namespace as NamespaceOriginal,
  OAuthAppType as OAuthAppTypeOriginal,
  OAuthTechnologies as OAuthTechnologiesOriginal,
  SiteCreatedContext as SiteCreatedContextOriginal,
  SortOrder as SortOrderOriginal,
  State as StateOriginal,
  WebhookIdentityType as WebhookIdentityTypeOriginal,
  createOAuthApp2 as createOAuthApp,
  deleteOAuthApp2 as deleteOAuthApp,
  generateOAuthAppSecret2 as generateOAuthAppSecret,
  getOAuthApp2 as getOAuthApp,
  queryOAuthApps2 as queryOAuthApps,
  updateOAuthApp2 as updateOAuthApp
};
