// @wix/auto_sdk_data_sharing/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  connectToSharedCollection: () => connectToSharedCollection2,
  createDataSharingPolicy: () => createDataSharingPolicy2,
  deleteDataSharingPolicy: () => deleteDataSharingPolicy2,
  disconnectFromSharedCollection: () => disconnectFromSharedCollection2,
  getDataSharingPolicy: () => getDataSharingPolicy2,
  listDataSharingPolicies: () => listDataSharingPolicies2,
  listSharedDataCollections: () => listSharedDataCollections2,
  updateDataSharingPolicy: () => updateDataSharingPolicy2
});
module.exports = __toCommonJS(meta_exports);

// src/data-v1-data-sharing-policy-sharing.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressCloudDataApiShareDataCollectionSharingServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/cloud-data",
        destPath: ""
      }
    ],
    "code._base_domain_": [
      {
        srcPath: "/_api/cloud-data/v1/data-settings",
        destPath: "/v1/data-settings"
      },
      {
        srcPath: "/_api/cloud-data/v1/schemas",
        destPath: "/v1/schemas"
      },
      {
        srcPath: "/_api/cloud-data/v1/connector",
        destPath: "/v1/connector"
      },
      {
        srcPath: "/_api/data/v1/data-settings",
        destPath: "/v1/data-settings"
      },
      {
        srcPath: "/_api/data/v1/schemas",
        destPath: "/v1/schemas"
      },
      {
        srcPath: "/_api/data/v1/connector",
        destPath: "/v1/connector"
      },
      {
        srcPath: "/_api/cloud-data/v1/wix-data",
        destPath: "/v1/wix-data"
      }
    ],
    "cloud-data.wix-code.com": [
      {
        srcPath: "",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/cloud-data",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/cloud-data",
        destPath: ""
      },
      {
        srcPath: "/_api/data",
        destPath: ""
      },
      {
        srcPath: "/_api/cloud-data",
        destPath: ""
      }
    ],
    "dev._base_domain_": [
      {
        srcPath: "/_api/cloud-data",
        destPath: ""
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/_api/cloud-data/v1",
        destPath: "/v1"
      },
      {
        srcPath: "/_api/cloud-data/v2",
        destPath: "/v2"
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/cloud-data/v1",
        destPath: "/v1"
      },
      {
        srcPath: "/_api/cloud-data/v2",
        destPath: "/v2"
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/cloud-data/v1",
        destPath: "/v1"
      },
      {
        srcPath: "/_api/cloud-data/v2",
        destPath: "/v2"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/cloud-data",
        destPath: ""
      },
      {
        srcPath: "/_api/data",
        destPath: ""
      },
      {
        srcPath: "/_api/cloud-data",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/cloud-data/v1/schemas",
        destPath: "/v1/schemas"
      },
      {
        srcPath: "/_api/cloud-data/dbs/tasks",
        destPath: "/dbs/tasks"
      },
      {
        srcPath: "/_api/data/v1/schemas",
        destPath: "/v1/schemas"
      },
      {
        srcPath: "/_api/data/dbs/tasks",
        destPath: "/dbs/tasks"
      },
      {
        srcPath: "/_api/cloud-data/v1/wix-data",
        destPath: "/v1/wix-data"
      },
      {
        srcPath: "/_api/cloud-data/v1/data-settings",
        destPath: "/v1/data-settings"
      },
      {
        srcPath: "/_api/data/v2/indexes",
        destPath: "/v2/indexes"
      },
      {
        srcPath: "/_api/cloud-data/v1/external-database-connections",
        destPath: "/v1/external-database-connections"
      },
      {
        srcPath: "/_api/data/v1/data-collection-sharing",
        destPath: "/v1/data-collection-sharing"
      },
      {
        srcPath: "/_api/cloud-data/v2/collections",
        destPath: "/v2/collections"
      },
      {
        srcPath: "/_api/cloud-data/v2/bulk",
        destPath: "/v2/bulk"
      },
      {
        srcPath: "/_api/cloud-data/v2/items",
        destPath: "/v2/items"
      },
      {
        srcPath: "/_api/cloud-data/v2/dbs/tasks",
        destPath: "/v2/dbs/tasks"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/cloud-data/v1/schemas",
        destPath: "/v1/schemas"
      },
      {
        srcPath: "/_api/cloud-data/dbs/tasks",
        destPath: "/dbs/tasks"
      },
      {
        srcPath: "/_api/data/v1/schemas",
        destPath: "/v1/schemas"
      },
      {
        srcPath: "/_api/data/dbs/tasks",
        destPath: "/dbs/tasks"
      },
      {
        srcPath: "/_api/cloud-data/v1/wix-data",
        destPath: "/v1/wix-data"
      },
      {
        srcPath: "/_api/cloud-data/v1/data-settings",
        destPath: "/v1/data-settings"
      },
      {
        srcPath: "/_api/data/v2/indexes",
        destPath: "/v2/indexes"
      },
      {
        srcPath: "/_api/cloud-data/v1/external-database-connections",
        destPath: "/v1/external-database-connections"
      },
      {
        srcPath: "/_api/data/v1/data-collection-sharing",
        destPath: "/v1/data-collection-sharing"
      },
      {
        srcPath: "/_api/cloud-data/v2/collections",
        destPath: "/v2/collections"
      },
      {
        srcPath: "/_api/cloud-data/v2/bulk",
        destPath: "/v2/bulk"
      },
      {
        srcPath: "/_api/cloud-data/v2/items",
        destPath: "/v2/items"
      },
      {
        srcPath: "/_api/cloud-data/v2/dbs/tasks",
        destPath: "/v2/dbs/tasks"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/cloud-data/v1/schemas",
        destPath: "/v1/schemas"
      },
      {
        srcPath: "/_api/cloud-data/dbs/tasks",
        destPath: "/dbs/tasks"
      },
      {
        srcPath: "/_api/data/v1/schemas",
        destPath: "/v1/schemas"
      },
      {
        srcPath: "/_api/data/dbs/tasks",
        destPath: "/dbs/tasks"
      },
      {
        srcPath: "/_api/cloud-data/v1/wix-data",
        destPath: "/v1/wix-data"
      },
      {
        srcPath: "/_api/cloud-data/v1/data-settings",
        destPath: "/v1/data-settings"
      },
      {
        srcPath: "/_api/data/v2/indexes",
        destPath: "/v2/indexes"
      },
      {
        srcPath: "/_api/cloud-data/v1/external-database-connections",
        destPath: "/v1/external-database-connections"
      },
      {
        srcPath: "/_api/data/v1/data-collection-sharing",
        destPath: "/v1/data-collection-sharing"
      },
      {
        srcPath: "/_api/cloud-data/v2/collections",
        destPath: "/v2/collections"
      },
      {
        srcPath: "/_api/cloud-data/v2/bulk",
        destPath: "/v2/bulk"
      },
      {
        srcPath: "/_api/cloud-data/v2/items",
        destPath: "/v2/items"
      },
      {
        srcPath: "/_api/cloud-data/v2/dbs/tasks",
        destPath: "/v2/dbs/tasks"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/wix-data/v1/collections",
        destPath: "/v1/wix-data/collections"
      },
      {
        srcPath: "/wix-data/v1/external-database-connections",
        destPath: "/v1/external-database-connections"
      },
      {
        srcPath: "/wix-data/v2/indexes",
        destPath: "/v2/indexes"
      },
      {
        srcPath: "/wix-data/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/wix-data/v1/bulk",
        destPath: "/v1/bulk"
      },
      {
        srcPath: "/wix-data/v1/external-databases",
        destPath: "/v1/external-databases"
      },
      {
        srcPath: "/wix-data/v2",
        destPath: "/v2"
      },
      {
        srcPath: "/data/v2/collections",
        destPath: "/v2/collections"
      },
      {
        srcPath: "/data/v2/items",
        destPath: "/v2/items"
      },
      {
        srcPath: "/data/v2/bulk/items",
        destPath: "/v2/bulk/items"
      },
      {
        srcPath: "/data/v2/indexes",
        destPath: "/v2/indexes"
      },
      {
        srcPath: "/data/v1/external-database-connections",
        destPath: "/v1/external-database-connections"
      },
      {
        srcPath: "/data/v1/data-collection-sharing",
        destPath: "/v1/data-collection-sharing"
      }
    ],
    "www.wixgateway.com": [
      {
        srcPath: "/wix-data/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/wix-data/v1/bulk",
        destPath: "/v1/bulk"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/cloud-data/v2",
        destPath: "/v2"
      },
      {
        srcPath: "/_api/cloud-data/v1/external-database-connections",
        destPath: "/v1/external-database-connections"
      },
      {
        srcPath: "/data/v1/data-collection-sharing",
        destPath: "/v1/data-collection-sharing"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_data_sharing";
function createDataSharingPolicy(payload) {
  function __createDataSharingPolicy({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "dataSharingPolicy.createdDate" },
          { path: "dataSharingPolicy.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.data.v1.data_sharing_policy",
      method: "POST",
      methodFqn: "com.wixpress.cloud.data.api.share.DataCollectionSharingService.CreateDataSharingPolicy",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressCloudDataApiShareDataCollectionSharingServiceUrl({
        protoPath: "/v1/data-collection-sharing/policies",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "dataSharingPolicy.createdDate" },
            { path: "dataSharingPolicy.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createDataSharingPolicy;
}
function updateDataSharingPolicy(payload) {
  function __updateDataSharingPolicy({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "dataSharingPolicy.createdDate" },
          { path: "dataSharingPolicy.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.data.v1.data_sharing_policy",
      method: "POST",
      methodFqn: "com.wixpress.cloud.data.api.share.DataCollectionSharingService.UpdateDataSharingPolicy",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressCloudDataApiShareDataCollectionSharingServiceUrl({
        protoPath: "/v1/data-collection-sharing/policies/{dataSharingPolicy.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "dataSharingPolicy.createdDate" },
            { path: "dataSharingPolicy.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateDataSharingPolicy;
}
function deleteDataSharingPolicy(payload) {
  function __deleteDataSharingPolicy({ host }) {
    const metadata = {
      entityFqdn: "wix.data.v1.data_sharing_policy",
      method: "DELETE",
      methodFqn: "com.wixpress.cloud.data.api.share.DataCollectionSharingService.DeleteDataSharingPolicy",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressCloudDataApiShareDataCollectionSharingServiceUrl({
        protoPath: "/v1/data-collection-sharing/policies/{dataSharingPolicyId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteDataSharingPolicy;
}
function getDataSharingPolicy(payload) {
  function __getDataSharingPolicy({ host }) {
    const metadata = {
      entityFqdn: "wix.data.v1.data_sharing_policy",
      method: "GET",
      methodFqn: "com.wixpress.cloud.data.api.share.DataCollectionSharingService.GetDataSharingPolicy",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressCloudDataApiShareDataCollectionSharingServiceUrl({
        protoPath: "/v1/data-collection-sharing/policies/{dataSharingPolicyId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "dataSharingPolicy.createdDate" },
            { path: "dataSharingPolicy.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getDataSharingPolicy;
}
function listDataSharingPolicies(payload) {
  function __listDataSharingPolicies({ host }) {
    const metadata = {
      entityFqdn: "wix.data.v1.data_sharing_policy",
      method: "GET",
      methodFqn: "com.wixpress.cloud.data.api.share.DataCollectionSharingService.ListDataSharingPolicies",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressCloudDataApiShareDataCollectionSharingServiceUrl({
        protoPath: "/v1/data-collection-sharing/policies",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "dataSharingPolicies.createdDate" },
            { path: "dataSharingPolicies.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listDataSharingPolicies;
}
function listSharedDataCollections(payload) {
  function __listSharedDataCollections({ host }) {
    const metadata = {
      entityFqdn: "wix.data.v1.data_sharing_policy",
      method: "GET",
      methodFqn: "com.wixpress.cloud.data.api.share.DataCollectionSharingService.ListSharedDataCollections",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressCloudDataApiShareDataCollectionSharingServiceUrl({
        protoPath: "/v1/data-collection-sharing/shared",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "connections.createdDate" },
            { path: "connections.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listSharedDataCollections;
}
function connectToSharedCollection(payload) {
  function __connectToSharedCollection({ host }) {
    const metadata = {
      entityFqdn: "wix.data.v1.data_sharing_policy",
      method: "POST",
      methodFqn: "com.wixpress.cloud.data.api.share.DataCollectionSharingService.ConnectToSharedCollection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressCloudDataApiShareDataCollectionSharingServiceUrl({
        protoPath: "/v1/data-collection-sharing/connect-to-shared-collection",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "connection.createdDate" },
            { path: "connection.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __connectToSharedCollection;
}
function disconnectFromSharedCollection(payload) {
  function __disconnectFromSharedCollection({ host }) {
    const metadata = {
      entityFqdn: "wix.data.v1.data_sharing_policy",
      method: "POST",
      methodFqn: "com.wixpress.cloud.data.api.share.DataCollectionSharingService.DisconnectFromSharedCollection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressCloudDataApiShareDataCollectionSharingServiceUrl({
        protoPath: "/v1/data-collection-sharing/disconnect-from-shared-collection",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __disconnectFromSharedCollection;
}

// src/data-v1-data-sharing-policy-sharing.types.ts
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/data-v1-data-sharing-policy-sharing.meta.ts
function createDataSharingPolicy2() {
  const payload = {};
  const getRequestOptions = createDataSharingPolicy(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/data-collection-sharing/policies",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateDataSharingPolicy2() {
  const payload = { dataSharingPolicy: { id: ":dataSharingPolicyId" } };
  const getRequestOptions = updateDataSharingPolicy(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/data-collection-sharing/policies/{dataSharingPolicy.id}",
    pathParams: { dataSharingPolicyId: "dataSharingPolicyId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteDataSharingPolicy2() {
  const payload = { dataSharingPolicyId: ":dataSharingPolicyId" };
  const getRequestOptions = deleteDataSharingPolicy(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/data-collection-sharing/policies/{dataSharingPolicyId}",
    pathParams: { dataSharingPolicyId: "dataSharingPolicyId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getDataSharingPolicy2() {
  const payload = { dataSharingPolicyId: ":dataSharingPolicyId" };
  const getRequestOptions = getDataSharingPolicy(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/data-collection-sharing/policies/{dataSharingPolicyId}",
    pathParams: { dataSharingPolicyId: "dataSharingPolicyId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listDataSharingPolicies2() {
  const payload = {};
  const getRequestOptions = listDataSharingPolicies(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/data-collection-sharing/policies",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listSharedDataCollections2() {
  const payload = {};
  const getRequestOptions = listSharedDataCollections(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/data-collection-sharing/shared",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function connectToSharedCollection2() {
  const payload = {};
  const getRequestOptions = connectToSharedCollection(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/data-collection-sharing/connect-to-shared-collection",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function disconnectFromSharedCollection2() {
  const payload = {};
  const getRequestOptions = disconnectFromSharedCollection(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/data-collection-sharing/disconnect-from-shared-collection",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  WebhookIdentityTypeOriginal,
  connectToSharedCollection,
  createDataSharingPolicy,
  deleteDataSharingPolicy,
  disconnectFromSharedCollection,
  getDataSharingPolicy,
  listDataSharingPolicies,
  listSharedDataCollections,
  updateDataSharingPolicy
});
