// @wix/bex-utils/@wix/ambassador-fedinfra-exportservice-v1-export-async-job/build/es/builders.impl.js
import { mergeDeep, someNumber, someString, someAny, someDate, } from '@wix/metro-common-builders';
import { MeasurementUnit, DiscountType, MediaItemType, OptionType, ProductType, InventoryStatus, SortOrder, Status, QueryFieldNumber, RuleType, ArrayFieldDelimiter, FileType, } from './types.impl';
// @ts-ignore
import stack from 'callsite';
export function aExportAsyncJob(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        id: someString({
            validations: [
                { type: 'FORMAT', value: 'GUID' },
                { type: 'READ_ONLY', value: 'true' },
            ],
        }),
        data: aExportAsyncJobData(),
        status: aStatus(),
        createDate: someDate({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        updateDate: someDate({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
    }, overrideObject);
}
export function aExportAsyncJobData(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        query: aExportQueryV2(),
        fields: [aFieldDescriptor()],
        arrayFieldDelimiter: aArrayFieldDelimiter(),
        methodMetadata: aMethodMetadata(),
        methodSpec: aMethodSpec(),
        userId: someString(),
        subDir: someString(),
        format: someString(),
        uploadId: someString(),
        saveAs: someString({
            validations: [{ type: 'MAX_LENGTH', value: '255' }],
        }),
        pagingMetadata: aPagingMetadataV2(),
        iterationsCount: someNumber(),
        processedItemsCount: someNumber(),
        completionStep: someString(),
        resetCount: someNumber(),
        signedUrl: someString(),
        error: aDetails(),
        testParams: {},
    }, overrideObject);
}
export function aExportQueryV2(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        paging: aPaging(),
        filter: {},
        sort: [aSorting()],
        fields: [someString()],
        fieldsets: [someString()],
    }, overrideObject);
}
export function aSorting(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        fieldName: someString({
            validations: [{ type: 'MAX_LENGTH', value: '512' }],
        }),
        order: aSortOrder(),
        selectItemsBy: [{}, {}, {}, {}, {}, {}, {}, {}, {}, {}],
    }, overrideObject);
}
export function aSortOrder(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = SortOrder.ASC; }
    return enumFirstItem;
}
export function aPaging(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        limit: someNumber({
            validations: [{ type: 'MIN' }],
        }),
        offset: someNumber({
            validations: [{ type: 'MIN' }],
        }),
    }, overrideObject);
}
export function aExportCursorPaging(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        limit: someNumber({
            validations: [
                { type: 'MIN' },
                { type: 'MAX', value: '10000' },
            ],
        }),
        cursor: someString(),
    }, overrideObject);
}
export function aFieldDescriptor(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        id: someString(),
        header: someString(),
    }, overrideObject);
}
export function aArrayFieldDelimiter(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = ArrayFieldDelimiter.SEMICOLON; }
    return enumFirstItem;
}
export function aMethodMetadata(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        artifact: someString(),
        service: someString(),
        method: someString(),
    }, overrideObject);
}
export function aMethodSpec(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        requestQueryFieldNumber: aQueryFieldNumber(),
        responseRepeatedFieldName: someString(),
        responsePagingMetadataFieldName: someString(),
    }, overrideObject);
}
export function aQueryFieldNumber(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = QueryFieldNumber.DEFAULT; }
    return enumFirstItem;
}
export function aPagingMetadataV2(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        count: someNumber(),
        offset: someNumber(),
        total: someNumber(),
        tooManyToCount: true,
        cursors: aCursors(),
        hasNext: true,
    }, overrideObject);
}
export function aCursors(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        next: someString({
            validations: [
                { type: 'MAX_LENGTH', value: '16000' },
            ],
        }),
        prev: someString({
            validations: [
                { type: 'MAX_LENGTH', value: '16000' },
            ],
        }),
    }, overrideObject);
}
export function aDetails(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        applicationError: aApplicationError(),
        tracing: { tracing: someString() },
    }, overrideObject);
}
export function aApplicationError(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        code: someString(),
        description: someString(),
        data: {},
    }, overrideObject);
}
export function aValidationError(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        fieldViolations: [aFieldViolation()],
    }, overrideObject);
}
export function aRuleType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = RuleType.VALIDATION; }
    return enumFirstItem;
}
export function aFieldViolation(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        field: someString(),
        description: someString(),
        violatedRule: aRuleType(),
        ruleName: someString(),
        data: {},
    }, overrideObject);
}
export function aSystemError(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        errorCode: someString(),
    }, overrideObject);
}
export function aStatus(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = Status.UNKNOWN; }
    return enumFirstItem;
}
export function aCreateExportAsyncJobRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        query: aExportQueryV2(),
        fields: [aFieldDescriptor()],
        arrayFieldDelimiter: aArrayFieldDelimiter(),
        methodMetadata: aMethodMetadata(),
        methodSpec: aMethodSpec(),
        saveAs: someString({
            validations: [{ type: 'MAX_LENGTH', value: '255' }],
        }),
        testParams: {},
    }, overrideObject);
}
export function aCreateExportAsyncJobResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        job: aExportAsyncJob(),
    }, overrideObject);
}
export function aGetExportAsyncJobRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        jobId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
export function aGetExportAsyncJobResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        job: aExportAsyncJob(),
    }, overrideObject);
}
export function aCancelExportAsyncJobRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        jobId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
export function aCancelExportAsyncJobResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        job: aExportAsyncJob(),
    }, overrideObject);
}
export function aGenerateExportAsyncJobDownloadUrlRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        jobId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
export function aGenerateExportAsyncJobDownloadUrlResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        job: aExportAsyncJob(),
    }, overrideObject);
}
export function aQueryRequestLoose(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        query: aExportQueryV2(),
    }, overrideObject);
}
export function aQueryVariantsExportSpiResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        items: [aProductOrVariant()],
        pagingMetadata: aPagingMetadataV2(),
    }, overrideObject);
}
export function aProductOrVariant(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        kind: someString(),
        product: aProduct(),
        variant: aStoreVariant(),
    }, overrideObject);
}
export function aProduct(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        id: someString({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        name: someString({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '80' },
            ],
        }),
        slug: someString({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        visible: true,
        productType: aProductType(),
        description: someString({
            validations: [{ type: 'MAX_LENGTH', value: '8000' }],
        }),
        sku: someString({
            validations: [{ type: 'MAX_LENGTH', value: '40' }],
        }),
        weight: someNumber({
            validations: [
                { type: 'MIN' },
                { type: 'MAX', value: '999999999.99' },
            ],
        }),
        weightRange: aNumericPropertyRange(),
        stock: aStock(),
        price: aPriceData(),
        priceData: aPriceData(),
        convertedPriceData: aPriceData(),
        priceRange: aNumericPropertyRange(),
        costAndProfitData: aCostAndProfitData(),
        costRange: aNumericPropertyRange(),
        pricePerUnitData: aPricePerUnitData(),
        additionalInfoSections: [aAdditionalInfoSection()],
        ribbons: [aRibbon()],
        media: aMedia(),
        customTextFields: [aCustomTextField()],
        manageVariants: true,
        productOptions: [
            aProductOption(),
            aProductOption(),
            aProductOption(),
            aProductOption(),
            aProductOption(),
            aProductOption(),
        ],
        productPageUrl: aPageUrl(),
        numericId: someString({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'DECIMAL_VALUE' },
            ],
            isUnsigned: true,
        }),
        inventoryItemId: someString({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        discount: aDiscount(),
        collectionIds: [
            someString({
                validations: [
                    { type: 'READ_ONLY', value: 'true' },
                ],
            }),
        ],
        variants: [
            aVariant(),
            aVariant(),
            aVariant(),
            aVariant(),
            aVariant(),
            aVariant(),
            aVariant(),
            aVariant(),
            aVariant(),
            aVariant(),
        ],
        lastUpdated: someDate({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        createdDate: someDate({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        seoData: aSeoSchema(),
        ribbon: someString({
            validations: [{ type: 'MAX_LENGTH', value: '30' }],
        }),
        brand: someString({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '50' },
            ],
        }),
        taxGroupId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        exportProductId: someString({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MAX_LENGTH', value: '100' },
            ],
        }),
        digitalFile: aSecuredMedia(),
    }, overrideObject);
}
export function aProductType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = ProductType.unspecified_product_type; }
    return enumFirstItem;
}
export function aNumericPropertyRange(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        minValue: someNumber(),
        maxValue: someNumber(),
    }, overrideObject);
}
export function aStock(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        trackInventory: true,
        quantity: someNumber({
            validations: [
                { type: 'MIN' },
                { type: 'MAX', value: '99999' },
            ],
        }),
        inStock: true,
        inventoryStatus: aInventoryStatus(),
    }, overrideObject);
}
export function aInventoryStatus(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = InventoryStatus.IN_STOCK; }
    return enumFirstItem;
}
export function aPriceData(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        currency: someString({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        price: someNumber({
            validations: [
                { type: 'MIN' },
                { type: 'MAX', value: '999999999.99' },
            ],
        }),
        discountedPrice: someNumber({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        formatted: aFormattedPrice(),
        pricePerUnit: someNumber({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
    }, overrideObject);
}
export function aFormattedPrice(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        price: someString(),
        discountedPrice: someString(),
        pricePerUnit: someString({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
    }, overrideObject);
}
export function aCostAndProfitData(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        itemCost: someNumber({
            validations: [
                { type: 'MIN' },
                { type: 'MAX', value: '999999999.99' },
            ],
        }),
        formattedItemCost: someString({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        profit: someNumber({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MIN' },
                { type: 'MAX', value: '999999999.99' },
            ],
        }),
        formattedProfit: someString({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        profitMargin: someNumber({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MIN' },
                { type: 'MAX', value: '1' },
            ],
        }),
    }, overrideObject);
}
export function aPricePerUnitData(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        totalQuantity: someNumber({
            validations: [
                { type: 'MIN', value: '0.01' },
                { type: 'MAX', value: '999999999.99' },
            ],
        }),
        totalMeasurementUnit: aMeasurementUnit(),
        baseQuantity: someNumber({
            validations: [
                { type: 'MIN', value: '0.01' },
                { type: 'MAX', value: '999999999.99' },
            ],
        }),
        baseMeasurementUnit: aMeasurementUnit(),
    }, overrideObject);
}
export function aMeasurementUnit(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = MeasurementUnit.UNSPECIFIED; }
    return enumFirstItem;
}
export function aAdditionalInfoSection(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        title: someString({
            validations: [{ type: 'MAX_LENGTH', value: '50' }],
        }),
        description: someString({
            validations: [
                { type: 'MAX_LENGTH', value: '16000' },
            ],
        }),
    }, overrideObject);
}
export function aRibbon(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        text: someString({
            validations: [{ type: 'MAX_LENGTH', value: '30' }],
        }),
    }, overrideObject);
}
export function aMedia(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        mainMedia: aMediaItem(),
        items: [aMediaItem()],
    }, overrideObject);
}
export function aMediaItem(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        image: aMediaItemUrlAndSize(),
        thumbnail: aMediaItemUrlAndSize(),
        mediaType: aMediaItemType(),
        title: someString(),
        id: someString(),
    }, overrideObject);
}
export function aMediaItemUrlAndSize(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        url: someString({
            validations: [{ type: 'FORMAT', value: 'WEB_URL' }],
        }),
        width: someNumber({
            validations: [{ type: 'MIN' }],
        }),
        height: someNumber({
            validations: [{ type: 'MIN' }],
        }),
        format: someString(),
        altText: someString(),
    }, overrideObject);
}
export function aMediaItemType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = MediaItemType.unspecified_media_item_type; }
    return enumFirstItem;
}
export function aMediaItemVideo(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        files: [aMediaItemUrlAndSize()],
        stillFrameMediaId: someString(),
    }, overrideObject);
}
export function aCustomTextField(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        title: someString({
            validations: [{ type: 'MAX_LENGTH', value: '100' }],
        }),
        maxLength: someNumber({
            validations: [{ type: 'MAX', value: '500' }],
        }),
        mandatory: true,
    }, overrideObject);
}
export function aProductOption(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        optionType: aOptionType(),
        name: someString({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '50' },
            ],
        }),
        choices: [aChoice()],
    }, overrideObject);
}
export function aOptionType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = OptionType.unspecified_option_type; }
    return enumFirstItem;
}
export function aChoice(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        value: someString({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '50' },
            ],
        }),
        description: someString({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '50' },
            ],
        }),
        media: aMedia(),
        inStock: true,
        visible: true,
    }, overrideObject);
}
export function aPageUrl(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        base: someString(),
        path: someString(),
    }, overrideObject);
}
export function aDiscount(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        type: aDiscountType(),
        value: someNumber(),
    }, overrideObject);
}
export function aDiscountType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = DiscountType.UNDEFINED; }
    return enumFirstItem;
}
export function aVariant(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        id: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        choices: { choices: someString() },
        variant: aVariantDataWithNoStock(),
        stock: aVariantStock(),
    }, overrideObject);
}
export function aVariantDataWithNoStock(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        priceData: aPriceData(),
        convertedPriceData: aPriceData(),
        costAndProfitData: aCostAndProfitData(),
        weight: someNumber(),
        sku: someString(),
        visible: true,
    }, overrideObject);
}
export function aVariantStock(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        trackQuantity: true,
        quantity: someNumber({
            validations: [
                { type: 'MIN' },
                { type: 'MAX', value: '99999' },
            ],
        }),
        inStock: true,
        locationId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
    }, overrideObject);
}
export function aSeoSchema(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        tags: [aTag()],
        settings: aSettings(),
    }, overrideObject);
}
export function aKeyword(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        term: someString(),
        isMain: true,
        origin: someString({
            validations: [{ type: 'MAX_LENGTH', value: '1000' }],
        }),
    }, overrideObject);
}
export function aTag(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        type: someString(),
        props: {},
        meta: {},
        children: someString(),
        custom: true,
        disabled: true,
    }, overrideObject);
}
export function aSettings(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        preventAutoRedirect: true,
        keywords: [aKeyword(), aKeyword(), aKeyword(), aKeyword(), aKeyword()],
    }, overrideObject);
}
export function aSecuredMedia(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        id: someString({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '100' },
            ],
        }),
        fileName: someString({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '1000' },
            ],
        }),
        fileType: aFileType(),
    }, overrideObject);
}
export function aFileType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = FileType.UNSPECIFIED; }
    return enumFirstItem;
}
export function aStoreVariant(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        id: someString(),
        variantId: someString({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        productId: someString(),
        variantName: someString({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '80' },
            ],
        }),
        productName: someString({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '80' },
            ],
        }),
        managedVariant: true,
        sku: someString({
            validations: [{ type: 'MAX_LENGTH', value: '40' }],
        }),
        stock: aVariantStock(),
        choices: { choices: someString() },
        collectionIds: [
            someString({
                validations: [{ type: 'FORMAT', value: 'GUID' }],
            }),
        ],
        media: aPlatformMedia(),
        customFields: { customFields: someAny() },
        preorderInfo: aPreorderInfo(),
    }, overrideObject);
}
export function aPlatformMedia(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        image: aImage(),
    }, overrideObject);
}
export function aImage(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        id: someString(),
        url: someString(),
        height: someNumber({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        width: someNumber({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        altText: someString(),
        urlExpirationDate: someDate({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        filename: someString({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        sizeInBytes: someString({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
            ],
        }),
        focalPoint: aFocalPoint(),
    }, overrideObject);
}
export function aFocalPoint(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        x: someNumber(),
        y: someNumber(),
        height: someNumber(),
        width: someNumber(),
    }, overrideObject);
}
export function aVideoV2(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        id: someString(),
        url: someString(),
        resolutions: [
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
            aVideoResolution(),
        ],
        filename: someString({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        posters: [aImage()],
        sizeInBytes: someString({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
            ],
        }),
        urlExpirationDate: someDate({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        durationInMilliseconds: someNumber({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MIN' },
            ],
        }),
        title: someString({
            validations: [{ type: 'MAX_LENGTH', value: '300' }],
        }),
        description: someString({
            validations: [{ type: 'MAX_LENGTH', value: '2000' }],
        }),
        height: someNumber({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MIN' },
            ],
        }),
        width: someNumber({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MIN' },
            ],
        }),
    }, overrideObject);
}
export function aVideoResolution(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        url: someString(),
        height: someNumber(),
        width: someNumber(),
        poster: aImage(),
        format: someString(),
        urlExpirationDate: someDate(),
        sizeInBytes: someString({
            validations: [
                { type: 'FORMAT', value: 'DECIMAL_VALUE' },
            ],
        }),
        quality: someString(),
        filename: someString(),
        durationInSeconds: someNumber({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MIN' },
            ],
        }),
        durationInMilliseconds: someNumber({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MIN' },
            ],
        }),
        private: true,
        assetKey: someString({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
    }, overrideObject);
}
export function aPreorderInfo(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return mergeDeep({
        enabled: true,
        message: someString({
            validations: [
                { type: 'MIN_LENGTH', value: '1' },
                { type: 'MAX_LENGTH', value: '250' },
            ],
        }),
        limit: someNumber({
            validations: [
                { type: 'MIN' },
                { type: 'MAX', value: '100000' },
            ],
        }),
    }, overrideObject);
}
export function aQueryProductsExportSpiResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        items: [aProductOrVariantV2()],
        pagingMetadata: aPagingMetadataV2(),
    }, overrideObject);
}
export function aProductOrVariantV2(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return mergeDeep({
        kind: someString(),
        product: aProduct(),
        variant: aVariant(),
    }, overrideObject);
}
function isCurrentBuilderWasPreviouslyCalled() {
    var callChainFunctionNames = stack().map(function (item) {
        return item.getFunctionName();
    });
    var callChainWithoutCurrentBuilder = callChainFunctionNames.splice(2);
    var currentBuilderName = callChainFunctionNames[1];
    return callChainWithoutCurrentBuilder.some(function (functionName) { return functionName === currentBuilderName; });
}
export * from '@wix/metro-common-builders';
