// @wix/bex-utils/@wix/ambassador-fedinfra-exportservice-v1-export-async-job/build/es/meta.impl.js
import * as httpWixFedinfraExportserviceV1ExportAsyncJob from './http.impl';
export function createExportAsyncJob() {
    var payload = {};
    var getRequestOptions = httpWixFedinfraExportserviceV1ExportAsyncJob.createExportAsyncJob(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/export-async-job',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getExportAsyncJob() {
    var payload = { jobId: ':jobId' };
    var getRequestOptions = httpWixFedinfraExportserviceV1ExportAsyncJob.getExportAsyncJob(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/v1/export-async-job/{jobId}',
        pathParams: { jobId: 'jobId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function cancelExportAsyncJob() {
    var payload = { jobId: ':jobId' };
    var getRequestOptions = httpWixFedinfraExportserviceV1ExportAsyncJob.cancelExportAsyncJob(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'PUT',
        path: '/v1/export-async-job/{jobId}/cancel',
        pathParams: { jobId: 'jobId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function generateExportAsyncJobDownloadUrl() {
    var payload = { jobId: ':jobId' };
    var getRequestOptions = httpWixFedinfraExportserviceV1ExportAsyncJob.generateExportAsyncJobDownloadUrl(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'PUT',
        path: '/v1/export-async-job/{jobId}/generate-download-url',
        pathParams: { jobId: 'jobId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
