// @wix/headless-stores/dist/react/Category.js
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import React from 'react';
import { AsChildSlot } from '@wix/headless-utils/react';
import * as CoreProductListFilters from './core/ProductListFilters.js';
import { DataComponentTags } from '../data-component-tags.js';
const CategoryContext = React.createContext(null);
function useCategoryContext() {
    const context = React.useContext(CategoryContext);
    if (!context) {
        throw new Error('useCategoryContext must be used within a Category.Root component');
    }
    return context;
}
var TestIds;
(function (TestIds) {
    TestIds["categoryRoot"] = "category-root";
    TestIds["categoryItem"] = "category-item";
    TestIds["categoryTrigger"] = "category-trigger";
    TestIds["categoryLabel"] = "category-label";
    TestIds["categoryId"] = "category-id";
    TestIds["categoryRaw"] = "category-raw";
    TestIds["categoryFilter"] = "category-filter";
})(TestIds || (TestIds = {}));
/**
 * Root container for a single category item.
 * This component sets up the necessary services for managing category state
 * and provides category context to child components.
 *
 * @order 1
 * @component
 * @example
 * ```tsx
 * import { Category } from '@wix/headless-stores/react';
 *
 * <Category.Root category={category}>
 *     <Category.Trigger />
 *     <Category.Label />
 *     <Category.ID />
 * </Category.Root>
 * ```
 */
export const Root = React.forwardRef((props, ref) => {
    const { category, children } = props;
    return (_jsx(CoreProductListFilters.CategoryFilter, { children: ({ selectedCategory, setSelectedCategory }) => {
            // Determine if this category is selected by comparing with selectedCategory
            const isSelected = selectedCategory?._id === category._id;
            const contextValue = {
                category,
                isSelected,
                selectedCategory,
                setSelectedCategory,
            };
            return (_jsx(CategoryContext.Provider, { value: contextValue, children: _jsx(AsChildSlot, { ref: ref, "data-component-tag": DataComponentTags.categoryRoot, "data-testid": TestIds.categoryRoot, children: children }) }));
        } }));
});
Root.displayName = 'Category.Root';
/**
 * Interactive element for selecting or triggering category actions.
 * Provides category data and selection state to custom render functions.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Category.Trigger className="px-4 py-2 rounded border hover:bg-surface-hover" />
 *
 * // Custom rendering with forwardRef
 * <Category.Trigger asChild>
 *   {React.forwardRef(({category, isSelected, onSelect, ...props}, ref) => (
 *     <button
 *       ref={ref}
 *       {...props}
 *       onClick={onSelect}
 *       className={`px-4 py-2 rounded transition-colors ${
 *         isSelected
 *           ? 'bg-brand-primary text-white'
 *           : 'border border-surface-subtle hover:bg-surface-hover'
 *       }`}
 *     >
 *       {category.name}
 *     </button>
 *   ))}
 * </Category.Trigger>
 * ```
 */
export const Trigger = React.forwardRef((props, ref) => {
    const { asChild, children, onSelect, className, ...otherProps } = props;
    const { category, isSelected, setSelectedCategory } = useCategoryContext();
    const handleSelect = () => {
        // Use CategoryFilter's setSelectedCategory function
        if (isSelected) {
            // Deselect by passing null
            setSelectedCategory(null);
        }
        else {
            // Select this category
            setSelectedCategory(category);
        }
        if (onSelect) {
            onSelect(category);
        }
    };
    return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, onClick: handleSelect, "data-testid": TestIds.categoryTrigger, "data-selected": isSelected ? 'true' : 'false', customElement: children, customElementProps: {
            category,
            isSelected,
            onSelect: handleSelect,
        }, content: category.name, ...otherProps, children: _jsx("button", { children: category.name }) }));
});
/**
 * Displays the category name or label.
 * Provides category name and full category data to custom render functions.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Category.Label className="text-lg font-medium text-content-primary" />
 *
 * // Custom rendering with forwardRef
 * <Category.Label asChild>
 *   {React.forwardRef(({name, category, ...props}, ref) => (
 *     <span ref={ref} {...props} className="text-lg font-medium text-content-primary">
 *       {name}
 *       {category.numberOfProducts > 0 && (
 *         <span className="text-sm text-content-muted ml-2">
 *           ({category.numberOfProducts})
 *         </span>
 *       )}
 *     </span>
 *   ))}
 * </Category.Label>
 * ```
 */
export const Label = React.forwardRef((props, ref) => {
    const { asChild, children, className } = props;
    const { category, isSelected } = useCategoryContext();
    return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.categoryLabel, "data-selected": isSelected ? 'true' : 'false', customElement: children, customElementProps: { name: category.name, category }, content: category.name, children: _jsx("span", { children: category.name }) }));
});
/**
 * Provides access to the category ID for advanced use cases.
 * Typically used for tracking, analytics, or hidden form fields.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage (hidden by default)
 * <Category.ID className="hidden" />
 *
 * // Custom rendering with forwardRef
 * <Category.ID asChild>
 *   {React.forwardRef(({id, category, ...props}, ref) => (
 *     <span
 *       ref={ref}
 *       {...props}
 *       data-category-id={id}
 *       className="sr-only"
 *     >
 *       Category ID: {id}
 *     </span>
 *   ))}
 * </Category.ID>
 * ```
 */
export const ID = React.forwardRef((props, ref) => {
    const { asChild, children, className } = props;
    const { category, isSelected } = useCategoryContext();
    const id = category._id || '';
    return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className || 'sr-only', "data-testid": TestIds.categoryId, "data-selected": isSelected ? 'true' : 'false', customElement: children, customElementProps: { id, category }, content: id, children: _jsx("span", { children: id }) }));
});
/**
 * Provides access to the full category data for advanced use cases.
 * Useful for custom implementations that need access to all category properties.
 *
 * @component
 * @example
 * ```tsx
 * // Custom rendering with forwardRef
 * <Category.Raw asChild>
 *   {React.forwardRef(({category, ...props}, ref) => (
 *     <div
 *       ref={ref}
 *       {...props}
 *       data-category-id={category._id}
 *       className="category-raw-data"
 *     >
 *       <pre>{JSON.stringify(category, null, 2)}</pre>
 *     </div>
 *   ))}
 * </Category.Raw>
 * ```
 */
export const Raw = React.forwardRef((props, ref) => {
    const { asChild, children, className } = props;
    const { category, isSelected } = useCategoryContext();
    return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className || 'sr-only', "data-testid": TestIds.categoryRaw, "data-selected": isSelected ? 'true' : 'false', customElement: children, customElementProps: { category, isSelected }, children: _jsx("span", { children: JSON.stringify(category) }) }));
});
/**
 * Category filter component that provides category selection functionality.
 * Provides selected category state and selection controls to custom render functions.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <CategoryFilter className="category-filter" />
 *
 * // With custom label
 * <CategoryFilter label="Current Category:" />
 *
 * // Custom rendering with forwardRef
 * <CategoryFilter asChild>
 *   {React.forwardRef(({selectedCategory, setSelectedCategory, ...props}, ref) => (
 *     <div ref={ref} {...props} className="custom-category-filter">
 *       {selectedCategory && (
 *         <span>Selected: {selectedCategory.name}</span>
 *       )}
 *       <button onClick={() => setSelectedCategory(null)}>
 *         Clear Selection
 *       </button>
 *     </div>
 *   ))}
 * </CategoryFilter>
 * ```
 */
export const CategoryFilter = React.forwardRef((props, ref) => {
    const { asChild, children, className } = props;
    const label = props.label || 'Selected:';
    return (_jsx(CoreProductListFilters.CategoryFilter, { children: ({ selectedCategory, setSelectedCategory }) => {
            return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.categoryFilter, "data-selected": selectedCategory ? 'true' : 'false', customElement: children, customElementProps: {
                    selectedCategory,
                    setSelectedCategory,
                }, children: _jsx("div", { children: selectedCategory && (_jsxs("span", { children: [label, " ", selectedCategory.name] })) }) }));
        } }));
});
CategoryFilter.displayName = 'CategoryFilter';
