// @wix/unidriver-common/dist/esm/__tests__/TestUniDriver/UniDriverClass.js
import { fireEvent, waitFor } from '@testing-library/dom';
import { PointerEventsCheckLevel, userEvent } from '@testing-library/user-event';
import { UniDriverBase } from '../../UniDriverBase/UniDriverBase';
import { enterText } from './enterText';
import { UniDriverClassList } from './UniDriverClassList';
import { enterValue } from './enterValue';
import { bindUniDriverApiMethods } from '../../utils/bindUniDriverApiMethods';
export class UniDriverClass extends UniDriverBase {
    constructor(ctx) {
        super(ctx);
        Object.defineProperty(this, "type", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: 'react'
        });
        Object.defineProperty(this, "api", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: {
                attr: async (base, attr) => base.getAttribute(attr),
                unwrap: async (base) => base,
                Class: UniDriverClass,
                click: async (base, options) => {
                    const clickParams = {
                        delay: null,
                        // We don't check pointer events for jsdom as we can't compute styles
                        pointerEventsCheck: PointerEventsCheckLevel.Never,
                    };
                    if (options?.type === 'double') {
                        return userEvent.dblClick(base, clickParams);
                    }
                    return userEvent.click(base, clickParams);
                },
                enterText: async (base, value, options) => {
                    await enterText(base, value, options);
                },
                enterValue: async (base, value, options) => {
                    await enterValue(base, value, options);
                },
                getElementsBySelector: async (base, selector) => [...base.querySelectorAll(selector)],
                getType: () => 'jsdom-react',
                hasClass: async (base, className) => base.classList.contains(className),
                ListClass: UniDriverClassList,
                mouse: {
                    hover: async (base) => userEvent.hover(base),
                    leave: async (base) => userEvent.unhover(base),
                    moveTo: async (base, to) => {
                        const { left, top } = (await to.unwrap()).getBoundingClientRect();
                        fireEvent.mouseMove(base, { clientX: left, clientY: top });
                    },
                    press: async (base) => {
                        fireEvent.mouseDown(base);
                    },
                    release: async (base) => {
                        fireEvent.mouseUp(base);
                    },
                },
                pressKey: async (base, keyMeta) => {
                    fireEvent.keyDown(base, keyMeta);
                    fireEvent.keyUp(base, keyMeta);
                },
                prop: async (base, prop) => base[prop],
                text: async (base) => base.textContent ?? '',
                value: async (base) => {
                    const value = base.value;
                    return typeof value === 'number' ? String(value) : (value ?? '');
                },
            }
        });
        Object.defineProperty(this, "get", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: async (selector, options) => {
                return waitFor(() => {
                    return super.get(selector, options);
                }, { timeout: this.getTimeBudget(options?.timeout) });
            }
        });
        Object.defineProperty(this, "getAll", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: async (selector, options) => {
                return waitFor(() => {
                    return super.getAll(selector, options);
                }, { timeout: this.getTimeBudget(options?.timeout) });
            }
        });
        bindUniDriverApiMethods(this);
    }
    driverSpecificMethod() {
        return 'test';
    }
}
