// @wix/patterns/dist/esm/components/CollectionPageNew/CollectionPage.js
import { Page } from '@wix/design-system';
import React, { useEffect, useState } from 'react';
import { useCollectionPageChild } from '../../hooks/useCollectionPageChild';
import { CollectionPageHeader } from '../CollectionPageHeaderNew';
import { PageWrapperBase } from '../PageWrapper';
import { CollectionPageContent } from '../CollectionPageContent';
import { observer } from 'mobx-react-lite';
import { CollectionPageContextProvider } from '../../providers';
import { usePageHeader } from '../../hooks/usePageHeaderNew';
import { st, classes } from './CollectionPage.st.css.js';
import { useCollectionPageState, useWixPatternsContainer, } from '@wix/bex-core/react';
import { useCollectionPageLoadReporting } from './useCollectionPageLoadReporting';
import { useIsMobile } from '../../hooks/useIsMobile';
function _CollectionPage({ children, scrollableContentRef, ...rest }) {
    const wrappedChildren = useCollectionPageChild(children);
    const childrenArr = React.Children.toArray(children);
    const { title: pageTitle, hideTotal: hideHeaderTotal, header, } = usePageHeader(childrenArr);
    const container = useWixPatternsContainer();
    const [reportBI] = useState(() => container.createBILogger({
        componentType: 'Collection page',
        pageType: 'Collection',
        route: window.location.pathname,
    }));
    const collectionPageState = useCollectionPageState({
        reportBI,
        pageTitle,
        hasHeader: header != null,
        hideHeaderTotal,
    });
    useEffect(() => collectionPageState.init(), []);
    useCollectionPageLoadReporting({
        startTime: collectionPageState.startTime,
        withinRouter: !!collectionPageState.withinRouter,
    });
    const isMobile = useIsMobile();
    return (React.createElement(CollectionPageContextProvider, { value: collectionPageState },
        React.createElement(PageWrapperBase, { pageState: collectionPageState.pageState },
            React.createElement(Page, { className: st(classes.root, { mobile: isMobile }), scrollableContentRef: (el) => {
                    collectionPageState.pageState.scrollableContentRef.current = el;
                    scrollableContentRef?.(el);
                }, ...rest, minWidth: isMobile ? 0 : undefined }, wrappedChildren))));
}
_CollectionPage.Header = CollectionPageHeader;
_CollectionPage.Content = CollectionPageContent;
_CollectionPage.displayName = 'CollectionPage';
export const CollectionPage = observer(_CollectionPage);
