// @wix/design-system/dist/cjs/AngleInput/AngleInput.st.css.js

require("../Foundation/stylable/colors.st.css.js");

require("../common/styles/visuallyHidden.st.css.js");
require("../common/styles/visuallyHidden.st.css.js");
const { classesRuntime, statesRuntime } = require("./../stylable-cjs-runtime.js");



var _namespace_ = "wds_1_226_0_AngleInput";
var _style_ = classesRuntime.bind(null, _namespace_);

module.exports.cssStates = statesRuntime.bind(null, _namespace_);
module.exports.style = _style_;
module.exports.st = _style_;

module.exports.namespace = _namespace_;
module.exports.classes = {"root":"wds_1_226_0_AngleInput__root","visuallyHidden":"wds_1_226_0_visuallyHidden__visuallyHidden","track":"wds_1_226_0_AngleInput__track","arm":"wds_1_226_0_AngleInput__arm","knob":"wds_1_226_0_AngleInput__knob","knobWrapper":"wds_1_226_0_AngleInput__knobWrapper","visuallyHiddenInput":"wds_1_226_0_AngleInput__visuallyHiddenInput wds_1_226_0_visuallyHidden__visuallyHidden"};
module.exports.keyframes = {}; 
module.exports.layers = {};
module.exports.containers = {};
module.exports.stVars = {"wrapperWidth":"58px","trackDiameter":"48px","armWidth":"2px","knobWrapperWidth":"18px","knobRegularWidth":"4px","knobLargeWidth":"12px","knobRegularBorderWidth":"4px","knobLargeBorderWidth":"3px","knobWrapperCenter":"calc(18px / 2)","trackRadius":"calc(48px / 2)"}; 
module.exports.vars = {"wds-color-fill-standard-primary":"--wds-color-fill-standard-primary","wds-color-border-standard-secondary":"--wds-color-border-standard-secondary","wds-color-border-standard-secondary-disabled":"--wds-color-border-standard-secondary-disabled","wds-color-fill-standard-primary-disabled":"--wds-color-fill-standard-primary-disabled","wds-angle-input-value":"--wds-angle-input-value"};
