// @wix/design-system/dist/cjs/AngleInput/AngleInput.uni.driver.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.angleInputDriverFactory = void 0;
var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _unidriver = require("../utils/test-utils/unidriver");
var _AngleInput = require("./AngleInput.constants");
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2.default)(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var angleInputDriverFactory = base => {
  var getElementByHook = hook => base.$("[data-hook=".concat(hook, "]"));
  var input = getElementByHook(_AngleInput.dataHooks.input);
  return _objectSpread(_objectSpread({}, (0, _unidriver.baseUniDriverFactory)(base)), {}, {
    /**
     * Checks if the input element is focused.
     */
    isFocused: function () {
      var _isFocused = (0, _asyncToGenerator2.default)(function* () {
        return yield (0, _unidriver.isElementFocused)(input);
      });
      function isFocused() {
        return _isFocused.apply(this, arguments);
      }
      return isFocused;
    }(),
    /**
     * Checks if the input element is disabled.
     */
    isDisabled: function () {
      var _isDisabled = (0, _asyncToGenerator2.default)(function* () {
        var disabledAttr = yield input.attr('disabled');
        return disabledAttr !== null;
      });
      function isDisabled() {
        return _isDisabled.apply(this, arguments);
      }
      return isDisabled;
    }(),
    /**
     * Focuses the input element.
     */

    focus: function () {
      var _focus = (0, _asyncToGenerator2.default)(function* () {
        return (yield input.getNative()).focus();
      });
      function focus() {
        return _focus.apply(this, arguments);
      }
      return focus;
    }(),
    /**
     * Blurs the input element.
     */
    blur: function () {
      var _blur = (0, _asyncToGenerator2.default)(function* () {
        return (yield input.getNative()).blur();
      });
      function blur() {
        return _blur.apply(this, arguments);
      }
      return blur;
    }(),
    /**
     * Retrieves the value of the input element.
     */
    getValue: function () {
      var _getValue = (0, _asyncToGenerator2.default)(function* () {
        return yield input.value();
      });
      function getValue() {
        return _getValue.apply(this, arguments);
      }
      return getValue;
    }(),
    /**
     * Sets the value of the input element.
     */
    setValue: function () {
      var _setValue = (0, _asyncToGenerator2.default)(function* (value) {
        return yield input.enterValue(value);
      });
      function setValue(_x) {
        return _setValue.apply(this, arguments);
      }
      return setValue;
    }(),
    /**
     * Increases the angle by pressing the 'ArrowUp' key a specified number of times.
     */
    incrementByStepCount: function () {
      var _incrementByStepCount = (0, _asyncToGenerator2.default)(function* (times) {
        (yield input.getNative()).focus();
        for (var i = 0; i < times; i++) {
          yield input.pressKey('ArrowUp');
        }
      });
      function incrementByStepCount(_x2) {
        return _incrementByStepCount.apply(this, arguments);
      }
      return incrementByStepCount;
    }(),
    /**
     * Decreases the angle by pressing the 'ArrowDown' key a specified number of times.
     */
    decrementByStepCount: function () {
      var _decrementByStepCount = (0, _asyncToGenerator2.default)(function* (times) {
        (yield input.getNative()).focus();
        for (var i = 0; i < times; i++) {
          yield input.pressKey('ArrowDown');
        }
      });
      function decrementByStepCount(_x3) {
        return _decrementByStepCount.apply(this, arguments);
      }
      return decrementByStepCount;
    }()
  });
};
exports.angleInputDriverFactory = angleInputDriverFactory;
