// @wix/auto_sdk_payments_saved-payment-methods/build/es/meta.mjs
// src/payments-saved-payment-methods-v1-saved-payment-method-saved-payment-methods.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixPaymentsSavedPaymentMethodsV1SavedPaymentMethodServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/payments/saved-payment-methods",
        destPath: ""
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/payments/saved-payment-methods",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/payments/saved-payment-methods",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_saved-payment-methods";
function upsertSavedPaymentMethod(payload) {
  function __upsertSavedPaymentMethod({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "savedPaymentMethod.createdDate" },
          { path: "savedPaymentMethod.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.payments.saved_payment_methods.v1.saved_payment_method",
      method: "POST",
      methodFqn: "wix.payments.saved_payment_methods.v1.SavedPaymentMethodService.UpsertSavedPaymentMethod",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsSavedPaymentMethodsV1SavedPaymentMethodServiceUrl({
        protoPath: "/v1/saved-payment-methods/upsert",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "savedPaymentMethod.createdDate" },
            { path: "savedPaymentMethod.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __upsertSavedPaymentMethod;
}
function getSavedPaymentMethod(payload) {
  function __getSavedPaymentMethod({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.saved_payment_methods.v1.saved_payment_method",
      method: "GET",
      methodFqn: "wix.payments.saved_payment_methods.v1.SavedPaymentMethodService.GetSavedPaymentMethod",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsSavedPaymentMethodsV1SavedPaymentMethodServiceUrl({
        protoPath: "/v1/saved-payment-methods/{savedPaymentMethodId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "savedPaymentMethod.createdDate" },
            { path: "savedPaymentMethod.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getSavedPaymentMethod;
}
function deleteSavedPaymentMethod(payload) {
  function __deleteSavedPaymentMethod({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.saved_payment_methods.v1.saved_payment_method",
      method: "DELETE",
      methodFqn: "wix.payments.saved_payment_methods.v1.SavedPaymentMethodService.DeleteSavedPaymentMethod",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsSavedPaymentMethodsV1SavedPaymentMethodServiceUrl({
        protoPath: "/v1/saved-payment-methods/{savedPaymentMethodId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteSavedPaymentMethod;
}
function listSavedPaymentMethods(payload) {
  function __listSavedPaymentMethods({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.saved_payment_methods.v1.saved_payment_method",
      method: "GET",
      methodFqn: "wix.payments.saved_payment_methods.v1.SavedPaymentMethodService.ListSavedPaymentMethods",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsSavedPaymentMethodsV1SavedPaymentMethodServiceUrl({
        protoPath: "/v1/saved-payment-methods",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "savedPaymentMethods.createdDate" },
            { path: "savedPaymentMethods.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listSavedPaymentMethods;
}
function updateSavedPaymentMethod(payload) {
  function __updateSavedPaymentMethod({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "savedPaymentMethod.createdDate" },
          { path: "savedPaymentMethod.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.payments.saved_payment_methods.v1.saved_payment_method",
      method: "PATCH",
      methodFqn: "wix.payments.saved_payment_methods.v1.SavedPaymentMethodService.UpdateSavedPaymentMethod",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsSavedPaymentMethodsV1SavedPaymentMethodServiceUrl({
        protoPath: "/v1/saved-payment-methods/{savedPaymentMethod.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "savedPaymentMethod.createdDate" },
            { path: "savedPaymentMethod.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateSavedPaymentMethod;
}
function markSavedPaymentMethodPrimary(payload) {
  function __markSavedPaymentMethodPrimary({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.saved_payment_methods.v1.saved_payment_method",
      method: "GET",
      methodFqn: "wix.payments.saved_payment_methods.v1.SavedPaymentMethodService.MarkSavedPaymentMethodPrimary",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsSavedPaymentMethodsV1SavedPaymentMethodServiceUrl({
        protoPath: "/v1/saved-payment-methods/{savedPaymentMethodId}/mark-primary",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __markSavedPaymentMethodPrimary;
}

// src/payments-saved-payment-methods-v1-saved-payment-method-saved-payment-methods.meta.ts
function upsertSavedPaymentMethod2() {
  const payload = {};
  const getRequestOptions = upsertSavedPaymentMethod(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/saved-payment-methods/upsert",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getSavedPaymentMethod2() {
  const payload = { savedPaymentMethodId: ":savedPaymentMethodId" };
  const getRequestOptions = getSavedPaymentMethod(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/saved-payment-methods/{savedPaymentMethodId}",
    pathParams: { savedPaymentMethodId: "savedPaymentMethodId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteSavedPaymentMethod2() {
  const payload = { savedPaymentMethodId: ":savedPaymentMethodId" };
  const getRequestOptions = deleteSavedPaymentMethod(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/saved-payment-methods/{savedPaymentMethodId}",
    pathParams: { savedPaymentMethodId: "savedPaymentMethodId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listSavedPaymentMethods2() {
  const payload = {};
  const getRequestOptions = listSavedPaymentMethods(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/saved-payment-methods",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateSavedPaymentMethod2() {
  const payload = {
    savedPaymentMethod: { id: ":savedPaymentMethodId" }
  };
  const getRequestOptions = updateSavedPaymentMethod(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/saved-payment-methods/{savedPaymentMethod.id}",
    pathParams: { savedPaymentMethodId: "savedPaymentMethodId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function markSavedPaymentMethodPrimary2() {
  const payload = { savedPaymentMethodId: ":savedPaymentMethodId" };
  const getRequestOptions = markSavedPaymentMethodPrimary(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/saved-payment-methods/{savedPaymentMethodId}/mark-primary",
    pathParams: { savedPaymentMethodId: "savedPaymentMethodId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  deleteSavedPaymentMethod2 as deleteSavedPaymentMethod,
  getSavedPaymentMethod2 as getSavedPaymentMethod,
  listSavedPaymentMethods2 as listSavedPaymentMethods,
  markSavedPaymentMethodPrimary2 as markSavedPaymentMethodPrimary,
  updateSavedPaymentMethod2 as updateSavedPaymentMethod,
  upsertSavedPaymentMethod2 as upsertSavedPaymentMethod
};
