// @wix/bex-core/dist/esm/state/ArrayFilterState/ArrayFilterState.js
import { EventEmitter } from 'events';
import { action, computed, makeObservable, observable } from 'mobx';
import { refreshFilter } from '../Filter/refreshFilter';
export class ArrayFilterState {
    constructor(params) {
        this.events = new EventEmitter();
        this._toArray = (value) => {
            return value;
        };
        this.name = params.name ?? '';
        this.persistent = params.persistent;
        this.defaultValue = params.defaultValue ?? [];
        this.parse = params.parse;
        this.simplify = params.simplify ?? ((value) => value);
        this.itemKey = params.itemKey;
        this.itemName = params.itemName;
        this._value = params.initialValue ?? this.defaultValue;
        this.matches =
            params.matches ??
                ((item, value) => {
                    const { itemKey } = this;
                    const key = typeof item === 'string' ? item : itemKey(item);
                    return (value == null ||
                        !value.length ||
                        value.some((v) => itemKey(v) === key));
                });
        this.equals = params.equals;
        this.isCustomField = params.isCustomField;
        makeObservable(this, {
            _value: observable.ref,
            defaultValue: observable.ref,
            persistent: observable.ref,
            isEmpty: computed,
            size: computed,
            value: computed,
            reset: action,
            setValue: action,
            changeValue: action,
            scheduleRefresh: action,
            clone: action,
        });
    }
    get value() {
        return this._value;
    }
    get toArray() {
        return this._value;
    }
    get isEmpty() {
        return !this.size;
    }
    get size() {
        return this._value.length;
    }
    get toQueryString() {
        return JSON.stringify(this._value.map(this.simplify));
    }
    applyFromQueryString(str) {
        try {
            const maybeArr = JSON.parse(str);
            if (Array.isArray(maybeArr)) {
                this.setValue(maybeArr.map(this.parse).filter((value) => value != null));
            }
        }
        catch (_) { }
    }
    encode(value) {
        return value.map(this.simplify);
    }
    decode(raw) {
        if (Array.isArray(raw)) {
            return raw.map(this.parse).filter((value) => value != null);
        }
        return [];
    }
    setValue(value, { emitEvents, ...options } = {}) {
        this._value = value;
        if (emitEvents) {
            for (const event of emitEvents) {
                this.events.emit(event, options);
            }
        }
    }
    changeValue(value, { emitEvents = [], ...options } = {}) {
        this.setValue(value, { ...options, emitEvents: ['change', ...emitEvents] });
    }
    refresh(value, options) {
        refreshFilter(this, value, options);
    }
    scheduleRefresh(value) {
        this.changeValue(value, { emitEvents: ['scheduleRefresh'] });
    }
    reset(options = {}) {
        this.setValue(this.defaultValue, options);
    }
    hasDiff(items) {
        const { value } = this;
        return (value.length !== items.length ||
            value.some((item) => items.every((i) => this.itemKey(i) !== this.itemKey(item))));
    }
    remove(items, options = {}) {
        const { itemKey } = this;
        const keysToRemove = new Set(items.map(itemKey));
        this.setValue(this._value.filter((item) => !keysToRemove.has(itemKey(item))), options);
    }
    clone(params) {
        const { itemName, itemKey, simplify, parse, defaultValue, value: initialValue, matches, persistent, name, } = this;
        return new ArrayFilterState({
            itemName,
            itemKey,
            simplify,
            parse,
            defaultValue,
            initialValue,
            matches,
            persistent,
            name,
            ...params,
        });
    }
}
export const stringsArrayFilter = (params) => new ArrayFilterState({
    itemKey: (item) => item,
    itemName: (item) => item,
    parse: (value) => {
        if (value == null || typeof value !== 'string') {
            return null;
        }
        return value;
    },
    simplify: (value) => value,
    ...params,
});
export const arrayFilter = (params) => new ArrayFilterState(params);
export const idNameArrayFilter = (params) => new ArrayFilterState({
    itemKey: (item) => item.id,
    itemName: (item) => item.name,
    parse: (value) => {
        if (value == null ||
            typeof value === 'string' ||
            typeof value.id !== 'string' ||
            typeof value.name !== 'string') {
            return null;
        }
        return value;
    },
    simplify: (value) => ({
        id: value.id,
        name: value.name,
    }),
    ...params,
});
export const idArrayFilter = (params) => new ArrayFilterState({
    itemKey: (item) => item.id,
    itemName: () => '',
    parse: (value) => {
        if (value == null ||
            typeof value === 'string' ||
            typeof value.id !== 'string') {
            return null;
        }
        return value;
    },
    simplify: (value) => ({
        id: value.id,
    }),
    ...params,
});
export const idNameArrayFilterNum = (params) => new ArrayFilterState({
    itemKey: (item) => String(item.id),
    itemName: (item) => item.name,
    parse: (value) => {
        if (value == null ||
            typeof value === 'string' ||
            typeof value.id !== 'number' ||
            typeof value.name !== 'string') {
            return null;
        }
        return value;
    },
    simplify: (value) => ({
        id: value.id,
        name: value.name,
    }),
    ...params,
});
