// @wix/design-system/dist/cjs/Input/components/Affix/Affix.st.css.js

require("../../../Foundation/stylable/typography.st.css.js");
require("../../../Foundation/stylable/colors.st.css.js");

const { classesRuntime, statesRuntime } = require("./../../../stylable-cjs-runtime.js");



var _namespace_ = "wds_1_226_0_Affix";
var _style_ = classesRuntime.bind(null, _namespace_);

module.exports.cssStates = statesRuntime.bind(null, _namespace_);
module.exports.style = _style_;
module.exports.st = _style_;

module.exports.namespace = _namespace_;
module.exports.classes = {"root":"wds_1_226_0_Affix__root","text-tiny-thin":"wds_1_226_0_StylableTypography__text-tiny-thin","text-small-normal":"wds_1_226_0_StylableTypography__text-small-normal","text-small-thin":"wds_1_226_0_StylableTypography__text-small-thin","text-medium-normal":"wds_1_226_0_StylableTypography__text-medium-normal","text-medium-thin":"wds_1_226_0_StylableTypography__text-medium-thin"};
module.exports.keyframes = {}; 
module.exports.layers = {};
module.exports.containers = {};
module.exports.stVars = {}; 
module.exports.vars = {"wds-input-value-font-line-height-tiny":"--wds-input-value-font-line-height-tiny","wds-input-value-font-line-height-small":"--wds-input-value-font-line-height-small","wds-input-value-font-line-height-medium":"--wds-input-value-font-line-height-medium","wds-input-value-font-line-height-large":"--wds-input-value-font-line-height-large","wds-input-value-font-size-tiny":"--wds-input-value-font-size-tiny","wds-input-value-font-size-small":"--wds-input-value-font-size-small","wds-input-value-font-size-medium":"--wds-input-value-font-size-medium","wds-input-value-font-size-large":"--wds-input-value-font-size-large","wds-color-text-standard-secondary-light":"--wds-color-text-standard-secondary-light","wds-color-text-placeholder":"--wds-color-text-placeholder","wds-color-text-disabled":"--wds-color-text-disabled","wds-space-50":"--wds-space-50","wds-space-100":"--wds-space-100"};
