// @wix/patterns/dist/esm/components/CollectionItemActions/CollectionItemActionsHelper.js
import { action } from 'mobx';
import { createElement } from 'react';
const calcCtaCount = (actionCellProps) => {
    const { primaryAction, secondaryActions } = actionCellProps;
    const primaryActionLength = primaryAction ? 1 : 0;
    return (secondaryActions?.length || 0) + primaryActionLength;
};
export const createSecondaryActionCellPropWithBi = ({ toolbarBIReporter, actionCellProps, keyedItem, extensionMenuItems, }) => {
    const { primaryAction, secondaryActions = [] } = actionCellProps;
    const primaryActionLength = primaryAction ? 1 : 0;
    const moreSecondaryItems = extensionMenuItems.menuItems.map((menuItem) => {
        return {
            biName: menuItem.biName,
            dataHook: menuItem.dataHook,
            onClick: () => {
                menuItem.onClick({ selectedId: keyedItem.id });
            },
            text: menuItem.text,
            icon: menuItem.prefixIcon || createElement('div'),
            componentId: menuItem.componentId,
            appId: menuItem.appId,
        };
    }) || [];
    const allActions = [
        ...secondaryActions,
        ...moreSecondaryItems,
    ];
    return (allActions.map((_action, index) => {
        const biIndex = index + 1;
        if (_action.divider) {
            return _action;
        }
        const onClick = action(() => {
            toolbarBIReporter.ctaClicked({
                location: 'actionCell',
                ctaIndex: primaryActionLength + biIndex,
                ctaName: _action?.biName || _action?.text || 'secondaryAction',
                moreActionsIndex: biIndex,
                numCtas: calcCtaCount(actionCellProps),
                itemIndex: keyedItem.index,
                itemId: keyedItem.id,
                componentId: _action.componentId,
                appId: _action.appId,
            });
            _action.onClick?.();
        });
        const { biName: _biName, ...wsrSecondaryAction } = _action;
        return { ...wsrSecondaryAction, onClick };
    }) || []);
};
export const createPrimaryActionCellPropsWithBI = (toolbarBIReporter, actionCellProps, keyedItem) => {
    const { primaryAction } = actionCellProps;
    if (primaryAction) {
        const onClick = action(() => {
            toolbarBIReporter.ctaClicked({
                location: 'actionCell',
                ctaIndex: 1,
                ctaName: primaryAction?.biName || primaryAction?.text || 'primaryAction',
                moreActionsIndex: 0,
                numCtas: calcCtaCount(actionCellProps),
                itemIndex: keyedItem.index,
                itemId: keyedItem.id,
            });
            primaryAction?.onClick();
        });
        const { biName, ...wsrPrimaryAction } = primaryAction;
        return { ...wsrPrimaryAction, onClick };
    }
    return primaryAction;
};
export const reportOnPopoverMenuShow = (toolbarBIReporter, actionCellProps, keyedItem) => {
    toolbarBIReporter.ctaClicked({
        location: 'actionCell',
        ctaIndex: 0,
        ctaName: 'More Actions Menu',
        numCtas: calcCtaCount(actionCellProps),
        itemId: keyedItem.id,
        itemIndex: keyedItem.index,
    });
};
