// @wix/auto_sdk_payments_site-payment-method-types/build/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  Format: () => Format,
  RequestedFields: () => RequestedFields,
  getSitePaymentMethodType: () => getSitePaymentMethodType4,
  listSitePaymentMethodTypes: () => listSitePaymentMethodTypes4
});
module.exports = __toCommonJS(index_exports);

// src/payments-site-payment-method-types-v1-site-payment-method-type-site-payment-method-types.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/payments-site-payment-method-types-v1-site-payment-method-type-site-payment-method-types.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixPaymentsSitePaymentMethodTypesV1SitePaymentMethodTypeServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/payment-method-types",
        destPath: ""
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      },
      {
        srcPath: "/_api/payments/payment-method-types",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/payments/site-payment-method-types",
        destPath: ""
      },
      {
        srcPath: "/_api/payments/payment-method-types",
        destPath: ""
      }
    ],
    "cashier._base_domain_": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "cashier.wixapps.net": [
      {
        srcPath: "/_api/payments/payment-method-types",
        destPath: ""
      },
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "cashier-services._base_domain_": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "events._base_domain_": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "events.wixapps.net": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "invoices._base_domain_": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "premium._base_domain_": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "pricing-plans.wixapps.net": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "vod-server._base_domain_": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "www.wixhotels.com": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/payments/payment-method-types",
        destPath: ""
      }
    ],
    "tpa.wixapps.net": [
      {
        srcPath: "/_api/payments/payment-method-types",
        destPath: ""
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/payments/payment-method-types",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/payments/payment-method-types",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/payments/payment-method-types",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/payments/payment-method-types",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/payments/payment-method-types",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/payments/site-payment-method-types/v1/site-payment-method-types",
        destPath: "/v1/site-payment-method-types"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/payments/site-payment-method-types/v1/site-payment-method-types",
        destPath: "/v1/site-payment-method-types"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_site-payment-method-types";
function getSitePaymentMethodType(payload) {
  function __getSitePaymentMethodType({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.site_payment_method_types.v1.site_payment_method_type",
      method: "GET",
      methodFqn: "wix.payments.site_payment_method_types.v1.SitePaymentMethodTypeService.GetSitePaymentMethodType",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsSitePaymentMethodTypesV1SitePaymentMethodTypeServiceUrl(
        {
          protoPath: "/v1/site-payment-method-types/{sitePaymentMethodTypeId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getSitePaymentMethodType;
}
function listSitePaymentMethodTypes(payload) {
  function __listSitePaymentMethodTypes({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.site_payment_method_types.v1.site_payment_method_type",
      method: "GET",
      methodFqn: "wix.payments.site_payment_method_types.v1.SitePaymentMethodTypeService.ListSitePaymentMethodTypes",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsSitePaymentMethodTypesV1SitePaymentMethodTypeServiceUrl(
        { protoPath: "/v1/site-payment-method-types", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __listSitePaymentMethodTypes;
}

// src/payments-site-payment-method-types-v1-site-payment-method-type-site-payment-method-types.universal.ts
var import_image = require("@wix/sdk-runtime/transformations/image");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var Format = /* @__PURE__ */ ((Format2) => {
  Format2["UNKNOWN_FORMAT"] = "UNKNOWN_FORMAT";
  Format2["SVG"] = "SVG";
  Format2["PNG"] = "PNG";
  return Format2;
})(Format || {});
var RequestedFields = /* @__PURE__ */ ((RequestedFields2) => {
  RequestedFields2["UNKNOWN_REQUESTED_FIELD"] = "UNKNOWN_REQUESTED_FIELD";
  RequestedFields2["ALLOWED_CURRENCIES"] = "ALLOWED_CURRENCIES";
  return RequestedFields2;
})(RequestedFields || {});
async function getSitePaymentMethodType2(sitePaymentMethodTypeId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    sitePaymentMethodTypeId,
    languageCode: options?.languageCode,
    countryCode: options?.countryCode,
    currencyCode: options?.currencyCode,
    requestedFields: options?.requestedFields
  });
  const reqOpts = getSitePaymentMethodType(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths.transformPaths)(result.data, [
        {
          transformFn: import_image.transformRESTImageToSDKImage,
          paths: [
            { path: "sitePaymentMethodType.merchantFlow.icons.image" },
            { path: "sitePaymentMethodType.buyerFlow.icons.image" }
          ]
        }
      ])
    )?.sitePaymentMethodType;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          sitePaymentMethodTypeId: "$[0]",
          languageCode: "$[1].languageCode",
          countryCode: "$[1].countryCode",
          currencyCode: "$[1].currencyCode",
          requestedFields: "$[1].requestedFields"
        },
        singleArgumentUnchanged: false
      },
      ["sitePaymentMethodTypeId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listSitePaymentMethodTypes2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    ids: options?.ids,
    languageCode: options?.languageCode,
    countryCode: options?.countryCode,
    currencyCode: options?.currencyCode,
    requestedFields: options?.requestedFields
  });
  const reqOpts = listSitePaymentMethodTypes(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths.transformPaths)(result.data, [
        {
          transformFn: import_image.transformRESTImageToSDKImage,
          paths: [
            { path: "sitePaymentMethodTypes.merchantFlow.icons.image" },
            { path: "sitePaymentMethodTypes.buyerFlow.icons.image" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          ids: "$[0].ids",
          languageCode: "$[0].languageCode",
          countryCode: "$[0].countryCode",
          currencyCode: "$[0].currencyCode",
          requestedFields: "$[0].requestedFields"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/payments-site-payment-method-types-v1-site-payment-method-type-site-payment-method-types.public.ts
function getSitePaymentMethodType3(httpClient) {
  return (sitePaymentMethodTypeId, options) => getSitePaymentMethodType2(
    sitePaymentMethodTypeId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function listSitePaymentMethodTypes3(httpClient) {
  return (options) => listSitePaymentMethodTypes2(
    options,
    // @ts-ignore
    { httpClient }
  );
}

// src/payments-site-payment-method-types-v1-site-payment-method-type-site-payment-method-types.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var getSitePaymentMethodType4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getSitePaymentMethodType3);
var listSitePaymentMethodTypes4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(listSitePaymentMethodTypes3);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Format,
  RequestedFields,
  getSitePaymentMethodType,
  listSitePaymentMethodTypes
});
