// @wix/auto_sdk_payments_site-payment-method-types/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  getSitePaymentMethodType: () => getSitePaymentMethodType2,
  listSitePaymentMethodTypes: () => listSitePaymentMethodTypes2
});
module.exports = __toCommonJS(meta_exports);

// src/payments-site-payment-method-types-v1-site-payment-method-type-site-payment-method-types.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixPaymentsSitePaymentMethodTypesV1SitePaymentMethodTypeServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/payment-method-types",
        destPath: ""
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      },
      {
        srcPath: "/_api/payments/payment-method-types",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/payments/site-payment-method-types",
        destPath: ""
      },
      {
        srcPath: "/_api/payments/payment-method-types",
        destPath: ""
      }
    ],
    "cashier._base_domain_": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "cashier.wixapps.net": [
      {
        srcPath: "/_api/payments/payment-method-types",
        destPath: ""
      },
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "cashier-services._base_domain_": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "events._base_domain_": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "events.wixapps.net": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "invoices._base_domain_": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "premium._base_domain_": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "pricing-plans.wixapps.net": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "vod-server._base_domain_": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "www.wixhotels.com": [
      {
        srcPath: "/_api/payment-method-types",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/payments/payment-method-types",
        destPath: ""
      }
    ],
    "tpa.wixapps.net": [
      {
        srcPath: "/_api/payments/payment-method-types",
        destPath: ""
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/payments/payment-method-types",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/payments/payment-method-types",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/payments/payment-method-types",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/payments/payment-method-types",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/payments/payment-method-types",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/payments/site-payment-method-types/v1/site-payment-method-types",
        destPath: "/v1/site-payment-method-types"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/payments/site-payment-method-types/v1/site-payment-method-types",
        destPath: "/v1/site-payment-method-types"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_site-payment-method-types";
function getSitePaymentMethodType(payload) {
  function __getSitePaymentMethodType({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.site_payment_method_types.v1.site_payment_method_type",
      method: "GET",
      methodFqn: "wix.payments.site_payment_method_types.v1.SitePaymentMethodTypeService.GetSitePaymentMethodType",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsSitePaymentMethodTypesV1SitePaymentMethodTypeServiceUrl(
        {
          protoPath: "/v1/site-payment-method-types/{sitePaymentMethodTypeId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getSitePaymentMethodType;
}
function listSitePaymentMethodTypes(payload) {
  function __listSitePaymentMethodTypes({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.site_payment_method_types.v1.site_payment_method_type",
      method: "GET",
      methodFqn: "wix.payments.site_payment_method_types.v1.SitePaymentMethodTypeService.ListSitePaymentMethodTypes",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsSitePaymentMethodTypesV1SitePaymentMethodTypeServiceUrl(
        { protoPath: "/v1/site-payment-method-types", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __listSitePaymentMethodTypes;
}

// src/payments-site-payment-method-types-v1-site-payment-method-type-site-payment-method-types.meta.ts
function getSitePaymentMethodType2() {
  const payload = {
    sitePaymentMethodTypeId: ":sitePaymentMethodTypeId"
  };
  const getRequestOptions = getSitePaymentMethodType(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/site-payment-method-types/{sitePaymentMethodTypeId}",
    pathParams: { sitePaymentMethodTypeId: "sitePaymentMethodTypeId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listSitePaymentMethodTypes2() {
  const payload = {};
  const getRequestOptions = listSitePaymentMethodTypes(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/site-payment-method-types",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getSitePaymentMethodType,
  listSitePaymentMethodTypes
});
