// @wix/headless-ecom/cjs/dist/react/core/CurrentCart.js
import { jsx as _jsx } from "react/jsx-runtime";
import { useService, WixServices } from '@wix/services-manager-react';
import { CurrentCartServiceDefinition, CurrentCartService, } from '../../services/current-cart-service.js';
import { createServicesMap } from '@wix/services-manager';
import { media } from '@wix/sdk';
/**
 * Root component that provides the CurrentCart service context to its children.
 * This component sets up the necessary services for managing current cart functionality.
 *
 * @order 1
 * @component
 * @example
 * ```tsx
 * import { CurrentCart } from '@wix/ecom/components';
 *
 * function CartProvider() {
 *   return (
 *     <CurrentCart.Root>
 *       <div>
 *         <CurrentCart.OpenTrigger>
 *           {({ totalItems, open }) => (
 *             <button onClick={open}>
 *               Cart ({totalItems})
 *             </button>
 *           )}
 *         </CurrentCart.OpenTrigger>
 *
 *         <CurrentCart.Content>
 *           {({ cart, isLoading }) => (
 *             <div>
 *               {isLoading ? 'Loading...' : 'Cart loaded'}
 *             </div>
 *           )}
 *         </CurrentCart.Content>
 *       </div>
 *     </CurrentCart.Root>
 *   );
 * }
 * ```
 */
export function Root(props) {
    return (_jsx(WixServices, { servicesMap: createServicesMap().addService(CurrentCartServiceDefinition, CurrentCartService, props.currentCartServiceConfig), children: props.children }));
}
/**
 * Component that renders content when the cart is empty.
 * Only displays its children when there are no items in cart, no loading state, and no errors.
 *
 * @component
 * @example
 * ```tsx
 * import { CurrentCart } from '@wix/ecom/components';
 *
 * function EmptyCartMessage() {
 *   return (
 *     <CurrentCart.EmptyState>
 *       {() => (
 *         <div className="empty-state">
 *           <h3>No items in cart</h3>
 *           <p>Items will appear here once they are added</p>
 *         </div>
 *       )}
 *     </CurrentCart.EmptyState>
 *   );
 * }
 * ```
 */
export function EmptyState(props) {
    const { isLoading, error, cart } = useService(CurrentCartServiceDefinition);
    const isLoadingValue = isLoading.get();
    const errorValue = error.get();
    const cartValue = cart.get();
    if (!isLoadingValue && !errorValue && cartValue?.lineItems?.length === 0) {
        return typeof props.children === 'function'
            ? props.children({})
            : props.children;
    }
    return null;
}
/**
 * Helper function to format currency properly
 */
function formatCurrency(amount, currencyCode) {
    try {
        return new Intl.NumberFormat('en-US', {
            style: 'currency',
            currency: currencyCode,
        }).format(amount);
    }
    catch (error) {
        // Fallback if currency code is invalid
        return `${amount.toFixed(2)} ${currencyCode}`;
    }
}
/**
 * Headless component for cart content/modal
 *
 * @example
 * // Complete cart implementation with essential headless features
 * <CurrentCart.Content>
 *   {({ cart, isLoading }) => (
 *     <div>
 *       <CurrentCart.OpenTrigger>
 *         {({ totalItems }) => <h1>Cart ({totalItems} items)</h1>}
 *       </CurrentCart.OpenTrigger>
 *
 *       {isLoading && <p>Loading cart...</p>}
 *
 *       <CurrentCart.LineItemsList>
 *         {({ items, totalItems }) => (
 *           <>
 *             {items.length === 0 ? (
 *               <p>Your cart is empty</p>
 *             ) : (
 *               <>
 *                 <CurrentCart.Clear>
 *                   {({ clear, totalItems, isLoading }) => (
 *                     totalItems > 0 && (
 *                       <button onClick={clear} disabled={isLoading}>
 *                         {isLoading ? 'Clearing...' : 'Clear Cart'}
 *                       </button>
 *                     )
 *                   )}
 *                 </CurrentCart.Clear>
 *
 *                 {items.map((item) => (
 *                   <CurrentCart.Item key={item._id} item={item}>
 *                     {({
 *                       title,
 *                       image,
 *                       price,
 *                       quantity,
 *                       selectedOptions,
 *                       increaseQuantity,
 *                       decreaseQuantity,
 *                       remove,
 *                       isLoading: itemLoading
 *                     }) => (
 *                       <div>
 *                         <h3>{title}</h3>
 *                         <p>{price}</p>
 *
 *                         {selectedOptions.map((option, index) => (
 *                           <span key={index}>
 *                             {option.name}: {typeof option.value === 'object' ? option.value.name : option.value}
 *                           </span>
 *                         ))}
 *
 *                         <button onClick={decreaseQuantity} disabled={itemLoading || quantity <= 1}>-</button>
 *                         <span>{quantity}</span>
 *                         <button onClick={increaseQuantity} disabled={itemLoading}>+</button>
 *
 *                         <button onClick={remove} disabled={itemLoading}>
 *                           {itemLoading ? 'Removing...' : 'Remove'}
 *                         </button>
 *                       </div>
 *                     )}
 *                   </CurrentCart.Item>
 *                 ))}
 *
 *                 <CurrentCart.Notes>
 *                   {({ notes, updateNotes }) => (
 *                     <textarea
 *                       value={notes}
 *                       onChange={e => updateNotes(e.target.value)}
 *                       placeholder="Special instructions for your order"
 *                     />
 *                   )}
 *                 </CurrentCart.Notes>
 *
 *                 <CurrentCart.Coupon>
 *                   {({ appliedCoupon, apply, remove, isLoading }) => (
 *                     <div>
 *                       {appliedCoupon ? (
 *                         <div>
 *                           <span>Coupon: {appliedCoupon}</span>
 *                           <button onClick={remove} disabled={isLoading}>
 *                             {isLoading ? 'Removing...' : 'Remove'}
 *                           </button>
 *                         </div>
 *                       ) : (
 *                         <form onSubmit={e => {
 *                           e.preventDefault();
 *                           const code = new FormData(e.currentTarget).get('couponCode');
 *                           if (code?.trim()) apply(code.trim());
 *                         }}>
 *                           <input name="couponCode" placeholder="Enter promo code" disabled={isLoading} />
 *                           <button type="submit" disabled={isLoading}>
 *                             {isLoading ? 'Applying...' : 'Apply'}
 *                           </button>
 *                         </form>
 *                       )}
 *                     </div>
 *                   )}
 *                 </CurrentCart.Coupon>
 *
 *                 <CurrentCart.Summary>
 *                   {({ subtotal, discount, appliedCoupon, shipping, tax, total, currency, totalItems, isTotalsLoading }) => (
 *                     <div>
 *                       <div>Subtotal ({totalItems} items): {isTotalsLoading ? 'Calculating...' : subtotal}</div>
 *                       {discount && <div>Discount: -{discount}</div>}
 *                       {appliedCoupon && <div>Applied Coupon: {appliedCoupon}</div>}
 *                       <div>Shipping: {isTotalsLoading ? 'Calculating...' : shipping}</div>
 *                       <div>Tax: {isTotalsLoading ? 'Calculating...' : tax}</div>
 *                       <div>Total: {isTotalsLoading ? 'Calculating...' : total}</div>
 *                       <div>Currency: {currency}</div>
 *                     </div>
 *                   )}
 *                 </CurrentCart.Summary>
 *
 *                 <CurrentCart.Checkout>
 *                   {({ proceedToCheckout, canCheckout, isLoading: checkoutLoading, error }) => (
 *                     <div>
 *                       {error && <p>Error: {error}</p>}
 *                       <button onClick={proceedToCheckout} disabled={!canCheckout || checkoutLoading}>
 *                         {checkoutLoading ? 'Processing...' : 'Proceed to Checkout'}
 *                       </button>
 *                     </div>
 *                   )}
 *                 </CurrentCart.Checkout>
 *               </>
 *             )}
 *           </>
 *         )}
 *       </CurrentCart.LineItemsList>
 *     </div>
 *   )}
 * </CurrentCart.Content>
 */
export const Content = (props) => {
    const service = useService(CurrentCartServiceDefinition);
    const cart = service.cart.get();
    const isLoading = service.isLoading.get();
    const error = service.error.get();
    return props.children({
        cart,
        isLoading,
        error,
    });
};
/**
 * Headless component for cart items collection
 *
 * @example
 * ```tsx
 * <CurrentCart.LineItemsList>
 *   {({ items, totalItems }) => (
 *     <div>
 *       <h1>Cart ({totalItems} items)</h1>
 *       <p>Items: {items.length}</p>
 *       <p>Has items: {items.length > 0 ? 'Yes' : 'No'}</p>
 *     </div>
 *   )}
 * </CurrentCart.LineItemsList>
 * ```
 */
export const LineItemsList = (props) => {
    const service = useService(CurrentCartServiceDefinition);
    const cart = service.cart.get();
    const items = cart?.lineItems || [];
    const totalItems = service.cartCount.get();
    return props.children({
        items,
        totalItems,
    });
};
/**
 * Headless component for individual cart item
 *
 * @example
 * ```tsx
 * <CurrentCart.Item item={item}>
 *   {({ quantity, title, image, price, selectedOptions, increaseQuantity, decreaseQuantity, remove, isLoading }) => (
 *     <div>
 *       <h3>{title}</h3>
 *       <p>{price}</p>
 *       <p>{quantity}</p>
 *       <p>{image}</p>
 *       <p>{selectedOptions}</p>
 *       <button onClick={increaseQuantity}>Increase</button>
 *       <button onClick={decreaseQuantity}>Decrease</button>
 *      <button onClick={remove}>Remove</button>
 *     </div>
 *   )}
 * </CurrentCart.Item>
 * ```
 */
export const Item = (props) => {
    const service = useService(CurrentCartServiceDefinition);
    const cart = service.cart.get();
    const item = props.item;
    const isLoading = service.isLoading.get();
    if (!item) {
        const currency = cart?.currency || 'USD';
        return props.children({
            quantity: 0,
            title: '',
            image: null,
            price: formatCurrency(0, currency),
            selectedOptions: [],
            onValueChange: async () => { },
            remove: async () => { },
            isLoading: false,
        });
    }
    // Fix image URL access - properly handle null/undefined image
    let image = null;
    if (item.image) {
        try {
            image = media.getImageUrl(item.image).url;
        }
        catch (error) {
            console.warn('Failed to get image URL:', error);
            image = null;
        }
    }
    // Extract variant information from description lines
    const selectedOptions = [];
    if (item.descriptionLines) {
        item.descriptionLines.forEach((line) => {
            if (line.name?.original) {
                const optionName = line.name.original;
                if (line.colorInfo) {
                    selectedOptions.push({
                        name: optionName,
                        value: {
                            name: line.colorInfo.original,
                            code: line.colorInfo.code,
                        },
                    });
                }
                else if (line.plainText) {
                    selectedOptions.push({
                        name: optionName,
                        value: line.plainText.original,
                    });
                }
            }
        });
    }
    // Calculate total price for this line item (unit price × quantity)
    const unitPrice = parseFloat(item.price?.amount || '0');
    const quantity = item.quantity || 0;
    const totalPrice = unitPrice * quantity;
    const currency = cart?.currency || 'USD';
    // Format price with proper currency
    const formattedPrice = formatCurrency(totalPrice, currency);
    const lineItemId = item._id || '';
    return props.children({
        quantity,
        title: item.productName?.original || '',
        image,
        price: formattedPrice,
        selectedOptions,
        onValueChange: (value) => service.updateLineItemQuantity(lineItemId, value),
        remove: () => service.removeLineItem(lineItemId),
        isLoading,
    });
};
/**
 * Headless component for cart summary/totals
 *
 * @example
 * ```tsx
 * <CurrentCart.Summary>
 *   {({ subtotal, discount, appliedCoupon, shipping, tax, total, currency, totalItems, isTotalsLoading }) => (
 *     <div>
 *       <h1>Cart Summary</h1>
 *       <p>Subtotal: {subtotal}</p>
 *       <p>Discount: {discount}</p>
 *       <p>Applied Coupon: {appliedCoupon}</p>
 *       <p>Shipping: {shipping}</p>
 *       <p>Tax: {tax}</p>
 *       <p>Total: {total}</p>
 *       <p>Currency: {currency}</p>
 *       <p>Item Count: {totalItems}</p>
 *       <p>Is Totals Loading: {isTotalsLoading ? 'Yes' : 'No'}</p>
 *     </div>
 *   )}
 * </CurrentCart.Summary>
 * ```
 */
export const Summary = (props) => {
    const service = useService(CurrentCartServiceDefinition);
    const cart = service.cart.get();
    const totalItems = service.cartCount.get();
    const cartTotals = service.cartTotals.get();
    const isTotalsLoading = service.isTotalsLoading.get();
    const currency = cart?.currency || cartTotals?.currency || 'USD';
    // Use SDK totals only
    const totals = cartTotals?.priceSummary || {};
    const subtotalAmount = totals.subtotal?.amount || 0;
    const subtotal = formatCurrency(subtotalAmount, currency);
    const shippingAmount = totals.shipping?.amount || 0;
    const shipping = formatCurrency(shippingAmount, currency);
    const taxAmount = totals.tax?.amount || 0;
    const tax = formatCurrency(taxAmount, currency);
    const totalAmount = totals.total?.amount || 0;
    const total = formatCurrency(totalAmount, currency);
    const appliedCoupon = cart?.appliedDiscounts?.find((discount) => discount.coupon?.code)
        ?.coupon?.code || null;
    const discountAmount = totals.discount?.amount || '0';
    const discount = discountAmount
        ? formatCurrency(parseFloat(discountAmount), currency)
        : null;
    const amountValues = {
        subtotal: subtotalAmount,
        discount: parseFloat(discountAmount),
        shipping: shippingAmount,
        tax: taxAmount,
        total: totalAmount,
    };
    return props.children({
        subtotal,
        discount,
        appliedCoupon,
        shipping,
        tax,
        total,
        currency,
        totalItems,
        isTotalsLoading,
        amountValues,
    });
};
/**
 * Headless component for clearing the cart
 *
 * @example
 * ```tsx
 * <CurrentCart.Clear>
 *   {({ clear, totalItems, isLoading }) => (
 *     <div>
 *       <h1>Cart Clear</h1>
 *       <p>Total items: {totalItems}</p>
 *       <p>Is loading: {isLoading ? 'Yes' : 'No'}</p>
 *       <button onClick={clear} disabled={isLoading}>Clear Cart</button>
 *     </div>
 *   )}
 * </CurrentCart.Clear>
 * ```
 */
export const Clear = (props) => {
    const service = useService(CurrentCartServiceDefinition);
    const totalItems = service.cartCount.get();
    const isLoading = service.isLoading.get();
    return props.children({
        clear: service.clearCart,
        totalItems,
        isLoading,
    });
};
/**
 * Headless component for checkout action
 *
 * @example
 * ```tsx
 * <CurrentCart.Checkout>
 *   {({ proceedToCheckout, canCheckout, isLoading, error }) => (
 *     <div>
 *       <h1>Checkout</h1>
 *       <p>Can checkout: {canCheckout ? 'Yes' : 'No'}</p>
 *       <p>Is loading: {isLoading ? 'Yes' : 'No'}</p>
 *       <p>Error: {error}</p>
 *       <button onClick={proceedToCheckout} disabled={!canCheckout || isLoading}>Proceed to Checkout</button>
 *     </div>
 *   )}
 * </CurrentCart.Checkout>
 * ```
 */
export const Checkout = (props) => {
    const service = useService(CurrentCartServiceDefinition);
    const totalItems = service.cartCount.get();
    const isLoading = service.isLoading.get();
    const error = service.error.get();
    const addToCart = service.addToCart;
    return props.children({
        proceedToCheckout: service.proceedToCheckout,
        canCheckout: totalItems > 0,
        isLoading,
        error,
        addToCart,
    });
};
/**
 * Headless component for notes
 *
 * @example
 * ```tsx
 * <CurrentCart.Notes>
 *   {({ notes, updateNotes }) => (
 *     <textarea
 *       value={notes}
 *       onChange={e => updateNotes(e.target.value)}
 *       placeholder="Special instructions for your order"
 *     />
 *   )}
 * </CurrentCart.Notes>
 * ```
 */
export const Notes = (props) => {
    const service = useService(CurrentCartServiceDefinition);
    const notes = service.buyerNotes.get();
    return props.children({
        notes,
        updateNotes: service.setBuyerNotes,
    });
};
/**
 * Headless component for coupon functionality
 *
 * @example
 * ```tsx
 * <CurrentCart.Coupon>
 *   {({ appliedCoupon, apply, remove, isLoading, error }) => (
 *     <div>
 *       {error && <p>Error: {error}</p>}
 *       {appliedCoupon ? (
 *         <div>
 *           <span>Coupon: {appliedCoupon}</span>
 *           <button onClick={remove} disabled={isLoading}>
 *             {isLoading ? 'Removing...' : 'Remove'}
 *           </button>
 *         </div>
 *       ) : (
 *         <form onSubmit={e => {
 *           e.preventDefault();
 *           const code = new FormData(e.currentTarget).get('couponCode');
 *           if (code?.trim()) apply(code.trim());
 *         }}>
 *           <input name="couponCode" placeholder="Enter promo code" disabled={isLoading} />
 *           <button type="submit" disabled={isLoading}>
 *             {isLoading ? 'Applying...' : 'Apply'}
 *           </button>
 *         </form>
 *       )}
 *     </div>
 *   )}
 * </CurrentCart.Coupon>
 * ```
 */
export const Coupon = (props) => {
    const service = useService(CurrentCartServiceDefinition);
    const cart = service.cart.get();
    const isLoading = service.isCouponLoading.get();
    const error = service.error.get();
    const appliedCoupon = cart?.appliedDiscounts?.find((discount) => discount.coupon?.code)
        ?.coupon?.code || null;
    return props.children({
        appliedCoupon,
        apply: service.applyCoupon,
        remove: service.removeCoupon,
        isLoading,
        error,
    });
};
/**
 * Headless component for line item added event subscription
 *
 * Provides a way to subscribe to cart addition events and receive notifications
 * when items are added to the current cart. The callback receives the updated
 * line items array, allowing you to show notifications, trigger animations,
 * or perform other actions when products are added.
 *
 * @example
 * ```tsx
 * // Basic usage - show global notification
 * <CurrentCart.LineItemAdded>
 *   {({ onAddedToCart }) => {
 *     useEffect(() => {
 *       return onAddedToCart((lineItems) => {
 *         setShowSuccessMessage(true);
 *         setTimeout(() => setShowSuccessMessage(false), 3000);
 *       });
 *     }, [onAddedToCart]);
 *     return null;
 *   }}
 * </CurrentCart.LineItemAdded>
 * ```
 *
 * @example
 * ```tsx
 * // Product-specific usage - show notification only for specific product
 * <CurrentCart.LineItemAdded>
 *   {({ onAddedToCart }) => {
 *     useEffect(() => {
 *       return onAddedToCart((lineItems) => {
 *         if (!lineItems) return;
 *         const myLineItemIsThere = lineItems.some(
 *           lineItem => lineItem.catalogReference?.catalogItemId === product._id
 *         );
 *         if (!myLineItemIsThere) return;
 *
 *         setShowSuccessMessage(true);
 *         setTimeout(() => setShowSuccessMessage(false), 3000);
 *       });
 *     }, [onAddedToCart, product._id]);
 *     return null;
 *   }}
 * </CurrentCart.LineItemAdded>
 * ```
 *
 * @example
 * ```tsx
 * // Combined with cart opening - show notification then open cart
 * <CurrentCart.OpenTrigger>
 *   {({ open }) => (
 *     <CurrentCart.LineItemAdded>
 *       {({ onAddedToCart }) => {
 *         useEffect(() => {
 *           return onAddedToCart((lineItems) => {
 *             setShowSuccessMessage(true);
 *             setTimeout(() => {
 *               setShowSuccessMessage(false);
 *               open(); // Open cart after notification
 *             }, 3000);
 *           });
 *         }, [onAddedToCart]);
 *         return null;
 *       }}
 *     </CurrentCart.LineItemAdded>
 *   )}
 * </CurrentCart.OpenTrigger>
 * ```
 */
export const LineItemAdded = (props) => {
    const service = useService(CurrentCartServiceDefinition);
    return props.children({
        onAddedToCart: service.onAddedToCart,
    });
};
