// @wix/bookings/build/es/meta.mjs
// meta.ts
import * as availabilityCalendar from "@wix/auto_sdk_bookings_availability-calendar/meta";
import * as availabilityTimeSlots from "@wix/auto_sdk_bookings_availability-time-slots/meta";
import * as eventTimeSlots from "@wix/auto_sdk_bookings_event-time-slots/meta";
import * as multiServiceAvailabilityTimeSlots from "@wix/auto_sdk_bookings_multi-service-availability-time-slots/meta";
import * as externalCalendars from "@wix/auto_sdk_bookings_external-calendars/meta";
import * as serviceOptionsAndVariants from "@wix/auto_sdk_bookings_service-options-and-variants/meta";
import * as categoriesV2 from "@wix/auto_sdk_bookings_categories-v-2/meta";
import * as bookingFees from "@wix/auto_sdk_bookings_booking-fees/meta";
import * as bookingPolicySnapshots from "@wix/auto_sdk_bookings_booking-policy-snapshots/meta";
import * as extendedBookings from "@wix/auto_sdk_bookings_extended-bookings/meta";
import * as resources from "@wix/auto_sdk_bookings_resources/meta";
import * as resourceTypes from "@wix/auto_sdk_bookings_resource-types/meta";
import * as services from "@wix/auto_sdk_bookings_services/meta";
import * as staffMembers from "@wix/auto_sdk_bookings_staff-members/meta";
import * as bookingPolicies from "@wix/auto_sdk_bookings_booking-policies/meta";
import * as categories from "@wix/auto_sdk_bookings_categories/meta";
import * as attendance from "@wix/auto_sdk_bookings_attendance/meta";
import * as bookings from "@wix/auto_sdk_bookings_bookings/meta";
import * as pricing from "@wix/auto_sdk_bookings_pricing/meta";
export {
  attendance,
  availabilityCalendar,
  availabilityTimeSlots,
  bookingFees,
  bookingPolicies,
  bookingPolicySnapshots,
  bookings,
  categories,
  categoriesV2,
  eventTimeSlots,
  extendedBookings,
  externalCalendars,
  multiServiceAvailabilityTimeSlots,
  pricing,
  resourceTypes,
  resources,
  serviceOptionsAndVariants,
  services,
  staffMembers
};
