// @wix/auto_sdk_restaurants_menu-ordering-settings/build/es/index.typings.mjs
// src/restaurants-menu-settings-v1-menu-ordering-settings-menu-ordering-settings.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/restaurants-menu-settings-v1-menu-ordering-settings-menu-ordering-settings.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixRestaurantsMenuSettingsV1MenuOrderingSettingsServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/_api/menu-ordering-settings",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/menu-ordering-settings",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/menu-ordering-settings",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/menu-ordering-settings",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/menu-ordering-settings",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/menu-ordering-settings",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/menu-ordering-settings",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/menu-ordering-settings",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/menu-ordering-settings",
        destPath: ""
      }
    ],
    _base_domain_: [
      {
        srcPath: "/_api/menu-ordering-settings",
        destPath: ""
      }
    ],
    "users._base_domain_": [
      {
        srcPath: "/_api/menu-ordering-settings",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/menu-ordering-settings",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/menu-ordering-settings",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/menu-ordering-settings",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/menu-ordering-settings",
        destPath: ""
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/menu-ordering-settings",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_menu-ordering-settings";
function getMenuOrderingSettings(payload) {
  function __getMenuOrderingSettings({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menu_settings.v1.menu_ordering_settings",
      method: "GET",
      methodFqn: "wix.restaurants.menu_settings.v1.MenuOrderingSettingsService.GetMenuOrderingSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenuSettingsV1MenuOrderingSettingsServiceUrl({
        protoPath: "/v1/menu-ordering-settings/{menuOrderingSettingsId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "menuOrderingSettings.createdDate" },
            { path: "menuOrderingSettings.updatedDate" },
            {
              path: "menuOrderingSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "menuOrderingSettings.availability.timestampRangesOptions.ranges.endTime"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getMenuOrderingSettings;
}
function upsertMenuOrderingSettingsByMenuId(payload) {
  function __upsertMenuOrderingSettingsByMenuId({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "menuOrderingSettings.createdDate" },
          { path: "menuOrderingSettings.updatedDate" },
          {
            path: "menuOrderingSettings.availability.timestampRangesOptions.ranges.startTime"
          },
          {
            path: "menuOrderingSettings.availability.timestampRangesOptions.ranges.endTime"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menu_settings.v1.menu_ordering_settings",
      method: "POST",
      methodFqn: "wix.restaurants.menu_settings.v1.MenuOrderingSettingsService.UpsertMenuOrderingSettingsByMenuId",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenuSettingsV1MenuOrderingSettingsServiceUrl({
        protoPath: "/v1/menu-ordering-settings/upsert/menu-id/{menuOrderingSettings.menuId}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "menuOrderingSettings.createdDate" },
            { path: "menuOrderingSettings.updatedDate" },
            {
              path: "menuOrderingSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "menuOrderingSettings.availability.timestampRangesOptions.ranges.endTime"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __upsertMenuOrderingSettingsByMenuId;
}
function updateMenuOrderingSettings(payload) {
  function __updateMenuOrderingSettings({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "menuOrderingSettings.createdDate" },
          { path: "menuOrderingSettings.updatedDate" },
          {
            path: "menuOrderingSettings.availability.timestampRangesOptions.ranges.startTime"
          },
          {
            path: "menuOrderingSettings.availability.timestampRangesOptions.ranges.endTime"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menu_settings.v1.menu_ordering_settings",
      method: "PATCH",
      methodFqn: "wix.restaurants.menu_settings.v1.MenuOrderingSettingsService.UpdateMenuOrderingSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenuSettingsV1MenuOrderingSettingsServiceUrl({
        protoPath: "/v1/menu-ordering-settings/{menuOrderingSettings.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "menuOrderingSettings.createdDate" },
            { path: "menuOrderingSettings.updatedDate" },
            {
              path: "menuOrderingSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "menuOrderingSettings.availability.timestampRangesOptions.ranges.endTime"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateMenuOrderingSettings;
}
function queryMenuOrderingSettings(payload) {
  function __queryMenuOrderingSettings({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menu_settings.v1.menu_ordering_settings",
      method: "POST",
      methodFqn: "wix.restaurants.menu_settings.v1.MenuOrderingSettingsService.QueryMenuOrderingSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenuSettingsV1MenuOrderingSettingsServiceUrl({
        protoPath: "/v1/menu-ordering-settings/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "menuOrderingSettings.createdDate" },
            { path: "menuOrderingSettings.updatedDate" },
            {
              path: "menuOrderingSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "menuOrderingSettings.availability.timestampRangesOptions.ranges.endTime"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryMenuOrderingSettings;
}
function bulkUpdateMenuOrderingSettings(payload) {
  function __bulkUpdateMenuOrderingSettings({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "menusOrderingSettings.mask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "menusOrderingSettings.menuOrderingSettings.createdDate" },
          { path: "menusOrderingSettings.menuOrderingSettings.updatedDate" },
          {
            path: "menusOrderingSettings.menuOrderingSettings.availability.timestampRangesOptions.ranges.startTime"
          },
          {
            path: "menusOrderingSettings.menuOrderingSettings.availability.timestampRangesOptions.ranges.endTime"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menu_settings.v1.menu_ordering_settings",
      method: "POST",
      methodFqn: "wix.restaurants.menu_settings.v1.MenuOrderingSettingsService.BulkUpdateMenuOrderingSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenuSettingsV1MenuOrderingSettingsServiceUrl({
        protoPath: "/v1/bulk/menu-ordering-settings/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.menuOrderingSettings.createdDate" },
            { path: "results.menuOrderingSettings.updatedDate" },
            {
              path: "results.menuOrderingSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "results.menuOrderingSettings.availability.timestampRangesOptions.ranges.endTime"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateMenuOrderingSettings;
}
function listMenusAvailabilityStatus(payload) {
  function __listMenusAvailabilityStatus({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "timeSlot.startTime" }, { path: "timeSlot.endTime" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menu_settings.v1.menu_ordering_settings",
      method: "GET",
      methodFqn: "wix.restaurants.menu_settings.v1.MenuOrderingSettingsService.ListMenusAvailabilityStatus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenuSettingsV1MenuOrderingSettingsServiceUrl({
        protoPath: "/v1/menu-ordering-settings/menus-availability-status",
        data: serializedData,
        host
      }),
      params: toURLSearchParams(serializedData)
    };
    return metadata;
  }
  return __listMenusAvailabilityStatus;
}
function updateExtendedFields(payload) {
  function __updateExtendedFields({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menu_settings.v1.menu_ordering_settings",
      method: "POST",
      methodFqn: "wix.restaurants.menu_settings.v1.MenuOrderingSettingsService.UpdateExtendedFields",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenuSettingsV1MenuOrderingSettingsServiceUrl({
        protoPath: "/v1/menu-ordering-settings/{id}/update-extended-fields",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "menuOrderingSettings.createdDate" },
            { path: "menuOrderingSettings.updatedDate" },
            {
              path: "menuOrderingSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "menuOrderingSettings.availability.timestampRangesOptions.ranges.endTime"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateExtendedFields;
}
function bulkUpdateMenuOrderingSettingsTags(payload) {
  function __bulkUpdateMenuOrderingSettingsTags({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menu_settings.v1.menu_ordering_settings",
      method: "POST",
      methodFqn: "wix.restaurants.menu_settings.v1.MenuOrderingSettingsService.BulkUpdateMenuOrderingSettingsTags",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenuSettingsV1MenuOrderingSettingsServiceUrl({
        protoPath: "/v1/bulk/menu-ordering-settings/update-tags",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateMenuOrderingSettingsTags;
}
function bulkUpdateMenuOrderingSettingsTagsByFilter(payload) {
  function __bulkUpdateMenuOrderingSettingsTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menu_settings.v1.menu_ordering_settings",
      method: "POST",
      methodFqn: "wix.restaurants.menu_settings.v1.MenuOrderingSettingsService.BulkUpdateMenuOrderingSettingsTagsByFilter",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenuSettingsV1MenuOrderingSettingsServiceUrl({
        protoPath: "/v1/bulk/menu-ordering-settings/update-tags-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateMenuOrderingSettingsTagsByFilter;
}

// src/restaurants-menu-settings-v1-menu-ordering-settings-menu-ordering-settings.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var AvailabilityType = /* @__PURE__ */ ((AvailabilityType2) => {
  AvailabilityType2["UNSPECIFIED_AVAILABILITY_OPTION"] = "UNSPECIFIED_AVAILABILITY_OPTION";
  AvailabilityType2["ALWAYS_AVAILABLE"] = "ALWAYS_AVAILABLE";
  AvailabilityType2["WEEKLY_SCHEDULE"] = "WEEKLY_SCHEDULE";
  AvailabilityType2["TIMESTAMP_RANGES"] = "TIMESTAMP_RANGES";
  return AvailabilityType2;
})(AvailabilityType || {});
var EntitiesDayOfWeek = /* @__PURE__ */ ((EntitiesDayOfWeek2) => {
  EntitiesDayOfWeek2["MON"] = "MON";
  EntitiesDayOfWeek2["TUE"] = "TUE";
  EntitiesDayOfWeek2["WED"] = "WED";
  EntitiesDayOfWeek2["THU"] = "THU";
  EntitiesDayOfWeek2["FRI"] = "FRI";
  EntitiesDayOfWeek2["SAT"] = "SAT";
  EntitiesDayOfWeek2["SUN"] = "SUN";
  return EntitiesDayOfWeek2;
})(EntitiesDayOfWeek || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var AvailabilityStatus = /* @__PURE__ */ ((AvailabilityStatus2) => {
  AvailabilityStatus2["UNKNOWN_AVAILABILITY_STATUS"] = "UNKNOWN_AVAILABILITY_STATUS";
  AvailabilityStatus2["AVAILABLE"] = "AVAILABLE";
  AvailabilityStatus2["UNAVAILABLE"] = "UNAVAILABLE";
  return AvailabilityStatus2;
})(AvailabilityStatus || {});
var PlacementType = /* @__PURE__ */ ((PlacementType2) => {
  PlacementType2["BEFORE"] = "BEFORE";
  PlacementType2["AFTER"] = "AFTER";
  PlacementType2["REPLACE"] = "REPLACE";
  return PlacementType2;
})(PlacementType || {});
var DayOfWeek = /* @__PURE__ */ ((DayOfWeek2) => {
  DayOfWeek2["MONDAY"] = "MONDAY";
  DayOfWeek2["TUESDAY"] = "TUESDAY";
  DayOfWeek2["WEDNESDAY"] = "WEDNESDAY";
  DayOfWeek2["THURSDAY"] = "THURSDAY";
  DayOfWeek2["FRIDAY"] = "FRIDAY";
  DayOfWeek2["SATURDAY"] = "SATURDAY";
  DayOfWeek2["SUNDAY"] = "SUNDAY";
  return DayOfWeek2;
})(DayOfWeek || {});
var ResolutionMethod = /* @__PURE__ */ ((ResolutionMethod2) => {
  ResolutionMethod2["QUERY_PARAM"] = "QUERY_PARAM";
  ResolutionMethod2["SUBDOMAIN"] = "SUBDOMAIN";
  ResolutionMethod2["SUBDIRECTORY"] = "SUBDIRECTORY";
  return ResolutionMethod2;
})(ResolutionMethod || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function getMenuOrderingSettings2(menuOrderingSettingsId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    menuOrderingSettingsId
  });
  const reqOpts = getMenuOrderingSettings(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.menuOrderingSettings;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { menuOrderingSettingsId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["menuOrderingSettingsId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function upsertMenuOrderingSettingsByMenuId2(menuId, menuOrderingSettings) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    menuOrderingSettings: { ...menuOrderingSettings, menuId }
  });
  const reqOpts = upsertMenuOrderingSettingsByMenuId(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { menuOrderingSettings: "$[1]" },
        explicitPathsToArguments: { "menuOrderingSettings.menuId": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["menuId", "menuOrderingSettings"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateMenuOrderingSettings2(_id, menuOrderingSettings) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    menuOrderingSettings: { ...menuOrderingSettings, id: _id }
  });
  const reqOpts = updateMenuOrderingSettings(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.menuOrderingSettings;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { menuOrderingSettings: "$[1]" },
        explicitPathsToArguments: { "menuOrderingSettings.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "menuOrderingSettings"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryMenuOrderingSettings2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryMenuOrderingSettings(
        payload
      );
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [])
      );
      return {
        items: transformedData?.menuOrderingSettings,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function bulkUpdateMenuOrderingSettings2(menusOrderingSettings, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    menusOrderingSettings,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkUpdateMenuOrderingSettings(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          menusOrderingSettings: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["menusOrderingSettings", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listMenusAvailabilityStatus2(timeSlot, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    timeSlot,
    operationId: options?.operationId,
    cursorPaging: options?.cursorPaging
  });
  const reqOpts = listMenusAvailabilityStatus(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          timeSlot: "$[0]",
          operationId: "$[1].operationId",
          cursorPaging: "$[1].cursorPaging"
        },
        singleArgumentUnchanged: false
      },
      ["timeSlot", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateExtendedFields2(_id, namespace, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    id: _id,
    namespace,
    namespaceData: options?.namespaceData
  });
  const reqOpts = updateExtendedFields(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          id: "$[0]",
          namespace: "$[1]",
          namespaceData: "$[2].namespaceData"
        },
        singleArgumentUnchanged: false
      },
      ["_id", "namespace", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateMenuOrderingSettingsTags2(menuOrderingSettingsIds, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    menuOrderingSettingsIds,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateMenuOrderingSettingsTags(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          menuOrderingSettingsIds: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["menuOrderingSettingsIds", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateMenuOrderingSettingsTagsByFilter2(filter, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    filter,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateMenuOrderingSettingsTagsByFilter(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          filter: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["filter", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
export {
  AvailabilityStatus,
  AvailabilityType,
  DayOfWeek,
  EntitiesDayOfWeek,
  PlacementType,
  ResolutionMethod,
  SortOrder,
  WebhookIdentityType,
  bulkUpdateMenuOrderingSettings2 as bulkUpdateMenuOrderingSettings,
  bulkUpdateMenuOrderingSettingsTags2 as bulkUpdateMenuOrderingSettingsTags,
  bulkUpdateMenuOrderingSettingsTagsByFilter2 as bulkUpdateMenuOrderingSettingsTagsByFilter,
  getMenuOrderingSettings2 as getMenuOrderingSettings,
  listMenusAvailabilityStatus2 as listMenusAvailabilityStatus,
  queryMenuOrderingSettings2 as queryMenuOrderingSettings,
  updateExtendedFields2 as updateExtendedFields,
  updateMenuOrderingSettings2 as updateMenuOrderingSettings,
  upsertMenuOrderingSettingsByMenuId2 as upsertMenuOrderingSettingsByMenuId
};
