// @wix/auto-patterns/dist/cjs/components/AutoPatternsRoute/AutoPatternsRoutes.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.AutoPatternsRoutes = void 0;
var _react = _interopRequireDefault(require("react"));
var _hooks = require("../../hooks");
var _router = require("@wix/patterns/router");
var _AutoPatternsPage = require("./AutoPatternsPage");
var _reactRouterDom = require("react-router-dom");
var _jsxFileName = "/home/builduser/work/73d19ce3378ce4dd/packages/auto-patterns/dist/cjs/components/AutoPatternsRoute/AutoPatternsRoutes.tsx";
const pageTypeToRouteType = {
  collectionPage: 'collection',
  entityPage: 'editEntity'
};
const AutoPatternsRoutes = ({
  config
}) => {
  const {
    getPagePath
  } = (0, _hooks.usePagePath)(config);
  const mainPage = config.pages.find(page => page.appMainPage);
  const routes = config.pages.map(page => {
    const routePath = getPagePath(page.id);
    if (!routePath) {
      return null;
    }
    return /*#__PURE__*/_react.default.createElement(_router.PatternsReactRoute, {
      key: page.id,
      type: pageTypeToRouteType[page.type] ?? 'other',
      path: routePath,
      element: /*#__PURE__*/_react.default.createElement(_AutoPatternsPage.AutoPatternsPage, {
        pageConfig: page,
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 31,
          columnNumber: 20
        }
      }),
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 27,
        columnNumber: 9
      }
    });
  }).filter(Boolean);
  if (mainPage) {
    const mainPagePath = getPagePath(mainPage.id);
    if (mainPagePath) {
      routes.push(/*#__PURE__*/_react.default.createElement(_router.PatternsReactRoute, {
        key: "main-redirect",
        type: "other",
        path: "/",
        element: /*#__PURE__*/_react.default.createElement(_reactRouterDom.Navigate, {
          to: mainPagePath,
          __self: void 0,
          __source: {
            fileName: _jsxFileName,
            lineNumber: 46,
            columnNumber: 20
          }
        }),
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 42,
          columnNumber: 9
        }
      }));
    }
  }
  return /*#__PURE__*/_react.default.createElement(_router.PatternsReactRouter, {
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 52,
      columnNumber: 10
    }
  }, routes);
};
exports.AutoPatternsRoutes = AutoPatternsRoutes;
