// @wix/auto-patterns/dist/cjs/components/AutoPatternsRoute/AutoPatternsPage.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.AutoPatternsPage = void 0;
var _react = _interopRequireDefault(require("react"));
var _AutoPatternsCollectionPage = require("../AutoPatternsCollectionPage");
var _AutoPatternsEntityPage = require("../AutoPatternsEntityPage");
var _SchemaContext = require("../../providers/SchemaContext");
var _SkeletonEntity = require("../AutoPatternsEntityPage/SkeletonEntity");
var _jsxFileName = "/home/builduser/work/73d19ce3378ce4dd/packages/auto-patterns/dist/cjs/components/AutoPatternsRoute/AutoPatternsPage.tsx";
const AutoPatternsPage = ({
  pageConfig
}) => {
  const {
    type,
    collectionPage,
    entityPage
  } = pageConfig;
  switch (type) {
    case 'collectionPage':
      if (collectionPage) {
        return /*#__PURE__*/_react.default.createElement(_AutoPatternsCollectionPage.AutoPatternsCollectionPage, {
          configuration: collectionPage,
          __self: void 0,
          __source: {
            fileName: _jsxFileName,
            lineNumber: 18,
            columnNumber: 16
          }
        });
      }
      break;
    case 'entityPage':
      if (entityPage) {
        return /*#__PURE__*/_react.default.createElement(_SchemaContext.SchemaProvider, {
          collection: entityPage,
          skeleton: /*#__PURE__*/_react.default.createElement(_SkeletonEntity.SkeletonEntity, {
            entityPage: entityPage,
            __self: void 0,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 26,
              columnNumber: 23
            }
          }),
          __self: void 0,
          __source: {
            fileName: _jsxFileName,
            lineNumber: 24,
            columnNumber: 11
          }
        }, /*#__PURE__*/_react.default.createElement(_AutoPatternsEntityPage.AutoPatternsEntityPage, {
          configuration: entityPage,
          __self: void 0,
          __source: {
            fileName: _jsxFileName,
            lineNumber: 28,
            columnNumber: 13
          }
        }));
      }
      break;
    default:
      break;
  }
  return null;
};
exports.AutoPatternsPage = AutoPatternsPage;
