// @wix/bex-core/dist/cjs/testkit/kanban/createKanbanStagesBackend.js
"use strict";

exports.__esModule = true;
exports.createKanbanStage = createKanbanStage;
exports.createKanbanStagesBackend = void 0;
var _lodash = require("lodash");
var _InMemoryBackend = require("../../InMemoryBackend");
var _kanbanStaticData = require("./kanbanStaticData");
function createKanbanStage(chance, {
  createdAt = chance.date({
    min: new Date(Date.now() - 365 * 24 * 60 * 60 * 1000),
    max: new Date(Date.now() - 2 * 24 * 60 * 60 * 1000)
  }),
  ...rest
} = {}) {
  return {
    id: chance.guid(),
    name: chance.name(),
    createdAt,
    updatedAt: createdAt,
    ...rest
  };
}
const createKanbanStagesBackend = ({
  createOne,
  ...params
} = {}) => {
  return new _InMemoryBackend.InMemoryBackend({
    enableTotal: params == null ? void 0 : params.enableTotal,
    createOne: i => {
      const raw = _kanbanStaticData.kanbanStaticStagesData[i];
      return {
        ...raw,
        createdAt: new Date(raw.createdAt ?? ''),
        updatedAt: new Date(raw.updatedAt ?? '')
      };
    },
    itemKey: item => item.id,
    total: _kanbanStaticData.kanbanStaticStagesData.length,
    predicate: query => {
      const {
        search
      } = query;
      // const filters = query.filters ?? {};
      const rgx = new RegExp((0, _lodash.escapeRegExp)(search), 'i');
      return item => {
        if (search && !rgx.test(item.name)) {
          return false;
        }
        return true;
      };
    },
    orderBy: query => {
      return [...(query.sort ?? [])];
    },
    ...params
  });
};
exports.createKanbanStagesBackend = createKanbanStagesBackend;
