// @wix/bex-core/dist/cjs/testkit/kanban/createKanbanItemsBackend.js
"use strict";

exports.__esModule = true;
exports.createKanbanItem = createKanbanItem;
exports.createKanbanItemsBackend = void 0;
var _lodash = require("lodash");
var _InMemoryBackend = require("../../InMemoryBackend");
var _kanbanStaticData = require("./kanbanStaticData");
function createKanbanItem(chance, {
  createdAt = chance.date({
    min: new Date(Date.now() - 365 * 24 * 60 * 60 * 1000),
    max: new Date(Date.now() - 2 * 24 * 60 * 60 * 1000)
  }),
  ...rest
} = {}) {
  return {
    id: chance.guid(),
    name: chance.name(),
    description: chance.sentence(),
    price: chance.integer({
      min: 0,
      max: 10000
    }),
    status: chance.pickone(['available', 'outOfStock']),
    createdAt,
    updatedAt: createdAt,
    stageId: chance.guid(),
    ...rest
  };
}
const createKanbanItemsBackend = ({
  createOne,
  ...params
} = {}) => {
  return new _InMemoryBackend.InMemoryBackend({
    enableTotal: params == null ? void 0 : params.enableTotal,
    createOne: i => {
      const raw = _kanbanStaticData.kanbanStaticLoadMoreCardsData[i];
      return {
        ...raw,
        createdAt: new Date(raw.createdAt ?? ''),
        updatedAt: new Date(raw.updatedAt ?? '')
      };
    },
    itemKey: item => item.id,
    total: 0,
    predicate: query => {
      const {
        search
      } = query;
      const filters = query.filters ?? {};
      const rgx = new RegExp((0, _lodash.escapeRegExp)(search), 'i');
      return item => {
        if (search && !rgx.test(item.name)) {
          return false;
        }
        if (filters.stageId && filters.stageId.length && filters.stageId.indexOf(item.id) === -1) {
          return false;
        }
        return true;
      };
    },
    orderBy: query => {
      return [...(query.sort ?? [])];
    },
    ...params
  });
};
exports.createKanbanItemsBackend = createKanbanItemsBackend;
