// @wix/ricos/dist/cjs/toolbars-v3/components/buttons/AddPluginMenuButtons/AddPluginMenuMobileButton.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var AddPluginMenuMobileButton_exports = {};
__export(AddPluginMenuMobileButton_exports, {
  AddPluginMenuMobileButton: () => AddPluginMenuMobileButton
});
module.exports = __toCommonJS(AddPluginMenuMobileButton_exports);
var import_react = __toESM(require("react"));
var import_ricos_context = require("./../../../../ricos-context/index.js");
var import_toolbar_button = require("./../../../../_shared/toolbar-button/toolbar-button.js");
var import_toggle_quick_actions_modal = require("./../../../../ricos-common-plugins/quick-actions/toggle-quick-actions-modal/toggle-quick-actions-modal.js");
var import_AddPluginMenuMobileButton_scss = __toESM(require("./AddPluginMenuMobileButton-scss.js"));
const AddPluginMenuMobileButton = ({
  toolbarItem
}) => {
  const { isMobile } = (0, import_react.useContext)(import_ricos_context.RicosContext);
  const modalService = (0, import_react.useContext)(import_ricos_context.ModalServiceContext) || {};
  return /* @__PURE__ */ import_react.default.createElement(
    import_toolbar_button.ToolbarButton,
    {
      dataHook: "add-plugin-button-mobile",
      onClick: () => {
        (0, import_toggle_quick_actions_modal.toggleQuickActionsModal)({ modalService, referenceElement: null, isMobile });
      },
      className: import_AddPluginMenuMobileButton_scss.default.toggleButton,
      prefix: toolbarItem.presentation?.icon,
      tabIndex: toolbarItem.presentation.tabIndex
    }
  );
};
