// @wix/unidriver-playwright/dist/cjs/UniDriverClass.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UniDriverClass = void 0;
const unidriver_common_1 = require("@wix/unidriver-common");
const UniDriverClassList_1 = require("./UniDriverClassList");
const enterText_1 = require("./utils/enterText");
class UniDriverClass extends unidriver_common_1.UniDriverBase {
    constructor(ctx) {
        super(ctx);
        Object.defineProperty(this, "type", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: 'puppeteer'
        });
        Object.defineProperty(this, "defaultTimeBudget", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: 5_000
        });
        Object.defineProperty(this, "api", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: {
                Class: UniDriverClass,
                ListClass: UniDriverClassList_1.UniDriverClassList,
                getType: () => 'playwright',
                unwrap: async (base) => base,
                getNative: async (base) => {
                    return base;
                },
                getElementsBySelector: async (base, selector) => {
                    return base.locator(selector).all();
                },
                click: async (base, options) => {
                    return base.click({ clickCount: options?.type === 'double' ? 2 : 1 });
                },
                text: async (base) => {
                    return base.evaluate((_base) => {
                        return (_base.textContent ?? '');
                    });
                },
                value: async (base) => {
                    return await base.evaluate((_base) => {
                        const value = _base.value;
                        return typeof value === 'number' ? String(value) : (value ?? '');
                    });
                },
                enterText: async (base, value, options) => {
                    return (0, enterText_1.enterText)(base, value, options);
                },
                enterValue: async (base, value, options) => {
                    return (0, enterText_1.enterText)(base, value, {
                        ...options,
                        clear: options.shouldClear,
                    });
                },
                pressKey: async (base, keyMeta) => {
                    return base.press(keyMeta.key);
                },
                attr: async (base, attr) => {
                    return base.evaluate((_base, _attr) => {
                        return _base.getAttribute(_attr);
                    }, attr);
                },
                hasClass: async (base, className) => {
                    return base.evaluate((_base, _className) => {
                        return _base.classList.contains(_className);
                    }, className);
                },
                prop: async (base, prop) => {
                    return base.evaluate((_base, _prop) => {
                        return _base[_prop];
                    }, prop);
                },
                mouse: {
                    hover: async (base) => {
                        return base.hover();
                    },
                    press: async (base) => {
                        return this.fireEvent(base, 'mousedown');
                    },
                    release: async (base) => {
                        return this.fireEvent(base, 'mouseup');
                    },
                    leave: async (base) => {
                        await this.fireEvent(base, 'mouseout');
                        await this.fireEvent(base, 'mouseleave');
                    },
                    moveTo: async (base, to) => {
                        const locator = await to.unwrap();
                        const boundingBox = await locator.boundingBox();
                        if (boundingBox) {
                            return base.evaluate((_base, _boundingBox) => {
                                const { x, y } = _boundingBox;
                                _base.dispatchEvent(new MouseEvent('mousemove', {
                                    clientX: x,
                                    clientY: y,
                                    bubbles: true,
                                    cancelable: false,
                                }));
                            }, {
                                x: boundingBox.x,
                                y: boundingBox.y,
                            });
                        }
                        else {
                            throw new Error(`Cannot find target element`);
                        }
                    },
                },
            }
        });
        (0, unidriver_common_1.bindUniDriverApiMethods)(this, ['isDisplayed', 'scrollIntoView']);
    }
    /** @deprecated use .unwrap() and native element */
    async isDisplayed() {
        const base = await this.getBaseElement();
        return base.isVisible();
    }
    /** @deprecated use .unwrap() + native element */
    async scrollIntoView() {
        const base = await this.getBaseElement();
        await base.evaluate((_base) => {
            // 'auto' is instant in other browsers, but chrome has additional 'instant' option
            _base.scrollIntoView({ behavior: 'instant' });
            _base.scrollIntoView({ behavior: 'auto' });
        });
        await base.hover();
    }
    fireEvent(base, eventName) {
        return base.evaluate((_base, _eventName) => {
            _base.dispatchEvent(new Event(_eventName, {
                bubbles: true,
                cancelable: true,
            }));
        }, eventName);
    }
}
exports.UniDriverClass = UniDriverClass;
