// @wix/patterns/dist/cjs/components/CollectionToolbars/CollectionToolbarsProvider.js
"use strict";

exports.__esModule = true;
exports.CollectionToolbarsProvider = CollectionToolbarsProvider;
var _FilterRenderingContext = require("../FilterRenderingContext");
var _react = _interopRequireWildcard(require("react"));
var _designSystem = require("@wix/design-system");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CollectionToolbars/CollectionToolbarsProvider.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function CollectionToolbarsProvider({
  children
}) {
  const [filterRenderingContext] = (0, _react.useState)(() => ({
    location: 'toolbar'
  }));
  const [wsrDefaults] = (0, _react.useState)(() => ({
    input: {
      size: 'small'
    },
    button: {
      size: 'small'
    }
  }));
  return /*#__PURE__*/_react.default.createElement(_FilterRenderingContext.FilterRenderingContextProvider, {
    value: filterRenderingContext,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 26,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.WixStyleReactDefaultsOverrideProvider, {
    value: wsrDefaults,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 27,
      columnNumber: 7
    }
  }, children));
}
