// @wix/patterns/dist/cjs/components/CollectionToolbars/CollectionToolbars.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.CollectionToolbars = void 0;
var _designSystem = require("@wix/design-system");
var _react = _interopRequireDefault(require("react"));
var _mobxReactLite = require("mobx-react-lite");
var _CollectionToolbar = require("../CollectionToolbar");
var _CollectionToolbarsProvider = require("./CollectionToolbarsProvider");
var _ToolbarAndTitlebarContainer = require("./ToolbarAndTitlebarContainer");
var _useIsMobile = require("../../hooks/useIsMobile");
var _CollectionToolbarsVisualization = require("./CollectionToolbarsVisualization");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CollectionToolbars/CollectionToolbars.tsx";
function BorderRadiusTopCard({
  children
}) {
  const isMobile = (0, _useIsMobile.useIsMobile)();
  return /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
    direction: "vertical",
    backgroundColor: "D80",
    borderTopLeftRadius: isMobile ? 0 : 8,
    borderTopRightRadius: isMobile ? 0 : 8,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 50,
      columnNumber: 5
    }
  }, children);
}
function BorderRadiusTopBottomCard({
  children
}) {
  const isMobile = (0, _useIsMobile.useIsMobile)();
  return /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
    direction: "vertical",
    backgroundColor: "D80",
    borderRadius: isMobile ? 0 : 8,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 64,
      columnNumber: 5
    }
  }, children);
}
function _CollectionToolbars(props) {
  var _featuredComponents$a;
  const {
    sticky,
    state,
    search,
    selectionToolbar,
    dragAndDropReorderModeToolbar,
    filters,
    title,
    tabs,
    customColumns,
    multiLevelSorting,
    views,
    extraToolbar,
    summaryBar,
    topNotification: topNotificationProp,
    titleBar,
    tableGridSwitchButton,
    exportButton,
    primaryActionButton,
    secondaryActions,
    moreActions,
    visualization: customSlot
  } = props;
  const {
    collection,
    featuredComponents,
    hasSubToolbarComponents,
    shouldDisplayFiltersTags,
    bulk: {
      selectedIds
    },
    internalScroll
  } = state;
  const isMobile = (0, _useIsMobile.useIsMobile)();
  const topNotification = typeof topNotificationProp === 'function' ? topNotificationProp() : topNotificationProp;
  const AppliedFiltersTagList = (_featuredComponents$a = featuredComponents.appliedFiltersTagList.value) == null ? void 0 : _featuredComponents$a.component;
  const {
    showRefreshLoader
  } = collection;
  const shouldDisplaySubToolbar = shouldDisplayFiltersTags && hasSubToolbarComponents;
  const toolbars = /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, selectionToolbar && !showRefreshLoader && selectedIds.length > 0 ? selectionToolbar : /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, extraToolbar, /*#__PURE__*/_react.default.createElement(_CollectionToolbar.CollectionToolbar, {
    dataHook: "collection-toolbar",
    search: search,
    customColumns: customColumns,
    multiLevelSorting: multiLevelSorting,
    views: views,
    state: state,
    filters: filters,
    title: title,
    tabs: tabs,
    tableGridSwitchButton: tableGridSwitchButton,
    exportButton: exportButton,
    primaryActionButton: primaryActionButton,
    dragAndDropReorderModeToolbar: dragAndDropReorderModeToolbar,
    secondaryActions: secondaryActions,
    moreActions: moreActions,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 132,
      columnNumber: 11
    }
  })), shouldDisplaySubToolbar && /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
    direction: "vertical",
    backgroundColor: "D80",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 152,
      columnNumber: 9
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.Table.SubToolbar, {
    dataHook: "collection-table-sub-toolbar",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 153,
      columnNumber: 11
    }
  }, AppliedFiltersTagList && shouldDisplayFiltersTags && /*#__PURE__*/_react.default.createElement(AppliedFiltersTagList, {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 155,
      columnNumber: 15
    }
  }))), summaryBar, topNotification);

  /* wrapping with Card is WSR way to make the sticky table header not to add extra border between the header and the content https://www.docs.wixdesignsystem.com/?path=/story/components-lists-table--table#Sticky_Toolbar_&_Sub-toolbar*/

  const child = /*#__PURE__*/_react.default.createElement(_ToolbarAndTitlebarContainer.ToolbarAndTitlebarContainer, {
    state: state.toolbar,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 168,
      columnNumber: 5
    }
  }, customSlot ? /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(BorderRadiusTopBottomCard, {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 171,
      columnNumber: 13
    }
  }, toolbars), /*#__PURE__*/_react.default.createElement(_CollectionToolbarsVisualization.CollectionToolbarsVisualization, {
    state: state,
    visualization: customSlot,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 172,
      columnNumber: 13
    }
  }), /*#__PURE__*/_react.default.createElement(BorderRadiusTopCard, {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 173,
      columnNumber: 13
    }
  }, titleBar)) : /*#__PURE__*/_react.default.createElement(BorderRadiusTopCard, {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 176,
      columnNumber: 11
    }
  }, toolbars, titleBar));
  return /*#__PURE__*/_react.default.createElement(_CollectionToolbarsProvider.CollectionToolbarsProvider, {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 185,
      columnNumber: 5
    }
  }, sticky && !internalScroll ? /*#__PURE__*/_react.default.createElement(_designSystem.Page.Sticky, {
    style: isMobile ? {
      top: 78
    } : undefined,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 187,
      columnNumber: 9
    }
  }, child) : child);
}
const CollectionToolbars = exports.CollectionToolbars = (0, _mobxReactLite.observer)(_CollectionToolbars);
