// @wix/headless-pricing-plans/cjs/dist/react/Plan.js
import { jsx as _jsx, Fragment as _Fragment } from "react/jsx-runtime";
import React from 'react';
export { WixMediaImage } from '@wix/headless-media/react';
import { AsChildSlot } from '@wix/headless-utils/react';
import { PlanServiceDefinition, PRICING_PLANS_APP_ID, } from '../services/index.js';
import { Root as CoreRoot, Plan as CorePlan, Image as CoreImage, Name as CoreName, Description as CoreDescription, Price as CorePrice, AdditionalFees as CoreAdditionalFees, AdditionalFeesRepeater as CoreAdditionalFeesRepeater, AdditionalFeeName as CoreAdditionalFeeName, AdditionalFeeAmount as CoreAdditionalFeeAmount, AdditionalFeeRoot as CoreAdditionalFeeRoot, Recurrence as CoreRecurrence, Duration as CoreDuration, Perks as CorePerks, PerksRepeater as CorePerksRepeater, PerkItemContext, PerkItem as CorePerkItem, FreeTrialDays as CoreFreeTrialDays, } from './core/Plan.js';
import { WixMediaImage } from '@wix/headless-media/react';
import { Commerce } from '@wix/headless-ecom/react';
import { useService } from '@wix/services-manager-react';
var PlanTestId;
(function (PlanTestId) {
    PlanTestId["Plan"] = "plan-plan";
    PlanTestId["Image"] = "plan-image";
    PlanTestId["Name"] = "plan-name";
    PlanTestId["Description"] = "plan-description";
    PlanTestId["Price"] = "plan-price";
    PlanTestId["AdditionalFees"] = "plan-additional-fees";
    PlanTestId["AdditionalFeeName"] = "plan-additional-fee-name";
    PlanTestId["AdditionalFeeAmount"] = "plan-additional-fee-amount";
    PlanTestId["Perks"] = "plan-perks";
    PlanTestId["PerkItem"] = "plan-perk-item";
    PlanTestId["ActionBuyNow"] = "plan-action-buy-now";
    PlanTestId["FreeTrialDays"] = "plan-free-trial-days";
})(PlanTestId || (PlanTestId = {}));
/**
 * The root container that provides plan context to child components
 *
 * @component
 * @example
 * ```tsx
 * <Plan.Root planServiceConfig={planServiceConfig}>
 *   <Plan.Plan>
 *     <Plan.Image />
 *     <Plan.Name />
 *     <Plan.Price />
 *     <Plan.AdditionalFees />
 *     <Plan.Recurrence />
 *     <Plan.Duration />
 *     <Plan.Action.BuyNow label="Select Plan" />
 *   </Plan.Plan>
 * </Plan.Root>
 * ```
 */
export const Root = ({ planServiceConfig, children }) => (_jsx(CoreRoot, { planServiceConfig: planServiceConfig, children: children }));
/**
 * Displays the plan data with support for loading state
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Plan.Plan className="flex flex-col gap-4" loadingState={<div>Loading...</div>}>
 *   <Plan.Image />
 *   <Plan.Name />
 * </Plan.Plan>
 *
 * // With asChild
 * <Plan.Plan asChild>
 *   {React.forwardRef(({isLoading, error, plan}, ref) => {
 *     if (isLoading) {
 *       return <div>Loading...</div>;
 *     }
 *     if (error) {
 *       return <div>Error!</div>;
 *     }
 *     return (
 *       <div ref={ref} className="flex flex-col gap-4">
 *         Plan {plan.name}
 *       </div>
 *     );
 *   })}
 * </Plan.Plan>
 *
 */
export const Plan = React.forwardRef(({ children, asChild, className, loadingState }, ref) => {
    return (_jsx(CorePlan, { children: (planData) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": PlanTestId.Plan, "data-is-loading": planData.isLoading, "data-has-error": planData.error !== null, customElement: children, customElementProps: planData, children: _jsx("div", { children: planData.isLoading
                    ? loadingState
                    : children }) })) }));
});
/**
 * Displays the plan image using WixMediaImage
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Plan.Image className="w-full h-full object-cover" />
 *
 * // asChild with primitive
 * <Plan.Image asChild>
 *   <img className="w-full h-full object-cover" />
 * </Plan.Image>
 *
 * // asChild with react component
 * <Plan.Image asChild>
 *   {React.forwardRef(({src, ...props}, ref) => (
 *     <img ref={ref} {...props} src={src} className="w-full h-full object-cover" />
 *   ))}
 * </Plan.Image>
 * ```
 */
export const Image = React.forwardRef((props, ref) => (_jsx(CoreImage, { children: (renderProps) => (_jsx(WixMediaImage, { ...props, ref: ref, media: { image: renderProps.image }, "data-testid": PlanTestId.Image })) })));
/**
 * Displays the plan name.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Plan.Name className="text-2xl font-bold" />
 *
 * // asChild with primitive
 * <Plan.Name asChild>
 *   <h1 className="text-2xl font-bold" />
 * </Plan.Name>
 *
 * // asChild with react component
 * <Plan.Name asChild>
 *   {React.forwardRef(({name, ...props}, ref) => (
 *     <h1 ref={ref} {...props} className="text-2xl font-bold">
 *       {name}
 *     </h1>
 *   ))}
 * </Plan.Name>
 * ```
 */
export const Name = React.forwardRef(({ children, asChild, className }, ref) => {
    return (_jsx(CoreName, { children: (nameData) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, customElement: children, customElementProps: nameData, className: className, content: nameData.name, "data-testid": PlanTestId.Name, children: _jsx("span", { children: nameData.name }) })) }));
});
/**
 * Displays the plan description.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Plan.Description className="text-sm" />
 *
 * // asChild with primitive
 * <Plan.Description asChild>
 *   <p className="text-sm" />
 * </Plan.Description>
 *
 * // asChild with react component
 * <Plan.Description asChild>
 *   {React.forwardRef(({description, ...props}, ref) => (
 *     <p ref={ref} {...props} className="text-sm">
 *       {description}
 *     </p>
 *   ))}
 * </Plan.Description>
 * ```
 */
export const Description = React.forwardRef(({ children, asChild, className }, ref) => {
    return (_jsx(CoreDescription, { children: (descriptionData) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, customElement: children, customElementProps: descriptionData, className: className, content: descriptionData.description, "data-testid": PlanTestId.Description, children: _jsx("span", { children: descriptionData.description }) })) }));
});
/**
 * Displays the plan price.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Plan.Price className="text-2xl font-bold" />
 *
 * // asChild with primitive
 * <Plan.Price asChild>
 *   <span className="text-2xl font-bold" />
 * </Plan.Price>
 *
 * // asChild with react component
 * <Plan.Price asChild>
 *   {React.forwardRef(({price, ...props}, ref) => (
 *     <span ref={ref} {...props} className="text-2xl font-bold">
 *       {price.formattedPrice}
 *       <br />
 *       <br />
 *       {price.amount} {price.currency}
 *     </span>
 *   ))}
 * </Plan.Price>
 * ```
 */
export const Price = React.forwardRef(({ children, asChild, className }, ref) => {
    return (_jsx(CorePrice, { children: (priceData) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, customElement: children, customElementProps: priceData, className: className, "data-testid": PlanTestId.Price, content: priceData.price.formattedPrice, children: _jsx("span", { children: priceData.price.formattedPrice }) })) }));
});
/**
 * Container for plan additional fees.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Plan.AdditionalFees className="flex flex-col gap-2">
 *   <Plan.AdditionalFeesRepeater>
 *     <Plan.AdditionalFeeName />
 *     <Plan.AdditionalFeeAmount />
 *   </Plan.AdditionalFeesRepeater>
 * </Plan.AdditionalFees>
 *
 * // asChild with react component
 * <Plan.AdditionalFees asChild>
 *   {React.forwardRef(({additionalFees, ...props}, ref) => (
 *     <div ref={ref} {...props} className="fees-container">
 *       {additionalFees.length > 0 && 'Additional fees apply'}
 *     </div>
 *   ))}
 * </Plan.AdditionalFees>
 * ```
 */
export const AdditionalFees = React.forwardRef(({ children, asChild, className }, ref) => {
    return (_jsx(CoreAdditionalFees, { children: (additionalFeesData) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, customElement: children, customElementProps: additionalFeesData, className: className, "data-testid": PlanTestId.AdditionalFees, children: _jsx("div", { children: children }) })) }));
});
/**
 * Repeater component that renders children for each additional fee with the additional fee context.
 *
 * @component
 * @example
 * ```tsx
 * <Plan.AdditionalFeesRepeater>
 *   <Plan.AdditionalFeeName />
 *   <Plan.AdditionalFeeAmount />
 * </Plan.AdditionalFeesRepeater>
 * ```
 */
export const AdditionalFeesRepeater = ({ children, }) => {
    return (_jsx(CoreAdditionalFeesRepeater, { children: ({ additionalFees }) => (_jsx(_Fragment, { children: additionalFees.map((fee, i) => (_jsx(CoreAdditionalFeeRoot, { fee: fee, children: children }, i))) })) }));
};
/**
 * Displays the additional fee name. Must be used within an AdditionalFeesRepeater.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Plan.AdditionalFeeName className="text-sm" />
 *
 * // asChild with primitive
 * <Plan.AdditionalFeeName asChild>
 *   <span className="text-sm" />
 * </Plan.AdditionalFeeName>
 *
 * // asChild with react component
 * <Plan.AdditionalFeeName asChild>
 *   {React.forwardRef(({name, ...props}, ref) => (
 *     <span ref={ref} {...props} className="text-sm">
 *       {name}
 *     </span>
 *   ))}
 * </Plan.AdditionalFeeName>
 * ```
 */
export const AdditionalFeeName = React.forwardRef(({ children, asChild, className }, ref) => {
    return (_jsx(CoreAdditionalFeeName, { children: (additionalFeeNameData) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, customElement: children, customElementProps: additionalFeeNameData, className: className, "data-testid": PlanTestId.AdditionalFeeName, content: additionalFeeNameData.name, children: _jsx("span", { children: additionalFeeNameData.name }) })) }));
});
/**
 * Displays the additional fee amount. Must be used within an AdditionalFeesRepeater.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Plan.AdditionalFeeAmount className="text-sm" />
 *
 * // asChild with primitive
 * <Plan.AdditionalFeeAmount asChild>
 *   <span className="text-sm" />
 * </Plan.AdditionalFeeAmount>
 *
 * // asChild with react component
 * <Plan.AdditionalFeeAmount asChild>
 *   {React.forwardRef(({amount, currency, formattedFee, ...props}, ref) => (
 *     <span ref={ref} {...props} className="text-sm">
 *       {amount} {currency}
 *       <br />
 *       <br />
 *       {formattedFee}
 *     </span>
 *   ))}
 * </Plan.AdditionalFeeAmount>
 * ```
 */
export const AdditionalFeeAmount = React.forwardRef(({ children, asChild, className }, ref) => {
    return (_jsx(CoreAdditionalFeeAmount, { children: (additionalFeeAmountData) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, customElement: children, customElementProps: additionalFeeAmountData, className: className, "data-testid": PlanTestId.AdditionalFeeAmount, content: additionalFeeAmountData.formattedFee, children: _jsx("span", { children: additionalFeeAmountData.formattedFee }) })) }));
});
/**
 * Provides the child component with the recurrence data. It will be null for one-time plans.
 *
 * @component
 * @example
 * ```tsx
 * <Plan.Recurrence>
 *   {React.forwardRef(({ recurrence }, ref) => {
 *     if (!recurrence) return null;
 *
 *     return <span ref={ref} className="text-content-secondary" data-testid="plan-recurrence">
 *       Renews every {recurrence.count} {recurrence.period}(s)
 *     </span>
 *   })}
 * </Plan.Recurrence>
 * ```
 */
export const Recurrence = React.forwardRef(({ children }, ref) => (_jsx(CoreRecurrence, { children: (recurrenceData) => (_jsx(AsChildSlot, { ref: ref, asChild: true, customElement: children, customElementProps: recurrenceData })) })));
/**
 * Provides the child component with the duration data. It will be null for unlimited plans.
 *
 * @component
 * @example
 * ```tsx
 * <Plan.Duration>
 *   {React.forwardRef(({ duration }, ref) => {
 *     if (!duration) return <span>Valid until canceled</span>;
 *
 *     return <span ref={ref} className="text-sm" data-testid="plan-duration">
 *       Valid for {duration.count} {duration.period}(s)
 *     </span>
 *   })}
 * </Plan.Duration>
 * ```
 */
export const Duration = React.forwardRef(({ children }, ref) => (_jsx(CoreDuration, { children: (durationData) => (_jsx(AsChildSlot, { ref: ref, asChild: true, customElement: children, customElementProps: durationData })) })));
/**
 * Displays the free trial days.
 *
 * @component
 * @example
 * ```tsx
 * <Plan.FreeTrialDays>
 *   {React.forwardRef(({ freeTrialDays }, ref) => {
 *     return <span ref={ref} className="text-sm" data-testid="plan-free-trial-days">
 *       Free trial for {freeTrialDays} days
 *     </span>
 *   })}
 * </Plan.FreeTrialDays>
 * ```
 */
export const FreeTrialDays = React.forwardRef(({ children }, ref) => (_jsx(CoreFreeTrialDays, { children: (freeTrialDaysData) => (_jsx(AsChildSlot, { ref: ref, asChild: true, customElement: children, customElementProps: freeTrialDaysData, "data-testid": PlanTestId.FreeTrialDays })) })));
/**
 * Container for plan perks.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Plan.Perks className="flex flex-col gap-2">
 *   <Plan.PerksRepeater>
 *     <Plan.PerkItem className="text-sm" />
 *   </Plan.PerksRepeater>
 * </Plan.Perks>
 *
 * // asChild with react component
 * <Plan.Perks asChild>
 *   {React.forwardRef(({perks, ...props}, ref) => (
 *     <div ref={ref} {...props} className="flex flex-col gap-2">
 *       {perks.map((perk) => <span key={perk}> 🎉 {perk}</span>)}
 *     </div>
 *   ))}
 * </Plan.Perks>
 * ```
 */
export const Perks = React.forwardRef(({ children, asChild, className }, ref) => {
    return (_jsx(CorePerks, { children: (perksData) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, customElement: children, customElementProps: perksData, className: className, "data-testid": PlanTestId.Perks, children: _jsx("div", { children: children }) })) }));
});
/**
 * Repeater component that renders children for each perk with the perk item context.
 *
 * @component
 * @example
 * ```tsx
 * <Plan.PerksRepeater>
 *   <Plan.PerkItem className="text-sm" />
 * </Plan.PerksRepeater>
 * ```
 */
export const PerksRepeater = ({ children }) => {
    return (_jsx(CorePerksRepeater, { children: (perksData) => (_jsx(_Fragment, { children: perksData.perks.map((perk) => (_jsx(PerkItemContext.Provider, { value: { perk: perk }, children: children }, perk))) })) }));
};
/**
 * Displays the perk item. Must be used within a PerksRepeater.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Plan.PerkItem className="text-sm" />
 *
 * // asChild with primitive
 * <Plan.PerkItem asChild>
 *   <span className="text-sm" />
 * </Plan.PerkItem>
 *
 * // asChild with react component
 * <Plan.PerkItem asChild>
 *   {React.forwardRef(({perk, ...props}, ref) => (
 *     <span ref={ref} {...props} className="text-sm">
 *       {perk}
 *     </span>
 *   ))}
 * </Plan.PerkItem>
 * ```
 */
export const PerkItem = React.forwardRef(({ children, asChild, className }, ref) => (_jsx(CorePerkItem, { children: (perkItemData) => (_jsx(AsChildSlot, { ref: ref, asChild: asChild, customElement: children, customElementProps: perkItemData, className: className, "data-testid": PlanTestId.PerkItem, content: perkItemData.perk, children: _jsx("span", { children: perkItemData.perk }) })) })));
/**
 * Initiates the plan purchase flow.
 *
 * @component
 * @example
 * ```tsx
 * // Default usage
 * <Plan.Action.BuyNow className="btn-primary" label="Buy Now" loadingState="Processing..." />
 *
 * // With custom button
 * <Plan.Action.BuyNow className="btn-primary" label="Buy Now" loadingState="Processing..." asChild>
 *   <button>Buy Now</button>
 * </Plan.Action.BuyNow>
 *
 * // With custom button with forwardRef
 * <Plan.Action.BuyNow className="btn-primary" label="Buy Now" loadingState="Processing..." asChild>
 *   {({disabled, isLoading, onClick, ...props}, ref) => (
 *     <button ref={ref} {...props} disabled={disabled} onClick={onClick} className="btn-primary">
 *       {isLoading ? 'Processing...' : 'Buy Now'}
 *     </button>
 *   )}
 * </Plan.Action.BuyNow>
 * ```
 */
const ActionBuyNow = React.forwardRef((props, ref) => {
    const { planSignal } = useService(PlanServiceDefinition);
    return (_jsx(Commerce.Actions.BuyNow, { ...props, lineItems: [
            {
                quantity: 1,
                catalogReference: {
                    appId: PRICING_PLANS_APP_ID,
                    catalogItemId: planSignal.get()._id,
                    options: {
                        type: 'PLAN',
                        planOptions: {
                            pricingVariantId: planSignal.get().enhancedData.price.pricingVariantId,
                        },
                    },
                },
            },
        ], ref: ref, "data-testid": PlanTestId.ActionBuyNow }));
});
export const Action = {
    BuyNow: ActionBuyNow,
};
