// @wix/auto-patterns/dist/esm/hooks/useBulkActionToolbar.js
import React, { useCallback } from 'react';
import { useActionsSDK } from './useActionsSDK';
import { MultiBulkActionToolbar } from '@wix/patterns';
import { bulkDeleteAction, customAction, resolveAction, groupActions } from '../utils/actions';
export const useBulkActionToolbar = _ref => {
  let {
    config,
    collectionId
  } = _ref;
  const sdk = useActionsSDK({
    collectionId
  });
  const bulkActionToolbar = useCallback(param => {
    const buildAction = actionConfig => {
      let actionToResolve = null;
      const baseParams = {
        actionParams: param,
        sdk
      };
      switch (actionConfig.type) {
        case 'bulkDelete':
          actionToResolve = bulkDeleteAction({
            action: actionConfig,
            ...baseParams
          });
          break;
        case 'custom':
          actionToResolve = customAction({
            action: actionConfig,
            ...baseParams
          });
          break;
        default:
          return null;
      }
      if (!actionToResolve) {
        return null;
      }
      const resolvedAction = resolveAction(actionConfig, actionToResolve);
      const {
        icon,
        ...resolvedActionParams
      } = resolvedAction;
      return {
        prefixIcon: /*#__PURE__*/React.isValidElement(icon) ? icon : undefined,
        dataHook: actionConfig.id,
        ...resolvedActionParams
      };
    };
    const buildPrimaryActions = () => {
      if (!(config != null && config.primaryActions)) {
        return [];
      }
      return config.primaryActions.map((primaryItem, index) => {
        if (primaryItem.type === 'action') {
          return buildAction(primaryItem.action.item);
        }
        if (primaryItem.type === 'menu') {
          const actionGroups = groupActions(primaryItem.menu.items);
          const subItems = actionGroups.map(group => group.map(buildAction).filter(action => Boolean(action))).filter(group => group.length > 0);
          if (subItems.length === 0) {
            return null;
          }
          return {
            label: primaryItem.menu.label,
            subItems,
            dataHook: `bulk-multi-action-${index}`
          };
        }
        return null;
      }).filter(action => Boolean(action));
    };
    const primaryActionItems = buildPrimaryActions();
    const secondaryActionItems = config != null && config.secondaryActions ? groupActions(config.secondaryActions).map(group => group.map(buildAction).filter(action => Boolean(action))).filter(group => group.length > 0) : [];
    return /*#__PURE__*/React.createElement(MultiBulkActionToolbar, {
      primaryActionItems: primaryActionItems,
      secondaryActionItems: secondaryActionItems
    });
  }, [config, sdk]);
  if (!config) {
    return undefined;
  }
  const hasPrimaryActions = config.primaryActions && config.primaryActions.length > 0;
  const hasSecondaryActions = config.secondaryActions && config.secondaryActions.length > 0;
  if (!hasPrimaryActions && !hasSecondaryActions) {
    return undefined;
  }
  return bulkActionToolbar;
};
