// @wix/auto-patterns/dist/esm/hooks/useBaseGridFeatures.js
import { useMemo } from 'react';
import { getImageUrl } from '../utils/media/getImageUrl';
import { useSchema } from '../providers';
export const useBaseGridFeatures = configuration => {
  const {
    item
  } = configuration;
  const schema = useSchema();
  const {
    titleFieldId,
    subtitleFieldId,
    imageFieldId,
    imagePlacement,
    cardContentMode: preset
  } = item;
  const visibleFieldIds = useMemo(() => [titleFieldId, subtitleFieldId, imageFieldId].filter(Boolean), [titleFieldId, subtitleFieldId, imageFieldId]);
  const renderItem = useMemo(() => gridItem => ({
    id: gridItem[schema.idField],
    title: `${gridItem[titleFieldId]}`,
    ...(subtitleFieldId ? {
      subtitle: `${gridItem[subtitleFieldId] ?? ''}`
    } : {}),
    ...(imageFieldId ? {
      backgroundImageUrl: getImageUrl(gridItem[imageFieldId])
    } : {})
  }), [titleFieldId, subtitleFieldId, imageFieldId, schema.idField]);
  return {
    imagePlacement,
    preset,
    renderItem,
    visibleFieldIds
  };
};
