// @wix/auto-patterns/dist/esm/hooks/useCommonCollectionFeatures.js
import React from 'react';
import { useEmptyStates } from './useEmptyStates';
import { ToolbarTitle } from '@wix/patterns';
import { Box, Text, TextButton } from '@wix/design-system';
import { useActionCell } from './useActionCell';
import { useBulkActionToolbar } from './useBulkActionToolbar';
export const useCommonCollectionFeatures = _ref => {
  var _toolbarTitle$subtitl, _toolbarTitle$subtitl2, _toolbarTitle$subtitl3, _toolbarTitle$subtitl4, _toolbarTitle$subtitl5, _toolbarTitle$subtitl6, _toolbarTitle$subtitl7;
  let {
    actionCell: actionCellConfig,
    bulkActionToolbar: bulkActionToolbarConfig,
    toolbarTitle,
    search,
    emptyState: emptyStateConfig,
    collection
  } = _ref;
  const {
    emptyState
  } = useEmptyStates(emptyStateConfig);
  const actionCell = useActionCell({
    config: actionCellConfig,
    collectionId: collection.collectionId
  });
  const bulkActionToolbar = useBulkActionToolbar({
    config: bulkActionToolbarConfig,
    collectionId: collection.collectionId
  });
  const title = toolbarTitle && /*#__PURE__*/React.createElement(ToolbarTitle, {
    title: toolbarTitle.title,
    subtitle: {
      text: ((_toolbarTitle$subtitl = toolbarTitle.subtitle) == null ? void 0 : _toolbarTitle$subtitl.text) ?? '',
      learnMore: (_toolbarTitle$subtitl2 = toolbarTitle.subtitle) == null ? void 0 : _toolbarTitle$subtitl2.learnMore,
      info: ((_toolbarTitle$subtitl3 = toolbarTitle.subtitle) == null ? void 0 : _toolbarTitle$subtitl3.info) && {
        content: /*#__PURE__*/React.createElement(Box, {
          direction: "vertical",
          gap: "6px"
        }, /*#__PURE__*/React.createElement(Text, {
          dataHook: "toolbar-title-subtitle-info-content",
          light: true,
          size: "small"
        }, (_toolbarTitle$subtitl4 = toolbarTitle.subtitle) == null || (_toolbarTitle$subtitl4 = _toolbarTitle$subtitl4.info) == null ? void 0 : _toolbarTitle$subtitl4.content.text), ((_toolbarTitle$subtitl5 = toolbarTitle.subtitle) == null || (_toolbarTitle$subtitl5 = _toolbarTitle$subtitl5.info) == null ? void 0 : _toolbarTitle$subtitl5.content.link) && /*#__PURE__*/React.createElement(TextButton, {
          dataHook: "toolbar-title-subtitle-info-link",
          size: "small",
          underline: "always",
          skin: "light",
          as: "a",
          rel: "noopener",
          target: "_blank",
          href: (_toolbarTitle$subtitl6 = toolbarTitle.subtitle) == null || (_toolbarTitle$subtitl6 = _toolbarTitle$subtitl6.info) == null ? void 0 : _toolbarTitle$subtitl6.content.link.url
        }, (_toolbarTitle$subtitl7 = toolbarTitle.subtitle) == null || (_toolbarTitle$subtitl7 = _toolbarTitle$subtitl7.info) == null ? void 0 : _toolbarTitle$subtitl7.content.link.label))
      }
    },
    itemsLimit: toolbarTitle.itemsLimit,
    showTotal: toolbarTitle.showTotal
  });
  return {
    search: (search == null ? void 0 : search.shown) !== false,
    title,
    emptyState,
    actionCell,
    bulkActionToolbar
  };
};
