// @wix/auto-patterns/dist/esm/hooks/useCollectionPageActions.js
import React, { useCallback, useMemo } from 'react';
import { useActionsSDK } from './useActionsSDK';
import { createAction, customAction, resolveAction, groupActions } from '../utils/actions';
import { PrimaryActions, SecondaryActions } from '@wix/patterns';
export const useCollectionPageActions = _ref => {
  let {
    collectionPageActions
  } = _ref;
  const sdk = useActionsSDK({
    collectionId: ''
  });
  const buildCollectionPageActions = useCallback(() => {
    if (!collectionPageActions) {
      return;
    }
    const buildAction = actionConfig => {
      let actionToResolve = null;
      const baseParams = {
        actionParams: {
          collectionId: actionConfig.collection.collectionId
        },
        sdk
      };
      switch (actionConfig.type) {
        case 'create':
          actionToResolve = createAction({
            action: actionConfig,
            ...baseParams
          });
          break;
        case 'custom':
          actionToResolve = customAction({
            action: actionConfig,
            ...baseParams
          });
          break;
        default:
          return null;
      }
      if (!actionToResolve) {
        return null;
      }
      const resolvedAction = resolveAction(actionConfig, actionToResolve);
      const {
        tooltip,
        icon,
        ...resolvedActionProps
      } = resolvedAction;
      return {
        ...resolvedActionProps,
        prefixIcon: icon,
        disabledDescription: resolvedAction.tooltip
      };
    };
    const handleActionsType = actions => {
      if (actions.type === 'action') {
        return buildAction(actions.action.item);
      }
      if (actions.type === 'menu') {
        const actionGroups = groupActions(actions.menu.items);
        const subItems = actionGroups.map(group => group.map(buildAction).filter(action => action !== null)).filter(group => group.length > 0);
        if (!subItems.length) {
          return null;
        }
        return {
          label: actions.menu.label,
          subItems
        };
      }
      return null;
    };
    const primaryActions = (collectionPageActions == null ? void 0 : collectionPageActions.primaryActions) && handleActionsType(collectionPageActions.primaryActions);
    const secondaryActions = (collectionPageActions == null ? void 0 : collectionPageActions.secondaryActions) && handleActionsType(collectionPageActions.secondaryActions);
    if (!primaryActions && !secondaryActions) {
      return;
    }
    return {
      primaryActions: primaryActions ? /*#__PURE__*/React.createElement(PrimaryActions, primaryActions) : undefined,
      secondaryActions: secondaryActions ? /*#__PURE__*/React.createElement(SecondaryActions, secondaryActions) : undefined
    };
  }, [collectionPageActions, sdk]);
  const actions = useMemo(() => {
    return {
      primaryActions: undefined,
      secondaryActions: undefined
    };
  }, []);
  if (!(collectionPageActions != null && collectionPageActions.primaryActions) && !(collectionPageActions != null && collectionPageActions.secondaryActions)) {
    return {
      ...actions
    };
  }
  return buildCollectionPageActions() ?? {
    ...actions
  };
};
