// @wix/auto-patterns/dist/esm/hooks/useEntityPageMoreActions.js
import { customAction, resolveAction, groupActions } from '../utils/actions';
import { useCallback } from 'react';
export function useEntityPageMoreActions(moreActions, entity, sdk, form) {
  const buildEntityPageActions = useCallback(() => {
    if (!moreActions) {
      return;
    }
    const buildAction = actionConfig => {
      let actionToResolve = null;
      const baseParams = {
        actionParams: {
          entity,
          form
        },
        sdk
      };
      switch (actionConfig.type) {
        case 'custom':
          actionToResolve = customAction({
            action: actionConfig,
            ...baseParams
          });
          break;
        default:
          return null;
      }
      if (!actionToResolve) {
        return null;
      }
      const resolved = resolveAction(actionConfig, actionToResolve);
      return {
        onClick: resolved.onClick,
        text: resolved.label,
        prefixIcon: resolved.icon,
        disabled: resolved.disabled
      };
    };
    const actionGroups = groupActions(moreActions);
    return actionGroups.map(group => group.map(buildAction).filter(action => Boolean(action))).filter(group => group.length > 0);
  }, [moreActions, sdk, entity, form]);
  return buildEntityPageActions() ?? [];
}
