// @wix/auto-patterns/dist/esm/hooks/useEmptyStates.js
import _extends from "@babel/runtime/helpers/extends";
import React from 'react';
import { CollectionEmptyState } from '@wix/patterns';
import { usePatternsWizardOverridesContext } from '../providers';
export const useEmptyStates = emptyState => {
  var _emptyState$addNewCta, _overrides$emptyState, _emptyState$addNewCta2, _emptyState$addNewCta3, _emptyState$addNewCta4, _emptyState$customCta, _overrides$emptyState2, _emptyState$customCta2, _emptyState$image, _overrides$emptyState3, _emptyState$image2;
  const {
    title,
    subtitle
  } = emptyState || {};
  const overrides = usePatternsWizardOverridesContext();
  const onClick = emptyState != null && (_emptyState$addNewCta = emptyState.addNewCta) != null && _emptyState$addNewCta.id ? overrides == null || (_overrides$emptyState = overrides.emptyState) == null ? void 0 : _overrides$emptyState[emptyState == null || (_emptyState$addNewCta2 = emptyState.addNewCta) == null ? void 0 : _emptyState$addNewCta2.id] : undefined;
  const addNewCta = onClick ? {
    addNewCta: {
      ...(emptyState != null && (_emptyState$addNewCta3 = emptyState.addNewCta) != null && _emptyState$addNewCta3.text ? {
        text: emptyState == null || (_emptyState$addNewCta4 = emptyState.addNewCta) == null ? void 0 : _emptyState$addNewCta4.text
      } : {}),
      onClick
    }
  } : {};
  const _customCta = emptyState != null && (_emptyState$customCta = emptyState.customCta) != null && _emptyState$customCta.id ? overrides == null || (_overrides$emptyState2 = overrides.emptyState) == null ? void 0 : _overrides$emptyState2[emptyState == null || (_emptyState$customCta2 = emptyState.customCta) == null ? void 0 : _emptyState$customCta2.id] : undefined;
  const customCta = _customCta ? {
    customCta: _customCta()
  } : {};
  const _image = emptyState != null && (_emptyState$image = emptyState.image) != null && _emptyState$image.id ? overrides == null || (_overrides$emptyState3 = overrides.emptyState) == null ? void 0 : _overrides$emptyState3[emptyState == null || (_emptyState$image2 = emptyState.image) == null ? void 0 : _emptyState$image2.id] : undefined;
  const image = _image ? {
    image: _image()
  } : {};
  return {
    emptyState: /*#__PURE__*/React.createElement(CollectionEmptyState, _extends({
      title: title,
      subtitle: subtitle
    }, addNewCta, customCta, image))
  };
};
