// @wix/auto-patterns/dist/esm/hooks/useFilters.js
import _extends from "@babel/runtime/helpers/extends";
import React from 'react';
import { CollectionToolbarFilters } from '@wix/patterns';
import { useSchema } from '../providers/SchemaContext';
import { useSchemaRegistry } from '../providers';
import { StaticCollectionFilter, DynamicCollectionFilter } from '../components/filters';
import { createDateTimeFilter, createBooleanFilter, createNumberFilter, createEnumFilter, createReferenceFilter } from '../utils/filterCreators';
export const useFilters = configuration => {
  const schema = useSchema();
  const schemaRegistry = useSchemaRegistry();
  const filters = React.useMemo(() => {
    const currFilters = [];
    configuration == null || configuration.items.forEach(filter => {
      const {
        id,
        fieldId
      } = filter;
      const fieldDefinition = schema.fields[fieldId];
      if (!fieldDefinition) {
        return;
      }
      const label = filter.displayName || fieldDefinition.displayName;
      const {
        openByDefault,
        tagLabel,
        sectionTitle
      } = filter;
      const baseParams = {
        id,
        fieldId,
        commonFilterProps: {
          sectionTitle,
          initiallyOpen: openByDefault,
          accordionItemProps: {
            label
          },
          toolbarItemProps: {
            label
          },
          toolbarTagProps: {
            label: tagLabel
          }
        }
      };
      switch (fieldDefinition.type) {
        case 'DATE':
        case 'DATETIME':
          currFilters.push(createDateTimeFilter({
            ...baseParams,
            config: filter.dateConfig
          }));
          break;
        case 'BOOLEAN':
          currFilters.push(createBooleanFilter({
            ...baseParams,
            config: filter.booleanConfig
          }));
          break;
        case 'NUMBER':
          currFilters.push(createNumberFilter({
            ...baseParams,
            config: filter.numberConfig
          }));
          break;
        case 'REFERENCE':
          {
            const {
              referencedCollectionId
            } = fieldDefinition.referenceMetadata;
            const referencedSchema = schemaRegistry.getSchema(referencedCollectionId);
            if (referencedSchema) {
              currFilters.push(createReferenceFilter({
                ...baseParams,
                config: filter.dynamicOptionsConfig,
                additionalParams: {
                  referencedSchema
                }
              }));
            }
            break;
          }
        default:
          if (filter != null && filter.enumConfig) {
            currFilters.push(createEnumFilter({
              ...baseParams,
              config: filter.enumConfig
            }));
          }
          break;
      }
    });
    return currFilters;
  }, [configuration == null ? void 0 : configuration.items, schema.fields, schemaRegistry]);
  const filterFieldMapping = React.useMemo(() => {
    return configuration == null ? void 0 : configuration.items.reduce((acc, filter) => {
      acc[filter.id] = filter;
      return acc;
    }, {});
  }, [configuration == null ? void 0 : configuration.items]);
  const collectionToolbarFiltersProps = React.useMemo(() => ({
    inline: (configuration == null ? void 0 : configuration.maxInlineFilters) || 0,
    panelTitle: configuration == null ? void 0 : configuration.panelTitle
  }), [configuration]);
  const filtersData = filters.reduce((acc, filter) => {
    acc.filters[filter.key] = filter.filter;
    acc.components.push(filter.collectionData ? /*#__PURE__*/React.createElement(StaticCollectionFilter, _extends({
      filter: filter.filter,
      key: filter.key,
      component: filter.component,
      collectionData: filter.collectionData
    }, filter.props)) : filter.collectionId ? /*#__PURE__*/React.createElement(DynamicCollectionFilter, _extends({
      filter: filter.filter,
      key: filter.key,
      component: filter.component,
      collectionId: filter.collectionId,
      componentName: filter.props.componentName
    }, filter.props)) : /*#__PURE__*/React.createElement(filter.component, _extends({
      key: filter.key,
      filter: filter.filter
    }, filter.props)));
    return acc;
  }, {
    filters: {},
    components: [],
    collectionToolbarFiltersProps
  });
  return {
    filterFieldMapping,
    filtersObject: filtersData.filters,
    filterComponent: filtersData.components.length > 0 ? /*#__PURE__*/React.createElement(CollectionToolbarFilters, collectionToolbarFiltersProps, filtersData.components) : undefined
  };
};
