// @wix/auto-patterns/dist/esm/hooks/useGridFeatures.js
import { useFetchData } from './useFetchData';
import { useGridCollection } from '@wix/patterns';
import { useCommonCollectionFeatures } from './useCommonCollectionFeatures';
import { useFilters } from './useFilters';
import { useSchema } from '../providers/SchemaContext';
import { useAutoPatternsOptimisticActions } from './useAutoPatternsOptimisticActions';
import { useBaseGridFeatures } from './useBaseGridFeatures';
export const useGridFeatures = configuration => {
  const {
    collectionId,
    limit,
    reflectQueryInUrl,
    selectAllScope,
    selectionUpdateMode
  } = configuration.collection;
  const {
    renderItem,
    imagePlacement,
    preset,
    visibleFieldIds
  } = useBaseGridFeatures(configuration);
  const {
    filterComponent,
    filtersObject,
    filterFieldMapping
  } = useFilters(configuration.filters);
  const {
    fetchData
  } = useFetchData({
    searchableFieldIds: visibleFieldIds,
    filterFieldMapping
  });
  const schema = useSchema();
  const state = useGridCollection({
    queryName: `${collectionId}`,
    itemKey: gridItem => gridItem[schema.idField],
    itemName: gridItem => gridItem[schema.displayField],
    filters: filtersObject,
    toExtendedFields: gridItem => gridItem.extendedFields,
    fetchErrorMessage: _ref => {
      let {
        err
      } = _ref;
      return String(err);
    },
    fetchData,
    limit,
    persistQueryToUrl: reflectQueryInUrl,
    disableAutoSelectAllCount: selectAllScope === 'page',
    selectionConsistencyMode: selectionUpdateMode
  });
  const optimisticActions = useAutoPatternsOptimisticActions(collectionId, state.collection);
  const commonFeatures = useCommonCollectionFeatures({
    ...configuration,
    optimisticActions
  });
  return {
    ...commonFeatures,
    state,
    imagePlacement,
    preset,
    optimisticActions,
    renderItem,
    filters: filterComponent
  };
};
