// @wix/auto-patterns/dist/esm/hooks/useTableFeatures.js
import { useFetchData } from './useFetchData';
import { useTableCollection } from '@wix/patterns';
import { useCommonCollectionFeatures } from './useCommonCollectionFeatures';
import { useFilters } from './useFilters';
import { useSchema } from '../providers/SchemaContext';
import { useAutoPatternsOptimisticActions } from './useAutoPatternsOptimisticActions';
import { useBaseTableFeatures } from './useBaseTableFeatures';
export const useTableFeatures = configuration => {
  const {
    collectionId,
    limit,
    reflectQueryInUrl,
    selectAllScope,
    selectionUpdateMode
  } = configuration.collection;
  const {
    filterComponent,
    filtersObject,
    filterFieldMapping
  } = useFilters(configuration.filters);
  const {
    customColumns,
    visibleFieldIds,
    columns,
    onRowClick,
    stickyColumns,
    stickySelectionColumn
  } = useBaseTableFeatures(configuration);
  const {
    fetchData
  } = useFetchData({
    searchableFieldIds: visibleFieldIds,
    filterFieldMapping
  });
  const schema = useSchema();
  const state = useTableCollection({
    queryName: `${collectionId}`,
    itemKey: item => item[schema.idField],
    itemName: item => item[schema.displayField],
    filters: filtersObject,
    toExtendedFields: item => item.extendedFields,
    fetchErrorMessage: _ref => {
      let {
        err
      } = _ref;
      return String(err);
    },
    fetchData,
    limit,
    persistQueryToUrl: reflectQueryInUrl,
    disableAutoSelectAllCount: selectAllScope === 'page',
    selectionConsistencyMode: selectionUpdateMode
  });
  const optimisticActions = useAutoPatternsOptimisticActions(collectionId, state.collection);
  const commonFeatures = useCommonCollectionFeatures({
    ...configuration,
    optimisticActions
  });
  return {
    ...commonFeatures,
    state,
    columns,
    filters: filterComponent,
    optimisticActions,
    customColumns,
    onRowClick,
    stickyColumns,
    stickySelectionColumn
  };
};
