// @wix/auto-patterns/dist/esm/hooks/useTableGridSwitchFeatures.js
import { useFetchData } from './useFetchData';
import { useTableGridSwitchCollection } from '@wix/patterns';
import { useCommonCollectionFeatures } from './useCommonCollectionFeatures';
import { useFilters } from './useFilters';
import { useSchema } from '../providers/SchemaContext';
import { useAutoPatternsOptimisticActions } from './useAutoPatternsOptimisticActions';
import { useBaseGridFeatures } from './useBaseGridFeatures';
import { useBaseTableFeatures } from './useBaseTableFeatures';
export const useTableGridSwitchFeatures = config => {
  const schema = useSchema();
  const {
    collectionId,
    limit,
    reflectQueryInUrl,
    selectAllScope,
    selectionUpdateMode
  } = config.collection;
  const {
    filterComponent,
    filtersObject,
    filterFieldMapping
  } = useFilters(config.filters);
  const {
    preset,
    imagePlacement,
    visibleFieldIds: gridVisibleFieldIds,
    renderItem
  } = useBaseGridFeatures(config);
  const {
    customColumns,
    visibleFieldIds: tableVisibleFieldIds,
    columns: tableColumns,
    onRowClick,
    stickyColumns,
    stickySelectionColumn
  } = useBaseTableFeatures(config);
  const {
    fetchData: tableFetchData
  } = useFetchData({
    searchableFieldIds: Array.from(new Set([...tableVisibleFieldIds, ...gridVisibleFieldIds])),
    filterFieldMapping
  });
  const state = useTableGridSwitchCollection({
    queryName: `${collectionId}`,
    itemKey: item => item[schema.idField],
    itemName: item => item[schema.displayField],
    filters: filtersObject,
    toExtendedFields: item => item.extendedFields,
    fetchErrorMessage: _ref => {
      let {
        err
      } = _ref;
      return String(err);
    },
    fetchData: tableFetchData,
    limit,
    persistQueryToUrl: reflectQueryInUrl,
    disableAutoSelectAllCount: selectAllScope === 'page',
    selectionConsistencyMode: selectionUpdateMode
  });
  const optimisticActions = useAutoPatternsOptimisticActions(collectionId, state.collection);
  const commonFeatures = useCommonCollectionFeatures({
    ...config,
    optimisticActions
  });
  return {
    ...commonFeatures,
    state,
    columns: tableColumns,
    filters: filterComponent,
    optimisticActions,
    customColumns,
    onRowClick,
    stickyColumns,
    stickySelectionColumn,
    horizontalScroll: true,
    // Grid specific features
    imagePlacement,
    preset,
    renderItem
  };
};
