// @wix/ricos/dist/cjs/plugin-ai/ai-content-modal/ai-content-modal.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var ai_content_modal_exports = {};
__export(ai_content_modal_exports, {
  AiContentModal: () => AiContentModal
});
module.exports = __toCommonJS(ai_content_modal_exports);
var import_design_system = require("@wix/design-system");
var import_wix_ui_icons_common = require("@wix/wix-ui-icons-common");
var import_classnames = __toESM(require("classnames"));
var import_react = __toESM(require("react"));
var import_ricos_context = require("./../../ricos-context/index.js");
var import_uncreachable = require("./../../_shared/uncreachable.js");
var import_ui_components = require("./../../ui-components/index.js");
var import_data_hooks = require("../data-hooks.js");
var import_error_message = require("../error-message/error-message.js");
var import_types = require("../types.js");
var import_get_prompt_bi = require("../utils/get-prompt-bi.js");
var import_get_selection_as_html = require("../utils/get-selection-as-html.js");
var import_ai_content_modal_scss = __toESM(require("./ai-content-modal-scss.js"));
var import_disclaimer = require("./disclaimer/disclaimer.js");
var import_feed = require("./feed/feed.js");
var import_no_selection_message = require("./no-selection-message/no-selection-message.js");
var import_prompts_list = require("./prompts-list.js");
var import_textual_prompt_input_area = require("./textual-prompt-input-area/textual-prompt-input-area.js");
var import_use_ai_modal_api = require("./use-ai-modal-api.js");
const PROMPT_CHAR_LIMIT = 500;
const AiContentModal = ({
  closeModal,
  openDiscardModal,
  config,
  zIndex,
  modalSessionId,
  initialPromptOptions,
  feedUserEntryFormatter
}) => {
  const { t } = (0, import_react.useContext)(import_ricos_context.RicosContext);
  const { adapter } = (0, import_react.useContext)(import_ricos_context.EditorContext);
  const inputRef = import_react.default.useRef(null);
  const containerEl = (0, import_ui_components.usePanelContainerElement)();
  const listRef = (0, import_react.useRef)(null);
  const selection = import_react.default.useMemo(() => (0, import_get_selection_as_html.getSelectionAsHTML)(adapter.tiptapEditor), []);
  const resolvedZIndex = typeof zIndex === "number" ? zIndex : 5500;
  const pluginsEvents = (0, import_react.useContext)(import_ricos_context.PluginsEventsContext);
  const publishModalAction = import_react.default.useMemo(() => {
    let counter = 0;
    return (actionName, value) => {
      counter++;
      pluginsEvents.publishModalAction({
        pluginId: import_types.AI_TYPE,
        modalId: import_types.AI_MODAL_ID,
        modalSessionId,
        actionName,
        value,
        actionNumber: counter
      });
    };
  }, [pluginsEvents.publishModalAction, modalSessionId]);
  const {
    submitPrompt,
    setFreePromptValue,
    setSelectedPromptIndex,
    triggerFreePrompt,
    cancelPrompt,
    retryPrompt,
    goBack,
    textualPrompt,
    promptsList,
    state,
    getResultActions,
    tryCloseModal,
    isDiscardModalOpen
  } = (0, import_use_ai_modal_api.useAiModalApi)({
    selection: selection || "",
    ...config,
    closeModal,
    openDiscardModal,
    modalRef: { current: containerEl },
    inputRef,
    listRef,
    publishModalAction,
    initialPromptOptions
  });
  const closeOnClickOutside = import_react.default.useCallback(() => {
    tryCloseModal();
    publishModalAction("close_on_click_outside");
  }, [tryCloseModal]);
  (0, import_ui_components.useClickOutside)({
    containerRef: { current: containerEl },
    onClickOutside: closeOnClickOutside,
    disable: isDiscardModalOpen
  });
  import_react.default.useEffect(() => {
    if (initialPromptOptions?.autoTrigger && state.type === "INITIAL") {
      submitPrompt(initialPromptOptions.prompt, selection);
    }
  }, []);
  return /* @__PURE__ */ import_react.default.createElement(
    import_ui_components.Panel,
    {
      closeButtonProps: {
        onClick: () => {
          closeModal();
          publishModalAction("close_on_button_click");
        }
      },
      width: 540,
      dataHook: import_data_hooks.AI_MODAL_ROOT,
      maxHeight: 456
    },
    /* @__PURE__ */ import_react.default.createElement(
      import_ui_components.PanelHeader,
      {
        title: /* @__PURE__ */ import_react.default.createElement(import_design_system.Box, { gap: "SP1" }, /* @__PURE__ */ import_react.default.createElement(import_design_system.Box, { color: "B10" }, /* @__PURE__ */ import_react.default.createElement(import_wix_ui_icons_common.SparklesFilled, null)), /* @__PURE__ */ import_react.default.createElement(import_design_system.Heading, { size: "medium" }, t("AIPlugin_Modal_Title")))
      }
    ),
    (() => {
      switch (state.type) {
        case "INITIAL": {
          return selection ? promptsList.length > 0 ? /* @__PURE__ */ import_react.default.createElement("div", { className: import_ai_content_modal_scss.default.content }, /* @__PURE__ */ import_react.default.createElement(
            import_prompts_list.PromptsList,
            {
              ref: listRef,
              visible: textualPrompt.length === 0,
              prompts: promptsList,
              highlightedIndex: state.selectedPromptIndex,
              onHighlight: setSelectedPromptIndex,
              onClick: (prompt) => {
                submitPrompt(prompt, selection);
                publishModalAction("prompt_selected", (0, import_get_prompt_bi.getPromptBiValue)(prompt));
              }
            }
          )) : null : /* @__PURE__ */ import_react.default.createElement(
            import_no_selection_message.NoSelectionMessage,
            {
              title: initialPromptOptions?.emptyStateTitle,
              text: initialPromptOptions?.emptyStateText
            }
          );
        }
        case "IN_PROGRESS": {
          return /* @__PURE__ */ import_react.default.createElement(
            "div",
            {
              className: (0, import_classnames.default)(import_ai_content_modal_scss.default.content, import_ai_content_modal_scss.default.contentScrollReverse),
              "data-hook": import_data_hooks.AI_MODAL_LOADING_INDICATOR
            },
            /* @__PURE__ */ import_react.default.createElement(
              import_feed.Feed,
              {
                entries: state.entries,
                zIndex: resolvedZIndex,
                getResultActions,
                modalRef: { current: containerEl },
                feedUserEntryFormatter,
                isLoading: true
              }
            )
          );
        }
        case "RESULTS": {
          return /* @__PURE__ */ import_react.default.createElement(
            "div",
            {
              className: (0, import_classnames.default)(import_ai_content_modal_scss.default.content, import_ai_content_modal_scss.default.contentScrollReverse),
              "data-hook": import_data_hooks.AI_MODAL_RESULTS_CONTENT
            },
            /* @__PURE__ */ import_react.default.createElement(
              import_feed.Feed,
              {
                entries: state.entries,
                zIndex: resolvedZIndex,
                getResultActions,
                modalRef: { current: containerEl },
                feedUserEntryFormatter
              }
            )
          );
        }
        case "ERROR": {
          return /* @__PURE__ */ import_react.default.createElement("div", { className: (0, import_classnames.default)(import_ai_content_modal_scss.default.content, import_ai_content_modal_scss.default.padded) }, /* @__PURE__ */ import_react.default.createElement(import_error_message.ErrorMessage, null));
        }
        default: {
          throw (0, import_uncreachable.unreachable)(state);
        }
      }
    })(),
    /* @__PURE__ */ import_react.default.createElement(
      import_design_system.SidePanel.Footer,
      {
        noPadding: true,
        showDivider: !(state.type === "INITIAL" && promptsList.length === 0) && textualPrompt.length === 0 || state.type !== "INITIAL" || !selection
      },
      /* @__PURE__ */ import_react.default.createElement(
        import_design_system.Box,
        {
          paddingTop: "SP3",
          paddingBottom: "SP2",
          paddingLeft: "SP4",
          paddingRight: "SP4",
          align: "right",
          className: import_ai_content_modal_scss.default.footer
        },
        (() => {
          switch (state.type) {
            case "ERROR": {
              return /* @__PURE__ */ import_react.default.createElement(import_design_system.Box, { gap: "SP2", paddingBottom: "SP1" }, /* @__PURE__ */ import_react.default.createElement(
                import_design_system.Button,
                {
                  dataHook: import_data_hooks.AI_MODAL_ERROR_CANCEL_BUTTON,
                  size: "small",
                  priority: "secondary",
                  onClick: () => {
                    goBack();
                    publishModalAction("back_from_error");
                  }
                },
                t("AIPlugin_Error_Button_GoBack")
              ), /* @__PURE__ */ import_react.default.createElement(
                import_design_system.Button,
                {
                  dataHook: import_data_hooks.AI_MODAL_ERROR_RETRY_BUTTON,
                  size: "small",
                  priority: "primary",
                  onClick: () => {
                    retryPrompt();
                    publishModalAction("retry_from_error");
                  }
                },
                t("AIPlugin_Error_Button_Retry")
              ));
            }
            default: {
              return /* @__PURE__ */ import_react.default.createElement(import_design_system.Box, { direction: "vertical", gap: "SP2", width: "100%" }, /* @__PURE__ */ import_react.default.createElement(
                import_textual_prompt_input_area.TextualPromptInputArea,
                {
                  busy: state.type === "IN_PROGRESS",
                  value: textualPrompt,
                  onChange: setFreePromptValue,
                  handleTextAreaEnter: state.type !== "INITIAL",
                  onSubmit: () => {
                    triggerFreePrompt();
                    publishModalAction("submit_free_prompt", textualPrompt);
                  },
                  onCancel: () => {
                    cancelPrompt();
                    publishModalAction("cancel_free_prompt");
                  },
                  ref: inputRef,
                  maxLength: PROMPT_CHAR_LIMIT,
                  zIndex: resolvedZIndex,
                  placeholder: state.type === "INITIAL" && initialPromptOptions?.placeholder ? initialPromptOptions.placeholder : void 0,
                  isSubmitButtonVisible: initialPromptOptions?.isSubmitButtonVisible
                }
              ), /* @__PURE__ */ import_react.default.createElement(import_disclaimer.Disclaimer, null));
            }
          }
        })()
      )
    )
  );
};
