// @wix/patterns/dist/cjs/components/CustomFields/CustomFieldsDisabledTooltip.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.CustomFieldsDisabledTooltip = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _DisabledTooltip = require("../DisabledTooltip");
var _react = _interopRequireDefault(require("react"));
var _mobxReactLite = require("mobx-react-lite");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CustomFields/CustomFieldsDisabledTooltip.tsx";
function _CustomFieldsDisabledTooltip(props) {
  const {
    children,
    disabled,
    content,
    ...rest
  } = props;
  const {
    key,
    props: componentProps,
    type: Component
  } = children;
  const {
    dataHook
  } = componentProps;
  return /*#__PURE__*/_react.default.createElement(_DisabledTooltip.DisabledTooltip, (0, _extends2.default)({
    dataHook: dataHook ? `${dataHook}-disabled-tooltip` : undefined,
    content: content,
    appendTo: "window",
    maxWidth: "226px"
  }, rest, {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 20,
      columnNumber: 5
    }
  }), /*#__PURE__*/_react.default.createElement(Component, (0, _extends2.default)({
    key: key
  }, componentProps, {
    disabled: disabled,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 27,
      columnNumber: 7
    }
  })));
}
const CustomFieldsDisabledTooltip = exports.CustomFieldsDisabledTooltip = (0, _mobxReactLite.observer)(_CustomFieldsDisabledTooltip);
