// @wix/patterns/dist/cjs/components/CustomFields/CustomFieldsDisabledHint.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.CustomFieldsDisabledHint = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _designSystem = require("@wix/design-system");
var _react = _interopRequireDefault(require("react"));
var _mobxReactLite = require("mobx-react-lite");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CustomFields/CustomFieldsDisabledHint.tsx";
function _CustomFieldsDisabledHint(props) {
  const {
    children,
    disabled,
    content
  } = props;
  const {
    key,
    props: componentProps,
    type: Component
  } = children;
  const {
    dataHook
  } = componentProps;
  return /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
    direction: "vertical",
    align: "center",
    gap: "SP1",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 16,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(Component, (0, _extends2.default)({
    key: key
  }, componentProps, {
    disabled: disabled,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 17,
      columnNumber: 7
    }
  })), disabled && /*#__PURE__*/_react.default.createElement(_designSystem.Text, {
    dataHook: dataHook ? `${dataHook}-disabled-hint` : undefined,
    size: "small",
    skin: "disabled",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 19,
      columnNumber: 9
    }
  }, content));
}
const CustomFieldsDisabledHint = exports.CustomFieldsDisabledHint = (0, _mobxReactLite.observer)(_CustomFieldsDisabledHint);
