// @wix/patterns/dist/cjs/components/CollectionToolbar/CollectionToolbar.st.css.js
"use strict";

exports.__esModule = true;
exports.vars = exports.style = exports.stVars = exports.st = exports.namespace = exports.layers = exports.keyframes = exports.cssStates = exports.containers = exports.classes = void 0;
var _stylableEsmRuntime = require("../../stylable-esm-runtime.js");
// @ts-nocheck

var _namespace_ = "CollectionToolbar3612629254";
var _style_ = _stylableEsmRuntime.classesRuntime.bind(null, _namespace_);
var cssStates = exports.cssStates = _stylableEsmRuntime.statesRuntime.bind(null, _namespace_);
var style = exports.style = _style_;
var st = exports.st = _style_;
var namespace = exports.namespace = _namespace_;
var classes = exports.classes = {
  "root": "CollectionToolbar3612629254__root",
  "left": "CollectionToolbar3612629254__left",
  "right": "CollectionToolbar3612629254__right",
  "fullWidth": "CollectionToolbar3612629254__fullWidth"
};
var keyframes = exports.keyframes = {};
var layers = exports.layers = {};
var containers = exports.containers = {};
var stVars = exports.stVars = {};
var vars = exports.vars = {};
