// @wix/patterns/dist/cjs/components/CollectionToolbar/CollectionToolbarSkeletonItem.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.CollectionToolbarSkeletonItem = CollectionToolbarSkeletonItem;
var _react = _interopRequireDefault(require("react"));
var _designSystem = require("@wix/design-system");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CollectionToolbar/CollectionToolbarSkeletonItem.tsx";
function CollectionToolbarSkeletonItem(props) {
  const {
    dataHook,
    width
  } = props;
  return /*#__PURE__*/_react.default.createElement(_designSystem.TableToolbar.ItemGroup, {
    dataHook: dataHook,
    position: "start",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 15,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.SkeletonGroup, {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 16,
      columnNumber: 7
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.SkeletonLine, {
    width: width,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 17,
      columnNumber: 9
    }
  })));
}
