// @wix/patterns/dist/cjs/components/ExportButton/ExportStepError.js
"use strict";

exports.__esModule = true;
exports.ExportStepError = void 0;
var _designSystem = require("@wix/design-system");
var _react = _interopRequireWildcard(require("react"));
var _mobxReactLite = require("mobx-react-lite");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/ExportButton/ExportStepError.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _ExportStepError({
  state,
  children
}) {
  const {
    counter
  } = state;
  return /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
    gap: "SP2",
    direction: "vertical",
    paddingBottom: "SP1",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 18,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.LinearProgressBar, {
    error: true,
    showProgressIndication: true,
    value: counter,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 19,
      columnNumber: 7
    }
  }), children && /*#__PURE__*/(0, _react.cloneElement)(children, {
    dataHook: 'export-error-message'
  }));
}
const ExportStepError = exports.ExportStepError = (0, _mobxReactLite.observer)(_ExportStepError);
