// @wix/patterns/dist/cjs/components/ExportButton/ExportModal.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.ExportModal = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _designSystem = require("@wix/design-system");
var _react = _interopRequireWildcard(require("react"));
var _ToolbarCollectionContext = require("../ToolbarCollectionContext");
var _bexCore = require("@wix/bex-core");
var _ExportButtonModalLayout = require("./ExportButtonModalLayout");
var _mobxReactLite = require("mobx-react-lite");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/ExportButton/ExportModal.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _ExportModal(props) {
  const table = (0, _ToolbarCollectionContext.useToolbarCollectionContext)();
  const [exportModalState] = (0, _react.useState)(() => table.exportState ?? (table.exportState = new _bexCore.ModalState()));
  return /*#__PURE__*/_react.default.createElement(_designSystem.Modal, {
    onRequestClose: exportModalState.close,
    isOpen: exportModalState.isOpen,
    dataHook: "export-modal",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 17,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(_ExportButtonModalLayout.ExportButtonModalLayout, (0, _extends2.default)({}, props, {
    exportModalState: exportModalState,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 22,
      columnNumber: 7
    }
  })));
}
const ExportModal = exports.ExportModal = (0, _mobxReactLite.observer)(_ExportModal);
