// @wix/patterns/dist/cjs/components/ExportButton/ExportModal.uni.driver.js
"use strict";

exports.__esModule = true;
exports.ExportModalUniDriver = ExportModalUniDriver;
exports.default = void 0;
var _unidriver = require("@wix/wix-ui-test-utils/unidriver");
var _unidriver2 = require("../../unidriver");
var _unidriver3 = require("@wix/design-system/dist/testkit/unidriver");
function ExportModalUniDriver(base, body) {
  const modalBase = () => body.$('.portal-export-modal');
  const footerBase = () => modalBase().$('[data-hook="export-modal-footer-text"]');
  const linearProgressBarBase = () => modalBase().$('[data-hook="export-linear-progress-bar"]');
  const getAbsoluteCounter = () => (0, _unidriver3.TextUniDriver)(linearProgressBarBase().$('[data-hook="progress-count"]'), body);
  const getTotalCounter = () => (0, _unidriver3.HeadingUniDriver)(linearProgressBarBase().$('[data-hook="progress-total"]'), body);
  const getExportRadioGroup = () => (0, _unidriver3.RadioGroupUniDriver)(modalBase().$('[data-hook="export-group"]'), body);
  const getModal = () => (0, _unidriver3.ModalUniDriver)(modalBase(), modalBase());
  const getModalLayout = () => (0, _unidriver3.CustomModalLayoutUniDriver)(modalBase().$('[data-hook="export-modal-layout"]'), body);
  const getFooterText = async () => (await (0, _unidriver3.TextUniDriver)(footerBase(), body)).getText();
  const footerExists = () => footerBase().exists();
  const getCsvMessageText = async () => (await (0, _unidriver3.TextUniDriver)(modalBase().$('[data-hook="export-csv-message"]'), body).getText()).replace('&nbsp;', ' ');
  return {
    wait: timeout => base.wait(timeout),
    ...(0, _unidriver2.baseUniDriverFactory)(base),
    getModal,
    /** Get RadioGroupDriver of export types list */
    getExportRadioGroup,
    getModalLayout,
    getCsvMessageText,
    waitModalClosed: timeout => (0, _unidriver.waitFor)(async () => !(await getModalLayout().exists()), timeout),
    /** Clicks export modal primary action button*/
    clickPrimaryAction: async () => (await getModalLayout().getPrimaryButtonDriver()).click(),
    /** Get export modal primary action text*/
    getPrimaryActionText: async () => (await getModalLayout().getPrimaryButtonDriver()).getButtonTextContent(),
    /** Get export modal secondary action text */
    getSecondaryButtonText: async () => (await getModalLayout().getSecondaryButtonDriver()).getButtonTextContent(),
    /** Clicks export modal secondary action button*/
    clickSecondaryAction: async () => (await getModalLayout().getSecondaryButtonDriver()).click(),
    /** Clicks export modal close button*/
    clickCloseButton: () => getModalLayout().clickCloseButton(),
    /** Get export modal title*/
    getModalTitle: () => getModalLayout().getTitleText(),
    /** Get export progress absolute counter*/
    getAbsoluteCounter: () => getAbsoluteCounter().getText(),
    /** Get export total counter*/
    getTotalCounter: () => getTotalCounter().getText(),
    /** Get text from error message without the space value that is added with the widows prop
    Tests are failing with this Hex valus, so it is replaced here */
    getErrorMessageText: async () => (await modalBase().$('[data-hook="export-error-message"]').text()).replace('\u00A0', ' '),
    getErrorMessageAction: async () => modalBase().$('[data-hook="export-error-message-action"]'),
    /** Get export modal footer text*/
    getFooterText,
    /** Indicates whether footer exists in the export modal */
    footerExists
  };
}
var _default = exports.default = ExportModalUniDriver;
