// @wix/patterns/dist/cjs/components/ExportButton/ExportButtonModalLayout.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.ExportButtonModalLayout = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _designSystem = require("@wix/design-system");
var _getExportModalStepProps = require("./getExportModalStepProps");
var _react = _interopRequireWildcard(require("react"));
var _mobxReactLite = require("mobx-react-lite");
var _ToolbarCollectionContext = require("../ToolbarCollectionContext");
var _bexCore = require("@wix/bex-core");
var _types = require("@wix/bex-utils/@wix/ambassador-fedinfra-exportservice-v1-export-async-job/types");
var _util = require("@wix/bex-core/util");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/ExportButton/ExportButtonModalLayout.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _ExportButtonModalLayout(props) {
  const {
    renderError,
    transformPlatformizedQuery,
    differentQueryEndpointThanCollection,
    customTotal,
    methodMetadata,
    methodSpec,
    fields,
    saveAs,
    exportModalState,
    arrayFieldDelimiter,
    exportModalFooter,
    exportModalTitle,
    onProgressModalTitle,
    exportToCSVMessage,
    ...rest
  } = props;
  const table = (0, _ToolbarCollectionContext.useToolbarCollectionContext)();
  const createExport = (0, _bexCore.useCreateExport)({
    transformPlatformizedQuery,
    differentQueryEndpointThanCollection,
    customTotal,
    reportBi: table.reportBi,
    exportModalState,
    createExportAsyncJobRequest: {
      methodMetadata,
      fields,
      arrayFieldDelimiter: arrayFieldDelimiter ? _types.ArrayFieldDelimiter[arrayFieldDelimiter] : undefined,
      methodSpec: methodSpec ? {
        ...methodSpec,
        requestQueryFieldNumber: methodSpec.requestQueryFieldNumber ? _types.QueryFieldNumber[methodSpec.requestQueryFieldNumber] : undefined
      } : undefined
    },
    saveAs
  });
  const [state] = (0, _react.useState)(() => createExport());
  const {
    onRequestClose,
    ...customModalLayoutProps
  } = (0, _getExportModalStepProps.getExportModalStepProps)({
    state,
    renderError,
    exportModalFooter,
    exportModalTitle,
    onProgressModalTitle,
    exportToCSVMessage
  });
  (0, _react.useEffect)(() => {
    if (onRequestClose) {
      return (0, _util.addEventListener)(exportModalState.events, 'requestClose', onRequestClose);
    }
    return undefined;
  }, [onRequestClose]);
  (0, _react.useEffect)(() => state.init(), []);
  return /*#__PURE__*/_react.default.createElement(_designSystem.CustomModalLayout, (0, _extends2.default)({
    width: "510px",
    dataHook: "export-modal-layout"
  }, rest, customModalLayoutProps, {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 89,
      columnNumber: 5
    }
  }));
}
const ExportButtonModalLayout = exports.ExportButtonModalLayout = (0, _mobxReactLite.observer)(_ExportButtonModalLayout);
