// @wix/patterns/dist/cjs/components/ExportButton/ExportStepProgress.js
"use strict";

exports.__esModule = true;
exports.ExportStepProgress = void 0;
var _bexCore = require("@wix/bex-core");
var _designSystem = require("@wix/design-system");
var _react = _interopRequireWildcard(require("react"));
var _mobxReactLite = require("mobx-react-lite");
var _mobx = require("mobx");
var _LinearAbsoluteProgressBar = require("./LinearAbsoluteProgressBar");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/ExportButton/ExportStepProgress.tsx"; // import { Collapse } from '../Collapse';
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _ExportStepProgress({
  state
}) {
  const {
    counter,
    totalForExport,
    absoluteCounter,
    translate: t
  } = state;
  const [collapse] = (0, _react.useState)(() => new _bexCore.TransitionState());
  (0, _react.useEffect)(() => {
    const disposers = [(0, _mobx.reaction)(() => state.iterationsCountPoll.pollCount > 7, show => {
      collapse.setShow(show);
    })];
    return () => {
      disposers.forEach(d => d());
    };
  }, []);
  return /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
    gap: "SP1",
    direction: "vertical",
    paddingTop: "SP1",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 36,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(_LinearAbsoluteProgressBar.LinearAbsoluteProgressBar, {
    value: counter,
    count: absoluteCounter,
    total: totalForExport,
    dataHook: "export-linear-progress-bar",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 37,
      columnNumber: 7
    }
  }), /*#__PURE__*/_react.default.createElement(_designSystem.Text, {
    widows: true,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 43,
      columnNumber: 7
    }
  }, t('cairo.export.exportingDescription')));
}
const ExportStepProgress = exports.ExportStepProgress = (0, _mobxReactLite.observer)(_ExportStepProgress);
