// @wix/metro-common-builders/dist/es/src/value-generator/validation-extractor.js
import { ValidationType } from '../domain';
import { toNumber } from './value-generator-utils';
var ValidationExtractor = /** @class */ (function () {
    function ValidationExtractor(validations) {
        this.validations = validations;
    }
    ValidationExtractor.prototype.getValidationOfType = function (validationType) {
        return this.validations.find(function (validation) { return validation.type === validationType; });
    };
    ValidationExtractor.prototype.getValidationValueOfType = function (validationType) {
        var _a;
        return (_a = this.getValidationOfType(validationType)) === null || _a === void 0 ? void 0 : _a.value;
    };
    ValidationExtractor.prototype.getNumberValidations = function () {
        var max = this.getValidationValueOfType(ValidationType.MAX);
        var min = this.getValidationValueOfType(ValidationType.MIN);
        var _a = this.getValidationValueOfType(ValidationType.DECIMAL_VALUE) || {}, lowerThanEqual = _a.lte, lowerThan = _a.lt, greaterThanEqual = _a.gte, greaterThan = _a.gt, maxScale = _a.maxScale;
        return {
            min: toNumber(min),
            max: toNumber(max),
            lowerThanEqual: toNumber(lowerThanEqual),
            greaterThanEqual: toNumber(greaterThanEqual),
            lowerThan: toNumber(lowerThan),
            greaterThan: toNumber(greaterThan),
            maxScale: toNumber(maxScale),
        };
    };
    ValidationExtractor.prototype.getRepeatedValidations = function () {
        var maxSize = this.getValidationValueOfType(ValidationType.MAX_SIZE);
        var minSize = this.getValidationValueOfType(ValidationType.MIN_SIZE);
        return { minSize: minSize, maxSize: maxSize };
    };
    ValidationExtractor.prototype.getStringValidations = function () {
        var maxLength = this.getValidationValueOfType(ValidationType.MAX_LENGTH);
        var minLength = this.getValidationValueOfType(ValidationType.MIN_LENGTH);
        var format = this.getValidationValueOfType(ValidationType.FORMAT);
        return { minLength: minLength, maxLength: maxLength, format: format };
    };
    return ValidationExtractor;
}());
export { ValidationExtractor };
