// @wix/metro-common-builders/dist/es/src/value-generator/value-generator-utils.js
var _a, _b;
import { Type } from '../domain';
import { FunctionNames, GeneratorType } from './constants';
export function applyScale(maxScale, num) {
    return maxScale ? parseFloat(num.toFixed(maxScale)) : num;
}
export function increment(maxScale, num) {
    return num + getAdditionOrSubtractionValue(maxScale);
}
export function decrement(maxScale, num) {
    return num - getAdditionOrSubtractionValue(maxScale);
}
export function getAdditionOrSubtractionValue(scale) {
    // if float - generate according to scale, eg, 0.01, 0.001... else return 1
    return scale ? parseFloat("0.".concat('0'.repeat(scale - 1), "1")) : 1;
}
export function padArray(array, minSize, fillValue) {
    return Object.assign(new Array(minSize).fill(fillValue), array);
}
export function toNumber(number) {
    return typeof number === 'string' ? parseFloat(number) : number;
}
export function generatorTypeToFunctionName(generatorType) {
    return GENERATOR_TYPE_TO_FUNCTION[generatorType];
}
var GENERATOR_TYPE_TO_FUNCTION = (_a = {},
    _a[GeneratorType.NUMBER] = FunctionNames.SOME_NUMBER,
    _a[GeneratorType.LONG] = FunctionNames.SOME_LONG,
    _a[GeneratorType.FLOAT] = FunctionNames.SOME_FLOAT,
    _a[GeneratorType.STRING] = FunctionNames.SOME_STRING,
    _a[GeneratorType.REPEATED] = FunctionNames.SOME_REPEATED,
    _a[GeneratorType.DATE] = FunctionNames.SOME_DATE,
    _a[GeneratorType.DURATION] = FunctionNames.SOME_DURATION,
    _a[GeneratorType.BYTES] = FunctionNames.SOME_BYTES,
    _a[GeneratorType.ANY] = FunctionNames.SOME_ANY,
    _a);
export function typeToGeneratorType(type) {
    return TYPE_TO_GENERATOR_TYPE[type];
}
var TYPE_TO_GENERATOR_TYPE = (_b = {},
    _b[Type.DOUBLE] = GeneratorType.NUMBER,
    _b[Type.FLOAT] = GeneratorType.FLOAT,
    _b[Type.INT32] = GeneratorType.NUMBER,
    _b[Type.UINT32] = GeneratorType.NUMBER,
    _b[Type.SINT32] = GeneratorType.NUMBER,
    _b[Type.FIXED32] = GeneratorType.FLOAT,
    _b[Type.SFIXED32] = GeneratorType.FLOAT,
    _b[Type.INT64] = GeneratorType.STRING,
    _b[Type.FIXED64] = GeneratorType.STRING,
    _b[Type.SFIXED64] = GeneratorType.STRING,
    _b[Type.SINT64] = GeneratorType.STRING,
    _b[Type.UINT64] = GeneratorType.STRING,
    _b[Type.STRING] = GeneratorType.STRING,
    _b.FIELD_MASK = GeneratorType.REPEATED,
    _b.LIST_VALUE = GeneratorType.REPEATED,
    _b[Type.BOOL] = GeneratorType.OTHER,
    _b[Type.BYTES] = GeneratorType.BYTES,
    _b.NULL = GeneratorType.OTHER,
    _b.DURATION = GeneratorType.DURATION,
    _b.DATE = GeneratorType.DATE,
    _b.ANY = GeneratorType.ANY,
    _b);
export function wrapperToBusinessSchemaType(protoType) {
    return WRAPPER_TO_BUSINESS_SCHEMA[protoType];
}
var WRAPPER_TO_BUSINESS_SCHEMA = {
    'google.protobuf.DoubleValue': Type.DOUBLE,
    'google.protobuf.FloatValue': Type.FLOAT,
    'google.protobuf.Int32Value': Type.INT32,
    'google.protobuf.UInt32Value': Type.UINT32,
    'google.protobuf.Int64Value': Type.INT64,
    'google.protobuf.UInt64Value': Type.UINT64,
    'google.protobuf.BoolValue': Type.BOOL,
    'google.protobuf.StringValue': Type.STRING,
    'google.protobuf.BytesValue': Type.BYTES,
    'google.protobuf.NullValue': 'NULL',
    'google.protobuf.Struct': 'ANY',
    // TODO - remove when rich content create V2 without copy/paste google types
    'wix.rich_content.v1.Struct': 'ANY',
    'wix.rich_content.v1.Value': 'ANY',
    'google.protobuf.Duration': 'DURATION',
    'google.protobuf.Timestamp': 'DATE',
    'google.protobuf.Value': 'ANY',
    'google.protobuf.ListValue': 'LIST_VALUE',
    'google.protobuf.FieldMask': 'FIELD_MASK',
};
