// @wix/auto_sdk_forms_interactive-form-sessions/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  ChunkTypeOriginal: () => ChunkType,
  RoleOriginal: () => Role,
  StyleOriginal: () => Style,
  TypeOriginal: () => Type,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  createInteractiveFormSession: () => createInteractiveFormSession2,
  createInteractiveFormSessionStreamed: () => createInteractiveFormSessionStreamed2,
  sendUserMessage: () => sendUserMessage2,
  sendUserMessageStreamed: () => sendUserMessageStreamed2
});
module.exports = __toCommonJS(meta_exports);

// src/forms-ai-v1-interactive-form-session-interactive-form-sessions.http.ts
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
function resolveWixFormsAiV1InteractiveFormSessionsServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/_api/form-ai-assistant",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/form-ai-assistant",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/form-ai-assistant",
        destPath: ""
      }
    ],
    "*.wixforms.com": [
      {
        srcPath: "/_api/form-ai-assistant",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/form-ai-assistant",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/form-ai-assistant",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/forms/ai/v1/chat-settings",
        destPath: "/v1/chat-settings"
      },
      {
        srcPath: "/forms/ai/v1/interactive-form-sessions",
        destPath: "/v1/interactive-form-sessions"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/form-ai-assistant",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_forms_interactive-form-sessions";
function createInteractiveFormSession(payload) {
  function __createInteractiveFormSession({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.ai.v1.interactive_form_session",
      method: "POST",
      methodFqn: "wix.forms.ai.v1.InteractiveFormSessionsService.CreateInteractiveFormSession",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsAiV1InteractiveFormSessionsServiceUrl({
        protoPath: "/v1/interactive-form-sessions",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [
            { path: "responseChunks.numberInputDetails.multipleOf" },
            {
              path: "responseChunks.numberInputDetails.rangeLimit.maxInclusiveValue"
            },
            {
              path: "responseChunks.numberInputDetails.rangeLimit.maxExclusiveValue"
            },
            {
              path: "responseChunks.numberInputDetails.rangeLimit.minInclusiveValue"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createInteractiveFormSession;
}
function createInteractiveFormSessionStreamed(payload) {
  function __createInteractiveFormSessionStreamed({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.ai.v1.interactive_form_session",
      method: "POST",
      methodFqn: "wix.forms.ai.v1.InteractiveFormSessionsService.CreateInteractiveFormSessionStreamed",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsAiV1InteractiveFormSessionsServiceUrl({
        protoPath: "/v1/interactive-form-sessions/create-streamed",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [
            { path: "responseChunk.numberInputDetails.multipleOf" },
            {
              path: "responseChunk.numberInputDetails.rangeLimit.maxInclusiveValue"
            },
            {
              path: "responseChunk.numberInputDetails.rangeLimit.maxExclusiveValue"
            },
            {
              path: "responseChunk.numberInputDetails.rangeLimit.minInclusiveValue"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createInteractiveFormSessionStreamed;
}
function sendUserMessage(payload) {
  function __sendUserMessage({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.ai.v1.interactive_form_session",
      method: "POST",
      methodFqn: "wix.forms.ai.v1.InteractiveFormSessionsService.SendUserMessage",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsAiV1InteractiveFormSessionsServiceUrl({
        protoPath: "/v1/interactive-form-sessions/{interactiveFormSessionId}/send-user-message",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [
            { path: "responseChunks.numberInputDetails.multipleOf" },
            {
              path: "responseChunks.numberInputDetails.rangeLimit.maxInclusiveValue"
            },
            {
              path: "responseChunks.numberInputDetails.rangeLimit.maxExclusiveValue"
            },
            {
              path: "responseChunks.numberInputDetails.rangeLimit.minInclusiveValue"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __sendUserMessage;
}
function sendUserMessageStreamed(payload) {
  function __sendUserMessageStreamed({ host }) {
    const metadata = {
      entityFqdn: "wix.forms.ai.v1.interactive_form_session",
      method: "POST",
      methodFqn: "wix.forms.ai.v1.InteractiveFormSessionsService.SendUserMessageStreamed",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixFormsAiV1InteractiveFormSessionsServiceUrl({
        protoPath: "/v1/interactive-form-sessions/{interactiveFormSessionId}/send-user-message-streamed",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [
            { path: "responseChunk.numberInputDetails.multipleOf" },
            {
              path: "responseChunk.numberInputDetails.rangeLimit.maxInclusiveValue"
            },
            {
              path: "responseChunk.numberInputDetails.rangeLimit.maxExclusiveValue"
            },
            {
              path: "responseChunk.numberInputDetails.rangeLimit.minInclusiveValue"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __sendUserMessageStreamed;
}

// src/forms-ai-v1-interactive-form-session-interactive-form-sessions.types.ts
var ChunkType = /* @__PURE__ */ ((ChunkType2) => {
  ChunkType2["UNKNOWN_CHUNK_TYPE"] = "UNKNOWN_CHUNK_TYPE";
  ChunkType2["TEXT"] = "TEXT";
  ChunkType2["TEXT_DATA"] = "TEXT_DATA";
  ChunkType2["MULTI_SELECT_INPUT"] = "MULTI_SELECT_INPUT";
  ChunkType2["NUMBER_INPUT"] = "NUMBER_INPUT";
  ChunkType2["SEPARATOR"] = "SEPARATOR";
  ChunkType2["SINGLE_SELECT_INPUT"] = "SINGLE_SELECT_INPUT";
  ChunkType2["ERROR"] = "ERROR";
  ChunkType2["SUBMISSION"] = "SUBMISSION";
  ChunkType2["IMPORTANT_TEXT"] = "IMPORTANT_TEXT";
  ChunkType2["DEBUG"] = "DEBUG";
  ChunkType2["END_OF_RESPONSE"] = "END_OF_RESPONSE";
  ChunkType2["FILE_UPLOAD"] = "FILE_UPLOAD";
  ChunkType2["SIGNATURE"] = "SIGNATURE";
  return ChunkType2;
})(ChunkType || {});
var Style = /* @__PURE__ */ ((Style2) => {
  Style2["UNKNOWN_STYLE"] = "UNKNOWN_STYLE";
  Style2["NORMAL"] = "NORMAL";
  Style2["BOLD"] = "BOLD";
  return Style2;
})(Style || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  Type2["PARAGRAPH"] = "PARAGRAPH";
  return Type2;
})(Type || {});
var Role = /* @__PURE__ */ ((Role2) => {
  Role2["UNKNOWN_ROLE"] = "UNKNOWN_ROLE";
  Role2["USER"] = "USER";
  Role2["ASSISTANT"] = "ASSISTANT";
  Role2["FUNCTION_CALL"] = "FUNCTION_CALL";
  Role2["FUNCTION_CALL_OUTPUT"] = "FUNCTION_CALL_OUTPUT";
  Role2["DEVELOPER"] = "DEVELOPER";
  return Role2;
})(Role || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/forms-ai-v1-interactive-form-session-interactive-form-sessions.meta.ts
function createInteractiveFormSession2() {
  const payload = {};
  const getRequestOptions = createInteractiveFormSession(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/interactive-form-sessions",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createInteractiveFormSessionStreamed2() {
  const payload = {};
  const getRequestOptions = createInteractiveFormSessionStreamed(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/interactive-form-sessions/create-streamed",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function sendUserMessage2() {
  const payload = {
    interactiveFormSessionId: ":interactiveFormSessionId"
  };
  const getRequestOptions = sendUserMessage(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/interactive-form-sessions/{interactiveFormSessionId}/send-user-message",
    pathParams: { interactiveFormSessionId: "interactiveFormSessionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function sendUserMessageStreamed2() {
  const payload = {
    interactiveFormSessionId: ":interactiveFormSessionId"
  };
  const getRequestOptions = sendUserMessageStreamed(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/interactive-form-sessions/{interactiveFormSessionId}/send-user-message-streamed",
    pathParams: { interactiveFormSessionId: "interactiveFormSessionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ChunkTypeOriginal,
  RoleOriginal,
  StyleOriginal,
  TypeOriginal,
  WebhookIdentityTypeOriginal,
  createInteractiveFormSession,
  createInteractiveFormSessionStreamed,
  sendUserMessage,
  sendUserMessageStreamed
});
