// @wix/patterns/dist/esm/state/EntityPage/EntityPageStateBIReporter.js
import { addEventListener, withoutDefaults } from '@wix/bex-core'; // withoutDefaults
import { action } from 'mobx';
import { cairoTryAgainClicked, cairoApiRequestDataLoadEnd, cairoPageCtaClicked, cairoPageDiscardChangesBeforeSave, loadStart, loadEnd, } from '@wix/bex-core/bi';
export class EntityPageStateBIReporter {
    constructor(params) {
        this.startTime = performance.now();
        this.entityPageState = params.entityPageState;
        this.reportBI = params.reportBI;
    }
    _getErrorType(e) {
        const isNetworkError = e?.code === 'ERR_NETWORK';
        return isNetworkError ? 'Network error' : 'Technical issue';
    }
    init() {
        const { events } = this.entityPageState;
        const disposers = [
            addEventListener(events, 'save', action(() => {
                events.once('saveError', (err) => {
                    this.reportSaveError(err, this.startTime);
                });
            })),
            (this.appLoaded = this.reportOnLoadStart(this.startTime)),
        ];
        return () => {
            disposers.forEach((disposer) => disposer());
        };
    }
    reportButtonClick({ ctaName, isValid, }) {
        this.reportBI(withoutDefaults(cairoPageCtaClicked)({
            ctaName,
            ctaType: 'Main',
            location: 'Entity page header',
            isValid,
        }));
    }
    reportSaveTryAgain() {
        this.reportBI(withoutDefaults(cairoTryAgainClicked)({
            actionName: 'Saving Entity',
            loaction: 'toast',
        }));
    }
    reportSaveError(e, startTime) {
        this.reportBI(withoutDefaults(cairoApiRequestDataLoadEnd)({
            errorType: this._getErrorType(e),
            duration: performance.now() - startTime,
        }));
    }
    reportOnBeforeUnload() {
        const values = this.entityPageState.form.getValues();
        const fields = Object.keys(values);
        this.reportBI(withoutDefaults(cairoPageDiscardChangesBeforeSave)({
            numOfFields: fields.length,
            numOfActions: fields.reduce((prev, key) => prev + +this.entityPageState.form.getFieldState(key)?.isDirty, 0),
        }));
    }
    reportOnLoadStart(startTime) {
        this.reportBI(withoutDefaults(loadStart)({
            ...this._commonDynamicBiParams(),
        }));
        return () => {
            this.reportOnLoadEnd(Math.round(performance.now() - startTime));
        };
    }
    reportOnLoadEnd(endTime) {
        this.reportBI(withoutDefaults(loadEnd)({
            ...this._commonDynamicBiParams(),
            loadingTime: endTime,
        }));
    }
    _commonDynamicBiParams() {
        return {
            url: window.location.href,
            route: window.location.pathname,
            routerUsage: this.entityPageState._withinRouter,
        };
    }
}
