// @wix/headless-blog/cjs/dist/react/core/Comments.js
import { jsx as _jsx } from "react/jsx-runtime";
import { useService } from '@wix/services-manager-react';
import React from 'react';
import { CommentsServiceDefinition } from '../../services/comments-service.js';
export const Comments = (props) => {
    const service = useService(CommentsServiceDefinition);
    return props.children({
        comments: service.getComments(),
        isEmpty: service.isEmpty,
        hasNextPage: service.hasNextPage,
        isLoading: service.isLoading,
        loadMore: service.loadMore,
        initialLoad: service.initialLoad,
    });
};
Comments.displayName = 'Comments.Comments (core)';
export const Sort = (props) => {
    const service = useService(CommentsServiceDefinition);
    const currentSort = service.sort.get();
    const sortOptions = [
        {
            fieldName: 'NEWEST_FIRST',
            label: 'Newest first',
        },
        {
            fieldName: 'OLDEST_FIRST',
            label: 'Oldest first',
        },
    ];
    return props.children({
        currentSort,
        sortOptions,
        setSort: service.setSort,
    });
};
Sort.displayName = 'Comments.Sort (core)';
export const CreateComment = (props) => {
    const service = useService(CommentsServiceDefinition);
    const { parentCommentId: parrentCommentIdOrNull, topCommentId } = props;
    const parentCommentId = parrentCommentIdOrNull ?? undefined;
    return props.children({
        createComment: (content) => {
            const contentAsRichContent = {
                richContent: {
                    nodes: [{ type: 'PARAGRAPH', nodes: [{ type: 'TEXT', textData: { text: content } }] }],
                },
            };
            if (parentCommentId && topCommentId) {
                return service.createReply(topCommentId, parentCommentId, contentAsRichContent);
            }
            return service.createComment(contentAsRichContent);
        },
        isLoading: service.isLoading(parentCommentId) === 'saving',
        error: service.getError(parentCommentId),
        clearError: () => service.clearError(parentCommentId),
    });
};
CreateComment.displayName = 'Comments.CreateComment (core)';
export const Comment = (props) => {
    const { commentId } = props;
    const service = useService(CommentsServiceDefinition);
    const parentComment = service.getComments().find((c) => c._id === commentId);
    const isLoading = service.isLoading(commentId) === 'more' || service.isLoading(commentId) === 'initial';
    const loadNextPage = React.useCallback(() => service.loadMoreReplies(commentId), [service, commentId]);
    return props.children({
        comment: service.getComment(commentId),
        replies: service.getComments(commentId),
        hasNextPage: service.hasNextPage(commentId),
        isLoading,
        loadNextPage,
        parentComment,
        deleteComment: () => service.deleteComment(commentId),
    });
};
Comment.displayName = 'Comment (core)';
export const TopLevelCommentContext = React.createContext(null);
export const TopLevelCommentRoot = (props) => {
    const { comment, children } = props;
    return (_jsx(TopLevelCommentContext.Provider, { value: comment, children: children }));
};
TopLevelCommentRoot.displayName = 'Comment.TopLevelCommentRoot (core)';
