// @wix/headless-blog/cjs/dist/react/core/Feed.js
import { jsx as _jsx } from "react/jsx-runtime";
import { useService } from '@wix/services-manager-react';
import React from 'react';
import { BlogFeedServiceDefinition } from '../../services/blog-feed-service.js';
/**
 * Core BlogFeed Root component that provides BlogFeed service context.
 * This is the service-connected component that should be wrapped by the public API.
 *
 * @component
 */
export const Root = React.forwardRef((props, ref) => {
    const { children } = props;
    return _jsx("div", { ref: ref, children: children });
});
Root.displayName = 'Blog.Feed.Root (core)';
/**
 * Core Posts component that provides posts data access
 */
export const Posts = (props) => {
    const service = useService(BlogFeedServiceDefinition);
    const posts = service.posts.get();
    const hasPosts = posts.length > 0;
    return props.children({
        posts,
        isLoading: service.isLoading.get(),
        isEmpty: service.isEmpty(),
        hasNextPage: service.hasNextPage(),
        loadNextPage: service.loadNextPage,
        hasPosts,
        totalPosts: service.totalPosts.get(),
    });
};
Posts.displayName = 'Blog.Feed.Posts (core)';
/**
 * Core LoadMore component for pagination functionality
 */
export const LoadMore = (props) => {
    const service = useService(BlogFeedServiceDefinition);
    return props.children({
        hasNextPage: service.hasNextPage(),
        isLoading: service.isLoading.get(),
        loadNextPage: service.loadNextPage,
    });
};
LoadMore.displayName = 'Blog.Feed.LoadMore (core)';
export const Sort = (props) => {
    const blogFeedService = useService(BlogFeedServiceDefinition);
    const currentSort = blogFeedService.sort.get();
    const sortOptions = [
        {
            fieldName: 'firstPublishedDate',
            order: 'DESC',
            label: 'Newest First',
        },
        {
            fieldName: 'firstPublishedDate',
            order: 'ASC',
            label: 'Oldest First',
        },
    ];
    return props.children({
        currentSort,
        sortOptions,
        setSort: (sort) => {
            blogFeedService.setSort(sort);
        },
    });
};
Sort.displayName = 'Blog.Feed.Sort/Core';
