// @wix/ricos/dist/cjs/plugin-ai/ai-image-modal/ai-image-modal.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var ai_image_modal_exports = {};
__export(ai_image_modal_exports, {
  AiImageModal: () => AiImageModal
});
module.exports = __toCommonJS(ai_image_modal_exports);
var import_design_system = require("@wix/design-system");
var import_wix_ui_icons_common = require("@wix/wix-ui-icons-common");
var import_react = __toESM(require("react"));
var import_ricos_context = require("./../../ricos-context/index.js");
var import_ui_components = require("./../../ui-components/index.js");
var import_data_hooks = require("../data-hooks.js");
var import_ts_utils = require("../ts-utils.js");
var import_types = require("../types.js");
var import_get_selection_as_text = require("../utils/get-selection-as-text.js");
var import_feed = require("./feed.js");
var import_footer = require("./footer.js");
var import_use_ai_image_modal_api = require("./use-ai-image-modal-api.js");
const AiImageModal = ({
  closeModal,
  openDiscardModal,
  config,
  modalSessionId,
  zIndex,
  convertSelectionToPrompt,
  insertActions
}) => {
  const { t } = import_react.default.useContext(import_ricos_context.RicosContext);
  const { adapter } = import_react.default.useContext(import_ricos_context.EditorContext);
  const footerRef = import_react.default.useRef(null);
  const selection = import_react.default.useMemo(() => (0, import_get_selection_as_text.getSelectionAsText)(adapter.tiptapEditor), []);
  const containerEl = (0, import_ui_components.usePanelContainerElement)();
  const resolvedZIndex = typeof zIndex === "number" ? zIndex : 5500;
  const pluginsEvents = import_react.default.useContext(import_ricos_context.PluginsEventsContext);
  const publishModalAction = import_react.default.useMemo(() => {
    let counter = 0;
    return (actionName, value, imageStyle) => {
      counter++;
      pluginsEvents.publishModalAction({
        pluginId: import_types.AI_TYPE,
        modalId: import_types.AI_IMAGE_MODAL_ID,
        modalSessionId,
        actionName,
        value,
        actionNumber: counter,
        initialSelection: selection,
        style: imageStyle
      });
    };
  }, [pluginsEvents.publishModalAction, modalSessionId]);
  const {
    state,
    goBack,
    retryPrompt,
    setPromptValue,
    setImageStyleValue,
    tryConvertingSelectionToPrompt,
    submitPrompt,
    cancelPrompt,
    insertImage,
    isConvertedPromptHelperVisible,
    closeOptimisedPromptHelper,
    tryCloseModal,
    isDiscardModalOpen
  } = (0, import_use_ai_image_modal_api.useAiImageModalApi)({
    ...config,
    selection,
    closeModal,
    openDiscardModal,
    focusInputArea: () => footerRef.current?.focus(),
    modalRef: containerEl,
    convertSelectionToPrompt,
    publishModalAction
  });
  const closeOnClickOutside = import_react.default.useCallback(() => {
    tryCloseModal();
    publishModalAction("close_on_click_outside");
  }, [tryCloseModal]);
  (0, import_ui_components.useClickOutside)({
    containerRef: { current: containerEl },
    onClickOutside: closeOnClickOutside,
    disable: isDiscardModalOpen
  });
  import_react.default.useEffect(() => {
    if ((0, import_ts_utils.isOfType)(["INITIAL"], state)) {
      setTimeout(() => {
        footerRef.current?.focus();
      }, 0);
    }
  }, [state.type]);
  import_react.default.useEffect(() => {
    if (convertSelectionToPrompt && state.type === "INITIALIZING") {
      tryConvertingSelectionToPrompt(selection);
    }
  }, []);
  return /* @__PURE__ */ import_react.default.createElement(
    import_ui_components.Panel,
    {
      closeButtonProps: { onClick: () => closeModal() },
      width: 540,
      dataHook: import_data_hooks.AI_IMAGE_MODAL_ROOT
    },
    /* @__PURE__ */ import_react.default.createElement(
      import_ui_components.PanelHeader,
      {
        title: /* @__PURE__ */ import_react.default.createElement(import_design_system.Box, { gap: "SP1" }, /* @__PURE__ */ import_react.default.createElement(import_design_system.Box, { color: "B10" }, /* @__PURE__ */ import_react.default.createElement(import_wix_ui_icons_common.SparklesFilled, null)), /* @__PURE__ */ import_react.default.createElement(import_design_system.Heading, { size: "medium" }, t("AIPlugin_Modal_ImageGeneratorTitle")))
      }
    ),
    /* @__PURE__ */ import_react.default.createElement(import_feed.Feed, { state, insertImage, insertActions }),
    /* @__PURE__ */ import_react.default.createElement(
      import_footer.Footer,
      {
        ref: footerRef,
        state,
        zIndex: resolvedZIndex,
        modalRef: containerEl,
        setPromptValue,
        cancelPrompt,
        setImageStyleValue,
        retryPrompt,
        goBack,
        submitPrompt,
        isConvertedPromptHelperVisible,
        closeOptimisedPromptHelper,
        publishModalAction
      }
    )
  );
};
