// @wix/fast-gallery-vibe/dist/cjs/components/GalleryWrapper.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.GalleryWrapper = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var React = _interopRequireWildcard(require("react"));
var _fastGalleryCore = require("@wix/fast-gallery-core");
var _clsx = require("clsx");
var _variants = require("./variants.js");
require("./gallery.css");
var _jsxFileName = "/home/builduser/work/d3ecddb38d0a3df1/packages/fast-gallery-packages/fast-gallery-vibe/dist/cjs/components/GalleryWrapper.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
const GalleryWrapper = exports.GalleryWrapper = /*#__PURE__*/React.forwardRef((props, ref) => {
  const {
    items = [],
    variant = 'grid',
    className,
    itemRenderer,
    emptyState,
    children,
    ...otherProps
  } = props;
  const hasItems = items.length > 0;

  // Convert items to gallery format
  const convertItemToGalleryItem = React.useCallback((item, index) => {
    return {
      id: item.id || item._id || `item-${index}`,
      index: index.toString(),
      mediaUrl: '',
      type: _fastGalleryCore.MediaTypes.IMAGE,
      metaData: {
        width: 1,
        // Minimal value for fast-gallery-core layout calculations
        height: 1,
        // Minimal value for fast-gallery-core layout calculations
        title: '',
        description: '',
        alt: '',
        link: {
          url: '',
          target: '_self'
        }
      }
    };
  }, []);
  const galleryItems = React.useMemo(() => items.map((item, index) => convertItemToGalleryItem(item, index)), [items, convertItemToGalleryItem]);

  // Create custom item renderer component
  const ItemRendererComponent = React.useMemo(() => {
    const Component = itemProps => {
      if (itemRenderer) {
        const originalItem = items[parseInt(itemProps.data.index, 10)];
        if (originalItem) {
          const rendered = itemRenderer(originalItem, parseInt(itemProps.data.index, 10));
          return /*#__PURE__*/React.createElement(React.Fragment, null, rendered);
        }
      }

      // Development warning if no itemRenderer is provided
      if (process.env.NODE_ENV === 'development' && !itemRenderer) {
        console.warn('GalleryWrapper: No itemRenderer provided. Gallery will render empty items. ' + 'Please provide an itemRenderer prop to render your content.');
      }
      return null;
    };
    return Component;
  }, [itemRenderer, items]);
  const customRenderers = React.useMemo(() => ({
    itemRenderer: ItemRendererComponent
  }), [ItemRendererComponent]);
  const finalClassName = (0, _clsx.clsx)((0, _variants.galleryVariants)({
    variant
  }), className);

  // Show empty state if no items
  if (!hasItems) {
    return emptyState || null;
  }
  return /*#__PURE__*/React.createElement("div", (0, _extends2.default)({
    ref: ref,
    className: finalClassName,
    "data-gallery-variant": variant
  }, otherProps, {
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 98,
      columnNumber: 5
    }
  }), /*#__PURE__*/React.createElement(_fastGalleryCore.Gallery, {
    dataList: galleryItems,
    customRenderers: customRenderers,
    options: {},
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 104,
      columnNumber: 7
    }
  }), children);
});
GalleryWrapper.displayName = 'GalleryWrapper';
