// @wix/ricos/dist/cjs/plugin-table/toolbar/modals/BorderSelectionDropdown.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var BorderSelectionDropdown_exports = {};
__export(BorderSelectionDropdown_exports, {
  BorderSelectionDropdown: () => BorderSelectionDropdown
});
module.exports = __toCommonJS(BorderSelectionDropdown_exports);
var import_wix_ui_icons_common = require("@wix/wix-ui-icons-common");
var import_react = __toESM(require("react"));
var import_ricos_context = require("./../../../ricos-context/index.js");
var import_modal_service_provider = require("./../../../ricos-modals/modal-service-provider.js");
var import_dropdown_modal = require("./../../../_shared/dropdown-modal/dropdown-modal.js");
var import_list_item_select = require("./../../../_shared/list-item-select/list-item-select.js");
var import_consts = require("../../consts.js");
var import_data_hooks = require("./data-hooks.js");
const BorderSelectionDropdown = ({ onClick }) => {
  const { t, isMobile } = (0, import_react.useContext)(import_ricos_context.RicosContext);
  const modalService = (0, import_modal_service_provider.useModalService)();
  return /* @__PURE__ */ import_react.default.createElement(
    import_dropdown_modal.DropdownModal,
    {
      onCloseButtonClick: () => modalService.closeModal(import_consts.TABLE_BUTTONS_MODALS_ID.BORDER),
      title: t("TablePlugin_Toolbar_BorderPanel_Header_Title"),
      options: [
        /* @__PURE__ */ import_react.default.createElement(
          import_list_item_select.ListItemSelect,
          {
            key: "border",
            dataHook: import_data_hooks.BORDER_COLOR_AROUND,
            prefix: isMobile ? /* @__PURE__ */ import_react.default.createElement(import_wix_ui_icons_common.SquareRatio, null) : /* @__PURE__ */ import_react.default.createElement(import_wix_ui_icons_common.SquareRatioSmall, null),
            title: t("TablePlugin_Toolbar_BorderPanel_OutsideBorder"),
            onSelect: () => onClick(import_consts.BORDER_TYPES.outsideBorders)
          }
        ),
        /* @__PURE__ */ import_react.default.createElement(
          import_list_item_select.ListItemSelect,
          {
            key: "outsideBorder",
            dataHook: import_data_hooks.BORDER_COLOR_ALL,
            prefix: isMobile ? /* @__PURE__ */ import_react.default.createElement(import_wix_ui_icons_common.AllBorders, null) : /* @__PURE__ */ import_react.default.createElement(import_wix_ui_icons_common.AllBordersSmall, null),
            title: t("TablePlugin_Toolbar_BorderPanel_AllBorders"),
            onSelect: () => onClick(import_consts.BORDER_TYPES.borders)
          }
        )
      ]
    }
  );
};
