// @wix/auto_sdk_pricing-plans_pricing-plans-settings/build/es/meta.mjs
// src/pricing-plans-settings-v1-pricing-plans-settings-pricing-plans-settings.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixPricingPlansSettingsV1PricingPlansSettingsServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/pricing-plans-settings",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_pricing-plans_pricing-plans-settings";
function getPricingPlansSettings(payload) {
  function __getPricingPlansSettings({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.settings.v1.pricing_plans_settings",
      method: "GET",
      methodFqn: "wix.pricing_plans.settings.v1.PricingPlansSettingsService.GetPricingPlansSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixPricingPlansSettingsV1PricingPlansSettingsServiceUrl({
        protoPath: "/v1/pricing-plans-settings",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "pricingPlansSettings.createdDate" },
            { path: "pricingPlansSettings.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getPricingPlansSettings;
}
function updatePricingPlansSettings(payload) {
  function __updatePricingPlansSettings({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "pricingPlansSettings.createdDate" },
          { path: "pricingPlansSettings.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.pricing_plans.settings.v1.pricing_plans_settings",
      method: "PATCH",
      methodFqn: "wix.pricing_plans.settings.v1.PricingPlansSettingsService.UpdatePricingPlansSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixPricingPlansSettingsV1PricingPlansSettingsServiceUrl({
        protoPath: "/v1/pricing-plans-settings",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "pricingPlansSettings.createdDate" },
            { path: "pricingPlansSettings.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updatePricingPlansSettings;
}

// src/pricing-plans-settings-v1-pricing-plans-settings-pricing-plans-settings.types.ts
var TaxableAddress = /* @__PURE__ */ ((TaxableAddress2) => {
  TaxableAddress2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  TaxableAddress2["BILLING"] = "BILLING";
  TaxableAddress2["BUSINESS"] = "BUSINESS";
  return TaxableAddress2;
})(TaxableAddress || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/pricing-plans-settings-v1-pricing-plans-settings-pricing-plans-settings.meta.ts
function getPricingPlansSettings2() {
  const payload = {};
  const getRequestOptions = getPricingPlansSettings(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/pricing-plans-settings",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updatePricingPlansSettings2() {
  const payload = {};
  const getRequestOptions = updatePricingPlansSettings(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/pricing-plans-settings",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  TaxableAddress as TaxableAddressOriginal,
  WebhookIdentityType as WebhookIdentityTypeOriginal,
  getPricingPlansSettings2 as getPricingPlansSettings,
  updatePricingPlansSettings2 as updatePricingPlansSettings
};
