// @wix/auto_sdk_events_schedule/build/internal/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  ScheduleStatus: () => ScheduleStatus,
  SortOrder: () => SortOrder,
  StateFilter: () => StateFilter,
  addScheduleItem: () => addScheduleItem4,
  deleteScheduleItem: () => deleteScheduleItem4,
  discardDraft: () => discardDraft4,
  getScheduleItem: () => getScheduleItem4,
  listScheduleItems: () => listScheduleItems4,
  publishDraft: () => publishDraft4,
  queryScheduleItems: () => queryScheduleItems4,
  rescheduleDraft: () => rescheduleDraft4,
  updateScheduleItem: () => updateScheduleItem4
});
module.exports = __toCommonJS(index_exports);

// src/events-schedule-v1-schedule-item-schedule.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/events-schedule-v1-schedule-item-schedule.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_duration = require("@wix/sdk-runtime/transformations/duration");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEventsScheduleScheduleManagementUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/wix-events-agenda",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    _: [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "editor-flow.wixapps.net": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/events/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "api._base_domain_": [
      {
        srcPath: "/events/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_events_schedule";
function listScheduleItems(payload) {
  function __listScheduleItems({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "startingFrom" }, { path: "startingBefore" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "GET",
      methodFqn: "wix.events.schedule.ScheduleManagement.ListScheduleItems",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleScheduleManagementUrl({
        protoPath: "/v1/schedule",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "items.createdDate" },
            { path: "items.updatedDate" },
            { path: "items.timeSlot.start" },
            { path: "items.timeSlot.end" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listScheduleItems;
}
function queryScheduleItems(payload) {
  function __queryScheduleItems({ host }) {
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "POST",
      methodFqn: "wix.events.schedule.ScheduleManagement.QueryScheduleItems",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleScheduleManagementUrl({
        protoPath: "/v1/schedule/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "items.createdDate" },
            { path: "items.updatedDate" },
            { path: "items.timeSlot.start" },
            { path: "items.timeSlot.end" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryScheduleItems;
}
function getScheduleItem(payload) {
  function __getScheduleItem({ host }) {
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "GET",
      methodFqn: "wix.events.schedule.ScheduleManagement.GetScheduleItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleScheduleManagementUrl({
        protoPath: "/v1/schedule/{itemId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "item.createdDate" },
            { path: "item.updatedDate" },
            { path: "item.timeSlot.start" },
            { path: "item.timeSlot.end" },
            { path: "draft.createdDate" },
            { path: "draft.updatedDate" },
            { path: "draft.timeSlot.start" },
            { path: "draft.timeSlot.end" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getScheduleItem;
}
function addScheduleItem(payload) {
  function __addScheduleItem({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "item.timeSlot.start" }, { path: "item.timeSlot.end" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "POST",
      methodFqn: "wix.events.schedule.ScheduleManagement.AddScheduleItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleScheduleManagementUrl({
        protoPath: "/v1/schedule/draft",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "item.createdDate" },
            { path: "item.updatedDate" },
            { path: "item.timeSlot.start" },
            { path: "item.timeSlot.end" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __addScheduleItem;
}
function updateScheduleItem(payload) {
  function __updateScheduleItem({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fields" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "item.timeSlot.start" }, { path: "item.timeSlot.end" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "PATCH",
      methodFqn: "wix.events.schedule.ScheduleManagement.UpdateScheduleItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleScheduleManagementUrl({
        protoPath: "/v1/schedule/draft/{itemId}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "item.createdDate" },
            { path: "item.updatedDate" },
            { path: "item.timeSlot.start" },
            { path: "item.timeSlot.end" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateScheduleItem;
}
function deleteScheduleItem(payload) {
  function __deleteScheduleItem({ host }) {
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "DELETE",
      methodFqn: "wix.events.schedule.ScheduleManagement.DeleteScheduleItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleScheduleManagementUrl({
        protoPath: "/v1/schedule/draft/items",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteScheduleItem;
}
function discardDraft(payload) {
  function __discardDraft({ host }) {
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "DELETE",
      methodFqn: "wix.events.schedule.ScheduleManagement.DiscardDraft",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleScheduleManagementUrl({
        protoPath: "/v1/schedule/draft",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __discardDraft;
}
function publishDraft(payload) {
  function __publishDraft({ host }) {
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "POST",
      methodFqn: "wix.events.schedule.ScheduleManagement.PublishDraft",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleScheduleManagementUrl({
        protoPath: "/v1/schedule/publish",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __publishDraft;
}
function rescheduleDraft(payload) {
  function __rescheduleDraft({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_duration.transformSDKDurationToRESTDuration,
        paths: [{ path: "timeSlotOffset" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "POST",
      methodFqn: "wix.events.schedule.ScheduleManagement.RescheduleDraft",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleScheduleManagementUrl({
        protoPath: "/v1/schedule/draft/reschedule",
        data: serializedData,
        host
      }),
      data: serializedData
    };
    return metadata;
  }
  return __rescheduleDraft;
}

// src/events-schedule-v1-schedule-item-schedule.universal.ts
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var ScheduleStatus = /* @__PURE__ */ ((ScheduleStatus2) => {
  ScheduleStatus2["SCHEDULED"] = "SCHEDULED";
  ScheduleStatus2["CANCELED"] = "CANCELED";
  return ScheduleStatus2;
})(ScheduleStatus || {});
var StateFilter = /* @__PURE__ */ ((StateFilter2) => {
  StateFilter2["PUBLISHED"] = "PUBLISHED";
  StateFilter2["DRAFT"] = "DRAFT";
  StateFilter2["VISIBLE"] = "VISIBLE";
  StateFilter2["HIDDEN"] = "HIDDEN";
  return StateFilter2;
})(StateFilter || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
async function listScheduleItems2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    eventId: options?.eventId,
    state: options?.state,
    startingFrom: options?.startingFrom,
    startingBefore: options?.startingBefore,
    offset: options?.offset,
    limit: options?.limit,
    facet: options?.facet,
    itemId: options?.itemId,
    tag: options?.tag,
    stageName: options?.stageName,
    paging: options?.paging
  });
  const reqOpts = listScheduleItems(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          eventId: "$[0].eventId",
          state: "$[0].state",
          startingFrom: "$[0].startingFrom",
          startingBefore: "$[0].startingBefore",
          offset: "$[0].offset",
          limit: "$[0].limit",
          facet: "$[0].facet",
          itemId: "$[0].itemId",
          tag: "$[0].tag",
          stageName: "$[0].stageName",
          paging: "$[0].paging"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryScheduleItems2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryScheduleItems(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [])
      );
      return {
        items: transformedData?.items,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function getScheduleItem2(itemId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    itemId,
    eventId: options?.eventId,
    includeDraft: options?.includeDraft
  });
  const reqOpts = getScheduleItem(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.item;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          itemId: "$[0]",
          eventId: "$[1].eventId",
          includeDraft: "$[1].includeDraft"
        },
        singleArgumentUnchanged: false
      },
      ["itemId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function addScheduleItem2(eventId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    eventId,
    item: options?.item
  });
  const reqOpts = addScheduleItem(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { eventId: "$[0]", item: "$[1].item" },
        singleArgumentUnchanged: false
      },
      ["eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateScheduleItem2(itemId, eventId, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    itemId,
    eventId,
    item: options?.item,
    fields: options?.fields
  });
  const reqOpts = updateScheduleItem(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          itemId: "$[0]",
          eventId: "$[1]",
          item: "$[2].item",
          fields: "$[2].fields"
        },
        singleArgumentUnchanged: false
      },
      ["itemId", "eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteScheduleItem2(eventId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    eventId,
    itemIds: options?.itemIds
  });
  const reqOpts = deleteScheduleItem(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { eventId: "$[0]", itemIds: "$[1].itemIds" },
        singleArgumentUnchanged: false
      },
      ["eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function discardDraft2(eventId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ eventId });
  const reqOpts = discardDraft(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { eventId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["eventId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function publishDraft2(eventId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ eventId });
  const reqOpts = publishDraft(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { eventId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["eventId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function rescheduleDraft2(eventId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    eventId,
    timeZoneId: options?.timeZoneId,
    timeSlotOffset: options?.timeSlotOffset
  });
  const reqOpts = rescheduleDraft(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          eventId: "$[0]",
          timeZoneId: "$[1].timeZoneId",
          timeSlotOffset: "$[1].timeSlotOffset"
        },
        singleArgumentUnchanged: false
      },
      ["eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/events-schedule-v1-schedule-item-schedule.public.ts
function listScheduleItems3(httpClient) {
  return (options) => listScheduleItems2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function queryScheduleItems3(httpClient) {
  return () => queryScheduleItems2(
    // @ts-ignore
    { httpClient }
  );
}
function getScheduleItem3(httpClient) {
  return (itemId, options) => getScheduleItem2(
    itemId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function addScheduleItem3(httpClient) {
  return (eventId, options) => addScheduleItem2(
    eventId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function updateScheduleItem3(httpClient) {
  return (itemId, eventId, options) => updateScheduleItem2(
    itemId,
    eventId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function deleteScheduleItem3(httpClient) {
  return (eventId, options) => deleteScheduleItem2(
    eventId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function discardDraft3(httpClient) {
  return (eventId) => discardDraft2(
    eventId,
    // @ts-ignore
    { httpClient }
  );
}
function publishDraft3(httpClient) {
  return (eventId) => publishDraft2(
    eventId,
    // @ts-ignore
    { httpClient }
  );
}
function rescheduleDraft3(httpClient) {
  return (eventId, options) => rescheduleDraft2(
    eventId,
    options,
    // @ts-ignore
    { httpClient }
  );
}

// src/events-schedule-v1-schedule-item-schedule.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var listScheduleItems4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(listScheduleItems3);
var queryScheduleItems4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(queryScheduleItems3);
var getScheduleItem4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getScheduleItem3);
var addScheduleItem4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(addScheduleItem3);
var updateScheduleItem4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(updateScheduleItem3);
var deleteScheduleItem4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(deleteScheduleItem3);
var discardDraft4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(discardDraft3);
var publishDraft4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(publishDraft3);
var rescheduleDraft4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(rescheduleDraft3);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ScheduleStatus,
  SortOrder,
  StateFilter,
  addScheduleItem,
  deleteScheduleItem,
  discardDraft,
  getScheduleItem,
  listScheduleItems,
  publishDraft,
  queryScheduleItems,
  rescheduleDraft,
  updateScheduleItem
});
