// @wix/auto_sdk_events_schedule/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  ScheduleStatusOriginal: () => ScheduleStatus,
  SortOrderOriginal: () => SortOrder,
  StateFilterOriginal: () => StateFilter,
  addScheduleItem: () => addScheduleItem2,
  deleteScheduleItem: () => deleteScheduleItem2,
  discardDraft: () => discardDraft2,
  getScheduleItem: () => getScheduleItem2,
  listScheduleItems: () => listScheduleItems2,
  publishDraft: () => publishDraft2,
  queryScheduleItems: () => queryScheduleItems2,
  rescheduleDraft: () => rescheduleDraft2,
  updateScheduleItem: () => updateScheduleItem2
});
module.exports = __toCommonJS(meta_exports);

// src/events-schedule-v1-schedule-item-schedule.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_duration = require("@wix/sdk-runtime/transformations/duration");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEventsScheduleScheduleManagementUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/wix-events-agenda",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    _: [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "editor-flow.wixapps.net": [
      {
        srcPath: "/_api/wix-events-schedule/v1/schedule",
        destPath: "/v1/schedule"
      },
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/events/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "api._base_domain_": [
      {
        srcPath: "/events/v1/schedule",
        destPath: "/v1/schedule"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/wix-events-web/v1/schedule",
        destPath: "/v1/schedule"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_events_schedule";
function listScheduleItems(payload) {
  function __listScheduleItems({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "startingFrom" }, { path: "startingBefore" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "GET",
      methodFqn: "wix.events.schedule.ScheduleManagement.ListScheduleItems",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleScheduleManagementUrl({
        protoPath: "/v1/schedule",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "items.createdDate" },
            { path: "items.updatedDate" },
            { path: "items.timeSlot.start" },
            { path: "items.timeSlot.end" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listScheduleItems;
}
function queryScheduleItems(payload) {
  function __queryScheduleItems({ host }) {
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "POST",
      methodFqn: "wix.events.schedule.ScheduleManagement.QueryScheduleItems",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleScheduleManagementUrl({
        protoPath: "/v1/schedule/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "items.createdDate" },
            { path: "items.updatedDate" },
            { path: "items.timeSlot.start" },
            { path: "items.timeSlot.end" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryScheduleItems;
}
function getScheduleItem(payload) {
  function __getScheduleItem({ host }) {
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "GET",
      methodFqn: "wix.events.schedule.ScheduleManagement.GetScheduleItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleScheduleManagementUrl({
        protoPath: "/v1/schedule/{itemId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "item.createdDate" },
            { path: "item.updatedDate" },
            { path: "item.timeSlot.start" },
            { path: "item.timeSlot.end" },
            { path: "draft.createdDate" },
            { path: "draft.updatedDate" },
            { path: "draft.timeSlot.start" },
            { path: "draft.timeSlot.end" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getScheduleItem;
}
function addScheduleItem(payload) {
  function __addScheduleItem({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "item.timeSlot.start" }, { path: "item.timeSlot.end" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "POST",
      methodFqn: "wix.events.schedule.ScheduleManagement.AddScheduleItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleScheduleManagementUrl({
        protoPath: "/v1/schedule/draft",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "item.createdDate" },
            { path: "item.updatedDate" },
            { path: "item.timeSlot.start" },
            { path: "item.timeSlot.end" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __addScheduleItem;
}
function updateScheduleItem(payload) {
  function __updateScheduleItem({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fields" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "item.timeSlot.start" }, { path: "item.timeSlot.end" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "PATCH",
      methodFqn: "wix.events.schedule.ScheduleManagement.UpdateScheduleItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleScheduleManagementUrl({
        protoPath: "/v1/schedule/draft/{itemId}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "item.createdDate" },
            { path: "item.updatedDate" },
            { path: "item.timeSlot.start" },
            { path: "item.timeSlot.end" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateScheduleItem;
}
function deleteScheduleItem(payload) {
  function __deleteScheduleItem({ host }) {
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "DELETE",
      methodFqn: "wix.events.schedule.ScheduleManagement.DeleteScheduleItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleScheduleManagementUrl({
        protoPath: "/v1/schedule/draft/items",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteScheduleItem;
}
function discardDraft(payload) {
  function __discardDraft({ host }) {
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "DELETE",
      methodFqn: "wix.events.schedule.ScheduleManagement.DiscardDraft",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleScheduleManagementUrl({
        protoPath: "/v1/schedule/draft",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __discardDraft;
}
function publishDraft(payload) {
  function __publishDraft({ host }) {
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "POST",
      methodFqn: "wix.events.schedule.ScheduleManagement.PublishDraft",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleScheduleManagementUrl({
        protoPath: "/v1/schedule/publish",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __publishDraft;
}
function rescheduleDraft(payload) {
  function __rescheduleDraft({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_duration.transformSDKDurationToRESTDuration,
        paths: [{ path: "timeSlotOffset" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.schedule.v1.schedule_item",
      method: "POST",
      methodFqn: "wix.events.schedule.ScheduleManagement.RescheduleDraft",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsScheduleScheduleManagementUrl({
        protoPath: "/v1/schedule/draft/reschedule",
        data: serializedData,
        host
      }),
      data: serializedData
    };
    return metadata;
  }
  return __rescheduleDraft;
}

// src/events-schedule-v1-schedule-item-schedule.types.ts
var ScheduleStatus = /* @__PURE__ */ ((ScheduleStatus2) => {
  ScheduleStatus2["SCHEDULED"] = "SCHEDULED";
  ScheduleStatus2["CANCELED"] = "CANCELED";
  return ScheduleStatus2;
})(ScheduleStatus || {});
var StateFilter = /* @__PURE__ */ ((StateFilter2) => {
  StateFilter2["PUBLISHED"] = "PUBLISHED";
  StateFilter2["DRAFT"] = "DRAFT";
  StateFilter2["VISIBLE"] = "VISIBLE";
  StateFilter2["HIDDEN"] = "HIDDEN";
  return StateFilter2;
})(StateFilter || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});

// src/events-schedule-v1-schedule-item-schedule.meta.ts
function listScheduleItems2() {
  const payload = {};
  const getRequestOptions = listScheduleItems(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/schedule",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryScheduleItems2() {
  const payload = {};
  const getRequestOptions = queryScheduleItems(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/schedule/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getScheduleItem2() {
  const payload = { itemId: ":itemId" };
  const getRequestOptions = getScheduleItem(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/schedule/{itemId}",
    pathParams: { itemId: "itemId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function addScheduleItem2() {
  const payload = {};
  const getRequestOptions = addScheduleItem(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/schedule/draft",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateScheduleItem2() {
  const payload = { itemId: ":itemId" };
  const getRequestOptions = updateScheduleItem(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/schedule/draft/{itemId}",
    pathParams: { itemId: "itemId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteScheduleItem2() {
  const payload = {};
  const getRequestOptions = deleteScheduleItem(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/schedule/draft/items",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function discardDraft2() {
  const payload = {};
  const getRequestOptions = discardDraft(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/schedule/draft",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function publishDraft2() {
  const payload = {};
  const getRequestOptions = publishDraft(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/schedule/publish",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function rescheduleDraft2() {
  const payload = {};
  const getRequestOptions = rescheduleDraft(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/schedule/draft/reschedule",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ScheduleStatusOriginal,
  SortOrderOriginal,
  StateFilterOriginal,
  addScheduleItem,
  deleteScheduleItem,
  discardDraft,
  getScheduleItem,
  listScheduleItems,
  publishDraft,
  queryScheduleItems,
  rescheduleDraft,
  updateScheduleItem
});
