// @wix/auto_sdk_bookings_attendance/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  AttendanceStatusOriginal: () => AttendanceStatus,
  SortOrderOriginal: () => SortOrder,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  bulkSetAttendance: () => bulkSetAttendance2,
  countAttendances: () => countAttendances2,
  getAttendance: () => getAttendance2,
  queryAttendance: () => queryAttendance2,
  setAttendance: () => setAttendance2
});
module.exports = __toCommonJS(meta_exports);

// src/bookings-v2-attendance-attendance.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressBookingsAttendanceV2AttendanceServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/bookings-attendance",
        destPath: ""
      },
      {
        srcPath: "/bookings-attendance",
        destPath: ""
      },
      {
        srcPath: "/_api/bookings-attendance/v2/bulk/attendance",
        destPath: "/v2/bulk/attendance"
      },
      {
        srcPath: "/bookings-attendance/v2/bulk/attendance",
        destPath: "/v2/bulk/attendance"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/bookings-attendance",
        destPath: "/v2/attendance"
      },
      {
        srcPath: "/bookings/attendance/v2/bulk/attendance",
        destPath: "/v2/bulk/attendance"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/bookings/bookings-attendance",
        destPath: "/v2/attendance"
      },
      {
        srcPath: "/_api/bookings-attendance",
        destPath: "/v2/attendance"
      },
      {
        srcPath: "/bookings/attendance",
        destPath: "/v2/attendance"
      },
      {
        srcPath: "/bookings/attendance/v2/attendance",
        destPath: "/v2/attendance"
      },
      {
        srcPath: "/bookings/v2/attendance",
        destPath: "/v2/attendance"
      },
      {
        srcPath: "/bookings/v2/bulk/attendance",
        destPath: "/v2/bulk/attendance"
      },
      {
        srcPath: "/bookings-attendance/v2/bulk/attendance",
        destPath: "/v2/bulk/attendance"
      },
      {
        srcPath: "/_api/bookings-attendance/v2/bulk/attendance",
        destPath: "/v2/bulk/attendance"
      },
      {
        srcPath: "/bookings/attendance/v2/bulk/attendance",
        destPath: "/v2/bulk/attendance"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/bookings-attendance",
        destPath: "/v2/attendance"
      },
      {
        srcPath: "/_api/bookings-attendance/v2/bulk/attendance",
        destPath: "/v2/bulk/attendance"
      }
    ],
    _: [
      {
        srcPath: "/bookings-attendance",
        destPath: "/v2/attendance"
      },
      {
        srcPath: "/_api/bookings-attendance/v2/bulk/attendance",
        destPath: "/v2/bulk/attendance"
      },
      {
        srcPath: "/bookings/attendance/v2/bulk/attendance",
        destPath: "/v2/bulk/attendance"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/bookings-attendance",
        destPath: "/v2/attendance"
      },
      {
        srcPath: "/api/bookings-attendance/v2/bulk/attendance",
        destPath: "/v2/bulk/attendance"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_bookings_attendance";
function getAttendance(payload) {
  function __getAttendance({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.v2.attendance",
      method: "GET",
      methodFqn: "com.wixpress.bookings.attendance.v2.AttendanceService.GetAttendance",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressBookingsAttendanceV2AttendanceServiceUrl({
        protoPath: "/v2/attendance/{attendanceId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "attendance.updatedDate" },
            { path: "attendance.statusUpdatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getAttendance;
}
function setAttendance(payload) {
  function __setAttendance({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "attendance.updatedDate" },
          { path: "attendance.statusUpdatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.bookings.v2.attendance",
      method: "POST",
      methodFqn: "com.wixpress.bookings.attendance.v2.AttendanceService.SetAttendance",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressBookingsAttendanceV2AttendanceServiceUrl({
        protoPath: "/v2/attendance/set",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "attendance.updatedDate" },
            { path: "attendance.statusUpdatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __setAttendance;
}
function bulkSetAttendance(payload) {
  function __bulkSetAttendance({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "attendanceList.updatedDate" },
          { path: "attendanceList.statusUpdatedDate" },
          { path: "attendanceDetails.attendance.updatedDate" },
          { path: "attendanceDetails.attendance.statusUpdatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.bookings.v2.attendance",
      method: "POST",
      methodFqn: "com.wixpress.bookings.attendance.v2.AttendanceService.BulkSetAttendance",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressBookingsAttendanceV2AttendanceServiceUrl({
        protoPath: "/v2/bulk/attendance/set",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.updatedDate" },
            { path: "results.item.statusUpdatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkSetAttendance;
}
function queryAttendance(payload) {
  function __queryAttendance({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.v2.attendance",
      method: "POST",
      methodFqn: "com.wixpress.bookings.attendance.v2.AttendanceService.QueryAttendance",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressBookingsAttendanceV2AttendanceServiceUrl({
        protoPath: "/v2/attendance/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "attendances.updatedDate" },
            { path: "attendances.statusUpdatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryAttendance;
}
function countAttendances(payload) {
  function __countAttendances({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.v2.attendance",
      method: "POST",
      methodFqn: "com.wixpress.bookings.attendance.v2.AttendanceService.CountAttendances",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressBookingsAttendanceV2AttendanceServiceUrl({
        protoPath: "/v2/attendance/count",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __countAttendances;
}

// src/bookings-v2-attendance-attendance.types.ts
var AttendanceStatus = /* @__PURE__ */ ((AttendanceStatus2) => {
  AttendanceStatus2["NOT_SET"] = "NOT_SET";
  AttendanceStatus2["ATTENDED"] = "ATTENDED";
  AttendanceStatus2["NOT_ATTENDED"] = "NOT_ATTENDED";
  return AttendanceStatus2;
})(AttendanceStatus || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/bookings-v2-attendance-attendance.meta.ts
function getAttendance2() {
  const payload = { attendanceId: ":attendanceId" };
  const getRequestOptions = getAttendance(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/attendance/{attendanceId}",
    pathParams: { attendanceId: "attendanceId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function setAttendance2() {
  const payload = {};
  const getRequestOptions = setAttendance(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/attendance/set",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkSetAttendance2() {
  const payload = {};
  const getRequestOptions = bulkSetAttendance(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/bulk/attendance/set",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryAttendance2() {
  const payload = {};
  const getRequestOptions = queryAttendance(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/attendance/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function countAttendances2() {
  const payload = {};
  const getRequestOptions = countAttendances(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/attendance/count",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AttendanceStatusOriginal,
  SortOrderOriginal,
  WebhookIdentityTypeOriginal,
  bulkSetAttendance,
  countAttendances,
  getAttendance,
  queryAttendance,
  setAttendance
});
