// @wix/patterns/dist/cjs/components/CustomColumns/CustomColumns.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.CustomColumns = CustomColumns;
var _ToolbarCollectionContext = require("../ToolbarCollectionContext");
var _react = _interopRequireDefault(require("react"));
var _initCustomColumns = require("../../state/initCustomColumns");
var _CustomColumnsCTAButton = require("../CustomColumnsPanel/CustomColumnsCTAButton");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CustomColumns/CustomColumns.tsx";
function CustomColumns({
  onColumnClick
}) {
  const state = (0, _ToolbarCollectionContext.useToolbarCollectionContext)();
  return /*#__PURE__*/_react.default.createElement(_CustomColumnsCTAButton.CustomColumnsCTAButton, {
    onColumnClick: onColumnClick,
    state: state,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 22,
      columnNumber: 10
    }
  });
}
CustomColumns.setCustomColumnsState = _initCustomColumns.setCustomColumnsState;
CustomColumns.dragAndDrop = undefined;
CustomColumns.displayName = 'CustomColumns';
