// @wix/auto_sdk_restaurants_sections/build/internal/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  AvailabilityType: () => AvailabilityType,
  DayOfWeek: () => DayOfWeek,
  SortOrder: () => SortOrder,
  WebhookIdentityType: () => WebhookIdentityType,
  bulkCreateSections: () => bulkCreateSections4,
  bulkDeleteSections: () => bulkDeleteSections4,
  bulkUpdateSection: () => bulkUpdateSection4,
  createSection: () => createSection4,
  deleteSection: () => deleteSection4,
  duplicateSection: () => duplicateSection4,
  getSection: () => getSection4,
  listSections: () => listSections4,
  onSectionCreated: () => onSectionCreated2,
  onSectionDeleted: () => onSectionDeleted2,
  onSectionUpdated: () => onSectionUpdated2,
  querySections: () => querySections4,
  updateSection: () => updateSection4
});
module.exports = __toCommonJS(index_exports);

// src/restaurants-menus-v1-section-sections.public.ts
var import_rename_all_nested_keys2 = require("@wix/sdk-runtime/rename-all-nested-keys");
var import_image3 = require("@wix/sdk-runtime/transformations/image");
var import_timestamp3 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths3 = require("@wix/sdk-runtime/transformations/transform-paths");
var import_sdk_types = require("@wix/sdk-types");

// src/restaurants-menus-v1-section-sections.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/restaurants-menus-v1-section-sections.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressRestaurantsMenusSectionV1RestaurantsMenusSectionUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/sections",
        destPath: "/v1/sections"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/sections",
        destPath: "/v1/bulk/sections"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/sections",
        destPath: "/v1/sections"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/sections",
        destPath: "/v1/bulk/sections"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-menus-section",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/menus-section",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/sections",
        destPath: "/v1/bulk/sections"
      },
      {
        srcPath: "/restaurants/menus/v1/sections",
        destPath: "/v1/sections"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
function resolveComWixpressRestaurantsMenusSectionV1RestaurantsDuplicateSectionRestaurantsDuplicateSectionUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/sections",
        destPath: "/v1/sections"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/sections",
        destPath: "/v1/bulk/sections"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/sections",
        destPath: "/v1/sections"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/sections",
        destPath: "/v1/bulk/sections"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-menus-section",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/menus-section",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/sections",
        destPath: "/v1/bulk/sections"
      },
      {
        srcPath: "/restaurants/menus/v1/sections",
        destPath: "/v1/sections"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_sections";
function duplicateSection(payload) {
  function __duplicateSection({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.section",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_section.v1.restaurants_duplicate_section.RestaurantsDuplicateSection.DuplicateSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusSectionV1RestaurantsDuplicateSectionRestaurantsDuplicateSectionUrl(
        { protoPath: "/v1/sections/{id}/duplicate", data: payload, host }
      ),
      data: payload
    };
    return metadata;
  }
  return __duplicateSection;
}
function createSection(payload) {
  function __createSection({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "section.createdDate" },
          { path: "section.updatedDate" },
          { path: "section.image.urlExpirationDate" },
          { path: "section.additionalImages.urlExpirationDate" },
          {
            path: "section.orderSettings.availability.timestampRangesOptions.ranges.startTime"
          },
          {
            path: "section.orderSettings.availability.timestampRangesOptions.ranges.endTime"
          },
          { path: "section.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.section",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_section.v1.RestaurantsMenusSection.CreateSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusSectionV1RestaurantsMenusSectionUrl(
        { protoPath: "/v1/sections", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "section.createdDate" },
            { path: "section.updatedDate" },
            { path: "section.image.urlExpirationDate" },
            { path: "section.additionalImages.urlExpirationDate" },
            {
              path: "section.orderSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "section.orderSettings.availability.timestampRangesOptions.ranges.endTime"
            },
            { path: "section.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createSection;
}
function bulkCreateSections(payload) {
  function __bulkCreateSections({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "sections.createdDate" },
          { path: "sections.updatedDate" },
          { path: "sections.image.urlExpirationDate" },
          { path: "sections.additionalImages.urlExpirationDate" },
          {
            path: "sections.orderSettings.availability.timestampRangesOptions.ranges.startTime"
          },
          {
            path: "sections.orderSettings.availability.timestampRangesOptions.ranges.endTime"
          },
          { path: "sections.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.section",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_section.v1.RestaurantsMenusSection.BulkCreateSections",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusSectionV1RestaurantsMenusSectionUrl(
        { protoPath: "/v1/bulk/sections/create", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.image.urlExpirationDate" },
            { path: "results.item.additionalImages.urlExpirationDate" },
            {
              path: "results.item.orderSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "results.item.orderSettings.availability.timestampRangesOptions.ranges.endTime"
            },
            { path: "results.item.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateSections;
}
function getSection(payload) {
  function __getSection({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.section",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.menus_section.v1.RestaurantsMenusSection.GetSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusSectionV1RestaurantsMenusSectionUrl(
        { protoPath: "/v1/sections/{sectionId}", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "section.createdDate" },
            { path: "section.updatedDate" },
            { path: "section.image.urlExpirationDate" },
            { path: "section.additionalImages.urlExpirationDate" },
            {
              path: "section.orderSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "section.orderSettings.availability.timestampRangesOptions.ranges.endTime"
            },
            { path: "section.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getSection;
}
function listSections(payload) {
  function __listSections({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.section",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.menus_section.v1.RestaurantsMenusSection.ListSections",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusSectionV1RestaurantsMenusSectionUrl(
        { protoPath: "/v1/sections", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "sections.createdDate" },
            { path: "sections.updatedDate" },
            { path: "sections.image.urlExpirationDate" },
            { path: "sections.additionalImages.urlExpirationDate" },
            {
              path: "sections.orderSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "sections.orderSettings.availability.timestampRangesOptions.ranges.endTime"
            },
            { path: "sections.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listSections;
}
function querySections(payload) {
  function __querySections({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.section",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_section.v1.RestaurantsMenusSection.QuerySections",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusSectionV1RestaurantsMenusSectionUrl(
        { protoPath: "/v1/sections/query", data: payload, host }
      ),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "sections.createdDate" },
            { path: "sections.updatedDate" },
            { path: "sections.image.urlExpirationDate" },
            { path: "sections.additionalImages.urlExpirationDate" },
            {
              path: "sections.orderSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "sections.orderSettings.availability.timestampRangesOptions.ranges.endTime"
            },
            { path: "sections.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __querySections;
}
function updateSection(payload) {
  function __updateSection({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "section.createdDate" },
          { path: "section.updatedDate" },
          { path: "section.image.urlExpirationDate" },
          { path: "section.additionalImages.urlExpirationDate" },
          {
            path: "section.orderSettings.availability.timestampRangesOptions.ranges.startTime"
          },
          {
            path: "section.orderSettings.availability.timestampRangesOptions.ranges.endTime"
          },
          { path: "section.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.section",
      method: "PATCH",
      methodFqn: "com.wixpress.restaurants.menus_section.v1.RestaurantsMenusSection.UpdateSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusSectionV1RestaurantsMenusSectionUrl(
        { protoPath: "/v1/sections/{section.id}", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "section.createdDate" },
            { path: "section.updatedDate" },
            { path: "section.image.urlExpirationDate" },
            { path: "section.additionalImages.urlExpirationDate" },
            {
              path: "section.orderSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "section.orderSettings.availability.timestampRangesOptions.ranges.endTime"
            },
            { path: "section.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateSection;
}
function bulkUpdateSection(payload) {
  function __bulkUpdateSection({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "sections.mask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "sections.section.createdDate" },
          { path: "sections.section.updatedDate" },
          { path: "sections.section.image.urlExpirationDate" },
          { path: "sections.section.additionalImages.urlExpirationDate" },
          {
            path: "sections.section.orderSettings.availability.timestampRangesOptions.ranges.startTime"
          },
          {
            path: "sections.section.orderSettings.availability.timestampRangesOptions.ranges.endTime"
          },
          { path: "sections.section.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.section",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_section.v1.RestaurantsMenusSection.BulkUpdateSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusSectionV1RestaurantsMenusSectionUrl(
        { protoPath: "/v1/bulk/sections/update", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.section.createdDate" },
            { path: "results.section.updatedDate" },
            { path: "results.section.image.urlExpirationDate" },
            { path: "results.section.additionalImages.urlExpirationDate" },
            {
              path: "results.section.orderSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "results.section.orderSettings.availability.timestampRangesOptions.ranges.endTime"
            },
            { path: "results.section.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateSection;
}
function deleteSection(payload) {
  function __deleteSection({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.section",
      method: "DELETE",
      methodFqn: "com.wixpress.restaurants.menus_section.v1.RestaurantsMenusSection.DeleteSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusSectionV1RestaurantsMenusSectionUrl(
        { protoPath: "/v1/sections/{sectionId}", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteSection;
}
function bulkDeleteSections(payload) {
  function __bulkDeleteSections({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.section",
      method: "DELETE",
      methodFqn: "com.wixpress.restaurants.menus_section.v1.RestaurantsMenusSection.BulkDeleteSections",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusSectionV1RestaurantsMenusSectionUrl(
        { protoPath: "/v1/bulk/sections/delete", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __bulkDeleteSections;
}

// src/restaurants-menus-v1-section-sections.universal.ts
var import_image = require("@wix/sdk-runtime/transformations/image");
var import_image2 = require("@wix/sdk-runtime/transformations/image");
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var AvailabilityType = /* @__PURE__ */ ((AvailabilityType2) => {
  AvailabilityType2["UNSPECIFIED_AVAILABILITY_OPTION"] = "UNSPECIFIED_AVAILABILITY_OPTION";
  AvailabilityType2["ALWAYS_AVAILABLE"] = "ALWAYS_AVAILABLE";
  AvailabilityType2["WEEKLY_SCHEDULE"] = "WEEKLY_SCHEDULE";
  AvailabilityType2["TIMESTAMP_RANGES"] = "TIMESTAMP_RANGES";
  return AvailabilityType2;
})(AvailabilityType || {});
var DayOfWeek = /* @__PURE__ */ ((DayOfWeek2) => {
  DayOfWeek2["MON"] = "MON";
  DayOfWeek2["TUE"] = "TUE";
  DayOfWeek2["WED"] = "WED";
  DayOfWeek2["THU"] = "THU";
  DayOfWeek2["FRI"] = "FRI";
  DayOfWeek2["SAT"] = "SAT";
  DayOfWeek2["SUN"] = "SUN";
  return DayOfWeek2;
})(DayOfWeek || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function duplicateSection2(_id, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    id: _id,
    businessLocationIds: options?.businessLocationIds,
    duplicateSubEntities: options?.duplicateSubEntities
  });
  const reqOpts = duplicateSection(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          id: "$[0]",
          businessLocationIds: "$[1].businessLocationIds",
          duplicateSubEntities: "$[1].duplicateSubEntities"
        },
        singleArgumentUnchanged: false
      },
      ["_id", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function createSection2(section) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ section }),
    [
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [
          { path: "section.image" },
          { path: "section.additionalImages", isRepeated: true }
        ]
      }
    ]
  );
  const reqOpts = createSection(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "section.image" },
            { path: "section.additionalImages", isRepeated: true }
          ]
        }
      ])
    )?.section;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { section: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["section"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCreateSections2(sections, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      sections,
      returnEntity: options?.returnEntity
    }),
    [
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [
          { path: "sections.image" },
          { path: "sections.additionalImages", isRepeated: true }
        ]
      }
    ]
  );
  const reqOpts = bulkCreateSections(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "results.item.image" },
            { path: "results.item.additionalImages", isRepeated: true }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          sections: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["sections", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getSection2(sectionId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    sectionId
  });
  const reqOpts = getSection(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "section.image" },
            { path: "section.additionalImages", isRepeated: true }
          ]
        }
      ])
    )?.section;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { sectionId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["sectionId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listSections2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    sectionIds: options?.sectionIds,
    paging: options?.paging,
    onlyVisible: options?.onlyVisible
  });
  const reqOpts = listSections(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "sections.image" },
            { path: "sections.additionalImages", isRepeated: true }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          sectionIds: "$[0].sectionIds",
          paging: "$[0].paging",
          onlyVisible: "$[0].onlyVisible"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function querySections2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = querySections(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [
          {
            transformFn: import_image2.transformRESTImageToSDKImage,
            paths: [
              { path: "sections.image" },
              { path: "sections.additionalImages", isRepeated: true }
            ]
          }
        ])
      );
      return {
        items: transformedData?.sections,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function updateSection2(_id, section) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ section: { ...section, id: _id } }),
    [
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [
          { path: "section.image" },
          { path: "section.additionalImages", isRepeated: true }
        ]
      }
    ]
  );
  const reqOpts = updateSection(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "section.image" },
            { path: "section.additionalImages", isRepeated: true }
          ]
        }
      ])
    )?.section;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { section: "$[1]" },
        explicitPathsToArguments: { "section.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "section"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateSection2(sections, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      sections,
      returnEntity: options?.returnEntity
    }),
    [
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [
          { path: "sections.section.image" },
          { path: "sections.section.additionalImages", isRepeated: true }
        ]
      }
    ]
  );
  const reqOpts = bulkUpdateSection(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "results.section.image" },
            { path: "results.section.additionalImages", isRepeated: true }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          sections: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["sections", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteSection2(sectionId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    sectionId
  });
  const reqOpts = deleteSection(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { sectionId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["sectionId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkDeleteSections2(ids) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ ids });
  const reqOpts = bulkDeleteSections(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { ids: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["ids"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/restaurants-menus-v1-section-sections.public.ts
function duplicateSection3(httpClient) {
  return (_id, options) => duplicateSection2(
    _id,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function createSection3(httpClient) {
  return (section) => createSection2(
    section,
    // @ts-ignore
    { httpClient }
  );
}
function bulkCreateSections3(httpClient) {
  return (sections, options) => bulkCreateSections2(
    sections,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function getSection3(httpClient) {
  return (sectionId) => getSection2(
    sectionId,
    // @ts-ignore
    { httpClient }
  );
}
function listSections3(httpClient) {
  return (options) => listSections2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function querySections3(httpClient) {
  return () => querySections2(
    // @ts-ignore
    { httpClient }
  );
}
function updateSection3(httpClient) {
  return (_id, section) => updateSection2(
    _id,
    section,
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdateSection3(httpClient) {
  return (sections, options) => bulkUpdateSection2(
    sections,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function deleteSection3(httpClient) {
  return (sectionId) => deleteSection2(
    sectionId,
    // @ts-ignore
    { httpClient }
  );
}
function bulkDeleteSections3(httpClient) {
  return (ids) => bulkDeleteSections2(
    ids,
    // @ts-ignore
    { httpClient }
  );
}
var onSectionCreated = (0, import_sdk_types.EventDefinition)(
  "wix.restaurants.menus.v1.section_created",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          {
            path: "entity.orderSettings.availability.timestampRangesOptions.ranges.startTime"
          },
          {
            path: "entity.orderSettings.availability.timestampRangesOptions.ranges.endTime"
          },
          { path: "entity.externalReferenceInfo.lastSyncDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_image3.transformRESTImageToSDKImage,
        paths: [
          { path: "entity.image" },
          { path: "entity.additionalImages", isRepeated: true }
        ]
      }
    ])
  )
)();
var onSectionDeleted = (0, import_sdk_types.EventDefinition)(
  "wix.restaurants.menus.v1.section_deleted",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "undefined.createdDate" },
          { path: "undefined.updatedDate" },
          {
            path: "undefined.orderSettings.availability.timestampRangesOptions.ranges.startTime"
          },
          {
            path: "undefined.orderSettings.availability.timestampRangesOptions.ranges.endTime"
          },
          { path: "undefined.externalReferenceInfo.lastSyncDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_image3.transformRESTImageToSDKImage,
        paths: [
          { path: "undefined.image" },
          { path: "undefined.additionalImages", isRepeated: true }
        ]
      }
    ])
  )
)();
var onSectionUpdated = (0, import_sdk_types.EventDefinition)(
  "wix.restaurants.menus.v1.section_updated",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          {
            path: "entity.orderSettings.availability.timestampRangesOptions.ranges.startTime"
          },
          {
            path: "entity.orderSettings.availability.timestampRangesOptions.ranges.endTime"
          },
          { path: "entity.externalReferenceInfo.lastSyncDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_image3.transformRESTImageToSDKImage,
        paths: [
          { path: "entity.image" },
          { path: "entity.additionalImages", isRepeated: true }
        ]
      }
    ])
  )
)();

// src/restaurants-menus-v1-section-sections.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var import_event_definition_modules = require("@wix/sdk-runtime/event-definition-modules");
var duplicateSection4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(duplicateSection3);
var createSection4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(createSection3);
var bulkCreateSections4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkCreateSections3);
var getSection4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getSection3);
var listSections4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(listSections3);
var querySections4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(querySections3);
var updateSection4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(updateSection3);
var bulkUpdateSection4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkUpdateSection3);
var deleteSection4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(deleteSection3);
var bulkDeleteSections4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkDeleteSections3);
var onSectionCreated2 = (0, import_event_definition_modules.createEventModule)(onSectionCreated);
var onSectionDeleted2 = (0, import_event_definition_modules.createEventModule)(onSectionDeleted);
var onSectionUpdated2 = (0, import_event_definition_modules.createEventModule)(onSectionUpdated);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AvailabilityType,
  DayOfWeek,
  SortOrder,
  WebhookIdentityType,
  bulkCreateSections,
  bulkDeleteSections,
  bulkUpdateSection,
  createSection,
  deleteSection,
  duplicateSection,
  getSection,
  listSections,
  onSectionCreated,
  onSectionDeleted,
  onSectionUpdated,
  querySections,
  updateSection
});
