// @wix/auto_sdk_restaurants_sections/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  AvailabilityTypeOriginal: () => AvailabilityType,
  DayOfWeekOriginal: () => DayOfWeek,
  SortOrderOriginal: () => SortOrder,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  bulkCreateSections: () => bulkCreateSections2,
  bulkDeleteSections: () => bulkDeleteSections2,
  bulkUpdateSection: () => bulkUpdateSection2,
  createSection: () => createSection2,
  deleteSection: () => deleteSection2,
  duplicateSection: () => duplicateSection2,
  getSection: () => getSection2,
  listSections: () => listSections2,
  querySections: () => querySections2,
  updateSection: () => updateSection2
});
module.exports = __toCommonJS(meta_exports);

// src/restaurants-menus-v1-section-sections.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressRestaurantsMenusSectionV1RestaurantsMenusSectionUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/sections",
        destPath: "/v1/sections"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/sections",
        destPath: "/v1/bulk/sections"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/sections",
        destPath: "/v1/sections"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/sections",
        destPath: "/v1/bulk/sections"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-menus-section",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/menus-section",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/sections",
        destPath: "/v1/bulk/sections"
      },
      {
        srcPath: "/restaurants/menus/v1/sections",
        destPath: "/v1/sections"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
function resolveComWixpressRestaurantsMenusSectionV1RestaurantsDuplicateSectionRestaurantsDuplicateSectionUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/sections",
        destPath: "/v1/sections"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/sections",
        destPath: "/v1/bulk/sections"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/sections",
        destPath: "/v1/sections"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/sections",
        destPath: "/v1/bulk/sections"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-menus-section",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/menus-section",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/sections",
        destPath: "/v1/bulk/sections"
      },
      {
        srcPath: "/restaurants/menus/v1/sections",
        destPath: "/v1/sections"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-section",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_sections";
function duplicateSection(payload) {
  function __duplicateSection({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.section",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_section.v1.restaurants_duplicate_section.RestaurantsDuplicateSection.DuplicateSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusSectionV1RestaurantsDuplicateSectionRestaurantsDuplicateSectionUrl(
        { protoPath: "/v1/sections/{id}/duplicate", data: payload, host }
      ),
      data: payload
    };
    return metadata;
  }
  return __duplicateSection;
}
function createSection(payload) {
  function __createSection({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "section.createdDate" },
          { path: "section.updatedDate" },
          { path: "section.image.urlExpirationDate" },
          { path: "section.additionalImages.urlExpirationDate" },
          {
            path: "section.orderSettings.availability.timestampRangesOptions.ranges.startTime"
          },
          {
            path: "section.orderSettings.availability.timestampRangesOptions.ranges.endTime"
          },
          { path: "section.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.section",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_section.v1.RestaurantsMenusSection.CreateSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusSectionV1RestaurantsMenusSectionUrl(
        { protoPath: "/v1/sections", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "section.createdDate" },
            { path: "section.updatedDate" },
            { path: "section.image.urlExpirationDate" },
            { path: "section.additionalImages.urlExpirationDate" },
            {
              path: "section.orderSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "section.orderSettings.availability.timestampRangesOptions.ranges.endTime"
            },
            { path: "section.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createSection;
}
function bulkCreateSections(payload) {
  function __bulkCreateSections({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "sections.createdDate" },
          { path: "sections.updatedDate" },
          { path: "sections.image.urlExpirationDate" },
          { path: "sections.additionalImages.urlExpirationDate" },
          {
            path: "sections.orderSettings.availability.timestampRangesOptions.ranges.startTime"
          },
          {
            path: "sections.orderSettings.availability.timestampRangesOptions.ranges.endTime"
          },
          { path: "sections.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.section",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_section.v1.RestaurantsMenusSection.BulkCreateSections",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusSectionV1RestaurantsMenusSectionUrl(
        { protoPath: "/v1/bulk/sections/create", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.image.urlExpirationDate" },
            { path: "results.item.additionalImages.urlExpirationDate" },
            {
              path: "results.item.orderSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "results.item.orderSettings.availability.timestampRangesOptions.ranges.endTime"
            },
            { path: "results.item.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateSections;
}
function getSection(payload) {
  function __getSection({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.section",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.menus_section.v1.RestaurantsMenusSection.GetSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusSectionV1RestaurantsMenusSectionUrl(
        { protoPath: "/v1/sections/{sectionId}", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "section.createdDate" },
            { path: "section.updatedDate" },
            { path: "section.image.urlExpirationDate" },
            { path: "section.additionalImages.urlExpirationDate" },
            {
              path: "section.orderSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "section.orderSettings.availability.timestampRangesOptions.ranges.endTime"
            },
            { path: "section.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getSection;
}
function listSections(payload) {
  function __listSections({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.section",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.menus_section.v1.RestaurantsMenusSection.ListSections",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusSectionV1RestaurantsMenusSectionUrl(
        { protoPath: "/v1/sections", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "sections.createdDate" },
            { path: "sections.updatedDate" },
            { path: "sections.image.urlExpirationDate" },
            { path: "sections.additionalImages.urlExpirationDate" },
            {
              path: "sections.orderSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "sections.orderSettings.availability.timestampRangesOptions.ranges.endTime"
            },
            { path: "sections.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listSections;
}
function querySections(payload) {
  function __querySections({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.section",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_section.v1.RestaurantsMenusSection.QuerySections",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusSectionV1RestaurantsMenusSectionUrl(
        { protoPath: "/v1/sections/query", data: payload, host }
      ),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "sections.createdDate" },
            { path: "sections.updatedDate" },
            { path: "sections.image.urlExpirationDate" },
            { path: "sections.additionalImages.urlExpirationDate" },
            {
              path: "sections.orderSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "sections.orderSettings.availability.timestampRangesOptions.ranges.endTime"
            },
            { path: "sections.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __querySections;
}
function updateSection(payload) {
  function __updateSection({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "section.createdDate" },
          { path: "section.updatedDate" },
          { path: "section.image.urlExpirationDate" },
          { path: "section.additionalImages.urlExpirationDate" },
          {
            path: "section.orderSettings.availability.timestampRangesOptions.ranges.startTime"
          },
          {
            path: "section.orderSettings.availability.timestampRangesOptions.ranges.endTime"
          },
          { path: "section.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.section",
      method: "PATCH",
      methodFqn: "com.wixpress.restaurants.menus_section.v1.RestaurantsMenusSection.UpdateSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusSectionV1RestaurantsMenusSectionUrl(
        { protoPath: "/v1/sections/{section.id}", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "section.createdDate" },
            { path: "section.updatedDate" },
            { path: "section.image.urlExpirationDate" },
            { path: "section.additionalImages.urlExpirationDate" },
            {
              path: "section.orderSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "section.orderSettings.availability.timestampRangesOptions.ranges.endTime"
            },
            { path: "section.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateSection;
}
function bulkUpdateSection(payload) {
  function __bulkUpdateSection({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "sections.mask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "sections.section.createdDate" },
          { path: "sections.section.updatedDate" },
          { path: "sections.section.image.urlExpirationDate" },
          { path: "sections.section.additionalImages.urlExpirationDate" },
          {
            path: "sections.section.orderSettings.availability.timestampRangesOptions.ranges.startTime"
          },
          {
            path: "sections.section.orderSettings.availability.timestampRangesOptions.ranges.endTime"
          },
          { path: "sections.section.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.section",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_section.v1.RestaurantsMenusSection.BulkUpdateSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusSectionV1RestaurantsMenusSectionUrl(
        { protoPath: "/v1/bulk/sections/update", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.section.createdDate" },
            { path: "results.section.updatedDate" },
            { path: "results.section.image.urlExpirationDate" },
            { path: "results.section.additionalImages.urlExpirationDate" },
            {
              path: "results.section.orderSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "results.section.orderSettings.availability.timestampRangesOptions.ranges.endTime"
            },
            { path: "results.section.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateSection;
}
function deleteSection(payload) {
  function __deleteSection({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.section",
      method: "DELETE",
      methodFqn: "com.wixpress.restaurants.menus_section.v1.RestaurantsMenusSection.DeleteSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusSectionV1RestaurantsMenusSectionUrl(
        { protoPath: "/v1/sections/{sectionId}", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteSection;
}
function bulkDeleteSections(payload) {
  function __bulkDeleteSections({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.section",
      method: "DELETE",
      methodFqn: "com.wixpress.restaurants.menus_section.v1.RestaurantsMenusSection.BulkDeleteSections",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusSectionV1RestaurantsMenusSectionUrl(
        { protoPath: "/v1/bulk/sections/delete", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __bulkDeleteSections;
}

// src/restaurants-menus-v1-section-sections.types.ts
var AvailabilityType = /* @__PURE__ */ ((AvailabilityType2) => {
  AvailabilityType2["UNSPECIFIED_AVAILABILITY_OPTION"] = "UNSPECIFIED_AVAILABILITY_OPTION";
  AvailabilityType2["ALWAYS_AVAILABLE"] = "ALWAYS_AVAILABLE";
  AvailabilityType2["WEEKLY_SCHEDULE"] = "WEEKLY_SCHEDULE";
  AvailabilityType2["TIMESTAMP_RANGES"] = "TIMESTAMP_RANGES";
  return AvailabilityType2;
})(AvailabilityType || {});
var DayOfWeek = /* @__PURE__ */ ((DayOfWeek2) => {
  DayOfWeek2["MON"] = "MON";
  DayOfWeek2["TUE"] = "TUE";
  DayOfWeek2["WED"] = "WED";
  DayOfWeek2["THU"] = "THU";
  DayOfWeek2["FRI"] = "FRI";
  DayOfWeek2["SAT"] = "SAT";
  DayOfWeek2["SUN"] = "SUN";
  return DayOfWeek2;
})(DayOfWeek || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/restaurants-menus-v1-section-sections.meta.ts
function duplicateSection2() {
  const payload = { id: ":id" };
  const getRequestOptions = duplicateSection(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/sections/{id}/duplicate",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createSection2() {
  const payload = {};
  const getRequestOptions = createSection(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/sections",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkCreateSections2() {
  const payload = {};
  const getRequestOptions = bulkCreateSections(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/sections/create",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getSection2() {
  const payload = { sectionId: ":sectionId" };
  const getRequestOptions = getSection(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/sections/{sectionId}",
    pathParams: { sectionId: "sectionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listSections2() {
  const payload = {};
  const getRequestOptions = listSections(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/sections",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function querySections2() {
  const payload = {};
  const getRequestOptions = querySections(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/sections/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateSection2() {
  const payload = { section: { id: ":sectionId" } };
  const getRequestOptions = updateSection(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/sections/{section.id}",
    pathParams: { sectionId: "sectionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateSection2() {
  const payload = {};
  const getRequestOptions = bulkUpdateSection(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/sections/update",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteSection2() {
  const payload = { sectionId: ":sectionId" };
  const getRequestOptions = deleteSection(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/sections/{sectionId}",
    pathParams: { sectionId: "sectionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkDeleteSections2() {
  const payload = {};
  const getRequestOptions = bulkDeleteSections(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/bulk/sections/delete",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AvailabilityTypeOriginal,
  DayOfWeekOriginal,
  SortOrderOriginal,
  WebhookIdentityTypeOriginal,
  bulkCreateSections,
  bulkDeleteSections,
  bulkUpdateSection,
  createSection,
  deleteSection,
  duplicateSection,
  getSection,
  listSections,
  querySections,
  updateSection
});
