// @wix/patterns/dist/cjs/components/EntityPageExtensions/EntityPageExtensions.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.EntityPageSlots = void 0;
exports._EntityPageSlots = _EntityPageSlots;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireWildcard(require("react"));
var _providers = require("../../providers");
var _SkeletonCard = require("../SkeletonCard");
var _Extensions = require("../Extensions");
var _ExtensionsState = require("../../state/ExtensionsState");
var _react2 = require("@wix/bex-core/react");
var _mobxReactLite = require("mobx-react-lite");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/EntityPageExtensions/EntityPageExtensions.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _EntityPageSlots(props) {
  const {
    containerId,
    dataHook,
    ...rest
  } = props;
  const pageState = (0, _providers.useEntityPageContext)();
  const container = (0, _react2.useWixPatternsContainer)();
  const [state] = (0, _react.useState)(() => {
    return new _ExtensionsState.ExtensionsState({
      container,
      containerId
    });
  });
  (0, _react.useEffect)(() => {
    return state.init();
  }, []);
  return !pageState || pageState.isFetching || state.initTask.status.isLoading ? /*#__PURE__*/_react.default.createElement(_SkeletonCard.SkeletonCard, {
    dataHook: dataHook,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 42,
      columnNumber: 5
    }
  }) : /*#__PURE__*/_react.default.createElement(_Extensions.Extensions, (0, _extends2.default)({
    form: pageState.form,
    entity: pageState.entity,
    dataHook: dataHook
  }, rest, {
    state: state,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 44,
      columnNumber: 5
    }
  }));
}
const EntityPageSlots = exports.EntityPageSlots = (0, _mobxReactLite.observer)(_EntityPageSlots);
