// @wix/auto-patterns/dist/cjs/utils/filterCreators.js
"use strict";

exports.__esModule = true;
exports.createReferenceFilter = exports.createNumberFilter = exports.createEnumFilter = exports.createDateTimeFilter = exports.createBooleanFilter = void 0;
var _patterns = require("@wix/patterns");
var _constants = require("../dataSourceAdapters/constants");
const createDateTimeFilter = ({
  id,
  fieldId,
  commonFilterProps,
  config
}) => ({
  key: id,
  fieldId,
  filter: (0, _patterns.dateRangeFilter)(),
  component: _patterns.DateRangeFilter,
  props: {
    ...commonFilterProps,
    mode: config == null ? void 0 : config.mode,
    dateRangeOptions: config == null ? void 0 : config.presets,
    filterByTime: config == null ? void 0 : config.includeTime
  }
});
exports.createDateTimeFilter = createDateTimeFilter;
const createBooleanFilter = ({
  id,
  fieldId,
  commonFilterProps,
  config
}) => ({
  key: id,
  fieldId,
  filter: (0, _patterns.idNameArrayFilter)({
    matches: (fieldValue, filterValue) => {
      if (fieldValue) {
        return filterValue.some(v => v.id === _constants.BooleanFilters.checked);
      }
      return filterValue.some(v => v.id === _constants.BooleanFilters.unchecked);
    }
  }),
  component: _patterns.RadioGroupFilter,
  props: {
    ...commonFilterProps,
    data: [{
      id: 'checked',
      name: (config == null ? void 0 : config.trueLabel) || 'Yes'
    }, {
      id: 'unchecked',
      name: (config == null ? void 0 : config.falseLabel) || 'No'
    }]
  }
});
exports.createBooleanFilter = createBooleanFilter;
const createNumberFilter = ({
  id,
  fieldId,
  commonFilterProps,
  config
}) => ({
  key: id,
  fieldId,
  filter: (0, _patterns.numberRangeFilter)(),
  component: _patterns.NumberRangeFilter,
  props: {
    ...commonFilterProps,
    min: config == null ? void 0 : config.min,
    max: config == null ? void 0 : config.max,
    decimal: config == null ? void 0 : config.allowedDecimals
  }
});
exports.createNumberFilter = createNumberFilter;
const createEnumFilter = ({
  id,
  fieldId,
  commonFilterProps,
  config
}) => {
  const {
    options,
    selectionMode,
    optionType
  } = config;
  const data = (options == null ? void 0 : options.map(option => ({
    id: option.value,
    name: option.label
  }))) ?? [];
  const inlineComponentThreshold = 5;
  let component;
  if (selectionMode === 'single') {
    component = data.length > inlineComponentThreshold ? _patterns.AutoCompleteFilter : _patterns.RadioGroupFilter;
    if (optionType === 'select') {
      component = _patterns.AutoCompleteFilter;
    } else if (optionType === 'radio') {
      component = _patterns.RadioGroupFilter;
    }
  } else {
    component = data.length > inlineComponentThreshold ? _patterns.MultiSelectCheckboxFilter : _patterns.MultiInlineCheckboxFilter;
    if (optionType === 'select') {
      component = _patterns.MultiSelectCollectionFilter;
    } else if (optionType === 'inlineCheckbox') {
      component = _patterns.MultiInlineCheckboxFilter;
    } else if (optionType === 'checkbox') {
      component = _patterns.MultiSelectCheckboxFilter;
    }
  }
  return {
    key: id,
    fieldId,
    filter: (0, _patterns.idNameArrayFilter)(),
    component,
    collectionData: data,
    props: {
      ...commonFilterProps,
      data
    }
  };
};
exports.createEnumFilter = createEnumFilter;
const createReferenceFilter = ({
  id,
  fieldId,
  commonFilterProps,
  config,
  additionalParams
}) => {
  const component = (config == null ? void 0 : config.selectionMode) === 'single' ? _patterns.AutoCompleteFilter : _patterns.MultiSelectCollectionFilter;
  return {
    key: id,
    fieldId,
    filter: (0, _patterns.idNameArrayFilter)(),
    component,
    collectionId: additionalParams.referencedSchema.id,
    props: commonFilterProps
  };
};
exports.createReferenceFilter = createReferenceFilter;
