// @wix/auto_sdk_benefit-programs_transactions/build/es/index.mjs
// src/benefit-programs-v1-transaction-transactions.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/benefit-programs-v1-transaction-transactions.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixBenefitProgramsV1TransactionTransactionServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/pool-transactions",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ],
    _: [
      {
        srcPath: "/_api/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_benefit-programs_transactions";
function getTransaction(payload) {
  function __getTransaction({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.transaction",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.transaction.TransactionService.GetTransaction",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1TransactionTransactionServiceUrl({
        protoPath: "/v1/transactions/{transactionId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "transaction.createdDate" },
            { path: "transaction.updatedDate" },
            { path: "transaction.details.effectiveDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getTransaction;
}
function queryTransactions(payload) {
  function __queryTransactions({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.transaction",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.transaction.TransactionService.QueryTransactions",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1TransactionTransactionServiceUrl({
        protoPath: "/v1/transactions/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "transactions.createdDate" },
            { path: "transactions.updatedDate" },
            { path: "transactions.details.effectiveDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryTransactions;
}

// src/benefit-programs-v1-transaction-transactions.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var BalanceType = /* @__PURE__ */ ((BalanceType2) => {
  BalanceType2["UNDEFINED"] = "UNDEFINED";
  BalanceType2["AVAILABLE"] = "AVAILABLE";
  BalanceType2["EXTERNAL"] = "EXTERNAL";
  return BalanceType2;
})(BalanceType || {});
var IdentityType = /* @__PURE__ */ ((IdentityType2) => {
  IdentityType2["UNKNOWN"] = "UNKNOWN";
  IdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  IdentityType2["MEMBER"] = "MEMBER";
  IdentityType2["WIX_USER"] = "WIX_USER";
  return IdentityType2;
})(IdentityType || {});
var TransactionStatus = /* @__PURE__ */ ((TransactionStatus2) => {
  TransactionStatus2["UNDEFINED"] = "UNDEFINED";
  TransactionStatus2["PENDING"] = "PENDING";
  TransactionStatus2["COMPLETED"] = "COMPLETED";
  TransactionStatus2["FAILED"] = "FAILED";
  return TransactionStatus2;
})(TransactionStatus || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function getTransaction2(transactionId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    transactionId
  });
  const reqOpts = getTransaction(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.transaction;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { transactionId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["transactionId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryTransactions2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryTransactions(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [])
      );
      return {
        items: transformedData?.transactions,
        pagingMetadata: transformedData?.metadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}

// src/benefit-programs-v1-transaction-transactions.public.ts
function getTransaction3(httpClient) {
  return (transactionId) => getTransaction2(
    transactionId,
    // @ts-ignore
    { httpClient }
  );
}
function queryTransactions3(httpClient) {
  return () => queryTransactions2(
    // @ts-ignore
    { httpClient }
  );
}

// src/benefit-programs-v1-transaction-transactions.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
var getTransaction4 = /* @__PURE__ */ createRESTModule(getTransaction3);
var queryTransactions4 = /* @__PURE__ */ createRESTModule(queryTransactions3);
export {
  BalanceType,
  IdentityType,
  SortOrder,
  TransactionStatus,
  WebhookIdentityType,
  getTransaction4 as getTransaction,
  queryTransactions4 as queryTransactions
};
