// @wix/auto_sdk_benefit-programs_transactions/build/es/meta.mjs
// src/benefit-programs-v1-transaction-transactions.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixBenefitProgramsV1TransactionTransactionServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/pool-transactions",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ],
    _: [
      {
        srcPath: "/_api/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/benefit-programs/v1/transactions",
        destPath: "/v1/transactions"
      },
      {
        srcPath: "/benefit-programs/v1/bulk/transactions",
        destPath: "/v1/bulk/transactions"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_benefit-programs_transactions";
function getTransaction(payload) {
  function __getTransaction({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.transaction",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.transaction.TransactionService.GetTransaction",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1TransactionTransactionServiceUrl({
        protoPath: "/v1/transactions/{transactionId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "transaction.createdDate" },
            { path: "transaction.updatedDate" },
            { path: "transaction.details.effectiveDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getTransaction;
}
function queryTransactions(payload) {
  function __queryTransactions({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.transaction",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.transaction.TransactionService.QueryTransactions",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1TransactionTransactionServiceUrl({
        protoPath: "/v1/transactions/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "transactions.createdDate" },
            { path: "transactions.updatedDate" },
            { path: "transactions.details.effectiveDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryTransactions;
}

// src/benefit-programs-v1-transaction-transactions.meta.ts
function getTransaction2() {
  const payload = { transactionId: ":transactionId" };
  const getRequestOptions = getTransaction(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/transactions/{transactionId}",
    pathParams: { transactionId: "transactionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryTransactions2() {
  const payload = {};
  const getRequestOptions = queryTransactions(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/transactions/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  getTransaction2 as getTransaction,
  queryTransactions2 as queryTransactions
};
