// pro-gallery-lib/dist/cjs/settings/options/behaviourParams_gallery_horizontal_autoSlide_behaviour.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const constants_js_1 = require("../utils/constants.js");
const index_js_1 = tslib_1.__importDefault(require("../../common/constants/index.js"));
const utils_js_1 = require("../utils/utils.js");
const optionsMap_js_1 = tslib_1.__importDefault(require("../../core/helpers/optionsMap.js"));
exports.default = {
    title: 'Auto slide Behaviour',
    isRelevant: (options) => {
        return (options[optionsMap_js_1.default.layoutParams.structure.scrollDirection] ===
            index_js_1.default[optionsMap_js_1.default.layoutParams.structure.scrollDirection].HORIZONTAL);
    },
    isRelevantDescription: 'Set a Horizontal gallery ("Scroll Direction" as "Horizontal") and set "Auto Slide" to "true".',
    type: constants_js_1.INPUT_TYPES.OPTIONS,
    default: index_js_1.default[optionsMap_js_1.default.behaviourParams.gallery.horizontal.autoSlide.behaviour].OFF,
    get options() {
        return (0, utils_js_1.createOptions)(optionsMap_js_1.default.behaviourParams.gallery.horizontal.autoSlide.behaviour);
    },
    description: `Choose the type of auto-scrolling to be used when navigation between items, "OFF" (auto-scrolling disabled) "INTERVAL" (time between navigation) or "CONTINOUS" (Continuous scrolling).`,
};
