// pro-gallery-lib/dist/cjs/settings/options/behaviourParams_item_overlay_hoverAnimation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const constants_js_1 = require("../utils/constants.js");
const index_js_1 = tslib_1.__importDefault(require("../../common/constants/index.js"));
const utils_js_1 = require("../utils/utils.js");
const optionsMap_js_1 = tslib_1.__importDefault(require("../../core/helpers/optionsMap.js"));
exports.default = {
    title: 'Overlay Hover Animation',
    isRelevantDescription: 'Set "Hover Effect" to anything but "No Change" or "Never Show".',
    isRelevant: (options) => options[optionsMap_js_1.default.behaviourParams.item.overlay.hoveringBehaviour] !==
        index_js_1.default[optionsMap_js_1.default.behaviourParams.item.overlay.hoveringBehaviour].ALWAYS_SHOW &&
        options[optionsMap_js_1.default.behaviourParams.item.overlay.hoveringBehaviour] !==
            index_js_1.default[optionsMap_js_1.default.behaviourParams.item.overlay.hoveringBehaviour].NEVER_SHOW,
    type: constants_js_1.INPUT_TYPES.OPTIONS,
    default: index_js_1.default[optionsMap_js_1.default.behaviourParams.item.overlay.hoverAnimation].NO_EFFECT,
    get options() {
        return (0, utils_js_1.createOptions)(optionsMap_js_1.default.behaviourParams.item.overlay.hoverAnimation);
    },
    description: `Choose the overlay animation effect to be used when hovering over an item`,
};
