// pro-gallery-lib/dist/cjs/settings/options/behaviourParams_gallery_horizontal_autoSlide_speed.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const constants_js_1 = require("../utils/constants.js");
const index_js_1 = tslib_1.__importDefault(require("../../common/constants/index.js"));
const optionsMap_js_1 = tslib_1.__importDefault(require("../../core/helpers/optionsMap.js"));
const behaviourParams_gallery_horizontal_autoSlide_behaviour_js_1 = tslib_1.__importDefault(require("./behaviourParams_gallery_horizontal_autoSlide_behaviour.js"));
exports.default = {
    title: 'Slide speed',
    description: `Set the slide speed when using "continuous" auto-scrolling.`,
    min: 100,
    max: 1000,
    isRelevant: (options) => {
        return (behaviourParams_gallery_horizontal_autoSlide_behaviour_js_1.default.isRelevant(options) &&
            options[optionsMap_js_1.default.behaviourParams.gallery.horizontal.autoSlide.behaviour] ===
                index_js_1.default[optionsMap_js_1.default.behaviourParams.gallery.horizontal.autoSlide.behaviour].CONTINUOUS);
    },
    type: constants_js_1.INPUT_TYPES.NUMBER,
    isRelevantDescription: 'Set a Horizontal gallery ("Scroll Direction" as "Horizontal") and set "Auto slide Behaviour" to "CONTINUOUS".',
    default: 200,
};
