// pro-gallery-lib/dist/cjs/settings/options/behaviourParams_item_secondaryMedia_behaviour.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const constants_js_1 = require("../utils/constants.js");
const index_js_1 = tslib_1.__importDefault(require("../../common/constants/index.js"));
const utils_js_1 = require("../utils/utils.js");
const behaviourParams_item_secondaryMedia_trigger_js_1 = tslib_1.__importDefault(require("./behaviourParams_item_secondaryMedia_trigger.js"));
const optionsMap_js_1 = tslib_1.__importDefault(require("../../core/helpers/optionsMap.js"));
exports.default = {
    title: 'Second Media Behaivour',
    isRelevant: (options) => {
        return (behaviourParams_item_secondaryMedia_trigger_js_1.default.isRelevant(options) &&
            options[optionsMap_js_1.default.behaviourParams.item.secondaryMedia.trigger] !==
                index_js_1.default[optionsMap_js_1.default.behaviourParams.item.secondaryMedia.trigger].OFF);
    },
    isRelevantDescription: 'Secondary Media Trigger is not "OFF" and when arrowsPosition is MOUSE_CURSOR there must be some space to hover (mouseCursorContainerMaxWidth less then 100)',
    type: constants_js_1.INPUT_TYPES.OPTIONS,
    default: index_js_1.default[optionsMap_js_1.default.behaviourParams.item.secondaryMedia.behaviour].APPEARS,
    get options() {
        return (0, utils_js_1.createOptions)(optionsMap_js_1.default.behaviourParams.item.secondaryMedia.behaviour);
    },
    description: `Select the behaviour for the second media on action.`,
};
