// pro-gallery-lib/dist/cjs/settings/options/behaviourParams_item_video_playTrigger.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const constants_js_1 = require("../utils/constants.js");
const index_js_1 = tslib_1.__importDefault(require("../../common/constants/index.js"));
const utils_js_1 = require("../utils/utils.js");
const optionsMap_js_1 = tslib_1.__importDefault(require("../../core/helpers/optionsMap.js"));
exports.default = {
    title: 'Play Video On',
    description: `Choose when video items will start playing: On hover, on click or autoplay`,
    isRelevant: (options, option) => {
        // Distribution of the specific isRelevant functions
        const videoPlayOptions = {
            [index_js_1.default[optionsMap_js_1.default.behaviourParams.item.video.playTrigger].AUTO]: () => {
                return true;
            },
            [index_js_1.default[optionsMap_js_1.default.behaviourParams.item.video.playTrigger].HOVER]: () => {
                return true;
            },
            [index_js_1.default[optionsMap_js_1.default.behaviourParams.item.video.playTrigger].CLICK]: () => {
                return true;
            },
        };
        // specific option isRelevant : general video play isRelevant (AUTO always true)
        return option ? videoPlayOptions[option](options) : Object.values(videoPlayOptions).some((val) => val(options));
    },
    isRelevantDescription: 'Always Relevant',
    get options() {
        return (0, utils_js_1.createOptions)(optionsMap_js_1.default.behaviourParams.item.video.playTrigger);
    },
    type: constants_js_1.INPUT_TYPES.OPTIONS,
    default: index_js_1.default[optionsMap_js_1.default.behaviourParams.item.video.playTrigger].HOVER, //one source
};
