// pro-gallery-lib/dist/cjs/settings/options/behaviourParams_gallery_vertical_loadMore_amount.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const constants_js_1 = require("../utils/constants.js");
const index_js_1 = tslib_1.__importDefault(require("../../common/constants/index.js"));
const utils_js_1 = require("../utils/utils.js");
const optionsMap_js_1 = tslib_1.__importDefault(require("../../core/helpers/optionsMap.js"));
const behaviourParams_gallery_vertical_loadMore_enable_js_1 = tslib_1.__importDefault(require("./behaviourParams_gallery_vertical_loadMore_enable.js"));
exports.default = {
    title: 'Load More Amount',
    isRelevant: (sp) => behaviourParams_gallery_vertical_loadMore_enable_js_1.default.isRelevant(sp) && sp[optionsMap_js_1.default.behaviourParams.gallery.vertical.loadMore.enable],
    isRelevantDescription: 'Set a Vertical gallery ("Scroll Direction" as "Vertical") and set "Enable load more button" to "true".',
    type: constants_js_1.INPUT_TYPES.OPTIONS,
    default: index_js_1.default[optionsMap_js_1.default.behaviourParams.gallery.vertical.loadMore.amount].ALL,
    get options() {
        return (0, utils_js_1.createOptions)(optionsMap_js_1.default.behaviourParams.gallery.vertical.loadMore.amount);
    },
    description: `Choose whether clicking the 'load more' button toggle an infinite scroll behaviour or adds a few more items and keep the button at the bottom.`,
};
