// pro-gallery-lib/dist/cjs/settings/options/behaviourParams_gallery_horizontal_slideAnimation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const constants_js_1 = require("../utils/constants.js");
const index_js_1 = tslib_1.__importDefault(require("../../common/constants/index.js"));
const utils_js_1 = require("../utils/utils.js");
const optionsMap_js_1 = tslib_1.__importDefault(require("../../core/helpers/optionsMap.js"));
exports.default = {
    title: 'Slide Animation',
    isRelevant: (options) => options[optionsMap_js_1.default.layoutParams.structure.scrollDirection] ===
        index_js_1.default[optionsMap_js_1.default.layoutParams.structure.scrollDirection].HORIZONTAL &&
        options[optionsMap_js_1.default.layoutParams.groups.groupSize] === 1 &&
        options[optionsMap_js_1.default.layoutParams.crop.enable] &&
        JSON.stringify(options[optionsMap_js_1.default.layoutParams.crop.ratios]) === JSON.stringify(['100%/100%']),
    isRelevantDescription: 'Set a Horizontal gallery ("Scroll Direction" as "Horizontal"), set "Max Group Size" to "1" and set "Crop Images" to true.',
    type: constants_js_1.INPUT_TYPES.OPTIONS,
    default: index_js_1.default[optionsMap_js_1.default.behaviourParams.gallery.horizontal.slideAnimation].SCROLL,
    get options() {
        return (0, utils_js_1.createOptions)(optionsMap_js_1.default.behaviourParams.gallery.horizontal.slideAnimation);
    },
    description: `Choose the slide animation effect to be used when navigating between items in a slideshow`,
};
