// pro-gallery-lib/dist/cjs/settings/options/behaviourParams_item_overlay_sizeUnits.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const constants_js_1 = require("../utils/constants.js");
const index_js_1 = tslib_1.__importDefault(require("../../common/constants/index.js"));
const utils_js_1 = require("../utils/utils.js");
const optionsMap_js_1 = tslib_1.__importDefault(require("../../core/helpers/optionsMap.js"));
exports.default = {
    title: 'Overlay Size Type',
    description: `choose pixel or percent`,
    isRelevantDescription: 'Set "Hover Effect" to anything but "Never Show".',
    isRelevant: (options) => options[optionsMap_js_1.default.behaviourParams.item.overlay.hoveringBehaviour] !==
        index_js_1.default[optionsMap_js_1.default.behaviourParams.item.overlay.hoveringBehaviour].NEVER_SHOW,
    type: constants_js_1.INPUT_TYPES.OPTIONS,
    default: index_js_1.default[optionsMap_js_1.default.behaviourParams.item.overlay.sizeUnits].PERCENT,
    get options() {
        return (0, utils_js_1.createOptions)(optionsMap_js_1.default.behaviourParams.item.overlay.sizeUnits);
    },
};
