// pro-gallery-lib/dist/cjs/settings/options/behaviourParams_gallery_scrollAnimation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const constants_js_1 = require("../utils/constants.js");
const index_js_1 = tslib_1.__importDefault(require("../../common/constants/index.js"));
const utils_js_1 = require("../utils/utils.js");
const optionsMap_js_1 = tslib_1.__importDefault(require("../../core/helpers/optionsMap.js"));
exports.default = {
    title: 'Scroll Animation',
    description: `Choose the type of animation to be used when items appear while scrolling vertically through the gallery`,
    isRelevantDescription: 'To enable "Scroll Animation" either set a Vertical gallery ("Scroll Direction" as "Vertical")\nor set a Horizontal gallery ("Scroll Direction" as "Horizontal") with "Slide Animation" set to "Scroll".',
    isRelevant: (options) => options[optionsMap_js_1.default.layoutParams.structure.scrollDirection] ===
        index_js_1.default[optionsMap_js_1.default.layoutParams.structure.scrollDirection].VERTICAL ||
        (options[optionsMap_js_1.default.layoutParams.structure.scrollDirection] ===
            index_js_1.default[optionsMap_js_1.default.layoutParams.structure.scrollDirection].HORIZONTAL &&
            options[optionsMap_js_1.default.behaviourParams.gallery.horizontal.slideAnimation] ===
                index_js_1.default[optionsMap_js_1.default.behaviourParams.gallery.horizontal.slideAnimation].SCROLL),
    get options() {
        return (0, utils_js_1.createOptions)(optionsMap_js_1.default.behaviourParams.gallery.scrollAnimation);
    },
    type: constants_js_1.INPUT_TYPES.OPTIONS,
    default: index_js_1.default[optionsMap_js_1.default.behaviourParams.gallery.scrollAnimation].NO_EFFECT, //one source
};
