// pro-gallery-lib/dist/cjs/settings/options/layoutParams_navigationArrows_position.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const constants_js_1 = require("../utils/constants.js");
const index_js_1 = tslib_1.__importDefault(require("../../common/constants/index.js"));
const utils_js_1 = require("../utils/utils.js");
const layoutParams_navigationArrows_enable_js_1 = tslib_1.__importDefault(require("./layoutParams_navigationArrows_enable.js"));
const optionsMap_js_1 = tslib_1.__importDefault(require("../../core/helpers/optionsMap.js"));
exports.default = {
    title: 'Navigation Arrows Position',
    isRelevant: (options) => layoutParams_navigationArrows_enable_js_1.default.isRelevant(options) &&
        options[optionsMap_js_1.default.layoutParams.navigationArrows.enable],
    isRelevantDescription: 'Set a Horizontal gallery ("Scroll Direction" as "Horizontal") and set "Show Navigation Arrows" to "true".',
    type: constants_js_1.INPUT_TYPES.OPTIONS,
    default: index_js_1.default.layoutParams_navigationArrows_position.ON_GALLERY,
    get options() {
        return (0, utils_js_1.createOptions)('layoutParams_navigationArrows_position');
    },
    description: `Set the position of the navigation arrows in sliders. You can choose to position the arrows on the gallery (ON_GALLERY)
  ,outside the gallery (OUTSIDE_GALLERY- in these case the gallery will margins from both sides to make room for the arrows), or on the mouse cursor (still in experimental mode).);
  `,
};
