// pro-gallery-lib/dist/cjs/settings/options/layoutParams_navigationArrows_type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const constants_js_1 = require("../utils/constants.js");
const index_js_1 = tslib_1.__importDefault(require("../../common/constants/index.js"));
const utils_js_1 = require("../utils/utils.js");
const optionsMap_js_1 = tslib_1.__importDefault(require("../../core/helpers/optionsMap.js"));
exports.default = {
    title: 'Arrows Type',
    isRelevant: (options) => {
        return (options[optionsMap_js_1.default.layoutParams.structure.scrollDirection] ===
            index_js_1.default[optionsMap_js_1.default.layoutParams.structure.scrollDirection].HORIZONTAL &&
            options[optionsMap_js_1.default.layoutParams.navigationArrows.enable]);
    },
    isRelevantDescription: 'Set a Horizontal gallery ("Scroll Direction" as "Horizontal") and set "Show Navigation Arrows" to "true".',
    type: constants_js_1.INPUT_TYPES.OPTIONS,
    default: index_js_1.default[optionsMap_js_1.default.layoutParams.navigationArrows.type].DEFAULT_ARROW,
    get options() {
        return (0, utils_js_1.createOptions)(optionsMap_js_1.default.layoutParams.navigationArrows.type);
    },
    description: `Set the type of arrows you want to display in horizontal galleries`,
};
