// pro-gallery-lib/dist/cjs/settings/options/layoutParams_navigationArrows_mouseCursorContainerMaxWidth.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const constants_js_1 = require("../utils/constants.js");
const index_js_1 = tslib_1.__importDefault(require("../../common/constants/index.js"));
const layoutParams_navigationArrows_position_js_1 = tslib_1.__importDefault(require("./layoutParams_navigationArrows_position.js"));
const optionsMap_js_1 = tslib_1.__importDefault(require("../../core/helpers/optionsMap.js"));
exports.default = {
    title: 'Mouse Cursor Container Max Width',
    isRelevant: (options) => {
        return (layoutParams_navigationArrows_position_js_1.default.isRelevant(options) &&
            options[optionsMap_js_1.default.layoutParams.navigationArrows.position] ===
                index_js_1.default[optionsMap_js_1.default.layoutParams.navigationArrows.position].MOUSE_CURSOR);
    },
    isRelevantDescription: 'Set a Horizontal gallery ("Scroll Direction" as "Horizontal"), set "Show Navigation Arrows" to "true" and set "Navigation Arrows Position" to "Mouse Cursor".',
    type: constants_js_1.INPUT_TYPES.NUMBER,
    min: 10,
    max: 100,
    default: 100,
    description: `Set the size of the container mouse cursor in percentage.`,
};
