// pro-gallery-lib/dist/cjs/settings/options/layoutParams_structure_layoutOrientation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const constants_js_1 = require("../utils/constants.js");
const utils_js_1 = require("../utils/utils.js");
const index_js_1 = tslib_1.__importDefault(require("../../common/constants/index.js"));
const optionsMap_js_1 = tslib_1.__importDefault(require("../../core/helpers/optionsMap.js"));
exports.default = {
    title: 'Layout Orientation',
    description: `Choose between a vertical to horizontal oriented gallery. Note that this option may affect
  other (e.g: "titlePlacement" - you will not be able to change this option when "isVertical" is "false")`,
    isRelevant: (options) => options[optionsMap_js_1.default.layoutParams.structure.scrollDirection] ===
        index_js_1.default[optionsMap_js_1.default.layoutParams.structure.scrollDirection].VERTICAL,
    isRelevantDescription: 'Set a Vertical gallery ("Scroll Direction" as "Vertical").',
    get options() {
        return (0, utils_js_1.createOptions)(optionsMap_js_1.default.layoutParams.structure.layoutOrientation);
    },
    type: constants_js_1.INPUT_TYPES.OPTIONS,
    default: index_js_1.default[optionsMap_js_1.default.layoutParams.structure.layoutOrientation].HORIZONTAL,
};
