// pro-gallery-lib/dist/cjs/settings/options/layoutParams_structure_groupsOrder.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const constants_js_1 = require("../utils/constants.js");
const index_js_1 = tslib_1.__importDefault(require("../../common/constants/index.js"));
const optionsMap_js_1 = tslib_1.__importDefault(require("../../core/helpers/optionsMap.js"));
const utils_js_1 = require("../utils/utils.js");
exports.default = {
    title: 'Arrange Items by Columns Order',
    description: `In a columns layout, items will usually arrange by the shortest column. Use this parametr to force a LTR / RTL order (usfull for Grid like layouts)`,
    type: constants_js_1.INPUT_TYPES.MULTISELECT,
    isRelevant: (options) => options[optionsMap_js_1.default.layoutParams.structure.scrollDirection] ===
        index_js_1.default[optionsMap_js_1.default.layoutParams.structure.scrollDirection].VERTICAL &&
        options[optionsMap_js_1.default.layoutParams.structure.layoutOrientation] ===
            index_js_1.default[optionsMap_js_1.default.layoutParams.structure.layoutOrientation].VERTICAL,
    isRelevantDescription: 'Set "Gallery Orientation" to "Columns" & "Scroll Direction" to "Vertical".',
    get options() {
        return (0, utils_js_1.createOptions)(optionsMap_js_1.default.layoutParams.structure.groupsOrder);
    },
    default: 'BY_HEIGHT',
};
