// pro-gallery-lib/dist/cjs/settings/options/layoutParams_structure_responsiveMode.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const constants_js_1 = require("../utils/constants.js");
const index_js_1 = tslib_1.__importDefault(require("../../common/constants/index.js"));
const utils_js_1 = require("../utils/utils.js");
const optionsMap_js_1 = tslib_1.__importDefault(require("../../core/helpers/optionsMap.js"));
exports.default = {
    title: 'Responsive Type',
    isRelevant: (options) => {
        return (options[optionsMap_js_1.default.layoutParams.structure.scrollDirection] ===
            index_js_1.default[optionsMap_js_1.default.layoutParams.structure.scrollDirection].VERTICAL &&
            options[optionsMap_js_1.default.layoutParams.structure.layoutOrientation] ===
                index_js_1.default[optionsMap_js_1.default.layoutParams.structure.layoutOrientation].VERTICAL);
    },
    isRelevantDescription: 'Set a Vertical gallery ("Scroll Direction" as "Vertical") and set "Layout Orientation" to "Columns".',
    type: constants_js_1.INPUT_TYPES.OPTIONS,
    default: index_js_1.default[optionsMap_js_1.default.layoutParams.structure.responsiveMode].FIT_TO_SCREEN,
    get options() {
        return (0, utils_js_1.createOptions)(optionsMap_js_1.default.layoutParams.structure.responsiveMode);
    },
    description: `Choose between adjusting the number of columns addording to the container
  size or setting it manually and keep it fixed.`,
    alert: 'This option will disable the responsive feature of the gallery!',
};
