// pro-gallery-lib/dist/cjs/settings/options/layoutParams_navigationArrows_verticalAlignment.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const constants_js_1 = require("../utils/constants.js");
const index_js_1 = tslib_1.__importDefault(require("../../common/constants/index.js"));
const utils_js_1 = require("../utils/utils.js");
const layoutParams_navigationArrows_position_js_1 = tslib_1.__importDefault(require("./layoutParams_navigationArrows_position.js"));
const optionsMap_js_1 = tslib_1.__importDefault(require("../../core/helpers/optionsMap.js"));
exports.default = {
    title: 'Navigation Arrows Vertical Position',
    isRelevant: (options) => layoutParams_navigationArrows_position_js_1.default.isRelevant(options) &&
        (options[optionsMap_js_1.default.layoutParams.navigationArrows.position] ===
            index_js_1.default[optionsMap_js_1.default.layoutParams.navigationArrows.position].ON_GALLERY ||
            options[optionsMap_js_1.default.layoutParams.navigationArrows.position] ===
                index_js_1.default[optionsMap_js_1.default.layoutParams.navigationArrows.position].OUTSIDE_GALLERY),
    isRelevantDescription: layoutParams_navigationArrows_position_js_1.default.isRelevantDescription +
        ' and make sure arrow position is set to "on gallery" or "outside gallery"',
    type: constants_js_1.INPUT_TYPES.OPTIONS,
    default: index_js_1.default[optionsMap_js_1.default.layoutParams.navigationArrows.verticalAlignment].ITEM_CENTER,
    get options() {
        return (0, utils_js_1.createOptions)('layoutParams_navigationArrows_verticalAlignment');
    },
    description: `Set the vertical position of the navigation arrows in sliders. You can choose to position the arrows on the center of the whole item, the center of the image or the center of the info`,
};
