// pro-gallery-lib/dist/cjs/settings/options/layoutParams_crop_alignment.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const constants_js_1 = require("../utils/constants.js");
const index_js_1 = tslib_1.__importDefault(require("../../common/constants/index.js"));
const utils_js_1 = require("../utils/utils.js");
const optionsMap_js_1 = tslib_1.__importDefault(require("../../core/helpers/optionsMap.js"));
exports.default = {
    title: 'Fitted Image Position',
    description: `image alignment when crop method is FIT`,
    isRelevant: (options) => options[optionsMap_js_1.default.layoutParams.crop.enable] &&
        options[optionsMap_js_1.default.layoutParams.crop.method] === index_js_1.default[optionsMap_js_1.default.layoutParams.crop.method].FIT,
    isRelevantDescription: 'Set `Crop Images` to true and `Crop Method` to FIT',
    type: constants_js_1.INPUT_TYPES.OPTIONS,
    get options() {
        return (0, utils_js_1.createOptions)('layoutParams_crop_alignment');
    },
    default: index_js_1.default[optionsMap_js_1.default.layoutParams.crop.alignment].CENTER,
};
