// pro-gallery-lib/dist/cjs/settings/options/layoutParams_groups_numberOfGroupsPerRow.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
// NEW STYPEPARAMS METHOD
const constants_js_1 = require("../utils/constants.js");
// import { default as GALLERY_CONSTS } from '../../common/constants/index.js';
const optionsMap_js_1 = tslib_1.__importDefault(require("../../core/helpers/optionsMap.js"));
const index_js_1 = require("../../index.js");
exports.default = {
    title: 'Groups per Row',
    isRelevant: (options) => {
        return (options[optionsMap_js_1.default.layoutParams.structure.scrollDirection] ===
            index_js_1.GALLERY_CONSTS[optionsMap_js_1.default.layoutParams.structure.scrollDirection].VERTICAL &&
            options[optionsMap_js_1.default.layoutParams.structure.layoutOrientation] ===
                index_js_1.GALLERY_CONSTS[optionsMap_js_1.default.layoutParams.structure.layoutOrientation].HORIZONTAL &&
            options[optionsMap_js_1.default.layoutParams.structure.responsiveMode] ===
                index_js_1.GALLERY_CONSTS[optionsMap_js_1.default.layoutParams.structure.responsiveMode].SET_ITEMS_PER_ROW);
    },
    isRelevantDescription: 'Set a Vertical gallery ("Scroll Direction" as "Vertical"), set "Layout Orientation" to "Rows" and set "Responsive Type" to "Set Items Per Row".',
    type: constants_js_1.INPUT_TYPES.NUMBER,
    default: 0,
    min: 0,
    max: 10,
    description: `Set the number of groups per row in the gallery (relevant only to galleries with vertical scroll).
  `,
};
