// pro-gallery-lib/dist/cjs/settings/options/layoutParams_groups_allowedGroupTypes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const constants_js_1 = require("../utils/constants.js");
const index_js_1 = tslib_1.__importDefault(require("../../common/constants/index.js"));
const utils_js_1 = require("../utils/utils.js");
const optionsMap_js_1 = tslib_1.__importDefault(require("../../core/helpers/optionsMap.js"));
exports.default = {
    title: 'Allowed Group Types',
    isRelevant: (options) => options[optionsMap_js_1.default.layoutParams.groups.groupSize] > 1,
    isRelevantDescription: 'Set "Max Group Size" to be greater than 1.',
    type: constants_js_1.INPUT_TYPES.MULTISELECT,
    default: Object.keys(index_js_1.default[optionsMap_js_1.default.layoutParams.groups.allowedGroupTypes]),
    get options() {
        return (0, utils_js_1.createOptions)(optionsMap_js_1.default.layoutParams.groups.allowedGroupTypes);
    },
    description: `The allowed group types in collage. This is an advance option that gives you more control over
  the layout of the gallery by specifying the groups you want in the gallery (e.g: "1" - groups of 1 item, "2v" - groups of 2 vertical items
  and more...).
  `,
};
