// pro-gallery-lib/dist/cjs/settings/options/layoutParams_info_height.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const constants_js_1 = require("../utils/constants.js");
const index_js_1 = tslib_1.__importDefault(require("../../common/constants/index.js"));
const optionsMap_js_1 = tslib_1.__importDefault(require("../../core/helpers/optionsMap.js"));
exports.default = {
    title: 'Text Box Height (Pixels)',
    description: `Set the text box Height when the texts are above or below the item`,
    isRelevantDescription: 'First set a Vertical gallery ("Scroll Direction" as "Vertical").\nThen set "Layout Orientation" to "Columns".\nThen set "Max Group Size" to "1".\nThen set "Texts Placement" to "Show Below" or "Show Above".',
    isRelevant: (options) => {
        return (((options[optionsMap_js_1.default.layoutParams.structure.layoutOrientation] ===
            index_js_1.default[optionsMap_js_1.default.layoutParams.structure.layoutOrientation].VERTICAL &&
            options[optionsMap_js_1.default.layoutParams.structure.scrollDirection] ===
                index_js_1.default[optionsMap_js_1.default.layoutParams.structure.scrollDirection].VERTICAL) ||
            options[optionsMap_js_1.default.layoutParams.structure.scrollDirection] ===
                index_js_1.default[optionsMap_js_1.default.layoutParams.structure.scrollDirection].HORIZONTAL) &&
            options[optionsMap_js_1.default.layoutParams.groups.groupSize] === 1 &&
            index_js_1.default.hasExternalVerticalPlacement(options[optionsMap_js_1.default.layoutParams.info.placement]));
    },
    type: constants_js_1.INPUT_TYPES.NUMBER,
    default: 200,
    min: 0,
    max: 1000,
};
