// pro-gallery-lib/dist/cjs/settings/options/layoutParams_navigationArrows_padding.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const constants_js_1 = require("../utils/constants.js");
const index_js_1 = tslib_1.__importDefault(require("../../common/constants/index.js"));
const layoutParams_navigationArrows_position_js_1 = tslib_1.__importDefault(require("./layoutParams_navigationArrows_position.js"));
const optionsMap_js_1 = tslib_1.__importDefault(require("../../core/helpers/optionsMap.js"));
exports.default = {
    title: 'Navigation Arrows Padding',
    isRelevant: (options) => layoutParams_navigationArrows_position_js_1.default.isRelevant(options) &&
        options[optionsMap_js_1.default.layoutParams.navigationArrows.position] ===
            index_js_1.default[optionsMap_js_1.default.layoutParams.navigationArrows.position].ON_GALLERY,
    isRelevantDescription: 'Set a Horizontal gallery ("Scroll Direction" as "Horizontal"), set "Show Navigation Arrows" to "true" ans set "Navigation Arrows Position" to "On Gallery"',
    type: constants_js_1.INPUT_TYPES.NUMBER,
    min: 0,
    max: 100,
    default: 23,
    description: `Set the left / right padding of the navigation arrows in pixels.`,
};
