// pro-gallery-lib/dist/cjs/settings/options/layoutParams_info_width.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const constants_js_1 = require("../utils/constants.js");
const index_js_1 = tslib_1.__importDefault(require("../../common/constants/index.js"));
const optionsMap_js_1 = tslib_1.__importDefault(require("../../core/helpers/optionsMap.js"));
exports.default = {
    title: 'Text Box Width',
    description: `Set the text box width (accourding to the selected unit) from the column width when on the right side or on the left side.`,
    isRelevant: (options) => {
        return (options[optionsMap_js_1.default.layoutParams.structure.layoutOrientation] ===
            index_js_1.default[optionsMap_js_1.default.layoutParams.structure.layoutOrientation].VERTICAL &&
            options[optionsMap_js_1.default.layoutParams.groups.groupSize] === 1 &&
            options[optionsMap_js_1.default.layoutParams.structure.scrollDirection] ===
                index_js_1.default[optionsMap_js_1.default.layoutParams.structure.scrollDirection].VERTICAL &&
            index_js_1.default.hasExternalHorizontalPlacement(options[optionsMap_js_1.default.layoutParams.info.placement]));
    },
    isRelevantDescription: `First set a Vertical gallery ("Scroll Direction" as "Vertical").\nThen set "Layout Orientation" to "Columns".\nThen set "Max Group Size" to "1".\nThen set "Texts Placement" to "Show on the Right" or "Show on the Left".`,
    type: constants_js_1.INPUT_TYPES.NUMBER,
    default: 50,
    min: 0,
    max: 100,
};
