// pro-gallery-lib/dist/cjs/settings/options/layoutParams_info_layout.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const constants_js_1 = require("../utils/constants.js");
const index_js_1 = tslib_1.__importDefault(require("../../common/constants/index.js"));
const utils_js_1 = require("../utils/utils.js");
const optionsMap_js_1 = tslib_1.__importDefault(require("../../core/helpers/optionsMap.js"));
exports.default = {
    title: 'Choose info layout',
    description: `Choose the layout you want for your texts, you can choose to separate the texts and the items
  so you can style them separately.`,
    isRelevantDescription: 'Set "Max Group Size" to "1", set "Layout Orientation" to "Columns" and set "Texts Placement" to anything but "Show On Hover".',
    isRelevant: (options) => options[optionsMap_js_1.default.layoutParams.groups.groupSize] === 1 &&
        options[optionsMap_js_1.default.layoutParams.structure.layoutOrientation] ===
            index_js_1.default[optionsMap_js_1.default.layoutParams.structure.layoutOrientation].VERTICAL &&
        !index_js_1.default.hasHoverPlacement(options[optionsMap_js_1.default.layoutParams.info.placement]),
    get options() {
        return (0, utils_js_1.createOptions)('layoutParams_info_layout');
    },
    type: constants_js_1.INPUT_TYPES.OPTIONS,
    default: index_js_1.default.layoutParams_info_layout.NO_BACKGROUND,
};
