// pro-gallery-lib/dist/cjs/settings/options/layoutParams_info_placement.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const constants_js_1 = require("../utils/constants.js");
const index_js_1 = tslib_1.__importDefault(require("../../common/constants/index.js"));
const utils_js_1 = require("../utils/utils.js");
const optionsMap_js_1 = tslib_1.__importDefault(require("../../core/helpers/optionsMap.js"));
exports.default = {
    title: 'Texts Placement',
    isRelevantDescription: 'Set "Layout Orientation" to "Columns" and set "Max Group Size" to "1".',
    isRelevant: (options, option) => {
        // specific isRelevant functions
        const isHorizontalInfoCompatibleLayout = (options) => {
            return (options[optionsMap_js_1.default.layoutParams.structure.layoutOrientation] ===
                index_js_1.default[optionsMap_js_1.default.layoutParams.structure.layoutOrientation].VERTICAL &&
                options[optionsMap_js_1.default.layoutParams.groups.groupSize] === 1 &&
                options[optionsMap_js_1.default.layoutParams.structure.scrollDirection] ===
                    index_js_1.default[optionsMap_js_1.default.layoutParams.structure.scrollDirection].VERTICAL);
        };
        const isVerticalInfoCompatibleLayout = (options) => {
            return {
                [index_js_1.default[optionsMap_js_1.default.layoutParams.structure.scrollDirection].VERTICAL]: options[optionsMap_js_1.default.layoutParams.structure.layoutOrientation] ===
                    index_js_1.default[optionsMap_js_1.default.layoutParams.structure.layoutOrientation].VERTICAL &&
                    options[optionsMap_js_1.default.layoutParams.groups.groupSize] === 1,
                [index_js_1.default[optionsMap_js_1.default.layoutParams.structure.scrollDirection].HORIZONTAL]: options[optionsMap_js_1.default.layoutParams.groups.groupSize] === 1,
            }[options[optionsMap_js_1.default.layoutParams.structure.scrollDirection]];
        };
        // Distribution of the specific isRelevant functions
        const placementOptions = {
            OVERLAY: () => {
                return true;
            },
            BELOW: isVerticalInfoCompatibleLayout,
            ABOVE: isVerticalInfoCompatibleLayout,
            RIGHT: isHorizontalInfoCompatibleLayout,
            LEFT: isHorizontalInfoCompatibleLayout,
            ALTERNATE_HORIZONTALLY: isHorizontalInfoCompatibleLayout,
            ALTERNATE_VERTICALLY: isVerticalInfoCompatibleLayout,
        };
        // specific option isRelevant : general titlePlacement isRelevant (Hover always true)
        return option ? placementOptions[option](options) : Object.values(placementOptions).some((val) => val(options));
    },
    type: constants_js_1.INPUT_TYPES.MULTISELECT,
    default: index_js_1.default[optionsMap_js_1.default.layoutParams.info.placement].OVERLAY,
    get options() {
        return (0, utils_js_1.createOptions)(optionsMap_js_1.default.layoutParams.info.placement);
    },
    description: `Choose the the placement of the texts (title and description) relative to the items in the gallery.
  Notes:
   - this option also deals with the hover effects and may overide "hoveringBehaviour" when set to anything but "SHOW_ON_HOVER".
   - you can select multiple values, but only one of each direction (ABOVE / BELOW, RIGHT / LEFT)
  `,
};
