// pro-gallery-lib/dist/cjs/settings/options/stylingParams_itemResolutionMode.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const constants_js_1 = require("../utils/constants.js");
const index_js_1 = tslib_1.__importDefault(require("../../common/constants/index.js"));
const utils_js_1 = require("../utils/utils.js");
const optionsMap_js_1 = tslib_1.__importDefault(require("../../core/helpers/optionsMap.js"));
exports.default = {
    title: 'Item Resolution Mode',
    isRelevant: () => true,
    isRelevantDescription: 'Always relevant',
    type: constants_js_1.INPUT_TYPES.OPTIONS,
    default: index_js_1.default[optionsMap_js_1.default.stylingParams.itemResolutionMode].SCALED_DOWN,
    get options() {
        return (0, utils_js_1.createOptions)(optionsMap_js_1.default.stylingParams.itemResolutionMode);
    },
    description: `Set the resolution loading mode for the images in the gallery,
  use "full" option to load the images in full resolution and the "sclaed down" option to load the images in thir container size.`,
};
