// @wix/patterns/dist/cjs/components/AutoCompleteFilter/AutoCompleteFilterState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.AutoCompleteFilterState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _state = require("../../state");
var _mobx = require("mobx");
var _debounce = _interopRequireDefault(require("lodash/debounce"));
class AutoCompleteFilterState {
  constructor(params) {
    (0, _defineProperty2.default)(this, "dropdown", void 0);
    (0, _defineProperty2.default)(this, "searchInputValue", '');
    (0, _defineProperty2.default)(this, "_scheduleOnInputBlur", (0, _debounce.default)((0, _mobx.action)(searchValue => {
      const {
        dropdown: {
          dropdownSelect
        },
        searchInputValue
      } = this;
      if (searchValue !== searchInputValue) {
        return;
      }
      if (!searchInputValue) {
        dropdownSelect.clear();
      } else {
        this.syncSelectedToInput();
      }
    }), 100));
    this.dropdown = new _state.DropdownCollectionFilterState(params);
    (0, _mobx.makeObservable)(this, {
      syncSelectedToInput: _mobx.action.bound,
      clearSelectionAndSearch: _mobx.action.bound,
      onInputBlur: _mobx.action.bound,
      searchInputValue: _mobx.observable.ref
    });
    const {
      events
    } = this.dropdown.dropdownSelect;
    events.on('syncSelectedItems', this.syncSelectedToInput);
    this.syncSelectedToInput();
  }
  syncSelectedToInput() {
    const {
      dropdown: {
        dropdownSelect,
        collectionFilter: {
          collection
        }
      }
    } = this;
    const [selected] = dropdownSelect.selectedItems;
    collection.search('');
    if (selected) {
      this.searchInputValue = collection.itemName(selected);
    } else {
      this.searchInputValue = '';
    }
  }
  clearSelectionAndSearch() {
    const {
      dropdown: {
        dropdownSelect
      },
      collection
    } = this;
    this.searchInputValue = '';
    dropdownSelect.clear();
    collection.clearSearch();
  }
  onInputBlur() {
    this._scheduleOnInputBlur(this.searchInputValue);
  }
  get collection() {
    const {
      dropdown: {
        collectionFilter: {
          collection
        }
      }
    } = this;
    return collection;
  }
  init() {
    return this.dropdown.init();
  }
}
exports.AutoCompleteFilterState = AutoCompleteFilterState;
