// @wix/unidriver-common/dist/cjs/UniDriverBase/UniDriverBaseList.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UniDriverBaseList = void 0;
const getSelectorsPath_1 = require("./getSelectorsPath");
const UniDriverBaseCommon_1 = require("./UniDriverBaseCommon/UniDriverBaseCommon");
const errors_1 = require("./errors");
class UniDriverBaseList extends UniDriverBaseCommon_1.UniDriverBaseCommon {
    constructor() {
        super(...arguments);
        Object.defineProperty(this, "api", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "createUniDriver", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: (element, idx) => {
                return new this.api.Class({
                    base: element,
                    idx,
                    parent: this.ctx,
                    selector: this.ctx.selector,
                });
            }
        });
    }
    /** @deprecated use .at() */
    get(idx) {
        return this.at(idx);
    }
    at(idx) {
        return new this.api.Class({
            base: async () => {
                const cont = await this.getBaseElement();
                if (!cont[idx]) {
                    throw new errors_1.NoElementWithSelectorError({
                        fullPath: await (0, getSelectorsPath_1.getSelectorsPath)(this.ctx),
                        selector: (await this.ctx.selector) ?? '',
                    });
                }
                else {
                    return cont[idx];
                }
            },
            idx,
            parent: this.ctx,
        });
    }
    async count() {
        try {
            const base = await this.getBaseElement();
            return base.length;
        }
        catch (e) {
            if (e instanceof errors_1.NoElementWithSelectorError &&
                JSON.stringify(e.failedPath) === JSON.stringify(await this.selectorsPath)) {
                return 0;
            }
            throw e;
        }
    }
    filter(fn) {
        return new this.api.ListClass({
            ...this.ctx,
            base: async () => {
                const base = await this.getBaseElement();
                const filterResults = await Promise.all(base.map((element, idx) => {
                    return fn(this.createUniDriver(element, idx), idx);
                }));
                return base.filter((_, i) => {
                    return filterResults[i];
                });
            },
        });
    }
    find(fn) {
        return this.filter(fn).at(0);
    }
    async some(fn) {
        const base = await this.getBaseElement();
        for (let idx = 0; idx < base.length; idx++) {
            if (await fn(this.createUniDriver(base[idx], idx), idx)) {
                return true;
            }
        }
        return false;
    }
    async every(fn) {
        const base = await this.getBaseElement();
        const results = await Promise.all(base.map((element, idx) => {
            return fn(this.createUniDriver(element, idx), idx);
        }));
        return results.every(Boolean);
    }
    async map(fn) {
        const base = await this.runWithinTimeBudget(async () => {
            return await this.getBaseElement();
        });
        return Promise.all(base.map((element, idx) => {
            return fn(this.createUniDriver(element, idx), idx);
        }));
    }
    async text() {
        return this.map((d) => d.text());
    }
    /** internal API */
    getType() {
        return this.api.getType();
    }
    async getBaseElement() {
        // eslint-disable-next-line @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-call
        return typeof this.ctx.base === 'function' ? await this.ctx.base() : this.ctx.base;
    }
}
exports.UniDriverBaseList = UniDriverBaseList;
