// @wix/headless-online-programs/dist/react/core/Program.js
import { jsx as _jsx } from "react/jsx-runtime";
import { useService, WixServices } from '@wix/services-manager-react';
import { createServicesMap } from '@wix/services-manager';
import { ProgramServiceDefinition, ProgramService, } from '../../services/program-service.js';
import { InstructorsService, InstructorsServiceDefinition, } from '../../services/instructors-service.js';
/**
 * Root component that provides the Program service context to its children.
 *
 * @order 1
 * @component
 * @example
 * ```tsx
 * import { Program } from '@wix/online-programs/components';
 *
 * function ProgramPage() {
 *  return (
 *    <Program.Root programServiceConfig={{ program: myProgram }}>
 *      <Program.Title />
 *    </Program.Root>
 *  );
 * }
 * ```
 */
export function Root(props) {
    return (_jsx(WixServices, { servicesMap: createServicesMap().addService(ProgramServiceDefinition, ProgramService, props.programServiceConfig), children: props.children }));
}
/**
 * Headless component that provides access to the complete program data
 *
 * @component
 * @example
 * ```tsx
 * <Program.Root programServiceConfig={{ program: myProgram }}>
 *  <Program.Raw asChild>
 *    {({ program }) => (
 *      <div>
 *        <p>Program ID: {program._id}</p>
 *        <p>Title: {program.description?.title}</p>
 *      </div>
 *    )}
 *  </Program.Raw>
 * </Program.Root>
 * ```
 */
export function Raw(props) {
    const service = useService(ProgramServiceDefinition);
    const program = service.program.get();
    return props.children({
        program,
    });
}
/**
 * Headless component for program title display
 *
 * @component
 * @example
 * ```tsx
 *  <Program.Title asChild>
 *    {({ title }) => (
 *      <h1>{title}</h1>
 *    )}
 *  </Program.Title>
 * ```
 */
export function Title(props) {
    const service = useService(ProgramServiceDefinition);
    const program = service.program.get();
    const title = program.description?.title || '';
    return props.children({
        title,
    });
}
/**
 * Headless component for program description display
 *
 * @component
 * @example
 * ```tsx
 * <Program.Description>
 *  {({ description }) => (
 *    <p>{description}</p>
 *  )}
 * </Program.Description>
 * ```
 */
export function Description(props) {
    const service = useService(ProgramServiceDefinition);
    const program = service.program.get();
    const description = program.description?.details || '';
    return props.children({
        description,
    });
}
/**
 * Headless component for program image display
 *
 * @component
 * @example
 * ```tsx
 * <Program.Image asChild>
 *  {({ src, alt }) => (
 *    <img src={src} alt={alt} />
 *  )}
 * </Program.Image>
 * ```
 */
export function Image(props) {
    const service = useService(ProgramServiceDefinition);
    const program = service.program.get();
    const src = program.description?.image || '';
    const alt = program.description?.title || '';
    return props.children({
        src,
        alt,
    });
}
/**
 * Headless component for program duration display
 *
 * @component
 * @example
 * ```tsx
 * <Program.DurationInDays asChild>
 *  {({ durationInDays, isSelfPaced }) => (
 *    <p>
 *      {isSelfPaced ? 'No Time Limit' : `${durationInDays} days`}
 *    </p>
 *  )}
 * </Program.DurationInDays>
 * ```
 */
export function DurationInDays(props) {
    const service = useService(ProgramServiceDefinition);
    const program = service.program.get();
    const timeline = program.timeline;
    const durationInDays = timeline?.durationInDays || null;
    const isSelfPaced = timeline?.selfPaced || false;
    return props.children({
        durationInDays,
        isSelfPaced,
    });
}
/**
 * Headless component for program price display
 *
 * @component
 * @example
 * ```tsx
 * <Program.Price asChild>
 *  {({ price, formattedPrice }) => (
 *    <div>
 *      {price ? (
 *        <span>{formattedPrice}</span>
 *      ) : (
 *        <span>Free</span>
 *      )}
 *    </div>
 *  )}
 * </Program.Price>
 * ```
 */
export function Price(props) {
    const service = useService(ProgramServiceDefinition);
    const program = service.program.get();
    const price = program.price?.value || null;
    const currency = program.price?.currency || '';
    // TODO: use formatCurrency from stores
    const formattedPrice = price && currency ? `${price} ${currency}` : '';
    return props.children({
        price,
        formattedPrice,
        currency,
    });
}
/**
 * Headless component for program steps count display
 *
 * @component
 * @example
 * ```tsx
 * <Program.StepCount asChild>
 *  {({ stepCount }) => (
 *    <p>{stepCount} Steps</p>
 *  )}
 * </Program.StepCount>
 * ```
 */
export function StepCount(props) {
    const service = useService(ProgramServiceDefinition);
    const program = service.program.get();
    const stepCount = program.contentSummary?.stepCount || 0;
    return props.children({
        stepCount,
    });
}
/**
 * Headless component for program section count display
 *
 * @component
 * @example
 * ```tsx
 * <Program.SectionCount asChild>
 *  {({ sectionCount }) => (
 *    <p>{sectionCount} Sections</p>
 *  )}
 * </Program.SectionCount>
 * ```
 */
export function SectionCount(props) {
    const service = useService(ProgramServiceDefinition);
    const program = service.program.get();
    const sectionCount = program.contentSummary?.sectionCount || 0;
    return props.children({
        sectionCount,
    });
}
/**
 * Headless component for program ID display
 *
 * @component
 * @example
 * ```tsx
 * import { Program } from '@wix/online-programs/components';
 *
 * function ProgramIdDisplay() {
 *   return (
 *     <Program.Id>
 *       {({ id }) => (
 *         <a href={`/program/${id}`}>
 *           View Program
 *         </a>
 *       )}
 *     </Program.Id>
 *   );
 * }
 * ```
 */
export function Id(props) {
    const service = useService(ProgramServiceDefinition);
    const program = service.program.get();
    const id = program._id;
    return props.children({
        id,
    });
}
export function Instructors(props) {
    const instructors = props.instructorsServiceConfig?.instructors || [];
    return (_jsx(WixServices, { servicesMap: createServicesMap().addService(InstructorsServiceDefinition, InstructorsService, props.instructorsServiceConfig), children: typeof props.children === 'function'
            ? props.children({
                instructors,
            })
            : props.children }));
}
