// @wix/auto_sdk_restaurants_item-labels/build/internal/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  SortOrder: () => SortOrder,
  WebhookIdentityType: () => WebhookIdentityType,
  bulkDeleteLabels: () => bulkDeleteLabels4,
  countLabels: () => countLabels4,
  createLabel: () => createLabel4,
  deleteLabel: () => deleteLabel4,
  getLabel: () => getLabel4,
  listLabels: () => listLabels4,
  onItemLabelCreated: () => onItemLabelCreated2,
  onItemLabelDeleted: () => onItemLabelDeleted2,
  onItemLabelUpdated: () => onItemLabelUpdated2,
  queryLabels: () => queryLabels4,
  updateLabel: () => updateLabel4
});
module.exports = __toCommonJS(index_exports);

// src/restaurants-menus-v1-item-label-item-labels.public.ts
var import_rename_all_nested_keys2 = require("@wix/sdk-runtime/rename-all-nested-keys");
var import_image3 = require("@wix/sdk-runtime/transformations/image");
var import_timestamp3 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths3 = require("@wix/sdk-runtime/transformations/transform-paths");
var import_sdk_types = require("@wix/sdk-types");

// src/restaurants-menus-v1-item-label-item-labels.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/restaurants-menus-v1-item-label-item-labels.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressRestaurantsMenusItemLabelsV1RestaurantsMenusItemLabelsUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-labels",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-menus-item-labels",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-labels",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-labels",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-menus-item-labels",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-menus-item-labels",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-menus-item-labels",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-labels",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-menus-item-labels",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/item-labels",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/labels",
        destPath: "/v1/labels"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-labels",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_item-labels";
function createLabel(payload) {
  function __createLabel({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "label.createdDate" },
          { path: "label.updatedDate" },
          { path: "label.icon.urlExpirationDate" },
          { path: "label.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_label",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_item_labels.v1.RestaurantsMenusItemLabels.CreateLabel",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemLabelsV1RestaurantsMenusItemLabelsUrl(
        { protoPath: "/v1/labels", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "label.createdDate" },
            { path: "label.updatedDate" },
            { path: "label.icon.urlExpirationDate" },
            { path: "label.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createLabel;
}
function getLabel(payload) {
  function __getLabel({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_label",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.menus_item_labels.v1.RestaurantsMenusItemLabels.GetLabel",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemLabelsV1RestaurantsMenusItemLabelsUrl(
        { protoPath: "/v1/labels/{labelId}", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "label.createdDate" },
            { path: "label.updatedDate" },
            { path: "label.icon.urlExpirationDate" },
            { path: "label.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getLabel;
}
function listLabels(payload) {
  function __listLabels({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_label",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.menus_item_labels.v1.RestaurantsMenusItemLabels.ListLabels",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemLabelsV1RestaurantsMenusItemLabelsUrl(
        { protoPath: "/v1/labels", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "labels.createdDate" },
            { path: "labels.updatedDate" },
            { path: "labels.icon.urlExpirationDate" },
            { path: "labels.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listLabels;
}
function queryLabels(payload) {
  function __queryLabels({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_label",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_item_labels.v1.RestaurantsMenusItemLabels.QueryLabels",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemLabelsV1RestaurantsMenusItemLabelsUrl(
        { protoPath: "/v1/labels/query", data: payload, host }
      ),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "labels.createdDate" },
            { path: "labels.updatedDate" },
            { path: "labels.icon.urlExpirationDate" },
            { path: "labels.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryLabels;
}
function countLabels(payload) {
  function __countLabels({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_label",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_item_labels.v1.RestaurantsMenusItemLabels.CountLabels",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemLabelsV1RestaurantsMenusItemLabelsUrl(
        { protoPath: "/v1/labels/count", data: payload, host }
      ),
      data: payload
    };
    return metadata;
  }
  return __countLabels;
}
function updateLabel(payload) {
  function __updateLabel({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "label.createdDate" },
          { path: "label.updatedDate" },
          { path: "label.icon.urlExpirationDate" },
          { path: "label.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_label",
      method: "PATCH",
      methodFqn: "com.wixpress.restaurants.menus_item_labels.v1.RestaurantsMenusItemLabels.UpdateLabel",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemLabelsV1RestaurantsMenusItemLabelsUrl(
        { protoPath: "/v1/labels/{label.id}", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "label.createdDate" },
            { path: "label.updatedDate" },
            { path: "label.icon.urlExpirationDate" },
            { path: "label.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateLabel;
}
function deleteLabel(payload) {
  function __deleteLabel({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_label",
      method: "DELETE",
      methodFqn: "com.wixpress.restaurants.menus_item_labels.v1.RestaurantsMenusItemLabels.DeleteLabel",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemLabelsV1RestaurantsMenusItemLabelsUrl(
        { protoPath: "/v1/labels/{labelId}", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteLabel;
}
function bulkDeleteLabels(payload) {
  function __bulkDeleteLabels({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_label",
      method: "DELETE",
      methodFqn: "com.wixpress.restaurants.menus_item_labels.v1.RestaurantsMenusItemLabels.BulkDeleteLabels",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemLabelsV1RestaurantsMenusItemLabelsUrl(
        { protoPath: "/v1/bulk/labels/delete", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __bulkDeleteLabels;
}

// src/restaurants-menus-v1-item-label-item-labels.universal.ts
var import_image = require("@wix/sdk-runtime/transformations/image");
var import_image2 = require("@wix/sdk-runtime/transformations/image");
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createLabel2(label) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ label }),
    [
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [{ path: "label.icon" }]
      }
    ]
  );
  const reqOpts = createLabel(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "label.icon" }]
        }
      ])
    )?.label;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { label: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["label"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getLabel2(labelId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ labelId });
  const reqOpts = getLabel(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "label.icon" }]
        }
      ])
    )?.label;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { labelId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["labelId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listLabels2() {
  const { httpClient, sideEffects } = arguments[0];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({});
  const reqOpts = listLabels(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "labels.icon" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {},
        singleArgumentUnchanged: false
      },
      []
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryLabels2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)(
    {
      func: async (payload) => {
        const reqOpts = queryLabels(payload);
        sideEffects?.onSiteCall?.();
        try {
          const result = await httpClient.request(reqOpts);
          sideEffects?.onSuccess?.(result);
          return result;
        } catch (err) {
          sideEffects?.onError?.(err);
          throw err;
        }
      },
      requestTransformer: (query) => {
        const args = [query, {}];
        return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
          ...args?.[1],
          query: args?.[0]
        });
      },
      responseTransformer: ({ data }) => {
        const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
          (0, import_transform_paths2.transformPaths)(data, [
            {
              transformFn: import_image2.transformRESTImageToSDKImage,
              paths: [{ path: "labels.icon" }]
            }
          ])
        );
        return {
          items: transformedData?.labels,
          pagingMetadata: transformedData?.pagingMetadata
        };
      },
      errorTransformer: (err) => {
        const transformedError = (0, import_transform_error.transformError)(err, {
          spreadPathsToArguments: {},
          explicitPathsToArguments: { query: "$[0]" },
          singleArgumentUnchanged: false
        });
        throw transformedError;
      },
      pagingMethod: "CURSOR",
      transformationPaths: {}
    }
  );
}
async function countLabels2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    filter: options?.filter
  });
  const reqOpts = countLabels(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { filter: "$[0].filter" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateLabel2(_id, label) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ label: { ...label, id: _id } }),
    [
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [{ path: "label.icon" }]
      }
    ]
  );
  const reqOpts = updateLabel(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "label.icon" }]
        }
      ])
    )?.label;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { label: "$[1]" },
        explicitPathsToArguments: { "label.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "label"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteLabel2(labelId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ labelId });
  const reqOpts = deleteLabel(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { labelId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["labelId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkDeleteLabels2(ids) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ ids });
  const reqOpts = bulkDeleteLabels(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { ids: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["ids"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/restaurants-menus-v1-item-label-item-labels.public.ts
function createLabel3(httpClient) {
  return (label) => createLabel2(
    label,
    // @ts-ignore
    { httpClient }
  );
}
function getLabel3(httpClient) {
  return (labelId) => getLabel2(
    labelId,
    // @ts-ignore
    { httpClient }
  );
}
function listLabels3(httpClient) {
  return () => listLabels2(
    // @ts-ignore
    { httpClient }
  );
}
function queryLabels3(httpClient) {
  return () => queryLabels2(
    // @ts-ignore
    { httpClient }
  );
}
function countLabels3(httpClient) {
  return (options) => countLabels2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function updateLabel3(httpClient) {
  return (_id, label) => updateLabel2(
    _id,
    label,
    // @ts-ignore
    { httpClient }
  );
}
function deleteLabel3(httpClient) {
  return (labelId) => deleteLabel2(
    labelId,
    // @ts-ignore
    { httpClient }
  );
}
function bulkDeleteLabels3(httpClient) {
  return (ids) => bulkDeleteLabels2(
    ids,
    // @ts-ignore
    { httpClient }
  );
}
var onItemLabelCreated = (0, import_sdk_types.EventDefinition)(
  "wix.restaurants.menus.v1.item_label_created",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.externalReferenceInfo.lastSyncDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_image3.transformRESTImageToSDKImage,
        paths: [{ path: "entity.icon" }]
      }
    ])
  )
)();
var onItemLabelDeleted = (0, import_sdk_types.EventDefinition)(
  "wix.restaurants.menus.v1.item_label_deleted",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "undefined.createdDate" },
          { path: "undefined.updatedDate" },
          { path: "undefined.externalReferenceInfo.lastSyncDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_image3.transformRESTImageToSDKImage,
        paths: [{ path: "undefined.icon" }]
      }
    ])
  )
)();
var onItemLabelUpdated = (0, import_sdk_types.EventDefinition)(
  "wix.restaurants.menus.v1.item_label_updated",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.externalReferenceInfo.lastSyncDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_image3.transformRESTImageToSDKImage,
        paths: [{ path: "entity.icon" }]
      }
    ])
  )
)();

// src/restaurants-menus-v1-item-label-item-labels.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var import_event_definition_modules = require("@wix/sdk-runtime/event-definition-modules");
var createLabel4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(createLabel3);
var getLabel4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getLabel3);
var listLabels4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(listLabels3);
var queryLabels4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(queryLabels3);
var countLabels4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(countLabels3);
var updateLabel4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(updateLabel3);
var deleteLabel4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(deleteLabel3);
var bulkDeleteLabels4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkDeleteLabels3);
var onItemLabelCreated2 = (0, import_event_definition_modules.createEventModule)(onItemLabelCreated);
var onItemLabelDeleted2 = (0, import_event_definition_modules.createEventModule)(onItemLabelDeleted);
var onItemLabelUpdated2 = (0, import_event_definition_modules.createEventModule)(onItemLabelUpdated);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  SortOrder,
  WebhookIdentityType,
  bulkDeleteLabels,
  countLabels,
  createLabel,
  deleteLabel,
  getLabel,
  listLabels,
  onItemLabelCreated,
  onItemLabelDeleted,
  onItemLabelUpdated,
  queryLabels,
  updateLabel
});
