// @wix/auto_sdk_payments_refunds/build/internal/es/index.typings.mjs
// src/payments-refunds-v1-refund-refunds.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/payments-refunds-v1-refund-refunds.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixPaymentsRefundsV1RefundServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/payments/refunds/v1",
        destPath: "/v1"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/payments/refunds/v1/refunds",
        destPath: "/v1/refunds"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/payments/refunds/v1/refunds",
        destPath: "/v1/refunds"
      }
    ],
    _: [
      {
        srcPath: "/payments/refunds/v1/refunds",
        destPath: "/v1/refunds"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_refunds";
function createRefund(payload) {
  function __createRefund({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "refund.createdDate" }, { path: "refund.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.payments.refunds.v1.refund",
      method: "POST",
      methodFqn: "wix.payments.refunds.v1.RefundService.CreateRefund",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsRefundsV1RefundServiceUrl({
        protoPath: "/v1/refunds",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "refund.createdDate" },
            { path: "refund.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createRefund;
}
function getRefund(payload) {
  function __getRefund({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.refunds.v1.refund",
      method: "GET",
      methodFqn: "wix.payments.refunds.v1.RefundService.GetRefund",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsRefundsV1RefundServiceUrl({
        protoPath: "/v1/refunds/{refundId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "refund.createdDate" },
            { path: "refund.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getRefund;
}
function queryRefunds(payload) {
  function __queryRefunds({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.refunds.v1.refund",
      method: "POST",
      methodFqn: "wix.payments.refunds.v1.RefundService.QueryRefunds",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsRefundsV1RefundServiceUrl({
        protoPath: "/v1/refunds/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "refunds.createdDate" },
            { path: "refunds.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryRefunds;
}

// src/payments-refunds-v1-refund-refunds.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["UNKNOWN_STATUS"] = "UNKNOWN_STATUS";
  Status2["PENDING"] = "PENDING";
  Status2["SUCCEEDED"] = "SUCCEEDED";
  Status2["FAILED"] = "FAILED";
  Status2["REVERSED"] = "REVERSED";
  return Status2;
})(Status || {});
var Initiator = /* @__PURE__ */ ((Initiator2) => {
  Initiator2["UNKNOWN_INITIATOR"] = "UNKNOWN_INITIATOR";
  Initiator2["WIX"] = "WIX";
  Initiator2["API"] = "API";
  Initiator2["PROVIDER"] = "PROVIDER";
  return Initiator2;
})(Initiator || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var RejectionReason = /* @__PURE__ */ ((RejectionReason2) => {
  RejectionReason2["UNKNOWN_REJECTION_REASON"] = "UNKNOWN_REJECTION_REASON";
  RejectionReason2["CHARGE_REFUNDED"] = "CHARGE_REFUNDED";
  RejectionReason2["CHARGE_REFUND_IN_PROGRESS"] = "CHARGE_REFUND_IN_PROGRESS";
  RejectionReason2["CHARGE_DISPUTED"] = "CHARGE_DISPUTED";
  RejectionReason2["CHARGE_REFUND_PERIOD_ENDED"] = "CHARGE_REFUND_PERIOD_ENDED";
  RejectionReason2["CHARGE_UNPAID"] = "CHARGE_UNPAID";
  RejectionReason2["PROVIDER_DOWN"] = "PROVIDER_DOWN";
  RejectionReason2["PROVIDER_NOT_SUPPORTED"] = "PROVIDER_NOT_SUPPORTED";
  RejectionReason2["PAYMENT_METHOD_NOT_SUPPORTED"] = "PAYMENT_METHOD_NOT_SUPPORTED";
  RejectionReason2["MERCHANT_BALANCE_INSUFFICIENT"] = "MERCHANT_BALANCE_INSUFFICIENT";
  RejectionReason2["ONLINE_REFUND_NOT_SUPPORTED"] = "ONLINE_REFUND_NOT_SUPPORTED";
  RejectionReason2["NOT_AUTHORIZED"] = "NOT_AUTHORIZED";
  return RejectionReason2;
})(RejectionReason || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createRefund2(refund, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    refund,
    previouslyRefundedAmount: options?.previouslyRefundedAmount
  });
  const reqOpts = createRefund(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.refund;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          refund: "$[0]",
          previouslyRefundedAmount: "$[1].previouslyRefundedAmount"
        },
        singleArgumentUnchanged: false
      },
      ["refund", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getRefund2(refundId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ refundId });
  const reqOpts = getRefund(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.refund;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { refundId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["refundId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryRefunds2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryRefunds(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [])
      );
      return {
        items: transformedData?.refunds,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
export {
  Initiator,
  RejectionReason,
  SortOrder,
  Status,
  WebhookIdentityType,
  createRefund2 as createRefund,
  getRefund2 as getRefund,
  queryRefunds2 as queryRefunds
};
