// @wix/auto_sdk_payments_refunds/build/internal/es/meta.mjs
// src/payments-refunds-v1-refund-refunds.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixPaymentsRefundsV1RefundServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/payments/refunds/v1",
        destPath: "/v1"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/payments/refunds/v1/refunds",
        destPath: "/v1/refunds"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/payments/refunds/v1/refunds",
        destPath: "/v1/refunds"
      }
    ],
    _: [
      {
        srcPath: "/payments/refunds/v1/refunds",
        destPath: "/v1/refunds"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_refunds";
function createRefund(payload) {
  function __createRefund({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "refund.createdDate" }, { path: "refund.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.payments.refunds.v1.refund",
      method: "POST",
      methodFqn: "wix.payments.refunds.v1.RefundService.CreateRefund",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsRefundsV1RefundServiceUrl({
        protoPath: "/v1/refunds",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "refund.createdDate" },
            { path: "refund.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createRefund;
}
function getRefund(payload) {
  function __getRefund({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.refunds.v1.refund",
      method: "GET",
      methodFqn: "wix.payments.refunds.v1.RefundService.GetRefund",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsRefundsV1RefundServiceUrl({
        protoPath: "/v1/refunds/{refundId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "refund.createdDate" },
            { path: "refund.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getRefund;
}
function queryRefunds(payload) {
  function __queryRefunds({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.refunds.v1.refund",
      method: "POST",
      methodFqn: "wix.payments.refunds.v1.RefundService.QueryRefunds",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsRefundsV1RefundServiceUrl({
        protoPath: "/v1/refunds/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "refunds.createdDate" },
            { path: "refunds.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryRefunds;
}

// src/payments-refunds-v1-refund-refunds.meta.ts
function createRefund2() {
  const payload = {};
  const getRequestOptions = createRefund(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/refunds",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getRefund2() {
  const payload = { refundId: ":refundId" };
  const getRequestOptions = getRefund(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/refunds/{refundId}",
    pathParams: { refundId: "refundId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryRefunds2() {
  const payload = {};
  const getRequestOptions = queryRefunds(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/refunds/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  createRefund2 as createRefund,
  getRefund2 as getRefund,
  queryRefunds2 as queryRefunds
};
