// @wix/auto_sdk_benefit-programs_pools/build/es/meta.mjs
// src/benefit-programs-v1-pool-pools.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixBenefitProgramsV1PoolPoolServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/pools",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/benefit-programs-pools",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/benefits",
        destPath: "/v1/bulk/benefits"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/pools",
        destPath: "/v1/bulk/pools"
      },
      {
        srcPath: "/_api/benefit-programs/v1/pools",
        destPath: "/v1/pools"
      },
      {
        srcPath: "/_api/benefit-programs/v1/benefits",
        destPath: "/v1/benefits"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pools",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/benefits",
        destPath: "/v1/bulk/benefits"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/pools",
        destPath: "/v1/bulk/pools"
      },
      {
        srcPath: "/_api/benefit-programs/v1/pools",
        destPath: "/v1/pools"
      },
      {
        srcPath: "/_api/benefit-programs/v1/benefits",
        destPath: "/v1/benefits"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pools",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/benefits",
        destPath: "/v1/bulk/benefits"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/pools",
        destPath: "/v1/bulk/pools"
      },
      {
        srcPath: "/_api/benefit-programs/v1/pools",
        destPath: "/v1/pools"
      },
      {
        srcPath: "/_api/benefit-programs/v1/benefits",
        destPath: "/v1/benefits"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/benefit-programs-pools",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/benefits",
        destPath: "/v1/bulk/benefits"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/pools",
        destPath: "/v1/bulk/pools"
      },
      {
        srcPath: "/_api/benefit-programs/v1/pools",
        destPath: "/v1/pools"
      },
      {
        srcPath: "/_api/benefit-programs/v1/benefits",
        destPath: "/v1/benefits"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/benefit-programs-pools",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/benefits",
        destPath: "/v1/bulk/benefits"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/pools",
        destPath: "/v1/bulk/pools"
      },
      {
        srcPath: "/_api/benefit-programs/v1/pools",
        destPath: "/v1/pools"
      },
      {
        srcPath: "/_api/benefit-programs/v1/benefits",
        destPath: "/v1/benefits"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pools",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/benefits",
        destPath: "/v1/bulk/benefits"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/pools",
        destPath: "/v1/bulk/pools"
      },
      {
        srcPath: "/_api/benefit-programs/v1/pools",
        destPath: "/v1/pools"
      },
      {
        srcPath: "/_api/benefit-programs/v1/benefits",
        destPath: "/v1/benefits"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pools",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/benefits",
        destPath: "/v1/benefits"
      },
      {
        srcPath: "/_api/benefit-programs/v1/pools",
        destPath: "/v1/pools"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/benefits",
        destPath: "/v1/bulk/benefits"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/pools",
        destPath: "/v1/bulk/pools"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/benefit-programs/v1/bulk/benefits",
        destPath: "/v1/bulk/benefits"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/pools",
        destPath: "/v1/bulk/pools"
      },
      {
        srcPath: "/_api/benefit-programs/v1/pools",
        destPath: "/v1/pools"
      },
      {
        srcPath: "/_api/benefit-programs/v1/benefits",
        destPath: "/v1/benefits"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/benefit-programs/v1/bulk/benefits",
        destPath: "/v1/bulk/benefits"
      },
      {
        srcPath: "/benefit-programs/v1/bulk/pools",
        destPath: "/v1/bulk/pools"
      },
      {
        srcPath: "/benefit-programs/v1/pools",
        destPath: "/v1/pools"
      },
      {
        srcPath: "/benefit-programs/v1/benefits",
        destPath: "/v1/benefits"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_benefit-programs_pools";
function getPool(payload) {
  function __getPool({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.pool.PoolService.GetPool",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolPoolServiceUrl({
        protoPath: "/v1/pools/{poolId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "pool.createdDate" }, { path: "pool.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __getPool;
}
function updatePool(payload) {
  function __updatePool({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "pool.createdDate" }, { path: "pool.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool",
      method: "PATCH",
      methodFqn: "wix.benefit_programs.v1.pool.PoolService.UpdatePool",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolPoolServiceUrl({
        protoPath: "/v1/pools/{pool.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "pool.createdDate" }, { path: "pool.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __updatePool;
}
function queryPools(payload) {
  function __queryPools({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.pool.PoolService.QueryPools",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolPoolServiceUrl({
        protoPath: "/v1/pools/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "pools.createdDate" },
            { path: "pools.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryPools;
}
function redeemBenefit(payload) {
  function __redeemBenefit({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "targetDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.pool.PoolService.RedeemBenefit",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolPoolServiceUrl({
        protoPath: "/v1/benefits/redeem",
        data: serializedData,
        host
      }),
      data: serializedData
    };
    return metadata;
  }
  return __redeemBenefit;
}
function checkBenefitEligibility(payload) {
  function __checkBenefitEligibility({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "targetDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.pool.PoolService.CheckBenefitEligibility",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolPoolServiceUrl({
        protoPath: "/v1/pools/check-eligibility",
        data: serializedData,
        host
      }),
      data: serializedData
    };
    return metadata;
  }
  return __checkBenefitEligibility;
}
function bulkCheckBenefitEligibility(payload) {
  function __bulkCheckBenefitEligibility({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "benefitSelectors.targetDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.pool.PoolService.BulkCheckBenefitEligibility",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolPoolServiceUrl({
        protoPath: "/v1/bulk/benefits/check-eligibility",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "results.benefitSelector.targetDate" }]
        }
      ])
    };
    return metadata;
  }
  return __bulkCheckBenefitEligibility;
}
function getEligibleBenefits(payload) {
  function __getEligibleBenefits({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "targetDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.pool.PoolService.GetEligibleBenefits",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolPoolServiceUrl({
        protoPath: "/v1/pools/eligible-pools",
        data: serializedData,
        host
      }),
      data: serializedData
    };
    return metadata;
  }
  return __getEligibleBenefits;
}
function bulkUpdatePools(payload) {
  function __bulkUpdatePools({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "pools.fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "pools.pool.createdDate" },
          { path: "pools.pool.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.pool.PoolService.BulkUpdatePools",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolPoolServiceUrl({
        protoPath: "/v1/bulk/pools",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.pool.createdDate" },
            { path: "results.pool.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdatePools;
}

// src/benefit-programs-v1-pool-pools.meta.ts
function getPool2() {
  const payload = { poolId: ":poolId" };
  const getRequestOptions = getPool(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/pools/{poolId}",
    pathParams: { poolId: "poolId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updatePool2() {
  const payload = { pool: { id: ":poolId" } };
  const getRequestOptions = updatePool(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/pools/{pool.id}",
    pathParams: { poolId: "poolId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryPools2() {
  const payload = {};
  const getRequestOptions = queryPools(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/pools/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function redeemBenefit2() {
  const payload = {};
  const getRequestOptions = redeemBenefit(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/benefits/redeem",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function checkBenefitEligibility2() {
  const payload = {};
  const getRequestOptions = checkBenefitEligibility(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/pools/check-eligibility",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkCheckBenefitEligibility2() {
  const payload = {};
  const getRequestOptions = bulkCheckBenefitEligibility(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/benefits/check-eligibility",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getEligibleBenefits2() {
  const payload = {};
  const getRequestOptions = getEligibleBenefits(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/pools/eligible-pools",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdatePools2() {
  const payload = {};
  const getRequestOptions = bulkUpdatePools(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/pools",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  bulkCheckBenefitEligibility2 as bulkCheckBenefitEligibility,
  bulkUpdatePools2 as bulkUpdatePools,
  checkBenefitEligibility2 as checkBenefitEligibility,
  getEligibleBenefits2 as getEligibleBenefits,
  getPool2 as getPool,
  queryPools2 as queryPools,
  redeemBenefit2 as redeemBenefit,
  updatePool2 as updatePool
};
