// @wix/auto_sdk_payments_onboarding-availability/build/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  PartnerFlowStatus: () => PartnerFlowStatus,
  RestrictedGoodsCategory: () => RestrictedGoodsCategory,
  RestrictedGoodsFlowStatus: () => RestrictedGoodsFlowStatus,
  Status: () => Status,
  WebhookIdentityType: () => WebhookIdentityType,
  getOnboardingAvailability: () => getOnboardingAvailability2,
  updateCbdFlow: () => updateCbdFlow2,
  updatePartnerFlow: () => updatePartnerFlow2,
  updateRestrictedGoodsFlow: () => updateRestrictedGoodsFlow2
});
module.exports = __toCommonJS(index_typings_exports);

// src/cashier-onboarding-availability-v1-onboarding-availability-onboarding-availability.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/cashier-onboarding-availability-v1-onboarding-availability-onboarding-availability.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixCashierOnboardingAvailabilityV1OnboardingAvailabilityServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/v1/onboarding-availability",
        destPath: "/v1/onboarding-availability"
      },
      {
        srcPath: "/ambassador/v1/onboarding-availability",
        destPath: "/v1/onboarding-availability"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_onboarding-availability";
function getOnboardingAvailability(payload) {
  function __getOnboardingAvailability({ host }) {
    const metadata = {
      entityFqdn: "wix.cashier.onboarding_availability.v1.onboarding_availability",
      method: "GET",
      methodFqn: "wix.cashier.onboarding_availability.v1.OnboardingAvailabilityService.GetOnboardingAvailability",
      packageName: PACKAGE_NAME,
      url: resolveWixCashierOnboardingAvailabilityV1OnboardingAvailabilityServiceUrl(
        { protoPath: "/v1/onboarding-availability", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            {
              path: "onboardingAvailability.cbdFlow.attestationInfo.attestationFormSignedDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getOnboardingAvailability;
}
function updateCbdFlow(payload) {
  function __updateCbdFlow({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "cbdFlow.attestationInfo.attestationFormSignedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.cashier.onboarding_availability.v1.onboarding_availability",
      method: "PATCH",
      methodFqn: "wix.cashier.onboarding_availability.v1.OnboardingAvailabilityService.UpdateCbdFlow",
      packageName: PACKAGE_NAME,
      url: resolveWixCashierOnboardingAvailabilityV1OnboardingAvailabilityServiceUrl(
        {
          protoPath: "/v1/onboarding-availability/cbd-flow",
          data: serializedData,
          host
        }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            {
              path: "onboardingAvailability.cbdFlow.attestationInfo.attestationFormSignedDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateCbdFlow;
}
function updateRestrictedGoodsFlow(payload) {
  function __updateRestrictedGoodsFlow({ host }) {
    const metadata = {
      entityFqdn: "wix.cashier.onboarding_availability.v1.onboarding_availability",
      method: "PATCH",
      methodFqn: "wix.cashier.onboarding_availability.v1.OnboardingAvailabilityService.UpdateRestrictedGoodsFlow",
      packageName: PACKAGE_NAME,
      url: resolveWixCashierOnboardingAvailabilityV1OnboardingAvailabilityServiceUrl(
        {
          protoPath: "/v1/onboarding-availability/restricted-goods-flow",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            {
              path: "onboardingAvailability.cbdFlow.attestationInfo.attestationFormSignedDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateRestrictedGoodsFlow;
}
function updatePartnerFlow(payload) {
  function __updatePartnerFlow({ host }) {
    const metadata = {
      entityFqdn: "wix.cashier.onboarding_availability.v1.onboarding_availability",
      method: "PATCH",
      methodFqn: "wix.cashier.onboarding_availability.v1.OnboardingAvailabilityService.UpdatePartnerFlow",
      packageName: PACKAGE_NAME,
      url: resolveWixCashierOnboardingAvailabilityV1OnboardingAvailabilityServiceUrl(
        {
          protoPath: "/v1/onboarding-availability/partner-flow",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            {
              path: "onboardingAvailability.cbdFlow.attestationInfo.attestationFormSignedDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updatePartnerFlow;
}

// src/cashier-onboarding-availability-v1-onboarding-availability-onboarding-availability.universal.ts
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["UNDEFINED"] = "UNDEFINED";
  Status2["DECLINED"] = "DECLINED";
  Status2["POSSIBLE"] = "POSSIBLE";
  Status2["CONFIRMED"] = "CONFIRMED";
  return Status2;
})(Status || {});
var RestrictedGoodsFlowStatus = /* @__PURE__ */ ((RestrictedGoodsFlowStatus2) => {
  RestrictedGoodsFlowStatus2["UNDEFINED"] = "UNDEFINED";
  RestrictedGoodsFlowStatus2["DECLINED"] = "DECLINED";
  RestrictedGoodsFlowStatus2["CONFIRMED"] = "CONFIRMED";
  return RestrictedGoodsFlowStatus2;
})(RestrictedGoodsFlowStatus || {});
var RestrictedGoodsCategory = /* @__PURE__ */ ((RestrictedGoodsCategory2) => {
  RestrictedGoodsCategory2["UNDEFINED"] = "UNDEFINED";
  RestrictedGoodsCategory2["TOBACCO_ALCOHOL"] = "TOBACCO_ALCOHOL";
  RestrictedGoodsCategory2["FIREARMS_WEAPONS"] = "FIREARMS_WEAPONS";
  RestrictedGoodsCategory2["ADULT"] = "ADULT";
  RestrictedGoodsCategory2["MEDICAL"] = "MEDICAL";
  RestrictedGoodsCategory2["FINANCIAL"] = "FINANCIAL";
  RestrictedGoodsCategory2["TRAVEL_AGENCIES"] = "TRAVEL_AGENCIES";
  RestrictedGoodsCategory2["GAMBLING_LOTTERIES_SKILL_GAMES"] = "GAMBLING_LOTTERIES_SKILL_GAMES";
  RestrictedGoodsCategory2["BINARY_OPTIONS_CRYPTOCURRENCIES"] = "BINARY_OPTIONS_CRYPTOCURRENCIES";
  RestrictedGoodsCategory2["MARKETPLACES"] = "MARKETPLACES";
  RestrictedGoodsCategory2["OTHER"] = "OTHER";
  RestrictedGoodsCategory2["CBD"] = "CBD";
  RestrictedGoodsCategory2["TOBACCO_E_CIGARETTES"] = "TOBACCO_E_CIGARETTES";
  RestrictedGoodsCategory2["ALCOHOL"] = "ALCOHOL";
  RestrictedGoodsCategory2["NUTRACEUTICALS"] = "NUTRACEUTICALS";
  RestrictedGoodsCategory2["MISLEADING_MARKETING"] = "MISLEADING_MARKETING";
  RestrictedGoodsCategory2["INTELLECTUAL_PROPERTY"] = "INTELLECTUAL_PROPERTY";
  return RestrictedGoodsCategory2;
})(RestrictedGoodsCategory || {});
var PartnerFlowStatus = /* @__PURE__ */ ((PartnerFlowStatus2) => {
  PartnerFlowStatus2["UNDEFINED"] = "UNDEFINED";
  PartnerFlowStatus2["DECLINED"] = "DECLINED";
  PartnerFlowStatus2["CONFIRMED"] = "CONFIRMED";
  return PartnerFlowStatus2;
})(PartnerFlowStatus || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function getOnboardingAvailability2() {
  const { httpClient, sideEffects } = arguments[0];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({});
  const reqOpts = getOnboardingAvailability(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {},
        singleArgumentUnchanged: false
      },
      []
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateCbdFlow2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    cbdFlow: options?.cbdFlow
  });
  const reqOpts = updateCbdFlow(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { cbdFlow: "$[0].cbdFlow" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateRestrictedGoodsFlow2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    restrictedGoods: options?.restrictedGoods
  });
  const reqOpts = updateRestrictedGoodsFlow(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { restrictedGoods: "$[0].restrictedGoods" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updatePartnerFlow2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    partnerFlow: options?.partnerFlow
  });
  const reqOpts = updatePartnerFlow(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { partnerFlow: "$[0].partnerFlow" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  PartnerFlowStatus,
  RestrictedGoodsCategory,
  RestrictedGoodsFlowStatus,
  Status,
  WebhookIdentityType,
  getOnboardingAvailability,
  updateCbdFlow,
  updatePartnerFlow,
  updateRestrictedGoodsFlow
});
