// @wix/auto_sdk_payments_onboarding-availability/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  getOnboardingAvailability: () => getOnboardingAvailability2,
  updateCbdFlow: () => updateCbdFlow2,
  updatePartnerFlow: () => updatePartnerFlow2,
  updateRestrictedGoodsFlow: () => updateRestrictedGoodsFlow2
});
module.exports = __toCommonJS(meta_exports);

// src/cashier-onboarding-availability-v1-onboarding-availability-onboarding-availability.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixCashierOnboardingAvailabilityV1OnboardingAvailabilityServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/v1/onboarding-availability",
        destPath: "/v1/onboarding-availability"
      },
      {
        srcPath: "/ambassador/v1/onboarding-availability",
        destPath: "/v1/onboarding-availability"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_onboarding-availability";
function getOnboardingAvailability(payload) {
  function __getOnboardingAvailability({ host }) {
    const metadata = {
      entityFqdn: "wix.cashier.onboarding_availability.v1.onboarding_availability",
      method: "GET",
      methodFqn: "wix.cashier.onboarding_availability.v1.OnboardingAvailabilityService.GetOnboardingAvailability",
      packageName: PACKAGE_NAME,
      url: resolveWixCashierOnboardingAvailabilityV1OnboardingAvailabilityServiceUrl(
        { protoPath: "/v1/onboarding-availability", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            {
              path: "onboardingAvailability.cbdFlow.attestationInfo.attestationFormSignedDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getOnboardingAvailability;
}
function updateCbdFlow(payload) {
  function __updateCbdFlow({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "cbdFlow.attestationInfo.attestationFormSignedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.cashier.onboarding_availability.v1.onboarding_availability",
      method: "PATCH",
      methodFqn: "wix.cashier.onboarding_availability.v1.OnboardingAvailabilityService.UpdateCbdFlow",
      packageName: PACKAGE_NAME,
      url: resolveWixCashierOnboardingAvailabilityV1OnboardingAvailabilityServiceUrl(
        {
          protoPath: "/v1/onboarding-availability/cbd-flow",
          data: serializedData,
          host
        }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            {
              path: "onboardingAvailability.cbdFlow.attestationInfo.attestationFormSignedDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateCbdFlow;
}
function updateRestrictedGoodsFlow(payload) {
  function __updateRestrictedGoodsFlow({ host }) {
    const metadata = {
      entityFqdn: "wix.cashier.onboarding_availability.v1.onboarding_availability",
      method: "PATCH",
      methodFqn: "wix.cashier.onboarding_availability.v1.OnboardingAvailabilityService.UpdateRestrictedGoodsFlow",
      packageName: PACKAGE_NAME,
      url: resolveWixCashierOnboardingAvailabilityV1OnboardingAvailabilityServiceUrl(
        {
          protoPath: "/v1/onboarding-availability/restricted-goods-flow",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            {
              path: "onboardingAvailability.cbdFlow.attestationInfo.attestationFormSignedDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateRestrictedGoodsFlow;
}
function updatePartnerFlow(payload) {
  function __updatePartnerFlow({ host }) {
    const metadata = {
      entityFqdn: "wix.cashier.onboarding_availability.v1.onboarding_availability",
      method: "PATCH",
      methodFqn: "wix.cashier.onboarding_availability.v1.OnboardingAvailabilityService.UpdatePartnerFlow",
      packageName: PACKAGE_NAME,
      url: resolveWixCashierOnboardingAvailabilityV1OnboardingAvailabilityServiceUrl(
        {
          protoPath: "/v1/onboarding-availability/partner-flow",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            {
              path: "onboardingAvailability.cbdFlow.attestationInfo.attestationFormSignedDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updatePartnerFlow;
}

// src/cashier-onboarding-availability-v1-onboarding-availability-onboarding-availability.meta.ts
function getOnboardingAvailability2() {
  const payload = {};
  const getRequestOptions = getOnboardingAvailability(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/onboarding-availability",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateCbdFlow2() {
  const payload = {};
  const getRequestOptions = updateCbdFlow(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/onboarding-availability/cbd-flow",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateRestrictedGoodsFlow2() {
  const payload = {};
  const getRequestOptions = updateRestrictedGoodsFlow(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/onboarding-availability/restricted-goods-flow",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updatePartnerFlow2() {
  const payload = {};
  const getRequestOptions = updatePartnerFlow(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/onboarding-availability/partner-flow",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getOnboardingAvailability,
  updateCbdFlow,
  updatePartnerFlow,
  updateRestrictedGoodsFlow
});
