// @wix/astro/build/dependencies/astro-custom-elements-extensions/browser-runtime/index.js
// src/runtime/handleHmr.ts
function handleHmr({
  customElement,
  tagName
}) {
  for (const node of document.querySelectorAll(tagName)) {
    if ("disconnectedCallback" in node && typeof node.disconnectedCallback === "function") {
      node.disconnectedCallback.apply(node);
    }
    applyCustomElementHmr(node, customElement);
    if ("connectedCallback" in node && typeof node.connectedCallback === "function") {
      node.connectedCallback.apply(node);
    }
  }
}
var replaceMethodsOnElement = (elementNode, newClass) => {
  const prototypeDescriptors = Object.getOwnPropertyDescriptors(
    newClass.prototype
  );
  for (const [name, descriptor] of Object.entries(prototypeDescriptors)) {
    Object.defineProperty(elementNode, name, descriptor);
  }
};
function applyCustomElementHmr(elementNode, NewElementClass) {
  visitPrototypeChain(NewElementClass, (classInheritance) => {
    replaceMethodsOnElement(elementNode, classInheritance);
  });
}
function visitPrototypeChain(rootClass, visitFn) {
  const inheritanceArray = [rootClass];
  let currentClass = Object.getPrototypeOf(
    rootClass
  );
  while (currentClass !== HTMLElement) {
    inheritanceArray.unshift(currentClass);
    currentClass = Object.getPrototypeOf(
      currentClass
    );
  }
  for (const classForVisiting of inheritanceArray) {
    visitFn(classForVisiting);
  }
}

// src/runtime/wrapCustomElement.ts
import { createClient } from "@wix/sdk";
import { site } from "@wix/site";
function wrapCustomElement({
  customElement,
  tagName
}) {
  const { injectAccessTokenFunction } = createContext();
  Object.defineProperty(customElement.prototype, "accessTokenListener", {
    configurable: false,
    value: injectAccessTokenFunction,
    writable: false
  });
  customElements.define(tagName, customElement);
}
function createContext() {
  const host = site.host({
    applicationId: import.meta.env.WIX_CLIENT_ID
  });
  const auth = site.auth();
  const wixClient = createClient({
    auth,
    host
  });
  wixClient.enableContext("module");
  return {
    injectAccessTokenFunction: auth.getAccessTokenInjector()
  };
}
export {
  handleHmr,
  wrapCustomElement
};
