// @wix/fast-gallery-core/dist/cjs/lib/types/customGridTypes/customGridTypesConverters.js
"use strict";

exports.__esModule = true;
exports.makePatternArrayPositive = makePatternArrayPositive;
exports.patternArrayToPatternObject = patternArrayToPatternObject;
exports.patternArrayToString = patternArrayToString;
exports.patternObjectToPatternArray = patternObjectToPatternArray;
exports.stringToPatternObject = stringToPatternObject;
function stringToPatternObject(patternString) {
  const regex = /\[([-?\d\s,]+)\]/g;
  const matches = patternString.matchAll(regex);
  const rows = Array.from(matches, match => {
    const row = match[1].split(',').map(cell => ({
      proportion: parseInt(cell.trim(), 10)
    }));
    return {
      cells: row
    };
  });
  return {
    rows
  };
}
function patternObjectToPatternArray(objModel) {
  return objModel.rows.map(row => row.cells.map(cell => cell.proportion));
}
function patternArrayToPatternObject(arrayModel) {
  return {
    rows: arrayModel.map(row => ({
      cells: row.map(proportion => ({
        proportion
      }))
    }))
  };
}
function patternArrayToString(patternArray) {
  return JSON.stringify(patternArray);
}
function makePatternArrayPositive(arr) {
  return arr.map(row => row.map(item => Math.abs(item)));
}
