// @wix/auto-patterns/dist/esm/components/AutoPatternsCollectionPage/AutoPatternsCollectionPage.js
import React from 'react';
import { CollectionPage } from '@wix/patterns/page';
import { AutoPatternsCollectionPageContent } from '../AutoPatternsCollectionPageContent';
import { usePatternsWizardOverridesContext } from '../../providers';
import { useCollectionPageActions } from '../../hooks/useCollectionPageActions';
export const AutoPatternsCollectionPage = _ref => {
  let {
    configuration
  } = _ref;
  const {
    title,
    subtitle
  } = configuration;
  const actions = useCollectionPageActions({
    collectionPageActions: configuration.actions
  });
  const context = usePatternsWizardOverridesContext();
  return /*#__PURE__*/React.createElement(CollectionPage, {
    height: "100vh",
    dataHook: "auto-patterns-page"
  }, /*#__PURE__*/React.createElement(CollectionPage.Header, {
    title: title,
    subtitle: subtitle,
    primaryAction: actions.primaryActions,
    secondaryActions: actions.secondaryActions
  }), /*#__PURE__*/React.createElement(CollectionPage.Content, null, context != null && context.HeaderAddon ? /*#__PURE__*/React.createElement(context.HeaderAddon, null) : null, /*#__PURE__*/React.createElement(AutoPatternsCollectionPageContent, {
    configuration: configuration
  })));
};
