// @wix/auto_sdk_restaurants_availability-exceptions/build/internal/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  AffectedMethods: () => AffectedMethods,
  SortOrder: () => SortOrder,
  Type: () => Type,
  WebhookIdentityType: () => WebhookIdentityType,
  bulkUpdateAvailabilityExceptionTags: () => bulkUpdateAvailabilityExceptionTags2,
  bulkUpdateAvailabilityExceptionTagsByFilter: () => bulkUpdateAvailabilityExceptionTagsByFilter2,
  bulkUpdateAvailabilityExceptions: () => bulkUpdateAvailabilityExceptions2,
  createAvailabilityException: () => createAvailabilityException2,
  deleteAvailabilityException: () => deleteAvailabilityException2,
  getAvailabilityException: () => getAvailabilityException2,
  queryAvailabilityExceptions: () => queryAvailabilityExceptions2,
  updateAvailabilityException: () => updateAvailabilityException2
});
module.exports = __toCommonJS(index_typings_exports);

// src/restaurants-v1-availability-exception-availability-exceptions.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/restaurants-v1-availability-exception-availability-exceptions.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    _base_domain_: [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "users._base_domain_": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants-availability-exceptions",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_availability-exceptions";
function createAvailabilityException(payload) {
  function __createAvailabilityException({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "availabilityException.createdDate" },
          { path: "availabilityException.updatedDate" },
          { path: "availabilityException.startTime" },
          { path: "availabilityException.endTime" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "POST",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.CreateAvailabilityException",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/availability-exceptions",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "availabilityException.createdDate" },
            { path: "availabilityException.updatedDate" },
            { path: "availabilityException.startTime" },
            { path: "availabilityException.endTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createAvailabilityException;
}
function getAvailabilityException(payload) {
  function __getAvailabilityException({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "GET",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.GetAvailabilityException",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/availability-exceptions/{availabilityExceptionId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "availabilityException.createdDate" },
            { path: "availabilityException.updatedDate" },
            { path: "availabilityException.startTime" },
            { path: "availabilityException.endTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getAvailabilityException;
}
function updateAvailabilityException(payload) {
  function __updateAvailabilityException({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "availabilityException.createdDate" },
          { path: "availabilityException.updatedDate" },
          { path: "availabilityException.startTime" },
          { path: "availabilityException.endTime" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "PATCH",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.UpdateAvailabilityException",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/availability-exceptions/{availabilityException.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "availabilityException.createdDate" },
            { path: "availabilityException.updatedDate" },
            { path: "availabilityException.startTime" },
            { path: "availabilityException.endTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateAvailabilityException;
}
function deleteAvailabilityException(payload) {
  function __deleteAvailabilityException({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "DELETE",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.DeleteAvailabilityException",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/availability-exceptions/{availabilityExceptionId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteAvailabilityException;
}
function queryAvailabilityExceptions(payload) {
  function __queryAvailabilityExceptions({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "POST",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.QueryAvailabilityExceptions",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/availability-exceptions/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "availabilityExceptions.createdDate" },
            { path: "availabilityExceptions.updatedDate" },
            { path: "availabilityExceptions.startTime" },
            { path: "availabilityExceptions.endTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryAvailabilityExceptions;
}
function bulkUpdateAvailabilityExceptions(payload) {
  function __bulkUpdateAvailabilityExceptions({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "availabilityExceptions.mask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "availabilityExceptions.availabilityException.createdDate" },
          { path: "availabilityExceptions.availabilityException.updatedDate" },
          { path: "availabilityExceptions.availabilityException.startTime" },
          { path: "availabilityExceptions.availabilityException.endTime" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "POST",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.BulkUpdateAvailabilityExceptions",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/bulk/availability-exceptions/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.startTime" },
            { path: "results.item.endTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateAvailabilityExceptions;
}
function bulkUpdateAvailabilityExceptionTags(payload) {
  function __bulkUpdateAvailabilityExceptionTags({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "POST",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.BulkUpdateAvailabilityExceptionTags",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/bulk/availability-exceptions/update-tags",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateAvailabilityExceptionTags;
}
function bulkUpdateAvailabilityExceptionTagsByFilter(payload) {
  function __bulkUpdateAvailabilityExceptionTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "POST",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.BulkUpdateAvailabilityExceptionTagsByFilter",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/bulk/availability-exceptions/update-tags-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateAvailabilityExceptionTagsByFilter;
}

// src/restaurants-v1-availability-exception-availability-exceptions.universal.ts
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var AffectedMethods = /* @__PURE__ */ ((AffectedMethods2) => {
  AffectedMethods2["UNKNOWN_AFFECTED_METHODS"] = "UNKNOWN_AFFECTED_METHODS";
  AffectedMethods2["ALL_FULFILLMENT_METHODS"] = "ALL_FULFILLMENT_METHODS";
  AffectedMethods2["SPECIFIC_FULFILLMENT_METHODS"] = "SPECIFIC_FULFILLMENT_METHODS";
  AffectedMethods2["ALL_PICKUP_FULFILLMENT_METHODS"] = "ALL_PICKUP_FULFILLMENT_METHODS";
  AffectedMethods2["ALL_DELIVERY_FULFILLMENT_METHODS"] = "ALL_DELIVERY_FULFILLMENT_METHODS";
  return AffectedMethods2;
})(AffectedMethods || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNSPECIFIED_AVAILABILITY_EXCEPTION_TYPE"] = "UNSPECIFIED_AVAILABILITY_EXCEPTION_TYPE";
  Type2["ORDERING_PAUSED"] = "ORDERING_PAUSED";
  Type2["CLOSURE"] = "CLOSURE";
  return Type2;
})(Type || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createAvailabilityException2(availabilityException) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    availabilityException
  });
  const reqOpts = createAvailabilityException(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.availabilityException;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { availabilityException: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["availabilityException"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getAvailabilityException2(availabilityExceptionId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    availabilityExceptionId
  });
  const reqOpts = getAvailabilityException(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.availabilityException;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { availabilityExceptionId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["availabilityExceptionId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateAvailabilityException2(_id, availabilityException) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    availabilityException: { ...availabilityException, id: _id }
  });
  const reqOpts = updateAvailabilityException(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.availabilityException;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { availabilityException: "$[1]" },
        explicitPathsToArguments: { "availabilityException.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "availabilityException"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteAvailabilityException2(availabilityExceptionId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    availabilityExceptionId
  });
  const reqOpts = deleteAvailabilityException(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { availabilityExceptionId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["availabilityExceptionId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryAvailabilityExceptions2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryAvailabilityExceptions(
        payload
      );
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [])
      );
      return {
        items: transformedData?.availabilityExceptions,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function bulkUpdateAvailabilityExceptions2(availabilityExceptions, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    availabilityExceptions,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkUpdateAvailabilityExceptions(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          availabilityExceptions: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["availabilityExceptions", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateAvailabilityExceptionTags2(availabilityExceptionIds, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    availabilityExceptionIds,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateAvailabilityExceptionTags(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          availabilityExceptionIds: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["availabilityExceptionIds", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateAvailabilityExceptionTagsByFilter2(filter, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    filter,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateAvailabilityExceptionTagsByFilter(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          filter: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["filter", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AffectedMethods,
  SortOrder,
  Type,
  WebhookIdentityType,
  bulkUpdateAvailabilityExceptionTags,
  bulkUpdateAvailabilityExceptionTagsByFilter,
  bulkUpdateAvailabilityExceptions,
  createAvailabilityException,
  deleteAvailabilityException,
  getAvailabilityException,
  queryAvailabilityExceptions,
  updateAvailabilityException
});
