// @wix/auto_sdk_restaurants_order-pacing-policy/build/internal/es/index.typings.mjs
// src/restaurants-order-pacing-v1-policy-order-pacing-policy.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/restaurants-order-pacing-v1-policy-order-pacing-policy.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressRestaurantsPacingPoliciesOrderPacingPoliciesUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/order-pacing/v1/policies",
        destPath: "/v1/policies"
      },
      {
        srcPath: "/restaurants/v1/order-pacing-policies",
        destPath: "/v1"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-order-pacing-policies",
        destPath: "/api"
      },
      {
        srcPath: "/order-pacing/v1/policies",
        destPath: "/v1/policies"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-order-pacing-policies",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-order-pacing-policies",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_order-pacing-policy";
function createPolicy(payload) {
  function __createPolicy({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.order_pacing.v1.policy",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.pacing.policies.OrderPacingPolicies.CreatePolicy",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsPacingPoliciesOrderPacingPoliciesUrl({
        protoPath: "/v1/policies",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __createPolicy;
}
function getPolicy(payload) {
  function __getPolicy({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.order_pacing.v1.policy",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.pacing.policies.OrderPacingPolicies.GetPolicy",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsPacingPoliciesOrderPacingPoliciesUrl({
        protoPath: "/v1/policies",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getPolicy;
}
function updatePolicy(payload) {
  function __updatePolicy({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.order_pacing.v1.policy",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.pacing.policies.OrderPacingPolicies.UpdatePolicy",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsPacingPoliciesOrderPacingPoliciesUrl({
        protoPath: "/v1/policies/{policy.id}",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __updatePolicy;
}

// src/restaurants-order-pacing-v1-policy-order-pacing-policy.universal.ts
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNKNOWN"] = "UNKNOWN";
  Type2["FROM_ZERO_LOCATIONS_TO_ONE_LOCATION"] = "FROM_ZERO_LOCATIONS_TO_ONE_LOCATION";
  Type2["FROM_ONE_LOCATION_TO_ZERO_LOCATIONS"] = "FROM_ONE_LOCATION_TO_ZERO_LOCATIONS";
  return Type2;
})(Type || {});
var SyncLocationsSource = /* @__PURE__ */ ((SyncLocationsSource2) => {
  SyncLocationsSource2["API"] = "API";
  SyncLocationsSource2["LOCATION_EVENT"] = "LOCATION_EVENT";
  SyncLocationsSource2["RESTAURANT_PROVISIONED_EVENT"] = "RESTAURANT_PROVISIONED_EVENT";
  return SyncLocationsSource2;
})(SyncLocationsSource || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createPolicy2(policy) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ policy });
  const reqOpts = createPolicy(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.policy;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { policy: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["policy"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getPolicy2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    locationId: options?.locationId
  });
  const reqOpts = getPolicy(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { locationId: "$[0].locationId" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updatePolicy2(_id, policy) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    policy: { ...policy, id: _id }
  });
  const reqOpts = updatePolicy(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { policy: "$[1]" },
        explicitPathsToArguments: { "policy.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "policy"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
export {
  SyncLocationsSource,
  Type,
  WebhookIdentityType,
  createPolicy2 as createPolicy,
  getPolicy2 as getPolicy,
  updatePolicy2 as updatePolicy
};
