// @wix/auto_sdk_restaurants_items/build/internal/es/index.mjs
// src/restaurants-menus-v1-item-items.public.ts
import { renameKeysFromRESTResponseToSDKResponse as renameKeysFromRESTResponseToSDKResponse2 } from "@wix/sdk-runtime/rename-all-nested-keys";
import { transformRESTImageToSDKImage as transformRESTImageToSDKImage2 } from "@wix/sdk-runtime/transformations/image";
import { transformRESTTimestampToSDKTimestamp as transformRESTTimestampToSDKTimestamp2 } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths as transformPaths3 } from "@wix/sdk-runtime/transformations/transform-paths";
import { EventDefinition } from "@wix/sdk-types";

// src/restaurants-menus-v1-item-items.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/restaurants-menus-v1-item-items.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKFloatToRESTFloat } from "@wix/sdk-runtime/transformations/float";
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressRestaurantsMenusItemV1RestaurantsMenusItemUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-menus-item",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/menus-item",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/items",
        destPath: "/v1/bulk/items"
      },
      {
        srcPath: "/restaurants/menus/v1/items",
        destPath: "/v1/items"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
function resolveComWixpressRestaurantsMenusItemV1RestaurantsDuplicateItemRestaurantsDuplicateItemUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-menus-item",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/menus-item",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/items",
        destPath: "/v1/bulk/items"
      },
      {
        srcPath: "/restaurants/menus/v1/items",
        destPath: "/v1/items"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_items";
function duplicateItem(payload) {
  function __duplicateItem({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_item.v1.restaurants_duplicate_item.RestaurantsDuplicateItem.DuplicateItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemV1RestaurantsDuplicateItemRestaurantsDuplicateItemUrl(
        { protoPath: "/v1/items/{id}/duplicate", data: payload, host }
      ),
      data: payload
    };
    return metadata;
  }
  return __duplicateItem;
}
function createItem(payload) {
  function __createItem({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "item.createdDate" },
          { path: "item.updatedDate" },
          { path: "item.image.urlExpirationDate" },
          { path: "item.additionalImages.urlExpirationDate" },
          {
            path: "item.orderSettings.availability.timestampRangesOptions.ranges.startTime"
          },
          {
            path: "item.orderSettings.availability.timestampRangesOptions.ranges.endTime"
          },
          { path: "item.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_item.v1.RestaurantsMenusItem.CreateItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemV1RestaurantsMenusItemUrl({
        protoPath: "/v1/items",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "item.createdDate" },
            { path: "item.updatedDate" },
            { path: "item.image.urlExpirationDate" },
            { path: "item.additionalImages.urlExpirationDate" },
            {
              path: "item.orderSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "item.orderSettings.availability.timestampRangesOptions.ranges.endTime"
            },
            { path: "item.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createItem;
}
function bulkCreateItems(payload) {
  function __bulkCreateItems({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "items.createdDate" },
          { path: "items.updatedDate" },
          { path: "items.image.urlExpirationDate" },
          { path: "items.additionalImages.urlExpirationDate" },
          {
            path: "items.orderSettings.availability.timestampRangesOptions.ranges.startTime"
          },
          {
            path: "items.orderSettings.availability.timestampRangesOptions.ranges.endTime"
          },
          { path: "items.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_item.v1.RestaurantsMenusItem.BulkCreateItems",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemV1RestaurantsMenusItemUrl({
        protoPath: "/v1/bulk/items/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.image.urlExpirationDate" },
            { path: "results.item.additionalImages.urlExpirationDate" },
            {
              path: "results.item.orderSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "results.item.orderSettings.availability.timestampRangesOptions.ranges.endTime"
            },
            { path: "results.item.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateItems;
}
function getItem(payload) {
  function __getItem({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.menus_item.v1.RestaurantsMenusItem.GetItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemV1RestaurantsMenusItemUrl({
        protoPath: "/v1/items/{itemId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "item.createdDate" },
            { path: "item.updatedDate" },
            { path: "item.image.urlExpirationDate" },
            { path: "item.additionalImages.urlExpirationDate" },
            {
              path: "item.orderSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "item.orderSettings.availability.timestampRangesOptions.ranges.endTime"
            },
            { path: "item.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getItem;
}
function listItems(payload) {
  function __listItems({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.menus_item.v1.RestaurantsMenusItem.ListItems",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemV1RestaurantsMenusItemUrl({
        protoPath: "/v1/items",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "items.createdDate" },
            { path: "items.updatedDate" },
            { path: "items.image.urlExpirationDate" },
            { path: "items.additionalImages.urlExpirationDate" },
            {
              path: "items.orderSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "items.orderSettings.availability.timestampRangesOptions.ranges.endTime"
            },
            { path: "items.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listItems;
}
function queryItems(payload) {
  function __queryItems({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_item.v1.RestaurantsMenusItem.QueryItems",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemV1RestaurantsMenusItemUrl({
        protoPath: "/v1/items/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "items.createdDate" },
            { path: "items.updatedDate" },
            { path: "items.image.urlExpirationDate" },
            { path: "items.additionalImages.urlExpirationDate" },
            {
              path: "items.orderSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "items.orderSettings.availability.timestampRangesOptions.ranges.endTime"
            },
            { path: "items.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryItems;
}
function searchItems(payload) {
  function __searchItems({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "search.aggregations.range.buckets.from" },
          { path: "search.aggregations.range.buckets.to" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.menus_item.v1.RestaurantsMenusItem.SearchItems",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemV1RestaurantsMenusItemUrl({
        protoPath: "/v1/items/search",
        data: serializedData,
        host
      }),
      params: toURLSearchParams(serializedData, true),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "items.createdDate" },
            { path: "items.updatedDate" },
            { path: "items.image.urlExpirationDate" },
            { path: "items.additionalImages.urlExpirationDate" },
            {
              path: "items.orderSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "items.orderSettings.availability.timestampRangesOptions.ranges.endTime"
            },
            { path: "items.externalReferenceInfo.lastSyncDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "aggregationData.results.ranges.results.from" },
            { path: "aggregationData.results.ranges.results.to" },
            { path: "aggregationData.results.scalar.value" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveComWixpressRestaurantsMenusItemV1RestaurantsMenusItemUrl({
            protoPath: "/v1/items/search",
            data: serializedData,
            host
          }),
          data: serializedData
        }
      ]
    };
    return metadata;
  }
  return __searchItems;
}
function countItems(payload) {
  function __countItems({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_item.v1.RestaurantsMenusItem.CountItems",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemV1RestaurantsMenusItemUrl({
        protoPath: "/v1/items/count",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __countItems;
}
function updateItem(payload) {
  function __updateItem({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "item.createdDate" },
          { path: "item.updatedDate" },
          { path: "item.image.urlExpirationDate" },
          { path: "item.additionalImages.urlExpirationDate" },
          {
            path: "item.orderSettings.availability.timestampRangesOptions.ranges.startTime"
          },
          {
            path: "item.orderSettings.availability.timestampRangesOptions.ranges.endTime"
          },
          { path: "item.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item",
      method: "PATCH",
      methodFqn: "com.wixpress.restaurants.menus_item.v1.RestaurantsMenusItem.UpdateItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemV1RestaurantsMenusItemUrl({
        protoPath: "/v1/items/{item.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "item.createdDate" },
            { path: "item.updatedDate" },
            { path: "item.image.urlExpirationDate" },
            { path: "item.additionalImages.urlExpirationDate" },
            {
              path: "item.orderSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "item.orderSettings.availability.timestampRangesOptions.ranges.endTime"
            },
            { path: "item.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateItem;
}
function bulkUpdateItem(payload) {
  function __bulkUpdateItem({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "items.mask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "items.item.createdDate" },
          { path: "items.item.updatedDate" },
          { path: "items.item.image.urlExpirationDate" },
          { path: "items.item.additionalImages.urlExpirationDate" },
          {
            path: "items.item.orderSettings.availability.timestampRangesOptions.ranges.startTime"
          },
          {
            path: "items.item.orderSettings.availability.timestampRangesOptions.ranges.endTime"
          },
          { path: "items.item.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_item.v1.RestaurantsMenusItem.BulkUpdateItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemV1RestaurantsMenusItemUrl({
        protoPath: "/v1/bulk/items/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.image.urlExpirationDate" },
            { path: "results.item.additionalImages.urlExpirationDate" },
            {
              path: "results.item.orderSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "results.item.orderSettings.availability.timestampRangesOptions.ranges.endTime"
            },
            { path: "results.item.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateItem;
}
function deleteItem(payload) {
  function __deleteItem({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item",
      method: "DELETE",
      methodFqn: "com.wixpress.restaurants.menus_item.v1.RestaurantsMenusItem.DeleteItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemV1RestaurantsMenusItemUrl({
        protoPath: "/v1/items/{itemId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteItem;
}
function bulkDeleteItems(payload) {
  function __bulkDeleteItems({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item",
      method: "DELETE",
      methodFqn: "com.wixpress.restaurants.menus_item.v1.RestaurantsMenusItem.BulkDeleteItems",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemV1RestaurantsMenusItemUrl({
        protoPath: "/v1/bulk/items/delete",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __bulkDeleteItems;
}

// src/restaurants-menus-v1-item-items.universal.ts
import { transformSDKImageToRESTImage } from "@wix/sdk-runtime/transformations/image";
import { transformRESTImageToSDKImage } from "@wix/sdk-runtime/transformations/image";
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var AvailabilityType = /* @__PURE__ */ ((AvailabilityType2) => {
  AvailabilityType2["UNSPECIFIED_AVAILABILITY_OPTION"] = "UNSPECIFIED_AVAILABILITY_OPTION";
  AvailabilityType2["ALWAYS_AVAILABLE"] = "ALWAYS_AVAILABLE";
  AvailabilityType2["WEEKLY_SCHEDULE"] = "WEEKLY_SCHEDULE";
  AvailabilityType2["TIMESTAMP_RANGES"] = "TIMESTAMP_RANGES";
  return AvailabilityType2;
})(AvailabilityType || {});
var EntitiesDayOfWeek = /* @__PURE__ */ ((EntitiesDayOfWeek2) => {
  EntitiesDayOfWeek2["MON"] = "MON";
  EntitiesDayOfWeek2["TUE"] = "TUE";
  EntitiesDayOfWeek2["WED"] = "WED";
  EntitiesDayOfWeek2["THU"] = "THU";
  EntitiesDayOfWeek2["FRI"] = "FRI";
  EntitiesDayOfWeek2["SAT"] = "SAT";
  EntitiesDayOfWeek2["SUN"] = "SUN";
  return EntitiesDayOfWeek2;
})(EntitiesDayOfWeek || {});
var Enum = /* @__PURE__ */ ((Enum2) => {
  Enum2["UNKNOWN"] = "UNKNOWN";
  Enum2["PROTECTED"] = "PROTECTED";
  Enum2["PRIVATE"] = "PRIVATE";
  Enum2["PUBLIC"] = "PUBLIC";
  Enum2["UNCHANGED"] = "UNCHANGED";
  Enum2["GROUP_PROTECTED"] = "GROUP_PROTECTED";
  return Enum2;
})(Enum || {});
var State = /* @__PURE__ */ ((State2) => {
  State2["Unknown"] = "Unknown";
  State2["Off"] = "Off";
  State2["On"] = "On";
  return State2;
})(State || {});
var PlacementType = /* @__PURE__ */ ((PlacementType2) => {
  PlacementType2["BEFORE"] = "BEFORE";
  PlacementType2["AFTER"] = "AFTER";
  PlacementType2["REPLACE"] = "REPLACE";
  return PlacementType2;
})(PlacementType || {});
var DayOfWeek = /* @__PURE__ */ ((DayOfWeek2) => {
  DayOfWeek2["MONDAY"] = "MONDAY";
  DayOfWeek2["TUESDAY"] = "TUESDAY";
  DayOfWeek2["WEDNESDAY"] = "WEDNESDAY";
  DayOfWeek2["THURSDAY"] = "THURSDAY";
  DayOfWeek2["FRIDAY"] = "FRIDAY";
  DayOfWeek2["SATURDAY"] = "SATURDAY";
  DayOfWeek2["SUNDAY"] = "SUNDAY";
  return DayOfWeek2;
})(DayOfWeek || {});
var ResolutionMethod = /* @__PURE__ */ ((ResolutionMethod2) => {
  ResolutionMethod2["QUERY_PARAM"] = "QUERY_PARAM";
  ResolutionMethod2["SUBDOMAIN"] = "SUBDOMAIN";
  ResolutionMethod2["SUBDIRECTORY"] = "SUBDIRECTORY";
  return ResolutionMethod2;
})(ResolutionMethod || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var SortType = /* @__PURE__ */ ((SortType2) => {
  SortType2["COUNT"] = "COUNT";
  SortType2["VALUE"] = "VALUE";
  return SortType2;
})(SortType || {});
var SortDirection = /* @__PURE__ */ ((SortDirection2) => {
  SortDirection2["DESC"] = "DESC";
  SortDirection2["ASC"] = "ASC";
  return SortDirection2;
})(SortDirection || {});
var ScalarType = /* @__PURE__ */ ((ScalarType2) => {
  ScalarType2["UNKNOWN_SCALAR_TYPE"] = "UNKNOWN_SCALAR_TYPE";
  ScalarType2["COUNT_DISTINCT"] = "COUNT_DISTINCT";
  ScalarType2["MIN"] = "MIN";
  ScalarType2["MAX"] = "MAX";
  return ScalarType2;
})(ScalarType || {});
var AggregationType = /* @__PURE__ */ ((AggregationType2) => {
  AggregationType2["UNKNOWN_AGGREGATION_TYPE"] = "UNKNOWN_AGGREGATION_TYPE";
  AggregationType2["VALUE"] = "VALUE";
  AggregationType2["RANGE"] = "RANGE";
  AggregationType2["SCALAR"] = "SCALAR";
  return AggregationType2;
})(AggregationType || {});
var Mode = /* @__PURE__ */ ((Mode2) => {
  Mode2["OR"] = "OR";
  Mode2["AND"] = "AND";
  return Mode2;
})(Mode || {});
async function duplicateItem2(_id, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    id: _id,
    businessLocationIds: options?.businessLocationIds
  });
  const reqOpts = duplicateItem(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          id: "$[0]",
          businessLocationIds: "$[1].businessLocationIds"
        },
        singleArgumentUnchanged: false
      },
      ["_id", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function createItem2(item) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({ item }),
    [
      {
        transformFn: transformSDKImageToRESTImage,
        paths: [
          { path: "item.image" },
          { path: "item.additionalImages", isRepeated: true }
        ]
      }
    ]
  );
  const reqOpts = createItem(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [
            { path: "item.image" },
            { path: "item.additionalImages", isRepeated: true }
          ]
        }
      ])
    )?.item;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { item: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["item"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCreateItems2(items, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({
      items,
      returnEntity: options?.returnEntity
    }),
    [
      {
        transformFn: transformSDKImageToRESTImage,
        paths: [
          { path: "items.image" },
          { path: "items.additionalImages", isRepeated: true }
        ]
      }
    ]
  );
  const reqOpts = bulkCreateItems(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [
            { path: "results.item.image" },
            { path: "results.item.additionalImages", isRepeated: true }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          items: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["items", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getItem2(itemId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ itemId });
  const reqOpts = getItem(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [
            { path: "item.image" },
            { path: "item.additionalImages", isRepeated: true }
          ]
        }
      ])
    )?.item;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { itemId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["itemId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listItems2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    itemIds: options?.itemIds,
    paging: options?.paging,
    onlyVisible: options?.onlyVisible
  });
  const reqOpts = listItems(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [
            { path: "items.image" },
            { path: "items.additionalImages", isRepeated: true }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          itemIds: "$[0].itemIds",
          paging: "$[0].paging",
          onlyVisible: "$[0].onlyVisible"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryItems2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryItems(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [
          {
            transformFn: transformRESTImageToSDKImage,
            paths: [
              { path: "items.image" },
              { path: "items.additionalImages", isRepeated: true }
            ]
          }
        ])
      );
      return {
        items: transformedData?.items,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function searchItems2(search) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ search });
  const reqOpts = searchItems(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [
            { path: "items.image" },
            { path: "items.additionalImages", isRepeated: true }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { search: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["search"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function countItems2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    filter: options?.filter,
    searchDetails: options?.searchDetails
  });
  const reqOpts = countItems(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          filter: "$[0].filter",
          searchDetails: "$[0].searchDetails"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateItem2(_id, item) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({ item: { ...item, id: _id } }),
    [
      {
        transformFn: transformSDKImageToRESTImage,
        paths: [
          { path: "item.image" },
          { path: "item.additionalImages", isRepeated: true }
        ]
      }
    ]
  );
  const reqOpts = updateItem(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [
            { path: "item.image" },
            { path: "item.additionalImages", isRepeated: true }
          ]
        }
      ])
    )?.item;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { item: "$[1]" },
        explicitPathsToArguments: { "item.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "item"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateItem2(items, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({
      items,
      returnEntity: options?.returnEntity
    }),
    [
      {
        transformFn: transformSDKImageToRESTImage,
        paths: [
          { path: "items.item.image" },
          { path: "items.item.additionalImages", isRepeated: true }
        ]
      }
    ]
  );
  const reqOpts = bulkUpdateItem(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [
            { path: "results.item.image" },
            { path: "results.item.additionalImages", isRepeated: true }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          items: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["items", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteItem2(itemId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ itemId });
  const reqOpts = deleteItem(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { itemId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["itemId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkDeleteItems2(ids) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ ids });
  const reqOpts = bulkDeleteItems(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { ids: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["ids"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/restaurants-menus-v1-item-items.public.ts
function duplicateItem3(httpClient) {
  return (_id, options) => duplicateItem2(
    _id,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function createItem3(httpClient) {
  return (item) => createItem2(
    item,
    // @ts-ignore
    { httpClient }
  );
}
function bulkCreateItems3(httpClient) {
  return (items, options) => bulkCreateItems2(
    items,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function getItem3(httpClient) {
  return (itemId) => getItem2(
    itemId,
    // @ts-ignore
    { httpClient }
  );
}
function listItems3(httpClient) {
  return (options) => listItems2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function queryItems3(httpClient) {
  return () => queryItems2(
    // @ts-ignore
    { httpClient }
  );
}
function searchItems3(httpClient) {
  return (search) => searchItems2(
    search,
    // @ts-ignore
    { httpClient }
  );
}
function countItems3(httpClient) {
  return (options) => countItems2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function updateItem3(httpClient) {
  return (_id, item) => updateItem2(
    _id,
    item,
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdateItem3(httpClient) {
  return (items, options) => bulkUpdateItem2(
    items,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function deleteItem3(httpClient) {
  return (itemId) => deleteItem2(
    itemId,
    // @ts-ignore
    { httpClient }
  );
}
function bulkDeleteItems3(httpClient) {
  return (ids) => bulkDeleteItems2(
    ids,
    // @ts-ignore
    { httpClient }
  );
}
var onItemCreated = EventDefinition(
  "wix.restaurants.menus.v1.item_created",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          {
            path: "entity.orderSettings.availability.timestampRangesOptions.ranges.startTime"
          },
          {
            path: "entity.orderSettings.availability.timestampRangesOptions.ranges.endTime"
          },
          { path: "entity.externalReferenceInfo.lastSyncDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: transformRESTImageToSDKImage2,
        paths: [
          { path: "entity.image" },
          { path: "entity.additionalImages", isRepeated: true }
        ]
      }
    ])
  )
)();
var onItemDeleted = EventDefinition(
  "wix.restaurants.menus.v1.item_deleted",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          {
            path: "entity.orderSettings.availability.timestampRangesOptions.ranges.startTime"
          },
          {
            path: "entity.orderSettings.availability.timestampRangesOptions.ranges.endTime"
          },
          { path: "entity.externalReferenceInfo.lastSyncDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: transformRESTImageToSDKImage2,
        paths: [
          { path: "entity.image" },
          { path: "entity.additionalImages", isRepeated: true }
        ]
      }
    ])
  )
)();
var onItemUpdated = EventDefinition(
  "wix.restaurants.menus.v1.item_updated",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          {
            path: "entity.orderSettings.availability.timestampRangesOptions.ranges.startTime"
          },
          {
            path: "entity.orderSettings.availability.timestampRangesOptions.ranges.endTime"
          },
          { path: "entity.externalReferenceInfo.lastSyncDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: transformRESTImageToSDKImage2,
        paths: [
          { path: "entity.image" },
          { path: "entity.additionalImages", isRepeated: true }
        ]
      }
    ])
  )
)();

// src/restaurants-menus-v1-item-items.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
import { createEventModule } from "@wix/sdk-runtime/event-definition-modules";
var duplicateItem4 = /* @__PURE__ */ createRESTModule(duplicateItem3);
var createItem4 = /* @__PURE__ */ createRESTModule(createItem3);
var bulkCreateItems4 = /* @__PURE__ */ createRESTModule(bulkCreateItems3);
var getItem4 = /* @__PURE__ */ createRESTModule(getItem3);
var listItems4 = /* @__PURE__ */ createRESTModule(listItems3);
var queryItems4 = /* @__PURE__ */ createRESTModule(queryItems3);
var searchItems4 = /* @__PURE__ */ createRESTModule(searchItems3);
var countItems4 = /* @__PURE__ */ createRESTModule(countItems3);
var updateItem4 = /* @__PURE__ */ createRESTModule(updateItem3);
var bulkUpdateItem4 = /* @__PURE__ */ createRESTModule(bulkUpdateItem3);
var deleteItem4 = /* @__PURE__ */ createRESTModule(deleteItem3);
var bulkDeleteItems4 = /* @__PURE__ */ createRESTModule(bulkDeleteItems3);
var onItemCreated2 = createEventModule(onItemCreated);
var onItemDeleted2 = createEventModule(onItemDeleted);
var onItemUpdated2 = createEventModule(onItemUpdated);
export {
  AggregationType,
  AvailabilityType,
  DayOfWeek,
  EntitiesDayOfWeek,
  Enum,
  Mode,
  PlacementType,
  ResolutionMethod,
  ScalarType,
  SortDirection,
  SortOrder,
  SortType,
  State,
  WebhookIdentityType,
  bulkCreateItems4 as bulkCreateItems,
  bulkDeleteItems4 as bulkDeleteItems,
  bulkUpdateItem4 as bulkUpdateItem,
  countItems4 as countItems,
  createItem4 as createItem,
  deleteItem4 as deleteItem,
  duplicateItem4 as duplicateItem,
  getItem4 as getItem,
  listItems4 as listItems,
  onItemCreated2 as onItemCreated,
  onItemDeleted2 as onItemDeleted,
  onItemUpdated2 as onItemUpdated,
  queryItems4 as queryItems,
  searchItems4 as searchItems,
  updateItem4 as updateItem
};
