// @wix/auto_sdk_restaurants_items/build/internal/es/meta.mjs
// src/restaurants-menus-v1-item-items.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKFloatToRESTFloat } from "@wix/sdk-runtime/transformations/float";
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressRestaurantsMenusItemV1RestaurantsMenusItemUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-menus-item",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/menus-item",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/items",
        destPath: "/v1/bulk/items"
      },
      {
        srcPath: "/restaurants/menus/v1/items",
        destPath: "/v1/items"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
function resolveComWixpressRestaurantsMenusItemV1RestaurantsDuplicateItemRestaurantsDuplicateItemUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/items",
        destPath: "/v1/items"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/items",
        destPath: "/v1/bulk/items"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-menus-item",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/menus-item",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/items",
        destPath: "/v1/bulk/items"
      },
      {
        srcPath: "/restaurants/menus/v1/items",
        destPath: "/v1/items"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_items";
function duplicateItem(payload) {
  function __duplicateItem({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_item.v1.restaurants_duplicate_item.RestaurantsDuplicateItem.DuplicateItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemV1RestaurantsDuplicateItemRestaurantsDuplicateItemUrl(
        { protoPath: "/v1/items/{id}/duplicate", data: payload, host }
      ),
      data: payload
    };
    return metadata;
  }
  return __duplicateItem;
}
function createItem(payload) {
  function __createItem({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "item.createdDate" },
          { path: "item.updatedDate" },
          { path: "item.image.urlExpirationDate" },
          { path: "item.additionalImages.urlExpirationDate" },
          {
            path: "item.orderSettings.availability.timestampRangesOptions.ranges.startTime"
          },
          {
            path: "item.orderSettings.availability.timestampRangesOptions.ranges.endTime"
          },
          { path: "item.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_item.v1.RestaurantsMenusItem.CreateItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemV1RestaurantsMenusItemUrl({
        protoPath: "/v1/items",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "item.createdDate" },
            { path: "item.updatedDate" },
            { path: "item.image.urlExpirationDate" },
            { path: "item.additionalImages.urlExpirationDate" },
            {
              path: "item.orderSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "item.orderSettings.availability.timestampRangesOptions.ranges.endTime"
            },
            { path: "item.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createItem;
}
function bulkCreateItems(payload) {
  function __bulkCreateItems({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "items.createdDate" },
          { path: "items.updatedDate" },
          { path: "items.image.urlExpirationDate" },
          { path: "items.additionalImages.urlExpirationDate" },
          {
            path: "items.orderSettings.availability.timestampRangesOptions.ranges.startTime"
          },
          {
            path: "items.orderSettings.availability.timestampRangesOptions.ranges.endTime"
          },
          { path: "items.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_item.v1.RestaurantsMenusItem.BulkCreateItems",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemV1RestaurantsMenusItemUrl({
        protoPath: "/v1/bulk/items/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.image.urlExpirationDate" },
            { path: "results.item.additionalImages.urlExpirationDate" },
            {
              path: "results.item.orderSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "results.item.orderSettings.availability.timestampRangesOptions.ranges.endTime"
            },
            { path: "results.item.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateItems;
}
function getItem(payload) {
  function __getItem({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.menus_item.v1.RestaurantsMenusItem.GetItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemV1RestaurantsMenusItemUrl({
        protoPath: "/v1/items/{itemId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "item.createdDate" },
            { path: "item.updatedDate" },
            { path: "item.image.urlExpirationDate" },
            { path: "item.additionalImages.urlExpirationDate" },
            {
              path: "item.orderSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "item.orderSettings.availability.timestampRangesOptions.ranges.endTime"
            },
            { path: "item.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getItem;
}
function listItems(payload) {
  function __listItems({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.menus_item.v1.RestaurantsMenusItem.ListItems",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemV1RestaurantsMenusItemUrl({
        protoPath: "/v1/items",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "items.createdDate" },
            { path: "items.updatedDate" },
            { path: "items.image.urlExpirationDate" },
            { path: "items.additionalImages.urlExpirationDate" },
            {
              path: "items.orderSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "items.orderSettings.availability.timestampRangesOptions.ranges.endTime"
            },
            { path: "items.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listItems;
}
function queryItems(payload) {
  function __queryItems({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_item.v1.RestaurantsMenusItem.QueryItems",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemV1RestaurantsMenusItemUrl({
        protoPath: "/v1/items/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "items.createdDate" },
            { path: "items.updatedDate" },
            { path: "items.image.urlExpirationDate" },
            { path: "items.additionalImages.urlExpirationDate" },
            {
              path: "items.orderSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "items.orderSettings.availability.timestampRangesOptions.ranges.endTime"
            },
            { path: "items.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryItems;
}
function searchItems(payload) {
  function __searchItems({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "search.aggregations.range.buckets.from" },
          { path: "search.aggregations.range.buckets.to" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.menus_item.v1.RestaurantsMenusItem.SearchItems",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemV1RestaurantsMenusItemUrl({
        protoPath: "/v1/items/search",
        data: serializedData,
        host
      }),
      params: toURLSearchParams(serializedData, true),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "items.createdDate" },
            { path: "items.updatedDate" },
            { path: "items.image.urlExpirationDate" },
            { path: "items.additionalImages.urlExpirationDate" },
            {
              path: "items.orderSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "items.orderSettings.availability.timestampRangesOptions.ranges.endTime"
            },
            { path: "items.externalReferenceInfo.lastSyncDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "aggregationData.results.ranges.results.from" },
            { path: "aggregationData.results.ranges.results.to" },
            { path: "aggregationData.results.scalar.value" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveComWixpressRestaurantsMenusItemV1RestaurantsMenusItemUrl({
            protoPath: "/v1/items/search",
            data: serializedData,
            host
          }),
          data: serializedData
        }
      ]
    };
    return metadata;
  }
  return __searchItems;
}
function countItems(payload) {
  function __countItems({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_item.v1.RestaurantsMenusItem.CountItems",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemV1RestaurantsMenusItemUrl({
        protoPath: "/v1/items/count",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __countItems;
}
function updateItem(payload) {
  function __updateItem({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "item.createdDate" },
          { path: "item.updatedDate" },
          { path: "item.image.urlExpirationDate" },
          { path: "item.additionalImages.urlExpirationDate" },
          {
            path: "item.orderSettings.availability.timestampRangesOptions.ranges.startTime"
          },
          {
            path: "item.orderSettings.availability.timestampRangesOptions.ranges.endTime"
          },
          { path: "item.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item",
      method: "PATCH",
      methodFqn: "com.wixpress.restaurants.menus_item.v1.RestaurantsMenusItem.UpdateItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemV1RestaurantsMenusItemUrl({
        protoPath: "/v1/items/{item.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "item.createdDate" },
            { path: "item.updatedDate" },
            { path: "item.image.urlExpirationDate" },
            { path: "item.additionalImages.urlExpirationDate" },
            {
              path: "item.orderSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "item.orderSettings.availability.timestampRangesOptions.ranges.endTime"
            },
            { path: "item.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateItem;
}
function bulkUpdateItem(payload) {
  function __bulkUpdateItem({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "items.mask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "items.item.createdDate" },
          { path: "items.item.updatedDate" },
          { path: "items.item.image.urlExpirationDate" },
          { path: "items.item.additionalImages.urlExpirationDate" },
          {
            path: "items.item.orderSettings.availability.timestampRangesOptions.ranges.startTime"
          },
          {
            path: "items.item.orderSettings.availability.timestampRangesOptions.ranges.endTime"
          },
          { path: "items.item.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_item.v1.RestaurantsMenusItem.BulkUpdateItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemV1RestaurantsMenusItemUrl({
        protoPath: "/v1/bulk/items/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.image.urlExpirationDate" },
            { path: "results.item.additionalImages.urlExpirationDate" },
            {
              path: "results.item.orderSettings.availability.timestampRangesOptions.ranges.startTime"
            },
            {
              path: "results.item.orderSettings.availability.timestampRangesOptions.ranges.endTime"
            },
            { path: "results.item.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateItem;
}
function deleteItem(payload) {
  function __deleteItem({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item",
      method: "DELETE",
      methodFqn: "com.wixpress.restaurants.menus_item.v1.RestaurantsMenusItem.DeleteItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemV1RestaurantsMenusItemUrl({
        protoPath: "/v1/items/{itemId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteItem;
}
function bulkDeleteItems(payload) {
  function __bulkDeleteItems({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item",
      method: "DELETE",
      methodFqn: "com.wixpress.restaurants.menus_item.v1.RestaurantsMenusItem.BulkDeleteItems",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemV1RestaurantsMenusItemUrl({
        protoPath: "/v1/bulk/items/delete",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __bulkDeleteItems;
}

// src/restaurants-menus-v1-item-items.types.ts
var AvailabilityType = /* @__PURE__ */ ((AvailabilityType2) => {
  AvailabilityType2["UNSPECIFIED_AVAILABILITY_OPTION"] = "UNSPECIFIED_AVAILABILITY_OPTION";
  AvailabilityType2["ALWAYS_AVAILABLE"] = "ALWAYS_AVAILABLE";
  AvailabilityType2["WEEKLY_SCHEDULE"] = "WEEKLY_SCHEDULE";
  AvailabilityType2["TIMESTAMP_RANGES"] = "TIMESTAMP_RANGES";
  return AvailabilityType2;
})(AvailabilityType || {});
var EntitiesDayOfWeek = /* @__PURE__ */ ((EntitiesDayOfWeek2) => {
  EntitiesDayOfWeek2["MON"] = "MON";
  EntitiesDayOfWeek2["TUE"] = "TUE";
  EntitiesDayOfWeek2["WED"] = "WED";
  EntitiesDayOfWeek2["THU"] = "THU";
  EntitiesDayOfWeek2["FRI"] = "FRI";
  EntitiesDayOfWeek2["SAT"] = "SAT";
  EntitiesDayOfWeek2["SUN"] = "SUN";
  return EntitiesDayOfWeek2;
})(EntitiesDayOfWeek || {});
var Enum = /* @__PURE__ */ ((Enum2) => {
  Enum2["UNKNOWN"] = "UNKNOWN";
  Enum2["PROTECTED"] = "PROTECTED";
  Enum2["PRIVATE"] = "PRIVATE";
  Enum2["PUBLIC"] = "PUBLIC";
  Enum2["UNCHANGED"] = "UNCHANGED";
  Enum2["GROUP_PROTECTED"] = "GROUP_PROTECTED";
  return Enum2;
})(Enum || {});
var State = /* @__PURE__ */ ((State2) => {
  State2["Unknown"] = "Unknown";
  State2["Off"] = "Off";
  State2["On"] = "On";
  return State2;
})(State || {});
var PlacementType = /* @__PURE__ */ ((PlacementType2) => {
  PlacementType2["BEFORE"] = "BEFORE";
  PlacementType2["AFTER"] = "AFTER";
  PlacementType2["REPLACE"] = "REPLACE";
  return PlacementType2;
})(PlacementType || {});
var DayOfWeek = /* @__PURE__ */ ((DayOfWeek2) => {
  DayOfWeek2["MONDAY"] = "MONDAY";
  DayOfWeek2["TUESDAY"] = "TUESDAY";
  DayOfWeek2["WEDNESDAY"] = "WEDNESDAY";
  DayOfWeek2["THURSDAY"] = "THURSDAY";
  DayOfWeek2["FRIDAY"] = "FRIDAY";
  DayOfWeek2["SATURDAY"] = "SATURDAY";
  DayOfWeek2["SUNDAY"] = "SUNDAY";
  return DayOfWeek2;
})(DayOfWeek || {});
var ResolutionMethod = /* @__PURE__ */ ((ResolutionMethod2) => {
  ResolutionMethod2["QUERY_PARAM"] = "QUERY_PARAM";
  ResolutionMethod2["SUBDOMAIN"] = "SUBDOMAIN";
  ResolutionMethod2["SUBDIRECTORY"] = "SUBDIRECTORY";
  return ResolutionMethod2;
})(ResolutionMethod || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var SortType = /* @__PURE__ */ ((SortType2) => {
  SortType2["COUNT"] = "COUNT";
  SortType2["VALUE"] = "VALUE";
  return SortType2;
})(SortType || {});
var SortDirection = /* @__PURE__ */ ((SortDirection2) => {
  SortDirection2["DESC"] = "DESC";
  SortDirection2["ASC"] = "ASC";
  return SortDirection2;
})(SortDirection || {});
var ScalarType = /* @__PURE__ */ ((ScalarType2) => {
  ScalarType2["UNKNOWN_SCALAR_TYPE"] = "UNKNOWN_SCALAR_TYPE";
  ScalarType2["COUNT_DISTINCT"] = "COUNT_DISTINCT";
  ScalarType2["MIN"] = "MIN";
  ScalarType2["MAX"] = "MAX";
  return ScalarType2;
})(ScalarType || {});
var AggregationType = /* @__PURE__ */ ((AggregationType2) => {
  AggregationType2["UNKNOWN_AGGREGATION_TYPE"] = "UNKNOWN_AGGREGATION_TYPE";
  AggregationType2["VALUE"] = "VALUE";
  AggregationType2["RANGE"] = "RANGE";
  AggregationType2["SCALAR"] = "SCALAR";
  return AggregationType2;
})(AggregationType || {});
var Mode = /* @__PURE__ */ ((Mode2) => {
  Mode2["OR"] = "OR";
  Mode2["AND"] = "AND";
  return Mode2;
})(Mode || {});

// src/restaurants-menus-v1-item-items.meta.ts
function duplicateItem2() {
  const payload = { id: ":id" };
  const getRequestOptions = duplicateItem(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/items/{id}/duplicate",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createItem2() {
  const payload = {};
  const getRequestOptions = createItem(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/items",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkCreateItems2() {
  const payload = {};
  const getRequestOptions = bulkCreateItems(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/items/create",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getItem2() {
  const payload = { itemId: ":itemId" };
  const getRequestOptions = getItem(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/items/{itemId}",
    pathParams: { itemId: "itemId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listItems2() {
  const payload = {};
  const getRequestOptions = listItems(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/items",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryItems2() {
  const payload = {};
  const getRequestOptions = queryItems(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/items/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function searchItems2() {
  const payload = {};
  const getRequestOptions = searchItems(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/items/search",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function countItems2() {
  const payload = {};
  const getRequestOptions = countItems(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/items/count",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateItem2() {
  const payload = { item: { id: ":itemId" } };
  const getRequestOptions = updateItem(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/items/{item.id}",
    pathParams: { itemId: "itemId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateItem2() {
  const payload = {};
  const getRequestOptions = bulkUpdateItem(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/items/update",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteItem2() {
  const payload = { itemId: ":itemId" };
  const getRequestOptions = deleteItem(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/items/{itemId}",
    pathParams: { itemId: "itemId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkDeleteItems2() {
  const payload = {};
  const getRequestOptions = bulkDeleteItems(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/bulk/items/delete",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  AggregationType as AggregationTypeOriginal,
  AvailabilityType as AvailabilityTypeOriginal,
  DayOfWeek as DayOfWeekOriginal,
  EntitiesDayOfWeek as EntitiesDayOfWeekOriginal,
  Enum as EnumOriginal,
  Mode as ModeOriginal,
  PlacementType as PlacementTypeOriginal,
  ResolutionMethod as ResolutionMethodOriginal,
  ScalarType as ScalarTypeOriginal,
  SortDirection as SortDirectionOriginal,
  SortOrder as SortOrderOriginal,
  SortType as SortTypeOriginal,
  State as StateOriginal,
  WebhookIdentityType as WebhookIdentityTypeOriginal,
  bulkCreateItems2 as bulkCreateItems,
  bulkDeleteItems2 as bulkDeleteItems,
  bulkUpdateItem2 as bulkUpdateItem,
  countItems2 as countItems,
  createItem2 as createItem,
  deleteItem2 as deleteItem,
  duplicateItem2 as duplicateItem,
  getItem2 as getItem,
  listItems2 as listItems,
  queryItems2 as queryItems,
  searchItems2 as searchItems,
  updateItem2 as updateItem
};
