// @wix/headless-events/cjs/dist/utils/date.js
export const formatShortDate = (date, timeZone, locale) => new Date(date).toLocaleDateString(locale, {
    timeZone,
    weekday: 'short',
    day: '2-digit',
    month: 'short',
});
export const formatFullDate = (date, timeZone, showTimeZone, locale) => new Date(date).toLocaleDateString(locale, {
    timeZone,
    year: 'numeric',
    month: 'short',
    day: 'numeric',
    hour: '2-digit',
    minute: '2-digit',
    timeZoneName: showTimeZone ? 'short' : undefined,
});
export const formatMonthDayYear = (date, timeZone, locale) => new Date(date).toLocaleDateString(locale, {
    timeZone,
    month: 'short',
    day: 'numeric',
    year: 'numeric',
});
export const formatTime = (date, timeZone, locale) => new Date(date).toLocaleTimeString(locale, {
    timeZone,
    hour: '2-digit',
    minute: '2-digit',
});
export const formatTimeRange = (startDate, endDate, timeZone, locale) => `${formatTime(startDate, timeZone, locale)} - ${formatTime(endDate, timeZone, locale)}`;
export const getDurationInMinutes = (startDate, endDate) => Math.round((new Date(endDate).getTime() - new Date(startDate).getTime()) / (1000 * 60));
