// @wix/auto-patterns/dist/cjs/components/AutoPatternsCollectionPageContent/AutoPatternsCollectionPageContent.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.AutoPatternsCollectionPageContent = void 0;
var _react = _interopRequireDefault(require("react"));
var _AutoPatternsTable = require("../AutoPatternsTable");
var _AutoPatternsGrid = require("../AutoPatternsGrid");
var _AutoPatternsTableGridSwitch = require("../AutoPatternsTableGridSwitch/AutoPatternsTableGridSwitch");
var _providers = require("../../providers");
var _SkeletonCollection = require("./SkeletonCollection");
var _jsxFileName = "/home/builduser/work/73d19ce3378ce4dd/packages/auto-patterns/dist/cjs/components/AutoPatternsCollectionPageContent/AutoPatternsCollectionPageContent.tsx";
const AutoPatternsCollectionPageContent = ({
  configuration
}) => {
  const {
    components
  } = configuration;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, components.map((component, index) => {
    var _layout$find, _layout$find2;
    const {
      layout,
      ...sharedConfig
    } = component;
    if (!(layout != null && layout.length)) {
      return null;
    }
    const tableConfig = (_layout$find = layout.find(item => item.type === 'Table')) == null ? void 0 : _layout$find.table;
    const gridConfig = (_layout$find2 = layout.find(item => item.type === 'Grid')) == null ? void 0 : _layout$find2.grid;
    let componentType = 'Table';
    let componentToRender = null;
    if (tableConfig && gridConfig) {
      const config = {
        ...sharedConfig,
        ...tableConfig,
        ...gridConfig
      };
      componentToRender = /*#__PURE__*/_react.default.createElement(_AutoPatternsTableGridSwitch.AutoPatternsTableGridSwitch, {
        configuration: config,
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 42,
          columnNumber: 15
        }
      });
    } else if (tableConfig) {
      const config = {
        ...sharedConfig,
        ...tableConfig
      };
      componentToRender = /*#__PURE__*/_react.default.createElement(_AutoPatternsTable.AutoPatternsTable, {
        configuration: config,
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 46,
          columnNumber: 33
        }
      });
    } else if (gridConfig) {
      componentType = 'Grid';
      const config = {
        ...sharedConfig,
        ...gridConfig
      };
      componentToRender = /*#__PURE__*/_react.default.createElement(_AutoPatternsGrid.AutoPatternsGrid, {
        configuration: config,
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 50,
          columnNumber: 33
        }
      });
    }
    return componentToRender ? /*#__PURE__*/_react.default.createElement(_providers.SchemaProvider, {
      key: `${sharedConfig.collection.collectionId}-${index}`,
      collection: sharedConfig.collection,
      skeleton: /*#__PURE__*/_react.default.createElement(_SkeletonCollection.SkeletonCollection, {
        componentType: componentType,
        __self: void 0,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 57,
          columnNumber: 25
        }
      }),
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 54,
        columnNumber: 13
      }
    }, componentToRender) : null;
  }).filter(Boolean));
};
exports.AutoPatternsCollectionPageContent = AutoPatternsCollectionPageContent;
