// @wix/auto-patterns/dist/cjs/components/AutoPatternsCollectionPageContent/SkeletonCollection.js
"use strict";

exports.__esModule = true;
exports.SkeletonCollection = void 0;
var _react = _interopRequireWildcard(require("react"));
var _patterns = require("@wix/patterns");
var _jsxFileName = "/home/builduser/work/73d19ce3378ce4dd/packages/auto-patterns/dist/cjs/components/AutoPatternsCollectionPageContent/SkeletonCollection.tsx";
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
const SkeletonCollection = ({
  componentType
}) => {
  const state = (0, _patterns.useTableGridSwitchCollection)({
    queryName: '',
    fetchData: async () => new Promise(() => {}),
    limit: 50
  });
  (0, _react.useEffect)(() => {
    state.viewType = componentType === 'Table' ? 'table' : 'grid';
  }, [componentType, state]);
  return /*#__PURE__*/_react.default.createElement(_patterns.TableGridSwitch, {
    state: state,
    columns: [],
    dataHook: `auto-patterns-${componentType.toLowerCase()}`,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 20,
      columnNumber: 5
    }
  });
};
exports.SkeletonCollection = SkeletonCollection;
