// @wix/ricos/dist/es/ricos-content/normalization/block-processors.js
import lodash from "lodash";
const fixAtomicBlockText = (block) => block.type === "atomic" ? { ...block, text: " " } : block;
const addInlineStyleRanges = (block) => ({
  ...block,
  inlineStyleRanges: block.inlineStyleRanges || []
});
const addLinkUnderlineRange = (block, range) => {
  const inlineStyleRange = {
    offset: range.offset,
    length: range.length,
    style: "UNDERLINE"
  };
  const inlineStyleRanges = lodash.uniqWith(
    [...block.inlineStyleRanges || [], inlineStyleRange],
    lodash.isEqual
  );
  return { ...block, inlineStyleRanges };
};
const isHexColor = (str) => /^#[0-9a-f]{3}(?:[0-9a-f]{3})?$/i.test(str);
const wrapTextColorStyle = (style) => isHexColor(style) ? JSON.stringify({ FG: style }) : style;
const fixTextColorStyle = (block) => {
  return {
    ...block,
    inlineStyleRanges: block.inlineStyleRanges.map((range) => ({
      ...range,
      style: wrapTextColorStyle(range.style)
    }))
  };
};
export {
  addInlineStyleRanges,
  addLinkUnderlineRange,
  fixAtomicBlockText,
  fixTextColorStyle
};
