// @wix/auto_sdk_benefit-programs_program-definitions/build/es/index.mjs
// src/benefit-programs-v1-program-definition-program-definitions.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/benefit-programs-v1-program-definition-program-definitions.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixBenefitProgramsV1ProgramDefinitionProgramDefinitionServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/program-definitions",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/benefit-programs-program-definitions",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-program-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/program-definitions",
        destPath: "/v1/program-definitions"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-program-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/program-definitions",
        destPath: "/v1/program-definitions"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/benefit-programs-program-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/program-definitions",
        destPath: "/v1/program-definitions"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/benefit-programs-program-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/program-definitions",
        destPath: "/v1/program-definitions"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-program-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/program-definitions",
        destPath: "/v1/program-definitions"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-program-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/program-definitions",
        destPath: "/v1/program-definitions"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/_api/benefit-programs/v1/program-definitions",
        destPath: "/v1/program-definitions"
      },
      {
        srcPath: "/benefit-programs/v1/program-definitions",
        destPath: "/v1/program-definitions"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/benefit-programs/v1/program-definitions",
        destPath: "/v1/program-definitions"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/benefit-programs/v1/program-definitions",
        destPath: "/v1/program-definitions"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_benefit-programs_program-definitions";
function createProgramDefinition(payload) {
  function __createProgramDefinition({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "programDefinition.createdDate" },
          { path: "programDefinition.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program_definition",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.program_definition.ProgramDefinitionService.CreateProgramDefinition",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramDefinitionProgramDefinitionServiceUrl(
        { protoPath: "/v1/program-definitions", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "programDefinition.createdDate" },
            { path: "programDefinition.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createProgramDefinition;
}
function updateProgramDefinition(payload) {
  function __updateProgramDefinition({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "programDefinition.createdDate" },
          { path: "programDefinition.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program_definition",
      method: "PATCH",
      methodFqn: "wix.benefit_programs.v1.program_definition.ProgramDefinitionService.UpdateProgramDefinition",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramDefinitionProgramDefinitionServiceUrl(
        {
          protoPath: "/v1/program-definitions/{programDefinition.id}",
          data: serializedData,
          host
        }
      ),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "programDefinition.createdDate" },
            { path: "programDefinition.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateProgramDefinition;
}
function deleteProgramDefinition(payload) {
  function __deleteProgramDefinition({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program_definition",
      method: "DELETE",
      methodFqn: "wix.benefit_programs.v1.program_definition.ProgramDefinitionService.DeleteProgramDefinition",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramDefinitionProgramDefinitionServiceUrl(
        {
          protoPath: "/v1/program-definitions/{programDefinitionId}",
          data: payload,
          host
        }
      ),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteProgramDefinition;
}
function getProgramDefinition(payload) {
  function __getProgramDefinition({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program_definition",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.program_definition.ProgramDefinitionService.GetProgramDefinition",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramDefinitionProgramDefinitionServiceUrl(
        {
          protoPath: "/v1/program-definitions/{programDefinitionId}",
          data: payload,
          host
        }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "programDefinition.createdDate" },
            { path: "programDefinition.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getProgramDefinition;
}
function getProgramDefinitionByExternalIdAndNamespace(payload) {
  function __getProgramDefinitionByExternalIdAndNamespace({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program_definition",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.program_definition.ProgramDefinitionService.GetProgramDefinitionByExternalIdAndNamespace",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramDefinitionProgramDefinitionServiceUrl(
        {
          protoPath: "/v1/program-definitions/by-namespace-and-external-id",
          data: payload,
          host
        }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "programDefinition.createdDate" },
            { path: "programDefinition.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getProgramDefinitionByExternalIdAndNamespace;
}
function queryProgramDefinitions(payload) {
  function __queryProgramDefinitions({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program_definition",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.program_definition.ProgramDefinitionService.QueryProgramDefinitions",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramDefinitionProgramDefinitionServiceUrl(
        { protoPath: "/v1/program-definitions/query", data: payload, host }
      ),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "programDefinitions.createdDate" },
            { path: "programDefinitions.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryProgramDefinitions;
}

// src/benefit-programs-v1-program-definition-program-definitions.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var Cascade = /* @__PURE__ */ ((Cascade2) => {
  Cascade2["UNKNOWN_CASCADE"] = "UNKNOWN_CASCADE";
  Cascade2["NEXT_RENEWAL"] = "NEXT_RENEWAL";
  Cascade2["IMMEDIATELY"] = "IMMEDIATELY";
  Cascade2["FUTURE_PROVISIONS"] = "FUTURE_PROVISIONS";
  return Cascade2;
})(Cascade || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createProgramDefinition2(programDefinition) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    programDefinition
  });
  const reqOpts = createProgramDefinition(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.programDefinition;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { programDefinition: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["programDefinition"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateProgramDefinition2(_id, programDefinition) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    programDefinition: { ...programDefinition, id: _id }
  });
  const reqOpts = updateProgramDefinition(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.programDefinition;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { programDefinition: "$[1]" },
        explicitPathsToArguments: { "programDefinition.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "programDefinition"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteProgramDefinition2(programDefinitionId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    programDefinitionId,
    cascadeType: options?.cascadeType
  });
  const reqOpts = deleteProgramDefinition(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          programDefinitionId: "$[0]",
          cascadeType: "$[1].cascadeType"
        },
        singleArgumentUnchanged: false
      },
      ["programDefinitionId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getProgramDefinition2(programDefinitionId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    programDefinitionId
  });
  const reqOpts = getProgramDefinition(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.programDefinition;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { programDefinitionId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["programDefinitionId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getProgramDefinitionByExternalIdAndNamespace2(namespace, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    namespace,
    externalId: options?.externalId
  });
  const reqOpts = getProgramDefinitionByExternalIdAndNamespace(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          namespace: "$[0]",
          externalId: "$[1].externalId"
        },
        singleArgumentUnchanged: false
      },
      ["namespace", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryProgramDefinitions2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryProgramDefinitions(
        payload
      );
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [])
      );
      return {
        items: transformedData?.programDefinitions,
        pagingMetadata: transformedData?.metadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}

// src/benefit-programs-v1-program-definition-program-definitions.public.ts
function createProgramDefinition3(httpClient) {
  return (programDefinition) => createProgramDefinition2(
    programDefinition,
    // @ts-ignore
    { httpClient }
  );
}
function updateProgramDefinition3(httpClient) {
  return (_id, programDefinition) => updateProgramDefinition2(
    _id,
    programDefinition,
    // @ts-ignore
    { httpClient }
  );
}
function deleteProgramDefinition3(httpClient) {
  return (programDefinitionId, options) => deleteProgramDefinition2(
    programDefinitionId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function getProgramDefinition3(httpClient) {
  return (programDefinitionId) => getProgramDefinition2(
    programDefinitionId,
    // @ts-ignore
    { httpClient }
  );
}
function getProgramDefinitionByExternalIdAndNamespace3(httpClient) {
  return (namespace, options) => getProgramDefinitionByExternalIdAndNamespace2(
    namespace,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function queryProgramDefinitions3(httpClient) {
  return () => queryProgramDefinitions2(
    // @ts-ignore
    { httpClient }
  );
}

// src/benefit-programs-v1-program-definition-program-definitions.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
var createProgramDefinition4 = /* @__PURE__ */ createRESTModule(createProgramDefinition3);
var updateProgramDefinition4 = /* @__PURE__ */ createRESTModule(updateProgramDefinition3);
var deleteProgramDefinition4 = /* @__PURE__ */ createRESTModule(deleteProgramDefinition3);
var getProgramDefinition4 = /* @__PURE__ */ createRESTModule(getProgramDefinition3);
var getProgramDefinitionByExternalIdAndNamespace4 = /* @__PURE__ */ createRESTModule(
  getProgramDefinitionByExternalIdAndNamespace3
);
var queryProgramDefinitions4 = /* @__PURE__ */ createRESTModule(queryProgramDefinitions3);
export {
  Cascade,
  SortOrder,
  WebhookIdentityType,
  createProgramDefinition4 as createProgramDefinition,
  deleteProgramDefinition4 as deleteProgramDefinition,
  getProgramDefinition4 as getProgramDefinition,
  getProgramDefinitionByExternalIdAndNamespace4 as getProgramDefinitionByExternalIdAndNamespace,
  queryProgramDefinitions4 as queryProgramDefinitions,
  updateProgramDefinition4 as updateProgramDefinition
};
