// @wix/auto_sdk_benefit-programs_program-definitions/build/es/meta.mjs
// src/benefit-programs-v1-program-definition-program-definitions.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixBenefitProgramsV1ProgramDefinitionProgramDefinitionServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/program-definitions",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/benefit-programs-program-definitions",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-program-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/program-definitions",
        destPath: "/v1/program-definitions"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-program-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/program-definitions",
        destPath: "/v1/program-definitions"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/benefit-programs-program-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/program-definitions",
        destPath: "/v1/program-definitions"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/benefit-programs-program-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/program-definitions",
        destPath: "/v1/program-definitions"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-program-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/program-definitions",
        destPath: "/v1/program-definitions"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-program-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/program-definitions",
        destPath: "/v1/program-definitions"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/_api/benefit-programs/v1/program-definitions",
        destPath: "/v1/program-definitions"
      },
      {
        srcPath: "/benefit-programs/v1/program-definitions",
        destPath: "/v1/program-definitions"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/benefit-programs/v1/program-definitions",
        destPath: "/v1/program-definitions"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/benefit-programs/v1/program-definitions",
        destPath: "/v1/program-definitions"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_benefit-programs_program-definitions";
function createProgramDefinition(payload) {
  function __createProgramDefinition({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "programDefinition.createdDate" },
          { path: "programDefinition.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program_definition",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.program_definition.ProgramDefinitionService.CreateProgramDefinition",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramDefinitionProgramDefinitionServiceUrl(
        { protoPath: "/v1/program-definitions", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "programDefinition.createdDate" },
            { path: "programDefinition.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createProgramDefinition;
}
function updateProgramDefinition(payload) {
  function __updateProgramDefinition({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "programDefinition.createdDate" },
          { path: "programDefinition.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program_definition",
      method: "PATCH",
      methodFqn: "wix.benefit_programs.v1.program_definition.ProgramDefinitionService.UpdateProgramDefinition",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramDefinitionProgramDefinitionServiceUrl(
        {
          protoPath: "/v1/program-definitions/{programDefinition.id}",
          data: serializedData,
          host
        }
      ),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "programDefinition.createdDate" },
            { path: "programDefinition.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateProgramDefinition;
}
function deleteProgramDefinition(payload) {
  function __deleteProgramDefinition({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program_definition",
      method: "DELETE",
      methodFqn: "wix.benefit_programs.v1.program_definition.ProgramDefinitionService.DeleteProgramDefinition",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramDefinitionProgramDefinitionServiceUrl(
        {
          protoPath: "/v1/program-definitions/{programDefinitionId}",
          data: payload,
          host
        }
      ),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteProgramDefinition;
}
function getProgramDefinition(payload) {
  function __getProgramDefinition({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program_definition",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.program_definition.ProgramDefinitionService.GetProgramDefinition",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramDefinitionProgramDefinitionServiceUrl(
        {
          protoPath: "/v1/program-definitions/{programDefinitionId}",
          data: payload,
          host
        }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "programDefinition.createdDate" },
            { path: "programDefinition.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getProgramDefinition;
}
function getProgramDefinitionByExternalIdAndNamespace(payload) {
  function __getProgramDefinitionByExternalIdAndNamespace({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program_definition",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.program_definition.ProgramDefinitionService.GetProgramDefinitionByExternalIdAndNamespace",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramDefinitionProgramDefinitionServiceUrl(
        {
          protoPath: "/v1/program-definitions/by-namespace-and-external-id",
          data: payload,
          host
        }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "programDefinition.createdDate" },
            { path: "programDefinition.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getProgramDefinitionByExternalIdAndNamespace;
}
function queryProgramDefinitions(payload) {
  function __queryProgramDefinitions({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program_definition",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.program_definition.ProgramDefinitionService.QueryProgramDefinitions",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramDefinitionProgramDefinitionServiceUrl(
        { protoPath: "/v1/program-definitions/query", data: payload, host }
      ),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "programDefinitions.createdDate" },
            { path: "programDefinitions.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryProgramDefinitions;
}

// src/benefit-programs-v1-program-definition-program-definitions.meta.ts
function createProgramDefinition2() {
  const payload = {};
  const getRequestOptions = createProgramDefinition(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/program-definitions",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateProgramDefinition2() {
  const payload = { programDefinition: { id: ":programDefinitionId" } };
  const getRequestOptions = updateProgramDefinition(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/program-definitions/{programDefinition.id}",
    pathParams: { programDefinitionId: "programDefinitionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteProgramDefinition2() {
  const payload = { programDefinitionId: ":programDefinitionId" };
  const getRequestOptions = deleteProgramDefinition(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/program-definitions/{programDefinitionId}",
    pathParams: { programDefinitionId: "programDefinitionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getProgramDefinition2() {
  const payload = { programDefinitionId: ":programDefinitionId" };
  const getRequestOptions = getProgramDefinition(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/program-definitions/{programDefinitionId}",
    pathParams: { programDefinitionId: "programDefinitionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getProgramDefinitionByExternalIdAndNamespace2() {
  const payload = {};
  const getRequestOptions = getProgramDefinitionByExternalIdAndNamespace(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/program-definitions/by-namespace-and-external-id",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryProgramDefinitions2() {
  const payload = {};
  const getRequestOptions = queryProgramDefinitions(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/program-definitions/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  createProgramDefinition2 as createProgramDefinition,
  deleteProgramDefinition2 as deleteProgramDefinition,
  getProgramDefinition2 as getProgramDefinition,
  getProgramDefinitionByExternalIdAndNamespace2 as getProgramDefinitionByExternalIdAndNamespace,
  queryProgramDefinitions2 as queryProgramDefinitions,
  updateProgramDefinition2 as updateProgramDefinition
};
