// @wix/auto_sdk_restaurants_availability-exceptions/build/internal/es/meta.mjs
// src/restaurants-v1-availability-exception-availability-exceptions.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    _base_domain_: [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "users._base_domain_": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-availability-exceptions",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants-availability-exceptions",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_availability-exceptions";
function createAvailabilityException(payload) {
  function __createAvailabilityException({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "availabilityException.createdDate" },
          { path: "availabilityException.updatedDate" },
          { path: "availabilityException.startTime" },
          { path: "availabilityException.endTime" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "POST",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.CreateAvailabilityException",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/availability-exceptions",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "availabilityException.createdDate" },
            { path: "availabilityException.updatedDate" },
            { path: "availabilityException.startTime" },
            { path: "availabilityException.endTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createAvailabilityException;
}
function getAvailabilityException(payload) {
  function __getAvailabilityException({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "GET",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.GetAvailabilityException",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/availability-exceptions/{availabilityExceptionId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "availabilityException.createdDate" },
            { path: "availabilityException.updatedDate" },
            { path: "availabilityException.startTime" },
            { path: "availabilityException.endTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getAvailabilityException;
}
function updateAvailabilityException(payload) {
  function __updateAvailabilityException({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "availabilityException.createdDate" },
          { path: "availabilityException.updatedDate" },
          { path: "availabilityException.startTime" },
          { path: "availabilityException.endTime" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "PATCH",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.UpdateAvailabilityException",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/availability-exceptions/{availabilityException.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "availabilityException.createdDate" },
            { path: "availabilityException.updatedDate" },
            { path: "availabilityException.startTime" },
            { path: "availabilityException.endTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateAvailabilityException;
}
function deleteAvailabilityException(payload) {
  function __deleteAvailabilityException({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "DELETE",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.DeleteAvailabilityException",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/availability-exceptions/{availabilityExceptionId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteAvailabilityException;
}
function queryAvailabilityExceptions(payload) {
  function __queryAvailabilityExceptions({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "POST",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.QueryAvailabilityExceptions",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/availability-exceptions/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "availabilityExceptions.createdDate" },
            { path: "availabilityExceptions.updatedDate" },
            { path: "availabilityExceptions.startTime" },
            { path: "availabilityExceptions.endTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryAvailabilityExceptions;
}
function bulkUpdateAvailabilityExceptions(payload) {
  function __bulkUpdateAvailabilityExceptions({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "availabilityExceptions.mask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "availabilityExceptions.availabilityException.createdDate" },
          { path: "availabilityExceptions.availabilityException.updatedDate" },
          { path: "availabilityExceptions.availabilityException.startTime" },
          { path: "availabilityExceptions.availabilityException.endTime" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "POST",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.BulkUpdateAvailabilityExceptions",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/bulk/availability-exceptions/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.startTime" },
            { path: "results.item.endTime" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateAvailabilityExceptions;
}
function bulkUpdateAvailabilityExceptionTags(payload) {
  function __bulkUpdateAvailabilityExceptionTags({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "POST",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.BulkUpdateAvailabilityExceptionTags",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/bulk/availability-exceptions/update-tags",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateAvailabilityExceptionTags;
}
function bulkUpdateAvailabilityExceptionTagsByFilter(payload) {
  function __bulkUpdateAvailabilityExceptionTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v1.availability_exception",
      method: "POST",
      methodFqn: "wix.restaurants.v1.AvailabilityExceptionsService.BulkUpdateAvailabilityExceptionTagsByFilter",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsV1AvailabilityExceptionsServiceUrl({
        protoPath: "/v1/bulk/availability-exceptions/update-tags-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateAvailabilityExceptionTagsByFilter;
}

// src/restaurants-v1-availability-exception-availability-exceptions.types.ts
var AffectedMethods = /* @__PURE__ */ ((AffectedMethods2) => {
  AffectedMethods2["UNKNOWN_AFFECTED_METHODS"] = "UNKNOWN_AFFECTED_METHODS";
  AffectedMethods2["ALL_FULFILLMENT_METHODS"] = "ALL_FULFILLMENT_METHODS";
  AffectedMethods2["SPECIFIC_FULFILLMENT_METHODS"] = "SPECIFIC_FULFILLMENT_METHODS";
  AffectedMethods2["ALL_PICKUP_FULFILLMENT_METHODS"] = "ALL_PICKUP_FULFILLMENT_METHODS";
  AffectedMethods2["ALL_DELIVERY_FULFILLMENT_METHODS"] = "ALL_DELIVERY_FULFILLMENT_METHODS";
  return AffectedMethods2;
})(AffectedMethods || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNSPECIFIED_AVAILABILITY_EXCEPTION_TYPE"] = "UNSPECIFIED_AVAILABILITY_EXCEPTION_TYPE";
  Type2["ORDERING_PAUSED"] = "ORDERING_PAUSED";
  Type2["CLOSURE"] = "CLOSURE";
  return Type2;
})(Type || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/restaurants-v1-availability-exception-availability-exceptions.meta.ts
function createAvailabilityException2() {
  const payload = {};
  const getRequestOptions = createAvailabilityException(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/availability-exceptions",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getAvailabilityException2() {
  const payload = {
    availabilityExceptionId: ":availabilityExceptionId"
  };
  const getRequestOptions = getAvailabilityException(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/availability-exceptions/{availabilityExceptionId}",
    pathParams: { availabilityExceptionId: "availabilityExceptionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateAvailabilityException2() {
  const payload = {
    availabilityException: { id: ":availabilityExceptionId" }
  };
  const getRequestOptions = updateAvailabilityException(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/availability-exceptions/{availabilityException.id}",
    pathParams: { availabilityExceptionId: "availabilityExceptionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteAvailabilityException2() {
  const payload = {
    availabilityExceptionId: ":availabilityExceptionId"
  };
  const getRequestOptions = deleteAvailabilityException(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/availability-exceptions/{availabilityExceptionId}",
    pathParams: { availabilityExceptionId: "availabilityExceptionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryAvailabilityExceptions2() {
  const payload = {};
  const getRequestOptions = queryAvailabilityExceptions(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/availability-exceptions/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateAvailabilityExceptions2() {
  const payload = {};
  const getRequestOptions = bulkUpdateAvailabilityExceptions(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/availability-exceptions/update",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateAvailabilityExceptionTags2() {
  const payload = {};
  const getRequestOptions = bulkUpdateAvailabilityExceptionTags(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/availability-exceptions/update-tags",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateAvailabilityExceptionTagsByFilter2() {
  const payload = {};
  const getRequestOptions = bulkUpdateAvailabilityExceptionTagsByFilter(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/availability-exceptions/update-tags-by-filter",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  AffectedMethods as AffectedMethodsOriginal,
  SortOrder as SortOrderOriginal,
  Type as TypeOriginal,
  WebhookIdentityType as WebhookIdentityTypeOriginal,
  bulkUpdateAvailabilityExceptionTags2 as bulkUpdateAvailabilityExceptionTags,
  bulkUpdateAvailabilityExceptionTagsByFilter2 as bulkUpdateAvailabilityExceptionTagsByFilter,
  bulkUpdateAvailabilityExceptions2 as bulkUpdateAvailabilityExceptions,
  createAvailabilityException2 as createAvailabilityException,
  deleteAvailabilityException2 as deleteAvailabilityException,
  getAvailabilityException2 as getAvailabilityException,
  queryAvailabilityExceptions2 as queryAvailabilityExceptions,
  updateAvailabilityException2 as updateAvailabilityException
};
