// @wix/design-system/dist/cjs/AudioPlayer/AudioPlayer.st.css.js

require("../Foundation/stylable/colors.st.css.js");
const { classesRuntime, statesRuntime } = require("./../stylable-cjs-runtime.js");



var _namespace_ = "wds_1_226_0_AudioPlayer";
var _style_ = classesRuntime.bind(null, _namespace_);

module.exports.cssStates = statesRuntime.bind(null, _namespace_);
module.exports.style = _style_;
module.exports.st = _style_;

module.exports.namespace = _namespace_;
module.exports.classes = {"root":"wds_1_226_0_AudioPlayer__root","playPauseButton":"wds_1_226_0_AudioPlayer__playPauseButton","slider":"wds_1_226_0_AudioPlayer__slider","track":"wds_1_226_0_AudioPlayer__track","handle":"wds_1_226_0_AudioPlayer__handle","tooltip":"wds_1_226_0_AudioPlayer__tooltip","tooltipTarget":"wds_1_226_0_AudioPlayer__tooltipTarget","timer":"wds_1_226_0_AudioPlayer__timer"};
module.exports.keyframes = {}; 
module.exports.layers = {};
module.exports.containers = {};
module.exports.stVars = {}; 
module.exports.vars = {"audio-player-position":"--wds_1_226_0_AudioPlayer-audio-player-position"};
