// @wix/auto_sdk_benefit-programs_programs/build/internal/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  IdentityType: () => IdentityType,
  OrderStatus: () => OrderStatus,
  ProgramStatus: () => ProgramStatus,
  ProvisionProgramRequestProgramStatus: () => ProvisionProgramRequestProgramStatus,
  SortOrder: () => SortOrder,
  WebhookIdentityType: () => WebhookIdentityType,
  endProgram: () => endProgram2,
  getProgram: () => getProgram2,
  getProgramByExternalIdAndNamespace: () => getProgramByExternalIdAndNamespace2,
  pauseProgram: () => pauseProgram2,
  provisionProgram: () => provisionProgram2,
  queryPrograms: () => queryPrograms2,
  renewProgram: () => renewProgram2,
  resumeProgram: () => resumeProgram2,
  updateProgram: () => updateProgram2
});
module.exports = __toCommonJS(index_typings_exports);

// src/benefit-programs-v1-program-programs.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/benefit-programs-v1-program-programs.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixBenefitProgramsV1ProgramProgramServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/programs",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/benefit-programs-programs",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-programs",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/programs",
        destPath: "/v1/programs"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-programs",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/programs",
        destPath: "/v1/programs"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/benefit-programs-programs",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/programs",
        destPath: "/v1/programs"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/benefit-programs-programs",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/programs",
        destPath: "/v1/programs"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-programs",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/programs",
        destPath: "/v1/programs"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-programs",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/programs",
        destPath: "/v1/programs"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/benefit-programs/v1/programs",
        destPath: "/v1/programs"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/benefit-programs/v1/programs",
        destPath: "/v1/programs"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_benefit-programs_programs";
function provisionProgram(payload) {
  function __provisionProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.program.ProgramService.ProvisionProgram",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramProgramServiceUrl({
        protoPath: "/v1/programs/provision",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "program.createdDate" },
            { path: "program.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __provisionProgram;
}
function renewProgram(payload) {
  function __renewProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.program.ProgramService.RenewProgram",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramProgramServiceUrl({
        protoPath: "/v1/programs/renew",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __renewProgram;
}
function pauseProgram(payload) {
  function __pauseProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.program.ProgramService.PauseProgram",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramProgramServiceUrl({
        protoPath: "/v1/programs/pause",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __pauseProgram;
}
function resumeProgram(payload) {
  function __resumeProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.program.ProgramService.ResumeProgram",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramProgramServiceUrl({
        protoPath: "/v1/programs/resume",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __resumeProgram;
}
function endProgram(payload) {
  function __endProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.program.ProgramService.EndProgram",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramProgramServiceUrl({
        protoPath: "/v1/programs/end",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __endProgram;
}
function updateProgram(payload) {
  function __updateProgram({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "program.createdDate" },
          { path: "program.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program",
      method: "PATCH",
      methodFqn: "wix.benefit_programs.v1.program.ProgramService.UpdateProgram",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramProgramServiceUrl({
        protoPath: "/v1/programs/{program.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "program.createdDate" },
            { path: "program.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateProgram;
}
function getProgram(payload) {
  function __getProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.program.ProgramService.GetProgram",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramProgramServiceUrl({
        protoPath: "/v1/programs/{programId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "program.createdDate" },
            { path: "program.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getProgram;
}
function queryPrograms(payload) {
  function __queryPrograms({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.program.ProgramService.QueryPrograms",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramProgramServiceUrl({
        protoPath: "/v1/programs/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "programs.createdDate" },
            { path: "programs.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryPrograms;
}
function getProgramByExternalIdAndNamespace(payload) {
  function __getProgramByExternalIdAndNamespace({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.program.ProgramService.GetProgramByExternalIdAndNamespace",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramProgramServiceUrl({
        protoPath: "/v1/programs/by-namespace-and-external-id",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "program.createdDate" },
            { path: "program.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getProgramByExternalIdAndNamespace;
}

// src/benefit-programs-v1-program-programs.universal.ts
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var ProgramStatus = /* @__PURE__ */ ((ProgramStatus2) => {
  ProgramStatus2["UNDEFINED"] = "UNDEFINED";
  ProgramStatus2["ACTIVE"] = "ACTIVE";
  ProgramStatus2["PAUSED"] = "PAUSED";
  ProgramStatus2["ENDED"] = "ENDED";
  ProgramStatus2["PROVISIONING"] = "PROVISIONING";
  ProgramStatus2["PAUSING"] = "PAUSING";
  ProgramStatus2["RESUMING"] = "RESUMING";
  ProgramStatus2["ENDING"] = "ENDING";
  return ProgramStatus2;
})(ProgramStatus || {});
var IdentityType = /* @__PURE__ */ ((IdentityType2) => {
  IdentityType2["UNKNOWN"] = "UNKNOWN";
  IdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  IdentityType2["MEMBER"] = "MEMBER";
  IdentityType2["WIX_USER"] = "WIX_USER";
  return IdentityType2;
})(IdentityType || {});
var ProvisionProgramRequestProgramStatus = /* @__PURE__ */ ((ProvisionProgramRequestProgramStatus2) => {
  ProvisionProgramRequestProgramStatus2["ACTIVE"] = "ACTIVE";
  ProvisionProgramRequestProgramStatus2["PAUSED"] = "PAUSED";
  return ProvisionProgramRequestProgramStatus2;
})(ProvisionProgramRequestProgramStatus || {});
var OrderStatus = /* @__PURE__ */ ((OrderStatus2) => {
  OrderStatus2["UNKNOWN"] = "UNKNOWN";
  OrderStatus2["ACTIVE"] = "ACTIVE";
  OrderStatus2["PAUSED"] = "PAUSED";
  OrderStatus2["ENDED"] = "ENDED";
  return OrderStatus2;
})(OrderStatus || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function provisionProgram2(poolDefinitionLookupId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    poolDefinitionLookupId,
    beneficiary: options?.beneficiary,
    programStatus: options?.programStatus,
    namespace: options?.namespace,
    displayName: options?.displayName,
    externalProgramId: options?.externalProgramId
  });
  const reqOpts = provisionProgram(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          poolDefinitionLookupId: "$[0]",
          beneficiary: "$[1].beneficiary",
          programStatus: "$[1].programStatus",
          namespace: "$[1].namespace",
          displayName: "$[1].displayName",
          externalProgramId: "$[1].externalProgramId"
        },
        singleArgumentUnchanged: false
      },
      ["poolDefinitionLookupId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function renewProgram2(programId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    programId,
    idempotencyKey: options?.idempotencyKey
  });
  const reqOpts = renewProgram(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          programId: "$[0]",
          idempotencyKey: "$[1].idempotencyKey"
        },
        singleArgumentUnchanged: false
      },
      ["programId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function pauseProgram2(programId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    programId,
    idempotencyKey: options?.idempotencyKey
  });
  const reqOpts = pauseProgram(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          programId: "$[0]",
          idempotencyKey: "$[1].idempotencyKey"
        },
        singleArgumentUnchanged: false
      },
      ["programId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function resumeProgram2(programId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    programId,
    idempotencyKey: options?.idempotencyKey
  });
  const reqOpts = resumeProgram(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          programId: "$[0]",
          idempotencyKey: "$[1].idempotencyKey"
        },
        singleArgumentUnchanged: false
      },
      ["programId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function endProgram2(programId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    programId
  });
  const reqOpts = endProgram(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { programId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["programId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateProgram2(_id, program) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    program: { ...program, id: _id }
  });
  const reqOpts = updateProgram(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.program;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { program: "$[1]" },
        explicitPathsToArguments: { "program.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "program"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getProgram2(programId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    programId
  });
  const reqOpts = getProgram(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.program;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { programId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["programId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryPrograms2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryPrograms(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [])
      );
      return {
        items: transformedData?.programs,
        pagingMetadata: transformedData?.metadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function getProgramByExternalIdAndNamespace2(namespace, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    namespace,
    externalId: options?.externalId
  });
  const reqOpts = getProgramByExternalIdAndNamespace(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          namespace: "$[0]",
          externalId: "$[1].externalId"
        },
        singleArgumentUnchanged: false
      },
      ["namespace", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  IdentityType,
  OrderStatus,
  ProgramStatus,
  ProvisionProgramRequestProgramStatus,
  SortOrder,
  WebhookIdentityType,
  endProgram,
  getProgram,
  getProgramByExternalIdAndNamespace,
  pauseProgram,
  provisionProgram,
  queryPrograms,
  renewProgram,
  resumeProgram,
  updateProgram
});
