// @wix/ricos/dist/es/plugin-ai/ai-actions-modal/ai-actions-modal.js
import { DropdownLayout, ListItemSection } from "@wix/design-system";
import { EditSmall, ImageSparklesSmall } from "@wix/wix-ui-icons-common";
import React, { useContext } from "react";
import { EditorContext, ModalServiceContext, RicosContext } from "./../../ricos-context/index.js";
import { unreachable } from "./../../_shared/uncreachable.js";
import { AI_MODAL_FLOATING_MENU_POPOVER } from "../data-hooks.js";
import { defaultPromptsConfiguration } from "../prompts.js";
import styles from "./ai-actions-modal-scss.js";
const AiActionsModal = ({
  toggleAiContentModal,
  toggleAiImageModal,
  enableImageGeneration,
  config
}) => {
  const { t } = useContext(RicosContext);
  const { adapter } = useContext(EditorContext);
  const modalService = useContext(ModalServiceContext);
  const [prompts] = React.useState(
    config.configurePrompts ? config.configurePrompts(defaultPromptsConfiguration) : defaultPromptsConfiguration
  );
  const openAiContentModal = React.useCallback(
    (initialPrompt) => {
      toggleAiContentModal(adapter.tiptapEditor, modalService, initialPrompt);
    },
    [adapter.tiptapEditor, modalService]
  );
  const openAiImageModal = React.useCallback(() => {
    toggleAiImageModal(adapter.tiptapEditor, modalService, true);
  }, [adapter.tiptapEditor, modalService]);
  const options = React.useMemo(() => {
    return [
      promptToDropdownOption(
        {
          promptSlug: "",
          type: "prompt",
          title: "AIPlugin_Action_WritePrompt",
          icon: EditSmall
        },
        t,
        () => openAiContentModal()
      ),
      ...enableImageGeneration ? [
        {
          disabled: true,
          overrideOptionStyle: true,
          id: "AIPlugin_Prompts_Section_Image",
          value: () => /* @__PURE__ */ React.createElement(ListItemSection, { title: t("AIPlugin_Prompts_Section_Image") })
        },
        promptToDropdownOption(
          {
            promptSlug: "",
            type: "prompt",
            title: "AIPlugin_Action_GenerateImage",
            icon: ImageSparklesSmall
          },
          t,
          () => openAiImageModal()
        )
      ] : [],
      ...prompts.floatingToolbarButtonPrompts.map((prompt) => {
        switch (prompt.type) {
          case "prompt": {
            return promptToDropdownOption(prompt, t, () => openAiContentModal(prompt));
          }
          case "section": {
            return {
              disabled: true,
              overrideOptionStyle: true,
              id: prompt.title,
              value: () => /* @__PURE__ */ React.createElement(ListItemSection, { title: t(prompt.title) })
            };
          }
          default: {
            throw unreachable(prompt);
          }
        }
      })
    ];
  }, [prompts.floatingToolbarButtonPrompts]);
  return /* @__PURE__ */ React.createElement(
    DropdownLayout,
    {
      dataHook: AI_MODAL_FLOATING_MENU_POPOVER,
      className: styles.dropdown,
      maxHeightPixels: 250,
      minWidthPixels: 242,
      visible: true,
      options
    }
  );
};
const promptToDropdownOption = (prompt, t, onClick) => {
  return {
    as: "button",
    dataHook: prompt.title,
    ellipsis: false,
    id: prompt.title,
    onClick,
    optionTitle: t(prompt.title),
    prefixIcon: prompt.icon ? /* @__PURE__ */ React.createElement(prompt.icon, null) : void 0,
    shouldFocusWithoutScroll: true,
    size: "small",
    skin: "standard",
    title: false,
    value: "action"
  };
};
export {
  AiActionsModal
};
