// @wix/ricos/dist/cjs/ricos-common-plugins/bold/bold.testkit.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var bold_testkit_exports = {};
__export(bold_testkit_exports, {
  createBoldTestkit: () => createBoldTestkit
});
module.exports = __toCommonJS(bold_testkit_exports);
var import_wix_rich_content_common = require("./../../wix-rich-content-common/index.js");
var import_data_hooks = require("./data-hooks.js");
const createBoldTestkit = () => {
  return {
    content: {},
    toolbars: {
      createTextButtonToolbars: (getTextButtonToolbarBase) => {
        const getButton = () => getTextButtonToolbarBase().$((0, import_wix_rich_content_common.dataHook)(import_data_hooks.TEXT_BUTTON_BOLD));
        return {
          click: () => getButton().click(),
          getElement: () => getTextButtonToolbarBase().get((0, import_wix_rich_content_common.dataHook)(import_data_hooks.TEXT_BUTTON_BOLD)),
          focus: async () => {
            const button = await getButton().unwrap();
            return button.focus();
          }
        };
      }
    }
  };
};
