// @wix/ricos/dist/cjs/ricos-common-plugins/bold/bold.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var bold_exports = {};
__export(bold_exports, {
  pluginBold: () => pluginBold
});
module.exports = __toCommonJS(bold_exports);
var import_wix_ui_icons_common = require("@wix/wix-ui-icons-common");
var import_v1 = require("./../../ricos-schema/legacy/wix/rich_content/v1/index.js");
var import_ricos_types = require("./../../ricos-types/index.js");
var import_is_always_visible = require("./../../_shared/resolvers/is-always-visible.js");
var import_is_bold_in_text = require("./../../_shared/resolvers/is-bold-in-text.js");
var import_is_node_selected = require("./../../_shared/resolvers/is-node-selected.js");
var import_bold_converter = require("./bold-converter.js");
var import_data_hooks = require("./data-hooks.js");
var import_extension = require("./extension.js");
const pluginBold = {
  type: import_v1.Decoration_Type.BOLD,
  config: {},
  pmNodeConverters: [],
  pmMarkConverters: [import_bold_converter.boldConverter],
  tiptapExtensions: [import_extension.bold],
  shortcuts: [
    (0, import_ricos_types.createShortcut)({
      name: import_v1.Decoration_Type.BOLD,
      description: "ShortcutsSheet_Formatting_Bold",
      keys: { macOs: "Meta+B", windows: "Ctrl+B" },
      command({ publicCommands }) {
        publicCommands.toggleInlineStyle("bold");
        return "continue";
      },
      group: "ShortcutsSheet_Section_Formatting",
      enabled: true,
      showInShortcutSheet: true
    })
  ],
  textButtons: [
    {
      id: import_v1.Decoration_Type.BOLD,
      type: "toggle",
      presentation: {
        dataHook: import_data_hooks.TEXT_BUTTON_BOLD,
        tooltip: "BoldButton_Tooltip",
        icon: import_wix_ui_icons_common.Bold
      },
      attributes: {
        visible: import_is_always_visible.isAlwaysVisible,
        active: import_is_bold_in_text.isBoldInText,
        disabled: import_is_node_selected.isNodeSelected
      },
      command: ({ publicCommands }) => () => {
        publicCommands.toggleInlineStyle("bold");
        return true;
      }
    }
  ]
};
