// @wix/ricos/dist/cjs/ricos-common-plugins/bold/bold-converter.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var bold_converter_exports = {};
__export(bold_converter_exports, {
  boldConverter: () => boldConverter
});
module.exports = __toCommonJS(bold_converter_exports);
var import_v1 = require("./../../ricos-schema/legacy/wix/rich_content/v1/index.js");
const boldConverter = {
  fromPm: {
    type: import_v1.Decoration_Type.BOLD,
    convert: (mark) => {
      const { type: _, attrs } = mark;
      return [{ type: import_v1.Decoration_Type.BOLD, ...attrs }];
    }
  },
  toPm: {
    type: import_v1.Decoration_Type.BOLD,
    convert: (decoration) => {
      const { type: _, ...data } = decoration;
      return {
        type: import_v1.Decoration_Type.BOLD,
        attrs: { ...data }
      };
    }
  }
};
