// @wix/auto-patterns/dist/esm/components/modals/actions/BulkDeleteModal.js
import { Box, CustomModalLayout, Modal, Text } from '@wix/design-system';
import React from 'react';
export const BulkDeleteModal = _ref => {
  var _modalParams$title, _modalParams$actions, _modalParams$actions2, _modalParams$descript;
  let {
    sdk,
    modalParams,
    actionParams
  } = _ref;
  const {
    getOptimisticActions,
    collectionId,
    closeModal
  } = sdk;
  const optimisticActions = getOptimisticActions(collectionId);
  return /*#__PURE__*/React.createElement(Modal, {
    isOpen: true,
    onRequestClose: closeModal
  }, /*#__PURE__*/React.createElement(CustomModalLayout, {
    title: ((_modalParams$title = modalParams.title) == null ? void 0 : _modalParams$title.text) || 'Delete item?',
    theme: "destructive",
    onCloseButtonClick: () => {
      closeModal();
    },
    primaryButtonOnClick: () => {
      var _modalParams$feedback;
      const schema = sdk.getSchema(sdk.collectionId);
      if (!schema) {
        throw new Error('Schema not found');
      }
      const selectedIds = actionParams.selectedValues.map(value => value[schema.idField]);
      optimisticActions == null || optimisticActions.deleteMany(actionParams.selectedValues, {
        successToast: ((_modalParams$feedback = modalParams.feedback) == null || (_modalParams$feedback = _modalParams$feedback.successToast) == null ? void 0 : _modalParams$feedback.text) || 'Items deleted successfully',
        errorToast: () => {
          var _modalParams$feedback2;
          return ((_modalParams$feedback2 = modalParams.feedback) == null || (_modalParams$feedback2 = _modalParams$feedback2.errorToast) == null ? void 0 : _modalParams$feedback2.text) || 'Failed to delete items';
        },
        submit: async () => {
          const result = await schema.actions.bulkDelete(selectedIds);
          return [result];
        }
      });
      closeModal();
    },
    secondaryButtonOnClick: () => {
      closeModal();
    },
    primaryButtonText: ((_modalParams$actions = modalParams.actions) == null || (_modalParams$actions = _modalParams$actions.submit) == null ? void 0 : _modalParams$actions.text) || 'Save',
    secondaryButtonText: ((_modalParams$actions2 = modalParams.actions) == null || (_modalParams$actions2 = _modalParams$actions2.cancel) == null ? void 0 : _modalParams$actions2.text) || 'Cancel',
    content: /*#__PURE__*/React.createElement(Box, {
      direction: "vertical",
      width: "552px",
      gap: "SP2",
      paddingTop: "SP1",
      paddingBottom: "SP1"
    }, /*#__PURE__*/React.createElement(Box, null, /*#__PURE__*/React.createElement(Text, null, ((_modalParams$descript = modalParams.description) == null ? void 0 : _modalParams$descript.text) || 'This items will be deleted')))
  }));
};
