// @wix/auto-patterns/dist/esm/components/modals/actions/CreateModal.js
import { Box, CustomModalLayout, Input, Modal, Text } from '@wix/design-system';
import React, { useState } from 'react';
export const CreateModal = _ref => {
  var _modalParams$title, _modalParams$actions, _modalParams$actions2;
  let {
    sdk,
    modalParams
  } = _ref;
  const [values, setValues] = useState(modalParams.fields.reduce((acc, field) => {
    acc[field.id] = '';
    return acc;
  }, {}));
  const {
    closeModal
  } = sdk;
  const {
    getOptimisticActions,
    collectionId
  } = sdk;
  const optimisticActions = getOptimisticActions(collectionId);
  return /*#__PURE__*/React.createElement(Modal, {
    isOpen: true,
    onRequestClose: closeModal
  }, /*#__PURE__*/React.createElement(CustomModalLayout, {
    title: ((_modalParams$title = modalParams.title) == null ? void 0 : _modalParams$title.text) || 'Create',
    onCloseButtonClick: () => {
      closeModal();
    },
    primaryButtonOnClick: () => {
      var _modalParams$feedback;
      const schema = sdk.getSchema(sdk.collectionId);
      const createdItem = {
        ...values
      };
      optimisticActions == null || optimisticActions.createOne(createdItem, {
        successToast: ((_modalParams$feedback = modalParams.feedback) == null || (_modalParams$feedback = _modalParams$feedback.successToast) == null ? void 0 : _modalParams$feedback.text) || 'Item created successfully',
        errorToast: () => {
          var _modalParams$feedback2;
          return ((_modalParams$feedback2 = modalParams.feedback) == null || (_modalParams$feedback2 = _modalParams$feedback2.errorToast) == null ? void 0 : _modalParams$feedback2.text) || 'Failed to create item';
        },
        submit: async () => {
          if (!schema) {
            throw new Error('Schema not found');
          }
          const result = await schema.actions.create(createdItem);
          return [result];
        }
      });
      closeModal();
    },
    secondaryButtonOnClick: () => {
      closeModal();
    },
    primaryButtonText: ((_modalParams$actions = modalParams.actions) == null || (_modalParams$actions = _modalParams$actions.submit) == null ? void 0 : _modalParams$actions.text) || 'Save',
    secondaryButtonText: ((_modalParams$actions2 = modalParams.actions) == null || (_modalParams$actions2 = _modalParams$actions2.cancel) == null ? void 0 : _modalParams$actions2.text) || 'Cancel',
    content: /*#__PURE__*/React.createElement(Box, {
      direction: "vertical",
      width: "552px",
      gap: "SP2",
      paddingTop: "SP1",
      paddingBottom: "SP1"
    }, modalParams.fields.map(field => {
      return /*#__PURE__*/React.createElement(Box, {
        key: field.id,
        direction: "vertical",
        width: "100%",
        gap: "SP2"
      }, /*#__PURE__*/React.createElement(Box, null, /*#__PURE__*/React.createElement(Text, null, field.label)), /*#__PURE__*/React.createElement(Box, null, /*#__PURE__*/React.createElement(Input, {
        value: values[field.id],
        onChange: e => {
          setValues(prev => ({
            ...prev,
            [field.id]: e.target.value
          }));
        }
      })));
    }))
  }));
};
