// @wix/monitoring-astro/src/client/setup-client.ts
import { outdent } from 'outdent';
import { METHODS } from '../constants.js';
import { log, overrideConsoleMethods } from '../monitoring.js';
import type { LogParams } from '../types.js';

declare global {
  interface Window {
    __wixMonitoring?: {
      originalMethods: Record<LogParams['method'], (...args: unknown[]) => void>;
      queue: Array<{
        args: unknown[];
        method: LogParams['method'];
        timestamp: number;
        stack: string;
      }>;
    };
  }
}

export const getInlineMonitoringSnippet = () => outdent`
  if (!window.__wixMonitoring && !location.pathname.startsWith('/_wix/extensions/backoffice')) {
    window.__wixMonitoring = {
      queue: [],
      originalMethods: {},
    };

    ${JSON.stringify(METHODS)}.forEach((method) => {
      window.__wixMonitoring.originalMethods[method] = console[method];

      console[method] = (...args) => {
        let stack = '';

        try { throw Error() } catch(e) { stack = e.stack; } 

        window.__wixMonitoring.queue.push({method, stack, timestamp: Date.now(), args });
        window.__wixMonitoring.originalMethods[method].apply(console, args);
      };
    });
  }
  `

export function setupClientMonitoring(): void {
  if (location.pathname.startsWith('/_wix/extensions/backoffice')) {
    return;
  }

  if (!window.__wixMonitoring) {
    overrideConsoleMethods(console);
    return;
  }

  const { originalMethods, queue } = window.__wixMonitoring;

  overrideConsoleMethods(originalMethods);

  for (const { method, stack, timestamp, args } of queue) {
    log({ method, stack, timestamp }, ...args);
  }

  delete window.__wixMonitoring;
}
