// @wix/auto_sdk_seo_seo-sitemap-entries/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  listSitemapPages: () => listSitemapPages2
});
module.exports = __toCommonJS(meta_exports);

// src/promote-seo-v1-headless-sitemap-entry-seo-sitemap-entries.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixPromoteSeoHeadlessSitemapServiceV1SeoHeadlessSitemapServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/seo-headless-sitemap/v1/list-static-sitemap-pages",
        destPath: "/v1/list-static-sitemap-pages"
      }
    ],
    _: [
      {
        srcPath: "/_api/seo-headless-sitemap/v1/list-sitemap-pages",
        destPath: "/v1/list-sitemap-pages"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/seo-headless-sitemap/v1/list-sitemap-pages",
        destPath: "/v1/list-sitemap-pages"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/v1/list-sitemap-pages",
        destPath: "/v1/list-sitemap-pages"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_seo_seo-sitemap-entries";
function listSitemapPages(payload) {
  function __listSitemapPages({ host }) {
    const metadata = {
      entityFqdn: "wix.promote.seo.v1.headless_sitemap_entry",
      method: "GET",
      methodFqn: "wix.promote.seo.headless.sitemap.service.v1.SeoHeadlessSitemapService.ListSitemapPages",
      packageName: PACKAGE_NAME,
      url: resolveWixPromoteSeoHeadlessSitemapServiceV1SeoHeadlessSitemapServiceUrl(
        { protoPath: "/v1/list-sitemap-pages", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __listSitemapPages;
}

// src/promote-seo-v1-headless-sitemap-entry-seo-sitemap-entries.meta.ts
function listSitemapPages2() {
  const payload = {};
  const getRequestOptions = listSitemapPages(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/list-sitemap-pages",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  listSitemapPages
});
