// @wix/auto_sdk_pricing-plans_orders/build/internal/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  CancellationCause: () => CancellationCause,
  CancellationEffectiveAt: () => CancellationEffectiveAt,
  OrderMethod: () => OrderMethod,
  OrderStatus: () => OrderStatus,
  OrderType: () => OrderType,
  PaymentStatus: () => PaymentStatus,
  PeriodUnit: () => PeriodUnit,
  ReasonNotSuspendable: () => ReasonNotSuspendable,
  Set: () => Set,
  SortOrder: () => SortOrder,
  Status: () => Status,
  WebhookIdentityType: () => WebhookIdentityType,
  cancelOrder: () => cancelOrder4,
  createOfflineOrder: () => createOfflineOrder4,
  createOnlineOrder: () => createOnlineOrder4,
  getOfflineOrderPreview: () => getOfflineOrderPreview4,
  getOnlineOrderPreview: () => getOnlineOrderPreview4,
  getPricePreview: () => getPricePreview4,
  managementGetOrder: () => managementGetOrder4,
  managementListOrders: () => managementListOrders4,
  markAsPaid: () => markAsPaid4,
  memberGetOrder: () => memberGetOrder4,
  memberListOrders: () => memberListOrders4,
  onOrderAutoRenewCanceled: () => onOrderAutoRenewCanceled2,
  onOrderCanceled: () => onOrderCanceled2,
  onOrderCreated: () => onOrderCreated2,
  onOrderCycleStarted: () => onOrderCycleStarted2,
  onOrderEndDatePostponed: () => onOrderEndDatePostponed2,
  onOrderEnded: () => onOrderEnded2,
  onOrderMarkedAsPaid: () => onOrderMarkedAsPaid2,
  onOrderPaused: () => onOrderPaused2,
  onOrderPurchased: () => onOrderPurchased2,
  onOrderResumed: () => onOrderResumed2,
  onOrderStartDateChanged: () => onOrderStartDateChanged2,
  onOrderStarted: () => onOrderStarted2,
  onOrderUpdated: () => onOrderUpdated2,
  pauseOrder: () => pauseOrder4,
  postponeEndDate: () => postponeEndDate4,
  requestCancellation: () => requestCancellation4,
  resumeOrder: () => resumeOrder4
});
module.exports = __toCommonJS(index_exports);

// src/pricing-plans-v2-order-orders.public.ts
var import_rename_all_nested_keys2 = require("@wix/sdk-runtime/rename-all-nested-keys");
var import_timestamp3 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var import_sdk_types = require("@wix/sdk-types");

// src/pricing-plans-v2-order-orders.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/pricing-plans-v2-order-orders.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressMembershipV2OrdersCheckoutServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/pricing-plans-checkout-proxy",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/paid-plans/v2/checkout",
        destPath: "/v2/checkout"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/paid-plans/v2/checkout",
        destPath: "/v2/checkout"
      }
    ],
    "*.wixapps.net": [
      {
        srcPath: "/_api/paid-plans/v2/checkout",
        destPath: "/v2/checkout"
      }
    ],
    _: [
      {
        srcPath: "/pricing-plans/v2/checkout",
        destPath: "/v2/checkout"
      },
      {
        srcPath: "/_api/paid-plans/v2/checkout",
        destPath: "/v2/checkout"
      },
      {
        srcPath: "/_api/pricing-plans/v2/checkout",
        destPath: "/v2/checkout"
      }
    ],
    "editor-flow.wixapps.net": [
      {
        srcPath: "/_api/paid-plans/v2/checkout",
        destPath: "/v2/checkout"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/paid-plans/v2/checkout",
        destPath: "/v2/checkout"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/paid-plans/v2/checkout",
        destPath: "/v2/checkout"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/paid-plans/v2/checkout",
        destPath: "/v2/checkout"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/paid-plans/v2/checkout",
        destPath: "/v2/checkout"
      }
    ],
    "forums._base_domain_": [
      {
        srcPath: "/_api/paid-plans/v2/checkout",
        destPath: "/v2/checkout"
      }
    ],
    "forums.wixapps.net": [
      {
        srcPath: "/_api/paid-plans/v2/checkout",
        destPath: "/v2/checkout"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/pricing-plans/v2/checkout",
        destPath: "/v2/checkout"
      },
      {
        srcPath: "/_api/paid-plans/v2/checkout",
        destPath: "/v2/checkout"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/paid-plans/v2/checkout",
        destPath: "/v2/checkout"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/pricing-plans/v2/checkout",
        destPath: "/v2/checkout"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
function resolveComWixpressMembershipV2OrdersMemberMemberOrdersServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/member-orders-service-proxy",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/pricing-plans-member-orders/v2/member/orders",
        destPath: "/v2/member/orders"
      },
      {
        srcPath: "/_api/pricing-plans/v2/member/orders",
        destPath: "/v2/member/orders"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/pricing-plans-member-orders/v2/member/orders",
        destPath: "/v2/member/orders"
      },
      {
        srcPath: "/_api/pricing-plans/v2/member/orders",
        destPath: "/v2/member/orders"
      }
    ],
    _: [
      {
        srcPath: "/_api/pricing-plans/v2/member/orders",
        destPath: "/v2/member/orders"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/pricing-plans/v2/member/orders",
        destPath: "/v2/member/orders"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/pricing-plans/v2/member/orders",
        destPath: "/v2/member/orders"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/pricing-plans/v2/member/orders",
        destPath: "/v2/member/orders"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/pricing-plans/v2/member/orders",
        destPath: "/v2/member/orders"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
function resolveComWixpressMembershipV2OrdersOrderManagementServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/order-management-service-proxy",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/paid-plans/v2/orders",
        destPath: "/v2/orders"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/paid-plans/v2/orders",
        destPath: "/v2/orders"
      }
    ],
    "*.wixapps.net": [
      {
        srcPath: "/_api/paid-plans/v2/orders",
        destPath: "/v2/orders"
      }
    ],
    _: [
      {
        srcPath: "/_api/paid-plans/v2/orders",
        destPath: "/v2/orders"
      }
    ],
    "editor-flow.wixapps.net": [
      {
        srcPath: "/_api/paid-plans/v2/orders",
        destPath: "/v2/orders"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/paid-plans/v2/orders",
        destPath: "/v2/orders"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/paid-plans/v2/orders",
        destPath: "/v2/orders"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/paid-plans/v2/orders",
        destPath: "/v2/orders"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/paid-plans/v2/orders",
        destPath: "/v2/orders"
      }
    ],
    "forums._base_domain_": [
      {
        srcPath: "/_api/paid-plans/v2/orders",
        destPath: "/v2/orders"
      }
    ],
    "forums.wixapps.net": [
      {
        srcPath: "/_api/paid-plans/v2/orders",
        destPath: "/v2/orders"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/pricing-plans/v2/orders",
        destPath: "/v2/orders"
      },
      {
        srcPath: "/_api/paid-plans/v2/orders",
        destPath: "/v2/orders"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/paid-plans/v2/orders",
        destPath: "/v2/orders"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/pricing-plans/v2/orders",
        destPath: "/v2/orders"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_pricing-plans_orders";
function managementGetOrder(payload) {
  function __managementGetOrder({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.order",
      method: "GET",
      methodFqn: "com.wixpress.membership.v2.orders.OrderManagementService.GetOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2OrdersOrderManagementServiceUrl({
        protoPath: "/v2/orders/{id}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.startDate" },
            { path: "order.endDate" },
            { path: "order.earliestEndDate" },
            { path: "order.createdDate" },
            { path: "order.updatedDate" },
            { path: "order.cancellation.requestedDate" },
            { path: "order.pausePeriods.pauseDate" },
            { path: "order.pausePeriods.resumeDate" },
            { path: "order.currentCycle.startedDate" },
            { path: "order.currentCycle.endedDate" },
            { path: "order.cycles.startedDate" },
            { path: "order.cycles.endedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __managementGetOrder;
}
function managementListOrders(payload) {
  function __managementListOrders({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.order",
      method: "GET",
      methodFqn: "com.wixpress.membership.v2.orders.OrderManagementService.ListOrders",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2OrdersOrderManagementServiceUrl({
        protoPath: "/v2/orders",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload, true),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "orders.startDate" },
            { path: "orders.endDate" },
            { path: "orders.earliestEndDate" },
            { path: "orders.createdDate" },
            { path: "orders.updatedDate" },
            { path: "orders.cancellation.requestedDate" },
            { path: "orders.pausePeriods.pauseDate" },
            { path: "orders.pausePeriods.resumeDate" },
            { path: "orders.currentCycle.startedDate" },
            { path: "orders.currentCycle.endedDate" },
            { path: "orders.cycles.startedDate" },
            { path: "orders.cycles.endedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __managementListOrders;
}
function postponeEndDate(payload) {
  function __postponeEndDate({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "endDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.order",
      method: "PATCH",
      methodFqn: "com.wixpress.membership.v2.orders.OrderManagementService.PostponeEndDate",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2OrdersOrderManagementServiceUrl({
        protoPath: "/v2/orders/{id}",
        data: serializedData,
        host
      }),
      data: serializedData
    };
    return metadata;
  }
  return __postponeEndDate;
}
function cancelOrder(payload) {
  function __cancelOrder({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.order",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.orders.OrderManagementService.CancelOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2OrdersOrderManagementServiceUrl({
        protoPath: "/v2/orders/{id}/cancel",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __cancelOrder;
}
function markAsPaid(payload) {
  function __markAsPaid({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.order",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.orders.OrderManagementService.MarkAsPaid",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2OrdersOrderManagementServiceUrl({
        protoPath: "/v2/orders/{id}/mark-as-paid",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __markAsPaid;
}
function pauseOrder(payload) {
  function __pauseOrder({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.order",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.orders.OrderManagementService.PauseOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2OrdersOrderManagementServiceUrl({
        protoPath: "/v2/orders/{id}/pause",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __pauseOrder;
}
function resumeOrder(payload) {
  function __resumeOrder({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.order",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.orders.OrderManagementService.ResumeOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2OrdersOrderManagementServiceUrl({
        protoPath: "/v2/orders/{id}/resume",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __resumeOrder;
}
function memberGetOrder(payload) {
  function __memberGetOrder({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.order",
      method: "GET",
      methodFqn: "com.wixpress.membership.v2.orders.member.MemberOrdersService.GetOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2OrdersMemberMemberOrdersServiceUrl({
        protoPath: "/v2/member/orders/{id}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.startDate" },
            { path: "order.endDate" },
            { path: "order.earliestEndDate" },
            { path: "order.createdDate" },
            { path: "order.updatedDate" },
            { path: "order.cancellation.requestedDate" },
            { path: "order.pausePeriods.pauseDate" },
            { path: "order.pausePeriods.resumeDate" },
            { path: "order.currentCycle.startedDate" },
            { path: "order.currentCycle.endedDate" },
            { path: "order.cycles.startedDate" },
            { path: "order.cycles.endedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __memberGetOrder;
}
function memberListOrders(payload) {
  function __memberListOrders({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.order",
      method: "GET",
      methodFqn: "com.wixpress.membership.v2.orders.member.MemberOrdersService.ListOrders",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2OrdersMemberMemberOrdersServiceUrl({
        protoPath: "/v2/member/orders",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload, true),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "orders.startDate" },
            { path: "orders.endDate" },
            { path: "orders.earliestEndDate" },
            { path: "orders.createdDate" },
            { path: "orders.updatedDate" },
            { path: "orders.cancellation.requestedDate" },
            { path: "orders.pausePeriods.pauseDate" },
            { path: "orders.pausePeriods.resumeDate" },
            { path: "orders.currentCycle.startedDate" },
            { path: "orders.currentCycle.endedDate" },
            { path: "orders.cycles.startedDate" },
            { path: "orders.cycles.endedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __memberListOrders;
}
function requestCancellation(payload) {
  function __requestCancellation({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.order",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.orders.member.MemberOrdersService.RequestCancellation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2OrdersMemberMemberOrdersServiceUrl({
        protoPath: "/v2/member/orders/{id}/cancel",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __requestCancellation;
}
function createOnlineOrder(payload) {
  function __createOnlineOrder({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "startDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.order",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.orders.CheckoutService.CreateOnlineOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2OrdersCheckoutServiceUrl({
        protoPath: "/v2/checkout/orders/online",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.startDate" },
            { path: "order.endDate" },
            { path: "order.earliestEndDate" },
            { path: "order.createdDate" },
            { path: "order.updatedDate" },
            { path: "order.cancellation.requestedDate" },
            { path: "order.pausePeriods.pauseDate" },
            { path: "order.pausePeriods.resumeDate" },
            { path: "order.currentCycle.startedDate" },
            { path: "order.currentCycle.endedDate" },
            { path: "order.cycles.startedDate" },
            { path: "order.cycles.endedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createOnlineOrder;
}
function createOfflineOrder(payload) {
  function __createOfflineOrder({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "startDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.order",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.orders.CheckoutService.CreateOfflineOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2OrdersCheckoutServiceUrl({
        protoPath: "/v2/checkout/orders/offline",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.startDate" },
            { path: "order.endDate" },
            { path: "order.earliestEndDate" },
            { path: "order.createdDate" },
            { path: "order.updatedDate" },
            { path: "order.cancellation.requestedDate" },
            { path: "order.pausePeriods.pauseDate" },
            { path: "order.pausePeriods.resumeDate" },
            { path: "order.currentCycle.startedDate" },
            { path: "order.currentCycle.endedDate" },
            { path: "order.cycles.startedDate" },
            { path: "order.cycles.endedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createOfflineOrder;
}
function getOnlineOrderPreview(payload) {
  function __getOnlineOrderPreview({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "startDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.order",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.orders.CheckoutService.GetOnlineOrderPreview",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2OrdersCheckoutServiceUrl({
        protoPath: "/v2/checkout/orders/preview-online",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.startDate" },
            { path: "order.endDate" },
            { path: "order.earliestEndDate" },
            { path: "order.createdDate" },
            { path: "order.updatedDate" },
            { path: "order.cancellation.requestedDate" },
            { path: "order.pausePeriods.pauseDate" },
            { path: "order.pausePeriods.resumeDate" },
            { path: "order.currentCycle.startedDate" },
            { path: "order.currentCycle.endedDate" },
            { path: "order.cycles.startedDate" },
            { path: "order.cycles.endedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getOnlineOrderPreview;
}
function getOfflineOrderPreview(payload) {
  function __getOfflineOrderPreview({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "startDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.order",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.orders.CheckoutService.GetOfflineOrderPreview",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2OrdersCheckoutServiceUrl({
        protoPath: "/v2/checkout/orders/preview-offline",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.startDate" },
            { path: "order.endDate" },
            { path: "order.earliestEndDate" },
            { path: "order.createdDate" },
            { path: "order.updatedDate" },
            { path: "order.cancellation.requestedDate" },
            { path: "order.pausePeriods.pauseDate" },
            { path: "order.pausePeriods.resumeDate" },
            { path: "order.currentCycle.startedDate" },
            { path: "order.currentCycle.endedDate" },
            { path: "order.cycles.startedDate" },
            { path: "order.cycles.endedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getOfflineOrderPreview;
}
function getPricePreview(payload) {
  function __getPricePreview({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.order",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.orders.CheckoutService.GetPricePreview",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2OrdersCheckoutServiceUrl({
        protoPath: "/v2/checkout/orders/price-preview",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __getPricePreview;
}

// src/pricing-plans-v2-order-orders.universal.ts
var PeriodUnit = /* @__PURE__ */ ((PeriodUnit2) => {
  PeriodUnit2["UNDEFINED"] = "UNDEFINED";
  PeriodUnit2["DAY"] = "DAY";
  PeriodUnit2["WEEK"] = "WEEK";
  PeriodUnit2["MONTH"] = "MONTH";
  PeriodUnit2["YEAR"] = "YEAR";
  return PeriodUnit2;
})(PeriodUnit || {});
var OrderType = /* @__PURE__ */ ((OrderType2) => {
  OrderType2["UNDEFINED"] = "UNDEFINED";
  OrderType2["ONLINE"] = "ONLINE";
  OrderType2["OFFLINE"] = "OFFLINE";
  OrderType2["EXTERNAL"] = "EXTERNAL";
  return OrderType2;
})(OrderType || {});
var OrderMethod = /* @__PURE__ */ ((OrderMethod2) => {
  OrderMethod2["UNKNOWN"] = "UNKNOWN";
  OrderMethod2["MOTO"] = "MOTO";
  OrderMethod2["POS"] = "POS";
  return OrderMethod2;
})(OrderMethod || {});
var OrderStatus = /* @__PURE__ */ ((OrderStatus2) => {
  OrderStatus2["UNDEFINED"] = "UNDEFINED";
  OrderStatus2["DRAFT"] = "DRAFT";
  OrderStatus2["PENDING"] = "PENDING";
  OrderStatus2["ACTIVE"] = "ACTIVE";
  OrderStatus2["PAUSED"] = "PAUSED";
  OrderStatus2["ENDED"] = "ENDED";
  OrderStatus2["CANCELED"] = "CANCELED";
  return OrderStatus2;
})(OrderStatus || {});
var CancellationCause = /* @__PURE__ */ ((CancellationCause2) => {
  CancellationCause2["UNDEFINED"] = "UNDEFINED";
  CancellationCause2["OWNER_ACTION"] = "OWNER_ACTION";
  CancellationCause2["MEMBER_ACTION"] = "MEMBER_ACTION";
  CancellationCause2["PAYMENT_FAILURE"] = "PAYMENT_FAILURE";
  CancellationCause2["PAYMENT_SETUP_FAILURE"] = "PAYMENT_SETUP_FAILURE";
  CancellationCause2["UNKNOWN"] = "UNKNOWN";
  return CancellationCause2;
})(CancellationCause || {});
var CancellationEffectiveAt = /* @__PURE__ */ ((CancellationEffectiveAt2) => {
  CancellationEffectiveAt2["UNDEFINED"] = "UNDEFINED";
  CancellationEffectiveAt2["IMMEDIATELY"] = "IMMEDIATELY";
  CancellationEffectiveAt2["NEXT_PAYMENT_DATE"] = "NEXT_PAYMENT_DATE";
  return CancellationEffectiveAt2;
})(CancellationEffectiveAt || {});
var PaymentStatus = /* @__PURE__ */ ((PaymentStatus2) => {
  PaymentStatus2["UNDEFINED"] = "UNDEFINED";
  PaymentStatus2["PAID"] = "PAID";
  PaymentStatus2["REFUNDED"] = "REFUNDED";
  PaymentStatus2["FAILED"] = "FAILED";
  PaymentStatus2["UNPAID"] = "UNPAID";
  PaymentStatus2["PENDING"] = "PENDING";
  PaymentStatus2["NOT_APPLICABLE"] = "NOT_APPLICABLE";
  return PaymentStatus2;
})(PaymentStatus || {});
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["UNDEFINED"] = "UNDEFINED";
  Status2["ACTIVE"] = "ACTIVE";
  Status2["ENDED"] = "ENDED";
  return Status2;
})(Status || {});
var Set = /* @__PURE__ */ ((Set2) => {
  Set2["UNKNOWN_SET"] = "UNKNOWN_SET";
  Set2["BASIC"] = "BASIC";
  Set2["FULL"] = "FULL";
  return Set2;
})(Set || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var ReasonNotSuspendable = /* @__PURE__ */ ((ReasonNotSuspendable2) => {
  ReasonNotSuspendable2["UNDEFINED"] = "UNDEFINED";
  ReasonNotSuspendable2["PENDING"] = "PENDING";
  ReasonNotSuspendable2["TRIAL"] = "TRIAL";
  ReasonNotSuspendable2["CANCELED"] = "CANCELED";
  ReasonNotSuspendable2["ENDED"] = "ENDED";
  ReasonNotSuspendable2["NOT_STARTED"] = "NOT_STARTED";
  ReasonNotSuspendable2["ALREADY_SUSPENDED"] = "ALREADY_SUSPENDED";
  ReasonNotSuspendable2["OLD_STRIPE"] = "OLD_STRIPE";
  return ReasonNotSuspendable2;
})(ReasonNotSuspendable || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function managementGetOrder2(_id, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    id: _id,
    fieldSet: options?.fieldSet
  });
  const reqOpts = managementGetOrder(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { id: "$[0]", fieldSet: "$[1].fieldSet" },
        singleArgumentUnchanged: false
      },
      ["_id", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function managementListOrders2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    buyerIds: options?.buyerIds,
    planIds: options?.planIds,
    autoRenewCanceled: options?.autoRenewCanceled,
    orderStatuses: options?.orderStatuses,
    paymentStatuses: options?.paymentStatuses,
    limit: options?.limit,
    offset: options?.offset,
    sorting: options?.sorting,
    fieldSet: options?.fieldSet
  });
  const reqOpts = managementListOrders(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          buyerIds: "$[0].buyerIds",
          planIds: "$[0].planIds",
          autoRenewCanceled: "$[0].autoRenewCanceled",
          orderStatuses: "$[0].orderStatuses",
          paymentStatuses: "$[0].paymentStatuses",
          limit: "$[0].limit",
          offset: "$[0].offset",
          sorting: "$[0].sorting",
          fieldSet: "$[0].fieldSet"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function postponeEndDate2(_id, endDate) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    id: _id,
    endDate
  });
  const reqOpts = postponeEndDate(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { id: "$[0]", endDate: "$[1]" },
        singleArgumentUnchanged: false
      },
      ["_id", "endDate"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function cancelOrder2(_id, effectiveAt) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    id: _id,
    effectiveAt
  });
  const reqOpts = cancelOrder(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { id: "$[0]", effectiveAt: "$[1]" },
        singleArgumentUnchanged: false
      },
      ["_id", "effectiveAt"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function markAsPaid2(_id) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ id: _id });
  const reqOpts = markAsPaid(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { id: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function pauseOrder2(_id) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ id: _id });
  const reqOpts = pauseOrder(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { id: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function resumeOrder2(_id) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ id: _id });
  const reqOpts = resumeOrder(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { id: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function memberGetOrder2(_id, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    id: _id,
    fieldSet: options?.fieldSet
  });
  const reqOpts = memberGetOrder(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.order;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { id: "$[0]", fieldSet: "$[1].fieldSet" },
        singleArgumentUnchanged: false
      },
      ["_id", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function memberListOrders2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    planIds: options?.planIds,
    autoRenewCanceled: options?.autoRenewCanceled,
    orderStatuses: options?.orderStatuses,
    paymentStatuses: options?.paymentStatuses,
    limit: options?.limit,
    offset: options?.offset,
    sorting: options?.sorting,
    fieldSet: options?.fieldSet
  });
  const reqOpts = memberListOrders(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          planIds: "$[0].planIds",
          autoRenewCanceled: "$[0].autoRenewCanceled",
          orderStatuses: "$[0].orderStatuses",
          paymentStatuses: "$[0].paymentStatuses",
          limit: "$[0].limit",
          offset: "$[0].offset",
          sorting: "$[0].sorting",
          fieldSet: "$[0].fieldSet"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function requestCancellation2(_id, effectiveAt) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    id: _id,
    effectiveAt
  });
  const reqOpts = requestCancellation(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { id: "$[0]", effectiveAt: "$[1]" },
        singleArgumentUnchanged: false
      },
      ["_id", "effectiveAt"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function createOnlineOrder2(planId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    planId,
    startDate: options?.startDate,
    couponCode: options?.couponCode,
    onBehalf: options?.onBehalf,
    submissionId: options?.submissionId
  });
  const reqOpts = createOnlineOrder(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          planId: "$[0]",
          startDate: "$[1].startDate",
          couponCode: "$[1].couponCode",
          onBehalf: "$[1].onBehalf",
          submissionId: "$[1].submissionId"
        },
        singleArgumentUnchanged: false
      },
      ["planId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function createOfflineOrder2(planId, memberId, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    planId,
    memberId,
    startDate: options?.startDate,
    paid: options?.paid,
    couponCode: options?.couponCode,
    submissionId: options?.submissionId
  });
  const reqOpts = createOfflineOrder(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          planId: "$[0]",
          memberId: "$[1]",
          startDate: "$[2].startDate",
          paid: "$[2].paid",
          couponCode: "$[2].couponCode",
          submissionId: "$[2].submissionId"
        },
        singleArgumentUnchanged: false
      },
      ["planId", "memberId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getOnlineOrderPreview2(planId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    planId,
    startDate: options?.startDate,
    couponCode: options?.couponCode
  });
  const reqOpts = getOnlineOrderPreview(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          planId: "$[0]",
          startDate: "$[1].startDate",
          couponCode: "$[1].couponCode"
        },
        singleArgumentUnchanged: false
      },
      ["planId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getOfflineOrderPreview2(planId, memberId, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    planId,
    memberId,
    startDate: options?.startDate,
    couponCode: options?.couponCode
  });
  const reqOpts = getOfflineOrderPreview(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          planId: "$[0]",
          memberId: "$[1]",
          startDate: "$[2].startDate",
          couponCode: "$[2].couponCode"
        },
        singleArgumentUnchanged: false
      },
      ["planId", "memberId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getPricePreview2(planId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    planId,
    couponCode: options?.couponCode
  });
  const reqOpts = getPricePreview(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          planId: "$[0]",
          couponCode: "$[1].couponCode"
        },
        singleArgumentUnchanged: false
      },
      ["planId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/pricing-plans-v2-order-orders.public.ts
function managementGetOrder3(httpClient) {
  return (_id, options) => managementGetOrder2(
    _id,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function managementListOrders3(httpClient) {
  return (options) => managementListOrders2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function postponeEndDate3(httpClient) {
  return (_id, endDate) => postponeEndDate2(
    _id,
    endDate,
    // @ts-ignore
    { httpClient }
  );
}
function cancelOrder3(httpClient) {
  return (_id, effectiveAt) => cancelOrder2(
    _id,
    effectiveAt,
    // @ts-ignore
    { httpClient }
  );
}
function markAsPaid3(httpClient) {
  return (_id) => markAsPaid2(
    _id,
    // @ts-ignore
    { httpClient }
  );
}
function pauseOrder3(httpClient) {
  return (_id) => pauseOrder2(
    _id,
    // @ts-ignore
    { httpClient }
  );
}
function resumeOrder3(httpClient) {
  return (_id) => resumeOrder2(
    _id,
    // @ts-ignore
    { httpClient }
  );
}
function memberGetOrder3(httpClient) {
  return (_id, options) => memberGetOrder2(
    _id,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function memberListOrders3(httpClient) {
  return (options) => memberListOrders2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function requestCancellation3(httpClient) {
  return (_id, effectiveAt) => requestCancellation2(
    _id,
    effectiveAt,
    // @ts-ignore
    { httpClient }
  );
}
function createOnlineOrder3(httpClient) {
  return (planId, options) => createOnlineOrder2(
    planId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function createOfflineOrder3(httpClient) {
  return (planId, memberId, options) => createOfflineOrder2(
    planId,
    memberId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function getOnlineOrderPreview3(httpClient) {
  return (planId, options) => getOnlineOrderPreview2(
    planId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function getOfflineOrderPreview3(httpClient) {
  return (planId, memberId, options) => getOfflineOrderPreview2(
    planId,
    memberId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function getPricePreview3(httpClient) {
  return (planId, options) => getPricePreview2(
    planId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
var onOrderAutoRenewCanceled = (0, import_sdk_types.EventDefinition)(
  "wix.pricing_plans.v2.order_auto_renew_canceled",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths2.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "data.order.startDate" },
          { path: "data.order.endDate" },
          { path: "data.order.earliestEndDate" },
          { path: "data.order.createdDate" },
          { path: "data.order.updatedDate" },
          { path: "data.order.cancellation.requestedDate" },
          { path: "data.order.pausePeriods.pauseDate" },
          { path: "data.order.pausePeriods.resumeDate" },
          { path: "data.order.currentCycle.startedDate" },
          { path: "data.order.currentCycle.endedDate" },
          { path: "data.order.cycles.startedDate" },
          { path: "data.order.cycles.endedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onOrderCanceled = (0, import_sdk_types.EventDefinition)(
  "wix.pricing_plans.v2.order_canceled",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths2.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "data.order.startDate" },
          { path: "data.order.endDate" },
          { path: "data.order.earliestEndDate" },
          { path: "data.order.createdDate" },
          { path: "data.order.updatedDate" },
          { path: "data.order.cancellation.requestedDate" },
          { path: "data.order.pausePeriods.pauseDate" },
          { path: "data.order.pausePeriods.resumeDate" },
          { path: "data.order.currentCycle.startedDate" },
          { path: "data.order.currentCycle.endedDate" },
          { path: "data.order.cycles.startedDate" },
          { path: "data.order.cycles.endedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onOrderCycleStarted = (0, import_sdk_types.EventDefinition)(
  "wix.pricing_plans.v2.order_cycle_started",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths2.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "data.order.startDate" },
          { path: "data.order.endDate" },
          { path: "data.order.earliestEndDate" },
          { path: "data.order.createdDate" },
          { path: "data.order.updatedDate" },
          { path: "data.order.cancellation.requestedDate" },
          { path: "data.order.pausePeriods.pauseDate" },
          { path: "data.order.pausePeriods.resumeDate" },
          { path: "data.order.currentCycle.startedDate" },
          { path: "data.order.currentCycle.endedDate" },
          { path: "data.order.cycles.startedDate" },
          { path: "data.order.cycles.endedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onOrderEndDatePostponed = (0, import_sdk_types.EventDefinition)(
  "wix.pricing_plans.v2.order_end_date_postponed",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths2.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "data.order.startDate" },
          { path: "data.order.endDate" },
          { path: "data.order.earliestEndDate" },
          { path: "data.order.createdDate" },
          { path: "data.order.updatedDate" },
          { path: "data.order.cancellation.requestedDate" },
          { path: "data.order.pausePeriods.pauseDate" },
          { path: "data.order.pausePeriods.resumeDate" },
          { path: "data.order.currentCycle.startedDate" },
          { path: "data.order.currentCycle.endedDate" },
          { path: "data.order.cycles.startedDate" },
          { path: "data.order.cycles.endedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onOrderEnded = (0, import_sdk_types.EventDefinition)(
  "wix.pricing_plans.v2.order_ended",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths2.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "data.order.startDate" },
          { path: "data.order.endDate" },
          { path: "data.order.earliestEndDate" },
          { path: "data.order.createdDate" },
          { path: "data.order.updatedDate" },
          { path: "data.order.cancellation.requestedDate" },
          { path: "data.order.pausePeriods.pauseDate" },
          { path: "data.order.pausePeriods.resumeDate" },
          { path: "data.order.currentCycle.startedDate" },
          { path: "data.order.currentCycle.endedDate" },
          { path: "data.order.cycles.startedDate" },
          { path: "data.order.cycles.endedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onOrderMarkedAsPaid = (0, import_sdk_types.EventDefinition)(
  "wix.pricing_plans.v2.order_marked_as_paid",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths2.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "data.order.startDate" },
          { path: "data.order.endDate" },
          { path: "data.order.earliestEndDate" },
          { path: "data.order.createdDate" },
          { path: "data.order.updatedDate" },
          { path: "data.order.cancellation.requestedDate" },
          { path: "data.order.pausePeriods.pauseDate" },
          { path: "data.order.pausePeriods.resumeDate" },
          { path: "data.order.currentCycle.startedDate" },
          { path: "data.order.currentCycle.endedDate" },
          { path: "data.order.cycles.startedDate" },
          { path: "data.order.cycles.endedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onOrderPaused = (0, import_sdk_types.EventDefinition)(
  "wix.pricing_plans.v2.order_paused",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths2.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "data.order.startDate" },
          { path: "data.order.endDate" },
          { path: "data.order.earliestEndDate" },
          { path: "data.order.createdDate" },
          { path: "data.order.updatedDate" },
          { path: "data.order.cancellation.requestedDate" },
          { path: "data.order.pausePeriods.pauseDate" },
          { path: "data.order.pausePeriods.resumeDate" },
          { path: "data.order.currentCycle.startedDate" },
          { path: "data.order.currentCycle.endedDate" },
          { path: "data.order.cycles.startedDate" },
          { path: "data.order.cycles.endedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onOrderPurchased = (0, import_sdk_types.EventDefinition)(
  "wix.pricing_plans.v2.order_purchased",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths2.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "data.order.startDate" },
          { path: "data.order.endDate" },
          { path: "data.order.earliestEndDate" },
          { path: "data.order.createdDate" },
          { path: "data.order.updatedDate" },
          { path: "data.order.cancellation.requestedDate" },
          { path: "data.order.pausePeriods.pauseDate" },
          { path: "data.order.pausePeriods.resumeDate" },
          { path: "data.order.currentCycle.startedDate" },
          { path: "data.order.currentCycle.endedDate" },
          { path: "data.order.cycles.startedDate" },
          { path: "data.order.cycles.endedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onOrderResumed = (0, import_sdk_types.EventDefinition)(
  "wix.pricing_plans.v2.order_resumed",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths2.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "data.order.startDate" },
          { path: "data.order.endDate" },
          { path: "data.order.earliestEndDate" },
          { path: "data.order.createdDate" },
          { path: "data.order.updatedDate" },
          { path: "data.order.cancellation.requestedDate" },
          { path: "data.order.pausePeriods.pauseDate" },
          { path: "data.order.pausePeriods.resumeDate" },
          { path: "data.order.currentCycle.startedDate" },
          { path: "data.order.currentCycle.endedDate" },
          { path: "data.order.cycles.startedDate" },
          { path: "data.order.cycles.endedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onOrderStarted = (0, import_sdk_types.EventDefinition)(
  "wix.pricing_plans.v2.order_started",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths2.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "data.order.startDate" },
          { path: "data.order.endDate" },
          { path: "data.order.earliestEndDate" },
          { path: "data.order.createdDate" },
          { path: "data.order.updatedDate" },
          { path: "data.order.cancellation.requestedDate" },
          { path: "data.order.pausePeriods.pauseDate" },
          { path: "data.order.pausePeriods.resumeDate" },
          { path: "data.order.currentCycle.startedDate" },
          { path: "data.order.currentCycle.endedDate" },
          { path: "data.order.cycles.startedDate" },
          { path: "data.order.cycles.endedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onOrderUpdated = (0, import_sdk_types.EventDefinition)(
  "wix.pricing_plans.v2.order_updated",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths2.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.startDate" },
          { path: "entity.endDate" },
          { path: "entity.earliestEndDate" },
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.cancellation.requestedDate" },
          { path: "entity.pausePeriods.pauseDate" },
          { path: "entity.pausePeriods.resumeDate" },
          { path: "entity.currentCycle.startedDate" },
          { path: "entity.currentCycle.endedDate" },
          { path: "entity.cycles.startedDate" },
          { path: "entity.cycles.endedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onOrderCreated = (0, import_sdk_types.EventDefinition)(
  "wix.pricing_plans.v2.order_created",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths2.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.startDate" },
          { path: "entity.endDate" },
          { path: "entity.earliestEndDate" },
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.cancellation.requestedDate" },
          { path: "entity.pausePeriods.pauseDate" },
          { path: "entity.pausePeriods.resumeDate" },
          { path: "entity.currentCycle.startedDate" },
          { path: "entity.currentCycle.endedDate" },
          { path: "entity.cycles.startedDate" },
          { path: "entity.cycles.endedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onOrderStartDateChanged = (0, import_sdk_types.EventDefinition)(
  "wix.pricing_plans.v2.order_start_date_changed",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths2.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "data.order.startDate" },
          { path: "data.order.endDate" },
          { path: "data.order.earliestEndDate" },
          { path: "data.order.createdDate" },
          { path: "data.order.updatedDate" },
          { path: "data.order.cancellation.requestedDate" },
          { path: "data.order.pausePeriods.pauseDate" },
          { path: "data.order.pausePeriods.resumeDate" },
          { path: "data.order.currentCycle.startedDate" },
          { path: "data.order.currentCycle.endedDate" },
          { path: "data.order.cycles.startedDate" },
          { path: "data.order.cycles.endedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();

// src/pricing-plans-v2-order-orders.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var import_event_definition_modules = require("@wix/sdk-runtime/event-definition-modules");
var managementGetOrder4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(managementGetOrder3);
var managementListOrders4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(managementListOrders3);
var postponeEndDate4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(postponeEndDate3);
var cancelOrder4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(cancelOrder3);
var markAsPaid4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(markAsPaid3);
var pauseOrder4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(pauseOrder3);
var resumeOrder4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(resumeOrder3);
var memberGetOrder4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(memberGetOrder3);
var memberListOrders4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(memberListOrders3);
var requestCancellation4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(requestCancellation3);
var createOnlineOrder4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(createOnlineOrder3);
var createOfflineOrder4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(createOfflineOrder3);
var getOnlineOrderPreview4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getOnlineOrderPreview3);
var getOfflineOrderPreview4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getOfflineOrderPreview3);
var getPricePreview4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getPricePreview3);
var onOrderAutoRenewCanceled2 = (0, import_event_definition_modules.createEventModule)(
  onOrderAutoRenewCanceled
);
var onOrderCanceled2 = (0, import_event_definition_modules.createEventModule)(onOrderCanceled);
var onOrderCycleStarted2 = (0, import_event_definition_modules.createEventModule)(
  onOrderCycleStarted
);
var onOrderEndDatePostponed2 = (0, import_event_definition_modules.createEventModule)(
  onOrderEndDatePostponed
);
var onOrderEnded2 = (0, import_event_definition_modules.createEventModule)(onOrderEnded);
var onOrderMarkedAsPaid2 = (0, import_event_definition_modules.createEventModule)(
  onOrderMarkedAsPaid
);
var onOrderPaused2 = (0, import_event_definition_modules.createEventModule)(onOrderPaused);
var onOrderPurchased2 = (0, import_event_definition_modules.createEventModule)(onOrderPurchased);
var onOrderResumed2 = (0, import_event_definition_modules.createEventModule)(onOrderResumed);
var onOrderStarted2 = (0, import_event_definition_modules.createEventModule)(onOrderStarted);
var onOrderUpdated2 = (0, import_event_definition_modules.createEventModule)(onOrderUpdated);
var onOrderCreated2 = (0, import_event_definition_modules.createEventModule)(onOrderCreated);
var onOrderStartDateChanged2 = (0, import_event_definition_modules.createEventModule)(
  onOrderStartDateChanged
);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  CancellationCause,
  CancellationEffectiveAt,
  OrderMethod,
  OrderStatus,
  OrderType,
  PaymentStatus,
  PeriodUnit,
  ReasonNotSuspendable,
  Set,
  SortOrder,
  Status,
  WebhookIdentityType,
  cancelOrder,
  createOfflineOrder,
  createOnlineOrder,
  getOfflineOrderPreview,
  getOnlineOrderPreview,
  getPricePreview,
  managementGetOrder,
  managementListOrders,
  markAsPaid,
  memberGetOrder,
  memberListOrders,
  onOrderAutoRenewCanceled,
  onOrderCanceled,
  onOrderCreated,
  onOrderCycleStarted,
  onOrderEndDatePostponed,
  onOrderEnded,
  onOrderMarkedAsPaid,
  onOrderPaused,
  onOrderPurchased,
  onOrderResumed,
  onOrderStartDateChanged,
  onOrderStarted,
  onOrderUpdated,
  pauseOrder,
  postponeEndDate,
  requestCancellation,
  resumeOrder
});
