// @wix/ricos/dist/es/plugin-ai/ai-content-modal/ai-content-modal.js
import { Box, Button, Heading, SidePanel } from "@wix/design-system";
import { SparklesFilled } from "@wix/wix-ui-icons-common";
import classNames from "classnames";
import React, { useContext, useRef } from "react";
import { EditorContext, PluginsEventsContext, RicosContext } from "./../../ricos-context/index.js";
import { unreachable } from "./../../_shared/uncreachable.js";
import {
  Panel,
  PanelHeader,
  useClickOutside,
  usePanelContainerElement
} from "./../../ui-components/index.js";
import {
  AI_MODAL_ERROR_CANCEL_BUTTON,
  AI_MODAL_ERROR_RETRY_BUTTON,
  AI_MODAL_LOADING_INDICATOR,
  AI_MODAL_RESULTS_CONTENT,
  AI_MODAL_ROOT
} from "../data-hooks.js";
import { ErrorMessage } from "../error-message/error-message.js";
import { AI_MODAL_ID, AI_TYPE } from "../types.js";
import { getPromptBiValue } from "../utils/get-prompt-bi.js";
import { getSelectionAsHTML } from "../utils/get-selection-as-html.js";
import styles from "./ai-content-modal-scss.js";
import { Disclaimer } from "./disclaimer/disclaimer.js";
import { Feed } from "./feed/feed.js";
import { NoSelectionMessage } from "./no-selection-message/no-selection-message.js";
import { PromptsList } from "./prompts-list.js";
import {
  TextualPromptInputArea
} from "./textual-prompt-input-area/textual-prompt-input-area.js";
import { useAiModalApi } from "./use-ai-modal-api.js";
const PROMPT_CHAR_LIMIT = 500;
const AiContentModal = ({
  closeModal,
  openDiscardModal,
  config,
  zIndex,
  modalSessionId,
  initialPromptOptions,
  feedUserEntryFormatter
}) => {
  const { t } = useContext(RicosContext);
  const { adapter } = useContext(EditorContext);
  const inputRef = React.useRef(null);
  const containerEl = usePanelContainerElement();
  const listRef = useRef(null);
  const selection = React.useMemo(() => getSelectionAsHTML(adapter.tiptapEditor), []);
  const resolvedZIndex = typeof zIndex === "number" ? zIndex : 5500;
  const pluginsEvents = useContext(PluginsEventsContext);
  const publishModalAction = React.useMemo(() => {
    let counter = 0;
    return (actionName, value) => {
      counter++;
      pluginsEvents.publishModalAction({
        pluginId: AI_TYPE,
        modalId: AI_MODAL_ID,
        modalSessionId,
        actionName,
        value,
        actionNumber: counter
      });
    };
  }, [pluginsEvents.publishModalAction, modalSessionId]);
  const {
    submitPrompt,
    setFreePromptValue,
    setSelectedPromptIndex,
    triggerFreePrompt,
    cancelPrompt,
    retryPrompt,
    goBack,
    textualPrompt,
    promptsList,
    state,
    getResultActions,
    tryCloseModal,
    isDiscardModalOpen
  } = useAiModalApi({
    selection: selection || "",
    ...config,
    closeModal,
    openDiscardModal,
    modalRef: { current: containerEl },
    inputRef,
    listRef,
    publishModalAction,
    initialPromptOptions
  });
  const closeOnClickOutside = React.useCallback(() => {
    tryCloseModal();
    publishModalAction("close_on_click_outside");
  }, [tryCloseModal]);
  useClickOutside({
    containerRef: { current: containerEl },
    onClickOutside: closeOnClickOutside,
    disable: isDiscardModalOpen
  });
  React.useEffect(() => {
    if (initialPromptOptions?.autoTrigger && state.type === "INITIAL") {
      submitPrompt(initialPromptOptions.prompt, selection);
    }
  }, []);
  return /* @__PURE__ */ React.createElement(
    Panel,
    {
      closeButtonProps: {
        onClick: () => {
          closeModal();
          publishModalAction("close_on_button_click");
        }
      },
      width: 540,
      dataHook: AI_MODAL_ROOT,
      maxHeight: 456
    },
    /* @__PURE__ */ React.createElement(
      PanelHeader,
      {
        title: /* @__PURE__ */ React.createElement(Box, { gap: "SP1" }, /* @__PURE__ */ React.createElement(Box, { color: "B10" }, /* @__PURE__ */ React.createElement(SparklesFilled, null)), /* @__PURE__ */ React.createElement(Heading, { size: "medium" }, t("AIPlugin_Modal_Title")))
      }
    ),
    (() => {
      switch (state.type) {
        case "INITIAL": {
          return selection ? promptsList.length > 0 ? /* @__PURE__ */ React.createElement("div", { className: styles.content }, /* @__PURE__ */ React.createElement(
            PromptsList,
            {
              ref: listRef,
              visible: textualPrompt.length === 0,
              prompts: promptsList,
              highlightedIndex: state.selectedPromptIndex,
              onHighlight: setSelectedPromptIndex,
              onClick: (prompt) => {
                submitPrompt(prompt, selection);
                publishModalAction("prompt_selected", getPromptBiValue(prompt));
              }
            }
          )) : null : /* @__PURE__ */ React.createElement(
            NoSelectionMessage,
            {
              title: initialPromptOptions?.emptyStateTitle,
              text: initialPromptOptions?.emptyStateText
            }
          );
        }
        case "IN_PROGRESS": {
          return /* @__PURE__ */ React.createElement(
            "div",
            {
              className: classNames(styles.content, styles.contentScrollReverse),
              "data-hook": AI_MODAL_LOADING_INDICATOR
            },
            /* @__PURE__ */ React.createElement(
              Feed,
              {
                entries: state.entries,
                zIndex: resolvedZIndex,
                getResultActions,
                modalRef: { current: containerEl },
                feedUserEntryFormatter,
                isLoading: true
              }
            )
          );
        }
        case "RESULTS": {
          return /* @__PURE__ */ React.createElement(
            "div",
            {
              className: classNames(styles.content, styles.contentScrollReverse),
              "data-hook": AI_MODAL_RESULTS_CONTENT
            },
            /* @__PURE__ */ React.createElement(
              Feed,
              {
                entries: state.entries,
                zIndex: resolvedZIndex,
                getResultActions,
                modalRef: { current: containerEl },
                feedUserEntryFormatter
              }
            )
          );
        }
        case "ERROR": {
          return /* @__PURE__ */ React.createElement("div", { className: classNames(styles.content, styles.padded) }, /* @__PURE__ */ React.createElement(ErrorMessage, null));
        }
        default: {
          throw unreachable(state);
        }
      }
    })(),
    /* @__PURE__ */ React.createElement(
      SidePanel.Footer,
      {
        noPadding: true,
        showDivider: !(state.type === "INITIAL" && promptsList.length === 0) && textualPrompt.length === 0 || state.type !== "INITIAL" || !selection
      },
      /* @__PURE__ */ React.createElement(
        Box,
        {
          paddingTop: "SP3",
          paddingBottom: "SP2",
          paddingLeft: "SP4",
          paddingRight: "SP4",
          align: "right",
          className: styles.footer
        },
        (() => {
          switch (state.type) {
            case "ERROR": {
              return /* @__PURE__ */ React.createElement(Box, { gap: "SP2", paddingBottom: "SP1" }, /* @__PURE__ */ React.createElement(
                Button,
                {
                  dataHook: AI_MODAL_ERROR_CANCEL_BUTTON,
                  size: "small",
                  priority: "secondary",
                  onClick: () => {
                    goBack();
                    publishModalAction("back_from_error");
                  }
                },
                t("AIPlugin_Error_Button_GoBack")
              ), /* @__PURE__ */ React.createElement(
                Button,
                {
                  dataHook: AI_MODAL_ERROR_RETRY_BUTTON,
                  size: "small",
                  priority: "primary",
                  onClick: () => {
                    retryPrompt();
                    publishModalAction("retry_from_error");
                  }
                },
                t("AIPlugin_Error_Button_Retry")
              ));
            }
            default: {
              return /* @__PURE__ */ React.createElement(Box, { direction: "vertical", gap: "SP2", width: "100%" }, /* @__PURE__ */ React.createElement(
                TextualPromptInputArea,
                {
                  busy: state.type === "IN_PROGRESS",
                  value: textualPrompt,
                  onChange: setFreePromptValue,
                  handleTextAreaEnter: state.type !== "INITIAL",
                  onSubmit: () => {
                    triggerFreePrompt();
                    publishModalAction("submit_free_prompt", textualPrompt);
                  },
                  onCancel: () => {
                    cancelPrompt();
                    publishModalAction("cancel_free_prompt");
                  },
                  ref: inputRef,
                  maxLength: PROMPT_CHAR_LIMIT,
                  zIndex: resolvedZIndex,
                  placeholder: state.type === "INITIAL" && initialPromptOptions?.placeholder ? initialPromptOptions.placeholder : void 0,
                  isSubmitButtonVisible: initialPromptOptions?.isSubmitButtonVisible
                }
              ), /* @__PURE__ */ React.createElement(Disclaimer, null));
            }
          }
        })()
      )
    )
  );
};
export {
  AiContentModal
};
