// @wix/auto_sdk_payments_dispute-history-records/build/internal/es/meta.mjs
// src/payments-dispute-history-records-v1-dispute-history-record-dispute-history-records.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixPaymentsDisputeHistoryRecordsV1DisputeHistoryRecordServiceUrl(opts) {
  const domainToMappings = {};
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_dispute-history-records";
function getDisputeHistoryRecord(payload) {
  function __getDisputeHistoryRecord({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_history_records.v1.dispute_history_record",
      method: "GET",
      methodFqn: "wix.payments.dispute_history_records.v1.DisputeHistoryRecordService.GetDisputeHistoryRecord",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeHistoryRecordsV1DisputeHistoryRecordServiceUrl(
        {
          protoPath: "/v1/dispute-history-records/{disputeHistoryRecordId}",
          data: payload,
          host
        }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "disputeHistoryRecord.createdDate" },
            { path: "disputeHistoryRecord.updatedDate" },
            { path: "disputeHistoryRecord.dueDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getDisputeHistoryRecord;
}
function queryDisputeHistoryRecords(payload) {
  function __queryDisputeHistoryRecords({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_history_records.v1.dispute_history_record",
      method: "POST",
      methodFqn: "wix.payments.dispute_history_records.v1.DisputeHistoryRecordService.QueryDisputeHistoryRecords",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeHistoryRecordsV1DisputeHistoryRecordServiceUrl(
        { protoPath: "/v1/dispute-history-records/query", data: payload, host }
      ),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "disputeHistoryRecords.createdDate" },
            { path: "disputeHistoryRecords.updatedDate" },
            { path: "disputeHistoryRecords.dueDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixPaymentsDisputeHistoryRecordsV1DisputeHistoryRecordServiceUrl(
            {
              protoPath: "/v1/dispute-history-records/query",
              data: payload,
              host
            }
          ),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryDisputeHistoryRecords;
}
function bulkUpdateDisputeHistoryRecordTags(payload) {
  function __bulkUpdateDisputeHistoryRecordTags({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_history_records.v1.dispute_history_record",
      method: "POST",
      methodFqn: "wix.payments.dispute_history_records.v1.DisputeHistoryRecordService.BulkUpdateDisputeHistoryRecordTags",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeHistoryRecordsV1DisputeHistoryRecordServiceUrl(
        {
          protoPath: "/v1/bulk/dispute-history-records/update-tags",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateDisputeHistoryRecordTags;
}
function bulkUpdateDisputeHistoryRecordTagsByFilter(payload) {
  function __bulkUpdateDisputeHistoryRecordTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.dispute_history_records.v1.dispute_history_record",
      method: "POST",
      methodFqn: "wix.payments.dispute_history_records.v1.DisputeHistoryRecordService.BulkUpdateDisputeHistoryRecordTagsByFilter",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsDisputeHistoryRecordsV1DisputeHistoryRecordServiceUrl(
        {
          protoPath: "/v1/bulk/dispute-history-records/update-tags-by-filter",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateDisputeHistoryRecordTagsByFilter;
}

// src/payments-dispute-history-records-v1-dispute-history-record-dispute-history-records.meta.ts
function getDisputeHistoryRecord2() {
  const payload = { disputeHistoryRecordId: ":disputeHistoryRecordId" };
  const getRequestOptions = getDisputeHistoryRecord(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/dispute-history-records/{disputeHistoryRecordId}",
    pathParams: { disputeHistoryRecordId: "disputeHistoryRecordId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryDisputeHistoryRecords2() {
  const payload = {};
  const getRequestOptions = queryDisputeHistoryRecords(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/dispute-history-records/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateDisputeHistoryRecordTags2() {
  const payload = {};
  const getRequestOptions = bulkUpdateDisputeHistoryRecordTags(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/dispute-history-records/update-tags",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateDisputeHistoryRecordTagsByFilter2() {
  const payload = {};
  const getRequestOptions = bulkUpdateDisputeHistoryRecordTagsByFilter(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/dispute-history-records/update-tags-by-filter",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  bulkUpdateDisputeHistoryRecordTags2 as bulkUpdateDisputeHistoryRecordTags,
  bulkUpdateDisputeHistoryRecordTagsByFilter2 as bulkUpdateDisputeHistoryRecordTagsByFilter,
  getDisputeHistoryRecord2 as getDisputeHistoryRecord,
  queryDisputeHistoryRecords2 as queryDisputeHistoryRecords
};
