// @wix/bex-core/dist/esm/testkit/kanban/createKanbanStagesBackend.js
import { escapeRegExp } from 'lodash';
import { InMemoryBackend } from '../../InMemoryBackend';
import { kanbanStaticStagesData } from './kanbanStaticData';
export function createKanbanStage(chance, { createdAt = chance.date({
    min: new Date(Date.now() - 365 * 24 * 60 * 60 * 1000),
    max: new Date(Date.now() - 2 * 24 * 60 * 60 * 1000),
}), ...rest } = {}) {
    return {
        id: chance.guid(),
        name: chance.name(),
        createdAt,
        updatedAt: createdAt,
        ...rest,
    };
}
export const createKanbanStagesBackend = ({ createOne, ...params } = {}) => {
    return new InMemoryBackend({
        enableTotal: params?.enableTotal,
        createOne: (i) => {
            const raw = kanbanStaticStagesData[i];
            return {
                ...raw,
                createdAt: new Date(raw.createdAt ?? ''),
                updatedAt: new Date(raw.updatedAt ?? ''),
            };
        },
        itemKey: (item) => item.id,
        total: kanbanStaticStagesData.length,
        predicate: (query) => {
            const { search } = query;
            // const filters = query.filters ?? {};
            const rgx = new RegExp(escapeRegExp(search), 'i');
            return (item) => {
                if (search && !rgx.test(item.name)) {
                    return false;
                }
                return true;
            };
        },
        orderBy: (query) => {
            return [...(query.sort ?? [])];
        },
        ...params,
    });
};
