// @wix/auto_sdk_benefit-programs_pools/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  bulkCheckBenefitEligibility: () => bulkCheckBenefitEligibility2,
  bulkUpdatePools: () => bulkUpdatePools2,
  checkBenefitEligibility: () => checkBenefitEligibility2,
  getEligibleBenefits: () => getEligibleBenefits2,
  getPool: () => getPool2,
  queryPools: () => queryPools2,
  redeemBenefit: () => redeemBenefit2,
  updatePool: () => updatePool2
});
module.exports = __toCommonJS(meta_exports);

// src/benefit-programs-v1-pool-pools.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixBenefitProgramsV1PoolPoolServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/pools",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/benefit-programs-pools",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/benefits",
        destPath: "/v1/bulk/benefits"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/pools",
        destPath: "/v1/bulk/pools"
      },
      {
        srcPath: "/_api/benefit-programs/v1/pools",
        destPath: "/v1/pools"
      },
      {
        srcPath: "/_api/benefit-programs/v1/benefits",
        destPath: "/v1/benefits"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pools",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/benefits",
        destPath: "/v1/bulk/benefits"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/pools",
        destPath: "/v1/bulk/pools"
      },
      {
        srcPath: "/_api/benefit-programs/v1/pools",
        destPath: "/v1/pools"
      },
      {
        srcPath: "/_api/benefit-programs/v1/benefits",
        destPath: "/v1/benefits"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pools",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/benefits",
        destPath: "/v1/bulk/benefits"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/pools",
        destPath: "/v1/bulk/pools"
      },
      {
        srcPath: "/_api/benefit-programs/v1/pools",
        destPath: "/v1/pools"
      },
      {
        srcPath: "/_api/benefit-programs/v1/benefits",
        destPath: "/v1/benefits"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/benefit-programs-pools",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/benefits",
        destPath: "/v1/bulk/benefits"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/pools",
        destPath: "/v1/bulk/pools"
      },
      {
        srcPath: "/_api/benefit-programs/v1/pools",
        destPath: "/v1/pools"
      },
      {
        srcPath: "/_api/benefit-programs/v1/benefits",
        destPath: "/v1/benefits"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/benefit-programs-pools",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/benefits",
        destPath: "/v1/bulk/benefits"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/pools",
        destPath: "/v1/bulk/pools"
      },
      {
        srcPath: "/_api/benefit-programs/v1/pools",
        destPath: "/v1/pools"
      },
      {
        srcPath: "/_api/benefit-programs/v1/benefits",
        destPath: "/v1/benefits"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pools",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/benefits",
        destPath: "/v1/bulk/benefits"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/pools",
        destPath: "/v1/bulk/pools"
      },
      {
        srcPath: "/_api/benefit-programs/v1/pools",
        destPath: "/v1/pools"
      },
      {
        srcPath: "/_api/benefit-programs/v1/benefits",
        destPath: "/v1/benefits"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-pools",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/benefits",
        destPath: "/v1/benefits"
      },
      {
        srcPath: "/_api/benefit-programs/v1/pools",
        destPath: "/v1/pools"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/benefits",
        destPath: "/v1/bulk/benefits"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/pools",
        destPath: "/v1/bulk/pools"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/benefit-programs/v1/bulk/benefits",
        destPath: "/v1/bulk/benefits"
      },
      {
        srcPath: "/_api/benefit-programs/v1/bulk/pools",
        destPath: "/v1/bulk/pools"
      },
      {
        srcPath: "/_api/benefit-programs/v1/pools",
        destPath: "/v1/pools"
      },
      {
        srcPath: "/_api/benefit-programs/v1/benefits",
        destPath: "/v1/benefits"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/benefit-programs/v1/bulk/benefits",
        destPath: "/v1/bulk/benefits"
      },
      {
        srcPath: "/benefit-programs/v1/bulk/pools",
        destPath: "/v1/bulk/pools"
      },
      {
        srcPath: "/benefit-programs/v1/pools",
        destPath: "/v1/pools"
      },
      {
        srcPath: "/benefit-programs/v1/benefits",
        destPath: "/v1/benefits"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_benefit-programs_pools";
function getPool(payload) {
  function __getPool({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.pool.PoolService.GetPool",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolPoolServiceUrl({
        protoPath: "/v1/pools/{poolId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "pool.createdDate" }, { path: "pool.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __getPool;
}
function updatePool(payload) {
  function __updatePool({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "pool.createdDate" }, { path: "pool.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool",
      method: "PATCH",
      methodFqn: "wix.benefit_programs.v1.pool.PoolService.UpdatePool",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolPoolServiceUrl({
        protoPath: "/v1/pools/{pool.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "pool.createdDate" }, { path: "pool.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __updatePool;
}
function queryPools(payload) {
  function __queryPools({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.pool.PoolService.QueryPools",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolPoolServiceUrl({
        protoPath: "/v1/pools/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "pools.createdDate" },
            { path: "pools.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryPools;
}
function redeemBenefit(payload) {
  function __redeemBenefit({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "targetDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.pool.PoolService.RedeemBenefit",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolPoolServiceUrl({
        protoPath: "/v1/benefits/redeem",
        data: serializedData,
        host
      }),
      data: serializedData
    };
    return metadata;
  }
  return __redeemBenefit;
}
function checkBenefitEligibility(payload) {
  function __checkBenefitEligibility({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "targetDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.pool.PoolService.CheckBenefitEligibility",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolPoolServiceUrl({
        protoPath: "/v1/pools/check-eligibility",
        data: serializedData,
        host
      }),
      data: serializedData
    };
    return metadata;
  }
  return __checkBenefitEligibility;
}
function bulkCheckBenefitEligibility(payload) {
  function __bulkCheckBenefitEligibility({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "benefitSelectors.targetDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.pool.PoolService.BulkCheckBenefitEligibility",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolPoolServiceUrl({
        protoPath: "/v1/bulk/benefits/check-eligibility",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "results.benefitSelector.targetDate" }]
        }
      ])
    };
    return metadata;
  }
  return __bulkCheckBenefitEligibility;
}
function getEligibleBenefits(payload) {
  function __getEligibleBenefits({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "targetDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.pool.PoolService.GetEligibleBenefits",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolPoolServiceUrl({
        protoPath: "/v1/pools/eligible-pools",
        data: serializedData,
        host
      }),
      data: serializedData
    };
    return metadata;
  }
  return __getEligibleBenefits;
}
function bulkUpdatePools(payload) {
  function __bulkUpdatePools({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "pools.fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "pools.pool.createdDate" },
          { path: "pools.pool.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.pool",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.pool.PoolService.BulkUpdatePools",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1PoolPoolServiceUrl({
        protoPath: "/v1/bulk/pools",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.pool.createdDate" },
            { path: "results.pool.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdatePools;
}

// src/benefit-programs-v1-pool-pools.meta.ts
function getPool2() {
  const payload = { poolId: ":poolId" };
  const getRequestOptions = getPool(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/pools/{poolId}",
    pathParams: { poolId: "poolId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updatePool2() {
  const payload = { pool: { id: ":poolId" } };
  const getRequestOptions = updatePool(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/pools/{pool.id}",
    pathParams: { poolId: "poolId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryPools2() {
  const payload = {};
  const getRequestOptions = queryPools(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/pools/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function redeemBenefit2() {
  const payload = {};
  const getRequestOptions = redeemBenefit(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/benefits/redeem",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function checkBenefitEligibility2() {
  const payload = {};
  const getRequestOptions = checkBenefitEligibility(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/pools/check-eligibility",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkCheckBenefitEligibility2() {
  const payload = {};
  const getRequestOptions = bulkCheckBenefitEligibility(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/benefits/check-eligibility",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getEligibleBenefits2() {
  const payload = {};
  const getRequestOptions = getEligibleBenefits(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/pools/eligible-pools",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdatePools2() {
  const payload = {};
  const getRequestOptions = bulkUpdatePools(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/pools",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  bulkCheckBenefitEligibility,
  bulkUpdatePools,
  checkBenefitEligibility,
  getEligibleBenefits,
  getPool,
  queryPools,
  redeemBenefit,
  updatePool
});
