// @wix/auto-patterns/dist/cjs/components/filters/DynamicCollectionFilter.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.DynamicCollectionFilter = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireDefault(require("react"));
var _patterns = require("@wix/patterns");
var _providers = require("../../providers");
var _fetchCmsData = require("../../dataSourceAdapters/cms/fetchCmsData");
var _jsxFileName = "/home/builduser/work/73d19ce3378ce4dd/packages/auto-patterns/dist/cjs/components/filters/DynamicCollectionFilter.tsx";
const DynamicCollectionFilter = ({
  filter,
  collectionId,
  component: FilterComponent,
  ...filterProps
}) => {
  const {
    getSchema
  } = (0, _providers.useSchemaRegistry)();
  const referencedSchema = getSchema(collectionId);
  const fetchData = _react.default.useCallback(async query => {
    if (!referencedSchema) {
      return {
        items: [],
        total: 0
      };
    }
    const result = await (0, _fetchCmsData.fetchCmsData)(collectionId, query, referencedSchema.fields, {
      searchableFieldIds: [referencedSchema.displayField],
      filterFieldMapping: {}
    });
    return {
      items: result.items.map(item => ({
        id: item[referencedSchema.idField],
        name: item[referencedSchema.displayField]
      }))
    };
  }, [collectionId, referencedSchema]);
  const collection = (0, _patterns.useFilterCollection)(filter, {
    queryName: `filter-collection-${collectionId}`,
    fetchData
  });
  return /*#__PURE__*/_react.default.createElement(FilterComponent, (0, _extends2.default)({
    filter: filter,
    collection: collection
  }, filterProps, {
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 52,
      columnNumber: 5
    }
  }));
};
exports.DynamicCollectionFilter = DynamicCollectionFilter;
