// @wix/auto-patterns/dist/cjs/components/filters/StaticCollectionFilter.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.StaticCollectionFilter = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireDefault(require("react"));
var _patterns = require("@wix/patterns");
var _jsxFileName = "/home/builduser/work/73d19ce3378ce4dd/packages/auto-patterns/dist/cjs/components/filters/StaticCollectionFilter.tsx";
const StaticCollectionFilter = ({
  filter,
  collectionData,
  component,
  ...filterProps
}) => {
  const FilterComponent = component;
  const collection = (0, _patterns.useStaticListFilterCollection)(filter, collectionData ?? []);
  return /*#__PURE__*/_react.default.createElement(FilterComponent, (0, _extends2.default)({
    filter: filter,
    collection: collection
  }, filterProps, {
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 25,
      columnNumber: 5
    }
  }));
};
exports.StaticCollectionFilter = StaticCollectionFilter;
