// @wix/wix-ui-test-utils/dist/cjs/driver-factory/createDriverFactory.js
"use strict";

exports.__esModule = true;
exports.createDriverFactory = createDriverFactory;
var React = _interopRequireWildcard(require("react"));
var _client = require("react-dom/client");
var _reactHelpers = require("../react-helpers");
var _jsxFileName = "/home/builduser/work/c98d836f8565ce5f/packages/wix-ui-test-utils/dist/cjs/driver-factory/createDriverFactory.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function componentFactory(Component) {
  let element;
  const eventTrigger = (0, _reactHelpers.reactEventTrigger)();
  const wrapperDiv = document.createElement('div');
  const root = (0, _client.createRoot)(wrapperDiv);
  root.render(/*#__PURE__*/React.createElement("div", {
    ref: r => element = r,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 29,
      columnNumber: 15
    }
  }, Component));
  return {
    element: element && element.childNodes[0],
    wrapper: wrapperDiv,
    component: Component,
    eventTrigger
  };
}
function createDriverFactory(driverFactory) {
  return Component => driverFactory(componentFactory(Component));
}
