// @wix/design-system/dist/cjs/BaseModalLayout/BaseModalLayout.st.css.js

require("../Foundation/stylable/colors.st.css.js");
require("../Foundation/stylable/spacing.st.css.js");
require("../Foundation/stylable/border.st.css.js");
require("../Foundation/stylable/shadows.st.css.js");
require("./LayoutBlocks/Header/Header.st.css.js");
require("./LayoutBlocks/Content/Content.st.css.js");
require("./LayoutBlocks/Footer/Footer.st.css.js");
require("./LayoutBlocks/Footnote/Footnote.st.css.js");
require("./LayoutBlocks/Illustration/Illustration.st.css.js");
require("../CloseButton/CloseButton.st.css.js");

const { classesRuntime, statesRuntime } = require("./../stylable-cjs-runtime.js");



var _namespace_ = "wds_1_226_0_BaseModalLayout";
var _style_ = classesRuntime.bind(null, _namespace_);

module.exports.cssStates = statesRuntime.bind(null, _namespace_);
module.exports.style = _style_;
module.exports.st = _style_;

module.exports.namespace = _namespace_;
module.exports.classes = {"root":"wds_1_226_0_BaseModalLayout__root","controlButtons":"wds_1_226_0_BaseModalLayout__controlButtons","helpButton":"wds_1_226_0_BaseModalLayout__helpButton","closeButton":"wds_1_226_0_BaseModalLayout__closeButton","helpIcon":"wds_1_226_0_BaseModalLayout__helpIcon","header":"wds_1_226_0_BaseModalLayout__header","content":"wds_1_226_0_BaseModalLayout__content","footer":"wds_1_226_0_BaseModalLayout__footer","footnote":"wds_1_226_0_BaseModalLayout__footnote","illustration":"wds_1_226_0_BaseModalLayout__illustration"};
module.exports.keyframes = {}; 
module.exports.layers = {};
module.exports.containers = {};
module.exports.stVars = {"helpIconSize":"24px"}; 
module.exports.vars = {"wds-space-50":"--wds-space-50","wds-color-fill-surface-modal":"--wds-color-fill-surface-modal","wds-border-radius-surface-modal":"--wds-border-radius-surface-modal","wds-shadow-surface-modal":"--wds-shadow-surface-modal","wds-custom-modal-header-padding-vertical":"--wds-custom-modal-header-padding-vertical","wds-custom-modal-header-padding-right":"--wds-custom-modal-header-padding-right"};
