// @wix/ricos/dist/cjs/plugin-button/button-converters.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var button_converters_exports = {};
__export(button_converters_exports, {
  actionButtonConverter: () => actionButtonConverter,
  linkButtonConverter: () => linkButtonConverter
});
module.exports = __toCommonJS(button_converters_exports);
var import_ricos_content = require("./../ricos-content/index.js");
var import_v1 = require("./../ricos-schema/legacy/wix/rich_content/v1/index.js");
var import_link_converters = require("./../_shared/link/link-converters.js");
const toTiptapButtonConvert = (node) => ({
  type: node.buttonData.type === "LINK" ? import_ricos_content.TIPTAP_LINK_BUTTON_TYPE : import_ricos_content.TIPTAP_ACTION_BUTTON_TYPE,
  attrs: {
    ...node.buttonData,
    styles: {
      ...node.buttonData.styles,
      border: void 0,
      colors: void 0,
      backgroundColor: node.buttonData.styles?.backgroundColor ?? node.buttonData.styles?.colors?.background,
      textColor: node.buttonData.styles?.textColor ?? node.buttonData.styles?.colors?.text,
      borderColor: node.buttonData.styles?.borderColor ?? node.buttonData.styles?.colors?.border,
      borderWidth: node.buttonData.styles?.borderWidth ?? node.buttonData.styles?.border?.width,
      borderRadius: node.buttonData.styles?.borderRadius ?? node.buttonData.styles?.border?.radius
    },
    id: node.id
  }
});
const fromTiptapButtonConvert = (node) => {
  const { id, ...data } = node.attrs || {};
  const linkData = data.link;
  const containerData = data.containerData;
  return {
    type: import_v1.Node_Type.BUTTON,
    id,
    nodes: [],
    buttonData: {
      ...data,
      link: (0, import_link_converters.fromTiptapLinkConverter)(linkData),
      containerData: {
        ...containerData,
        alignment: containerData?.alignment || import_v1.PluginContainerData_Alignment.CENTER,
        width: containerData?.width ? {
          ...containerData?.width,
          size: containerData?.width?.custom ? void 0 : containerData?.width?.size
        } : void 0
      },
      text: data.text || ""
    }
  };
};
const linkButtonConverter = {
  toPm: {
    types: [import_v1.Node_Type.BUTTON],
    // @ts-ignore $TSFixMe
    convert: toTiptapButtonConvert
  },
  fromPm: {
    types: [import_ricos_content.TIPTAP_LINK_BUTTON_TYPE],
    convert: fromTiptapButtonConvert
  }
};
const actionButtonConverter = {
  toPm: {
    types: [import_v1.Node_Type.BUTTON],
    // @ts-ignore $TSFixMe
    convert: toTiptapButtonConvert
  },
  fromPm: {
    types: [import_ricos_content.TIPTAP_ACTION_BUTTON_TYPE],
    convert: fromTiptapButtonConvert
  }
};
