// @wix/bex-utils/@wix/ambassador-os-tags-v1-tag/build/cjs/builders.impl.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.aMigrateTagResponse = exports.aMigrateTagRequest = exports.aListTagsResponse = exports.aListTagsRequest = exports.aDeleteTagResponse = exports.aDeleteTagRequest = exports.aUpdateTagResponse = exports.aUpdateTagRequest = exports.aGetTagResponse = exports.aGetTagRequest = exports.aCreateTagResponse = exports.aCreateTagRequest = exports.aWebhookIdentityType = exports.aIdentificationData = exports.aMessageEnvelope = exports.aActionEvent = exports.aEntityDeletedEvent = exports.aEntityUpdatedEvent = exports.aRestoreInfo = exports.aEntityCreatedEvent = exports.aDomainEvent = exports.aListTagsForAccountResponse = exports.aListTagsForAccountRequest = exports.aDeleteTagForAccountResponse = exports.aDeleteTagForAccountRequest = exports.aUpdateTagForAccountResponse = exports.aUpdateTagForAccountRequest = exports.aGetTagForAccountResponse = exports.aGetTagForAccountRequest = exports.aCreateTagForAccountResponse = exports.aCreateTagForAccountRequest = exports.aTagExposure = exports.aTag = void 0;
var metro_common_builders_1 = require("@wix/metro-common-builders");
var types_impl_1 = require("./types.impl");
// @ts-ignore
var callsite_1 = __importDefault(require("callsite"));
function aTag(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        id: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'MAX_LENGTH', value: '5' },
            ],
        }),
        revision: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'READ_ONLY', value: 'true' },
                { type: 'DECIMAL_VALUE' },
            ],
        }),
        createdDate: (0, metro_common_builders_1.someDate)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        updatedDate: (0, metro_common_builders_1.someDate)({
            validations: [{ type: 'READ_ONLY', value: 'true' }],
        }),
        name: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '255' }],
        }),
        fqdn: (0, metro_common_builders_1.someString)({
            validations: [
                { type: 'IMMUTABLE', value: 'true' },
                { type: 'MAX_LENGTH', value: '255' },
            ],
        }),
        exposure: aTagExposure(),
        deleted: true,
    }, overrideObject);
}
exports.aTag = aTag;
function aTagExposure(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.TagExposure.UNKNOWN; }
    return enumFirstItem;
}
exports.aTagExposure = aTagExposure;
function aCreateTagForAccountRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        tag: aTag(),
    }, overrideObject);
}
exports.aCreateTagForAccountRequest = aCreateTagForAccountRequest;
function aCreateTagForAccountResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        tag: aTag(),
    }, overrideObject);
}
exports.aCreateTagForAccountResponse = aCreateTagForAccountResponse;
function aGetTagForAccountRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        tagId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '5' }],
        }),
    }, overrideObject);
}
exports.aGetTagForAccountRequest = aGetTagForAccountRequest;
function aGetTagForAccountResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        tag: aTag(),
    }, overrideObject);
}
exports.aGetTagForAccountResponse = aGetTagForAccountResponse;
function aUpdateTagForAccountRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        tag: aTag(),
    }, overrideObject);
}
exports.aUpdateTagForAccountRequest = aUpdateTagForAccountRequest;
function aUpdateTagForAccountResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        tag: aTag(),
    }, overrideObject);
}
exports.aUpdateTagForAccountResponse = aUpdateTagForAccountResponse;
function aDeleteTagForAccountRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        tagId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '5' }],
        }),
    }, overrideObject);
}
exports.aDeleteTagForAccountRequest = aDeleteTagForAccountRequest;
function aDeleteTagForAccountResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({}, overrideObject);
}
exports.aDeleteTagForAccountResponse = aDeleteTagForAccountResponse;
function aListTagsForAccountRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        fqdn: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '255' }],
        }),
        exposure: aTagExposure(),
    }, overrideObject);
}
exports.aListTagsForAccountRequest = aListTagsForAccountRequest;
function aListTagsForAccountResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        tags: [aTag()],
    }, overrideObject);
}
exports.aListTagsForAccountResponse = aListTagsForAccountResponse;
function aDomainEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        createdEvent: aEntityCreatedEvent(),
        id: (0, metro_common_builders_1.someString)(),
        entityFqdn: (0, metro_common_builders_1.someString)(),
        slug: (0, metro_common_builders_1.someString)(),
        entityId: (0, metro_common_builders_1.someString)(),
        eventTime: (0, metro_common_builders_1.someDate)(),
        triggeredByAnonymizeRequest: true,
        originatedFrom: (0, metro_common_builders_1.someString)(),
        entityEventSequence: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'DECIMAL_VALUE' }],
        }),
    }, overrideObject);
}
exports.aDomainEvent = aDomainEvent;
function aEntityCreatedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        entityAsJson: (0, metro_common_builders_1.someString)(),
        triggeredByUndelete: true,
        restoreInfo: aRestoreInfo(),
        additionalMetadataAsJson: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aEntityCreatedEvent = aEntityCreatedEvent;
function aRestoreInfo(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        deletedDate: (0, metro_common_builders_1.someDate)(),
    }, overrideObject);
}
exports.aRestoreInfo = aRestoreInfo;
function aEntityUpdatedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        currentEntityAsJson: (0, metro_common_builders_1.someString)(),
        previousEntityAsJson: (0, metro_common_builders_1.someString)(),
        modifiedFields: { modifiedFields: (0, metro_common_builders_1.someAny)() },
        additionalMetadataAsJson: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aEntityUpdatedEvent = aEntityUpdatedEvent;
function aEntityDeletedEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        movedToTrash: true,
        deletedEntityAsJson: (0, metro_common_builders_1.someString)(),
        additionalMetadataAsJson: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aEntityDeletedEvent = aEntityDeletedEvent;
function aActionEvent(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        bodyAsJson: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aActionEvent = aActionEvent;
function aMessageEnvelope(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        instanceId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        eventType: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '150' }],
        }),
        identity: aIdentificationData(),
        data: (0, metro_common_builders_1.someString)(),
    }, overrideObject);
}
exports.aMessageEnvelope = aMessageEnvelope;
function aIdentificationData(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        anonymousVisitorId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'FORMAT', value: 'GUID' }],
        }),
        identityType: aWebhookIdentityType(),
    }, overrideObject);
}
exports.aIdentificationData = aIdentificationData;
function aWebhookIdentityType(enumFirstItem) {
    if (enumFirstItem === void 0) { enumFirstItem = types_impl_1.WebhookIdentityType.UNKNOWN; }
    return enumFirstItem;
}
exports.aWebhookIdentityType = aWebhookIdentityType;
function aCreateTagRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        tag: aTag(),
    }, overrideObject);
}
exports.aCreateTagRequest = aCreateTagRequest;
function aCreateTagResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        tag: aTag(),
    }, overrideObject);
}
exports.aCreateTagResponse = aCreateTagResponse;
function aGetTagRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        tagId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '5' }],
        }),
    }, overrideObject);
}
exports.aGetTagRequest = aGetTagRequest;
function aGetTagResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        tag: aTag(),
    }, overrideObject);
}
exports.aGetTagResponse = aGetTagResponse;
function aUpdateTagRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        tag: aTag(),
    }, overrideObject);
}
exports.aUpdateTagRequest = aUpdateTagRequest;
function aUpdateTagResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        tag: aTag(),
    }, overrideObject);
}
exports.aUpdateTagResponse = aUpdateTagResponse;
function aDeleteTagRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({
        tagId: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '5' }],
        }),
    }, overrideObject);
}
exports.aDeleteTagRequest = aDeleteTagRequest;
function aDeleteTagResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    return (0, metro_common_builders_1.mergeDeep)({}, overrideObject);
}
exports.aDeleteTagResponse = aDeleteTagResponse;
function aListTagsRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        fqdn: (0, metro_common_builders_1.someString)({
            validations: [{ type: 'MAX_LENGTH', value: '255' }],
        }),
        exposure: aTagExposure(),
    }, overrideObject);
}
exports.aListTagsRequest = aListTagsRequest;
function aListTagsResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        tags: [aTag()],
    }, overrideObject);
}
exports.aListTagsResponse = aListTagsResponse;
function aMigrateTagRequest(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        tag: aTag(),
    }, overrideObject);
}
exports.aMigrateTagRequest = aMigrateTagRequest;
function aMigrateTagResponse(overrideObject) {
    if (overrideObject === void 0) { overrideObject = {}; }
    if (isCurrentBuilderWasPreviouslyCalled()) {
        return {};
    }
    return (0, metro_common_builders_1.mergeDeep)({
        tag: aTag(),
    }, overrideObject);
}
exports.aMigrateTagResponse = aMigrateTagResponse;
function isCurrentBuilderWasPreviouslyCalled() {
    var callChainFunctionNames = (0, callsite_1.default)().map(function (item) {
        return item.getFunctionName();
    });
    var callChainWithoutCurrentBuilder = callChainFunctionNames.splice(2);
    var currentBuilderName = callChainFunctionNames[1];
    return callChainWithoutCurrentBuilder.some(function (functionName) { return functionName === currentBuilderName; });
}
__exportStar(require("@wix/metro-common-builders"), exports);
