// @wix/bex-utils/@wix/ambassador-os-tags-v1-tag/build/cjs/meta.impl.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listTags = exports.deleteTag = exports.updateTag = exports.getTag = exports.createTag = exports.listTagsForAccount = exports.deleteTagForAccount = exports.updateTagForAccount = exports.getTagForAccount = exports.createTagForAccount = void 0;
var httpWixOsTagsV1Tag = __importStar(require("./http.impl"));
function createTagForAccount() {
    var payload = {};
    var getRequestOptions = httpWixOsTagsV1Tag.createTagForAccount(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/account/tags',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createTagForAccount = createTagForAccount;
function getTagForAccount() {
    var payload = { tagId: ':tagId' };
    var getRequestOptions = httpWixOsTagsV1Tag.getTagForAccount(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/v1/account/tags/{tagId}',
        pathParams: { tagId: 'tagId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getTagForAccount = getTagForAccount;
function updateTagForAccount() {
    var payload = { tag: { id: ':tagId' } };
    var getRequestOptions = httpWixOsTagsV1Tag.updateTagForAccount(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'PATCH',
        path: '/v1/account/tags/{tag.id}',
        pathParams: { tagId: 'tagId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateTagForAccount = updateTagForAccount;
function deleteTagForAccount() {
    var payload = { tagId: ':tagId' };
    var getRequestOptions = httpWixOsTagsV1Tag.deleteTagForAccount(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'DELETE',
        path: '/v1/account/tags/{tagId}',
        pathParams: { tagId: 'tagId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteTagForAccount = deleteTagForAccount;
function listTagsForAccount() {
    var payload = {};
    var getRequestOptions = httpWixOsTagsV1Tag.listTagsForAccount(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/v1/account/tags',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listTagsForAccount = listTagsForAccount;
function createTag() {
    var payload = {};
    var getRequestOptions = httpWixOsTagsV1Tag.createTag(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'POST',
        path: '/v1/tags',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createTag = createTag;
function getTag() {
    var payload = { tagId: ':tagId' };
    var getRequestOptions = httpWixOsTagsV1Tag.getTag(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/v1/tags/{tagId}',
        pathParams: { tagId: 'tagId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getTag = getTag;
function updateTag() {
    var payload = { tag: { id: ':tagId' } };
    var getRequestOptions = httpWixOsTagsV1Tag.updateTag(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'PATCH',
        path: '/v1/tags/{tag.id}',
        pathParams: { tagId: 'tagId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateTag = updateTag;
function deleteTag() {
    var payload = { tagId: ':tagId' };
    var getRequestOptions = httpWixOsTagsV1Tag.deleteTag(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'DELETE',
        path: '/v1/tags/{tagId}',
        pathParams: { tagId: 'tagId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteTag = deleteTag;
function listTags() {
    var payload = {};
    var getRequestOptions = httpWixOsTagsV1Tag.listTags(payload);
    var getUrl = function (context) {
        var url = getRequestOptions(context).url;
        return url;
    };
    return {
        getUrl: getUrl,
        httpMethod: 'GET',
        path: '/v1/tags',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listTags = listTags;
