// @wix/auto_sdk_events_tickets/build/internal/es/index.typings.mjs
// src/events-v1-ticket-tickets.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/events-v1-ticket-tickets.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKFloatToRESTFloat } from "@wix/sdk-runtime/transformations/float";
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixEventsTicketingTicketManagementUrl(opts) {
  const domainToMappings = {
    "events._base_domain_": [
      {
        srcPath: "",
        destPath: ""
      },
      {
        srcPath: "/_api/wix-events-web",
        destPath: "/api"
      }
    ],
    "www.wixevents.com": [
      {
        srcPath: "/_api/wix-events-web",
        destPath: "/api"
      },
      {
        srcPath: "/api",
        destPath: "/api"
      },
      {
        srcPath: "/viewEvent",
        destPath: "/viewEvent"
      },
      {
        srcPath: "/viewRsvpDashboard",
        destPath: "/viewRsvpDashboard"
      },
      {
        srcPath: "/viewOrderDashboard",
        destPath: "/viewOrderDashboard"
      },
      {
        srcPath: "/viewSummarySubscription",
        destPath: "/viewSummarySubscription"
      },
      {
        srcPath: "/check-in",
        destPath: "/check-in"
      },
      {
        srcPath: "/v2",
        destPath: "/v2"
      },
      {
        srcPath: "/export",
        destPath: "/export"
      },
      {
        srcPath: "/tickets",
        destPath: "/tickets"
      },
      {
        srcPath: "/payment-redirect",
        destPath: "/payment-redirect"
      },
      {
        srcPath: "/admin",
        destPath: "/admin"
      },
      {
        srcPath: "/ticket",
        destPath: "/ticket"
      },
      {
        srcPath: "/media",
        destPath: "/media"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/wix-events-web",
        destPath: ""
      }
    ],
    "events.wixapps.net": [
      {
        srcPath: "/_api/wix-events-web",
        destPath: "/api"
      },
      {
        srcPath: "/events",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/wix-events-web",
        destPath: "/api"
      },
      {
        srcPath: "/_api/wix-events-web/v2",
        destPath: "/v2"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/wix-events-web",
        destPath: "/api"
      },
      {
        srcPath: "/events",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/wix-events-web",
        destPath: "/api"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v2/events",
        destPath: "/v2/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/events",
        destPath: "/v1/bulk/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      },
      {
        srcPath: "/events/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/events/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/events/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/events/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/events/v1/bulk/events",
        destPath: "/v1/bulk/events"
      },
      {
        srcPath: "/events/v1/reports",
        destPath: "/v1/reports"
      },
      {
        srcPath: "/events/v1/tickets",
        destPath: "/v1/tickets"
      },
      {
        srcPath: "/events/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/events/v1/web",
        destPath: "/v1/web"
      },
      {
        srcPath: "/events/v2/tickets",
        destPath: "/v2/tickets"
      },
      {
        srcPath: "/events/v2/events",
        destPath: "/v2/events"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/events/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/events/v1/tickets",
        destPath: "/v1/tickets"
      },
      {
        srcPath: "/events/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/events/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/events/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/events/v2/events",
        destPath: "/v2/events"
      },
      {
        srcPath: "/events/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      }
    ],
    "api._base_domain_": [
      {
        srcPath: "/events/v2",
        destPath: "/v2"
      },
      {
        srcPath: "/events/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/events/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/events/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/events/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/events/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/events/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    _: [
      {
        srcPath: "/_api/wix-events-web/v2",
        destPath: "/v2"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/events/v2/events",
        destPath: "/v2/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/wix-events-web/v2",
        destPath: "/v2"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "editor-flow.wixapps.net": [
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/wix-events-web/v1/bulk-async/categories",
        destPath: "/v1/bulk-async/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/bulk/categories",
        destPath: "/v1/bulk/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/wix-events-web/v1/events",
        destPath: "/v1/events"
      },
      {
        srcPath: "/_api/wix-events-web/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets",
        destPath: "/v1/tickets"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_events_tickets";
function listTickets(payload) {
  function __listTickets({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.ticket",
      method: "GET",
      methodFqn: "wix.events.ticketing.TicketManagement.ListTickets",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingTicketManagementUrl({
        protoPath: "/v1/events/{eventId}/tickets",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "tickets.checkIn.created" },
            { path: "tickets.guestDetails.form.inputValues.dateTime" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "tickets.guestDetails.form.inputValues.number" },
            {
              path: "tickets.guestDetails.form.inputValues.address.address.geocode.latitude"
            },
            {
              path: "tickets.guestDetails.form.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ]),
      fallback: [
        {
          method: "GET",
          url: resolveWixEventsTicketingTicketManagementUrl({
            protoPath: "/v1/events/{eventId}/tickets",
            data: payload,
            host
          }),
          params: toURLSearchParams(payload)
        }
      ]
    };
    return metadata;
  }
  return __listTickets;
}
function getTicket(payload) {
  function __getTicket({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.ticket",
      method: "GET",
      methodFqn: "wix.events.ticketing.TicketManagement.GetTicket",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingTicketManagementUrl({
        protoPath: "/v1/events/{eventId}/tickets/{ticketNumber}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "ticket.checkIn.created" },
            { path: "ticket.guestDetails.form.inputValues.dateTime" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "ticket.guestDetails.form.inputValues.number" },
            {
              path: "ticket.guestDetails.form.inputValues.address.address.geocode.latitude"
            },
            {
              path: "ticket.guestDetails.form.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getTicket;
}
function checkInTicket(payload) {
  function __checkInTicket({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.ticket",
      method: "POST",
      methodFqn: "wix.events.ticketing.TicketManagement.CheckInTicket",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingTicketManagementUrl({
        protoPath: "/v1/tickets/check-in",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "tickets.checkIn.created" },
            { path: "tickets.guestDetails.form.inputValues.dateTime" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "tickets.guestDetails.form.inputValues.number" },
            {
              path: "tickets.guestDetails.form.inputValues.address.address.geocode.latitude"
            },
            {
              path: "tickets.guestDetails.form.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __checkInTicket;
}
function deleteTicketCheckIn(payload) {
  function __deleteTicketCheckIn({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.ticket",
      method: "DELETE",
      methodFqn: "wix.events.ticketing.TicketManagement.DeleteTicketCheckIn",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingTicketManagementUrl({
        protoPath: "/v1/tickets/check-in",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "tickets.checkIn.created" },
            { path: "tickets.guestDetails.form.inputValues.dateTime" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "tickets.guestDetails.form.inputValues.number" },
            {
              path: "tickets.guestDetails.form.inputValues.address.address.geocode.latitude"
            },
            {
              path: "tickets.guestDetails.form.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __deleteTicketCheckIn;
}
function updateTicket(payload) {
  function __updateTicket({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fields" }]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "checkoutForm.inputValues.number" },
          { path: "checkoutForm.inputValues.address.address.geocode.latitude" },
          {
            path: "checkoutForm.inputValues.address.address.geocode.longitude"
          },
          { path: "guestDetails.form.inputValues.number" },
          {
            path: "guestDetails.form.inputValues.address.address.geocode.latitude"
          },
          {
            path: "guestDetails.form.inputValues.address.address.geocode.longitude"
          }
        ]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "checkoutForm.inputValues.dateTime" },
          { path: "guestDetails.form.inputValues.dateTime" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v1.ticket",
      method: "PATCH",
      methodFqn: "wix.events.ticketing.TicketManagement.UpdateTicket",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingTicketManagementUrl({
        protoPath: "/v1/events/{eventId}/tickets/{ticketNumber}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "ticket.checkIn.created" },
            { path: "ticket.guestDetails.form.inputValues.dateTime" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "ticket.guestDetails.form.inputValues.number" },
            {
              path: "ticket.guestDetails.form.inputValues.address.address.geocode.latitude"
            },
            {
              path: "ticket.guestDetails.form.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateTicket;
}
function bulkUpdateTickets(payload) {
  function __bulkUpdateTickets({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.ticket",
      method: "PATCH",
      methodFqn: "wix.events.ticketing.TicketManagement.BulkUpdateTickets",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingTicketManagementUrl({
        protoPath: "/v1/events/{eventId}/tickets",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "tickets.checkIn.created" },
            { path: "tickets.guestDetails.form.inputValues.dateTime" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "tickets.guestDetails.form.inputValues.number" },
            {
              path: "tickets.guestDetails.form.inputValues.address.address.geocode.latitude"
            },
            {
              path: "tickets.guestDetails.form.inputValues.address.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateTickets;
}

// src/events-v1-ticket-tickets.universal.ts
import { transformSDKAddressToRESTAddress } from "@wix/sdk-runtime/transformations/address";
import { transformRESTAddressToSDKAddress } from "@wix/sdk-runtime/transformations/address";
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var OrderStatus = /* @__PURE__ */ ((OrderStatus2) => {
  OrderStatus2["NA_ORDER_STATUS"] = "NA_ORDER_STATUS";
  OrderStatus2["FREE"] = "FREE";
  OrderStatus2["PENDING"] = "PENDING";
  OrderStatus2["PAID"] = "PAID";
  OrderStatus2["OFFLINE_PENDING"] = "OFFLINE_PENDING";
  OrderStatus2["INITIATED"] = "INITIATED";
  OrderStatus2["CANCELED"] = "CANCELED";
  OrderStatus2["DECLINED"] = "DECLINED";
  OrderStatus2["AUTHORIZED"] = "AUTHORIZED";
  OrderStatus2["VOIDED"] = "VOIDED";
  OrderStatus2["PARTIALLY_PAID"] = "PARTIALLY_PAID";
  return OrderStatus2;
})(OrderStatus || {});
var SubdivisionType = /* @__PURE__ */ ((SubdivisionType2) => {
  SubdivisionType2["UNKNOWN_SUBDIVISION_TYPE"] = "UNKNOWN_SUBDIVISION_TYPE";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_1"] = "ADMINISTRATIVE_AREA_LEVEL_1";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_2"] = "ADMINISTRATIVE_AREA_LEVEL_2";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_3"] = "ADMINISTRATIVE_AREA_LEVEL_3";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_4"] = "ADMINISTRATIVE_AREA_LEVEL_4";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_5"] = "ADMINISTRATIVE_AREA_LEVEL_5";
  SubdivisionType2["COUNTRY"] = "COUNTRY";
  return SubdivisionType2;
})(SubdivisionType || {});
var ChannelType = /* @__PURE__ */ ((ChannelType2) => {
  ChannelType2["ONLINE"] = "ONLINE";
  ChannelType2["OFFLINE_POS"] = "OFFLINE_POS";
  return ChannelType2;
})(ChannelType || {});
var TicketFieldset = /* @__PURE__ */ ((TicketFieldset2) => {
  TicketFieldset2["GUEST_DETAILS"] = "GUEST_DETAILS";
  TicketFieldset2["TICKET_DETAILS"] = "TICKET_DETAILS";
  TicketFieldset2["GUEST_FORM"] = "GUEST_FORM";
  return TicketFieldset2;
})(TicketFieldset || {});
var State = /* @__PURE__ */ ((State2) => {
  State2["ORDER_ARCHIVED"] = "ORDER_ARCHIVED";
  State2["ORDER_ACTIVE"] = "ORDER_ACTIVE";
  State2["TICKET_ARCHIVED"] = "TICKET_ARCHIVED";
  State2["TICKET_ACTIVE"] = "TICKET_ACTIVE";
  State2["CHECKED_IN"] = "CHECKED_IN";
  State2["NON_CHECKED_IN"] = "NON_CHECKED_IN";
  State2["FREE"] = "FREE";
  State2["PAID"] = "PAID";
  State2["MEMBER"] = "MEMBER";
  return State2;
})(State || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function listTickets2(eventId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    eventId,
    offset: options?.offset,
    limit: options?.limit,
    orderNumber: options?.orderNumber,
    ticketNumber: options?.ticketNumber,
    searchPhrase: options?.searchPhrase,
    orderStatus: options?.orderStatus,
    fieldset: options?.fieldset,
    state: options?.state,
    memberId: options?.memberId,
    facet: options?.facet,
    sort: options?.sort,
    contactId: options?.contactId,
    ticketDefinitionId: options?.ticketDefinitionId,
    seatId: options?.seatId
  });
  const reqOpts = listTickets(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTAddressToSDKAddress,
          paths: [
            { path: "tickets.guestDetails.form.inputValues.address.address" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          eventId: "$[0]",
          offset: "$[1].offset",
          limit: "$[1].limit",
          orderNumber: "$[1].orderNumber",
          ticketNumber: "$[1].ticketNumber",
          searchPhrase: "$[1].searchPhrase",
          orderStatus: "$[1].orderStatus",
          fieldset: "$[1].fieldset",
          state: "$[1].state",
          memberId: "$[1].memberId",
          facet: "$[1].facet",
          sort: "$[1].sort",
          contactId: "$[1].contactId",
          ticketDefinitionId: "$[1].ticketDefinitionId",
          seatId: "$[1].seatId"
        },
        singleArgumentUnchanged: false
      },
      ["eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getTicket2(identifiers, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    eventId: identifiers?.eventId,
    ticketNumber: identifiers?.ticketNumber,
    fieldset: options?.fieldset
  });
  const reqOpts = getTicket(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTAddressToSDKAddress,
          paths: [
            { path: "ticket.guestDetails.form.inputValues.address.address" }
          ]
        }
      ])
    )?.ticket;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          eventId: "$[0].eventId",
          ticketNumber: "$[0].ticketNumber",
          fieldset: "$[1].fieldset"
        },
        singleArgumentUnchanged: false
      },
      ["identifiers", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function checkInTickets(eventId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    eventId,
    ticketNumber: options?.ticketNumber
  });
  const reqOpts = checkInTicket(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTAddressToSDKAddress,
          paths: [
            { path: "tickets.guestDetails.form.inputValues.address.address" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          eventId: "$[0]",
          ticketNumber: "$[1].ticketNumber"
        },
        singleArgumentUnchanged: false
      },
      ["eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteTicketCheckIns(eventId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    eventId,
    ticketNumber: options?.ticketNumber
  });
  const reqOpts = deleteTicketCheckIn(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTAddressToSDKAddress,
          paths: [
            { path: "tickets.guestDetails.form.inputValues.address.address" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          eventId: "$[0]",
          ticketNumber: "$[1].ticketNumber"
        },
        singleArgumentUnchanged: false
      },
      ["eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateTicket2(identifiers, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({
      eventId: identifiers?.eventId,
      ticketNumber: identifiers?.ticketNumber,
      fields: options?.fields,
      archived: options?.archived,
      guestDetails: options?.guestDetails
    }),
    [
      {
        transformFn: transformSDKAddressToRESTAddress,
        paths: [
          { path: "checkoutForm.inputValues.address.address" },
          { path: "guestDetails.form.inputValues.address.address" }
        ]
      }
    ]
  );
  const reqOpts = updateTicket(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTAddressToSDKAddress,
          paths: [
            { path: "ticket.guestDetails.form.inputValues.address.address" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          eventId: "$[0].eventId",
          ticketNumber: "$[0].ticketNumber",
          fields: "$[1].fields",
          archived: "$[1].archived",
          guestDetails: "$[1].guestDetails"
        },
        singleArgumentUnchanged: false
      },
      ["identifiers", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateTickets2(eventId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    eventId,
    ticketNumber: options?.ticketNumber,
    archived: options?.archived
  });
  const reqOpts = bulkUpdateTickets(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTAddressToSDKAddress,
          paths: [
            { path: "tickets.guestDetails.form.inputValues.address.address" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          eventId: "$[0]",
          ticketNumber: "$[1].ticketNumber",
          archived: "$[1].archived"
        },
        singleArgumentUnchanged: false
      },
      ["eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
export {
  ChannelType,
  OrderStatus,
  State,
  SubdivisionType,
  TicketFieldset,
  WebhookIdentityType,
  bulkUpdateTickets2 as bulkUpdateTickets,
  checkInTickets,
  deleteTicketCheckIns,
  getTicket2 as getTicket,
  listTickets2 as listTickets,
  updateTicket2 as updateTicket
};
