// @wix/auto_sdk_events_ticket-definitions/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  FeeTypeOriginal: () => FeeType,
  GroupByOriginal: () => GroupBy,
  SortOrderOriginal: () => SortOrder,
  StateOriginal: () => State,
  TicketDefinitionFieldsetOriginal: () => TicketDefinitionFieldset,
  TicketDefinitionStateEnumStateOriginal: () => TicketDefinitionStateEnumState,
  TicketSaleStatusOriginal: () => TicketSaleStatus,
  TypeOriginal: () => Type,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  changeCurrency: () => changeCurrency2,
  createTicketDefinition: () => createTicketDefinition2,
  deleteTicketDefinition: () => deleteTicketDefinition2,
  getTicketDefinition: () => getTicketDefinition2,
  listTicketDefinitions: () => listTicketDefinitions2,
  queryTicketDefinitions: () => queryTicketDefinitions2,
  queryTicketDefinitionsV2: () => queryTicketDefinitionsV22,
  updateTicketDefinition: () => updateTicketDefinition2
});
module.exports = __toCommonJS(meta_exports);

// src/events-v1-ticket-definition-ticket-definitions.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEventsTicketingTicketDefinitionManagementUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/events/v2/ticket-definitions",
        destPath: "/v2/ticket-definitions"
      },
      {
        srcPath: "/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v2/ticket-definitions",
        destPath: "/v2/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/events-ticketing-v1-proxy",
        destPath: "/api"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/events/v2/ticket-definitions",
        destPath: "/v2/ticket-definitions"
      },
      {
        srcPath: "/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    _: [
      {
        srcPath: "/_api/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v2/ticket-definition",
        destPath: "/v2/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/events-ticketing-v1-proxy/v1",
        destPath: "/v1"
      },
      {
        srcPath: "/_api/events-ticketing-v1-proxy/v2",
        destPath: "/v2"
      },
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v2/ticket-definitions",
        destPath: "/v2/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v2/ticket-definitions",
        destPath: "/v2/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v2/ticket-definitions",
        destPath: "/v2/ticket-definitions"
      },
      {
        srcPath: "/_api/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    "api._base_domain_": [
      {
        srcPath: "/events/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/events/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/events/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/events/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    "editor-flow.wixapps.net": [
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/wix-events-web/v1/checkout",
        destPath: "/v1/checkout"
      },
      {
        srcPath: "/_api/wix-events-web/v1/ticket-definitions",
        destPath: "/v1/ticket-definitions"
      },
      {
        srcPath: "/_api/wix-events-web/v1/tickets/available",
        destPath: "/v1/tickets/available"
      },
      {
        srcPath: "/_api/wix-events-web/v1/pos-checkout",
        destPath: "/v1/pos-checkout"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_events_ticket-definitions";
function queryTicketDefinitions(payload) {
  function __queryTicketDefinitions({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.ticket_definition",
      method: "POST",
      methodFqn: "wix.events.ticketing.TicketDefinitionManagement.QueryTicketDefinitions",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingTicketDefinitionManagementUrl({
        protoPath: "/v1/ticket-definitions/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "definitions.salePeriod.startDate" },
            { path: "definitions.salePeriod.endDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryTicketDefinitions;
}
function queryTicketDefinitionsV2(payload) {
  function __queryTicketDefinitionsV2({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.ticket_definition",
      method: "POST",
      methodFqn: "wix.events.ticketing.TicketDefinitionManagement.QueryTicketDefinitionsV2",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingTicketDefinitionManagementUrl({
        protoPath: "/v2/ticket-definitions/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "definitions.salePeriod.startDate" },
            { path: "definitions.salePeriod.endDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryTicketDefinitionsV2;
}
function listTicketDefinitions(payload) {
  function __listTicketDefinitions({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.ticket_definition",
      method: "GET",
      methodFqn: "wix.events.ticketing.TicketDefinitionManagement.ListTicketDefinitions",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingTicketDefinitionManagementUrl({
        protoPath: "/v1/ticket-definitions",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "definitions.salePeriod.startDate" },
            { path: "definitions.salePeriod.endDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listTicketDefinitions;
}
function getTicketDefinition(payload) {
  function __getTicketDefinition({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.ticket_definition",
      method: "GET",
      methodFqn: "wix.events.ticketing.TicketDefinitionManagement.GetTicketDefinition",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingTicketDefinitionManagementUrl({
        protoPath: "/v1/ticket-definitions/{definitionId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "definition.salePeriod.startDate" },
            { path: "definition.salePeriod.endDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getTicketDefinition;
}
function createTicketDefinition(payload) {
  function __createTicketDefinition({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "definition.salePeriod.startDate" },
          { path: "definition.salePeriod.endDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v1.ticket_definition",
      method: "POST",
      methodFqn: "wix.events.ticketing.TicketDefinitionManagement.CreateTicketDefinition",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingTicketDefinitionManagementUrl({
        protoPath: "/v1/ticket-definitions",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "definition.salePeriod.startDate" },
            { path: "definition.salePeriod.endDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createTicketDefinition;
}
function updateTicketDefinition(payload) {
  function __updateTicketDefinition({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fields" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "definition.salePeriod.startDate" },
          { path: "definition.salePeriod.endDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v1.ticket_definition",
      method: "PATCH",
      methodFqn: "wix.events.ticketing.TicketDefinitionManagement.UpdateTicketDefinition",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingTicketDefinitionManagementUrl({
        protoPath: "/v1/ticket-definitions/{definitionId}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "definition.salePeriod.startDate" },
            { path: "definition.salePeriod.endDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateTicketDefinition;
}
function deleteTicketDefinition(payload) {
  function __deleteTicketDefinition({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.ticket_definition",
      method: "DELETE",
      methodFqn: "wix.events.ticketing.TicketDefinitionManagement.DeleteTicketDefinition",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingTicketDefinitionManagementUrl({
        protoPath: "/v1/ticket-definitions",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteTicketDefinition;
}
function changeCurrency(payload) {
  function __changeCurrency({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v1.ticket_definition",
      method: "PUT",
      methodFqn: "wix.events.ticketing.TicketDefinitionManagement.ChangeCurrency",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsTicketingTicketDefinitionManagementUrl({
        protoPath: "/v1/ticket-definitions/currency",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __changeCurrency;
}

// src/events-v1-ticket-definition-ticket-definitions.types.ts
var FeeType = /* @__PURE__ */ ((FeeType2) => {
  FeeType2["FEE_ADDED"] = "FEE_ADDED";
  FeeType2["FEE_INCLUDED"] = "FEE_INCLUDED";
  FeeType2["FEE_ADDED_AT_CHECKOUT"] = "FEE_ADDED_AT_CHECKOUT";
  return FeeType2;
})(FeeType || {});
var TicketSaleStatus = /* @__PURE__ */ ((TicketSaleStatus2) => {
  TicketSaleStatus2["SALE_SCHEDULED"] = "SALE_SCHEDULED";
  TicketSaleStatus2["SALE_STARTED"] = "SALE_STARTED";
  TicketSaleStatus2["SALE_ENDED"] = "SALE_ENDED";
  return TicketSaleStatus2;
})(TicketSaleStatus || {});
var TicketDefinitionStateEnumState = /* @__PURE__ */ ((TicketDefinitionStateEnumState2) => {
  TicketDefinitionStateEnumState2["INCLUDE_HIDDEN_NOT_ON_SALE"] = "INCLUDE_HIDDEN_NOT_ON_SALE";
  return TicketDefinitionStateEnumState2;
})(TicketDefinitionStateEnumState || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["STANDARD"] = "STANDARD";
  Type2["DONATION"] = "DONATION";
  return Type2;
})(Type || {});
var TicketDefinitionFieldset = /* @__PURE__ */ ((TicketDefinitionFieldset2) => {
  TicketDefinitionFieldset2["POLICY"] = "POLICY";
  TicketDefinitionFieldset2["DASHBOARD"] = "DASHBOARD";
  return TicketDefinitionFieldset2;
})(TicketDefinitionFieldset || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var State = /* @__PURE__ */ ((State2) => {
  State2["VISIBLE"] = "VISIBLE";
  State2["HIDDEN"] = "HIDDEN";
  State2["FREE"] = "FREE";
  State2["PAID"] = "PAID";
  return State2;
})(State || {});
var GroupBy = /* @__PURE__ */ ((GroupBy2) => {
  GroupBy2["NONE"] = "NONE";
  GroupBy2["EVENT_CREATED_DESC"] = "EVENT_CREATED_DESC";
  GroupBy2["EVENT_CREATED_ASC"] = "EVENT_CREATED_ASC";
  return GroupBy2;
})(GroupBy || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/events-v1-ticket-definition-ticket-definitions.meta.ts
function queryTicketDefinitions2() {
  const payload = {};
  const getRequestOptions = queryTicketDefinitions(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/ticket-definitions/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryTicketDefinitionsV22() {
  const payload = {};
  const getRequestOptions = queryTicketDefinitionsV2(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/ticket-definitions/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listTicketDefinitions2() {
  const payload = {};
  const getRequestOptions = listTicketDefinitions(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/ticket-definitions",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getTicketDefinition2() {
  const payload = { definitionId: ":definitionId" };
  const getRequestOptions = getTicketDefinition(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/ticket-definitions/{definitionId}",
    pathParams: { definitionId: "definitionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createTicketDefinition2() {
  const payload = {};
  const getRequestOptions = createTicketDefinition(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/ticket-definitions",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateTicketDefinition2() {
  const payload = { definitionId: ":definitionId" };
  const getRequestOptions = updateTicketDefinition(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/ticket-definitions/{definitionId}",
    pathParams: { definitionId: "definitionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteTicketDefinition2() {
  const payload = {};
  const getRequestOptions = deleteTicketDefinition(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/ticket-definitions",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function changeCurrency2() {
  const payload = {};
  const getRequestOptions = changeCurrency(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PUT",
    path: "/v1/ticket-definitions/currency",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  FeeTypeOriginal,
  GroupByOriginal,
  SortOrderOriginal,
  StateOriginal,
  TicketDefinitionFieldsetOriginal,
  TicketDefinitionStateEnumStateOriginal,
  TicketSaleStatusOriginal,
  TypeOriginal,
  WebhookIdentityTypeOriginal,
  changeCurrency,
  createTicketDefinition,
  deleteTicketDefinition,
  getTicketDefinition,
  listTicketDefinitions,
  queryTicketDefinitions,
  queryTicketDefinitionsV2,
  updateTicketDefinition
});
