// @wix/patterns/dist/esm/components/CustomFieldsPanel/CustomFieldItem.js
import { Box, TableListItem, Text } from '@wix/design-system';
import React, { useMemo } from 'react';
import { observer } from 'mobx-react-lite';
import { ManageFieldsPopoverMenu } from './ManageFieldsPopoverMenu';
function _CustomFieldItem(props) {
    const { field, showDivider, state } = props;
    const style = useMemo(() => ({
        display: 'flex',
        flexDirection: 'column',
        position: 'relative',
    }), []);
    return (React.createElement("div", { style: style, "data-hook": `custom-field-${field.id}` },
        React.createElement(TableListItem, { showDivider: showDivider, verticalPadding: "tiny", options: [
                {
                    value: (React.createElement(Box, null,
                        React.createElement("div", { style: {
                                padding: '0 6px',
                                wordBreak: 'break-word',
                            } },
                            React.createElement(Text, { size: "small" }, field.label || field.id)))),
                },
                {
                    value: (React.createElement(Box, { height: "24px" },
                        React.createElement(ManageFieldsPopoverMenu, { origin: "Custom fields panel", state: state, id: field.id, location: "Manage custom fields action cell" }))),
                    align: 'right',
                },
            ] })));
}
export const CustomFieldItem = observer(_CustomFieldItem);
