// @wix/patterns/dist/cjs/state/CustomFields/CustomFieldsFormState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.CustomFieldsFormState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _mobx = require("mobx");
class CustomFieldsFormState {
  constructor(params) {
    (0, _defineProperty2.default)(this, "_namespace", void 0);
    (0, _defineProperty2.default)(this, "_customFieldsWidgetState", void 0);
    (0, _defineProperty2.default)(this, "_fields", void 0);
    (0, _defineProperty2.default)(this, "_extendedFieldsDataMap", void 0);
    (0, _defineProperty2.default)(this, "_initialValues", void 0);
    (0, _defineProperty2.default)(this, "_translate", void 0);
    (0, _defineProperty2.default)(this, "_inputRefs", void 0);
    const {
      namespace,
      customFieldsWidgetState,
      translate
    } = params;
    this._namespace = namespace;
    this._customFieldsWidgetState = customFieldsWidgetState;
    const {
      fields,
      extendedFieldsDataMap
    } = customFieldsWidgetState.getNamespaceFormSchemaAndData(namespace);
    this._fields = fields ?? [];
    this._extendedFieldsDataMap = _mobx.observable.map(extendedFieldsDataMap);
    this._initialValues = {
      ...extendedFieldsDataMap
    };
    this._inputRefs = {};
    this._translate = translate;
    (0, _mobx.makeObservable)(this, {
      setValue: _mobx.action,
      setNewFields: _mobx.action,
      values: _mobx.computed,
      fields: _mobx.computed,
      isDirty: _mobx.computed,
      _fields: _mobx.observable.ref,
      writeableFields: _mobx.computed,
      _isUserFieldsNamespace: _mobx.computed,
      _inputRefs: _mobx.observable.ref,
      appName: _mobx.computed
    });
  }
  _getFullyQualifiedFieldId(originalId) {
    return `extendedFields.namespaces.${this._namespace}.${originalId}`;
  }
  get fields() {
    return this._fields;
  }
  get writeableFields() {
    const {
      _isUserFieldsNamespace,
      _fields
    } = this;
    if (_isUserFieldsNamespace) {
      return _fields;
    }
    return _fields.filter(field => {
      var _field$permissions;
      return (_field$permissions = field.permissions) == null || (_field$permissions = _field$permissions.write) == null ? void 0 : _field$permissions.includes('users');
    });
  }
  get _isUserFieldsNamespace() {
    const {
      _customFieldsWidgetState,
      _namespace
    } = this;
    const {
      _userFieldsNamespace
    } = _customFieldsWidgetState;
    return _namespace === _userFieldsNamespace;
  }
  init() {
    const {
      _customFieldsWidgetState
    } = this;
    const disposers = [(0, _mobx.reaction)(() => _customFieldsWidgetState.getNamespaceFormSchemaAndData(this._namespace), ({
      fields
    }) => {
      if (!fields) {
        return;
      }
      this.setNewFields(fields);
    }, {
      fireImmediately: true
    })];
    return () => {
      disposers.forEach(d => d == null ? void 0 : d());
    };
  }
  setValue({
    id,
    value,
    onChange
  }) {
    this._extendedFieldsDataMap.set(id, value);
    onChange == null || onChange({
      id: `extendedFields.namespaces.${this._namespace}.${id}`,
      value,
      allFields: Object.keys(this.values).reduce((acc, fieldId) => {
        acc[fieldId] = {
          id: this._getFullyQualifiedFieldId(fieldId),
          value: this.values[fieldId]
        };
        return acc;
      }, {}),
      namespace: this._namespace
    });
  }
  getInitialValue(id) {
    return this._initialValues[id];
  }
  setInputRef(id, input) {
    this._inputRefs[id] = input;
  }
  setNewFields(fields) {
    this._fields = fields;
    this._adjustExtendedFields();
  }
  _adjustExtendedFields() {
    const existingFields = this._fields.map(field => field.id);
    this._extendedFieldsDataMap.forEach((_, key) => {
      if (!existingFields.includes(key)) {
        this._extendedFieldsDataMap.delete(key);
      }
    });
  }
  validate() {
    const invalid = Object.keys(this._inputRefs).find(fieldId => {
      const ref = this._inputRefs[fieldId];
      return ref.invalid;
    });
    if (invalid) {
      var _this$_inputRefs$inva, _this$_inputRefs$inva2;
      (_this$_inputRefs$inva = (_this$_inputRefs$inva2 = this._inputRefs[invalid]).focus) == null || _this$_inputRefs$inva.call(_this$_inputRefs$inva2);
      return {
        isValid: false
      };
    }
    return {
      isValid: true,
      extendedFields: this.values
    };
  }
  get isDirty() {
    return !!Array.from(this._extendedFieldsDataMap.entries()).find(([key, value]) => {
      // Files field is not considered dirty since it cannot be changed
      const customField = this._fields.find(field => field.id === key);
      if ((customField == null ? void 0 : customField.type) === 'files') {
        return false;
      }
      if (this._initialValues.hasOwnProperty(key)) {
        return this._initialValues[key] !== value;
      } else {
        // includes both any string value and boolean types (all existing field types values)
        return !!value;
      }
    });
  }
  get appName() {
    var _dataExtension$appSch;
    const {
      _customFieldsWidgetState,
      _namespace
    } = this;
    const {
      apps,
      dataExtension
    } = _customFieldsWidgetState;
    const {
      dataExtensionService: {
        userFieldsNamespace
      }
    } = dataExtension;
    if (_namespace === userFieldsNamespace) {
      return this._translate('cairo.customFields.entityPage.sectionTitle');
    }
    const appSchema = (_dataExtension$appSch = dataExtension.appSchemas) == null ? void 0 : _dataExtension$appSch.find(schema => schema.namespace === _namespace);
    return apps.appNames[(appSchema == null ? void 0 : appSchema.appDefId) || ''];
  }
  get values() {
    return Object.fromEntries(this._extendedFieldsDataMap);
  }
}
exports.CustomFieldsFormState = CustomFieldsFormState;
