// @wix/patterns/dist/cjs/state/CustomFields/CustomFieldsPanelState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.CustomFieldsPanelState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _bexCore = require("@wix/bex-core");
var _mobx = require("mobx");
class CustomFieldsPanelState {
  constructor(params) {
    (0, _defineProperty2.default)(this, "reportBi", void 0);
    (0, _defineProperty2.default)(this, "closeSidePanel", void 0);
    (0, _defineProperty2.default)(this, "dataExtension", void 0);
    (0, _defineProperty2.default)(this, "customFieldModal", void 0);
    (0, _defineProperty2.default)(this, "archiveFieldModal", void 0);
    this.reportBi = params.reportBi;
    this.closeSidePanel = params.closeSidePanel;
    this.dataExtension = params.dataExtension;
    this.archiveFieldModal = new _bexCore.ArchiveFieldModalState({
      reportBi: this.reportBi,
      dataExtension: this.dataExtension
    });
    this.customFieldModal = new _bexCore.CustomFieldModalState({
      reportBi: this.reportBi,
      dataExtensionState: this.dataExtension
    });
    (0, _mobx.makeAutoObservable)(this);
  }
  openCustomFieldModal(params) {
    const {
      customFieldModal
    } = this;
    const {
      customFieldModalBiReporter
    } = customFieldModal;
    customFieldModal.reset();
    customFieldModal.mode = params.fieldId ? 'edit' : 'new';
    customFieldModal.origin = params.origin;
    customFieldModal.customOrigin = params.customOrigin;
    !params.fieldId && customFieldModalBiReporter.reportAddCustomField({
      origin: params.origin,
      customOrigin: params.customOrigin
    });
    customFieldModalBiReporter.formSnapshot = null;
    customFieldModal.setDefaultPermissions();
    if (this.dataExtension.userSchema) {
      if (params.fieldId) {
        const formValues = this.dataExtension.extractFieldValuesFromSchema(params.fieldId);
        if (!formValues) {
          return;
        }
        customFieldModal.setFormValues(formValues, {
          disablePermissionCheckboxes: true
        });
        customFieldModalBiReporter.formSnapshot = formValues;
      }
    }
    customFieldModal.modalState.modal.open();
  }
}
exports.CustomFieldsPanelState = CustomFieldsPanelState;
