// @wix/auto_sdk_events_policies/build/internal/es/meta.mjs
// src/events-v2-policy-policies.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixEventsV2PolicyManagementUrl(opts) {
  const domainToMappings = {
    _: [
      {
        srcPath: "/_api/events-policies",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/events-policies",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/events-policies",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/events-policies",
        destPath: ""
      },
      {
        srcPath: "/events/v2/policies",
        destPath: "/v2/policies"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/events-policies",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/events-policies",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/events-policies",
        destPath: ""
      }
    ],
    "events.wixapps.net": [
      {
        srcPath: "/_api/events-policies",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_events_policies";
function createPolicy(payload) {
  function __createPolicy({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "policy.createdDate" }, { path: "policy.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v2.policy",
      method: "POST",
      methodFqn: "wix.events.v2.PolicyManagement.CreatePolicy",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsV2PolicyManagementUrl({
        protoPath: "/v2/policies",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "policy.createdDate" },
            { path: "policy.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createPolicy;
}
function updatePolicy(payload) {
  function __updatePolicy({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "policy.createdDate" }, { path: "policy.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v2.policy",
      method: "PATCH",
      methodFqn: "wix.events.v2.PolicyManagement.UpdatePolicy",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsV2PolicyManagementUrl({
        protoPath: "/v2/policies/{policy.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "policy.createdDate" },
            { path: "policy.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updatePolicy;
}
function deletePolicy(payload) {
  function __deletePolicy({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v2.policy",
      method: "DELETE",
      methodFqn: "wix.events.v2.PolicyManagement.DeletePolicy",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsV2PolicyManagementUrl({
        protoPath: "/v2/policies/{policyId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deletePolicy;
}
function queryPolicies(payload) {
  function __queryPolicies({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v2.policy",
      method: "POST",
      methodFqn: "wix.events.v2.PolicyManagement.QueryPolicies",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsV2PolicyManagementUrl({
        protoPath: "/v2/policies/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "policies.createdDate" },
            { path: "policies.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryPolicies;
}
function reorderEventPolicies(payload) {
  function __reorderEventPolicies({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v2.policy",
      method: "POST",
      methodFqn: "wix.events.v2.PolicyManagement.ReorderEventPolicies",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsV2PolicyManagementUrl({
        protoPath: "/v2/policies/reorder",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "policies.createdDate" },
            { path: "policies.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __reorderEventPolicies;
}
function getPolicy(payload) {
  function __getPolicy({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v2.policy",
      method: "GET",
      methodFqn: "wix.events.v2.PolicyManagement.GetPolicy",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsV2PolicyManagementUrl({
        protoPath: "/v2/policies/{policyId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "policy.createdDate" },
            { path: "policy.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getPolicy;
}

// src/events-v2-policy-policies.types.ts
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var LocationType = /* @__PURE__ */ ((LocationType2) => {
  LocationType2["VENUE"] = "VENUE";
  LocationType2["ONLINE"] = "ONLINE";
  return LocationType2;
})(LocationType || {});
var SubdivisionType = /* @__PURE__ */ ((SubdivisionType2) => {
  SubdivisionType2["UNKNOWN_SUBDIVISION_TYPE"] = "UNKNOWN_SUBDIVISION_TYPE";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_1"] = "ADMINISTRATIVE_AREA_LEVEL_1";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_2"] = "ADMINISTRATIVE_AREA_LEVEL_2";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_3"] = "ADMINISTRATIVE_AREA_LEVEL_3";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_4"] = "ADMINISTRATIVE_AREA_LEVEL_4";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_5"] = "ADMINISTRATIVE_AREA_LEVEL_5";
  SubdivisionType2["COUNTRY"] = "COUNTRY";
  return SubdivisionType2;
})(SubdivisionType || {});
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["ONE_TIME"] = "ONE_TIME";
  Status2["RECURRING"] = "RECURRING";
  Status2["RECURRING_NEXT"] = "RECURRING_NEXT";
  Status2["RECURRING_LAST_ENDED"] = "RECURRING_LAST_ENDED";
  Status2["RECURRING_LAST_CANCELED"] = "RECURRING_LAST_CANCELED";
  return Status2;
})(Status || {});
var EventStatus = /* @__PURE__ */ ((EventStatus2) => {
  EventStatus2["SCHEDULED"] = "SCHEDULED";
  EventStatus2["STARTED"] = "STARTED";
  EventStatus2["ENDED"] = "ENDED";
  EventStatus2["CANCELED"] = "CANCELED";
  return EventStatus2;
})(EventStatus || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/events-v2-policy-policies.meta.ts
function createPolicy2() {
  const payload = {};
  const getRequestOptions = createPolicy(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/policies",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updatePolicy2() {
  const payload = { policy: { id: ":policyId" } };
  const getRequestOptions = updatePolicy(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v2/policies/{policy.id}",
    pathParams: { policyId: "policyId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deletePolicy2() {
  const payload = { policyId: ":policyId" };
  const getRequestOptions = deletePolicy(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v2/policies/{policyId}",
    pathParams: { policyId: "policyId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryPolicies2() {
  const payload = {};
  const getRequestOptions = queryPolicies(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/policies/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function reorderEventPolicies2() {
  const payload = {};
  const getRequestOptions = reorderEventPolicies(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/policies/reorder",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getPolicy2() {
  const payload = { policyId: ":policyId" };
  const getRequestOptions = getPolicy(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/policies/{policyId}",
    pathParams: { policyId: "policyId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  EventStatus as EventStatusOriginal,
  LocationType as LocationTypeOriginal,
  SortOrder as SortOrderOriginal,
  Status as StatusOriginal,
  SubdivisionType as SubdivisionTypeOriginal,
  WebhookIdentityType as WebhookIdentityTypeOriginal,
  createPolicy2 as createPolicy,
  deletePolicy2 as deletePolicy,
  getPolicy2 as getPolicy,
  queryPolicies2 as queryPolicies,
  reorderEventPolicies2 as reorderEventPolicies,
  updatePolicy2 as updatePolicy
};
