// @wix/design-system/dist/esm/AreaChart/AreaChart.st.css.js

import "../Foundation/stylable/colors.st.css.js";
import "../Foundation/stylable/easing.st.css.js";
import { classesRuntime, statesRuntime } from "./../stylable-esm-runtime.js";



var _namespace_ = "wds_1_226_0_AreaChart";
var _style_ = classesRuntime.bind(null, _namespace_);

export var cssStates = statesRuntime.bind(null, _namespace_);
export var style = _style_;
export var st = _style_;

export var namespace = _namespace_;
export var classes = {"root":"wds_1_226_0_AreaChart__root","lineChart":"wds_1_226_0_AreaChart__lineChart","lineChartCursor":"wds_1_226_0_AreaChart__lineChartCursor","lineChartAxis":"wds_1_226_0_AreaChart__lineChartAxis"};
export var keyframes = {}; 
export var layers = {};
export var containers = {};
export var stVars = {"canvas-top-gap":"6px","canvas-bottom-gap":"31px","datasetBg":"rgba(51, 112, 251, 0.06)","pointBorderColor":"rgba(51, 112, 251, 0)","pointBackgroundColor":"rgba(51, 112, 251, 0)","datasetLineColor":"#3370FB","pointHoverBackgroundColor":"#FFFFFF","pointHoverBorderColor":"#3370FB","gridLineZeroLineColor":"#7A92A5","gridLineFontColor":"#CBD3DC","tooltipBg":"#162D3D"}; 
export var vars = {};
