// @wix/patterns/dist/cjs/components/CollectionPage/CollectionPage.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.CollectionPage = CollectionPage;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _designSystem = require("@wix/design-system");
var _react = _interopRequireWildcard(require("react"));
var _CollectionPageHeader = require("../CollectionPageHeader");
var _InfiniteScrollPage = require("../PageWrapper/InfiniteScrollPage");
var _CollectionContext = require("../CollectionContext");
var _react2 = require("@wix/bex-core/react");
var _usePageHeader = require("../../hooks/usePageHeader");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CollectionPage/CollectionPage.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
/**
 * @deprecated Use [CollectionPage](./?path=%2Fstory%2Fbase-components-pages-collection-page--collectionpage) from
 * `@wix/patterns/page`
 */
function CollectionPage({
  children,
  state,
  hideTotal,
  ...rest
}) {
  const childrenArr = _react.Children.toArray(children);
  const {
    header,
    title: pageTitle
  } = (0, _usePageHeader.usePageHeader)(childrenArr);
  const collectionPageState = (0, _react2.useCollectionPageState)({
    pageTitle
  });
  return /*#__PURE__*/_react.default.createElement(_CollectionContext.CollectionProvider, {
    value: state,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 40,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(_InfiniteScrollPage.InfiniteScrollPage, (0, _extends2.default)({
    state: collectionPageState
  }, rest, {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 41,
      columnNumber: 7
    }
  }), header && /*#__PURE__*/_react.default.createElement(_designSystem.Page.Header, (0, _extends2.default)({}, header.props, {
    title: hideTotal ? header.props.title : /*#__PURE__*/_react.default.createElement(_CollectionPageHeader.CollectionPageHeader, {
      state: state,
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 49,
        columnNumber: 17
      }
    }, header.props.title),
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 43,
      columnNumber: 11
    }
  })), childrenArr.filter(e => e !== header)));
}
