// @wix/auto_sdk_restaurants_item-variants/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  SortOrderOriginal: () => SortOrder,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  bulkCreateVariants: () => bulkCreateVariants2,
  bulkDeleteVariants: () => bulkDeleteVariants2,
  bulkUpdateVariants: () => bulkUpdateVariants2,
  countVariants: () => countVariants2,
  createVariant: () => createVariant2,
  deleteVariant: () => deleteVariant2,
  getVariant: () => getVariant2,
  listVariants: () => listVariants2,
  queryVariants: () => queryVariants2,
  updateVariant: () => updateVariant2
});
module.exports = __toCommonJS(meta_exports);

// src/restaurants-menus-v1-item-variant-item-variants.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-variant",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-variant",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-variant",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-menus-item-variant",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-menus-item-variant",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-menus-item-variant",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/variants",
        destPath: "/v1/variants"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/variants/create",
        destPath: "/v1/bulk/variants/create"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/variants/update",
        destPath: "/v1/bulk/variants/update"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/variants/delete",
        destPath: "/v1/bulk/variants/delete"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-variant",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-menus-item-variant",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/variants",
        destPath: "/v1/variants"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/variants/create",
        destPath: "/v1/bulk/variants/create"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/variants/update",
        destPath: "/v1/bulk/variants/update"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/variants/delete",
        destPath: "/v1/bulk/variants/delete"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/item-variants",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/variants/create",
        destPath: "/v1/bulk/variants/create"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/variants/update",
        destPath: "/v1/bulk/variants/update"
      },
      {
        srcPath: "/restaurants/menus/v1/variants",
        destPath: "/v1/variants"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/variants/delete",
        destPath: "/v1/bulk/variants/delete"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-menus-item-variant",
        destPath: ""
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-variant",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_item-variants";
function createVariant(payload) {
  function __createVariant({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "variant.createdDate" },
          { path: "variant.updatedDate" },
          { path: "variant.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_variant",
      method: "POST",
      methodFqn: "wix.restaurants.menus_item_variant.v1.VariantsService.CreateVariant",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl({
        protoPath: "/v1/variants",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "variant.createdDate" },
            { path: "variant.updatedDate" },
            { path: "variant.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createVariant;
}
function getVariant(payload) {
  function __getVariant({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_variant",
      method: "GET",
      methodFqn: "wix.restaurants.menus_item_variant.v1.VariantsService.GetVariant",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl({
        protoPath: "/v1/variants/{variantId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "variant.createdDate" },
            { path: "variant.updatedDate" },
            { path: "variant.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getVariant;
}
function listVariants(payload) {
  function __listVariants({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_variant",
      method: "GET",
      methodFqn: "wix.restaurants.menus_item_variant.v1.VariantsService.ListVariants",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl({
        protoPath: "/v1/variants",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "variants.createdDate" },
            { path: "variants.updatedDate" },
            { path: "variants.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listVariants;
}
function updateVariant(payload) {
  function __updateVariant({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "variant.createdDate" },
          { path: "variant.updatedDate" },
          { path: "variant.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_variant",
      method: "PATCH",
      methodFqn: "wix.restaurants.menus_item_variant.v1.VariantsService.UpdateVariant",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl({
        protoPath: "/v1/variants/{variant.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "variant.createdDate" },
            { path: "variant.updatedDate" },
            { path: "variant.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateVariant;
}
function bulkCreateVariants(payload) {
  function __bulkCreateVariants({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "variants.createdDate" },
          { path: "variants.updatedDate" },
          { path: "variants.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_variant",
      method: "POST",
      methodFqn: "wix.restaurants.menus_item_variant.v1.VariantsService.BulkCreateVariants",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl({
        protoPath: "/v1/bulk/variants/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.variant.createdDate" },
            { path: "results.variant.updatedDate" },
            { path: "results.variant.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateVariants;
}
function bulkUpdateVariants(payload) {
  function __bulkUpdateVariants({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "variants.fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "variants.variant.createdDate" },
          { path: "variants.variant.updatedDate" },
          { path: "variants.variant.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_variant",
      method: "POST",
      methodFqn: "wix.restaurants.menus_item_variant.v1.VariantsService.BulkUpdateVariants",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl({
        protoPath: "/v1/bulk/variants/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.variant.createdDate" },
            { path: "results.variant.updatedDate" },
            { path: "results.variant.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateVariants;
}
function deleteVariant(payload) {
  function __deleteVariant({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_variant",
      method: "DELETE",
      methodFqn: "wix.restaurants.menus_item_variant.v1.VariantsService.DeleteVariant",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl({
        protoPath: "/v1/variants/{variantId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteVariant;
}
function queryVariants(payload) {
  function __queryVariants({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_variant",
      method: "POST",
      methodFqn: "wix.restaurants.menus_item_variant.v1.VariantsService.QueryVariants",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl({
        protoPath: "/v1/variants/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "variants.createdDate" },
            { path: "variants.updatedDate" },
            { path: "variants.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryVariants;
}
function countVariants(payload) {
  function __countVariants({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_variant",
      method: "POST",
      methodFqn: "wix.restaurants.menus_item_variant.v1.VariantsService.CountVariants",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl({
        protoPath: "/v1/variants/count",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __countVariants;
}
function bulkDeleteVariants(payload) {
  function __bulkDeleteVariants({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_variant",
      method: "DELETE",
      methodFqn: "wix.restaurants.menus_item_variant.v1.VariantsService.BulkDeleteVariants",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl({
        protoPath: "/v1/bulk/variants/delete",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __bulkDeleteVariants;
}

// src/restaurants-menus-v1-item-variant-item-variants.types.ts
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/restaurants-menus-v1-item-variant-item-variants.meta.ts
function createVariant2() {
  const payload = {};
  const getRequestOptions = createVariant(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/variants",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getVariant2() {
  const payload = { variantId: ":variantId" };
  const getRequestOptions = getVariant(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/variants/{variantId}",
    pathParams: { variantId: "variantId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listVariants2() {
  const payload = {};
  const getRequestOptions = listVariants(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/variants",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateVariant2() {
  const payload = { variant: { id: ":variantId" } };
  const getRequestOptions = updateVariant(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/variants/{variant.id}",
    pathParams: { variantId: "variantId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkCreateVariants2() {
  const payload = {};
  const getRequestOptions = bulkCreateVariants(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/variants/create",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateVariants2() {
  const payload = {};
  const getRequestOptions = bulkUpdateVariants(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/variants/update",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteVariant2() {
  const payload = { variantId: ":variantId" };
  const getRequestOptions = deleteVariant(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/variants/{variantId}",
    pathParams: { variantId: "variantId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryVariants2() {
  const payload = {};
  const getRequestOptions = queryVariants(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/variants/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function countVariants2() {
  const payload = {};
  const getRequestOptions = countVariants(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/variants/count",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkDeleteVariants2() {
  const payload = {};
  const getRequestOptions = bulkDeleteVariants(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/bulk/variants/delete",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  SortOrderOriginal,
  WebhookIdentityTypeOriginal,
  bulkCreateVariants,
  bulkDeleteVariants,
  bulkUpdateVariants,
  countVariants,
  createVariant,
  deleteVariant,
  getVariant,
  listVariants,
  queryVariants,
  updateVariant
});
