// @wix/patterns/dist/esm/state/NestedTableState/createNestedOptimisticActions.js
import { CollectionOptimisticActions, } from '@wix/bex-core';
import { createParentIdPredicate } from './createParentIdPredicate';
export function createNestedOptimisticActions(container, collection, state, params) {
    const parentIdPredicate = createParentIdPredicate(state);
    const optimisticActions = collection._optimisticActions ??
        new CollectionOptimisticActions({
            container,
            collection,
            orderBy: () => [],
            ...params,
            predicate: (query) => {
                const originalPredicateForQuery = params?.predicate?.(query) ??
                    state?.levelDescriptor?.optimisticActionsPredicate?.(query);
                const parentIdPredicateForQuery = parentIdPredicate(query);
                return (item) => parentIdPredicateForQuery(item) &&
                    (originalPredicateForQuery?.(item) ?? true);
            },
            cacheNamespace: state?.parentId,
            sequences: state?.sequences,
            query: state?.query,
        });
    optimisticActions._commitMode = false;
    return optimisticActions;
}
