// @wix/auto_sdk_media_enterprise-media-items/build/internal/es/index.mjs
// src/media-enterprise-public-media-v1-enterprise-item-enterprise-media-items.public.ts
import { renameKeysFromRESTResponseToSDKResponse as renameKeysFromRESTResponseToSDKResponse2 } from "@wix/sdk-runtime/rename-all-nested-keys";
import { transformRESTAudioToSDKAudio as transformRESTAudioToSDKAudio2 } from "@wix/sdk-runtime/transformations/audio";
import { transformRESTDocumentToSDKDocument as transformRESTDocumentToSDKDocument2 } from "@wix/sdk-runtime/transformations/document";
import { transformRESTImageToSDKImage as transformRESTImageToSDKImage2 } from "@wix/sdk-runtime/transformations/image";
import { transformRESTTimestampToSDKTimestamp as transformRESTTimestampToSDKTimestamp2 } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths as transformPaths3 } from "@wix/sdk-runtime/transformations/transform-paths";
import { transformRESTVideoV2ToSDKVideoV2 as transformRESTVideoV2ToSDKVideoV22 } from "@wix/sdk-runtime/transformations/video-v2";
import { EventDefinition } from "@wix/sdk-types";

// src/media-enterprise-public-media-v1-enterprise-item-enterprise-media-items.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/media-enterprise-public-media-v1-enterprise-item-enterprise-media-items.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixMediaEnterprisePublicMediaV1EnterpriseItemsServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/enterprise-public-media-server",
        destPath: ""
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/_api/enterprise-public-media-server",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/enterprise-public-media-server",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/enterprise-public-media-server",
        destPath: ""
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/enterprise-public-media-server",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_media_enterprise-media-items";
function itemUploadCallback(payload) {
  function __itemUploadCallback({ host }) {
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_item",
      method: "POST",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseItemsService.ItemUploadCallback",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseItemsServiceUrl({
        protoPath: "/v1/enterprise/items/upload/callback",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __itemUploadCallback;
}
function generateFileUploadUrl(payload) {
  function __generateFileUploadUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_item",
      method: "POST",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseItemsService.GenerateFileUploadUrl",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseItemsServiceUrl({
        protoPath: "/v1/enterprise/items/upload/generate-url",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __generateFileUploadUrl;
}
function importFile(payload) {
  function __importFile({ host }) {
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_item",
      method: "POST",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseItemsService.ImportFile",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseItemsServiceUrl({
        protoPath: "/v1/enterprise/items/upload/import",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "item.createdDate" },
            { path: "item.updatedDate" },
            { path: "item.assets.image.urlExpirationDate" },
            { path: "item.assets.video.urlExpirationDate" },
            { path: "item.assets.video.resolutions.urlExpirationDate" },
            {
              path: "item.assets.video.resolutions.poster.urlExpirationDate"
            },
            { path: "item.assets.video.posters.urlExpirationDate" },
            { path: "item.assets.vector.urlExpirationDate" },
            { path: "item.assets.audio.urlExpirationDate" },
            { path: "item.assets.document.urlExpirationDate" },
            { path: "item.assets.archive.urlExpirationDate" },
            { path: "item.assets.model3d.urlExpirationDate" },
            { path: "item.assets.model3d.thumbnail.urlExpirationDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __importFile;
}
function bulkImportFiles(payload) {
  function __bulkImportFiles({ host }) {
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_item",
      method: "POST",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseItemsService.BulkImportFiles",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseItemsServiceUrl({
        protoPath: "/v1/bulk/files/import",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.assets.image.urlExpirationDate" },
            { path: "results.item.assets.video.urlExpirationDate" },
            {
              path: "results.item.assets.video.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.assets.video.resolutions.poster.urlExpirationDate"
            },
            { path: "results.item.assets.video.posters.urlExpirationDate" },
            { path: "results.item.assets.vector.urlExpirationDate" },
            { path: "results.item.assets.audio.urlExpirationDate" },
            { path: "results.item.assets.document.urlExpirationDate" },
            { path: "results.item.assets.archive.urlExpirationDate" },
            { path: "results.item.assets.model3d.urlExpirationDate" },
            {
              path: "results.item.assets.model3d.thumbnail.urlExpirationDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkImportFiles;
}
function searchItems(payload) {
  function __searchItems({ host }) {
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_item",
      method: "POST",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseItemsService.SearchItems",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseItemsServiceUrl({
        protoPath: "/v1/enterprise/items/search",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "items.createdDate" },
            { path: "items.updatedDate" },
            { path: "items.assets.image.urlExpirationDate" },
            { path: "items.assets.video.urlExpirationDate" },
            { path: "items.assets.video.resolutions.urlExpirationDate" },
            {
              path: "items.assets.video.resolutions.poster.urlExpirationDate"
            },
            { path: "items.assets.video.posters.urlExpirationDate" },
            { path: "items.assets.vector.urlExpirationDate" },
            { path: "items.assets.audio.urlExpirationDate" },
            { path: "items.assets.document.urlExpirationDate" },
            { path: "items.assets.archive.urlExpirationDate" },
            { path: "items.assets.model3d.urlExpirationDate" },
            { path: "items.assets.model3d.thumbnail.urlExpirationDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __searchItems;
}
function queryItems(payload) {
  function __queryItems({ host }) {
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_item",
      method: "POST",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseItemsService.QueryItems",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseItemsServiceUrl({
        protoPath: "/v1/enterprise/items/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "items.createdDate" },
            { path: "items.updatedDate" },
            { path: "items.assets.image.urlExpirationDate" },
            { path: "items.assets.video.urlExpirationDate" },
            { path: "items.assets.video.resolutions.urlExpirationDate" },
            {
              path: "items.assets.video.resolutions.poster.urlExpirationDate"
            },
            { path: "items.assets.video.posters.urlExpirationDate" },
            { path: "items.assets.vector.urlExpirationDate" },
            { path: "items.assets.audio.urlExpirationDate" },
            { path: "items.assets.document.urlExpirationDate" },
            { path: "items.assets.archive.urlExpirationDate" },
            { path: "items.assets.model3d.urlExpirationDate" },
            { path: "items.assets.model3d.thumbnail.urlExpirationDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryItems;
}
function updateItem(payload) {
  function __updateItem({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "item.createdDate" },
          { path: "item.updatedDate" },
          { path: "item.assets.image.urlExpirationDate" },
          { path: "item.assets.video.urlExpirationDate" },
          { path: "item.assets.video.resolutions.urlExpirationDate" },
          { path: "item.assets.video.resolutions.poster.urlExpirationDate" },
          { path: "item.assets.video.posters.urlExpirationDate" },
          { path: "item.assets.vector.urlExpirationDate" },
          { path: "item.assets.audio.urlExpirationDate" },
          { path: "item.assets.document.urlExpirationDate" },
          { path: "item.assets.archive.urlExpirationDate" },
          { path: "item.assets.model3d.urlExpirationDate" },
          { path: "item.assets.model3d.thumbnail.urlExpirationDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_item",
      method: "PATCH",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseItemsService.UpdateItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseItemsServiceUrl({
        protoPath: "/v1/enterprise/items/{item.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "item.createdDate" },
            { path: "item.updatedDate" },
            { path: "item.assets.image.urlExpirationDate" },
            { path: "item.assets.video.urlExpirationDate" },
            { path: "item.assets.video.resolutions.urlExpirationDate" },
            {
              path: "item.assets.video.resolutions.poster.urlExpirationDate"
            },
            { path: "item.assets.video.posters.urlExpirationDate" },
            { path: "item.assets.vector.urlExpirationDate" },
            { path: "item.assets.audio.urlExpirationDate" },
            { path: "item.assets.document.urlExpirationDate" },
            { path: "item.assets.archive.urlExpirationDate" },
            { path: "item.assets.model3d.urlExpirationDate" },
            { path: "item.assets.model3d.thumbnail.urlExpirationDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateItem;
}
function bulkUpdateItem(payload) {
  function __bulkUpdateItem({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "updateItemRequests.fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "updateItemRequests.item.createdDate" },
          { path: "updateItemRequests.item.updatedDate" },
          { path: "updateItemRequests.item.assets.image.urlExpirationDate" },
          { path: "updateItemRequests.item.assets.video.urlExpirationDate" },
          {
            path: "updateItemRequests.item.assets.video.resolutions.urlExpirationDate"
          },
          {
            path: "updateItemRequests.item.assets.video.resolutions.poster.urlExpirationDate"
          },
          {
            path: "updateItemRequests.item.assets.video.posters.urlExpirationDate"
          },
          { path: "updateItemRequests.item.assets.vector.urlExpirationDate" },
          { path: "updateItemRequests.item.assets.audio.urlExpirationDate" },
          { path: "updateItemRequests.item.assets.document.urlExpirationDate" },
          { path: "updateItemRequests.item.assets.archive.urlExpirationDate" },
          { path: "updateItemRequests.item.assets.model3d.urlExpirationDate" },
          {
            path: "updateItemRequests.item.assets.model3d.thumbnail.urlExpirationDate"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_item",
      method: "PATCH",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseItemsService.BulkUpdateItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseItemsServiceUrl({
        protoPath: "/v1/bulk/enterprise/items/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.assets.image.urlExpirationDate" },
            { path: "results.item.assets.video.urlExpirationDate" },
            {
              path: "results.item.assets.video.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.assets.video.resolutions.poster.urlExpirationDate"
            },
            { path: "results.item.assets.video.posters.urlExpirationDate" },
            { path: "results.item.assets.vector.urlExpirationDate" },
            { path: "results.item.assets.audio.urlExpirationDate" },
            { path: "results.item.assets.document.urlExpirationDate" },
            { path: "results.item.assets.archive.urlExpirationDate" },
            { path: "results.item.assets.model3d.urlExpirationDate" },
            {
              path: "results.item.assets.model3d.thumbnail.urlExpirationDate"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateItem;
}
function getItem(payload) {
  function __getItem({ host }) {
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_item",
      method: "GET",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseItemsService.GetItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseItemsServiceUrl({
        protoPath: "/v1/enterprise/items/{itemId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "item.createdDate" },
            { path: "item.updatedDate" },
            { path: "item.assets.image.urlExpirationDate" },
            { path: "item.assets.video.urlExpirationDate" },
            { path: "item.assets.video.resolutions.urlExpirationDate" },
            {
              path: "item.assets.video.resolutions.poster.urlExpirationDate"
            },
            { path: "item.assets.video.posters.urlExpirationDate" },
            { path: "item.assets.vector.urlExpirationDate" },
            { path: "item.assets.audio.urlExpirationDate" },
            { path: "item.assets.document.urlExpirationDate" },
            { path: "item.assets.archive.urlExpirationDate" },
            { path: "item.assets.model3d.urlExpirationDate" },
            { path: "item.assets.model3d.thumbnail.urlExpirationDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getItem;
}
function linkItemToCategories(payload) {
  function __linkItemToCategories({ host }) {
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_item",
      method: "POST",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseItemsService.LinkItemToCategories",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseItemsServiceUrl({
        protoPath: "/v1/enterprise/items/{itemId}/link-categories",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __linkItemToCategories;
}
function unlinkItemFromCategories(payload) {
  function __unlinkItemFromCategories({ host }) {
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_item",
      method: "POST",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseItemsService.UnlinkItemFromCategories",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseItemsServiceUrl({
        protoPath: "/v1/enterprise/items/{itemId}/unlink-categories",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __unlinkItemFromCategories;
}
function overwriteItemCategories(payload) {
  function __overwriteItemCategories({ host }) {
    const metadata = {
      entityFqdn: "wix.media.enterprise_public_media.v1.enterprise_item",
      method: "POST",
      methodFqn: "com.wix.media.enterprise_public_media.v1.EnterpriseItemsService.OverwriteItemCategories",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaEnterprisePublicMediaV1EnterpriseItemsServiceUrl({
        protoPath: "/v1/enterprise/items/{itemId}/overwrite-categories",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __overwriteItemCategories;
}

// src/media-enterprise-public-media-v1-enterprise-item-enterprise-media-items.universal.ts
import { transformSDKAudioToRESTAudio } from "@wix/sdk-runtime/transformations/audio";
import { transformRESTAudioToSDKAudio } from "@wix/sdk-runtime/transformations/audio";
import { transformSDKImageToRESTImage } from "@wix/sdk-runtime/transformations/image";
import { transformRESTImageToSDKImage } from "@wix/sdk-runtime/transformations/image";
import { transformSDKVideoV2ToRESTVideoV2 } from "@wix/sdk-runtime/transformations/video-v2";
import { transformRESTVideoV2ToSDKVideoV2 } from "@wix/sdk-runtime/transformations/video-v2";
import { transformSDKDocumentToRESTDocument } from "@wix/sdk-runtime/transformations/document";
import { transformRESTDocumentToSDKDocument } from "@wix/sdk-runtime/transformations/document";
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var MediaType = /* @__PURE__ */ ((MediaType2) => {
  MediaType2["MIXED"] = "MIXED";
  MediaType2["IMAGE"] = "IMAGE";
  MediaType2["VIDEO"] = "VIDEO";
  MediaType2["AUDIO"] = "AUDIO";
  MediaType2["DOCUMENT"] = "DOCUMENT";
  MediaType2["VECTOR"] = "VECTOR";
  MediaType2["ARCHIVE"] = "ARCHIVE";
  MediaType2["MODEL3D"] = "MODEL3D";
  return MediaType2;
})(MediaType || {});
var PublishStatus = /* @__PURE__ */ ((PublishStatus2) => {
  PublishStatus2["UNDEFINED"] = "UNDEFINED";
  PublishStatus2["UNPUBLISHED"] = "UNPUBLISHED";
  PublishStatus2["PUBLISHED"] = "PUBLISHED";
  PublishStatus2["WIX_ONLY"] = "WIX_ONLY";
  return PublishStatus2;
})(PublishStatus || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function itemUploadCallback2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    itemId: options?.itemId,
    callbackToken: options?.callbackToken
  });
  const reqOpts = itemUploadCallback(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          itemId: "$[0].itemId",
          callbackToken: "$[0].callbackToken"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function generateFileUploadUrl2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    fileName: options?.fileName,
    contentType: options?.contentType,
    sizeInBytes: options?.sizeInBytes,
    categoryIds: options?.categoryIds
  });
  const reqOpts = generateFileUploadUrl(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          fileName: "$[0].fileName",
          contentType: "$[0].contentType",
          sizeInBytes: "$[0].sizeInBytes",
          categoryIds: "$[0].categoryIds"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function importFile2(url, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    url,
    fileName: options?.fileName,
    contentType: options?.contentType,
    sizeInBytes: options?.sizeInBytes,
    categoryIds: options?.categoryIds,
    mediaType: options?.mediaType,
    uploaderSystemTag: options?.uploaderSystemTag,
    uploaderInfoSystemTag: options?.uploaderInfoSystemTag
  });
  const reqOpts = importFile(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [
            { path: "item.assets.image" },
            { path: "item.assets.vector" },
            { path: "item.assets.model3d.thumbnail" }
          ]
        },
        {
          transformFn: transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "item.assets.video" }]
        },
        {
          transformFn: transformRESTAudioToSDKAudio,
          paths: [{ path: "item.assets.audio" }]
        },
        {
          transformFn: transformRESTDocumentToSDKDocument,
          paths: [{ path: "item.assets.document" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          url: "$[0]",
          fileName: "$[1].fileName",
          contentType: "$[1].contentType",
          sizeInBytes: "$[1].sizeInBytes",
          categoryIds: "$[1].categoryIds",
          mediaType: "$[1].mediaType",
          uploaderSystemTag: "$[1].uploaderSystemTag",
          uploaderInfoSystemTag: "$[1].uploaderInfoSystemTag"
        },
        singleArgumentUnchanged: false
      },
      ["url", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkImportFiles2(importFileRequests, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    importFileRequests,
    returnEntity: options?.returnEntity,
    categoryIds: options?.categoryIds
  });
  const reqOpts = bulkImportFiles(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [
            { path: "results.item.assets.image" },
            { path: "results.item.assets.vector" },
            { path: "results.item.assets.model3d.thumbnail" }
          ]
        },
        {
          transformFn: transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "results.item.assets.video" }]
        },
        {
          transformFn: transformRESTAudioToSDKAudio,
          paths: [{ path: "results.item.assets.audio" }]
        },
        {
          transformFn: transformRESTDocumentToSDKDocument,
          paths: [{ path: "results.item.assets.document" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          importFileRequests: "$[0]",
          returnEntity: "$[1].returnEntity",
          categoryIds: "$[1].categoryIds"
        },
        singleArgumentUnchanged: false
      },
      ["importFileRequests", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function searchItems2(query) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ query });
  const reqOpts = searchItems(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [
            { path: "items.assets.image" },
            { path: "items.assets.vector" },
            { path: "items.assets.model3d.thumbnail" }
          ]
        },
        {
          transformFn: transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "items.assets.video" }]
        },
        {
          transformFn: transformRESTAudioToSDKAudio,
          paths: [{ path: "items.assets.audio" }]
        },
        {
          transformFn: transformRESTDocumentToSDKDocument,
          paths: [{ path: "items.assets.document" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["query"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryItems2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryItems(
        payload
      );
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [
          {
            transformFn: transformRESTImageToSDKImage,
            paths: [
              { path: "items.assets.image" },
              { path: "items.assets.vector" },
              { path: "items.assets.model3d.thumbnail" }
            ]
          },
          {
            transformFn: transformRESTVideoV2ToSDKVideoV2,
            paths: [{ path: "items.assets.video" }]
          },
          {
            transformFn: transformRESTAudioToSDKAudio,
            paths: [{ path: "items.assets.audio" }]
          },
          {
            transformFn: transformRESTDocumentToSDKDocument,
            paths: [{ path: "items.assets.document" }]
          }
        ])
      );
      return {
        items: transformedData?.items,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "OFFSET",
    transformationPaths: {}
  });
}
async function updateItem2(_id, item) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({ item: { ...item, id: _id } }),
    [
      {
        transformFn: transformSDKImageToRESTImage,
        paths: [
          { path: "item.assets.image" },
          { path: "item.assets.vector" },
          { path: "item.assets.model3d.thumbnail" }
        ]
      },
      {
        transformFn: transformSDKVideoV2ToRESTVideoV2,
        paths: [{ path: "item.assets.video" }]
      },
      {
        transformFn: transformSDKAudioToRESTAudio,
        paths: [{ path: "item.assets.audio" }]
      },
      {
        transformFn: transformSDKDocumentToRESTDocument,
        paths: [{ path: "item.assets.document" }]
      }
    ]
  );
  const reqOpts = updateItem(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [
            { path: "item.assets.image" },
            { path: "item.assets.vector" },
            { path: "item.assets.model3d.thumbnail" }
          ]
        },
        {
          transformFn: transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "item.assets.video" }]
        },
        {
          transformFn: transformRESTAudioToSDKAudio,
          paths: [{ path: "item.assets.audio" }]
        },
        {
          transformFn: transformRESTDocumentToSDKDocument,
          paths: [{ path: "item.assets.document" }]
        }
      ])
    )?.item;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { item: "$[1]" },
        explicitPathsToArguments: { "item.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "item"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateItem2(updateItemRequests, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({
      updateItemRequests,
      returnEntity: options?.returnEntity
    }),
    [
      {
        transformFn: transformSDKImageToRESTImage,
        paths: [
          { path: "updateItemRequests.item.assets.image" },
          { path: "updateItemRequests.item.assets.vector" },
          { path: "updateItemRequests.item.assets.model3d.thumbnail" }
        ]
      },
      {
        transformFn: transformSDKVideoV2ToRESTVideoV2,
        paths: [{ path: "updateItemRequests.item.assets.video" }]
      },
      {
        transformFn: transformSDKAudioToRESTAudio,
        paths: [{ path: "updateItemRequests.item.assets.audio" }]
      },
      {
        transformFn: transformSDKDocumentToRESTDocument,
        paths: [{ path: "updateItemRequests.item.assets.document" }]
      }
    ]
  );
  const reqOpts = bulkUpdateItem(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [
            { path: "results.item.assets.image" },
            { path: "results.item.assets.vector" },
            { path: "results.item.assets.model3d.thumbnail" }
          ]
        },
        {
          transformFn: transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "results.item.assets.video" }]
        },
        {
          transformFn: transformRESTAudioToSDKAudio,
          paths: [{ path: "results.item.assets.audio" }]
        },
        {
          transformFn: transformRESTDocumentToSDKDocument,
          paths: [{ path: "results.item.assets.document" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          updateItemRequests: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["updateItemRequests", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getItem2(itemId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ itemId });
  const reqOpts = getItem(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [
            { path: "item.assets.image" },
            { path: "item.assets.vector" },
            { path: "item.assets.model3d.thumbnail" }
          ]
        },
        {
          transformFn: transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "item.assets.video" }]
        },
        {
          transformFn: transformRESTAudioToSDKAudio,
          paths: [{ path: "item.assets.audio" }]
        },
        {
          transformFn: transformRESTDocumentToSDKDocument,
          paths: [{ path: "item.assets.document" }]
        }
      ])
    )?.item;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { itemId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["itemId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function linkItemToCategories2(itemId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    itemId,
    categoryIds: options?.categoryIds
  });
  const reqOpts = linkItemToCategories(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          itemId: "$[0]",
          categoryIds: "$[1].categoryIds"
        },
        singleArgumentUnchanged: false
      },
      ["itemId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function unlinkItemFromCategories2(itemId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    itemId,
    categoryIds: options?.categoryIds
  });
  const reqOpts = unlinkItemFromCategories(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          itemId: "$[0]",
          categoryIds: "$[1].categoryIds"
        },
        singleArgumentUnchanged: false
      },
      ["itemId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function overwriteItemCategories2(itemId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    itemId,
    categoryIds: options?.categoryIds
  });
  const reqOpts = overwriteItemCategories(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          itemId: "$[0]",
          categoryIds: "$[1].categoryIds"
        },
        singleArgumentUnchanged: false
      },
      ["itemId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/media-enterprise-public-media-v1-enterprise-item-enterprise-media-items.public.ts
function itemUploadCallback3(httpClient) {
  return (options) => itemUploadCallback2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function generateFileUploadUrl3(httpClient) {
  return (options) => generateFileUploadUrl2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function importFile3(httpClient) {
  return (url, options) => importFile2(
    url,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkImportFiles3(httpClient) {
  return (importFileRequests, options) => bulkImportFiles2(
    importFileRequests,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function searchItems3(httpClient) {
  return (query) => searchItems2(
    query,
    // @ts-ignore
    { httpClient }
  );
}
function queryItems3(httpClient) {
  return () => queryItems2(
    // @ts-ignore
    { httpClient }
  );
}
function updateItem3(httpClient) {
  return (_id, item) => updateItem2(
    _id,
    item,
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdateItem3(httpClient) {
  return (updateItemRequests, options) => bulkUpdateItem2(
    updateItemRequests,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function getItem3(httpClient) {
  return (itemId) => getItem2(
    itemId,
    // @ts-ignore
    { httpClient }
  );
}
function linkItemToCategories3(httpClient) {
  return (itemId, options) => linkItemToCategories2(
    itemId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function unlinkItemFromCategories3(httpClient) {
  return (itemId, options) => unlinkItemFromCategories2(
    itemId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function overwriteItemCategories3(httpClient) {
  return (itemId, options) => overwriteItemCategories2(
    itemId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
var onEnterpriseItemCreated = EventDefinition(
  "wix.media.enterprise_public_media.v1.enterprise_item_created",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.assets.archive.urlExpirationDate" },
          { path: "entity.assets.model3d.urlExpirationDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: transformRESTImageToSDKImage2,
        paths: [
          { path: "entity.assets.image" },
          { path: "entity.assets.vector" },
          { path: "entity.assets.model3d.thumbnail" }
        ]
      },
      {
        transformFn: transformRESTVideoV2ToSDKVideoV22,
        paths: [{ path: "entity.assets.video" }]
      },
      {
        transformFn: transformRESTAudioToSDKAudio2,
        paths: [{ path: "entity.assets.audio" }]
      },
      {
        transformFn: transformRESTDocumentToSDKDocument2,
        paths: [{ path: "entity.assets.document" }]
      }
    ])
  )
)();
var onEnterpriseItemItemCategoriesChanged = EventDefinition(
  "wix.media.enterprise_public_media.v1.enterprise_item_item_categories_changed",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "data.entity.createdDate" },
          { path: "data.entity.updatedDate" },
          { path: "data.entity.assets.archive.urlExpirationDate" },
          { path: "data.entity.assets.model3d.urlExpirationDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: transformRESTImageToSDKImage2,
        paths: [
          { path: "data.entity.assets.image" },
          { path: "data.entity.assets.vector" },
          { path: "data.entity.assets.model3d.thumbnail" }
        ]
      },
      {
        transformFn: transformRESTVideoV2ToSDKVideoV22,
        paths: [{ path: "data.entity.assets.video" }]
      },
      {
        transformFn: transformRESTAudioToSDKAudio2,
        paths: [{ path: "data.entity.assets.audio" }]
      },
      {
        transformFn: transformRESTDocumentToSDKDocument2,
        paths: [{ path: "data.entity.assets.document" }]
      }
    ])
  )
)();
var onEnterpriseItemPublishStatusChanged = EventDefinition(
  "wix.media.enterprise_public_media.v1.enterprise_item_publish_status_changed",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "data.entity.createdDate" },
          { path: "data.entity.updatedDate" },
          { path: "data.entity.assets.archive.urlExpirationDate" },
          { path: "data.entity.assets.model3d.urlExpirationDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: transformRESTImageToSDKImage2,
        paths: [
          { path: "data.entity.assets.image" },
          { path: "data.entity.assets.vector" },
          { path: "data.entity.assets.model3d.thumbnail" }
        ]
      },
      {
        transformFn: transformRESTVideoV2ToSDKVideoV22,
        paths: [{ path: "data.entity.assets.video" }]
      },
      {
        transformFn: transformRESTAudioToSDKAudio2,
        paths: [{ path: "data.entity.assets.audio" }]
      },
      {
        transformFn: transformRESTDocumentToSDKDocument2,
        paths: [{ path: "data.entity.assets.document" }]
      }
    ])
  )
)();
var onEnterpriseItemUpdated = EventDefinition(
  "wix.media.enterprise_public_media.v1.enterprise_item_updated",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.assets.archive.urlExpirationDate" },
          { path: "entity.assets.model3d.urlExpirationDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: transformRESTImageToSDKImage2,
        paths: [
          { path: "entity.assets.image" },
          { path: "entity.assets.vector" },
          { path: "entity.assets.model3d.thumbnail" }
        ]
      },
      {
        transformFn: transformRESTVideoV2ToSDKVideoV22,
        paths: [{ path: "entity.assets.video" }]
      },
      {
        transformFn: transformRESTAudioToSDKAudio2,
        paths: [{ path: "entity.assets.audio" }]
      },
      {
        transformFn: transformRESTDocumentToSDKDocument2,
        paths: [{ path: "entity.assets.document" }]
      }
    ])
  )
)();

// src/media-enterprise-public-media-v1-enterprise-item-enterprise-media-items.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
import { createEventModule } from "@wix/sdk-runtime/event-definition-modules";
var itemUploadCallback4 = /* @__PURE__ */ createRESTModule(itemUploadCallback3);
var generateFileUploadUrl4 = /* @__PURE__ */ createRESTModule(generateFileUploadUrl3);
var importFile4 = /* @__PURE__ */ createRESTModule(importFile3);
var bulkImportFiles4 = /* @__PURE__ */ createRESTModule(bulkImportFiles3);
var searchItems4 = /* @__PURE__ */ createRESTModule(searchItems3);
var queryItems4 = /* @__PURE__ */ createRESTModule(queryItems3);
var updateItem4 = /* @__PURE__ */ createRESTModule(updateItem3);
var bulkUpdateItem4 = /* @__PURE__ */ createRESTModule(bulkUpdateItem3);
var getItem4 = /* @__PURE__ */ createRESTModule(getItem3);
var linkItemToCategories4 = /* @__PURE__ */ createRESTModule(linkItemToCategories3);
var unlinkItemFromCategories4 = /* @__PURE__ */ createRESTModule(unlinkItemFromCategories3);
var overwriteItemCategories4 = /* @__PURE__ */ createRESTModule(overwriteItemCategories3);
var onEnterpriseItemCreated2 = createEventModule(
  onEnterpriseItemCreated
);
var onEnterpriseItemItemCategoriesChanged2 = createEventModule(
  onEnterpriseItemItemCategoriesChanged
);
var onEnterpriseItemPublishStatusChanged2 = createEventModule(
  onEnterpriseItemPublishStatusChanged
);
var onEnterpriseItemUpdated2 = createEventModule(
  onEnterpriseItemUpdated
);
export {
  MediaType,
  PublishStatus,
  SortOrder,
  WebhookIdentityType,
  bulkImportFiles4 as bulkImportFiles,
  bulkUpdateItem4 as bulkUpdateItem,
  generateFileUploadUrl4 as generateFileUploadUrl,
  getItem4 as getItem,
  importFile4 as importFile,
  itemUploadCallback4 as itemUploadCallback,
  linkItemToCategories4 as linkItemToCategories,
  onEnterpriseItemCreated2 as onEnterpriseItemCreated,
  onEnterpriseItemItemCategoriesChanged2 as onEnterpriseItemItemCategoriesChanged,
  onEnterpriseItemPublishStatusChanged2 as onEnterpriseItemPublishStatusChanged,
  onEnterpriseItemUpdated2 as onEnterpriseItemUpdated,
  overwriteItemCategories4 as overwriteItemCategories,
  queryItems4 as queryItems,
  searchItems4 as searchItems,
  unlinkItemFromCategories4 as unlinkItemFromCategories,
  updateItem4 as updateItem
};
