// @wix/patterns/dist/cjs/components/Tags/BulkAssignTagsPopover/AsyncJobProgressModal.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports._AsyncJobProgressModal = exports.AsyncJobProgressModal = void 0;
var _mobxReactLite = require("mobx-react-lite");
var _react = _interopRequireDefault(require("react"));
var _designSystem = require("@wix/design-system");
var _react2 = require("@wix/bex-core/react");
var _ActionModal = require("../../ActionModal");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/Tags/BulkAssignTagsPopover/AsyncJobProgressModal.tsx";
const _AsyncJobProgressModal = ({
  state
}) => {
  const {
    asyncJobProgressModalState,
    closeAsyncJobProgressModal,
    updatedItemsPercentage
  } = state.bulkAssignAsyncState;
  const {
    translate: t
  } = (0, _react2.useWixPatternsContainer)();
  return /*#__PURE__*/_react.default.createElement(_ActionModal.CustomActionModal, {
    state: asyncJobProgressModalState,
    title: t('cairo.tags.bulkItemUpdate.updatingModal.title'),
    primaryButtonText: t('cairo.tags.bulkItemUpdate.updatingModal.CTA'),
    primaryButtonOnClick: closeAsyncJobProgressModal,
    onCloseButtonClick: closeAsyncJobProgressModal,
    dataHook: "async-job-progress-modal-layout",
    content: /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
      direction: "vertical",
      width: "552px",
      gap: "SP2",
      paddingTop: "SP1",
      paddingBottom: "SP1",
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 31,
        columnNumber: 9
      }
    }, /*#__PURE__*/_react.default.createElement(_designSystem.LinearProgressBar, {
      dataHook: "async-job-progress-modal-progress-bar",
      value: updatedItemsPercentage,
      showProgressIndication: true,
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 38,
        columnNumber: 11
      }
    }), /*#__PURE__*/_react.default.createElement(_designSystem.Text, {
      dataHook: "async-job-progress-modal-body-text",
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 43,
        columnNumber: 11
      }
    }, t('cairo.tags.bulkItemUpdate.updatingModal.body'))),
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 23,
      columnNumber: 5
    }
  });
};
exports._AsyncJobProgressModal = _AsyncJobProgressModal;
const AsyncJobProgressModal = exports.AsyncJobProgressModal = (0, _mobxReactLite.observer)(_AsyncJobProgressModal);
