// @wix/patterns/dist/cjs/components/Tags/BulkAssignTagsPopover/BulkAssignTagsPopover.uni.driver.js
"use strict";

exports.__esModule = true;
exports.BulkAssignTagsPopoverUniDriver = BulkAssignTagsPopoverUniDriver;
var _unidriver = require("../../../unidriver");
var _unidriver2 = require("@wix/design-system/dist/testkit/unidriver");
var _ManageTagsModalUni = require("../ManageTags/ManageTagsModal/ManageTagsModal.uni.driver");
var _AsyncJobProgressModalUni = require("./AsyncJobProgressModal.uni.driver");
function BulkAssignTagsPopoverUniDriver(base, body) {
  const popoverBase = () => (0, _unidriver2.PopoverUniDriver)(base.$('[data-hook="bulk-assign-tags-popover"]'), body);
  const getContentElement = async () => popoverBase().getContentElementUniDriver();
  const getInnerPopoverDriver = async (dataHook, driverConstructor) => {
    const contentElement = await getContentElement();
    return driverConstructor(contentElement.$(`[data-hook=${dataHook}]`), body);
  };
  const getTagsListDriver = async () => getInnerPopoverDriver('bulk-assign-tags-selector-list', _unidriver2.SelectorListUniDriver);
  const getCloseButtonDriver = async () => getInnerPopoverDriver('bulk-assign-tags-close-button', _unidriver2.ButtonUniDriver);
  const getApplyButtonDriver = async () => getInnerPopoverDriver('bulk-assign-tags-apply-button', _unidriver2.ButtonUniDriver);
  const getCreateTagButtonDriver = async () => getInnerPopoverDriver('bulk-assign-tags-create-tag-button', _unidriver2.ButtonUniDriver);
  const getManageTagsButtonDriver = async () => getInnerPopoverDriver('bulk-assign-tags-manage-tags-button', _unidriver2.TextButtonUniDriver);
  const getEmptyStateDriver = async () => getInnerPopoverDriver('bulk-assign-tags-empty-state', _unidriver2.EmptyStateUniDriver);
  const getErrorStateDriver = async () => getInnerPopoverDriver('bulk-assign-tags-error-state', _unidriver2.EmptyStateUniDriver);
  const getErrorStateRetryButtonDriver = async () => getInnerPopoverDriver('bulk-assign-tags-error-state-retry-button', _unidriver2.ButtonUniDriver);
  const getLoaderDriver = async () => getInnerPopoverDriver('bulk-assign-tags-loading-state', _unidriver2.LoaderUniDriver);
  const enterSearchValue = async value => {
    const contentElement = await getContentElement();
    const tagsListDriver = await getTagsListDriver();
    const isEmptyState = await tagsListDriver.showsEmptyState();
    if (isEmptyState) {
      await (0, _unidriver2.SearchUniDriver)(contentElement.$('[data-hook="bulk-assign-tags-empty-state-search"]'), body).inputDriver.enterText(value);
    } else {
      await tagsListDriver.enterSearchValue(value);
    }
  };
  const getTagDriverAt = async idx => getInnerPopoverDriver(`bulk-assign-tags-tag-${idx}`, _unidriver2.TagUniDriver);
  return {
    ...(0, _unidriver.baseUniDriverFactory)(base),
    asyncJobProgressModal: () => (0, _AsyncJobProgressModalUni.AsyncJobProgressModalUniDriver)(base, body),
    popoverExists: () => popoverBase().exists(),
    clickOutside: () => popoverBase().clickOutside(),
    tagsSelectorlistExist: async () => (await getTagsListDriver()).exists(),
    enterSearchValue,
    clickSearchInputClear: async () => (await getTagsListDriver()).clickSearchInputClear(),
    isSelectorCheckedAt: async idx => (await getTagsListDriver()).isSelectorCheckedAt(idx),
    isSelectorIsIndeterminateAt: async idx => (await getTagsListDriver()).isSelectorIsIndeterminateAt(idx),
    tagsListExists: async () => (await getTagsListDriver()).listExists(),
    numberOfItemsInList: async () => (await getTagsListDriver()).numberOfItemsInList(),
    getTagLabelAt: async idx => (await getTagDriverAt(idx)).getLabel(),
    toggleSelectorAt: async idx => (await getTagsListDriver()).toggleSelectorAt(idx),
    closeButtonExists: async () => (await getCloseButtonDriver()).exists(),
    clickCloseButton: async () => (await getCloseButtonDriver()).click(),
    applyButtonExists: async () => (await getApplyButtonDriver()).exists(),
    clickApplyButton: async () => (await getApplyButtonDriver()).click(),
    createTagButtonExists: async () => (await getCreateTagButtonDriver()).exists(),
    clickCreateTagButton: async () => (await getCreateTagButtonDriver()).click(),
    manageTagsButtonExists: async () => (await getManageTagsButtonDriver()).exists(),
    clickManageTagsButton: async () => (await getManageTagsButtonDriver()).click(),
    manageTagsModal: () => (0, _ManageTagsModalUni.ManageTagsModalUniDriver)(base, body),
    emptyStateExists: async () => (await getEmptyStateDriver()).exists(),
    errorStateExists: async () => (await getErrorStateDriver()).exists(),
    errorStateRetryButtonExists: async () => (await getErrorStateRetryButtonDriver()).exists(),
    clickErrorStateRetryButton: async () => (await getErrorStateRetryButtonDriver()).click(),
    isLoading: async () => (await getLoaderDriver()).exists(),
    isContentElementExists: () => popoverBase().isContentElementExists(),
    getContentElement: () => popoverBase().getContentElement()
  };
}
