// @wix/patterns/dist/cjs/components/Tags/BulkAssignTagsPopover/CreateTagButton.js
"use strict";

exports.__esModule = true;
exports.CreateTagButton = void 0;
var _react = _interopRequireWildcard(require("react"));
var _designSystem = require("@wix/design-system");
var _wixUiIconsCommon = require("@wix/wix-ui-icons-common");
var _react2 = require("@wix/bex-core/react");
var _bexCore = require("@wix/bex-core");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/Tags/BulkAssignTagsPopover/CreateTagButton.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
const CreateTagButton = ({
  state,
  searchValue
}) => {
  const {
    translate: t
  } = (0, _react2.useWixPatternsContainer)();
  const createAndAssignTag = (0, _react.useCallback)(tagName => {
    state.createAndAssignTag(tagName);
  }, [state]);
  return /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
    height: "18px",
    width: "100%",
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 26,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.TextButton, {
    dataHook: "bulk-assign-tags-create-tag-button",
    prefixIcon: /*#__PURE__*/_react.default.createElement(_wixUiIconsCommon.Add, {
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 29,
        columnNumber: 21
      }
    }),
    onClick: () => createAndAssignTag(searchValue),
    size: "small",
    ellipsis: true,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 27,
      columnNumber: 7
    }
  }, t('cairo.tags.collectionPage.popover.create.CTA', {
    tagName: searchValue.slice(0, _bexCore.TAG_MAX_LENGTH)
  })));
};
exports.CreateTagButton = CreateTagButton;
