// @wix/patterns/dist/cjs/fields/Field/Field.uni.driver.js
"use strict";

exports.__esModule = true;
exports.default = void 0;
var _unidriver = require("../../unidriver");
var _unidriver2 = require("@wix/design-system/dist/testkit/unidriver");
var _dataHooks = require("../dataHooks");
var _default = (base, body) => {
  const fieldType = () => base.attr('data-field-type');
  const inputBase = (suffix = '') => base.$(`[data-hook="${_dataHooks.dataHooks.Input}${suffix}"]`);
  const dropdown = () => (0, _unidriver2.DropdownUniDriver)(inputBase(), body);
  const checkbox = () => (0, _unidriver2.CheckboxUniDriver)(inputBase(), body);
  const formField = () => (0, _unidriver2.FormFieldUniDriver)(base.$(`[data-hook="${_dataHooks.dataHooks.Label}"]`), body);
  return {
    /**
     * Gets field label
     */
    getLabel: async () => {
      const type = await fieldType();
      if (type === 'checkbox') {
        return checkbox().getLabel();
      }
      const element = await formField().getLabel();
      return element == null ? void 0 : element.textContent;
    },
    /**
     * Gets field value
     */
    getValue: async () => {
      const type = await fieldType();
      switch (type) {
        case 'url':
        case 'shortText':
          return (0, _unidriver2.InputUniDriver)(inputBase(), body).getValue();
        case 'date':
          return (0, _unidriver2.DatePickerUniDriver)(inputBase(), body).inputDriver.getValue();
        case 'dateTime':
          const date = await (0, _unidriver2.DatePickerUniDriver)(inputBase('-date'), body).inputDriver.getValue();
          const format = Intl.DateTimeFormat([], {
            hour: 'numeric',
            minute: 'numeric'
          });
          const time = await (0, _unidriver2.TimeInputUniDriver)(inputBase('-time'), body).getValue();
          if (isNaN(time.getTime())) {
            return date;
          }
          return `${date} ${format.format(time)}`;
        case 'checkbox':
          return checkbox().isChecked();
        case 'longText':
          return (0, _unidriver2.InputAreaUniDriver)(inputBase(), body).getValue();
        case 'dropdown':
          return dropdown().inputDriver.getValue();
        case 'integer':
        case 'decimal':
          return (0, _unidriver2.NumberInputUniDriver)(inputBase(), body).getValue();
        default:
          throw new Error(`Driver for type '${type}' not found.`);
      }
    },
    /**
     * Gets field status message
     */
    getStatusMessage: async () => {
      const type = await fieldType();
      switch (type) {
        case 'shortText':
        case 'url':
          return (0, _unidriver2.FormFieldUniDriver)(base, body).getStatusMessage();
        case 'date':
          return (0, _unidriver2.DatePickerUniDriver)(inputBase(), body).inputDriver.getStatusMessage();
        case 'dateTime':
          let dateStatusMessage;
          let timeStatusMessage;
          try {
            dateStatusMessage = await (0, _unidriver2.DatePickerUniDriver)(inputBase('-date'), body).inputDriver.getStatusMessage();
          } catch {}
          try {
            timeStatusMessage = await (0, _unidriver2.TimeInputUniDriver)(inputBase('-time'), body).getStatusMessage();
          } catch {}
          return [dateStatusMessage, timeStatusMessage].join('\n').trim();
        case 'longText':
          return (0, _unidriver2.InputAreaUniDriver)(inputBase(), body).getStatusMessage();
        case 'dropdown':
          return dropdown().inputDriver.getStatusMessage();
        case 'integer':
        case 'decimal':
          return (0, _unidriver2.NumberInputUniDriver)(inputBase(), body).getStatusMessage();
        default:
          throw new Error(`Driver for type '${type}' not found.`);
      }
    },
    /**
     * Enters field value, handles different value types based on field type
     */
    enterValue: async (value, {
      locale = 'en'
    } = {}) => {
      var _element$page;
      const type = await fieldType();
      switch (type) {
        case 'url':
          const inputDriver = (0, _unidriver2.InputUniDriver)(inputBase(), body);
          await inputDriver.enterText(value);
          const {
            page
          } = await inputBase().getNative();
          // jsdom
          if (!(page != null && page.evaluate)) {
            await inputDriver.blur();
          }
          return;
        case 'shortText':
          return (0, _unidriver2.InputUniDriver)(inputBase(), body).enterText(value);
        case 'date':
          const datePickerUniDriver = (0, _unidriver2.DatePickerUniDriver)(inputBase(), body);
          await datePickerUniDriver.inputDriver.enterText(typeof value === 'string' ? value : `${Intl.DateTimeFormat().format(value)}`);
          const {
            page: pageWithDate
          } = await inputBase().getNative();
          // jsdom
          if (!(pageWithDate != null && pageWithDate.evaluate)) {
            await datePickerUniDriver.inputDriver.blur();
          }
          return;
        case 'dateTime':
          let dateValue = '';
          let timeValue = '';
          if (typeof value === 'string') {
            const parts = value.split('T');
            dateValue = parts[0];
            timeValue = parts[1];
          } else {
            dateValue = `${Intl.DateTimeFormat([locale]).format(value)}`;
            timeValue = Intl.DateTimeFormat([locale], {
              hour: 'numeric',
              minute: 'numeric'
            }).format(value);
          }
          const datetimePickerUniDriver = (0, _unidriver2.DatePickerUniDriver)(inputBase('-date'), body);
          await datetimePickerUniDriver.inputDriver.enterText(dateValue);
          const {
            page: pageWithDatetime
          } = await inputBase('-date').getNative();
          // jsdom
          if (!(pageWithDatetime != null && pageWithDatetime.evaluate)) {
            // We do it twice because we have a bug when removing the value in first time
            await datetimePickerUniDriver.inputDriver.enterText(dateValue);
            await datetimePickerUniDriver.inputDriver.blur();
          }
          const timeInput = (0, _unidriver2.TimeInputUniDriver)(inputBase('-time'), body);
          await timeInput.setValue(timeValue);
          const element = await inputBase('-time').getNative();
          if (!((_element$page = element.page) != null && _element$page.evaluate)) {
            await timeInput.confirmValue();
          }
          return;
        case 'checkbox':
          const checkboxDriver = checkbox();
          const isChecked = await checkboxDriver.isChecked();
          if (value && !isChecked) {
            await checkboxDriver.click();
          } else if (isChecked && !value) {
            await checkboxDriver.click();
          }
          return;
        case 'longText':
          return (0, _unidriver2.InputAreaUniDriver)(inputBase(), body).enterText(value);
        case 'dropdown':
          const dropdownDriver = dropdown();
          await dropdownDriver.dropdownLayoutDriver.click();
          return dropdownDriver.dropdownLayoutDriver.clickAtOptionWithValue(value);
        case 'integer':
        case 'decimal':
          return (0, _unidriver2.NumberInputUniDriver)(inputBase(), body).enterText(value);
        default:
          throw new Error(`Driver for type '${type}' not found.`);
      }
    },
    /**
     * Returns available dropdown options
     */
    getOptions: async () => {
      const type = await fieldType();
      if (type !== 'dropdown') {
        throw new Error(`Field type '${type}' does not support this method.`);
      }
      return dropdown().dropdownLayoutDriver.optionsContent();
    },
    /**
     * Returns remaining chars count on fields with maxLength
     */
    getLengthLeft: async () => {
      return formField().getLengthLeft();
    },
    ...(0, _unidriver.baseUniDriverFactory)(base)
  };
};
exports.default = _default;
