// @wix/auto_sdk_online-programs_sections/build/es/index.typings.mjs
// src/online-programs-v3-section-sections.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/online-programs-v3-section-sections.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKFloatToRESTFloat } from "@wix/sdk-runtime/transformations/float";
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixOnlineprogramsSectionsV3SectionsServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/program-sections-service",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/program-sections-service",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/program-sections-service",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/program-sections-service",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/program-sections-service",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/program-sections-service",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/program-sections-service",
        destPath: ""
      },
      {
        srcPath: "/program-sections-service",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/program-sections-service",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/program-sections-service",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/program-sections-service",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/program-sections-service",
        destPath: ""
      },
      {
        srcPath: "/program-sections-service",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/program-sections-service",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/program-sections-service",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_online-programs_sections";
function createSection(payload) {
  function __createSection({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "section.createdDate" },
          { path: "section.updatedDate" },
          { path: "section.description.image.urlExpirationDate" },
          { path: "section.description.video.urlExpirationDate" },
          { path: "section.description.video.resolutions.urlExpirationDate" },
          {
            path: "section.description.video.resolutions.poster.urlExpirationDate"
          },
          { path: "section.description.video.posters.urlExpirationDate" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "section.ordering" },
          { path: "section.description.image.focalPoint.x" },
          { path: "section.description.image.focalPoint.y" },
          { path: "section.description.video.resolutions.poster.focalPoint.x" },
          { path: "section.description.video.resolutions.poster.focalPoint.y" },
          { path: "section.description.video.posters.focalPoint.x" },
          { path: "section.description.video.posters.focalPoint.y" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "POST",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.CreateSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/sections",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "section.createdDate" },
            { path: "section.updatedDate" },
            { path: "section.description.image.urlExpirationDate" },
            { path: "section.description.video.urlExpirationDate" },
            {
              path: "section.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "section.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "section.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "section.ordering" },
            { path: "section.description.image.focalPoint.x" },
            { path: "section.description.image.focalPoint.y" },
            {
              path: "section.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "section.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "section.description.video.posters.focalPoint.x" },
            { path: "section.description.video.posters.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createSection;
}
function bulkCreateSection(payload) {
  function __bulkCreateSection({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "sections.createdDate" },
          { path: "sections.updatedDate" },
          { path: "sections.description.image.urlExpirationDate" },
          { path: "sections.description.video.urlExpirationDate" },
          { path: "sections.description.video.resolutions.urlExpirationDate" },
          {
            path: "sections.description.video.resolutions.poster.urlExpirationDate"
          },
          { path: "sections.description.video.posters.urlExpirationDate" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "sections.ordering" },
          { path: "sections.description.image.focalPoint.x" },
          { path: "sections.description.image.focalPoint.y" },
          {
            path: "sections.description.video.resolutions.poster.focalPoint.x"
          },
          {
            path: "sections.description.video.resolutions.poster.focalPoint.y"
          },
          { path: "sections.description.video.posters.focalPoint.x" },
          { path: "sections.description.video.posters.focalPoint.y" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "POST",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.BulkCreateSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/bulk/sections/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.description.image.urlExpirationDate" },
            { path: "results.item.description.video.urlExpirationDate" },
            {
              path: "results.item.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.description.video.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.description.video.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.ordering" },
            { path: "results.item.description.image.focalPoint.x" },
            { path: "results.item.description.image.focalPoint.y" },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "results.item.description.video.posters.focalPoint.x" },
            { path: "results.item.description.video.posters.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateSection;
}
function bulkCreateSectionMigration(payload) {
  function __bulkCreateSectionMigration({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "sections.createdDate" },
          { path: "sections.updatedDate" },
          { path: "sections.description.image.urlExpirationDate" },
          { path: "sections.description.video.urlExpirationDate" },
          { path: "sections.description.video.resolutions.urlExpirationDate" },
          {
            path: "sections.description.video.resolutions.poster.urlExpirationDate"
          },
          { path: "sections.description.video.posters.urlExpirationDate" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "sections.ordering" },
          { path: "sections.description.image.focalPoint.x" },
          { path: "sections.description.image.focalPoint.y" },
          {
            path: "sections.description.video.resolutions.poster.focalPoint.x"
          },
          {
            path: "sections.description.video.resolutions.poster.focalPoint.y"
          },
          { path: "sections.description.video.posters.focalPoint.x" },
          { path: "sections.description.video.posters.focalPoint.y" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "POST",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.BulkCreateSectionMigration",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/bulk/sections/create/migration",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.description.image.urlExpirationDate" },
            { path: "results.item.description.video.urlExpirationDate" },
            {
              path: "results.item.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.description.video.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.description.video.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.ordering" },
            { path: "results.item.description.image.focalPoint.x" },
            { path: "results.item.description.image.focalPoint.y" },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "results.item.description.video.posters.focalPoint.x" },
            { path: "results.item.description.video.posters.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateSectionMigration;
}
function getSection(payload) {
  function __getSection({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "GET",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.GetSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/sections/{sectionId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "section.createdDate" },
            { path: "section.updatedDate" },
            { path: "section.description.image.urlExpirationDate" },
            { path: "section.description.video.urlExpirationDate" },
            {
              path: "section.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "section.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "section.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "section.ordering" },
            { path: "section.description.image.focalPoint.x" },
            { path: "section.description.image.focalPoint.y" },
            {
              path: "section.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "section.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "section.description.video.posters.focalPoint.x" },
            { path: "section.description.video.posters.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getSection;
}
function updateSection(payload) {
  function __updateSection({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "section.createdDate" },
          { path: "section.updatedDate" },
          { path: "section.description.image.urlExpirationDate" },
          { path: "section.description.video.urlExpirationDate" },
          { path: "section.description.video.resolutions.urlExpirationDate" },
          {
            path: "section.description.video.resolutions.poster.urlExpirationDate"
          },
          { path: "section.description.video.posters.urlExpirationDate" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "section.ordering" },
          { path: "section.description.image.focalPoint.x" },
          { path: "section.description.image.focalPoint.y" },
          { path: "section.description.video.resolutions.poster.focalPoint.x" },
          { path: "section.description.video.resolutions.poster.focalPoint.y" },
          { path: "section.description.video.posters.focalPoint.x" },
          { path: "section.description.video.posters.focalPoint.y" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "PATCH",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.UpdateSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/sections/{section.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "section.createdDate" },
            { path: "section.updatedDate" },
            { path: "section.description.image.urlExpirationDate" },
            { path: "section.description.video.urlExpirationDate" },
            {
              path: "section.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "section.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "section.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "section.ordering" },
            { path: "section.description.image.focalPoint.x" },
            { path: "section.description.image.focalPoint.y" },
            {
              path: "section.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "section.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "section.description.video.posters.focalPoint.x" },
            { path: "section.description.video.posters.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateSection;
}
function deleteSection(payload) {
  function __deleteSection({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "DELETE",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.DeleteSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/sections/{sectionId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteSection;
}
function querySections(payload) {
  function __querySections({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "POST",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.QuerySections",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/sections/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "sections.createdDate" },
            { path: "sections.updatedDate" },
            { path: "sections.description.image.urlExpirationDate" },
            { path: "sections.description.video.urlExpirationDate" },
            {
              path: "sections.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "sections.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "sections.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "sections.ordering" },
            { path: "sections.description.image.focalPoint.x" },
            { path: "sections.description.image.focalPoint.y" },
            {
              path: "sections.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "sections.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "sections.description.video.posters.focalPoint.x" },
            { path: "sections.description.video.posters.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __querySections;
}
function listSections(payload) {
  function __listSections({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "GET",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.ListSections",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/sections",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "sections.createdDate" },
            { path: "sections.updatedDate" },
            { path: "sections.description.image.urlExpirationDate" },
            { path: "sections.description.video.urlExpirationDate" },
            {
              path: "sections.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "sections.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "sections.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "sections.ordering" },
            { path: "sections.description.image.focalPoint.x" },
            { path: "sections.description.image.focalPoint.y" },
            {
              path: "sections.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "sections.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "sections.description.video.posters.focalPoint.x" },
            { path: "sections.description.video.posters.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listSections;
}
function cloneSection(payload) {
  function __cloneSection({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "POST",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.CloneSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/sections/{sectionId}/clone",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "section.createdDate" },
            { path: "section.updatedDate" },
            { path: "section.description.image.urlExpirationDate" },
            { path: "section.description.video.urlExpirationDate" },
            {
              path: "section.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "section.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "section.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "section.ordering" },
            { path: "section.description.image.focalPoint.x" },
            { path: "section.description.image.focalPoint.y" },
            {
              path: "section.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "section.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "section.description.video.posters.focalPoint.x" },
            { path: "section.description.video.posters.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __cloneSection;
}
function moveSection(payload) {
  function __moveSection({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "PATCH",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.MoveSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/sections/move",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "section.createdDate" },
            { path: "section.updatedDate" },
            { path: "section.description.image.urlExpirationDate" },
            { path: "section.description.video.urlExpirationDate" },
            {
              path: "section.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "section.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "section.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "section.ordering" },
            { path: "section.description.image.focalPoint.x" },
            { path: "section.description.image.focalPoint.y" },
            {
              path: "section.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "section.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "section.description.video.posters.focalPoint.x" },
            { path: "section.description.video.posters.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __moveSection;
}
function publishSection(payload) {
  function __publishSection({ host }) {
    const metadata = {
      entityFqdn: "wix.online_programs.v3.section",
      method: "PATCH",
      methodFqn: "wix.onlineprograms.sections.v3.SectionsService.PublishSection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixOnlineprogramsSectionsV3SectionsServiceUrl({
        protoPath: "/v3/sections/publish",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "section.createdDate" },
            { path: "section.updatedDate" },
            { path: "section.description.image.urlExpirationDate" },
            { path: "section.description.video.urlExpirationDate" },
            {
              path: "section.description.video.resolutions.urlExpirationDate"
            },
            {
              path: "section.description.video.resolutions.poster.urlExpirationDate"
            },
            { path: "section.description.video.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "section.ordering" },
            { path: "section.description.image.focalPoint.x" },
            { path: "section.description.image.focalPoint.y" },
            {
              path: "section.description.video.resolutions.poster.focalPoint.x"
            },
            {
              path: "section.description.video.resolutions.poster.focalPoint.y"
            },
            { path: "section.description.video.posters.focalPoint.x" },
            { path: "section.description.video.posters.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __publishSection;
}

// src/online-programs-v3-section-sections.universal.ts
import { transformSDKImageToRESTImage } from "@wix/sdk-runtime/transformations/image";
import { transformRESTImageToSDKImage } from "@wix/sdk-runtime/transformations/image";
import { transformSDKVideoV2ToRESTVideoV2 } from "@wix/sdk-runtime/transformations/video-v2";
import { transformRESTVideoV2ToSDKVideoV2 } from "@wix/sdk-runtime/transformations/video-v2";
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var State = /* @__PURE__ */ ((State2) => {
  State2["UNKNOWN_STATE"] = "UNKNOWN_STATE";
  State2["DRAFT"] = "DRAFT";
  State2["PUBLISHED"] = "PUBLISHED";
  return State2;
})(State || {});
var Enum = /* @__PURE__ */ ((Enum2) => {
  Enum2["UNKNOWN"] = "UNKNOWN";
  Enum2["PROTECTED"] = "PROTECTED";
  Enum2["PRIVATE"] = "PRIVATE";
  Enum2["PUBLIC"] = "PUBLIC";
  Enum2["UNCHANGED"] = "UNCHANGED";
  Enum2["GROUP_PROTECTED"] = "GROUP_PROTECTED";
  return Enum2;
})(Enum || {});
var RequestedFields = /* @__PURE__ */ ((RequestedFields2) => {
  RequestedFields2["UNKNOWN_REQUESTED_FIELD"] = "UNKNOWN_REQUESTED_FIELD";
  RequestedFields2["TOTAL_STEPS"] = "TOTAL_STEPS";
  return RequestedFields2;
})(RequestedFields || {});
var BulkActionType = /* @__PURE__ */ ((BulkActionType2) => {
  BulkActionType2["UNKNOWN_ACTION_TYPE"] = "UNKNOWN_ACTION_TYPE";
  BulkActionType2["INSERT"] = "INSERT";
  BulkActionType2["UPDATE"] = "UPDATE";
  BulkActionType2["DELETE"] = "DELETE";
  return BulkActionType2;
})(BulkActionType || {});
var DescriptionFieldSet = /* @__PURE__ */ ((DescriptionFieldSet2) => {
  DescriptionFieldSet2["STANDARD"] = "STANDARD";
  DescriptionFieldSet2["EXTENDED"] = "EXTENDED";
  return DescriptionFieldSet2;
})(DescriptionFieldSet || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var SearchIndexingNotificationState = /* @__PURE__ */ ((SearchIndexingNotificationState2) => {
  SearchIndexingNotificationState2["Unknown"] = "Unknown";
  SearchIndexingNotificationState2["Off"] = "Off";
  SearchIndexingNotificationState2["On"] = "On";
  return SearchIndexingNotificationState2;
})(SearchIndexingNotificationState || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createSection2(section, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({
      section,
      fields: options?.fields
    }),
    [
      {
        transformFn: transformSDKImageToRESTImage,
        paths: [{ path: "section.description.image" }]
      },
      {
        transformFn: transformSDKVideoV2ToRESTVideoV2,
        paths: [{ path: "section.description.video" }]
      }
    ]
  );
  const reqOpts = createSection(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [{ path: "section.description.image" }]
        },
        {
          transformFn: transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "section.description.video" }]
        }
      ])
    )?.section;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { section: "$[0]", fields: "$[1].fields" },
        singleArgumentUnchanged: false
      },
      ["section", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCreateSection2(sections, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({
      sections,
      fields: options?.fields
    }),
    [
      {
        transformFn: transformSDKImageToRESTImage,
        paths: [{ path: "sections.description.image" }]
      },
      {
        transformFn: transformSDKVideoV2ToRESTVideoV2,
        paths: [{ path: "sections.description.video" }]
      }
    ]
  );
  const reqOpts = bulkCreateSection(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [{ path: "results.item.description.image" }]
        },
        {
          transformFn: transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "results.item.description.video" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { sections: "$[0]", fields: "$[1].fields" },
        singleArgumentUnchanged: false
      },
      ["sections", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCreateSectionMigration2(sections) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({ sections }),
    [
      {
        transformFn: transformSDKImageToRESTImage,
        paths: [{ path: "sections.description.image" }]
      },
      {
        transformFn: transformSDKVideoV2ToRESTVideoV2,
        paths: [{ path: "sections.description.video" }]
      }
    ]
  );
  const reqOpts = bulkCreateSectionMigration(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [{ path: "results.item.description.image" }]
        },
        {
          transformFn: transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "results.item.description.video" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { sections: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["sections"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getSection2(sectionId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    sectionId,
    descriptionFieldSet: options?.descriptionFieldSet,
    fields: options?.fields
  });
  const reqOpts = getSection(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [{ path: "section.description.image" }]
        },
        {
          transformFn: transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "section.description.video" }]
        }
      ])
    )?.section;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          sectionId: "$[0]",
          descriptionFieldSet: "$[1].descriptionFieldSet",
          fields: "$[1].fields"
        },
        singleArgumentUnchanged: false
      },
      ["sectionId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateSection2(_id, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({
      section: { ...options?.section, id: _id },
      fields: options?.fields
    }),
    [
      {
        transformFn: transformSDKImageToRESTImage,
        paths: [{ path: "section.description.image" }]
      },
      {
        transformFn: transformSDKVideoV2ToRESTVideoV2,
        paths: [{ path: "section.description.video" }]
      }
    ]
  );
  const reqOpts = updateSection(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [{ path: "section.description.image" }]
        },
        {
          transformFn: transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "section.description.video" }]
        }
      ])
    )?.section;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { section: "$[1].section" },
        explicitPathsToArguments: {
          "section.id": "$[0]",
          fields: "$[1].fields"
        },
        singleArgumentUnchanged: false
      },
      ["_id", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteSection2(sectionId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    sectionId
  });
  const reqOpts = deleteSection(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { sectionId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["sectionId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function querySections2(options) {
  const { httpClient, sideEffects } = arguments[1];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = querySections({
        ...payload,
        ...options ?? {}
      });
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, options];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [
          {
            transformFn: transformRESTImageToSDKImage,
            paths: [{ path: "sections.description.image" }]
          },
          {
            transformFn: transformRESTVideoV2ToSDKVideoV2,
            paths: [{ path: "sections.description.video" }]
          }
        ])
      );
      return {
        items: transformedData?.sections,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function listSections2(programId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    programId,
    descriptionFieldSet: options?.descriptionFieldSet,
    fields: options?.fields
  });
  const reqOpts = listSections(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [{ path: "sections.description.image" }]
        },
        {
          transformFn: transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "sections.description.video" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          programId: "$[0]",
          descriptionFieldSet: "$[1].descriptionFieldSet",
          fields: "$[1].fields"
        },
        singleArgumentUnchanged: false
      },
      ["programId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function cloneSection2(sectionId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    sectionId,
    fields: options?.fields
  });
  const reqOpts = cloneSection(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [{ path: "section.description.image" }]
        },
        {
          transformFn: transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "section.description.video" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { sectionId: "$[0]", fields: "$[1].fields" },
        singleArgumentUnchanged: false
      },
      ["sectionId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function moveSection2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    sectionId: options?.sectionId,
    afterSectionId: options?.afterSectionId,
    revision: options?.revision,
    fields: options?.fields
  });
  const reqOpts = moveSection(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [{ path: "section.description.image" }]
        },
        {
          transformFn: transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "section.description.video" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          sectionId: "$[0].sectionId",
          afterSectionId: "$[0].afterSectionId",
          revision: "$[0].revision",
          fields: "$[0].fields"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function publishSection2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    sectionId: options?.sectionId,
    revision: options?.revision,
    fields: options?.fields
  });
  const reqOpts = publishSection(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [{ path: "section.description.image" }]
        },
        {
          transformFn: transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "section.description.video" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          sectionId: "$[0].sectionId",
          revision: "$[0].revision",
          fields: "$[0].fields"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
export {
  BulkActionType,
  DescriptionFieldSet,
  Enum,
  RequestedFields,
  SearchIndexingNotificationState,
  SortOrder,
  State,
  WebhookIdentityType,
  bulkCreateSection2 as bulkCreateSection,
  bulkCreateSectionMigration2 as bulkCreateSectionMigration,
  cloneSection2 as cloneSection,
  createSection2 as createSection,
  deleteSection2 as deleteSection,
  getSection2 as getSection,
  listSections2 as listSections,
  moveSection2 as moveSection,
  publishSection2 as publishSection,
  querySections2 as querySections,
  updateSection2 as updateSection
};
