// @wix/auto_sdk_restaurants_item-labels/build/es/index.mjs
// src/restaurants-menus-v1-item-label-item-labels.public.ts
import { renameKeysFromRESTResponseToSDKResponse as renameKeysFromRESTResponseToSDKResponse2 } from "@wix/sdk-runtime/rename-all-nested-keys";
import { transformRESTImageToSDKImage as transformRESTImageToSDKImage2 } from "@wix/sdk-runtime/transformations/image";
import { transformRESTTimestampToSDKTimestamp as transformRESTTimestampToSDKTimestamp2 } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths as transformPaths3 } from "@wix/sdk-runtime/transformations/transform-paths";
import { EventDefinition } from "@wix/sdk-types";

// src/restaurants-menus-v1-item-label-item-labels.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/restaurants-menus-v1-item-label-item-labels.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressRestaurantsMenusItemLabelsV1RestaurantsMenusItemLabelsUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-labels",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-menus-item-labels",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-labels",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-labels",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-menus-item-labels",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-menus-item-labels",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-menus-item-labels",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-labels",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-menus-item-labels",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/item-labels",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/labels",
        destPath: "/v1/labels"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-labels",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_item-labels";
function createLabel(payload) {
  function __createLabel({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "label.createdDate" },
          { path: "label.updatedDate" },
          { path: "label.icon.urlExpirationDate" },
          { path: "label.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_label",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_item_labels.v1.RestaurantsMenusItemLabels.CreateLabel",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemLabelsV1RestaurantsMenusItemLabelsUrl(
        { protoPath: "/v1/labels", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "label.createdDate" },
            { path: "label.updatedDate" },
            { path: "label.icon.urlExpirationDate" },
            { path: "label.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createLabel;
}
function getLabel(payload) {
  function __getLabel({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_label",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.menus_item_labels.v1.RestaurantsMenusItemLabels.GetLabel",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemLabelsV1RestaurantsMenusItemLabelsUrl(
        { protoPath: "/v1/labels/{labelId}", data: payload, host }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "label.createdDate" },
            { path: "label.updatedDate" },
            { path: "label.icon.urlExpirationDate" },
            { path: "label.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getLabel;
}
function listLabels(payload) {
  function __listLabels({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_label",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.menus_item_labels.v1.RestaurantsMenusItemLabels.ListLabels",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemLabelsV1RestaurantsMenusItemLabelsUrl(
        { protoPath: "/v1/labels", data: payload, host }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "labels.createdDate" },
            { path: "labels.updatedDate" },
            { path: "labels.icon.urlExpirationDate" },
            { path: "labels.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listLabels;
}
function queryLabels(payload) {
  function __queryLabels({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_label",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_item_labels.v1.RestaurantsMenusItemLabels.QueryLabels",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemLabelsV1RestaurantsMenusItemLabelsUrl(
        { protoPath: "/v1/labels/query", data: payload, host }
      ),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "labels.createdDate" },
            { path: "labels.updatedDate" },
            { path: "labels.icon.urlExpirationDate" },
            { path: "labels.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryLabels;
}
function countLabels(payload) {
  function __countLabels({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_label",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.menus_item_labels.v1.RestaurantsMenusItemLabels.CountLabels",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemLabelsV1RestaurantsMenusItemLabelsUrl(
        { protoPath: "/v1/labels/count", data: payload, host }
      ),
      data: payload
    };
    return metadata;
  }
  return __countLabels;
}
function updateLabel(payload) {
  function __updateLabel({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "label.createdDate" },
          { path: "label.updatedDate" },
          { path: "label.icon.urlExpirationDate" },
          { path: "label.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_label",
      method: "PATCH",
      methodFqn: "com.wixpress.restaurants.menus_item_labels.v1.RestaurantsMenusItemLabels.UpdateLabel",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemLabelsV1RestaurantsMenusItemLabelsUrl(
        { protoPath: "/v1/labels/{label.id}", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "label.createdDate" },
            { path: "label.updatedDate" },
            { path: "label.icon.urlExpirationDate" },
            { path: "label.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateLabel;
}
function deleteLabel(payload) {
  function __deleteLabel({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_label",
      method: "DELETE",
      methodFqn: "com.wixpress.restaurants.menus_item_labels.v1.RestaurantsMenusItemLabels.DeleteLabel",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemLabelsV1RestaurantsMenusItemLabelsUrl(
        { protoPath: "/v1/labels/{labelId}", data: payload, host }
      ),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteLabel;
}
function bulkDeleteLabels(payload) {
  function __bulkDeleteLabels({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_label",
      method: "DELETE",
      methodFqn: "com.wixpress.restaurants.menus_item_labels.v1.RestaurantsMenusItemLabels.BulkDeleteLabels",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsMenusItemLabelsV1RestaurantsMenusItemLabelsUrl(
        { protoPath: "/v1/bulk/labels/delete", data: payload, host }
      ),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __bulkDeleteLabels;
}

// src/restaurants-menus-v1-item-label-item-labels.universal.ts
import { transformSDKImageToRESTImage } from "@wix/sdk-runtime/transformations/image";
import { transformRESTImageToSDKImage } from "@wix/sdk-runtime/transformations/image";
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createLabel2(label) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({ label }),
    [
      {
        transformFn: transformSDKImageToRESTImage,
        paths: [{ path: "label.icon" }]
      }
    ]
  );
  const reqOpts = createLabel(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [{ path: "label.icon" }]
        }
      ])
    )?.label;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { label: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["label"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getLabel2(labelId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ labelId });
  const reqOpts = getLabel(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [{ path: "label.icon" }]
        }
      ])
    )?.label;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { labelId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["labelId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listLabels2() {
  const { httpClient, sideEffects } = arguments[0];
  const payload = renameKeysFromSDKRequestToRESTRequest({});
  const reqOpts = listLabels(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [{ path: "labels.icon" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {},
        singleArgumentUnchanged: false
      },
      []
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryLabels2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder(
    {
      func: async (payload) => {
        const reqOpts = queryLabels(payload);
        sideEffects?.onSiteCall?.();
        try {
          const result = await httpClient.request(reqOpts);
          sideEffects?.onSuccess?.(result);
          return result;
        } catch (err) {
          sideEffects?.onError?.(err);
          throw err;
        }
      },
      requestTransformer: (query) => {
        const args = [query, {}];
        return renameKeysFromSDKRequestToRESTRequest({
          ...args?.[1],
          query: args?.[0]
        });
      },
      responseTransformer: ({ data }) => {
        const transformedData = renameKeysFromRESTResponseToSDKResponse(
          transformPaths2(data, [
            {
              transformFn: transformRESTImageToSDKImage,
              paths: [{ path: "labels.icon" }]
            }
          ])
        );
        return {
          items: transformedData?.labels,
          pagingMetadata: transformedData?.pagingMetadata
        };
      },
      errorTransformer: (err) => {
        const transformedError = sdkTransformError(err, {
          spreadPathsToArguments: {},
          explicitPathsToArguments: { query: "$[0]" },
          singleArgumentUnchanged: false
        });
        throw transformedError;
      },
      pagingMethod: "CURSOR",
      transformationPaths: {}
    }
  );
}
async function countLabels2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    filter: options?.filter
  });
  const reqOpts = countLabels(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { filter: "$[0].filter" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateLabel2(_id, label) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = transformPaths2(
    renameKeysFromSDKRequestToRESTRequest({ label: { ...label, id: _id } }),
    [
      {
        transformFn: transformSDKImageToRESTImage,
        paths: [{ path: "label.icon" }]
      }
    ]
  );
  const reqOpts = updateLabel(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(
      transformPaths2(result.data, [
        {
          transformFn: transformRESTImageToSDKImage,
          paths: [{ path: "label.icon" }]
        }
      ])
    )?.label;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { label: "$[1]" },
        explicitPathsToArguments: { "label.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "label"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteLabel2(labelId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ labelId });
  const reqOpts = deleteLabel(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { labelId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["labelId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkDeleteLabels2(ids) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ ids });
  const reqOpts = bulkDeleteLabels(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { ids: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["ids"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/restaurants-menus-v1-item-label-item-labels.public.ts
function createLabel3(httpClient) {
  return (label) => createLabel2(
    label,
    // @ts-ignore
    { httpClient }
  );
}
function getLabel3(httpClient) {
  return (labelId) => getLabel2(
    labelId,
    // @ts-ignore
    { httpClient }
  );
}
function listLabels3(httpClient) {
  return () => listLabels2(
    // @ts-ignore
    { httpClient }
  );
}
function queryLabels3(httpClient) {
  return () => queryLabels2(
    // @ts-ignore
    { httpClient }
  );
}
function countLabels3(httpClient) {
  return (options) => countLabels2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function updateLabel3(httpClient) {
  return (_id, label) => updateLabel2(
    _id,
    label,
    // @ts-ignore
    { httpClient }
  );
}
function deleteLabel3(httpClient) {
  return (labelId) => deleteLabel2(
    labelId,
    // @ts-ignore
    { httpClient }
  );
}
function bulkDeleteLabels3(httpClient) {
  return (ids) => bulkDeleteLabels2(
    ids,
    // @ts-ignore
    { httpClient }
  );
}
var onItemLabelCreated = EventDefinition(
  "wix.restaurants.menus.v1.item_label_created",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.externalReferenceInfo.lastSyncDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: transformRESTImageToSDKImage2,
        paths: [{ path: "entity.icon" }]
      }
    ])
  )
)();
var onItemLabelDeleted = EventDefinition(
  "wix.restaurants.menus.v1.item_label_deleted",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "undefined.createdDate" },
          { path: "undefined.updatedDate" },
          { path: "undefined.externalReferenceInfo.lastSyncDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: transformRESTImageToSDKImage2,
        paths: [{ path: "undefined.icon" }]
      }
    ])
  )
)();
var onItemLabelUpdated = EventDefinition(
  "wix.restaurants.menus.v1.item_label_updated",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.externalReferenceInfo.lastSyncDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: transformRESTImageToSDKImage2,
        paths: [{ path: "entity.icon" }]
      }
    ])
  )
)();

// src/restaurants-menus-v1-item-label-item-labels.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
import { createEventModule } from "@wix/sdk-runtime/event-definition-modules";
var createLabel4 = /* @__PURE__ */ createRESTModule(createLabel3);
var getLabel4 = /* @__PURE__ */ createRESTModule(getLabel3);
var listLabels4 = /* @__PURE__ */ createRESTModule(listLabels3);
var queryLabels4 = /* @__PURE__ */ createRESTModule(queryLabels3);
var countLabels4 = /* @__PURE__ */ createRESTModule(countLabels3);
var updateLabel4 = /* @__PURE__ */ createRESTModule(updateLabel3);
var deleteLabel4 = /* @__PURE__ */ createRESTModule(deleteLabel3);
var bulkDeleteLabels4 = /* @__PURE__ */ createRESTModule(bulkDeleteLabels3);
var onItemLabelCreated2 = createEventModule(onItemLabelCreated);
var onItemLabelDeleted2 = createEventModule(onItemLabelDeleted);
var onItemLabelUpdated2 = createEventModule(onItemLabelUpdated);
export {
  SortOrder,
  WebhookIdentityType,
  bulkDeleteLabels4 as bulkDeleteLabels,
  countLabels4 as countLabels,
  createLabel4 as createLabel,
  deleteLabel4 as deleteLabel,
  getLabel4 as getLabel,
  listLabels4 as listLabels,
  onItemLabelCreated2 as onItemLabelCreated,
  onItemLabelDeleted2 as onItemLabelDeleted,
  onItemLabelUpdated2 as onItemLabelUpdated,
  queryLabels4 as queryLabels,
  updateLabel4 as updateLabel
};
