// @wix/wix-ui-test-utils/dist/cjs/unidriver/stylable-unidriver-util-legacy.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.StylableUnidriverUtilLegacy = exports.StylableUnidriverUtilCompat = exports.StylableCompatUniDriver = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _uniDriver = require("@stylable/uni-driver");
/**
 * LEGACY SUPPORT
 */
class StylableUnidriverUtilLegacy {
  constructor(style) {
    this.style = style;
  }
  async hasStyleState(base, stateName, param = true) {
    const {
      stateKey,
      styleState
    } = this.getStateDataAttrKey(stateName, param);
    const actual = await base.attr(stateKey);
    return String(styleState[stateKey]) === actual;
  }
  /**
   * Get style state value
   *
   * @returns state or null if not found
   */
  async getStyleState(base, stateName) {
    const {
      stateKey
    } = this.getStateDataAttrKey(stateName);
    return base.attr(stateKey);
  }
  getStateDataAttrKey(state, param = true) {
    const styleState = this.style.$cssStates({
      [state]: param
    });
    return {
      stateKey: Object.keys(styleState)[0],
      styleState
    };
  }
}
exports.StylableUnidriverUtilLegacy = StylableUnidriverUtilLegacy;
const stateMiddleDelimiter = '-';
class StylableUnidriverUtilCompat {
  constructor(style) {
    this.style = style;
  }
  async hasStyleState(base, stateName, param = true) {
    const stateClass = this.style.$cssStates({
      [stateName]: param
    }).className;
    return base.hasClass(stateClass);
  }
  /**
   * Get style state value
   *
   * @returns state or null if not found
   */
  async getStyleState(base, stateName) {
    const className = (await base.attr('class')) || '';
    if (!className.includes(stateName)) {
      return null;
    }
    const classList = className.trim().split(/\s+/);
    const booleanState = this.style.$cssStates({
      [stateName]: true
    }).className;
    if (classList.includes(booleanState)) {
      return true;
    }
    const baseState = this.getBaseStateWithParam(stateName);
    let paramValue = '';
    classList.forEach(cls => {
      if (!paramValue) {
        paramValue = this.getStateValueFromClassName(cls, baseState);
      }
    });
    return paramValue ? paramValue : null;
  }
  getStateValueFromClassName(cls, baseState) {
    if (cls.startsWith(baseState)) {
      const param = cls.slice(baseState.length);
      const paramIndex = param.indexOf(stateMiddleDelimiter);
      if (paramIndex !== -1) {
        return param.slice(paramIndex + 1);
      }
    }
    return '';
  }
  getBaseStateWithParam(stateName) {
    const singleCharState = 'x';
    return this.style.$cssStates({
      [stateName]: singleCharState
    }).className.slice(0, -3);
  }
}
exports.StylableUnidriverUtilCompat = StylableUnidriverUtilCompat;
class StylableCompatUniDriver {
  // TODO - make sure to fix this - due to type mismatch....!
  constructor(style) {
    this.style = style;
    (0, _defineProperty2.default)(this, "internal", void 0);
    const mode = getStylesheetMode(this.style || {
      $cssStates() {
        return {};
      }
    });
    if (mode === 'v2') {
      this.internal = new _uniDriver.StylableUnidriverUtil(this.style);
    } else if (mode === 'compat') {
      this.internal = new StylableUnidriverUtilCompat(this.style);
    } else {
      this.internal = new StylableUnidriverUtilLegacy(this.style);
    }
  }
  async hasStyleState(base, stateName, param = true) {
    return this.internal.hasStyleState(base, stateName, param);
  }
  /**
   * Get style state value
   *
   * @returns state or null if not found
   */
  async getStyleState(base, stateName) {
    return this.internal.getStyleState(base, stateName);
  }
}
exports.StylableCompatUniDriver = StylableCompatUniDriver;
function getStylesheetMode(sheet) {
  if (sheet.$cssStates) {
    const res = typeof sheet.$cssStates === 'function' ? sheet.$cssStates({}) : {};
    if (res.hasOwnProperty('className')) {
      return 'compat';
    } else {
      return 'legacy';
    }
  } else {
    return 'v2';
  }
}
