// @wix/patterns/dist/cjs/components/CustomFieldsWidget/ActionButtons.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.ActionButtons = void 0;
var _react = _interopRequireDefault(require("react"));
var _designSystem = require("@wix/design-system");
var _mobxReactLite = require("mobx-react-lite");
var _wixUiIconsCommon = require("@wix/wix-ui-icons-common");
var _ManageFieldsCTAButton = require("../CustomFieldsPanel/ManageFieldsCTAButton");
var _CustomFields = require("../CustomFields");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CustomFieldsWidget/ActionButtons.tsx";
const _ActionButtons = ({
  state
}) => {
  const {
    container: {
      translate: t
    }
  } = state;
  return /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
    gap: "SP2",
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 19,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(_CustomFields.CustomFieldsDisabledTooltip, {
    disabled: state.dataExtension.disableAddCustomField,
    content: state.dataExtension.disableAddCustomFieldContent,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 20,
      columnNumber: 7
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.Button, {
    dataHook: "add-custom-field-btn",
    prefixIcon: /*#__PURE__*/_react.default.createElement(_wixUiIconsCommon.Add, {
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 26,
        columnNumber: 23
      }
    }),
    size: "small",
    priority: "secondary",
    onClick: () => {
      state.customFieldsPanel.openCustomFieldModal({
        origin: 'Custom fields widget'
      });
    },
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 24,
      columnNumber: 9
    }
  }, t('cairo.customFields.addCustomField.CTA'))), /*#__PURE__*/_react.default.createElement(_ManageFieldsCTAButton.ManageFieldsCTAButton, {
    state: state,
    origin: "Custom fields widget",
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 38,
      columnNumber: 7
    }
  }));
};
const ActionButtons = exports.ActionButtons = (0, _mobxReactLite.observer)(_ActionButtons);
