// @wix/patterns/dist/cjs/components/CustomFieldsWidget/CustomFieldsNamespaceForm.js
"use strict";

exports.__esModule = true;
exports.CustomFieldsNamespaceForm = void 0;
var _react = _interopRequireWildcard(require("react"));
var _mobxReactLite = require("mobx-react-lite");
var _CustomFieldsForm = require("../CustomFieldsForm");
var _useCustomFieldsFormState = require("./useCustomFieldsFormState");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CustomFieldsWidget/CustomFieldsNamespaceForm.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _CustomFieldsNamespaceForm(props) {
  const {
    state: allNamespacesFormState,
    onChange,
    namespace
  } = props;
  const state = (0, _useCustomFieldsFormState.useCustomFieldsFormState)({
    state: allNamespacesFormState,
    namespace
  });
  (0, _react.useEffect)(() => state.init(), []);
  if (!state.writeableFields.length) {
    return null;
  }
  return /*#__PURE__*/_react.default.createElement(_CustomFieldsForm.CustomFieldsForm, {
    state: state,
    onChange: onChange,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 28,
      columnNumber: 10
    }
  });
}
const CustomFieldsNamespaceForm = exports.CustomFieldsNamespaceForm = (0, _mobxReactLite.observer)(_CustomFieldsNamespaceForm);
