// @wix/patterns/dist/cjs/components/CustomFieldsWidget/CustomFieldsWidgetContentLazy.js
"use strict";

exports.__esModule = true;
exports.CustomFieldsWidgetContentLazy = void 0;
var _mobxReactLite = require("mobx-react-lite");
var _react = _interopRequireWildcard(require("react"));
var _CustomFieldsWidgetContext = require("../CustomFieldsWidgetContext");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CustomFieldsWidget/CustomFieldsWidgetContentLazy.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
const _CustomFieldsWidgetContentLazy = ({
  emptyStateOverrides,
  onChange
}) => {
  const state = (0, _CustomFieldsWidgetContext.useCustomFieldsWidgetContext)();
  const WidgetContent = state.CustomFieldsWidgetContent;
  (0, _react.useEffect)(() => {
    return state.initContent();
  }, []);
  return /*#__PURE__*/_react.default.createElement("div", {
    ref: widgetContentRef => {
      state.setWidgetContentRef(widgetContentRef);
    },
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 21,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(WidgetContent, {
    state: state,
    emptyStateOverrides: emptyStateOverrides,
    onChange: onChange,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 26,
      columnNumber: 7
    }
  }));
};
const CustomFieldsWidgetContentLazy = exports.CustomFieldsWidgetContentLazy = (0, _mobxReactLite.observer)(_CustomFieldsWidgetContentLazy);
