// @wix/patterns/dist/cjs/components/CustomFieldsWidget/CustomFieldsViewWidget.uni.driver.js
"use strict";

exports.__esModule = true;
exports.default = void 0;
var _unidriver = require("../../unidriver");
var _unidriver2 = require("@wix/design-system/dist/testkit/unidriver");
var _default = (base, body) => {
  const errorState = () => (0, _unidriver2.EmptyStateUniDriver)(base.$(`[data-hook="error-card-state"]`), body);
  const emptyState = () => (0, _unidriver2.EmptyStateUniDriver)(base.$(`[data-hook="empty-state"]`), body);
  const tryAgainButton = () => (0, _unidriver2.TextButtonUniDriver)(base.$(`[data-hook="error-card-state-action-button"]`), body);
  const getCustomFieldLabels = () => base.$$(`[data-hook^="custom-field-label"]`).map(async label => {
    const text = await label.text();
    return text.substring(0, text.length - 1);
  });
  const getCustomFieldValues = () => base.$$(`[data-hook^="custom-field-value"]`).map(value => value.text());
  const getSectionTitles = () => base.$$('[data-hook^="custom-field-section-title"]').map(value => value.text());
  const getCustomFieldLabelsForNamespace = namespace => base.$$(`[data-hook^="custom-field-label-${namespace}"]`).map(async label => {
    const text = await label.text();
    return text.substring(0, text.length - 1);
  });
  const getCustomFieldValuesForNamespace = namespace => base.$$(`[data-hook^="custom-field-value-${namespace}"]`).map(value => value.text());
  const getCustomFieldLabel = async (namespace, fieldId) => {
    const wholeLabel = await base.$(`[data-hook="custom-field-label-${namespace}-${fieldId}"]`).text();
    return wholeLabel.substring(0, wholeLabel.length - 1);
  };
  const getCustomFieldValue = (namespace, fieldId) => base.$(`[data-hook="custom-field-value-${namespace}-${fieldId}"]`).text();
  const sectionTitleForNamespaceExists = namespace => base.$(`[data-hook="custom-field-section-title-${namespace}"]`).exists();
  const getSectionTitleForNamespace = namespace => base.$(`[data-hook="custom-field-section-title-${namespace}"]`).text();
  const getFileCard = mediaId => {
    return (0, _unidriver2.BoxUniDriver)(base.$(`[data-hook="file-card-${mediaId}"]`), body);
  };
  const getFileThumbnail = ({
    mediaId,
    mediaType
  }) => {
    return (0, _unidriver2.BoxUniDriver)(base.$(`[data-hook="file-thumbnail-${mediaType}-${mediaId}"]`), body);
  };
  const getFileDownloadButton = mediaId => {
    return (0, _unidriver2.BoxUniDriver)(base.$(`[data-hook="download-button-${mediaId}"]`), body);
  };
  const getFileDetails = mediaId => {
    return (0, _unidriver2.BoxUniDriver)(base.$(`[data-hook="file-details-${mediaId}"]`), body);
  };
  const getFileDetailsSize = mediaId => {
    return (0, _unidriver2.TextUniDriver)(base.$(`[data-hook="file-details-size-${mediaId}"]`), body);
  };
  const getFileDetailsName = mediaId => {
    return (0, _unidriver2.TextUniDriver)(base.$(`[data-hook="file-details-name-${mediaId}"]`), body);
  };
  const getTitleElement = () => base.$(`[data-hook="title"]`);
  return {
    getFileCard,
    getFileThumbnail,
    getFileDownloadButton,
    getFileDetails,
    getFileDetailsName,
    getFileDetailsSize,
    /** Returns title text */
    getTitle: () => getTitleElement().text(),
    getTitleElement,
    /** Indicates whether the widget loading state shows */
    loadingStateExists: () => base.$(`[data-hook="widget-inline-loading"]`).exists(),
    fieldsContentExists: () => base.$(`[data-hook="fields-content"]`).exists(),
    /** Returns the error state title */
    getErrorTitle: () => errorState().getTitleText(),
    /** Returns the error state subtitle */
    getErrorSubtitle: () => errorState().getSubtitleText(),
    /** Returns the empty state title */
    getEmptyTitle: () => emptyState().getTitleText(),
    /** Returns the empty state subtitle */
    getEmptySubtitle: () => base.$(`[data-hook="empty-state-subtitle"]`).text(),
    /** Clicks on try again button when error state visible */
    clickTryAgain: () => tryAgainButton().click(),
    /** Gets try again button text */
    getTryAgainContent: () => tryAgainButton().getButtonTextContent(),
    /** Returns all custom field labels */
    getCustomFieldLabels,
    /** Returns custom field labels for specific namespace */
    getCustomFieldLabelsForNamespace,
    /** Returns specific custom field label by namespace and field id */
    getCustomFieldLabel,
    /** Returns the custom field label by index */
    getCustomFieldLabelByIndex: async index => (await getCustomFieldLabels())[index],
    /** Returns all custom field values */
    getCustomFieldValues,
    /** Returns custom field values for specific namespace */
    getCustomFieldValuesForNamespace,
    /** Returns specific custom field value by namespace and field id */
    getCustomFieldValue,
    /** Returns the custom field value by index */
    getCustomFieldValueByIndex: async index => (await getCustomFieldValues())[index],
    /** Return all section titles */
    getSectionTitles,
    /** Return specific section title for namespace */
    getSectionTitleForNamespace,
    /** Return the section title by index */
    getSectionTitleByIndex: async index => (await getSectionTitles())[index],
    sectionTitleForNamespaceExists,
    ...(0, _unidriver.baseUniDriverFactory)(base)
  };
};
exports.default = _default;
