// @wix/patterns/dist/cjs/components/CustomFieldsWidget/CustomFieldsAllNamespacesForm.js
"use strict";

exports.__esModule = true;
exports.CustomFieldsAllNamespacesForm = void 0;
var _mobxReactLite = require("mobx-react-lite");
var _state = require("../../state");
var _react = _interopRequireWildcard(require("react"));
var _designSystem = require("@wix/design-system");
var _CustomFieldsNamespaceSectionForm = require("./CustomFieldsNamespaceSectionForm");
var _CustomFieldsNamespaceForm = require("./CustomFieldsNamespaceForm");
var _EmptyStateContent = require("./EmptyStateContent");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CustomFieldsWidget/CustomFieldsAllNamespacesForm.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _CustomFieldsAllNamespacesForm(props) {
  const {
    state: customFieldsWidgetState,
    onChange,
    emptyStateOverrides
  } = props;
  const {
    _userFieldsNamespace
  } = customFieldsWidgetState;
  const [state] = (0, _react.useState)(() => customFieldsWidgetState._formState ?? (customFieldsWidgetState._formState = new _state.CustomFieldsAllNamespacesFormState({
    customFieldsWidgetState
  })));
  const {
    schemasSections,
    tpaFieldsExist
  } = state;
  if (!schemasSections.length) {
    return /*#__PURE__*/_react.default.createElement(_EmptyStateContent.EmptyStateContent, {
      state: customFieldsWidgetState,
      emptyStateOverrides: emptyStateOverrides,
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 41,
        columnNumber: 7
      }
    });
  }
  let child;
  if (tpaFieldsExist) {
    child = /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, schemasSections.map((schema, index) => /*#__PURE__*/_react.default.createElement(_CustomFieldsNamespaceSectionForm.CustomFieldsNamespaceSectionForm, {
      key: schema.namespace,
      state: state,
      namespace: schema.namespace,
      index: index,
      onChange: onChange,
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 54,
        columnNumber: 11
      }
    })));
  } else {
    child = /*#__PURE__*/_react.default.createElement(_CustomFieldsNamespaceForm.CustomFieldsNamespaceForm, {
      state: state,
      namespace: _userFieldsNamespace,
      onChange: onChange,
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 66,
        columnNumber: 7
      }
    });
  }
  return /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
    direction: "vertical",
    width: "100%",
    gap: "SP3",
    dataHook: "custom-fields-form",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 75,
      columnNumber: 5
    }
  }, child);
}
const CustomFieldsAllNamespacesForm = exports.CustomFieldsAllNamespacesForm = (0, _mobxReactLite.observer)(_CustomFieldsAllNamespacesForm);
