// @wix/patterns/dist/cjs/components/CustomFieldsWidget/CustomFieldsNamespaceSectionForm.js
"use strict";

exports.__esModule = true;
exports.CustomFieldsNamespaceSectionForm = void 0;
var _react = _interopRequireWildcard(require("react"));
var _mobxReactLite = require("mobx-react-lite");
var _CustomFieldsForm = require("../CustomFieldsForm");
var _designSystem = require("@wix/design-system");
var _useCustomFieldsFormState = require("./useCustomFieldsFormState");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CustomFieldsWidget/CustomFieldsNamespaceSectionForm.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _CustomFieldsNamespaceSectionForm(props) {
  const {
    index,
    state: allNamespacesFormState,
    onChange,
    namespace
  } = props;
  const state = (0, _useCustomFieldsFormState.useCustomFieldsFormState)({
    state: allNamespacesFormState,
    namespace
  });
  const {
    _customFieldsWidgetState: {
      container: {
        translate: t
      },
      _userFieldsNamespace
    }
  } = allNamespacesFormState;
  (0, _react.useEffect)(() => state.init(), []);
  if (!state.writeableFields.length) {
    return null;
  }
  return /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
    key: namespace,
    dataHook: `custom-fields-namespace-form-${namespace}`,
    direction: "vertical",
    width: "100%",
    gap: "SP3",
    paddingTop: index > 0 ? 'SP3' : undefined,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 40,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.SectionHeader, {
    dataHook: "section-header",
    title: /*#__PURE__*/_react.default.createElement(_designSystem.Text, {
      secondary: true,
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 50,
        columnNumber: 16
      }
    }, state.appName),
    suffix: namespace !== _userFieldsNamespace && /*#__PURE__*/_react.default.createElement(_designSystem.InfoIcon, {
      dataHook: "info-icon",
      content: t('cairo.customFields.tpaSection.tooltip'),
      size: "small",
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 53,
        columnNumber: 13
      }
    }),
    skin: "light",
    divider: index > 0 ? 'top' : 'none',
    size: "medium",
    horizontalPadding: "xxTiny",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 48,
      columnNumber: 7
    }
  }), /*#__PURE__*/_react.default.createElement(_CustomFieldsForm.CustomFieldsForm, {
    state: state,
    onChange: onChange,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 66,
      columnNumber: 7
    }
  }));
}
const CustomFieldsNamespaceSectionForm = exports.CustomFieldsNamespaceSectionForm = (0, _mobxReactLite.observer)(_CustomFieldsNamespaceSectionForm);
