// @wix/patterns/dist/cjs/components/CustomFieldsWidget/CustomFieldsWidgetContent.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.CustomFieldsWidgetContent = void 0;
var _react = _interopRequireDefault(require("react"));
var _mobxReactLite = require("mobx-react-lite");
var _CustomFieldsAllNamespacesForm = require("./CustomFieldsAllNamespacesForm");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CustomFieldsWidget/CustomFieldsWidgetContent.tsx";
const _CustomFieldsWidgetContent = ({
  state,
  onChange,
  emptyStateOverrides = {}
}) => {
  return /*#__PURE__*/_react.default.createElement(_CustomFieldsAllNamespacesForm.CustomFieldsAllNamespacesForm, {
    state: state,
    onChange: onChange,
    emptyStateOverrides: emptyStateOverrides,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 20,
      columnNumber: 5
    }
  });
};
const CustomFieldsWidgetContent = exports.CustomFieldsWidgetContent = (0, _mobxReactLite.observer)(_CustomFieldsWidgetContent);
