// @wix/patterns/dist/cjs/components/CustomFieldsWidget/EmptyStateViewContent.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.EmptyStateViewContent = void 0;
var _react = _interopRequireDefault(require("react"));
var _designSystem = require("@wix/design-system");
var _mobxReactLite = require("mobx-react-lite");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CustomFieldsWidget/EmptyStateViewContent.tsx";
function _EmptyStateViewContent({
  title,
  subtitle
}) {
  return /*#__PURE__*/_react.default.createElement(_designSystem.EmptyState, {
    dataHook: "empty-state",
    title: title,
    subtitle: /*#__PURE__*/_react.default.createElement(_designSystem.Text, {
      dataHook: "empty-state-subtitle",
      size: "small",
      secondary: true,
      widows: true,
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 19,
        columnNumber: 9
      }
    }, subtitle),
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 15,
      columnNumber: 5
    }
  });
}
const EmptyStateViewContent = exports.EmptyStateViewContent = (0, _mobxReactLite.observer)(_EmptyStateViewContent);
