// @wix/patterns/dist/cjs/components/CustomFieldsWidget/EmptyStateContent.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.EmptyStateContent = void 0;
var _react = _interopRequireDefault(require("react"));
var _designSystem = require("@wix/design-system");
var _mobxReactLite = require("mobx-react-lite");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CustomFieldsWidget/EmptyStateContent.tsx";
function _EmptyStateContent(props) {
  const {
    state,
    emptyStateOverrides = {}
  } = props;
  const {
    container: {
      translate: t
    }
  } = state;
  return /*#__PURE__*/_react.default.createElement(_designSystem.EmptyState, {
    dataHook: "empty-state",
    title: state.hasWritePermissions ? emptyStateOverrides.title || t('cairo.customFields.emptyState.title') : t('cairo.customFields.emptyStatePassive.title'),
    subtitle: state.hasWritePermissions ? emptyStateOverrides.subtitle || t('cairo.customFields.emptyState.subtitle') : t('cairo.customFields.emptyStatePassive.subtitle'),
    theme: "page-no-border",
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 21,
      columnNumber: 5
    }
  });
}
const EmptyStateContent = exports.EmptyStateContent = (0, _mobxReactLite.observer)(_EmptyStateContent);
