// @wix/patterns/dist/cjs/components/CustomFieldsWidget/ErrorStateContent.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.ErrorStateContent = void 0;
var _react = _interopRequireDefault(require("react"));
var _designSystem = require("@wix/design-system");
var _mobxReactLite = require("mobx-react-lite");
var _wixUiIconsCommon = require("@wix/wix-ui-icons-common");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CustomFieldsWidget/ErrorStateContent.tsx";
const _ErrorStateContent = ({
  state
}) => {
  const {
    container: {
      translate: t
    }
  } = state;
  return /*#__PURE__*/_react.default.createElement(_designSystem.EmptyState, {
    dataHook: "error-state",
    theme: "page-no-border",
    title: t('cairo.customFields.technical.error.title'),
    subtitle: t('cairo.technical.error.text'),
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 18,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.TextButton, {
    dataHook: "retry-button",
    prefixIcon: /*#__PURE__*/_react.default.createElement(_wixUiIconsCommon.Refresh, {
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 26,
        columnNumber: 21
      }
    }),
    onClick: () => state.retry(),
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 24,
      columnNumber: 7
    }
  }, t('cairo.error.tryAgain.cta')));
};
const ErrorStateContent = exports.ErrorStateContent = (0, _mobxReactLite.observer)(_ErrorStateContent);
