// @wix/auto_sdk_payments_psp-callbacks/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  submitEvent: () => submitEvent2
});
module.exports = __toCommonJS(meta_exports);

// src/payments-psp-v1-provider-platform-event-psp-callbacks.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
function resolveWixPaymentsV1ProviderPlatformEventsProviderPlatformEventsServiceUrl(opts) {
  const domainToMappings = {
    "cashier-services._base_domain_": [
      {
        srcPath: "",
        destPath: ""
      },
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      }
    ],
    "difm._base_domain_": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      }
    ],
    "invoices._base_domain_": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      }
    ],
    "ecom._base_domain_": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      }
    ],
    "www.cartscheckout.com": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      }
    ],
    "cartscheckout.com": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      }
    ],
    "cartscheckout._base_domain_": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/payment-services-web",
        destPath: ""
      }
    ],
    "progallery._base_domain_": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: ""
      }
    ],
    "progallerycdn._base_domain_": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: ""
      }
    ],
    "scheduler._base_domain_": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      }
    ],
    "events._base_domain_": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      },
      {
        srcPath: "/payment-services-web",
        destPath: ""
      }
    ],
    "bookings.wixapps.net": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      }
    ],
    "cashier._base_domain_": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      }
    ],
    "ecom.wixapps.net": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      }
    ],
    "cashier.wixapps.net": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      },
      {
        srcPath: "",
        destPath: ""
      }
    ],
    "vod-server._base_domain_": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      }
    ],
    "vod.wixapps.net": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      }
    ],
    "www.wixhotels.com": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      },
      {
        srcPath: "/.well-known/apple-developer-merchantid-domain-association",
        destPath: "/domain-verification/apple-pay"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      },
      {
        srcPath: "/.well-known/apple-developer-merchantid-domain-association",
        destPath: "/domain-verification/apple-pay"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      },
      {
        srcPath: "/.well-known/apple-developer-merchantid-domain-association",
        destPath: "/domain-verification/apple-pay"
      }
    ],
    "test-applepay.wixapps.net": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/payments",
        destPath: "/api/merchant"
      },
      {
        srcPath: "/payments/v1/provider-platform-events",
        destPath: "/payments/v1/provider-platform-events"
      },
      {
        srcPath: "/api/plugin/v1/transactions",
        destPath: "/api/plugin/v1/transactions"
      },
      {
        srcPath: "/payments/api/merchant",
        destPath: "/api/merchant"
      }
    ],
    "us-applepay.wixapps.net": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      }
    ],
    "eu-applepay.wixapps.net": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      }
    ],
    "payments-sb.wixapps.net": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      },
      {
        srcPath: "/ambassador/payment-services-web",
        destPath: ""
      },
      {
        srcPath: "/.well-known/apple-developer-merchantid-domain-association",
        destPath: "/domain-verification/apple-pay"
      }
    ],
    _: [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      },
      {
        srcPath: "/.well-known/apple-developer-merchantid-domain-association",
        destPath: "/domain-verification/apple-pay"
      },
      {
        srcPath: "/payments/psp/v1/provider-platform-events",
        destPath: "/payments/v1/provider-platform-events"
      }
    ],
    "events.wixapps.net": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      }
    ],
    "tpa.wixapps.net": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      },
      {
        srcPath: "/payment-services-web",
        destPath: ""
      }
    ],
    "pricing-plans.wixapps.net": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      }
    ],
    "editor-flow.wixapps.net": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      }
    ],
    "premium._base_domain_": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      },
      {
        srcPath: "/payment-services-web",
        destPath: ""
      },
      {
        srcPath: "/lib-mbapi/include/modules/gateway/return/paypal.php",
        destPath: "/api/payPal/payments/legacy-ipn-to-ignore"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      },
      {
        srcPath: "/.well-known/apple-developer-merchantid-domain-association",
        destPath: "/domain-verification/apple-pay"
      }
    ],
    "payment-webhooks.wixapps.net": [
      {
        srcPath: "/cashier/accounts/stripe",
        destPath: "/api/merchant/stripe"
      },
      {
        srcPath: "/cashier/transactions/stripe",
        destPath: "/api/stripe"
      }
    ],
    "cashier-services.wixapis.com": [
      {
        srcPath: "/payments/v1/provider-platform-events",
        destPath: "/payments/v1/provider-platform-events"
      }
    ],
    "currency-converter.wixapps.net": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/payment-services-web",
        destPath: "/api"
      },
      {
        srcPath: "/payments/psp/v1/provider-platform-events",
        destPath: "/payments/v1/provider-platform-events"
      }
    ]
  };
  return (0, import_rest_modules.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_payments_psp-callbacks";
function submitEvent(payload) {
  function __submitEvent({ host }) {
    const metadata = {
      entityFqdn: "wix.payments.psp.v1.provider_platform_event",
      method: "POST",
      methodFqn: "wix.payments.v1.provider_platform_events.ProviderPlatformEventsService.SubmitEvent",
      packageName: PACKAGE_NAME,
      url: resolveWixPaymentsV1ProviderPlatformEventsProviderPlatformEventsServiceUrl(
        {
          protoPath: "/payments/v1/provider-platform-events",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __submitEvent;
}

// src/payments-psp-v1-provider-platform-event-psp-callbacks.meta.ts
function submitEvent2() {
  const payload = {};
  const getRequestOptions = submitEvent(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/payments/v1/provider-platform-events",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  submitEvent
});
