// @wix/auto_sdk_pricing-plans_plans/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  AppliedAtOriginal: () => AppliedAt,
  ArchivedFilterOriginal: () => ArchivedFilter,
  PeriodUnitOriginal: () => PeriodUnit,
  PublicFilterOriginal: () => PublicFilter,
  SortOrderOriginal: () => SortOrder,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  archivePlan: () => archivePlan2,
  clearPrimary: () => clearPrimary2,
  createPlan: () => createPlan2,
  getPlan: () => getPlan2,
  getPlanStats: () => getPlanStats2,
  listPlans: () => listPlans2,
  listPublicPlans: () => listPublicPlans2,
  makePlanPrimary: () => makePlanPrimary2,
  queryPublicPlans: () => queryPublicPlans2,
  setPlanVisibility: () => setPlanVisibility2,
  updatePlan: () => updatePlan2
});
module.exports = __toCommonJS(meta_exports);

// src/pricing-plans-v2-plan-plans.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressMembershipV2PlansPlansServiceV2Url(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/plan-proxy",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "*.wixapps.net": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    _: [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "editor-flow.wixapps.net": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "forums._base_domain_": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "forums.wixapps.net": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      },
      {
        srcPath: "/_api/pricing-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/paid-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/pricing-plans/v2/plans",
        destPath: "/v2/plans"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_pricing-plans_plans";
function listPublicPlans(payload) {
  function __listPublicPlans({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "GET",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.ListPublicPlans",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans/public",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "plans.createdDate" },
            { path: "plans.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listPublicPlans;
}
function queryPublicPlans(payload) {
  function __queryPublicPlans({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.QueryPublicPlans",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans/public/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "plans.createdDate" },
            { path: "plans.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryPublicPlans;
}
function getPlan(payload) {
  function __getPlan({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "GET",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.GetPlan",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans/{id}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __getPlan;
}
function listPlans(payload) {
  function __listPlans({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "GET",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.ListPlans",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "plans.createdDate" },
            { path: "plans.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listPlans;
}
function getPlanStats(payload) {
  function __getPlanStats({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "GET",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.GetPlanStats",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans/stats",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getPlanStats;
}
function createPlan(payload) {
  function __createPlan({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.CreatePlan",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __createPlan;
}
function updatePlan(payload) {
  function __updatePlan({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "PATCH",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.UpdatePlan",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans/{plan.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __updatePlan;
}
function setPlanVisibility(payload) {
  function __setPlanVisibility({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "PUT",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.SetPlanVisibility",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans/{id}/visibility",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __setPlanVisibility;
}
function makePlanPrimary(payload) {
  function __makePlanPrimary({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.MakePlanPrimary",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans/{id}/make-primary",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __makePlanPrimary;
}
function clearPrimary(payload) {
  function __clearPrimary({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.ClearPrimary",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans/clear-primary",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __clearPrimary;
}
function archivePlan(payload) {
  function __archivePlan({ host }) {
    const metadata = {
      entityFqdn: "wix.pricing_plans.v2.plan",
      method: "POST",
      methodFqn: "com.wixpress.membership.v2.plans.PlansServiceV2.ArchivePlan",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
        protoPath: "/v2/plans/{id}/archive",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "plan.createdDate" }, { path: "plan.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __archivePlan;
}

// src/pricing-plans-v2-plan-plans.types.ts
var PeriodUnit = /* @__PURE__ */ ((PeriodUnit2) => {
  PeriodUnit2["UNDEFINED"] = "UNDEFINED";
  PeriodUnit2["DAY"] = "DAY";
  PeriodUnit2["WEEK"] = "WEEK";
  PeriodUnit2["MONTH"] = "MONTH";
  PeriodUnit2["YEAR"] = "YEAR";
  return PeriodUnit2;
})(PeriodUnit || {});
var AppliedAt = /* @__PURE__ */ ((AppliedAt2) => {
  AppliedAt2["UNKNOWN_CHARGE_EVENT"] = "UNKNOWN_CHARGE_EVENT";
  AppliedAt2["FIRST_PAYMENT"] = "FIRST_PAYMENT";
  return AppliedAt2;
})(AppliedAt || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var ArchivedFilter = /* @__PURE__ */ ((ArchivedFilter2) => {
  ArchivedFilter2["ACTIVE"] = "ACTIVE";
  ArchivedFilter2["ARCHIVED"] = "ARCHIVED";
  ArchivedFilter2["ARCHIVED_AND_ACTIVE"] = "ARCHIVED_AND_ACTIVE";
  return ArchivedFilter2;
})(ArchivedFilter || {});
var PublicFilter = /* @__PURE__ */ ((PublicFilter2) => {
  PublicFilter2["PUBLIC_AND_HIDDEN"] = "PUBLIC_AND_HIDDEN";
  PublicFilter2["PUBLIC"] = "PUBLIC";
  PublicFilter2["HIDDEN"] = "HIDDEN";
  return PublicFilter2;
})(PublicFilter || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/pricing-plans-v2-plan-plans.meta.ts
function listPublicPlans2() {
  const payload = {};
  const getRequestOptions = listPublicPlans(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/plans/public",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryPublicPlans2() {
  const payload = {};
  const getRequestOptions = queryPublicPlans(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/plans/public/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getPlan2() {
  const payload = { id: ":id" };
  const getRequestOptions = getPlan(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/plans/{id}",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listPlans2() {
  const payload = {};
  const getRequestOptions = listPlans(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/plans",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getPlanStats2() {
  const payload = {};
  const getRequestOptions = getPlanStats(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/plans/stats",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createPlan2() {
  const payload = {};
  const getRequestOptions = createPlan(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/plans",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updatePlan2() {
  const payload = { plan: { id: ":planId" } };
  const getRequestOptions = updatePlan(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v2/plans/{plan.id}",
    pathParams: { planId: "planId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function setPlanVisibility2() {
  const payload = { id: ":id" };
  const getRequestOptions = setPlanVisibility(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PUT",
    path: "/v2/plans/{id}/visibility",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function makePlanPrimary2() {
  const payload = { id: ":id" };
  const getRequestOptions = makePlanPrimary(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/plans/{id}/make-primary",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function clearPrimary2() {
  const payload = {};
  const getRequestOptions = clearPrimary(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/plans/clear-primary",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function archivePlan2() {
  const payload = { id: ":id" };
  const getRequestOptions = archivePlan(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/plans/{id}/archive",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AppliedAtOriginal,
  ArchivedFilterOriginal,
  PeriodUnitOriginal,
  PublicFilterOriginal,
  SortOrderOriginal,
  WebhookIdentityTypeOriginal,
  archivePlan,
  clearPrimary,
  createPlan,
  getPlan,
  getPlanStats,
  listPlans,
  listPublicPlans,
  makePlanPrimary,
  queryPublicPlans,
  setPlanVisibility,
  updatePlan
});
