// @wix/auto_sdk_events_staff-members/build/internal/es/index.mjs
// src/events-staffmembers-v1-staff-member-staff-members.public.ts
import { renameKeysFromRESTResponseToSDKResponse as renameKeysFromRESTResponseToSDKResponse2 } from "@wix/sdk-runtime/rename-all-nested-keys";
import { transformRESTTimestampToSDKTimestamp as transformRESTTimestampToSDKTimestamp2 } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths as transformPaths3 } from "@wix/sdk-runtime/transformations/transform-paths";
import { EventDefinition } from "@wix/sdk-types";

// src/events-staffmembers-v1-staff-member-staff-members.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/events-staffmembers-v1-staff-member-staff-members.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixEventsStaffmembersV1StaffMemberManagementUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/events-staff-members",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/events-staff-members",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/events-staff-members",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/events-staff-members",
        destPath: ""
      },
      {
        srcPath: "/_api/events-staff-members",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/events-staff-members",
        destPath: ""
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/events-staff-members",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/events-staff-members",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/events/v1/events-staff-members",
        destPath: "/v1/events-staff-members"
      },
      {
        srcPath: "/events/v1/staff-members",
        destPath: "/v1/staff-members"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_events_staff-members";
function createStaffMember(payload) {
  function __createStaffMember({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "staffMember.createdDate" },
          { path: "staffMember.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.staffmembers.v1.staff_member",
      method: "POST",
      methodFqn: "wix.events.staffmembers.v1.StaffMemberManagement.CreateStaffMember",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsStaffmembersV1StaffMemberManagementUrl({
        protoPath: "/v1/staff-members",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "staffMember.createdDate" },
            { path: "staffMember.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createStaffMember;
}
function getStaffMember(payload) {
  function __getStaffMember({ host }) {
    const metadata = {
      entityFqdn: "wix.events.staffmembers.v1.staff_member",
      method: "GET",
      methodFqn: "wix.events.staffmembers.v1.StaffMemberManagement.GetStaffMember",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsStaffmembersV1StaffMemberManagementUrl({
        protoPath: "/v1/staff-members/{staffMemberId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "staffMember.createdDate" },
            { path: "staffMember.updatedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "GET",
          url: resolveWixEventsStaffmembersV1StaffMemberManagementUrl({
            protoPath: "/v1/staff-members/{staffMemberId}",
            data: payload,
            host
          }),
          params: toURLSearchParams(payload)
        }
      ]
    };
    return metadata;
  }
  return __getStaffMember;
}
function updateStaffMember(payload) {
  function __updateStaffMember({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "staffMember.createdDate" },
          { path: "staffMember.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.staffmembers.v1.staff_member",
      method: "PATCH",
      methodFqn: "wix.events.staffmembers.v1.StaffMemberManagement.UpdateStaffMember",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsStaffmembersV1StaffMemberManagementUrl({
        protoPath: "/v1/staff-members/{staffMember.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "staffMember.createdDate" },
            { path: "staffMember.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateStaffMember;
}
function deleteStaffMember(payload) {
  function __deleteStaffMember({ host }) {
    const metadata = {
      entityFqdn: "wix.events.staffmembers.v1.staff_member",
      method: "DELETE",
      methodFqn: "wix.events.staffmembers.v1.StaffMemberManagement.DeleteStaffMember",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsStaffmembersV1StaffMemberManagementUrl({
        protoPath: "/v1/staff-members/{staffMemberId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteStaffMember;
}
function queryStaffMembers(payload) {
  function __queryStaffMembers({ host }) {
    const metadata = {
      entityFqdn: "wix.events.staffmembers.v1.staff_member",
      method: "POST",
      methodFqn: "wix.events.staffmembers.v1.StaffMemberManagement.QueryStaffMembers",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsStaffmembersV1StaffMemberManagementUrl({
        protoPath: "/v1/staff-members/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "staffMembers.createdDate" },
            { path: "staffMembers.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryStaffMembers;
}
function joinStaffMember(payload) {
  function __joinStaffMember({ host }) {
    const metadata = {
      entityFqdn: "wix.events.staffmembers.v1.staff_member",
      method: "POST",
      methodFqn: "wix.events.staffmembers.v1.StaffMemberManagement.JoinStaffMember",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsStaffmembersV1StaffMemberManagementUrl({
        protoPath: "/v1/staff-members/join",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "staffMember.createdDate" },
            { path: "staffMember.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __joinStaffMember;
}
function loginStaffMember(payload) {
  function __loginStaffMember({ host }) {
    const metadata = {
      entityFqdn: "wix.events.staffmembers.v1.staff_member",
      method: "POST",
      methodFqn: "wix.events.staffmembers.v1.StaffMemberManagement.LoginStaffMember",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsStaffmembersV1StaffMemberManagementUrl({
        protoPath: "/v1/staff-members/login",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "staffMember.createdDate" },
            { path: "staffMember.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __loginStaffMember;
}
function accessStaffMember(payload) {
  function __accessStaffMember({ host }) {
    const metadata = {
      entityFqdn: "wix.events.staffmembers.v1.staff_member",
      method: "POST",
      methodFqn: "wix.events.staffmembers.v1.StaffMemberManagement.AccessStaffMember",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsStaffmembersV1StaffMemberManagementUrl({
        protoPath: "/v1/staff-members/access",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "staffMember.createdDate" },
            { path: "staffMember.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __accessStaffMember;
}

// src/events-staffmembers-v1-staff-member-staff-members.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var Scope = /* @__PURE__ */ ((Scope2) => {
  Scope2["UNKNOWN_SCOPE"] = "UNKNOWN_SCOPE";
  Scope2["CHECK_IN_GUESTS"] = "CHECK_IN_GUESTS";
  Scope2["SELL_TICKETS"] = "SELL_TICKETS";
  Scope2["ADD_GUESTS"] = "ADD_GUESTS";
  Scope2["READ_ANALYTICS"] = "READ_ANALYTICS";
  return Scope2;
})(Scope || {});
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["PENDING"] = "PENDING";
  Status2["HAS_ACCESS"] = "HAS_ACCESS";
  Status2["ACCESS_PAUSED"] = "ACCESS_PAUSED";
  return Status2;
})(Status || {});
var Permission = /* @__PURE__ */ ((Permission2) => {
  Permission2["UNKNOWN_PERMISSION"] = "UNKNOWN_PERMISSION";
  Permission2["CHECK_IN_GUESTS"] = "CHECK_IN_GUESTS";
  Permission2["SELL_TICKETS"] = "SELL_TICKETS";
  Permission2["ADD_GUESTS"] = "ADD_GUESTS";
  Permission2["READ_ANALYTICS"] = "READ_ANALYTICS";
  Permission2["READ_GUEST_FORM"] = "READ_GUEST_FORM";
  return Permission2;
})(Permission || {});
var Fieldset = /* @__PURE__ */ ((Fieldset2) => {
  Fieldset2["LOGIN_LINK"] = "LOGIN_LINK";
  return Fieldset2;
})(Fieldset || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createStaffMember2(staffMember) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    staffMember
  });
  const reqOpts = createStaffMember(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.staffMember;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { staffMember: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["staffMember"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getStaffMember2(staffMemberId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    staffMemberId,
    fieldsets: options?.fieldsets
  });
  const reqOpts = getStaffMember(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.staffMember;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          staffMemberId: "$[0]",
          fieldsets: "$[1].fieldsets"
        },
        singleArgumentUnchanged: false
      },
      ["staffMemberId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateStaffMember2(_id, staffMember) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    staffMember: { ...staffMember, id: _id }
  });
  const reqOpts = updateStaffMember(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.staffMember;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { staffMember: "$[1]" },
        explicitPathsToArguments: { "staffMember.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "staffMember"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteStaffMember2(staffMemberId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    staffMemberId
  });
  const reqOpts = deleteStaffMember(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { staffMemberId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["staffMemberId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryStaffMembers2(options) {
  const { httpClient, sideEffects } = arguments[1];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryStaffMembers({
        ...payload,
        ...options ?? {}
      });
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, options];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [])
      );
      return {
        items: transformedData?.staffMembers,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function joinStaffMember2(joinToken) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    joinToken
  });
  const reqOpts = joinStaffMember(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { joinToken: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["joinToken"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function loginStaffMember2(loginToken) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    loginToken
  });
  const reqOpts = loginStaffMember(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { loginToken: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["loginToken"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function accessStaffMember2(accessToken) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    accessToken
  });
  const reqOpts = accessStaffMember(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { accessToken: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["accessToken"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/events-staffmembers-v1-staff-member-staff-members.public.ts
function createStaffMember3(httpClient) {
  return (staffMember) => createStaffMember2(
    staffMember,
    // @ts-ignore
    { httpClient }
  );
}
function getStaffMember3(httpClient) {
  return (staffMemberId, options) => getStaffMember2(
    staffMemberId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function updateStaffMember3(httpClient) {
  return (_id, staffMember) => updateStaffMember2(
    _id,
    staffMember,
    // @ts-ignore
    { httpClient }
  );
}
function deleteStaffMember3(httpClient) {
  return (staffMemberId) => deleteStaffMember2(
    staffMemberId,
    // @ts-ignore
    { httpClient }
  );
}
function queryStaffMembers3(httpClient) {
  return (options) => queryStaffMembers2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function joinStaffMember3(httpClient) {
  return (joinToken) => joinStaffMember2(
    joinToken,
    // @ts-ignore
    { httpClient }
  );
}
function loginStaffMember3(httpClient) {
  return (loginToken) => loginStaffMember2(
    loginToken,
    // @ts-ignore
    { httpClient }
  );
}
function accessStaffMember3(httpClient) {
  return (accessToken) => accessStaffMember2(
    accessToken,
    // @ts-ignore
    { httpClient }
  );
}
var onStaffMemberCreated = EventDefinition(
  "wix.events.staffmembers.v1.staff_member_created",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onStaffMemberDeleted = EventDefinition(
  "wix.events.staffmembers.v1.staff_member_deleted",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onStaffMemberUpdated = EventDefinition(
  "wix.events.staffmembers.v1.staff_member_updated",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();

// src/events-staffmembers-v1-staff-member-staff-members.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
import { createEventModule } from "@wix/sdk-runtime/event-definition-modules";
var createStaffMember4 = /* @__PURE__ */ createRESTModule(createStaffMember3);
var getStaffMember4 = /* @__PURE__ */ createRESTModule(getStaffMember3);
var updateStaffMember4 = /* @__PURE__ */ createRESTModule(updateStaffMember3);
var deleteStaffMember4 = /* @__PURE__ */ createRESTModule(deleteStaffMember3);
var queryStaffMembers4 = /* @__PURE__ */ createRESTModule(queryStaffMembers3);
var joinStaffMember4 = /* @__PURE__ */ createRESTModule(joinStaffMember3);
var loginStaffMember4 = /* @__PURE__ */ createRESTModule(loginStaffMember3);
var accessStaffMember4 = /* @__PURE__ */ createRESTModule(accessStaffMember3);
var onStaffMemberCreated2 = createEventModule(
  onStaffMemberCreated
);
var onStaffMemberDeleted2 = createEventModule(
  onStaffMemberDeleted
);
var onStaffMemberUpdated2 = createEventModule(
  onStaffMemberUpdated
);
export {
  Fieldset,
  Permission,
  Scope,
  SortOrder,
  Status,
  WebhookIdentityType,
  accessStaffMember4 as accessStaffMember,
  createStaffMember4 as createStaffMember,
  deleteStaffMember4 as deleteStaffMember,
  getStaffMember4 as getStaffMember,
  joinStaffMember4 as joinStaffMember,
  loginStaffMember4 as loginStaffMember,
  onStaffMemberCreated2 as onStaffMemberCreated,
  onStaffMemberDeleted2 as onStaffMemberDeleted,
  onStaffMemberUpdated2 as onStaffMemberUpdated,
  queryStaffMembers4 as queryStaffMembers,
  updateStaffMember4 as updateStaffMember
};
