// @wix/auto_sdk_events_staff-members/build/internal/es/meta.mjs
// src/events-staffmembers-v1-staff-member-staff-members.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixEventsStaffmembersV1StaffMemberManagementUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/events-staff-members",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/events-staff-members",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/events-staff-members",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/events-staff-members",
        destPath: ""
      },
      {
        srcPath: "/_api/events-staff-members",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/events-staff-members",
        destPath: ""
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/events-staff-members",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/events-staff-members",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/events/v1/events-staff-members",
        destPath: "/v1/events-staff-members"
      },
      {
        srcPath: "/events/v1/staff-members",
        destPath: "/v1/staff-members"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_events_staff-members";
function createStaffMember(payload) {
  function __createStaffMember({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "staffMember.createdDate" },
          { path: "staffMember.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.staffmembers.v1.staff_member",
      method: "POST",
      methodFqn: "wix.events.staffmembers.v1.StaffMemberManagement.CreateStaffMember",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsStaffmembersV1StaffMemberManagementUrl({
        protoPath: "/v1/staff-members",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "staffMember.createdDate" },
            { path: "staffMember.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createStaffMember;
}
function getStaffMember(payload) {
  function __getStaffMember({ host }) {
    const metadata = {
      entityFqdn: "wix.events.staffmembers.v1.staff_member",
      method: "GET",
      methodFqn: "wix.events.staffmembers.v1.StaffMemberManagement.GetStaffMember",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsStaffmembersV1StaffMemberManagementUrl({
        protoPath: "/v1/staff-members/{staffMemberId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "staffMember.createdDate" },
            { path: "staffMember.updatedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "GET",
          url: resolveWixEventsStaffmembersV1StaffMemberManagementUrl({
            protoPath: "/v1/staff-members/{staffMemberId}",
            data: payload,
            host
          }),
          params: toURLSearchParams(payload)
        }
      ]
    };
    return metadata;
  }
  return __getStaffMember;
}
function updateStaffMember(payload) {
  function __updateStaffMember({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "staffMember.createdDate" },
          { path: "staffMember.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.staffmembers.v1.staff_member",
      method: "PATCH",
      methodFqn: "wix.events.staffmembers.v1.StaffMemberManagement.UpdateStaffMember",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsStaffmembersV1StaffMemberManagementUrl({
        protoPath: "/v1/staff-members/{staffMember.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "staffMember.createdDate" },
            { path: "staffMember.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateStaffMember;
}
function deleteStaffMember(payload) {
  function __deleteStaffMember({ host }) {
    const metadata = {
      entityFqdn: "wix.events.staffmembers.v1.staff_member",
      method: "DELETE",
      methodFqn: "wix.events.staffmembers.v1.StaffMemberManagement.DeleteStaffMember",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsStaffmembersV1StaffMemberManagementUrl({
        protoPath: "/v1/staff-members/{staffMemberId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteStaffMember;
}
function queryStaffMembers(payload) {
  function __queryStaffMembers({ host }) {
    const metadata = {
      entityFqdn: "wix.events.staffmembers.v1.staff_member",
      method: "POST",
      methodFqn: "wix.events.staffmembers.v1.StaffMemberManagement.QueryStaffMembers",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsStaffmembersV1StaffMemberManagementUrl({
        protoPath: "/v1/staff-members/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "staffMembers.createdDate" },
            { path: "staffMembers.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryStaffMembers;
}
function joinStaffMember(payload) {
  function __joinStaffMember({ host }) {
    const metadata = {
      entityFqdn: "wix.events.staffmembers.v1.staff_member",
      method: "POST",
      methodFqn: "wix.events.staffmembers.v1.StaffMemberManagement.JoinStaffMember",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsStaffmembersV1StaffMemberManagementUrl({
        protoPath: "/v1/staff-members/join",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "staffMember.createdDate" },
            { path: "staffMember.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __joinStaffMember;
}
function loginStaffMember(payload) {
  function __loginStaffMember({ host }) {
    const metadata = {
      entityFqdn: "wix.events.staffmembers.v1.staff_member",
      method: "POST",
      methodFqn: "wix.events.staffmembers.v1.StaffMemberManagement.LoginStaffMember",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsStaffmembersV1StaffMemberManagementUrl({
        protoPath: "/v1/staff-members/login",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "staffMember.createdDate" },
            { path: "staffMember.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __loginStaffMember;
}
function accessStaffMember(payload) {
  function __accessStaffMember({ host }) {
    const metadata = {
      entityFqdn: "wix.events.staffmembers.v1.staff_member",
      method: "POST",
      methodFqn: "wix.events.staffmembers.v1.StaffMemberManagement.AccessStaffMember",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsStaffmembersV1StaffMemberManagementUrl({
        protoPath: "/v1/staff-members/access",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "staffMember.createdDate" },
            { path: "staffMember.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __accessStaffMember;
}

// src/events-staffmembers-v1-staff-member-staff-members.types.ts
var Scope = /* @__PURE__ */ ((Scope2) => {
  Scope2["UNKNOWN_SCOPE"] = "UNKNOWN_SCOPE";
  Scope2["CHECK_IN_GUESTS"] = "CHECK_IN_GUESTS";
  Scope2["SELL_TICKETS"] = "SELL_TICKETS";
  Scope2["ADD_GUESTS"] = "ADD_GUESTS";
  Scope2["READ_ANALYTICS"] = "READ_ANALYTICS";
  return Scope2;
})(Scope || {});
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["PENDING"] = "PENDING";
  Status2["HAS_ACCESS"] = "HAS_ACCESS";
  Status2["ACCESS_PAUSED"] = "ACCESS_PAUSED";
  return Status2;
})(Status || {});
var Permission = /* @__PURE__ */ ((Permission2) => {
  Permission2["UNKNOWN_PERMISSION"] = "UNKNOWN_PERMISSION";
  Permission2["CHECK_IN_GUESTS"] = "CHECK_IN_GUESTS";
  Permission2["SELL_TICKETS"] = "SELL_TICKETS";
  Permission2["ADD_GUESTS"] = "ADD_GUESTS";
  Permission2["READ_ANALYTICS"] = "READ_ANALYTICS";
  Permission2["READ_GUEST_FORM"] = "READ_GUEST_FORM";
  return Permission2;
})(Permission || {});
var Fieldset = /* @__PURE__ */ ((Fieldset2) => {
  Fieldset2["LOGIN_LINK"] = "LOGIN_LINK";
  return Fieldset2;
})(Fieldset || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/events-staffmembers-v1-staff-member-staff-members.meta.ts
function createStaffMember2() {
  const payload = {};
  const getRequestOptions = createStaffMember(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/staff-members",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getStaffMember2() {
  const payload = { staffMemberId: ":staffMemberId" };
  const getRequestOptions = getStaffMember(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/staff-members/{staffMemberId}",
    pathParams: { staffMemberId: "staffMemberId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateStaffMember2() {
  const payload = { staffMember: { id: ":staffMemberId" } };
  const getRequestOptions = updateStaffMember(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/staff-members/{staffMember.id}",
    pathParams: { staffMemberId: "staffMemberId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteStaffMember2() {
  const payload = { staffMemberId: ":staffMemberId" };
  const getRequestOptions = deleteStaffMember(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/staff-members/{staffMemberId}",
    pathParams: { staffMemberId: "staffMemberId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryStaffMembers2() {
  const payload = {};
  const getRequestOptions = queryStaffMembers(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/staff-members/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function joinStaffMember2() {
  const payload = {};
  const getRequestOptions = joinStaffMember(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/staff-members/join",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function loginStaffMember2() {
  const payload = {};
  const getRequestOptions = loginStaffMember(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/staff-members/login",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function accessStaffMember2() {
  const payload = {};
  const getRequestOptions = accessStaffMember(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/staff-members/access",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  Fieldset as FieldsetOriginal,
  Permission as PermissionOriginal,
  Scope as ScopeOriginal,
  SortOrder as SortOrderOriginal,
  Status as StatusOriginal,
  WebhookIdentityType as WebhookIdentityTypeOriginal,
  accessStaffMember2 as accessStaffMember,
  createStaffMember2 as createStaffMember,
  deleteStaffMember2 as deleteStaffMember,
  getStaffMember2 as getStaffMember,
  joinStaffMember2 as joinStaffMember,
  loginStaffMember2 as loginStaffMember,
  queryStaffMembers2 as queryStaffMembers,
  updateStaffMember2 as updateStaffMember
};
