// @wix/auto_sdk_restaurants_fulfillemt-methods/build/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  BestDeliveryByCategory: () => BestDeliveryByCategory,
  DayOfWeek: () => DayOfWeek,
  FulfillmentMethodType: () => FulfillmentMethodType,
  TimeDisplayUnit: () => TimeDisplayUnit,
  Type: () => Type,
  createFulfillmentMethod: () => createFulfillmentMethod4,
  deleteFulfillmentMethod: () => deleteFulfillmentMethod4,
  getFulfillmentMethod: () => getFulfillmentMethod4,
  listFulfillmentMethods: () => listFulfillmentMethods4,
  updateFulfillmentMethod: () => updateFulfillmentMethod4
});
module.exports = __toCommonJS(index_exports);

// src/restaurants-fulfillments-v3-fulfillment-method-fulfillemt-methods.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/restaurants-fulfillments-v3-fulfillment-method-fulfillemt-methods.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressRestaurantsFulfillmentsV3FulfillmentMethodsUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/v3/fulfillment-methods",
        destPath: "/v3/fulfillment-methods"
      }
    ],
    _: [
      {
        srcPath: "/_api/fulfillment-methods",
        destPath: "/v3/fulfillment-methods"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/fulfillment-methods",
        destPath: "/v3/fulfillment-methods"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/fulfillment-methods",
        destPath: "/v3/fulfillment-methods"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_fulfillemt-methods";
function createFulfillmentMethod(payload) {
  function __createFulfillmentMethod({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          {
            path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.radiusDetails.geocode.latitude"
          },
          {
            path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.radiusDetails.geocode.longitude"
          },
          {
            path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.customDetails.geocodes.latitude"
          },
          {
            path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.customDetails.geocodes.longitude"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.fulfillments.v3.fulfillment_method",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.fulfillments.v3.FulfillmentMethods.CreateFulfillmentMethod",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsFulfillmentsV3FulfillmentMethodsUrl({
        protoPath: "/v3/fulfillment-methods",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.radiusDetails.geocode.latitude"
            },
            {
              path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.radiusDetails.geocode.longitude"
            },
            {
              path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.customDetails.geocodes.latitude"
            },
            {
              path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.customDetails.geocodes.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createFulfillmentMethod;
}
function getFulfillmentMethod(payload) {
  function __getFulfillmentMethod({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.fulfillments.v3.fulfillment_method",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.fulfillments.v3.FulfillmentMethods.GetFulfillmentMethod",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsFulfillmentsV3FulfillmentMethodsUrl({
        protoPath: "/v3/fulfillment-methods/{fulfillmentMethodId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.radiusDetails.geocode.latitude"
            },
            {
              path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.radiusDetails.geocode.longitude"
            },
            {
              path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.customDetails.geocodes.latitude"
            },
            {
              path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.customDetails.geocodes.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getFulfillmentMethod;
}
function updateFulfillmentMethod(payload) {
  function __updateFulfillmentMethod({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          {
            path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.radiusDetails.geocode.latitude"
          },
          {
            path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.radiusDetails.geocode.longitude"
          },
          {
            path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.customDetails.geocodes.latitude"
          },
          {
            path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.customDetails.geocodes.longitude"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.fulfillments.v3.fulfillment_method",
      method: "PATCH",
      methodFqn: "com.wixpress.restaurants.fulfillments.v3.FulfillmentMethods.UpdateFulfillmentMethod",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsFulfillmentsV3FulfillmentMethodsUrl({
        protoPath: "/v3/fulfillment-methods/{fulfillmentMethod.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.radiusDetails.geocode.latitude"
            },
            {
              path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.radiusDetails.geocode.longitude"
            },
            {
              path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.customDetails.geocodes.latitude"
            },
            {
              path: "fulfillmentMethod.deliveryInfo.restaurant.deliveryRange.customDetails.geocodes.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateFulfillmentMethod;
}
function deleteFulfillmentMethod(payload) {
  function __deleteFulfillmentMethod({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.fulfillments.v3.fulfillment_method",
      method: "DELETE",
      methodFqn: "com.wixpress.restaurants.fulfillments.v3.FulfillmentMethods.DeleteFulfillmentMethod",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsFulfillmentsV3FulfillmentMethodsUrl({
        protoPath: "/v3/fulfillment-methods/{fulfillmentMethodId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteFulfillmentMethod;
}
function listFulfillmentMethods(payload) {
  function __listFulfillmentMethods({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.fulfillments.v3.fulfillment_method",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.fulfillments.v3.FulfillmentMethods.ListFulfillmentMethods",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsFulfillmentsV3FulfillmentMethodsUrl({
        protoPath: "/v3/fulfillment-methods",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "fulfillmentMethods.deliveryInfo.restaurant.deliveryRange.radiusDetails.geocode.latitude"
            },
            {
              path: "fulfillmentMethods.deliveryInfo.restaurant.deliveryRange.radiusDetails.geocode.longitude"
            },
            {
              path: "fulfillmentMethods.deliveryInfo.restaurant.deliveryRange.customDetails.geocodes.latitude"
            },
            {
              path: "fulfillmentMethods.deliveryInfo.restaurant.deliveryRange.customDetails.geocodes.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listFulfillmentMethods;
}

// src/restaurants-fulfillments-v3-fulfillment-method-fulfillemt-methods.universal.ts
var FulfillmentMethodType = /* @__PURE__ */ ((FulfillmentMethodType2) => {
  FulfillmentMethodType2["UNSPECIFIED_FULFILLMENT_TYPE"] = "UNSPECIFIED_FULFILLMENT_TYPE";
  FulfillmentMethodType2["PICKUP"] = "PICKUP";
  FulfillmentMethodType2["DELIVERY"] = "DELIVERY";
  FulfillmentMethodType2["DINE_IN"] = "DINE_IN";
  return FulfillmentMethodType2;
})(FulfillmentMethodType || {});
var DayOfWeek = /* @__PURE__ */ ((DayOfWeek2) => {
  DayOfWeek2["MON"] = "MON";
  DayOfWeek2["TUE"] = "TUE";
  DayOfWeek2["WED"] = "WED";
  DayOfWeek2["THU"] = "THU";
  DayOfWeek2["FRI"] = "FRI";
  DayOfWeek2["SAT"] = "SAT";
  DayOfWeek2["SUN"] = "SUN";
  return DayOfWeek2;
})(DayOfWeek || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNDEFINED"] = "UNDEFINED";
  Type2["RADIUS"] = "RADIUS";
  Type2["POSTAL_CODE"] = "POSTAL_CODE";
  Type2["CUSTOM"] = "CUSTOM";
  return Type2;
})(Type || {});
var TimeDisplayUnit = /* @__PURE__ */ ((TimeDisplayUnit2) => {
  TimeDisplayUnit2["UNDEFINED"] = "UNDEFINED";
  TimeDisplayUnit2["MINUTES"] = "MINUTES";
  TimeDisplayUnit2["HOURS"] = "HOURS";
  TimeDisplayUnit2["DAYS"] = "DAYS";
  return TimeDisplayUnit2;
})(TimeDisplayUnit || {});
var BestDeliveryByCategory = /* @__PURE__ */ ((BestDeliveryByCategory2) => {
  BestDeliveryByCategory2["DELIVERY_FEE"] = "DELIVERY_FEE";
  return BestDeliveryByCategory2;
})(BestDeliveryByCategory || {});
async function createFulfillmentMethod2(fulfillmentMethod) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    fulfillmentMethod
  });
  const reqOpts = createFulfillmentMethod(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.fulfillmentMethod;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { fulfillmentMethod: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["fulfillmentMethod"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getFulfillmentMethod2(fulfillmentMethodId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    fulfillmentMethodId
  });
  const reqOpts = getFulfillmentMethod(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.fulfillmentMethod;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { fulfillmentMethodId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["fulfillmentMethodId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateFulfillmentMethod2(_id, fulfillmentMethod) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    fulfillmentMethod: { ...fulfillmentMethod, id: _id }
  });
  const reqOpts = updateFulfillmentMethod(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.fulfillmentMethod;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { fulfillmentMethod: "$[1]" },
        explicitPathsToArguments: { "fulfillmentMethod.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "fulfillmentMethod"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteFulfillmentMethod2(fulfillmentMethodId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    fulfillmentMethodId
  });
  const reqOpts = deleteFulfillmentMethod(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { fulfillmentMethodId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["fulfillmentMethodId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listFulfillmentMethods2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    restaurantLocationIds: options?.restaurantLocationIds
  });
  const reqOpts = listFulfillmentMethods(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          restaurantLocationIds: "$[0].restaurantLocationIds"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/restaurants-fulfillments-v3-fulfillment-method-fulfillemt-methods.public.ts
function createFulfillmentMethod3(httpClient) {
  return (fulfillmentMethod) => createFulfillmentMethod2(
    fulfillmentMethod,
    // @ts-ignore
    { httpClient }
  );
}
function getFulfillmentMethod3(httpClient) {
  return (fulfillmentMethodId) => getFulfillmentMethod2(
    fulfillmentMethodId,
    // @ts-ignore
    { httpClient }
  );
}
function updateFulfillmentMethod3(httpClient) {
  return (_id, fulfillmentMethod) => updateFulfillmentMethod2(
    _id,
    fulfillmentMethod,
    // @ts-ignore
    { httpClient }
  );
}
function deleteFulfillmentMethod3(httpClient) {
  return (fulfillmentMethodId) => deleteFulfillmentMethod2(
    fulfillmentMethodId,
    // @ts-ignore
    { httpClient }
  );
}
function listFulfillmentMethods3(httpClient) {
  return (options) => listFulfillmentMethods2(
    options,
    // @ts-ignore
    { httpClient }
  );
}

// src/restaurants-fulfillments-v3-fulfillment-method-fulfillemt-methods.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var createFulfillmentMethod4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(createFulfillmentMethod3);
var getFulfillmentMethod4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getFulfillmentMethod3);
var updateFulfillmentMethod4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(updateFulfillmentMethod3);
var deleteFulfillmentMethod4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(deleteFulfillmentMethod3);
var listFulfillmentMethods4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(listFulfillmentMethods3);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  BestDeliveryByCategory,
  DayOfWeek,
  FulfillmentMethodType,
  TimeDisplayUnit,
  Type,
  createFulfillmentMethod,
  deleteFulfillmentMethod,
  getFulfillmentMethod,
  listFulfillmentMethods,
  updateFulfillmentMethod
});
