// @wix/auto_sdk_stores_catalog-imports-v-3/build/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  Alignment: () => Alignment,
  AppType: () => AppType,
  AspectRatio: () => AspectRatio,
  BackgroundType: () => BackgroundType,
  ButtonDataType: () => ButtonDataType,
  ChoiceType: () => ChoiceType,
  Crop: () => Crop,
  DecorationType: () => DecorationType,
  Direction: () => Direction,
  DiscountType: () => DiscountType,
  DividerDataAlignment: () => DividerDataAlignment,
  FileType: () => FileType,
  FontType: () => FontType,
  GIFType: () => GIFType,
  ImagePosition: () => ImagePosition,
  InitialExpandedItems: () => InitialExpandedItems,
  Layout: () => Layout,
  LayoutType: () => LayoutType,
  LineStyle: () => LineStyle,
  MapType: () => MapType,
  MeasurementUnit: () => MeasurementUnit,
  MediaType: () => MediaType,
  ModifierRenderType: () => ModifierRenderType,
  NodeType: () => NodeType,
  NullValue: () => NullValue,
  Orientation: () => Orientation,
  Placement: () => Placement,
  PluginContainerDataAlignment: () => PluginContainerDataAlignment,
  PollLayoutDirection: () => PollLayoutDirection,
  PollLayoutType: () => PollLayoutType,
  Position: () => Position,
  ProductOptionRenderType: () => ProductOptionRenderType,
  ProductType: () => ProductType,
  Resizing: () => Resizing,
  ResponsivenessBehaviour: () => ResponsivenessBehaviour,
  Scaling: () => Scaling,
  Source: () => Source,
  StylesPosition: () => StylesPosition,
  SubscriptionFrequency: () => SubscriptionFrequency,
  Target: () => Target,
  TextAlignment: () => TextAlignment,
  ThumbnailsAlignment: () => ThumbnailsAlignment,
  Type: () => Type,
  VerticalAlignment: () => VerticalAlignment,
  VerticalAlignmentAlignment: () => VerticalAlignmentAlignment,
  ViewMode: () => ViewMode,
  ViewRole: () => ViewRole,
  VoteRole: () => VoteRole,
  WebhookIdentityType: () => WebhookIdentityType,
  WeightUnit: () => WeightUnit,
  Width: () => Width,
  WidthType: () => WidthType,
  addProductsToImport: () => addProductsToImport2,
  createCatalogImport: () => createCatalogImport2,
  submitCatalogImport: () => submitCatalogImport2
});
module.exports = __toCommonJS(index_typings_exports);

// src/stores-catalog-v3-catalog-import-catalog-imports-v-3.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/stores-catalog-v3-catalog-import-catalog-imports-v-3.http.ts
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
function resolveWixStoresCatalogCatalogImportV3CatalogImportsUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/stores/v3/catalog-imports",
        destPath: "/v3/catalog-imports"
      }
    ]
  };
  return (0, import_rest_modules.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_stores_catalog-imports-v-3";
function createCatalogImport(payload) {
  function __createCatalogImport({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "catalogImport.createdDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.stores.catalog.v3.catalog_import",
      method: "POST",
      methodFqn: "wix.stores.catalog.catalog_import.v3.CatalogImports.CreateCatalogImport",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixStoresCatalogCatalogImportV3CatalogImportsUrl({
        protoPath: "/v3/catalog-imports",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "catalogImport.createdDate" }]
        }
      ])
    };
    return metadata;
  }
  return __createCatalogImport;
}
function addProductsToImport(payload) {
  function __addProductsToImport({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "products.fieldMask" }]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "products.product.physicalProperties.pricePerUnit.quantity" },
          {
            path: "products.product.physicalProperties.shippingWeightRange.minValue"
          },
          {
            path: "products.product.physicalProperties.shippingWeightRange.maxValue"
          },
          {
            path: "products.product.variantsInfo.variants.physicalProperties.pricePerUnit.settings.quantity"
          },
          {
            path: "products.product.variantsInfo.variants.revenueDetails.profitMargin"
          },
          {
            path: "products.product.subscriptionDetails.subscriptions.discount.percentOff"
          },
          {
            path: "products.product.description.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "products.product.description.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "products.product.description.nodes.galleryData.items.video.thumbnail.duration"
          },
          {
            path: "products.product.description.nodes.galleryData.options.item.ratio"
          },
          {
            path: "products.product.description.nodes.imageData.image.duration"
          },
          {
            path: "products.product.description.nodes.mapData.mapSettings.lat"
          },
          {
            path: "products.product.description.nodes.mapData.mapSettings.lng"
          },
          {
            path: "products.product.description.nodes.pollData.poll.image.duration"
          },
          {
            path: "products.product.description.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "products.product.description.nodes.pollData.design.poll.background.image.duration"
          },
          {
            path: "products.product.description.nodes.appEmbedData.image.duration"
          },
          {
            path: "products.product.description.nodes.videoData.video.duration"
          },
          {
            path: "products.product.description.nodes.videoData.thumbnail.duration"
          },
          {
            path: "products.product.description.nodes.audioData.audio.duration"
          },
          {
            path: "products.product.description.nodes.audioData.coverImage.duration"
          },
          {
            path: "products.product.description.nodes.layoutData.backgroundImage.media.duration"
          },
          {
            path: "products.product.description.nodes.layoutData.backdropImage.media.duration"
          },
          {
            path: "products.product.infoSections.description.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "products.product.infoSections.description.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "products.product.infoSections.description.nodes.galleryData.items.video.thumbnail.duration"
          },
          {
            path: "products.product.infoSections.description.nodes.galleryData.options.item.ratio"
          },
          {
            path: "products.product.infoSections.description.nodes.imageData.image.duration"
          },
          {
            path: "products.product.infoSections.description.nodes.mapData.mapSettings.lat"
          },
          {
            path: "products.product.infoSections.description.nodes.mapData.mapSettings.lng"
          },
          {
            path: "products.product.infoSections.description.nodes.pollData.poll.image.duration"
          },
          {
            path: "products.product.infoSections.description.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "products.product.infoSections.description.nodes.pollData.design.poll.background.image.duration"
          },
          {
            path: "products.product.infoSections.description.nodes.appEmbedData.image.duration"
          },
          {
            path: "products.product.infoSections.description.nodes.videoData.video.duration"
          },
          {
            path: "products.product.infoSections.description.nodes.videoData.thumbnail.duration"
          },
          {
            path: "products.product.infoSections.description.nodes.audioData.audio.duration"
          },
          {
            path: "products.product.infoSections.description.nodes.audioData.coverImage.duration"
          },
          {
            path: "products.product.infoSections.description.nodes.layoutData.backgroundImage.media.duration"
          },
          {
            path: "products.product.infoSections.description.nodes.layoutData.backdropImage.media.duration"
          },
          {
            path: "products.product.variantsInfo.variants.physicalProperties.weight"
          }
        ]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "products.product.description.metadata.createdTimestamp" },
          { path: "products.product.description.metadata.updatedTimestamp" },
          { path: "products.product.media.main.image.urlExpirationDate" },
          { path: "products.product.media.main.video.urlExpirationDate" },
          {
            path: "products.product.media.main.video.resolutions.urlExpirationDate"
          },
          {
            path: "products.product.media.main.video.resolutions.poster.urlExpirationDate"
          },
          {
            path: "products.product.media.main.video.posters.urlExpirationDate"
          },
          {
            path: "products.product.media.itemsInfo.items.image.urlExpirationDate"
          },
          {
            path: "products.product.media.itemsInfo.items.video.urlExpirationDate"
          },
          {
            path: "products.product.media.itemsInfo.items.video.resolutions.urlExpirationDate"
          },
          {
            path: "products.product.media.itemsInfo.items.video.resolutions.poster.urlExpirationDate"
          },
          {
            path: "products.product.media.itemsInfo.items.video.posters.urlExpirationDate"
          },
          {
            path: "products.product.options.choicesSettings.choices.image.urlExpirationDate"
          },
          {
            path: "products.product.options.choicesSettings.choices.linkedMedia.image.urlExpirationDate"
          },
          {
            path: "products.product.options.choicesSettings.choices.linkedMedia.video.urlExpirationDate"
          },
          {
            path: "products.product.options.choicesSettings.choices.linkedMedia.video.resolutions.urlExpirationDate"
          },
          {
            path: "products.product.options.choicesSettings.choices.linkedMedia.video.resolutions.poster.urlExpirationDate"
          },
          {
            path: "products.product.options.choicesSettings.choices.linkedMedia.video.posters.urlExpirationDate"
          },
          {
            path: "products.product.modifiers.choicesSettings.choices.image.urlExpirationDate"
          },
          {
            path: "products.product.modifiers.choicesSettings.choices.linkedMedia.image.urlExpirationDate"
          },
          {
            path: "products.product.modifiers.choicesSettings.choices.linkedMedia.video.urlExpirationDate"
          },
          {
            path: "products.product.modifiers.choicesSettings.choices.linkedMedia.video.resolutions.urlExpirationDate"
          },
          {
            path: "products.product.modifiers.choicesSettings.choices.linkedMedia.video.resolutions.poster.urlExpirationDate"
          },
          {
            path: "products.product.modifiers.choicesSettings.choices.linkedMedia.video.posters.urlExpirationDate"
          },
          {
            path: "products.product.infoSections.description.metadata.createdTimestamp"
          },
          {
            path: "products.product.infoSections.description.metadata.updatedTimestamp"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.stores.catalog.v3.catalog_import",
      method: "POST",
      methodFqn: "wix.stores.catalog.catalog_import.v3.CatalogImports.AddProductsToImport",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixStoresCatalogCatalogImportV3CatalogImportsUrl({
        protoPath: "/v3/catalog-imports/{catalogImportId}/add-products",
        data: serializedData,
        host
      }),
      data: serializedData
    };
    return metadata;
  }
  return __addProductsToImport;
}
function submitCatalogImport(payload) {
  function __submitCatalogImport({ host }) {
    const metadata = {
      entityFqdn: "wix.stores.catalog.v3.catalog_import",
      method: "POST",
      methodFqn: "wix.stores.catalog.catalog_import.v3.CatalogImports.SubmitCatalogImport",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixStoresCatalogCatalogImportV3CatalogImportsUrl({
        protoPath: "/v3/catalog-imports/{catalogImportId}/submit",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __submitCatalogImport;
}

// src/stores-catalog-v3-catalog-import-catalog-imports-v-3.universal.ts
var import_image = require("@wix/sdk-runtime/transformations/image");
var import_video_v2 = require("@wix/sdk-runtime/transformations/video-v2");
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var NodeType = /* @__PURE__ */ ((NodeType2) => {
  NodeType2["PARAGRAPH"] = "PARAGRAPH";
  NodeType2["TEXT"] = "TEXT";
  NodeType2["HEADING"] = "HEADING";
  NodeType2["BULLETED_LIST"] = "BULLETED_LIST";
  NodeType2["ORDERED_LIST"] = "ORDERED_LIST";
  NodeType2["LIST_ITEM"] = "LIST_ITEM";
  NodeType2["BLOCKQUOTE"] = "BLOCKQUOTE";
  NodeType2["CODE_BLOCK"] = "CODE_BLOCK";
  NodeType2["VIDEO"] = "VIDEO";
  NodeType2["DIVIDER"] = "DIVIDER";
  NodeType2["FILE"] = "FILE";
  NodeType2["GALLERY"] = "GALLERY";
  NodeType2["GIF"] = "GIF";
  NodeType2["HTML"] = "HTML";
  NodeType2["IMAGE"] = "IMAGE";
  NodeType2["LINK_PREVIEW"] = "LINK_PREVIEW";
  NodeType2["MAP"] = "MAP";
  NodeType2["POLL"] = "POLL";
  NodeType2["APP_EMBED"] = "APP_EMBED";
  NodeType2["BUTTON"] = "BUTTON";
  NodeType2["COLLAPSIBLE_LIST"] = "COLLAPSIBLE_LIST";
  NodeType2["TABLE"] = "TABLE";
  NodeType2["EMBED"] = "EMBED";
  NodeType2["COLLAPSIBLE_ITEM"] = "COLLAPSIBLE_ITEM";
  NodeType2["COLLAPSIBLE_ITEM_TITLE"] = "COLLAPSIBLE_ITEM_TITLE";
  NodeType2["COLLAPSIBLE_ITEM_BODY"] = "COLLAPSIBLE_ITEM_BODY";
  NodeType2["TABLE_CELL"] = "TABLE_CELL";
  NodeType2["TABLE_ROW"] = "TABLE_ROW";
  NodeType2["EXTERNAL"] = "EXTERNAL";
  NodeType2["AUDIO"] = "AUDIO";
  NodeType2["CAPTION"] = "CAPTION";
  NodeType2["LAYOUT"] = "LAYOUT";
  NodeType2["LAYOUT_CELL"] = "LAYOUT_CELL";
  return NodeType2;
})(NodeType || {});
var WidthType = /* @__PURE__ */ ((WidthType2) => {
  WidthType2["CONTENT"] = "CONTENT";
  WidthType2["SMALL"] = "SMALL";
  WidthType2["ORIGINAL"] = "ORIGINAL";
  WidthType2["FULL_WIDTH"] = "FULL_WIDTH";
  return WidthType2;
})(WidthType || {});
var PluginContainerDataAlignment = /* @__PURE__ */ ((PluginContainerDataAlignment2) => {
  PluginContainerDataAlignment2["CENTER"] = "CENTER";
  PluginContainerDataAlignment2["LEFT"] = "LEFT";
  PluginContainerDataAlignment2["RIGHT"] = "RIGHT";
  return PluginContainerDataAlignment2;
})(PluginContainerDataAlignment || {});
var ButtonDataType = /* @__PURE__ */ ((ButtonDataType2) => {
  ButtonDataType2["LINK"] = "LINK";
  ButtonDataType2["ACTION"] = "ACTION";
  return ButtonDataType2;
})(ButtonDataType || {});
var Target = /* @__PURE__ */ ((Target2) => {
  Target2["SELF"] = "SELF";
  Target2["BLANK"] = "BLANK";
  Target2["PARENT"] = "PARENT";
  Target2["TOP"] = "TOP";
  return Target2;
})(Target || {});
var TextAlignment = /* @__PURE__ */ ((TextAlignment2) => {
  TextAlignment2["AUTO"] = "AUTO";
  TextAlignment2["LEFT"] = "LEFT";
  TextAlignment2["RIGHT"] = "RIGHT";
  TextAlignment2["CENTER"] = "CENTER";
  TextAlignment2["JUSTIFY"] = "JUSTIFY";
  return TextAlignment2;
})(TextAlignment || {});
var LineStyle = /* @__PURE__ */ ((LineStyle2) => {
  LineStyle2["SINGLE"] = "SINGLE";
  LineStyle2["DOUBLE"] = "DOUBLE";
  LineStyle2["DASHED"] = "DASHED";
  LineStyle2["DOTTED"] = "DOTTED";
  return LineStyle2;
})(LineStyle || {});
var Width = /* @__PURE__ */ ((Width2) => {
  Width2["LARGE"] = "LARGE";
  Width2["MEDIUM"] = "MEDIUM";
  Width2["SMALL"] = "SMALL";
  return Width2;
})(Width || {});
var DividerDataAlignment = /* @__PURE__ */ ((DividerDataAlignment2) => {
  DividerDataAlignment2["CENTER"] = "CENTER";
  DividerDataAlignment2["LEFT"] = "LEFT";
  DividerDataAlignment2["RIGHT"] = "RIGHT";
  return DividerDataAlignment2;
})(DividerDataAlignment || {});
var ViewMode = /* @__PURE__ */ ((ViewMode2) => {
  ViewMode2["NONE"] = "NONE";
  ViewMode2["FULL"] = "FULL";
  ViewMode2["MINI"] = "MINI";
  return ViewMode2;
})(ViewMode || {});
var LayoutType = /* @__PURE__ */ ((LayoutType2) => {
  LayoutType2["COLLAGE"] = "COLLAGE";
  LayoutType2["MASONRY"] = "MASONRY";
  LayoutType2["GRID"] = "GRID";
  LayoutType2["THUMBNAIL"] = "THUMBNAIL";
  LayoutType2["SLIDER"] = "SLIDER";
  LayoutType2["SLIDESHOW"] = "SLIDESHOW";
  LayoutType2["PANORAMA"] = "PANORAMA";
  LayoutType2["COLUMN"] = "COLUMN";
  LayoutType2["MAGIC"] = "MAGIC";
  LayoutType2["FULLSIZE"] = "FULLSIZE";
  return LayoutType2;
})(LayoutType || {});
var Orientation = /* @__PURE__ */ ((Orientation2) => {
  Orientation2["ROWS"] = "ROWS";
  Orientation2["COLUMNS"] = "COLUMNS";
  return Orientation2;
})(Orientation || {});
var Crop = /* @__PURE__ */ ((Crop2) => {
  Crop2["FILL"] = "FILL";
  Crop2["FIT"] = "FIT";
  return Crop2;
})(Crop || {});
var ThumbnailsAlignment = /* @__PURE__ */ ((ThumbnailsAlignment2) => {
  ThumbnailsAlignment2["TOP"] = "TOP";
  ThumbnailsAlignment2["RIGHT"] = "RIGHT";
  ThumbnailsAlignment2["BOTTOM"] = "BOTTOM";
  ThumbnailsAlignment2["LEFT"] = "LEFT";
  ThumbnailsAlignment2["NONE"] = "NONE";
  return ThumbnailsAlignment2;
})(ThumbnailsAlignment || {});
var GIFType = /* @__PURE__ */ ((GIFType2) => {
  GIFType2["NORMAL"] = "NORMAL";
  GIFType2["STICKER"] = "STICKER";
  return GIFType2;
})(GIFType || {});
var Source = /* @__PURE__ */ ((Source2) => {
  Source2["HTML"] = "HTML";
  Source2["ADSENSE"] = "ADSENSE";
  Source2["AI"] = "AI";
  return Source2;
})(Source || {});
var StylesPosition = /* @__PURE__ */ ((StylesPosition2) => {
  StylesPosition2["START"] = "START";
  StylesPosition2["END"] = "END";
  StylesPosition2["TOP"] = "TOP";
  StylesPosition2["HIDDEN"] = "HIDDEN";
  return StylesPosition2;
})(StylesPosition || {});
var MapType = /* @__PURE__ */ ((MapType2) => {
  MapType2["ROADMAP"] = "ROADMAP";
  MapType2["SATELITE"] = "SATELITE";
  MapType2["HYBRID"] = "HYBRID";
  MapType2["TERRAIN"] = "TERRAIN";
  return MapType2;
})(MapType || {});
var ViewRole = /* @__PURE__ */ ((ViewRole2) => {
  ViewRole2["CREATOR"] = "CREATOR";
  ViewRole2["VOTERS"] = "VOTERS";
  ViewRole2["EVERYONE"] = "EVERYONE";
  return ViewRole2;
})(ViewRole || {});
var VoteRole = /* @__PURE__ */ ((VoteRole2) => {
  VoteRole2["SITE_MEMBERS"] = "SITE_MEMBERS";
  VoteRole2["ALL"] = "ALL";
  return VoteRole2;
})(VoteRole || {});
var PollLayoutType = /* @__PURE__ */ ((PollLayoutType2) => {
  PollLayoutType2["LIST"] = "LIST";
  PollLayoutType2["GRID"] = "GRID";
  return PollLayoutType2;
})(PollLayoutType || {});
var PollLayoutDirection = /* @__PURE__ */ ((PollLayoutDirection2) => {
  PollLayoutDirection2["LTR"] = "LTR";
  PollLayoutDirection2["RTL"] = "RTL";
  return PollLayoutDirection2;
})(PollLayoutDirection || {});
var BackgroundType = /* @__PURE__ */ ((BackgroundType2) => {
  BackgroundType2["COLOR"] = "COLOR";
  BackgroundType2["IMAGE"] = "IMAGE";
  BackgroundType2["GRADIENT"] = "GRADIENT";
  return BackgroundType2;
})(BackgroundType || {});
var DecorationType = /* @__PURE__ */ ((DecorationType2) => {
  DecorationType2["BOLD"] = "BOLD";
  DecorationType2["ITALIC"] = "ITALIC";
  DecorationType2["UNDERLINE"] = "UNDERLINE";
  DecorationType2["SPOILER"] = "SPOILER";
  DecorationType2["ANCHOR"] = "ANCHOR";
  DecorationType2["MENTION"] = "MENTION";
  DecorationType2["LINK"] = "LINK";
  DecorationType2["COLOR"] = "COLOR";
  DecorationType2["FONT_SIZE"] = "FONT_SIZE";
  DecorationType2["EXTERNAL"] = "EXTERNAL";
  DecorationType2["STRIKETHROUGH"] = "STRIKETHROUGH";
  DecorationType2["SUPERSCRIPT"] = "SUPERSCRIPT";
  DecorationType2["SUBSCRIPT"] = "SUBSCRIPT";
  return DecorationType2;
})(DecorationType || {});
var FontType = /* @__PURE__ */ ((FontType2) => {
  FontType2["PX"] = "PX";
  FontType2["EM"] = "EM";
  return FontType2;
})(FontType || {});
var Position = /* @__PURE__ */ ((Position2) => {
  Position2["START"] = "START";
  Position2["END"] = "END";
  Position2["TOP"] = "TOP";
  return Position2;
})(Position || {});
var AspectRatio = /* @__PURE__ */ ((AspectRatio2) => {
  AspectRatio2["SQUARE"] = "SQUARE";
  AspectRatio2["RECTANGLE"] = "RECTANGLE";
  return AspectRatio2;
})(AspectRatio || {});
var Resizing = /* @__PURE__ */ ((Resizing2) => {
  Resizing2["FILL"] = "FILL";
  Resizing2["FIT"] = "FIT";
  return Resizing2;
})(Resizing || {});
var Placement = /* @__PURE__ */ ((Placement2) => {
  Placement2["IMAGE"] = "IMAGE";
  Placement2["PRODUCT_INFO"] = "PRODUCT_INFO";
  return Placement2;
})(Placement || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["CONTAINED"] = "CONTAINED";
  Type2["FRAMELESS"] = "FRAMELESS";
  return Type2;
})(Type || {});
var Alignment = /* @__PURE__ */ ((Alignment2) => {
  Alignment2["START"] = "START";
  Alignment2["CENTER"] = "CENTER";
  Alignment2["END"] = "END";
  return Alignment2;
})(Alignment || {});
var Layout = /* @__PURE__ */ ((Layout2) => {
  Layout2["STACKED"] = "STACKED";
  Layout2["SIDE_BY_SIDE"] = "SIDE_BY_SIDE";
  return Layout2;
})(Layout || {});
var AppType = /* @__PURE__ */ ((AppType2) => {
  AppType2["PRODUCT"] = "PRODUCT";
  AppType2["EVENT"] = "EVENT";
  AppType2["BOOKING"] = "BOOKING";
  return AppType2;
})(AppType || {});
var InitialExpandedItems = /* @__PURE__ */ ((InitialExpandedItems2) => {
  InitialExpandedItems2["FIRST"] = "FIRST";
  InitialExpandedItems2["ALL"] = "ALL";
  InitialExpandedItems2["NONE"] = "NONE";
  return InitialExpandedItems2;
})(InitialExpandedItems || {});
var Direction = /* @__PURE__ */ ((Direction2) => {
  Direction2["LTR"] = "LTR";
  Direction2["RTL"] = "RTL";
  return Direction2;
})(Direction || {});
var VerticalAlignment = /* @__PURE__ */ ((VerticalAlignment2) => {
  VerticalAlignment2["TOP"] = "TOP";
  VerticalAlignment2["MIDDLE"] = "MIDDLE";
  VerticalAlignment2["BOTTOM"] = "BOTTOM";
  return VerticalAlignment2;
})(VerticalAlignment || {});
var NullValue = /* @__PURE__ */ ((NullValue2) => {
  NullValue2["NULL_VALUE"] = "NULL_VALUE";
  return NullValue2;
})(NullValue || {});
var Scaling = /* @__PURE__ */ ((Scaling2) => {
  Scaling2["AUTO"] = "AUTO";
  Scaling2["CONTAIN"] = "CONTAIN";
  Scaling2["COVER"] = "COVER";
  return Scaling2;
})(Scaling || {});
var ImagePosition = /* @__PURE__ */ ((ImagePosition2) => {
  ImagePosition2["CENTER"] = "CENTER";
  ImagePosition2["CENTER_LEFT"] = "CENTER_LEFT";
  ImagePosition2["CENTER_RIGHT"] = "CENTER_RIGHT";
  ImagePosition2["TOP"] = "TOP";
  ImagePosition2["TOP_LEFT"] = "TOP_LEFT";
  ImagePosition2["TOP_RIGHT"] = "TOP_RIGHT";
  ImagePosition2["BOTTOM"] = "BOTTOM";
  ImagePosition2["BOTTOM_LEFT"] = "BOTTOM_LEFT";
  ImagePosition2["BOTTOM_RIGHT"] = "BOTTOM_RIGHT";
  return ImagePosition2;
})(ImagePosition || {});
var VerticalAlignmentAlignment = /* @__PURE__ */ ((VerticalAlignmentAlignment2) => {
  VerticalAlignmentAlignment2["TOP"] = "TOP";
  VerticalAlignmentAlignment2["MIDDLE"] = "MIDDLE";
  VerticalAlignmentAlignment2["BOTTOM"] = "BOTTOM";
  return VerticalAlignmentAlignment2;
})(VerticalAlignmentAlignment || {});
var ResponsivenessBehaviour = /* @__PURE__ */ ((ResponsivenessBehaviour2) => {
  ResponsivenessBehaviour2["STACK"] = "STACK";
  ResponsivenessBehaviour2["WRAP"] = "WRAP";
  return ResponsivenessBehaviour2;
})(ResponsivenessBehaviour || {});
var MediaType = /* @__PURE__ */ ((MediaType2) => {
  MediaType2["UNKNOWN_MEDIA_TYPE"] = "UNKNOWN_MEDIA_TYPE";
  MediaType2["IMAGE"] = "IMAGE";
  MediaType2["VIDEO"] = "VIDEO";
  return MediaType2;
})(MediaType || {});
var ProductOptionRenderType = /* @__PURE__ */ ((ProductOptionRenderType2) => {
  ProductOptionRenderType2["UNKNOWN_OPTION_RENDER_TYPE"] = "UNKNOWN_OPTION_RENDER_TYPE";
  ProductOptionRenderType2["TEXT_CHOICES"] = "TEXT_CHOICES";
  ProductOptionRenderType2["SWATCH_CHOICES"] = "SWATCH_CHOICES";
  return ProductOptionRenderType2;
})(ProductOptionRenderType || {});
var ChoiceType = /* @__PURE__ */ ((ChoiceType2) => {
  ChoiceType2["UNKNOWN_CHOICE_TYPE"] = "UNKNOWN_CHOICE_TYPE";
  ChoiceType2["CHOICE_TEXT"] = "CHOICE_TEXT";
  ChoiceType2["ONE_COLOR"] = "ONE_COLOR";
  ChoiceType2["MULTIPLE_COLORS"] = "MULTIPLE_COLORS";
  ChoiceType2["IMAGE"] = "IMAGE";
  return ChoiceType2;
})(ChoiceType || {});
var ModifierRenderType = /* @__PURE__ */ ((ModifierRenderType2) => {
  ModifierRenderType2["UNKNOWN_MODIFIER_RENDER_TYPE"] = "UNKNOWN_MODIFIER_RENDER_TYPE";
  ModifierRenderType2["FREE_TEXT"] = "FREE_TEXT";
  ModifierRenderType2["TEXT_CHOICES"] = "TEXT_CHOICES";
  ModifierRenderType2["SWATCH_CHOICES"] = "SWATCH_CHOICES";
  return ModifierRenderType2;
})(ModifierRenderType || {});
var ProductType = /* @__PURE__ */ ((ProductType2) => {
  ProductType2["UNKNOWN_PRODUCT_TYPE"] = "UNKNOWN_PRODUCT_TYPE";
  ProductType2["PHYSICAL"] = "PHYSICAL";
  ProductType2["DIGITAL"] = "DIGITAL";
  return ProductType2;
})(ProductType || {});
var MeasurementUnit = /* @__PURE__ */ ((MeasurementUnit2) => {
  MeasurementUnit2["UNSPECIFIED"] = "UNSPECIFIED";
  MeasurementUnit2["ML"] = "ML";
  MeasurementUnit2["CL"] = "CL";
  MeasurementUnit2["L"] = "L";
  MeasurementUnit2["CBM"] = "CBM";
  MeasurementUnit2["MG"] = "MG";
  MeasurementUnit2["G"] = "G";
  MeasurementUnit2["KG"] = "KG";
  MeasurementUnit2["MM"] = "MM";
  MeasurementUnit2["CM"] = "CM";
  MeasurementUnit2["M"] = "M";
  MeasurementUnit2["SQM"] = "SQM";
  MeasurementUnit2["OZ"] = "OZ";
  MeasurementUnit2["LB"] = "LB";
  MeasurementUnit2["FLOZ"] = "FLOZ";
  MeasurementUnit2["PT"] = "PT";
  MeasurementUnit2["QT"] = "QT";
  MeasurementUnit2["GAL"] = "GAL";
  MeasurementUnit2["IN"] = "IN";
  MeasurementUnit2["FT"] = "FT";
  MeasurementUnit2["YD"] = "YD";
  MeasurementUnit2["SQFT"] = "SQFT";
  return MeasurementUnit2;
})(MeasurementUnit || {});
var WeightUnit = /* @__PURE__ */ ((WeightUnit2) => {
  WeightUnit2["UNSPECIFIED_WEIGHT_UNIT"] = "UNSPECIFIED_WEIGHT_UNIT";
  WeightUnit2["KG"] = "KG";
  WeightUnit2["LB"] = "LB";
  return WeightUnit2;
})(WeightUnit || {});
var FileType = /* @__PURE__ */ ((FileType2) => {
  FileType2["UNSPECIFIED"] = "UNSPECIFIED";
  FileType2["SECURE_PICTURE"] = "SECURE_PICTURE";
  FileType2["SECURE_VIDEO"] = "SECURE_VIDEO";
  FileType2["SECURE_DOCUMENT"] = "SECURE_DOCUMENT";
  FileType2["SECURE_MUSIC"] = "SECURE_MUSIC";
  FileType2["SECURE_ARCHIVE"] = "SECURE_ARCHIVE";
  return FileType2;
})(FileType || {});
var SubscriptionFrequency = /* @__PURE__ */ ((SubscriptionFrequency2) => {
  SubscriptionFrequency2["UNDEFINED"] = "UNDEFINED";
  SubscriptionFrequency2["DAY"] = "DAY";
  SubscriptionFrequency2["WEEK"] = "WEEK";
  SubscriptionFrequency2["MONTH"] = "MONTH";
  SubscriptionFrequency2["YEAR"] = "YEAR";
  return SubscriptionFrequency2;
})(SubscriptionFrequency || {});
var DiscountType = /* @__PURE__ */ ((DiscountType2) => {
  DiscountType2["UNKNOWN_DISCOUNT"] = "UNKNOWN_DISCOUNT";
  DiscountType2["AMOUNT"] = "AMOUNT";
  DiscountType2["PERCENT"] = "PERCENT";
  return DiscountType2;
})(DiscountType || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createCatalogImport2(catalogImport) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogImport
  });
  const reqOpts = createCatalogImport(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.catalogImport;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { catalogImport: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["catalogImport"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function addProductsToImport2(catalogImportId, products) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      catalogImportId,
      products
    }),
    [
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [
          { path: "products.product.media.main.image" },
          { path: "products.product.media.itemsInfo.items.image" },
          { path: "products.product.options.choicesSettings.choices.image" },
          {
            path: "products.product.options.choicesSettings.choices.linkedMedia.image"
          },
          { path: "products.product.modifiers.choicesSettings.choices.image" },
          {
            path: "products.product.modifiers.choicesSettings.choices.linkedMedia.image"
          }
        ]
      },
      {
        transformFn: import_video_v2.transformSDKVideoV2ToRESTVideoV2,
        paths: [
          { path: "products.product.media.main.video" },
          { path: "products.product.media.itemsInfo.items.video" },
          {
            path: "products.product.options.choicesSettings.choices.linkedMedia.video"
          },
          {
            path: "products.product.modifiers.choicesSettings.choices.linkedMedia.video"
          }
        ]
      }
    ]
  );
  const reqOpts = addProductsToImport(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { catalogImportId: "$[0]", products: "$[1]" },
        singleArgumentUnchanged: false
      },
      ["catalogImportId", "products"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function submitCatalogImport2(catalogImportId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    catalogImportId
  });
  const reqOpts = submitCatalogImport(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { catalogImportId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["catalogImportId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Alignment,
  AppType,
  AspectRatio,
  BackgroundType,
  ButtonDataType,
  ChoiceType,
  Crop,
  DecorationType,
  Direction,
  DiscountType,
  DividerDataAlignment,
  FileType,
  FontType,
  GIFType,
  ImagePosition,
  InitialExpandedItems,
  Layout,
  LayoutType,
  LineStyle,
  MapType,
  MeasurementUnit,
  MediaType,
  ModifierRenderType,
  NodeType,
  NullValue,
  Orientation,
  Placement,
  PluginContainerDataAlignment,
  PollLayoutDirection,
  PollLayoutType,
  Position,
  ProductOptionRenderType,
  ProductType,
  Resizing,
  ResponsivenessBehaviour,
  Scaling,
  Source,
  StylesPosition,
  SubscriptionFrequency,
  Target,
  TextAlignment,
  ThumbnailsAlignment,
  Type,
  VerticalAlignment,
  VerticalAlignmentAlignment,
  ViewMode,
  ViewRole,
  VoteRole,
  WebhookIdentityType,
  WeightUnit,
  Width,
  WidthType,
  addProductsToImport,
  createCatalogImport,
  submitCatalogImport
});
