// @wix/auto_sdk_benefit-programs_program-definitions/build/internal/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  Cascade: () => Cascade,
  SortOrder: () => SortOrder,
  WebhookIdentityType: () => WebhookIdentityType,
  createProgramDefinition: () => createProgramDefinition2,
  deleteProgramDefinition: () => deleteProgramDefinition2,
  getProgramDefinition: () => getProgramDefinition2,
  getProgramDefinitionByExternalIdAndNamespace: () => getProgramDefinitionByExternalIdAndNamespace2,
  queryProgramDefinitions: () => queryProgramDefinitions2,
  updateProgramDefinition: () => updateProgramDefinition2
});
module.exports = __toCommonJS(index_typings_exports);

// src/benefit-programs-v1-program-definition-program-definitions.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/benefit-programs-v1-program-definition-program-definitions.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixBenefitProgramsV1ProgramDefinitionProgramDefinitionServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/program-definitions",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/benefit-programs-program-definitions",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-program-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/program-definitions",
        destPath: "/v1/program-definitions"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-program-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/program-definitions",
        destPath: "/v1/program-definitions"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/benefit-programs-program-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/program-definitions",
        destPath: "/v1/program-definitions"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/benefit-programs-program-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/program-definitions",
        destPath: "/v1/program-definitions"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-program-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/program-definitions",
        destPath: "/v1/program-definitions"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/benefit-programs-program-definitions",
        destPath: ""
      },
      {
        srcPath: "/_api/benefit-programs/v1/program-definitions",
        destPath: "/v1/program-definitions"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/_api/benefit-programs/v1/program-definitions",
        destPath: "/v1/program-definitions"
      },
      {
        srcPath: "/benefit-programs/v1/program-definitions",
        destPath: "/v1/program-definitions"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/benefit-programs/v1/program-definitions",
        destPath: "/v1/program-definitions"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/benefit-programs/v1/program-definitions",
        destPath: "/v1/program-definitions"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_benefit-programs_program-definitions";
function createProgramDefinition(payload) {
  function __createProgramDefinition({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "programDefinition.createdDate" },
          { path: "programDefinition.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program_definition",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.program_definition.ProgramDefinitionService.CreateProgramDefinition",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramDefinitionProgramDefinitionServiceUrl(
        { protoPath: "/v1/program-definitions", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "programDefinition.createdDate" },
            { path: "programDefinition.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createProgramDefinition;
}
function updateProgramDefinition(payload) {
  function __updateProgramDefinition({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "programDefinition.createdDate" },
          { path: "programDefinition.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program_definition",
      method: "PATCH",
      methodFqn: "wix.benefit_programs.v1.program_definition.ProgramDefinitionService.UpdateProgramDefinition",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramDefinitionProgramDefinitionServiceUrl(
        {
          protoPath: "/v1/program-definitions/{programDefinition.id}",
          data: serializedData,
          host
        }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "programDefinition.createdDate" },
            { path: "programDefinition.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateProgramDefinition;
}
function deleteProgramDefinition(payload) {
  function __deleteProgramDefinition({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program_definition",
      method: "DELETE",
      methodFqn: "wix.benefit_programs.v1.program_definition.ProgramDefinitionService.DeleteProgramDefinition",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramDefinitionProgramDefinitionServiceUrl(
        {
          protoPath: "/v1/program-definitions/{programDefinitionId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteProgramDefinition;
}
function getProgramDefinition(payload) {
  function __getProgramDefinition({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program_definition",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.program_definition.ProgramDefinitionService.GetProgramDefinition",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramDefinitionProgramDefinitionServiceUrl(
        {
          protoPath: "/v1/program-definitions/{programDefinitionId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "programDefinition.createdDate" },
            { path: "programDefinition.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getProgramDefinition;
}
function getProgramDefinitionByExternalIdAndNamespace(payload) {
  function __getProgramDefinitionByExternalIdAndNamespace({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program_definition",
      method: "GET",
      methodFqn: "wix.benefit_programs.v1.program_definition.ProgramDefinitionService.GetProgramDefinitionByExternalIdAndNamespace",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramDefinitionProgramDefinitionServiceUrl(
        {
          protoPath: "/v1/program-definitions/by-namespace-and-external-id",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "programDefinition.createdDate" },
            { path: "programDefinition.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getProgramDefinitionByExternalIdAndNamespace;
}
function queryProgramDefinitions(payload) {
  function __queryProgramDefinitions({ host }) {
    const metadata = {
      entityFqdn: "wix.benefit_programs.v1.program_definition",
      method: "POST",
      methodFqn: "wix.benefit_programs.v1.program_definition.ProgramDefinitionService.QueryProgramDefinitions",
      packageName: PACKAGE_NAME,
      url: resolveWixBenefitProgramsV1ProgramDefinitionProgramDefinitionServiceUrl(
        { protoPath: "/v1/program-definitions/query", data: payload, host }
      ),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "programDefinitions.createdDate" },
            { path: "programDefinitions.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryProgramDefinitions;
}

// src/benefit-programs-v1-program-definition-program-definitions.universal.ts
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var Cascade = /* @__PURE__ */ ((Cascade2) => {
  Cascade2["UNKNOWN_CASCADE"] = "UNKNOWN_CASCADE";
  Cascade2["NEXT_RENEWAL"] = "NEXT_RENEWAL";
  Cascade2["IMMEDIATELY"] = "IMMEDIATELY";
  Cascade2["FUTURE_PROVISIONS"] = "FUTURE_PROVISIONS";
  return Cascade2;
})(Cascade || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createProgramDefinition2(programDefinition) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    programDefinition
  });
  const reqOpts = createProgramDefinition(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.programDefinition;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { programDefinition: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["programDefinition"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateProgramDefinition2(_id, programDefinition) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    programDefinition: { ...programDefinition, id: _id }
  });
  const reqOpts = updateProgramDefinition(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.programDefinition;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { programDefinition: "$[1]" },
        explicitPathsToArguments: { "programDefinition.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "programDefinition"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteProgramDefinition2(programDefinitionId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    programDefinitionId,
    cascadeType: options?.cascadeType
  });
  const reqOpts = deleteProgramDefinition(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          programDefinitionId: "$[0]",
          cascadeType: "$[1].cascadeType"
        },
        singleArgumentUnchanged: false
      },
      ["programDefinitionId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getProgramDefinition2(programDefinitionId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    programDefinitionId
  });
  const reqOpts = getProgramDefinition(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.programDefinition;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { programDefinitionId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["programDefinitionId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getProgramDefinitionByExternalIdAndNamespace2(namespace, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    namespace,
    externalId: options?.externalId
  });
  const reqOpts = getProgramDefinitionByExternalIdAndNamespace(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          namespace: "$[0]",
          externalId: "$[1].externalId"
        },
        singleArgumentUnchanged: false
      },
      ["namespace", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryProgramDefinitions2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryProgramDefinitions(
        payload
      );
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [])
      );
      return {
        items: transformedData?.programDefinitions,
        pagingMetadata: transformedData?.metadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Cascade,
  SortOrder,
  WebhookIdentityType,
  createProgramDefinition,
  deleteProgramDefinition,
  getProgramDefinition,
  getProgramDefinitionByExternalIdAndNamespace,
  queryProgramDefinitions,
  updateProgramDefinition
});
