// @wix/patterns/dist/cjs/components/CollectionViewsDropdown/AutoCompleteReadonly.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.AutoCompleteReadonly = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _designSystem = require("@wix/design-system");
var _react = _interopRequireWildcard(require("react"));
var _mobxReactLite = require("mobx-react-lite");
var _AutoCompleteReadonlyState = require("./AutoCompleteReadonlyState");
var _AutoCompleteReadonlyStCss = require("./AutoCompleteReadonly.st.css.js");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CollectionViewsDropdown/AutoCompleteReadonly.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _AutoCompleteReadonly(props) {
  var _String;
  const {
    value,
    selectedLabel,
    dropdownWidth,
    clearInput
  } = props;
  const [state] = (0, _react.useState)(() => props.state ?? new _AutoCompleteReadonlyState.AutoCompleteReadonlyState());
  (0, _react.useEffect)(() => {
    state.clearInput = clearInput;
  }, [clearInput]);
  const {
    disableEditing
  } = state;
  const valueOrSelected = (0, _react.useMemo)(() => {
    return disableEditing ? selectedLabel || '' : value;
  }, [value, disableEditing, selectedLabel]);
  return /*#__PURE__*/_react.default.createElement(_designSystem.AutoComplete, (0, _extends2.default)({}, props, {
    dropdownWidth: dropdownWidth,
    onSelect: (option, sameOptionWasPicked) => {
      props.onSelect == null || props.onSelect(option, sameOptionWasPicked);
      state.onSelect();
    },
    onFocus: state.onFocus,
    onBlur: e => {
      props.onBlur == null || props.onBlur(e);
      state.onBlur();
    },
    onClickOutside: e => {
      props.onClickOutside == null || props.onClickOutside(e);
      state.onClickOutside();
    },
    suffix: !disableEditing && (_String = String(value)) != null && _String.trim() ? undefined : props.suffix,
    placeholder: !disableEditing && selectedLabel != null ? String(selectedLabel) : props.placeholder,
    onKeyDown: e => {
      if (e.nativeEvent.code === 'Tab') {
        state.onTabKeyDown();
      }
    },
    clearButton: !disableEditing,
    onClear: !disableEditing ? e => {
      props.onClear == null || props.onClear(e);
      state.onInputClearButtonClick();
    } : undefined,
    value: valueOrSelected,
    onOptionsShow: state.onOptionsShow,
    focusOnClearClick: true,
    className: (0, _AutoCompleteReadonlyStCss.st)(_AutoCompleteReadonlyStCss.classes.root, {
      disableEditing
    }),
    inputElement: /*#__PURE__*/_react.default.createElement(_designSystem.Input, {
      disableEditing: disableEditing,
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 75,
        columnNumber: 21
      }
    }),
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 31,
      columnNumber: 5
    }
  }));
}
const AutoCompleteReadonly = exports.AutoCompleteReadonly = (0, _mobxReactLite.observer)(_AutoCompleteReadonly);
