// @wix/patterns/dist/cjs/components/CollectionViewsDropdown/CollectionViewsDropdownBase.js
"use strict";

exports.__esModule = true;
exports.CollectionViewsDropdownBase = void 0;
var _react = _interopRequireWildcard(require("react"));
var _ViewsDropdown = require("./ViewsDropdown");
var _setViewsState = require("../../state/setViewsState");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/CollectionViewsDropdown/CollectionViewsDropdownBase.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
const CollectionViewsDropdownBase = props => {
  const {
    toolbar,
    showTotal,
    onSelect,
    popoverProps,
    ...rest
  } = props;
  const [state] = (0, _react.useState)(() => (0, _setViewsState.setViewsState)(toolbar, rest));
  return /*#__PURE__*/_react.default.createElement(_ViewsDropdown.ViewsDropdown, {
    dataHook: "collection-views",
    state: state,
    showTotal: showTotal,
    onSelect: onSelect,
    popoverProps: state ? popoverProps : undefined,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 94,
      columnNumber: 5
    }
  });
};
exports.CollectionViewsDropdownBase = CollectionViewsDropdownBase;
