// @wix/ricos/dist/cjs/plugin-vertical-embed/components/Card.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var Card_exports = {};
__export(Card_exports, {
  default: () => Card_default
});
module.exports = __toCommonJS(Card_exports);
var import_classnames = __toESM(require("classnames"));
var import_react = __toESM(require("react"));
var import_ricos_context = require("./../../ricos-context/index.js");
var import_imageUtils = require("./../../wix-rich-content-common/lib/imageUtils.js");
var import_widget_scss = __toESM(require("../statics/styles/widget-scss.js"));
var import_CardContent = require("./CardContent.js");
const getImageTargetHeight = (width, height, targetWidth) => {
  const proportion = height / width;
  const targetHeight = proportion * targetWidth;
  return targetHeight;
};
const Card = (props) => {
  const { tabIndex } = (0, import_react.useContext)(import_ricos_context.RicosContext);
  const getImageSrc = ({ targetWidth }) => {
    if (props.image && props.image.width && props.image.height && props.image.src?.url) {
      const targetHeight = getImageTargetHeight(props.image.width, props.image.height, targetWidth);
      return (0, import_imageUtils.getScaleImageSrc)(props.image.src.url, targetWidth, targetHeight);
    }
    return props.image?.src?.url;
  };
  const [imageSrc, setImageSrc] = (0, import_react.useState)(
    () => (
      // value for ssr and first render
      getImageSrc({ targetWidth: 185 })
    )
  );
  const setRef = (ref) => {
    if (ref && props.image?.width && props.image?.height) {
      const targetWidth = ref.getBoundingClientRect()?.width;
      setImageSrc(getImageSrc({ targetWidth }));
    }
  };
  const imageStyle = imageSrc ? { backgroundImage: `url(${imageSrc})` } : {};
  return (
    // eslint-disable-next-line react/jsx-no-target-blank
    /* @__PURE__ */ import_react.default.createElement(
      "a",
      {
        className: import_widget_scss.default.link,
        href: props.disabled ? void 0 : props.url,
        target: "_blank",
        ref: setRef,
        tabIndex
      },
      /* @__PURE__ */ import_react.default.createElement(
        "div",
        {
          style: { direction: props.direction },
          className: (0, import_classnames.default)(import_widget_scss.default[props.direction], import_widget_scss.default.container, import_widget_scss.default.cardLayout)
        },
        imageSrc && /* @__PURE__ */ import_react.default.createElement("div", { style: imageStyle, className: import_widget_scss.default.image }),
        /* @__PURE__ */ import_react.default.createElement(
          import_CardContent.CardContent,
          {
            info: props.subtitleInfo,
            title: props.title,
            buttonText: props.buttonText,
            disabled: props.disabled,
            t: props.t
          }
        )
      )
    )
  );
};
var Card_default = Card;
