// @wix/auto_sdk_restaurants_recipients/build/internal/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  Channel: () => Channel,
  SortOrder: () => SortOrder,
  WebhookIdentityType: () => WebhookIdentityType,
  bulkUpdateRecipientTags: () => bulkUpdateRecipientTags2,
  bulkUpdateRecipientTagsByFilter: () => bulkUpdateRecipientTagsByFilter2,
  createRecipient: () => createRecipient2,
  deleteRecipient: () => deleteRecipient2,
  getRecipient: () => getRecipient2,
  queryRecipients: () => queryRecipients2,
  updateRecipient: () => updateRecipient2
});
module.exports = __toCommonJS(index_typings_exports);

// src/restaurants-recipient-v1-recipient-recipients.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/restaurants-recipient-v1-recipient-recipients.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/rest-notification-recipients",
        destPath: ""
      }
    ],
    "dev._base_domain_": [
      {
        srcPath: "/_api/rest-notification-recipients",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/rest-notification-recipients/v1",
        destPath: "/v1"
      },
      {
        srcPath: "/restaurants/recipient/v1/recipients",
        destPath: "/v1/recipients"
      }
    ],
    _: [
      {
        srcPath: "/_api/rest-notification-recipients",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/rest-notification-recipients",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_recipients";
function createRecipient(payload) {
  function __createRecipient({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "recipient.createdDate" },
          { path: "recipient.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.recipient.v1.recipient",
      method: "POST",
      methodFqn: "wix.restaurants.recipient.v1.NotificationRecipientsService.CreateRecipient",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl({
        protoPath: "/v1/recipients",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "recipient.createdDate" },
            { path: "recipient.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createRecipient;
}
function getRecipient(payload) {
  function __getRecipient({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.recipient.v1.recipient",
      method: "GET",
      methodFqn: "wix.restaurants.recipient.v1.NotificationRecipientsService.GetRecipient",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl({
        protoPath: "/v1/recipients/{recipientId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "recipient.createdDate" },
            { path: "recipient.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getRecipient;
}
function updateRecipient(payload) {
  function __updateRecipient({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "recipient.createdDate" },
          { path: "recipient.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.recipient.v1.recipient",
      method: "PATCH",
      methodFqn: "wix.restaurants.recipient.v1.NotificationRecipientsService.UpdateRecipient",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl({
        protoPath: "/v1/recipients/{recipient.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "recipient.createdDate" },
            { path: "recipient.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateRecipient;
}
function deleteRecipient(payload) {
  function __deleteRecipient({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.recipient.v1.recipient",
      method: "DELETE",
      methodFqn: "wix.restaurants.recipient.v1.NotificationRecipientsService.DeleteRecipient",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl({
        protoPath: "/v1/recipients/{recipientId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteRecipient;
}
function queryRecipients(payload) {
  function __queryRecipients({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.recipient.v1.recipient",
      method: "GET",
      methodFqn: "wix.restaurants.recipient.v1.NotificationRecipientsService.QueryRecipients",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl({
        protoPath: "/v1/recipients/query",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload, true),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "recipients.createdDate" },
            { path: "recipients.updatedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl(
            { protoPath: "/v1/recipients/query", data: payload, host }
          ),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryRecipients;
}
function bulkUpdateRecipientTags(payload) {
  function __bulkUpdateRecipientTags({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.recipient.v1.recipient",
      method: "POST",
      methodFqn: "wix.restaurants.recipient.v1.NotificationRecipientsService.BulkUpdateRecipientTags",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl({
        protoPath: "/v1/bulk/recipients/update-tags",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateRecipientTags;
}
function bulkUpdateRecipientTagsByFilter(payload) {
  function __bulkUpdateRecipientTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.recipient.v1.recipient",
      method: "POST",
      methodFqn: "wix.restaurants.recipient.v1.NotificationRecipientsService.BulkUpdateRecipientTagsByFilter",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl({
        protoPath: "/v1/bulk/recipients/update-tags-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateRecipientTagsByFilter;
}

// src/restaurants-recipient-v1-recipient-recipients.universal.ts
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var Channel = /* @__PURE__ */ ((Channel2) => {
  Channel2["UNKNOWN_CHANNEL"] = "UNKNOWN_CHANNEL";
  Channel2["SMS"] = "SMS";
  Channel2["PHONE"] = "PHONE";
  return Channel2;
})(Channel || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createRecipient2(recipient) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    recipient
  });
  const reqOpts = createRecipient(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.recipient;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { recipient: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["recipient"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getRecipient2(recipientId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    recipientId
  });
  const reqOpts = getRecipient(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.recipient;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { recipientId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["recipientId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateRecipient2(_id, recipient) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    recipient: { ...recipient, id: _id }
  });
  const reqOpts = updateRecipient(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.recipient;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { recipient: "$[1]" },
        explicitPathsToArguments: { "recipient.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "recipient"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteRecipient2(recipientId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    recipientId
  });
  const reqOpts = deleteRecipient(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { recipientId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["recipientId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryRecipients2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryRecipients(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [])
      );
      return {
        items: transformedData?.recipients,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function bulkUpdateRecipientTags2(recipientIds, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    recipientIds,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateRecipientTags(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          recipientIds: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["recipientIds", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateRecipientTagsByFilter2(filter, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    filter,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateRecipientTagsByFilter(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          filter: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["filter", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Channel,
  SortOrder,
  WebhookIdentityType,
  bulkUpdateRecipientTags,
  bulkUpdateRecipientTagsByFilter,
  createRecipient,
  deleteRecipient,
  getRecipient,
  queryRecipients,
  updateRecipient
});
