// @wix/auto_sdk_restaurants_recipients/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  ChannelOriginal: () => Channel,
  SortOrderOriginal: () => SortOrder,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  bulkUpdateRecipientTags: () => bulkUpdateRecipientTags2,
  bulkUpdateRecipientTagsByFilter: () => bulkUpdateRecipientTagsByFilter2,
  createRecipient: () => createRecipient2,
  deleteRecipient: () => deleteRecipient2,
  getRecipient: () => getRecipient2,
  queryRecipients: () => queryRecipients2,
  updateRecipient: () => updateRecipient2
});
module.exports = __toCommonJS(meta_exports);

// src/restaurants-recipient-v1-recipient-recipients.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/rest-notification-recipients",
        destPath: ""
      }
    ],
    "dev._base_domain_": [
      {
        srcPath: "/_api/rest-notification-recipients",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/rest-notification-recipients/v1",
        destPath: "/v1"
      },
      {
        srcPath: "/restaurants/recipient/v1/recipients",
        destPath: "/v1/recipients"
      }
    ],
    _: [
      {
        srcPath: "/_api/rest-notification-recipients",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/rest-notification-recipients",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_recipients";
function createRecipient(payload) {
  function __createRecipient({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "recipient.createdDate" },
          { path: "recipient.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.recipient.v1.recipient",
      method: "POST",
      methodFqn: "wix.restaurants.recipient.v1.NotificationRecipientsService.CreateRecipient",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl({
        protoPath: "/v1/recipients",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "recipient.createdDate" },
            { path: "recipient.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createRecipient;
}
function getRecipient(payload) {
  function __getRecipient({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.recipient.v1.recipient",
      method: "GET",
      methodFqn: "wix.restaurants.recipient.v1.NotificationRecipientsService.GetRecipient",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl({
        protoPath: "/v1/recipients/{recipientId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "recipient.createdDate" },
            { path: "recipient.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getRecipient;
}
function updateRecipient(payload) {
  function __updateRecipient({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "recipient.createdDate" },
          { path: "recipient.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.recipient.v1.recipient",
      method: "PATCH",
      methodFqn: "wix.restaurants.recipient.v1.NotificationRecipientsService.UpdateRecipient",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl({
        protoPath: "/v1/recipients/{recipient.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "recipient.createdDate" },
            { path: "recipient.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateRecipient;
}
function deleteRecipient(payload) {
  function __deleteRecipient({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.recipient.v1.recipient",
      method: "DELETE",
      methodFqn: "wix.restaurants.recipient.v1.NotificationRecipientsService.DeleteRecipient",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl({
        protoPath: "/v1/recipients/{recipientId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteRecipient;
}
function queryRecipients(payload) {
  function __queryRecipients({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.recipient.v1.recipient",
      method: "GET",
      methodFqn: "wix.restaurants.recipient.v1.NotificationRecipientsService.QueryRecipients",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl({
        protoPath: "/v1/recipients/query",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload, true),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "recipients.createdDate" },
            { path: "recipients.updatedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl(
            { protoPath: "/v1/recipients/query", data: payload, host }
          ),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryRecipients;
}
function bulkUpdateRecipientTags(payload) {
  function __bulkUpdateRecipientTags({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.recipient.v1.recipient",
      method: "POST",
      methodFqn: "wix.restaurants.recipient.v1.NotificationRecipientsService.BulkUpdateRecipientTags",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl({
        protoPath: "/v1/bulk/recipients/update-tags",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateRecipientTags;
}
function bulkUpdateRecipientTagsByFilter(payload) {
  function __bulkUpdateRecipientTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.recipient.v1.recipient",
      method: "POST",
      methodFqn: "wix.restaurants.recipient.v1.NotificationRecipientsService.BulkUpdateRecipientTagsByFilter",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl({
        protoPath: "/v1/bulk/recipients/update-tags-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateRecipientTagsByFilter;
}

// src/restaurants-recipient-v1-recipient-recipients.types.ts
var Channel = /* @__PURE__ */ ((Channel2) => {
  Channel2["UNKNOWN_CHANNEL"] = "UNKNOWN_CHANNEL";
  Channel2["SMS"] = "SMS";
  Channel2["PHONE"] = "PHONE";
  return Channel2;
})(Channel || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/restaurants-recipient-v1-recipient-recipients.meta.ts
function createRecipient2() {
  const payload = {};
  const getRequestOptions = createRecipient(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/recipients",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getRecipient2() {
  const payload = { recipientId: ":recipientId" };
  const getRequestOptions = getRecipient(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/recipients/{recipientId}",
    pathParams: { recipientId: "recipientId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateRecipient2() {
  const payload = { recipient: { id: ":recipientId" } };
  const getRequestOptions = updateRecipient(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/recipients/{recipient.id}",
    pathParams: { recipientId: "recipientId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteRecipient2() {
  const payload = { recipientId: ":recipientId" };
  const getRequestOptions = deleteRecipient(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/recipients/{recipientId}",
    pathParams: { recipientId: "recipientId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryRecipients2() {
  const payload = {};
  const getRequestOptions = queryRecipients(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/recipients/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateRecipientTags2() {
  const payload = {};
  const getRequestOptions = bulkUpdateRecipientTags(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/recipients/update-tags",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateRecipientTagsByFilter2() {
  const payload = {};
  const getRequestOptions = bulkUpdateRecipientTagsByFilter(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/recipients/update-tags-by-filter",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ChannelOriginal,
  SortOrderOriginal,
  WebhookIdentityTypeOriginal,
  bulkUpdateRecipientTags,
  bulkUpdateRecipientTagsByFilter,
  createRecipient,
  deleteRecipient,
  getRecipient,
  queryRecipients,
  updateRecipient
});
