// pro-gallery/dist/cjs/components/helpers/cssLayoutsHelper.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createCssLayouts = void 0;
const pro_gallery_lib_1 = require("pro-gallery-lib");
const cssScrollHelper_js_1 = require("./cssScrollHelper.js");
// // const CDN_URL = 'https://static.wixstatic.com/media/';
// const desktopWidths = [480, 768, 1024, 1280, 1440, 1680, 1920, 2560];
// const mobileWidths = [320]; //, 375, 414, 480, 600, 768, 900]; (mobile is currently fixed to 320px)
const getImageStyle = (item) => ({
    top: item.offset.top,
    left: item.offset.left,
    width: item.width + item.infoWidth,
    height: item.height + item.infoHeight,
    innerHeight: item.height,
});
const createItemId = ({ galleryId, item }) => {
    return `#pro-gallery-${galleryId} #${cssScrollHelper_js_1.cssScrollHelper.getSellectorDomId(item)}`;
};
const createExactCssForItems = (id = '', galleryItems, options) => {
    const isRTL = options[pro_gallery_lib_1.optionsMap.behaviourParams.gallery.layoutDirection] ===
        pro_gallery_lib_1.GALLERY_CONSTS[pro_gallery_lib_1.optionsMap.behaviourParams.gallery.layoutDirection].RIGHT_TO_LEFT;
    let cssStr = '';
    galleryItems.forEach((item) => {
        const itemId = createItemId({ galleryId: id, item });
        const style = getImageStyle(item, options);
        const T = `top:${style.top}px;`;
        const L = isRTL ? `right:${style.left}px;left:auto;` : `left:${style.left}px;`;
        const W = `width:${style.width}px;`;
        const H = `height:${style.height}px;`;
        cssStr += `${itemId} {${T}${L}${W}${H}}`;
        // cssStr += `${id} .gallery-item-wrapper, ${id} .gallery-item-hover, ${id} .gallery-item {${Wvw}${Hvw}}`;
    });
    return cssStr;
};
const createCssLayouts = ({ galleryItems, layoutParams, id }) => {
    const exactCss = [];
    exactCss.push(createExactCssForItems(id, galleryItems, layoutParams.options));
    return exactCss;
};
exports.createCssLayouts = createCssLayouts;
