// pro-gallery/dist/cjs/components/helpers/mouseCursorPosition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.clickable = exports.clickableFactory = exports.ArrowFollower = exports.MouseCursor = exports.MouseFollowerProvider = exports.mouseFollower = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importStar(require("react"));
const emitter_js_1 = tslib_1.__importDefault(require("./emitter.js"));
const proxy_js_1 = require("./proxy.js");
const pro_gallery_lib_1 = require("pro-gallery-lib");
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
function mouseFollower(container) {
    const emitter = new emitter_js_1.default();
    let state = {
        mouseIn: false,
        position: [0, 0],
    };
    emitter.listen.mouseEnterState((mouseIn, ...position) => {
        state = {
            mouseIn,
            position,
        };
    });
    emitter.listen.mouseMove((x, y) => {
        state = Object.assign(Object.assign({}, state), { position: [x, y] });
    });
    const getMousePosition = (event) => {
        const bounding = container.getBoundingClientRect();
        const position = [event.clientX - bounding.left, event.clientY - bounding.top];
        return position;
    };
    function onMouseEnter(event) {
        const position = getMousePosition(event);
        emitter.call.mouseEnterState(!isHoveringClickableElement(container, event, ...position), ...position);
    }
    function onMouseMove(event) {
        const position = getMousePosition(event);
        emitter.call.mouseEnterState(!isHoveringClickableElement(container, event, ...position), ...position);
        emitter.call.mouseMove(...position);
    }
    function onMouseLeave() {
        emitter.call.mouseEnterState(false, 0, 0);
    }
    function onMouseClickInteractionEvent(event) {
        emitter.call.mouseClickInteractionEvent(event);
    }
    container.addEventListener('mouseenter', onMouseEnter);
    container.addEventListener('mousemove', onMouseMove);
    container.addEventListener('mouseleave', onMouseLeave);
    container.addEventListener('click', onMouseClickInteractionEvent);
    container.addEventListener('mousedown', onMouseClickInteractionEvent);
    container.addEventListener('mouseup', onMouseClickInteractionEvent);
    return {
        listen: emitter.listen,
        get state() {
            return state;
        },
        destroy: () => {
            container.removeEventListener('mouseenter', onMouseEnter);
            container.removeEventListener('mousemove', onMouseMove);
            container.removeEventListener('mouseleave', onMouseLeave);
            container.removeEventListener('click', onMouseClickInteractionEvent);
            container.removeEventListener('mousedown', onMouseClickInteractionEvent);
            container.removeEventListener('mouseup', onMouseClickInteractionEvent);
        },
    };
}
exports.mouseFollower = mouseFollower;
const getContainerById = (id) => document.getElementById(`pro-gallery-container-${id}`);
const MouseFollowerContext = react_1.default.createContext(undefined);
const MouseFollowerProvider = ({ children, id }) => {
    const [mouseFollowerValue, setMouseFollowerValue] = react_1.default.useState();
    (0, react_1.useEffect)(() => {
        const container = getContainerById(id);
        const mouseFollowerValue = mouseFollower(container);
        setMouseFollowerValue(mouseFollowerValue);
        return () => {
            mouseFollowerValue.destroy();
        };
    }, [id]);
    return (react_1.default.createElement(MouseFollowerContext.Provider, { value: mouseFollowerValue }, mouseFollowerValue && children));
};
exports.MouseFollowerProvider = MouseFollowerProvider;
class MouseCursor extends react_1.default.Component {
    constructor() {
        var _a, _b;
        super(...arguments);
        this.state = {
            position: ((_a = this.context) === null || _a === void 0 ? void 0 : _a.state.position) || [0, 0],
            mouseIn: ((_b = this.context) === null || _b === void 0 ? void 0 : _b.state.mouseIn) || false,
        };
    }
    componentDidMount() {
        if (!this.context) {
            throw new Error('MouseFollowerContext is undefined');
        }
        const removeMoveListener = this.context.listen.mouseMove(
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        pro_gallery_lib_1.utils.throttle(() => {
            this.setState({
                position: this.context.state.position,
            });
        }, this.props.throttle));
        const removeEnterListener = this.context.listen.mouseEnterState(() => {
            const { mouseIn, position } = this.context.state;
            if (this.state.mouseIn !== mouseIn) {
                this.setState({
                    mouseIn: mouseIn,
                    position,
                });
                if (this.props.onEnterState) {
                    this.props.onEnterState(mouseIn);
                }
            }
        });
        const removeClickListener = this.context.listen.mouseClickInteractionEvent((e) => {
            if (!this.state.mouseIn) {
                return;
            }
            const [x, y] = this.state.position;
            if (!this.props.shouldRenderAtPosition(x, y)) {
                return;
            }
            if (e.type === 'mousedown' || e.type === 'mouseup') {
                this.props.onMouseDownOrUp(e);
                return;
            }
            this.props.onClick(e);
        });
        this.cleanup = () => {
            removeMoveListener();
            removeEnterListener();
            removeClickListener();
        };
    }
    componentWillUnmount() {
        var _a;
        (_a = this.cleanup) === null || _a === void 0 ? void 0 : _a.call(this);
    }
    render() {
        if (!this.state.mouseIn) {
            return null;
        }
        const [x, y] = this.state.position;
        if (!this.props.shouldRenderAtPosition(x, y)) {
            return null;
        }
        return this.props.render(this.state.position[0], this.state.position[1]);
    }
}
exports.MouseCursor = MouseCursor;
MouseCursor.contextType = MouseFollowerContext;
const CLICKABLE_ATTR = 'pg-clickable';
const isHoveringClickableElement = (element, e, x, y) => {
    // cancel the click event
    for (const ele of e.composedPath()) {
        if (ele instanceof HTMLElement && ele.getAttribute(CLICKABLE_ATTR) === 'true') {
            return true;
        }
    }
    const bounding = element.getBoundingClientRect();
    const elementUnderMouse = document.elementFromPoint(x + bounding.left, y + bounding.top);
    if (!elementUnderMouse) {
        return false;
    }
    let parent = elementUnderMouse;
    while (parent) {
        if (parent.getAttribute(CLICKABLE_ATTR) === 'true') {
            return true;
        }
        parent = parent.parentElement;
    }
    return false;
};
class ArrowFollower extends react_1.default.Component {
    constructor() {
        super(...arguments);
        this.shouldRender = (x) => {
            const element = getContainerById(this.props.id);
            const containerWidth = element.offsetWidth;
            const realMaxWidth = Math.min(this.props.mouseCursorContainerMaxWidth, this.props.isTheOnlyArrow ? 100 : 50);
            const amountOfPixelsNeeded = containerWidth * (realMaxWidth / 100);
            const isLeft = amountOfPixelsNeeded >= x;
            const isRight = x > containerWidth - amountOfPixelsNeeded;
            return this.props.direction === 'left' ? isLeft : isRight;
        };
        this.onNavigation = (e) => {
            e.preventDefault();
            e.stopPropagation();
            this.props.onNavigate();
        };
    }
    render() {
        return (react_1.default.createElement(MouseCursor, { getElement: () => getContainerById(this.props.id), render: (x, y) => {
                var _a;
                return (react_1.default.createElement(react_1.default.Fragment, null,
                    this.props.children(x, y),
                    react_1.default.createElement("style", null, `
                  #${(_a = getContainerById(this.props.id)) === null || _a === void 0 ? void 0 : _a.id} * {
                    cursor: none !important;
                  }
              `)));
            }, shouldRenderAtPosition: this.shouldRender, onClick: this.onNavigation, onMouseDownOrUp: (e) => e.stopPropagation(), throttle: 50 }));
    }
}
exports.ArrowFollower = ArrowFollower;
function clickableFactory() {
    const instances = new Map();
    return (0, proxy_js_1.proxy)((name) => {
        if (!instances.get(name)) {
            instances.set(name, (props) => {
                return react_1.default.createElement(name, Object.assign(Object.assign({}, props), { [CLICKABLE_ATTR]: 'true' }));
            });
        }
        return instances.get(name);
    });
}
exports.clickableFactory = clickableFactory;
exports.clickable = clickableFactory();
