// pro-gallery/dist/cjs/components/helpers/navigationArrowUtils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getArrowBoxStyle = exports.shouldRenderNavArrows = exports.getArrowsRenderData = void 0;
const tslib_1 = require("tslib");
const react_1 = tslib_1.__importDefault(require("react"));
const pro_gallery_lib_1 = require("pro-gallery-lib");
const index_js_1 = require("../svgs/index.js");
const getArrowsSizeData = ({ customNavArrowsRenderer, arrowsSize, svgData, containerStyleType }) => {
    const isLandscape = svgData.width / svgData.height > 1;
    if (containerStyleType === pro_gallery_lib_1.GALLERY_CONSTS[pro_gallery_lib_1.optionsMap.layoutParams.navigationArrows.container.type].BOX) {
        const sizeData = {
            navArrowsContainerWidth: arrowsSize,
            navArrowsContainerHeight: arrowsSize,
            scalePercentage: arrowsSize / 2.4 / (isLandscape ? svgData.width : svgData.height),
        };
        return sizeData;
    }
    if (customNavArrowsRenderer) {
        return {
            navArrowsContainerWidth: arrowsSize,
            navArrowsContainerHeight: arrowsSize,
        };
    }
    const scalePercentage = arrowsSize / svgData.width;
    return {
        scalePercentage,
        navArrowsContainerWidth: arrowsSize,
        navArrowsContainerHeight: svgData.height * scalePercentage,
    };
};
const getArrowsRenderData = ({ customNavArrowsRenderer, arrowsColor, arrowsSize, arrowsType, containerStyleType, }) => {
    const arrowData = getArrowIconData(arrowsType);
    const { navArrowsContainerWidth, navArrowsContainerHeight, scalePercentage } = getArrowsSizeData({
        customNavArrowsRenderer,
        arrowsSize,
        svgData: arrowData,
        containerStyleType,
    });
    if (customNavArrowsRenderer) {
        const size = containerStyleType === pro_gallery_lib_1.GALLERY_CONSTS[pro_gallery_lib_1.optionsMap.layoutParams.navigationArrows.container.type].BOX
            ? arrowsSize / 2.4
            : arrowsSize;
        const customRenderer = (position) => (react_1.default.createElement("div", { style: {
                width: size,
                height: size,
            } }, customNavArrowsRenderer(position)));
        return {
            arrowRenderer: customRenderer,
            navArrowsContainerWidth,
            navArrowsContainerHeight,
        };
    }
    const arrowRenderer = (position) => {
        const scaleX = position === 'right' ? 1 : -1;
        const style = {
            transform: `scaleX(${scaleX}) scale(${scalePercentage})`,
            fill: pro_gallery_lib_1.utils.isMobile() && (arrowsColor === null || arrowsColor === void 0 ? void 0 : arrowsColor.value) ? arrowsColor.value : '',
        };
        return react_1.default.createElement(arrowData.SvgComp, { style: style });
    };
    return { arrowRenderer, navArrowsContainerWidth, navArrowsContainerHeight };
};
exports.getArrowsRenderData = getArrowsRenderData;
// Function that checks if the nav arrows parent-container is large enough for them
const arrowsWillFitPosition = ({ options, container, customNavArrowsRenderer }) => {
    const { height } = container;
    // Calc of Nav arrows container's height
    const arrowData = getArrowIconData(options[pro_gallery_lib_1.optionsMap.layoutParams.navigationArrows.type]);
    const { navArrowsContainerHeight } = getArrowsSizeData({
        customNavArrowsRenderer,
        arrowsSize: options[pro_gallery_lib_1.optionsMap.layoutParams.navigationArrows.size],
        svgData: arrowData,
        containerStyleType: options[pro_gallery_lib_1.optionsMap.layoutParams.navigationArrows.container.type],
    });
    const infoHeight = options[pro_gallery_lib_1.optionsMap.layoutParams.info.height];
    const parentHeightByVerticalPosition = {
        [pro_gallery_lib_1.GALLERY_CONSTS[pro_gallery_lib_1.optionsMap.layoutParams.navigationArrows.verticalAlignment].INFO_CENTER]: infoHeight,
        [pro_gallery_lib_1.GALLERY_CONSTS[pro_gallery_lib_1.optionsMap.layoutParams.navigationArrows.verticalAlignment].IMAGE_CENTER]: height - infoHeight,
        [pro_gallery_lib_1.GALLERY_CONSTS[pro_gallery_lib_1.optionsMap.layoutParams.navigationArrows.verticalAlignment].ITEM_CENTER]: height,
    };
    const parentHeight = parentHeightByVerticalPosition[options[pro_gallery_lib_1.optionsMap.layoutParams.navigationArrows.verticalAlignment]];
    return parentHeight >= navArrowsContainerHeight;
};
// function to Determine whether we should render the navigation arrows
const shouldRenderNavArrows = (props) => {
    const shouldRenderArrowsRelevantArgs = getShouldRenderArrowsArgs(props);
    const navigationArrowsEnabled = shouldRenderArrowsRelevantArgs.options[pro_gallery_lib_1.optionsMap.layoutParams.navigationArrows.enable];
    const { galleryWidth } = shouldRenderArrowsRelevantArgs.container;
    const { isPrerenderMode, galleryStructure, customNavArrowsRenderer } = shouldRenderArrowsRelevantArgs;
    const arrowsWillFitPositionRelevantArgs = {
        options: shouldRenderArrowsRelevantArgs.options,
        container: shouldRenderArrowsRelevantArgs.container,
        customNavArrowsRenderer,
    };
    const isGalleryWiderThanRenderedItems = galleryStructure.width <= galleryWidth;
    return (!!navigationArrowsEnabled &&
        !isPrerenderMode &&
        arrowsWillFitPosition(arrowsWillFitPositionRelevantArgs) &&
        !isGalleryWiderThanRenderedItems);
};
exports.shouldRenderNavArrows = shouldRenderNavArrows;
const getShouldRenderArrowsArgs = (props) => {
    const { isPrerenderMode, galleryStructure, customNavArrowsRenderer } = props;
    return {
        options: props.options,
        container: props.container,
        isPrerenderMode,
        galleryStructure,
        customNavArrowsRenderer,
    };
};
const getArrowIconData = (arrowType = pro_gallery_lib_1.GALLERY_CONSTS[pro_gallery_lib_1.optionsMap.layoutParams.navigationArrows.type].DEFAULT_ARROW) => {
    const { DEFAULT_ARROW, ARROW_2, ARROW_3, ARROW_4, ARROW_5, ARROW_6, ARROW_7 } = pro_gallery_lib_1.GALLERY_CONSTS[pro_gallery_lib_1.optionsMap.layoutParams.navigationArrows.type];
    let arrowData;
    switch (arrowType) {
        case ARROW_2:
            arrowData = index_js_1.ARROWS_DATA.ARROW_2;
            break;
        case ARROW_3:
            arrowData = index_js_1.ARROWS_DATA.ARROW_3;
            break;
        case ARROW_4:
            arrowData = index_js_1.ARROWS_DATA.ARROW_4;
            break;
        case ARROW_5:
            arrowData = index_js_1.ARROWS_DATA.ARROW_5;
            break;
        case ARROW_6:
            arrowData = index_js_1.ARROWS_DATA.ARROW_6;
            break;
        case ARROW_7:
            arrowData = index_js_1.ARROWS_DATA.ARROW_7;
            break;
        case DEFAULT_ARROW:
        default:
            arrowData = index_js_1.ARROWS_DATA.DEFAULT_ARROW;
            break;
    }
    return arrowData;
};
const getArrowBoxStyle = ({ type, backgroundColor, borderRadius }) => {
    return type === pro_gallery_lib_1.GALLERY_CONSTS[pro_gallery_lib_1.optionsMap.layoutParams.navigationArrows.container.type].BOX
        ? {
            backgroundColor: backgroundColor,
            borderRadius: `${borderRadius}%`,
        }
        : {};
};
exports.getArrowBoxStyle = getArrowBoxStyle;
