// @wix/bex-core/dist/cjs/state/CustomFilterState/CustomFilterState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.customFilter = exports.CustomFilterState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _events = require("events");
var _mobx = require("mobx");
var _refreshFilter = require("../Filter/refreshFilter");
class CustomFilterState {
  constructor(params) {
    (0, _defineProperty2.default)(this, "name", void 0);
    (0, _defineProperty2.default)(this, "persistent", void 0);
    (0, _defineProperty2.default)(this, "defaultValue", void 0);
    (0, _defineProperty2.default)(this, "_value", void 0);
    (0, _defineProperty2.default)(this, "_isEmpty", void 0);
    (0, _defineProperty2.default)(this, "_size", void 0);
    (0, _defineProperty2.default)(this, "_toQueryString", void 0);
    (0, _defineProperty2.default)(this, "_fromQueryString", void 0);
    (0, _defineProperty2.default)(this, "_toArray", void 0);
    (0, _defineProperty2.default)(this, "_remove", void 0);
    (0, _defineProperty2.default)(this, "_hasDiff", void 0);
    (0, _defineProperty2.default)(this, "decode", void 0);
    (0, _defineProperty2.default)(this, "encode", void 0);
    (0, _defineProperty2.default)(this, "itemKey", void 0);
    (0, _defineProperty2.default)(this, "itemName", void 0);
    (0, _defineProperty2.default)(this, "matches", void 0);
    (0, _defineProperty2.default)(this, "equals", void 0);
    (0, _defineProperty2.default)(this, "events", new _events.EventEmitter());
    this.name = params.name ?? '';
    this.persistent = params.persistent;
    this.defaultValue = params.defaultValue;
    this._value = params.initialValue ?? params.defaultValue;
    this._isEmpty = params.isEmpty;
    this._size = params.size ?? (value => this._isEmpty(value) ? 0 : 1);
    this._toQueryString = params.toQueryString;
    this._fromQueryString = params.fromQueryString;
    this._toArray = params.toArray;
    this._remove = params.remove ?? (() => this.defaultValue);
    this._hasDiff = params.hasDiff ?? (data => data !== this.value);
    this.itemKey = params.itemKey;
    this.itemName = params.itemName;
    this.decode = params.decode;
    this.encode = params.encode;
    this.matches = params.matches ?? (() => false);
    this.equals = params.equals;
    (0, _mobx.makeObservable)(this, {
      _value: _mobx.observable.ref,
      persistent: _mobx.observable.ref,
      isEmpty: _mobx.computed,
      size: _mobx.computed,
      value: _mobx.computed,
      reset: _mobx.action,
      setValue: _mobx.action,
      changeValue: _mobx.action,
      scheduleRefresh: _mobx.action,
      clone: _mobx.action
    });
  }
  get value() {
    return this._value;
  }
  get toArray() {
    return this._toArray(this._value);
  }
  get isEmpty() {
    return this._isEmpty(this.value);
  }
  get size() {
    return this._size(this.value);
  }
  setValue(value, {
    emitEvents,
    ...options
  } = {}) {
    this._value = value;
    if (emitEvents) {
      for (const event of emitEvents) {
        this.events.emit(event, options);
      }
    }
  }
  changeValue(value, {
    emitEvents = [],
    ...options
  } = {}) {
    this.setValue(value, {
      ...options,
      emitEvents: ['change', ...emitEvents]
    });
  }
  refresh(value, options) {
    (0, _refreshFilter.refreshFilter)(this, value, options);
  }
  scheduleRefresh(value) {
    this.changeValue(value, {
      emitEvents: ['scheduleRefresh']
    });
  }
  get toQueryString() {
    const {
      _toQueryString,
      encode,
      value
    } = this;
    if (_toQueryString) {
      return _toQueryString(value);
    }
    return JSON.stringify(encode(value));
  }
  applyFromQueryString(str) {
    const {
      _fromQueryString,
      decode
    } = this;
    if (_fromQueryString) {
      this.setValue(_fromQueryString(str));
    } else {
      try {
        const raw = JSON.parse(str);
        this.setValue(decode(raw));
      } catch (e) {}
    }
  }
  reset(options = {}) {
    this.setValue(this.defaultValue, options);
  }
  remove(items, options = {}) {
    this.setValue(this._remove(items, this.value), options);
  }
  hasDiff(data) {
    return this._hasDiff(data);
  }
  clone(params) {
    const {
      itemName,
      itemKey,
      encode,
      decode,
      _isEmpty: isEmpty,
      _remove: remove,
      _size: size,
      _fromQueryString: fromQueryString,
      _toArray: toArray,
      _toQueryString: toQueryString,
      defaultValue,
      value: initialValue,
      matches,
      persistent,
      name
    } = this;
    return new CustomFilterState({
      itemName,
      itemKey,
      encode,
      decode,
      isEmpty,
      remove,
      size,
      fromQueryString,
      toArray,
      toQueryString,
      defaultValue,
      initialValue,
      matches,
      persistent,
      name,
      ...params
    });
  }
}
exports.CustomFilterState = CustomFilterState;
const customFilter = params => new CustomFilterState(params);
exports.customFilter = customFilter;
