// @wix/bex-core/dist/cjs/services/AccountPreferencesMergeStrategy.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.AccountPreferencesMergeStrategy = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _merge = _interopRequireDefault(require("lodash/merge"));
class AccountPreferencesMergeStrategy {
  constructor(params) {
    (0, _defineProperty2.default)(this, "storage", void 0);
    this.storage = params.storage;
  }
  extendScope(scope) {
    return scope;
  }
  async setItem(key, value, options) {
    var _data$views, _partialData$views;
    const {
      storage
    } = this;
    const data = await this.getItem(key, options).catch(() => undefined);
    const partialData = value;
    if (data != null && (_data$views = data.views) != null && _data$views.list && partialData != null && (_partialData$views = partialData.views) != null && _partialData$views.list) {
      for (const [key, value] of Object.entries(partialData.views.list)) {
        if (value === null) {
          // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
          delete data.views.list[key];
          // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
          delete partialData.views.list[key];
        }
      }
    }
    return storage.setItem(key, (0, _merge.default)(data, value), options);
  }
  async removeItem(key, options) {
    const {
      storage
    } = this;
    const data = await this.getItem(key, options).catch(() => undefined);
    return storage.setItem(key, (0, _merge.default)(data, {
      [key]: null
    }), options);
  }
  async getItem(key, options) {
    return this.storage.getItem(key, options);
  }
  async getAllItems(options) {
    return this.storage.getAllItems(options);
  }
}
exports.AccountPreferencesMergeStrategy = AccountPreferencesMergeStrategy;
