// @wix/bex-core/dist/cjs/services/AccountPreferencesStrategy.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.AccountPreferencesStrategy = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _isHttpError = require("../util/isHttpError");
class AccountPreferencesStrategy {
  constructor(params) {
    (0, _defineProperty2.default)(this, "httpClient", void 0);
    (0, _defineProperty2.default)(this, "accountPrefHttpPath", '/_api/app-settings-service/v1/accountPreferences');
    this.httpClient = params.httpClient;
  }
  extendScope(scope) {
    return scope;
  }
  async getAllItems(options) {
    const url = [this.accountPrefHttpPath, 'getVolatilePrefs', options.namespace].filter(x => x).join('/');
    const {
      data
    } = await this.httpClient.request({
      url,
      method: 'GET'
    });
    return data;
  }
  async getItem(key, options) {
    const url = [this.accountPrefHttpPath, 'getVolatilePrefForKey', options.namespace, key].filter(x => x).join('/');
    try {
      const {
        data
      } = await this.httpClient.request({
        url,
        method: 'GET'
      });
      return data[key];
    } catch (e) {
      var _e$response;
      if ((0, _isHttpError.isHttpError)(e) && ((_e$response = e.response) == null ? void 0 : _e$response.status) === 404) {
        throw new Error('Key was not found in capsule');
      }
      throw e;
    }
  }
  async removeItem(key, options) {
    const payload = {
      namespace: options.namespace,
      key
    };
    const url = [this.accountPrefHttpPath, 'delete'].filter(x => x).join('/');
    await this.httpClient.request({
      url,
      data: payload,
      method: 'POST'
    });
  }
  async setItem(key, value, options) {
    const payload = {
      namespace: options.namespace,
      key,
      blob: value
    };
    if (options.expiration) {
      payload.ttlInDays = Math.ceil(options.expiration / (60 * 60 * 24));
    }
    const url = [this.accountPrefHttpPath, 'set'].filter(x => x).join('/');
    await this.httpClient.request({
      url,
      data: payload,
      method: 'POST'
    });
  }
}
exports.AccountPreferencesStrategy = AccountPreferencesStrategy;
