// @wix/bex-core/dist/cjs/services/AppSettingsCapsule.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.AppSettingsCapsule = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _http = require("@wix/bex-utils/@wix/ambassador-app-settings-v1-settings/http");
var _core = require("react-query/core");
const queryName = 'app-settings-capsule';
class AppSettingsCapsule {
  constructor(params) {
    (0, _defineProperty2.default)(this, "httpClient", void 0);
    (0, _defineProperty2.default)(this, "context", void 0);
    (0, _defineProperty2.default)(this, "queryClient", void 0);
    (0, _defineProperty2.default)(this, "queryCache", void 0);
    this.httpClient = params.httpClient;
    this.context = params.context;
    this.queryCache = params.queryCache;
    this.queryClient = new _core.QueryClient({
      queryCache: this.queryCache
    });
  }
  _buildAppSettingsPayload(payload) {
    const {
      context: {
        componentId
      }
    } = this;
    return {
      ...payload,
      ...(componentId && {
        componentId
      })
    };
  }
  _getQueryKey() {
    return [queryName, this.context];
  }
  async getItem(key) {
    var _data$settings;
    const {
      context,
      httpClient,
      queryClient
    } = this;
    const {
      data
    } = await queryClient.fetchQuery({
      queryFn: () => httpClient.request((0, _http.getGlobalSettings)({
        host: context.host
      })),
      queryKey: this._getQueryKey(),
      staleTime: 120000
    });
    return (_data$settings = data.settings) == null || (_data$settings = _data$settings[context.componentId]) == null || (_data$settings = _data$settings[context.namespace]) == null ? void 0 : _data$settings[key];
  }
  async setItem(key, value) {
    const {
      httpClient,
      queryCache,
      context
    } = this;
    queryCache.findAll({
      queryKey: this._getQueryKey()
    }).forEach(q => q.invalidate());
    await httpClient.request((0, _http.setGlobalSettings)({
      host: context.host,
      settings: {
        [context.componentId]: {
          [context.namespace]: {
            [key]: value
          }
        }
      }
    }));
  }
  async removeItem(key) {
    await this.setItem(key, null);
  }
}
exports.AppSettingsCapsule = AppSettingsCapsule;
