// @wix/bex-core/dist/cjs/services/cairoMonitor.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.WixPatternsMonitor = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _WixPatternsError = require("./WixPatternsError");
class WixPatternsMonitor {
  constructor({
    internalErrorMonitor,
    errorMonitor,
    internalFedops,
    environment,
    csid
  }) {
    (0, _defineProperty2.default)(this, "_internalErrorMonitor", void 0);
    (0, _defineProperty2.default)(this, "_errorMonitor", void 0);
    (0, _defineProperty2.default)(this, "_internalFedops", void 0);
    (0, _defineProperty2.default)(this, "_reports", {});
    (0, _defineProperty2.default)(this, "_environment", void 0);
    (0, _defineProperty2.default)(this, "_csid", void 0);
    this._internalErrorMonitor = internalErrorMonitor;
    this._errorMonitor = errorMonitor;
    this._internalFedops = internalFedops;
    this._environment = environment;
    this._csid = csid;
  }

  /**
   * This method is used to report errors that occur within the application. It takes in an error object with the following properties:
   * @param errorCode A WixPatternsErrorCode enum value representing the error that occurred.
   * @param exception The exception that was thrown.
   * @param extraData (optional) Additional data related to the error.
   * @param propagate (optional) Propagates report to vertical error monitor.
   * @default propagate true
   */
  reportError({
    error,
    propagate = true,
    category
  }) {
    const options = {
      tags: {
        ...this.extraInteractionParams
      }
    };
    if (category) {
      options.tags.category = category;
    }
    this._internalErrorMonitor.captureException(error, options);
    if (this._errorMonitor && propagate) {
      const originalException = (0, _WixPatternsError.isWixPatternsError)(error) ? error.originalException : error;
      this._errorMonitor.captureException(originalException);
    }
  }
  get extraInteractionParams() {
    return {
      cairoVersion: this._environment.uiLibVersion,
      componentId: this._environment.componentId,
      csid: this._csid
    };
  }
  interactionStarted(interactionName) {
    var _this$_internalFedops;
    (_this$_internalFedops = this._internalFedops) == null || _this$_internalFedops.interactionStarted(interactionName, {
      customParams: this.extraInteractionParams
    });
  }
  interactionEnded(interactionName) {
    var _this$_internalFedops2;
    (_this$_internalFedops2 = this._internalFedops) == null || _this$_internalFedops2.interactionEnded(interactionName, {
      customParams: this.extraInteractionParams
    });
  }
  reportInteractionStart(componentName, event, customParams) {
    const interaction = `${componentName.toLowerCase()}-${event}`;
    const reportKey = `${interaction}-start`;
    if (!this._reports[reportKey]) {
      var _this$_internalFedops3;
      this._reports[reportKey] = true;
      (_this$_internalFedops3 = this._internalFedops) == null || _this$_internalFedops3.interactionStarted(interaction, {
        customParams: {
          ...this.extraInteractionParams,
          ...customParams
        }
      });
    }
  }
  reportInteractionEnd(componentName, event, customParams) {
    const interaction = `${componentName.toLowerCase()}-${event}`;
    const reportKey = `${interaction}-end`;
    if (!this._reports[reportKey]) {
      var _this$_internalFedops4;
      this._reports[reportKey] = true;
      (_this$_internalFedops4 = this._internalFedops) == null || _this$_internalFedops4.interactionEnded(interaction, {
        customParams: {
          ...this.extraInteractionParams,
          ...customParams
        }
      });
    }
  }
  reportErrorThrownOncePerFlow() {
    var _this$_internalFedops5;
    (_this$_internalFedops5 = this._internalFedops) == null || _this$_internalFedops5.reportErrorThrownOncePerFlow();
  }
}
exports.WixPatternsMonitor = WixPatternsMonitor;
