// @wix/bex-core/dist/cjs/services/createAccountComponentSettings.js
"use strict";

exports.__esModule = true;
exports.createAccountComponentSettings = void 0;
var _AccountPreferencesStrategy = require("./AccountPreferencesStrategy");
var _AccountPreferencesMergeStrategy = require("./AccountPreferencesMergeStrategy");
function isDataCapsuleStorage(storageStrategy) {
  return 'storageStrategy' in storageStrategy;
}
const createAccountComponentSettings = ({
  httpClient,
  environment,
  createEssentials
}) => {
  return options => {
    const scope = environment.componentId || environment.artifactId;
    const {
      dataCapsule: _dataCapsule
    } = createEssentials({
      dataCapsule: {
        storageStrategy: 'userPreferences',
        namespace: [scope, options.namespace].join('.')
      }
    });
    const {
      dataCapsule: oldLocalStorageComponentSettingsDataCapsule,
      fedopsLogger
    } = createEssentials({
      dataCapsule: {
        namespace: options.namespace,
        scope
      },
      fedops: {
        appName: 'cairo'
      }
    });
    const dataCapsule = _dataCapsule;
    if (isDataCapsuleStorage(dataCapsule.storageStrategy)) {
      dataCapsule.storageStrategy = new _AccountPreferencesMergeStrategy.AccountPreferencesMergeStrategy({
        storage: new _AccountPreferencesStrategy.AccountPreferencesStrategy({
          httpClient
        })
      });
    }
    _dataCapsule.migrateItem = async key => {
      var _localData$views, _data$views;
      const localData = await oldLocalStorageComponentSettingsDataCapsule.getItem(key).catch(() => null);
      if (!localData) {
        return;
      }
      fedopsLogger == null || fedopsLogger.interactionStarted('account-component-settings-local-storage-migration');
      const data = await dataCapsule.getItem(key).catch(() => null);
      await dataCapsule.setItem(key, {
        views: {
          list: {
            ...(localData == null || (_localData$views = localData.views) == null ? void 0 : _localData$views.list),
            ...(data == null || (_data$views = data.views) == null ? void 0 : _data$views.list)
          }
        }
      });
      await oldLocalStorageComponentSettingsDataCapsule.removeItem(key);
      fedopsLogger == null || fedopsLogger.interactionEnded('account-component-settings-local-storage-migration');
    };
    return _dataCapsule;
  };
};
exports.createAccountComponentSettings = createAccountComponentSettings;
