// @wix/bex-core/dist/cjs/services/DataCapsuleMergeStrategy.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.DataCapsuleMergeStrategy = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _merge = _interopRequireDefault(require("lodash/merge"));
// This "merge" layer allows to mimic the automatic "merge" behavior of app-settings also in user-preferences storage strategy.
// https://github.com/wix-private/app-settings/blob/64eef870a0958a117a8325511f57df9d73857359/app-settings-service/src/services/mysql-storage.ts#L273
class DataCapsuleMergeStrategy {
  constructor(params) {
    (0, _defineProperty2.default)(this, "storage", void 0);
    this.storage = params.storage;
  }
  extendScope(scope) {
    return scope;
  }
  async setItem(key, value, options) {
    const {
      storage
    } = this;
    const data = await this.getItem(key, options).catch(() => undefined);
    return storage.setItem(key, (0, _merge.default)(data, value), options);
  }
  async removeItem(key, options) {
    const {
      storage
    } = this;
    const data = await this.getItem(key, options).catch(() => undefined);
    return storage.setItem(key, (0, _merge.default)(data, {
      [key]: null
    }), options);
  }
  async getItem(key, options) {
    return this.storage.getItem(key, options);
  }
  async getAllItems(options) {
    return this.storage.getAllItems(options);
  }
}
exports.DataCapsuleMergeStrategy = DataCapsuleMergeStrategy;
