// @wix/bex-core/dist/cjs/services/navigationManager.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.NavigationManager = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _events = _interopRequireDefault(require("events"));
class NavigationManager {
  constructor(window) {
    (0, _defineProperty2.default)(this, "window", void 0);
    (0, _defineProperty2.default)(this, "events", new _events.default());
    this.window = window;
  }
  navigate(url) {
    const state = {
      prevented: false,
      pause: () => {
        return {
          resume: () => {
            this.window.location.href = url;
          }
        };
      }
    };
    for (const navigation of this.events.listeners('navigation')) {
      navigation(state);
    }
    if (!state.prevented) {
      this.window.location.href = url;
    }
  }
  navigateToPage({
    pageId
  }) {
    const url = new URL(this.window.location.href);
    url.pathname = `/page/${pageId}`;
    this.navigate(url.href);
  }
  confirmLeavePage() {
    this.events.emit('navigationPromptResult', true);
  }
  subscribe(onNavigation) {
    this.events.on('navigation', onNavigation);
    return () => {
      this.events.off('navigation', onNavigation);
    };
  }
}
exports.NavigationManager = NavigationManager;
