// @wix/bex-core/dist/cjs/services/WixPatternsError.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.WixPatternsError = void 0;
exports.isError = isError;
exports.isWixPatternsError = isWixPatternsError;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
function isWixPatternsError(err) {
  return err instanceof WixPatternsError;
}
function isError(err) {
  return err instanceof Error;
}
class WixPatternsError extends Error {
  /**
   * This class is used to create a cairo error that occur within the application. It takes in an object with the following properties:
   * @param errorCode A WixPatternsErrorCode enum value representing the error that occurred.
   * @param originalException (optional) The exception that was thrown.
   * @param extraData (optional) Additional data related to the error.
   */
  constructor({
    errorCode,
    originalException,
    extraData
  }) {
    super(isError(originalException) ? (originalException == null ? void 0 : originalException.message) || (originalException == null ? void 0 : originalException.name) : '');
    (0, _defineProperty2.default)(this, "originalException", void 0);
    (0, _defineProperty2.default)(this, "extraData", void 0);
    this.name = errorCode;
    this.originalException = originalException;
    this.extraData = extraData;
  }
}
exports.WixPatternsError = WixPatternsError;
