// @wix/patterns/dist/cjs/essentials/buildWixPatternsEssentials.js
"use strict";

exports.__esModule = true;
exports.buildWixPatternsEssentials = buildWixPatternsEssentials;
var _react = require("@wix/fe-essentials/react");
var _bexCore = require("@wix/bex-core");
function buildWixPatternsEssentials(essentials, {
  overrides
} = {}) {
  const {
    errorMonitor,
    httpClient,
    environment
  } = essentials;
  return {
    createEssentials: options => essentials.createChildEssentials({
      ...options,
      // currently avoid client-side conduction for cairo experiments
      experiments: {
        bag: {}
      }
    }),
    errorMonitor: errorMonitor ?? {
      captureException: () => {}
    },
    useTranslation: _react.useTranslation,
    Trans: _react.Trans,
    useExperiments: _react.useExperiments,
    showToast: () => ({
      remove: () => {}
    }),
    httpClient,
    experiments: (0, _bexCore.getExperimentsCompat)(essentials),
    ...overrides,
    user: {
      permissions: [],
      ...(overrides == null ? void 0 : overrides.user)
    },
    environment: {
      ...environment,
      ...(overrides == null ? void 0 : overrides.environment),
      providerType: 'essentials'
    },
    environmentBI: {
      artifactId: environment.artifactId
    }
  };
}
