// @wix/patterns/dist/cjs/components/EmptyState/CollectionPremiumEmptyState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.CollectionPremiumEmptyState = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireDefault(require("react"));
var _mobxReactLite = require("mobx-react-lite");
var _EmptyStateBase = require("./EmptyStateBase");
var _CollectionPremiumEmptyStateIllustration = require("../Illustration/CollectionPremiumEmptyStateIllustration");
var _react2 = require("@wix/bex-core/react");
var _PremiumButton = require("./PremiumButton");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/EmptyState/CollectionPremiumEmptyState.tsx";
function _CollectionPremiumEmptyState(props) {
  const {
    translate: t
  } = (0, _react2.useWixPatternsContainer)();
  const {
    upgradeCta,
    ...rest
  } = props;
  return /*#__PURE__*/_react.default.createElement(_EmptyStateBase.EmptyStateBase, (0, _extends2.default)({
    dataHook: "premium-empty-state",
    title: t('cairo.emptyState.premium.title'),
    subtitle: t('cairo.emptyState.premium.text'),
    image: /*#__PURE__*/_react.default.createElement(_CollectionPremiumEmptyStateIllustration.CollectionPremiumEmptyStateIllustration, {
      "data-hook": "empty-state-premium-illustration",
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 30,
        columnNumber: 9
      }
    })
  }, rest, {
    cta: /*#__PURE__*/_react.default.createElement(_PremiumButton.PremiumButton, (0, _extends2.default)({}, upgradeCta, {
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 33,
        columnNumber: 12
      }
    })),
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 25,
      columnNumber: 5
    }
  }));
}
const CollectionPremiumEmptyState = exports.CollectionPremiumEmptyState = (0, _mobxReactLite.observer)(_CollectionPremiumEmptyState);
