// @wix/patterns/dist/cjs/components/EmptyState/CollectionEmptyState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.CollectionEmptyState = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireDefault(require("react"));
var _mobxReactLite = require("mobx-react-lite");
var _AddItemEmptyState = require("../Illustration/AddItemEmptyState");
var _react2 = require("@wix/bex-core/react");
var _EmptyStateBase = require("./EmptyStateBase");
var _NewItemButton = require("./NewItemButton");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/EmptyState/CollectionEmptyState.tsx";
function _CollectionEmptyState(props) {
  const {
    translate: t
  } = (0, _react2.useWixPatternsContainer)();
  const {
    addNewCta
  } = props;
  return /*#__PURE__*/_react.default.createElement(_EmptyStateBase.EmptyStateBase, (0, _extends2.default)({
    dataHook: "collection-empty-state",
    title: t('cairo.emptyState.additem.title'),
    subtitle: t('cairo.emptyState.additem.text'),
    image: /*#__PURE__*/_react.default.createElement(_AddItemEmptyState.AddItemEmptyState, {
      "data-hook": "empty-state-new-item-illustration",
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 30,
        columnNumber: 9
      }
    })
  }, props, {
    cta: addNewCta && /*#__PURE__*/_react.default.createElement(_NewItemButton.NewItemButton, (0, _extends2.default)({}, addNewCta, {
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 33,
        columnNumber: 25
      }
    })),
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 25,
      columnNumber: 5
    }
  }));
}
const CollectionEmptyState = exports.CollectionEmptyState = (0, _mobxReactLite.observer)(_CollectionEmptyState);
