// @wix/patterns/dist/cjs/components/EmptyState/EmptyStateBase.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.EmptyStateBase = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireDefault(require("react"));
var _designSystem = require("@wix/design-system");
var _mobxReactLite = require("mobx-react-lite");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/EmptyState/EmptyStateBase.tsx";
function _EmptyStateBase(props) {
  const {
    children,
    customCta,
    title,
    subtitle,
    cta,
    image,
    ...rest
  } = props;
  let child;
  if (cta || customCta) {
    child = /*#__PURE__*/_react.default.createElement(_designSystem.Box, {
      gap: "SP5",
      verticalAlign: "middle",
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 49,
        columnNumber: 7
      }
    }, customCta, cta);
  }
  if (children) {
    child = /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, child, children);
  }
  return /*#__PURE__*/_react.default.createElement(_designSystem.EmptyState, (0, _extends2.default)({
    title: title,
    subtitle: subtitle,
    theme: "page-no-border",
    image: image ? /*#__PURE__*/_react.default.createElement("div", {
      style: {
        display: 'flex'
      },
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 70,
        columnNumber: 22
      }
    }, image) : undefined
  }, rest, {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 66,
      columnNumber: 5
    }
  }), child);
}
const EmptyStateBase = exports.EmptyStateBase = (0, _mobxReactLite.observer)(_EmptyStateBase);
