// @wix/bex-utils/dist/esm/PromiseState.js
import { reactive } from './reactive';
export class PromiseState {
    constructor(params) {
        this.status = reactive({ type: 'idle' });
        this.fn = params.fn;
        this.onError = params.onError;
    }
    async once() {
        const { status: { value: { type, promise }, }, } = this;
        if (type === 'loading') {
            await promise;
            return;
        }
        if (type === 'success') {
            return;
        }
        return this.run();
    }
    async run() {
        const { fn, status, onError } = this;
        const promise = fn();
        status.value = {
            type: 'loading',
            promise,
        };
        try {
            const data = await promise;
            if (promise === status.value.promise) {
                status.value = {
                    type: 'success',
                    data,
                    promise,
                };
            }
        }
        catch (error) {
            onError?.(error);
            if (promise === status.value.promise) {
                status.value = {
                    type: 'error',
                    error,
                };
            }
        }
    }
}
