// @wix/headless-bookings/dist/services/service-service.js
import { defineService, implementService } from '@wix/services-definitions';
import { SignalsServiceDefinition, } from '@wix/services-definitions/core-services/signals';
import { services } from '@wix/bookings';
/**
 * Service definition for the Service service.
 * This defines the contract that the ServiceService must implement.
 *
 * @constant
 */
export const ServiceServiceDefinition = defineService('service');
/**
 * Implementation of the Service service that manages reactive service data.
 * This service provides signals for service data, loading state, and error handling,
 * along with methods to dynamically load services.
 *
 * @example
 * ```tsx
 * import { ServiceService, ServiceServiceDefinition } from '@wix/services/services';
 * import { useService } from '@wix/services-manager-react';
 *
 * function ServiceComponent({ serviceConfig }) {
 *   return (
 *     <ServiceProvider services={createServicesMap([
 *       [ServiceServiceDefinition, ServiceService.withConfig(serviceConfig)]
 *     ])}>
 *       <ServiceDisplay />
 *     </ServiceProvider>
 *   );
 * }
 *
 * function ServiceDisplay() {
 *   const serviceService = useService(ServiceServiceDefinition);
 *   const service = serviceService.service.get();
 *   const isLoading = serviceService.isLoading.get();
 *   const error = serviceService.error.get();
 *
 *   if (isLoading) return <div>Loading...</div>;
 *   if (error) return <div>Error: {error}</div>;
 *
 *   return <h1>{service.name}</h1>;
 * }
 * ```
 */
export const ServiceService = implementService.withConfig()(ServiceServiceDefinition, ({ getService, config }) => {
    const signalsService = getService(SignalsServiceDefinition);
    const service = signalsService.signal(config.service);
    const isLoading = signalsService.signal(!!config.serviceId);
    const error = signalsService.signal(null);
    const loadService = async (id) => {
        isLoading.set(true);
        const serviceResponse = await loadServiceById(id);
        if (!serviceResponse.service) {
            error.set('Service not found');
        }
        else {
            service.set(serviceResponse.service);
            error.set(null);
        }
        isLoading.set(false);
    };
    if (config.serviceId) {
        loadService(config.serviceId);
    }
    return {
        service,
        isLoading,
        error,
        loadService,
    };
});
/**
 * Internal helper function to load a service by its ID from the Wix Services API.
 * Fetches comprehensive service data including description, categories, media, etc.
 *
 * @private
 * @param {string} id - The service ID to load
 * @returns {Promise} Service response from the API
 */
const loadServiceById = async (id) => {
    try {
        const service = await services.getService(id);
        return { service };
    }
    catch (error) {
        console.error(`Failed to load service for id "${id}":`, error);
        return { service: null };
    }
};
/**
 * Loads service service configuration from the Wix Services API for SSR initialization.
 * This function is designed to be used during Server-Side Rendering (SSR) to preload
 * a specific service by id that will be used to configure the ServiceService.
 *
 * @param serviceId The service id to load
 * @returns Promise that resolves to ServiceServiceConfigResult (success with config or notFound)
 *
 * @example
 * ```astro
 * ---
 * // Astro page example - pages/service/[id].astro
 * import { loadServiceServiceConfig } from '@wix/services/services';
 * import { Service } from '@wix/services/components';
 *
 * // Get service id from URL params
 * const { id } = Astro.params;
 *
 * // Load service data during SSR
 * const serviceResult = await loadServiceServiceConfig(id);
 *
 * // Handle not found case
 * if (serviceResult.type === 'notFound') {
 *   return Astro.redirect('/404');
 * }
 * ---
 *
 * <Service.Root serviceConfig={serviceResult.config}>
 *   <Service.Name>
 *     {({ name }) => <h1>{name}</h1>}
 *   </Service.Name>
 * </Service.Root>
 * ```
 */
export async function loadServiceServiceConfig(serviceId) {
    try {
        const serviceResponse = await loadServiceById(serviceId);
        if (!serviceResponse.service) {
            return { type: 'notFound' };
        }
        return {
            type: 'success',
            config: {
                service: serviceResponse.service,
            },
        };
    }
    catch (error) {
        console.error(`Failed to load service for id "${serviceId}":`, error);
        return { type: 'notFound' };
    }
}
