// @wix/design-system/dist/cjs/Box/Box.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _react = _interopRequireWildcard(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _BoxSt = require("./Box.st.css.js");
var _colorsSt = require("../Foundation/stylable/colors.st.css.js");
var _filterObject = require("../utils/filterObject");
var _formatSpacingValues = require("./utils/formatSpacingValues");
var _formatSizeValues = require("./utils/formatSizeValues");
var _Box = require("./Box.constants");
var _excluded = ["inline", "direction", "children", "className", "align", "verticalAlign", "dataHook", "gap", "padding", "paddingTop", "paddingRight", "paddingBottom", "paddingLeft", "margin", "marginTop", "marginRight", "marginBottom", "marginLeft", "minWidth", "maxWidth", "width", "minHeight", "maxHeight", "height", "color", "backgroundColor", "border", "borderColor", "borderTopColor", "borderRightColor", "borderBottomColor", "borderLeftColor", "style", "data-hook"];
var _jsxFileName = "/home/builduser/work/57e038ea7326c1ec/packages/wix-design-system/dist/cjs/Box/Box.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function _interopRequireWildcard(e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (var _t in e) "default" !== _t && {}.hasOwnProperty.call(e, _t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, _t)) && (i.get || i.set) ? o(f, _t, i) : f[_t] = e[_t]); return f; })(e, t); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2.default)(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var getColorStyles = props => {
  var styles = {};
  for (var propName in props) {
    var propValue = props[propName];
    if (propValue) {
      styles[propName] = _colorsSt.stVars[propValue] || propValue;
    }
  }
  return styles;
};
var Box = /*#__PURE__*/(0, _react.forwardRef)((_ref, ref) => {
  var {
      inline = false,
      direction = _Box.DIRECTION.horizontal,
      children,
      className,
      align,
      verticalAlign,
      dataHook,
      gap,
      padding,
      paddingTop,
      paddingRight,
      paddingBottom,
      paddingLeft,
      margin,
      marginTop,
      marginRight,
      marginBottom,
      marginLeft,
      minWidth,
      maxWidth,
      width,
      minHeight,
      maxHeight,
      height,
      color,
      backgroundColor,
      border,
      borderColor,
      borderTopColor,
      borderRightColor,
      borderBottomColor,
      borderLeftColor,
      style,
      'data-hook': dataHookByKebabCase
    } = _ref,
    nativeStyles = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  var rootClassNames = (0, _BoxSt.st)(_BoxSt.classes.root, {
    inline,
    direction,
    alignItems: align,
    justifyContent: verticalAlign
  }, className);
  var rootStyles = _objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread({}, style), (0, _formatSpacingValues.getSpacingValues)({
    padding,
    paddingTop,
    paddingRight,
    paddingBottom,
    paddingLeft,
    margin,
    marginTop,
    marginRight,
    marginBottom,
    marginLeft
  })), (0, _formatSizeValues.getSizeValues)({
    minWidth,
    maxWidth,
    width,
    minHeight,
    maxHeight,
    height
  })), {}, {
    border
  }, getColorStyles({
    color,
    backgroundColor,
    borderColor,
    borderTopColor,
    borderRightColor,
    borderBottomColor,
    borderLeftColor
  })), {}, {
    [_BoxSt.vars['gap']]: (0, _formatSpacingValues.formatSpacingValue)(gap) || 0
  }, nativeStyles);
  var rootStylesFiltered = (0, _filterObject.filterObject)(rootStyles, (_key, value) => typeof value !== 'undefined');
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-hook": dataHook,
    className: rootClassNames,
    style: rootStylesFiltered,
    ref: ref,
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 129,
      columnNumber: 7
    }
  }, children);
});
Box.displayName = 'Box';
Box.propTypes = {
  children: _propTypes.default.any,
  className: _propTypes.default.any,
  inline: _propTypes.default.any,
  direction: _propTypes.default.any,
  align: _propTypes.default.any,
  verticalAlign: _propTypes.default.any,
  gap: _propTypes.default.any,
  padding: _propTypes.default.any,
  paddingTop: _propTypes.default.any,
  paddingRight: _propTypes.default.any,
  paddingBottom: _propTypes.default.any,
  paddingLeft: _propTypes.default.any,
  margin: _propTypes.default.any,
  marginTop: _propTypes.default.any,
  marginRight: _propTypes.default.any,
  marginBottom: _propTypes.default.any,
  marginLeft: _propTypes.default.any,
  minWidth: _propTypes.default.any,
  maxWidth: _propTypes.default.any,
  width: _propTypes.default.any,
  minHeight: _propTypes.default.any,
  maxHeight: _propTypes.default.any,
  height: _propTypes.default.any,
  color: _propTypes.default.any,
  backgroundColor: _propTypes.default.any,
  border: _propTypes.default.any,
  borderColor: _propTypes.default.any,
  borderTopColor: _propTypes.default.any,
  borderRightColor: _propTypes.default.any,
  borderBottomColor: _propTypes.default.any,
  borderLeftColor: _propTypes.default.any,
  dataHook: _propTypes.default.any,
  style: _propTypes.default.any
};
var _default = exports.default = Box;
