// @wix/headless-faq/dist/react/core/Faq.js
import { jsx as _jsx } from "react/jsx-runtime";
import { useService, WixServices } from '@wix/services-manager-react';
import { createServicesMap } from '@wix/services-manager';
import { FaqServiceDefinition, FaqService, } from '../../services/index.js';
/**
 * Root component that provides the FAQ service context to its children.
 * This component sets up the necessary services for rendering and managing a single FAQ entry's data.
 *
 * @component
 * @example
 * ```tsx
 * import { Faq } from '@wix/faq/components';
 *
 * function FaqPage() {
 *   return (
 *     <Faq.Root faqServiceConfig={{ faqs: [myFaq], categoryId: 'cat-1' }}>
 *       <Faq.Name>
 *         {({ question }) => (
 *           <h1 className="text-4xl font-bold">{question}</h1>
 *         )}
 *       </Faq.Name>
 *     </Faq.Root>
 *   );
 * }
 * ```
 */
export function Root(props) {
    return (_jsx(WixServices, { servicesMap: createServicesMap().addService(FaqServiceDefinition, FaqService, props.faqServiceConfig), children: props.children }));
}
/**
 * Headless component for FAQ question display
 *
 * @component
 * @example
 * ```tsx
 * import { Faq } from '@wix/faq/components';
 *
 * function FaqHeader() {
 *   return (
 *     <Faq.Name>
 *       {({ question }) => (
 *         <h1>{question}</h1>
 *       )}
 *     </Faq.Name>
 *   );
 * }
 * ```
 */
export function Name(props) {
    const service = useService(FaqServiceDefinition);
    const faqs = service.faqs.get();
    const faq = faqs[0]; // For single FAQ display, take the first FAQ
    const question = faq?.question || '';
    return props.children({
        question,
    });
}
/**
 * Headless component for FAQ item wrapper
 *
 * @component
 * @example
 * ```tsx
 * import { Faq } from '@wix/faq/components';
 *
 * function FaqItem() {
 *   return (
 *     <Faq.Item>
 *       {({ id }) => (
 *         <AccordionItem value={id}>
 *           {children}
 *         </AccordionItem>
 *       )}
 *     </Faq.Item>
 *   );
 * }
 * ```
 */
export function Item(props) {
    const service = useService(FaqServiceDefinition);
    const faqs = service.faqs.get();
    const faq = faqs[0]; // For single FAQ display, take the first FAQ
    const id = faq?._id || `faq-${Math.random()}`;
    return props.children({
        id,
    });
}
/**
 * Headless component for FAQ answer display
 *
 * @component
 * @example
 * ```tsx
 * import { Faq } from '@wix/faq/components';
 *
 * function FaqContent() {
 *   return (
 *     <Faq.Answer>
 *       {({ answer }) => (
 *         <p>{answer}</p>
 *       )}
 *     </Faq.Answer>
 *   );
 * }
 * ```
 */
export function Answer(props) {
    const service = useService(FaqServiceDefinition);
    const faqs = service.faqs.get();
    const faq = faqs[0]; // For single FAQ display, take the first FAQ
    const answer = faq?.plainText || faq?.draftjs || '';
    return props.children({
        answer,
    });
}
