// @wix/headless-site/dist/esm/host-module.js
import { createVeloMonitoringClient } from '@wix/monitoring-velo';
import { getPassThroughHeaders } from './headers';
import { resolveAuthToken } from './utils/auth-utils';
export const createHostModule = () => {
  return {
    __type: 'host',
    create: () => {
      return {};
    },
    host: function (_temp) {
      let {
        essentials
      } = _temp === void 0 ? {} : _temp;
      return {
        essentials: {
          get passThroughHeaders() {
            return getPassThroughHeaders();
          },
          get language() {
            var _window$essentials;
            return (essentials == null ? void 0 : essentials.language) ?? ((_window$essentials = window.essentials) == null ? void 0 : _window$essentials.language) ?? 'en';
          },
          get locale() {
            var _window$essentials2;
            return (essentials == null ? void 0 : essentials.locale) ?? ((_window$essentials2 = window.essentials) == null ? void 0 : _window$essentials2.locale) ?? 'en-US';
          },
          get timezone() {
            var _window$essentials3;
            return (essentials == null ? void 0 : essentials.timezone) ?? ((_window$essentials3 = window.essentials) == null ? void 0 : _window$essentials3.timeZone) ?? 'UTC';
          }
        },
        getMonitoringClient: () => {
          let producer = 'unknown';
          let req;
          if (typeof document !== 'undefined') {
            producer = document.location.href;
            req = new Request(document.location.href);
          }
          return createVeloMonitoringClient({
            namespace: 'Headless',
            producer,
            getAuthToken: async () => resolveAuthToken(),
            req
          });
        }
      };
    }
  };
};
