// @wix/design-system/dist/cjs/Notification/ActionButton/ActionButton.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.default = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireDefault(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _Button = _interopRequireDefault(require("../../Button"));
var _TextButton = _interopRequireDefault(require("../../TextButton"));
var _constants = require("../constants");
var _NotificationSt = require("../Notification.st.css.js");
var _jsxFileName = "/home/builduser/work/57e038ea7326c1ec/packages/wix-design-system/dist/cjs/Notification/ActionButton/ActionButton.js";
var ActionButton = _ref => {
  var {
    children,
    onClick,
    type = 'button',
    link,
    target = '_self'
  } = _ref;
  var commonProps = {
    dataHook: _constants.dataHooks.notificationCtaButton,
    onClick
  };
  if (type === 'textLink') {
    return /*#__PURE__*/_react.default.createElement(_TextButton.default, (0, _extends2.default)({
      className: _NotificationSt.classes.textLink,
      underline: "always",
      skin: "light",
      as: "a",
      href: link,
      target: target
    }, commonProps, {
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 22,
        columnNumber: 7
      }
    }), children);
  } else {
    return /*#__PURE__*/_react.default.createElement(_Button.default, (0, _extends2.default)({
      className: _NotificationSt.classes.button,
      type: "button",
      size: "small",
      skin: "transparent"
    }, commonProps, {
      __self: void 0,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 36,
        columnNumber: 7
      }
    }), children);
  }
};
ActionButton.propTypes = {
  children: _propTypes.default.any,
  onClick: _propTypes.default.func,
  link: _propTypes.default.string,
  type: _propTypes.default.string,
  target: _propTypes.default.string
};
ActionButton.displayName = 'Notification.ActionButton';
var _default = exports.default = ActionButton;
