// @wix/auto_sdk_online-programs_instructors/build/es/index.typings.mjs
// src/achievements-instructors-v2-instructor-instructors.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/achievements-instructors-v2-instructor-instructors.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixAchievementsInstructorsV2InstructorsServiceUrl(opts) {
  const domainToMappings = {
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/instructors-service",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/instructors-service",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/instructors-service",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/instructors-service",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/instructors-service",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/instructors-service",
        destPath: ""
      },
      {
        srcPath: "/_api/instructors-service",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/instructors-service",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/instructors-service",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/instructors-service",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_online-programs_instructors";
function createInstructor(payload) {
  function __createInstructor({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "instructor.createdDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.achievements.instructors.v2.instructor",
      method: "POST",
      methodFqn: "wix.achievements.instructors.v2.InstructorsService.CreateInstructor",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAchievementsInstructorsV2InstructorsServiceUrl({
        protoPath: "/v2/instructors",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "instructor.createdDate" }]
        }
      ])
    };
    return metadata;
  }
  return __createInstructor;
}
function inviteInstructor(payload) {
  function __inviteInstructor({ host }) {
    const metadata = {
      entityFqdn: "wix.achievements.instructors.v2.instructor",
      method: "POST",
      methodFqn: "wix.achievements.instructors.v2.InstructorsService.InviteInstructor",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAchievementsInstructorsV2InstructorsServiceUrl({
        protoPath: "/v2/instructors/invite",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __inviteInstructor;
}
function listInstructors(payload) {
  function __listInstructors({ host }) {
    const metadata = {
      entityFqdn: "wix.achievements.instructors.v2.instructor",
      method: "POST",
      methodFqn: "wix.achievements.instructors.v2.InstructorsService.ListInstructors",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAchievementsInstructorsV2InstructorsServiceUrl({
        protoPath: "/v2/instructors/list",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "instructors.createdDate" }]
        }
      ])
    };
    return metadata;
  }
  return __listInstructors;
}
function queryInstructors(payload) {
  function __queryInstructors({ host }) {
    const metadata = {
      entityFqdn: "wix.achievements.instructors.v2.instructor",
      method: "POST",
      methodFqn: "wix.achievements.instructors.v2.InstructorsService.QueryInstructors",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAchievementsInstructorsV2InstructorsServiceUrl({
        protoPath: "/v2/instructors/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "instructors.createdDate" }]
        }
      ])
    };
    return metadata;
  }
  return __queryInstructors;
}
function updateInstructor(payload) {
  function __updateInstructor({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "instructor.createdDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.achievements.instructors.v2.instructor",
      method: "PATCH",
      methodFqn: "wix.achievements.instructors.v2.InstructorsService.UpdateInstructor",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAchievementsInstructorsV2InstructorsServiceUrl({
        protoPath: "/v2/instructors/{instructor.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "instructor.createdDate" }]
        }
      ])
    };
    return metadata;
  }
  return __updateInstructor;
}
function deleteInstructor(payload) {
  function __deleteInstructor({ host }) {
    const metadata = {
      entityFqdn: "wix.achievements.instructors.v2.instructor",
      method: "DELETE",
      methodFqn: "wix.achievements.instructors.v2.InstructorsService.DeleteInstructor",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAchievementsInstructorsV2InstructorsServiceUrl({
        protoPath: "/v2/instructors/{instructorId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "instructor.createdDate" }]
        }
      ])
    };
    return metadata;
  }
  return __deleteInstructor;
}
function assignInstructorToProgram(payload) {
  function __assignInstructorToProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.achievements.instructors.v2.instructor",
      method: "POST",
      methodFqn: "wix.achievements.instructors.v2.InstructorsService.AssignInstructorToProgram",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAchievementsInstructorsV2InstructorsServiceUrl({
        protoPath: "/v2/instructors/{instructorId}/assign",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __assignInstructorToProgram;
}
function unassignInstructorFromProgram(payload) {
  function __unassignInstructorFromProgram({ host }) {
    const metadata = {
      entityFqdn: "wix.achievements.instructors.v2.instructor",
      method: "POST",
      methodFqn: "wix.achievements.instructors.v2.InstructorsService.UnassignInstructorFromProgram",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAchievementsInstructorsV2InstructorsServiceUrl({
        protoPath: "/v2/instructors/{instructorId}/unassign",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __unassignInstructorFromProgram;
}
function changeProgramInstructors(payload) {
  function __changeProgramInstructors({ host }) {
    const metadata = {
      entityFqdn: "wix.achievements.instructors.v2.instructor",
      method: "POST",
      methodFqn: "wix.achievements.instructors.v2.InstructorsService.ChangeProgramInstructors",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixAchievementsInstructorsV2InstructorsServiceUrl({
        protoPath: "/v2/assignments",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __changeProgramInstructors;
}

// src/achievements-instructors-v2-instructor-instructors.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var Role = /* @__PURE__ */ ((Role2) => {
  Role2["DEFAULT"] = "DEFAULT";
  Role2["OWNER"] = "OWNER";
  Role2["INSTRUCTOR"] = "INSTRUCTOR";
  Role2["MANAGER"] = "MANAGER";
  return Role2;
})(Role || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var State = /* @__PURE__ */ ((State2) => {
  State2["UNKNOWN"] = "UNKNOWN";
  State2["ENABLED"] = "ENABLED";
  State2["DISABLED"] = "DISABLED";
  State2["PENDING"] = "PENDING";
  State2["DEMO"] = "DEMO";
  return State2;
})(State || {});
var SiteCreatedContext = /* @__PURE__ */ ((SiteCreatedContext2) => {
  SiteCreatedContext2["OTHER"] = "OTHER";
  SiteCreatedContext2["FROM_TEMPLATE"] = "FROM_TEMPLATE";
  SiteCreatedContext2["DUPLICATE_BY_SITE_TRANSFER"] = "DUPLICATE_BY_SITE_TRANSFER";
  SiteCreatedContext2["DUPLICATE"] = "DUPLICATE";
  SiteCreatedContext2["OLD_SITE_TRANSFER"] = "OLD_SITE_TRANSFER";
  SiteCreatedContext2["FLASH"] = "FLASH";
  return SiteCreatedContext2;
})(SiteCreatedContext || {});
var Namespace = /* @__PURE__ */ ((Namespace2) => {
  Namespace2["UNKNOWN_NAMESPACE"] = "UNKNOWN_NAMESPACE";
  Namespace2["WIX"] = "WIX";
  Namespace2["SHOUT_OUT"] = "SHOUT_OUT";
  Namespace2["ALBUMS"] = "ALBUMS";
  Namespace2["WIX_STORES_TEST_DRIVE"] = "WIX_STORES_TEST_DRIVE";
  Namespace2["HOTELS"] = "HOTELS";
  Namespace2["CLUBS"] = "CLUBS";
  Namespace2["ONBOARDING_DRAFT"] = "ONBOARDING_DRAFT";
  Namespace2["DEV_SITE"] = "DEV_SITE";
  Namespace2["LOGOS"] = "LOGOS";
  Namespace2["VIDEO_MAKER"] = "VIDEO_MAKER";
  Namespace2["PARTNER_DASHBOARD"] = "PARTNER_DASHBOARD";
  Namespace2["DEV_CENTER_COMPANY"] = "DEV_CENTER_COMPANY";
  Namespace2["HTML_DRAFT"] = "HTML_DRAFT";
  Namespace2["SITELESS_BUSINESS"] = "SITELESS_BUSINESS";
  Namespace2["CREATOR_ECONOMY"] = "CREATOR_ECONOMY";
  Namespace2["DASHBOARD_FIRST"] = "DASHBOARD_FIRST";
  Namespace2["ANYWHERE"] = "ANYWHERE";
  Namespace2["HEADLESS"] = "HEADLESS";
  Namespace2["ACCOUNT_MASTER_CMS"] = "ACCOUNT_MASTER_CMS";
  Namespace2["RISE"] = "RISE";
  Namespace2["BRANDED_FIRST"] = "BRANDED_FIRST";
  Namespace2["NOWNIA"] = "NOWNIA";
  Namespace2["UGC_TEMPLATE"] = "UGC_TEMPLATE";
  Namespace2["CODUX"] = "CODUX";
  Namespace2["MEDIA_DESIGN_CREATOR"] = "MEDIA_DESIGN_CREATOR";
  Namespace2["SHARED_BLOG_ENTERPRISE"] = "SHARED_BLOG_ENTERPRISE";
  Namespace2["STANDALONE_FORMS"] = "STANDALONE_FORMS";
  Namespace2["STANDALONE_EVENTS"] = "STANDALONE_EVENTS";
  Namespace2["MIMIR"] = "MIMIR";
  Namespace2["TWINS"] = "TWINS";
  Namespace2["NANO"] = "NANO";
  return Namespace2;
})(Namespace || {});
var DeleteStatus = /* @__PURE__ */ ((DeleteStatus2) => {
  DeleteStatus2["UNKNOWN"] = "UNKNOWN";
  DeleteStatus2["TRASH"] = "TRASH";
  DeleteStatus2["DELETED"] = "DELETED";
  DeleteStatus2["PENDING_PURGE"] = "PENDING_PURGE";
  DeleteStatus2["PURGED_EXTERNALLY"] = "PURGED_EXTERNALLY";
  return DeleteStatus2;
})(DeleteStatus || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createInstructor2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    instructor: options?.instructor,
    actionId: options?.actionId
  });
  const reqOpts = createInstructor(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.instructor;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          instructor: "$[0].instructor",
          actionId: "$[0].actionId"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function inviteInstructor2(email) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ email });
  const reqOpts = inviteInstructor(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { email: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["email"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listInstructors2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    programIdsFilter: options?.programIdsFilter,
    paging: options?.paging
  });
  const reqOpts = listInstructors(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          programIdsFilter: "$[0].programIdsFilter",
          paging: "$[0].paging"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryInstructors2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryInstructors(
        payload
      );
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [])
      );
      return {
        items: transformedData?.instructors,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function updateInstructor2(_id, instructor, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    instructor: { ...instructor, id: _id },
    actionId: options?.actionId
  });
  const reqOpts = updateInstructor(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.instructor;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { instructor: "$[1]" },
        explicitPathsToArguments: {
          "instructor.id": "$[0]",
          actionId: "$[2].actionId"
        },
        singleArgumentUnchanged: false
      },
      ["_id", "instructor", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteInstructor2(instructorId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    instructorId,
    actionId: options?.actionId
  });
  const reqOpts = deleteInstructor(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          instructorId: "$[0]",
          actionId: "$[1].actionId"
        },
        singleArgumentUnchanged: false
      },
      ["instructorId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function assignInstructorToProgram2(instructorId, programId, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    instructorId,
    programId,
    actionId: options?.actionId
  });
  const reqOpts = assignInstructorToProgram(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          instructorId: "$[0]",
          programId: "$[1]",
          actionId: "$[2].actionId"
        },
        singleArgumentUnchanged: false
      },
      ["instructorId", "programId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function unassignInstructorFromProgram2(instructorId, programId, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    instructorId,
    programId,
    actionId: options?.actionId
  });
  const reqOpts = unassignInstructorFromProgram(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          instructorId: "$[0]",
          programId: "$[1]",
          actionId: "$[2].actionId"
        },
        singleArgumentUnchanged: false
      },
      ["instructorId", "programId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function changeProgramInstructors2(programId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    programId,
    assignInstructorIds: options?.assignInstructorIds,
    unassignInstructorIds: options?.unassignInstructorIds,
    actionId: options?.actionId
  });
  const reqOpts = changeProgramInstructors(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          programId: "$[0]",
          assignInstructorIds: "$[1].assignInstructorIds",
          unassignInstructorIds: "$[1].unassignInstructorIds",
          actionId: "$[1].actionId"
        },
        singleArgumentUnchanged: false
      },
      ["programId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
export {
  DeleteStatus,
  Namespace,
  Role,
  SiteCreatedContext,
  SortOrder,
  State,
  WebhookIdentityType,
  assignInstructorToProgram2 as assignInstructorToProgram,
  changeProgramInstructors2 as changeProgramInstructors,
  createInstructor2 as createInstructor,
  deleteInstructor2 as deleteInstructor,
  inviteInstructor2 as inviteInstructor,
  listInstructors2 as listInstructors,
  queryInstructors2 as queryInstructors,
  unassignInstructorFromProgram2 as unassignInstructorFromProgram,
  updateInstructor2 as updateInstructor
};
