// @wix/ricos/dist/cjs/ricos-content/velo-adapter/builder.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var builder_exports = {};
__export(builder_exports, {
  createBuilder: () => createBuilder
});
module.exports = __toCommonJS(builder_exports);
var import_builder_utils = require("../RicosContentAPI/builder-utils.js");
class Builder {
  constructor(content, onRender) {
    this.content = content;
    this.onRender = onRender;
  }
  add(addNodeParams) {
    this.content = (0, import_builder_utils.addNode)(addNodeParams);
    return this;
  }
  append(node) {
    return this.add({ node, content: this.content });
  }
  insertBefore(id, node) {
    return this.add({ node, content: this.content, before: id });
  }
  insertAfter(id, node) {
    return this.add({ node, content: this.content, after: id });
  }
  insertAt(index, node) {
    return this.add({ node, content: this.content, index });
  }
  render() {
    this.onRender?.(this.content);
    return this.content;
  }
}
const createBuilder = (content, onRender) => new Builder(content, onRender);
