// @wix/auto_sdk_restaurants_pos-tpa/build/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  disconnectAccount: () => disconnectAccount4,
  disconnectConfiguration: () => disconnectConfiguration4,
  getConfiguration: () => getConfiguration4,
  listConfigurations: () => listConfigurations4
});
module.exports = __toCommonJS(index_exports);

// src/restaurants-postpaapi-v1-configuration-pos-tpa.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/restaurants-postpaapi-v1-configuration-pos-tpa.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressRestaurantsPostpaapiRestaurantsPosTpaApiUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/restaurants/v1/pos-spi-host",
        destPath: "/v1/pos-spi-host"
      }
    ],
    "com.wixpress.restaurants.restaurants-pos-spi-host.wixapps.net": [
      {
        srcPath: "/restaurants-pos-spi-host",
        destPath: "/v1/pos-spi-host"
      }
    ],
    "www.wixapps.com": [
      {
        srcPath: "/restaurants/v1/pos-integration",
        destPath: "/v1/pos-integration"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/v1/pos-integration",
        destPath: "/v1/pos-integration"
      }
    ],
    "users._base_domain": [
      {
        srcPath: "/_api/restaurants/v1/pos-spi-host",
        destPath: "/v1/pos-spi-host"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_pos-tpa";
function getConfiguration(payload) {
  function __getConfiguration({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.postpaapi.v1.configuration",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.postpaapi.RestaurantsPosTpaApi.GetConfiguration",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsPostpaapiRestaurantsPosTpaApiUrl({
        protoPath: "/v1/pos-integration/configurations/{id}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getConfiguration;
}
function listConfigurations(payload) {
  function __listConfigurations({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.postpaapi.v1.configuration",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.postpaapi.RestaurantsPosTpaApi.ListConfigurations",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsPostpaapiRestaurantsPosTpaApiUrl({
        protoPath: "/v1/pos-integration/configurations",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __listConfigurations;
}
function disconnectAccount(payload) {
  function __disconnectAccount({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.postpaapi.v1.configuration",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.postpaapi.RestaurantsPosTpaApi.DisconnectAccount",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsPostpaapiRestaurantsPosTpaApiUrl({
        protoPath: "/v1/pos-integration/accounts/{accountId}/disconnect",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __disconnectAccount;
}
function disconnectConfiguration(payload) {
  function __disconnectConfiguration({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.postpaapi.v1.configuration",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.postpaapi.RestaurantsPosTpaApi.DisconnectConfiguration",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsPostpaapiRestaurantsPosTpaApiUrl({
        protoPath: "/v1/pos-integration/configurations/{configurationId}/disconnect",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __disconnectConfiguration;
}

// src/restaurants-postpaapi-v1-configuration-pos-tpa.universal.ts
async function getConfiguration2(_id) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ id: _id });
  const reqOpts = getConfiguration(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.configuration;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { id: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listConfigurations2() {
  const { httpClient, sideEffects } = arguments[0];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({});
  const reqOpts = listConfigurations(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {},
        singleArgumentUnchanged: false
      },
      []
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function disconnectAccount2(accountId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    accountId,
    appDefId: options?.appDefId
  });
  const reqOpts = disconnectAccount(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          accountId: "$[0]",
          appDefId: "$[1].appDefId"
        },
        singleArgumentUnchanged: false
      },
      ["accountId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function disconnectConfiguration2(configurationId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    configurationId
  });
  const reqOpts = disconnectConfiguration(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { configurationId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["configurationId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/restaurants-postpaapi-v1-configuration-pos-tpa.public.ts
function getConfiguration3(httpClient) {
  return (_id) => getConfiguration2(
    _id,
    // @ts-ignore
    { httpClient }
  );
}
function listConfigurations3(httpClient) {
  return () => listConfigurations2(
    // @ts-ignore
    { httpClient }
  );
}
function disconnectAccount3(httpClient) {
  return (accountId, options) => disconnectAccount2(
    accountId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function disconnectConfiguration3(httpClient) {
  return (configurationId) => disconnectConfiguration2(
    configurationId,
    // @ts-ignore
    { httpClient }
  );
}

// src/restaurants-postpaapi-v1-configuration-pos-tpa.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var getConfiguration4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getConfiguration3);
var listConfigurations4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(listConfigurations3);
var disconnectAccount4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(disconnectAccount3);
var disconnectConfiguration4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(disconnectConfiguration3);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  disconnectAccount,
  disconnectConfiguration,
  getConfiguration,
  listConfigurations
});
