// @wix/auto_sdk_restaurants_pos-tpa/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  disconnectAccount: () => disconnectAccount2,
  disconnectConfiguration: () => disconnectConfiguration2,
  getConfiguration: () => getConfiguration2,
  listConfigurations: () => listConfigurations2
});
module.exports = __toCommonJS(meta_exports);

// src/restaurants-postpaapi-v1-configuration-pos-tpa.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressRestaurantsPostpaapiRestaurantsPosTpaApiUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/restaurants/v1/pos-spi-host",
        destPath: "/v1/pos-spi-host"
      }
    ],
    "com.wixpress.restaurants.restaurants-pos-spi-host.wixapps.net": [
      {
        srcPath: "/restaurants-pos-spi-host",
        destPath: "/v1/pos-spi-host"
      }
    ],
    "www.wixapps.com": [
      {
        srcPath: "/restaurants/v1/pos-integration",
        destPath: "/v1/pos-integration"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/v1/pos-integration",
        destPath: "/v1/pos-integration"
      }
    ],
    "users._base_domain": [
      {
        srcPath: "/_api/restaurants/v1/pos-spi-host",
        destPath: "/v1/pos-spi-host"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_pos-tpa";
function getConfiguration(payload) {
  function __getConfiguration({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.postpaapi.v1.configuration",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.postpaapi.RestaurantsPosTpaApi.GetConfiguration",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsPostpaapiRestaurantsPosTpaApiUrl({
        protoPath: "/v1/pos-integration/configurations/{id}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getConfiguration;
}
function listConfigurations(payload) {
  function __listConfigurations({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.postpaapi.v1.configuration",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.postpaapi.RestaurantsPosTpaApi.ListConfigurations",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsPostpaapiRestaurantsPosTpaApiUrl({
        protoPath: "/v1/pos-integration/configurations",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __listConfigurations;
}
function disconnectAccount(payload) {
  function __disconnectAccount({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.postpaapi.v1.configuration",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.postpaapi.RestaurantsPosTpaApi.DisconnectAccount",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsPostpaapiRestaurantsPosTpaApiUrl({
        protoPath: "/v1/pos-integration/accounts/{accountId}/disconnect",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __disconnectAccount;
}
function disconnectConfiguration(payload) {
  function __disconnectConfiguration({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.postpaapi.v1.configuration",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.postpaapi.RestaurantsPosTpaApi.DisconnectConfiguration",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsPostpaapiRestaurantsPosTpaApiUrl({
        protoPath: "/v1/pos-integration/configurations/{configurationId}/disconnect",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __disconnectConfiguration;
}

// src/restaurants-postpaapi-v1-configuration-pos-tpa.meta.ts
function getConfiguration2() {
  const payload = { id: ":id" };
  const getRequestOptions = getConfiguration(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/pos-integration/configurations/{id}",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listConfigurations2() {
  const payload = {};
  const getRequestOptions = listConfigurations(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/pos-integration/configurations",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function disconnectAccount2() {
  const payload = { accountId: ":accountId" };
  const getRequestOptions = disconnectAccount(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/pos-integration/accounts/{accountId}/disconnect",
    pathParams: { accountId: "accountId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function disconnectConfiguration2() {
  const payload = { configurationId: ":configurationId" };
  const getRequestOptions = disconnectConfiguration(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/pos-integration/configurations/{configurationId}/disconnect",
    pathParams: { configurationId: "configurationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  disconnectAccount,
  disconnectConfiguration,
  getConfiguration,
  listConfigurations
});
