// @wix/patterns/dist/esm/components/MoreActions/ExploreAppsState.js
import React from 'react';
import { action, makeObservable, observable } from 'mobx';
import { wixPatternsAppWasInstalledSrc144Evid200 } from '@wix/bex-core/bi';
import { ExploreAppsModal } from './ExploreAppsModal';
import { RelatedApps } from '@wix/wix-ui-icons-common';
export class ExploreAppsState {
    constructor({ modalProps, reportBi, translatedText, }) {
        this.isModalOpen = false;
        this.reportBi = (_reportProps) => { };
        this.modalProps = modalProps;
        this.moreActionsMenuItem = {
            biName: 'Explore Apps',
            text: translatedText,
            prefixIcon: React.createElement(RelatedApps, null),
            onClick: this.openExploreAppsModal.bind(this),
            modalProps,
        };
        this.reportBi = reportBi;
        this.Modal = ExploreAppsModal;
        makeObservable(this, {
            modalProps: observable.ref,
            moreActionsMenuItem: observable.ref,
            isModalOpen: observable,
            openExploreAppsModal: action,
            closeExploreAppsModal: action,
        });
    }
    openExploreAppsModal() {
        this.isModalOpen = true;
    }
    closeExploreAppsModal() {
        this.isModalOpen = false;
    }
    onInstallationSuccess({ appId }) {
        this.reportBi(wixPatternsAppWasInstalledSrc144Evid200({ appId }));
    }
}
