// @wix/auto_sdk_restaurants_orders/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  DiscountTypeOriginal: () => DiscountType,
  DisplayTypeOriginal: () => DisplayType,
  EventTypeOriginal: () => EventType,
  FieldNameOriginal: () => FieldName,
  FulfillmentTypeOriginal: () => FulfillmentType,
  OnArrivalOriginal: () => OnArrival,
  PaymentTypeOriginal: () => PaymentType,
  SortOrderOriginal: () => SortOrder,
  StatusOriginal: () => Status,
  TypeOriginal: () => Type,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  acceptOrder: () => acceptOrder2,
  cancelOrder: () => cancelOrder2,
  fulfillOrder: () => fulfillOrder2,
  getOrder: () => getOrder2,
  listOrders: () => listOrders2
});
module.exports = __toCommonJS(meta_exports);

// src/restaurants-v3-order-orders.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressRestaurantsOrdersV3OrdersApiUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/v3/admin/orders/notify-order-event",
        destPath: "/v3/admin/orders/notify-order-event"
      },
      {
        srcPath: "/restaurants/v3/orders",
        destPath: "/v3/orders"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/restaurants/v3/orders",
        destPath: "/v3/orders"
      }
    ],
    "users._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/orders",
        destPath: "/v3/orders"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/orders",
        destPath: "/v3/orders"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants/v3/orders",
        destPath: "/v3/orders"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants/v3/orders",
        destPath: "/v3/orders"
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants/v3/orders",
        destPath: "/v3/orders"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants/v3/orders",
        destPath: "/v3/orders"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_orders";
function getOrder(payload) {
  function __getOrder({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v3.order",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.orders.v3.OrdersApi.GetOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsOrdersV3OrdersApiUrl({
        protoPath: "/v3/orders/{id}",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.createdDate" },
            { path: "order.updatedDate" },
            { path: "order.fulfillment.promisedTime" },
            {
              path: "order.fulfillment.deliveryDetails.externalProvider.pickupTime"
            },
            {
              path: "order.fulfillment.deliveryDetails.externalProvider.dropOffTime"
            },
            {
              path: "order.fulfillment.deliveryDetails.pickupInfo.windowStartTime"
            },
            {
              path: "order.fulfillment.deliveryDetails.pickupInfo.windowEndTime"
            },
            {
              path: "order.fulfillment.deliveryDetails.dropOffInfo.windowStartTime"
            },
            {
              path: "order.fulfillment.deliveryDetails.dropOffInfo.windowEndTime"
            },
            { path: "order.activities.timestamp" }
          ]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "order.fulfillment.deliveryDetails.address.location.latitude"
            },
            {
              path: "order.fulfillment.deliveryDetails.address.location.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getOrder;
}
function listOrders(payload) {
  function __listOrders({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v3.order",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.orders.v3.OrdersApi.ListOrders",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsOrdersV3OrdersApiUrl({
        protoPath: "/v3/orders",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "orders.createdDate" },
            { path: "orders.updatedDate" },
            { path: "orders.fulfillment.promisedTime" },
            {
              path: "orders.fulfillment.deliveryDetails.externalProvider.pickupTime"
            },
            {
              path: "orders.fulfillment.deliveryDetails.externalProvider.dropOffTime"
            },
            {
              path: "orders.fulfillment.deliveryDetails.pickupInfo.windowStartTime"
            },
            {
              path: "orders.fulfillment.deliveryDetails.pickupInfo.windowEndTime"
            },
            {
              path: "orders.fulfillment.deliveryDetails.dropOffInfo.windowStartTime"
            },
            {
              path: "orders.fulfillment.deliveryDetails.dropOffInfo.windowEndTime"
            },
            { path: "orders.activities.timestamp" }
          ]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "orders.fulfillment.deliveryDetails.address.location.latitude"
            },
            {
              path: "orders.fulfillment.deliveryDetails.address.location.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listOrders;
}
function acceptOrder(payload) {
  function __acceptOrder({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "pickupTime" }, { path: "dropOffTime" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.v3.order",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.orders.v3.OrdersApi.AcceptOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsOrdersV3OrdersApiUrl({
        protoPath: "/v3/orders/{id}/accept",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.createdDate" },
            { path: "order.updatedDate" },
            { path: "order.fulfillment.promisedTime" },
            {
              path: "order.fulfillment.deliveryDetails.externalProvider.pickupTime"
            },
            {
              path: "order.fulfillment.deliveryDetails.externalProvider.dropOffTime"
            },
            {
              path: "order.fulfillment.deliveryDetails.pickupInfo.windowStartTime"
            },
            {
              path: "order.fulfillment.deliveryDetails.pickupInfo.windowEndTime"
            },
            {
              path: "order.fulfillment.deliveryDetails.dropOffInfo.windowStartTime"
            },
            {
              path: "order.fulfillment.deliveryDetails.dropOffInfo.windowEndTime"
            },
            { path: "order.activities.timestamp" }
          ]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "order.fulfillment.deliveryDetails.address.location.latitude"
            },
            {
              path: "order.fulfillment.deliveryDetails.address.location.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __acceptOrder;
}
function cancelOrder(payload) {
  function __cancelOrder({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v3.order",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.orders.v3.OrdersApi.CancelOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsOrdersV3OrdersApiUrl({
        protoPath: "/v3/orders/{id}/cancel",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.createdDate" },
            { path: "order.updatedDate" },
            { path: "order.fulfillment.promisedTime" },
            {
              path: "order.fulfillment.deliveryDetails.externalProvider.pickupTime"
            },
            {
              path: "order.fulfillment.deliveryDetails.externalProvider.dropOffTime"
            },
            {
              path: "order.fulfillment.deliveryDetails.pickupInfo.windowStartTime"
            },
            {
              path: "order.fulfillment.deliveryDetails.pickupInfo.windowEndTime"
            },
            {
              path: "order.fulfillment.deliveryDetails.dropOffInfo.windowStartTime"
            },
            {
              path: "order.fulfillment.deliveryDetails.dropOffInfo.windowEndTime"
            },
            { path: "order.activities.timestamp" }
          ]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "order.fulfillment.deliveryDetails.address.location.latitude"
            },
            {
              path: "order.fulfillment.deliveryDetails.address.location.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __cancelOrder;
}
function fulfillOrder(payload) {
  function __fulfillOrder({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.v3.order",
      method: "POST",
      methodFqn: "com.wixpress.restaurants.orders.v3.OrdersApi.FulfillOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsOrdersV3OrdersApiUrl({
        protoPath: "/v3/orders/{id}/fulfill",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.createdDate" },
            { path: "order.updatedDate" },
            { path: "order.fulfillment.promisedTime" },
            {
              path: "order.fulfillment.deliveryDetails.externalProvider.pickupTime"
            },
            {
              path: "order.fulfillment.deliveryDetails.externalProvider.dropOffTime"
            },
            {
              path: "order.fulfillment.deliveryDetails.pickupInfo.windowStartTime"
            },
            {
              path: "order.fulfillment.deliveryDetails.pickupInfo.windowEndTime"
            },
            {
              path: "order.fulfillment.deliveryDetails.dropOffInfo.windowStartTime"
            },
            {
              path: "order.fulfillment.deliveryDetails.dropOffInfo.windowEndTime"
            },
            { path: "order.activities.timestamp" }
          ]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "order.fulfillment.deliveryDetails.address.location.latitude"
            },
            {
              path: "order.fulfillment.deliveryDetails.address.location.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __fulfillOrder;
}

// src/restaurants-v3-order-orders.types.ts
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["UNSPECIFIED_ORDER_STATUS"] = "UNSPECIFIED_ORDER_STATUS";
  Status2["PENDING"] = "PENDING";
  Status2["NEW"] = "NEW";
  Status2["ACCEPTED"] = "ACCEPTED";
  Status2["CANCELED"] = "CANCELED";
  Status2["FULFILLED"] = "FULFILLED";
  return Status2;
})(Status || {});
var DisplayType = /* @__PURE__ */ ((DisplayType2) => {
  DisplayType2["UNSPECIFIED_DISPLAY_TYPE"] = "UNSPECIFIED_DISPLAY_TYPE";
  DisplayType2["SELECTION"] = "SELECTION";
  DisplayType2["EXTRAS"] = "EXTRAS";
  DisplayType2["DESELECTION"] = "DESELECTION";
  return DisplayType2;
})(DisplayType || {});
var DiscountType = /* @__PURE__ */ ((DiscountType2) => {
  DiscountType2["UNSPECIFIED_TYPE"] = "UNSPECIFIED_TYPE";
  DiscountType2["OFF_ITEM"] = "OFF_ITEM";
  DiscountType2["OFF_ORDER"] = "OFF_ORDER";
  DiscountType2["OFF_ORDER_MANAGER_DISCOUNT"] = "OFF_ORDER_MANAGER_DISCOUNT";
  return DiscountType2;
})(DiscountType || {});
var PaymentType = /* @__PURE__ */ ((PaymentType2) => {
  PaymentType2["UNSPECIFIED_PAYMENT_TYPE"] = "UNSPECIFIED_PAYMENT_TYPE";
  PaymentType2["CASH"] = "CASH";
  PaymentType2["CREDIT"] = "CREDIT";
  PaymentType2["DELIVERY_CLUB"] = "DELIVERY_CLUB";
  PaymentType2["DELIVERY_COM"] = "DELIVERY_COM";
  PaymentType2["BITPAY"] = "BITPAY";
  PaymentType2["CELLARIX"] = "CELLARIX";
  PaymentType2["BITSOFGOLD"] = "BITSOFGOLD";
  PaymentType2["MULTIPASS"] = "MULTIPASS";
  PaymentType2["TENBIS"] = "TENBIS";
  PaymentType2["PAYPAL"] = "PAYPAL";
  PaymentType2["MYSODEXO"] = "MYSODEXO";
  PaymentType2["WIX_PAYMENTS"] = "WIX_PAYMENTS";
  return PaymentType2;
})(PaymentType || {});
var FulfillmentType = /* @__PURE__ */ ((FulfillmentType2) => {
  FulfillmentType2["UNSPECIFIED_FULFILLMENT_TYPE"] = "UNSPECIFIED_FULFILLMENT_TYPE";
  FulfillmentType2["PICKUP"] = "PICKUP";
  FulfillmentType2["DELIVERY"] = "DELIVERY";
  FulfillmentType2["DINE_IN"] = "DINE_IN";
  return FulfillmentType2;
})(FulfillmentType || {});
var OnArrival = /* @__PURE__ */ ((OnArrival2) => {
  OnArrival2["UNSPECIFIED_ON_ARRIVAL_TYPE"] = "UNSPECIFIED_ON_ARRIVAL_TYPE";
  OnArrival2["BUZZ_DOOR"] = "BUZZ_DOOR";
  OnArrival2["CALL_ME"] = "CALL_ME";
  return OnArrival2;
})(OnArrival || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNSPECIFIED_CHANNEL_TYPE"] = "UNSPECIFIED_CHANNEL_TYPE";
  Type2["WEB"] = "WEB";
  Type2["MOBILE_WEB"] = "MOBILE_WEB";
  Type2["MOBILE"] = "MOBILE";
  Type2["CALL_CENTER"] = "CALL_CENTER";
  Type2["FACEBOOK"] = "FACEBOOK";
  Type2["TPA"] = "TPA";
  return Type2;
})(Type || {});
var FieldName = /* @__PURE__ */ ((FieldName2) => {
  FieldName2["CREATED_DATE"] = "CREATED_DATE";
  FieldName2["UPDATED_DATE"] = "UPDATED_DATE";
  FieldName2["PROMISED_TIME"] = "PROMISED_TIME";
  return FieldName2;
})(FieldName || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
var EventType = /* @__PURE__ */ ((EventType2) => {
  EventType2["UNKNOWN"] = "UNKNOWN";
  EventType2["CREATED"] = "CREATED";
  EventType2["ACCEPTED"] = "ACCEPTED";
  EventType2["CANCELED"] = "CANCELED";
  EventType2["FULFILLED"] = "FULFILLED";
  return EventType2;
})(EventType || {});

// src/restaurants-v3-order-orders.meta.ts
function getOrder2() {
  const payload = { id: ":id" };
  const getRequestOptions = getOrder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/orders/{id}",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listOrders2() {
  const payload = {};
  const getRequestOptions = listOrders(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/orders",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function acceptOrder2() {
  const payload = { id: ":id" };
  const getRequestOptions = acceptOrder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/orders/{id}/accept",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function cancelOrder2() {
  const payload = { id: ":id" };
  const getRequestOptions = cancelOrder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/orders/{id}/cancel",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function fulfillOrder2() {
  const payload = { id: ":id" };
  const getRequestOptions = fulfillOrder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/orders/{id}/fulfill",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  DiscountTypeOriginal,
  DisplayTypeOriginal,
  EventTypeOriginal,
  FieldNameOriginal,
  FulfillmentTypeOriginal,
  OnArrivalOriginal,
  PaymentTypeOriginal,
  SortOrderOriginal,
  StatusOriginal,
  TypeOriginal,
  WebhookIdentityTypeOriginal,
  acceptOrder,
  cancelOrder,
  fulfillOrder,
  getOrder,
  listOrders
});
