// @wix/wix-ui-test-utils/dist/cjs/react-helpers/helpers.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.isClassExists = void 0;
exports.makeControlled = makeControlled;
exports.sleep = exports.reactEventTrigger = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var React = _interopRequireWildcard(require("react"));
var _testUtils = require("react-dom/test-utils");
var _jsxFileName = "/home/builduser/work/c98d836f8565ce5f/packages/wix-ui-test-utils/dist/cjs/react-helpers/helpers.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
const isClassExists = (element, className) => !!element && !!element.className.match(new RegExp('\\b' + className + '\\b'));
exports.isClassExists = isClassExists;
const sleep = ms => {
  return new Promise(resolve => setTimeout(resolve, ms));
};

// HOC that makes underlying component "controlled"
/** @deprecated */
exports.sleep = sleep;
function makeControlled(Component) {
  var _ControlledComponent;
  return _ControlledComponent = class ControlledComponent extends React.Component {
    constructor(...args) {
      super(...args);
      (0, _defineProperty2.default)(this, "state", {
        value: this.props.value
      });
      (0, _defineProperty2.default)(this, "_onChange", e => {
        const {
          onChange
        } = this.props;
        this.setState({
          value: e.target.value
        });
        onChange && onChange(e);
      });
    }
    render() {
      const bindedPropMethods = {};
      for (const propName of Object.keys(this.props)) {
        const propValue = this.props[propName];
        if (typeof propValue === 'function') {
          bindedPropMethods[propName] = this.props[propName].bind(this);
        }
      }
      return /*#__PURE__*/React.createElement(Component, (0, _extends2.default)({}, this.props, bindedPropMethods, {
        value: this.state.value,
        onChange: this._onChange,
        __self: this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 69,
          columnNumber: 9
        }
      }));
    }
  }, (0, _defineProperty2.default)(_ControlledComponent, "defaultProps", {
    value: ''
  }), _ControlledComponent;
}
const reactEventTrigger = () => _testUtils.Simulate;
exports.reactEventTrigger = reactEventTrigger;
