// @wix/ricos/dist/es/ricos-common-plugins/blockquote/blockquote-converter.js
import { Node_Type } from "./../../ricos-schema/legacy/wix/rich_content/v1/index.js";
import blockquoteDefaultData from "./../../ricos-schema/legacy/statics/blockquote.defaults.json.js";
const toParagraphFields = (node) => {
  const { id: paragraphId, paragraphData = {} } = node;
  return { paragraphData, paragraphId };
};
const toParagraphNode = (fields) => (nodes) => {
  return {
    id: fields.paragraphId,
    type: Node_Type.PARAGRAPH,
    paragraphData: fields.paragraphData,
    nodes
  };
};
const blockquoteConverter = {
  toPm: {
    types: [Node_Type.BLOCKQUOTE],
    // @ts-expect-error $TSFixMe
    convert: (node, visit) => {
      const { id, style, blockquoteData } = node || {};
      const firstNode = node.nodes?.[0] || { id: "", paragraphData: {} };
      const { paragraphData, paragraphId } = toParagraphFields(firstNode);
      return {
        type: Node_Type.BLOCKQUOTE,
        attrs: {
          paragraphId,
          textStyle: paragraphData.textStyle,
          indentation: blockquoteData?.indentation ?? blockquoteDefaultData.indentation,
          ...style ? { style } : {},
          id
        },
        // skip internal paragraph, go to text nodes
        content: node.nodes?.[0] ? visit(node.nodes[0]) : []
      };
    }
  },
  fromPm: {
    types: [Node_Type.BLOCKQUOTE],
    // @ts-expect-error $TSFixMe
    convert: (node, visit) => {
      const { id, style, paragraphId, textStyle, indentation } = node.attrs || {};
      const paragraphNode = toParagraphNode({
        paragraphData: { indentation: 0, textStyle },
        paragraphId
      })(visit(node));
      return {
        type: Node_Type.BLOCKQUOTE,
        id,
        nodes: [paragraphNode],
        ...style ? { style } : {},
        blockquoteData: {
          indentation
        }
      };
    }
  }
};
export {
  blockquoteConverter
};
