// @wix/ricos/dist/cjs/ricos-common-plugins/layout/cell-converter.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var cell_converter_exports = {};
__export(cell_converter_exports, {
  cellConverter: () => cellConverter
});
module.exports = __toCommonJS(cell_converter_exports);
var import_v1 = require("./../../ricos-schema/legacy/wix/rich_content/v1/index.js");
const cellConverter = {
  toPm: {
    types: [import_v1.Node_Type.LAYOUT_CELL],
    // @ts-expect-error $TSFixMe
    convert: (node, visit) => {
      return {
        type: import_v1.Node_Type.LAYOUT_CELL,
        attrs: {
          colSpan: node.layoutCellData.colSpan,
          id: node.id
        },
        content: visit(node)
      };
    }
  },
  fromPm: {
    types: [import_v1.Node_Type.LAYOUT_CELL],
    // @ts-expect-error $TSFixMe
    convert: (node, visit) => {
      const { attrs = {} } = node;
      const { id, colSpan } = attrs;
      return {
        type: import_v1.Node_Type.LAYOUT_CELL,
        id,
        nodes: visit(node),
        layoutCellData: {
          colSpan
        }
      };
    }
  }
};
