// @wix/auto_sdk_restaurants_item-variants/build/es/index.mjs
// src/restaurants-menus-v1-item-variant-item-variants.public.ts
import { renameKeysFromRESTResponseToSDKResponse as renameKeysFromRESTResponseToSDKResponse2 } from "@wix/sdk-runtime/rename-all-nested-keys";
import { transformRESTTimestampToSDKTimestamp as transformRESTTimestampToSDKTimestamp2 } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths as transformPaths3 } from "@wix/sdk-runtime/transformations/transform-paths";
import { EventDefinition } from "@wix/sdk-types";

// src/restaurants-menus-v1-item-variant-item-variants.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/restaurants-menus-v1-item-variant-item-variants.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-variant",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-variant",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-variant",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/restaurants-menus-item-variant",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/restaurants-menus-item-variant",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/restaurants-menus-item-variant",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/variants",
        destPath: "/v1/variants"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/variants/create",
        destPath: "/v1/bulk/variants/create"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/variants/update",
        destPath: "/v1/bulk/variants/update"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/variants/delete",
        destPath: "/v1/bulk/variants/delete"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-variant",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/restaurants-menus-item-variant",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/variants",
        destPath: "/v1/variants"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/variants/create",
        destPath: "/v1/bulk/variants/create"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/variants/update",
        destPath: "/v1/bulk/variants/update"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/variants/delete",
        destPath: "/v1/bulk/variants/delete"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/item-variants",
        destPath: ""
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/variants/create",
        destPath: "/v1/bulk/variants/create"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/variants/update",
        destPath: "/v1/bulk/variants/update"
      },
      {
        srcPath: "/restaurants/menus/v1/variants",
        destPath: "/v1/variants"
      },
      {
        srcPath: "/restaurants/menus/v1/bulk/variants/delete",
        destPath: "/v1/bulk/variants/delete"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants-menus-item-variant",
        destPath: ""
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/restaurants-menus-item-variant",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_item-variants";
function createVariant(payload) {
  function __createVariant({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "variant.createdDate" },
          { path: "variant.updatedDate" },
          { path: "variant.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_variant",
      method: "POST",
      methodFqn: "wix.restaurants.menus_item_variant.v1.VariantsService.CreateVariant",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl({
        protoPath: "/v1/variants",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "variant.createdDate" },
            { path: "variant.updatedDate" },
            { path: "variant.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createVariant;
}
function getVariant(payload) {
  function __getVariant({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_variant",
      method: "GET",
      methodFqn: "wix.restaurants.menus_item_variant.v1.VariantsService.GetVariant",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl({
        protoPath: "/v1/variants/{variantId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "variant.createdDate" },
            { path: "variant.updatedDate" },
            { path: "variant.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getVariant;
}
function listVariants(payload) {
  function __listVariants({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_variant",
      method: "GET",
      methodFqn: "wix.restaurants.menus_item_variant.v1.VariantsService.ListVariants",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl({
        protoPath: "/v1/variants",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "variants.createdDate" },
            { path: "variants.updatedDate" },
            { path: "variants.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listVariants;
}
function updateVariant(payload) {
  function __updateVariant({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "variant.createdDate" },
          { path: "variant.updatedDate" },
          { path: "variant.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_variant",
      method: "PATCH",
      methodFqn: "wix.restaurants.menus_item_variant.v1.VariantsService.UpdateVariant",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl({
        protoPath: "/v1/variants/{variant.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "variant.createdDate" },
            { path: "variant.updatedDate" },
            { path: "variant.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateVariant;
}
function bulkCreateVariants(payload) {
  function __bulkCreateVariants({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "variants.createdDate" },
          { path: "variants.updatedDate" },
          { path: "variants.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_variant",
      method: "POST",
      methodFqn: "wix.restaurants.menus_item_variant.v1.VariantsService.BulkCreateVariants",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl({
        protoPath: "/v1/bulk/variants/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.variant.createdDate" },
            { path: "results.variant.updatedDate" },
            { path: "results.variant.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateVariants;
}
function bulkUpdateVariants(payload) {
  function __bulkUpdateVariants({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "variants.fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "variants.variant.createdDate" },
          { path: "variants.variant.updatedDate" },
          { path: "variants.variant.externalReferenceInfo.lastSyncDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_variant",
      method: "POST",
      methodFqn: "wix.restaurants.menus_item_variant.v1.VariantsService.BulkUpdateVariants",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl({
        protoPath: "/v1/bulk/variants/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.variant.createdDate" },
            { path: "results.variant.updatedDate" },
            { path: "results.variant.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateVariants;
}
function deleteVariant(payload) {
  function __deleteVariant({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_variant",
      method: "DELETE",
      methodFqn: "wix.restaurants.menus_item_variant.v1.VariantsService.DeleteVariant",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl({
        protoPath: "/v1/variants/{variantId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteVariant;
}
function queryVariants(payload) {
  function __queryVariants({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_variant",
      method: "POST",
      methodFqn: "wix.restaurants.menus_item_variant.v1.VariantsService.QueryVariants",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl({
        protoPath: "/v1/variants/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "variants.createdDate" },
            { path: "variants.updatedDate" },
            { path: "variants.externalReferenceInfo.lastSyncDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryVariants;
}
function countVariants(payload) {
  function __countVariants({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_variant",
      method: "POST",
      methodFqn: "wix.restaurants.menus_item_variant.v1.VariantsService.CountVariants",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl({
        protoPath: "/v1/variants/count",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __countVariants;
}
function bulkDeleteVariants(payload) {
  function __bulkDeleteVariants({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.menus.v1.item_variant",
      method: "DELETE",
      methodFqn: "wix.restaurants.menus_item_variant.v1.VariantsService.BulkDeleteVariants",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsMenusItemVariantV1VariantsServiceUrl({
        protoPath: "/v1/bulk/variants/delete",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __bulkDeleteVariants;
}

// src/restaurants-menus-v1-item-variant-item-variants.universal.ts
import { transformPaths as transformPaths2 } from "@wix/sdk-runtime/transformations/transform-paths";
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createVariant2(variant) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ variant });
  const reqOpts = createVariant(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.variant;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { variant: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["variant"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getVariant2(variantId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    variantId
  });
  const reqOpts = getVariant(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.variant;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { variantId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["variantId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listVariants2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    variantIds: options?.variantIds,
    paging: options?.paging
  });
  const reqOpts = listVariants(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          variantIds: "$[0].variantIds",
          paging: "$[0].paging"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateVariant2(_id, variant) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    variant: { ...variant, id: _id }
  });
  const reqOpts = updateVariant(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.variant;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { variant: "$[1]" },
        explicitPathsToArguments: { "variant.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "variant"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCreateVariants2(variants, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    variants,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkCreateVariants(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          variants: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["variants", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateVariants2(variants, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    variants,
    returnEntity: options?.returnEntity
  });
  const reqOpts = bulkUpdateVariants(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          variants: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["variants", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteVariant2(variantId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    variantId
  });
  const reqOpts = deleteVariant(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { variantId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["variantId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryVariants2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryVariants(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({ data }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(
        transformPaths2(data, [])
      );
      return {
        items: transformedData?.variants,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function countVariants2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    filter: options?.filter
  });
  const reqOpts = countVariants(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { filter: "$[0].filter" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkDeleteVariants2(ids) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ ids });
  const reqOpts = bulkDeleteVariants(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { ids: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["ids"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/restaurants-menus-v1-item-variant-item-variants.public.ts
function createVariant3(httpClient) {
  return (variant) => createVariant2(
    variant,
    // @ts-ignore
    { httpClient }
  );
}
function getVariant3(httpClient) {
  return (variantId) => getVariant2(
    variantId,
    // @ts-ignore
    { httpClient }
  );
}
function listVariants3(httpClient) {
  return (options) => listVariants2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function updateVariant3(httpClient) {
  return (_id, variant) => updateVariant2(
    _id,
    variant,
    // @ts-ignore
    { httpClient }
  );
}
function bulkCreateVariants3(httpClient) {
  return (variants, options) => bulkCreateVariants2(
    variants,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdateVariants3(httpClient) {
  return (variants, options) => bulkUpdateVariants2(
    variants,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function deleteVariant3(httpClient) {
  return (variantId) => deleteVariant2(
    variantId,
    // @ts-ignore
    { httpClient }
  );
}
function queryVariants3(httpClient) {
  return () => queryVariants2(
    // @ts-ignore
    { httpClient }
  );
}
function countVariants3(httpClient) {
  return (options) => countVariants2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkDeleteVariants3(httpClient) {
  return (ids) => bulkDeleteVariants2(
    ids,
    // @ts-ignore
    { httpClient }
  );
}
var onItemVariantCreated = EventDefinition(
  "wix.restaurants.menus.v1.item_variant_created",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.externalReferenceInfo.lastSyncDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onItemVariantDeleted = EventDefinition(
  "wix.restaurants.menus.v1.item_variant_deleted",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "undefined.createdDate" },
          { path: "undefined.updatedDate" },
          { path: "undefined.externalReferenceInfo.lastSyncDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onItemVariantUpdated = EventDefinition(
  "wix.restaurants.menus.v1.item_variant_updated",
  true,
  (event) => renameKeysFromRESTResponseToSDKResponse2(
    transformPaths3(event, [
      {
        transformFn: transformRESTTimestampToSDKTimestamp2,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.externalReferenceInfo.lastSyncDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();

// src/restaurants-menus-v1-item-variant-item-variants.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
import { createEventModule } from "@wix/sdk-runtime/event-definition-modules";
var createVariant4 = /* @__PURE__ */ createRESTModule(createVariant3);
var getVariant4 = /* @__PURE__ */ createRESTModule(getVariant3);
var listVariants4 = /* @__PURE__ */ createRESTModule(listVariants3);
var updateVariant4 = /* @__PURE__ */ createRESTModule(updateVariant3);
var bulkCreateVariants4 = /* @__PURE__ */ createRESTModule(bulkCreateVariants3);
var bulkUpdateVariants4 = /* @__PURE__ */ createRESTModule(bulkUpdateVariants3);
var deleteVariant4 = /* @__PURE__ */ createRESTModule(deleteVariant3);
var queryVariants4 = /* @__PURE__ */ createRESTModule(queryVariants3);
var countVariants4 = /* @__PURE__ */ createRESTModule(countVariants3);
var bulkDeleteVariants4 = /* @__PURE__ */ createRESTModule(bulkDeleteVariants3);
var onItemVariantCreated2 = createEventModule(
  onItemVariantCreated
);
var onItemVariantDeleted2 = createEventModule(
  onItemVariantDeleted
);
var onItemVariantUpdated2 = createEventModule(
  onItemVariantUpdated
);
export {
  SortOrder,
  WebhookIdentityType,
  bulkCreateVariants4 as bulkCreateVariants,
  bulkDeleteVariants4 as bulkDeleteVariants,
  bulkUpdateVariants4 as bulkUpdateVariants,
  countVariants4 as countVariants,
  createVariant4 as createVariant,
  deleteVariant4 as deleteVariant,
  getVariant4 as getVariant,
  listVariants4 as listVariants,
  onItemVariantCreated2 as onItemVariantCreated,
  onItemVariantDeleted2 as onItemVariantDeleted,
  onItemVariantUpdated2 as onItemVariantUpdated,
  queryVariants4 as queryVariants,
  updateVariant4 as updateVariant
};
