// @wix/ricos/dist/cjs/ricos-content/normalization/block-processors.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var block_processors_exports = {};
__export(block_processors_exports, {
  addInlineStyleRanges: () => addInlineStyleRanges,
  addLinkUnderlineRange: () => addLinkUnderlineRange,
  fixAtomicBlockText: () => fixAtomicBlockText,
  fixTextColorStyle: () => fixTextColorStyle
});
module.exports = __toCommonJS(block_processors_exports);
var import_lodash = __toESM(require("lodash"));
const fixAtomicBlockText = (block) => block.type === "atomic" ? { ...block, text: " " } : block;
const addInlineStyleRanges = (block) => ({
  ...block,
  inlineStyleRanges: block.inlineStyleRanges || []
});
const addLinkUnderlineRange = (block, range) => {
  const inlineStyleRange = {
    offset: range.offset,
    length: range.length,
    style: "UNDERLINE"
  };
  const inlineStyleRanges = import_lodash.default.uniqWith(
    [...block.inlineStyleRanges || [], inlineStyleRange],
    import_lodash.default.isEqual
  );
  return { ...block, inlineStyleRanges };
};
const isHexColor = (str) => /^#[0-9a-f]{3}(?:[0-9a-f]{3})?$/i.test(str);
const wrapTextColorStyle = (style) => isHexColor(style) ? JSON.stringify({ FG: style }) : style;
const fixTextColorStyle = (block) => {
  return {
    ...block,
    inlineStyleRanges: block.inlineStyleRanges.map((range) => ({
      ...range,
      style: wrapTextColorStyle(range.style)
    }))
  };
};
