// @wix/patterns/dist/cjs/state/CustomColumnsHistoryState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.CustomColumnsHistoryState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _mobx = require("mobx");
var _bexCore = require("@wix/bex-core");
function parseSelectedColumnsString(selectedColumnsFromUrl) {
  return selectedColumnsFromUrl.split(',').map(columnExpression => {
    const [id = '', isSelected] = columnExpression.split(' ');
    return {
      id: decodeURIComponent(id),
      isSelected: isSelected !== 'false'
    };
  }).filter(({
    id
  }) => id);
}
class CustomColumnsHistoryState {
  constructor(params) {
    (0, _defineProperty2.default)(this, "customColumns", void 0);
    (0, _defineProperty2.default)(this, "history", void 0);
    (0, _defineProperty2.default)(this, "selectedColumnsSearchParamName", 'selectedColumns');
    (0, _defineProperty2.default)(this, "_selectedColumnsFromUrl", void 0);
    this.customColumns = params.customColumns;
    this.history = params.history;
    this._preInitHistoryPersistence();
  }
  _preInitHistoryPersistence() {
    const {
      history,
      customColumns
    } = this;
    const searchParams = new URLSearchParams(history.location.search);
    this._selectedColumnsFromUrl = searchParams.get(this.selectedColumnsSearchParamName);
    if (this._selectedColumnsFromUrl) {
      this.customColumns.table.selectedColumns.setValue(parseSelectedColumnsString(this._selectedColumnsFromUrl));
      customColumns.table.multi.collections.forEach(({
        query
      }) => {
        query._initializedFromHistory = true;
      });
    }
  }
  init() {
    const disposers = [(0, _mobx.reaction)(() => this.customColumns.table.selectedColumns.value, selectedColumns => {
      (0, _bexCore.replaceSearchParamsWith)(this.history, searchParams => {
        if (selectedColumns.length) {
          searchParams.set(this.selectedColumnsSearchParamName, selectedColumns.map(c => `${encodeURIComponent(c.id)}${!c.isSelected ? ' false' : ''}`).join(','));
        } else {
          searchParams.delete(this.selectedColumnsSearchParamName);
        }
      });
    }, {
      fireImmediately: true,
      delay: 250
    })];
    return () => {
      disposers.forEach(d => d());
    };
  }
}
exports.CustomColumnsHistoryState = CustomColumnsHistoryState;
