// @wix/patterns/dist/cjs/state/DateRangeFilterState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.DateRangeFilterState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _DateRangePickerState = require("./DateRangePickerState");
var _mobx = require("mobx");
class DateRangeFilterState {
  constructor(params) {
    (0, _defineProperty2.default)(this, "dateRangePicker", void 0);
    this.dateRangePicker = new _DateRangePickerState.DateRangePickerState({
      toolbar: params.toolbar,
      translate: params.translate,
      filter: params.filter,
      filterByTime: params.filterByTime,
      mode: params.mode
    });
    (0, _mobx.makeObservable)(this, {
      init: _mobx.action.bound
    });
  }
  init() {
    const {
      dateRangePicker
    } = this;
    const dispose = dateRangePicker.init();
    return () => {
      dispose();
    };
  }
}
exports.DateRangeFilterState = DateRangeFilterState;
