// @wix/patterns/dist/cjs/state/DropdownSelectFilterState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.DropdownSelectFilterState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _bexCore = require("@wix/bex-core");
var _DropdownSelectState = require("./DropdownSelectState");
var _mobx = require("mobx");
class DropdownSelectFilterState {
  constructor(params) {
    (0, _defineProperty2.default)(this, "select", void 0);
    (0, _defineProperty2.default)(this, "dropdownSelect", void 0);
    this.select = new _bexCore.SelectState({
      keyGetter: value => value.key
    });
    this.dropdownSelect = new _DropdownSelectState.DropdownSelectState({
      select: this.select,
      container: params.container,
      toolbar: params.toolbar,
      filter: params.filter
    });
    (0, _mobx.makeObservable)(this, {
      init: _mobx.action.bound
    });
  }
  init() {
    const {
      dropdownSelect
    } = this;
    const dispose = dropdownSelect.init();
    return () => {
      dispose();
    };
  }
}
exports.DropdownSelectFilterState = DropdownSelectFilterState;
