// @wix/patterns/dist/cjs/state/ExtensionMenuItemsState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.ExtensionMenuItemsState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = require("react");
var _mobx = require("mobx");
var _ExtensionsState = require("./ExtensionsState");
class ExtensionMenuItemsState {
  constructor(params) {
    (0, _defineProperty2.default)(this, "_extensionsState", void 0);
    this._extensionsState = new _ExtensionsState.ExtensionsState(params);
    (0, _mobx.makeObservable)(this, {
      menuItems: _mobx.computed
    });
  }
  get menuItems() {
    return this._extensionsState.items.filter(item => {
      var _item$menuItem, _item$menuItem2;
      return Boolean(item.title && ((_item$menuItem = item.menuItem) == null ? void 0 : _item$menuItem.IconComponent) && ((_item$menuItem2 = item.menuItem) == null ? void 0 : _item$menuItem2.structuredAction));
    }).map(extension => {
      const {
        componentId,
        appId,
        menuItem: {
          IconComponent,
          structuredAction
        }
      } = extension;
      return {
        biName: extension.title || `menu-item-extension-${componentId}`,
        // extension.title was requested explicitly by BI, it's not a mistake
        dataHook: `menu-item-extension-${componentId}`,
        prefixIcon: IconComponent && /*#__PURE__*/(0, _react.createElement)(IconComponent),
        text: extension.title,
        onClick: structuredAction,
        appId,
        componentId,
        type: 'TPA'
      };
    });
  }
  get containerProps() {
    return this._extensionsState._containerProps;
  }
  init() {
    return this._extensionsState.init();
  }
}
exports.ExtensionMenuItemsState = ExtensionMenuItemsState;
