// @wix/patterns/dist/cjs/state/FilterValueTotalState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.FilterValueTotalState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _bexCore = require("@wix/bex-core");
var _util = require("@wix/bex-core/util");
var _mapValues = _interopRequireDefault(require("lodash/mapValues"));
var _mobx = require("mobx");
class FilterValueTotalState {
  constructor(params) {
    var _params$collection$re;
    (0, _defineProperty2.default)(this, "collection", void 0);
    (0, _defineProperty2.default)(this, "totalsCollection", void 0);
    (0, _defineProperty2.default)(this, "optimisticActions", void 0);
    (0, _defineProperty2.default)(this, "ignoreSearch", void 0);
    (0, _defineProperty2.default)(this, "tabFilter", void 0);
    (0, _defineProperty2.default)(this, "value", void 0);
    this.collection = params.collection;
    this.optimisticActions = params.optimisticActions;
    this.ignoreSearch = params.ignoreSearch;
    this.value = params.value;
    const {
      errorMonitor,
      errorHandler,
      onlineState,
      history,
      translate,
      queryCache,
      lodash,
      queryName,
      itemKey,
      itemName,
      paginationMode,
      selectionConsistencyMode,
      query: {
        customFilters,
        search
      },
      fetchData: fetchDataCollection
    } = ((_params$collection$re = params.collection.result.fetchTotal) == null ? void 0 : _params$collection$re._totalsCollection) ?? params.collection;
    const fetchData = params.fetchData ?? fetchDataCollection;
    const filters = (0, _mapValues.default)(customFilters, filter => {
      if (filter === params.filter) {
        const tabFilter = filter.clone();
        this.tabFilter = tabFilter;
        if (params.value != null) {
          tabFilter.setValue([params.value]);
        } else {
          tabFilter.reset();
        }
        return tabFilter;
      }
      return filter;
    });
    this.totalsCollection = new _bexCore.CollectionState({
      errorMonitor,
      errorHandler,
      onlineState,
      history,
      translate,
      queryCache,
      lodash,
      queryName,
      itemKey,
      itemName,
      limit: 1,
      fetchData: async query => {
        // const fetchDataPromise = fetchData(query);
        //
        // const collectionUpdateResultPromise = new Promise<void>((resolve) => {
        //   this.collection.emitter.on('refreshUpdateResult', resolve);
        // });
        //
        // await Promise.all([fetchDataPromise, collectionUpdateResultPromise]);

        return fetchData(query);
      },
      paginationMode: paginationMode.id,
      selectionConsistencyMode,
      search: params.ignoreSearch ? undefined : search,
      filters
    });
    (0, _mobx.makeObservable)(this, {
      init: _mobx.action
    });
  }
  refresh(value) {
    const {
      tabFilter
    } = this;
    this.value = value;
    if (tabFilter) {
      if (value != null) {
        tabFilter.refresh([value]);
      } else {
        tabFilter.reset();
      }
    }
  }
  init() {
    const {
      totalsCollection,
      optimisticActions,
      ignoreSearch,
      collection
    } = this;
    const disposers = [totalsCollection.init(), (0, _util.addEventListener)(collection.emitter, 'search', () => {
      if (!ignoreSearch) {
        totalsCollection.clearResultAndMoveToStart();
      }
    }), (0, _util.addEventListener)(collection.emitter, 'clearSearch', () => {
      if (!ignoreSearch) {
        totalsCollection.clearResultAndMoveToStart();
      }
    }), (0, _util.addEventListener)(collection.emitter, 'refreshAllPages', () => {
      totalsCollection.refreshAllPages();
    }), (0, _util.addEventListener)(collection.emitter, 'refreshCurrentPage', () => {
      totalsCollection.refreshCurrentPage();
    }), ...(optimisticActions ? [optimisticActions.initCollection(totalsCollection)] : [])];
    return () => {
      disposers.forEach(d => d());
    };
  }
}
exports.FilterValueTotalState = FilterValueTotalState;
