// @wix/patterns/dist/cjs/state/FoldersAndItemsCollectionsState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.FoldersAndItemsCollectionsState = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _bexCore = require("@wix/bex-core");
var _util = require("@wix/bex-core/util");
var _mobx = require("mobx");
const isActiveFilterAndNotInCollection = ({
  query: {
    filtersArray
  }
}) => f1 => !f1.isEmpty && !filtersArray.some(f2 => f2 === f1);
class FoldersAndItemsCollectionsState {
  constructor(params) {
    (0, _defineProperty2.default)(this, "toolbar", void 0);
    (0, _defineProperty2.default)(this, "items", void 0);
    (0, _defineProperty2.default)(this, "folders", void 0);
    this.toolbar = params.toolbar;
    this.items = params.items;
    this.folders = params.folders;
    this.toolbar.multi.collections.push(this.folders);
    (0, _bexCore.syncWithSharedQuery)(this.folders.query, this.items.query);
    this.items.bulkSelect.useNewSelectAllLogic = true;
    this.folders.bulkSelect.useNewSelectAllLogic = true;
    (0, _mobx.makeObservable)(this, {
      hasItemsOnlyActiveFilters: _mobx.computed,
      hasFoldersOnlyActiveFilters: _mobx.computed,
      showEmptyState: _mobx.computed,
      hasSelectedFolders: _mobx.computed
    });
  }
  get hasFoldersOnlyActiveFilters() {
    const {
      folders,
      items
    } = this;
    return folders.query.filtersArray.some(isActiveFilterAndNotInCollection(items));
  }
  get hasItemsOnlyActiveFilters() {
    const {
      folders,
      items
    } = this;
    return items.query.filtersArray.some(isActiveFilterAndNotInCollection(folders));
  }
  get showEmptyState() {
    const {
      toolbar: {
        multi: {
          showEmptyState
        }
      },
      items,
      folders,
      hasFoldersOnlyActiveFilters,
      hasItemsOnlyActiveFilters
    } = this;
    return showEmptyState || folders.showEmptyState && hasFoldersOnlyActiveFilters || items.showEmptyState && hasItemsOnlyActiveFilters;
  }
  get hasAvailableItems() {
    const {
      toolbar: {
        multi: {
          hasAvailableItems
        }
      },
      items,
      folders
    } = this;
    return hasAvailableItems || folders.hasAvailableItems || items.hasAvailableItems;
  }
  get resultOriginQuerySearch() {
    const {
      toolbar: {
        multi: {
          resultOriginQuerySearch
        }
      }
    } = this;
    return resultOriginQuerySearch;
  }
  get hasNonPersistentActiveFilters() {
    const {
      toolbar: {
        multi: {
          hasNonPersistentActiveFilters
        }
      }
    } = this;
    return hasNonPersistentActiveFilters;
  }
  get query() {
    return this.items.query;
  }
  get hasSelectedFolders() {
    return !!this.folders.bulkSelect.selectedIds.length;
  }
  init(initParams) {
    const {
      items,
      folders,
      toolbar
    } = this;
    const disposers = [toolbar.init(initParams), (0, _util.addEventListener)(items.emitter, 'search', ({
      clearResult
    }) => {
      folders._runSearch(clearResult);
    }), (0, _util.addEventListener)(items.emitter, 'clearSearch', () => {
      folders.clearResultAndMoveToStart();
    }), (0, _util.addEventListener)(items.emitter, 'sort', params => {
      folders._runSort(params);
    }), (0, _util.addEventListener)(items.emitter, 'beforeSortStart', params => {
      folders.emitter.emit('beforeSortStart', params);
    }), (0, _util.addEventListener)(items.emitter, 'sortStart', params => {
      folders.emitter.emit('sortStart', params);
    })];
    return () => {
      disposers.forEach(d => d());
    };
  }
}
exports.FoldersAndItemsCollectionsState = FoldersAndItemsCollectionsState;
