// @wix/auto_sdk_restaurants_order-settings/build/internal/es/index.typings.mjs
// src/restaurants-ordersettings-v3-order-settings-order-settings.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/restaurants-ordersettings-v3-order-settings-order-settings.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressRestaurantsOrdersettingsV3OrderSettingsApiUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/restaurants/v3/order-settings",
        destPath: "/v3/order-settings"
      }
    ],
    _: [
      {
        srcPath: "/_api/order-settings",
        destPath: "/v3/order-settings"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/order-settings",
        destPath: "/v3/order-settings"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/order-settings",
        destPath: "/v3/order-settings"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/restaurants/v3/order-settings",
        destPath: "/v3/order-settings"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_order-settings";
function getOrderSettings(payload) {
  function __getOrderSettings({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.ordersettings.v3.order_settings",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.ordersettings.v3.OrderSettingsApi.GetOrderSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsOrdersettingsV3OrderSettingsApiUrl({
        protoPath: "/v3/order-settings/{orderSettingsId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getOrderSettings;
}
function updateOrderSettings(payload) {
  function __updateOrderSettings({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.ordersettings.v3.order_settings",
      method: "PATCH",
      methodFqn: "com.wixpress.restaurants.ordersettings.v3.OrderSettingsApi.UpdateOrderSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsOrdersettingsV3OrderSettingsApiUrl({
        protoPath: "/v3/order-settings/{orderSettings.id}",
        data: serializedData,
        host
      }),
      data: serializedData
    };
    return metadata;
  }
  return __updateOrderSettings;
}
function listOrderSettings(payload) {
  function __listOrderSettings({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.ordersettings.v3.order_settings",
      method: "GET",
      methodFqn: "com.wixpress.restaurants.ordersettings.v3.OrderSettingsApi.ListOrderSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressRestaurantsOrdersettingsV3OrderSettingsApiUrl({
        protoPath: "/v3/order-settings",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __listOrderSettings;
}

// src/restaurants-ordersettings-v3-order-settings-order-settings.universal.ts
var ApproximateAddresses = /* @__PURE__ */ ((ApproximateAddresses2) => {
  ApproximateAddresses2["EMPTY"] = "EMPTY";
  ApproximateAddresses2["BLOCK"] = "BLOCK";
  ApproximateAddresses2["ALLOW"] = "ALLOW";
  return ApproximateAddresses2;
})(ApproximateAddresses || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["URL"] = "URL";
  Type2["TEXT"] = "TEXT";
  return Type2;
})(Type || {});
var TipsType = /* @__PURE__ */ ((TipsType2) => {
  TipsType2["AMOUNT"] = "AMOUNT";
  TipsType2["PERCENTAGE"] = "PERCENTAGE";
  return TipsType2;
})(TipsType || {});
var DayOfWeek = /* @__PURE__ */ ((DayOfWeek2) => {
  DayOfWeek2["MON"] = "MON";
  DayOfWeek2["TUE"] = "TUE";
  DayOfWeek2["WED"] = "WED";
  DayOfWeek2["THU"] = "THU";
  DayOfWeek2["FRI"] = "FRI";
  DayOfWeek2["SAT"] = "SAT";
  DayOfWeek2["SUN"] = "SUN";
  return DayOfWeek2;
})(DayOfWeek || {});
var ChargeRoundingStrategy = /* @__PURE__ */ ((ChargeRoundingStrategy2) => {
  ChargeRoundingStrategy2["UNDEFINED"] = "UNDEFINED";
  ChargeRoundingStrategy2["HALF_AWAY_FROM_ZERO"] = "HALF_AWAY_FROM_ZERO";
  ChargeRoundingStrategy2["HALF_EVEN"] = "HALF_EVEN";
  return ChargeRoundingStrategy2;
})(ChargeRoundingStrategy || {});
async function getOrderSettings2(orderSettingsId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    orderSettingsId
  });
  const reqOpts = getOrderSettings(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.orderSettings;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { orderSettingsId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["orderSettingsId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateOrderSettings2(_id, orderSettings) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    orderSettings: { ...orderSettings, id: _id }
  });
  const reqOpts = updateOrderSettings(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.orderSettings;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: { orderSettings: "$[1]" },
        explicitPathsToArguments: { "orderSettings.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "orderSettings"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listOrderSettings2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    restaurantLocationIds: options?.restaurantLocationIds
  });
  const reqOpts = listOrderSettings(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          restaurantLocationIds: "$[0].restaurantLocationIds"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
export {
  ApproximateAddresses,
  ChargeRoundingStrategy,
  DayOfWeek,
  TipsType,
  Type,
  getOrderSettings2 as getOrderSettings,
  listOrderSettings2 as listOrderSettings,
  updateOrderSettings2 as updateOrderSettings
};
