// pro-gallery/dist/cjs/components/helpers/mediaScrollHelper/mediaScrollHelperWrapper.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const pro_gallery_lib_1 = require("pro-gallery-lib");
class MediaScrollHelper {
    constructor(config) {
        this.config = config;
        this.scrollHelpers = [];
        this.stop = (idx) => {
            for (const scrollHelper of this.scrollHelpers) {
                if (scrollHelper.findItem(idx)) {
                    scrollHelper.stop(idx);
                }
            }
        };
        this.initializePlayState = () => {
            for (const scrollHelper of this.scrollHelpers) {
                scrollHelper.initializePlayState();
            }
        };
        this.onScroll = ({ top, left }) => {
            if (!this.scrollHelpers.length) {
                this.top = top || this.top;
                this.left = left || this.left;
            }
            for (const scrollHelper of this.scrollHelpers) {
                scrollHelper.onScroll({ top, left });
            }
        };
        this.handleEvent = (event) => {
            for (const scrollHelper of this.scrollHelpers) {
                scrollHelper.handleEvent(event);
            }
        };
        this.updateGalleryStructure = (data) => {
            if (data.isSSR) {
                //TODO - This could be broken in hydrate since we use inPrerenderMode. (diff between hydrate and ssr are breaking)
                return;
            }
            for (const scrollHelper of this.scrollHelpers) {
                scrollHelper.updateGalleryStructure(data);
            }
            if (this.scrollHelperPromise) {
                this.latestGalleryStructure = data;
                return;
            }
            this.latestGalleryStructure = data;
            if ((0, pro_gallery_lib_1.isEditMode)()) {
                return;
            }
            if (!data.galleryStructure.galleryItems.some((item) => this.config.some((c) => c.supportedItemsFilter(item)))) {
                return;
            }
            this.scrollHelperPromise = Promise.resolve().then(() => __importStar(require(/* webpackChunkName: "proGallery_videoScrollHelper" */ './mediaScrollHelper.js'))).then(({ default: VideoScrollHelper }) => {
                for (const config of this.config) {
                    this.scrollHelpers.push(
                    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
                    // @ts-ignore
                    new VideoScrollHelper(config.onSetPlayingIdx, config.supportedItemsFilter, config.getPlayTrigger));
                }
                this.updateGalleryStructure(this.latestGalleryStructure);
                this.onScroll({ top: this.top, left: this.left });
            })
                .catch((e) => {
                console.error('Failed to load videoScrollHelper. error: ' + e);
            });
        };
        this.left = 0;
        this.top = 0;
    }
}
exports.default = MediaScrollHelper;
