// pro-gallery/dist/cjs/components/helpers/mediaScrollHelper/utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isWithinPaddingVertically = exports.isWithinPaddingHorizontally = void 0;
function getDistanceFromScreen({ offset, scroll, itemStart, itemEnd, screenSize }) {
    const before = scroll - offset - itemEnd;
    const after = offset + itemStart - screenSize - scroll;
    return { before, after };
}
function isWithinPaddingVertically({ target, scrollBase, top, bottom, screenHeight, padding }) {
    const res = getDistanceFromScreen({
        offset: scrollBase || 0,
        scroll: target.scrollY,
        itemStart: top,
        itemEnd: bottom,
        screenSize: screenHeight,
    });
    return res.before < padding && res.after < padding;
}
exports.isWithinPaddingVertically = isWithinPaddingVertically;
function isWithinPaddingHorizontally({ target, left, right, screenWidth, padding }) {
    const res = getDistanceFromScreen({
        offset: 0,
        scroll: target.scrollLeft,
        itemStart: left,
        itemEnd: right,
        screenSize: screenWidth,
    });
    return res.before < padding && res.after < padding;
}
exports.isWithinPaddingHorizontally = isWithinPaddingHorizontally;
