// @wix/auto_sdk_media_folders/build/internal/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  Namespace: () => Namespace,
  RootFolder: () => RootFolder,
  SortOrder: () => SortOrder,
  State: () => State,
  WebhookIdentityType: () => WebhookIdentityType,
  bulkDeleteFolders: () => bulkDeleteFolders4,
  bulkRestoreFoldersFromTrashBin: () => bulkRestoreFoldersFromTrashBin4,
  createFolder: () => createFolder4,
  generateFolderDownloadUrl: () => generateFolderDownloadUrl4,
  getFolder: () => getFolder4,
  listDeletedFolders: () => listDeletedFolders4,
  listFolders: () => listFolders4,
  onFolderCreated: () => onFolderCreated2,
  onFolderDeleted: () => onFolderDeleted2,
  onFolderUpdated: () => onFolderUpdated2,
  searchFolders: () => searchFolders4,
  updateFolder: () => updateFolder4
});
module.exports = __toCommonJS(index_exports);

// src/media-site-media-v1-folder-folders.public.ts
var import_rename_all_nested_keys2 = require("@wix/sdk-runtime/rename-all-nested-keys");
var import_timestamp3 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var import_sdk_types = require("@wix/sdk-types");

// src/media-site-media-v1-folder-folders.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/media-site-media-v1-folder-folders.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixMediaSiteMediaV1FoldersServiceUrl(opts) {
  const domainToMappings = {
    "apps._base_domain_": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: "/api"
      },
      {
        srcPath: "/_api/wix-privatemedia-proxy-server/v1",
        destPath: "/v1"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/site-media/v1",
        destPath: "/v1"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "platform.rise.ai": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_media_folders";
function createFolder(payload) {
  function __createFolder({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.CreateFolder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/folders",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "folder.createdDate" },
            { path: "folder.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createFolder;
}
function getFolder(payload) {
  function __getFolder({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "GET",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.GetFolder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/folders/{folderId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "folder.createdDate" },
            { path: "folder.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getFolder;
}
function listFolders(payload) {
  function __listFolders({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "GET",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.ListFolders",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/folders",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload, true),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "folders.createdDate" },
            { path: "folders.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listFolders;
}
function searchFolders(payload) {
  function __searchFolders({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.SearchFolders",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/folders/search",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "folders.createdDate" },
            { path: "folders.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __searchFolders;
}
function updateFolder(payload) {
  function __updateFolder({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "folder.createdDate" }, { path: "folder.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "PATCH",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.UpdateFolder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/folders/{folder.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "folder.createdDate" },
            { path: "folder.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateFolder;
}
function generateFolderDownloadUrl(payload) {
  function __generateFolderDownloadUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.GenerateFolderDownloadUrl",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/folders/{folderId}/generate-download-url",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __generateFolderDownloadUrl;
}
function bulkDeleteFolders(payload) {
  function __bulkDeleteFolders({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.BulkDeleteFolders",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/bulk/folders/delete",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkDeleteFolders;
}
function bulkRestoreFoldersFromTrashBin(payload) {
  function __bulkRestoreFoldersFromTrashBin({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.BulkRestoreFoldersFromTrashBin",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/bulk/trash-bin/folders/restore",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkRestoreFoldersFromTrashBin;
}
function listDeletedFolders(payload) {
  function __listDeletedFolders({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "GET",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.ListDeletedFolders",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/trash-bin/folders",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload, true),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "folders.createdDate" },
            { path: "folders.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listDeletedFolders;
}

// src/media-site-media-v1-folder-folders.universal.ts
var State = /* @__PURE__ */ ((State2) => {
  State2["OK"] = "OK";
  State2["DELETED"] = "DELETED";
  return State2;
})(State || {});
var Namespace = /* @__PURE__ */ ((Namespace2) => {
  Namespace2["NO_NAMESPACE"] = "NO_NAMESPACE";
  Namespace2["OTHERS"] = "OTHERS";
  Namespace2["WIX_VIDEO"] = "WIX_VIDEO";
  Namespace2["WIX_MUSIC"] = "WIX_MUSIC";
  Namespace2["ALBUMS_AND_ART_STORE"] = "ALBUMS_AND_ART_STORE";
  Namespace2["WIX_ECOM"] = "WIX_ECOM";
  Namespace2["PHOTO_SHARE_APP"] = "PHOTO_SHARE_APP";
  Namespace2["SHARING_APP"] = "SHARING_APP";
  Namespace2["CHAT"] = "CHAT";
  Namespace2["LOGO_BUILDER"] = "LOGO_BUILDER";
  Namespace2["ALBUMS_OLD"] = "ALBUMS_OLD";
  Namespace2["CHAT_MOBILE"] = "CHAT_MOBILE";
  Namespace2["WIX_FORMS"] = "WIX_FORMS";
  Namespace2["WIX_LEGENDS"] = "WIX_LEGENDS";
  Namespace2["PORTFOLIO"] = "PORTFOLIO";
  return Namespace2;
})(Namespace || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var RootFolder = /* @__PURE__ */ ((RootFolder2) => {
  RootFolder2["MEDIA_ROOT"] = "MEDIA_ROOT";
  RootFolder2["TRASH_ROOT"] = "TRASH_ROOT";
  RootFolder2["VISITOR_UPLOADS_ROOT"] = "VISITOR_UPLOADS_ROOT";
  return RootFolder2;
})(RootFolder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createFolder2(displayName, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    displayName,
    parentFolderId: options?.parentFolderId
  });
  const reqOpts = createFolder(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          displayName: "$[0]",
          parentFolderId: "$[1].parentFolderId"
        },
        singleArgumentUnchanged: false
      },
      ["displayName", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getFolder2(folderId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ folderId });
  const reqOpts = getFolder(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.folder;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { folderId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["folderId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listFolders2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    parentFolderId: options?.parentFolderId,
    sort: options?.sort,
    paging: options?.paging
  });
  const reqOpts = listFolders(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          parentFolderId: "$[0].parentFolderId",
          sort: "$[0].sort",
          paging: "$[0].paging"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function searchFolders2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    rootFolder: options?.rootFolder,
    sort: options?.sort,
    paging: options?.paging,
    search: options?.search
  });
  const reqOpts = searchFolders(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          rootFolder: "$[0].rootFolder",
          sort: "$[0].sort",
          paging: "$[0].paging",
          search: "$[0].search"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateFolder2(_id, folder) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    folder: { ...folder, id: _id }
  });
  const reqOpts = updateFolder(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.folder;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { folder: "$[1]" },
        explicitPathsToArguments: { "folder.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "folder"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function generateFolderDownloadUrl2(folderId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ folderId });
  const reqOpts = generateFolderDownloadUrl(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { folderId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["folderId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkDeleteFolders2(folderIds, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    folderIds,
    permanent: options?.permanent
  });
  const reqOpts = bulkDeleteFolders(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          folderIds: "$[0]",
          permanent: "$[1].permanent"
        },
        singleArgumentUnchanged: false
      },
      ["folderIds", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkRestoreFoldersFromTrashBin2(folderIds) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    folderIds
  });
  const reqOpts = bulkRestoreFoldersFromTrashBin(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { folderIds: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["folderIds"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listDeletedFolders2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    parentFolderId: options?.parentFolderId,
    sort: options?.sort,
    paging: options?.paging
  });
  const reqOpts = listDeletedFolders(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          parentFolderId: "$[0].parentFolderId",
          sort: "$[0].sort",
          paging: "$[0].paging"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/media-site-media-v1-folder-folders.public.ts
function createFolder3(httpClient) {
  return (displayName, options) => createFolder2(
    displayName,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function getFolder3(httpClient) {
  return (folderId) => getFolder2(
    folderId,
    // @ts-ignore
    { httpClient }
  );
}
function listFolders3(httpClient) {
  return (options) => listFolders2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function searchFolders3(httpClient) {
  return (options) => searchFolders2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function updateFolder3(httpClient) {
  return (_id, folder) => updateFolder2(
    _id,
    folder,
    // @ts-ignore
    { httpClient }
  );
}
function generateFolderDownloadUrl3(httpClient) {
  return (folderId) => generateFolderDownloadUrl2(
    folderId,
    // @ts-ignore
    { httpClient }
  );
}
function bulkDeleteFolders3(httpClient) {
  return (folderIds, options) => bulkDeleteFolders2(
    folderIds,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkRestoreFoldersFromTrashBin3(httpClient) {
  return (folderIds) => bulkRestoreFoldersFromTrashBin2(
    folderIds,
    // @ts-ignore
    { httpClient }
  );
}
function listDeletedFolders3(httpClient) {
  return (options) => listDeletedFolders2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
var onFolderCreated = (0, import_sdk_types.EventDefinition)(
  "wix.media.site_media.v1.folder_created",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths2.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onFolderDeleted = (0, import_sdk_types.EventDefinition)(
  "wix.media.site_media.v1.folder_deleted",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths2.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "undefined.createdDate" },
          { path: "undefined.updatedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onFolderUpdated = (0, import_sdk_types.EventDefinition)(
  "wix.media.site_media.v1.folder_updated",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths2.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();

// src/media-site-media-v1-folder-folders.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var import_event_definition_modules = require("@wix/sdk-runtime/event-definition-modules");
var createFolder4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(createFolder3);
var getFolder4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getFolder3);
var listFolders4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(listFolders3);
var searchFolders4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(searchFolders3);
var updateFolder4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(updateFolder3);
var generateFolderDownloadUrl4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(generateFolderDownloadUrl3);
var bulkDeleteFolders4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkDeleteFolders3);
var bulkRestoreFoldersFromTrashBin4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkRestoreFoldersFromTrashBin3);
var listDeletedFolders4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(listDeletedFolders3);
var onFolderCreated2 = (0, import_event_definition_modules.createEventModule)(onFolderCreated);
var onFolderDeleted2 = (0, import_event_definition_modules.createEventModule)(onFolderDeleted);
var onFolderUpdated2 = (0, import_event_definition_modules.createEventModule)(onFolderUpdated);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Namespace,
  RootFolder,
  SortOrder,
  State,
  WebhookIdentityType,
  bulkDeleteFolders,
  bulkRestoreFoldersFromTrashBin,
  createFolder,
  generateFolderDownloadUrl,
  getFolder,
  listDeletedFolders,
  listFolders,
  onFolderCreated,
  onFolderDeleted,
  onFolderUpdated,
  searchFolders,
  updateFolder
});
