// @wix/auto_sdk_media_folders/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  bulkDeleteFolders: () => bulkDeleteFolders2,
  bulkRestoreFoldersFromTrashBin: () => bulkRestoreFoldersFromTrashBin2,
  createFolder: () => createFolder2,
  generateFolderDownloadUrl: () => generateFolderDownloadUrl2,
  getFolder: () => getFolder2,
  listDeletedFolders: () => listDeletedFolders2,
  listFolders: () => listFolders2,
  searchFolders: () => searchFolders2,
  updateFolder: () => updateFolder2
});
module.exports = __toCommonJS(meta_exports);

// src/media-site-media-v1-folder-folders.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixMediaSiteMediaV1FoldersServiceUrl(opts) {
  const domainToMappings = {
    "apps._base_domain_": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: "/api"
      },
      {
        srcPath: "/_api/wix-privatemedia-proxy-server/v1",
        destPath: "/v1"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/site-media/v1",
        destPath: "/v1"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ],
    "platform.rise.ai": [
      {
        srcPath: "/_api/wix-privatemedia-proxy-server",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_media_folders";
function createFolder(payload) {
  function __createFolder({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.CreateFolder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/folders",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "folder.createdDate" },
            { path: "folder.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createFolder;
}
function getFolder(payload) {
  function __getFolder({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "GET",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.GetFolder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/folders/{folderId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "folder.createdDate" },
            { path: "folder.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getFolder;
}
function listFolders(payload) {
  function __listFolders({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "GET",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.ListFolders",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/folders",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload, true),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "folders.createdDate" },
            { path: "folders.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listFolders;
}
function searchFolders(payload) {
  function __searchFolders({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.SearchFolders",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/folders/search",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "folders.createdDate" },
            { path: "folders.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __searchFolders;
}
function updateFolder(payload) {
  function __updateFolder({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "folder.createdDate" }, { path: "folder.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "PATCH",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.UpdateFolder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/folders/{folder.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "folder.createdDate" },
            { path: "folder.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateFolder;
}
function generateFolderDownloadUrl(payload) {
  function __generateFolderDownloadUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.GenerateFolderDownloadUrl",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/folders/{folderId}/generate-download-url",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __generateFolderDownloadUrl;
}
function bulkDeleteFolders(payload) {
  function __bulkDeleteFolders({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.BulkDeleteFolders",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/bulk/folders/delete",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkDeleteFolders;
}
function bulkRestoreFoldersFromTrashBin(payload) {
  function __bulkRestoreFoldersFromTrashBin({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "POST",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.BulkRestoreFoldersFromTrashBin",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/bulk/trash-bin/folders/restore",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkRestoreFoldersFromTrashBin;
}
function listDeletedFolders(payload) {
  function __listDeletedFolders({ host }) {
    const metadata = {
      entityFqdn: "wix.media.site_media.v1.folder",
      method: "GET",
      methodFqn: "com.wix.media.site_media.v1.FoldersService.ListDeletedFolders",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixMediaSiteMediaV1FoldersServiceUrl({
        protoPath: "/v1/trash-bin/folders",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload, true),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "folders.createdDate" },
            { path: "folders.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listDeletedFolders;
}

// src/media-site-media-v1-folder-folders.meta.ts
function createFolder2() {
  const payload = {};
  const getRequestOptions = createFolder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/folders",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getFolder2() {
  const payload = { folderId: ":folderId" };
  const getRequestOptions = getFolder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/folders/{folderId}",
    pathParams: { folderId: "folderId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listFolders2() {
  const payload = {};
  const getRequestOptions = listFolders(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/folders",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function searchFolders2() {
  const payload = {};
  const getRequestOptions = searchFolders(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/folders/search",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateFolder2() {
  const payload = { folder: { id: ":folderId" } };
  const getRequestOptions = updateFolder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/folders/{folder.id}",
    pathParams: { folderId: "folderId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function generateFolderDownloadUrl2() {
  const payload = { folderId: ":folderId" };
  const getRequestOptions = generateFolderDownloadUrl(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/folders/{folderId}/generate-download-url",
    pathParams: { folderId: "folderId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkDeleteFolders2() {
  const payload = {};
  const getRequestOptions = bulkDeleteFolders(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/folders/delete",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkRestoreFoldersFromTrashBin2() {
  const payload = {};
  const getRequestOptions = bulkRestoreFoldersFromTrashBin(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/trash-bin/folders/restore",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listDeletedFolders2() {
  const payload = {};
  const getRequestOptions = listDeletedFolders(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/trash-bin/folders",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  bulkDeleteFolders,
  bulkRestoreFoldersFromTrashBin,
  createFolder,
  generateFolderDownloadUrl,
  getFolder,
  listDeletedFolders,
  listFolders,
  searchFolders,
  updateFolder
});
