// @wix/patterns/dist/cjs/components/ErrorCard/ErrorCard.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.ErrorCard = ErrorCard;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireWildcard(require("react"));
var _designSystem = require("@wix/design-system");
var _jsxFileName = "/home/builduser/work/db7ea24852bc3350/packages/cairo/dist/cjs/components/ErrorCard/ErrorCard.tsx";
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function ErrorCard({
  children,
  title,
  subtitle,
  action,
  ...rest
}) {
  const childrenArray = _react.Children.toArray(children);
  return /*#__PURE__*/_react.default.createElement(_designSystem.Card, (0, _extends2.default)({}, rest, {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 48,
      columnNumber: 5
    }
  }), childrenArray.filter(el => {
    var _el$type;
    return /*#__PURE__*/(0, _react.isValidElement)(el) && ((_el$type = el.type) == null ? void 0 : _el$type.displayName) !== 'Card.Content';
  }), /*#__PURE__*/_react.default.createElement(_designSystem.Card.Content, {
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 54,
      columnNumber: 7
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.EmptyState, {
    dataHook: "error-card-state",
    title: title,
    subtitle: subtitle,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 55,
      columnNumber: 9
    }
  }, /*#__PURE__*/_react.default.createElement(_designSystem.TextButton, {
    dataHook: "error-card-state-action-button",
    prefixIcon: action == null ? void 0 : action.prefixIcon,
    onClick: action == null ? void 0 : action.onClick,
    __self: this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 60,
      columnNumber: 11
    }
  }, action == null ? void 0 : action.text))));
}
