// @wix/patterns/dist/esm/state/CollectionReachBottom.js
import { ReachBottomListener, ScrollableWindow, } from '../reach-bottom-listener';
export class CollectionReachBottom {
    constructor(params) {
        this.triggerListenerIfReached = () => {
            return this.reachBottom.triggerListenerIfReached();
        };
        this.waitForScrollIdle = () => {
            const { reachBottom } = this;
            // if currently scrolling, return a promise for when the scrolling is idle
            if (!reachBottom.isScrolling) {
                return;
            }
            return new Promise((resolve) => {
                reachBottom.events.once('idle', resolve);
            });
        };
        const { reachBottomParams, lodash } = params;
        this.collection = params.collection;
        this.reachBottom = new ReachBottomListener({
            onReachBottom: this.collection.fetchNextPageIfNeeded,
            delay: 200,
            // this constance is a naive heuristic: amount of pixels covered in 250ms (avg server latency) when scrolling 1 pixel per millisecond (avg scrolling speed)
            offsetPX: 250,
            lodash,
            ...reachBottomParams,
            container: reachBottomParams?.container ?? new ScrollableWindow(),
        });
    }
    subscribe() {
        const { reachBottom } = this;
        reachBottom.subscribe();
    }
    unsubscribe() {
        const { reachBottom, collection } = this;
        collection.beforeSetData = undefined;
        reachBottom.unsubscribe();
    }
}
