// @wix/patterns/dist/esm/state/ExtensionMenuItemsState.js
import { createElement } from 'react';
import { computed, makeObservable } from 'mobx';
import { ExtensionsState } from './ExtensionsState';
export class ExtensionMenuItemsState {
    constructor(params) {
        this._extensionsState = new ExtensionsState(params);
        makeObservable(this, {
            menuItems: computed,
        });
    }
    get menuItems() {
        return this._extensionsState.items
            .filter((item) => {
            return Boolean(item.title &&
                item.menuItem?.IconComponent &&
                item.menuItem?.structuredAction);
        })
            .map((extension) => {
            const { componentId, appId, menuItem: { IconComponent, structuredAction }, } = extension;
            return {
                biName: extension.title || `menu-item-extension-${componentId}`,
                dataHook: `menu-item-extension-${componentId}`,
                prefixIcon: IconComponent && createElement(IconComponent),
                text: extension.title,
                onClick: structuredAction,
                appId,
                componentId,
                type: 'TPA',
            };
        });
    }
    get containerProps() {
        return this._extensionsState._containerProps;
    }
    init() {
        return this._extensionsState.init();
    }
}
