// @wix/patterns/dist/esm/state/FoldersAndItemsCollectionsState.js
import { syncWithSharedQuery, } from '@wix/bex-core';
import { addEventListener } from '@wix/bex-core/util';
import { computed, makeObservable } from 'mobx';
const isActiveFilterAndNotInCollection = ({ query: { filtersArray } }) => (f1) => !f1.isEmpty && !filtersArray.some((f2) => f2 === f1);
export class FoldersAndItemsCollectionsState {
    constructor(params) {
        this.toolbar = params.toolbar;
        this.items = params.items;
        this.folders = params.folders;
        this.toolbar.multi.collections.push(this.folders);
        syncWithSharedQuery(this.folders.query, this.items.query);
        this.items.bulkSelect.useNewSelectAllLogic = true;
        this.folders.bulkSelect.useNewSelectAllLogic = true;
        makeObservable(this, {
            hasItemsOnlyActiveFilters: computed,
            hasFoldersOnlyActiveFilters: computed,
            showEmptyState: computed,
            hasSelectedFolders: computed,
        });
    }
    get hasFoldersOnlyActiveFilters() {
        const { folders, items } = this;
        return folders.query.filtersArray.some(isActiveFilterAndNotInCollection(items));
    }
    get hasItemsOnlyActiveFilters() {
        const { folders, items } = this;
        return items.query.filtersArray.some(isActiveFilterAndNotInCollection(folders));
    }
    get showEmptyState() {
        const { toolbar: { multi: { showEmptyState }, }, items, folders, hasFoldersOnlyActiveFilters, hasItemsOnlyActiveFilters, } = this;
        return (showEmptyState ||
            (folders.showEmptyState && hasFoldersOnlyActiveFilters) ||
            (items.showEmptyState && hasItemsOnlyActiveFilters));
    }
    get hasAvailableItems() {
        const { toolbar: { multi: { hasAvailableItems }, }, items, folders, } = this;
        return (hasAvailableItems || folders.hasAvailableItems || items.hasAvailableItems);
    }
    get resultOriginQuerySearch() {
        const { toolbar: { multi: { resultOriginQuerySearch }, }, } = this;
        return resultOriginQuerySearch;
    }
    get hasNonPersistentActiveFilters() {
        const { toolbar: { multi: { hasNonPersistentActiveFilters }, }, } = this;
        return hasNonPersistentActiveFilters;
    }
    get query() {
        return this.items.query;
    }
    get hasSelectedFolders() {
        return !!this.folders.bulkSelect.selectedIds.length;
    }
    init(initParams) {
        const { items, folders, toolbar } = this;
        const disposers = [
            toolbar.init(initParams),
            addEventListener(items.emitter, 'search', ({ clearResult }) => {
                folders._runSearch(clearResult);
            }),
            addEventListener(items.emitter, 'clearSearch', () => {
                folders.clearResultAndMoveToStart();
            }),
            addEventListener(items.emitter, 'sort', (params) => {
                folders._runSort(params);
            }),
            addEventListener(items.emitter, 'beforeSortStart', (params) => {
                folders.emitter.emit('beforeSortStart', params);
            }),
            addEventListener(items.emitter, 'sortStart', (params) => {
                folders.emitter.emit('sortStart', params);
            }),
        ];
        return () => {
            disposers.forEach((d) => d());
        };
    }
}
