// @wix/auto_sdk_restaurants_recipients/build/es/meta.mjs
// src/restaurants-recipient-v1-recipient-recipients.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/rest-notification-recipients",
        destPath: ""
      }
    ],
    "dev._base_domain_": [
      {
        srcPath: "/_api/rest-notification-recipients",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/rest-notification-recipients/v1",
        destPath: "/v1"
      },
      {
        srcPath: "/restaurants/recipient/v1/recipients",
        destPath: "/v1/recipients"
      }
    ],
    _: [
      {
        srcPath: "/_api/rest-notification-recipients",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/rest-notification-recipients",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_restaurants_recipients";
function createRecipient(payload) {
  function __createRecipient({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "recipient.createdDate" },
          { path: "recipient.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.recipient.v1.recipient",
      method: "POST",
      methodFqn: "wix.restaurants.recipient.v1.NotificationRecipientsService.CreateRecipient",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl({
        protoPath: "/v1/recipients",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "recipient.createdDate" },
            { path: "recipient.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createRecipient;
}
function getRecipient(payload) {
  function __getRecipient({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.recipient.v1.recipient",
      method: "GET",
      methodFqn: "wix.restaurants.recipient.v1.NotificationRecipientsService.GetRecipient",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl({
        protoPath: "/v1/recipients/{recipientId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "recipient.createdDate" },
            { path: "recipient.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getRecipient;
}
function updateRecipient(payload) {
  function __updateRecipient({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "recipient.createdDate" },
          { path: "recipient.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.restaurants.recipient.v1.recipient",
      method: "PATCH",
      methodFqn: "wix.restaurants.recipient.v1.NotificationRecipientsService.UpdateRecipient",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl({
        protoPath: "/v1/recipients/{recipient.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "recipient.createdDate" },
            { path: "recipient.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateRecipient;
}
function deleteRecipient(payload) {
  function __deleteRecipient({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.recipient.v1.recipient",
      method: "DELETE",
      methodFqn: "wix.restaurants.recipient.v1.NotificationRecipientsService.DeleteRecipient",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl({
        protoPath: "/v1/recipients/{recipientId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteRecipient;
}
function queryRecipients(payload) {
  function __queryRecipients({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.recipient.v1.recipient",
      method: "GET",
      methodFqn: "wix.restaurants.recipient.v1.NotificationRecipientsService.QueryRecipients",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl({
        protoPath: "/v1/recipients/query",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload, true),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "recipients.createdDate" },
            { path: "recipients.updatedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl(
            { protoPath: "/v1/recipients/query", data: payload, host }
          ),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryRecipients;
}
function bulkUpdateRecipientTags(payload) {
  function __bulkUpdateRecipientTags({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.recipient.v1.recipient",
      method: "POST",
      methodFqn: "wix.restaurants.recipient.v1.NotificationRecipientsService.BulkUpdateRecipientTags",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl({
        protoPath: "/v1/bulk/recipients/update-tags",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateRecipientTags;
}
function bulkUpdateRecipientTagsByFilter(payload) {
  function __bulkUpdateRecipientTagsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.restaurants.recipient.v1.recipient",
      method: "POST",
      methodFqn: "wix.restaurants.recipient.v1.NotificationRecipientsService.BulkUpdateRecipientTagsByFilter",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixRestaurantsRecipientV1NotificationRecipientsServiceUrl({
        protoPath: "/v1/bulk/recipients/update-tags-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateRecipientTagsByFilter;
}

// src/restaurants-recipient-v1-recipient-recipients.types.ts
var Channel = /* @__PURE__ */ ((Channel2) => {
  Channel2["UNKNOWN_CHANNEL"] = "UNKNOWN_CHANNEL";
  Channel2["SMS"] = "SMS";
  Channel2["PHONE"] = "PHONE";
  return Channel2;
})(Channel || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/restaurants-recipient-v1-recipient-recipients.meta.ts
function createRecipient2() {
  const payload = {};
  const getRequestOptions = createRecipient(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/recipients",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getRecipient2() {
  const payload = { recipientId: ":recipientId" };
  const getRequestOptions = getRecipient(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/recipients/{recipientId}",
    pathParams: { recipientId: "recipientId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateRecipient2() {
  const payload = { recipient: { id: ":recipientId" } };
  const getRequestOptions = updateRecipient(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/recipients/{recipient.id}",
    pathParams: { recipientId: "recipientId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteRecipient2() {
  const payload = { recipientId: ":recipientId" };
  const getRequestOptions = deleteRecipient(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/recipients/{recipientId}",
    pathParams: { recipientId: "recipientId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryRecipients2() {
  const payload = {};
  const getRequestOptions = queryRecipients(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/recipients/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateRecipientTags2() {
  const payload = {};
  const getRequestOptions = bulkUpdateRecipientTags(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/recipients/update-tags",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateRecipientTagsByFilter2() {
  const payload = {};
  const getRequestOptions = bulkUpdateRecipientTagsByFilter(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/recipients/update-tags-by-filter",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  Channel as ChannelOriginal,
  SortOrder as SortOrderOriginal,
  WebhookIdentityType as WebhookIdentityTypeOriginal,
  bulkUpdateRecipientTags2 as bulkUpdateRecipientTags,
  bulkUpdateRecipientTagsByFilter2 as bulkUpdateRecipientTagsByFilter,
  createRecipient2 as createRecipient,
  deleteRecipient2 as deleteRecipient,
  getRecipient2 as getRecipient,
  queryRecipients2 as queryRecipients,
  updateRecipient2 as updateRecipient
};
