// @wix/patterns/dist/cjs/components/CollectionViewsDropdown/modals/SaveView/CollectionSaveViewModal.uni.driver.js
"use strict";

exports.__esModule = true;
exports.CollectionSaveViewModalUniDriver = void 0;
var _unidriver = require("@wix/design-system/dist/testkit/unidriver");
var _unidriver2 = require("../../../../unidriver");
const CollectionSaveViewModalUniDriver = (base, body) => {
  const getModalBase = () => body.$('[data-hook="save-view-modal"]');
  const saveViewModalDriver = () => (0, _unidriver.CustomModalLayoutUniDriver)(getModalBase(), body);
  const viewNameFormFiledDriver = () => (0, _unidriver.FormFieldUniDriver)(body.$('[data-hook="save-view-modal-form-field"]'), body);
  const viewNameInputDriver = () => (0, _unidriver.InputUniDriver)(body.$('[data-hook="save-view-modal-input"]'), body);
  const getSaveButton = () => {
    return saveViewModalDriver().getPrimaryButtonDriver();
  };
  const getCancelButton = () => {
    return saveViewModalDriver().getSecondaryButtonDriver();
  };
  const getLearnMoreButton = () => (0, _unidriver.TextButtonUniDriver)(body.$('[data-hook="save-view-modal-learn-more-link"]'), body);
  return {
    ...(0, _unidriver2.baseUniDriverFactory)(base),
    exists: () => saveViewModalDriver().exists(),
    getTitle: () => saveViewModalDriver().getTitleText(),
    getSubTitle: () => saveViewModalDriver().getSubtitleText(),
    getSaveButtonText: async () => {
      const saveButton = await getSaveButton();
      return saveButton.getButtonTextContent();
    },
    isSaveButtonDisabled: async () => {
      const saveButton = await getSaveButton();
      return saveButton.isButtonDisabled();
    },
    clickSaveButton: async () => {
      const saveButton = await getSaveButton();
      await saveButton.click();
    },
    getCancelButtonText: async () => {
      const cancelButton = await getCancelButton();
      return cancelButton.getButtonTextContent();
    },
    clickCancelButton: async () => {
      const cancelButton = await getCancelButton();
      await cancelButton.click();
    },
    clickCloseButton: () => saveViewModalDriver().clickCloseButton(),
    viewNameFormFieldExists: () => viewNameFormFiledDriver().exists(),
    getViewNameInputValue: () => viewNameInputDriver().getValue(),
    getViewNameFormFieldTitle: () => viewNameFormFiledDriver().getLabel(),
    getViewNameInputPlaceholder: () => viewNameInputDriver().getPlaceholder(),
    fillViewNameInput: value => viewNameInputDriver().enterText(value),
    getViewNameInputStatusMessage: () => viewNameInputDriver().getStatusMessage(),
    getViewNameHasErrorStatus: () => viewNameInputDriver().hasStatus('error'),
    getLearnMoreButton
  };
};
exports.CollectionSaveViewModalUniDriver = CollectionSaveViewModalUniDriver;
