// @wix/auto_sdk_members_member-followers/build/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  WebhookIdentityType: () => WebhookIdentityType,
  followMember: () => followMember2,
  listCurrentMemberFollowers: () => listCurrentMemberFollowers,
  listCurrentMemberFollowing: () => listCurrentMemberFollowing,
  listMemberFollowers: () => listMemberFollowers2,
  listMemberFollowing: () => listMemberFollowing2,
  queryCurrentMemberConnections: () => queryCurrentMemberConnections,
  queryMemberConnections: () => queryMemberConnections2,
  unfollowMember: () => unfollowMember2
});
module.exports = __toCommonJS(index_typings_exports);

// src/members-v3-follow-member-followers.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/members-v3-follow-member-followers.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressMembersFollowV3MemberFollowersUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/members/v3/followers",
        destPath: "/v3/followers"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/members/v3/followers",
        destPath: "/v3/followers"
      },
      {
        srcPath: "/members/v3/followers",
        destPath: "/v3/followers"
      }
    ],
    _: [
      {
        srcPath: "/_api/members/v3/followers",
        destPath: "/v3/followers"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/members/v3/followers",
        destPath: "/v3/followers"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/members/v3/followers",
        destPath: "/v3/followers"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/members/v3/followers",
        destPath: "/v3/followers"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/members/v3/followers",
        destPath: "/v3/followers"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/members/v3/followers",
        destPath: "/v3/followers"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_member-followers";
function followMember(payload) {
  function __followMember({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v3.follow",
      method: "POST",
      methodFqn: "com.wixpress.members.follow.v3.MemberFollowers.FollowMember",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembersFollowV3MemberFollowersUrl({
        protoPath: "/v3/followers/{memberId}",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __followMember;
}
function unfollowMember(payload) {
  function __unfollowMember({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v3.follow",
      method: "DELETE",
      methodFqn: "com.wixpress.members.follow.v3.MemberFollowers.UnfollowMember",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembersFollowV3MemberFollowersUrl({
        protoPath: "/v3/followers/{memberId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __unfollowMember;
}
function listMyMemberFollowing(payload) {
  function __listMyMemberFollowing({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v3.follow",
      method: "GET",
      methodFqn: "com.wixpress.members.follow.v3.MemberFollowers.ListMyMemberFollowing",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembersFollowV3MemberFollowersUrl({
        protoPath: "/v3/followers/my/following",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __listMyMemberFollowing;
}
function listMemberFollowing(payload) {
  function __listMemberFollowing({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v3.follow",
      method: "GET",
      methodFqn: "com.wixpress.members.follow.v3.MemberFollowers.ListMemberFollowing",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembersFollowV3MemberFollowersUrl({
        protoPath: "/v3/followers/{memberId}/following",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __listMemberFollowing;
}
function listMyMemberFollowers(payload) {
  function __listMyMemberFollowers({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v3.follow",
      method: "GET",
      methodFqn: "com.wixpress.members.follow.v3.MemberFollowers.ListMyMemberFollowers",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembersFollowV3MemberFollowersUrl({
        protoPath: "/v3/followers/my",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __listMyMemberFollowers;
}
function listMemberFollowers(payload) {
  function __listMemberFollowers({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v3.follow",
      method: "GET",
      methodFqn: "com.wixpress.members.follow.v3.MemberFollowers.ListMemberFollowers",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembersFollowV3MemberFollowersUrl({
        protoPath: "/v3/followers/{memberId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __listMemberFollowers;
}
function queryMyMemberConnections(payload) {
  function __queryMyMemberConnections({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v3.follow",
      method: "POST",
      methodFqn: "com.wixpress.members.follow.v3.MemberFollowers.QueryMyMemberConnections",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembersFollowV3MemberFollowersUrl({
        protoPath: "/v3/followers/my/connections",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __queryMyMemberConnections;
}
function queryMemberConnections(payload) {
  function __queryMemberConnections({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v3.follow",
      method: "POST",
      methodFqn: "com.wixpress.members.follow.v3.MemberFollowers.QueryMemberConnections",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressMembersFollowV3MemberFollowersUrl({
        protoPath: "/v3/followers/{memberId}/connections",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __queryMemberConnections;
}

// src/members-v3-follow-member-followers.universal.ts
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function followMember2(memberId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ memberId });
  const reqOpts = followMember(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { memberId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["memberId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function unfollowMember2(memberId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ memberId });
  const reqOpts = unfollowMember(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { memberId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["memberId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listCurrentMemberFollowing(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    paging: options?.paging
  });
  const reqOpts = listMyMemberFollowing(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return {
      memberIds: (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.memberIds,
      metadata: (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.pagingMetadata
    };
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { paging: "$[0].paging" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listMemberFollowing2(memberId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    memberId,
    paging: options?.paging
  });
  const reqOpts = listMemberFollowing(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return {
      memberIds: (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.memberIds,
      metadata: (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.pagingMetadata
    };
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { memberId: "$[0]", paging: "$[1].paging" },
        singleArgumentUnchanged: false
      },
      ["memberId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listCurrentMemberFollowers(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    paging: options?.paging
  });
  const reqOpts = listMyMemberFollowers(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return {
      memberIds: (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.memberIds,
      metadata: (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.pagingMetadata
    };
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { paging: "$[0].paging" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listMemberFollowers2(memberId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    memberId,
    paging: options?.paging
  });
  const reqOpts = listMemberFollowers(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return {
      memberIds: (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.memberIds,
      metadata: (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.pagingMetadata
    };
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { memberId: "$[0]", paging: "$[1].paging" },
        singleArgumentUnchanged: false
      },
      ["memberId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function queryCurrentMemberConnections(connectedMemberIds) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    connectedMemberIds
  });
  const reqOpts = queryMyMemberConnections(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { connectedMemberIds: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["connectedMemberIds"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function queryMemberConnections2(memberId, connectedMemberIds) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    memberId,
    connectedMemberIds
  });
  const reqOpts = queryMemberConnections(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          memberId: "$[0]",
          connectedMemberIds: "$[1]"
        },
        singleArgumentUnchanged: false
      },
      ["memberId", "connectedMemberIds"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  WebhookIdentityType,
  followMember,
  listCurrentMemberFollowers,
  listCurrentMemberFollowing,
  listMemberFollowers,
  listMemberFollowing,
  queryCurrentMemberConnections,
  queryMemberConnections,
  unfollowMember
});
