// @wix/ricos/dist/es/ricos-common-plugins/layout/cell.testkit.js
import { RicosPluginToolbarNames } from "./../../ricos-types/index.js";
import { dataHook } from "./../../wix-rich-content-common/index.js";
import {
  LAYOUT_CELL_HANDLE,
  LAYOUT_CELL_TOOLBAR_ADD_LEFT_BUTTON,
  LAYOUT_CELL_TOOLBAR_ADD_RIGHT_BUTTON,
  LAYOUT_CELL_TOOLBAR_DELETE_BUTTON,
  LAYOUT_CELL_TOOLBAR_MOVE_LEFT_BUTTON,
  LAYOUT_CELL_TOOLBAR_MOVE_RIGHT_BUTTON
} from "./data-hooks.js";
const createLayoutCellTestkit = (getBase) => {
  return {
    content: {},
    clickCellHandle: async (cellId) => {
      await (await getBase().get(dataHook(`${LAYOUT_CELL_HANDLE}-${cellId}`))).click();
    },
    toolbars: {
      createPluginToolbar: (getPluginToolbarBase) => ({
        clickAddLeft: async () => {
          await (await getPluginToolbarBase(RicosPluginToolbarNames.LAYOUT_CELL).get(
            dataHook(LAYOUT_CELL_TOOLBAR_ADD_LEFT_BUTTON)
          )).click();
        },
        clickAddRight: async () => {
          await (await getPluginToolbarBase(RicosPluginToolbarNames.LAYOUT_CELL).get(
            dataHook(LAYOUT_CELL_TOOLBAR_ADD_RIGHT_BUTTON)
          )).click();
        },
        clickMoveLeft: async () => {
          await (await getPluginToolbarBase(RicosPluginToolbarNames.LAYOUT_CELL).get(
            dataHook(LAYOUT_CELL_TOOLBAR_MOVE_LEFT_BUTTON)
          )).click();
        },
        clickMoveRight: async () => {
          await (await getPluginToolbarBase(RicosPluginToolbarNames.LAYOUT_CELL).get(
            dataHook(LAYOUT_CELL_TOOLBAR_MOVE_RIGHT_BUTTON)
          )).click();
        },
        clickDelete: async () => {
          await (await getPluginToolbarBase(RicosPluginToolbarNames.LAYOUT_CELL).get(
            dataHook(LAYOUT_CELL_TOOLBAR_DELETE_BUTTON)
          )).click();
        },
        exists: async () => {
          try {
            return getPluginToolbarBase(RicosPluginToolbarNames.LAYOUT_CELL).exists();
          } catch (error) {
            return false;
          }
        }
      })
    }
  };
};
export {
  createLayoutCellTestkit
};
