// @wix/headless-bookings/cjs/dist/react/ServiceList.js
import { jsx as _jsx, Fragment as _Fragment } from "react/jsx-runtime";
import React from 'react';
import { useService } from '@wix/services-manager-react';
import { TestIds } from './test-ids.js';
import { AsChildSlot } from '@wix/headless-utils/react';
import { Root as CoreServiceListRoot, Error as CoreServiceListError, } from './core/ServiceList.js';
import { Root as CoreServiceRoot } from './core/Service.js';
import { ServicesListServiceDefinition, } from '../services/services-list-service.js';
/**
 * Root component that provides the ServiceList context for rendering service lists.
 * This component manages the service data and provides it to child components through context.
 */
export const List = React.forwardRef((props, ref) => {
    const { children, services = [], servicesListConfig, className } = props;
    const serviceConfig = servicesListConfig || {
        services: services,
        searchOptions: {
            cursorPaging: { limit: 10 },
        },
        pagingMetadata: {
            count: services?.length || 0,
        },
        aggregations: {
            results: [],
        },
        customizations: [],
    };
    return (_jsx(CoreServiceListRoot, { servicesListConfig: serviceConfig, children: _jsx(ListContent, { children: children, className: className, ref: ref }) }));
});
List.displayName = 'Services.List';
const ListContent = React.forwardRef((props, ref) => {
    const { children, className } = props;
    const servicesListService = useService(ServicesListServiceDefinition);
    const services = servicesListService.services.get();
    const pagingMetadata = servicesListService.pagingMetadata.get();
    const displayedServices = services.length;
    const totalServices = pagingMetadata?.count || services.length;
    const isFiltered = Object.keys(servicesListService.filters.get()).length > 0;
    const attributes = {
        'data-testid': TestIds.servicesList,
        'data-total-services': totalServices,
        'data-displayed-services': displayedServices,
        'data-filtered': isFiltered,
        className,
    };
    return (_jsx("div", { ...attributes, ref: ref, children: typeof children === 'function' ? children({ services }) : children }));
});
export const Options = React.forwardRef((props, ref) => {
    const { children, emptyState, infiniteScroll = true, pageSize = 0, className, } = props;
    const servicesListService = useService(ServicesListServiceDefinition);
    const services = servicesListService.services.get();
    const hasServices = services.length > 0;
    if (!hasServices) {
        return emptyState || null;
    }
    const attributes = {
        'data-testid': TestIds.servicesOptions,
        'data-empty': !hasServices,
        'data-infinite-scroll': infiniteScroll,
        'data-page-size': pageSize,
        className,
    };
    return (_jsx("div", { ...attributes, ref: ref, children: typeof children === 'function' ? children({ services }) : children }));
});
Options.displayName = 'Services.Options';
export const ServiceRepeater = React.forwardRef((props, _ref) => {
    const { children, className } = props;
    const servicesListService = useService(ServicesListServiceDefinition);
    const services = servicesListService.services.get();
    const hasServices = services.length > 0;
    if (!hasServices)
        return null;
    return (_jsx(_Fragment, { children: services.map((service) => (_jsx(CoreServiceRoot, { serviceConfig: { service }, "data-testid": TestIds.serviceRepeater, "data-service-id": service._id, className: className, children: typeof children === 'function' ? children({ service }) : children }, service._id))) }));
});
ServiceRepeater.displayName = 'Services.ServiceRepeater';
export const Raw = React.forwardRef((props, ref) => {
    const { children } = props;
    const servicesListService = useService(ServicesListServiceDefinition);
    const services = servicesListService.services.get();
    const pagingMetadata = servicesListService.pagingMetadata.get();
    const displayedServices = services.length;
    const totalServices = pagingMetadata?.count || services.length;
    const isFiltered = Object.keys(servicesListService.filters.get()).length > 0;
    return (_jsx("div", { ref: ref, children: typeof children === 'function'
            ? children({ totalServices, displayedServices, isFiltered })
            : children }));
});
Raw.displayName = 'Services.Raw';
export const Error = React.forwardRef((props, ref) => {
    const { asChild, children, className, ...otherProps } = props;
    return (_jsx(CoreServiceListError, { children: ({ error }) => {
            if (!error) {
                return null;
            }
            return (_jsx(AsChildSlot, { ref: ref, asChild: asChild, className: className, "data-testid": TestIds.serviceError, "data-error": error, customElement: children, customElementProps: {
                    error,
                }, content: error, ...otherProps, children: _jsx("div", { className: "text-status-error text-sm sm:text-base", children: error }) }));
        } }));
});
Error.displayName = 'Services.Error';
