// @wix/runtime-fetch-adapter/dist/entry/server.js
import { App } from 'astro/app';
import { setGetEnv } from 'astro/env/setup';
import { createGetEnv } from '../utils.ts/env';
export function createExports(manifest, args) {
    const app = new App(manifest);
    return {
        default: {
            fetch: async (request, env = {}) => {
                setGetEnv(createGetEnv(env));
                try {
                    const locals = { ...env, env };
                    return await app.render(request, { locals });
                }
                catch (error) {
                    const errorInfo = {
                        url: request.url,
                        method: request.method,
                        error: error instanceof Error ? error.message : 'Unknown error',
                        stack: error instanceof Error ? error.stack : undefined,
                        timestamp: new Date().toISOString(),
                        adapter: 'wix-runtime-fetch-adapter'
                    };
                    console.error('🚨 Wix Runtime Fetch Adapter - Request failed:', errorInfo);
                    let statusCode = 500;
                    let errorType = 'Internal Server Error';
                    if (error instanceof Error) {
                        if (error.message.includes('Failed to initialize Astro App')) {
                            statusCode = 503;
                            errorType = 'Service Initialization Error';
                        }
                        else if (error.message.includes('Cannot find module')) {
                            statusCode = 502;
                            errorType = 'Build Configuration Error';
                        }
                    }
                    return new Response(JSON.stringify({
                        error: errorType,
                        message: error instanceof Error ? error.message : 'Unknown error occurred',
                        timestamp: new Date().toISOString(),
                        requestUrl: request.url,
                        ...(typeof process !== 'undefined' && process.env?.NODE_ENV !== 'production' && {
                            debug: {
                                stack: error instanceof Error ? error.stack : undefined,
                                adapter: 'wix-runtime-fetch-adapter@1.0.0'
                            }
                        })
                    }), { status: statusCode, headers: { 'Content-Type': 'application/json', 'Cache-Control': 'no-cache, no-store, must-revalidate' } });
                }
            }
        }
    };
}
export default createExports;
