// @wix/ricos/dist/cjs/toolbars-v3/components/Clickoutside/ClickOutside.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var ClickOutside_exports = {};
__export(ClickOutside_exports, {
  ClickOutside: () => ClickOutside
});
module.exports = __toCommonJS(ClickOutside_exports);
var import_react = __toESM(require("react"));
class ClickOutside extends import_react.default.Component {
  constructor() {
    super(...arguments);
    this.clickCaptured = false;
    this.innerClick = () => {
      this.clickCaptured = true;
    };
    this.documentClick = (event) => {
      if (!this.clickCaptured && this.props.onClickOutside) {
        this.props.onClickOutside(event);
      }
      this.clickCaptured = false;
    };
  }
  render() {
    return this.renderComponent();
  }
  renderComponent() {
    if (typeof this.props.children === "function") {
      return this.props.children(this.getProps());
    }
    return import_react.default.createElement(this.props.wrapper || "span", this.getProps(), this.props.children);
  }
  getProps() {
    return {
      onMouseDown: this.innerClick,
      onTouchStart: this.innerClick
    };
  }
  componentDidMount() {
    document.addEventListener("mousedown", this.documentClick);
    document.addEventListener("touchstart", this.documentClick);
  }
  componentWillUnmount() {
    document.removeEventListener("mousedown", this.documentClick);
    document.removeEventListener("touchstart", this.documentClick);
  }
}
