// @wix/auto-patterns/dist/esm/utils/filterCreators.js
import { dateRangeFilter, idNameArrayFilter, numberRangeFilter, DateRangeFilter, RadioGroupFilter, NumberRangeFilter, AutoCompleteFilter, MultiSelectCheckboxFilter, MultiInlineCheckboxFilter, MultiSelectCollectionFilter } from '@wix/patterns';
import { BooleanFilters } from '../dataSourceAdapters/constants';
export const createDateTimeFilter = _ref => {
  let {
    id,
    fieldId,
    commonFilterProps,
    config
  } = _ref;
  return {
    key: id,
    fieldId,
    filter: dateRangeFilter(),
    component: DateRangeFilter,
    props: {
      ...commonFilterProps,
      mode: config == null ? void 0 : config.mode,
      dateRangeOptions: config == null ? void 0 : config.presets,
      filterByTime: config == null ? void 0 : config.includeTime
    }
  };
};
export const createBooleanFilter = _ref2 => {
  let {
    id,
    fieldId,
    commonFilterProps,
    config
  } = _ref2;
  return {
    key: id,
    fieldId,
    filter: idNameArrayFilter({
      matches: (fieldValue, filterValue) => {
        if (fieldValue) {
          return filterValue.some(v => v.id === BooleanFilters.checked);
        }
        return filterValue.some(v => v.id === BooleanFilters.unchecked);
      }
    }),
    component: RadioGroupFilter,
    props: {
      ...commonFilterProps,
      data: [{
        id: 'checked',
        name: (config == null ? void 0 : config.trueLabel) || 'Yes'
      }, {
        id: 'unchecked',
        name: (config == null ? void 0 : config.falseLabel) || 'No'
      }]
    }
  };
};
export const createNumberFilter = _ref3 => {
  let {
    id,
    fieldId,
    commonFilterProps,
    config
  } = _ref3;
  return {
    key: id,
    fieldId,
    filter: numberRangeFilter(),
    component: NumberRangeFilter,
    props: {
      ...commonFilterProps,
      min: config == null ? void 0 : config.min,
      max: config == null ? void 0 : config.max,
      decimal: config == null ? void 0 : config.allowedDecimals
    }
  };
};
export const createEnumFilter = _ref4 => {
  let {
    id,
    fieldId,
    commonFilterProps,
    config
  } = _ref4;
  const {
    options,
    selectionMode,
    optionType
  } = config;
  const data = (options == null ? void 0 : options.map(option => ({
    id: option.value,
    name: option.label
  }))) ?? [];
  const inlineComponentThreshold = 5;
  let component;
  if (selectionMode === 'single') {
    component = data.length > inlineComponentThreshold ? AutoCompleteFilter : RadioGroupFilter;
    if (optionType === 'select') {
      component = AutoCompleteFilter;
    } else if (optionType === 'radio') {
      component = RadioGroupFilter;
    }
  } else {
    component = data.length > inlineComponentThreshold ? MultiSelectCheckboxFilter : MultiInlineCheckboxFilter;
    if (optionType === 'select') {
      component = MultiSelectCollectionFilter;
    } else if (optionType === 'inlineCheckbox') {
      component = MultiInlineCheckboxFilter;
    } else if (optionType === 'checkbox') {
      component = MultiSelectCheckboxFilter;
    }
  }
  return {
    key: id,
    fieldId,
    filter: idNameArrayFilter(),
    component,
    collectionData: data,
    props: {
      ...commonFilterProps,
      data
    }
  };
};
export const createReferenceFilter = _ref5 => {
  let {
    id,
    fieldId,
    commonFilterProps,
    config,
    additionalParams
  } = _ref5;
  const component = (config == null ? void 0 : config.selectionMode) === 'single' ? AutoCompleteFilter : MultiSelectCollectionFilter;
  return {
    key: id,
    fieldId,
    filter: idNameArrayFilter(),
    component,
    collectionId: additionalParams.referencedSchema.id,
    props: commonFilterProps
  };
};
