// @wix/auto_sdk_bookings_categories/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  EventOriginal: () => Event,
  StatusOriginal: () => Status,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  createCategory: () => createCategory,
  deleteCategory: () => deleteCategory,
  listCategories: () => listCategories,
  updateCategory: () => updateCategory
});
module.exports = __toCommonJS(meta_exports);

// src/bookings-v1-category-categories.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressBookingsServicesApiV1CategoriesServiceUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/bookings/v1/categories",
        destPath: "/v1/categories"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/bookings/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/services-server/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/services-server/v1/batch/categories",
        destPath: "/v1/batch/categories"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/bookings/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/bookings/v1/batch/categories",
        destPath: "/v1/batch/categories"
      },
      {
        srcPath: "/_api/services-server/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/services-server/v1/batch/categories",
        destPath: "/v1/batch/categories"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/bookings/v1/categories",
        destPath: "/v1/categories"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/bookings/v1/categories",
        destPath: "/v1/categories"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/bookings/v1/categories",
        destPath: "/v1/categories"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/categories-proxy",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/services-server/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/services-server/v1/batch/categories",
        destPath: "/v1/batch/categories"
      }
    ],
    "bookings._base_domain_": [
      {
        srcPath: "/_api/services-server/v1/categories",
        destPath: "/v1/categories"
      },
      {
        srcPath: "/_api/services-server/v1/batch/categories",
        destPath: "/v1/batch/categories"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_bookings_categories";
function list(payload) {
  function __list({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.v1.category",
      method: "GET",
      methodFqn: "com.wixpress.bookings.services.api.v1.CategoriesService.List",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressBookingsServicesApiV1CategoriesServiceUrl({
        protoPath: "/v1/categories",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __list;
}
function create(payload) {
  function __create({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.v1.category",
      method: "POST",
      methodFqn: "com.wixpress.bookings.services.api.v1.CategoriesService.Create",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressBookingsServicesApiV1CategoriesServiceUrl({
        protoPath: "/v1/categories",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __create;
}
function update(payload) {
  function __update({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.v1.category",
      method: "PUT",
      methodFqn: "com.wixpress.bookings.services.api.v1.CategoriesService.Update",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressBookingsServicesApiV1CategoriesServiceUrl({
        protoPath: "/v1/categories/{category.id}",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __update;
}
function _delete(payload) {
  function ___delete({ host }) {
    const metadata = {
      entityFqdn: "wix.bookings.v1.category",
      method: "DELETE",
      methodFqn: "com.wixpress.bookings.services.api.v1.CategoriesService._delete",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressBookingsServicesApiV1CategoriesServiceUrl({
        protoPath: "/v1/categories/{id}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return ___delete;
}

// src/bookings-v1-category-categories.types.ts
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["CREATED"] = "CREATED";
  Status2["DELETED"] = "DELETED";
  return Status2;
})(Status || {});
var Event = /* @__PURE__ */ ((Event2) => {
  Event2["Updated"] = "Updated";
  Event2["Deleted"] = "Deleted";
  Event2["Created"] = "Created";
  return Event2;
})(Event || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/bookings-v1-category-categories.meta.ts
function listCategories() {
  const payload = {};
  const getRequestOptions = list(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/categories",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createCategory() {
  const payload = {};
  const getRequestOptions = create(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/categories",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateCategory() {
  const payload = { category: { id: ":categoryId" } };
  const getRequestOptions = update(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PUT",
    path: "/v1/categories/{category.id}",
    pathParams: { categoryId: "categoryId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteCategory() {
  const payload = { id: ":id" };
  const getRequestOptions = _delete(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/categories/{id}",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  EventOriginal,
  StatusOriginal,
  WebhookIdentityTypeOriginal,
  createCategory,
  deleteCategory,
  listCategories,
  updateCategory
});
