// @wix/ricos/dist/cjs/ricos-content/RicosContentAPI/collapsible-list-builder-api.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var collapsible_list_builder_api_exports = {};
__export(collapsible_list_builder_api_exports, {
  addCollapsibleList: () => addCollapsibleList,
  makeCollapsibleList: () => makeCollapsibleList
});
module.exports = __toCommonJS(collapsible_list_builder_api_exports);
var A = __toESM(require("fp-ts-esm/Array"));
var import_function = require("fp-ts-esm/function");
var import_v1 = require("./../../ricos-schema/legacy/wix/rich_content/v1/index.js");
var import_builder_utils = require("./builder-utils.js");
var import_node_builder_methods = require("./node-builder-methods.js");
const DEFAULT_COLLAPSIBLE_LIST_DATA = {
  initialExpandedItems: import_v1.CollapsibleListData_InitialExpandedItems.FIRST,
  direction: import_v1.CollapsibleListData_Direction.LTR,
  containerData: import_node_builder_methods.DEFAULT_CONTAINER_DATA,
  expandOnlyOne: false
};
const toCollapsibleItemNode = (generateId) => (item) => (0, import_node_builder_methods.createNode)(
  generateId
)(import_v1.Node_Type.COLLAPSIBLE_ITEM, {}, [
  (0, import_node_builder_methods.createNode)(generateId)(
    import_v1.Node_Type.COLLAPSIBLE_ITEM_TITLE,
    {},
    item.title.nodes
  ),
  (0, import_node_builder_methods.createNode)(generateId)(
    import_v1.Node_Type.COLLAPSIBLE_ITEM_BODY,
    {},
    item.content.nodes
  )
]);
const toCollapsibleListNode = (generateId) => (data = DEFAULT_COLLAPSIBLE_LIST_DATA) => (nodes) => (0, import_node_builder_methods.createNode)(generateId)(
  import_v1.Node_Type.COLLAPSIBLE_LIST,
  data,
  nodes
);
const toContentWithCollapsibleList = (args) => (node) => (0, import_builder_utils.addNode)({ ...args, node });
const makeCollapsibleList = (generateId) => ({
  items,
  data
}) => {
  return (0, import_function.pipe)(
    items,
    A.map(toCollapsibleItemNode(generateId)),
    toCollapsibleListNode(generateId)(data)
  );
};
const addCollapsibleList = (generateId) => ({
  items,
  data,
  index,
  before,
  after,
  content
}) => {
  return (0, import_function.pipe)(
    makeCollapsibleList(generateId)({ items, data }),
    toContentWithCollapsibleList({ index, before, after, content })
  );
};
