// @wix/headless-site/dist/cjs/host-module.js
"use strict";

exports.__esModule = true;
exports.createHostModule = void 0;
var _monitoringVelo = require("@wix/monitoring-velo");
var _headers = require("./headers");
var _authUtils = require("./utils/auth-utils");
const createHostModule = () => {
  return {
    __type: 'host',
    create: () => {
      return {};
    },
    host: ({
      essentials
    } = {}) => {
      return {
        essentials: {
          get passThroughHeaders() {
            return (0, _headers.getPassThroughHeaders)();
          },
          get language() {
            var _window$essentials;
            return (essentials == null ? void 0 : essentials.language) ?? ((_window$essentials = window.essentials) == null ? void 0 : _window$essentials.language) ?? 'en';
          },
          get locale() {
            var _window$essentials2;
            return (essentials == null ? void 0 : essentials.locale) ?? ((_window$essentials2 = window.essentials) == null ? void 0 : _window$essentials2.locale) ?? 'en-US';
          },
          get timezone() {
            var _window$essentials3;
            return (essentials == null ? void 0 : essentials.timezone) ?? ((_window$essentials3 = window.essentials) == null ? void 0 : _window$essentials3.timeZone) ?? 'UTC';
          }
        },
        getMonitoringClient: () => {
          let producer = 'unknown';
          let req;
          if (typeof document !== 'undefined') {
            producer = document.location.href;
            req = new Request(document.location.href);
          }
          return (0, _monitoringVelo.createVeloMonitoringClient)({
            namespace: 'Headless',
            producer,
            getAuthToken: async () => (0, _authUtils.resolveAuthToken)(),
            req
          });
        }
      };
    }
  };
};
exports.createHostModule = createHostModule;
