// @wix/ricos/dist/cjs/plugin-table/tiptap/plugins/addRowColPlugin.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var addRowColPlugin_exports = {};
__export(addRowColPlugin_exports, {
  addRowColPlugin: () => addRowColPlugin
});
module.exports = __toCommonJS(addRowColPlugin_exports);
var import_view = require("@tiptap/pm/view");
var import_prosemirror_utils = require("prosemirror-utils");
var import_decorations = require("../decorations/index.js");
const addRowColPlugin = (Plugin, PluginKey, editor, services) => {
  const key = new PluginKey("table-add-row-col");
  return new Plugin({
    key,
    state: {
      init: () => {
        return {};
      },
      apply(tr, prev, oldState, newState) {
        const newParentTable = (0, import_prosemirror_utils.findTable)(newState.selection);
        const oldParentTable = (0, import_prosemirror_utils.findTable)(oldState.selection);
        if (!newParentTable) {
          return {};
        }
        if (oldParentTable?.start === newParentTable.start) {
          return prev;
        }
        const decorations = import_view.DecorationSet.create(newState.doc, [
          (0, import_decorations.addRowDecoration)(newState, editor, services),
          (0, import_decorations.addColDecoration)(newState, editor, services)
        ]);
        return { decorations };
      }
    },
    props: {
      decorations(state) {
        return key.getState(state).decorations;
      }
    }
  });
};
