// @wix/ricos/dist/cjs/plugin-table/tiptap/plugins/cellStatesPlugin.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var cellStatesPlugin_exports = {};
__export(cellStatesPlugin_exports, {
  cellStatesPlugin: () => cellStatesPlugin
});
module.exports = __toCommonJS(cellStatesPlugin_exports);
var import_state = require("@tiptap/pm/state");
var import_tables = require("@tiptap/pm/tables");
var import_view = require("@tiptap/pm/view");
var import_consts = require("../consts.js");
var import_decorations = require("../decorations/index.js");
var import_is_selection_type = require("../utilities/is-selection-type.js");
const cellStatesPlugin = (editor, services) => {
  const key = new import_state.PluginKey("cell-states");
  return new import_state.Plugin({
    key,
    state: {
      init: () => {
        return {};
      },
      apply(tr, prev, oldState, newState) {
        if (!(0, import_tables.isInTable)(newState)) return {};
        if ((0, import_is_selection_type.isCellSelection)(newState.selection)) return {};
        const { pos: editCell } = tr.getMeta(import_consts.TRANSACTION_META_KEYS.EDITING_CELL) || {};
        const { pos: selectCell } = tr.getMeta(import_consts.TRANSACTION_META_KEYS.SELECTED_CELL) || {};
        if (newState.selection.ranges.length > 1) return {};
        else if (selectCell) {
          return {
            selectCell
          };
        } else if (editCell) {
          return {
            editCell
          };
        }
        return prev;
      }
    },
    props: {
      handleClick: (view) => {
        const $anchor = (0, import_tables.cellAround)(view.state.selection.$anchor);
        if ($anchor !== null) {
          const cellState = key.getState(view.state);
          if (cellState?.editCell?.pos === $anchor.pos) return false;
          editor.commands.setCellSelection($anchor.pos);
          return true;
        }
        return false;
      },
      handleDoubleClick: (view) => {
        const $anchor = (0, import_tables.cellAround)(view.state.selection.$anchor);
        if ($anchor !== null) {
          editor.commands.setEditCell();
          return true;
        }
        return false;
      },
      decorations(state) {
        const tablePluginState = state && key.getState(state);
        if (tablePluginState.editCell) {
          const editDecorations = (0, import_decorations.cellEditingDecorations)(tablePluginState.editCell);
          const pluginAddMenuDecoration = (0, import_decorations.pluginMenuDecoration)(
            tablePluginState.editCell,
            editor,
            services
          );
          return import_view.DecorationSet.create(state.doc, [...editDecorations, pluginAddMenuDecoration]);
        }
      }
    }
  });
};
