// @wix/patterns/dist/cjs/components/CollectionPageNew/CollectionPage.uni.driver.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.default = void 0;
var _unidriver = require("../../unidriver");
var _unidriver2 = require("@wix/design-system/dist/testkit/unidriver");
var _PageSubTitleUni = _interopRequireDefault(require("../PageSubTitle/PageSubTitle.uni.driver"));
var _MoreActionsUni = require("../MoreActions/MoreActions.uni.driver");
var _ActionControlUni = require("../ActionControl/ActionControl.uni.driver");
var _default = (base, body) => {
  return {
    /**
     * @deprecated Use `getPrimaryActions` instead. Works with `PrimaryActions` component.
     * Returns primary action button driver
     */
    getPrimaryAction: dataHook => (0, _unidriver2.ButtonUniDriver)(base.$(`[data-hook="${dataHook}"]`), body),
    /** Returns primary actions drivers */
    getPrimaryActions: (dataHook = 'primary-actions') => (0, _ActionControlUni.ActionControlUniDriver)(base, body).getActionControl(dataHook),
    /** Returns secondary actions drivers */
    getSecondaryActions: (dataHook = 'secondary-actions') => (0, _ActionControlUni.ActionControlUniDriver)(base, body).getActionControl(dataHook),
    /** Returns more actions button driver */
    getMoreActions: (dataHook = 'more-actions') => (0, _MoreActionsUni.MoreActionsUniDriver)(base, body).getMoreActions(dataHook),
    getPageHeader: () => (0, _unidriver2.PageHeaderUniDriver)(base.$(`[data-hook="collection-page-header"]`), body),
    getTotal: async () => Number(await base.$('[data-hook="page-header-total-count"]').text()),
    subTitleDriver: (0, _PageSubTitleUni.default)(base, body),
    ...(0, _unidriver.baseUniDriverFactory)(base)
  };
};
exports.default = _default;
