// @wix/auto_sdk_events_wix-events-v-2/build/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  Alignment: () => Alignment,
  AppType: () => AppType,
  AspectRatio: () => AspectRatio,
  BackgroundType: () => BackgroundType,
  BadgeTypeType: () => BadgeTypeType,
  ButtonDataType: () => ButtonDataType,
  CardStylesType: () => CardStylesType,
  CheckoutType: () => CheckoutType,
  CommonSortOrder: () => CommonSortOrder,
  ConferenceTypeType: () => ConferenceTypeType,
  Crop: () => Crop,
  DecorationType: () => DecorationType,
  Direction: () => Direction,
  DividerDataAlignment: () => DividerDataAlignment,
  FontType: () => FontType,
  GIFType: () => GIFType,
  GuestTypeType: () => GuestTypeType,
  ImagePosition: () => ImagePosition,
  InitialExpandedItems: () => InitialExpandedItems,
  InitialRegistrationTypeType: () => InitialRegistrationTypeType,
  InputControlType: () => InputControlType,
  Layout: () => Layout,
  LayoutType: () => LayoutType,
  LineStyle: () => LineStyle,
  LocationType: () => LocationType,
  MapType: () => MapType,
  NodeType: () => NodeType,
  NullValue: () => NullValue,
  Orientation: () => Orientation,
  Placement: () => Placement,
  PluginContainerDataAlignment: () => PluginContainerDataAlignment,
  PollLayoutDirection: () => PollLayoutDirection,
  PollLayoutType: () => PollLayoutType,
  Position: () => Position,
  RecurrenceStatusStatus: () => RecurrenceStatusStatus,
  RegistrationStatusStatus: () => RegistrationStatusStatus,
  RegistrationTypeType: () => RegistrationTypeType,
  Relation: () => Relation,
  RequestedFields: () => RequestedFields,
  Resizing: () => Resizing,
  ResponseType: () => ResponseType,
  ResponsivenessBehaviour: () => ResponsivenessBehaviour,
  Scaling: () => Scaling,
  SortOrder: () => SortOrder,
  Source: () => Source,
  Status: () => Status,
  StylesPosition: () => StylesPosition,
  SubdivisionType: () => SubdivisionType,
  Target: () => Target,
  TaxType: () => TaxType,
  TextAlignment: () => TextAlignment,
  ThumbnailsAlignment: () => ThumbnailsAlignment,
  Type: () => Type,
  ValueType: () => ValueType,
  VerticalAlignment: () => VerticalAlignment,
  VerticalAlignmentAlignment: () => VerticalAlignmentAlignment,
  ViewMode: () => ViewMode,
  ViewRole: () => ViewRole,
  VoteRole: () => VoteRole,
  WebhookIdentityType: () => WebhookIdentityType,
  Width: () => Width,
  WidthType: () => WidthType,
  bulkCancelEventsByFilter: () => bulkCancelEventsByFilter4,
  bulkDeleteEventsByFilter: () => bulkDeleteEventsByFilter4,
  cancelEvent: () => cancelEvent4,
  cloneEvent: () => cloneEvent4,
  countEventsByStatus: () => countEventsByStatus4,
  createEvent: () => createEvent4,
  deleteEvent: () => deleteEvent4,
  getEvent: () => getEvent4,
  getEventBySlug: () => getEventBySlug4,
  listEventsByCategory: () => listEventsByCategory4,
  onEventCanceled: () => onEventCanceled2,
  onEventCloned: () => onEventCloned2,
  onEventCreated: () => onEventCreated2,
  onEventDeleted: () => onEventDeleted2,
  onEventEnded: () => onEventEnded2,
  onEventPublished: () => onEventPublished2,
  onEventReminder: () => onEventReminder2,
  onEventStarted: () => onEventStarted2,
  onEventUpdated: () => onEventUpdated2,
  publishDraftEvent: () => publishDraftEvent4,
  queryEvents: () => queryEvents4,
  updateEvent: () => updateEvent4
});
module.exports = __toCommonJS(index_exports);

// src/events-v3-event-wix-events-v-2.public.ts
var import_rename_all_nested_keys2 = require("@wix/sdk-runtime/rename-all-nested-keys");
var import_address3 = require("@wix/sdk-runtime/transformations/address");
var import_float3 = require("@wix/sdk-runtime/transformations/float");
var import_image3 = require("@wix/sdk-runtime/transformations/image");
var import_page_url3 = require("@wix/sdk-runtime/transformations/page-url");
var import_timestamp3 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths3 = require("@wix/sdk-runtime/transformations/transform-paths");
var import_sdk_types = require("@wix/sdk-types");

// src/events-v3-event-wix-events-v-2.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/events-v3-event-wix-events-v-2.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEventsEventsV3EventManagementUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/events/v3/events",
        destPath: "/v3/events"
      },
      {
        srcPath: "/events/v3/bulk/events",
        destPath: "/v3/bulk/events"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/events-events-v3-adapter",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/events-events-v3-adapter",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/events-events-v3-adapter",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/events-events-v3-adapter",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/events-events-v3-adapter",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/events-events-v3-adapter",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/events-events-v3-adapter",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/events-events-v3-adapter",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_events_wix-events-v-2";
function createEvent(payload) {
  function __createEvent({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "event.createdDate" },
          { path: "event.updatedDate" },
          { path: "event.publishedDate" },
          { path: "event.dateAndTimeSettings.startDate" },
          { path: "event.dateAndTimeSettings.endDate" },
          {
            path: "event.dateAndTimeSettings.recurringEvents.individualEventDates.startDate"
          },
          {
            path: "event.dateAndTimeSettings.recurringEvents.individualEventDates.endDate"
          },
          { path: "event.mainImage.urlExpirationDate" },
          { path: "event.registration.rsvp.startDate" },
          { path: "event.registration.rsvp.endDate" },
          { path: "event.categories.categories.createdDate" },
          { path: "event.description.metadata.createdTimestamp" },
          { path: "event.description.metadata.updatedTimestamp" },
          { path: "event.badges.badges.functionUpdateDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "event.location.address.geocode.latitude" },
          { path: "event.location.address.geocode.longitude" },
          {
            path: "event.description.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "event.description.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "event.description.nodes.galleryData.items.video.thumbnail.duration"
          },
          { path: "event.description.nodes.galleryData.options.item.ratio" },
          { path: "event.description.nodes.imageData.image.duration" },
          { path: "event.description.nodes.mapData.mapSettings.lat" },
          { path: "event.description.nodes.mapData.mapSettings.lng" },
          { path: "event.description.nodes.pollData.poll.image.duration" },
          {
            path: "event.description.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "event.description.nodes.pollData.design.poll.background.image.duration"
          },
          { path: "event.description.nodes.appEmbedData.image.duration" },
          { path: "event.description.nodes.videoData.video.duration" },
          { path: "event.description.nodes.videoData.thumbnail.duration" },
          { path: "event.description.nodes.audioData.audio.duration" },
          { path: "event.description.nodes.audioData.coverImage.duration" },
          {
            path: "event.description.nodes.layoutData.backgroundImage.media.duration"
          },
          {
            path: "event.description.nodes.layoutData.backdropImage.media.duration"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v3.event",
      method: "POST",
      methodFqn: "wix.events.events.v3.EventManagement.CreateEvent",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsEventsV3EventManagementUrl({
        protoPath: "/v3/events",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "event.createdDate" },
            { path: "event.updatedDate" },
            { path: "event.publishedDate" },
            { path: "event.dateAndTimeSettings.startDate" },
            { path: "event.dateAndTimeSettings.endDate" },
            {
              path: "event.dateAndTimeSettings.recurringEvents.individualEventDates.startDate"
            },
            {
              path: "event.dateAndTimeSettings.recurringEvents.individualEventDates.endDate"
            },
            { path: "event.mainImage.urlExpirationDate" },
            { path: "event.registration.rsvp.startDate" },
            { path: "event.registration.rsvp.endDate" },
            { path: "event.categories.categories.createdDate" },
            { path: "event.description.metadata.createdTimestamp" },
            { path: "event.description.metadata.updatedTimestamp" },
            { path: "event.badges.badges.functionUpdateDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "event.location.address.geocode.latitude" },
            { path: "event.location.address.geocode.longitude" },
            {
              path: "event.description.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "event.description.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "event.description.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "event.description.nodes.galleryData.options.item.ratio"
            },
            { path: "event.description.nodes.imageData.image.duration" },
            { path: "event.description.nodes.mapData.mapSettings.lat" },
            { path: "event.description.nodes.mapData.mapSettings.lng" },
            { path: "event.description.nodes.pollData.poll.image.duration" },
            {
              path: "event.description.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "event.description.nodes.pollData.design.poll.background.image.duration"
            },
            { path: "event.description.nodes.appEmbedData.image.duration" },
            { path: "event.description.nodes.videoData.video.duration" },
            { path: "event.description.nodes.videoData.thumbnail.duration" },
            { path: "event.description.nodes.audioData.audio.duration" },
            { path: "event.description.nodes.audioData.coverImage.duration" },
            {
              path: "event.description.nodes.layoutData.backgroundImage.media.duration"
            },
            {
              path: "event.description.nodes.layoutData.backdropImage.media.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createEvent;
}
function cloneEvent(payload) {
  function __cloneEvent({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldmask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "event.createdDate" },
          { path: "event.updatedDate" },
          { path: "event.publishedDate" },
          { path: "event.dateAndTimeSettings.startDate" },
          { path: "event.dateAndTimeSettings.endDate" },
          {
            path: "event.dateAndTimeSettings.recurringEvents.individualEventDates.startDate"
          },
          {
            path: "event.dateAndTimeSettings.recurringEvents.individualEventDates.endDate"
          },
          { path: "event.mainImage.urlExpirationDate" },
          { path: "event.registration.rsvp.startDate" },
          { path: "event.registration.rsvp.endDate" },
          { path: "event.categories.categories.createdDate" },
          { path: "event.description.metadata.createdTimestamp" },
          { path: "event.description.metadata.updatedTimestamp" },
          { path: "event.badges.badges.functionUpdateDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "event.location.address.geocode.latitude" },
          { path: "event.location.address.geocode.longitude" },
          {
            path: "event.description.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "event.description.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "event.description.nodes.galleryData.items.video.thumbnail.duration"
          },
          { path: "event.description.nodes.galleryData.options.item.ratio" },
          { path: "event.description.nodes.imageData.image.duration" },
          { path: "event.description.nodes.mapData.mapSettings.lat" },
          { path: "event.description.nodes.mapData.mapSettings.lng" },
          { path: "event.description.nodes.pollData.poll.image.duration" },
          {
            path: "event.description.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "event.description.nodes.pollData.design.poll.background.image.duration"
          },
          { path: "event.description.nodes.appEmbedData.image.duration" },
          { path: "event.description.nodes.videoData.video.duration" },
          { path: "event.description.nodes.videoData.thumbnail.duration" },
          { path: "event.description.nodes.audioData.audio.duration" },
          { path: "event.description.nodes.audioData.coverImage.duration" },
          {
            path: "event.description.nodes.layoutData.backgroundImage.media.duration"
          },
          {
            path: "event.description.nodes.layoutData.backdropImage.media.duration"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v3.event",
      method: "POST",
      methodFqn: "wix.events.events.v3.EventManagement.CloneEvent",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsEventsV3EventManagementUrl({
        protoPath: "/v3/events/{eventId}/clone",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "event.createdDate" },
            { path: "event.updatedDate" },
            { path: "event.publishedDate" },
            { path: "event.dateAndTimeSettings.startDate" },
            { path: "event.dateAndTimeSettings.endDate" },
            {
              path: "event.dateAndTimeSettings.recurringEvents.individualEventDates.startDate"
            },
            {
              path: "event.dateAndTimeSettings.recurringEvents.individualEventDates.endDate"
            },
            { path: "event.mainImage.urlExpirationDate" },
            { path: "event.registration.rsvp.startDate" },
            { path: "event.registration.rsvp.endDate" },
            { path: "event.categories.categories.createdDate" },
            { path: "event.description.metadata.createdTimestamp" },
            { path: "event.description.metadata.updatedTimestamp" },
            { path: "event.badges.badges.functionUpdateDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "event.location.address.geocode.latitude" },
            { path: "event.location.address.geocode.longitude" },
            {
              path: "event.description.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "event.description.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "event.description.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "event.description.nodes.galleryData.options.item.ratio"
            },
            { path: "event.description.nodes.imageData.image.duration" },
            { path: "event.description.nodes.mapData.mapSettings.lat" },
            { path: "event.description.nodes.mapData.mapSettings.lng" },
            { path: "event.description.nodes.pollData.poll.image.duration" },
            {
              path: "event.description.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "event.description.nodes.pollData.design.poll.background.image.duration"
            },
            { path: "event.description.nodes.appEmbedData.image.duration" },
            { path: "event.description.nodes.videoData.video.duration" },
            { path: "event.description.nodes.videoData.thumbnail.duration" },
            { path: "event.description.nodes.audioData.audio.duration" },
            { path: "event.description.nodes.audioData.coverImage.duration" },
            {
              path: "event.description.nodes.layoutData.backgroundImage.media.duration"
            },
            {
              path: "event.description.nodes.layoutData.backdropImage.media.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __cloneEvent;
}
function updateEvent(payload) {
  function __updateEvent({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldmask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "event.createdDate" },
          { path: "event.updatedDate" },
          { path: "event.publishedDate" },
          { path: "event.dateAndTimeSettings.startDate" },
          { path: "event.dateAndTimeSettings.endDate" },
          {
            path: "event.dateAndTimeSettings.recurringEvents.individualEventDates.startDate"
          },
          {
            path: "event.dateAndTimeSettings.recurringEvents.individualEventDates.endDate"
          },
          { path: "event.mainImage.urlExpirationDate" },
          { path: "event.registration.rsvp.startDate" },
          { path: "event.registration.rsvp.endDate" },
          { path: "event.categories.categories.createdDate" },
          { path: "event.description.metadata.createdTimestamp" },
          { path: "event.description.metadata.updatedTimestamp" },
          { path: "event.badges.badges.functionUpdateDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "event.location.address.geocode.latitude" },
          { path: "event.location.address.geocode.longitude" },
          {
            path: "event.description.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "event.description.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "event.description.nodes.galleryData.items.video.thumbnail.duration"
          },
          { path: "event.description.nodes.galleryData.options.item.ratio" },
          { path: "event.description.nodes.imageData.image.duration" },
          { path: "event.description.nodes.mapData.mapSettings.lat" },
          { path: "event.description.nodes.mapData.mapSettings.lng" },
          { path: "event.description.nodes.pollData.poll.image.duration" },
          {
            path: "event.description.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "event.description.nodes.pollData.design.poll.background.image.duration"
          },
          { path: "event.description.nodes.appEmbedData.image.duration" },
          { path: "event.description.nodes.videoData.video.duration" },
          { path: "event.description.nodes.videoData.thumbnail.duration" },
          { path: "event.description.nodes.audioData.audio.duration" },
          { path: "event.description.nodes.audioData.coverImage.duration" },
          {
            path: "event.description.nodes.layoutData.backgroundImage.media.duration"
          },
          {
            path: "event.description.nodes.layoutData.backdropImage.media.duration"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.events.v3.event",
      method: "PATCH",
      methodFqn: "wix.events.events.v3.EventManagement.UpdateEvent",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsEventsV3EventManagementUrl({
        protoPath: "/v3/events/{event.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "event.createdDate" },
            { path: "event.updatedDate" },
            { path: "event.publishedDate" },
            { path: "event.dateAndTimeSettings.startDate" },
            { path: "event.dateAndTimeSettings.endDate" },
            {
              path: "event.dateAndTimeSettings.recurringEvents.individualEventDates.startDate"
            },
            {
              path: "event.dateAndTimeSettings.recurringEvents.individualEventDates.endDate"
            },
            { path: "event.mainImage.urlExpirationDate" },
            { path: "event.registration.rsvp.startDate" },
            { path: "event.registration.rsvp.endDate" },
            { path: "event.categories.categories.createdDate" },
            { path: "event.description.metadata.createdTimestamp" },
            { path: "event.description.metadata.updatedTimestamp" },
            { path: "event.badges.badges.functionUpdateDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "event.location.address.geocode.latitude" },
            { path: "event.location.address.geocode.longitude" },
            {
              path: "event.description.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "event.description.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "event.description.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "event.description.nodes.galleryData.options.item.ratio"
            },
            { path: "event.description.nodes.imageData.image.duration" },
            { path: "event.description.nodes.mapData.mapSettings.lat" },
            { path: "event.description.nodes.mapData.mapSettings.lng" },
            { path: "event.description.nodes.pollData.poll.image.duration" },
            {
              path: "event.description.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "event.description.nodes.pollData.design.poll.background.image.duration"
            },
            { path: "event.description.nodes.appEmbedData.image.duration" },
            { path: "event.description.nodes.videoData.video.duration" },
            { path: "event.description.nodes.videoData.thumbnail.duration" },
            { path: "event.description.nodes.audioData.audio.duration" },
            { path: "event.description.nodes.audioData.coverImage.duration" },
            {
              path: "event.description.nodes.layoutData.backgroundImage.media.duration"
            },
            {
              path: "event.description.nodes.layoutData.backdropImage.media.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateEvent;
}
function publishDraftEvent(payload) {
  function __publishDraftEvent({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v3.event",
      method: "POST",
      methodFqn: "wix.events.events.v3.EventManagement.PublishDraftEvent",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsEventsV3EventManagementUrl({
        protoPath: "/v3/events/{eventId}/publish",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "event.createdDate" },
            { path: "event.updatedDate" },
            { path: "event.publishedDate" },
            { path: "event.dateAndTimeSettings.startDate" },
            { path: "event.dateAndTimeSettings.endDate" },
            {
              path: "event.dateAndTimeSettings.recurringEvents.individualEventDates.startDate"
            },
            {
              path: "event.dateAndTimeSettings.recurringEvents.individualEventDates.endDate"
            },
            { path: "event.mainImage.urlExpirationDate" },
            { path: "event.registration.rsvp.startDate" },
            { path: "event.registration.rsvp.endDate" },
            { path: "event.categories.categories.createdDate" },
            { path: "event.description.metadata.createdTimestamp" },
            { path: "event.description.metadata.updatedTimestamp" },
            { path: "event.badges.badges.functionUpdateDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "event.location.address.geocode.latitude" },
            { path: "event.location.address.geocode.longitude" },
            {
              path: "event.description.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "event.description.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "event.description.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "event.description.nodes.galleryData.options.item.ratio"
            },
            { path: "event.description.nodes.imageData.image.duration" },
            { path: "event.description.nodes.mapData.mapSettings.lat" },
            { path: "event.description.nodes.mapData.mapSettings.lng" },
            { path: "event.description.nodes.pollData.poll.image.duration" },
            {
              path: "event.description.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "event.description.nodes.pollData.design.poll.background.image.duration"
            },
            { path: "event.description.nodes.appEmbedData.image.duration" },
            { path: "event.description.nodes.videoData.video.duration" },
            { path: "event.description.nodes.videoData.thumbnail.duration" },
            { path: "event.description.nodes.audioData.audio.duration" },
            { path: "event.description.nodes.audioData.coverImage.duration" },
            {
              path: "event.description.nodes.layoutData.backgroundImage.media.duration"
            },
            {
              path: "event.description.nodes.layoutData.backdropImage.media.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __publishDraftEvent;
}
function cancelEvent(payload) {
  function __cancelEvent({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v3.event",
      method: "POST",
      methodFqn: "wix.events.events.v3.EventManagement.CancelEvent",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsEventsV3EventManagementUrl({
        protoPath: "/v3/events/{eventId}/cancel",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "event.createdDate" },
            { path: "event.updatedDate" },
            { path: "event.publishedDate" },
            { path: "event.dateAndTimeSettings.startDate" },
            { path: "event.dateAndTimeSettings.endDate" },
            {
              path: "event.dateAndTimeSettings.recurringEvents.individualEventDates.startDate"
            },
            {
              path: "event.dateAndTimeSettings.recurringEvents.individualEventDates.endDate"
            },
            { path: "event.mainImage.urlExpirationDate" },
            { path: "event.registration.rsvp.startDate" },
            { path: "event.registration.rsvp.endDate" },
            { path: "event.categories.categories.createdDate" },
            { path: "event.description.metadata.createdTimestamp" },
            { path: "event.description.metadata.updatedTimestamp" },
            { path: "event.badges.badges.functionUpdateDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "event.location.address.geocode.latitude" },
            { path: "event.location.address.geocode.longitude" },
            {
              path: "event.description.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "event.description.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "event.description.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "event.description.nodes.galleryData.options.item.ratio"
            },
            { path: "event.description.nodes.imageData.image.duration" },
            { path: "event.description.nodes.mapData.mapSettings.lat" },
            { path: "event.description.nodes.mapData.mapSettings.lng" },
            { path: "event.description.nodes.pollData.poll.image.duration" },
            {
              path: "event.description.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "event.description.nodes.pollData.design.poll.background.image.duration"
            },
            { path: "event.description.nodes.appEmbedData.image.duration" },
            { path: "event.description.nodes.videoData.video.duration" },
            { path: "event.description.nodes.videoData.thumbnail.duration" },
            { path: "event.description.nodes.audioData.audio.duration" },
            { path: "event.description.nodes.audioData.coverImage.duration" },
            {
              path: "event.description.nodes.layoutData.backgroundImage.media.duration"
            },
            {
              path: "event.description.nodes.layoutData.backdropImage.media.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __cancelEvent;
}
function bulkCancelEventsByFilter(payload) {
  function __bulkCancelEventsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v3.event",
      method: "POST",
      methodFqn: "wix.events.events.v3.EventManagement.BulkCancelEventsByFilter",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsEventsV3EventManagementUrl({
        protoPath: "/v3/bulk/events/cancel-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkCancelEventsByFilter;
}
function deleteEvent(payload) {
  function __deleteEvent({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v3.event",
      method: "DELETE",
      methodFqn: "wix.events.events.v3.EventManagement.DeleteEvent",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsEventsV3EventManagementUrl({
        protoPath: "/v3/events/{eventId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteEvent;
}
function bulkDeleteEventsByFilter(payload) {
  function __bulkDeleteEventsByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v3.event",
      method: "POST",
      methodFqn: "wix.events.events.v3.EventManagement.BulkDeleteEventsByFilter",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsEventsV3EventManagementUrl({
        protoPath: "/v3/bulk/events/delete-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkDeleteEventsByFilter;
}
function queryEvents(payload) {
  function __queryEvents({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v3.event",
      method: "POST",
      methodFqn: "wix.events.events.v3.EventManagement.QueryEvents",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsEventsV3EventManagementUrl({
        protoPath: "/v3/events/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "events.createdDate" },
            { path: "events.updatedDate" },
            { path: "events.publishedDate" },
            { path: "events.dateAndTimeSettings.startDate" },
            { path: "events.dateAndTimeSettings.endDate" },
            {
              path: "events.dateAndTimeSettings.recurringEvents.individualEventDates.startDate"
            },
            {
              path: "events.dateAndTimeSettings.recurringEvents.individualEventDates.endDate"
            },
            { path: "events.mainImage.urlExpirationDate" },
            { path: "events.registration.rsvp.startDate" },
            { path: "events.registration.rsvp.endDate" },
            { path: "events.categories.categories.createdDate" },
            { path: "events.description.metadata.createdTimestamp" },
            { path: "events.description.metadata.updatedTimestamp" },
            { path: "events.badges.badges.functionUpdateDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "events.location.address.geocode.latitude" },
            { path: "events.location.address.geocode.longitude" },
            {
              path: "events.description.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "events.description.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "events.description.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "events.description.nodes.galleryData.options.item.ratio"
            },
            { path: "events.description.nodes.imageData.image.duration" },
            { path: "events.description.nodes.mapData.mapSettings.lat" },
            { path: "events.description.nodes.mapData.mapSettings.lng" },
            { path: "events.description.nodes.pollData.poll.image.duration" },
            {
              path: "events.description.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "events.description.nodes.pollData.design.poll.background.image.duration"
            },
            { path: "events.description.nodes.appEmbedData.image.duration" },
            { path: "events.description.nodes.videoData.video.duration" },
            { path: "events.description.nodes.videoData.thumbnail.duration" },
            { path: "events.description.nodes.audioData.audio.duration" },
            {
              path: "events.description.nodes.audioData.coverImage.duration"
            },
            {
              path: "events.description.nodes.layoutData.backgroundImage.media.duration"
            },
            {
              path: "events.description.nodes.layoutData.backdropImage.media.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryEvents;
}
function countEventsByStatus(payload) {
  function __countEventsByStatus({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v3.event",
      method: "POST",
      methodFqn: "wix.events.events.v3.EventManagement.CountEventsByStatus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsEventsV3EventManagementUrl({
        protoPath: "/v3/events/count-by-status",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __countEventsByStatus;
}
function listEventsByCategory(payload) {
  function __listEventsByCategory({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v3.event",
      method: "GET",
      methodFqn: "wix.events.events.v3.EventManagement.ListEventsByCategory",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsEventsV3EventManagementUrl({
        protoPath: "/v3/events/category/{categoryId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "events.createdDate" },
            { path: "events.updatedDate" },
            { path: "events.publishedDate" },
            { path: "events.dateAndTimeSettings.startDate" },
            { path: "events.dateAndTimeSettings.endDate" },
            {
              path: "events.dateAndTimeSettings.recurringEvents.individualEventDates.startDate"
            },
            {
              path: "events.dateAndTimeSettings.recurringEvents.individualEventDates.endDate"
            },
            { path: "events.mainImage.urlExpirationDate" },
            { path: "events.registration.rsvp.startDate" },
            { path: "events.registration.rsvp.endDate" },
            { path: "events.categories.categories.createdDate" },
            { path: "events.description.metadata.createdTimestamp" },
            { path: "events.description.metadata.updatedTimestamp" },
            { path: "events.badges.badges.functionUpdateDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "events.location.address.geocode.latitude" },
            { path: "events.location.address.geocode.longitude" },
            {
              path: "events.description.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "events.description.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "events.description.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "events.description.nodes.galleryData.options.item.ratio"
            },
            { path: "events.description.nodes.imageData.image.duration" },
            { path: "events.description.nodes.mapData.mapSettings.lat" },
            { path: "events.description.nodes.mapData.mapSettings.lng" },
            { path: "events.description.nodes.pollData.poll.image.duration" },
            {
              path: "events.description.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "events.description.nodes.pollData.design.poll.background.image.duration"
            },
            { path: "events.description.nodes.appEmbedData.image.duration" },
            { path: "events.description.nodes.videoData.video.duration" },
            { path: "events.description.nodes.videoData.thumbnail.duration" },
            { path: "events.description.nodes.audioData.audio.duration" },
            {
              path: "events.description.nodes.audioData.coverImage.duration"
            },
            {
              path: "events.description.nodes.layoutData.backgroundImage.media.duration"
            },
            {
              path: "events.description.nodes.layoutData.backdropImage.media.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listEventsByCategory;
}
function getEvent(payload) {
  function __getEvent({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v3.event",
      method: "GET",
      methodFqn: "wix.events.events.v3.EventManagement.GetEvent",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsEventsV3EventManagementUrl({
        protoPath: "/v3/events/{eventId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "event.createdDate" },
            { path: "event.updatedDate" },
            { path: "event.publishedDate" },
            { path: "event.dateAndTimeSettings.startDate" },
            { path: "event.dateAndTimeSettings.endDate" },
            {
              path: "event.dateAndTimeSettings.recurringEvents.individualEventDates.startDate"
            },
            {
              path: "event.dateAndTimeSettings.recurringEvents.individualEventDates.endDate"
            },
            { path: "event.mainImage.urlExpirationDate" },
            { path: "event.registration.rsvp.startDate" },
            { path: "event.registration.rsvp.endDate" },
            { path: "event.categories.categories.createdDate" },
            { path: "event.description.metadata.createdTimestamp" },
            { path: "event.description.metadata.updatedTimestamp" },
            { path: "event.badges.badges.functionUpdateDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "event.location.address.geocode.latitude" },
            { path: "event.location.address.geocode.longitude" },
            {
              path: "event.description.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "event.description.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "event.description.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "event.description.nodes.galleryData.options.item.ratio"
            },
            { path: "event.description.nodes.imageData.image.duration" },
            { path: "event.description.nodes.mapData.mapSettings.lat" },
            { path: "event.description.nodes.mapData.mapSettings.lng" },
            { path: "event.description.nodes.pollData.poll.image.duration" },
            {
              path: "event.description.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "event.description.nodes.pollData.design.poll.background.image.duration"
            },
            { path: "event.description.nodes.appEmbedData.image.duration" },
            { path: "event.description.nodes.videoData.video.duration" },
            { path: "event.description.nodes.videoData.thumbnail.duration" },
            { path: "event.description.nodes.audioData.audio.duration" },
            { path: "event.description.nodes.audioData.coverImage.duration" },
            {
              path: "event.description.nodes.layoutData.backgroundImage.media.duration"
            },
            {
              path: "event.description.nodes.layoutData.backdropImage.media.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getEvent;
}
function getEventBySlug(payload) {
  function __getEventBySlug({ host }) {
    const metadata = {
      entityFqdn: "wix.events.v3.event",
      method: "GET",
      methodFqn: "wix.events.events.v3.EventManagement.GetEventBySlug",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEventsEventsV3EventManagementUrl({
        protoPath: "/v3/events/slug/{slug}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "event.createdDate" },
            { path: "event.updatedDate" },
            { path: "event.publishedDate" },
            { path: "event.dateAndTimeSettings.startDate" },
            { path: "event.dateAndTimeSettings.endDate" },
            {
              path: "event.dateAndTimeSettings.recurringEvents.individualEventDates.startDate"
            },
            {
              path: "event.dateAndTimeSettings.recurringEvents.individualEventDates.endDate"
            },
            { path: "event.mainImage.urlExpirationDate" },
            { path: "event.registration.rsvp.startDate" },
            { path: "event.registration.rsvp.endDate" },
            { path: "event.categories.categories.createdDate" },
            { path: "event.description.metadata.createdTimestamp" },
            { path: "event.description.metadata.updatedTimestamp" },
            { path: "event.badges.badges.functionUpdateDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "event.location.address.geocode.latitude" },
            { path: "event.location.address.geocode.longitude" },
            {
              path: "event.description.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "event.description.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "event.description.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "event.description.nodes.galleryData.options.item.ratio"
            },
            { path: "event.description.nodes.imageData.image.duration" },
            { path: "event.description.nodes.mapData.mapSettings.lat" },
            { path: "event.description.nodes.mapData.mapSettings.lng" },
            { path: "event.description.nodes.pollData.poll.image.duration" },
            {
              path: "event.description.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "event.description.nodes.pollData.design.poll.background.image.duration"
            },
            { path: "event.description.nodes.appEmbedData.image.duration" },
            { path: "event.description.nodes.videoData.video.duration" },
            { path: "event.description.nodes.videoData.thumbnail.duration" },
            { path: "event.description.nodes.audioData.audio.duration" },
            { path: "event.description.nodes.audioData.coverImage.duration" },
            {
              path: "event.description.nodes.layoutData.backgroundImage.media.duration"
            },
            {
              path: "event.description.nodes.layoutData.backdropImage.media.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getEventBySlug;
}

// src/events-v3-event-wix-events-v-2.universal.ts
var import_address = require("@wix/sdk-runtime/transformations/address");
var import_address2 = require("@wix/sdk-runtime/transformations/address");
var import_image = require("@wix/sdk-runtime/transformations/image");
var import_image2 = require("@wix/sdk-runtime/transformations/image");
var import_page_url = require("@wix/sdk-runtime/transformations/page-url");
var import_page_url2 = require("@wix/sdk-runtime/transformations/page-url");
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var LocationType = /* @__PURE__ */ ((LocationType2) => {
  LocationType2["UNKNOWN_LOCATION"] = "UNKNOWN_LOCATION";
  LocationType2["VENUE"] = "VENUE";
  LocationType2["ONLINE"] = "ONLINE";
  return LocationType2;
})(LocationType || {});
var SubdivisionType = /* @__PURE__ */ ((SubdivisionType2) => {
  SubdivisionType2["UNKNOWN_SUBDIVISION_TYPE"] = "UNKNOWN_SUBDIVISION_TYPE";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_1"] = "ADMINISTRATIVE_AREA_LEVEL_1";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_2"] = "ADMINISTRATIVE_AREA_LEVEL_2";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_3"] = "ADMINISTRATIVE_AREA_LEVEL_3";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_4"] = "ADMINISTRATIVE_AREA_LEVEL_4";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_5"] = "ADMINISTRATIVE_AREA_LEVEL_5";
  SubdivisionType2["COUNTRY"] = "COUNTRY";
  return SubdivisionType2;
})(SubdivisionType || {});
var RecurrenceStatusStatus = /* @__PURE__ */ ((RecurrenceStatusStatus2) => {
  RecurrenceStatusStatus2["UNKNOWN_STATUS"] = "UNKNOWN_STATUS";
  RecurrenceStatusStatus2["ONE_TIME"] = "ONE_TIME";
  RecurrenceStatusStatus2["RECURRING"] = "RECURRING";
  RecurrenceStatusStatus2["RECURRING_UPCOMING"] = "RECURRING_UPCOMING";
  RecurrenceStatusStatus2["RECURRING_RECENTLY_ENDED"] = "RECURRING_RECENTLY_ENDED";
  RecurrenceStatusStatus2["RECURRING_RECENTLY_CANCELED"] = "RECURRING_RECENTLY_CANCELED";
  return RecurrenceStatusStatus2;
})(RecurrenceStatusStatus || {});
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["UNKNOWN_EVENT_STATUS"] = "UNKNOWN_EVENT_STATUS";
  Status2["UPCOMING"] = "UPCOMING";
  Status2["STARTED"] = "STARTED";
  Status2["ENDED"] = "ENDED";
  Status2["CANCELED"] = "CANCELED";
  Status2["DRAFT"] = "DRAFT";
  return Status2;
})(Status || {});
var RegistrationTypeType = /* @__PURE__ */ ((RegistrationTypeType2) => {
  RegistrationTypeType2["UNKNOWN_REGISTRATION_TYPE"] = "UNKNOWN_REGISTRATION_TYPE";
  RegistrationTypeType2["RSVP"] = "RSVP";
  RegistrationTypeType2["TICKETING"] = "TICKETING";
  RegistrationTypeType2["EXTERNAL"] = "EXTERNAL";
  RegistrationTypeType2["NONE"] = "NONE";
  return RegistrationTypeType2;
})(RegistrationTypeType || {});
var RegistrationStatusStatus = /* @__PURE__ */ ((RegistrationStatusStatus2) => {
  RegistrationStatusStatus2["UNKNOWN_REGISTRATION_STATUS"] = "UNKNOWN_REGISTRATION_STATUS";
  RegistrationStatusStatus2["CLOSED_AUTOMATICALLY"] = "CLOSED_AUTOMATICALLY";
  RegistrationStatusStatus2["CLOSED_MANUALLY"] = "CLOSED_MANUALLY";
  RegistrationStatusStatus2["OPEN_RSVP"] = "OPEN_RSVP";
  RegistrationStatusStatus2["OPEN_RSVP_WAITLIST_ONLY"] = "OPEN_RSVP_WAITLIST_ONLY";
  RegistrationStatusStatus2["OPEN_TICKETS"] = "OPEN_TICKETS";
  RegistrationStatusStatus2["OPEN_EXTERNAL"] = "OPEN_EXTERNAL";
  RegistrationStatusStatus2["SCHEDULED_RSVP"] = "SCHEDULED_RSVP";
  return RegistrationStatusStatus2;
})(RegistrationStatusStatus || {});
var ResponseType = /* @__PURE__ */ ((ResponseType2) => {
  ResponseType2["UNKNOWN_RESPONSE_TYPE"] = "UNKNOWN_RESPONSE_TYPE";
  ResponseType2["YES_ONLY"] = "YES_ONLY";
  ResponseType2["YES_AND_NO"] = "YES_AND_NO";
  return ResponseType2;
})(ResponseType || {});
var TaxType = /* @__PURE__ */ ((TaxType2) => {
  TaxType2["UNKNOWN_TAX_TYPE"] = "UNKNOWN_TAX_TYPE";
  TaxType2["INCLUDED_IN_PRICE"] = "INCLUDED_IN_PRICE";
  TaxType2["ADDED_AT_CHECKOUT"] = "ADDED_AT_CHECKOUT";
  return TaxType2;
})(TaxType || {});
var CheckoutType = /* @__PURE__ */ ((CheckoutType2) => {
  CheckoutType2["UNKNOWN_CHECKOUT_TYPE"] = "UNKNOWN_CHECKOUT_TYPE";
  CheckoutType2["EVENTS_APP"] = "EVENTS_APP";
  CheckoutType2["ECOMM_PLATFORM"] = "ECOMM_PLATFORM";
  return CheckoutType2;
})(CheckoutType || {});
var GuestTypeType = /* @__PURE__ */ ((GuestTypeType2) => {
  GuestTypeType2["UNKNOWN_GUEST_TYPE"] = "UNKNOWN_GUEST_TYPE";
  GuestTypeType2["VISITOR_OR_MEMBER"] = "VISITOR_OR_MEMBER";
  GuestTypeType2["MEMBER"] = "MEMBER";
  return GuestTypeType2;
})(GuestTypeType || {});
var InitialRegistrationTypeType = /* @__PURE__ */ ((InitialRegistrationTypeType2) => {
  InitialRegistrationTypeType2["UNKNOWN_INITIAL_REGISTRATION_TYPE"] = "UNKNOWN_INITIAL_REGISTRATION_TYPE";
  InitialRegistrationTypeType2["RSVP"] = "RSVP";
  InitialRegistrationTypeType2["TICKETING"] = "TICKETING";
  return InitialRegistrationTypeType2;
})(InitialRegistrationTypeType || {});
var InputControlType = /* @__PURE__ */ ((InputControlType2) => {
  InputControlType2["INPUT"] = "INPUT";
  InputControlType2["TEXTAREA"] = "TEXTAREA";
  InputControlType2["DROPDOWN"] = "DROPDOWN";
  InputControlType2["RADIO"] = "RADIO";
  InputControlType2["CHECKBOX"] = "CHECKBOX";
  InputControlType2["NAME"] = "NAME";
  InputControlType2["GUEST_CONTROL"] = "GUEST_CONTROL";
  InputControlType2["ADDRESS_SHORT"] = "ADDRESS_SHORT";
  InputControlType2["ADDRESS_FULL"] = "ADDRESS_FULL";
  InputControlType2["DATE"] = "DATE";
  return InputControlType2;
})(InputControlType || {});
var ValueType = /* @__PURE__ */ ((ValueType2) => {
  ValueType2["TEXT"] = "TEXT";
  ValueType2["NUMBER"] = "NUMBER";
  ValueType2["TEXT_ARRAY"] = "TEXT_ARRAY";
  ValueType2["DATE_TIME"] = "DATE_TIME";
  ValueType2["ADDRESS"] = "ADDRESS";
  return ValueType2;
})(ValueType || {});
var ConferenceTypeType = /* @__PURE__ */ ((ConferenceTypeType2) => {
  ConferenceTypeType2["UNKNOWN_CONFERENCE_TYPE"] = "UNKNOWN_CONFERENCE_TYPE";
  ConferenceTypeType2["MEETING"] = "MEETING";
  ConferenceTypeType2["WEBINAR"] = "WEBINAR";
  return ConferenceTypeType2;
})(ConferenceTypeType || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["MANUAL"] = "MANUAL";
  Type2["RECURRING_EVENT"] = "RECURRING_EVENT";
  return Type2;
})(Type || {});
var NodeType = /* @__PURE__ */ ((NodeType2) => {
  NodeType2["PARAGRAPH"] = "PARAGRAPH";
  NodeType2["TEXT"] = "TEXT";
  NodeType2["HEADING"] = "HEADING";
  NodeType2["BULLETED_LIST"] = "BULLETED_LIST";
  NodeType2["ORDERED_LIST"] = "ORDERED_LIST";
  NodeType2["LIST_ITEM"] = "LIST_ITEM";
  NodeType2["BLOCKQUOTE"] = "BLOCKQUOTE";
  NodeType2["CODE_BLOCK"] = "CODE_BLOCK";
  NodeType2["VIDEO"] = "VIDEO";
  NodeType2["DIVIDER"] = "DIVIDER";
  NodeType2["FILE"] = "FILE";
  NodeType2["GALLERY"] = "GALLERY";
  NodeType2["GIF"] = "GIF";
  NodeType2["HTML"] = "HTML";
  NodeType2["IMAGE"] = "IMAGE";
  NodeType2["LINK_PREVIEW"] = "LINK_PREVIEW";
  NodeType2["MAP"] = "MAP";
  NodeType2["POLL"] = "POLL";
  NodeType2["APP_EMBED"] = "APP_EMBED";
  NodeType2["BUTTON"] = "BUTTON";
  NodeType2["COLLAPSIBLE_LIST"] = "COLLAPSIBLE_LIST";
  NodeType2["TABLE"] = "TABLE";
  NodeType2["EMBED"] = "EMBED";
  NodeType2["COLLAPSIBLE_ITEM"] = "COLLAPSIBLE_ITEM";
  NodeType2["COLLAPSIBLE_ITEM_TITLE"] = "COLLAPSIBLE_ITEM_TITLE";
  NodeType2["COLLAPSIBLE_ITEM_BODY"] = "COLLAPSIBLE_ITEM_BODY";
  NodeType2["TABLE_CELL"] = "TABLE_CELL";
  NodeType2["TABLE_ROW"] = "TABLE_ROW";
  NodeType2["EXTERNAL"] = "EXTERNAL";
  NodeType2["AUDIO"] = "AUDIO";
  NodeType2["CAPTION"] = "CAPTION";
  NodeType2["LAYOUT"] = "LAYOUT";
  NodeType2["LAYOUT_CELL"] = "LAYOUT_CELL";
  return NodeType2;
})(NodeType || {});
var WidthType = /* @__PURE__ */ ((WidthType2) => {
  WidthType2["CONTENT"] = "CONTENT";
  WidthType2["SMALL"] = "SMALL";
  WidthType2["ORIGINAL"] = "ORIGINAL";
  WidthType2["FULL_WIDTH"] = "FULL_WIDTH";
  return WidthType2;
})(WidthType || {});
var PluginContainerDataAlignment = /* @__PURE__ */ ((PluginContainerDataAlignment2) => {
  PluginContainerDataAlignment2["CENTER"] = "CENTER";
  PluginContainerDataAlignment2["LEFT"] = "LEFT";
  PluginContainerDataAlignment2["RIGHT"] = "RIGHT";
  return PluginContainerDataAlignment2;
})(PluginContainerDataAlignment || {});
var ButtonDataType = /* @__PURE__ */ ((ButtonDataType2) => {
  ButtonDataType2["LINK"] = "LINK";
  ButtonDataType2["ACTION"] = "ACTION";
  return ButtonDataType2;
})(ButtonDataType || {});
var Target = /* @__PURE__ */ ((Target2) => {
  Target2["SELF"] = "SELF";
  Target2["BLANK"] = "BLANK";
  Target2["PARENT"] = "PARENT";
  Target2["TOP"] = "TOP";
  return Target2;
})(Target || {});
var TextAlignment = /* @__PURE__ */ ((TextAlignment2) => {
  TextAlignment2["AUTO"] = "AUTO";
  TextAlignment2["LEFT"] = "LEFT";
  TextAlignment2["RIGHT"] = "RIGHT";
  TextAlignment2["CENTER"] = "CENTER";
  TextAlignment2["JUSTIFY"] = "JUSTIFY";
  return TextAlignment2;
})(TextAlignment || {});
var LineStyle = /* @__PURE__ */ ((LineStyle2) => {
  LineStyle2["SINGLE"] = "SINGLE";
  LineStyle2["DOUBLE"] = "DOUBLE";
  LineStyle2["DASHED"] = "DASHED";
  LineStyle2["DOTTED"] = "DOTTED";
  return LineStyle2;
})(LineStyle || {});
var Width = /* @__PURE__ */ ((Width2) => {
  Width2["LARGE"] = "LARGE";
  Width2["MEDIUM"] = "MEDIUM";
  Width2["SMALL"] = "SMALL";
  return Width2;
})(Width || {});
var DividerDataAlignment = /* @__PURE__ */ ((DividerDataAlignment2) => {
  DividerDataAlignment2["CENTER"] = "CENTER";
  DividerDataAlignment2["LEFT"] = "LEFT";
  DividerDataAlignment2["RIGHT"] = "RIGHT";
  return DividerDataAlignment2;
})(DividerDataAlignment || {});
var ViewMode = /* @__PURE__ */ ((ViewMode2) => {
  ViewMode2["NONE"] = "NONE";
  ViewMode2["FULL"] = "FULL";
  ViewMode2["MINI"] = "MINI";
  return ViewMode2;
})(ViewMode || {});
var LayoutType = /* @__PURE__ */ ((LayoutType2) => {
  LayoutType2["COLLAGE"] = "COLLAGE";
  LayoutType2["MASONRY"] = "MASONRY";
  LayoutType2["GRID"] = "GRID";
  LayoutType2["THUMBNAIL"] = "THUMBNAIL";
  LayoutType2["SLIDER"] = "SLIDER";
  LayoutType2["SLIDESHOW"] = "SLIDESHOW";
  LayoutType2["PANORAMA"] = "PANORAMA";
  LayoutType2["COLUMN"] = "COLUMN";
  LayoutType2["MAGIC"] = "MAGIC";
  LayoutType2["FULLSIZE"] = "FULLSIZE";
  return LayoutType2;
})(LayoutType || {});
var Orientation = /* @__PURE__ */ ((Orientation2) => {
  Orientation2["ROWS"] = "ROWS";
  Orientation2["COLUMNS"] = "COLUMNS";
  return Orientation2;
})(Orientation || {});
var Crop = /* @__PURE__ */ ((Crop2) => {
  Crop2["FILL"] = "FILL";
  Crop2["FIT"] = "FIT";
  return Crop2;
})(Crop || {});
var ThumbnailsAlignment = /* @__PURE__ */ ((ThumbnailsAlignment2) => {
  ThumbnailsAlignment2["TOP"] = "TOP";
  ThumbnailsAlignment2["RIGHT"] = "RIGHT";
  ThumbnailsAlignment2["BOTTOM"] = "BOTTOM";
  ThumbnailsAlignment2["LEFT"] = "LEFT";
  ThumbnailsAlignment2["NONE"] = "NONE";
  return ThumbnailsAlignment2;
})(ThumbnailsAlignment || {});
var GIFType = /* @__PURE__ */ ((GIFType2) => {
  GIFType2["NORMAL"] = "NORMAL";
  GIFType2["STICKER"] = "STICKER";
  return GIFType2;
})(GIFType || {});
var Source = /* @__PURE__ */ ((Source2) => {
  Source2["HTML"] = "HTML";
  Source2["ADSENSE"] = "ADSENSE";
  Source2["AI"] = "AI";
  return Source2;
})(Source || {});
var StylesPosition = /* @__PURE__ */ ((StylesPosition2) => {
  StylesPosition2["START"] = "START";
  StylesPosition2["END"] = "END";
  StylesPosition2["TOP"] = "TOP";
  StylesPosition2["HIDDEN"] = "HIDDEN";
  return StylesPosition2;
})(StylesPosition || {});
var MapType = /* @__PURE__ */ ((MapType2) => {
  MapType2["ROADMAP"] = "ROADMAP";
  MapType2["SATELITE"] = "SATELITE";
  MapType2["HYBRID"] = "HYBRID";
  MapType2["TERRAIN"] = "TERRAIN";
  return MapType2;
})(MapType || {});
var ViewRole = /* @__PURE__ */ ((ViewRole2) => {
  ViewRole2["CREATOR"] = "CREATOR";
  ViewRole2["VOTERS"] = "VOTERS";
  ViewRole2["EVERYONE"] = "EVERYONE";
  return ViewRole2;
})(ViewRole || {});
var VoteRole = /* @__PURE__ */ ((VoteRole2) => {
  VoteRole2["SITE_MEMBERS"] = "SITE_MEMBERS";
  VoteRole2["ALL"] = "ALL";
  return VoteRole2;
})(VoteRole || {});
var PollLayoutType = /* @__PURE__ */ ((PollLayoutType2) => {
  PollLayoutType2["LIST"] = "LIST";
  PollLayoutType2["GRID"] = "GRID";
  return PollLayoutType2;
})(PollLayoutType || {});
var PollLayoutDirection = /* @__PURE__ */ ((PollLayoutDirection2) => {
  PollLayoutDirection2["LTR"] = "LTR";
  PollLayoutDirection2["RTL"] = "RTL";
  return PollLayoutDirection2;
})(PollLayoutDirection || {});
var BackgroundType = /* @__PURE__ */ ((BackgroundType2) => {
  BackgroundType2["COLOR"] = "COLOR";
  BackgroundType2["IMAGE"] = "IMAGE";
  BackgroundType2["GRADIENT"] = "GRADIENT";
  return BackgroundType2;
})(BackgroundType || {});
var DecorationType = /* @__PURE__ */ ((DecorationType2) => {
  DecorationType2["BOLD"] = "BOLD";
  DecorationType2["ITALIC"] = "ITALIC";
  DecorationType2["UNDERLINE"] = "UNDERLINE";
  DecorationType2["SPOILER"] = "SPOILER";
  DecorationType2["ANCHOR"] = "ANCHOR";
  DecorationType2["MENTION"] = "MENTION";
  DecorationType2["LINK"] = "LINK";
  DecorationType2["COLOR"] = "COLOR";
  DecorationType2["FONT_SIZE"] = "FONT_SIZE";
  DecorationType2["EXTERNAL"] = "EXTERNAL";
  DecorationType2["STRIKETHROUGH"] = "STRIKETHROUGH";
  DecorationType2["SUPERSCRIPT"] = "SUPERSCRIPT";
  DecorationType2["SUBSCRIPT"] = "SUBSCRIPT";
  return DecorationType2;
})(DecorationType || {});
var FontType = /* @__PURE__ */ ((FontType2) => {
  FontType2["PX"] = "PX";
  FontType2["EM"] = "EM";
  return FontType2;
})(FontType || {});
var Position = /* @__PURE__ */ ((Position2) => {
  Position2["START"] = "START";
  Position2["END"] = "END";
  Position2["TOP"] = "TOP";
  return Position2;
})(Position || {});
var AspectRatio = /* @__PURE__ */ ((AspectRatio2) => {
  AspectRatio2["SQUARE"] = "SQUARE";
  AspectRatio2["RECTANGLE"] = "RECTANGLE";
  return AspectRatio2;
})(AspectRatio || {});
var Resizing = /* @__PURE__ */ ((Resizing2) => {
  Resizing2["FILL"] = "FILL";
  Resizing2["FIT"] = "FIT";
  return Resizing2;
})(Resizing || {});
var Placement = /* @__PURE__ */ ((Placement2) => {
  Placement2["IMAGE"] = "IMAGE";
  Placement2["PRODUCT_INFO"] = "PRODUCT_INFO";
  return Placement2;
})(Placement || {});
var CardStylesType = /* @__PURE__ */ ((CardStylesType2) => {
  CardStylesType2["CONTAINED"] = "CONTAINED";
  CardStylesType2["FRAMELESS"] = "FRAMELESS";
  return CardStylesType2;
})(CardStylesType || {});
var Alignment = /* @__PURE__ */ ((Alignment2) => {
  Alignment2["START"] = "START";
  Alignment2["CENTER"] = "CENTER";
  Alignment2["END"] = "END";
  return Alignment2;
})(Alignment || {});
var Layout = /* @__PURE__ */ ((Layout2) => {
  Layout2["STACKED"] = "STACKED";
  Layout2["SIDE_BY_SIDE"] = "SIDE_BY_SIDE";
  return Layout2;
})(Layout || {});
var AppType = /* @__PURE__ */ ((AppType2) => {
  AppType2["PRODUCT"] = "PRODUCT";
  AppType2["EVENT"] = "EVENT";
  AppType2["BOOKING"] = "BOOKING";
  return AppType2;
})(AppType || {});
var InitialExpandedItems = /* @__PURE__ */ ((InitialExpandedItems2) => {
  InitialExpandedItems2["FIRST"] = "FIRST";
  InitialExpandedItems2["ALL"] = "ALL";
  InitialExpandedItems2["NONE"] = "NONE";
  return InitialExpandedItems2;
})(InitialExpandedItems || {});
var Direction = /* @__PURE__ */ ((Direction2) => {
  Direction2["LTR"] = "LTR";
  Direction2["RTL"] = "RTL";
  return Direction2;
})(Direction || {});
var VerticalAlignment = /* @__PURE__ */ ((VerticalAlignment2) => {
  VerticalAlignment2["TOP"] = "TOP";
  VerticalAlignment2["MIDDLE"] = "MIDDLE";
  VerticalAlignment2["BOTTOM"] = "BOTTOM";
  return VerticalAlignment2;
})(VerticalAlignment || {});
var NullValue = /* @__PURE__ */ ((NullValue2) => {
  NullValue2["NULL_VALUE"] = "NULL_VALUE";
  return NullValue2;
})(NullValue || {});
var Scaling = /* @__PURE__ */ ((Scaling2) => {
  Scaling2["AUTO"] = "AUTO";
  Scaling2["CONTAIN"] = "CONTAIN";
  Scaling2["COVER"] = "COVER";
  return Scaling2;
})(Scaling || {});
var ImagePosition = /* @__PURE__ */ ((ImagePosition2) => {
  ImagePosition2["CENTER"] = "CENTER";
  ImagePosition2["CENTER_LEFT"] = "CENTER_LEFT";
  ImagePosition2["CENTER_RIGHT"] = "CENTER_RIGHT";
  ImagePosition2["TOP"] = "TOP";
  ImagePosition2["TOP_LEFT"] = "TOP_LEFT";
  ImagePosition2["TOP_RIGHT"] = "TOP_RIGHT";
  ImagePosition2["BOTTOM"] = "BOTTOM";
  ImagePosition2["BOTTOM_LEFT"] = "BOTTOM_LEFT";
  ImagePosition2["BOTTOM_RIGHT"] = "BOTTOM_RIGHT";
  return ImagePosition2;
})(ImagePosition || {});
var VerticalAlignmentAlignment = /* @__PURE__ */ ((VerticalAlignmentAlignment2) => {
  VerticalAlignmentAlignment2["TOP"] = "TOP";
  VerticalAlignmentAlignment2["MIDDLE"] = "MIDDLE";
  VerticalAlignmentAlignment2["BOTTOM"] = "BOTTOM";
  return VerticalAlignmentAlignment2;
})(VerticalAlignmentAlignment || {});
var ResponsivenessBehaviour = /* @__PURE__ */ ((ResponsivenessBehaviour2) => {
  ResponsivenessBehaviour2["STACK"] = "STACK";
  ResponsivenessBehaviour2["WRAP"] = "WRAP";
  return ResponsivenessBehaviour2;
})(ResponsivenessBehaviour || {});
var BadgeTypeType = /* @__PURE__ */ ((BadgeTypeType2) => {
  BadgeTypeType2["UNKNOWN_BADGE_TYPE"] = "UNKNOWN_BADGE_TYPE";
  BadgeTypeType2["FIRST_PRIORITY"] = "FIRST_PRIORITY";
  BadgeTypeType2["SECOND_PRIORITY"] = "SECOND_PRIORITY";
  BadgeTypeType2["THIRD_PRIORITY"] = "THIRD_PRIORITY";
  return BadgeTypeType2;
})(BadgeTypeType || {});
var RequestedFields = /* @__PURE__ */ ((RequestedFields2) => {
  RequestedFields2["UNKNOWN_REQUESTED_FIELD"] = "UNKNOWN_REQUESTED_FIELD";
  RequestedFields2["DETAILS"] = "DETAILS";
  RequestedFields2["TEXTS"] = "TEXTS";
  RequestedFields2["REGISTRATION"] = "REGISTRATION";
  RequestedFields2["URLS"] = "URLS";
  RequestedFields2["FORM"] = "FORM";
  RequestedFields2["DASHBOARD"] = "DASHBOARD";
  RequestedFields2["FEED"] = "FEED";
  RequestedFields2["ONLINE_CONFERENCING_SESSION"] = "ONLINE_CONFERENCING_SESSION";
  RequestedFields2["SEO_SETTINGS"] = "SEO_SETTINGS";
  RequestedFields2["AGENDA"] = "AGENDA";
  RequestedFields2["CATEGORIES"] = "CATEGORIES";
  return RequestedFields2;
})(RequestedFields || {});
var CommonSortOrder = /* @__PURE__ */ ((CommonSortOrder2) => {
  CommonSortOrder2["ASC"] = "ASC";
  CommonSortOrder2["DESC"] = "DESC";
  return CommonSortOrder2;
})(CommonSortOrder || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var Relation = /* @__PURE__ */ ((Relation2) => {
  Relation2["ATTENDING"] = "ATTENDING";
  return Relation2;
})(Relation || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createEvent2(event, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(
      { event, draft: options?.draft, fields: options?.fields },
      ["event.description"]
    ),
    [
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [{ path: "event.mainImage" }]
      },
      {
        transformFn: import_page_url.transformSDKPageURLToRESTPageURL,
        paths: [
          { path: "event.eventPageUrl" },
          { path: "event.agendaSettings.pageUrl" }
        ]
      },
      {
        transformFn: import_address.transformSDKAddressToRESTAddress,
        paths: [{ path: "event.location.address" }]
      }
    ]
  );
  const reqOpts = createEvent(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "event.mainImage" }]
        },
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [
            { path: "event.eventPageUrl" },
            { path: "event.agendaSettings.pageUrl" }
          ]
        },
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [{ path: "event.location.address" }]
        }
      ]),
      ["event.description"]
    )?.event;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          event: "$[0]",
          draft: "$[1].draft",
          fields: "$[1].fields"
        },
        singleArgumentUnchanged: false
      },
      ["event", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function cloneEvent2(eventId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(
      {
        eventId,
        event: options?.event,
        draft: options?.draft,
        fields: options?.fields
      },
      ["event.description"]
    ),
    [
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [{ path: "event.mainImage" }]
      },
      {
        transformFn: import_page_url.transformSDKPageURLToRESTPageURL,
        paths: [
          { path: "event.eventPageUrl" },
          { path: "event.agendaSettings.pageUrl" }
        ]
      },
      {
        transformFn: import_address.transformSDKAddressToRESTAddress,
        paths: [{ path: "event.location.address" }]
      }
    ]
  );
  const reqOpts = cloneEvent(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "event.mainImage" }]
        },
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [
            { path: "event.eventPageUrl" },
            { path: "event.agendaSettings.pageUrl" }
          ]
        },
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [{ path: "event.location.address" }]
        }
      ]),
      ["event.description"]
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          eventId: "$[0]",
          event: "$[1].event",
          draft: "$[1].draft",
          fields: "$[1].fields"
        },
        singleArgumentUnchanged: false
      },
      ["eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateEvent2(_id, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(
      { event: { ...options?.event, id: _id }, fields: options?.fields },
      ["event.description"]
    ),
    [
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [{ path: "event.mainImage" }]
      },
      {
        transformFn: import_page_url.transformSDKPageURLToRESTPageURL,
        paths: [
          { path: "event.eventPageUrl" },
          { path: "event.agendaSettings.pageUrl" }
        ]
      },
      {
        transformFn: import_address.transformSDKAddressToRESTAddress,
        paths: [{ path: "event.location.address" }]
      }
    ]
  );
  const reqOpts = updateEvent(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "event.mainImage" }]
        },
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [
            { path: "event.eventPageUrl" },
            { path: "event.agendaSettings.pageUrl" }
          ]
        },
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [{ path: "event.location.address" }]
        }
      ]),
      ["event.description"]
    )?.event;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { event: "$[1].event" },
        explicitPathsToArguments: { "event.id": "$[0]", fields: "$[1].fields" },
        singleArgumentUnchanged: false
      },
      ["_id", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function publishDraftEvent2(eventId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(
    { eventId, fields: options?.fields },
    []
  );
  const reqOpts = publishDraftEvent(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "event.mainImage" }]
        },
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [
            { path: "event.eventPageUrl" },
            { path: "event.agendaSettings.pageUrl" }
          ]
        },
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [{ path: "event.location.address" }]
        }
      ]),
      ["event.description"]
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { eventId: "$[0]", fields: "$[1].fields" },
        singleArgumentUnchanged: false
      },
      ["eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function cancelEvent2(eventId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(
    { eventId, fields: options?.fields },
    []
  );
  const reqOpts = cancelEvent(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "event.mainImage" }]
        },
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [
            { path: "event.eventPageUrl" },
            { path: "event.agendaSettings.pageUrl" }
          ]
        },
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [{ path: "event.location.address" }]
        }
      ]),
      ["event.description"]
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { eventId: "$[0]", fields: "$[1].fields" },
        singleArgumentUnchanged: false
      },
      ["eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCancelEventsByFilter2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(
    { filter: options?.filter },
    []
  );
  const reqOpts = bulkCancelEventsByFilter(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { filter: "$[0].filter" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteEvent2(eventId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(
    { eventId },
    []
  );
  const reqOpts = deleteEvent(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data, []);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { eventId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["eventId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkDeleteEventsByFilter2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(
    { filter: options?.filter },
    []
  );
  const reqOpts = bulkDeleteEventsByFilter(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { filter: "$[0].filter" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryEvents2(options) {
  const { httpClient, sideEffects } = arguments[1];
  return (0, import_query_builder.queryBuilder)(
    {
      func: async (payload) => {
        const reqOpts = queryEvents({
          ...payload,
          ...options ?? {}
        });
        sideEffects?.onSiteCall?.();
        try {
          const result = await httpClient.request(reqOpts);
          sideEffects?.onSuccess?.(result);
          return result;
        } catch (err) {
          sideEffects?.onError?.(err);
          throw err;
        }
      },
      requestTransformer: (query) => {
        const args = [query, options];
        return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(
          { ...args?.[1], query: args?.[0] },
          []
        );
      },
      responseTransformer: ({ data }) => {
        const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
          (0, import_transform_paths2.transformPaths)(data, [
            {
              transformFn: import_image2.transformRESTImageToSDKImage,
              paths: [{ path: "events.mainImage" }]
            },
            {
              transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
              paths: [
                { path: "events.eventPageUrl" },
                { path: "events.agendaSettings.pageUrl" }
              ]
            },
            {
              transformFn: import_address2.transformRESTAddressToSDKAddress,
              paths: [{ path: "events.location.address" }]
            }
          ]),
          ["events.description"]
        );
        return {
          items: transformedData?.events,
          pagingMetadata: transformedData?.pagingMetadata
        };
      },
      errorTransformer: (err) => {
        const transformedError = (0, import_transform_error.transformError)(err, {
          spreadPathsToArguments: {},
          explicitPathsToArguments: { query: "$[0]" },
          singleArgumentUnchanged: false
        });
        throw transformedError;
      },
      pagingMethod: "OFFSET",
      transformationPaths: {}
    }
  );
}
async function countEventsByStatus2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(
    {
      filter: options?.filter,
      facet: options?.facet,
      includeDrafts: options?.includeDrafts
    },
    []
  );
  const reqOpts = countEventsByStatus(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data, []);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          filter: "$[0].filter",
          facet: "$[0].facet",
          includeDrafts: "$[0].includeDrafts"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listEventsByCategory2(categoryId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(
    {
      categoryId,
      paging: options?.paging,
      fields: options?.fields
    },
    []
  );
  const reqOpts = listEventsByCategory(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "events.mainImage" }]
        },
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [
            { path: "events.eventPageUrl" },
            { path: "events.agendaSettings.pageUrl" }
          ]
        },
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [{ path: "events.location.address" }]
        }
      ]),
      ["events.description"]
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          categoryId: "$[0]",
          paging: "$[1].paging",
          fields: "$[1].fields"
        },
        singleArgumentUnchanged: false
      },
      ["categoryId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getEvent2(eventId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(
    { eventId, fields: options?.fields },
    []
  );
  const reqOpts = getEvent(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "event.mainImage" }]
        },
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [
            { path: "event.eventPageUrl" },
            { path: "event.agendaSettings.pageUrl" }
          ]
        },
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [{ path: "event.location.address" }]
        }
      ]),
      ["event.description"]
    )?.event;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { eventId: "$[0]", fields: "$[1].fields" },
        singleArgumentUnchanged: false
      },
      ["eventId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getEventBySlug2(slug, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(
    { slug, fields: options?.fields },
    []
  );
  const reqOpts = getEventBySlug(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "event.mainImage" }]
        },
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [
            { path: "event.eventPageUrl" },
            { path: "event.agendaSettings.pageUrl" }
          ]
        },
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [{ path: "event.location.address" }]
        }
      ]),
      ["event.description"]
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { slug: "$[0]", fields: "$[1].fields" },
        singleArgumentUnchanged: false
      },
      ["slug", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/events-v3-event-wix-events-v-2.public.ts
function createEvent3(httpClient) {
  return (event, options) => createEvent2(
    event,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function cloneEvent3(httpClient) {
  return (eventId, options) => cloneEvent2(
    eventId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function updateEvent3(httpClient) {
  return (_id, options) => updateEvent2(
    _id,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function publishDraftEvent3(httpClient) {
  return (eventId, options) => publishDraftEvent2(
    eventId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function cancelEvent3(httpClient) {
  return (eventId, options) => cancelEvent2(
    eventId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkCancelEventsByFilter3(httpClient) {
  return (options) => bulkCancelEventsByFilter2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function deleteEvent3(httpClient) {
  return (eventId) => deleteEvent2(
    eventId,
    // @ts-ignore
    { httpClient }
  );
}
function bulkDeleteEventsByFilter3(httpClient) {
  return (options) => bulkDeleteEventsByFilter2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function queryEvents3(httpClient) {
  return (options) => queryEvents2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function countEventsByStatus3(httpClient) {
  return (options) => countEventsByStatus2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function listEventsByCategory3(httpClient) {
  return (categoryId, options) => listEventsByCategory2(
    categoryId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function getEvent3(httpClient) {
  return (eventId, options) => getEvent2(
    eventId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function getEventBySlug3(httpClient) {
  return (slug, options) => getEventBySlug2(
    slug,
    options,
    // @ts-ignore
    { httpClient }
  );
}
var onEventCanceled = (0, import_sdk_types.EventDefinition)(
  "wix.events.v3.event_canceled",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [{ path: "data.timestamp" }, { path: "metadata.eventTime" }]
      }
    ])
  )
)();
var onEventCloned = (0, import_sdk_types.EventDefinition)(
  "wix.events.v3.event_cloned",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "data.timestamp" },
          { path: "data.dateAndTimeSettings.startDate" },
          { path: "data.dateAndTimeSettings.endDate" },
          {
            path: "data.dateAndTimeSettings.recurringEvents.individualEventDates.startDate"
          },
          {
            path: "data.dateAndTimeSettings.recurringEvents.individualEventDates.endDate"
          },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_address3.transformRESTAddressToSDKAddress,
        paths: [{ path: "data.location.address" }]
      }
    ])
  )
)();
var onEventCreated = (0, import_sdk_types.EventDefinition)(
  "wix.events.v3.event_created",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_image3.transformRESTImageToSDKImage,
        paths: [{ path: "entity.mainImage" }]
      },
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.publishedDate" },
          { path: "entity.dateAndTimeSettings.startDate" },
          { path: "entity.dateAndTimeSettings.endDate" },
          {
            path: "entity.dateAndTimeSettings.recurringEvents.individualEventDates.startDate"
          },
          {
            path: "entity.dateAndTimeSettings.recurringEvents.individualEventDates.endDate"
          },
          { path: "entity.registration.rsvp.startDate" },
          { path: "entity.registration.rsvp.endDate" },
          { path: "entity.categories.categories.createdDate" },
          { path: "entity.description.metadata.createdTimestamp" },
          { path: "entity.description.metadata.updatedTimestamp" },
          { path: "entity.badges.badges.functionUpdateDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_page_url3.transformRESTPageURLToSDKPageURL,
        paths: [
          { path: "entity.eventPageUrl" },
          { path: "entity.agendaSettings.pageUrl" }
        ]
      },
      {
        transformFn: import_address3.transformRESTAddressToSDKAddress,
        paths: [{ path: "entity.location.address" }]
      },
      {
        transformFn: import_float3.transformRESTFloatToSDKFloat,
        paths: [
          {
            path: "entity.description.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "entity.description.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "entity.description.nodes.galleryData.items.video.thumbnail.duration"
          },
          { path: "entity.description.nodes.galleryData.options.item.ratio" },
          { path: "entity.description.nodes.imageData.image.duration" },
          { path: "entity.description.nodes.mapData.mapSettings.lat" },
          { path: "entity.description.nodes.mapData.mapSettings.lng" },
          { path: "entity.description.nodes.pollData.poll.image.duration" },
          {
            path: "entity.description.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "entity.description.nodes.pollData.design.poll.background.image.duration"
          },
          { path: "entity.description.nodes.appEmbedData.image.duration" },
          { path: "entity.description.nodes.videoData.video.duration" },
          { path: "entity.description.nodes.videoData.thumbnail.duration" },
          { path: "entity.description.nodes.audioData.audio.duration" },
          { path: "entity.description.nodes.audioData.coverImage.duration" },
          {
            path: "entity.description.nodes.layoutData.backgroundImage.media.duration"
          },
          {
            path: "entity.description.nodes.layoutData.backdropImage.media.duration"
          }
        ]
      }
    ])
  )
)();
var onEventDeleted = (0, import_sdk_types.EventDefinition)(
  "wix.events.v3.event_deleted",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_image3.transformRESTImageToSDKImage,
        paths: [{ path: "undefined.mainImage" }]
      },
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "undefined.createdDate" },
          { path: "undefined.updatedDate" },
          { path: "undefined.publishedDate" },
          { path: "undefined.dateAndTimeSettings.startDate" },
          { path: "undefined.dateAndTimeSettings.endDate" },
          {
            path: "undefined.dateAndTimeSettings.recurringEvents.individualEventDates.startDate"
          },
          {
            path: "undefined.dateAndTimeSettings.recurringEvents.individualEventDates.endDate"
          },
          { path: "undefined.registration.rsvp.startDate" },
          { path: "undefined.registration.rsvp.endDate" },
          { path: "undefined.categories.categories.createdDate" },
          { path: "undefined.description.metadata.createdTimestamp" },
          { path: "undefined.description.metadata.updatedTimestamp" },
          { path: "undefined.badges.badges.functionUpdateDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_page_url3.transformRESTPageURLToSDKPageURL,
        paths: [
          { path: "undefined.eventPageUrl" },
          { path: "undefined.agendaSettings.pageUrl" }
        ]
      },
      {
        transformFn: import_address3.transformRESTAddressToSDKAddress,
        paths: [{ path: "undefined.location.address" }]
      },
      {
        transformFn: import_float3.transformRESTFloatToSDKFloat,
        paths: [
          {
            path: "undefined.description.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "undefined.description.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "undefined.description.nodes.galleryData.items.video.thumbnail.duration"
          },
          {
            path: "undefined.description.nodes.galleryData.options.item.ratio"
          },
          { path: "undefined.description.nodes.imageData.image.duration" },
          { path: "undefined.description.nodes.mapData.mapSettings.lat" },
          { path: "undefined.description.nodes.mapData.mapSettings.lng" },
          {
            path: "undefined.description.nodes.pollData.poll.image.duration"
          },
          {
            path: "undefined.description.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "undefined.description.nodes.pollData.design.poll.background.image.duration"
          },
          { path: "undefined.description.nodes.appEmbedData.image.duration" },
          { path: "undefined.description.nodes.videoData.video.duration" },
          {
            path: "undefined.description.nodes.videoData.thumbnail.duration"
          },
          { path: "undefined.description.nodes.audioData.audio.duration" },
          {
            path: "undefined.description.nodes.audioData.coverImage.duration"
          },
          {
            path: "undefined.description.nodes.layoutData.backgroundImage.media.duration"
          },
          {
            path: "undefined.description.nodes.layoutData.backdropImage.media.duration"
          }
        ]
      }
    ])
  )
)();
var onEventEnded = (0, import_sdk_types.EventDefinition)(
  "wix.events.v3.event_ended",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [{ path: "data.timestamp" }, { path: "metadata.eventTime" }]
      }
    ])
  )
)();
var onEventPublished = (0, import_sdk_types.EventDefinition)(
  "wix.events.v3.event_published",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "data.timestamp" },
          { path: "data.dateAndTimeSettings.startDate" },
          { path: "data.dateAndTimeSettings.endDate" },
          {
            path: "data.dateAndTimeSettings.recurringEvents.individualEventDates.startDate"
          },
          {
            path: "data.dateAndTimeSettings.recurringEvents.individualEventDates.endDate"
          },
          { path: "data.registration.rsvp.startDate" },
          { path: "data.registration.rsvp.endDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_page_url3.transformRESTPageURLToSDKPageURL,
        paths: [{ path: "data.eventPageUrl" }]
      },
      {
        transformFn: import_address3.transformRESTAddressToSDKAddress,
        paths: [{ path: "data.location.address" }]
      }
    ])
  )
)();
var onEventReminder = (0, import_sdk_types.EventDefinition)(
  "wix.events.v3.event_reminder",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "data.timestamp" },
          { path: "data.dateAndTimeSettings.startDate" },
          { path: "data.dateAndTimeSettings.endDate" },
          {
            path: "data.dateAndTimeSettings.recurringEvents.individualEventDates.startDate"
          },
          {
            path: "data.dateAndTimeSettings.recurringEvents.individualEventDates.endDate"
          },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_address3.transformRESTAddressToSDKAddress,
        paths: [{ path: "data.location.address" }]
      }
    ])
  )
)();
var onEventStarted = (0, import_sdk_types.EventDefinition)(
  "wix.events.v3.event_started",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [{ path: "data.timestamp" }, { path: "metadata.eventTime" }]
      }
    ])
  )
)();
var onEventUpdated = (0, import_sdk_types.EventDefinition)(
  "wix.events.v3.event_updated",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_image3.transformRESTImageToSDKImage,
        paths: [{ path: "entity.mainImage" }]
      },
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.publishedDate" },
          { path: "entity.dateAndTimeSettings.startDate" },
          { path: "entity.dateAndTimeSettings.endDate" },
          {
            path: "entity.dateAndTimeSettings.recurringEvents.individualEventDates.startDate"
          },
          {
            path: "entity.dateAndTimeSettings.recurringEvents.individualEventDates.endDate"
          },
          { path: "entity.registration.rsvp.startDate" },
          { path: "entity.registration.rsvp.endDate" },
          { path: "entity.categories.categories.createdDate" },
          { path: "entity.description.metadata.createdTimestamp" },
          { path: "entity.description.metadata.updatedTimestamp" },
          { path: "entity.badges.badges.functionUpdateDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_page_url3.transformRESTPageURLToSDKPageURL,
        paths: [
          { path: "entity.eventPageUrl" },
          { path: "entity.agendaSettings.pageUrl" }
        ]
      },
      {
        transformFn: import_address3.transformRESTAddressToSDKAddress,
        paths: [{ path: "entity.location.address" }]
      },
      {
        transformFn: import_float3.transformRESTFloatToSDKFloat,
        paths: [
          {
            path: "entity.description.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "entity.description.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "entity.description.nodes.galleryData.items.video.thumbnail.duration"
          },
          { path: "entity.description.nodes.galleryData.options.item.ratio" },
          { path: "entity.description.nodes.imageData.image.duration" },
          { path: "entity.description.nodes.mapData.mapSettings.lat" },
          { path: "entity.description.nodes.mapData.mapSettings.lng" },
          { path: "entity.description.nodes.pollData.poll.image.duration" },
          {
            path: "entity.description.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "entity.description.nodes.pollData.design.poll.background.image.duration"
          },
          { path: "entity.description.nodes.appEmbedData.image.duration" },
          { path: "entity.description.nodes.videoData.video.duration" },
          { path: "entity.description.nodes.videoData.thumbnail.duration" },
          { path: "entity.description.nodes.audioData.audio.duration" },
          { path: "entity.description.nodes.audioData.coverImage.duration" },
          {
            path: "entity.description.nodes.layoutData.backgroundImage.media.duration"
          },
          {
            path: "entity.description.nodes.layoutData.backdropImage.media.duration"
          }
        ]
      }
    ])
  )
)();

// src/events-v3-event-wix-events-v-2.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var import_event_definition_modules = require("@wix/sdk-runtime/event-definition-modules");
var createEvent4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(createEvent3);
var cloneEvent4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(cloneEvent3);
var updateEvent4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(updateEvent3);
var publishDraftEvent4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(publishDraftEvent3);
var cancelEvent4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(cancelEvent3);
var bulkCancelEventsByFilter4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkCancelEventsByFilter3);
var deleteEvent4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(deleteEvent3);
var bulkDeleteEventsByFilter4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkDeleteEventsByFilter3);
var queryEvents4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(queryEvents3);
var countEventsByStatus4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(countEventsByStatus3);
var listEventsByCategory4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(listEventsByCategory3);
var getEvent4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getEvent3);
var getEventBySlug4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getEventBySlug3);
var onEventCanceled2 = (0, import_event_definition_modules.createEventModule)(onEventCanceled);
var onEventCloned2 = (0, import_event_definition_modules.createEventModule)(onEventCloned);
var onEventCreated2 = (0, import_event_definition_modules.createEventModule)(onEventCreated);
var onEventDeleted2 = (0, import_event_definition_modules.createEventModule)(onEventDeleted);
var onEventEnded2 = (0, import_event_definition_modules.createEventModule)(onEventEnded);
var onEventPublished2 = (0, import_event_definition_modules.createEventModule)(onEventPublished);
var onEventReminder2 = (0, import_event_definition_modules.createEventModule)(onEventReminder);
var onEventStarted2 = (0, import_event_definition_modules.createEventModule)(onEventStarted);
var onEventUpdated2 = (0, import_event_definition_modules.createEventModule)(onEventUpdated);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Alignment,
  AppType,
  AspectRatio,
  BackgroundType,
  BadgeTypeType,
  ButtonDataType,
  CardStylesType,
  CheckoutType,
  CommonSortOrder,
  ConferenceTypeType,
  Crop,
  DecorationType,
  Direction,
  DividerDataAlignment,
  FontType,
  GIFType,
  GuestTypeType,
  ImagePosition,
  InitialExpandedItems,
  InitialRegistrationTypeType,
  InputControlType,
  Layout,
  LayoutType,
  LineStyle,
  LocationType,
  MapType,
  NodeType,
  NullValue,
  Orientation,
  Placement,
  PluginContainerDataAlignment,
  PollLayoutDirection,
  PollLayoutType,
  Position,
  RecurrenceStatusStatus,
  RegistrationStatusStatus,
  RegistrationTypeType,
  Relation,
  RequestedFields,
  Resizing,
  ResponseType,
  ResponsivenessBehaviour,
  Scaling,
  SortOrder,
  Source,
  Status,
  StylesPosition,
  SubdivisionType,
  Target,
  TaxType,
  TextAlignment,
  ThumbnailsAlignment,
  Type,
  ValueType,
  VerticalAlignment,
  VerticalAlignmentAlignment,
  ViewMode,
  ViewRole,
  VoteRole,
  WebhookIdentityType,
  Width,
  WidthType,
  bulkCancelEventsByFilter,
  bulkDeleteEventsByFilter,
  cancelEvent,
  cloneEvent,
  countEventsByStatus,
  createEvent,
  deleteEvent,
  getEvent,
  getEventBySlug,
  listEventsByCategory,
  onEventCanceled,
  onEventCloned,
  onEventCreated,
  onEventDeleted,
  onEventEnded,
  onEventPublished,
  onEventReminder,
  onEventStarted,
  onEventUpdated,
  publishDraftEvent,
  queryEvents,
  updateEvent
});
